<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>rate4site - detector of conserved amino-acid sites</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/rate4site">rate4site_3.0.0-8_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       rate4site - detector of conserved amino-acid sites

</pre><h4><b>SYNOPSIS</b></h4><pre>
       rate4site [OPTIONS] -s &lt;MSA FILE&gt;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The rate of evolution is not constant among amino acid sites: some positions evolve slowly and are
       commonly referred to as "conserved", while others evolve rapidly and are referred to as "variable". The
       rate variations correspond to different levels of purifying selection acting on these sites. The
       purifying selection can be the result of geometrical constraints on the folding of the protein into its
       3D structure, constraints at amino acid sites involved in enzymatic activity or in ligand binding or,
       alternatively, at amino acid sites that take part in protein-protein interactions. Rate4Site calculates
       the relative evolutionary rate at each site using a probabilistic-based evolutionary model. This allows
       taking into account the stochastic process underlying sequence evolution within protein families and the
       phylogenetic tree of the proteins in the family. The conservation score at a site corresponds to the
       site's evolutionary rate.

</pre><h4><b>METHODOLOGY</b></h4><pre>
       The sole obligatory input to Rate4Site is an MSA file. The program then computes a phylogenetic tree that
       is consistent with the available MSA (the user can also input a pre-calculated tree). It then calculates
       the relative conservation score for each site in the MSA. This is carried out using either an empirical
       Bayesian method or a maximum likelihood method (Pupko et al., 2002). The differences between the two
       methods are explained in details in Mayrose et al (2004).

</pre><h4><b>REFERENCES</b></h4><pre>
       Mayrose, I., Graur, D., Ben-Tal, N., and Pupko, T. 2004. Comparison of site-specific rate-inference
       methods: Bayesian methods are superior. Mol Biol Evol 21: 1781-1791.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-s</b> <b>MSA_FILE</b>
           The  input  sequence  file  name. The following formats are supported: Mase, Molphy, Phylip, Clustal,
           Fasta

       -t  The input tree file name (in Newick format)

       <b>-o</b> <b>OUTPUT_FILE</b>
           The results output file

       -a  Reference sequence name in the MSA. The conservation scores are printed based on the  amino-acids  in
           this sequence.

       -k  The number of discrete Gamma categories

       -m  Evolutionary model. The following amino-acids models are supported:

                   DAY (-md), JTT (-mj), REV (-mr), aaJC (-ma), LG (-Ml), WAG (-Mw) .

                   For nucleotides, the following models are supported: JC (-mn), HKY (-Mh), Tamura92 (-Mt), GTR (-Mg).

       -b  Branch lengths optimization flag:

                   -bn = no Branch lengths optimization

                   -bh = optimization using a homogeneous model (no among-site-rate-variation)

                   -bg = optimization using a Gamma model

       -i  Rate inference method flag:

                   -Im = rates are inferred using the maximum likelihood method

                   -Ib = rates are inferred using the empirical Bayes method

       -z  Tree constructing method

                   zj = Neighbor-joining tree with Jukes-Cantor distances

                   zn = Neighbor-joining tree with maximum likelihood distances

       -h  Short help message

</pre><h4><b>AUTHOR</b></h4><pre>
       Nir Ben-Tal &lt;<a href="mailto:NirB@tauex.tau.ac.il">NirB@tauex.tau.ac.il</a>&gt;

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Main website
           &lt;<a href="http://www.tau.ac.il/~itaymay/cp/rate4site.html">http://www.tau.ac.il/~itaymay/cp/rate4site.html</a>&gt;

3.0.0                                              2022-01-20                                       <u><a href="../man1/RATE4SITE.1.html">RATE4SITE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>