<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>snp-sites - finds SNP sites from a multi FASTA alignment file</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/snp-sites">snp-sites_2.5.1-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       snp-sites - finds SNP sites from a multi FASTA alignment file

</pre><h4><b>SYNOPSIS</b></h4><pre>
       snp-sites [-mvph] [-o output_filename] [input file]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This application takes in a multi fasta alignment, finds all the SNP sites, then outputs the SNP sites in
       either of the following formats:

       •   multi fasta alignment

       •   VCF

       •   relaxed phylip format

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-r</b>
           Output internal pseudo reference sequence

       <b>-m</b>
           Output a multi fasta alignment file (default)

       <b>-v</b>
           Output a VCF file

       <b>-p</b>
           Output a phylip file

       <b>-c</b>
           Only output columns containing exclusively ACGT

       <b>-b</b>
           Output monomorphic sites, useful for BEAST

       <b>-o</b>
           Specify an output filename

       <b>-h</b>
           Show help message

       <b>-V</b>
           Show version and exit

</pre><h4><b>EXAMPLES</b></h4><pre>
           snp-sites my-alignment.aln

           snp-sites my-gzipped-alignment.aln.gz

           snp-sites -cb -o output_beast_format.aln inputfile.aln

</pre><h4><b>FORMAT</b> <b>OF</b> <b>THE</b> <b>INPUT</b> <b>FILE</b></h4><pre>
       Input files should look like this:

           &gt;reference_sequence
           aaccggtt
           &gt;comparison_sequence
           AACCGGTT
           &gt;another_comparison_sequence
           AACCGCTT

       For more examples, visit: https://github.com/sanger-pathogens/snp-sites/tree/master/tests/data

</pre><h4><b>AUTHOR</b></h4><pre>
       snp-sites was originally written by Andrew Page (<a href="mailto:ap13@sanger.ac.uk">ap13@sanger.ac.uk</a>)

</pre><h4><b>CITATION</b></h4><pre>
       If you use this program, please cite: "SNP-sites: rapid efficient extraction of SNPs from multi-FASTA
       alignments", Andrew J. Page, Ben Taylor, Aidan J. Delaney, Jorge Soares, Torsten Seemann, Jacqueline A.
       Keane, Simon R. Harris (2016), bioRxiv doi: <a href="http://dx.doi.org/10.1101/038190">http://dx.doi.org/10.1101/038190</a>

</pre><h4><b>COPYING</b></h4><pre>
       Wellcome Trust Sanger Institute Copyright © 2013 Wellcome Trust Sanger Institute This program is free
       software; you can redistribute it and/or modify it under the terms of the GNU General Public License as
       published by the Free Software Foundation; either version 3 of the License, or (at your option) any later
       version.

                                                                                                    <u><a href="../man1/SNP-SITES.1.html">SNP-SITES</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>