<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>dawg - DNA assembly with gaps, a DNA sequence simulator</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/dawg">dawg_1.2-4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       dawg - DNA assembly with gaps, a DNA sequence simulator

</pre><h4><b>DESCRIPTION</b></h4><pre>
       dawg 1.2-release DNA Assembly With Gaps Copyright Â© 2004-2009 Reed A. Cartwright

       dawg <b>-[scubvhqew</b>?] [-o outputfile] file1 [file2...]

       <b>-s</b>: process files serially [default]

       <b>-c</b>: process files combined together

       <b>-u</b>: unbuffered output

       <b>-b</b>: buffered output [default]

       <b>-q</b>: disable error and warning reports (quiet)

       <b>-e</b>: enable error reports [default]

       <b>-w</b>: enable warning reports [default]

       <b>-v</b>: display version information

       <b>-h</b>: display help information

       -?: same as <b>-h</b>

       <b>-o</b> outputfile: override output filename in the configuration file

              Dawg will read stdin if filename is "-".

       FILE FORMAT

              The  file  format  takes  a  series  of  statements in the form of "name = value," where "name" is
              alphanumeric and value can be a string, number, boolean, tree, or  vector  of  values.   A  single
              variable is equivalent to a vector of a single entry.

       string:  "[char-sequence]"

              '[char-sequence]'  """[multi-line char-sequence]""" (rm initial and final newlines) '''[multi-line
              char-sequence]''' (kp initial and final newlines)

       number:  [sign]digits[.digits][(e|E)[sign]digits] boolean: true|false tree:    Newick Format  vector:   {
       value, value, ...}

       OPTIONS

       Name   Type            Description

       <b>---------------------------------</b>

       Tree   VT  phylogeny

       TreeScale
              N   coefficient to scale branch lengths by

       Sequence
              VS  root sequences

       Length VN  length of generated root sequences

       Rates  VVN rate of evolution of each root nucleotide

       Model  S   model of evolution: GTR|JC|K2P|K3P|HKY|F81|F84|TN

       Freqs  VN  nucleotide (ACGT) frequencies

       Params VN  parameters for the model of evolution

       Width  N   block width for indels and recombination

       Scale  VN  block position scales

       Gamma  VN  coefficients of variance for rate heterogenity

       Alpha  VN  shape parameters

       Iota   VN  proportions of invariant sites

       GapModel
              VS  models of indel formation: NB|PL|US

       Lambda VN  rates of indel formation

       GapParams
              VVN parameter for the indel model

       Reps   N   number of data sets to output

       File   S   output file

       Format S   output format: Fasta|Nexus|Phylip|Clustal

       GapSingleChar
              B   output gaps as a single character

       GapPlus
              B   distinguish insertions from deletions in alignment

       KeepFlank
              N   undeletable flanking regions N nucs from sequence

       KeepEmpty
              B   preserve empty columns in final alignment

       LowerCase
              B   output sequences in lowercase

       Translate
              B   translate outputed sequences to amino acids

       Seed   VN  pseudo-random-number-generator seed (integers)

       Out.Block.Head
              S   string to insert at the start of the output

       Out.Block.Tail
              S   string to insert at the end of the output

       Out.Block.Before S
              string to insert before a sequence set in the output

       Out.Block.After
              S   string to insert after a sequence set in the output

       Out.Subst
              B   do variable subsitution in Out.Block.*

</pre><h4><b>LICENSE</b></h4><pre>
       This  program  is  free  software;  you  can  redistribute it and/or modify it under the terms of the GNU
       General Public License as published by the Free Software Foundation; either version 2 of the License,  or
       (at your option) any later version.

       This  program  is  distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
       the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General  Public
       License for more details.

       You  should have received a copy of the GNU General Public License along with this program; if not, write
       to the Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA

       Send Bug Reports to <a href="mailto:reed@scit.us">reed@scit.us</a>.

dawg 1.2-release                                  December 2015                                          <u><a href="../man1/DAWG.1.html">DAWG</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>