<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>skesa - strategic Kmer extension for scrupulous assemblies</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/skesa">skesa_2.4.0-6build2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       skesa - strategic Kmer extension for scrupulous assemblies

</pre><h4><b>DESCRIPTION</b></h4><pre>
       SKESA  is  a  DeBruijn  graph-based  de-novo assembler designed for assembling reads of microbial genomes
       sequenced using Illumina.  Comparison with SPAdes and MegaHit shows that SKESA produces  assemblies  that
       have  high  sequence  quality  and  contiguity,  handles  low-level  contamination in reads, is fast, and
       produces an identical assembly for the same  input  when  assembled  multiple  times  with  the  same  or
       different  compute  resources.  SKESA has been used for assembling over 272,000 read sets in the Sequence
       Read Archive at NCBI and for real-time pathogen detection.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>General</b> <b>options:</b>
       <b>-h</b> [ <b>--help</b> ]
              Produce help message

       <b>-v</b> [ <b>--version</b> ]
              Print version

       <b>--cores</b> arg (=0)
              Number of cores to use (default all) [integer]

       <b>--memory</b> arg (=32)
              Memory available (GB, only for sorted counter) [integer]

       <b>--hash_count</b>
              Use hash counter [flag]

       <b>--estimated_kmers</b> arg (=100)
              Estimated number of unique kmers for bloom filter (M, only for hash counter) [integer]

       <b>--skip_bloom_filter</b>
              Don't do bloom filter; use <b>--estimated_kmers</b> as the hash table size (only for hash counter) [flag]

   <b>Input/output</b> <b>options</b> <b>:</b> <b>at</b> <b>least</b> <b>one</b> <b>input</b> <b>providing</b> <b>reads</b> <b>for</b> <b>assembly</b> <b>must</b> <b>be</b> <b>specified:</b>
       <b>--fasta</b> arg
              Input fasta file(s) (could be used multiple times for different runs) [string]

       <b>--fastq</b> arg
              Input fastq file(s) (could be used multiple times for different runs) [string]

       <b>--use_paired_ends</b>
              Indicates that a single (not comma separated) fasta/fastq file contains paired reads [flag]

       <b>--sra_run</b> arg
              Input sra run accession (could be used multiple times for different runs) [string]

       <b>--contigs_out</b> arg
              Output file for contigs (stdout if not specified) [string]

   <b>Assembly</b> <b>options:</b>
       <b>--kmer</b> arg (=21)
              Minimal kmer length for assembly [integer]

       <b>--min_count</b> arg
              Minimal count for kmers retained for comparing alternate choices [integer]

       <b>--max_kmer_count</b> arg
              Minimum acceptable average count for estimating the maximal kmer length in reads [integer]

       <b>--vector_percent</b> arg (=0.05)
              Count for  vectors as a fraction of the read number (1. disables) [float (0,1]]

       <b>--insert_size</b> arg
              Expected insert size for paired reads (if not provided, it will be estimated) [integer]

       <b>--steps</b> arg (=11)
              Number of assembly iterations from minimal to maximal kmer length in reads [integer]

       <b>--fraction</b> arg (=0.1)
              Maximum noise to signal ratio acceptable for extension [float [0,1)]

       <b>--max_snp_len</b> arg (=150)
              Maximal snp length [integer]

       <b>--min_contig</b> arg (=200)
              Minimal contig length reported in output [integer]

       <b>--allow_snps</b>
              Allow additional step for snp discovery [flag]

   <b>Debugging</b> <b>options:</b>
       <b>--force_single_ends</b>
              Don't use paired-end information [flag]

       <b>--seeds</b> arg
              Input file with seeds [string]

       <b>--all</b> arg
              Output fasta for each iteration [string]

       <b>--dbg_out</b> arg
              Output kmer file [string]

       <b>--hist</b> arg
              File for histogram [string]

       <b>--connected_reads</b> arg
              File for connected paired reads [string]

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

skesa 2.3.0                                         July 2019                                           <u><a href="../man1/SKESA.1.html">SKESA</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>