<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>grinder - Versatile omics shotgun and amplicon sequencing read simulator</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/grinder">grinder_0.5.4-6_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       grinder - Versatile omics shotgun and amplicon sequencing read simulator

</pre><h4><b>DESCRIPTION</b></h4><pre>
   <b>Usage:</b>
              grinder   <b>-rf</b>  &lt;reference_file&gt;  |  <b>-reference_file</b>  &lt;reference_file&gt;  |  <b>-gf</b>  &lt;reference_file&gt;  |
              <b>-genome_file</b> &lt;reference_file&gt; [cli  optional  arguments]  grinder  <b>--help</b>  grinder  <b>--man</b>  grinder
              <b>--usage</b> grinder <b>--version</b>

   <b>Cli</b> <b>required</b> <b>arguments:</b>

       <b>-rf</b> &lt;reference_file&gt; | <b>-reference_file</b> &lt;reference_file&gt; | <b>-gf</b>

              &lt;reference_file&gt; | <b>-genome_file</b> &lt;reference_file&gt;

              FASTA file that contains the input reference sequences (full genomes, 16S rRNA genes, transcripts,
              proteins...)  or  '-'  to  read  them from the standard input. See the README file for examples of
              databases you can use and where to get them from. Default: -

   <b>Cli</b> <b>optional</b> <b>arguments:</b>

       <b>-tr</b> &lt;total_reads&gt; | <b>-total_reads</b> &lt;total_reads&gt;

              Number of shotgun or amplicon reads to generate for each library.  Do  not  specify  this  if  you
              specify the fold coverage. Default: 100

       <b>-cf</b> &lt;coverage_fold&gt; | <b>-coverage_fold</b> &lt;coverage_fold&gt;

              Desired  fold  coverage  of  the input reference sequences (the output FASTA length divided by the
              input FASTA length). Do not specify this if you specify the number of reads directly.

       <b>-rd</b> &lt;read_dist&gt;... | <b>-read_dist</b> &lt;read_dist&gt;...

              Desired shotgun or amplicon read length distribution specified as:  average  length,  distribution
              ('uniform' or 'normal') and standard deviation.

              Only the first element is required. Examples:

              All  reads  exactly 101 bp long (Illumina GA 2x): 101 Uniform read distribution around 100+-10 bp:
              100 uniform 10 Reads normally distributed with an average of 800 and a standard deviation of 100

              bp (Sanger reads): 800 normal 100

              Reads normally distributed with an average of 450 and a standard deviation of 50

              bp (454 GS-FLX Ti): 450 normal 50

              Reference sequences smaller than the specified read length are not used. Default: 100

       <b>-id</b> &lt;insert_dist&gt;... | <b>-insert_dist</b> &lt;insert_dist&gt;...

              Create paired-end or mate-pair reads spanning the given insert length. Important:  the  insert  is
              defined  in  the  biological  sense,  i.e. its length includes the length of both reads and of the
              stretch of DNA between them: 0 : off, or: insert size distribution in bp, in the  same  format  as
              the  read  length distribution (a typical value is 2,500 bp for mate pairs) Two distinct reads are
              generated whether or not the mate pair overlaps. Default: 0

       <b>-mo</b> &lt;mate_orientation&gt; | <b>-mate_orientation</b> &lt;mate_orientation&gt;

              When generating paired-end or mate-pair reads (see &lt;insert_dist&gt;), specify the orientation of  the
              reads (F: forward, R: reverse):

       FR:    <b>---</b>&gt; &lt;---  e.g. Sanger, Illumina paired-end, IonTorrent mate-pair

       FF:    <b>---</b>&gt; <b>---</b>&gt;  e.g. 454

       RF:    &lt;--- <b>---</b>&gt;  e.g. Illumina mate-pair

       RR:    &lt;--- &lt;---

              Default: FR

       <b>-ec</b> &lt;exclude_chars&gt; | <b>-exclude_chars</b> &lt;exclude_chars&gt;

              Do  not  create  reads containing any of the specified characters (case insensitive). For example,
              use 'NX' to prevent reads with ambiguities (N or X). Grinder will error if  it  fails  to  find  a
              suitable  read  (or  pair of reads) after 10 attempts. Consider using &lt;delete_chars&gt;, which may be
              more appropriate for your case.  Default: ''

       <b>-dc</b> &lt;delete_chars&gt; | <b>-delete_chars</b> &lt;delete_chars&gt;

              Remove the specified characters from the reference sequences  (case-insensitive),  e.g.  '-~*'  to
              remove  gaps  (- or ~) or terminator (*). Removing these characters is done once, when reading the
              reference sequences, prior to taking reads. Hence  it  is  more  efficient  than  &lt;exclude_chars&gt;.
              Default:

       <b>-fr</b> &lt;forward_reverse&gt; | <b>-forward_reverse</b> &lt;forward_reverse&gt;

              Use  DNA  amplicon  sequencing using a forward and reverse PCR primer sequence provided in a FASTA
              file. The reference sequences and their  reverse  complement  will  be  searched  for  PCR  primer
              matches.  The  primer  sequences  should  use the IUPAC convention for degenerate residues and the
              reference sequences that that do not match the specified primers are excluded. If  your  reference
              sequences  are  full  genomes,  it  is recommended to use &lt;copy_bias&gt; = 1 and &lt;length_bias&gt; = 0 to
              generate amplicon reads. To sequence from the forward strand, set &lt;unidirectional&gt; to  1  and  put
              the forward primer first and reverse primer second in the FASTA file. To sequence from the reverse
              strand,  invert  the  primers  in  the FASTA file and use &lt;unidirectional&gt; = <b>-1</b>. The second primer
              sequence in the FASTA file is always optional. Example: AAACTYAAAKGAATTGRCGG  and  ACGGGCGGTGTGTRC
              for the 926F and 1392R primers that target the V6 to V9 region of the 16S rRNA gene.

       <b>-un</b> &lt;unidirectional&gt; | <b>-unidirectional</b> &lt;unidirectional&gt;

              Instead  of producing reads bidirectionally, from the reference strand and its reverse complement,
              proceed unidirectionally, from one  strand  only  (forward  or  reverse).  Values:  0  (off,  i.e.
              bidirectional),   1   (forward),   <b>-1</b>  (reverse).  Use  &lt;unidirectional&gt;  =  1  for  amplicon  and
              strand-specific transcriptomic or proteomic datasets. Default: 0

       <b>-lb</b> &lt;length_bias&gt; | <b>-length_bias</b> &lt;length_bias&gt;

              In shotgun libraries, sample reference sequences proportionally to their length. For  example,  in
              simulated  microbial  datasets,  this  means  that  at the same relative abundance, larger genomes
              contribute more reads than smaller genomes (and all genomes have the same fold coverage). 0 =  no,
              1 = yes. Default: 1

       <b>-cb</b> &lt;copy_bias&gt; | <b>-copy_bias</b> &lt;copy_bias&gt;

              In  amplicon  libraries where full genomes are used as input, sample species proportionally to the
              number of copies of the target gene: at equal  relative  abundance,  genomes  that  have  multiple
              copies of the target gene contribute more amplicon reads than genomes that have a single copy. 0 =
              no, 1 = yes. Default: 1

       <b>-md</b> &lt;mutation_dist&gt;... | <b>-mutation_dist</b> &lt;mutation_dist&gt;...

              Introduce  sequencing  errors in the reads, under the form of mutations (substitutions, insertions
              and deletions) at positions  that  follow  a  specified  distribution  (with  replacement):  model
              (uniform,  linear,  poly4),  model  parameters.  For  example, for a uniform 0.1% error rate, use:
              uniform 0.1. To simulate Sanger errors, use a linear model where the errror rate is 1% at  the  5'
              end  of  reads  and  2%  at  the 3' end: linear 1 2. To model Illumina errors using the 4th degree
              polynome 3e-3 + 3.3e-8 * i^4 (Korbel et al 2009), use: poly4 3e-3 3.3e-8. Use the &lt;mutation_ratio&gt;
              option to alter how many of these mutations are substitutions or indels.  Default: uniform 0 0

       <b>-mr</b> &lt;mutation_ratio&gt;... | <b>-mutation_ratio</b> &lt;mutation_ratio&gt;...

              Indicate the percentage of substitutions and the number of indels (insertions and deletions).  For
              example,  use  '80 20' (4 substitutions for each indel) for Sanger reads. Note that this parameter
              has no effect unless you specify the &lt;mutation_dist&gt; option. Default: 80 20

       <b>-hd</b> &lt;homopolymer_dist&gt; | <b>-homopolymer_dist</b> &lt;homopolymer_dist&gt;

              Introduce sequencing errors in the reads under the form  of  homopolymeric  stretches  (e.g.  AAA,
              CCCCC)  using a specified model where the homopolymer length follows a normal distribution N(mean,
              standard deviation) that is function of the homopolymer length n:

       Margulies: N(n, 0.15 * n)
              ,  Margulies et al. 2005.

       Richter
              : N(n, 0.15 * sqrt(n))        ,  Richter et al. 2008.

       Balzer : N(n, 0.03494 + n * 0.06856) ,  Balzer et al. 2010.

              Default: 0

       <b>-cp</b> &lt;chimera_perc&gt; | <b>-chimera_perc</b> &lt;chimera_perc&gt;

              Specify the percent of reads  in  amplicon  libraries  that  should  be  chimeric  sequences.  The
              'reference'  field  in  the description of chimeric reads will contain the ID of all the reference
              sequences forming the chimeric template. A typical value is 10% for amplicons.  This option can be
              used to generate chimeric shotgun reads as well.  Default: 0 %

       <b>-cd</b> &lt;chimera_dist&gt;... | <b>-chimera_dist</b> &lt;chimera_dist&gt;...

              Specify the distribution of chimeras: bimeras, trimeras,  quadrameras  and  multimeras  of  higher
              order. The default is the average values from Quince et al. 2011: '314 38 1', which corresponds to
              89%  of  bimeras, 11% of trimeras and 0.3% of quadrameras. Note that this option only takes effect
              when you request the generation of chimeras with the &lt;chimera_perc&gt; option. Default: 314 38 1

       <b>-ck</b> &lt;chimera_kmer&gt; | <b>-chimera_kmer</b> &lt;chimera_kmer&gt;

              Activate a method to form chimeras by picking  breakpoints  at  places  where  k-mers  are  shared
              between  sequences.  &lt;chimera_kmer&gt; represents k, the length of the k-mers (in bp). The longer the
              kmer, the more similar the sequences have to be  to  be  eligible  to  form  chimeras.   The  more
              frequent  a  k-mer  is  in  the  pool  of  reference sequences (taking into account their relative
              abundance), the more often this k-mer will be chosen. For example, CHSIM (Edgar et al. 2011)  uses
              this  method  with  a  k-mer  length of 10 bp. If you do not want to use k-mer information to form
              chimeras, use 0, which will result in the reference sequences and breakpoints to be taken randomly
              on the "aligned" reference sequences. Note that this option only takes effect when you request the
              generation of chimeras with  the  &lt;chimera_perc&gt;  option.  Also,  this  options  is  quite  memory
              intensive,  so  you  should  probably  limit  yourself  to  a relatively small number of reference
              sequences if you want to use it. Default: 10 bp

       <b>-af</b> &lt;abundance_file&gt; | <b>-abundance_file</b> &lt;abundance_file&gt;

              Specify the relative abundance of the reference sequences manually in an input file. Each line  of
              the  file  should  contain  a  sequence name and its relative abundance (%), e.g. 'seqABC 82.1' or
              'seqABC 82.1 10.2' if you are specifying two different libraries.

       <b>-am</b> &lt;abundance_model&gt;... | <b>-abundance_model</b> &lt;abundance_model&gt;...

              Relative abundance model for the input reference sequences: uniform, linear, powerlaw, logarithmic
              or exponential. The uniform and linear models do not require a parameter,  but  the  other  models
              take  a  parameter  in  the  range  [0,  infinity). If this parameter is not specified, then it is
              randomly chosen. Examples:

              uniform distribution: uniform powerlaw distribution with parameter 0.1: powerlaw  0.1  exponential
              distribution with automatically chosen parameter: exponential

              Default: uniform 1

       <b>-nl</b> &lt;num_libraries&gt; | <b>-num_libraries</b> &lt;num_libraries&gt;

              Number  of  independent  libraries  to create. Specify how diverse and similar they should be with
              &lt;diversity&gt;,  &lt;shared_perc&gt;  and  &lt;permuted_perc&gt;.   Assign   them   different   MID   tags   with
              &lt;multiplex_mids&gt;. Default: 1

       <b>-mi</b> &lt;multiplex_ids&gt; | <b>-multiplex_ids</b> &lt;multiplex_ids&gt;

              Specify  an  optional  FASTA  file  that  contains  multiplex  sequence identifiers (a.k.a MIDs or
              barcodes) to add to the sequences (one sequence per library, in the order  given).  The  MIDs  are
              included  in  the  length  specified  with  the <b>-read_dist</b> option and can be altered by sequencing
              errors. See the MIDesigner or BarCrawl programs to generate MID sequences.

       <b>-di</b> &lt;diversity&gt;... | <b>-diversity</b> &lt;diversity&gt;...

              This option specifies alpha  diversity,  specifically  the  richness,  i.e.  number  of  reference
              sequences  to  take  randomly and include in each library. Use 0 for the maximum richness possible
              (based on the number of reference sequences available). Provide one value to  make  all  libraries
              have the same diversity, or one richness value per library otherwise. Default: 0

       <b>-sp</b> &lt;shared_perc&gt; | <b>-shared_perc</b> &lt;shared_perc&gt;

              This  option  controls  an aspect of beta-diversity. When creating multiple libraries, specify the
              percent of reference sequences they should have in common (relative to the diversity of the  least
              diverse library). Default: 0 %

       <b>-pp</b> &lt;permuted_perc&gt; | <b>-permuted_perc</b> &lt;permuted_perc&gt;

              This  option controls another aspect of beta-diversity. For multiple libraries, choose the percent
              of the most-abundant reference sequences to permute  (randomly  shuffle)  the  rank-abundance  of.
              Default: 100 %

       <b>-rs</b> &lt;random_seed&gt; | <b>-random_seed</b> &lt;random_seed&gt;

              Seed number to use for the pseudo-random number generator.

       <b>-dt</b> &lt;desc_track&gt; | <b>-desc_track</b> &lt;desc_track&gt;

              Track  read  information  (reference  sequence,  position, errors, ...)  by writing it in the read
              description. Default: 1

       <b>-ql</b> &lt;qual_levels&gt;... | <b>-qual_levels</b> &lt;qual_levels&gt;...

              Generate basic quality scores for the simulated reads. Good residues are given  a  specified  good
              score  (e.g.  30)  and  residues  that  are the result of an insertion or substitution are given a
              specified bad score (e.g. 10). Specify first the  good  score  and  then  the  bad  score  on  the
              command-line, e.g.: 30 10. Default:

       <b>-fq</b> &lt;fastq_output&gt; | <b>-fastq_output</b> &lt;fastq_output&gt;

              Whether  to write the generated reads in FASTQ format (with Sanger-encoded quality scores) instead
              of FASTA and QUAL or not (1: yes, 0: no). &lt;qual_levels&gt; need to be specified for this option to be
              effective. Default: 0

       <b>-bn</b> &lt;base_name&gt; | <b>-base_name</b> &lt;base_name&gt;

              Prefix of the output files. Default: grinder

       <b>-od</b> &lt;output_dir&gt; | <b>-output_dir</b> &lt;output_dir&gt;

              Directory where the results should be written. This folder will be created if needed. Default: .

       <b>-pf</b> &lt;profile_file&gt; | <b>-profile_file</b> &lt;profile_file&gt;

              A file that contains Grinder arguments. This is useful if you use many options or  often  use  the
              same   options.   Lines   with   comments   (#)   are   ignored.   Consider   the   profile  file,
              'simple_profile.txt':

              # A simple Grinder profile <b>-read_dist</b> 105 normal 12 <b>-total_reads</b> 1000

              Running: grinder <b>-reference_file</b> viral_genomes.fa <b>-profile_file</b> simple_profile.txt

              Translates into: grinder <b>-reference_file</b> viral_genomes.fa <b>-read_dist</b> 105  normal  12  <b>-total_reads</b>
              1000

              Note  that  the  arguments  specified  in the profile should not be specified again on the command
              line.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man7/grinder.7.html">grinder</a></b>(7), <b><a href="../man1/grinder.1.html">grinder</a></b>(1), <b><a href="../man1/average_genome_size.1.html">average_genome_size</a></b>(1) and <b><a href="../man1/change_paired_read_orientation.1.html">change_paired_read_orientation</a></b>(1).

grinder 0.5.4                                     December 2020                                       <u><a href="../man1/GRINDER.1.html">GRINDER</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>