<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>art_SOLiD - Simulation of Applied Biosystems SOLiD Sequencing</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/art-nextgen-simulation-tools">art-nextgen-simulation-tools_20160605+dfsg-5build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       art_SOLiD - Simulation of Applied Biosystems SOLiD Sequencing

</pre><h4><b>DESCRIPTION</b></h4><pre>
       ART  is  a  set of simulation tools to generate synthetic next-generation sequencing reads. ART simulates
       sequencing reads by mimicking real sequencing process with empirical error  models  or  quality  profiles
       summarized from large recalibrated sequencing data.

       art_SOLiD can be used for Simulation of Applied Biosystems SOLiD Sequencing.

</pre><h4><b>USAGE</b></h4><pre>
   <b>SINGLE-END</b> <b>(F3</b> <b>READ)</b> <b>SIMULATION</b>
       <b>art_SOLiD</b> [ options ] &lt;INPUT_SEQ_FILE&gt; &lt;OUTPUT_FILE_PREFIX&gt; &lt;LEN_READ&gt; &lt;FOLD_COVERAGE&gt;

   <b>MATE-PAIR</b> <b>READS</b> <b>(F3-R3</b> <b>PAIR)</b> <b>SIMULATION</b>
       <b>art_SOLiD</b>  [  options  ] &lt;INPUT_SEQ_FILE&gt; &lt;OUTPUT_FILE_PREFIX&gt; &lt;LEN_READ&gt; &lt;FOLD_COVERAGE&gt; &lt;MEAN_FRAG_LEN&gt;
       &lt;STD_DEV&gt;

   <b>PAIRED-END</b> <b>READS</b> <b>(F3-F5</b> <b>PAIR)</b> <b>SIMULATION</b>
       <b>art_SOLiD</b> [ options ] &lt;INPUT_SEQ_FILE&gt; &lt;OUTPUT_FILE_PREFIX&gt; &lt;LEN_READ_F3&gt;  &lt;LEN_READ_F5&gt;  &lt;FOLD_COVERAGE&gt;
       &lt;MEAN_FRAG_LEN&gt; &lt;STD_DEV&gt;

   <b>AMPLICON</b> <b>SEQUENCING</b> <b>SIMULATION</b>
       <b>art_SOLiD</b> [ options ] <b>-A</b> s &lt;INPUT_SEQ_FILE&gt; &lt;OUTPUT_FILE_PREFIX&gt; &lt;LEN_READ&gt; &lt;READS_PER_AMPLICON&gt;

       <b>art_SOLiD</b> [ options ] <b>-A</b> m &lt;INPUT_SEQ_FILE&gt; &lt;OUTPUT_FILE_PREFIX&gt; &lt;LEN_READ&gt; &lt;READ_PAIRS_PER_AMPLICON&gt;

       <b>art_SOLiD</b>   [   options   ]   <b>-A</b>  p  &lt;INPUT_SEQ_FILE&gt;  &lt;OUTPUT_FILE_PREFIX&gt;  &lt;LEN_READ_F3&gt;  &lt;LEN_READ_F5&gt;
       &lt;READ_PAIRS_PER_AMPLICON&gt;

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>MANDATORY</b> <b>OPTIONS</b>
       INPUT_SEQ_FILE            -  filename of DNA/RNA reference sequences in FASTA format

       OUTPUT_FILE_PREFIX        -  prefix or directory for all output read data files

       FOLD_COVERAGE             -  fold of read coverage over the reference sequences

       LEN_READ                  -  length of F3/R3 reads

       LEN_READ_F3               -  length of F3 reads for paired-end read simulation

       LEN_READ_F5               -  length of F5 reads for paired-end read simulation

       READS_PER_AMPLICON        -  number of reads per amplicon

       READ_PAIRS_PER_AMPLICON   -  number of read pairs per amplicon

       MEAN_FRAG_LEN             -  mean DNA/RNA fragment size for matepair/paired-end read simulation

       STD_DEV                   -  standard deviation of the DNA/RNA fragment sizes for matepair/paired-end
       read simulation

   <b>OPTIONAL</b> <b>PARAMETERS</b>
       <b>-A</b> specify the read type for amplicon sequencing simulation (s:single-end, m: matepair, p: paired-end)

       <b>-M</b> indicate to use CIGAR 'M' instead of '=/X' for alignment match/mismatch

       <b>-s</b> indicate to generate a SAM alignment file

       <b>-r</b> specify the random seed for the simulation

       <b>-f</b> specify the scale factor adjusting error rate (e.g., <b>-f</b> 0 for zero-error rate simulation)

       <b>-p</b> specify user's own read profile for simulation

</pre><h4><b>EXAMPLES</b></h4><pre>
       1) singl-end 25bp reads simulation at 10X coverage

              art_SOLiD <b>-s</b> seq_reference.fa ./outdir/single_dat 25 10

       2) singl-end 75bp reads simulation at 20X coverage with user's error profile

              art_SOLiD <b>-s</b> <b>-p</b> ../SOLiD_profiles/profile_pseudo ./seq_reference.fa ./dat_userProfile 75 20

       3) matepair 35bp (F3-R3) reads simulation at 20X coverage with DNA/RNA MEAN fragment size 2000bp and STD
       50

              art_SOLiD <b>-s</b> seq_reference.fa ./outdir/matepair_dat 35 20 2000 50

       4) matepair reads simulation with a fixed random seed

              art_SOLiD <b>-r</b> 777 <b>-s</b> seq_reference.fa ./outdir/matepair_fs 50 10 1500 50

       5) paired-end reads (75bp F3, 35bp F5) simulation with the MEAN fragment size 250 and STD 10 at 20X
       coverage

              art_SOLiD <b>-s</b> seq_reference.fa ./outdir/paired_dat 75 35 50 250 10

       6) amplicon sequencing with 25bp single-end reads at 100 reads per amplicon

              art_SOLiD <b>-A</b> s <b>-s</b> amp_reference.fa ./outdir/amp_single 25 100

       7) amplicon sequencing with 50bp matepair reads at 80 read pairs per amplicon

              art_SOLiD <b>-A</b> m <b>-s</b> amp_reference.fa ./outdir/amp_matepair 50 80

       8) amplicon sequencing with paired-end reads (35bp F3, 25bp F5 reads) at 50 pairs per amplicon

              art_SOLiD <b>-A</b> p <b>-s</b> amp_reference.fa ./outdir/amp_pair 35 25 50

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

art_SOLiD 3.19.15                                 February 2016                                     <u><a href="../man1/ART_SOLID.1.html">ART_SOLID</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>