<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>bowtie2-align-s  -  ultrafast  and  memory-efficient  backend  tool for aligning sequencing reads to long</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/bowtie2">bowtie2_2.5.4-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       bowtie2-align-s  -  ultrafast  and  memory-efficient  backend  tool for aligning sequencing reads to long
       reference sequences

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bowtie 2 version 2.5.4 by Ben Langmead (<a href="mailto:langmea@cs.jhu.edu">langmea@cs.jhu.edu</a>, www.cs.jhu.edu/~langmea)

</pre><h4><b>USAGE</b></h4><pre>
       bowtie2-align [options]* <b>-x</b> &lt;bt2-idx&gt; {-1 &lt;m1&gt; <b>-2</b> &lt;m2&gt; | <b>-U</b> &lt;r&gt; |  <b>--interleaved</b>  &lt;i&gt;  |  <b>-b</b>  &lt;bam&gt;}  [-S
       &lt;sam&gt;]

       &lt;bt2-idx&gt;
              Index  filename  prefix  (minus  trailing  .X.bt2).   NOTE:  Bowtie 1 and Bowtie 2 indexes are not
              compatible.

       &lt;m1&gt;   Files with #1 mates, paired with files in &lt;m2&gt;.

       &lt;m2&gt;   Files with #2 mates, paired with files in &lt;m1&gt;.

       &lt;r&gt;    Files with unpaired reads.

       &lt;i&gt;    Files with interleaved paired-end FASTQ/FASTA reads

       &lt;bam&gt;  Files are unaligned BAM sorted by read name.

       &lt;sam&gt;  File for SAM output (default: stdout)

              &lt;m1&gt;, &lt;m2&gt;, &lt;r&gt; can be comma-separated lists (no whitespace) and  can  be  specified  many  times.
              E.g. '-U file1.fq,file2.fq <b>-U</b> file3.fq'.

</pre><h4><b>OPTIONS</b> <b>(defaults</b> <b>in</b> <b>parentheses)</b></h4><pre>
   <b>Input:</b>
       <b>-q</b>     query input files are FASTQ .fq/.fastq (default)

       <b>--tab5</b> query input files are TAB5 .tab5

       <b>--tab6</b> query input files are TAB6 .tab6

       <b>--qseq</b> query input files are in Illumina's qseq format

       <b>-f</b>     query input files are (multi-)FASTA .fa/.mfa

       <b>-r</b>     query input files are raw one-sequence-per-line

       <b>-F</b> k:&lt;int&gt;,i:&lt;int&gt; query input files are continuous FASTA where reads
              are  substrings (k-mers) extracted from the FASTA file and aligned at offsets 1, 1+i, 1+2i ... end
              of reference

       <b>-c</b>     &lt;m1&gt;, &lt;m2&gt;, &lt;r&gt; are sequences themselves, not files

       <b>-s</b>/--skip &lt;int&gt;
              skip the first &lt;int&gt; reads/pairs in the input (none)

       <b>-u</b>/--upto &lt;int&gt;
              stop after first &lt;int&gt; reads/pairs (no limit)

       <b>-5</b>/--trim5 &lt;int&gt;
              trim &lt;int&gt; bases from 5'/left end of reads (0)

       <b>-3</b>/--trim3 &lt;int&gt;
              trim &lt;int&gt; bases from 3'/right end of reads (0)

       <b>--trim-to</b> [3:|5:]&lt;int&gt; trim reads exceeding &lt;int&gt; bases from either 3' or 5' end
              If the read end is not specified then it defaults to 3 (0)

       <b>--phred33</b>
              qualities are Phred+33 (default)

       <b>--phred64</b>
              qualities are Phred+64

       <b>--int-quals</b>
              qualities encoded as space-delimited integers

   <b>Presets:</b>
       Same as:

              For <b>--end-to-end</b>:

       <b>--very-fast</b>            <b>-D</b> 5 <b>-R</b> 1 <b>-N</b> 0 <b>-L</b> 22 <b>-i</b> S,0,2.50

       <b>--fast</b>                 <b>-D</b> 10 <b>-R</b> 2 <b>-N</b> 0 <b>-L</b> 22 <b>-i</b> S,0,2.50

       <b>--sensitive</b>            <b>-D</b> 15 <b>-R</b> 2 <b>-N</b> 0 <b>-L</b> 22 <b>-i</b> S,1,1.15 (default)

       <b>--very-sensitive</b>       <b>-D</b> 20 <b>-R</b> 3 <b>-N</b> 0 <b>-L</b> 20 <b>-i</b> S,1,0.50

              For <b>--local</b>:

       <b>--very-fast-local</b>      <b>-D</b> 5 <b>-R</b> 1 <b>-N</b> 0 <b>-L</b> 25 <b>-i</b> S,1,2.00

       <b>--fast-local</b>           <b>-D</b> 10 <b>-R</b> 2 <b>-N</b> 0 <b>-L</b> 22 <b>-i</b> S,1,1.75

       <b>--sensitive-local</b>      <b>-D</b> 15 <b>-R</b> 2 <b>-N</b> 0 <b>-L</b> 20 <b>-i</b> S,1,0.75 (default)

       <b>--very-sensitive-local</b> <b>-D</b> 20 <b>-R</b> 3 <b>-N</b> 0 <b>-L</b> 20 <b>-i</b> S,1,0.50

   <b>Alignment:</b>
       <b>-N</b> &lt;int&gt;
              max # mismatches in seed alignment; can be 0 or 1 (0)

       <b>-L</b> &lt;int&gt;
              length of seed substrings; must be &gt;3, &lt;32 (22)

       <b>-i</b> &lt;func&gt;
              interval between seed substrings w/r/t read len (S,1,1.15)

       <b>--n-ceil</b> &lt;func&gt;
              func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)

       <b>--dpad</b> &lt;int&gt;
              include &lt;int&gt; extra ref chars on sides of DP table (15)

       <b>--gbar</b> &lt;int&gt;
              disallow gaps within &lt;int&gt; nucs of read extremes (4)

       <b>--ignore-quals</b>
              treat all quality values as 30 on Phred scale (off)

       <b>--nofw</b> do not align forward (original) version of read (off)

       <b>--norc</b> do not align reverse-complement version of read (off)

       <b>--no-1mm-upfront</b>
              do not allow 1 mismatch alignments before attempting to scan for the optimal seeded alignments

       <b>--end-to-end</b>
              entire read must align; no clipping (on)

              OR

       <b>--local</b>
              local alignment; ends might be soft clipped (off)

   <b>Scoring:</b>
       <b>--ma</b> &lt;int&gt;
              match bonus (0 for <b>--end-to-end</b>, 2 for <b>--local</b>)

       <b>--mp</b> &lt;int&gt;
              max penalty for mismatch; lower qual = lower penalty (6)

       <b>--np</b> &lt;int&gt;
              penalty for non-A/C/G/Ts in read/ref (1)

       <b>--rdg</b> &lt;int&gt;,&lt;int&gt;
              read gap open, extend penalties (5,3)

       <b>--rfg</b> &lt;int&gt;,&lt;int&gt;
              reference gap open, extend penalties (5,3)

       <b>--score-min</b> &lt;func&gt; min acceptable alignment score w/r/t read length
              (G,20,8 for local, L,-0.6,-0.6 for end-to-end)

   <b>Reporting:</b>
       (default)
              look for multiple alignments, report best, with MAPQ

              OR

       <b>-k</b> &lt;int&gt;
              report up to &lt;int&gt; alns per read; MAPQ not meaningful

              OR

       <b>-a</b>/--all
              report all alignments; very slow, MAPQ not meaningful

   <b>Effort:</b>
       <b>-D</b> &lt;int&gt;
              give up extending after &lt;int&gt; failed extends in a row (15)

       <b>-R</b> &lt;int&gt;
              for reads w/ repetitive seeds, try &lt;int&gt; sets of seeds (2)

              Paired-end:

       <b>-I</b>/--minins &lt;int&gt;
              minimum fragment length (0)

       <b>-X</b>/--maxins &lt;int&gt;
              maximum fragment length (500)

       <b>--fr</b>/--rf/--ff     <b>-1</b>, <b>-2</b> mates align fw/rev, rev/fw, fw/fw (<b>--fr</b>)

       <b>--no-mixed</b>
              suppress unpaired alignments for paired reads

       <b>--no-discordant</b>
              suppress discordant alignments for paired reads

       <b>--dovetail</b>
              concordant when mates extend past each other

       <b>--no-contain</b>
              not concordant when one mate alignment contains other

       <b>--no-overlap</b>
              not concordant when mates overlap at all

              BAM:

       <b>--align-paired-reads</b>
              Bowtie2 will, by default, attempt  to  align  unpaired  BAM  reads.   Use  this  option  to  align
              paired-end reads instead.

       <b>--preserve-tags</b>
              Preserve  tags  from the original BAM record by appending them to the end of the corresponding SAM
              output.

   <b>Output:</b>
       <b>-t</b>/--time
              print wall-clock time taken by search phases

       <b>--quiet</b>
              print nothing to stderr except serious errors

       <b>--met-file</b> &lt;path&gt;
              send metrics to file at &lt;path&gt; (off)

       <b>--met-stderr</b>
              send metrics to stderr (off)

       <b>--met</b> &lt;int&gt;
              report internal counters &amp; metrics every &lt;int&gt; secs (1)

       <b>--no-unal</b>
              suppress SAM records for unaligned reads

       <b>--no-head</b>
              suppress header lines, i.e. lines starting with @

       <b>--no-sq</b>
              suppress @SQ header lines

       <b>--rg-id</b> &lt;text&gt;
              set read group id, reflected in @RG line and RG:Z: opt field

       <b>--rg</b> &lt;text&gt;
              add &lt;text&gt; ("lab:value") to @RG line of SAM header.  Note: @RG line only printed when  <b>--rg-id</b>  is
              set.

       <b>--omit-sec-seq</b>
              put '*' in SEQ and QUAL fields for secondary alignments.

       <b>--sam-no-qname-trunc</b>
              Suppress standard behavior of truncating readname at first whitespace at the expense of generating
              non-standard SAM.

       <b>--xeq</b>  Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.

       <b>--soft-clipped-unmapped-tlen</b>
              Exclude soft-clipped bases when reporting TLEN.

       <b>--sam-append-comment</b>
              Append FASTA/FASTQ comment to SAM record.

       <b>--sam-opt-config</b> &lt;config&gt;
              Use &lt;config&gt;, example '-MD,YP,-AS', to toggle SAM Optional fields.

   <b>Performance:</b>

       <b>-p</b>/--threads &lt;int&gt; number of alignment threads to launch (1)

       <b>--reorder</b>
              force SAM output order to match order of input reads

       <b>--mm</b>   use memory-mapped I/O for index; many 'bowtie's can share

   <b>Other:</b>
       <b>--qc-filter</b>
              filter out reads that are bad according to QSEQ filter

       <b>--seed</b> &lt;int&gt;
              seed for random number generator (0)

       <b>--non-deterministic</b>
              seed rand. gen. arbitrarily instead of using read attributes

       <b>--version</b>
              print version information and quit

       <b>-h</b>/--help
              print this usage message

bowtie2-align-s 2.5.4                              August 2024                                <u><a href="../man1/BOWTIE2-ALIGN-S.1.html">BOWTIE2-ALIGN-S</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>