<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>htseq-qa - Perform simple quality assessment of high-throughput sequencing reads</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/python3-htseq">python3-htseq_2.0.9+dfsg-1build2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       htseq-qa - Perform simple quality assessment of high-throughput sequencing reads

       The  Python script <b>htseq-qa</b> takes a file with sequencing reads (either raw or aligned reads) and produces
       a PDF file with useful plots to assess the technical quality of a run.

</pre><h4><b>PLOT</b></h4><pre>
       Here is a typical plot: [image]

       The plot is made from a SAM file, which contained aligned and unalignable reads.  The left column is made
       from the non-aligned, the right column from the aligned reads. The header informs you about the  name  of
       the SAM file, and the number of reads.

       The  upper  row  shows how often which base was called for each position in the read. In this sample, the
       non-alignable reads have a clear excess in A. The aligned reads  have  a  balance  between  complementing
       reads:  A  and C (reddish colours) have equal levels, and so do C and G (greenish colours). The sequences
       seem to be AT rich. Furthermore, nearly all aligned reads start with a T, followed by an A, and then, a C
       in 70% and an A in 30% of the reads. Such an imbalance would be reason for concern  if  it  has  no  good
       explanation. Here, the reason is that the fragmentation of the sample was done by enzyme digestion.

       The  lower  half  shows the abundance of base-call quality scores at the different positions in the read.
       Nearly all aligned reads have a quality of 34 over their whole length, while for the  non-aligned  reads,
       some reads have lower quality scores towards their ends.

</pre><h4><b>USAGE</b></h4><pre>
       Note that <b>htseq-qa</b> needs matplotlib to produce the plot, so you need to install this module, as described
       <u>here</u> on the matplotlib web site.

       After you have installed HTSeq (see install) and matplotlib, you can run <b>htseq-qa</b> from the command line:

          htseq-qa [options] read_file

       If the file <b>htseq-qa</b> is not in your path, you can, alternatively, call the script with

          python -m HTSeq.scripts.qa [options] read_file

       The  <u>read_file</u>  is either a FASTQ file or a SAM file. For a SAM file, a plot with two columns is produced
       as above, for a FASTQ file, you get only one column.

       The output is written into a file with the same name as <u>read_file</u>, with the suffix <b>.pdf</b>  added.  View  it
       with a PDF viewer such as the Acrobat Reader.

   <b>Options</b>
       <b>-t</b> <b>&lt;type&gt;,</b> <b>--type=&lt;type&gt;</b>
              The file type of the <u>read_file</u>. Supported values for <u>&lt;type&gt;</u> are:

              • <b>sam</b>:  a  SAM file (Note that the <u>SAMtools</u> contain Perl scripts to convert most alignment formats
                to SAM)

              • <b>solexa-export</b>: an <b>_export.txt</b> file as produced by the  SolexaPipeline  software  after  aligning
                with Eland (<b>htseq-qa</b> expects the new Solexa quality encoding as produced by version 1.3 or newer
                of the SolexaPipeline)

              • <b>fastq</b>: a FASTQ file with standard (Sanger or Phred) quality encoding

              • <b>solexa-fastq</b>: a FASTQ file with Solexa quality encoding, as produced by the SolexaPipeline after
                base-calling  with  Bustard  (<b>htseq-qa</b>  expects  the  new Solexa quality encoding as produced by
                version 1.3 or newer of the SolexaPipeline)

       <b>-o</b> <b>&lt;outfile&gt;,</b> <b>--outfile=&lt;outfile&gt;</b>
              output filename (default is <u>&lt;read_file&gt;``</u>.pdf``)

       <b>-r</b> <b>&lt;readlen&gt;,</b> <b>--readlength=&lt;readlen&gt;</b>
              the maximum read length (when not specified, the script guesses from the file

       <b>-g</b> <b>&lt;gamma&gt;,</b> <b>--gamma=&lt;gamma&gt;</b>
              the gamma factor for the contrast adjustment of the quality score plot

       <b>-n,</b> <b>--nosplit</b>
              do not split reads in unaligned and aligned ones, i.e., produce a one-column plot

       <b>-m,</b> <b>--maxqual</b>
              the maximum quality score that appears in the data (default: 40)

       <b>-h,</b> <b>--help</b>
              Show a usage summary and exit

</pre><h4><b>AUTHOR</b></h4><pre>
       Simon Anders

</pre><h4><b>COPYRIGHT</b></h4><pre>
       2017, Simon Anders

0.6.1p1                                           Oct 01, 2017                                       <u><a href="../man1/HTSEQ-QA.1.html">HTSEQ-QA</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>