<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>macs2_pileup - Model-based Analysis for ChIP-Sequencing</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/macs">macs_2.2.7.1-5ubuntu1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       macs2_pileup - Model-based Analysis for ChIP-Sequencing

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: macs2 pileup [-h] <b>-i</b> IFILE [IFILE ...] <b>-o</b> OUTPUTFILE [--outdir OUTDIR]

       [-f {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE}]
              [-B] [--extsize EXTSIZE] [--buffer-size BUFFER_SIZE] [--verbose VERBOSE]

   <b>options:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>-i</b> IFILE [IFILE ...], <b>--ifile</b> IFILE [IFILE ...]
              Alignment  file.  If  multiple  files  are  given  as  '-t  A B C', then they will all be read and
              combined. Note that pair-end data is not supposed to work with this command. REQUIRED.

       <b>-o</b> OUTPUTFILE, <b>--ofile</b> OUTPUTFILE
              Output bedGraph file name. If not specified, will write to standard output. REQUIRED.

       <b>--outdir</b> OUTDIR
              If specified all output files will be written to that  directory.  Default:  the  current  working
              directory

       <b>-f</b> {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE}, <b>--format</b>
       {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE}
              Format  of  tag file, "AUTO", "BED", "ELAND", "ELANDMULTI", "ELANDEXPORT", "SAM", "BAM", "BOWTIE",
              "BAMPE", or "BEDPE". The default AUTO option will let 'macs2 pileup' decide which format the  file
              is.   DEFAULT:  "AUTO",  MACS2  will  pick  a  format  from  "AUTO", "BED", "ELAND", "ELANDMULTI",
              "ELANDEXPORT", "SAM", "BAM" and "BOWTIE". If the format is  BAMPE  or  BEDPE,  please  specify  it
              explicitly. Please note that when the format is BAMPE or BEDPE, the <b>-B</b> and <b>--extsize</b> options would
              be ignored.

       <b>-B</b>, <b>--both-direction</b>
              By  default,  any  read  will  be extended towards downstream direction by extension size. So it's
              [0,size-1] (1-based index system) for plus strand read and [-size+1,0] for minus strand read where
              position 0 is 5' end of the aligned read. Default behavior can simulate MACS2  way  of  piling  up
              ChIP  sample  reads  where  extension size is set as fragment size/d. If this option is set as on,
              aligned reads will be extended in both upstream and downstream directions by  extension  size.  It
              means [-size,size] where 0 is the 5' end of a aligned read. It can partially simulate MACS2 way of
              piling  up control reads. However MACS2 local bias is calculated by maximizing the expected pileup
              over a ChIP fragment size/d estimated from 10kb, 1kb, d and whole genome background.  This  option
              will be ignored when the format is set as BAMPE or BEDPE. DEFAULT: False

       <b>--extsize</b> EXTSIZE
              The  extension  size  in bps. Each alignment read will become a EXTSIZE of fragment, then be piled
              up. Check description for <b>-B</b> for detail. It's twice the `shiftsize` in old MACSv1  language.  This
              option will be ignored when the format is set as BAMPE or BEDPE.  DEFAULT: 200

       <b>--buffer-size</b> BUFFER_SIZE
              Buffer size for incrementally increasing internal array size to store reads alignment information.
              In  most  cases,  you  don't have to change this parameter.  However, if there are large number of
              chromosomes/contigs/scaffolds in your alignment, it's recommended to specify a smaller buffer size
              in order to decrease memory usage (but it will take longer time to read alignment files).  Minimum
              memory  requested  for reading an alignment file is about # of CHROMOSOME * BUFFER_SIZE * 8 Bytes.
              DEFAULT: 100000

       <b>--verbose</b> VERBOSE
              Set verbose level. 0: only show critical message, 1: show  additional  warning  message,  2:  show
              process  information,  3:  show debug messages. If you want to know where are the duplicate reads,
              use 3. DEFAULT:2

macs2 pileup 2.2.7.1                               April 2022                                    <u><a href="../man1/MACS2_PILEUP.1.html">MACS2_PILEUP</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>