<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>macs2_randsample - Model-based Analysis for ChIP-Sequencing</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/macs">macs_2.2.7.1-5ubuntu1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       macs2_randsample - Model-based Analysis for ChIP-Sequencing

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: macs2 randsample [-h] <b>-i</b> IFILE [IFILE ...] (<b>-p</b> PERCENTAGE | <b>-n</b> NUMBER)

       [--seed SEED] [-o OUTPUTFILE] [--outdir OUTDIR]
              [-s  TSIZE] [-f {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE}] [--buffer-size
              BUFFER_SIZE] [--verbose VERBOSE]

   <b>options:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>-i</b> IFILE [IFILE ...], <b>--ifile</b> IFILE [IFILE ...]
              Alignment file. If multiple files are given as '-t A  B  C',  then  they  will  all  be  read  and
              combined. Note that pair-end data is not supposed to work with this command. REQUIRED.

       <b>-p</b> PERCENTAGE, <b>--percentage</b> PERCENTAGE
              Percentage  of  tags  you  want to keep. Input 80.0 for 80%. This option can't be used at the same
              time with <b>-n</b>/--num. REQUIRED

       <b>-n</b> NUMBER, <b>--number</b> NUMBER
              Number of tags you want to keep. Input 8000000 or 8e+6 for 8 million. This option can't be used at
              the same time with <b>-p</b>/--percent. Note that the number of tags in  output  is  approximate  as  the
              number specified here.  REQUIRED

       <b>--seed</b> SEED
              Set  the  random  seed  while  down  sampling  data. Must be a non-negative integer in order to be
              effective.  DEFAULT: not set

       <b>-o</b> OUTPUTFILE, <b>--ofile</b> OUTPUTFILE
              Output BED file name. If not specified, will write to standard output. Note, if the  input  format
              is BAMPE or BEDPE, the output will be in BEDPE format. DEFAULT: stdout

       <b>--outdir</b> OUTDIR
              If  specified  all  output  files  will be written to that directory. Default: the current working
              directory

       <b>-s</b> TSIZE, <b>--tsize</b> TSIZE
              Tag size. This will override the auto detected tag size. DEFAULT: Not set

       <b>-f</b> {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE}, <b>--format</b>
       {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE}
              Format of tag file, "AUTO", "BED" or "ELAND" or "ELANDMULTI" or "ELANDEXPORT" or "SAM" or "BAM" or
              "BOWTIE" or "BAMPE" or "BEDPE". The default AUTO option will macs2 randsample decide which  format
              the    file    is.    Please    check    the   definition   in   README   file   if   you   choose
              ELAND/ELANDMULTI/ELANDEXPORT/SAM/BAM/BOWTIE or BAMPE/BEDPE. DEFAULT: "AUTO"

       <b>--buffer-size</b> BUFFER_SIZE
              Buffer size for incrementally increasing internal array size to store reads alignment information.
              In most cases, you don't have to change this parameter.  However, if there  are  large  number  of
              chromosomes/contigs/scaffolds in your alignment, it's recommended to specify a smaller buffer size
              in  order to decrease memory usage (but it will take longer time to read alignment files). Minimum
              memory requested for reading an alignment file is about # of CHROMOSOME * BUFFER_SIZE *  8  Bytes.
              DEFAULT: 100000

       <b>--verbose</b> VERBOSE
              Set  verbose  level.  0:  only  show critical message, 1: show additional warning message, 2: show
              process information, 3: show debug messages. If you want to know where are  the  duplicate  reads,
              use 3. DEFAULT:2

macs2 randsample 2.2.7.1                           April 2022                                <u><a href="../man1/MACS2_RANDSAMPLE.1.html">MACS2_RANDSAMPLE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>