<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>macs2_bdgcmp - Model-based Analysis for ChIP-Sequencing</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/macs">macs_2.2.7.1-5ubuntu1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       macs2_bdgcmp - Model-based Analysis for ChIP-Sequencing

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: macs2 bdgcmp [-h] <b>-t</b> TFILE <b>-c</b> CFILE [-S SFACTOR] [-p PSEUDOCOUNT]

       [-m {ppois,qpois,subtract,logFE,FE,logLR,slogLR,max} [{ppois,qpois,subtract,logFE,FE,logLR,slogLR,max}
       ...]]
              [--outdir OUTDIR] (<b>--o-prefix</b> OPREFIX | <b>-o</b> OFILE [OFILE ...])

   <b>options:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>-t</b> TFILE, <b>--tfile</b> TFILE
              Treatment bedGraph file, e.g. *_treat_pileup.bdg from MACSv2. REQUIRED

       <b>-c</b> CFILE, <b>--cfile</b> CFILE
              Control bedGraph file, e.g. *_control_lambda.bdg from MACSv2. REQUIRED

       <b>-S</b> SFACTOR, <b>--scaling-factor</b> SFACTOR
              Scaling  factor  for  treatment and control track. Keep it as 1.0 or default in most cases. Set it
              ONLY while you have SPMR output from MACS2  callpeak,  and  plan  to  calculate  scores  as  MACS2
              callpeak  module. If you want to simulate 'callpeak' w/o '--to-large', calculate effective smaller
              sample size after filtering redudant reads in million (e.g., put 31.415926 if effective reads  are
              31,415,926)  and input it for '-S'; for 'callpeak <b>--to-large</b>', calculate effective reads in larger
              sample. DEFAULT: 1.0

       <b>-p</b> PSEUDOCOUNT, <b>--pseudocount</b> PSEUDOCOUNT
              The pseudocount used for calculating  logLR,  logFE  or  FE.  The  count  will  be  applied  after
              normalization of sequencing depth. DEFAULT: 0.0, no pseudocount is applied.

       <b>-m</b> {ppois,qpois,subtract,logFE,FE,logLR,slogLR,max} [{ppois,qpois,subtract,logFE,FE,logLR,slogLR,max}
       ...], <b>--method</b> {ppois,qpois,subtract,logFE,FE,logLR,slogLR,max}
       [{ppois,qpois,subtract,logFE,FE,logLR,slogLR,max} ...]
              Method to use while calculating a score in any bin by comparing treatment value and control value.
              Available  choices  are:  ppois, qpois, subtract, logFE, logLR, and slogLR. They represent Poisson
              Pvalue (<b>-log10</b>(pvalue) form) using control as lambda and treatment as observation, q-value through
              a BH process for poisson pvalues, subtraction from treatment, linear scale fold enrichment,  log10
              fold  enrichment(need  to  set pseudocount), log10 likelihood between ChIP-enriched model and open
              chromatin model(need to set pseudocount), symmetric log10 likelihood between  two  ChIP-enrichment
              models, or maximum value between the two tracks. Default option is ppois.

       <b>--outdir</b> OUTDIR
              If  specified  all  output  files  will be written to that directory. Default: the current working
              directory

       <b>--o-prefix</b> OPREFIX
              The PREFIX of output bedGraph file to write scores. If it is given as A, and  method  is  'ppois',
              output file will be A_ppois.bdg. Mutually exclusive with <b>-o</b>/--ofile.

       <b>-o</b> OFILE [OFILE ...], <b>--ofile</b> OFILE [OFILE ...]
              Output  filename.  Mutually  exclusive with <b>--o-prefix</b>.  The number and the order of arguments for
              <b>--ofile</b> must be the same as for <b>-m</b>.

macs2 bdgcmp 2.2.7.1                               April 2022                                    <u><a href="../man1/MACS2_BDGCMP.1.html">MACS2_BDGCMP</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>