<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>macs2_refinepeak - Model-based Analysis for ChIP-Sequencing</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/macs">macs_2.2.7.1-5ubuntu1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       macs2_refinepeak - Model-based Analysis for ChIP-Sequencing

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: macs2 refinepeak [-h] <b>-b</b> BEDFILE <b>-i</b> IFILE [IFILE ...]

       [-f {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE}]
              [-c  CUTOFF] [-w WINDOWSIZE] [--buffer-size BUFFER_SIZE] [--verbose VERBOSE] [--outdir OUTDIR] (<b>-o</b>
              OFILE | <b>--o-prefix</b> OPREFIX)

   <b>options:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>-b</b> BEDFILE
              Candidate peak file in BED format. REQUIRED.

       <b>-i</b> IFILE [IFILE ...], <b>--ifile</b> IFILE [IFILE ...]
              ChIP-seq alignment file. If multiple files are given as '-t A B C', then they will all be read and
              combined. Note that pair-end data is not supposed to work with this command. REQUIRED.

       <b>-f</b> {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE}, <b>--format</b>
       {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE}
              Format of tag file, "AUTO", "BED" or "ELAND" or "ELANDMULTI" or "ELANDEXPORT" or "SAM" or "BAM" or
              "BOWTIE". The default AUTO option will let 'macs2 refinepeak' decide which  format  the  file  is.
              Please      check      the      definition      in      README      file     if     you     choose
              ELAND/ELANDMULTI/ELANDEXPORT/SAM/BAM/BOWTIE. DEFAULT: "AUTO"

       <b>-c</b> CUTOFF, <b>--cutoff</b> CUTOFF
              Cutoff DEFAULT: 5

       <b>-w</b> WINDOWSIZE, <b>--window-size</b> WINDOWSIZE
              Scan window size on both side of the summit (default: 100bp)

       <b>--buffer-size</b> BUFFER_SIZE
              Buffer size for incrementally increasing internal array size to store reads alignment information.
              In most cases, you don't have to change this parameter.  However, if there  are  large  number  of
              chromosomes/contigs/scaffolds in your alignment, it's recommended to specify a smaller buffer size
              in  order to decrease memory usage (but it will take longer time to read alignment files). Minimum
              memory requested for reading an alignment file is about # of CHROMOSOME * BUFFER_SIZE *  8  Bytes.
              DEFAULT: 100000

       <b>--verbose</b> VERBOSE
              Set  verbose  level.  0:  only  show critical message, 1: show additional warning message, 2: show
              process information, 3: show debug messages. If you want to know where are  the  duplicate  reads,
              use 3. DEFAULT:2

       <b>--outdir</b> OUTDIR
              If  specified  all  output  files  will be written to that directory. Default: the current working
              directory

       <b>-o</b> OFILE, <b>--ofile</b> OFILE
              Output file name. Mutually exclusive with <b>--o-prefix</b>.

       <b>--o-prefix</b> OPREFIX
              Output file prefix. Mutually exclusive with <b>-o</b>/--ofile.

macs2 refinepeak 2.2.7.1                           April 2022                                <u><a href="../man1/MACS2_REFINEPEAK.1.html">MACS2_REFINEPEAK</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>