<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Variation::RNAChange - Sequence change class for RNA level</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-variation-perl">libbio-variation-perl_1.7.5-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Variation::RNAChange - Sequence change class for RNA level

</pre><h4><b>SYNOPSIS</b></h4><pre>
          $rnachange = Bio::Variation::RNAChange-&gt;new
              ('-start'         =&gt; $start,
               '-end'           =&gt; $end,
               '-length'        =&gt; $len,
               '-codon_pos'     =&gt; $cp,
               '-upStreamSeq'   =&gt; $upflank,
               '-dnStreamSeq'   =&gt; $dnflank,
               '-proof'         =&gt; $proof,
               '-isMutation'    =&gt; 1,
               '-mut_number'    =&gt; $mut_number
              );
          $a1 = Bio::Variation::Allele-&gt;new;
          $a1-&gt;seq('a');
          $rnachange-&gt;allele_ori($a1);
          my $a2 = Bio::Variation::Allele-&gt;new;
          $a2-&gt;seq('t');
          $rnachange-&gt;add_Allele($a2);
          $rnachange-&gt;allele_mut($a2);

          print "The codon change is ", $rnachange-&gt;codon_ori,
              "&gt;", $rnachange-&gt;codon_mut, "\n";

          # add it to a SeqDiff container object
          $seqdiff-&gt;add_Variant($rnachange);

          # and create links to and from DNA level mutation objects
          $rnachange-&gt;DNAMutation($dnamut);
          $dnamut-&gt;RNAChange($rnachange);

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The instantiable class Bio::Variation::DNAMutation describes basic sequence changes at RNA molecule
       level. It uses methods defined in superclass Bio::Variation::VariantI. See Bio::Variation::VariantI for
       details.

       You are normally expected to create a corresponding Bio::Variation::DNAMutation object even if mutation
       is defined at RNA level. The numbering follows then cDNA numbering.  Link the DNAMutation object to the
       RNAChange object using the method <b>DNAMutation()</b>. If the variation described by a RNAChange object is
       translated, link the corresponding Bio::Variation::AAChange object to it using method <b>AAChange()</b>. See
       Bio::Variation::DNAMutation and Bio::Variation::AAChange for more information.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing lists  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Heikki</b> <b>Lehvaslaiho</b></h4><pre>
       Email:  heikki-at-bioperl-dot-org

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>codon_ori</b>
        Title   : codon_ori
        Usage   : $obj-&gt;codon_ori();
        Function:

                   Sets and returns codon_ori triplet.  If value is not set,
                   creates the codon triplet from the codon position and
                   flanking sequences.  The string has to be three characters
                   long. The character content is not checked.

        Example :
        Returns : string
        Args    : string

   <b>codon_mut</b>
        Title   : codon_mut
        Usage   : $obj-&gt;codon_mut();
        Function:

                   Sets and returns codon_mut triplet.  If value is not
                   set, creates the codon triplet from the codon position and
                   flanking sequences. Return undef for other than point mutations.

        Example :
        Returns : string
        Args    : string

   <b>codon_pos</b>
        Title   : codon_pos
        Usage   : $obj-&gt;codon_pos();
        Function:

                   Sets and returns the position of the mutation start in the
                   codon. If value is not set, returns false.

        Example :
        Returns : 1,2,3
        Args    : none if get, the new value if set

   <b>codon_table</b>
        Title   : codon_table
        Usage   : $obj-&gt;codon_table();
        Function:

                   Sets and returns the codon table id of the RNA
                   If value is not set, returns 1, 'universal' code, as the default.

        Example :
        Returns : integer
        Args    : none if get, the new value if set

   <b>DNAMutation</b>
        Title   : DNAMutation
        Usage   : $mutobj = $obj-&gt;DNAMutation;
                : $mutobj = $obj-&gt;DNAMutation($objref);
        Function: Returns or sets the link-reference to a mutation/change object.
                  If there is no link, it will return undef
        Returns : an obj_ref or undef

   <b>AAChange</b>
        Title   : AAChange
        Usage   : $mutobj = $obj-&gt;AAChange;
                : $mutobj = $obj-&gt;AAChange($objref);
        Function: Returns or sets the link-reference to a mutation/change object.
                  If there is no link, it will return undef
        Returns : an obj_ref or undef

   <b>exons_modified</b>
        Title   : exons_modified
        Usage   : $modified = $obj-&gt;exons_modified;
                : $modified = $obj-&gt;<a href="../man1/exons_modified.1.html">exons_modified</a>(1);
        Function: Returns or sets information (example: a simple boolean flag) about
                  the modification of exons as a result of a mutation.

   <b>region</b>
        Title   : region
        Usage   : $obj-&gt;region();
        Function:

                   Sets and returns the name of the sequence region type or
                   protein domain at this location.  If value is not set,
                   returns false.

        Example :
        Returns : string
        Args    : string

   <b>cds_end</b>
        Title   : cds_end
        Usage   : $cds_end = $obj-&gt;get_cds_end();
        Function:

                  Sets or returns the cds_end from the beginning of the DNA sequence
                  to the coordinate start used to describe variants.
                  Should be the location of the last nucleotide of the
                  terminator codon of the gene.

        Example :
        Returns : value of cds_end, a scalar
        Args    :

   <b>label</b>
        Title   : label
        Usage   : $obj-&gt;label();
        Function:

                   Sets and returns mutation event label(s).  If value is not
                   set, or no argument is given returns false.  Each
                   instantiable subclass of L&lt;Bio::Variation::VariantI&gt; needs
                   to implement this method. Valid values are listed in
                   'Mutation event controlled vocabulary' in
                   <a href="http://www.ebi.ac.uk/mutations/recommendations/mutevent.html">http://www.ebi.ac.uk/mutations/recommendations/mutevent.html</a>.

        Example :
        Returns : string
        Args    : string

   <b>_change_codon_pos</b>
        Title   : _change_codon_pos
        Usage   : $newCodonPos = _change_codon_pos($myCodonPos, 5)
        Function:

                  Keeps track of the codon position in a changeing sequence

        Returns : codon_pos = integer 1, 2 or 3
        Args    : valid codon position
                  signed integer offset to a new location in sequence

perl v5.32.1                                       2021-12-02                     <u>Bio::Variation::<a href="../man3pm/RNAChange.3pm.html">RNAChange</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>