<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Variation::SNP - submitted SNP</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-variation-perl">libbio-variation-perl_1.7.5-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Variation::SNP - submitted SNP

</pre><h4><b>SYNOPSIS</b></h4><pre>
         $SNP = Bio::Variation::SNP-&gt;new ();

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Inherits from Bio::Variation::SeqDiff and Bio::Variation::Allele, with additional methods that are
       (db)SNP specific (ie, refSNP/subSNP IDs, batch IDs, validation methods).

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists. Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution. Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b></h4><pre>
       Allen Day &lt;<a href="mailto:allenday@ucla.edu">allenday@ucla.edu</a>&gt;

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>get/set-able</b> <b>methods</b>
        Usage   : $is = $snp-&gt;method()
        Function: for getting/setting attributes
        Returns : a value.  probably a scalar.
        Args    : if you're trying to set an attribute, pass in the new value.

        Methods:
        --------
        id
        type
        observed
        seq_5
        seq_3
        ncbi_build
        ncbi_chr_hits
        ncbi_ctg_hits
        ncbi_seq_loc
        ucsc_build
        ucsc_chr_hits
        ucsc_ctg_hits
        heterozygous
        heterozygous_SE
        validated
        genotype
        handle
        batch_id
        method
        locus_id
        symbol
        mrna
        protein
        functional_class

   <b>is_subsnp</b>
        Title   : is_subsnp
        Usage   : $is = $snp-&gt;is_subsnp()
        Function: returns 1 if $snp is a subSNP
        Returns : 1 or undef
        Args    : NONE

   <b>subsnp</b>
        Title   : subsnp
        Usage   : $subsnp = $snp-&gt;subsnp()
        Function: returns the currently active subSNP of $snp
        Returns : Bio::Variation::SNP
        Args    : NONE

   <b>add_subsnp</b>
        Title   : add_subsnp
        Usage   : $subsnp = $snp-&gt;add_subsnp()
        Function: pushes the previous value returned by subsnp() onto a stack,
                  accessible with each_subsnp().
                  Sets return value of subsnp() to a new Bio::Variation::SNP
                  object, and returns that object.
        Returns : Bio::Varitiation::SNP
        Args    : NONE

   <b>each_subsnp</b>
        Title   : each_subsnp
        Usage   : @subsnps = $snp-&gt;each_subsnp()
        Function: returns a list of the subSNPs of a refSNP
        Returns : list
        Args    : NONE

perl v5.32.1                                       2021-12-02                           <u>Bio::Variation::<a href="../man3pm/SNP.3pm.html">SNP</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>