<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Variation::SeqDiff - Container class for mutation/variant descriptions</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-variation-perl">libbio-variation-perl_1.7.5-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Variation::SeqDiff - Container class for mutation/variant descriptions

</pre><h4><b>SYNOPSIS</b></h4><pre>
         $seqDiff = Bio::Variation::SeqDiff-&gt;new (
                                                  -id =&gt; $M20132,
                                                  -alphabet =&gt; 'rna',
                                                  -gene_symbol =&gt; 'AR'
                                                  -chromosome =&gt; 'X',
                                                  -numbering =&gt; 'coding'
                                                  );
         # get a DNAMutation object somehow
         $seqDiff-&gt;add_Variant($dnamut);
         print  $seqDiff-&gt;sys_name(), "\n";

</pre><h4><b>DESCRIPTION</b></h4><pre>
       SeqDiff stores Bio::Variation::VariantI object references and descriptive information common to all
       changes in a sequence. Mutations are understood to be any kind of sequence markers and are expected to
       occur in the same chromosome. See Bio::Variation::VariantI for details.

       The methods of SeqDiff are geared towards describing mutations in human genes using gene-based coordinate
       system where 'A' of the initiator codon has number 1 and the one before it -1. This is according to
       conventions of human genetics.

       There will be class Bio::Variation::Genotype to describe markers in different chromosomes and diploid
       genototypes.

       Classes implementing Bio::Variation::VariantI interface are Bio::Variation::DNAMutation,
       Bio::Variation::RNAChange, and Bio::Variation::AAChange. See Bio::Variation::VariantI,
       Bio::Variation::DNAMutation, Bio::Variation::RNAChange, and Bio::Variation::AAChange for more
       information.

       Variant objects can be added using two ways: an array passed to the constructor or as individual Variant
       objects with add_Variant method.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing lists  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Heikki</b> <b>Lehvaslaiho</b></h4><pre>
       Email:  heikki-at-bioperl-dot-org

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
       Eckhard Lehmann, <a href="mailto:ecky@e-lehmann.de">ecky@e-lehmann.de</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>new</b>
         Title   : new
         Usage   : $seqDiff = Bio::Variation::SeqDiff-&gt;new;
         Function: generates a new Bio::Variation::SeqDiff
         Returns : reference to a new object of class SeqDiff
         Args    :

   <b>id</b>
        Title   : id
        Usage   : $obj-&gt;id(H0001); $id = $obj-&gt;id();
        Function:

                  Sets or returns the id of the seqDiff.
                  Should be used to give the collection of variants a UID
                  without semantic associations.

        Example :
        Returns : value of id, a scalar
        Args    : newvalue (optional)

   <b>sysname</b>
        Title   : sysname
        Usage   : $obj-&gt;sysname('5C&gt;G'); $sysname = $obj-&gt;sysname();
        Function:

                  Sets or returns the systematic name of the seqDiff.  The
                  name should follow the HUGO Mutation Database Initiative
                  approved nomenclature. If called without first setting the
                  value, will generate it from L&lt;Bio::Variation::DNAMutation&gt;
                  objects attached.

        Example :
        Returns : value of sysname, a scalar
        Args    : newvalue (optional)

   <b>trivname</b>
        Title   : trivname
        Usage   : $obj-&gt;trivname('[A2G;T56G]'); $trivname = $obj-&gt;trivname();
        Function:

                  Sets or returns the trivial name of the seqDiff.
                  The name should follow the HUGO Mutation Database Initiative
                  approved nomenclature. If called without first setting the
                  value, will generate it from L&lt;Bio::Variation::AAChange&gt;
                  objects attached.

        Example :
        Returns : value of trivname, a scalar
        Args    : newvalue (optional)

   <b>chromosome</b>
        Title   : chromosome
        Usage   : $obj-&gt;chromosome('X'); $chromosome = $obj-&gt;chromosome();
        Function:

                  Sets or returns the chromosome ("linkage group") of the seqDiff.

        Example :
        Returns : value of chromosome, a scalar
        Args    : newvalue (optional)

   <b>gene_symbol</b>
        Title   : gene_symbol
        Usage   : $obj-&gt;gene_symbol('FOS'); $gene_symbol = $obj-&gt;gene_symbol;
        Function:

                  Sets or returns the gene symbol for the studied CDS.

        Example :
        Returns : value of gene_symbol, a scalar
        Args    : newvalue (optional)

   <b>description</b>
        Title   : description
        Usage   : $obj-&gt;description('short description'); $descr = $obj-&gt;description();
        Function:

                  Sets or returns the short description of the seqDiff.

        Example :
        Returns : value of description, a scalar
        Args    : newvalue (optional)

   <b>alphabet</b>
        Title   : alphabet
        Usage   : if( $obj-&gt;alphabet eq 'dna' ) { /Do Something/ }
        Function: Returns the type of primary reference sequence being one of
                  'dna', 'rna' or 'protein'. This is case sensitive.

        Returns : a string either 'dna','rna','protein'.
        Args    : none

   <b>numbering</b>
        Title   : numbering
        Usage   : $obj-&gt;numbering('coding'); $numbering = $obj-&gt;numbering();
        Function:

                  Sets or returns the string giving the numbering schema used
                  to describe the variants.

        Example :
        Returns : value of numbering, a scalar
        Args    : newvalue (optional)

   <b>offset</b>
        Title   : offset
        Usage   : $obj-&gt;<a href="../man124/offset.124.html">offset</a>(124); $offset = $obj-&gt;offset();
        Function:

                  Sets or returns the offset from the beginning of the DNA sequence
                  to the coordinate start used to describe variants. Typically
                  the beginning of the coding region of the gene.
                  The cds_start should be 1 + offset.

        Example :
        Returns : value of offset, a scalar
        Args    : newvalue (optional)

   <b>cds_start</b>
        Title   : cds_start
        Usage   : $obj-&gt;<a href="../man123/cds_start.123.html">cds_start</a>(123); $cds_start = $obj-&gt;cds_start();
        Function:

                  Sets or returns the cds_start from the beginning of the DNA
                  sequence to the coordinate start used to describe
                  variants. Typically the beginning of the coding region of
                  the gene. Needs to be and is implemented as 1 + offset.

        Example :
        Returns : value of cds_start, a scalar
        Args    : newvalue (optional)

   <b>cds_end</b>
        Title   : cds_end
        Usage   : $obj-&gt;<a href="../man321/cds_end.321.html">cds_end</a>(321); $cds_end = $obj-&gt;cds_end();
        Function:

                  Sets or returns the position of the last nucleotitide of the
                  termination codon. The coordinate system starts from cds_start.

        Example :
        Returns : value of cds_end, a scalar
        Args    : newvalue (optional)

   <b>rna_offset</b>
        Title   : rna_offset
        Usage   : $obj-&gt;<a href="../man124/rna_offset.124.html">rna_offset</a>(124); $rna_offset = $obj-&gt;rna_offset();
        Function:

                  Sets or returns the rna_offset from the beginning of the RNA sequence
                  to the coordinate start used to describe variants. Typically
                  the beginning of the coding region of the gene.

        Example :
        Returns : value of rna_offset, a scalar
        Args    : newvalue (optional)

   <b>rna_id</b>
        Title   : rna_id
        Usage   : $obj-&gt;rna_id('transcript#3'); $rna_id = $obj-&gt;rna_id();
        Function:

                   Sets or returns the ID for original RNA sequence of the seqDiff.

        Example :
        Returns : value of rna_id, a scalar
        Args    : newvalue (optional)

   <b>add_Variant</b>
        Title   : add_Variant
        Usage   : $obj-&gt;add_Variant($variant)
        Function:

                  Pushes one Bio::Variation::Variant into the list of variants.
                  At the same time, creates a link from the Variant to SeqDiff
                  using its SeqDiff method.

        Example :
        Returns : 1 when succeeds, 0 for failure.
        Args    : Variant object

   <b>each_Variant</b>
        Title   : each_Variant
        Usage   : $obj-&gt;each_Variant();
        Function:

                   Returns a list of Variants.

        Example :
        Returns : list of Variants
        Args    : none

   <b>add_Gene</b>
        Title   : add_Gene
        Usage   : $obj-&gt;add_Gene($gene)
        Function:

                  Pushes one L&lt;Bio::LiveSeq::Gene&gt; into the list of genes.

        Example :
        Returns : 1 when succeeds, 0 for failure.
        Args    : Bio::LiveSeq::Gene object

       See Bio::LiveSeq::Gene for more information.

   <b>each_Gene</b>
        Title   : each_Gene
        Usage   : $obj-&gt;each_Gene();
        Function:

                   Returns a list of L&lt;Bio::LiveSeq::Gene&gt;s.

        Example :
        Returns : list of Genes
        Args    : none

   <b>dna_ori</b>
        Title   : dna_ori
        Usage   : $obj-&gt;dna_ori('atgctgctgctgct'); $dna_ori = $obj-&gt;dna_ori();
        Function:

                   Sets or returns the original DNA sequence string of the seqDiff.

        Example :
        Returns : value of dna_ori, a scalar
        Args    : newvalue (optional)

   <b>dna_mut</b>
        Title   : dna_mut
        Usage   : $obj-&gt;dna_mut('atgctggtgctgct'); $dna_mut = $obj-&gt;dna_mut();
        Function:

                   Sets or returns the mutated DNA sequence of the seqDiff.
                   If sequence has not been set generates it from the
                   original sequence and DNA mutations.

        Example :
        Returns : value of dna_mut, a scalar
        Args    : newvalue (optional)

   <b>rna_ori</b>
        Title   : rna_ori
        Usage   : $obj-&gt;rna_ori('atgctgctgctgct'); $rna_ori = $obj-&gt;rna_ori();
        Function:

                   Sets or returns the original RNA sequence of the seqDiff.

        Example :
        Returns : value of rna_ori, a scalar
        Args    : newvalue (optional)

   <b>rna_mut</b>
        Title   : rna_mut
        Usage   : $obj-&gt;rna_mut('atgctggtgctgct'); $rna_mut = $obj-&gt;rna_mut();
        Function:

                   Sets or returns the mutated RNA sequence of the seqDiff.

        Example :
        Returns : value of rna_mut, a scalar
        Args    : newvalue (optional)

   <b>aa_ori</b>
        Title   : aa_ori
        Usage   : $obj-&gt;aa_ori('MAGVLL*'); $aa_ori = $obj-&gt;aa_ori();
        Function:

                   Sets or returns the original protein sequence of the seqDiff.

        Example :
        Returns : value of aa_ori, a scalar
        Args    : newvalue (optional)

   <b>aa_mut</b>
        Title   : aa_mut
        Usage   : $obj-&gt;aa_mut('MA*'); $aa_mut = $obj-&gt;aa_mut();
        Function:

                   Sets or returns the mutated protein sequence of the seqDiff.

        Example :
        Returns : value of aa_mut, a scalar
        Args    : newvalue (optional)

   <b>seqobj</b>
        Title   : seqobj
        Usage   : $dnaobj = $obj-&gt;seqobj('dna_mut');
        Function:

                   Returns the any original or mutated sequences as a
                   Bio::PrimarySeq object.

        Example :
        Returns : Bio::PrimarySeq object for the requested sequence
        Args    : string, method name for the sequence requested

       See Bio::PrimarySeq for more information.

   <b>alignment</b>
        Title   : alignment
        Usage   : $obj-&gt;alignment
        Function:

                  Returns a pretty RNA/AA sequence alignment from linked
                  objects.  Under construction: Only simple coding region
                  point mutations work.

        Example :
        Returns :
        Args    : none

perl v5.32.1                                       2021-12-02                       <u>Bio::Variation::<a href="../man3pm/SeqDiff.3pm.html">SeqDiff</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>