<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>run_discoSnpRad.sh - pipelining kissnp2 and kissreads and clustering per locus for calling SNPs and small</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/discosnp">discosnp_2.6.2-4build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       run_discoSnpRad.sh - pipelining kissnp2 and kissreads and clustering per locus for calling SNPs and small
       indels from RAD-seq data without the need of a reference genome

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>run_discoSnpRad.sh</b> <u>-r</u> <u>read_file_of_files</u> [<u>OPTIONS</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       run_discoSnpRad.sh,  pipelining kissnp2 and kissreads and clustering per locus for calling SNPs and small
       indels from RAD-seq data without the need of a reference genome Version 2.3.X

              MANDATORY:

       <b>-r</b> read_file_of_files
              Example: <b>-r</b> bank.fof with bank.fof containing the two lines

       data_sample/reads_sequence1.fasta
              data_sample/reads_sequence2.fasta.gz

       <b>-S</b> **absolute** path to short_read_connector directory, containing the short_read_connector.sh'' file.
              <b>-Note1</b>: short read connector must be compiled.  <b>-Note2</b>: if this  option  is  missing,  discoSnpRad
              will  still  however  provide  a fasta file containing SNPs and INDELS, that won't be clustered by
              locus

              DISCOSNPRAD OPTIONS:

              <b>-g</b>: reuse a previously created graph (.h5 file) with same prefix and same k and c parameters.   <b>-b</b>
              value.

       1: (smart branching) forbid SNPs for which the two paths are branching (e.g. the two paths can be created
       either with a 'A' or a 'C' at the same position Default value
              2: No limitation on branching (lowers the precision, high recall)

              <b>-s</b>  value.  In  b2  mode  only: maximal number of symmetrical croasroads traversed while trying to
              close a bubble. Default: no limit <b>-D</b> value. discoSnpRad will search for deletions of size  from  1
              to  D  included. Default=100 <b>-a</b> value. Maximal size of ambiguity of INDELs. INDELS whose ambiguity
              is higher than this value are not output  [default '20'] <b>-P</b> value. discoSnpRad will search up to P
              SNPs in a unique bubble. Default=5  <b>-p</b>  prefix.  All  out  files  will  start  with  this  prefix.
              Default="discoRad"  <b>-l</b>: remove low complexity bubbles <b>-k</b> value. Set the length of used kmers. Must
              fit the compiled value. Default=31 <b>-c</b> value. Set the  minimal  coverage  per  read  set:  Used  by
              kissnp2  (don't  use  kmers  with  lower  coverage) and kissreads (read coherency threshold). This
              coverage can be  automatically  detected  per  read  set  or  specified  per  read  set,  see  the
              documentation.  Default=auto <b>-C</b> value. Set the maximal coverage for each read set: Used by kissnp2
              (don't use kmers with higher coverage). Default=2^31-1 <b>-d</b> value.  Set  the  number  of  authorized
              substitutions used while mapping reads on found SNPs (kissreads). Default=1 <b>-u</b>: max number of used
              threads <b>-v</b>: verbose 0 (avoids progress output) or 1 (enables progress output) <b>--</b> default=1.

              <b>-h</b>: Prints this message and exist

       Any   further   question:   read   the   readme   file   or   contact   us   via   the   Biostar   forum:
       https://www.biostars.org/t/discosnp/

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

run_discoSnpRad.sh 2.3.0                          January 2019                             <u><a href="../man1/RUN_DISCOSNPRAD.SH.1.html">RUN_DISCOSNPRAD.SH</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>