<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::DB::SeqFeature::Store::Loader -- Loader</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-db-seqfeature-perl">libbio-db-seqfeature-perl_1.7.5-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::DB::SeqFeature::Store::Loader -- Loader

</pre><h4><b>SYNOPSIS</b></h4><pre>
        # non-instantiable base class

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This is the base class for Bio::DB::SeqFeature::Loader::GFF3Loader,
       Bio::DB::SeqFeature::Loader::GFFLoader, and Bio::DB::SeqFeature::FeatureFileLoader. Please see the manual
       pages for these modules.

   <b>new</b>
        Title   : new
        Usage   : $loader = Bio::DB::SeqFeature::Store::GFF3Loader-&gt;new(@options)
        Function: create a new parser
        Returns : a Bio::DB::SeqFeature::Store::GFF3Loader gff3 parser and loader
        Args    : several - see below
        Status  : public

       This method creates a new GFF3 loader and establishes its connection with a Bio::DB::SeqFeature::Store
       database. Arguments are -name=&gt;$value pairs as described in this table:

        Name               Value
        ----               -----

        -store             A writable Bio::DB::SeqFeature::Store database handle.

        -seqfeature_class  The name of the type of Bio::SeqFeatureI object to create
                             and store in the database (Bio::DB::SeqFeature by default)

        -sf_class          A shorter alias for -seqfeature_class

        -verbose           Send progress information to standard error.

        -fast              If true, activate fast loading (see below)

        -chunk_size        Set the storage chunk size for nucleotide/protein sequences
                              (default 2000 bytes)

        -tmp               Indicate a temporary directory to use when loading non-normalized
                              features.

        -map_coords        A code ref that will transform a list of ($ref,[$start1,$end1]...)
                              coordinates into a list of ($newref,[$newstart1,$newend1]...)

        -index_subfeatures Indicate true if subfeatures should be indexed. Default is true.

        -summary_stats     Rebuild summary stats at the end of loading (not incremental,
                            so takes a long time)

       When you call <b>new()</b>, a connection to a Bio::DB::SeqFeature::Store database should already have been
       established and the database initialized (if appropriate).

       Some combinations of Bio::SeqFeatures and Bio::DB::SeqFeature::Store databases support a fast loading
       mode. Currently the only reliable implementation of fast loading is the combination of DBI::mysql with
       Bio::DB::SeqFeature. The other important restriction on fast loading is the requirement that a feature
       that contains subfeatures must occur in the GFF3 file before any of its subfeatures. Otherwise the
       subfeatures that occurred before the parent feature will not be attached to the parent correctly. This
       restriction does not apply to normal (slow) loading.

       If you use an unnormalized feature class, such as Bio::SeqFeature::Generic, then the loader needs to
       create a temporary database in which to cache features until all their parts and subparts have been seen.
       This temporary databases uses the "berkeleydb" adaptor. The -tmp option specifies the directory in which
       that database will be created. If not present, it defaults to the system default tmp directory specified
       by File::Spec-&gt;<b>tmpdir()</b>.

       The -chunk_size option allows you to tune the representation of DNA/Protein sequence in the Store
       database. By default, sequences are split into 2000 base/residue chunks and then reassembled as needed.
       This avoids the problem of pulling a whole chromosome into memory in order to fetch a short subsequence
       from somewhere in the middle. Depending on your usage patterns, you may wish to tune this parameter using
       a chunk size that is larger or smaller than the default.

   <b>load</b>
        Title   : load
        Usage   : $count = $loader-&gt;load(@ARGV)
        Function: load the indicated files or filehandles
        Returns : number of feature lines loaded
        Args    : list of files or filehandles
        Status  : public

       Once the loader is created, invoke its <b>load()</b> method with a list of GFF3 or FASTA file paths or
       previously-opened filehandles in order to load them into the database. Compressed files ending with .gz,
       .Z and .bz2 are automatically recognized and uncompressed on the fly. Paths beginning with http: or ftp:
       are treated as URLs and opened using the LWP GET program (which must be on your path).

       FASTA files are recognized by their initial "&gt;" character. Do not feed the loader a file that is neither
       GFF3 nor FASTA; I don't know what will happen, but it will probably not be what you expect.

   <b>accessors</b>
       The following read-only accessors return values passed or created during <b>new()</b>:

        store()          the long-term Bio::DB::SeqFeature::Store object

        tmp_store()      the temporary Bio::DB::SeqFeature::Store object used
                           during loading

        sfclass()        the Bio::SeqFeatureI class

        fast()           whether fast loading is active

        seq_chunk_size() the sequence chunk size

        verbose()        verbose progress messages

   <b>Internal</b> <b>Methods</b>
       The following methods are used internally and may be overridden by subclasses.

       default_seqfeature_class
             $class = $loader-&gt;default_seqfeature_class

           Return the default SeqFeatureI class (Bio::DB::SeqFeature).

       subfeatures_normalized
             $flag = $loader-&gt;subfeatures_normalized([$new_flag])

           Get  or  set  a  flag  that  indicates  that the subfeatures are normalized. This is deduced from the
           SeqFeature class information.

       subfeatures_in_table
             $flag = $loader-&gt;subfeatures_in_table([$new_flag])

           Get or set a flag that indicates that feature/subfeature relationships are stored in a table. This is
           deduced from the SeqFeature class and Store information.

       load_fh
             $count = $loader-&gt;load_fh($filehandle)

           Load the GFF3 data at the other end of the filehandle and  return  true  if  successful.  Internally,
           <b>load_fh()</b> invokes:

             start_load();
             do_load($filehandle);
             finish_load();

       start_load, finish_load
           These methods are called at the start and end of a filehandle load.

       build_summary
             $loader-&gt;build_summary

           Call  this to rebuild the summary coverage statistics. This is done automatically if <b>new()</b> was passed
           a true value for -summary_stats at create time.

       do_load
             $count = $loader-&gt;do_load($fh)

           This is called by <b>load_fh()</b> to load the GFF3 file's filehandle and return the number of lines loaded.

       load_line
               $loader-&gt;load_line($data);

           Load a line of a GFF3 file. You must bracket this with calls to <b>start_load()</b> and <b>finish_load()</b>!

               $loader-&gt;start_load();
               $loader-&gt;load_line($_) while &lt;FH&gt;;
               $loader-&gt;finish_load();

       handle_feature
             $loader-&gt;handle_feature($data_line)

           This method is called to process a single  data  line.  It  manipulates  information  stored  a  data
           structure called $self-&gt;{load_data}.

       handle_meta
             $loader-&gt;handle_meta($data_line)

           This  method  is  called  to  process  a  single  data line. It manipulates information stored a data
           structure called $self-&gt;{load_data}.

       store_current_feature
             $loader-&gt;store_current_feature()

           This method is called to store the currently active feature in the database. It uses a data structure
           stored in $self-&gt;{load_data}.

       parse_attributes
            ($reserved,$unreserved) = $loader-&gt;parse_attributes($attribute_line)

           This method parses the information contained in the $attribute_line into two hashrefs, one containing
           the values of reserved attribute tags (e.g. ID) and the other containing  the  values  of  unreserved
           ones.

       start_or_finish_sequence
             $loader-&gt;start_or_finish_sequence('Chr9')

           This method is called at the beginning and end of a fasta section.

       load_sequence
             $loader-&gt;load_sequence('gatttcccaaa')

           This  method  is  called  to  load  some amount of sequence after <b>start_or_finish_sequence()</b> is first
           called.

       open_fh
            my $io_file = $loader-&gt;open_fh($filehandle_or_path)

           This method opens up the indicated file or pipe, using some  intelligence  to  recognized  compressed
           files and URLs and doing the right thing.

       loaded_ids
            my $ids    = $loader-&gt;loaded_ids;
            my $id_cnt = @$ids;

           After  performing  a load, this returns an array ref containing all the feature primary ids that were
           created during the load.

       local_ids
            my $ids    = $self-&gt;local_ids;
            my $id_cnt = @$ids;

           After performing a load, this returns an array ref  containing  all  the  load  file  IDs  that  were
           contained within the file just loaded.

       time
            my $time = $loader-&gt;time

           This method returns the current time in seconds, using Time::HiRes if available.

       unescape
            my $unescaped = GFF3Loader::unescape($escaped)

           This  is  an internal utility.  It is the same as CGI::Util::unescape, but doesn't change pluses into
           spaces and ignores unicode escapes.

</pre><h4><b>BUGS</b></h4><pre>
       This is an early version, so there are certainly some bugs. Please use the BioPerl bug tracking system to
       report bugs.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       bioperl,                    Bio::DB::SeqFeature::Store,                     Bio::DB::SeqFeature::Segment,
       Bio::DB::SeqFeature::NormalizedFeature,                           Bio::DB::SeqFeature::Store::GFF3Loader,
       Bio::DB::SeqFeature::Store::DBI::mysql, Bio::DB::SeqFeature::Store::berkeleydb

</pre><h4><b>AUTHOR</b></h4><pre>
       Lincoln Stein &lt;<a href="mailto:lstein@cshl.org">lstein@cshl.org</a>&gt;.

       Copyright (c) 2006 Cold Spring Harbor Laboratory.

       This library is free software; you can redistribute it and/or modify it under  the  same  terms  as  Perl
       itself.

perl v5.40.0                                       2024-10-11             <u>Bio::DB::SeqFea...::Store::<a href="../man3pm/Loader.3pm.html">Loader</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>