<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>aragorn - detect tRNA genes in nucleotide sequences</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/aragorn">aragorn_1.2.41-3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       aragorn - detect tRNA genes in nucleotide sequences

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>aragorn</b> [<u>OPTION</u>]... <u>FILE</u>

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-m</b>
           Search for tmRNA genes.

       <b>-t</b>
           Search for tRNA genes. By default, all are detected. If one of <b>-m</b> or <b>-t</b> is specified, then the other
           is not detected unless specified as well.

       <b>-mt</b>
           Search for Metazoan mitochondrial tRNA genes. tRNA genes with introns not detected.  <b>-i</b>, <b>-sr</b> switches
           ignored. Composite Metazoan mitochondrial genetic code used.

       <b>-mtmam</b>
           Search for Mammalian mitochondrial tRNA genes.  <b>-i</b>, <b>-sr</b> switches ignored.  <b>-tv</b> switch set. Mammalian
           mitochondrial genetic code used.

       <b>-mtx</b>
           Same as <b>-mt</b> but low scoring tRNA genes are not reported.

       <b>-mtd</b>
           Overlapping metazoan mitochondrial tRNA genes on opposite strands are reported.

       <b>-gc</b>[<u>num</u>]
           Use the GenBank transl_table = [<u>num</u>] genetic code. Individual modifications can be appended using
           <u>,BBB</u>=&lt;aa&gt; B = A,C,G, or T. &lt;aa&gt; is the three letter code for an amino-acid. More than one
           modification can be specified. eg <b>-gcvert</b>,aga=Trp,agg=Trp uses the Vertebrate Mitochondrial code and
           the codons AGA and AGG changed to Tryptophan.

       <b>-gcstd</b>
           Use standard genetic code.

       <b>-gcmet</b>
           Use composite Metazoan mitochondrial genetic code.

       <b>-gcvert</b>
           Use Vertebrate mitochondrial genetic code.

       <b>-gcinvert</b>
           Use Invertebrate mitochondrial genetic code.

       <b>-gcyeast</b>
           Use Yeast mitochondrial genetic code.

       <b>-gcprot</b>
           Use Mold/Protozoan/Coelenterate mitochondrial genetic code.

       <b>-gcciliate</b>
           Use Ciliate genetic code.

       <b>-gcflatworm</b>
           Use Echinoderm/Flatworm mitochondrial genetic code

       <b>-gceuplot</b>
           Use Euplotid genetic code.

       <b>-gcbact</b>
           Use Bacterial/Plant Chloroplast genetic code.

       <b>-gcaltyeast</b>
           Use alternative Yeast genetic code.

       <b>-gcascid</b>
           Use Ascidian Mitochondrial genetic code.

       <b>-gcaltflat</b>
           Use alternative Flatworm Mitochondrial genetic code.

       <b>-gcblep</b>
           Use Blepharisma genetic code.

       <b>-gcchloroph</b>
           Use Chlorophycean Mitochondrial genetic code.

       <b>-gctrem</b>
           Use Trematode Mitochondrial genetic code.

       <b>-gcscen</b>
           Use Scenedesmus obliquus Mitochondrial genetic code.

       <b>-gcthraust</b>
           Use Thraustochytrium Mitochondrial genetic code.

       <b>-tv</b>
           Do not search for mitochondrial TV replacement loop tRNA genes. Only relevant if <b>-mt</b> used.

       <b>-c7</b>
           Search for tRNA genes with 7 base C-loops only.

       <b>-i</b>
           Search for tRNA genes with introns in anticodon loop with maximum length 3000 bases. Minimum intron
           length is 0 bases. Ignored if <b>-m</b> is specified.

       <b>-i</b>[<u>max</u>]
           Search for tRNA genes with introns in anticodon loop with maximum length [<u>max</u>] bases. Minimum intron
           length is 0 bases. Ignored if <b>-m</b> is specified.

       <b>-i</b>[<u>min</u>],[<u>max</u>]
           Search for tRNA genes with introns in anticodon loop with maximum length [<u>max</u>] bases, and minimum
           length [<u>min</u>] bases. Ignored if <b>-m</b> is specified.

       <b>-io</b>
           Same as <b>-i</b>, but allow tRNA genes with long introns to overlap shorter tRNA genes.

       <b>-if</b>
           Same as <b>-i</b>, but fix intron between positions 37 and 38 on C-loop (one base after anticodon).

       <b>-ifo</b>
           Same as <b>-if</b> and <b>-io</b> combined.

       <b>-ir</b>
           Same as <b>-i</b>, but report tRNA genes with minimum length [<u>min</u>] bases rather than search for tRNA genes
           with minimum length [<u>min</u>] bases. With this switch, [<u>min</u>] acts as an output filter, minimum intron
           length for searching is still 0 bases.

       <b>-c</b>
           Assume that each sequence has a circular topology. Search wraps around each end. Default setting.

       <b>-l</b>
           Assume that each sequence has a linear topology. Search does not wrap.

       <b>-d</b>
           Double. Search both strands of each sequence. Default setting.

       <b>-s</b> or <b>-s+</b>
           Single. Do not search the complementary (antisense) strand of each sequence.

       <b>-sc</b> or <b>-s-</b>
           Single complementary. Do not search the sense strand of each sequence.

       <b>-ps</b>
           Lower scoring thresholds to 95% of default levels.

       <b>-ps</b>[<u>num</u>]
           Change scoring thresholds to [<u>num</u>] percent of default levels.

       <b>-rp</b>
           Flag possible pseudogenes (score &lt; 100 or tRNA anticodon loop &lt;&gt; 7 bases long). Note that genes with
           score &lt; 100 will not be detected or flagged if scoring thresholds are not also changed to below 100%
           (see -ps switch).

       <b>-seq</b>
           Print out primary sequence.

       <b>-br</b>
           Show secondary structure of tRNA gene primary sequence using round brackets.

       <b>-fasta</b>
           Print out primary sequence in fasta format.

       <b>-fo</b>
           Print out primary sequence in fasta format only (no secondary structure).

       <b>-fon</b>
           Same as <b>-fo</b>, with sequence and gene numbering in header.

       <b>-fos</b>
           Same as <b>-fo</b>, with no spaces in header.

       <b>-fons</b>
           Same as <b>-fo</b>, with sequence and gene numbering, but no spaces.

       <b>-w</b>
           Print out in Batch mode.

       <b>-ss</b>
           Use the stricter canonical 1-2 bp spacer1 and 1 bp spacer2. Ignored if <b>-mt</b> set. Default is to allow 3
           bp spacer1 and 0-2 bp spacer2, which may degrade selectivity.

       <b>-v</b>
           Verbose. Prints out information during search to STDERR.

       <b>-a</b>
           Print out tRNA domain for tmRNA genes.

       <b>-a7</b>
           Restrict tRNA astem length to a maximum of 7 bases

       <b>-aa</b>
           Display message if predicted iso-acceptor species does not match species in sequence name (if
           present).

       <b>-j</b>
           Display 4-base sequence on 3' end of astem regardless of predicted amino-acyl acceptor length.

       <b>-jr</b>
           Allow some divergence of 3' amino-acyl acceptor sequence from NCCA.

       <b>-jr4</b>
           Allow some divergence of 3' amino-acyl acceptor sequence from NCCA, and display 4 bases.

       <b>-q</b>
           Do not print configuration line (which switches and files were used).

       <b>-rn</b>
           Repeat sequence name before summary information.

       <b>-O</b> [<u>outfile</u>]
           Print output to <u>.</u> <u>If</u> <u>['outfile</u>] already exists, it is overwritten. By default all output goes to
           stdout.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       aragorn detects tRNA, mtRNA, and tmRNA genes. A minimum requirement is at least a 32 bit compiler
       architecture (variable types int and unsigned int are at least 4 bytes long).

       [<u>FILE</u>] is assumed to contain one or more sequences in FASTA format. Results of the search are printed to
       STDOUT. All switches are optional and case-insensitive. Unless -i is specified, tRNA genes containing
       introns are not detected.

</pre><h4><b>AUTHORS</b></h4><pre>
       Bjorn Canback &lt;<a href="mailto:bcanback@acgt.se">bcanback@acgt.se</a>&gt;, Dean Laslett &lt;<a href="mailto:gaiaquark@gmail.com">gaiaquark@gmail.com</a>&gt;

</pre><h4><b>REFERENCES</b></h4><pre>
       Laslett, D. and Canback, B. (2004) ARAGORN, a program for the detection of transfer RNA and
       transfer-messenger RNA genes in nucleotide sequences Nucleic Acids Research, 32;11-16

       Laslett, D. and Canback, B. (2008) ARWEN: a program to detect tRNA genes in metazoan mitochondrial
       nucleotide sequences Bioinformatics, <a href="../man2/24.2.html">24</a>(2); 172-175.

                                                   02/24/2013                                         <u><a href="../man1/ARAGORN.1.html">ARAGORN</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>