<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>cuteSV - prediction of structural variants from sequence alignments</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/cutesv">cutesv_2.1.1-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       cuteSV - prediction of structural variants from sequence alignments

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: cuteSV [-h] [--version] [-t THREADS] [-b BATCHES] [-S SAMPLE]

              [--retain_work_dir]  [--report_readid]  [-p  MAX_SPLIT_PARTS] [-q MIN_MAPQ] [-r MIN_READ_LEN] [-md
              MERGE_DEL_THRESHOLD] [-mi MERGE_INS_THRESHOLD] [-s MIN_SUPPORT] [-l MIN_SIZE] [-L  MAX_SIZE]  [-sl
              MIN_SIGLENGTH]   [--genotype]   [--gt_round   GT_ROUND]   [-Ivcf   IVCF]   [--max_cluster_bias_INS
              MAX_CLUSTER_BIAS_INS]  [--diff_ratio_merging_INS  DIFF_RATIO_MERGING_INS]  [--max_cluster_bias_DEL
              MAX_CLUSTER_BIAS_DEL]  [--diff_ratio_merging_DEL  DIFF_RATIO_MERGING_DEL]  [--max_cluster_bias_INV
              MAX_CLUSTER_BIAS_INV]   [--max_cluster_bias_DUP   MAX_CLUSTER_BIAS_DUP]    [--max_cluster_bias_TRA
              MAX_CLUSTER_BIAS_TRA] [--diff_ratio_filtering_TRA DIFF_RATIO_FILTERING_TRA] [BAM] reference output
              work_dir

              Current version: v1.0.11 Author: Tao Jiang Contact: <a href="mailto:tjiang@hit.edu.cn">tjiang@hit.edu.cn</a>

              If you use cuteSV in your work, please cite:

              Jiang  T  et al. Long-read-based human genomic structural variation detection with cuteSV.  Genome
              Biol 21,<a href="../man2020/189.2020.html">189</a>(2020). https://doi.org/10.1186/s13059-020-02107-y

              Suggestions:

              For PacBio CLR data:

       <b>--max_cluster_bias_INS</b>
              100

       <b>--diff_ratio_merging_INS</b>
              0.3

       <b>--max_cluster_bias_DEL</b>
              200

       <b>--diff_ratio_merging_DEL</b>
              0.5

              For PacBio CCS(HIFI) data:

       <b>--max_cluster_bias_INS</b>
              1000

       <b>--diff_ratio_merging_INS</b>
              0.9

       <b>--max_cluster_bias_DEL</b>
              1000

       <b>--diff_ratio_merging_DEL</b>
              0.5

              For ONT data:

       <b>--max_cluster_bias_INS</b>
              100

       <b>--diff_ratio_merging_INS</b>
              0.3

       <b>--max_cluster_bias_DEL</b>
              100

       <b>--diff_ratio_merging_DEL</b>
              0.3

   <b>positional</b> <b>arguments:</b>
       [BAM]  Sorted .bam file form NGMLR or Minimap2.

       reference
              The reference genome in fasta format.

       output Output VCF format file.

       work_dir
              Work-directory for distributed jobs

   <b>optional</b> <b>arguments:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>--version</b>, <b>-v</b>
              show program's version number and exit

       <b>-t</b> THREADS, <b>--threads</b> THREADS
              Number of threads to use.[16]

       <b>-b</b> BATCHES, <b>--batches</b> BATCHES
              Batch of genome segmentation interval.[10000000]

       <b>-S</b> SAMPLE, <b>--sample</b> SAMPLE
              Sample name/id

       <b>--retain_work_dir</b>
              Enable to retain temporary folder and files.

       <b>--report_readid</b>
              Enable to report supporting read ids for each SV.

   <b>Collection</b> <b>of</b> <b>SV</b> <b>signatures:</b>
       <b>-p</b> MAX_SPLIT_PARTS, <b>--max_split_parts</b> MAX_SPLIT_PARTS
              Maximum number of split segments a read may be aligned before it is ignored.  All  split  segments
              are considered when using <b>-1</b>. (Recommend <b>-1</b> when applying assembly-based alignment.)[7]

       <b>-q</b> MIN_MAPQ, <b>--min_mapq</b> MIN_MAPQ
              Minimum mapping quality value of alignment to be taken into account.[20]

       <b>-r</b> MIN_READ_LEN, <b>--min_read_len</b> MIN_READ_LEN
              Ignores reads that only report alignments with not longer than bp.[500]

       <b>-md</b> MERGE_DEL_THRESHOLD, <b>--merge_del_threshold</b> MERGE_DEL_THRESHOLD
              Maximum  distance  of deletion signals to be merged. In our paper, I used <b>-md</b> 500 to process HG002
              real human sample data.[0]

       <b>-mi</b> MERGE_INS_THRESHOLD, <b>--merge_ins_threshold</b> MERGE_INS_THRESHOLD
              Maximum distance of insertion signals to be merged. In our paper, I used <b>-mi</b> 500 to process  HG002
              real human sample data.[100]

   <b>Generation</b> <b>of</b> <b>SV</b> <b>clusters:</b>
       <b>-s</b> MIN_SUPPORT, <b>--min_support</b> MIN_SUPPORT
              Minimum number of reads that support a SV to be reported.[10]

       <b>-l</b> MIN_SIZE, <b>--min_size</b> MIN_SIZE
              Minimum size of SV to be reported.[30]

       <b>-L</b> MAX_SIZE, <b>--max_size</b> MAX_SIZE
              Maximum size of SV to be reported.[100000]

       <b>-sl</b> MIN_SIGLENGTH, <b>--min_siglength</b> MIN_SIGLENGTH
              Minimum length of SV signal to be extracted.[10]

   <b>Computing</b> <b>genotypes:</b>
       <b>--genotype</b>
              Enable to generate genotypes.

       <b>--gt_round</b> GT_ROUND
              Maximum round of iteration for alignments searching if perform genotyping.[500]

   <b>Force</b> <b>calling:</b>
       <b>-Ivcf</b> IVCF
              Optional given vcf file. Enable to perform force calling. [NULL]

   <b>Advanced:</b>
       <b>--max_cluster_bias_INS</b> MAX_CLUSTER_BIAS_INS
              Maximum distance to cluster read together for insertion.[100]

       <b>--diff_ratio_merging_INS</b> DIFF_RATIO_MERGING_INS
              Do not merge breakpoints with basepair identity more than [0.3] for insertion.

       <b>--max_cluster_bias_DEL</b> MAX_CLUSTER_BIAS_DEL
              Maximum distance to cluster read together for deletion.[200]

       <b>--diff_ratio_merging_DEL</b> DIFF_RATIO_MERGING_DEL
              Do not merge breakpoints with basepair identity more than [0.5] for deletion.

       <b>--max_cluster_bias_INV</b> MAX_CLUSTER_BIAS_INV
              Maximum distance to cluster read together for inversion.[500]

       <b>--max_cluster_bias_DUP</b> MAX_CLUSTER_BIAS_DUP
              Maximum distance to cluster read together for duplication.[500]

       <b>--max_cluster_bias_TRA</b> MAX_CLUSTER_BIAS_TRA
              Maximum distance to cluster read together for translocation.[50]

       <b>--diff_ratio_filtering_TRA</b> DIFF_RATIO_FILTERING_TRA
              Filter breakpoints with basepair identity less than [0.6] for translocation.

cuteSV 1.0.11                                       July 2021                                          <u><a href="../man1/CUTESV.1.html">CUTESV</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>