<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>gt-eval - Compare annotation files and show accuracy measures (prediction vs. reference).</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/genometools">genometools_1.6.5+ds-2.2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       gt-eval - Compare annotation files and show accuracy measures (prediction vs. reference).

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>gt</b> <b>eval</b> reference_file prediction_file

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>-nuc</b> [<u>yes|no</u>]
           evaluate nucleotide level (memory consumption is proportional to the input file sizes) (default: yes)

       <b>-ltr</b> [<u>yes|no</u>]
           evaluate a LTR retrotransposon prediction instead of a gene prediction (all LTR_retrotransposon
           elements are considered to have an undetermined strand) (default: no)

       <b>-ltrdelta</b> [<u>value</u>]
           set allowed delta for LTR borders to be considered equal (default: 20)

       <b>-v</b> [<u>yes|no</u>]
           be verbose (default: no)

       <b>-o</b> [<u>filename</u>]
           redirect output to specified file (default: undefined)

       <b>-gzip</b> [<u>yes|no</u>]
           write gzip compressed output file (default: no)

       <b>-bzip2</b> [<u>yes|no</u>]
           write bzip2 compressed output file (default: no)

       <b>-force</b> [<u>yes|no</u>]
           force writing to output file (default: no)

       <b>-help</b>
           display help and exit

       <b>-version</b>
           display version information and exit

       The program shows sensitivity and specificity values for certain feature types (e.g., gene, mRNA, and
       exon). For some feature types the number of missing and wrong features of that type is also shown.
       Thereby, “missing” means the number of features of that type from the “reference” without overlap to a
       feature of that type from the “prediction”. Vice versa, “wrong” denotes the number of features of that
       type from the “prediction” without overlap to a feature of that type from the “reference”.

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       Report bugs to https://github.com/genometools/genometools/issues.

GenomeTools 1.6.5                                  04/27/2024                                         <u><a href="../man1/GT-EVAL.1.html">GT-EVAL</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>