<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>vcfcreatemulti - collates single ALT allele records into multi-allele records while tracking genotypes</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/libvcflib-tools">libvcflib-tools_1.0.12+dfsg-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       vcfcreatemulti - collates single ALT allele records into multi-allele records while tracking genotypes

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>vcfcreatemulti</b>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>vcfcreatemulti</b>  merges VCF records into one line by combining ALT alleles into a single VCF record.  This
       tool is a great companion to vcfwave.

       In 2022 <b>vcfcreatemulti</b> has been upgraded to track INFO records and genotypes (samples) so they are updat‐
       ed in the output.

       Note that the tool is not perfect: See below EXAMPLES, the caveat on `too many  variants'  <b>MULTI=ALTPROB‐</b>
       <b>LEM</b>, and vcfwave for more information.

   <b>Options</b>
       -h, –help
              shows help message and exits.

       See more below.

</pre><h4><b>EXIT</b> <b>VALUES</b></h4><pre>
       <b>0</b>      Success

       <b>not</b> <b>0</b>  Failure

</pre><h4><b>EXAMPLES</b></h4><pre>
              &gt;&gt;&gt; head("vcfcreatemulti -h",25)
              &gt;
              Usage: vcfcreatemulti [options] [file]
              &gt;
              Go through sorted VCF and when overlapping alleles are represented across multiple records, merge them into a single multi-ALT record. See the documentation for more information.
              &gt;
              options:
              &gt;
                  --quiet           no progress bar
                  --legacy          legacy mode (old C++ implementation does not do genotypes)
              &gt;
              Type: transformation
              &gt;

       The  original  `legacy' vcfcreatemulti can combine overlapping alleles onto one record (VCF line), but it
       does not correct the INFO fields and sample (genotypes).  For example:

              &gt;&gt;&gt; sh("cat ../samples/10158243-after-vcfwave.vcf|grep -v ^\#")
              grch38#chr4     10158244        &gt;3655&gt;3662_1    CCCCCACCCCCAC   C       60      .       AC=1;AF=0.011236;AN=89;AT=&gt;3655&gt;3656&gt;3657&gt;3660&gt;3662;NS=45;LV=0;ORIGIN=grch38#chr4:10158243;LEN=12;INV=0;TYPE=del        GT      0|0     0|0     0|0     0|0     1|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0
              grch38#chr4     10158244        &gt;3655&gt;3662_2    CCCCCACCCCCACC  C       60      .       AC=3;AF=0.033708;AN=89;AT=&gt;3655&gt;3656&gt;3660&gt;3662;NS=45;LV=0;ORIGIN=grch38#chr4:10158243;LEN=13;INV=0;TYPE=del     GT      0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     1|0     0|1     0|0     0|0     0|0     0|0     0|0     0|0     0|1     0|0     0
              grch38#chr4     10158245        &gt;3655&gt;3662_3    CCCCACCCCCACC   C       60      .       AC=64;AF=0.719101;AN=89;AT=&gt;3655&gt;3656&gt;3657&gt;3658&gt;3659&gt;3660&gt;3662;NS=45;LV=0;ORIGIN=grch38#chr4:10158243;LEN=12;INV=0;TYPE=del     GT      0|0     1|1     1|1     1|0     0|1     0|0     0|1     0|1     1|1     1|1     1|1     1|1     1|1     1|1     1|1     0|0     1|1     1|1     1|1     1|0     1|0     1|0     1|0     1|1     1|1     1|0     1|1     1|1     0|0     1|0     1|1     0|1     1|1     1|1     0|1     1|0     1|1     1|1     0|1     1|1     1|1     1|0     1|0     1|1     0
              grch38#chr4     10158251        &gt;3655&gt;3662_4    CCCCACC C       60      .       AC=3;AF=0.033708;AN=89;AT=&gt;3655&gt;3656&gt;3657&gt;3658&gt;3660&gt;3662;NS=45;LV=0;ORIGIN=grch38#chr4:10158243;LEN=6;INV=0;TYPE=del    GT      0|0     0|0     0|0     0|0     0|0     0|1     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     1|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|1     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0
              grch38#chr4     10158256        &gt;3655&gt;3662_5    CC      C       60      .       AC=2;AF=0.022472;AN=89;AT=&gt;3655&gt;3660&gt;3662;NS=45;LV=0;ORIGIN=grch38#chr4:10158243;LEN=1;INV=0;TYPE=del   GT      0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|1     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     1|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0
              grch38#chr4     10158257        &gt;3655&gt;3662_6    C       A       60      .       AC=1;AF=0.011236;AN=89;AT=&gt;3655&gt;3656&gt;3657&gt;3660&gt;3662;NS=45;LV=0;ORIGIN=grch38#chr4:10158243;LEN=1;INV=0;TYPE=snp GT      0|0     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     .|.     0

       this now gets converted into the following:

              &gt;&gt;&gt; sh("../build/vcfcreatemulti ../samples/10158243-after-vcfwave.vcf|grep -v ^\#")
              grch38#chr4     10158244        &gt;3655&gt;3662_1    CCCCCACCCCCACC  CC,C,CC,CCCCCACC,CCCCCACCCCCAC,CCCCCACCCCCACA   60      .       AC=1,3,64,3,2,1;AF=0.011236,0.033708,0.719101,0.033708,0.022472,0.011236;AN=89,89,89,89,89,89;AT=&gt;3655&gt;3656&gt;3657&gt;3660&gt;3662,&gt;3655&gt;3656&gt;3660&gt;3662,&gt;3655&gt;3656&gt;3657&gt;3658&gt;3659&gt;3660&gt;3662,&gt;3655&gt;3656&gt;3657&gt;3658&gt;3660&gt;3662,&gt;3655&gt;3660&gt;3662,&gt;3655&gt;3656&gt;3657&gt;3660&gt;3662;NS=45;LV=0;ORIGIN=grch38#chr4:10158243;LEN=12;INV=0,0,0,0,0,0;TYPE=del,del,del,del,del,snp;combined=10158244-10158257      GT      0|0     3|3     3|3     3|0     1|3     0|4     0|3     0|3     3|3     3|3     3|3     3|3     3|3     3|3     3|3     4|5     3|3     3|3     3|3     3|0     3|0     3|0     3|0     3|3     3|3     3|4     3|3     3|3     5|0     3|0     3|3     0|3     3|3     3|3     2|3     3|2     3|3     3|3     0|3     3|3     3|3     3|0     3|2     3|3     0

   <b>Too</b> <b>many</b> <b>variants</b>
       <u>Or</u> <u>the</u> <u>MULTI=ALTPROBLEM.</u>

       That looks proper.  There is one caveat or blatant problem, however.  If a variant sequence  is  long  (a
       large `bubble') and with the other alleles more (small) INDELs are scored than there are samples then the
       genotypes represent only the last match.  Resulting in something ugly:

              ...,del,del,snp,ins,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,s np,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,s np,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp,snp;combined=36390210-36409660 GT

              509|49 8 500|500 20|251  500|238 238|498 653|387 102|1 500|498 9|509 498|69  500|297 498|725 498|660 500|472 204|500 50 0|846 654|653 500|500 500|500 18|18 430|498 214|500 499|299 67|500  18|386  47|154  508|47  500|385 42|47 579|47 47|18 47|47 219|500 18|47 53|213  500|18  500|18  500|500 47|846  47|47 500|47  500|47  839|500 498|47  500

       this is a simple artefact resulting from the fact that complex structures do not map (easily) on the sim‐
       ple VCF layout.  Another problem is that multiple SNPs don’t get incorporated in the ALTs - the algorithm
       uses the reference to build up the longer ALTs for every single SNP from the start.

       In  this  example you see that the ALT for SNP2 does not contain SNP1 even though they may be in the same
       individual/sample:

                                     sample
              REF      ACTGACTGACTG
              ALT-SNP1 ACTGC          1/0
              ALT-SNP2 ACTGACTA       1/0
                           ^

       In words: the result is incorrect.

       At this point, for analysis, there is little else to do but go to the original data  (pangenome  or  VCF)
       and  compare  the  results.   What <b>vcfcreatemulti</b> helps to do is point out that there is a complex region
       here with ample variation and the resulting layout is a problem (too many ALTs as in `too many cooks'!).

       To help vcflib show’s a `WARNING: Too many ALT alleles to fit in sample(s)’ and we add an INFO tag  “MUL‐
       TI=ALTPROBLEM”.  Searching for these will give an idea of this issue.  E.g.

              grep MULTI= ./test/tmp/vcfcreatemulti_2.vcf -c

       Finds 3 marked records.  One of them is derived from the combination:

              grch38#chr8 36377478  &gt;601&gt;606  GTTTCTTGAAAAACCAAATGT GTTTCTTGAAAAACCAAAGGT,G 60  . AC=20,1;AF=0.224719,0.011236
              ;AN=89;AT=&gt;601&gt;602&gt;603&gt;605&gt;606,&gt;601&gt;602&gt;604&gt;605&gt;606,&gt;601&gt;606;NS=45;LV=0 GT  0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0
              0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 1|1 0|0 0|0 0|0 0|0 1|0 1|0 0|2 0|0 0|1 0|1 1|1 1|1 0|0 1|1 0|0 0|1 0|1
              0|0 1|0 1|1 0|1 0|1 0|0 0|1 0
              grch38#chr8 36377496  &gt;602&gt;605  T G 60  . AC=20;AF=0.227273;AN=88;AT=&gt;602&gt;603&gt;605,&gt;602&gt;604&gt;605;NS=45;LV=1;PS=&gt;60
              1&gt;606 GT  0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 1|1 0|0 0|0 0|0 0|0 1|0 1|
              0 0|. 0|0 0|1 0|1 1|1 1|1 0|0 1|1 0|0 0|1 0|1 0|0 1|0 1|1 0|1 0|1 0|0 0|1 0

       resulting in

              grch38#chr8 36377478  &gt;601&gt;606  GTTTCTTGAAAAACCAAATGT GTTTCTTGAAAAACCAAAGGT,G,GTTTCTTGAAAAACCAAAGGT 60  . AC=20,
              1,20;AF=0.224719,0.011236,0.227273;AN=89,89,88;AT=&gt;601&gt;602&gt;603&gt;605&gt;606,&gt;601&gt;602&gt;604&gt;605&gt;606,&gt;601&gt;602&gt;603&gt;605&gt;606
              ,&gt;601&gt;606,&gt;602&gt;603&gt;605,&gt;602&gt;604&gt;605;NS=45;LV=0;MULTI=ALTPROBLEM;combined=36377478-36377496  GT  0|0 0|0 0|0 0|0
              0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 3|3 0|0 0|0 0|0 0|0 3|0 3|0 0|2 0|0 0|3 0|3 3|3 3|3
              0|0 3|3 0|0 0|3 0|3 0|0 3|0 3|3 0|3 0|3 0|0 0|3 0

       This is a combination of:

              grch38#chr8 36377478  &gt;601&gt;606  GTTTCTTGAAAAACCAAATGT GTTTCTTGAAAAACCAAAGGT,G 60  . AC=20,1;AF=0.224719,0.011236
              ;AN=89;AT=&gt;601&gt;602&gt;603&gt;605&gt;606,&gt;601&gt;602&gt;604&gt;605&gt;606,&gt;601&gt;606;NS=45;LV=0 GT  0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0
              0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 1|1 0|0 0|0 0|0 0|0 1|0 1|0 0|2 0|0 0|1 0|1 1|1 1|1 0|0 1|1 0|0 0|1 0|1
              0|0 1|0 1|1 0|1 0|1 0|0 0|1 0
              grch38#chr8 36377496  &gt;602&gt;605  T G 60  . AC=20;AF=0.227273;AN=88;AT=&gt;602&gt;603&gt;605,&gt;602&gt;604&gt;605;NS=45;LV=1;PS=&gt;60
              1&gt;606 GT  0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 0|0 1|1 0|0 0|0 0|0 0|0 1|0 1|
              0 0|. 0|0 0|1 0|1 1|1 1|1 0|0 1|1 0|0 0|1 0|1 0|0 1|0 1|1 0|1 0|1 0|0 0|1 0

       Where the ALTs end up being a duplication and there is some overlap in the genotype calling.

       One  future  solution  might  be to have vcfcreatemulti ignore SNPs, or only take the first one, but that
       somewhat would do away with pointing out complex arrangements.  Another solution might  be  to  edit  the
       ALTs and merge ALT-SNP1 into ALT-SNP2 so we get <b>ACTGCCTA</b>.  Contributions and ideas are welcome!

       Having  a  think  about  this: the safest approach is to backtrack on a conflict and leave it alone.  So,
       when a variant comes up that conflicts with the combined record (so far)  we  should  drop  merging  that
       variant  and  leave  it  alone.   This will typically happen with a long ALT that overlaps many SNPs.  We
       could come up with all types of solutions, but the point of this algorithm is to `fix' the obvious cases.
       At this point we continue and show the MULTI=ALTPROBLEM info field.  It is not  satisfactory  and  it  is
       slow too.  We can have a stab at the backtrack in the future.

</pre><h4><b>./vcfcreatemulti</b> <b>../samples/grch38#chr8_36353854-36453166.vcf</b> <b>&gt;</b> <b>../test/data/regression/vcfcreatemulti_2.vcf</b></h4><pre>
                            run_stdout(“vcfcreatemulti    ../samples/grch38#chr8_36353854-36453166-bcftools-nor‐
                            malised.vcf”, ext=“vcf”, uniq=2) output in vcfcreatemulti_2.vcf

                            run_stdout(“vcfcreatemulti  ../samples/sample.vcf”,  ext=“vcf”,  uniq=3)  output  in
                            vcfcreatemulti_3.vcf

              Check if the legacy version is still the same. Note it only retains the first genotype and has duplicate 'CC' alt alleles. INFO fields are not correct either.

              ```python
              &gt;&gt;&gt; sh("../build/vcfcreatemulti --legacy ../samples/10158243-after-vcfwave.vcf|grep -v ^\#")
              grch38#chr4     10158244        &gt;3655&gt;3662_1    CCCCCACCCCCACC  CC,C,CC,CCCCCACC,CCCCCACCCCCAC,CCCCCACCCCCACA   60      .       AC=1;AF=0.011236;AN=89;AT=&gt;3655&gt;3656&gt;3657&gt;3660&gt;3662;NS=45;LV=0;ORIGIN=grch38#chr4:10158243;LEN=12;INV=0;TYPE=del;combined=10158244-10158257     GT      0|0     0|0     0|0     0|0     1|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0|0     0

   <b>Trouble</b> <b>shooting</b>
       If you get an error like

              thread 502 panic: attempt to unwrap error: MultiAltNotSupported

       It  means the input file already contains multi-allele VCF records.  To split these you can run a command
       such as <b>bcftools</b> <b>norm</b> <b>-m-</b> to normalise the VCF records and split out multiple ALT alleles  into  separate
       VCF records.  Finally use <b>vcfcreatemulti</b> to create multi-allele VCF records again.

   <b>Warning:</b> <b>Too</b> <b>many</b> <b>ALT</b> <b>alleles</b> <b>to</b> <b>fit</b> <b>in</b> <b>sample(s)</b>
       See `caveat' section above.

   <b>Warning:</b>  <b>This</b>  <b>code</b>  <b>only</b>  <b>supports</b>  <b>one</b>  <b>ALT</b> <b>allele</b> <b>per</b> <b>record:</b> <b>bailing</b> <b>out</b> <b>—</b> <b>try</b> <b>normalising</b> <b>the</b> <b>data</b> <b>with</b>
       <b>bcftools</b> <b>norm</b> <b>-m-</b>
       Your VCF already contains multi-allele entries - bring them back to one single ALT per record/line.

</pre><h4><b>LICENSE</b></h4><pre>
       Copyright 2022-2024 (C) Erik Garrison, Pjotr Prins and vcflib contributors.  MIT licensed.

</pre><h4><b>AUTHORS</b></h4><pre>
       Erik Garrison, Pjotr Prins and other vcflib contributors.

vcfcreatemulti (vcflib)                                                                        <u><a href="../man1/VCFCREATEMULTI.1.html">VCFCREATEMULTI</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>