<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>bamintervalcomment - sort BAM files by coordinate or query name</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/biobambam2">biobambam2_2.0.185+ds-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       bamintervalcomment - sort BAM files by coordinate or query name

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>bamintervalcomment</b> [options]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       bamintervalcomment  reads  a BAM, SAM or CRAM file and a file containing a list of named intervals, marks
       each line in the input with the list of all matching intervals and stores the resulting file in BAM,  SAM
       or  CRAM  format.   The  intervals file needs to be given using the intervals key. The file can be either
       plain or compressed using gzip. The intervals file is expected to contain one  interval  per  line.  Each
       line  is  assumed  to contain a tab separated list of values, where the following columns are used by the
       program:

       first and second column
              contain a pair of names which form the id of the interval

       third column
              gives the name of the reference sequence containing the interval

       fifth and sixth column
              give the interval on the reference sequence designated by the  third  column  as  a  pair  of  non
              negative integers. Both borders are included.

       For  each  alignment  the matching interval designators are stored in the CO (comment) auxiliary field in
       the form of a semicolon separated list, where each list element is a pair (A,B) given the two id  columns
       of    the    respective    interval.     An   example   of   an   interval   file   can   be   found   at
       <a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/refFlat.txt.gz">http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/refFlat.txt.gz</a> .

       The following key=value pairs can be given when running the program:

       <b>level=&lt;-1|0|1|9|11&gt;:</b> set compression level of the output BAM file. Valid values are

       -1:    zlib/gzip default compression level

       0:     uncompressed

       1:     zlib/gzip level 1 (fast) compression

       9:     zlib/gzip level 9 (best) compression

       If libmaus has been compiled with support for igzip (see https://software.intel.com/en-us/articles/igzip-
       a-high-performance-deflate-compressor-with-optimizations-for-genomic-data) then an additional valid value
       is

       11:    igzip compression

       <b>verbose=&lt;1&gt;:</b> Valid values are

       1:     print progress report on standard error

       0:     do not print progress report

       <b>tmpfile=&lt;filename&gt;:</b> set the prefix for temporary file names

       <b>disablevalidation=&lt;0|1&gt;:</b> sets whether input validation is performed. Valid values are

       0:     validation is enabled (default)

       1:     validation is disabled

       <b>md5=&lt;0|1&gt;:</b> md5 checksum creation for output file. This option can only be given if outputformat=bam. Then
       valid values are

       0:     do not compute checksum. This is the default.

       1:     compute checksum. If the md5filename key is set, then the checksum is written to the  given  file.
              If md5filename is unset, then no checksum will be computed.

       <b>md5filename</b> file name for md5 checksum if md5=1.

       <b>index=&lt;0|1&gt;:</b>  compute  BAM index for output file. This option can only be given if outputformat=bam. Then
       valid values are

       0:     do not compute BAM index. This is the default.

       1:     compute BAM index. If the indexfilename key is set, then the BAM index is  written  to  the  given
              file. If indexfilename is unset, then no BAM index will be computed.

       <b>indexfilename</b> file name for output BAM index if index=1.

       <b>inputformat=&lt;bam&gt;:</b>  <b>input</b>  <b>file</b> <b>format.</b>  All versions of bamintervalcomment come with support for the BAM
       input format. If the program in addition is linked to the io_lib package, then the following options  are
       valid:

       bam:   BAM (see <a href="http://samtools.sourceforge.net/SAM1.pdf">http://samtools.sourceforge.net/SAM1.pdf</a>)

       sam:   SAM (see <a href="http://samtools.sourceforge.net/SAM1.pdf">http://samtools.sourceforge.net/SAM1.pdf</a>)

       cram:  CRAM (see <a href="http://www.ebi.ac.uk/ena/about/cram_toolkit">http://www.ebi.ac.uk/ena/about/cram_toolkit</a>)

       <b>outputformat=&lt;bam&gt;:</b> <b>output</b> <b>file</b> <b>format.</b>  All versions of bamintervalcomment come with support for the BAM
       output format. If the program in addition is linked to the io_lib package, then the following options are
       valid:

       bam:   BAM (see <a href="http://samtools.sourceforge.net/SAM1.pdf">http://samtools.sourceforge.net/SAM1.pdf</a>)

       sam:   SAM (see <a href="http://samtools.sourceforge.net/SAM1.pdf">http://samtools.sourceforge.net/SAM1.pdf</a>)

       cram:  CRAM  (see  <a href="http://www.ebi.ac.uk/ena/about/cram_toolkit">http://www.ebi.ac.uk/ena/about/cram_toolkit</a>).  This  format  is not advisable for data
              sorted by query name.

       <b>I=&lt;[stdin]&gt;:</b> input filename, standard input if unset.

       <b>O=&lt;[stdout]&gt;:</b> output filename, standard output if unset.

       <b>inputthreads=&lt;[1]&gt;:</b> input helper threads, only valid for inputformat=bam.

       <b>outputthreads=&lt;[1]&gt;:</b> output helper threads, only valid for outputformat=bam.

       <b>reference=&lt;[]&gt;:</b> reference FastA file for inputformat=cram and outputformat=cram. An index file (.fai)  is
       required.

       <b>range=&lt;&gt;:</b>  input range to be processed. This option is only valid if the input is a coordinate sorted and
       indexed BAM file

       <b>intervals=&lt;&gt;:</b> file name of intervals file

</pre><h4><b>AUTHOR</b></h4><pre>
       Written by German Tischler.

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       Report bugs to &lt;<a href="mailto:germant@miltenyibiotec.de">germant@miltenyibiotec.de</a>&gt;

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright © 2009-2014 German Tischler, © 2011-2014 Genome Research  Limited.   License  GPLv3+:  GNU  GPL
       version 3 &lt;<a href="http://gnu.org/licenses/gpl.html">http://gnu.org/licenses/gpl.html</a>&gt;
       This  is  free software: you are free to change and redistribute it.  There is NO WARRANTY, to the extent
       permitted by law.

BIOBAMBAM                                           June 2014                              <u><a href="../man1/BAMINTERVALCOMMENT.1.html">BAMINTERVALCOMMENT</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>