<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Run::Alignment::Exonerate - wrapper for the exonerate alignment program</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-perl-run-perl">libbio-perl-run-perl_1.7.3-8_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Run::Alignment::Exonerate - wrapper for the exonerate alignment program

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::Tools::Run::Alignment::Exonerate;
         use Bio::SeqIO;

         my $qio = Bio::SeqIO-&gt;new(-file=&gt;$ARGV[0],-format=&gt;'fasta');
         my $query = $qio-&gt;next_seq();
         my $tio = Bio::SeqIO-&gt;new(-file=&gt;$ARGV[1],-format=&gt;'fasta');
         my $target = $sio-&gt;next_seq();

         #exonerate parameters can all be passed via arguments parameter.
         #parameters passed are not checked for validity

         my $run = Bio::Tools::Run::Alignment::Exonerate-&gt;
             new(arguments=&gt;'--model est2genome --bestn 10');
         my $searchio_obj = $run-&gt;run($query,$target);

         while(my $result = $searchio-&gt;next_result){
           while( my $hit = $result-&gt;next_hit ) {
             while( my $hsp = $hit-&gt;next_hsp ) {
               print $hsp-&gt;start."\t".$hsp-&gt;end."\n";
             }
           }
         }

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Wrapper for Exonerate alignment program. You can get exonerate at <a href="http://www.ebi.ac.uk/~guy/exonerate/">http://www.ebi.ac.uk/~guy/exonerate/</a>.
       This wrapper is written without parameter checking. All parameters are passed via the arugment parameter
       that is passed in the constructor. See SYNOPSIS.  For exonerate parameters, run exonerate --help for more
       details.

</pre><h4><b>PROGRAM</b> <b>VERSIONS</b></h4><pre>
       The tests have been shown to pass with exonorate versions 2.0 - 2.2.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         <a href="http://redmine.open-bio.org/projects/bioperl/">http://redmine.open-bio.org/projects/bioperl/</a>

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Shawn</b> <b>Hoon</b></h4><pre>
         Email shawnh-at-stanford.edu

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>program_name</b>
        Title   : program_name
        Usage   : $factory&gt;program_name()
        Function: holds the program name
        Returns:  string
        Args    : None

   <b>program_dir</b>
        Title   : program_dir
        Usage   : $factory-&gt;program_dir(@params)
        Function: returns the program directory, obtained from ENV variable.
        Returns:  string
        Args    :

   <b>new</b>
        Title   : new
        Usage   : my $factory= Bio::Tools::Run::Phrap-&gt;new();
        Function: creates a new Phrap factory
        Returns:  Bio::Tools::Run::Phrap
        Args    :

   <b>version</b>
        Title   : version
        Usage   : exit if $prog-&gt;version() &lt; 1.8
        Function: Determine the version number of the program
        Example :
        Returns : float or undef
        Args    : none

   <b>run</b>
        Title   :   run()
        Usage   :   my $feats = $factory-&gt;run($seq)
        Function:   Runs Phrap
        Returns :   An array of Bio::SeqFeature::Generic objects
        Args    :   A Bio::PrimarySeqI

   <b>_input</b>
        Title   :   _input
        Usage   :   $factory-&gt;_input($seqFile)
        Function:   get/set for input file
        Returns :
        Args    :

   <b>_run</b>
        Title   :   _run
        Usage   :   $factory-&gt;_run()
        Function:   Makes a system call and runs Phrap
        Returns :   An array of Bio::SeqFeature::Generic objects
        Args    :

   <b>_writeInput</b>
        Title   :   _writeInput
        Usage   :   $factory-&gt;_writeInput($query,$target)
        Function:   Creates a file from the given seq object
        Returns :   A string(filename)
        Args    :   Bio::PrimarySeqI

   <b>_setparams</b>
        Title   :  _setparams
        Usage   :  Internal function, not to be called directly
        Function:  creates a string of params to be used in the command string
        Example :
        Returns :  string of params
        Args    :

perl v5.34.0                                       2022-02-10              <u>Bio::Tools::Ru...ment::<a href="../man3pm/Exonerate.3pm.html">Exonerate</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>