<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Run::Alignment::Lagan - Object for the local execution of the LAGAN suite of tools (including</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-perl-run-perl">libbio-perl-run-perl_1.7.3-8_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Run::Alignment::Lagan - Object for the local execution of the LAGAN suite of tools (including
       MLAGAN for multiple sequence alignments)

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::Tools::Run::Alignment::Lagan;

         @params =
             ('chaos' =&gt; "The contents of this string will be passed as args to chaos",

              #Read you chaos README file for more info/This functionality
              #has not been tested and will be integrated in future versions.

              'order' =&gt; "\"-gs -7 -gc -2 -mt 2 -ms -1\"",
              #Where gap start penalty of- 7, gap continue of -2, match of 2,
              #and mismatch of -1.

              'recurse' =&gt; "\"(12,25),(7,25),(4,30)"\",
              #A list of (wordlength,score cutoff) pairs to be used in the
              #recursive anchoring

              'tree' =&gt; "\"(sample1 (sample2 sample3))"\",
              #Used by mlagan / tree can also be passed when calling mlagan directly

              #SCORING PARAMETERS FOR MLAGAN:
              'match' =&gt; 12,
              'mismatch' =&gt; -8,
              'gapstart' =&gt; -50,
              'gapend' =&gt; -50,
              'gapcont' =&gt; -2,
         );

</pre><h4><b>DESCRIPTION</b></h4><pre>
       To run mlagan/lagan, you must have an environment variable that points to the executable directory with
       files lagan.pl etc.  "LAGAN_DIR=/opt/lagan_executables/"

       Simply having the executables in your path is not supported because the executables themselves only work
       with the environment variable set.

       All lagan and mlagan parameters listed in their Readmes can be set except for the mfa flag which has been
       turned on by default to prevent parsing of the alignment format.

       TO USE LAGAN:

         my $lagan = Bio::Tools::Run::Alignment::Lagan-&gt;new(@params);
         my $report_out = $lagan-&gt;lagan($seq1, $seq2);

       A SimpleAlign object is returned.

       TO USE MLAGAN:

         my $lagan = Bio::Tools::Run::Alignment::Lagan-&gt;new();
         my $tree = "(($seqname1 $seqname2) $seqname3)";
         my @sequence_objs;    #an array of bioperl Seq objects

         ##If you use an unblessed seq array
         my $seq_ref = \@sequence_objs;
         bless $seq_ref, "ARRAY";

         my $report_out = $lagan-&gt;mlagan($seq_ref, $tree);

         A SimpleAlign object is returned

       Only basic mlagan/lagan functionality has been implemented due to the iterative development of their
       project.  Future maintenance upgrades will include enhanced features and scoring.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         <a href="http://redmine.open-bio.org/projects/bioperl/">http://redmine.open-bio.org/projects/bioperl/</a>

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Stephen</b> <b>Montgomery</b></h4><pre>
       Email <a href="mailto:smontgom@bcgsc.bc.ca">smontgom@bcgsc.bc.ca</a>

       Genome Sciences Centre in beautiful Vancouver, British Columbia CANADA

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
       MLagan/Lagan is the hard work of Michael Brudno et al.

       Sendu Bala <a href="mailto:bix@sendu.me.uk">bix@sendu.me.uk</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>lagan</b>
         Runs the Lagan pairwise alignment algorithm
         Inputs should be two PrimarySeq objects.

         Returns an SimpleAlign object / preloaded with the tmp file of the
         Lagan multifasta output.

   <b>mlagan</b>
         Runs the Mlagan multiple sequence alignment algorithm.
         Inputs should be an Array of Primary Seq objects and a Phylogenetic Tree in
         String format or as a Bio::Tree::TreeI compliant object.
         Returns an SimpleAlign object / preloaded with the tmp file of the Mlagan
         multifasta output.

   <b>nuc_matrix</b>
        Title   : nuc_matrix
        Usage   : my $matrix_obj = $obj-&gt;nuc_matrix();
                  -or-
                  $obj-&gt;nuc_matrix($matrix_obj);
                  -or-
                  $obj-&gt;nuc_matrix($matrix_file);
        Function: Get/set the substitution matrix for use by mlagan. By default the
                  file $LAGAN_DIR/nucmatrix.txt is used by mlagan. By default this
                  method returns a corresponding Matrix.
        Returns : Bio::Matrix::Mlagan object
        Args    : none to get, OR to set:
                  Bio::Matrix::MLagan object
                  OR
                  filename of an mlagan substitution matrix file

                  NB: due to a bug in mlagan 2.0, the -nucmatrixfile option does not
                  work, so this Bioperl wrapper is unable to simply point mlagan to
                  your desired matrix file (or to a temp file generated from your
                  matrix object). Instead the $LAGAN_DIR/nucmatrix.txt file must
                  actually be replaced. This wrapper will make a back-up copy of that
                  file, write the new file in its place, then revert things back to the
                  way they were after the alignment has been produced. For this reason,
                  $LAGAN_DIR must be writable, as must $LAGAN_DIR/nucmatrix.txt.

   <b>_setinput</b>
        Title   : _setinput
        Usage   : Internal function, not to be called directly
        Function: Create input file(s) for Lagan executables
        Returns : name of files containing Lagan data input /
                  or array of files and phylo tree for Mlagan data input

   <b>_generic_lagan</b>
        Title   : _generic_lagan
        Usage   :  internal function not called directly
        Returns :  SimpleAlign object

   <b>_setparams</b>
        Title   : _setparams
        Usage   : Internal function, not to be called directly
        Function: Create parameter inputs for (m)Lagan program
        Returns : parameter string to be passed to Lagan
        Args    : Reference to calling object and name of (m)Lagan executable

   <b>_runlagan</b>
        Title   :  _runlagan
        Usage   :  Internal function, not to be called directly
        Function:   makes actual system call to (m)Lagan program
        Example :
        Returns : Report object in the SimpleAlign object

   <b>executable</b>
        Title   : executable
        Usage   : my $exe = $lagan-&gt;executable('mlagan');
        Function: Finds the full path to the 'lagan' executable
        Returns : string representing the full path to the exe
        Args    : [optional] name of executable to set path to
                  [optional] boolean flag whether or not warn when exe is not found

        Thanks to Jason Stajich for providing the framework for this subroutine

   <b>program_path</b>
        Title   : program_path
        Usage   : my $path = $lagan-&gt;program_path();
        Function: Builds path for executable
        Returns : string representing the full path to the exe

        Thanks to Jason Stajich for providing the framework for this subroutine

   <b>program_dir</b>
        Title   : program_dir
        Usage   : my $dir = $lagan-&gt;program_dir();
        Function: Abstract get method for dir of program. To be implemented
                  by wrapper.
        Returns : string representing program directory

        Thanks to Jason Stajich for providing the framework for this subroutine

   <b>version</b>
        Title   : version
        Usage   : my $version = $lagan-&gt;version;
        Function: returns the program version
        Returns : number
        Args    : none

perl v5.34.0                                       2022-02-10             <u>Bio::Tools::Run::Alignment::<a href="../man3pm/Lagan.3pm.html">Lagan</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>