<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Align::Utilities - A collection of utilities regarding converting and manipulating alignment objects</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Align::Utilities - A collection of utilities regarding converting and manipulating alignment objects

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::Align::Utilities qw(:all);

         # Even if the protein alignments are local make sure the start/end
         # stored in the LocatableSeq objects are to the full length protein.
         # The coding sequence that is passed in should still be the full
         # length CDS as the nt alignment will be generated.
         # %dnaseqs is a hash of CDS sequences (spliced)
         my $dna_aln = aa_to_dna_aln($aa_aln,\%dnaseqs);

         # The reverse, which is simpler. The input alignment has to be
         # translate-able, with gap lengths and an overall length divisible by 3
         my $aa_aln = dna_to_aa_aln($dna_al);

         # Generate bootstraps
         my $replicates = bootstrap_replicates($aln,$count);

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This module contains utility methods for manipulating sequence alignments (Bio::Align::AlignI) objects.

       The <b>aa_to_dna_aln</b> utility is essentially the same as the <b>mrtrans</b> program by Bill Pearson available at
       <a href="ftp://ftp.virginia.edu/pub/fasta/other/mrtrans.shar">ftp://ftp.virginia.edu/pub/fasta/other/mrtrans.shar</a>.  Of course this is a pure-Perl implementation, but
       just to mention that if anything seems odd you can check the alignments generated against Bill's program.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Jason</b> <b>Stajich</b></h4><pre>
       Email jason-at-bioperl-dot-org

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>aa_to_dna_aln</b>
        Title   : aa_to_dna_aln
        Usage   : my $dnaaln = aa_to_dna_aln($aa_aln, \%seqs);
        Function: Will convert an AA alignment to DNA space given the
                  corresponding DNA sequences.  Note that this method expects
                  the DNA sequences to be in frame +1 (GFF frame 0) as it will
                  start to project into coordinates starting at the first base of
                  the DNA sequence, if this alignment represents a different
                  frame for the cDNA you will need to edit the DNA sequences
                  to remove the 1st or 2nd bases (and revcom if things should be).
        Returns : Bio::Align::AlignI object
        Args    : 2 arguments, the alignment and a hashref.
                  Alignment is a Bio::Align::AlignI of amino acid sequences.
                  The hash reference should have keys which are
                  the display_ids for the aa
                  sequences in the alignment and the values are a
                  Bio::PrimarySeqI object for the corresponding
                  spliced cDNA sequence.

       See also: Bio::Align::AlignI, Bio::SimpleAlign, Bio::PrimarySeq

   <b>dna_to_aa_aln</b>
        Title   : dna_to_aa_aln
        Usage   : my $aa_aln = dna_to_aa_aln($dna_aln);
        Function: Convert a DNA alignment to an amino acid alignment where
                  the length of all alignment strings and the lengths of any
                  gaps must be divisible by 3
        Returns : Bio::Align::AlignI object
        Args    : the DNA alignment, a Bio::Align::AlignI of DNA sequences

       See also: Bio::Align::AlignI, Bio::SimpleAlign, Bio::PrimarySeq

   <b>bootstrap_replicates</b>
        Title   : bootstrap_replicates
        Usage   : my $alns = &amp;bootstrap_replicates($aln,100);
        Function: Generate a pseudo-replicate of the data by randomly
                  sampling, with replacement, the columns from an alignment for
                  the non-parametric bootstrap.
        Returns : Arrayref of L&lt;Bio::SimpleAlign&gt; objects
        Args    : L&lt;Bio::SimpleAlign&gt; object
                  Number of replicates to generate

   <b>bootstrap_replicates_codons</b>
        Title   : bootstrap_replicates_codons
        Usage   : my $alns = &amp;bootstrap_replicates_codons($aln,100);
        Function: Generate a pseudo-replicate of the data by randomly
                  sampling, with replacement, the columns from a codon alignment for
                  the non-parametric bootstrap. The alignment is assumed to start on
                  the first position of a codon.
        Returns : Arrayref of L&lt;Bio::SimpleAlign&gt; objects
        Args    : L&lt;Bio::SimpleAlign&gt; object
                  Number of replicates to generate

   <b>cat</b>
        Title     : cat
        Usage     : $aln123 = cat($aln1, $aln2, $aln3)
        Function  : Concatenates alignment objects. Sequences are identified by id.
                    An error will be thrown if the sequence ids are not unique in the
                    first alignment. If any ids are not present or not unique in any
                    of the additional alignments then those sequences are omitted from
                    the concatenated alignment, and a warning is issued. An error will
                    be thrown if any of the alignments are not flush, since
                    concatenating such alignments is unlikely to make biological
                    sense.
        Returns   : A new Bio::SimpleAlign object
        Args      : A list of Bio::SimpleAlign objects

   <b>most_common_sequences</b>
        Title     : most_common_sequences
        Usage     : @common = most_common_sequences ($align, $case_sensitivity)
        Function  : Returns an array of the sequences that appear most often in the
                    alignment (although this probably makes more sense when there is
                    only a single most common sequence).  Sequences are compared after
                    removing any "-" (gap characters), and ambiguous units (e.g., R
                    for purines) are only compared to themselves.  The returned
                    sequence is also missing the "-" since they don't actually make
                    part of the sequence.
        Returns   : Array of text strings.
        Arguments : Optional argument defining whether the comparison between sequences
                    to find the most common should be case sensitive. Defaults to
                    false, i.e, not case sensitive.

perl v5.32.1                                       2021-08-15                         <u>Bio::Align::<a href="../man3pm/Utilities.3pm.html">Utilities</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>