<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::DB::GFF::Aggregator::alignment -- Alignment aggregator</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-db-gff-perl">libbio-db-gff-perl_1.7.4-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::DB::GFF::Aggregator::alignment -- Alignment aggregator

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::DB::GFF;

         # Open the sequence database
         my $db      = Bio::DB::GFF-&gt;new( -adaptor =&gt; 'dbi:mysql',
                                          -dsn     =&gt; 'dbi:mysql:elegans42',
                                          -aggregator =&gt; ['alignment'],
                                        );

        -----------------------------
        Aggregator method: alignment
        Main method:       (none)
        Sub methods:       nucleotide_match,EST_match,cDNA_match,expressed_sequence_match,
                           translated_nucleotide_match,protein_match,HSP
        -----------------------------

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::DB::GFF::Aggregator::alignment is one of the default aggregators, and was written to be compatible
       with the C elegans GFF files.  It aggregates raw "similarity" features into composite features of type
       "alignment".  A better name for this class might be "gapped_alignment."

       This aggregator does not insist that there be a single top-level feature that spans one end of the
       alignment to the other.  As a result, it can produce truncated alignments if the entire alignment is not
       contained within the segment of interest.

   <b>aggregate</b>
        Title   : aggregate
        Usage   : $features = $a-&gt;aggregate($features,$factory)
        Function: aggregate a feature list into composite features
        Returns : an array reference containing modified features
        Args    : see L&lt;Bio::DB::GFF::Aggregator&gt;
        Status  : Public

       Because of the large number of similarity features, the <b>aggregate()</b> method is overridden in order to
       perform some optimizations.

   <b>method</b>
        Title   : method
        Usage   : $aggregator-&gt;method
        Function: return the method for the composite object
        Returns : the string "alignment"
        Args    : none
        Status  : Public

   <b>part_names</b>
        Title   : part_names
        Usage   : $aggregator-&gt;part_names
        Function: return the methods for the sub-parts
        Returns : the full list of aggregated methods
        Args    : none
        Status  : Public

perl v5.30.0                                       2020-01-13              <u>Bio::DB::GFF::...ator::<a href="../man3pm/alignment.3pm.html">alignment</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>