<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Alignment::Consed - A module to work with objects from consed .ace files</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Alignment::Consed - A module to work with objects from consed .ace files

</pre><h4><b>SYNOPSIS</b></h4><pre>
         # a report for sequencing stuff
         my $o_consed = Bio::Tools::Alignment::Consed-&gt;new(
             -acefile =&gt; "/path/to/an/acefile.ace.1",
             -verbose =&gt; 1);
         my $foo = $o_consed-&gt;set_reverse_designator("r");
         my $bar = $o_consed-&gt;set_forward_designator("f");

         # get the contig numbers
         my @keys = $o_consed-&gt;get_contigs();

         # construct the doublets
         my $setter_doublets = $o_consed-&gt;choose_doublets();

         # get the doublets
         my @doublets = $o_consed-&gt;get_doublets();

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::Tools::Alignment::Consed provides methods and objects to deal with the output from the Consed
       software suite. Specifically, takes an ".ace" file and provides objects for the results.

       A word about doublets: This module was written to accommodate a large EST sequencing operation. In this
       case, EST's were sequenced from the 3' and from the 5' end of the EST. The objective was to find a
       consensus sequence for these two reads.  Thus, a contig of two is what we wanted, and this contig should
       consist of the forward and reverse reads of a getn clone. For example, for a forward designator of "F"
       and a reverse designator of "R", if the two reads chad1F and chad1R were in a single contig (for example
       Contig 5) it will be determined that the consensus sequence for Contig 5 will be the sequence for clone
       chad1.

       Doublets are good!

       This module parses ".ace" and related files. A detailed list of methods can be found at the end of this
       document.

       I wrote a detailed rationale for design that may explain the reasons why some things were done the way
       they were done. That document is beyond the scope of this pod and can probably be found in the directory
       from which this module was 'made' or at
       &lt;<a href="http://www.dieselwurks.com/bioinformatics/consedpm_documentation.pdf">http://www.dieselwurks.com/bioinformatics/consedpm_documentation.pdf</a>&gt;.

       Note that the POD in that document might be old but the original rationale still stands.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Chad</b> <b>Matsalla</b></h4><pre>
       Email chad-at-dieselwurks.com

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>new()</b>
        Title   : new(-acefile =&gt; $path_to_some_acefile, -verbose =&gt; "1")
        Usage   : $o_consed = Bio::Tools::Alignment::Consed-&gt;
                     new(-acefile =&gt; $path_to_some_acefile, -verbose =&gt; "1");
        Function: Construct the Bio::Tools::Alignment::Consed object. Sets
                  verbosity for the following procedures, if necessary:
                  1. Construct a new Bio::Tools::Alignment::Trim object, to
                  handle quality trimming 2. Read in the acefile and parse it

        Returns : A reference to a Bio::Tools::Alignment::Consed object.
        Args    : A hash. (-acefile) is the filename of an acefile. If a full path
                  is not specified "./" is prepended to the filename and used from
                  instantiation until destruction. If you want
                  Bio::Tools::Alignment::Consed to be noisy during parsing of
                  the acefile, specify some value for (-verbose).

   <b>set_verbose()</b>
        Title   : set_verbose()
        Usage   : $o_consed-&gt;<a href="../man1/set_verbose.1.html">set_verbose</a>(1);
        Function: Set the verbosity level for debugging messages. On instantiation
                  of the Bio::Tools::Alignment::Consed object the verbosity level
                  is set to 0 (quiet).
        Returns : 1 or 0.
        Args    : The verbosity levels are:
                     0 - quiet
                     1 - noisy
                     2 - noisier
                     3 - annoyingly noisy

       This method for setting verbosity has largely been superseded by a sub-by-sub way, where for every sub
       you can provide a (-verbose) switch. I am doing converting this bit-by-bit so do not be surprised if some
       subs do not honour this.

   <b>get_filename()</b>
        Title   : get_filename()
        Usage   : $o_consed-&gt;get_filename();
        Function: Returns the name of the acefile being used by the
                  Bio::Tools::Alignment::Consed object.
        Returns : A scalar containing the name of a file.
        Args    : None.

   <b>count_sequences_with_grep()</b>
        Title   : count_sequences_with_grep()
        Usage   : $o_consed-&gt;count_sequences_with_grep();
        Function: Use <a href="file:/bin/grep">/bin/grep</a> to scan through the files in the ace project dir
                  and count sequences in those files. I used this method in the
                  development of this module to verify that I was getting all of the
                  sequences. It works, but it is (I think) unix-like platform
                  dependent.
        Returns : A scalar containing the number of sequences in the ace project
                  directory.
        Args    : None.

       If you are on a non-UNIX platform, you really do not have to use this. It is more of a debugging routine
       designed to address very specific problems.

       This method was reimplemented to be platform independent with a pure perl implementation.  The above note
       can be ignored.

   <b>get_path()</b>
        Title   : get_path()
        Usage   : $o_consed-&gt;get_path();
        Function: Returns the path to the acefile this object is working with.
        Returns : Scalar. The path to the working acefile.
        Args    : None.

   <b>get_contigs()</b>
        Title   : get_contigs()
        Usage   : $o_consed-&gt;get_contigs();
        Function: Return the keys to the Bio::Tools::Alignment::Consed object.
        Returns : An array containing the keynames in the
                  Bio::Tools::Alignment::Consed object.
        Args    : None.

       This would normally be used to get the keynames for some sort of iterator. These keys are worthless in
       general day-to-day use because in the Consed acefile they are simply Contig1, Contig2, ...

   <b>get_class($contig_keyname)</b>
        Title   : get_class($contig_keyname)
        Usage   : $o_consed-&gt;get_class($contig_keyname);
        Function: Return the class name for this contig
        Returns : A scalar representing the class of this contig.
        Args    : None.
        Notes   :

   <b>get_quality_array($contig_keyname)</b>
        Title   : get_quality_array($contig_keyname)
        Usage   : $o_consed-&gt;get_quality_array($contig_keyname);
        Function: Returns the quality for the consensus sequence for the given
                  contig as an array. See get_quality_scalar to get this as a scalar.
        Returns : An array containing the quality for the consensus sequence with
                  the given keyname.
        Args    : The keyname of a contig. Note: This is a keyname. The key would
                  normally come from get_contigs.

       Returns an array, not a reference. Is this a bug? <u>thinking</u> No.  Well, maybe.  Why was this developed like
       this? I was using FreezeThaw for object persistence, and when it froze out these arrays it took a long
       time to thaw it. Much better as a scalar.

       See <b>get_quality_scalar()</b>

   <b>get_quality_scalar($contig_keyname)</b>
        Title   : get_quality_scalar($contig_keyname)
        Usage   : $o_consed-&gt;get_quality_scalar($contig_keyname);
        Function: Returns the quality for the consensus sequence for the given
                  contig as a scalar. See get_quality_array to get this as an array.
        Returns : An scalar containing the quality for the consensus sequence with
                  the given keyname.
        Args    : The keyname of a contig. Note this is a _keyname_. The key would
                  normally come from get_contigs.

       Why was this developed like this? I was using FreezeThaw for object persistence, and when it froze out
       these arrays it took a coon's age to thaw it. Much better as a scalar.

       See <b>get_quality_array()</b>

   <b>freeze_hash()</b>
        Title   : freeze_hash()
        Usage   : $o_consed-&gt;freeze_hash();

        Function: Use Ilya's FreezeThaw module to create a persistent data
                  object for this Bio::Tools::Alignment::Consed data
                  structure. In the case of AAFC, we use
                  Bio::Tools::Alignment::Consed to pre-process bunches of
                  sequences, freeze the structures, and send in a harvesting
                  robot later to do database stuff.
        Returns : 0 or 1;
        Args    : None.

       This procedure was removed so Consed.pm won't require FreezeThaw.

   <b>get_members($contig_keyname)</b>
        Title   : get_members($contig_keyname)
        Usage   : $o_consed-&gt;get_members($contig_keyname);
        Function: Return the _names_ of the reads in this contig.
        Returns : An array containing the names of the reads in this contig.
        Args    : The keyname of a contig. Note this is a keyname. The keyname
                  would normally come from get_contigs.

       See <b>get_contigs()</b>

   <b>get_members_by_name($some_arbitrary_name)</b>
        Title   : get_members_by_name($some_arbitrary_name)
        Usage   : $o_consed-&gt;get_members_by_name($some_arbitrary_name);
        Function: Return the names of the reads in a contig. This is the name given
                  to $contig{key} based on what is in the contig. This is different
                  from the keys retrieved through get_contigs().
        Returns : An array containing the names of the reads in the contig with this
                  name.
        Args    : The name of a contig. Not a key, but a name.

       Highly inefficient. use some other method if possible.  See <b>get_contigs()</b>

   <b>get_contig_number_by_name($some_arbitrary_name)</b>
        Title   : get_contig_number_by_name($some_arbitrary_name)
        Usage   : $o_consed-&gt;get_contig_number_by_name($some_arbitrary_name);
        Function: Return the names of the reads in a contig. This is the name given
                  to $contig{key} based on what is in the contig. This is different
                  from the keys retrieved through get_contigs().
        Returns : An array containing the names of the reads in the contig with this
                  name.
        Args    : The name of a contig. Not a key, but a name.

       See <b>get_contigs()</b>

   <b>get_sequence($contig_keyname)</b>
        Title   : get_sequence($contig_keyname)
        Usage   : $o_consed-&gt;get_sequence($contig_keyname);
        Function: Returns the consensus sequence for a given contig.
        Returns : A scalar containing a sequence.
        Args    : The keyname of a contig. Note this is a key. The key would
                  normally come from get_contigs.

       See <b>get_contigs()</b>

   <b>set_final_sequence($some_sequence)</b>
        Title   : set_final_sequence($name,$some_sequence)
        Usage   : $o_consed-&gt;set_final_sequence($name,$some_sequence);
        Function: Provides a manual way to set the sequence for a given key in the
                  contig hash. Rarely used.
        Returns : 0 or 1;
        Args    : The name (not the keyname) of a contig and an arbitrary string.

       A method with a questionable and somewhat mysterious origin. May raise the dead or something like that.

   <b>_read_file()</b>
        Title   : _read_file();
        Usage   : _read_file();
        Function: An internal subroutine used to read in an acefile and parse it
                  into a Bio::Tools::Alignment::Consed object.
        Returns : 0 or 1.
        Args    : Nothing.

       This routine creates and saves the filhandle for reading the files in {fh}

   <b>set_reverse_designator($some_string)</b>
        Title   : set_reverse_designator($some_string)
        Usage   : $o_consed-&gt;set_reverse_designator($some_string);
        Function: Set the designator for the reverse read of contigs in this
                  Bio::Tools::Alignment::Consed object. Used to determine if
                  contigs containing two reads can be named.
        Returns : The value of $o_consed-&gt;{reverse_designator} so you can check
                  to see that it was set properly.
        Args    : An arbitrary string.

       May be useful only to me. <u>shrug</u>

   <b>set_forward_designator($some_string)</b>
        Title   : set_forward_designator($some_string)
        Usage   : $o_consed-&gt;set_forward_designator($some_string);
        Function: Set the designator for the forward read of contigs in this
                  Bio::Tools::Alignment::Consed object. Used to determine if
                  contigs containing two reads can be named.
        Returns : The value of $o_consed-&gt;{forward_designator} so you can check
                  to see that it was set properly.
        Args    : An arbitrary string.

       May be useful only to me. <u>shrug</u>

   <b>set_designator_ignore_case("yes")</b>
        Title   : set_designator_ignore_case("yes")
        Usage   : $o_consed-&gt;set_designator_ignore_case("yes");
        Function: Deprecated.
        Returns : Deprecated.
        Args    : Deprecated.

       Deprecated. Really. Trust me.

   <b>set_trim_points_singlets_and_singletons()</b>
        Title   : set_trim_points_singlets_and_singletons()
        Usage   : $o_consed-&gt;set_trim_points_singlets_and_singletons();
        Function: Set the trim points for singlets and singletons based on
                  quality.  Uses the Bio::Tools::Alignment::Trim object. Use
                  at your own risk because the Bio::Tools::Alignment::Trim
                  object was designed specifically for me and is mysterious
                  in its ways. Every time somebody other then me uses it a
                  swarm of locusts decends on a small Central American
                  village so do not say you weren't warned.
        Returns : Nothing.
        Args    : None.

       Working on exceptions and warnings here.

       See Bio::Tools::Alignment::Trim for more information

   <b>set_trim_points_doublets()</b>
        Title   : set_trim_points_doublets()
        Usage   : $o_consed-&gt;set_trim_points_doublets();
        Function: Set the trim points for doublets based on quality. Uses the
                  Bio::Tools::Alignment::Trim object. Use at your own risk because
                  the Bio::Tools::Alignment::Trim object was designed specifically
                  for me and is mysterious in its ways. Every time somebody other
                  then me uses it you risk a biblical plague being loosed on your
                  city.
        Returns : Nothing.
        Args    : None.
        Notes   : Working on exceptions here.

       See Bio::Tools::Alignment::Trim for more information

   <b>get_trimmed_sequence_by_name($name)</b>
        Title   : get_trimmed_sequence_by_name($name)
        Usage   : $o_consed-&gt;get_trimmed_sequence_by_name($name);
        Function: Returns the trimmed_sequence of a contig with {name} eq $name.
        Returns : A scalar- the trimmed sequence.
        Args    : The {name} of a contig.
        Notes   :

   <b>set_dash_present_in_sequence_name("yes")</b>
        Title   : set_dash_present_in_sequence_name("yes")
        Usage   : $o_consed-&gt;set_dash_present_in_sequence_name("yes");
        Function: Deprecated. Part of an uncompleted thought. ("Oooh! Shiny!")
        Returns : Nothing.
        Args    : "yes" to set {dash_present_in_sequence_name} to 1
        Notes   :

   <b>set_doublets()</b>
        Title   : set_doublets()
        Usage   : $o_consed-&gt;set_doublets();
        Function: Find pairs that have similar names and mark them as doublets
                  and set the {name}.
        Returns : 0 or 1.
        Args    : None.

       A complicated subroutine that iterates over the Bio::Tools::Alignment::Consed looking for contigs of 2.
       If the forward and reverse designator are removed from each of the reads in {'member_array'} and the
       remaining reads are the same, {name} is set to that name and the contig's class is set as "doublet".  If
       any of those cases fail the contig is marked as a "pair".

   <b>set_singlets</b>
        Title   : set_singlets
        Usage   : $o_consed-&gt;set_singlets();
        Function: Read in a singlets file and place them into the
                  Bio::Tools::Alignment::Consed object.
        Returns : Nothing.
        Args    : A scalar to turn on verbose parsing of the singlets file.
        Notes   :

   <b>get_singlets()</b>
        Title   : get_singlets()
        Usage   : $o_consed-&gt;get_singlets();
        Function: Return the keynames of the singlets.
        Returns : An array containing the keynames of all
                  Bio::Tools::Alignment::Consed sequences in the class "singlet".
        Args    : None.
        Notes   :

   <b>set_quality_by_name($name,$quality)</b>
        Title   : set_quality_by_name($name,$quality)
        Usage   : $o_consed-&gt;set_quality_by_name($name,$quality);
        Function: Deprecated. Make the contig with {name} have {'quality'} $quality.
                  Probably used for testing.
        Returns : Nothing.
        Args    : The name of a contig and a scalar for its quality.
        Notes   : Deprecated.

   <b>set_singlet_quality()</b>
        Title   : set_singlet_quality()
        Usage   : $o_consed-&gt;set_singlet_quality();
        Function: For each singlet, go to the appropriate file in phd_dir and read
                  in the phred quality for that read and place it into {'quality'}
        Returns : 0 or 1.
        Args    : None.
        Notes   : This is the next subroutine that will receive substantial revision
                  in the next little while. It really should eval the creation of
                  Bio::Tools::Alignment::Phred objects and go from there.

   <b>set_contig_quality()</b>
        Title   : set_contig_quality()
        Usage   : $o_consed-&gt;set_contig_quality();
        Function: Deprecated.
        Returns : Deprecated.
        Args    : Deprecated.
        Notes   : Deprecated. Really. Trust me.

   <b>get_multiplets()</b>
        Title   : get_multiplets()
        Usage   : $o_consed-&gt;get_multiplets();
        Function: Return the keynames of the multiplets.
        Returns : Returns an array containing the keynames of all
                  Bio::Tools::Alignment::Consed sequences in the class "multiplet".
        Args    : None.
        Notes   :

   <b>get_all_members()</b>
         Title   : get_all_members()
         Usage   : @all_members = $o_consed-&gt;get_all_members();
         Function: Return a list of all of the read names in the
                   Bio::Tools::Alignment::Consed object.
         Returns : An array containing all of the elements in all of the
                   {'member_array'}s.
         Args    : None.
         Notes   :

   <b>sum_lets($total_only)</b>
        Title   : sum_lets($total_only)
        Usage   : $statistics = $o_consed-&gt;sum_lets($total_only);
        Function: Provide numbers for how many sequences were accounted for in the
                  Bio::Tools::Alignment::Consed object.
        Returns : If a scalar is present, returns the total number of
                  sequences accounted for in all classes. If no scalar passed
                  then returns a string that looks like this:
                  Singt/singn/doub/pair/mult/total : 2,0,<a href="../man2/1.2.html">1</a>(2),<a href="../man0/0.0.html">0</a>(0),<a href="../man0/0.0.html">0</a>(0),4
                  This example means the following: There were 1 singlets.
                  There were 0 singletons.  There were 1 doublets for a total
                  of 2 sequences in this class.  There were 0 pairs for a
                  total of 0 sequences in this class.  There were 0
                  multiplets for a total of 0 sequences in this class.  There
                  were a total of 4 sequences accounted for in the
                  Bio::Tools::Alignment::Consed object.
        Args : A scalar is optional to change the way the numbers are returned.
        Notes:

   <b>write_stats()</b>
        Title   : write_stats()
        Usage   : $o_consed-&gt;write_stats();
        Function: Write a file called "statistics" containing numbers similar to
                  those provided in sum_lets().
        Returns : Nothing. Write a file in $o_consed-&gt;{path} containing something
                  like this:

                  0,0,<a href="../man100/50.100.html">50</a>(100),<a href="../man0/0.0.html">0</a>(0),<a href="../man0/0.0.html">0</a>(0),100

                  Where the numbers provided are in the format described in the
                  documentation for sum_lets().
        Args    : None.
        Notes   : This might break platform independence, I do not know.

       See <b>sum_lets()</b>

   <b>get_singletons()</b>
        Title   : get_singletons()
        Usage   : @singletons = $o_consed-&gt;get_singletons();
        Function: Return the keynames of the singletons.
        Returns : Returns an array containing the keynames of all
                  Bio::Tools::Alignment::Consed sequences in the class "singleton".
        Args    : None.
        Notes   :

   <b>get_pairs()</b>
        Title   : get_pairs()
        Usage   : @pairs = $o_consed-&gt;get_pairs();
        Function: Return the keynames of the pairs.
        Returns : Returns an array containing the keynames of all
                  Bio::Tools::Alignment::Consed sequences in the class "pair".
        Args    : None.
        Notes   :

   <b>get_name($contig_keyname)</b>
        Title   : get_name($contig_keyname)
        Usage   : $name = $o_consed-&gt;get_name($contig_keyname);
        Function: Return the {name} for $contig_keyname.
        Returns : A string. ({name})
        Args    : A contig keyname.
        Notes   :

   <b>_get_contig_name(\@array_containing_reads)</b>
        Title   : _get_contig_name(\@array_containing_reads)
        Usage   : $o_consed-&gt;_get_contig_name(\@array_containing_reads);
        Function: The logic for the set_doublets subroutine.
        Returns : The name for this contig.
        Args    : A reference to an array containing read names.
        Notes   : Depends on reverse_designator. Be sure this is set the way you
                  intend.

   <b>get_doublets()</b>
        Title   : get_doublets()
        Usage   : @doublets = $o_consed-&gt;get_doublets();
        Function: Return the keynames of the doublets.
        Returns : Returns an array containing the keynames of all
                  Bio::Tools::Alignment::Consed sequences in the class "doublet".
        Args    : None.
        Notes   :

   <b>dump_hash()</b>
        Title   : dump_hash()
        Usage   : $o_consed-&gt;dump_hash();
        Function: Use dumpvar.pl to dump out the Bio::Tools::Alignment::Consed
                  object to STDOUT.
        Returns : Nothing.
        Args    : None.
        Notes   : I used this a lot in debugging.

   <b>dump_hash_compact()</b>
        Title   : dump_hash_compact()
        Usage   : $o_consed-&gt;dump_hash_compact();
        Function: Dump out the Bio::Tools::Alignment::Consed object in a compact way.
        Returns : Nothing.
        Args    : Nothing.
        Notes   : Cleaner then dumpValue(), dumpHash(). I used this a lot in
                  debugging.

   <b>get_phreds()</b>
        Title   : get_phreds()
        Usage   : @phreds = $o_consed-&gt;get_phreds();
        Function: For each doublet in the Bio::Tools::Alignment::Consed hash, go
                  and get the phreds for the top and bottom reads. Place them into
                  {top_phreds} and {bottom_phreds}.
        Returns : Nothing.
        Args    : Nothing.

       Requires <b>parse_phd()</b> and <b>reverse_and_complement()</b>. I realize that it would be much more elegant to pull
       qualities as required but there were certain "features" in the acefile that required a bit more detailed
       work be done to get the qualities for certain parts of the consensus sequence. In order to make _sure_
       that this was done properly I wrote things to do all steps and then I used <b>dump_hash()</b> and checked each
       one to ensure expected behavior. I have never changed this, so there you are.

   <b>parse_phd($read_name)</b>
        Title   : parse_phd($read_name)
        Usage   : $o_consed-&gt;parse_phd($read_name);
        Function: Suck in the contents of a .phd file.
        Returns : A reference to an array containing the quality values for the read.
        Args    : The name of a read.
        Notes   : This is a significantly weak subroutine because it was always
                  intended that these functions, along with the functions provided by
                  get_phreds() be put into the Bio::SeqIO:phd module. This is done
                  now but the Bio::Tools::Alignment::Consed module has not be
                  rewritten to reflect this change.

       See Bio::SeqIO::phd for more information.

   <b>reverse_and_complement(\@source)</b>
        Title   : reverse_and_complement(\@source)
        Usage   : $reference_to_array = $o_consed-&gt;reverse_and_complement(\@source);
        Function: A stub for the recursive routine reverse_recurse().
        Returns : A reference to a reversed and complemented array of phred data.
        Args    : A reference to an array of phred data.
        Notes   :

   <b>reverse_recurse($r_source,$r_destination)</b>
        Title   : reverse_recurse(\@source,\@destination)
        Usage   : $o_consed-&gt;reverse_recurse(\@source,\@destination);
        Function: A recursive routine to reverse and complement an array of
                  phred data.
        Returns : A reference to an array containing reversed phred data.
        Args    : A reference to a source array and a reverence to a destination
                  array.

       Recursion is kewl, but this sub should likely be _reverse_recurse.

   <b>show_missing_sequence()</b>
        Title   : show_missing_sequence();
        Usage   : $o_consed-&gt;show_missing_sequence();
        Function: Used by set_trim_points_doublets() to fill in quality values where
                  consed (phrap?) set them to 0 at the beginning and/or end of the
                  consensus sequences.
        Returns : Nothing.
        Args    : None.

       Acts on doublets only. Really very somewhat quite ugly. A disgusting kludge. <u>insert</u> <u>pride</u> <u>here</u> It was
       written stepwise with no real plan because it was not really evident why consed (phrap?)  was doing this.

perl v5.32.1                                       2021-08-15                 <u>Bio::Tools::Alignment::<a href="../man3pm/Consed.3pm.html">Consed</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>