<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Run::Alignment::MSAProbs - Object for the calculation of a multiple sequence alignment (MSA)</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-perl-run-perl">libbio-perl-run-perl_1.7.3-8_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Run::Alignment::MSAProbs - Object for the calculation of a multiple sequence alignment (MSA)
       from a set of unaligned sequences using the MSAProbs program

</pre><h4><b>SYNOPSIS</b></h4><pre>
         # Build a msaprobs alignment factory
         $factory = Bio::Tools::Run::Alignment::MSAProbs-&gt;new(@params);

         # Pass the factory a list of sequences to be aligned.
         $inputfilename = 't/cysprot.fa';
         # $aln is a SimpleAlign object.
         $aln = $factory-&gt;align($inputfilename);

         # or where @seq_array is an array of Bio::Seq objects
         $seq_array_ref = \@seq_array;
         $aln = $factory-&gt;align($seq_array_ref);

         #There are various additional options and input formats available.
         #See the DESCRIPTION section that follows for additional details.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       MSAProbs is Liu, Schmidt, and Maskell's (2010) alignment program using HMM and partition function
       posterior probabilities.  For more a more in-depth description see the original publication:

           Liu, Y., Schmidt, B., and Maskell, D. L. (2010) MSAProbs: multiple
           sequence alignment based on pair hidden Markov models and partition
           function posterior probabilities. I&lt;Bioinformatics&gt; <a href="../man16/26.16.html">26</a>(16): 1958-1964
           doi:10.1093/bioinformatics/btq338


                                       -OR-
           <a href="http://bioinformatics.oxfordjournals.org/content/26/16/1958.abstract">http://bioinformatics.oxfordjournals.org/content/26/16/1958.abstract</a>

       You can download the source code from <a href="http://sourceforge.net/projects/msaprobs/">http://sourceforge.net/projects/msaprobs/</a>

       It is recommended you use at least version 0.9; behaviour with earlier versions is questionable.

   <b>Helping</b> <b>the</b> <b>module</b> <b>find</b> <b>your</b> <b>executable</b>
       You will need to help MSAProbs to find the 'msaprobs' executable. This can be done in (at least) three
       ways:

         1. Make sure the msaprobs executable is in your path (i.e.
            'which msaprobs' returns a valid program)
         2. define an environmental variable MSAPROBSDIR which points to a
            directory containing the 'msaprobs' app:
          In bash
               export MSAPROBSDIR=/home/progs/msaprobs   or
          In csh/tcsh
               setenv MSAPROBSDIR /home/progs/msaprobs

         3. include a definition of an environmental variable MSAPROBSDIR
             in every script that will
            BEGIN {$ENV{MSAPROBSDIR} = '/home/progs/msaprobs'; }
            use Bio::Tools::Run::Alignment::MSAProbs;

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

        <a href="http://bugzilla.open-bio.org/">http://bugzilla.open-bio.org/</a>

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Jessen</b> <b>Bredeson</b></h4><pre>
       Email <a href="mailto:jessenbredeson@berkeley.edu">jessenbredeson@berkeley.edu</a>

</pre><h4><b>CONTRIBUTIONS</b></h4><pre>
       This MSAProbs module was adapted from the Bio::Tools::Run::Alignment::Muscle module, written by Jason
       Stajich and almost all of the credit should be given to him.

       Email jason-at-bioperl-dot-org

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>program_name</b>
        Title   : program_name
        Usage   : $factory-&gt;program_name()
        Function: holds the program name
        Returns:  string
        Args    : None

   <b>program_dir</b>
        Title   : program_dir
        Usage   : $factory-&gt;program_dir(@params)
        Function: returns the program directory, obtained from ENV variable.
        Returns:  string
        Args    :

   <b>version</b>
        Title   : version
        Usage   : exit if $prog-&gt;version() &lt; 0.9.4
        Function: Determine the version number of the program
        Example :
        Returns : float or undef
        Args    : none

   <b>new</b>
        Title   : new
        Usage   : my $msaprobs = Bio::Tools::Run::Alignment::MSAProbs-&gt;new();
        Function: Constructor
        Returns : Bio::Tools::Run::Alignment::MSAProbs
        Args    : -outfile =&gt; $outname

   <b>run</b>
        Title   : run
        Usage   : my $output = $application-&gt;run(\@seqs);
        Function: Generic run of an application
        Returns : Bio::SimpleAlign object
        Args    : Arrayref of Bio::PrimarySeqI objects or
                  a filename to run on

   <b>align</b>
        Title   : align
        Usage   :
               $inputfilename = 't/data/cysprot.fa';
               $aln = $factory-&gt;align($inputfilename);
       or
               $seq_array_ref = \@seq_array;
               # @seq_array is array of Seq objs
               $aln = $factory-&gt;align($seq_array_ref);
        Function: Perform a multiple sequence alignment
        Returns : Reference to a SimpleAlign object containing the
                  sequence alignment.
        Args    : Name of a file containing a set of unaligned fasta sequences
                  or else an array of references to Bio::Seq objects.

        Throws an exception if argument is not either a string (eg a
        filename) or a reference to an array of Bio::Seq objects.  If
        argument is string, throws exception if file corresponding to string
        name can not be found. If argument is Bio::Seq array, throws
        exception if less than two sequence objects are in array.

   <b>error_string</b>
        Title   : error_string
        Usage   : $obj-&gt;error_string($newval)
        Function: Where the output from the last analysus run is stored.
        Returns : value of error_string
        Args    : newvalue (optional)

   <b>infile</b>
        Title   : infile
        Usage   : $prog-&gt;infile($filename)
        Function: get/set the fasta (and only a fasta) file to run on
                  or the array reference containing the Bio::SeqI objects
        Returns : name of input sequence file or object array ref
        Args    : name of input sequence file or object array ref

   <b>outfile</b>
        Title   : outfile
        Usage   : $prog-&gt;outfile($filename)
        Function: get/set the file to save output to
        Returns : outfile name if set
        Args    : newvalue (optional)

   <b>annot_file</b>
        Title   : annot_file
        Usage   : $prog-&gt;annot_file($filename)
        Function: get/set the file name to write the MSA annotation to
        Returns : filename or undef
        Args    : filename (optional)

   <b>num_threads</b>
        Title   : num_threads
        Usage   : $prog-&gt;num_threads($cores)
        Function: get/set number of cores on your machine
        Returns : integer
        Args    : integer (optional; executable auto-detects)

   <b>consistency</b>
        Title   : consistency
        Usage   : $prog-&gt;consistency($passes)
        Function: get/set the number of consistency transformation passes
        Returns : integer
        Args    : integer 0..5, [default 2] (optional)

   <b>iterations</b>
        Title   : iterations
        Usage   : $prog-&gt;iterations($passes)
        Function: get/set the number of iterative-refinement passes
        Returns : integer
        Args    : integer 0..1000, [default 10] (optional)

   <b>alignment_order</b>
        Title   : alignment_order
        Usage   : $prog-&gt;alignment_order($bool)
        Function: specify whether or not to output aligned sequences in
                  alignment order, not input order
        Returns : boolean
        Args    : boolean [default: off] (optional)

   <b>clustalw</b>
        Title   : clustalw
        Usage   : $prog-&gt;clustalw($bool)
        Function: write output in clustalw format; makes no sense unless
                  outfile() is also specified
        Returns : boolean
        Args    : boolean [default: off] (optional)

</pre><h4><b>Bio::Tools::Run::WrapperBase</b> <b>methods</b></h4><pre>
   <b>no_param_checks</b>
        Title   : no_param_checks
        Usage   : $obj-&gt;no_param_checks($newval)
        Function: Boolean flag as to whether or not we should
                  trust the sanity checks for parameter values
        Returns : value of no_param_checks
        Args    : newvalue (optional)

   <b>save_tempfiles</b>
        Title   : save_tempfiles
        Usage   : $obj-&gt;save_tempfiles($newval)
        Function:
        Returns : value of save_tempfiles
        Args    : newvalue (optional)

   <b>outfile_name</b>
        Title   : outfile_name
        Usage   : my $outfile = $msaprobs-&gt;outfile_name();
        Function: Get the name of the output file from a run
                  (if you wanted to do something special)
        Returns : string
        Args    : none

   <b>tempdir</b>
        Title   : tempdir
        Usage   : my $tmpdir = $self-&gt;tempdir();
        Function: Retrieve a temporary directory name (which is created)
        Returns : string which is the name of the temporary directory
        Args    : none

   <b>cleanup</b>
        Title   : cleanup
        Usage   : $msaprobs-&gt;cleanup();
        Function: Will cleanup the tempdir directory
        Returns : none
        Args    : none

   <b>io</b>
        Title   : io
        Usage   : $obj-&gt;io($newval)
        Function:  Gets a L&lt;Bio::Root::IO&gt; object
        Returns : L&lt;Bio::Root::IO&gt;
        Args    : none

</pre><h4><b>Private</b> <b>Methods</b></h4><pre>
   <b>_run</b>
        Title   :  _run
        Usage   :  Internal function, not to be called directly
        Function:  makes actual system call to msaprobs program
        Example :
        Returns : nothing; msaprobs output is written to a
                  temporary file OR specified output file
        Args    : Name of a file containing a set of unaligned fasta sequences
                  and hash of parameters to be passed to msaprobs

   <b>_setinput</b>
        Title   :  _setinput
        Usage   :  Internal function, not to be called directly
        Function:  Create input file for msaprobs program
        Example :
        Returns : name of file containing msaprobs data input AND
        Args    : Arrayref of Seqs or input file name

   <b>_setparams</b>
        Title   :  _setparams
        Usage   :  Internal function, not to be called directly
        Function:  Create parameter inputs for msaprobs program
        Example :
        Returns : parameter string to be passed to msaprobs
                  during align
        Args    : name of calling object

   <b>aformat</b>
        Title   : aformat
        Usage   : my $alignmentformat = $self-&gt;aformat();
        Function: Get/Set alignment format
        Returns : string
        Args    : string

perl v5.34.0                                       2022-02-10             <u>Bio::Tools::Run...nment::<a href="../man3pm/MSAProbs.3pm.html">MSAProbs</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>