<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Run::Alignment::Pal2Nal - Wrapper for Pal2Nal</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-perl-run-perl">libbio-perl-run-perl_1.7.3-8_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Run::Alignment::Pal2Nal - Wrapper for Pal2Nal

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::Tools::Run::Alignment::Pal2Nal;

         # Make a Pal2Nal factory
         $factory = Bio::Tools::Run::Alignment::Pal2Nal-&gt;new();

         # Run Pal2Nal with a protein alignment file and a multi-fasta nucleotide
         # file
         my $aln = $factory-&gt;run($protein_alignfilename, $nucleotide_filename);

         # or with Bioperl objects
         $aln = $factory-&gt;run($protein_bio_simplalign, [$nucleotide_bio_seq1,
                                                        $nucleotide_bio_seq2]);

         # combinations of files/ objects are possible

         # $aln isa Bio::SimpleAlign of the nucleotide sequences aligned according to
         # the protein alignment

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This is a wrapper for running the Pal2Nal perl script by Mikita Suyama. You can get details here:
       <a href="http://coot.embl.de/pal2nal/">http://coot.embl.de/pal2nal/</a>. Pal2Nal is used for aligning a set of nucleotide sequences based on an
       alignment of their translations.

       You can try supplying normal pal2nal command-line arguments to <b>new()</b>, eg.  <b>new()</b> or calling arg-named
       methods (excluding the initial hyphen, eg.  $factory-&gt;(1) to set the - arg).

       You will need to enable this Pal2Nal wrapper to find the pal2nal.pl script.  This can be done in (at
       least) three ways:

        1. Make sure the script is in your path.
        2. Define an environmental variable PAL2NALDIR which is a
           directory which contains the script:
           In bash:

           export PAL2NALDIR=/home/username/pal2nal/

           In csh/tcsh:

           setenv PAL2NALDIR /home/username/pal2nal

        3. Include a definition of an environmental variable PAL2NALDIR in
           every script that will use this Pal2Nal wrapper module, e.g.:

           BEGIN { $ENV{PAL2NALDIR} = '/home/username/pal2nal/' }
           use Bio::Tools::Run::Alignment::Pal2Nal;

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         <a href="http://redmine.open-bio.org/projects/bioperl/">http://redmine.open-bio.org/projects/bioperl/</a>

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Sendu</b> <b>Bala</b></h4><pre>
       Email <a href="mailto:bix@sendu.me.uk">bix@sendu.me.uk</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>program_name</b>
        Title   : program_name
        Usage   : $factory&gt;program_name()
        Function: holds the program name
        Returns : string
        Args    : None

   <b>program_dir</b>
        Title   : program_dir
        Usage   : $factory-&gt;program_dir(@params)
        Function: returns the program directory, obtained from ENV variable.
        Returns : string
        Args    : None

   <b>new</b>
        Title   : new
        Usage   : $factory = Bio::Tools::Run::Alignment::Pal2Nal-&gt;new()
        Function: creates a new Pal2Nal factory.
        Returns : Bio::Tools::Run::Alignment::Pal2Nal
        Args    : Most options understood by pal2nal.pl can be supplied as key =&gt;
                  value pairs.

                  These options can NOT be used with this wrapper:
                  -output
                  -html
                  -h
                  -nostderr

   <b>run</b>
        Title   : run
        Usage   : $result = $factory-&gt;run($protein_align_file, $multi_fasta_nucleotide);
                  -or-
                  $result = $factory-&gt;run($prot_align_object, [$bioseq_object1, ...]);
        Function: Runs pal2nal on a protein alignment and set of nucleotide sequences.
        Returns : Bio::SimpleAlign;
        Args    : The first argument represents a protein alignment, the second
                  argument a set of nucleotide sequences.
                  The alignment can be provided as an alignment file readable by
                  Bio::AlignIO, or a Bio::Align::AlignI compliant object (eg. a
                  Bio::SimpleAlign).
                  The nucleotide sequences can be provided as a single filename of a
                  fasta file containing multiple nucleotide sequences, or an array ref
                  of filenames, each file containing one sequence. Alternatively, an
                  array ref of Bio::PrimarySeqI compliant objects can be supplied.

                  In all cases, the protein alignment sequence names must correspond to
                  the names of the supplied nucleotide sequences.

   <b>_setparams</b>
        Title   : _setparams
        Usage   : Internal function, not to be called directly
        Function: Creates a string of params to be used in the command string
        Returns : string of params
        Args    : alignment and tree file names

perl v5.34.0                                       2022-02-10              <u>Bio::Tools::Ru...gnment::<a href="../man3pm/Pal2Nal.3pm.html">Pal2Nal</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>