<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::DB::Bam::Alignment -- The SAM/BAM alignment object</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-samtools-perl">libbio-samtools-perl_1.43-5build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::DB::Bam::Alignment -- The SAM/BAM alignment object

</pre><h4><b>SYNOPSIS</b></h4><pre>
        use Bio::DB::Sam;

        my $sam = Bio::DB::Sam-&gt;new(-fasta=&gt;"data/ex1.fa",
                                    -bam  =&gt;"data/ex1.bam");

        my @alignments = $sam-&gt;get_features_by_location(-seq_id =&gt; 'seq2',
                                                        -start  =&gt; 500,
                                                        -end    =&gt; 800);
        for my $a (@alignments) {
           my $seqid  = $a-&gt;seq_id;
           my $start  = $a-&gt;start;
           my $end    = $a-&gt;end;
           my $strand = $a-&gt;strand;
           my $ref_dna= $a-&gt;dna;

           my $query_start  = $a-&gt;query-&gt;start;
           my $query_end    = $a-&gt;query-&gt;end;
           my $query_strand = $a-&gt;query-&gt;strand;
           my $query_dna    = $a-&gt;query-&gt;dna;

           my $cigar     = $a-&gt;cigar_str;
           my @scores    = $a-&gt;qscore;     # per-base quality scores
           my $score     = $a-&gt;qstring;    # TAM-style quality string
           my $match_qual= $a-&gt;qual;       # quality of the match

           my $paired = $a-&gt;get_tag_values('PAIRED');
        }

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The Bio::DB::Bam::Alignment and Bio::DB::Bam::AlignWrapper classes together represent an alignment
       between a sequence read (the "query") and a reference sequence (the "target"). Bio::DB::Bam::Alignment
       adheres strictly to the C-level BAM library's definition of a bam1_t* and is used in the Bio::DB::Sam
       low-level API The latter adds convenience methods that make it similar to a BioPerl Bio::SeqFeatureI
       object. This manual page describes both.

</pre><h4><b>High-level</b> <b>Bio::DB::Bam::Alignment</b> <b>methods</b></h4><pre>
       These methods are provided by Bio::DB::Bam::Alignment, and are intended to be compatible with the
       Bio::SeqFeatureI interfaces. Note that these objects are <b>not</b> compatible with Bio::Align::AlignI, as the
       BAM API is fundamentally incompatible with the BioPerl API for alignments (the first deals with the
       alignment of a single read against the reference sequence, while the second deals with a multiple
       alignment).

       Note that the high-level API return Bio::DB::Bam::AlignWrapper objects <b>except</b> in the case of the callback
       to the <b>fast_pileup()</b> method. In this case only, the object returned by calling $pileup-&gt;b() is a
       Bio::DB::Bam::Alignment object for performance reasons.

       $seq_id = $align-&gt;seq_id
           Return  the  seq_id  of  the  reference  (target)  sequence.  This  method  is  only available in the
           Bio::DB::Bam::AlignWrapper extension.

       $start = $align-&gt;start
           Return the start of the alignment in 1-based reference sequence coordinates.

       $end = $align-&gt;end
           Return the end of the alignment in 1-based reference sequence coordinates.

       $len = $align-&gt;length
           Return the length of the alignment on the reference sequence.

       $mseqid = $align-&gt;mate_seq_id
           Return the seq_id of the mate's reference (target) sequence. This method is  only  available  in  the
           Bio::DB::AlignWrapper extension.

       $mstart = $align-&gt;mate_start
           For paired reads, return the start of the mate's alignment in 1-based reference sequence coordinates.

       $mend = $align-&gt;mate_end
           For  paired  reads,  return  the  end  position of the mate's alignment in 1-based reference sequence
           coordinates.

       $mlen = $align-&gt;mate_len
           For mate-pairs, retrieve the length of the mate's alignment on the reference sequence.

       $strand = $align-&gt;strand
           Return the strand of the alignment as -1 for reversed, +1 for forward.

           NOTE: In versions 1.00-1.06, this method always returned +1. As of version  1.07,  this  behavior  is
           fixed.

       $mstrand = $align-&gt;mstrand
           If the read has a mate pair, return the strand of the mate in the format -1 or +1.

       $ref_dna        = $align-&gt;dna
           Returns  the  <b>reference</b>  sequence's  DNA  across  the  aligned region. If an MD tag is present in the
           alignment, it will be used preferentially  to  reconstruct  the  reference  sequence.  Otherwise  the
           reference DNA access object passed to Bio::DB::Sam-&gt;<b>new()</b> will be used.

       $ref_dna        = $align-&gt;seq
           The  <b>reference</b>  sequence's  DNA as a Bio::PrimarySeqI object (useful for passing to BioPerl functions
           and for calculating subsequences and reverse complements).

       $query = $align-&gt;query
           This method returns a Bio::DB::Alignment::Query object that can be used to retrieve information about
           the query sequence. The next few entries show how to use this object.

       $read_name = $align-&gt;query-&gt;name
           The name of the read.

       $q_start   = $align-&gt;query-&gt;start
           This returns the start position of the query (read) sequence in 1-based coordinates. It  acts  via  a
           transient   Bio::DB::Bam::Query   object  that  is  provided  for  Bio::Graphics  compatibility  (see
           Bio::Graphics).

       $q_end     = $align-&gt;query-&gt;end
           This returns the end position of the query sequence in 1-based coordinates.

       $q_len     = $align-&gt;query-&gt;length
           Return the length of the alignment on the read.

       $scores = $align-&gt;query-&gt;score
           Return an array reference containing the unpacked quality scores for each base of the query sequence.
           The length of this array reference will be equal to the length of the read.

       $read_dna = $align-&gt;query-&gt;dna
           The read's DNA string.

       $read_seq = $align-&gt;query-&gt;seq
           The read's DNA as a Bio::PrimarySeqI object.

       $target  = $align-&gt;target;
           The <b>target()</b> method is similar to <b>query()</b>, except that it follows Bio::AlignIO conventions for how to
           represent minus strand alignments. The object returned has <b>start()</b>, <b>end()</b>, <b>qscore()</b>, <b>dna()</b> and  <b>seq()</b>
           methods,  but  for  minus  strand  alignments  the  sequence will be represented as it appears on the
           reverse strand, rather than on the forward strand. This has the advantage of giving you the  read  as
           it came off the machine, before being reverse complemented for use in the SAM file.

       $query   = $align-&gt;hit
           The  <b>hit()</b>  method is identical to <b>target()</b> and returns information about the read. It is present for
           compatibility with some of the Bio::Graphics glyphs, which use <b>hit()</b> to represent  the  non-reference
           sequence in aligned sequences.

       $primary_id = $align-&gt;primary_id
           This   method   synthesizes   a   unique   ID   for   the   alignment   which   can   be   passed  to
           $sam-&gt;<b>get_feature_by_id()</b> to retrieve the alignment at a later date.

       @tags = $align-&gt;get_all_tags
           Return all tag names known to this alignment. This includes SAM flags such as M_UNMAPPED, as well  as
           auxiliary  flags  such  as H0. The behavior of this method depends on the value of -expand_flags when
           the SAM object was created. If false (the default), then the standard SAM flags will be  concatenated
           together  into a single string and stored in a tag named 'FLAGS'. The format of this tag value is the
           list   of   one   or   more   flag   constants   separated   by   the   "|"   character,    as    in:
           "PAIRED|MAP_PAIR|REVERSED|SECOND_MATE".  If  -expand_flags  was  true, then each flag becomes its own
           named tag, such as "MAP_PAIR".

       @values = $align-&gt;get_tag_values($tag)
           Given a tag name, such as 'PAIRED' or 'H0', return its value(s). -expand_flags must be true in  order
           to  use  the  standard  SAM  flag constants as tags. Otherwise, they can be fetched by asking for the
           "FLAGS" tag, or by using the low-level methods described below.

       $is_true = $align-&gt;has_tag($tag)
           Return true if the alignment has the indicated tag.

       $string = $align-&gt;cigar_str
           Return the CIGAR string for this alignment in conventional human readable format (e.g. "M34D1M1").

       $arrayref = $align-&gt;cigar_array
           Return a reference to an array representing the CIGAR string. This is an array of  arrays,  in  which
           each subarray consists of a CIGAR operation and a count. Example:

            [ ['M',34], ['D',1], ['M1',1] ]

       ($ref,$matches,$query) = $align-&gt;padded_alignment
           Return  three  strings  that  show  the alignment between the reference sequence (the target) and the
           query. It will look like this:

            $ref     AGTGCCTTTGTTCA-----ACCCCCTTGCAACAACC
            $matches ||||||||||||||     |||||||||||||||||
            $query   AGTGCCTTTGTTCACATAGACCCCCTTGCAACAACC

       $str = $align-&gt;aux
           Returns the text version of the SAM tags, e.g.  "XT:A:M NM:i:2 SM:i:37 AM:i:37 XM:i:1  XO:i:1  XG:i:1
           MD:Z:6^C0A47"

       $str = $align-&gt;tam_line
           Returns  the  TAM  (text) representation of the alignment (available in the high-level "AlignWrapper"
           interface only).

       $tag = $align-&gt;primary_tag
           This is provided for Bio::SeqFeatureI compatibility. Return the string "match".

       $tag = $align-&gt;source_tag
           This is provided for Bio::SeqFeatureI compatibility. Return the string "sam/bam".

       @parts = $align-&gt;get_SeqFeatures
           Return subfeatures of this alignment. If you have fetched a "read_pair" feature, this will be the two
           mate pair objects (both of type Bio::DB::Bam::AlignWrapper). If you have -split_splices set  to  true
           in  the  Bio::DB::Sam  database,  calling  <b>get_SeqFeatures()</b>  will  return  the  components  of split
           alignments. See "Bio::DB::Sam Constructor and basic accessors" in Bio::DB::Sam for an example of  how
           to use this.

</pre><h4><b>Low-level</b> <b>Bio::DB::Bam::Alignment</b> <b>methods</b></h4><pre>
       These    methods    are   available   to   objects   of   type   Bio::DB::Bam::Alignment   as   well   as
       Bio::DB::Bam::AlignWrapper and closely mirror the native C API.

       $align = Bio::DB::Bam::Alignment-&gt;new
           Create a new, empty alignment object. This is usually only needed when iterating through a  TAM  file
           using Bio::DB::Tam-&gt;<b>read1()</b>.

       $tid = $align-&gt;tid( [$new_tid] )
           Return  the  target  ID  of  the  alignment.  Optionally you may change the tid by providing it as an
           argument (currently this is the only field that you can change; the functionality was implemented  as
           a proof of principle).

       $read_name = $align-&gt;qname
           Returns the name of the read.

       $pos = $align-&gt;pos
           0-based leftmost coordinate of the aligned sequence on the reference sequence.

       $end = $align-&gt;calend
           The  0-based  rightmost  coordinate  of  the  aligned sequence on the reference sequence after taking
           alignment gaps into account.

       $len = $align-&gt;cigar2qlen
           The length of the query sequence calculated from the CIGAR string.

       $quality = $align-&gt;qual
           The quality score for the alignment as a whole.

       $flag = $align-&gt;flag
           The bitwise flag field (see the SAM documentation).

       $mtid = $align-&gt;mtid
           For paired reads, the target ID of the mate's alignment.

       $mpos = $align-&gt;mpos
           For paired reads, the 0-based leftmost coordinate of the mate's alignment on the reference sequence.

       $n_cigar = $align-&gt;n_cigar
           Number of CIGAR operations in this alignment.

       $length = $align-&gt;l_qseq
           The length of the query sequence (the read).

       $dna = $align-&gt;qseq
           The actual DNA sequence of the query. As in the SAM file, reads that are aligned to the minus  strand
           of the reference are returned in reverse complemented form.

       $score_str = $align-&gt;_qscore
           A  packed  binary string containing the quality scores for each base of the read. It will be the same
           length as the DNA. You may unpack it using unpack('C*',$score_str), or use  the  high-level  <b>qscore()</b>
           method.

       $score_arry = $align-&gt;qscore
       @score_arry = $align-&gt;qscore
           In a scalar context return an array reference containing the unpacked quality scores for each base of
           the  query  sequence.  In  a  list  context  return  a  list of the scores. This array is in the same
           orientation as the reference sequence.

       $score_str = $align-&gt;qstring
           Returns the quality string in the same format used in the SAM (TAM) file.

       $length = $align-&gt;isize
           The calculated insert size for mapped paired reads.

       $length = $align-&gt;l_aux
           The length of the align "auxiliary" data.

       $value = $align-&gt;aux_get("tag")
           Given an auxiliary tag, such as "H0", return its value.

       @keys  = $align-&gt;aux_keys
           Return the list of auxiliary tags known to this alignment.

       $data = $align-&gt;data
           Return a packed string containing the alignment data (sequence, quality scores and cigar string).

       $length = $align-&gt;data_len
           Return the current length of the alignment data.

       $length = $align-&gt;m_data
           Return the maximum length of the alignment data.

       $is_paired = $align-&gt;paired
           Return true if the aligned read is part of a mate/read pair (regardless of whether the mate mapped).

       $is_proper = $align-&gt;proper_pair
           Return true if the aligned read is part of a mate/read pair and both partners mapped to the reference
           sequence.

       $is_unmapped = $align-&gt;unmapped
           Return true if the read failed to align.

       $mate_is_unmapped = $align-&gt;munmapped
           Return true if the read's mate failed to align.

       $reversed = $align-&gt;reversed
           Return true if the aligned read was reverse complemented prior to aligning.

       $mate_reversed = $align-&gt;mreversed
           Return true if the aligned read's mate was reverse complemented prior to aligning.

       $isize = $align-&gt;isize
           For mate-pairs, return the computed insert size.

       $arrayref = $align-&gt;cigar
           This returns the CIGAR data in its native BAM format. You will receive  an  arrayref  in  which  each
           operation  and count are packed together into an 8-bit structure. To decode each element you must use
           the following operations:

            use Bio::DB::Sam::Constants;
            my $c   = $align-&gt;cigar;
            my $op  = $c-&gt;[0] &amp; BAM_CIGAR_MASK;
            my $len = $c-&gt;[0] &gt;&gt; BAM_CIGAR_SHIFT;

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Bio::Perl, Bio::DB::Sam, Bio::DB::Bam::Constants

</pre><h4><b>AUTHOR</b></h4><pre>
       Lincoln Stein &lt;<a href="mailto:lincoln.stein@oicr.on.ca">lincoln.stein@oicr.on.ca</a>&gt;.  &lt;<a href="mailto:lincoln.stein@bmail.com">lincoln.stein@bmail.com</a>&gt;

       Copyright (c) 2009-2015 Ontario Institute for Cancer Research.

       This package and its accompanying libraries are free software; you can redistribute it and/or  modify  it
       under  the  terms  of the Artistic License 2.0, the Apache 2.0 License, or the GNU General Public License
       (version 1 or higher).  Refer to LICENSE for the full license text.

perl v5.40.0                                       2024-10-20                       <u>Bio::DB::Bam::<a href="../man3pm/Alignment.3pm.html">Alignment</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>