<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>TFBS::Matrix::Alignment - class for alignment of PFM objects</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libtfbs-perl">libtfbs-perl_0.7.1+ds-4build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       TFBS::Matrix::Alignment - class for alignment of PFM objects

</pre><h4><b>SYNOPSIS</b></h4><pre>
       â€¢
        Making  an  alignment:  (See  documentation of individual TFBS::DB::* modules to learn how to connect to
        different types of pattern databases and retrieve TFBS::Matrix::* objects from them.)

            my $db_obj = TFBS::DB::JASPAR2-&gt;new
                            (-connect =&gt; ["dbi:mysql:JASPAR2:myhost",
                                          "myusername", "mypassword"]);
            my $pfm1 = $db_obj-&gt;get_Matrix_by_ID("M0001", "PFM");
            my $pfm2 = $db_obj-&gt;get_Matrix_by_ID("M0002", "PFM");

            my $alignment= new TFBS::Matrix::Alignment(
                                              -pfm1=&gt;$pfm1,
                                              -pfm2=&gt;$pfm2,
                                              -binary=&gt;"/TFBS/Ext/matrix_aligner",
                                            );

</pre><h4><b>DESCRIPTION</b></h4><pre>
       TFBS::Matrix::Alignment is a class for representing and performing pairwise alignments  of  profiles  (in
       the  form  of  TFBS::PFM  objects) Alignments are preformed using a semi-global variant of the Needleman-
       Wunsch algorithm that only permits the opening of one internal gap.  Fore  reference,  the  algorithm  is
       described in Sandelin et al Funct Integr Genomics. 2003 Jun 25

</pre><h4><b>FEEDBACK</b></h4><pre>
       Please send bug reports and other comments to the author.

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Boris</b> <b>Lenhard</b></h4><pre>
       Boris Lenhard &lt;<a href="mailto:Boris.Lenhard@cgb.ki.se">Boris.Lenhard@cgb.ki.se</a>&gt;

</pre><h4><b>APPENDIX</b></h4><pre>
       The  rest  of the documentation details each of the object methods. Internal methods are preceded with an
       underscore.

   <b>new</b>
        Title   : new
        Usage   : my $alignment = TFBS::Matrix::Alignment-&gt;new(%args)
        Function: constructor for the TFBS::Matrix::Alignment object
        Returns : a new TFBS::Matrix::Alignment object
        Args    : # you must specify:

                  -pfm1,      # a TFBS::Matrix::PFM object
                  -pfm2,      # another TFBS::Matrix::PFM object
                   -binary,  # a valid path to the comparison algorithm (matrixalign)

                  #######

                  -ext_penalty            #OPTIONAL gap extension penalty in Needleman-Wunsch
                                           algorithmstring. Default 0.01
                  -open_penalty,          #OPTIONAL gap opening penalty in Needleman-Wunsch
                                           algorithmstring. Default 3.0

   <b>score</b>
        Title   : score
        Usage   : my $score = $alignmentobject-&gt;score();
        Function: access an alignment score (where each aligned position can contribute max 2)
        Returns : a floating point number
        Args    : none

   <b>score</b>
        Title   : gaps
        Usage   : my $nr_of_gaps = $alignmentobject-&gt;gaps();
        Function: access the number of gaps in an alignment
        Returns : an integer
        Args    : none

   <b>length</b>
        Title   : length
        Usage   : my $length = $alignmentobject-&gt;length();
        Function: access the length of an alignment (ie thenumber of aligned positions)
        Returns : an integer
        Args    : none

   <b>strand</b>
        Title   : strand
        Usage   : my $strand = $alignmentobject-&gt;strand();
        Function: access the oriantation of the aligned patterns:
                  ++= oriented as input
                  +-= second pattern is reverse-complemented
        Returns : a string
        Args    : none

   <b>alignment</b>
        Title   : alignment
        Usage   : my $alignment_string = $alignmentobject-&gt;alignment();
        Function: access a string describing the alignment
        Returns : an string, where each number refers to a position in respective PFM.
        Position numbering is according to orientation: ie if the second profile is
        reversed, position 1 corresponds to the last position in the input profile.
        Gaps are denoted as - .

        RXR-VDR           -       1       2       3       -     4      5     -
        PPARgamma-RXRal   1       2       3       4       5     6      7     8

        Args    : none

</pre><h4><b>POD</b> <b>ERRORS</b></h4><pre>
       Hey! <b>The</b> <b>above</b> <b>document</b> <b>had</b> <b>some</b> <b>coding</b> <b>errors,</b> <b>which</b> <b>are</b> <b>explained</b> <b>below:</b>

       Around line 39:
           You forgot a '=back' before '=head1'

perl v5.40.0                                       2024-10-20                       <u>TFBS::Matrix::<a href="../man3pm/Alignment.3pm.html">Alignment</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>