<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>indexdb_rna - tool for filtering, mapping and OTU-picking NGS reads (indexdb)</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/sortmerna">sortmerna_2.1-5_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       indexdb_rna - tool for filtering, mapping and OTU-picking NGS reads (indexdb)

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>indexdb_rna</b> <b>--ref</b> db.fasta,db.idx [OPTIONS]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       SortMeRNA  is  a  biological sequence analysis tool for filtering, mapping and OTU-picking NGS reads. The
       core algorithm is based on approximate seeds and allows for fast and  sensitive  analyses  of  nucleotide
       sequences.  The main application of SortMeRNA is filtering rRNA from metatranscriptomic data.  Additional
       applications include OTU-picking and taxonomy assignation available through QIIME v1.9+ (<a href="http://qiime.org">http://qiime.org</a>
       - v1.9.0-rc1).

       SortMeRNA takes as input a file of reads (fasta or fastq  format)  and  one  or  multiple  rRNA  database
       file(s), and sorts apart rRNA and rejected reads into two files specified by the user. Optionally, it can
       provide  high  quality  local  alignments  of  rRNA reads against the rRNA database. SortMeRNA works with
       Illumina, 454, Ion Torrent and PacBio data, and can produce SAM and BLAST-like alignments.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>MANDATORY</b> <b>OPTIONS</b>
       <b>--ref</b> <u>STRING,STRING</u>
              FASTA reference file, index file
              Example:
              <b>--ref</b> <u>/path/to/file1.fasta,/path/to/index1</u>
              If passing multiple reference sequence files, separate them by ':'
              Example:
              <b>--ref</b> <u>/path/f1.fasta,/path/index1:/path/f2.fasta,path/index2</u>

   <b>OPTIONAL</b> <b>OPTIONS</b>
       <b>--fast</b> <u>BOOL</u>
              suggested option for aligning ~99% related species (default: off)

       <b>--sensitive</b> <u>BOOL</u>
              suggested option for aligning ~75-98% related species (default: on)

       <b>--tmpdir</b> <u>STRING</u>
              directory where to write temporary files

       <b>-m</b> <u>INT</u> the amount of memory (in Mbytes) for building the index (default: 3072)

       <b>-L</b> <u>INT</u> seed length (default: 18)

       <b>--max_pos</b> <u>INT</u>
              maximum number of positions to store for each unique L-mer (default: 10000,  setting  <b>--max_pos</b>  0
              will store all positions)

       <b>-v</b> <u>BOOL</u>
              verbose

       <b>-h</b> <u>BOOL</u>
              help

indexdb_rna 2.0                                    August 2015                                    <u><a href="../man1/INDEXDB_RNA.1.html">INDEXDB_RNA</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>