<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>NHGRI::Blastall - Perl extension for running and parsing NCBI's BLAST 2.x</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libnhgri-blastall-perl">libnhgri-blastall-perl_0.66-4_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       NHGRI::Blastall - Perl extension for running and parsing NCBI's BLAST 2.x

</pre><h4><b>SYNOPSIS</b></h4><pre>
</pre><h4><b>DESCRIPTION</b></h4><pre>
       If you have NCBI's BLAST2 or WU-BLAST installed locally and your environment is already setup you can use
       Perl's object-oriented capabilities to run your BLASTs.  Also if you have a blastcl3 binary from the
       toolkit (or binaries from our FTP site) you can run BLAST over the network.  There are also methods to
       blast single sequences against each other using the bl2seq binaries (also in the toolkit and binaries).
       You can blast one sequence against a library of sequences using the blast_one_to_many method.  You can
       format databases with formatdb method.  You can also have NHGRI::Blastall read existing BLAST reports.
       If you have a database of repetitive DNA or other DNA you would like to mask out, you can use the mask
       method to mask the data against these databases. You can then use either the filter or result methods to
       parse the report and access the various elements of the data.

       RUNNING NEW BLASTS
             use NHGRI::Blastall;
             my $b = new NHGRI::Blastall;
             # If you are running NCBI's Local BLAST
             $b-&gt;blastall( p =&gt; 'blastn',
                           d =&gt; 'nr',
                           i =&gt; 'infile',
                           o =&gt; 'outfile'
                         );
             # If you are running NCBI's blastcl3 network client
             $b-&gt;blastcl3( p =&gt; 'blastn',
                           d =&gt; 'nr',
                           i =&gt; 'infile',
                           o =&gt; 'outfile'
                         );
             # If you are running WU-BLAST locally
             $b-&gt;wu_blastall( p      =&gt; 'blastn',
                              d      =&gt; 'nr',
                              nogap  =&gt; '!',     #use ! for arguments w/o parameter
                              i      =&gt; 'infile',
                              o      =&gt; 'outfile'
                            );

             See BLASTALL for more info

       BLASTING 2 SEQUENCES
             use NHGRI::Blastall;
             my $b = new NHGRI::Blastall;
             $b-&gt;bl2seq(i =&gt; 'seq1',
                        j =&gt; 'seq2',
                        p =&gt; 'tblastx'
                       );

             See BL2SEQ for more info

       BLASTING 1 SEQUENCE AGAINST A FASTA LIBRARY OF SEQUENCES
             # a library is a FASTA file with multiple FASTA formatted sequences
             use NHGRI::Blastall;
             my $b = new NHGRI::Blastall;
             $b-&gt;blast_one_to_many(i =&gt; 'seq1',
                                   d =&gt; 'seq2.lib',
                                   p =&gt; 'tblastx',
                                  );

             See BLAST_ONE_TO_MANY for more info

       INITIALIZING EXISTING BLAST REPORTS
             use NHGRI::Blastall;
             my $b = new NHGRI::Blastall;
             $b-&gt;read_report('/path/to/report');

       MASKING SEQUENCES
             use NHGRI::Blastall;
             my $b = new NHGRI::Blastall;
             $masked_seq = $b-&gt;mask( type =&gt; 'wu_blastall',
                                     p    =&gt; 'blastn',
                                     d    =&gt; 'alu',
                                     i    =&gt; 'infile'
                                   );

             See MASKING for more info

       CREATING BLAST INDEXES
             use NHGRI::Blastall;
             my $b = new NHGRI::Blastall;
             $b-&gt;formatdb( i =&gt; 'est',
                           p =&gt; 'F',
                           o =&gt; 'T',
                         );

             See FORMATDB for more info

       PRINTING REPORTS
             $b-&gt;print_report();
             # this method only opens the report and prints.  It does not print
             # summary reports

       FILTERING BLAST RESULTS
             @hits = $b-&gt;filter( scores     =&gt; '38.2',
                                 identities =&gt; '.98'
                               );

             # returns an array of hash references.
             See HASHREF for more info on manipulating the results.
             See FILTERING for more info on using the filter method

       GETTING AT ELEMENTS
             @ids = $b-&gt;result('id');
             @scores = $b-&gt;result('scores',$ids[0]);  # second param must be an id

             See RESULT for more info on using the result method
             See ELEMENTS for element names

       GETTING AT ALL THE DATA
             @results = $b-&gt;result();  # returns an array of hashes

             See HASHREF for information on the array of hashes that is returned.
             See DUMP RESULTS to see how to work with the array of hashes

       ADJUSTING THE DEFLINE REGEX
             $b = new NHGRI::Blastall (-DB_ID_REGEX =&gt; '[^ ]+');

             See DB_ID_REGEX for more info

</pre><h4><b>BLASTALL</b></h4><pre>
       This  method  provides  a simple object oriented frontend to BLAST.  This module works with either NCBI's
       blastall binary distributed with BLAST 2.x, WU-BLAST or over  the  web  through  NCBI's  Web  Site.   The
       blastall  function  accepts  as a parameter an anonymous hash with keys that are the command line options
       (See BLASTALL OPTIONS) and values which are the corresponding values to those options.  You may  want  to
       set  the  BLASTALL  variable in Blastall.pm to the full path of your `blastall' binary, especially if you
       will be running scripts as cron jobs or if blastall is not in the system path.

</pre><h4><b>BLASTALL</b> <b>OPTIONS</b></h4><pre>
       For wu_blastall you need to use NCBI type switches for the following
          [C<b>-i</b>] for infile
          [C<b>-o</b>] for outfile
          [C<b>-p</b>] for program
          [C<b>-d</b>] for database the rest of the parameters MUST be the parameters available through WU-BLAST  (e.g.
       -sump,  -nogap  -compat1.4,  etc.)   use  a  `!'  to specify that an argument has no parameters.  See the
       example at the top of the manpage.

       These are the options that NCBI's blastall and binary accepts and these are the  same  options  that  are
       accepted by the blastall and blastcl3 methods.  NOTE: You must set the proper environmental variables for
       the blastall method to work (BLASTDB,BLASTMAT).

       •   <b>p</b> =&gt; Program Name

       •   <b>d</b> =&gt; Database                                           default=nr

       •   <b>i</b> =&gt; QueryFile

       •   <b>e</b> =&gt; Expectation value (E)                              default=10.0

       •   <b>m</b> =&gt; alignment view                                     default=0
               0 = pairwise,
               1 = master-slave showing identities,
               2 = master-slave no identities,
               3 = flat master-slave, show identities,
               4 = flat master-slave, no identities,
               5 = master-slave no identities and blunt ends,
               6 = flat master-slave, no identities and blunt ends

       •   <b>o</b> =&gt; BLAST report Output File                           default=stdout

       •   <b>F</b> =&gt; Filter query sequence                              default=T
               (DUST with blastn, SEG with others)

       •   <b>G</b> =&gt; Cost to open a gap                                 default=0
               (zero invokes default behavior)

       •   <b>E</b> =&gt; Cost to extend a gap                               default=0
               (zero invokes default behavior)

       •   <b>X</b> =&gt; X dropoff value for gapped alignment (in bits)     default=0
               (zero invokes default behavior)

       •   <b>I</b> =&gt; Show GI's in deflines                              default=F

       •   <b>q</b> =&gt; Penalty for a nucleotide mismatch (blastn only)    default=-3

       •   <b>r</b> =&gt; Reward for a nucleotide match (blastn only)        default=1

       •   <b>v</b> =&gt; Number of one-line descriptions (V)                default=500

       •   <b>b</b> =&gt; Number of alignments to show (B)                   default=250

       •   <b>f</b> =&gt; Threshold for extending hits, default if zero      default=0

       •   <b>g</b> =&gt; Perform gapped alignment (NA with tblastx)         default=T

       •   <b>Q</b> =&gt; Query Genetic code to use                          default=1

       •   <b>D</b> =&gt; DB Genetic code (for tblast[nx] only)              default=1

       •   <b>a</b> =&gt; Number of processors to use                        default=1

       •   <b>O</b> =&gt; SeqAlign file                                      Optional

       •   <b>J</b> =&gt; Believe the query defline                          default=F

       •   <b>M</b> =&gt; Matrix                                             default=BLOSUM62

       •   <b>W</b> =&gt; Word size, default if zero                         default=0

       •   <b>z</b> =&gt; Effective length of the database                   default=0
                (use zero for the real size)

       •   <b>K</b> =&gt; Number of best hits from a region to keep          default=100

       •   <b>L</b> =&gt; Length of region used to judge hits                default=20

       •   <b>Y</b> =&gt; Effective length of the search space               default=0
                (use zero for the real size)

       •   <b>S</b> =&gt; Query strands to search against database           default=3
                (for blast[nx], and tblastx).
                3 is both, 1 is top, 2 is bottom

       •   <b>T</b> =&gt; Produce HTML output [T/F]                          default=F

       •   <b>l</b> =&gt; Restrict search of database to list of GI's [String]

       NOTE: If you do not supply an `o' option (outfile),
             the following environment variables are checked in order:
             `TMPDIR', `TEMP', and `TMP'.
             If one of them is set, outfiles are created relative to the
             directory it specifies.  If none of them are set, the first
             possible one of the following directories is used:
             <a href="file:/var/tmp">/var/tmp</a> , /usr/tmp , /temp , <a href="file:/tmp">/tmp</a> ,
             This file is deleted after the NHGRI::Blastall object is destroyed.
             It is recommended that you create a tmp directory in your home
             directory and set one of the above environmental vars to point
             to this directory and then set the permissions on this directory
             to 0700.  Writing to a "public" tmp directory can have
             security ramifications.

</pre><h4><b>BL2SEQ</b></h4><pre>
       This  method uses the bl2seq binary (distributed with BLAST executables and source) to BLAST one sequence
       against another sequence.  Like the blastall method the bl2seq method accepts the same options  that  the
       bl2seq  binary  accepts.  Run bl2seq without options from the command line to get a full list of options.
       An important note about the options, when running blastx 1st sequence should be nucleotide; when  running
       tblastn 2nd sequence should be nucleotide.

         use NHGRI::Blastall;
         my $b = new NHGRI::Blastall;
         $b-&gt;bl2seq(i =&gt; 'seq1.nt',
                    j =&gt; 'seq2.aa',
                    p =&gt; 'blastx'
                   );

</pre><h4><b>BLAST_ONE_TO_MANY</b></h4><pre>
       This  method  allows  for blasting one sequence against a FASTA library of sequences.  Behind the scenes,
       BLAST indexes are created (in the same directory as the FASTA library) using the provided  FASTA  library
       and the one sequence is used to search against this database.  If the program completes successfully, the
       databases  are  removed.   To compare two sequences, use the bl2seq method which is faster and less messy
       (no tmp indexes).   This method accepts the same options  as  the  blastall  binary  with  the  d  option
       corresponding to the FASTA library.

         use NHGRI::Blastall;
         my $b = new NHGRI::Blastall;
         $b-&gt;blast_one_to_many(i =&gt; 'seq.aa',
                               d =&gt; 'seq.nt.lib',
                               e =&gt; '0.001',
                               p =&gt; 'tblastn',
                              );

</pre><h4><b>MASKING</b></h4><pre>
       Screens  DNA  sequences  in  fasta format against the database specified in the blastall 'd' option.  The
       mask method accepts the same parameters as the blastall method.  Any matches to the masking database will
       be substituted with "N"s.  The mask method returns the masked sequence.   Performs  similar  function  as
       xblast, an old NCBI program written in C.

       Set the type parameter to wu_blastall, blastcl3 or blastall depending on your configuration.

         $masked_seq = $b-&gt;mask( type =&gt; 'blastcl3', # defaults to blastall
                                 p    =&gt; 'blastn',
                                 d    =&gt; 'alu',
                                 i    =&gt; 'infile'
                               );

       To get the mask coordinates back call the mask method in an array context.

           @mask = $b-&gt;mask(p    =&gt; 'blastn',
                            d    =&gt; 'alu',
                            i    =&gt; 'infile'
                           );
           $masked_seq = $mask[0];        # same as above masked seq
           $ra_masked_coords = $mask[1];  # reference to array of mask coordinates

</pre><h4><b>FORMATDB</b></h4><pre>
       This  method creates BLAST indexes using the formatdb binary which is distributed with BLAST.  It accepts
       the same parameters as formatdb.  The remove_formatdb_indexes method will remove databases created  using
       the  formatdb  method  (if  called  by  the  same object).  formatdb leaves a file called formatdb.log by
       default in the current working directory (if it has permission).  To change  this  behavior,  use  the  l
       option to direct the sequence to /dev/null or somewhere else.

         use NHGRI::Blastall;
         my $b = new NHGRI::Blastall;
         $b-&gt;formatdb( i =&gt; 'swissprot',
                       p =&gt; 'T',
                       l =&gt; '/dev/null',
                       o =&gt; 'T',
                     );

</pre><h4><b>DB_ID_REGEX</b></h4><pre>
       By  default  Blastall.pm  expects FASTA deflines of BLAST databases to be formatted like Genbank database
       (gi|GINUMBER|DATABASE|ACCESSION|SUMMARY).  The default regular expression is  [^\|]+(?:\|[^\|,\s]*){1,10}
       When  using  non-genbankformatted deflines, it may become necessary to adjust the regular expression that
       identifies the unique identifier in a defline. This can be done with the -DB_ID_REGEX  parameter  to  the
       new method.  For example

           $b = new NHGRI::Blastall( -DB_ID_REGEX =&gt; '[^ ]+' );

</pre><h4><b>FILTERING</b></h4><pre>
       The  filter  method  accepts an anonymous hash in which the keys are elements of the blast report and the
       values are limits that are put on the result set.

       The following are the Filter elements and their default operations.

           id                  =&gt; regular expression match
           defline             =&gt; regular expression match
           subject_length      =&gt; greater than
           scores              =&gt; greater than
           expects             =&gt; less than
           identities          =&gt; greater than
           match_length        =&gt; greater than
           subject_strand      =&gt; equals
           query_frames        =&gt; equals
           subject_frames      =&gt; equals

       so if you would like to limit your results to entries that have scores greater than 38.2  and  identities
       greater than 98% you would say...

           @hits = $b-&gt;filter( scores      =&gt; '38.2',
                               identities  =&gt; '.98'
                             );

       you  can  also  override  the  defaults.  if you would like only scores that are less than 38.2 you could
       say...

           @hits = $b-&gt;filter( scores =&gt; '&lt;38.2' );

       or if you wanted only identities that were equal to 1 and you didn't care about the hits array you  could
       say...

           $b-&gt;filter( identities =&gt; '=1' );

       Regular expression matches are case insensitive.  If you wanted only records with the word "human" in the
       defline you could say...

           @hits = $b-&gt;filter( defline =&gt; 'HuMaN' );

       After  you  run  the  filter  method on an object the object only contains those results which passed the
       filter. This will effect additional calls to the filter method as well as calls to  other  methods  (e.g.
       result).  To reset the NHGRI::Blastall object you can use the unfilter method.

           $b-&gt;unfilter();

       See DUMP RESULTS for info on how to manipulate the array of hash refs.

</pre><h4><b>RESULT</b></h4><pre>
         The result method has 3 possible invocations.  The first invocation
         is when it is called without parameters.

         @results = $b-&gt;result();

         This invocation returns an array of hash references.
         See HASHREF for further explanation of this structure.

         To get a list of all the ids do...

         @ids = $b-&gt;result('id');

         These ids can be used to get at specific elements.  If 2 parameters
         are present and the first one is an element (See ELEMENTS for a list
         of ELEMENTS) and the second one is an id then the routine will
         return a list of elements corresponding to the id.

         @scores = $b-&gt;result('scores',$ids[0]);  # second param must be an id

         If more than 2 elements are passed the function will return undef.

</pre><h4><b>ACCESSOR</b> <b>METHODS</b></h4><pre>
       <b>get_report</b>
               returns the filename of the BLAST report.

       <b>get_database_description</b>
               returns description given to the database during formatting of db.
               e.g. All non-redundant GenBank CDStranslations+PDB+SwissProt+PIR

       <b>get_database_sequence_count</b>
               returns the number of sequences in the database.

       <b>get_database_letter_count</b>
               returns the number of total letters in the database.

       <b>get_blast_program</b>
               returns the BLAST program name that appears at the top of the report.
               either BLASTN, BLASTP, BLASTX, TBLASTN or TBLASTX

       <b>get_blast_version</b>
               returns the version of the BLAST program that was used.

</pre><h4><b>ELEMENTS</b></h4><pre>
       <b>id</b>
                an example of an id is `&gt;gb|U19386|MMU19386' the initial `&gt;'
                is just a flag.  The next characters up until the first pipe
                is the database the subject was taken from.  The next characters
                up to the next pipe is the Genbank accession number.  The last
                characters are the locus.  This element is used as a unique
                identifier by the NHGRI::Blastall module.
                (SCALAR)

       <b>defline</b>
               The definition line taken from the subject
                (SCALAR)

       <b>subject_length</b>
               This is the length of the full subject, not the
                length of the match.
                (SCALAR)

       <b>scores</b>
               This is score (in bits) of the match.
                (ARRAY)

       <b>expects</b>
               This is the statistical significance (`E value') for the match.
                (ARRAY)

       <b>identities</b>
               This is the number of identities divided by the match
               length in decimal format. (Listed as a fraction and a percentage
               in a BLAST report.)
               (ARRAY)

       <b>match_lengths</b>
               this is the number of base pairs that match up.
               (ARRAY)

       <b>query_starts</b>
               This is the number of the first base which matched
               with the subject.
               (ARRAY)

       <b>query_ends</b>
               This is the number of the last base which matched
               with the subject.
               (ARRAY)

       <b>subject_starts</b>
               This is the number of the first base which matched
               with the query.
               (ARRAY)

       <b>subject_ends</b>
               This is the number of the last base which matched
               with the query.
               (ARRAY)

       <b>subject_strands</b>
               This is either plus or minus depending on the orientation
               of the subject sequence in the match.
               (ARRAY)

       <b>query_strands</b>
               This is either plus or minus depending on the orientation
               of the query sequence in the match.
               (ARRAY)

       <b>query_frames</b>
               If you are running a blastx or tblastx search in which the
               query_sequence is translated this is the frame the query
               sequence matched.
               (ARRAY)

       <b>subject_frames</b>
               If you are running a tblastn or tblastx search in which the
               subject sequence is translated, this is the frame where the
               subject sequence matched.
               (ARRAY)

</pre><h4><b>HASHREF</b></h4><pre>
         Each hash ref contains an id, defline and subject Length.  Because
         there can be multiple scores, expect values, Identities, match_lengths,
         query_starts, query_strands and subject_starts, these are stored
         as array references.  The following is an array containing two hash
         refs.

         @hits = (
             {'id'                 =&gt; '&gt;gb|U79716|HSU79716',
              'defline'            =&gt; 'Human reelin (RELN) mRNA, complete cds',
              'subject_length'     =&gt; '11580',
              'scores'             =&gt; [ 684, 123               ],
              'expects'            =&gt; [ 0.0, 3e-26             ],
              'identities'         =&gt; [ .99430199, .89256198   ],
              'match_lengths'      =&gt; [ 351, 121               ],
              'query_starts'       =&gt; [ 3, 404                 ],
              'query_ends'         =&gt; [ 303, 704                 ],
              'subject_starts'     =&gt; [ 5858, 6259             ],
              'subject_ends'       =&gt; [ 6158, 6559             ],
              'subject_strands'    =&gt; [ 'plus', 'minus'        ],
              'query_strands'      =&gt; [ 'plus', 'plus'         ],
              'query_frames'       =&gt; [ '+1', '-3'             ],
              'subject_frames'     =&gt; [ '+2', '-1'             ],
             },
             {'id'                 =&gt; '&gt;gb|U24703|MMU24703',
              'defline'            =&gt; 'Mus musculus reelin mRNA, complete cds',
              'subject_length'     =&gt; '11673',
              'scores'             =&gt; [ 319, 38.2              ],
              'expects'            =&gt; [ 2e-85, 1.2             ],
              'identities'         =&gt; [ .86455331, 1           ],
              'match_lengths'      =&gt; [ 347, 19                ],
              'query_starts'       =&gt; [ 3, 493                 ],
              'query_ends'         =&gt; [ 303, 793                 ],
              'subject_starts'     =&gt; [ 5968, 6457             ]
              'subject_ends'       =&gt; [ 6268, 6757             ],
              'subject_strands'    =&gt; [ 'plus', 'minus'        ],
              'query_strands'      =&gt; [ 'plus', 'plus'         ],
              'query_frames'       =&gt; [ '+3', '-3'             ],
              'subject_frames'     =&gt; [ '+1', '-2'             ],
             }
         );

         See ELEMENTS for explanation of each element.
         See DUMP RESULTS and/or the <a href="../man1/perlref.1.html">perlref</a>(1) manpage for clues on working
             with this structure.

</pre><h4><b>DUMP</b> <b>RESULTS</b></h4><pre>
         When calling the result function or with no parameters, or calling the
         filter function, an array of references to hashes is returned.
         Each element of the array is a reference to a hash containing 1 record.
         See HASHREF for details on this structure.  The following
         routine will go through each element of the array of hashes and
         then print out the element and it's corresponding value or values.
         See <a href="../man1/perlref.1.html">perlref</a>(1) for more info on references.

         sub dump_results {
             foreach $rh_r (@results) {
                 while (($key,$value) = each %$rh_r) {
                     if (ref($value) eq "ARRAY") {
                         print "$key: ";
                         foreach $v (@$value) {
                             print "$v ";
                         }
                         print "\n";
                     } else {
                         print "$key: $value\n";
                     }
                 }
             }
         }

</pre><h4><b>AUTHOR</b></h4><pre>
       •   Joseph Ryan (<a href="mailto:jfryan@nhgri.nih.gov">jfryan@nhgri.nih.gov</a>)

</pre><h4><b>CONTACT</b> <b>ADDRESS</b></h4><pre>
       If you have problems, questions, comments send to <a href="mailto:webblaster@nhgri.nih.gov">webblaster@nhgri.nih.gov</a>

</pre><h4><b>COPYRIGHT</b> <b>INFORMATION</b></h4><pre>
       This  software/database is "United States Government Work" under the terms of the United States Copyright
       Act. It was written as part of the authors' official duties for the United  States  Government  and  thus
       cannot  be  copyrighted.  This  software/database  is  freely  available  to the public for use without a
       copyright notice. Restrictions cannot be placed on its present or future use.

       Although all reasonable efforts have been taken to ensure the accuracy and reliability  of  the  software
       and  data,  the  National  Human  Genome Research Institute (NHGRI) and the U.S.  Government does not and
       cannot warrant the performance or results that may be obtained by using this software or data. NHGRI  and
       the  U.S.   Government  disclaims  all  warranties  as to performance, merchantability or fitness for any
       particular purpose.

       In any work or product derived from this material, proper attribution of the authors as the source of the
       software or data should be made, using <a href="http://genome.nhgri.nih.gov/blastall">http://genome.nhgri.nih.gov/blastall</a> as the citation.

</pre><h4><b>ENVIRONMENT</b> <b>VARIABLES</b></h4><pre>
       <b>BLASTDB</b>
           location of BLAST formatted databases

       <b>BLASTMAT</b>
           location of BLAST matrices

       <b>TMPDIR</b> <b>TEMP</b> <b>TMP</b>
           If the `o' option is not passed to the blastall method than NHGRI::Blastall looks for  one  of  these
           vars  (in  order)  to  store the BLAST report.  This report is destroyed after the NHGRI::Blastall.pm
           object is destroyed.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/perl.1.html">perl</a></b>(1) <b><a href="../man1/perlref.1.html">perlref</a></b>(1)

       <a href="http://www.ncbi.nlm.nih.gov/BLAST/newblast.html">http://www.ncbi.nlm.nih.gov/BLAST/newblast.html</a>

       <a href="ftp://ncbi.nlm.nih.gov/blast/db/README">ftp://ncbi.nlm.nih.gov/blast/db/README</a>

       <a href="http://www.ncbi.nlm.nih.gov/BLAST/tutorial/Altschul-1.html">http://www.ncbi.nlm.nih.gov/BLAST/tutorial/Altschul-1.html</a>

perl v5.30.3                                       2020-11-06                                      <u><a href="../man3pm/Blastall.3pm.html">Blastall</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>