<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>compare_genomes.py - find methylation in nanopore reads</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/mcaller">mcaller_1.0.3+git20210624.b415090-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       compare_genomes.py - find methylation in nanopore reads

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: compare_genomes.py [-h] <b>--bed1</b> BED1 <b>--bed2</b> BED2 [-g GENOME_ALIGNMENT]

              [-v]

       Compare methylation between two genomes by probabilities of methylation for aligned positions

   <b>optional</b> <b>arguments:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>--bed1</b> BED1
              bed file 1 with verbose output from make_bed.py

       <b>--bed2</b> BED2
              bed file 2 with verbose output from make_bed.py

       <b>-g</b> GENOME_ALIGNMENT, <b>--genome_alignment</b> GENOME_ALIGNMENT
              an xmfa file from mauve (if absent, alignments assumed to be to the same reference genome)

       <b>-v</b>, <b>--version</b>
              print version

compare_genomes.py 1.0.3+git20210331.f7a616a        June 2021                              <u><a href="../man1/COMPARE_GENOMES.PY.1.html">COMPARE_GENOMES.PY</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>