<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>make_bed.py - find methylation in nanopore reads</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/mcaller">mcaller_1.0.3+git20210624.b415090-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       make_bed.py - find methylation in nanopore reads

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <u>/usr/lib/python3/dist-packages/joblib/_multiprocessing_helpers</u>.py:53:  UserWarning: [Errno 13] Permission
       denied.  joblib will operate in serial mode

       warnings.warn('%s.
              joblib will operate in serial mode' % (e,))

       usage: make_bed.py [-h] [-d MIN_READ_DEPTH] [-t MOD_THRESHOLD] <b>-f</b> MCALLER_FILE

              [-p POSITIONS] [--control] [--gff] [--ref REF] [--plot] [--plotsummary] [--plotdir PLOTDIR] [--vo]
              [-v]

       Produce bed file of methylated positions based on mCaller output

   <b>optional</b> <b>arguments:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>-d</b> MIN_READ_DEPTH, <b>--min_read_depth</b> MIN_READ_DEPTH
              minimum coverage of position to determine methylation (default = 15)

       <b>-t</b> MOD_THRESHOLD, <b>--mod_threshold</b> MOD_THRESHOLD
              minimum fraction of observations with probability of methylation &gt;=50% at a position to include in
              report (default = 0.5)

       <b>-f</b> MCALLER_FILE, <b>--mCaller_file</b> MCALLER_FILE
              the output file from mCaller to summarize

       <b>-p</b> POSITIONS, <b>--positions</b> POSITIONS
              ~bed file of positions for which to calculate % of methylated reads (chromosome,start,end,strand);
              ignores other thresholds

       <b>--control</b>
              take unmethylated positions as a control for motif detection

       <b>--gff</b>  output PacBio-style gff instead of bed ("identificationQv" score will be  average  probability  of
              methylation)

       <b>--ref</b> REF
              use reference fasta to output longer contexts surrounding a base, from <b>-20</b> to +20

       <b>--plot</b> plot currents deviations at the positions included (not recommended for many positions)

       <b>--plotsummary</b>
              plot currents deviations summarized across the positions included

       <b>--plotdir</b> PLOTDIR
              output directory for plots, default=mCaller_position_plots

       <b>--vo</b>   verbose output including probabilities for each position

       <b>-v</b>, <b>--version</b>
              print version

make_bed.py 1.0.3+git20210331.f7a616a               June 2021                                     <u><a href="../man1/MAKE_BED.PY.1.html">MAKE_BED.PY</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>