<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>ABYSS - assemble short reads into contigs</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/abyss">abyss_2.3.1-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       ABYSS - assemble short reads into contigs

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>ABYSS</b> [<u>OPTION</u>]... <u>FILE</u>...

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Assemble  all  input  files, FILE, which may be in FASTA, FASTQ, qseq, export, SRA, SAM or BAM format and
       may be compressed with gz, bz2 or xz and may be tarred.

       Users wishing to run a full assembly with ABySS are recommended to use  abyss-pe  as  their  entry  point
       rather  than  the  ABYSS  program.   abyss-pe  is  a  Makefile  that  coordinates many of the ABySS tools
       (including ABYSS) in order to run the full ABySS assembly pipeline.  abyss-pe is capable  of  doing  many
       things  that  ABYSS  is not, such as leveraging the distance information provided by paired end reads and
       mate pair reads.

       <b>--chastity</b>
              discard unchaste reads [default]

       <b>--no-chastity</b>
              do not discard unchaste reads

       <b>--trim-masked</b>
              trim masked bases from the ends of reads [default]

       <b>--no-trim-masked</b>
              do not trim masked bases from the ends of reads

       <b>-q</b>, <b>--trim-quality</b>=<u>THRESHOLD</u>
              trim bases from the ends of reads whose quality is less than the threshold

       <b>-Q</b>, <b>--mask-quality</b>=<u>THRESHOLD</u>
              mask all bases in reads whose quality is less than the threshold

       <b>--standard-quality</b>
              zero quality is `!' (33)
              default for FASTQ and SAM files

       <b>--illumina-quality</b>
              zero quality is `@' (64)
              default for qseq and export files

       <b>-o</b>, <b>--out</b>=<u>FILE</u>
              write the contigs to FILE

       <b>-k</b>, <b>--kmer</b>=<u>KMER_SIZE</u>
              k-mer size

       <b>-t</b>, <b>--trim-length</b>=<u>TRIM_LENGTH</u>
              maximum length of dangling edges to trim

       <b>-c</b>, <b>--coverage</b>=<u>COVERAGE</u>
              remove contigs with mean k-mer coverage less than this threshold

       <b>-b</b>, <b>--bubbles</b>=<u>N</u>
              pop bubbles shorter than N bp (default: 3*k).

       <b>-b</b>0, <b>--no-bubbles</b>
              do not pop bubbles

       <b>-e</b>, <b>--erode</b>=<u>COVERAGE</u>
              erode bases at the ends of blunt contigs with coverage less than this threshold

       <b>-E</b>, <b>--erode-strand</b>=<u>COVERAGE</u>
              erode bases at the ends of blunt contigs with coverage less than this threshold on either strand

       <b>--coverage-hist</b>=<u>FILE</u>
              record the k-mer coverage histogram in FILE

       <b>-g</b>, <b>--graph</b>=<u>FILE</u>
              generate a graph in dot format

       <b>-s</b>, <b>--snp</b>=<u>FILE</u>
              record popped bubbles in FILE

       <b>-v</b>, <b>--verbose</b>
              display verbose output

       <b>--help</b> display this help and exit

       <b>--version</b>
              output version information and exit

</pre><h4><b>AUTHOR</b></h4><pre>
       Written by Jared Simpson and Shaun Jackman.

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       Report bugs to &lt;<a href="mailto:abyss-users@bcgsc.ca">abyss-users@bcgsc.ca</a>&gt;.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright 2015 Canada's Michael Smith Genome Sciences Centre

ABYSS (ABySS) 2.3.1                                 2015-May                                            <u><a href="../man1/ABYSS.1.html">ABYSS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>