<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>abyss-pe - assemble reads into contigs</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/abyss">abyss_2.3.10-1ubuntu1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       abyss-pe - assemble reads into contigs

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>abyss-pe</b> [<u>OPTION</u>]...  [<u>PARAMETER</u>=<u>VALUE</u>]...  [<u>MAKE_TARGET</u>]...

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Assemble  the reads of the input files into contigs. The reads may be in FASTA, FASTQ, qseq, export, SRA,
       SAM or BAM format and may be compressed with gz, bz2 or xz and may be tarred.

       abyss-pe is a Makefile script. Any options of make may also be used with abyss-pe.

   <b>Parameters</b> <b>of</b> <b>abyss-pe</b>
       <b>name</b>, <b>JOB_NAME</b>
              The name of this assembly. The resulting scaffolds will be stored in ${name}-scaffolds.fa.

       <b>in</b>     input files. Use this variable when assembling data from a single library.

       <b>lib</b>    a quoted list of whitespace-separated paired-end library names. Use this variable when  assembling
              data  from  multiple  paired-end  libraries.  For each library name in lib, the user must define a
              variable on the command line with the same name, which indicates the read files for that  library.
              See <b>EXAMPLES</b> below for a concrete example of usage.

       <b>pe</b>     list  of paired-end libraries that will be used only for merging unitigs into contigs and will not
              contribute toward the consensus sequence.

       <b>mp</b>     list of mate-pair libraries that will  be  used  for  scaffolding.   Mate-pair  libraries  do  not
              contribute toward the consensus sequence.

       <b>long</b>   list  of  long sequence libraries that will be used for rescaffolding.  long sequence libraries do
              not contribute toward the consensus sequence.

       <b>se</b>     files containing single-end reads

       <b>a</b>      maximum number of branches of a bubble [2]

       <b>b</b>      maximum length of a bubble (bp) [""]
              abyss-pe has two bubble popping stages. The default limits are 3*k bp for ABYSS and 10000  bp  for
              PopBubbles.

       <b>c</b>      minimum mean k-mer coverage of a unitig [<a href="../manmedian/sqrt.median.html">sqrt</a>(median)]

       <b>d</b>      allowable error of a distance estimate (bp) [6]

       <b>e</b>      minimum erosion k-mer coverage [round(<a href="../manmedian/sqrt.median.html">sqrt</a>(median))]

       <b>E</b>      minimum erosion k-mer coverage per strand [1 if <a href="../manmedian/sqrt.median.html">sqrt</a>(median) &gt; 2 else 0]

       <b>j</b>      number of threads [2]

       <b>k</b>      size of a k-mer (when K is not set) or the span of a k-mer pair (when K is set)

       <b>K</b>      size of a single k-mer in a k-mer pair (bp)

       <b>l</b>      minimum alignment length of a read (bp) [40]

       <b>m</b>      minimum overlap of two unitigs (bp) [k-1]

       <b>n</b>      minimum number of pairs required for building contigs [10]

       <b>N</b>      minimum number of pairs required for building scaffolds [n]

       <b>p</b>      minimum sequence identity of a bubble [0.9]

       <b>q</b>      minimum base quality when trimming [3]
              Trim bases from the ends of reads whose quality is less q.

       <b>Q</b>      minimum base quality [0]
              Mask all bases of reads whose quality is less than Q as `N'.

       <b>s</b>      minimum unitig size required for building contigs (bp) [1000]
              The  seed  length  should be at least twice the value of k. If more sequence is assembled than the
              expected genome size, try increasing s.

       <b>S</b>      minimum contig size required for building scaffolds (bp) [1000-10000]

       <b>SS</b>     SS=--SS to assemble in strand-specific mode
              Requires that all libraries are strand-specific RNA-Seq libraries.  Assumes that the first read in
              a read pair is reversed WRT the transcripts sequenced.

       <b>t</b>      maximum length of blunt contigs to trim [k]

       <b>v</b>      v=-v to enable verbose logging

       <b>np</b>, <b>NSLOTS</b>
              the number of processes of an MPI assembly

       <b>mpirun</b> the path to mpirun

       <b>aligner</b>
              The program to use to align the reads to the contigs [map].
              Permitted values are: map, kaligner, bwa, bwasw, bowtie, bowtie2,  dida.   See  the  <b>DIDA</b>  section
              below for further info on the dida option.

       <b>cs</b>     convert colour-space contigs to nucleotide contigs following assembly

   <b>Options</b> <b>of</b> <b>make</b>
       <b>-n</b>, <b>--dry-run</b>
              Print the commands that would be executed, but do not execute them.

   <b>Make</b> <b>targets</b> <b>for</b> <b>abyss-pe</b>
       <b>default</b>
              Equivalent to `scaffolds scaffolds-dot stats'.

       <b>unitigs</b>
              Assemble unitigs.

       <b>unitigs-dot</b>
              Output the unitig overlap graph.

       <b>pe-sam</b> Map  paired-end  reads  to  the  unitigs  and  output a SAM file. SAM file will only contain reads
              mapping to different contigs, and the read ID, sequence and quality strings will be replaced  with
              '*' characters.

       <b>pe-bam</b> Map  paired-end  reads  to  the  unitigs  and  output a BAM file. BAM file will only contain reads
              mapping to different contigs, and the read ID, sequence and quality strings will be replaced  with
              '*' characters.

       <b>pe-index</b>
              Generate an index of the unitigs used by abyss-map.

       <b>contigs</b>
              Assemble contigs.

       <b>contigs-dot</b>
              Output the contig overlap graph.

       <b>mp-sam</b> Map mate-pair reads to the contigs and output a SAM file. SAM file will only contain reads mapping
              to  different  contigs,  and  the  read ID, sequence and quality strings will be replaced with '*'
              characters.

       <b>mp-bam</b> Map mate-pair reads to the contigs and output a BAM file. BAM file will only contain reads mapping
              to different contigs, and the read ID, sequence and quality strings  will  be  replaced  with  '*'
              characters.

       <b>mp-index</b>
              Generate an index of the contigs used by abyss-map.

       <b>scaffolds</b>
              Assemble scaffolds.

       <b>scaffolds-dot</b>
              Output the scaffold overlap graph.

       <b>scaftigs</b>
              Break scaffolds and generate AGP file.

       <b>long-scaffs</b>
              Rescaffold using RNA-Seq assembled contigs.

       <b>long-scaffs-dot</b>
              Output the RNA scaffold overlap graph.

       <b>stats</b>  Display assembly contiguity statistics.

       <b>clean</b>  Remove intermediate files.

       <b>version</b>
              Display the version of abyss-pe.

       <b>versions</b>
              Display the versions of all programs used by abyss-pe.

       <b>help</b>   Display a helpful message.

</pre><h4><b>DIDA</b></h4><pre>
       ABySS  supports  the  use  of  DIDA  (Distributed  Indexing Dispatched Alignment), an MPI-based alignment
       framework for computing sequence alignments across multiple machines.  To  use  DIDA  with  ABySS,  first
       download and install DIDA from <a href="http://www.bcgsc.ca/platform/bioinfo/software/dida">http://www.bcgsc.ca/platform/bioinfo/software/dida</a>, then specify `dida` as
       the value of the <b>aligner</b> parameter to <b>abyss-pe</b>.

   <b>DIDA-related</b> <b>abyss-pe</b> <b>parameters</b>
       <b>DIDA_MPIRUN</b>
              The `mpirun` command used to run DIDA jobs.

       <b>DIDA_RUN_OPTIONS</b>
              Runtime  options such as number of threads per MPI rank and values for environment variables (e.g.
              PATH, LD_LIBRARY_PATH).  Run `abyss-dida --help` for a list of available options.

       <b>DIDA_OPTIONS</b>
              Options that are passed directly to the DIDA binary. For example, this can be used to control  the
              minimum alignment length threshold.  Run `dida-wrapper --help` for a list of available options.

   <b>MPI</b> <b>COMPATIBILITY</b>
       Due  to  its use of multi-threading, DIDA has known deadlocking issues with OpenMPI.  Using the MPICH MPI
       library is strongly recommended when running assemblies with DIDA. Testing was  done  with  MPICH  3.1.3,
       compiled with --enable-threads=funneled.

   <b>EXAMPLE</b>
       The  recommended  runtime configuration for DIDA is 1 MPI rank per machine and 1 thread per CPU core. For
       example, to run an assembly across 3 cluster nodes with 12 cores each, do:

            abyss-pe   k=64   name=ecoli   in='reads1.fa   reads2.fa'    aligner=dida    DIDA_RUN_OPTIONS='-j12'
       DIDA_MPIRUN='mpirun -np 3 -ppn 1 -bind-to board'

       This example uses the MPICH command line options for `mpirun`.  Here, `-np 3` indicates the number of MPI
       ranks, `-ppn 1` indicates the number of MPI ranks per "node", and `-bind-to board` defines a "node" to be
       a motherboard (i.e. a full machine).

</pre><h4><b>ENVIRONMENT</b> <b>VARIABLES</b></h4><pre>
       Any parameter that may be specified on the command line may also be specified in an environment variable.

       <b>PATH</b>   must  contain the directory where the ABySS executables are installed.  Use `abyss-pe versions` to
              check that PATH is configured correctly.

       <b>TMPDIR</b> specifies a directory to use for temporary files

   <b>Scheduler</b> <b>integration</b>
       ABySS integrates well with cluster job schedulers, such as:
        * SGE (Sun Grid Engine)
        * Portable Batch System (PBS)
        * Load Sharing Facility (LSF)
        * IBM LoadLeveler

       The SGE environment variables JOB_NAME, SGE_TASK_ID and NSLOTS may be  used  to  specify  the  parameters
       name, k and np, respectively, and similarly for other schedulers.

</pre><h4><b>EXAMPLES</b></h4><pre>
   <b>One</b> <b>paired-end</b> <b>library</b>
        abyss-pe k=64 name=ecoli in='reads1.fa reads2.fa'

   <b>Multiple</b> <b>paired-end</b> <b>libraries</b>
        abyss-pe k=64 name=ecoli lib='lib1 lib2' \
            lib1='lib1_1.fa lib1_2.fa' lib2='lib2_1.fa lib2_2.fa' \
            se='se1.fa se2.fa'

   <b>Paired-end</b> <b>and</b> <b>mate-pair</b> <b>libraries</b>
        abyss-pe k=64 name=ecoli lib='pe1 pe2' mp='mp1 mp2' \
            pe1='pe1_1.fa pe1_2.fa' pe2='pe2_1.fa pe2_2.fa' \
            mp1='mp1_1.fa mp1_2.fa' mp2='mp2_1.fa mp2_2.fa' \
            se='se1.fa se2.fa'

   <b>Including</b> <b>RNA-Seq</b> <b>assemblies</b>
        abyss-pe k=64 name=ecoli lib=pe1 mp=mp1 long=long1 \
            pe1='pe1_1.fa pe1_2.fa' mp1='mp1_1.fa mp1_2.fa' \
            long1=long1.fa

   <b>MPI</b>
        abyss-pe np=8 k=64 name=ecoli in='reads1.fa reads2.fa'

   <b>SGE</b>
        qsub -N ecoli -t 64 -pe openmpi 8 \
            abyss-pe n=10 in='reads1.fa reads2.fa'

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <a href="../man1/make.1.html">make</a>(1), <a href="../man1/ABYSS.1.html">ABYSS</a>(1)

</pre><h4><b>AUTHOR</b></h4><pre>
       Written by Shaun Jackman.

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       Report bugs to &lt;<a href="mailto:abyss-users@googlegroups.com">abyss-users@googlegroups.com</a>&gt;.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright 2015 Canada's Michael Smith Genome Sciences Centre

abyss-pe (ABySS) 2.3.10                             2015-May                                         <u><a href="../man1/abyss-pe.1.html">abyss-pe</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>