<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Beagle - Genotype calling, genotype phasing and imputation of ungenotyped markers</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/beagle">beagle_241217-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Beagle - Genotype calling, genotype phasing and imputation of ungenotyped markers

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>java</b> <b>-Xmx[GB]g</b> <b>-jar</b> <b>/usr/share/beagle/beagle.jar</b> [<u>options</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Beagle  performs  genotype calling, genotype phasing, imputation of ungenotyped markers, and identity-by-
       descent segment detection. Genotypic imputation works on  phased  haplotypes  using  a  Li  and  Stephens
       haplotype  frequency model.  Beagle also implements the Refined IBD algorithm for detecting homozygosity-
       by-descent (HBD) and identity-by-descent (IBD) segments.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Data</b> <b>input/output</b> <b>parameters</b>
       <b>gt</b>=<u>filename</u>
              Optional
              Specifies a VCF file containing a GT (genotype) format field  for  each  marker.   If  a  genotype
              contains  the  phased  allele  separator, "|", then Beagle will preserve the phase of the genotype
              during the analysis.  If you use the gt argument, all genotypes in the output file will be  phased
              and non-missing.

       <b>gl</b>=<u>filename</u>
              Optional
              Specifies  a  VCF  file  containing a GL or PL (genotype likelihood) format field for each marker.
              Any data in the GT format field will be ignored.  If both GL and PL format fields are present  for
              a marker, the GL format will be used.

       <b>gtgl</b>=<u>filename</u>
              Optional
              Specifies  a  VCF  file  containing a GT, GL or PL format field for each marker.  If a genotype is
              non-missing, Beagle will ignore the genotype likelihood.  If both GL  and  PL  format  fields  are
              present for a marker, the GL field will be used.

       <b>ref</b>=<u>filename</u>
              Optional
              Specifies a VCF file containing phased reference genotypes.  See the impute parameter.

       <b>out</b>=<u>prefix</u>
              Required
              Specifies  the output filename prefix.  The prefix may be an absolute or relative filename, but it
              cannot be a directory name.

       <b>excludesamples</b>=<u>filename</u>
              Optional
              Specifies a file containing non-reference samples (one sample per line) to be  excluded  from  the
              analysis and output files.

       <b>excludemarkers</b>=<u>filename</u>
              Optional
              Specifies a file containing markers (one marker per line) to be excluded from the analysis and the
              output  files.  An excluded marker identifier can either be an identifier from the VCF record’s ID
              field or a genomic coordinate in the format: CHROM:POS.

       <b>map</b>=<u>filename</u>
              Optional
              Specifies a PLINK format genetic map on the cM scale.  HapMap GrCh36 and GrCh37  genetic  maps  in
              PLINK  format  are  available  for  download  from  the  Beagle  website.  Use of a genetic map is
              recommended if you are imputing ungenotyped markers.  If no genetic map is specified, Beagle  will
              assume a constant recombination rate of 1 cM / Mb.

       <b>chrom</b>=<u>chrom:start-end</u>
              Optional
              Specifies  a  chromosome  or chromosome interval using a chromosome identifier in the VCF file and
              the starting and ending positions of the interval.  The entire chromosome, the  beginning  of  the
              chromosome, and the end of a chromosome can be specified by <b>chrom</b>=[chrom], <b>chrom</b>=[chrom:-end], and
              <b>chrom</b>=[chrom:start-] respectively.

       <b>maxlr</b>=<u>number_≥_1</u>
              Default = 5000
              Specifies  the  maximum likelihood ratio at a genotype.  If M is the maximum of the likelihoods of
              each possible genotype, any likelihood that is less than (M ⁄ maxlr) is  set  to  0.0  to  improve
              computational efficiency.

   <b>General</b> <b>parameters</b>
       <b>nthreads</b>=<u>positive_integer</u>
              Default: machine-dependent
              Specifies the number of threads of execution.  If no <b>nthreads</b> parameter is specified, the <b>nthreads</b>
              parameter will be set equal to the number of CPU cores on the host machine.

       <b>lowmem</b>=<u>true/false</u>
              Default = false
              Specifies  whether  a  memory  efficient algorithm should be used.  The memory efficient algorithm
              increases run-time by a factor less than 2.0.

       <b>window</b>=<u>positive_integer</u>
              Default = 50000
              Specifies the number of markers to include in each sliding window.  The <b>window</b> parameter  must  be
              at  least  twice  as  large as the overlap parameter.  The <b>window</b> parameter controls the amount of
              memory used in the analysis.  For human data, it is  recommended  that  the  <b>window</b>  parameter  be
              greater than or equal to the typical number of markers in 5 cM.

       <b>overlap</b>=<u>positive_integer</u>
              Default = 3000
              Specifies  the  number  of  markers  of  overlap  between  sliding windows.  For human data, it is
              recommended that the overlap be set to the typical number of markers in 0.5 cM (when <b>ibd</b>=false) or
              2.0 cM (when <b>ibd</b>=true).

       <b>seed</b>=<u>integer</u>
              Default = -99999
              Specifies the seed for the random number generator.

   <b>Phasing</b> <b>and</b> <b>imputation</b> <b>parameters</b>
       <b>niterations</b>=<u>non-negative_integer</u>
              Default = 5
              Specifies the number of phasing iterations.  The phasing iterations are  preceded  by  10  burn-in
              iterations  which  carry  out the Beagle version 4.0 phasing algorithm.  If you want to phase your
              data with the Beagle 4.0 phasing algorithm, use <b>niterations</b>=0.  Accuracy and compute time increase
              with the number of iterations.

       <b>impute</b>=<u>true/false</u>
              Default = true
              Specifies whether markers that are present in the reference panel but absent in your data will  be
              imputed.  This option has no effect if the <b>ref</b> and <b>gt</b> arguments are not used.

       <b>gprobs</b>=<u>true/false</u>
              Default = false
              Specifies whether a GP (genotype probability) format field will be included in the output VCF file
              when imputing ungenotyped markers.  By default, a GP fields is not printed because a DS (alternate
              allele dose) format field is always printed when imputing ungenotyped markers.

       <b>ne</b>=<u>integer</u>
              Default = 1000000
              Specifies  the  effective population size when imputing ungenotyped markers.  The default value is
              suitable for a large outbred human population.  Smaller values in the hundreds  or  thousands  for
              the <b>ne</b> parameter are suggested for inbred human and animal populations.

       <b>err</b>=<u>non-negative_number</u>
              Default = 0.0001
              Specifies  the  allele miscall rate.  The default value should give good results for most sequence
              and SNP array data.

       <b>cluster</b>=<u>non-negative_number</u>
              Default = 0.005
              Specifies the maximum cM distance between individual markers that are combined into  an  aggregate
              marker when imputing ungenotyped markers.

   <b>IBD</b> <b>parameters</b>
       <b>ibd</b>=<u>true/false</u>
              Default = false
              Specifies whether IBD analysis will be performed when the <b>gt</b> argument is used.

       <b>ibdlod</b>=<u>non-negative_integer</u>
              Default = 3.0
              Specifies the minimum LOD score for reported IBD.

       <b>ibdscale</b>=<u>non-negative_number</u>
              Default: data-dependent
              Specifies the scale parameter used to build the haplotype frequency model for IBD analysis.  If no
              <b>ibdscale</b>  parameter  is  specified  the scale parameter for the IBD analysis will be set to max{2,
              sqrt[sample size]/100}, which we have found to work well for outbred populations.

       <b>ibdtrim</b>=<u>non-negative_integer</u>
              Default = 40
              Specifies the number of markers trimmed from the end of a shared haplotype when testing  for  IBD.
              Note: The default <b>ibdtrim</b> parameter is designed for European samples genotyped with a 1M SNP array
              (~  1  marker per 3 kb).  For human SNP array data, it is recommended to set the <b>ibdtrim</b> parameter
              to the typical number of markers in a 0.15 cM region.  Pilot studies of randomly selected  genomic
              regions can be used to fine-tune the values of the <b>ibdtrim</b> parameter.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b>https://faculty.washington.edu/browning/beagle/beagle.html</b>

</pre><h4><b>AUTHOR</b></h4><pre>
       Beagle was written by Brian L. Browning.

       This  manual  page  was written by Dylan Aïssi &lt;<a href="mailto:bob.dybian@gmail.com">bob.dybian@gmail.com</a>&gt;, for the Debian project (but may be
       used by others).

4.1                                               February 2016                                        <u><a href="../man1/Beagle.1.html">Beagle</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>