<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>kissreads2 - enhances the kissnp2 results by computing per read set</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/discosnp">discosnp_2.6.2-4build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       kissreads2 - enhances the kissnp2 results by computing per read set

</pre><h4><b>DESCRIPTION</b></h4><pre>
       [Kissreads2 options]

       <b>-x</b>     (0 arg) :    radseq option, homogeneous read depth along the prediction

       <b>-predictions</b>
              (1 arg) :    Input predictions

       <b>-reads</b> (1 arg) :    Input reads

       <b>-unco</b>  (1 arg) :    Output uncoherent file name  [default '/dev/null']

       <b>-co</b>    (1 arg) :    Output coherent file name

       <b>-hamming</b>
              (1 arg) :    Maximal hamming distance authorized while maping  [default '1']

       <b>-coverage_file</b> (1 arg) :
              File  (.h5)  generated  by  kissnp2,  containing  the  coverage  threshold  per read set  [default
              '_removemeplease']

       <b>-k</b>     (1 arg) :    Size of k, used as minial overlap and kmer spanning read coherence  [default '31']

       <b>-index_stride</b>
              (1 arg) :    Index Stride  [default '2']

       <b>-size_seeds</b>
              (1 arg) :    Size of the used seeds (distinct from the size of k)  [default '25']

       <b>-output_fasta</b>
              (0 arg) :    Output standard Fasta.  By  default  the  output  is  formatted  especially  for  the
              discoSnp++ pipeline

       <b>-genotype</b>
              (0 arg) :    Compute genotypes

       <b>-nb-cores</b>
              (1 arg) :    number of cores  [default '0']

       <b>-verbose</b>
              (1 arg) :    verbosity level  [default '1']

       <b>-version</b>
              (0 arg) :    version

       <b>-help</b>  (0 arg) :    help

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

kissreads2 2.3.0                                  January 2019                                     <u><a href="../man1/KISSREADS2.1.html">KISSREADS2</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>