<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>fa2htgs - formatter for high throughput genome sequencing project submissions</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/ncbi-tools-bin">ncbi-tools-bin_6.1.20170106+dfsg2-5_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       fa2htgs - formatter for high throughput genome sequencing project submissions

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>fa2htgs</b>  [<b>-</b>]  [<b>-6</b> <u>str</u>]  [<b>-7</b> <u>str</u>]  [<b>-A</b> <u>filename</u>]  [<b>-C</b> <u>str</u>]  [<b>-D</b>] [<b>-L</b> <u>filename</u>] [<b>-M</b> <u>str</u>] [<b>-N</b>] [<b>-O</b> <u>filename</u>]
       [<b>-P</b> <u>str</u>] [<b>-Q</b> <u>filename</u>] [<b>-S</b> <u>str</u>] [<b>-T</b> <u>filename</u>] [<b>-X</b>] [<b>-a</b> <u>str</u>] [<b>-b</b> <u>N</u>] [<b>-c</b> <u>str</u>] [<b>-d</b> <u>str</u>]  [<b>-e</b> <u>filename</u>]  [<b>-f</b>]
       <b>-g</b> <u>str</u>  [<b>-h</b> <u>str</u>]  [<b>-i</b> <u>filename</u>]  [<b>-k</b> <u>str</u>]  [<b>-l</b> <u>N</u>] [<b>-m</b>] [<b>-n</b> <u>str</u>] [<b>-o</b> <u>filename</u>] [<b>-p</b> <u>N</u>] [<b>-q</b>] [<b>-r</b> <u>str</u>] <b>-s</b> <u>str</u>
       [<b>-t</b> <u>filename</u>] [<b>-u</b>] [<b>-v</b>] [<b>-w</b>] [<b>-x</b> <u>str</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>fa2htgs</b> is a program used to generate Seq-submits (an ASN.1 sequence submission file) for high throughput
       genome sequencing projects.

       <b>fa2htgs</b> will read a FASTA file (or an Ace Contig file with  Phrap  sequence  quality  values),  a  Sequin
       submission  template  file, (to get contact and citation information for the submission), and a series of
       command line arguments (see below).  This  program  will  then  combines  these  information  to  make  a
       submission  suitable  for  GenBank.  Once you have generated your submission file, you need to follow the
       submission protocol (see the README present on your FTP account or mailed out to your Center).

       <b>fa2htgs</b> is intended for the automation by scripts for bulk submission of unannotated genome sequence.  It
       can  easily  be extended from its current simple form to allow more complicated processing.  A submission
       prepared with <b>fa2htgs</b> can also be read into <b><a href="../man1/Psequin.1.html">Psequin</a></b>(1), and then annotated more extensively.

       Questions and concerns about this processing protocol, or how to use this tool  should  be  forwarded  to
       &lt;<a href="mailto:htgs@ncbi.nlm.nih.gov">htgs@ncbi.nlm.nih.gov</a>&gt;.

</pre><h4><b>OPTIONS</b></h4><pre>
       A summary of options is included below.

       <b>-</b>      Print usage message

       <b>-6</b> <u>str</u> SP6 clone (e.g., Contig1,left)

       <b>-7</b> <u>str</u> T7 clone (e.g., Contig2,right)

       <b>-A</b> <u>filename</u>
              Filename  for accession list input (mutually exclusive with <b>-T</b> and <b>-i</b>).  The input file contains a
              tab-delimited table with three to five columns, which are accession number, start  position,  stop
              position, and (optionally) length and strand.  If start &gt; stop, the minus strand on the referenced
              accession  is used.  A gap is indicated by the word "gap" instead of an accession, 0 for the start
              and stop positions, and a number for the length.

       <b>-C</b> <u>str</u> Clone library name (will appear as <b>/clone-lib="</b><u>str</u><b>"</b> on the source feature)

       <b>-D</b>     HTGS_DRAFT sequence

       <b>-L</b> <u>filename</u>
              Read phrap contig order from <u>filename</u>.  This is a tab-delimited file that can be used to drive the
              order of contigs (normally specified by <b>-P</b>), as well as indicating the SP6 and T7  ends.   It  can
              also be used when contigs are known to be in opposite orientation.  For example:

                  Contig2     +       1       SP6     left
                  Contig3     +       1
                  Contig1     -               T7      right

              The  first  column  is  the  contig  name,  the  second  is  the  orientation,  the  third  is the
              fragment_group, the fourth indicates the SP6 or T7 end, and the fifth says which side of SP6 or T7
              end had vector removed.

       <b>-M</b> <u>str</u> Map name (will appear as <b>/map="</b><u>str</u><b>"</b> on the source feature)

       <b>-N</b>     Annotate assembly_fragments

       <b>-O</b> <u>filename</u>
              Read comment from <u>filename</u> (100-character-per-line maximum; <b>~</b> is a linebreak and <b>`~</b> is  a  literal
              <b>~</b>.  You can check the format with <b><a href="../man1/PSequin.1.html">PSequin</a></b>(1).)

       <b>-P</b> <u>str</u> Contigs  to  use,  separated  by  commas.   If  <b>-P</b>  is  not indicated with the <b>-T</b> option, then the
              fragments will go in in the order that they are in the ace file (which is appropriate for a  phase
              1  record,  but  not for a phase 2 or 3).  If you need to set the order of the segments of the ace
              file, you need to set it with the <b>-P</b> flag, like this: <b>-P</b> <b>"Contig1,Contig4,Contig3,Contig2,Contig5"</b>

       <b>-Q</b> <u>filename</u>
              Read quality scores from <u>filename</u>

       <b>-S</b> <u>str</u> Strain name

       <b>-T</b> <u>filename</u>
              Filename for phrap input (mutually exclusive with <b>-A</b> and <b>-i</b>)

       <b>-X</b>     The coordinates in the input file are on the resulting segmented sequence.  (Bases 1 through <u>n</u>  of
              each accession are used.)  Otherwise, the coordinates are on the individual accessions, which need
              not start at base 1 of the record.

       <b>-a</b> <u>str</u> GenBank accession; use if and only if updating a sequence.

       <b>-b</b> <u>N</u>   Gap length (default = 100; anything from 0 to 1000000000 is legal)

       <b>-c</b> <u>str</u> Clone name (will appear as <b>/clone</b> in the source feature; can be the same as <b>-s</b>)

       <b>-d</b> <u>str</u> Title for sequence (will appear in GenBank <b>DEFINITION</b> line)

       <b>-e</b> <u>filename</u>
              Log errors to <u>filename</u>

       <b>-f</b>     htgs_fulltop keyword

       <b>-g</b> <u>str</u> Genome Center tag (probably the same as your login name on the NCBI FTP server)

       <b>-h</b> <u>str</u> Chromosome (will appear as <b>/chromosome</b> in the source feature)

       <b>-i</b> <u>filename</u>
              Filename for fasta input (default is stdin; mutually exclusive with <b>-A</b> and <b>-T</b>)

       <b>-k</b> <u>str</u> Add the supplied string as a keyword.

       <b>-l</b> <u>N</u>   Length  of  sequence in bp (default = 0). The length is checked against the actual number of bases
              we get. For phase 1 and 2 sequence it is also used to estimate gap lengths.  For  phase  1  and  2
              records, it is important to use a number GREATER than the amount of provided nucleotide, otherwise
              this  will  generate  false  `gaps'.   Here is assumed that the putative full length of the BAC or
              cosmid will be used.  There should be at least 20 to 30 `n' in between the segments (you can check
              for these in Sequin), as this will ensure proper behavior when this sequence is used  with  BLAST.
              Otherwise `artifactual' unrelated segment neighbors may be brought into proximity of each other.

       <b>-m</b>     Take comment from template

       <b>-n</b> <u>str</u> Organism name (default = Homo sapiens)

       <b>-o</b> <u>filename</u>
              Filename for asn.1 output (default = stdout)

       <b>-p</b> <u>N</u>   HTGS phase:
              1      A  collection  of  unordered contigs with gaps of unknown length.  A Phase 1 record must at
                     the very least have two segments with one gap.  (default)
              2      A series of ordered contigs, possibly with known gap  lengths.   This  could  be  a  single
                     sequence without gaps, if the sequence has ambiguities to resolve.
              3      A single contiguous sequence.  This sequence is finished, but not necessarily annotated.

       <b>-q</b>     htgs_cancelled keyword

       <b>-r</b> <u>str</u> Remark for update (brief comment describing the nature of the update, such as "new sequence", "new
              citation", or "updated features")

       <b>-s</b> <u>str</u> Sequence  name.  The sequence must have a name that is unique within the genome center. We use the
              combination of the genome center name (<b>-g</b> argument) and the  sequence  name  (<b>-s</b>)  to  track  this
              sequence  and  to  talk  to  you about it.  The name can have any form you like but must be unique
              within your center.

       <b>-t</b> <u>filename</u>
              Filename for Seq-submit template (default = template.sub)

       <b>-u</b>     Take biosource from template

       <b>-v</b>     htgs_activefin keyword

       <b>-w</b>     Whole Genome Shotgun flag

       <b>-x</b> <u>str</u> Secondary accession numbers, separated by commas, s.t. U10000,L11000.

              In some cases a large  segment  will  supersede  another  or  group  of  other  accession  numbers
              (records).   These  records  which are no longer wanted in GenBank should be made secondary. Using
              the <b>-x</b> argument you can list the Accession Numbers you want to make secondary.  This will instruct
              us to remove the accession number(s) from GenBank, and will no  longer  be  part  of  the  GenBank
              release. They will nonetheless be available from Entrez.

              <b>GREAT</b>  <b>CARE</b>  should  be  taken when using this argument!!!  Improper use of accession numbers here
              will result in the inappropriate withdrawal of GenBank records from GenBank, EMBL  and  DDBJ.   We
              provide  this parameter as a convenience to submitting centers, but this may need to be removed if
              it is not used carefully.

</pre><h4><b>AUTHOR</b></h4><pre>
       The National Center for Biotechnology Information.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/Psequin.1.html">Psequin</a></b>(1), /usr/share/doc/ncbi-tools-bin/README.fa2htgs.gz

NCBI                                               2006-05-29                                         <u><a href="../man1/FA2HTGS.1.html">FA2HTGS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>