<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Filters - Low quality overlap filtering</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/mecat2">mecat2_0.0+git20200428.f54c542+ds-4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Filters - Low quality overlap filtering

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>fsa_ol_filter</b> [<u>options</u>] <u>overlaps</u> <u>filtered_overlaps</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Command line tool to filters out low-quality overlaps prior to assembling.

       This takes the following parameters:

              overlaps

                   The input filename

              filtered_overlaps

                   The resulting ouput filename

</pre><h4><b>OPTIONS</b></h4><pre>

       <b>--min_length</b>=<u>INT</u>

              minimum length of reads

              default: 2500

       <b>--max_length</b>=<u>INT</u>

              maximum length of reads default: 2147483647

       <b>--min_identity</b>=<u>DOUBLE</u>

              minimum identity of overlaps

              default: 90

       <b>--min_aligned_length</b>=<u>INT</u>

              minimum aligned length of overlaps default: 2500

       <b>--max_overhang</b>=<u>INT</u>

              maximum overhang of overlaps, negative number = determined by the program.

              default: 10

       <b>--min_coverage</b>=<u>INT</u>

              minimum base coverage, negative number = determined by the program.  default: <b>-1</b>

       <b>--max_coverage</b>=<u>INT</u>

              maximum base coverage, negative number = determined by the program default: <b>-1</b>

       <b>--max_diff_coverage</b>=<u>INT</u>

              maximum difference of base coverage, negative number = determined by the program default: <b>-1</b>

       <b>--coverage_discard</b>=<u>DOUBLE</u>

              discard ratio of base coverage. If max_coverage or max_diff_coverage is negative, it will be reset
              to (100-coverage_discard)th percentile.  default: 0.01

       <b>--overlap_file_type=</b>"|m4|m4a|paf|ovl"

              overlap  file  format.  "" = filename extension, "m4" = M4 format, "paf" = PAF format generated by
              minimap2, "ovl" = OVL format generated by FALCON.  default: ""

       <b>--bestn</b>=<u>INT</u>

              output best n overlaps on 5' or 3' end for each read

              default: 10

       <b>--genome_size</b>=<u>INT</u>

              genome size. It determines the maximum length of reads with coverage together default: 0

       <b>--coverage</b>=<u>INT</u>

              coverage. It determines the maximum length of reads with genome_size together default: 0

       <b>--output_directory</b>=<u>STRING</u>

              directory for output files default: "."

       <b>--thread_size</b>=<u>INT</u>

              number of threads default: 4

       Usage: fsa_ol_filter [options] overlaps filtered_overlaps

              overlaps

              input filename

              filtered_overlaps

              ouput filename

       <b>--min_length</b>=<u>INT</u>

              minimum length of reads

              default: 2500

       <b>--max_length</b>=<u>INT</u>

              maximum length of reads default: 2147483647

       <b>--min_identity</b>=<u>DOUBLE</u>

              minimum identity of overlaps

              default: 90

       <b>--min_aligned_length</b>=<u>INT</u>

              minimum aligned length of overlaps default: 2500

       <b>--max_overhang</b>=<u>INT</u>

              maximum overhang of overlaps, negative number = determined by the program.

              default: 10

       <b>--min_coverage</b>=<u>INT</u>

              minimum base coverage, negative number = determined by the program.  default: <b>-1</b>

       <b>--max_coverage</b>=<u>INT</u>

              maximum base coverage, negative number = determined by the program default: <b>-1</b>

       <b>--max_diff_coverage</b>=<u>INT</u>

              maximum difference of base coverage, negative number = determined by the program default: <b>-1</b>

       <b>--coverage_discard</b>=<u>DOUBLE</u>

              discard ratio of base coverage. If max_coverage or max_diff_coverage is negative, it will be reset
              to (100-coverage_discard)th percentile.  default: 0.01

       <b>--overlap_file_type=</b>"|m4|m4a|paf|ovl"

              overlap file format. "" = filename extension, "m4" = M4 format, "paf" = PAF  format  generated  by
              minimap2, "ovl" = OVL format generated by FALCON.  default: ""

       <b>--bestn</b>=<u>INT</u>

              output best n overlaps on 5' or 3' end for each read

              default: 10

       <b>--genome_size</b>=<u>INT</u>

              genome size. It determines the maximum length of reads with coverage together default: 0

       <b>--coverage</b>=<u>INT</u>

              coverage. It determines the maximum length of reads with genome_size together default: 0

       <b>--output_directory</b>=<u>STRING</u>

              directory for output files default: "."

       <b>--thread_size</b>=<u>INT</u>

              number of threads default: 4

Filters out low-quality overlaps for assembling     July 2020                                         <u><a href="../man1/FILTERS.1.html">FILTERS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>