<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>blasr - Map SMRT Sequences to a reference genome</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/blasr">blasr_5.3.5+dfsg-7build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       blasr - Map SMRT Sequences to a reference genome

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>blasr</b> reads.bam genome.fasta <b>--bam</b> <b>--out</b> out.bam

       <b>blasr</b> reads.fasta genome.fasta

       <b>blasr</b> reads.fasta genome.fasta <b>--sa</b> genome.fasta.sa

       <b>blasr</b> reads.bax.h5 genome.fasta [--sa genome.fasta.sa]

       <b>blasr</b> reads.bax.h5 genome.fasta <b>--sa</b> genome.fasta.sa <b>--maxScore</b> 100 <b>--minMatch</b> 15 ...

       <b>blasr</b> reads.bax.h5 genome.fasta <b>--sa</b> genome.fasta.sa <b>--nproc</b> 24 <b>--out</b> alignment.out ...

</pre><h4><b>DESCRIPTION</b></h4><pre>
       blasr  is  a  read  mapping  program  that  maps reads to positions in a genome by clustering short exact
       matches between the read and the genome, and scoring clusters using alignment. The matches are  generated
       by  searching all suffixes of a read against the genome using a suffix array. Global chaining methods are
       used to score clusters of matches.

       The only required inputs to blasr are a file of reads and a reference genome.  It is exremely  useful  to
       have read filtering information, and mapping runtime may decrease substantially when a precomputed suffix
       array index on the reference sequence is specified.

       Although  reads  may  be  input  in FASTA format, the recommended input is PacBio BAM files because these
       contain quality value information that is used in the  alignment  and  produces  higher  quality  variant
       detection.  Although alignments can be output in various formats, the recommended output format is PacBio
       BAM.   Support to bax.h5 and plx.h5 files will be DEPRECATED.  Support to region tables for h5 files will
       be DEPRECATED.

       When suffix array index of a genome is  not  specified,  the  suffix  array  is  built  before  producing
       alignment.    This  may  be  prohibitively  slow  when  the  genome is large (e.g. Human).  It is best to
       precompute the suffix array of a genome using the program sawriter, and then specify the suffix array  on
       the command line using <b>-sa</b> genome.fa.sa.

       The  optional  parameters  are  roughly  divided into three categories: control over anchoring, alignment
       scoring, and output.

       The default anchoring parameters are optimal for small genomes and samples with up to 5% divergence  from
       the  reference  genome.   The  main parameter governing speed and sensitivity is the <b>-minMatch</b> parameter.
       For human genome alignments, a value of 11 or higher is recommended.  Several  methods  may  be  used  to
       speed up alignments, at the expense of possibly decreasing sensitivity.

       Regions  that are too repetitive may be ignored during mapping by limiting the number of positions a read
       maps to with the <b>-maxAnchorsPerPosition</b> option.  Values between 500 and 1000 are effective in  the  human
       genome.

       For  small  genomes  such as bacterial genomes or BACs, the default parameters are sufficient for maximal
       sensitivity and good speed.

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

blasr 5.3.2                                       January 2019                                          <u><a href="../man1/BLASR.1.html">BLASR</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>