<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>mapsembler2_kissreads  - Checks for each sequence contained into the toCheck.fasta if it is read coherent</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/mapsembler2">mapsembler2_2.2.4+dfsg1-4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       mapsembler2_kissreads  - Checks for each sequence contained into the toCheck.fasta if it is read coherent
       with reads from readsA.fasta or readsB.fasta

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>mapsembler2_kissreads</b> &lt;toCheck.fasta&gt; &lt;readsC1.fasta&gt; [&lt;readsC2.fasta&gt; [&lt;readsC3.fasta] ...]  [-k  value]
       [-c  value]  [-d  value]  [-O value] [-o name] [-u name] [-n] [-I] [-i index_stride] [-m align_file] [-s]
       [-f] [-h]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Checks for each sequence contained into the toCheck.fasta if  it  is  read  coherent  (each  position  is
       covered  by  at  least  "min_coverage" read(s)) with reads from readsA.fasta or readsB.fasta A sequence s
       from toCheck is treated as follow:

              if (s coherent with at least one read set): output the sequence as follows

              &gt;original fasta comment|C1:min&lt;avg-corr_avg&lt;max|C2:min&lt;avg-cor_avg&lt;max|C3...: &gt;s

              With A:min&lt;avg-cor_avg&lt;max standing for :  value  of  the  position  having  minimal  coverage  in
              readsA.fasta  &lt;  average  coverage  in  readsA.fasta - R-squarred corrected average in readsA.fa &lt;
              value of the position having maximal coverage in readsA.fasta The coverage is the number of  reads
              that  perfectly  mapped  a position Any other situation (s not coherent with any): couple non read
              coherent, not outputed

</pre><h4><b>OPTIONS</b></h4><pre>

       <b>-k</b> size_seed: will use seeds of length size_seed. Default: 25.

       <b>-O</b> minimal_read_overlap: a read can be mapped if its overlap is a least "minimal_read_overlap".  Default:
              k

       <b>-c</b> min_coverage: a sequence is covered by at least min_coverage coherent reads. Default: 2

       <b>-d</b> max_substitutions: Maximal number of substitutions authorized between a read and a fragment. Note that
              no substitution is allowed on the central position while anaylizing the kissnp output. Default: 1.

       <b>-o</b> file_name: write read-coherent outputs. Default: standard output

       <b>-u</b> file_name: write unread-coherent outputs. Default: standard output

       <b>-n</b> the input file (toCheck.fasta) is a kissnp output (incompatible with <b>-I</b> option)

              in  this case: 1/ only the upper characters are considered (no mapping done on the extensions) and
              2/ the central position (where the SNP occurs) is strictly mapped, no substitution  is  authorized
              on this position.

       <b>-I</b> the input file (toCheck.fasta) is an Intl output (incompatible with <b>-n</b> option)

       <b>-i</b>  index_stride  (int value). This is a heuristic for limiting the memory footprint. Instead of indexing
              each kmer of the sequences contained into the toCheck.fasta,  kissreads  indexes  kmers  occurring
              each "index_stride" position. Default = 1 (no heuristic)

       <b>-t</b> max number of threads (also limited by number of input files)

       <b>-m</b> align_file, write a file of reads mapped to sequences in file align_file

       <b>-s</b> silent mode

       <b>-f</b> outputs coherent events in a standard fasta file format

       <b>-h</b> prints this message and exit

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

mapsembler2_kissreads 2.2.4                       October 2018                          <u><a href="../man1/MAPSEMBLER2_KISSREADS.1.html">MAPSEMBLER2_KISSREADS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>