<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>mapsembler_extend  -  extends  a  DNA  sequence of interest given a set of reads, by computing a targeted</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/mapsembler2">mapsembler2_2.2.4+dfsg1-4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       mapsembler_extend  -  extends  a  DNA  sequence of interest given a set of reads, by computing a targeted
       micro assembly

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>mapsembler_extend</b>  &lt;extrem_kmers.fasta&gt;  &lt;readsC1.fasta&gt;  [&lt;readsC2.fasta&gt;  [&lt;readsC3.fasta]   ...]   [-t
       extension_type] [-k value] [-c value] [-g value] [-i index_name] [-o name] [-h]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Mapsembler  2  extends  a  DNA  sequence  of interest given a set of reads, by computing a targeted micro
       assembly.

</pre><h4><b>OPTIONS</b></h4><pre>

       <b>-t</b> extension_type. Default: 1

              1: a strict sequence: any branching stops the extension

              2: a consensus sequence: contiging approach

              3: a strict graph: any branching is conserved in the graph

              4: a consensus graph: "small" polymorphism is merged, but "large" structures are represented

       <b>-k</b> size_kmers: Size of the k-mers used duriung the extension phase Default: 31. Accepted  range,  depends
              on the compilation (make k=42 for instance)

       <b>-c</b> min_coverage: a sequence is covered by at least min_coverage coherent reads. Default: 2

       <b>-g</b> estimated_genome_size: estimation of the size of the genome whose reads come from.

              It is in bp, does not need to be accurate, only controls memory usage. Default: 3 billion

       <b>-x</b> node_len: limit max of nodes length. Default: 40

       <b>-y</b> graph_max_depth: limit max of graph depth.Default: 10000

       <b>-i</b>  index_name:  stores  the  index files in files starting with this prefix name. Can be re-used latter.
              Default: "index"

              IF THE FILE "index_name.bloom" EXISTS: the index is not re-created

       <b>-o</b> file_name_prefix: where to write outputs. Default: "res_mapsembler"

       <b>-p</b> search_mod: kind of prosses Breadth or Depth. Default: Breadth

       <b>-h</b> prints this message and exit

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

mapsembler_extend 2.2.4                           October 2018                              <u><a href="../man1/MAPSEMBLER_EXTEND.1.html">MAPSEMBLER_EXTEND</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>