<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>mhap - probabilistic sequence overlapping</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/mhap">mhap_2.1.3+dfsg-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       mhap - probabilistic sequence overlapping

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>mhap</b> <b>-s</b>&lt;fasta/dat from/self file&gt; [-q&lt;fasta/dat to file&gt;] [-f&lt;kmer filter list, must be sorted&gt;]
       <b>java</b>  <b>-server</b>  <b>-Xmx</b>&lt;memory&gt;  <b>-jar</b>  $(which  mhap)  <b>-s</b>&lt;fasta/dat  from/self  file&gt; [-q&lt;fasta/dat to file&gt;]
       [-f&lt;kmer filter list, must be sorted&gt;]
       generate precomputed binaries:
       <b>java</b> <b>-server</b> <b>-Xmx</b>&lt;memory&gt; <b>-jar</b> $(which mhap) <b>-p</b>&lt;directory of fasta files&gt; <b>-q</b> &lt;output directory&gt;  [-f&lt;kmer
       filter list, must be sorted&gt;]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       MHAP  (MinHash  Alignment Protocol) is a tool for finding overlaps of long-read sequences (such as PacBio
       or Nanopore) in bioinformatics.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b> <b>--help</b>
              Show summary of available options.

       <b>--version</b>
              Displays the version.

mhap                                                July 2016                                            <u><a href="../man1/MHAP.1.html">MHAP</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>