<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>samtools-fasta,-samtools-fastq - converts a SAM/BAM/CRAM file to FASTA or FASTQ</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/samtools">samtools_1.21-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       samtools-fasta,-samtools-fastq - converts a SAM/BAM/CRAM file to FASTA or FASTQ

</pre><h4><b>SYNOPSIS</b></h4><pre>
       samtools fastq [<u>options</u>] <u>in.bam</u>
       samtools fasta [<u>options</u>] <u>in.bam</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Converts a BAM or CRAM into either FASTQ or FASTA format depending on the command invoked. The files will
       be automatically compressed if the file names have a .gz, .bgz, or .bgzf extension.

       Note  this  command is attempting to reverse the alignment process, so if the aligner took a single input
       FASTQ and produced multiple SAM records via supplementary and/or secondary  alignments,  then  converting
       back  to  FASTQ  again  should  produce the original single FASTA / FASTQ record.  By default it will not
       attempt to records for supplementary and secondary alignments, but see the <b>-F</b> option for more details.

       If the input contains read-pairs which are to be interleaved or written to separate  files  in  the  same
       order,  then  the  input  should  be first collated by name.  Use <b>samtools</b> <b>collate</b> or <b>samtools</b> <b>sort</b> <b>-n</b> to
       ensure this.

       For each different QNAME, the input records are categorised according to the state of the READ1 and READ2
       flag bits.  The three categories used are:

       1 : Only READ1 is set.

       2 : Only READ2 is set.

       0 : Either both READ1 and READ2 are set; or neither is set.

       The exact meaning of these categories depends on the sequencing technology used.   It  is  expected  that
       ordinary  single and paired-end sequencing reads will be in categories 1 and 2 (in the case of paired-end
       reads, one read of the pair will be in category 1, the other in category 2).  Category 0 is essentially a
       “catch-all” for reads that do not fit into a simple paired-end sequencing model.

       For each category only one sequence will be written for a given  QNAME.   If  more  than  one  record  is
       available  for  a given QNAME and category, the first in input file order that has quality values will be
       used.  If none of the candidate records has quality values, then the first in input file  order  will  be
       used instead.

       Sequences  will  be  written  to  standard output unless one of the <b>-1</b>, <b>-2</b>, <b>-o</b>, or <b>-0</b> options is used, in
       which case sequences for that category will be written to the specified file.  The same filename  may  be
       specified with multiple options, in which case the sequences will be multiplexed in order of occurrence.

       If  a  singleton  file  is  specified  using  the <b>-s</b> option then only paired sequences will be output for
       categories 1 and 2; paired meaning that for a given QNAME there are sequences for both category 1 <b>and</b>  2.
       If  there  is  a  sequence  for only one of categories 1 or 2 then it will be diverted into the specified
       singletons file.  This can be used to prepare fastq files for programs that cannot handle  a  mixture  of
       paired and singleton reads.

       The  <b>-s</b>  option  only  affects  category  1  and  2  records.  The output for category 0 will be the same
       irrespective of the use of this option.

       The sequence generated will be for the entire sequence  recorded  in  the  SAM  record  (and  quality  if
       appropriate).   This means if it has soft-clipped CIGAR records then the soft-clipped data will be in the
       output FASTA/FASTQ.  Hard-clipped data is, by definition, absent from the SAM record and  hence  will  be
       absent in any FASTA/FASTQ produced.

       The filter options order of precedence is -d/-D, -f, -F, --rf and -G.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-n</b>      By  default,  either '/1' or '/2' is added to the end of read names where the corresponding READ1
               or READ2 FLAG bit is set.  Using <b>-n</b> causes read names to be left as they are.

       <b>-N</b>      Always add either '/1' or '/2' to the end of read names even when put into different files.

       <b>-O</b>      Use quality values from OQ tags in preference to standard quality string if available.

       <b>-s</b> <b>FILE</b> Write singleton reads to FILE.

       <b>-t</b>      Copy RG, BC and QT tags to the FASTQ header line, if they exist.

       <b>-T</b> <b>TAGLIST</b>
               Specify a comma-separated list of tags to copy to the FASTQ header line, if they exist.   <b>TAGLIST</b>
               can be blank or <b>*</b> to indicate all tags should be copied to the output.  If using <b>*</b>, be careful to
               quote it to avoid unwanted shell expansion.

       <b>-1</b> <b>FILE</b> Write  reads  with the READ1 FLAG set (and READ2 not set) to FILE instead of outputting them.  If
               the <b>-s</b> option is used, only paired reads will be written to this file.

       <b>-2</b> <b>FILE</b> Write reads with the READ2 FLAG set (and READ1 not set) to FILE instead of outputting  them.   If
               the <b>-s</b> option is used, only paired reads will be written to this file.

       <b>-o</b> <b>FILE</b> Write  reads  with  either  READ1  FLAG  or  READ2 flag set to FILE instead of outputting them to
               stdout.  This is equivalent to <b>-1</b> FILE <b>-2</b> FILE.

       <b>-0</b> <b>FILE</b> Write reads where the READ1 and READ2 FLAG bits set are either both set or  both  unset  to  FILE
               instead of outputting them.

       <b>-f</b> <u>INT</u>  Only  output alignments with all bits set in <u>INT</u> present in the FLAG field.  <u>INT</u> can be specified
               in hex by beginning with `0x' (i.e. /^0x[0-9A-F]+/) or in  octal  by  beginning  with  `0'  (i.e.
               /^0[0-7]+/) [0].

       <b>-F</b> <u>INT</u><b>,</b> <u>,</u><b>--excl-flags</b> <u>INT</u> <b>,</b><u>--exclude-flags</u> <b>INT</b>
               Do  not  output  alignments  with  any  bits  set  in  <u>INT</u> present in the FLAG field.  <u>INT</u> can be
               specified in hex by beginning with `0x' (i.e. /^0x[0-9A-F]+/) or in octal by beginning  with  `0'
               (i.e.  /^0[0-7]+/)  [0x900].   This  defaults  to  0x900  representing filtering of secondary and
               supplementary alignments.

       <b>--rf</b> <u>INT</u> <b>,</b> <b>--incl-flags</b> <u>INT</u><b>,</b> <b>--include-flags</b> <u>INT</u>
               Only output alignments with any bits set in <u>INT</u> present in the FLAG field.  <u>INT</u> can be  specified
               in  hex  by  beginning  with  `0x'  (i.e.  /^0x[0-9A-F]+/),  in octal by beginning with `0' (i.e.
               /^0[0-7]+/), as a decimal number not beginning with '0' or as  a  comma-separated  list  of  flag
               names [0].

       <b>-G</b> <u>INT</u>  Only  EXCLUDE  reads  with  all  of  the  bits  set in <u>INT</u> present in the FLAG field.  <u>INT</u> can be
               specified in hex by beginning with `0x' (i.e. /^0x[0-9A-F]+/) or in octal by beginning  with  `0'
               (i.e. /^0[0-7]+/) [0].

       <b>-d</b> <u>TAG</u><b>[:</b><u>VAL</u><b>]</b>
               Only  output alignments containing an auxiliary tag matching both <u>TAG</u> and <u>VAL</u>.  If <u>VAL</u> is omitted
               then any value is accepted.  The tag types supported are i, f, Z, A and H.  "B"  arrays  are  not
               supported.   This  is  comparable  to  the  method  used  in <b>samtools</b> <b>view</b> <b>-d</b>.  The option may be
               specified multiple times and is equivalent to using the <b>-D</b> option.

       <b>-D</b> <u>TAG:FILE</u>
               Only output alignments containing an auxiliary tag matching <u>TAG</u> and  having  a  value  listed  in
               <u>FILE</u>.   The  format  of  the  file  is  one  line per value.  This is equivalent to specifying <b>-d</b>
               multiple times.

       <b>-i</b>      add Illumina Casava 1.8 format entry to header (eg 1:N:0:ATCACG)

       <b>-c</b> <b>[0..9]</b>
               set compression level when writing gz or bgzf fastq files.

       <b>--i1</b> <b>FILE</b>
               write first index reads to FILE

       <b>--i2</b> <b>FILE</b>
               write second index reads to FILE

       <b>--barcode-tag</b> <b>TAG</b>
               aux tag to find index reads in [default: BC]

       <b>--quality-tag</b> <b>TAG</b>
               aux tag to find index quality in [default: QT]

       <b>-@,</b> <b>--threads</b> <u>INT</u>
               Number of input/output compression threads to use in addition to main thread [0].

       <b>--index-format</b> <b>STR</b>
               string to describe how to parse the barcode and quality tags. For example:

               <b>i14i8</b>   the first 14 characters are index 1, the next 8 characters are index 2

               <b>n8i14</b>   ignore the first 8 characters, and use the next 14 characters for index 1

                       If the tag contains a separator, then the numeric part can be replaced with '*'  to  mean
                       'read until the separator or end of tag', for example:

               <b>n*i*</b>    ignore the left part of the tag until the separator, then use the second part

</pre><h4><b>EXAMPLES</b></h4><pre>
       Starting  from  a  coordinate  sorted file, output paired reads to separate files, discarding singletons,
       supplementary and secondary reads.  The resulting files can be used with, for example, the <b>bwa</b> aligner.

           samtools collate -u -O in_pos.bam | \
           samtools fastq -1 paired1.fq -2 paired2.fq -0 /dev/null -s /dev/null -n

       Starting with a name collated file, output paired and  singleton  reads  in  a  single  file,  discarding
       supplementary and secondary reads.  To get all of the reads in a single file, it is necessary to redirect
       the  output  of  samtools  fastq.   The output file is suitable for use with <b>bwa</b> <b>mem</b> <b>-p</b> which understands
       interleaved files containing a mixture of paired and singleton reads.

           samtools fastq -0 /dev/null in_name.bam &gt; all_reads.fq

       Output paired reads in a single file, discarding supplementary and secondary reads.  Save any  singletons
       in  a separate file.  Append /1 and /2 to read names.  This format is suitable for use by <b>NextGenMap</b> when
       using its <b>-p</b> and <b>-q</b> options.   With  this  aligner,  paired  reads  must  be  mapped  separately  to  the
       singletons.

           samtools fastq -0 /dev/null -s single.fq -N in_name.bam &gt; paired.fq

</pre><h4><b>BUGS</b></h4><pre>
       o The way of specifying output files is far too complicated and easy to get wrong.

</pre><h4><b>AUTHOR</b></h4><pre>
       Written  by  Heng  Li,  with  modifications  by  Martin  Pollard and Jennifer Liddle, all from the Sanger
       Institute.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/samtools.1.html">samtools</a></u>(1), <u><a href="../man1/samtools-faidx.1.html">samtools-faidx</a></u>(1), <u><a href="../man1/samtools-fqidx.1.html">samtools-fqidx</a></u>(1) <u><a href="../man1/samtools-import.1.html">samtools-import</a></u>(1)

       Samtools website: &lt;<a href="http://www.htslib.org/">http://www.htslib.org/</a>&gt;

samtools-1.21                                   12 September 2024                              <u><a href="../man1/samtools-fasta.1.html">samtools-fasta</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>