<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>samtools-dict - create a sequence dictionary file from a fasta file</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/samtools">samtools_1.21-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       samtools-dict - create a sequence dictionary file from a fasta file

</pre><h4><b>SYNOPSIS</b></h4><pre>
       samtools dict <u>ref.fasta</u>|<u>ref.fasta.gz</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Create a sequence dictionary file from a fasta file.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-a,</b> <b>--assembly</b> <u>STR</u>
                  Specify the assembly for the AS tag.

       <b>-A,</b> <b>--alias,</b> <b>--alternative-name</b>
                  Add  an  AN tag with the same value as the SN tag, except that a “chr” prefix is removed if SN
                  has one or added if it does not.  For mitochondria (i.e., when SN is  “M”  or  “MT”,  with  or
                  without a “chr” prefix), also adds the remaining combinations of “chr/M/MT” to the AN tag.

       <b>-H,</b> <b>--no-header</b>
                  Do not print the @HD header line.

       <b>-l,</b> <b>--alt</b> <u>FILE</u>
                  Add  an  AH  tag to each sequence listed in the specified <u><a href="../man1/bwa.1.html">bwa</a></u>(1)-style <b>.alt</b> file.  These files
                  use SAM records to represent alternate locus sequences (as named in the <b>QNAME</b> field) and their
                  mappings to the primary assembly.

       <b>-o,</b> <b>--output</b> <u>FILE</u>
                  Output to <u>FILE</u> [stdout].

       <b>-s,</b> <b>--species</b> <u>STR</u>
                  Specify the species for the SP tag.

       <b>-u,</b> <b>--uri</b> <u>STR</u>
                  Specify the URI for the UR tag. Defaults to the absolute path of <u>ref.fasta</u> unless reading from
                  stdin.

</pre><h4><b>AUTHOR</b></h4><pre>
       Written by Shane McCarthy from the Sanger Institute.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/samtools.1.html">samtools</a></u>(1), <u><a href="../man1/bcftools.1.html">bcftools</a></u>(1), <u><a href="../man1/bwa.1.html">bwa</a></u>(1), <u><a href="../man5/sam.5.html">sam</a></u>(5), <u><a href="../man1/tabix.1.html">tabix</a></u>(1)

       Samtools website: &lt;<a href="http://www.htslib.org/">http://www.htslib.org/</a>&gt;
       File format specification of SAM/BAM,CRAM,VCF/BCF: &lt;<a href="http://samtools.github.io/hts-specs">http://samtools.github.io/hts-specs</a>&gt;
       Samtools latest source: &lt;https://github.com/samtools/samtools&gt;
       HTSlib latest source: &lt;https://github.com/samtools/htslib&gt;
       Bcftools website: &lt;<a href="http://samtools.github.io/bcftools">http://samtools.github.io/bcftools</a>&gt;

samtools-1.21                                   12 September 2024                               <u><a href="../man1/samtools-dict.1.html">samtools-dict</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>