<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Search::Tiling::TilingI - Abstract interface for an HSP tiling module</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Search::Tiling::TilingI - Abstract interface for an HSP tiling module

</pre><h4><b>SYNOPSIS</b></h4><pre>
       Not used directly. Useful POD here for developers, however.

       The interface is designed to make the following code conversion as simple as possible:

       From:

        # Bio::Search::SearchUtils-based
        while ( local $_ = $result-&gt;next_hit ) {
           printf( "E-value: %g; Fraction aligned: %f; Number identical: %d\n",
             $hit-&gt;significance, $hit-&gt;frac_aligned_query, $hit-&gt;num_identical);
        }

       To:

        # TilingI-based
        while ( local $_ = $result-&gt;next_hit ) {
           my $tiling = Bio::Search::Tiling::MyTiling($_);
           printf( "E-value: %g; Fraction aligned: %f; Number identical: %d\n",
             $hit-&gt;significance, $tiling-&gt;frac_aligned_query, $tiling-&gt;num_identical);
        }

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This module provides strong suggestions for any intended HSP tiling object implementation. An object
       subclassing TilingI should override the methods defined here according to their descriptions below.

       See the section STATISTICS METHODS for hints on implementing methods that are valid across different
       algorithms and report types.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Mark</b> <b>A.</b> <b>Jensen</b></h4><pre>
       Email <a href="mailto:maj@fortinbras.us">maj@fortinbras.us</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>STATISTICS</b> <b>METHODS</b>
       The tiling statistics can be thought of as global counterparts to similar statistics defined for the
       individual HSPs. We therefore prescribe definitions for many of the synonymous methods defined in
       Bio::Search::HSP::HSPI.

       The tiling statistics must be able to keep track of the coordinate systems in which both the query and
       subject sequences exist; i.e., either nucleotide or amino acid. This information is typically inferred
       from the name of the algorithm used to perform the original search (contained in
       "$hit_object-&gt;algorithm"). Here is a table of algorithm information that may be useful (if you trust us).

        algorithm   query on hit   coordinates(q/h)
        ---------   ------------   ---------------
         blastn      dna on dna         dna/dna
         blastp      aa  on aa           aa/aa
         blastx      xna on aa          dna/aa
        tblastn      aa  on xna          aa/dna
        tblastx      xna on xna         dna/dna
          fasta      dna on dna         dna/dna
          fasta      aa  on aa           aa/aa
          fastx      xna on aa          dna/aa
          fasty      xna on aa          dna/aa
         tfasta      aa  on xna          aa/dna
         tfasty      aa  on xna          aa/dna
        megablast    dna on dna         dna/dna

         xna: translated nucleotide data

       Statistics methods must also be aware of differences in reporting among the algorithms. Hit attributes
       are not necessarily normalized over all algorithms. Devs, please feel free to add examples to the list
       below.

       NCBI BLAST vs WU-BLAST (AB-BLAST) lengths
           The   total   length   of   the   alignment  is  reported  differently  between  these  two  flavors.
           "$hit_object-&gt;length()" will contain the number in the denominator of the stats line; i.e., 120 in

            Identical = 34/120 Positives = 67/120

           NCBI BLAST uses the total length of the query sequence as input by the user (a.k.a. "with gaps"). WU-
           BLAST uses the length of the query sequence  actually  aligned  by  the  algorithm  (a.k.a.  "without
           gaps").

       Finally,  developers  should  remember  that  sequence  data  may  or may not be associated with the HSPs
       contained in the hit object. This will typically depend on whether a full report (e.g, "blastall -m0") or
       a summary (e.g., "blastall -m8") was parsed. Statistics methods that depend directly on the sequence data
       will need to check that that data is present.

   <b>identities</b>
        Title   : identities
        Alias   : num_identical
        Usage   : $num_identities = $tiling-&gt;identities()
        Function: Return the estimated or exact number of identities in the
                  tiling, accounting for overlapping HSPs
        Example :
        Returns : number of identical residue pairs
        Args    :

   <b>conserved</b>
        Title   : conserved
        Alias   : num_conserved
        Usage   : $num_conserved = $tiling-&gt;conserved()
        Function: Return the estimated or exact number of conserved sites in the
                  tiling, accounting for overlapping HSPs
        Example :
        Returns : number of conserved residue pairs
        Args    :

   <b>length</b>
        Title   : length
        Usage   : $max_length = $tiling-&gt;length($type)
        Function: Return the total number of residues of the subject or query
                  sequence covered by the tiling
        Returns : number of "logical" residues covered
        Args    : scalar $type, one of 'hit', 'subject', 'query'

   <b>frac_identical</b>
        Title   : frac_identical
        Usage   : $tiling-&gt;frac_identical($type)
        Function: Return the fraction of sequence length consisting
                  of identical pairs
        Returns : scalar float
        Args    : scalar $type, one of 'hit', 'subject', 'query'
        Note    : This method must take account of the $type coordinate
                  system and the length reporting method (see STATISTICS
                  METHODS above)

   <b>percent_identity</b>
        Title   : percent_identity
        Usage   : $tiling-&gt;percent_identity($type)
        Function: Return the fraction of sequence length consisting
                  of identical pairs as a percentage
        Returns : scalar float
        Args    : scalar $type, one of 'hit', 'subject', 'query'

   <b>frac_conserved</b>
        Title   : frac_conserved
        Usage   : $tiling-&gt;frac_conserved($type)
        Function: Return the fraction of sequence length consisting
                  of conserved pairs
        Returns : scalar float
        Args    : scalar $type, one of 'hit', 'subject', 'query'
        Note    : This method must take account of the $type coordinate
                  system and the length reporting method (see STATISTICS
                  METHODS above)

   <b>percent_conserved</b>
        Title   : percent_conserved
        Usage   : $tiling-&gt;percent_conserved($type)
        Function: Return the fraction of sequence length consisting
                  of conserved pairs as a percentage
        Returns : scalar float
        Args    : scalar $type, one of 'hit', 'subject', 'query'

   <b>frac_aligned</b>
        Title   : frac_aligned
        Usage   : $tiling-&gt;frac_aligned($type)
        Function: Return the fraction of B&lt;input&gt; sequence length consisting
                  that was aligned by the algorithm
        Returns : scalar float
        Args    : scalar $type, one of 'hit', 'subject', 'query'
        Note    : This method must take account of the $type coordinate
                  system and the length reporting method (see STATISTICS
                  METHODS above)

   <b>range</b>
        Title   : range
        Usage   : $tiling-&gt;range($type)
        Function: Returns the extent of the longest tiling
                  as ($min_coord, $max_coord)
        Returns : array of two scalar integers
        Args    : scalar $type, one of 'hit', 'subject', 'query'

</pre><h4><b>TILING</b> <b>ITERATORS</b></h4><pre>
   <b>next_tiling</b>
        Title   : next_tiling
        Usage   : @hsps = $self-&gt;next_tiling($type);
        Function: Obtain a tiling of HSPs over the $type ('hit', 'subject',
                  'query') sequence
        Example :
        Returns : an array of HSPI objects
        Args    : scalar $type: one of 'hit', 'subject', 'query', with
                  'subject' an alias for 'hit'

   <b>rewind_tilings</b>
        Title   : rewind_tilings
        Usage   : $self-&gt;rewind_tilings($type)
        Function: Reset the next_tilings($type) iterator
        Example :
        Returns : True on success
        Args    : scalar $type: one of 'hit', 'subject', 'query', with
                  'subject' an alias for 'hit'

</pre><h4><b>INFORMATIONAL</b> <b>ACCESSORS</b></h4><pre>
   <b>algorithm</b>
        Title   : algorithm
        Usage   : $tiling-&gt;algorithm
        Function: Retrieve the algorithm name associated with the
                  invocant's hit object
        Returns : scalar string
        Args    :

perl v5.32.1                                       2021-08-15                  <u>Bio::Search::Tiling::<a href="../man3pm/TilingI.3pm.html">TilingI</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>