<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>create_matrix - calculate the genome abundance similarity matrix</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/gasic">gasic_0.0.r19-8_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       create_matrix - calculate the genome abundance similarity matrix

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>create_matrix</b> [<u>options</u>] <u>NAMES</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Calculate the similarity matrix.

       First,  a  set of reads is simulated for every reference genome using a read simulator from core/tools.py
       specified via <b>-s</b>.  Second, the simulated reads of each species are mapped against all  reference  genomes
       using  the  mapper  specified  with  <b>-m</b>.   Third,  the  resulting SAM-files are analyzed to calculate the
       similarity matrix. The similarity matrix is stored as a numpy file (<b>-o</b>).

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>NAMES</b>  Filename of the names file; the plain text names file should contain one name per line.  The  name
              is used as identifier in the whole algorithm.

       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>-s</b> SIMULATOR, <b>--simulator</b>=<u>SIMULATOR</u>
              Identifier of read simulator defined in core/tools.py [default: none]

       <b>-r</b> REF, <b>--reference</b>=<u>REF</u>
              Reference  sequence  file  pattern  for  the  read  simulator.  Placeholder  for the name is "%s".
              [default: ./ref/%s.fasta]

       <b>-m</b> MAPPER, <b>--mapper</b>=<u>MAPPER</u>
              Identifier of mapper defined in core/tools.py [default: none]

       <b>-i</b> INDEX, <b>--index</b>=<u>INDEX</u>
              Reference index  files  for  the  read  mapper.  Placeholder  for  the  name  is  "%s".  [default:
              ./ref/%s.fasta]

       <b>-t</b> TEMP, <b>--temp</b>=<u>TEMP</u>
              Directory to store temporary simulated datasets and SAM files. [default: ./temp]

       <b>-o</b> OUT, <b>--output</b>=<u>OUT</u>
              Output similarity matrix file. [default: ./similarity_matrix.npy]

create_matrix SVNr18                              February 2014                                 <u><a href="../man1/CREATE_MATRIX.1.html">CREATE_MATRIX</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>