<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>run_gubbins - phylogenetic analysis of genome sequences</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/gubbins">gubbins_3.4-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       run_gubbins - phylogenetic analysis of genome sequences

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>run_gubbins</b>  [-h]  [--outgroup  OUTGROUP]  [--starting_tree STARTING_TREE] [--use_time_stamp] [--verbose]
       [--no_cleanup]   [--tree_builder   TREE_BUILDER]   [--iterations   ITERATIONS]   [--min_snps    MIN_SNPS]
       [--filter_percentage   FILTER_PERCENTAGE]   [--prefix   PREFIX]  [--threads  THREADS]  [--converge_method
       CONVERGE_METHOD]  [--version]  [--min_window_size  MIN_WINDOW_SIZE]  [--max_window_size  MAX_WINDOW_SIZE]
       alignment_filename

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Gubbins  supports  rapid  phylogenetic  analysis  of  large samples of recombinant bacterial whole genome
       sequences.

       Gubbins (Genealogies Unbiased By recomBinations In Nucleotide Sequences) is an algorithm that iteratively
       identifies loci containing elevated densities of base substitutions  while  concurrently  constructing  a
       phylogeny  based  on  the  putative point mutations outside of these regions. Simulations demonstrate the
       algorithm generates highly accurate  reconstructions  under  realistic  models  of  short-term  bacterial
       evolution, and can be run in only a few hours on alignments of hundreds of bacterial genome sequences.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>positional</b> <b>arguments:</b>
       alignment_filename
              Multifasta alignment file

   <b>optional</b> <b>arguments:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>--outgroup</b> OUTGROUP, <b>-o</b> OUTGROUP
              Outgroup name for rerooting. A list of comma separated names can be used if they form a clade

       <b>--starting_tree</b> STARTING_TREE, <b>-s</b> STARTING_TREE
              Starting tree

       <b>--use_time_stamp</b>, <b>-u</b>
              Use a time stamp in file names

       <b>--verbose</b>, <b>-v</b>
              Turn on debugging

       <b>--no_cleanup</b>, <b>-n</b>
              Don't cleanup intermediate files

       <b>--tree_builder</b> TREE_BUILDER, <b>-t</b> TREE_BUILDER
              Application to use for tree building [raxml|fasttree|hybrid], default RAxML

       <b>--iterations</b> ITERATIONS, <b>-i</b> ITERATIONS
              Maximum No. of iterations, default is 5

       <b>--min_snps</b> MIN_SNPS, <b>-m</b> MIN_SNPS
              Min SNPs to identify a recombination block, default is 3

       <b>--filter_percentage</b> FILTER_PERCENTAGE, <b>-f</b> FILTER_PERCENTAGE
              Filter out taxa with more than this percentage of gaps, default is 25

       <b>--prefix</b> PREFIX, <b>-p</b> PREFIX
              Add a prefix to the final output filenames

       <b>--threads</b> THREADS, <b>-c</b> THREADS
              Number of threads to run with RAXML, but only if a PTHREADS version is available

       <b>--converge_method</b> CONVERGE_METHOD, <b>-z</b> CONVERGE_METHOD
              Criteria      to      use      to      know      when      to      halt      iterations     [weigh
              ted_robinson_foulds|robinson_foulds|recombination]

       <b>--version</b>
              show program's version number and exit

       <b>--min_window_size</b> MIN_WINDOW_SIZE, <b>-a</b> MIN_WINDOW_SIZE
              Minimum window size, default 100

       <b>--max_window_size</b> MAX_WINDOW_SIZE, <b>-b</b> MAX_WINDOW_SIZE
              Maximum window size, default 10000

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Croucher N. J., Page A. J., Connor T. R., Delaney A. J., Keane J. A., Bentley S. D., Parkhill J.,  Harris
       S.R.  "Rapid phylogenetic analysis of large samples of recombinant bacterial whole genome sequences using
       Gubbins". Nucleic Acids Res. 2015 Feb 18;<a href="../man3/43.3.html">43</a>(3):e15. doi: 10.1093/nar/gku1196.

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

run_gubbins 2.0.0                                   Julu 2016                                     <u><a href="../man1/RUN_GUBBINS.1.html">RUN_GUBBINS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>