<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Variation::AAReverseMutate - point mutation and codon</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-variation-perl">libbio-variation-perl_1.7.5-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Variation::AAReverseMutate - point mutation and codon
         information from single amino acid changes

</pre><h4><b>SYNOPSIS</b></h4><pre>
           $aamut = Bio::Variation::AAReverseMutate-&gt;new
                        (-aa_ori =&gt; 'F',
                         -aa_mut =&gt; 'S',
                         -codon_ori =&gt; 'ttc', # optional
                         -codon_table =&gt; '3'  # defaults to 1
                         );

          @points = $aamut-&gt;each_Variant;

          if (scalar @points &gt; 0 ) {
              foreach $rnachange ( @points ) {
                  # $rnachange is a Bio::Variation::RNAChange object
                  print " ", $rnachange-&gt;allele_ori-&gt;seq, "&gt;",
                  $rnachange-&gt;allele_mut-&gt;seq, " in ",
                  $rnachange-&gt;codon_ori, "&gt;", $rnachange-&gt;codon_mut,
                  " at position ", $rnachange-&gt;codon_pos, "\n";
              }
          } else  {
              print "No point mutations possible\n",
          }

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::Variation::AAReverseMutate objects take in reference and mutated amino acid information and deduces
       potential point mutations at RNA level leading to this change. The choice can be further limited by
       letting the object know what is the the codon in the reference sequence. The results are returned as
       Bio::Variation::RNAChange objects.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing lists  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Heikki</b> <b>Lehvaslaiho</b></h4><pre>
       Email:  heikki-at-bioperl-dot-org

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>aa_ori</b>
        Title   : aa_ori
        Usage   : $obj-&gt;aa_ori();
        Function:

                   Sets and returns original aa sequence.  If value is not
                   set, returns false.

                   Amino acid sequences are stored in upper case characters,
                   others in lower case.

        Example :
        Returns : string
        Args    : single character amino acid code

   <b>aa_mut</b>
        Title   : aa_mut
        Usage   : $obj-&gt;aa_mut();
        Function:

                   Sets and returns the mutated allele sequence.  If value is not
                   set, returns false.

        Example :
        Returns : string
        Args    : single character amino acid code

   <b>codon_ori</b>
        Title   : codon_ori
        Usage   : $obj-&gt;codon_ori();
        Function:

                   Sets and returns codon_ori triplet.  If value is not set,
                   returns false.  The string has to be three characters
                   long. The character content is not checked.

        Example :
        Returns : string
        Args    : string

   <b>codon_table</b>
        Title   : codon_table
        Usage   : $obj-&gt;codon_table();
        Function:

                   Sets and returns the codon table id of the RNA
                   If value is not set, returns 1, 'universal' code, as the default.

        Example :
        Returns : integer
        Args    : none if get, the new value if set

   <b>each_Variant</b>
        Title   : each_Variant
        Usage   : $obj-&gt;each_Variant();
        Function:

                   Returns a list of Variants.

        Example :
        Returns : list of Variants
        Args    : none

perl v5.32.1                                       2021-12-02               <u>Bio::Variation::<a href="../man3pm/AAReverseMutate.3pm.html">AAReverseMutate</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>