<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>bam12auxmerge - merge two BAM files</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/biobambam2">biobambam2_2.0.185+ds-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       bam12auxmerge - merge two BAM files

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>bam12auxmerge</b> [options] &lt;input.bam&gt;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       bam12auxmerge  merges  the contents of two BAM files containing the same set of reads together and writes
       the resulting data as a BAM file.  It reads one of the BAM files from standard input, the  other  one  is
       given via its file name.

       It  is  assumed  that  the file (A) given by file name was obtained by processing another BAM file by the
       sequence

       bamcollate2 collate=1 | bamcollate2 collate=2

       where the first run of bamcollate2 makes sure that the ends of pairs appear adjacently and the second run
       attaches rank pairs to the read names.

       The other file (B) read by bam12auxmerge via standard input is assumed be a (re)aligned version  of  file
       (A), which is processed by

       bamsort SO=queryname

       right before being passed to bam12auxmerge.

       This  preprocessing ensures that bam12auxmerge observes the reads in the same order, with the possibility
       of file (B) containing multiple lines for each read (i.e. secondary alignments).

       bam12auxmerge merge the SAM headers of (A) and (B) by starting with the header for (B) and prepending the
       PG type lines from (A), thus restoring the full history of the file to  (B).  Alignments  are  merged  by
       inserting  the  auxiliary  tags from lines in (A) into the corresponding lines in (B).  Priority is given
       here to (B), i.e. if a tag appears in both than the version in (B) will be retained.

       The following key=value pairs can be given:

       <b>level=&lt;-1|0|1|9|11&gt;:</b> set compression level of the output BAM file. Valid values are

       -1:    zlib/gzip default compression level

       0:     uncompressed

       1:     zlib/gzip level 1 (fast) compression

       9:     zlib/gzip level 9 (best) compression

       If libmaus has been compiled with support for igzip (see https://software.intel.com/en-us/articles/igzip-
       a-high-performance-deflate-compressor-with-optimizations-for-genomic-data) then an additional valid value
       is

       11:    igzip compression

       <b>verbose=&lt;1&gt;:</b> Valid values are

       1:     print progress report on standard error (default)

       0:     do not print progress report

       <b>mod=&lt;1048576&gt;:</b> Print progress report after processing each mod'th alignment if verbose is active.

       <b>ranksplit=&lt;1&gt;:</b> Split rank pairs into single ranks per read/alignment. This acts  like  preprocessing  the
       file with bam12split before handing it to bam12auxmerge.

       <b>rankstrip=&lt;1&gt;:</b>  Strip ranks from reads/alignments. This acts like postprocessing the file with bam12strip
       after handing it to bam12auxmerge.

       <b>clipreinsert=&lt;1&gt;:</b> Reinsert clipped sequence and quality string parts while marking them as soft  clipped.
       This acts like postprocessing the file with bamclipreinsert after handing it to bam12auxmerge.

       <b>zztoname=&lt;1&gt;:</b>  Insert  rank  stored in zz auxiliary field to name as a prefix and remove the zz auxiliary
       field. This acts like postprocessing the file with bamzztoname after handing it to bam12auxmerge.

       <b>tmpfile=&lt;filename&gt;:</b> prefix for temporary files. By default the temporary files are created in the current
       directory

       <b>md5=&lt;0|1&gt;:</b> md5 checksum creation for output file. Valid values are

       0:     do not compute checksum. This is the default.

       1:     compute checksum. If the md5filename key is set, then the checksum is written to the  given  file.
              If md5filenameis unset, then no checksum will be computed.

       <b>md5filename</b> file name for md5 checksum if md5=1.

       <b>index=&lt;0|1&gt;:</b> compute BAM index for output file. Valid values are

       0:     do not compute BAM index. This is the default.

       1:     compute  BAM  index.  If  the indexfilename key is set, then the BAM index is written to the given
              file. If indexfilename is unset, then no BAM index will be computed.

       <b>indexfilename</b> file name for BAM index if index=1.

</pre><h4><b>AUTHOR</b></h4><pre>
       Written by German Tischler.

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       Report bugs to &lt;<a href="mailto:germant@miltenyibiotec.de">germant@miltenyibiotec.de</a>&gt;

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright © 2009-2013 German Tischler, © 2011-2013 Genome Research  Limited.   License  GPLv3+:  GNU  GPL
       version 3 &lt;<a href="http://gnu.org/licenses/gpl.html">http://gnu.org/licenses/gpl.html</a>&gt;
       This  is  free software: you are free to change and redistribute it.  There is NO WARRANTY, to the extent
       permitted by law.

BIOBAMBAM                                           July 2013                                   <u><a href="../man1/BAM12AUXMERGE.1.html">BAM12AUXMERGE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>