<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Recent versions (v7.1xx; 2013 Jan.) have more features than those described below.  See also the tips</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/mafft">mafft_7.505-1_amd64</a> <br><br><pre>
</pre><h4><b>THIS</b> <b>MANUAL</b> <b>IS</b> <b>FOR</b> <b>V6.2XX</b> <b>(2007)</b></h4><pre>
       Recent versions (v7.1xx; 2013 Jan.) have more features than those described below.  See also the tips
       page at <a href="http://mafft.cbrc.jp/alignment/software/tips0.html">http://mafft.cbrc.jp/alignment/software/tips0.html</a>

</pre><h4><b>NAME</b></h4><pre>
       mafft - Multiple alignment program for amino acid or nucleotide sequences

</pre><h4><b>SYNOPSIS</b></h4><pre>

       <b>mafft</b> [<b>options</b>] <u>input</u> [&gt; <u>output</u>]

       <b>linsi</b> <u>input</u> [&gt; <u>output</u>]

       <b>ginsi</b> <u>input</u> [&gt; <u>output</u>]

       <b>einsi</b> <u>input</u> [&gt; <u>output</u>]

       <b>fftnsi</b> <u>input</u> [&gt; <u>output</u>]

       <b>fftns</b> <u>input</u> [&gt; <u>output</u>]

       <b>nwns</b> <u>input</u> [&gt; <u>output</u>]

       <b>nwnsi</b> <u>input</u> [&gt; <u>output</u>]

       <b>mafft-profile</b> <u>group1</u> <u>group2</u> [&gt; <u>output</u>]

                     <u>input</u>, <u>group1</u> and <u>group2</u> must be in FASTA format.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>MAFFT</b> is a multiple sequence alignment program for unix-like operating systems. It offers a range of
       multiple alignment methods.

   <b>Accuracy-oriented</b> <b>methods:</b>
       •   L-INS-i (probably most accurate; recommended for &lt;200 sequences; iterative refinement method
           incorporating local pairwise alignment information):

           <b>mafft</b> <b>--localpair</b> <b>--maxiterate</b> <u>1000</u> <u>input</u> [&gt; <u>output</u>]

           <b>linsi</b> <u>input</u> [&gt; <u>output</u>]

       •   G-INS-i (suitable for sequences of similar lengths; recommended for &lt;200 sequences; iterative
           refinement method incorporating global pairwise alignment information):

           <b>mafft</b> <b>--globalpair</b> <b>--maxiterate</b> <u>1000</u> <u>input</u> [&gt; <u>output</u>]

           <b>ginsi</b> <u>input</u> [&gt; <u>output</u>]

       •   E-INS-i (suitable for sequences containing large unalignable regions; recommended for &lt;200
           sequences):

           <b>mafft</b> <b>--ep</b> <u>0</u> <b>--genafpair</b> <b>--maxiterate</b> <u>1000</u> <u>input</u> [&gt; <u>output</u>]

           <b>einsi</b> <u>input</u> [&gt; <u>output</u>]

                 For E-INS-i, the <b>--ep</b> <u>0</u> option is recommended to allow large gaps.

   <b>Speed-oriented</b> <b>methods:</b>
       •   FFT-NS-i (iterative refinement method; two cycles only):

           <b>mafft</b> <b>--retree</b> <u>2</u> <b>--maxiterate</b> <u>2</u> <u>input</u> [&gt; <u>output</u>]

           <b>fftnsi</b> <u>input</u> [&gt; <u>output</u>]

       •   FFT-NS-i (iterative refinement method; max. 1000 iterations):

           <b>mafft</b> <b>--retree</b> <u>2</u> <b>--maxiterate</b> <u>1000</u> <u>input</u> [&gt; <u>output</u>]

       •   FFT-NS-2 (fast; progressive method):

           <b>mafft</b> <b>--retree</b> <u>2</u> <b>--maxiterate</b> <u>0</u> <u>input</u> [&gt; <u>output</u>]

           <b>fftns</b> <u>input</u> [&gt; <u>output</u>]

       •   FFT-NS-1 (very fast; recommended for &gt;2000 sequences; progressive method with a rough guide tree):

           <b>mafft</b> <b>--retree</b> <u>1</u> <b>--maxiterate</b> <u>0</u> <u>input</u> [&gt; <u>output</u>]

       •   NW-NS-i (iterative refinement method without FFT approximation; two cycles only):

           <b>mafft</b> <b>--retree</b> <u>2</u> <b>--maxiterate</b> <u>2</u> <b>--nofft</b> <u>input</u> [&gt; <u>output</u>]

           <b>nwnsi</b> <u>input</u> [&gt; <u>output</u>]

       •   NW-NS-2 (fast; progressive method without the FFT approximation):

           <b>mafft</b> <b>--retree</b> <u>2</u> <b>--maxiterate</b> <u>0</u> <b>--nofft</b> <u>input</u> [&gt; <u>output</u>]

           <b>nwns</b> <u>input</u> [&gt; <u>output</u>]

       •   NW-NS-PartTree-1 (recommended for ~10,000 to ~50,000 sequences; progressive method with the PartTree
           algorithm):

           <b>mafft</b> <b>--retree</b> <u>1</u> <b>--maxiterate</b> <u>0</u> <b>--nofft</b> <b>--parttree</b> <u>input</u> [&gt; <u>output</u>]

   <b>Group-to-group</b> <b>alignments</b>

           <b>mafft-profile</b> <u>group1</u> <u>group2</u> [&gt; <u>output</u>]

           or:

           <b>mafft</b> <b>--maxiterate</b> <u>1000</u> <b>--seed</b> <u>group1</u> <b>--seed</b> <u>group2</u> /dev/null [&gt; <u>output</u>]

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Algorithm</b>
       <b>--auto</b>
           Automatically selects an appropriate strategy from L-INS-i, FFT-NS-i and FFT-NS-2, according to data
           size.  Default: off (always FFT-NS-2)

       <b>--6merpair</b>
           Distance is calculated based on the number of shared 6mers.  Default: on

       <b>--globalpair</b>
           All pairwise alignments are computed with the Needleman-Wunsch algorithm.  More accurate but slower
           than --6merpair.  Suitable for a set of globally alignable sequences.  Applicable to up to ~200
           sequences.  A combination with --maxiterate 1000 is recommended (G-INS-i).  Default: off (6mer
           distance is used)

       <b>--localpair</b>
           All pairwise alignments are computed with the Smith-Waterman algorithm.  More accurate but slower
           than --6merpair.  Suitable for a set of locally alignable sequences.  Applicable to up to ~200
           sequences.  A combination with --maxiterate 1000 is recommended (L-INS-i).  Default: off (6mer
           distance is used)

       <b>--genafpair</b>
           All pairwise alignments are computed with a local algorithm with the generalized affine gap cost
           (Altschul 1998).  More accurate but slower than --6merpair.  Suitable when large internal gaps are
           expected.  Applicable to up to ~200 sequences.  A combination with --maxiterate 1000 is recommended
           (E-INS-i).  Default: off (6mer distance is used)

       <b>--fastapair</b>
           All pairwise alignments are computed with FASTA (Pearson and Lipman 1988).  FASTA is required.
           Default: off (6mer distance is used)

       <b>--weighti</b> <u>number</u>
           Weighting factor for the consistency term calculated from pairwise alignments.  Valid when either of
           --globalpair, --localpair,  --genafpair, --fastapair or --blastpair is selected.  Default: 2.7

       <b>--retree</b> <u>number</u>
           Guide tree is built <u>number</u> times in the progressive stage.  Valid with 6mer distance.  Default: 2

       <b>--maxiterate</b> <u>number</u>
           <u>number</u> cycles of iterative refinement are performed.  Default: 0

       <b>--fft</b>
           Use FFT approximation in group-to-group alignment.  Default: on

       <b>--nofft</b>
           Do not use FFT approximation in group-to-group alignment.  Default: off

       <b>--noscore</b>
           Alignment score is not checked in the iterative refinement stage.  Default: off (score is checked)

       <b>--memsave</b>
           Use the Myers-Miller (1988) algorithm.  Default: automatically turned on when the alignment length
           exceeds 10,000 (aa/nt).

       <b>--parttree</b>
           Use a fast tree-building method (PartTree, Katoh and Toh 2007) with the 6mer distance.  Recommended
           for a large number (&gt; ~10,000) of sequences are input.  Default: off

       <b>--dpparttree</b>
           The PartTree algorithm is used with distances based on DP.  Slightly more accurate and slower than
           --parttree.  Recommended for a large number (&gt; ~10,000) of sequences are input.   Default: off

       <b>--fastaparttree</b>
           The PartTree algorithm is used with distances based on FASTA.  Slightly more accurate and slower than
           --parttree.  Recommended for a large number (&gt; ~10,000) of sequences are input.  FASTA is required.
           Default: off

       <b>--partsize</b> <u>number</u>
           The number of partitions in the PartTree algorithm.  Default: 50

       <b>--groupsize</b> <u>number</u>
           Do not make alignment larger than <u>number</u> sequences. Valid only with the --*parttree options.
           Default: the number of input sequences

   <b>Parameter</b>
       <b>--op</b> <u>number</u>
           Gap opening penalty at group-to-group alignment.  Default: 1.53

       <b>--ep</b> <u>number</u>
           Offset value, which works like gap extension penalty, for group-to-group alignment.  Default: 0.123

       <b>--lop</b> <u>number</u>
           Gap opening penalty at local pairwise alignment.  Valid when the --localpair or --genafpair option is
           selected.  Default: -2.00

       <b>--lep</b> <u>number</u>
           Offset value at local pairwise alignment.  Valid when the --localpair or --genafpair option is
           selected.  Default: 0.1

       <b>--lexp</b> <u>number</u>
           Gap extension penalty at local pairwise alignment.  Valid when the --localpair or --genafpair option
           is selected.  Default: -0.1

       <b>--LOP</b> <u>number</u>
           Gap opening penalty to skip the alignment.  Valid when the --genafpair option is selected.   Default:
           -6.00

       <b>--LEXP</b> <u>number</u>
           Gap extension penalty to skip the alignment.  Valid when the --genafpair option is selected.
           Default: 0.00

       <b>--bl</b> <u>number</u>
           BLOSUM <u>number</u> matrix (Henikoff and Henikoff 1992) is used.  <u>number</u>=30, 45, 62 or 80.  Default: 62

       <b>--jtt</b> <u>number</u>
           JTT PAM <u>number</u> (Jones et al. 1992) matrix is used.  <u>number</u>&gt;0.  Default: BLOSUM62

       <b>--tm</b> <u>number</u>
           Transmembrane PAM <u>number</u> (Jones et al. 1994) matrix is used.  <u>number</u>&gt;0.  Default: BLOSUM62

       <b>--aamatrix</b> <u>matrixfile</u>
           Use a user-defined AA scoring matrix.  The format of <u>matrixfile</u> is the same to that of BLAST.
           Ignored when nucleotide sequences are input.   Default: BLOSUM62

       <b>--fmodel</b>
           Incorporate the AA/nuc composition information into the scoring matrix.  Default: off

   <b>Output</b>
       <b>--clustalout</b>
           Output format: clustal format.  Default: off (fasta format)

       <b>--inputorder</b>
           Output order: same as input.  Default: on

       <b>--reorder</b>
           Output order: aligned.  Default: off (inputorder)

       <b>--treeout</b>
           Guide tree is output to the <u>input</u>.tree file.  Default: off

       <b>--quiet</b>
           Do not report progress.  Default: off

   <b>Input</b>
       <b>--nuc</b>
           Assume the sequences are nucleotide.  Default: auto

       <b>--amino</b>
           Assume the sequences are amino acid.  Default: auto

       <b>--seed</b> <u>alignment1</u> [<b>--seed</b> <u>alignment2</u> <b>--seed</b> <u>alignment3</u> ...]
           Seed alignments given in <u>alignment_n</u> (fasta format) are aligned with sequences in <u>input</u>.  The
           alignment within every seed is preserved.

</pre><h4><b>FILES</b></h4><pre>
       Mafft stores the input sequences and other files in a temporary directory, which by default is located in
       <u><a href="file:/tmp">/tmp</a></u>.

</pre><h4><b>ENVIONMENT</b></h4><pre>
       <b>MAFFT_BINARIES</b>
           Indicates the location of the binary files used by mafft. By default, they are searched in
           <u>/usr/local/lib/mafft</u>, but on Debian systems, they are searched in <u>/usr/lib/mafft</u>.

       <b>FASTA_4_MAFFT</b>
           This variable can be set to indicate to mafft the location to the fasta34 program if it is not in the
           PATH.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/mafft-homologs.1.html">mafft-homologs</a></b>(1)

</pre><h4><b>REFERENCES</b></h4><pre>
   <b>In</b> <b>English</b>
       •   Katoh and Toh (Bioinformatics 23:372-374, 2007) PartTree: an algorithm to build an approximate tree
           from a large number of unaligned sequences (describes the PartTree algorithm).

       •   Katoh, Kuma, Toh and Miyata (Nucleic Acids Res. 33:511-518, 2005) MAFFT version 5: improvement in
           accuracy of multiple sequence alignment (describes [ancestral versions of] the G-INS-i, L-INS-i and
           E-INS-i strategies)

       •   Katoh, Misawa, Kuma and Miyata (Nucleic Acids Res. 30:3059-3066, 2002) MAFFT: a novel method for
           rapid multiple sequence alignment based on fast Fourier transform (describes the FFT-NS-1, FFT-NS-2
           and FFT-NS-i strategies)

   <b>In</b> <b>Japanese</b>
       •   Katoh and Misawa (Seibutsubutsuri 46:312-317, 2006) Multiple Sequence Alignments: the Next Generation

       •   Katoh and Kuma (Kagaku to Seibutsu 44:102-108, 2006) Jissen-teki Multiple Alignment

</pre><h4><b>AUTHORS</b></h4><pre>
       <b>Kazutaka</b> <b>Katoh</b> &lt;kazutaka.katoh_at_aist.go.jp&gt;
           Wrote Mafft.

       <b>Charles</b> <b>Plessy</b> &lt;charles-debian-nospam_at_plessy.org&gt;
           Wrote this manpage in DocBook XML for the Debian distribution, using Mafft's homepage as a template.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright © 2002-2007 Kazutaka Katoh (mafft)
       Copyright © 2007 Charles Plessy (this manpage)

       Mafft and its manpage are offered under the following conditions:

       Redistribution  and  use in source and binary forms, with or without modification, are permitted provided
       that the following conditions are met:

        1.  Redistributions of source code must retain the above copyright notice, this list of  conditions  and
           the following disclaimer.

         2.  Redistributions  in  binary form must reproduce the above copyright notice, this list of conditions
           and the  following  disclaimer  in  the  documentation  and/or  other  materials  provided  with  the
           distribution.

         3.  The  name  of  the author may not be used to endorse or promote products derived from this software
           without specific prior written permission.

       THIS SOFTWARE IS PROVIDED BY THE AUTHOR "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
       LIMITED TO, THE  IMPLIED  WARRANTIES  OF  MERCHANTABILITY  AND  FITNESS  FOR  A  PARTICULAR  PURPOSE  ARE
       DISCLAIMED.  IN  NO  EVENT  SHALL  THE  AUTHOR  BE  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
       EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF  SUBSTITUTE  GOODS  OR
       SERVICES;  LOSS  OF  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
       LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING  IN
       ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

mafft 6.240                                        2007-06-09                                           <u><a href="../man1/MAFFT.1.html">MAFFT</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>