<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>sambamba-view - tool for extracting information from SAM/BAM files</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/sambamba">sambamba_1.0.1+dfsg-2build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       <b>sambamba-view</b> - tool for extracting information from SAM/BAM files

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>sambamba</b> <b>view</b> <u>OPTIONS</u> &lt;input.bam | input.sam&gt; [region1 [...]]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>sambamba</b> <b>view</b> allows to efficiently filter SAM/BAM files for alignments satisfying various conditions, as
       well  as  access  its  SAM  header and information about reference sequences. In order to make these data
       readily available for consumption by scripts in Perl/Python/Ruby, JSON output is provided.

       By default, the tool expects BAM file as an input. In order to work with SAM, specify <b>-S</b>|<b>--sam-input</b> as a
       command-line option, the tool does NOT try to guess file format from  the  extension.  Beware  that  when
       reading  SAM,  the  tool will skip tags which don´t conform to the SAM/BAM specification, and set invalid
       fields to their default values.

</pre><h4><b>FILTERING</b></h4><pre>
       Filtering is presented in two ways. First, you can specify a condition with <b>-F</b> option,  using  a  special
       language for filtering, described at

       https://github.com/lomereiter/sambamba/wiki/%5Bsambamba-view%5D-Filter-expression-syntax

       Second, if you have an indexed BAM file, several regions can be specified as well. The syntax for regions
       is  the  same  as  in  samtools:  <u>chr</u>:<u>beg</u>-<u>end</u> where <u>beg</u> and <u>end</u> are 1-based start and end of a closed-end
       interval on the reference <u>chr</u>.

</pre><h4><b>JSON</b></h4><pre>
       Alignment record JSON representation is a  hash  with  keys  ´qname´,  ´flag´,  ´rname´,  ´pos´,  ´mapq´,
       ´cigar´, ´rnext´, ´qual´, ´tags´, e.g.

       {"qname":"EAS56_57:6:190:289:82","flag":69,"rname":"chr1","pos":100,
       "mapq":0,"cigar":"*","rnext":"=","pnext":100,"tlen":0,
       "seq":"CTCAAGGTTGTTGCAAGGGGGTCTATGTGAACAAA",
       "qual":[27,27,27,22,27,27,27,26,27,27,27,27,27,27,27,27,23,26,26,27,
       22,26,19,27,26,27,26,26,26,26,26,24,19,27,26],"tags":{"MF":192}}

       JSON representation mimics SAM format except quality is given as an array of integers.

       Postprocessing JSON output is best accomplished with https://stedolan.github.io/jq/

       The output is one line per read, for building a proper JSON array pipe the output into <b>jq</b> <b>--slurp</b>.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-F</b>, <b>--filter</b>=<u>FILTER</u>
              Set custom filter for alignments.

       <b>-f</b>, <b>--format</b>=<u>FORMAT</u>
              Specify output format. <u>FORMAT</u> must be one of sam, bam, or json (in lowercase). Default is SAM.

       <b>-h</b>, <b>--with-header</b>
              Print SAM header before reads. This is always done for BAM output.

       <b>-H</b>, <b>--header</b>
              Print  only  SAM header to STDOUT. If <u>FORMAT</u> is sam or bam, its text version is printed, otherwise
              JSON object is written.

       <b>-I</b>, <b>--reference-info</b>
              Output to STDOUT reference sequence names and lengths in JSON (see <u>EXAMPLES</u>).

       <b>-L</b>, <b>--regions</b>=<u>BEDFILE</u>
              Intersect a file with regions specified in the BED file.

       <b>-c</b>, <b>--count</b>
              Output to STDOUT only the number of matching records, -hHI options are ignored.

       <b>-v</b>, <b>--valid</b>
              Output only valid reads.

       <b>-S</b>, <b>--sam-input</b>
              Specify that the input is SAM file (default is BAM for all operations).

       <b>-p</b>, <b>--show-progress</b>
              Show progressbar in STDERR. Works only for BAM files, and with no  regions  specified,  i.e.  only
              when reading full file.

       <b>-l</b>, <b>--compression-level</b>=<u>COMPRESSION_LEVEL</u>
              Set compression level for BAM output, a number from 0 to 9.

       <b>-o</b>, <b>--output-filename</b>=<u>FILENAME</u>
              Specify output filename (by default everything is written to STDOUT).

       <b>-t</b>, <b>--nthreads</b>=<u>NTHREADS</u>
              Number of threads to use.

</pre><h4><b>EXAMPLES</b></h4><pre>
       Print basic reference sequence information:

            $ sambamba view --reference-info ex1_header.bam
            [{"name":"chr1","length":1575},{"name":"chr2","length":1584}]

       Count reads with mapping quality not less than 50:

            $ sambamba view -c -F "mapping_quality &gt;= 50" ex1_header.bam
            3124

       Count properly paired reads overlapping 100..200 on chr1:

            $ sambamba view -c -F "proper_pair" ex1_header.bam chr1:100-200
            39

       Output header in JSON format:

            $ sambamba view --header --format=json ex1_header.bam
            {"format_version":"1.3","rg_lines":[],
             "sq_lines":[{"sequence_length":1575,"species":"","uri":"",
             "sequence_name":"chr1","assembly":"","md5":""},
             {"sequence_length":1584,"species":"","uri":"",
             "sequence_name":"chr2","assembly":"","md5":""}],
             "sorting_order":"coordinate","pg_lines":[]}

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       For  more  information  on  the  original  samtools  VIEW behaviour, check out the samtools documentation
       <u><a href="http://samtools.sourceforge.net/samtools.shtml">http://samtools.sourceforge.net/samtools.shtml</a></u>.

                                                    June 2016                                   <u><a href="../man1/SAMBAMBA-VIEW.1.html">SAMBAMBA-VIEW</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>