<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>art - Artemis genome browser and annotation tool</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/artemis">artemis_18.2.0+dfsg-4_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       art - Artemis genome browser and annotation tool

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>art</b> [<u>options</u>] <u>SEQUENCE_FILE</u> [<u>+FEATURE_FILE</u>...]

</pre><h4><b>OPTIONS</b></h4><pre>
               SEQUENCE_FILE                  An EMBL, GenBank, FASTA, or GFF3 file
               FEATURE_FILE                   An Artemis TAB file, or GFF file

               -options FILE                  Read a text file of options from FILE
               -debug                         Run using the debugging JVM instead
               -fast | -fast64                Use the FastVM (hp Tru64 UNIX) with 32/64 bit pointers

               -Dblack_belt_mode=?            Keep warning messages to a minimum [true,false]
               -Doffset=XXX                   Open viewer at base position XXX [integer &gt;= 1]
               -Duserplot=FILE[,FILE2]        Open one or more userplots
               -Dloguserplot=FILE[,FILE2]     Open one or more userplots, take log(data)
               -Dbam=FILE[,FILE2,...]         Open one or more BAM, VCF or BCF files
               -DbamClone=n                   Open all BAMs in multiple (n &gt; 1) panels
               -Dbam[1,2,..]=FILE[,FILE2,..]  Open BAMs in separate panels
               -Dshow_snps                    Show SNP marks in BamView
               -Dshow_snp_plot                Open SNP plot in BamView
               -Dshow_cov_plot                Open coverage plot in BamView
               -Dshow_forward_lines=?         Hide/show forward frame lines [true,false]
               -Dshow_reverse_lines=?         Hide/show reverse frame lines [true,false]
               -Dchado="h:p/d?u"              Get Artemis to open this CHADO database
               -Dread_only                    Open CHADO database read-only

</pre><h4><b>EXAMPLES</b></h4><pre>
               % art AJ006275.embl
               % art contigs.fa +annotation.gff +islands.tab
               % art -Dblack_belt_mode=true -Dbam=ecoli_hiseq.bam E_coli_K12.gbk
               % art -Duserplot=repeatmap.plot,geecee.plot Plasmid.gff3

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
               <a href="http://www.sanger.ac.uk/resources/software/artemis/">http://www.sanger.ac.uk/resources/software/artemis/</a>

       <b><a href="../man1/act.1.html">act</a></b>(1), <b><a href="../man1/dnaplotter.1.html">dnaplotter</a></b>(1)

                                                  February 2014                                           <u><a href="../man1/ART.1.html">ART</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>