<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>rsem-plot-transcript-wiggles - Generate PDF wiggle plots from transcript or gene ids</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/rsem">rsem_1.3.3+dfsg-3build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       rsem-plot-transcript-wiggles - Generate PDF wiggle plots from transcript or gene ids

</pre><h4><b>SYNOPSIS</b></h4><pre>
        rsem-plot-transcript-wiggles [options] sample_name input_list output_plot_file

</pre><h4><b>ARGUMENTS</b></h4><pre>
       <b>sample_name</b>
           The name of the sample analyzed.

       <b>input_list</b>
           A  list  of  transcript ids or gene ids. But it cannot be a mixture of transcript &amp; gene ids. Each id
           occupies one line without extra spaces.

       <b>output_plot_file</b>
           The file name of the pdf file which contains all plots.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>--gene-list</b>
           The input-list is a list of gene ids. (Default: off)

       <b>--transcript-list</b>
           The input-list is a list of transcript ids. This option can only  be  turned  on  if  allele-specific
           expression is calculated. (Default: off)

       <b>--show-unique</b>
           Show the wiggle plots as stacked bar plots. See description section for details. (Default: off)

       <b>-p/--num-threads</b> &lt;int&gt;
           Set the number of threads we can use. (Default: 1)

       <b>--memory-per-thread</b> &lt;string&gt;
           Set  the  maximum  allowable  memory  per thread. &lt;string&gt; represents the memory and accepts suffices
           'K/M/G'. (Default: 1G)

       <b>-h/--help</b>
           Show help information.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This program generates transcript wiggle plots and outputs them in a pdf file. This  program  can  accept
       either  a  list of transcript ids or gene ids (if transcript to gene mapping information is provided) and
       has two modes of showing wiggle plots. If '--show-unique' is not specified,  the  wiggle  plot  for  each
       transcript is a histogram where each position has the expected read depth at this position as its height.
       If  '--show-unique' is specified, for each transcript a stacked bar plot is generated. For each position,
       the read depth of unique reads, which have only one alignment, is showed in  black.  The  read  depth  of
       multi-reads,  which  align  to  more than one places, is showed in red on top of the read depth of unique
       reads.This program will use some files RSEM generated previouslly. So please do not delete/move any  file
       'rsem-calculate-expression'  generated.  If  allele-specific expression is calculated, the basic unit for
       plotting  is  an  allele-specific  transcript  and  plots  can  be  grouped  by  either  transcript   ids
       (--transcript-list) or gene ids (--gene-list).

</pre><h4><b>OUTPUT</b></h4><pre>
       <b>output_plot_file</b>
           This is a pdf file containing all plots generated. If a list of transcript ids is provided, each page
           display  at most 6 plots in 3 rows and 2 columns. If gene ids are provided, each page display a gene.
           The gene's id is showed at the top and all its transcripts' wiggle plots are showed in this page. The
           arrangement of plots is determined automatically. For each transcript wiggle plot, the transcript  id
           is  displayed  as  title.  x-axis  is position in the transcript and y-axis is read depth. If allele-
           specific expression  is  calculated,  the  basin  unit  becomes  an  allele-specific  transcript  and
           transcript ids and gene ids can be used to group allele-specific transcripts.

       <b>sample_name.transcript.sorted.bam</b> <b>and</b> <b>sample_name.transcript.readdepth</b>
           If these files do not exist, 'rsem-plot-transcript-wiggles' will automatically generate them.

       <b>sample_name.uniq.transcript.bam,</b> <b>sample_name.uniq.transcript.sorted.bam</b> <b>and</b>
       <b>sample_name.uniq.transcript.readdepth</b>
           If  '--show-unique'  option is specified and these files do not exist, 'rsem-plot-transcript-wiggles'
           will automatically generate them.

</pre><h4><b>EXAMPLES</b></h4><pre>
       Suppose sample_name and output_plot_file are set to 'mmliver_single_quals' and 'output.pdf' respectively.
       input_list is set to 'transcript_ids.txt' if transcript ids are provided, and is set to 'gene_ids.txt' if
       gene ids are provided.

       1) Transcript ids are provided and we just want normal wiggle plots:

        rsem-plot-transcript-wiggles mmliver_single_quals transcript_ids.txt output.pdf

       2) Gene ids are provided and we want to show stacked bar plots:

        rsem-plot-transcript-wiggles --gene-list --show-unique mmliver_single_quals gene_ids.txt output.pdf

perl v5.38.2                                       2024-04-14                    <u><a href="../man1/RSEM-PLOT-TRANSCRIPT-WIGGLES.1.html">RSEM-PLOT-TRANSCRIPT-WIGGLES</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>