<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>salmon_index - highly-accurate, transcript-level quantification estimates from RNA-seq data</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/salmon">salmon_1.10.2+ds1-1build2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       salmon_index - highly-accurate, transcript-level quantification estimates from RNA-seq data

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Index ========== Creates a salmon index.

   <b>Command</b> <b>Line</b> <b>Options:</b>
       <b>-v</b> [ <b>--version</b> ]
              print version string

       <b>-h</b> [ <b>--help</b> ]
              produce help message

       <b>-t</b> [ <b>--transcripts</b> ] arg
              Transcript fasta file.

       <b>-k</b> [ <b>--kmerLen</b> ] arg (=31)
              The size of k-mers that should be used for the quasi index.

       <b>-i</b> [ <b>--index</b> ] arg
              salmon index.

       <b>--gencode</b>
              This  flag  will  expect  the  input  transcript fasta to be in GENCODE format, and will split the
              transcript name at the first '|' character.  These reduced names will be used in  the  output  and
              when looking for these transcripts in a gene to transcript GTF.

       <b>--features</b>
              This flag will expect the input reference to be in the tsv file format, and will split the feature
              name  at  the  first  'tab'  character.   These  reduced names will be used in the output and when
              looking for the sequence of the features.GTF.

       <b>--keepDuplicates</b>
              This flag will disable the default indexing behavior of  discarding  sequence-identical  duplicate
              transcripts.   If this flag is passed, then duplicate transcripts that appear in the input will be
              retained and quantified separately.

       <b>-p</b> [ <b>--threads</b> ] arg (=2)
              Number of threads to use during indexing.

       <b>--keepFixedFasta</b>
              Retain the fixed fasta file (without short transcripts and duplicates, clipped,  etc.)   generated
              during indexing

       <b>-f</b> [ <b>--filterSize</b> ] arg (=-1) The size of the Bloom filter that will be used
              by  TwoPaCo  during  indexing.  The filter will be of size 2^{filterSize}. The default value of <b>-1</b>
              means that the filter size will be automatically set based on the number of distinct k-mers in the
              input, as estimated by nthll.

       <b>--tmpdir</b> arg
              The directory location that will be used for TwoPaCo temporary files; it will be created  if  need
              be  and  be  removed  prior  to  indexing  completion.  The default value will cause a (temporary)
              subdirectory of the salmon index directory to be used for this purpose.

       <b>--sparse</b>
              Build the index using a  sparse  sampling  of  k-mer  positions  This  will  require  less  memory
              (especially  during quantification), but will take longer to construct and can slow down mapping /
              alignment

       <b>-d</b> [ <b>--decoys</b> ] arg
              Treat these sequences ids from the reference as the decoys that may have  sequence  homologous  to
              some  known  transcript.  for example in case of the genome, provide a list of chromosome name <b>---</b>
              one per line

       <b>-n</b> [ <b>--no-clip</b> ]
              Don't clip poly-A tails from the ends of target sequences

       <b>--type</b> arg (=puff)
              The type of index to build; the only option is "puff" in this version of salmon.

salmon --no-version-check index 1.10.2+ds1         April 2024                                    <u><a href="../man1/SALMON_INDEX.1.html">SALMON_INDEX</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>