<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Genemark - Results of one Genemark run</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Genemark - Results of one Genemark run

</pre><h4><b>SYNOPSIS</b></h4><pre>
          $Genemark = Bio::Tools::Genemark-&gt;new(-file =&gt; 'result.Genemark');
          # filehandle:
          $Genemark = Bio::Tools::Genemark-&gt;new( -fh  =&gt; \*INPUT );

          # parse the results
          # note: this class is-a Bio::Tools::AnalysisResult which implements
          # Bio::SeqAnalysisParserI, i.e., $Genemark-&gt;next_feature() is the same
          while($gene = $Genemark-&gt;next_prediction()) {
              # $gene is an instance of Bio::Tools::Prediction::Gene, which inherits
              # off Bio::SeqFeature::Gene::Transcript.
              #
              # $gene-&gt;exons() returns an array of
              # Bio::Tools::Prediction::Exon objects
              # all exons:
              @exon_arr = $gene-&gt;exons();

              # initial exons only
              @init_exons = $gene-&gt;exons('Initial');
              # internal exons only
              @intrl_exons = $gene-&gt;exons('Internal');
              # terminal exons only
              @term_exons = $gene-&gt;exons('Terminal');
              # singleton exons:
              ($single_exon) = $gene-&gt;exons();
          }

          # essential if you gave a filename at initialization (otherwise the file
          # will stay open)
          $Genemark-&gt;close();

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The Genemark module provides a parser for Genemark gene structure prediction output. It parses one gene
       prediction into a Bio::SeqFeature::Gene::Transcript- derived object.

       This module has been developed around genemark.hmm for eukaryots v2.2a and will probably not work with
       other versions.

       This module also implements the Bio::SeqAnalysisParserI interface, and thus can be used wherever such an
       object fits. See Bio::SeqAnalysisParserI.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Hilmar</b> <b>Lapp,</b> <b>Mark</b> <b>Fiers</b></h4><pre>
       Email <a href="mailto:hlapp@gmx.net">hlapp@gmx.net</a>
             <a href="mailto:m.w.e.j.fiers@plant.wag-ur.nl">m.w.e.j.fiers@plant.wag-ur.nl</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : my $obj = Bio::Tools::Genemark-&gt;new();
        Function: Builds a new Bio::Tools::Genemark object
        Returns : an instance of Bio::Tools::Genemark
        Args    : seqname

   <b>analysis_method</b>
        Usage     : $Genemark-&gt;analysis_method();
        Purpose   : Inherited method. Overridden to ensure that the name matches
                    /GeneMark.hmm/i.
        Returns   : String
        Argument  : n/a

   <b>next_feature</b>
        Title   : next_feature
        Usage   : while($gene = $Genemark-&gt;next_feature()) {
                         # do something
                  }
        Function: Returns the next gene structure prediction of the Genemark result
                  file. Call this method repeatedly until FALSE is returned.

                  The returned object is actually a SeqFeatureI implementing object.
                  This method is required for classes implementing the
                  SeqAnalysisParserI interface, and is merely an alias for
                  next_prediction() at present.

        Example :
        Returns : A Bio::Tools::Prediction::Gene object.
        Args    :

   <b>next_prediction</b>
        Title   : next_prediction
        Usage   : while($gene = $Genemark-&gt;next_prediction()) {
                         # do something
                  }
        Function: Returns the next gene structure prediction of the Genemark result
                  file. Call this method repeatedly until FALSE is returned.

        Example :
        Returns : A Bio::Tools::Prediction::Gene object.
        Args    :

   <b>_parse_predictions</b>
        Title   : _parse_predictions()
        Usage   : $obj-&gt;_parse_predictions()
        Function: Parses the prediction section. Automatically called by
                  next_prediction() if not yet done.
        Example :
        Returns :

   <b>_prediction</b>
        Title   : _prediction()
        Usage   : $gene = $obj-&gt;_prediction()
        Function: internal
        Example :
        Returns :

   <b>_add_prediction</b>
        Title   : _add_prediction()
        Usage   : $obj-&gt;_add_prediction($gene)
        Function: internal
        Example :
        Returns :

   <b>_predictions_parsed</b>
        Title   : _predictions_parsed
        Usage   : $obj-&gt;_predictions_parsed
        Function: internal
        Example :
        Returns : TRUE or FALSE

   <b>_has_cds</b>
        Title   : _has_cds()
        Usage   : $obj-&gt;_has_cds()
        Function: Whether or not the result contains the predicted CDSs, too.
        Example :
        Returns : TRUE or FALSE

   <b>_read_fasta_seq</b>
        Title   : _read_fasta_seq()
        Usage   : ($id,$seqstr) = $obj-&gt;_read_fasta_seq();
        Function: Simple but specialised FASTA format sequence reader. Uses
                  $self-&gt;_readline() to retrieve input, and is able to strip off
                  the traling description lines.
        Example :
        Returns : An array of two elements.

   <b>_seqname</b>
        Title   : _seqname
        Usage   : $obj-&gt;_seqname($seqname)
        Function: internal
        Example :
        Returns : String

perl v5.32.1                                       2021-08-15                          <u>Bio::Tools::<a href="../man3pm/Genemark.3pm.html">Genemark</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>