<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::SeqFeature::Gene::Intron - An intron feature</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::SeqFeature::Gene::Intron - An intron feature

</pre><h4><b>SYNOPSIS</b></h4><pre>
       Give standard usage here

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Describe the object here

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>David</b> <b>Block</b></h4><pre>
       Email <a href="mailto:dblock@gene.pbi.nrc.ca">dblock@gene.pbi.nrc.ca</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>upstream_Exon</b>
         Title   : upstream_Exon
         Usage   : $intron-&gt;upstream_Exon()
         Function: exon upstream of the intron
         Returns : Bio::EnsEMBL::Exon
         Args    :

   <b>downstream_Exon</b>
         Title   : downstream_Exon
         Usage   : $intron-&gt;downstream_Exon()
         Function: exon downstream of the intron
         Returns : Bio::EnsEMBL::Exon
         Args    :

   <b>phase</b>
         Title   : phase
         Usage   : $intron-&gt;phase()
         Function: returns the phase of the intron(where it interrupts the codon)
         Returns : int(0,1,2)
         Args    :

   <b>acceptor_splice_site</b>
         Title   : acceptor_splice_site
         Usage   : $intron-&gt;acceptor_splice_site(21,3)
         Function: returns the sequence corresponding to the
                   consensus acceptor splice site. If start and
                   end are provided, it will number of base pairs
                   left and right of the canonical AG. Here 21 means
                   21 bp into intron and 3 means 3 bp into the exon.
                   --Intron--21----|AG|-3-----Exon
                   Defaults to 21,3

         Returns : Bio::Seq
         Args    : start and end

   <b>donor_splice_site</b>
         Title   : donor_splice_site
         Usage   : $intron-&gt;donor_splice_site(3,6)
         Function: returns the sequence corresponding to the
                   consensus donor splice site. If start and
                   end are provided, it will number of base pairs
                   left and right of the canonical GT. Here 3 means
                   3 bp into exon and 6 means 6 bp into the intron.
                   --Exon-3--|GT|-6----Intron-
                   Defaults to 3,6

         Returns : Bio::Seq
         Args    : start and end

perl v5.32.1                                       2021-08-15                 <u>Bio::SeqFeature::Gene::<a href="../man3pm/Intron.3pm.html">Intron</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>