<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::SeqFeature::Gene::Exon - a feature representing an exon</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::SeqFeature::Gene::Exon - a feature representing an exon

</pre><h4><b>SYNOPSIS</b></h4><pre>
           # obtain an exon instance $exon somehow
           print "exon from ", $exon-&gt;start(), " to ", $exon-&gt;end(),
                 " on seq ", $exon-&gt;seq_id(), ", strand ", $exon-&gt;strand(),
                 ", encodes the peptide sequence ",
                 $exon-&gt;cds()-&gt;translate()-&gt;seq(), "\n";

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This module implements a feature representing an exon by implementing the Bio::SeqFeature::Gene::ExonI
       interface. By default an Exon is coding. Supply -is_coding =&gt; 0 to the constructor or call
       $exon-&gt;<b><a href="../man0/is_coding.0.html">is_coding</a></b>(0) otherwise.

       Apart from that, this class also implements Bio::SeqFeatureI by inheriting off Bio::SeqFeature::Generic.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably
        to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Hilmar</b> <b>Lapp</b></h4><pre>
       Email <a href="mailto:hlapp@gmx.net">hlapp@gmx.net</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>is_coding</b>
        Title   : is_coding
        Usage   : if($exon-&gt;is_coding()) {
                          # do something
                  }
                  if($is_utr) {
                      $exon-&gt;<a href="../man0/is_coding.0.html">is_coding</a>(0);
                  }
        Function: Get/set whether or not the exon codes for amino acid.
        Returns : TRUE if the object represents a feature translated into protein,
                  and FALSE otherwise.
        Args    : A boolean value on set.

   <b>primary_tag</b>
        Title   : primary_tag
        Usage   : $tag = $feat-&gt;primary_tag()
                  $feat-&gt;primary_tag('exon')
        Function: Get/set the primary tag for the exon feature.

                  This method is overridden here in order to allow only for
                  tag values following a certain convention. For consistency reasons,
                  the tag value must either contain the string 'exon' or the string
                  'utr' (both case-insensitive). In the case of 'exon', a string
                  describing the type of exon may be appended or prefixed. Presently,
                  the following types are allowed: initial, internal, and terminal
                  (all case-insensitive).

                  If the supplied tag value matches 'utr' (case-insensitive),
                  is_coding() will automatically be set to FALSE, and to TRUE
                  otherwise.

        Returns : A string.
        Args    : A string on set.

   <b>location</b>
        Title   : location
        Usage   : my $location = $exon-&gt;location()
        Function: Returns a location object suitable for identifying the location
                  of the exon on the sequence or parent feature.

                  This method is overridden here to restrict allowed location types
                  to non-compound locations.

        Returns : Bio::LocationI object
        Args    : none

   <b>cds</b>
        Title   : cds()
        Usage   : $cds = $exon-&gt;cds();
        Function: Get the coding sequence of the exon as a sequence object.

                  The sequence of the returned object is prefixed by Ns (lower case)
                  if the frame of the exon is defined and different from zero. The
                  result is that the first base starts a codon (frame 0).

                  This implementation returns undef if the particular exon is
                  not translated to protein, i.e., is_coding() returns FALSE. Undef
                  will also be returned if no sequence is attached to this exon
                  feature.

        Returns : A Bio::PrimarySeqI implementing object.
        Args    :

perl v5.32.1                                       2021-08-15                   <u>Bio::SeqFeature::Gene::<a href="../man3pm/Exon.3pm.html">Exon</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>