<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Sim4::Exon - A single exon determined by an alignment</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Sim4::Exon - A single exon determined by an alignment

</pre><h4><b>SYNOPSIS</b></h4><pre>
         # See Bio::Tools::Sim4::Results for a description of the context.

         # an instance of this class is-a Bio::SeqFeature::SimilarityPair

         # coordinates of the exon (recommended way):
         print "exon from ", $exon-&gt;start(),
               " to ", $exon-&gt;end(), "\n";

         # the same (feature1() inherited from Bio::SeqFeature::FeaturePair)
         print "exon from ", $exon-&gt;feature1()-&gt;start(),
               " to ", $exon-&gt;feature1()-&gt;end(), "\n";
         # also the same (query() inherited from Bio::SeqFeature::SimilarityPair):
         print "exon from ", $exon-&gt;query()-&gt;start(),
               " to ", $exon-&gt;query()-&gt;end(), "\n";

         # coordinates on the matching EST (recommended way):
         print "matches on EST from ", $exon-&gt;est_hit()-&gt;start(),
               " to ", $exon-&gt;est_hit()-&gt;end(), "\n";

         # the same (feature2() inherited from Bio::SeqFeature::FeaturePair)
         print "matches on EST from ", $exon-&gt;feature2()-&gt;start(),
               " to ", $exon-&gt;feature2()-&gt;end(), "\n";
         # also the same (subject() inherited from Bio::SeqFeature::SimilarityPair):
         print "exon from ", $exon-&gt;subject()-&gt;start(),
               " to ", $exon-&gt;subject()-&gt;end(), "\n";

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This class inherits from Bio::SeqFeature::SimilarityPair and represents an exon on a genomic sequence
       determined by similarity, that is, by aligning an EST sequence (using Sim4 in this case). Consequently,
       the notion of query and subject is always from the perspective of the genomic sequence: query refers to
       the genomic seq, subject to the aligned EST hit. Because of this, $exon-&gt;<b>start()</b>, $exon-&gt;<b>end()</b> etc will
       always return what you expect.

       To get the coordinates on the matching EST, refer to the properties of the feature returned by est_hit().

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably
        to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Ewan</b> <b>Birney,</b> <b>Hilmar</b> <b>Lapp</b></h4><pre>
       Ewan Birney &lt;birney-at-sanger.ac.uk&gt; Hilmar Lapp &lt;hlapp-at-gmx.net&gt; or
       &lt;hilmar.lapp-at-pharma.novartis.com&gt;.

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>percentage_id</b>
        Title   : percentage_id
        Usage   : $obj-&gt;percentage_id($newval)
        Function: This is a synonym for 100 * $obj-&gt;est_hit()-&gt;frac_identical().
        Returns : value of percentage_id
        Args    : newvalue (optional)

   <b>est_hit</b>
        Title   : est_hit
        Usage   : $est_feature = $obj-&gt;est_hit();
        Function: Returns the EST hit pointing to (i.e., aligned to by Sim4) this
                  exon (i.e., genomic region). At present, merely a synonym for
                  $obj-&gt;feature2().
        Returns : An Bio::SeqFeatureI implementing object.
        Args    :

perl v5.32.1                                       2021-08-15                        <u>Bio::Tools::Sim4::<a href="../man3pm/Exon.3pm.html">Exon</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>