<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::SeqFeature::Gene::ExonI - Interface for a feature representing an exon</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::SeqFeature::Gene::ExonI - Interface for a feature representing an exon

</pre><h4><b>SYNOPSIS</b></h4><pre>
       See documentation of methods.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       A feature representing an exon. An exon in this definition is transcribed and at least for one particular
       transcript not spliced out of the pre-mRNA. However, it does not necessarily code for amino acid.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably
        to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Hilmar</b> <b>Lapp</b></h4><pre>
       Email <a href="mailto:hlapp@gmx.net">hlapp@gmx.net</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>is_coding</b>
        Title   : is_coding
        Usage   : if($exon-&gt;is_coding()) {
                          # do something
                  }
        Function: Whether or not the exon codes for amino acid.
        Returns : TRUE if the object represents a feature translated into protein,
                  and FALSE otherwise.
        Args    :

   <b>cds</b>
        Title   : cds()
        Usage   : $cds = $exon-&gt;cds();
        Function: Get the coding sequence of the exon as a sequence object.

                  The returned sequence object must be in frame 0, i.e., the first
                  base starts a codon.

                  An implementation may return undef, indicating that a coding
                  sequence does not exist, e.g. for a UTR (untranslated region).

        Returns : A L&lt;Bio::PrimarySeqI&gt; implementing object.
        Args    :

perl v5.32.1                                       2021-08-15                  <u>Bio::SeqFeature::Gene::<a href="../man3pm/ExonI.3pm.html">ExonI</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>