<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>eprimer32 - Picks PCR primers and hybridization oligos</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/emboss">emboss_6.6.0+dfsg-15ubuntu2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       eprimer32 - Picks PCR primers and hybridization oligos

</pre><h4><b>SYNOPSIS</b></h4><pre>

       <b>eprimer32</b> <b>-sequence</b> <u>seqall</u> [<b>-primer</b> <u>toggle</u>] <b>-task</b> <u>list</u> [<b>-hybridprobe</b> <u>toggle</u>] <b>-mishyblibraryfile</b> <u>infile</u>
                 <b>-mispriminglibraryfile</b> <u>infile</u> [<b>-numreturn</b> <u>integer</u>] [<b>-includedregion</b> <u>range</u>]
                 [<b>-targetregion</b> <u>range</u>] [<b>-excludedregion</b> <u>range</u>] [<b>-forwardinput</b> <u>string</u>] [<b>-reverseinput</b> <u>string</u>]
                 <b>-gcclamp</b> <u>integer</u> <b>-osize</b> <u>integer</u> <b>-minsize</b> <u>integer</u> <b>-maxsize</b> <u>integer</u> <b>-otm</b> <u>float</u> <b>-mintm</b> <u>float</u>
                 <b>-maxtm</b> <u>float</u> <b>-maxdifftm</b> <u>float</u> <b>-ogcpercent</b> <u>float</u> <b>-mingc</b> <u>float</u> <b>-maxgc</b> <u>float</u> <b>-saltconc</b> <u>float</u>
                 <b>-dnaconc</b> <u>float</u> <b>-maxpolyx</b> <u>integer</u> <b>-psizeopt</b> <u>integer</u> <b>-prange</b> <u>range</u> <b>-ptmopt</b> <u>float</u> <b>-ptmmin</b> <u>float</u>
                 <b>-ptmmax</b> <u>float</u> <b>-oexcludedregion</b> <u>range</u> <b>-oligoinput</b> <u>string</u> <b>-osizeopt</b> <u>integer</u> <b>-ominsize</b> <u>integer</u>
                 <b>-omaxsize</b> <u>integer</u> <b>-otmopt</b> <u>float</u> <b>-otmmin</b> <u>float</u> <b>-otmmax</b> <u>float</u> <b>-ogcopt</b> <u>float</u> <b>-ogcmin</b> <u>float</u>
                 <b>-ogcmax</b> <u>float</u> <b>-osaltconc</b> <u>float</u> <b>-odnaconc</b> <u>float</u> <b>-oanyself</b> <u>float</u> <b>-oendself</b> <u>float</u>
                 <b>-opolyxmax</b> <u>integer</u> <b>-omishybmax</b> <u>float</u> <b>-explainflag</b> <u>boolean</u> <b>-fileflag</b> <u>boolean</u>
                 <b>-firstbaseindex</b> <u>integer</u> <b>-pickanyway</b> <u>boolean</u> <b>-maxmispriming</b> <u>float</u> <b>-pairmaxmispriming</b> <u>float</u>
                 <b>-numnsaccepted</b> <u>integer</u> <b>-selfany</b> <u>float</u> <b>-selfend</b> <u>float</u> <b>-scorrection</b> <u>list</u> <b>-tmformula</b> <u>list</u>
                 <b>-maxendstability</b> <u>float</u> <b>-outfile</b> <u>outfile</u>

       <b>eprimer32</b> <b>-help</b>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>eprimer32</b> is a command line program from EMBOSS (“the European Molecular Biology Open Software Suite”).
       It is part of the "Nucleic:Primers" command group(s).

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Input</b> <b>section</b>
       <b>-sequence</b> <u>seqall</u>
           The sequence from which to choose primers. The sequence must be presented 5' to 3'

       <b>-primer</b> <u>toggle</u>
           Tell Eprimer32 to pick primer(s) Default value: Y

       <b>-task</b> <u>list</u>
           Tell Eprimer32 what task to perform. Legal values are 1: 'Pick PCR primers', 2: 'Pick forward primer
           only', 3: 'Pick reverse primer only', 4: 'No primers needed'. Default value: 1

       <b>-hybridprobe</b> <u>toggle</u>
           An 'internal oligo' is intended to be used as a hybridization probe (hyb probe) to detect the PCR
           product after amplification. Default value: N

       <b>-mishyblibraryfile</b> <u>infile</u>
           Similar to MISPRIMING-LIBRARY, except that the event we seek to avoid is hybridization of the
           internal oligo to sequences in this library rather than priming from them. The file must be in (a
           slightly restricted) FASTA format (W. B. Pearson and D.J. Lipman, PNAS 85:8 pp 2444-2448 [1988]); we
           briefly discuss the organization of this file below. If this parameter is specified then Eprimer32
           locally aligns each candidate oligo against each library sequence and rejects those primers for which
           the local alignment score times a specified weight (see below) exceeds INTERNAL-OLIGO-MAX-MISHYB.
           (The maximum value of the weight is arbitrarily set to 12.0.) Each sequence entry in the FASTA-format
           file must begin with an 'id line' that starts with '&gt;'. The contents of the id line is 'slightly
           restricted' in that Eprimer32 parses everything after any optional asterisk ('*') as a floating point
           number to use as the weight mentioned above. If the id line contains no asterisk then the weight
           defaults to 1.0. The alignment scoring system used is the same as for calculating complementarity
           among oligos (e.g. SELF-ANY). The remainder of an entry contains the sequence as lines following the
           id line up until a line starting with '&gt;' or the end of the file. Whitespace and newlines are
           ignored. Characters 'A', 'T', 'G', 'C', 'a', 't', 'g', 'c' are retained and any other character is
           converted to 'N' (with the consequence that any IUB / IUPAC codes for ambiguous bases are converted
           to 'N'). There are no restrictions on line length. An empty value for this parameter indicates that
           no library should be used.

       <b>-mispriminglibraryfile</b> <u>infile</u>
           The name of a file containing a nucleotide sequence library of sequences to avoid amplifying (for
           example repetitive sequences, or possibly the sequences of genes in a gene family that should not be
           amplified.) The file must be in (a slightly restricted) FASTA format (W. B. Pearson and D.J. Lipman,
           PNAS 85:8 pp 2444-2448 [1988]); we briefly discuss the organization of this file below. If this
           parameter is specified then Eprimer32 locally aligns each candidate primer against each library
           sequence and rejects those primers for which the local alignment score times a specified weight (see
           below) exceeds MAX-MISPRIMING. (The maximum value of the weight is arbitrarily set to 100.0.) Each
           sequence entry in the FASTA-format file must begin with an 'id line' that starts with '&gt;'. The
           contents of the id line is 'slightly restricted' in that Eprimer32 parses everything after any
           optional asterisk ('*') as a floating point number to use as the weight mentioned above. If the id
           line contains no asterisk then the weight defaults to 1.0. The alignment scoring system used is the
           same as for calculating complementarity among oligos (e.g. SELF-ANY). The remainder of an entry
           contains the sequence as lines following the id line up until a line starting with '&gt;' or the end of
           the file. Whitespace and newlines are ignored. Characters 'A', 'T', 'G', 'C', 'a', 't', 'g', 'c' are
           retained and any other character is converted to 'N' (with the consequence that any IUB / IUPAC codes
           for ambiguous bases are converted to 'N'). There are no restrictions on line length. An empty value
           for this parameter indicates that no repeat library should be used.

   <b>Additional</b> <b>section</b>
   <b>Program</b> <b>options</b>
       <b>-numreturn</b> <u>integer</u>
           The maximum number of primer pairs to return. Primer pairs returned are sorted by their 'quality', in
           other words by the value of the objective function (where a lower number indicates a better primer
           pair). Caution: setting this parameter to a large value will increase running time. Default value: 5

   <b>Sequence</b> <b>options</b>
       <b>-includedregion</b> <u>range</u>
           A sub-region of the given sequence in which to pick primers. For example, often the first dozen or so
           bases of a sequence are vector, and should be excluded from consideration. The value for this
           parameter has the form (start),(end) where (start) is the index of the first base to consider, and
           (end) is the last in the primer-picking region.

       <b>-targetregion</b> <u>range</u>
           If one or more Targets is specified then a legal primer pair must flank at least one of them. A
           Target might be a simple sequence repeat site (for example a CA repeat) or a single-base-pair
           polymorphism. The value should be a space-separated list of (start),(end) pairs where (start) is the
           index of the first base of a Target, and (end) is the last E.g. 50,51 requires primers to surround
           the 2 bases at positions 50 and 51.

       <b>-excludedregion</b> <u>range</u>
           Primer oligos may not overlap any region specified in this tag. The associated value must be a
           space-separated list of (start),(end) pairs where (start) is the index of the first base of the
           excluded region, and and (end) is the last. This tag is useful for tasks such as excluding regions of
           low sequence quality or for excluding regions containing repetitive elements such as ALUs or LINEs.
           E.g. 401,407 68,70 forbids selection of primers in the 7 bases starting at 401 and the 3 bases at 68.

       <b>-forwardinput</b> <u>string</u>
           The sequence of a forward primer to check and around which to design reverse primers and optional
           internal oligos. Must be a substring of SEQUENCE.

       <b>-reverseinput</b> <u>string</u>
           The sequence of a reverse primer to check and around which to design forward primers and optional
           internal oligos. Must be a substring of the reverse strand of SEQUENCE.

   <b>Primer</b> <b>options</b>
       <b>-gcclamp</b> <u>integer</u>
           Require the specified number of consecutive Gs and Cs at the 3' end of both the forward and reverse
           primer. (This parameter has no effect on the internal oligo if one is requested.)

       <b>-osize</b> <u>integer</u>
           Optimum length (in bases) of a primer oligo. Eprimer32 will attempt to pick primers close to this
           length. Default value: 20

       <b>-minsize</b> <u>integer</u>
           Minimum acceptable length of a primer. Must be greater than 0 and less than or equal to MAX-SIZE.
           Default value: 18

       <b>-maxsize</b> <u>integer</u>
           Maximum acceptable length (in bases) of a primer. Currently this parameter cannot be larger than 35.
           This limit is governed by the maximum oligo size for which Eprimer32's melting-temperature is valid.
           Default value: 27

       <b>-otm</b> <u>float</u>
           Optimum melting temperature(Celsius) for a primer oligo. Eprimer32 will try to pick primers with
           melting temperatures are close to this temperature. The oligo melting temperature formula in
           Eprimer32 is that given in Rychlik, Spencer and Rhoads, Nucleic Acids Research, vol 18, num 21, pp
           6409-6412 and Breslauer, Frank, Bloecker and Marky, Proc. Natl. Acad. Sci. USA, vol 83, pp 3746-3750.
           Please refer to the former paper for background discussion. Default value: 60.0

       <b>-mintm</b> <u>float</u>
           Minimum acceptable melting temperature(Celsius) for a primer oligo. Default value: 57.0

       <b>-maxtm</b> <u>float</u>
           Maximum acceptable melting temperature(Celsius) for a primer oligo. Default value: 63.0

       <b>-maxdifftm</b> <u>float</u>
           Maximum acceptable (unsigned) difference between the melting temperatures of the forward and reverse
           primers. Default value: 100.0

       <b>-ogcpercent</b> <u>float</u>
           Primer optimum GC percent. Default value: 50.0

       <b>-mingc</b> <u>float</u>
           Minimum allowable percentage of Gs and Cs in any primer. Default value: 20.0

       <b>-maxgc</b> <u>float</u>
           Maximum allowable percentage of Gs and Cs in any primer generated by Primer. Default value: 80.0

       <b>-saltconc</b> <u>float</u>
           The millimolar concentration of salt (usually KCl) in the PCR. Eprimer32 uses this argument to
           calculate oligo melting temperatures. Default value: 50.0

       <b>-dnaconc</b> <u>float</u>
           The nanomolar concentration of annealing oligos in the PCR. Eprimer32 uses this argument to calculate
           oligo melting temperatures. The default (50nM) works well with the standard protocol used at the
           Whitehead/MIT Center for Genome Research--0.5 microliters of 20 micromolar concentration for each
           primer oligo in a 20 microliter reaction with 10 nanograms template, 0.025 units/microliter Taq
           polymerase in 0.1 mM each dNTP, 1.5mM MgCl2, 50mM KCl, 10mM Tris-HCL (pH 9.3) using 35 cycles with an
           annealing temperature of 56 degrees Celsius. This parameter corresponds to 'c' in Rychlik, Spencer
           and Rhoads' equation (ii) (Nucleic Acids Research, vol 18, num 21) where a suitable value (for a
           lower initial concentration of template) is 'empirically determined'. The value of this parameter is
           less than the actual concentration of oligos in the reaction because it is the concentration of
           annealing oligos, which in turn depends on the amount of template (including PCR product) in a given
           cycle. This concentration increases a great deal during a PCR; fortunately PCR seems quite robust for
           a variety of oligo melting temperatures. See ADVICE FOR PICKING PRIMERS. Default value: 50.0

       <b>-maxpolyx</b> <u>integer</u>
           The maximum allowable length of a mononucleotide repeat in a primer, for example AAAAAA. Default
           value: 5

   <b>Product</b> <b>options</b>
       <b>-psizeopt</b> <u>integer</u>
           The optimum size for the PCR product. 0 indicates that there is no optimum product size. Default
           value: 200

       <b>-prange</b> <u>range</u>
           The associated values specify the lengths of the product that the user wants the primers to create,
           and is a space separated list of elements of the form (x)-(y) where an (x)-(y) pair is a legal range
           of lengths for the product. For example, if one wants PCR products to be between 100 to 150 bases
           (inclusive) then one would set this parameter to 100-150. If one desires PCR products in either the
           range from 100 to 150 bases or in the range from 200 to 250 bases then one would set this parameter
           to 100-150 200-250. Eprimer32 favours ranges to the left side of the parameter string. Eprimer32 will
           return legal primers pairs in the first range regardless the value of the objective function for
           these pairs. Only if there are an insufficient number of primers in the first range will Eprimer32
           return primers in a subsequent range. Default value: 100-300

       <b>-ptmopt</b> <u>float</u>
           The optimum melting temperature for the PCR product. 0 indicates that there is no optimum
           temperature. Default value: 0.0

       <b>-ptmmin</b> <u>float</u>
           The minimum allowed melting temperature of the amplicon. Please see the documentation on the maximum
           melting temperature of the product for details. Default value: -1000000.0

       <b>-ptmmax</b> <u>float</u>
           The maximum allowed melting temperature of the amplicon. Product Tm is calculated using the formula
           from Bolton and McCarthy, PNAS 84:1390 (1962) as presented in Sambrook, Fritsch and Maniatis,
           Molecular Cloning, p 11.46 (1989, CSHL Press). Tm = 81.5 + 16.6(log10([Na+])) + .41*(%GC) -
           600/length Where [Na+} is the molar sodium concentration, (%GC) is the percent of Gs and Cs in the
           sequence, and length is the length of the sequence. A similar formula is used by the prime primer
           selection program in GCG, which instead uses 675.0/length in the last term (after F. Baldino, Jr,
           M.-F. Chesselet, and M.E. Lewis, Methods in Enzymology 168:766 (1989) eqn (1) on page 766 without the
           mismatch and formamide terms). The formulas here and in Baldino et al. assume Na+ rather than K+.
           According to J.G. Wetmur, Critical Reviews in BioChem. and Mol. Bio. 26:227 (1991) 50 mM K+ should be
           equivalent in these formulae to .2 M Na+. Eprimer32 uses the same salt concentration value for
           calculating both the primer melting temperature and the oligo melting temperature. If you are
           planning to use the PCR product for hybridization later this behavior will not give you the Tm under
           hybridization conditions. Default value: 1000000.0

   <b>Internal</b> <b>oligo</b> <b>input</b>
       <b>-oexcludedregion</b> <u>range</u>
           Middle oligos may not overlap any region specified by this tag. The associated value must be a
           space-separated list of (start),(end) pairs, where (start) is the index of the first base of an
           excluded region, and (end) is the last. Often one would make Target regions excluded regions for
           internal oligos.

       <b>-oligoinput</b> <u>string</u>
           The sequence of an internal oligo to check and around which to design forward and reverse primers.
           Must be a substring of SEQUENCE.

   <b>Internal</b> <b>oligo</b> <b>options</b>
       <b>-osizeopt</b> <u>integer</u>
           Optimum length (in bases) of an internal oligo. Eprimer32 will attempt to pick primers close to this
           length. Default value: 20

       <b>-ominsize</b> <u>integer</u>
           Minimum acceptable length of an internal oligo. Must be greater than 0 and less than or equal to
           INTERNAL-OLIGO-MAX-SIZE. Default value: 18

       <b>-omaxsize</b> <u>integer</u>
           Maximum acceptable length (in bases) of an internal oligo. Currently this parameter cannot be larger
           than 35. This limit is governed by maximum oligo size for which Eprimer32's melting-temperature is
           valid. Default value: 27

       <b>-otmopt</b> <u>float</u>
           Optimum melting temperature (Celsius) for an internal oligo. Eprimer32 will try to pick oligos with
           melting temperatures that are close to this temperature. The oligo melting temperature formula in
           Eprimer32 is that given in Rychlik, Spencer and Rhoads, Nucleic Acids Research, vol 18, num 21, pp
           6409-6412 and Breslauer, Frank, Bloecker and Marky, Proc. Natl. Acad. Sci. USA, vol 83, pp 3746-3750.
           Please refer to the former paper for background discussion. Default value: 60.0

       <b>-otmmin</b> <u>float</u>
           Minimum acceptable melting temperature(Celsius) for an internal oligo. Default value: 57.0

       <b>-otmmax</b> <u>float</u>
           Maximum acceptable melting temperature (Celsius) for an internal oligo. Default value: 63.0

       <b>-ogcopt</b> <u>float</u>
           Internal oligo optimum GC percent. Default value: 50.0

       <b>-ogcmin</b> <u>float</u>
           Minimum allowable percentage of Gs and Cs in an internal oligo. Default value: 20.0

       <b>-ogcmax</b> <u>float</u>
           Maximum allowable percentage of Gs and Cs in any internal oligo generated by Primer. Default value:
           80.0

       <b>-osaltconc</b> <u>float</u>
           The millimolar concentration of salt (usually KCl) in the hybridization. Eprimer32 uses this argument
           to calculate internal oligo melting temperatures. Default value: 50.0

       <b>-odnaconc</b> <u>float</u>
           The nanomolar concentration of annealing internal oligo in the hybridization. Default value: 50.0

       <b>-oanyself</b> <u>float</u>
           The maximum allowable local alignment score when testing an internal oligo for (local)
           self-complementarity. Local self-complementarity is taken to predict the tendency of oligos to anneal
           to themselves The scoring system gives 1.00 for complementary bases, -0.25 for a match of any base
           (or N) with an N, -1.00 for a mismatch, and -2.00 for a gap. Only single-base-pair gaps are allowed.
           For example, the alignment 5' ATCGNA 3' || | | 3' TA-CGT 5' is allowed (and yields a score of 1.75),
           but the alignment 5' ATCCGNA 3' || | | 3' TA--CGT 5' is not considered. Scores are non-negative, and
           a score of 0.00 indicates that there is no reasonable local alignment between two oligos. Default
           value: 12.00

       <b>-oendself</b> <u>float</u>
           The maximum allowable 3'-anchored global alignment score when testing a single oligo for
           self-complementarity. The scoring system is as for the Maximum Complementarity argument. In the
           examples above the scores are 7.00 and 6.00 respectively. Scores are non-negative, and a score of
           0.00 indicates that there is no reasonable 3'-anchored global alignment between two oligos. In order
           to estimate 3'-anchored global alignments for candidate oligos, Primer assumes that the sequence from
           which to choose oligos is presented 5' to 3'. INTERNAL-OLIGO-SELF-END is meaningless when applied to
           internal oligos used for hybridization-based detection, since primer-dimer will not occur. We
           recommend that INTERNAL-OLIGO-SELF-END be set at least as high as INTERNAL-OLIGO-SELF-ANY. Default
           value: 12.00

       <b>-opolyxmax</b> <u>integer</u>
           The maximum allowable length of an internal oligo mononucleotide repeat, for example AAAAAA. Default
           value: 5

       <b>-omishybmax</b> <u>float</u>
           Similar to MAX-MISPRIMING except that this parameter applies to the similarity of candidate internal
           oligos to the library specified in INTERNAL-OLIGO-MISHYB-LIBRARY. Default value: 12.0

   <b>Advanced</b> <b>section</b>
       <b>-explainflag</b> <u>boolean</u>
           If this flag is true, produce LEFT-EXPLAIN, RIGHT-EXPLAIN, and INTERNAL-OLIGO-EXPLAIN output tags,
           which are intended to provide information on the number of oligos and primer pairs that Eprimer32
           examined, and statistics on the number discarded for various reasons. Default value: N

       <b>-fileflag</b> <u>boolean</u>
           If the associated value is true, then Eprimer32 creates two output files for each input SEQUENCE.
           File (sequence-id).for lists all acceptable forward primers for (sequence-id), and (sequence-id).rev
           lists all acceptable reverse primers for (sequence-id), where (sequence-id) is the value of the
           SEQUENCE-ID tag (which must be supplied). In addition, if the input tag TASK is 1 or 4, Eprimer32
           produces a file (sequence-id).int, which lists all acceptable internal oligos. Default value: N

       <b>-firstbaseindex</b> <u>integer</u>
           This parameter is the index of the first base in the input sequence. For input and output using
           1-based indexing (such as that used in GenBank and to which many users are accustomed) set this
           parameter to 1. For input and output using 0-based indexing set this parameter to 0. (This parameter
           also affects the indexes in the contents of the files produced when the primer file flag is set.)
           Default value: 1

       <b>-pickanyway</b> <u>boolean</u>
           If true pick a primer pair even if LEFT-INPUT, RIGHT-INPUT, or INTERNAL-OLIGO-INPUT violates specific
           constraints. Default value: N

       <b>-maxmispriming</b> <u>float</u>
           The maximum allowed weighted similarity with any sequence in MISPRIMING-LIBRARY. Default value: 12.00

       <b>-pairmaxmispriming</b> <u>float</u>
           The maximum allowed sum of weighted similarities of a primer pair (one similarity for each primer)
           with any single sequence in MISPRIMING-LIBRARY. Default value: 24.00

       <b>-numnsaccepted</b> <u>integer</u>
           Maximum number of unknown bases (N) allowable in any primer.

       <b>-selfany</b> <u>float</u>
           The maximum allowable local alignment score when testing a single primer for (local)
           self-complementarity and the maximum allowable local alignment score when testing for complementarity
           between forward and reverse primers. Local self-complementarity is taken to predict the tendency of
           primers to anneal to each other without necessarily causing self-priming in the PCR. The scoring
           system gives 1.00 for complementary bases, -0.25 for a match of any base (or N) with an N, -1.00 for
           a mismatch, and -2.00 for a gap. Only single-base-pair gaps are allowed. For example, the alignment
           5' ATCGNA 3' ...|| | | 3' TA-CGT 5' is allowed (and yields a score of 1.75), but the alignment 5'
           ATCCGNA 3' ...|| | | 3' TA--CGT 5' is not considered. Scores are non-negative, and a score of 0.00
           indicates that there is no reasonable local alignment between two oligos. Default value: 8.00

       <b>-selfend</b> <u>float</u>
           The maximum allowable 3'-anchored global alignment score when testing a single primer for
           self-complementarity, and the maximum allowable 3'-anchored global alignment score when testing for
           complementarity between forward and reverse primers. The 3'-anchored global alignment score is taken
           to predict the likelihood of PCR-priming primer-dimers, for example 5' ATGCCCTAGCTTCCGGATG 3'
           .............||| ||||| ..........3' AAGTCCTACATTTAGCCTAGT 5' or 5' AGGCTATGGGCCTCGCGA 3'
           ...............|||||| ............3' AGCGCTCCGGGTATCGGA 5' The scoring system is as for the Maximum
           Complementarity argument. In the examples above the scores are 7.00 and 6.00 respectively. Scores are
           non-negative, and a score of 0.00 indicates that there is no reasonable 3'-anchored global alignment
           between two oligos. In order to estimate 3'-anchored global alignments for candidate primers and
           primer pairs, Primer assumes that the sequence from which to choose primers is presented 5' to 3'. It
           is nonsensical to provide a larger value for this parameter than for the Maximum (local)
           Complementarity parameter because the score of a local alignment will always be at least as great as
           the score of a global alignment. Default value: 3.00

       <b>-scorrection</b> <u>list</u>
           Specifies the salt correction formula for the melting temperature calculation. Default value: 1

       <b>-tmformula</b> <u>list</u>
           Specifies details of melting temperature calculation. Default value: 1

   <b>Primer</b> <b>penalty</b> <b>weights</b>
       <b>-maxendstability</b> <u>float</u>
           The maximum stability for the five 3' bases of a forward or reverse primer. Bigger numbers mean more
           stable 3' ends. The value is the maximum delta G for duplex disruption for the five 3' bases as
           calculated using the nearest neighbor parameters published in Breslauer, Frank, Bloecker and Marky,
           Proc. Natl. Acad. Sci. USA, vol 83, pp 3746-3750. Eprimer32 uses a completely permissive default
           value for backward compatibility (which we may change in the next release). Rychlik recommends a
           maximum value of 9 (Wojciech Rychlik, 'Selection of Primers for Polymerase Chain Reaction' in BA
           White, Ed., 'Methods in Molecular Biology, Vol. 15: PCR Protocols: Current Methods and Applications',
           1993, pp 31-40, Humana Press, Totowa NJ). Default value: 9.0

   <b>Output</b> <b>section</b>
       <b>-outfile</b> <u>outfile</u>

</pre><h4><b>BUGS</b></h4><pre>
       Bugs can be reported to the Debian Bug Tracking system (<a href="http://bugs.debian.org/emboss">http://bugs.debian.org/emboss</a>), or directly to
       the EMBOSS developers (<a href="http://sourceforge.net/tracker/">http://sourceforge.net/tracker/</a>?group_id=93650&amp;atid=605031).

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       eprimer32 is fully documented via the <b><a href="../man1/tfm.1.html">tfm</a></b>(1) system.

</pre><h4><b>AUTHOR</b></h4><pre>
       <b>Debian</b> <b>Med</b> <b>Packaging</b> <b>Team</b> &lt;<a href="mailto:debian-med-packaging@lists.alioth.debian.org">debian-med-packaging@lists.alioth.debian.org</a>&gt;
           Wrote the script used to autogenerate this manual page.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       This manual page was autogenerated from an Ajax Control Definition of the EMBOSS package. It can be
       redistributed under the same terms as EMBOSS itself.

EMBOSS 6.4.0                                       05/11/2012                                      <u><a href="../man1e/EPRIMER32.1e.html">EPRIMER32</a></u>(1e)
</pre>
 </div>
</div></section>
</div>
</body>
</html>