<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>RNAhybrid - calculate secondary structure hybridisations of RNAs</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/rnahybrid">rnahybrid_2.1.2-8_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       RNAhybrid - calculate secondary structure hybridisations of RNAs

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>RNAhybrid</b>  <b>[-h]</b>  <b>[-b</b>  <u>hit_number</u><b>]</b> <b>[-e</b> <u>energy_cutoff</u><b>]</b> <b>[-p</b> <u>p-value_cutoff</u><b>]</b> <b>[-c]</b> <b>[-d</b> <u>xi,theta</u><b>]</b> <b>[-s</b> <u>set_name</u><b>]</b>
       <b>[-f</b> <u>from,to</u><b>]</b> <b>[-m</b> <u>max_target_length</u><b>]</b> <b>[-n</b> <u>max_query_length</u><b>]</b> <b>[-u</b> <u>iloop_upper_limit</u><b>]</b>  <b>[-v</b>  <u>bloop_upper_limit</u><b>]</b>
       <b>[-g</b> <b>(ps|png|jpg|all)]</b> <b>[-t</b> <u>target_file</u><b>]</b> <b>[-q</b> <u>query_file</u><b>]</b> <b>[</b><u>target</u><b>]</b> <b>[</b><u>query</u><b>]</b>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <u>RNAhybrid</u>  is  a tool for finding minimum free energy (mfe) hybridisations of a long (target) and a short
       (query) RNA. The hybridisation is performed in  a  kind  of  domain  mode,  ie.  the  short  sequence  is
       hybridised to the best fitting parts of the long one. The tool is primarily meant as a means for microRNA
       target  prediction.   In  addition  to  mfes,  the  program  calculates  p-values  based on extreme value
       distributions of length normalised energies.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>     Give a short summary of command line options.

       <b>-b</b> <u>hit_number</u>
              Maximal number of hits to show. <u>hit_number</u> hits with increasing  minimum  free  energy  (reminder:
              larger energies are worse) are shown, unless the -e option is used and the energy cut-off has been
              exceeded  (see -e option below) or there are no more hits. Hits may only overlap at dangling bases
              (5' or 3' unpaired end of target).

       <b>-c</b>     Produce compact output. For each target/query pair one line of output is generated. Each line is a
              colon (:) separated list of the following fields: target name, query name,  minimum  free  energy,
              position  in  target, alignment line 1, line 2, line 3, line 4. If a target or a query is given on
              command line (ie. no -t or -q respectively), its name in the output will be "command line".

       <b>-d</b> <u>xi,theta</u>
              xi and theta are the position and shape parameters, respectively, of an extreme value distribution
              (evd). p-values of duplex energies are assumed to be distributed according to such an evd.  For  a
              length  normalised  energy  en,  we  have  P[X &lt;= en] = 1 - exp(-exp(-(-en-xi)/theta)), where en =
              e/log(m*n) with m and n being the lengths of the target and the query,  respectively.  If  the  -d
              option  is  omitted,  xi  and  theta  are  estimated  from the maximal duplex energy of the query,
              assuming a linear dependence. The parameters of this linear dependence are coded into the program,
              but the option -s has to be given to choose from  the  appropriate  set.  Note  that  the  evd  is
              mirrored, since good mfes are large negative values.

       <b>-s</b> <u>set_name</u>
              Used  for  quick  estimate  of  extreme value distribution parameters (see -d option above). Tells
              RNAhybrid which target dataset to assume. Valid parameters are 3utr_fly, 3utr_worm and 3utr_human.

       <b>-e</b> <u>energy_cutoff</u>
              Hits with increasing minimum free energy (reminder: larger energies are worse) less than or  equal
              to  <u>energy_cutoff</u>  are shown, unless the -b option is used and the number of already reported hits
              has reached the maximal <u>hit_number</u> (see -b option above). Hits may only overlap at dangling  bases
              (5' or 3' unpaired end of target).

       <b>-p</b> <u>p-value_cutoff</u>
              Only hits with p-values not larger than <u>p-value_cutoff</u> are reported.  See also options -d and -s.

       <b>-f</b> <u>from,to</u>
              Forces all structures to have a helix from position <u>from</u> to position <u>to</u> with respect to the query.
              The first base has position 1.

       <b>-m</b> <u>max_target_length</u>
              The  maximum  allowed length of a target sequence. The default value is 2000. This option only has
              an effect if a target file is given with the -t option (see below).

       <b>-n</b> <u>max_query_length</u>
              The maximum allowed length of a query sequence. The default value is 30. This option only  has  an
              effect if a query file is given with the -q option (see below).

       <b>-u</b> <u>iloop_upper_limit</u>
              The maximally allowed number of unpaired nucleotides in either side of an internal loop.

       <b>-v</b> <u>bloop_upper_limit</u>
              The maximally allowed number of unpaired nucleotides in a bulge loop.

       <b>-g</b> <b>(ps|png|jpg|all)</b>
              Produce a plot of the hybridisation, either in ps, png or jpg format, or for all formats together.
              The  plots  are  saved  in  files  whose  names  are  created  from  the  target  and  query names
              ("command_line" if given on the command line). This option only works, if the appropriate graphics
              libraries are present.

       <b>-t</b> <u>target_file</u>
              Each of the target sequences in <u>target_file</u> is subject to hybridisation with each of  the  queries
              (which  either  are  from the <u>query_file</u> or is the one query given on command line; see -q below).
              The sequences in the <u>target_file</u> have to be in FASTA format, ie. one line starting with  a  &gt;  and
              directly  followed  by  a  name,  then  one or more following lines with the sequence itself. Each
              individual sequence line must not have more than 1000 characters. If no -t is  given,  either  the
              last  argument  (if a -q is given) or the second last argument (if no -q is given) to RNAhybrid is
              taken as a target.

       <b>-q</b> <u>query_file</u>
              See -t option above.

</pre><h4><b>REFERENCES</b></h4><pre>
       The energy parameters are taken from:

       Mathews DH, Sabina J, Zuker M, Turner DH.  "Expanded  sequence  dependence  of  thermodynamic  parameters
       improves prediction of RNA secondary structure" J Mol Biol., 288 (5), pp 911-940, 1999

       The graphical output uses code from the Vienna RNA package:

       Hofacker  IL.   "Vienna  RNA secondary structure server."  Nucleic Acids Research, 31 (13), pp 3429-3431,
       2003

</pre><h4><b>VERSION</b></h4><pre>
       This man page documents version 2.0 of RNAhybrid.

</pre><h4><b>AUTHORS</b></h4><pre>
       Marc Rehmsmeier, Peter Steffen, Matthias Hoechsmann.

</pre><h4><b>LIMITATIONS</b></h4><pre>
       Character dependent energy values are only defined for [acgtuACGTU].  All other characters lead to values
       of zero in these cases.

</pre><h4><b>BUGS</b></h4><pre>
       In suboptimal hits, dangling ends appear as Ns if they were in the first or last hybridising position  of
       a previous hit.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       RNAcalibrate, RNAeffective

                                                                                                    <u><a href="../man1/RNAHYBRID.1.html">RNAHYBRID</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>