<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>sniffles - structural variation caller using third-generation sequencing</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/sniffles">sniffles_2.6.0-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       sniffles - structural variation caller using third-generation sequencing

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage:  sniffles  <b>--input</b>  SORTED_INPUT.bam [--vcf OUTPUT.vcf] [--snf MERGEABLE_OUTPUT.snf] [--threads 4]
       [--non-germline]

       Sniffles2: A fast structural variant (SV) caller for long-read sequencing data

              Version 2.0.2 Contact: <a href="mailto:moritz.g.smolka@gmail.com">moritz.g.smolka@gmail.com</a>

              Usage example A - Call SVs for a single sample:

              sniffles <b>--input</b> sorted_indexed_alignments.bam <b>--vcf</b> output.vcf

              ... OR, with CRAM input and bgzipped+tabix indexed VCF output:

              sniffles <b>--input</b> sample.cram <b>--vcf</b> output.vcf.gz

              ... OR, producing only a SNF file with SV candidates for later multi-sample calling:

              sniffles <b>--input</b> sample1.bam <b>--snf</b> sample1.snf

              ... OR, simultaneously producing a single-sample VCF and SNF file for later multi-sample calling:

              sniffles <b>--input</b> sample1.bam <b>--vcf</b> sample1.vcf.gz <b>--snf</b> sample1.snf

              ... OR, with additional options to specify tandem repeat annotations (for improved call accuracy),
              reference (for DEL sequences) and non-germline mode for detecting rare SVs:

              sniffles <b>--input</b> sample1.bam <b>--vcf</b> sample1.vcf.gz <b>--tandem-repeats</b> tandem_repeats.bed  <b>--reference</b>
              genome.fa <b>--non-germline</b>

              Usage example B - Multi-sample calling:

              Step  1.  Create  .snf  for  each  sample:  sniffles <b>--input</b> sample1.bam <b>--snf</b> sample1.snf Step 2.
              Combined calling: sniffles <b>--input</b> sample1.snf sample2.snf ... sampleN.snf <b>--vcf</b> multisample.vcf

              ... OR, using a .tsv file containing a list of .snf files, and custom sample ids  in  an  optional
              second column (one sample per line): Step 2. Combined calling: sniffles <b>--input</b> snf_files_list.tsv
              <b>--vcf</b> multisample.vcf

              Usage example C - Determine genotypes for a set of known SVs (force calling):

              sniffles <b>--input</b> sample.bam <b>--genotype-vcf</b> input_known_svs.vcf <b>--vcf</b> output_genotypes.vcf

              Use <b>--help</b> for full parameter/usage information

   <b>optional</b> <b>arguments:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>--version</b>
              show program's version number and exit

   <b>Common</b> <b>parameters:</b>
       <b>-i</b> IN [IN ...], <b>--input</b> IN [IN ...]
              For  single-sample  calling:  A  coordinate-sorted  and  indexed .bam/.cram (BAM/CRAM format) file
              containing aligned reads. - OR - For multi-sample calling: Multiple .snf files  (generated  before
              by running Sniffles2 for individual samples with <b>--snf</b>) (default: None)

       <b>-v</b> OUT.vcf, <b>--vcf</b> OUT.vcf
              VCF  output  filename to write the called and refined SVs to. If the given filename ends with .gz,
              the VCF file will be automatically bgzipped and a .tbi index built for it. (default: None)

       <b>--snf</b> OUT.snf
              Sniffles2 file (.snf) output filename to store candidates for later multi-sample calling (default:
              None)

       <b>--reference</b> reference.fasta
              (Optional) Reference sequence the reads were aligned against. To  enable  output  of  deletion  SV
              sequences, this parameter must be set. (default: None)

       <b>--tandem-repeats</b> IN.bed
              (Optional)  Input  .bed  file  containing  tandem  repeat  annotations  for  the reference genome.
              (default: None)

       <b>--non-germline</b>
              Call non-germline SVs (rare, somatic or mosaic SVs) (default: False)

       <b>--phase</b>
              Determine phase for SV calls (requires the input alignments to be phased) (default: False)

       <b>-t</b> N, <b>--threads</b> N
              Number of parallel threads to use (speed-up for multi-core CPUs) (default: 4)

   <b>SV</b> <b>Filtering</b> <b>parameters:</b>
       <b>--minsupport</b> auto
              Minimum number of supporting reads for a SV to be reported (default: automatically choose based on
              coverage) (default: auto)

       <b>--minsupport-auto-mult</b> 0.1/0.025
              Coverage  based  minimum  support  multiplier  for  germline/non-germline  modes  (only  for  auto
              minsupport) (default: None)

       <b>--minsvlen</b> N
              Minimum SV length (in bp) (default: 35)

       <b>--minsvlen-screen-ratio</b> N
              Minimum length for SV candidates (as fraction of <b>--minsvlen</b>) (default: 0.95)

       <b>--mapq</b> N
              Alignments with mapping quality lower than this value will be ignored (default: 25)

       <b>--no-qc</b>
              Output all SV candidates, disregarding quality control steps. (default: False)

       <b>--qc-stdev</b> True
              Apply filtering based on SV start position and length standard deviation (default: True)

       <b>--qc-stdev-abs-max</b> N
              Maximum standard deviation for SV length and size (in bp) (default: 500)

       <b>--qc-strand</b> False
              Apply filtering based on strand support of SV calls (default: False)

       <b>--qc-coverage</b> N
              Minimum surrounding region coverage of SV calls (default: 1)

       <b>--long-ins-length</b> 2500
              Insertion  SVs  longer  than  this value are considered as hard to detect based on the aligner and
              read length and subjected to more sensitive filtering. (default: 2500)

       <b>--long-del-length</b> 50000
              Deletion SVs longer than this value are subjected to central coverage  drop-based  filtering  (Not
              applicable for <b>--non-germline</b>) (default: 50000)

       <b>--long-del-coverage</b> 0.66
              Long  deletions  with  central  coverage (in relation to upstream/downstream coverage) higher than
              this value will be filtered (Not applicable for <b>--non-germline</b>) (default: 0.66)

       <b>--long-dup-length</b> 50000
              Duplication SVs longer than this value are subjected to central coverage increase-based  filtering
              (Not applicable for <b>--non-germline</b>) (default: 50000)

       <b>--long-dup-coverage</b> 1.33
              Long  duplications  with central coverage (in relation to upstream/downstream coverage) lower than
              this value will be filtered (Not applicable for <b>--non-germline</b>) (default: 1.33)

       <b>--max-splits-kb</b> N
              Additional number of splits per kilobase read sequence allowed before reads are ignored  (default:
              0.1)

       <b>--max-splits-base</b> N
              Base  number of splits allowed before reads are ignored (in addition to <b>--max-splits-kb</b>) (default:
              3)

       <b>--min-alignment-length</b> N
              Reads with alignments shorter than this length (in bp) will be ignored (default: 1000)

       <b>--phase-conflict-threshold</b> F
              Maximum fraction of conflicting reads permitted for SV phase information to be  labelled  as  PASS
              (only for <b>--phase</b>) (default: 0.1)

       <b>--detect-large-ins</b> True
              Infer insertions that are longer than most reads and therefore are spanned by few alignments only.
              (default: True)

   <b>SV</b> <b>Clustering</b> <b>parameters:</b>
       <b>--cluster-binsize</b> N
              Initial screening bin size in bp (default: 100)

       <b>--cluster-r</b> R
              Multiplier for SV start position standard deviation criterion in cluster merging (default: 2.5)

       <b>--cluster-repeat-h</b> H
              Multiplier for mean SV length criterion for tandem repeat cluster merging (default: 1.5)

       <b>--cluster-repeat-h-max</b> N
              Max.  merging  distance  based  on SV length criterion for tandem repeat cluster merging (default:
              1000)

       <b>--cluster-merge-pos</b> N
              Max. merging distance for insertions and deletions on the same  read  and  cluster  in  non-repeat
              regions (default: 150)

       <b>--cluster-merge-len</b> F
              Max. size difference for merging SVs as fraction of SV length (default: 0.33)

       <b>--cluster-merge-bnd</b> N
              Max. merging distance for breakend SV candidates. (default: 1500)

   <b>SV</b> <b>Genotyping</b> <b>parameters:</b>
       <b>--genotype-ploidy</b> N
              Sample ploidy (currently fixed at value 2) (default: 2)

       <b>--genotype-error</b> N
              Estimated false positive rate for leads (relating to total coverage) (default: 0.05)

       <b>--sample-id</b> SAMPLE_ID
              Custom ID for this sample, used for later multi-sample calling (stored in .snf) (default: None)

       <b>--genotype-vcf</b> IN.vcf
              Determine  the  genotypes  for all SVs in the given input .vcf file (forced calling). Re-genotyped
              .vcf will be written to the output file specified with <b>--vcf</b>. (default: None)

   <b>Multi-Sample</b> <b>Calling</b> <b>/</b> <b>Combine</b> <b>parameters:</b>
       <b>--combine-high-confidence</b> F
              Minimum fraction of samples in which a SV needs to have  individually  passed  QC  for  it  to  be
              reported  in  combined output (a value of zero will report all SVs that pass QC in at least one of
              the input samples) (default: 0.0)

       <b>--combine-low-confidence</b> F
              Minimum fraction of samples in which a SV needs to be present (failed QC) for it to be reported in
              combined output (default: 0.2)

       <b>--combine-low-confidence-abs</b> N
              Minimum absolute number of samples in which a SV needs to be present (failed  QC)  for  it  to  be
              reported in combined output (default: 3)

       <b>--combine-null-min-coverage</b> N
              Minimum coverage for a sample genotype to be reported as 0/0 (sample genotypes with coverage below
              this threshold at the SV location will be output as .<a href="file:/.">/.</a>) (default: 5)

       <b>--combine-match</b> N
              Maximum  deviation  of  multiple  SV's  start/end position for them to be combined across samples.
              Given by max_dev=M*sqrt(min(SV_length_a,SV_length_b)), where M is this parameter. (default: 500)

       <b>--combine-consensus</b>
              Output the consensus genotype of all samples (default: False)

       <b>--combine-separate-intra</b>
              Disable combination of SVs within the same sample (default: False)

       <b>--combine-output-filtered</b>
              Include low-confidence / putative non-germline SVs in multi-calling (default: False)

   <b>SV</b> <b>Postprocessing,</b> <b>QC</b> <b>and</b> <b>output</b> <b>parameters:</b>
       <b>--output-rnames</b>
              Output names of all supporting reads for each SV in the RNAMEs info field (default: False)

       <b>--no-consensus</b>
              Disable consensus sequence generation for insertion SV calls (may improve  performance)  (default:
              False)

       <b>--no-sort</b>
              Do not sort output VCF by genomic coordinates (may slightly improve performance) (default: False)

       <b>--no-progress</b>
              Disable progress display (default: False)

       <b>--quiet</b>
              Disable all logging, except errors (default: False)

       <b>--max-del-seq-len</b> N
              Maximum deletion sequence length to be output. Deletion SVs longer than this value will be written
              to the output as symbolic SVs. (default: 50000)

       <b>--symbolic</b>
              Output  all  SVs  as symbolic, including insertions and deletions, instead of reporting nucleotide
              sequences.  (default: False)

              Usage example A - Call SVs for a single sample:

              sniffles <b>--input</b> sorted_indexed_alignments.bam <b>--vcf</b> output.vcf

              ... OR, with CRAM input and bgzipped+tabix indexed VCF output:

              sniffles <b>--input</b> sample.cram <b>--vcf</b> output.vcf.gz

              ... OR, producing only a SNF file with SV candidates for later multi-sample calling:

              sniffles <b>--input</b> sample1.bam <b>--snf</b> sample1.snf

              ... OR, simultaneously producing a single-sample VCF and SNF file for later multi-sample calling:

              sniffles <b>--input</b> sample1.bam <b>--vcf</b> sample1.vcf.gz <b>--snf</b> sample1.snf

              ... OR, with additional options to specify tandem repeat annotations (for improved call accuracy),
              reference (for DEL sequences) and non-germline mode for detecting rare SVs:

              sniffles <b>--input</b> sample1.bam <b>--vcf</b> sample1.vcf.gz <b>--tandem-repeats</b> tandem_repeats.bed  <b>--reference</b>
              genome.fa <b>--non-germline</b>

              Usage example B - Multi-sample calling:

              Step  1.  Create  .snf  for  each  sample:  sniffles <b>--input</b> sample1.bam <b>--snf</b> sample1.snf Step 2.
              Combined calling: sniffles <b>--input</b> sample1.snf sample2.snf ... sampleN.snf <b>--vcf</b> multisample.vcf

              ... OR, using a .tsv file containing a list of .snf files, and custom sample ids  in  an  optional
              second column (one sample per line): Step 2. Combined calling: sniffles <b>--input</b> snf_files_list.tsv
              <b>--vcf</b> multisample.vcf

              Usage example C - Determine genotypes for a set of known SVs (force calling):

              sniffles <b>--input</b> sample.bam <b>--genotype-vcf</b> input_known_svs.vcf <b>--vcf</b> output_genotypes.vcf

</pre><h4><b>AUTHOR</b></h4><pre>
        This manpage was written by Andreas Tille for the Debian distribution and
        can be used for any other usage of the program.

sniffles 2.0.2                                    February 2022                                      <u><a href="../man1/SNIFFLES.1.html">SNIFFLES</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>