<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Run::Phylo::PAML::Evolver - Wrapper aroud the PAML program evolver</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-tools-phylo-paml-perl">libbio-tools-phylo-paml-perl_1.7.3-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Run::Phylo::PAML::Evolver - Wrapper aroud the PAML program evolver

</pre><h4><b>VERSION</b></h4><pre>
       version 1.7.3

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::Tools::Run::Phylo::PAML::Evolver;

         my $evolver = Bio::Tools::Run::Phylo::PAML::Evolver-&gt;new();

         # Get a $tree object somehow
         $evolver-&gt;tree($tree);

         # FIXME: evolver generates a tree (first run with option 1 or 2)?

         # One or more alns are created
         my @alns = $evolver-&gt;run();

         ####

         # Or with all the data coming from a previous PAML run
         my $parser = Bio::Tools::Phylo::PAML-&gt;new
           (
            -file =&gt; "$inputfile",
           );
         my $result = $parser-&gt;next_result();
         my $tree = $result-&gt;next_tree;
         $evolver-&gt;tree($tree);
         my @codon_freqs = $result-&gt;get_CodonFreqs();
         $evolver-&gt;set_CodonFreqs(\@codon_freqs);

         my $val = $evolver-&gt;prepare();

         # FIXME: something similar for nucleotide frequencies:
         # Option (5) Simulate nucleotide data sets (use MCbase.dat)?

         # FIXME: something similar for aa parameters:
         # Option (7) Simulate amino acid data sets (use MCaa.dat)?

         # FIXME: With an initial RootSeq.txt

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This is a wrapper around the evolver program of PAML (Phylogenetic Analysis by Maximum Likelihood)
       package of Ziheng Yang.  See <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">http://abacus.gene.ucl.ac.uk/software/paml.html</a> for more information.

       This module is more about generating the properl MCmodel.ctl file and will run the program in a separate
       temporary directory to avoid creating temp files all over the place.

   <b>Default</b> <b>Values</b>
       Valid and default values for evolver programs are listed below.  The default values are always the first
       one listed.  These descriptions are essentially lifted from the example MCcodon.dat file and pamlDOC
       documentation provided by the author.

       Stub:

       <b>option1</b> specifies something.

       <b>option2</b> specifies something else.

       INCOMPLETE DOCUMENTATION OF ALL METHODS

</pre><h4><b>ATTRIBUTES</b></h4><pre>
   <b>program_name</b>
        Title   : program_name
        Usage   : $factory-&gt;program_name()
        Function: holds the program name
        Returns:  string
        Args    : None

   <b>program_dir</b>
        Title   : program_dir
        Usage   : -&gt;program_dir()
        Function: returns the program directory, obtained from ENV variable.
        Returns:  string
        Args    :

   <b>error_string</b>
        Title   : error_string
        Usage   : $obj-&gt;error_string($newval)
        Function: Where the output from the last analysis run is stored.
        Returns : value of error_string
        Args    : newvalue (optional)

   <b>alignment</b>
        Title   : alignment
        Usage   : $evolver-&gt;align($aln);
        Function: Get/Set the L&lt;Bio::Align::AlignI&gt; object
        Returns : L&lt;Bio::Align::AlignI&gt; object
        Args    : [optional] L&lt;Bio::Align::AlignI&gt;
        Comment : We could potentially add support for running directly on a file
                  but we shall keep it simple
        See also: L&lt;Bio::SimpleAlign&gt;

   <b>tree</b>
        Title   : tree
        Usage   : $evolver-&gt;tree($tree, %params);
        Function: Get/Set the L&lt;Bio::Tree::TreeI&gt; object
        Returns : L&lt;Bio::Tree::TreeI&gt;
        Args    : [optional] $tree =&gt; L&lt;Bio::Tree::TreeI&gt;,
                  [optional] %parameters =&gt; hash of tree-specific parameters:
                         branchLengths: 0, 1 or 2
                         out

        Comment : We could potentially add support for running directly on a file
                  but we shall keep it simple
        See also: L&lt;Bio::Tree::Tree&gt;

   <b>no_param_checks</b>
        Title   : no_param_checks
        Usage   : $obj-&gt;no_param_checks($newval)
        Function: Boolean flag as to whether or not we should
                  trust the sanity checks for parameter values
        Returns : value of no_param_checks
        Args    : newvalue (optional)

   <b>save_tempfiles</b>
        Title   : save_tempfiles
        Usage   : $obj-&gt;save_tempfiles($newval)
        Function:
        Returns : value of save_tempfiles
        Args    : newvalue (optional)

   <b>outfile_name</b>
        Title   : outfile_name
        Usage   : my $outfile = $evolver-&gt;outfile_name();
        Function: Get/Set the name of the output file for this run
                  (if you wanted to do something special)
        Returns : string
        Args    : [optional] string to set value to

   <b>tempdir</b>
        Title   : tempdir
        Usage   : my $tmpdir = $self-&gt;tempdir();
        Function: Retrieve a temporary directory name (which is created)
        Returns : string which is the name of the temporary directory
        Args    : none

   <b>indel</b>
        Title   : indel
        Usage   : $obj-&gt;indel($newval)
        Function: this is only useful if using evolver_indel instead of main
                  evolver package:
                  Exploring the Relationship between Sequence Similarity and
                  Accurate Phylogenetic Trees Brandi L. Cantarel, Hilary
                  G. Morrison and William Pearson
        Example :
        Returns : value of indel (a scalar)
        Args    : on set, new value (a scalar or undef, optional)

</pre><h4><b>METHODS</b></h4><pre>
   <b>new</b>
        Title   : new
        Usage   : my $obj = Bio::Tools::Run::Phylo::PAML::Evolver-&gt;new();
        Function: Builds a new Bio::Tools::Run::Phylo::PAML::Evolver object
        Returns : Bio::Tools::Run::Phylo::PAML::Evolver
                  -save_tempfiles =&gt; boolean to save the generated tempfiles and
                                     NOT cleanup after onesself (default FALSE)
                  -tree =&gt; the Bio::Tree::TreeI object (FIXME: optional if this is done in a first run)
                  -params =&gt; a hashref of PAML parameters (all passed to set_parameter)
                  -executable =&gt; where the evolver executable resides

       See also: Bio::Tree::TreeI

   <b>prepare</b>
        Title   : prepare
        Usage   : my $rundir = $evolver-&gt;prepare($aln);
        Function: prepare the evolver analysis using the default or updated parameters
                  the alignment parameter must have been set
        Returns : value of rundir
        Args    : L&lt;Bio::Align::AlignI&gt; object,
              L&lt;Bio::Tree::TreeI&gt; object [optional]

   <b>run</b>
        Title   : run
        Usage   : my ($rc,$parser) = $evolver-&gt;run();
        Function: run the evolver analysis using the default or updated parameters
                  the alignment parameter must have been set
        Returns : Return code, L&lt;Bio::Tools::Phylo::PAML&gt;
        Args    : L&lt;Bio::Align::AlignI&gt; object,
              L&lt;Bio::Tree::TreeI&gt; object [optional]

   <b>get_parameters</b>
        Title   : get_parameters
        Usage   : my %params = $self-&gt;get_parameters();
        Function: returns the list of parameters as a hash
        Returns : associative array keyed on parameter names
        Args    : none

   <b>set_parameter</b>
        Title   : set_parameter
        Usage   : $evolver-&gt;set_parameter($param,$val);
        Function: Sets a evolver parameter, will be validated against
                  the valid values as set in the %VALIDVALUES class variable.
                  The checks can be ignored if one turns off param checks like this:
                    $evolver-&gt;<a href="../man1/no_param_checks.1.html">no_param_checks</a>(1)
        Returns : boolean if set was success, if verbose is set to -1
                  then no warning will be reported
        Args    : $param =&gt; name of the parameter
                  $value =&gt; value to set the parameter to
        See also: L&lt;no_param_checks()&gt;

   <b>set_default_parameters</b>
        Title   : set_default_parameters
        Usage   : $evolver-&gt;<a href="../man0/set_default_parameters.0.html">set_default_parameters</a>(0);
        Function: (Re)set the default parameters from the defaults
                  (the first value in each array in the
               %VALIDVALUES class variable)
        Returns : none
        Args    : boolean: keep existing parameter values

   <b>set_CodonFreqs</b>
        Title   : set_CodonFreqs
        Usage   : $obj-&gt;set_CodonFreqs($newval)
        Function: Get/Set the Codon Frequence table
        Returns : value of set_CodonFreqs (a scalar)
        Args    : on set, new value (a scalar or undef, optional)

   <b>get_CodonFreqs</b>
        Title   : get_CodonFreqs
        Usage   : my @codon_freqs = $evolver-&gt;get_CodonFreqs()
        Function: Get the Codon freqs
        Returns : Array
        Args    : none

   <b>cleanup</b>
        Title   : cleanup
        Usage   : $evolver-&gt;cleanup();
        Function: Will cleanup the tempdir directory after a PAML run
        Returns : none
        Args    : none

   <b>io</b>
        Title   : io
        Usage   : $obj-&gt;io($newval)
        Function:  Gets a L&lt;Bio::Root::IO&gt; object
        Returns : L&lt;Bio::Root::IO&gt;
        Args    : none

</pre><h4><b>Bio::Tools::Run::WrapperBase</b> <b>methods</b></h4><pre>
</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>              - General discussion
         <a href="http://bioperl.org/Support.html">http://bioperl.org/Support.html</a>    - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list: <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bio-tools-phylo-paml/issues

</pre><h4><b>AUTHOR</b></h4><pre>
       Albert Vilella &lt;<a href="mailto:avilella@gmail.com">avilella@gmail.com</a>&gt;

</pre><h4><b>COPYRIGHT</b></h4><pre>
       This software is copyright (c) by Albert Vilella &lt;<a href="mailto:avilella@gmail.com">avilella@gmail.com</a>&gt;.

       This software is available under the same terms as the perl 5 programming language system itself.

perl v5.32.0                                       2020-11-10              <u>Bio::Tools::Ru...::PAML::<a href="../man3pm/Evolver.3pm.html">Evolver</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>