<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>cnvkit_reference - Compile a coverage reference from the given files (normal samples).</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/cnvkit">cnvkit_0.9.12-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       cnvkit_reference - Compile a coverage reference from the given files (normal samples).

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: cnvkit.py reference [-h] [-f FASTA] [-o FILENAME] [-c]

       [--min-cluster-size NUM]
              [-x  {m,y,male,Male,f,x,female,Female}]  [-y]  [-t TARGETS] [-a ANTITARGETS] [--no-gc] [--no-edge]
              [--no-rmask] [references ...]

   <b>positional</b> <b>arguments:</b>
       references
              Normal-sample target or antitarget .cnn files, or the directory that contains them.

   <b>options:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>-f</b> FASTA, <b>--fasta</b> FASTA
              Reference genome, FASTA format (e.g. UCSC hg19.fa)

       <b>-o</b> FILENAME, <b>--output</b> FILENAME
              Output file name.

       <b>-c</b>, <b>--cluster</b>
              Calculate and store summary stats for  clustered  subsets  of  the  normal  samples  with  similar
              coverage profiles.

       <b>--min-cluster-size</b> NUM
              Minimum cluster size to keep in reference profiles.  [Default: 4]

       <b>-x</b> {m,y,male,Male,f,x,female,Female}, <b>--sample-sex</b> {m,y,male,Male,f,x,female,Female}, <b>-g</b>
       {m,y,male,Male,f,x,female,Female}, <b>--gender</b> {m,y,male,Male,f,x,female,Female}
              Specify  the  chromosomal  sex of all given samples as male or female. (Default: guess each sample
              from coverage of X and Y chromosomes).

       <b>-y</b>, <b>--male-reference</b>, <b>--haploid-x-reference</b>
              Create a male reference: shift female samples' chrX log-coverage by  <b>-1</b>,  so  the  reference  chrX
              average is <b>-1</b>. Otherwise, shift male samples' chrX by +1, so the reference chrX average is 0.

   <b>To</b> <b>construct</b> <b>a</b> <b>generic,</b>  <b>flat"</b> <b>copy</b> <b>number</b> <b>reference</b> <b>with</b> <b>neutral</b> <b>expected</b> <b>coverage:"</b>
       <b>-t</b> TARGETS, <b>--targets</b> TARGETS
              Target intervals (.bed or .list)

       <b>-a</b> ANTITARGETS, <b>--antitargets</b> ANTITARGETS
              Antitarget intervals (.bed or .list)

   <b>To</b> <b>disable</b> <b>specific</b> <b>automatic</b> <b>bias</b> <b>corrections:</b>
       <b>--no-gc</b>
              Skip GC correction.

       <b>--no-edge</b>
              Skip edge-effect correction.

       <b>--no-rmask</b>
              Skip RepeatMasker correction.

cnvkit.py reference 0.9.10                          July 2023                                <u><a href="../man1/CNVKIT_REFERENCE.1.html">CNVKIT_REFERENCE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>