<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>razers3 - Faster, fully sensitive read mapping</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/seqan-apps">seqan-apps_2.4.0+dfsg-16_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       razers3 - Faster, fully sensitive read mapping

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>razers3</b> [<u>OPTIONS</u>] &lt;<u>GENOME</u> <u>FILE</u>&gt; &lt;<u>READS</u> <u>FILE</u>&gt;
       <b>razers3</b> [<u>OPTIONS</u>] &lt;<u>GENOME</u> <u>FILE</u>&gt; &lt;<u>PE-READS</u> <u>FILE1</u>&gt; &lt;<u>PE-READS</u> <u>FILE2</u>&gt;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       RazerS  3  is  a  versatile  full-sensitive  read  mapper based on k-mer counting and seeding filters. It
       supports single and paired-end mapping, shared-memory parallelism, and optimally parametrizes the  filter
       based   on   a   user-defined  minimal  sensitivity.  See  <u><a href="http://www.seqan.de/projects/razers">http://www.seqan.de/projects/razers</a></u>  for  more
       information.

       Input to RazerS 3 is a reference genome file and either one file  with  single-end  reads  or  two  files
       containing left or right mates of paired-end reads. Use - to read single-end reads from stdin.

       (c) Copyright 2009-2014 by David Weese.

</pre><h4><b>REQUIRED</b> <b>ARGUMENTS</b></h4><pre>
       <b>ARGUMENT</b> <b>0</b> <u>INPUT_FILE</u>
              A  reference  genome  file.  Valid filetypes are: <u>.sam[.*]</u>, <u>.raw[.*]</u>, <u>.gbk[.*]</u>, <u>.frn[.*]</u>, <u>.fq[.*]</u>,
              <u>.fna[.*]</u>, <u>.ffn[.*]</u>, <u>.fastq[.*]</u>, <u>.fasta[.*]</u>, <u>.faa[.*]</u>, <u>.fa[.*]</u>, <u>.embl[.*]</u>, and <u>.bam</u>, where * is any
              of the following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent (de)compression.

       <b>READS</b> List of <u>INPUT_FILE</u>'s
              Either one (single-end) or two (paired-end) read files. Valid filetypes are:  <u>.sam[.*]</u>,  <u>.raw[.*]</u>,
              <u>.gbk[.*]</u>,  <u>.frn[.*]</u>,  <u>.fq[.*]</u>,  <u>.fna[.*]</u>,  <u>.ffn[.*]</u>,  <u>.fastq[.*]</u>,  <u>.fasta[.*]</u>,  <u>.faa[.*]</u>, <u>.fa[.*]</u>,
              <u>.embl[.*]</u>, and <u>.bam</u>, where * is any of the following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent
              (de)compression.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              Display the help message.

       <b>--version</b>
              Display version information.

   <b>Main</b> <b>Options:</b>
       <b>-i</b>, <b>--percent-identity</b> <u>DOUBLE</u>
              Percent identity threshold. In range [50..100]. Default: <u>95</u>.

       <b>-rr</b>, <b>--recognition-rate</b> <u>DOUBLE</u>
              Percent recognition rate. In range [80..100]. Default: <u>100</u>.

       <b>-ng</b>, <b>--no-gaps</b>
              Allow only mismatches, no indels. Default: allow both.

       <b>-f</b>, <b>--forward</b>
              Map reads only to forward strands.

       <b>-r</b>, <b>--reverse</b>
              Map reads only to reverse strands.

       <b>-m</b>, <b>--max-hits</b> <u>INTEGER</u>
              Output only &lt;<u>NUM</u>&gt; of the best hits. In range [1..inf]. Default: <u>100</u>.

       <b>--unique</b>
              Output only unique best matches (-m 1 -dr 0 -pa).

       <b>-tr</b>, <b>--trim-reads</b> <u>INTEGER</u>
              Trim reads to given length. Default: off. In range [14..inf].

       <b>-o</b>, <b>--output</b> <u>OUTPUT_FILE</u>
              Mapping result filename (use - to dump to stdout in razers format). Default: &lt;<u>READS</u>  <u>FILE</u>&gt;.razers.
              Valid filetypes are: <u>.sam</u>, <u>.razers</u>, <u>.gff</u>, <u>.fasta</u>, <u>.fa</u>, <u>.eland</u>, <u>.bam</u>, and <u>.afg</u>.

       <b>-v</b>, <b>--verbose</b>
              Verbose mode.

       <b>-vv</b>, <b>--vverbose</b>
              Very verbose mode.

   <b>Paired-end</b> <b>Options:</b>
       <b>-ll</b>, <b>--library-length</b> <u>INTEGER</u>
              Paired-end library length. In range [1..inf]. Default: <u>220</u>.

       <b>-le</b>, <b>--library-error</b> <u>INTEGER</u>
              Paired-end library length tolerance. In range [0..inf]. Default: <u>50</u>.

   <b>Output</b> <b>Format</b> <b>Options:</b>
       <b>-a</b>, <b>--alignment</b>
              Dump the alignment for each match (only <u>razer</u> or <u>fasta</u> format).

       <b>-pa</b>, <b>--purge-ambiguous</b>
              Purge reads with more than &lt;<u>max-hits</u>&gt; best matches.

       <b>-dr</b>, <b>--distance-range</b> <u>INTEGER</u>
              Only consider matches with at most NUM more errors compared to the best. Default: output all.

       <b>-gn</b>, <b>--genome-naming</b> <u>INTEGER</u>
              Select how genomes are named (see Naming section below). In range [0..1]. Default: <u>0</u>.

       <b>-rn</b>, <b>--read-naming</b> <u>INTEGER</u>
              Select how reads are named (see Naming section below). In range [0..3]. Default: <u>0</u>.

       <b>--full-readid</b>
              Use the whole read id (don't clip after whitespace).

       <b>-so</b>, <b>--sort-order</b> <u>INTEGER</u>
              Select how matches are sorted (see Sorting section below). In range [0..1]. Default: <u>0</u>.

       <b>-pf</b>, <b>--position-format</b> <u>INTEGER</u>
              Select begin/end position numbering (see Coordinate section below). In range [0..1]. Default: <u>0</u>.

       <b>-ds</b>, <b>--dont-shrink-alignments</b>
              Disable alignment shrinking in SAM.  This is required for generating a gold mapping for Rabema.

   <b>Filtration</b> <b>Options:</b>
       <b>-fl</b>, <b>--filter</b> <u>STRING</u>
              Select k-mer filter. One of <u>pigeonhole</u> and <u>swift</u>. Default: <u>pigeonhole</u>.

       <b>-mr</b>, <b>--mutation-rate</b> <u>DOUBLE</u>
              Set the percent mutation rate (<u>pigeonhole</u>). In range [0..20]. Default: <u>5</u>.

       <b>-ol</b>, <b>--overlap-length</b> <u>INTEGER</u>
              Manually set the overlap length of adjacent k-mers (<u>pigeonhole</u>). In range [0..inf].

       <b>-pd</b>, <b>--param-dir</b> <u>STRING</u>
              Read user-computed parameter files in the directory &lt;<u>DIR</u>&gt; (<u>swift</u>).

       <b>-t</b>, <b>--threshold</b> <u>INTEGER</u>
              Manually set minimum k-mer count threshold (<u>swift</u>). In range [1..inf].

       <b>-tl</b>, <b>--taboo-length</b> <u>INTEGER</u>
              Set taboo length (<u>swift</u>). In range [1..inf]. Default: <u>1</u>.

       <b>-s</b>, <b>--shape</b> <u>STRING</u>
              Manually set k-mer shape.

       <b>-oc</b>, <b>--overabundance-cut</b> <u>INTEGER</u>
              Set k-mer overabundance cut ratio. In range [0..1]. Default: <u>1</u>.

       <b>-rl</b>, <b>--repeat-length</b> <u>INTEGER</u>
              Skip simple-repeats of length &lt;<u>NUM</u>&gt;. In range [1..inf]. Default: <u>1000</u>.

       <b>-lf</b>, <b>--load-factor</b> <u>DOUBLE</u>
              Set the load factor for the open addressing k-mer index. In range [1..inf]. Default: <u>1.6</u>.

   <b>Verification</b> <b>Options:</b>
       <b>-mN</b>, <b>--match-N</b>
              N matches all other characters. Default: N matches nothing.

       <b>-ed</b>, <b>--error-distr</b> <u>STRING</u>
              Write error distribution to <u>FILE</u>.

       <b>-mf</b>, <b>--mismatch-file</b> <u>STRING</u>
              Write mismatch patterns to <u>FILE</u>.

   <b>Misc</b> <b>Options:</b>
       <b>-cm</b>, <b>--compact-mult</b> <u>DOUBLE</u>
              Multiply  compaction  threshold  by  this  value after reaching and compacting. In range [0..inf].
              Default: <u>2.2</u>.

       <b>-ncf</b>, <b>--no-compact-frac</b> <u>DOUBLE</u>
              Don't compact if in this last fraction of genome. In range [0..1]. Default: <u>0.05</u>.

   <b>Parallelism</b> <b>Options:</b>
       <b>-tc</b>, <b>--thread-count</b> <u>INTEGER</u>
              Set the number of threads to use (0 to force sequential mode). In range [0..inf]. Default: <u>1</u>.

       <b>-pws</b>, <b>--parallel-window-size</b> <u>INTEGER</u>
              Collect candidates in windows of this length. In range [1..inf]. Default: <u>500000</u>.

       <b>-pvs</b>, <b>--parallel-verification-size</b> <u>INTEGER</u>
              Verify candidates in packages of this size. In range [1..inf]. Default: <u>100</u>.

       <b>-pvmpc</b>, <b>--parallel-verification-max-package-count</b> <u>INTEGER</u>
              Largest number of packages to create for verification per thread-1. In  range  [1..inf].  Default:
              <u>100</u>.

       <b>-amms</b>, <b>--available-matches-memory-size</b> <u>INTEGER</u>
              Bytes of main memory available for storing matches. In range [-1..inf]. Default: <u>0</u>.

       <b>-mhst</b>, <b>--match-histo-start-threshold</b> <u>INTEGER</u>
              When to start histogram. In range [1..inf]. Default: <u>5</u>.

</pre><h4><b>FORMATS,</b> <b>NAMING,</b> <b>SORTING,</b> <b>AND</b> <b>COORDINATE</b> <b>SCHEMES</b></h4><pre>
       RazerS  3 supports various output formats. The output format is detected automatically from the file name
       suffix.

       .razers
              Razer format

       .fa, .fasta
              Enhanced Fasta format

       .eland Eland format

       .gff   GFF format

       .sam   SAM format

       .bam   BAM format

       .afg   Amos AFG format

              By default, reads and contigs are referred by their Fasta ids given in the input files.  With  the
              <b>-gn</b> and <b>-rn</b> options this behaviour can be changed:

       0      Use Fasta id.

       1      Enumerate beginning with 1.

       2      Use the read sequence (only for short reads!).

       3      Use the Fasta id, do NOT append /L or /R for mate pairs.

              The way matches are sorted in the output file can be changed with the <b>-so</b> option for the following
              formats: <b>razers</b>, <b>fasta</b>, <b>sam</b>, and <b>afg</b>. Primary and secondary sort keys are:

       0      1. read number, 2. genome position

       1      1. genome position, 2. read number

              The  coordinate  space used for begin and end positions can be changed with the <b>-pf</b> option for the
              <b>razer</b> and <b>fasta</b> formats:

       0      Gap space. Gaps between characters are counted from 0.

       1      Position space. Characters are counted from 1.

</pre><h4><b>EXAMPLES</b></h4><pre>
       <b>razers3</b> <b>-i</b> <b>96</b> <b>-tc</b> <b>12</b> <b>-o</b> <b>mapped.razers</b> <b>hg18.fa</b> <b>reads.fq</b>
              Map single-end reads with 4% error rate using 12 threads.

       <b>razers3</b> <b>-i</b> <b>95</b> <b>-no-gaps</b> <b>-o</b> <b>mapped.razers</b> <b>hg18.fa</b> <b>reads.fq.gz</b>
              Map single-end gzipped reads with 5% error rate and no indels.

       <b>razers3</b> <b>-i</b> <b>94</b> <b>-rr</b> <b>95</b> <b>-tc</b> <b>12</b> <b>-ll</b> <b>280</b> <b>--le</b> <b>80</b> <b>-o</b> <b>mapped.razers</b> <b>hg18.fa</b> <b>reads_1.fq</b> <b>reads_2.fq</b>
              Map paired-end reads with up to 6% errors, 95% sensitivity, 12 threads, and  only  output  aligned
              pairs with an outer distance of 200-360bp.

razers3 3.5.8 [tarball]                                                                               <u><a href="../man1/RAZERS3.1.html">RAZERS3</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>