<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>gth - predict genome structures</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/genomethreader">genomethreader_1.7.3+dfsg-10build2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       gth - predict genome structures

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>gth</b> [option ...] -genomic file [...] -cdna file [...] -protein file [...]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Computes similarity-based gene structure predictions (spliced alignments) using cDNA/EST and/or protein
       sequences and assemble the resulting spliced alignments to consensus spliced alignments.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-genomic</b> &lt;file&gt;
           specify input files containing genomic sequences (mandatory option)

       <b>-cdna</b> &lt;file&gt;
           specify input files containing cDNA/EST sequences

       <b>-protein</b> &lt;file&gt;
           specify input files containing protein sequences

       <b>-species</b> &lt;species&gt;
           specify species to select splice site model which is most appropriate; possible species: "human"
           "mouse" "rat" "chicken" "drosophila" "nematode" "fission_yeast" "aspergillus" "arabidopsis" "maize"
           "rice" "medicago" default: undefined

       <b>-bssm</b>
           read bssm parameter from file in the path given by the environment variable BSSMDIR, default:
           undefined

       <b>-scorematrix</b>
           read amino acid substitution scoring matrix from file in the path given by the environment variable
           GTHDATADIR default: BLOSUM62

       <b>-translationtable</b>
           set the codon translation table used for codon translation in matching, DP, and output default: 1

       <b>-f</b>
           analyze only forward strand of genomic sequences default: no

       <b>-r</b>
           analyze only reverse strand of genomic sequences default: no

       <b>-cdnaforward</b>
           align only forward strand of cDNAs default: no

       <b>-frompos</b>
           analyze genomic sequence from this position requires -topos or -width; counting from 1 on default: 0

       <b>-topos</b>
           analyze genomic sequence to this position requires -frompos; counting from 1 on default: 0

       <b>-width</b>
           analyze only this width of genomic sequence requires -frompos default: 0

       <b>-v</b>
           be verbose default: no

       <b>-xmlout</b>
           show output in XML format default: no

       <b>-gff3out</b>
           show output in GFF3 format default: no

       <b>-md5ids</b>
           show MD5 fingerprints as sequence IDs default: no

       <b>-o</b>
           redirect output to specified file default: undefined

       <b>-gzip</b>
           write gzip compressed output file default: no

       <b>-bzip2</b>
           write bzip2 compressed output file default: no

       <b>-force</b>
           force writing to output file default: no

       <b>-skipalignmentout</b>
           skip output of spliced alignments default: no

       <b>-mincutoffs</b>
           show full spliced alignments i.e., cutoffs mode for leading and terminal bases is MINIMAL default: no

       <b>-showintronmaxlen</b>
           set the maximum length of a fully shown intron If set to 0, all introns are shown completely default:
           120

       <b>-minorflen</b>
           set the minimum length of an ORF to be shown default: 64

       <b>-startcodon</b>
           require than an ORF must begin with a start codon default: no

       <b>-finalstopcodon</b>
           require that the final ORF must end with a stop codon default: no

       <b>-showseqnums</b>
           show sequence numbers in output default: no

       <b>-pglgentemplate</b>
           show genomic template in PGL lines (switch off for backward compatibility) default: yes

       <b>-gs2out</b>
           output in old GeneSeqer2 format default: no

       <b>-maskpolyatails</b>
           mask poly(A) tails in cDNA/EST files default: no

       <b>-proteinsmap</b>
           specify smap file used for protein files default: protein

       <b>-noautoindex</b>
           do not create indices automatically except for the .dna.* files used for the DP. existence is not
           tested before an index is actually used! default: no

       <b>-createindicesonly</b>
           stop program flow after the indices have been created default: no

       <b>-skipindexcheck</b>
           skip index check (in preprocessing phase) default: no

       <b>-minmatchlen</b>
           specify minimum match length (cDNA matching) default: 20

       <b>-seedlength</b>
           specify the seed length (cDNA matching) default: 18

       <b>-exdrop</b>
           specify the Xdrop value for edit distance extension (cDNA matching) default: 2

       <b>-prminmatchlen</b>
           specify minimum match length (protein matches) default: 24

       <b>-prseedlength</b>
           specify seed length (protein matching) default: 10

       <b>-prhdist</b>
           specify Hamming distance (protein matching) default: 4

       <b>-online</b>
           run the similarity filter online without using the complete index (increases runtime) default: no

       <b>-inverse</b>
           invert query and index in vmatch call default: no

       <b>-exact</b>
           use exact matches in the similarity filter default: no

       <b>-gcmaxgapwidth</b>
           set the maximum gap width for global chains defines approximately the maximum intron length set to 0
           to allow for unlimited length in order to avoid false-positive exons (lonely exons) at the sequence
           ends, it is very important to set this parameter appropriately! default: 1000000

       <b>-gcmincoverage</b>
           set the minimum coverage of global chains regarding to the reference sequence default: 50

       <b>-paralogs</b>
           compute paralogous genes (different chaining procedure) default: no

       <b>-enrichchains</b>
           enrich genomic sequence part of global chains with additional matches default: no

       <b>-introncutout</b>
           enable the intron cutout technique default: no

       <b>-fastdp</b>
           use jump table to increase speed of DP calculation default: no

       <b>-autointroncutout</b>
           set the automatic intron cutout matrix size in megabytes and enable the automatic intron cutout
           technique default: 0

       <b>-icinitialdelta</b>
           set the initial delta used for intron cutouts default: 50

       <b>-iciterations</b>
           set the number of intron cutout iterations default: 2

       <b>-icdeltaincrease</b>
           set the delta increase during every iteration default: 50

       <b>-icminremintronlen</b>
           set the minimum remaining intron length for an intron to be cut out default: 10

       <b>-nou12intronmodel</b>
           disable the U12-type intron model default: no

       <b>-u12donorprob</b>
           set the probability for perfect U12-type donor sites default: 0.99

       <b>-u12donorprob1mism</b>
           set the prob. for U12-type donor w. 1 mismatch default: 0.90

       <b>-probies</b>
           set the initial exon state probability default: 0.50

       <b>-probdelgen</b>
           set the genomic sequence deletion probability default: 0.03

       <b>-identityweight</b>
           set the pairs of identical characters weight default: 2.00

       <b>-mismatchweight</b>
           set the weight for mismatching characters default: -2.00

       <b>-undetcharweight</b>
           set the weight for undetermined characters default: 0.00

       <b>-deletionweight</b>
           set the weight for deletions default: -5.00

       <b>-dpminexonlen</b>
           set the minimum exon length for the DP default: 5

       <b>-dpminintronlen</b>
           set the minimum intron length for the DP default: 50

       <b>-shortexonpenal</b>
           set the short exon penalty default: 100.00

       <b>-shortintronpenal</b>
           set the short intron penalty default: 100.00

       <b>-wzerotransition</b>
           set the zero transition weights window size default: 80

       <b>-wdecreasedoutput</b>
           set the decreased output weights window size default: 80

       <b>-leadcutoffsmode</b>
           set the cutoffs mode for leading bases can be either RELAXED, STRICT, or MINIMAL default: RELAXED

       <b>-termcutoffsmode</b>
           set the cutoffs mode for terminal bases can be either RELAXED, STRICT, or MINIMAL default: STRICT

       <b>-cutoffsminexonlen</b>
           set the cutoffs minimum exon length default: 5

       <b>-scoreminexonlen</b>
           set the score minimum exon length default: 50

       <b>-minaveragessp</b>
           set the minimum average splice site prob. default: 0.50

       <b>-duplicatecheck</b>
           criterion used to check for spliced alignment duplicates, choose from none|id|desc|seq|both default:
           both

       <b>-minalignmentscore</b>
           set the minimum alignment score for spliced alignments to be included into the set of spliced
           alignments default: 0.00

       <b>-maxalignmentscore</b>
           set the maximum alignment score for spliced alignments to be included into the set of spliced
           alignments default: 1.00

       <b>-mincoverage</b>
           set the minimum coverage for spliced alignments to be included into the set of spliced alignments
           default: 0.00

       <b>-maxcoverage</b>
           set the maximum coverage for spliced alignments to be included into the set of spliced alignments
           default: 9999.99

       <b>-intermediate</b>
           stop after calculation of spliced alignments and output results in reusable XML format. Do not
           process this output yourself, use the ``normal'' XML output instead! default: no

       <b>-sortags</b>
           sort alternative gene structures according to the weighted mean of the average exon score and the
           average splice site probability default: no

       <b>-sortagswf</b>
           set the weight factor for the sorting of AGSs default: 1.00

       <b>-exondistri</b>
           show the exon length distribution default: no

       <b>-introndistri</b>
           show the intron length distribution default: no

       <b>-refseqcovdistri</b>
           show the reference sequence coverage distribution default: no

       <b>-first</b>
           set the maximum number of spliced alignments per genomic DNA input. Set to 0 for unlimited number.
           default: 0

       <b>-help</b>
           display help for basic options and exit

       <b>-help+</b>
           display help for all options and exit

       <b>-version</b>
           display version information and exit

                                                                                                          <u><a href="../man1/GTH.1.html">GTH</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>