<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>QTLtools extract - Data extraction mode.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/qtltools">qtltools_1.3.1+dfsg-4build4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       QTLtools extract - Data extraction mode.

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>QTLtools</b> <b>extract</b> <b>[--vcf</b> <b>--bed</b> <b>--cov]</b> <u>relevant_file</u> <b>--out</b> <u>output_prefix</u> <u>[OPTIONS]</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Extract  all  the data from the provided files into one flat file.  This will also impute all the missing
       data.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>--vcf</b> <b>[</b><u>in.vcf</u>|<u>in.bcf</u>|<u>in.vcf.gz</u><b>]</b>
              Genotypes in VCF/BCF format.  At least one of <b>--vcf</b> <b>--bed</b> <b>--cov</b> is REQUIRED.

       <b>--bed</b> <u>quantifications.bed.gz</u>
              Molecular phenotype quantifications in BED format.  At least one of <b>--vcf</b> <b>--bed</b> <b>--cov</b> is REQUIRED.

       <b>--cov</b> <u>covariates.txt</u>
              Covariates data.  At least one of <b>--vcf</b> <b>--bed</b> <b>--cov</b> is REQUIRED.

       <b>--out</b> <u>filename_prefix</u>
              Output file name prefix.  REQUIRED.

       <b>--region</b> <u>chr:start-end</u>
              Genomic region to be processed.  E.g. chr4:12334456-16334456, or chr5.

</pre><h4><b>OUTPUT</b> <b>FILES</b></h4><pre>
       <b>.content.txt.gz</b>
        Flat file with samples in rows and all the extracted variables in columns.

       <b>.header.txt</b>
        Genomic locations of the extracted variables.

</pre><h4><b>EXAMPLE</b></h4><pre>
       o Extract genotypes, gene quantifications, and covariates from a certain region:

         QTLtools   extract    --vcf    genotypes.chr22.vcf.gz    --bed    genes.50percent.chr22.bed.gz    --cov
         genes.covariates.pc50.txt.gz --region chr22:17000000-18000000 --out extract_output

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/QTLtools.1.html">QTLtools</a></u>(1)

       QTLtools website: &lt;https://qtltools.github.io/qtltools&gt;

</pre><h4><b>BUGS</b></h4><pre>
       o Versions  up  to  and  including  1.2, suffer from a bug in reading missing genotypes in VCF/BCF files.
         This bug affects variants with a DS field in their genotype's FORMAT and have a  missing  genotype  (DS
         field  is  .)  in  one  of the samples, in which case genotypes for all the samples are set to missing,
         effectively removing this variant from the analyses.

       Please submit bugs to &lt;https://github.com/qtltools/qtltools&gt;

</pre><h4><b>CITATION</b></h4><pre>
       Delaneau, O., Ongen, H., Brown, A. et al. A complete tool set for molecular QTL discovery  and  analysis.
       <u>Nat</u> <u>Commun</u> <b>8</b>, 15452 (2017).  &lt;https://doi.org/10.1038/ncomms15452&gt;

</pre><h4><b>AUTHORS</b></h4><pre>
       Olivier Delaneau (<a href="mailto:olivier.delaneau@gmail.com">olivier.delaneau@gmail.com</a>), Halit Ongen (<a href="mailto:halitongen@gmail.com">halitongen@gmail.com</a>)

QTLtools-v1.3                                      06 May 2020                               <u><a href="../man1/QTLtools-extract.1.html">QTLtools-extract</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>