<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>dless - Attempts to identify elements under selection in all species or in</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/phast">phast_1.7+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       dless - Attempts to identify elements under selection in all species or in

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Attempts  to  identify  elements  under selection in all species or in some subset of species, based on a
       multiple alignment and a phylo-HMM.  In particular, detects elements that have been under selection since
       the divergence of all species in the given set, that were "born" on some branch of the tree  since  their
       divergence  and  have  been under selection since, or that were present in the common ancestor but "died"
       (ceased to be under selection) on some branch of the tree.  Currently only  detects  negative  selection,
       but extensions to detect positive selection as well are planned.

</pre><h4><b>EXAMPLE</b></h4><pre>
</pre><h4><b>OPTIONS</b></h4><pre>

       <b>--rho</b>, <b>-R</b> &lt;rho&gt;

              (default 0.3)

       <b>--transitions</b>, <b>-t</b> [~]&lt;mu&gt;,&lt;nu&gt;

              Set  the transition probabilities of the two-state HMM using the specified values of &lt;mu&gt; and &lt;nu&gt;
              (both between 0 and 1).

       <b>--phi</b>, <b>-p</b> [~]&lt;phi&gt;

              (default 0.5)

       <b>--target-coverage</b>, <b>-C</b> [~]&lt;gamma&gt;

              (Alternative to transitions, use with <b>--expected-length</b>) Set the transition parameters  such  that
              the  expected  fraction  of  sites  in  conserved elements is &lt;gamma&gt; (betwen 0 and 1).  This is a
              *prior* rather than *posterior* expectation  and  assumes  stationarity  of  the  state-transition
              process.   This  option  causes  the ratio mu/nu to be fixed at (1-gamma)/gamma, and together with
              <b>--expected-length</b>, completely defines the transition probabilities.

       <b>--expected-length</b>, <b>-E</b> [~]&lt;omega&gt;

              (Alternative to <b>--transitions</b>, use with <b>--target-coverage</b>) Set transition probabilities such  that
              the  (prior)  expected  length  of  a  conserved  element  is &lt;omega&gt;.  The parameter mu is set to
              1/omega.

       <b>--msa-format</b>, <b>-i</b> FASTA|PHYLIP|MPM|MAF|SS

              Alignment format (default is to guess format from file contents).  Note that the program  msa_view
              can be used for conversion.

       <b>--refseq</b>, <b>-M</b> &lt;fname&gt;

              (for  use  with  <b>--msa-format</b>  MAF)  Read the complete text of the reference sequence from &lt;fname&gt;
              (FASTA format) and combine it with the contents of the MAF file to  produce  a  complete,  ordered
              representation  of the alignment.  The reference sequence of the MAF file is assumed to be the one
              that appears first in each block.

       <b>--refidx</b>, <b>-r</b> &lt;refseq_idx&gt; Use coordinate frame of specified sequence in output.  Default

              value is 1, first  sequence  in  alignment;  0  indicates  coordinate  frame  of  entire  multiple
              alignment.

       <b>--seqname</b>,  <b>-N</b>  &lt;name&gt; Use specified string for 'seqname' (GFF) or 'chrom' field in output file.  Default
              is obtained from  input  file  name  (double  filename  root,  e.g.,  "chr22"  if  input  file  is
              "chr22.35.ss").

       <b>--idpref</b>,  <b>-P</b>  &lt;name&gt;  Use  specified  string  as prefix of generated ids in output file.  Can be used to
              ensure ids are unique.  Default is obtained from input file  name  (single  filename  root,  e.g.,
              "chr22.35" if input file is "chr22.35.ss").

       <b>--indel-model</b>, <b>-I</b> alpha,beta,tau[,alpha2,beta2,tau2]

              Use  a simple model of insertions and deletions that assumes a known indel history and at most one
              indel per branch of the tree at any given position.  The parameters alpha and beta  are  rates  of
              insertion and deletion, respectively, per expected substitution per site, and the parameter tau is
              approximately the inverse of the expected indel length (see indelFit).  If two sets are parameters
              are  given  the  first will be used for nonconserved regions and the second for conserved regions.
              If <b>--indel-history</b> is not used, a history will be inferred on the fly  using  a  simple  parsimony
              algorithm.

       <b>--indel-history</b>, <b>-H</b> &lt;file.ih&gt;

              (for use with <b>--indel-model</b>) Use the specified indel history (see indelHistory).

       <b>--help</b>, <b>-h</b>

              Show this help message and exit.

dless 1.4                                           May 2016                                            <u><a href="../man1/DLESS.1.html">DLESS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>