<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>bwa - Burrows-Wheeler Alignment Tool</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/bwa">bwa_0.7.18-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       bwa - Burrows-Wheeler Alignment Tool

</pre><h4><b>SYNOPSIS</b></h4><pre>
       bwa index ref.fa

       bwa mem ref.fa reads.fq &gt; aln-se.sam

       bwa mem ref.fa read1.fq read2.fq &gt; aln-pe.sam

       bwa aln ref.fa short_read.fq &gt; aln_sa.sai

       bwa samse ref.fa aln_sa.sai short_read.fq &gt; aln-se.sam

       bwa sampe ref.fa aln_sa1.sai aln_sa2.sai read1.fq read2.fq &gt; aln-pe.sam

       bwa bwasw ref.fa long_read.fq &gt; aln.sam

</pre><h4><b>DESCRIPTION</b></h4><pre>
       BWA  is  a software package for mapping low-divergent sequences against a large reference genome, such as
       the human genome. It consists of three algorithms: BWA-backtrack, BWA-SW and BWA-MEM. The first algorithm
       is designed for Illumina sequence reads up to 100bp, while the rest two for longer sequences ranged  from
       70bp  to  1Mbp.  BWA-MEM and BWA-SW share similar features such as long-read support and split alignment,
       but BWA-MEM, which is the latest, is generally recommended for high-quality queries as it is  faster  and
       more accurate.  BWA-MEM also has better performance than BWA-backtrack for 70-100bp Illumina reads.

       For  all  the  algorithms,  BWA first needs to construct the FM-index for the reference genome (the <b>index</b>
       command). Alignment  algorithms  are  invoked  with  different  sub-commands:  <b>aln</b>/<b>samse</b>/<b>sampe</b>  for  BWA-
       backtrack, <b>bwasw</b> for BWA-SW and <b>mem</b> for the BWA-MEM algorithm.

</pre><h4><b>COMMANDS</b> <b>AND</b> <b>OPTIONS</b></h4><pre>
       <b>index</b>  <b>bwa</b> <b>index</b> [<b>-p</b> <u>prefix</u>] [<b>-a</b> <u>algoType</u>] <u>db.fa</u>

              Index database sequences in the FASTA format.

              <b>OPTIONS:</b>

              <b>-p</b> <u>STR</u>    Prefix of the output database [same as db filename]

              <b>-a</b> <u>STR</u>    Algorithm   for  constructing  BWT  index.  BWA  implements  three  algorithms  for  BWT
                        construction: <b>is</b>, <b>bwtsw</b> and <b>rb2</b>.  The first algorithm  is  a  little  faster  for  small
                        database but requires large RAM and does not work for databases with total length longer
                        than  2GB.  The  second  algorithm  is adapted from the BWT-SW source code. It in theory
                        works with database with trillions of bases. When this  option  is  not  specified,  the
                        appropriate algorithm will be chosen automatically.

       <b>mem</b>    <b>bwa</b>  <b>mem</b> [<b>-aCHjMpP</b>] [<b>-t</b> <u>nThreads</u>] [<b>-k</b> <u>minSeedLen</u>] [<b>-w</b> <u>bandWidth</u>] [<b>-d</b> <u>zDropoff</u>] [<b>-r</b> <u>seedSplitRatio</u>]
              [<b>-c</b> <u>maxOcc</u>] [<b>-D</b> <u>chainShadow</u>] [<b>-m</b> <u>maxMateSW</u>] [<b>-W</b> <u>minSeedMatch</u>] [<b>-A</b> <u>matchScore</u>] [<b>-B</b>  <u>mmPenalty</u>]  [<b>-O</b>
              <u>gapOpenPen</u>]  [<b>-E</b> <u>gapExtPen</u>] [<b>-L</b> <u>clipPen</u>] [<b>-U</b> <u>unpairPen</u>] [<b>-x</b> <u>readType</u>] [<b>-R</b> <u>RGline</u>] [<b>-H</b> <u>HDlines</u>] [<b>-v</b>
              <u>verboseLevel</u>] <u>db.prefix</u> <u>reads.fq</u> [<u>mates.fq</u>]

              Align 70bp-1Mbp query sequences with the  BWA-MEM  algorithm.  Briefly,  the  algorithm  works  by
              seeding  alignments with maximal exact matches (MEMs) and then extending seeds with the affine-gap
              Smith-Waterman algorithm (SW).

              If <u>mates.fq</u> file is absent and option <b>-p</b> is not set, this command regards input reads are  single-
              end.  If  <u>mates.fq</u> is present, this command assumes the <u>i</u>-th read in <u>reads.fq</u> and the <u>i</u>-th read in
              <u>mates.fq</u> constitute a read pair. If <b>-p</b> is used, the command assumes the 2<u>i</u>-th  and  the  (2<u>i</u>+1)-th
              read in <u>reads.fq</u> constitute a read pair (such input file is said to be interleaved). In this case,
              <u>mates.fq</u>  is  ignored. In the paired-end mode, the <b>mem</b> command will infer the read orientation and
              the insert size distribution from a batch of reads.

              The BWA-MEM algorithm performs local alignment. It may produce  multiple  primary  alignments  for
              different  part  of  a query sequence. This is a crucial feature for long sequences. However, some
              tools may not work with split alignments.  One may consider to use option <b>-M</b> to flag shorter split
              hits as secondary.

              <b>ALGORITHM</b> <b>OPTIONS:</b>

              <b>-t</b> <u>INT</u>    Number of threads [1]

              <b>-k</b> <u>INT</u>    Minimum seed length. Matches shorter than <u>INT</u> will be missed.  The  alignment  speed  is
                        usually insensitive to this value unless it significantly deviates from 20. [19]

              <b>-w</b> <u>INT</u>    Band  width.  Essentially, gaps longer than <u>INT</u> will not be found. Note that the maximum
                        gap length is also affected by the  scoring  matrix  and  the  hit  length,  not  solely
                        determined by this option. [100]

              <b>-d</b> <u>INT</u>    Off-diagonal  X-dropoff (Z-dropoff). Stop extension when the difference between the best
                        and the current extension score is above |<u>i</u>-<u>j</u>|*<u>A</u>+<u>INT</u>, where <u>i</u>  and  <u>j</u>  are  the  current
                        positions  of  the  query  and  reference, respectively, and <u>A</u> is the matching score. Z-
                        dropoff is similar to BLAST's X-dropoff except that it doesn't penalize gaps in  one  of
                        the  sequences  in  the  alignment. Z-dropoff not only avoids unnecessary extension, but
                        also reduces poor alignments inside a long good alignment. [100]

              <b>-r</b> <u>FLOAT</u>  Trigger re-seeding for a MEM longer than <u>minSeedLen</u>*<u>FLOAT</u>.   This  is  a  key  heuristic
                        parameter  for  tuning  the performance. Larger value yields fewer seeds, which leads to
                        faster alignment speed but lower accuracy. [1.5]

              <b>-c</b> <u>INT</u>    Discard a MEM if it has more than <u>INT</u> occurrence in the genome. This is  an  insensitive
                        parameter. [500]

              <b>-D</b> <u>FLOAT</u>  Drop chains shorter than <u>FLOAT</u> fraction of the longest overlapping chain [0.5]

              <b>-m</b> <u>INT</u>    Perform at most <u>INT</u> rounds of mate-SW [50]

              <b>-W</b> <u>INT</u>    Drop  a  chain  if  the  number  of  bases in seeds is smaller than <u>INT</u>.  This option is
                        primarily used for longer contigs/reads. When positive, it also affects seed  filtering.
                        [0]

              <b>-P</b>        In  the  paired-end  mode, perform SW to rescue missing hits only but do not try to find
                        hits that fit a proper pair.

              <b>SCORING</b> <b>OPTIONS:</b>

              <b>-A</b> <u>INT</u>    Matching score. [1]

              <b>-B</b> <u>INT</u>    Mismatch penalty. The sequence error rate is approximately: {.75 * exp[-<a href="../man4/log.4.html">log</a>(4) *  B/A]}.
                        [4]

              <b>-O</b> <u>INT[,INT]</u>
                        Gap  open  penalty.  If two numbers are specified, the first is the penalty of opening a
                        deletion and the second for opening an insertion. [6]

              <b>-E</b> <u>INT[,INT]</u>
                        Gap extension penalty. If two numbers  are  specified,  the  first  is  the  penalty  of
                        extending  a deletion and second for extending an insertion. A gap of length k costs O +
                        k*E (i.e.  <b>-O</b> is for opening a zero-length gap). [1]

              <b>-L</b> <u>INT[,INT]</u>
                        Clipping penalty. When performing SW extension, BWA-MEM keeps track of  the  best  score
                        reaching  the  end  of  query.  If this score is larger than the best SW score minus the
                        clipping penalty, clipping will not be applied. Note that in this case, the SAM  AS  tag
                        reports the best SW score; clipping penalty is not deduced. If two numbers are provided,
                        the first is for 5'-end clipping and second for 3'-end clipping. [5]

              <b>-U</b> <u>INT</u>    Penalty   for   an  unpaired  read  pair.  BWA-MEM  scores  an  unpaired  read  pair  as
                        scoreRead1+scoreRead2-<u>INT</u> and scores a paired as scoreRead1+scoreRead2-insertPenalty. It
                        compares these two scores to determine whether we should force pairing. A  larger  value
                        leads to more aggressive read pair. [17]

              <b>-x</b> <u>STR</u>    Read type. Changes multiple parameters unless overridden [null]

                        <b>pacbio</b>:   <b>-k17</b> <b>-W40</b> <b>-r10</b> <b>-A1</b> <b>-B1</b> <b>-O1</b> <b>-E1</b> <b>-L0</b> (PacBio reads to ref)

                        <b>ont2d</b>:    <b>-k14</b> <b>-W20</b> <b>-r10</b> <b>-A1</b> <b>-B1</b> <b>-O1</b> <b>-E1</b> <b>-L0</b> (Oxford Nanopore 2D-reads to ref)

                        <b>intractg</b>: <b>-B9</b> <b>-O16</b> <b>-L5</b> (intra-species contigs to ref)

              <b>INPUT/OUTPUT</b> <b>OPTIONS:</b>

              <b>-p</b>        Smart  pairing.  If two adjacent reads have the same name, they are considered to form a
                        read pair. This way, paired-end and single-end reads can be mixed in  a  single  FASTA/Q
                        stream.

              <b>-R</b> <u>STR</u>    Complete  read group header line. '\t' can be used in <u>STR</u> and will be converted to a TAB
                        in the output SAM. The read group ID will be attached to every read in  the  output.  An
                        example is '@RG\tID:foo\tSM:bar'.  [null]

              <b>-H</b> <u>ARG</u>    If  ARG  starts  with @, it is interpreted as a string and gets inserted into the output
                        SAM header; otherwise, ARG is interpreted as a file with all lines starting  with  @  in
                        the file inserted into the SAM header. [null]

              <b>-o</b> <u>FILE</u>   Write  the  output  SAM  file  to <u>FILE</u>.  For compatibility with other BWA commands, this
                        option may also be given as <b>-f</b> <u>FILE</u>.  [standard output]

              <b>-q</b>
                         Don't reduce the mapping quality of split alignment of lower alignment score.

              <b>-5</b>        For split alignment, mark the segment with the smallest coordinate as  the  primary.  It
                        automatically  applies  option  <b>-q</b> as well. This option may help some Hi-C pipelines. By
                        default, BWA-MEM marks highest scoring segment as primary.

              <b>-K</b>  <b>INT</b>   Process <u>INT</u> input bases in each batch  regardless  of  the  number  of  threads  in  use
                        [10000000*<u>nThreads</u>].   By  default,  the  batch  size  is  proportional to the number of
                        threads in use.  Because the inferred insert size distribution slightly depends  on  the
                        batch  size, using different number of threads may produce different output.  Specifying
                        this option helps reproducibility.

              <b>-T</b> <u>INT</u>    Don't output alignment with score lower  than  <u>INT</u>.   This  option  affects  output  and
                        occasionally SAM flag 2. [30]

              <b>-j</b>        Treat ALT contigs as part of the primary assembly (i.e. ignore the <u>db.prefix.alt</u> file).

              <b>-h</b> <u>INT[,INT2]</u>
                        If a query has not more than <u>INT</u> hits with score higher than 80% of the best hit, output
                        them  all  in  the XA tag.  If <u>INT2</u> is specified, BWA-MEM outputs up to <u>INT2</u> hits if the
                        list contains a hit to an ALT contig. [5,200]

              <b>-a</b>        Output  all  found  alignments  for  single-end  or  unpaired  paired-end  reads.  These
                        alignments will be flagged as secondary alignments.

              <b>-C</b>        Append  FASTA/Q  comment  to  SAM  output. This option can be used to transfer read meta
                        information (e.g. barcode) to the SAM output. Note that the FASTA/Q comment (the  string
                        after  a  space  in  the  header  line)  must  conform  the  SAM spec (e.g. BC:Z:CGTAC).
                        Malformated comments lead to incorrect SAM output.

              <b>-Y</b>        Use soft clipping CIGAR operation for supplementary alignments. By default, BWA-MEM uses
                        soft clipping for the primary alignment and hard clipping for supplementary alignments.

              <b>-M</b>        Mark shorter split hits as secondary (for Picard compatibility).

              <b>-v</b> <u>INT</u>    Control the verbosity level of the output. This option  has  not  been  fully  supported
                        throughout  BWA.  Ideally,  a  value  0  for  disabling  all the output to stderr; 1 for
                        outputting errors only; 2 for warnings and errors; 3  for  all  normal  messages;  4  or
                        higher for debugging. When this option takes value 4, the output is not SAM. [3]

              <b>-I</b> <u>FLOAT[,FLOAT[,INT[,INT]]]</u>
                        Specify  the mean, standard deviation (10% of the mean if absent), max (4 sigma from the
                        mean if absent) and min (4 sigma if  absent)  of  the  insert  size  distribution.  Only
                        applicable  to the FR orientation. By default, BWA-MEM infers these numbers and the pair
                        orientations given enough reads. [inferred]

       <b>aln</b>    bwa aln [-n maxDiff] [-o maxGapO] [-e maxGapE] [-d nDelTail] [-i nIndelEnd] [-k  maxSeedDiff]  [-l
              seedLen]  [-t  nThrds]  [-cRN]  [-M  misMsc]  [-O  gapOsc] [-E gapEsc] [-q trimQual] &lt;in.db.fasta&gt;
              &lt;in.query.fq&gt; &gt; &lt;out.sai&gt;

              Find the SA coordinates of the input reads. Maximum <u>maxSeedDiff</u> differences  are  allowed  in  the
              first <u>seedLen</u> subsequence and maximum <u>maxDiff</u> differences are allowed in the whole sequence.

              <b>OPTIONS:</b>

              <b>-n</b> <u>NUM</u>    Maximum  edit  distance if the value is INT, or the fraction of missing alignments given
                        2% uniform base error rate if FLOAT. In the latter case, the maximum  edit  distance  is
                        automatically chosen for different read lengths. [0.04]

              <b>-o</b> <u>INT</u>    Maximum number of gap opens [1]

              <b>-e</b> <u>INT</u>    Maximum number of gap extensions, -1 for k-difference mode (disallowing long gaps) [-1]

              <b>-d</b> <u>INT</u>    Disallow a long deletion within INT bp towards the 3'-end [16]

              <b>-i</b> <u>INT</u>    Disallow an indel within INT bp towards the ends [5]

              <b>-l</b> <u>INT</u>    Take  the  first  INT  subsequence  as  seed.  If INT is larger than the query sequence,
                        seeding will be disabled. For long reads, this option is typically ranged from 25 to  35
                        for `-k 2'. [inf]

              <b>-k</b> <u>INT</u>    Maximum edit distance in the seed [2]

              <b>-t</b> <u>INT</u>    Number of threads (multi-threading mode) [1]

              <b>-M</b> <u>INT</u>    Mismatch  penalty.  BWA  will  not  search  for  suboptimal hits with a score lower than
                        (bestScore-misMsc). [3]

              <b>-O</b> <u>INT</u>    Gap open penalty [11]

              <b>-E</b> <u>INT</u>    Gap extension penalty [4]

              <b>-R</b> <u>INT</u>    Proceed with suboptimal alignments if there are no more than INT equally best hits. This
                        option only affects paired-end mapping. Increasing this threshold helps to  improve  the
                        pairing accuracy at the cost of speed, especially for short reads (~32bp).

              <b>-c</b>        Reverse query but not complement it, which is required for alignment in the color space.
                        (Disabled since 0.6.x)

              <b>-N</b>        Disable  iterative search. All hits with no more than <u>maxDiff</u> differences will be found.
                        This mode is much slower than the default.

              <b>-q</b> <u>INT</u>    Parameter for read trimming. BWA trims a read down to  argmax_x{\sum_{i=x+1}^l(INT-q_i)}
                        if q_l&lt;INT where l is the original read length. [0]

              <b>-I</b>        The input is in the Illumina 1.3+ read format (quality equals ASCII-64).

              <b>-B</b> <u>INT</u>    Length  of  barcode  starting from the 5'-end. When <u>INT</u> is positive, the barcode of each
                        read will be trimmed before mapping and will be written at the <b>BC</b> SAM tag.  For  paired-
                        end reads, the barcode from both ends are concatenated. [0]

              <b>-b</b>        Specify the input read sequence file is the BAM format. For paired-end data, two ends in
                        a  pair  must  be  grouped  together and options <b>-1</b> or <b>-2</b> are usually applied to specify
                        which end should be mapped. Typical command lines for mapping pair-end data in  the  BAM
                        format are:

                            bwa aln ref.fa -b1 reads.bam &gt; 1.sai
                            bwa aln ref.fa -b2 reads.bam &gt; 2.sai
                            bwa sampe ref.fa 1.sai 2.sai reads.bam reads.bam &gt; aln.sam

              <b>-0</b>        When <b>-b</b> is specified, only use single-end reads in mapping.

              <b>-1</b>        When <b>-b</b> is specified, only use the first read in a read pair in mapping (skip single-end
                        reads and the second reads).

              <b>-2</b>        When <b>-b</b> is specified, only use the second read in a read pair in mapping.

       <b>samse</b>  bwa samse [-n maxOcc] &lt;in.db.fasta&gt; &lt;in.sai&gt; &lt;in.fq&gt; &gt; &lt;out.sam&gt;

              Generate  alignments  in  the  SAM format given single-end reads. Repetitive hits will be randomly
              chosen.

              <b>OPTIONS:</b>

              <b>-n</b> <u>INT</u>    Maximum number of alignments to output in the XA tag for reads  paired  properly.  If  a
                        read has more than INT hits, the XA tag will not be written. [3]

              <b>-r</b> <u>STR</u>    Specify the read group in a format like `@RG\tID:foo\tSM:bar'. [null]

       <b>sampe</b>  bwa  sampe  [-a  maxInsSize]  [-o  maxOcc]  [-n  maxHitPaired]  [-N  maxHitDis] [-P] &lt;in.db.fasta&gt;
              &lt;in1.sai&gt; &lt;in2.sai&gt; &lt;in1.fq&gt; &lt;in2.fq&gt; &gt; &lt;out.sam&gt;

              Generate alignments in the SAM format given paired-end reads. Repetitive read pairs will be placed
              randomly.

              <b>OPTIONS:</b>

              <b>-a</b> <u>INT</u>  Maximum insert size for a read pair to be considered being mapped properly.  Since  0.4.5,
                      this  option  is  only  used  when  there  are  not  enough  good  alignment  to infer the
                      distribution of insert sizes. [500]

              <b>-o</b> <u>INT</u>  Maximum occurrences of a read for pairing. A read with more occurrneces will be treated as
                      a single-end read. Reducing this parameter helps faster pairing. [100000]

              <b>-P</b>      Load the entire FM-index into memory to reduce disk operations  (base-space  reads  only).
                      With  this  option,  at least 1.25N bytes of memory are required, where N is the length of
                      the genome.

              <b>-n</b> <u>INT</u>  Maximum number of alignments to output in the XA tag for reads paired properly. If a  read
                      has more than INT hits, the XA tag will not be written. [3]

              <b>-N</b> <u>INT</u>  Maximum  number  of  alignments  to  output  in  the  XA  tag for disconcordant read pairs
                      (excluding singletons). If a read has more than INT hits, the XA tag will not be  written.
                      [10]

              <b>-r</b> <u>STR</u>  Specify the read group in a format like `@RG\tID:foo\tSM:bar'. [null]

       <b>bwasw</b>  bwa  bwasw  [-a matchScore] [-b mmPen] [-q gapOpenPen] [-r gapExtPen] [-t nThreads] [-w bandWidth]
              [-T thres] [-s hspIntv] [-z zBest] [-N nHspRev] [-c thresCoef] &lt;in.db.fasta&gt; &lt;in.fq&gt; [mate.fq]

              Align query sequences in the <u>in.fq</u> file. When <u>mate.fq</u> is present,  perform  paired-end  alignment.
              The  paired-end mode only works for reads Illumina short-insert libraries. In the paired-end mode,
              BWA-SW may still output split alignments but they are all marked as not properly paired; the  mate
              positions will not be written if the mate has multiple local hits.

              <b>OPTIONS:</b>

              <b>-a</b> <u>INT</u>    Score of a match [1]

              <b>-b</b> <u>INT</u>    Mismatch penalty [3]

              <b>-q</b> <u>INT</u>    Gap open penalty [5]

              <b>-r</b> <u>INT</u>    Gap extension penalty. The penalty for a contiguous gap of size k is q+k*r. [2]

              <b>-t</b> <u>INT</u>    Number of threads in the multi-threading mode [1]

              <b>-w</b> <u>INT</u>    Band width in the banded alignment [33]

              <b>-T</b> <u>INT</u>    Minimum score threshold divided by a [37]

              <b>-c</b> <u>FLOAT</u>  Coefficient  for  threshold adjustment according to query length. Given an l-long query,
                        the threshold for a hit to be retained is a*max{T,c*log(l)}. [5.5]

              <b>-z</b> <u>INT</u>    Z-best heuristics. Higher -z increases accuracy at the cost of speed. [1]

              <b>-s</b> <u>INT</u>    Maximum SA interval size for initiating a seed. Higher -s increases accuracy at the cost
                        of speed. [3]

              <b>-N</b> <u>INT</u>    Minimum number of seeds supporting the resultant alignment to  skip  reverse  alignment.
                        [5]

</pre><h4><b>SAM</b> <b>ALIGNMENT</b> <b>FORMAT</b></h4><pre>
       The  output of the <b>`aln'</b> command is binary and designed for BWA use only. BWA outputs the final alignment
       in the SAM (Sequence Alignment/Map) format. Each line consists of:

                      ┌─────┬───────┬──────────────────────────────────────────────────────────┐
                      │ <b>Col</b> │ <b>Field</b> │                       <b>Description</b>                        │
                      ├─────┼───────┼──────────────────────────────────────────────────────────┤
                      │  1  │ QNAME │ Query (pair) NAME                                        │
                      │  2  │ FLAG  │ bitwise FLAG                                             │
                      │  3  │ RNAME │ Reference sequence NAME                                  │
                      │  4  │ POS   │ 1-based leftmost POSition/coordinate of clipped sequence │
                      │  5  │ MAPQ  │ MAPping Quality (Phred-scaled)                           │
                      │  6  │ CIAGR │ extended CIGAR string                                    │
                      │  7  │ MRNM  │ Mate Reference sequence NaMe (`=' if same as RNAME)      │
                      │  8  │ MPOS  │ 1-based Mate POSistion                                   │
                      │  9  │ ISIZE │ Inferred insert SIZE                                     │
                      │ 10  │ SEQ   │ query SEQuence on the same strand as the reference       │
                      │ 11  │ QUAL  │ query QUALity (ASCII-33 gives the Phred base quality)    │
                      │ 12  │ OPT   │ variable OPTional fields in the format TAG:VTYPE:VALUE   │
                      └─────┴───────┴──────────────────────────────────────────────────────────┘

       Each bit in the FLAG field is defined as:

                               ┌─────┬────────┬───────────────────────────────────────┐
                               │ <b>Chr</b> │  <b>Flag</b>  │              <b>Description</b>              │
                               ├─────┼────────┼───────────────────────────────────────┤
                               │  p  │ 0x0001 │ the read is paired in sequencing      │
                               │  P  │ 0x0002 │ the read is mapped in a proper pair   │
                               │  u  │ 0x0004 │ the query sequence itself is unmapped │
                               │  U  │ 0x0008 │ the mate is unmapped                  │
                               │  r  │ 0x0010 │ strand of the query (1 for reverse)   │
                               │  R  │ 0x0020 │ strand of the mate                    │
                               │  1  │ 0x0040 │ the read is the first read in a pair  │
                               │  2  │ 0x0080 │ the read is the second read in a pair │
                               │  s  │ 0x0100 │ the alignment is not primary          │
                               │  f  │ 0x0200 │ QC failure                            │
                               │  d  │ 0x0400 │ optical or PCR duplicate              │
                               │  S  │ 0x0800 │ supplementary alignment               │
                               └─────┴────────┴───────────────────────────────────────┘

       The Please check &lt;<a href="http://samtools.sourceforge.net">http://samtools.sourceforge.net</a>&gt; for the format specification and the tools  for  post-
       processing the alignment.

       BWA generates the following optional fields. Tags starting with `X' are specific to BWA.

                              ┌─────┬──────────────────────────────────────────────────┐
                              │ <b>Tag</b> │                     <b>Meaning</b>                      │
                              ├─────┼──────────────────────────────────────────────────┤
                              │ <b>NM</b>  │ Edit distance                                    │
                              │ <b>MD</b>  │ Mismatching positions/bases                      │
                              │ <b>AS</b>  │ Alignment score                                  │
                              │ <b>BC</b>  │ Barcode sequence                                 │
                              │ <b>SA</b>  │ Supplementary alignments                         │
                              ├─────┼──────────────────────────────────────────────────┤
                              │ <b>X0</b>  │ Number of best hits                              │
                              │ <b>X1</b>  │ Number of suboptimal hits found by BWA           │
                              │ <b>XN</b>  │ Number of ambiguous bases in the referenece      │
                              │ <b>XM</b>  │ Number of mismatches in the alignment            │
                              │ <b>XO</b>  │ Number of gap opens                              │
                              │ <b>XG</b>  │ Number of gap extensions                         │
                              │ <b>XT</b>  │ Type: Unique/Repeat/N/Mate-sw                    │
                              │ <b>XA</b>  │ Alternative hits; format: /(chr,pos,CIGAR,NM;)*/ │
                              ├─────┼──────────────────────────────────────────────────┤
                              │ <b>XS</b>  │ Suboptimal alignment score                       │
                              │ <b>XF</b>  │ Support from forward/reverse alignment           │
                              │ <b>XE</b>  │ Number of supporting seeds                       │
                              └─────┴──────────────────────────────────────────────────┘

       Note that XO and XG are generated by BWT search while the CIGAR string by Smith-Waterman alignment. These
       two tags may be inconsistent with the CIGAR string. This is not a bug.

</pre><h4><b>NOTES</b> <b>ON</b> <b>SHORT-READ</b> <b>ALIGNMENT</b></h4><pre>
   <b>Alignment</b> <b>Accuracy</b>
       When  seeding  is  disabled,  BWA  guarantees to find an alignment containing maximum <u>maxDiff</u> differences
       including <u>maxGapO</u> gap opens which do not occur within <u>nIndelEnd</u> bp  towards  either  end  of  the  query.
       Longer  gaps may be found if <u>maxGapE</u> is positive, but it is not guaranteed to find all hits. When seeding
       is enabled, BWA further requires that the first <u>seedLen</u> subsequence contains  no  more  than  <u>maxSeedDiff</u>
       differences.

       When gapped alignment is disabled, BWA is expected to generate the same alignment as Eland version 1, the
       Illumina  alignment  program.  However, as BWA change `N' in the database sequence to random nucleotides,
       hits to these random sequences will also be counted. As a consequence, BWA may mark a  unique  hit  as  a
       repeat,  if  the  random  sequences happen to be identical to the sequences which should be unqiue in the
       database.

       By default, if the best hit is not highly repetitive (controlled by -R), BWA also finds all hits contains
       one more mismatch; otherwise, BWA finds all equally best hits only. Base quality  is  NOT  considered  in
       evaluating  hits. In the paired-end mode, BWA pairs all hits it found. It further performs Smith-Waterman
       alignment for unmapped reads to rescue reads with a high erro rate, and for high-quality anomalous  pairs
       to fix potential alignment errors.

   <b>Estimating</b> <b>Insert</b> <b>Size</b> <b>Distribution</b>
       BWA estimates the insert size distribution per 256*1024 read pairs. It first collects pairs of reads with
       both ends mapped with a single-end quality 20 or higher and then calculates median (Q2), lower and higher
       quartile  (Q1  and Q3). It estimates the mean and the variance of the insert size distribution from pairs
       whose insert sizes are within interval [Q1-2(Q3-Q1), Q3+2(Q3-Q1)]. The maximum  distance  x  for  a  pair
       considered   to   be   properly  paired  (SAM  flag  0x2)  is  calculated  by  solving  equation  Phi((x-
       mu)/sigma)=x/L*p0, where mu is the mean, sigma is the standard error of the insert size  distribution,  L
       is  the  length  of  the  genome,  p0  is  prior  of  anomalous pair and Phi() is the standard cumulative
       distribution function. For mapping Illumina short-insert reads to the human genome, x is about 6-7  sigma
       away from the mean. Quartiles, mean, variance and x will be printed to the standard error output.

   <b>Memory</b> <b>Requirement</b>
       With  bwtsw algorithm, 5GB memory is required for indexing the complete human genome sequences. For short
       reads, the <b>aln</b> command uses ~3.2GB memory and the <b>sampe</b> command uses ~5.4GB.

   <b>Speed</b>
       Indexing the human genome sequences takes 3 hours with bwtsw algorithm. Indexing smaller genomes with  IS
       algorithms is faster, but requires more memory.

       The  speed  of alignment is largely determined by the error rate of the query sequences (r). Firstly, BWA
       runs much faster for near perfect hits than for hits with many differences, and it stops searching for  a
       hit  with  l+2  differences if a l-difference hit is found. This means BWA will be very slow if r is high
       because in this case BWA has to visit hits with many differences and looking for these hits is expensive.
       Secondly, the alignment algorithm behind makes the speed sensitive  to  [k  log(N)/m],  where  k  is  the
       maximum allowed differences, N the size of database and m the length of a query. In practice, we choose k
       w.r.t. r and therefore r is the leading factor. I would not recommend to use BWA on data with r&gt;0.02.

       Pairing  is  slower  for  shorter reads. This is mainly because shorter reads have more spurious hits and
       converting SA coordinates to chromosomal coordinates are very costly.

</pre><h4><b>CHANGES</b> <b>IN</b> <b>BWA-0.6</b></h4><pre>
       Since version 0.6, BWA has been able to work with a reference genome longer than 4GB.  This feature makes
       it possible to integrate the forward and reverse complemented genome in one  FM-index,  which  speeds  up
       both  BWA-short  and BWA-SW. As a tradeoff, BWA uses more memory because it has to keep all positions and
       ranks in 64-bit integers, twice larger than 32-bit integers used in the previous versions.

       The latest BWA-SW also works for paired-end reads longer than 100bp. In comparison to  BWA-short,  BWA-SW
       tends  to be more accurate for highly unique reads and more robust to relative long INDELs and structural
       variants.  Nonetheless, BWA-short usually has higher power to  distinguish  the  optimal  hit  from  many
       suboptimal hits. The choice of the mapping algorithm may depend on the application.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       BWA website &lt;<a href="http://bio-bwa.sourceforge.net">http://bio-bwa.sourceforge.net</a>&gt;, Samtools website &lt;<a href="http://samtools.sourceforge.net">http://samtools.sourceforge.net</a>&gt;

</pre><h4><b>AUTHOR</b></h4><pre>
       Heng  Li  at  the  Sanger Institute wrote the key source codes and integrated the following codes for BWT
       construction: bwtsw &lt;<a href="http://i.cs.hku.hk/~ckwong3/bwtsw/">http://i.cs.hku.hk/~ckwong3/bwtsw/</a>&gt;, implemented by Chi-Kwong Wong at the University
       of  Hong  Kong  and  IS   &lt;<a href="http://yuta.256.googlepages.com/sais">http://yuta.256.googlepages.com/sais</a>&gt;   originally   proposed   by   Nong   Ge
       &lt;<a href="http://www.cs.sysu.edu.cn/nong/">http://www.cs.sysu.edu.cn/nong/</a>&gt; at the Sun Yat-Sen University and implemented by Yuta Mori.

</pre><h4><b>LICENSE</b> <b>AND</b> <b>CITATION</b></h4><pre>
       The  full  BWA package is distributed under GPLv3 as it uses source codes from BWT-SW which is covered by
       GPL. Sorting, hash table, BWT and IS libraries are distributed under the MIT license.

       If you use the BWA-backtrack algorithm, please cite the following paper:

       Li H. and Durbin R. (2009) Fast  and  accurate  short  read  alignment  with  Burrows-Wheeler  transform.
       Bioinformatics, 25, 1754-1760. [PMID: 19451168]

       If you use the BWA-SW algorithm, please cite:

       Li  H.  and  Durbin  R.  (2010)  Fast  and  accurate  long-read alignment with Burrows-Wheeler transform.
       Bioinformatics, 26, 589-595. [PMID: 20080505]

       If you use BWA-MEM or the fastmap component of BWA, please cite:

       Li  H.  (2013)  Aligning  sequence  reads,  clone  sequences   and   assembly   contigs   with   BWA-MEM.
       arXiv:1303.3997v1 [q-bio.GN].

       It is likely that the BWA-MEM manuscript will not appear in a peer-reviewed journal.

</pre><h4><b>HISTORY</b></h4><pre>
       BWA is largely influenced by BWT-SW. It uses source codes from BWT-SW and mimics its binary file formats;
       BWA-SW  resembles  BWT-SW  in several ways. The initial idea about BWT-based alignment also came from the
       group who developed BWT-SW. At the same time,  BWA  is  different  enough  from  BWT-SW.  The  short-read
       alignment  algorithm  bears  no similarity to Smith-Waterman algorithm any more. While BWA-SW learns from
       BWT-SW, it introduces heuristics that can hardly be applied to the original algorithm. In all,  BWA  does
       not  guarantee to find all local hits as what BWT-SW is designed to do, but it is much faster than BWT-SW
       on both short and long query sequences.

       I started to write the first piece of codes on 24 May 2008 and got the initial stable version on 02  June
       2008.  During this period, I was acquainted that Professor Tak-Wah Lam, the first author of BWT-SW paper,
       was collaborating with Beijing Genomics Institute on SOAP2, the successor to SOAP (Short  Oligonucleotide
       Analysis  Package).  SOAP2 has come out in November 2008. According to the SourceForge download page, the
       third BWT-based short read aligner, bowtie, was first released in August 2008. At  the  time  of  writing
       this manual, at least three more BWT-based short-read aligners are being implemented.

       The  BWA-SW  algorithm  is  a new component of BWA. It was conceived in November 2008 and implemented ten
       months later.

       The BWA-MEM algorithm is based on an algorithm finding super-maximal exact  matches  (SMEMs),  which  was
       first  published  with the fermi assembler paper in 2012. I first implemented the basic SMEM algorithm in
       the <b>fastmap</b> command for an experiment and then extended the basic algorithm and added the extension  part
       in Feburary 2013 to make BWA-MEM a fully featured mapper.

bwa-0.7.18-r1243                                  14 April 2023                                           <u><a href="../man1/bwa.1.html">bwa</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>