<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>perm - Efficient mapping of short reads with periodic spaced seeds</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/perm">perm_0.4.0-8_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       <b>perm</b> - Efficient mapping of short reads with periodic spaced seeds

       If you have any usage questions, please email "yanghoch at usc dot edu".

</pre><h4><b>SYNOPSIS</b></h4><pre>
       To use the command line, type <u>perm</u> with the args in the order.

</pre><h4><b>EXAMPLES</b></h4><pre>
       <b>For</b> <b>single-end</b> <b>reads:</b>

       <b>perm</b> <u>Ref</u> <u>Reads</u> [<u>options</u>]

       Examples:

       <b>perm</b> Ref.fasta Reads.fasta -v 5 -o out.mapping -u ummappedReads.fa

       <b>perm</b> RefFilesList.txt ReadsSetFilesList.txt -v 5 -u unmappedReads.fa -E

       <b>perm</b> Ref.fasta Reads.csfasta -v 5 -m -s my.index --delimiter ´,´ --seed F3

       <b>perm</b> my.index SingleEndReads.csfasta -v 5 -o out.sam -k 10 -a ambiguous10.csfasta

       <b>For</b> <b>paired-end</b> <b>reads:</b>

       <b>perm</b> <u>Ref</u> <b>-1</b> <u>F3_Reads</u> <b>-2</b> <u>R3_Reads</u> [<u>options</u>]

       Examples:

       <b>perm</b> ref.fa -1 F3.fa -2 R3.fa -U 3000 -L 100 -v 5 -A -m -s -o out.sam

       <b>perm</b> ref.txt -1 F3.fq -2 R3.fq -v 5 -m -s my.index -o out.mapping --seed F3

       <b>perm</b> my.index -1 F3.fq -2 R3.fq -U 3000 -L 100 -v 5 -A -o out.sam

       <b>To</b> <b>build</b> <b>an</b> <b>index</b> <b>only:</b>

       <b>perm</b> <u>Ref</u> <u>Read_Length</u> <b>--readFormat</b> &lt;.csfasta|.fasta&gt; <b>-m</b> <b>-s</b> <u>index</u> <u>path</u> <b>--seed</b> F3

       Example:

       <b>perm</b> hg18.txt 50 <b>--readFormat</b> .csfasta <b>-m</b> <b>-s</b> hg18_50_SOLiD.index

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>Required</b> <b>Arguments</b>

       •   The reference file should be in FASTA format with the either the <u>.fasta</u>, <u>.fna</u>, or <u>.fa</u> file extension.
           For  a transcriptome with multiple genes or isoforms as reference, concatenate all FASTA sequences in
           a single FASTA file. Alternatively, if there are many files, for  example  one  per  chromosome,  ex:
           chr1.fa to chrY.fa, list the FASTA filenames one per line in a file which has the <u>.txt</u> extension. The
           <u>.txt</u>  is  important  because  PerM examines the file extension to know if the input file is a list of
           filenames. The filenames need to include the file path (relative or absolute) unless the FASTA  files
           are all in the same directory that PerM is run from.

       •   The  read  file(s)  should  be in the .fasta, .fastq, .csfasta or .csfastq format. PerM parses a file
           according to its extension, or the format explicitly specified by the <b>--readFormat</b> <b>&lt;format&gt;</b> flag.  If
           there are multiple read files, list each filename, one per line, in a .txt file. PerM takes it as the
           input and can map multiple read sets in parallel by [<a href="http://en.wikipedia.org/wiki/OpenMP">http://en.wikipedia.org/wiki/OpenMP</a> OpenMP].

       <b>Short</b> <b>Options</b> <b>(grouped</b> <b>by</b> <b>related</b> <b>functionality)</b>

       <b>-A</b>     Output <u>all</u> alignments within the mismatch threshold (see <u>-v</u> option), end-to-end.

       <b>-B</b>     Output  <u>best</u> alignments in terms of mismatches in the threshold (see <u>-v</u> option). For example, if a
              read has no perfect  match  alignments,  two  single  base  mismatch  alignments,  and  additional
              alignments  with  more mismatches, only the two single base mismatch alignments will be output. <u>-B</u>
              is the default mode if neither <u>-A</u> or <u>-B</u> is specified.

       <b>-E</b>     Output only uniquely mapped reads <u>remaining</u> <u>after</u> <u>the</u> <u>best</u> down  selection  has  been  applied  if
              applicable.  When  combined  with  the  <u>-A</u>  option,  only reads with a single alignment within the
              mismatch threshold (see <u>-v</u> option) will be output.

       <b>-v</b>     Maximum number of mismatches allowed (or allowed in each end  for  pair-end  reads).  The  default
              value is the number of mismatches that the seed used is fully sensitive to.

       <b>-k</b>     Specifies  maximum  number of alignments to output. The default value is 200 if the -k flag is not
              given. Alignments for reads mapping to more than the maximum number positions will not be  output.
              Use the -a option to collect reads which exceeded the maximum.

       <b>-t</b>     Number  of  bases at the 5´ end of each read to ignore. For example, if the first 5 bases are used
              as a barcode or to index multiple samples together, use -t 5. If not specified, no  initial  bases
              will be ignored.

       <b>-T</b>     Number of bases in each read to use, starting after any bases ignored by -t option. Later bases at
              the  3´  of the read are ignored. For example, -T 30 means use only first 30 bases (signals) after
              the any bases ignored due to the -t option.

       <b>-m</b>     Create the reference index without reusing the saved index even if available.

       <b>-s</b> <u>path</u>
              Save the reference index to accelerate the mapping in the future. If <u>path</u> is  not  specified,  the
              index  will  be  created  in the current working directory (i.e. where PerM is run from) using the
              default index name. If <u>path</u> is a directory, the index will be created in the  specified  directory
              using the default index name (directory must exist; it will not automatically be created). If <u>path</u>
              is a file path, the index will be created with the specified name.

       <b>-o</b> <u>filepath</u>
              Name  of mapping output file when mapping a single read set. The output file format will be either
              the .mapping tab delimited text format or the SAM format as determined by  the  extension  of  the
              output   filename.   For   example   {{{-o   out.sam}}}   will   output   in   SAM  format;  {{{-o
              /path/to/out.mapping}}} will output in  .mapping  format.  Use  <u>--outputFormat</u>  to  override  this
              behavior.  The  <u>-o</u> option does not apply when multiple reads sets are being mapped at once to take
              advantage of multiple CPUs (cores); see the <u>-d</u> option for that case.

       <b>-d</b> <u>dirpath</u>
              Output directory for mapping output files when mapping multiple read sets (output  files  will  be
              named  automatically).  If  the  directory  specified does not exist, the output directory will be
              created provided the parent directory exists. If the <u>-d</u> switch is not  specified,  files  will  be
              written to the directory PerM is run from. Note: if -d <u>filepath</u> is specified when mapping a single
              read set, <u>dirpath</u> will be prepended to <u>filepath</u>; however, this usage is not recommended.

       <b>-a</b> <u>filepath</u>
              Create  a FASTA (FASTQ) file for reads mapped to more positions than the threshold specified by -k
              or the default of 200.

       <b>-b</b> <u>filepath</u>
              Create a FASTA (FASTQ) file for reads that  is  shorter  than  expected  length  or  with  strange
              characters.

       <b>-u</b> <u>filepath</u>
              Create  a  FASTA  (FASTAQ)  file  of  unmapped  reads.  When a single read set is mapped, <u>filename</u>
              specifies the name of the output file. When multiple read sets are mapped, <u>filename</u> is  irrelevant
              and  should be omitted; the files of unmapped sequences will automatically be named and created in
              the directory PerM is run from.

       <b>Long</b> <b>Options</b>

       <b>--ambiguosReadOnly</b>
              Output only ambiguous mapping to find repeats (similar  regions  within  substitution  threshold).
              When  this  option is specified, reads that mapped to over mapping number threshold that specified
              by -k will still be printed.

       <b>--ambiguosReadInOneLine</b>
              utput reads mapped to more than k places in one line. When this option is  specified,  reads  that
              mapped  to  over mapping number threshold specified by -k will still be printed but printed in one
              line.

       <b>--noSamHeader</b>
              Do not include a SAM header. This makes it easier to concatenate multiple SAM output files.

       <b>--includeReadsWN</b>
              Map reads with equal or fewer N or ´.´ bases than the specified threshold by encoding N or ´.´  as
              A or 3. Reads with more ´N´ will be discarded. The default setting discards read with any ´N´.

       <b>--statsOnly</b>
              Output the mapping statistics to stdout only, without saving alignments to files.

       <b>--ignoreQS</b>
              Ignore the quality scores in FASTQ or QUAL files.

       <b>--printNM</b>
              When  quality  scores  are  available,  use  this  flag  to print number of mismatches, instead of
              mismatch scores in mapping format.

       <b>--seed</b> {F,,0,, | F,,1,, | F,,2,, | F,,3,, | F,,4,, | S,,11,, | S,,20,, | S,,12,,}
              Specify the seed pattern. The F,,0,,, F,,1,,, F,,2,,, F,,3,,, and F,,4,, seeds are fully sensitive
              to 0-4 mismatches respectively. The S,,11,, S,,20,, S,,12,,  seeds  are  designed  for  the  SOLiD
              sequencer.  An S,,kj,, seed is full sensitive to k adjacent mismatch pairs (SNP signature is color
              space) and j isolated mismatches. See [<a href="http://code.google.com/p/perm/wiki/Algorithms">http://code.google.com/p/perm/wiki/Algorithms</a> the algorithm
              page] for more information about the seed patterns.

       <b>--refFormat</b> <u>{fasta</u> <u>|</u> <u>list</u> <u>|</u> <u>index</u> <u>}</u>
              Assume references sequence(s) are in the specified format, instead of guessing according to file´s
              extension.

       <b>--readFormat</b> |<u>{fasta</u> <u>|</u> <u>fastq</u> <u>|</u> <u>csfasta</u> <u>|</u> <u>csfastq}</u>
              Assume reads are in the specified format, instead of guessing according to the file(s)´ extension.

       <b>--outputFormat</b> <u>{</u> <u>sam</u> <u>|</u> <u>mapping</u> <u>}</u>
              Override the default output mapping format option or specify it explicitly when  the  output  file
              extension is not .sam or .mapping.

       <b>--delimiter</b> <u>char</u>
              <u>char</u>  is a character used as the delimiter to separate the the read id, and the additional info in
              the line with &gt; when reading a FASTA or CSFASTA file.

       <b>--log</b> <u>filepath</u>
              <u>filepath</u> specifies the name of the log file which contains the mapping statistics that  will  also
              be printed on the screen.

       <b>--forwardOnly</b>
              Map reads to the forward strand only: (This is for SOLiD Strand specific sequencing).

       <b>--reverseOnly</b>
              Map reads to the reverse strand only: (This is for SOLiD Strand specific sequencing)

       <b>Options</b> <b>for</b> <b>Paired-end</b> <b>Reads</b>

       PerM deals with mate-paired reads by mapping each end separately. All combinations of mated pairs mapping
       to  the same reference sequence will be output if their separation are in the allowed ranged as specified
       by the <u>-L</u> and <u>-U</u> flags.

       <b>-e</b>     Exclude ambiguous paired.

       <b>-L</b> / <b>--lowerBound</b> <u>Int</u>
              lower bound for mate-paired separation distance

       <b>-U</b> / <b>--upperBound</b> <u>Int</u>
              upper bound for mate-paired separation distance

       The upper bound and lower bound can be negative, which may catch the re-arrangement variations.  Use  the
       <b>-A</b>  argument  to  avoid missing the correct pairs. However, this may greatly increase the running time if
       both ends are in repetitive regions.

       <b>--fr</b>   Map paired-end reads to different strand only

       <b>--ff</b>   Map paired-end reads to the same strand only

       <b>--printRefSeq</b>
              Print the mapped reference paired sequence as the two last  columns  in  .mapping  format.  |  The
              default option output mapping in both the same or different strand.

</pre><h4><b>DEFAULT</b> <b>SETTINGS</b></h4><pre>
       The  following  are  the  default  settings  when the corresponding command line option is not specified.
       Please specify the option to change the default settings.

       •   Allow only two mismatches only in each end and use seed F,,2,, S,,11,, or F,,3,, ,selected  according
           to the read lengths and types.

       •   Print the best alignments for each read in terms of the number of mismatches.

       •   Output files in <b>*</b>.mapping format.

       •   Searches for a saved index with the default file name before building the new index.

       •   Won´t save the index in file, unless {{{-s}}} is specified.

       •   For paired end reads, the default allowed separation distance is 0-3000 bp. Change with the <b>-L</b> and <b>-U</b>
           options.

       <b>Parallel</b> <b>Mapping</b>

       PerM  simultaneously  maps  multiple  reads sets in a list by querying the same index. It will detect how
       many CPUs (cores) are available and assign each of them a read set. If a read set is done, the next  read
       set  in the list will be processed automatically. Each read set will have its own mapping output file. To
       better utilize all CPUs on a node, large reads set should be split into many small read sets and put in a
       list. When multiple nodes are used in the same file system, the index should be pre-built  first  by  one
       node;  the  other  nodes  will  read  the pre-built index without building index again. Without pre-built
       index, each machine will try to build its own index, wasting CPU time and storage space.

</pre><h4><b>Exit</b> <b>codes</b></h4><pre>
       PerM sets the exit code to 0 upon successful completion, the normal Unix  behavior.  If  the  program  is
       terminated  via  Ctrl-C  (SIGINT),  the exit code will be 2, the number for SIGINT (see <b>man</b> <b>kill</b>). If you
       invoke PerM from another language, you can check the return code and do something intelligent. Here is  a
       Perl pseudo-code example:

           while (... some sort of loop ...) {
             my $cmd = "PerM ... arguments and switches";
             my $ec = system($cmd);
             if ($ec == 2) {
           print STDERR "PerM terminated via Ctrl-C. Stopping run.\n\n";
           # Maybe do some cleanup such as deleting the small files the read file was
           # split into for parallel processing.
           exit($ec);
             }
           }

</pre><h4><b>Use</b> <b>PerM</b> <b>on</b> <b>Galaxy</b></h4><pre>
       Thanks   to   Prof   Anton   Nekrutenko   and   Kelly   Vincent   at   PSU,  you  can  now  use  PerM  on
       [<a href="http://test.g2.bx.psu.edu/">http://test.g2.bx.psu.edu/</a> Galaxy´s <u>test</u> server]. Follow the  hyperlink  to  Galaxy´s  page,  and  click
       NGS:Mapping in the tool menu. Please choose <u>Map</u> <u>with</u> <u>PerM</u> <u>for</u> <u>SOLiD</u> <u>and</u> <u>Illumina</u>. You can upload your own
       reference  or  use  the  pre-built  hg19  index  in  the  system.  Please  email  me if you encounter any
       difficulties. Once the system proves its stability, it will be moved to Galaxy´s main  server  with  more
       pre-built reference index.

</pre><h4><b>Unit</b> <b>Test</b></h4><pre>
       When  PerM was developed, a unit cppUnit test module was also prepared. If you are interested in the test
       code for PerM, please email me.

Yangho Chen                                        April 2014                                            <u><a href="../man1/PERM.1.html">PERM</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>