<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>bamadapterfind - find adapter contamination in sequencing reads</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/biobambam2">biobambam2_2.0.185+ds-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       bamadapterfind - find adapter contamination in sequencing reads

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>bamdapterfind</b> [options]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       bamdapterfind  scans  a  BAM file for contaminations by sequencing adapters. It uses two separate methods
       for this detection:

       list:  each read is matched against a predefined list of adapter sequences. A sequence is  considered  as
              matching  if  there is an overlap of a least <u>adpmatchminscore</u> bases, the overlap covers at least a
              factor of <u>adpmatchminfrac</u> of the adapter's length and the indel free local alignment  between  the
              adapter  and  the  read covers at least a factor of <u>adpmatchminpfrac</u> of the length of the possible
              overlap between the two. If such a match is found, then the auxiliary field <u>as</u> is filled with  the
              length  of  the  match,  <u>af</u>  is filled with the fraction of the adapter sequence matched and <u>aa</u> is
              filled with the name of the matched adapter sequence.

       overlap:
              the two mates need to have a match similar to the following two lines

                  s0s1s2s3s4s5s6s7s8s9s10s11s12s13s14s15s16t0t1t2t3
          x3x2x1x0s0s1s2s3s4s5s6s7s8s9s10s11s12s13s14s15s16

              where an infix s0s1s2... of the first read matches a suffix  of  the  reverse  complement  of  the
              second  read.  In  this case it is likely that the first read has been sequenced beyond the end of
              the payload sequence and into the attached adapter. This overlap needs to be at least  <u>MIN_OVERLAP</u>
              bases  long to be considered. If such an overlap is found, then the adjacent sequences are checked
              for a match, where in the example x3x2x1x0 needs to be the reverse  complement  of  t0t1t2t3.  The
              adjacent sequences are checked up to a limit of <u>ADAPTER_MATCH</u> base pairs. If such a match is found
              then  the  auxiliary  field  <u>ah</u>  is  set  to 1 and <u>a3</u> is used to store the length of the suspected
              adapter sequence.

       The following key=value pairs can be given at the program start:

       <b>level=&lt;-1|0|1|9|11&gt;:</b> set compression level of the output BAM file. Valid values are

       -1:    zlib/gzip default compression level

       0:     uncompressed

       1:     zlib/gzip level 1 (fast) compression

       9:     zlib/gzip level 9 (best) compression

       If libmaus has been compiled with support for igzip (see https://software.intel.com/en-us/articles/igzip-
       a-high-performance-deflate-compressor-with-optimizations-for-genomic-data) then an additional valid value
       is

       11:    igzip compression

       <b>verbose=&lt;1&gt;:</b> Valid values are

       1:     print progress report on standard error

       0:     do not print progress report

       <b>mod=&lt;1048576&gt;:</b> if verbose=1 then this sets the frequency of progress reports, i.e. a report is given  for
       each mod'th input read/alignment

       <b>adaptersbam=&lt;&gt;:</b>  file  name  of the BAM file containing the list of adapter used for the adapter matching
       described above under list. The program contains an internal list which is used if this key is not given.

       <b>SEED_LENGTH=&lt;12&gt;:</b> length of the seed used for detecting overlaps in overlap based matching  (see  overlap
       above, default value is 12 base pairs).

       <b>PCT_MISMATCH=&lt;10&gt;:</b> percentage of mismatches allowed for overlap matching. This only includes the overlap,
       not the suspected attached adapter sequence. The default value is 10.

       <b>MAX_SEED_MISMATCHES=&lt;SEED_LENGTH*PCT_MISMATCH&gt;:</b>  maximum  number  of  mismatches  allowed in the seed. By
       default this value is computed as SEED_LENGTH*PCT_MISMATCH.

       <b>MIN_OVERLAP=&lt;32&gt;:</b> minimum length of overlap for overlap matching in base pairs (see above).  The  default
       value is 32.

       <b>ADAPTER_MATCH=&lt;12&gt;:</b>  maximum  number  of  base  pairs  to  check  for  matching adapters in overlap based
       matching. The default value is 12.

       <b>adpmatchminscore=&lt;16&gt;</b> minimum score for list based adapter matching (see above, default value is 16)

       <b>adpmatchminfrac=&lt;0.75&gt;</b> minimum fraction of adapter sequence which needs  to  match  (see  above,  default
       value is 0.75=75%)

       <b>adpmatchminpfrac=&lt;0.8&gt;</b> minimum fraction of overlap for adapter list matching (see above, default value is
       0.8=80%)

       <b>clip=&lt;0&gt;</b> clip the adapters off and move the corresponding sequence part to the qs auxiliary field and the
       corresponding quality string part to the qq auxiliary field

       <b>reflen=&lt;3000000000&gt;</b> length of reference sequence/genome

       <b>pA=&lt;0.25&gt;</b> relative frequency of base A in reference sequence/genome

       <b>pC=&lt;0.25&gt;</b> relative frequency of base C in reference sequence/genome

       <b>pG=&lt;0.25&gt;</b> relative frequency of base G in reference sequence/genome

       <b>pT=&lt;0.25&gt;</b> relative frequency of base T in reference sequence/genome

</pre><h4><b>AUTHOR</b></h4><pre>
       Written by German Tischler.

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       Report bugs to &lt;<a href="mailto:germant@miltenyibiotec.de">germant@miltenyibiotec.de</a>&gt;

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright  ©  2009-2013  German  Tischler,  © 2011-2013 Genome Research Limited.  License GPLv3+: GNU GPL
       version 3 &lt;<a href="http://gnu.org/licenses/gpl.html">http://gnu.org/licenses/gpl.html</a>&gt;
       This is free software: you are free to change and redistribute it.  There is NO WARRANTY, to  the  extent
       permitted by law.

BIOBAMBAM                                           July 2013                                  <u><a href="../man1/BAMADAPTERFIND.1.html">BAMADAPTERFIND</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>