<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>fastq-mcf  -  ea-utils:  detect  levels  of  adapter  presence,  compute likelihoods and locations of the</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/ea-utils">ea-utils_1.1.2+dfsg-9build2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       fastq-mcf  -  ea-utils:  detect  levels  of  adapter  presence,  compute likelihoods and locations of the
       adapters

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>fastq-mcf</b> [<u>options</u>] <u>&lt;adapters.fa&gt;</u> <u>&lt;reads.fq&gt;</u> [<u>mates1.fq</u> ...]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Version: 1.04.676

       Detects levels of adapter presence, computes likelihoods and locations  (start,  end)  of  the  adapters.
       Removes the adapter sequences from the fastq file(s).

       Stats go to stderr, unless <b>-o</b> is specified.

       Specify <b>-0</b> to turn off all default settings

       If  you specify multiple 'paired-end' inputs, then a <b>-o</b> option is required for each.  IE: <b>-o</b> read1.clip.q
       <b>-o</b> read2.clip.fq

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>     This help

       <b>-o</b> FIL Output file (stats to stdout)

       <b>-s</b> N.N Log scale for adapter minimum-length-match (2.2)

       <b>-t</b> N   % occurance threshold before adapter clipping (0.25)

       <b>-m</b> N   Minimum clip length, overrides scaled auto (1)

       <b>-p</b> N   Maximum adapter difference percentage (10)

       <b>-l</b> N   Minimum remaining sequence length (19)

       <b>-L</b> N   Maximum remaining sequence length (none)

       <b>-D</b> N   Remove duplicate reads : Read_1 has an identical N bases (0)

       <b>-k</b> N   sKew percentage-less-than causing cycle removal (2)

       <b>-x</b> N   'N' (Bad read) percentage causing cycle removal (20)

       <b>-q</b> N   quality threshold causing base removal (10)

       <b>-w</b> N   window-size for quality trimming (1)

       <b>-H</b>     remove &gt;95% homopolymer reads (no)

       <b>-X</b>     remove low complexity reads (no)

       <b>-0</b>     Set all default parameters to zero/do nothing

       <b>-U</b>|u   Force disable/enable Illumina PF filtering (auto)

       <b>-P</b> N   Phred-scale (auto)

       <b>-R</b>     Don't remove N's from the fronts/ends of reads

       <b>-n</b>     Don't clip, just output what would be done

       <b>-C</b> N   Number of reads to use for subsampling (300k)

       <b>-S</b>     Save all discarded reads to '.skip' files

       <b>-d</b>     Output lots of random debugging stuff

   <b>Quality</b> <b>adjustment</b> <b>options:</b>
       <b>--cycle-adjust</b>
              CYC,AMT   Adjust cycle CYC (negative = offset from end) by amount AMT

       <b>--phred-adjust</b>
              SCORE,AMT Adjust score SCORE by amount AMT

       <b>--phred-adjust-max</b>
              SCORE     Adjust scores &gt; SCORE to SCOTE

   <b>Filtering</b> <b>options*:</b>
       <b>--[mate-]qual-mean</b>
              NUM       Minimum mean quality score

       <b>--[mate-]qual-gt</b>
              NUM,THR   At least NUM quals &gt; THR

       <b>--[mate-]max-ns</b>
              NUM       Maxmium N-calls in a read (can be a %)

       <b>--[mate-]min-len</b>
              NUM       Minimum remaining length (same as <b>-l</b>)

       <b>--homopolymer-pct</b>
              PCT       Homopolymer filter percent (95)

       <b>--lowcomplex-pct</b>
              PCT       Complexity filter percent (95)

       If mate- prefix is used, then applies to second non-barcode read only

       Adapter files are 'fasta' formatted:

       Specify n/a to turn off adapter clipping, and just use filters

       Increasing the scale makes recognition-lengths longer, a scale of 100 will force full-length  recognition
       of adapters.

       Adapter  sequences  with _5p in their label will match 'end's, and sequences with _3p in their label will
       match 'start's, otherwise the 'end' is auto-determined.

       Skew is when one cycle is poor, 'skewed' toward a particular base.  If any nucleotide is  less  than  the
       skew percentage, then the whole cycle is removed.  Disable for methyl-seq, etc.

       Set  the  skew  (<b>-k</b>)  or  N-pct  (<b>-x</b>)  to  0 to turn it off (should be done for miRNA, amplicon and other
       low-complexity situations!)

       Duplicate read filtering is appropriate for assembly  tasks,  and  never  when  read  length  &lt;  expected
       coverage.  <b>-D</b> 50 will use 4.5GB RAM on 100m DNA reads - be careful. Great for RNA assembly.

       *Quality filters are evaluated after clipping/trimming

       Homopolymer  filtering  is a subset of low-complexity, but will not be separately tracked unless both are
       turned on.

fastq-mcf 1.1.2                                     July 2015                                       <u><a href="../man1/FASTQ-MCF.1.html">FASTQ-MCF</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>