<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::DB::GFF::Aggregator::clone -- Clone aggregator</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-db-gff-perl">libbio-db-gff-perl_1.7.4-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::DB::GFF::Aggregator::clone -- Clone aggregator

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::DB::GFF;

         # Open the sequence database
         my $db      = Bio::DB::GFF-&gt;new( -adaptor =&gt; 'dbi:mysql',
                                          -dsn     =&gt; 'dbi:mysql:elegans42',
                                          -aggregator =&gt; ['transcript','clone'],
                                        );

        ----------------------------------------------------------------------------
        Aggregator method: clone
        Main method:       -none-
        Sub methods:       Clone_left_end Clone_right_end region:Genomic_canonical
        ----------------------------------------------------------------------------

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::DB::GFF::Aggregator::clone is one of the default aggregators, and was written to be compatible with
       the C elegans GFF files.  It aggregates raw "Clone_left_end", "Clone_right_end", and
       "region:Genomic_canonical" features into composite features of type "clone".

   <b>aggregate</b>
        Title   : aggregate
        Usage   : $features = $a-&gt;aggregate($features,$factory)
        Function: aggregate a feature list into composite features
        Returns : an array reference containing modified features
        Args    : see L&lt;Bio::DB::GFF::Aggregator&gt;
        Status  : Public

       The WormBase GFF model is unusual in that clones aren't identified as a single feature with start and
       stop positions, but as two features, a "left end" and a "right end".  One or both of these features may
       be absent.  In order to accommodate this, the aggregator will return undef for the start and/or stop if
       one or both of the ends are missing.

   <b>method</b>
        Title   : method
        Usage   : $aggregator-&gt;method
        Function: return the method for the composite object
        Returns : the string "clone"
        Args    : none
        Status  : Public

   <b>part_names</b>
        Title   : part_names
        Usage   : $aggregator-&gt;part_names
        Function: return the methods for the sub-parts
        Returns : the list ("Clone_left_end", "Clone_right_end", "region:Genomic_canonical")
        Args    : none
        Status  : Public

</pre><h4><b>BUGS</b></h4><pre>
       None reported.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Bio::DB::GFF, Bio::DB::GFF::Aggregator

</pre><h4><b>AUTHOR</b></h4><pre>
       Lincoln Stein &lt;<a href="mailto:lstein@cshl.org">lstein@cshl.org</a>&gt;.

       Copyright (c) 2001 Cold Spring Harbor Laboratory.

       This library is free software; you can redistribute it and/or modify it under the same terms as Perl
       itself.

perl v5.30.0                                       2020-01-13               <u>Bio::DB::GFF::Aggregator::<a href="../man3pm/clone.3pm.html">clone</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>