<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>cufflinks - component of cufflinks suite</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/cufflinks">cufflinks_2.2.1+dfsg.1-10_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       cufflinks - component of cufflinks suite

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>cufflinks</b> [<u>options</u>] <u>&lt;hits.sam&gt;</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       cufflinks v2.2.1 linked against Boost version 108300 <b>-----------------------------</b>

   <b>General</b> <b>Options:</b>
       <b>-o</b>/--output-dir
              write all output files to this directory              [ default:     ./ ]

       <b>-p</b>/--num-threads
              number of threads used during analysis                [ default:      1 ]

       <b>--seed</b> value of random number generator seed                 [ default:      0 ]

       <b>-G</b>/--GTF
              quantitate against reference transcript annotations

       <b>-g</b>/--GTF-guide
              use reference transcript annotation to guide assembly

       <b>-M</b>/--mask-file
              ignore all alignment within transcripts in this file

       <b>-b</b>/--frag-bias-correct
              use bias correction - reference fasta required        [ default:   NULL ]

       <b>-u</b>/--multi-read-correct
              use 'rescue method' for multi-reads (more accurate)   [ default:  FALSE ]

       <b>--library-type</b>
              library prep used for input reads                     [ default:  below ]

       <b>--library-norm-method</b>
              Method used to normalize library sizes                [ default:  below ]

   <b>Advanced</b> <b>Abundance</b> <b>Estimation</b> <b>Options:</b>
       <b>-m</b>/--frag-len-mean
              average fragment length (unpaired reads only)         [ default:    200 ]

       <b>-s</b>/--frag-len-std-dev
              fragment length std deviation (unpaired reads only)   [ default:     80 ]

       <b>--max-mle-iterations</b>
              maximum iterations allowed for MLE calculation        [ default:   5000 ]

       <b>--compatible-hits-norm</b>
              count hits compatible with reference RNAs only        [ default:  FALSE ]

       <b>--total-hits-norm</b>
              count all hits for normalization                      [ default:  TRUE  ]

       <b>--num-frag-count-draws</b>
              Number of fragment generation samples                 [ default:    100 ]

       <b>--num-frag-assign-draws</b>
              Number of fragment assignment samples per generation  [ default:     50 ]

       <b>--max-frag-multihits</b>
              Maximum number of alignments allowed per fragment     [ default: unlim  ]

       <b>--no-effective-length-correction</b>
              No effective length correction                  [ default:  FALSE ]

       <b>--no-length-correction</b>
              No length correction                                  [ default:  FALSE ]

       <b>-N</b>/--upper-quartile-norm
              Deprecated, use <b>--library-norm-method</b>                 [    DEPRECATED   ]

       <b>--raw-mapped-norm</b>
              Deprecated, use <b>--library-norm-method</b>                 [    DEPRECATED   ]

   <b>Advanced</b> <b>Assembly</b> <b>Options:</b>
       <b>-L</b>/--label
              assembled transcripts have this ID prefix             [ default:   CUFF ]

       <b>-F</b>/--min-isoform-fraction
              suppress transcripts below this abundance level       [ default:   0.10 ]

       <b>-j</b>/--pre-mrna-fraction
              suppress intra-intronic transcripts below this level  [ default:   0.15 ]

       <b>-I</b>/--max-intron-length
              ignore alignments with gaps longer than this          [ default: 300000 ]

       <b>-a</b>/--junc-alpha
              alpha for junction binomial test filter               [ default:  0.001 ]

       <b>-A</b>/--small-anchor-fraction
              percent read overhang taken as 'suspiciously small'   [ default:   0.09 ]

       <b>--min-frags-per-transfrag</b>
              minimum number of fragments needed for new transfrags [ default:     10 ]

       <b>--overhang-tolerance</b>
              number of terminal exon bp to tolerate in introns     [ default:      8 ]

       <b>--max-bundle-length</b>
              maximum genomic length allowed for a given bundle     [ default:3500000 ]

       <b>--max-bundle-frags</b>
              maximum fragments allowed in a bundle before skipping [ default: 500000 ]

       <b>--min-intron-length</b>
              minimum intron size allowed in genome                 [ default:     50 ]

       <b>--trim-3-avgcov-thresh</b>
              minimum avg coverage required to attempt 3' trimming  [ default:     10 ]

       <b>--trim-3-dropoff-frac</b>
              fraction of avg coverage below which to trim 3' end   [ default:    0.1 ]

       <b>--max-multiread-fraction</b>
              maximum fraction of allowed multireads per transcript [ default:   0.75 ]

       <b>--overlap-radius</b>
              maximum gap size to fill between transfrags (in bp)   [ default:     50 ]

   <b>Advanced</b> <b>Reference</b> <b>Annotation</b> <b>Guided</b> <b>Assembly</b> <b>Options:</b>
       <b>--no-faux-reads</b>
              disable tiling by faux reads                          [ default:  FALSE ]

       <b>--3-overhang-tolerance</b>
              overhang allowed on 3' end when merging with reference[ default:    600 ]

       <b>--intron-overhang-tolerance</b>
              overhang allowed inside reference intron when merging [ default:     30 ]

   <b>Advanced</b> <b>Program</b> <b>Behavior</b> <b>Options:</b>
       <b>-v</b>/--verbose
              log-friendly verbose processing (no progress bar)     [ default:  FALSE ]

       <b>-q</b>/--quiet
              log-friendly quiet processing (no progress bar)       [ default:  FALSE ]

       <b>--no-update-check</b>
              do not contact server to check for update availability[ default:  FALSE ]

   <b>Supported</b> <b>library</b> <b>types:</b>
              ff-firststrand   ff-secondstrand   ff-unstranded   fr-firststrand   fr-secondstrand  fr-unstranded
              (default) transfrags

   <b>Supported</b> <b>library</b> <b>normalization</b> <b>methods:</b>
              classic-fpkm

cufflinks 2.2.1+dfsg.1-10                          August 2024                                      <u><a href="../man1/CUFFLINKS.1.html">CUFFLINKS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>