<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>mauveAligner - efficiently constructing multiple genome alignments</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/progressivemauve">progressivemauve_1.2.0+4713+dfsg-5build3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       mauveAligner - efficiently constructing multiple genome alignments

</pre><h4><b>SYNOPSIS</b></h4><pre>
       mauveAligner [options] &lt;seq1 filename&gt; &lt;sml1 filename&gt; ...  &lt;seqN filename&gt; &lt;smlN filename&gt;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The mauveAligner and progressiveMauve alignment algorithms have been implemented as command-line programs
       included  with  the  downloadable Mauve software.  When run from the command-line, these programs provide
       options not yet available in the graphical interface.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>--output=</b>&lt;file&gt; Output file name.
              Prints to screen by default

       <b>--mums</b> Find MUMs only, do not attempt to determine locally collinear blocks (LCBs)

       <b>--no-recursion</b> Don't perform recursive anchor identification (implies <b>--no-gapped-alignment</b>)

       <b>--no-lcb-extension</b> If determining LCBs, don't attempt to extend the LCBs

       <b>--seed-size=</b>&lt;number&gt; Initial seed match size, default is log_2( average seq. length )

       <b>--max-extension-iterations=</b>&lt;number&gt; Limit LCB extensions to this number of attempts, default is 4

       <b>--eliminate-inclusions</b> Eliminate linked inclusions in subset matches.

       <b>--weight=</b>&lt;number&gt; Minimum LCB weight in base pairs per sequence

       <b>--match-input=</b>&lt;file&gt; Use specified match file instead of searching for matches

       <b>--lcb-match-input</b>  Indicates that the match input file contains matches that have been clustered into
       LCBs

       <b>--lcb-input=</b>&lt;file&gt; Use specified lcb file instead of constructing LCBs (skips LCB generation)

       <b>--scratch-path=</b>&lt;path&gt;  For large genomes, use a directory for storage of temporary data.  Should be given
       two or more times to with different paths.

       <b>--id-matrix=</b>&lt;file&gt; Generate LCB stats and write them to the specified file

       <b>--island-size=</b>&lt;number&gt; Find islands larger than the given number

       <b>--island-output=</b>&lt;file&gt; Output islands the given file (requires <b>--island-size</b>)

       <b>--backbone-size=</b>&lt;number&gt; Find stretches of backbone longer than the given number of b.p.

       <b>--max-backbone-gap=</b>&lt;number&gt; Allow backbone to be interrupted by gaps up to this length in b.p.

       <b>--backbone-output=</b>&lt;file&gt; Output islands the given file (requires <b>--island-size</b>)

       <b>--coverage-output=</b>&lt;file&gt; Output a coverage list to the specified file (- for stdout)

       <b>--repeats</b> Generates a repeat map.  Only one sequence can be specified

       <b>--output-guide-tree=</b>&lt;file&gt; Write out a guide tree to the designated file

       <b>--collinear</b> Assume that input sequences are collinear--they have no rearrangements

   <b>Gapped</b> <b>alignment</b> <b>controls:</b>
       <b>--no-gapped-alignment</b> Don't perform a gapped alignment

       <b>--max-gapped-aligner-length=</b>&lt;number&gt; Maximum number of base pairs to attempt aligning with the gapped
       aligner

       <b>--min-recursive-gap-length=</b>&lt;number&gt; Minimum size of gaps that Mauve will perform recursive MUM anchoring
       on (Default is 200)

   <b>Signed</b> <b>permutation</b> <b>matrix</b> <b>options:</b>
       <b>--permutation-matrix-output=</b>&lt;file&gt; Write out the LCBs as a signed permutation matrix to the given file

       <b>--permutation-matrix-min-weight=</b>&lt;number&gt; A permutation matrix will be written for every set of LCBs with
       weight between this value and the value of <b>--weight</b>

   <b>Alignment</b> <b>output</b> <b>options:</b>
       <b>--alignment-output-dir=</b>&lt;directory&gt; Outputs a set of alignment files (one per LCB) to a given directory

       <b>--alignment-output-format=</b>&lt;directory&gt; Selects the output format for <b>--alignment-output-dir</b>

       <b>--output-alignment=</b>&lt;file&gt; Write out an XMFA format alignment to the designated file

       Supported alignment output formats are: phylip, clustal, msf, nexus, mega, codon

mauveAligner 1.2.0+4713                            April 2015                                    <u><a href="../man1/MAUVEALIGNER.1.html">MAUVEALIGNER</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>