<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>dialign2-2 - Multiple alignment program using the segment-to-segment approach</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/dialign">dialign_2.2.1-13_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       dialign2-2 - Multiple alignment program using the segment-to-segment approach

</pre><h4><b>SYNOPSIS</b></h4><pre>

       <b>dialign2-2</b> [options] [seq_file]

       <u>seq_file</u> is the name of the input sequence file; this must be a multiple FASTA file (all sequences in one
       file).

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>dialign2-2</b> is a program that constructs alignments from gapfree pairs of similar segments of the
       sequences. If (possibly) coding nucleic acid sequences are to be aligned, DIALIGN optionally translates
       the compared `nucleic acid segments´ to `peptide segments´ according to the genetic code -- without
       presupposing any of the three possible reading frames, so all combinations of reading frames get checked
       for significant similarity.

       By default, DIALIGN creates a single file containing

       •   An alignment of the input sequences in DIALIGN format.

       •   The same alignment in FASTA format.

       •   A sequence tree in PHYLIP format. This tree is constructed by applying the UPGMA clustering method to
           the DIALIGN similarity scores. It roughly reflects the different degrees of similarity among
           sequences. For detailed phylogenetic analysis, we recommend the usual methods for phylogenetic
           reconstruction.

       The format of the output files is documented in <u>/usr/share/doc/dialign/USER_GUIDE.gz</u>. The FASTA, CLUSTALW
       and MSF output formats are optionally available (see OPTIONS).

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-afc</b>
           Creates additional output file "*.afc" containing data of all fragments considered for alignment.
           WARNING: this file can be HUGE!

       <b>-afc_v</b>
           Like "<b>-afc</b>" but verbose: fragments are explicitly printed. WARNING: this file can be EVEN BIGGER!

       <b>-anc</b>
           Anchored alignment. Requires a file <u>seq_file.anc</u> containing anchor points.

       <b>-cs</b>
           If segments are translated, not only the `Watson strand´ but also the `Crick strand´ is looked at.

       <b>-cw</b>
           Additional output file in CLUSTAL W format.

       <b>-ds</b>
           `DNA alignment speed up´. Non-translated nucleic acid fragments are taken into account only if they
           start with at least two matches. Speeds up DNA alignment at the expense of sensitivity.

       <b>-fa</b>
           Additional output file in FASTA format.

       <b>-ff</b>
           Creates file <u>*.frg</u> containing information about all fragments that are part of the respective optimal
           pairwise alignmnets plus information about consistency in the multiple alignment.

       <b>-fn</b> <u>out_file</u>
           Output files are named <u>out_file.extension</u>.

       <b>-fop</b>
           Creates file <u>*.fop</u> containing coordinates of all fragments that are part of the respective pairwise
           alignments.

       <b>-fsm</b>
           Creates file <u>*.fsm</u> containing coordinates of all fragments that are part of the final alignment

       <b>-iw</b>
           Overlap weights switched off (by default, overlap weights are used if up to 35 sequences are
           aligned). This option speeds up the alignment but may lead to reduced alignment quality.

       <b>-lgs</b>
           `Long genomic sequences´ - combines the following options: <b>-ma</b>, <b>-thr</b> <u>2</u>, <b>-lmax</b> <u>30</u>, <b>-smin</b> <u>8</u>, <b>-nta</b>, <b>-ff</b>,
           <b>-fop</b>, <b>-ff</b>, <b>-cs</b>, <b>-ds</b>, <b>-pst</b>.

       <b>-lgs_t</b>
           Like "<b>-lgs</b>" but with all segment pairs assessed at the peptide level (rather than ´mixed alignments´
           as with the "-lgs" option). Therefore faster than -lgs but not very sensitive for non-coding regions.

       <b>-lmax</b> <u>x</u>
           Maximum fragment length = <u>x</u> (default: <u>x</u> = 40 or <u>x</u> = 120 for `translated´ fragments). Shorter <u>x</u> speeds
           up the program but may affect alignment quality.

       <b>-lo</b>
           (Long Output) Additional file <u>*.log</u> with information abut fragments selected for pairwise alignment
           and about consistency in multi-alignment proceedure.

       <b>-ma</b>
           `mixed alignments´ consisting of P-fragments and N-fragments if nucleic acid sequences are aligned.

       <b>-mask</b>
           Residues not belonging to selected fragments are replaced by `*´ characters in output alignment
           (rather than being printed in lower-case characters)

       <b>-mat</b>
           Creates file <u>*mat</u> with substitution counts derived from the fragments that have been selected for
           alignment.

       <b>-mat_thr</b> <u>t</u>
           Like "-mat" but only fragments with weight score &gt; <u>t</u> are considered.

       <b>-max_link</b>
           "Maximum linkage" clustering used to construct sequence tree (instead of UPGMA).

       <b>-min_link</b>
           "Minimum linkage" clustering used.

       <b>-mot</b>
           "Motif" option.

       <b>-msf</b>
           Separate output file in MSF format.

       <b>-n</b>
           Input sequences are nucleic acid sequences. No translation of fragments.

       <b>-nt</b>
           Input sequences are nucleic acid sequences and `nucleic acid segments´ are translated to `peptide
           segments´.

       <b>-nta</b>
           `No textual alignment´. Textual alignment suppressed. This option makes sense if other output files
           are of intrest -- e.g. the fragment files created with <b>-ff</b>, <b>-fop</b>, <b>-fsm</b> or <b>-lo</b>.

       <b>-o</b>
           Fast version, resulting alignments may be slightly different.

       <b>-ow</b>
           Overlap weights enforced (By default, overlap weights are used only if up to 35 sequences are aligned
           since calculating overlap weights is time consuming). Warning: overlap weights generally improve
           alignment quality but the running time increases in the order O(n^4) with the number of sequences.
           This is why, by default, overlap weights are used only for sequence sets with &lt; 36 sequences.

       <b>-pst</b>
           "Print status". Creates and updates a file <u>*.sta</u> with information about the current status of the
           program run. This option is recommended if large data sets are aligned since it allows the user to
           estimate the remaining running time.

       <b>-smin</b> <u>x</u>
           Minimum similarity value for first residue pair (or codon pair) in fragments. Speeds up protein
           alignment or alignment of translated DNA fragments at the expense of sensitivity.

       <b>-stars</b> <u>x</u>
           Maximum number of `*´ characters indicating degree of local similarity among sequences. By default,
           no stars are used but numbers between 0 and 9, instead.

       <b>-stdo</b>
           Results written to standard output.

       <b>-ta</b>
           Standard textual alignment printed (overrides suppression of textual alignments in special options,
           e.g. -lgs).

       <b>-thr</b> <u>x</u>
           Threshold T = x.

       <b>-xfr</b>
           "Exclude fragments". List of fragments can be specified that are NOT considered for pairwise
           alignment.

       General remark: If contradictory options are used, subsequent options override previous ones, e.g.:
       dialign2-2 -nt -n <u>seq_file</u> runs the program with the "-n" option (no translation!), while dialign2-2 -n
       -nt <u>seq_file</u> runs it with the "-nt" option (translation!).

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       The full documentation is in <u>/usr/share/doc/dialign/</u>.

       The website of dialign: <a href="http://dialign.gobics.de/">http://dialign.gobics.de/</a>

       DIALIGN2 has been re-implemented in <b><a href="../man1/dialign-tx.1.html">dialign-tx</a></b>(1). See <a href="http://dialign-tx.gobics.de/">http://dialign-tx.gobics.de/</a>

</pre><h4><b>ENVIRONMENT</b> <b>VARIABLES</b></h4><pre>
       You can create an environment variable `<b>DIALIGN2_DIR</b>´ pointing to a directory where the substitution
       matrices are (see FILES). When installed from the Debian package, it is not necessary to set this
       environnement variable to run DIALIGN.

</pre><h4><b>FILES</b></h4><pre>
       DIALIGN2 needs the files <u>tp400_dna</u>, <u>tp400_prot</u>, <u>tp400_trans</u> and <u>BLOSUM</u>. When DIALIGN is installed from
       the Debian package, they are stored in <u>/usr/share/dialign/</u>.

       DIALIGN 2 uses the BLOSUM62 amino acid substitution matrix. In the current version, it is NOT possible to
       replace BLOSUM62 by other similarity matrices.

</pre><h4><b>REFERENCE</b></h4><pre>
       B. Morgenstern (1999). DIALIGN 2: improvement of the segment-to-segment approach to multiple sequence
       alignment. Bioinformatics 15, 211 - 218. Public research assisted by DIALIGN should cite this article.

</pre><h4><b>AUTHORS</b></h4><pre>
       <b>Burkhard</b> <b>Morgenstern</b> &lt;<a href="mailto:bmorgen@gwdg.de">bmorgen@gwdg.de</a>&gt;
           Author of DIALIGN

       <b>Said</b> <b>Abdeddaim</b>
           Author of DIALIGN

       <b>Charles</b> <b>Plessy</b> &lt;<a href="mailto:plessy@debian.org">plessy@debian.org</a>&gt;
           Wrote this manpage

</pre><h4><b>COPYRIGHT</b></h4><pre>
       DIALIGN  was  written  by  Burkhard  Morgenstern  and Said Abdeddaim at University of Bielefeld (FSPM and
       International Graduate School in Bioinformatics and Genome Research), GSF  (ISG,  IBB,  MIPS/IBI),  North
       Carolina  State  University,  Universite  de  Rouen,  MPI  fuer  Biochemie  (Martinsried),  University of
       Goettingen, Institute of Microbiology and Genetics.

       This manual page was adapted from the DIALIGN manual by Charles Plessy &lt;<a href="mailto:plessy@debian.org">plessy@debian.org</a>&gt; for the Debian
       system (but may be used by others). Permission is granted to copy, distribute and/or modify this document
       under the terms of the GNU Lesser General Public License, Version 2.1 any later version published by  the
       Free Software Foundation.

       On  Debian  systems,  the  complete  text  of  the  GNU  Lesser  General  Public  License can be found in
       /usr/share/common-licenses/LGPL.

       Copyright © 1999 Burkhard Morgenstern (for DIALIGN)
       Copyright © 2006, 2007, 2008 Charles Plessy (for this manpage)

dialign2-2 2.2.1                                  avril 9, 2006                                       <u><a href="../man1/DIALIGN.1.html">DIALIGN</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>