<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>needle - Needleman-Wunsch global alignment of two sequences</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/emboss">emboss_6.6.0+dfsg-15ubuntu2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       needle - Needleman-Wunsch global alignment of two sequences

</pre><h4><b>SYNOPSIS</b></h4><pre>

       <b>needle</b> <b>-asequence</b> <u>sequence</u> <b>-bsequence</b> <u>seqall</u> [<b>-datafile</b> <u>matrixf</u>] <b>-gapopen</b> <u>float</u> <b>-gapextend</b> <u>float</u>
              [<b>-endweight</b> <u>boolean</u>] [<b>-endopen</b> <u>float</u>] [<b>-endextend</b> <u>float</u>] <b>-brief</b> <u>boolean</u> <b>-outfile</b> <u>align</u>

       <b>needle</b> <b>-help</b>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>needle</b> is a command line program from EMBOSS (“the European Molecular Biology Open Software Suite”). It
       is part of the "Alignment:Global" command group(s).

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Input</b> <b>section</b>
       <b>-asequence</b> <u>sequence</u>

       <b>-bsequence</b> <u>seqall</u>

       <b>-datafile</b> <u>matrixf</u>
           This is the scoring matrix file used when comparing sequences. By default it is the file 'EBLOSUM62'
           (for proteins) or the file 'EDNAFULL' (for nucleic sequences). These files are found in the 'data'
           directory of the EMBOSS installation.

   <b>Required</b> <b>section</b>
       <b>-gapopen</b> <u>float</u>
           The gap open penalty is the score taken away when a gap is created. The best value depends on the
           choice of comparison matrix. The default value assumes you are using the EBLOSUM62 matrix for protein
           sequences, and the EDNAFULL matrix for nucleotide sequences. Default value: @($(acdprotein)? 10.0 :
           10.0 )

       <b>-gapextend</b> <u>float</u>
           The gap extension, penalty is added to the standard gap penalty for each base or residue in the gap.
           This is how long gaps are penalized. Usually you will expect a few long gaps rather than many short
           gaps, so the gap extension penalty should be lower than the gap penalty. An exception is where one or
           both sequences are single reads with possible sequencing errors in which case you would expect many
           single base gaps. You can get this result by setting the gap open penalty to zero (or very low) and
           using the gap extension penalty to control gap scoring. Default value: @($(acdprotein)? 0.5 : 0.5 )

   <b>Additional</b> <b>section</b>
       <b>-endweight</b> <u>boolean</u>
           Default value: N

       <b>-endopen</b> <u>float</u>
           The end gap open penalty is the score taken away when an end gap is created. The best value depends
           on the choice of comparison matrix. The default value assumes you are using the EBLOSUM62 matrix for
           protein sequences, and the EDNAFULL matrix for nucleotide sequences. Default value: @($(acdprotein)?
           10.0 : 10.0 )

       <b>-endextend</b> <u>float</u>
           The end gap extension, penalty is added to the end gap penalty for each base or residue in the end
           gap. This is how long end gaps are penalized. Default value: @($(acdprotein)? 0.5 : 0.5 )

   <b>Output</b> <b>section</b>
       <b>-brief</b> <u>boolean</u>
           Brief identity and similarity Default value: Y

       <b>-outfile</b> <u>align</u>

</pre><h4><b>BUGS</b></h4><pre>
       Bugs can be reported to the Debian Bug Tracking system (<a href="http://bugs.debian.org/emboss">http://bugs.debian.org/emboss</a>), or directly to
       the EMBOSS developers (<a href="http://sourceforge.net/tracker/">http://sourceforge.net/tracker/</a>?group_id=93650&amp;atid=605031).

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       needle is fully documented via the <b><a href="../man1/tfm.1.html">tfm</a></b>(1) system.

</pre><h4><b>AUTHOR</b></h4><pre>
       <b>Debian</b> <b>Med</b> <b>Packaging</b> <b>Team</b> &lt;<a href="mailto:debian-med-packaging@lists.alioth.debian.org">debian-med-packaging@lists.alioth.debian.org</a>&gt;
           Wrote the script used to autogenerate this manual page.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       This manual page was autogenerated from an Ajax Control Definition of the EMBOSS package. It can be
       redistributed under the same terms as EMBOSS itself.

EMBOSS 6.4.0                                       05/11/2012                                         <u><a href="../man1e/NEEDLE.1e.html">NEEDLE</a></u>(1e)
</pre>
 </div>
</div></section>
</div>
</body>
</html>