<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>pychopper - package documentation</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/python3-pychopper">python3-pychopper_2.7.10-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       pychopper - package documentation

</pre><h4><b>COMMAND</b> <b>LINE</b> <b>TOOLS</b></h4><pre>
</pre><h4><b>FULL</b> <b>API</b> <b>REFERENCE</b></h4><pre>
   <b>pychopper</b>
   <b>pychopper</b> <b>package</b>
   <b>Subpackages</b>
   <b>pychopper.phmm_data</b> <b>package</b>
   <b>Module</b> <b>contents</b>
   <b>pychopper.primer_data</b> <b>package</b>
   <b>Module</b> <b>contents</b>
   <b>pychopper.scripts</b> <b>package</b>
   <b>Submodules</b>
   <b>pychopper.scripts.pychopper</b> <b>module</b>
       <b>pychopper.scripts.pychopper.main()</b>
              Parse command line arguments.

   <b>Module</b> <b>contents</b>
   <b>pychopper.tests</b> <b>package</b>
   <b>Submodules</b>
   <b>pychopper.tests.test_detector</b> <b>module</b>
       <b>class</b> <b>pychopper.tests.test_detector.TestDetector(methodName='runTest')</b>
              Bases: <b>TestCase</b>

              Create  an  instance  of  the  class  that  will use the named test method when executed. Raises a
              ValueError if the instance does not have a method with the specified name.

              <b>testPairAlign()</b>

              <b>testScoreCutoff()</b>

   <b>pychopper.tests.test_regression_simple</b> <b>module</b>
       <b>class</b> <b>pychopper.tests.test_regression_simple.TestIntegration(methodName='runTest')</b>
              Bases: <b>TestCase</b>

              Create an instance of the class that will use the  named  test  method  when  executed.  Raises  a
              ValueError if the instance does not have a method with the specified name.

              <b>testIntegration()</b>
                     Integration test.

              <b>testIntegration_umi()</b>
                     Integration test.

   <b>Module</b> <b>contents</b>
   <b>Submodules</b>
   <b>pychopper.alignment_hits</b> <b>module</b>
       <b>pychopper.alignment_hits.process_hits(hits,</b> <b>max_score)</b>
              Process alignment hits by removing overlaps

   <b>pychopper.chopper</b> <b>module</b>
       <b>pychopper.chopper.analyse_hits(hits,</b> <b>config)</b>
              Segment  reads  based  on alignment hits using dynamic programming.  The algorithm is based on the
              rule that each primer alignment hit can be used only once.  Hence if a segment  is  included,  the
              next one has to be excluded.

       <b>pychopper.chopper.chopper_edlib(reads,</b> <b>primers,</b> <b>config,</b> <b>max_ed,</b> <b>cutoff,</b> <b>pool,</b> <b>min_batch)</b>
              Segment using the edlib/parasail backend

       <b>pychopper.chopper.chopper_phmm(reads,</b> <b>phmm_file,</b> <b>config,</b> <b>cutoff,</b> <b>threads,</b> <b>pool,</b> <b>min_batch)</b>
              Segment using the profile HMM backend

       <b>pychopper.chopper.segments_to_reads(read,</b> <b>segments,</b> <b>keep_primers,</b> <b>bam_tags,</b> <b>detect_umis)</b>
              Convert segments to output reads with annotation

   <b>pychopper.common_structures</b> <b>module</b>
       <b>class</b> <b>pychopper.common_structures.Hit(Ref,</b> <b>RefStart,</b> <b>RefEnd,</b> <b>Query,</b> <b>QueryStart,</b> <b>QueryEnd,</b> <b>Score)</b>
              Bases: <b>tuple</b>

              Create new instance of Hit(Ref, RefStart, RefEnd, Query, QueryStart, QueryEnd, Score)

              <b>Query</b>  Alias for field number 3

              <b>QueryEnd</b>
                     Alias for field number 5

              <b>QueryStart</b>
                     Alias for field number 4

              <b>Ref</b>    Alias for field number 0

              <b>RefEnd</b> Alias for field number 2

              <b>RefStart</b>
                     Alias for field number 1

              <b>Score</b>  Alias for field number 6

       <b>class</b> <b>pychopper.common_structures.Segment(Left,</b> <b>Start,</b> <b>End,</b> <b>Right,</b> <b>Strand,</b> <b>Len)</b>
              Bases: <b>tuple</b>

              Create new instance of Segment(Left, Start, End, Right, Strand, Len)

              <b>End</b>    Alias for field number 2

              <b>Left</b>   Alias for field number 0

              <b>Len</b>    Alias for field number 5

              <b>Right</b>  Alias for field number 3

              <b>Start</b>  Alias for field number 1

              <b>Strand</b> Alias for field number 4

       <b>class</b> <b>pychopper.common_structures.Seq(Id,</b> <b>Name,</b> <b>Seq,</b> <b>Qual,</b> <b>Umi)</b>
              Bases: <b>tuple</b>

              Create new instance of Seq(Id, Name, Seq, Qual, Umi)

              <b>Id</b>     Alias for field number 0

              <b>Name</b>   Alias for field number 1

              <b>Qual</b>   Alias for field number 3

              <b>Seq</b>    Alias for field number 2

              <b>Umi</b>    Alias for field number 4

   <b>pychopper.edlib_backend</b> <b>module</b>
       <b>pychopper.edlib_backend.find_locations(reads,</b> <b>all_primers,</b> <b>max_ed,</b> <b>pool,</b> <b>min_batch)</b>
              Find alignment hits of all primers in all reads using the edlib/parasail backend

       <b>pychopper.edlib_backend.find_umi_single(params)</b>
              Find UMI in a single reads using the edlib/parasail backend

   <b>pychopper.hmmer_backend</b> <b>module</b>
       <b>pychopper.hmmer_backend.find_locations(reads,</b> <b>phmm_file,</b> <b>E,</b> <b>pool,</b> <b>min_batch)</b>
              Find alignment hits of all primers in all reads using the pHMM/nhmmscan backend

   <b>pychopper.parasail_backend</b> <b>module</b>
       <b>pychopper.parasail_backend.first_cigar(cigar)</b>
              Extract details of the first operation in a cigar string.

       <b>pychopper.parasail_backend.pair_align(reference,</b> <b>query,</b> <b>query_name,</b> <b>subs_mat,</b> <b>params)</b>
              Perform pairwise local alignment using parsail-python

       <b>pychopper.parasail_backend.process_alignment(aln,</b> <b>query,</b> <b>query_name,</b> <b>aln_params)</b>
              Process an alignment, extracting score, start and end.

       <b>pychopper.parasail_backend.refine_locations(read,</b> <b>all_primers,</b> <b>locations,</b> <b>aln_params={'gap_extend':</b> <b>1,</b>
       <b>'gap_open':</b> <b>1,</b> <b>'match':</b> <b>1,</b> <b>'mismatch':</b> <b>-2},</b> <b>subs_mat=&lt;parasail.bindings_v2.Matrix</b> <b>object&gt;)</b>
              Refine alignment edges based on local alignment

   <b>pychopper.report</b> <b>module</b>
       <b>class</b> <b>pychopper.report.Report(pdf)</b>
              Bases: <b>object</b>

              Class  for  plotting  utilities  on  the  top of matplotlib. Plots are saved in the specified file
              through the PDF backend.

              <b>Parameters</b>

                     • <b>self</b> -- object.

                     • <b>pdf</b> -- Output pdf.

              <b>Returns</b>
                     The report object.

              <b>Return</b> <b>type</b>
                     <u>Report</u>

              <b>close()</b>
                     Close PDF backend. Do not forget to call this at the end of your script or your output will
                     be damaged!

                     <b>Parameters</b>
                            <b>self</b> -- object

                     <b>Returns</b>
                            None

                     <b>Return</b> <b>type</b>
                            object

              <b>plot_arrays(data_map,</b> <b>title='',</b> <b>xlab='',</b> <b>ylab='',</b> <b>marker='.',</b> <b>legend_loc='best',</b> <b>legend=True,</b>
              <b>vlines=None,</b> <b>vlcolor='green',</b> <b>vlwitdh=0.5)</b>
                     Plot multiple pairs of data arrays.

                     <b>Parameters</b>

                            • <b>self</b> -- object.

                            • <b>data_map</b> -- A dictionary with labels as keys and tupples of data arrays  (x,y)  as
                              values.

                            • <b>title</b> -- Figure title.

                            • <b>xlab</b> -- X axis label.

                            • <b>ylab</b> -- Y axis label.

                            • <b>marker</b> -- Marker passed to the plot function.

                            • <b>legend_loc</b> -- Location of legend.

                            • <b>legend</b> -- Plot legend if True

                            • <b>vlines</b> -- Dictionary with labels and positions of vertical lines to draw.

                            • <b>vlcolor</b> -- Color of vertical lines drawn.

                            • <b>vlwidth</b> -- Width of vertical lines drawn.

                     <b>Returns</b>
                            None

                     <b>Return</b> <b>type</b>
                            object

              <b>plot_bars_simple(data_map,</b> <b>title='',</b> <b>xlab='',</b> <b>ylab='',</b> <b>alpha=0.6,</b> <b>xticks_rotation=0,</b>
              <b>auto_limit=False)</b>
                     Plot simple bar chart from input dictionary.

                     <b>Parameters</b>

                            • <b>self</b> -- object.

                            • <b>data_map</b> -- A dictionary with labels as keys and data as values.

                            • <b>title</b> -- Figure title.

                            • <b>xlab</b> -- X axis label.

                            • <b>ylab</b> -- Y axis label.

                            • <b>alpha</b> -- Alpha value.

                            • <b>xticks_rotation</b> -- Rotation value for x tick labels.

                            • <b>auto_limit</b> -- Set y axis limits automatically.

                     <b>Returns</b>
                            None

                     <b>Return</b> <b>type</b>
                            object

              <b>plot_histograms(data_map,</b> <b>title='',</b> <b>xlab='',</b> <b>ylab='',</b> <b>bins=50,</b> <b>alpha=0.7,</b> <b>legend_loc='best',</b>
              <b>legend=True,</b> <b>vlines=None)</b>
                     Plot histograms of multiple data arrays.

                     <b>Parameters</b>

                            • <b>self</b> -- object.

                            • <b>data_map</b> -- A dictionary with labels as keys and data arrays as values.

                            • <b>title</b> -- Figure title.

                            • <b>xlab</b> -- X axis label.

                            • <b>ylab</b> -- Y axis label.

                            • <b>bins</b> -- Number of bins.

                            • <b>alpha</b> -- Transparency value for histograms.

                            • <b>legend_loc</b> -- Location of legend.

                            • <b>legend</b> -- Plot legend if True.

                            • <b>vlines</b> -- Dictionary with labels and positions of vertical lines to draw.

                     <b>Returns</b>
                            None

                     <b>Return</b> <b>type</b>
                            object

              <b>save_close()</b>
                     Utility method to save and close figure.

   <b>pychopper.seq_utils</b> <b>module</b>
       Utilities manipulating biological sequences and formats. Extensions to biopython functionality.

       <b>pychopper.seq_utils.base_complement(k)</b>
              Return complement of base.

              Performs  the  subsitutions: A&lt;=&gt;T, C&lt;=&gt;G, X=&gt;X for both upper and lower case. The return value is
              identical to the argument for all other values.

              <b>Parameters</b>
                     <b>k</b> -- A base.

              <b>Returns</b>
                     Complement of base.

              <b>Return</b> <b>type</b>
                     str

       <b>pychopper.seq_utils.errs_tab(n)</b>
              Generate list of error rates for qualities less than equal than n.

       <b>pychopper.seq_utils.get_primers(primers)</b>
              Load primers from fasta file

       <b>pychopper.seq_utils.get_runid(desc)</b>
              Parse out runid from sequence description.

       <b>pychopper.seq_utils.mean_qual(quals,</b> <b>qround=False,</b> <b>tab=[1.0,</b> <b>0.7943282347242815,</b> <b>0.6309573444801932,</b>
       <b>0.5011872336272722,</b> <b>0.3981071705534972,</b> <b>0.31622776601683794,</b> <b>0.251188643150958,</b> <b>0.19952623149688797,</b>
       <b>0.15848931924611134,</b> <b>0.12589254117941673,</b> <b>0.1,</b> <b>0.07943282347242814,</b> <b>0.06309573444801933,</b>
       <b>0.05011872336272722,</b> <b>0.039810717055349734,</b> <b>0.03162277660168379,</b> <b>0.025118864315095794,</b> <b>0.0199526231496888,</b>
       <b>0.015848931924611134,</b> <b>0.012589254117941675,</b> <b>0.01,</b> <b>0.007943282347242814,</b> <b>0.00630957344480193,</b>
       <b>0.005011872336272725,</b> <b>0.003981071705534973,</b> <b>0.0031622776601683794,</b> <b>0.0025118864315095794,</b>
       <b>0.001995262314968879,</b> <b>0.001584893192461114,</b> <b>0.0012589254117941675,</b> <b>0.001,</b> <b>0.0007943282347242813,</b>
       <b>0.000630957344480193,</b> <b>0.0005011872336272725,</b> <b>0.00039810717055349735,</b> <b>0.00031622776601683794,</b>
       <b>0.00025118864315095795,</b> <b>0.00019952623149688788,</b> <b>0.00015848931924611142,</b> <b>0.00012589254117941674,</b> <b>0.0001,</b>
       <b>7.943282347242822e-05,</b> <b>6.309573444801929e-05,</b> <b>5.011872336272725e-05,</b> <b>3.9810717055349695e-05,</b>
       <b>3.1622776601683795e-05,</b> <b>2.5118864315095822e-05,</b> <b>1.9952623149688786e-05,</b> <b>1.584893192461114e-05,</b>
       <b>1.2589254117941661e-05,</b> <b>1e-05,</b> <b>7.943282347242822e-06,</b> <b>6.30957344480193e-06,</b> <b>5.011872336272725e-06,</b>
       <b>3.981071705534969e-06,</b> <b>3.162277660168379e-06,</b> <b>2.5118864315095823e-06,</b> <b>1.9952623149688787e-06,</b>
       <b>1.584893192461114e-06,</b> <b>1.2589254117941661e-06,</b> <b>1e-06,</b> <b>7.943282347242822e-07,</b> <b>6.30957344480193e-07,</b>
       <b>5.011872336272725e-07,</b> <b>3.981071705534969e-07,</b> <b>3.162277660168379e-07,</b> <b>2.5118864315095823e-07,</b>
       <b>1.9952623149688787e-07,</b> <b>1.584893192461114e-07,</b> <b>1.2589254117941662e-07,</b> <b>1e-07,</b> <b>7.943282347242822e-08,</b>
       <b>6.30957344480193e-08,</b> <b>5.011872336272725e-08,</b> <b>3.981071705534969e-08,</b> <b>3.162277660168379e-08,</b>
       <b>2.511886431509582e-08,</b> <b>1.9952623149688786e-08,</b> <b>1.5848931924611143e-08,</b> <b>1.2589254117941661e-08,</b> <b>1e-08,</b>
       <b>7.943282347242822e-09,</b> <b>6.309573444801943e-09,</b> <b>5.011872336272715e-09,</b> <b>3.981071705534969e-09,</b>
       <b>3.1622776601683795e-09,</b> <b>2.511886431509582e-09,</b> <b>1.9952623149688828e-09,</b> <b>1.584893192461111e-09,</b>
       <b>1.2589254117941663e-09,</b> <b>1e-09,</b> <b>7.943282347242822e-10,</b> <b>6.309573444801942e-10,</b> <b>5.011872336272714e-10,</b>
       <b>3.9810717055349694e-10,</b> <b>3.1622776601683795e-10,</b> <b>2.511886431509582e-10,</b> <b>1.9952623149688828e-10,</b>
       <b>1.584893192461111e-10,</b> <b>1.2589254117941662e-10,</b> <b>1e-10,</b> <b>7.943282347242822e-11,</b> <b>6.309573444801942e-11,</b>
       <b>5.011872336272715e-11,</b> <b>3.9810717055349695e-11,</b> <b>3.1622776601683794e-11,</b> <b>2.5118864315095823e-11,</b>
       <b>1.9952623149688828e-11,</b> <b>1.5848931924611107e-11,</b> <b>1.2589254117941662e-11,</b> <b>1e-11,</b> <b>7.943282347242821e-12,</b>
       <b>6.309573444801943e-12,</b> <b>5.011872336272715e-12,</b> <b>3.9810717055349695e-12,</b> <b>3.1622776601683794e-12,</b>
       <b>2.5118864315095823e-12,</b> <b>1.9952623149688827e-12,</b> <b>1.584893192461111e-12,</b> <b>1.258925411794166e-12,</b> <b>1e-12,</b>
       <b>7.943282347242822e-13,</b> <b>6.309573444801942e-13,</b> <b>5.011872336272715e-13,</b> <b>3.981071705534969e-13,</b>
       <b>3.162277660168379e-13,</b> <b>2.511886431509582e-13,</b> <b>1.9952623149688827e-13,</b> <b>1.584893192461111e-13])</b>
              Calculate average basecall quality of a read.  Receive the ascii quality  scores  of  a  read  and
              return  the  average  quality for that read First convert Phred scores to probabilities, calculate
              average error probability convert average back to Phred scale

       <b>pychopper.seq_utils.random(size=None)</b>
              Return random floats in the half-open  interval  [0.0,  1.0).  Alias  for  <u>random_sample</u>  to  ease
              forward-porting to the new random API.

       <b>pychopper.seq_utils.readfq(fastq,</b> <b>sample=None,</b> <b>min_qual=0,</b> <b>rfq_sup={})</b>
              Read fastx files.

              This  is  a  generator  function that yields sequtils.Seq objects.  Optionally filter by a minimum
              mean quality (min_qual).  Optionally subsample the fastx file using sample (0.0 - 1.0)

       <b>pychopper.seq_utils.record_size(read,</b> <b>in_format='fastq')</b>
              Calculate record size.

       <b>pychopper.seq_utils.revcomp_seq(seq)</b>
              Reverse complement sequence record

       <b>pychopper.seq_utils.reverse_complement(seq)</b>
              Reverse complement sequence.

              <b>Param</b>  input sequence string.

              <b>Returns</b>
                     reverse-complemented string.

       <b>pychopper.seq_utils.writefq(r,</b> <b>fh)</b>
              Write read to fastq file

   <b>pychopper.utils</b> <b>module</b>
       <b>pychopper.utils.batch(iterable,</b> <b>size)</b>

       <b>pychopper.utils.check_command(cmd)</b>

       <b>pychopper.utils.check_min_hmmer_version(major,</b> <b>minor)</b>

       <b>pychopper.utils.count_fastq_records(fname,</b> <b>size=128000000,</b> <b>opener=&lt;built-in</b> <b>function</b> <b>open&gt;)</b>

       <b>pychopper.utils.hit2bed(hit,</b> <b>read)</b>

       <b>pychopper.utils.parse_config_string(s)</b>

   <b>Module</b> <b>contents</b>
       • <u>Index</u>

       • <u>Module</u> <u>Index</u>

       • <u>Search</u> <u>Page</u>

</pre><h4><b>AUTHOR</b></h4><pre>
       ONT Applications Group

</pre><h4><b>COPYRIGHT</b></h4><pre>
       2024, Oxford Nanopore Technologies Ltd.

2.7.10                                            Jul 31, 2024                                      <u><a href="../man1/PYCHOPPER.1.html">PYCHOPPER</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>