<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>hmm_train  -  Estimate  the  transition  probabilities  of  an HMM, based on multiple -g <gff_fname_list></title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/phast">phast_1.7+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       hmm_train  -  Estimate  the  transition  probabilities  of  an HMM, based on multiple <b>-g</b> &lt;gff_fname_list&gt;
       [OPTIONS] &gt; out.hmm

</pre><h4><b>DESCRIPTION:</b></h4><pre>
       Estimate the transition probabilities of an HMM, based on multiple alignments, sequence annotations,  and
       a category map.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>required</b> <b>options</b>

       <b>-m</b> &lt;msa_fname_list&gt;

              List of multiple sequence alignment files.  Currently, in testing mode, the list must be of length
              one.

       <b>-c</b> &lt;category_map_fname&gt;

              File defining mapping of feature types to category numbers.

       <b>-g</b> &lt;gff_fname_list&gt;

              Files  in  GFF  defining  sequence  features to be used in labeling sites.   Frame of reference of
              feature indices is determined feature-by-feature according to

       'seqname' attribute.
              Filenames must correspond in number and order

              to the elements of &lt;msa_fname_list&gt;.

   <b>alignment</b> <b>options</b>

       <b>-M</b> &lt;msa_length_list&gt;

              (Mutually exclusive with <b>-m</b>) Assume alignments of the specified lengths (comma-separated list) and
              do not not attempt to map the coordinates in the specified GFFs (assume they are  in  the  desired
              coordinate  frame).   This  option  allows  an  HMM  to  be  trained  directly  from GFFs, without
              alignments.  Not permitted with <b>-I</b>.

       <b>-i</b> PHYLIP|FASTA|MPM|SS

              (default SS) Alignment format.

       <b>-R</b> &lt;tag&gt;

              Before estimating transition probabilities, group features  by  &lt;tag&gt;  (e.g.,  "transcript_id"  or
              "exon_id") and reverse complement segments of the alignment corresponding to groups on the reverse
              strand.  Groups must be non-overlapping (see refeature <b>--unique</b>).

   <b>indel</b> <b>options</b>

       <b>-I</b> &lt;indel_cat_list&gt;

       Model indels for specified categories.
              To have

              nonzero  probability  for  the  states corresponding to a specified category range, indels must be
              "clean" (nonoverlapping), must be assignable by parsimony to a single branch in  the  phylogenetic
              tree,  and  must  have lengths that are exact multiples of the category range size.  Avoid <b>-G</b> with
              this option.  If used in training mode, requires <b>-T</b>.

       <b>-t</b> &lt;tree_fname&gt;

              Use the specified tree topology when training for indels.

       <b>-n</b> &lt;nseqs&gt;

              Train an indel model for &lt;nseqs&gt; sequences, despite that the training alignment  has  a  different
              number.  All (non-trivial) gap patterns are assumed to be equally frequent.

   <b>other</b> <b>options</b>

       <b>-q</b>

              Proceed quietly (without updates to stderr).

       <b>-h</b>

              Print this help message and exit.

hmm_train 1.4                                       May 2016                                        <u><a href="../man1/HMM_TRAIN.1.html">HMM_TRAIN</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>