<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>theseus - Maximum likelihood, multiple simultaneous superpositions with statistical analysis</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/theseus">theseus_3.3.0-14build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       <b>theseus</b> - Maximum likelihood, multiple simultaneous superpositions with statistical analysis

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>theseus</b> [options] <u>pdbfile1</u> <u>[pdbfile2</u> <u>...]</u>

       and

       <b>theseus_align</b> [options] -f <u>pdbfile1</u> <u>[pdbfile2</u> <u>...]</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>Theseus</b>  superposes  a  set  of  macromolecular  structures  simultaneously  using  the method of maximum
       likelihood (ML), rather  than  the  conventional  least-squares  criterion.   <b>Theseus</b>  assumes  that  the
       structures  are  distributed  according to a matrix Gaussian distribution and that the eigenvalues of the
       atomic covariance matrix are hierarchically distributed according to an inverse gamma distribution.  This
       ML superpositioning model produces much more  accurate  results  by  essentially  downweighting  variable
       regions of the structures and by correcting for correlations among atoms.

       <b>Theseus</b>  operates in two main modes: (1) a mode for superimposing structures with identical sequences and
       (2) a mode for structures with different sequences but similar structures:

              (1) A mode for superpositioning macromolecules with identical sequences and numbers  of  residues,
              for instance, multiple models in an NMR family or multiple structures from different crystal forms
              of the same protein.

              In this mode, <b>Theseus</b> will read every model in every file on the command line and superpose them.

              Example:

              <b>theseus</b> <u>1s40.pdb</u>

              In the above example, <u>1s40.pdb</u> is a pdb file of 10 NMR models.

              (2)  An  ``alignment'' mode for superpositioning structures with different sequences, for example,
              multiple structures of the cytochrome  c  protein  from  different  species  or  multiple  mutated
              structures of hen egg white lysozyme.

              This  mode  requires  the  user  to  supply  a  sequence  alignment  file  of the structures being
              superpositioned (see option <b>-A</b> and ``FILE FORMATS'' below).  Additionally, it may be necessary  to
              supply a mapfile that tells <b>theseus</b> which PDB structure files correspond to which sequences in the
              alignment  (see option <b>-M</b> and ``FILE FORMATS'' below).  The mapfile is unnecessary if the sequence
              names and corresponding pdb filenames  are  identical.   In  this  mode,  if  there  are  multiple
              structural  models  in  a PDB file, <b>theseus</b> only reads the first model in each file on the command
              line. In other words, <b>theseus</b> treats the files on the command line  as  if  there  were  only  one
              structure per file.

              Example 1:

              <b>theseus</b> -A cytc.aln -M cytc.filemap d1cih__.pdb d1csu__.pdb d1kyow_.pdb

              In  the  above  example,  d1cih__.pdb,  d1csu__.pdb, and d1kyow_.pdb are pdb files of cytochrome c
              domains from the SCOP database.

              Example 2:

              <b>theseus_align</b> -f d1cih__.pdb d1csu__.pdb d1kyow_.pdb

              In this example, the <b>theseus_align</b> script is called  to  do  the  hard  work  for  you.   It  will
              calculate  a  sequence  alignment  and  then  superpose  based  on  that  alignment.   The  script
              <b>theseus_align</b> takes the same options as the <b>theseus</b> program.  Note, the first few  lines  of  this
              script  must  be  modified for your system, since it calls an external multiple sequence alignment
              program to do the alignment.  See the <b>examples/</b> directory  for  more  details,  including  example
              files.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Algorithmic</b> <b>options,</b> <b>defaults</b> <b>in</b> <b>{brackets}:</b>
       <b>--amber</b>
              Do special processing for AMBER8 formatted PDB files

              Most  people  will  never  need  to use this long option, unless you are processing MD traces from
              AMBER.  AMBER puts the atom names in the wrong column in the PDB file.

       <b>-a</b> <b>[</b><u>selection</u><b>]</b>
              Atoms to include in the superposition.  This option takes two types of  arguments,  either  (1)  a
              number  specifying  a  preselected set of atom types, or (2) an explict PDB-style, colon-delimited
              list of the atoms to include.

              For the preselected atom type subsets, the following integer options are available:

               • 0, alpha carbons for proteins, C1´ atoms for nucleic acids
               • 1, backbone
               • 2, all
               • 3, alpha and beta carbons
               • 4, all heavy atoms (no hydrogens)

              Note, only the <b>-a0</b> option is available when superpositioning structures with different sequences.

              To custom select an explicit set of atom types, the atom types must be specified exactly as  given
              in  the PDB file field, including spaces, and the atom-types must encapsulated in quotation marks.
              Multiple atom types must be delimited by a colon.  For example,

              <b>-a</b> <b>`</b> <b>N</b>  <b>:</b> <b>CA</b> <b>:</b> <b>C</b>  <b>:</b> <b>O</b>  <b>'</b>

              would specify the atom types in the peptide backbone.

       <b>-f</b>     Only read the first model of a multi-model PDB file

       <b>-h</b>     Help/usage

       <b>-i</b> <b>[</b><u>nnn</u><b>]</b>
              Maximum iterations, {200}

       <b>-p</b> <b>[</b><u>precision</u><b>]</b>
              Requested relative precision for convergence, {1e-7}

       <b>-r</b> <b>[</b><u>root</u> <u>name</u><b>]</b>
              Root name to be used in naming the output files, {theseus}

       <b>-s</b> <b>[</b><u>n-n:...</u><b>]</b>
              Residue selection (e.g. -s15-45:50-55), {all}

       <b>-S</b> <b>[</b><u>n-n:...</u><b>]</b>
              Residues to exclude (e.g. -S15-45:50-55) {none}

              The previous two options have the same format. Residue (or alignment column) ranges are  indicated
              by  beginning and end separated by a dash.  Multiple ranges, in any arbitrary order, are separated
              by a colon.  Chains may also be selected by giving the chain ID immediately preceding the  residue
              range.   For  example, <b>-sA1-20:A40-71</b> will only include residues 1 through 20 and 40 through 70 in
              chain A. Chains cannot be specified when superposing structures with different sequences.

       <b>-v</b>     use ML variance weighting (no correlations) {default}

   <b>Input/output</b> <b>options:</b>
       <b>-A</b> <b>[</b><u>sequence</u> <u>alignment</u> <u>file</u><b>]</b>
              Sequence alignment file to use as a guide (CLUSTAL or A2M format)

              For use when superposing structures with different sequences.  See ``FILE FORMATS'' below.

       <b>-E</b>     Print expert options

       <b>-F</b>     Print FASTA files of the sequences in PDB files and quit

              A useful option when superposing structures with different sequences.  The files output with  this
              option  can  be  aligned with a multiple sequence alignment program such as CLUSTAL or MUSCLE, and
              the resulting output alignment file used as <b>theseus</b> input with the <b>-A</b> option.

       <b>-h</b>     Help/usage

       <b>-I</b>     Just calculate statistics for input file; don't superpose

       <b>-M</b> <b>[</b><u>mapfile</u><b>]</b>
              File that maps PDB files to sequences in the alignment.

              A simple two-column formatted file; see ``FILE FORMATS'' below. Used with mode 2.

       <b>-n</b>     Don't write transformed pdb file

       <b>-o</b> <b>[</b><u>reference</u> <u>structure</u><b>]</b>
              Reference file to superpose on, all rotations are relative to the first model in this file

              For example, 'theseus -o cytc1.pdb cytc1.pdb cytc2.pdb cytc3.pdb' will  superpose  the  structures
              and  rotate  the  entire  final  superposition so that the structure from cytc1.pdb is in the same
              orientation as the structure in the original cytc1.pdb PDB file.

       <b>-V</b>     Version

   <b>Principal</b> <b>components</b> <b>analysis:</b>
       <b>-C</b>     Use covariance matrix for PCA (correlation matrix is default)

       <b>-P</b> <b>[</b><u>nnn</u><b>]</b>
              Number of principal components to calculate {0}

              In both of the above, the corresponding principal component is written in the  B-factor  field  of
              the output PDB file. Usually only the first few PCs are of any interest (maybe up to six).

               EXAMPLES <b>theseus</b> <u>2sdf.pdb</u>

       <b>theseus</b> -l -r new2sdf <u>2sdf.pdb</u>

       <b>theseus</b> -s15-45 -P3 <u>2sdf.pdb</u>

       <b>theseus</b> -A <u>cytc.aln</u> -M <u>cytc.mapfile</u> -o <u>cytc1.pdb</u> -s1-40 <u>cytc1.pdb</u> <u>cytc2.pdb</u> <u>cytc3.pdb</u> <u>cytc4.pdb</u>

</pre><h4><b>ENVIRONMENT</b></h4><pre>
       You  can  set  the  environment  variable 'PDBDIR' to your PDB file directory and <b>theseus</b> will look there
       after the present working directory.  For example, in the C shell (tcsh or csh), you  can  put  something
       akin to this in your .cshrc file:

       setenv PDBDIR '/usr/share/pdbs/'

</pre><h4><b>FILE</b> <b>FORMATS</b></h4><pre>
       <b>Theseus</b>  will  read standard PDB formatted files (see &lt;<a href="http://www.rcsb.org/pdb/">http://www.rcsb.org/pdb/</a>&gt;).  Every effort has been
       made for the program to accept nonstandard CNS and X-PLOR file formats also.

       Two other files deserve mention, a sequence alignment file and a mapfile.

   <b>Sequence</b> <b>alignment</b> <b>file</b>
       When  superposing  structures  with  different  residue  identities  (where  the  lengths  of  each   the
       macromolecules  in  terms  of  residues  are  not  necessarily  equal), a sequence alignment file must be
       included for <b>theseus</b> to use as a guide (specified by the <b>-A</b> option).  <b>Theseus</b> accepts  both  CLUSTAL  and
       A2M (FASTA) formatted multiple sequence alignment files.

       NOTE  1:  The  residue  sequence  in  the  alignment must match exactly the residue sequence given in the
       coordinates of the PDB file. That is, there can be no missing or extra residues that do not correspond to
       the sequence in the PDB file. An easy way to ensure that your sequences exactly match the PDB files is to
       generate the sequences using <b>theseus'</b> <b>-F</b> option, which writes out a FASTA formatted sequence file of  the
       chain(s) in the PDB files. The files output with this option can then be aligned with a multiple sequence
       alignment  program  such  as  CLUSTAL  or MUSCLE, and the resulting output alignment file used as <b>theseus</b>
       input with the <b>-A</b> option.

       NOTE 2: Every PDB file must have a corresponding sequence in the alignment.  However, not every  sequence
       in  the  alignment  needs  to have a corresponding PDB file. That is, there can be extra sequences in the
       alignment that are not used for guiding the superposition.

   <b>PDB</b> <b>-&gt;</b> <b>Sequence</b> <b>mapfile</b>
       If the names of the PDB files and  the  names  of  the  corresponding  sequences  in  the  alignemnt  are
       identical, the mapfile may be omitted.  Otherwise, <b>Theseus</b> needs to know which sequences in the alignment
       file  correspond  to  which  PDB  structure  files. This information is included in a mapfile with a very
       simple format (specified with the <b>-M</b> option). There are only two columns  separated  by  whitespace:  the
       first  column lists the names of the PDB structure files, while the second column lists the corresponding
       sequence names exactly as given in the multiple sequence alignment file.

       An example of the mapfile:

       cytc1.pdb    seq1
       cytc2.pdb    seq2
       cytc3.pdb    seq3

</pre><h4><b>SCREEN</b> <b>OUTPUT</b></h4><pre>
       Theseus provides output describing both the progress of the superposing and several  statistics  for  the
       final result:

       <b>Classical</b> <b>LS</b> <b>pairwise</b> <b>&lt;RMSD&gt;:</b>
              The  conventional  RMSD  for  the superposition, the average RMSD for all pairwise combinations of
              structures in the ensemble.

       <b>Least-squares</b> <b>&lt;sigma&gt;:</b>
              The standard deviation  for  the  superposition,  based  on  the  conventional  assumption  of  no
              correlation and equal variances. Basically equal to the RMSD from the average structure.

       <b>Maximum</b> <b>Likelihood</b> <b>&lt;sigma&gt;:</b>
              The ML analog of the standard deviation for the superposition. When assuming that the correlations
              are  zero (a diagonal covariance matrix), this is equal to the square root of the harmonic average
              of the variances for each atom. In contrast, the ``Least-squares &lt;sigma&gt;'' given above reports the
              square root of the arithmetic average of the variances.  The harmonic average is always less  than
              the  arithmetic  average,  and the harmonic average downweights large values proportional to their
              magnitude. This makes sense statistically, because when combining values one should weight them by
              the reciprocal of their variance (which is in fact what the ML superposing method does).

       <b>Marginal</b> <b>Log</b> <b>Likelihood:</b>
              The final marginal log likelihood of the superposition, assuming the matrix Gaussian  distribution
              of  the  structures  and  the  hierarchical  inverse  gamma distribution of the eigenvalues of the
              covariance matrix.  The marginal log likelihood is  the  likelihood  with  the  covariance  matrix
              integrated out.

       <b>AIC:</b>   The  Akaike  Information  Criterion for the final superposition. This is an important statistic in
              likelihood analysis and model selection theory. It allows  an  objective  comparison  of  multiple
              theoretical  models  with different numbers of parameters. In this case, the higher the number the
              better. There is a tradeoff between fit to the data  and  the  number  of  parameters  being  fit.
              Increasing  the  number of parameters in a model will always give a better fit to the data, but it
              also increases the uncertainty of  the  estimated  values.   The  AIC  criterion  finds  the  best
              combination  by (1) maximizing the fit to the data while (2) minimizing the uncertainty due to the
              number of parameters. In the superposition case, one can compare the least  squares  superposition
              to the maximum likelihood superposition. The method (or model) with the higher AIC is preferred. A
              difference in the AIC of 2 or more is considered strong statistical evidence for the better model.

       <b>BIC:</b>   The Bayesian Information Criterion. Similar to the AIC, but with a Bayesian emphasis.

       <b>Omnibus</b> <b>chi2:</b>
              The  overall  reduced  chi2  statistic  for the entire fit, including the rotations, translations,
              covariances, and the inverse gamma parameters. This is probably the most important  statistic  for
              the  superposition. In some cases, the inverse gamma fit may be poor, yet the overall fit is still
              very good. Again, it should ideally be close to 1.0, which would indicate a perfect fit.  However,
              if  you think it is too large, make sure to compare it to the chi2 for the least-squares fit; it's
              probably not that bad after all.  A large chi2 often indicates a violation of the  assumptions  of
              the model.  The most common violation is when superposing two or more independent domains that can
              rotate  relative  to  each  other.  If  this  is  the case, then there will likely be not just one
              Gaussian distribution, but several mixed Gaussians, one for each domain.  Then, it would be better
              to superpose each domain independently.

       <b>Hierarchical</b> <b>var</b> <b>(alpha,</b> <b>gamma)</b> <b>chi2:</b>
              The reduced chi2 for the inverse gamma fit of the covariance matrix  eigenvalues.  As  before,  it
              should  ideally  be  close  to 1.0.  The two values in the parentheses are the ML estimates of the
              scale and shape parameters, respectively, for the inverse gamma distribtuion.

       <b>Rotational,</b> <b>translational,</b> <b>covar</b> <b>chi2:</b>
              The reduced chi2 statistic for the fit of the structures to the model.  With a good fit it  should
              be  close to 1.0, which indicates a perfect fit of the data to the statistical model.  In the case
              of least-squares, the assumed model is a matrix Gaussian distribution of the structures with equal
              variances and no correlations.  For the ML fits, the assumed model is  unequal  variances  and  no
              correlations, as calculated with the <b>-v</b> option [default].  This statistic is for the superposition
              only,  and  does  not  include  the  fit  of the covariance matrix eigenvalues to an inverse gamma
              distribution.  See ``Omnibus chi2'' below.

       <b>Hierarchical</b> <b>minimum</b> <b>var:</b>
              The hierarchical fit of the inverse gamma distribution constrains the variances of  the  atoms  by
              making  large  ones  smaller  and  small ones larger.  This statistic reports the minimum possible
              variance given the inferred inverse gamma parameters.

       <b>skewness,</b> <b>skewness</b> <b>Z-value,</b> <b>kurtosis</b> <b>&amp;</b> <b>kurtosis</b> <b>Z-value:</b>
              The skewness and kurtosis of the residuals. Both should be 0.0 if the  residuals  fit  a  Gaussian
              distribution  perfectly.   They  are  followed  by  the P-value for the statistics. This is a very
              stringent test; residuals can be very non-Gaussian and yet the estimated rotations,  translations,
              and covariance matrix may still be rather accurate.

       <b>Data</b> <b>pts,</b> <b>Free</b> <b>params,</b> <b>D/P:</b>
              The  total number of data points given all observed structures, the number of parameters being fit
              in the model, and the data-to-parameter ratio.

       <b>Median</b> <b>structure:</b>
              The structure that is overall most similar to the average structure. This can be considered to  be
              the most ``typical'' structure in the ensemble.

       <b>Total</b> <b>rounds:</b>
              The number of iterations that the algorithm took to converge.

       <b>Fractional</b> <b>precision:</b>
              The actual precision that the algorithm converged to.

</pre><h4><b>OUTPUT</b> <b>FILES</b></h4><pre>
       Theseus writes out the following files:

       <b>theseus_sup.pdb</b>
              The final superposition, rotated to the principle axes of the mean structure.

       <b>theseus_ave.pdb</b>
              The estimate of the mean structure.

       <b>theseus_residuals.txt</b>
              The normalized residuals of the superposition. These can be analyzed for deviations from normality
              (whether  they  fit  a  standard  Gaussian  distribution).  E.g., the chi2, skewness, and kurtosis
              statistics are based on these values.

       <b>theseus_transf.txt</b>
              The final transformation rotation matrices and translation vectors.

       <b>theseus_variances.txt</b>
              The vector of estimated variances for each atom.

       When Principal Components are calculated (with the <b>-P</b> option), the following files are also produced:

       <b>theseus_pcvecs.txt</b>
              The principal component vectors.

       <b>theseus_pcstats.txt</b>
              Simple statistics for each principle component (loadings, variance explained, etc.).

       <b>theseus_pcN_ave.pdb</b>
              The average structure with the Nth principal component written in the temperature factor field.

       <b>theseus_pcN.pdb</b>
              The final superposition with the Nth principal component written in the temperature factor  field.
              This file is omitted when superposing molecules with different residue sequences (mode 2).

       <b>theseus_cor.mat,</b> <b>theseus_cov.mat</b>
              The  atomic  correlation  matrix  and  covariance  matrices, based on the final superposition. The
              format is suitable for input to GNU's <b>octave.</b>  These  are  the  matrices  used  in  the  Principal
              Components Analysis.

</pre><h4><b>BUGS</b></h4><pre>
       Please send me (DLT) reports of all problems.

</pre><h4><b>RESTRICTIONS</b></h4><pre>
       <b>Theseus</b>  is  <u>not</u>  a  structural  alignment  program.  The structure-based alignment problem is completely
       different from the structural superposition problem.  In order to do a  structural  superposition,  there
       must  be  a  1-to-1  mapping  that  associates  the  atoms  in  one structure with the atoms in the other
       structures.  In the simplest case, this means that structures must have equivalent numbers of atoms, such
       as the models in an NMR PDB file.  For structures with different numbers of  residues/atoms,  superposing
       is  only  possible  when the sequences have been aligned previously.  Finding the best sequence alignment
       based on only structural information is a difficult problem, and one for  which  there  is  currently  no
       maximum likelihood approach.  Extending <b>theseus</b> to address the structural alignment problem is an ongoing
       research project.

</pre><h4><b>AUTHOR</b></h4><pre>
       Douglas L. Theobald
       <a href="mailto:dtheobald@brandeis.edu">dtheobald@brandeis.edu</a>

</pre><h4><b>CITATION</b></h4><pre>
       When using <b>theseus</b> in publications please cite:

       Douglas L. Theobaldand Phillip A. Steindel (2012)
       ``Optimal simultaneous superpositioning of multiple structures with missing data.''
       Bioinformatics <a href="../man15/28.15.html">28</a>(15):1972-1979

       The following papers also report <b>theseus</b> developments:

       Douglas L. Theobald and Deborah S. Wuttke (2008)
       ``Accurate structural correlations from maximum likelihood superpositions.''
       PLoS Computational Biology <a href="../man2/4.2.html">4</a>(2):e43

       Douglas L. Theobald and Deborah S. Wuttke (2006)
       ``THESEUS: Maximum likelihood superpositioning and analysis of macromolecular structures."
       Bioinformatics <a href="../man17/22.17.html">22</a>(17):2171-2172

       Douglas L. Theobald and Deborah S. Wuttke (2006)
       ``Empirical Bayes models for regularizing maximum likelihood estimation in the matrix Gaussian Procrustes
       problem.''
       PNAS <a href="../man49/103.49.html">103</a>(49):18521-18527

</pre><h4><b>HISTORY</b></h4><pre>
       Long, tedious, and sordid.

Brandeis University                               25 March 2015                                       <u><a href="../man1/THESEUS.1.html">THESEUS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>