<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>RNAeffective - calculation of effective numbers of orthologous miRNA targets</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/rnahybrid">rnahybrid_2.1.2-8_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       RNAeffective - calculation of effective numbers of orthologous miRNA targets

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>RNAeffective</b>  <b>[-h]</b> <b>[-d</b> <u>frequency_file</u><b>]</b> <b>[-f</b> <u>from,to</u><b>]</b> <b>[-k</b> <u>sample_size</u><b>]</b> <b>[-l</b> <u>mean,std</u><b>]</b> <b>[-m</b> <u>max_target_length</u><b>]</b>
       <b>[-n</b> <u>max_query_length</u><b>]</b> <b>[-u</b> <u>iloop_upper_limit</u><b>]</b> <b>[-v</b> <u>bloop_upper_limit</u><b>]</b> <b>[-s]</b> <b>[-t</b> <u>target_file</u><b>]</b> <b>[-q</b> <u>query_file</u><b>]</b>
       <b>[</b><u>query</u><b>]</b>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <u>RNAeffective</u> is a tool for determining the effective number of orthologous miRNA  targets.   This  number
       can  be  used for the calculation of more accurate joint p-values in multi-species analyses. RNAeffective
       searches a set of target sequences with random miRNAs that can be given on the command line or  otherwise
       generates   random  sequences  according  to  given  sample  size,  length  distribution  parameters  and
       dinucleotide frequencies. The empirical distribution of  joint  p-values  is  compared  to  the  p-values
       themselves, and the effective number of independent targets is the one that reduces the deviation between
       the two distributions.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>     Give a short summary of command line options.

       <b>-d</b> <u>frequency_file</u>
              Generate  random  sequences  according  to  dinucleotide  frequencies given in <u>frequency_file</u>. See
              example directory for example files.

       <b>-f</b> <u>from,to</u>
              Forces all structures to have a helix from position <u>from</u> to position <u>to</u> with respect to the query.
              The first base has position 1.

       <b>-k</b> <u>sample_size</u>
              Generate <u>sample_size</u> random sequences. Default value is 5000.

       <b>-l</b> <u>mean,std</u>
              Generate random sequences with a normal length distribution of mean <u>mean</u>  and  standard  deviation
              <u>std</u>. Default values are 22 and 0, respectively.

       <b>-m</b> <u>max_target_length</u>
              The  maximum  allowed length of a target sequence. The default value is 2000. This option only has
              an effect if a target file is given with the -t option (see below).

       <b>-n</b> <u>max_query_length</u>
              The maximum allowed length of a query sequence. The default value is 30. This option only  has  an
              effect if a query file is given with the -q option (see below).

       <b>-u</b> <u>iloop_upper_limit</u>
              The maximally allowed number of unpaired nucleotides in either side of an internal loop.

       <b>-v</b> <u>bloop_upper_limit</u>
              The maximally allowed number of unpaired nucleotides in a bulge loop.

       <b>-s</b>     Generate random sequences according to the dinucleotide distribution of given queries (either with
              the -q option or on command line. If no -q is given, the last argument to RNAeffective is taken as
              a query). See -q option.

       <b>-q</b> <u>query_file</u>
              Without  the -s option, each of the query sequences in <u>query_file</u> is subject to hybridisation with
              each of the targets (which are  from  the  <u>target_file</u>;  see  -t  below).  The  sequences  in  the
              <u>query_file</u>  have  to be in FASTA format, ie. one line starting with a &gt; and directly followed by a
              name, then one or more following lines with the sequence itself.  Each  individual  sequence  line
              must not have more than 1000 characters.

              With  the  -s  option,  the query (or query file) dinucleotide distribution is counted, and random
              sequences are generated according to this distribution.

              If no -q is given, random sequences are generated as described above (see -d option).

       <b>-t</b> <u>target_file</u>
              See -q option above.

</pre><h4><b>REFERENCES</b></h4><pre>
       The energy parameters are taken from:

       Mathews DH, Sabina J, Zuker M, Turner DH.  "Expanded  sequence  dependence  of  thermodynamic  parameters
       improves prediction of RNA secondary structure" J Mol Biol., 288 (5), pp 911-940, 1999

</pre><h4><b>VERSION</b></h4><pre>
       This man page documents version 2.0 of RNAeffective.

</pre><h4><b>AUTHORS</b></h4><pre>
       Marc Rehmsmeier, Peter Steffen, Matthias Hoechsmann.

</pre><h4><b>LIMITATIONS</b></h4><pre>
       Character dependent energy values are only defined for [acgtuACGTU].  All other characters lead to values
       of zero in these cases.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       RNAhybrid, RNAcalibrate

                                                                                                 <u><a href="../man1/RNAEFFECTIVE.1.html">RNAEFFECTIVE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>