<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Primer3 - Create input for and work with the output from the program primer3</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Primer3 - Create input for and work with the output from the program primer3

</pre><h4><b>SYNOPSIS</b></h4><pre>
        # parse primer3 output to get some data
        # this is also called from Bio::Tools::Run::Primer3
        use Bio::Tools::Primer3;

        # read a primer3 output file
        my $p3 = Bio::Tools::Primer3-&gt;new(-file=&gt;"data/primer3_output.txt");

        # how many results were there?
        my $num = $p3-&gt;number_of_results;
        print "There were $num results\n";

        # get all the results
        my $all_results = $p3-&gt;all_results;
        print "ALL the results\n";
        foreach my $key (keys %{$all_results}) {
           print "$key\t${$all_results}{$key}\n";
        }

        # get specific results
        my $result1 = $p3-&gt;<a href="../man1/primer_results.1.html">primer_results</a>(1);
        print "The first primer is\n";
        foreach my $key (keys %{$result1}) {
           print "$key\t${$result1}{$key}\n";
        }

        # get the results as a Bio::Seq::PrimedSeq stream
        my $primer = $p3-&gt;next_primer;
        print "The left primer in the stream is ",
          $primer-&gt;get_primer('-left_primer')-&gt;seq-&gt;seq, "\n";

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::Tools::Primer3 creates the input files needed to design primers using primer3 and provides
       mechanisms to access data in the primer3 output files.

       This module provides a bioperl interface to the program primer3. See
       <a href="http://www-genome.wi.mit.edu/genome_software/other/primer3.html">http://www-genome.wi.mit.edu/genome_software/other/primer3.html</a> for details and to download the software.

       This module is based on one written by Chad Matsalla (<a href="mailto:bioinformatics1@dieselwurks.com">bioinformatics1@dieselwurks.com</a>)

       I have ripped some of his code, and added a lot of my own. I hope he is not mad at me!

       This is probably best run in one of the two following ways:

         i. To parse the output from Bio::Tools::Run::Primer3.
            You will most likely just use next_primer to get the results from
            Bio::Tools::Run::Primer3.
         ii. To parse the output of primer3 handed to it as a file name.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b></h4><pre>
         Rob Edwards

         <a href="mailto:redwards@utmem.edu">redwards@utmem.edu</a>

         Based heavily on work of

         Chad Matsalla

         <a href="mailto:bioinformatics1@dieselwurks.com">bioinformatics1@dieselwurks.com</a>

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
         Brian Osborne bosborne at alum.mit.edu

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>new</b>
         Title   : new()
         Usage   : my $primer3 = Bio::Tools::Primer3-&gt;new(-file=&gt;$file);
         Function: Parse primer3 output
         Returns : Does not return anything. If called with a filename will
                   allow you to retrieve the results
         Args    : -file (optional) file of primer3 results to parse -verbose
                   (optional) set verbose output
         Notes   :

   <b>number_of_results</b>
         Title   : number_of_results()
         Usage   : my $count = $primer3-&gt;number_of_results();
         Function: Retrieve the number of primers returned from Primer3.
         Returns : A scalar
         Args    : None
         Notes   : This returns the count of the primers returned by Primer3
                    (aka how many of them there are).
                    This is one more than the maximum offset into the zero
                    based list of primers that is accessed by primer_results().

   <b>all_results</b>
         Title   : all_results()
         Usage   : my $results = $primer3-&gt;all_results();
                      or
                   my $results = $primer3-&gt;all_results('primer3 result name', 'other results');
         Function: Retrieve the results returned from Primer3.
         Returns : A reference to a hash
         Args    : Optional array of specific results to retrieve

   <b>primer_results</b>
         Title   : primer_results()
         Usage   : my $results = $primer3-&gt;<a href="../man2/primer_results.2.html">primer_results</a>(2); # results for third primer
         Function: Retrieve the results returned from Primer3 for specific primer pairs.
         Returns : A reference to a hash
         Args    : A number between 0 and the maximum number of primers to retrieve

   <b>_readfile</b>
         Title   : _readfile()
         Usage   : $self-&gt;_readfile();
         Function: An internal function that reads a file and sets up the results
         Returns : Nothing.
         Args    : None
         Notes   :

   <b>next_primer</b>
         Title   : next_primer()
         Usage   : while (my $primed_seq  = $primer3-&gt;next_primer()) {
         Function: Retrieve the primed sequence and a primer pair, one at a time
         Returns : Returns a Bio::Seq::PrimedSeq object, one at a time
         Args    : None
         Notes   : Use $primed_seq-&gt;annotated_seq to get an annotated sequence
                   object you can write out.

   <b>primer_stream</b>
         Title   : primer_stream()
         Usage   : while (my $primed_seq  = $primer3-&gt;primer_stream()) {
         Function: Retrieve the primer/sequences one at a time
         Returns : Returns a Bio::Seq::PrimedSeq object, one at a time
         Args    : None
         Notes   : Deprecated, just a link to next_primer

   <b>_separate</b>
         Title   : _separate()
         Usage   : $self-&gt;_separate();
         Function: An internal function that groups the results by number
                   (e.g. primer pair 1, etc)
         Returns : Nothing.
         Args    : None
         Notes   :

   <b>_set_variable</b>
         Title   : _set_variable()
         Usage   : $self-&gt;_set_variable('variable name', 'value');
         Function: An internal function that sets a variable
         Returns : Nothing.
         Args    : None
         Notes   : Used to set $self-&gt;{results} and $self-&gt;seqobject

perl v5.32.1                                       2021-08-15                           <u>Bio::Tools::<a href="../man3pm/Primer3.3pm.html">Primer3</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>