<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>bamflagsplit - sort BAM files by coordinate or query name</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/biobambam2">biobambam2_2.0.185+ds-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       bamflagsplit - sort BAM files by coordinate or query name

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>bamflagsplit</b> [options]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       bamflagsplit  reads  a name collated BAM, SAM or CRAM file and distributes the contained reads into a set
       of BAM files. The reads are distributed as follows:

       single single end reads are placed in the single file

       unmapped
              (partially) unmapped read pairs (pairs where at least one of the two mates is unmapped) are placed
              in the unmapped file

       orphan orphan reads (pairs where the other mate is missing in the file) are placed in the orphan file

       supplementary
              supplementary mappings (mappings which  have  the  supplementary  flag  set)  are  placed  in  the
              supplementary file

       split  split  pair  reads (pairs such that the two ends map to different reference sequences/chromosomes)
              are placed in the split file

       samestrand
              read pairs such that both ends map to the same strand are placed in the samestrand file

       improper
              improperly mapped read pairs (read pairs where the mapped as a proper pair flag is  not  set)  are
              place in the improper file

       proper properly mapped pairs are placed in the proper file

       The following key=value pairs can be given:

       <b>I=&lt;[stdin]&gt;:</b> input filename, standard input if unset.

       <b>level=&lt;-1|0|1|9|11&gt;:</b> set compression level of the output BAM files. Valid values are

       -1:    zlib/gzip default compression level

       0:     uncompressed

       1:     zlib/gzip level 1 (fast) compression

       9:     zlib/gzip level 9 (best) compression

       If libmaus has been compiled with support for igzip (see https://software.intel.com/en-us/articles/igzip-
       a-high-performance-deflate-compressor-with-optimizations-for-genomic-data) then an additional valid value
       is

       11:    igzip compression

       <b>verbose=&lt;1&gt;:</b> Valid values are

       1:     print progress report on standard error

       0:     do not print progress report

       <b>disablevalidation=&lt;0|1&gt;:</b> sets whether input validation is performed. Valid values are

       0:     validation is enabled (default)

       1:     validation is disabled

       <b>single=&lt;filename&gt;:</b> file name for the single file

       <b>singlemd5=&lt;0|1&gt;:</b> md5 checksum creation for single output file. Valid values are

       0:     do not compute checksum. This is the default.

       1:     compute  checksum.  If  the splitmd5filename key is set, then the checksum is written to the given
              file. If splitmd5filename, then no checksum will be computed.

       <b>unmapped=&lt;filename&gt;:</b> file name for the unmapped file

       <b>unmappedmd5=&lt;0|1&gt;:</b> md5 checksum creation for unmapped output file. Valid values are

       0:     do not compute checksum. This is the default.

       1:     compute checksum. If the splitmd5filename key is set, then the checksum is written  to  the  given
              file. If splitmd5filename, then no checksum will be computed.

       <b>orphan=&lt;filename&gt;:</b> file name for the orphan file

       <b>orphanmd5=&lt;0|1&gt;:</b> md5 checksum creation for orphan output file. Valid values are

       0:     do not compute checksum. This is the default.

       1:     compute  checksum.  If  the splitmd5filename key is set, then the checksum is written to the given
              file. If splitmd5filename, then no checksum will be computed.

       <b>supplementary=&lt;filename&gt;:</b> file name for the supplementary file

       <b>supplementarymd5=&lt;0|1&gt;:</b> md5 checksum creation for supplementary output file. Valid values are

       0:     do not compute checksum. This is the default.

       1:     compute checksum. If the splitmd5filename key is set, then the checksum is written  to  the  given
              file. If splitmd5filename, then no checksum will be computed.

       <b>split=&lt;filename&gt;:</b> file name for the split file

       <b>splitmd5=&lt;0|1&gt;:</b> md5 checksum creation for split output file. Valid values are

       0:     do not compute checksum. This is the default.

       1:     compute  checksum.  If  the splitmd5filename key is set, then the checksum is written to the given
              file. If splitmd5filename, then no checksum will be computed.

       <b>samestrand=&lt;filename&gt;:</b> file name for the samestrand file

       <b>samestrandmd5=&lt;0|1&gt;:</b> md5 checksum creation for samestrand output file. Valid values are

       0:     do not compute checksum. This is the default.

       1:     compute checksum. If the samestrandmd5filename key is set, then the checksum  is  written  to  the
              given file. If samestrandmd5filename, then no checksum will be computed.

       <b>improper=&lt;filename&gt;:</b> file name for the improper file

       <b>impropermd5=&lt;0|1&gt;:</b> md5 checksum creation for improper output file. Valid values are

       0:     do not compute checksum. This is the default.

       1:     compute checksum. If the impropermd5filename key is set, then the checksum is written to the given
              file. If impropermd5filename, then no checksum will be computed.

       <b>proper=&lt;filename&gt;:</b> file name for the proper file

       <b>propermd5=&lt;0|1&gt;:</b> md5 checksum creation for proper output file. Valid values are

       0:     do not compute checksum. This is the default.

       1:     compute  checksum.  If the propermd5filename key is set, then the checksum is written to the given
              file. If propermd5filename, then no checksum will be computed.

       <b>inputformat=&lt;bam&gt;:</b> <b>input</b> <b>file</b> <b>format.</b>  All versions of bamflagsplit come with support for the  BAM  input
       format. If the program in addition is linked to the io_lib package, then the following options are valid:

       bam:   BAM (see <a href="http://samtools.sourceforge.net/SAM1.pdf">http://samtools.sourceforge.net/SAM1.pdf</a>)

       sam:   SAM (see <a href="http://samtools.sourceforge.net/SAM1.pdf">http://samtools.sourceforge.net/SAM1.pdf</a>)

       cram:  CRAM (see <a href="http://www.ebi.ac.uk/ena/about/cram_toolkit">http://www.ebi.ac.uk/ena/about/cram_toolkit</a>)

</pre><h4><b>AUTHOR</b></h4><pre>
       Written by German Tischler.

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       Report bugs to &lt;<a href="mailto:germant@miltenyibiotec.de">germant@miltenyibiotec.de</a>&gt;

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright  ©  2009-2014  German  Tischler,  © 2011-2014 Genome Research Limited.  License GPLv3+: GNU GPL
       version 3 &lt;<a href="http://gnu.org/licenses/gpl.html">http://gnu.org/licenses/gpl.html</a>&gt;
       This is free software: you are free to change and redistribute it.  There is NO WARRANTY, to  the  extent
       permitted by law.

BIOBAMBAM                                           June 2014                                    <u><a href="../man1/BAMFLAGSPLIT.1.html">BAMFLAGSPLIT</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>