<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>fastaq_to_tiling_bam - Make a BAM file of reads uniformly spread across the input reference</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/fastaq">fastaq_3.17.0-9_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       fastaq_to_tiling_bam - Make a BAM file of reads uniformly spread across the input reference

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: fastaq_to_tiling_bam [options] &lt;infile&gt; &lt;read_length&gt; &lt;read_step&gt; &lt;read_prefix&gt; &lt;outfile&gt;

       Takes a sequence file. Makes a BAM file containing perfect (unpaired) reads tiling the whole genome

   <b>positional</b> <b>arguments:</b>
       infile Name of input fasta/q file

       read_length
              Length of reads

       read_step
              Distance between start of each read

       read_prefix
              Prefix of read names

       outfile
              Name of output BAM file

   <b>options:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>--qual_char</b> QUAL_CHAR
              Character to use for quality score [I]

       <b>--read_group</b> READ_GROUP
              Add the given read group ID to all reads [42]

       Important: assumes that samtools is in your path

fastaq_to_tiling_bam 3.17.0                       December 2024                          <u><a href="../man1/FASTAQ-TO_TILING_BAM.1.html">FASTAQ-TO_TILING_BAM</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>