<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>gmap - Genomic Mapping and Alignment Program</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/gmap">gmap_2024-11-20+ds-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       gmap - Genomic Mapping and Alignment Program

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>gmap</b> [<u>OPTIONS</u>...] <u>&lt;FASTA</u> <u>files</u>...<u>&gt;,</u> <u>or</u> cat &lt;FASTA files...&gt; | gmap [OPTIONS...]

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Input</b> <b>options</b> <b>(must</b> <b>include</b> <b>-d</b> or <b>-g</b>)
       <b>-D</b>, <b>--dir</b>=<u>directory</u>
              Genome   directory.   Default  (as  specified  by  <b>--with-gmapdb</b>  to  the  configure  program)  is
              <u>/var/cache/gmap</u>

       <b>-d</b>, <b>--db</b>=<u>STRING</u>
              Genome database.  If argument is '?' (with the quotes), this command lists available databases.

       <b>-k</b>, <b>--kmer</b>=<u>INT</u>
              kmer size to use in genome database (allowed values: 16 or less).  If not specified,  the  program
              will find the highest available kmer size in the genome database

       <b>--sampling</b>=<u>INT</u>
              Sampling  to  use  in  genome  database.   If  not  specified,  the program will find the smallest
              available sampling value in the genome database within selected k-mer size

       <b>-g</b>, <b>--gseg</b>=<u>filename</u>
              User-supplied genomic segments.  If multiple segments are provided, then every query  sequence  is
              aligned against every genomic segment

       <b>-1</b>, <b>--selfalign</b>
              Align  one  sequence  against  itself  in  FASTA  format  via  stdin  (Useful  for getting protein
              translation of a nucleotide sequence)

       <b>-2</b>, <b>--pairalign</b>
              Align two sequences in FASTA format via stdin, first one being genomic and second one being cDNA

       <b>--cmdline</b>=<u>STRING</u>,STRING
              Align these two sequences provided on the command line, first one being  genomic  and  second  one
              being cDNA

       <b>--align-fraction</b>=<u>FLOAT</u>
              Process  only  the  given fraction of reads, selected at random If <b>--align-fraction</b> and <b>--part</b> are
              given, <b>--align-fraction</b> takes precedence

       <b>-q</b>, <b>--part</b>=<u>INT</u>/INT
              Process only the i-th out of every n sequences e.g., 0/100 or 99/100 (useful for distributing jobs
              to a computer farm).

       <b>--input-buffer-size</b>=<u>INT</u>
              Size of input buffer (program reads this many sequences at a time for efficiency) (default 1000)

       Computation options

       <b>-B</b>, <b>--batch</b>=<u>INT</u>
              Batch mode (default = 2)

                       Mode     Positions       Genome
                         0      mmap            mmap
                         1      mmap &amp; preload  mmap
               (default) 2      mmap &amp; preload  mmap &amp; preload
                         3      allocate        mmap &amp; preload
                         4      allocate        allocate
                         5      allocate        allocate     (same as 4)

       Note: For a single sequence, all data structures use mmap
              If mmap not available and allocate not chosen, then will use fileio (very slow)

       <b>--use-shared-memory</b>=<u>INT</u>
              If 1, then allocated memory is shared among all processes on this node If 0 (default),  then  each
              process has private allocated memory

       <b>--nosplicing</b>
              Turns off splicing (useful for aligning genomic sequences onto a genome)

       <b>--max-deletionlength</b>=<u>INT</u>
              Max  length  for  a  deletion  (default 30).  Above this size, a genomic gap will be considered an
              intron rather than a deletion.  If the genomic gap is less than <b>--max-deletionlength</b>  and  greater
              than  <b>--min-intronlength</b>,  a  known splice site or splice site probabilities of 0.80 on both sides
              will be reported as an intron.

       <b>--min-intronlength</b>=<u>INT</u>
              Min length for one internal intron (default 9).  Below this size, a genomic gap will be considered
              a deletion rather than an intron.  If the  genomic  gap  is  less  than  <b>--max-deletionlength</b>  and
              greater  than <b>--min-intronlength</b>, a known splice site or splice site probabilities of 0.80 on both
              sides will be reported as an intron.

       <b>--max-intronlength-middle</b>=<u>INT</u>
              Max length for one internal intron (default 500000).  Note: for backward compatibility, the <b>-K</b>  or
              <b>--intronlength</b> flag will set both <b>--max-intronlength-middle</b> and <b>--max-intronlength-ends</b>.  Also see
              <b>--split-large-introns</b> below.

       <b>--max-intronlength-ends</b>=<u>INT</u>
              Max  length for first or last intron (default 10000).  Note: for backward compatibility, the <b>-K</b> or
              <b>--intronlength</b> flag will set both <b>--max-intronlength-middle</b> and <b>--max-intronlength-ends</b>.

       <b>--split-large-introns</b>
              Sometimes GMAP will exceed the value for <b>--max-intronlength-middle</b>, if  it  finds  a  good  single
              alignment.  However, you can force GMAP to split such alignments by using this flag

       <b>--end-trimming-score</b>=<u>INT</u>
              Trim ends if the alignment score is below this value where a match scores +1 and a mismatch scores
              <b>-3</b>  The value should be 0 (default) or negative.  A negative allows some mismatches at the ends of
              the alignment

       <b>--trim-end-exons</b>=<u>INT</u>
              Trim end exons with fewer than given number of matches (in nt, default 12)

       <b>-w</b>, <b>--localsplicedist</b>=<u>INT</u>
              Max length for known splice sites at ends of sequence (default 2000000)

       <b>-L</b>, <b>--totallength</b>=<u>INT</u>
              Max total intron length (default 2400000)

       <b>-x</b>, <b>--chimera-margin</b>=<u>INT</u>
              Amount of unaligned sequence that  triggers  search  for  the  remaining  sequence  (default  30).
              Enables  alignment of chimeric reads, and may help with some non-chimeric reads.  To turn off, set
              to zero.

       <b>--no-chimeras</b>
              Turns off finding of chimeras.  Same effect as <b>--chimera-margin</b>=<u>0</u>

       <b>-t</b>, <b>--nthreads</b>=<u>INT</u>
              Number of worker threads

       <b>-c</b>, <b>--chrsubset</b>=<u>string</u>
              Limit search to given chromosome

       <b>--strand</b>=<u>STRING</u>
              Genome strand to try aligning to (plus, minus, or both default)

       <b>-z</b>, <b>--direction</b>=<u>STRING</u>
              cDNA direction (sense_force, antisense_force, sense_filter, antisense_filter,or auto (default))

       <b>--canonical-mode</b>=<u>INT</u>
              Reward for canonical and semi-canonical introns  0=low  reward,  1=high  reward  (default),  2=low
              reward for high-identity sequences and high reward otherwise

       <b>--cross-species</b>
              Use  a  more  sensitive  search  for  canonical splicing, which helps especially for cross-species
              alignments and other difficult cases

       <b>--allow-close-indels</b>=<u>INT</u>
              Allow an  insertion  and  deletion  close  to  each  other  (0=no,  1=yes  (default),  2=only  for
              high-quality alignments)

       <b>--microexon-spliceprob</b>=<u>FLOAT</u>
              Allow  microexons only if one of the splice site probabilities is greater than this value (default
              0.95)

       <b>--indel-open</b>
              In dynamic programming, opening penalty for indel

       <b>--indel-extend</b>
              In dynamic programming, extension penalty for indel Values  for  <b>--indel-open</b>  and  <b>--indel-extend</b>
              should  be  in  [-127,-1].   If  value is &lt; <b>-127</b>, then will use <b>-127</b> instead.  If <b>--indel-open</b> and
              <b>--indel-extend</b> are not specified, values are chosen adaptively, based on the  differences  between
              the query and reference

       <b>--cmetdir</b>=<u>STRING</u>
              Directory  for methylcytosine index files (created using cmetindex) (default is location of genome
              index files specified using <b>-D</b>, <b>-V</b>, and <b>-d</b>)

       <b>--atoidir</b>=<u>STRING</u>
              Directory for A-to-I RNA editing index files (created using atoiindex)  (default  is  location  of
              genome index files specified using <b>-D</b>, <b>-V</b>, and <b>-d</b>)

       <b>--mode</b>=<u>STRING</u>
              Alignment    mode:    standard    (default),   cmet-stranded,   cmet-nonstranded,   atoi-stranded,
              atoi-nonstranded, ttoc-stranded, or ttoc-nonstranded.  Non-standard modes  requires  you  to  have
              previously run the cmetindex or atoiindex programs (which also cover the ttoc modes) on the genome

       <b>-p</b>, <b>--prunelevel</b>
              Pruning level: 0=no pruning (default), 1=poor seqs, 2=repetitive seqs, 3=poor and repetitive

       Output types

       <b>-S</b>, <b>--summary</b>
              Show summary of alignments only

       <b>-A</b>, <b>--align</b>
              Show alignments

       <b>-3</b>, <b>--continuous</b>
              Show alignment in three continuous lines

       <b>-4</b>, <b>--continuous-by-exon</b>
              Show alignment in three lines per exon

       <b>-E</b>, <b>--exons</b>=<u>STRING</u>
              Print exons ("cdna" or "genomic") Will also print introns with "cdna+introns" or "genomic+introns"

       <b>-P</b>, <b>--protein_dna</b>
              Print protein sequence (cDNA)

       <b>-Q</b>, <b>--protein_gen</b>
              Print protein sequence (genomic)

       <b>-f</b>, <b>--format</b>=<u>INT</u>
              Other  format  for  output  (also note the <b>-A</b> and <b>-S</b> options and other options listed under Output
              types):
               mask_introns,
               mask_utr_introns,
               psl (or 1) = PSL (BLAT) format,
               gff3_gene (or 2) = GFF3 gene format,
               gff3_match_cdna (or 3) = GFF3 cDNA_match format,
               gff3_match_est (or 4) = GFF3 EST_match format,
               splicesites (or 6) = splicesites output (for GSNAP splicing file),
               introns = introns output (for GSNAP splicing file),
               map_exons (or 7) = IIT FASTA exon map format,
               map_ranges (or 8) = IIT FASTA range map format,
               coords (or 9) = coords in table format,
               sampe = SAM format (setting paired_read bit in flag),
               samse = SAM format (without setting paired_read bit),
               bedpe = indels and gaps in BEDPE format

       Output options

       <b>-n</b>, <b>--npaths</b>=<u>INT</u>
              Maximum number of paths to show (default  5).   If  set  to  1,  GMAP  will  not  report  chimeric
              alignments, since those imply two paths.  If you want a single alignment plus chimeric alignments,
              then set this to be 0.

       <b>--suboptimal-score</b>=<u>FLOAT</u>
              Report only paths whose score is within this value of the best path.

       If specified between 0.0 and 1.0, then treated as a fraction
              of  the  score  of  the  best  alignment  (matches  minus  penalties  for  mismatches and indels).
              Otherwise, treated as an integer number to be subtracted from the score  of  the  best  alignment.
              Default value is 0.50.

       <b>-O</b>, <b>--ordered</b>
              Print output in same order as input (relevant only if there is more than one worker thread)

       <b>-5</b>, <b>--md5</b>
              Print MD5 checksum for each query sequence

       <b>-o</b>, <b>--chimera-overlap</b>
              Overlap to show, if any, at chimera breakpoint

       <b>--failsonly</b>
              Print only failed alignments, those with no results

       <b>--nofails</b>
              Exclude printing of failed alignments

       <b>-V</b>, <b>--snpsdir</b>=<u>STRING</u>
              Directory for SNPs index files (created using snpindex) (default is location of genome index files
              specified using <b>-D</b> and <b>-d</b>)

       <b>-v</b>, <b>--use-snps</b>=<u>STRING</u>
              Use  database  containing  known  SNPs  (in  &lt;STRING&gt;.iit,  built  previously  using snpindex) for
              tolerance to SNPs

       <b>--split-output</b>=<u>STRING</u>
              Basename for multiple-file output, separately for nomapping,
               uniq, mult, (and chimera, if <b>--chimera-margin</b> is selected)

       <b>--failed-input</b>=<u>STRING</u>
              Print completely failed alignments as input FASTA or FASTQ format  to  the  given  file.   If  the
              <b>--split-output</b>  flag  is  also  given,  this  file  is  generated in addition to the output in the
              .nomapping file.

       <b>--append-output</b>
              When <b>--split-output</b> or <b>--failedinput</b> is given, this flag will append output to the existing files.
              Otherwise, the default is to create new files.

       <b>--output-buffer-size</b>=<u>INT</u>
              Buffer size, in queries, for output thread (default 1000).  When  the  number  of  results  to  be
              printed exceeds this size, worker threads wait until the backlog is cleared

       <b>--translation-code</b>=<u>INT</u>
              Genetic  code  used  for  translating  codons  to  amino  acids  and  computing  CDS Integer value
              (default=1)         corresponds         to         an          available          code          at
              <a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi">http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi</a>

       <b>--alt-start-codons</b>
              Also,  use  the  alternate  initiation codons shown in the above Web site By default, without this
              option, only ATG is considered an initiation codon

       <b>-F</b>, <b>--fulllength</b>
              Assume full-length protein, starting with Met

       <b>-a</b>, <b>--cdsstart</b>=<u>INT</u>
              Translate codons from given nucleotide (1-based)

       <b>-T</b>, <b>--truncate</b>
              Truncate alignment around full-length protein, Met to Stop Implies <b>-F</b> flag.

       <b>-Y</b>, <b>--tolerant</b>
              Translates cDNA with corrections for frameshifts

       Options for GFF3 output

       <b>--gff3-add-separators</b>=<u>INT</u>
              Whether to add a ### separator after each query sequence Values: 0 (no), 1 (yes, default)

       <b>--gff3-swap-phase</b>=<u>INT</u>
              Whether to swap phase (0 =&gt; 0, 1 =&gt; 2, 2 =&gt;  1)  in  gff3_gene  format  Needed  by  some  analysis
              programs, but deviates from GFF3 specification Values: 0 (no, default), 1 (yes)

       <b>--gff3-fasta-annotation</b>=<u>INT</u>
              Whether  to  include annotation from the FASTA header into the GFF3 output Values: 0 (default): Do
              not include

       1: Wrap all annotation as Annot="&lt;header&gt;"
              2: Include key=value pairs, replacing brackets with quotation marks

              and replacing spaces between key=value pairs with semicolons

       <b>--gff3-cds</b>=<u>STRING</u>
              Whether to use cDNA or genomic translation for the CDS coordinates Values: cdna (default), genomic

       Options for SAM output

       <b>--no-sam-headers</b>
              Do not print headers beginning with '@'

       <b>--sam-use-0M</b>
              Insert 0M in CIGAR between adjacent insertions and deletions Required by  Picard,  but  can  cause
              errors in other tools

       <b>--sam-extended-cigar</b>
              Use extended CIGAR format (using X and = symbols instead of M,
               to indicate matches and mismatches, respectively

       <b>--sam-flipped</b>
              Flip  the query and genomic positions in the SAM output.  Potentially useful with the <b>-g</b> flag when
              short reads are picked as query sequences and longer reads as picked as genomic sequences

       <b>--force-xs-dir</b>
              For RNA-Seq alignments, disallows XS:A:? when the sense direction is unclear,  and  replaces  this
              value  arbitrarily  with  XS:A:+.  May be useful for some programs, such as Cufflinks, that cannot
              handle XS:A:?.  However, if you use this flag, the reported value of XS:A:+ in  these  cases  will
              not be meaningful.

       <b>--md-lowercase-snp</b>
              In MD string, when known SNPs are given by the <b>-v</b> flag,
               prints difference nucleotides as lower-case when they,
               differ from reference but match a known alternate allele

       <b>--action-if-cigar-error</b>
              Action to take if there is a disagreement between CIGAR length and sequence length Allowed values:
              ignore,  warning  (default),  noprint, abort Note that the noprint option does not print the CIGAR
              string at all if there is an error, so it may break a SAM parser

       <b>--read-group-id</b>=<u>STRING</u>
              Value to put into read-group id (RG-ID) field

       <b>--read-group-name</b>=<u>STRING</u>
              Value to put into read-group name (RG-SM) field

       <b>--read-group-library</b>=<u>STRING</u>
              Value to put into read-group library (RG-LB) field

       <b>--read-group-platform</b>=<u>STRING</u>
              Value to put into read-group library (RG-PL) field

       Options for quality scores

       <b>--quality-protocol</b>=<u>STRING</u>
              Protocol for input quality scores.  Allowed values: illumina (ASCII 64-126) (equivalent to  <b>-J</b>  64
              <b>-j</b> <b>-31</b>) sanger   (ASCII 33-126) (equivalent to <b>-J</b> 33 <b>-j</b> 0)

       Default is sanger (no quality print shift)
              SAM output files should have quality scores in sanger protocol

              Or you can specify the print shift with this flag:

       <b>-j</b>, <b>--quality-print-shift</b>=<u>INT</u>
              Shift  FASTQ  quality scores by this amount in output (default is 0 for sanger protocol; to change
              Illumina input to Sanger output, select <b>-31</b>)

       External map file options

       <b>-M</b>, <b>--mapdir</b>=<u>directory</u>
              Map directory

       <b>-m</b>, <b>--map</b>=<u>iitfile</u>
              Map file.  If argument is '?' (with the quotes),
               this lists available map files.

       <b>-e</b>, <b>--mapexons</b>
              Map each exon separately

       <b>-b</b>, <b>--mapboth</b>
              Report hits from both strands of genome

       <b>-u</b>, <b>--flanking</b>=<u>INT</u>
              Show flanking hits (default 0)

       <b>--print-comment</b>
              Show comment line for each hit

       Alignment output options

       <b>--nolengths</b>
              No intron lengths in alignment

       <b>--nomargin</b>
              No left margin in GMAP standard output (with the <b>-A</b> flag)

       <b>-I</b>, <b>--invertmode</b>=<u>INT</u>
              Mode for alignments to genomic (-) strand: 0=Don't invert the cDNA  (default)  1=Invert  cDNA  and
              print genomic (-) strand 2=Invert cDNA and print genomic (+) strand

       <b>-i</b>, <b>--introngap</b>=<u>INT</u>
              Nucleotides to show on each end of intron (default 3)

       <b>-l</b>, <b>--wraplength</b>=<u>INT</u>
              Wrap length for alignment (default 50)

       Filtering output options

       <b>--min-trimmed-coverage</b>=<u>FLOAT</u>
              Do  not  print  alignments  with  trimmed  coverage  less  this value (default=0.0, which means no
              filtering) Note that chimeric alignments will be output regardless of this filter

       <b>--min-identity</b>=<u>FLOAT</u>
              Do not print alignments with identity less this value (default=0.0, which means no filtering) Note
              that chimeric alignments will be output regardless of this filter Help options

       <b>--check</b>
              Check compiler assumptions

       <b>--version</b>
              Show version

       <b>--help</b> Show this help message

       Other tools of GMAP suite are located in /usr/lib/gmap

gmap 2024-11-20+ds-2                              February 2025                                          <u><a href="../man1/GMAP.1.html">GMAP</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>