<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>swarm — find clusters of nearly-identical nucleotide amplicons</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/swarm">swarm_3.1.5+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       swarm — find clusters of nearly-identical nucleotide amplicons

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>swarm</b> <b>-h|v</b>

       High-precision clustering:

       <b>swarm</b> [<u>filename</u>]

       <b>swarm</b> [<b>-d</b> 1<b>]</b> [<b>-nrz</b>] [<b>-a</b> <u>int</u>] [<b>-i</b> <u>filename</u>] [<b>-j</b> <u>filename</u>] [<b>-l</b> <u>filename</u>] [<b>-o</b> <u>filename</u>] [<b>-s</b> <u>filename</u>]
             [<b>-t</b> <u>int</u>] [<b>-u</b> <u>filename</u>] [<b>-w</b> <u>filename</u>] [<u>filename</u>]

       <b>swarm</b> [<b>-d</b> 1<b>]</b> <b>-f</b> [<b>-nrz</b>] [<b>-a</b> <u>int</u>] [<b>-b</b> <u>int</u>] [<b>-c|y</b> <u>int</u>] [<b>-i</b> <u>filename</u>] [<b>-j</b> <u>filename</u>] [<b>-l</b> <u>filename</u>]
             [<b>-o</b> <u>filename</u>] [<b>-s</b> <u>filename</u>] [<b>-t</b> <u>int</u>] [<b>-u</b> <u>filename</u>] [<b>-w</b> <u>filename</u>] [<u>filename</u>]

       Conservative clustering:

       <b>swarm</b> -d <b>2+</b> [<b>-nrxz</b>] [<b>-a</b> <u>int</u>] [<b>-e</b> <u>int</u>] [<b>-g</b> <u>int</u>] [<b>-i</b> <u>filename</u>] [<b>-l</b> <u>filename</u>] [<b>-m</b> <u>int</u>] [<b>-o</b> <u>filename</u>]
             [<b>-p</b> <u>int</u>] [<b>-s</b> <u>filename</u>] [<b>-t</b> <u>int</u>] [<b>-u</b> <u>filename</u>] [<b>-w</b> <u>filename</u>] [<u>filename</u>]

       Dereplication (merge strictly identical sequences):

       <b>swarm</b> -d <b>0</b> [<b>-rz</b>] [<b>-a</b> <u>int</u>] [<b>-i</b> <u>filename</u>] [<b>-l</b> <u>filename</u>] [<b>-o</b> <u>filename</u>] [<b>-s</b> <u>filename</u>] [<b>-u</b> <u>filename</u>]
             [<b>-w</b> <u>filename</u>] [<u>filename</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Environmental  or  clinical  molecular studies generate large volumes of amplicons (e.g., 16S or 18S SSU-
       rRNA sequences) that need to be grouped into  clusters.  Traditional  clustering  methods  are  based  on
       greedy,  input-order  dependent  algorithms,  with  arbitrary  selection of cluster centroids and cluster
       limits (often 97%-similarity). To address that problem, we developed <b>swarm</b>, a fast and robust method that
       recursively groups amplicons with <u>d</u> or less differences (i.e. substitutions,  insertions  or  deletions).
       <b>swarm</b>  produces natural and stable clusters centered on local peaks of abundance, mostly free from input-
       order dependency induced by centroid selection.

       Exact clustering is impractical on large data sets when using a naïve all-vs-all approach (more precisely
       a 2-combination without repetitions), as it implies unrealistic numbers of pairwise comparisons. <b>swarm</b> is
       based on a maximum number of differences <u>d</u> between two amplicons, and focuses only on  very  close  local
       relationships.  For <u>d</u> = 1, the default value, <b>swarm</b> uses an algorithm of linear complexity that generates
       all possible single mutations and performs exact-string matching by comparing hash-values. For <u>d</u> =  2  or
       greater,  <b>swarm</b>  uses  an algorithm of quadratic complexity that performs pairwise string comparisons. An
       efficient <u>k</u>-mer-based filtering and an astute use of comparisons results obtained during  the  clustering
       process  allows  <b>swarm</b>  to avoid most of the amplicon comparisons needed in a naïve approach. To speed up
       the remaining amplicon comparisons, <b>swarm</b> implements an extremely fast Needleman-Wunsch algorithm  making
       use  of  the  Streaming  SIMD  Extensions  (SSE2)  of  x86-64  CPUs,  NEON instructions of ARM64 CPUs, or
       Altivec/VMX instructions of POWER8 CPUs. If SSE2 instructions are not  available,  <b>swarm</b>  exits  with  an
       error message.

       <b>swarm</b>  can read nucleotide amplicons in fasta format from a normal file or from the standard input (using
       a pipe or a redirection). The amplicon <u>header</u> is defined as the string comprised between the  '&gt;'  symbol
       and the first space or the end of the line, whichever comes first. Each header must end with an <u>abundance</u>
       <u>annotation</u>  representing  the amplicon copy number and defined as '_' followed by a positive integer. See
       option -z for input data using usearch/vsearch's abundance annotation format  (';size=<u>integer</u>[;]').  Once
       stripped  from  the abundance annotation, the remaining part of the header is call the <u>label</u>. In summary,
       using regular expression patterns:

                                &gt;header[[:blank:]]   and   header = label_[1-9][0-9]*$

       Abundance annotations play a crucial role in the clustering  process,  and  swarm  exits  with  an  error
       message  if that information is not available. As <b>swarm</b> outputs lists of amplicon labels, amplicon labels
       must be unique to avoid any ambiguity; swarm exits with an error message if labels are  not  unique.  The
       amplicon  sequence  is defined as a string of [ACGT] or [ACGU] symbols (case insensitive, 'U' is replaced
       with 'T' internally), starting after the end of the header line and ending before the next header line or
       the file end; <b>swarm</b> silently removes newline symbols ('\n' or '\r') and exits with an  error  message  if
       any other symbol is present. Accepted sequence lengths range from 1 nucleotide to 67 million nucleotides.
       Please  note  that  processing  67-Mb  sequences  requires  at  least  32 gigabytes of memory. Lastly, if
       sequences are not all unique, i.e. were not properly dereplicated, swarm will exit with an error message.

       Clusters are written to output files (specified with -i, -o, -s and -u) by decreasing abundance of  their
       seed  sequences,  and  then by alphabetical order of seed sequence labels. An exception to that is the -w
       (--seeds) output, which is sorted by decreasing <u>cluster</u> <u>abundance</u> (sum of abundances of all sequences  in
       the  cluster),  and  then  by alphabetical order of seed sequence labels. This is particularly useful for
       post-clustering steps, such as <u>de</u>  <u>novo</u>  chimera  detection,  that  require  clusters  to  be  sorted  by
       decreasing abundances.

   <b>General</b> <b>options</b>
       <b>-h</b>, <b>--help</b>
                display this help and exit successfully.

       <b>-t</b>, <b>--threads</b> <u>positive</u> <u>integer</u>
                number of computation threads to use. Values between 1 and 512 are accepted, but we recommend to
                use  a number of threads lesser or equal to the number of available CPU cores. Default number of
                threads is 1.

       <b>-v</b>, <b>--version</b>
                output version information and exit successfully.

       <b>--</b>       delimit the option list. Later arguments, if any, are treated as operands  even  if  they  begin
                with '-'. For example, 'swarm -- -file.fasta' reads from the file '-file.fasta'.

   <b>Clustering</b> <b>options</b>
       <b>-d</b>, <b>--differences</b> <u>zero</u> <u>or</u> <u>positive</u> <u>integer</u>
                maximum  number of differences allowed between two amplicons, meaning that two amplicons will be
                grouped if they have <u>integer</u> (or less) differences. This is <b>swarm</b>'s  most  important  parameter.
                The  number  of differences is calculated as the number of mismatches (substitutions, insertions
                or deletions) between the two amplicons once the optimal  pairwise  global  alignment  has  been
                found (see 'pairwise alignment advanced options' to influence that step).

                Any  <u>integer</u>  from  0  to  255  can  be  used,  but  high <u>d</u> values will decrease the taxonomical
                resolution of <b>swarm</b> results. Commonly used <u>d</u> values are 1, 2 or 3, rarely higher. When using <u>d</u> =
                0, <b>swarm</b> will output results corresponding to  a  strict  dereplication  of  the  dataset,  i.e.
                merging  identical  amplicons.  Warning,  whatever  the <u>d</u> value, <b>swarm</b> requires fasta entries to
                present abundance values. Default number of differences <u>d</u> is 1.

       <b>-n</b>, <b>--no-otu-breaking</b>
                when working with <u>d</u> = 1, deactivate the built-in cluster refinement (not recommended).  Amplicon
                abundance  values  are  used  to  identify transitions among in-contact clusters and to separate
                them, yielding higher-resolution clustering results. That option prevents that  separation,  and
                in practice, allows the creation of a link between amplicons A and B, even if the abundance of B
                is higher than the abundance of A.

   <b>Fastidious</b> <b>options</b>
       <b>-b</b>, <b>--boundary</b> <u>positive</u> <u>integer</u>
                when  using  the  option  --fastidious  (-f),  define  the  minimum  abundance of what should be
                considered a <u>large</u> cluster. By default, a cluster with  a  total  abundance  of  3  or  more  is
                considered  large.  Conversely,  a  cluster  is  small if it has a total abundance of 2 or less,
                meaning that it is composed of either one amplicon of abundance 2, or two amplicons of abundance
                1, or one amplicon of abundance 1. Any positive value greater than 1  can  be  specified.  Using
                higher  boundary  values  can reduce the number of clusters (up to a point), and will reduce the
                taxonomical resolution of <b>swarm</b> results. It will also slightly increase computation time.

       <b>-c</b>, <b>--ceiling</b> <u>positive</u> <u>integer</u>
                when using the option --fastidious (-f), define <b>swarm</b>'s maximum memory footprint (in megabytes).
                <b>swarm</b> will adjust the --bloom-bits (-y) value of the Bloom filter to fit  within  the  specified
                amount of memory. Values accepted range from 40 to 1,073,741,824 megabytes. See the --bloom-bits
                (-y) option for an alternative way to control the memory footprint.

       <b>-f</b>, <b>--fastidious</b>
                when working with <u>d</u> = 1, perform a second clustering pass to reduce the number of small clusters
                (recommended  option). During the first clustering pass, an intermediate amplicon can be missing
                for purely stochastic reasons, interrupting the aggregation process. The fastidious option  will
                create  virtual  amplicons,  allowing  to  graft  small clusters upon larger ones. By default, a
                cluster is considered large if it has a total abundance of 3 or more (see the --boundary  option
                to modify that value).

                To speed things up, <b>swarm</b> uses a Bloom filter to store intermediate results. Warning, the second
                clustering  pass can be 2 to 3 times slower than the first pass and requires much more memory to
                store the virtual amplicons in Bloom filters. See the options  --bloom-bits  (-y)  or  --ceiling
                (-c) to control the memory footprint of the Bloom filter.

                The  fastidious  option  modifies  clustering  results: the output files produced by the options
                --log (-l), --output-file (-o), --mothur (-r), --uclust-file, and --seeds (-w)  are  updated  to
                reflect these modifications; the file --statistics-file (-s) is partially updated (columns 6 and
                7  are not updated); the output file --internal-structure (-i) is partially updated (column 5 is
                not updated for amplicons that belonged to the small cluster).

       <b>-y</b>, <b>--bloom-bits</b> <u>positive</u> <u>integer</u>
                when using the option --fastidious (-f), define the size (in bits) of each entry  in  the  Bloom
                filter.  That  option  allows to balance the efficiency (i.e. speed) and the memory footprint of
                the Bloom filter. Large values will make the Bloom filter more efficient but will  require  more
                memory.  Any  value  between  2  and 64 can be used. Default value is 16. See the --ceiling (-c)
                option for an alternative way to control the memory footprint.

   <b>Input/output</b> <b>options</b>
       <b>-a</b>, <b>--append-abundance</b> <u>positive</u> <u>integer</u>
                set abundance value to use when some or all amplicons in the input file  lack  abundance  values
                (_<u>integer</u>,  or  ;size=<u>integer</u>;  when  using  -z). Warning, it is not recommended to use <b>swarm</b> on
                datasets where abundance values are all identical. We provide  that  option  as  a  courtesy  to
                advanced  users,  please use it carefully. <b>swarm</b> exits with an error message if abundance values
                are missing and if this option is not used.

       <b>-i</b>, <b>--internal-structure</b> <u>filename</u>
                output all pairs of nearly-identical amplicons to <u>filename</u>  using  a  five-column  tab-delimited
                format:

                       1.  amplicon A label (header without abundance annotations).

                       2.  amplicon B label (header without abundance annotations).

                       3.  number of differences between amplicons A and B (<u>positive</u> <u>integer</u>).

                       4.  cluster   number  (<u>positive</u>  <u>integer</u>).  Clusters  are  numbered  in  their  order  of
                           delineation, starting from 1. All pairs of amplicons belonging to  the  same  cluster
                           will receive the same number.

                       5.  cummulated  number  of  steps from the cluster seed to amplicon B (<u>positive</u> <u>integer</u>).
                           When using the option --fastidious (-f), the actual number of steps  between  grafted
                           amplicons and the cluster seed cannot be re-computed efficiently and is always set to
                           2  for  the  amplicon pair linking the small cluster to the large cluster. Cummulated
                           number of steps in the small cluster (if any) are left unchanged.

       <b>-j</b>, <b>--network-file</b> <u>filename</u>
                (advanced users) when working with <u>d</u> = 1, dump raw amplicon network to  <u>filename</u>  using  a  two-
                column  tab-delimited  table  of  headers  with  abundance  annotations.  Each line represents a
                connection between two similar amplicons, from the most abundant to the  lesser  abundant.  When
                amplicons  have  the same abundance value, connections are bi-directional and are represented on
                two lines: A to B, then B to A.

                In order to delineate clusters and to compute the equivalent of a minimal spanning tree for each
                cluster (see option --internal-structure), swarm first builds a network  of  similar  amplicons.
                This option is for advanced users who would like to explore this raw network.

       <b>-l</b>, <b>--log</b> <u>filename</u>
                output  all messages to <u>filename</u> instead of <u>standard</u> <u>error</u>, with the exception of error messages
                of course. That option is useful in situations where writing to <u>standard</u>  <u>error</u>  is  problematic
                (for example, with certain job schedulers).

       <b>-o</b>, <b>--output-file</b> <u>filename</u>
                output  clustering  results  to <u>filename</u>. Results consist of a list of clusters, one cluster per
                line. A cluster is a list of amplicon headers separated by spaces. That  output  format  can  be
                modified by the option --mothur (-r). Default is to write to <u>standard</u> <u>output</u>.

       <b>-r</b>, <b>--mothur</b>
                output  clustering  results  in  a  format  compatible with Mothur. That option modifies <b>swarm</b>'s
                default output format.

       <b>-s</b>, <b>--statistics-file</b> <u>filename</u>
                output statistics to <u>filename</u>. The file is a tab-separated table with one cluster  per  row  and
                seven columns of information:

                       1.  number of unique amplicons in the cluster,

                       2.  total abundance of amplicons in the cluster,

                       3.  label of the initial seed (header without abundance annotations),

                       4.  abundance of the initial seed,

                       5.  number of amplicons with an abundance of 1 in the cluster,

                       6.  maximum number of iterations before the cluster reached its natural limit,

                       7.  cummulated  number  of  steps  along  the  path  joining the seed and the furthermost
                           amplicon in the cluster. Please note that the actual number  of  differences  between
                           the  seed and the furthermost amplicon is usually much smaller. When using the option
                           --fastidious (-f), grafted amplicons are not taken into account.

       <b>-u</b>, <b>--uclust-file</b> <u>filename</u>
                output clustering results in <u>filename</u> using a tab-separated uclust-like format with  10  columns
                and  3 different type of entries (S, H or C). That option does not modify <b>swarm</b>'s default output
                format. Each fasta sequence in the input file can be either a cluster centroid (S) or a hit  (H)
                assigned  to  a  cluster.  Cluster records (C) summarize information for each cluster (number of
                hits, centroid header). Column content varies with the type of entry (S, H or C):

                       1.  Record type: S, H, or C.

                       2.  Cluster number (zero-based).

                       3.  Centroid length (S), query length (H), or number of hits (C).

                       4.  Percentage of similarity with the centroid sequence (H), or set to '*' (S, C).

                       5.  Match orientation + or - (H), or set to '*' (S, C).

                       6.  Not used, always set to '*' (S, C) or to zero (H).

                       7.  Not used, always set to '*' (S, C) or to zero (H).

                       8.  set to '*' (S, C) or, for H, compact representation of the pairwise  alignment  using
                           the  CIGAR  format  (Compact  Idiosyncratic  Gapped  Alignment  Report): M (match), D
                           (deletion) and I (insertion).  The  equal  sign  '='  indicates  that  the  query  is
                           identical to the centroid sequence.

                       9.  Header of the query sequence (H), or of the centroid sequence (S, C).

                       10. Header of the centroid sequence (H), or set to '*' (S, C).

       <b>-w</b>, <b>--seeds</b> <u>filename</u>
                output cluster representative sequences to <u>filename</u> in fasta format. The abundance value of each
                cluster  representative  is the sum of the abundances of all the amplicons in the cluster. Fasta
                headers are formated as follows: '&gt;label_<u>integer</u>', or '&gt;label;size=<u>integer</u>;' if the -z option is
                used, and sequences are uppercased. Sequences are sorted by decreasing abundance,  and  then  by
                alphabetical order of sequence labels.

       <b>-z</b>, <b>--usearch-abundance</b>
                accept  amplicon  abundance  values  in  usearch/vsearch's  style (&gt;label;size=<u>integer</u>[;]). That
                option influences the abundance annotation style used in swarm's <u>standard</u> <u>output</u> (-o),  as  well
                as the output of options -r, -u and -w.

   <b>Pairwise</b> <b>alignment</b> <b>advanced</b> <b>options</b>
       when  using <u>d</u> &gt; 1, <b>swarm</b> recognizes advanced command-line options modifying the pairwise global alignment
       scoring parameters:

              <b>-m</b>, <b>--match-reward</b> <u>positive</u> <u>integer</u>
                       Default reward for a nucleotide match is 5.

              <b>-p</b>, <b>--mismatch-penalty</b> <u>positive</u> <u>integer</u>
                       Default penalty for a nucleotide mismatch is 4.

              <b>-g</b>, <b>--gap-opening-penalty</b> <u>positive</u> <u>integer</u>
                       Default gap opening penalty is 12.

              <b>-e</b>, <b>--gap-extension-penalty</b> <u>positive</u> <u>integer</u>
                       Default gap extension penalty is 4.

              <b>-x</b>, <b>--disable-sse3</b>
                       On the x86-64 CPU architecture, disable SSE3 and later instructions. This option is meant
                       for developers, not for regular users.

       As <b>swarm</b> focuses on close relationships (e.g., <u>d</u> = 2 or 3), clustering results are resilient to  pairwise
       alignment  model  parameters  modifications.  When  clustering  using  a  higher <u>d</u> value, modifying model
       parameters has a stronger impact.

</pre><h4><b>EXAMPLES</b></h4><pre>
       Clusterize the compressed data set <u>myfile.fasta</u> using the finest resolution  possible  (1  difference  by
       default,  built-in  breaking, fastidious option) using 4 computation threads. Clusters are written to the
       file <u>myfile.swarms</u>, and cluster representatives are written to <u>myfile.representatives.fasta</u>:
              zcat myfile.fasta.gz | \
                  swarm \
                      -t 4 \
                      -f \
                      -w myfile.representatives.fasta \
                      -o myfile.swarms

</pre><h4><b>AUTHORS</b></h4><pre>
       Concept by Frédéric Mahé, implementation by Torbjørn Rognes.

</pre><h4><b>CITATION</b></h4><pre>
       Mahé F, Rognes T, Quince C, de Vargas C, Dunthorn M. (2014) Swarm: robust and fast clustering method  for
       amplicon-based studies.  <u>PeerJ</u> 2:e593 https://doi.org/10.7717/peerj.593.

       Mahé F, Rognes T, Quince C, de Vargas C, Dunthorn M. (2015) Swarm v2: highly-scalable and high-resolution
       amplicon clustering.  <u>PeerJ</u> 3:e1420 https://doi.org/10.7717/peerj.1420.

       Mahé  F,  Czech  L,  Stamatakis  A, Quince C, de Vargas C, Dunthorn M, Rognes T. (2021) Swarm v3: towards
       tera-scale amplicon clustering.  <u>Bioinformatics</u> https://doi.org/10.1093/bioinformatics/btab493.

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       Submit suggestions and bug-reports at https://github.com/torognes/swarm/issues, send a  pull  request  at
       https://github.com/torognes/swarm/pulls,  or  compose  a friendly or curmudgeonly e-mail to Frédéric Mahé
       and Torbjørn Rognes.

</pre><h4><b>AVAILABILITY</b></h4><pre>
       Source code and binaries available at https://github.com/torognes/swarm.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright (C) 2012-2024 Frédéric Mahé &amp; Torbjørn Rognes

       This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero
       General Public License as published by the Free Software Foundation, either version 3 of the License,  or
       any later version.

       This  program  is  distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
       the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU  Affero  General
       Public License for more details.

       You  should  have  received  a copy of the GNU Affero General Public License along with this program.  If
       not, see https://www.gnu.org/licenses/.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b>swipe</b>,  an  extremely  fast  Smith-Waterman  database  search  tool  by  Torbjørn  Rognes  (available  at
       https://github.com/torognes/swipe).

       <b>vsearch</b>,  an  open-source re-implementation of the classic uclust clustering method (by Robert C. Edgar),
       along with other amplicon filtering and searching tools. <b>vsearch</b> is implemented by  Torbjørn  Rognes  and
       documented by Frédéric Mahé, and is available at https://github.com/torognes/vsearch.

</pre><h4><b>VERSION</b> <b>HISTORY</b></h4><pre>
       New features and important modifications of <b>swarm</b> (short lived or minor bug releases are not mentioned):

              <b>v3.1.5</b> released March 31, 2024
                     Version  3.1.5  changes  the  minimal  value  for the ceiling option from 8 megabytes to 40
                     megabytes, and fixes four minor bugs. Warning, peak RSS memory increased by 5 to 10% when <u>d</u>
                     &gt;= 2. Version 3.1.5 improves documentation (now covering option --network_file), adds  more
                     compilation checks and eliminates 50 compilation warnings with GCC 13, GCC 14 and clang 19,
                     as well as 1,677 static analysis warnings.

              <b>v3.1.4</b> released September 20, 2023
                     Version  3.1.4  fixes a minor bug. It eliminates compilation warnings with GCC 13 and clang
                     18, as well as 1,040 static analysis warnings. The maximal number of threads swarm can  run
                     is  now  512,  instead  of  256.  Compilation  with  runtime checks (`-DNDEBUG`) is now the
                     default. When d &gt; 1, overall memory allocations  remain  unchanged,  but  peak  RSS  memory
                     increased  by  6  to  10%,  due to a change in the timing of memory deallocations. Peak RSS
                     memory is expected to regress to its prior levels as refactoring continues.

              <b>v3.1.3</b> released December 5, 2022
                     Version 3.1.3 fixes a regression introduced in version 3.1.1 (memory over-allocation when d
                     &gt; 1). It also fixes a minor off-by-one error when allocating memory  for  a  Bloom  filter,
                     compilation  warnings  with  GCC  12  and  clang  13,  as well as static analysis warnings.
                     Documentation was improved, as well as our test suite (swarm-tests).

              <b>v3.1.2</b> released November 10, 2022
                     Fix a bug with fastidious mode introduced in version  3.1.1,  that  could  cause  Swarm  to
                     crash. Probably due to allocating too much memory.

              <b>v3.1.1</b> released September 29, 2022
                     Version 3.1.1 eliminates a risk of segmentation fault with extremely long sequence headers.
                     Documentation and error messages have been improved, and code cleaning continued.

              <b>v3.1.0</b> released March 1, 2021
                     Version  3.1.0  includes a fix for a bug in the 16-bit SIMD alignment code that was exposed
                     with a combination of d&gt;1, long sequences, and very high gap penalties. The code  has  also
                     been  been  cleaned  up,  tested  and  improved  substantially,  and  it is now fully C++11
                     compliant. Support for macOS on Apple Silicon (ARM64) has been added.

              <b>v3.0.0</b> released October 24, 2019
                     Version 3.0.0 introduces a faster algorithm for <u>d</u> = 1,  and  a  reduced  memory  footprint.
                     Swarm  has  been ported to Windows x86-64, GNU/Linux ARM 64, and GNU/Linux POWER8. Internal
                     code has been modernized, hardened, and thoroughly tested. Strict  dereplication  of  input
                     sequences  is  now  mandatory.  The  --seeds  option  (-w)  now  outputs  results sorted by
                     decreasing abundance, and then by alphabetical order of sequence labels.

              <b>v2.2.2</b> released December 12, 2017
                     Version 2.2.2 fixes a bug that would cause swarm to wait forever in very  rare  cases  when
                     multiple threads were used.

              <b>v2.2.1</b> released October 27, 2017
                     Version 2.2.1 fixes a memory allocation bug for <u>d</u> = 1 and duplicated sequences.

              <b>v2.2.0</b> released October 17, 2017
                     Version  2.2.0 fixes several problems and improves usability. Corrected output to structure
                     and uclust files when using fastidious mode. Corrected  abundance  output  in  some  cases.
                     Added  check  for  duplicated sequences and fixed check for duplicated sequence IDs. Checks
                     for empty sequences. Sorts sequences by additional fields to  improve  stability.  Improves
                     compatibility  with  compilers  and  operating  systems.   Outputs sequences in upper case.
                     Allows 64-bit abundances. Shows message when waiting for input from stdin.  Improves  error
                     messages  and  warnings.  Improves checking of command line options. Fixes remaining errors
                     reported by test suite. Updates documentation.

              <b>v2.1.13</b> released March 8, 2017
                     Version 2.1.13 removes a bug with the progress bar when writing seeds.

              <b>v2.1.12</b> released January 16, 2017
                     Version 2.1.12 removes a debugging message.

              <b>v2.1.11</b> released January 16, 2017
                     Version 2.1.11 fixes two bugs related to the SIMD implementation of  alignment  that  might
                     result in incorrect alignments and scores.  The bug only applies when <u>d</u> &gt; 1.

              <b>v2.1.10</b> released December 22, 2016
                     Version  2.1.10  fixes  two bugs related to gap penalties of alignments.  The first bug may
                     lead to wrong aligments and similarity percentages reported  in  UCLUST  (.uc)  files.  The
                     second  bug  makes  swarm  use  a slightly higher gap extension penalty than specified. The
                     default gap extension penalty used have actually been 4.5 instead of 4.

              <b>v2.1.9</b> released July 6, 2016
                     Version 2.1.9 fixes errors when compiling with GCC version 6.

              <b>v2.1.8</b> released March 11, 2016
                     Version 2.1.8 fixes a rare bug triggered when  clustering  extremely  short  undereplicated
                     sequences. Also, alignment parameters are not shown when <u>d</u> = 1.

              <b>v2.1.7</b> released February 24, 2016
                     Version 2.1.7 fixes a bug in the output of seeds with the -w option when <u>d</u> &gt; 1 that was not
                     properly  fixed  in  version 2.1.6. It also handles ascii character #13 (CR) in FASTA files
                     better. Swarm will now exit with status 0 if the -h or the -v option is specified. The help
                     text and some error messages have been improved.

              <b>v2.1.6</b> released December 14, 2015
                     Version 2.1.6 fixes problems with older compilers that do not have the  x86intrin.h  header
                     file. It also fixes a bug in the output of seeds with the -w option when <u>d</u> &gt; 1.

              <b>v2.1.5</b> released September 8, 2015
                     Version 2.1.5 fixes minor bugs.

              <b>v2.1.4</b> released September 4, 2015
                     Version 2.1.4 fixes minor bugs in the swarm algorithm used for <u>d</u> = 1.

              <b>v2.1.3</b> released August 28, 2015
                     Version 2.1.3 adds checks of numeric option arguments.

              <b>v2.1.1</b> released March 31, 2015
                     Version  2.1.1  fixes  a  bug  with  the  fastidious  option  that caused it to ignore some
                     connections between large and small clusters.

              <b>v2.1.0</b> released March 24, 2015
                     Version 2.1.0 marks the first official release of swarm v2.

              <b>v2.0.7</b> released March 18, 2015
                     Version 2.0.7 writes abundance information in usearch style when using options -w (--seeds)
                     in combination with -z (--usearch-abundance).

              <b>v2.0.6</b> released March 13, 2015
                     Version 2.0.6 fixes a minor bug.

              <b>v2.0.5</b> released March 13, 2015
                     Version 2.0.5 improves the implementation of the fastidious  option  and  adds  options  to
                     control  memory  usage of the Bloom filter (-y and -c).  In addition, an option (-w) allows
                     to output cluster representatives sequences with updated abundances (sum of all  abundances
                     inside each cluster). This version also enables <b>swarm</b> to run with <u>d</u> = 0.

              <b>v2.0.4</b> released March 6, 2015
                     Version 2.0.4 includes a fully parallelised implementation of the fastidious option.

              <b>v2.0.3</b> released March 4, 2015
                     Version  2.0.3  includes  a  working  implementation of the fastidious option, but only the
                     initial clustering is parallelized.

              <b>v2.0.2</b> released February 26, 2015
                     Version 2.0.2 fixes SSSE3 problems.

              <b>v2.0.1</b> released February 26, 2015
                     Version 2.0.1 is a development version  that  contains  a  partial  implementation  of  the
                     fastidious option, but it is not usable yet.

              <b>v2.0.0</b> released December 3, 2014
                     Version   2.0.0   is   faster   and   easier   to   use,   providing   new  output  options
                     (--internal-structure  and  --log),  new   control   options   (--boundary,   --fastidious,
                     --no-otu-breaking),  and  built-in  cluster  refinement  (no  need to use the python script
                     anymore). When using default  parameters,  a  novel  and  considerably  faster  algorithmic
                     approach is used, guaranteeing <b>swarm</b>'s scalability.

              <b>v1.2.21</b> released February 26, 2015
                     Version  1.2.21  is  supposed  to  fix  some  problems  related to the use of the SSSE3 CPU
                     instructions which are not always available.

              <b>v1.2.20</b> released November 6, 2014
                     Version 1.2.20 presents a production-ready version of  the  alternative  algorithm  (option
                     -a),  with  optional  built-in  cluster  breaking (option -n). That alternative algorithmic
                     approach (usable only with <u>d</u> = 1) is considerably faster  than  currently  used  clustering
                     algorithms,  and  can  deal  with datasets of 100 million unique amplicons or more in a few
                     hours. Of course, results are rigourously identical to the results previously produced with
                     swarm. That release also introduces new options to control swarm  output  (options  -i  and
                     -l).

              <b>v1.2.19</b> released October 3, 2014
                     Version  1.2.19  fixes  a  problem related to abundance information when the sequence label
                     includes multiple underscore characters.

              <b>v1.2.18</b> released September 29, 2014
                     Version 1.2.18 reenables the possibility of reading sequences from <u>stdin</u> if no file name is
                     specified on the command line. It also fixes a bug related to CPU features detection.

              <b>v1.2.17</b> released September 28, 2014
                     Version 1.2.17 fixes a memory allocation bug introduced in version 1.2.15.

              <b>v1.2.16</b> released September 27, 2014
                     Version 1.2.16 fixes a bug in the abundance sort introduced in version 1.2.15.

              <b>v1.2.15</b> released September 27, 2014
                     Version 1.2.15 sorts the input sequences in order of decreasing abundance unless  they  are
                     detected to be sorted already. When using the alternative algorithm for <u>d</u> = 1 it also sorts
                     all subseeds in order of decreasing abundance.

              <b>v1.2.14</b> released September 27, 2014
                     Version  1.2.14  fixes  a bug in the output with the --swarm_breaker option (-b) when using
                     the alternative algorithm (-a).

              <b>v1.2.12</b> released August 18, 2014
                     Version 1.2.12 introduces an option  --alternative-algorithm  to  use  an  extremely  fast,
                     experimental clustering algorithm for the special case <u>d</u> = 1. Multithreading scalability of
                     the default algorithm has been noticeably improved.

              <b>v1.2.10</b> released August 8, 2014
                     Version  1.2.10  allows  amplicon abundances to be specified using the usearch style in the
                     sequence header (e.g. '&gt;id;size=1') when the -z option is chosen.

              <b>v1.2.8</b> released August 5, 2014
                     Version 1.2.8 fixes an error with the gap extension penalty. Previous versions used  a  gap
                     penalty twice as large as intended. That bug correction induces small changes in clustering
                     results.

              <b>v1.2.6</b> released May 23, 2014
                     Version  1.2.6  introduces  an  option  --mothur  to  output clustering results in a format
                     compatible  with  the  microbial  ecology  community  analysis  software  suite  Mothur   (
                     https://www.mothur.org/).

              <b>v1.2.5</b> released April 11, 2014
                     Version  1.2.5  removes the need for a POPCNT hardware instruction to be present. <b>swarm</b> now
                     automatically checks whether POPCNT is  available  and  uses  a  slightly  slower  software
                     implementation if not. Only basic SSE2 instructions are now required to run <b>swarm</b>.

              <b>v1.2.4</b> released January 30, 2014
                     Version  1.2.4  introduces an option --break-swarms to output all pairs of amplicons with <u>d</u>
                     differences  to  <u>standard</u>  <u>error</u>.  That  option   is   used   by   the   companion   script
                     `swarm_breaker.py`  to  refine  <b>swarm</b>  results.  The  syntax of the inline assembly code is
                     changed for compatibility with more compilers.

              <b>v1.2</b> released May 16, 2013
                     Version 1.2 greatly improves speed by using alignment-free comparisons of  amplicons  based
                     on  <u>k</u>-mer  word  content. For each amplicon, the presence-absence of all possible 5-mers is
                     computed and recorded in a 1024-bits vector. Vector  comparisons  are  extremely  fast  and
                     drastically  reduce  the  number  of  costly  pairwise alignments performed by <b>swarm</b>. While
                     remaining exact, <b>swarm</b> 1.2 can be more than 100-times faster than <b>swarm</b> 1.1, when  using  a
                     single  thread  with  a  large  set  of  sequences. The minor version 1.1.1, published just
                     before, adds compatibility with Apple computers, and corrects  an  issue  in  the  pairwise
                     global alignment step that could lead to sub-optimal alignments.

              <b>v1.1</b> released February 26, 2013
                     Version  1.1  introduces  two  new  important options: the possibility to output clustering
                     results using the uclust output format, and the possibility to output  detailed  statistics
                     on  each  cluster.  <b>swarm</b>  1.1  is  also  faster: new filterings based on pairwise amplicon
                     sequence lengths and composition comparisons  reduce  the  number  of  pairwise  alignments
                     needed and speed up the clustering.

              <b>v1.0</b> released November 10, 2012
                     First public release.

version 3.1.5                                    March 31, 2024                                         <u><a href="../man1/swarm.1.html">swarm</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>