<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>TFBS::Word - IUPAC DNA consensus word-based pattern class =head1 DESCRIPTION</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libtfbs-perl">libtfbs-perl_0.7.1+ds-4build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       TFBS::Word - IUPAC DNA consensus word-based pattern class =head1 DESCRIPTION

       TFBS::Word is a base class consisting of universal constructor called by its subclasses
       (TFBS::Matrix::*), and word pattern manipulation methods that are independent of the word type. It is not
       meant to be instantiated itself.

</pre><h4><b>FEEDBACK</b></h4><pre>
       Please send bug reports and other comments to the author.

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Boris</b> <b>Lenhard</b></h4><pre>
       Boris Lenhard &lt;<a href="mailto:Boris.Lenhard@cgb.ki.se">Boris.Lenhard@cgb.ki.se</a>&gt;

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are preceded with an
       underscore.

   <b>new</b>
        Title   : new
        Usage   : my $pwm = TFBS::Matrix::PWM-&gt;new(%args)
        Function: constructor for the TFBS::Matrix::PWM object
        Returns : a new TFBS::Matrix::PWM object
        Args    : # you must specify the -word argument:
                  -word,       # a strig consisting of letters in
                               # IUPAC degenerate DNA alphabet
                               # (any of ACGTSWKMPYBDHVN)

                  #######

                  -name,        # string, OPTIONAL
                  -ID,          # string, OPTIONAL
                  -class,       # string, OPTIONAL
                  -tags         # a hash reference reference, OPTIONAL

   <b>search_seq</b>
        Title   : search_seq
        Usage   : my $siteset = $pwm-&gt;search_seq(%args)
        Function: scans a nucleotide sequence with the pattern represented
                  by the PWM
        Returns : a TFBS::SiteSet object
        Args    : # you must specify either one of the following three:

                  -file,       # the name od a fasta file (single sequence)
                     #or
                  -seqobj      # a Bio::Seq object
                               # (more accurately, a Bio::PrimarySeqobject or a
                               #  subclass thereof)
                     #or
                  -seqstring # a string containing the sequence

                  -max_mismatches,  # number of allowed positions in the site that do
                                    # not match the consensus
                                            # OPTIONAL: default 0

   <b>search_aln</b>
        Title   : search_aln
        Usage   : my $site_pair_set = $pwm-&gt;search_aln(%args)
        Function: Scans a pairwise alignment of nucleotide sequences
                  with the pattern represented by the word: it reports only
                  those hits that are present in equivalent positions of both
                  sequences and exceed a specified threshold score in both, AND
                  are found in regions of the alignment above the specified
                  conservation cutoff value.
        Returns : a TFBS::SitePairSet object
        Args    : # you must specify either one of the following three:

                  -file,       # the name of the alignment file in Clustal
                                      format
                     #or
                  -alignobj      # a Bio::SimpleAlign object
                               # (more accurately, a Bio::PrimarySeqobject or a
                               #  subclass thereof)
                     #or
                  -alignstring # a multi-line string containing the alignment
                               # in clustal format
                  #############

                  -max_mismatches,  # number of allowed positions in the site that do
                                     # not match the consensus
                                             # OPTIONAL: default 0

                  -window,     # size of the sliding window (inn nucleotides)
                               # for calculating local conservation in the
                               # alignment
                               # OPTIONAL: default 50

                  -cutoff      # conservation cutoff (%) for including the
                               # region in the results of the pattern search
                               # OPTIONAL: default "70%"

   <b>to_PWM</b>
   <b>validate_word</b>
   <b>length</b>
perl v5.40.0                                       2024-10-20                         <u>TFBS::Word::<a href="../man3pm/Consensus.3pm.html">Consensus</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>