<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>plast - Parallel Local Sequence Alignment Search Tool</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/plast">plast_2.3.2+dfsg-10_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       plast - Parallel Local Sequence Alignment Search Tool

</pre><h4><b>DESCRIPTION</b></h4><pre>
       plast 2.3.1

              - Build date: 2016-02-09 09:47:45 - OS: Linux-4.3.0-1-amd64 - Compiler: <u>/usr/bin/cc</u> (5.3.1) - Host
              CPU: 4 cores available

       [*] denotes mandatory argument.

       <b>-p</b> [*]:
              Program Name [plastp, tplastn, plastx, tplastx or plastn]

       <b>-d</b> [*]:
              Subject database file

       <b>-i</b> [*]:
              Query database file

       <b>-o</b> :   PLAST report Output File

       <b>-e</b> :   Expectation value

       <b>-n</b> :   Size of neighbourhood peforming ungapped extension

       <b>-s</b> :   Ungapped threshold trigger a small gapped extension

       <b>-g</b> :   threshold for small gapped extension

       <b>-b</b> :   bandwidth for small gapped extension

       <b>-a</b> :   Number of processors to use

       <b>-G</b> :   Cost to open a gap

       <b>-E</b> :   Cost to extend a gap

       <b>-xdrop-ungap</b> :
              X dropoff value for Ungapped alignment (in bits) (zero invokes default behavior 20 bits)

       <b>-X</b> :   X dropoff value for gapped alignment (in bits) (zero invokes default behavior)

       <b>-Z</b> :   X dropoff value for final gapped alignment in bits (0.0 invokes default behavior)

       <b>-index-threshold</b> :
              Index threshold to calculate the similarity between neighbour

       <b>-F</b> :   Filter query sequence

       <b>-M</b> :   Score matrix (BLOSUM62 or BLOSUM50)

       <b>-strand</b> :
              strands for plastn: 'plus', 'minus' or 'both' (default)

       <b>-r</b> :   reward for a nucleotide match (plastn)

       <b>-q</b> :   penalty for a nucleotide mismatch (plastn)

       <b>-force-query-order</b> :
              Force queries ordering in output file.

       <b>-max-database-size</b> :
              Maximum allowed size (in bytes) for a database. If greater, database is segmented.

       <b>-max-hit-per-query</b> :
              Maximum hits per query. 0 value will dump all hits (default)

       <b>-max-hsp-per-hit</b> :
              Maximum alignments per hit. 0 value will dump all hits (default)

       <b>-outfmt</b> :
              Output format: 1 for tabulated (default), 2 for extended tubulated, 4 for NCBI Blast-like.

       <b>-strands-list</b> :
              List of the strands (ex: "1,2,6") to be used when using algo using nucleotids databases.

       <b>-optim-codon-stop</b> :
              size of the allowed range between the last invalid character and the next stop codon

       <b>-factory-dispatcher</b> :
              Factory that creates dispatcher.

       <b>-factory-statistics</b> :
              Factory that creates statistics builder.

       <b>-factory-indexation</b> :
              Factory that creates indexation builder.

       <b>-factory-hit-ungap</b> :
              Factory that creates ungap hits iterator.

       <b>-factory-hit-smallgap</b> :
              Factory that creates small gap hits iterator.

       <b>-factory-hit-fullgap</b> :
              Factory that creates full gap hits iterator.

       <b>-factory-hit-composition</b> :
              Factory that creates composition hits iterator.

       <b>-factory-gap-result</b> :
              Factory that creates gap alignments result.

       <b>-factory-ungap-result</b> :
              Factory that creates ungap alignments result.

       <b>-splitter</b> :
              Factory that creates an alignment splitter. String: 'normal' or 'banded' (default)

       <b>-optim-filter-ungap</b> :
              Optimization that filters out through ungap alignments.

       <b>-bargraph</b> :
              Display a progress bar during execution.

       <b>-bargraph-size</b> :
              Nb of characters of the bargraph.

       <b>-progression-file</b> :
              Dump in a file the current execution percentage.

       <b>-verbose</b> :
              Display information during algorithm execution.

       <b>-full-stats</b> :
              Dump algorithm statistics.

       <b>-stats</b> :
              Dump generic statistics.

       <b>-stats-fmt</b> :
              Format of statistics: 'raw' (default) or 'xml'

       <b>-stats-auto</b> :
              Automatic stats file creation

       <b>-alignment-progress</b> :
              Dump in a file the growing number of ungap/ungap alignments during algorithm.

       <b>-resources-progress</b> :
              Dump in a file information about resources during algorithm.

       <b>-plastrc</b> :
              Pathname of the plast config file.

       <b>-xmlfilter</b> :
              Uri of a XML filter file.

       <b>-seeds-use-ratio</b> :
              Ratio of seeds to be used.

       <b>-seeds-index-filter</b> :
              seeds length to be used for the indexation filter.

       <b>-complete-subject-database-stats-file</b> :
              File path to the stats of the complete subject database

       <b>-W</b> :   size of the seeds

       <b>-h</b> :   help

       Citing  PLAST:  Nguyen  VH,  Lavenier  D. (2009) PLAST: parallel local alignment search tool for database
       comparison. BMC Bioinformatics, vol 10, no 329.

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

plast 2.3.1                                       February 2016                                         <u><a href="../man1/PLAST.1.html">PLAST</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>