<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>sga - String Graph Assembler: de novo genome assembler that uses string graphs</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/sga">sga_0.10.15-7build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       sga - String Graph Assembler: de novo genome assembler that uses string graphs

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>sga</b> <u>&lt;command&gt;</u> [<u>options</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Program: sga Version: 0.10.15 Contact: Jared Simpson [<a href="mailto:js18@sanger.ac.uk">js18@sanger.ac.uk</a>]

   <b>Commands:</b>
       preprocess
              filter and quality-trim reads

       index  build the BWT and FM-index for a set of reads

       merge  merge multiple BWT/FM-index files into a single index

       bwt2fa transform a bwt back into a set of sequences

       correct
              correct sequencing errors in a set of reads

       fm-merge
              merge unambiguously overlapped sequences using the FM-index

       overlap
              compute overlaps between reads

       assemble
              generate contigs from an assembly graph

       oview  view overlap alignments

       subgraph
              extract a subgraph from a graph

       filter remove reads from a data set

       rmdup  duplicate read removal

       gen-ssa
              generate a sampled suffix array for the given set of reads

       scaffold
              generate ordered sets of contigs using distance estimates

       scaffold2fasta
              convert the output of the scaffold subprogram into a fasta file

       gapfill
              fill intra-scaffold gaps

   <b>Variant</b> <b>Calling</b> <b>Commands:</b>
       graph-diff
              compare reads to find sequence variants

       graph-concordance
              check called variants for representation in the assembly graph

       rewrite-evidence-bam
              fill in sequence and quality information for a variant evidence BAM

       haplotype-filter
              filter out low-quality haplotypes

       somatic-variant-filters
              filter out low-quality variants

   <b>Experimental</b> <b>commands:</b>
       preqc  perform pre-assembly quality checks on a set of reads

       stats  print summary statistics about a read set

       filterBAM
              filter out contaminating mate-pair data in a BAM file

       cluster
              find clusters of reads belonging to the same connected component in an assembly graph

       kmer-count
              extract all kmers from a BWT file

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       Report bugs to <a href="mailto:js18@sanger.ac.uk">js18@sanger.ac.uk</a>

sga 0.10.15                                       November 2022                                           <u><a href="../man1/SGA.1.html">SGA</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>