<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>wtdbg2 - de novo sequence assembler for long noisy reads</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/wtdbg2">wtdbg2_2.5-10_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       wtdbg2 - de novo sequence assembler for long noisy reads

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>wtdbg2</b> [<u>options</u>] <u>-i</u> <u>&lt;reads.fa&gt;</u> <u>-o</u> <u>&lt;prefix&gt;</u> [<u>reads.fa</u> ...]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       WTDBG:  De  novo  assembler  for  long  noisy sequences Author: Jue Ruan &lt;<a href="mailto:ruanjue@gmail.com">ruanjue@gmail.com</a>&gt; Version: 2.5
       (20190621)

</pre><h4><b>OPTIONS</b></h4><pre>

       <b>-i</b> &lt;string&gt; Long reads sequences file (REQUIRED; can be multiple), []

       <b>-o</b> &lt;string&gt; Prefix of output files (REQUIRED), []

       <b>-t</b> &lt;int&gt;
              Number of threads, 0 for all cores, [4]

       <b>-f</b>     Force to overwrite output files

       <b>-x</b> &lt;string&gt; Presets, comma delimited, []

       preset1/rsII/rs: <b>-p</b> 21 <b>-S</b> 4 <b>-s</b> 0.05 <b>-L</b> 5000
              preset2: <b>-p</b> 0 <b>-k</b> 15 <b>-AS</b> 2 <b>-s</b> 0.05 <b>-L</b> 5000 preset3: <b>-p</b> 19 <b>-AS</b> 2 <b>-s</b> 0.05 <b>-L</b> 5000

              sequel/sq

              nanopore/ont:

              (genome size &lt; 1G: preset2) <b>-p</b> 0 <b>-k</b> 15 <b>-AS</b> 2 <b>-s</b> 0.05 <b>-L</b> 5000 (genome size &gt;= 1G:  preset3)  <b>-p</b>  19
              <b>-AS</b> 2 <b>-s</b> 0.05 <b>-L</b> 5000

              preset4/corrected/ccs: <b>-p</b> 21 <b>-k</b> 0 <b>-AS</b> 4 <b>-K</b> 0.05 <b>-s</b> 0.5

       <b>-g</b> &lt;number&gt; Approximate genome size (k/m/g suffix allowed) [0]

       <b>-X</b> &lt;float&gt;
              Choose the best &lt;float&gt; depth from input reads(effective with <b>-g</b>) [50.0]

       <b>-L</b> &lt;int&gt;
              Choose  the  longest  subread  and drop reads shorter than &lt;int&gt; (5000 recommended for PacBio) [0]
              Negative integer indicate tidying read names too, e.g. <b>-5000</b>.

       <b>-k</b> &lt;int&gt;
              Kmer fsize, 0 &lt;= k &lt;= 23, [0]

       <b>-p</b> &lt;int&gt;
              Kmer psize, 0 &lt;= p &lt;= 23, [21] k + p &lt;= 25, seed is &lt;k-mer&gt;+&lt;p-homopolymer-compressed&gt;

       <b>-K</b> &lt;float&gt;
              Filter high frequency kmers, maybe repetitive, [1000.05] &gt;= 1000 and indexing  &gt;=  (1  -  0.05)  *
              total_kmers_count

       <b>-S</b> &lt;float&gt;
              Subsampling  kmers,  1/(&lt;-S&gt;)  kmers  are  indexed,  [4.00] <b>-S</b> is very useful in saving memory and
              speeding up please note that subsampling kmers will have less matched length

       <b>-l</b> &lt;float&gt;
              Min length of alignment, [2048]

       <b>-m</b> &lt;float&gt;
              Min matched length by kmer matching, [200]

       <b>-R</b>     Enable realignment mode

       <b>-A</b>     Keep contained reads during alignment

       <b>-s</b> &lt;float&gt;
              Min similarity, calculated by kmer matched length / aligned length, [0.05]

       <b>-e</b> &lt;int&gt;
              Min read depth of a valid edge, [3]

       <b>-q</b>     Quiet

       <b>-v</b>     Verbose (can be multiple)

       <b>-V</b>     Print version information and then exit

       <b>--help</b> Show more options

              ** more options ** <b>--cpu</b> &lt;int&gt;

              See <b>-t</b> 0, default: all cores

       <b>--input</b> &lt;string&gt; +

              See <b>-i</b>

       <b>--force</b>

              See <b>-f</b>

       <b>--prefix</b> &lt;string&gt;

              See <b>-o</b>

       <b>--preset</b> &lt;string&gt;

              See <b>-x</b>

       <b>--kmer-fsize</b> &lt;int&gt;

              See <b>-k</b> 0

       <b>--kmer-psize</b> &lt;int&gt;

              See <b>-p</b> 21

       <b>--kmer-depth-max</b> &lt;float&gt;

              See <b>-K</b> 1000.05

       <b>-E</b>, <b>--kmer-depth-min</b> &lt;int&gt;

              Min kmer frequency, [2]

       <b>--kmer-subsampling</b> &lt;float&gt;

              See <b>-S</b> 4.0

       <b>--kbm-parts</b> &lt;int&gt;

              Split total reads into multiple parts, index one part by one to save memory, [1]

       <b>--aln-kmer-sampling</b> &lt;int&gt;

              Select no more than n seeds in a query bin, default: 256

       <b>--dp-max-gap</b> &lt;int&gt;

              Max number of <a href="../man256bp/bin.256bp.html">bin</a>(256bp) in one gap, [4]

       <b>--dp-max-var</b> &lt;int&gt;

              Max number of <a href="../man256bp/bin.256bp.html">bin</a>(256bp) in one deviation, [4]

       <b>--dp-penalty-gap</b> &lt;int&gt;

              Penalty for BIN gap, [-7]

       <b>--dp-penalty-var</b> &lt;int&gt;

              Penalty for BIN deviation, [-21]

       <b>--aln-min-length</b> &lt;int&gt;

              See <b>-l</b> 2048

       <b>--aln-min-match</b> &lt;int&gt;

              See <b>-m</b> 200. Here the num of matches counting basepair of the matched kmer's regions

       <b>--aln-min-similarity</b> &lt;float&gt;

              See <b>-s</b> 0.05

       <b>--aln-max-var</b> &lt;float&gt;

              Max length variation of two aligned fragments, default: 0.25

       <b>--aln-dovetail</b> &lt;int&gt;

              Retain dovetail overlaps only, the max overhang size is  &lt;--aln-dovetail&gt;,  the  value  should  be
              times of 256, <b>-1</b> to disable filtering, default: 256

       <b>--aln-strand</b> &lt;int&gt;

              1:  forward,  2: reverse, 3: both. Please don't change the deault value 3, unless you exactly know
              what you are doing

       <b>--aln-maxhit</b> &lt;int&gt;

              Max n hits for each read in build graph, default: 1000

       <b>--aln-bestn</b> &lt;int&gt;

              Use best n hits for each read in build graph, 0: keep all, default: 500 &lt;prefix&gt;.alignments always
              store all alignments

       <b>-R</b>, <b>--realign</b>

              Enable      re-alignment,       see       <b>--realn-kmer-psize</b>=<u>15</u>,       <b>--realn-kmer-subsampling</b>=<u>1</u>,
              <b>--realn-min-length</b>=<u>2048</u>, <b>--realn-min-match</b>=<u>200</u>, <b>--realn-min-similarity</b>=<u>0</u>.1, <b>--realn-max-var</b>=<u>0</u>.25

       <b>--realn-kmer-psize</b> &lt;int&gt;

              Set kmer-psize in realignment, (kmer-ksize always eq 0), default:15

       <b>--realn-kmer-subsampling</b> &lt;int&gt;

              Set kmer-subsampling in realignment, default:1

       <b>--realn-min-length</b> &lt;int&gt;

              Set aln-min-length in realignment, default: 2048

       <b>--realn-min-match</b> &lt;int&gt;

              Set aln-min-match in realignment, default: 200

       <b>--realn-min-similarity</b> &lt;float&gt;

              Set aln-min-similarity in realignment, default: 0.1

       <b>--realn-max-var</b> &lt;float&gt;

              Set aln-max-var in realignment, default: 0.25

       <b>-A</b>, <b>--aln-noskip</b>

              Even a read was contained in previous alignment, still align it against other reads

       <b>--keep-multiple-alignment-parts</b>

              By default, wtdbg will keep only the best alignment between two reads after chainning. This option
              will disable it, and keep multiple

       <b>--verbose</b> +

              See <b>-v</b>. <b>-vvvv</b> will display the most detailed information

       <b>--quiet</b>

              See <b>-q</b>

       <b>--limit-input</b> &lt;int&gt;

              Limit the input sequences to at most &lt;int&gt; M bp. Usually for test

       <b>-L</b> &lt;int&gt;, <b>--tidy-reads</b> &lt;int&gt;

              Default:  0.  Pick longest subreads if possible. Filter reads less than &lt;--tidy-reads&gt;. Please add
              <b>--tidy-name</b> or set <b>--tidy-reads</b> to nagetive value if want to rename reads. Set to 0 bp to  disable
              tidy. Suggested value is 5000 for pacbio RSII reads

       <b>--tidy-name</b>

              Rename reads into 'S%010d' format. The first read is named as S0000000001

       <b>--rdname-filter</b> &lt;string&gt;

              A  file  contains  lines  of reads name to be discarded in loading. If you want to filter reads by
              yourself, please also set <b>-X</b> 0

       <b>--rdname-includeonly</b> &lt;string&gt;

              Reverse manner with <b>--rdname-filter</b>

       <b>-g</b> &lt;number&gt;, <b>--genome-size</b> &lt;number&gt;

              Provide genome size, e.g. 100.4m, 2.3g. In this version, it  is  used  with  <b>-X</b>/--rdcov-cutoff  in
              selecting reads just after readed all.

       <b>-X</b> &lt;float&gt;, <b>--rdcov-cutoff</b> &lt;float&gt;

              Default: 50.0. Retaining 50.0 folds of genome coverage, combined with <b>-g</b> and <b>--rdcov-filter</b>.

       <b>--rdcov-filter</b> [0|1]

              Default 0. Strategy 0: retaining longest reads. Strategy 1: retaining medain length reads.

       <b>--err-free-nodes</b>

              Select  nodes  from error-free-sequences only. E.g. you have contigs assembled from NGS-WGS reads,
              and long noisy  reads.   You  can  type  '--err-free-seq  your_ctg.fa  <b>--input</b>  your_long_reads.fa
              <b>--err-free-nodes</b>' to perform assembly somehow act as long-reads scaffolding

       <b>--node-len</b> &lt;int&gt;

              The  default value is 1024, which is times of KBM_BIN_SIZE(always equals 256 bp). It specifies the
              length of intervals (or call nodes after selecting).  kbm indexs sequences into BINs of 256 bp  in
              size,  so  that  many  parameter  should  be  times  of 256 bp. There are: <b>--node-len</b>, <b>--node-ovl</b>,
              <b>--aln-min-length</b>,  <b>--aln-dovetail</b>  .    Other  parameters  are  counted  in  BINs,   <b>--dp-max-gap</b>,
              <b>--dp-max-var</b> .

       <b>--node-matched-bins</b> &lt;int&gt;

              Min matched bins in a node, default:1

       <b>--node-ovl</b> &lt;int&gt;

              Default: 256. Max overlap size between two adjacent intervals in any read. It is used in selecting
              best nodes representing reads in graph

       <b>--node-drop</b> &lt;float&gt;

              Default:  0.25.  Will  discard  an  node  when  has  more this ratio intervals are conflicted with
              previous generated node

       <b>-e</b> &lt;int&gt;, <b>--edge-min=</b>&lt;int&gt;

              Default: 3. The minimal depth of a valid edge is set to 3. In another word, Valid  edges  must  be
              supported  by  at  least  3 reads When the sequence depth is low, have a try with <b>--edge-min</b> 2. Or
              very high, try <b>--edge-min</b> 4

       <b>--edge-max-span</b> &lt;int&gt;

              Default: 1024 BINs. Program will build edges of length no large than 1024

       <b>--drop-low-cov-edges</b>

              Don't attempt to rescue low coverage edges

       <b>--node-min</b> &lt;int&gt;

              Min depth of an interval to be selected as valid node. Defaultly, this value is automatically  the
              same with <b>--edge-min</b>.

       <b>--node-max</b> &lt;int&gt;

              Nodes  with  too high depth will be regarded as repetitive, and be masked. Default: 200, more than
              200 reads contain this node

       <b>--ttr-cutoff-depth</b> &lt;int&gt;, 0

       <b>--ttr-cutoff-ratio</b> &lt;float&gt;, 0.5

              Tiny Tandom Repeat. A node located inside ttr will bring noisy in graph,  should  be  masked.  The
              pattern  of  such  nodes  is:  depth  &gt;=  &lt;--ttr-cutoff-depth&gt;, and none of their edges have depth
              greater than depth * &lt;--ttr-cutoff-ratio 0.5&gt; set <b>--ttr-cutoff-depth</b> 0 to disable ttr masking

       <b>--dump-kbm</b> &lt;string&gt;

              Dump kbm index into file for loaded by `kbm` or `wtdbg`

       <b>--dump-seqs</b> &lt;string&gt;

              Dump kbm index (only sequences, no k-mer index) into file for loaded by `kbm`  or  `wtdbg`  Please
              note: normally load it with <b>--load-kbm</b>, not with <b>--load-seqs</b>

       <b>--load-kbm</b> &lt;string&gt;

              Instead  of  reading  sequences  and building kbm index, which is time-consumed, loading kbm-index
              from already dumped file.  Please note that, once kbm-index is mmaped by kbm <b>-R</b> &lt;kbm-index&gt; start,
              will just get the shared memory in minute time.  See `kbm` <b>-R</b> &lt;your_seqs.kbmidx&gt; [start | stop]

       <b>--load-seqs</b> &lt;string&gt;

              Similar with <b>--load-kbm</b>, but only use the sequences in kbmidx, and rebuild index in process's RAM.

       <b>--load-alignments</b> &lt;string&gt; +

              `wtdbg` output reads' alignments into &lt;--prefix&gt;.alignments, program can load them to fastly build
              assembly graph. Or you can offer other source of alignments  to  `wtdbg`.  When  <b>--load-alignment</b>,
              will only reading long sequences but skip building kbm index You can type <b>--load-alignments</b> &lt;file&gt;
              more than once to load alignments from many files

       <b>--load-clips</b> &lt;string&gt;

              Combined with <b>--load-nodes</b>. Load reads clips. You can find it in `wtdbg`'s &lt;--prefix&gt;.clps

       <b>--load-nodes</b> &lt;sting&gt;

              Load  dumped  nodes  from  previous  execution  for  fast  construct the assembly graph, should be
              combined with <b>--load-clips</b>. You can find it in `wtdbg`'s &lt;--prefix&gt;.1.nodes

       <b>--bubble-step</b> &lt;int&gt;

              Max step to search a bubble, meaning the max step from the  starting  node  to  the  ending  node.
              Default: 40

       <b>--tip-step</b> &lt;int&gt;

              Max step to search a tip, 10

       <b>--ctg-min-length</b> &lt;int&gt;

              Min length of contigs to be output, 5000

       <b>--ctg-min-nodes</b> &lt;int&gt;

              Min num of nodes in a contig to be output, 3

       <b>--minimal-output</b>

              Will generate as less output files (&lt;--prefix&gt;.*) as it can

       <b>--bin-complexity-cutoff</b> &lt;int&gt;

              Used  in  filtering  BINs. If a BIN has less indexed valid kmers than &lt;--bin-complexity-cutoff 2&gt;,
              masks it.

       <b>--no-local-graph-analysis</b>

              Before building edges, for each node, local-graph-analysis reads all related reads  and  according
              nodes,  and  builds a local graph to judge whether to mask it The analysis aims to find repetitive
              nodes

       <b>--no-read-length-sort</b>

              Defaultly, `wtdbg` sorts input sequences by length DSC. The order of reads affects the  generating
              of nodes in selecting important intervals

       <b>--keep-isolated-nodes</b>

              In graph clean, `wtdbg` normally masks isolated (orphaned) nodes

       <b>--no-read-clip</b>

              Defaultly,  `wtdbg` clips a input sequence by analyzing its overlaps to remove high error endings,
              rolling-circle repeats (see PacBio CCS), and chimera.  When building edges, clipped  region  won't
              contribute. However, `wtdbg` will use them in the final linking of unitigs

       <b>--no-chainning-clip</b>

              Defaultly,  performs  alignments  chainning in read clipping ** If '--aln-bestn 0 <b>--no-read-clip</b>',
              alignments will be parsed directly, and less RAM spent on recording alignments

</pre><h4><b>AUTHOR</b></h4><pre>
        This manpage was written by Andreas Tille for the Debian distribution and
        can be used for any other usage of the program.

wtdbg2 2.5                                         April 2020                                          <u><a href="../man1/WTDBG2.1.html">WTDBG2</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>