<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>mac3_predictd - Model-based Analysis for ChIP-Sequencing</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/macs">macs_3.0.2-1build3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       mac3_predictd - Model-based Analysis for ChIP-Sequencing

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: macs3 predictd [-h] <b>-i</b> IFILE [IFILE ...]

       [-f {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE}]
              [-g GSIZE] [-s TSIZE] [--bw BW] [--d-min D_MIN] [-m MFOLD MFOLD] [--outdir OUTDIR] [--rfile RFILE]
              [--buffer-size BUFFER_SIZE] [--verbose VERBOSE]

   <b>options:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>-i</b>, <b>--ifile</b> IFILE [IFILE ...]
              ChIP-seq alignment file. If multiple files are given as '-t A B C', then they will all be read and
              combined. REQUIRED.

       <b>-f</b>, <b>--format</b> {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE}
              Format of tag file, "AUTO", "BED" or "ELAND" or "ELANDMULTI" or "ELANDEXPORT" or "SAM" or "BAM" or
              "BOWTIE" or "BAMPE" or "BEDPE". The default AUTO option will let MACS decide which format the file
              is.   However, if you want to decide the average insertion size/fragment size from PE data such as
              BEDPE or BAMPE, please specify the format as  BAMPE  or  BEDPE  since  MACS3  won't  automatically
              recognize  three two formats with <b>-f</b> AUTO. Please be aware that in PE mode, <b>-g</b>, <b>-s</b>, <b>--bw</b>, <b>--d-min</b>,
              <b>-m</b>, and <b>--rfile</b> have NO effect.  DEFAULT: "AUTO"

       <b>-g</b>, <b>--gsize</b> GSIZE
              Effective  genome  size.  It  can  be  1.0e+9  or  1000000000,   or   shortcuts:'hs'   for   human
              (2,913,022,398),  'mm'  for  mouse (2,652,783,500), 'ce' for C. elegans (100,286,401) and 'dm' for
              fruitfly (142,573,017), Default:hs. The effective genome size numbers for the above  four  species
              are   collected   from   Deeptools  https:  //deeptools.readthedocs.io/en/develop/content/feature/
              effectiveGenomeSize.html Please refer to deeptools to define the best genome size you plan to use.

       <b>-s</b>, <b>--tsize</b> TSIZE
              Tag size. This will override the auto detected tag size. DEFAULT: Not set

       <b>--bw</b> BW
              Band width for picking regions to compute fragment size. This value is only  used  while  building
              the shifting model. DEFAULT: 300

       <b>--d-min</b> D_MIN
              Minimum  fragment  size  in basepair. Any predicted fragment size less than this will be excluded.
              DEFAULT: 20

       <b>-m</b>, <b>--mfold</b> MFOLD MFOLD
              Select the regions within MFOLD range of highconfidence enrichment  ratio  against  background  to
              build  model. Fold-enrichment in regions must be lower than upper limit, and higher than the lower
              limit. Use as "-m 10 30". DEFAULT:5 50

       <b>--outdir</b> OUTDIR
              If specified all output files will be written to that  directory.  Default:  the  current  working
              directory

       <b>--rfile</b> RFILE
              PREFIX  of filename of R script for drawing X-correlation figure. DEFAULT:'predictd_model.R' and R
              file will be predicted_model.R

       <b>--buffer-size</b> BUFFER_SIZE
              Buffer size for incrementally increasing internal array size to store reads alignment information.
              In most cases, you don't have to change this parameter.  However, if there  are  large  number  of
              chromosomes/contigs/scaffolds in your alignment, it's recommended to specify a smaller buffer size
              in  order to decrease memory usage (but it will take longer time to read alignment files). Minimum
              memory requested for reading an alignment file is about # of CHROMOSOME * BUFFER_SIZE *  8  Bytes.
              DEFAULT: 100000

       <b>--verbose</b> VERBOSE
              Set  verbose  level  of runtime message. 0: only show critical message, 1: show additional warning
              message, 2: show process information, 3: show debug messages.  DEFAULT:2

macs3 predictd 3.0.2                               March 2025                                   <u><a href="../man1/MAC3_PREDICTD.1.html">MAC3_PREDICTD</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>