<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>faidx - an index enabling random access to FASTA and FASTQ files</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libhts3t64">libhts3t64_1.21+ds-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       faidx - an index enabling random access to FASTA and FASTQ files

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <u>file.fa</u>.fai, <u>file.fasta</u>.fai, <u>file.fq</u>.fai, <u>file.fastq</u>.fai

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Using  an  <b>fai</b>  <b>index</b>  file in conjunction with a FASTA/FASTQ file containing reference sequences enables
       efficient access to arbitrary regions within those reference sequences.  The index file typically has the
       same filename as the corresponding FASTA/FASTQ file, with <b>.fai</b> appended.

       An <b>fai</b> <b>index</b> file is a text file consisting of lines each with five TAB-delimited  columns  for  a  FASTA
       file and six for FASTQ:
       <b>NAME</b>         Name of this reference sequence
       <b>LENGTH</b>       Total length of this reference sequence, in bases
       <b>OFFSET</b>       Offset in the FASTA/FASTQ file of this sequence's first base
       <b>LINEBASES</b>    The number of bases on each line
       <b>LINEWIDTH</b>    The number of bytes in each line, including the newline
       <b>QUALOFFSET</b>   Offset of sequence's first quality within the FASTQ file

       The  <b>NAME</b>  and  <b>LENGTH</b> columns contain the same data as would appear in the <b>SN</b> and <b>LN</b> fields of a SAM <b>@SQ</b>
       header for the same reference sequence.

       The <b>OFFSET</b> column contains the offset within the FASTA/FASTQ file, in bytes starting from  zero,  of  the
       first  base  of  this  reference sequence, i.e., of the character following the newline at the end of the
       header line (the "<b>&gt;</b>" line in FASTA, "<b>@</b>" in FASTQ). Typically the lines of a <b>fai</b> <b>index</b> file appear in  the
       order in which the reference sequences appear in the FASTA/FASTQ file, so <b>.fai</b> files are typically sorted
       according to this column.

       The  <b>LINEBASES</b>  column  contains  the number of bases in each of the sequence lines that form the body of
       this reference sequence, apart from the final line which may be shorter.  The <b>LINEWIDTH</b>  column  contains
       the  number  of  <u>bytes</u> in each of the sequence lines (except perhaps the final line), thus differing from
       <b>LINEBASES</b> in that it also counts the bytes forming the line terminator.

       The <b>QUALOFFSET</b> works the same way as <b>OFFSET</b> but for the first quality score of this  reference  sequence.
       This  would  be  the  first  character following the newline at the end of the "<b>+</b>" line.  For FASTQ files
       only.

   <b>FASTA</b> <b>Files</b>
       In order to be indexed with <b>samtools</b> <b>faidx</b>, a FASTA file must be a text file of the form

              &gt;<u>name</u> [<u>description</u>...]
              ATGCATGCATGCATGCATGCATGCATGCAT
              GCATGCATGCATGCATGCATGCATGCATGC
              ATGCAT
              &gt;<u>name</u> [<u>description</u>...]
              ATGCATGCATGCAT
              GCATGCATGCATGC
              [...]

       In particular, each reference sequence must be "well-formatted", i.e., all of its sequence lines must  be
       the  same  length,  apart  from  the final sequence line which may be shorter.  (While this sequence line
       length must be the same within each sequence, it may vary between different reference  sequences  in  the
       same FASTA file.)

       This  also  means that although the FASTA file may have Unix- or Windows-style or other line termination,
       the newline characters present must be consistent, at least within each reference sequence.

       The <b>samtools</b> implementation uses the first word of the "<b>&gt;</b>" header  line  text  (i.e.,  up  to  the  first
       whitespace character, having skipped any initial whitespace after the "&gt;") as the <b>NAME</b> column.

   <b>FASTQ</b> <b>Files</b>
       FASTQ files for indexing work in the same way as the FASTA files.

              @<u>name</u> [<u>description...]</u>
              ATGCATGCATGCATGCATGCATGCATGCAT
              GCATGCATGCATGCATGCATGCATGCATGC
              ATGCAT
              +
              FFFA@@FFFFFFFFFFHHB:::@BFFFFGG
              HIHIIIIIIIIIIIIIIIIIIIIIIIFFFF
              8011&lt;&lt;
              @<u>name</u> [<u>description...]</u>
              ATGCATGCATGCAT
              GCATGCATGCATGC
              +
              IIA94445EEII==
              =&gt;IIIIIIIIICCC
              [...]

       Quality lines must be wrapped at the same length as the corresponding sequence lines.

</pre><h4><b>EXAMPLE</b></h4><pre>
       For example, given this FASTA file

              &gt;one
              ATGCATGCATGCATGCATGCATGCATGCAT
              GCATGCATGCATGCATGCATGCATGCATGC
              ATGCAT
              &gt;two another chromosome
              ATGCATGCATGCAT
              GCATGCATGCATGC

       formatted with Unix-style (LF) line termination, the corresponding fai index would be
              one   66    5   30   31
              two   28   98   14   15

       If the FASTA file were formatted with Windows-style (CR-LF) line termination, the fai index would be
              one   66     6   30   32
              two   28   103   14   16

       An example FASTQ file

              @fastq1
              ATGCATGCATGCATGCATGCATGCATGCAT
              GCATGCATGCATGCATGCATGCATGCATGC
              ATGCAT
              +
              FFFA@@FFFFFFFFFFHHB:::@BFFFFGG
              HIHIIIIIIIIIIIIIIIIIIIIIIIFFFF
              8011&lt;&lt;
              @fastq2
              ATGCATGCATGCAT
              GCATGCATGCATGC
              +
              IIA94445EEII==
              =&gt;IIIIIIIIICCC

       Formatted with Unix-style line termination would give this fai index
              fastq1   66     8   30   31    79
              fastq2   28   156   14   15   188

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/samtools.1.html">samtools</a></u>(1)

       https://en.wikipedia.org/wiki/FASTA_format

       https://en.wikipedia.org/wiki/FASTQ_format

              Further description of the FASTA and FASTQ formats

htslib                                              June 2018                                           <u><a href="../man5/faidx.5.html">faidx</a></u>(5)
</pre>
 </div>
</div></section>
</div>
</body>
</html>