<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>vcf - Variant Call Format</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libhts3t64">libhts3t64_1.21+ds-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       vcf - Variant Call Format

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The  Variant  Call Format (VCF) is a TAB-delimited format with each data line consisting of the following
       fields:
        1    <b>CHROM</b>    CHROMosome name
        2    <b>POS</b>      the left-most POSition of the variant
        3    <b>ID</b>       unique variant IDentifier
        4    <b>REF</b>      the REFerence allele
        5    <b>ALT</b>      the ALTernate allele(s) (comma-separated)
        6    <b>QUAL</b>     variant/reference QUALity
        7    <b>FILTER</b>   FILTERs applied
        8    <b>INFO</b>     INFOrmation related to the variant (semicolon-separated)
        9    <b>FORMAT</b>   FORMAT of the genotype fields (optional; colon-separated)
       10+   <b>SAMPLE</b>   SAMPLE genotypes and per-sample information (optional)

       The following table gives the <b>INFO</b> tags used by samtools and bcftools.

       <b>AF1</b>    Max-likelihood estimate of the site allele frequency (AF) of the first ALT allele (double)

       <b>DP</b>     Raw read depth (without quality filtering) (int)

       <b>DP4</b>    # high-quality reference forward bases, ref reverse, alternate for and alt rev bases (int[4])

       <b>FQ</b>     Consensus quality. Positive: sample genotypes different; negative: otherwise (int)

       <b>MQ</b>     Root-Mean-Square mapping quality of covering reads (int)

       <b>PC2</b>    Phred probability of AF in group1 samples being larger (,smaller) than in group2 (int[2])

       <b>PCHI2</b>  Posterior weighted chi^2 P-value between group1 and group2 samples (double)

       <b>PV4</b>    P-value for strand bias, baseQ bias, mapQ bias and tail distance bias (double[4])

       <b>QCHI2</b>  Phred-scaled PCHI2 (int)

       <b>RP</b>     # permutations yielding a smaller PCHI2 (int)

       <b>CLR</b>    Phred log ratio of genotype likelihoods with and without the trio/pair constraint (int)

       <b>UGT</b>    Most probable genotype configuration without the trio constraint (string)

       <b>CGT</b>    Most probable configuration with the trio constraint (string)

       <b>VDB</b>    Tests variant positions within reads. Intended for filtering RNA-seq artifacts around splice sites
              (float)

       <b>RPB</b>    Mann-Whitney rank-sum test for tail distance bias (float)

       <b>HWE</b>    Hardy-Weinberg equilibrium test (Wigginton et al) (float)

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       https://github.com/samtools/hts-specs
              The full VCF/BCF file format specification

       <u>A</u> <u>note</u> <u>on</u> <u>exact</u> <u>tests</u> <u>of</u> <u>Hardy-Weinberg</u> <u>equilibrium</u>
              Wigginton JE et al PMID:15789306

htslib                                             August 2013                                            <u><a href="../man5/vcf.5.html">vcf</a></u>(5)
</pre>
 </div>
</div></section>
</div>
</body>
</html>