<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Maq - Mapping and Assembly with Qualities</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/maq">maq_0.7.1-10_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Maq - Mapping and Assembly with Qualities

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>maq</b> <u>command</u> [<u>options</u>] <u>arguments</u>

       <b>maq.pl</b> <u>command</u> [<u>options</u>] <u>arguments</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Maq  is  a  software  that  builds  mapping  assemblies from short reads generated by the next-generation
       sequencing machines. It is particularly designed for Illumina-Solexa  1G  Genetic  Analyzer,  and  has  a
       preliminary functionality to handle AB SOLiD data.

       With Maq you can:

       • Fast align Illumina/SOLiD reads to the reference genome. With the default options, one million pairs of
         reads can be mapped to the human genome in about 10 CPU hours with less than 1G memory.

       • Accurately measure the error probability of the alignment of each individual read.

       • Call  the  consensus  genotypes,  including  homozygous  and  heterozygous  polymorphisms, with a Phred
         probabilistic quality assigned to each base.

       • Find short indels with paired end reads.

       • Accurately find large scale genomic deletions and translocations with paired end reads.

       • Discover potential CNVs by checking read depth.

       • Evaluate the accuracy of raw base qualities from sequencers and help to check the systematic errors.

       However, Maq can <b>NOT</b>:

       • Do <u>de</u> <u>novo</u> assembly. (Maq can only call the consensus by mapping reads to a known reference.)

       • Map shorts reads against themselves. (Maq can only find complete overlap between reads.)

       • Align capillary reads or 454 reads to the reference. (Maq cannot align reads longer than 63bp.)

</pre><h4><b>MAQ</b> <b>COMMANDS</b></h4><pre>
       <b>Key</b> <b>Commands</b>

       <b>fasta2bfa</b>  <b>maq</b> <b>fasta2bfa</b> <u>in.ref.fasta</u> <u>out.ref.bfa</u>

                  Convert sequences in FASTA format to Maq's BFA (binary FASTA) format.

       <b>fastq2bfq</b>  <b>maq</b> <b>fastq2bfq</b> [<b>-n</b> <u>nreads</u>] <u>in.read.fastq</u> <u>out.read.bfq</u>⎪<u>out.prefix</u>

                  Convert reads in FASTQ format to Maq's BFQ (binary FASTQ) format.

                  <b>OPTIONS:</b>

                  <b>-n</b> <u>INT</u>   number of reads per file [not specified]

       <b>map</b>        <b>maq</b> <b>map</b> [<b>-n</b> <u>nmis</u>] [<b>-a</b> <u>maxins</u>] [<b>-c</b>] [<b>-1</b> <u>len1</u>] [<b>-2</b> <u>len2</u>] [<b>-d</b> <u>adap3</u>] [<b>-m</b> <u>mutrate</u>]  [<b>-u</b>  <u>unmapped</u>]
                  [-e  <u>maxerr</u>]  [<b>-M</b>  c⎪g]  [<b>-N</b>]  [<b>-H</b>  <u>allhits</u>]  [<b>-C</b> <u>maxhits</u>] <u>out.aln.map</u> <u>in.ref.bfa</u> <u>in.read1.bfq</u>
                  [<u>in.read2.bfq</u>] 2&gt; <u>out.map.log</u>

                  Map reads to the reference sequences.

                  <b>OPTIONS:</b>

                  <b>-n</b> <u>INT</u>   Number of maximum mismatches that can always be found [2]

                  <b>-a</b> <u>INT</u>   Maximum outer distance for a correct read pair [250]

                  <b>-A</b> <u>INT</u>   Maximum outer distance of two RF paied read (0 for disable) [0]

                  <b>-c</b>       Map reads in the colour space (for SOLiD only)

                  <b>-1</b> <u>INT</u>   Read length for the first read, 0 for auto [0]

                  <b>-2</b> <u>INT</u>   Read length for the second read, 0 for auto [0]

                  <b>-m</b> <u>FLOAT</u> Mutation rate between the reference sequences and the reads [0.001]

                  <b>-d</b> <u>FILE</u>  Specify a file containing a single line of the 3'-adapter sequence [null]

                  <b>-u</b> <u>FILE</u>  Dump unmapped reads and reads containing more than <u>nmis</u> mismatches to a separate file
                           [null]

                  <b>-e</b> <u>INT</u>   Threshold on the sum of mismatching base qualities [70]

                  <b>-H</b> <u>FILE</u>  Dump multiple/all 01-mismatch hits to <u>FILE</u> [null]

                  <b>-C</b> <u>INT</u>   Maximum number of hits to output. Unlimited if larger than 512. [250]

                  <b>-M</b> c⎪g   methylation alignment mode. All C (or G) on the forward strand will be changed  to  T
                           (or A). This option is for testing only.

                  <b>-N</b>       store  the  mismatch  position in the output file <u>out.aln.map</u>. When this option is in
                           use, the maximum allowed read length is 55bp.

                  <b>NOTE:</b>

                  * Paired end reads should be prepared in two files, one for each end, with reads are sorted in
                    the same order. This means the k-th read in the first file is mated with the  k-th  read  in
                    the  second  file.  The corresponding read names must be identical up to the tailing `/1' or
                    `/2'. For example,  such  a  pair  of  read  names  are  allowed:  `EAS1_1_5_100_200/1'  and
                    `EAS1_1_5_100_200/2'.  The  tailing  `/[12]'  is  usually  generated  by  the  GAPipeline to
                    distinguish the two ends in a pair.

                  * The output is a compressed binary file. It is affected by the endianness.

                  * The best way to run this command is to provide about 1 to 3 million  reads  as  input.  More
                    reads consume more memory.

                  * Option  <b>-n</b>  controls  the  sensitivity  of  the  alignment.  By  default, a hit with up to 2
                    mismatches can be always found. Higher <b>-n</b> finds more hits and also improves the accuracy  of
                    mapping qualities. However, this is done at the cost of speed.

                  * Alignments  with  many  high-quality  mismatches  should be discarded as false alignments or
                    possible contaminations. This behaviour is controlled by option <b>-e</b>. The <b>-e</b> threshold is only
                    calculated approximately because base qualities are divided by 10 at a certain stage of  the
                    alignment. The <b>-Q</b> option in the <b>assemble</b> command precisely set the threshold.

                  * A pair of reads are said to be correctly paired if and only if the orientation is <u>FR</u> and the
                    outer distance of the pair is no larger than <u>maxins</u>. There is no limit on the minimum insert
                    size.  This  setting  is  determined  by  the  paired  end  alignment algorithm used in Maq.
                    Requiring a minimum insert size will lead to some wrong alignments with highly overestimated
                    mapping qualities.

                  * Currently, read pairs from Illumina/Solexa long-insert library have RF read orientation. The
                    maximum insert size is set by option <b>-A</b>. However, long-insert library is also mixed  with  a
                    small fraction of short-insert read pairs. <b>-a</b> should also be set correctly.

                  * Sometimes  5'-end  or  even  the  entire  3'-adapter sequence may be sequenced. Providing <b>-d</b>
                    renders Maq to eliminate the adapter contaminations.

                  * Given 2 million reads as input, <b>maq</b> usually takes 800MB memory.

       <b>mapmerge</b>   <b>maq</b> <b>mapmerge</b> <u>out.aln.map</u> <u>in.aln1.map</u> <u>in.aln2.map</u> [...]

                  Merge a batch of read alignments together.

                  <b>NOTE:</b>

                  * In theory, this command can merge unlimited number of alignments. However, as mapmerge  will
                    be  reading  all  the inputs at the same time, it may hit the limit of the maximum number of
                    opening files set by the OS. At present, this has to be manually solved by endusers.

                  * Command <b>mapmerge</b> can be used to merge alignment files with different read lengths.  All  the
                    subsequent analyses do not assume fixed length any more.

       <b>rmdup</b>      <b>maq</b> <b>rmdup</b> <u>out.rmdup.map</u> <u>in.ori.map</u>

                  Remove  pairs  with  identical  outer  coordinates.  In  principle, pairs with identical outer
                  coordinates should happen rarely. However, due to the  amplification  in  sample  preparation,
                  this  occurs  much  more  frequently  than  by  chance.  Practical analyses show that removing
                  duplicates helps to improve the overall accuracy of SNP calling.

       <b>assemble</b>   <b>maq</b> <b>assemble</b> [<b>-sp</b>] [<b>-m</b> <u>maxmis</u>] [<b>-Q</b> <u>maxerr</u>] [<b>-r</b> <u>hetrate</u>] [<b>-t</b> <u>coef</u>] [<b>-q</b> <u>minQ</u>] [<b>-N</b> <u>nHap</u>]  <u>out.cns</u>
                  <u>in.ref.bfa</u> <u>in.aln.map</u> 2&gt; <u>out.cns.log</u>

                  Call the consensus sequences from read mapping.

                  <b>OPTIONS:</b>

                  <b>-t</b> <u>FLOAT</u> Error dependency coefficient [0.93]

                  <b>-r</b> <u>FLOAT</u> Fraction of heterozygotes among all sites [0.001]

                  <b>-s</b>       Take  single  end  mapping quality as the final mapping quality; otherwise paired end
                           mapping quality will be used

                  <b>-p</b>       Discard paired end reads that are not mapped in correct pairs

                  <b>-m</b> <u>INT</u>   Maximum number of mismatches allowed for a read to be used in consensus calling [7]

                  <b>-Q</b> <u>INT</u>   Maximum allowed sum of quality values of mismatched bases [60]

                  <b>-q</b> <u>INT</u>   Minimum mapping quality allowed for a read to be used in consensus calling [0]

                  <b>-N</b> <u>INT</u>   Number of haplotypes in the pool (&gt;=2) [2]

                  <b>NOTE:</b>

                  * Option <b>-Q</b> sets a limit on the maximum sum of mismatching base  qualities.  Reads  containing
                    many high-quality mismatches should be discarded.

                  * Option  <b>-N</b>  sets  the  number  of  haplotypes  in a pool. It is designed for resequencing of
                    samples by pooling multiple strains/individuals together. For diploid  genome  resequencing,
                    this option equals 2.

       <b>glfgen</b>     <b>maq</b>  <b>glfgen</b>  [<b>-sp</b>]  [<b>-m</b> <u>maxmis</u>] [<b>-Q</b> <u>maxerr</u>] [<b>-r</b> <u>hetrate</u>] [<b>-t</b> <u>coef</u>] [<b>-q</b> <u>minQ</u>] [<b>-N</b> <u>nHap</u>] <u>out.cns</u>
                  <u>in.ref.bfa</u> <u>in.aln.map</u> 2&gt; <u>out.cns.log</u>

                  Calculate log-likelihood for all genotypes and store the results  in  GLF  format  (Genotyping
                  Likelihood  Format). Please check MAQ website for detailed descriptions of the file format and
                  the related utilities.

       <b>indelpe</b>    <b>maq</b> <b>indelpe</b> <u>in.ref.bfa</u> <u>in.aln.map</u> &gt; <u>out.indelpe</u>

                  Call consistent indels from paired end reads. The output  is  TAB  delimited  with  each  line
                  consisting of chromosome, start position, type of the indel, number of reads across the indel,
                  size  of  the indel and inserted/deleted nucleotides (separated by colon), number of indels on
                  the reverse strand, number of indels on the forward strand, 5' sequence ahead of the indel, 3'
                  sequence following the indel, number of reads aligned  without  indels  and  three  additional
                  columns for filters.

                  At  the  3rd  column, type of the indel, a star indicates the indel is confirmed by reads from
                  both strands, a plus means the indel is hit by at least two reads but from the same strand,  a
                  minus  shows  the  indel  is only found on one read, and a dot means the indel is too close to
                  another indel and is filtered out.

                  Users are recommended to run through `maq.pl indelpe' to correct the number  of  reads  mapped
                  without indels. For more details, see the `maq.pl indelpe' section.

       <b>indelsoa</b>   <b>maq</b> <b>indelsoa</b> <u>in.ref.bfa</u> <u>in.aln.map</u> &gt; <u>out.indelsoa</u>

                  Call  potential homozygous indels and break points by detecting the abnormal alignment pattern
                  around indels and break points. The output is also TAB delimited with each line consisting  of
                  chromosome,  approximate  coordinate,  length  of  the abnormal region, number of reads mapped
                  across the position, number of reads on the left-hand side of the position and number of reads
                  on the right-hand side. The last column can be ignored.

                  The output contains many false positives. A recommended filter could be:

                    awk '$5+$6-$4 &gt;= 3 &amp;&amp; $4 &lt;= 1' in.indelsoa

                  Note that this command does not aim to be an accurate indel  detector,  but  mainly  helps  to
                  avoid some false positives in substitution calling. In addition, it only works well given deep
                  depth (~40X for example); otherwise the false negative rate would be very high.

       <b>Format</b> <b>Converting</b>

       <b>sol2sanger</b> <b>maq</b> <b>sol2sanger</b> <u>in.sol.fastq</u> <u>out.sanger.fastq</u>

                  Convert Solexa FASTQ to standard/Sanger FASTQ format.

       <b>bfq2fastq</b>  <b>maq</b> <b>bfq2fastq</b> <u>in.read.bfq</u> <u>out.read.fastq</u>

                  Convert Maq's BFQ format to standard FASTQ format.

       <b>mapass2maq</b> <b>maq</b> <b>mapass2maq</b> <u>in.mapass2.map</u> <u>out.maq.map</u>

                  Convert  obsolete  mapass2's  map  format to Maq's map format. The old format does not contain
                  read names.

       <b>Information</b> <b>Extracting</b>

       <b>mapview</b>    <b>maq</b> <b>mapview</b> [<b>-bN</b>] <u>in.aln.map</u> &gt; <u>out.aln.txt</u>

                  Display the read alignment  in  plain  text.  For  reads  aligned  before  the  Smith-Waterman
                  alignment, each line consists of read name, chromosome, position, strand, insert size from the
                  outer  coorniates  of  a  pair,  paired  flag,  mapping  quality,  single-end mapping quality,
                  alternative mapping quality, number of mismatches  of  the  best  hit,  sum  of  qualities  of
                  mismatched  bases  of  the  best  hit,  number of 0-mismatch hits of the first 24bp, number of
                  1-mismatch hits of the first 24bp on the reference, length of the read, read sequence and  its
                  quality.   Alternative  mapping  quality always equals to mapping quality if the reads are not
                  paired. If reads are paired, it equals to the smaller mapping quality of the  two  ends.  This
                  alternative mapping quality is actually the mapping quality of an abnormal pair.

                  The  fifth  column,  paired  flag, is a bitwise flag. Its lower 4 bits give the orientation: 1
                  stands for FF, 2 for FR, 4 for RF, and 8 for RR, where FR means that  the  read  with  smaller
                  coordinate is on the forward strand, and its mate is on the reverse strand. Only FR is allowed
                  for  a  correct pair. The higher bits of this flag give further information. If the pair meets
                  the paired end requirement, 16 will  be  set.  If  the  two  reads  are  mapped  to  different
                  chromosomes,  32 will be set. If one of the two reads cannot be mapped at all, 64 will be set.
                  The flag for a correct pair always equals to 18.

                  For reads aligned by the Smith-Waterman alignment afterwards, the flag is always 130.  A  line
                  consists  of read name, chromosome, position, strand, insert size, flag (always 130), position
                  of the indel on the read (0 if no indel), length of the indels (positive  for  insertions  and
                  negative  for  deletions),  mapping quality of its mate, number of mismatches of the best hit,
                  sum of qualities of mismatched bases of the best hit, two zeros,  length  of  the  read,  read
                  sequence and its quality. The mate of a 130-flagged read always gets a flag 18.

                  Flag  192  indicates that the read is not mapped but its mate is mapped. For such a read pair,
                  one read has flag 64 and the other has 192.

                  <b>OPTIONS:</b>

                  <b>-b</b>       do not display the read sequence and the quality

                  <b>-N</b>       display the positions where mismatches occur. This flag only works with a  .map  file
                           generated by `maq map -N'.

       <b>mapcheck</b>   <b>maq</b> <b>mapcheck</b> [<b>-s</b>] [<b>-m</b> <u>maxmis</u>] [<b>-q</b> <u>minQ</u>] <u>in.ref.bfa</u> <u>in.aln.map</u> &gt; <u>out.mapcheck</u>

                  Read quality check. The mapcheck first reports the composition and the depth of the reference.
                  After  that there is a form. The first column indicates the position on a read. Following four
                  columns which show the nucleotide composition, substitution rates between  the  reference  and
                  reads  will  be  given. These rates and the numbers in the following columns are scaled to 999
                  and rounded to nearest integer. The next group  of  columns  show  the  distribution  of  base
                  qualities  along  the  reads  at  a  quality interval of 10. A decay in quality can usually be
                  observed, which means bases at the end of read are less accurate. The last  group  of  columns
                  present  the fraction of substitutions for read bases at a quality interval. This measures the
                  accuracy of base quality estimation. Idealy, we expect to see 1 in the 3? column, 10 in the 2?
                  column and 100 in the 1? column.

                  <b>OPTIONS:</b>

                  <b>-s</b>       Take single end mapping quality as the final mapping quality

                  <b>-m</b> <u>INT</u>   Maximum number of mismatahces allowed for a read to be counted [4]

                  <b>-q</b> <u>INT</u>   Minimum mapping quality allowed for a read to be counted [30]

       <b>pileup</b>     <b>maq</b> <b>pileup</b> [<b>-spvP</b>] [<b>-m</b> <u>maxmis</u>] [<b>-Q</b> <u>maxerr</u>] [<b>-q</b> <u>minQ</u>] [<b>-l</b>  <u>sitefile</u>]  <u>in.ref.bfa</u>  <u>in.aln.map</u>  &gt;
                  <u>out.pileup</u>

                  Display  the  alignment in a `pileup' text format. Each line consists of chromosome, position,
                  reference base, depth and the bases on reads that cover this position. If <b>-v</b> is added  on  the
                  command  line, base qualities and mapping qualities will be presented in the sixth and seventh
                  columns in order.

                  The fifth column always starts with `@'. In this column, read bases identical to the reference
                  are showed in comma `,' or dot `.', and read bases different from the reference in letters.  A
                  comma or a upper case indicates that the base comes from a read aligned on the forward strand,
                  while a dot or a lower case on the reverse strand.

                  This command is for users who want to develop their own SNP callers.

                  <b>OPTIONS:</b>

                  <b>-s</b>       Take single end mapping quality as the final mapping quality

                  <b>-p</b>       Discard paired end reads that are not mapped as correct pairs

                  <b>-v</b>       Output verbose information including base qualities and mapping qualities

                  <b>-m</b> <u>INT</u>   Maximum number of mismatches allowed for a read to be used [7]

                  <b>-Q</b> <u>INT</u>   Maximum allowed number of quality values of mismatches [60]

                  <b>-q</b> <u>INT</u>   Minimum mapping quality allowed for a read to be used [0]

                  <b>-l</b> <u>FILE</u>  File containing the sites at which pileup will be printed out. In this file the first
                           column  gives  the  names of the reference and the second the coordinates. Additional
                           columns will be ignored. [null]

                  <b>-P</b>       also output the base position on the read

       <b>cns2fq</b>     <b>maq</b> <b>cns2fq</b> [<b>-Q</b> <u>minMapQ</u>] [<b>-n</b> <u>minNeiQ</u>] [<b>-d</b> <u>minDepth</u>] [<b>-D</b> <u>maxDepth</u>] <u>in.cns</u> &gt; <u>out.cns.fastq</u>

                  Extract the consensus sequences in FASTQ format. In the sequence lines, bases  in  lower  case
                  are  essentially  repeats  or  do  not  have sufficient coverage; bases in upper case indicate
                  regions where SNPs can be reliably called. In the quality lines, ASCII of a character minus 33
                  gives the PHRED quality.

                  <b>OPTIONS:</b>

                  <b>-Q</b> <u>INT</u>   Minimum mapping quality [40]

                  <b>-d</b> <u>INT</u>   Minimum read depth [3]

                  <b>-n</b> <u>INT</u>   Minimum neighbouring quality [20]

                  <b>-D</b> <u>INT</u>   Maximum read dpeth. &gt;=255 for unlimited. [255]

       <b>cns2snp</b>    <b>maq</b> <b>cns2snp</b> <u>in.cns</u> &gt; <u>out.snp</u>

                  Extract SNP sites. Each line consists of chromosome, position, reference base, consensus base,
                  Phred-like consensus quality, read depth, the average number of hits of  reads  covering  this
                  position,  the  highest  mapping  quality  of  the  reads  covering  the position, the minimum
                  consensus quality in the 3bp flanking regions at each side of the site  (6bp  in  total),  the
                  second  best  call,  log  likelihood ratio of the second best and the third best call, and the
                  third best call.

                  The 5th column is the key criterion when you judge the reliability of a SNP. However, as  this
                  quality  is only calculated assuming site independency, you should also consider other columns
                  to get more accurate SNP calls. Script command `<b>maq.pl</b> <b>SNPfilter</b>' is designed  for  this  (see
                  below).

                  The  7th column implies whether the site falls in a repetitive region. If no read covering the
                  site can be mapped with high mapping quality, the flanking region is possibly repetitive or in
                  the lack of good reads. A SNP at such site is usually not reliable.

                  The 8th column roughly gives the copy number of the flanking region in the  reference  genome.
                  In  most cases, this number approaches 1.00, which means the region is about unique. Sometimes
                  you may see non-zero read depth but 0.00 at the 7th column. This indicates that all the  reads
                  covering  the  position  have  at  least  two mismatches. Maq only counts the number of 0- and
                  1-mismatch hits to the reference. This is due to a complex technical issue.

                  The 9th column gives the neighbouring quality. Filtering on this column is  also  required  to
                  get  reliable  SNPs.  This  idea  is inspired by NQS, although NQS is initially designed for a
                  single read instead of a consensus.

       <b>cns2view</b>   <b>maq</b> <b>cns2view</b> <u>in.cns</u> &gt; <u>out.view</u>

                  Show detailed information at all sites. The output format is identical to <b>cns2snp</b> report.

       <b>cns2ref</b>    <b>maq</b> <b>cns2ref</b> <u>in.cns</u> &gt; <u>out.ref.fasta</u>

                  Extract the reference sequence.

       <b>cns2win</b>    <b>maq</b> <b>cns2win</b> [<b>-w</b> <u>winsize</u>] [<b>-c</b> <u>chr</u>] [<b>-b</b> <u>begin</u>] [<b>-e</b> <u>end</u>] [<b>-q</b> <u>minQ</u>] <u>in.cns</u> &gt; <u>out.win</u>

                  Extract information averaged in a tilling window. The output is TAB delimited, which  consists
                  of  reference  name, coordinate divided by 1,000,000, SNP rate, het rate, raw read depth, read
                  depth in approximately unique regions, the average number of hits of reads in the  window  and
                  percent GC.

                  <b>OPTIONS:</b>

                  <b>-w</b> <u>INT</u>   Size of a window [1000]

                  <b>-c</b> <u>STR</u>   Destinated reference sequence; otherwise all references will be used [null]

                  <b>-b</b> <u>INT</u>   Start position, 0 for no constraint [0]

                  <b>-e</b> <u>INT</u>   End position, 0 for no constraint [0]

                  <b>-q</b> <u>INT</u>   Minimum consensus quality of the sites to be used [0]

       <b>Simulation</b> <b>Related</b>

       <b>fakemut</b>    <b>maq</b> <b>fakemut</b> [<b>-r</b> <u>mutrate</u>] [<b>-R</b> <u>indelfrac</u>] <u>in.ref.fasta</u> &gt; <u>out.fakeref.fasta</u> 2&gt; <u>out.fake.snp</u>

                  Randomly  introduce  substitutions and indels to the reference. Substitutions and sinlge base-
                  pair indels can be added.

                  <b>OPTIONS:</b>

                  <b>-r</b> <u>FLOAT</u>  Mutation rate [0.001]

                  <b>-R</b> <u>FLOAT</u>  Fraction of mutations to be indels [0.1]

       <b>simutrain</b>  <b>maq</b> <b>simutrain</b> <u>out.simupars.dat</u> <u>in.read.fastq</u>

                  Estimate/train parameters for read simulation.

       <b>simulate</b>   <b>maq</b> <b>simulate</b> [<b>-d</b> <u>insize</u>] [<b>-s</b> <u>stdev</u>] [<b>-N</b> <u>nReads</u>] [<b>-1</b> <u>readLen1</u>] [<b>-2</b> <u>readLen2</u>] [<b>-r</b>  <u>mutRate</u>]  [<b>-R</b>
                  <u>indelFrac</u>] [<b>-h</b>] <u>out.read1.fastq</u> <u>out.read2.fastq</u> <u>in.ref.fasta</u> <u>in.simupars.dat</u>

                  Simulate  paired  end  reads.  File  <u>in.simupars.dat</u>  determines  the read lengths and quality
                  distribution. It is generated from <b>simutrain</b>, or can be downloaded from Maq  website.  In  the
                  output  read  files,  a  read  name  consists  of  the  reference  sequence name and the outer
                  coordinates of the pair of simulated reads. By default, <b>simulate</b> assumes  reads  come  from  a
                  diploid  sequence  which is generated by adding two different sets of mutations, including one
                  base-pair indels, to <u>in.ref.fasta</u>.

                  <b>OPTIONS:</b>

                  <b>-d</b> <u>INT</u>   mean of the outer distance of insert sizes [170]

                  <b>-s</b> <u>INT</u>   standard deviation of insert sizes [20]

                  <b>-N</b> <u>INT</u>   number of pairs of reads to be generated [1000000]

                  <b>-1</b> <u>INT</u>   length of the first read [set by <u>in.simupars.dat</u>]

                  <b>-2</b> <u>INT</u>   length of the second read [set by <u>in.simupars.dat</u>]

                  <b>-r</b> <u>FLOAT</u> mutation rate [0.001]

                  <b>-R</b> <u>FLOAT</u> fraction of 1bp indels [0.1]

                  <b>-h</b>       add all mutations to <u>in.ref.fasta</u> and generate reads from the single mutated sequence
                           (haploid mode)

                  <b>NOTE:</b>

                  * Reads generated from this command are independent, which deviates from  the  truth.  Whereas
                    alignment evaluation is less affected by this, evaluation on SNP calling should be performed
                    with caution. Error dependency may be one of the major causes of wrong SNP calls.

       <b>simustat</b>   <b>maq</b> <b>simustat</b> <u>in.simu-aln.map</u> &gt; <u>out.simustat</u>

                  Evaluate mapping qualities from simulated reads.

       <b>SOLiD</b> <b>Related</b>

       <b>fasta2csfa</b> <b>maq</b> <b>fasta2csfa</b> <u>in.nucl-ref.fasta</u> &gt; <u>out.colour-ref.fasta</u>

                  Convert nucleotide FASTA to colour-coded FASTA. Flag <b>-c</b> should be then applied to <b>map</b> command.
                  In  the  output,  letter  `A'  stands  for  color  0, `C' for 1, `G' for 2 and `T' for 3. Each
                  sequence in the output is 1bp shorter than the input.

       <b>csmap2nt</b>   <b>maq</b> <b>csmap2nt</b> <u>out.nt.map</u> <u>in.ref.nt.bfa</u> <u>in.cs.map</u>

                  Convert color alignment to nucleotide alignment. The input  <u>in.ref.nt.bfa</u>  is  the  nucleotide
                  binary  FASTA  reference  file.  It  must correspond to the original file from which the color
                  reference is converted. Nucleotide consensus can be called from the resultant alignment.

       <b>Miscellaneous/Advanced</b> <b>Commands</b>

       <b>submap</b>     <b>maq</b> <b>submap</b> [<b>-q</b> <u>minMapQ</u>] [<b>-Q</b> <u>maxSumErr</u>] [<b>-m</b> <u>maxMM</u>] [<b>-p</b>] <u>out.map</u> <u>in.map</u>

                  Filter bad alignments in <u>in.map</u>. Command-line options are described in the `<b>assemble</b>' command.

       <b>eland2maq</b>  <b>maq</b> <b>eland2maq</b> [<b>-q</b> <u>defqual</u>] <u>out.map</u> <u>in.list</u> <u>in.eland</u>

                  Convert eland alignment to maq's .map format. File <u>in.list</u> consists of the sequence names that
                  appear at the seventh column of the eland alignment file <u>in.eland</u> and the name you  expect  to
                  see in maq alignment. The following is an example:

                    cX.fa chrX
                    c1.fa chr1
                    c2.fa chr2

                  If  you  are  aligning  reads  in several batches using eland, it is important to use the same
                  <u>in.list</u> for the conversion. In addition, maq will load all the alignments and sort them in the
                  memory. If you have concatenate several eland outputs into one huge file, you should  separate
                  it into smaller files to prevent maq from eating all your machine memory.

                  This  command  actually  aims to show Eland alignment in Maqview. As no quality information is
                  available, the resultant maq alignment file should not be used to call consensus genotypes.

       <b>export2maq</b> <b>maq</b> <b>export2maq</b> [<b>-1</b> <u>read1len</u>] [<b>-2</b> <u>read2len</u>] [<b>-a</b> <u>maxdist</u>] [<b>-n</b>] <u>out.map</u> <u>in.list</u> <u>in.export</u>

                  Convert Illumina's Export format to Maq's <u>.map</u> format. Export format is a new alignment format
                  since SolexaPipeline-0.3.0 which also calculates mapping qualities  like  maq.  The  resultant
                  file can be used to call consensus genotypes as most of necessary information is available for
                  maq to do this accurately.

                  <b>OPTIONS:</b>

                  <b>-1</b> <u>INT</u>   Length of the first read [0]

                  <b>-2</b> <u>INT</u>   Length of the second read [0]

                  <b>-a</b> <u>INT</u>   Maximum outer distance for a correct read pair [250]

                  <b>-n</b>       Retain filtered reads

</pre><h4><b>MAQ-PERL</b> <b>COMMANDS</b></h4><pre>
       <b>demo</b>       <b>maq.pl</b> <b>demo</b> [<b>-h</b>] [<b>-s</b>] [<b>-N</b> <u>nPairs</u>] [<b>-d</b> <u>outDir</u>] <u>in.fasta</u> <u>in.simudat</u>

                  Demonstrate  the use of <b>maq</b> and its companion scripts. This command will simulate reads from a
                  FASTA file <u>in.fasta</u>. The sequence length and qualities are determined by <u>in.simudat</u>  which  is
                  generated  from  <b>maq</b> <b>simutrain</b> or can be downloaded from Maq website. The simulated reads will
                  then be mapped with <b>maq.pl</b> <b>easyrun</b>. The alignment accuracy is evaluated by <b>maq</b>  <b>simustat</b>,  the
                  consensus accuracy by <b>maq</b> <b>simucns</b>, and the SNP accuracy by <b>maq_eval.pl</b>.

                  By  default,  paired end reads will be simulated and a diploid sequence will be generated from
                  the input by adding mutations to either haploid type. The insert size and  mutation  rate  are
                  controlled by <b>maq</b> <b>simulate</b>.

                  <b>OPTIONS:</b>

                  <b>-h</b>       simulate a haploid sequence instead of a diploid sequence

                  <b>-s</b>       use single-end mode to align reads instead of paired-end mode

                  <b>-N</b> <u>INT</u>   number of pairs of reads to be simulated [1000000]

                  <b>-d</b> <u>DIR</u>   output directory [maqdemo]

                  <b>NOTE:</b>

                  * The output files from <b>maq_eval.pl</b> have not been documented, but you may make a good guess at
                    some of these files.

                  * This command just demonstrates the use of the maq suite. The accuracy on real data is almost
                    always lower than what you see from pure simulation.

       <b>easyrun</b>    <b>maq.pl</b>  <b>easyrun</b>  [<b>-1</b> <u>read1Len</u>] [<b>-d</b> <u>out.dir</u>] [<b>-n</b> <u>nReads</u>] [<b>-A</b> <u>3adapter</u>] [<b>-e</b> <u>minDep</u>] [<b>-q</b> <u>minCnsQ</u>]
                  [<b>-p</b>] [<b>-2</b> <u>read2Len</u>] [<b>-a</b> <u>maxIns</u>] [<b>-S</b>] [<b>-N</b>] <u>in.ref.fasta</u> <u>in1.fastq</u> [<u>in2.fastq</u>]

                  Analyses pipeline for small genomes. Easyrun command will run most of analyses implemented  in
                  <b>maq</b>.  By  default,  <b>easyrun</b>  assumes  all  the  input  read sequences files are single-end and
                  independent; when <b>-p</b> is specified, two read sequence files are required, one for each end.

                  Several files will be generated in <u>out.dir</u>, among  which  the  following  files  are  the  key
                  output:

                  <u>cns.final.snp</u>   final SNP calls with low quality ones filtered out

                  <u>cns.fq</u>          consensus sequences and qualities in the FASTQ format

                  <b>OPTIONS:</b>

                  <b>-d</b> <u>DIR</u>   output directory [easyrun]

                  <b>-n</b> <u>INT</u>   number of reads/pairs in one batch of alignment [2000000]

                  <b>-S</b>       apply split-read analysis of short indels (maybe very slow)

                  <b>-N</b> <u>INT</u>   number of haplotypes/strains in the pool (&gt;=2) [2]

                  <b>-A</b> <u>FILE</u>  file for 3'-adapter. The file should contain a single line of sequence [null]

                  <b>-1</b> <u>INT</u>   length of the first read, 0 for auto [0]

                  <b>-e</b> <u>INT</u>   minimum read depth required to call a SNP (for SNPfilter) [3]

                  <b>-q</b> <u>INT</u>   minimum consensus quality for SNPs in <u>cns.final.snp</u> [30]

                  <b>-p</b>       switch to paired end alignment mode

                  <b>-2</b> <u>INT</u>   length of the second read when <b>-p</b> is applied [0]

                  <b>-a</b> <u>INT</u>   maximum insert size when <b>-p</b> is applied [250]

                  <b>NOTES:</b>

                  * For SNP calling on pooled samples, users should set correct `<b>-N</b>' as well as `<b>-E</b> 0'.

                  * The input file can be maq's binary format. <b>maq.pl</b> will automatically detect the file format.

       <b>SNPfilter</b>  <b>maq.pl</b>  <b>SNPfilter</b>  [<b>-d</b>  <u>minDep</u>]  [<b>-D</b>  <u>maxDep</u>]  [<b>-Q</b> <u>maxMapQ</u>] [<b>-q</b> <u>minCnsQ</u>] [<b>-w</b> <u>indelWinSize</u>] [<b>-n</b>
                  <u>minNeiQ</u>] [<b>-F</b> <u>in.indelpe</u>] [<b>-f</b> <u>in.indelsoa</u>] [<b>-s</b> <u>minScore</u>] [<b>-m</b> <u>maxAcross</u>] [<b>-a</b>] [<b>-N</b> <u>maxWinSNP</u>] [<b>-W</b>
                  <u>densWinSize</u>] <u>in.cns2snp.snp</u> &gt; <u>out.filtered.snp</u>

                  Rule out SNPs that are covered by few reads (specified by <b>-d</b>), by too many reads (specified by
                  <b>-D</b>), near (specified by <b>-w</b>) to a potential indel,  falling  in  a  possible  repetitve  region
                  (characterized  by  <b>-Q</b>),  or  having  low-quality  neighbouring  bases  (specified  by <b>-n</b>). If
                  <u>maxWinSNP</u> or more SNPs appear in any <u>densWinSize</u>  window,  they  will  also  be  filtered  out
                  together.

                  <b>OPTIONS:</b>

                  <b>-d</b> <u>INT</u>    Minimum read depth required to call a SNP [3]

                  <b>-D</b> <u>INT</u>    Maximum read depth required to call a SNP (&lt;255, otherwise ignored) [256]

                  <b>-Q</b> <u>INT</u>    Required maximum mapping quality of reads covering the SNP [40]

                  <b>-q</b> <u>INT</u>    Minimum consensus quality [20]

                  <b>-n</b> <u>INT</u>    Minimum adjacent consensus quality [20]

                  <b>-w</b> <u>INT</u>    Size  of  the window around the potential indels. SNPs that are close to indels will
                            be suppressed [3]

                  <b>-F</b> <u>FILE</u>   The <b>indelpe</b> output [null]

                  <b>-f</b> <u>FILE</u>   The <b>indelsoa</b> output [null]

                  <b>-s</b> <u>INT</u>    Minimum score for a soa-indel to be considered [3]

                  <b>-m</b> <u>INT</u>    Maximum number of reads that can be mapped across a soa-indel [1]

                  <b>-a</b>        Alternative filter for single end alignment

       <b>indelpe</b>    <b>maq.pl</b> <b>indelpe</b> <u>in.indelpe</u> &gt; <u>out.indelpe</u>

                  Correct the number of reads mapped without indels for homopolymer tracts. This command  modify
                  the  4th,  10th and the last three columns of <u>in.indelpe</u> and output the result in <u>out.indelpe</u>.
                  After the correction, the following <b>awk</b> command gives putative homozygous indels:

                    awk '($3=="*"⎪⎪$3=="+") &amp;&amp; $6+$7&gt;=3 &amp;&amp; ($6+$7)/$4&gt;=0.75'

                  and the following gives heterozygotes:

                    awk '($3=="*"⎪⎪$3=="+") &amp;&amp; $6+$7&gt;=3 &amp;&amp; ($6+$7)/$4&lt;0.75'

                  Please note that this <b>indelpe</b> command just implements several heuristic  rules.  It  does  not
                  correct  for  impure  homopolymer runs or di-nucleotide/triplet repeats. Consequently, the two
                  awk commands only give approximate hom/het indels.

</pre><h4><b>EXAMPLES</b></h4><pre>
       • Easyrun script:
           maq.pl easyrun -d easyrun ref.fasta part1.fastq part2.fastq

       • Key commands behind easyrun:
           maq fasta2bfa ref.fasta ref.bfa;
           maq fastq2bfq part1.fastq part1.bfq;
           maq fastq2bfq part2.fastq part2.bfq;
           maq map part1.map ref.bfa part1.bfq;
           maq map part2.map ref.bfa part2.bfq;
           maq mapmerge aln.map part1.map part2.map;
           maq assemble cns.cns ref.bfa aln.map;

</pre><h4><b>LICENSE</b></h4><pre>
       GNU General Public License, version 3 (GPLv3)

</pre><h4><b>AVAILABILITY</b></h4><pre>
       &lt;<a href="http://maq.sourceforge.net">http://maq.sourceforge.net</a>&gt;

</pre><h4><b>AUTHOR</b></h4><pre>
       Heng Li &lt;<a href="mailto:lh3@sanger.ac.uk">lh3@sanger.ac.uk</a>&gt;

maq-0.7.1                                          2008-09-22                                             <u><a href="../man1/MAQ.1.html">MAQ</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>