<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>SOAPaligner/soap2 - Short Oligonucleotide Analysis Package aligner</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/soapaligner">soapaligner_2.20-5_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       SOAPaligner/soap2 - Short Oligonucleotide Analysis Package aligner

</pre><h4><b>SYNOPSIS</b></h4><pre>
       soap reference.index short_reads.fast[a|q] alignment.out [options]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       SOAPaligner/soap2  is  a  member  of  the SOAP (Short Oligonucleotide Analysis Package). It is an updated
       version of SOAP software for short oligonucleotide alignment. The new program features in super fast  and
       accurate alignment for huge amounts of short reads generated by Illumina/Solexa Genome Analyzer. Compared
       to  soap  v1, it is one order of magnitude faster. It require only 2 minutes aligning one million single-
       end reads onto the human reference genome. Another remarkable improvement of SOAPaligner is that  it  now
       supports a wide range of the read length.

       SOAPaligner  benefitted  in  time  and  space efficiency by a revolution in the basic data structures and
       algorithms used.The core algorithms and the indexing data structures  (2way-BWT)  are  developed  by  the
       algorithms  research  group of the Department of Computer Science, the University of Hong Kong (T.W. Lam,
       Alan Tam, Simon Wong, Edward Wu and S.M. Yiu).

</pre><h4><b>COMMAND</b> <b>AND</b> <b>OPTIONS</b></h4><pre>
       <b>soap</b>  -D   &lt;in.fasta.index&gt;   -a   &lt;query.file.a&gt;   [-b   &lt;query.file.b&gt;]   -o   &lt;alignment.output&gt;   [-2
       &lt;unpaired.output&gt;] [options]

       <b>OPTIONS:</b>

              <b>-D</b> <b>STR</b> Prefix name for reference index [*.index]. See <b>APPENDIX</b> How to build the reference index

              <b>-a</b> <b>STR</b> Query file, for SE reads alignment or one end of PE reads

              <b>-b</b> <b>STR</b> Query b file, one end of PE reads

              <b>-o</b> <b>STR</b> Output file for alignment results

              <b>-2</b> <b>STR</b> Output file contains mapped but unpaired reads when do PE alignment

              <b>-u</b> <b>STR</b> Output file for unmapped reads, [none]

              <b>-m</b> <b>INT</b> Minimal insert size INT allowed for PE, [400]

              <b>-x</b> <b>INT</b> Maximal insert size INT allowed for PE, [600]

              <b>-n</b> <b>INT</b> Filter low quality reads containing more INT bp Ns, [5]

              <b>-t</b>     Output reads id instead reads name, [none]

              <b>-r</b> <b>INT</b> How to report repeat hits, 0=none; 1=random one; 2=all, [1]

              <b>-R</b>     RF alignment for long insert size(&gt;= 2k bps) PE data, [none] FR alignment

              <b>-l</b> <b>INT</b> For  long  reads with high error rate at 3'-end, those can't align whole length, then first
                     align 5' INT bp subsequence as a seed, [256] use whole length of the read

              <b>-s</b> <b>INT</b> minimal alignment length (for soft clip)

              <b>-v</b> <b>INT</b> Totally allowed mismatches in one read, when use subsequence as a seed, [5]

              <b>-g</b> <b>INT</b> Allow gap size in one read, [0]

              <b>-M</b> <b>INT</b> Match mode for each read or the seed part of read, which  shouldn't  contain  more  than  2
                     mismatches, [4]

                     0: exact match only

                     1: 1 mismatch match only

                     2: 2 mismatch match only

                     4: find the best hits
              <b>-p</b> <b>INT</b> Multithreads, n threads, [1]

</pre><h4><b>OUTPUT</b> <b>FORMAT</b></h4><pre>
       SOAP2 output format contains following column information:

       1. reads name / reads ID (if -t is available)

       2. reads sequence (if read align to reverse strand, here is the reverse sequence of original read)

       3. quality sequence (if input is fasta reads, the column will be all 'h', and the sequence is backward if
       reads mapping reverse )

       4.

</pre><h4><b>APPENDIX</b></h4><pre>
       Before use soap2 to do alignment, the reference index must be generated by 2bwt-builder.

              <b>2bwt-builder</b> &lt;reference.fasta&gt;

              <b>NOTE:</b>  1.  the  reference  input should only be FASTA format; 2. the program wil auto generate the
              index files in the directory where the fasta file is located, so confirm the permission at first.

</pre><h4><b>ENVIRONMENT</b></h4><pre>
       The datastructure is imcompatible with 32bit, so it can't be migrated on any  32bit  platforms.   Due  to
       using  the  MMX  instruction  to  opitimize  parts  of  code,  the current version can only run on <b>x86_64</b>
       <b>platform.</b>  We will provide a universal version for most of the 64bit platform later.

       <b>HARDWARE</b> <b>REQUIREMENT</b>
              1.8Gb RAM (for a genome as large as human's)

              2.at least 8Gb hard disk to store index (for a genome as large as human's)

       <b>SYSTEM</b> <b>REQUIREMENT</b>
              Linux x86_64

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Website for SOAP &lt;<a href="http://soap.genomics.org.cn">http://soap.genomics.org.cn</a>&gt;,

       Google Group for SOAP &lt;<a href="http://groups.google.com/group/bgi-soap">http://groups.google.com/group/bgi-soap</a>&gt;

       <b>Publication:</b>
              "SOAP: short oligonucleotide alignment program" (2008) BIOINFORMATICS,Vol.  24  no.5  2008,  pages
              713-714

</pre><h4><b>ATHOUR</b></h4><pre>
       <b>BGI</b>  <b>Shenzhen</b>  SOAP  team.  The  core algorithm Bidirect-BWT is wrotten by Prof. T.W. Lam and his team at
       HongKong University.

</pre><h4><b>REPORT</b> <b>BUGS</b></h4><pre>
       Report bugs to &lt;<a href="mailto:soap@genomics.org.cn">soap@genomics.org.cn</a>&gt;

</pre><h4><b>ACKNOWLEDGEMENTS</b></h4><pre>
       We appreciate Prof. T.W. Lam, Alan Tam, Simon Wong, Edward Wu and S.M. Yiu prominent  work  on  Bidirect-
       BWT.

SOAPaligner-2.1X                                   25 May 2009                              <u>SOAPaligner/<a href="../man1/soap2.1.html">soap2</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>