<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>rockhopper - system for analyzing bacterial RNA-seq data (command line tool)</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/rockhopper">rockhopper_2.0.3+dfsg2-5_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       rockhopper - system for analyzing bacterial RNA-seq data (command line tool)

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>rockhopper</b> [<u>options</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>rockhopper</b>  is  a  comprehensive and user-friendly system for computational analysis of bacterial RNA-seq
       data. As input, it takes RNA sequencing reads output by  high-throughput  sequencing  technology  (FASTQ,
       QSEQ, FASTA, SAM, or BAM files).

</pre><h4><b>REQUIRED</b> <b>ARGUMENTS</b></h4><pre>
       <u>exp1A.fastq,exp1B.fastq,exp1C.fastq</u> <u>exp2A.fastq,exp2B.fastq</u>
              a  comma  separated  list  of  sequencing  files  (in  FASTQ, QSEQ, FASTA, SAM, or BAM format) for
              replicate experiments, one list per experimental condition (mate-pair files should be delimited by
              '%')

</pre><h4><b>REFERENCE</b> <b>BASED</b> <b>ASSEMBLY</b> <b>VS.</b> <b>DE</b> <b>NONO</b> <b>ASSEMBLY</b></h4><pre>
       If the -g option is used, then rockhopper aligns reads to  one  or  more  reference  genomes,  otherwise,
       rockhopper performs de novo transcript assembly.

       <b>-g</b> <b>&lt;DIR1,DIR2&gt;</b>
              a  comma  separated list of directories, each containing a genome file (*.fna), gene file (*.ptt),
              and rna file (*.rnt)

</pre><h4><b>OPTIONAL</b> <b>ARGUMENTS</b> <b>FOR</b> <b>EITHER</b> <b>REFERENCE</b> <b>BASED</b> <b>ASSEMBLY</b> <b>OR</b> <b>DE</b> <b>NOVO</b> <b>ASSEMBLY</b></h4><pre>
       <b>-c</b> <b>&lt;boolean&gt;</b>
              reverse complement single-end reads (default is false)

       <b>-ff,</b> <b>-fr,</b> <b>-rf,</b> <b>-rr</b>
              orientation of two mate reads for paired-end read, f=forward and r=reverse_complement (default  is
              fr)

       <b>-d</b> <b>&lt;integer&gt;</b>
              maximum number of bases between mate pairs for paired-end reads (default is 500)

       <b>-a</b> <b>&lt;boolean&gt;</b>
              identify 1 alignment (true) or identify all optimal alignments (false), (default is true)

       <b>-p</b> <b>&lt;integer&gt;</b>
              number of processors (default is self-identification of processors)

       <b>-e</b> <b>&lt;boolean&gt;</b>
              compute  differential  expression  for transcripts in pairs of experimental conditions (default is
              true)

       <b>-s</b> <b>&lt;boolean&gt;</b>
              RNA-seq experiments are strand specific (true) or strand ambiguous (false), (default is true)

       <b>-L</b> <b>&lt;comma</b> <b>separated</b> <b>list&gt;</b>
              labels for each condition

       <b>-o</b> <b>&lt;DIR&gt;</b>
              directory where output files are written (default is Rockhopper_Results/)

       <b>-v</b> <b>&lt;boolean&gt;</b>
              verbose output including raw/normalized counts aligning to each gene (default is false)

       <b>-SAM</b>   output a SAM format file

       <b>-TIME</b>  output time taken to execute program

</pre><h4><b>OPTIONAL</b> <b>ARGUMENTS</b> <b>FOR</b> <b>REFERENCE</b> <b>BASED</b> <b>ASSEMBLY</b> <b>ONLY</b></h4><pre>
       <b>-m</b> <b>&lt;number&gt;</b>
              allowed mismatches as percent of read length (default is 0.15)

       <b>-l</b> <b>&lt;number&gt;</b>
              minimum seed as percent of read length (default is 0.33)

       <b>-y</b> <b>&lt;boolean&gt;</b>
              compute operons (default is true)

       <b>-t</b> <b>&lt;boolean&gt;</b>
              identify transcript boundaries including UTRs and ncRNAs (default is true)

       <b>-z</b> <b>&lt;number&gt;</b>
              minimum expression of UTRs and ncRNAs, a number in range [0.0, 1.0] (default is 0.5)

</pre><h4><b>OPTIONAL</b> <b>ARGUMENTS</b> <b>FOR</b> <b>DE</b> <b>NOVO</b> <b>ASSEMBLY</b> <b>ONLY</b></h4><pre>
       <b>-k</b> <b>&lt;integer&gt;</b>
              size of k-mer, range of values is 15 to 31 (default is 25)

       <b>-j</b> <b>&lt;integer&gt;</b>
              minimum length required to use a sequencing read after trimming/processing (default is 35)

       <b>-n</b> <b>&lt;integer&gt;</b>
              size of k-mer hashtable is ~ 2^n (default is 25). HINT: should normally be 25 or, if  more  memory
              is available, 26. WARNING: if increased above 25 then more than 1.2M of memory must be allocated

       <b>-b</b> <b>&lt;integer&gt;</b>
              minimum  number  of full length reads required to map to a de novo assembled trancript (default is
              20)

       <b>-u</b> <b>&lt;integer&gt;</b>
              minimum length of de novo assembled transcripts (default is 2*k)

       <b>-w</b> <b>&lt;integer&gt;</b>
              minimum count of k-mer to use it to seed a new de novo assembled transcript (default is 50)

       <b>-x</b> <b>&lt;integer&gt;</b>
              minimum count of k-mer to use it to extend an existing de novo assembled transcript (default is 5)

</pre><h4><b>EXAMPLES</b></h4><pre>
       reference based assembly with single-end reads
               %             rockhopper             &lt;options&gt;             -g             genome_DIR1,genome_DIR2
       aerobic_replicate1.fastq,aerobic_replicate2.fastq anaerobic_replicate1.fastq,anaerobic_replicate2.fastq

       de novo assembly with single-end reads
               %          rockhopper         &lt;options&gt;         aerobic_replicate1.fastq,aerobic_replicate2.fastq
       anaerobic_replicate1.fastq,anaerobic_replicate2.fastq

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       https://cs.wellesley.edu/~btjaden/Rockhopper/

       <b><a href="../man1/rockhoppergui.1.html">rockhoppergui</a></b>(1)

                                                  February 2022                                    <u><a href="../man1/ROCKHOPPER.1.html">ROCKHOPPER</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>