<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>berkeley-express - Streaming quantification for high-throughput sequencing</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/berkeley-express">berkeley-express_1.5.3+dfsg-3build5_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       berkeley-express - Streaming quantification for high-throughput sequencing

</pre><h4><b>DESCRIPTION</b></h4><pre>
       File Usage:  express [options] &lt;target_seqs.fa&gt; &lt;hits.(sam/bam)&gt; Piped Usage: bowtie [options] <b>-S</b> &lt;index&gt;
       &lt;reads.fq&gt; | express [options] &lt;target_seqs.fa&gt;

   <b>Required</b> <b>arguments:</b>
       &lt;target_seqs.fa&gt;
              target sequence file in fasta format

       &lt;hits.(sam/bam)&gt;
              read alignment file in SAM or BAM format

   <b>Standard</b> <b>Options:</b>
       <b>-h</b> [ <b>--help</b> ]
              produce help message

       <b>-o</b> [ <b>--output-dir</b> ] arg (=.)
              write all output files to this directory

       <b>-D</b> [ <b>--preprocess</b> ]
              run preprocess script for eXpressD

       <b>-m</b> [ <b>--frag-len-mean</b> ] arg (=200)
              prior estimate for average fragment length

       <b>-s</b> [ <b>--frag-len-stddev</b> ] arg (=80)
              prior estimate for fragment length std deviation

       <b>-H</b> [ <b>--haplotype-file</b> ] arg
              path to a file containing haplotype pairs

       <b>-B</b> [ <b>--additional-batch</b> ] arg (=0)
              number of additional batch EM rounds after initial online round

       <b>-O</b> [ <b>--additional-online</b> ] arg (=0) number of additional online EM rounds
              after initial online round

       <b>-L</b> [ <b>--max-read-len</b> ] arg (=250)
              maximum allowed length of a read

       <b>--output-align-prob</b>
              output alignments (sam/bam) with probabilistic assignments

       <b>--output-align-samp</b>
              output alignments (sam/bam) with sampled assignments

       <b>--fr-stranded</b>
              accept only forward-&gt;reverse alignments (second-stranded protocols)

       <b>--rf-stranded</b>
              accept only reverse-&gt;forward alignments (first-stranded protocols)

       <b>--f-stranded</b>
              accept only forward single-end alignments (second-stranded protocols)

       <b>--r-stranded</b>
              accept only reverse single-end alignments (first-stranded protocols)

       <b>--no-update-check</b>
              disables automatic check for update via web

       <b>--logtostderr</b>
              prints all logging messages to stderr

   <b>Advanced</b> <b>Options:</b>
       <b>-f</b> [ <b>--forget-param</b> ] arg (=0.84999999999999998)
              sets the 'forgetting factor' parameter (0.5 &lt; c &lt;= 1)

       <b>--library-size</b> arg
              specifies library size for FPKM instead of calculating from alignments

       <b>--max-indel-size</b> arg (=10)
              sets the maximum allowed indel size, affecting geometric indel prior

       <b>--calc-covar</b>
              calculate and output covariance matrix

       <b>--expr-alpha</b> arg (=0.0050000000000000001)
              sets the strength of the prior, per bp

       <b>--stop-at</b> arg (=0)
              sets the number of fragments to process, disabled with 0

       <b>--burn-out</b> arg (=5000000)
              sets number of fragments after which to stop updating auxiliary parameters

       <b>--no-bias-correct</b>
              disables bias correction

       <b>--no-error-model</b>
              disables error modelling

       <b>--aux-param-file</b> arg
              path to file containing auxiliary parameters to use instead of learning

express 1.5.1                                     February 2015                                       <u><a href="../man1/EXPRESS.1.html">EXPRESS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>