<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>maf_parse - Reads a MAF file and perform various operations on it.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/phast">phast_1.7+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       maf_parse - Reads a MAF file and perform various operations on it.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Reads  a  MAF  file  and  perform  various  operations on it.  Performs parsing operations block-by-block
       whenever possible, rather than storing entire alignment in memory.  Can extract a sub-alignment  from  an
       alignment  (by  row  or  by  column).   Can  extract features given GFF, BED, or genepred file.  Can also
       extract sub-features such as CDS1,2,3 or 4d sites.  Can perform various functions such as  gap  stripping
       or re-ordering of sequences.  Capable of reading and

              writing  in  a  few  common  formats,  but will not load input or output alignments into memory if
              output format is MAF.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Output</b> <b>format</b>

       <b>--out-format</b>, <b>-o</b> MAF|PHYLIP|FASTA|MPM|SS (Default MAF).  Output file format.  SS format is only available
              un-ordered.  Note that some options, which  involve  reversing  alignments  based  on  strand,  or
              stripping  gaps,  cannot  be  output  in MAF format and use FASTA by default.  Also note that when
              output format is not MAF, the entire output must be loaded into memory.

       <b>--pretty</b>, <b>-p</b>

              Pretty-print alignment (use '.' when character matches corresponding character in first sequence).
              Ignored if <b>--out-format</b> SS is selected.

   <b>Obtaining</b> <b>sub-alignments</b> <b>and</b> <b>re-ordering</b> <b>rows</b>

       <b>--start</b>, <b>-s</b> &lt;start_col&gt; Start index of sub-alignment (indexing starts with 1).  Coordinates are in  terms
              of  the  reference sequence unless the <b>--no-refseq</b> option is used, in which case they are in terms
              of alignment columns.  Default is 1.

       <b>--end</b>, <b>-e</b> &lt;end_col&gt; End index of sub-alignment.  Default is length of alignment.

              Coordinates defined as in <b>--start</b> option, above.

       <b>--seqs</b>, <b>-l</b> &lt;seq_list&gt;

              Comma-separated list of sequences to  include  (default)  exclude  (if  <b>--exclude</b>).   Indicate  by
              sequence number or name (numbering starts with 1 and is evaluated *after* <b>--order</b> is applied).

       <b>--exclude</b>, <b>-x</b> Exclude rather than include specified sequences.

       <b>--order</b>, <b>-O</b> &lt;name_list&gt;

              Change  order of rows in alignment to match sequence names specified in name_list.  The first name
              in the alignment becomes the reference sequence.

       <b>--no-refseq</b>, <b>-n</b> Do not assume first sequence in  MAF  is  refseq.   Instead,  use  coordinates  given  by
              absolute position in alignment (starting from 1).

   <b>Splitting</b> <b>into</b> <b>multiple</b> <b>MAFs</b> <b>by</b> <b>length</b>

       <b>--split</b>, <b>-S</b> length

              Split  MAF  into  pieces  by  length,  and puts output in outRootX.maf, where X=1,2,...,numPieces.
              outRoot can be modified with <b>--out-root</b>, and the minimum number of digits in  X  can  be  modified
              with <b>--out-root-digits</b>.  Splits between blocks, so that each output file does not exceed specified
              length.   By  default,  length  is  counted  by  distance  spanned  in alignment by refseq, unless
              <b>--no-refseq</b> is specified.

       <b>--out-root</b>, <b>-r</b> &lt;name&gt;

              Filename root for output files produced by <b>--split</b> (default "maf_parse").

       <b>--out-root-digits</b>, <b>-d</b> &lt;numdigits&gt; (for use with <b>--split</b>).  The minimum number of digits used to

              index each output file produced by split.

   <b>Extracting</b> <b>features</b> <b>from</b> <b>MAF</b>

       <b>--features</b>, <b>-g</b> &lt;fname&gt; Annotations file.  May be GFF, BED, or genepred format.

              Coordinates assumed to be in frame of  first  sequence  of  alignment  (reference  sequence).   By
              default,  outputs  subset  of  MAF  which  are  labeled in annotations file.  But can be used with
              <b>--by-category</b>, <b>--by-group</b>, and/or <b>--do-cats</b> to split MAF by annotation  type.   Or  if  used  with
              <b>--mask-features</b>,   is   only   used  to  determine  regions  to  mask.   Implies  <b>--strip-i-lines</b>,
              <b>--strip-e-lines</b>

       <b>--by-category</b>, <b>-L</b>

       (Requires <b>--features</b>).
              Split by category, as defined by annotations file and (optionally) category map (see <b>--catmap</b>).

       <b>--do-cats</b>, <b>-C</b> &lt;cat_list&gt; (For use with  <b>--by-category</b>)  Output  sub-alignments  for  only  the  specified
              categories.

       <b>--catmap</b>, <b>-c</b> &lt;fname&gt;|&lt;string&gt;

              (Optionally use with <b>--by-category</b>) Mapping of feature types to category numbers.  Can either give
              a filename or an "inline" description of a simple category map, e.g.,

       <b>--catmap</b> "NCATS = 3 ; CDS 1-3" or

       <b>--catmap</b> "NCATS = 1; UTR 1".

       <b>--by-group</b>,  <b>-P</b> &lt;tag&gt; (Requires <b>--features</b>).  Split by groups in annotation file, as defined by specified
              tag.

   <b>Masking</b> <b>by</b> <b>quality</b> <b>score</b>

       <b>--mask-bases</b>, <b>-b</b> &lt;qscore&gt; Mask all bases with quality score &lt;= n.  Note that n is in the  same  units  as
              displayed  in  the  MAF  (ranging  from  0-9), and represents min(9, floor(PHRED_score/5)).  Bases
              without any quality score will not be masked.

       <b>--masked-file</b>, <b>-m</b> &lt;filename&gt; (For use with <b>--mask-bases</b>).  Write a file containing all the regions masked
              for low quality.  The file will be  in  0-based  coordinates  relative  to  the  refseq,  with  an
              additional  column  giving  the name of the species masked.  Note that low-quality bases masked at
              alignment columns with a gap in the reference sequence may not be represented in the output  file.

       <b>--mask-features</b>  <b>-M</b>  &lt;spec&gt;  (Requires  <b>--features</b>).   Mask  all bases annotated in features in the given
              species (can be a comma-delimited list of species).  Note that

              coordinates are always in terms of refseq, even if a different species is being masked.

   <b>Other</b>

       <b>--strip-i-lines</b>, <b>-I</b>

              Remove lines in MAF starting with i.

       <b>--strip-e-lines</b>, <b>-E</b> Remove lines in MAF starting with e.

       <b>--help</b>, <b>-h</b>

              Print this help message.

maf_parse 1.4                                       May 2016                                        <u><a href="../man1/MAF_PARSE.1.html">MAF_PARSE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>