<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>rsem-control-fdr - Filter EBSeq output for statistical significance.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/rsem">rsem_1.3.3+dfsg-3build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       rsem-control-fdr - Filter EBSeq output for statistical significance.

</pre><h4><b>SYNOPSIS</b></h4><pre>
       rsem-control-fdr [options] input_file fdr_rate output_file

</pre><h4><b>ARGUMENTS</b></h4><pre>
       <b>input_file</b>
           This   should   be   the   main  result  file  generated  by  'rsem-run-ebseq',  which  contains  all
           genes/transcripts and their associated statistics.

       <b>fdr_rate</b>
           The desire false discovery rate (FDR).

       <b>output_file</b>
           This file is a subset  of  the  'input_file'.  It  only  contains  the  genes/transcripts  called  as
           differentially expressed (DE). When more than 2 conditions exist, DE is defined as not all conditions
           are  equally  expressed.   Because  statistical  significance  does  not  necessarily mean biological
           significance, users should also refer to the fold  changes  to  decide  which  genes/transcripts  are
           biologically significant. When more than two conditions exist, this file will not contain fold change
           information and users need to calculate it from 'input_file.condmeans' by themselves.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>--hard-threshold</b>
           Use  hard  threshold  method to control FDR. If this option is set, only those genes/transcripts with
           their PPDE &gt;= 1 - fdr_rate are called as DE. (Default: on)

       <b>--soft-threshold</b>
           Use soft threshold method to control FDR. If this option is set, this program will try to  report  as
           many  genes/transcripts  as  possible,  as long as their average PPDE &gt;= 1 - fdr_rate. This option is
           equivalent to use EBSeq's 'crit_fun' for FDR control. (Default: off)

       <b>-h/--help</b>
           Show help information.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This program controls the false discovery rate and reports differentially expressed genes/transcripts.

</pre><h4><b>EXAMPLES</b></h4><pre>
       We assume that we have 'GeneMat.results' as input. We want to control FDR at 0.05  using  hard  threshold
       method and name the output file as 'GeneMat.de.txt':

        rsem-control-fdr GeneMat.results 0.05 GeneMat.de.txt

perl v5.38.2                                       2024-04-14                                <u><a href="../man1/RSEM-CONTROL-FDR.1.html">RSEM-CONTROL-FDR</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>