<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>asn2gb - convert ASN.1 biological data to a GenBank-style flat format</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/ncbi-tools-bin">ncbi-tools-bin_6.1.20170106+dfsg2-5_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       asn2gb - convert ASN.1 biological data to a GenBank-style flat format

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>asn2gb</b>  [<b>-</b>]  [<b>-A</b> <u>accession</u>]  [<b>-F</b>]  [<b>-a</b> <u>asn-type</u>]  [<b>-b</b>]  [<b>-c</b>] [<b>-d</b>] [<b>-f</b> <u>format</u>] [<b>-g</b> <u>N</u>] [<b>-h</b> <u>N</u>] [<b>-i</b> <u>filename</u>]
       [<b>-j</b> <u>N</u>] [<b>-k</b> <u>N</u>] [<b>-l</b> <u>filename</u>] [<b>-m</b> <u>mode</u>] [<b>-n</b> <u>filename</u>]  [<b>-o</b> <u>filename</u>]  [<b>-p</b>]  [<b>-q</b> <u>filename</u>]  [<b>-r</b>]  [<b>-s</b> <u>style</u>]
       [<b>-t</b> <u>N</u>] [<b>-u</b> <u>N</u>] [<b>-y</b> <u>N</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>asn2gb</b> converts descriptions of biological sequences from NCBI's ASN.1 format to one of several flat-file
       formats, and is the successor to <b><a href="../man1/asn2ff.1.html">asn2ff</a></b>(1).

</pre><h4><b>OPTIONS</b></h4><pre>
       A summary of options is included below.

       <b>-</b>      Print usage message

       <b>-A</b> <u>accession</u>
              Accession  to fetch; may take the form <u>accession</u><b>,</b><u>complexity</u><b>,</b><u>flags</u> where <u>complexity</u> should normally
              be <b>0</b> and a <u>flags</u> value of <b>-1</b> enables fetching of external features (as with the legacy <b>-F</b> option)

       <b>-F</b>     Fetch remote annotations (equivalent to specifying <b>-A</b> <u>accession</u><b>,0,-1</b>)

       <b>-a</b> <u>asn-type</u>
              ASN.1 Type:
              [Single record]
              a      Any (autodetected; default)
              e      seq-Entry
              b      Bioseq
              s      bioseq-Set
              m      seq-subMit
              q      Catenated
              [Release file; components individually processed and freed]
              t      baTch bioseq-set
              u      batch seq-sUbmit

       <b>-b</b>     Input file is binary

       <b>-c</b>     Batch file is compressed

       <b>-d</b>     Seq-loc minus strand

       <b>-f</b> <u>format</u>
              Format:
              b      GenBank (default)
              bp or pb
                     GenBank and GenPept
              e      EMBL
              p      GenPept
              q      nucleotide GBSet (XML)
              r      protein GBSet (XML)
              t      Feature table only
              x      nucleotide INSDSet (XML)
              y      tiny seq (XML)
              Y      FASTA
              z      protein INSDSet (XML)

       <b>-g</b> <u>N</u>   Bit flags (all default to off):
              1      HTML
              2      XML
              4      ContigFeats
              8      ContigSrcs
              16     FarTransl

       <b>-h</b> <u>N</u>   Lock/Lookup Flags (all default to off):
              8      LockProd
              16     LookupComp
              64     LookupProd

       <b>-i</b> <u>filename</u>
              Input file name (default = stdin)

       <b>-j</b> <u>N</u>   Start location (default is 0, beginning of sequence)

       <b>-k</b> <u>N</u>   End location (default is 0, end of sequence)

       <b>-l</b> <u>filename</u>
              Log file

       <b>-m</b> <u>mode</u>
              Mode:
              r      Release
              e      Entrez
              s      Sequin (default)
              d      Dump

       <b>-n</b> <u>filename</u>
              Asn2Flat Executable (default = asn2flat)

       <b>-o</b> <u>filename</u>
              Output file name (default = stdout)

       <b>-p</b>     Propagate top descriptors

       <b>-q</b> <u>filename</u>
              Ffdiff Executable (default = /netopt/genbank/subtool/bin/ffdiff)

       <b>-r</b>     Enable remote fetching

       <b>-s</b> <u>style</u>
              Style:
              n      Normal (default)
              s      Segment
              m      Master
              c      Contig

       <b>-t</b> <u>N</u>   Batch:
              1      Report
              2      Sequin/Release
              3      asn2gb SSEC/nocleanup
              4      asn2flat BSEC/nocleanup
              5      asn2gb/asn2flat
              6      asn2gb NEW dbxref/OLD dbxref
              7      oldasn2gb/newasn2gb
       <b>-u</b> <u>N</u> Custom flags (all default to off):
              4      Hide features
              1792   Hide references
              8192   Hide sources
              262144 Hide translations

       <b>-y</b> <u>N</u>   Feature itemID

</pre><h4><b>AUTHOR</b></h4><pre>
       The National Center for Biotechnology Information.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/asn2all.1.html">asn2all</a></b>(1),    <b><a href="../man1/asn2asn.1.html">asn2asn</a></b>(1),    <b><a href="../man1/asn2ff.1.html">asn2ff</a></b>(1),    <b><a href="../man1/asn2fsa.1.html">asn2fsa</a></b>(1),    <b><a href="../man1/asn2xml.1.html">asn2xml</a></b>(1),    <b><a href="../man1/asndhuff.1.html">asndhuff</a></b>(1),     <b><a href="../man1/insdseqget.1.html">insdseqget</a></b>(1),
       /usr/share/doc/libncbi6-dev/asn2gb.txt.gz.

NCBI                                               2016-09-01                                          <u><a href="../man1/ASN2GB.1.html">ASN2GB</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>