<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>cd-hit-est-2d - run CD-HIT algorithm on RNA/DNA sequences in db1 or db2 format</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/cd-hit">cd-hit_4.8.1-4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       cd-hit-est-2d - run CD-HIT algorithm on RNA/DNA sequences in db1 or db2 format

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>cd-hit-est-2d</b> [<u>Options</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
              ====== CD-HIT version 4.8.1 (built on Aug 20 2021) ======

       Options

       <b>-i</b>     input filename for db1 in fasta format, required, can be in .gz format

       <b>-i2</b>    input filename for db2 in fasta format, required, can be in .gz format

       <b>-j</b>, <b>-j2</b>
              input filename in fasta/fastq format for R2 reads if input are paired end (PE) files

       <b>-i</b> db1-R1.fq <b>-j</b> db1-R2.fq <b>-i2</b> db2-R1.fq <b>-j2</b> db2-R2.fq <b>-o</b> output_R1 <b>-op</b> output_R2 or

       <b>-i</b> db1-R1.fa <b>-j</b> db1-R2.fa <b>-i2</b> db2-R1.fq <b>-j2</b> db2-R2.fq <b>-o</b> output_R1 <b>-op</b> output_R2

       <b>-o</b>     output filename, required

       <b>-op</b>    output filename for R2 reads if input are paired end (PE) files

       <b>-c</b>     sequence  identity  threshold, default 0.9 this is the default cd-hit's "global sequence identity"
              calculated as: number of identical amino acids or bases in alignment divided by the full length of
              the shorter sequence

       <b>-G</b>     use global sequence identity, default 1 if set to 0, then use local sequence identity,  calculated
              as  : number of identical amino acids or bases in alignment divided by the length of the alignment
              NOTE!!! don't use <b>-G</b> 0 unless you use alignment coverage controls see options <b>-aL</b>, <b>-AL</b>, <b>-aS</b>, <b>-AS</b>

       <b>-b</b>     band_width of alignment, default 20

       <b>-M</b>     memory limit (in MB) for the program, default 800; 0 for unlimitted;

       <b>-T</b>     number of threads, default 1; with 0, all CPUs will be used

       <b>-n</b>     word_length, default 10, see user's guide for choosing it

       <b>-l</b>     length of throw_away_sequences, default 10

       <b>-d</b>     length of description in .clstr file, default 20 if set to 0, it takes the fasta defline and stops
              at first space

       <b>-s</b>     length difference cutoff, default 0.0 if set to 0.9, the shorter sequences need to be at least 90%
              length of the representative of the cluster

       <b>-S</b>     length difference cutoff in amino acid, default 999999 if set to 60, the length difference between
              the shorter sequences and the representative of the cluster can not be bigger than 60

       <b>-s2</b>    length difference cutoff for db1, default 1.0 by default, seqs in db1 &gt;= seqs in  db2  in  a  same
              cluster if set to 0.9, seqs in db1 may just &gt;= 90% seqs in db2

       <b>-S2</b>    length  difference  cutoff,  default 0 by default, seqs in db1 &gt;= seqs in db2 in a same cluster if
              set to 60, seqs in db2 may 60aa longer than seqs in db1

       <b>-aL</b>    alignment coverage for the longer sequence, default 0.0 if set to 0.9, the alignment  must  covers
              90% of the sequence

       <b>-AL</b>    alignment  coverage control for the longer sequence, default 99999999 if set to 60, and the length
              of the sequence is 400, then the alignment must be &gt;= 340 (400-60) residues

       <b>-aS</b>    alignment coverage for the shorter sequence, default 0.0 if set to 0.9, the alignment must  covers
              90% of the sequence

       <b>-AS</b>    alignment coverage control for the shorter sequence, default 99999999 if set to 60, and the length
              of the sequence is 400, then the alignment must be &gt;= 340 (400-60) residues

       <b>-A</b>     minimal  alignment coverage control for the both sequences, default 0 alignment must cover &gt;= this
              value for both sequences

       <b>-uL</b>    maximum unmatched percentage for the longer sequence, default 1.0 if set  to  0.1,  the  unmatched
              region (excluding leading and tailing gaps) must not be more than 10% of the sequence

       <b>-uS</b>    maximum  unmatched  percentage  for the shorter sequence, default 1.0 if set to 0.1, the unmatched
              region (excluding leading and tailing gaps) must not be more than 10% of the sequence

       <b>-U</b>     maximum unmatched length, default 99999999 if set to 10, the unmatched region  (excluding  leading
              and tailing gaps) must not be more than 10 bases

       <b>-B</b>     1  or  0,  default  0, by default, sequences are stored in RAM if set to 1, sequence are stored on
              hard drive !! No longer supported !!

       <b>-P</b>     input paired end (PE) reads, default 0, single file if set to 1, please use <b>-i</b> R1 <b>-j</b> R2  to  input
              both PE files

       <b>-cx</b>    length  to  keep  after  trimming  the tail of sequence, default 0, not trimming if set to 50, the
              program only uses the first 50 letters of input sequence

       <b>-cy</b>    length to keep after trimming the tail of R2 sequence, default 0, not trimming if set to  50,  the
              program  only  uses  the  first 50 letters of input R2 sequence e.g. <b>-cx</b> 100 <b>-cy</b> 80 for paired end
              reads

       <b>-p</b>     1 or 0, default 0 if set to 1, print alignment overlap in .clstr file

       <b>-g</b>     1 or 0, default 0 by cd-hit's default algorithm, a sequence is clustered to the first cluster that
              meet the threshold (fast cluster). If set to 1, the program will cluster it into the most  similar
              cluster  that  meet  the  threshold  (accurate  but  slow mode) but either 1 or 0 won't change the
              representatives of final clusters

       <b>-r</b>     1 or 0, default 1, by default do both +/+ &amp; +/- alignments if set to 0, only +/+ strand alignment

       <b>-mask</b>  masking letters (e.g. <b>-mask</b> NX, to mask out both 'N' and 'X')

       <b>-match</b> matching score, default 2 (1 for T-U and N-N)

       <b>-mismatch</b>
              mismatching score, default <b>-2</b>

       <b>-gap</b> gap opening score, default <b>-6</b>

       <b>-gap-ext</b>
              gap extension score, default <b>-1</b>

       <b>-bak</b> write backup cluster file (1 or 0, default 0)

       <b>-h</b>     print this help

              Questions, bugs, contact Weizhong Li at <a href="mailto:liwz@sdsc.edu">liwz@sdsc.edu</a> For updated versions and information, please
              visit: <a href="http://cd-hit.org">http://cd-hit.org</a>

              or https://github.com/weizhongli/cdhit

              cd-hit web server is also available from <a href="http://cd-hit.org">http://cd-hit.org</a>

              If you find cd-hit useful, please kindly cite:

              "CD-HIT: a fast program  for  clustering  and  comparing  large  sets  of  protein  or  nucleotide
              sequences",  Weizhong  Li  &amp; Adam Godzik. Bioinformatics, (2006) 22:1658-1659 "CD-HIT: accelerated
              for clustering the next generation sequencing data", Limin Fu, Beifang Niu, Zhengwei Zhu, Sitao Wu
              &amp; Weizhong Li. Bioinformatics, (2012) 28:3150-3152

cd-hit-est-2d 4.8.1                                August 2021                                  <u><a href="../man1/CD-HIT-EST-2D.1.html">CD-HIT-EST-2D</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>