<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>hhmake - build an HMM from an input alignment or convert between HMMER format and HHsearch format</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/hhsuite">hhsuite_3.3.0+ds-8_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       hhmake - build an HMM from an input alignment or convert between HMMER format and HHsearch format

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>hhmake</b> <u>-i</u> <u>&lt;file&gt;</u> <u>-o</u> <u>&lt;file&gt;</u> [<u>options</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       HHmake  3.3.0 Build an HMM from an input alignment in A2M, A3M, or FASTA format, or convert between HMMER
       format (.hmm) and HHsearch format (.hhm).  Steinegger M, Meier M, Mirdita M, V??hringer H, Haunsberger  S
       J,  and  S??ding  J (2019) HH-suite3 for fast remote homology detection and deep protein annotation.  BMC
       Bioinformatics, doi:10.1186/s12859-019-3019-7 (c) The HH-suite development team

       <b>-i</b> &lt;file&gt;
              query alignment (A2M, A3M, or FASTA), or query HMM

       &lt;file&gt; may be 'stdin' or 'stdout' throughout.

   <b>Output</b> <b>options:</b>
       <b>-o</b> &lt;file&gt;
              HMM file to be written to  (default=&lt;infile.hhm&gt;)

       <b>-a</b> &lt;file&gt;
              HMM file to be appended to

       <b>-v</b> &lt;int&gt;
              verbose mode: 0:no screen output  1:only warings  2: verbose

       <b>-seq</b> &lt;int&gt;
              max. number of  query/template  sequences  displayed  (def=10)  Beware  of  overflows!  All  these
              sequences are stored in memory.

       <b>-add_cons</b>
              make consensus sequence master sequence of query MSA

       <b>-name</b> &lt;name&gt;
              use this name for HMM (default: use name of first sequence)

       Filter query multiple sequence alignment

       <b>-id</b>    [0,100]  maximum pairwise sequence identity (%) (def=90)

       <b>-diff</b> [0,inf[
              filter MSA by selecting most diverse set of sequences, keeping at least this many seqs in each MSA
              block of length 50 (def=100)

       <b>-cov</b>   [0,100]  minimum coverage with query (%) (def=0)

       <b>-qid</b>   [0,100]  minimum sequence identity with query (%) (def=0)

       <b>-qsc</b>   [0,100]  minimum score per column with query  (def=-20.0)

       <b>-neff</b> [1,inf]
              target diversity of alignment (default=off)

   <b>Input</b> <b>alignment</b> <b>format:</b>
       <b>-M</b> a2m use  A2M/A3M  (default): upper case = Match; lower case = Insert; '-' = Delete; '.' = gaps aligned
              to inserts (may be omitted)

       <b>-M</b> first
              use FASTA: columns with residue in 1st sequence are match states

       <b>-M</b> [0,100]
              use FASTA: columns with fewer than X% gaps are match states

   <b>Pseudocount</b> <b>(pc)</b> <b>options:</b>
              Context specific hhm pseudocounts:

       <b>-pc_hhm_contxt_mode</b> {0,..,3}
              position dependence of pc admixture 'tau' (pc mode, default=0)

       0: no pseudo counts:
              tau = 0

       1: constant
              tau = a

              2:  diversity-dependent:  tau  =   a/(1+((Neff[i]-1)/b)^c)   3:   CSBlast   admixture:     tau   =
              a(1+b)/(Neff[i]+b) (Neff[i]: number of effective seqs in local MSA around column i)

       <b>-pc_hhm_contxt_a</b>
              [0,1]        overall pseudocount admixture (def=0.9)

       <b>-pc_hhm_contxt_b</b>
              [1,inf[      Neff threshold value for mode 2 (def=4.0)

       <b>-pc_hhm_contxt_c</b>
              [0,3]        extinction exponent c for mode 2 (def=1.0)

              Context  independent  hhm  pseudocounts  (used for templates; used for query if contxt file is not
              available):

       <b>-pc_hhm_nocontxt_mode</b> {0,..,3}
              position dependence of pc admixture 'tau' (pc mode, default=2)

       0: no pseudo counts:
              tau = 0

       1: constant
              tau = a

              2: diversity-dependent: tau = a/(1+((Neff[i]-1)/b)^c) (Neff[i]: number of effective seqs in  local
              MSA around column i)

       <b>-pc_hhm_nocontxt_a</b>
              [0,1]        overall pseudocount admixture (def=1.0)

       <b>-pc_hhm_nocontxt_b</b>
              [1,inf[      Neff threshold value for mode 2 (def=1.5)

       <b>-pc_hhm_nocontxt_c</b>
              [0,3]        extinction exponent c for mode 2 (def=1.0)

              Context-specific pseudo-counts:

       <b>-nocontxt</b>
              use substitution-matrix instead of context-specific pseudocounts

       <b>-contxt</b> &lt;file&gt; context file for computing context-specific pseudocounts (default=)

   <b>Other</b> <b>options:</b>
       <b>-maxseq</b> &lt;int&gt;
              max number of input rows (def=65535)

       <b>-maxres</b> &lt;int&gt;
              max number of HMM columns (def=20001)

       Example: hhmake <b>-i</b> test.a3m <b>-o</b> stdout

hhmake 3.3.0+ds                                    August 2023                                         <u><a href="../man1/HHMAKE.1.html">HHMAKE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>