<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>phastCons - Identify conserved elements or produce conservation scores, given</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/phast">phast_1.7+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       phastCons - Identify conserved elements or produce conservation scores, given

</pre><h4><b>SYNOPSIS</b></h4><pre>
       The  alignment  file  can  be in any of several file formats (see <b>--msa-format</b>).  The phylogenetic models
       must be in the .mod format produced by the phyloFit program.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Identify conserved elements or produce conservation scores, given a multiple alignment and  a  phylo-HMM.
       By  default,  a  phylo-HMM consisting of two states is assumed: a "conserved" state and a "non-conserved"
       state.  Separate phylogenetic models can be specified for these two states, e.g.,

              phastCons myfile.ss cons.mod,noncons.mod &gt; scores.wig

       or a single model can be given for the non-conserved state, e.g.,

              phastCons myfile.ss <b>--rho</b> 0.5 noncons.mod &gt; scores.wig

       in which case the model for the conserved state will be obtained by multiplying all branch lengths by the
       scaling parameter rho (0 &lt; rho &lt; 1).  If the <b>--rho</b> option is not used, rho will be  set  to  its  default
       value of 0.3.

       By  default,  the phylogenetic models will be left unaltered, but if the <b>--estimate-trees</b> option is used,
       e.g.,

              phastCons myfile.ss init.mod <b>--estimate-trees</b> newtree &gt; scores.wig

       then the phylogenetic models for the two states will be estimated from the data, and the given tree model
       (there must be only one in this case) will be used for initialization  only.   It  is  also  possible  to
       estimate  only the scale factor <b>--rho</b>, using the <b>--estimate-rho</b> option.  The transition probabilities for
       the HMM can either be specified at the command line or estimated from the data using an EM algorithm.  To
       specify them at the command line, use either  the  <b>--transitions</b>  option  or  the  <b>--target-coverage</b>  and
       <b>--expected-length</b>  options.   The  recommended  method is to use <b>--target-coverage</b> and <b>--expected-length</b>,
       e.g.,

              phastCons <b>--target-coverage</b> 0.25 <b>--expected-length</b> 12 myfile.ss cons.mod,noncons.mod &gt; scores.wig

   <b>The</b> <b>program</b> <b>produces</b> <b>two</b> <b>main</b> <b>types</b> <b>of</b> <b>output.</b>
       The      primary      output,      sent      to      stdout      in      fixed-step      WIG       format
       (<a href="http://genome.ucsc.edu/goldenPath/help/wiggle.html">http://genome.ucsc.edu/goldenPath/help/wiggle.html</a>),  is a set of base-by-base conservation scores.  The
       score at each base is equal to the posterior probability that that base was "generated" by the  conserved
       state  of  the  phylo-HMM.   The  scores  are  reported in the coordinate frame of a designated reference
       sequence (see <b>--refidx</b>), which is by default the first sequence in the alignment.  They can be suppressed
       with the <b>--no-post-probs</b> option.  The secondary type of output, activated with the <b>--most-conserved</b>  (aka
       <b>--viterbi</b>)  option,  is a set of discrete conserved elements.  These elements are output in either BED or
       GFF format, also in the coordinate system of the reference sequence (see <b>--most-conserved</b>).  They can  be
       assigned log-odds scores using the <b>--score</b> option.

       Other  uses are also supported, but will not be described in detail here.  For example, it is possible to
       produce conservation scores and conserved elements using a k-state phylo-HMM of  the  kind  described  by
       Felsenstein  and  Churchill  (1996)  (see <b>--FC</b>), and it is possible to produce a "coding potential" score
       instead of a conservation score (see <b>--coding-potential</b>).  It is also possible  to  give  the  program  a
       custom HMM and to specify any subset of its states to use for prediction (see <b>--hmm</b> and <b>--states</b>).

       See the phastCons HOWTO for additional details.

</pre><h4><b>EXAMPLE</b></h4><pre>
       1.  Given  phylogenetic  models  for  conserved  and  nonconserved regions and HMM transition parameters,
       compute a set of conservation scores.

              phastCons <b>--transitions</b> 0.01,0.01 mydata.ss cons.mod,noncons.mod &gt; scores.wig

       2. Similar to (1), but define the conserved model as a scaled version of  the  nonconserved  model,  with
       rho=0.4  as  the  scaling parameter.  Also predict conserved elements as well as conservation scores, and
       assign log-odds scores to predictions.

              phastCons <b>--transitions</b>  0.01,0.01  <b>--most-conserved</b>  mostcons.bed  <b>--score</b>  <b>--rho</b>  0.4  mydata.ss
              noncons.mod &gt; scores.wig

       (if output file were "mostcons.gff," then output would be in GFF instead of BED format)

       3.  This  time,  estimate  the  parameter  rho from the data.  Suppress both the scores and the conserved
       elements.  Specify the transition probabilities using <b>--target-coverage</b> and <b>--expected-length</b> instead  of
       <b>--transitions</b>.

              phastCons  <b>--target-coverage</b>  0.25  <b>--expected-length</b>  12  <b>--estimate-rho</b>  newtree <b>--no-post-probs</b>
              mydata.ss noncons.mod

       4. This time estimate all free parameters of the tree models.

              phastCons <b>--target-coverage</b> 0.25 <b>--expected-length</b>  12  <b>--estimate-trees</b>  newtree  <b>--no-post-probs</b>
              mydata.ss noncons.mod

       5.  Estimate the state-transition parameters but not the tree models.  Output the conservation scores but
       not the conserved elements.

              phastCons mydata.ss cons.mod,noncons.mod &gt; scores.wig

       6. Estimate just the expected-length parameter and also estimate rho.

              phastCons <b>--target-coverage</b> 0.25 <b>--estimate-rho</b> newtree mydata.ss noncons.mod &gt; scores.wig

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Tree</b> <b>models</b>

       <b>--rho</b>, <b>-R</b> &lt;rho&gt;

              Set the *scale* (overall evolutionary rate) of the model for the conserved state to be &lt;rho&gt; times
              that of the model for the non-conserved state (0  &lt;  &lt;rho&gt;  &lt;  1;  default  0.3).   If  used  with
              <b>--estimate-trees</b>  or <b>--estimate-rho</b>, the specified value will be used for initialization only (rho
              will be estimated).  This option is ignored if two tree models are given.

       <b>--estimate-trees</b>, <b>-T</b> &lt;fname_root&gt; Estimate free parameters  of  tree  models  and  write  new  models  to
              &lt;fname_root&gt;.cons.mod and &lt;fname_root&gt;.noncons.mod.

       <b>--estimate-rho</b>, <b>-O</b> &lt;fname_root&gt;

              Like <b>--estimate-trees</b>, but estimate only the parameter rho.

       <b>--gc</b>,  <b>-G</b>  &lt;val&gt;  (Optionally use with <b>--estimate-trees</b> or <b>--estimate-rho</b>) Assume a background nucleotide
              distribution consistent with the given average G+C content (0 &lt; &lt;val&gt; &lt; 1)  when  estimating  tree
              models.  (The frequencies of G and C will be set to &lt;val&gt;/2 and the frequencies of A and T will be
              set  to  (1-&lt;val&gt;)/2.)   This  option  overrides the default behavior of estimating the background
              distribution from the data (if <b>--estimate-trees</b>) or  obtaining  them  from  the  input  model  (if
              <b>--estimate-rho</b>).

       <b>--nrates</b>,  <b>-k</b>  &lt;nrates&gt;  |  &lt;nrates_conserved,nrates_nonconserved&gt;  (Optionally use with a discrete-gamma
              model and <b>--estimate-trees</b>) Assume the specified number of rate categories, instead of the  number
              given  in the *.mod file.  The shape parameter 'alpha' will be as given in the *.mod file.  In the
              case of the default two-state HMM, two values can be specified, for the numbers of rates  for  the
              conserved and the nonconserved states, resp.

   <b>State-transition</b> <b>parameters</b>

       <b>--transitions</b>, <b>-t</b> [~]&lt;mu&gt;,&lt;nu&gt;

              Fix the transition probabilities of the two-state HMM as specified, rather than estimating them by
              maximum  likelihood.   Alternatively,  if first character of argument is '~', estimate parameters,
              but initialize to specified values.  The argument &lt;mu&gt; is the probability  of  transitioning  from
              the  conserved  to the non-conserved state, and &lt;nu&gt; is the probability of the reverse transition.
              The probabilities of self transitions are thus 1-&lt;mu&gt; and  1-&lt;nu&gt;  and  the  expected  lengths  of
              conserved and nonconserved elements are 1/&lt;mu&gt; and 1/&lt;nu&gt;, respectively.

       <b>--target-coverage</b>, <b>-C</b> &lt;gamma&gt;

              (Alternative  to <b>--transitions</b>) Constrain transition parameters such that the expected fraction of
              sites in conserved elements is &lt;gamma&gt; (0  &lt;  &lt;gamma&gt;  &lt;  1).   This  is  a  *prior*  rather  than
              *posterior*  expectation  and  assumes  stationarity of the state-transition process.  Adding this
              constraint  causes  the  ratio  mu/nu  to  be  fixed  at  (1-&lt;gamma&gt;)/&lt;gamma&gt;.    If   used   with
              <b>--expected-length</b>,   the   transition  probabilities  will  be  completely  fixed;  otherwise  the
              expected-length parameter &lt;omega&gt; will be estimated by maximum likelihood.  <b>--expected-length</b>,  <b>-E</b>
              [~]&lt;omega&gt; {--expected-lengths also allowed, for backward compatibility}

              (For  use  with <b>--target-coverage</b>, alternative to <b>--transitions</b>) Set transition probabilities such
              that the expected length of a conserved element is &lt;omega&gt;.  Specifically, the parameter mu is set
              to 1/&lt;omega&gt;.  If preceded by '~', &lt;omega&gt; will be estimated,  but  will  be  initialized  to  the
              specified value.

   <b>Input/output</b>

       <b>--msa-format</b>, <b>-i</b> PHYLIP|FASTA|MPM|SS|MAF

       Alignment file format.
              Default is to guess format based on

       file contents.
              Note that the msa_view program can be used to

              convert between formats.

       <b>--viterbi</b>  [alternatively  <b>--most-conserved]</b>,  <b>-V</b>  &lt;fname&gt;  Predict  discrete  elements using the Viterbi
              algorithm and write to specified file.  Output is in BED format, unless &lt;fname&gt; has suffix ".gff",
              in which case output is in GFF.

       <b>--score</b>, <b>-s</b> (Optionally use with <b>--viterbi</b>) Assign a log-odds score to each prediction.

       <b>--lnl</b>, <b>-L</b> &lt;fname&gt;

              Compute total log likelihood using the forward algorithm and write to specified file.

       <b>--no-post-probs</b>, <b>-n</b> Suppress output of posterior probabilities.  Useful  if  only  discrete  elements  or
              likelihood is of interest.

       <b>--log</b>, <b>-g</b> &lt;log_fname&gt;

              (Optionally  use when estimating free parameters) Write log of optimization procedure to specified
              file.

       <b>--refidx</b>, <b>-r</b> &lt;refseq_idx&gt; Use coordinate frame of specified sequence in output.  Default

              value is 1, first  sequence  in  alignment;  0  indicates  coordinate  frame  of  entire  multiple
              alignment.

       <b>--seqname</b>,  <b>-N</b> &lt;name&gt; (Optionally use with <b>--viterbi</b>) Use specified string for 'seqname' (GFF) or 'chrom'
              field in output file.  Default is obtained from input  file  name  (double  filename  root,  e.g.,
              "chr22" if input file is "chr22.35.ss").

       <b>--idpref</b>, <b>-P</b> &lt;name&gt;

              (Optionally  use  with  <b>--viterbi</b>) Use specified string as prefix of generated ids in output file.
              Can be used to ensure ids are unique.  Default is obtained from input file name  (single  filename
              root, e.g., "chr22.35" if input file is "chr22.35.ss").

       <b>--quiet</b>, <b>-q</b> Proceed quietly (without updates to stderr).

       <b>--help</b>, <b>-h</b>

              Print this help message.  (Indels) [experimental]

       <b>--indels</b>, <b>-I</b>

              Expand HMM state space to model indels as described in Siepel &amp; Haussler (2004).

       <b>--max-micro-indel</b>,  <b>-Y</b>  &lt;length&gt;  (Optionally use with <b>--indels</b>) Maximum length of an alignment gap to be
              considered a "micro-indel" and therefore addressed by the indel  model.   Gaps  longer  than  this
              threshold will be treated as missing data.  Default value is 20.

       <b>--indel-params</b>, <b>-D</b> [~]&lt;alpha_0,beta_0,tau_0,alpha_1,beta_1,tau_1&gt;

              (Optionally  use  with  <b>--indels</b>  and default two-state HMM) Fix the indel parameters at (alpha_0,
              beta_0, tau_0) for the conserved state and at  (alpha_1,  beta_1,  tau_1)  for  the  non-conserved
              state,  rather  than  estimating them by maximum likelihood.  Alternatively, if first character of
              argument is '~', estimate parameters, but initialize with specified values.  Alpha_j is  the  rate
              of  insertion events per substitution per site in state j (typically ~0.05), beta_j is the rate of
              deletion events per substitution per site in state j (typically ~0.05), and tau_j is approximately
              the inverse of the expected indel length in state j (typically 0.2-0.5).

       <b>--indels-only</b>, <b>-J</b> Like <b>--indels</b> but force the use of a single-state HMM.  This option allows  the  effect
              of  the  indel  model  in  isolation  to  be  observed.  Implies <b>--no-post-probs</b>.  Use with <b>--lnl</b>.
              (Felsenstein/Churchill model) [rarely used]

       <b>--FC</b>, <b>-X</b>

              (Alternative to <b>--hmm</b>; specify only one *.mod file with this option) Use an HMM with a  state  for
              every  rate  category  in the given phylogenetic model, and transition probabilities defined by an
              autocorrelation parameter lambda (as described  by  Felsenstein  and  Churchill,  1996).   A  rate
              constant  for  each  state  (rate  category)  will  be  multiplied  by  the  branch lengths of the
              phylogenetic model, to create a "scaled" version of the model for that state.  If the phylogenetic
              model was estimated using Yang's discrete gamma method (<b>-k</b> option  to  phyloFit),  then  the  rate
              constants will be defined according to the estimated shape parameter 'alpha', as described by Yang
              (1994).   Otherwise,  a  nonparameteric  model of rate variation must have been used (<b>-K</b> option to
              phyloFit), and the rate constants will be as defined (explicitly) in the *.mod file.  By  default,
              the parameter lambda will be estimated by maximum likelihood (see <b>--lambda</b>).

       <b>--lambda</b>, <b>-l</b> [~]&lt;lambda&gt;

              (Optionally  use with <b>--FC</b>) Fix lambda at the specified value rather than estimating it by maximum
              likelihood.  Alternatively, if first character is '~', estimate but initialize at specified value.
              Allowable range is 0-1.  With k rate categories, the transition probability between  state  i  and
              state  j  will  be  lambda  *  I(i == j) + (1-lambda)/k, where I is the indicator function.  Thus,
              lambda = 0 implies no autocorrelation and lambda = 1  implies  perfect  autocorrelation.   (Coding
              potential) [experimental]

       <b>--coding-potential</b>, <b>-p</b>

              Use  parameter  settings  that  cause  output to be interpretable as a coding potential score.  By
              default, a simplified version of exoniphy's phylo-HMM  is  used,  with  a  noncoding  (background)
              state,  a conserved non-coding (CNS) state, and states for the three codon positions.  This option
              implies <b>--catmap</b> "NCATS=4; CNS 1; CDS 2-4" <b>--hmm</b> &lt;default-HMM-file&gt; <b>--states</b> CDS  <b>--reflect-strand</b>
              background,CNS and a set of default *.mod files (all of which can be overridden).  This option can
              be used with or without <b>--indels</b>.

       <b>--extrapolate</b>, <b>-e</b> &lt;phylog.nh&gt; | default

              Extrapolate to a larger set of species based on the given phylogeny (Newick-format).  The trees in
              the given tree models (*.mod files) must be subtrees of the larger phylogeny.  For each tree model
              M,  a  copy will be created of the larger phylogeny, then scaled such that the total branch length
              of the subtree corresponding to M's tree equals the total branch length  of  M's  tree;  this  new
              version will then be used in place of M's tree.  (Any species name present in this tree but not in
              the  data  will  be  ignored.)   If  the  string  "default" is given instead of a filename, then a
              phylogeny for 25 vertebrate species, estimated from sequence data for Target 1 (CFTR) of the  NISC
              Comparative Sequencing Program (Thomas et al., 2003), will be assumed.

       <b>--alias</b>, <b>-A</b> &lt;alias_def&gt;

              Alias  names  in  input  alignment  according  to  given definition, e.g., "hg17=human; mm5=mouse;
              rn3=rat".  Useful with default *.mod files, e.g., with <b>--coding-potential</b>.   (Default  models  use
              generic  common  names  such  as  "human", "mouse", and "rat".  This option allows a mapping to be
              established between the leaves of trees in these files and the sequences of an alignment that uses
              an alternative naming convention.)

   <b>Custom</b> <b>HMMs</b> <b>[rarely</b> <b>used]</b>

       <b>--hmm</b>, <b>-H</b> &lt;hmm_fname&gt;

              Name of HMM file explicitly defining the probabilities of all state transitions.   States  in  the
              file  must  correspond  in  number and order to phylogenetic models in &lt;mod_fname_list&gt;.  Expected
              file format is as produced by 'hmm_train.'

       <b>--catmap</b>, <b>-c</b> &lt;fname&gt;|&lt;string&gt; (Optionally use with <b>--hmm</b>) Mapping of feature types to  category  numbers.
              Can  give  either  a  filename or an "inline" description of a simple category map, e.g., <b>--catmap</b>
              "NCATS = 3 ; CDS 1-3".

       <b>--states</b>, <b>-S</b> &lt;state_list&gt;

              States of interest in the phylo-HMM, specified by number (indexing starts with 0), or if <b>--catmap</b>,
              by category name.  Default value is 1.  Choosing <b>--states</b> "0,1,2" will cause output of the sum  of
              the  posterior  probabilities  for states 0, 1, and 2, and/or of regions in which the Viterbi path
              coincides with (any of) states 0, 1, or 2 (see <b>--viterbi</b>).

       <b>--reflect-strand</b>, <b>-U</b> &lt;pivot_states&gt;

              (Optionally use with <b>--hmm</b>) Given an HMM describing the forward strand, create a larger  HMM  that
              allows  for  features on both strands by "reflecting" the original HMM about the specified "pivot"
              states.  The new HMM will be used for prediction on both strands.   States  can  be  specified  by
              number (indexing starts with 0), or if <b>--catmap</b>, by category name.

   <b>Missing</b> <b>data</b> <b>[rarely</b> <b>used]</b>

       <b>--require-informative</b>,  <b>-M</b>  &lt;states&gt;  Require  "informative"  columns  (i.e.,  columns with more than two
              non-missing-data characters, excluding sequences specified by <b>--not-informative</b>) in specified  HMM
              states, to help eliminate false positive predictions.  States can be specified by number (indexing
              starts  with  0) or, if <b>--catmap</b> is used, by category name.  Non-informative columns will be given
              emission probabilities of zero.  By default, this option is active, with &lt;states&gt; equal to the set
              of states of  interest  for  prediction  (as  specified  by  <b>--states</b>).   Use  "none"  to  disable
              completely.

       <b>--not-informative</b>, <b>-F</b> &lt;list&gt;

              Do  not  consider  the  specified  sequences  (listed  by  name) when deciding whether a column is
              informative.  This option may be useful when sequences are present that  are  very  close  to  the
              reference  sequence and thus do not contribute much in the way of phylogenetic information.  E.g.,
              one might use "--not-informative chimp" with a human-referenced multiple alignment including chimp
              sequence, to avoid false-positive predictions based only  on  human/chimp  alignments  (can  be  a
              problem, e.g., with <b>--coding-potential</b>).

       <b>--ignore-missing</b>, <b>-z</b>

              (For use when estimating transition probabilities) Ignore regions of missing data in all sequences
              but  the  reference  sequence (excluding sequences specified by <b>--not-informative</b>) when estimating
              transition probabilities.  Can help avoid too-low estimates of &lt;mu&gt; and &lt;nu&gt; or too-high estimates
              of &lt;lambda&gt;.  Warning: this option should not be used with <b>--viterbi</b> because coordinates in output
              will be unrecognizable.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       J. Felsenstein and G. Churchill.  1996. A hidden Markov model approach to variation among sites  in  rate
       of evolution.  Mol. Biol. Evol., 13:93-104.  A. Siepel, G. Bejerano, J. S. Pedersen, et al.  2005.

       Evolutionarily conserved elements in vertebrate, insect, worm, and yeast genomes.  Genome Res. (in press)
       A. Siepel and D. Haussler.  2004.  Computational identification of evolutionarily conserved exons.  Proc.
       8th Annual Int'l Conf.  on Research in Computational Biology (RECOMB '04), pp. 177-186.

       J.  Thomas  et al.  2003.  Comparative analyses of multi-species sequences from targeted genomic regions.
       Nature 424:788-793.

       Z. Yang. 1994. Maximum likelihood phylogenetic estimation from DNA sequences  with  variable  rates  over
       sites: approximate methods. J. Mol. Evol., 39:306-314.

phastCons 1.4                                       May 2016                                        <u><a href="../man1/PHASTCONS.1.html">PHASTCONS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>