<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>phastOdds - Compute log-odds scores based on two phylogenetic models or phylo-HMMs,</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/phast">phast_1.7+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       phastOdds - Compute log-odds scores based on two phylogenetic models or phylo-HMMs,

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Compute  log-odds  scores  based  on  two phylogenetic models or phylo-HMMs, one for features of interest
       (e.g., coding exons, conserved regions) and one for background.  Will either (1) compute a score for each
       feature in an input set, and output the same set of features with scores; or (2) output a separate  score
       for  each  position in fixed-step WIG format (<a href="http://genome.ucsc.edu/goldenPath/help/wiggle.html">http://genome.ucsc.edu/goldenPath/help/wiggle.html</a>); or (3)
       compute scores in a sliding window of designated size, and output a three-column file, with the index  of
       the  center  of  each window followed by the score for that window on the positive strand, then the score
       for that window on the negative strand.  The default is to assume a reference  sequence  alignment,  with
       the reference sequence appearing first; feature coordinates are assumed to be defined with respect to the
       reference sequence (see <b>--refidx</b>).

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>phastOdds</b>   [OPTIONS]   <b>--background-mods</b>   &lt;bmods&gt;   [--background-hmm  &lt;bhmm&gt;]  <b>--feature-mods</b>  &lt;fmods&gt;
       [--feature-hmm &lt;fhmm&gt;] ( <b>--features</b> &lt;feats&gt; | <b>--window</b> &lt;size&gt; ) &lt;alignment&gt;

       Arguments &lt;bmods&gt; and &lt;fmods&gt; should be comma-delimited lists of phylogenetic models in .mod  format  (as
       produced  by  phyloFit),  &lt;feats&gt; may be in GFF, BED, or genepred format, and &lt;alignment&gt; may be in FASTA
       format or an alternative format specified by <b>--msa-format</b>.  HMM files should be in  the  format  used  by
       exoniphy.

</pre><h4><b>EXAMPLE</b></h4><pre>
       (See below for more details on options)

       1. Compute conservation scores for features in a GFF file, based on a

       model  for conserved sites (conserved.mod) vs. a model of neutral evolution (neutral.mod).  (These models
       may be estimated with phyloFit or phastCons.)

              phastOdds  <b>--background-mods</b>  neutral.mod  <b>--feature-mods</b>  conserved.mod  <b>--features</b>  features.gff
              alignment.fa &gt; scores.gff

       Features  could  alternatively  be  specified in BED or genepred format (format will be auto-recognized).
       The program can be made to produce BED-formatted output with <b>--output-bed</b>.

       2. Compute conservation scores in a sliding window of size 100.

              phastOdds <b>--background-mods</b> neutral.mod <b>--feature-mods</b> conserved.mod <b>--window</b> 100  alignment.fa  &gt;
              scores.dat

       (Window  is  advanced  one  site at a time.  Window boundaries are defined in the coordinate frame of the
       multiple alignment, but center coordinates are converted to the frame of the reference sequence  as  they
       are output.)

       3. Compute a "coding potential" score for features in a BED file, based on a phylo-HMM for coding regions
       versus a phylo-HMM for noncoding DNA, with states for conserved and nonconserved sequences.

              phastOdds    <b>--background-mods</b>    codon1.mod,codon2.mod,codon3.mod   <b>--background-hmm</b>   coding.hmm
              <b>--feature-mods</b>   neutral.mod,conserved-noncoding.mod   <b>--feature-hmm</b>   noncoding.hmm    <b>--features</b>
              features.bed <b>--output-bed</b> alignment.fa &gt; scores.bed

</pre><h4><b>OPTIONS</b></h4><pre>

       <b>--background-mods</b>, <b>-b</b> &lt;backgd_mods&gt;

              (Required)  Comma-delimited  list  of  tree  model  (*.mod)  files  for  background.  If used with
              <b>--background-hmm</b>, order of models must correspond to order of states in HMM.

       <b>--background-hmm</b>, <b>-B</b> &lt;backgd.hmm&gt;

       HMM for background.
              If there is only one backgound tree

              model, a trivial (single-state) HMM will be assumed.

       <b>--feature-mods</b>, <b>-f</b> &lt;feat_mods&gt; (Required) Comma-delimited list of tree model (*.mod) files for  features.
              If used with <b>--feature-hmm</b>, order of models must correspond to order of states in HMM.

       <b>--feature-hmm</b>,  <b>-F</b>  &lt;feat.hmm&gt; HMM for features.  If there is only one tree model for features, a trivial
              (single-state) HMM will be assumed.

       <b>--features</b>, <b>-g</b> &lt;feats.gff&gt;

              (Required unless <b>-w</b> or <b>-y</b>) File defining features to be scored (GFF, BED, or genepred).

       <b>--window</b>, <b>-w</b> &lt;size&gt; (Can be used instead of <b>-g</b> or <b>-y</b>) Compute scores in a sliding window of the specified
              size.

       <b>--base-by-base</b>, <b>-y</b>

              (Can be used instead of <b>-g</b> or <b>-y</b>) Output base-by-base scores,  in  the  coordinate  frame  of  the
              reference sequence (or of the sequence specified by <b>--refidx</b>).  Output is in fixed-step WIG format
              (<a href="http://genome.ucsc.edu/goldenPath/help/wiggle.html">http://genome.ucsc.edu/goldenPath/help/wiggle.html</a>).    This   option   can  only  be  used  with
              individual phylogenetic models, not with sets of models and a (nontrivial) HMM.

       <b>--window-wig</b>, <b>-W</b> &lt;size&gt;

              (Can be used instead of <b>-g</b> or <b>-y</b>) Like <b>--window</b> but outputs scores in fixed-step  WIG  format,  as
              with <b>--base-by-base</b>.  Scores for the positive strand only are output.

       <b>--msa-format</b>, <b>-i</b> &lt;type&gt;

       Input format for alignment.
              May be FASTA, PHYLIP, MPM, SS, or

              MAF (default is to guess format from file contents).

       <b>--refidx</b>,  <b>-r</b>  &lt;ref_seq&gt;  Index  of reference sequence for coordinates.  Use 0 to indicate the coordinate
              system of the alignment as a whole.  Default is 1, for first sequence.

       <b>--output-bed</b>, <b>-d</b>

              (For use with <b>-g</b>) Generate output in bed format rather than GFF.

       <b>--verbose</b>, <b>-v</b> Verbose mode.  Print messages to stderr describing what the program is doing.

       <b>--help</b>, <b>-h</b>

              Print this help message.

phastOdds 1.4                                       May 2016                                        <u><a href="../man1/PHASTODDS.1.html">PHASTODDS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>