<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>phyloFit - Fits one or more tree models to a multiple alignment of DNA</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/phast">phast_1.7+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       phyloFit - Fits one or more tree models to a multiple alignment of DNA

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Fits  one  or  more tree models to a multiple alignment of DNA sequences by maximum likelihood, using the
       specified tree topology and substitution model.  If categories of sites are defined  via  <b>--features</b>  and
       <b>--catmap</b>  (see  below), then a separate model will be estimated for each category.  A description of each
       model will be written to a separate file, with the suffix ".mod".  These .mod files minimally  include  a
       substitution  rate  matrix,  a  tree  with  branch  lengths,  and  estimates  of  nucleotide  equilibrium
       frequencies.  They may also include information about parameters for modeling rate variation.

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>phyloFit</b> [OPTIONS] &lt;msa_fname&gt;

       &lt;msa_fname&gt; should be a multiple alignment in FASTA format or one of  several  alternative  formats  (see
       <b>--msa-format</b>).   For backward compatibility, this argument may be preceded by '-m' or '--msa'.  Note that
       <b>--tree</b> is required in most cases.  By default, all output files will  have  the  prefix  "phyloFit"  (see
       <b>--out-root</b>).

</pre><h4><b>EXAMPLE</b></h4><pre>
       (If you're like me, you want some basic examples first, and a list of all options later.)

       1. Compute the distance between two aligned sequences (in FASTA file pair.fa) under the REV model.

              phyloFit pair.fa

       (output  is  to  phyloFit.mod;  distance in substitutions per site appears in the TREE line in the output
       file)

       2. Fit a phylogenetic model to an alignment of human, chimp, mouse, and rat  sequences.   Use  the  HKY85
       substitution model.  Write output to files with prefix "myfile".

              phyloFit    <b>--tree</b>    "((human,chimp),(mouse,rat))"    <b>--subst-mod</b>    HKY85    <b>--out-root</b>   myfile
              primate-rodent.fa

       3. As above, but use the discrete-gamma model for rate variation, with 4 rate categories.

              phyloFit <b>--tree</b> "((human,chimp),(mouse,rat))"  <b>--subst-mod</b>  HKY85  <b>--out-root</b>  myfile  <b>--nrates</b>  4
              primate-rodent.fa

       4.  As  above,  but  use  genome-wide  data, stored in the compact "sufficient-statistics" format (can be
       produced with "msa_view <b>-o</b> SS").

              phyloFit <b>--tree</b> "((human,chimp),(mouse,rat))"  <b>--subst-mod</b>  HKY85  <b>--out-root</b>  myfile  <b>--nrates</b>  4
              <b>--msa-format</b> SS primate-rodent.ss

       5.  Fit  a context-dependent phylogenetic model (U2S) to an alignment of human, mouse, and rat sequences.
       Use an EM algorithm for parameter optimization and relax the convergence criteria a bit (recommended with
       context-dependent  models).   Write  a  log  file  for  the  optimization   procedure.    Consider   only
       non-overlapping pairs of sites.

              phyloFit <b>--tree</b> "(human,(mouse,rat))" <b>--subst-mod</b> U2S <b>--EM</b> <b>--precision</b> MED <b>--non-overlapping</b> <b>--log</b>
              u2s.log <b>--out-root</b> hmr-u2s hmr.fa

       6.  As above, but allow overlapping pairs of sites, and compute likelihoods by assuming Markov-dependence
       of columns (see Siepel &amp; Haussler, 2004).  The EM algorithm can no longer be used (optimization  will  be
       much slower).

              phyloFit  <b>--tree</b>  "(human,(mouse,rat))"  <b>--subst-mod</b>  U2S  <b>--precision</b>  MED  <b>--log</b>  u2s-markov.log
              <b>--markov</b> hmr.fa

       7. Compute a likelihood using parameter estimates obtained in (5) and an assumption of Markov dependence.
       This provides a lower bound on the likelihood of the Markov-dependent model.

              phyloFit <b>--init-model</b> hmr-u2s.mod <b>--lnl</b> <b>--markov</b> hmr.fa

       8. Given an alignment of several mammalian sequences (mammals.fa), a tree topology (tree.nh), and  a  set
       of  gene  annotations  in  GFF  (genes.gff),  fit  separate  models  to  sites in 1st, 2nd, and 3rd codon
       positions.  Use the REV substitution model.  Assume coding regions have feature type 'CDS'.

              phyloFit <b>--tree</b> tree.nh <b>--features</b> genes.gff <b>--out-root</b> mammals-rev <b>--catmap</b> "NCATS = 3; CDS  1-3"
              <b>--do-cats</b> 1,2,3 mammals.fa

       (output will be to mammals-rev.cds-1.mod, mammals-rev.cds-2.mod, and mammals-rev.cds-3.mod)

</pre><h4><b>OPTIONS</b></h4><pre>

       <b>--tree</b>, <b>-t</b> &lt;tree_fname&gt;|&lt;tree_string&gt;

              (Required  if  more than three species, or more than two species and a non-reversible substitution
              model, e.g., UNREST, U2, U3) Name of file or literal string defining tree topology.  Tree must  be
              in  Newick  format,  with  the  label at each leaf equal to the index or name of the corresponding
              sequence in the  alignment  (indexing  begins  with  1).   Examples:  <b>--tree</b>  "(1,(2,3))",  <b>--tree</b>
              "(human,(mouse,rat))".   Currently,  the  topology must be rooted.  When a reversible substitution
              model is used, the root is ignored during the optimization procedure.

       <b>--subst-mod</b>, <b>-s</b> JC69|F81|HKY85|HKY85+Gap|REV|SSREV|UNREST|R2|R2S|U2|U2S|R3|R3S|U3|U3S

       (default REV).
              Nucleotide substitution model.  JC69, F81, HKY85

              REV, and UNREST have the usual meanings (see, e.g., Yang, Goldman, and Friday, 1994).  SSREV is  a
              strand-symmetric  version  of  REV.  HKY85+Gap is an adaptation of HKY that treats gaps as a fifth
              character (courtesy of James Taylor).  The others,  all  considered  "context-dependent",  are  as
              defined  in  Siepel and Haussler, 2004.  The options <b>--EM</b> and <b>--precision</b> MED are recommended with
              context-dependent models (see below).

       <b>--msa-format</b>, <b>-i</b> FASTA|PHYLIP|MPM|MAF|SS (default is  to  guess  format  from  file  contents)  Alignment
              format.   FASTA  is  as  usual.  PHYLIP is compatible with the formats used in the PHYLIP and PAML
              packages.  MPM is the format used by the MultiPipMaker aligner and some  other  of  Webb  Miller's
              older tools.  MAF ("Multiple Alignment Format") is used by MULTIZ/TBA and the UCSC Genome Browser.
              SS  is  a  simple format describing the sufficient statistics for phylogenetic inference (distinct
              columns or tuple of columns and their counts).  Note that the program "msa_view" can be  used  for
              file conversion.

       <b>--out-root</b>,  <b>-o</b>  &lt;output_fname_root&gt; (default "phyloFit").  Use specified string as root filename for all
              files created.

       <b>--min-informative</b>, <b>-I</b> &lt;ninf_sites&gt;

              Require at least &lt;ninf_sites&gt; "informative" sites <b>--</b> i.e., sites at which at least two non-gap and
              non-missing-data ('N' or '*') characters are present.  Default is 50.

       <b>--gaps-as-bases</b>, <b>-G</b> Treat alignment gap characters ('-') like ordinary bases.  By

              default, they are treated as missing data.

       <b>--ignore-branches</b>, <b>-b</b> &lt;branches&gt;

              Ignore specified branches in likelihood computations  and  parameter  estimation,  and  treat  the
              induced  subtrees  as  independent.   Can  be  useful  for  likelihood  ratio tests.  The argument
              &lt;branches&gt; should be a comma-separated list of nodes in the tree, indicating  the  branches  above
              these  nodes,  e.g.,  human-chimp,cow-dog.   (See tree_doctor <b>--name-ancestors</b> regarding names for
              ancestral nodes.)  This option does not currently work with <b>--EM</b>.

       <b>--quiet</b>, <b>-q</b> Proceed quietly.

       <b>--help</b>, <b>-h</b>

              Print this help message.  (Options for controlling and monitoring the optimization procedure)

       <b>--lnl</b>, <b>-L</b>

              (for use with <b>--init-model</b>) Simply evaluate the  log  likelihood  of  the  specified  tree  model,
              without  performing any further optimization.  Can be used with <b>--post-probs</b>, <b>--expected-subs</b>, and
              <b>--expected-total-subs</b>.

       <b>--EM</b>, <b>-E</b>

              Fit model(s) using EM rather than the BFGS quasi-Newton algorithm (the default).

       <b>--precision</b>, <b>-p</b> HIGH|MED|LOW

              (default HIGH) Level of precision to use in  estimating  model  parameters.   Affects  convergence
              criteria  for  iterative  algorithms:  higher precision means more iterations and longer execution
              time.

       <b>--log</b>, <b>-l</b> &lt;log_fname&gt;

              Write log to &lt;log_fname&gt; describing details of the optimization procedure.

       <b>--init-model</b>, <b>-M</b> &lt;mod_fname&gt; Initialize with specified tree model.  By choosing good

              starting values for parameters, it is possible to reduce execution  time  dramatically.   If  this
              option  is  chosen, <b>--tree</b> is not allowed.  The substitution model used in the given model will be
              used unless <b>--subst-mod</b> is also specified.  Note: currently only one mod_fname may  be  specified;
              it will be used for all categories.

       <b>--init-random</b>, <b>-r</b> Initialize parameters randomly.  Can be used multiple times to test

              whether the m.l.e. is real.

       <b>--seed</b>, <b>-D</b> &lt;seed&gt;

              Provide  a  random  number seed for choosing initial parameter values (usually with <b>--init-random</b>,
              though random values are used in some other cases as well).  Should be an  integer  &gt;=1.   If  not
              provided, seed is chosen based on current time.

       <b>--init-parsimony</b>,  <b>-y</b>  Initialize  branch  lengths using parsimony counts for given data.  Only currently
              implemented for models with single character state (ie, not di- or tri-nucleotides).  Other <b>--init</b>
              options  such  as  <b>--init-random</b>  or  <b>--init-model</b>  can  be  used  in  conjunction  to  initialize
              substitution matrix parameters.

       <b>--print-parsimony</b>, <b>-Y</b> &lt;filename&gt;

       Print parsimony score to given file, and quit.
              (Does not optimize

              or report likelihoods).

       <b>--clock</b>,  <b>-z</b> Assume a molecular clock in estimation.  Causes the distances to all descendant leaves to be
              equal for each ancestral node and cuts the number of  free  branch-length  parameters  roughly  in
              half.

       <b>--scale-only</b>, <b>-B</b>

              (for  use  with  <b>--init-model</b>)  Estimate only the scale of the tree, rather than individual branch
              lengths (branch proportions fixed).  Equilibrium frequencies and rate-matrix parameters will still
              be estimated unless <b>--no-freqs</b> and <b>--no-rates</b> are used.

       <b>--scale-subtree</b>, <b>-S</b> &lt;node_name&gt; (for use with <b>--scale-only</b>) Estimate separate scale factors  for  subtree
              beneath  identified node and rest of tree.  The branch leading to the subtree is included with the
              subtree.  If ":loss" or ":gain" is appended to &lt;node_name&gt;, subtree scale  is  constrained  to  be
              greater than or less than (respectively) scale for rest of tree.

       <b>--estimate-freqs</b>, <b>-F</b>

              Estimate  equilibrium  frequencies  by  maximum  likelihood, rather than approximating them by the
              relative frequencies in the data.  If using the SSREV model, this option implies <b>--sym-freqs</b>.

       <b>--sym-freqs</b>, <b>-W</b>

              Estimate equilibrium frequencies, assuming freq(A)=freq(T) and freq(C)=freq(G).  This  only  works
              for an alphabet ACGT (and possibly gap).  This option implies <b>--estimate-freqs</b>.

       <b>--no-freqs</b>, <b>-f</b>

              (for  use  with  <b>--init-model</b>) Do not estimate equilibrium frequencies; just use the ones from the
              given tree model.

       <b>--no-rates</b>, <b>-n</b>

              (for use with <b>--init-model</b>) Do not estimate rate-matrix parameters; just use  the  ones  from  the
              given tree model.

       <b>--ancestor</b>, <b>-A</b> &lt;seqname&gt; Treat specified sequence as the root of the tree.  The tree topology must define
              this  sequence  to  be a child of the root (in practice, the branch from the root to the specified
              sequence will be retained, but will be constrained to have length zero).

       <b>--error</b>, <b>-e</b> &lt;filename&gt;

              For each parameter, report estimate, variance, and 95%%  confidence  interval,  printed  to  given
              filename, one parameter per line.

       <b>--no-opt</b>,  <b>-O</b>  &lt;param_list&gt;  Hold  parameters  listed  in  comma-separated param_list constant at initial
              values.  This applies only to the "main" model, and not to any models defined with  the  <b>--alt-mod</b>
              option.   Param  list  can  contain  values  such  as  "branches" to hold branch lengths constant,
              "ratematrix", "backgd", or "ratevar" to hold entire rate matrix, equilibrium frequencies, or  rate
              variation   parameters  constant  (respectively).   There  are  also  substitution  model-specific
              parameters such as "kappa" (transition/transversion rate ratio).  Note: to hold  certain  branches
              constant,  but  optimize  others,  put an exclamation point in the newick-formatted tree after the
              branch lengths that should be held constant.  This can be useful for enforcing  a  star-phylogeny.
              However,  note  that the two branches coming from root of tree are treated as one.  So they should
              both be held constant, or not held constant.  This option does *not*  work  with  <b>--scale-only</b>  or
              <b>--clock</b>.

       <b>--bound</b>  &lt;param_name[lower_bound,upper_bound]&gt;  Set boundaries for parameter.  lower_bound or upper_bound
              may be empty string to keep default.  For example <b>--bound</b> gc_param[1,] will

              set the lower bound for gc_param to 1 (keeping upper bound at infinity), for  a  GC  model.   Only
              applies  to  parameters  for  model  in the "main" tree, but similar syntax can be used within the
              <b>--alt-mod</b> arguments.  Can be used multiple times to set boundaries for different parameters.

       <b>--selection</b> &lt;selection_param&gt;

              Use selection in the model (is also  implied  if  <b>--init-model</b>  is  used  and  contains  selection
              parameter).   Selection  scales  rate matrix entries by selection_param/(1-exp(<b>-selection-param</b>));
              this is done after rate matrix is scaled to set expected number of substitutions per unit time  to
              1.   If using codon models selection acts only on nonysynonymous mutations.  (Options for modeling
              rate variation)

       <b>--nrates</b>, <b>-k</b> &lt;nratecats&gt; (default 1).  Number of rate categories to use.  Specifying a

              value of greater than one causes the discrete gamma model for rate variation  to  be  used  (Yang,
              1994).

       <b>--alpha</b>,  <b>-a</b>  &lt;alpha&gt; (for use with <b>--nrates</b>).  Initial value for alpha, the shape parameter of the gamma
              distribution.  Default is 1.

       <b>--rate-constants</b>, <b>-K</b> &lt;rate_consts&gt;

              Use a non-parameteric mixture model for rates, instead of  assuming  a  gamma  distribution.   The
              argument &lt;rate_consts&gt; must be a comma-delimited list explicitly defining the rate constants to be
              used.  The "weight" (mixing proportion) associated with each rate constant will be estimated by EM
              (this option implies <b>--EM</b>).  If <b>--alpha</b> is used with this option, then the mixing proportions will
              be  initialized  to reflect a gamma distribution with the specified shape parameter.  (Options for
              separate handling of sites in different annotation categories)

       <b>--features</b>, <b>-g</b> &lt;fname&gt;

              Annotations file (GFF or BED  format)  describing  features  on  one  or  more  sequences  in  the
              alignment.   Together with a category map (see <b>--catmap</b>), will be taken to define site categories,
              and a separate model will be estimated for each category.  If no  category  map  is  specified,  a
              category  will  be  assumed  for  each  type of feature, and they will be numbered in the order of
              appearance of the features.  Features are assumed  to  use  the  coordinate  frame  of  the  first
              sequence in the alignment and should be non-overlapping (see 'refeature <b>--unique</b>').

       <b>--catmap</b>, <b>-c</b> &lt;fname&gt;|&lt;string&gt;

              (optionally  use with <b>--features</b>) Mapping of feature types to category numbers.  Can either give a
              filename or an "inline" description of a simple category map, e.g., <b>--catmap</b> "NCATS = 3 ; CDS 1-3"
              or <b>--catmap</b> "NCATS = 1 ; UTR 1".  Note that category 0 is reserved  for  "background"  (everything
              that is not described by a defined feature type).

       <b>--do-cats</b>, <b>-C</b> &lt;cat_list&gt;

              (optionally   use   with   <b>--features</b>)   Estimate   models   for  only  the  specified  categories
              (comma-delimited list categories, by name or numbera).  Default  is  to  fit  a  model  for  every
              category.

       <b>--reverse-groups</b>, <b>-R</b> &lt;tag&gt;

              (optionally  use with <b>--features</b>) Group features by &lt;tag&gt; (e.g., "transcript_id" or "exon_id") and
              reverse complement segments of the alignment  corresponding  to  groups  on  the  reverse  strand.
              Groups  must be non-overlapping (see refeature <b>--unique</b>).  Useful with categories corresponding to
              strand-specific phenomena (e.g., codon positions).  (Options  for  context-dependent  substitution
              models)

       <b>--markov</b>, <b>-N</b>

              (for  use with context-dependent substitutions models and not available with <b>--EM</b>.)  Assume Markov
              dependence of alignment columns, and compute the conditional probability of each column given  its
              N-1  predecessors  using the two-pass algorithm described by Siepel and Haussler (2004).  (Here, N
              is the "order" of the model, as defined by <b>--subst-mod</b>; e.g., N=1 for REV, N=2 for  U2S,  N=3  for
              U3S.)  The  alternative  (the  default)  is  simply  to work with joint probabilities of tuples of
              columns.  (You can ensure  that  these  tuples  are  non-overlapping  with  the  <b>--non-overlapping</b>
              option.)   The  use  of joint probabilities during parameter estimation allows the use of the <b>--EM</b>
              option and can be much faster; in addition, it appears to produce nearly equivalent estimates.  If
              desired, parameters can be estimated without <b>--markov</b>, and then the likelihood  can  be  evaluated
              using  <b>--lnl</b>  and  <b>--markov</b>  together.   This  gives  a  lower  bound  on  the  likelihood  of the
              Markov-dependent model.

       <b>--non-overlapping</b>, <b>-V</b> (for use with context-dependent substitution models; not compatible with  <b>--markov</b>,
              <b>--features</b>,  or  <b>--msa-format</b> SS) Avoid using overlapping tuples of sites in parameter estimation.
              If a dinucleotide model is selected, every other tuple will be considered,  and  if  a  nucleotide
              triplet  model is selected, every third tuple will be considered.  This option cannot be used with
              an alignment represented only by unordered sufficient statistics.   (Option  for  lineage-specific
              models)

       <b>--label-branches</b> branch1,branch2,branch3...:label

              Create a group of branches by giving a set of branches a single label.  The label should be a word
              which  does not contain special characters (in particular, no spaces, brackets, parentheses, pound
              signs, commas, or colons).  The label is for  use  with  <b>--alt-model</b>  option  below,  so  that  an
              alternate  model  can  be defined for a set of branches.  A branch is specified by the name of the
              node which is a descendant of that branch.

              For  example,  <b>--label-branches</b>  hg18,chimp,hg18-chimp:HC  will  apply  the  label  "HC"  to   the
              hg18,chimp,and    hg18-chimp   branches   in   the   following   tree:   (((hg18,chimp)hg18-chimp,
              (mouse,rat)mouse-rat)

              The same label could be defined without using <b>--label-branches</b> by specifying the tree  (either  on
              the command-line or within init-model) as follows: (((hg18 # HC, chimp #HC)#HC, (mouse,rat))

       <b>--label-subtree</b>  node[+]:label  Similar  to label-branches, except labels the entire subtree of the named
              node.  If the node name is followed by a "+" sign, then includes the branch leading up to the node
              in the subtree.

       <b>--alt-model</b>, <b>-d</b> &lt;label:(model|param_list)&gt; Create a lineage-specific substitution model  on  a  group  of
              branches.   The  group is defined by a label, which can be specified within the tree string (using
              the # sign notation), or by using the  <b>--label-branches</b>  or  <b>--label-subtree</b>  arguments.   If  the
              alt-model  applies  to  only  a single branch, labelling is not necessary and the name of the node
              descending from the branch can be used instead.  See <b>--label-branches</b> above for  more  details  on
              labelling  groups  of  branches.   If  a name of a substitution model (HKY85, REV, UNREST, etc) is
              given after the colon, then this model will be used for the  group  of  branches,  and  parameters
              relevant to the model will be estimated separately in this group.  This model may be different (or
              the  same)  as  the model used in the rest of the tree, but it must have the same number of states
              and be of the same order as the model used for the rest of the tree.

              Alternately, a list of parameter names can be given after the  colon.   In  this  case,  the  same
              substitution  model  will be used for the entire tree, but the parameters listed will be estimated
              separately in the specified group of branches.

              The parameter names are model-specific, and include "kappa" for HKY models, "alpha" for GC models,
              "ratematrix" to specify all rate-matrix  parameters  in  general  models,  and  "backgd"  for  the
              equilibrium  background frequencies.  The parameter names may optionally be followed by boundaries
              in square brackets to specify parameter bounds, as described in <b>--bound</b> option.

              The <b>--alt-model</b> argument may be used multiple times, if one wishes to  (for  example)  optimize  a
              parameter  independently  on  several  different  groups  of branches.  Example: phyloFit align.fa
              <b>--subst-mod</b> HKY85 \ <b>--tree</b> "(human, (mouse#MR, rat#MR)#MR, cow)"\ <b>--alt-model</b> "MR:kappa[0, 1]"

              will estimate the HKY85 parameter kappa separately on the mouse/rat subtree, and  constrain  kappa
              between  0 and 1.  The quotes are often necessary to prevent the square brakcets from being parsed
              by the shell.  The same model could be achieved with:

              phyloFit   align.fa   <b>--subst-mod</b>   HKY85   \   <b>--tree</b>   "(human,   (mouse,rat)mouse-rat,   cow)"\
              <b>--label-branches</b> mouse,rat,mouse-rat:MR \ <b>--alt-model</b> "MR:kappa[0,1]" or

              phyloFit   align.fa   <b>--subst-mod</b>   HKY85   \   <b>--tree</b>   "(human,  (mouse,rat)mouse-rat,  cow)"  \
              <b>--label-subtree</b> "mouse-rat+:MR" \ <b>--alt-model</b> "MR:kappa[0,1]"

              (Options for posterior probabilities)

       <b>--post-probs</b>, <b>-P</b>

              Output posterior probabilities of all bases at all ancestral nodes.  Output will be  to  auxiliary
              file(s) with suffix ".postprobs".

       <b>--expected-subs</b>, <b>-X</b>

              Output  posterior  expected number of substitutions on each branch at each site, summed across all
              types of substitutions.  Output will be to auxiliary file(s) with suffix ".expsub".

       <b>--expected-subs-col</b>, <b>-J</b>

              Output posterior expected number of substitutions of each type on  each  branch,  for  each  site.
              Output will be to auxiliary file(s) with suffix .expcolsub

       <b>--expected-total-subs</b>, <b>-Z</b>

              Output  posterior  expected number of substitutions of each type on each branch, summed across all
              sites.  Output will be to auxiliary file(s) with suffix ".exptotsub".

       <b>--column-probs</b>, <b>-U</b>

       (for use with <b>-init-model</b>; implies <b>--lnl</b>)
              Output a separate log

       probability for each type of column in the input.
              Output will

       be to a file with suffix ".colprobs".
              Values are log base 2.

              (Options for estimation in sliding window)

       <b>--windows</b>, <b>-w</b> &lt;size,shift&gt; Apply a sliding window to the alignment, and  fit  a  separate  tree  to  each
              window.   Arguments specify size of window and amount by which to shift it on each iteration, both
              in bases of the first sequence in the alignment (assumed to be the reference sequence).   Separate
              versions of all output files will be created for each window.

       <b>--windows-explicit</b>, <b>-v</b> &lt;window_coord_list&gt; Like <b>--windows</b>, except that all start and end coordinates must
              be explicitly specified.  Each successive pair of numbers is interpreted as defining the start and
              end   of   a   window.   Can  be  used  with  a  two-column  file  and  the  '*'  operator,  e.g.,
              <b>--windows-explicit</b> '*mycoords'.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       A. Siepel and D. Haussler.
              2004.  Phylogenetic estimation of context-dependent  substitution  rates  by  maximum  likelihood.
              Mol. Biol. Evol., 21:468-488.

       Z. Yang, N. Goldman, and A. Friday.
              1994.  Comparison  of  models  for  nucleotide  substution used in maximum likelihood phylogenetic
              estimation. Mol. Biol. Evol., 11:316-324.

       Z. Yang. 1994. Maximum likelihood phylogenetic estimation from
              DNA sequences with variable rates over sites: approximate methods. J. Mol. Evol., 39:306-314.

phyloFit 1.4                                        May 2016                                         <u><a href="../man1/PHYLOFIT.1.html">PHYLOFIT</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>