<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>sreformat - convert sequence file to different format</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/biosquid">biosquid_1.9g+cvs20050121-15.1build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       sreformat - convert sequence file to different format

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>sreformat</b> <u>[options]</u> <u>format</u> <u>seqfile</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>sreformat</b>  reads  the  sequence  file  <u>seqfile</u>  in  any  supported format, reformats it into a new format
       specified by <u>format,</u> then prints the reformatted text.

       Supported input formats include (but are not limited to) the  unaligned  formats  FASTA,  Genbank,  EMBL,
       SWISS-PROT, PIR, and GCG, and the aligned formats Stockholm, Clustal, GCG MSF, and Phylip.

       Available  unaligned output file format codes include <u>fasta</u> (FASTA format); <u>embl</u> (EMBL/SWISSPROT format);
       <u>genbank</u> (Genbank format); <u>gcg</u> (GCG single sequence format); <u>gcgdata</u> (GCG flatfile database  format);  <u>pir</u>
       (PIR/CODATA flatfile format); <u>raw</u> (raw sequence, no other information).

       The  available  aligned  output file format codes include <u>stockholm</u> (PFAM/Stockholm format); <u>msf</u> (GCG MSF
       format); <u>a2m</u> (an aligned FASTA format); <u>PHYLIP</u> (Felsenstein's PHYLIP format); and <u>clustal</u> (Clustal  V/W/X
       format); and <u>selex</u> (the old SELEX/HMMER/Pfam annotated alignment format);

       All thee codes are interpreted case-insensitively (e.g. MSF, Msf, or msf all work).

       Unaligned  format  files  cannot  be  reformatted  to  aligned  formats.  However, aligned formats can be
       reformatted to unaligned formats -- gap characters are simply stripped out.

       This program was originally named <b>reformat,</b> but that name clashes with a GCG program of the same name.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-d</b>     DNA; convert U's to T's, to make sure a nucleic acid sequence is shown as DNA not RNA. See <b>-r.</b>

       <b>-h</b>     Print brief help; includes version number and summary of all options, including expert options.

       <b>-l</b>     Lowercase; convert all sequence residues to lower case.  See <b>-u.</b>

       <b>-n</b>     For DNA/RNA sequences, converts any character that's not unambiguous RNA/DNA (e.g. ACGTU/acgtu) to
              an N. Used to convert IUPAC ambiguity codes to N's, for software that can't handle all IUPAC codes
              (some public RNA folding codes, for example). If the file is an alignment, gap characters are also
              left unchanged. If sequences are not nucleic acid sequences, this option will corrupt the data  in
              a predictable fashion.

       <b>-r</b>     RNA; convert T's to U's, to make sure a nucleic acid sequence is shown as RNA not DNA. See <b>-d.</b>

       <b>-u</b>     Uppercase; convert all sequence residues to upper case.  See <b>-l.</b>

       <b>-x</b>     For  DNA  sequences, convert non-IUPAC characters (such as X's) to N's.  This is for compatibility
              with benighted people who insist on using X instead of the IUPAC ambiguity character N. (X is  for
              ambiguity in an amino acid residue).

              Warning: like the <b>-n</b> option, the code doesn't check that you are actually giving it DNA. It simply
              literally  just  converts  non-IUPAC  DNA  symbols  to  N.  So if you accidentally give it protein
              sequence, it will happily convert most every amino acid residue to an N.

</pre><h4><b>EXPERT</b> <b>OPTIONS</b></h4><pre>
       <b>--gapsym</b> <u>&lt;c&gt;</u>
              Convert all gap characters to <u>&lt;c&gt;.</u>  Used to prepare  alignment  files  for  programs  with  strict
              requirements for gap symbols. Only makes sense if the input <u>seqfile</u> is an alignment.

       <b>--informat</b> <u>&lt;s&gt;</u>
              Specify  that  the  sequence file is in format <u>&lt;s&gt;,</u> rather than allowing the program to autodetect
              the file format. Common examples include Genbank, EMBL, GCG,  PIR,  Stockholm,  Clustal,  MSF,  or
              PHYLIP; see the printed documentation for a complete list of accepted format names.

       <b>--mingap</b>
              If  <u>seqfile</u>  is  an alignment, remove any columns that contain 100% gap characters, minimizing the
              overall length of the alignment.  (Often useful if you've extracted a subset of aligned  sequences
              from a larger alignment.)

       <b>--nogap</b>
              Remove  any  aligned  columns  that  contain  any  gap  symbols  at  all.  Useful  as a prelude to
              phylogenetic analyses, where you only want to analyze columns containing  100%  residues,  so  you
              want  to  strip  out  any columns with gaps in them.  Only makes sense if the file is an alignment
              file.

       <b>--pfam</b> For SELEX alignment output format only, put the entire alignment in one  block  (don't  wrap  into
              multiple blocks).  This is close to the format used internally by Pfam in Stockholm and Cambridge.

       <b>--sam</b>  Try  to  convert  gap  characters  to UC Santa Cruz SAM style, where a .  means a gap in an insert
              column, and a - means a deletion in a consensus/match  column.  This  only  works  for  converting
              aligned  file  formats,  and  only if the alignment already adheres to the SAM convention of upper
              case for residues in consensus/match columns, and lower case for residues in insert columns.  This
              is  true,  for  instance,  of  all  alignments produced by old versions of HMMER. (HMMER2 produces
              alignments that adhere to SAM's conventions even in gap character choice.)  This option was  added
              to  allow  Pfam  alignments  to  be  reformatted  into  something  more  suitable  for profile HMM
              construction using the UCSC SAM software.

       <b>--samfrac</b> <u>&lt;x&gt;</u>
              Try to convert the alignment gap characters and residue cases to UC Santa Cruz SAM style, where  a
              .  means a gap in an insert column and a - means a deletion in a consensus/match column, and upper
              case  means  match/consensus  residues and lower case means inserted resiudes. This will only work
              for converting aligned file formats, but unlike the <b>--sam</b>  option,  it  will  work  regardless  of
              whether  the  file  adheres  to  the  upper/lower  case  residue  convention.  Instead, any column
              containing more than a fraction <u>&lt;x&gt;</u> of gap characters is interpreted as an insert column, and  all
              other columns are interpreted as match columns.  This option was added to allow Pfam alignments to
              be  reformatted  into  something  more  suitable  for  profile HMM construction using the UCSC SAM
              software.

       <b>--wussify</b>
              Convert RNA secondary structure annotation strings (both consensus and individual) from old  "KHS"
              format,  &gt;&lt;,  to the new WUSS notation, &lt;&gt;. If the notation is already in WUSS format, this option
              will screw it up, without warning. Only SELEX and Stockholm format files have secondary  structure
              markup at present.

       <b>--dewuss</b>
              Convert RNA secondary structure annotation strings from the new WUSS notation, &lt;&gt;, back to the old
              KHS format, &gt;&lt;. If the annotation is already in KHS, this option will corrupt it, without warning.
              Only SELEX and Stockholm format files have secondary structure markup.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/afetch.1.html">afetch</a></b>(1),  <b><a href="../man1/alistat.1.html">alistat</a></b>(1),  <b><a href="../man1/compalign.1.html">compalign</a></b>(1),  <b><a href="../man1/compstruct.1.html">compstruct</a></b>(1),  <b><a href="../man1/revcomp.1.html">revcomp</a></b>(1),  <b><a href="../man1/seqsplit.1.html">seqsplit</a></b>(1),  <b><a href="../man1/seqstat.1.html">seqstat</a></b>(1), <b><a href="../man1/sfetch.1.html">sfetch</a></b>(1),
       <b><a href="../man1/shuffle.1.html">shuffle</a></b>(1), <b><a href="../man1/sindex.1.html">sindex</a></b>(1), <b><a href="../man1/stranslate.1.html">stranslate</a></b>(1), <b><a href="../man1/weight.1.html">weight</a></b>(1).

</pre><h4><b>AUTHOR</b></h4><pre>
       Biosquid and its documentation are Copyright (C) 1992-2003 HHMI/Washington University School of  Medicine
       Freely distributed under the GNU General Public License (GPL) See COPYING in the source code distribution
       for more details, or contact me.

       Sean Eddy
       HHMI/Department of Genetics
       Washington University School of Medicine
       4444 Forest Park Blvd., Box 8510
       St Louis, MO 63108 USA
       Phone: 1-314-362-7666
       FAX  : 1-314-362-2157
       Email: <a href="mailto:eddy@genetics.wustl.edu">eddy@genetics.wustl.edu</a>

Biosquid 1.9g                                     January 2003                                      <u><a href="../man1/sreformat.1.html">sreformat</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>