<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>vcftools - Utilities for the variant call format (VCF) and binary variant call format (BCF)</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/vcftools">vcftools_0.1.16-3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       vcftools - Utilities for the variant call format (VCF) and binary variant call format (BCF)

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>vcftools</b>  [  <b>--vcf</b>  FILE  |  <b>--gzvcf</b>  FILE | <b>--bcf</b> FILE] [ <b>--out</b> OUTPUT PREFIX ] [ FILTERING OPTIONS ]  [
       OUTPUT OPTIONS ]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       vcftools is a suite of functions for use on genetic variation data in the form of VCF and BCF files.  The
       tools  provided  will  be  used  mainly to summarize data, run calculations on data, filter out data, and
       convert data into other useful file formats.

</pre><h4><b>EXAMPLES</b></h4><pre>
       Output allele frequency for all sites in the input vcf file from chromosome 1
         <b>vcftools</b> --gzvcf input_file.vcf.gz --freq --chr 1 --out chr1_analysis

       Output a new vcf file from the input vcf file that removes any indel sites
         <b>vcftools</b> --vcf input_file.vcf --remove-indels --recode --recode-INFO-all --out SNPs_only

       Output file comparing the sites in two vcf files
         <b>vcftools</b> --gzvcf input_file1.vcf.gz --gzdiff input_file2.vcf.gz --diff-site --out in1_v_in2

       Output a new vcf file to standard out without any sites that have a filter tag,  then  compress  it  with
       gzip
         <b>vcftools</b>   --gzvcf   input_file.vcf.gz   --remove-filtered-all   --recode   --stdout   |   gzip   -c  &gt;
         output_PASS_only.vcf.gz

       Output a Hardy-Weinberg p-value for every site in the bcf file that does not have any missing genotypes
         <b>vcftools</b> --bcf input_file.bcf --hardy --max-missing 1.0 --out output_noMissing

       Output nucleotide diversity at a list of positions
         zcat input_file.vcf.gz | <b>vcftools</b> --vcf - --site-pi --positions SNP_list.txt --out nucleotide_diversity

</pre><h4><b>BASIC</b> <b>OPTIONS</b></h4><pre>
       These options are used to specify the input and output files.

   <b>INPUT</b> <b>FILE</b> <b>OPTIONS</b>
         <b>--vcf</b> <u>&lt;input_filename&gt;</u>
           This option defines the VCF file to be processed. VCFtools expects files in VCF format v4.0, v4.1  or
           v4.2. The latter two are supported with some small limitations. If the user provides a dash character
           '-' as a file name, the program expects a VCF file to be piped in through standard in.

         <b>--gzvcf</b> <u>&lt;input_filename&gt;</u>
           This option can be used in place of the --vcf option to read compressed (gzipped) VCF files directly.

         <b>--bcf</b> <u>&lt;input_filename&gt;</u>
           This  option can be used in place of the --vcf option to read BCF2 files directly. You do not need to
           specify if this file is compressed with BGZF encoding. If the user provides a dash character '-' as a
           file name, the program expects a BCF2 file to be piped in through standard in.

   <b>OUTPUT</b> <b>FILE</b> <b>OPTIONS</b>
         <b>--out</b> <u>&lt;output_prefix&gt;</u>
           This option defines the output filename prefix for all files generated by vcftools. For  example,  if
           &lt;prefix&gt;  is  set to output_filename, then all output files will be of the form output_filename.*** .
           If this option is omitted, all output files will have  the  prefix  "out."  in  the  current  working
           directory.

         <b>--stdout</b>
         <b>-c</b>
           These  options  direct the vcftools output to standard out so it can be piped into another program or
           written directly to a filename of choice. However, a select few output functions cannot be written to
           standard out.

         <b>--temp</b> <u>&lt;temporary_directory&gt;</u>
           This option can be used to redirect any temporary  files  that  vcftools  creates  into  a  specified
           directory.

</pre><h4><b>SITE</b> <b>FILTERING</b> <b>OPTIONS</b></h4><pre>
       These  options  are  used  to  include  or exclude certain sites from any analysis being performed by the
       program.

   <b>POSITION</b> <b>FILTERING</b>
         <b>--chr</b> <u>&lt;chromosome&gt;</u>
         <b>--not-chr</b> <u>&lt;chromosome&gt;</u>
           Includes or excludes sites with  indentifiers  matching  &lt;chromosome&gt;.  These  options  may  be  used
           multiple times to include or exclude more than one chromosome.

         <b>--from-bp</b> <u>&lt;integer&gt;</u>
         <b>--to-bp</b> <u>&lt;integer&gt;</u>
           These  options specify a lower bound and upper bound for a range of sites to be processed. Sites with
           positions less than or greater than these values will be excluded. These options can only be used  in
           conjunction with a single usage of --chr. Using one of these does not require use of the other.

         <b>--positions</b> <u>&lt;filename&gt;</u>
         <b>--exclude-positions</b> <u>&lt;filename&gt;</u>
           Include  or  exclude  a  set of sites on the basis of a list of positions in a file. Each line of the
           input file should contain a (tab-separated) chromosome and position. The file can have comment  lines
           that start with a "#", they will be ignored.

         <b>--positions-overlap</b> <u>&lt;filename&gt;</u>
         <b>--exclude-positions-overlap</b> <u>&lt;filename&gt;</u>
           Include  or  exclude  a  set of sites on the basis of the reference allele overlapping with a list of
           positions in a file. Each line of the input file should  contain  a  (tab-separated)  chromosome  and
           position. The file can have comment lines that start with a "#", they will be ignored.

         <b>--bed</b> <u>&lt;filename&gt;</u>
         <b>--exclude-bed</b> <u>&lt;filename&gt;</u>
           Include  or  exclude  a set of sites on the basis of a BED file. Only the first three columns (chrom,
           chromStart and chromEnd) are required. The BED file is expected to have a header line. A site will be
           kept or excluded if any part of any allele (REF or ALT) at a site is within the range of one  of  the
           BED entries.

         <b>--thin</b> <u>&lt;integer&gt;</u>
           Thin sites so that no two sites are within the specified distance from one another.

         <b>--mask</b> <u>&lt;filename&gt;</u>
         <b>--invert-mask</b> <u>&lt;filename&gt;</u>
         <b>--mask-min</b> <u>&lt;integer&gt;</u>
           These  options  are  used  to specify a FASTA-like mask file to filter with. The mask file contains a
           sequence of integer digits (between 0 and 9) for each position on a chromosome that specify if a site
           at that position should be filtered or not.
           An example mask file would look like:
             <u>&gt;1</u>
             <u>0000011111222...</u>
             <u>&gt;2</u>
             <u>2222211111000...</u>
           In this example, sites in the VCF file located within the first 5 bases of the start of chromosome  1
           would  be  kept,  whereas sites at position 6 onwards would be filtered out. And sites after the 11th
           position on chromosome 2 would be filtered out as well.
           The "--invert-mask" option takes the same format mask file as the "--mask" option, however it inverts
           the mask file before filtering with it.
           And the "--mask-min" option specifies a threshold mask value between 0 and 9 to filter positions  by.
           The default threshold is 0, meaning only sites with that value or lower will be kept.

   <b>SITE</b> <b>ID</b> <b>FILTERING</b>
         <b>--snp</b> <u>&lt;string&gt;</u>
           Include SNP(s) with matching ID (e.g. a dbSNP rsID). This command can be used multiple times in order
           to include more than one SNP.

         <b>--snps</b> <u>&lt;filename&gt;</u>
         <b>--exclude</b> <u>&lt;filename&gt;</u>
           Include  or  exclude  a list of SNPs given in a file. The file should contain a list of SNP IDs (e.g.
           dbSNP rsIDs), with one ID per line. No header line is expected.

   <b>VARIANT</b> <b>TYPE</b> <b>FILTERING</b>
         <b>--keep-only-indels</b>
         <b>--remove-indels</b>
           Include or exclude sites that contain an indel. For these options  "indel"  means  any  variant  that
           alters the length of the REF allele.

   <b>FILTER</b> <b>FLAG</b> <b>FILTERING</b>
         <b>--remove-filtered-all</b>
           Removes all sites with a FILTER flag other than PASS.

         <b>--keep-filtered</b> <u>&lt;string&gt;</u>
         <b>--remove-filtered</b> <u>&lt;string&gt;</u>
           Includes  or  excludes  all  sites marked with a specific FILTER flag. These options may be used more
           than once to specify multiple FILTER flags.

   <b>INFO</b> <b>FIELD</b> <b>FILTERING</b>
         <b>--keep-INFO</b> <u>&lt;string&gt;</u>
         <b>--remove-INFO</b> <u>&lt;string&gt;</u>
           Includes or excludes all sites with a specific INFO flag. These options only filter on  the  presence
           of  the  flag  and  not  its value. These options can be used multiple times to specify multiple INFO
           flags.

   <b>ALLELE</b> <b>FILTERING</b>
         <b>--maf</b> <u>&lt;float&gt;</u>
         <b>--max-maf</b> <u>&lt;float&gt;</u>
           Include only sites with a Minor Allele Frequency greater than or equal to the "--maf" value and  less
           than  or  equal  to the "--max-maf" value. One of these options may be used without the other. Allele
           frequency is defined as the number of times an allele appears over  all  individuals  at  that  site,
           divided by the total number of non-missing alleles at that site.

         <b>--non-ref-af</b> <u>&lt;float&gt;</u>
         <b>--max-non-ref-af</b> <u>&lt;float&gt;</u>
         <b>--non-ref-ac</b> <u>&lt;integer&gt;</u>
         <b>--max-non-ref-ac</b> <u>&lt;integer&gt;</u>

         <b>--non-ref-af-any</b> <u>&lt;float&gt;</u>
         <b>--max-non-ref-af-any</b> <u>&lt;float&gt;</u>
         <b>--non-ref-ac-any</b> <u>&lt;integer&gt;</u>
         <b>--max-non-ref-ac-any</b> <u>&lt;integer&gt;</u>
           Include  only  sites  with  all Non-Reference (ALT) Allele Frequencies (af) or Counts (ac) within the
           range specified, and including the specified value. The default options require <b>all</b> alleles  to  meet
           the  specified  criteria, whereas the options appended with "any" require only one allele to meet the
           criteria. The Allele frequency is defined  as  the  number  of  times  an  allele  appears  over  all
           individuals at that site, divided by the total number of non-missing alleles at that site.

         <b>--mac</b> <u>&lt;integer&gt;</u>
         <b>--max-mac</b> <u>&lt;integer&gt;</u>
           Include  only  sites with Minor Allele Count greater than or equal to the "--mac" value and less than
           or equal to the "--max-mac" value. One of these options may be used without the other.  Allele  count
           is simply the number of times that allele appears over all individuals at that site.

         <b>--min-alleles</b> <u>&lt;integer&gt;</u>
         <b>--max-alleles</b> <u>&lt;integer&gt;</u>
           Include  only  sites  with a number of alleles greater than or equal to the "--min-alleles" value and
           less than or equal to the "--max-alleles" value. One of these options may be used without the other.
           For example, to include only bi-allelic sites, one could use:
             <b>vcftools</b> --vcf file1.vcf --min-alleles 2 --max-alleles 2

   <b>GENOTYPE</b> <b>VALUE</b> <b>FILTERING</b>
         <b>--min-meanDP</b> <u>&lt;float&gt;</u>
         <b>--max-meanDP</b> <u>&lt;float&gt;</u>
           Includes only sites with mean depth values (over all included individuals) greater than or  equal  to
           the "--min-meanDP" value and less than or equal to the "--max-meanDP" value. One of these options may
           be used without the other. These options require that the "DP" FORMAT tag is included for each site.

         <b>--hwe</b> <u>&lt;float&gt;</u>
           Assesses  sites  for  Hardy-Weinberg Equilibrium using an exact test, as defined by Wigginton, Cutler
           and Abecasis (2005). Sites with a p-value below the threshold defined by this option are taken to  be
           out of HWE, and therefore excluded.

         <b>--max-missing</b> <u>&lt;float&gt;</u>
           Exclude  sites on the basis of the proportion of missing data (defined to be between 0 and 1, where 0
           allows sites that are completely missing and 1 indicates no missing data allowed).

         <b>--max-missing-count</b> <u>&lt;integer&gt;</u>
           Exclude sites with more than this number of missing genotypes over all individuals.

         <b>--phased</b>
           Excludes all sites that contain unphased genotypes.

   <b>MISCELLANEOUS</b> <b>FILTERING</b>
         <b>--minQ</b> <u>&lt;float&gt;</u>
           Includes only sites with Quality value above this threshold.

</pre><h4><b>INDIVIDUAL</b> <b>FILTERING</b> <b>OPTIONS</b></h4><pre>
       These options are used to include or exclude certain individuals from any analysis being performed by the
       program.
         <b>--indv</b> <u>&lt;string&gt;</u>
         <b>--remove-indv</b> <u>&lt;string&gt;</u>
           Specify an individual to be kept or removed from the analysis. This option can be used multiple times
           to specify multiple individuals. If both options are specified, then the "--indv" option is  executed
           before the "--remove-indv option".

         <b>--keep</b> <u>&lt;filename&gt;</u>
         <b>--remove</b> <u>&lt;filename&gt;</u>
           Provide  files  containing a list of individuals to either include or exclude in subsequent analysis.
           Each individual ID (as defined in the VCF headerline) should be included on a separate line. If  both
           options  are  used,  then the "--keep" option is executed before the "--remove" option. When multiple
           files are provided, the union of  individuals  from  all  keep  files  subtracted  by  the  union  of
           individuals from all remove files are kept. No header line is expected.

         <b>--max-indv</b> <u>&lt;integer&gt;</u>
           Randomly thins individuals so that only the specified number are retained.

</pre><h4><b>GENOTYPE</b> <b>FILTERING</b> <b>OPTIONS</b></h4><pre>
       These  options  are  used  to  exclude  genotypes  from  any  analysis being performed by the program. If
       excluded, these values will be treated as missing.
         <b>--remove-filtered-geno-all</b>
           Excludes all genotypes with a FILTER flag not equal to "." (a missing value) or PASS.

         <b>--remove-filtered-geno</b> <u>&lt;string&gt;</u>
           Excludes genotypes with a specific FILTER flag.

         <b>--minGQ</b> <u>&lt;float&gt;</u>
           Exclude all genotypes with a quality below the threshold specified. This  option  requires  that  the
           "GQ" FORMAT tag is specified for all sites.

         <b>--minDP</b> <u>&lt;float&gt;</u>
         <b>--maxDP</b> <u>&lt;float&gt;</u>
           Includes  only  genotypes  greater than or equal to the "--minDP" value and less than or equal to the
           "--maxDP" value. This option requires that the "DP" FORMAT tag is specified for all sites.

</pre><h4><b>OUTPUT</b> <b>OPTIONS</b></h4><pre>
       These options specify which analyses or conversions to perform  on  the  data  that  passed  through  all
       specified filters.

   <b>OUTPUT</b> <b>ALLELE</b> <b>STATISTICS</b>
         <b>--freq</b>
         <b>--freq2</b>
           Outputs  the  allele  frequency  for each site in a file with the suffix ".frq". The second option is
           used to suppress output of any information about the alleles.

         <b>--counts</b>
         <b>--counts2</b>
           Outputs the raw allele counts for each site in a file with the suffix ".frq.count". The second option
           is used to suppress output of any information about the alleles.

         <b>--derived</b>
           For use with the previous four frequency and count options only. Re-orders the output file columns so
           that the ancestral allele appears first. This option relies on the ancestral allele  being  specified
           in the VCF file using the AA tag in the INFO field.

   <b>OUTPUT</b> <b>DEPTH</b> <b>STATISTICS</b>
         <b>--depth</b>
           Generates a file containing the mean depth per individual. This file has the suffix ".idepth".

         <b>--site-depth</b>
           Generates  a  file  containing the depth per site summed across all individuals. This output file has
           the suffix ".ldepth".

         <b>--site-mean-depth</b>
           Generates a file containing the mean depth per site averaged across all individuals. This output file
           has the suffix ".ldepth.mean".

         <b>--geno-depth</b>
           Generates a (possibly very large) file containing the depth  for  each  genotype  in  the  VCF  file.
           Missing entries are given the value -1. The file has the suffix ".gdepth".

   <b>OUTPUT</b> <b>LD</b> <b>STATISTICS</b>
         <b>--hap-r2</b>
           Outputs  a  file  reporting  the  r2,  D,  and  D'  statistics using phased haplotypes. These are the
           traditional measures of LD often reported in the population genetics literature. The output file  has
           the suffix ".hap.ld". This option assumes that the VCF input file has phased haplotypes.

         <b>--geno-r2</b>
           Calculates  the  squared correlation coefficient between genotypes encoded as 0, 1 and 2 to represent
           the number of non-reference alleles in each individual. This is the same as the LD  measure  reported
           by  PLINK.  The  D and D' statistics are only available for phased genotypes. The output file has the
           suffix ".geno.ld".

         <b>--geno-chisq</b>
           If your data contains sites with more than two alleles, then this option can  be  used  to  test  for
           genotype independence via the chi-squared statistic. The output file has the suffix ".geno.chisq".

         <b>--hap-r2-positions</b> <u>&lt;positions</u> <u>list</u> <u>file&gt;</u>
         <b>--geno-r2-positions</b> <u>&lt;positions</u> <u>list</u> <u>file&gt;</u>
           Outputs  a  file  reporting  the r2 statistics of the sites contained in the provided file verses all
           other sites. The output files have the suffix ".list.hap.ld" or ".list.geno.ld", depending  on  which
           option is used.

         <b>--ld-window</b> <u>&lt;integer&gt;</u>
           This  optional  parameter  defines the maximum number of SNPs between the SNPs being tested for LD in
           the "--hap-r2", "--geno-r2", and "--geno-chisq" functions.

         <b>--ld-window-bp</b> <u>&lt;integer&gt;</u>
           This optional parameter defines the maximum number of physical bases between the  SNPs  being  tested
           for LD in the "--hap-r2", "--geno-r2", and "--geno-chisq" functions.

         <b>--ld-window-min</b> <u>&lt;integer&gt;</u>
           This  optional  parameter  defines the minimum number of SNPs between the SNPs being tested for LD in
           the "--hap-r2", "--geno-r2", and "--geno-chisq" functions.

         <b>--ld-window-bp-min</b> <u>&lt;integer&gt;</u>
           This optional parameter defines the minimum number of physical bases between the  SNPs  being  tested
           for LD in the "--hap-r2", "--geno-r2", and "--geno-chisq" functions.

         <b>--min-r2</b> <u>&lt;float&gt;</u>
           This  optional parameter sets a minimum value for r2, below which the LD statistic is not reported by
           the "--hap-r2", "--geno-r2", and "--geno-chisq" functions.

         <b>--interchrom-hap-r2</b>
         <b>--interchrom-geno-r2</b>
           Outputs a file reporting the r2 statistics for sites on different chromosomes. The output files  have
           the suffix ".interchrom.hap.ld" or ".interchrom.geno.ld", depending on the option used.

   <b>OUTPUT</b> <b>TRANSITION/TRANSVERSION</b> <b>STATISTICS</b>
         <b>--TsTv</b> <u>&lt;integer&gt;</u>
           Calculates  the Transition / Transversion ratio in bins of size defined by this option. Only uses bi-
           allelic SNPs. The resulting output file has the suffix ".TsTv".

         <b>--TsTv-summary</b>
           Calculates a simple summary of all Transitions and Transversions. The  output  file  has  the  suffix
           ".TsTv.summary".

         <b>--TsTv-by-count</b>
           Calculates  the  Transition / Transversion ratio as a function of alternative allele count. Only uses
           bi-allelic SNPs. The resulting output file has the suffix ".TsTv.count".

         <b>--TsTv-by-qual</b>
           Calculates the Transition / Transversion ratio as a function of SNP quality threshold. Only uses  bi-
           allelic SNPs. The resulting output file has the suffix ".TsTv.qual".

         <b>--FILTER-summary</b>
           Generates  a  summary of the number of SNPs and Ts/Tv ratio for each FILTER category. The output file
           has the suffix ".FILTER.summary".

   <b>OUTPUT</b> <b>NUCLEOTIDE</b> <b>DIVERGENCE</b> <b>STATISTICS</b>
         <b>--site-pi</b>
           Measures nucleotide divergency on a per-site basis. The output file has the suffix ".sites.pi".

         <b>--window-pi</b> <u>&lt;integer&gt;</u>
         <b>--window-pi-step</b> <u>&lt;integer&gt;</u>
           Measures the nucleotide diversity in windows, with the number provided as the window size. The output
           file has the suffix ".windowed.pi". The latter is an optional argument used to specify the step  size
           in between windows.

   <b>OUTPUT</b> <b>FST</b> <b>STATISTICS</b>
         <b>--weir-fst-pop</b> <u>&lt;filename&gt;</u>
           This  option  is  used to calculate an Fst estimate from Weir and Cockerham's 1984 paper. This is the
           preferred calculation of Fst. The provided file must contain a list of  individuals  (one  individual
           per line) from the VCF file that correspond to one population. This option can be used multiple times
           to  calculate  Fst  for  more  than  two  populations.  These files will also be included as "--keep"
           options. By default, calculations are done on a per-site  basis.  The  output  file  has  the  suffix
           ".weir.fst".

         <b>--fst-window-size</b> <u>&lt;integer&gt;</u>
         <b>--fst-window-step</b> <u>&lt;integer&gt;</u>
           These  options  can  be  used  with  "--weir-fst-pop"  to do the Fst calculations on a windowed basis
           instead of a per-site basis. These arguments specify the desired window size  and  the  desired  step
           size between windows.

   <b>OUTPUT</b> <b>OTHER</b> <b>STATISTICS</b>
         <b>--het</b>
           Calculates  a  measure  of  heterozygosity  on  a  per-individual  basis. Specfically, the inbreeding
           coefficient, F, is estimated for each individual using a method of moments. The  resulting  file  has
           the suffix ".het".

         <b>--hardy</b>
           Reports  a  p-value  for  each  site from a Hardy-Weinberg Equilibrium test (as defined by Wigginton,
           Cutler and Abecasis (2005)). The resulting file (with  suffix  ".hwe")  also  contains  the  Observed
           numbers of Homozygotes and Heterozygotes and the corresponding Expected numbers under HWE.

         <b>--TajimaD</b> <u>&lt;integer&gt;</u>
           Outputs  Tajima's  D  statistic  in  bins  with size of the specified number. The output file has the
           suffix ".Tajima.D".

         <b>--indv-freq-burden</b>
           This option calculates the number of variants within each individual of  a  specific  frequency.  The
           resulting file has the suffix ".ifreqburden".

         <b>--LROH</b>
           This  option  will  identify  and  output  Long  Runs of Homozygosity. The output file has the suffix
           ".LROH". This function is experimental, and will use a lot of memory if applied to large datasets.

         <b>--relatedness</b>
           This option is used to calculate and output a relatedness statistic based on the method  of  Yang  et
           al,  Nature Genetics 2010 (doi:10.1038/ng.608). Specifically, calculate the unadjusted Ajk statistic.
           Expectation of Ajk is zero for individuals within a populations,  and  one  for  an  individual  with
           themselves. The output file has the suffix ".relatedness".

         <b>--relatedness2</b>
           This  option  is  used  to  calculate  and  output  a  relatedness  statistic  based on the method of
           Manichaikul et al., BIOINFORMATICS 2010 (doi:10.1093/bioinformatics/btq559). The output file has  the
           suffix ".relatedness2".

         <b>--site-quality</b>
           Generates  a  file  containing the per-site SNP quality, as found in the QUAL column of the VCF file.
           This file has the suffix ".lqual".

         <b>--missing-indv</b>
           Generates a file reporting the missingness on  a  per-individual  basis.  The  file  has  the  suffix
           ".imiss".

         <b>--missing-site</b>
           Generates a file reporting the missingness on a per-site basis. The file has the suffix ".lmiss".

         <b>--SNPdensity</b> <u>&lt;integer&gt;</u>
           Calculates  the  number  and  density  of  SNPs in bins of size defined by this option. The resulting
           output file has the suffix ".snpden".

         <b>--kept-sites</b>
           Creates a file listing all sites that have been  kept  after  filtering.  The  file  has  the  suffix
           ".kept.sites".

         <b>--removed-sites</b>
           Creates  a  file  listing  all  sites that have been removed after filtering. The file has the suffix
           ".removed.sites".

         <b>--singletons</b>
           This option will generate a file detailing the location of singletons, and the individual they  occur
           in.  The  file reports both true singletons, and private doubletons (i.e. SNPs where the minor allele
           only occurs in a single individual and that individual is homozygotic for that  allele).  The  output
           file has the suffix ".singletons".

         <b>--hist-indel-len</b>
           This  option  will  generate  a histogram file of the length of all indels (including SNPs). It shows
           both the count and the percentage of all indels for indel lengths that occur at  least  once  in  the
           input  file.  SNPs  are  considered  indels  with  length  zero.  The  output  file  has  the  suffix
           ".indel.hist".

         <b>--hapcount</b> <u>&lt;BED</u> <u>file&gt;</u>
           This option will output the number of unique haplotypes within user specified bins, as defined by the
           BED file. The output file has the suffix ".hapcount".

         <b>--mendel</b> <u>&lt;PED</u> <u>file&gt;</u>
           This option is use to report mendel errors identified in trios. The command  requires  a  PLINK-style
           PED  file,  with  the first four columns specifying a family ID, the child ID, the father ID, and the
           mother ID. The output of this command has the suffix ".mendel".

         <b>--extract-FORMAT-info</b> <u>&lt;string&gt;</u>
           Extract information from the genotype  fields  in  the  VCF  file  relating  to  a  specified  FORMAT
           identifier.  The  resulting  output  file  has  the  suffix  ".&lt;FORMAT_ID&gt;.FORMAT".  For example, the
           following command would extract the all of the GT (i.e. Genotype) entries:
             <b>vcftools</b> --vcf file1.vcf --extract-FORMAT-info GT

         <b>--get-INFO</b> <u>&lt;string&gt;</u>
           This option is used to extract information from the INFO field in the VCF file. The &lt;string&gt; argument
           specifies the INFO tag to be extracted, and the option can be used multiple times in order to extract
           multiple INFO  entries.  The  resulting  file,  with  suffix  ".INFO",  contains  the  required  INFO
           information  in a tab-separated table. For example, to extract the NS and DB flags, one would use the
           command:
             <b>vcftools</b> --vcf file1.vcf --get-INFO NS --get-INFO DB

   <b>OUTPUT</b> <b>VCF</b> <b>FORMAT</b>
         <b>--recode</b>
         <b>--recode-bcf</b>
           These options are used to generate a new file in either VCF or BCF from the input  VCF  or  BCF  file
           after  applying  the  filtering  options  specified  by  the  user.  The  output  file has the suffix
           ".recode.vcf" or ".recode.bcf". By default, the INFO fields are removed from the output file, as  the
           INFO  values  may be invalidated by the recoding (e.g. the total depth may need to be recalculated if
           individuals are removed). This behavior may be overridden by the following options. By  default,  BCF
           files are written out as BGZF compressed files.

         <b>--recode-INFO</b> <u>&lt;string&gt;</u>
         <b>--recode-INFO-all</b>
           These  options  can  be  used with the above recode options to define an INFO key name to keep in the
           output file. This option can be used multiple times to keep more  of  the  INFO  fields.  The  second
           option is used to keep all INFO values in the original file.

         <b>--contigs</b> <u>&lt;string&gt;</u>
           This  option  can  be used in conjunction with the --recode-bcf when the input file does not have any
           contig declarations. This option expects a file name with one contig header per line. These lines are
           included in the output file.

   <b>OUTPUT</b> <b>OTHER</b> <b>FORMATS</b>
         <b>--012</b>
           This option outputs the genotypes as a large matrix. Three files are produced. The first, with suffix
           ".012", contains the genotypes of each individual on a separate line. Genotypes are represented as 0,
           1 and 2, where the number represent that number  of  non-reference  alleles.  Missing  genotypes  are
           represented  by  -1. The second file, with suffix ".012.indv" details the individuals included in the
           main file. The third file, with suffix ".012.pos" details the site locations  included  in  the  main
           file.

         <b>--IMPUTE</b>
           This  option  outputs  phased  haplotypes in IMPUTE reference-panel format. As IMPUTE requires phased
           data, using this option also implies --phased.  Unphased  individuals  and  genotypes  are  therefore
           excluded.  Only bi-allelic sites are included in the output. Using this option generates three files.
           The IMPUTE haplotype file has the suffix ".impute.hap", and the IMPUTE legend  file  has  the  suffix
           ".impute.hap.legend".  The  third  file,  with  suffix  ".impute.hap.indv",  details  the individuals
           included in the haplotype file, although this file is not needed by IMPUTE.

         <b>--ldhat</b>
         <b>--ldhelmet</b>
         <b>--ldhat-geno</b>
           These options output data in LDhat/LDhelmet format. This option requires the "--chr" filter option to
           also be used. The two first options output phased data only, and therefore also implies "--phased" be
           used, leading to unphased individuals and genotypes being excluded. For LDhelmet, only snps  will  be
           considered,  and  therefore it implies "--remove-indels". The second option treats all of the data as
           unphased, and therefore outputs LDhat  files  in  genotype/unphased  format.  Two  output  files  are
           generated  with  the suffixes ".ldhat.sites" and ".ldhat.locs", which correspond to the LDhat "sites"
           and "locs" input files respectively;  for  LDhelmet,  the  two  files  generated  have  the  suffixes
           ".ldhelmet.snps" and ".ldhelmet.pos", which corresponds to the "SNPs" and "positions" files.

         <b>--BEAGLE-GL</b>
         <b>--BEAGLE-PL</b>
           These  options output genotype likelihood information for input into the BEAGLE program. The VCF file
           is required to contain FORMAT fields with "GL" or "PL" tags, which can generally  be  output  by  SNP
           callers  such  as  the GATK. Use of this option requires a chromosome to be specified via the "--chr"
           option. The resulting output file has the suffix ".BEAGLE.GL" or ".BEAGLE.PL" and  contains  genotype
           likelihoods  for  biallelic  sites.  This  file  is  suitable  for  input into BEAGLE via the "like="
           argument.

         <b>--plink</b>
         <b>--plink-tped</b>
         <b>--chrom-map</b>
           These options output the genotype data in PLINK PED format. With the  first  option,  two  files  are
           generated,  with  suffixes  ".ped" and ".map". Note that only bi-allelic loci will be output. Further
           details of these files can be found in the PLINK documentation.
           Note: The first option can be very slow on large datasets. Using the --chr option to  divide  up  the
           dataset is advised, or alternatively use the --plink-tped option which outputs the files in the PLINK
           transposed format with suffixes ".tped" and ".tfam".
           For  usage  with  variant  sites  in species other than humans, the --chrom-map option may be used to
           specify a file name that has a tab-delimited mapping of chromosome name to a  desired  integer  value
           with  one  line  per chromosome. This file must contain a mapping for every chromosome value found in
           the file.

</pre><h4><b>COMPARISON</b> <b>OPTIONS</b></h4><pre>
       These options are used to compare the original variant file  to  another  variant  file  and  output  the
       results.  All of the diff functions require both files to contain the same chromosomes and that the files
       be sorted in the same order. If one of the files contains chromosomes that the other file does  not,  use
       the --not-chr filter to remove them from the analysis.

   <b>DIFF</b> <b>VCF</b> <b>FILE</b>
         <b>--diff</b> <u>&lt;filename&gt;</u>
         <b>--gzdiff</b> <u>&lt;filename&gt;</u>
         <b>--diff-bcf</b> <u>&lt;filename&gt;</u>
           These  options compare the original input file to this specified VCF, gzipped VCF, or BCF file. These
           options must be specified with one additional option described below in order to specify what type of
           comparison is to be performed. See the examples section for typical usage.

   <b>DIFF</b> <b>OPTIONS</b>
         <b>--diff-site</b>
           Outputs the sites  that  are  common  /  unique  to  each  file.  The  output  file  has  the  suffix
           ".diff.sites_in_files".

         <b>--diff-indv</b>
           Outputs  the  individuals  that  are  common  /  unique  to each file. The output file has the suffix
           ".diff.indv_in_files".

         <b>--diff-site-discordance</b>
           This option calculates discordance on a site by site basis. The resulting output file has the  suffix
           ".diff.sites".

         <b>--diff-indv-discordance</b>
           This  option  calculates  discordance  on  a  per-individual basis. The resulting output file has the
           suffix ".diff.indv".

         <b>--diff-indv-map</b> <u>&lt;filename&gt;</u>
           This option allows the user to specify a mapping of individual IDs in the second file to those in the
           first file. The program expects the file to contain a tab-delimited line containing  an  individual's
           name in file one followed by that same individual's name in file two with one mapping per line.

         <b>--diff-discordance-matrix</b>
           This  option  calculates  a  discordance  matrix.  This  option  only works with bi-allelic loci with
           matching alleles that  are  present  in  both  files.  The  resulting  output  file  has  the  suffix
           ".diff.discordance.matrix".

         <b>--diff-switch-error</b>
           This  option  calculates phasing errors (specifically "switch errors"). This option creates an output
           file describing switch errors found between sites, with suffix ".diff.switch".

</pre><h4><b>AUTHORS</b></h4><pre>
       Adam Auton
       Anthony Marcketta

0.1.16                                            2 August 2018                                      <u><a href="../man1/vcftools.1.html">vcftools</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>