<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>extractTranscriptInfo.py - extracts information about transcripts from reference Fasta file</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/bitseq">bitseq_0.7.5+dfsg-6_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       extractTranscriptInfo.py - extracts information about transcripts from reference Fasta file

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>extractTranscriptInfo.py</b> [<u>options</u>] <u>&lt;inputFile&gt;</u> <u>&lt;outputFile&gt;</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
              This  program extracts information about transcripts from reference Fasta file.  This is partially
              replaced by using SAM header, which however does not  include  information  about  transcript-gene
              grouping.   Current  version of parseAlignment extracts this information from a reference sequence
              file (making this script obsolete).

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>-v</b>, <b>--verbose</b>
              Verbose output

       <b>-t</b> TYPE, <b>--type</b>=<u>TYPE</u>
              Type of file to parse: ensembl, cuff, other

extractTranscriptInfo.py 0.7.5+dfsg              September 2021                      <u><a href="../man1/EXTRACTTRANSCRIPTINFO.PY.1.html">EXTRACTTRANSCRIPTINFO.PY</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>