<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>samtools-sort - sorts SAM/BAM/CRAM files</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/samtools">samtools_1.21-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       samtools-sort - sorts SAM/BAM/CRAM files

</pre><h4><b>SYNOPSIS</b></h4><pre>
       samtools sort [<u>options</u>] [<u>in.sam</u>|<u>in.bam</u>|<u>in.cram</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Sort alignments by leftmost coordinates, by read name when <b>-n</b> or <b>-N</b> are used, by tag contents with <b>-t</b>, or
       a  minimiser-based collation order with <b>-M</b>.  An appropriate <b>@HD</b> <b>SO</b> sort order header tag will be added or
       an existing one updated if necessary, along with the <b>@HD</b> <b>SS</b> sub-sort header tag where appropriate.

       The sorted output is written to standard output by default, or to the specified file (<u>out.bam</u>) when <b>-o</b> is
       used.  This command will also create temporary files <u>tmpprefix</u><b>.</b><u>%d</u><b>.bam</b> as needed when the entire alignment
       data cannot fit into memory (as controlled via the <b>-m</b> option).

       Consider using <b>samtools</b> <b>collate</b> instead if you need name collated data  without  a  full  lexicographical
       sort.

       Note  that  if  the  sorted output file is to be indexed with <b>samtools</b> <b>index</b>, the default coordinate sort
       must be used.  Thus the <b>-n</b>, <b>-N</b>, <b>-t</b> and <b>-M</b> options are incompatible with <b>samtools</b> <b>index</b>.

       When sorting by minimisier (<b>-M</b>), the sort order is defined by the  whole-read  minimiser  value  and  the
       offset  into  the  read that this minimiser was observed.  This produces small clusters (contig-like, but
       unaligned) and helps to improve compression with LZ algorithms.  This can  be  improved  by  supplying  a
       known reference to build a minimiser index (<b>-I</b> and <b>-w</b> options).

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-l</b> <u>INT</u>     Set  the desired compression level for the final output file, ranging from 0 (uncompressed) or
                  1 (fastest but minimal compression) to 9 (best compression but slowest to write), similarly to
                  <b><a href="../man1/gzip.1.html">gzip</a></b>(1)'s compression level setting.

                  If <b>-l</b> is not used, the default compression level will apply.

       <b>-u</b>         Set the compression level to 0, for uncompressed output.  This is a synonym for <b>-l</b> <b>0</b>.

       <b>-m</b> <u>INT</u>     Approximately the maximum required memory per thread, specified either in bytes or with  a  <b>K</b>,
                  <b>M</b>, or <b>G</b> suffix.  [768 MiB]

                  To prevent sort from creating a huge number of temporary files, it enforces a minimum value of
                  1M for this setting.

       <b>-n</b>         Sort  by  read  names  (i.e., the <b>QNAME</b> field) using an alpha-numeric ordering, rather than by
                  chromosomal coordinates.  The alpha-numeric or “natural” sort order detects runs of digits  in
                  the strings and sorts these numerically.  Hence "a7b" appears before "a12b".  Note this is not
                  suitable  where  hexadecimal  values  are  in  use.   Sets the header sub-sort (<b>@HD</b> <b>SS</b>) tag to
                  <b>queryname:natural</b>.

       <b>-N</b>         Sort by read names (i.e., the <b>QNAME</b> field) using the lexicographical ordering, rather than  by
                  chromosomal  coordinates.   Unlike  <b>-n</b>  no  detection  of  numeric components is used, instead
                  relying purely on the ASCII value of each character.  Hence "x12" comes before "x7" as "1"  is
                  before "7" in ASCII.  This is a more appropriate name sort order where all digits in names are
                  already  zero-padded  and/or hexadecimal values are being used.  Sets the header sub-sort (<b>@HD</b>
                  <b>SS</b>) tag to <b>queryname:lexicographical</b>.

       <b>-t</b> <u>TAG</u>     Sort first by the value in the alignment tag TAG, then by position or name (if also  using  <b>-n</b>
                  or <b>-N</b>).

       <b>-M</b>         Sort  unmapped  reads (those in chromosome "*") by their sequence minimiser (Schleimer et al.,
                  2003; Roberts et al., 2004), also reverse complementing as appropriate.  This has  the  effect
                  of  collating  some  similar  data  together,  improving  the  compressibility of the unmapped
                  sequence.  The minimiser kmer size is adjusted using the <b>-K</b> option.  Note data  compressed  in
                  this manner may need to be name collated prior to conversion back to fastq.

                  Mapped sequences are sorted by chromosome and position.

       <b>-R</b>         Do not use reverse strand with minimiser sort (only compatible with -M).

       <b>-K</b> <u>INT</u>     Sets the kmer size to be used in the <b>-M</b> option. [20]

       <b>-I</b> <u>FILE</u>    Build  a  minimiser  index  over  <u>FILE</u>.   The per-read minimisers produced by <b>-M</b> are no longer
                  sorted by their numeric value, but by the reference coordinate this  minimiser  was  found  to
                  come from (if found in the index).  This further improves compression due to improved sequence
                  similarity between sequences, albeit with a small CPU cost of building and querying the index.
                  Specifying <b>-I</b> automatically implies <b>-M</b>.

       <b>-w</b> <u>INT</u>     Specifies  the window size for building the minimiser index on the file specified in <b>-I</b>.  This
                  defaults to 100.  It may be better to set this closer to 50 for short-read  data  sets  (at  a
                  higher CPU and memory cost), or for more speed up to 1000 for long-read data sets.

       <b>-H</b>         Squashes base homopolymers down to a single base pair before constructing the minimiser.  This
                  is useful for instruments where the primary source of error is in the length of homopolymer.

       <b>-o</b> <u>FILE</u>    Write the final sorted output to <u>FILE</u>, rather than to standard output.

       <b>-O</b> <u>FORMAT</u>  Write the final output as <b>sam</b>, <b>bam</b>, or <b>cram</b>.

                  By default, samtools tries to select a format based on the <b>-o</b> filename extension; if output is
                  to standard output or no format can be deduced, <b>bam</b> is selected.

       <b>-T</b> <u>PREFIX</u>  Write temporary files to <u>PREFIX</u><b>.</b><u>nnnn</u><b>.bam,</b> or if the specified <u>PREFIX</u> is an existing directory,
                  to  <u>PREFIX</u><b>/samtools.</b><u>mmm</u><b>.</b><u>mmm</u><b>.tmp.</b><u>nnnn</u><b>.bam,</b>  where  <u>mmm</u> is unique to this invocation of the <b>sort</b>
                  command.

                  By  default,   any   temporary   files   are   written   alongside   the   output   file,   as
                  <u>out.bam</u><b>.tmp.</b><u>nnnn</u><b>.bam,</b>  or  if  output  is  to  standard  output,  in  the current directory as
                  <b>samtools.</b><u>mmm</u><b>.</b><u>mmm</u><b>.tmp.</b><u>nnnn</u><b>.bam.</b>

       <b>-@</b> <u>INT</u>     Set number of sorting and compression threads.  By default, operation is single-threaded.

       <b>--no-PG</b>    Do not add a @PG line to the header of the output file.

       <b>--template-coordinate</b>
                  Sorts by template-coordinate, whereby the sort order (@HD SO) is  <b>unsorted</b>,  the  group  order
                  (GO) is <b>query</b>, and the sub-sort (SS) is <b>template-coordinate</b>.

       <b>Ordering</b> <b>Rules</b>

       The following rules are used for ordering records.

       If  option  <b>-t</b>  is  in use, records are first sorted by the value of the given alignment tag, and then by
       position or name (if using <b>-n</b> or <b>-N</b>).  For example, “-t RG” will make read group the  primary  sort  key.
       The rules for ordering by tag are:

       •   Records that do not have the tag are sorted before ones that do.

       •   If  the  types  of the tags are different, they will be sorted so that single character tags (type A)
           come before array tags (type B), then string tags (types H and Z), then numeric tags (types f and i).

       •   Numeric tags (types f and i) are compared by value.  Note that comparisons of  floating-point  values
           are subject to issues of rounding and precision.

       •   String  tags  (types  H  and  Z)  are  compared  based  on the binary contents of the tag using the C
           <b><a href="../man3/strcmp.3.html">strcmp</a></b>(3) function.

       •   Character tags (type A) are compared by binary character value.

       •   No attempt is made to compare tags of other types — notably type B array values will not be compared.

       When the <b>-n</b> or <b>-N</b> option is present, records are sorted  by  name.   Historically  samtools  has  used  a
       “natural” ordering — i.e. sections consisting of digits are compared numerically while all other sections
       are  compared  based on their binary representation.  This means “a1” will come before “b1” and “a9” will
       come before “a10”.  However this alpha-numeric sort can be confused by runs of hexadecimal  digits.   The
       newer  <b>-N</b>  option  adds a simpler lexicographical based name collation which does not attempt any numeric
       comparisons and may be more appropriate for some data sets.  Note care must be taken when using  <b>samtools</b>
       <b>merge</b> to ensure all files are using the same collation order.  Records with the same name will be ordered
       according  to the values of the READ1 and READ2 flags (see <b>samtools</b> <b>flags</b>). When that flag is also equal,
       ties are resolved with primary alignments first, then SUPPLEMENTARY, SECONDARY, and finally SUPPLEMENTARY
       plus SECONDARY.  Any remaining ties are reported in the same order as the input data.

       When the <b>--template-coordinate</b> option is in use, the reads are sorted by:

       1. The earlier unclipped 5' coordinate of the template.

       2. The higher unclipped 5' coordinate of the template.

       3. The library (from the read group).

       4. The molecular identifier (MI tag if present).

       5. The read name.

       6. If unpaired, or if R1 has the lower coordinates of the pair.

       When none of the above options are in use, reads are sorted by reference (according to the order  of  the
       @SQ header records), then by position in the reference, and then by the REVERSE flag.

       <b>Note</b>

       Historically <b>samtools</b> <b>sort</b> also accepted a less flexible way of specifying the final and temporary output
       filenames:

              samtools sort [<b>-f</b>] [<b>-o</b>] <u>in.bam</u> <u>out.prefix</u>

       This has now been removed.  The previous <u>out.prefix</u> argument (and <b>-f</b> option, if any) should be changed to
       an appropriate combination of <b>-T</b> <u>PREFIX</u> and <b>-o</b> <u>FILE</u>.  The previous <b>-o</b> option should be removed, as output
       defaults to standard output.

</pre><h4><b>AUTHOR</b></h4><pre>
       Written by Heng Li from the Sanger Institute with numerous subsequent modifications.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/samtools.1.html">samtools</a></u>(1), <u><a href="../man1/samtools-collate.1.html">samtools-collate</a></u>(1), <u><a href="../man1/samtools-merge.1.html">samtools-merge</a></u>(1)

       Samtools website: &lt;<a href="http://www.htslib.org/">http://www.htslib.org/</a>&gt;

samtools-1.21                                   12 September 2024                               <u><a href="../man1/samtools-sort.1.html">samtools-sort</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>