<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::tRNAscanSE - A parser for tRNAscan-SE output</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::tRNAscanSE - A parser for tRNAscan-SE output

</pre><h4><b>SYNOPSIS</b></h4><pre>
          use Bio::Tools::tRNAscanSE;

          my $parser = Bio::Tools::tRNAscanSE-&gt;new(-file =&gt; 'result.tRNAscanSE');

          # parse the results
          while( my $gene = $parser-&gt;next_prediction ) {

              @exon_arr = $gene-&gt;get_SeqFeatures();

          }

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This script will parse tRNAscan-SE output.  Just the tabular output of the tRNA locations in the genome
       for now.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Jason</b> <b>Stajich</b></h4><pre>
       Email jason-at-bioperl.org

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : my $obj = Bio::Tools::tRNAscanSE-&gt;new();
        Function: Builds a new Bio::Tools::tRNAscanSE object
        Returns : an instance of Bio::Tools::tRNAscanSE
        Args    : -fh/-file for input filename
                  -genetag =&gt; primary tag used in gene features (default 'tRNA_gene')
                  -exontag =&gt; primary tag used in exon features (default 'tRNA_exon')
                  -srctag  =&gt; source tag used in all features (default 'tRNAscan-SE')

   <b>gene_tag</b>
        Title   : gene_tag
        Usage   : $obj-&gt;gene_tag($newval)
        Function: Get/Set the value used for the 'gene_tag' of genes
                  Default is 'tRNA_gene' as set by the global $GeneTag
        Returns : value of gene_tag (a scalar)
        Args    : on set, new value (a scalar or undef, optional)

   <b>source_tag</b>
        Title   : source_tag
        Usage   : $obj-&gt;source_tag($newval)
        Function: Get/Set the value used for the 'source_tag' of exons and genes
                  Default is 'tRNAscan-SE' as set by the global $SrcTag
        Returns : value of source_tag (a scalar)
        Args    : on set, new value (a scalar or undef, optional)

   <b>exon_tag</b>
        Title   : exon_tag
        Usage   : $obj-&gt;exon_tag($newval)
        Function: Get/Set the value used for the 'primary_tag' of exons
                  Default is 'tRNA_exon' as set by the global $ExonTag
        Returns : value of exon_tag (a scalar)
        Args    : on set, new value (a scalar or undef, optional)

   <b>analysis_method</b>
        Usage     : $genscan-&gt;analysis_method();
        Purpose   : Inherited method. Overridden to ensure that the name matches
                    /tRNAscan-SE/i.
        Returns   : String
        Argument  : n/a

   <b>next_feature</b>
        Title   : next_feature
        Usage   : while($gene = $genscan-&gt;next_feature()) {
                         # do something
                  }
        Function: Returns the next gene structure prediction of the Genscan result
                  file. Call this method repeatedly until FALSE is returned.

                  The returned object is actually a SeqFeatureI implementing object.
                  This method is required for classes implementing the
                  SeqAnalysisParserI interface, and is merely an alias for
                  next_prediction() at present.

        Example :
        Returns : A Bio::SeqFeature::Generic object.
        Args    :
       See also : L&lt;Bio::SeqFeature::Generic&gt;

   <b>next_prediction</b>
        Title   : next_prediction
        Usage   : while($gene = $genscan-&gt;next_prediction()) {
                         # do something
                  }
        Function: Returns the next gene structure prediction of the Genscan result
                  file. Call this method repeatedly until FALSE is returned.

        Example :
        Returns : A Bio::SeqFeature::Generic object.
        Args    :
       See also : L&lt;Bio::SeqFeature::Generic&gt;

perl v5.32.1                                       2021-08-15                        <u>Bio::Tools::<a href="../man3pm/tRNAscanSE.3pm.html">tRNAscanSE</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>