<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>rabema_evaluate - RABEMA Evaluation</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/seqan-apps">seqan-apps_2.4.0+dfsg-16_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       rabema_evaluate - RABEMA Evaluation

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>rabema_evaluate</b> [<u>OPTIONS</u>] <b>--reference</b> <u>REF.fa</u> <b>--in-gsi</b> <u>IN.gsi</u> <b>--in-bam</b> <u>MAPPING.{sam,bam}</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Compare  the  SAM/bam  output <u>MAPPING.sam</u>/<u>MAPPING.bam</u> of any read mapper against the RABEMA gold standard
       previously built with <b>rabema_build_gold_standard</b>.  The input is a reference FASTA file, a  gold  standard
       interval (GSI) file and the SAM/BAM input to evaluate.

       The  input  SAM/BAM  file  must  be  <u>sorted</u>  <u>by</u>  <u>queryname</u>.   The  program will create a FASTA index file
       <u>REF.fa.fai</u> for fast random access to the reference.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              Display the help message.

       <b>--version</b>
              Display version information.

       <b>-v</b>, <b>--verbose</b>
              Enable verbose output.

       <b>-vv</b>, <b>--very-verbose</b>
              Enable even more verbose output.

   <b>Input</b> <b>/</b> <b>Output:</b>
       <b>-r</b>, <b>--reference</b> <u>INPUT_FILE</u>
              Path to load reference FASTA from. Valid filetypes are: <u>.sam[.*]</u>,  <u>.raw[.*]</u>,  <u>.gbk[.*]</u>,  <u>.frn[.*]</u>,
              <u>.fq[.*]</u>, <u>.fna[.*]</u>, <u>.ffn[.*]</u>, <u>.fastq[.*]</u>, <u>.fasta[.*]</u>, <u>.faa[.*]</u>, <u>.fa[.*]</u>, <u>.embl[.*]</u>, and <u>.bam</u>, where
              * is any of the following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent (de)compression.

       <b>-g</b>, <b>--in-gsi</b> <u>INPUT_FILE</u>
              Path to load gold standard intervals from. If compressed using gzip, the file will be decompressed
              on  the  fly.  Valid  filetype  is:  <u>.gsi[.*]</u>,  where * is any of the following extensions: <u>gz</u> for
              transparent (de)compression.

       <b>-b</b>, <b>--in-bam</b> <u>INPUT_FILE</u>
              Path to load the read mapper SAM or BAM output from. Valid filetypes are: <u>.sam[.*]</u> and <u>.bam</u>, where
              * is any of the following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent (de)compression.

       <b>--out-tsv</b> <u>OUTPUT_FILE</u>
              Path to write the statistics to as TSV. Valid filetype is: <u>.rabema_report_tsv</u>.

       <b>--dont-check-sorting</b>
              Do not check sortedness (by name) of input SAM/BAM files.  This is required if the reads  are  not
              sorted by name in the original FASTQ files.  Files from the SRA and ENA generally are sorted.

   <b>Benchmark</b> <b>Parameters:</b>
       <b>--oracle-mode</b>
              Enable  oracle  mode.   This  is used for simulated data when the input GSI file gives exactly one
              position that is considered as the true sample position.  For simulated data.

       <b>--only-unique-reads</b>
              Consider only reads that a single alignment in the  mapping  result  file.  Useful  for  precision
              computation.

       <b>--match-N</b>
              When set, N matches all characters without penalty.

       <b>--distance-metric</b> <u>STRING</u>
              Set  distance  metric.   Valid  values:  hamming,  edit.   Default: edit. One of <u>hamming</u> and <u>edit</u>.
              Default: <u>edit</u>.

       <b>-e</b>, <b>--max-error</b> <u>INTEGER</u>
              Maximal error rate to build gold standard for in  percent.   This  parameter  is  an  integer  and
              relative  to  the read length.  The error rate is ignored in oracle mode, here the distance of the
              read at the sample position is taken, individually for each read.  Default: 0 Default: <u>0</u>.

       <b>-c</b>, <b>--benchmark-category</b> <u>STRING</u>
              Set benchmark category.  One of {all, all-best, any-best.  Default: all One of <u>all</u>, <u>all-best</u>,  and
              <u>any-best</u>. Default: <u>all</u>.

       <b>--trust-NM</b>
              When  set,  we trust the alignment and distance from SAM/BAM file and no realignment is performed.
              Off by default.

       <b>--extra-pos-tag</b> <u>STRING</u>
              If the CIGAR string is absent, the missing alignment end position can be provided by this BAM tag.

       <b>--ignore-paired-flags</b>
              When set, we ignore all SAM/BAM flags related to pairing.  This is necessary  when  analyzing  SAM
              from SOAP's soap2sam.pl script.

       <b>--DONT-PANIC</b>
              Do not stop program execution if an additional hit was found that indicates that the gold standard
              is incorrect.

   <b>Logging:</b>
       <b>--show-missed-intervals</b>
              Show details for each missed interval from the GSI.

       <b>--show-invalid-hits</b>
              Show details for invalid hits (with too high error rate).

       <b>--show-additional-hits</b>
              Show details for additional hits (low enough error rate but not in gold standard.

       <b>--show-hits</b>
              Show details for hit intervals.

       <b>--show-try-hit</b>
              Show details for each alignment in SAM/BAM input.

              The  occurrence  of  "invalid"  hits  in  the  read mapper's output is not an error.  If there are
              additional hits, however, this shows an error in the gold standard.

</pre><h4><b>RETURN</b> <b>VALUES</b></h4><pre>
       A return value of 0 indicates success, any other value indicates an error.

</pre><h4><b>MEMORY</b> <b>REQUIREMENTS</b></h4><pre>
       From version 1.1, great care has been taken to keep the memory requirements as low as possible.

       The evaluation step needs to store the whole reference sequence in memory but little  more  memory.   So,
       for  the  human  genome,  the  memory  requirements  are below 4 GB, regardless of the size of the GSI or
       SAM/BAM file.

</pre><h4><b>REFERENCES</b></h4><pre>
       M. Holtgrewe, A.-K. Emde, D. Weese and K. Reinert.  A Novel  And  Well-Defined  Benchmarking  Method  For
       Second Generation Read Mapping, BMC Bioinformatics 2011, 12:210.

       <u><a href="http://www.seqan.de/rabema">http://www.seqan.de/rabema</a></u>
              RABEMA Homepage

       <u><a href="http://www.seqan.de/mason">http://www.seqan.de/mason</a></u>
              Mason Homepage

rabema_evaluate 1.2.10 [tarball]                                                              <u><a href="../man1/RABEMA_EVALUATE.1.html">RABEMA_EVALUATE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>