<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>art_illumina - Simulation of Illumina sequencers</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/art-nextgen-simulation-tools">art-nextgen-simulation-tools_20160605+dfsg-5build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       art_illumina - Simulation of Illumina sequencers

</pre><h4><b>DESCRIPTION</b></h4><pre>
       ART  is  a  set of simulation tools to generate synthetic next-generation sequencing reads. ART simulates
       sequencing reads by mimicking real sequencing process with empirical error  models  or  quality  profiles
       summarized from large recalibrated sequencing data.

       art_illumina can be used for Simulation of Illumina sequencers

</pre><h4><b>USAGE</b></h4><pre>
       <b>art_illumina</b> [options] <b>-sam</b> <b>-i</b> &lt;seq_ref_file&gt; <b>-l</b> &lt;read_length&gt; <b>-f</b> &lt;fold_coverage&gt; <b>-ss</b> &lt;sequencing_system&gt;
       <b>-o</b> &lt;outfile_prefix&gt;

       <b>art_illumina</b> [options] <b>-sam</b> <b>-i</b> &lt;seq_ref_file&gt; <b>-l</b> &lt;read_length&gt; <b>-f</b> &lt;fold_coverage&gt; <b>-o</b> &lt;outfile_prefix&gt;

       <b>art_illumina</b> [options] <b>-sam</b> <b>-i</b> &lt;seq_ref_file&gt; <b>-l</b> &lt;read_length&gt; <b>-c</b> &lt;total_num_reads&gt; <b>-o</b> &lt;outfile_prefix&gt;

       <b>art_illumina</b>  [options]  <b>-sam</b> <b>-i</b> &lt;seq_ref_file&gt; <b>-l</b> &lt;read_length&gt; <b>-f</b> &lt;fold_coverage&gt; <b>-m</b> &lt;mean_fragsize&gt; <b>-s</b>
       &lt;std_fragsize&gt; <b>-o</b> &lt;outfile_prefix&gt;

       <b>art_illumina</b> [options] <b>-sam</b> <b>-i</b> &lt;seq_ref_file&gt; <b>-l</b> &lt;read_length&gt; <b>-c</b> &lt;total_num_reads&gt; <b>-m</b> &lt;mean_fragsize&gt; <b>-s</b>
       &lt;std_fragsize&gt; <b>-o</b> &lt;outfile_prefix&gt;

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-1</b>   <b>--qprof1</b>
              the first-read quality profile

       <b>-2</b>   <b>--qprof2</b>
              the second-read quality profile

       <b>-amp</b> <b>--amplicon</b> amplicon sequencing simulation

       <b>-c</b>   <b>--rcount</b>
              total number of reads/read pairs to be generated [per amplicon if for amplicon simulation](not  be
              used together with <b>-f</b>/--fcov)

       <b>-d</b>   <b>--id</b>
              the prefix identification tag for read ID

       <b>-ef</b>  <b>--errfree</b>
              indicate to generate the zero sequencing errors SAM file as well the regular one

              NOTE:  the  reads  in  the  zero-error  SAM file have the same alignment positions as those in the
              regular SAM file, but have no sequencing errors

       <b>-f</b>   <b>--fcov</b>
              the fold of read coverage to be simulated  or  number  of  reads/read  pairs  generated  for  each
              amplicon

       <b>-h</b>   <b>--help</b>
              print out usage information

       <b>-i</b>   <b>--in</b>
              the filename of input DNA/RNA reference

       <b>-ir</b>  <b>--insRate</b>
              the first-read insertion rate (default: 0.00009)

       <b>-ir2</b> <b>--insRate2</b> the second-read insertion rate (default: 0.00015)

       <b>-dr</b>  <b>--delRate</b>
              the first-read deletion rate (default:  0.00011)

       <b>-dr2</b> <b>--delRate2</b> the second-read deletion rate (default: 0.00023)

       <b>-l</b>   <b>--len</b>
              the length of reads to be simulated

       <b>-m</b>   <b>--mflen</b>
              the mean size of DNA/RNA fragments for paired-end simulations

       <b>-mp</b>  <b>--matepair</b> indicate a mate-pair read simulation

       <b>-nf</b>  <b>--maskN</b>
              the cutoff frequency of 'N' in a window size of the read length for masking genomic regions

              NOTE: default: '-nf 1' to mask all regions with 'N'. Use '-nf 0' to turn off masking

       <b>-na</b>  <b>--noALN</b>
              do not output ALN alignment file

       <b>-o</b>   <b>--out</b>
              the prefix of output filename

       <b>-p</b>   <b>--paired</b>
              indicate a paired-end read simulation or to generate reads from both ends of amplicons

              NOTE:  art  will automatically switch to a mate-pair simulation if the given mean fragment size &gt;=
              2000

       <b>-q</b>   <b>--quiet</b>
              turn off end of run summary

       <b>-qs</b>  <b>--qShift</b>
              the amount to shift every first-read quality score by

       <b>-qs2</b> <b>--qShift2</b>
              the amount to shift every second-read quality score by

              NOTE: For <b>-qs</b>/-qs2 option, a positive number will shift up quality scores (the  max  is  93)  that
              reduce  substitution  sequencing  errors and a negative number will shift down quality scores that
              increase sequencing errors. If shifting scores by x, the error rate will be 1/(10^(x/10))  of  the
              default profile.

       <b>-rs</b>  <b>--rndSeed</b>
              the seed for random number generator (default: system time in second)

              NOTE: using a fixed seed to generate two identical datasets from different runs

       <b>-s</b>   <b>--sdev</b>
              the standard deviation of DNA/RNA fragment size for paired-end simulations.

       <b>-sam</b> <b>--samout</b>
              indicate to generate SAM alignment file

       <b>-sp</b>  <b>--sepProf</b>
              indicate to use separate quality profiles for different bases (ATGC)

       <b>-ss</b>  <b>--seqSys</b>
              The name of Illumina sequencing system of the built-in profile used for simulation

              NOTE: sequencing system id names are:

       GA1 - Genome Analyzer I, GA2 - Genome Analyzer II

       HS10 - HiSeq 1000, HS20 - HiSeq 2000, HS25 - HiSeq 2500, MS - MiSeq

       <b>-M</b>  <b>--cigarM</b>
              indicate to use CIGAR 'M' instead of '=/X' for alignment match/mismatch

</pre><h4><b>NOTES</b></h4><pre>
       *  ART by default selects a built-in quality score profile according to the read length specified for the
       run.

       * For single-end simulation, ART requires input sequence file, outputfile prefix, read length,  and  read
       count/fold coverage.

       * For paired-end simulation (except for amplicon sequencing), ART also requires the parameter values of

              the mean and standard deviation of DNA/RNA fragment lengths

</pre><h4><b>EXAMPLES</b></h4><pre>
       1) single-end read simulation

              art_illumina <b>-sam</b> <b>-i</b> reference.fa <b>-l</b> 150 <b>-ss</b> HS25 <b>-f</b> 10 <b>-o</b> single_dat

       2) paired-end read simulation

              art_illumina <b>-sam</b> <b>-i</b> reference.fa <b>-p</b> <b>-l</b> 150 <b>-ss</b> HS25 <b>-f</b> 20 <b>-m</b> 200 <b>-s</b> 10 <b>-o</b> paired_dat

       3) mate-pair read simulation

              art_illumina <b>-sam</b> <b>-i</b> reference.fa <b>-mp</b> <b>-l</b> 50 <b>-f</b> 20 <b>-m</b> 2500 <b>-s</b> 50 <b>-o</b> matepair_dat

       4) amplicon sequencing simulation with 5' end single-end reads

              art_illumina <b>-amp</b> <b>-sam</b> <b>-na</b> <b>-i</b> amp_reference.fa <b>-l</b> 50 <b>-f</b> 10 <b>-o</b> amplicon_5end_dat

       5) amplicon sequencing simulation with paired-end reads

              art_illumina <b>-amp</b> <b>-p</b> <b>-sam</b> <b>-na</b> <b>-i</b> amp_reference.fa <b>-l</b> 50 <b>-f</b> 10 <b>-o</b> amplicon_pair_dat

       6) amplicon sequencing simulation with matepair reads

              art_illumina <b>-amp</b> <b>-mp</b> <b>-sam</b> <b>-na</b> <b>-i</b> amp_reference.fa <b>-l</b> 50 <b>-f</b> 10 <b>-o</b> amplicon_mate_dat

       7) generate an extra SAM file with zero-sequencing errors for a paired-end read simulation

              art_illumina <b>-ef</b> <b>-i</b> reference.fa <b>-p</b> <b>-l</b> 50 <b>-f</b> 20 <b>-m</b> 200 <b>-s</b> 10 <b>-o</b> paired_twosam_dat

       8) reduce the substitution error rate to one 10th of the default profile

              art_illumina <b>-i</b> reference.fa <b>-qs</b> 10 <b>-qs2</b> 10 <b>-l</b> 50 <b>-f</b> 10 <b>-p</b> <b>-m</b> 500 <b>-s</b> 10 <b>-sam</b> <b>-o</b> reduce_error

       9) turn off the masking of genomic regions with unknown nucleotides 'N'

              art_illumina <b>-nf</b> 0 <b>-sam</b> <b>-i</b> reference.fa <b>-p</b> <b>-l</b> 50 <b>-f</b> 20 <b>-m</b> 200 <b>-s</b> 10 <b>-o</b> paired_nomask

       10) masking genomic regions with &gt;=5 'N's within the read length 50

              art_illumina <b>-nf</b> 5 <b>-sam</b> <b>-i</b> reference.fa <b>-p</b> <b>-l</b> 50 <b>-f</b> 20 <b>-m</b> 200 <b>-s</b> 10 <b>-o</b> paired_maskN5

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

art_illumina 3.19.15                              February 2016                                  <u><a href="../man1/ART_ILLUMINA.1.html">ART_ILLUMINA</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>