<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>art_454 - Simulation of 454 Pyrosequencing</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/art-nextgen-simulation-tools">art-nextgen-simulation-tools_20160605+dfsg-5build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       art_454 - Simulation of 454 Pyrosequencing

</pre><h4><b>DESCRIPTION</b></h4><pre>
       ART  is  a  set of simulation tools to generate synthetic next-generation sequencing reads. ART simulates
       sequencing reads by mimicking real sequencing process with empirical error  models  or  quality  profiles
       summarized from large recalibrated sequencing data.

       art_454 can be used for Simulation of 454 Pyrosequencing.

</pre><h4><b>USAGE</b></h4><pre>
   <b>SINGLE-END</b> <b>SIMULATION</b>
       <b>art_454</b>  [-s]  [-a  ]  [-t]  [-r  rand_seed]  [ <b>-p</b> read_profile ] [ <b>-c</b> num_flow_cycles ] &lt;INPUT_SEQ_FILE&gt;
       &lt;OUTPUT_FILE_PREFIX&gt; &lt;FOLD_COVERAGE&gt;

   <b>PAIRED-END</b> <b>SIMULATION</b>
       <b>art_454</b>
        [-s]  [-a  ]  [-t]  [-r  rand_seed]  [  <b>-p</b>  read_profile  ]  [  <b>-c</b>  num_flow_cycles  ]  &lt;INPUT_SEQ_FILE&gt;
       &lt;OUTPUT_FILE_PREFIX&gt; &lt;FOLD_COVERAGE&gt; &lt;MEAN_FRAG_LEN&gt; &lt;STD_DEV&gt;

   <b>AMPLICON</b> <b>SEQUENCING</b> <b>SIMULATION</b>
       <b>art_454</b>   [-s]  [-a  ]  [-t]  [-r  rand_seed]  [  <b>-p</b>  read_profile  ]  [  <b>-c</b>  num_flow_cycles  ]  &lt;-A|-B&gt;
       &lt;INPUT_SEQ_FILE&gt; &lt;OUTPUT_FILE_PREFIX&gt; &lt;#_READS/#_READ_PAIRS_PER_AMPLICON&gt;

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>MANDATORY</b> <b>OPTIONS</b>
       INPUT_SEQ_FILE           -  the filename of DNA/RNA reference sequences in FASTA format

       OUTPUT_FILE_PREFIX       -  the prefix or directory of output read data file (*.fq) and read alignment
       file (*.aln)

       FOLD_COVERAGE            -  the fold of read coverage over the reference sequences

       MEAN_FRAG_LEN            -  the average DNA fragment size for paired-end read simulation

       STD_DEV                  -  the standard deviation of the DNA fragment size for paired-end read
       simulation

       #READS_PER_AMPLICON      -  number of reads per amplicon (for 5'end amplicon sequencing)

       #READ_PAIRS_PER_AMPLICON -  number of read pairs per amplicon (for two-end amplicon sequencing)

   <b>OPTIONAL</b> <b>PARAMETERS</b>
       <b>-A</b> indicate to perform single-end amplicon sequencing simulation

       <b>-B</b> indicate to perform paired-end amplicon sequencing simulation

       <b>-M</b> indicate to use CIGAR 'M' instead of '=/X' for alignment match/mismatch

       <b>-a</b> indicate to output the ALN alignment file

       <b>-s</b> indicate to output the SAM alignment file

       <b>-d</b> print out warning messages for debugging

       <b>-t</b> indicate to simulate reads from the built-in GS FLX Titanium profile [default: GS FLX profile]

       <b>-r</b> specify a fixed random seed for the simulation (to generate two identical datasets from two different
       runs)

       <b>-c</b> specify the number of flow cycles by the sequencer [ default: 100 for GS-FLX, and 200 for GS-FLX
       Titanium ]

       <b>-p</b> specify user's own read profile for simulation

       NOTE: the name of a read profile is the directory containing read profile data files.   please  read  the
       REAME  file about the format of 454 read profile data files and.  and the default filenames of these data
       files.

</pre><h4><b>EXAMPLES</b></h4><pre>
       1) singl-end simulation with 20X coverage

              art_454 <b>-s</b> seq_reference.fa ./outdir/single_dat 20

       2) paired-end simulation with the mean fragment size 1500 and STD 20 using GS FLX Titanium platform

              art_454 <b>-s</b> <b>-t</b> seq_reference.fa ./outdir/paired_dat 10 1500 20

       3) paired-end simulation with a fixed random seed

              art_454 <b>-s</b> <b>-r</b> 777 seq_reference.fa ./outdir/paired_fxSeed 10 2500 50

       4) single-end amplicon sequencing with 10 reads per amplicon

              art_454 <b>-A</b> <b>-s</b> amplicon_ref.fa ./outdir/amp_single 10

       5) paired-end amplicon sequencing with 10 read pairs per amplicon

              art_454 <b>-B</b> <b>-s</b> amplicon_ref.fa ./outdir/amp_paired 10

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

art_454 3.19.15                                   February 2016                                       <u><a href="../man1/ART_454.1.html">ART_454</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>