<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>damapper - long read to reference genome mapping tool</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/damapper">damapper_0.0+git20240314.b025cf9-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       damapper - long read to reference genome mapping tool

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Recognised as the Damapper Library, this is a long read to reference genome mapping command line tool.

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>damapper</b>   [<u>-vpzCN</u>]  [<u>-k&lt;<a href="../man20/int.20.html">int</a>(20)&gt;</u>]  [<u>-t&lt;int&gt;</u>]  [<u>-M&lt;int&gt;</u>]  [<u>-T&lt;<a href="../man4/int.4.html">int</a>(4)&gt;</u>]  [<u>-P&lt;dir(<a href="file:/tmp">/tmp</a>)&gt;</u>]  [-e&lt;double(.85)]
       [-s&lt;<a href="../man100/int.100.html">int</a>(100)&gt;] [-n&lt;double(1.00)&gt;] [-m&lt;track&gt;]+ &lt;reference:dam&gt; &lt;reads:db&gt; ...

       <b>-k</b>: k-mer size (must be &lt;= 32).

       <b>-t</b>: Ignore k-mers that occur &gt;= <b>-t</b> times in a block.

       <b>-M</b>: Use only <b>-M</b> GB of memory by ignoring most frequent k-mers.

       <b>-e</b>: Look for alignments with <b>-e</b> percent similarity.

       <b>-s</b>: Use <b>-s</b> as the trace point spacing for encoding alignments.

       <b>-n</b>: Output all matches within this % of the best

       <b>-T</b>: Use <b>-T</b> threads.

       <b>-P</b>: Do sorts and merges in directory <b>-P</b>.

       <b>-m</b>: Soft mask the blocks with the specified mask.

       <b>-v</b>: Verbose mode, output statistics as proceed.

       <b>-z</b>: sort .las by A,B-read pairs (overlap piles)

              off =&gt; sort .las by A-read,A-position pairs (default for mapping)

       <b>-p</b>: Output repeat profile track

       <b>-C</b>: Output reference vs reads .las.

       <b>-N</b>: Do not output reads vs reference .las.

damapper                                           August 2020                                       <u><a href="../man1/DAMAPPER.1.html">DAMAPPER</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>