<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>velvetg_63 - de Bruijn graph construction, error removal and repeat resolution (MAXKMERLENGTH = 63)</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/velvet-long">velvet-long_1.2.10+dfsg1-9_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       velvetg_63 - de Bruijn graph construction, error removal and repeat resolution (MAXKMERLENGTH = 63)

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Usage: ./velvetg directory [options]

       directory
              : working directory name

   <b>Standard</b> <b>options:</b>
       <b>-cov_cutoff</b> &lt;floating-point|auto&gt;
              : removal of low coverage nodes AFTER tour bus or allow the system to infer it

              (default: no removal)

       <b>-ins_length</b> &lt;integer&gt;
              : expected distance between two paired end reads (default: no read pairing)

       <b>-read_trkg</b> &lt;yes|no&gt;
              : tracking of short read positions in assembly (default: no tracking)

       <b>-min_contig_lgth</b> &lt;integer&gt;
              : minimum contig length exported to contigs.fa file (default: hash length * 2)

       <b>-amos_file</b> &lt;yes|no&gt;
              : export assembly to AMOS file (default: no export)

       <b>-exp_cov</b> &lt;floating point|auto&gt;
              : expected coverage of unique regions or allow the system to infer it

              (default: no long or paired-end read resolution)

       <b>-long_cov_cutoff</b> &lt;floating-point&gt;: removal of nodes with low long-read coverage AFTER tour bus

              (default: no removal)

   <b>Advanced</b> <b>options:</b>
       <b>-ins_length</b>* &lt;integer&gt;
              : expected distance between two paired-end reads in the respective short-read dataset (default: no
              read pairing)

       <b>-ins_length_long</b> &lt;integer&gt;
              : expected distance between two long paired-end reads (default: no read pairing)

       <b>-ins_length</b>*_sd &lt;integer&gt;
              : est. standard deviation of respective dataset (default: 10% of corresponding length)

              [replace '*' by nothing, '2' or '_long' as necessary]

       <b>-scaffolding</b> &lt;yes|no&gt;
              : scaffolding of contigs used paired end information (default: on)

       <b>-max_branch_length</b> &lt;integer&gt;
              : maximum length in base pair of bubble (default: 100)

       <b>-max_divergence</b> &lt;floating-point&gt;: maximum divergence rate between two branches in a bubble (default: 0.2)

       <b>-max_gap_count</b> &lt;integer&gt;
              : maximum number of gaps allowed in the alignment of the two branches of a bubble (default: 3)

       <b>-min_pair_count</b> &lt;integer&gt;
              :  minimum  number  of  paired  end  connections  to  justify  the scaffolding of two long contigs
              (default: 5)

       <b>-max_coverage</b> &lt;floating point&gt;
              : removal of high coverage nodes AFTER tour bus (default: no removal)

       <b>-coverage_mask</b> &lt;int&gt;
              : minimum coverage required for confident regions of contigs (default: 1)

       <b>-long_mult_cutoff</b> &lt;int&gt;
              : minimum number of long reads required to merge contigs (default: 2)

       <b>-unused_reads</b> &lt;yes|no&gt;
              : export unused reads in UnusedReads.fa file (default: no)

       <b>-alignments</b> &lt;yes|no&gt;
              : export a summary of contig alignment to the reference sequences (default: no)

       <b>-exportFiltered</b> &lt;yes|no&gt;
              : export the long nodes which were eliminated by the coverage filters (default: no)

       <b>-clean</b> &lt;yes|no&gt;
              : remove all the intermediary files which are useless for recalculation (default : no)

       <b>-very_clean</b> &lt;yes|no&gt;
              : remove all the intermediary files (no recalculation possible) (default: no)

       <b>-paired_exp_fraction</b> &lt;double&gt;
              : remove all the paired end connections which less than the specified  fraction  of  the  expected
              count (default: 0.1)

       <b>-shortMatePaired</b>* &lt;yes|no&gt;
              :  for  mate-pair libraries, indicate that the library might be contaminated with paired-end reads
              (default no)

       <b>-conserveLong</b> &lt;yes|no&gt;
              : preserve sequences with long reads in them (default no)

   <b>Output:</b>
       directory/contigs.fa
              : fasta file of contigs longer than twice hash length

       directory/stats.txt
              : stats file (tab-spaced) useful for determining appropriate coverage cutoff

       directory/LastGraph
              : special formatted file with all the information on the final graph

       directory/velvet_asm.afg
              : (if requested) AMOS compatible assembly file

velvetg_63 1.2.10+dfsg1                             July 2018                                      <u><a href="../man1/VELVETG_63.1.html">VELVETG_63</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>