<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>anfo-tool - process native ANFO binary files</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/anfo">anfo_0.98-9_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       anfo-tool - process native ANFO binary files

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>anfo-tool</b> <b>[</b> <u>option</u> <b>|</b> <u>pattern</u> <b>...</b> <b>]</b>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>anfo-tool</b> is used to filter, process and convert the files created by <b>anfo</b>.  Every pattern on the command
       line  is  wildcard  expanded,  then for every input file (or the standard input, if no pattern is given),
       <b>anfo-tool</b> builds a chain of input filters, it then merges these input streams in  one  of  several  ways,
       splits  the  result  up  into  multiple  output  streams, each of which can have a chain of output filter
       applied.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>General</b> <b>Options</b>
       These options apply globally and modify the behavior of the whole program.  They can be  placed  anywhere
       in the command line.

       -V, --version
              Print version number and exit.

       -q, --quiet
              Suppress all output except fatal error messages.

       -v, --verbose
              Produce more output, including progress indicators for most operations.

       --debug
              Produce debugging output in addition to progress information.

       -n, --dry-run
              Parse command line, optionally print a description of the intended operations, then exit.

       --vmem X
              Limit  virtual  memory  to  <u>X</u> megabytes.  If memory runs out, <u>anfo-tool</u> tries to free up memory by
              forgetting about big files, e.g. genomes.  Use this option  to  avoid  swapping  or  out-of-memory
              conditions when operations involve big or multiple genomes.

   <b>Setting</b> <b>Parameters</b>
       A  parameter  can  be  set  multiple times on the command line and will overwrite previous settings.  Any
       filter option that needs a parameter picks up the last definition that appeared before the filter option.

       --set-slope S
              Set the slope parameter to <u>S</u>.  The <u>slope</u> is used together with the <u>intercept</u> where  filters  apply
              to  alignment scores; alignments scoring no worse than <u>slope</u> <u>*</u> <u>(length</u> <u>-</u> <u>intercept)</u> are considered
              good.  The default is 7.5.

       --set-intercept L
              Set the intercept parameter to <u>L</u>.  The <u>intercept</u> is used together with  the  <u>slope</u>  where  filters
              apply  to  alignment  scores;  alignments  scoring  no worse than <u>slope</u> <u>*</u> <u>(length</u> <u>-</u> <u>intercept)</u> are
              considered good.  The default is 20.

       --set-context C
              Set the context parameter to <u>C</u>.  The context is the number of surrounding bases of  the  reference
              included when printing alignments in text form.  The default is 0.

       --set-genome G
              Set  the  genome  parameter  to  <u>G</u>.   Many  filters will only consider the best alignments to this
              specific genome if it is set.  If no genome is set, the globally best alignment is used.

       --clear-genome
              Clear the genome parameter.  Filters apply to the globally best alignment afterwards.

   <b>Filter</b> <b>Options</b>
       Filters can be applied before merging the inputs or after splitting the back up.

       -s, --sort-pos=n
              Sort by alignment position while buffering no more than <u>n</u> MiB in memory.   If  a  genome  is  set,
              alignments to that genome are used.

       -S, --sort-name=n
              Sort by read name while buffering no more than <u>n</u> MiB in memory.

       -l, --filter-length=L
              Retain alignments only for reads of at least <u>L</u> bases length.  The reads themselves are kept.

       -f, --filter-score
              Retain alignments only if their score is good enough.  <u>Uses</u>slope<u>and</u>intercept<u>.</u>

       --filter-mapq=Q
              Remove alignments with mapping quality below <u>Q</u>.

       -h, --filter-hit=SEQ
              Keep  only reads that have a hit to a sequence named <u>SEQ</u>.  If <u>SEQ</u> is empty, reads are kept if they
              have any hit.  If the <u>genome</u> parameter is set, only hits to that genome count.

       --delete-hit=SEQ
              Delete alignments to <u>SEQ</u>.  If <u>SEQ</u> is empty, all alignments are deleted.  If the  <u>genome</u>  parameter
              is set, only alignments to that genome are deleted.

       --filter-qual=Q
              Mask out bases with quality below <u>Q</u>.  Such a base is replaced by the <u>N</u> ambiguity code.

       --multiplicity=N
              Keep  only  reads of molecules that have been sequenced at least <u>N</u> times.  Reads are considered to
              come from the same original molecule if their aligned coordinates are identical.

       --subsample=F
              Subsample a fraction <u>F</u> of the results.  Every read is independently and  randomly  choosen  to  be
              kept or not.

       --inside-regions=FILE
              Read a list of regions from <u>FILE</u>, then keep only alignments that overlap an annotated region.

       --outside-regions=FILE
              Read  a  list  of  regions  from  <u>FILE</u>, then keep only alignments that do not overlap an annotated
              region.

   <b>Special</b> <b>Filters</b>
       -d, --rmdup=Q
              Remove PCR duplicates, clamp quality scores to <u>Q</u>.  Two reads are considered to be  duplicates,  if
              their aligned coordinates are identical.  If a <u>genome</u> is set, the best alignment to that genome is
              used,  else  the  globally  best  alignment.   Both  alignments  must  be  good,  as determined by
              <u>slope</u>and<u>intercept</u>.  For a set of duplicates, a  consensus  is  called,  generally  increasing  the
              quality scores.  If a resulting quality score exceeds <u>Q</u>, it is set to <u>Q</u>.  This filter requires the
              input to be sorted by alignment coordinate on the selected <u>genome</u>.

              <u>--duct-tape=NAME</u>  Duct-tape overlapping alignments into contigs and call a consensus for them.  If
              a <u>genome</u> is set, alignments to that genome are used, else  the  globally  best  alignments.   This
              filter  requires  input  to  be  sorted by alignment coordinate on the genome.  Output is a set of
              contigs, every position gets assigned a consensus base, a quality score and likelihoods for  every
              possible  diallele.   (It  is called duct-taping because it kind of looks like an assembly, but is
              not nearly as solid.)

       --edit-header=ED
              Invoke the editor <u>ED</u> on the text representaion of the stream's header.  This can be used to  clean
              up header that have accumulated too much cruft.

   <b>Merging</b> <b>Filters</b>
       Exactly  one  merging filter should be given on the command line, all filter options occuring before that
       are part of the input filter chains, all further filters become output chains.  If no merging  filter  is
       given, <u>--concat</u> is assumed, and all filters are input filters.

       -c, --concat
              Concatenate all input streams in the order they appear on the command line.

       -m, --merge
              Merge sorted input streams, producing a sorted result.  All inputs must be sorted in the same way.

       -j, --join
              Join  input  streams  and  retain  the  single  best hits to each genome.  Every input stream must
              contain a record for every read, reads are  buffered  in  memory  until  all  of  their  hits  are
              collected.  This way, joining works well if all inputs are nearly in the same order.  If reads are
              missing from some streams, joining them will waste memory.

       --mega-merge
              Merge  many streams such as those produced by running <b>anfo-sge</b>.  Streams that operated on the same
              reads are joined, then everything is merged.

   <b>Output</b> <b>Options</b>
       If an output option is given on the command line, the current output filter chain is ended and a new  one
       is  started.   If  no  output option is given, a textual representation of the final stream is written to
       <u>stdout</u>.  All output options accept <u>-</u> to write to stdout.

       -o, --output FILE
              Write native binary stream (a compressed protobuf message) to <u>FILE</u>.  Writing a binary  stream  and
              reading it back in is lossless.

       --output-text FILE
              Write  protobuf  text  stream  to  <u>FILE</u>.   If  the  necessary  genomes  are  available,  a textual
              representation of the alignments is  included.   If  the  <u>context</u>  parameter  is  set,  that  many
              additional bases of the reference upstream and downstream from the alignment are included.

       --output-sam=FILE
              Write alignments in SAM format to <u>FILE</u>.

       --output-glz FILE
              Write  contigs  in GLZ 0.9 format to <u>FILE</u>.  Generating GLZ only works after application of <u>--duct-</u>
              <u>tape</u>, every contigs becomes a GLZ record.

       --output-3aln FILE
              Write contigs in a table based format to <u>FILE</u>.  The format is still subject  to  change,  see  the
              source code for detailed documentation.

       --output-fasta FILE
              Write alignments(!) in FastA format to <u>FILE</u>.  Alignments are writte as pair of reference and query
              sequence,  aligned  coordinates  are  indicated  in the description of the query sequence.  If the
              <u>context</u> parameter is set, that many additional bases of the reference upstream and downstream from
              the alignment are included.  This format is not suggested  for  any  serious  use,  it  exists  to
              support legacy applications.

       --output-fastq FILE
              Write  sequences(!) in FastQ format to <u>FILE</u>.  Writing FastQ effectively reconstitutes the input to
              <b>ANFO</b> if no filtering was done on the results.

       --output-table FILE
              Write per-alignment statistics to <u>FILE</u>.  The file has  three  colums:Âsequence  length,  alignment
              score,  difference  to  next  best  alignment.   It  is  mainly  useful  to  analyze/visualize the
              distribution of alignment scores.

       --stats FILE
              Write simple statistics to <u>FILE</u>.  This results in  some  simple  summary  statistics  of  a  whole
              stream: number of aligned sequences, average length, GC content.

</pre><h4><b>ENVIRONMENT</b></h4><pre>
       ANFO_PATH
              Colon separated list of directories searched for genome and index files.

       ANFO_TEMP
              Temporary space used for sorting of large files.

</pre><h4><b>FILES</b></h4><pre>
       <u>/etc/popt</u>
              The  system  wide  configuration  file for <b><a href="../man3/popt.3.html">popt</a></b>(3).  <b>anfo-tool</b> identifies itself as "anfo-tool" to
              popt.

       <u><a href="file:~/.popt">~/.popt</a></u>
              Per user configuration file for <b><a href="../man3/popt.3.html">popt</a></b>(3).

</pre><h4><b>BUGS</b></h4><pre>
       The command line of this tools is way too complicated and  its  semantics  are  counterintuitive.   Using
       <b>anfo-tool</b>  is probably best avoided in most cases, the <b>guile</b> bindings should provide a much more scalable
       and easier to understand interface.

</pre><h4><b>AUTHOR</b></h4><pre>
       Udo Stenzel &lt;<a href="mailto:udo_stenzel@eva.mpg.de">udo_stenzel@eva.mpg.de</a>&gt;

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/anfo.1.html">anfo</a></b>(1), <b><a href="../man1/fa2dna.1.html">fa2dna</a></b>(1) <b><a href="../man3/popt.3.html">popt</a></b>(3), <b><a href="../man5/fasta.5.html">fasta</a></b>(5)

Applications                                      OCTOBER 2009                                      <u><a href="../man1/ANFO-TOOL.1.html">ANFO-TOOL</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>