<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>QTLtools correct - Corrects genotypes or phenotypes with the provided covariates</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/qtltools">qtltools_1.3.1+dfsg-4build4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       QTLtools correct - Corrects genotypes or phenotypes with the provided covariates

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>QTLtools</b>  <b>correct</b> <b>--vcf</b> <u>[in.vcf</u>|<u>in.vcf.gz</u>|<u>in.bcf]</u> <b>|</b> <b>--bed</b> <u>in.bed.gz</u> <b>--cov</b> <u>covariates.txt</u> <b>|</b> <b>--normal</b> <b>--out</b>
       <u>output.txt</u> <u>[OPTIONS]</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This mode corrects genotypes or phenotypes in VCF, BCF, or BED files with covariates provided and/or rank
       normal transforms the data and writes out the new corrected values to a file.  This  mode  regresses  out
       the covariates from the data, and writes the residuals to a file.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>--vcf</b> <b>[</b><u>in.vcf</u>|<u>in.bcf</u>|<u>in.vcf.gz</u>|<u>in.bed.gz</u><b>]</b>
              Genotypes in VCF/BCF/BED format.  REQUIRED unless <b>--bed</b>.

       <b>--bed</b> <u>quantifications.bed.gz</u>
              Quantifications in BED format.  REQUIRED unless <b>--vcf</b>.

       <b>--out</b> <u>output_file</u>
              Output file name.  REQUIRED.

       <b>--cov</b> <u>covariates.txt</u>
              Covariates to correct the data with.  REQUIRED unless <b>--normal</b>.

       <b>--qtl</b> <u>qtl_file</u> <u>[in.vcf|in.vcf.gz|in.bcf]</u>
              Corrects given phenotypes with the given genotypes.  Takes two arguments.  The first one is a text
              file  listing  genotype-phenotype pairs (separated by a space e.g. snp1 gene1), one pair per line.
              A phenotype will be corrected for all the genotypes  provided  for  that  phenotype.   The  second
              argument  is  a VCF file containing the genotypes.  The variant IDs in the qtl_file must match the
              variant IDs in the VCF file, and the phenotype IDs must  match  with  the  <b>--bed</b>  file.   Requires
              <b>--bed</b>.

       <b>--normal</b>
              Rank normal transform the data.  REQUIRED unless <b>--cov</b>.

</pre><h4><b>EXAMPLES</b></h4><pre>
       o Correct  gene quantifications with technical covariates, rank normal transform the quantifications, and
         create a new bed file:

         QTLtools   correct   --bed    quantifications.bed.gz    --out    quantifications_corrected.bed    --cov
         technical_covars.txt --normal

       o Correct the gene quantifications for eQTLs

         QTLtools    correct    --bed    quantifications.bed.gz    --out   quantifications_corrected.bed   --qtl
         eQTL_variant_gene_pairs.txt

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/QTLtools.1.html">QTLtools</a></u>(1)

       QTLtools website: &lt;https://qtltools.github.io/qtltools&gt;

</pre><h4><b>BUGS</b></h4><pre>
       o Versions up to and including 1.2, suffer from a bug in reading  missing  genotypes  in  VCF/BCF  files.
         This  bug  affects  variants with a DS field in their genotype's FORMAT and have a missing genotype (DS
         field is .) in one of the samples, in which case genotypes for all the  samples  are  set  to  missing,
         effectively removing this variant from the analyses.

       Please submit bugs to &lt;https://github.com/qtltools/qtltools&gt;

</pre><h4><b>CITATION</b></h4><pre>
       Delaneau,  O.,  Ongen, H., Brown, A. et al. A complete tool set for molecular QTL discovery and analysis.
       <u>Nat</u> <u>Commun</u> <b>8</b>, 15452 (2017).  &lt;https://doi.org/10.1038/ncomms15452&gt;

</pre><h4><b>AUTHORS</b></h4><pre>
       Olivier Delaneau (<a href="mailto:olivier.delaneau@gmail.com">olivier.delaneau@gmail.com</a>), Halit Ongen (<a href="mailto:halitongen@gmail.com">halitongen@gmail.com</a>)

QTLtools-v1.3                                      06 May 2020                               <u><a href="../man1/QTLtools-correct.1.html">QTLtools-correct</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>