<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>QTLtools rep - Replicate QTL associations in an independent dataset</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/qtltools">qtltools_1.3.1+dfsg-4build4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       QTLtools rep - Replicate QTL associations in an independent dataset

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>QTLtools</b>  <b>rep</b>  <b>--bed</b> <u>quantifications.bed.gz</u> <b>--vcf</b> <u>[in.vcf</u>|<u>in.vcf.gz</u>|<u>in.bcf]</u> <b>--qtl</b> <u>qtls_external.txt</u> <b>--out</b>
       <u>output.txt</u> <u>[OPTIONS]</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This mode reads phenotype-genotype pairs that were identified in another dataset and  checks  association
       between the same genotype and phenotype in this dataset.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>--vcf</b> <b>[</b><u>in.vcf</u>|<u>in.bcf</u>|<u>in.vcf.gz</u><b>|</b><u>in.bed.gz</u><b>]</b> <b>...</b>
              Genotypes in VCF/BCF format, or another molecular phenotype in BED format.  If there is a DS field
              in   the  genotype  FORMAT  of  a  variant  (dosage  of  the  genotype  calculated  from  genotype
              probabilities, e.g. after imputation), then this is used as the genotype.  If there is only the GT
              field in the genotype FORMAT then this is used and it is converted to a dosage.  REQUIRED.

       <b>--bed</b> <u>quantifications.bed.gz</u> ...
              Molecular phenotype quantifications in BED format.  REQUIRED.

       <b>--qtl</b> <u>external_qtls.txt</u> ...
              The file listing the QTLs identified  in  the  external  dataset.   One  phenotype  genotype  pair
              (separated by space, phenotype first) per line.  REQUIRED.

       <b>--out</b> <u>filename.txt</u>
              The output file name.  REQUIRED.

       <b>--cov</b> <u>covariates.txt</u>
              Covariates to correct the phenotype data with.

       <b>--normal</b>
              Rank  normal  transform  the  phenotype  data  so  that  each  phenotype  is normally distributed.
              RECOMMENDED.

</pre><h4><b>OUTPUT</b> <b>FILE</b></h4><pre>
       <b>output</b> <b>file</b>
        Space separated output file with the following columns.
         1   The phenotype ID
         2   The phenotype's chromosome
         3   The phenotype's start position
         4   The phenotype's end position
         5   The phenotype's strand
         6   The genotype ID
         7   The genotype's chromosome
         8   The genotype's start position
         9   The genotype's end position
        10   The p-value of the association
        11   The slope of the association

</pre><h4><b>EXAMPLE</b></h4><pre>
       o Replicate the  results  of  dataset1  in  dataset2,  correcting  dataset2  phenotype's  with  technical
         covariates and normal transforming them:

         QTLtools   rep   --bed   dataset2.bed.gz   --vcf   dataset2.bcf   --cov  dataset2.covariates.txt  --qtl
         significant_qtls_dataset1.txt --normal --out rep_results.txt

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/QTLtools.1.html">QTLtools</a></u>(1)

       QTLtools website: &lt;https://qtltools.github.io/qtltools&gt;

</pre><h4><b>BUGS</b></h4><pre>
       Versions up to and including 1.2, suffer from a bug in reading missing genotypes in VCF/BCF files.   This
       bug  affects variants with a DS field in their genotype's FORMAT and have a missing genotype (DS field is
       .) in one of the samples, in which case genotypes for all the samples are  set  to  missing,  effectively
       removing this variant from the analyses.

       Please submit bugs to &lt;https://github.com/qtltools/qtltools&gt;

</pre><h4><b>CITATION</b></h4><pre>
       Delaneau  O., Ongen H., Brown A. A., et al. A complete tool set for molecular QTL discovery and analysis.
       <u>Nat</u> <u>Commun</u> <b>8</b>, 15452 (2017).  &lt;https://doi.org/10.1038/ncomms15452&gt;

</pre><h4><b>AUTHORS</b></h4><pre>
       Olivier Delaneau (<a href="mailto:olivier.delaneau@gmail.com">olivier.delaneau@gmail.com</a>), Halit Ongen (<a href="mailto:halitongen@gmail.com">halitongen@gmail.com</a>)

QTLtools-v1.3                                      06 May 2020                                   <u><a href="../man1/QTLtools-rep.1.html">QTLtools-rep</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>