<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>QTLtools pca - Conducts PCA</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/qtltools">qtltools_1.3.1+dfsg-4build4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       QTLtools pca - Conducts PCA

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>QTLtools</b> <b>pca</b> <b>--vcf</b> <u>[in.vcf</u>|<u>in.vcf.gz</u>|<u>in.bcf]</u> <b>|</b> <b>--bed</b> <u>in.bed.gz</u> <b>--out</b> <u>output.txt</u> <u>[OPTIONS]</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This  mode  allows  performing  a  Principal  Component  Analysis  (PCA)  either  on  molecular phenotype
       quantifications or genotype data.  It is typically used (i) to detect  outliers  in  the  data,  (ii)  to
       detect stratification in the data or (iii) to build a covariate matrix before QTL mapping.  QTLtools' PCA
       implementation  utilizes singular value decomposition (SVD).  When building a covariate matrix to account
       for technical covariates we recommend using <b>--center</b> and  <b>--scale</b>.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>--vcf</b> <b>[</b><u>in.vcf</u>|<u>in.bcf</u>|<u>in.vcf.gz</u>|<u>in.bed.gz</u><b>]</b>
              Genotypes in VCF/BCF/BED format.  REQUIRED unless <b>--bed</b>.

       <b>--bed</b> <u>quantifications.bed.gz</u>
              Quantifications in BED format.  REQUIRED unless <b>--vcf</b>.

       <b>--out</b> <u>output_prefix</u>
              Output file prefix.  REQUIRED.

       <b>--center</b>
              Center the variables (genotypes or phenotypes) by subtracting the mean from each value

       <b>--scale</b>
              Scale the variables (genotypes or phenotypes) by dividing each value by the standard deviation

       <b>--region</b> <u>chr:start-end</u>
              Genomic region to be processed.  E.g. chr4:12334456-16334456, or chr5

       <b>--exclude-chrs</b> <u>string</u>
              The chromosomes to exclude given as a space separated list.  Only applies to <b>--vcf</b>.  DEFAULT="X  Y
              M MT XY chrX chrY chrM chrMT chrXY"

       <b>--maf</b> <u>float</u>
              Exclude sites with minor allele frequency less than this.  Only applies to <b>--vcf</b>.  DEFAULT=0.0

       <b>--distance</b> <u>integer</u>
              Only include sites separated with this many base pairs.  Only applies to <b>--vcf</b>.  DEFAULT=0

</pre><h4><b>OUTPUT</b> <b>FILES</b></h4><pre>
       <b>.pca</b>
        This  file  contains  the  principal  components  that  were  calculated.   The  names  of the principal
        components, which is given in the first column, is composed of the output file prefix, whether the  data
        was centered, whether the data was scaled, and the principal component number.

       <b>.pca_stats</b>
        This  file  contains  the  standard  deviation  of  each  principal  component, and the variance and the
        cumulative variance explained by each PC.

</pre><h4><b>EXAMPLES</b></h4><pre>
       o Running pca on RNAseq quantifications to calculate technical covariates:

         QTLtools pca --bed genes.50percent.chr22.bed.gz --out genes.50percent.chr22 --center --scale

       o Running pca on genotypes to detect population stratification:

         QTLtools pca --vcf genotypes.chr22.vcf.gz --out genotypes.chr22 --center --scale --maf 0.05  --distance
         5000

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/QTLtools.1.html">QTLtools</a></u>(1)

       QTLtools website: &lt;https://qtltools.github.io/qtltools&gt;

</pre><h4><b>BUGS</b></h4><pre>
       o Versions  up  to  and  including  1.2, suffer from a bug in reading missing genotypes in VCF/BCF files.
         This bug affects variants with a DS field in their genotype's FORMAT and have a  missing  genotype  (DS
         fiels  is  .)  in  one  of the samples, in which case genotypes for all the samples are set to missing,
         effectively removing this variant from the analyses.

       Please submit bugs to &lt;https://github.com/qtltools/qtltools&gt;

</pre><h4><b>AUTHORS</b></h4><pre>
       Halit Ongen (<a href="mailto:halitongen@gmail.com">halitongen@gmail.com</a>), Olivier Delaneau (<a href="mailto:olivier.delaneau@gmail.com">olivier.delaneau@gmail.com</a>)

QTLtools-v1.3                                      06 May 2020                                   <u><a href="../man1/QTLtools-pca.1.html">QTLtools-pca</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>