<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>vcf-isec  -  create intersections, unions, complements on bgzipped and tabix indexed VCF or tab-delimited</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/vcftools">vcftools_0.1.16-3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       vcf-isec  -  create intersections, unions, complements on bgzipped and tabix indexed VCF or tab-delimited
       files

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>vcf-isec</b> [<u>OPTIONS</u>] <u>file1.vcf</u> <u>file2.vcf</u> ...

</pre><h4><b>DESCRIPTION</b></h4><pre>
       About: Create intersections, unions, complements on bgzipped  and  tabix  indexed  VCF  or  tab-delimited
       files.

              Note  that  lines  from  all  files  can  be  intermixed  together  on the output, which can yield
              unexpected results.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-C</b>, <b>--chromosomes</b> &lt;list|file&gt;
              Process the given chromosomes (comma-separated list or one chromosome per line in a file).

       <b>-c</b>, <b>--complement</b>
              Output positions present in the first file but missing from the other files.

       <b>-d</b>, <b>--debug</b>
              Debugging information

       <b>-f</b>, <b>--force</b>
              Continue even if the script complains about differing columns.

       <b>-o</b>, <b>--one-file-only</b>
              Print only entries from the left-most file. Without <b>-o</b>, all unique positions will be printed.

       <b>-n</b>, <b>--nfiles</b> [+-=]&lt;int&gt;
              Output positions present in this many (=), this many or more (+), or this many or fewer (-) files.

       <b>-p</b>, <b>--prefix</b> &lt;path&gt;
              If present, multiple files will be created with all possible isec combinations. (Suitable for Venn
              Diagram analysis.)

       <b>-t</b>, <b>--tab</b> &lt;chr:pos:file&gt;
              Tab-delimited file with indexes of chromosome and position columns. (1-based indexes)

       <b>-w</b>, <b>--win</b> &lt;int&gt;
              In repetitive sequences, the same indel can be called at  different  positions.  Consider  records
              this far apart as matching (be it a SNP or an indel).

       <b>-h</b>, -?, <b>--help</b>
              This help message.

</pre><h4><b>EXAMPLES</b></h4><pre>
              bgzip file.vcf; tabix <b>-p</b> vcf file.vcf.gz bgzip file.tab; tabix <b>-s</b> 1 <b>-b</b> 2 <b>-e</b> 2 file.tab.gz

vcf-isec 0.1.5                                      July 2011                                        <u><a href="../man1/VCF-ISEC.1.html">VCF-ISEC</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>