<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>QTLtools trans - trans QTL analysis</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/qtltools">qtltools_1.3.1+dfsg-4build4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       QTLtools trans - trans QTL analysis

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>QTLtools</b>  <b>trans</b>  <b>--vcf</b>  <u>[in.vcf</u>|<u>in.vcf.gz</u>|<u>in.bcf</u>|<u>in.bed.gz]</u>  <b>--bed</b>  <u>quantifications.bed.gz</u>  <b>[--nominal</b>  <b>|</b>
       <b>--permute</b> <b>|</b> <b>--sample</b> <u>integer</u> <b>|</b> <b>--adjust</b> <u>in.txt</u>] <b>--out</b> <u>output.txt</u> <u>[OPTIONS]</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This mode maps trans (distal) quantitative trait loci (QTLs) that affect  the  phenotypes,  using  linear
       regression.   The  method is detailed in &lt;https://www.nature.com/articles/ncomms15452&gt;.  We first regress
       out the provided covariates from the phenotype data, followed by running the  linear  regression  between
       the  phenotype residuals and the genotype.  If <b>--normal</b> and <b>--cov</b> are provided at the same time, then the
       residuals after the covariate correction are rank  normal  transformed.   It  incorporates  an  efficient
       permutation  scheme.   You can run a nominal pass (<b>--nominal</b>) listing all genotype-phenotype associations
       below a certain threshold, a permutation pass (<b>--permute</b> or <b>--sample</b> <u>no_genes_to_sample</u>)  to  empirically
       characterize  the null distribution of associations, or adjust the nominal p-values based on permutations
       (<b>--adjust</b>).

       In the full permutation scheme (<b>--permute</b>) we  permute  all  phenotypes  using  the  same  random  number
       sequence  to  preserve the correlation structure.  By doing so, the only association we actually break in
       the data is between the genotype and the phenotype data.  Then, we proceed with  a  standard  association
       scan  identical to the one used in the nominal pass.  In practice, we repeat this for 100 permutations of
       the phenotype data.  Subsequently, we can proceed with FDR correction by ranking all the nominal p-values
       in ascending order and by counting how many p-values  in  the  permuted  data  sets  are  smaller.   This
       provides  an  FDR  estimate:  if we have 500 p-values in the permuted data sets that are smaller than the
       100th smallest nominal p-value, we can then assume that the FDR for the 100 first associations is  around
       5% (=500/(100 Ã— 100)).

       To  enable  fast screening in trans, we also designed an approximation of the method described just above
       based on what we already do in cis.  To make it possible, we assume that the phenotypes  are  independent
       and  normally  distributed  (which can be enforced with <b>--normal</b>).  The idea is that since all phenotypes
       are normally distributed, effectively they are the same, and  also  the  cis  region  removed  from  each
       phenotype  is  so  small compared to rest of the genome that its phenotype specific impact is negligible.
       Hence the number of and the correlation amongst variants for each phenotype is  approximately  the  same,
       and  each  phenotype  is  approximately  the  same;  thus  we can run permutations with a small number of
       phenotypes  rather  then  all,  which  drastically  decreases  the  computational  burden  and  the  null
       distribution  generated  can  be  applied  to  all phenotypes.  The implementation draws from the null by
       permuting some randomly chosen phenotypes, testing for  associations  with  all  variants  in  trans  and
       storing  the  smallest  p-value.  When we repeat this many times (typically 1000), effectively building a
       null distribution of the strongest associations for a single phenotype.  We then make  it  continuous  by
       fitting  a  beta  distribution as we do in cis and use it to adjust every nominal p-value coming from the
       initial pass for the number of variants being tested.  To correct for  the  number  of  phenotypes  being
       tested,  we  estimate FDR as we do in cis; that is from the best adjusted p-values per phenotype (one per
       phenotype).  This also gives an adjusted p-value threshold that we use to identify all  phenotype-variant
       pairs  that are whole-genome significant.  In our experiments, this approach gives similar results to the
       full permutation scheme both in term of FDR estimates and number of discoveries, while running faster.

       Since linear regressions assumes normally distributed data, we <b>highly</b> <b>recommend</b> using the <b>--normal</b> option
       to rank normal transform the phenotype quantifications in order to avoid false positive associations  due
       to  outliers.   If  you are using the approximate permutation scheme (<b>--sample</b>) <b>you</b> <b>MUST</b> <b>use</b> <b>the</b> <b>--normal</b>
       <b>option</b> <b>or</b> <b>make</b> <b>sure</b> <b>that</b> <b>your</b> <b>phenotypes</b> <b>are</b> <b>normally</b> <b>distributed</b>.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>--vcf</b> <b>[</b><u>in.vcf</u>|<u>in.bcf</u>|<u>in.vcf.gz</u><b>|</b><u>in.bed.gz</u><b>]</b>
              Genotypes in VCF/BCF format, or another molecular phenotype in BED format.  If there is a DS field
              in  the  genotype  FORMAT  of  a  variant  (dosage  of  the  genotype  calculated  from   genotype
              probabilities, e.g. after imputation), then this is used as the genotype.  If there is only the GT
              field in the genotype FORMAT then this is used and it is converted to a dosage.  REQUIRED.

       <b>--bed</b> <u>quantifications.bed.gz</u>
              Molecular phenotype quantifications in BED format.  REQUIRED.

       <b>--out</b> <u>output.txt</u>
              Output file.  REQUIRED.

       <b>--cov</b> <u>covariates.txt</u>
              Covariates to correct the phenotype data with.

       <b>--normal</b>
              Rank  normal  transform  the  phenotype  data  so  that  each  phenotype  is normally distributed.
              RECOMMENDED.

       <b>--window</b> <u>integer</u>
              Size of the cis window to remove flanking each phenotype's start position.  DEFAULT=5000000.

       <b>--threshold</b> <u>float</u>
              P-value threshold below which hits are reported.  Give 1.0 to print everything, which may generate
              a huge file.  When <b>--adjust</b>  is  provided,  this  threshold  applies  to  the  adjusted  p-values.
              DEFAULT=1e-5.

       <b>--bins</b> <u>integer</u>
              Number of bins to use to categorize all p-values above --threshold.  DEFAULT=1000.

       <b>--nominal</b>
              Calculate  the nominal p-value for the genotype-phenotype associations and print out the ones that
              pass the provided threshold.  Mutually exclusive with --permute, --sample and --adjust.

       <b>--permute</b>
              Permute all phenotypes together, once.  For multiple permutations you need to  change  the  random
              seed using <b>--seed</b> for each permutation.  Mutually exclusive with --nominal, --sample and --adjust.

       <b>--sample</b> <u>integer</u>
              Permute  randomly  chosen phenotypes <u>integer</u> times.  Mutually exclusive with --nominal, --permute,
              --adjust, and --chunk.

       <b>--adjust</b> <u>filename</u>
              Test and adjust p-values using  the  null  distribution  in  <u>filename</u>.   Mutually  exclusive  with
              --nominal, --permute, and --sample.

       <b>--chunk</b> <u>integer1</u> <u>integer2</u>
              For  parallelization.   Divide  the  data  into <u>integer2</u> number of chunks and process chunk number
              <u>integer1</u>.  <b>Minimum</b> <b>number</b> <b>of</b> <b>chunks</b> <b>has</b> <b>to</b> <b>be</b> <b>at</b> <b>least</b> <b>the</b> <b>same</b> <b>number</b> <b>of</b> <b>chromosomes</b> <b>in</b> <b>the</b> <b>--bed</b>
              <b>file.</b>

</pre><h4><b>OUTPUT</b> <b>FILES</b></h4><pre>
       <b>.hits.txt.gz</b>
        Space separated results output file detailing the variant-phenotype pairs that pass the  threshold  with
        the following columns:
        1   The phenotype ID
        2   The phenotype chromosome
        3   Start position of the phenotype
        4   The variant ID
        5   The variant chromosome
        6   The start position of the variant
        7   The nominal p-value of the association between the variant and the phenotype.
        8   The adjusted p-value of the association between the variant and the phenotype.  Requires <b>--adjust</b>
        9   Correlation coefficient

       <b>.best.txt.gz</b>
        Space separated output file listing the most significant variant per phenotype.
        1   The phenotype ID
        2   The adjusted p-value of the association between the variant and the phenotype.  Requires <b>--adjust</b>
        3   The nominal p-value of the association between the variant and the phenotype.
        4   The variant ID

       <b>.bins.txt.gz</b>
        Space  separated  output  file  containing  the  binning  of all hits with a p-value below the specified
        <b>--threshold</b>.
        1   The index of the bin
        2   The lower bound of the correlation coefficient for this bin
        3   The upper bound of the correlation coefficient for this bin
        4   The upper bound of the p-value for this bin
        5   The lower bound of the p-value for this bin

</pre><h4><b>FULL</b> <b>PERMUTATION</b> <b>ANALYSIS</b> <b>EXAMPLE</b></h4><pre>
       1 Run a nominal analysis, rank normal transforming the phenotypes and outputting all associations with  a
         p-value below 1e-5:

         QTLtools trans --vcf genotypes.chr22.vcf.gz --bed genes.simulated.chr22.bed.gz --nominal --normal --out
         trans.nominal

       2 Run  a full permutation analysis with 100 jobs on a compute cluster, run the following <b>making</b> <b>sure</b> <b>that</b>
         <b>you</b> <b>change</b> <b>the</b> <b>seed</b> <b>for</b> <b>each</b> <b>permutation</b> <b>iteration</b> (qsub needs to be  changed  to  the  job  submission
         system used [bsub, psub, etc...])

         for j in $(seq 1 100); do
             echo  "QTLtools  trans  --vcf  genotypes.chr22.vcf.gz  --bed genes.simulated.chr22.bed.gz --permute
             --normal --out trans.perm$j.txt --seed $j" | qsub
         done

</pre><h4><b>APPROXIMATE</b> <b>PERMUTATION</b> <b>ANALYSIS</b> <b>EXAMPLE</b></h4><pre>
       1 Build the null distribution randomly selecting  1000  phenotypes,  and  rank  normal  transforming  the
         phenotypes:

         QTLtools  trans  --vcf genotypes.chr22.vcf.gz --bed genes.simulated.chr22.bed.gz --sample 1000 --normal
         --out trans.sample

       2 Run the nominal pass adjusting the p-values with the given null distribution, rank normal  transforming
         the phenotypes, and printing out associations with an adjusted p-value less than 0.1:

         QTLtools    trans    --vcf    genotypes.chr22.vcf.gz    --bed   genes.simulated.chr22.bed.gz   --adjust
         trans.sample.best.txt.gz --threshold 0.1 --normal --out trans.adjust

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/QTLtools.1.html">QTLtools</a></u>(1)

       QTLtools website: &lt;https://qtltools.github.io/qtltools&gt;

</pre><h4><b>BUGS</b></h4><pre>
       o Versions up to and including 1.2, suffer from a bug in reading  missing  genotypes  in  VCF/BCF  files.
         This  bug  affects  variants with a DS field in their genotype's FORMAT and have a missing genotype (DS
         field is .) in one of the samples, in which case genotypes for all the  samples  are  set  to  missing,
         effectively removing this variant from the analyses.

       Please submit bugs to &lt;https://github.com/qtltools/qtltools&gt;

</pre><h4><b>CITATION</b></h4><pre>
       Delaneau,  O.,  Ongen, H., Brown, A. et al. A complete tool set for molecular QTL discovery and analysis.
       <u>Nat</u> <u>Commun</u> <b>8</b>, 15452 (2017).  &lt;https://doi.org/10.1038/ncomms15452&gt;

</pre><h4><b>AUTHORS</b></h4><pre>
       Halit Ongen (<a href="mailto:halitongen@gmail.com">halitongen@gmail.com</a>), Olivier Delaneau (<a href="mailto:olivier.delaneau@gmail.com">olivier.delaneau@gmail.com</a>)

QTLtools-v1.3                                      06 May 2020                                 <u><a href="../man1/QTLtools-trans.1.html">QTLtools-trans</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>