<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>iva - iterative virus assembler</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/iva">iva_1.0.11+ds-6_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       iva - iterative virus assembler

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: iva [options] {-f reads_fwd <b>-r</b> reads_rev | <b>--fr</b> reads} &lt;output directory&gt;

   <b>positional</b> <b>arguments:</b>
       Output directory
              Name of output directory (must not already exist)

   <b>options:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

   <b>Input</b> <b>and</b> <b>output:</b>
       <b>-f</b> filename[.gz], <b>--reads_fwd</b> filename[.gz]
              Name of forward reads fasta/q file. Must be used in conjunction with <b>--reads_rev</b>

       <b>-r</b> filename[.gz], <b>--reads_rev</b> filename[.gz]
              Name of reverse reads fasta/q file. Must be used in conjunction with <b>--reads_fwd</b>

       <b>--fr</b> filename[.gz]
              Name of interleaved fasta/q file

       <b>--keep_files</b>
              Keep intermediate files (could be many!). Default is to delete all unnecessary files

       <b>--contigs</b> filename[.gz]
              Fasta file of contigs to be extended. Incompatible with <b>--reference</b>

       <b>--reference</b> filename[.gz]
              EXPERIMENTAL! This option is EXPERIMENTAL, not recommended, and has not been tested! Fasta file of
              reference genome, or parts thereof. IVA will try to assemble one contig per sequence in this file.
              Incompatible with <b>--contigs</b>

       <b>-v</b>, <b>--verbose</b>
              Be verbose by printing messages to stdout. Use up to three times for increasing verbosity.

   <b>SMALT</b> <b>mapping</b> <b>options:</b>
       <b>-k</b> INT, <b>--smalt_k</b> INT
              kmer hash length in SMALT (the <b>-k</b> option in smalt index) [19]

       <b>-s</b> INT, <b>--smalt_s</b> INT
              kmer hash step size in SMALT (the <b>-s</b> option in smalt index) [11]

       <b>-y</b> FLOAT, <b>--smalt_id</b> FLOAT
              Minimum identity threshold for mapping to be reported (the <b>-y</b> option in smalt map) [0.5]

   <b>Contig</b> <b>options:</b>
       <b>--ctg_first_trim</b> INT
              Number of bases to trim off the end of every contig before extending for the first time [25]

       <b>--ctg_iter_trim</b> INT
              During  iterative  extension, number of bases to trim off the end of a contig when extension fails
              (then try extending again) [10]

       <b>--ext_min_cov</b> INT
              Minimum kmer depth needed to use that kmer to extend a contig [10]

       <b>--ext_min_ratio</b> FLOAT
              Sets N, where kmer for extension must be at least N times more abundant than next most common kmer
              [4]

       <b>--ext_max_bases</b> INT
              Maximum number of bases to try to extend on each iteration [100]

       <b>--ext_min_clip</b> INT
              Set minimum number of bases soft clipped off a read for those bases to be used for extension [3]

       <b>--max_contigs</b> INT
              Maximum number of contigs allowed in the assembly. No  more  seeds  generated  if  the  cutoff  is
              reached [50]

   <b>Seed</b> <b>generation</b> <b>options:</b>
       <b>--make_new_seeds</b>
              When  no  more  contigs  can  be  extended,  generate  a  new seed. This is forced to be true when
              <b>--contigs</b> is not used

       <b>--seed_start_length</b> INT
              When making a seed sequence, use the most common kmer of  this  length.  Default  is  to  use  the
              minimum of (median read length, 95). Warning: it is not recommended to set this higher than 95

       <b>--seed_stop_length</b> INT
              Stop  extending  seed  using  perfect  matches  from  reads  when  this  length is reached. Future
              extensions are then made by treating the seed as a contig [0.9*max_insert]

       <b>--seed_min_kmer_cov</b> INT
              Minimum kmer coverage of initial seed [25]

       <b>--seed_max_kmer_cov</b> INT
              Maximum kmer coverage of initial seed [1000000]

       <b>--seed_ext_max_bases</b> INT
              Maximum number of bases to try to extend on each iteration [50]

       <b>--seed_overlap_length</b> INT
              Number  of  overlapping  bases  needed  between  read  and  seed  to  use  that  read  to   extend
              [seed_start_length]

       <b>--seed_ext_min_cov</b> INT
              Minimum kmer depth needed to use that kmer to extend a contig [10]

       <b>--seed_ext_min_ratio</b> FLOAT
              Sets N, where kmer for extension must be at least N times more abundant than next most common kmer
              [4]

   <b>Read</b> <b>trimming</b> <b>options:</b>
       <b>--trimmomatic</b> FILENAME
              Provide location of trimmomatic.jar file to enable read trimming. Required if <b>--adapters</b> used

       <b>--trimmo_qual</b> STRING
              Trimmomatic options used to quality trim reads [LEADING:10 TRAILING:10 SLIDINGWINDOW:4:20]

       <b>--adapters</b> FILENAME
              Fasta  file  of  adapter  sequences to be trimmed off reads. If used, must also use <b>--trimmomatic</b>.
              Default is file of adapters supplied with IVA

       <b>--min_trimmed_length</b> INT
              Minimum length of read after trimming [50]

       <b>--pcr_primers</b> FILENAME
              FASTA file of primers. The first perfect match found to a sequence in the  primers  file  will  be
              trimmed off the start of each read. This is run after trimmomatic (if <b>--trimmomatic</b> used)

   <b>Other</b> <b>options:</b>
       <b>-i</b> INT, <b>--max_insert</b> INT
              Maximum  insert size (includes read length). Reads with inferred insert size more than the maximum
              will not be used to extend contigs [800]

       <b>-t</b> INT, <b>--threads</b> INT
              Number of threads to use [1]

       <b>--kmc_onethread</b>
              Force kmc to use one thread. By default the value of <b>-t</b>/--threads is used when running kmc

       <b>--strand_bias</b> FLOAT in [0,0.5]
              Set strand bias cutoff of mapped reads when trimming contig ends, in the interval [0,0.5]. A value
              of x means that a base needs min(fwd_depth, rev_depth) / total_depth &lt;=  x.  The  only  time  this
              should  be used is with libraries with overlapping reads (ie fragment length &lt; 2*read length), and
              even then, it can make results worse. If used, try a low value like 0.1 first [0]

       <b>--test</b> Run using built in test data. All other options will  be  ignored,  except  the  mandatory  output
              directory, and <b>--trimmomatic</b> and <b>--threads</b> can be also be used

       <b>--version</b>
              show program's version number and exit

iva 1.0.8                                         December 2024                                           <u><a href="../man1/IVA.1.html">IVA</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>