<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>csv_to_gene_db - generate the appropriate headers for srst2</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/srst2">srst2_0.2.0-13_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       csv_to_gene_db - generate the appropriate headers for srst2

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>csv_to_gene_db</b> [<u>options</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This tool is part of the SRST2 suite.

       take  csv  table  detailing  clustering  etc  and sequences for gene DB, write as fasta expected csv file
       format:

              seqID,clusterid,gene,allele,(DNAseq),other....

       headers in output will be  srst2  compatible,  ie  [clusterID]__[gene]__[allele]__[seqID]  [other  stuff]
       sequence  can  be  read from a specified column or from a fasta file (specify which column contains fasta
       header to match in seqs file)

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>-t</b> TABLE_FILE, <b>--table</b>=<u>TABLE_FILE</u>
              table to read (csv)

       <b>-o</b> OUTPUT_FILE, <b>--out</b>=<u>OUTPUT_FILE</u>
              output file (fasta)

       <b>-s</b> SEQ_COL, <b>--seq_col</b>=<u>SEQ_COL</u>
              column number containing sequences

       <b>-f</b> FASTA_FILE, <b>--fasta</b>=<u>FASTA_FILE</u>
              fasta file to read sequences from (must specify which column in the table  contains  the  sequence
              names that match the fasta file headers)

       <b>-c</b> HEADERS_COL, <b>--headers_col</b>=<u>HEADERS_COL</u>
              column number that contains the sequence names that match the fasta file headers

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

csv_to_gene_db 0.1.8                                June 2016                                  <u><a href="../man1/CSV_TO_GENE_DB.1.html">CSV_TO_GENE_DB</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>