<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>cnvkit_heatmap - Plot copy number for multiple samples as a heatmap.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/cnvkit">cnvkit_0.9.12-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       cnvkit_heatmap - Plot copy number for multiple samples as a heatmap.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: cnvkit.py heatmap [-h] [-c CHROMOSOME] [-y]

       [-x {m,y,male,Male,f,x,female,Female}]
              [--no-shift-xy]  [-o  FILENAME] [-b] [-d] [-v] [--delimit-samples] [-t TITLE] filenames [filenames
              ...]

   <b>positional</b> <b>arguments:</b>
       filenames
              Sample coverages as raw probes (.cnr) or segments (.cns).

   <b>options:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>-c</b> CHROMOSOME, <b>--chromosome</b> CHROMOSOME
              Chromosome (e.g. 'chr1') or chromosomal range (e.g.   'chr1:2333000-2444000')  to  display.  If  a
              range  is  given,  all targeted genes in this range will be shown, unless '--gene'/'-g' is already
              given.

       <b>-y</b>, <b>--male-reference</b>, <b>--haploid-x-reference</b>
              Assume inputs were normalized to a male reference (i.e. female samples will  have  +1  log-CNR  of
              chrX; otherwise male samples would have <b>-1</b> chrX).

       <b>-x</b> {m,y,male,Male,f,x,female,Female}, <b>--sample-sex</b> {m,y,male,Male,f,x,female,Female}, <b>-g</b>
       {m,y,male,Male,f,x,female,Female}, <b>--gender</b> {m,y,male,Male,f,x,female,Female}
              Specify  the  chromosomal  sex of all given samples as male or female. [Default: guess each sample
              from coverage of X and Y chromosomes].

       <b>--no-shift-xy</b>
              Don't adjust the X and Y chromosomes according to sample sex.

       <b>-o</b> FILENAME, <b>--output</b> FILENAME
              Output PDF file name.

   <b>Plot</b> <b>aesthetics:</b>
       <b>-b</b>, <b>--by-bin</b>
              Plot data x-coordinates by bin indices instead of genomic coordinates. All bins will be shown with
              equal width, no blank regions will be  shown,  and  x-axis  values  indicate  bin  number  (within
              chromosome) instead of genomic position.

       <b>-d</b>, <b>--desaturate</b>
              Tweak color saturation to focus on significant changes.

       <b>-v</b>, <b>--vertical</b>
              Plot heatmap with samples as X-axis (instead of Y-axis).

       <b>--delimit-samples</b>
              Add an horizontal delimitation line between each sample.

       <b>-t</b> TITLE, <b>--title</b> TITLE
              Plot title. [Default: Range if provided, otherwise none]

cnvkit.py heatmap 0.9.10                            July 2023                                  <u><a href="../man1/CNVKIT_HEATMAP.1.html">CNVKIT_HEATMAP</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>