<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>kmerresistance - correlates mapped genes with the predicted species of WGS samples</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/kmerresistance">kmerresistance_2.2.0-4ubuntu1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       kmerresistance - correlates mapped genes with the predicted species of WGS samples

</pre><h4><b>SYNOPSIS</b></h4><pre>
</pre><h4><b>DESCRIPTION</b></h4><pre>
       KmerResistance  correlates mapped genes with the predicted species of WGS samples, where this this allows
       for identification of genes in samples which have been poorly sequenced or high accuracy predictions  for
       samples  with  contamination. KmerResistance has one dependency, namely KMA to perform the mapping, which
       is also freely available.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-i</b>     Input/query file name                                   REQUIRED

       <b>-o</b>     Output file                                             REQUIRED

       <b>-t_db</b>  Template DB                                             REQUIRED

       <b>-s_db</b>  Species DB                                              REQUIRED

       <b>-id</b>    ID threshhold                   70.0

       <b>-dct</b>   Depth correction threshhold     0.1

       <b>-kma</b>   alternative KMA         kma

   <b>KMA</b> <b>options:</b>
       <b>-i</b>     Input/query file name           None            REQUIRED

       <b>-o</b>     Output file                     None            REQUIRED

       <b>-t_db</b>  Template DB                     None            REQUIRED

       <b>-k</b>     Kmersize                        16

       <b>-e</b>     evalue                          0.05

       <b>-delta</b> Align in pieces of delta        511

       <b>-mem_mode</b>
              Use kmers to choose best template, and save memory       False

       <b>-ex_mode</b>
              Searh kmers exhaustively        False

       <b>-deCon</b> Remove contamination            False

       <b>-dense</b> Do not allow insertions in assembly                     False

       <b>-ref_fsa</b>
              Consensus sequnce will have "n" instead of gaps        False

       <b>-matrix</b>
              Print assembly matrix           False

       <b>-mp</b>    Minimum phred score             30

       <b>-5p</b>    Cut a constant number of nucleotides from the 5 prime.   0

       <b>-Sparse</b>
              Run KmerFinder                  False

       <b>-ID</b>    Minimum ID                      1.0%

       <b>-ss</b>    Sparse sorting (q,c,d)          q

       <b>-shm</b>   Use shared DB made by kma_shm   0 (lvl)

       <b>-swap</b>  Swap DB to disk                 0 (lvl)

       <b>-1t1</b>   Skip HMM                        False

       <b>-boot</b>  Bootstrap sequence              False

       <b>-mrs</b>   Minimum alignment score, normalized to alignment length  0.5

       <b>-reward</b>
              Score for match                 1

       <b>-penalty</b>
              Penalty for mismatch            <b>-2</b>

       <b>-gapopen</b>
              Penalty for gap opening         <b>-3</b>

       <b>-gapextend</b>
              Penalty for gap extension       <b>-1</b>

       <b>-h</b>     Shows this help message

</pre><h4><b>EXAMPLES</b></h4><pre>
       Mapping reads against resistance genes:

                kmerresistance -i sample_1.fastq sample_2.fastq -o out -t_db ResFinder -s_db bacteria

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

kmerresistance 2.0+git20180205.26467e9              June 2018                                  <u><a href="../man1/KMERRESISTANCE.1.html">KMERRESISTANCE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>