<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>QTLtools bamstat - Calculate stats of overlap between an RNAseq BAM file and an annotation</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/qtltools">qtltools_1.3.1+dfsg-4build4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       QTLtools bamstat - Calculate stats of overlap between an RNAseq BAM file and an annotation

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>QTLtools</b> <b>bamstat</b> <b>--bam</b> <b>[</b><u>sample.bam|sample.sam|sample.cram</u><b>]</b> <b>--bed</b> <b>[</b><u>gene_annotation.bed</u><b>]</b>  <b>--out</b> <u>output_file</u>
       [<u>OPTIONS</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This  mode  counts  the  number  of  RNAseq reads, and the ones that overlap with an annotation file.  We
       recommend using uniquely mapping reads only by specifying the correct <b>--filter-mapping-quality</b>.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>--bed</b>  <u>annotation.bed</u>
              Annotation of interest REQUIRED.

       <b>--bam,</b> <b>-b</b> <b>[</b><u>in.bam</u>|<u>in.sam</u>|<u>in.cram</u><b>]</b>
              Sequence data in BAM/SAM/CRAM format.  REQUIRED.

       <b>--out,</b> <b>-o</b> <u>output</u>
              Output file name REQUIRED.

       <b>--filter-mapping-quality</b> <u>integer</u>
              Minimum mapping quality for a read or read pair to  be  considered.   Set  this  to  only  include
              uniquely mapped reads.  DEFAULT=10

       <b>--filter-keep-duplicates</b>
              Keep reads designated as duplicate by the aligner.  RECOMMENDED for RNAseq

</pre><h4><b>OUTPUT</b> <b>FILE</b> <b>COLUMNS</b></h4><pre>
       <b>--out</b> <u>filename</u>
        This file does not have header and it contains the following columns:

        1   The total number of reads in the BAM file
        2   The number of mapped sequencing reads passing the <b>--filter-mapping-quality</b>
        3   The number of mapped sequencing reads falling within the annotations specified with <b>--bed</b>
        4   The total number of annotations in the <b>--bed</b> file
        5   The number of annotations covered by at least one sequencing read

</pre><h4><b>EXAMPLES</b></h4><pre>
       o Running bamstat on an RNAseq sample mapped with GEM and GENCODE gene annotations:

         QTLtools     bamstat     --bam     HG00381.chr22.bam      --out     HG00381.chr22.bamstat.txt     --bed
         gencode.v19.annotation.bed.gz --filter-mapping-quality 150 --filter-keep-duplicates

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/QTLtools.1.html">QTLtools</a></u>(1)

       QTLtools website: &lt;https://qtltools.github.io/qtltools&gt;

</pre><h4><b>BUGS</b></h4><pre>
       Please submit bugs to &lt;https://github.com/qtltools/qtltools&gt;

</pre><h4><b>CITATION</b></h4><pre>
       Delaneau, O., Ongen, H., Brown, A. et al. A complete tool set for molecular QTL discovery  and  analysis.
       <u>Nat</u> <u>Commun</u> <b>8</b>, 15452 (2017).  &lt;https://doi.org/10.1038/ncomms15452&gt;

</pre><h4><b>AUTHORS</b></h4><pre>
       Olivier Delaneau (<a href="mailto:olivier.delaneau@gmail.com">olivier.delaneau@gmail.com</a>), Halit Ongen (<a href="mailto:halitongen@gmail.com">halitongen@gmail.com</a>)

QTLtools-v1.3                                      06 May 2020                               <u><a href="../man1/QTLtools-bamstat.1.html">QTLtools-bamstat</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>