<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>QTLtools mbv - Match genotypes in a VCF to a BAM file</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/qtltools">qtltools_1.3.1+dfsg-4build4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       QTLtools mbv - Match genotypes in a VCF to a BAM file

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>QTLtools</b>  <b>mbv</b> <b>--bam</b> <b>[</b><u>sample.bam|sample.sam|sample.cram</u><b>]</b> <b>--vcf</b> <b>[</b><u>in.vcf</u>|<u>in.bcf</u>|<u>in.vcf.gz</u><b>]</b> <b>--out</b> <u>output_file</u>
       [<u>OPTIONS</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This mode checks if the genotypes in the VCF are observed in the RNAseq reads in the BAM file to  quickly
       solve  sample  mislabeling and detect cross-sample contamination and PCR amplification bias.  The details
       of the  method  are  described  &lt;https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6044394/&gt;.   In  brief,  we
       measure,  for  each  individual  in the VCF, the proportions of heterozygous and homozygous genotypes for
       which both alleles are captured by the sequencing reads in the BAM file.  A 'match' would have  close  to
       100%  concordance  for  both measures, whereas a 'mismatch' will have significantly lower concordance for
       both metrics.  Increased cross-sample contaminations leads to  decreased  homozygous  concordance  values
       with  no  change  in  heterozygous  concordance  while  increased  amplification  bias leads to decreased
       heterozygous concordance with no change in homozygous concordance.  We recommend using  uniquely  mapping
       reads only by specifying the correct <b>--filter-mapping-quality</b>.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>--vcf</b> <b>[</b><u>in.vcf</u>|<u>in.bcf</u>|<u>in.vcf.gz</u><b>]</b>
              Genotypes in VCF/BCF format.  Should contain all the samples in the dataset.  REQUIRED.

       <b>--bam</b> <b>[</b><u>in.bam</u>|<u>in.sam</u>|<u>in.cram</u><b>]</b>
              Sequence data in BAM/SAM/CRAM format.  REQUIRED.

       <b>--out</b> <u>output</u>
              Output file name REQUIRED.

       <b>--reg</b> <u>chr:start-end</u>
              Genomic region to be processed.  E.g. chr4:12334456-16334456, or chr5

       <b>--filter-mapping-quality</b> <u>integer</u>
              Minimum  mapping  quality  for  a  read  or  read pair to be considered.  Set this to only include
              uniquely mapped reads.  DEFAULT=10

       <b>--filter-base-quality</b> <u>integer</u>
              Minimum phred quality for a base to be considered.  DEFAULT=5

       <b>--filter-binomial-pvalue</b> <u>float</u>
              Binomial p-value threshold below which a heterozygous genotype is considered as exhibiting allelic
              imbalance.  DEFAULT=0.05

       <b>--filter-minimal-coverage</b> <u>integer</u>
              Minimum number of reads overlapping a genotype for it to be considered.  DEFAULT=10

       <b>--filter-imputation-qual</b> <u>float</u>
              Minimum imputation information score for a variant to be considered.  DEFAULT=0.9

       <b>--filter-imputation-prob</b> <u>float</u>
              Minimum posterior probability for a genotype to be considered.  DEFAULT=0.99

       <b>--filter-keep-duplicates</b>
              Keep reads designated as duplicate by the aligner.

</pre><h4><b>OUTPUT</b> <b>FILE</b> <b>COLUMNS</b></h4><pre>
       <b>--out</b> <u>filename</u>
        This file does not have header and it contains the following columns:

         1   The sample ID in the VCF against which the sequence data has been matched
         2   The number of missing genotypes for this sample
         3   The total number of heterozygous genotypes examined
         4   The total number of homozygous genotypes examined
         5   The number of heterozygous genotypes considered for the matching, i.e. those that  are  covered  by
             more than <b>--filter-minimal-coverage</b>
         6   The number of homozygous genotypes considered for the matching, i.e. those that are covered by more
             than <b>--filter-minimal-coverage</b>
         7   The number of heterozygous genotypes that match between this sample and the BAM file
         8   The number of homozygous genotypes that match between this sample and the BAM file
         9   The percentage of heterozygous genotypes that match between this sample and the BAM file
        10   The percentage of homozygous genotypes that match between this sample and the BAM file
        11   The number of heterozygous genotypes with significant allelic imbalance

</pre><h4><b>EXAMPLES</b></h4><pre>
       o Running mbv on an RNAseq sample mapped with GEM:

         QTLtools   mbv   --bam   HG00381.chr22.bam  --out  HG00381.chr22.mbv.txt  --vcf  genotypes.chr22.vcf.gz
         --filter-mapping-quality 150

         You can then plot column 9 vs. 10 to identify the genotyped sample in the VCF that  matches  best  your
         sequence data.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/QTLtools.1.html">QTLtools</a></u>(1)

       QTLtools website: &lt;https://qtltools.github.io/qtltools&gt;

</pre><h4><b>BUGS</b></h4><pre>
       Please submit bugs to &lt;https://github.com/qtltools/qtltools&gt;

</pre><h4><b>CITATION</b></h4><pre>
       Fort A., Panousis N. I., Garieri M. et al. MBV: a method to solve sample mislabeling and detect technical
       bias  in  large  combined  genotype  and  sequencing  assay  datasets,  <u>Bioinformatics</u> <b><a href="../man12/33.12.html">33</a></b>(12), 1895 2017.
       &lt;https://doi.org/10.1093/bioinformatics/btx074&gt;

</pre><h4><b>AUTHORS</b></h4><pre>
       Olivier Delaneau (<a href="mailto:olivier.delaneau@gmail.com">olivier.delaneau@gmail.com</a>), Halit Ongen (<a href="mailto:halitongen@gmail.com">halitongen@gmail.com</a>)

QTLtools-v1.3                                      06 May 2020                                   <u><a href="../man1/QTLtools-mbv.1.html">QTLtools-mbv</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>