<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>QTLtools ase - Measure ASE from RNA-seq</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/qtltools">qtltools_1.3.1+dfsg-4build4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       QTLtools ase - Measure ASE from RNA-seq

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>QTLtools</b>   <b>ase</b>   <b>--bam</b>   <b>[</b><u>sample.bam|sample.sam|sample.cram</u><b>]</b>   <b>--vcf</b>  <b>[</b><u>in.vcf</u>|<u>in.bcf</u>|<u>in.vcf.gz</u><b>]</b>  <b>--sample</b>
       <u>sample_name_in_vcf</u> <b>--mapq</b>  <u>integer</u> <b>--out</b> <u>output_file_prefix</u> [<u>OPTIONS</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This mode measures allele specific expression (ASE) from RNAseq  for  transcribed  heterozygous  SNPs  as
       detailed  in  &lt;https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3918453/&gt;.   In  brief,  the reference allele
       mapping bias is calculated for each of the 12 REF/ALT pairs separately provided that there are  at  least
       <b>--sites</b>  number of REF/ALT sites that have a minimum <b>--cov-bias</b> number of reads overlapping.  For REF/ALT
       pairs that fail these criteria, reference allele mapping bias is calculated from  all  sites.   Reference
       allele  mapping bias is the number of all reads across many sites that contained the reference allele for
       a given REF/ALT pair over the total number of reads overlapping.  <b>--subsample</b> controls  which  percentile
       of the highest covered sites are subsampled, which is necessary so that the reference allele mapping bias
       is  not  estimated mostly from very high coverage sites, but from sites covering all the genome.  The ASE
       p-value for each site is then calculated as a two-tailed binomial test checking if the observed number of
       reference allele reads is significantly different from random, given the total number of reads,  and  the
       probability  of  observing  a  reference allele, which is the reference allele mapping bias for a certain
       REF/ALT pair.

       The defaults for options should work well for most RNAseq experiments.  It is <b>NOT</b> advisable  to  decrease
       the  <b>--baseq</b> below 10, <b>--cov</b> and <b>--cov-bias</b> below 8, and setting <b>--subsample</b> to 1.  It is <b>NOT</b> recommended
       to use  the  following  options  regarding  filtering  <b>--keep-bans-for-bias</b>,  <b>--keep-discordant-for-bias</b>,
       <b>--filter-duplicates</b>,  <b>--ignore-orientation</b>,  and  <b>--legacy-options</b>.  Also refrain from using <b>--auto-flip</b>,
       but rather create correct VCF/BCF files.

       We highly recommended using a BCF file rather than a VCF due to performance benefits, <b>--fasta</b> to  provide
       the  genome sequence used, <b>--blacklist</b> to filter low mappability regions, and <b>--gtf</b> to annotate the SNPs.
       If you are using unfiltered imputed genotypes  then  consider  using  <b>--imp-qual</b>  and  <b>--geno-prob</b>.   For
       trouble  shooting purposes you can use <b>--filtered</b>, which will list why certain variants are filtered from
       the analysis.  If you are I/O bound you may  try  using  <b>--group-by</b>  for  better  performance.   If  your
       blacklist  file  contains  many  overlapping or contiguous regions you can decrease the memory usage with
       <b>--merge-on-the-fly</b>.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>--vcf,</b> <b>-v</b> <b>[</b><u>in.vcf</u>|<u>in.bcf</u>|<u>in.vcf.gz</u><b>]</b>
              Genotypes in VCF/BCF format sorted by chromosome and then position.  Can contain multiple samples.
              REQUIRED and RECOMMENDED to use a BCF file for performance.

       <b>--bam,</b> <b>-b</b> <b>[</b><u>in.bam</u>|<u>in.sam</u>|<u>in.cram</u><b>]</b>
              Sequence data in BAM/SAM/CRAM format sorted by chromosome and then position.  One sample  per  BAM
              file.  REQUIRED.

       <b>--ind,</b> <b>-i</b> <u>sample_name</u>
              Sample name in the VCF corresponding to the BAM file.  REQUIRED.

       <b>--out,</b> <b>-o</b> <u>output</u>
              Output prefix.  This will generate output.ase and output.ref_bias files.  Or if you give output.gz
              or output.bz2 then output.ase.gz etc.  REQUIRED.

       <b>--mapq,</b> <b>-q</b> <u>integer</u>
              Minimum  mapping  quality  for  a  read  or  read pair to be considered.  Set this to only include
              uniquely mapped reads.  REQUIRED.

       <b>--fasta,</b> <b>-f</b> <u>genome_sequence.fa</u>
              Genome sequence in FASTA format.  Make sure this  the  sequence  for  the  correct  genome  build.
              RECOMMENDED.

       <b>--blacklist,</b> <b>-B</b> <u>poor_mappability_regions.bed</u>
              Poor  mappability  regions  in  BED  format.  ASE estimates will be unreliable in regions with low
              mappability.  RECOMMENDED.

       <b>--merge-on-the-fly,</b> <b>-t</b>
              Merges the blacklisted regions on the fly.  This  can  reduce  memory  usage  if  there  are  many
              overlapping or contiguous regions in the blacklist.

       <b>--gtf,</b> <b>-g</b> <u>gene_annotation.gtf</u>
              Gene  annotations  in  GTF  format.   These  can be obtained from &lt;https://www.gencodegenes.org/&gt;.
              RECOMMENDED.

       <b>--filtered,</b> <b>-l</b> <u>filename</u>
              File to output filtered variants.   RECOMMENDED  for  troubleshooting  especially  if  <b>--suppress-</b>
              <b>warnings.</b>

       <b>--reg,</b> <b>-r</b> <u>chr:start-end</u>
              Genomic region to be processed.  E.g. chr4:12334456-16334456, or chr5

       <b>--fix-chr,</b> <b>-F</b>
              Attempt  to  match chromosome names to the BAM file by adding or removing chr to chromosome names.
              Does not apply to --{include,exclude}-positions options.  These should be in  the  VCF  chromosome
              names.

       <b>--fix-id,</b> <b>-R</b>
              Convert missing VCF variant IDs to chr_pos_refalt.

       <b>--auto-flip,</b> <b>-x</b>
              Attempt  to  fix reference allele mismatches.  Requires a fasta file for the reference sequence by
              <b>--fasta</b>.  NOT RECOMMENDED.

       <b>--print-stats,</b> <b>-P</b>
              Print out stats for the filtered reads for ASE sites.  This will be slower and if there are  sites
              with more reads than <b>--max-depth</b>, then will potentially give different results.

       <b>--suppress-warnings,</b> <b>-k</b>
              Suppress the warnings about individual variants.

       <b>--illumina13,</b> <b>-j</b>
              Base quality is in the Illumina-1.3+ encoding.

       <b>--group-by,</b> <b>-G</b> <u>integer</u>
              Group  variants  separated by this much into batches.  This allows you not to stream the whole BAM
              file and may improve running time.

       <b>--max-depth,</b> <b>-d</b> <u>integer</u>
              Pileup max-depth DEFAULT=1000000.  Set to more  reasonable  value  if  you  experience  memory  or
              performance  issues.   This is set to a high value by default since with RNA-seq you can have very
              high coverage sites.  Set to 0 for max.

       <b>--baseq,</b> <b>-Q</b> <u>integer</u>
              Minimum phred quality for a base to be considered DEFAULT=10.

       <b>--pvalue,</b> <b>-p</b> <u>float</u>
              Binomial p-value threshold for ASE output DEFAULT=1.0.

       <b>--cov,</b> <b>-c</b> <u>integer</u>
              Minimum coverage for a genotype to be considered in ASE analysis DEFAULT=16.

       <b>--cov-bias,</b> <b>-C</b> <u>integer</u>
              Minimum coverage for a genotype to  be  considered  in  reference  allele  mapping  bias  analysis
              DEFAULT=10.

       <b>--sites,</b> <b>-s</b> <u>integer</u>
              Minimum  number  of sites to calculate a reference allele mapping bias from for a specific REF/ALT
              pair DEFAULT=200.  The reference allele mapping bias for pairs with less than this many sites will
              be calculated from all sites.

       <b>--imp-qual-id,</b> <b>-I</b> <u>string</u>
              The VCF INFO field ID of the imputation score in the VCF DEFAULT=INFO.

       <b>--geno-prob-id,</b> <b>-L</b> <u>string</u>
              The VCF FORMAT field  ID  of  the  genotype  posterior  probabilities  for  RR/RA/AA  in  the  VCF
              DEFAULT=GP.

       <b>--imp-qual,</b> <b>-W</b> <u>float</u>
              Minimum imputation score for a variant to be considered DEFAULT=0.0.

       <b>--geno-prob,</b> <b>-V</b> <u>float</u>
              Minimum posterior probability for a genotype to be considered DEFAULT=0.0.

       <b>--subsample,</b> <b>-S</b> <u>float</u>
              Randomly  subsample  sites  that  have  greater  coverage than this percentile of all the sites in
              reference allele mapping bias calculations DEFAULT=0.75.  Set to  1  to  turn  off  which  is  NOT
              RECOMMENDED.  Set to 0 to subsample all sites to <b>--cov-bias</b>.

       <b>--both-alleles-seen,</b> <b>-a</b>
              Require both alleles to be observed in RNA-seq reads for a site for ASE calculations.

       <b>--keep-bans-for-bias,</b> <b>-A</b>
              DON'T  require both alleles to be observed in RNAseq reads for a site for reference allele mapping
              bias calculations.  NOT RECOMMENDED.

       <b>--keep-discordant-for-bias,</b> <b>-E</b>
              If given, sites with more discordant alleles than REF or ALT  alleles  will  be  included  in  the
              reference allele mapping bias bias calculations.  NOT RECOMMENDED.

       <b>--filter-indel-reads,</b> <b>-D</b>
              Remove reads that contain indels.

       <b>--keep-failed-qc,</b> <b>-e</b>
              Keep fastq reads that fail sequencing QC (as indicated by the sequencer).

       <b>--keep-orphan-reads,</b> <b>-O</b>
              Keep paired end reads where one of mates is unmapped.

       <b>--check-proper-pairing,</b> <b>-y</b>
              If provided only properly paired reads according to the aligner will be considered.

       <b>--ignore-orientation,</b> <b>-X</b>
              If  NOT  provided  only mate pairs where both mates are on the same chromosome and where the first
              mate is on the +ve strand  and  the  second  is  on  the  -ve  strand  will  be  considered.   NOT
              RECOMMENDED.

       <b>--filter-duplicates,</b> <b>-u</b>
              Remove reads designated as duplicate by the aligner.  NOT RECOMMENDED.

       <b>--filter-supp,</b> <b>-m</b>
              Remove supplementary (non-linear) alignments.

       <b>--legacy-options,</b> <b>-J</b>
              Replicate legacy options used.  NOT RECOMMENDED.

</pre><h4><b>OUTPUT</b> <b>FILES</b></h4><pre>
       <b>.ase</b>
        This  file  is  the main ASE results file with the following columns.  Columns after the 23rd column are
        only printed if <b>--print-stats</b> is provided.

         1   INDIVIDUAL              The sample id
         2   RSID                    The SNP ID from the VCF file
         3   CHR                     Chromosome of the SNP
         4   POS                     Position of the SNP
         5   ALLELES                 The SNP's alleles
         6   BOTH_ALLELES_SEEN       Whether or not both of the SNP's alleles were seen in the RNAseq reads
         7   MIN_ALLELE_RATIO        The minor allele ration among RNAseq reads
         8   REF_COUNT               Number of reference alleles in reads
         9   NONREF_COUNT            Number of alternative alleles in reads
        10   TOTAL_COUNT             Number of read overlapping the SNP
        11   WEIGHTED_REF_COUNT      REF_COUNT adjusted for ref mapping bias
        12   WEIGHTED_NONREF_COUNT   NONREF_COUNT adjusted for ref mapping bias
        13   WRC_MINUS_WNC           WEIGHTED_REF_COUNT - WEIGHTED_NONREF_COUNT
        14   ALLELES_SEEN            Alleles observed in RNAseq reads
        15   REF_ALLELE              Reference allele
        16   ALT_ALLELE              Alternative allele
        17   OTHER_COUNT             Number of discordant reads (not REF or ALT)
        18   EXPECTED_DISCORDANT     Expected number of discordant alleles.  Calculated by  adding  up  all  the
                                     base error probabilities of RNAseq read positions overlapping the SNP
        19   DISCORDANT_PVAL         P-value  for  observed number of discordant reads being more than expected.
                                     One can Bonferroni correct these p-values, and exclude the significant ones
                                     from downstream analyses.
        20   REF_RATIO               Reference allele mapping bias
        21   PVALUE                  ASE p-value
        22   CONCERN                 If there were any concerns potentially rendering this  SNP  unusable,  they
                                     will be coded here.  See below <b>OUTPUT</b> <b>FILE</b> <b>CODES</b> for how to decode
        23   EXON_INFO               Exons  that overlap this SNP will be listed here if a GTF file is provided.
                                     Exon names, which are formed by concatenating gene id, transcript id,  exon
                                     position,  and  gene  name  delimited  with  a  colon,   are separated with
                                     semicolons
        24   SECONDARY               The number of secondary alignments that were filtered out
        25   SUPPLEMENTARY           If <b>--filter-supp</b> is provided, then the number of  supplementary  alignments
                                     that were filtered out
        26   FAIL_MAPQ               The number of alignments that were filtered out due to low mapping quality
        27   FAILQC                  If <b>--keep-failed-qc</b> is not provided, then the number of filtered reads that
                                     failed qc according to the sequencer
        28   DUPLICATE               If  <b>--filter-duplicates</b> is provided, then the number of filtered reads that
                                     were labeled as duplicate by the aligner
        29   MATE_UNMAPPED           If <b>--keep-orphan-reads</b> is not provided, then the number of reads that  were
                                     filtered since one mate was unmapped
        30   WRONG_ORIENTATION       If <b>--ignore-orientation</b> is not provided, then the number of reads that were
                                     filtered since they were in the wrong orientation
        31   NOT_PROPER_PAIR         If  <b>--check-proper-pairing</b>  is  provided, then the number of filtered reads
                                     that were not properly paired according to the aligner
        32   SKIPPED                 The number of alignments that were filtered out since  they  did  not  have
                                     actual bases overlapping the SNP
        33   FAIL_BASEQ              The  number of filtered reads where the base overlapping the SNP had a base
                                     quality less than <b>--baseq</b>
        34   INDEL                   If <b>--filter-indel-reads</b> is provided, then the number of filtered reads that
                                     contained indels
        35   DEPTH                   The number alignments that overlap with the SNP position

       <b>.ref_bias</b>
        Details the reference allele mapping bias results, and has the following columns:

        1   INDIVIDUAL         The sample id
        2   ALLELES            The reference alternative allele pair
        3   REF_ALL            Total number of reference alleles observed across all sites
        4   NONREF_ALL         Total number of alternative alleles observed across all sites
        5   SITES              Number of sites for this REF/ALT pair that pass the thresholds
        6   SUBSAMPLED_SITES   Number of sites that were subsampled to SUBSAMPLED_TO since they had too  high  a
                               coverage
        7   SUBSAMPLED_TO      The coverage SUBSAMPLED_SITES were subsampled to
        8   PERC               Reference allele mapping bias
        9   SOURCE             Whether  the  reference  allele  mapping bias was calculated from allele specific
                               sites of all sites

       <b>--filtered</b> <u>filename</u>
        This file lists the variants that were omitted from the analysis and why.  The first column is the coded
        omission reason, followed by the variant ID.  The codes and their meaning are listed  in  the  following
        section.

</pre><h4><b>OUTPUT</b> <b>FILE</b> <b>CODES</b></h4><pre>
       <b>.ase</b> <b>file</b> <b>codes</b> <b>for</b> <b>the</b> <b>CONCERN</b> <b>column</b>
        You  probably  want  to  exclude  SNPs  with RM, NRA, MDTA, or MDTR concerns from your analyses as these
        likely have wrong genotypes.  SNPs with other concern may be OK

        RM     Reference allele in the VCF file mismatches the reference sequence.  Requires <b>--fasta</b>
        DP     Multiple variants observed at the same position in the VCF file
        NRA    No reference or alternative allele observed in the RNAseq reads
        MDTA   More discordant (not reference or alternative) alleles than alternative  alleles  in  the  RNAseq
               reads
        MDTR   More discordant (not reference or alternative) alleles than reference alleles in the RNAseq reads
        BANS   Both alleles of the SNP were not observed in RNAseq reads
        LMAR   Among the RNAseq reads the minor allele ratio was less than 2%
        PD     Pileup  depth  (<b>--max-depth</b>)  was  potentially  exceeded  around this SNP.  This may prevent some
               overlapping reads from being counted.  We recommend rerunning with a higher <b>--max-depth</b>.

       <b>--filtered</b> <b>file</b> <b>codes</b>
        VMA      Multi-allelic variant
        VU       Variant position or ID is excluded by the user
        VCNIB    Variant chromosome is not in the BAM file
        VB       Variant is in a blacklisted region
        VI       Variant is an indel
        VMRA     Variant is missing either the reference or the alternative allele
        VNIF     Variant is not in the reference genome.  Only if <b>--fasta</b> was provided
        VS       Reference and alternative alleles were  swapped   (not  excluded).   Only  if  <b>--auto-flip</b>  and
                 <b>--fasta</b> were provided
        VF       Reference and alternative alleles' strand was flipped  (not excluded).  Only if <b>--auto-flip</b> and
                 <b>--fasta</b> were provided
        VWR      Reference  allele  does not match the reference sequence.  Only if <b>--auto-flip</b> and <b>--fasta</b> were
                 provided
        VBI      Variant failed the imputation quality filter
        VMGT     Variant with missing GT field or variant is not diploid
        VMG      Missing genotype
        VH       Homozygous variant
        VBG      Variant failed the genotype probability filter
        VD       Duplicate variant position
        VMI      Variant with a missing ID (not excluded).  Will be renamed if <b>--fix-id</b> is provided
        VDK      Heterozygous variant has another variant with the same position (not excluded)
        BC       Variant did not have enough coverage for reference allele mapping bias calculations
        BBANS    Both alleles were not observed for reference allele mapping bias calculations
        BMDTRA   Variant had more discordant alleles than reference or alternative  alleles  thus  was  excluded
                 from reference allele mapping bias calculations
        AC       Variant did not have enough coverage ASE calculations
        ABANS    Both alleles were not observed for ASE calculations

</pre><h4><b>EXAMPLES</b></h4><pre>
       o ASE analysis of a sample mapped with STAR:

         QTLtools ase --vcf multi_sample.bcf --bam sample1.bam --ind sample1 --mapq 255 --out sample1 --filtered
         sample1.filtered.gz   --gtf   gencode.v19.annotation.gtf.gz   --blacklist  poor_mappability_regions.bed
         --fasta hg19.fa --fix-chr --fix-id

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/QTLtools.1.html">QTLtools</a></u>(1)

       QTLtools website: &lt;https://qtltools.github.io/qtltools&gt;

</pre><h4><b>BUGS</b></h4><pre>
       Please submit bugs to &lt;https://github.com/qtltools/qtltools&gt;

</pre><h4><b>CITATION</b></h4><pre>
       Delaneau, O., Ongen, H., Brown, A. et al. A complete tool set for molecular QTL discovery  and  analysis.
       <u>Nat</u> <u>Commun</u> <b>8</b>, 15452 (2017).  &lt;https://doi.org/10.1038/ncomms15452&gt;

</pre><h4><b>AUTHORS</b></h4><pre>
       Halit Ongen (<a href="mailto:halitongen@gmail.com">halitongen@gmail.com</a>), Olivier Delaneau (<a href="mailto:olivier.delaneau@gmail.com">olivier.delaneau@gmail.com</a>)

QTLtools-v1.3                                      06 May 2020                                   <u><a href="../man1/QTLtools-ase.1.html">QTLtools-ase</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>