<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>map_gff_coords  -  Map  coordinates  of GFF features from one genome to another using a Mercator multiple</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/fsa">fsa_1.15.9+dfsg-6_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       map_gff_coords  -  Map  coordinates  of GFF features from one genome to another using a Mercator multiple
       alignment

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>map_gff_coords</b> [<u>options</u>] <u>&lt;source</u> <u>genome&gt;</u> <u>&lt;source</u> <u>GFF</u> <u>file&gt;</u> <u>&lt;target</u> <u>genome&gt;</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       map_gff_coords from FSA 1.15.9

       Map coordinates of GFF features from one genome to another using a Mercator multiple alignment.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              show this message

       <b>-t</b>, <b>--type</b> &lt;string&gt;
              only look at features of particular type

       <b>-D</b>, <b>--data</b> &lt;directory&gt;
              path to map, genome and alignment files

       <b>-M</b>, <b>--map</b> &lt;directory&gt;
              path to map and genome files

       <b>-A</b>, <b>--align</b> &lt;directory&gt;
              path to alignment files

       <b>-L</b>, <b>--lazy</b>
              warn, rather than die, if the subalignment can't be obtained

       <b>-U</b>, <b>--truncate</b>
              truncate unmappable sequence (rather than skipping) and show truncated subalignment

       <b>-f</b>, <b>--force-entry</b>
              if a feature can't be mapped, then add an empty entry to the GFF file  (rather  than  skipping  it
              entirely); implies <b>--lazy</b>

       <b>-e</b>, <b>--verbose</b>
              report progress

       PLEASE  NOTE:  While this program is reasonably fast if the GFF is properly ordered by chromosome and the
       start and end coordinates of features, it will be *very slow* if the GFF is not sorted.  Assumes that the
       "seqid" or "name" field (the first field) of the GFF entries holds the chromosome name.

       Note  that  the  GFF  specification  defines  coordinates  to  be  1-based  and  fully-closed,  therefore
       representing the interval [start, end].  Conformance to this specification is assumed internally.

       If  requested,  unmappable  sequence  will be truncated to the mappable portion; note that the truncation
       will favor the beginning of the requested sequence.

       If a GFF feature is on the + strand for the source genome but the corresponding homologous region in  the
       target genome is on the - strand, then the mapped GFF feature will be reported as on the - strand.

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

map_gff_coords 1.15.9                             December 2015                                <u><a href="../man1/MAP_GFF_COORDS.1.html">MAP_GFF_COORDS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>