<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>diamond - accelerated BLAST compatible local sequence aligner</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/diamond-aligner">diamond-aligner_2.1.11-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       diamond - accelerated BLAST compatible local sequence aligner

</pre><h4><b>DESCRIPTION</b></h4><pre>
       diamond  v2.0.13.151  (C)  Max  Planck  Society for the Advancement of Science Documentation, support and
       updates         available         at          <a href="http://www.diamondsearch.org">http://www.diamondsearch.org</a>          Please          cite:
       <a href="http://dx.doi.org/10.1038/s41592-021-01101-x">http://dx.doi.org/10.1038/s41592-021-01101-x</a> Nature Methods (2021)

       Syntax: diamond COMMAND [OPTIONS]

       Commands:  makedb   Build  DIAMOND  database  from  a FASTA file blastp  Align amino acid query sequences
       against a protein reference database blastx  Align  DNA  query  sequences  against  a  protein  reference
       database view    View DIAMOND alignment archive (DAA) formatted file help    Produce help message version
       Display  version  information  getseq   Retrieve  sequences  from  a  DIAMOND database file dbinfo  Print
       information about a DIAMOND database file test    Run regression tests makeidx Make database index

       General options: <b>--threads</b> (<b>-p</b>)           number of CPU threads <b>--db</b>  (<b>-d</b>)                 database  file
       <b>--out</b> (<b>-o</b>)               output file <b>--outfmt</b> (<b>-f</b>)            output format

       0      = BLAST pairwise

       5      = BLAST XML

       6      = BLAST tabular

              100 = DIAMOND alignment archive (DAA) 101 = SAM

              Value 6 may be followed by a space-separated list of these keywords:

              qseqid  means  Query  Seq  -  id  qlen  means  Query sequence length sseqid means Subject Seq - id
              sallseqid means All subject Seq - id(s), separated by a ';' slen  means  Subject  sequence  length
              qstart  means  Start of alignment in query qend means End of alignment in query sstart means Start
              of alignment in subject send means End of alignment in subject qseq means Aligned  part  of  query
              sequence  qseq_translated  means Aligned part of query sequence (translated) full_qseq means Query
              sequence full_qseq_mate means Query sequence of the  mate  sseq  means  Aligned  part  of  subject
              sequence full_sseq means Subject sequence evalue means Expect value bitscore means Bit score score
              means  Raw score length means Alignment length pident means Percentage of identical matches nident
              means Number of identical matches mismatch means Number of mismatches  positive  means  Number  of
              positive  -  scoring  matches gapopen means Number of gap openings gaps means Total number of gaps
              ppos means Percentage of positive - scoring matches qframe means  Query  frame  btop  means  Blast
              traceback  operations(BTOP)  cigar means CIGAR string staxids means unique Subject Taxonomy ID(s),
              separated by a ';' (in  numerical  order)  sscinames  means  unique  Subject  Scientific  Name(s),
              separated  by a ';' sskingdoms means unique Subject Super Kingdom(s), separated by a ';' skingdoms
              means unique Subject Kingdom(s), separated by a  ';'  sphylums  means  unique  Subject  Phylum(s),
              separated  by a ';' stitle means Subject Title salltitles means All Subject Title(s), separated by
              a '&lt;&gt;' qcovhsp means Query Coverage Per HSP scovhsp means Subject Coverage Per  HSP  qtitle  means
              Query  title  qqual  means Query quality values for the aligned part of the query full_qqual means
              Query quality values qstrand means Query strand

              Default: qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore

       <b>--verbose</b> (<b>-v</b>)            verbose  console  output  <b>--log</b>                     enable  debug  log  <b>--quiet</b>
       disable console output <b>--header</b>                 Write header lines to blast tabular format.

       Makedb   options:   <b>--in</b>                       input   reference   file   in   FASTA   format  <b>--taxonmap</b>
       protein  accession  to  taxid  mapping  file  <b>--taxonnodes</b>              taxonomy  nodes.dmp   from   NCBI
       <b>--taxonnames</b>             taxonomy names.dmp from NCBI

       Aligner  options:  <b>--query</b>  (<b>-q</b>)              input  query file <b>--strand</b>                 query strands to
       search    (both/minus/plus)    <b>--un</b>                        file    for     unaligned     queries     <b>--al</b>
       file  or  aligned  queries  <b>--unfmt</b>                  format of unaligned query file (fasta/fastq) <b>--alfmt</b>
       format of aligned query file  (fasta/fastq)  <b>--unal</b>                    report  unaligned  queries  (0=no,
       1=yes)  <b>--max-target-seqs</b> (<b>-k</b>)   maximum number of target sequences to report alignments for (default=25)
       <b>--top</b>                    report alignments within this percentage range of top alignment score (overrides
       <b>--max-target-seqs</b>) <b>--max-hsps</b>               maximum number of HSPs per target sequence to report for each
       query (default=1) <b>--range-culling</b>          restrict hit culling to overlapping  query  ranges  <b>--compress</b>
       compression  for  output  files (0=none, 1=gzip, zstd) <b>--evalue</b> (<b>-e</b>)            maximum e-value to report
       alignments (default=0.001) <b>--min-score</b>              minimum bit score  to  report  alignments  (overrides
       e-value  setting)  <b>--id</b>                      minimum  identity%  to  report  an  alignment  <b>--query-cover</b>
       minimum query cover% to report an alignment <b>--subject-cover</b>          minimum subject cover% to report  an
       alignment  <b>--fast</b>                    enable  fast mode <b>--mid-sensitive</b>          enable mid-sensitive mode
       <b>--sensitive</b>              enable sensitive  mode)  <b>--more-sensitive</b>          enable  more  sensitive  mode
       <b>--very-sensitive</b>          enable very sensitive mode <b>--ultra-sensitive</b>        enable ultra sensitive mode
       <b>--iterate</b>                iterated search with increasing sensitivity <b>--global-ranking</b> (<b>-g</b>)     number  of
       targets  for  global  ranking  <b>--block-size</b>  (<b>-b</b>)         sequence  block  size  in  billions  of letters
       (default=2.0) <b>--index-chunks</b> (<b>-c</b>)      number of chunks for index processing  (default=4)  <b>--tmpdir</b>  (<b>-t</b>)
       directory   for   temporary   files  <b>--parallel-tmpdir</b>         directory  for  temporary  files  used  by
       multiprocessing <b>--gapopen</b>                gap open penalty <b>--gapextend</b>              gap extension  penalty
       <b>--frameshift</b>  (<b>-F</b>)         frame  shift  penalty  (default=disabled)  <b>--long-reads</b>              short for
       <b>--range-culling</b>  <b>--top</b>  10  <b>-F</b>  15  <b>--matrix</b>                   score   matrix   for   protein   alignment
       (default=BLOSUM62)  <b>--custom-matrix</b>           file  containing  custom  scoring matrix <b>--comp-based-stats</b>
       composition  based  statistics  mode  (0-4)  <b>--masking</b>                 masking  algorithm   (none,   seg,
       tantan=default)  <b>--query-gencode</b>           genetic  code  to  use  to  translate  query (see user manual)
       <b>--salltitles</b>             include full subject titles in DAA  file  <b>--sallseqid</b>               include  all
       subject  ids  in  DAA file <b>--no-self-hits</b>           suppress reporting of identical self hits <b>--taxonlist</b>
       restrict search to list of taxon ids (comma-separated) <b>--taxon-exclude</b>          exclude list of taxon ids
       (comma-separated)   <b>--seqidlist</b>                filter   the    database    by    list    of    accessions
       <b>--skip-missing-seqids</b>    ignore accessions missing in the database

       Advanced          options:         <b>--algo</b>                           Seed         search         algorithm
       (0=double-indexed/1=query-indexed/ctg=contiguous-seed) <b>--bin</b>                    number of query bins  for
       seed  search  <b>--min-orf</b>  (<b>-l</b>)            ignore  translated sequences without an open reading frame of at
       least this length <b>--seed-cut</b>               cutoff for seed complexity <b>--freq-masking</b>           mask seeds
       based on frequency <b>--freq-sd</b>                number of standard deviations  for  ignoring  frequent  seeds
       <b>--motif-masking</b>           softmask  abundant  motifs  (0/1)  <b>--id2</b>                     minimum  number of
       identities for stage 1 hit <b>--xdrop</b> (<b>-x</b>)             xdrop for ungapped  alignment  <b>--gapped-filter-evalue</b>
       E-value  threshold  for  gapped  filter  (auto)  <b>--band</b>                    band  for  dynamic programming
       computation  <b>--shapes</b>  (<b>-s</b>)             number  of  seed  shapes  (default=all  available)   <b>--shape-mask</b>
       seed   shapes   <b>--multiprocessing</b>          enable   distributed-memory   parallel   processing  <b>--mp-init</b>
       initialize   multiprocessing   run   <b>--mp-recover</b>               enable   continuation   of    interrupted
       multiprocessing   run   <b>--mp-query-chunk</b>           process   only  a  single  query  chunk  as  specified
       <b>--ext-chunk-size</b>         chunk size for adaptive ranking (default=auto) <b>--no-ranking</b>              disable
       ranking     heuristic     <b>--ext</b>                       Extension    mode    (banded-fast/banded-slow/full)
       <b>--culling-overlap</b>        minimum range overlap with higher scoring hit  to  delete  a  hit  (default=50%)
       <b>--taxon-k</b>                   maximum   number   of   targets   to   report   per   species   <b>--range-cover</b>
       percentage   of   query   range   to   be   covered   for   range    culling    (default=50%)    <b>--dbsize</b>
       effective database size (in letters) <b>--no-auto-append</b>         disable auto appending of DAA and DMND file
       extensions  <b>--xml-blord-format</b>        Use  gnl|BL_ORD_ID|  style  format in XML output <b>--stop-match-score</b>
       Set the match  score  of  stop  codons  against  each  other.   <b>--tantan-minMaskProb</b>      minimum  repeat
       probability   for   masking   (default=0.9)   <b>--file-buffer-size</b>         file   buffer   size   in  bytes
       (default=67108864)  <b>--memory-limit</b>  (<b>-M</b>)       Memory  limit  for  extension  stage  in  GB   <b>--no-unlink</b>
       Do  not  unlink  temporary  files.  <b>--target-indexed</b>         Enable target-indexed mode <b>--ignore-warnings</b>
       Ignore warnings

       View options: <b>--daa</b> (<b>-a</b>)               DIAMOND alignment archive (DAA) file <b>--forwardonly</b>            only
       show alignments of forward strand

       Getseq options: <b>--seq</b>                    Space-separated list of sequence numbers to display.

       Online documentation at <a href="http://www.diamondsearch.org">http://www.diamondsearch.org</a>

</pre><h4><b>AUTHOR</b></h4><pre>
        This manpage was written by Nilesh Patra for the Debian distribution and
        can be used for any other usage of the program.

diamond 2.0.13                                    October 2021                                        <u><a href="../man1/DIAMOND.1.html">DIAMOND</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>