<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>insegt - INtersecting SEcond Generation sequencing daTa with annotation</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/seqan-apps">seqan-apps_2.4.0+dfsg-16_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       insegt - INtersecting SEcond Generation sequencing daTa with annotation

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>insegt</b> [<u>OPTIONS</u>] &lt;<u>ALIGMENTS-FILE</u>&gt; &lt;<u>ANNOTATIONS-FILE</u>&gt;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       INSEGT  is  a  tool  to  analyze  alignments  of  RNA-Seq reads (single-end or paired-end) by using gene-
       annotations.

       Input to INSEGT is a SAM file containing the alignments and a file  containing  the  annotations  of  the
       reference genome, either in GFF or GTF format.

</pre><h4><b>REQUIRED</b> <b>ARGUMENTS</b></h4><pre>
       <b>ARGUMENT</b> <b>0</b> <u>INPUT_FILE</u>
               Valid filetype is: <u>.sam</u>.

       <b>ARGUMENT</b> <b>1</b> <u>INPUT_FILE</u>
               Valid filetypes are: <u>.gtf</u> and <u>.gff</u>.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              Display the help message.

       <b>--version</b>
              Display version information.

   <b>Options:</b> <b>:</b>
       <b>-ro</b>, <b>--read-output</b> <u>OUTPUT_FILE</u>
              Output  filename  for  read-output, which contains the mapped annotations followed by their parent
              annotation. Valid filetype is: <u>.gff</u>.

       <b>-ao</b>, <b>--anno-output</b> <u>OUTPUT_FILE</u>
              Output filename for anno-output, which contains the annotations  similar  to  the  GFF  input  and
              additionally  the  counts  of the mapped reads and the normalized expression levels in RPKM. Valid
              filetype is: <u>.gff</u>.

       <b>-to</b>, <b>--tuple-output</b> <u>OUTPUT_FILE</u>
              Output filename for tuple-output, which contains exon tuples  connected  by  reads  or  matepairs.
              Valid filetype is: <u>.gff</u>.

       <b>-fo</b>, <b>--fusion-output</b> <u>STRING</u>
              Output  filename  for  fusion-output,  which contains exon tuple of gene fusions (Advanced option,
              currently no output port for KNIME). One of <u>gff</u>.

       <b>-n</b>, <b>--ntuple</b> <u>INTEGER</u>
              ntuple Default: <u>2</u>.

       <b>-o</b>, <b>--offset-interval</b> <u>INTEGER</u>
              Offset to short alignment-intervals for search. Default: <u>5</u>.

       <b>-t</b>, <b>--threshold-gaps</b> <u>INTEGER</u>
              Threshold for allowed gaps in alignment (not introns). Default: <u>5</u>.

       <b>-c</b>, <b>--threshold-count</b> <u>INTEGER</u>
              Threshold for min. count of tuple for output. Default: <u>1</u>.

       <b>-r</b>, <b>--threshold-rpkm</b> <u>DOUBLE</u>
              Threshold for min. RPKM of tuple for output. Default: <u>0.0</u>.

       <b>-m</b>, <b>--max-tuple</b>
              Create only maxTuple (which are spanned by the whole read).

       <b>-e</b>, <b>--exact-ntuple</b>
              Create only Tuple of exact length n. By default all tuple up to the given length are computed  (if
              -m is set, -e will be ignored).

       <b>-u</b>, <b>--unknown-orientation</b>
              Orientation of reads is unknown.

</pre><h4><b>EXAMPLES</b></h4><pre>
       <b>insegt</b>  <b>example/alignments.sam</b> <b>example/annotations.gff</b>
              Run INSEGT on example files with default parameters.

       <b>insegt</b> <b>-m</b> <b>example/alignments.sam</b> <b>example/annotations.gff</b>
              Run INSEGT on example files and only compute maxTuple.

       <b>insegt</b> <b>-c</b> <b>2</b> <b>example/alignments.sam</b> <b>example/annotations.gff</b>
              Run INSEGT on example files and only output tuple with a min. count of 2.

insegt 1.1.10 [tarball]                                                                                <u><a href="../man1/INSEGT.1.html">INSEGT</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>