<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>bp_search2alnblocks - Turn SearchIO parseable reports(s) into a set of aligned blocks</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/bioperl">bioperl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       bp_search2alnblocks - Turn SearchIO parseable reports(s) into a set of aligned blocks

</pre><h4><b>SYNOPSIS</b></h4><pre>
         bp_search2alnblocks --minid PERCENTID --minlen LEN --minevalue EVALUE file1.
         blast file2.blast ...&gt; out.fas

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This script will parse and filter BLAST (or other formats Bio::SearchIO can parse) output and format the
       alignment as blocks of alignments based on the HSPs.  Note this can only work if the input file parsed
       contains the necessary.

       Typically this can be used to turn BLAST output into a FASTA alignment format for input into the QRNA
       comparative gene finder for RNA genes (E.Rivas).

</pre><h4><b>OPTIONS</b></h4><pre>
        --maxevalue   Maximum E-value for an HSP
        --minevalue   Minimum E-value for an HSP
        --minlen      Minimum length of an HSP [default 0]
        --maxid       Maximum Percent Id [default 100]
                      (to help remove sequences which are really close)
        --minid       Minimum Percent Identity for an HSP [default 0]
        -i/--input    An optional input filename (expects input on STDIN by default)
        -o/--output   An optional output filename (exports to STDOUT by default)
        -f/--format   Specify a different Search Alignment format-
                      {fasta, axt, waba, blast, blastxml} are all permitted
                      although the format must have actual alignment
                      sequence for this script to work
                      See L&lt;Bio::SearchIO&gt; for more information.
        -of/--outformat Output format for the alignment blocks, anything
                      L&lt;Bio::AlignIO&gt; supports.
        -v/--verbose  Turn on debugging

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Jason</b> <b>Stajich</b></h4><pre>
       Jason Stajich, jason-at-bioperl-dot-org.

perl v5.32.1                                       2021-08-15                            <u><a href="../man1p/BP_SEARCH2ALNBLOCKS.1p.html">BP_SEARCH2ALNBLOCKS</a></u>(1p)
</pre>
 </div>
</div></section>
</div>
</body>
</html>