<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>xpsa - extended psa header</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/pftools">pftools_3.2.12-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       xpsa - extended psa header

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>xpsa</b>  is  an  extension  of  the  <b><a href="../man5/psa.5.html">psa</a></b>(5)  file  format  used by the <b>pftools</b> package to describe and store
       biological sequences.

       <b>xpsa</b> uses <u>keyword</u>=<u>value</u> pairs in the header to include information about the sequence  or  the  alignment
       between  the  sequence and a <u>PROSITE</u> profile (or any other kind of motif). The syntax is therefore easily
       extensible. In this man-page, we focus on the keywords defined and used by the <b>pftools</b> package.

       In the following text we will use the more general term  '<u>motif</u>'  when  in  fact  our  discussion  refers
       specifically  to  <u>PROSITE</u>  <u>profiles</u>.   Nevertheless  the keywords defined here can be used by any kind of
       sequence analysis tool to store and transfer information.

       None of the defined keywords are mandatory and analysis tools have often a length limit  imposed  on  the
       header  line. In order to keep the header line reasonably short and user readable, these tools can easily
       remove individual <u>keyword</u>=<u>value</u> pairs which are not important to the specific task at hand.

</pre><h4><b>SYNTAX</b></h4><pre>
       The general syntax of the <b>xpsa</b> header is given below. For examples  please  refer  to  the  corresponding
       <u>examples</u> section.

       The  biological  sequence itself starts on the next line following the header and may extend over several
       lines.  If several sequences are contained in the same file, each must be preceded by a header line.

       Syntax of the header line:

              <b>&gt;</b><u>seq_id</u><b>/</b><u>seq_pos</u> { [ <u>keyword</u>=<u>value</u> | <u>free_text</u> ] }

       The header must start with a '<b>&gt;</b>' followed by the fields detailed below:

              <u>seq_id</u> typically the <u>accession</u> number and <u>identifier</u> of the sequence.  This text field should  not
                     contain any spaces.

              Note:  If  the input sequences have a well defined accession number and identifier (as is the case
                     with SWISS-PROT entries) <b><a href="../man1/pfsearch.1.html">pfsearch</a></b>(1) and <b><a href="../man1/pfscan.1.html">pfscan</a></b>(1) will use as <u>seq_id</u> the accession  number
                     separated by a '<b>|</b>' from the identifier.
                     If the input sequences are in FASTA format, no guess can be made about the accession number
                     or  the  identifier.  Therefore  <b><a href="../man1/pfsearch.1.html">pfsearch</a></b>(1)  and  <b><a href="../man1/pfscan.1.html">pfscan</a></b>(1)  will  use  as <u>seq_id</u> all text
                     following the '<b>&gt;</b>' character up to the first whitespace.

              <u>seq_pos</u>
                     this field describes the sequence coordinates where a  motif  matches.  The  positions  are
                     generally given as <u>start_pos</u>-<u>end_pos</u> pairs.
                     If present, this field is separated from the <u>seq_id</u> by a '<b>/</b>' character.

              <u>keyword</u>=<u>value</u>
                     is  an optional list of defined <u>keyword</u>=<u>value</u> pairs. Each pair should be separated from the
                     next or from free text by whitespaces.
                     The <u>keyword</u> is case sensitive in the current implementation  of  the  <b>pftools</b>  package.  It
                     should not exceed 24 characters in length.
                     The  <u>value</u> can be numeric or alphanumeric. Values containing whitespaces should be enclosed
                     in either <u>single</u> <u>quotes</u> or <u>double</u> <u>quotes</u>.  If the same type of  quote  appears  inside  the
                     value it must be escaped using the '<b>\</b>' character.

                     Note:  the <b>pftools</b> do not currently produce quote enclosed values.

              <u>free_text</u>
                     typically  this  is a description of the sequence. It should not contain any of the defined
                     <u>keyword</u>=<u>value</u> pairs.

   <b>Keywords</b>
       Keywords used by the <b><a href="../man1/pfsearch.1.html">pfsearch</a></b>(1) and <b><a href="../man1/pfscan.1.html">pfscan</a></b>(1) programs:

              <u>level</u>  the highest cut-off level (as a value) exceeded by the alignment.
                     The characters '<b>NA</b>' indicate that the alignment score does not exceed any  of  the  cut-off
                     levels defined in the motif.

              <u>level_tag</u>
                     the highest cut-off level (as a character string) exceeded by the alignment.
                     The  characters  '<b>NA</b>'  indicate that the alignment score does not exceed any of the cut-off
                     levels defined in the motif.

                     Note:  <b><a href="../man1/pfsearch.1.html">pfsearch</a></b>(1) and <b><a href="../man1/pfscan.1.html">pfscan</a></b>(1) only report the first  2  characters  of  the  level  text
                            string.

              <u>match_nb</u>
                     if  the  motif  matches  several  times  on  the  same sequence, each alignment is numbered
                     incrementally.
                     If the motif is circular, each  single  repeat  is  numbered  incrementally  with  the  key
                     <u>repeat_nb</u> (see below).

              <u>match_parent</u>
                     for  each  single match of a circular motif, this key references the number of the parental
                     total match of the circular motif.

                     Note:  if a circular motif matches only once on a given sequence, <b><a href="../man1/pfsearch.1.html">pfsearch</a></b>(1) and <b><a href="../man1/pfscan.1.html">pfscan</a></b>(1)
                            do not report this key.

              <u>match_type</u>
                     identifies the type of match. Either <b>region</b> for a complete match of a motif to a  sequence,
                     or <b>repeat</b> for a single repeat of a circular motif.

              <u>motif</u>  the name and/or identifier of the motif.

              <u>motif_start</u>
                     the motif position where the alignment begins.

              <u>motif_end</u>
                     negative offset from the end of the alignment to the end of the motif.

              <u>norm_score</u>
                     the normalized score of the alignment.

              <u>raw_score</u>
                     the raw score of the alignment.

              <u>repeat_nb</u>
                     if  the  motif  is  circular,  each  individual  repeat is numbered incrementally with this
                     keyword.

              <u>seq_end</u>
                     negative offset from the end of the alignment to the end of the sequence.
                     In combination with the information given by <u>seq_pos</u> this allows to deduce  the  length  of
                     the query sequence.

              <u>strand</u> the  sequence  strand  on  which  the  motif  matches, when the search includes the reverse
                     complement of a DNA sequence.  The <u>value</u> is either <b>s</b> for the sens  or  <b>r</b>  for  the  reverse
                     strand.

       Keywords used by the <b><a href="../man1/pfmake.1.html">pfmake</a></b>(1) and <b><a href="../man1/pfw.1.html">pfw</a></b>(1) programs:

              <u>weight</u> the weight of a given sequence in a multiple alignment.

</pre><h4><b>EXAMPLES</b></h4><pre>
       (1)    &gt;O00628|PEX7/73-315 <u>motif</u>=PS50294|WD_REP <u>raw_score</u>=1336 <u>match_nb</u>=1 <u>match_type</u>=region <u>seq_end</u>=-491
              VTW[...]IYD
              &gt;O00628|PEX7/540-801 <u>motif</u>=PS50294|WD_REP <u>raw_score</u>=1378 <u>match_nb</u>=2 <u>match_type</u>=region <u>seq_end</u>=-5
              SFD[...]PAS

              The  2  headers  above  describe  2 matches of the motif called '<u>WD_REP</u>' onto the sequence '<u>PEX7</u>'.
              Each of the matches onto this single sequence is numbered using the the <u>match_nb</u>  keyword.   These
              matches are not individual repeats of a circular motif as can be seen with the <u>region</u> value of the
              <u>match_type</u> keyword.
              The  first  match starts at position 73 of the sequence and ends at position 315. This position is
              491 residues away from the end of the input sequence (<u>seq_end</u>).
              The next line following the <b><a href="../man5/xpsa.5.html">xpsa</a></b>(5) header line  is  the  sequence  of  the  match  (it  has  been
              truncated here to help readability).
              The  second  match begins at position 540 of the sequence and terminates 5 residues before the end
              of the input sequence, that is at position 801.

       (2)    &gt;O00628|PEX7/540-582   <u>motif</u>=PS50294|WD_REP    <u>norm_score</u>=7.437    <u>raw_score</u>=180    <u>match_parent</u>=2
              <u>repeat_nb</u>=1 <u>match_type</u>=repeat <u>level</u>=-1 <u>seq_end</u>=-224 <u>motif_start</u>=1 <u>motif_end</u>=-1
              SFD[...]PLQ

              This example illustrates the kind of header obtained when aligning a circular motif to a sequence.
              Each  match  of  this motif (which we will call <u>total</u> match) can be composed of several individual
              repeats of the motif. Tools like <b><a href="../man1/pfsearch.1.html">pfsearch</a></b>(1) and <b><a href="../man1/pfscan.1.html">pfscan</a></b>(1) can output each total match followed by
              all its individual repeats. In this example we only show one of the indiviual repeats that is part
              of a total match between a circular profile and a sequence.
              The <b><a href="../man5/xpsa.5.html">xpsa</a></b>(5) header above describes a single repeat of a match  between  a  circular  motif  called
              '<u>WD_REP</u>' and the sequence '<u>PEX7</u>'.
              This  is  the  first  individual  repeat  of  a  match of the circular motif, as identified by the
              <u>repeat_nb</u> keyword. The other individual repeats have not been listed in this example.
              The total circular motif has at least 2 distinct matches on  the  '<u>PEX7</u>'  sequence,  because  this
              single  repeat  is part of the second match as described by the <u>match_parent</u> keyword. The parental
              matches have been omitted from this example, they would be numbered using the <u>match_nb</u> keyword.
              The normalized score of this motif exceeds the cut-off level number -1 (<u>level</u>  keyword)  which  is
              specified in the motif.
              This  match starts at position 1 of the profile (<u>motif_start</u>) and position 540 of the sequence, it
              ends at the end of the motif (<u>motif_end</u>=-1) and position 582 of the sequence.
              The next line following the <b><a href="../man5/xpsa.5.html">xpsa</a></b>(5) header line  is  the  sequence  of  the  match  (it  has  been
              truncated here to help readability).

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man5/psa.5.html">psa</a></b>(5), <b><a href="../man1/pfsearch.1.html">pfsearch</a></b>(1), <b><a href="../man1/pfscan.1.html">pfscan</a></b>(1), <b><a href="../man1/pfw.1.html">pfw</a></b>(1), <b><a href="../man1/pfmake.1.html">pfmake</a></b>(1), <b><a href="../man1/psa2msa.1.html">psa2msa</a></b>(1)

</pre><h4><b>AUTHOR</b></h4><pre>
       This manual page was originally written by Volker Flegel.
       The <b>pftools</b> package was developed by Philipp Bucher.
       Any comments or suggestions should be addressed to &lt;<a href="mailto:pftools@sib.swiss">pftools@sib.swiss</a>&gt;.

pftools 2.3                                         July 2003                                            <u><a href="../man5/XPSA.5.html">XPSA</a></u>(5)
</pre>
 </div>
</div></section>
</div>
</body>
</html>