<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Ace::Sequence::Multi - Combine Feature Tables from Multiple Databases</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libace-perl">libace-perl_1.92-12build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Ace::Sequence::Multi - Combine Feature Tables from Multiple Databases

</pre><h4><b>SYNOPSIS</b></h4><pre>
           use Ace::Sequence::Multi;

           # open reference database
           $ref = Ace-&gt;connect(-host=&gt;'stein.cshl.org',-port=&gt;200009);

           # open some secondary databases
           $db1 = Ace-&gt;connect(-host=&gt;'stein.cshl.org',-port=&gt;200010);
           $db2 = Ace-&gt;connect(-path=&gt;'/usr/local/acedb/mydata');

           # Make an Ace::Sequence::Multi object
           $seq = Ace::Sequence::Multi-&gt;new(-name   =&gt; 'CHROMOSOME_I,
                                            -db     =&gt; $ref,
                                            -offset =&gt; 3_000_000,
                                            -length =&gt; 1_000_000);

           # add the secondary databases
           $seq-&gt;add_secondary($db1,$db2);

           # get all the homologies (a list of Ace::Sequence::Feature objs)
           @homol = $seq-&gt;features('Similarity');

           # Get information about the first one -- goes to the correct db
           $feature = $homol[0];
           $type    = $feature-&gt;type;
           $subtype = $feature-&gt;subtype;
           $start   = $feature-&gt;start;
           $end     = $feature-&gt;end;
           $score   = $feature-&gt;score;

           # Follow the target
           $target  = $feature-&gt;info;

           # print the target's start and end positions
           print $target-&gt;start,'-',$target-&gt;end, "\n";

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <u>Ace::Sequence::Multi</u> transparently combines information stored about a sequence in a reference database
       with features tables from any number of annotation databases.  The resulting object can be used just like
       an Ace::Sequence object, except that the features remember their database of origin and go back to that
       database for information.

       This class will only work properly if the reference database and all annotation databases share the same
       cosmid map.

</pre><h4><b>OBJECT</b> <b>CREATION</b></h4><pre>
       You will use the <b>new()</b> method to create new Ace::Sequence::Multi objects.  The arguments are identical to
       the those in the Ace::Sequence parent class, with the addition of an option <b>-secondary</b> argument, which
       points to one or more secondary databases from which to fetch annotation information.

       -source
           The  sequence  source.   This  must  be an <u>Ace::Object</u> of the "Sequence" class, or be a sequence-like
           object containing the SMap tag (see below).

       -offset
           An offset from the beginning of the source sequence.  The retrieved <u>Ace::Sequence</u> will begin at  this
           position.  The offset can be any positive or negative integer.  Offsets are <b>0-based</b>.

       -length
           The  length of the sequence to return.  Either a positive or negative integer can be specified.  If a
           negative length is given, the returned sequence will be complemented relative to the source sequence.

       -refseq
           The sequence to use to establish the coordinate system  for  the  returned  sequence.   Normally  the
           source  sequence  is  used  to establish the coordinate system, but this can be used to override that
           choice.  You can provide either an <u>Ace::Object</u> or just a sequence name for this argument.  The source
           and reference sequences must share a common ancestor, but do not have to  be  directly  related.   An
           attempt to use a disjunct reference sequence, such as one on a different chromosome, will fail.

       -name
           As  an  alternative  to  using  an  <u>Ace::Object</u>  with  the <b>-source</b> argument, you may specify a source
           sequence using <b>-name</b> and <b>-db</b>.  The <u>Ace::Sequence</u> module will use the provided  database  accessor  to
           fetch  a Sequence object with the specified name. <b>new()</b> will return undef is no Sequence by this name
           is known.

       -db This argument is required if the  source  sequence  is  specified  by  name  rather  than  by  object
           reference.  It must be a previously opened handle to the reference database.

       -secondary
           This  argument points to one or more previously-opened annotation databases.  You may use a scalar if
           there is only one annotation database.  Otherwise, use an array reference.  You may  add  and  delete
           annotation  databases after the object is created by using the <b>add_secondary()</b> and <b>delete_secondary()</b>
           methods.

       If <b>new()</b> is successful, it will create an <u>Ace::Sequence::Multi</u> object and return it.  Otherwise  it  will
       return  undef  and  return  a descriptive message in Ace-&gt;<b>error()</b>.  Certain programming errors, such as a
       failure to provide required arguments, cause a fatal error.

</pre><h4><b>OBJECT</b> <b>METHODS</b></h4><pre>
       Most methods are inherited from <u>Ace::Sequence</u>.  The following additional methods are supported:

       <b>secondary()</b>
             @databases = $seq-&gt;secondary;

           Return a list of the secondary databases currently in use, or an empty list if none.

       <b>add_secondary()</b>
             $seq-&gt;add_secondary($db1,$db2,...)

           Add one or more secondary databases to the list of annotation databases.  Duplicate databases will be
           silently ignored.

       <b>delete_secondary()</b>
             $seq-&gt;delete_secondary($db1,$db2,...)

           Delete one or more secondary databases from the list of annotation databases.  Databases not  already
           in use will be silently ignored.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Ace, Ace::Object, Ace::Sequence,Ace::Sequence::Homol, Ace::Sequence::FeatureList, Ace::Sequence::Feature,
       GFF

</pre><h4><b>AUTHOR</b></h4><pre>
       Lincoln Stein &lt;<a href="mailto:lstein@w3.org">lstein@w3.org</a>&gt; with extensive help from Jean Thierry-Mieg &lt;<a href="mailto:mieg@kaa.crbm.cnrs-mop.fr">mieg@kaa.crbm.cnrs-mop.fr</a>&gt;

       Copyright (c) 1999, Lincoln D. Stein

       This  library  is  free  software;  you can redistribute it and/or modify it under the same terms as Perl
       itself.  See DISCLAIMER.txt for disclaimers of warranty.

perl v5.40.0                                       2024-10-20                          <u>Ace::Sequence::<a href="../man3pm/Multi.3pm.html">Multi</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>