<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Ace::Sequence::Homol - Temporary Sequence Homology Class</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libace-perl">libace-perl_1.92-12build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Ace::Sequence::Homol - Temporary Sequence Homology Class

</pre><h4><b>SYNOPSIS</b></h4><pre>
           # Get all similarity features from an Ace::Sequence
           @homol = $seq-&gt;features('Similarity');

           # sort by score
           @sorted = sort { $a-&gt;score &lt;=&gt; $b-&gt;score } @homol;

           # the last one has the highest score
           $best = $sorted[$#sorted];

           # fetch its associated Ace::Sequence::Homol
           $homol = $best-&gt;target;

           # print out the sequence name, DNA, start and end
           print $homol-&gt;name,' ',$homol-&gt;start,'-',$homol-&gt;end,"\n";
           print $homol-&gt;asDNA;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <u>Ace::Sequence::Homol</u> is a subclass of Ace::Object (<b>not</b> Ace::Sequence) which is specialized for returning
       information about a DNA or protein homology.  This is a temporary placeholder for a more sophisticated
       homology class which will include support for alignments.

</pre><h4><b>OBJECT</b> <b>CREATION</b></h4><pre>
       You will not ordinarily create an <u>Ace::Sequence::Homol</u> object directly.  Instead, objects will be created
       in response to an <b>info()</b> or <b>group()</b> method call on a similarity feature in an <u>Ace::Sequence::Feature</u>
       object.  If you wish to create an <u>Ace::Sequence::Homol</u> object directly, please consult the source code
       for the <u><b>new()</b></u> method.

</pre><h4><b>OBJECT</b> <b>METHODS</b></h4><pre>
       Most methods are inherited from <u>Ace::Object</u>.  The following methods are also supported:

       <b>start()</b>
             $start = $homol-&gt;start;

           Returns  the  start of the area that is similar to the <u>Ace::Sequence::Feature</u> from which his homology
           was derived.  Coordinates are relative to the target homology.

       <b>end()</b>
             $end = $homol-&gt;end;

           Returns the end of the area that is similar to the <u>Ace::Sequence::Feature</u> from which his homology was
           derived.  Coordinates are relative to the target homology.

       <b>asString()</b>
             $label = $homol-&gt;asString;

           Returns a human-readable identifier describing the nature of the feature.  The format is:

            $name/$start-$end

           for example:

            HUMGEN13/1-67

           This method is also called automatically when the object is treated in a string context.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Ace, Ace::Object, Ace::Sequence,Ace::Sequence::FeatureList, Ace::Sequence::Feature, GFF

</pre><h4><b>AUTHOR</b></h4><pre>
       Lincoln Stein &lt;<a href="mailto:lstein@w3.org">lstein@w3.org</a>&gt; with extensive help from Jean Thierry-Mieg &lt;<a href="mailto:mieg@kaa.crbm.cnrs-mop.fr">mieg@kaa.crbm.cnrs-mop.fr</a>&gt;

       Copyright (c) 1999, Lincoln D. Stein

       This library is free software; you can redistribute it and/or modify it under  the  same  terms  as  Perl
       itself.  See DISCLAIMER.txt for disclaimers of warranty.

perl v5.40.0                                       2024-10-20                          <u>Ace::Sequence::<a href="../man3pm/Homol.3pm.html">Homol</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>