<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::ASN1::Sequence::Indexer - Indexes NCBI Sequence files.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-asn1-entrezgene-perl">libbio-asn1-entrezgene-perl_1.730-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::ASN1::Sequence::Indexer - Indexes NCBI Sequence files.

</pre><h4><b>VERSION</b></h4><pre>
       version 1.73

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::ASN1::Sequence::Indexer;

         # creating &amp; using the index is just a few lines
         my $inx = Bio::ASN1::Sequence::Indexer-&gt;new(
           -filename =&gt; 'seq.idx',
           -write_flag =&gt; 'WRITE'); # needed for make_index call, but if opening
                                    # existing index file, don't set write flag!
         $inx-&gt;make_index('seq1.asn', 'seq2.asn');
         my $seq = $inx-&gt;fetch('AF093062'); # Bio::Seq obj for Sequence (doesn't work yet)
         # alternatively, if one prefers just a data structure instead of objects
         $seq = $inx-&gt;fetch_hash('AF093062'); # a hash produced by Bio::ASN1::Sequence
                                   # that contains all data in the Sequence record

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::ASN1::Sequence::Indexer is a Perl Indexer for NCBI Sequence genome databases. It processes an
       ASN.1-formatted Sequence record and stores the file position for each record in a way compliant with
       Bioperl standard (in fact its a subclass of Bioperl's index objects).

       Note that this module does not parse record, because it needs to run fast and grab only the gene ids.
       For parsing record, use Bio::ASN1::Sequence.

       As with Bio::ASN1::Sequence, this module is best thought of as beta version - it works, but is not fully
       tested.

</pre><h4><b>METHODS</b></h4><pre>
   <b>fetch</b>
         Parameters: $geneid - id for the Sequence record to be retrieved
         Example:    my $hash = $indexer-&gt;<a href="../man10/fetch.10.html">fetch</a>(10); # get Sequence #10
         Function:   fetch the data for the given Sequence id.
         Returns:    A Bio::Seq object produced by Bio::SeqIO::sequence
         Notes:      Bio::SeqIO::sequence does not exist and probably won't
                       exist for a while!  So call fetch_hash instead

   <b>fetch_hash</b>
         Parameters: $seqid - id for the Sequence record to be retrieved
         Example:    my $hash = $indexer-&gt;fetch_hash('AF093062');
         Function:   fetch a hash produced by Bio::ASN1::Sequence for given id
         Returns:    A data structure containing all data items from the Sequence
                       record.
         Notes:      Alternative to fetch()

</pre><h4><b>INTERNAL</b> <b>METHODS</b></h4><pre>
   <b>_version</b>
   <b>_type_stamp</b>
   <b>_index_file</b>
   <b>_file_format</b>
   <b>_file_handle</b>
         Title   : _file_handle
         Usage   : $fh = $index-&gt;_file_handle( INT )
         Function: Returns an open filehandle for the file
                   index INT.  On opening a new filehandle it
                   caches it in the @{$index-&gt;_filehandle} array.
                   If the requested filehandle is already open,
                   it simply returns it from the array.
         Example : $fist_file_indexed = $index-&gt;_file_handle( 0 );
         Returns : ref to a filehandle
         Args    : INT
         Notes   : This function is copied from Bio::Index::Abstract. Once that module
                     changes file handle code like I do below to fit perl 5.005_03, this
                     sub would be removed from this module

</pre><h4><b>PREREQUISITE</b></h4><pre>
       Bio::ASN1::Sequence, Bioperl and all dependencies therein.

</pre><h4><b>INSTALLATION</b></h4><pre>
       Same as Bio::ASN1::EntrezGene

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Please check out perldoc for Bio::ASN1::EntrezGene for more info.

</pre><h4><b>CITATION</b></h4><pre>
       Liu, Mingyi, and Andrei Grigoriev. "Fast parsers for Entrez Gene."  Bioinformatics 21, no. 14 (2005):
       3189-3190.

</pre><h4><b>OPERATION</b> <b>SYSTEMS</b> <b>SUPPORTED</b></h4><pre>
       Any OS that Perl &amp; Bioperl run on.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>              - General discussion
         <a href="http://bioperl.org/Support.html">http://bioperl.org/Support.html</a>    - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list: <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bio-asn1-entrezgene/issues

</pre><h4><b>AUTHOR</b></h4><pre>
       Dr. Mingyi Liu &lt;<a href="mailto:mingyiliu@gmail.com">mingyiliu@gmail.com</a>&gt;

</pre><h4><b>COPYRIGHT</b></h4><pre>
       This software is copyright (c) 2005 by Mingyi Liu, 2005 by GPC Biotech AG, and 2005 by Altana Research
       Institute.

       This software is available under the same terms as the perl 5 programming language system itself.

perl v5.38.2                                       2024-03-04                  <u>Bio::ASN1::Sequence::<a href="../man3pm/Indexer.3pm.html">Indexer</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>