<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Coordinate::Collection - Noncontinuous match between two coordinate sets.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-coordinate-perl">libbio-coordinate-perl_1.7.1-4_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Coordinate::Collection - Noncontinuous match between two coordinate sets.

</pre><h4><b>VERSION</b></h4><pre>
       version 1.007001

</pre><h4><b>SYNOPSIS</b></h4><pre>
         # create Bio::Coordinate::Pairs or other Bio::Coordinate::MapperIs somehow
         $pair1; $pair2;

         # add them into a Collection
         $collection = Bio::Coordinate::Collection-&gt;new;
         $collection-&gt;add_mapper($pair1);
         $collection-&gt;add_mapper($pair2);

         # create a position and map it
         $pos = Bio::Location::Simple-&gt;new (-start =&gt; 5, -end =&gt; 9 );
         $res = $collection-&gt;map($pos);
         $res-&gt;match-&gt;start == 1;
         $res-&gt;match-&gt;end == 5;

         # if mapping is many to one (*&gt;1) or many-to-many (*&gt;*)
         # you have to give seq_id not get unrelevant entries
         $pos = Bio::Location::Simple-&gt;new
             (-start =&gt; 5, -end =&gt; 9 -seq_id=&gt;'clone1');

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Generic, context neutral mapper to provide coordinate transforms between two <b>disjoint</b> coordinate systems.
       It brings into Bioperl the functionality from Ewan Birney's Bio::EnsEMBL::Mapper ported into current
       bioperl.

       This class is aimed for representing mapping between whole chromosomes and contigs, or between contigs
       and clones, or between sequencing reads and assembly. The submaps are automatically sorted, so they can
       be added in any order.

       To map coordinates to the other direction, you have to <b>swap()</b> the collection. Keeping track of the
       direction and ID restrictions are left to the calling code.

</pre><h4><b>ATTRIBUTES</b></h4><pre>
   <b>mappers</b>
        Title   : mappers
        Usage   : $obj-&gt;mappers();
        Function: Returns or sets a list of mappers.
        Example :
        Returns : array of mappers
        Args    : array of mappers

   <b>each_mapper</b>
        Title   : each_mapper
        Usage   : $obj-&gt;each_mapper();
        Function: Returns a list of mappers.
        Example :
        Returns : list of mappers
        Args    : none

   <b>mapper_count</b>
        Title   : mapper_count
        Usage   : my $count = $collection-&gt;mapper_count;
        Function: Get the count of the number of mappers stored
                  in this collection
        Example :
        Returns : integer
        Args    : none

</pre><h4><b>METHODS</b></h4><pre>
   <b>new</b>
   <b>add_mapper</b>
        Title   : add_mapper
        Usage   : $obj-&gt;add_mapper($mapper)
        Function: Pushes one Bio::Coordinate::MapperI into the list of mappers.
                  Sets _is_sorted() to false.
        Example :
        Returns : 1 when succeeds, 0 for failure.
        Args    : mapper object

   <b>swap</b>
        Title   : swap
        Usage   : $obj-&gt;swap;
        Function: Swap the direction of mapping;input &lt;-&gt; output
        Example :
        Returns : 1
        Args    :

   <b>test</b>
        Title   : test
        Usage   : $obj-&gt;test;
        Function: test that both components of all pairs are of the same length.
                  Ran automatically.
        Example :
        Returns : boolean
        Args    :

   <b>map</b>
        Title   : map
        Usage   : $newpos = $obj-&gt;map($pos);
        Function: Map the location from the input coordinate system
                  to a new value in the output coordinate system.
        Example :
        Returns : new value in the output coordinate system
        Args    : integer

   <b>sort</b>
        Title   : sort
        Usage   : $obj-&gt;sort;
        Function: Sort function so that all mappings are sorted by
                  input coordinate start
        Example :
        Returns : 1
        Args    :

</pre><h4><b>INTERNAL</b> <b>METHODS</b></h4><pre>
   <b>_map</b>
        Title   : _map
        Usage   : $newpos = $obj-&gt;_map($simpleloc);
        Function: Internal method that does the actual mapping. Called multiple times
                  by map() if the location  to be mapped is a split location

        Example :
        Returns : new location in the output coordinate system or undef
        Args    : Bio::Location::Simple

   <b>_is_sorted</b>
        Title   : _is_sorted
        Usage   : $newpos = $obj-&gt;_is_sorted;
        Function: toggle for whether the (internal) coordinate mapper data are sorted
        Example :
        Returns : boolean
        Args    : boolean

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list: <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/%%7Bdist%7D

</pre><h4><b>AUTHOR</b></h4><pre>
       Heikki Lehvaslaiho &lt;<a href="mailto:heikki@bioperl.org">heikki@bioperl.org</a>&gt;

</pre><h4><b>COPYRIGHT</b></h4><pre>
       This software is copyright (c) by Heikki Lehvaslaiho.

       This software is available under the same terms as the perl 5 programming language system itself.

perl v5.32.0                                       2020-11-24                   <u>Bio::Coordinate::<a href="../man3pm/Collection.3pm.html">Collection</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>