<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Seq::MetaI - Interface for sequence objects with residue-based meta information</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Seq::MetaI - Interface for sequence objects with residue-based meta information

</pre><h4><b>SYNOPSIS</b></h4><pre>
         # get a Bio::Seq::MetaI compliant object somehow

         # to test this is a meta seq object
         $obj-&gt;isa("Bio::Seq::MetaI")
            || $obj-&gt;throw("$obj not a Bio::Seq::MetaI");

         # accessors
         $string     = $obj-&gt;meta;
         $string     = $obj-&gt;meta_text;
         $substring  = $obj-&gt;submeta(12,50);
         $unique_key = $obj-&gt;accession_number();

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This class defines an abstract interface for basic residue-based meta information. Examples of this kind
       of meta data are secondary structures (RNA and protein), protein hydrophobicity assignments, or other
       alternative alphabets for polypeptides, sequence quality data and nucleotide alignments with
       translations.

       The length of the meta data sequence is not dependent on the amount of the meta information. The meta
       information always covers all the residues, but a blank value is used to denote unavailable information.
       If necessary the implementation quietly truncates or extends meta information with blank values.
       Definition of blank is implementation dependent. Gaps in MSAs should not have meta information.

       At this point a residue in a sequence object can have only one meta value. If you need more, use multiple
       copies of the sequence object.

       Meta data storage can be implemented in various ways, e.g: string, array of scalars, array of hashes,
       array of objects.

       If the implementation so chooses, there can be more then one meta values associated to each residue. See
       named_meta and names_submeta. Note that use of arbitrary names is very prone to typos leading to creation
       of additional copies of meta data sets.

       Bio::Seq::Meta provides basic, pure perl implementation of sequences with meta information. See
       Bio::Seq::Meta. Application specific implementations will override and add to these methods.

   <b>Method</b> <b>naming</b>
       Character based meta data is read and set by method <b>meta()</b> and its variants. These are the suffixes and
       prefixes used in the variants:

           [named_] [sub] meta [_text]

       _text
          Suffix <b>_text</b> guaranties that output is a string. Note that it does not limit the input.

       sub
          Prefix  <b>sub</b>,  like  in <b>subseq()</b>, means that the method applies to sub region of the sequence range and
          takes start and end as arguments.  Unlike <b>subseq()</b>, these methods are able  to  set  values.   If  the
          range is not defined, it defaults to the complete sequence.

       named_
          Prefix  <b>named_</b>  in  method  names  allows  the used to attach multiple meta strings to one sequence by
          explicitly naming them. The name is always the first argument to the method. The "unnamed" methods use
          the class wide default name for the meta data and are thus special cases "named" methods.

          Note that internally names are keys in a hash and any misspelling of a name will  silently  store  the
          data  under  a  wrong  name.  The  used  names  (keys) can be retrieved using method <b>meta_names()</b>. See
          meta_names.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Bio::Seq::Meta, Bio::Seq::Meta::Array, Bio::Seq::EncodedSeq, Bio::Tools::OddCodes, Bio::Seq::Quality

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your  comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather  than  to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.   Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Heikki</b> <b>Lehvaslaiho</b></h4><pre>
       Email heikki-at-bioperl-dot-org

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
       Chad  Matsalla,  <a href="mailto:bioinformatics@dieselwurks.com">bioinformatics@dieselwurks.com</a>;  Aaron  Mackey,  <a href="mailto:amackey@virginia.edu">amackey@virginia.edu</a>;  Peter  Schattner
       <a href="mailto:schattner@alum.mit.edu">schattner@alum.mit.edu</a>; Richard Adams, <a href="mailto:Richard.Adams@ed.ac.uk">Richard.Adams@ed.ac.uk</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually  preceded
       with a _

   <b>meta</b>
        Title   : meta
        Usage   : $meta_values  = $obj-&gt;meta($values_string);
        Function:

                  Get and set method for the unnamed meta data starting from
                  residue position one. Since it is dependent on the length
                  of the sequence, it needs to be manipulated after the
                  sequence.

                  The implementation may choose to accept argument values in
                  a string or in an array (reference) or in a hash
                  (reference).

                  The return value may be a string or an array reference,
                  depending on the implementation. If in doubt, use meta_text()
                  which is a variant guarantied to return a string.  See
                  L&lt;meta_text&gt;.

                  The length of the returned value always matches the length
                  of the sequence.

        Returns : A reference to an array or a string
        Args    : new value, optional

   <b>meta_text</b>
        Title   : meta_text()
        Usage   : $meta_values  = $obj-&gt;meta_text($values_arrayref);
        Function: Variant of meta() guarantied to return a textual
                  representation of the meta data. For details, see L&lt;meta&gt;.
        Returns : a string
        Args    : new value, optional

   <b>named_meta</b>
        Title   : named_meta()
        Usage   : $meta_values  = $obj-&gt;named_meta($name, $values_arrayref);
        Function: A more general version of meta(). Each meta data set needs
                  to be named. See also L&lt;meta_names&gt;.
        Returns : a string
        Args    : scalar, name of the meta data set
                  new value, optional

   <b>named_meta_text</b>
        Title   : named_meta_text()
        Usage   : $meta_values  = $obj-&gt;named_meta_text($name, $values_arrayref);
        Function: Variant of named_meta() guarantied to return a textual
                  representation  of the named meta data.
                  For details, see L&lt;meta&gt;.
        Returns : a string
        Args    : scalar, name of the meta data set
                  new value, optional

   <b>submeta</b>
        Title   : submeta
        Usage   : $subset_of_meta_values = $obj-&gt;submeta(10, 20, $value_string);
                  $subset_of_meta_values = $obj-&gt;submeta(10, undef, $value_string);
        Function:

                  Get and set method for meta data for subsequences.

                  Numbering starts from 1 and the number is inclusive, ie 1-2
                  are the first two residue of the sequence. Start cannot be
                  larger than end but can be equal.

                  If the second argument is missing the returned values
                  should extend to the end of the sequence.

                  If implementation tries to set values beyond the current
                  sequence, they should be ignored.

                  The return value may be a string or an array reference,
                  depending on the implementation. If in doubt, use
                  submeta_text() which is a variant guarantied to return a
                  string.  See L&lt;submeta_text&gt;.

        Returns : A reference to an array or a string
        Args    : integer, start position, optional
                  integer, end position, optional
                  new value, optional

   <b>submeta_text</b>
        Title   : submeta_text
        Usage   : $meta_values  = $obj-&gt;submeta_text(20, $value_string);
        Function: Variant of submeta() guarantied to return a textual
                  representation  of meta data. For details, see L&lt;meta&gt;.
        Returns : a string
        Args    : integer, start position, optional
                  integer, end position, optional
                  new value, optional

   <b>named_submeta</b>
        Title   : named_submeta
        Usage   : $subset_of_meta_values = $obj-&gt;named_submeta($name, 10, 20, $value_string);
                  $subset_of_meta_values = $obj-&gt;named_submeta($name, 10);
        Function: Variant of submeta() guarantied to return a textual
                  representation  of meta data. For details, see L&lt;meta&gt;.
        Returns : A reference to an array or a string
        Args    : scalar, name of the meta data set
                  integer, start position
                  integer, end position, optional when a third argument present
                  new value, optional

   <b>named_submeta_text</b>
        Title   : named_submeta_text
        Usage   : $meta_values  = $obj-&gt;named_submeta_text($name, 20, $value_string);
        Function: Variant of submeta() guarantied to return a textual
                  representation  of meta data. For details, see L&lt;meta&gt;.
        Returns : a string
        Args    : scalar, name of the meta data
        Args    : integer, start position, optional
                  integer, end position, optional
                  new value, optional

   <b>meta_names</b>
        Title   : meta_names
        Usage   : @meta_names  = $obj-&gt;meta_names()
        Function: Retrieves an array of meta data set names. The default (unnamed)
                  set name is guarantied to be the first name.
        Returns : an array of names
        Args    : none

   <b>force_flush</b>
        Title   : force_flush()
        Usage   : $force_flush = $obj-&gt;<a href="../man1/force_flush.1.html">force_flush</a>(1);
        Function: Automatically pad with empty values or truncate meta values to
                   sequence length
        Returns : boolean 1 or 0
        Args    : optional boolean value

   <b>is_flush</b>
        Title   : is_flush
        Usage   : $is_flush  = $obj-&gt;is_flush()
                  or  $is_flush = $obj-&gt;is_flush($my_meta_name)
        Function: Boolean to tell if all meta values are in
                  flush with the sequence length.
                  Returns true if force_flush() is set
                  Set verbosity to a positive value to see failed meta sets
        Returns : boolean 1 or 0
        Args    : optional name of the meta set

   <b>meta_length</b>
        Title   : meta_length()
        Usage   : $meeta_len  = $obj-&gt;meta_length();
        Function: return the number of elements in the meta set
        Returns : integer
        Args    : -

   <b>named_meta_length</b>
        Title   : named_meta_length()
        Usage   : $meeta_len  = $obj-&gt;named_meta_length($name);
        Function: return the number of elements in the named meta set
        Returns : integer
        Args    : -

</pre><h4><b>Bio::PrimarySeqI</b> <b>methods</b></h4><pre>
       Implementing classes will need to rewrite these Bio::PrimaryI methods.

   <b>revcom</b>
        Title   : revcom
        Usage   : $newseq = $seq-&gt;revcom();
        Function: Produces a new Bio::Seq::MetaI implementing object where
                  the order of residues and their meta information is reversed.
        Returns : A new (fresh) Bio::Seq::MetaI object
        Args    : none

   <b>trunc</b>
        Title   : trunc
        Usage   : $subseq = $myseq-&gt;trunc(10,100);
        Function: Provides a truncation of a sequence
        Returns : a fresh Bio::Seq::MetaI implementing object
        Args    : Two integers denoting first and last residue of the sub-sequence.

perl v5.32.1                                       2021-08-15                               <u>Bio::Seq::<a href="../man3pm/MetaI.3pm.html">MetaI</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>