<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>prank - Computes probabilistic multiple sequence alignments</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/prank">prank_0.0.170427+dfsg-3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       prank - Computes probabilistic multiple sequence alignments

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>prank</b> <u>sequence_file</u>

       <b>prank</b> [optional parameters] -d=<u>sequence_file</u> [optional parameters]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The Probabilistic Alignment Kit (PRANK) is a probabilistic multiple alignment program for DNA, codon and
       amino-acid sequences. It's based on a novel algorithm that treats insertions correctly and avoids over-
       estimation of the number of deletion events.

       In addition, PRANK borrows ideas from maximum likelihood methods used in phylogenetics and correctly
       takes into account the evolutionary distances between sequences. Lastly, PRANK allows for defining a
       potential structure for sequences to be aligned and then, simultaneously with the alignment, predicts the
       locations of structural units in the sequences.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>INPUT/OUTPUT</b> <b>PARAMETERS</b>
       <b>-d=</b><u><b>sequence_file</b></u>
               The input sequence file in FASTA format.

       <b>-t=</b><u><b>tree_file</b></u>
               The tree file to use. If unset, an appriximated NJ tree is generated.

       <b>-o=</b><u><b>output_file</b></u>
               Set the name of the output file. If unset, <u>output_file</u> is set to <b>output</b>.

       <b>-f=</b><u><b>output_format</b></u>
               Set  the  output  format. <u>output_format</u> can be one of <b>fasta</b> (default), <b>phylipi</b>, <b>phylips</b>, <b>paml</b>, or
               <b>nexus</b>.

       <b>-m=</b><u><b>model_file</b></u>
               The model file to use. If unset, <u>model_file</u> is set to <b>HKY2/WAG</b>.

       <b>-support</b>
               Compute posterior support.

       <b>-showxml</b>
               Output alignment xml-file.

       <b>-showtree</b>
               Output alignment guidetree.

       <b>-showanc</b>
               Output ancestral sequences.

       <b>-showall</b>
               Output all of these.

       <b>-noanchors</b>
               Do not use Exonerate anchoring. (Exonerate to be installed separately.)

       <b>-nomafft</b>
               Do not use MAFFT for guide tree. (MAFFT to be installed separately.)

       <b>-njtree</b> Estimate tree from an input alignment (and realign).

       <b>-shortnames</b>
               Truncate names at first space character.

       <b>-quiet</b>  Reduce output.

   <b>ALIGNMENT</b> <b>MERGE</b>
       <b>-d1=</b><u><b>alignment_file</b></u>
               The first input alignment file in FASTA format.

       <b>-d2=</b><u><b>alignment_file</b></u>
               The second input alignment file in FASTA format.

       <b>-t1=</b><u><b>tree_file</b></u>
               The tree file for the first alignment. If unset, an appriximated NJ tree is generated.

       <b>-t2=</b><u><b>tree_file</b></u>
               The tree file for the second alignment. If unset, an appriximated NJ tree is generated.

   <b>MODEL</b> <b>PARAMETERS</b>
       <b>-F</b>, <b>+F</b>  Force insertions to be always skipped.

       <b>-gaprate=</b><u><b>#</b></u>
               Set the gap opening rate. The default is <b>0.025</b> for DNA and <b>0.005</b> for proteins.

       <b>-gapext=</b><u><b>#</b></u>
               Set the gap extension probability. The default is <b>0.75</b> for DNA and <b>0.5</b> for proteins.

       <b>-codon</b>  Use empirical codon model for coding DNA.

       <b>-DNA</b>, <b>-protein</b>
               Use DNA or protein model, respectively. Disables auto-detection of model.

       <b>-termgap</b>
               Penalise terminal gaps normally.

       <b>-nomissing</b>
               No missing data. Use <b>-F</b> for terminal gaps.

       <b>-keep</b>   Do not remove gaps from pre-aligned sequences.

   <b>OTHER</b> <b>PARAMETERS</b>
       <b>-iterate=#</b>
               Rounds of re-alignment iteration; by default, iterate five times and keep the best result.

       <b>-once</b>   Run only once. Same as -iterate=1.

       <b>-prunetree</b>
               Prune guide tree branches with no sequence data.

       <b>-prunedata</b>
               Prune sequence data with no guide tree leaves.

       <b>-uselogs</b>
               Slower but should work for a greater number of sequences.

       <b>-translate</b>
               Translate input data to protein sequences.

       <b>-mttranslate</b>
               Translate input data to protein sequencess using mt table.

       <b>-convert</b>
               Do not align, just convert to a different format.

       <b>-dna=</b><u><b>dna_sequence_file</b></u>
               DNA sequence file for backtranslation of protein alignment.

       <b>-help</b>   Show an extended help page with more options.

       <b>-version</b>
               Show version and check for updates.

</pre><h4><b>AUTHORS</b></h4><pre>
       <b>prank</b> was written by Ari Loytynoja.

       This manual page was originally written by Manuel Prinz &lt;<a href="mailto:manuel@debian.org">manuel@debian.org</a>&gt; for the Debian  project  (and
       may be used by others).

v.121211                                           2020-10-30                                           <u><a href="../man1/PRANK.1.html">PRANK</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>