<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>alf - Alignment free sequence comparison</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/seqan-apps">seqan-apps_2.4.0+dfsg-16_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       alf - Alignment free sequence comparison

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>alf</b> [<u>OPTIONS</u>] <b>-i</b> <u>IN.FASTA</u> [<b>-o</b> <u>OUT.TXT</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Compute  pairwise  similarity  of  sequences  using alignment-free methods in <u>IN.FASTA</u> and write out tab-
       delimited matrix with pairwise scores to <u>OUT.TXT</u>.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              Display the help message.

       <b>--version</b>
              Display version information.

       <b>-v</b>, <b>--verbose</b>
              When given, details about the progress are printed to the screen.

   <b>Input</b> <b>/</b> <b>Output:</b>
       <b>-i</b>, <b>--input-file</b> <u>INPUT_FILE</u>
              Name of the multi-FASTA input file. Valid filetypes are: <u>.sam[.*]</u>, <u>.raw[.*]</u>,  <u>.gbk[.*]</u>,  <u>.frn[.*]</u>,
              <u>.fq[.*]</u>, <u>.fna[.*]</u>, <u>.ffn[.*]</u>, <u>.fastq[.*]</u>, <u>.fasta[.*]</u>, <u>.faa[.*]</u>, <u>.fa[.*]</u>, <u>.embl[.*]</u>, and <u>.bam</u>, where
              * is any of the following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent (de)compression.

       <b>-o</b>, <b>--output-file</b> <u>OUTPUT_FILE</u>
              Name  of  the  file  to  which  the  tab-delimtied matrix with pairwise scores will be written to.
              Default is to write to stdout. Valid filetype is: <u>.alf[.*]</u>,  where  *  is  any  of  the  following
              extensions: <u>tsv</u> for transparent (de)compression.

   <b>General</b> <b>Algorithm</b> <b>Parameters:</b>
       <b>-m</b>, <b>--method</b> <u>STRING</u>
              Select method to use. One of <u>N2</u>, <u>D2</u>, <u>D2Star</u>, and <u>D2z</u>. Default: <u>N2</u>.

       <b>-k</b>, <b>--k-mer-size</b> <u>INTEGER</u>
              Size of the k-mers. Default: <u>4</u>.

       <b>-mo</b>, <b>--bg-model-order</b> <u>INTEGER</u>
              Order of background Markov Model. Default: <u>1</u>.

   <b>N2</b> <b>Algorithm</b> <b>Parameters:</b>
       <b>-rc</b>, <b>--reverse-complement</b> <u>STRING</u>
              Which  strand  to  score.   Use  <u>both_strands</u>  to score both strands simultaneously. One of <u>input</u>,
              <u>both_strands</u>, <u>mean</u>, <u>min</u>, and <u>max</u>. Default: <u>input</u>.

       <b>-mm</b>, <b>--mismatches</b> <u>INTEGER</u>
              Number of mismatches, one of <u>0</u> and <u>1</u>.  When <u>1</u> is  used,  N2  uses  the  k-mer-neighbour  with  one
              mismatch. Default: <u>0</u>.

       <b>-mmw</b>, <b>--mismatch-weight</b> <u>DOUBLE</u>
              Real-valued weight of counts for words with mismatches. Default: <u>0.1</u>.

       <b>-kwf</b>, <b>--k-mer-weights-file</b> <u>OUTPUT_FILE</u>
              Print k-mer weights for every sequence to this file if given. Valid filetype is: <u>.txt</u>.

</pre><h4><b>CONTACT</b> <b>AND</b> <b>REFERENCES</b></h4><pre>
       For questions or comments, contact:
              Jonathan Goeke &lt;<a href="mailto:goeke@molgen.mpg.de">goeke@molgen.mpg.de</a>&gt;

       Please reference the following publication if you used ALF or the N2 method for your analysis:
              Jonathan  Goeke,  Marcel  H.  Schulz,  Julia  Lasserre, and Martin Vingron. Estimation of Pairwise
              Sequence Similarity of Mammalian Enhancers with Word Neighbourhood Counts. Bioinformatics (2012).

       Project Homepage:
              <a href="http://www.seqan.de/projects/alf">http://www.seqan.de/projects/alf</a>

alf 1.1.10 [tarball]                                                                                      <u><a href="../man1/ALF.1.html">ALF</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>