<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>spades - SPAdes genome assembler</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/spades">spades_3.15.5+dfsg-8_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       spades - SPAdes genome assembler

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>spades</b> [options] -o &lt;output_dir&gt;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>spades</b> is the main executable for the <b>SPAdes</b> software. It assembles genomic reads given to it and places
       the resulting assembly in &lt;output_dir&gt;.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Basic</b> <b>options</b>
       <b>-o</b> &lt;output_dir&gt;
           directory to store all the resulting files (required)

       <b>--sc</b>
           this flag is required for MDA (single-cell) data

       <b>--meta</b>
           this flag is required for metagenomic sample data

       <b>--rna</b>
           this flag is required for RNA-seq data

       <b>--plasmid</b>
           runs plasmidSPAdes pipeline for plasmid detection

       <b>--iontorrent</b>
           this flag is required for IonTorrent data

       <b>--test</b>
           runs SPAdes on toy dataset

       <b>-h</b>/<b>--help</b>
           prints this usage message

   <b>Input</b> <b>data</b>
       <b>--12</b> &lt;filename&gt;
           file with interlaced forward and reverse paired-end reads

       <b>-1</b> &lt;filename&gt;
           file with forward paired-end reads

       <b>-2</b> &lt;filename&gt;
           file with reverse paired-end reads

       <b>-s</b> &lt;filename&gt;
           file with unpaired reads

       <b>--pe&lt;#&gt;-12</b> &lt;filename&gt;
           file with interlaced reads for paired-end library number &lt;#&gt; (&lt;#&gt; = 1,2,3,4,5)

       <b>--pe&lt;#&gt;-1</b> &lt;filename&gt;
           file with forward reads for paired-end library number &lt;#&gt; (&lt;#&gt; = 1,2,3,4,5)

       <b>--pe&lt;#&gt;-2</b> &lt;filename&gt;
           file with reverse reads for paired-end library number &lt;#&gt; (&lt;#&gt; = 1,2,3,4,5)

       <b>--pe&lt;#&gt;-s</b> &lt;filename&gt;
           file with unpaired reads for paired-end library number &lt;#&gt; (&lt;#&gt; = 1,2,3,4,5)

       <b>--pe&lt;#&gt;-&lt;or&gt;</b>
           orientation of reads for paired-end library number &lt;#&gt; (&lt;#&gt; = 1,2,3,4,5; &lt;or&gt; = fr, rf, ff)

       <b>--s&lt;#&gt;</b> &lt;filename&gt;
           file with unpaired reads for single reads library number &lt;#&gt; (&lt;#&gt; = 1,2,3,4,5)

       <b>--mp&lt;#&gt;-12</b>  &lt;filename&gt;
           file with interlaced reads for mate-pair library number &lt;#&gt; (&lt;#&gt; = 1,2,3,4,5)

       <b>--mp&lt;#&gt;-1</b> &lt;filename&gt;
           file with forward reads for mate-pair library number &lt;#&gt; (&lt;#&gt; = 1,2,3,4,5)

       <b>--mp&lt;#&gt;-2</b> &lt;filename&gt;
           file with reverse reads for mate-pair library number &lt;#&gt; (&lt;#&gt; = 1,2,3,4,5)

       <b>--mp&lt;#&gt;-s</b> &lt;filename&gt;
           file with unpaired reads for mate-pair library number &lt;#&gt; (&lt;#&gt; = 1,2,3,4,5)

       <b>--mp&lt;#&gt;-&lt;or&gt;</b>
           orientation of reads for mate-pair library number &lt;#&gt; (&lt;#&gt; = 1,2,3,4,5; &lt;or&gt; = fr, rf, ff)

       <b>--hqmp&lt;#&gt;-12</b> &lt;filename&gt;
           file with interlaced reads for high-quality mate-pair library number &lt;#&gt; (&lt;#&gt; = 1,2,3,4,5)

       <b>--hqmp&lt;#&gt;-1</b> &lt;filename&gt;
           file with forward reads for high-quality mate-pair library number &lt;#&gt; (&lt;#&gt; = 1,2,3,4,5)

       <b>--hqmp&lt;#&gt;-2</b> &lt;filename&gt;
           file with reverse reads for high-quality mate-pair library number &lt;#&gt; (&lt;#&gt; = 1,2,3,4,5)

       <b>--hqmp&lt;#&gt;-s</b> &lt;filename&gt;
           file with unpaired reads for high-quality mate-pair library number &lt;#&gt; (&lt;#&gt; = 1,2,3,4,5)

       <b>--hqmp&lt;#&gt;-&lt;or&gt;</b>
           orientation of reads for high-quality mate-pair library number &lt;#&gt; (&lt;#&gt; = 1,2,3,4,5; &lt;or&gt; = fr, rf,
           ff)

       <b>--nxmate&lt;#&gt;-1</b> &lt;filename&gt;
           file with forward reads for Lucigen NxMate library number &lt;#&gt; (&lt;#&gt; = 1,2,3,4,5)

       <b>--nxmate&lt;#&gt;-2</b> &lt;filename&gt;
           file with reverse reads for Lucigen NxMate library number &lt;#&gt; (&lt;#&gt; = 1,2,3,4,5)

       <b>--sanger</b> &lt;filename&gt;
           file with Sanger reads

       <b>--pacbio</b> &lt;filename&gt;
           file with PacBio reads

       <b>--nanopore</b> &lt;filename&gt;
           file with Nanopore reads

       <b>--trusted-contigs</b> &lt;filename&gt;
           file with trusted contigs

       <b>--untrusted-contigs</b> &lt;filename&gt;
           file with untrusted contigs

   <b>Pipeline</b> <b>options</b>
       <b>--only-error-correction</b>
           runs only read error correction (without assembling)

       <b>--only-assembler</b>
           runs only assembling (without read error correction)

       <b>--careful</b>
           tries to reduce number of mismatches and short indels

       <b>--continue</b>
           continue run from the last available check-point

       <b>--restart-from</b> &lt;cp&gt;
           restart run with updated options and from the specified check-point ('ec', 'as', 'k&lt;int&gt;', 'mc')

       <b>--disable-gzip-output</b>
           forces error correction not to compress the corrected reads

       <b>--disable-rr</b>
           disables repeat resolution stage of assembling

   <b>Advanced</b> <b>options</b>
       <b>--dataset</b> &lt;filename&gt;
           file with dataset description in YAML format

       <b>-t</b>/<b>--threads</b> &lt;int&gt;
           number of threads [default: 16]

       <b>-m</b>/<b>--memory</b> &lt;int&gt;
           RAM limit for SPAdes in Gb (terminates if exceeded) [default: 250]

       <b>--tmp-dir</b> &lt;dirname&gt;
           directory for temporary files [default: &lt;output_dir&gt;<a href="file:/tmp">/tmp</a>]

       <b>-k</b> &lt;int,int,...&gt;
           comma-separated list of k-mer sizes (must be odd and less than 128) [default: 'auto']

       <b>--cov-cutoff</b> &lt;float&gt;
           coverage cutoff value (a positive float number, or 'auto', or 'off') [default: 'off']

       <b>--phred-offset</b> &lt;33 or 64&gt;
           PHRED quality offset in the input reads (33 or 64) [default: auto-detect]

                                                                                                       <u><a href="../man1/SPADES.1.html">SPADES</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>