<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>cnvkit_scatter - Plot probe log2 coverages and segmentation calls together.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/cnvkit">cnvkit_0.9.12-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       cnvkit_scatter - Plot probe log2 coverages and segmentation calls together.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: cnvkit.py scatter [-h] [-s FILENAME] [-c RANGE] [-g GENE]

       [-l RANGE_LIST] [-w WIDTH] [-o FILENAME]
              [-a  CHARACTER]  [--by-bin]  [--segment-color  SEGMENT_COLOR] [--title TITLE] [-t] [--y-max Y_MAX]
              [--y-min Y_MIN] [--fig-size  WIDTH  HEIGHT]  [-v  FILENAME]  [-i  SAMPLE_ID]  [-n  NORMAL_ID]  [-m
              MIN_VARIANT_DEPTH] [-z [ALT_FREQ]] [filename]

   <b>positional</b> <b>arguments:</b>
       filename
              Processed bin-level copy ratios (*.cnr), the output of the 'fix' sub-command.

   <b>options:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>-s</b> FILENAME, <b>--segment</b> FILENAME
              Segmentation calls (.cns), the output of the 'segment' command.

       <b>-c</b> RANGE, <b>--chromosome</b> RANGE
              Chromosome  or chromosomal range, e.g. 'chr1' or 'chr1:2333000-2444000', to display. If a range is
              given, all targeted genes in this range will be shown, unless <b>-g</b>/--gene is also given.

       <b>-g</b> GENE, <b>--gene</b> GENE
              Name of gene or genes (comma-separated) to display.

       <b>-l</b> RANGE_LIST, <b>--range-list</b> RANGE_LIST
              File listing the chromosomal ranges to display, as BED, interval  list  or  'chr:start-end'  text.
              Creates  focal plots similar to <b>-c</b>/--chromosome for each listed region, combined into a multi-page
              PDF. The output filename must also be specified (<b>-o</b>/--output).

       <b>-w</b> WIDTH, <b>--width</b> WIDTH
              Width of margin to show around the  selected  gene(s)  (<b>-g</b>/--gene)  or  small  chromosomal  region
              (<b>-c</b>/--chromosome). [Default: 1000000]

       <b>-o</b> FILENAME, <b>--output</b> FILENAME
              Output PDF file name.

   <b>Plot</b> <b>aesthetics:</b>
       <b>-a</b> CHARACTER, <b>--antitarget-marker</b> CHARACTER
              Plot  antitargets  using  this symbol when plotting in a selected chromosomal region (<b>-g</b>/--gene or
              <b>-c</b>/--chromosome). [Default: same as targets]

       <b>--by-bin</b>
              Plot data x-coordinates by bin indices instead of genomic coordinates. All bins will be shown with
              equal width, no blank regions will be  shown,  and  x-axis  values  indicate  bin  number  (within
              chromosome) instead of genomic position.

       <b>--segment-color</b> SEGMENT_COLOR
              Plot  segment  lines  in this color. Value can be any string accepted by matplotlib, e.g. 'red' or
              '#CC0000'.

       <b>--title</b> TITLE
              Plot title. [Default: sample ID, from filename or <b>-i]</b>

       <b>-t</b>, <b>--trend</b>
              Draw a smoothed local trendline on the scatter plot.

       <b>--y-max</b> Y_MAX
              y-axis upper limit.

       <b>--y-min</b> Y_MIN
              y-axis lower limit.

       <b>--fig-size</b> WIDTH HEIGHT
              Width and height of the plot in inches. [Default: Predefined  in  Matplotlib  'rcParams'  variable
              (most of the time: '6.4 4.8')]

   <b>To</b> <b>plot</b> <b>SNP</b> <b>b-allele</b> <b>frequencies:</b>
       <b>-v</b> FILENAME, <b>--vcf</b> FILENAME
              VCF file name containing variants to plot for SNV b-allele frequencies.

       <b>-i</b> SAMPLE_ID, <b>--sample-id</b> SAMPLE_ID
              Name  of  the  sample  in the VCF to use for b-allele frequency extraction and as the default plot
              title.

       <b>-n</b> NORMAL_ID, <b>--normal-id</b> NORMAL_ID
              Corresponding normal sample ID in the input VCF. This sample is used to select only germline  SNVs
              to plot.

       <b>-m</b> MIN_VARIANT_DEPTH, <b>--min-variant-depth</b> MIN_VARIANT_DEPTH
              Minimum read depth for a SNV to be used in the b-allele frequency calculation. [Default: 20]

       <b>-z</b> [ALT_FREQ], <b>--zygosity-freq</b> [ALT_FREQ]
              Ignore  VCF's genotypes (GT field) and instead infer zygosity from allele frequencies. [Default if
              used without a number: 0.25]

cnvkit.py scatter 0.9.10                            July 2023                                  <u><a href="../man1/CNVKIT_SCATTER.1.html">CNVKIT_SCATTER</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>