<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>iqtree2-omp - efficient phylogenetic software by maximum likelihood (multiprocessor version)</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/iqtree">iqtree_2.0.7+dfsg-1build3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       iqtree2-omp - efficient phylogenetic software by maximum likelihood (multiprocessor version)

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>iqtree</b> [<u>-s</u> <u>ALIGNMENT</u>] [<u>-p</u> <u>PARTITION</u>] [<u>-m</u> <u>MODEL</u>] [<u>-t</u> <u>TREE</u>] ...

</pre><h4><b>DESCRIPTION</b></h4><pre>
       IQ-TREE  multicore  version  2.0.7 for Linux 64-bit built Nov 30 2024 Developed by Bui Quang Minh, Nguyen
       Lam Tung, Olga Chernomor, Heiko Schmidt, Dominik Schrempf, Michael Woodhams.

   <b>GENERAL</b> <b>OPTIONS:</b>
       <b>-h</b>, <b>--help</b>
              Print (more) help usages

       <b>-s</b> FILE[,...,FILE]
              PHYLIP/FASTA/NEXUS/CLUSTAL/MSF alignment file(s)

       <b>-s</b> DIR Directory of alignment files

       <b>--seqtype</b> STRING
              BIN, DNA, AA, NT2AA, CODON, MORPH (default: auto-detect)

       <b>-t</b> FILE|PARS|RAND
              Starting tree (default: 99 parsimony and BIONJ)

       <b>-o</b> TAX[,...,TAX]
              Outgroup taxon (list) for writing .treefile

       <b>--prefix</b> STRING
              Prefix for all output files (default: aln/partition)

       <b>--seed</b> NUM
              Random seed number, normally used for debugging purpose

       <b>--safe</b> Safe likelihood kernel to avoid numerical underflow

       <b>--mem</b> NUM[G|M|%]
              Maximal RAM usage in GB | MB | %

       <b>--runs</b> NUM
              Number of indepedent runs (default: 1)

       <b>-v</b>, <b>--verbose</b>
              Verbose mode, printing more messages to screen

       <b>-V</b>, <b>--version</b>
              Display version number

       <b>--quiet</b>
              Quiet mode, suppress printing to screen (stdout)

       <b>-fconst</b> f1,...,fN
              Add constant patterns into alignment (N=no. states)

       <b>--epsilon</b> NUM
              Likelihood epsilon for parameter estimate (default 0.01)

       <b>-T</b> NUM|AUTO
              No. cores/threads or AUTO-detect (default: 1)

       <b>--threads-max</b> NUM
              Max number of threads for <b>-T</b> AUTO (default: all cores)

   <b>CHECKPOINT:</b>
       <b>--redo</b> Redo both ModelFinder and tree search

       <b>--redo-tree</b>
              Restore ModelFinder and only redo tree search

       <b>--undo</b> Revoke finished run, used when changing some options

       <b>--cptime</b> NUM
              Minimum checkpoint interval (default: 60 sec and adapt)

   <b>PARTITION</b> <b>MODEL:</b>
       <b>-p</b> FILE|DIR
              NEXUS/RAxML partition file or directory with alignments Edge-linked proportional partition model

       <b>-q</b> FILE|DIR
              Like <b>-p</b> but edge-linked equal partition model

       <b>-Q</b> FILE|DIR
              Like <b>-p</b> but edge-unlinked partition model

       <b>-S</b> FILE|DIR
              Like <b>-p</b> but separate tree inference

       <b>--subsample</b> NUM
              Randomly sub-sample partitions (negative for complement)

       <b>--subsample-seed</b> NUM Random number seed for <b>--subsample</b>

   <b>LIKELIHOOD/QUARTET</b> <b>MAPPING:</b>
       <b>--lmap</b> NUM
              Number of quartets for likelihood mapping analysis

       <b>--lmclust</b> FILE
              NEXUS file containing clusters for likelihood mapping

       <b>--quartetlh</b>
              Print quartet log-likelihoods to .quartetlh file

   <b>TREE</b> <b>SEARCH</b> <b>ALGORITHM:</b>
       <b>--ninit</b> NUM
              Number of initial parsimony trees (default: 100)

       <b>--ntop</b> NUM
              Number of top initial trees (default: 20)

       <b>--nbest</b> NUM
              Number of best trees retained during search (defaut: 5)

       <b>-n</b> NUM Fix number of iterations to stop (default: OFF)

       <b>--nstop</b> NUM
              Number of unsuccessful iterations to stop (default: 100)

       <b>--perturb</b> NUM
              Perturbation strength for randomized NNI (default: 0.5)

       <b>--radius</b> NUM
              Radius for parsimony SPR search (default: 6)

       <b>--allnni</b>
              Perform more thorough NNI search (default: OFF)

       <b>-g</b> FILE
              (Multifurcating) topological constraint tree file

       <b>--fast</b> Fast search to resemble FastTree

       <b>--polytomy</b>
              Collapse near-zero branches into polytomy

       <b>--tree-fix</b>
              Fix <b>-t</b> tree (no tree search performed)

       <b>--treels</b>
              Write locally optimal trees into .treels file

       <b>--show-lh</b>
              Compute tree likelihood without optimisation

       <b>--terrace</b>
              Check if the tree lies on a phylogenetic terrace

   <b>ULTRAFAST</b> <b>BOOTSTRAP/JACKKNIFE:</b>
       <b>-B</b>, <b>--ufboot</b> NUM
              Replicates for ultrafast bootstrap (&gt;=1000)

       <b>-J</b>, <b>--ufjack</b> NUM
              Replicates for ultrafast jackknife (&gt;=1000)

       <b>--jack-prop</b> NUM
              Subsampling proportion for jackknife (default: 0.5)

       <b>--sampling</b> STRING
              GENE|GENESITE resampling for partitions (default: SITE)

       <b>--boot-trees</b>
              Write bootstrap trees to .ufboot file (default: none)

       <b>--wbtl</b> Like <b>--boot-trees</b> but also writing branch lengths

       <b>--nmax</b> NUM
              Maximum number of iterations (default: 1000)

       <b>--nstep</b> NUM
              Iterations for UFBoot stopping rule (default: 100)

       <b>--bcor</b> NUM
              Minimum correlation coefficient (default: 0.99)

       <b>--beps</b> NUM
              RELL epsilon to break tie (default: 0.5)

       <b>--bnni</b> Optimize UFBoot trees by NNI on bootstrap alignment

   <b>NON-PARAMETRIC</b> <b>BOOTSTRAP/JACKKNIFE:</b>
       <b>-b</b>, <b>--boot</b> NUM
              Replicates for bootstrap + ML tree + consensus tree

       <b>-j</b>, <b>--jack</b> NUM
              Replicates for jackknife + ML tree + consensus tree

       <b>--jack-prop</b> NUM
              Subsampling proportion for jackknife (default: 0.5)

       <b>--bcon</b> NUM
              Replicates for bootstrap + consensus tree

       <b>--bonly</b> NUM
              Replicates for bootstrap only

       <b>--tbe</b>  Transfer bootstrap expectation

   <b>SINGLE</b> <b>BRANCH</b> <b>TEST:</b>
       <b>--alrt</b> NUM
              Replicates for SH approximate likelihood ratio test

       <b>--alrt</b> 0
              Parametric aLRT test (Anisimova and Gascuel 2006)

       <b>--abayes</b>
              approximate Bayes test (Anisimova et al. 2011)

       <b>--lbp</b> NUM
              Replicates for fast local bootstrap probabilities

   <b>MODEL-FINDER:</b>
       <b>-m</b> TESTONLY
              Standard model selection (like jModelTest, ProtTest)

       <b>-m</b> TEST
              Standard model selection followed by tree inference

       <b>-m</b> MF  Extended model selection with FreeRate heterogeneity

       <b>-m</b> MFP Extended model selection followed by tree inference

       <b>-m</b> ...+LM
              Additionally test Lie Markov models

       <b>-m</b> ...+LMRY
              Additionally test Lie Markov models with RY symmetry

       <b>-m</b> ...+LMWS
              Additionally test Lie Markov models with WS symmetry

       <b>-m</b> ...+LMMK
              Additionally test Lie Markov models with MK symmetry

       <b>-m</b> ...+LMSS
              Additionally test strand-symmetric models

       <b>--mset</b> STRING
              Restrict search to models supported by other programs (raxml, phyml or mrbayes)

       <b>--mset</b> STR,...
              Comma-separated model list (e.g. <b>-mset</b> WAG,LG,JTT)

       <b>--msub</b> STRING
              Amino-acid model source (nuclear, mitochondrial, chloroplast or viral)

       <b>--mfreq</b> STR,...
              List of state frequencies

       <b>--mrate</b> STR,...
              List of rate heterogeneity among sites (e.g. <b>-mrate</b> E,I,G,I+G,R is used for <b>-m</b> MF)

       <b>--cmin</b> NUM
              Min categories for FreeRate model [+R] (default: 2)

       <b>--cmax</b> NUM
              Max categories for FreeRate model [+R] (default: 10)

       <b>--merit</b> AIC|AICc|BIC
              Akaike|Bayesian information criterion (default: BIC)

       <b>--mtree</b>
              Perform full tree search for every model

       <b>--madd</b> STR,...
              List of mixture models to consider

       <b>--mdef</b> FILE
              Model definition NEXUS file (see Manual)

       <b>--modelomatic</b>
              Find best codon/protein/DNA models (Whelan et al. 2015)

   <b>PARTITION-FINDER:</b>
       <b>--merge</b>
              Merge partitions to increase model fit

       <b>--merge</b> greedy|rcluster|rclusterf
              Set merging algorithm (default: rclusterf)

       <b>--merge-model</b> 1|all
              Use only 1 or all models for merging (default: 1)

       <b>--merge-model</b> STR,...
              Comma-separated model list for merging

       <b>--merge-rate</b> 1|all
              Use only 1 or all rate heterogeneity (default: 1)

       <b>--merge-rate</b> STR,...
              Comma-separated rate list for merging

       <b>--rcluster</b> NUM
              Percentage of partition pairs for rcluster algorithm

       <b>--rclusterf</b> NUM
              Percentage of partition pairs for rclusterf algorithm

       <b>--rcluster-max</b> NUM
              Max number of partition pairs (default: 10*partitions)

   <b>SUBSTITUTION</b> <b>MODEL:</b>
       <b>-m</b> STRING
              Model name string (e.g. GTR+F+I+G)

       DNA:   HKY (default), JC, F81, K2P, K3P, K81uf, TN/TrN, TNef,  TIM,  TIMef,  TVM,  TVMef,  SYM,  GTR,  or
              6-digit model specification (e.g., 010010 = HKY)

       Protein:
              LG  (default),  Poisson,  cpREV,  mtREV, Dayhoff, mtMAM, JTT, WAG, mtART, mtZOA, VT, rtREV, DCMut,
              PMB, HIVb, HIVw, JTTDCMut, FLU, Blosum62, GTR20, mtMet, mtVer, mtInv, FLAVI,

              Q.LG, Q.pfam, Q.pfam_gb, Q.bird, Q.mammal, Q.insect, Q.plant, Q.yeast

       Protein mixture:
              C10,...,C60, EX2, EX3, EHO, UL2, UL3, EX_EHO, LG4M, LG4X

       Binary:
              JC2 (default), GTR2

       Empirical codon:
              KOSI07, SCHN05

       Mechanistic codon:
              GY (default), MG, MGK, GY0K, GY1KTS, GY1KTV, GY2K, MG1KTS, MG1KTV, MG2K

       Semi-empirical codon:  XX_YY where XX is empirical and YY is mechanistic model

       Morphology/SNP:
              MK (default), ORDERED, GTR

       Lie Markov DNA:
              1.1, 2.2b, 3.3a, 3.3b, 3.3c, 3.4, 4.4a, 4.4b, 4.5a, 4.5b, 5.6a, 5.6b,  5.7a,  5.7b,  5.7c,  5.11a,
              5.11b,  5.11c,  5.16,  6.6,  6.7a,  6.7b, 6.8a, 6.8b, 6.17a, 6.17b, 8.8, 8.10a, 8.10b, 8.16, 8.17,
              8.18, 9.20a, 9.20b, 10.12, 10.34, 12.12 (optionally prefixed by RY, WS or MK)

       Non-reversible:
              STRSYM (strand symmetric model, equiv. WS6.6), NONREV, UNREST (unrestricted model, equiv. 12.12)

       Otherwise:
              Name of file containing user-model parameters

   <b>STATE</b> <b>FREQUENCY:</b>
       <b>-m</b> ...+F
              Empirically counted frequencies from alignment

       <b>-m</b> ...+FO
              Optimized frequencies by maximum-likelihood

       <b>-m</b> ...+FQ
              Equal frequencies

       <b>-m</b> ...+FRY
              For DNA, freq(A+G)=1/2=freq(C+T)

       <b>-m</b> ...+FWS
              For DNA, freq(A+T)=1/2=freq(C+G)

       <b>-m</b> ...+FMK
              For DNA, freq(A+C)=1/2=freq(G+T)

       <b>-m</b> ...+Fabcd
              4-digit constraint on ACGT frequency (e.g. +F1221 means f_A=f_T, f_C=f_G)

       <b>-m</b> ...+FU
              Amino-acid frequencies given protein matrix

       <b>-m</b> ...+F1x4
              Equal NT frequencies over three codon positions

       <b>-m</b> ...+F3x4
              Unequal NT frequencies over three codon positions

   <b>RATE</b> <b>HETEROGENEITY</b> <b>AMONG</b> <b>SITES:</b>
       <b>-m</b> ...+I
              A proportion of invariable sites

       <b>-m</b> ...+G[n]
              Discrete Gamma model with n categories (default n=4)

       <b>-m</b> ...*G[n]
              Discrete Gamma model with unlinked model parameters

       <b>-m</b> ...+I+G[n]
              Invariable sites plus Gamma model with n categories

       <b>-m</b> ...+R[n]
              FreeRate model with n categories (default n=4)

       <b>-m</b> ...*R[n]
              FreeRate model with unlinked model parameters

       <b>-m</b> ...+I+R[n]
              Invariable sites plus FreeRate model with n categories

       <b>-m</b> ...+Hn
              Heterotachy model with n classes

       <b>-m</b> ...*Hn
              Heterotachy model with n classes and unlinked parameters

       <b>--alpha-min</b> NUM
              Min Gamma shape parameter for site rates (default: 0.02)

       <b>--gamma-median</b>
              Median approximation for +G site rates (default: mean)

       <b>--rate</b> Write empirical Bayesian site rates to .rate file

       <b>--mlrate</b>
              Write maximum likelihood site rates to .mlrate file

   <b>POLYMORPHISM</b> <b>AWARE</b> <b>MODELS</b> <b>(PoMo):</b>
       <b>-s</b> FILE
              Input counts file (see manual)

       <b>-m</b> ...+P
              DNA substitution model (see above) used with PoMo

       <b>-m</b> ...+N&lt;POPSIZE&gt;
              Virtual population size (default: 9)

       <b>-m</b> ...+WB|WH|S]
              Weighted binomial sampling

       <b>-m</b> ...+WH
              Weighted hypergeometric sampling

       <b>-m</b> ...+S
              Sampled sampling

       <b>-m</b> ...+G[n]
              Discrete Gamma rate with n categories (default n=4)

   <b>COMPLEX</b> <b>MODELS:</b>
       <b>-m</b> "MIX{m1,...,mK}"
              Mixture model with K components

       <b>-m</b> "FMIX{f1,...fK}"
              Frequency mixture model with K components

       <b>--mix-opt</b>
              Optimize mixture weights (default: detect)

       <b>-m</b> ...+ASC
              Ascertainment bias correction

       <b>--tree-freq</b> FILE
              Input tree to infer site frequency model

       <b>--site-freq</b> FILE
              Input site frequency model file

       <b>--freq-max</b>
              Posterior maximum instead of mean approximation

   <b>TREE</b> <b>TOPOLOGY</b> <b>TEST:</b>
       <b>--trees</b> FILE
              Set of trees to evaluate log-likelihoods

       <b>--test</b> NUM
              Replicates for topology test

       <b>--test-weight</b>
              Perform weighted KH and SH tests

       <b>--test-au</b>
              Approximately unbiased (AU) test (Shimodaira 2002)

       <b>--sitelh</b>
              Write site log-likelihoods to .sitelh file

   <b>ANCESTRAL</b> <b>STATE</b> <b>RECONSTRUCTION:</b>
       <b>--ancestral</b>
              Ancestral state reconstruction by empirical Bayes

       <b>--asr-min</b> NUM
              Min probability of ancestral state (default: equil freq)

   <b>TEST</b> <b>OF</b> <b>SYMMETRY:</b>
       <b>--symtest</b>
              Perform three tests of symmetry

       <b>--symtest-only</b>
              Do <b>--symtest</b> then exist

       <b>--symtest-remove-bad</b>
              Do <b>--symtest</b> and remove bad partitions

       <b>--symtest-remove-good</b>
              Do <b>--symtest</b> and remove good partitions

       <b>--symtest-type</b> MAR|INT
              Use MARginal/INTernal test when removing partitions

       <b>--symtest-pval</b> NUMER
              P-value cutoff (default: 0.05)

       <b>--symtest-keep-zero</b>
              Keep NAs in the tests

   <b>CONCORDANCE</b> <b>FACTOR</b> <b>ANALYSIS:</b>
       <b>-t</b> FILE
              Reference tree to assign concordance factor

       <b>--gcf</b> FILE
              Set of source trees for gene concordance factor (gCF)

       <b>--df-tree</b>
              Write discordant trees associated with gDF1

       <b>--scf</b> NUM
              Number of quartets for site concordance factor (sCF)

       <b>-s</b> FILE
              Sequence alignment for <b>--scf</b>

       <b>-p</b> FILE|DIR
              Partition file or directory for <b>--scf</b>

       <b>--cf-verbose</b>
              Write CF per tree/locus to cf.stat_tree/_loci

       <b>--cf-quartet</b>
              Write sCF for all resampled quartets to .cf.quartet

iqtree2-omp 2.0.7+dfsg                            November 2024                                   <u><a href="../man1/IQTREE2-OMP.1.html">IQTREE2-OMP</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>