<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>gt-seed_extend - Calculate local alignments using the seed and extend algorithm.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/genometools">genometools_1.6.5+ds-2.2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       gt-seed_extend - Calculate local alignments using the seed and extend algorithm.

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>gt</b> <b>seed_extend</b> [option ...] encseq_basename [encseq_basename]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>-ii</b> [<u>string</u>]
           Input index for encseq encoded sequences

       <b>-qii</b> [<u>string</u>]
           Query input index (encseq)

       <b>-seedlength</b> [<u>value</u>]
           Minimum length of a seed default: logarithm of input length with alphabet size as log-base

       <b>-spacedseed</b> [<u>value</u>]
           use spaced seed of length specified by option -seedlength (optional argument specifies weight of
           spaced seed) (default: 0)

       <b>-diagbandwidth</b> [<u>value</u>]
           Logarithm of diagonal band width in the range from 0 to 10 (for filter) (default: 6)

       <b>-mincoverage</b> [<u>value</u>]
           Minimum coverage in two neighbouring diagonal bands (for filter) default: 2.5 x seedlength

       <b>-diagband-stat</b> [<u>...</u>]
           Compute statistics from diagonal band scores; parameter specifies kind of statistics, possible
           choices are sum (default: sum)

       <b>-maxfreq</b> [<u>value</u>]
           Maximum frequency of a k-mer (for filter) (default: undefined)

       <b>-memlimit</b> [<u>string</u>]
           Maximum memory usage to determine the maximum frequency of a k-mer (for filter) (default: undefined)

       <b>-extendxdrop</b> [<u>value</u>]
           Extend seed to both sides using xdrop algorithm, optional parameter specifies sensitivity (default:
           97)

       <b>-xdropbelow</b> [<u>value</u>]
           Specify xdrop cutoff score (0 means automatically defined depending on minidentity)

       <b>-extendgreedy</b> [<u>value</u>]
           Extend seed to both sides using greedy algorithm, optional parameter specifies sensitivity (default:
           97)

       <b>-maxmat</b> [<u>value</u>]
           compute maximal matches of minimum length specified by option -l (default: 1)

       <b>-l</b> [<u>value</u>]
           Minimum length of aligned sequences (default: undefined)

       <b>-minidentity</b> [<u>value</u>]
           Minimum identity of matches (for seed extension) (default: 80)

       <b>-evalue</b> [<u>value</u>]
           switch on evalue filtering of matches (optional argument specifies evalue threshold) (default:
           10.000000)

       <b>-outfmt</b>
           specify what information about the matches to display alignment: show alignment (possibly followed by
           =&lt;number&gt; to specify width of alignment columns, default is ) seed_in_algn: mark the seed in
           alignment polinfo: add polishing information for shown alignment seed: abbreviation for seed.len
           seed.s seed.q failed_seed: show the coordinates of a seed extension, which does not satisfy the
           filter conditions fstperquery: output only the first found match per query tabsep: separate columns
           by tabulator, instead of space as default blast: output matches in blast format 7 (tabular with
           comment lines; instead of gap opens, indels are displayed) gfa2: output matches in gfa2 format
           custom: output matches in custom format, i.e. no columns are pre-defined; all columns have to be
           specified by the user cigar: display cigar string representing alignment (no distinction between
           match and mismatch) cigarX: display cigar string representing alignment (distinction between match
           (=) and mismatch (X)) trace: display trace, i.e. a compact representation of an alignment (possibly
           followed by =&lt;delta&gt;) to specify the delta-value; default value of delta is 50 dtrace: display trace
           as differences; like trace, but instead of an absolute value x, report the difference delta-x. This
           leads to smaller numbers and thus a more compact representation s.len: display length of match on
           subject sequence s.seqnum: display sequence number of subject sequence subject id: display sequence
           description of subject sequence s.start: display start position of match on subject sequence s.end:
           display end position of match on subject sequence strand: display strand of match using symbols F
           (forward) and P (reverse complement) q.len: display length of match on query sequence q.seqnum:
           display sequence number of query sequence query id: display sequence description of query sequence
           q.start: display start position of match on query sequence q.end: display end position of match on
           query sequence alignment length: display length of alignment mismatches: display number of mismatches
           in alignment indels: display number of indels in alignment gap opens: display number of indels in
           alignment score: display score of match editdist: display unit edit distance identity: display
           percent identity of match seed.len: display length seed of the match seed.s: display start position
           of seed in subject seed.q: display start position of seed in query s.seqlen: display length of
           subject sequence in which match occurs q.seqlen: display length of query sequence in which match
           occurs evalue: display evalue bit score: display bit score

       <b>-ani</b> [<u>yes|no</u>]
           output average nucleotide identity determined from the computed matches (which are not output)
           (default: no)

       <b>-no-reverse</b> [<u>yes|no</u>]
           do not compute matches on reverse complemented strand (default: no)

       <b>-no-forward</b> [<u>yes|no</u>]
           do not compute matches on forward strand (default: no)

       <b>-use-apos</b> [<u>yes|no</u>]
           Discard a seed only if both apos and bpos overlap with a previous successful alignment (default: no)

       <b>-parts</b> [<u>value</u>]
           Divide data into specified number of parts (default: 1)

       <b>-kmerfile</b> [<u>yes|no</u>]
           Use pre-calculated k-mers from file (if exist) (default: yes)

       <b>-v</b> [<u>yes|no</u>]
           be verbose (default: no)

       <b>-help</b>
           display help and exit

       <b>-version</b>
           display version information and exit

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       Report bugs to https://github.com/genometools/genometools/issues.

GenomeTools 1.6.5                                  04/27/2024                                  <u><a href="../man1/GT-SEED_EXTEND.1.html">GT-SEED_EXTEND</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>