<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Copyright [2015-2018] EMBL-European Bioinformatics Institute</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-db-hts-perl">libbio-db-hts-perl_3.01-5_amd64</a> <br><br><pre>
</pre><h4><b>LICENSE</b></h4><pre>
       Copyright [2015-2018] EMBL-European Bioinformatics Institute

       Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
       compliance with the License.  You may obtain a copy of the License at

            <a href="http://www.apache.org/licenses/LICENSE-2.0">http://www.apache.org/licenses/LICENSE-2.0</a>

       Unless required by applicable law or agreed to in writing, software distributed under the License is
       distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
       See the License for the specific language governing permissions and limitations under the License.

</pre><h4><b>NAME</b></h4><pre>
       Faidx -- Perl extension for accessing bgzip compressed and indexed FASTA using htslib

</pre><h4><b>SYNOPSIS</b></h4><pre>
        #include the module
        use Bio::DB::HTS::Faidx;

        #create the index object
        my $fasta = "$Bin/data/Saccharomyces_cerevisiae.R64-1-1.dna.chromosome.I.fa.gz" ;
        my $location = "I:1-100" ;
        my $seq_id = "I" ;
        my $index = Bio::DB::HTS::Faidx-&gt;new($fasta);

        #get sequence using the location
        my $seq = "" ;
        my $length = 0 ;
        ($seq, $length) = $index-&gt;get_sequence($location);

        $seq = $index-&gt;get_sequence_no_length($location);
        $length = $index-&gt;length($seq_id);

        #get sequence using the separate parameters.
        #Note here that the sequence start and end points are zero indexed, so code accordingly.
        ($seq, $length) = $index-&gt;get_sequence2("I",1,99);
        $seq = $index-&gt;get_sequence2_no_length("I",1,99);

        my @seq_ids = $index-&gt;get_all_sequence_ids();

        #returns 1 if sequence ID is present, 0 if not
        my $has_seq = $index-&gt;has_sequence('I');

</pre><h4><b>AUTHOR</b></h4><pre>
       Rishi Nag

perl v5.40.0                                       2024-10-31                           <u>Bio::DB::HTS::<a href="../man3pm/Faidx.3pm.html">Faidx</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>