<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>tbl2asn - prepare a GenBank submission using an ASCII feature table</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/ncbi-tools-bin">ncbi-tools-bin_6.1.20170106+dfsg2-5_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       tbl2asn - prepare a GenBank submission using an ASCII feature table

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>tbl2asn</b>  [<b>-</b>]  [<b>-A</b> <u>str</u>]  [<b>-C</b> <u>str</u>]  [<b>-D</b> <u>filename</u>]  [<b>-E</b>]  [<b>-F</b> <u>str</u>] [<b>-G</b> <u>str</u>] [<b>-H</b> <u>str</u>] [<b>-J</b>] [<b>-K</b>] [<b>-L</b>] [<b>-M</b> <u>str</u>]
       [<b>-N</b> <u>n</u>] [<b>-O</b>] [<b>-P</b>] [<b>-Q</b> <u>str</u>] [<b>-R</b>] [<b>-S</b>] [<b>-T</b>] [<b>-U</b>] [<b>-V</b> <u>str</u>] [<b>-W</b>] [<b>-X</b> <u>str</u>] [<b>-Y</b> <u>filename</u>] [<b>-Z</b> <u>filename</u>] [<b>-a</b> <u>str</u>]
       [<b>-b</b>] [<b>-c</b> <u>str</u>]  [<b>-f</b> <u>filename</u>]  [<b>-g</b>]  [<b>-h</b>]  [<b>-i</b> <u>filename</u>]  [<b>-j</b> <u>str</u>]  [<b>-k</b> <u>str</u>]  [<b>-l</b> <u>str</u>]  [<b>-m</b> <u>str</u>]  [<b>-n</b> <u>str</u>]
       [<b>-o</b> <u>filename</u>] [<b>-p</b> <u>str</u>] [<b>-q</b>] [<b>-r</b> <u>str</u>] [<b>-s</b>] [<b>-t</b> <u>filename</u>] [<b>-u</b>] [<b>-v</b>] [<b>-w</b> <u>filename</u>] [<b>-x</b> <u>str</u>] [<b>-y</b> <u>str</u>] [<b>-z</b>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>tbl2asn</b>  reads  a  template  along  with  sequence  and  table files, and outputs ASN.1 for submission to
       GenBank. Thus, the submitter does not need to read each set of table  and  sequence  files  into  Sequin.
       Furthermore,  the template file can contain the organism and submitter information common to all records,
       obviating the need to input these data for each sequence/table pair.

</pre><h4><b>OPTIONS</b></h4><pre>
       A summary of options is included below.

       <b>-</b>      Print usage message

       <b>-a</b> <u>str</u> Accession

       <b>-C</b> <u>str</u> Genome Center tag

       <b>-D</b> <u>filename</u>
              Descriptors file

       <b>-E</b>     Recurse

       <b>-F</b>     Feature ID links:
              o      By Overlap
              p      By Product
              l      By Label and Location
              s      Suppress links forced by <b>-M</b>

       <b>-G</b> <u>str</u> Alignment Gap Flags (comma separated fields, e.g., <b>p,-,-,-,?,.</b> ) <b>n</b> Nucleotide or <b>p</b> Protein, Begin,
              Middle, End Gap Characters, Missing Characters, Match Characters Alignment middle Gap characters

       <b>-H</b> <u>str</u> Hold until publication:
              y      For one year
              <u>mm/dd/yyyy</u>
                     Until the specified date

       <b>-J</b>     Delayed genomic product set

       <b>-K</b>     Safe Bioseq-set

       <b>-L</b>     Force Local protein_id/transcript_id

       <b>-M</b> <u>str</u> Master genome flags:
              n      Normal
              b      Big sequence
              p      Power option
              t      TSA

       <b>-N</b> <u>n</u>   Project version number

       <b>-O</b>     Allow run-on ORFs

       <b>-P</b>     Remote publication lookup

       <b>-Q</b>     mRNA title policy
              s      Special mRNA titles
              r      RefSeq mRNA titles

       <b>-R</b>     Remote sequence record fetching from ID

       <b>-S</b>     Smart feature annotation

       <b>-T</b>     Remote Taxonomy lookup

       <b>-U</b>     Remove Unnecessary gene xref

       <b>-V</b> <u>str</u> Verification (combine any of the following letters)
              v      Validate with Normal Stringency
              r      Validate without Country Check
              c      BarCode validation
              b      Generate GenBank Flatfile
              g      Generate Gene Report
              t      Validate with TSA check

       <b>-W</b>     Log progress

       <b>-X</b> <u>str</u> Extra flags (combine any of the following letters)
              A      Automatic definition line generation
              C      Apply Comments in <b>.cmt</b> files to all sequences
              E      Treat like Eukarypota in the discrepancy report

       <b>-Y</b> <u>filename</u>
              Read a comment string from <u>filename</u>

       <b>-Z</b> <u>filename</u>
              Write a discrepancy report to <u>filename</u>

       <b>-a</b> <u>str</u> File type:
              a      Any (default)
              r20u   Runs of 20+ Ns are gaps, 100 Ns are unknown length
              r20k   Runs of 20+ Ns are gaps, 100 Ns are known length
              r10u   Runs of 10+ Ns are gaps, 100 Ns are unknown length
              r10k   Runs of 10+ Ns are gaps, 100 Ns are known length
              s      FASTA Set (<b>s</b> Batch, <b>s1</b> Pop, <b>s2</b> Phy, <b>s3</b> Mut, <b>s4</b> Eco, <b>s9</b> Small-genome)
              d      FASTA Delta
              di     FASTA Delta with Implicit Gaps
              l      FASTA+Gap Alignment (<b>l</b> Batch, <b>l1</b> Pop, <b>l2</b> Phy, <b>l3</b> Mut, <b>l4</b> Eco, <b>l9</b> Small-genome)
              z      FASTA with Gap Lines
              e      PHRAP/ACE                  -
              b      ASN.1 (in conjunction with M
                                                )
       <b>-b</b>     Generate GenBank file (deprecated in favor of <b>-V</b> <b>b</b>)

       <b>-c</b> <u>str</u> Cleanup (combine any of the following letters)
              d      Correct Collection Dates (assume month first)
              D      Correct Collection Dates (assume day first)
              b      Append note to coding regions that overlap other coding regions with similar product  names
                     and do not contain 'ABC'
              x      Extend partial ends of features by one or two nucleotides to abut gaps or sequence ends
              p      Add exception to non-extendable partials
              s      Add exception to short introns
              f      Fix product names

       <b>-f</b> <u>filename</u>
              Single table file

       <b>-g</b>     Input is a genomic product set

       <b>-h</b>     Convert general ID to note

       <b>-i</b> <u>filename</u>
              Single input file

       <b>-j</b> <u>str</u> Source qualifiers

       <b>-k</b> <u>str</u> CDS flags (combine any of the following letters)
              c      Annotate Longest ORF
              r      Allow Runon ORFs
              m      Allow Alternative Starts
              k      Set Conflict on Mismatch

       <b>-l</b> <u>str</u> Add  type of evidence used to assert linkage across assembly gaps (only for TSA records).  Must be
              one of the following:
              paired-ends
              align-genus
              align-xgenus
              align-trnscpt
              within-clone
              clone-contig
              map
              strobe

       <b>-m</b> <u>str</u> Lineage to use for discrepancy report tests

       <b>-n</b> <u>str</u> Organism name

       <b>-o</b> <u>filename</u>
              Single output file

       <b>-p</b> <u>str</u> Path to files

       <b>-q</b>     Set sequence ID from input file name

       <b>-r</b> <u>str</u> Path for results

       <b>-s</b>     Read FASTAs as Set

       <b>-t</b> <u>filename</u>
              Read template from <u>filename</u>

       <b>-u</b>     Convert GenProdSet to NucProtSet

       <b>-v</b>     Validate (deprecated in favor of <b>-V</b> <b>v</b>)

       <b>-w</b> <u>filename</u>
              Single structured comment file (overrides the use of <b>-X</b> <b>C</b>)

       <b>-x</b> <u>str</u> Suffix (default = <b>.fsa</b>)

       <b>-y</b> <u>str</u>

       <b>-z</b>     Clean up log file Comment

</pre><h4><b>AUTHOR</b></h4><pre>
       The National Center for Biotechnology Information.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/Psequin.1.html">Psequin</a></b>(1), <b><a href="../man1/sbtedit.1.html">sbtedit</a></b>(1), /usr/share/doc/ncbi-tools-bin/tbl2asn.txt.gz,
       &lt;<a href="http://www.ncbi.nlm.nih.gov/Sequin/table.html">http://www.ncbi.nlm.nih.gov/Sequin/table.html</a>&gt;.

NCBI                                               2016-09-01                                         <u><a href="../man1/TBL2ASN.1.html">TBL2ASN</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>