<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>hhsearch - search a database of HMMs with a query alignment or query HMM</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/hhsuite">hhsuite_3.3.0+ds-8_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       hhsearch - search a database of HMMs with a query alignment or query HMM

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>hhsearch</b> <u>-i</u> <u>query</u> <u>-d</u> <u>database</u> [<u>options</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       HHsearch 3.3.0 Search a database of HMMs with a query alignment or query HMM (c) The HH-suite development
       team  Steinegger M, Meier M, Mirdita M, V??hringer H, Haunsberger S J, and S??ding J (2019) HH-suite3 for
       fast   remote   homology    detection    and    deep    protein    annotation.     BMC    Bioinformatics,
       doi:10.1186/s12859-019-3019-7

       <b>-i</b> &lt;file&gt;
              input/query multiple sequence alignment (a2m, a3m, FASTA) or HMM

       &lt;file&gt; may be 'stdin' or 'stdout' throughout.  Options:

       <b>-d</b> &lt;name&gt;
              database  name  (e.g.  uniprot20_29Feb2012)  Multiple databases may be specified with '-d &lt;db1&gt; <b>-d</b>
              &lt;db2&gt; ...'

       <b>-e</b>     [0,1]   E-value cutoff for inclusion in result alignment (def=0.001)

   <b>Input</b> <b>alignment</b> <b>format:</b>
       <b>-M</b> a2m use A2M/A3M (default): upper case = Match; lower case = Insert;

              '-' = Delete; '.' = gaps aligned to inserts (may be omitted)

       <b>-M</b> first
              use FASTA: columns with residue in 1st sequence are match states

       <b>-M</b> [0,100]
              use FASTA: columns with fewer than X% gaps are match states

       <b>-tags</b>/-notags
              do NOT / do neutralize His-, C-myc-, FLAG-tags, and trypsin  recognition  sequence  to  background
              distribution (def=-notags)

   <b>Output</b> <b>options:</b>
       <b>-o</b> &lt;file&gt;
              write results in standard format to file (default=&lt;infile.hhr&gt;)

       <b>-oa3m</b> &lt;file&gt;
              write result MSA with significant matches in a3m format

       <b>-blasttab</b> &lt;name&gt; write result in tabular BLAST format (compatible to <b>-m</b> 8 or <b>-outfmt</b> 6 output)

       1      2      3           4      5         6        7      8    9      10   11   12

              query target #match/tLen alnLen #mismatch #gapOpen qstart qend tstart tend eval score

       <b>-opsi</b> &lt;file&gt;
              write result MSA of significant matches in PSI-BLAST format

       <b>-ohhm</b> &lt;file&gt;
              write HHM file for result MSA of significant matches

       <b>-add_cons</b>
              generate consensus sequence as master sequence of query MSA (default=don't)

       <b>-hide_cons</b>
              don't show consensus sequence in alignments (default=show)

       <b>-hide_pred</b>
              don't show predicted 2ndary structure in alignments (default=show)

       <b>-hide_dssp</b>
              don't show DSSP 2ndary structure in alignments (default=show)

       <b>-show_ssconf</b>
              show confidences for predicted 2ndary structure in alignments

       <b>-Ofas</b> &lt;file&gt;
              write pairwise alignments in FASTA xor A2M (<b>-Oa2m</b>) xor A3M (<b>-Oa3m</b>) format

       <b>-seq</b> &lt;int&gt;
              max. number of query/template sequences displayed (default=1)

       <b>-aliw</b> &lt;int&gt;
              number of columns per line in alignment list (default=80)

       <b>-p</b> [0,100]
              minimum probability in summary and alignment list (default=20)

       <b>-E</b> [0,inf[
              maximum E-value in summary and alignment list (default=1E+06)

       <b>-Z</b> &lt;int&gt;
              maximum number of lines in summary hit list (default=500)

       <b>-z</b> &lt;int&gt;
              minimum number of lines in summary hit list (default=10)

       <b>-B</b> &lt;int&gt;
              maximum number of alignments in alignment list (default=500)

       <b>-b</b> &lt;int&gt;
              minimum number of alignments in alignment list (default=10)

       Filter options applied to query MSA, database MSAs, and result MSA

       <b>-all</b>   show all sequences in result MSA; do not filter result MSA

       <b>-id</b>    [0,100]  maximum pairwise sequence identity (def=90)

       <b>-diff</b> [0,inf[
              filter  MSAs  by  selecting most diverse set of sequences, keeping at least this many seqs in each
              MSA block of length 50 Zero and non-numerical values turn off the filtering. (def=100)

       <b>-cov</b>   [0,100]  minimum coverage with master sequence (%) (def=0)

       <b>-qid</b>   [0,100]  minimum sequence identity with master sequence (%) (def=0)

       <b>-qsc</b>   [0,100]  minimum score per column with master sequence (default=-20.0)

       <b>-neff</b> [1,inf]
              target diversity of multiple sequence alignment (default=off)

       <b>-mark</b>  do not filter out sequences marked by "&gt;@"in their name line

   <b>HMM-HMM</b> <b>alignment</b> <b>options:</b>
       <b>-norealign</b>
              do NOT realign displayed hits with MAC algorithm (def=realign)

       <b>-ovlp</b> &lt;int&gt;
              banded alignment: forbid &lt;ovlp&gt; largest diagonals |i-j| of DP matrix (def=0)

       <b>-mact</b> [0,1[
              posterior prob threshold for MAC realignment controlling greediness at  alignment  ends:  0:global
              &gt;0.1:local (default=0.35)

       <b>-glob</b>/-loc
              use global/local alignment mode for searching/ranking (def=local)

       <b>-realign</b>
              realign displayed hits with max. accuracy (MAC) algorithm

       <b>-excl</b> &lt;range&gt;
              exclude query positions from the alignment, e.g. '1-33,97-168'

       <b>-realign_max</b> &lt;int&gt;
              realign max. &lt;int&gt; hits (default=500)

       <b>-alt</b> &lt;int&gt;
              show up to this many alternative alignments with raw score &gt; smin(def=4)

       <b>-smin</b> &lt;float&gt;
              minimum raw score for alternative alignments (def=20.0)

       <b>-shift</b> [-1,1]
              profile-profile score offset (def=-0.03)

       <b>-corr</b> [0,1]
              weight of term for pair correlations (def=0.10)

       <b>-sc</b>    &lt;int&gt;         amino acid score         (tja: template HMM at column j) (def=1)

       0      = log2 Sum(tja*qia/pa)   (pa: aa background frequencies)

       1      = log2 Sum(tja*qia/pqa)  (pqa = 1/2*(pa+ta) )

       2      = log2 Sum(tja*qia/ta)   (ta: av. aa freqs in template)

       3      = log2 Sum(tja*qia/qa)   (qa: av. aa freqs in query)

       5      local amino acid composition correction

       <b>-ssm</b> {0,..,4}
              0:   no ss scoring

       1,2: ss scoring after or during alignment
              [default=2]

              3,4: ss scoring after or during alignment, predicted vs. predicted

       <b>-ssw</b> [0,1]
              weight of ss score  (def=0.11)

       <b>-ssa</b> [0,1]
              SS substitution matrix = (1-ssa)*I + ssa*full-SS-substition-matrix [def=1.00)

       <b>-wg</b>    use global sequence weighting for realignment!

   <b>Gap</b> <b>cost</b> <b>options:</b>
       <b>-gapb</b> [0,inf[
              Transition pseudocount admixture (def=1.00)

       <b>-gapd</b> [0,inf[
              Transition pseudocount admixture for open gap (default=0.15)

       <b>-gape</b> [0,1.5]
              Transition pseudocount admixture for extend gap (def=1.00)

       <b>-gapf</b> ]0,inf]
              factor to increase/reduce gap open penalty for deletes (def=0.60)

       <b>-gapg</b> ]0,inf]
              factor to increase/reduce gap open penalty for inserts (def=0.60)

       <b>-gaph</b> ]0,inf]
              factor to increase/reduce gap extend penalty for deletes(def=0.60)

       <b>-gapi</b> ]0,inf]
              factor to increase/reduce gap extend penalty for inserts(def=0.60)

       <b>-egq</b>   [0,inf[  penalty (bits) for end gaps aligned to query residues (def=0.00)

       <b>-egt</b>   [0,inf[  penalty (bits) for end gaps aligned to template residues (def=0.00)

   <b>Pseudocount</b> <b>(pc)</b> <b>options:</b>
              Context specific hhm pseudocounts:

       <b>-pc_hhm_contxt_mode</b> {0,..,3}
              position dependence of pc admixture 'tau' (pc mode, default=2)

       0: no pseudo counts:
              tau = 0

       1: constant
              tau = a

              2:   diversity-dependent:   tau   =   a/(1+((Neff[i]-1)/b)^c)   3:   CSBlast  admixture:    tau  =
              a(1+b)/(Neff[i]+b) (Neff[i]: number of effective seqs in local MSA around column i)

       <b>-pc_hhm_contxt_a</b>
              [0,1]        overall pseudocount admixture (def=0.9)

       <b>-pc_hhm_contxt_b</b>
              [1,inf[      Neff threshold value for mode 2 (def=4.0)

       <b>-pc_hhm_contxt_c</b>
              [0,3]        extinction exponent c for mode 2 (def=1.0)

              Context independent hhm pseudocounts (used for templates; used for query if  contxt  file  is  not
              available):

       <b>-pc_hhm_nocontxt_mode</b> {0,..,3}
              position dependence of pc admixture 'tau' (pc mode, default=2)

       0: no pseudo counts:
              tau = 0

       1: constant
              tau = a

              2:  diversity-dependent: tau = a/(1+((Neff[i]-1)/b)^c) (Neff[i]: number of effective seqs in local
              MSA around column i)

       <b>-pc_hhm_nocontxt_a</b>
              [0,1]        overall pseudocount admixture (def=1.0)

       <b>-pc_hhm_nocontxt_b</b>
              [1,inf[      Neff threshold value for mode 2 (def=1.5)

       <b>-pc_hhm_nocontxt_c</b>
              [0,3]        extinction exponent c for mode 2 (def=1.0)

              Context-specific pseudo-counts:

       <b>-nocontxt</b>
              use substitution-matrix instead of context-specific pseudocounts

       <b>-contxt</b> &lt;file&gt; context file for computing context-specific pseudocounts (default=)

       <b>-csw</b>   [0,inf]  weight of central position in cs pseudocount mode (def=1.6)

       <b>-csb</b>   [0,1]    weight decay parameter for positions in cs pc mode (def=0.9)

   <b>Other</b> <b>options:</b>
       <b>-v</b> &lt;int&gt;
              verbose mode: 0:no screen output  1:only warnings  2: verbose (def=2)

       <b>-cpu</b> &lt;int&gt;
              number of CPUs to use (for shared memory SMPs) (default=2)

       <b>-scores</b> &lt;file&gt; write scores for all pairwise comparisons to file

       <b>-atab</b>  &lt;file&gt; write all alignments in tabular layout to file

       <b>-maxseq</b> &lt;int&gt;
              max number of input rows (def=65535)

       <b>-maxres</b> &lt;int&gt;
              max number of HMM columns (def=20001)

       <b>-maxmem</b> [1,inf[ limit memory for realignment (in GB) (def=3.0)

       Example: hhsearch <b>-i</b> a.1.1.1.a3m <b>-d</b> scop70_1.71

       Download databases from &lt;<a href="http://wwwuser.gwdg.de/~compbiol/data/hhsuite/databases/hhsuite_dbs/">http://wwwuser.gwdg.de/~compbiol/data/hhsuite/databases/hhsuite_dbs/</a>&gt;.

hhsearch 3.3.0+ds                                  August 2023                                       <u><a href="../man1/HHSEARCH.1.html">HHSEARCH</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>