<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>mash-dist - estimate the distance of query sequences to references</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/mash">mash_2.3+dfsg-6build2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       mash-dist - estimate the distance of query sequences to references

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>mash</b> <b>dist</b> [options] &lt;reference&gt; &lt;query&gt; [&lt;query&gt;] ...

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Estimate the distance of each query sequence to the reference. Both the reference and queries can be
       fasta or fastq, gzipped or not, or Mash sketch files (.msh) with matching k-mer sizes. Query files can
       also be files of file names (see <b>-l</b>). Whole files are compared by default (see <b>-i</b>). The output fields are
       [reference-ID, query-ID, distance, p-value, shared-hashes].

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>
           Help

       <b>-p</b> &lt;int&gt;
           Parallelism. This many threads will be spawned for processing. [1]

   <b>Input</b>
       <b>-l</b>
           List input. Each query file contains a list of sequence files, one per line. The reference file is
           not affected.

   <b>Output</b>
       <b>-t</b>
           Table output (will not report p-values, but fields will be blank if they do not meet the p-value
           threshold).

       <b>-v</b> &lt;num&gt;
           Maximum p-value to report. (0-1) [1.0]

       <b>-d</b> &lt;num&gt;
           Maximum distance to report. (0-1) [1.0]

   <b>Sketching</b>
       <b>-k</b> &lt;int&gt;
           K-mer size. Hashes will be based on strings of this many nucleotides. Canonical nucleotides are used
           by default (see Alphabet options below). (1-32) [21]

       <b>-s</b> &lt;int&gt;
           Sketch size. Each sketch will have at most this many non-redundant min-hashes. [1000]

       <b>-i</b>
           Sketch individual sequences, rather than whole files.

       <b>-w</b> &lt;num&gt;
           Probability threshold for warning about low k-mer size. (0-1) [0.01]

       <b>-r</b>
           Input is a read set. See Reads options below. Incompatible with <b>-i</b>.

   <b>Sketching</b> <b>(reads)</b>
       <b>-b</b> &lt;size&gt;
           Use a Bloom filter of this size (raw bytes or with K/M/G/T) to filter out unique k-mers. This is
           useful if exact filtering with <b>-m</b> uses too much memory. However, some unique k-mers may pass
           erroneously, and copies cannot be counted beyond 2. Implies <b>-r</b>.

       <b>-m</b> &lt;int&gt;
           Minimum copies of each k-mer required to pass noise filter for reads. Implies <b>-r</b>. [1]

       <b>-c</b> &lt;num&gt;
           Target coverage. Sketching will conclude if this coverage is reached before the end of the input file
           (estimated by average k-mer multiplicity). Implies <b>-r</b>.

       <b>-g</b> &lt;size&gt;
           Genome size. If specified, will be used for p-value calculation instead of an estimated size from
           k-mer content. Implies <b>-r</b>.

   <b>Sketching</b> <b>(alphabet)</b>
       <b>-n</b>
           Preserve strand (by default, strand is ignored by using canonical DNA k-mers, which are alphabetical
           minima of forward-reverse pairs). Implied if an alphabet is specified with <b>-a</b> or <b>-z</b>.

       <b>-a</b>
           Use amino acid alphabet (A-Z, except BJOUXZ). Implies <b>-n</b>, <b>-k</b> 9.

       <b>-z</b> &lt;text&gt;
           Alphabet to base hashes on (case ignored by default; see <b>-Z</b>). K-mers with other characters will be
           ignored. Implies <b>-n</b>.

       <b>-Z</b>
           Preserve case in k-mers and alphabet (case is ignored by default). Sequence letters whose case is not
           in the current alphabet will be skipped when sketching.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <a href="../man1/mash.1.html">mash</a>(1)

                                                   2019-12-13                                       <u><a href="../man1/MASH-DIST.1.html">MASH-DIST</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>