<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>ssw-align - fast Smith-Waterman aligner</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/ssw-align">ssw-align_1.1-15_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       ssw-align - fast Smith-Waterman aligner

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>ssw-align</b> [<u>options</u>] ... <u>&lt;target.fasta&gt;</u> <u>&lt;query.fasta&gt;(or</u> <u>&lt;query.fastq&gt;)</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This  command line tool uses SSW, a fast SIMD optimized alignment library, to perform alignments on input
       files.

       The input files can be in FASTA or FASTQ format.  Both  target  and  query  files  can  contain  multiple
       sequences.  Each sequence in the query file will be aligned with all sequences in the target file. If the
       target file has N sequences and the query file has M sequences, the results will have M*N alignments.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-m</b> N   N is a positive integer for weight match in genome sequence alignment. [default: 2]

       <b>-x</b> N   N is a positive integer. <b>-N</b> will  be  used  as  weight  mismatch  in  genome  sequence  alignment.
              [default: 2]

       <b>-o</b> N   N is a positive integer. <b>-N</b> will be used as the weight for the gap opening. [default: 3]

       <b>-e</b> N   N is a positive integer. <b>-N</b> will be used as the weight for the gap extension. [default: 1]

       <b>-p</b>     Do  protein  sequence  alignment.  Without  this  option,  the  ssw_test  will  do genome sequence
              alignment.

       <b>-a</b> FILE FILE is either the Blosum or Pam weight matrix. [default: Blosum50]

       <b>-c</b>     Return the alignment path.

       <b>-f</b> N   N is a positive integer. Only output the alignments with the Smith-Waterman score &gt;= N.

       <b>-r</b>     The best alignment will be picked between the original read alignment and the  reverse  complement
              read alignment.

       <b>-s</b>     Output in SAM format. [default: no header]

       <b>-h</b>     If <b>-s</b> is used, include header in SAM output.

</pre><h4><b>OUTPUT</b> <b>FORMATS</b></h4><pre>
       The  software can output SAM format or BLAST like format results.  For a specification of the SAM format,
       please see <a href="http://samtools.sourceforge.net/SAM1.pdf">http://samtools.sourceforge.net/SAM1.pdf</a>. The additional optional  field  "ZS"  indicates  the
       suboptimal alignment score.

ssw-align                                           July 2024                                       <u><a href="../man1/SSW-ALIGN.1.html">SSW-ALIGN</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>