<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>YAHA - find split-read mappings on single-end queries</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/yaha">yaha_0.1.83-3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       YAHA - find split-read mappings on single-end queries

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Usage (Default parameter values shown in parenthesis):

       To  create an index: yaha <b>-g</b> genomeFilename [-H maxHits (65525)] [-L wordLen (15)] [-S Skip-distance (1)]
       The genome file can be a FASTA file, or a nib2 file (created by a previous yaha index operation).

       To    align    queries:    yaha    <b>-x</b>    yahaIndexFile    [-q     queryFile|(stdin)]     [-o8|(<b>-osh</b>)|-oss
       outputFile|(stdout)][AdditionalOptions]  The  query file can be either a FASTA file or a FASTQ file.  <b>-o8</b>
       produces alignment output in modified Blast8 format.  <b>-osh</b>    produces alignment  output  in  SAM  format
       with  hard clipping.  <b>-oss</b>    produces alignment output in SAM format with soft clipping.  [-t numThreads
       (1)]

   <b>Additional</b> <b>General</b> <b>Alignment</b> <b>Options:</b>
              [-BW BandWidth (5)] [-G maxGap (50)] [-H maxHits (650)] [-M minMatch (25)]  [-MD  MaxDesert  (50)]
              [-P minPercent-identity (0.9)] [-X Xdropoff (25)]

       [-AGS  (Y)|N]  controls use of Affine Gap Scoring.  If <b>-AGS</b> is off, a simple edit distance calculation is
       done.  If on, the following are used:

              [-GEC GapExtensionCost (2)] [-GOC GapOpenCost (5)] [-MS MatchScore (1)] [-RC ReplacementCost (3)]

       [-OQC (Y)|N] controls use of the Optimal Query Coverage  Algorithm.   If  <b>-OQC</b>  if  off,  all  alignments
       meeting  above  criteria are output.  If on, a set of "primary" alignments are found that optimally cover
       the length of the query, using the following options:

              [-BP BreakpointPenalty (5)] [-MGDP MaxGenomicDistancePenalty (5)] [-MNO MinNonOverlap (minMatch)]

       The cost of inserting a breakpoint in the Optimal Coverage Set  is  BP*MIN(MGDP,  Log10(genomic  distance
       between reference loci)).

       [-FBS Y|(N)] controls inclusion of "secondary" alignments similar to a primary alignment found using OQC.
       If <b>-FBS</b> is on, the following are used.  A "secondary" alignemnt must satisfy BOTH criteria.

              [-PRL PercentReciprocalLength (0.9)] [-PSS PercentSimilarScore (0.9)]

       Additional  experimental  parameters:  To compress a FASTA file to a nib2 file without creating an index:
       yaha <b>-g</b> fastaGenomeFile <b>-c</b> To uncompress a nib2 file back into a FASTA file: yaha  <b>-g</b>  nib2GenomeFile  <b>-u</b>
       For finer control of alignments: [-I maxIntron (maxGap)] allows separate control of max deletion size vs.
       maxGap for insertion size.  For a more detailed help message, type yaha <b>-xh</b>.

       <b>--version</b> is not a valid option.

       Usage (Default parameter values shown in parenthesis):

       To  create an index: yaha <b>-g</b> genomeFilename [-H maxHits (65525)] [-L wordLen (15)] [-S Skip-distance (1)]
       The genome file can be a FASTA file, or a nib2 file (created by a previous yaha index operation).

       To    align    queries:    yaha    <b>-x</b>    yahaIndexFile    [-q     queryFile|(stdin)]     [-o8|(<b>-osh</b>)|-oss
       outputFile|(stdout)][AdditionalOptions]  The  query file can be either a FASTA file or a FASTQ file.  <b>-o8</b>
       produces alignment output in modified Blast8 format.  <b>-osh</b>    produces alignment  output  in  SAM  format
       with  hard clipping.  <b>-oss</b>    produces alignment output in SAM format with soft clipping.  [-t numThreads
       (1)]

   <b>Additional</b> <b>General</b> <b>Alignment</b> <b>Options:</b>
              [-BW BandWidth (5)] [-G maxGap (50)] [-H maxHits (650)] [-M minMatch (25)]  [-MD  MaxDesert  (50)]
              [-P minPercent-identity (0.9)] [-X Xdropoff (25)]

       [-AGS  (Y)|N]  controls use of Affine Gap Scoring.  If <b>-AGS</b> is off, a simple edit distance calculation is
       done.  If on, the following are used:

              [-GEC GapExtensionCost (2)] [-GOC GapOpenCost (5)] [-MS MatchScore (1)] [-RC ReplacementCost (3)]

       [-OQC (Y)|N] controls use of the Optimal Query Coverage  Algorithm.   If  <b>-OQC</b>  if  off,  all  alignments
       meeting  above  criteria are output.  If on, a set of "primary" alignments are found that optimally cover
       the length of the query, using the following options:

              [-BP BreakpointPenalty (5)] [-MGDP MaxGenomicDistancePenalty (5)] [-MNO MinNonOverlap (minMatch)]

       The cost of inserting a breakpoint in the Optimal Coverage Set  is  BP*MIN(MGDP,  Log10(genomic  distance
       between reference loci)).

       [-FBS Y|(N)] controls inclusion of "secondary" alignments similar to a primary alignment found using OQC.
       If <b>-FBS</b> is on, the following are used.  A "secondary" alignemnt must satisfy BOTH criteria.

              [-PRL PercentReciprocalLength (0.9)] [-PSS PercentSimilarScore (0.9)]

       Additional  experimental  parameters:  To compress a FASTA file to a nib2 file without creating an index:
       yaha <b>-g</b> fastaGenomeFile <b>-c</b> To uncompress a nib2 file back into a FASTA file: yaha  <b>-g</b>  nib2GenomeFile  <b>-u</b>
       For finer control of alignments: [-I maxIntron (maxGap)] allows separate control of max deletion size vs.
       maxGap for insertion size.  For a more detailed help message, type yaha <b>-xh</b>.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       https://github.com/GregoryFaust/yaha

YAHA version 0.1.83                                 July 2018                                            <u><a href="../man1/YAHA.1.html">YAHA</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>