<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>mason_frag_sequencing - Fragment Sequencing Simulation</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/seqan-apps">seqan-apps_2.4.0+dfsg-16_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       mason_frag_sequencing - Fragment Sequencing Simulation

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>mason_frag_sequencing</b> [<u>OPTIONS</u>] <b>-i</b> <u>IN.fa</u> <b>-o</b> <u>OUT.{fa,fq}</u> [<b>-or</b> <u>OUT2.{fa,fq}</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Given a FASTA file with fragments, simulate sequencing thereof.

       This program is a more lightweight version of mason_sequencing without support for the application of VCF
       and  fragment  sampling.   Output  of  SAM is also not available.  However, it uses the same code for the
       simulation of the reads as the more powerful mason_simulator.

       You can use mason_frag_sequencing if you want to implement you rown fragmentation behaviour, e.g. if  you
       have implemented your own bias models.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              Display the help message.

       <b>--version</b>
              Display version information.

       <b>-q</b>, <b>--quiet</b>
              Low verbosity.

       <b>-v</b>, <b>--verbose</b>
              Higher verbosity.

       <b>-vv</b>, <b>--very-verbose</b>
              Highest verbosity.

       <b>--seed</b> <u>INTEGER</u>
              Seed to use for random number generator. Default: <u>0</u>.

       <b>-i</b>, <b>--in</b> <u>INPUT_FILE</u>
              Path  to  input  file.  Valid  filetypes  are:  <u>.sam[.*]</u>,  <u>.raw[.*]</u>,  <u>.gbk[.*]</u>, <u>.frn[.*]</u>, <u>.fq[.*]</u>,
              <u>.fna[.*]</u>, <u>.ffn[.*]</u>, <u>.fastq[.*]</u>, <u>.fasta[.*]</u>, <u>.faa[.*]</u>, <u>.fa[.*]</u>, <u>.embl[.*]</u>, and <u>.bam</u>, where * is any
              of the following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent (de)compression.

       <b>-o</b>, <b>--out</b> <u>OUTPUT_FILE</u>
              Output of single-end/left end reads. Valid filetypes are: <u>.sam[.*]</u>, <u>.raw[.*]</u>,  <u>.frn[.*]</u>,  <u>.fq[.*]</u>,
              <u>.fna[.*]</u>,  <u>.ffn[.*]</u>,  <u>.fastq[.*]</u>,  <u>.fasta[.*]</u>,  <u>.faa[.*]</u>, <u>.fa[.*]</u>, and <u>.bam</u>, where * is any of the
              following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent (de)compression.

       <b>-or</b>, <b>--out-right</b> <u>OUTPUT_FILE</u>
              Output of right reads.  Giving this options enables paired-end simulation.  Valid  filetypes  are:
              <u>.sam[.*]</u>,  <u>.raw[.*]</u>,  <u>.frn[.*]</u>,  <u>.fq[.*]</u>,  <u>.fna[.*]</u>,  <u>.ffn[.*]</u>,  <u>.fastq[.*]</u>, <u>.fasta[.*]</u>, <u>.faa[.*]</u>,
              <u>.fa[.*]</u>, and <u>.bam</u>, where * is any of the following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u>  for  transparent
              (de)compression.

       <b>--force-single-end</b>
              Force single-end simulation although --out-right is given.

   <b>Global</b> <b>Read</b> <b>Simulation</b> <b>Options:</b>
       <b>--seq-technology</b> <u>STRING</u>
              Set sequencing technology to simulate. One of <u>illumina</u>, <u>454</u>, and <u>sanger</u>. Default: <u>illumina</u>.

       <b>--seq-mate-orientation</b> <u>STRING</u>
              Orientation  for  paired  reads.   See section Read Orientation below. One of <u>FR</u>, <u>RF</u>, <u>FF</u>, and <u>FF2</u>.
              Default: <u>FR</u>.

       <b>--seq-strands</b> <u>STRING</u>
              Strands to simulate from, only  applicable  to  paired  sequencing  simulation.  One  of  <u>forward</u>,
              <u>reverse</u>, and <u>both</u>. Default: <u>both</u>.

       <b>--embed-read-info</b>
              Whether or not to embed read information.

       <b>--read-name-prefix</b> <u>STRING</u>
              Read names will have this prefix. Default: <u>simulated.</u>.

   <b>BS-Seq</b> <b>Options:</b>
       <b>--enable-bs-seq</b>
              Enable BS-seq simulation.

       <b>--bs-seq-protocol</b> <u>STRING</u>
              Protocol to use for BS-Seq simulation. One of <u>directional</u> and <u>undirectional</u>. Default: <u>directional</u>.

       <b>--bs-seq-conversion-rate</b> <u>DOUBLE</u>
              Conversion rate for unmethylated Cs to become Ts. In range [0..1]. Default: <u>0.99</u>.

   <b>Illumina</b> <b>Options:</b>
       <b>--illumina-read-length</b> <u>INTEGER</u>
              Read length for Illumina simulation. In range [1..inf]. Default: <u>100</u>.

       <b>--illumina-error-profile-file</b> <u>INPUT_FILE</u>
              Path  to  file  with  Illumina  error  profile.   The file must be a text file with floating point
              numbers separated by space, each giving a positional error rate. Valid filetype is: <u>.txt</u>.

       <b>--illumina-prob-insert</b> <u>DOUBLE</u>
              Insert per-base probability for insertion  in  Illumina  sequencing.  In  range  [0..1].  Default:
              <u>0.00005</u>.

       <b>--illumina-prob-deletion</b> <u>DOUBLE</u>
              Insert  per-base  probability  for  deletion  in  Illumina  sequencing.  In range [0..1]. Default:
              <u>0.00005</u>.

       <b>--illumina-prob-mismatch-scale</b> <u>DOUBLE</u>
              Scaling factor for Illumina mismatch probability. In range [0..inf]. Default: <u>1.0</u>.

       <b>--illumina-prob-mismatch</b> <u>DOUBLE</u>
              Average per-base mismatch probability in Illumina sequencing. In range [0.0..1.0]. Default: <u>0.004</u>.

       <b>--illumina-prob-mismatch-begin</b> <u>DOUBLE</u>
              Per-base mismatch probability of first base in Illumina sequencing. In range [0.0..1.0].  Default:
              <u>0.002</u>.

       <b>--illumina-prob-mismatch-end</b> <u>DOUBLE</u>
              Per-base  mismatch  probability of last base in Illumina sequencing. In range [0.0..1.0]. Default:
              <u>0.012</u>.

       <b>--illumina-position-raise</b> <u>DOUBLE</u>
              Point where the error curve raises in relation to read length. In range [0.0..1.0]. Default: <u>0.66</u>.

       <b>--illumina-quality-mean-begin</b> <u>DOUBLE</u>
              Mean PHRED quality for non-mismatch bases of first base in Illumina sequencing. Default: <u>40.0</u>.

       <b>--illumina-quality-mean-end</b> <u>DOUBLE</u>
              Mean PHRED quality for non-mismatch bases of last base in Illumina sequencing. Default: <u>39.5</u>.

       <b>--illumina-quality-stddev-begin</b> <u>DOUBLE</u>
              Standard deviation of PHRED quality for non-mismatch bases of first base in  Illumina  sequencing.
              Default: <u>0.05</u>.

       <b>--illumina-quality-stddev-end</b> <u>DOUBLE</u>
              Standard  deviation  of  PHRED quality for non-mismatch bases of last base in Illumina sequencing.
              Default: <u>10.0</u>.

       <b>--illumina-mismatch-quality-mean-begin</b> <u>DOUBLE</u>
              Mean PHRED quality for mismatch bases of first base in Illumina sequencing. Default: <u>40.0</u>.

       <b>--illumina-mismatch-quality-mean-end</b> <u>DOUBLE</u>
              Mean PHRED quality for mismatch bases of last base in Illumina sequencing. Default: <u>30.0</u>.

       <b>--illumina-mismatch-quality-stddev-begin</b> <u>DOUBLE</u>
              Standard deviation of PHRED quality for mismatch bases  of  first  base  in  Illumina  sequencing.
              Default: <u>3.0</u>.

       <b>--illumina-mismatch-quality-stddev-end</b> <u>DOUBLE</u>
              Standard  deviation  of  PHRED  quality  for  mismatch  bases of last base in Illumina sequencing.
              Default: <u>15.0</u>.

       <b>--illumina-left-template-fastq</b> <u>INPUT_FILE</u>
              FASTQ file to use for a template for left-end reads.  Valid  filetypes  are:  <u>.sam[.*]</u>,  <u>.raw[.*]</u>,
              <u>.gbk[.*]</u>,  <u>.frn[.*]</u>,  <u>.fq[.*]</u>,  <u>.fna[.*]</u>,  <u>.ffn[.*]</u>,  <u>.fastq[.*]</u>,  <u>.fasta[.*]</u>,  <u>.faa[.*]</u>, <u>.fa[.*]</u>,
              <u>.embl[.*]</u>, and <u>.bam</u>, where * is any of the following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent
              (de)compression.

       <b>--illumina-right-template-fastq</b> <u>INPUT_FILE</u>
              FASTQ file to use for a template for right-end reads. Valid  filetypes  are:  <u>.sam[.*]</u>,  <u>.raw[.*]</u>,
              <u>.gbk[.*]</u>,  <u>.frn[.*]</u>,  <u>.fq[.*]</u>,  <u>.fna[.*]</u>,  <u>.ffn[.*]</u>,  <u>.fastq[.*]</u>,  <u>.fasta[.*]</u>,  <u>.faa[.*]</u>, <u>.fa[.*]</u>,
              <u>.embl[.*]</u>, and <u>.bam</u>, where * is any of the following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent
              (de)compression.

   <b>Sanger</b> <b>Sequencing</b> <b>Options:</b>
       <b>--sanger-read-length-model</b> <u>STRING</u>
              The model to use for sampling the Sanger read length. One of <u>normal</u> and <u>uniform</u>. Default: <u>normal</u>.

       <b>--sanger-read-length-min</b> <u>INTEGER</u>
              The minimal read length when the read length is sampled uniformly.  In  range  [0..inf].  Default:
              <u>400</u>.

       <b>--sanger-read-length-max</b> <u>INTEGER</u>
              The  maximal  read  length  when the read length is sampled uniformly. In range [0..inf]. Default:
              <u>600</u>.

       <b>--sanger-read-length-mean</b> <u>DOUBLE</u>
              The mean read length when the read length is sampled with normal distribution. In range  [0..inf].
              Default: <u>400</u>.

       <b>--sanger-read-length-error</b> <u>DOUBLE</u>
              The  read  length standard deviation when the read length is sampled uniformly. In range [0..inf].
              Default: <u>40</u>.

       <b>--sanger-prob-mismatch-scale</b> <u>DOUBLE</u>
              Scaling factor for Sanger mismatch probability. In range [0..inf]. Default: <u>1.0</u>.

       <b>--sanger-prob-mismatch-begin</b> <u>DOUBLE</u>
              Per-base mismatch probability of first base in Sanger sequencing. In  range  [0.0..1.0].  Default:
              <u>0.005</u>.

       <b>--sanger-prob-mismatch-end</b> <u>DOUBLE</u>
              Per-base  mismatch  probability  of  last base in Sanger sequencing. In range [0.0..1.0]. Default:
              <u>0.001</u>.

       <b>--sanger-prob-insertion-begin</b> <u>DOUBLE</u>
              Per-base insertion probability of first base in Sanger sequencing. In range  [0.0..1.0].  Default:
              <u>0.0025</u>.

       <b>--sanger-prob-insertion-end</b> <u>DOUBLE</u>
              Per-base  insertion  probability  of last base in Sanger sequencing. In range [0.0..1.0]. Default:
              <u>0.005</u>.

       <b>--sanger-prob-deletion-begin</b> <u>DOUBLE</u>
              Per-base deletion probability of first base in Sanger sequencing. In  range  [0.0..1.0].  Default:
              <u>0.0025</u>.

       <b>--sanger-prob-deletion-end</b> <u>DOUBLE</u>
              Per-base  deletion  probability  of  last base in Sanger sequencing. In range [0.0..1.0]. Default:
              <u>0.005</u>.

       <b>--sanger-quality-match-start-mean</b> <u>DOUBLE</u>
              Mean PHRED quality for non-mismatch bases of first base in Sanger sequencing. Default: <u>40.0</u>.

       <b>--sanger-quality-match-end-mean</b> <u>DOUBLE</u>
              Mean PHRED quality for non-mismatch bases of last base in Sanger sequencing. Default: <u>39.5</u>.

       <b>--sanger-quality-match-start-stddev</b> <u>DOUBLE</u>
              Mean PHRED quality for non-mismatch bases of first base in Sanger sequencing. Default: <u>0.1</u>.

       <b>--sanger-quality-match-end-stddev</b> <u>DOUBLE</u>
              Mean PHRED quality for non-mismatch bases of last base in Sanger sequencing. Default: <u>2</u>.

       <b>--sanger-quality-error-start-mean</b> <u>DOUBLE</u>
              Mean PHRED quality for erroneous bases of first base in Sanger sequencing. Default: <u>30</u>.

       <b>--sanger-quality-error-end-mean</b> <u>DOUBLE</u>
              Mean PHRED quality for erroneous bases of last base in Sanger sequencing. Default: <u>20</u>.

       <b>--sanger-quality-error-start-stddev</b> <u>DOUBLE</u>
              Mean PHRED quality for erroneous bases of first base in Sanger sequencing. Default: <u>2</u>.

       <b>--sanger-quality-error-end-stddev</b> <u>DOUBLE</u>
              Mean PHRED quality for erroneous bases of last base in Sanger sequencing. Default: <u>5</u>.

   <b>454</b> <b>Sequencing</b> <b>Options:</b>
       <b>--454-read-length-model</b> <u>STRING</u>
              The model to use for sampling the 454 read length. One of <u>normal</u> and <u>uniform</u>. Default: <u>normal</u>.

       <b>--454-read-length-min</b> <u>INTEGER</u>
              The minimal read length when the read length is sampled uniformly. In range [0..inf]. Default: <u>10</u>.

       <b>--454-read-length-max</b> <u>INTEGER</u>
              The maximal read length when the read length is sampled uniformly.  In  range  [0..inf].  Default:
              <u>600</u>.

       <b>--454-read-length-mean</b> <u>DOUBLE</u>
              The  mean read length when the read length is sampled with normal distribution. In range [0..inf].
              Default: <u>400</u>.

       <b>--454-read-length-stddev</b> <u>DOUBLE</u>
              The read length standard deviation when the read length is sampled with  normal  distribution.  In
              range [0..inf]. Default: <u>40</u>.

       <b>--454-no-sqrt-in-std-dev</b>
              For error model, if set then (sigma = k * r)) is used, otherwise (sigma = k * sqrt(r)).

       <b>--454-proportionality-factor</b> <u>DOUBLE</u>
              Proportionality  factor for calculating the standard deviation proportional to the read length. In
              range [0..inf]. Default: <u>0.15</u>.

       <b>--454-background-noise-mean</b> <u>DOUBLE</u>
              Mean of lognormal distribution to use for the noise. In range [0..inf]. Default: <u>0.23</u>.

       <b>--454-background-noise-stddev</b> <u>DOUBLE</u>
              Standard deviation of lognormal distribution to use for the noise.  In  range  [0..inf].  Default:
              <u>0.15</u>.

</pre><h4><b>SEQUENCING</b> <b>SIMULATION</b></h4><pre>
       Simulation  of  base  qualities  is  disabled  when  writing  out  FASTA files.  Simulation of paired-end
       sequencing is enabled when specifying two output files.

</pre><h4><b>READ</b> <b>ORIENTATION</b></h4><pre>
       You can use the <b>--mate-orientation</b> to set the relative orientation when doing paired-end sequencing.  The
       valid values are given in the following.

       FR     Reads are inward-facing, the same as Illumina paired-end reads: R1 --&gt; &lt;-- R2.

       RF     Reads are outward-facing, the same as Illumina mate-pair reads: R1 &lt;-- --&gt; R2.

       FF     Reads are on the same strand: R1 --&gt; --&gt; R2.

       FF2    Reads are on the same strand but the "right" reads are sequenced to the left of the "left"  reads,
              same as 454 paired: R2 --&gt; --&gt; R1.

mason_frag_sequencing 2.0.9 [tarball]                                                   <u><a href="../man1/MASON_FRAG_SEQUENCING.1.html">MASON_FRAG_SEQUENCING</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>