<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>gt-select - Select certain features (specified by the used options) from given GFF3 file(s).</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/genometools">genometools_1.6.5+ds-2.2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       gt-select - Select certain features (specified by the used options) from given GFF3 file(s).

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>gt</b> <b>select</b> [option ...] [GFF3_file ...]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>-retainids</b> [<u>yes|no</u>]
           when available, use the original IDs provided in the source file (memory consumption is proportional
           to the input file size(s)) (default: no)

       <b>-seqid</b> [<u>string</u>]
           select feature with the given sequence ID (all comments are selected). (default: undefined)

       <b>-source</b> [<u>string</u>]
           select feature with the given source (the source is column 2 in regular GFF3 lines) (default:
           undefined)

       <b>-contain</b> [<u>start</u> <u>end</u>]
           select all features which are contained in the given range (default: undefined)

       <b>-overlap</b> [<u>start</u> <u>end</u>]
           select all features which do overlap with the given range (default: undefined)

       <b>-strand</b> [<u>string</u>]
           select all top-level features(i.e., features without parents) whose strand equals the given one (must
           be one of <u>+-.?</u>) (default: undefined)

       <b>-targetstrand</b> [<u>string</u>]
           select all top-level features (i.e., features without parents) which have exactly one target
           attribute whose strand equals the given one (must be one of <u>+-.?</u>) (default: undefined)

       <b>-targetbest</b> [<u>yes|no</u>]
           if multiple top-level features (i.e., features without parents) with exactly one target attribute
           have the same target_id, keep only the feature with the best score. If -targetstrand is used at the
           same time, this option is applied after -targetstrand. Memory consumption is proportional to the
           input file size(s). (default: no)

       <b>-hascds</b> [<u>yes|no</u>]
           select all top-level features which do have a CDS child (default: no)

       <b>-maxgenelength</b> [<u>value</u>]
           select genes up to the given maximum length (default: undefined)

       <b>-maxgenenum</b> [<u>value</u>]
           select the first genes up to the given maximum number (default: undefined)

       <b>-mingenescore</b> [<u>value</u>]
           select genes with the given minimum score (default: undefined)

       <b>-maxgenescore</b> [<u>value</u>]
           select genes with the given maximum score (default: undefined)

       <b>-minaveragessp</b> [<u>value</u>]
           set the minimum average splice site probability (default: undefined)

       <b>-rule_files</b>
           specify Lua filter rule files to be used for selection (terminate list with <u>--</u>)

       <b>-rule_logic</b> [<u>...</u>]
           select how multiple Lua files should be combined choose from AND|OR (default: AND)

       <b>-dropped_file</b> [<u>filename</u>]
           save non-selected features to file (default: undefined)

       <b>-v</b> [<u>yes|no</u>]
           be verbose (default: no)

       <b>-o</b> [<u>filename</u>]
           redirect output to specified file (default: undefined)

       <b>-gzip</b> [<u>yes|no</u>]
           write gzip compressed output file (default: no)

       <b>-bzip2</b> [<u>yes|no</u>]
           write bzip2 compressed output file (default: no)

       <b>-force</b> [<u>yes|no</u>]
           force writing to output file (default: no)

       <b>-help</b>
           display help and exit

       <b>-version</b>
           display version information and exit

       File format for option <u>-rule_files</u>:

       The files supplied to option <u>-rule_files</u> define a function for filtering by user given criteria (see
       example below):

           function filter(gn)
             target = "exon"
             for curnode in gn:children() do
               if (curnode:get_type() == target) then
                 return false
               end
             end
             return true
           end

       The above function iterates over all children of <u>gn</u> and checks whether there is a node of type <u>exon</u>. If
       there is such a node the function returns <u>false</u>, indicating that the parent node <u>gn</u> will not be sorted
       out.

       NOTE: The function must be named <u>filter</u> and must return <u>false</u>, indicating that the node survived the
       filtering process.

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       Report bugs to https://github.com/genometools/genometools/issues.

GenomeTools 1.6.5                                  04/27/2024                                       <u><a href="../man1/GT-SELECT.1.html">GT-SELECT</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>