<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>mason_genome - Random Genome Simulation</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/seqan-apps">seqan-apps_2.4.0+dfsg-16_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       mason_genome - Random Genome Simulation

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>mason_genome</b> [<u>OPTIONS</u>] [<b>-l</b> <u>LEN</u>]+ <b>-o</b> <u>OUT.fa</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Simulate  a random genome to the output file.  For each <b>-l</b>/<b>--contig-length</b> entry, a contig with the given
       length will be simulated.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              Display the help message.

       <b>--version</b>
              Display version information.

       <b>-q</b>, <b>--quiet</b>
              Set verbosity to a minimum.

       <b>-v</b>, <b>--verbose</b>
              Enable verbose output.

       <b>-vv</b>, <b>--very-verbose</b>
              Enable very verbose output.

   <b>Simulation</b> <b>Configuration:</b>
       <b>-l</b>, <b>--contig-length</b> List of <u>INTEGER</u>'s
              Length of the contig to simulate. Give one  <b>-l</b>  value  for  each  contig  to  simulate.  In  range
              [1..inf].

       <b>-s</b>, <b>--seed</b> <u>INTEGER</u>
              The seed to use for the random number generator. Default: <u>0</u>.

   <b>Output</b> <b>Options:</b>
       <b>-o</b>, <b>--out-file</b> <u>OUTPUT_FILE</u>
              Output  file.  Valid  filetypes  are:  <u>.sam[.*]</u>,  <u>.raw[.*]</u>, <u>.frn[.*]</u>, <u>.fq[.*]</u>, <u>.fna[.*]</u>, <u>.ffn[.*]</u>,
              <u>.fastq[.*]</u>, <u>.fasta[.*]</u>, <u>.faa[.*]</u>, <u>.fa[.*]</u>, and <u>.bam</u>, where * is any of the  following  extensions:
              <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent (de)compression.

</pre><h4><b>EXAMPLES</b></h4><pre>
       <b>mason_genome</b> <b>-l</b> 1000 <b>-l</b> 4000 <b>-o</b> <u>genome.fa</u>
              Simulate a genome with two contigs of lengths 1000 and 4000 and write it to genome.fa.

mason_genome 2.0.9 [tarball]                                                                     <u><a href="../man1/MASON_GENOME.1.html">MASON_GENOME</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>