<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>bp_load_gff.pl - Load a Bio::DB::GFF database from GFF files.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/libbio-db-gff-perl">libbio-db-gff-perl_1.7.4-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       bp_load_gff.pl - Load a Bio::DB::GFF database from GFF files.

</pre><h4><b>SYNOPSIS</b></h4><pre>
         % bp_load_gff.pl -d testdb -u user -p pw
            --dsn 'dbi:mysql:database=dmel_r5_1;host=myhost;port=myport'
               dna1.fa dna2.fa features1.gff features2.gff ...

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This script loads a Bio::DB::GFF database with the features contained in a list of GFF files and/or FASTA
       sequence files.  You must use the exact variant of GFF described in Bio::DB::GFF.  Various command-line
       options allow you to control which database to load and whether to allow an existing database to be
       overwritten.

       This script uses the Bio::DB::GFF interface, and so works with all database adaptors currently supported
       by that module (MySQL, Oracle, PostgreSQL soon).  However, it is slow.  For faster loading, see the
       MySQL-specific bp_bulk_load_gff.pl and bp_fast_load_gff.pl scripts.

   <b>NOTES</b>
       If the filename is given as "-" then the input is taken from standard input. Compressed files (.gz, .Z,
       .bz2) are automatically uncompressed.

       FASTA format files are distinguished from GFF files by their filename extensions.  Files ending in .fa,
       .fasta, .fast, .seq, .dna and their uppercase variants are treated as FASTA files.  Everything else is
       treated as a GFF file.  If you wish to load -fasta files from STDIN, then use the -f command-line swith
       with an argument of '-', as in

           gunzip my_data.fa.gz | bp_fast_load_gff.pl -d test -f -

       On the first load of a database, you will see a number of "unknown table" errors.  This is normal.

       About maxfeature: the default value is 100,000,000 bases.  If you have features that are close to or
       greater that 100Mb in length, then the value of maxfeature should be increased to 1,000,000,000, or
       another power of 10.

</pre><h4><b>COMMAND-LINE</b> <b>OPTIONS</b></h4><pre>
       Command-line options can be abbreviated to single-letter options.  e.g. -d instead of --database.

          --dsn     &lt;dsn&gt;       Data source (default dbi:mysql:test)
          --adaptor &lt;adaptor&gt;   Schema adaptor (default dbi::mysqlopt)
          --user    &lt;user&gt;      Username for mysql authentication
          --pass    &lt;password&gt;  Password for mysql authentication
          --fasta   &lt;path&gt;      Fasta file or directory containing fasta files for the DNA
          --create              Force creation and initialization of database
          --maxfeature          Set the value of the maximum feature size (default 100 Mb; must be a power of 10)
          --group               A list of one or more tag names (comma or space separated)
                                 to be used for grouping in the 9th column.
          --upgrade             Upgrade existing database to current schema
          --gff3_munge          Activate GFF3 name munging (see Bio::DB::GFF)
          --quiet               No progress reports
          --summary             Generate summary statistics for drawing coverage histograms.
                                  This can be run on a previously loaded database or during
                                  the load.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Bio::DB::GFF, bulk_load_gff.pl, bp_load_gff.pl

</pre><h4><b>AUTHOR</b></h4><pre>
       Lincoln Stein, <a href="mailto:lstein@cshl.org">lstein@cshl.org</a>

       Copyright (c) 2002 Cold Spring Harbor Laboratory

       This library is free software; you can redistribute it and/or modify it under the same terms as Perl
       itself.  See DISCLAIMER.txt for disclaimers of warranty.

perl v5.30.0                                       2020-01-13                                    <u><a href="../man1p/BP_LOAD_GFF.1p.html">BP_LOAD_GFF</a></u>(1p)
</pre>
 </div>
</div></section>
</div>
</body>
</html>