<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>fa2dna - format fasta database for use with ANFO</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/anfo">anfo_0.98-9_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       fa2dna - format fasta database for use with ANFO

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>fa2dna</b> <b>[</b> <u>option</u> <b>|</b> <u>file</u> <b>...</b> <b>]</b>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>fa2dna</b>  reads  one or more files in fasta format and reformats them into a database suitable for <b><a href="../man1/anfo.1.html">anfo</a></b>(1).
       The input file(s) can contain more than one sequence, and each sequence can be broken  up  into  multiple
       contigs by a stretch of Ns.  All IUPAC ambiguity codes are fully supported.

</pre><h4><b>OPTIONS</b></h4><pre>
       -V, --version
              Print version number and exit.

       -o file, --output file
              Write output to <u>file</u>.<u>file</u> should end in <u>.dna</u>, because <b>anfo</b> and some downstream tools might stumble
              over a different extension.  Default is the genome name with the <u>.dna</u> extension.

       -m N, --maxn N
              Sets  the maximum number of <u>N</u>s to be interpreted as IUPAC ambiguity codes to <u>N</u>; any longer stretch
              is interpreted as separing contigs.  Default is 2.

       -g name, --genome name
              Sets the genome name to <u>name</u>.  This name is stored in the output file and will be used by <b>anfo</b>  to
              identify  the matched genome in its output.  The genome should be a prefix of the output file name
              to allow downstream tools to find it.  The name should be short, but unique, something like "hg18"
              or "pt2" for human or chimpanzee genomes would work fine.

       -d text, --description text
              Adds <u>text</u> as description to the metadata.  This is purely informational.

       -v, --verbose
              Causes a progress indicator to be printed.

</pre><h4><b>FILES</b></h4><pre>
       <u>/etc/popt</u>
              The system wide configuration file for <b><a href="../man3/popt.3.html">popt</a></b>(3).  <b>fa2dna</b> identifies itself as "fa2dna" to popt.

       <u><a href="file:~/.popt">~/.popt</a></u>
              Per user configuration file for <b><a href="../man3/popt.3.html">popt</a></b>(3).

</pre><h4><b>BUGS</b></h4><pre>
       None known.

</pre><h4><b>AUTHOR</b></h4><pre>
       Udo Stenzel &lt;<a href="mailto:udo_stenzel@eva.mpg.de">udo_stenzel@eva.mpg.de</a>&gt;

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/anfo.1.html">anfo</a></b>(1), <b><a href="../man1/fa2dna.1.html">fa2dna</a></b>(1) <b><a href="../man3/popt.3.html">popt</a></b>(3), <b><a href="../man5/fasta.5.html">fasta</a></b>(5)

Applications                                      OCTOBER 2009                                         <u><a href="../man1/FA2DNA.1.html">FA2DNA</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>