<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>fasts3,  fasts3_t  -  compare several short peptide sequences against a protein database using a modified</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/fasta3">fasta3_36.3.8i.14-Nov-2020-3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       fasts3,  fasts3_t  -  compare several short peptide sequences against a protein database using a modified
       fasta algorithm.

       tfasts3, tfasts3_t - compare short pepides against a translated DNA database.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>fasts3</b> and <b>tfasts3</b> are designed to compare set of  (presumably  non-contiguous)  peptides  to  a  protein
       (fasts3)  or  translated  DNA  (tfasts3)  database.   fasts3/tfasts3  are designed particularly for short
       peptide data from mass-spec analysis of protein digests.  Unlike the  traditional  <b>fasta3</b>  search,  which
       uses a protein or DNA sequence, <b>fasts3</b> and <b>tfasts3</b> work with a query sequence of the form:
            &gt;tests from mgstm1
            MLLE,
            MILGYW,
            MGADP,
            MLCYNP
(included with the distribution), the result is:
     testf    MILGYW----------MLLE------------MGDAP-----------
              ::::::          ::::            :::::
     GT8.7  MPMILGYWNVRGLTHPIRMLLEYTDSSYDEKRYTMGDAPDFDRSQWLNEK
                    10        20        30        40        50

     testf  --------------------------------------------------

     GT8.7  FKLGLDFPNLPYLIDGSHKITQSNAILRYLARKHHLDGETEEERIRADIV
                    60        70        80        90       100

                           20
     testf  ------------MLCYNP
                        ::::::
     GT8.7  ENQVMDTRMQLIMLCYNPDFEKQKPEFLKTIPEKMKLYSEFLGKRPWFAG
                   110       120       130       140       150

</pre><h4><b>Options</b></h4><pre>
       <b>fasts3</b> and <b>tfasts3</b> can accept a query sequence from the unix "stdin" data stream.   This  makes  it  much
       easier  to  use fasta3 and its relatives as part of a WWW page. To indicate that stdin is to be used, use
       "-" or "@" as the query sequence file name.

       -b #   number of best scores to show (must be &lt; -E cutoff)

       -d #   number of best alignments to show ( must be &lt; -E cutoff)

       -D     turn on debugging mode.  Enables checks on sequence alphabet that  cause  problems  with  tfastx3,
              tfasty3, tfasta3.

       -E #   Expectation  value  limit for displaying scores and alignments.  Expectation values for <b>fasts3</b> and
              <b>tfasts3</b> are not as accurate as those for the other <b>fasta3</b> programs.

       -H     turn off histogram display

       -i     compare against only the reverse complement of the library sequence.

       -L     report long sequence description in alignments

       -m 0,1,2,3,4,5,6,9,10
              alignment display options

       -N #   break long library sequences into blocks of # residues.  Useful for bacterial genomes, which  have
              only one sequence entry.  -N 2000 works well for well for bacterial genomes.

       -O file
              send output to file

       -q/-Q  quiet option; do not prompt for input

       -R file
              save all scores to statistics file

       -S #   offset substitution matrix values by  a constant #

       -s name
              specify  substitution  matrix.   BLOSUM50  is used by default; PAM250, PAM120, and BLOSUM62 can be
              specified by setting -s P120, P250, or BL62.  With this version, many more  scoring  matrices  are
              available,  including  BLOSUM80 (BL80), and MDM_10, MDM_20, MDM_40 (M10, M20, M40). Alternatively,
              BLASTP1.4 format scoring matrix files can be specified.

       -T #   (threaded, parallel only) number of threads or workers to use (set by  default  to  4  at  compile
              time).

       -t #   Translation    table    -    tfasts3    can    use    the    BLAST    tranlation    tables.    See
              <a href="http://www.ncbi.nih.gov/htbin-post/Taxonomy/wprintgc">http://www.ncbi.nih.gov/htbin-post/Taxonomy/wprintgc</a>?mode=c/.

       -w #   line width for similarity score, sequence alignment, output.

       -x "#,#"
              offsets query, library sequence for numbering alignments

       -z #   Specify statistical calculation. Default is -z 1, which uses regression against the length of  the
              library  sequence.  -z  0  disables  statistics.   -z 2 uses the ln() length correction. -z 3 uses
              Altschul and Gish's statistical estimates for specific protein BLOSUM  scoring  matrices  and  gap
              penalties. -z 4: an alternate regression method.

       -Z db_size
              Set the apparent database size used for expectation value calculations.

       -3     (TFASTS3 only) use only forward frame translations

</pre><h4><b>Environment</b> <b>variables:</b></h4><pre>
       FASTLIBS
              location of library choice file (-l FASTLIBS)

       SMATRIX
              default scoring matrix (-s SMATRIX)

       SRCH_URL
              the format string used to define the option to re-search the database.

       REF_URL
              the  format  string  used  to  define the option to lookup the library sequence in entrez, or some
              other database.

</pre><h4><b>AUTHOR</b></h4><pre>
       Bill Pearson
       <a href="mailto:wrp@virginia.EDU">wrp@virginia.EDU</a>

                                                      local                                    <u>FASTS/<a href="../man1/TFASTSv3.1.html">TFASTSv3</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>