<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>load2sqlitedb - load genome sequences and extrinsic evidence hints into a SQLite database</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/augustus">augustus_3.5.0+dfsg-4build7_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       load2sqlitedb - load genome sequences and extrinsic evidence hints into a SQLite database

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>load2sqlitedb</b> [OPTIONS] --species=<u>SPECIES</u> --dbaccess=<u>database.db</u> <u>inputfilename</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>load2sqlitedb</b> is a tool to load genome sequences and extrinsic evidence hints into a SQLite database.
       When storing genomes/hints of multiple organisms call this program repeatedly for each one.

</pre><h4><b>OPTIONS</b></h4><pre>
       <u>inputfilename</u> refers to a genome file in FASTA format or a hints file in GFF format.

   <b>Mandatory</b> <b>options</b>
       <b>-s</b>, <b>--species</b>=<u>SPECIES</u>
           SPECIES is a name for your species and the same identifier as is used in the treefile and alnfile
           parameters to augustus.

       <b>-d</b>, <b>--dbaccess</b>=<u>database.db</u>
           The file name of the SQLite database that will be opened or created if it does not already exist.

   <b>Additional</b> <b>options</b>
       <b>-c</b>, <b>--chunksize</b>=<u>size</u>
           This option is only relevant when loading a sequence file.
           The sequences in the input genome are split into chunks of this size so that subsequent retrievals of
           small sequence ranges do not require to read the complete - potentially much longer - chromosome. (⇐
           1000000). Default Value: 50000

       <b>-i</b>, <b>--noIdx</b>
           Use this flag to suppress the building of indices on the database tables.
           If you are going to load several genomes and/or hint files in a row, this option is recommended to
           speed up the loading. But make sure to build indices with --makeIdx after all genomes/hints are
           loaded. Otherwise, data retrieval operations can be very slow.

       <b>-m</b>, <b>--makeIdx</b>
           Use this flag to build the indices on the database tables after loading several genomes and/or hint
           files with --noIdx.
           Only call this once for all species, e.g. load2sqlitedb --makeIdx --dbaccess=database.db

       <b>-r</b>, <b>--clean</b>
           Makes a clean load deleting existing hints/genome for the species from the database.+ When called
           with a gff file, only the hints for the species are delete, but not the genome.
           When called with a fasta file, both hints and genome for the species are deleted.

       <b>-h</b>, <b>--help</b>
           Produce help message.

</pre><h4><b>EXAMPLE</b></h4><pre>
             load2sqlitedb --species=mouse --dbaccess=vertebrates.db mouse.fa
             load2sqlitedb --species=mouse --dbaccess=vertebrates.db mouse.hints.gff
             load2sqlitedb --species=human --dbaccess=vertebrates.db human.fa
             load2sqlitedb --species=human --dbaccess=vertebrates.db human.hints.gff

</pre><h4><b>AUTHORS</b></h4><pre>
       AUGUSTUS was written by M. Stanke, O. Keller, S. König, L. Gerischer and L. Romoth.

                                                                                                <u><a href="../man1/LOAD2SQLITEDB.1.html">LOAD2SQLITEDB</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>