<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>load2db - load genome sequences and extrinsic evidence hints into a MySQL database</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/augustus">augustus_3.5.0+dfsg-4build7_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       load2db - load genome sequences and extrinsic evidence hints into a MySQL database

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>load2db</b> [OPTIONS] --species=<u>SPECIES</u> --dbaccess=<u>dbname,host,user,passwd</u> <u>inputfilename</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>load2db</b> is a tool to load genome sequences and extrinsic evidence hints into a MySQL database.
       This database must be created before calling <b>load2db</b>.
       When storing genomes/hints of multiple organisms call this program repeatedly for each one.

</pre><h4><b>OPTIONS</b></h4><pre>
       <u>inputfilename</u> refers to a genome file in FASTA format or a hints file in GFF format.

   <b>Mandatory</b> <b>options</b>
       <b>-s</b>, <b>--species</b>=<u>SPECIES</u>
           SPECIES is a name for your species and the same identifier as is used in the treefile and alnfile
           parameters to augustus.

       <b>-d</b>, <b>--dbaccess</b>=<u>dbname,host,user,passwd</u>
           dbname,host,user,passwd are the name of the SQL database, the host name or IP, the database user and
           password

   <b>Additional</b> <b>options</b>
       <b>-c</b>, <b>--chunksize</b>=<u>size</u> (⇐ 1000000)
           This option is only relevant when loading a sequence file.
           The sequences in the input genome are split into chunks of this size so that subsequent retrievals of
           small sequence ranges do not require to read the complete - potentially much longer - chromosome.
           Default Value: 50000

       <b>-h</b>, <b>--help</b>
           Produce help message.

</pre><h4><b>EXAMPLE</b></h4><pre>
             create database before calling load2db:
              $ mysql -u root -p
              &gt; CREATE DATABASE vertebrates;
              &gt; CREATE USER 'myuser'@'%' IDENTIFIED BY 'mypasswd'; /* or whatever the password code is*/
              &gt; GRANT ALL ON vertebrates.* TO 'myuser'@'%' IDENTIFIED BY 'mypasswd';

             load2db --species=mouse --dbaccess=vertebrates,localhost,myuser,mypasswd mouse.fa
             load2db --species=mouse --dbaccess=vertebrates,localhost,myuser,mypasswd mouse.hints.gff
             load2db --species=human --dbaccess=vertebrates,localhost,myuser,mypasswd human.fa
             load2db --species=human --dbaccess=vertebrates,localhost,myuser,mypasswd human.hints.gff

</pre><h4><b>AUTHORS</b></h4><pre>
       AUGUSTUS was written by M. Stanke, O. Keller, S. König, L. Gerischer and L. Romoth.

                                                                                                      <u><a href="../man1/LOAD2DB.1.html">LOAD2DB</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>