<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>samtools-bedcov - reports coverage over regions in a supplied BED file</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/samtools">samtools_1.21-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       samtools-bedcov - reports coverage over regions in a supplied BED file

</pre><h4><b>SYNOPSIS</b></h4><pre>
       samtools bedcov [<u>options</u>] <u>region.bed</u> <u>in1.sam</u>|<u>in1.bam</u>|<u>in1.cram</u>[...]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Reports  the  total  read  base  count  (i.e.  the  sum  of per base read depths) for each genomic region
       specified in the supplied BED file. The regions are output as  they  appear  in  the  BED  file  and  are
       0-based.   Columns  1-3 are chrom/start/end as per the input BED file, followed by N columns of coverages
       (for N input BAMs), then (if given -d), N columns of bases-at-depth-X, then (if given -c)  N  columns  of
       read counts.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-Q,</b> <b>--min-MQ</b> <u>INT</u>
              Only count reads with mapping quality greater than or equal to <u>INT</u>

       <b>-g</b> <u>FLAGS</u>
              By  default, reads that have any of the flags UNMAP, SECONDARY, QCFAIL, or DUP set are skipped. To
              include these reads back in the analysis, use this option together with the desired flag  or  flag
              combination.  <u>FLAGS</u> can be specified in hex by beginning with `0x' (i.e. /^0x[0-9A-F]+/), in octal
              by beginning with `0' (i.e. /^0[0-7]+/), as a decimal number not beginning with '0' or as a comma-
              separated list of flag names. [0]

              For a list of flag names see <u><a href="../man1/samtools-flags.1.html">samtools-flags</a></u>(1).

       <b>-G</b> <u>FLAGS</u>
              Discard any read that has any of the flags specified by <u>FLAGS</u> set.  FLAGS are specified as for the
              <b>-g</b> option. [UNMAP,SECONDARY,QCFAIL,DUP]

       <b>-j</b>     Do not include deletions (D) and ref skips (N) in bedcov computation.

       <b>-d</b> <u>INT</u> Print an additional column, for each file, containing the number of bases having a depth above and
              including  the  given threshold. If the option is not used, the extra column is not displayed. The
              option value must be an integer &gt;= 0.

       <b>--max-depth</b> <u>INT</u>
              Specifies the maximum depth used for the mpileup algorithm.  If <b>-d</b> is used and is larger then this
              value will be used instead.  Defaults to 2 billion, but smaller values may be used when we do  not
              require  an  exact  count  in  excessively  deep regions and are interested in maximising speed of
              results.

       <b>-c</b>     Print an additional column with the read count for this region.  This will be +1  for  every  read
              covering  the region, not just starting within in.  The whole read filtering options <b>-Q</b>, <b>-g</b> and <b>-G</b>
              options will also have an affect on this count, but <b>-d</b> will not.

       <b>-X</b>     If this option is set, it will allows user to specify customized index  file  location(s)  if  the
              data  folder does not contain any index file. Example usage: samtools bedcov [options] -X &lt;in.bed&gt;
              &lt;/data_folder/in1.bam&gt; [...] &lt;/index_folder/index1.bai&gt; [...]

       <b>-H</b>     print a comment/header describing columns

</pre><h4><b>AUTHOR</b></h4><pre>
       Written by Heng Li from the Sanger Institute.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/samtools.1.html">samtools</a></u>(1)

       Samtools website: &lt;<a href="http://www.htslib.org/">http://www.htslib.org/</a>&gt;

samtools-1.21                                   12 September 2024                             <u><a href="../man1/samtools-bedcov.1.html">samtools-bedcov</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>