<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Run::Alignment::Gmap - Wrapper for running gmap.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-perl-run-perl">libbio-perl-run-perl_1.7.3-8_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Run::Alignment::Gmap - Wrapper for running gmap.

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::Tools::Run::Alignment::Gmap;
         use Bio::SeqIO;

         my $sio = Bio::SeqIO-&gt;new(-file=&gt;$filename ,-format=&gt;'fasta');
         my @seq;
         while(my $seq = $sio-&gt;next_seq()){
           push @seq,$seq;
         }
         my $mapper =Bio::Tools::Run::Gmap-&gt;new();
         my $result = $mapper-&gt;run(\@seq);

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bioperl-run wrapper around gmap.  See &lt;<a href="http://www.gene.com/share/gmap/">http://www.gene.com/share/gmap/</a>&gt; for information about gmap.

       It requires a reference to an array of bioperl SeqI objects and returns a reference to a filehandle from
       which the gmap output can be read.

       One can explicitly set the name of the genome database (defaults to NHGD_R36) using the '<b>genome_db()</b>'
       method.  One can also explicitly set the flags that are passed to gmap (defaults to '-f 9 -5 -e') using
       the '<b>flags()</b>' method.

       The name of the gmap executable can be overridden using the <b>program_name()</b> method and the directory in
       which to find that executable can be overridden using the <b>program_dir()</b> method.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         <a href="http://redmine.open-bio.org/projects/bioperl/">http://redmine.open-bio.org/projects/bioperl/</a>

</pre><h4><b>AUTHOR</b> <b>-</b> <b>George</b> <b>Hartzell</b></h4><pre>
       Email <a href="mailto:hartzell@alerce.com">hartzell@alerce.com</a>

       Describe contact details here

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
       Additional contributors names and emails here

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : my $obj = new Bio::Tools::Run::Alignment::Gmap();
        Function: Builds a new Bio::Tools::Run::Alignment::Gmap object
        Returns : an instance of Bio::Tools::Run::Alignment::Gmap
        Args    :

   <b>version</b>
        Title   : version
        Usage   : print "gmap version: " . $mapper-&gt;version() . "\n";
        Function: retrieves and returns the version of the gmap package.
        Example :
        Returns : scalar string containing the version number.  Probably looks
                  like YYYY-MM-DD.
        Args    : none.

   <b>program_name</b>
        Title   : program_name
        Usage   : $mapper-&gt;program_name('gmap-dev');
                  my $pname = $mapper-&gt;program_name();
        Function: sets/gets the name of the program to run.
        Returns : string representing the name of the executable.
        Args    : [optional] string representing the name of the executable
                  to set.

   <b>program_dir</b>
        Title   : program_dir
        Usage   : $mapper-&gt;program_dir('/usr/local/sandbox/gmap/bin');
                  my $pdir = $mapper-&gt;program_dir();
        Function: sets/gets the directory path in which
                  to find the gmap executable.
        Returns : string representing the path to the directory.
        Args    : [optional] string representing the directory path to set.

   <b>input_file</b>
        Title   : input_file
        Usage   : $mapper-&gt;input_file('/tmp/moose.fasta');
                  my $filename = $mapper-&gt;input_file();
        Function: sets/gets the name of a file containing sequences
                  to be mapped.
        Returns : string containing the name of the query sequence.
        Args    : [optional] string representing the directory path to set.

   <b>genome_db</b>
        Title   : genome_db
        Usage   : $mapper-&gt;genome_db('NHGD_R36');
                  my $genome_db = $mapper-&gt;genome_db();
        Function: sets/gets the name of the genome database, this will be
                  passed to gmap using its '-d' flag.
        Returns : name of the genome database.
        Args    : [optional] string representing the genome db to set.

   <b>flags</b>
        Title   : flags
        Usage   : $mapper-&gt;flags('-A -e -5');
                  my $flags = $mapper-&gt;flags();
        Function: sets/gets the flags that will be passed to gmap.
        Returns : the current value of the flags that will be passed to gmap.
        Args    : [optional] the flags to set.

   <b>run</b>
        Title   : run
        Usage   : $mapper-&gt;run()
        Function: runs gmap
        Example :
        Returns : a file handle, opened for reading, for gmap's output.
        Args    : An array of references query sequences (as Bio::Seq objects)

   <b>_build_fasta_input_file</b>
        Title   : _build_fasta_input_file
        Usage   : my $seq_file = $self-&gt;_build_fasta_input_file(@_);
        Function:
        Example :
        Returns : The name of the temporary file that contains the sequence.
        Args    : A reference to an array of Bio::Seq objects.

perl v5.34.0                                       2022-02-10              <u>Bio::Tools::Run::Alignment::<a href="../man3pm/Gmap.3pm.html">Gmap</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>