<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>augustus - a gene prediction tool</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/augustus">augustus_3.5.0+dfsg-4build7_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       augustus - a gene prediction tool

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>augustus</b> [parameters] --species=SPECIES queryfilename

</pre><h4><b>OPTIONS</b></h4><pre>
       The mandatory option 'queryfilename' specifies the filename (including relative path) to the file
       containing the query sequence(s) in FASTA format.

       SPECIES is an identifier for the species. Use <b>--species=help</b> to see a list.

   <b>Further</b> <b>parameters:</b>
       <b>--strand=both</b>, <b>--strand=forward</b> or <b>--strand=backward</b>
           define the strand to search on

       <b>--genemodel=genemodel</b>
           where genemodel is one of:

           •   partial      - allow prediction of incomplete genes at the sequence boundaries (default)

           •   intronless   - only predict single-exon genes like in prokaryotes and some eukaryotes

           •   complete     - only predict complete genes

           •   atleastone   - predict at least one complete gene

           •   exactlyone   - predict exactly one complete gene

       <b>--singlestrand=true</b>
           predict genes independently on each strand, allow overlapping genes on opposite strands. This option
           is turned off by default.

       <b>--hintsfile=hintsfilename</b>
           When this option is used the prediction considering hints (extrinsic information) is turned on.
           hintsfilename contains the hints in gff format.

       <b>--AUGUSTUS_CONFIG_PATH=path</b>
           path to config directory (if not specified as environment variable)

       <b>--alternatives-from-evidence=true/false</b>
           report alternative transcripts when they are suggested by hints

       <b>--alternatives-from-sampling=true/false</b>
           report alternative transcripts generated through probabilistic sampling

       <b>--maxtracks=n</b>
           For a description of these parameters see section "SAMPLING: ALTERNATIVE TRANSCRIPTS AND POSTERIOR
           PROBABILITIES" of docs/RUNNING-AUGUSTUS.md.

       <b>--proteinprofile=filename</b>
           When this option is used the prediction will consider the protein profile provided as parameter. The
           protein profile extension is described in section "AUGUSTUS-PPX" of docs/RUNNING-AUGUSTUS.md.

       <b>--progress=true</b>
           show a progressmeter

       <b>--gff3=on/off</b>
           output in gff3 format

       <b>--predictionStart=A</b>, <b>--predictionEnd=B</b>
           A and B define the range of the sequence for which predictions should be found.

       <b>--UTR=on/off</b>
           predict the untranslated regions in addition to the coding sequence. This currently works only for a
           subset of species.

       <b>--noInFrameStop=true/false</b>
           Do not report transcripts with in-frame stop codons. Otherwise, intron-spanning stop codons could
           occur. Default: false

       <b>--noprediction=true/false</b>
           If true and input is in genbank format, no prediction is made. Useful for getting the annotated
           protein sequences.

       <b>--uniqueGeneId=true/false</b>
           If true, output gene identifyers like this: seqname.gN

       <b>--paramlist</b>
           Show full list of supported parameters.

</pre><h4><b>AUTHORS</b></h4><pre>
       AUGUSTUS was written by M. Stanke, O. Keller, S. König, L. Gerischer and L. Romoth.

</pre><h4><b>ADDITIONAL</b> <b>DOCUMENTATION</b></h4><pre>
       An exhaustive documentation can be found in the file /usr/share/doc/augustus/README.md.gz.

                                                                                                     <u><a href="../man1/AUGUSTUS.1.html">AUGUSTUS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>