<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>pique-input - software pipeline for performing genome wide association studies</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/pique">pique_1.0-7_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       pique-input - software pipeline for performing genome wide association studies

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>pique</b> <u>-i</u> <u>in_prefix</u> <u>-o</u> <u>out_prefix</u> [-d] [-v] [-f in_format] [-k kinship] [-p pheno_file] [-c covar_file [-e
       n_pc]]

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Mandatory</b> <b>parameters:</b>

       <b>-i</b> in_prefix: input prefix for the input files

       <b>-o</b> out_prefix: output prefix for the output files

   <b>Optional</b> <b>parameters:</b>

       <b>-d</b> debug: saves intermediate files for debugging

       <b>-v</b> verbose: display all output from plink and smartpca

       <b>-f</b> in_format: input file format (default = ped)

              ped = requires in_prefix.ped &amp; in_prefix.map files tped = requires in_prefix.tped &amp; in_prefix.tfam
              files vcf = requires .vcf filename

       <b>-p</b> pheno_file: phenotype file name

              Defaults to [in_prefix].pheno

       <b>-g</b> group[,group...]: genotype sub-groups to be analysed

       <b>-n</b> number: minimum number of genotypes in sub-group (default = 20)

       <b>-k</b> kinship: method to calculate kinship matrix

              Either 'IBS' (default) or 'BN' methods

       <b>-c</b> covar_file: generate covariate file [out_prefix].covar

              from SNP information in .ped file using smartpca (EIGENSOFT)

       <b>-e</b> n_pc: number of eigenvectors to keep from smartpca

       <b>-m</b> MAF: Minor Allele Frequency

       <b>-x</b> missing: maximum per-SNP missing

       Further details and examples are given in the documentation

</pre><h4><b>AUTHOR</b></h4><pre>
       Tony Travis &lt;<a href="mailto:tony.travis@abdn.ac.uk">tony.travis@abdn.ac.uk</a>&gt; and Alex Douglas &lt;<a href="mailto:a.douglas@abdn.ac.uk">a.douglas@abdn.ac.uk</a>&gt;

       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

pique-input 1.0                                     July 2020                                     <u><a href="../man1/PIQUE-INPUT.1.html">PIQUE-INPUT</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>