<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::DB::GFF::Aggregator::transcript -- Transcript aggregator</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-db-gff-perl">libbio-db-gff-perl_1.7.4-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::DB::GFF::Aggregator::transcript -- Transcript aggregator

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::DB::GFF;

         # Open the sequence database
         my $db      = Bio::DB::GFF-&gt;new( -adaptor =&gt; 'dbi:mysql',
                                          -dsn     =&gt; 'dbi:mysql:elegans42',
                                          -aggregator =&gt; ['transcript','clone'],
                                        );

        -------------------------------------------------
        Aggregator method: transcript
        Main method:       transcript
        Sub methods:       exon CDS 5'UTR 3'UTR TSS PolyA
        -------------------------------------------------

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::DB::GFF::Aggregator::transcript is one of the default aggregators, and was written to be compatible
       with the C elegans GFF files.  It aggregates raw ""exon", "CDS", "5'UTR", "3'UTR", "polyA" and "TSS"
       features into "transcript" features.  For compatibility with the idiosyncrasies of the Sanger GFF format,
       it expects that the full range of the transcript is contained in a main feature of type "Transcript"
       (notice the capital "T").

       Internally this module is very simple.  To override it with one that recognizes a main feature named
       "gene", simply follow this template:

        my $db = Bio::DB::GFF-&gt;new(...etc...)
        my $aggregator = Bio::DB::GFF::Aggregator-&gt;new(-method =&gt; 'transcript',
                                                       -main_method =&gt; 'gene',
                                                       -sub_parts =&gt; ['exon','CDS']);
        $db-&gt;add_aggregator($aggregator);

   <b>method</b>
        Title   : method
        Usage   : $aggregator-&gt;method
        Function: return the method for the composite object
        Returns : the string "transcript"
        Args    : none
        Status  : Public

   <b>part_names</b>
        Title   : part_names
        Usage   : $aggregator-&gt;part_names
        Function: return the methods for the sub-parts
        Returns : the list "intron", "exon" and "CDS"
        Args    : none
        Status  : Public

   <b>main_name</b>
        Title   : main_name
        Usage   : $aggregator-&gt;main_name
        Function: return the method for the main component
        Returns : the string "transcript"
        Args    : none
        Status  : Public

</pre><h4><b>BUGS</b></h4><pre>
       None reported.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Bio::DB::GFF, Bio::DB::GFF::Aggregator

</pre><h4><b>AUTHOR</b></h4><pre>
       Lincoln Stein &lt;<a href="mailto:lstein@cshl.org">lstein@cshl.org</a>&gt;.

       Copyright (c) 2001 Cold Spring Harbor Laboratory.

       This library is free software; you can redistribute it and/or modify it under the same terms as Perl
       itself.

perl v5.30.0                                       2020-01-13             <u>Bio::DB::GFF::A...tor::<a href="../man3pm/transcript.3pm.html">transcript</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>