<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::SeqFeature::Gene::Transcript - A feature representing a transcript</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::SeqFeature::Gene::Transcript - A feature representing a transcript

</pre><h4><b>SYNOPSIS</b></h4><pre>
         # See documentation of methods.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       A feature representing a transcript.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track
        the bugs and their resolution.  Bug reports can be submitted via the
        web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Hilmar</b> <b>Lapp</b></h4><pre>
       Email <a href="mailto:hlapp@gmx.net">hlapp@gmx.net</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>promoters</b>
        Title   : promoters()
        Usage   : @proms = $transcript-&gt;promoters();
        Function: Get the promoter features/sites of this transcript.

                  Note that OO-modeling of regulatory elements is not stable yet.
                  This means that this method might change or even disappear in a
                  future release. Be aware of this if you use it.

        Returns : An array of Bio::SeqFeatureI implementing objects representing the
                  promoter regions or sites.
        Args    :

   <b>add_promoter</b>
        Title   : add_promoter()
        Usage   : $transcript-&gt;add_promoter($feature);
        Function: Add a promoter feature/site to this transcript.

                  Note that OO-modeling of regulatory elements is not stable yet.
                  This means that this method might change or even disappear in a
                  future release. Be aware of this if you use it.

        Returns :
        Args    : A Bio::SeqFeatureI implementing object.

   <b>flush_promoters</b>
        Title   : flush_promoters()
        Usage   : $transcript-&gt;flush_promoters();
        Function: Remove all promoter features/sites from this transcript.

                  Note that OO-modeling of regulatory elements is not stable yet.
                  This means that this method might change or even disappear in a
                  future release. Be aware of this if you use it.

        Returns : the removed features as a list
        Args    : none

   <b>exons</b>
        Title   : exons()
        Usage   : @exons = $gene-&gt;exons();
                  ($inital_exon) = $gene-&gt;exons('Initial');
        Function: Get all exon features or all exons of specified type of this
                  transcript.

                  Exon type is treated as a case-insensitive regular expression and
                  is optional. For consistency, use only the following types:
                  initial, internal, terminal.

        Returns : An array of Bio::SeqFeature::Gene::ExonI implementing objects.
        Args    : An optional string specifying the primary_tag of the feature.

   <b>exons_ordered</b>
        Title   : exons_ordered
        Usage   : @exons = $gene-&gt;exons_ordered();
                  @exons = $gene-&gt;exons_ordered("Internal");
        Function: Get an ordered list of all exon features or all exons of specified
                  type of this transcript.

                  Exon type is treated as a case-insensitive regular expression and
                  is optional. For consistency, use only the following types:

        Returns : An array of Bio::SeqFeature::Gene::ExonI implementing objects.
        Args    : An optional string specifying the primary_tag of the feature.

   <b>add_exon</b>
        Title   : add_exon()
        Usage   : $transcript-&gt;add_exon($exon,'initial');
        Function: Add a exon feature to this transcript.

                  The second argument denotes the type of exon. Mixing exons with and
                  without a type is likely to cause trouble in exons(). Either
                  leave out the type for all exons or for none.

                  Presently, the following types are known: initial, internal,
                  terminal, utr, utr5prime, and utr3prime (all case-insensitive).
                  UTR should better be added through utrs()/add_utr().

                  If you wish to use other or additional types, you will almost
                  certainly have to call exon_type_sortorder() in order to replace
                  the default sort order, or mrna(), cds(), protein(), and exons()
                  may yield unexpected results.

        Returns :
        Args    : A Bio::SeqFeature::Gene::ExonI implementing object.
                  A string indicating the type of the exon (optional).

   <b>flush_exons</b>
        Title   : flush_exons()
        Usage   : $transcript-&gt;flush_exons();
                  $transcript-&gt;flush_exons('terminal');
        Function: Remove all or a certain type of exon features from this transcript.

                  See add_exon() for documentation about types.

                  Calling without a type will not flush UTRs. Call flush_utrs() for
                  this purpose.
        Returns : the deleted features as a list
        Args    : A string indicating the type of the exon (optional).

   <b>introns</b>
        Title   : introns()
        Usage   : @introns = $gene-&gt;introns();
        Function: Get all intron features this gene structure.

                  Note that this implementation generates these features
                  on-the-fly, that is, it simply treats all regions between
                  exons as introns, assuming that exons do not overlap. A
                  consequence is that a consistent correspondence between the
                  elements in the returned array and the array that exons()
                  returns will exist only if the exons are properly sorted
                  within their types (forward for plus- strand and reverse
                  for minus-strand transcripts). To ensure correctness the
                  elements in the array returned will always be sorted.

        Returns : An array of Bio::SeqFeature::Gene::Intron objects representing
                  the intron regions.
        Args    :

   <b>poly_A_site</b>
        Title   : poly_A_site()
        Usage   : $polyAsite = $transcript-&gt;poly_A_site();
        Function: Get/set the poly-adenylation feature/site of this transcript.
        Returns : A Bio::SeqFeatureI implementing object representing the
                  poly-adenylation region.
        Args    : A Bio::SeqFeatureI implementing object on set, or FALSE to flush
                  a previously set object.

   <b>utrs</b>
        Title   : utrs()
        Usage   : @utr_sites = $transcript-&gt;utrs('utr3prime');
                  @utr_sites = $transcript-&gt;utrs('utr5prime');
                  @utr_sites = $transcript-&gt;utrs();
        Function: Get the features representing untranslated regions (UTR) of this
                  transcript.

                  You may provide an argument specifying the type of UTR. Currently
                  the following types are recognized: utr5prime utr3prime for UTR on the
                  5' and 3' end of the CDS, respectively.

        Returns : An array of Bio::SeqFeature::Gene::UTR objects
                  representing the UTR regions or sites.
        Args    : Optionally, either utr3prime, or utr5prime for the the type of UTR
                  feature.

   <b>add_utr</b>
        Title   : add_utr()
        Usage   : $transcript-&gt;add_utr($utrobj, 'utr3prime');
                  $transcript-&gt;add_utr($utrobj);
        Function: Add a UTR feature/site to this transcript.

                  The second parameter is optional and denotes the type of the UTR
                  feature. Presently recognized types include 'utr5prime' and 'utr3prime'
                  for UTR on the 5' and 3' end of a gene, respectively.

                  Calling this method is the same as calling
                  add_exon($utrobj, 'utr'.$type). In this sense a UTR object is a
                  special exon object, which is transcribed, not spliced out, but
                  not translated.

                  Note that the object supplied should return FALSE for is_coding().
                  Otherwise cds() and friends will become confused.

        Returns :
        Args    : A Bio::SeqFeature::Gene::UTR implementing object.

   <b>flush_utrs</b>
        Title   : flush_utrs()
        Usage   : $transcript-&gt;flush_utrs();
                  $transcript-&gt;flush_utrs('utr3prime');
        Function: Remove all or a specific type of UTR features/sites from this
                  transcript.

                  Cf. add_utr() for documentation about recognized types.
        Returns : a list of the removed features
        Args    : Optionally a string denoting the type of UTR feature.

   <b>sub_SeqFeature</b>
        Title   : sub_SeqFeature
        Usage   : @feats = $transcript-&gt;sub_SeqFeature();
        Function: Returns an array of all subfeatures.

                  This method is defined in Bio::SeqFeatureI. We override this here
                  to include the exon etc features.

        Returns : An array Bio::SeqFeatureI implementing objects.
        Args    : none

   <b>flush_sub_SeqFeature</b>
        Title   : flush_sub_SeqFeature
        Usage   : $transcript-&gt;flush_sub_SeqFeature();
                  $transcript-&gt;<a href="../man1/flush_sub_SeqFeature.1.html">flush_sub_SeqFeature</a>(1);
        Function: Removes all subfeatures.

                  This method is overridden from Bio::SeqFeature::Generic to flush
                  all additional subfeatures like exons, promoters, etc., which is
                  almost certainly not what you want. To remove only features added
                  through $transcript-&gt;add_sub_SeqFeature($feature) pass any
                  argument evaluating to TRUE.

        Example :
        Returns : none
        Args    : Optionally, an argument evaluating to TRUE will suppress flushing
                  of all transcript-specific subfeatures (exons etc.).

   <b>cds</b>
        Title   : cds
        Usage   : $seq = $transcript-&gt;cds();
        Function: Returns the CDS (coding sequence) as defined by the exons
                  of this transcript and the attached sequence.

                  If no sequence is attached this method will return false.

                  Note that the implementation provided here returns a
                  concatenation of all coding exons, thereby assuming that
                  exons do not overlap.

                  Note also that you cannot set the CDS via this method. Set
                  a single CDS feature as a single exon, or derive your own
                  class if you want to store a predicted CDS.

        Example :
        Returns : A Bio::PrimarySeqI implementing object.
        Args    :

   <b>protein</b>
        Title   : protein()
        Usage   : $protein = $transcript-&gt;protein();
        Function: Get the protein encoded by the transcript as a sequence object.

                  The implementation provided here simply calls translate() on the
                  object returned by cds().

        Returns : A Bio::PrimarySeqI implementing object.
        Args    :

   <b>mrna</b>
        Title   : mrna()
        Usage   : $mrna = $transcript-&gt;mrna();
        Function: Get the mRNA of the transcript as a sequence object.

                  The difference to cds() is that the sequence object returned by
                  this methods will also include UTR and the poly-adenylation site,
                  but not promoter sequence (TBD).

                  HL: do we really need this method?

        Returns : A Bio::PrimarySeqI implementing object.
        Args    :

   <b>features</b>
        Title   : features
        Usage   : my @features=$transcript-&gt;features;
        Function: returns all the features associated with this transcript
        Returns : a list of SeqFeatureI implementing objects
        Args    : none

   <b>features_ordered</b>
        Title   : features_ordered
        Usage   : my @features=$transcript-&gt;features_ordered;
        Function: returns all the features associated with this transcript,
                  in order by feature start, according to strand
        Returns : a list of SeqFeatureI implementing objects
        Args    : none

perl v5.32.1                                       2021-08-15             <u>Bio::SeqFeature::Gene::<a href="../man3pm/Transcript.3pm.html">Transcript</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>