<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::SeqFeature::Gene::TranscriptI - Interface for a feature representing a</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::SeqFeature::Gene::TranscriptI - Interface for a feature representing a
                transcript of exons, promoter(s), UTR, and a poly-adenylation site.

</pre><h4><b>SYNOPSIS</b></h4><pre>
         #documentation needed

</pre><h4><b>DESCRIPTION</b></h4><pre>
       A feature representing a transcript.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably
        to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Hilmar</b> <b>Lapp</b></h4><pre>
       Email <a href="mailto:hlapp@gmx.net">hlapp@gmx.net</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>promoters</b>
        Title   : promoters()
        Usage   : @proms = $transcript-&gt;promoters();
        Function: Get the promoter features of this transcript.

                  Note that OO-modeling of regulatory elements is not stable yet.
                  This means that this method might change or even disappear in a
                  future release. Be aware of this if you use it.

        Returns : An array of Bio::SeqFeatureI implementing objects representing the
                  promoter regions or sites.
        Args    :

   <b>exons</b>
        Title   : exons()
        Usage   : @exons = $transcript-&gt;exons();
                  @inital = $transcript-&gt;exons('Initial');
        Function: Get the individual exons this transcript comprises of, or all exons
                  of a specified type.

                  Refer to the documentation of the class that produced this
                  transcript object for information about the possible types.

                  See Bio::SeqFeature::Gene::ExonI for properties of the
                  returned objects.

        Returns : An array of Bio::SeqFeature::Gene::ExonI implementing objects
        Args    : An optional string specifying the type of the exon.

   <b>introns</b>
        Title   : introns()
        Usage   : @introns = $transcript-&gt;introns();
        Function: Get all introns this transcript comprises of.
        Returns : An array of Bio::SeqFeatureI implementing objects representing the
                  introns.
        Args    :

   <b>poly_A_site</b>
        Title   : poly_A_site()
        Usage   : $polyAsite = $transcript-&gt;poly_A_site();
        Function: Get the poly-adenylation site of this transcript.
        Returns : A Bio::SeqFeatureI implementing object.
        Args    :

   <b>utrs</b>
        Title   : utrs()
        Usage   : @utr_sites = $transcript-&gt;utrs();
        Function: Get the UTR regions this transcript comprises of.

                  See Bio::SeqFeature::Gene::ExonI for properties of the
                  returned objects.

        Returns : An array of Bio::SeqFeature::Gene::ExonI implementing objects
        Args    :

   <b>mrna</b>
        Title   : mrna()
        Usage   : $mrna = $transcript-&gt;mrna();
        Function: Get the mRNA of the transcript as a sequence object.

        Returns : A Bio::PrimarySeqI implementing object.
        Args    :

   <b>cds</b>
        Title   : cds()
        Usage   : $cds = $transcript-&gt;cds();
        Function: Get the CDS (coding sequence) of the transcript as a sequence
                  object.

        Returns : A Bio::PrimarySeqI implementing object.
        Args    :

   <b>protein</b>
        Title   : protein()
        Usage   : $protein = $transcript-&gt;protein();
        Function: Get the protein encoded by the transcript as a sequence object.

        Returns : A Bio::PrimarySeqI implementing object.
        Args    :

   <b>parent</b>
        Title   : parent
        Usage   : $obj-&gt;parent($newval)
        Function: get the parent gene of the transcript
        Returns : value of parent - a Bio::SeqFeature::Gene::GeneStructureI-compliant object
        Args    : a Bio::SeqFeature::Gene::GeneStructureI-compliant object (optional)

perl v5.32.1                                       2021-08-15             <u>Bio::SeqFeature...ne::<a href="../man3pm/TranscriptI.3pm.html">TranscriptI</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>