<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>stellar - the SwifT Exact LocaL AligneR</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/seqan-apps">seqan-apps_2.4.0+dfsg-16_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       stellar - the SwifT Exact LocaL AligneR

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>stellar</b> [<u>OPTIONS</u>] &lt;<u>FASTA</u> <u>FILE</u> <u>1</u>&gt; &lt;<u>FASTA</u> <u>FILE</u> <u>2</u>&gt;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       STELLAR  implements  the  SWIFT filter algorithm (Rasmussen et al., 2006) and a verification step for the
       SWIFT hits that applies local alignment, gapped X-drop extension, and extraction of the longest  epsilon-
       match.

       Input to STELLAR are two files, each containing one or more sequences in FASTA format. Each sequence from
       file  1  will be compared to each sequence in file 2. The sequences from file 1 are used as database, the
       sequences from file 2 as queries.

       (c) 2010-2012 by Birte Kehr

</pre><h4><b>REQUIRED</b> <b>ARGUMENTS</b></h4><pre>
       <b>FASTA_FILE_1</b> <u>INPUT_FILE</u>
               Valid filetypes are: <u>.fasta</u> and <u>.fa</u>.

       <b>FASTA_FILE_2</b> <u>INPUT_FILE</u>
               Valid filetypes are: <u>.fasta</u> and <u>.fa</u>.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              Display the help message.

       <b>--version</b>
              Display version information.

   <b>Main</b> <b>Options:</b>
       <b>-e</b>, <b>--epsilon</b> <u>DOUBLE</u>
              Maximal error rate (max 0.25). In range [0.0000001..0.25]. Default: <u>0.05</u>.

       <b>-l</b>, <b>--minLength</b> <u>INTEGER</u>
              Minimal length of epsilon-matches. In range [0..inf]. Default: <u>100</u>.

       <b>-f</b>, <b>--forward</b>
              Search only in forward strand of database.

       <b>-r</b>, <b>--reverse</b>
              Search only in reverse complement of database.

       <b>-a</b>, <b>--alphabet</b> <u>STRING</u>
              Alphabet type of input sequences (dna, rna, dna5, rna5, protein, char). One  of  <u>dna</u>,  <u>dna5</u>,  <u>rna</u>,
              <u>rna5</u>, <u>protein</u>, and <u>char</u>.

       <b>-v</b>, <b>--verbose</b>
              Set verbosity mode.

   <b>Filtering</b> <b>Options:</b>
       <b>-k</b>, <b>--kmer</b> <u>INTEGER</u>
              Length of the q-grams (max 32). In range [1..32].

       <b>-rp</b>, <b>--repeatPeriod</b> <u>INTEGER</u>
              Maximal period of low complexity repeats to be filtered. Default: <u>1</u>.

       <b>-rl</b>, <b>--repeatLength</b> <u>INTEGER</u>
              Minimal length of low complexity repeats to be filtered. Default: <u>1000</u>.

       <b>-c</b>, <b>--abundanceCut</b> <u>DOUBLE</u>
              k-mer overabundance cut ratio. In range [0..1]. Default: <u>1</u>.

   <b>Verification</b> <b>Options:</b>
       <b>-x</b>, <b>--xDrop</b> <u>DOUBLE</u>
              Maximal x-drop for extension. Default: <u>5</u>.

       <b>-vs</b>, <b>--verification</b> <u>STRING</u>
              Verification   strategy:  exact  or  bestLocal  or  bandedGlobal  One  of  <u>exact</u>,  <u>bestLocal</u>,  and
              <u>bandedGlobal</u>. Default: <u>exact</u>.

       <b>-dt</b>, <b>--disableThresh</b> <u>INTEGER</u>
              Maximal number of verified matches before disabling verification for one query  sequence  (default
              infinity). In range [0..inf].

       <b>-n</b>, <b>--numMatches</b> <u>INTEGER</u>
              Maximal  number  of  kept  matches per query and database. If STELLAR finds more matches, only the
              longest ones are kept. Default: <u>50</u>.

       <b>-s</b>, <b>--sortThresh</b> <u>INTEGER</u>
              Number of matches triggering removal of duplicates. Choose  a  smaller  value  for  saving  space.
              Default: <u>500</u>.

   <b>Output</b> <b>Options:</b>
       <b>-o</b>, <b>--out</b> <u>OUTPUT_FILE</u>
              Name of output file. Valid filetypes are: <u>.txt</u> and <u>.gff</u>. Default: <u>stellar.gff</u>.

       <b>-od</b>, <b>--outDisabled</b> <u>OUTPUT_FILE</u>
              Name  of  output  file  for  disabled  query  sequences.  Valid filetypes are: <u>.sam[.*]</u>, <u>.raw[.*]</u>,
              <u>.frn[.*]</u>, <u>.fq[.*]</u>, <u>.fna[.*]</u>, <u>.ffn[.*]</u>, <u>.fastq[.*]</u>, <u>.fasta[.*]</u>, <u>.faa[.*]</u>, <u>.fa[.*]</u>, and <u>.bam</u>,  where
              *  is any of the following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent (de)compression. Default:
              <u>stellar.disabled.fasta</u>.

</pre><h4><b>REFERENCES</b></h4><pre>
       Kehr, B., Weese, D., Reinert, K.: STELLAR: fast and exact local alignments. BMC Bioinformatics,  12(Suppl
       9):S15, 2011.

stellar 1.4.11 [tarball]                                                                              <u><a href="../man1/STELLAR.1.html">STELLAR</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>