<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>bolt - Efficient large cohorts genome-wide Bayesian mixed-model association testing</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/bolt-lmm">bolt-lmm_2.4.1+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       bolt - Efficient large cohorts genome-wide Bayesian mixed-model association testing

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>bolt</b> [<u>options</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The BOLT-LMM software package currently consists of two main algorithms, the BOLT-LMM algorithm for mixed
       model  association  testing,  and  the  BOLT-REML  algorithm  for  variance  components  analysis  (i.e.,
       partitioning of SNP-heritability and estimation of genetic correlations).

       The BOLT-LMM algorithm computes statistics for testing association between phenotype and genotypes  using
       a  linear  mixed  model.  By default, BOLT-LMM assumes a Bayesian mixture-of-normals prior for the random
       effect attributed to SNPs  other  than  the  one  being  tested.  This  model  generalizes  the  standard
       infinitesimal  mixed model used by previous mixed model association methods, providing an opportunity for
       increased power to detect associations while controlling false positives. Additionally, BOLT-LMM  applies
       algorithmic  advances  to compute mixed model association statistics much faster than eigendecomposition-
       based methods, both when using the Bayesian mixture model and when specialized to  standard  mixed  model
       association.

       The BOLT-REML algorithm estimates heritability explained by genotyped SNPs and genetic correlations among
       multiple  traits  measured on the same set of individuals. BOLT-REML applies variance components analysis
       to perform these tasks, supporting both multi-component modeling to partition SNP-heritability and multi-
       trait modeling to estimate correlations. BOLT-REML applies a Monte Carlo algorithm that  is  much  faster
       than eigendecomposition-based methods for variance components analysis at large sample sizes.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b> [ <b>--help</b> ] print help message with typical options

       <b>--helpFull</b>
              print help message with full option list

       <b>--bfile</b> arg
              prefix of PLINK .fam, .bim, .bed files

       <b>--bfilegz</b> arg
              prefix of PLINK .fam.gz, .bim.gz, .bed.gz files

       <b>--fam</b> arg
              PLINK .fam file (note: file names ending in .gz are auto-[de]compressed)

       <b>--bim</b> arg
              PLINK .bim file(s); for &gt;1, use multiple <b>--bim</b> and/or {i:j}, e.g., data.chr{1:22}.bim

       <b>--bed</b> arg
              PLINK .bed file(s); for &gt;1, use multiple <b>--bim</b> and/or {i:j} expansion

       <b>--geneticMapFile</b> arg
              Oxford-format file for interpolating genetic distances: tables/genetic_map_hg##.txt.gz

       <b>--remove</b> arg
              file(s) listing individuals to ignore (no header; FID IID must be first two columns)

       <b>--exclude</b> arg
              file(s) listing SNPs to ignore (no header; SNP ID must be first column)

       <b>--maxMissingPerSnp</b> arg (=0.1)
              QC filter: max missing rate per SNP

       <b>--maxMissingPerIndiv</b> arg (=0.1) QC filter: max missing rate per person

       <b>--phenoFile</b> arg
              phenotype file (header required; FID IID must be first two columns)

       <b>--phenoCol</b> arg
              phenotype column header

       <b>--phenoUseFam</b>
              use last (6th) column of .fam file as phenotype

       <b>--covarFile</b> arg
              covariate file (header required; FID IID must be first two columns)

       <b>--covarCol</b> arg
              categorical covariate column(s); for &gt;1, use multiple <b>--covarCol</b> and/or {i:j} expansion

       <b>--qCovarCol</b> arg
              quantitative covariate column(s); for &gt;1, use multiple <b>--qCovarCol</b> and/or {i:j} expansion

       <b>--covarUseMissingIndic</b>
              include  samples with missing covariates in analysis via missing indicator method (default: ignore
              such samples)

       <b>--reml</b> run variance components analysis to precisely estimate heritability (but not compute assoc stats)

       <b>--lmm</b>  compute assoc stats under the inf model and with Bayesian non-inf prior (VB approx), if power gain
              expected

       <b>--lmmInfOnly</b>
              compute mixed model assoc stats under the infinitesimal model

       <b>--lmmForceNonInf</b>
              compute non-inf assoc stats even if BOLT-LMM expects no power gain

       <b>--modelSnps</b> arg
              file(s) listing SNPs to use in model (i.e., GRM) (default: use all non-excluded SNPs)

       <b>--LDscoresFile</b> arg
              LD Scores for calibration of Bayesian assoc stats: tables/LDSCORE.1000G_EUR.tab.gz

       <b>--numThreads</b> arg (=1)
              number of computational threads

       <b>--statsFile</b> arg
              output file for assoc stats at PLINK genotypes

       <b>--dosageFile</b> arg
              file(s) containing imputed SNP dosages to test for association (see manual for format)

       <b>--dosageFidIidFile</b> arg
              file listing FIDs and IIDs of samples in dosageFile(s), one line per sample

       <b>--statsFileDosageSnps</b> arg
              output file for assoc stats at dosage format genotypes

       <b>--impute2FileList</b> arg
              list of [chr file] pairs containing IMPUTE2 SNP probabilities to test for association

       <b>--impute2FidIidFile</b> arg
              file listing FIDs and IIDs of samples in IMPUTE2 files, one line per sample

       <b>--impute2MinMAF</b> arg (=0)
              MAF threshold on IMPUTE2 genotypes; lower-MAF SNPs will be ignored

       <b>--bgenFile</b> arg
              file(s) containing Oxford BGEN-format genotypes to test for association

       <b>--sampleFile</b> arg
              file containing Oxford sample file corresponding to BGEN file(s)

       <b>--bgenSampleFileList</b> arg
              list of [bgen sample] file pairs containing BGEN imputed variants to test for association

       <b>--bgenMinMAF</b> arg (=0)
              MAF threshold on Oxford BGEN-format genotypes; lower-MAF SNPs will be ignored

       <b>--bgenMinINFO</b> arg (=0)
              INFO threshold on Oxford BGEN-format genotypes; lower-INFO SNPs will be ignored

       <b>--statsFileBgenSnps</b> arg
              output file for assoc stats at BGEN-format genotypes

       <b>--statsFileImpute2Snps</b> arg
              output file for assoc stats at IMPUTE2 format genotypes

       <b>--dosage2FileList</b> arg
              list of [map  dosage]  file  pairs  with  2-dosage  SNP  probabilities  (Ricopili/plink2  <b>--dosage</b>
              format=2) to test for association

       <b>--statsFileDosage2Snps</b> arg
              output file for assoc stats at 2-dosage format genotypes

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b>https://data.broadinstitute.org/alkesgroup/BOLT-LMM/</b>

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright Â© 2014-2018 Harvard University.  Distributed under the GNU GPLv3+ open source license.

2.3.2                                               May 2018                                             <u><a href="../man1/BOLT.1.html">BOLT</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>