<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>ktClassifyBLAST - explore hierarchical metagenomic data with zoomable pie charts</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/radiant">radiant_2.8.1+dfsg-2_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       ktClassifyBLAST - explore hierarchical metagenomic data with zoomable pie charts

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Assigns each query in tabular BLAST results to an NCBI taxonomy ID. If the results contain comment lines,
       queries with no hits will be included in the output (with taxonomy IDs of <b>-1</b> for consistency with MEGAN).

       ktClassifyBLAST \

              [options] \ blast_output_1 \ [blast_output_2] \ ...

       blast_output
              File  containing  BLAST results in tabular format ("Hit table (text)" when downloading from NCBI).
              If running BLAST locally, subject IDs in the local database must contain accession numbers, either
              bare or in the fourth field of the pipe-separated ("gi|12345|xx|ABC123.1|") format.

       [-o &lt;string&gt;] Output file name. [Default: 'blast.taxonomy.tab']

       [-t &lt;number&gt;]
              Threshold for bit score differences when determining "best" hits. Hits with scores that are within
              this distance of the highest score will be included when computing the lowest common ancestor  (or
              picking randomly if <b>-r</b> is specified). [Default: '3']

       [-f]   If any best hits have unknown accessions, force classification to root instead of ignoring them.

       [-r]   Pick from the best hits randomly instead of finding the lowest common ancestor.

       [-p]   Use percent identity for average scores instead of log[10] e-value.

       [-b]   Use bit score for average scores instead of log[10] e-value.

       [-s]   Summarize counts and average scores by taxonomy ID.

       [-e &lt;number&gt;]
              E-value  factor  for determining "best" hits. A bit score difference threshold (<b>-t</b>) is recommended
              instead to avoid comparing e-values that BLAST reports as  0  due  to  floating  point  underflow.
              However,  an  e-value  factor  should  be  used  if the input is a concatination of BLASTs against
              different databases.

       Default:          &lt;queryID&gt; &lt;taxID&gt; &lt;score&gt;

       Summarized (<b>-s</b>):  &lt;count&gt; &lt;taxID&gt; &lt;score&gt;

       queryID
              The query ID as it appears in the BLAST results.

       taxID  The NCBI taxonomy ID the query was assigned to (or <b>-1</b> if it has no hits).

       score  The score of the assignment(s); by default, the average E-value of "best" hits (see <b>-p</b>, <b>-b</b>).

       count  The number of assignments.

ktClassifyBLAST 2.8                                 June 2021                                 <u><a href="../man1/KTCLASSIFYBLAST.1.html">KTCLASSIFYBLAST</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>