<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>bcftools - utilities for variant calling and manipulating VCFs and BCFs.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/bcftools">bcftools_1.21-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       bcftools - utilities for variant calling and manipulating VCFs and BCFs.

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>bcftools</b> [--version|--version-only] [--help] [<u>COMMAND</u>] [<u>OPTIONS</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       BCFtools  is  a set of utilities that manipulate variant calls in the Variant Call Format (VCF) and its
       binary counterpart BCF. All commands work transparently with both VCFs and BCFs, both uncompressed and
       BGZF-compressed.

       Most commands accept VCF, bgzipped VCF and BCF with filetype detected automatically even when streaming
       from a pipe. Indexed VCF and BCF will work in all situations. Un-indexed VCF and BCF and streams will
       work in most, but not all situations. In general, whenever multiple VCFs are read simultaneously, they
       must be indexed and therefore also compressed. (Note that files with non-standard index names can be
       accessed as e.g. "bcftools view -r X:2928329 file.vcf.gz##idx##non-standard-index-name".)

       BCFtools is designed to work on a stream. It regards an input file "-" as the standard input (stdin) and
       outputs to the standard output (stdout). Several commands can thus be  combined  with  Unix pipes.

   <b>VERSION</b>
       This manual page was last updated <b>2024-09-12</b> and refers to bcftools git version <b>1.21</b>.

   <b>BCF1</b>
       The obsolete BCF1 format output by versions of samtools &lt;= 0.1.19 is <b>not</b> compatible with this version of
       bcftools. To read BCF1 files one can use the view command from old versions of bcftools packaged with
       samtools versions &lt;= 0.1.19 to convert to VCF, which can then be read by this version of bcftools.

               samtools-0.1.19/bcftools/bcftools view file.bcf1 | bcftools view

   <b>VARIANT</b> <b>CALLING</b>
       See <u>bcftools</u> <u>call</u> for variant calling from the output of the <u>samtools</u> <u>mpileup</u> command. In versions of
       samtools &lt;= 0.1.19 calling was done with <u>bcftools</u> <u>view</u>. Users are now required to choose between the old
       samtools calling model (<u>-c/--consensus-caller</u>) and the new multiallelic calling model
       (<u>-m/--multiallelic-caller</u>). The multiallelic calling model is recommended for most tasks.

   <b>FILTERING</b> <b>EXPRESSIONS</b>
       See <b>EXPRESSIONS</b>

</pre><h4><b>LIST</b> <b>OF</b> <b>COMMANDS</b></h4><pre>
       For a full list of available commands, run <b>bcftools</b> without arguments. For a full list of available
       options, run <b>bcftools</b> <u>COMMAND</u> without arguments.

       •   <b>annotate</b>  .. edit VCF files, add or remove annotations

       •   <b>call</b>      .. SNP/indel calling (former "view")

       •   <b>cnv</b>       .. Copy Number Variation caller

       •   <b>concat</b>    .. concatenate VCF/BCF files from the same set of samples

       •   <b>consensus</b> .. create consensus sequence by applying VCF variants

       •   <b>convert</b>   .. convert VCF/BCF to other formats and back

       •   <b>csq</b>       .. haplotype aware consequence caller

       •   <b>filter</b>    .. filter VCF/BCF files using fixed thresholds

       •   <b>gtcheck</b>   .. check sample concordance, detect sample swaps and contamination

       •   <b>head</b>      .. view VCF/BCF file headers

       •   <b>index</b>     .. index VCF/BCF

       •   <b>isec</b>      .. intersections of VCF/BCF files

       •   <b>merge</b>     .. merge VCF/BCF files files from non-overlapping sample sets

       •   <b>mpileup</b>   .. multi-way pileup producing genotype likelihoods

       •   <b>norm</b>      .. normalize indels

       •   <b>plugin</b>    .. run user-defined plugin

       •   <b>polysomy</b>  .. detect contaminations and whole-chromosome aberrations

       •   <b>query</b>     .. transform VCF/BCF into user-defined formats

       •   <b>reheader</b>  .. modify VCF/BCF header, change sample names

       •   <b>roh</b>       .. identify runs of homo/auto-zygosity

       •   <b>sort</b>      .. sort VCF/BCF files

       •   <b>stats</b>     .. produce VCF/BCF stats (former vcfcheck)

       •   <b>view</b>      .. subset, filter and convert VCF and BCF files

</pre><h4><b>LIST</b> <b>OF</b> <b>SCRIPTS</b></h4><pre>
       Some helper scripts are bundled with the bcftools code.

       •   <b>gff2gff</b>  .. converts a GFF file to the format required by <b>csq</b>

       •   <b>plot-vcfstats</b>  .. plots the output of <b>stats</b>

</pre><h4><b>COMMANDS</b> <b>AND</b> <b>OPTIONS</b></h4><pre>
   <b>Common</b> <b>Options</b>
       The following options are common to many bcftools commands. See usage for specific commands to see if
       they apply.

       <u>FILE</u>
           Files can be both VCF or BCF, uncompressed or BGZF-compressed. The file "-" is interpreted as
           standard input. Some tools may require tabix- or CSI-indexed files.

       <b>-c,</b> <b>--collapse</b> <u>snps</u>|<u>indels</u>|<u>both</u>|<u>all</u>|<u>some</u>|<u>none</u>|<u>id</u>
           Controls  how to treat records with duplicate positions and defines compatible records across
           multiple input files. Here by "compatible" we mean records which should be considered as identical by
           the tools. For example, when performing line intersections, the desire may be to consider as
           identical all sites with matching positions (<b>bcftools</b> <b>isec</b> <b>-c</b> <u>all</u>), or only sites with  matching
           variant type (<b>bcftools</b> <b>isec</b> <b>-c</b> <u>snps</u>  <b>-c</b> <u>indels</u>), or only sites with all alleles identical (<b>bcftools</b>
           <b>isec</b> <b>-c</b> <u>none</u>).

           <u>none</u>
               only records with identical REF and ALT alleles are compatible

           <u>some</u>
               only records where some subset of ALT alleles match are compatible

           <u>all</u>
               all records are compatible, regardless of whether the ALT alleles match or not. In the case of
               records with the same position, only the first will be considered and appear on output.

           <u>snps</u>
               any SNP records are compatible, regardless of whether the ALT alleles match or not. For duplicate
               positions, only the first SNP record will be considered and appear on output.

           <u>indels</u>
               all  indel records are compatible, regardless of whether the REF and ALT alleles match or not.
               For duplicate positions, only the first indel record will be considered and appear on output.

           <u>both</u>
               abbreviation of "<b>-c</b> <u>indels</u>  <b>-c</b> <u>snps</u>"

           <u>id</u>
               only records with identical ID column are compatible. Supported by <b>bcftools</b> <b>merge</b> only.

       <b>-f,</b> <b>--apply-filters</b> <u>LIST</u>
           Skip sites where FILTER column does not contain any of the strings listed in <u>LIST</u>. For example, to
           include only sites which have no filters set, use <b>-f</b> <u>.,PASS</u>.

       <b>--no-version</b>
           Do not append version and command line information to the output VCF header.

       <b>-o,</b> <b>--output</b> <u>FILE</u>
           When output consists of a single stream, write it to <u>FILE</u> rather than to standard output, where it is
           written by default. The file type is determined automatically from the file name suffix and in case a
           conflicting <b>-O</b> option is given, the file name suffix takes precedence.

       <b>-O,</b> <b>--output-type</b> <u>b</u>|<u>u</u>|<u>z</u>|<u>v</u>[0-9]
           Output compressed BCF (<u>b</u>), uncompressed BCF (<u>u</u>), compressed VCF (<u>z</u>), uncompressed VCF (<u>v</u>). Use the
           -Ou option when piping between bcftools subcommands to speed up performance by removing unnecessary
           compression/decompression and VCF←→BCF conversion. The compression level of the compressed formats (<u>b</u>
           and <u>z</u>) can be set by by appending a number between 0-9.

       <b>-r,</b> <b>--regions</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:beg-end</u>|<u>chr:beg-</u>[,...]
           Comma-separated list of regions, see also <b>-R,</b> <b>--regions-file</b>. Overlapping records are matched even
           when the starting coordinate is outside of the region, unlike the <b>-t/-T</b> options where only the POS
           coordinate is checked. Note that <b>-r</b> cannot be used in combination with <b>-R</b>.

       <b>-R,</b> <b>--regions-file</b> <u>FILE</u>
           Regions can be specified either on command line or in a VCF, BED, or tab-delimited file (the
           default). The columns of the tab-delimited file can contain either positions (two-column format:
           CHROM, POS) or intervals (three-column format: CHROM, BEG, END), but not both. Positions are 1-based
           and inclusive. The columns of the tab-delimited BED file are also CHROM, POS and END (trailing
           columns are ignored), but coordinates are 0-based, half-open. To indicate that a file be treated as
           BED rather than the 1-based tab-delimited file, the file must have the ".bed" or ".bed.gz" suffix
           (case-insensitive). Uncompressed files are stored in memory, while bgzip-compressed and tabix-indexed
           region files are streamed. Note that sequence names must match exactly, "chr20" is not the same as
           "20". Also note that chromosome ordering in <u>FILE</u> will be respected, the VCF will be processed in the
           order in which chromosomes first appear in <u>FILE</u>. However, within chromosomes, the VCF will always be
           processed in ascending genomic coordinate order no matter what order they appear in <u>FILE</u>. Note that
           overlapping regions in <u>FILE</u> can result in duplicated out of order positions in the output. This
           option requires indexed VCF/BCF files. Note that <b>-R</b> cannot be used in combination with <b>-r</b>.

       <b>--regions-overlap</b> <u>pos</u>|<u>record</u>|<u>variant</u>|<u>0</u>|<u>1</u>|<u>2</u>
           This option controls how overlapping records are determined: set to <b>pos</b> or <b>0</b> if the VCF record has to
           have POS inside a region (this corresponds to the default behavior of <b>-t/-T</b>); set to <b>record</b> or <b>1</b> if
           also overlapping records with POS outside a region should be included (this is the default behavior
           of <b>-r/-R</b>, and includes indels with POS at the end of a region, which are technically outside the
           region); or set to <b>variant</b> or <b>2</b> to include only true overlapping variation (compare the full VCF
           representation "TA&gt;T-" vs the true sequence variation "A&gt;-").

       <b>-s,</b> <b>--samples</b> [^]<u>LIST</u>
           Comma-separated list of samples to include or exclude if prefixed with "^." (Note that when multiple
           samples are to be excluded, the "^" prefix is still present only once, e.g. "^SAMPLE1,SAMPLE2".) The
           sample order is updated to reflect that given on the command line. Note that in general tags such as
           INFO/AC, INFO/AN, etc are not updated to correspond to the subset samples. <b>bcftools</b> <b>view</b> is the
           exception where some tags will be updated (unless the <b>-I,</b> <b>--no-update</b> option is used; see <b>bcftools</b>
           <b>view</b> documentation). To use updated tags for the subset in another command one can pipe from <b>view</b>
           into that command. For example:

               bcftools view -Ou -s sample1,sample2 file.vcf | bcftools query -f %INFO/AC\t%INFO/AN\n

       <b>-S,</b> <b>--samples-file</b> [^]<u>FILE</u>
           File of sample names to include or exclude if prefixed with "^". One sample per line. See also the
           note above for the <b>-s,</b> <b>--samples</b> option. The sample order is updated to reflect that given in the
           input file. The command <b>bcftools</b> <b>call</b> accepts an optional second column indicating ploidy (0, 1 or 2)
           or sex (as defined by <b>--ploidy</b>, for example "F" or "M"), for example:

               sample1    1
               sample2    2
               sample3    2

       or

               sample1    M
               sample2    F
               sample3    F

       If the second column is not present, the sex "F" is assumed. With <b>bcftools</b> <b>call</b> <b>-C</b> <u>trio</u>, PED file is
       expected. The program ignores the first column and the last indicates sex (1=male, 2=female), for
       example:

               ignored_column  daughterA fatherA  motherA  2
               ignored_column  sonB      fatherB  motherB  1

       <b>-t,</b> <b>--targets</b> [^]<u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           Similar as <b>-r,</b> <b>--regions</b>, but the next position is accessed by streaming the whole VCF/BCF rather
           than using the tbi/csi index. Both <b>-r</b> and <b>-t</b> options can be applied simultaneously: <b>-r</b>  uses  the
           index  to  jump  to  a  region and <b>-t</b> discards positions which are not in the targets. Unlike <b>-r</b>,
           targets can be prefixed with "^" to request logical complement. For example, "^X,Y,MT" indicates that
           sequences X, Y and MT should be skipped. Yet another difference between the <b>-t/-T</b> and <b>-r/-R</b> is that
           <b>-r/-R</b> checks for proper overlaps and considers both POS and the end position of an indel, while <b>-t/-T</b>
           considers the POS coordinate only (by default; see also <b>--regions-overlap</b> and <b>--targets-overlap</b>).
           Note that <b>-t</b> cannot be used in combination with <b>-T</b>.

       <b>-T,</b> <b>--targets-file</b> [^]<u>FILE</u>
           Same <b>-t,</b> <b>--targets</b>, but reads regions from a file. Note that <b>-T</b> cannot be used in combination with
           <b>-t</b>.

           With the <b>call</b> <b>-C</b> <u>alleles</u> command, third column of the targets file must be comma-separated list of
           alleles, starting with the reference allele. Note that the file must be compressed and indexed. Such
           a file can be easily created from a VCF using:

               bcftools query -f'%CHROM\t%POS\t%REF,%ALT\n' file.vcf | bgzip -c &gt; als.tsv.gz &amp;&amp; tabix -s1 -b2 -e2 als.tsv.gz

       <b>--targets-overlap</b> <u>pos</u>|<u>record</u>|<u>variant</u>|<u>0</u>|<u>1</u>|<u>2</u>
           Same as <b>--regions-overlap</b> but for <b>-t/-T</b>.

       <b>--threads</b> <u>INT</u>
           Use multithreading with <u>INT</u> worker threads. The option is currently used only for the compression of
           the output stream, only when <u>--output-type</u> is <u>b</u> or <u>z</u>. Default: 0.

       <b>-W</b>[<u>FMT</u>]<b>,</b> <b>-W</b>[=<u>FMT</u>]<b>,</b> <b>--write-index</b>[=<u>FMT</u>]
           Automatically index the output files. <u>FMT</u> is optional and can be one of "tbi" or "csi" depending on
           output file format. Defaults to CSI unless specified otherwise. Can be used only for compressed BCF
           and VCF output.

   <b>bcftools</b> <b>annotate</b> <u><b>[OPTIONS]</b></u> <u><b>FILE</b></u>
       Add or remove annotations.

       <b>-a,</b> <b>--annotations</b> <u>file</u>
           Bgzip-compressed and tabix-indexed file with annotations. The file can be VCF, BED, or a
           tab-delimited file with mandatory columns CHROM, POS (or, alternatively, FROM and TO), optional
           columns REF and ALT, and arbitrary number of annotation columns. BED files are expected to have the
           ".bed" or ".bed.gz" suffix (case-insensitive), otherwise a tab-delimited file is assumed. Note that
           in case of tab-delimited file, the coordinates POS, FROM and TO are one-based and inclusive. When REF
           and ALT are present, only matching VCF records will be annotated. If the END coordinate is present in
           the annotation file and given on command line as "-c ~INFO/END", then VCF records will be matched
           also by the INFO/END coordinate. If ID is present in the annotation file and given as "-c ~ID", then
           VCF records will be matched also by the ID column.

           When multiple ALT alleles are present in the annotation file (given as comma-separated list of
           alleles), at least one must match one of the alleles in the corresponding VCF record. Similarly, at
           least one alternate allele from a multi-allelic VCF record must be present in the annotation file.

           Missing values can be added by providing "." in place of actual value and using the missing value
           modifier with <b>-c</b>, such as ".TAG".

           Note that flag types, such as "INFO/FLAG", can be annotated by including a field with the value "1"
           to set the flag, "0" to remove it, or "." to keep existing flags. See also <b>-c,</b> <b>--columns</b> and <b>-h,</b>
           <b>--header-lines</b>.

               # Sample annotation file with columns CHROM, POS, STRING_TAG, NUMERIC_TAG
               1  752566  SomeString      5
               1  798959  SomeOtherString 6

       <b>-c,</b> <b>--columns</b> <u>list</u>
           Comma-separated list of columns or tags to carry over from the annotation file (see also <b>-a,</b>
           <b>--annotations</b>). If the annotation file is not a VCF/BCF, <u>list</u> describes the columns of the annotation
           file and must include CHROM, POS (or, alternatively, FROM,TO or BEG,END), and optionally REF and ALT.
           Unused columns which should be ignored can be indicated by "-".

           If the annotation file is a VCF/BCF, only the edited columns/tags must be present and their order
           does not matter. The columns ID, QUAL, FILTER, INFO and FORMAT can be edited, where INFO tags can be
           written both as "INFO/TAG" or simply "TAG", and FORMAT tags can be written as "FORMAT/TAG" or
           "FMT/TAG". The imported VCF annotations can be renamed as "DST_TAG:=SRC_TAG" or
           "FMT/DST_TAG:=FMT/SRC_TAG".

           To carry over all INFO annotations, use "INFO". To add all INFO annotations except "TAG", use
           "^INFO/TAG". By default, existing values are replaced.

           By default, existing tags are overwritten unless the source value is a missing value (i.e. "."). If
           also missing values should be carried over (and overwrite existing tags), use ".TAG" instead of
           "TAG". To add annotations without overwriting existing values (that is, to add tags that are absent
           or to add values to existing tags with missing values), use "+TAG" instead of "TAG". These can be
           combined, for example ".+TAG" can be used to add TAG even if the source value is missing but only if
           TAG does not exist in the target file; existing tags will not be overwritten. To append to existing
           values (rather than replacing or leaving untouched), use "=TAG" (instead of "TAG" or "+TAG"). To
           replace only existing values without modifying missing annotations, use "-TAG". As a special case of
           this, if position needs to be replaced, mark the column with the new coordinate as "-POS". (Note that
           in previous releases this used to be "~POS", now deprecated.)

           To match the record also by ID or INFO/END, in addition to REF and ALT, use "~ID" or "~INFO/END".
           Note that this works only for ID and POS, for other fields see the description of <b>-i</b> below.

           If the annotation file is not a VCF/BCF, all new annotations must be defined via <b>-h,</b> <b>--header-lines</b>.

           See also the <b>-l,</b> <b>--merge-logic</b> option.

           <b>Summary</b> <b>of</b> -c, --columns<b>:</b>

               CHROM,POS,TAG       .. match by chromosome and position, transfer annotation from TAG
               CHROM,POS,-,TAG     .. same as above, but ignore the third column of the annotation file
               CHROM,BEG,END,TAG   .. match by region (BEG,END are synonymous to FROM,TO)
               CHROM,POS,REF,ALT   .. match by CHROM, POS, REF and ALT

               DST_TAG:=SRC_TAG    .. transfer the SRC_TAG using the new name DST_TAG
               INFO                .. transfer all INFO annotations
               ^INFO/TAG           .. transfer all INFO annotations except "TAG"

               TAG       .. add or overwrite existing target value if source is not "." and skip otherwise
               +TAG      .. add or overwrite existing target value only it is "."
               .TAG      .. add or overwrite existing target value even if source is "."
               .+TAG     .. add new but never overwrite existing tag, regardless of its value; can transfer "." if target does not exist
               -TAG      .. overwrite existing value, never add new if target does not exist
               =TAG      .. do not overwrite but append value to existing tags

               ~FIELD    .. use this column to match lines with -i/-e expression (see the description of -i below)
               ~ID       .. in addition to CHROM,POS,REF,ALT match by also ID
               ~INFO/END .. in addition to CHROM,POS,REF,ALT match by also INFO/END

       <b>-C,</b> <b>--columns-file</b> <u>file</u>
           Read the list of columns from a file (normally given via the <b>-c,</b> <b>--columns</b> option). "-" to skip a
           column of the annotation file. One column name per row, an additional space- or tab-separated field
           can be present to indicate the merge logic (normally given via the <b>-l,</b> <b>--merge-logic</b> option). This is
           useful when many annotations are added at once.

       <b>-e,</b> <b>--exclude</b> <u>EXPRESSION</u>
           exclude sites for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b> and the extension
           described in <b>-i,</b> <b>--include</b> below.

       <b>--force</b>
           continue even when parsing errors, such as undefined tags, are encountered. Note this can be an
           unsafe operation and can result in corrupted BCF files. If this option is used, make sure to sanity
           check the result thoroughly.

       <b>-h,</b> <b>--header-lines</b> <u>file</u>
           Lines to append to the VCF header, see also <b>-c,</b> <b>--columns</b> and <b>-a,</b> <b>--annotations</b>. For example:

               ##INFO=&lt;ID=NUMERIC_TAG,Number=1,Type=Integer,Description="Example header line"&gt;
               ##INFO=&lt;ID=STRING_TAG,Number=1,Type=String,Description="Yet another header line"&gt;

       <b>-I,</b> <b>--set-id</b> [+]<u>FORMAT</u>
           assign ID on the fly. The format is the same as in the <b>query</b> command (see below). By default all
           existing IDs are replaced. If the format string is preceded by "+", only missing IDs will be set. For
           example, one can use

               bcftools annotate --set-id +'%CHROM\_%POS\_%REF\_%FIRST_ALT' file.vcf

       <b>-i,</b> <b>--include</b> <u>EXPRESSION</u>
           include only sites for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.

           Additionally, the command <b>bcftools</b> <b>annotate</b> supports expressions updated from the annotation file
           dynamically for each record:

               # The field 'STR' from the -a file is required to match INFO/TAG in VCF. In the first example
               # the alleles REF,ALT must match, in the second example they are ignored. The option -k is required
               # to output also records that are not annotated. The third example shows the same concept with
               # a numerical expression.
               bcftools annotate -a annots.tsv.gz -c CHROM,POS,REF,ALT,SCORE,~STR -i'TAG={STR}' -k input.vcf
               bcftools annotate -a annots.tsv.gz -c CHROM,POS,-,-,SCORE,~STR     -i'TAG={STR}' -k input.vcf
               bcftools annotate -a annots.tsv.gz -c CHROM,POS,-,-,SCORE,~INT     -i'TAG&gt;{INT}' -k input.vcf

       <b>-k,</b> <b>--keep-sites</b>
           keep sites which do not pass <b>-i</b> and <b>-e</b> expressions instead of discarding them

       <b>-l,</b> <b>--merge-logic</b> <u>tag:first</u>|<u>append</u>|<u>append-missing</u>|<u>unique</u>|<u>sum</u>|<u>avg</u>|<u>min</u>|<u>max</u>[,...]
           When multiple regions overlap a single record, this option defines how to treat multiple annotation
           values when setting <u>tag</u> in the destination file: use the first encountered value ignoring the rest
           (<u>first</u>); append allowing duplicates (<u>append</u>); append even if the appended value is missing, i.e. is a
           dot (<u>append-missing</u>); append discarding duplicate values (<u>unique</u>); sum the values (<u>sum</u>, numeric
           fields only); average the values (<u>avg</u>); use the minimum value (<u>min</u>) or the maximum (<u>max</u>). + Note that
           this option is intended for use with BED or TAB-delimited annotation files only. Moreover, it is
           effective only when either <u>REF</u> and <u>ALT</u> or <u>BEG</u> and <u>END</u> <b>--columns</b> are present . + Multiple rules can be
           given either as a comma-separated list or giving the option multiple times. This is an experimental
           feature.

       <b>-m,</b> <b>--mark-sites</b> <u>TAG</u>
           annotate sites which are present ("+") or absent ("-") in the <b>-a</b> file with a new INFO/TAG flag

       <b>--min-overlap</b> <u>ANN</u>:'VCF'
           minimum overlap required as a fraction of the variant in the annotation <b>-a</b> file (<u>ANN</u>), in the target
           VCF file (<u>:VCF</u>), or both for reciprocal overlap (<u>ANN:VCF</u>). By default overlaps of arbitrary length
           are sufficient. The option can be used only with the tab-delimited annotation <b>-a</b> file and with <u>BEG</u>
           and <u>END</u> columns present.

       <b>--no-version</b>
           see <b>Common</b> <b>Options</b>

       <b>-o,</b> <b>--output</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>-O,</b> <b>--output-type</b> <u>b</u>|<u>u</u>|<u>z</u>|<u>v</u>[0-9]
           see <b>Common</b> <b>Options</b>

       <b>--pair-logic</b> <u>snps</u>|<u>indels</u>|<u>both</u>|<u>all</u>|<u>some</u>|<u>exact</u>
           Controls how to match records from the annotation file to the target VCF. Effective only when <b>-a</b> is a
           VCF or BCF. The option replaces the former uninuitive <b>--collapse</b>. See <b>Common</b> <b>Options</b> for more.

       <b>-r,</b> <b>--regions</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-R,</b> <b>--regions-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--regions-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>--rename-annots</b> <u>file</u>
           rename annotations according to the map in <u>file</u>, with "old_name new_name\n" pairs separated by
           whitespaces, each on a separate line. The old name must be prefixed with the annotation type: INFO,
           FORMAT, or FILTER.

       <b>--rename-chrs</b> <u>file</u>
           rename chromosomes according to the map in <u>file</u>, with "old_name new_name\n" pairs separated by
           whitespaces, each on a separate line.

       <b>-s,</b> <b>--samples</b> [^]<u>LIST</u>
           subset of samples to annotate, see also <b>Common</b> <b>Options</b>

       <b>-S,</b> <b>--samples-file</b> <u>FILE</u>
           subset of samples to annotate. If the samples are named differently in the target VCF and the <b>-a,</b>
           <b>--annotations</b> VCF, the name mapping can be given as "src_name dst_name\n", separated by whitespaces,
           each pair on a separate line.

       <b>--single-overlaps</b>
           use this option to keep memory requirements low with very large annotation files. Note, however, that
           this comes at a cost, only single overlapping intervals are considered in this mode. This was the
           default mode until the commit af6f0c9 (Feb 24 2019).

       <b>--threads</b> <u>INT</u>
           see <b>Common</b> <b>Options</b>

       <b>-x,</b> <b>--remove</b> <u>list</u>
           List of annotations to remove. Use "FILTER" to remove all filters or "FILTER/SomeFilter" to remove a
           specific filter. Similarly, "INFO" can be used to remove all INFO tags and "FORMAT" to remove all
           FORMAT tags except GT. To remove all INFO tags except "FOO" and "BAR", use "^INFO/FOO,INFO/BAR" (and
           similarly for FORMAT and FILTER). "INFO" can be abbreviated to "INF" and "FORMAT" to "FMT".

       <b>-W</b>[<u>FMT</u>]<b>,</b> <b>-W</b>[=<u>FMT</u>]<b>,</b> <b>--write-index</b>[=<u>FMT</u>]
           Automatically index the output file. <u>FMT</u> is optional and can be one of "tbi" or "csi" depending on
           output file format.

       <b>Examples:</b>

               # Remove three fields
               bcftools annotate -x ID,INFO/DP,FORMAT/DP file.vcf.gz

               # Remove all INFO fields and all FORMAT fields except for GT and PL
               bcftools annotate -x INFO,^FORMAT/GT,FORMAT/PL file.vcf

               # Add ID, QUAL and INFO/TAG, not replacing TAG if already present
               bcftools annotate -a src.bcf -c ID,QUAL,+TAG dst.bcf

               # Carry over all INFO and FORMAT annotations except FORMAT/GT
               bcftools annotate -a src.bcf -c INFO,^FORMAT/GT dst.bcf

               # Annotate from a tab-delimited file with six columns (the fifth is ignored),
               # first indexing with tabix. The coordinates are 1-based.
               tabix -s1 -b2 -e2 annots.tab.gz
               bcftools annotate -a annots.tab.gz -h annots.hdr -c CHROM,POS,REF,ALT,-,TAG file.vcf

               # Annotate from a tab-delimited file with regions (1-based coordinates, inclusive)
               tabix -s1 -b2 -e3 annots.tab.gz
               bcftools annotate -a annots.tab.gz -h annots.hdr -c CHROM,FROM,TO,TAG input.vcf

               # Annotate from a bed file (0-based coordinates, half-closed, half-open intervals)
               bcftools annotate -a annots.bed.gz -h annots.hdr -c CHROM,FROM,TO,TAG input.vcf

               # Transfer the INFO/END tag, matching by POS,REF,ALT and ID. This example assumes
               # that INFO/END is already present in the VCF header.
               bcftools annotate -a annots.tab.gz  -c CHROM,POS,~ID,REF,ALT,INFO/END input.vcf

               # For (many) more examples see <a href="http://samtools.github.io/bcftools/howtos/annotate.html">http://samtools.github.io/bcftools/howtos/annotate.html</a>

   <b>bcftools</b> <b>call</b> <u><b>[OPTIONS]</b></u> <u><b>FILE</b></u>
       This command replaces the former <b>bcftools</b> <b>view</b> caller. Some of the original functionality has been
       temporarily lost in the process of transition under htslib &lt;<a href="http://github.com/samtools/htslib">http://github.com/samtools/htslib</a>&gt;, but will
       be added back on popular demand. The original calling model can be invoked with the <b>-c</b> option.

   <b>File</b> <b>format</b> <b>options:</b>
       <b>--no-version</b>
           see <b>Common</b> <b>Options</b>

       <b>-o,</b> <b>--output</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>-O,</b> <b>--output-type</b> <u>b</u>|<u>u</u>|<u>z</u>|<u>v</u>[0-9]
           see <b>Common</b> <b>Options</b>

       <b>--ploidy</b> <u>ASSEMBLY</u>[<u>?</u>]
           predefined ploidy, use <u>list</u> (or any other unused word) to print a list of all predefined assemblies.
           Append a question mark to print the actual definition. See also <b>--ploidy-file</b>.

       <b>--ploidy-file</b> <u>FILE</u>
           ploidy definition given as a space/tab-delimited list of CHROM, FROM, TO, SEX, PLOIDY. The SEX codes
           are arbitrary and correspond to the ones used by <b>--samples-file</b>. The default ploidy can be given
           using the starred records (see below), unlisted regions have ploidy 2. The default ploidy definition
           is

               X 1 60000 M 1
               X 2699521 154931043 M 1
               Y 1 59373566 M 1
               Y 1 59373566 F 0
               MT 1 16569 M 1
               MT 1 16569 F 1
               *  * *     M 2
               *  * *     F 2

       <b>-r,</b> <b>--regions</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-R,</b> <b>--regions-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--regions-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-s,</b> <b>--samples</b> <u>LIST</u>
           see <b>Common</b> <b>Options</b>

       <b>-S,</b> <b>--samples-file</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>-t,</b> <b>--targets</b> <u>LIST</u>
           see <b>Common</b> <b>Options</b>

       <b>-T,</b> <b>--targets-file</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>--targets-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>--threads</b> <u>INT</u>
           see <b>Common</b> <b>Options</b>

       <b>-W</b>[<u>FMT</u>]<b>,</b> <b>-W</b>[=<u>FMT</u>]<b>,</b> <b>--write-index</b>[=<u>FMT</u>]
           Automatically index the output file. <u>FMT</u> is optional and can be one of "tbi" or "csi" depending on
           output file format.

   <b>Input/output</b> <b>options:</b>
       <b>-A,</b> <b>--keep-alts</b>
           output all alternate alleles present in the alignments even if they do not appear in any of the
           genotypes

       <b>-</b>*<b>,</b> <b>--keep-unseen-allele</b>
           keep the unobserved allele &lt;*&gt; or &lt;NON_REF&gt;, useful mainly for gVCF output

       <b>-f,</b> <b>--format-fields</b> <u>list</u>
           comma-separated list of FORMAT fields to output for each sample. Currently GQ and GP fields are
           supported. For convenience, the fields can be given as lower case letters. Prefixed with "^"
           indicates a request for tag removal of auxiliary tags useful only for calling.

       <b>-F,</b> <b>--prior-freqs</b> <u>AN</u>,<u>AC</u>
           take advantage of prior knowledge of population allele frequencies. The workflow looks like this:

               # Extract AN,AC values from an existing VCF, such 1000Genomes
               bcftools query -f'%CHROM\t%POS\t%REF\t%ALT\t%AN\t%AC\n' 1000Genomes.bcf | bgzip -c &gt; AFs.tab.gz

               # If the tags AN,AC are not already present, use the +fill-tags plugin
               bcftools +fill-tags 1000Genomes.bcf | bcftools query -f'%CHROM\t%POS\t%REF\t%ALT\t%AN\t%AC\n' | bgzip -c &gt; AFs.tab.gz
               tabix -s1 -b2 -e2 AFs.tab.gz

               # Create a VCF header description, here we name the tags REF_AN,REF_AC
               cat AFs.hdr
               ##INFO=&lt;ID=REF_AN,Number=1,Type=Integer,Description="Total number of alleles in reference genotypes"&gt;
               ##INFO=&lt;ID=REF_AC,Number=A,Type=Integer,Description="Allele count in reference genotypes for each ALT allele"&gt;

               # Now before calling, stream the raw mpileup output through `bcftools annotate` to add the frequencies
               bcftools mpileup [...] -Ou | bcftools annotate -a AFs.tab.gz -h AFs.hdr -c CHROM,POS,REF,ALT,REF_AN,REF_AC -Ou | bcftools call -mv -F REF_AN,REF_AC [...]

       <b>-G,</b> <b>--group-samples</b> <u>FILE</u>|<u>-</u>
           by default, all samples are assumed to come from a single population. This option groups samples into
           populations and apply the HWE assumption within but not across the populations. <u>FILE</u> is a
           tab-delimited text file with sample names in the first column and group names in the second column.
           If <u>-</u> is given instead, no HWE assumption is made at all and single-sample calling is performed. (Note
           that in low coverage data this inflates the rate of false positives.) The <b>-G</b> option requires the
           presence of per-sample FORMAT/QS or FORMAT/AD tag generated with <b>bcftools</b> <b>mpileup</b> <b>-a</b> <b>QS</b> (or <b>-a</b> <b>AD</b>).

       <b>-g,</b> <b>--gvcf</b> <u>INT</u>[,...]
           output gVCF blocks of homozygous REF calls, with depth (DP) ranges specified by the list of integers.
           For example, passing <u>5,15</u> will group sites into two types of gVCF blocks, the first with minimum
           per-sample DP from the interval [5,15) and the latter with minimum depth 15 or more. In this example,
           sites with minimum per-sample depth less than 5 will be printed as separate records, outside of gVCF
           blocks.

       <b>-i,</b> <b>--insert-missed</b> <u>INT</u>
           output also sites missed by mpileup but present in <b>-T,</b> <b>--targets-file</b>.

       <b>-M,</b> <b>--keep-masked-ref</b>
           output sites where REF allele is N

       <b>-V,</b> <b>--skip-variants</b> <u>snps</u>|<u>indels</u>
           skip indel/SNP sites

       <b>-v,</b> <b>--variants-only</b>
           output variant sites only

   <b>Consensus/variant</b> <b>calling</b> <b>options:</b>
       <b>-c,</b> <b>--consensus-caller</b>
           the original <b>samtools</b>/<b>bcftools</b> calling method (conflicts with <b>-m</b>)

       <b>-C,</b> <b>--constrain</b> <u>alleles</u>|<u>trio</u>

           <u>alleles</u>
               call genotypes given alleles. See also <b>-T,</b> <b>--targets-file</b>.

           <u>trio</u>
               call genotypes given the father-mother-child constraint. See also <b>-s,</b> <b>--samples</b> and <b>-n,</b>
               <b>--novel-rate</b>.

       <b>-m,</b> <b>--multiallelic-caller</b>
           alternative model for multiallelic and rare-variant calling designed to overcome known limitations in
           <b>-c</b> calling model (conflicts with <b>-c</b>)

       <b>-n,</b> <b>--novel-rate</b> <u>float</u>[,...]
           likelihood of novel mutation for constrained <b>-C</b> <u>trio</u> calling. The trio genotype calling maximizes
           likelihood of a particular combination of genotypes for father, mother and the child P(F=i,M=j,C=k) =
           P(unconstrained) * Pn + P(constrained) * (1-Pn). By providing three values, the mutation rate Pn is
           set explicitly for SNPs, deletions and insertions, respectively. If two values are given, the first
           is interpreted as the mutation rate of SNPs and the second is used to calculate the mutation rate of
           indels according to their length as Pn=<u>float</u>*exp(-a-b*len), where a=22.8689, b=0.2994 for insertions
           and a=21.9313, b=0.2856 for deletions [pubmed:23975140]. If only one value is given, the same
           mutation rate Pn is used for SNPs and indels.

       <b>-p,</b> <b>--pval-threshold</b> <u>float</u>
           with <b>-c</b>, accept variant if P(ref|D) &lt; <u>float</u>.

       <b>-P,</b> <b>--prior</b> <u>float</u>
           expected substitution rate, or 0 to disable the prior. Only with <b>-m</b>.

       <b>-t,</b> <b>--targets</b> <u>file</u>|<u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-X,</b> <b>--chromosome-X</b>
           haploid output for male samples (requires PED file with <b>-s</b>)

       <b>-Y,</b> <b>--chromosome-Y</b>
           haploid output for males and skips females (requires PED file with <b>-s</b>)

   <b>bcftools</b> <b>cnv</b> <u><b>[OPTIONS]</b></u> <u><b>FILE</b></u>
       Copy number variation caller, requires a VCF annotated with the Illumina’s B-allele frequency (BAF) and
       Log R Ratio intensity (LRR) values. The HMM considers the following copy number states: CN 2 (normal), 1
       (single-copy loss), 0 (complete loss), 3 (single-copy gain).

   <b>General</b> <b>Options:</b>
       <b>-c,</b> <b>--control-sample</b> <u>string</u>
           optional control sample name. If given, pairwise calling is performed and the <b>-P</b>  option can be used

       <b>-f,</b> <b>--AF-file</b> <u>file</u>
           read allele frequencies from  a tab-delimited file with the columns CHR,POS,REF,ALT,AF

       <b>-o,</b> <b>--output-dir</b> <u>path</u>
           output directory

       <b>-p,</b> <b>--plot-threshold</b> <u>float</u>
           call <b>matplotlib</b> to produce plots for chromosomes with quality at least <u>float</u>, useful for visual
           inspection of the calls. With <b>-p</b> <b>0</b>, plots for all chromosomes will be generated. If not given, a
           <b>matplotlib</b> script will be created but not called.

       <b>-r,</b> <b>--regions</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-R,</b> <b>--regions-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--regions-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-s,</b> <b>--query-sample</b> <u>string</u>
           query sample name

       <b>-t,</b> <b>--targets</b> <u>LIST</u>
           see <b>Common</b> <b>Options</b>

       <b>-T,</b> <b>--targets-file</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>--targets-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

   <b>HMM</b> <b>Options:</b>
       <b>-a,</b> <b>--aberrant</b> <u>float</u>[,<u>float</u>]
           fraction of aberrant cells in query and control. The hallmark of duplications and contaminations is
           the BAF value of heterozygous markers which is dependent on the fraction of aberrant cells.
           Sensitivity to smaller fractions of cells can be increased by setting <b>-a</b> to a lower value. Note
           however, that this comes at the cost of increased false discovery rate.

       <b>-b,</b> <b>--BAF-weight</b> <u>float</u>
           relative contribution from BAF

       <b>-d,</b> <b>--BAF-dev</b> <u>float</u>[,<u>float</u>]
           expected BAF deviation in query and control, i.e. the noise observed in the data.

       <b>-e,</b> <b>--err-prob</b> <u>float</u>
           uniform error probability

       <b>-l,</b> <b>--LRR-weight</b> <u>float</u>
           relative contribution from LRR. With noisy data, this option can have big effect on the number of
           calls produced. In truly random noise (such as in simulated data), the value should be set high
           (1.0), but in the presence of systematic noise when LRR are not informative, lower values result in
           cleaner calls (0.2).

       <b>-L,</b> <b>--LRR-smooth-win</b> <u>int</u>
           reduce LRR noise by applying moving average given this window size

       <b>-O,</b> <b>--optimize</b> <u>float</u>
           iteratively estimate the fraction of aberrant cells, down to the given fraction. Lowering this value
           from the default 1.0 to say, 0.3, can help discover more events but also increases noise

       <b>-P,</b> <b>--same-prob</b> <u>float</u>
           the prior probability of the query and the control sample being the same. Setting to 0 calls both
           independently, setting to 1 forces the same copy number state in both.

       <b>-x,</b> <b>--xy-prob</b> <u>float</u>
           the HMM probability of transition to another copy number state. Increasing this values leads to
           smaller and more frequent calls.

   <b>bcftools</b> <b>concat</b> <u><b>[OPTIONS]</b></u> <u><b>FILE1</b></u> <u><b>FILE2</b></u> <b>[...]</b>
       Concatenate or combine VCF/BCF files. All source files must have the same sample columns appearing in the
       same order. Can be used, for example, to concatenate chromosome VCFs into one VCF, or combine a SNP VCF
       and an indel VCF into one. The input files must be sorted by chr and position. The files must be given in
       the correct order to produce sorted VCF on output unless the <b>-a,</b> <b>--allow-overlaps</b> option is specified.
       With the --naive option, the files are concatenated without being recompressed, which is very fast..

       <b>-a,</b> <b>--allow-overlaps</b>
           First coordinate of the next file can precede last record of the current file.

       <b>-c,</b> <b>--compact-PS</b>
           Do not output PS tag at each site, only at the start of a new phase set block.

       <b>-d,</b> <b>--rm-dups</b> <u>snps</u>|<u>indels</u>|<u>both</u>|<u>all</u>|<u>exact</u>
           Output duplicate records of specified type present in multiple files only once. Note that records
           duplicate within one file are not removed with this option, for that use <b>bcftools</b> <b>norm</b> <b>-d</b> instead.
           In other words, the default behavior of the program is similar to unix "cat" in that when two files
           contain a record with the same position, that position will appear twice on output. With <b>-d</b>, every
           line that finds a matching record in another file will be printed only once.
           Requires <b>-a,</b> <b>--allow-overlaps</b>.

       <b>-D,</b> <b>--remove-duplicates</b>
           Alias for <b>-d</b> <b>exact</b>

       <b>-f,</b> <b>--file-list</b> <u>FILE</u>
           Read file names from <u>FILE</u>, one file name per line.

       <b>-l,</b> <b>--ligate</b>
           Ligate phased VCFs by matching phase at overlapping haplotypes. Note that the option is intended for
           VCFs with perfect overlap, sites in overlapping regions present in one but missing in the other are
           dropped.

       <b>--ligate-force</b>
           Keep all sites and ligate even non-overlapping chunks and chunks with imperfect overlap

       <b>--ligate-warn</b>
           Drop sites in imperfect overlaps

       <b>--no-version</b>
           see <b>Common</b> <b>Options</b>

       <b>-n,</b> <b>--naive</b>
           Concatenate VCF or BCF files without recompression. This is very fast but requires that all files are
           of the same type (all VCF or all BCF) and have the same headers. This is because all tags and
           chromosome names in the BCF body rely on the order of the contig and tag definitions in the header. A
           header check compatibility is performed and the program throws an error if it is not safe to use the
           option.

       <b>--naive-force</b>
           Same as --naive, but header compatibility is not checked. Dangerous, use with caution.

       <b>-o,</b> <b>--output</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>-O,</b> <b>--output-type</b> <u>b</u>|<u>u</u>|<u>z</u>|<u>v</u>[0-9]
           see <b>Common</b> <b>Options</b>

       <b>-q,</b> <b>--min-PQ</b> <u>INT</u>
           Break phase set if phasing quality is lower than <u>INT</u>

       <b>-r,</b> <b>--regions</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>. Requires <b>-a,</b> <b>--allow-overlaps</b>.

       <b>-R,</b> <b>--regions-file</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>. Requires <b>-a,</b> <b>--allow-overlaps</b>.

       <b>--regions-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>--threads</b> <u>INT</u>
           see <b>Common</b> <b>Options</b>

       <b>-W</b>[<u>FMT</u>]<b>,</b> <b>-W</b>[=<u>FMT</u>]<b>,</b> <b>--write-index</b>[=<u>FMT</u>]
           Automatically index the output file. <u>FMT</u> is optional and can be one of "tbi" or "csi" depending on
           output file format.

   <b>bcftools</b> <b>consensus</b> <u><b>[OPTIONS]</b></u> <u><b>FILE</b></u>
       Create consensus sequence by applying VCF variants to a reference fasta file. By default, the program
       will apply all ALT variants to the reference fasta to obtain the consensus sequence. Using the <b>--sample</b>
       (and, optionally, <b>--haplotype</b>) option will apply genotype (haplotype) calls from FORMAT/GT. Note that the
       program does not act as a primitive variant caller and ignores allelic depth information, such as INFO/AD
       or FORMAT/AD. For that, consider using the <b>setGT</b> plugin.

       <b>-a,</b> <b>--absent</b> <u>CHAR</u>
           replace positions absent from VCF with CHAR

       <b>-c,</b> <b>--chain</b> <u>FILE</u>
           write a chain file for liftover

       <b>-e,</b> <b>--exclude</b> <u>EXPRESSION</u>
           exclude sites for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.

       <b>-f,</b> <b>--fasta-ref</b> <u>FILE</u>
           reference sequence in fasta format

       <b>-H,</b> <b>--haplotype</b> N|<u>R</u>|<u>A</u>|<u>I</u>|<u>LR</u>|<u>LA</u>|<u>SR</u>|<u>SA</u>|<u>NpIu</u>
           choose which allele from the FORMAT/GT field to use (the codes are case-insensitive):

           <u>N</u>
               N={1,2,3,...}, the allele index within the genotype, regardless of phasing

           <u>R</u>
               the REF allele (in heterozygous genotypes)

           <u>A</u>
               the ALT allele (in heterozygous genotypes)

           <u>I</u>
               IUPAC code for all genotypes

           <u>LR,</u> <u>LA</u>
               the longer allele. If both have the same length, use the REF allele (LR), or the ALT allele  (LA)

           <u>SR,</u> <u>SA</u>
               the shorter allele. If both have the same length, use the REF allele (SR), or the ALT allele
               (SA)

           <u>NpIu</u>
               N={1,2,3,...}, the allele index within genotype for phased genotypes and IUPAC code for unphased
               genotypes. For example, <u>1pIu</u> or <u>2pIu</u>

       Note that the <b>-H,</b> <b>--haplotype</b> option requires the <b>-s,</b> <b>--samples</b> option, unless exactly one sample is
       present in the VCF

       <b>-i,</b> <b>--include</b> <u>EXPRESSION</u>
           include only sites for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.

       <b>-I,</b> <b>--iupac-codes</b>
           output variants in the form of IUPAC ambiguity codes determined from FORMAT/GT fields. By default all
           samples are used and can be subset with <b>-s,</b> <b>--samples</b> and <b>-S,</b> <b>--samples-file</b>. Use <b>-s</b> <b>-</b> to ignore
           samples and use only the REF and ALT columns. NOTE: prior to version 1.17 the IUPAC codes were
           determined solely from REF,ALT columns and sample genotypes were not considered.

       <b>--mark-del</b> <u>CHAR</u>
           instead of removing sequence, insert character CHAR for deletions

       <b>--mark-ins</b> <u>uc</u>|<u>lc</u>|<u>CHAR</u>
           highlight inserted sequence in uppercase (uc), lowercase (lc), or a provided character CHAR, leaving
           the rest of the sequence as is

       <b>--mark-snv</b> <u>uc</u>|<u>lc</u>
           highlight substitutions in uppercase (uc), lowercase (lc), or a provided character CHAR, leaving the
           rest of the sequence as is

       <b>-m,</b> <b>--mask</b> <u>FILE</u>
           BED file or TAB file with regions to be replaced with N (the default) or as specified by the next
           <b>--mask-with</b> option. See discussion of <b>--regions-file</b> in <b>Common</b> <b>Options</b> for file format details.

       <b>--mask-with</b> <u>CHAR</u>|<u>lc</u>|<u>uc</u>
           replace sequence from <b>--mask</b> with CHAR, skipping overlapping variants, or change to lowercase (lc) or
           uppercase (uc)

       <b>-M,</b> <b>--missing</b> <u>CHAR</u>
           instead of skipping the missing genotypes, output the character CHAR (e.g. "?")

       <b>-o,</b> <b>--output</b> <u>FILE</u>
           write output to a file

       <b>--regions-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           how to treat VCF variants overlapping the target region in the fasta file: see <b>Common</b> <b>Options</b>

       <b>-s,</b> <b>--samples</b> <u>LIST</u>
           apply variants of the listed samples. See also the option <b>-I,</b> <b>--iupac-codes</b>

       <b>-S,</b> <b>--samples-file</b> <u>FILE</u>
           apply variants of the samples listed in the file. See also the option <b>-I,</b> <b>--iupac-codes</b>

       <b>Examples:</b>

               # Apply variants present in sample "NA001", output IUPAC codes for hets
               bcftools consensus -i -s NA001 -f in.fa in.vcf.gz &gt; out.fa

               # Create consensus for one region. The fasta header lines are then expected
               # in the form "&gt;chr:from-to". Ignore samples and consider only the REF and ALT columns
               samtools faidx ref.fa 8:11870-11890 | bcftools consensus -s - in.vcf.gz -o out.fa

               # For more examples see <a href="http://samtools.github.io/bcftools/howtos/consensus-sequence.html">http://samtools.github.io/bcftools/howtos/consensus-sequence.html</a>

       <b>Notes:</b>
           Masking options are applied in the following order

            1. mask regions with <b>--mask-with</b> character if <b>--mask</b> is given. All overlapping VCF variants are
               ignored

            2. replace sequence not mentioned in the VCF with the requested character if <b>--absent</b> is given

            3. finally apply <b>--mark-del</b>, <b>--mark-ins</b>, <b>--mark-snv</b> masks

   <b>bcftools</b> <b>convert</b> <u><b>[OPTIONS]</b></u> <u><b>FILE</b></u>
   <b>VCF</b> <b>input</b> <b>options:</b>
       <b>-e,</b> <b>--exclude</b> <u>EXPRESSION</u>
           exclude sites for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.

       <b>-i,</b> <b>--include</b> <u>EXPRESSION</u>
           include only sites for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.

       <b>-r,</b> <b>--regions</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-R,</b> <b>--regions-file</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>--regions-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-s,</b> <b>--samples</b> <u>LIST</u>
           see <b>Common</b> <b>Options</b>

       <b>-S,</b> <b>--samples-file</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>-t,</b> <b>--targets</b> <u>LIST</u>
           see <b>Common</b> <b>Options</b>

       <b>-T,</b> <b>--targets-file</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>--targets-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-W</b>[<u>FMT</u>]<b>,</b> <b>-W</b>[=<u>FMT</u>]<b>,</b> <b>--write-index</b>[=<u>FMT</u>]
           Automatically index the output file. <u>FMT</u> is optional and can be one of "tbi" or "csi" depending on
           output file format.

   <b>VCF</b> <b>output</b> <b>options:</b>
       <b>--no-version</b>
           see <b>Common</b> <b>Options</b>

       <b>-o,</b> <b>--output</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>-O,</b> <b>--output-type</b> <u>b</u>|<u>u</u>|<u>z</u>|<u>v</u>[0-9]
           see <b>Common</b> <b>Options</b>

       <b>--threads</b> <u>INT</u>
           see <b>Common</b> <b>Options</b>

   <b>GEN/SAMPLE</b> <b>conversion:</b>
       <b>-G,</b> <b>--gensample2vcf</b> <u>prefix</u> or <u>gen-file</u>,<u>sample-file</u>
           convert IMPUTE2 output to VCF. One of the ID columns ("SNP ID" or "rsID" in
           https://www.cog-genomics.org/plink/2.0/formats#gen) must be of the form "CHROM:POS_REF_ALT" to detect
           possible strand swaps.
           When the <b>--vcf-ids</b> option is given, the other column (autodetected) is used to fill the ID column of
           the VCF.
           See also <b>-g</b> and <b>--3N6</b> options.

       <b>-g,</b> <b>--gensample</b> <u>prefix</u> or <u>gen-file</u>,<u>sample-file</u>
           convert from VCF to gen/sample format used by IMPUTE2 and SHAPEIT. The columns of .gen file format
           are ID1,ID2,POS,A,B followed by three genotype probabilities P(AA), P(AB), P(BB) for each sample. In
           order to prevent strand swaps, the program uses IDs of the form "CHROM:POS_REF_ALT". When the
           <b>--vcf-ids</b> option is given, the second column is set to match the ID column of the VCF.
           See also <b>-G</b> and <b>--3N6</b> options.
           The file .gen and .sample file format are:

             .gen (with --3N6 --vcf-ids)
             ---------------------------
             chr1 1:111485207_G_A rsID1 111485207 G A 0 1 0 0 1 0
             chr1 1:111494194_C_T rsID2 111494194 C T 0 1 0 0 0 1

             .gen (with --vcf-ids)
             ---------------------------
             1:111485207_G_A rsID1 111485207 G A 0 1 0 0 1 0
             1:111494194_C_T rsID2 111494194 C T 0 1 0 0 0 1

             .gen (the default)
             ------------------------------
             1:111485207_G_A 1:111485207_G_A 111485207 G A 0 1 0 0 1 0
             1:111494194_C_T 1:111494194_C_T 111494194 C T 0 1 0 0 0 1

             .sample
             -------
             ID_1 ID_2 missing
             0 0 0
             sample1 sample1 0
             sample2 sample2 0

       <b>--3N6</b>
           Expect/Create files in the 3*N+6 column format. This is the new .gen file format with the first
           column containing the chromosome name, see https://www.cog-genomics.org/plink/2.0/formats#gen

       <b>--tag</b> <u>STRING</u>
           tag to take values for .gen file: GT,PL,GL,GP

       <b>--sex</b> <u>FILE</u>
           output sex column in the sample file. The FILE format is

               MaleSample    M
               FemaleSample  F

       <b>--vcf-ids</b>
           output VCF IDs in the second column instead of CHROM:POS_REF_ALT

   <b>gVCF</b> <b>conversion:</b>
       <b>--gvcf2vcf</b>
           convert gVCF to VCF, expanding REF blocks into sites. Note that the <b>-i</b> and <b>-e</b> options work
           differently with this switch. In this situation the filtering expressions define which sites should
           be expanded and which sites should be left unmodified, but all sites are printed on output. In order
           to drop sites, stream first through <b>bcftools</b> <b>view</b>.

       <b>-f,</b> <b>--fasta-ref</b> <u>file</u>
           reference sequence in fasta format. Must be indexed with samtools faidx

   <b>HAP/SAMPLE</b> <b>conversion:</b>
       <b>--hapsample2vcf</b> <u>prefix</u> or <u>hap-file</u>,<u>sample-file</u>
           convert from hap/sample format to VCF. The columns of .hap file are similar to .gen file above, but
           there are only two haplotype columns per sample. Note that the first or the second column of the .hap
           file is expected to be in the form "CHR:POS_REF_ALT[_END]", with the _END being optional for defining
           the INFO/END tag when ALT is a symbolic allele. For example:

             .hap (with --vcf-ids)
             ---------------------
             1:111485207_G_A rsID1 111485207 G A 0 1 0 0
             1:111495231_A_&lt;DEL&gt;_111495784 rsID3 111495231 A &lt;DEL&gt; 0 0 1 0

             .hap (the default)
             ------------------
             1 1:111485207_G_A 111485207 G A 0 1 0 0
             1 1:111495231_A_&lt;DEL&gt;_111495784 111495231 A &lt;DEL&gt; 0 0 1 0

       <b>--hapsample</b> <u>prefix</u> or <u>hap-file</u>,<u>sample-file</u>
           convert from VCF to hap/sample format used by IMPUTE2 and SHAPEIT. The columns of .hap file begin
           with ID,RSID,POS,REF,ALT. In order to prevent strand swaps, the program uses IDs of the form
           "CHROM:POS_REF_ALT".

       <b>--haploid2diploid</b>
           with <b>-h</b> option converts haploid genotypes to homozygous diploid genotypes. For example, the program
           will print <u>0</u> <u>0</u> instead of the default <u>0</u> <u>-</u>. This is useful for programs which do not handle haploid
           genotypes correctly.

       <b>--sex</b> <u>FILE</u>
           output sex column in the sample file. The FILE format is

               MaleSample    M
               FemaleSample  F

       <b>--vcf-ids</b>
           the second column of the .hap file holds the VCF ids, the first column is of the form
           "CHR:POS_REF_ALT[_END]". Without the option, the format follows
           https://www.cog-genomics.org/plink/2.0/formats#haps with ids (the second column) of the form
           "CHR:POS_REF_ALT[_END]"

   <b>HAP/LEGEND/SAMPLE</b> <b>conversion:</b>
       <b>-H,</b> <b>--haplegendsample2vcf</b> <u>prefix</u> or <u>hap-file</u>,<u>legend-file</u>,<u>sample-file</u>
           convert from hap/legend/sample format used by IMPUTE2 to VCF. See also <b>-h,</b> <b>--hapslegendsample</b> below.

       <b>-h,</b> <b>--haplegendsample</b> <u>prefix</u> or <u>hap-file</u>,<u>legend-file</u>,<u>sample-file</u>
           convert from VCF to hap/legend/sample format used by IMPUTE2 and SHAPEIT. The columns of .legend file
           ID,POS,REF,ALT. In order to prevent strand swaps, the program uses IDs of the form
           "CHROM:POS_REF_ALT". The .sample file is quite basic at the moment with columns for population, group
           and sex expected to be edited by the user. For example:

             .hap
             -----
             0 1 0 0 1 0
             0 1 0 0 0 1

             .legend
             -------
             id position a0 a1
             1:111485207_G_A 111485207 G A
             1:111494194_C_T 111494194 C T

             .sample
             -------
             sample population group sex
             sample1 sample1 sample1 2
             sample2 sample2 sample2 2

       <b>--haploid2diploid</b>
           with <b>-h</b> option converts haploid genotypes to homozygous diploid genotypes. For example, the program
           will print <u>0</u> <u>0</u> instead of the default <u>0</u> <u>-</u>. This is useful for programs which do not handle haploid
           genotypes correctly.

       <b>--sex</b> <u>FILE</u>
           output sex column in the sample file. The FILE format is

               MaleSample    M
               FemaleSample  F

       <b>--vcf-ids</b>
           output VCF IDs instead of "CHROM:POS_REF_ALT". Note that this option can be used with
           <b>--haplegendsample</b> but not with <b>--haplegendsample2vcf</b>.

   <b>TSV</b> <b>conversion:</b>
       <b>--tsv2vcf</b> <u>file</u>
           convert from TSV (tab-separated values) format (such as generated by 23andMe) to VCF. The input file
           fields can be tab- or space- delimited

       <b>-c,</b> <b>--columns</b> <u>list</u>
           comma-separated list of fields in the input file. In the current version, the fields CHROM, POS, ID,
           and AA or REF, ALT are expected and can appear in arbitrary order. Columns which should be ignored in
           the input file can be indicated by "-". The AA field lists alleles on the forward reference strand,
           for example "CC" or "CT" for diploid genotypes or "C" for haploid genotypes (sex chromosomes).
           Insertions and deletions are supported only with REF and ALT but not with AA. Missing data can be
           indicated with "--" or ".".

       <b>-f,</b> <b>--fasta-ref</b> <u>file</u>
           reference sequence in fasta format. Must be indexed with samtools faidx

       <b>-s,</b> <b>--samples</b> <u>LIST</u>
           list of sample names. See <b>Common</b> <b>Options</b>

       <b>-S,</b> <b>--samples-file</b> <u>FILE</u>
           file of sample names. See <b>Common</b> <b>Options</b>

       <b>Example:</b>

           # Convert 23andme results into VCF
           bcftools convert -c ID,CHROM,POS,AA -s SampleName -f 23andme-ref.fa --tsv2vcf 23andme.txt -o out.vcf.gz

           # Convert tab-delimited file into a sites-only VCF (no genotypes), in this example first column to be ignored
           bcftools convert -c -,CHROM,POS,REF,ALT -f ref.fa --tsv2vcf calls.txt -o out.bcf

   <b>bcftools</b> <b>csq</b> <u><b>[OPTIONS]</b></u> <u><b>FILE</b></u>
       Haplotype aware consequence predictor which correctly handles combined variants such as MNPs split over
       multiple VCF records, SNPs separated by an intron (but adjacent in the spliced transcript) or nearby
       frame-shifting indels which in combination in fact are not frame-shifting.

       The output VCF is annotated with INFO/BCSQ and FORMAT/BCSQ tag (configurable with the <b>-c</b> option). The
       latter is a bitmask of indexes to INFO/BCSQ, with interleaved haplotypes. See the usage examples below
       for using the %TBCSQ converter in <b>query</b> for extracting a more human readable form from this bitmask. The
       construction of the bitmask limits the number of consequences that can be referenced per sample in the
       FORMAT/BCSQ tags. By default this is 15, but if more are required, see the <b>--ncsq</b> option.

       Note that the program annotates only records with a functional consequence and intergenic regions will
       pass through unchanged.

       The program requires on input a VCF/BCF file, the reference genome in fasta format (<b>--fasta-ref</b>) and
       genomic features in the GFF3 format downloadable from the Ensembl website (<b>--gff-annot</b>), and outputs an
       annotated VCF/BCF file. Currently, only Ensembl GFF3 files are supported.

       By default, the input VCF should be phased. If phase is unknown, or only partially known, the <b>--phase</b>
       option can be used to indicate how to handle unphased data. Alternatively, haplotype aware calling can be
       turned off with the <b>--local-csq</b> option.

       If conflicting (overlapping) variants within one haplotype are detected, a warning will be emitted and
       predictions will be based on only the first variant in the analysis.

       Symbolic alleles are not supported. They will remain unannotated in the output VCF and are ignored for
       the prediction analysis.

       <b>-c,</b> <b>--custom-tag</b> <u>STRING</u>
           use this custom tag to store consequences rather than the default BCSQ tag

       <b>-B,</b> <b>--trim-protein-seq</b> <u>INT</u>
           abbreviate protein-changing predictions to maximum of INT aminoacids. For example, instead of writing
           the whole modified protein sequence with potentially hundreds of aminoacids, with <b>-B</b> <b>1</b> only an
           abbreviated version such as <u>25E..329&gt;25G..94</u> will be written.

       <b>--dump-gff</b> <u>FILE</u>
           dump the parsed GFF into a gzipped FILE. Intended for debugging purposes, shows how is the input GFF
           viewed by the program.

       <b>-e,</b> <b>--exclude</b> <u>EXPRESSION</u>
           exclude sites for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.

       <b>-f,</b> <b>--fasta-ref</b> <u>FILE</u>
           reference sequence in fasta format (required)

       <b>--force</b>
           run even if some sanity checks fail. Currently the option enables skipping transcripts in
           malformatted GFFs with incorrect phase

       <b>-g,</b> <b>--gff-annot</b> <u>FILE</u>
           GFF3 annotation file (required), such as <a href="http://ftp.ensembl.org/pub/current_gff3/homo_sapiens/">http://ftp.ensembl.org/pub/current_gff3/homo_sapiens/</a>. The
           script <b>gff2gff</b> can help with conversion from non-standard GFF formats. An example of a minimal
           working GFF file:

               # The program looks for "CDS", "exon", "three_prime_UTR" and "five_prime_UTR" lines,
               # looks up their parent transcript (determined from the "Parent=transcript:" attribute),
               # the gene (determined from the transcript's "Parent=gene:" attribute), and the biotype
               # (the most interesting is "protein_coding").
               #
               # Empty and commented lines are skipped, the following GFF columns are required
               #   1. chromosome
               #   2. IGNORED
               #   3. type (CDS, exon, three_prime_UTR, five_prime_UTR, gene, transcript, etc.)
               #   4. start of the feature (1-based)
               #   5. end of the feature (1-based)
               #   6. IGNORED
               #   7. strand (+ or -)
               #   8. phase (0, 1, 2 or .)
               #   9. semicolon-separated attributes (see below)
               #
               # Attributes required for
               #   gene lines:
               #   - ID=gene:&lt;gene_id&gt;
               #   - biotype=&lt;biotype&gt;
               #   - Name=&lt;gene_name&gt;      [optional]
               #
               #   transcript lines:
               #   - ID=transcript:&lt;transcript_id&gt;
               #   - Parent=gene:&lt;gene_id&gt;
               #   - biotype=&lt;biotype&gt;
               #
               #   other lines (CDS, exon, five_prime_UTR, three_prime_UTR):
               #   - Parent=transcript:&lt;transcript_id&gt;
               #
               # Supported biotypes:
               #   - see the function gff_parse_biotype() in bcftools/csq.c

               1   ignored_field  gene            21  2148  . -   . ID=gene:GeneId;biotype=protein_coding;Name=GeneName
               1   ignored_field  transcript      21  2148  . -   . ID=transcript:TranscriptId;Parent=gene:GeneId;biotype=protein_coding
               1   ignored_field  three_prime_UTR 21  2054  . -   . Parent=transcript:TranscriptId
               1   ignored_field  exon            21  2148  . -   . Parent=transcript:TranscriptId
               1   ignored_field  CDS             21  2148  . -   1   Parent=transcript:TranscriptId
               1   ignored_field  five_prime_UTR  210 2148  . -   . Parent=transcript:TranscriptId

       <b>-i,</b> <b>--include</b> <u>EXPRESSION</u>
           include only sites for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.

       <b>-l,</b> <b>--local-csq</b>
           switch off haplotype-aware calling, run localized predictions considering only one VCF record at a
           time

       <b>-n,</b> <b>--ncsq</b> <u>INT</u>
           maximum number of per-haplotype consequences to consider for each site. The INFO/BCSQ column includes
           all consequences, but only the first <u>INT</u> will be referenced by the FORMAT/BCSQ fields. The default
           value is 15 which corresponds to one 32-bit integer per diploid sample, after accounting for values
           reserved by the BCF specification. Note that increasing the value leads to increased size of the
           output BCF.

       <b>--no-version</b>
           see <b>Common</b> <b>Options</b>

       <b>-o,</b> <b>--output</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>-O,</b> <b>--output-type</b> <u>t</u>|<u>b</u>|<u>u</u>|<u>z</u>|<u>v</u>[0-9]
           see <b>Common</b> <b>Options</b>. In addition, a custom tab-delimited plain text output can be printed (<u>t</u>).

       <b>-p,</b> <b>--phase</b> <u>a</u>|<u>m</u>|<u>r</u>|<u>R</u>|<u>s</u>
           how to handle unphased heterozygous genotypes:

           <u>a</u>
               take GTs as is, create haplotypes regardless of phase (0/1 → 0|1)

           <u>m</u>
               merge all GTs into a single haplotype (0/1 → 1, 1/2 → 1)

           <u>r</u>
               require phased GTs, throw an error on unphased heterozygous GTs

           <u>R</u>
               create non-reference haplotypes if possible (0/1 → 1|1, 1/2 → 1|2)

           <u>s</u>
               skip unphased heterozygous GTs

       <b>-q,</b> <b>--quiet</b>
           suppress warning messages

       <b>-r,</b> <b>--regions</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-R,</b> <b>--regions-file</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>--regions-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-s,</b> <b>--samples</b> <u>LIST</u>
           samples to include or "-" to apply all variants and ignore samples

       <b>-S,</b> <b>--samples-file</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>-t,</b> <b>--targets</b> <u>LIST</u>
           see <b>Common</b> <b>Options</b>

       <b>-T,</b> <b>--targets-file</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>--targets-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>--unify-chr-names</b> <u>0</u>|<u>1</u>
           Automatically detect and unify chromosome naming conventions in the GFF, fasta and VCF, such as
           "chrX" vs "X". The chromosome names in the output VCF will match that of the input VCF. The default
           is to attempt the automatic translation.

       <b>-v,</b> <b>--verbose</b> <u>INT</u>
           verbosity level (0-2)

       <b>-W</b>[<u>FMT</u>]<b>,</b> <b>-W</b>[=<u>FMT</u>]<b>,</b> <b>--write-index</b>[=<u>FMT</u>]
           Automatically index the output file. <u>FMT</u> is optional and can be one of "tbi" or "csi" depending on
           output file format.

       <b>Examples:</b>

               # Basic usage
               bcftools csq -f hs37d5.fa -g Homo_sapiens.GRCh37.82.gff3.gz in.vcf -Ob -o out.bcf

               # Extract the translated haplotype consequences. The following TBCSQ variations
               # are recognised:
               #   %TBCSQ    .. print consequences in all haplotypes in separate columns
               #   %TBCSQ{0} .. print the first haplotype only
               #   %TBCSQ{1} .. print the second haplotype only
               #   %TBCSQ{*} .. print a list of unique consequences present in either haplotype
               bcftools query -f'[%CHROM\t%POS\t%SAMPLE\t%TBCSQ\n]' out.bcf

       <b>Examples</b> <b>of</b> <b>BCSQ</b> <b>annotation:</b>

               # Two separate VCF records at positions 2:122106101 and 2:122106102
               # change the same codon. This UV-induced C&gt;T dinucleotide mutation
               # has been annotated fully at the position 2:122106101 with
               #   - consequence type
               #   - gene name
               #   - ensembl transcript ID
               #   - coding strand (+ fwd, - rev)
               #   - amino acid position (in the coding strand orientation)
               #   - list of corresponding VCF variants
               # The annotation at the second position gives the position of the full
               # annotation
               BCSQ=missense|CLASP1|ENST00000545861|-|1174P&gt;1174L|122106101G&gt;A+122106102G&gt;A
               BCSQ=@122106101

               # A frame-restoring combination of two frameshift insertions C&gt;CG and T&gt;TGG
               BCSQ=@46115084
               BCSQ=inframe_insertion|COPZ2|ENST00000006101|-|18AGRGP&gt;18AQAGGP|46115072C&gt;CG+46115084T&gt;TGG

               # Stop gained variant
               BCSQ=stop_gained|C2orf83|ENST00000264387|-|141W&gt;141*|228476140C&gt;T

               # The consequence type of a variant downstream from a stop are prefixed with *
               BCSQ=*missense|PER3|ENST00000361923|+|1028M&gt;1028T|7890117T&gt;C

       <b>Supported</b> <b>consequence</b> <b>types</b>

           3_prime_utr
           5_prime_utr
           coding_sequence
           feature_elongation
           frameshift
           inframe_altering
           inframe_deletion
           inframe_insertion
           intergenic
           intron
           missense
           non_coding
           splice_acceptor
           splice_donor
           splice_region
           start_lost
           start_retained
           stop_gained
           stop_lost
           stop_retained
           synonymous

       See also https://ensembl.org/info/genome/variation/prediction/predicted_data.html

   <b>bcftools</b> <b>filter</b> <u><b>[OPTIONS]</b></u> <u><b>FILE</b></u>
       Apply fixed-threshold filters.

       <b>-e,</b> <b>--exclude</b> <u>EXPRESSION</u>
           exclude sites for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.

       <b>-g,</b> <b>--SnpGap</b> <u>INT</u>[:'indel',<u>mnp</u>,<u>bnd</u>,<u>other</u>,<u>overlap</u>]
           filter SNPs within <u>INT</u> base pairs of an indel or other other variant type. The following example
           demonstrates the logic of <b>--SnpGap</b> <u>3</u> applied on a deletion and an insertion:

           The SNPs at positions 1 and 7 are filtered, positions 0 and 8 are not:
                    0123456789
               ref  .G.GT..G..
               del  .A.G-..A..
           Here the positions 1 and 6 are filtered, 0 and 7 are not:
                    0123-456789
               ref  .G.G-..G..
               ins  .A.GT..A..

       <b>-G,</b> <b>--IndelGap</b> <u>INT</u>
           filter clusters of indels separated by <u>INT</u> or fewer base pairs allowing only one to pass. The
           following example demonstrates the logic of <b>--IndelGap</b> <u>2</u> applied on a deletion and an insertion:

           The second indel is filtered:
                    012345678901
               ref  .GT.GT..GT..
               del  .G-.G-..G-..
           And similarly here, the second is filtered:
                    01 23 456 78
               ref  .A-.A-..A-..
               ins  .AT.AT..AT..

       <b>-i,</b> <b>--include</b> <u>EXPRESSION</u>
           include only sites for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.

       <b>--mask</b> [^]<u>REGION</u>
           Soft filter regions, prepepend "^" to negate. Requires <b>-s,</b> <b>--soft-filter</b>.

       <b>-M,</b> <b>--mask-file</b> [^]<u>FILE</u>
           Soft filter regions listed in a file, "^" to negate. Requires <b>-s,</b> <b>--soft-filter</b>.

       <b>--mask-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           Same as <b>--regions-overlap</b> but for <b>--mask/--mask-file</b>. See <b>Common</b> <b>Options</b>. [1]

       <b>-m,</b> <b>--mode</b> [<u>+x</u>]
           define behaviour at sites with existing FILTER annotations. The default mode replaces existing
           filters of failed sites with a new FILTER string while leaving sites which pass untouched when
           non-empty and setting to "PASS" when the FILTER string is absent. The "+" mode appends new FILTER
           strings of failed sites instead of replacing them. The "x" mode resets filters of sites which pass to
           "PASS". Modes "+" and "x" can both be set.

       <b>--no-version</b>
           see <b>Common</b> <b>Options</b>

       <b>-o,</b> <b>--output</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>-O,</b> <b>--output-type</b> <u>b</u>|<u>u</u>|<u>z</u>|<u>v</u>[0-9]
           see <b>Common</b> <b>Options</b>

       <b>-r,</b> <b>--regions</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-R,</b> <b>--regions-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--regions-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-s,</b> <b>--soft-filter</b> <u>STRING</u>|<u>+</u>
           annotate FILTER column with <u>STRING</u> or, with <u>+</u>, a unique filter name generated by the program
           ("Filter%d"). Applies to records that do not meet filter expression.

       <b>-S,</b> <b>--set-GTs</b> <u>.</u>|<u>0</u>
           set genotypes of failed samples to missing value (<u>.</u>) or reference allele (<u>0</u>)

       <b>-t,</b> <b>--targets</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-T,</b> <b>--targets-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--targets-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>--threads</b> <u>INT</u>
           see <b>Common</b> <b>Options</b>

       <b>-W</b>[<u>FMT</u>]<b>,</b> <b>-W</b>[=<u>FMT</u>]<b>,</b> <b>--write-index</b>[=<u>FMT</u>]
           Automatically index the output file. <u>FMT</u> is optional and can be one of "tbi" or "csi" depending on
           output file format.

   <b>bcftools</b> <b>gtcheck</b> <b>[</b><u><b>OPTIONS</b></u><b>]</b> <b>[-g</b> <u><b>genotypes.vcf.gz</b></u><b>]</b> <u><b>query.vcf.gz</b></u>
       Checks sample identity. The program can operate in two modes. If the <b>-g</b> option is given, the identity of
       samples from <u>query.vcf.gz</u> is checked against the samples in the <b>-g</b> file. Without the <b>-g</b> option,
       multi-sample cross-check of samples in <u>query.vcf.gz</u> is performed.

       Note that the interpretation of the discordance score depends on the options provided (specifically <b>-e</b>
       and <b>-u</b>) and on the available annotations (FORMAT/PL vs FORMAT/GT). The discordance score can be
       interpreted as the number of mismatching genotypes only if GT-vs-GT matching is performed.

       <b>--distinctive-sites</b> <u>NUM[,MEM[,DIR]]</u>
           Find sites that can distinguish between at least NUM sample pairs. If the number is smaller or equal
           to 1, it is interpreted as the fraction of pairs. The optional MEM string sets the maximum memory
           used for in-memory sorting and DIR is the temporary directory for external sorting. This option
           requires also <b>--pairs</b> to be given.

       <b>--dry-run</b>
           Stop after first record to estimate required time.

       <b>-e,</b> <b>--exclude</b> [<u>qry</u>|<u>gt</u>]:'EXPRESSION'
           Exclude sites from query file (<u>qry:</u>) or genotype file (<u>gt:</u>) for which <u>EXPRESSION</u> is true. For valid
           expressions see <b>EXPRESSIONS</b>.

       <b>-E,</b> <b>--error-probability</b> <u>INT</u>
           Interpret genotypes and genotype likelihoods probabilistically. The value of <u>INT</u> represents genotype
           quality when GT tag is used (e.g. Q=30 represents one error in 1,000 genotypes and Q=40 one error in
           10,000 genotypes) and is ignored when PL tag is used (in that case an arbitrary non-zero integer can
           be provided).

           If <b>-E</b> is set to 0, the discordance score can be interpreted as the number of mismatching genotypes,
           but only in the GT-vs-GT matching mode. See the <b>-u,</b> <b>--use</b> option below for additional notes and
           caveats.

           If performance is an issue, set <b>-E</b> <b>0</b> for faster run times but less accurate results.

           Note that in previous versions of bcftools (⇐1.18), this option used to be a smaller case <b>-e</b>. It
           changed to make room for the filtering option <b>-e,</b> <b>--exclude</b> to stay consistent across other commands.

       <b>-g,</b> <b>--genotypes</b> <u>FILE</u>
           VCF/BCF file with reference genotypes to compare against

       <b>-H,</b> <b>--homs-only</b>
           Homozygous genotypes only, useful with low coverage data (requires <b>-g,</b> <b>--genotypes</b>)

       <b>-i,</b> <b>--include</b> [<u>qry</u>|<u>gt</u>]:'EXPRESSION'
           Include sites from query file (<u>qry:</u>) or genotype file (<u>gt:</u>) for which <u>EXPRESSION</u> is true. For valid
           expressions see <b>EXPRESSIONS</b>.

       <b>--n-matches</b> <u>INT</u>
           Print only top INT matches for each sample, 0 for unlimited. Use negative value to sort by HWE
           probability rather than the number of discordant sites. Note that average score is used to determine
           the top matches, not absolute values.

       <b>--no-HWE-prob</b>
           Disable calculation of HWE probability to reduce memory requirements with comparisons between very
           large number of sample pairs.

       <b>-o,</b> <b>--output</b> <u>FILE</u>
           Write to <u>FILE</u> rather than to standard output, where it is written by default.

       <b>-O,</b> <b>--output-type</b> <u>t</u>|<u>z</u>
           Write a plain (<u>t</u>) or compressed (<u>z</u>) text tab-delimited output.

       <b>-p,</b> <b>--pairs</b> <u>LIST</u>
           A comma-separated list of sample pairs to compare. When the <b>-g</b> option is given, the first sample must
           be from the query file, the second from the <b>-g</b> file, third from the query file etc
           (qry,gt[,qry,gt..]). Without the <b>-g</b> option, the pairs are created the same way but both samples are
           from the query file (qry,qry[,qry,qry..])

       <b>-P,</b> <b>--pairs-file</b> <u>FILE</u>
           A file with tab-delimited sample pairs to compare. The first sample in the pair must come from the
           query file, the second from the genotypes file when <b>-g</b> is given

       <b>-r,</b> <b>--regions</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           Restrict to comma-separated list of regions, see <b>Common</b> <b>Options</b>

       *-R, --regions-file' <u>FILE</u>
           Restrict to regions listed in a file, see <b>Common</b> <b>Options</b>

       <b>--regions-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-s,</b> <b>--samples</b> [<u>qry</u>|<u>gt</u>]:'LIST': List of query samples or <b>-g</b> samples. If neither <b>-s</b> nor <b>-S</b> are given, all
       possible sample pair combinations are compared

       <b>-S,</b> <b>--samples-file</b> [<u>qry</u>|<u>gt</u>]:'FILE' File with the query or <b>-g</b> samples to compare. If neither <b>-s</b> nor <b>-S</b> are
       given, all possible sample pair combinations are compared

       <b>-t,</b> <b>--targets</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>-T,</b> <b>--targets-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--targets-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-u,</b> <b>--use</b> <u>TAG1</u>[,<u>TAG2</u>]
           specifies which tag to use in the query file (<u>TAG1</u>) and the <b>-g</b> (<u>TAG2</u>) file. By default, the PL tag is
           used in the query file and, when available, the GT tags in the <b>-g</b> file.

           Note that when the requested tag is not available, the program will attempt to use the other tag. The
           output includes the number of sites that were matched by the four possible modes (for example
           GT-vs-GT or GT-vs-PL).

       <b>Examples:</b>

              # Check discordance of all samples from B against all samples in A
              bcftools gtcheck -g A.bcf B.bcf

              # Limit comparisons to the given list of samples
              bcftools gtcheck -s gt:a1,a2,a3 -s qry:b1,b2 -g A.bcf B.bcf

              # Compare only two pairs a1,b1 and a1,b2
              bcftools gtcheck -p a1,b1,a1,b2 -g A.bcf B.bcf

   <b>bcftools</b> <b>head</b> <b>[</b><u><b>OPTIONS</b></u><b>]</b> <b>[</b><u><b>FILE</b></u><b>]</b>
       By default, prints all headers from the specified input file to standard output in VCF format. The input
       file may be in VCF or BCF format; if no <u>FILE</u> is specified, standard input will be read. With appropriate
       options, only some of the headers and/or additionally some of the variant records will be printed.

       The <b>bcftools</b> <b>head</b> command outputs VCF headers almost exactly as they appear in the input file: it may add
       a ##FILTER=&lt;ID=PASS&gt; header if not already present, but it never adds version or command line information
       itself.

   <b>Options:</b>
       <b>-h,</b> <b>--header</b> <u>INT</u>
           Display only the first <u>INT</u> header lines. By default, all header lines are displayed.

       <b>-n,</b> <b>--records</b> <u>INT</u>
           Also display the first <u>INT</u> variant records. By default, no variant records are displayed.

       <b>-s,</b> <b>--samples</b> <u>INT</u>
           Display the first <u>INT</u> variant records including the last #CHROM header line with samples. Running
           with <b>-s</b> <b>0</b> alone outputs the #CHROM header line only. Note that the list of samples, with each sample
           per line, can be obtained with bcftools query using the option <b>-l,</b> <b>--list-samples</b>.

   <b>bcftools</b> <b>index</b> <b>[</b><u><b>OPTIONS</b></u><b>]</b> <u><b>in.bcf</b></u><b>|</b><u><b>in.vcf.gz</b></u>
       Creates index for bgzip compressed VCF/BCF files for random access. CSI (coordinate-sorted index) is
       created by default. The CSI format supports indexing of chromosomes up to length 2^31. TBI (tabix index)
       index files, which support chromosome lengths up to 2^29, can be created by using the <u>-t/--tbi</u> option or
       using the <u>tabix</u> program packaged with htslib. When loading an index file, bcftools will try the CSI first
       and then the TBI.

   <b>Indexing</b> <b>options:</b>
       <b>-c,</b> <b>--csi</b>
           generate CSI-format index for VCF/BCF files [default]

       <b>-f,</b> <b>--force</b>
           overwrite index if it already exists

       <b>-m,</b> <b>--min-shift</b> <u>INT</u>
           <u>set</u> <u>minimal</u> <u>interval</u> <u>size</u> <u>for</u> <u>CSI</u> <u>indices</u> <u>to</u> <u>2^INT;</u> <u>default:</u> <u>14</u>

       <b>-o,</b> <b>--output</b> <u>FILE</u>
           <u>output</u> <u>file</u> <u>name.</u> <u>If</u> <u>not</u> <u>set,</u> <u>then</u> <u>the</u> <u>index</u> <u>will</u> <u>be</u> <u>created</u> <u>using</u> <u>the</u> <u>input</u> <u>file</u> <u>name</u> <u>plus</u> <u>a</u> <u>.csi</u> <u>or</u>
           <u>.tbi</u> <u>extension</u>

       <b>-t,</b> <b>--tbi</b>
           <u>generate</u> <u>TBI-format</u> <u>index</u> <u>for</u> <u>VCF</u> <u>files</u>

       <b>--threads</b> <u>INT</u>
           <u>see</u> <b>Common</b> <b>Options</b>

   <b>Stats</b> <b>options:</b>
       <b>-a,</b> <b>--all</b>
           Used in conjunction with <b>-s,</b> <b>--stats</b>, print per contig stats for all contigs, even those with zero
           records and those for which no stats are recorded in the index file (shown as <u>.</u>).

       <b>-n,</b> <b>--nrecords</b>
           print the number of records based on the CSI or TBI index files

       <b>-s,</b> <b>--stats</b>
           Print per contig stats based on the CSI or TBI index files. Output format is three tab-delimited
           columns listing the contig name, contig length (<u>.</u> if unknown) and number of records for the contig.
           Contigs with zero records are not printed by default.

   <b>bcftools</b> <b>isec</b> <b>[</b><u><b>OPTIONS</b></u><b>]</b> <u><b>A.vcf.gz</b></u> <u><b>B.vcf.gz</b></u> <b>[...]</b>
       Creates intersections, unions and complements of VCF files. Depending on the options, the program can
       output records from one (or more) files which have (or do not have) corresponding records with the same
       position in the other files.

       <b>-c,</b> <b>--collapse</b> <u>snps</u>|<u>indels</u>|<u>both</u>|<u>all</u>|<u>some</u>|<u>none</u>
           see <b>Common</b> <b>Options</b>

       <b>-C,</b> <b>--complement</b>
           output positions present only in the first file but missing in the others

       <b>-e,</b> <b>--exclude</b> <u>-</u>|<u>EXPRESSION</u>
           exclude sites for which <u>EXPRESSION</u> is true. If <b>-e</b> (or <b>-i</b>) appears only once, the same filtering
           expression will be applied to all input files. Otherwise, <b>-e</b> or <b>-i</b> must be given for each input file.
           To indicate that no filtering should be performed on a file, use "-" in place of <u>EXPRESSION</u>, as shown
           in the example below. For valid expressions see <b>EXPRESSIONS</b>.

       <b>-f,</b> <b>--apply-filters</b> <u>LIST</u>
           see <b>Common</b> <b>Options</b>

       <b>-i,</b> <b>--include</b> <u>EXPRESSION</u>
           include only sites for which <u>EXPRESSION</u> is true. See discussion of <b>-e,</b> <b>--exclude</b> above.

       <b>-f,</b> <b>--file-list</b> <u>FILE</u>
           Read file names from <u>FILE</u>, one file name per line.

       <b>-n,</b> <b>--nfiles</b> [+-=]<u>INT</u>|~<u>BITMAP</u>
           output positions present in this many (=), this many or more (+), this many or fewer (-), or the
           exact same (~) files

       <b>-o,</b> <b>--output</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>. When several files are being output, their names are controlled via <b>-p</b> instead.

       <b>-O,</b> <b>--output-type</b> <u>b</u>|<u>u</u>|<u>z</u>|<u>v</u>[0-9]
           see <b>Common</b> <b>Options</b>

       <b>-p,</b> <b>--prefix</b> <u>DIR</u>
           if given, subset each of the input files accordingly. See also <b>-w</b>.

       <b>-r,</b> <b>--regions</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-R,</b> <b>--regions-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--regions-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-t,</b> <b>--targets</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-T,</b> <b>--targets-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--targets-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-w,</b> <b>--write</b> <u>LIST</u>
           comma-separated list of input files to output given as 1-based indices. With <b>-p</b> and no <b>-w</b>, all files
           are written.

       <b>-W</b>[<u>FMT</u>]<b>,</b> <b>-W</b>[=<u>FMT</u>]<b>,</b> <b>--write-index</b>[=<u>FMT</u>]
           Automatically index the output file. <u>FMT</u> is optional and defaults to tbi for vcf.gz and csi for bcf.
           This is done automatically with the <b>-p</b> option if the output format is compressed.

   <b>Examples:</b>
       Create intersection and complements of two sets saving the output in dir/*

               bcftools isec -p dir A.vcf.gz B.vcf.gz

       Filter sites in A (require INFO/MAF&gt;=0.01) and B (require INFO/dbSNP) but not in C, and create an
       intersection, including only sites which appear in at least two of the files after filters have been
       applied

               bcftools isec -e'MAF&lt;0.01' -i'dbSNP=1' -e- A.vcf.gz B.vcf.gz C.vcf.gz -n +2 -p dir

       Extract and write records from A shared by both A and B using exact allele match

               bcftools isec -p dir -n=2 -w1 A.vcf.gz B.vcf.gz

       Extract records private to A or B comparing by position only

               bcftools isec -p dir -n-1 -c all A.vcf.gz B.vcf.gz

       Print a list of records which are present in A and B but not in C and D

               bcftools isec -n~1100 -c all A.vcf.gz B.vcf.gz C.vcf.gz D.vcf.gz

   <b>bcftools</b> <b>merge</b> <b>[</b><u><b>OPTIONS</b></u><b>]</b> <u><b>A.vcf.gz</b></u> <u><b>B.vcf.gz</b></u> <b>[...]</b>
       Merge multiple VCF/BCF files from non-overlapping sample sets to create one multi-sample file. For
       example, when merging file <u>A.vcf.gz</u> containing samples <u>S1</u>, <u>S2</u> and <u>S3</u> and file <u>B.vcf.gz</u> containing samples
       <u>S3</u> and <u>S4</u>, the output file will contain five samples named <u>S1</u>, <u>S2</u>, <u>S3</u>, <u>2:S3</u> and <u>S4</u>.

       Note that it is responsibility of the user to ensure that the sample names are unique across all files.
       If they are not, the program will exit with an error unless the option <b>--force-samples</b> is given. The
       sample names can be also given explicitly using the <b>--print-header</b> and <b>--use-header</b> options.

       Note that only records from different files can be merged, never from the same file. For "vertical" merge
       take a look at <b>bcftools</b> <b>concat</b> or <b>bcftools</b> <b>norm</b> <b>-m</b> instead.

       <b>--force-no-index</b>
           synonymous to <b>--no-index</b>

       <b>--force-samples</b>
           if the merged files contain duplicate samples names, proceed anyway. Duplicate sample names will be
           resolved by prepending the index of the file as it appeared on the command line to the conflicting
           sample name (see <u>2:S3</u> in the above example).

       <b>--force-single</b>
           run even if only one file is given on input

       <b>--print-header</b>
           print only merged header and exit

       <b>--use-header</b> <u>FILE</u>
           use the VCF header in the provided text <u>FILE</u>

       <b>-0</b>  <b>--missing-to-ref</b>
           assume genotypes at missing sites are 0/0

       <b>-f,</b> <b>--apply-filters</b> <u>LIST</u>
           see <b>Common</b> <b>Options</b>

       <b>-F,</b> <b>--filter-logic</b> <u>x</u>|<u>+</u>
           Set the output record to PASS if any of the inputs is PASS (<u>x</u>), or apply all filters (<u>+</u>), which is
           the default.

       <b>-g,</b> <b>--gvcf</b> <u>-</u>|<u>FILE</u>
           merge gVCF blocks, INFO/END tag is expected. If the reference fasta file <u>FILE</u> is not given and the
           dash (<u>-</u>) is given, unknown reference bases generated at gVCF block splits will be substituted with
           N’s. The <b>--gvcf</b> option uses the following default INFO rules: <b>-i</b>
           <b>QS:sum,MinDP:min,I16:sum,IDV:max,IMF:max</b> and the following missing rules: <b>-M</b> <b>PL:max,AD:0</b>.

       <b>-i,</b> <b>--info-rules</b> <u>-</u>|<u>TAG:METHOD</u>[,...]
           Rules for merging INFO fields (scalars or vectors) or <u>-</u> to disable the default rules. <u>METHOD</u> is one
           of <u>sum</u>, <u>avg</u>, <u>min</u>, <u>max</u>, <u>join</u>. Default is <u>DP:sum,DP4:sum</u> if these fields exist in the input files.
           Fields with no specified rule will take the value from the first input file. The merged QUAL value is
           currently set to the maximum. This behaviour is not user controllable at the moment.

       <b>-l,</b> <b>--file-list</b> <u>FILE</u>
           Read file names from <u>FILE</u>, one file name per line.

       <b>-L,</b> <b>--local-alleles</b> <u>INT</u>
           Sites with many alternate alleles can require extremely large storage space which can exceed the 2GB
           size limit representable by BCF. This is caused by Number=G tags (such as FORMAT/PL) which store a
           value for each combination of reference and alternate alleles. The <b>-L,</b> <b>--local-alleles</b> option allows
           replacement of such tags with a localized tag (FORMAT/LPL) which only includes a subset of alternate
           alleles relevant for that sample. A new FORMAT/LAA tag is added which lists 1-based indices of the
           alternate alleles relevant (local) for the current sample. The number <u>INT</u> gives the maximum number of
           alternate alleles that can be included in the PL tag. The default value is 0 which disables the
           feature and outputs values for all alternate alleles.

       <b>-m,</b> <b>--merge</b> <u>snps</u>|<u>indels</u>|<u>both</u>|<u>snp-ins-del</u>|<u>all</u>|<u>none</u>|<u>id</u>[,<u>*</u>]
           The option controls what types of multiallelic records can be created. If single asterisk <u>*</u> is
           appended, the unobserved allele <u>&lt;*&gt;</u> or <u>&lt;NON_REF&gt;</u> will be removed at variant sites; if two asterisks
           <u>**</u> are appended, the unobserved allele will be removed all sites.

           -m none        .. no new multiallelics, output multiple records instead
           -m snps        .. allow multiallelic SNP records
           -m indels      .. allow multiallelic indel records
           -m both        .. both SNP and indel records can be multiallelic
           -m both,*      .. same as above but remove &lt;*&gt; (or &lt;NON_REF&gt;) from variant sites
           -m both,**     .. same as above but remove &lt;*&gt; (or &lt;NON_REF&gt;) at all sites
           -m all         .. SNP records can be merged with indel records
           -m snp-ins-del .. allow multiallelic SNVs, insertions, deletions, but don't mix them
           -m id          .. merge by ID

       <b>-M,</b> <b>--missing-rules</b> <u>-</u>|<u>TAG:METHOD</u>[,...]
           Rules for merging vector tags at multiallelic sites. When input files have different alternate
           alleles, vector fields pertaining to unobserved alleles are set to missing (<u>.</u>) by default. The <u>METHOD</u>
           is one of <u>.</u> (the default, use missing values), <u>NUMBER</u> (use a constant value, e.g. 0), <u>max</u> (the
           maximum value observed for other alleles in the sample). When <b>--gvcf</b> option is set, the rule <b>-M</b>
           <b>PL:max,AD:0</b> is implied. This can be overridden with providing <b>-M</b> <b>-</b> or <b>-M</b> <b>PL:.,AD:.</b>. Note that if the
           unobserved allele is explicitly present as <u>&lt;*&gt;</u> or <u>&lt;NON_REF&gt;</u>, then its corresponding value will be
           used regardless of <b>-M</b> settings.

       <b>--no-index</b>
           the option allows files to be merged without indexing them first. In order for this option to work,
           the user must ensure that the input files have chromosomes in the same order and consistent with the
           order of sequences in the VCF header.

       <b>--no-version</b>
           see <b>Common</b> <b>Options</b>

       <b>-o,</b> <b>--output</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>-O,</b> <b>--output-type</b> <u>b</u>|<u>u</u>|<u>z</u>|<u>v</u>[0-9]
           see <b>Common</b> <b>Options</b>

       <b>-r,</b> <b>--regions</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-R,</b> <b>--regions-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--regions-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>--threads</b> <u>INT</u>
           see <b>Common</b> <b>Options</b>

       <b>-W</b>[<u>FMT</u>]<b>,</b> <b>-W</b>[=<u>FMT</u>]<b>,</b> <b>--write-index</b>[=<u>FMT</u>]
           Automatically index the output file. <u>FMT</u> is optional and can be one of "tbi" or "csi" depending on
           output file format.

   <b>bcftools</b> <b>mpileup</b> <b>[</b><u><b>OPTIONS</b></u><b>]</b> <b>-f</b> <u><b>ref.fa</b></u> <u><b>in.bam</b></u> <b>[</b><u><b>in2.bam</b></u> <b>[...]]</b>
       Generate VCF or BCF containing genotype likelihoods for one or multiple alignment (BAM or CRAM) files.
       This is based on the original <b>samtools</b> <b>mpileup</b> command (with the <b>-v</b> or <b>-g</b> options) producing genotype
       likelihoods in VCF or BCF format, but not the textual pileup output. The <b>mpileup</b> command was transferred
       to bcftools in order to avoid errors resulting from use of incompatible versions of samtools and bcftools
       when using in the mpileup+bcftools call pipeline.

       Individuals are identified from the SM tags in the @RG header lines. Multiple individuals can be pooled
       in one alignment file, also one individual can be separated into multiple files. If sample identifiers
       are absent, each input file is regarded as one sample.

       Note that there are two orthogonal ways to specify locations in the input file; via <b>-r</b> <u>region</u> and <b>-t</b>
       <u>positions</u>. The former uses (and requires) an index to do random access while the latter streams through
       the file contents filtering out the specified regions, requiring no index. The two may be used in
       conjunction. For example a BED file containing locations of genes in chromosome 20 could be specified
       using <b>-r</b> <b>20</b> <b>-t</b> <b>chr20.bed</b>, meaning that the index is used to find chromosome 20 and then it is filtered
       for the regions listed in the BED file. Also note that the <b>-r</b> option can be much slower than <b>-t</b> with many
       regions and can require more memory when multiple regions and many alignment files are processed.

   <b>Input</b> <b>options</b>
       <b>-6,</b> <b>--illumina1.3+</b>
           Assume the quality is in the Illumina 1.3+ encoding.

       <b>-A,</b> <b>--count-orphans</b>
           Include anomalous read pairs in variant calling, i.e. reads with flag PAIRED but not PROPER_PAIR set.
           By default such reads are discarded.

       <b>-b,</b> <b>--bam-list</b> <u>FILE</u>
           List of input alignment files, one file per line [null]

       <b>-B,</b> <b>--no-BAQ</b>
           Disable probabilistic realignment for the computation of base alignment quality (BAQ). BAQ is the
           Phred-scaled probability of a read base being misaligned. Applying this option greatly helps to
           reduce false SNPs caused by misalignments.

       <b>-C,</b> <b>--adjust-MQ</b> <u>INT</u>
           Coefficient for downgrading mapping quality for reads containing excessive mismatches. Mismatches are
           counted as a proportion of the number of aligned bases ("M", "X" or "=" CIGAR operations), along with
           their quality, to derive an upper-bound of the mapping quality. Original mapping qualities lower than
           this are left intact, while higher ones are capped at the new adjusted score.

           The exact formula is complex and likely tuned to specific instruments and specific alignment tools,
           so this option is disabled by default (indicated as having a zero value). Variables in the formulae
           and their meaning are defined below.

               Variable    Meaning / formula
               M           The number of matching CIGAR bases (operation "M", "X" or "=").
               X           The number of substitutions with quality &gt;= 13.
               SubQ        The summed quality of substitution bases included in X, capped
                           at a maximum of quality 33 per mismatching base.
               ClipQ       The summed quality of soft-clipped or hard-clipped bases. This
                           has no minimum or maximum quality threshold per base. For
                           hard-clipped bases the per-base quality is taken as 13.

               T           SubQ - 10 * log10(M^X / X!) + ClipQ/5
               Cap         MAX(0, INT * sqrt((INT - T) / INT))

           Some notes on the impact of this.

           •   As the number of mismatches increases, the mapping quality cap reduces, eventually resulting in
               discarded alignments.

           •   High quality mismatches reduces the cap faster than low quality mismatches.

           •   The starting INT value also acts as a hard cap on mapping quality, even when zero mismatches are
               observed.

           •   Indels have no impact on the mapping quality.

               The intent of this option is to work around aligners that compute a mapping quality using a local
               alignment without having any regard to the degree of clipping required or consideration of
               potential contamination or large scale insertions with respect to the reference. A record may
               align uniquely and have no close second match, but having a high number of mismatches may still
               imply that the reference is not the correct site.

               However we do not recommend use of this parameter unless you fully understand the impact of it
               and have determined that it is appropriate for your sequencing technology.

       <b>-D,</b> <b>--full-BAQ</b>
           Run the BAQ algorithm on all reads, not just those in problematic regions. This matches the behaviour
           for Bcftools 1.12 and earlier.

           By default mpileup uses heuristics to decide when to apply the BAQ algorithm. Most sequences will not
           be BAQ adjusted, giving a CPU time closer to --no-BAQ, but it will still be applied in regions with
           suspected problematic alignments. This has been tested to work well on single sample data with even
           allele frequency, but the reliability is unknown for multi-sample calling and for low allele
           frequency variants so full BAQ is still recommended in those scenarios.

       <b>-d,</b> <b>--max-depth</b> <u>INT</u>
           At a position, read maximally <u>INT</u> reads per input file. Note that the original <b>samtools</b> <b>mpileup</b>
           command had a minimum value of <u>8000/n</u> where <u>n</u> was the number of input files given to mpileup. This
           means that in <b>samtools</b> <b>mpileup</b> the default was highly likely to be increased and the <b>-d</b> parameter
           would have an effect only once above the cross-sample minimum of 8000. This behavior was problematic
           when working with a combination of single- and multi-sample bams, therefore in <b>bcftools</b> <b>mpileup</b> the
           user is given the full control (and responsibility), and an informative message is printed instead
           [250]

       <b>-E,</b> <b>--redo-BAQ</b>
           Recalculate BAQ on the fly, ignore existing BQ tags

       <b>-f,</b> <b>--fasta-ref</b> <u>FILE</u>
           The <b>faidx</b>-indexed reference file in the FASTA format. The file can be optionally compressed by <b>bgzip</b>.
           Reference is required by default unless the <b>--no-reference</b> option is set [null]

       <b>--no-reference</b>
           Do not require the <b>--fasta-ref</b> option.

       <b>-G,</b> <b>--read-groups</b> <u>FILE</u>
           list of read groups to include or exclude if prefixed with "^". One read group per line. This file
           can also be used to assign new sample names to read groups by giving the new sample name as a second
           white-space-separated field, like this: "read_group_id new_sample_name". If the read group name is
           not unique, also the bam file name can be included: "read_group_id file_name sample_name". If all
           reads from the alignment file should be treated as a single sample, the asterisk symbol can be used:
           "* file_name sample_name". Alignments without a read group ID can be matched with "?". <b>NOTE:</b> The
           meaning of <b>bcftools</b> <b>mpileup</b> <b>-G</b> is the opposite of <b>samtools</b> <b>mpileup</b> <b>-G</b>.

               RG_ID_1
               RG_ID_2  SAMPLE_A
               RG_ID_3  SAMPLE_A
               RG_ID_4  SAMPLE_B
               RG_ID_5  FILE_1.bam  SAMPLE_A
               RG_ID_6  FILE_2.bam  SAMPLE_A
               *        FILE_3.bam  SAMPLE_C
               ? FILE_3.bam  SAMPLE_D

       <b>--indels-2.0</b>
           A new EXPERIMENTAL indel calling model which aims to address some known deficiencies of the current
           indel calling algorithm. Specifically, it uses diploid reference consensus sequence. Note that in the
           current version it has the potential to increase sensitivity but at the cost of decreased
           specificity. Only works with short-read sequencing technologies.

       <b>--indels-cns</b>
           Another EXPERIMENTAL indel calling method, predating indels-2.0 in PR form, but merged more recently.
           It also uses a diploid reference consensus, but with added parameters and heuristics to optimise for
           a variety of sequencing platforms. This is usually faster and more accurate than the default caller
           and --indels-2.0, but has not been tested on non-diploid samples and samples without approximately
           even allele frequency.

       <b>--no-indels-cns</b>
           May be used to turn off --indels-cns mode when using one of the newer profiles that has this enabled
           by default.

       <b>-q,</b> <b>-min-MQ</b> <u>INT</u>
           Minimum mapping quality for an alignment to be used [0]

       <b>-Q,</b> <b>--min-BQ</b> <u>INT</u>
           Minimum base quality for a base to be considered [13]

       <b>--max-BQ</b> <u>INT</u>
           Caps the base quality to a maximum value [60]. This can be particularly useful on technologies that
           produce overly optimistic high qualities, leading to too many false positives or incorrect genotype
           assignments.

       <b>-r,</b> <b>--regions</b> <u>CHR</u>|<u>CHR:POS</u>|<u>CHR:FROM-TO</u>|<u>CHR:FROM-</u>[,...]
           Only generate mpileup output in given regions. Requires the alignment files to be indexed. If used in
           conjunction with -l then considers the intersection; see <b>Common</b> <b>Options</b>

       <b>-R,</b> <b>--regions-file</b> <u>FILE</u>
           As for <b>-r,</b> <b>--regions</b>, but regions read from FILE; see <b>Common</b> <b>Options</b>

       <b>--regions-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>--ignore-RG</b>
           Ignore RG tags. Treat all reads in one alignment file as one sample.

       <b>--ls,</b> <b>--skip-all-set</b>
           Skip reads with all of the FLAG bits set [null]

       <b>--ns,</b> <b>--skip-any-set</b>
           Skip reads with any of the FLAG bits set. This option replaces and is synonymous to the deprecated
           <b>--ff,</b> <b>--excl-flags</b> [UNMAP,SECONDARY,QCFAIL,DUP]

       <b>--lu,</b> <b>--skip-all-unset</b>
           Skip reads with all of the FLAG bits unset. This option replaces and is synonymous to the deprecated
           <b>--rf,</b> <b>--incl-flags</b> [null]

       <b>--nu,</b> <b>--skip-any-unset</b>
           Skip reads with any of the FLAG bits unset [null]

       <b>-s,</b> <b>--samples</b> <u>LIST</u>
           list of sample names. See <b>Common</b> <b>Options</b>

       <b>-S,</b> <b>--samples-file</b> <u>FILE</u>
           file of sample names to include or exclude if prefixed with "^". One sample per line. This file can
           also be used to rename samples by giving the new sample name as a second white-space-separated
           column, like this: "old_name new_name". If a sample name contains spaces, the spaces can be escaped
           using the backslash character, for example "Not\ a\ good\ sample\ name".

       <b>-t,</b> <b>--targets</b> <u>LIST</u>
           see <b>Common</b> <b>Options</b>

       <b>-T,</b> <b>--targets-file</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>--targets-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-x,</b> <b>--ignore-overlaps</b>
           Disable read-pair overlap detection.

       <b>--seed</b> <u>INT</u>
           Set the random number seed used when sub-sampling deep regions [0].

   <b>Output</b> <b>options</b>
       <b>-a,</b> <b>--annotate</b> <u>LIST</u>
           Comma-separated list of FORMAT and INFO tags to output. (case-insensitive, the "FORMAT/" prefix is
           optional, and use "?" to list available annotations on the command line) [null]:

           FORMAT/AD   .. Allelic depth (Number=R,Type=Integer)
           FORMAT/ADF  .. Allelic depths on the forward strand (Number=R,Type=Integer)
           FORMAT/ADR  .. Allelic depths on the reverse strand (Number=R,Type=Integer)
           FORMAT/DP   .. Number of high-quality bases (Number=1,Type=Integer)
           FORMAT/SP   .. Phred-scaled strand bias P-value (Number=1,Type=Integer)
           FORMAT/SCR  .. Number of soft-clipped reads (Number=1,Type=Integer)

           INFO/AD     .. Total allelic depth (Number=R,Type=Integer)
           INFO/ADF    .. Total allelic depths on the forward strand (Number=R,Type=Integer)
           INFO/ADR    .. Total allelic depths on the reverse strand (Number=R,Type=Integer)
           INFO/SCR    .. Number of soft-clipped reads (Number=1,Type=Integer)

           FORMAT/DV   .. Deprecated in favor of FORMAT/AD; Number of high-quality non-reference bases, (Number=1,Type=Integer)
           FORMAT/DP4  .. Deprecated in favor of FORMAT/ADF and FORMAT/ADR; Number of high-quality ref-forward, ref-reverse,
                          alt-forward and alt-reverse bases (Number=4,Type=Integer)
           FORMAT/DPR  .. Deprecated in favor of FORMAT/AD; Number of high-quality bases for each observed allele (Number=R,Type=Integer)
           INFO/DPR    .. Deprecated in favor of INFO/AD; Number of high-quality bases for each observed allele (Number=R,Type=Integer)

       <b>-g,</b> <b>--gvcf</b> <u>INT</u>[,...]
           output gVCF blocks of homozygous REF calls, with depth (DP) ranges specified by the list of integers.
           For example, passing <u>5,15</u> will group sites into two types of gVCF blocks, the first with minimum
           per-sample DP from the interval [5,15) and the latter with minimum depth 15 or more. In this example,
           sites with minimum per-sample depth less than 5 will be printed as separate records, outside of gVCF
           blocks.

       <b>--no-version</b>
           see <b>Common</b> <b>Options</b>

       <b>-o,</b> <b>--output</b> <u>FILE</u>
           Write output to <u>FILE</u>, rather than the default of standard output. (The same short option is used for
           both <b>--open-prob</b> and <b>--output</b>. If <b>-o</b>'s argument contains any non-digit characters other than a
           leading + or - sign,  it  is  interpreted  as <b>--output</b>. Usually the filename extension will take care
           of this, but to write to an entirely numeric filename use <b>-o</b> <b>./123</b> or <b>--output</b> <b>123</b>.)

       <b>-O,</b> <b>--output-type</b> <u>b</u>|<u>u</u>|<u>z</u>|<u>v</u>[0-9]
           see <b>Common</b> <b>Options</b>

       <b>--threads</b> <u>INT</u>
           see <b>Common</b> <b>Options</b>

       <b>-U,</b> <b>--mwu-u</b>
           The the previous Mann-Whitney U test score from version 1.12 and earlier. This is a probability
           score, but importantly it folds probabilities above or below the desired score into the same P. The
           new Mann-Whitney U test score is a "Z score", expressing the score as the number of standard
           deviations away from the mean (with zero being matching the mean). It keeps both positive and
           negative values. This can be important for some tests where errors are asymmetric.

               This option changes the INFO field names produced back to the ones
               used by the earlier Bcftools releases. For excample BQBZ becomes
               BQB.

       <b>-W</b>[<u>FMT</u>]<b>,</b> <b>-W</b>[=<u>FMT</u>]<b>,</b> <b>--write-index</b>[=<u>FMT</u>]
           Automatically index the output file. <u>FMT</u> is optional and can be one of "tbi" or "csi" depending on
           output file format.

   <b>Options</b> <b>for</b> <b>SNP/INDEL</b> <b>genotype</b> <b>likelihood</b> <b>computation</b>
       <b>-X,</b> <b>--config</b> <u>STR</u>
           Specify a platform specific configuration profile. Specifying the profile as "list" will list the
           available profile names and the parameters they change. There are profiles named after a release,
           which should be used if you wish to ensure forward compatibility of results. The non-versioned names
           (eg "illumina") will always point to the most recent set of parameters for that instrument type. The
           current values are:

               1.12            -Q13 -h100 -m1

               bgi
               bgi-1.20        --indels-cns -B --indel-size 80 -F0.1 --indel-bias 0.9
                               --seqq-offset 120

               illumina-1.18   [ default values ]

               illumina
               illumina-1.20   --indels-cns --seqq-offset 125

               ont             -B -Q5 --max-BQ 30 -I

               ont-sup
               ont-sup-1.20    --indels-cns -B -Q1 --max-BQ 35 --delta-BQ 99 -F0.2
                               -o15 -e1 -h110 --del-bias 0.4 --indel-bias 0.7
                               --poly-mqual --seqq-offset 130 --indel-size 80

               pacbio-ccs-1.18 -D -Q5 --max-BQ 50 -F0.1 -o25 -e1 -M99999

               pacbio-ccs
               pacbio-ccs-1.20  --indels-cns -B -Q5 --max-BQ 50 -F0.1 -o25 -e1 -h300
                                --delta-BQ 10 --del-bias 0.4 --poly-mqual
                                --indel-bias 0.9 --seqq-offset 118 --indel-size 80
                                --score-vs-ref 0.7

               ultima
               ultima-1.20      --indels-cns -B -Q1 --max-BQ 30 --delta-BQ 10 -F0.15
                                -o20 -e10 -h250 --del-bias 0.3 --indel-bias 0.7
                                --poly-mqual --seqq-offset 140 --score-vs-ref 0.3
                                --indel-size 80

       <b>--ar,</b> <b>--ambig-reads</b> <u>drop</u>|<u>incAD</u>|<u>incAD0</u>
           What to do with ambiguous indel reads that do not span an entire short tandem repeat region: discard
           ambiguous reads from calling and do not increment high-quality AD depth counters (<u>drop</u>), exclude from
           calling but increment AD counters proportionally (<u>incAD</u>), exclude from calling and increment the
           first value of the AD counter (<u>incAD0</u>) [<u>drop</u>]

       <b>-e,</b> <b>--ext-prob</b> <u>INT</u>
           Phred-scaled gap extension sequencing error probability. Reducing <u>INT</u> leads to longer indels [20]

       <b>-F,</b> <b>--gap-frac</b> <u>FLOAT</u>
           Minimum fraction of gapped reads [0.002]

       <b>-h,</b> <b>--tandem-qual</b> <u>INT</u>
           Coefficient for modeling homopolymer errors. Given an <u>l</u>-long homopolymer run, the sequencing error of
           an indel of size s is modeled as <u>INT</u>*s/l [500] Increasing this informs the caller that indels in long
           homopolymers are more likely genuine and less likely to be sequencing artifacts. Hence increasing
           tandem-qual will have higher recall and lower precision. Bcftools 1.12 and earlier had a default of
           100, which was tuned around more error prone instruments. Note changing this may have a minor impact
           on SNP calling too. For maximum SNP calling accuracy, it may be preferable to adjust this lower
           again, although this will adversely affect indels.

       <b>--indel-bias</b> <u>FLOAT</u>
           Skews the indel scores up or down, trading recall (low false-negative) vs precision (low
           false-positive) [1.0]. In Bcftools 1.12 and earlier this parameter didn’t exist, but had an implied
           value of 1.0. If you are planning to do heavy filtering of variants, selecting the best quality ones
           only (favouring precision over recall), it is advisable to set this lower (such as 0.75) while higher
           depth samples or where you favour recall rates over precision may work better with a higher value
           such as 2.0.

       <b>--del-bias</b> <u>FLOAT</u>
           Skews the likelihood of deletions over insertions. Defaults to an even distribution value of 1.0.
           Lower values imply a higher rate of false positive deletions (meaning candidate deletions are less
           likely to be real).

       <b>--indel-size</b> <u>INT</u>
           Indel window size to use when assessing the quality of candidate indels. Note that although the
           window size approximately corresponds to the maximum indel size considered, it is not an exact
           threshold [110]

       <b>--seqq-offset</b> <u>INT</u>
           Tunes the importance of indel sequence quality per depth. The final "seqQ" quality used is "offset -
           5*MIN(depth,20)". [120]

       <b>--poly-mqual</b>
           Use the lowest quality value within a homopolymer run, instead of the quality immediately adjacent to
           the indel. This may be important for unclocked instruments, particularly ones with a flow chemistry
           where runs of bases of identical type are incorporated together.

       <b>-I,</b> <b>--skip-indels</b>
           Do not perform INDEL calling

       <b>-L,</b> <b>--max-idepth</b> <u>INT</u>
           Skip INDEL calling if the average per-sample depth is above <u>INT</u> [250]

       <b>-m,</b> <b>--min-ireads</b> <u>INT</u>
           Minimum number gapped reads for indel candidates <u>INT</u> [1]

       <b>-M,</b> <b>--max-read-len</b> <u>INT</u>
           The maximum read length permitted by the BAQ algorithm [500]. Variants are still called on longer
           reads, but they will not be passed through the BAQ method. This limit exists to prevent excessively
           long BAQ times and high memory usage. Note if partial BAQ is enabled with <u>-D</u> then raising this
           parameter will likely not have a significant a CPU cost.

       <b>-o,</b> <b>--open-prob</b> <u>INT</u>
           Phred-scaled gap open sequencing error probability. Reducing <u>INT</u> leads to more indel calls. (The same
           short option is used for both <b>--open-prob</b> and <b>--output</b>. When -o’s argument contains only an optional
           + or - sign followed by the digits 0 to 9, it is interpreted  as <b>--open-prob</b>.) [40]

       <b>-p,</b> <b>--per-sample-mF</b>
           Apply <b>-m</b> and <b>-F</b> thresholds per sample to increase sensitivity of calling. By default both options are
           applied to reads pooled from all samples.

       <b>-P,</b> <b>--platforms</b> <u>STR</u>
           Comma-delimited  list  of  platforms (determined by <b>@RG-PL</b>) from which indel candidates are obtained.
           It is recommended to collect indel candidates from sequencing technologies that have low indel error
           rate such as ILLUMINA [all]

   <b>Examples:</b>
       Call SNPs and short INDELs, then mark low quality sites and sites with the read depth exceeding a limit.
       (The read depth should be adjusted to about twice the average read depth as higher read depths usually
       indicate problematic regions which are often enriched for artefacts.) One may consider to add <b>-C50</b> to
       mpileup if mapping quality is overestimated  for reads containing  excessive mismatches. Applying this
       option usually helps for BWA-backtrack alignments, but may not other aligners.

               bcftools mpileup -Ou -f ref.fa aln.bam | \
               bcftools call -Ou -mv | \
               bcftools filter -s LowQual -e '%QUAL&lt;20 || DP&gt;100' &gt; var.flt.vcf

   <b>bcftools</b> <b>norm</b> <b>[</b><u><b>OPTIONS</b></u><b>]</b> <u><b>file.vcf.gz</b></u>
       Left-align and normalize indels, check if REF alleles match the reference, split multiallelic sites into
       multiple rows; recover multiallelics from multiple rows. Left-alignment and normalization will only be
       applied if the <b>--fasta-ref</b> option is supplied.

       <b>-a,</b> <b>--atomize</b>
           Decompose complex variants, e.g. split MNVs into consecutive SNVs. See also <b>--atom-overlaps</b> and
           <b>--old-rec-tag</b>.

       <b>--atom-overlaps</b> <u>.</u>|<u>*</u>
           Alleles missing because of an overlapping variant can be set either to missing (.) or to the star
           alele (*), as recommended by the VCF specification. IMPORTANT: Note that asterisk is expaneded by
           shell and must be put in quotes or escaped by a backslash:

               # Before atomization:
               100  CC  C,GG   1/2

               # After:
               #   bcftools norm -a --atom-overlaps .
               100         C         G      ./1
               100         CC         C      1/.
               101         C         G      ./1

               # After:
               #   bcftools norm -a --atom-overlaps '*'
               #   bcftools norm -a --atom-overlaps \*
               100         C         G,*    2/1
               100         CC         C,*    1/2
               101         C         G,*    2/1

       <b>-c,</b> <b>--check-ref</b> <u>e</u>|<u>w</u>|<u>x</u>|<u>s</u>
           what to do when incorrect or missing REF allele is encountered: exit (<u>e</u>), warn (<u>w</u>), exclude (<u>x</u>), or
           set/fix (<u>s</u>) bad sites. The <u>w</u> option can be combined with <u>x</u> and <u>s</u>. Note that <u>s</u> can swap alleles and
           will update genotypes (GT) and AC counts, but will not attempt to fix PL or other fields. Also note,
           and this cannot be stressed enough, that <u>s</u> will NOT fix strand issues in your VCF, do NOT use it for
           that purpose!!! (Instead see <a href="http://samtools.github.io/bcftools/howtos/plugin.af-dist.html">http://samtools.github.io/bcftools/howtos/plugin.af-dist.html</a> and
           <a href="http://samtools.github.io/bcftools/howtos/plugin.fixref.html">http://samtools.github.io/bcftools/howtos/plugin.fixref.html</a>.)

       <b>-d,</b> <b>--rm-dup</b> <u>snps</u>|<u>indels</u>|<u>both</u>|<u>all</u>|<u>exact</u>
           If a record is present multiple times, output only the first instance. See also <b>--collapse</b> in <b>Common</b>
           <b>Options</b>.

       <b>-D,</b> <b>--remove-duplicates</b>
           If a record is present in multiple files, output only the first instance. Alias for <b>-d</b> <b>none</b>,
           deprecated.

       <b>-e,</b> <b>--exclude</b> <u>EXPRESSION</u>
           do not normalize input records for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.
           Note that duplicate removal ignores this option.

       <b>-f,</b> <b>--fasta-ref</b> <u>FILE</u>
           reference sequence. Supplying this option will turn on left-alignment and normalization, however, see
           also the <b>--do-not-normalize</b> option below.

       <b>--force</b>
           try to proceed with <b>-m-</b> even if malformed tags with incorrect number of fields are encountered,
           discarding such tags. (Experimental, use at your own risk.)

       <b>-g,</b> <b>--gff-annot</b> <u>FILE</u>
           when a GFF file is provided, follow HGVS 3’rule and right-align variants in transcripts on the
           forward strand. In case of overlapping transcripts, the default mode is to left-align the variant.
           For a description of the supported GFF3 file format see <b>bcftools</b> <b>csq</b>.

       <b>-i,</b> <b>--include</b> <u>EXPRESSION</u>
           normalize only input records for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.
           Note that duplicate removal ignores this option.

       <b>--keep-sum</b> <u>TAG</u>[,...]
           keep vector sum constant when splitting multiallelic sites. Only AD tag is currently supported. See
           also https://github.com/samtools/bcftools/issues/360

       <b>-m,</b> <b>--multiallelics</b> <b>-</b>|<b>+</b>[<u>snps</u>|<u>indels</u>|<u>both</u>|<u>any</u>]
           split multiallelic sites into biallelic records (<b>-</b>) or join biallelic sites into multiallelic records
           (<b>+</b>). An optional type string can follow which controls variant types which should be split or merged
           together: If only SNP records should be split or merged, specify <u>snps</u>; if both SNPs and indels should
           be merged separately into two records, specify <u>both</u>; if SNPs and indels should be merged into a
           single record, specify <u>any</u>.

           Note that multiallelic sites with both SNPs and indels will be split into biallelic sites with both
           <b>-m</b> <b>-snps</b> and <b>-m</b> <b>-indels</b>.

       <b>--multi-overlaps</b> <u>0</u>|<u>.</u>
           use the reference (<u>0</u>) or missing (<u>.</u>) allele for overlapping alleles after splitting multiallelic
           sites

       <b>--no-version</b>
           see <b>Common</b> <b>Options</b>

       <b>-N,</b> <b>--do-not-normalize</b>
           the <u>-c</u> <u>s</u> option can be used to fix or set the REF allele from the reference <u>-f</u>. The <u>-N</u> option will
           not turn on indel normalisation as the <u>-f</u> option normally implies

       <b>--old-rec-tag</b> <u>STR</u>
           Add INFO/STR annotation with the original record. The format of the annotation is
           CHROM|POS|REF|ALT|USED_ALT_IDX.

       <b>-o,</b> <b>--output</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>-O,</b> <b>--output-type</b> <u>b</u>|<u>u</u>|<u>z</u>|<u>v</u>[0-9]
           see <b>Common</b> <b>Options</b>

       <b>-r,</b> <b>--regions</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-R,</b> <b>--regions-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--regions-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-s,</b> <b>--strict-filter</b>
           when merging (<u>-m+</u>), merged site is PASS only if all sites being merged PASS

       <b>-S,</b> <b>--sort</b> <u>pos</u>|<u>lex</u>
           when splitting sites or processing duplicates, sort records on output by POS only (<u>pos</u>, the default)
           or by POS and lexicographically by REF+ALT (<u>lex</u>)

       <b>-t,</b> <b>--targets</b> <u>LIST</u>
           see <b>Common</b> <b>Options</b>

       <b>-T,</b> <b>--targets-file</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>--targets-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>--threads</b> <u>INT</u>
           see <b>Common</b> <b>Options</b>

       <b>-v,</b> <b>--verbose</b> <u>INT</u>
           verbosity level of GFF parsing (0-2)

       <b>-w,</b> <b>--site-win</b> <u>INT</u>
           maximum distance between two records to consider when locally sorting variants which changed position
           during the realignment

       <b>-W</b>[<u>FMT</u>]<b>,</b> <b>-W</b>[=<u>FMT</u>]<b>,</b> <b>--write-index</b>[=<u>FMT</u>]
           Automatically index the output file. <u>FMT</u> is optional and can be one of "tbi" or "csi" depending on
           output file format.

   <b>bcftools</b> <b>[plugin</b> <u><b>NAME</b></u><b>|+</b><u><b>NAME</b></u><b>]</b> <u><b>[OPTIONS]</b></u> <u><b>FILE</b></u> <b>—</b> <u><b>[PLUGIN</b></u> <u><b>OPTIONS]</b></u>
       A common framework for various utilities. The plugins can be used the same way as normal commands only
       their name is prefixed with "+". Most plugins accept two types of parameters: general options shared by
       all plugins followed by a separator, and a list of plugin-specific options. There are some exceptions to
       this rule, some plugins do not accept the common options and implement their own parameters. Therefore
       please pay attention to the usage examples that each plugin comes with.

   <b>VCF</b> <b>input</b> <b>options:</b>
       <b>-e,</b> <b>--exclude</b> <u>EXPRESSION</u>
           exclude sites for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.

       <b>-i,</b> <b>--include</b> <u>EXPRESSION</u>
           include only sites for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.

       <b>-r,</b> <b>--regions</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-R,</b> <b>--regions-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--regions-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-t,</b> <b>--targets</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-T,</b> <b>--targets-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--targets-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

   <b>VCF</b> <b>output</b> <b>options:</b>
       <b>--no-version</b>
           see <b>Common</b> <b>Options</b>

       <b>-o,</b> <b>--output</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>-O,</b> <b>--output-type</b> <u>b</u>|<u>u</u>|<u>z</u>|<u>v</u>[0-9]
           see <b>Common</b> <b>Options</b>

       <b>--threads</b> <u>INT</u>
           see <b>Common</b> <b>Options</b>

       <b>-W</b>[<u>FMT</u>]<b>,</b> <b>-W</b>[=<u>FMT</u>]<b>,</b> <b>--write-index</b>[=<u>FMT</u>]
           Automatically index the output file. <u>FMT</u> is optional and can be one of "tbi" or "csi" depending on
           output file format.

   <b>Plugin</b> <b>options:</b>
       <b>-h,</b> <b>--help</b>
           list plugin’s options

       <b>-l,</b> <b>--list-plugins</b>
           List all available plugins.

           By default, appropriate system directories are searched for installed plugins. You can override this
           by setting the BCFTOOLS_PLUGINS environment variable to a colon-separated list of directories to
           search. If BCFTOOLS_PLUGINS begins with a colon, ends with a colon, or contains adjacent colons, the
           system directories are also searched at that position in the list of directories.

       <b>-v,</b> <b>--verbose</b>
           print debugging information to debug plugin failure

       <b>-V,</b> <b>--version</b>
           print version string and exit

   <b>List</b> <b>of</b> <b>plugins</b> <b>coming</b> <b>with</b> <b>the</b> <b>distribution:</b>
       <b>ad-bias</b>
           find positions with wildly varying ALT allele frequency (Fisher test on FMT/AD)

       <b>add-variantkey</b>
           add VariantKey INFO fields VKX and RSX

       <b>af-dist</b>
           collect AF deviation stats and GT probability distribution given AF and assuming HWE

       <b>allele-length</b>
           count the frequency of the length of REF, ALT and REF+ALT

       <b>check-ploidy</b>
           check if ploidy of samples is consistent for all sites

       <b>check-sparsity</b>
           print samples without genotypes in a region or chromosome

       <b>color-chrs</b>
           color shared chromosomal segments, requires trio VCF with phased GTs

       <b>contrast</b>
           runs a basic association test, per-site or in a region, and checks for novel alleles and genotypes in
           two groups of samples. Adds the following INFO annotations:

           •   PASSOC  .. Fisher’s exact test probability of genotypic association (REF vs non-REF allele)

           •   FASSOC  .. proportion of non-REF allele in controls and cases

           •   NASSOC  .. number of control-ref, control-alt, case-ref and case-alt alleles

           •   NOVELAL .. lists samples with a novel allele not observed in the control group

           •   NOVELGT .. lists samples with a novel genotype not observed in the control group

       <b>counts</b>
           a minimal plugin which counts number of SNPs, Indels, and total number of sites.

       <b>dosage</b>
           print genotype dosage. By default the plugin searches for PL, GL and GT, in that order.

       <b>fill-from-fasta</b>
           fill INFO or REF field based on values in a fasta file

       <b>fill-tags</b>
           set various INFO tags. The list of tags supported in this version:

           •   INFO/AC         Number:A  Type:Integer  .. Allele count in genotypes

           •   INFO/AC_Hom     Number:A  Type:Integer  .. Allele counts in homozygous genotypes

           •   INFO/AC_Het     Number:A  Type:Integer  .. Allele counts in heterozygous genotypes

           •   INFO/AC_Hemi    Number:A  Type:Integer  .. Allele counts in hemizygous genotypes

           •   INFO/AF         Number:A  Type:Float    .. Allele frequency

           •   INFO/AN         Number:1  Type:Integer  .. Total number of alleles in called genotypes

           •   INFO/ExcHet     Number:A  Type:Float    .. Test excess heterozygosity; 1=good, 0=bad

           •   INFO/END        Number:1  Type:Integer  .. End position of the variant

           •   INFO/F_MISSING  Number:1  Type:Float    .. Fraction of missing genotypes

           •   INFO/HWE        Number:A  Type:Float    .. HWE test (PMID:15789306); 1=good, 0=bad

           •   INFO/MAF        Number:A  Type:Float    .. Minor Allele frequency

           •   INFO/NS         Number:1  Type:Integer  .. Number of samples with data

           •   INFO/TYPE       Number:. Type:String   .. The record type (REF,SNP,MNP,INDEL,etc)

           •   FORMAT/VAF      Number:A  Type:Float    .. The fraction of reads with the alternate allele,
               requires FORMAT/AD or ADF+ADR

           •   FORMAT/VAF1     Number:1  Type:Float    .. The same as FORMAT/VAF but for all alternate alleles
               cumulatively

           •   TAG=func(TAG)   Number:1  Type:Integer  .. Experimental support for user-defined expressions such
               as "DP=sum(DP)"

       <b>fixploidy</b>
           sets correct ploidy

       <b>fixref</b>
           determine and fix strand orientation

       <b>frameshifts</b>
           annotate frameshift indels

       <b>GTisec</b>
           count genotype intersections across all possible sample subsets in a vcf file

       <b>GTsubset</b>
           output only sites where the requested samples all exclusively share a genotype

       <b>guess-ploidy</b>
           determine sample sex by checking genotype likelihoods (GL,PL) or genotypes (GT) in the non-PAR region
           of chrX.

       <b>gvcfz</b>
           compress gVCF file by resizing non-variant blocks according to specified criteria

       <b>impute-info</b>
           add imputation information metrics to the INFO field based on selected FORMAT tags

       <b>indel-stats</b>
           calculates per-sample or de novo indels stats. The usage and format is similar to <b>smpl-stats</b> and
           <b>trio-stats</b>

       <b>isecGT</b>
           compare two files and set non-identical genotypes to missing

       <b>mendelian</b>
           count Mendelian consistent / inconsistent genotypes (deprecated, use +mendelian2 instead)

       <b>mendelian2</b>
           count Mendelian consistent / inconsistent genotypes.

       <b>missing2ref</b>
           sets missing genotypes (".<a href="file:/.">/.</a>") to ref allele ("0/0" or "0|0")

       <b>parental-origin</b>
           determine parental origin of a CNV region

       <b>prune</b>
           prune sites by missingness, allele frequency or linkage disequilibrium. Alternatively, annotate sites
           with r2, Lewontin’s D' (PMID:19433632), Ragsdale’s D (PMID:31697386).

       <b>remove-overlaps</b>
           remove overlapping variants and duplicate sites

       <b>scatter</b>
           intended as an inverse to bcftools concat, scatter VCF by chunks or regions, creating multiple VCFs.

       <b>setGT</b>
           general tool to set genotypes according to rules requested by the user

       <b>smpl-stats</b>
           calculates basic per-sample stats. The usage and format is similar to <b>indel-stats</b> and <b>trio-stats</b>.

       <b>split</b>
           split VCF by sample, creating single- or multi-sample VCFs

       <b>split-vep</b>
           extract fields from structured annotations such as INFO/CSQ created by VEP or INFO/BCSQ created by
           bcftools/csq. These can be added as a new INFO field to the VCF or in a custom text format. See
           <a href="http://samtools.github.io/bcftools/howtos/plugin.split-vep.html">http://samtools.github.io/bcftools/howtos/plugin.split-vep.html</a> for more.

       <b>tag2tag</b>
           Convert between similar tags, such as GL,PL,GP or QR,QA,QS or tags with localized alleles e.g.
           LPL,LAD. See <a href="http://samtools.github.io/bcftools/howtos/plugin.tag2tag.html">http://samtools.github.io/bcftools/howtos/plugin.tag2tag.html</a> for more.

       <b>trio-dnm2</b>
           screen variants for possible de-novo mutations in trios

       <b>trio-stats</b>
           calculate transmission rate in trio children. The usage and format is similar to <b>indel-stats</b> and
           <b>smpl-stats</b>.

       <b>trio-switch-rate</b>
           calculate phase switch rate in trio samples, children samples must have phased GTs

       <b>variant-distance</b>
           annotate sites with the distance to the nearest variant

       <b>variantkey-hex</b>
           generate unsorted VariantKey-RSid index files in hexadecimal format

   <b>Examples:</b>
           # List options common to all plugins
           bcftools plugin

           # List available plugins
           bcftools plugin -l

           # Run a plugin
           bcftools plugin counts in.vcf

           # Run a plugin using the abbreviated "+" notation
           bcftools +counts in.vcf

           # Run a plugin from an explicit location
           bcftools +/path/to/counts.so in.vcf

           # The input VCF can be streamed just like in other commands
           cat in.vcf | bcftools +counts

           # Print usage information of plugin "dosage"
           bcftools +dosage -h

           # Replace missing genotypes with 0/0
           bcftools +missing2ref in.vcf

           # Replace missing genotypes with 0|0
           bcftools +missing2ref in.vcf -- -p

   <b>Plugins</b> <b>troubleshooting:</b>
       Things to check if your plugin does not show up in the <b>bcftools</b> <b>plugin</b> <b>-l</b> output:

       •   Run with the <b>-v</b> option for verbose output: <b>bcftools</b> <b>plugin</b> <b>-lv</b>

       •   Does the environment variable BCFTOOLS_PLUGINS include the correct path?

   <b>Plugins</b> <b>API:</b>
           // Short description used by 'bcftools plugin -l'
           const char *about(void);

           // Longer description used by 'bcftools +name -h'
           const char *usage(void);

           // Called once at startup, allows initialization of local variables.
           // Return 1 to suppress normal VCF/BCF header output, -1 on critical
           // errors, 0 otherwise.
           int init(int argc, char **argv, bcf_hdr_t *in_hdr, bcf_hdr_t *out_hdr);

           // Called for each VCF record, return NULL to suppress the output
           bcf1_t *process(bcf1_t *rec);

           // Called after all lines have been processed to clean up
           void destroy(void);

   <b>bcftools</b> <b>polysomy</b> <b>[</b><u><b>OPTIONS</b></u><b>]</b> <u><b>file.vcf.gz</b></u>
       Detect number of chromosomal copies in VCFs annotates with the Illumina’s B-allele frequency (BAF)
       values. Note that this command is not compiled in by default, see the section <b>Optional</b> <b>Compilation</b> <b>with</b>
       <b>GSL</b> in the INSTALL file for help.

   <b>General</b> <b>options:</b>
       <b>-o,</b> <b>--output-dir</b> <u>path</u>
           output directory

       <b>-r,</b> <b>--regions</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-R,</b> <b>--regions-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--regions-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-s,</b> <b>--sample</b> <u>string</u>
           sample name

       <b>-t,</b> <b>--targets</b> <u>LIST</u>
           see <b>Common</b> <b>Options</b>

       <b>-T,</b> <b>--targets-file</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>--targets-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-v,</b> <b>--verbose</b>
           verbose debugging output which gives hints about the thresholds and decisions made by the program.
           Note that the exact output can change between versions.

   <b>Algorithm</b> <b>options:</b>
       <b>-b,</b> <b>--peak-size</b> <u>float</u>
           the minimum peak size considered as a good match can be from the interval [0,1] where larger is
           stricter

       <b>-c,</b> <b>--cn-penalty</b> <u>float</u>
           a penalty for increasing copy number state. How this works: multiple peaks are always a better fit
           than a single peak, therefore the program prefers a single peak (normal copy number) unless the
           absolute deviation of the multiple peaks fit is significantly smaller. Here the meaning of
           "significant" is given by the <u>float</u> from the interval [0,1] where larger is stricter.

       <b>-f,</b> <b>--fit-th</b> <u>float</u>
           threshold for goodness of fit (normalized absolute deviation), smaller is stricter

       <b>-i,</b> <b>--include-aa</b>
           include also the AA peak in CN2 and CN3 evaluation. This usually requires increasing <b>-f</b>.

       <b>-m,</b> <b>--min-fraction</b> <u>float</u>
           minimum distinguishable fraction of aberrant cells. The experience shows that trustworthy are
           estimates of 20% and more.

       <b>-p,</b> <b>--peak-symmetry</b> <u>float</u>
           a heuristics to filter failed fits where the expected peak symmetry is violated. The <u>float</u> is from
           the interval [0,1] and larger is stricter

   <b>bcftools</b> <b>query</b> <b>[</b><u><b>OPTIONS</b></u><b>]</b> <u><b>file.vcf.gz</b></u> <b>[</b><u><b>file.vcf.gz</b></u> <b>[...]]</b>
       Extracts fields from VCF or BCF files and outputs them in user-defined format.

       <b>-e,</b> <b>--exclude</b> <u>EXPRESSION</u>
           exclude sites for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.

       <b>--force-samples</b>
           continue even when some samples requested via <b>-s/-S</b> do not exist

       <b>-f,</b> <b>--format</b> <u>FORMAT</u>
           learn by example, see below

       <b>-F,</b> <b>--print-filtered</b> <u>STR</u>
           by default, samples failing <b>-i/-e</b> filtering expressions are suppressed from output when FORMAT fields
           are queried (for example <u>%CHROM</u> <u>%POS</u> <u>[</u> <u>%GT]</u>). With <b>-F</b>, such fields will be still printed but instead
           of their actual value, <u>STR</u> will be used.

       <b>-H,</b> <b>--print-header</b>
           print header. By default, the header is printed with column indices, e.g. "#[1]CHROM". These can be
           suppressed by giving the option twice, "-HH".

       <b>-i,</b> <b>--include</b> <u>EXPRESSION</u>
           include only sites for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.

       <b>-l,</b> <b>--list-samples</b>
           list sample names and exit

       <b>-N,</b> <b>--disable-automatic-newline</b>
           disable automatic addition of a missing newline character at the end of the formatting expression. By
           default, the program checks if the expression contains a newline and appends it if not, to prevent
           formatting the entire output into a single line by mistake. Note that versions prior to 1.18 had no
           automatic check and newline had to be included explicitly.

       <b>-o,</b> <b>--output</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>-r,</b> <b>--regions</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-R,</b> <b>--regions-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--regions-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-s,</b> <b>--samples</b> <u>LIST</u>
           see <b>Common</b> <b>Options</b>

       <b>-S,</b> <b>--samples-file</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>-t,</b> <b>--targets</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-T,</b> <b>--targets-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--targets-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-u,</b> <b>--allow-undef-tags</b>
           do not throw an error if there are undefined tags in the format string, print "." instead

       <b>-v,</b> <b>--vcf-list</b> <u>FILE</u>
           process multiple VCFs listed in the file

   <b>Format:</b>
           %CHROM          The CHROM column (similarly also other columns: POS, ID, REF, ALT, QUAL, FILTER)
           %END            End position of the REF allele
           %END0           End position of the REF allele in 0-based coordinates
           %FIRST_ALT      Alias for %ALT{0}
           %FORMAT         Prints all FORMAT fields or a subset of samples with -s or -S
           %GT             Genotype (e.g. 0/1)
           %INFO           Prints the whole INFO column
           %INFO/TAG       Any tag in the INFO column
           %IUPACGT        Genotype translated to IUPAC ambiguity codes (e.g. M instead of C/A)
           %LINE           Prints the whole line
           %MASK           Indicates presence of the site in other files (with multiple files)
           %N_PASS(expr)   Number of samples that pass the filtering expression (see *&lt;&lt;expressions,EXPRESSIONS&gt;&gt;*)
           %POS0           POS in 0-based coordinates
           %PBINOM(TAG)    Calculate phred-scaled binomial probability, the allele index is determined from GT
           %SAMPLE         Sample name
           %TAG{INT}       Curly brackets to print a subfield (e.g. INFO/TAG{1}, the indexes are 0-based)
           %TBCSQ          Translated FORMAT/BCSQ. See the csq command above for explanation and examples.
           %TGT            Translated genotype (e.g. C/A)
           %TYPE           Variant type (REF, SNP, MNP, INDEL, BND, OTHER)
           %VKX            VariantKey, biallelic hexadecimal encoding of CHROM,POS,REF,ALT (https://github.com/tecnickcom/variantkey)
           []              Format fields must be enclosed in brackets to loop over all samples
           \n              new line
           \t              tab character

           Everything else is printed verbatim.

   <b>Examples:</b>
           # Print chromosome, position, ref allele and the first alternate allele
           bcftools query -f '%CHROM  %POS  %REF  %ALT{0}\n' file.vcf.gz

           # Similar to above, but use tabs instead of spaces, add sample name and genotype
           bcftools query -f '%CHROM\t%POS\t%REF\t%ALT[\t%SAMPLE=%GT]\n' file.vcf.gz

           # Print FORMAT/GT fields followed by FORMAT/GT fields
           bcftools query -f 'GQ:[ %GQ] \t GT:[ %GT]\n' file.vcf

           # Make a BED file: chr, pos (0-based), end pos (1-based), id
           bcftools query -f'%CHROM\t%POS0\t%END\t%ID\n' file.bcf

           # Print only samples with alternate (non-reference) genotypes
           bcftools query -f'[%CHROM:%POS %SAMPLE %GT\n]' -i'GT="alt"' file.bcf

           # Print all samples at sites with at least one alternate genotype
           bcftools view -i'GT="alt"' file.bcf -Ou | bcftools query -f'[%CHROM:%POS %SAMPLE %GT\n]'

           # Print phred-scaled binomial probability from FORMAT/AD tag for all heterozygous genotypes
           bcftools query -i'GT="het"' -f'[%CHROM:%POS %SAMPLE %GT %PBINOM(AD)\n]' file.vcf

           # Print the second value of AC field if bigger than 10. Note the (unfortunate) difference in
           # index subscript notation: formatting expressions (-f) uses "{}" while filtering expressions
           # (-i) use "[]". This is for historic reasons and backward-compatibility.
           bcftools query -f '%AC{1}\n' -i 'AC[1]&gt;10' file.vcf.gz

           # Print all samples at sites where at least one sample has DP=1 or DP=2. In the second case
           # print only samples with DP=1 or DP=2, the difference is in the logical operator used, || vs |.
           bcftools query -f '[%SAMPLE %GT %DP\n]' -i 'FMT/DP=1 || FMT/DP=2' file.vcf
           bcftools query -f '[%SAMPLE %GT %DP\n]' -i 'FMT/DP=1 |  FMT/DP=2' file.vcf

   <b>bcftools</b> <b>reheader</b> <b>[</b><u><b>OPTIONS</b></u><b>]</b> <u><b>file.vcf.gz</b></u>
       Modify header of VCF/BCF files, change sample names.

       <b>-f,</b> <b>--fai</b> <u>FILE</u>
           add to the header contig names and their lengths from the provided fasta index file (.fai). Lengths
           of existing contig lines will be updated and contig lines not present in the fai file will be removed

       <b>-h,</b> <b>--header</b> <u>FILE</u>
           new VCF header

       <b>-o,</b> <b>--output</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>-s,</b> <b>--samples</b> <u>FILE</u>
           new sample names, one name per line, in the same order as they appear in the VCF file. Alternatively,
           only samples which need to be renamed can be listed as "old_name new_name\n" pairs separated by
           whitespaces, each on a separate line. If a sample name contains spaces, the spaces can be escaped
           using the backslash character, for example "Not\ a\ good\ sample\ name".

       <b>-T,</b> <b>--temp-prefix</b> <u>PATH</u>
           this option is ignored, but left for compatibility with earlier versions of bcftools.

       <b>--threads</b> <u>INT</u>
           see <b>Common</b> <b>Options</b>

   <b>bcftools</b> <b>roh</b> <b>[</b><u><b>OPTIONS</b></u><b>]</b> <u><b>file.vcf.gz</b></u>
       A program for detecting runs of homo/autozygosity. Only bi-allelic sites are considered.

   <b>The</b> <b>HMM</b> <b>model:</b>
           Notation:
             D  = Data, AZ = autozygosity, HW = Hardy-Weinberg (non-autozygosity),
             f  = non-ref allele frequency

           Emission probabilities:
             oAZ = P_i(D|AZ) = (1-f)*P(D|RR) + f*P(D|AA)
             oHW = P_i(D|HW) = (1-f)^2 * P(D|RR) + f^2 * P(D|AA) + 2*f*(1-f)*P(D|RA)

           Transition probabilities:
             tAZ = P(AZ|HW)  .. from HW to AZ, the -a parameter
             tHW = P(HW|AZ)  .. from AZ to HW, the -H parameter

             ci  = P_i(C)  .. probability of cross-over at site i, from genetic map
             AZi = P_i(AZ) .. probability of site i being AZ/non-AZ, scaled so that AZi+HWi = 1
             HWi = P_i(HW)

             P_{i+1}(AZ) = oAZ * max[(1 - tAZ * ci) * AZ{i-1} , tAZ * ci * (1-AZ{i-1})]
             P_{i+1}(HW) = oHW * max[(1 - tHW * ci) * (1-AZ{i-1}) , tHW * ci * AZ{i-1}]

   <b>General</b> <b>Options:</b>
       <b>--AF-dflt</b> <u>FLOAT</u>
           in case allele frequency is not known, use the <u>FLOAT</u>. By default, sites where allele frequency cannot
           be determined, or is 0, are skipped.

       <b>--AF-tag</b> <u>TAG</u>
           use the specified INFO tag <u>TAG</u> as an allele frequency estimate instead of the default AC and AN tags.
           Sites which do not have <u>TAG</u> will be skipped.

       <b>--AF-file</b> <u>FILE</u>
           Read allele frequencies from a tab-delimited file containing the columns: CHROM\tPOS\tREF,ALT\tAF.
           The file can be compressed with <b>bgzip</b> and indexed with tabix -s1 -b2 -e2. Sites which are not present
           in the <u>FILE</u> or have different reference or alternate allele will be skipped. Note that such a file
           can be easily created from a VCF using:

               bcftools query -f'%CHROM\t%POS\t%REF,%ALT\t%INFO/TAG\n' file.vcf | bgzip -c &gt; freqs.tab.gz

       <b>-b,</b> <b>--buffer-size</b> <u>INT</u>[,<u>INT</u>]
           when the entire many-sample file cannot fit into memory, a sliding buffer approach can be used. The
           first value is the number of sites to keep in memory. If negative, it is interpreted as the maximum
           memory to use, in MB. The second, optional, value sets the number of overlapping sites. The default
           overlap is set to roughly 1% of the buffer size.

       <b>-e,</b> <b>--estimate-AF</b> <u>FILE</u>
           estimate the allele frequency by recalculating INFO/AC and INFO/AN on the fly, using the specified
           <u>TAG</u> which can be either FORMAT/GT ("GT") or FORMAT/PL ("PL"). If <u>TAG</u> is not given, "GT" is assumed.
           Either all samples ("-") or samples listed in <u>FILE</u> will be included. For example, use "PL,-" to
           estimate AF from FORMAT/PL of all samples. If neither <b>-e</b> nor the other <b>--AF-...</b> options are given,
           the allele frequency is estimated from AC and AN counts which are already present in the INFO field.

       <b>--exclude</b> <u>EXPRESSION</u>
           exclude sites for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.

       <b>-G,</b> <b>--GTs-only</b> <u>FLOAT</u>
           use genotypes (FORMAT/GT fields) ignoring genotype likelihoods (FORMAT/PL), setting PL of unseen
           genotypes to <u>FLOAT</u>. Safe value to use is 30 to account for GT errors.

       <b>--include</b> <u>EXPRESSION</u>
           include only sites for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.

       <b>-I,</b> <b>--skip-indels</b>
           skip indels as their genotypes are usually enriched for errors

       <b>-m,</b> <b>--genetic-map</b> <u>FILE</u>
           genetic map in the format required also by IMPUTE2. Only the first and third column are used
           (position and Genetic_Map(cM)). The <u>FILE</u> can be a single file or a file mask, where string "{CHROM}"
           is replaced with chromosome name.

       <b>-M,</b> <b>--rec-rate</b> <u>FLOAT</u>
           constant recombination rate per bp. In combination with <b>--genetic-map</b>, the <b>--rec-rate</b> parameter is
           interpreted differently, as <u>FLOAT</u>-fold increase of transition probabilities, which allows the model
           to become more sensitive yet still account for recombination hotspots. Note that also the range of
           the values is therefore different in both cases: normally the parameter will be in the range
           (1e-3,1e-9) but with <b>--genetic-map</b> it will be in the range (10,1000).

       <b>-o,</b> <b>--output</b> <u>FILE</u>
           Write output to the <u>FILE</u>, by default the output is printed on stdout

       <b>-O,</b> <b>--output-type</b> <u>s</u>|<u>r</u>[<u>z</u>]
           Generate per-site output (<u>s</u>) or per-region output (<u>r</u>). By default both types are printed and the
           output is uncompressed. Add <u>z</u> for a compressed output.

               # Output fields:
               RG = predicted homo/autozygous regions
                   - Sample
                   - Chromosome
                   - Start
                   - End
                   - Length (bp)
                   - Number of markers
                   - Quality .. average phred score in the region from the forward-backward algorithm

               ST = per-site output showing:
                   - Sample
                   - Chromosome
                   - Position
                   - State .. predicted state from the Viterbi algorithm, 0 for normal (HW, Hardy-Weinberg) or 1 for autozygous (AZ)
                   - Quality .. quality score from the forward-backward algorithm

       <b>-r,</b> <b>--regions</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-R,</b> <b>--regions-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--regions-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-s,</b> <b>--samples</b> <u>LIST</u>
           see <b>Common</b> <b>Options</b>

       <b>-S,</b> <b>--samples-file</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>-t,</b> <b>--targets</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-T,</b> <b>--targets-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--targets-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

   <b>HMM</b> <b>Options:</b>
       <b>-a,</b> <b>--hw-to-az</b> <u>FLOAT</u>
           P(AZ|HW) transition probability from AZ (autozygous) to HW (Hardy-Weinberg) state

       <b>-H,</b> <b>--az-to-hw</b> <u>FLOAT</u>
           P(HW|AZ) transition probability from HW to AZ state

       <b>-V,</b> <b>--viterbi-training</b> <u>FLOAT</u>
           estimate HMM parameters using Baum-Welch algorithm, using the convergence threshold <u>FLOAT</u>, e.g. 1e-10
           (experimental)

   <b>bcftools</b> <b>sort</b> <b>[</b><u><b>OPTIONS</b></u><b>]</b> <b>file.bcf</b>
       <b>-m,</b> <b>--max-mem</b> <u>FLOAT</u>[<u>kMG</u>]
           Maximum memory to use. Approximate, affects the number of temporary files written to the disk. Note
           that if the command fails at this step because of too many open files, your system limit on the
           number of open files ("ulimit") may need to be increased.

       <b>-o,</b> <b>--output</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>-O,</b> <b>--output-type</b> <u>b</u>|<u>u</u>|<u>z</u>|<u>v</u>[0-9]
           see <b>Common</b> <b>Options</b>

       <b>-T,</b> <b>--temp-dir</b> <u>DIR</u>
           Use this directory to store temporary files. If the last six characters of the string DIR are XXXXXX,
           then these are replaced with a string that makes the directory name unique.

       <b>-W</b>[<u>FMT</u>]<b>,</b> <b>-W</b>[=<u>FMT</u>]<b>,</b> <b>--write-index</b>[=<u>FMT</u>]
           Automatically index the output file. <u>FMT</u> is optional and can be one of "tbi" or "csi" depending on
           output file format.

   <b>bcftools</b> <b>stats</b> <b>[</b><u><b>OPTIONS</b></u><b>]</b> <u><b>A.vcf.gz</b></u> <b>[</b><u><b>B.vcf.gz</b></u><b>]</b>
       Parses VCF or BCF and produces text file stats which is suitable for machine processing and can be
       plotted using <b>plot-vcfstats</b>. When two files are given, the program generates separate stats for
       intersection and the complements. By default only sites are compared, <b>-s</b>/<b>-S</b> must given to include also
       sample columns. When one VCF file is specified on the command line, then stats by non-reference allele
       frequency, depth distribution, stats by quality and per-sample counts, singleton stats, etc. are printed.
       When two VCF files are given, then stats such as concordance (Genotype concordance by non-reference
       allele frequency, Genotype concordance by sample, Non-Reference Discordance) and correlation are also
       printed. Per-site discordance (PSD) is also printed in <b>--verbose</b> mode.

       <b>--af-bins</b> <u>LIST</u>|<u>FILE</u>
           comma separated list of allele frequency bins (e.g. 0.1,0.5,1) or a file listing the allele frequency
           bins one per line (e.g. 0.1\n0.5\n1)

       <b>--af-tag</b> <u>TAG</u>
           allele frequency INFO tag to use for binning. By default the allele frequency is estimated from
           AC/AN, if available, or directly from the genotypes (GT) if not.

       <b>-1,</b> <b>--1st-allele-only</b>
           consider only the 1st alternate allele at multiallelic sites

       <b>-c,</b> <b>--collapse</b> <u>snps</u>|<u>indels</u>|<u>both</u>|<u>all</u>|<u>some</u>|<u>none</u>
           see <b>Common</b> <b>Options</b>

       <b>-d,</b> <b>--depth</b> <u>INT</u>,<u>INT</u>,<u>INT</u>
           ranges of depth distribution: min, max, and size of the bin

       <b>--debug</b>
           produce verbose per-site and per-sample output

       <b>-e,</b> <b>--exclude</b> <u>EXPRESSION</u>
           exclude sites for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.

       <b>-E,</b> <b>--exons</b> <u>file.gz</u>
           tab-delimited file with exons for indel frameshifts statistics. The columns of the file are CHR,
           FROM, TO, with 1-based, inclusive, positions. The file is BGZF-compressed and indexed with tabix

               tabix -s1 -b2 -e3 file.gz

       <b>-f,</b> <b>--apply-filters</b> <u>LIST</u>
           see <b>Common</b> <b>Options</b>

       <b>-F,</b> <b>--fasta-ref</b> <u>ref.fa</u>
           faidx indexed reference sequence file to determine INDEL context

       <b>-i,</b> <b>--include</b> <u>EXPRESSION</u>
           include only sites for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.

       <b>-I,</b> <b>--split-by-ID</b>
           collect stats separately for sites which have the ID column set ("known sites") or which do not have
           the ID column set ("novel sites").

       <b>-r,</b> <b>--regions</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-R,</b> <b>--regions-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--regions-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-s,</b> <b>--samples</b> <u>LIST</u>
           see <b>Common</b> <b>Options</b>

       <b>-S,</b> <b>--samples-file</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>

       <b>-t,</b> <b>--targets</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-T,</b> <b>--targets-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--targets-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-u,</b> <b>--user-tstv</b> <u>&lt;TAG[:min:max:n]&gt;</u>
           collect Ts/Tv stats for any tag using the given binning [0:1:100]

       <b>-v,</b> <b>--verbose</b>
           produce verbose per-site and per-sample output

   <b>bcftools</b> <b>view</b> <b>[</b><u><b>OPTIONS</b></u><b>]</b> <u><b>file.vcf.gz</b></u> <b>[</b><u><b>REGION</b></u> <b>[...]]</b>
       View, subset and filter VCF or BCF files by position and filtering expression. Convert between VCF and
       BCF. Former <b>bcftools</b> <b>subset</b>.

   <b>Output</b> <b>options</b>
       <b>-G,</b> <b>--drop-genotypes</b>
           drop individual genotype information (after subsetting if <b>-s</b> option is set)

       <b>-h,</b> <b>--header-only</b>
           output the VCF header only (see also <b>bcftools</b> <b>head</b>)

       <b>-H,</b> <b>--no-header</b>
           suppress the header in VCF output

       <b>--with-header</b>
           output both VCF header and records (this is the default, but the option is useful for explicitness or
           to reset the effects of <b>-h</b> or <b>-H</b>)

       <b>-l,</b> <b>--compression-level</b> [<u>0-9</u>]
           compression level. 0 stands for uncompressed, 1 for best speed and 9 for best compression.

       <b>--no-version</b>
           see <b>Common</b> <b>Options</b>

       <b>-O,</b> <b>--output-type</b> <u>b</u>|<u>u</u>|<u>z</u>|<u>v</u>[0-9]
           see <b>Common</b> <b>Options</b>

       <b>-o,</b> <b>--output</b> <u>FILE</u>: output file name. If not present, the default is to print to standard output (stdout).

       <b>-r,</b> <b>--regions</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-R,</b> <b>--regions-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--regions-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>-t,</b> <b>--targets</b> <u>chr</u>|<u>chr:pos</u>|<u>chr:from-to</u>|<u>chr:from-</u>[,...]
           see <b>Common</b> <b>Options</b>

       <b>-T,</b> <b>--targets-file</b> <u>file</u>
           see <b>Common</b> <b>Options</b>

       <b>--targets-overlap</b> <u>0</u>|<u>1</u>|<u>2</u>
           see <b>Common</b> <b>Options</b>

       <b>--threads</b> <u>INT</u>
           see <b>Common</b> <b>Options</b>

       <b>-W</b>[<u>FMT</u>]<b>,</b> <b>-W</b>[=<u>FMT</u>]<b>,</b> <b>--write-index</b>[=<u>FMT</u>]
           Automatically index the output file. <u>FMT</u> is optional and can be one of "tbi" or "csi" depending on
           output file format.

   <b>Subset</b> <b>options:</b>
       <b>-A,</b> <b>--trim-unseen-alleles</b>
           remove the unseen allele <u>&lt;*&gt;</u> or <u>&lt;NON_REF&gt;</u> at variant sites when the option is given once (-A) or at
           all sites when the options is given twice (<u>-AA</u>).

       <b>-a,</b> <b>--trim-alt-alleles</b>
           remove alleles not seen in the genotype fields from the ALT column. Note that if no alternate allele
           remains after trimming, the record itself is not removed but ALT is set to ".". If the option <b>-s</b> or
           <b>-S</b> is given, removes alleles not seen in the subset. INFO and FORMAT tags declared as Type=A, G or R
           will be trimmed as well.

       <b>--force-samples</b>
           only warn about unknown subset samples

       <b>-I,</b> <b>--no-update</b>
           do not (re)calculate INFO fields for the subset (currently INFO/AC and INFO/AN)

       <b>-s,</b> <b>--samples</b> <u>LIST</u>
           see <b>Common</b> <b>Options</b>. Note that it is possible to create multiple subsets simultaneously using the
           <b>split</b> plugin.

       <b>-S,</b> <b>--samples-file</b> <u>FILE</u>
           see <b>Common</b> <b>Options</b>. Note that it is possible to create multiple subsets simultaneously using the
           <b>split</b> plugin.

   <b>Filter</b> <b>options:</b>
       Note that filter options below dealing with counting the number of alleles will, for speed, first check
       for the values of AC and AN in the INFO column to avoid parsing all the genotype (FORMAT/GT) fields in
       the VCF. This means that filters like <u>--uncalled</u>, --exclude-uncalled', or <u>--min-af</u> <u>0.1</u> will be calculated
       from INFO/AC and INFO/AN when available or FORMAT/GT otherwise. However, it will not attempt to use any
       other existing field, like INFO/AF for example. For that, use <u>--exclude</u> <u>AF&lt;0.1</u> instead.

       Also note that one must be careful when sample subsetting and filtering is performed in a single command
       because the order of internal operations can influence the result. For example, the <b>-i/-e</b> filtering is
       performed before sample removal, but the <b>-P</b> filtering is performed after, and some are inherently
       ambiguous, for example allele counts can be taken from the INFO column when present but calculated on the
       fly when absent. Therefore it is strongly recommended to spell out the required order explicitly by
       separating such commands into two steps. (Make sure to use the <b>-O</b> <b>u</b> option when piping!)

       <b>-c,</b> <b>--min-ac</b> <u>INT</u>[<u>:nref</u>|<u>:alt1</u>|<u>:minor</u>|<u>:major</u>|:'nonmajor']
           minimum allele count (INFO/AC) of sites to be printed. Specifying the type of allele is optional and
           can be set to non-reference (<u>nref</u>, the default), 1st alternate  (<u>alt1</u>), the least frequent (<u>minor</u>),
           the most frequent (<u>major</u>) or sum of all but the most frequent (<u>nonmajor</u>) alleles.

       <b>-C,</b> <b>--max-ac</b> <u>INT</u>[<u>:nref</u>|<u>:alt1</u>|<u>:minor</u>|:'major'|:'nonmajor']
           maximum allele count (INFO/AC) of sites to be printed. Specifying the type of allele is optional and
           can be set to non-reference (<u>nref</u>, the default), 1st alternate  (<u>alt1</u>), the least frequent (<u>minor</u>),
           the most frequent (<u>major</u>) or sum of all but the most frequent (<u>nonmajor</u>) alleles.

       <b>-e,</b> <b>--exclude</b> <u>EXPRESSION</u>
           exclude sites for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.

       <b>-f,</b> <b>--apply-filters</b> <u>LIST</u>
           see <b>Common</b> <b>Options</b>

       <b>-g,</b> <b>--genotype</b> [^][<u>hom</u>|<u>het</u>|<u>miss</u>]
           include only sites with one or more homozygous (<u>hom</u>), heterozygous (<u>het</u>) or missing (<u>miss</u>) genotypes.
           When prefixed with <u>^</u>, the logic is reversed; thus <u>^het</u> excludes sites with heterozygous genotypes.

       <b>-i,</b> <b>--include</b> <u>EXPRESSION</u>
           include sites for which <u>EXPRESSION</u> is true. For valid expressions see <b>EXPRESSIONS</b>.

       <b>-k,</b> <b>--known</b>
           print known sites only (ID column is not ".")

       <b>-m,</b> <b>--min-alleles</b> <u>INT</u>
           print sites with at least <u>INT</u> alleles listed in REF and ALT columns

       <b>-M,</b> <b>--max-alleles</b> <u>INT</u>
           print sites with at most <u>INT</u> alleles listed in REF and ALT columns. Use <b>-m2</b> <b>-M2</b> <b>-v</b> <b>snps</b> to only view
           biallelic SNPs.

       <b>-n,</b> <b>--novel</b>
           print novel sites only (ID column is ".")

       <b>-p,</b> <b>--phased</b>
           print sites where all samples are phased. Haploid genotypes are considered phased. Missing genotypes
           considered unphased unless the phased bit is set.

       <b>-P,</b> <b>--exclude-phased</b>
           exclude sites where all samples are phased

       <b>-q,</b> <b>--min-af</b> <u>FLOAT</u>[<u>:nref</u>|<u>:alt1</u>|<u>:minor</u>|<u>:major</u>|<u>:nonmajor</u>]
           minimum allele frequency (INFO/AC / INFO/AN) of sites to be printed. Specifying the type of allele is
           optional and can be set to non-reference (<u>nref</u>, the default), 1st alternate  (<u>alt1</u>), the least
           frequent (<u>minor</u>), the most frequent (<u>major</u>) or sum of all but the most frequent (<u>nonmajor</u>) alleles.

       <b>-Q,</b> <b>--max-af</b> <u>FLOAT</u>[<u>:nref</u>|<u>:alt1</u>|<u>:minor</u>|<u>:major</u>|<u>:nonmajor</u>]
           maximum allele frequency (INFO/AC / INFO/AN) of sites to be printed. Specifying the type of allele is
           optional and can be set to non-reference (<u>nref</u>, the default), 1st alternate  (<u>alt1</u>), the least
           frequent (<u>minor</u>), the most frequent (<u>major</u>) or sum of all but the most frequent (<u>nonmajor</u>) alleles.

       <b>-u,</b> <b>--uncalled</b>
           print sites without a called genotype, i.e. print sites with all genotypes missing. Note that the
           missingness is determined from INFO/AN and AC tags when available to avoid parsing sample fields.

       <b>-U,</b> <b>--exclude-uncalled</b>
           exclude sites without a called genotype, i.e. print sites with at least one non-missing genotype.
           Note that the missingness is determined from INFO/AN and AC tags when available to avoid parsing
           sample fields.

       <b>-v,</b> <b>--types</b> <u>snps</u>|<u>indels</u>|<u>mnps</u>|<u>other</u>
           comma-separated list of variant types to select. Site is selected if any of the ALT alleles is of the
           type requested. Types are determined by comparing the REF and ALT alleles in the VCF record not INFO
           tags like INFO/INDEL or INFO/VT. Use <b>--include</b> to select based on INFO tags.

       <b>-V,</b> <b>--exclude-types</b> <u>snps</u>|<u>indels</u>|<u>mnps</u>|<u>ref</u>|<u>bnd</u>|<u>other</u>
           comma-separated list of variant types to exclude. Site is excluded if any of the ALT alleles is of
           the type requested. Types are determined by comparing the REF and ALT alleles in the VCF record not
           INFO tags like INFO/INDEL or INFO/VT. Use <b>--exclude</b> to exclude based on INFO tags.

       <b>-x,</b> <b>--private</b>
           print sites where only the subset samples carry an non-reference allele. Requires <b>--samples</b> or
           <b>--samples-file</b>.

       <b>-X,</b> <b>--exclude-private</b>
           exclude sites where only the subset samples carry an non-reference allele

   <b>bcftools</b> <b>help</b> <b>[</b><u><b>COMMAND</b></u><b>]</b> <b>|</b> <b>bcftools</b> <b>--help</b> <b>[</b><u><b>COMMAND</b></u><b>]</b>
       Display  a  brief usage message listing the bcftools commands available. If the name of a command is also
       given, e.g., bcftools help view, the detailed usage message for that particular command is displayed.

   <b>bcftools</b> <b>[</b><u><b>--version</b></u><b>|</b><u><b>-v</b></u><b>]</b>
       Display the version numbers and copyright information for bcftools and the important libraries used by
       bcftools.

   <b>bcftools</b> <b>[</b><u><b>--version-only</b></u><b>]</b>
       Display the full bcftools version number in a machine-readable format.

</pre><h4><b>SCRIPTS</b></h4><pre>
   <b>gff2gff</b>
       Attempts to fix a GFF file to be correctly parsed by <b>csq</b>.

           zcat in.gff.gz | gff2gff | gzip -c &gt; out.gff.gz

   <b>plot-vcfstats</b> <b>[</b><u><b>OPTIONS</b></u><b>]</b> <u><b>file.vchk</b></u> <b>[...]</b>
       Script for processing output of <b>bcftools</b> <b>stats</b>. It can merge results from multiple outputs (useful when
       running the stats for each chromosome separately), plots graphs and creates a PDF presentation.

       <b>-m,</b> <b>--merge</b>
           Merge vcfstats files to STDOUT, skip plotting.

       <b>-p,</b> <b>--prefix</b> <u>DIR</u>
           The output directory. This directory will be created if it does not exist.

       <b>-P,</b> <b>--no-PDF</b>
           Skip the PDF creation step.

       <b>-r,</b> <b>--rasterize</b>
           Rasterize PDF images for faster rendering. This is the default and the opposite of <b>-v,</b> <b>--vectors</b>.

       <b>-s,</b> <b>--sample-names</b>
           Use sample names for xticks rather than numeric IDs.

       <b>-t,</b> <b>--title</b> <u>STRING</u>
           Identify files by these titles in plots. The option can be given multiple times, for each ID in the
           <b>bcftools</b> <b>stats</b> output. If not present, the script will use abbreviated source file names for the
           titles.

       <b>-v,</b> <b>--vectors</b>
           Generate vector graphics for PDF images, the opposite of <b>-r,</b> <b>--rasterize</b>.

       <b>-T,</b> <b>--main-title</b> <u>STRING</u>
           Main title for the PDF.

       <b>Example:</b>

           # Generate the stats
           bcftools stats -s - &gt; file.vchk

           # Plot the stats
           plot-vcfstats -p outdir file.vchk

           # The final looks can be customized by editing the generated
           # 'outdir/plot.py' script and re-running manually
           cd outdir &amp;&amp; python plot.py &amp;&amp; pdflatex summary.tex

</pre><h4><b>FILTERING</b> <b>EXPRESSIONS</b></h4><pre>
       These filtering expressions are accepted by most of the commands.

       <b>Valid</b> <b>expressions</b> <b>may</b> <b>contain:</b>

       •   numerical constants, string constants, file names (indicated by the prefix <u>@</u>)

               1, 1.0, 1e-4
               "String"
               @file_name

       •   arithmetic operators (addition, multiplication, subtraction, division, modulo)

               +, *, -, /, %

       •   comparison operators

               == (same as =), &gt;, &gt;=, &lt;=, &lt;, !=

       •   regex operators "\~" and its negation "!~". The expressions are case sensitive unless "/i" is added.

               INFO/HAYSTACK ~ "needle"
               INFO/HAYSTACK ~ "NEEDless/i"

       •   parentheses

               (, )

       •   logical operators. See also the examples below and the filtering tutorial
           &lt;<a href="http://samtools.github.io/bcftools/howtos/filtering.html">http://samtools.github.io/bcftools/howtos/filtering.html</a>&gt; about the distinction between "&amp;&amp;" vs "&amp;"
           and "||" vs "|".

               &amp;&amp;,  &amp;, ||,  |

       •   INFO tags, FORMAT tags, column names

               INFO/DP or DP
               FORMAT/DV, FMT/DV, or DV
               FILTER, QUAL, ID, CHROM, POS, REF, ALT[0]

       •   the FILTER column can be queried as follows:

               FILTER="PASS"
               FILTER="."
               FILTER="A"          .. exact match, for example "A;B" does not pass
               FILTER="A;B"        .. exact match, "A;B" and "B;A" pass, everything else fails
               FILTER!="A"         .. exact match, for example "A;B" does pass
               FILTER~"A"          .. subset match, for example both "A" and "A;B" pass
               FILTER~"A;B"        .. subset match, pass only if both "A" and "B" are present
               FILTER!~"A"         .. complement match, for example both "A" and "A;B" fail
               FILTER!~"A;B"       .. complement match, fail if both "A" and "B" are present

       •   1 (or 0) to test the presence (or absence) of a flag

               FlagA=1 &amp;&amp; FlagB=0

       •   "." to test missing values

               DP=".", DP!=".", ALT="."

       •   missing genotypes can be matched regardless of phase and ploidy (".|.", ".<a href="file:/.">/.</a>", ".", "0|.") using
           these expressions

               GT="mis", GT~"\.", GT!~"\."

       •   missing genotypes can be matched including the phase and ploidy (".|.", ".<a href="file:/.">/.</a>", ".") using these
           expressions

               GT=".|.", GT=".<a href="file:/.">/.</a>", GT="."

       •   sample genotype: reference (haploid or diploid), alternate (hom or het, haploid or diploid), missing
           genotype, homozygous, heterozygous, haploid, ref-ref hom, alt-alt hom, ref-alt het, alt-alt het,
           haploid ref, haploid alt (case-insensitive)

               GT="ref"
               GT="alt"
               GT="mis"
               GT="hom"
               GT="het"
               GT="hap"
               GT="RR"
               GT="AA"
               GT="RA" or GT="AR"
               GT="Aa" or GT="aA"
               GT="R"
               GT="A"

       •   TYPE for variant type in REF,ALT columns (indel,snp,mnp,ref,bnd,other,overlap, see <b>TERMINOLOGY</b>). Use
           the regex operator "\~" to require at least one allele of the given type or the equal sign "=" to
           require that all alleles are of the given type. Compare

               TYPE="snp"
               TYPE~"snp"
               TYPE!="snp"
               TYPE!~"snp"

       •   array subscripts (0-based), "*" for any element, "-" to indicate a range. Note that for querying
           FORMAT vectors, the colon ":" can be used to select a sample and an element of the vector, as shown
           in the examples below.

               INFO/AF[0] &gt; 0.3             .. first AF value bigger than 0.3
               FORMAT/AD[0:0] &gt; 30          .. first AD value of the first sample bigger than 30
               FORMAT/AD[0:1]               .. first sample, second AD value
               FORMAT/AD[1:0]               .. second sample, first AD value
               DP4[*] == 0                  .. any DP4 value
               FORMAT/DP[0]   &gt; 30          .. DP of the first sample bigger than 30
               FORMAT/DP[1-3] &gt; 10          .. samples 2-4
               FORMAT/DP[1-]  &lt; 7           .. all samples but the first
               FORMAT/DP[0,2-4] &gt; 20        .. samples 1, 3-5
               FORMAT/AD[0:1]               .. first sample, second AD field
               FORMAT/AD[0:*], AD[0:] or AD[0] .. first sample, any AD field
               FORMAT/AD[*:1] or AD[:1]        .. any sample, second AD field
               (DP4[0]+DP4[1])/(DP4[2]+DP4[3]) &gt; 0.3
               CSQ[*] ~ "missense_variant.*deleterious"

       •   in addition to array subscripts shown above, it is possible to subscript arrays of Number=R tags by
           alleles found in FORMAT/GT (starting with version 1.17). For example

               FORMAT/AD[GT] &gt; 10        .. require support of more than 10 reads for each allele
               FORMAT/AD[0:GT] &gt; 10      .. same as above, but in the first sample
               sSUM(FORMAT/AD[GT]) &gt; 20  .. require total sample depth bigger than 20

       •   with many samples it can be more practical to provide a file with sample names, one sample name per
           line

               GT[@samples.txt]="het" &amp; binom(AD)&lt;0.01

       •   function on FORMAT tags (over samples) and INFO tags (over vector fields): maximum; minimum;
           arithmetic mean (AVG is synonymous with MEAN); median; standard deviation from mean; sum; string
           length; absolute value; number of elements:

               MAX, MIN, AVG, MEAN, MEDIAN, STDEV, SUM, STRLEN, ABS, COUNT

           Note that functions above evaluate to a single value across all samples and are intended to select
           sites, not samples, even when applied on FORMAT tags. However, when prefixed with SMPL_ (or "s" for
           brevity, e.g. SMPL_MAX or sMAX), they will evaluate to a vector of per-sample values when applied on
           FORMAT tags:

               SMPL_MAX, SMPL_MIN, SMPL_AVG, SMPL_MEAN, SMPL_MEDIAN, SMPL_STDEV, SMPL_SUM,
               sMAX, sMIN, sAVG, sMEAN, sMEDIAN, sSTDEV, sSUM

       •   two-tailed binomial test. Note that for N=0 the test evaluates to a missing value and when FORMAT/GT
           is used to determine the vector indices, it evaluates to 1 for homozygous genotypes.

               binom(FMT/AD)                .. GT can be used to determine the correct index
               binom(AD[0],AD[1])           .. or the fields can be given explicitly
               phred(binom())               .. the same as binom but phred-scaled

       •   variables calculated on the fly if not present: number of alternate alleles; number of samples; count
           of alternate alleles; minor allele count (similar to AC but is always smaller than 0.5); frequency of
           alternate alleles (AF=AC/AN); frequency of minor alleles (MAF=MAC/AN); number of alleles in called
           genotypes; number of samples with missing genotype; fraction of samples with missing genotype; indel
           length (deletions negative, insertions positive, balanced substitutions zero)

               N_ALT, N_SAMPLES, AC, MAC, AF, MAF, AN, N_MISSING, F_MISSING, ILEN

       •   the number (N_PASS) or fraction (F_PASS) of samples which pass the expression

               N_PASS(GQ&gt;90 &amp; GT!="mis") &gt; 90
               F_PASS(GQ&gt;90 &amp; GT!="mis") &gt; 0.9

       •   custom perl filtering. Note that this command is not compiled in by default, see the section <b>Optional</b>
           <b>Compilation</b> <b>with</b> <b>Perl</b> in the INSTALL file for help and misc/demo-flt.pl for a working example. The
           demo defined the perl subroutine "severity" which can be invoked from the command line as follows:

               perl:path/to/script.pl; perl.severity(INFO/CSQ) &gt; 3

       <b>Notes:</b>

       •   String comparisons and regular expressions are case-insensitive

       •   Comma in strings is interpreted as a separator and when multiple values are compared, the OR logic is
           used. Consequently, the following two expressions are equivalent but not the third:

               -i 'TAG="hello,world"'
               -i 'TAG="hello" || TAG="world"'
               -i 'TAG="hello" &amp;&amp; TAG="world"'

       •   Variables and function names are case-insensitive, but not tag names. For example, "qual" can be used
           instead of "QUAL", "strlen()" instead of "STRLEN()" , but not "dp" instead of "DP".

       •   When querying multiple values, all elements are tested and the OR logic is used on the result. For
           example, when querying "TAG=1,2,3,4", it will be evaluated as follows:

               -i 'TAG[*]=1'   .. true, the record will be printed
               -i 'TAG[*]!=1'  .. true
               -e 'TAG[*]=1'   .. false, the record will be discarded
               -e 'TAG[*]!=1'  .. false
               -i 'TAG[0]=1'   .. true
               -i 'TAG[0]!=1'  .. false
               -e 'TAG[0]=1'   .. false
               -e 'TAG[0]!=1'  .. true

       •   When arithmetic operators are used on vectors A and B, the following logic is used to compute the
           resulting vector C:

           •   C_i = A_i + B_i when length(A)==B(A) and sets length©=length(A)

           •   C_i = A_i + B_0 when length(B)=1 and sets length©=length(A)

           •   C_i = A_0 + B_i when length(A)=1 and sets length©=length(B)

           •   throw an error when length(A)!=length(B) AND length(A)!=1 AND length(B)!=1

       <b>Examples:</b>

           MIN(DV)&gt;5       .. selects the whole site, evaluates min across all values and samples

           SMPL_MIN(DV)&gt;5  .. selects matching samples, evaluates within samples

           MIN(DV/DP)&gt;0.3

           MIN(DP)&gt;10 &amp; MIN(DV)&gt;3

           FMT/DP&gt;10  &amp; FMT/GQ&gt;10 .. both conditions must be satisfied within one sample

           FMT/DP&gt;10 &amp;&amp; FMT/GQ&gt;10 .. the conditions can be satisfied in different samples

           QUAL&gt;10 |  FMT/GQ&gt;10   .. true for sites with QUAL&gt;10 or a sample with GQ&gt;10, but selects only samples with GQ&gt;10

           QUAL&gt;10 || FMT/GQ&gt;10   .. true for sites with QUAL&gt;10 or a sample with GQ&gt;10, plus selects all samples at such sites

           TYPE="snp" &amp;&amp; QUAL&gt;=10 &amp;&amp; (DP4[2]+DP4[3] &gt; 2)

           COUNT(GT="hom")=0      .. no homozygous genotypes at the site

           AVG(GQ)&gt;50             .. average (arithmetic mean) of genotype qualities bigger than 50

           ID=@file               .. selects lines with ID present in the file

           ID!=@<a href="file:~/file">~/file</a>            .. skip lines with ID present in the <a href="file:~/file">~/file</a>

           INFO/TAG=@file         .. selects lines with INFO/TAG value present in the file

           MAF[0]&lt;0.05    .. select rare variants at 5% cutoff

           POS&gt;=100   .. restrict your range query, e.g. 20:100-200 to strictly sites with POS in that range.

       <b>Shell</b> <b>expansion:</b>

       Note that expressions must often be quoted because some characters have special meaning in the shell. An
       example of expression enclosed in single quotes which cause that the whole expression is passed to the
       program as intended:

           bcftools view -i '%ID!="." &amp; MAF[0]&lt;0.01'

       Please refer to the documentation of your shell for details.

</pre><h4><b>TERMINOLOGY</b></h4><pre>
       The program and the documentation uses the following terminology, multiple terms can be used
       interchangeably for the same VCF record type

           REF   ALT
           ---------
           C     . .. reference allele / non-variant site / ref-only site
           C     T         .. SNP or SNV (single-nucleotide polymorphism or variant), used interchangeably
           CC    TT        .. MNP (multi-nucleotide polymorphism)
           CAAA  C         .. indel, deletion (regardless of length)
           C     CAAA      .. indel, insertion (regardless of length)
           C     &lt;*&gt;       .. gVCF block, the allele &lt;*&gt; is a placeholder for alternate allele possibly missed because of low coverage
           C     &lt;NON_REF&gt; .. synonymous to &lt;*&gt;
           C     *         .. overlapping deletion
           C     &lt;INS&gt;     .. symbolic allele, known also as 'other [than above]'

</pre><h4><b>PERFORMANCE</b></h4><pre>
       HTSlib was designed with BCF format in mind. When parsing VCF files, all records are internally converted
       into BCF representation. Simple operations, like removing a single column from a VCF file, can be
       therefore done much faster with standard UNIX commands, such as <b>awk</b> or <b>cut</b>. Therefore it is recommended
       to use BCF as input/output format whenever possible to avoid large overhead of the VCF → BCF → VCF
       conversion.

</pre><h4><b>BUGS</b></h4><pre>
       Please report any bugs you encounter on the github website: <a href="http://github.com/samtools/bcftools">http://github.com/samtools/bcftools</a>

</pre><h4><b>AUTHORS</b></h4><pre>
       Heng Li from the Sanger Institute wrote the original C version of htslib, samtools and bcftools. Bob
       Handsaker from the Broad Institute implemented the BGZF library. Petr Danecek is maintaining and further
       developing bcftools, together with the rest of the samtools team
       &lt;https://www.sanger.ac.uk/tool/samtools-bcftools-htslib&gt;. Many other people contributed to the program
       and to the file format specifications, both directly and indirectly by providing patches, testing and
       reporting bugs. We thank them all.

</pre><h4><b>RESOURCES</b></h4><pre>
       BCFtools GitHub website: <a href="http://github.com/samtools/bcftools">http://github.com/samtools/bcftools</a>

       Samtools GitHub website: <a href="http://github.com/samtools/samtools">http://github.com/samtools/samtools</a>

       HTSlib GitHub website: <a href="http://github.com/samtools/htslib">http://github.com/samtools/htslib</a>

       File format specifications: <a href="http://samtools.github.io/hts-specs">http://samtools.github.io/hts-specs</a>

       BCFtools documentation: <a href="http://samtools.github.io/bcftools">http://samtools.github.io/bcftools</a>

       BCFtools wiki page: https://github.com/samtools/bcftools/wiki

</pre><h4><b>COPYING</b></h4><pre>
       The MIT/Expat License or GPL License, see the LICENSE document for details. Copyright (c) Genome Research
       Ltd.

                                                   2024-09-12                                        <u><a href="../man1/BCFTOOLS.1.html">BCFTOOLS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>