<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>cnvkit_call - Call copy number variants from segmented log2 ratios.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/cnvkit">cnvkit_0.9.12-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       cnvkit_call - Call copy number variants from segmented log2 ratios.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: cnvkit.py call [-h] [--center [{mean,median,mode,biweight}]]

       [--center-at CENTER_AT] [--filter {ampdel,cn,ci,sem}]
              [-m    {threshold,clonal,none}]    [-t    THRESHOLDS]    [--ploidy   PLOIDY]   [--purity   PURITY]
              [--drop-low-coverage] [-x {m,y,male,Male,f,x,female,Female}] [-y] [-o FILENAME] [-v FILENAME]  [-i
              SAMPLE_ID] [-n NORMAL_ID] [--min-variant-depth MIN_VARIANT_DEPTH] [-z [ALT_FREQ]] filename

   <b>positional</b> <b>arguments:</b>
       filename
              Copy ratios (.cnr or .cns).

   <b>options:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>--center</b> [{mean,median,mode,biweight}]
              Re-center  the log2 ratio values using this estimator of the center or average value. ('median' if
              no argument given.)

       <b>--center-at</b> CENTER_AT
              Subtract a constant number from all log2 ratios. For "manual" re-centering, in case  the  <b>--center</b>
              option gives unsatisfactory results.)

       <b>--filter</b> {ampdel,cn,ci,sem}
              Merge segments flagged by the specified filter(s) with the adjacent segment(s).

       <b>-m</b> {threshold,clonal,none}, <b>--method</b> {threshold,clonal,none}
              Calling method. [Default: threshold]

       <b>-t</b> THRESHOLDS, <b>--thresholds</b> THRESHOLDS
              Hard thresholds for calling each integer copy number, separated by commas. Use the '=' sign on the
              command line, e.g.: <b>-t</b>=<u>-1</u>,0,1 [Default: <b>-1</b>.1,-0.25,0.2,0.7]

       <b>--ploidy</b> PLOIDY
              Ploidy of the sample cells. [Default: 2]

       <b>--purity</b> PURITY
              Estimated tumor cell fraction, a.k.a. purity or cellularity.

       <b>--drop-low-coverage</b>
              Drop  very-low-coverage bins before segmentation to avoid false-positive deletions in poor-quality
              tumor samples.

       <b>-x</b> {m,y,male,Male,f,x,female,Female}, <b>--sample-sex</b> {m,y,male,Male,f,x,female,Female}, <b>-g</b>
       {m,y,male,Male,f,x,female,Female}, <b>--gender</b> {m,y,male,Male,f,x,female,Female}
              Specify the sample's chromosomal sex as male or female. (Otherwise guessed from X and Y coverage).

       <b>-y</b>, <b>--male-reference</b>, <b>--haploid-x-reference</b>
              Was a male reference used? If so, expect half ploidy on chrX and chrY; otherwise,  only  chrY  has
              half  ploidy.  In CNVkit, if a male reference was used, the "neutral" copy number (ploidy) of chrX
              is 1; chrY is haploid for either reference sex.

       <b>-o</b> FILENAME, <b>--output</b> FILENAME
              Output table file name (CNR-like table of segments, .cns).

   <b>To</b> <b>additionally</b> <b>process</b> <b>SNP</b> <b>b-allele</b> <b>frequencies</b> <b>for</b> <b>allelic</b> <b>copy</b> <b>number:</b>
       <b>-v</b> FILENAME, <b>--vcf</b> FILENAME
              VCF file name containing variants for calculation of b-allele frequencies.

       <b>-i</b> SAMPLE_ID, <b>--sample-id</b> SAMPLE_ID
              Name of the sample in the VCF (<b>-v</b>/--vcf) to use for b-allele frequency extraction.

       <b>-n</b> NORMAL_ID, <b>--normal-id</b> NORMAL_ID
              Corresponding normal sample ID in the input VCF (<b>-v</b>/--vcf). This sample is  used  to  select  only
              germline SNVs to calculate b-allele frequencies.

       <b>--min-variant-depth</b> MIN_VARIANT_DEPTH
              Minimum read depth for a SNV to be used in the b-allele frequency calculation. [Default: 20]

       <b>-z</b> [ALT_FREQ], <b>--zygosity-freq</b> [ALT_FREQ]
              Ignore  VCF's genotypes (GT field) and instead infer zygosity from allele frequencies. [Default if
              used without a number: 0.25]

cnvkit.py call 0.9.10                               July 2023                                     <u><a href="../man1/CNVKIT_CALL.1.html">CNVKIT_CALL</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>