<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>svim - Structural variant caller for long sequencing reads</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/svim">svim_2.0.0-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       svim - Structural variant caller for long sequencing reads

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: svim [-h] [--version] {reads,alignment} ...

       SVIM  (pronounced  SWIM)  is  a structural variant caller for long reads.  It discriminates six different
       variant  classes:  deletions,  tandem  and  interspersed   duplications,   inversions,   insertions   and
       translocations. SVIM is unique in its capability of extracting both the genomic origin and destination of
       duplications.

       SVIM consists of four major steps: - COLLECT detects signatures for SVs in long read alignments - CLUSTER
       merges  signatures  that come from the same SV - COMBINE combines clusters from different genomic regions
       and classifies them into distinct SV types - GENOTYPE uses alignments spanning  SVs  to  determine  their
       genotype

       SVIM  can  process  two  types  of input. Firstly, it can detect SVs from raw reads by aligning them to a
       given reference genome first ("SVIM.py reads [options] working_dir reads genome").  Alternatively, it can
       detect SVs from existing reads alignments in SAM/BAM format  ("SVIM.py  alignment  [options]  working_dir
       bam_file").

   <b>positional</b> <b>arguments:</b>
       {reads,alignment}
              modes

       reads  Detect SVs from raw reads. Align reads to given reference genome first.

       alignment
              Detect SVs from an existing alignment

   <b>optional</b> <b>arguments:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>--version</b>, <b>-v</b>
              show program's version number and exit

</pre><h4><b>AUTHOR</b></h4><pre>
        This manpage was written by Nilesh Patra for the Debian distribution and
        can be used for any other usage of the program.

svim 2.0.0                                          July 2021                                            <u><a href="../man1/SVIM.1.html">SVIM</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>