<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>varscan - variant detection in next-generation sequencing data</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/varscan">varscan_2.4.3+dfsg-4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       varscan - variant detection in next-generation sequencing data

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>varscan</b> [<b>COMMAND</b>] [OPTIONS]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Variant  detection  in  massively  parallel sequencing. For one sample, calls SNPs, indels, and consensus
       genotypes. For tumor-normal pairs, further classifies each variant as Germline, Somatic, or LOH, and also
       detects somatic copy number changes.

</pre><h4><b>USAGE</b></h4><pre>
   <b>COMMANDS:</b>
       <b>pileup2snp</b>
              Identify SNPs from a pileup file

       <b>pileup2indel</b>
              Identify indels a pileup file

       <b>pileup2cns</b>
              Call consensus and variants from a pileup file

       <b>mpileup2snp</b>
              Identify SNPs from an mpileup file

       <b>mpileup2indel</b>
              Identify indels an mpileup file

       <b>mpileup2cns</b>
              Call consensus and variants from an mpileup file

       <b>somatic</b>
              Call germline/somatic variants from tumor-normal pileups

       <b>copynumber</b>
              Determine relative tumor copy number from tumor-normal pileups

       <b>readcounts</b>
              Obtain read counts for a list of variants from a pileup file

       <b>filter</b> Filter SNPs by coverage, frequency, p-value, etc.

       <b>somaticFilter</b>
              Filter somatic variants for clusters/indels

       <b>processSomatic</b>
              Isolate Germline/LOH/Somatic calls from output

       <b>copyCaller</b>
              GC-adjust and process copy number changes from VarScan copynumber output

       <b>compare</b>
              Compare two lists of positions/variants

       <b>limit</b>  Restrict pileup/snps/indels to ROI positions

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <a href="http://varscan.sourceforge.net/using-varscan.html">http://varscan.sourceforge.net/using-varscan.html</a>

varscan 2.3.6                                      April 2014                                         <u><a href="../man1/VARSCAN.1.html">VARSCAN</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>