<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>remap - Display restriction enzyme binding sites in a nucleotide sequence</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/emboss">emboss_6.6.0+dfsg-15ubuntu2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       remap - Display restriction enzyme binding sites in a nucleotide sequence

</pre><h4><b>SYNOPSIS</b></h4><pre>

       <b>remap</b> <b>-sequence</b> <u>seqall</u> <b>-mfile</b> <u>datafile</u> <b>-enzymes</b> <u>string</u> <b>-sitelen</b> <u>integer</u> [<b>-mincuts</b> <u>integer</u>]
             [<b>-maxcuts</b> <u>integer</u>] [<b>-single</b> <u>boolean</u>] [<b>-blunt</b> <u>boolean</u>] [<b>-sticky</b> <u>boolean</u>] [<b>-ambiguity</b> <u>boolean</u>]
             [<b>-plasmid</b> <u>boolean</u>] [<b>-methylation</b> <u>boolean</u>] [<b>-commercial</b> <u>boolean</u>] [<b>-table</b> <u>list</u>] [<b>-frame</b> <u>list</u>]
             <b>-outfile</b> <u>outfile</u> [<b>-cutlist</b> <u>boolean</u>] [<b>-flatreformat</b> <u>boolean</u>] [<b>-limit</b> <u>boolean</u>] <b>-translation</b> <u>boolean</u>
             <b>-reverse</b> <u>boolean</u> <b>-orfminsize</b> <u>integer</u> <b>-uppercase</b> <u>range</u> <b>-highlight</b> <u>range</u> <b>-threeletter</b> <u>boolean</u>
             <b>-number</b> <u>boolean</u> <b>-width</b> <u>integer</u> <b>-length</b> <u>integer</u> <b>-margin</b> <u>integer</u> <b>-name</b> <u>boolean</u> <b>-description</b> <u>boolean</u>
             <b>-offset</b> <u>integer</u> <b>-html</b> <u>boolean</u>

       <b>remap</b> <b>-help</b>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>remap</b> is a command line program from EMBOSS (“the European Molecular Biology Open Software Suite”). It is
       part of the "Display,Nucleic:Restriction,Nucleic:Translation" command group(s).

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Input</b> <b>section</b>
       <b>-sequence</b> <u>seqall</u>

       <b>-mfile</b> <u>datafile</u>
           Default value: Emethylsites.dat

   <b>Required</b> <b>section</b>
       <b>-enzymes</b> <u>string</u>
           The name 'all' reads in all enzyme names from the REBASE database. You can specify enzymes by giving
           their names with commas between then, such as: 'HincII,hinfI,ppiI,hindiii'. The case of the names is
           not important. You can specify a file of enzyme names to read in by giving the name of the file
           holding the enzyme names with a '@' character in front of it, for example, '@enz.list'. Blank lines
           and lines starting with a hash character or '!' are ignored and all other lines are concatenated
           together with a comma character ',' and then treated as the list of enzymes to search for. An example
           of a file of enzyme names is: ! my enzymes HincII, ppiII ! other enzymes hindiii HinfI PpiI Default
           value: all

       <b>-sitelen</b> <u>integer</u>
           This sets the minimum length of the restriction enzyme recognition site. Any enzymes with sites
           shorter than this will be ignored. Default value: 4

   <b>Additional</b> <b>section</b>
       <b>-mincuts</b> <u>integer</u>
           This sets the minimum number of cuts for any restriction enzyme that will be considered. Any enzymes
           that cut fewer times than this will be ignored. Default value: 1

       <b>-maxcuts</b> <u>integer</u>
           This sets the maximum number of cuts for any restriction enzyme that will be considered. Any enzymes
           that cut more times than this will be ignored. Default value: 2000000000

       <b>-single</b> <u>boolean</u>
           If this is set then this forces the values of the mincuts and maxcuts qualifiers to both be 1. Any
           other value you may have set them to will be ignored. Default value: N

       <b>-blunt</b> <u>boolean</u>
           This allows those enzymes which cut at the same position on the forward and reverse strands to be
           considered. Default value: Y

       <b>-sticky</b> <u>boolean</u>
           This allows those enzymes which cut at different positions on the forward and reverse strands,
           leaving an overhang, to be considered. Default value: Y

       <b>-ambiguity</b> <u>boolean</u>
           This allows those enzymes which have one or more 'N' ambiguity codes in their pattern to be
           considered Default value: Y

       <b>-plasmid</b> <u>boolean</u>
           If this is set then this allows searches for restriction enzyme recognition site and cut positions
           that span the end of the sequence to be considered. Default value: N

       <b>-methylation</b> <u>boolean</u>
           If this is set then RE recognition sites will not match methylated bases. Default value: N

       <b>-commercial</b> <u>boolean</u>
           If this is set, then only those enzymes with a commercial supplier will be searched for. This
           qualifier is ignored if you have specified an explicit list of enzymes to search for, rather than
           searching through 'all' the enzymes in the REBASE database. It is assumed that, if you are asking for
           an explicit enzyme, then you probably know where to get it from and so all enzymes names that you
           have asked to be searched for, and which cut, will be reported whether or not they have a commercial
           supplier. Default value: Y

       <b>-table</b> <u>list</u>

       <b>-frame</b> <u>list</u>
           This allows you to specify the frames that are translated. If you are not displaying cut sites on the
           reverse sense, then the reverse sense translations will not be displayed even if you have requested
           frames 4, 5 or 6. By default, all six frames will be displayed. Default value: 6

   <b>Output</b> <b>section</b>
       <b>-outfile</b> <u>outfile</u>

       <b>-cutlist</b> <u>boolean</u>
           This produces lists in the output of the enzymes that cut, those that cut but are excluded because
           that cut fewer times than mincut or more times than maxcut and those enzymes that do not cut. Default
           value: Y

       <b>-flatreformat</b> <u>boolean</u>
           This changes the output format to one where the recognition site is indicated by a row of '==='
           characters and the cut site is pointed to by a '&gt;' character in the forward sense, or a '&lt;' in the
           reverse sense strand. Default value: N

       <b>-limit</b> <u>boolean</u>
           This limits the reporting of enzymes to just one enzyme from each group of isoschizomers. The enzyme
           chosen to represent an isoschizomer group is the prototype indicated in the data file 'embossre.equ',
           which is created by the program 'rebaseextract'. If you prefer different prototypes to be used, make
           a copy of embossre.equ in your home directory and edit it. If this value is set to be false then all
           of the input enzymes will be reported. You might like to set this to false if you are supplying an
           explicit set of enzymes rather than searching 'all' of them. Default value: Y

       <b>-translation</b> <u>boolean</u>
           This displays the 6-frame translations of the sequence in the output. Default value: Y

       <b>-reverse</b> <u>boolean</u>
           This displays the cut sites and translation of the reverse sense. Default value: Y

       <b>-orfminsize</b> <u>integer</u>
           This sets the minimum size of Open Reading Frames (ORFs) to display in the translations. All other
           translation regions are masked by changing the amino acids to '-' characters.

       <b>-uppercase</b> <u>range</u>
           Regions to put in uppercase. If this is left blank, then the sequence case is left alone. A set of
           regions is specified by a set of pairs of positions. The positions are integers. They are separated
           by any non-digit, non-alpha character. Examples of region specifications are: 24-45, 56-78 1:45,
           67=99;765..888 1,5,8,10,23,45,57,99

       <b>-highlight</b> <u>range</u>
           Regions to colour if formatting for HTML. If this is left blank, then the sequence is left alone. A
           set of regions is specified by a set of pairs of positions. The positions are integers. They are
           followed by any valid HTML font colour. Examples of region specifications are: 24-45 blue 56-78
           orange 1-100 green 120-156 red A file of ranges to colour (one range per line) can be specified as
           '@filename'.

       <b>-threeletter</b> <u>boolean</u>
           Default value: N

       <b>-number</b> <u>boolean</u>
           Default value: N

       <b>-width</b> <u>integer</u>
           Default value: 60

       <b>-length</b> <u>integer</u>

       <b>-margin</b> <u>integer</u>
           Default value: 10

       <b>-name</b> <u>boolean</u>
           Set this to be false if you do not wish to display the ID name of the sequence Default value: Y

       <b>-description</b> <u>boolean</u>
           Set this to be false if you do not wish to display the description of the sequence Default value: Y

       <b>-offset</b> <u>integer</u>
           Default value: 1

       <b>-html</b> <u>boolean</u>
           Default value: N

</pre><h4><b>BUGS</b></h4><pre>
       Bugs can be reported to the Debian Bug Tracking system (<a href="http://bugs.debian.org/emboss">http://bugs.debian.org/emboss</a>), or directly to
       the EMBOSS developers (<a href="http://sourceforge.net/tracker/">http://sourceforge.net/tracker/</a>?group_id=93650&amp;atid=605031).

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       remap is fully documented via the <b><a href="../man1/tfm.1.html">tfm</a></b>(1) system.

</pre><h4><b>AUTHOR</b></h4><pre>
       <b>Debian</b> <b>Med</b> <b>Packaging</b> <b>Team</b> &lt;<a href="mailto:debian-med-packaging@lists.alioth.debian.org">debian-med-packaging@lists.alioth.debian.org</a>&gt;
           Wrote the script used to autogenerate this manual page.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       This manual page was autogenerated from an Ajax Control Definition of the EMBOSS package. It can be
       redistributed under the same terms as EMBOSS itself.

EMBOSS 6.4.0                                       05/11/2012                                          <u><a href="../man1e/REMAP.1e.html">REMAP</a></u>(1e)
</pre>
 </div>
</div></section>
</div>
</body>
</html>