<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>restrict - Report restriction enzyme cleavage sites in a nucleotide sequence</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/emboss">emboss_6.6.0+dfsg-15ubuntu2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       restrict - Report restriction enzyme cleavage sites in a nucleotide sequence

</pre><h4><b>SYNOPSIS</b></h4><pre>

       <b>restrict</b> <b>-sequence</b> <u>seqall</u> <b>-datafile</b> <u>datafile</u> <b>-mfile</b> <u>datafile</u> <b>-sitelen</b> <u>integer</u> <b>-enzymes</b> <u>string</u>
                <b>-min</b> <u>integer</u> <b>-max</b> <u>integer</u> <b>-solofragment</b> <u>boolean</u> <b>-single</b> <u>boolean</u> <b>-blunt</b> <u>boolean</u> <b>-sticky</b> <u>boolean</u>
                <b>-ambiguity</b> <u>boolean</u> <b>-plasmid</b> <u>boolean</u> <b>-methylation</b> <u>boolean</u> <b>-commercial</b> <u>boolean</u> <b>-limit</b> <u>boolean</u>
                <b>-alphabetic</b> <u>boolean</u> <b>-fragments</b> <u>boolean</u> <b>-name</b> <u>boolean</u> <b>-outfile</b> <u>report</u>

       <b>restrict</b> <b>-help</b>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>restrict</b> is a command line program from EMBOSS (“the European Molecular Biology Open Software Suite”). It
       is part of the "Nucleic:Restriction" command group(s).

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Input</b> <b>section</b>
       <b>-sequence</b> <u>seqall</u>

       <b>-datafile</b> <u>datafile</u>

       <b>-mfile</b> <u>datafile</u>
           Default value: Emethylsites.dat

   <b>Required</b> <b>section</b>
       <b>-sitelen</b> <u>integer</u>
           This sets the minimum length of the restriction enzyme recognition site. Any enzymes with sites
           shorter than this will be ignored. Default value: 4

       <b>-enzymes</b> <u>string</u>
           The name 'all' reads in all enzyme names from the REBASE database. You can specify enzymes by giving
           their names with commas between then, such as: 'HincII,hinfI,ppiI,hindiii'. The case of the names is
           not important. You can specify a file of enzyme names to read in by giving the name of the file
           holding the enzyme names with a '@' character in front of it, for example, '@enz.list'. Blank lines
           and lines starting with a hash character or '!' are ignored and all other lines are concatenated
           together with a comma character ',' and then treated as the list of enzymes to search for. An example
           of a file of enzyme names is: ! my enzymes HincII, ppiII ! other enzymes hindiii HinfI PpiI Default
           value: all

   <b>Advanced</b> <b>section</b>
       <b>-min</b> <u>integer</u>
           This sets the minimum number of cuts for any restriction enzyme that will be considered. Any enzymes
           that cut fewer times than this will be ignored. Default value: 1

       <b>-max</b> <u>integer</u>
           This sets the maximum number of cuts for any restriction enzyme that will be considered. Any enzymes
           that cut more times than this will be ignored. Default value: 2000000000

       <b>-solofragment</b> <u>boolean</u>
           This gives the fragment lengths of the forward sense strand produced by complete restriction by each
           restriction enzyme on its own. Results are added to the tail section of the report. Default value: N

       <b>-single</b> <u>boolean</u>
           If this is set then this forces the values of the mincuts and maxcuts qualifiers to both be 1. Any
           other value you may have set them to will be ignored. Default value: N

       <b>-blunt</b> <u>boolean</u>
           This allows those enzymes which cut at the same position on the forward and reverse strands to be
           considered. Default value: Y

       <b>-sticky</b> <u>boolean</u>
           This allows those enzymes which cut at different positions on the forward and reverse strands,
           leaving an overhang, to be considered. Default value: Y

       <b>-ambiguity</b> <u>boolean</u>
           This allows those enzymes which have one or more 'N' ambiguity codes in their pattern to be
           considered Default value: Y

       <b>-plasmid</b> <u>boolean</u>
           If this is set then this allows searches for restriction enzyme recognition site and cut positions
           that span the end of the sequence to be considered. Default value: N

       <b>-methylation</b> <u>boolean</u>
           If this is set then RE recognition sites will not match methylated bases. Default value: N

       <b>-commercial</b> <u>boolean</u>
           If this is set, then only those enzymes with a commercial supplier will be searched for. This
           qualifier is ignored if you have specified an explicit list of enzymes to search for, rather than
           searching through 'all' the enzymes in the REBASE database. It is assumed that, if you are asking for
           an explicit enzyme, then you probably know where to get it from and so all enzymes names that you
           have asked to be searched for, and which cut, will be reported whether or not they have a commercial
           supplier. Default value: Y

   <b>Output</b> <b>section</b>
       <b>-limit</b> <u>boolean</u>
           This limits the reporting of enzymes to just one enzyme from each group of isoschizomers. The enzyme
           chosen to represent an isoschizomer group is the prototype indicated in the data file 'embossre.equ',
           which is created by the program 'rebaseextract'. If you prefer different prototypes to be used, make
           a copy of embossre.equ in your home directory and edit it. If this value is set to be false then all
           of the input enzymes will be reported. You might like to set this to false if you are supplying an
           explicit set of enzymes rather than searching 'all' of them. Default value: Y

       <b>-alphabetic</b> <u>boolean</u>
           Default value: N

       <b>-fragments</b> <u>boolean</u>
           This gives the fragment lengths of the forward sense strand produced by complete restriction using
           all of the input enzymes together. Results are added to the tail section of the report. Default
           value: N

       <b>-name</b> <u>boolean</u>
           Default value: N

       <b>-outfile</b> <u>report</u>

</pre><h4><b>BUGS</b></h4><pre>
       Bugs can be reported to the Debian Bug Tracking system (<a href="http://bugs.debian.org/emboss">http://bugs.debian.org/emboss</a>), or directly to
       the EMBOSS developers (<a href="http://sourceforge.net/tracker/">http://sourceforge.net/tracker/</a>?group_id=93650&amp;atid=605031).

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       restrict is fully documented via the <b><a href="../man1/tfm.1.html">tfm</a></b>(1) system.

</pre><h4><b>AUTHOR</b></h4><pre>
       <b>Debian</b> <b>Med</b> <b>Packaging</b> <b>Team</b> &lt;<a href="mailto:debian-med-packaging@lists.alioth.debian.org">debian-med-packaging@lists.alioth.debian.org</a>&gt;
           Wrote the script used to autogenerate this manual page.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       This manual page was autogenerated from an Ajax Control Definition of the EMBOSS package. It can be
       redistributed under the same terms as EMBOSS itself.

EMBOSS 6.4.0                                       05/11/2012                                       <u><a href="../man1e/RESTRICT.1e.html">RESTRICT</a></u>(1e)
</pre>
 </div>
</div></section>
</div>
</body>
</html>