<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::FeatureIO::ptt - read/write features in PTT format</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-featureio-perl">libbio-featureio-perl_1.6.905-2_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::FeatureIO::ptt - read/write features in PTT format

</pre><h4><b>SYNOPSIS</b></h4><pre>
        # read features
        my $fin = Bio::FeatureIO-&gt;new(-file=&gt;'genes.ptt', -format=&gt;'ptt');
        my @cds;
        while (my $f = $fin-&gt;next_feature) {
          push @cds, $f if $f-&gt;strand &gt; 0;
        }

        # write features (NOT IMPLEMENTED)
        my $fout = Bio::FeatureIO-&gt;new(-fh=&gt;\*STDOUT, -format=&gt;'ptt');
        for my $f (@cds) {
          $fout-&gt;write_feature($f);
        }

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The PTT file format is a table of protein features.  It is used mainly by NCBI who produce PTT files for
       all their published genomes found in &lt;<a href="ftp://ftp.ncbi.nih.gov/genomes/">ftp://ftp.ncbi.nih.gov/genomes/</a>&gt;.  It has the following format:

       Line 1
           Description of sequence to which the features belong
            eg. "Leptospira interrogans chromosome II, complete sequence - 0..358943"

           It  is  usually  equivalent to the DEFINITION line of a Genbank file, with the length of the sequence
           appended. It is unclear why "0" is used as a starting range, it should be "1".

       Line 2
           Number of feature lines in the table
            eg. "367 proteins"

       Line 3
           Column headers, tab separated
            eg. "Location  Strand  Length  PID Gene  Synonym Code  COG Product"

            Location : "begin..end" span of feature
            Strand   : "+" or "-"
            Length   : number of amino acids excluding the stop codon
            PID      : analogous to Genbank /db_xref="GI:xxxxxxxxx"
            Gene     : analogous to Genbank /gene="xxxx"
            Synonym  : analogous to Genbank /locus_tag="xxxx"
            Synonym  : analogous to Genbank /locus_tag="xxxx"
            COG      : CDD COG code with COG letter categories appended
            Product  : analogous to Genbank /product="xxxx"

       Line 4 onwards
           Feature lines, nine columns, tab separated, "-" used for empty fields
            eg. "2491..3423  + 310 24217063  metF  LB002 - COG0685E  5,10-methylenetetrahydrofolate reductase"

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your  comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather  than  to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs  and  their  resolution.
       Bug reports can be submitted via the web:

         <a href="http://bugzilla.open-bio.org/">http://bugzilla.open-bio.org/</a>

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Torsten</b> <b>Seemann</b></h4><pre>
       Email torsten.seemann AT infotech.monash.edu.au

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
       Based on bed.pm and gff.pm by Allen Day.

</pre><h4><b>APPENDIX</b></h4><pre>
       The  rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>_initialize</b>
        Title   : _initialize
        Function: Reading? parses the header of the input
                  Writing?

   <b>next_feature</b>
        Title   : next_feature
        Usage   : $io-&gt;next_feature()
        Function: read the next feature from the PTT file
        Example :
        Args    :
        Returns : Bio::SeqFeatureI object

   <b>write_feature</b> <b>(NOT</b> <b>IMPLEMENTED)</b>
        Title   : write_feature
        Usage   : $io-&gt;write_feature($feature)
        Function: write a Bio::SeqFeatureI object in PTT format
        Example :
        Args    : Bio::SeqFeatureI object
        Returns :

   <b>description</b>
        Title   : description
        Usage   : $obj-&gt;description($newval)
        Function: set/get the PTT file description for/from line one
        Example :
        Returns : value of description (a scalar)
        Args    : on set, new value (a scalar or undef, optional)

   <b>protein_count</b>
        Title   : protein_count
        Usage   : $obj-&gt;protein_count($newval)
        Function: set/get the PTT protein count for/from line two
        Example :
        Args    : on set, new value (a scalar or undef, optional)
        Returns : value of protein_count (a scalar)

perl v5.30.0                                       2020-01-13                           <u>Bio::FeatureIO::<a href="../man3pm/ptt.3pm.html">ptt</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>