<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>samtools-collate - shuffles and groups reads together by their names</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/samtools">samtools_1.21-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       samtools-collate - shuffles and groups reads together by their names

</pre><h4><b>SYNOPSIS</b></h4><pre>
       samtools collate [<u>options</u>] <u>in.sam</u>|<u>in.bam</u>|<u>in.cram</u> [<u>&lt;prefix&gt;</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Shuffles  and  groups  reads  together  by  their names.  A faster alternative to a full query name sort,
       <b>collate</b> ensures that reads of the same name are grouped together in contiguous groups, but  doesn't  make
       any guarantees about the order of read names between groups.

       The  output  from this command should be suitable for any operation that requires all reads from the same
       template to be grouped together.

       Temporary files are written to &lt;prefix&gt;, specified either as the last argument or with the <b>-T</b> option.  If
       prefix is unspecified then one will be derived from the output filename (<b>-o</b> option).  If no  output  file
       was  given then the <b>TMPDIR</b> environment variable will be used, and finally if that is unset then "<a href="file:/tmp">/tmp</a>" is
       used.

       Conversely, if prefix is specified but no output filename has been given then the output will be  written
       to &lt;prefix&gt;.&lt;fmt&gt; where &lt;fmt&gt; is appropriate to the file format is use (e.g. "bam" or "cram").

       Using  <b>-f</b> for fast mode will output <b>only</b> primary alignments that have either the READ1 <b>or</b> READ2 flags set
       (but not both).  Any other alignment records  will  be  filtered  out.   The  collation  will  only  work
       correctly if there are no more than two reads for any given QNAME after filtering.

       Fast  mode keeps a buffer of alignments in memory so that it can write out most pairs as soon as they are
       found instead of storing them in temporary files.  This allows collate to avoid some work and  so  finish
       more  quickly  compared  to  the  standard  mode.  The number of alignments held can be changed using <b>-r</b>,
       storing more alignments uses more memory but increases the number of pairs that can be written early.

       While collate normally randomises the ordering of read pairs, fast  mode  does  not.   Position-dependent
       biases  that  would  normally  be broken up can remain in the fast collate output.  It is therefore not a
       good idea to use fast mode when preparing data for programs that expect randomly  ordered  paired  reads.
       For  example  using fast collate instead of the standard mode may lead to significantly different results
       from aligners that estimate library insert sizes on batches of reads.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-O</b>      Output to stdout.  This option cannot be used with <b>-o</b>.

       <b>-o</b> <u>FILE</u> Write output to FILE.  This option cannot be used with <b>-O</b>.  If unspecified and <b>-O</b> is not set, the
               temporary file &lt;prefix&gt; is used, appended by the the appropriate file-format suffix.

       <b>-T</b> <u>PREFIX</u>
               Use <u>PREFIX</u> for temporary files.  This is the same as specifying <u>PREFIX</u> as the  last  argument  on
               the command line.  This option is included for consistency with <b>samtools</b> <b>sort</b>.

       <b>-u</b>      Write uncompressed BAM output

       <b>-l</b> <u>INT</u>  Compression level.  [1]

       <b>-n</b> <u>INT</u>  Number of temporary files to use.  [64]

       <b>-f</b>      Fast mode (primary alignments only).

       <b>-r</b> <u>INT</u>  Number of reads to store in memory (for use with -f).  [10000]

       <b>--no-PG</b> Do not add a @PG line to the header of the output file.

       <b>-@,</b> <b>--threads</b> <u>INT</u>
               Number of input/output compression threads to use in addition to main thread [0].

</pre><h4><b>AUTHOR</b></h4><pre>
       Written by Heng Li from the Sanger Institute and extended by Andrew Whitwham.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/samtools.1.html">samtools</a></u>(1), <u><a href="../man1/samtools-sort.1.html">samtools-sort</a></u>(1)

       Samtools website: &lt;<a href="http://www.htslib.org/">http://www.htslib.org/</a>&gt;

samtools-1.21                                   12 September 2024                            <u><a href="../man1/samtools-collate.1.html">samtools-collate</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>