<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>alignment-cat - Concatenate several alignments (with the same sequence names) end-to-end.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/bali-phy">bali-phy_3.6.1+dfsg-3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       <b>alignment-cat</b> - Concatenate several alignments (with the same sequence names) end-to-end.

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>alignment-cat</b> <u>file1</u> [<u>file2</u> ...]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Concatenate several alignments (with the same sequence names) end-to-end.

</pre><h4><b>ALL</b> <b>OPTIONS:</b></h4><pre>
       <b>-h</b>, <b>–help</b>
              Produce help message

       <b>–output</b> <u>arg</u> (=fasta)
              Which output format: fasta or phylip?

       <b>-c</b> <u>arg</u>, <b>–columns</b> <u>arg</u>

              Ranges of columns to keep, like: 1-10,30-
       <b>-t</b> <u>arg</u>, <b>–taxa</b> <u>arg</u>
              Taxa to keep, comma-separated

       <b>-p</b>, <b>–pad</b>
              Add gaps to make sequence lengths identical

       <b>-r</b>, <b>–reverse</b>
              Reverse the sequences

       <b>-e</b>, <b>–erase-empty-columns</b>
              Remove columns with no characters (all gaps).

       <b>–missing</b> <u>arg</u> (=-?)
              What letters are not characters (e.g. gaps)?

       <b>–strip-gaps</b>
              Remove all non-character letters from sequences.

       <b>–reorder-by-tree</b> <u>arg</u>
              Reorder the sequences given a tree

       <b>–use-root</b>
              use the root specified in the tree file to reorder

       <b>–reorder-by-alignment</b> <u>arg</u>
              Reorder the sequences following an alignment

       <b>–align-by-amino</b> <u>arg</u>
              Arrange nucleotides into codon alignment

</pre><h4><b>EXAMPLES:</b></h4><pre>
       To select columns from an alignment:

              % alignment-cat -c1-10,50-100,600- filename.fasta &gt; result.fasta
              % alignment-cat -c5-250/3 filename.fasta &gt; first_codon_position.fasta
              % alignment-cat -c6-250/3 filename.fasta &gt; second_codon_position.fasta

       To concatenate two or more alignments:

              % alignment-cat filename1.fasta filename2.fasta &gt; all.fasta

</pre><h4><b>REPORTING</b> <b>BUGS:</b></h4><pre>
       BAli-Phy online help: &lt;<a href="http://www.bali-phy.org/docs.php">http://www.bali-phy.org/docs.php</a>&gt;.

       Please send bug reports to &lt;<a href="mailto:bali-phy-users@googlegroups.com">bali-phy-users@googlegroups.com</a>&gt;.

</pre><h4><b>AUTHORS</b></h4><pre>
       Benjamin Redelings.

                                                    Feb 2018                                    <u><a href="../man1/alignment-cat.1.html">alignment-cat</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>