<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>alimask - calculate and add column mask to a multiple sequence alignment</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/hmmer">hmmer_3.4+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       alimask - calculate and add column mask to a multiple sequence alignment

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>alimask</b> [<u>options</u>] <u>msafile</u> <u>postmsafile</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>alimask</b>  is  used  to  apply a mask line to a multiple sequence alignment, based on provided alignment or
       model coordinates.  When <b>hmmbuild</b> receives a masked alignment as input, it produces a  profile  model  in
       which  the  emission  probabilities at masked positions are set to match the background frequency, rather
       than being set based on observed frequencies in the alignment.  Position-specific insertion and  deletion
       rates  are  not  altered,  even in masked regions.  <b>alimask</b> autodetects input format, and produces masked
       alignments in Stockholm format.  <u>msafile</u> may contain only one sequence alignment.

       A common motivation for masking a region in an alignment is that the  region  contains  a  simple  tandem
       repeat that is observed to cause an unacceptably high rate of false positive hits.

       In  the  simplest  case,  a  mask  range  is  given in coordinates relative to the input alignment, using
       <b>--alirange</b> <u>&lt;s&gt;</u><b>.</b>  However it is more often the case that the region to be masked has  been  identified  in
       coordinates relative to the profile model (e.g. based on recognizing a simple repeat pattern in false hit
       alignments  or in the HMM logo).  Not all alignment columns are converted to match state positions in the
       profile (see the <b>--symfrac</b> flag for <b>hmmbuild</b> for discussion), so model positions do not necessarily match
       up to alignment column positions.  To remove the  burden  of  converting  model  positions  to  alignment
       positions,  <b>alimask</b>  accepts  the  mask range input in model coordinates as well, using <b>--modelrange</b> <u>&lt;s&gt;</u><b>.</b>
       When using this flag, <u>alimask</u> determines which alignment positions would be  identified  by  <b>hmmbuild</b>  as
       match  states,  a  process  that  requires that all <b>hmmbuild</b> flags impacting that decision be supplied to
       <b>alimask</b>.  It is for this reason that many of the <b>hmmbuild</b> flags are also used by <b>alimask</b>.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>     Help; print a brief reminder of command line usage and all available options.

       <b>-o</b> <u>&lt;f&gt;</u> Direct the summary output to file <u>&lt;f&gt;</u>, rather than to stdout.

</pre><h4><b>OPTIONS</b> <b>FOR</b> <b>SPECIFYING</b> <b>MASK</b> <b>RANGE</b></h4><pre>
       A single mask range is given as a dash-separated pair, like <b>--modelrange</b> <b>10-20</b> and multiple ranges may be
       submitted as a comma-separated list, <b>--modelrange</b> <b>10-20,30-42</b>.

       <b>--modelrange</b> <u>&lt;s&gt;</u>
              Supply the given range(s) in model coordinates.

       <b>--alirange</b> <u>&lt;s&gt;</u>
              Supply the given range(s) in alignment coordinates.

       <b>--appendmask</b>
              Add to the existing mask found with the alignment.  The default is to overwrite any existing mask.

       <b>--model2ali</b> <u>&lt;s&gt;</u>
              Print model range(s) and the corresponding alignment range(s).  No masked alignment  is  produced.
              The output is a single line for each input range, of the form
                 i..j -&gt; m..n
              with i &amp; j representing model range values, and m &amp; n representing alignment range values.

       <b>--ali2model</b> <u>&lt;s&gt;</u>
              Print  alignment  range(s) and the corresponding model range(s).  No masked alignment is produced.
              Because some alignment positions may not map to model positions, the range(s) produced will  begin
              with  the  first  alignment  position  between  &lt;from&gt;  and  &lt;to&gt; (inclusive) that maps to a model
              position, and end with the final alignment position in that range that maps to a  model  position.
              The output is a single line for each input range, of the form
                i..j -&gt; m..n
              with  i &amp; j representing alignment range values, and m &amp; n representing model range values.  If no
              alignment positions in the range &lt;from&gt;..&lt;to&gt; map to a model position, the output prints the input
              &lt;from&gt; and &lt;to&gt; mapping to nothing, with the format:
               i..j  -&gt;   -..-  (no map) .

</pre><h4><b>OPTIONS</b> <b>FOR</b> <b>SPECIFYING</b> <b>THE</b> <b>ALPHABET</b></h4><pre>
       <b>--amino</b>
              Assert that sequences in <u>msafile</u> are protein, bypassing alphabet autodetection.

       <b>--dna</b>  Assert that sequences in <u>msafile</u> are DNA, bypassing alphabet autodetection.

       <b>--rna</b>  Assert that sequences in <u>msafile</u> are RNA, bypassing alphabet autodetection.

</pre><h4><b>OPTIONS</b> <b>CONTROLLING</b> <b>PROFILE</b> <b>CONSTRUCTION</b></h4><pre>
       These options control how consensus columns are defined in an alignment.

       <b>--fast</b> Define consensus columns as those that have a fraction &gt;= <b>symfrac</b> of residues as opposed to  gaps.
              (See below for the <b>--symfrac</b> option.) This is the default.

       <b>--hand</b> Define  consensus  columns  in  next profile using reference annotation to the multiple alignment.
              This allows you to define any consensus columns you like.

       <b>--symfrac</b> <u>&lt;x&gt;</u>
              Define the residue fraction threshold necessary to define a consensus column when using the <b>--fast</b>
              option. The default is 0.5. The symbol fraction in each column is calculated after taking relative
              sequence weighting into account, and ignoring gap characters corresponding  to  ends  of  sequence
              fragments  (as  opposed  to  internal insertions/deletions).  Setting this to 0.0 means that every
              alignment column will be assigned as consensus, which may be useful in some cases. Setting  it  to
              1.0  means  that only columns that include 0 gaps (internal insertions/deletions) will be assigned
              as consensus.

       <b>--fragthresh</b> <u>&lt;x&gt;</u>
              We only want to count terminal gaps as deletions if the aligned sequence  is  known  to  be  full-
              length,  not if it is a fragment (for instance, because only part of it was sequenced). HMMER uses
              a simple rule to infer fragments: if the sequence length L is less than or equal to a fraction <u>&lt;x&gt;</u>
              times the alignment length in columns, then the sequence is handled as a fragment. The default  is
              0.5.   Setting  <b>--fragthresh</b> <b>0</b> will define no (nonempty) sequence as a fragment; you might want to
              do this if you know you've got a carefully curated alignment of  full-length  sequences.   Setting
              <b>--fragthresh</b>  <b>1</b> will define all sequences as fragments; you might want to do this if you know your
              alignment is entirely composed of fragments, such as translated short reads in metagenomic shotgun
              data.

</pre><h4><b>OPTIONS</b> <b>CONTROLLING</b> <b>RELATIVE</b> <b>WEIGHTS</b></h4><pre>
       HMMER uses an ad hoc sequence weighting algorithm to downweight closely related  sequences  and  upweight
       distantly  related  ones.  This  has  the  effect  of  making  models  less biased by uneven phylogenetic
       representation. For example, two identical sequences would typically each receive half  the  weight  that
       one sequence would.  These options control which algorithm gets used.

       <b>--wpb</b>  Use  the  Henikoff  position-based sequence weighting scheme [Henikoff and Henikoff, J. Mol. Biol.
              243:574, 1994].  This is the default.

       <b>--wgsc</b> Use the Gerstein/Sonnhammer/Chothia weighting algorithm [Gerstein et al, J. Mol.  Biol.  235:1067,
              1994].

       <b>--wblosum</b>
              Use  the  same  clustering  scheme that was used to weight data in calculating BLOSUM substitution
              matrices [Henikoff and Henikoff, Proc. Natl. Acad. Sci  89:10915,  1992].  Sequences  are  single-
              linkage  clustered at an identity threshold (default 0.62; see <b>--wid</b>) and within each cluster of c
              sequences, each sequence gets relative weight 1/c.

       <b>--wnone</b>
              No relative weights. All sequences are assigned uniform weight.

       <b>--wid</b> <u>&lt;x&gt;</u>
              Sets the identity threshold used by single-linkage clustering when using <b>--wblosum</b>.  Invalid  with
              any other weighting scheme. Default is 0.62.

</pre><h4><b>OTHER</b> <b>OPTIONS</b></h4><pre>
       <b>--informat</b> <u>&lt;s&gt;</u>
              Assert  that  input  <u>msafile</u>  is  in alignment format <u>&lt;s&gt;</u>, bypassing format autodetection.  Common
              choices for <u>&lt;s&gt;</u> include: <b>stockholm</b>, <b>a2m</b>, <b>afa</b>, <b>psiblast</b>, <b>clustal</b>, <b>phylip</b>.   For  more  information,
              and  for  codes  for  some  less  common formats, see main documentation.  The string <u>&lt;s&gt;</u> is case-
              insensitive (<b>a2m</b> or <b>A2M</b> both work).

       <b>--outformat</b> <u>&lt;s&gt;</u>
              Write the output <u>postmsafile</u> in alignment format <u>&lt;s&gt;</u>.  Common choices for <u>&lt;s&gt;</u> include:  <b>stockholm</b>,
              <b>a2m</b>,  <b>afa</b>,  <b>psiblast</b>, <b>clustal</b>, <b>phylip</b>.  The string <u>&lt;s&gt;</u> is case-insensitive (<b>a2m</b> or <b>A2M</b> both work).
              Default is <b>stockholm</b>.

       <b>--seed</b> <u>&lt;n&gt;</u>
              Seed the random number generator with <u>&lt;n&gt;</u>, an integer &gt;= 0.  If <u>&lt;n&gt;</u>  is  nonzero,  any  stochastic
              simulations  will  be reproducible; the same command will give the same results.  If <u>&lt;n&gt;</u> is 0, the
              random number generator is seeded arbitrarily, and stochastic simulations will vary  from  run  to
              run of the same command.  The default seed is 42.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       See  <b><a href="../man1/hmmer.1.html">hmmer</a></b>(1) for a master man page with a list of all the individual man pages for programs in the HMMER
       package.

       For complete documentation, see the user guide that came with your HMMER distribution (Userguide.pdf); or
       see the HMMER web page (<a href="http://hmmer.org/">http://hmmer.org/</a>).

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright (C) 2023 Howard Hughes Medical Institute.
       Freely distributed under the BSD open source license.

       For additional information on copyright and licensing, see the file called COPYRIGHT in your HMMER source
       distribution, or see the HMMER web page (<a href="http://hmmer.org/">http://hmmer.org/</a>).

</pre><h4><b>AUTHOR</b></h4><pre>
       <a href="http://eddylab.org">http://eddylab.org</a>

HMMER 3.4                                           Aug 2023                                          <u><a href="../man1/alimask.1.html">alimask</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>