<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>alter-sequence-alignment - genomic sequences ALignment Transformation EnviRonment</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/alter-sequence-alignment">alter-sequence-alignment_1.3.4-8_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       alter-sequence-alignment - genomic sequences ALignment Transformation EnviRonment

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>alter-sequence-alignment</b> <u>[option]</u> <u>[sequence]</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
        ALTER (ALignment Transformation EnviRonment) is a tool to transform
        between multiple sequence alignment formats. ALTER focuses on the
        specifications of mainstream alignment and analysis programs rather than
        on the conversion among more or less specific formats.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-c</b> (<b>--collapse</b>)
              Collapse sequences to haplotypes.

       <b>-cg</b> (<b>--collapseGaps</b>)
              Treat gaps as missing data when collapsing.

       <b>-cl</b> (<b>--collapseLimit</b>) N
              Connection limit (sequences differing at &lt;= l sites will be collapsed) (default is l=0).

       <b>-cm</b> (<b>--collapseMissing</b>)
              Count missing data as differences when collapsing.

       <b>-i</b> (<b>--input</b>) FILE
              Input file.

       <b>-ia</b> (<b>--inputAutodetect</b>)
              Autodetect format (other input options are omitted).

       <b>-if</b> (<b>--inputFormat</b>) VAL
              Input format (ALN, FASTA, GDE, MEGA, MSF, NEXUS, PHYLIP or PIR).

       <b>-io</b> (<b>--inputOS</b>) VAL
              Input operating system (Linux, MacOS or Windows)

       <b>-ip</b> (<b>--inputProgram</b>) VAL
              Input program (Clustal, MAFFT, MUSCLE, PROBCONS or TCoffee).

       <b>-o</b> (<b>--output</b>) FILE
              Output file.

       <b>-of</b> (<b>--outputFormat</b>) VAL
              Output format (ALN, FASTA, GDE, MEGA, MSF, NEXUS, PHYLIP or PIR).

       <b>-ol</b> (<b>--outputLowerCase</b>)
              Lowe case output.

       <b>-om</b> (<b>--outputMatch</b>)
              Output match characters.

       <b>-on</b> (<b>--outputResidueNumbers</b>)
              Output residue numbers (only ALN format).

       <b>-oo</b> (<b>--outputOS</b>) VAL
              Output operating system (Linux, MacOS or Windows).

       <b>-op</b> (<b>--outputProgram</b>) VAL
              Output  program  (jModelTest,  MrBayes,  PAML, PAUP, PhyML, ProtTest, RAxML, TCS, CodABC, BioEdit,
              MEGA, dnaSP, Se-Al, Mesquite, SplitsTree, Clustal , MAFFT,  MUSCLE,  PROBCONS,  TCoffee,  Gblocks,
              SeaView, trimAl or GENERAL)

       <b>-os</b> (<b>--outputSequential</b>)
              Sequential output (only NEXUS and PHYLIP formats).

alter-sequence-alignment 1.3.3                   September 2015                      <u><a href="../man1/ALTER-SEQUENCE-ALIGNMENT.1.html">ALTER-SEQUENCE-ALIGNMENT</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>