<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>amap - Protein multiple alignment by sequence annealing</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/amap-align">amap-align_2.2+git20080214.600fc29+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       amap - Protein multiple alignment by sequence annealing

</pre><h4><b>SYNOPSIS</b></h4><pre>

       <b>amap</b> [<u>OPTION</u>] [<u>MFAFILE</u>] [<u>MFAFILE</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       AMAP is a tool to perform multiple alignment of peptidic sequences. It utilizes posterior decoding, and a
       sequence-annealing alignment, instead of the traditional progressive alignment method. It is the only
       alignment program that allows one to control the sensitivity / specificity tradeoff. It is based on the
       ProbCons source code, but uses alignment metric accuracy and eliminates the consistency transformation.

       In its default configuration, AMAP is tuned to maximize the expected Alignment Metric Accuracy (AMA)
       score - a new alignment accuracy measure, based on a metric for the multiple-alignment space, which
       integrates sensitivity and specificity into a single balanced measure. AMA is defined as the fraction of
       correctly aligned residues (either to another residue or to a gap) out of the total number of residues in
       all the sequences.

       <b>amap</b> aligns sequences provided in MFA format. This format consists of multiple sequences. Each sequence
       in MFA format begins with a single-line description, followed by lines of sequence data. The description
       line is distinguished from the sequence data by a greater-than (“&gt;”) symbol in the first column.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-clustalw</b>
           use CLUSTALW output format instead of MFA

       <b>-c</b> <b>--consistency</b> <u>REPS</u>
           use 0 &lt;= <u>REPS</u> &lt;= 5 (default: 0) passes of consistency transformation

       <b>-ir</b> <b>--iterative-refinement</b> <u>REPS</u>
           use 0 &lt;= <u>REPS</u> &lt;=1000 (default: 0) passes of iterative-refinement

       <b>-pre</b> <b>--pre-training</b> <u>REPS</u>
           use 0 &lt;= <u>REPS</u> &lt;= 20 (default: 0) rounds of pretraining

       <b>-pairs</b>
           generate all-pairs pairwise alignments

       <b>-viterbi</b>
           use Viterbi algorithm to generate all pairs (automatically enables <b>-pairs</b>)

       <b>-v</b> <b>--verbose</b>
           Report progress while aligning (default: off)

       <b>-annot</b> <u>FILENAME</u>
           write annotation for multiple alignment to <u>FILENAME</u>

       <b>-t</b> <b>--train</b> <u>FILENAME</u>
           compute EM transition probabilities, store in <u>FILENAME</u> (default: no training)

       <b>-e</b> <b>--emissions</b>
           also reestimate emission probabilities (default: off)

       <b>-p</b> <b>--paramfile</b> <u>FILENAME</u>
           read parameters from <u>FILENAME</u> (default: )

       <b>-a</b> <b>--alignment-order</b>
           print sequences in alignment order rather than input order (default: off)

       <b>-g</b> <b>--gap-factor</b> <u>GF</u>
           use <u>GF</u> as the gap-factor parameter, set to 0 for best sensitivity, higher values for better
           specificity (default: 0.5)

       <b>-w</b> <b>--edge-weight-threshold</b> <u>W</u>
           stop the sequence annealing process when best edge has lower weight than <u>W</u>, set to 0 for best
           sensitivity, higher values for better specificity (default: 0)

       <b>-prog</b> <b>--progressive</b>
           use progressive alignment instead of sequence annealing alignment (default: off)

       <b>-noreorder</b> <b>--no-edge-reordering</b>
           disable reordering of edges during sequence annealing alignment (default: off)

       <b>-maxstep</b> <b>--use-max-stepsize</b>
           use maximum improvement step size instead of tGf edge ranking (default: off)

       <b>-print</b> <b>--print-posteriors</b>
           only print the posterior probability matrices (default: off)

       <b>-gui</b> <u>START</u> <u>STEP</u>
           print output for the AMAP Display Java based GUI (default: ) starting at weight <u>START</u> (default:
           infinity) with step size <u>STEP</u> (default: )

</pre><h4><b>EXAMPLES</b></h4><pre>
       To run AMAP with the default options change to the <u>align</u> directory and type:

       <b>%</b> <b>amap</b> <b>&lt;multi-fasta-file-name&gt;</b>

       If no file name is provided the list of options are printed.

       In order to use the AMAP Display run AMAP with the -gui option, and save the output to a file, then use
       the file as the input to AmapDisplay. For example, type:

       <b>%</b> <b>align/amap</b> <b>-gui</b> <b>examples/BB12020.tfa</b> <b>&gt;</b> <b>examples/BB12020.tfa.out</b>

       <b>%</b> <b>java</b> <b>-jar</b> <b>display/AmapDisplay.jar</b> <b>examples/BB12020.tfa.out</b>

       (on Debian systems, the <u>examples</u> directory is in <u>/usr/share/doc/amap-align/examples</u>

</pre><h4><b>NOTE</b></h4><pre>
       In older versions ( &lt; 2.0-1) of the package for Debian(TM) systems, the <b>amap</b> command was renamed
       <b>amap-align</b> because there was already another tool called <b>amap</b> (which performs some computer network
       diagnostics). A symbolic link <b>amap-align</b> is still provided for upgrade purposes but will be removed in
       Debian releases posterior to Etch (Debian 4.0).

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       The current version of AMAP uses the PROBCONS 1.09 code base for some of the input/output procedures, and
       for the calculation of posterior probabilities (see PROBCONS.README in <u>/usr/share/doc/amap-align/</u>).
       Future releases might implement the algorithm using a new independent code base.

       On Debian(TM) systems, <b><a href="../man1/probcons.1.html">probcons</a></b>(1) is available in the probcons package.

</pre><h4><b>REFERENCES</b></h4><pre>
       For more details on AMAP and AMA, see Schwartz, Ariel S., Myers, Eugene W., and Pachter, Lior. Alignment
       Metric Accuracy (Submitted for publication). For more details on sequence-annealing, see Schwartz, Ariel
       S. and Pachter, Lior. Multiple Alignment by Sequence Annealing (Submitted for publication).

       PROBCONS was published in Do, C.B., Mahabhashyam, M.S.P., Brudno, M., and Batzoglou, S. 2005. PROBCONS:
       Probabilistic Consistency-based Multiple Sequence Alignment. Genome Research 15: 330-340.

</pre><h4><b>AUTHORS</b></h4><pre>
       <b>Ariel</b> <b>Schwartz</b> &lt;<a href="mailto:sariel@cs.berkeley.edu">sariel@cs.berkeley.edu</a>&gt;
           Upstream author of AMAP

       <b>Chuong</b> <b>Do</b>
           Wrote Probcons, on which AMAP is based.

       <b>Charles</b> <b>Plessy&gt;</b> &lt;<a href="mailto:charles-debian-nospam@plessy.org">charles-debian-nospam@plessy.org</a>&gt;
           Wrote this manpage in DocBook XML for the Debian distribution.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       AMAP,  PROBCONS, and this manual page have been made freely available as PUBLIC DOMAIN software and hence
       are not subject to copyright in the United States. This system and/or any portion of the source code  may
       be  used,  modified,  or  redistributed  without  restrictions.  AMAP,  PROBCONS and this manual page are
       distributed WITHOUT WARRANTY, express or implied. The authors accept NO LEGAL LIABILITY OR RESPONSIBILITY
       for loss due to reliance on the program.

amap 2.2                                           03/31/2008                                            <u><a href="../man1/AMAP.1.html">AMAP</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>