<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>bam2hints - convert mRNA-to-genome alignments in BAM format into a hint file for AUGUSTUS</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/augustus">augustus_3.5.0+dfsg-4build7_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       bam2hints - convert mRNA-to-genome alignments in BAM format into a hint file for AUGUSTUS

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>bam2hints</b> [parameters] [--in=example.bam] [--out=hints.gff]

</pre><h4><b>OPTIONS</b></h4><pre>
       The input BAM file must be sorted by target (=genome) sequence names and within the sequences by begin
       coordinates.

   <b>Parameters:</b>
       <b>--in=example.bam</b>
           Input BAM file. This option can be omitted only on UNIX platforms. Then bam2hints reads from the
           standard input stream.

       <b>--out=hints.gff</b>
           Output hints gff file. If this option is omitted, then bam2hints will write to the standard output
           stream.

       <b>--priority=n</b>/<b>-p</b>
           priority of hint group (set to 4)

       <b>--maxgaplen=n</b>/<b>-g</b>
           gaps at most this length are simply closed (set to 14)

       <b>--minintronlen=n</b>/<b>-m</b>
           alignments with gaps shorter than this and longer than maxgaplen are discarded (set to 32)

       <b>--maxintronlen=n</b>/<b>-M</b>
           alignments with longer gaps are discarded (set to 350000)

       <b>--MinEndBlockLen=n</b>/<b>-b</b>
           minimum length of a 'dangling' exon (set to 8)

       <b>--maxQgaplen=n</b>/<b>-q</b>
           maximum length of gap in query (cDNA) sequence (set to 5)

       <b>--exonhints</b>/<b>-x</b>
           compute exonpart, exon and splice site hints in addition to intron hints (set to 0=Off). You should
           generate exonpart hints from RNA-Seq using wiggle (.wig) input to wig2hints.

       <b>--ep_cutoff=n</b>/<b>-e</b>
           this many bp are cut off of each exonpart hint at end of alignment (set to 10)

       <b>--source=s</b>/<b>-s</b>
           source identifier (set to 'E')

       <b>--intronsonly</b>/<b>-I</b>
           only retrieve intron hints (e.g. because the exon(part) hints are retrieved by converting to a wig
           track, set to 1=On). Deprecated as this is the default now.

       <b>--nomult</b>/<b>-n</b>
           do not summarize multiple identical intron hints to a single one (set to 0=Off)

       <b>--remove_redundant</b>/<b>-r</b>
           only keep the strongest hint for a region (set to 0=Off)

       <b>--maxcoverage=n</b>/<b>-C</b>
           maximal number of hints at a given position (0: filtering deactivated). A high value causes long
           running time of AUGUSTUS in regions with thousands of cDNA alignments. (set to 0)

       <b>--ssOn</b>/<b>-S</b>
           include splice site (dss, ass) hints in output (set to 0=Off)

       <b>--trunkSS</b>/<b>-T</b>
           include splice sites hints from the ends of a truncated alignment (contig too short, set to 0=Off)

       <b>--score=f</b>/<b>-s</b>
           fill this number in in the score column (set to 0)

       <b>--maxgenelen=n</b>/<b>-G</b>
           alignments of the same clone are considered to be of the same gene if not separated by more than this
           (set to 400000). Alignments that span more than this are ignored, but better filter long introns
           through an alignment program.

</pre><h4><b>AUTHORS</b></h4><pre>
       AUGUSTUS was written by M. Stanke, O. Keller, S. KÃ¶nig, L. Gerischer and L. Romoth.

</pre><h4><b>ADDITIONAL</b> <b>DOCUMENTATION</b></h4><pre>
       An exhaustive documentation can be found in the file /usr/share/doc/augustus/README.md.gz.

                                                                                                    <u><a href="../man1/BAM2HINTS.1.html">BAM2HINTS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>