<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>bammarkduplicates2 - mark duplicate reads/alignments in BAM files</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/biobambam2">biobambam2_2.0.185+ds-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       bammarkduplicates2 - mark duplicate reads/alignments in BAM files

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>bammarkduplicates2</b> [options]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       bammarkduplicates2 reads a coordinate sorted BAM file containing alignments computed by an aligner, marks
       duplicate  reads/alignments using the coordinates of the alignments and writes the marked alignments to a
       BAM file. By default input is via standard input and output via standard output. Duplication metrics  are
       provided  on the standard error channel by default. bammarkduplicates2 scans the input BAM file twice, it
       is thus of benefit to set the I key (described below) to give the name of an input file. If the input BAM
       file is given via standard input, then a copy of the input will be stored in a temporary file during  the
       first  run for processing in a second run. bammarkduplicates2 assumes that each read end of a pair occurs
       in exactly one entry in the file.

       The following key=value pairs can be given:

       <b>I=&lt;stdin&gt;:</b> name of the input file (input is read from standard input if not set). This key can  be  given
       multiple  times.  If it is given multiple times, then all the input files need to be in coordinate sorted
       order and will be merged for output such that the output will be coordinate sorted.

       <b>O=&lt;stdout&gt;:</b> name of the output file (output is written to standard output if not set)

       <b>M=&lt;stderr&gt;:</b> name of the metrics file (metrics are written to standard error if not set)

       <b>tmpfile=&lt;filename&gt;:</b> prefix for temporary files. By default the temporary files are created in the current
       directory

       <b>level=&lt;-1|0|1|9|11&gt;:</b> set compression level of the output BAM file. Valid values are

       -1:    zlib/gzip default compression level

       0:     uncompressed

       1:     zlib/gzip level 1 (fast) compression

       9:     zlib/gzip level 9 (best) compression

       If libmaus has been compiled with support for igzip (see https://software.intel.com/en-us/articles/igzip-
       a-high-performance-deflate-compressor-with-optimizations-for-genomic-data) then an additional valid value
       is

       11:    igzip compression

       <b>markthreads=&lt;1&gt;:</b> Number of threads used during marking duplicate alignments.

       <b>verbose=&lt;1&gt;:</b> Valid values are

       1:     print progress report on standard error

       0:     do not print progress report

       <b>mod=&lt;1048576&gt;:</b> print progress after processing every mod'th input read/alignment (default is 1M)

       <b>rewritebam=&lt;0|1|2&gt;:</b> compression type of the temporary file stored if I key is not  given  (input  is  via
       standard input). Valid values are

       0:     temporary data is compressed using snappy (if available, see <a href="http://code.google.com/p/snappy/">http://code.google.com/p/snappy/</a>)

       1:     gzip/BAM, recompress input and store it as a BAM file

       2:     copy, produce a 1/1 copy of the input stream as a temporary file

       <b>rewritebamlevel=&lt;-1|0|1|9|11&gt;:</b>  compression level of temporary BAM file if rewritebam is 1 (see level key
       for possible values).

       <b>colhashbits=&lt;20&gt;:</b> base two logarithm of the size of the hash table used for collation (the default  value
       is  18  and  should  work  reasonably  well  for  most  input  files.   Please see the biobambam paper at
       arxiv.org/abs/1306.0836 for details).

       <b>collistsize=&lt;33554432&gt;:</b> size of hash table overflow list in bytes (the default is 128MB and  should  work
       reasonably  well  for  most  input  files.  Please see the biobambam paper at arxiv.org/abs/1306.0836 for
       details).

       <b>fragbufsize=&lt;50331648&gt;:</b> size of each fragment/pair file buffer in bytes (bammarkduplicates2 uses two such
       buffer for detecting duplicates)

       <b>rmdup=&lt;0|1&gt;:</b> sets how duplicates are handled

       0:     duplicates will be retained in the output file and have the duplication flag set

       1:     duplicates will be removed when writing the output file

       <b>maxreadlength=&lt;[500]&gt;:</b> maximum read length in input. This value can be set higher than the actual maximum
       in the file but not lower.

       <b>md5=&lt;0|1&gt;:</b> md5 checksum creation for output file. Valid values are

       0:     do not compute checksum. This is the default.

       1:     compute checksum. If the md5filename key is set, then the checksum is written to the  given  file.
              If  the  md5filename  key is not given but set O key is set the checksum will be written as output
              file name with .md5 appended.  If md5filename and O are unset, then no checksum will be computed.

       <b>md5filename</b> file name for md5 checksum if md5=1.

       <b>index=&lt;0|1&gt;:</b> compute BAM index for output file. Valid values are

       0:     do not compute BAM index. This is the default.

       1:     compute BAM index. If the indexfilename key is set, then the BAM index is  written  to  the  given
              file.  If  the indexfilename key is not given but set O key is set the checksum will be written as
              output file name with .bai appended.  If indexfilename and O are unset, then no BAM index will  be
              computed.

       <b>indexfilename</b> file name for BAM index if index=1.

       <b>tag=&lt;tag&gt;</b>  name  of  auxiliary field storing tag information in string form. Read fragments or pairs with
       different tags will not be considered as duplicates, even  they  would  be  according  to  their  mapping
       coordinates.  For pairs the tag field information of the first and second mate are concatenated to obtain
       the tag of the pair.

       <b>nucltag=&lt;tag&gt;</b> this option works like the tag option but is restricted to sequences of nucleotides  (A,C,G
       or  T)  as tags. The length of each tag sequence is not allowed to exceed 15 bases. All tags are required
       to have the same length.  Each non nucleotide symbol is mapped to A.  In  contrast  to  the  tag  option,
       nucltag uses less memory for processing and can be expected to be faster.

       <b>D</b>  output  file  name  for  removed  duplicates if rmdup=1. By default the reads and read pairs marked as
       duplicates are discarded when rmdup=1. If the D key is set, then the  sequence  of  reads  which  is  not
       written  to the output file is written to a separate file. The name of this separate file is the value of
       the D key.

       <b>dupmd5=&lt;0|1&gt;:</b> compute md5 checksum of file given by D key if rmdup=1.  By  default  no  md5  checksum  is
       computed for this file.

       <b>dupmd5filename:</b> file name for storing the md5 checksum of the D file if rmdup=1, D is set and dupmd5=1.

       <b>dupindex=&lt;0|1&gt;:</b>  compute BAM index for file given by D key if rmdup=1. By default no index is compute for
       this file.

       <b>dupindexfilename:</b> file name for storing the BAM index of the D file if rmdup=1, D is set and dupmd5=1.

       <b>optminpixeldif=&lt;100&gt;:</b> distance (x and y inside same tile) inside which reads are  considered  as  optical
       duplicates

</pre><h4><b>AUTHOR</b></h4><pre>
       Written by German Tischler.

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       Report bugs to &lt;<a href="mailto:germant@miltenyibiotec.de">germant@miltenyibiotec.de</a>&gt;

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright  ©  2009-2013  German  Tischler,  © 2011-2013 Genome Research Limited.  License GPLv3+: GNU GPL
       version 3 &lt;<a href="http://gnu.org/licenses/gpl.html">http://gnu.org/licenses/gpl.html</a>&gt;
       This is free software: you are free to change and redistribute it.  There is NO WARRANTY, to  the  extent
       permitted by law.

BIOBAMBAM                                         October 2013                             <u><a href="../man1/BAMMARKDUPLICATES2.1.html">BAMMARKDUPLICATES2</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>