<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>belvu - View multiple alignments in good-looking colours</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/belvu">belvu_4.44.1+dfsg-7.1build3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       belvu - View multiple alignments in good-looking colours

</pre><h4><b>DESCRIPTION</b></h4><pre>
              Belvu - View multiple alignments in good-looking colours.

              Usage: belvu [options] &lt;multiple_alignment&gt;|-

              &lt;multiple_alignment&gt;|- = alignment file or pipe.

              Options:

       <b>-c</b>     Print Conservation table.

       <b>-l</b> &lt;file&gt;
              Load residue color code file.

       <b>-L</b> &lt;file&gt;
              Load markup and organism color code file.

       <b>-m</b> &lt;file&gt;
              Read file with matching sequence segments.

       <b>-r</b>     Read alignment in 'raw' format (Name sequence).

       <b>-R</b>     Do not parse coordinates when reading alignment.

       <b>-o</b> &lt;format&gt; Write alignment or tree to stdout in this format and exit.

              Valid formats: MSF, Mul(Stockholm), Selex, FastaAlign, Fasta, tree.

       <b>-X</b> &lt;cutoff&gt; Print UPGMA-based subfamilies at cutoff &lt;cutoff&gt;.

       <b>-n</b> &lt;cutoff&gt; Make non-redundant to &lt;cutoff&gt; %identity at startup.

       <b>-Q</b> &lt;cutoff&gt; Remove columns more gappy than &lt;cutoff&gt;.

       <b>-q</b> &lt;cutoff&gt; Remove sequences more gappy than &lt;cutoff&gt;.

       <b>-G</b>     Penalize gaps in pairwise comparisons.

       <b>-i</b>     Ignore gaps in conservation calculation.

       <b>-P</b>     Remove partial sequences at startup.

       <b>-C</b>     Don't write coordinates to saved file.

       <b>-z</b> &lt;char&gt;
              Separator char between name and coordinates in saved file.

       <b>-a</b>     Show alignment annotations on screen (Stockholm format only).

       <b>-p</b>     Output random model probabilites for HMMER.  (Based on all residues.)

       <b>-S</b> &lt;order&gt;
              Sort sequences in this order.

              a  -&gt;  alphabetically  o  -&gt;  by  Swissprot  organism,  alphabetically  s  -&gt;  by  score  n  -&gt; by
              Neighbor-joining tree u -&gt; by UPGMA tree S -&gt; by similarity to first sequence i -&gt; by identity  to
              first sequence

       <b>-s</b> &lt;file&gt;
              Read in file of scores.

       <b>-T</b> &lt;method&gt; Tree options:

              i  -&gt;  Start  up  showing  tree  I  -&gt; Start up showing only tree d -&gt; Show distances in tree n -&gt;
              Neighbor-joining u -&gt; UPGMA c -&gt; Don't  color  tree  by  organism  o  -&gt;  Don't  display  sequence
              coordinates  in  tree  b  -&gt;  Use  Scoredist  distance  correction (default) j -&gt; Use Jukes-Cantor
              distance correction k -&gt; Use Kimura distance correction s  -&gt;  Use  Storm  &amp;  Sonnhammer  distance
              correction  r  -&gt; Use uncorrected distances p -&gt; Print distance matrix and exit R -&gt; Read distance
              matrix instead of alignment

              (only in combination with Tree routines)

       <b>-b</b> &lt;n&gt;
              Apply boostrap analysis with &lt;n&gt; bootstrap samples

       <b>-B</b>     Print out bootstrap trees and exit (Negative value -&gt; display bootstrap trees on screen)

       <b>-O</b> &lt;label&gt;
              Read organism info after this label (default OS)

       <b>-t</b> &lt;title&gt;
              Set window title.

       <b>-u</b>     Start up with uncoloured alignment (faster).

       <b>-h</b>, <b>--help</b>
              Show more detailed usage information

       <b>--compiled</b>
              Show package compile date

       <b>--version</b>
              Show package version number

       <b>--abbrev-title-on</b>
              Abbreviate window title prefixes

       <b>--abbrev-title-off</b>
              Do not abbreviate window title prefixes

              setenv BELVU_FETCH to desired sequence fetching program.  setenv BELVU_FONT_SIZE to specify window
              font size.  setenv BELVU_STATUSBAR_SIZE to specify statusbar font size (0 =&gt; hide statusbar).

       <b>-----</b>

              Written  by  Gemma  Barson  &lt;<a href="mailto:gb10@sanger.ac.uk">gb10@sanger.ac.uk</a>&gt;  Based  on  original  code  by   Erik   Sonnhammer
              &lt;<a href="mailto:Erik.Sonnhammer@sbc.su.se">Erik.Sonnhammer@sbc.su.se</a>&gt;

              Reference: Scoredist: A simple and robust protein sequence distance estimator.

              Erik LL Sonnhammer and Volker Hollich.  BMC Bioinformatics 6:108 (2005)

              See <a href="http://www.sanger.ac.uk/resources/software/seqtools/">http://www.sanger.ac.uk/resources/software/seqtools/</a> for more info.

              Copyright  (c)  2011-2015: Genome Research Ltd.  Belvu is distributed under the GNU General Public
              License; see <a href="http://www.gnu.org/copyleft/gpl.txt">http://www.gnu.org/copyleft/gpl.txt</a>

       Version 4.44.1
              14:27:56 Oct 19 2017

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

belvu 4.44.1                                      October 2017                                          <u><a href="../man1/BELVU.1.html">BELVU</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>