<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>blixem - display multiple alignments against a reference sequence</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/blixem">blixem_4.44.1+dfsg-7.1build3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       blixem - display multiple alignments against a reference sequence

</pre><h4><b>DESCRIPTION</b></h4><pre>
              Blixem - display multiple alignments against a reference sequence.

              Usage: blixem [options] [&lt;sequence_file&gt;] &lt;data_file&gt; [X options]

              &lt;sequence_file&gt;  contains  the  reference  sequence in FASTA format.  &lt;data_file&gt; is a GFF v3 file
              containing alignments and other features.   If  &lt;sequence_file&gt;  is  omitted,  &lt;data_file&gt;  should
              contain the reference sequence in FASTA format, below a comment line that reads ##FASTA.

              Both  &lt;sequence_file&gt; and &lt;data_file&gt; can be substituted by "-" for reading from stdin (pipe).  If
              &lt;sequence_file&gt; is piped, the first line should contain the  sequence  name  and  the  second  the
              sequence itself.

              Options:

       <b>-t</b> &lt;type&gt;, <b>--display-type=</b>&lt;type&gt;
              MANDATORY

              Whether to display sequences in nucleotide or protein mode. Must be one of:

              N = nucleotide P = protein

       <b>-a</b> &lt;names&gt;, <b>--alignment-names=</b>&lt;names&gt;

              Specify a string giving the names of the alignments, e.g. "EST_mouse EST_human" etc.

       <b>-c</b> &lt;file&gt;, <b>--config-file=</b>&lt;file&gt;

              Read configuration options from 'file'.

       <b>--abbrev-title-on</b>

              Abbreviate window title prefixes

       <b>--abbrev-title-off</b>

              Do not abbreviate window title prefixes

       <b>--compiled</b>

              Show package compile date.

       <b>-d</b> &lt;data_file&gt;, <b>--data-file=</b>&lt;data_file&gt;

              Alternative way of specifying &lt;data_file&gt; using an argument

       <b>--dataset</b>

              Optional string to indicate a data-set that the alignments are from.

       <b>-e</b> &lt;sequence_file&gt;, <b>--sequence-file=</b>&lt;sequence_file&gt;

              Alternative way of specifying &lt;sequence_file&gt; using an argument

       <b>--dotter-first-match</b>

              Call Dotter on the first match to the right of the default start coord.

       <b>--fetch-server</b> &lt;nodeid:port&gt;

              Causes  Blixem to get sequences from a fetch server at machine 'nodeid' on the given port (default
              22100).

       <b>-h</b>, <b>--help</b>

              More detailed usage information.

       <b>--hide-big-picture</b>

              Hide the big picture section on start-up.

       <b>--hide-inactive-strand</b>

              Hide the inactive strand (i.e. the reverse strand, or the forward  strand  if  the  <b>-R</b>  option  is
              used).

       <b>--highlight-diffs</b>

              Enable 'highlight differences' mode, where mismatches (rather than matches) are highlighted.

       <b>--invert-sort</b>

              Invert sorting order

       <b>-m</b> &lt;from[:to]&gt;, <b>--map-coords=</b>&lt;from[:to]&gt;

              Map  the  coordinate  system so that the given 'from' coordinate maps to the given 'to' coordinate
              (or to '1' if 'to' is not given).

       <b>-n</b>, <b>--negate-coords</b>

              When showing the reverse strand, negate the display coordinates.

       <b>-o</b> &lt;n&gt;, <b>--offset=</b>&lt;n&gt;

              Offset the reference sequence coordinate system by n.

       <b>--optional-data</b>

              Parse additional data such as organism and tissue-type on start-up.

       <b>--remove-input-files</b>

              Delete the input files after they have been parsed.

       <b>-r</b>, <b>--reverse-strand</b>

              Indicates that the given reference sequence is the reverse strand.

       <b>--save-temp-files</b>

              Save any temporary files created by Blixem.

       <b>--show-coverage</b>

              Display the coverage section on start-up.

       <b>--sort-mode=</b>&lt;mode&gt;

              Default sort mode. Use <b>--help</b> option to see details.

       <b>--squash-matches</b>

              Compress the alignment lists on start-up.

       <b>-s</b> &lt;n&gt;, <b>--start-coord=</b>&lt;n&gt;

              Start with the display centred on coordinate n.

       <b>--start-next-match</b>

              Start with the display centred on the first match to the right of the default start coord.

       <b>-y</b> &lt;file&gt;, <b>--styles-file=</b>&lt;file&gt;

              Read color options from a key-value file. Use <b>--help</b> option to see details.

       <b>--version</b>

              Show package version number.

       <b>-z</b> &lt;start:end&gt;, <b>--zoom-range=</b>&lt;start:end&gt;

              Specify the initial range of coordinates to zoom the big picture in to.

       <b>--zoom-whole</b>

              Start with the big picture zoomed out to view the full reference sequence range.

              Some X options: <b>-acefont</b> &lt;font&gt; Main font.  <b>-font</b>    &lt;font&gt; Menu font.

       <b>-----</b>

              Written  by  Gemma  Barson  &lt;<a href="mailto:gb10@sanger.ac.uk">gb10@sanger.ac.uk</a>&gt;  Based  on  original  code  by   Erik   Sonnhammer
              &lt;<a href="mailto:Erik.Sonnhammer@sbc.su.se">Erik.Sonnhammer@sbc.su.se</a>&gt;

              Reference: Sonnhammer ELL &amp; Durbin R (1994). A workbench for Large Scale

              Sequence Homology Analysis. Comput. Applic. Biosci. 10:301-307.

              See <a href="http://www.sanger.ac.uk/resources/software/seqtools/">http://www.sanger.ac.uk/resources/software/seqtools/</a> for more info.

              Copyright  (c) 2009-2015: Genome Research Ltd.  Blixem is distributed under the GNU General Public
              License; see <a href="http://www.gnu.org/copyleft/gpl.txt">http://www.gnu.org/copyleft/gpl.txt</a>

       Version 4.44.1
              14:27:56 Oct 19 2017

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

blixem 4.44.1                                     October 2017                                         <u><a href="../man1/BLIXEM.1.html">BLIXEM</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>