<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>cmbuild - construct covariance model(s) from structurally annotated RNA multiple sequence alignment(s)</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/infernal">infernal_1.1.5-3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       cmbuild - construct covariance model(s) from structurally annotated RNA multiple sequence alignment(s)

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>cmbuild</b> <u>[options]</u> <u>&lt;cmfile_out&gt;</u> <u>&lt;msafile&gt;</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       For  each  multiple  sequence  alignment  in <u>&lt;msafile&gt;</u> build a covariance model and save it to a new file
       <u>&lt;cmfile_out&gt;.</u>

       The alignment file must be in Stockholm or SELEX format, and must contain consensus  secondary  structure
       annotation.  <b>cmbuild</b> uses the consensus structure to determine the architecture of the CM.

       <u>&lt;msafile&gt;</u>  may  be '-' (dash), which means reading this input from <u>stdin</u> rather than a file.  To use '-',
       you must also specify the alignment file format with <b>--informat</b> <u>&lt;s&gt;</u><b>,</b> as in <b>--informat</b> <b>stockholm</b>  (because
       of a current limitation in our implementation, MSA file formats cannot be autodetected in a nonrewindable
       input stream.)

       <u>&lt;cmfile_out&gt;</u> may not be '-' <u>(stdout),</u> because sending the CM file to <u>stdout</u> would conflict with the other
       text output of the program.

       In  addition  to writing CM(s) to <u>&lt;cmfile_out&gt;,</u> <b>cmbuild</b> also outputs a single line for each model created
       to stdout. Each line has the following fields: "aln": the index of the alignment used to  build  the  CM;
       "idx":  the  index  of  the  CM  in  the  <u>&lt;cmfile_out&gt;;</u> "name": the name of the CM; "nseq": the number of
       sequences in the alignment used to build the CM; "eff_nseq": the effective number of  sequences  used  to
       build  the model; "alen": the length of the alignment used to build the CM; "clen": the number of columns
       from the alignment defined as consensus (match) columns; "bps":  the  number  of  basepairs  in  the  CM;
       "bifs":  the number of bifurcations in the CM; "rel entropy: CM": the total relative entropy of the model
       divided by the number of consensus columns; "rel entropy: HMM": the total relative entropy of  the  model
       ignoring  secondary  structure divided by the number of consensus columns.  "description": description of
       the model/alignment.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>     Help; print a brief reminder of command line usage and available options.

       <b>-n</b> <u>&lt;s&gt;</u> Name the new CM <u>&lt;s&gt;.</u>  The default is to use the name of the alignment (if one is  present  in  the
              <u>&lt;msafile&gt;),</u>  or,  failing  that,  the  name of the <u>&lt;msafile&gt;.</u>  If <u>&lt;msafile&gt;</u> contains more than one
              alignment, <u>-n</u> doesn't work, and every alignment must have a name annotated in the <u>&lt;msafile&gt;</u> (as in
              Stockholm #=GF ID annotation).

       <b>-F</b>     Allow <u>&lt;cmfile_out&gt;</u> to be overwritten. Without this option, if <u>&lt;cmfile_out&gt;</u> already exists, <b>cmbuild</b>
              exits with an error.

       <b>-o</b> <u>&lt;f&gt;</u> Direct the summary output to file <u>&lt;f&gt;,</u> rather than to <u>stdout.</u>

       <b>-O</b> <u>&lt;f&gt;</u> After each model is constructed, resave annotated source alignments to a  file  <u>&lt;f&gt;</u>  in  Stockholm
              format.   Sequences are annoted with what relative sequence weights were assigned.  The alignments
              are also annotated with a reference annotation line indicating  which  columns  were  assigned  as
              consensus.  If  the source alignment had reference annotation ("#=GC RF") it will be replaced with
              the consensus residue of the model for consensus columns and '.' for insert  columns,  unless  the
              <b>--hand</b> option was used for specifying consensus positions, in which case it will be unchanged. Any
              sequences defined as fragments will be annotated as well, using only <b>~</b> characters before the first
              residue and after the final residue, unless the <b>--fraggiven</b> option was used.

              <b>--devhelp</b>  Print help, as with <b>-h</b> <b>,</b> but also include expert options that are not displayed with <b>-h</b>
              <b>.</b>  These expert options are not expected to be relevant for the vast majority of users and so  are
              not  described in the manual page.  The only resources for understanding what they actually do are
              the brief one-line descriptions output when <b>--devhelp</b> is enabled, and the source code.

</pre><h4><b>OPTIONS</b> <b>CONTROLLING</b> <b>MODEL</b> <b>CONSTRUCTION</b></h4><pre>
       <b>--fast</b> Define consensus columns automatically as those that have a fraction &gt;=  <b>symfrac</b>  of  residues  as
              opposed to gaps. (See below for the <b>--symfrac</b> option.) This is the default.

       <b>--hand</b> Use  reference  coordinate  annotation (#=GC RF line, in Stockholm) to determine which columns are
              consensus, and which are inserts.  Any  non-gap  character  indicates  a  consensus  column.  (For
              example,  mark  consensus  columns  with "x", and insert columns with ".".) This option was called
              <b>--rf</b> in previous versions of Infernal (0.1 through 1.0.2).

       <b>--symfrac</b> <u>&lt;x&gt;</u>
              Define the residue fraction threshold necessary to  define  a  consensus  column  when  not  using
              <b>--hand.</b>   The  default  is  0.5.  The  symbol  fraction  in each column is calculated after taking
              relative sequence weighting into account.  Setting this to 0.0 means that every  alignment  column
              will  be  assigned  as  consensus, which may be useful in some cases. Setting it to 1.0 means that
              only columns that include 0 gaps  will  be  assigned  as  consensus.   This  option  replaces  the
              <b>--gapthresh</b>  <u>&lt;y&gt;</u>  option from previous versions of Infernal (0.1 through 1.0.2), with <u>&lt;x&gt;</u> equal to
              (1.0 - <u>&lt;y&gt;).</u>  For example to reproduce behavior for a command of <b>cmbuild</b>  <b>--gapthresh</b>   <b>0.8</b>  in  a
              previous version, use <b>cmbuild</b> <b>--symfrac</b>  <b>0.2</b> with this version.

       <b>--fragthresh</b> <u>&lt;x&gt;</u>
              We  only  want  to  count  terminal gaps as deletions if the aligned sequence is known to be full-
              length, not if it is a fragment (for instance, because only part of it was sequenced). A  fragment
              is  defined  as  any aligned sequence for which the fractional span, defined as its aligned length
              from its first to last residue divided by the total alignment  length,  is  less  than  "0.8"  (by
              default).  Note  that  this  differs from the way HMMER defines fragments (as of v3.3.2).  Setting
              <b>--fragthresh</b> <u>0</u> will define no sequence as a fragment; you might want to do this if  you  know  you
              have a carefully curated alignment of full-length sequences or want to mimic the behavior of older
              versions  of  Infernal  (v1.1  to  v1.1.4).  Setting  <b>--fragthresh</b>  <u>1</u> will define all sequences as
              fragments.  The <b>--fragnrfpos</b> and <b>--fraggiven</b> offer alternative ways to define fragments.

       <b>--fragnrfpos</b> <u>&lt;n&gt;</u>
              Define a sequence as a fragment if it has more than <u>&lt;n&gt;</u> gaps in terminal  consensus  positions  at
              the  5'  <b>or</b> 3' ends. This option can only be used in combination with the <b>--hand</b> option, and if it
              is used, the <b>--fragthresh</b> option is ignored.

       <b>--fraggiven</b>
              Do not infer which sequences are fragments based on their lengths but <b>do</b> use fragment  information
              in the input alignment, if there is any.  For a sequence in the input alignment to be considered a
              fragment,  all  positions  before (5' of) the first nucleotide and all positions after (3' of) the
              final nucleotide must be <b>~</b> symbols. Importantly, <b>~</b> symbols are not allowed anywhere  else  in  the
              alignment.

       <b>--noss</b> Ignore  the  secondary  structure  annotation,  if  any,  in  <u>&lt;msafile&gt;</u>  and  build a CM with zero
              basepairs. This model will be similar to a profile HMM and the <b>cmsearch</b> and <b>cmscan</b>  programs  will
              use  HMM  algorithms  which  are faster than CM ones for this model. Additionally, a zero basepair
              model need not be calibrated with <b>cmcalibrate</b> prior to running <b>cmsearch</b> with it. The <b>--noss</b> option
              must be used if there is no secondary structure annotation in <b>&lt;msafile&gt;.</b>

       <b>--rsearch</b> <u>&lt;f&gt;</u>
              Parameterize emission scores a la RSEARCH, using the RIBOSUM matrix in file <u>&lt;f&gt;.</u>   With  <b>--rsearch</b>
              enabled,  all  alignments in <u>&lt;msafile&gt;</u> must contain exactly one sequence or the <b>--call</b> option must
              also be enabled. All positions in each sequence will be considered consensus "columns".  Actually,
              the emission scores for these models will not be identical to RIBOSUM scores due of differences in
              the modelling strategy between Infernal and RSEARCH, but they will  be  as  similar  as  possible.
              RIBOSUM  matrix  files are included with Infernal in the "matrices/" subdirectory of the top-level
              "infernal-xxx" directory.  RIBOSUM matrices are substitution score matrices  trained  specifically
              for  structural  RNAs with separate single stranded residue and base pair substitution scores. For
              more information see the RSEARCH publication (Klein and Eddy, BMC Bioinformatics 4:44, 2003).

       <b>--consrf</b>
              With <b>--hand</b> use the model's consensus sequence for the model reference annotation instead  of  the
              RF annotation from the input alignment.

</pre><h4><b>OTHER</b> <b>MODEL</b> <b>CONSTRUCTION</b> <b>OPTIONS</b></h4><pre>
       <b>--null</b> <u>&lt;f&gt;</u>
              Read  a  null  model  from  <u>&lt;f&gt;.</u>  The null model defines the probability of each RNA nucleotide in
              background sequence, the default is to use 0.25 for each nucleotide.  The format of null files  is
              specified in the user guide.

       <b>--prior</b> <u>&lt;f&gt;</u>
              Read  a  Dirichlet  prior  from <u>&lt;f&gt;,</u> replacing the default mixture Dirichlet.  The format of prior
              files is specified in the user guide.

       Use <b>--devhelp</b> to see additional, otherwise undocumented, model construction options.

</pre><h4><b>OPTIONS</b> <b>CONTROLLING</b> <b>RELATIVE</b> <b>WEIGHTS</b></h4><pre>
       <b>cmbuild</b> uses an ad hoc sequence weighting algorithm to downweight closely related sequences and  upweight
       distantly  related  ones.  This  has  the  effect  of  making  models  less biased by uneven phylogenetic
       representation. For example, two identical sequences would typically each receive half  the  weight  that
       one sequence would.  These options control which algorithm gets used.

       <b>--wpb</b>  Use  the  Henikoff  position-based sequence weighting scheme [Henikoff and Henikoff, J. Mol. Biol.
              243:574, 1994].  This is the default.

       <b>--wgsc</b> Use the Gerstein/Sonnhammer/Chothia weighting algorithm [Gerstein et al, J. Mol.  Biol.  235:1067,
              1994].

       <b>--wnone</b>
              Turn sequence weighting off; e.g. explicitly set all sequence weights to 1.0.

       <b>--wgiven</b>
              Use sequence weights as given in annotation in the input alignment file. If no weights were given,
              assume   they   are   all  1.0.   The  default  is  to  determine  new  sequence  weights  by  the
              Gerstein/Sonnhammer/Chothia algorithm, ignoring any annotated weights.

       <b>--wblosum</b>
              Use the BLOSUM filtering algorithm to weight the sequences, instead of the default GSC  weighting.
              Cluster  the  sequences  at  a  given percentage identity (see <b>--wid);</b> assign each cluster a total
              weight of 1.0, distributed equally amongst the members of that cluster.

       <b>--wid</b> <u>&lt;x&gt;</u>
              Controls the behavior of the <u>--wblosum</u> weighting  option  by  setting  the  percent  identity  for
              clustering the alignment to <u>&lt;x&gt;.</u>

</pre><h4><b>OPTIONS</b> <b>CONTROLLING</b> <b>EFFECTIVE</b> <b>SEQUENCE</b> <b>NUMBER</b></h4><pre>
       After  relative  weights are determined, they are normalized to sum to a total effective sequence number,
       <u>eff_nseq.</u>  This number may be the actual number of sequences in the alignment, but it  is  almost  always
       smaller  than  that.  The default entropy weighting method <u>(--eent)</u> reduces the effective sequence number
       to reduce the information content (relative entropy, or average expected  score  on  true  homologs)  per
       consensus  position. The target relative entropy is controlled by a two-parameter function, where the two
       parameters are settable with <u>--ere</u> and <u>--esigma.</u>

       <b>--eent</b> Use the entropy weighting strategy to determine the effective sequence number that gives a  target
              mean match state relative entropy. This option is the default, and can be turned off with <b>--enone.</b>
              The  default  target  mean  match  state  relative entropy is 0.59 bits for models with at least 1
              basepair and 0.38 bits for models with zero basepairs, but can be changed with <b>--ere.</b>  The default
              of 0.59 or 0.38 bits is automatically changed if the total relative entropy of the  model  (summed
              match  state  relative entropy) is less than a cutoff, which is controlled by the <b>--esigma</b> option.
              If you really want to play with that option, consult the source code.  Additionally, the effective
              sequence number cannot be larger than the number of sequences in the alignment, although this  can
              be overridden to set the maximum possible effective sequence number with the <b>--emaxseq</b> option.

       <b>--enone</b>
              Turn  off  the  entropy  weighting  strategy.  The effective sequence number is just the number of
              sequences in the alignment.

       <b>--ere</b> <u>&lt;x&gt;</u>
              Set the target mean match state relative entropy as <u>&lt;x&gt;.</u>  By default the target  relative  entropy
              per  match position is 0.59 bits for models with at least 1 basepair and 0.38 for models with zero
              basepairs.

       <b>--eminseq</b> <u>&lt;x&gt;</u>
              Define the minimum allowed effective sequence number as <u>&lt;x&gt;.</u>

       <b>--emaxseq</b> <u>&lt;x&gt;</u>
              Define the maximum allowed effective sequence number as <u>&lt;x&gt;.</u>  This number can be larger  than  the
              number of sequences in the alignment.

       <b>--ehmmre</b> <u>&lt;x&gt;</u>
              Set the target HMM mean match state relative entropy as <u>&lt;x&gt;.</u>  Entropy for basepairing match states
              is calculated using marginalized basepair emission probabilities.

       <b>--eset</b> <u>&lt;x&gt;</u>
              Set the effective sequence number for entropy weighting as <u>&lt;x&gt;.</u>

</pre><h4><b>OPTIONS</b> <b>CONTROLLING</b> <b>FILTER</b> <b>P7</b> <b>HMM</b> <b>CONSTRUCTION</b></h4><pre>
       For  each  CM that <b>cmbuild</b> constructs, an accompanying filter p7 HMM is built from the input alignment as
       well. These options control filter HMM construction:

       <b>--p7ere</b> <u>&lt;x&gt;</u>
              Set the target mean match state relative entropy for the filter p7 HMM as  <u>&lt;x&gt;.</u>   By  default  the
              target relative entropy per match position is 0.38 bits.

       <b>--p7ml</b> Use  a  maximum likelihood p7 HMM built from the CM as the filter HMM. This HMM will be as similar
              as possible to the CM (while necessarily ignorant of secondary structure).

       Use <b>--devhelp</b> to see additional, otherwise undocumented, filter HMM construction options.

</pre><h4><b>OPTIONS</b> <b>CONTROLLING</b> <b>FILTER</b> <b>P7</b> <b>HMM</b> <b>CALIBRATION</b></h4><pre>
       After building each filter HMM, <b>cmbuild</b> determines appropriate E-value parameters to use during filtering
       in <b>cmsearch</b> and <b>cmscan</b> by  sampling  a  set  of  sequences  and  searching  them  with  each  HMM  filter
       configuration and algorithm.

       <b>--EmN</b>  <u>&lt;n&gt;</u>  Set  the  number  of  sampled  sequences for local MSV filter HMM calibration to <u>&lt;n&gt;.</u>  200 by
       default.

       <b>--EvN</b> <u>&lt;n&gt;</u> Set the number of sampled sequences for local Viterbi filter HMM calibration to  <u>&lt;n&gt;.</u>   200  by
       default.

       <b>--ElfN</b>  <u>&lt;n&gt;</u>  Set the number of sampled sequences for local Forward filter HMM calibration to <u>&lt;n&gt;.</u>  200 by
       default.

       <b>--EgfN</b> <u>&lt;n&gt;</u> Set the number of sampled sequences for glocal Forward filter HMM calibration to <u>&lt;n&gt;.</u>  200  by
       default.

       Use <b>--devhelp</b> to see additional, otherwise undocumented, filter HMM calibration options.

</pre><h4><b>OPTIONS</b> <b>FOR</b> <b>REFINING</b> <b>THE</b> <b>INPUT</b> <b>ALIGNMENT</b></h4><pre>
       <b>--refine</b> <u>&lt;f&gt;</u>
              Attempt  to  refine the alignment before building the CM using expectation-maximization (EM). A CM
              is first built from the initial alignment as usual. Then,  the  sequences  in  the  alignment  are
              realigned  optimally (with the HMM banded CYK algorithm, optimal means optimal given the bands) to
              the CM, and a new CM is built from the resulting alignment. The sequences are  then  realigned  to
              the  new  CM,  and  a  new  CM  is built from that alignment. This is continued until convergence,
              specifically when the alignments for two successive iterations  are  not  significantly  different
              (the  summed  bit  scores  of  all the sequences in the alignment changes less than 1% between two
              successive iterations). The final alignment (the alignment used to build the CM that gets  written
              to <u>&lt;cmfile_out&gt;)</u> is written to <u>&lt;f&gt;.</u>

       <b>-l</b>     With  <b>--refine,</b>  turn  on the local alignment algorithm, which allows the alignment to span two or
              more subsequences if necessary (e.g. if the structures of the query model and target sequence  are
              only  partially  shared),  allowing  certain large insertions and deletions in the structure to be
              penalized differently than normal indels.  The default is to globally align the query model to the
              target sequences.

       <b>--gibbs</b>
              Modifies the behavior of <b>--refine</b> so Gibbs sampling is used instead of EM. The difference is  that
              during  the  alignment  stage  the  alignment  is  not  necessarily  optimal, instead an alignment
              (parsetree) for each sequences is  sampled  from  the  posterior  distribution  of  alignments  as
              determined  by  the  Inside  algorithm. Due to this sampling step <b>--gibbs</b> is non-deterministic, so
              different runs with the same alignment may yield different results. This is not true when <b>--refine</b>
              is used without the <b>--gibbs</b> option, in which case the final alignment and CM will  always  be  the
              same.  When  <b>--gibbs</b>  is  enabled,  the  <b>--seed</b>   <b>&lt;n&gt;</b> option can be used to seed the random number
              generator predictably, making the results reproducible.  The goal of the <b>--gibbs</b> option is to help
              expert RNA alignment curators refine structural alignments by allowing them to observe alternative
              high scoring alignments.

       <b>--seed</b> <u>&lt;n&gt;</u>
              Seed the random number generator with <u>&lt;n&gt;,</u> an integer &gt;= 0.  This  option  can  only  be  used  in
              combination  with  <b>--gibbs.</b>   If  <u>&lt;n&gt;</u>  is  nonzero,  stochastic  sampling  of  alignments  will be
              reproducible; the same command will give the same  results.   If  <u>&lt;n&gt;</u>  is  0,  the  random  number
              generator  is  seeded  arbitrarily,  and stochastic samplings may vary from run to run of the same
              command.  The default seed is 0.

       <b>--cyk</b>  With <b>--refine,</b> align with the CYK algorithm. By default the optimal accuracy  algorithm  is  used.
              There is more information on this in the <b>cmalign</b> manual page.

       <b>--notrunc</b>
              With  <b>--refine,</b>  turn off the the truncated alignment algorithm. There is more information on this
              in the <b>cmalign</b> manual page.

       <b>--miss</b> With <b>--refine,</b> in the final alignment and each intermediate alignment, consider all sequences with
              terminal gaps as fragments for purposes of building models from those alignments. You may want  to
              do this if you have many sequences that are not full length, e.g. fragmentary because only part of
              it was sequenced.

       Use  <b>--devhelp</b>  to  see additional, otherwise undocumented, alignment refinement options as well as other
       output file options and options for building multiple models for a single alignment.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       See <b><a href="../man1/infernal.1.html">infernal</a>(1)</b> for a master man page with a list of all the individual man pages  for  programs  in  the
       Infernal package.

       For complete documentation, see the user guide that came with your Infernal distribution (Userguide.pdf);
       or see the Infernal web page (<a href="http://eddylab.org/infernal/">http://eddylab.org/infernal/</a>).

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright (C) 2023 Howard Hughes Medical Institute.
       Freely distributed under the BSD open source license.

       For  additional  information  on  copyright and licensing, see the file called COPYRIGHT in your Infernal
       source distribution, or see the Infernal web page (<a href="http://eddylab.org/infernal/">http://eddylab.org/infernal/</a>).

</pre><h4><b>AUTHOR</b></h4><pre>
       <a href="http://eddylab.org">http://eddylab.org</a>

Infernal 1.1.5                                      Sep 2023                                          <u><a href="../man1/cmbuild.1.html">cmbuild</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>