<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>dialign-tx - Segment-based multiple sequence alignment</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/dialign-tx">dialign-tx_1.0.2-15build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       dialign-tx - Segment-based multiple sequence alignment

</pre><h4><b>SYNOPSIS</b></h4><pre>

       <b>dialign-tx</b> [<b>OPTIONS</b>] {<u>conf-directory</u>} {<u>fasta-file</u>} [<u>fasta-out-file</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       DIALIGN-TX is an improved algorithm for segment-based multiple protein alignments. DIALIGN-TX is a
       complete reimplementation of the segment-base approach including several new improvements and heuristics
       that significantly enhance the quality of the output alignments compared to DIALIGN 2.2. This significant
       superiority has been observed on local as well on global alignment benchmarks.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-d</b>
           Debug-Mode [DEFAULT 0]

               0 no debug statements

               1 debugs the current phase of the processing

               2 very loquacious debugging

               5 hardcore debugging

       <b>-s</b>
           Maximum amount of input sequences [DEFAULT 5000].

       <b>-a</b>
           Maximum number of characters per line in a FASTA file [DEFAULT 100].

       <b>-c</b>
           Maximum amount of characters per line when printing a sequence [DEFAULT 80].

       <b>-l</b>
           sensitivity mode, the higher the level the less likely spurious random fragments are aligned in local
           alignments [DEFAULT 0]

               0 switched off

               1 level-1, reduced sensitivity

               2 level-2, strongly reduced sensitivity

       <b>-m</b>
           Score matrix file name (in the configuration directory) [DEFAULT PROTEIN: BLOSUM.scr] / [DEFAULT DNA:
           dna_matrix.scr].

       <b>-w</b>
           Defines  the  minimum  weight  when  the  weight formula is changed to 1-pow(1-prob, factor) [DEFAULT
           0.000000065].

       <b>-p</b>
           Probability  distribution  file   name   (in   the   configuration   directory)   [DEFAULT   PROTEIN:
           BLOSUM.diag_prob_t10] / [DEFAULT DNA: dna_diag_prob_100_exp_550000].

       <b>-v</b>
           Add to each score (to prevent negative values) [DEFAULT 0].

       <b>-t</b>

           “Even” threshold for low score for sequences alignment [DEFAULT PROTEIN: 4] / [DEFAULT DNA: 0].

       <b>-n</b>
           Maximum number of consecutive positions for window containing low scoring positions [DEFAULT PROTEIN:
           4] / [DEFAULT DNA: 1].

       <b>-g</b>
           Global minimum fragment length for stop criterion [DEFAULT PROTEIN: 40] / [DEFAULT DNA: 1].

       <b>-m</b>
           Minimal  allowed average score in frag window containing low scoring positions [DEFAULT PROTEIN: 4.0]
           / [DEFAULT DNA: 0.9].

       <b>-o</b>
           Whether overlap weights are calculated or not [DEFAULT 0].

       <b>-f</b>
           Minimum fragment length [DEFAULT 1].

       <b>-r</b>
           Threshold weight to consider the fragment at all [DEFAULT 0.0].

       <b>-u</b>
           [DEFAULT 0]

                             1: Only use a sqrt(amount_of_seqs)  stripe  of  neighbour  sequences  to  calculate
               pairwise alignments (increase performance).

                             0: All pairwise alignments will be calculated.

       <b>-A</b>
           Optional anchor file. [DEFAULT none]

       <b>-D</b>
           Input is DNA-sequence.

       <b>-T</b>
           Translate DNA into aminoacids from begin to end (length will be cut to mod 3 = 0).

           Warning Do not use <b>-D</b> with this option (Default values for PROTEIN input will be loaded).

       <b>-L</b>
           Compare only longest Open Reading Frame.

           Warning Do not use <b>-D</b> with this option (Default values for PROTEIN input will be loaded).

       <b>-O</b>
           Translate DNA to aminoacids, reading frame for each sequence calculated due to its longest ORF.

           Warning Do not use <b>-D</b> with this option (Default values for PROTEIN input will be loaded).

       <b>-P</b>
           Output in aminoacids, no retranslation of DNA sequences [DEFAULT: input = output].

       <b>-F</b>
           Fast mode (implies -l0, since it already significantly reduces sensitivity).

       <b>-C</b>
           Generate probability table saved in /usr/share/dialign-tx/prob_table and exit.

       <b>-H</b>, <b>-h</b>
           Print this message.

</pre><h4><b>FILES</b></h4><pre>
       <u>/usr/share/dialign-tx</u>
           This  is the default conf-directory that <b>dialign-tx</b> expects as its first argument, as supplied in the
           upstream sources.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       DIALIGN-TX is a re-implementation of <b><a href="../man1/dialign2-2.1.html">dialign2-2</a></b>(1). (See <a href="http://dialign.gobics.de/">http://dialign.gobics.de/</a> for  more  information
       about DIALIGN2).

       The website of DIALIGN-TX is <a href="http://dialign-tx.gobics.de/">http://dialign-tx.gobics.de/</a>

</pre><h4><b>REFERENCES</b></h4><pre>
       Amarendran  R.  Subramanian,  Michael  Kaufmann,  Burkhard  Morgenstern:  DIALIGN-TX:  improvement of the
       segment-based approach for multiple sequence alignment by  combining  greedy  and  progressive  alignment
       strategies, Algorithms for Molecular Biology 3:6, 2008.

       Amarendran  R.  Subramanian,  Jan  Weyer-Menkhoff,  Michael Kaufmann, Burkhard Morgenstern: DIALIGN-T: An
       improved algorithm for segment-based multiple sequence alignment. BMC Bioinformatics 2005, 6:66.

</pre><h4><b>AUTHORS</b></h4><pre>
       <b>Amarendran</b> <b>R.</b> <b>Subramanian</b> &lt;<a href="mailto:subraman@informatik.uni-tuebingen.de">subraman@informatik.uni-tuebingen.de</a>&gt;
           Author of dialign-tx

       <b>Volker</b> <b>Menrad</b>
           Co-author of dialign-tx

       <b>Dorothea</b> <b>Emig</b>
           Co-author of dialign-tx

       <b>Charles</b> <b>Plessy</b> &lt;<a href="mailto:plessy@debian.org">plessy@debian.org</a>&gt;
           Converted this guide in DocBook XML for the Debian distribution.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright © 2004, 2005, 2006, 2007, 2008 Amarendran R. Subramanian (DIALIGN-TX)
       Copyright © 2004 Volker Menrad (DIALIGN-TX)
       Copyright © 2004 Dorothea Emig (DIALIGN-TX)
       Copyright © 2007, 2008 Charles Plessy (This document and its XML source.)

       DIALIGN-TX is free software; you can redistribute it and/or modify it under the terms of the  GNU  Lesser
       General  Public  License as published by the Free Software Foundation; either version 2.1 of the License,
       or (at your option) any later version.

       DIALIGN-TX is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even  the
       implied  warranty  of  MERCHANTABILITY  or  FITNESS  FOR A PARTICULAR PURPOSE. See the GNU Lesser General
       Public License for more details.

       You should have received a copy of the GNU Lesser General Public License along with this library; if not,
       write to the Free Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA

       On  Debian  system,  a  copy   of   the   GNU   Lesser   General   Public   License   is   available   in
       <u>/usr/share/common-licences</u>.

       This  documentation  and its XML source file can be used, modified and redistributed under the same terms
       as DIALIGN-TX itself.

dialign-tx 1.0.2                                   12/15/2008                                      <u><a href="../man1/DIALIGN-TX.1.html">DIALIGN-TX</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>