<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>gff2aplot - Converting GFF files for pairwise alignments to PostScript.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/gff2aplot">gff2aplot_2.0-15_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       gff2aplot - Converting GFF files for pairwise alignments to PostScript.

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>gff2aplot</b> [options] &lt;GFF_files|STDIN&gt;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This manual page documents briefly the <b>gff2aplot</b> command.

       <b>gff2aplot</b>  is a program that draws color-filled alignment plots from GFF files for that alignment and two
       sequences annotations.

</pre><h4><b>OPTIONS</b></h4><pre>
       A double dash on itself "--" signals end of the options and start of file names (if present). You can use
       a single dash "-" as STDIN placeholder. Available options and a short description are listed here:

       <b>-h,</b> <b>--help</b>
              Show summary of options.

       <b>--version</b>
              Shows current version and exits.

       <b>-v,</b> <b>--verbose</b>
              Verbose mode, a full report is sent to standard error (default is set to showing only WARNINGS).

       <b>-V,</b> <b>--logs-filename</b> <b>&lt;filename&gt;</b>
              Report is written to a log file.

       <b>-q,</b> <b>--quiet</b>
              Quiet mode, do not show any message/warning to standard error (reporting only ERRORS).

       <b>-p,</b> <b>--page-size</b> <b>&lt;format_name&gt;</b>
              Setting a page size among pre-defined ones see below for a list of available page formats).

       <b>--margin-left</b> <b>&lt;length&gt;</b>

       <b>--margin-right</b> <b>&lt;length&gt;</b>

       <b>--margin-top</b> <b>&lt;length&gt;</b>

       <b>--margin-bottom</b> <b>&lt;length&gt;</b>
              Setting each page margin to &lt;length&gt;. If no units are provided, points are assumed,  you  can  use
              points, milimeters, centimeters or inches (pt, mm, cm or in, respectively).

       <b>-B,</b> <b>--background-color</b> <b>&lt;color&gt;</b>
              Background color.

       <b>-F,</b> <b>--foreground-color</b> <b>&lt;color&gt;</b>
              Foreground color.

       <b>-T,</b> <b>--title</b> <b>&lt;string&gt;</b>
              Definning plot title.

       <b>-t,</b> <b>--subtitle</b> <b>&lt;string&gt;</b>
              Definning plot subtitle.

       <b>-X,</b> <b>--x-label</b> <b>&lt;string&gt;</b>
              Defining X-axis label.

       <b>-Y,</b> <b>--y-label</b> <b>&lt;Y-Label&gt;</b>
              Defining Y-axis label.

       <b>-L,</b> <b>--percent-box-label</b> <b>&lt;string&gt;</b>
              Defining percent-box label.

       <b>-l,</b> <b>--extra-box-label</b> <b>&lt;XBox-Label&gt;</b>
              Definning Extra-Box Label.

       <b>-x,</b> <b>--x-sequence-coords</b> <b>&lt;pos..pos&gt;</b>

       <b>-S,</b> <b>--start-x-sequence</b> <b>&lt;pos&gt;</b>
              Sets X-sequence first nucleotide.

       <b>-E,</b> <b>--end-x-sequence</b> <b>&lt;pos&gt;</b>
              Sets X-sequence last nucleotide.

       <b>-y,</b> <b>---y-sequence-coords</b> <b>&lt;pos..pos&gt;</b>

       <b>-s,</b> <b>--start-y-sequence</b> <b>&lt;pos&gt;</b>
              Sets Y-sequence first nucleotide.

       <b>-e,</b> <b>--end-y-sequence</b> <b>&lt;pos&gt;</b>
              Sets Y-sequence last nucleotide.

       <b>--x-sequence-zoom</b> <b>&lt;pos..pos&gt;</b>

       <b>--y-sequence-zoom</b> <b>&lt;pos..pos&gt;</b>

       <b>-Z,</b> <b>--zoom</b> <b>[</b> <b>[-S</b> <b>&lt;pos&gt;]</b> <b>[-E</b> <b>&lt;pos&gt;]</b> <b>[-s</b> <b>&lt;pos&gt;]</b> <b>[-e</b> <b>&lt;pos&gt;]</b> <b>]</b>
              This option zooms an area you have selected with -S,-E,-s,-e (all 4 are optional).

       <b>-z,</b> <b>--zoom-area</b> <b>[</b> <b>[-S</b> <b>&lt;pos&gt;]</b> <b>[-E</b> <b>&lt;pos&gt;]</b> <b>[-s</b> <b>&lt;pos&gt;]</b> <b>[-e</b> <b>&lt;pos&gt;]</b> <b>]</b>
              This option marks a zoom area on your plot, but does not make a zoom.

       <b>-A,</b> <b>--alignment-name</b> <b>&lt;SeqXName:SeqYName&gt;</b>
              Defining  which  alignment  is going to be plotted if you have more than one alignment in your gff
              input.

       <b>-N,</b> <b>--x-sequence-name</b> <b>&lt;SeqXName&gt;</b>
              Defining which sequence is going to be plotted at X-axis.

       <b>-n,</b> <b>--y-sequence-name</b> <b>&lt;SeqYName&gt;</b>
              Defining which sequence is going to be plotted at Y-axis.

       <b>-r,</b> <b>--aplot-xy-noteq</b>
              By default X and Y axes have same length, this option disables such behaviour, so X and Y sequence
              will have axes-lengths proportional to their nucleotide lengths.

       <b>-R,</b> <b>--xy-axes-scale</b> <b>&lt;X/Y</b> <b>ratio&gt;</b>
              This option allows one to set a different scale between X and Y axes lengths (by default is  '1').
              Below  1  values make Y larger than X, and larger than 1 result in getting X larger than Y. # Must
              be explained better.

       <b>-W,</b> <b>--aln-scale-width</b>
              Scaling score on width for Aplot lines.

       <b>-w,</b> <b>--aln-scale-color</b>
              Scaling score on color for Aplot lines.

       <b>-K,</b> <b>--show-ribbons</b> <b>&lt;ribbon_type&gt;</b>
              Force Ribbons for all features on axes to be: (N)one, (L)ines, (R)ibbons, (B)oth.

       <b>-G,</b> <b>--show-grid</b>
              Switches 'on' grid (default is 'off').

       <b>-g,</b> <b>--hide-grid</b>
              Switches 'off' grid (if switched on from customization files).

       <b>-I,</b> <b>--show-percent-box</b>
              Switches 'on' Percent box (default is 'off').

       <b>-i,</b> <b>--hide-percent-box</b>
              Switches 'off' Percent box (if set to 'on' on custom files).

       <b>-O,</b> <b>--show-extra-box</b>
              Switches 'on' Extra box (default is 'off').

       <b>-o,</b> <b>--hide-extra-box</b>
              Switches 'off' Extra box (if set to 'on' on custom files).

       <b>-D,</b> <b>--aplot-box-color</b> <b>&lt;color&gt;</b>
              Aplot main box background color.

       <b>-d,</b> <b>--percent-box-color</b> <b>&lt;color&gt;</b>
              Percent box background color.

       <b>-b,</b> <b>--extra-box-color</b> <b>&lt;color&gt;</b>
              Extra box background color.

       <b>--nopswarnings</b>
              Switch off warnings that may appear on the final PostScript figure when  X  sequence,  Y  sequence
              and/or alignment input data is missing.

       <b>-a,</b> <b>--hide-credits</b>
              Switch off gff2aplot credits line on plot.

       <b>--debug</b>
              Reporting  variable  contents  when  testing the program. Requires that log report file option was
              also activated.

       <b>--layout-var</b> <b>'&lt;variable=value&gt;'</b>

       <b>--sequence-var</b> <b>'&lt;sequence::variable=value&gt;'</b>

       <b>--source-var</b> <b>'&lt;source::variable=value&gt;'</b>

       <b>--strand-var</b> <b>'&lt;strand::variable=value&gt;'</b>

       <b>--group-var</b> <b>'&lt;group::variable=value&gt;'</b>

       <b>--feature-var</b> <b>'&lt;feature::variable=value&gt;'</b>
              Loading a feature/group/strand/source/sequence/layout customization  variable  from  command-line.
              You can set several variables by repeating any of these options, i.e.:
                 ... --feature-var 'cds::feature_shape=box'
                     --feature-var 'cds::feature_color=blue' ...

       <b>-C,</b> <b>--custom-filename</b> <b>&lt;filename&gt;</b>
              Loading  customization  parameters  from  a given file (if default ".gff2aplotrc" exists is loaded
              before it).

       Those are the colors defined in gff2aplot:
               + Default Colors: black white.
               + Color Palette:
                 grey pink magenta violet blue skyblue cyan seagreen
                 green limegreen yellow orange red brown
                 You can get up to nine color shades from Variable Colors with "verydeep",  "deep",  "verydark",
              "dark", "light", "verylight", "pale" and "verypale" prefixes, as example:
                 verydeepblue,  deepblue,  verydarkblue,  darkblue, blue, lightblue, verylightblue, paleblue and
              verypaleblue.
               + Dynamic colors: gradient rainbow.

              The following page sizes are available: from A0 to A10, from B0 to B10, 10x14,  executive,  folio,
              ledger, legal, letter, quarto, statement and tabloid.

</pre><h4><b>ENVIRONMENT</b> <b>VARIABLES</b></h4><pre>
       There are two environmental variables that can be set by users to their preferences:
        +  You  can  specify  the  path  where  gff2aplot  can  find  the  default files with the shell variable
       "GFF2APLOT_CUSTOMPATH". Default value is the path where you are running gff2aplot.
        +  You  can  also  define  the   default   custom   filename   you   will   like   with   the   variable
       "GFF2APLOT_CUSTOMFILE", program default filename for custom file is ".gff2aplotrc".
        +  Now  gff2aplot does not need to write any temporary file, so that previous versions default temporary
       directory path variable ("GFF2APLOT_TMP") is no longer used.
        + Setting those vars in Bourne-shell and C-shell:
          o Using a Bourne-Shell (e.g. bash):
               export GFF2APLOT_CUSTOMPATH="path"      export GFF2APLOT_CUSTOMFILE="file_name"
          o Using a C-Shell:
               setenv GFF2APLOT_CUSTOMPATH "path"      setenv GFF2APLOT_CUSTOMFILE "file_name"

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/ali2gff.1.html">ali2gff</a></b>(1), <b><a href="../man1/blat2gff.1.html">blat2gff</a></b>(1), <b><a href="../man1/parseblast.1.html">parseblast</a></b>(1), <b><a href="../man1/sim2gff.1.html">sim2gff</a></b>(1).

</pre><h4><b>AUTHOR</b></h4><pre>
       gff2aplot was written by Josep Francesc ABRIL FERRANDO &lt;<a href="mailto:jabril@imim.es">jabril@imim.es</a>&gt;, Thomas  WIEHE  &lt;<a href="mailto:twiehe@zedat.fu">twiehe@zedat.fu</a>-
       berlin.de&gt; and Roderic GUIGO SERRA &lt;<a href="mailto:rguigo@imim.es">rguigo@imim.es</a>&gt;.

       This manual page was written by Nelson A. de Oliveira &lt;<a href="mailto:naoliv@gmail.com">naoliv@gmail.com</a>&gt;, for the Debian project (but may
       be used by others).

                                         Mon, 21 Mar 2005 21:44:15 -0300                            <u><a href="../man1/GFF2APLOT.1.html">GFF2APLOT</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>