<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>gt-csa - Transform spliced alignments from GFF3 file into consensus spliced alignments.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/genometools">genometools_1.6.5+ds-2.2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       gt-csa - Transform spliced alignments from GFF3 file into consensus spliced alignments.

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>gt</b> <b>csa</b> [option ...] [GFF3_file]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>-join-length</b> [<u>value</u>]
           set join length for the spliced alignment clustering (default: 300)

       <b>-v</b> [<u>yes|no</u>]
           be verbose (default: no)

       <b>-o</b> [<u>filename</u>]
           redirect output to specified file (default: undefined)

       <b>-gzip</b> [<u>yes|no</u>]
           write gzip compressed output file (default: no)

       <b>-bzip2</b> [<u>yes|no</u>]
           write bzip2 compressed output file (default: no)

       <b>-force</b> [<u>yes|no</u>]
           force writing to output file (default: no)

       <b>-help</b>
           display help and exit

       <b>-version</b>
           display version information and exit

</pre><h4><b>EXAMPLE:</b></h4><pre>
       Letâ€™s assume we have a GFF3 file <u>csa_example_spliced_alignments.gff3</u> containing the following four
       overlapping spliced alignments (represented as genes with exons as children):

           ##gff-version 3
           ##sequence-region   seq 1 290
           seq .       gene    1       209     .       +       .       ID=gene1
           seq .       exon    1       90      .       +       .       Parent=gene1
           seq .       exon    110     190     .       +       .       Parent=gene1
           seq .       exon    201     209     .       +       .       Parent=gene1
           ###
           seq .       gene    1       290     .       +       .       ID=gene2
           seq .       exon    1       90      .       +       .       Parent=gene2
           seq .       exon    101     190     .       +       .       Parent=gene2
           seq .       exon    201     290     .       +       .       Parent=gene2
           ###
           seq .       gene    10      290     .       +       .       ID=gene3
           seq .       exon    10      90      .       +       .       Parent=gene3
           seq .       exon    110     190     .       +       .       Parent=gene3
           seq .       exon    201     290     .       +       .       Parent=gene3
           ###
           seq .       gene    181     290     .       +       .       ID=gene4
           seq .       exon    181     190     .       +       .       Parent=gene4
           seq .       exon    201     290     .       +       .       Parent=gene4
           ###

       To compute the consensus spliced alignments we call:

           $ gt csa csa_example_spliced_alignments.gff3

       Which returns:

           ##gff-version 3
           ##sequence-region   seq 1 290
           seq gt csa  gene    1       290     .       +       .       ID=gene1
           seq gt csa  mRNA    1       290     .       +       .       ID=mRNA1;Parent=gene1
           seq gt csa  exon    1       90      .       +       .       Parent=mRNA1
           seq gt csa  exon    110     190     .       +       .       Parent=mRNA1
           seq gt csa  exon    201     290     .       +       .       Parent=mRNA1
           seq gt csa  mRNA    1       290     .       +       .       ID=mRNA2;Parent=gene1
           seq gt csa  exon    1       90      .       +       .       Parent=mRNA2
           seq gt csa  exon    101     190     .       +       .       Parent=mRNA2
           seq gt csa  exon    201     290     .       +       .       Parent=mRNA2
           ###

       As one can see, they have been combined into a consensus spliced alignment (represented as genes with
       mRNAs as children which in turn have exons as children) with two alternative splice forms. The first and
       the third spliced alignment have been combined into the first alternative splice form (mRNA1) and the the
       second and the fourth spliced alignment into the second alternative splice form (mRNA2).

       As one can see, the second exon from the first alternative splice form is shorter than the corresponding
       exon from the second alternative splice form.

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       Report bugs to https://github.com/genometools/genometools/issues.

GenomeTools 1.6.5                                  04/27/2024                                          <u><a href="../man1/GT-CSA.1.html">GT-CSA</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>