<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>hmm2build - build a profile HMM from an alignment</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/hmmer2">hmmer2_2.3.2+dfsg-12_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       hmm2build - build a profile HMM from an alignment

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>hmm2build</b> <u>[options]</u> <u>hmmfile</u> <u>alignfile</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>hmm2build</b>  reads  a  multiple sequence alignment file <u>alignfile</u> , builds a new profile HMM, and saves the
       HMM in <u>hmmfile.</u>

       <u>alignfile</u> may be in ClustalW, GCG MSF, SELEX, Stockholm, or aligned FASTA alignment format. The format is
       automatically detected.

       By default, the model is configured to find one or more nonoverlapping alignments to the complete  model:
       multiple  global  alignments  with respect to the model, and local with respect to the sequence.  This is
       analogous to the behavior of the <b>hmmls</b> program of HMMER 1.  To configure the  model  for  multiple  <u>local</u>
       alignments  with respect to the model and local with respect to the sequence, a la the old program <b>hmmfs,</b>
       use the <b>-f</b> (fragment) option. More rarely, you may want to  configure  the  model  for  a  single  global
       alignment  (global  with  respect  to  both model and sequence), using the <b>-g</b> option; or to configure the
       model for a single local/local alignment (a la standard Smith/Waterman, or the old  <b>hmmsw</b>  program),  use
       the <b>-s</b> option.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-f</b>     Configure  the  model  for finding multiple domains per sequence, where each domain can be a local
              (fragmentary) alignment. This is analogous to the old <b>hmmfs</b> program of HMMER 1.

       <b>-g</b>     Configure the model for finding a single global alignment to a target sequence, analogous  to  the
              old <b>hmms</b> program of HMMER 1.

       <b>-h</b>     Print brief help; includes version number and summary of all options, including expert options.

       <b>-n</b> <u>&lt;s&gt;</u> Name  this  HMM <u>&lt;s&gt;.</u>  <u>&lt;s&gt;</u> can be any string of non-whitespace characters (e.g. one "word").  There
              is no length limit (at least not one imposed by HMMER; your shell will complain about command line
              lengths first).

       <b>-o</b> <u>&lt;f&gt;</u> Re-save the starting alignment to <u>&lt;f&gt;,</u> in Stockholm format.  The columns which  were  assigned  to
              match  states  will be marked with x's in an #=RF annotation line.  If either the <b>--hand</b> or <b>--fast</b>
              construction options were chosen, the alignment may have been slightly altered  to  be  compatible
              with Plan 7 transitions, so saving the final alignment and comparing to the starting alignment can
              let  you  view  these  alterations.  See the User's Guide for more information on this arcane side
              effect.

       <b>-s</b>     Configure the model for finding a single local alignment per target sequence. This is analogous to
              the standard Smith/Waterman algorithm or the <b>hmmsw</b> program of HMMER 1.

       <b>-A</b>     Append this model to an existing <u>hmmfile</u> rather than creating <u>hmmfile.</u>  Useful  for  building  HMM
              libraries (like Pfam).

       <b>-F</b>     Force  overwriting  of  an existing <u>hmmfile.</u>  Otherwise HMMER will refuse to clobber your existing
              HMM files, for safety's sake.

</pre><h4><b>EXPERT</b> <b>OPTIONS</b></h4><pre>
       <b>--amino</b>
              Force the sequence alignment to be interpreted as amino acid sequences. Normally HMMER autodetects
              whether the alignment is protein or DNA, but sometimes alignments are so small that  autodetection
              is ambiguous. See <b>--nucleic.</b>

       <b>--archpri</b> <u>&lt;x&gt;</u>
              Set  the  "architecture  prior"  used  by  MAP  architecture  construction  to <u>&lt;x&gt;,</u> where <u>&lt;x&gt;</u> is a
              probability between 0 and 1. This parameter governs a  geometric  prior  distribution  over  model
              lengths.  As  <u>&lt;x&gt;</u>  increases, longer models are favored a priori.  As <u>&lt;x&gt;</u> decreases, it takes more
              residue conservation in a column to make  a  column  a  "consensus"  match  column  in  the  model
              architecture.  The 0.85 default has been chosen empirically as a reasonable setting.

       <b>--binary</b>
              Write the HMM to <u>hmmfile</u> in HMMER binary format instead of readable ASCII text.

       <b>--cfile</b> <u>&lt;f&gt;</u>
              Save the observed emission and transition counts to <u>&lt;f&gt;</u> after the architecture has been determined
              (e.g. after residues/gaps have been assigned to match, delete, and insert states).  This option is
              used  in HMMER development for generating data files useful for training new Dirichlet priors. The
              format of count files is documented in the User's Guide.

       <b>--fast</b> Quickly and heuristically determine the architecture of the model by assigning  all  columns  will
              more  than a certain fraction of gap characters to insert states. By default this fraction is 0.5,
              and it can be changed using the <b>--gapmax</b> option.  The default construction algorithm is a  maximum
              a posteriori (MAP) algorithm, which is slower.

       <b>--gapmax</b> <u>&lt;x&gt;</u>
              Controls  the <u>--fast</u> model construction algorithm, but if <u>--fast</u> is not being used, has no effect.
              If a column has more than a fraction <u>&lt;x&gt;</u> of gap symbols in it,  it  gets  assigned  to  an  insert
              column.   <u>&lt;x&gt;</u>  is a frequency from 0 to 1, and by default is set to 0.5. Higher values of <u>&lt;x&gt;</u> mean
              more columns get assigned to consensus, and models get longer; smaller values of  <u>&lt;x&gt;</u>  mean  fewer
              columns get assigned to consensus, and models get smaller.  <u>&lt;x&gt;</u>

       <b>--hand</b> Specify  the  architecture  of the model by hand: the alignment file must be in SELEX or Stockholm
              format, and the reference annotation line (#=RF in SELEX, #=GC RF in Stockholm) is used to specify
              the architecture. Any column marked with a non-gap symbol  (such  as  an  'x',  for  instance)  is
              assigned as a consensus (match) column in the model.

       <b>--idlevel</b> <u>&lt;x&gt;</u>
              Controls  both  the  determination  of effective sequence number and the behavior of the <u>--wblosum</u>
              weighting option. The sequence alignment is clustered by  percent  identity,  and  the  number  of
              clusters  at a cutoff threshold of <u>&lt;x&gt;</u> is used to determine the effective sequence number.  Higher
              values of <u>&lt;x&gt;</u> give more clusters and higher effective sequence numbers; lower values of  <u>&lt;x&gt;</u>  give
              fewer  clusters  and  lower  effective  sequence  numbers.   <u>&lt;x&gt;</u> is a fraction from 0 to 1, and by
              default is set to 0.62 (corresponding to the clustering level used in  constructing  the  BLOSUM62
              substitution matrix).

       <b>--informat</b> <u>&lt;s&gt;</u>
              Assert  that  the  input  <u>seqfile</u>  is in format <u>&lt;s&gt;;</u> do not run Babelfish format autodection. This
              increases the reliability of the program  somewhat,  because  the  Babelfish  can  make  mistakes;
              particularly  recommended  for  unattended,  high-throughput  runs  of HMMER. Valid format strings
              include FASTA, GENBANK, EMBL, GCG, PIR, STOCKHOLM, SELEX, MSF, CLUSTAL, and PHYLIP. See the User's
              Guide for a complete list.

       <b>--noeff</b>
              Turn off the effective sequence number calculation, and use the true number of sequences  instead.
              This will usually reduce the sensitivity of the final model (so don't do it without good reason!)

       <b>--nucleic</b>
              Force  the alignment to be interpreted as nucleic acid sequence, either RNA or DNA. Normally HMMER
              autodetects whether the alignment is protein or DNA, but sometimes alignments are  so  small  that
              autodetection is ambiguous. See <b>--amino.</b>

       <b>--null</b> <u>&lt;f&gt;</u>
              Read a null model from <u>&lt;f&gt;.</u>  The default for protein is to use average amino acid frequencies from
              Swissprot 34 and p1 = 350/351; for nucleic acid, the default is to use 0.25 for each base and p1 =
              1000/1001.  For  documentation of the format of the null model file and further explanation of how
              the null model is used, see the User's Guide.

       <b>--pam</b> <u>&lt;f&gt;</u>
              Apply a heuristic PAM- (substitution matrix-) based prior on match emission probabilities  instead
              of the default mixture Dirichlet. The substitution matrix is read from <u>&lt;f&gt;.</u>  See <b>--pamwgt.</b>

              The  default  Dirichlet state transition prior and insert emission prior are unaffected. Therefore
              in principle you could combine <b>--prior</b> with <b>--pam</b> but this isn't recommended, as  it  hasn't  been
              tested. ( <b>--pam</b> itself hasn't been tested much!)

       <b>--pamwgt</b> <u>&lt;x&gt;</u>
              Controls  the weight on a PAM-based prior. Only has effect if <b>--pam</b> option is also in use.  <u>&lt;x&gt;</u> is
              a positive real number, 20.0 by default.  <u>&lt;x&gt;</u> is the number of "pseudocounts" contriubuted by  the
              heuristic prior. Very high values of <u>&lt;x&gt;</u> can force a scoring system that is entirely driven by the
              substitution matrix, making HMMER somewhat approximate Gribskov profiles.

       <b>--pbswitch</b> <u>&lt;n&gt;</u>
              For  alignments  with  a  very  large  number of sequences, the GSC, BLOSUM, and Voronoi weighting
              schemes are slow; they're O(N^2) for N sequences. Henikoff position-based weights (PB weights) are
              more efficient. At or above a certain threshold sequence number <u>&lt;n&gt;</u>  <b>hmm2build</b>  will  switch  from
              GSC,  BLOSUM, or Voronoi weights to PB weights. To disable this switching behavior (at the cost of
              compute time, set <u>&lt;n&gt;</u> to be something larger than the number of sequences in your alignment.   <u>&lt;n&gt;</u>
              is a positive integer; the default is 1000.

       <b>--prior</b> <u>&lt;f&gt;</u>
              Read  a  Dirichlet  prior  from <u>&lt;f&gt;,</u> replacing the default mixture Dirichlet.  The format of prior
              files is documented in the User's Guide, and an example is given in the  Demos  directory  of  the
              HMMER distribution.

       <b>--swentry</b> <u>&lt;x&gt;</u>
              Controls  the  total  probability  that  is  distributed  to  local entries into the model, versus
              starting at the beginning of the model as in a global alignment.  <u>&lt;x&gt;</u> is a probability from  0  to
              1,  and by default is set to 0.5.  Higher values of <u>&lt;x&gt;</u> mean that hits that are fragments on their
              left (N or 5'-terminal) side will be penalized  less,  but  complete  global  alignments  will  be
              penalized  more.   Lower values of <u>&lt;x&gt;</u> mean that fragments on the left will be penalized more, and
              global alignments on this side will be favored.  This option only affects the configurations  that
              allow  local  alignments,  e.g.   <b>-s</b>  and  <b>-f;</b> unless one of these options is also activated, this
              option has no effect.  You have independent control over local/global alignment behavior  for  the
              N/C (5'/3') termini of your target sequences using <b>--swentry</b> and <b>--swexit.</b>

       <b>--swexit</b> <u>&lt;x&gt;</u>
              Controls the total probability that is distributed to local exits from the model, versus ending an
              alignment at the end of the model as in a global alignment.  <u>&lt;x&gt;</u> is a probability from 0 to 1, and
              by  default  is set to 0.5.  Higher values of <u>&lt;x&gt;</u> mean that hits that are fragments on their right
              (C or 3'-terminal) side will be penalized less, but complete global alignments will  be  penalized
              more.   Lower  values  of  <u>&lt;x&gt;</u> mean that fragments on the right will be penalized more, and global
              alignments on this side will be favored.  This option only affects the configurations  that  allow
              local  alignments, e.g.  <b>-s</b> and <b>-f;</b> unless one of these options is also activated, this option has
              no effect.  You have independent control over local/global alignment behavior for the N/C  (5'/3')
              termini of your target sequences using <b>--swentry</b> and <b>--swexit.</b>

       <b>--verbose</b>
              Print  more  possibly  useful  stuff,  such  as  the  individual  scores  for each sequence in the
              alignment.

       <b>--wblosum</b>
              Use the BLOSUM filtering algorithm to weight the sequences, instead of the default.   Cluster  the
              sequences  at  a  given percentage identity (see <b>--idlevel);</b> assign each cluster a total weight of
              1.0, distributed equally amongst the members of that cluster.

       <b>--wgsc</b> Use the Gerstein/Sonnhammer/Chothia ad hoc sequence  weighting  algorithm.  This  is  already  the
              default,  so  this  option  has no effect (unless it follows another option in the -\-w family, in
              which case it overrides it).

       <b>--wme</b>  Use the Krogh/Mitchison maximum entropy algorithm to "weight" the sequences. This  supersedes  the
              Eddy/Mitchison/Durbin  maximum  discrimination algorithm, which gives almost identical weights but
              is less robust. ME weighting seems to give a marginal increase in sensitivity over the default GSC
              weights, but takes a fair amount of time.

       <b>--wnone</b>
              Turn off all sequence weighting.

       <b>--wpb</b>  Use the Henikoff position-based weighting scheme.

       <b>--wvoronoi</b>
              Use the Sibbald/Argos Voronoi sequence weighting algorithm in place of the default GSC weighting.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Master man page, with full list of and guide to the individual man pages: see <b><a href="../man1/hmmer2.1.html">hmmer2</a>(1).</b>

       For          complete           documentation,           see           the           user           guide
       (<a href="ftp://selab.janelia.org/pub/software/hmmer/2.3.2/Userguide.pdf">ftp://selab.janelia.org/pub/software/hmmer/2.3.2/Userguide.pdf</a>);    or   see   the   HMMER   web   page,
       <a href="http://hmmer.janelia.org/">http://hmmer.janelia.org/</a>.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright (C) 1992-2003 HHMI/Washington University School of Medicine.
       Freely distributed under the GNU General Public License (GPL).
       See the file COPYING in your distribution for details on redistribution conditions.

</pre><h4><b>AUTHOR</b></h4><pre>
       Sean Eddy
       HHMI/Dept. of Genetics
       Washington Univ. School of Medicine
       4566 Scott Ave.
       St Louis, MO 63110 USA
       <a href="http://www.genetics.wustl.edu/eddy/">http://www.genetics.wustl.edu/eddy/</a>

HMMER 2.3.2                                         Oct 2003                                        <u><a href="../man1/hmm2build.1.html">hmm2build</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>