<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>indelHistory - Obtain an "indel history" for an alignment and write it to a file</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/phast">phast_1.7+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       indelHistory - Obtain an "indel history" for an alignment and write it to a file

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Obtain  an "indel history" for an alignment and write it to a file describing insertions and deletions in
       all species using a compact format.  The specified alignment may either contain  sequences  corresponding
       only  to  leaves  of  the  specified tree, in which case an indel history will be inferred using a simple
       parsimony algorithm, or it may contain sequences for all nodes in the  tree,  in  which  case  the  indel
       history will simply be extracted from the file.

</pre><h4><b>OPTIONS</b></h4><pre>

       <b>--msa-format</b>, <b>-i</b> FASTA|MAF|SS|....

              Read alignment in specified file format (default FASTA).

       <b>--output-alignment</b>, <b>-A</b>

              Instead  of  a  summary  of  indels only, output an alignment in FASTA format of sequences for all
              ancestral and leaf nodes.

       <b>--read-history</b>, <b>-H</b> &lt;fname&gt; Read an indel history directly from the specified file.  Useful for debugging.
              The alignment and tree arguments are not required;

              however in an alignment is given with <b>--read-history</b> and <b>--output-alignment</b>, then actual bases can
              be output for leaf species.

       <b>--ia-names</b>, <b>-I</b>

              Assume ancestral sequences in alignment.fa are named according to the convention used  by  Mathieu
              Blanchette's  inferAncestors  program,  e.g.,  "RAT+MOUSE+RABBIT+" for the last common ancestor of
              "rat", "mouse", and "rabbit".

       <b>--help</b>, <b>-h</b>

              Display this help message.

indelHistory 1.4                                    May 2016                                     <u><a href="../man1/INDELHISTORY.1.html">INDELHISTORY</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>