<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>malt-run - Sequences aligner using MALT (MEGAN alignment tool)</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/malt">malt_0.5.2-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       malt-run - Sequences aligner using MALT (MEGAN alignment tool)

</pre><h4><b>DESCRIPTION</b></h4><pre>
       SYNOPSIS

              malt-run [options]

       DESCRIPTION

              Aligns sequences using MALT (MEGAN alignment tool)

       OPTIONS

              Mode:

       <b>-m</b>, <b>--mode</b> [string]
              Program mode. Mandatory option. Legal values: Unknown, BlastN, BlastP, BlastX, Classifier

       <b>-at</b>, <b>--alignmentType</b> [string]
              Type of alignment to be performed. Default value: Local Legal values: Local, SemiGlobal

              Input:

       <b>-i</b>, <b>--inFile</b> [string(s)]
              Input file(s) containing queries in FastA or FastQ format (gzip or zip ok). Mandatory option.

       <b>-d</b>, <b>--index</b> [string]
              Index directory as generated by malt-build. Mandatory option.

              Output:

       <b>-o</b>, <b>--output</b> [string(s)]
              Output RMA file(s) or directory or stdout.

       <b>-iu</b>, <b>--includeUnaligned</b>
              Include unaligned queries in RMA output file. Default value: false.

       <b>-a</b>, <b>--alignments</b> [string(s)]
              Output alignment file(s) or directory or stdout.

       <b>-f</b>, <b>--format</b> [string]
              Alignment output format. Default value: SAM Legal values: SAM, Tab, Text

       <b>-za</b>, <b>--gzipAlignments</b>
              Compress alignments using gzip. Default value: true.

       <b>-ssc</b>, <b>--samSoftClip</b>
              Use soft clipping in SAM files (BlastN mode only). Default value: false.

       <b>-sps</b>, <b>--sparseSAM</b>
              Produce sparse SAM format (smaller, faster, but only suitable for MEGAN). Default value: false.

       <b>-oa</b>, <b>--outAligned</b> [string(s)]
              Aligned reads output file(s) or directory or stdout.

       <b>-zal</b>, <b>--gzipAligned</b>
              Compress aligned reads output using gzip. Default value: true.

       <b>-ou</b>, <b>--outUnaligned</b> [string(s)]
              Unaligned reads output file(s) or directory or stdout.

       <b>-zul</b>, <b>--gzipUnaligned</b>
              Compress unaligned reads output using gzip. Default value: true.

              Performance:

       <b>-t</b>, <b>--numThreads</b> [number]
              Number of worker threads. Default value: 4.

       <b>-mem</b>, <b>--memoryMode</b> [string]
              Memory mode. Default value: load Legal values: load, page, map

       <b>-mt</b>, <b>--maxTables</b> [number]
              Set the maximum number of seed tables to use (0=all). Default value: 0.

       <b>-rqc</b>, <b>--replicateQueryCache</b>
              Cache results for replicated queries. Default value: false.

              Filter:

       <b>-b</b>, <b>--minBitScore</b> [number]
              Minimum bit score. Default value: 50.0.

       <b>-e</b>, <b>--maxExpected</b> [number]
              Maximum expected score. Default value: 1.0.

       <b>-id</b>, <b>--minPercentIdentity</b> [number]
              Minimum percent identity. Default value: 0.0.

       <b>-mq</b>, <b>--maxAlignmentsPerQuery</b> [number]
              Maximum number of alignments per query. Default value: 25.

       <b>-mrf</b>, <b>--maxAlignmentsPerRef</b> [number]
              Maximum number of (non-overlapping) alignments per reference. Default value: 1.

              BlastN parameters:

       <b>-ma</b>, <b>--matchScore</b> [number]
              Match score. Default value: 2.

       <b>-mm</b>, <b>--mismatchScore</b> [number]
              Mismatch score. Default value: <b>-3</b>.

       <b>-la</b>, <b>--setLambda</b> [number]
              Parameter Lambda for BLASTN statistics. Default value: 0.625.

       <b>-K</b>, <b>--setK</b> [number]
              Parameter K for BLASTN statistics. Default value: 0.41.

              BlastP and BlastX parameters:

       <b>-psm</b>, <b>--subMatrix</b> [string]
              Protein  substitution  matrix  to  use.  Default value: BLOSUM62 Legal values: BLOSUM45, BLOSUM50,
              BLOSUM62, BLOSUM80, BLOSUM90

              DNA query parameters:

       <b>-fo</b>, <b>--forwardOnly</b>
              Align query forward strand only. Default value: false.

       <b>-ro</b>, <b>--reverseOnly</b>
              Align query reverse strand only. Default value: false.

              LCA parameters:

       <b>-top</b>, <b>--topPercent</b> [number]
              Top percent value for LCA algorithm. Default value: 10.0.

       <b>-supp</b>, <b>--minSupportPercent</b> [number]
              Min support value for LCA algorithm as a percent of assigned reads (0==off). Default value: 0.001.

       <b>-sup</b>, <b>--minSupport</b> [number]
              Min support value for LCA algorithm (overrides <b>--minSupportPercent</b>). Default value: 0.

       <b>-mpi</b>, <b>--minPercentIdentityLCA</b> [number]
              Min percent identity used by LCA algorithm. Default value: 0.0.

       <b>-mif</b>, <b>--useMinPercentIdentityFilterLCA</b>
              Use percent identity assignment filter. Default value: false.

       <b>-wlca</b>, <b>--weightedLCA</b>
              Use the weighted LCA for taxonomic assignment. Default value: false.

       <b>-lcp</b>, <b>--lcaCoveragePercent</b> [number]
              Set the percent for the LCA to cover. Default value: 100.0.

       <b>-mag</b>, <b>--magnitudes</b>
              Reads have magnitudes (to be used in taxonomic or functional analysis). Default value: false.

       <b>-cf</b>, <b>--conFile</b> [string]
              File of contaminant taxa (one Id or name per line).

              Heuristics:

       <b>-spf</b>, <b>--maxSeedsPerFrame</b> [number]
              Maximum number of seed matches per offset per read frame. Default value: 100.

       <b>-spr</b>, <b>--maxSeedsPerRef</b> [number]
              Maximum number of seed matches per read and reference. Default value: 20.

       <b>-sh</b>, <b>--seedShift</b> [number]
              Seed shift. Default value: 1.

              Banded alignment parameters:

       <b>-go</b>, <b>--gapOpen</b> [number]
              Gap open penalty. Default value: 11.

       <b>-ge</b>, <b>--gapExtend</b> [number]
              Gap extension penalty. Default value: 1.

       <b>-bd</b>, <b>--band</b> [number]
              Band width/2 for banded alignment. Default value: 4.

              Other:

       <b>-rqcb</b>, <b>--replicateQueryCacheBits</b> [number]
              Bits used for caching replicate queries (size is then 2^bits). Default value: 20.

       <b>-xP</b>, <b>--xPart</b>
              Show part of the table in human readable form for debugging. Default value: false.

       <b>-v</b>, <b>--verbose</b>
              Echo commandline options and be verbose. Default value: false.

       <b>-h</b>, <b>--help</b>
              Show program usage and quit.

</pre><h4><b>AUTHOR(s)</b></h4><pre>
              Daniel H. Huson.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/malt-build.1.html">malt-build</a></b>(1)

malt-run                                          January 2022                                       <u><a href="../man1/MALT-RUN.1.html">MALT-RUN</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>