<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Maqview - Maq alignment indexer and viewer</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/maqview">maqview_0.2.5-12_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Maqview - Maq alignment indexer and viewer

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>maqindex</b> <b>-i</b>|<b>-v</b>|<b>-b</b> [<b>-c</b> <u>in.cns</u>] <u>in.map</u> [<u>chr</u>[:<u>start</u>[-<u>end</u>]] [...]]

       <b>maqview</b> [<b>-c</b> <u>in.cns</u>] <u>in.map</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>Maqview</b> is a set of programs that achieves fast random access to the alignment file generated by `<b>maq</b>
       map', and displays the alignment with a nice GUI (Graphical User Interface). It is highly efficient in
       speed, memory and disk usage. <b>Maqview</b> is based on OpenGL and is known to work on both Mac OS X and Linux.
       Porting to Windows is in principle easy.

   <b>maqindex</b>
       <b>maqindex</b> <b>-i</b>|<b>-v</b>|<b>-b</b> [<b>-c</b> <u>in.cns</u>] <u>in.map</u> [<u>chr</u>[:<u>start</u>[-<u>end</u>]] [...]]

       <b>-i</b>       Index the alignment file

       <b>-v</b>       Print the alignment in the specified region in the `maq mapview' format

       <b>-b</b>       Dump the alignment in the specified region in the binary .map format

       <b>-c</b> <u>FILE</u>  Index the Maq consensus file for viewing in maqview at the same time

       Program  <b>maqindex</b>  indexes  a  maq  alignment  file  <u>in.map</u>  or quickly retrieves all the reads in one or
       multiple regions. One of <b>-i</b>, <b>-v</b> and <b>-b</b> must be used. For <b>-v</b> or <b>-b</b>,  at  least  one  region  like  `chrX',
       `chrX:1000' or `chrX:1,000-2,000' must be specified. Multiple regions are allowed.

   <b>maqview</b>
       <b>maqview</b> [<b>-c</b> <u>in.cns</u>] <u>in.map</u>

       <b>-c</b> <u>FILE</u>  The Maq consensus file

       Program  <b>maqview</b>  displays  the  read  alignment in a graphical window. When <u>in.cns</u> is specified, the top
       sequence is the reference, followed by the Maq consensus sequence; otherwise, the majority-rule consensus
       will be calculated.

       Maqview has two views: sequence view and box view. In the sequence view, read sequences will  be  printed
       on the screen. Darker bases indicate lower base qualities and red ones show the differences in comparison
       to  the  reference  if  <u>in.cns</u>  is  specified  or to the majority-rule consensus if not. In the box view,
       different types of nucleotides are represented as colour boxes with green for A, cyan for C,  orange  for
       G,  red  for  T  and  dark  gray  for  N.  The saturation of colours indicates the base qualities and the
       thickness lines of reads shows the mapping qualities of read alignments. Zooming in/out is supported only
       in the box view. In both views, the status bar at the bottom of the window  will  show  some  information
       about the key touches, and read names and base qualities pointed by the mouse.

       Navigating the alignment is accomplished by vaious key bindings.

       • Key bindings:

          ?                Display help
          &lt;Esc&gt;            Exit
          &lt;F1&gt;             Sequence view
          &lt;F2&gt;             Box view
          p                Switch to the previous reference sequence
          n                Switch to the next reference sequence
          r                Refresh

          &lt;digits&gt; &lt;Enter&gt; Move to the position pointed by &lt;digits&gt;
          h / &lt;Left&gt;       Move left by one base
          l / &lt;Right&gt;      Move right by one base
          k / &lt;Up&gt;         Move up by one line
          j / &lt;Down&gt;       Move down by one line
          &gt;                Move to the next read
          &lt;                Move to the previous read
          g / &lt;Home&gt;       Move to the beginning of the current reference
          G / &lt;End&gt;        Move to the end of the current reference
          u / &lt;PageUp&gt;     Move left by one page
          &lt;Space&gt;/&lt;PageDn&gt; Move right by one page
          &lt;Shift&gt; &lt;Arrow&gt;  Move by 100 bases
          &lt;Ctrl&gt; &lt;Arrow&gt;   Move by 1000 bases

          &lt;Ctrl&gt; +         Add a new view
          &lt;Ctrl&gt; -         Remove the current view

          &lt;Enter&gt;          Toggle on/off right auto-scrolling
          ^b               Toggle on/off left auto-scrolling
          +                Zoom in
          -                Zoom out
          0                Zoom to the default scale
          q                Toggle show/hide mapping qualities
          e                Toggle display SE/PE mapping qualities
          o                Open a new alignment file

       • Enter an integer and then press &lt;Enter&gt; to jump to the required position.

       • Enter an integer and then press a key other than &lt;Enter&gt; to speed up scrolling by the specified factor.

       • Left  click  and  drag  the  yellow  cross  in  the  square to move along the sequence. Right click the
         reference to highlight a column.

</pre><h4><b>LICENSE</b></h4><pre>
       GNU General Public License (GPL)

</pre><h4><b>AVAILABILITY</b></h4><pre>
       &lt;<a href="http://maq.sourceforge.net">http://maq.sourceforge.net</a>&gt;

</pre><h4><b>AUTHORS</b></h4><pre>
       Jue Ruan &lt;<a href="mailto:ruanjue@genomics.org.cn">ruanjue@genomics.org.cn</a>&gt; for writing and maintaining the whole software.

       Heng Li &lt;<a href="mailto:lh3@sanger.ac.uk">lh3@sanger.ac.uk</a>&gt; for testing the program and drafting the documentations.

       Mengyao Zhao &lt;<a href="mailto:zhaomengyao@gmail.com">zhaomengyao@gmail.com</a>&gt; for designing the color schemes.

maqview-0.2.4                                      2024-08-06                                         <u><a href="../man1/MAQVIEW.1.html">MAQVIEW</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>