<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>minimap2 - mapping and alignment between collections of DNA sequences</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/minimap2">minimap2_2.27+dfsg-1build2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       minimap2 - mapping and alignment between collections of DNA sequences

</pre><h4><b>SYNOPSIS</b></h4><pre>
       * Indexing the target sequences (optional):
           minimap2 [<b>-x</b> <u>preset</u>] <b>-d</b> <u>target.mmi</u> <u>target.fa</u>
           minimap2 [<b>-H</b>] [<b>-k</b> <u>kmer</u>] [<b>-w</b> <u>miniWinSize</u>] [<b>-I</b> <u>batchSize</u>] <b>-d</b> <u>target.mmi</u> <u>target.fa</u>

       * Long-read alignment with CIGAR:
           minimap2 <b>-a</b> [<b>-x</b> <u>preset</u>] <u>target.mmi</u> <u>query.fa</u> &gt; <u>output.sam</u>
           minimap2 <b>-c</b> [<b>-H</b>] [<b>-k</b> <u>kmer</u>] [<b>-w</b> <u>miniWinSize</u>] [<b>...</b>]  <u>target.fa</u> <u>query.fa</u> &gt; <u>output.paf</u>

       * Long-read overlap without CIGAR:
           minimap2 <b>-x</b> ava-ont [<b>-t</b> <u>nThreads</u>] <u>target.fa</u> <u>query.fa</u> &gt; <u>output.paf</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Minimap2  is  a  fast  sequence  mapping  and alignment program that can find overlaps between long noisy
       reads, or map long reads or their assemblies to a reference genome  optionally  with  detailed  alignment
       (i.e.  CIGAR).  At  present,  it  works  efficiently  with  query  sequences from a few kilobases to ~100
       megabases in length at a error rate ~15%. Minimap2 outputs in the PAF or the SAM format.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Indexing</b> <b>options</b>
       <b>-k</b> <u>INT</u>    Minimizer k-mer length [15]

       <b>-w</b> <u>INT</u>    Minimizer window size [10]. A minimizer is the smallest k-mer in a window of w  consecutive  k-
                 mers.

       <b>-j</b> <u>INT</u>    Syncmer  submer  size  [10]. Option <b>-j</b> and <b>-w</b> will override each: if <b>-w</b> is applied after <b>-j</b>, <b>-j</b>
                 will have no effect, and vice versa.

       <b>-H</b>        Use homopolymer-compressed (HPC) minimizers. An HPC  sequence  is  constructed  by  contracting
                 homopolymer runs to a single base. An HPC minimizer is a minimizer on the HPC sequence.

       <b>-I</b> <u>NUM</u>    Load  at  most <u>NUM</u> target bases into RAM for indexing [8G]. If there are more than <u>NUM</u> bases in
                 <u>target.fa</u>, minimap2 needs to read <u>query.fa</u> multiple times to  map  it  against  each  batch  of
                 target  sequences.  This  would create a multi-part index.  <u>NUM</u> may be ending with k/K/m/M/g/G.
                 NB: mapping quality is incorrect given a multi-part index. See also option <b>--split-prefix</b>.

       <b>--idx-no-seq</b>
                 Don't store target sequences in the index. It  saves  disk  space  and  memory  but  the  index
                 generated  with  this  option  will  not  work with <b>-a</b> or <b>-c</b>.  When base-level alignment is not
                 requested, this option is automatically applied.

       <b>-d</b> <u>FILE</u>   Save the minimizer index of <u>target.fa</u> to <u>FILE</u> [no dump]. Minimap2  indexing  is  fast.  It  can
                 index  the  human  genome  in a couple of minutes. If even shorter startup time is desired, use
                 this option to save the index. Indexing options are fixed in the index file. When an index file
                 is provided as the target sequences, options <b>-H</b>, <b>-k</b>, <b>-w</b>, <b>-I</b> will be effectively  overridden  by
                 the options stored in the index file.

       <b>--alt</b> <u>FILE</u>
                 List of ALT contigs [null]

       <b>--alt-drop</b> <u>FLOAT</u>
                 Drop ALT hits by <u>FLOAT</u> fraction when ranking and computing mapping quality [0.15]

   <b>Mapping</b> <b>options</b>
       <b>-f</b> <u>FLOAT</u><b>|</b><u>INT1</u><b>[,</b><u>INT2</u><b>]</b>
                 If fraction, ignore top <u>FLOAT</u> fraction of most frequent minimizers [0.0002]. If integer, ignore
                 minimizers  occuring  more  than  <u>INT1</u> times.  <u>INT2</u> is only effective in the <b>--sr</b> or <b>-xsr</b> mode,
                 which sets the threshold for a second round of seeding.

       <b>-U</b> <u>INT1</u><b>[,</b><u>INT2</u><b>]</b>
                 Lower and upper bounds of k-mer occurrences [10,1000000]. The final k-mer occurrence  threshold
                 is max{<u>INT1</u>, min{<u>INT2</u>, <b>-f</b>}}.  This option prevents excessively small or large <b>-f</b> estimated from
                 the  input reference. Available since r1034 and deprecating <b>--min-occ-floor</b> in earlier versions
                 of minimap2.

       <b>--q-occ-frac</b> <u>FLOAT</u>
                 Discard a query minimizer if its occurrence is higher than <u>FLOAT</u> fraction of  query  minimizers
                 and than the reference occurrence threshold [0.01]. Set 0 to disable. Available since r1105.

       <b>-e</b> <u>INT</u>    Sample a high-frequency minimizer every <u>INT</u> basepairs [500].

       <b>-g</b> <u>NUM</u>    Stop chain enlongation if there are no minimizers within <u>NUM</u>-bp [10k].

       <b>-r</b> <u>NUM1</u><b>[,</b><u>NUM2</u><b>]</b>
                 Bandwidth  for  chaining  and  base alignment [500,20k].  <u>NUM1</u> is used for initial chaining and
                 alignment extension; <u>NUM2</u> for RMQ-based re-chaining and closing gaps in alignments.

       <b>-n</b> <u>INT</u>    Discard chains consisting of &lt;<u>INT</u> number of minimizers [3]

       <b>-m</b> <u>INT</u>    Discard chains with chaining score &lt;<u>INT</u> [40]. Chaining score equals the approximate  number  of
                 matching bases minus a concave gap penalty. It is computed with dynamic programming.

       <b>-D</b>        If query sequence name/length are identical to the target name/length, ignore diagonal anchors.
                 This option also reduces DP-based extension along the diagonal.

       <b>-P</b>        Retain  all  chains  and  don't attempt to set primary chains. Options <b>-p</b> and <b>-N</b> have no effect
                 when this option is in use.

       <b>--dual</b>=<b>yes</b>|<b>no</b>
                 If <b>no</b>, skip query-target pairs wherein the query name is  lexicographically  greater  than  the
                 target name [yes]

       <b>-X</b>        Equivalent to '<b>-DP</b> <b>--dual</b>=<b>no</b> <b>--no-long-join</b>'.  Primarily used for all-vs-all read overlapping.

       <b>-p</b> <u>FLOAT</u>  Minimal  secondary-to-primary  score  ratio  to  output  secondary mappings [0.8].  Between two
                 chains overlaping over half of the shorter chain (controlled by <b>-M</b>), the  chain  with  a  lower
                 score  is  secondary  to  the  chain  with a higher score.  If the ratio of the scores is below
                 <u>FLOAT</u>, the secondary chain will not be outputted or extended with  DP  alignment  later.   This
                 option has no effect when <b>-X</b> is applied.

       <b>-N</b> <u>INT</u>    Output at most <u>INT</u> secondary alignments [5]. This option has no effect when <b>-X</b> is applied.

       <b>-G</b> <u>NUM</u>    Maximum  gap on the reference (effective with <b>-xsplice</b>/<b>--splice</b>).  This option also changes the
                 chaining and alignment band width to <u>NUM</u>.  Increasing this option slows down spliced alignment.
                 [200k]

       <b>-F</b> <u>NUM</u>    Maximum fragment length (aka insert size; effective with <b>-xsr</b>/<b>--frag</b>=<b>yes</b>) [800]

       <b>-M</b> <u>FLOAT</u>  Mark as secondary a chain that overlaps with a better chain by <u>FLOAT</u> or  more  of  the  shorter
                 chain [0.5]

       <b>--rmq</b>=<b>no</b>|<b>yes</b>
                 Use  the  minigraph  chaining  algorithm  [no].  The minigraph algorithm is better for aligning
                 contigs through long INDELs.

       <b>--hard-mask-level</b>
                 Honor option <b>-M</b> and disable a heurstic to save unmapped subsequences and disables <b>--mask-len</b>.

       <b>--mask-len</b> <u>NUM</u>
                 Keep an alignment if dropping it leaves an unaligned region on query  longer  than  <u>INT</u>  [inf].
                 Effective without <b>--hard-mask-level</b>.

       <b>--max-chain-skip</b> <u>INT</u>
                 A  heuristics  that  stops chaining early [25]. Minimap2 uses dynamic programming for chaining.
                 The time complexity is quadratic in the number of seeds. This option makes minimap2  exits  the
                 inner  loop  if it repeatedly sees seeds already on chains. Set <u>INT</u> to a large number to switch
                 off this heurstics.

       <b>--max-chain-iter</b> <u>INT</u>
                 Check up to <u>INT</u> partial chains during chaining [5000]. This is a heuristic to  avoid  quadratic
                 time complexity in the worst case.

       <b>--chain-gap-scale</b> <u>FLOAT</u>
                 Scale of gap cost during chaining [1.0]

       <b>--no-long-join</b>
                 Disable the long gap patching heuristic. When this option is applied, the maximum alignment gap
                 is mostly controlled by <b>-r</b>.

       <b>--splice</b>  Enable the splice alignment mode.

       <b>--sr</b>      Enable short-read alignment heuristics. In the short-read mode, minimap2 applies a second round
                 of  chaining  with  a  higher  minimizer  occurrence  threshold  if  no good chain is found. In
                 addition, minimap2 attempts to patch gaps between seeds with ungapped alignment.

       <b>--split-prefix</b> <u>STR</u>
                 Prefix to create temporary files. Typically used for a multi-part index.

       <b>--frag</b>=<b>no</b>|<b>yes</b>
                 Whether to enable the fragment mode [no]

       <b>--for-only</b>
                 Only map to the forward strand of the reference sequences. For paired-end reads in the forward-
                 reverse orientation, the first read is mapped to forward strand of the reference and the second
                 read to the reverse stand.

       <b>--rev-only</b>
                 Only map to the reverse complement strand of the reference sequences.

       <b>--heap-sort</b>=<b>no</b>|<b>yes</b>
                 If yes, sort anchors with heap merge, instead of radix sort. Heap merge  is  faster  for  short
                 reads, but slower for long reads. [no]

       <b>--no-pairing</b>
                 Treat  two  reads  in  a  pair  as  independent reads. The mate related fields in SAM are still
                 properly populated.

       <b>--no-hash-name</b>
                 Produce the same alignment for identical sequences regardless of their sequence names.

   <b>Alignment</b> <b>options</b>
       <b>-A</b> <u>INT</u>    Matching score [2]

       <b>-B</b> <u>INT</u>    Mismatching penalty [4]

       <b>-b</b> <u>INT</u>    Mismatching penalty for transitions [same as <b>-B</b>].

       <b>-O</b> <u>INT1[,INT2]</u>
                 Gap open penalty [4,24]. If <u>INT2</u> is not specified, it is set to <u>INT1</u>.

       <b>-E</b> <u>INT1[,INT2]</u>
                 Gap extension penalty [2,1]. A gap of length <u>k</u> costs min{<u>O1</u>+<u>k</u>*<u>E1</u>,<u>O2</u>+<u>k</u>*<u>E2</u>}.  In the splice mode,
                 the second gap penalties are not used.

       <b>-J</b> <u>INT</u>    Splice model [1]. 0 for the original minimap2 splice  model  that  always  penalizes  non-GT-AG
                 splicing;  1  for the miniprot model that considers non-GT-AG. Option <b>-C</b> has no effect with the
                 default <b>-J1</b>.  <b>-J0</b>.

       <b>-C</b> <u>INT</u>    Cost for a non-canonical GT-AG splicing (effective with <b>--splice</b> <b>-J0</b>) [0].

       <b>-z</b> <u>INT1[,INT2]</u>
                 Truncate an alignment if the running alignment score drops too quickly along  the  diagonal  of
                 the  DP  matrix  (diagonal  X-drop,  or  Z-drop) [400,200]. If the drop of score is above <u>INT2</u>,
                 minimap2 will reverse complement the query in the related region and align again to test  small
                 inversions.  Minimap2  truncates  alignment  if  there  is an inversion or the drop of score is
                 greater than <u>INT1</u>.  Decrease <u>INT2</u> to find small inversions at the cost of performance and false
                 positives.  Increase <u>INT1</u> to improves the contiguity of alignment at the cost of poor alignment
                 in the middle.

       <b>-s</b> <u>INT</u>    Minimal peak DP alignment score to output [40]. The peak  score  is  computed  from  the  final
                 CIGAR.  It  is  the score of the max scoring segment in the alignment and may be different from
                 the total alignment score.

       <b>-u</b> <u>CHAR</u>   How to find canonical splicing sites GT-AG - <b>f</b>: transcript  strand;  <b>b</b>:  both  strands;  <b>n</b>:  no
                 attempt to match GT-AG [n]

       <b>--end-bonus</b> <u>INT</u>
                 Score bonus when alignment extends to the end of the query sequence [0].

       <b>--score-N</b> <u>INT</u>
                 Score of a mismatch involving ambiguous bases [1].

       <b>--splice-flank</b>=<b>yes</b>|<b>no</b>
                 Assume the next base to a <b>GT</b> donor site tends to be A/G (91% in human and 92% in mouse) and the
                 preceding  base  to  a  <b>AG</b>  acceptor  tends  to  be  C/T  [no].   This  trend is evolutionarily
                 conservative, all the way to S. cerevisiae (PMID:18688272). Specifying  this  option  generally
                 leads  to  higher  junction  accuracy  by  several  percents,  so it is applied by default with
                 <b>--splice</b>.  However, the SIRV control does not honor this trend (only ~60%). This option reduces
                 accuracy. If you are benchmarking minimap2 on SIRV data, please add  <b>--splice-flank=no</b>  to  the
                 command line.

       <b>--junc-bed</b> FILE
                 Gene  annotations in the BED12 format (aka 12-column BED), or intron positions in 5-column BED.
                 With this option, minimap2 prefers splicing in annotations.  BED12 file can be  converted  from
                 GTF/GFF3 with `paftools.js gff2bed anno.gtf' [].

       <b>--junc-bonus</b> INT
                 Score bonus for a splice donor or acceptor found in annotation (effective with <b>--junc-bed</b>) [9].

       <b>--end-seed-pen</b> <u>INT</u>
                 Drop  a terminal anchor if <u>s</u>&lt;log(<u>g</u>)+<u>INT</u>, where <u>s</u> is the local alignment score around the anchor
                 and <u>g</u> the length of the terminal gap in the chain. This option is only effective with <b>--splice</b>.
                 It helps to avoid tiny terminal exons. [6]

       <b>--no-end-flt</b>
                 Don't filter seeds towards the ends of chains before performing base-level alignment.

       <b>--cap-sw-mem</b> <u>NUM</u>
                 Skip alignment if the DP matrix size is above <u>NUM</u>.  Set 0 to disable [100m].

       <b>--cap-kalloc</b> <u>NUM</u>
                 Free thread-local kalloc memory reservoir if after the alignment  the  size  of  the  reservoir
                 above <u>NUM</u>.  Set 0 to disable [0].

   <b>Input/output</b> <b>options</b>
       <b>-a</b>        Generate CIGAR and output alignments in the SAM format. Minimap2 outputs in PAF by default.

       <b>-o</b> <u>FILE</u>   Output alignments to <u>FILE</u> [stdout].

       <b>-Q</b>        Ignore base quality in the input file.

       <b>-L</b>        Write  CIGAR  with &gt;65535 operators at the CG tag. Older tools are unable to convert alignments
                 with &gt;65535 CIGAR ops to BAM. This option makes minimap2 SAM compatible with older tools. Newer
                 tools recognizes this tag and reconstruct the real CIGAR in memory.

       <b>-R</b> <u>STR</u>    SAM read group line in a format like <b>@RG\tID:foo\tSM:bar</b> [].

       <b>-y</b>        Copy input FASTA/Q comments to output.

       <b>-c</b>        Generate CIGAR. In PAF, the CIGAR is written to the `cg' custom tag.

       <b>--cs[=</b><u>STR</u><b>]</b>
                 Output the <b>cs</b> tag.  <u>STR</u> can be either <u>short</u> or <u>long</u>.  If no <u>STR</u> is  given,  <u>short</u>  is  assumed.
                 [none]

       <b>--MD</b>      Output the MD tag (see the SAM spec).

       <b>--eqx</b>     Output =/X CIGAR operators for sequence match/mismatch.

       <b>-Y</b>        In SAM output, use soft clipping for supplementary alignments.

       <b>--secondary-seq</b>
                 In SAM output, show query sequences for secondary alignments.

       <b>--seed</b> <u>INT</u>
                 Integer seed for randomizing equally best hits. Minimap2 hashes <u>INT</u> and read name when choosing
                 between equally best hits. [11]

       <b>-t</b> <u>INT</u>    Number  of threads [3]. Minimap2 uses at most three threads when indexing target sequences, and
                 uses up to <u>INT</u>+1 threads when mapping (the extra thread is for I/O, which  is  frequently  idle
                 and takes little CPU time).

       <b>-2</b>        Use two I/O threads during mapping. By default, minimap2 uses one I/O thread.  When I/O is slow
                 (e.g.  piping  to gzip, or reading from a slow pipe), the I/O thread may become the bottleneck.
                 Apply this option to use one thread for input and another thread for output,  at  the  cost  of
                 increased peak RAM.

       <b>-K</b> <u>NUM</u>    Number  of  bases  loaded into memory to process in a mini-batch [500M].  Similar to option <b>-I</b>,
                 K/M/G/k/m/g suffix is accepted. A large <u>NUM</u> helps load balancing in the  multi-threading  mode,
                 at the cost of increased memory.

       <b>--secondary</b>=<b>yes</b>|<b>no</b>
                 Whether to output secondary alignments [yes]

       <b>--max-qlen</b> <u>NUM</u>
                 Filter out query sequences longer than <u>NUM</u>.

       <b>--paf-no-hit</b>
                 In  PAF,  output  unmapped  queries;  the  strand and the reference name fields are set to `*'.
                 Warning: some paftools.js commands may not work with such output for the moment.

       <b>--sam-hit-only</b>
                 In SAM, don't output unmapped reads.

       <b>--version</b> Print version number to stdout

   <b>Preset</b> <b>options</b>
       <b>-x</b> <u>STR</u>    Preset []. This option applies multiple options at the same time. It should be  applied  before
                 other options because options applied later will overwrite the values set by <b>-x</b>.  Available <u>STR</u>
                 are:

                 <b>map-ont</b>   Align  noisy long reads of ~10% error rate to a reference genome. This is the default
                           mode.

                 <b>lr:hq</b>     Align accurate long reads (error rate &lt;1%) to a reference genome (<b>-k19</b> <b>-w19</b>  <b>-U50,500</b>
                           <b>-g10k</b>).   This  was  recommended by ONT developers for recent Nanopore reads produced
                           with chemistry v14 that can reach ~99% in accuracy.  It was shown to work better  for
                           accurate Nanopore reads than <b>map-hifi</b>.

                 <b>map-hifi</b>  Align PacBio high-fidelity (HiFi) reads to a reference genome (<b>-xlr:hq</b> <b>-A1</b> <b>-B4</b> <b>-O6,26</b>
                           <b>-E2,1</b>  <b>-s200</b>).  It differs from <b>lr:hq</b> only in scoring. It has not been tested whether
                           <b>lr:hq</b> would work better for PacBio HiFi reads.

                 <b>map-pb</b>    Align older PacBio continuous long (CLR) reads to a reference genome  (<b>-Hk19</b>).   Note
                           that  this  data type is effectively deprecated by HiFi.  Unless you work on very old
                           data, you probably want to use <b>map-hifi</b> or <b>lr:hq</b>.

                 <b>map-iclr</b>  Align Illumina Complete Long Reads  (ICLR)  to  a  reference  genome  (<b>-k19</b>  <b>-B6</b>  <b>-b4</b>
                           <b>-O10,50</b>).  This was recommended by Illumina developers.

                 <b>asm5</b>      Long assembly to reference mapping (<b>-k19</b> <b>-w19</b> <b>-U50,500</b> <b>--rmq</b> <b>-r1k,100k</b> <b>-g10k</b> <b>-A1</b> <b>-B19</b>
                           <b>-O39,81</b> <b>-E3,1</b> <b>-s200</b> <b>-z200</b> <b>-N50</b>).  Typically, the alignment will not extend to regions
                           with  5%  or higher sequence divergence. Use this preset if the average divergence is
                           not much higher than 0.1%.

                 <b>asm10</b>     Long assembly to reference mapping (<b>-k19</b> <b>-w19</b> <b>-U50,500</b> <b>--rmq</b> <b>-r1k,100k</b> <b>-g10k</b> <b>-A1</b>  <b>-B9</b>
                           <b>-O16,41</b> <b>-E2,1</b> <b>-s200</b> <b>-z200</b> <b>-N50</b>).  Use this if the average divergence is around 1%.

                 <b>asm20</b>     Long  assembly to reference mapping (<b>-k19</b> <b>-w10</b> <b>-U50,500</b> <b>--rmq</b> <b>-r1k,100k</b> <b>-g10k</b> <b>-A1</b> <b>-B4</b>
                           <b>-O6,26</b> <b>-E2,1</b> <b>-s200</b> <b>-z200</b> <b>-N50</b>).  Use this if the average divergence is around several
                           percent.

                 <b>splice</b>    Long-read spliced alignment (<b>-k15</b> <b>-w5</b> <b>--splice</b> <b>-g2k</b> <b>-G200k</b> <b>-A1</b> <b>-B2</b> <b>-O2,32</b>  <b>-E1,0</b>  <b>-C9</b>
                           <b>-z200</b>  <b>-ub</b> <b>--junc-bonus=9</b> <b>--cap-sw-mem=0</b> <b>--splice-flank=yes</b>).  In the splice mode, 1)
                           long deletions are taken as introns and represented as the  `<b>N</b>'  CIGAR  operator;  2)
                           long  insertions  are  disabled;  3)  deletion  and insertion gap costs are different
                           during chaining; 4) the computation of the `<b>ms</b>' tag ignores introns to demote hits to
                           pseudogenes.

                 <b>splice:hq</b> Spliced alignment for accurate long RNA-seq reads such as  PacBio  iso-seq  (<b>-xsplice</b>
                           <b>-C5</b> <b>-O6,24</b> <b>-B4</b>).

                 <b>sr</b>        Short-read  alignment  without  splicing  (<b>-k21</b>  <b>-w11</b> <b>--sr</b> <b>--frag=yes</b> <b>-A2</b> <b>-B8</b> <b>-O12,32</b>
                           <b>-E2,1</b> <b>-b0</b> <b>-r100</b> <b>-p.5</b> <b>-N20</b> <b>-f1000,5000</b> <b>-n2</b>  <b>-m25</b>  <b>-s40</b>  <b>-g100</b>  <b>-2K50m</b>  <b>--heap-sort=yes</b>
                           <b>--secondary=no</b>).

                 <b>ava-pb</b>    PacBio CLR all-vs-all overlap mapping (<b>-Hk19</b> <b>-Xw5</b> <b>-e0</b> <b>-m100</b>).

                 <b>ava-ont</b>   Oxford Nanopore all-vs-all overlap mapping (<b>-k15</b> <b>-Xw5</b> <b>-e0</b> <b>-m100</b> <b>-r2k</b>).

   <b>Miscellaneous</b> <b>options</b>
       <b>--no-kalloc</b>
                 Use  the  libc  default allocator instead of the kalloc thread-local allocator.  This debugging
                 option is mostly used with Valgrind to detect invalid memory  accesses.  Minimap2  runs  slower
                 with this option, especially in the multi-threading mode.

       <b>--print-qname</b>
                 Print query names to stderr, mostly to see which query is crashing minimap2.

       <b>--print-seeds</b>
                 Print seed positions to stderr, for debugging only.

</pre><h4><b>OUTPUT</b> <b>FORMAT</b></h4><pre>
       Minimap2  outputs  mapping  positions  in  the  Pairwise  mApping  Format (PAF) by default. PAF is a TAB-
       delimited text format with each line consisting of at least 12 fields as are described in  the  following
       table:
                      ┌─────┬────────┬─────────────────────────────────────────────────────────┐
                      │ <b>Col</b> │  <b>Type</b>  │                       <b>Description</b>                       │
                      ├─────┼────────┼─────────────────────────────────────────────────────────┤
                      │   1 │ string │ Query sequence name                                     │
                      │   2 │  int   │ Query sequence length                                   │
                      │   3 │  int   │ Query start coordinate (0-based)                        │
                      │   4 │  int   │ Query end coordinate (0-based)                          │
                      │   5 │  char  │ `+' if query/target on the same strand; `-' if opposite │
                      │   6 │ string │ Target sequence name                                    │
                      │   7 │  int   │ Target sequence length                                  │
                      │   8 │  int   │ Target start coordinate on the original strand          │
                      │   9 │  int   │ Target end coordinate on the original strand            │
                      │  10 │  int   │ Number of matching bases in the mapping                 │
                      │  11 │  int   │ Number bases, including gaps, in the mapping            │
                      │  12 │  int   │ Mapping quality (0-255 with 255 for missing)            │
                      └─────┴────────┴─────────────────────────────────────────────────────────┘

       When alignment is available, column 11 gives the total number of sequence matches, mismatches and gaps in
       the  alignment; column 10 divided by column 11 gives the BLAST-like alignment identity. When alignment is
       unavailable, these two columns are approximate. PAF may optionally have additional fields in the SAM-like
       typed key-value format. Minimap2 may output the following tags:
                        ┌─────┬──────┬───────────────────────────────────────────────────────┐
                        │ <b>Tag</b> │ <b>Type</b> │                      <b>Description</b>                      │
                        ├─────┼──────┼───────────────────────────────────────────────────────┤
                        │  tp │  A   │ Type of aln: P/primary, S/secondary and I,i/inversion │
                        │  cm │  i   │ Number of minimizers on the chain                     │
                        │  s1 │  i   │ Chaining score                                        │
                        │  s2 │  i   │ Chaining score of the best secondary chain            │
                        │  NM │  i   │ Total number of mismatches and gaps in the alignment  │
                        │  MD │  Z   │ To generate the ref sequence in the alignment         │
                        │  AS │  i   │ DP alignment score                                    │
                        │  SA │  Z   │ List of other supplementary alignments                │
                        │  ms │  i   │ DP score of the max scoring segment in the alignment  │
                        │  nn │  i   │ Number of ambiguous bases in the alignment            │
                        │  ts │  A   │ Transcript strand (splice mode only)                  │
                        │  cg │  Z   │ CIGAR string (only in PAF)                            │
                        │  cs │  Z   │ Difference string                                     │
                        │  dv │  f   │ Approximate per-base sequence divergence              │
                        │  de │  f   │ Gap-compressed per-base sequence divergence           │
                        │  rl │  i   │ Length of query regions harboring repetitive seeds    │
                        └─────┴──────┴───────────────────────────────────────────────────────┘

       The <b>cs</b> tag encodes difference sequences in the short form or the entire query <u>AND</u> reference sequences  in
       the long form. It consists of a series of operations:
                        ┌────┬────────────────────────────┬─────────────────────────────────┐
                        │ <b>Op</b> │           <b>Regex</b>            │           <b>Description</b>           │
                        ├────┼────────────────────────────┼─────────────────────────────────┤
                        │  = │ [ACGTN]+                   │ Identical sequence (long form)  │
                        │  : │ [0-9]+                     │ Identical sequence length       │
                        │  * │ [acgtn][acgtn]             │ Substitution: ref to query      │
                        │  + │ [acgtn]+                   │ Insertion to the reference      │
                        │  - │ [acgtn]+                   │ Deletion from the reference     │
                        │  ~ │ [acgtn]{2}[0-9]+[acgtn]{2} │ Intron length and splice signal │
                        └────┴────────────────────────────┴─────────────────────────────────┘

</pre><h4><b>LIMITATIONS</b></h4><pre>
       * Minimap2 may produce suboptimal alignments through long low-complexity regions where seed positions may
         be suboptimal. This should not be a big concern because even the optimal alignment may be wrong in such
         regions.

       * Minimap2  requires  SSE2  or NEON instructions to compile. It is possible to add non-SSE2/NEON support,
         but it would make minimap2 slower by several times.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <a href="../man1/miniasm.1.html">miniasm</a>(1), <a href="../man1/minimap.1.html">minimap</a>(1), <a href="../man1/bwa.1.html">bwa</a>(1).

minimap2-2.27 (r1193)                             12 March 2024                                      <u><a href="../man1/minimap2.1.html">minimap2</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>