<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>msa_split - Partitions a multiple sequence alignment either at designated</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/phast">phast_1.7+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       msa_split - Partitions a multiple sequence alignment either at designated

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Partitions a multiple sequence alignment either at designated columns, or according to specified category
       labels, and outputs sub-alignments for the partitions.  Optionally splits an associated annotations file.

</pre><h4><b>EXAMPLE</b></h4><pre>
       (See below for details on options)

       1.  Read  an  alignment  for  a  whole human chromosome from a MAF file and extract sub-alignments in 1Mb
       windows overlapping by 1kb.  Use sufficient statistics (SS) format for output (can be used  by  phyloFit,
       phastCons, or exoniphy).  Set window boundaries between alignment blocks, if possible.

              msa_split  chr1.maf  <b>--refseq</b>  chr1.fa  <b>--in-format</b>  MAF  <b>--windows</b>  1000000,1000  <b>--out-format</b> SS
              <b>--between-blocks</b> 5000 <b>--out-root</b> chr1

       (Windows will be defined using the coordinate system of the first sequence in the alignment,  assumed  to
       be the reference sequence; output will be to chr1.1-1000000.ss, chr1.999001-1999000.ss, ...)

       2.  As  in  (1),  but report unordered sufficient statistics (much more compact and adequate for use with
       phyloFit).

              msa_split chr1.maf  <b>--refseq</b>  chr1.fa  <b>--in-format</b>  MAF  <b>--windows</b>  1000000,1000  <b>--out-format</b>  SS
              <b>--between-blocks</b> 5000 <b>--out-root</b> chr1 <b>--unordered-ss</b>

       3.  Extract  sub-alignments of sites in conserved elements and not in conserved elements, as defined by a
       BED file (coordinates assumed to be for 1st sequence).  Read multiple alignment in FASTA format.

              msa_split mydata.fa <b>--features</b> conserved.bed <b>--by-category</b> <b>--out-root</b> mydata

       (Output will be to mydata.background-0.fa and  mydata.bed_feature-1.fa  [latter  has  sites  of  category
       number  1,  defined by bed file] 3. Extract sub-alignments of sites in each of the three codon positions,
       as defined by a GFF file (coordinates assumed to be for 1st sequence).  Reverse complement genes on minus
       strand.

              msa_split chr22.maf <b>--in-format</b> MAF <b>--features</b> chr22.gff <b>--by-category</b> <b>--catmap</b>  "NCATS  3  ;  CDS
              1-3" <b>--do-cats</b> CDS <b>--reverse-compl</b> <b>--out-root</b> chr22 <b>--out-format</b> SS

       (Output will be to chr22.cds-1.ss, chr22.cds-2.ss, chr22.cds-3.ss)

       4.  Split an alignment into pieces corresponding to the genes in a GFF file.  Assume genes are defined by
       the tag "transcript_id".

              msa_split cftr.fa <b>--features</b> cftr.gff <b>--by-group</b> transcript_id

       5. Obtain a sub-alignment for each of a set of regulatory regions, as defined in a BED file.

              msa_split chr22.maf <b>--in-format</b> MAF  <b>--refseq</b>  chr22.fa  <b>--features</b>  chr22.reg.bed  <b>--for-features</b>
              <b>--out-root</b> chr22.reg

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Splitting</b> <b>options</b>

       <b>--windows</b>, <b>-w</b> &lt;win_size,win_overlap&gt;

              Split the alignment into "windows" of size &lt;win_size&gt; bases, overlapping by &lt;win_overlap&gt;.

       <b>--by-category</b>, <b>-L</b>

              (Requires  <b>--features</b>) Split by category, as defined by annotations file and (optionally) category
              map (see <b>--catmap</b>)

       <b>--by-group</b>, <b>-P</b> &lt;tag&gt;

              (Requires <b>--features</b>) Split by groups in annotation file, as defined  by  specified  tag.   Splits
              midway  between every pair of consecutive groups.  Features will be sorted by group.  There should
              be no overlapping features (see 'refeature <b>--unique</b>').

       <b>--for-features</b>, <b>-F</b> (Requires <b>--features</b>) Extract section of alignment  corresponding  to  every  feature.
              There will be no output for regions not covered by features.

       <b>--by-index</b>,  <b>-p</b> &lt;indices&gt; List of explicit indices at which to split alignment (comma-separated).  If the
              list of indices is "10,20",  then  sub-alignments  will  be  output  for  sites  1-9,  10-19,  and
              20-&lt;msa_len&gt;.   Note  that the indices are relative to the input alignment, and not necessarily in
              genomic coordinates.

       <b>--npartitions</b>, <b>-n</b> &lt;number&gt;

              Split alignment equally into specified number of partitions.

       <b>--between-blocks</b>, <b>-B</b> &lt;radius&gt; (Not for use with <b>--by-category</b> or  <b>--for-features</b>)  Try  to  partition  at
              sites  between  alignment blocks.  Assumes a reference sequence alignment, with the first sequence
              as the reference seq (as created by multiz).  Blocks of 30 sites with gaps in  all  sequences  but
              the  reference seq are assumed to indicate boundaries between alignment blocks.  Partition indices
              will not be moved more than &lt;radius&gt; sites.

       <b>--features</b>, <b>-g</b> &lt;fname&gt;

              (For use with <b>--by-category</b>, <b>--by-group</b>, <b>--for-features</b>, or <b>--windows</b>) Annotations file.   May  be
              GFF,  BED, or genepred format.  Coordinates are assumed to be in the coordinate frame of the first
              sequence in the alignment (assumed to be the reference sequence).

       <b>--catmap</b>, <b>-c</b> &lt;fname&gt;|&lt;string&gt; (Optionally use with <b>--by-category</b>) Mapping of feature  types  to  category
              numbers.   Can  either  give a filename or an "inline" description of a simple category map, e.g.,
              <b>--catmap</b> "NCATS = 3 ; CDS 1-3" or <b>--catmap</b> "NCATS = 1 ; UTR 1".

       <b>--refidx</b>, <b>-d</b> &lt;frame_index&gt;

              (For use with <b>--windows</b> or <b>--by-index</b>) Index of frame of reference for split indices.  Default  is
              1 (1st sequence assumed reference).

</pre><h4><b>File</b> <b>names</b> <b>&amp;</b> <b>formats,</b> <b>type</b> <b>of</b> <b>output,</b> <b>etc.</b></h4><pre>

       <b>--in-format</b>, <b>-i</b> FASTA|PHYLIP|MPM|MAF|SS Input alignment file format.  Default is to guess format from

              file contents.

       <b>--refseq</b>, <b>-M</b> &lt;fname&gt;

              (For use with <b>--in-format</b> MAF) Name of file containing reference sequence, in FASTA format.

       <b>--out-format</b>, <b>-o</b> FASTA|PHYLIP|MPM|SS Output alignment file format.  Default is FASTA.

       <b>--out-root</b>, <b>-r</b> &lt;name&gt; Filename root for output files (default "msa_split").

       <b>--sub-features</b>,  <b>-f</b>  (For use with <b>--features</b>) Output subsets of features corresponding to subalignments.
              Features overlapping partition boundaries will be discarded.  Not permitted with

       <b>--by-category</b>.

       <b>--reverse-compl</b>, <b>-s</b>

              Reverse complement all segments having at least one feature on the reverse strand and none on  the
              positive  strand.   For  use  with  <b>--by-group</b>.  Can also be used with <b>--by-category</b> to ensure all
              sites in a category are represented in the same strand orientation.

       <b>--gap-strip</b>, <b>-G</b> ALL|ANY|&lt;seqno&gt;

              Strip columns in output alignments containing all  gaps,  any  gaps,  or  gaps  in  the  specified
              sequence (&lt;seqno&gt;; indexing begins with one).  Default is not to strip any columns.

       <b>--seqs</b>, <b>-l</b> &lt;seq_list&gt; Include only specified sequences in output.  Indicate by

              sequence number or name (numbering starts with 1 and is evaluated *after* <b>--order</b> is applied).

       <b>--exclude</b>, <b>-x</b> Exclude rather than include specified sequences.

       <b>--order</b>, <b>-O</b> &lt;name_list&gt;

              Change  order  of  rows  in  alignment  to match sequence names specified in name_list.  If a name
              appears in name_list but not in the alignment, a row of gaps will be inserted.

       <b>--min-informative</b>, <b>-I</b> &lt;n&gt;

              Only output alignments having at least &lt;n&gt; informative sites (sites at which at least two  non-gap
              and non-N gaps are present).

       <b>--do-cats</b>,  <b>-C</b>  &lt;cat_list&gt;  (For  use  with  <b>--by-category</b>)  Output sub-alignments for only the specified
              categories (column-delimited list).

       <b>--tuple-size</b>, <b>-T</b> &lt;tuple_size&gt;

              (for use with <b>--by-category</b> or  <b>--out-format</b>  SS)  Size  of  tuples  of  columns  to  consider  in
              downstream  analysis  (e.g.,  with  context-dependent  phylogenetic models; see 'phyloFit').  With
              <b>--by-category</b>, insert tuple_size-1 columns of missing data between sites that were not adjacent in
              the original alignment, to avoid creating  artificial  context.   With  <b>--out-format</b>  SS,  express
              sufficient statistics in terms of tuples of specified size.

       <b>--unordered-ss</b>,  <b>-z</b>  (For  use  with  <b>--out-format</b>  SS) Suppress the portion of the sufficient statistics
              concerned with the order in which columns appear.

       <b>--summary</b>, <b>-S</b>

              Output summary of each output alignment to a file with suffix ".sum"  (includes  base  frequencies
              and numbers of gapped columns).

   <b>Other</b>

       <b>--quiet</b>, <b>-q</b> Proceed quietly.

       <b>--help</b>, <b>-h</b>

              Print this help message.

msa_split 1.4                                       May 2016                                        <u><a href="../man1/MSA_SPLIT.1.html">MSA_SPLIT</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>