<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>msa_view - Provides various kinds of "views" of one or more multiple</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/phast">phast_1.7+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       msa_view - Provides various kinds of "views" of one or more multiple

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Provides  various  kinds of "views" of one or more multiple alignments.  Can extract a sub-alignment from
       an alignment (by row or by column) or  combine  several  alignments  into  one.   Also  can  extract  the
       sufficient  statistics  for  phylogenetic  analysis  from  an  alignment,  optionally accounting for site
       categories that are defined  by  an  auxiliary  annotations  file.   Supports  various  other  functions,
       including  gap  stripping,  column  randomization,  and  reordering of sequences.  Capable of reading and
       writing in a few common formats.  Can be used for file conversion (by default, output is the entire input
       alignment).

</pre><h4><b>EXAMPLE</b></h4><pre>
       (See below for more details on options)

       1. Convert alignment formats (default input and output is FASTA)

              msa_view myfile.fa <b>--out-format</b> PHYLIP &gt; myfile.ph

              msa_view myfile2.raw <b>--in-format</b> MPM &gt; myfile2.fa

       2. Obtain a sub-alignment by position, using the coordinate frame of the first sequence in the alignment.

              msa_view myfile.fa <b>--start</b> 1234 <b>--end</b> 5678 <b>--refidx</b> 1 &gt; mysub.fa

       3. Obtain a sub-alignment by sequence.

              msa_view myfile.fa <b>--seqs</b> 1,4,5 &gt; seqs145.fa

              msa_view myfile.fa <b>--seqs</b> 1,4,5 <b>--exclude</b> &gt; seqs236.fa

       (can also specify sequences by name, e.g., <b>--seqs</b> cow,rat,pig)

       4. Concatenate alignments.

              msa_view <b>--aggregate</b> human,mouse,rat myf1.fa myf2.fa myf3.fa &gt; concat.fa

       (source alignments may have different subsets of sequences and may use different sequence  orders;  here,
       human,mouse,rat defines full set and order in output alignment)

       5. Extract sufficient statistics from a FASTA file.

              msa_view myfile.fa <b>--out-format</b> SS &gt; myfile.ss

       6.  Extract  sufficient  statistics  from  a  MAF  file for a complete human chromosome.  (Can be used by
       phyloFit.)

              msa_view chr1.maf <b>--out-format</b> SS &gt; chr1.ss

       7. As in (6), but include information about regions of the reference sequence  not  present  in  the  MAF
       file, and include a representation of the order in which alignment columns occur (needed by programs such
       as phastCons or exoniphy).

              msa_view chr1.maf <b>--refseq</b> chr1.fa <b>--out-format</b> SS &gt; chr1.ordered.ss

       8.  As  in (6), but collect statistics for pairs of adjacent sites (can be used by phyloFit to estimate a
       dinucleotide model).

              msa_view chr1.maf <b>--out-format</b> SS <b>--tuple-size</b> 2 &gt; chr1.pairs.ss

       9. Pool sufficient statistics from several human chromosomes.

              msa_view <b>--aggregate</b> human,mouse,rat <b>--out-format</b> SS chr1.ss chr2.ss chr3.ss &gt; chr123.ss

       10. Extract separate sufficient statistics for the three codon positions, as defined by annotations in  a
       GFF file.

              msa_view  chr1.maf  <b>--features</b>  chr22.gff  <b>--catmap</b>  "NCATS  =  3;  CDS  1-3"  <b>--out-format</b>  SS  &gt;
              chr22.pos.ss

       11. As in (10), but re-orient genes on - strand so that stats reflect + strand.  Assume genes are defined
       by tag "transcript_id".

              msa_view  chr1.maf  <b>--features</b>  chr22.gff  <b>--catmap</b>  "NCATS  =  3;   CDS   1-3"   <b>--reverse-groups</b>
              transcript_id <b>--out-format</b> SS &gt; chr22.pos.ss

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Obtaining</b> <b>sub-alignments</b> <b>and</b> <b>combining</b> <b>alignments</b>

       <b>--start</b>, <b>-s</b> &lt;start_col&gt;

              Starting  column  of sub-alignment (indexing starts with 1).  Default is 1.  Note that coordinates
              use the frame of reference of the entire alignment unless <b>--refidx</b> 1 is specified.

       <b>--end</b>, <b>-e</b> &lt;end_col&gt;

       Ending column of sub-alignment.
              Default is length of

       alignment.
              Note that coordinates use the frame of reference

              of the entire alignment unless <b>--refidx</b> 1 is specified.

       <b>--seqs</b>, <b>-l</b> &lt;seq_list&gt; Comma-separated list of sequences to  include  (default)  exclude  (if  <b>--exclude</b>).
              Indicate  by  sequence number or name (numbering starts with 1 and is evaluated *after* <b>--order</b> is
              applied).

       <b>--exclude</b>, <b>-x</b>

              Exclude rather than include specified sequences.

       <b>--refidx</b>, <b>-r</b> &lt;ref_seq&gt;

       Index of reference sequence for coordinates.
              Use 0 to

              indicate the coordinate system of the alignment as a whole (this is the default).

       <b>--aggregate</b>, <b>-A</b> &lt;name_list&gt;

              (Not compatible with <b>--start</b> or <b>--end</b>) Create an aggregate  alignment  from  a  set  of  alignment
              files,  by  concatenating individual alignments.  If used with <b>--out-format</b> SS and <b>--unordered-ss</b>,
              the aggregate alignment will never be created explicitly (recommended for large data  sets).   The
              argument  &lt;name_list&gt;  must  be  a  list  of  sequence names, including all names in all specified
              alignments (missing sequences will be replaced by rows of missing data).  The standard &lt;msa_fname&gt;
              argument should be replaced with a list of (whitespaceseparated) file names.

       <b>--split-all</b>, <b>-X</b> &lt;filename root&gt;

              Split output  alignment  into  separate  fasta  files  by  species.   File  naming  convention  is
              filename_root.species.fa.  If  used  with  <b>--gap-strip</b>,  gap  characters will be stripped from all
              output files.  In this case, '--gap-strip &lt;s&gt;' should NOT be used (ALL or  ANY  should  both  work
              fine).

   <b>File</b> <b>formats,</b> <b>gap</b> <b>stripping,</b> <b>reordering,</b> <b>etc.</b>

       <b>--in-format</b>, <b>-i</b> PHYLIP|FASTA|MPM|MAF|SS

       (Default is to guess format from file contents).
              Input file

       format.
              FASTA is as usual.  PHYLIP is compatible with the formats

       used in the PHYLIP and PAML packages.
              MPM is the format used by the

              MultiPipMaker  aligner  and  some  other  of  Webb Miller's older tools.  MAF ("Multiple Alignment
              Format") is used by MULTIZ/TBA and the UCSC Genome Browser.  SS is a simple format describing  the
              sufficient  statistics  for phylogenetic inference (distinct columns or tuple of columns and their
              counts).  Use <b>--out-format</b> SS with <b>--in-format</b> MAF for  best  efficiency  (explicit  alignment  is
              never created).  Also, use <b>--unordered-ss</b> if possible.

       <b>--out-format</b>, <b>-o</b> PHYLIP|FASTA|MPM|SS (Default FASTA) Output file format.

       <b>--alphabet</b>, <b>-a</b> &lt;alphabet_string&gt;

       Use the specified alphabet (default "ACGT").
              In addition,

              '-' characters are assumed to represent alignment gaps, and '*' and 'N' characters are allowed for
              missing  data.   Alphabetical  letters  not  in the alphabet will be converted to 'N's upon input.
              This option is ignored with SS input (alphabet specified within SS files.)

       <b>--soft-masked</b>, <b>-f</b>

              Implies <b>--alphabet</b> 'ACGTNacgtn', useful for soft-masked sequences.

       <b>--unmask</b>, <b>-u</b>

              Remove soft-masking; convert to uppercase.

       <b>--pretty</b>, <b>-P</b> Pretty-print alignment (use '.' when character  matches  corresponding  character  in  first
              sequence).  Ignored if <b>--out-format</b> SS is selected.

       <b>--gap-strip</b>,  <b>-G</b>  ALL|ANY|&lt;s&gt;  Strip  columns  containing  all  gaps, any gaps, or a gap in the specified
              sequence (&lt;s&gt;).  Indexing starts at one and refers to the list *after*  any  sequences  have  been
              added or subtracted (via <b>--seqs</b> and <b>--exclude</b> or <b>--order</b>).

       <b>--collapse-missing</b>, <b>-p</b>

              (For  use with <b>-o</b> SS) Convert all missing-data characters and gaps to "*" characters.  Can be used
              to make sufficient statistics more compact, which can improve the  performance  of  phyloFit  (all
              missing data and gap characters are typically treated the same by phyloFit anyway).

       <b>--mark-missing</b>,  <b>-K</b>  &lt;maxlen&gt;  Convert  all  gaps  of length greater than &lt;maxlen&gt; to "*" characters.  If
              <b>--refidx</b> is specified (with a positive index), gaps in the designated reference sequence will  not
              be  altered.   This  is  a  useful heuristic for distinguishing between microindels and regions of
              missing data  (e.g.,  due  to  large-scale  indels,  incomplete  assemblies,  or  highly  diverged
              sequences).

       <b>--missing-as-indels</b>, <b>-m</b>

              Convert  all  missing  data characters (Ns and *s) to gap characters, except for Ns in a reference
              sequence specified by <b>--refidx</b>, which will be replaced by randomly  selected  nucleotides.   (This
              allows  the  coordinate  frame  for  the  reference sequence to be maintained; this option is only
              recommended if such Ns are rare.)  If <b>--refidx</b> is not used, all Ns will be replaced by gaps.   You
              may want to use <b>--gap-strip</b> ALL with this option.

       <b>--order</b>, <b>-O</b> &lt;name_list&gt; Change order of rows in alignment to match sequence names specified in name_list.
              If  a  name  appears  in name_list but not in the alignment, a row of gaps will be inserted.  This
              option is applied to the alignment *before* <b>--seqs</b>, <b>--refidx</b>, and <b>--gap-strip</b> are applied.

       <b>--reverse-complement</b>, <b>-V</b>

              Reverse complement output alignment.

       <b>--randomize</b>,  <b>-R</b>  Randomly  permute  the  columns  of  the  source  alignment   (done   *before*   taking
              sub-alignment).   Requires  an  ordered  representation  of  the  alignment  (careful  using  with
              <b>--in-format</b> SS|MAF <b>--</b> will create full alignment from sufficient statistics).

       <b>--fill-Ns</b>, <b>-N</b> &lt;s:b-e&gt;

              Fill sequence no.  &lt;s&gt;  with  Ns,  from  &lt;b&gt;  to  &lt;e&gt;.  Applied  before  <b>--start</b>,  <b>--end</b>,  <b>--seqs</b>,
              <b>--gap-strip</b>,  but  after  <b>--order</b>.   Coordinate  frame  depends on <b>--refidx</b>.  Can be used multiple
              times.

       <b>--summary-only</b> <b>-S</b> Report only summary statistics, rather than complete  alignment.   Statistics  are  for
              alignment that would otherwise be output (i.e., after other options have been applied).

       <b>--window-summary</b>, <b>-w</b> &lt;win_size&gt; Like <b>-S</b>, but output summary statistics for non-overlapping windows of the
              specified size.  (Sufficient statistics)

       <b>--tuple-size</b>, <b>-T</b> &lt;tup_size&gt; (For use with <b>--out-format</b> SS).  Represent an alignment in terms of tuples of
              columns of the designated size.  Useful

              with context-dependent phylogenetic models.

       <b>--unordered-ss</b>, <b>-z</b>

       (For use with <b>--out-format</b> SS).
              Suppress the portion of the

              sufficient  statistics  concerned with the order in which columns appear.  Useful for analyses for
              which order is unimportant.  (MAF input)

       <b>--refseq</b>, <b>-M</b> &lt;fname&gt;

              Read the complete text of the reference sequence from &lt;fname&gt; (FASTA format) and combine  it  with
              the  contents  of  the  MAF  file  to  produce a complete, ordered representation of the alignment
              (unaligned regions will be represented by gaps).  Best used with <b>--out-format</b> SS.   The  reference
              sequence of the MAF file is assumed to be the one that appears first in each block.

       <b>--keep-overlapping</b>, <b>-k</b>

              Keep  blocks in MAF that have overlapping coordinates in the reference (1st) sequence (by default,
              only the first one is kept).  Useful in extracting unordered stats from a  jumbled  collection  of
              MAF  blocks  (e.g.,  output  of  Jim  Kent's  mafFrags  program).   Cannot  be used with <b>--refseq</b>,
              <b>--features</b>, or

       <b>--cats-cycle</b>.  (Site categories: all options require <b>--out-format</b> SS)

       <b>--features</b>, <b>-g</b> &lt;gff_fname&gt;

              (Requires <b>--catmap</b>) Read sequence annotations from the specified file (GFF) and label the  columns
              of the alignment accordingly.  Note: UCSC BED and genepred formats are now recognized as well.

       <b>--catmap</b>, <b>-c</b> &lt;fname&gt;|&lt;string&gt;

              (optionally  use with <b>--features</b>) Mapping of feature types to category numbers.  Can either give a
              filename or an "inline" description of a simple category map, e.g., <b>--catmap</b> "NCATS = 3 ; CDS 1-3"
              or <b>--catmap</b> "NCATS = 1 ; UTR 1".

       <b>--cats-cycle</b>, <b>-Y</b> &lt;cycle_size&gt; (alternative to <b>--features</b> and <b>--catmap</b>) Assign site categories  in  cycles
              of the specified size, e.g., as 1,2,3,...,1,2,3 (for cycle_size == 3).  Useful for in-frame coding
              sequence, or to partition a data set into nonoverlapping tuples of columns (use with <b>--do-cats</b>).

       <b>--do-cats</b>, <b>-C</b> &lt;cat_list&gt;

       (For use with <b>--features</b> or <b>--cats-cycle</b>)
              Obtain

              sufficient statistics only for the specified categories (comma-delimited list, by number).

       <b>--codons</b>,  <b>-D</b>  Extract  sufficient statistics for in-frame codons.  Implies <b>--tuple-size</b> 3 <b>--cats-cycle</b> 3
              <b>--do-cats</b> 3.  Not appropriate

              for use with <b>--features</b>/--catmap.

       <b>--reverse-groups</b>, <b>-W</b> &lt;tag&gt;

              (For use with <b>--features</b>) Group features by &lt;tag&gt; (e.g., "transcript_id" or "exon_id") and reverse
              complement segments of the alignment corresponding to groups on the reverse strand.   Groups  must
              be  non-overlapping  (see  refeature  <b>--unique</b>).  Useful when extracting sufficient statistics for
              strand-specific site categories (e.g., codon positions).

       <b>--4d</b>, <b>-4</b>

              (For use with <b>--features</b>; assumes coding regions have  feature  type  'CDS')   Extract  sufficient
              statistics  for  fourfold  degenerate synonymous sites.  Implies <b>--out-format</b> SS <b>--unordered-stats</b>
              <b>--tuple-size</b> 3 <b>--reverse-groups</b> transcript_id.

</pre><h4><b>Alignment</b> <b>cleaning</b></h4><pre>

       <b>--clean-coding</b>, <b>-L</b> &lt;seqname&gt;

              Clean an alignment of coding sequences with respect to a named reference sequence.  Removes  sites
              with  gaps  and  blocks  of  gapless sites smaller than 10 codons in length, ensures everything is
              in-frame wrt reference sequence, prohibits in-frame stop codons.  Reference  sequence  must  begin
              with a start codon and end with a stop codon.

       <b>--clean-indels</b>, <b>-I</b> &lt;nseqs&gt;

       Clean an alignment with special attention to indels.
              Sites

              with  fewer  than  &lt;nseqs&gt;  bases  are  removed;  bases  adjacent  to  indels,  and  short gapless
              subsequences, are replaced with Ns.  If used with <b>--tuple-size</b>, then &lt;tup_size&gt;-1  columns  of  Ns
              will be retained between columns not adjacent in the original alignment.  Frame is not considered.

   <b>Other</b>

       <b>--help</b>, <b>-h</b> Print this help message.

msa_view 1.4                                        May 2016                                         <u><a href="../man1/MSA_VIEW.1.html">MSA_VIEW</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>