<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>muscle - Multiple alignment program of protein sequences</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/muscle">muscle_5.1.0-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       muscle - Multiple alignment program of protein sequences

</pre><h4><b>DESCRIPTION</b></h4><pre>
       MUSCLE  is  a  multiple  alignment  program  for  protein  sequences. MUSCLE stands for multiple sequence
       comparison by log-expectation. In the authors tests, MUSCLE achieved the highest  scores  of  all  tested
       programs on several alignment accuracy benchmarks, and is also one of the fastest programs out there.

</pre><h4><b>USAGE</b></h4><pre>
   <b>Align</b> <b>FASTA</b> <b>input,</b> <b>write</b> <b>aligned</b> <b>FASTA</b> <b>(AFA)</b> <b>output:</b>
              muscle <b>-align</b> input.fa <b>-output</b> aln.afa

       Align  large  input  using Super5 algorithm if <b>-align</b> is too expensive, typically needed with more than a
       few hundred sequences:

              muscle <b>-super5</b> input.fa <b>-output</b> aln.afa

   <b>Single</b> <b>replicate</b> <b>alignment:</b>
              muscle <b>-align</b> input.fa <b>-perm</b> PERM <b>-perturb</b> SEED <b>-output</b> aln.afa muscle <b>-super5</b> input.fa <b>-perm</b> PERM
              <b>-perturb</b> SEED <b>-output</b> aln.afa

              PERM is guide tree permutation none, abc, acb, bca (default none).  SEED is perturbation  seed  0,
              1, 2... (default 0 = don't perturb).

       Ensemble  of  replicate  alignments, output in Ensemble FASTA (EFA) format, EFA has one aligned FASTA for
       each replicate with header line "&lt;PERM.SEED":

              muscle  <b>-align</b>  input.fa  <b>-stratified</b>  <b>-output</b>  stratified_ensemble.efa  muscle  <b>-align</b>   input.fa
              <b>-diversified</b> <b>-output</b> diversified_ensemble.afa

       <b>-replicates</b> N

              Number of replicates, defaults 4, 100, 100 for stratified,

              diversified,  resampled. With <b>-stratified</b> there is one replicate per guide tree permutation, total
              is 4 x N.

       Generate resampled ensemble from existing ensemble by sampling columns with replacement:

              muscle <b>-resample</b> ensemble.efa <b>-output</b> resampled.efa

       <b>-maxgapfract</b> F

              Maximum fraction of gaps in a column (F=0..1, default 0.5).

       <b>-minconf</b> CC

              Minimum column confidence (CC=0..1, default 0.5).

       If ensemble output filename has @, then one FASTA file  is  generated  for  each  replicate  where  @  is
       replaced by perm.s, otherwise all replicates are written to one EFA file.

   <b>Calculate</b> <b>disperson</b> <b>of</b> <b>an</b> <b>ensemble:</b>
              muscle <b>-disperse</b> ensemble.efa

   <b>Extract</b> <b>replicate</b> <b>with</b> <b>highest</b> <b>total</b> <b>CC</b> <b>(diversified</b> <b>input</b> <b>recommended):</b>
              muscle <b>-maxcc</b> ensemble.efa <b>-output</b> maxcc.afa

   <b>Extract</b> <b>aligned</b> <b>FASTA</b> <b>files</b> <b>from</b> <b>EFA</b> <b>file:</b>
              muscle <b>-efa_explode</b> ensemble.efa

   <b>Convert</b> <b>FASTA</b> <b>to</b> <b>EFA,</b> <b>input</b> <b>has</b> <b>one</b> <b>filename</b> <b>per</b> <b>line:</b>
              muscle <b>-fa2efa</b> filenames.txt <b>-output</b> ensemble.efa

       Update  ensemble  by  adding two sequences of digits to each replicate, digits are column confidence (CC)
       values, e.g. "73" means CC=0.73, "++" is CC=1.0:

              muscle <b>-addconfseqs</b> ensemble.efa <b>-output</b> ensemble_cc.efa

       Calculate letter confidence (LC) values, <b>-ref</b> specifies the alignment to  compare  against  the  ensemble
       (e.g. from <b>-maxcc</b>), output is in aligned FASTA format with LC values 0, 1 ... 9 instead of letters:

              muscle <b>-letterconf</b> ensemble.efa <b>-ref</b> aln.afa <b>-output</b> letterconf.afa

       <b>-html</b> aln.html

              Alignment colored by LC in HTML format.

       <b>-jalview</b> aln.features

              Jalview feature file with LC values and colors.

   <b>More</b> <b>documentation</b> <b>at:</b>
              https://drive5.com/muscle

</pre><h4><b>AUTHOR</b></h4><pre>
        This manpage was written by Andreas Tille for the Debian distribution and
        can be used for any other usage of the program.

muscle 5.1                                        January 2022                                         <u><a href="../man1/MUSCLE.1.html">MUSCLE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>