<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>pal2nal.pl - use insight from protein alignment for nucleotide alignment</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/pal2nal">pal2nal_14.1-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       pal2nal.pl - use insight from protein alignment for nucleotide alignment

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>pal2nal.pl</b> <u>pep.aln</u>  <u>nuc.fasta</u>  [<u>nuc.fasta</u>...]  [<u>options</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       pal2nal.pl  (v13)

       pep.aln:
              protein alignment either in CLUSTAL or FASTA format

       nuc.fasta:
              DNA sequences (single multi-fasta or separated files)

       Options:
              <b>-h</b>            Show help

              <b>-output</b> (clustal|paml|fasta|codon)

              Output format; default = clustal

       <b>-blockonly</b>
              Show only user specified blocks '#' under CLUSTAL alignment (see example)

       <b>-nogap</b> remove columns with gaps and inframe stop codons

       <b>-nomismatch</b>
              remove mismatched codons (mismatch between pep and cDNA) from the output

              <b>-codontable</b> (universal|vmitochondria)

              Codon table

              - universal (default) - vmitochondria -&gt; vertebrate mitochondrial

       <b>-html</b>  HTML output (only for the web server)

       <b>-nostderr</b>
              No STDERR messages (only for the web server)

              - sequence order in pep.aln and nuc.fasta should be the same.

              - IDs in pep.aln are used in the output.

       pal2nal.pl  (v13)

       Usage:  pal2nal.pl  pep.aln  nuc.fasta  [nuc.fasta...]  [options]

       pep.aln:
              protein alignment either in CLUSTAL or FASTA format

       nuc.fasta:
              DNA sequences (single multi-fasta or separated files)

       Options:
              <b>-h</b>            Show help

              <b>-output</b> (clustal|paml|fasta|codon)

              Output format; default = clustal

       <b>-blockonly</b>
              Show only user specified blocks '#' under CLUSTAL alignment (see example)

       <b>-nogap</b> remove columns with gaps and inframe stop codons

       <b>-nomismatch</b>
              remove mismatched codons (mismatch between pep and cDNA) from the output

              <b>-codontable</b> (universal|vmitochondria)

              Codon table

              - universal (default) - vmitochondria -&gt; vertebrate mitochondrial

       <b>-html</b>  HTML output (only for the web server)

       <b>-nostderr</b>
              No STDERR messages (only for the web server)

              - sequence order in pep.aln and nuc.fasta should be the same.

              - IDs in pep.aln are used in the output.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       The  full  documentation  for  <b>pal2nal.pl</b>  is maintained as a Texinfo manual.  If the <b>info</b> and <b>pal2nal.pl</b>
       programs are properly installed at your site, the command

              <b>info</b> <b>pal2nal.pl</b>

       should give you access to the complete manual.

pal2nal.pl                                        November 2010                                    <u><a href="../man1/PAL2NAL.PL.1.html">PAL2NAL.PL</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>