<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>parseAlignment - pre-compute probabilities of (observed) reads alignments</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/bitseq">bitseq_0.7.5+dfsg-6_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       parseAlignment - pre-compute probabilities of (observed) reads alignments

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>parseAlignment</b> <u>-o</u> <u>&lt;outFileName&gt;</u> <u>-s</u> <u>&lt;trSeqFileName&gt;</u>  [<u>OPTIONS</u>] [<u>alignment</u> <u>file</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Pre-computes probabilities of (observed) reads' alignments.

              [alignment file] should be in either SAM or BAM format.

</pre><h4><b>OPTIONS</b></h4><pre>

       <b>--help</b>

              Show this help information.

       <b>--distributionFile=</b>&lt;distributionFileName&gt;

              Name of file to which read-distribution should be saved.

       <b>--excludeSingletons</b>

              Exclude single mate alignments for paired-end reads. (default: Off)

       <b>-e</b> &lt;expFileName&gt; ,   <b>--expressionFile=</b>&lt;expFileName&gt;

              Transcript  relative expression estimates <b>---</b> for better non-uniform read distribution estimation.

       <b>--failed=</b>&lt;failed&gt;

              File name where to save names of reads that failed to align.

       <b>-f</b> &lt;format&gt; ,   <b>--format=</b>&lt;format&gt;

              Input format: either SAM, BAM.

       <b>--lenMu=</b>&lt;lenMu&gt;

              Set mean of log fragment length distribution. (l_frag ~ LogNormal(mu,sigma^2))

       <b>--lenSigma=</b>&lt;lenSigma&gt;

              Set sigma^2 (or variance) of log fragment length distribution. (l_frag ~ LogNormal(mu,sigma^2))

       <b>--mateNamesDiffer</b>

              Mates from paired-end reads have different names. (default: Off)

       <b>-l</b> &lt;maxAlignments&gt; ,   <b>--limitA=</b>&lt;maxAlignments&gt;

              Limit maximum number of alignments per read. (Reads with more alignments are skipped.)

       <b>--noiseMismatches=</b>&lt;numNoiseMismatches&gt;

              Number of mismatches to be considered as noise. (default: 6)

       <b>-o</b> &lt;outFileName&gt; ,   <b>--outFile=</b>&lt;outFileName&gt;

              Name of the output file.

       <b>-P</b> &lt;procN&gt; ,   <b>--procN=</b>&lt;procN&gt;

              Maximum number of threads to be used. This provides speedup mostly  when  using  non-uniform  read
              distribution model (i.e. no <b>--uniform</b> flag). (default: 4)

       <b>-N</b> &lt;readsN&gt; ,   <b>--readsN=</b>&lt;readsN&gt;

              Total  number  of  reads.  This  is  not  necessary  if  [SB]AM  contains also reads with no valid
              alignments.

       <b>--show1warning</b>

              Show first alignments that are  considered  wrong  (TID  unknown,  TID  mismatch,  wrong  strand).
              (default: Off)

       <b>-t</b> &lt;trInfoFileName&gt; ,   <b>--trInfoFile=</b>&lt;trInfoFileName&gt;

              File to save transcript information extracted from [BS]AM file and reference.

       <b>-s</b> &lt;trSeqFileName&gt; ,   <b>--trSeqFile=</b>&lt;trSeqFileName&gt;

              Transcript sequence in FASTA format <b>---</b> for non-uniform read distribution estimation.

       <b>--trSeqHeader=</b>&lt;trSeqHeader&gt;

              Transcript  sequence  header  format  enables  gene  name extraction (standard/gencode). (default:
              standard)

       <b>--uniform</b>

              Use uniform read distribution. (default: Off)

       <b>--unstranded</b>

              Paired read are not strand specific. (default: Off)

       <b>-v</b> ,   <b>--verbose</b>

              Verbose output. (default: Off)

       <b>-V</b> ,   <b>--veryVerbose</b>

              Very verbose output. (default: Off)

parseAlignment 0.7.5+dfsg                        September 2021                                <u><a href="../man1/PARSEALIGNMENT.1.html">PARSEALIGNMENT</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>