<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>phastMotif - Predicts motifs from a set of multiple alignments.  Uses</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/phast">phast_1.7+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       phastMotif - Predicts motifs from a set of multiple alignments.  Uses

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Predicts  motifs  from a set of multiple alignments.  Uses an EM algorithm similar to that of MEME, but a
       motif is defined by phylogenetic models rather than multinomial distributions.   The  specified  multiple
       alignments  may  actually  be  single  sequences  (see  <b>-m</b>).  Various parameters control the strategy for
       initialization (see below).  Currently, the F81 substitution model is assumed.

</pre><h4><b>USAGE</b></h4><pre>
       phastMotif [-t &lt;treefile&gt;] [OPTIONS] &lt;msa_list&gt;

</pre><h4><b>OPTIONS</b></h4><pre>

       <b>-t</b> &lt;file&gt; (Required unless <b>-m</b> or <b>-p</b>) Use specified tree topology  for  all  phylogenetic  models  (Newick
              format).

       <b>-i</b> &lt;fmt&gt;
              Input format for alignment.  May be FASTA, PHYLIP, MPM, SS, or MAF (default FASTA).

       <b>-b</b> &lt;file&gt; Read background model from specified file (.mod format).

              By default, the background model is estimated in a preprocessing step, by pooling all data.

       <b>-s</b>     Estimate a separate background model for each multiple alignment.  (Not yet implemented.)

       <b>-k</b> &lt;size&gt; Learn motifs of the specified size (default is 10).

       <b>-B</b> &lt;n&gt;
              Report best &lt;n&gt; motifs (default 3).

       <b>-m</b>     MEME  mode.   Use  multinomial  rather than phylogenetic models.  Causes multiple alignments to be
              ignored <b>--</b> any gaps are discarded and all sequences are assumed independent.

       <b>-d</b> &lt;+lst&gt; Use the discriminative training method of Segal  et  al.  (RECOMB'02),  rather  than  EM.   The
              specified list

              should  contain  the  filenames  from  msa_list  that  are  to  be  considered *positive* examples
              (containing the desired motif); all others will be considered negative examples.  Can be used with
              or without <b>-m</b>.  <b>-p</b> Use "profile" models  rather  than  phylogenetic  models  (characters  in  each
              alignment  column  assumed  independent).   The  resulting model is a hybrid of the full model and
              MEME's model.  Essentially, it uses the multiple  alignments  but  not  the  phylogeny.   NOT  YET
              IMPLEMENTED.   <b>-n</b>  &lt;n&gt;  Perform  &lt;n&gt; random restarts and report the motif with highest likelihood.
              Default number is 10.  Ignored with <b>-I</b>, <b>-P</b>, and <b>-R</b> unless <b>-S</b> is specified (see below).

       <b>-I</b> &lt;mlst&gt; Run the algorithm after a "soft" initialization with

              each of the consensus sequences in the specified list.  At each position, &lt;pc&gt;  pseudocounts  (see
              <b>-c</b>)  are  given to the consensus base and 1 pseudocount to all other bases.  Each string must have
              length at most equal to the size of the motif.  If shorter, it is used as a "seed"  for  a  motif,
              with  flanking  positions  treated  as  wildcards.   <b>-P</b> &lt;x,y&gt; Initialize with the x most prevalent
              y-tuples.  A soft initialization is performed, as above.  If  y  is  less  than  the  motif  size,
              y-tuples  are used as a "seed" for a motif, as above.  <b>-R</b> &lt;x,y&gt; Initialize with a random sample of
              x y-tuples.  A soft initialization is performed, as above.  If y is  less  than  the  motif  size,
              y-tuples  are  used  as  a  "seed" for a motif, as above.  <b>-w</b> &lt;n&gt; (for use with <b>-I</b>, <b>-P</b>, <b>-R</b>) Winnow
              initialization sequences to the top &lt;n&gt; based on the unmaximized likelihood.

       <b>-c</b> &lt;pc&gt;
              (for use with <b>-I</b>, <b>-P</b>, <b>-R</b>) Number of pseudocounts for consensus bases (default  5).   <b>-S</b>  (for  use
              with  <b>-I</b>,  <b>-P</b>,  <b>-R</b>) Instead of doing a deterministic initialization based on a consensus sequence,
              sample parameters from a Dirichlet distribution defined by the pseudocounts  (see  <b>-c</b>).   In  this
              case, random restarts are performed, as specified by <b>-n</b>.

       <b>-o</b>  &lt;pref&gt;  Use  the  specified  prefix for all output files (dflt. "phastm").  <b>-H</b> Produce HTML formatted
              output, in addition to ordinary output.  One file is produced per predicted motif, as  well  as  a
              single HTML-formatted summary file.

       <b>-D</b>     Produce  a BED file with predicted motifs, for use in the UCSC browser.  Currently, sequence names
              must be formatted such as "chr10:102553847-102554897+", with  the  final  '+'  or  '-'  indicating
              strand.

       <b>-x</b>     (For use with <b>-H</b> or <b>-D</b>) Suppress ordinary output to stdout.

       <b>-h</b>     Print this help message.

phastMotif 1.4                                      May 2016                                       <u><a href="../man1/PHASTMOTIF.1.html">PHASTMOTIF</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>