<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>phastBias - Identify regions of the alignment which are affected by gBGC,</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/phast">phast_1.7+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       phastBias - Identify regions of the alignment which are affected by gBGC,

</pre><h4><b>SYNOPSIS</b></h4><pre>
       The  alignment  file can be in any of several file formats (see <b>--msa-format</b>).  The neutral model must be
       in the .mod format produced by the phyloFit program.  The foreground_branch should identify a  branch  of
       the tree (internal branches can be named with tree_doctor <b>--name-ancestors</b>).

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Identify  regions  of  the alignment which are affected by gBGC, indicated by a cluster of weak-to-strong
       (A/T -&gt; G/C) substitutions amidst a deficit of strong-to-weak substitutions on a particular branch of the
       tree.  The regions are identified by a phylo-HMM with four states: neutral, conserved, neutral with gBGC,
       and conserved with gBGC.

   <b>OUTPUT:</b>
       phastBias produces a wig file with scores for every position in the alignment indicating the  probability
       of  being in one of the gBGC states.  It can also produce gBGC tracts by thresholding this probability at
       0.5, or a matrix of probabilities for all four states.  See OUTPUT OPTIONS below.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>GENERAL</b> <b>OPTIONS:</b>

       <b>--help</b>,-h Print this help message.

       TUNING PARAMETER OPTIONS:

              gBGC PARAMETERS:

       <b>--bgc</b> &lt;B&gt;

       The B parameter describes the strength of gBGC.
              It must be &gt; 0.

              Too low of a value may yield false positives, as the gBGC model becomes indistinguishable from the
              non-gBGC model.

              Default: 3

       <b>--estimate-bgc</b> &lt;0|1&gt; Use "--estimate-bgc 1" to estimate B by maximum likelihood.  Default: 0

       <b>--bgc-exp-length</b> &lt;length&gt;

       Set the prior expected length of gBGC tracts.
              This is equivalent to

              1/alpha in the parametrization defined by Capra et al, where alpha is the rate out of gBGC states.

              Default: 1000

       <b>--estimate-bgc-exp-length</b> &lt;0|1&gt; Use "--estimate-bgc-exp-length 1" to estimate this parameter by an

              expectation-maximization algorithm.

              Default: 0

       <b>--bgc-target-coverage</b> &lt;coverage&gt;

              Set the prior for gBGC tract coverage (as a fraction between 0 and 1).

              This is represented in the model as beta/(alpha+beta), where beta is the rate into the gBGC state,
              and alpha is the rate out of the gBGC state.

              Default: 0.01

       <b>--estimate-bgc-target-coverage</b> &lt;0|1&gt;  Use  "--estimate-bgc-target-coverage  0"  to  hold  this  parameter
              constant.  Default: 1 (This is the only parameter estimated by default.)

   <b>CONSERVATION</b> <b>PARAMETERS:</b>
       Note: it is not recommended to tune these parameters with phastBias.

       Rather,  phastCons  may be used to determine the best values for rho and the transition rates into/out of
       conserved elements.  See phastCons <b>--help</b> and the phastCons  HOWTO  (available  online)  to  learn  about
       tuning these parameters.

       <b>--rho</b> &lt;rho&gt;

       Set the scaling factor for branch lengths in conserved states.
              Rho should

              be between 0 and 1.

              Default: 0.31

       <b>--cons-exp-length</b> &lt;length&gt;

       Set the prior expected length of conserved elements.
              This parameter is

              held  constant;  if  you  want to tune it, it is recommended to do this with the phastCons program
              under a non-gBGC model (see the <b>--expected-length</b> option in phastCons).  Default: 45

       <b>--cons-target-coverage</b> &lt;cov&gt;

              Set the prior for coverage of conserved elements (as a  fraction  between  0  and  1).   Like  the
              <b>--cons-exp-length</b>  above,  this  parameter  is also held constant, but can be tuned with phastCons
              (see phastCons <b>--transitions</b>).  Default: 0.3

   <b>OTHER</b> <b>PARAMETERS:</b>

       <b>--scale</b> &lt;scale&gt; Set an overall scaling factor for the branch lengths in all states.  Default: 1

       <b>--estimate-scale</b> &lt;0|1&gt;

              Rescale the branches in all states by a scaling factor determined by

              maximum likelihood (initialized by <b>--scale</b> above).  Default: 0

       <b>--eqfreqs-from-msa</b> &lt;0|1&gt;

              Reset equilibrium  frequencies  of  A,C,G,T  based  on  frequencies  observed  in  the  alignment.
              Otherwise will not be altered from input model.  Default: 1

   <b>OUTPUT</b> <b>OPTIONS</b>

       <b>--output-tracts</b> &lt;file.gff&gt;

              Print  a  GFF  file  identifying all regions with posterior probability of being in a gBGC state &gt;
              0.5.

       <b>--posteriors</b> &lt;none|wig|full&gt;

              Use this option to control posterior probability output,  which  is  written  to  stdout.   "none"
              implies  do  not  output  anything;  wig  outputs  a  standard  fixed-step  wiggle file giving the
              probability that each base is assigned to a gBGC state; "full" outputs a table with five  columns.
              The  first  column  is  the  coordinate (1-based relative to the first sequence in the alignment),
              followed by the probabilities of each of the four states: neutral, conserved, gBGC  neutral,  gBGC
              conserved.

              Default: wig

       <b>--output-mods</b> &lt;output_root&gt;

              Print    out    the    tree    models    for    all   four   states   to   &lt;output_root&gt;.cons.mod,
              &lt;output_root&gt;.neutral.mod, &lt;output_root&gt;.gBGC_cons.mod, and &lt;output_root&gt;.gBGC_neutral.mod.

       <b>--informative-fn</b>,-i &lt;file.gff&gt;

              Print a GFF containing regions of the alignment which are informative for gBGC. Note:  only  works
              properly if foreground branch is a single branch (not a group of branches).

       <b>--informative-only</b>,-o

              (To be used with <b>--informative-fn</b>). Print the informative regions, then quit.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Capra  JA, Hubisz MJ, Kostka D, Pollard KS, Siepel A: A Model-Based Analysis of GC-Biased Gene Conversion
       in the Human and Chimpanzee Genomes.  (Manuscript in submission).

phastBias 1.4                                       May 2016                                        <u><a href="../man1/PHASTBIAS.1.html">PHASTBIAS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>