<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>phyloP  -  Compute conservation or acceleration p-values based on an alignment and The phylogenetic model</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/phast">phast_1.7+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       phyloP  -  Compute conservation or acceleration p-values based on an alignment and The phylogenetic model
       must be in the .mod format produced by the phyloFit program.  The alignment file can be in any of several
       file formats (see <b>--msa-format</b>).  No alignment is required with the <b>--null</b> option.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Compute conservation or acceleration p-values based on an alignment and a  model  of  neutral  evolution.
       Will  also  compute p-values of conservation/acceleration in a subtree and in its complementary supertree
       given the whole tree (see <b>--subtree</b>).   P-values  can  be  produced  for  entire  input  alignments  (the
       default),  pre-specified  intervals  within  an  alignment  (see  <b>--features</b>),  or  individual sites (see
       <b>--wig-scores</b> and <b>--base-by-base</b>).

       The default behavior is to compute a null distribution for the total number  of  substitutions  from  the
       tree  model,  an  estimate of the number of substitutions that have actually occurred, and the p-value of
       this estimate wrt the null distribution.  These  computations  are  performed  as  described  by  Siepel,
       Pollard,  and  Haussler  (2006).   In  addition  to  the  SPH  method,  phyloP  can  compute  p-values or
       conservation/acceleration scores using a  likelihood  ratio  test  (<b>--method</b>  LRT),  a  score-based  test
       (<b>--method</b>  SCORE),  or  a  procedure  similar to that used by GERP (Cooper et al., 2005) (<b>--method</b> GERP).
       These alternative methods are currently supported only with <b>--base-by-base</b>, <b>--wig-scores</b>, or <b>--features</b>.

       The main advantage of the SPH method is  that  it  can  provide  a  complete  and  exact  description  of
       distributions  over  numbers  of substitutions.  However, simulation experiments suggest that the LRT and
       SCORE methods have somewhat better power than SPH for identifying selection, especially when the expected
       number of substitutions is small (e.g., with  short  branch  lengths  and/or  short  intervals/individual
       sites).   These  two methods are also faster.  They are generally similar to one another in power, but in
       many cases SCORE is considerably faster than LRT.  On the other hand, SCORE appears to have slightly less
       power than LRT at low false positive rates, i.e., for cases  of  extreme  selection.   Thus,  when  using
       <b>--base-by-base</b>,  <b>--wig-scores</b>,  or  <b>--features</b>, LRT is recommended for most purposes, but SCORE is a good
       alternative if speed is an issue.  When computing p-values with the SPH method, the default is to use the
       posterior expected number of substitutions as an estimate of the actual number.  This is  a  conservative
       estimate, because it is biased toward the mean of the null distribution by the prior.  These p-values can
       be made less conservative with <b>--fit-model</b> and more conservative with <b>--confidence-interval</b> (see below).

</pre><h4><b>EXAMPLE</b></h4><pre>
       1.  Using  the  SPH  method,  compute  and  report  p-values of conservation and acceleration for a given
       alignment with respect to a neutral model of evolution.  Estimated  numbers  of  substitutions  are  also
       reported.

              phyloP neutral.mod alignment.fa &gt; report.txt

       The  file  neutral.mod  could  be produced by running phyloFit on data from ancestral repeats or fourfold
       degenerate sites with an appropriate tree topology and substitution model.

       2. Compute and report p-values of conservation and acceleration for  a  particular  subtree  of  interest
       (using SPH).

              phyloP <b>--subtree</b> human-mouse_lemur neutral.mod alignment.fa &gt; report.txt

       Here human-mouse_lemur denote the most recent common ancestor of human and mouse_lemur, which is the node
       that  defines  the  primate  clade  in this phylogeny.  The tree_doctor program with the <b>--name-ancestors</b>
       option can be used to assign names to ancestral nodes of the tree.

       3. Describe the complete null distribution over the number of substitutions for a  10bp  alignment  given
       the specified neutral model (using SPH).

              phyloP <b>--null</b> 10 neutral.mod &gt; null.txt

       A  two-column  table  is  produced  with  numbers  of  substitutions  and  their  probabilities, up to an
       appropriate upper limit.

       4. Describe the complete posterior distribution over the number of substitutions  in  a  given  alignment
       (using SPH).

              phyloP <b>--posterior</b> neutral.mod alignment.fa &gt; posterior.txt

       5.  Compute  conservation  scores  (<b>-log10</b> p-values) for each site in an alignment and output them in the
       fixed-step wig format (see <a href="http://genome.ucsc.edu/goldenPath/help/wiggle.html">http://genome.ucsc.edu/goldenPath/help/wiggle.html</a>).  Use the likelihood ratio
       test (LRT) method.

              phyloP <b>--wig-scores</b> <b>--method</b> LRT neutral.mod alignment.fa &gt; scores.wig

       The <b>--mode</b> option can be used instead to produce  acceleration  scores  (ACC),  scores  of  nonneutrality
       (NNEUT),  or scores that summarize conservation and acceleration (CONACC).  The <b>--base-by-base</b> option can
       be used to output additional statistics of interest (estimated scale factors,  log10  likelihood  ratios,
       etc.).  As discussed above, several arguments to <b>--method</b> are possible.

       6. Similarly, compute scores describing lineage-specific conservation in primates.

              phyloP <b>--wig-scores</b> <b>--method</b> LRT <b>--subtree</b> human-mouse_lemur neutral.mod alignment.fa &gt; scores.wig

       7. Compute conservation p-values and associated statistics for each element in a BED file.  This time use
       a  score test and allow for acceleration as well as conservation, flagging elements under acceleration by
       making their p-values negative (CONACC mode).

              phyloP  <b>--features</b>  elements.bed  <b>--method</b>  SCORE  <b>--mode</b>  CONACC   neutral.mod   alignment.fa   &gt;
              element-scores.txt

       This  option can also be used with <b>--subtree</b>.  The <b>--gff-scores</b> option can be used to output the original
       features in GFF format with scores equal to <b>-log10</b> p.  Note that the input file can be in GFF instead  of
       BED format.

</pre><h4><b>OPTIONS</b></h4><pre>

       <b>--msa-format</b>, <b>-i</b> FASTA|PHYLIP|MPM|MAF|SS

              Alignment format (default is to guess format from file contents).

       <b>--method</b>, <b>-m</b> SPH|LRT|SCORE|GERP

              Method used to compute p-values or conservation/acceleration scores (Default SPH).  The likelihood
              ratio test (LRT) and score test (SCORE) compare an alternative model having a free scale parameter
              with  the  given  neutral  model, or, if <b>--subtree</b> is used, an alternative model having free scale
              parameters for the supertree and subtree with a null model having a single free  scale  parameter.
              P-values  are computed by comparing test statistics with asymptotic chi-square null distributions.
              The GERP-like method (GERP) estimates the number of "rejected substitutions" per base by comparing
              the (per-site) maximum likelihood expected number of substitutions with the expected number  under
              the  neutral  model.   Currently  LRT,  SCORE,  and  GERP  can  be  used only with <b>--base-by-base</b>,
              <b>--wig-scores</b>, or <b>--features</b>.

       <b>--wig-scores</b>, <b>-w</b>

              Compute separate p-values per site, and then  compute  site-specific  conservation  (acceleration)
              scores  as  <b>-log10</b>(p).   Output base-by-base scores in fixed-step wig format, using the coordinate
              system of the reference sequence (see <b>--refidx</b>).  In GERP mode, outputs rejected substitutions per
              site instead of <b>-log10</b> p-values.

       <b>--base-by-base</b>, <b>-b</b>

              Like <b>--wig-scores</b>, but outputs multiple values per site, in a method-dependent way.   With  'SPH',
              output  includes  mean  and variance of posterior distribution, with LRT and SCORE it includes the
              estimated scale factor(s) and test statistics, and with GERP it includes the estimated numbers  of
              neutral,  observed, and rejected substitutions, along with the number of species available at each
              site.

       <b>--refidx</b>, <b>-r</b> &lt;refseq_idx&gt;

              (for use with <b>--wig-scores</b> or <b>--base-by-base</b>)  Use  coordinate  frame  of  specified  sequence  in
              output.   Default  value is 1, first sequence in alignment; 0 indicates coordinate frame of entire
              multiple alignment.

       <b>--mode</b>, <b>-o</b> CON|ACC|NNEUT|CONACC

              (For use with <b>--wig-scores</b>, <b>--base-by-base</b>, or <b>--features</b>) Whether to compute  one-sided  p-values
              so  that  small  p  (large  <b>-log10</b>  p)  indicates  unexpected  conservation  (CON; the default) or
              acceleration (ACC); or two-sided p-values such that small p indicates an unexpected departure from
              neutrality (NNEUT).  The fourth option (CONACC) uses  positive  values  (p-values  or  scores)  to
              indicate  conservation and negative values to indicate acceleration.  In GERP mode, CON and CONACC
              both report the number of rejected substitutions R (which may be negative), while ACC reports  <b>-R</b>,
              and NNEUT reports abs(R).

       <b>--features</b>, <b>-f</b> &lt;file&gt;

              Read  features  from  &lt;file&gt;  (GFF  or  BED  format)  and  output  a table of p-values and related
              statistics with one row per feature.  The features are assumed to use the coordinate frame of  the
              first  sequence in the alignment.  Not for use with <b>--null</b> or <b>--posterior</b>.  See also <b>--gff-scores</b>.

       <b>--gff-scores</b>, <b>-g</b>

              (For use with features) Instead of a table, output a GFF and assign each feature a score equal  to
              its <b>-log10</b> p-value.

       <b>--subtree</b>, <b>-s</b> &lt;node-name&gt;

              (Not  available  in  GERP mode) Partition the tree into the subtree beneath the node whose name is
              given and the complementary supertree, and consider conservation/acceleration in the subtree given
              the supertree.  The branch above the specified node is included with the subtree.  Thus, given the
              tree "((human,chimp)primate,(mouse,rat)rodent)", the option "--subtree primate"  will  create  one
              partition  consisting  of  human,  chimp,  and  the  branch leading to them, and another partition
              consisting of the rest of the tree; "--subtree human" will create one partition consisting only of
              human and the branch leading to it and another partition consisting of the rest of the  tree.   In
              'SPH' mode, a reversible substitution model is assumed.

       <b>--branch</b>, <b>-B</b> &lt;node-name(s)&gt;

              (Not  available in GERP or SPH mode).  Like subtree, but partitions the tree into the set of named
              branches (each named by its  child  node),  and  all  the  remaining  branches.   Then  tests  for
              conservation/ acceleration in the set of named branches relative to the others.  The argument is a
              comma-delimited list of child nodes.

       <b>--chrom</b>, <b>-N</b> &lt;name&gt;

              (Optionally  use  with <b>--wig-scores</b> or <b>--base-by-base</b>) Chromosome name for wig output.  Default is
              root of multiple alignment filename.

       <b>--log</b>, <b>-l</b> &lt;fname&gt;

              Write log to  &lt;fname&gt;  describing  details  of  parameter  optimization.   Useful  for  debugging.
              (Warning: may produce large file.)

       <b>--seed</b>, <b>-d</b> &lt;seed&gt;

              Provide  a random number seed, should be an integer &gt;=1.  Random numbers are used in some cases to
              generate starting values for optimization.  If not specified will use a seed based on the  current
              time.

       <b>--no-prune</b>,-P

              Do  not prune species from tree which are not in alignment.  Rather, treat these species as having
              missing data in the alignment.  Missing data does have an effect on the results when <b>--method</b>  SPH
              is used.

       <b>--help</b>, <b>-h</b>

              Produce this help message.

   <b>Options</b> <b>for</b> <b>SPH</b> <b>mode</b> <b>only</b>

       <b>--null</b>, <b>-n</b> &lt;nsites&gt; Compute just the null (prior) distribution of the number of substitutions, as defined
              by  the tree model and the given number of sites, and output as a table.  The 'alignment' argument
              will be ignored.  If used with <b>--subtree</b>, the joint distribution over the number of  substitutions
              in the specified supertree and subtree will be output instead.

       <b>--posterior</b>,  <b>-p</b>  Compute  just  the  posterior  distribution  of  the number of substitutions, given the
              alignment and the model, and output as a table.  If used with <b>--subtree</b>,  the  joint  distribution
              over the number of substitutions in the specified supertree and subtree will be output instead.

       <b>--fit-model</b>, <b>-F</b>

              Fit  model  to  data before computing posterior distribution, by estimating a scale factor for the
              whole tree or (if <b>--subtree</b>) separate scale factors  for  the  specified  subtree  and  supertree.
              Makes  p-values  less conservative.  This option has no effect with <b>--null</b> and currently cannot be
              used with <b>--features</b>.  It can be used with <b>--wig-scores</b> and <b>--base-by-base</b>.

       <b>--epsilon</b>, <b>-e</b> &lt;val&gt;

              (Default 1e-10 or 1e-6 if <b>--wig-scores</b> or <b>--base-by-base</b>) Threshold used in  truncating  tails  of
              distributions;  tail  probabilities  less than this value are discarded.  To get accurate p-values
              smaller than 1e-10, this option will need to be used, at some cost in speed.  Note that truncation
              affects only *right* tails, not left tails, so it  should  be  an  issue  only  with  p-values  of
              acceleration.

       <b>--confidence-interval</b>, <b>-c</b> &lt;val&gt;

              Allow  for  uncertainty in the estimate of the actual number of substitutions by using a (central)
              confidence interval about the mean of the specified size (0 &lt; val &lt; 1).  To be  conservative,  the
              maximum of this interval is used when computing a p-value of conservation, and the minimum is used
              when  computing a p-value of acceleration.  The variance of the posterior is computed exactly, but
              the confidence interval is based  on  the  assumption  that  the  combined  distribution  will  be
              approximately normal (true for large numbers of sites by central limit theorem).

       <b>--quantiles</b>, <b>-q</b>

              (For  use  with  <b>--null</b>  or  <b>--posterior</b>)  Report  quantiles  of  distribution  rather  than whole
              distribution.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Cooper GM, Stone EA, Asimenos G, NISC Comparative Sequencing Program, Green ED,  Batzoglou  S,  Sidow  A.
       Distribution and intensity of constraint in mammalian genomic sequence.  Genome Res. 2005 <a href="../man7/15.7.html">15</a>(7):901-13.

       Siepel  A,  Pollard  KS,  and  Haussler  D.  New  methods  for  detecting  lineage-specific selection. In
       Proceedings of the 10th International Conference on Research in Computational Molecular  Biology  (RECOMB
       2006), pp. 190-205.

phyloP 1.4                                          May 2016                                           <u><a href="../man1/PHYLOP.1.html">PHYLOP</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>