<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>poa - align a set of sequences or alignments.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/poa">poa_2.0+20060928-10_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       poa - align a set of sequences or alignments.

</pre><h4><b>SYNOPSIS</b></h4><pre>

       <b>poa</b> [OPTIONS] [<u>MATRIXFILE</u>]

       One of the <b>-read_fasta</b>, <b>-read_msa</b>, or <b>-read_msa_list</b> arguments must be used, since a sequence or
       alignment file is required.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>POA</b> is Partial Order Alignment, a fast program for multiple sequence alignment (MSA) in bioinformatics.
       Its advantages are speed, scalability, sensitivity, and the superior ability to handle branching / indels
       in the alignment. Partial order alignment is an approach to MSA, which can be combined with existing
       methods such as progressive alignment. POA optimally aligns a pair of MSAs and which therefore can be
       applied directly to progressive alignment methods such as CLUSTAL. For large alignments, Progressive POA
       is 10 to 30 times faster than CLUSTALW.

</pre><h4><b>EXAMPLES</b></h4><pre>
       <b>poa</b> <b>-read_fasta</b> <u>multidom.seq</u> <b>-clustal</b> <u>m.aln</u> <u>blosum80.mat</u>

       On Debian systems, <b>poa</b> can be tested using the following command:

       <b>poa</b> <b>-read_fasta</b> <u>/usr/share/doc/poa/examples/multidom.seq</u> <b>-clustal</b> <u>/dev/stdout</u> <b>-v</b>
       <u>/usr/share/poa/blosum80.mat</u>

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>INPUT</b>
       <b>-read_fasta</b> <u>FILE</u>
           Read in FASTA sequence file.

       <b>-read_msa</b> <u>FILE</u>
           Read in MSA alignment file.

       <b>-read_msa2</b> <u>FILE</u>
           Read in second MSA file.

       <b>-subset</b> <u>FILE</u>
           Filter MSA to include list of seqs in file.

       <b>-subset2</b> <u>FILE</u>
           Filter second MSA to include list of seqs in file.

       <b>-remove</b> <u>FILE</u>
           Filter MSA to include list of seqs in file.

       <b>-remove2</b> <u>FILE</u>
           Filter second MSA to include list of seqs in file.

       <b>-read_msa_list</b> <u>FILE</u>
           Read an MSA from each filename listed in file.

       <b>-tolower</b>
           Force FASTA/MSA sequences to lowercase (nucleotides in our matrix files).

       <b>-toupper</b>
           Force FASTA/MSA sequences to UPPERCASE (amino acids in our matrix files).

   <b>ALIGNMENT</b>
       <b>-do_global</b>
           Do global alignment.

       <b>-do_progressive</b>
           Perform progressive alignment using a guide tree built by neighbor joining from a set of
           sequence-sequence similarity scores.

       <b>-read_pairscores</b> <u>FILE</u>
           Read tab-delimited file of similarity scores (If not provided, scores are constructed using pairwise
           sequence alignment.)

       <b>-fuse_all</b>
           Fuse identical letters on align rings.

   <b>ANALYSIS</b>
       <b>-hb</b>
           Perform heaviest bundling to generate consensi.

       <b>-hbmin</b> <u>VALUE</u>
           Include in heaviest bundle sequences with percent ID (as a fraction) &gt;= <u>VALUE</u>.

   <b>OUTPUT</b>
       <b>-pir</b> <u>FILE</u>
           Write out MSA in PIR format.

       <b>-clustal</b> <u>FILE</u>
           Write out MSA in CLUSTAL format.

       <b>-po</b> <u>FILE</u>
           Write out MSA in PO format.

       <b>-preserve_seqorder</b>
           Write out MSA with sequences in their input order.

       <b>-printmatrix</b> <u>LETTERS</u>
           Print score matrix to stdout.

       <b>-best</b>
           Restrict MSA output to heaviest bundles (PIR only).

       <b>-v</b>
           Run in verbose mode (e.g. output gap penalties).

</pre><h4><b>REFERENCE</b></h4><pre>
       Please cite Grasso C, Lee C. (2004) Combining partial order alignment and progressive multiple sequence
       alignment increases alignment speed and scalability to very large alignment problems. Bioinformatics.
       2004 Jul 10;<a href="../man10/20.10.html">20</a>(10):1546-56. Epub 2004 Feb 12.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       The homepage of POA is <u><a href="http://www.bioinformatics.ucla.edu/poa">http://www.bioinformatics.ucla.edu/poa</a></u>

</pre><h4><b>COPYRIGHTS</b></h4><pre>
       Copyright (C) 2001, 2006 Christopher Lee &lt;<a href="mailto:leec@mbi.ucla.edu">leec@mbi.ucla.edu</a>&gt;. POA is free software. You can redistribute
       it and/or modify it under the terms of the GNU General Public License as published by the Free Software
       Foundation.

       This manual page was written by Charles Plessy &lt;<a href="mailto:charles-debian-nospam@plessy.org">charles-debian-nospam@plessy.org</a>&gt; for the Debian(TM)
       system (but may be used by others). Permission is granted to copy, distribute and/or modify this document
       under the terms of the GNU General Public License, Version 2 any later version published by the Free
       Software Foundation.

       On Debian systems, the complete text of the GNU General Public License can be found in
       /usr/share/common-licenses/GPL.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright Â© 2006 Charles Plessy

                                               september 26, 2006                                         <u><a href="../man1/POA.1.html">POA</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>