<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>probcons - align multiple protein sequences and print to standard output</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/probcons">probcons_1.12-14_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       probcons - align multiple protein sequences and print to standard output

</pre><h4><b>SYNOPSIS</b></h4><pre>

       <b>probcons</b> [<u>OPTION</u>] [<u>MFAFILE</u>] [<u>MFAFILE</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>probcons</b> is a tool for generating multiple alignments of protein sequences. Using a combination of
       probabilistic modeling and consistency-based alignment techniques, PROBCONS has achieved the highest
       accuracies of all alignment methods to date. On the BAliBASE benchmark alignment database, alignments
       produced by PROBCONS show statistically significant improvement over current programs, containing an
       average of 7% more correctly aligned columns than those of T-Coffee, 11% more correctly aligned columns
       than those of CLUSTAL W, and 14% more correctly aligned columns than those of DIALIGN.

       <b>probcons</b> aligns sequences provided in MFA format. This format consists of multiple sequences. Each
       sequence in MFA format begins with a single-line description, followed by lines of sequence data. The
       description line is distinguished from the sequence data by a greater-than (“&gt;”) symbol in the first
       column.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-clustalw</b>
           use CLUSTALW output format instead of MFA

       <b>-c</b> <b>--consistency</b> <u>REPS</u>
           use 0 &lt;= <u>REPS</u> &lt;= 5 (default: 2) passes of consistency transformation

       <b>-ir</b> <b>--iterative-refinement</b> <u>REPS</u>
           use 0 &lt;= <u>REPS</u> &lt;=1000 (default: 100) passes of iterative-refinement

       <b>-pre</b> <b>--pre-training</b> <u>REPS</u>
           use 0 &lt;= <u>REPS</u> &lt;= 20 (default: 0) rounds of pretraining

       <b>-pairs</b>
           generate all-pairs pairwise alignments

       <b>-viterbi</b>
           use Viterbi algorithm to generate all pairs (automatically enables <b>-pairs</b>)

       <b>-v</b> <b>--verbose</b>
           Report progress while aligning (default: off)

       <b>-annot</b> <u>FILENAME</u>
           write annotation for multiple alignment to <u>FILENAME</u>

       <b>-t</b> <b>--train</b> <u>FILENAME</u>
           compute EM transition probabilities, store in <u>FILENAME</u> (default: no training)

       <b>-e</b> <b>--emissions</b>
           also reestimate emission probabilities (default: off)

       <b>-p</b> <b>--paramfile</b> <u>FILENAME</u>
           read parameters from <u>FILENAME</u> (default: )

       <b>-a</b> <b>--alignment-order</b>
           print sequences in alignment order rather than input order (default: off)

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       •   You can find more information in the manual of ProbCons, which is located in
           <u>/usr/share/doc/probcons/manual.pdf</u> in Debian systems.

       •   An experimental version of ProbCons compiled with parameters estimated via unsupervised training on
           BRAliBASE, <b><a href="../man1/probcons-RNA.1.html">probcons-RNA</a></b>(1), is also distributed in the ProbCons Debian package.

       •    <b><a href="../man1/pc-makegnufile.1.html">pc-makegnufile</a></b>(1), <b><a href="../man1/pc-compare.1.html">pc-compare</a></b>(1), <b><a href="../man1/pc-project.1.html">pc-project</a></b>(1), which are distributed separately in the
           probcons-extra package.

</pre><h4><b>REFERENCE</b></h4><pre>
       Please cite Do, C.B., Mahabhashyam, M.S.P., Brudno, M., and Batzoglou, S. 2005. PROBCONS: Probabilistic
       Consistency-based Multiple Sequence Alignment. Genome Research 15: 330-340.

</pre><h4><b>AUTHORS</b></h4><pre>
       <b>Chuong</b> <b>Do</b> &lt;<a href="mailto:chuongdo@cs.stanford.edu">chuongdo@cs.stanford.edu</a>&gt;
           Wrote  probcons  in  collaboration  with  Michael  Brudno in the research group of Serafim Batzoglou,
           Department of Computer Science, Stanford University.

       <b>Charles</b> <b>Plessy</b> &lt;<a href="mailto:charles-debian-nospam@plessy.org">charles-debian-nospam@plessy.org</a>&gt;
           Wrote this manpage in DocBook XML for the Debian distribution.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       This program and its manpage are in the public domain.

probcons 1.12                                      2007-04-04                                        <u><a href="../man1/PROBCONS.1.html">PROBCONS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>