<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>samtools-tview - display alignments in a curses-based interactive viewer.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/samtools">samtools_1.21-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       samtools-tview - display alignments in a curses-based interactive viewer.

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>samtools</b> <b>tview</b> [<b>-p</b> <u>chr:pos</u>] [<b>-s</b> <u>STR</u>] [<b>-d</b> <u>display</u>] <u>in.sorted.bam</u> [<u>ref.fasta</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Text  alignment viewer (based on the ncurses library). In the viewer, press `?' for help and press `g' to
       check the alignment start from a region in the  format  like  `chr10:10,000,000'  or  `=10,000,000'  when
       viewing the same reference sequence.

       The top line shows the reference sequence, or '<b>N</b>'s if unknown.  Underneath this is the consensus, derived
       from  the  sequence alignments.  Below the consensus the sequence alignment records are shown.  Uppercase
       and lowercase is used to distinguish the sequence strand, with uppercase being the top/forward strand.

       When the reference is known, both consensus and alignment  record  sequences  are  displayed  in  a  dot-
       notation  where  a  matching character is shown as '<b>.</b>'  (forward strand) or '<b>,</b>' (reverse strand) and only
       mismatching bases and missing bases are shown.  This mode can be toggled with the "."  command.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-d</b> <u>display</u>    Output as (H)tml, (C)urses or (T)ext.

                     The width of generated text is controlled by the COLUMNS environment  variable  or  the  <b>-w</b>
                     option  for  non-curses  outputs.   Note  this may be a local shell variable so it may need
                     exporting first or specifying on the command line prior to the command.  For example

                       export COLUMNS ; samtools tview -d T -p 1:234567 in.bam

       <b>-p</b> <u>chr:pos</u>    Go directly to this position

       <b>-s</b> <u>STR</u>        Display only alignments from this sample or read group.  <b>STR</b> must match either an <b>ID</b> or  <b>SM</b>
                     field in an <b>@RG</b> header record.  For example

                       samtools tview -p chr20:10M -s NA12878 grch38.fa

       <b>-w</b> <u>INT</u>        Specifies the display width when using the HTML or Text output modes.

       <b>-X</b>            If  this option is set, it will allows user to specify customized index file location(s) if
                     the data folder does not contain any index file. Example usage: samtools tview [options] -X
                     &lt;/data_folder/data.bam&gt; [/index_folder/index.bai] [ref.fasta]

</pre><h4><b>AUTHOR</b></h4><pre>
       Written by Heng Li from the Sanger Institute.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/samtools.1.html">samtools</a></u>(1)

       Samtools website: &lt;<a href="http://www.htslib.org/">http://www.htslib.org/</a>&gt;

samtools-1.21                                   12 September 2024                              <u><a href="../man1/samtools-tview.1.html">samtools-tview</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>