<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>2ft - 2-frame translation of a DNA sequence into protein</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/pftools">pftools_3.2.12-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       2ft - 2-frame translation of a DNA sequence into protein

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>2ft</b>       [ <b>-h</b> ] [ <b>-r</b> | <b>-s</b> ] [ <b>-W</b> <u>width</u> ] <b>&lt;</b> <u>sequence_library</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>2ft</b>  converts  both strands of one or several DNA sequences into so-called "<u>interleaved</u> <u>frame-translated</u>"
       DNA sequences.  An "<u>interleaved</u> <u>frame-translated</u>" DNA sequence is an amino acid  sequence  of  translated
       overlapping  codons.  For instance, the DNA sequence "ATGGCA" consisting of four overlapping codons would
       be translated into the amino acid sequence "MWGA", representing the translation of the 1st codon  of  the
       1st reading frame, the 1st codon of the 2nd reading frame, the 1st codon of the 3rd reading frame and the
       2nd  codon  of the 1st reading frame.  Stop codons are represented by the character "<b>O</b>".  The program <b>2ft</b>
       reads fasta formatted DNA sequences from standard input. If the sequences are contained in  a  file,  the
       input  can  be  redirected from this file using the shell's input redirector '<b>&lt;</b>' as shown in the <u>synopsis</u>
       section.  The <u>sequence_library</u> contains one or several sequences in Pearson/Fasta format.

       This program is used for searching genomic or EST sequence databases with frame-search profiles generated
       by pftools program <b><a href="../man1/ptof.1.html">ptof</a></b>(1).

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>     Display usage help text.

       <b>-r</b>     Translate only reverse (antisense) strand.
              This option is mutualy exclusive with option <b>-s</b>.
              Default: translate both strands

       <b>-s</b>     Translate only sense strand.
              This option is mutualy exclusive with option <b>-r</b>.
              Default: translate both strands

       <b>-W</b> <u>width</u>
              Set sequence output width.
              The value of '<u>width</u>' specifies how many residues will be output on one line.
              Type: integer
              Default: 60

</pre><h4><b>EXAMPLES</b></h4><pre>
       <u>See</u> manual pages of pftools program <b><a href="../man1/ptof.1.html">ptof</a></b>(1).

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/ptof.1.html">ptof</a></b>(1)

</pre><h4><b>AUTHOR</b></h4><pre>
       The <b>pftools</b> package was developed by Philipp Bucher.
       Any comments or suggestions should be addressed to &lt;<a href="mailto:pftools@sib.swiss">pftools@sib.swiss</a>&gt;.

pftools 2.3                                         May 2003                                              <u><a href="../man1/2FT.1.html">2FT</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>