<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>AnalyseSeqs - Analyse a set of sequences of common length</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/vienna-rna">vienna-rna_2.6.4+dfsg-1build2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       AnalyseSeqs - Analyse a set of sequences of common length

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>AnalyseSeqs</b> <b>[-X[</b><u>bswn</u><b>]]</b> <b>[-Q]</b> <b>[-M{mask}[+|!]]</b> <b>[-D{H|A|G}]</b> <b>[-d{S|H|D|B}]</b>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <u>AnalyseSeqs</u>  reads  a set of sequences from stdin and tries a variety of methods for sequence analysis on
       them. Currently available are:
       Statistical geometry for quadruples of sequences; THIS IS PRELIMINARY AND NOT WELL TESTED BY NOW.
       split decomposition; neighbour joining and Ward's variance method for  reconstructing  phylogenies  using
       various  distance  measures.   For  statistical  geometry  and  the  cluster methods PostScript output is
       available.
       The program continues reading until it encounters one of  the  separator  characters  '@'  or  '%'.  Only
       sequences  of  alphabetical  characters  or  of  a  specified alphabet are processed, all other lines are
       ignored. The program stops reading if it either encounters an EOF condition, or if  there  are  no  valid
       sequence data between two lines beginning with separator characters.
       A  list  of  taxa  names can be specified in the input stream. The list begins with a line beginning with
       '*'. Optionally, a file name prefix [fn] for the PostScript output can be specified in  this  line.   The
       entries  have  the  form 'x : Taxon', where x is the number of taxon, i.e., of the corresponding entry in
       the list of input sequences. The taxa list need not be complete.  It  must  end,  however,  with  a  line
       beginning with '*' or any of the separator characters. The taxa list is printed on top of the output. The
       specified taxa names are used as labels in the PostScript output.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-X[bswn]</b>
              specifies the analysis methods to be used.

       <b>[b]</b>    Statistical  Geometry.  A PostScript file named '[fn_]box.ps' giving a graphical representation of
              the statistical geometry is created. The resulting box is a good measure of 'tree likeness' of the
              data set.  This is the default.

       <b>[s]</b>    Split decomposition.

       <b>[w]</b>    Cluster analysis  using  Ward's  method.  A  PostScript  file  named  '[fn_]wards.ps'  is  created
              containing a drawing of the tree.

       <b>[n]</b>    Cluster  analysis using Saitou's neighbour joining method. A PostScript file named '[fn_]nj.ps' is
              created containing a drawing of the tree.

       <b>-Q</b>     indicates that a statistical geometry analysis is to be performed comparing four  data  sets,  for
              instance  to  confirm  the  significance  of  a proposed phylogeny. This option is only useful for
              statistical geometry analysis and hence the -X option is ignored. Each of the four data sets  must
              be of the form
              * [filename_prefix]
              # number
              [list of taxa names]
              *
              list of sequences
              %
              where number is 1,2,3,4 for the four groups to be compared.

       <b>-M{mask}[+|!]</b>
              allows  one  to  specify  a  mask for the input file. '{mask}' can be one of the following letters
              indicating a predefined alphabet or the %-sign followed by all characters to be accepted. A + sign
              at the very end of the mask indicates that the input is to be handled case sensitive.  Default  is
              conversion  of the input to upper case. A ! sign can be used to convert the input data to RY code:
              GgAaXx -&gt; R, UuCcKkTt -&gt; Y, all other letters are converted to *.

       <b>-Ma</b>    all letters A-Z and a-z.

       <b>-Mu</b>    uppercase letters.

       <b>-Ml</b>    lowercase letters.

       <b>-Mc</b>    digits [0-9].

       <b>-Mn</b>    all alphanumeric characters.

       <b>-MR</b>    RNA alphabet (GCAUgcau).

       <b>-MD</b>    DNA alphabet (GCATgcat).

       <b>-MA</b>    Amino acids in one-letter code.

       <b>-MS</b>    Secondary strcutures coded as '^.()'

       <b>-M%alphabet</b>
              use the specified alphabet.

       <b>-D</b>     specifies the algorithm to be used for calculating the distance matrix  of  the  input  data  set.
              Available are

       <b>-DH</b>    Hamming Distance

       <b>-DA[,cost]</b>
              Simple  alignment distance according to Needleman and Wunsch.  A gap cost different from 1. can be
              specified after the comma.

       <b>-DG[,cost1,cost2]</b>
              Gotoh's distance with  gap  cost  function  g(k)  =  cost2+cost1*(k-1).  cost2&lt;=cost1  has  to  be
              fulfilled.  Default values are cost1=1., cost2=1., yielding the same distance as option A.
              ONLY THE HAMMING DISTANCE IS WELL TESTED BY NOW !!!

       <b>-d</b>     specifies the edit cost matrix to be used. Available are

       <b>-dS</b>    simple  distance.  Indel  and substitution of different characters all have cost 1. The indel cost
              can be set by specifying the gap costs with the  algorithm  options  -DA  and  -DG.  This  is  the
              default.

       <b>-dH</b>    A  distance  matrix  for  RNA  secondary  structures.  Inspired  by  Hogeweg's  similarity measure
              (J.Mol.Biol 1988).  Gap-function is set automatically.

       <b>-dD</b>    Dayhoff's matrix for amino acid distances.

       <b>-dB</b>    Distinguish purines and pyrimidines only.  CAUTION this  option  of  course  influences  only  the
              calculation  of  distances.   It  does NOT affect computation of the statistical geometry. This is
              done directly on the sequences. If you want to do statistical geometry on RY sequences use  the  !
              sign with the -M option, for instance -MR!.

</pre><h4><b>REFERENCES</b></h4><pre>
       The  method  of  statistical  geometry  has been introduced by M. Eigen, R. Winkler-Oswatitsch and A.W.M.
       Dress (Proc Natl Acad Sci, 85:1988,5912).  The method of split decomposition was proposed by H.J. Bandelt
       and A.W.M. Dress (Adv Math, 92:1992,47).  The variance method for cluster analysis is due to H.J. Ward (J
       Amer Stat Ass, 58:1963,236).  The neighbour joining method was published by  Saitou  and  Nei  (Mol  Biol
       Evol, 4:1987,406).

       This program is part of the Vienna RNA Package

</pre><h4><b>WARNING</b></h4><pre>
       This is the beta test version. Some options or combinations of options may still produce nonsense. Please
       send bug reports to <a href="mailto:ivo@tbi.univie.ac.at">ivo@tbi.univie.ac.at</a>.

</pre><h4><b>VERSION</b></h4><pre>
       This man page is part of the Vienna RNA Package version 1.2.

</pre><h4><b>AUTHOR</b></h4><pre>
       Peter F Stadler, Ivo L. Hofacker.

</pre><h4><b>BUGS</b></h4><pre>
       Comments should be sent to <a href="mailto:ivo@itc.univie.ac.at">ivo@itc.univie.ac.at</a>.

                                                                                                  <u><a href="../man1/ANALYSESEQS.1.html">ANALYSESEQS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>