<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>bp_flanks - finding flanking sequences for a variant in a sequence position</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/libbio-variation-perl">libbio-variation-perl_1.7.5-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       bp_flanks - finding flanking sequences for a variant in a sequence position

</pre><h4><b>SYNOPSIS</b></h4><pre>
         bp_flanks --position POS [-p POS ...]  [--flanklen INT]
                accession | filename

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This script allows you to extract a subsequence around a region of interest from an existing sequence.
       The output if fasta formatted sequence entry where the header line contains additional information about
       the location.

</pre><h4><b>OPTIONS</b></h4><pre>
       The script takes one unnamed argument which be either a file name in the local file system or a
       nucleotide sequence accession number.

         -p         Position uses simple nucleotide sequence feature table
         --position notation to define the region of interest, typically a
                    SNP or microsatellite repeat around which the flanks are
                    defined.

                    There can be more than one position option or you can
                    give a comma separated list to one position option.

                    The format of a position is:

                        [id:] int | range | in-between [-]

                    The optional id is the name you want to call the new
                    sequence. If it not given in joins running number to the
                    entry name with an underscore.

                    The position is either a point (e.g. 234), a range (e.g
                    250..300) or insertion point between nucleotides
                    (e.g. 234^235)

                    If the position is not completely within the source
                    sequence the output sequence will be truncated and it
                    will print a warning.

                    The optional hyphen [-] at the end of the position
                    indicates that that you want the retrieved sequence to be
                    in the opposite strand.

         -f         Defaults to 100. This is the length of the nucleotides
         --flanklen sequence retrieved on both sides of the given position.

                    If the source file does not contain

</pre><h4><b>OUTPUT</b> <b>FORMAT</b></h4><pre>
       The output is a fasta formatted entry where the description file contains tag=value pairs for information
       about where in the original sequence the subsequence was taken.

       The ID of the fasta entry is the name given at the command line joined by hyphen to the filename or
       accesion of the source sequence. If no id is given a series of consecutive integers is used.

       The tag=value pairs are:

       oripos=int
          position in the source file

       strand=1|-1
          strand of the sequence compared to the source sequence

       allelepos=int
          position of the region of interest in the current entry.  The tag is the same as used by dbSNP@NCBI

       The  sequence highlights the allele variant position by showing it in upper case and rest of the sequence
       in lower case characters.

</pre><h4><b>EXAMPLE</b></h4><pre>
         % bp_flanks <a href="file:~/seq/ar.embl">~/seq/ar.embl</a>

         &gt;1_/HOME/HEIKKI/SEQ/AR.EMBL oripos=100 strand=1 allelepos=100
         taataactcagttcttatttgcacctacttcagtggacactgaatttggaaggtggagga
         ttttgtttttttcttttaagatctgggcatcttttgaatCtacccttcaagtattaagag
         acagactgtgagcctagcagggcagatcttgtccaccgtgtgtcttcttctgcacgagac
         tttgaggctgtcagagcgct

</pre><h4><b>TODO</b></h4><pre>
       The input files are assumed to be in EMBL format and the  sequences  are  retrieved  only  from  the  EMB
       database. Make this more generic and use the registry.

       head1 FEEDBACK

   <b>Mailing</b> <b>Lists</b>
       User  feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing lists  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.   Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Heikki</b> <b>Lehvaslaiho</b></h4><pre>
       Email:  &lt;heikki-at-bioperl-dot-org&gt;

perl v5.32.1                                       2021-12-02                                      <u><a href="../man1p/BP_FLANKS.1p.html">BP_FLANKS</a></u>(1p)
</pre>
 </div>
</div></section>
</div>
</body>
</html>