<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>bp_mask_by_search - mask sequence(s) based on its alignment results</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/bioperl">bioperl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       bp_mask_by_search - mask sequence(s) based on its alignment results

</pre><h4><b>SYNOPSIS</b></h4><pre>
         bp_mask_by_search.pl -f blast genomefile blastfile.bls &gt; maskedgenome.fa

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Mask sequence based on significant alignments of another sequence.  You need to provide the report file
       and the entire sequence data which you want to mask.  By default this will assume you have done a TBLASTN
       (or TFASTY) and try and mask the hit sequence assuming you've provided the sequence file for the hit
       database.  If you would like to do the reverse and mask the query sequence specify the -t/--type query
       flag.

       This is going to read in the whole sequence file into memory so for large genomes this may fall over.
       I'm using DB_File to prevent keeping everything in memory, one solution is to split the genome into
       pieces (BEFORE you run the DB search though, you want to use the exact file you BLASTed with as input to
       this program).

       Below the double dash (--) options are of the form --format=fasta or --format fasta or you can just say
       -f fasta

       By -f/--format I mean either are acceptable options.  The =s or =n or =c specify these arguments expect a
       'string'

       Options:
           -f/--format=s    Search report format (fasta,blast,axt,hmmer,etc)
           -sf/--sformat=s  Sequence format (fasta,genbank,embl,swissprot)
           --hardmask       (booelean) Hard mask the sequence
                            with the maskchar [default is lowercase mask]
           --maskchar=c     Character to mask with [default is N], change
                            to 'X' for protein sequences
           -e/--evalue=n    Evalue cutoff for HSPs and Hits, only
                            mask sequence if alignment has specified evalue
                            or better
           -o/--out/
           --outfile=file   Output file to save the masked sequence to.
           -t/--type=s      Alignment seq type you want to mask, the
                            'hit' or the 'query' sequence. [default is 'hit']
           --minlen=n       Minimum length of an HSP for it to be used
                            in masking [default 0]
           -h/--help        See this help information

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Jason</b> <b>Stajich</b></h4><pre>
       Jason Stajich, jason-at-bioperl-dot-org.

perl v5.32.1                                       2021-08-15                              <u><a href="../man1p/BP_MASK_BY_SEARCH.1p.html">BP_MASK_BY_SEARCH</a></u>(1p)
</pre>
 </div>
</div></section>
</div>
</body>
</html>