<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>cd-hit-454 - quickly group sequences, optimised for 454 data</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/cd-hit">cd-hit_4.8.1-4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       cd-hit-454 - quickly group sequences, optimised for 454 data

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>cd-hit-454</b> [<u>Options</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
              ====== CD-HIT version 4.8.1 (built on Aug 20 2021) ======

       Options

       <b>-i</b>     input filename in fasta format, required, can be in .gz format

       <b>-o</b>     output filename, required

       <b>-c</b>     sequence  identity  threshold,  default  0.98 this is a "global sequence identity" calculated as :
              number of identical amino acids or bases in alignment divided by the full length  of  the  shorter
              sequence + gaps

       <b>-b</b>     band_width of alignment, default 10

       <b>-M</b>     memory limit (in MB) for the program, default 800; 0 for unlimitted;

       <b>-T</b>     number of threads, default 1; with 0, all CPUs will be used

       <b>-n</b>     word_length, default 10, see user's guide for choosing it

       <b>-aL</b>    alignment  coverage  for the longer sequence, default 0.0 if set to 0.9, the alignment must covers
              90% of the sequence

       <b>-AL</b>    alignment coverage control for the longer sequence, default 99999999 if set to 60, and the  length
              of the sequence is 400, then the alignment must be &gt;= 340 (400-60) residues

       <b>-aS</b>    alignment  coverage for the shorter sequence, default 0.0 if set to 0.9, the alignment must covers
              90% of the sequence

       <b>-AS</b>    alignment coverage control for the shorter sequence, default 99999999 if set to 60, and the length
              of the sequence is 400, then the alignment must be &gt;= 340 (400-60) residues

       <b>-B</b>     1 or 0, default 0, by default, sequences are stored in RAM if set to 1,  sequence  are  stored  on
              hard drive !! No longer supported !!

       <b>-g</b>     1 or 0, default 0 by cd-hit's default algorithm, a sequence is clustered to the first cluster that
              meet  the threshold (fast cluster). If set to 1, the program will cluster it into the most similar
              cluster that meet the threshold (accurate but slow mode) but  either  1  or  0  won't  change  the
              representatives of final clusters

       <b>-D</b>     max size per indel, default 1

       <b>-match</b> matching score, default 2

       <b>-mismatch</b>
              mismatching score, default <b>-1</b>

       <b>-gap</b> gap opening score, default <b>-3</b>

       <b>-gap-ext</b>
              gap extension score, default <b>-1</b>

       <b>-bak</b> write backup cluster file (1 or 0, default 0)

       <b>-h</b>     print this help

              Questions, bugs, contact Weizhong Li at <a href="mailto:liwz@sdsc.edu">liwz@sdsc.edu</a>

              If you find cd-hit useful, please kindly cite:

              "CD-HIT:  a  fast  program  for  clustering  and  comparing  large  sets  of protein or nucleotide
              sequences", Weizhong Li &amp; Adam Godzik. Bioinformatics, (2006)  22:1658-1659  "CD-HIT:  accelerated
              for clustering the next generation sequencing data", Limin Fu, Beifang Niu, Zhengwei Zhu, Sitao Wu
              &amp; Weizhong Li. Bioinformatics, (2012) 28:3150-3152 "Beifang Niu, Limin Fu, Shulei Sun and Weizhong
              Li.   Artificial  and  natural  duplicates  in  pyrosequencing  reads  of  metagenomic  data.  BMC
              Bioinformatics (2010) 11:187

cd-hit-454 4.8.1                                   August 2021                                     <u><a href="../man1/CD-HIT-454.1.html">CD-HIT-454</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>