<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>chimeraslayer - detects likely chimeras in PCR amplified DNA</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/chimeraslayer">chimeraslayer_20101212+dfsg1-6_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       chimeraslayer - detects likely chimeras in PCR amplified DNA

</pre><h4><b>DESCRIPTION</b></h4><pre>
       ChimeraSlayer is a chimeric sequence detection utility, compatible with near-full length Sanger sequences
       and shorter 454-FLX sequences (~500bp).

       Chimera Slayer involves the following series of steps that operate to flag chimeric 16S rRNA sequences:

       1.     the  ends  of a query sequence are searched against an included database of reference chimera-free
              16S sequences to identify potential parents of a chimera

       2.     candidate parents of a chimera are selected as those that form a branched best  scoring  alignment
              to the NAST-formatted query sequence

       3.     the  NAST  alignment  of  the  query  sequence is improved in a ‘chimera-aware’ profile-based NAST
              realignment to the selected reference parent sequences

       4.     an evolutionary framework is used to flag  query  sequences  found  to  exhibit  greater  sequence
              homology  to  an  in  silico  chimera  formed  between  any  two  of the selected reference parent
              sequences.

       To run Chimera Slayer, you need NAST-formatted sequences generated by the nast-ier utility.

       ChimeraSlayer is part of the microbiomeutil suite.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Required</b>
       <b>--query_NAST</b>
              multi-fasta file containing query sequences in alignment format

   <b>Common</b> <b>options</b>
       <b>--db_NAST</b>
              db                  in                   NAST                   format                   (default:
              <u>/usr/share/microbiomeutil-data/RESOURCES/rRNA16S.gold.NAST_ALIGNED.fasta</u>)

       <b>--db_FASTA</b>
              db         in         fasta         format         (megablast         formatted)         (default:
              <u>/usr/share/microbiomeutil-data/RESOURCES/rRNA16S.gold.fasta</u>)

       <b>-n</b>     number of top matching database sequences to compare to (default 15)

       <b>-R</b>     min divergence ratio default: 1.007

       <b>-P</b>     min percent identity among matching sequences (default: 90)

   <b>Parameters</b> <b>to</b> <b>tune</b> <b>ChimeraParentSelector:</b>
       Scoring parameters:

       <b>-M</b>     match score   (default: +5)

       <b>-N</b>     mismatch penalty  (default: <b>-4</b>)

       <b>-Q</b>     min query coverage by matching database sequence (default: 70)

       <b>-T</b>     maximum traverses of the multiple alignment  (default: 1)

   <b>Parameters</b> <b>to</b> <b>tune</b> <b>ChimeraPhyloChecker</b>
       <b>--windowSize</b>
              default 50

       <b>--windowStep</b>
              default 5

       <b>--minBS</b>
              minimum bootstrap support for calling chimera (default: 90)

       <b>--num_BS_replicates</b>
              default: 100

       <b>--low_range_finer_BS</b>
              (default:  10)  If  computed  BS  is  between  minBS  and  (minBS  -   low_range_finer_BS),   then
              num_finer_BS_replicates computed.

       <b>--num_finer_BS_replicates</b>
              (default: 1000)

       <b>-S</b>     percent of SNPs to sample on each side of breakpoint for computing bootstrap support (default: 10)

       <b>--num_parents_test</b>
              number of potential parents to test for chimeras (default: 3)

       <b>--MAX_CHIMERA_PARENT_PER_ID</b>
              Chimera/Parent  alignments  with perID above this are considered non-chimeras (default 100; turned
              off)

   <b>Misc</b> <b>options</b>
       <b>--printFinalAlignments</b>
              shows alignment between query sequence and pair of candidate chimera parents

       <b>--printCSalignments</b>
              print ChimeraSlayer alignments in ChimeraSlayer output

       <b>--exec_dir</b>
              chdir to here before running

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <a href="http://microbiomeutil.sourceforge.net/">http://microbiomeutil.sourceforge.net/</a>#A_CS

</pre><h4><b>AUTHOR</b></h4><pre>
       This manual page was written by Andreas Tille &lt;<a href="mailto:tille@debian.org">tille@debian.org</a>&gt; but can be freely  used  for  any  other
       distribution.

chimeraslayer                                       June 2013                                   <u><a href="../man1/CHIMERASLAYER.1.html">CHIMERASLAYER</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>