<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>clustalo - General purpose multiple sequence alignment program for proteins</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/clustalo">clustalo_1.2.4-8build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       clustalo - General purpose multiple sequence alignment program for proteins

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>clustalo</b> [-h]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Clustal-Omega  is  a  general purpose multiple sequence alignment (MSA) program for proteins. It produces
       high quality MSAs and is capable  of  handling  data-sets  of  hundreds  of  thousands  of  sequences  in
       reasonable time.

       In  default mode, users give a file of sequences to be aligned and these are clustered to produce a guide
       tree and this is used to guide a "progressive alignment" of the sequences.  There are also facilities for
       aligning existing alignments to each other, aligning a sequence to an alignment and for  using  a  hidden
       Markov  model (HMM) to help guide an alignment of new sequences that are homologous to the sequences used
       to make the HMM.  This latter procedure is referred to as "external profile alignment" or EPA.

       Clustal-Omega uses HMMs for the alignment engine, based on the HHalign package from Johannes Soeding [1].
       Guide trees are made using an enhanced version of mBed [2]  which  can  cluster  very  large  numbers  of
       sequences  in O(N*log(N)) time. Multiple alignment then proceeds by aligning larger and larger alignments
       using HHalign, following the clustering given by the guide tree.

       In its current form Clustal-Omega can only align protein sequences  but  not  DNA/RNA  sequences.  It  is
       envisioned that DNA/RNA will become available in a future version.

</pre><h4><b>USAGE</b></h4><pre>
       Tool usage is available in /usr/share/doc/clustalo/README.

</pre><h4><b>DEVELOPMENT</b></h4><pre>
       Headers and libraries are available in libclustalo-dev package.

</pre><h4><b>CITING</b></h4><pre>
       Sievers F, Wilm A, Dineen DG, Gibson TJ, Karplus K, Li W, Lopez R, McWilliam H,
        Remmert  M, SÃ¶ding J, Thompson JD, Higgins DG (2011).  Fast, scalable generation of high-quality protein
       multiple sequence alignments
        using Clustal Omega. Mol Syst Biol 7.

</pre><h4><b>AUTHOR</b></h4><pre>
       Olivier Sallou (olivier.sallou (at) irisa.fr) - Man page and packaging

       Conway Institute UCD Dublin (clustalw (at) ucd.ie) - clustalo

version 1.0.3                                   December 14, 2011                                    <u><a href="../man1/clustalo.1.html">clustalo</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>