<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>cmalign - align sequences to a covariance model</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/infernal">infernal_1.1.5-3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       cmalign - align sequences to a covariance model

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>cmalign</b>
              <u>[options]</u> <u>&lt;cmfile&gt;</u> <u>&lt;seqfile&gt;</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>cmalign</b>  aligns  the  RNA  sequences  in  <u>&lt;seqfile&gt;</u>  to  the  covariance model (CM) in <u>&lt;cmfile&gt;.</u>  The new
       alignment is output to <u>stdout</u> in Stockholm format, but can be redirected to a file <u>&lt;f&gt;</u> with  the  <b>-o</b>  <u>&lt;f&gt;</u>
       option.

       Either  <u>&lt;cmfile&gt;</u> or <u>&lt;seqfile&gt;</u> (but not both) may be '-' (dash), which means reading this input from <u>stdin</u>
       rather than a file.

       The sequence file <u>&lt;seqfile&gt;</u> must be in FASTA or Genbank format.

       <b>cmalign</b> uses an HMM banding technique to accelerate alignment by  default  as  described  below  for  the
       <b>--hbanded</b> option. HMM banding can be turned off with the <b>--nonbanded</b> option.

       By  default,  <b>cmalign</b>  computes  the  alignment  with  maximum  expected accuracy that is consistent with
       constraints (bands) derived from an HMM, using a banded version of  the  Durbin/Holmes  optimal  accuracy
       algorithm.  This behavior can be changed with the <b>--cyk</b> or <b>--sample</b> options.

       <b>cmalign</b>  takes  special  care  to  correctly  align  truncated sequences, where some nucleotides from the
       beginning (5') and/or end (3') of the actual full length biological sequence are not present in the input
       sequence (see DL Kolbe and SR Eddy, Bioinformatics, 25:1236-1243, 2009). This behavior is on by  default,
       but  can  be turned off with <b>--notrunc.</b>  In previous versions of <b>cmalign</b> the <b>--sub</b> option was required to
       appropriately handle truncated sequences. The <b>--sub</b> option is still available in this  version,  but  the
       new  default  method  for handling truncated sequences should be as good or superior to the sub method in
       nearly all cases.

       The <b>--mapali</b> <u>&lt;s&gt;</u> option allows inclusion of the fixed training alignment used to build the CM  from  file
       <u>&lt;s&gt;</u> within the output alignment of <b>cmalign.</b>

       It  is  possible  to  merge  two  or  more alignments created by the same CM using the Easel miniapp <b>esl-</b>
       <b>alimerge</b> (included in the  easel/miniapps/  subdirectory  of  Infernal).  Previous  versions  of  <b>cmalign</b>
       included  options to merge alignments but they were deprecated upon development of <b>esl-alimerge,</b> which is
       significantly more memory efficient.

       By default, <b>cmalign</b> will output the alignment to stdout.  The alignment can be redirected  to  an  output
       file <u>&lt;f&gt;</u> with the <b>-o</b> <u>&lt;f&gt;</u> option. With <b>-o,</b> information on each aligned sequence, including score and model
       alignment boundaries will be printed to stdout (more on this below).

       The  output  alignment will be in Stockholm format by default. This can be changed to Pfam, aligned FASTA
       (AFA), A2M, Clustal, or Phylip format using the <b>--outformat</b> <u>&lt;s&gt;</u> option, where <u>&lt;s&gt;</u>  is  the  name  of  the
       desired  format.  As a special case, if the output alignment is large (more than 10,000 sequences or more
       than 10,000,000 total nucleotides) than the output  format  will  be  Pfam  format,  with  each  sequence
       appearing  on  a  single  line,  for reasons of memory efficiency. For alignments larger than this, using
       <b>--ileaved</b> will force interleaved Stockholm format, but the user should be aware that this may  require  a
       lot  of  memory.   <b>--ileaved</b>  will  only work for alignments up to 100,000 sequences or 100,000,000 total
       nucleotides.

       If the output alignment format is Stockholm  or  Pfam,  the  output  alignment  will  be  annotated  with
       posterior  probabilities which estimate the confidence level of each aligned nucleotide.  This annotation
       appears as lines beginning with "#=GR &lt;seq name&gt; PP",  one  per  sequence,  each  immediately  below  the
       corresponding  aligned sequence "&lt;seq name&gt;". Characters in PP lines have 12 possible values: "0-9", "*",
       or ".". If ".", the position corresponds to a gap in the sequence. A value of "0" indicates  a  posterior
       probability  of between 0.0 and 0.05, "1" indicates between 0.05 and 0.15, "2" indicates between 0.15 and
       0.25 and so on up to "9" which indicates between 0.85 and 0.95. A value  of  "*"  indicates  a  posterior
       probability of between 0.95 and 1.0. Higher posterior probabilities correspond to greater confidence that
       the  aligned  nucleotide belongs where it appears in the alignment.  With <b>--nonbanded,</b> the calculation of
       the posterior probabilities considers all possible alignments of the target sequence to the  CM.  Without
       <b>--nonbanded</b>  (i.e.  in  default  mode), the calculation considers only possible alignments within the HMM
       bands. Further, the posterior probabilities are conditional on the truncation mode of the alignment.  For
       example,  if the sequence alignment is truncated 5', a PP value of "9" indicates between 0.85 and 0.95 of
       all 5' truncated alignments include the given nucleotide at the given position.  The posterior annotation
       can be turned off with the <b>--noprob</b> option. If <b>--small</b> is enabled,  posterior  annotation  must  also  be
       turned off using <b>--noprob.</b>

       The  tabular output that is printed to stdout if the <b>-o</b> option is used includes one line per sequence and
       twelve fields per line: "idx": the index of the sequence in the input  file,  "seq  name":  the  sequence
       name;  "length":  the length of the sequence; "cm from" and "cm to": the model start and end positions of
       the alignment; "trunc": "no" if the sequence is not truncated, "5'" if  the  beginning  of  the  sequence
       truncated 5', "3'" if the end of the sequence is truncated, and "5'&amp;3'" if both the beginning and the end
       are  truncated;  "bit  sc": the bit score of the alignment, "avg pp" the average posterior probability of
       all aligned nucleotides in the alignment; "band calc", "alignment"  and  "total":  the  time  in  seconds
       required  for  calculating  HMM  bands, computing the alignment, and complete processing of the sequence,
       respectively; "mem (Mb)": the size in Mb of all dynamic programming matrices required  for  aligning  the
       sequence.  This tabular data can be saved to file <u>&lt;f&gt;</u> with the <b>--sfile</b> <u>&lt;f&gt;</u> option.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>     Help; print a brief reminder of command line usage and available options.

       <b>-o</b> <u>&lt;f&gt;</u> Save  the  alignment  in  Stockholm  format to a file <u>&lt;f&gt;.</u>  The default is to write it to standard
              output.

       <b>-g</b>     Configure the model for global alignment of the query model to the target sequences.  By  default,
              the  model  is  configured  for local alignment. Local alignments can contain large insertions and
              deletions called "local ends" in the structure to be penalized  differently  than  normal  indels.
              These  are  annotated  as "~" columns in the RF line of the output alignment. The <b>-g</b> option can be
              used to disallow these local ends.  The <b>-g</b> option is required if the <b>--sub</b> option is also used.

</pre><h4><b>OPTIONS</b> <b>FOR</b> <b>CONTROLLING</b> <b>THE</b> <b>ALIGNMENT</b> <b>ALGORITHM</b></h4><pre>
       <b>--optacc</b>
              Align sequences using the Durbin/Holmes optimal accuracy algorithm.  This  is  the  default.   The
              optimal  accuracy  alignment  will  be  constrained  by  HMM  bands  for  acceleration  unless the
              <b>--nonbanded</b> option is enabled.  The optimal  accuracy  algorithm  determines  the  alignment  that
              maximizes  the  posterior  probabilities  of  the  aligned  nucleotides  within it.  The posterior
              probabilites are determined using (possibly  HMM  banded)  variants  of  the  Inside  and  Outside
              algorithms.

       <b>--cyk</b>  Do  not  use  the Durbin/Holmes optimal accuracy alignment to align the sequences, instead use the
              CYK algorithm which determines  the  optimally  scoring  (maximum  likelihood)  alignment  of  the
              sequence to the model, given the HMM bands (unless <b>--nonbanded</b> is also enabled).

       <b>--sample</b>
              Sample  an alignment from the posterior distribution of alignments.  The posterior distribution is
              determined using an HMM banded (unless <b>--nonbanded)</b> variant of the Inside algorithm.

       <b>--seed</b> <u>&lt;n&gt;</u>
              Seed the random number generator with <u>&lt;n&gt;,</u> an integer &gt;= 0.  This  option  can  only  be  used  in
              combination  with  <b>--sample.</b>   If  <u>&lt;n&gt;</u>  is  nonzero,  stochastic  sampling  of  alignments will be
              reproducible; the same command will give the same  results.   If  <u>&lt;n&gt;</u>  is  0,  the  random  number
              generator  is  seeded  arbitrarily,  and stochastic samplings may vary from run to run of the same
              command.  The default seed is 181.

       <b>--notrunc</b>
              Turn off truncated alignment algorithms.  All sequences in the input file will be  assumed  to  be
              full  length,  unless  <b>--sub</b>  is  also  used, in which case the program can still handle truncated
              sequences but will use an alternative strategy for their alignment.

       <b>--sub</b>  Turn on the sub model construction and alignment procedure. For each sequence,  an  HMM  is  first
              used  to  predict  the model start and end consensus columns, and a new sub CM is constructed that
              only models consensus columns from start to end. The sequence is then aligned to this sub CM.  Sub
              alignment is an older method than the  default  one  for  aligning  sequences  that  are  possibly
              truncated.  By  default,  <b>cmalign</b>  uses  special DP algorithms to handle truncated sequences which
              should be more accurate than the sub method in most cases.  <b>--sub</b> is still included as  an  option
              mainly  for  testing against this default truncated sequence handling.  This "sub CM" procedure is
              not the same as the "sub CMs" described by Weinberg and Ruzzo.

</pre><h4><b>OPTIONS</b> <b>FOR</b> <b>CONTROLLING</b> <b>SPEED</b> <b>AND</b> <b>MEMORY</b> <b>REQUIREMENTS</b></h4><pre>
       <b>--hbanded</b>
              This option is turned on by default. Accelerate alignment by pruning away regions  of  the  CM  DP
              matrix  that are deemed negligible by an HMM.  First, each sequence is scored with a CM plan 9 HMM
              derived from the CM  using  the  Forward  and  Backward  HMM  algorithms  to  calculate  posterior
              probabilities  that each nucleotide aligns to each state of the HMM. These posterior probabilities
              are used to derive constraints (bands) on the CM  DP  matrix.  Finally,  the  target  sequence  is
              aligned  to  the  CM using the banded DP matrix, during which cells outside the bands are ignored.
              Usually most of the full DP matrix lies outside the bands  (often  more  than  95%),  making  this
              technique  faster  because  fewer  DP calculations are required, and more memory efficient because
              only cells within the bands need be allocated.

              Importantly, HMM banding sacrifices the  guarantee  of  determining  the  optimally  accurarte  or
              optimal  alignment,  which  will  be missed if it lies outside the bands. The tau parameter is the
              amount of probability mass considered negligible during HMM band calculation; lower values of  tau
              yield greater speedups but also a greater chance of missing the optimal alignment. The default tau
              is  1E-7,  determined  empirically  as  a good tradeoff between sensitivity and speed, though this
              value can be changed with the <b>--tau</b>  <b>&lt;x&gt;</b> option. The level of acceleration increases with both the
              length and primary sequence conservation level of the family. For example, with the default tau of
              1E-7, tRNA models (low primary sequence conservation with length of  about  75  nucleotides)  show
              about  10X  acceleration,  and  SSU bacterial rRNA models (high primary sequence conservation with
              length of about 1500 nucleotides) show about 700X.   HMM  banding  can  be  turned  off  with  the
              <b>--nonbanded</b> option.

       <b>--tau</b> <u>&lt;x&gt;</u>
              Set  the  tail  loss  probability  used during HMM band calculation to <u>&lt;x&gt;.</u>  This is the amount of
              probability mass within the HMM posterior probabilities that is considered negligible. The default
              value is 1E-7.  In general, higher values will result in greater acceleration,  but  increase  the
              chance of missing the optimal alignment due to the HMM bands.

       <b>--mxsize</b> <u>&lt;x&gt;</u>
              Set  the maximum allowable total DP matrix size to <u>&lt;x&gt;</u> megabytes. By default this size is 1024 Mb.
              This should be large enough for the vast majority of alignments, however if it is not <b>cmalign</b> will
              attempt to iteratively tighten the HMM bands it uses to constrain the alignment by raising the tau
              parameter and recalculating the bands until the total matrix size needed falls below <u>&lt;x&gt;</u> megabytes
              or the maximum allowable tau value (0.05 by default, but changeable with <b>--maxtau)</b> is reached.  At
              each iteration of band tightening, tau is multiplied by a 2.0. The band tightening strategy can be
              turned off with the <b>--fixedtau</b> option.  If the maximum tau is reached and the required matrix size
              still  exceeds  <u>&lt;x&gt;</u>  or  if HMM banding is not being used and the required matrix size exceeds <u>&lt;x&gt;</u>
              then <b>cmalign</b> will exit prematurely and report an  error  message  that  the  matrix  exceeded  its
              maximum  allowable  size.  In  this  case, the <b>--mxsize</b> can be used to raise the size limit or the
              maximum tau can be raised with <b>--maxtau.</b>  The limit will commonly be exceeded when the <b>--nonbanded</b>
              option is used without the <b>--small</b> option, but can still occur when <b>--nonbanded</b> is not used.  Note
              that  if  <b>cmalign</b> is being run in <u>&lt;n&gt;</u> multiple threads on a multicore machine then each thread may
              have an allocated matrix of up to size <u>&lt;x&gt;</u> Mb at any given time.

       <b>--fixedtau</b>
              Turn off the HMM band tightening strategy described in the  explanation  of  the  <b>--mxsize</b>  option
              above.

       <b>--maxtau</b> <u>&lt;x&gt;</u>
              Set  the  maximum  allowed  value  for tau during band tightening, described in the explanation of
              <b>--mxsize</b> above, to <u>&lt;x&gt;.</u>  By default this value is 0.05.

       <b>--nonbanded</b>
              Turns off HMM banding. The returned alignment is guaranteed to be the globally optimally  accurate
              one  (by default) or the globally optimally scoring one (if <b>--cyk</b> is enabled).  The <b>--small</b> option
              is recommended in combination with this option, because standard  alignment  without  HMM  banding
              requires a lot of memory (see <b>--small</b> ).

       <b>--small</b>
              Use  the divide and conquer CYK alignment algorithm described in SR Eddy, BMC Bioinformatics 3:18,
              2002. The <b>--nonbanded</b> option must  be  used  in  combination  with  this  options.   Also,  it  is
              recommended  whenever <b>--nonbanded</b> is used that <b>--small</b> is also used  because standard CM alignment
              without HMM banding requires a lot of memory,  especially  for  large  RNAs.   <b>--small</b>  allows  CM
              alignment within practical memory limits, reducing the memory required for alignment LSU rRNA, the
              largest  known RNAs, from 150 Gb to less than 300 Mb.  This option can only be used in combination
              with <b>--noprob,</b> <b>--nonbanded,</b> <b>--notrunc,</b> and <b>--cyk.</b>

</pre><h4><b>OPTIONAL</b> <b>OUTPUT</b> <b>FILES</b></h4><pre>
       <b>--sfile</b> <u>&lt;f&gt;</u>
              Dump per-sequence alignment score and timig information to file <u>&lt;f&gt;.</u>  The format of this  file  is
              described  above  (it's  the same data in the same format as the tabular stdout output when the <b>-o</b>
              option is used).

       <b>--tfile</b> <u>&lt;f&gt;</u>
              Dump tabular sequence tracebacks for each individual sequence to a file <u>&lt;f&gt;.</u>  Primarily useful for
              debugging.

       <b>--ifile</b> <u>&lt;f&gt;</u>
              Dump per-sequence insert information to file  <u>&lt;f&gt;.</u>   The  format  of  the  file  is  described  by
              "#"-prefixed  comment  lines included at the top of the file <u>&lt;f&gt;.</u>  The insert information is valid
              even when the <b>--matchonly</b> option is used.

       <b>--elfile</b> <u>&lt;f&gt;</u>
              Dump per-sequence EL state (local end) insert information to file <u>&lt;f&gt;.</u>  The format of the file  is
              described  by  "#"-prefixed  comment  lines  included  at  the top of the file <u>&lt;f&gt;.</u>  The EL insert
              information is valid even when the <b>--matchonly</b> option is used.

</pre><h4><b>OTHER</b> <b>OPTIONS</b></h4><pre>
       <b>--mapali</b> <u>&lt;f&gt;</u>
              Reads the alignment from file <u>&lt;f&gt;</u> used to build the model aligns it as a single object to the  CM;
              e.g.  the  alignment  in  <u>&lt;f&gt;</u>  is  held fixed.  This allows you to align sequences to a model with
              <b>cmalign</b> and view them in the context of an existing trusted multiple alignment.  <u>&lt;f&gt;</u> must  be  the
              alignment  file  that  the  CM  was built from. The program verifies that the checksum of the file
              matches that of the file used to construct the CM.  A  similar  option  to  this  one  was  called
              <b>--withali</b> in previous versions of <b>cmalign.</b>

       <b>--mapstr</b>
              Must  be  used  in  combination  with  <b>--mapali</b>  <u>&lt;f&gt;</u><b>.</b>   Propagate  structural  information for any
              pseudoknots that exist in <u>&lt;f&gt;</u> to the output alignment. A similar option to  this  one  was  called
              <b>--withstr</b> in previous versions of <b>cmalign.</b>

       <b>--informat</b> <u>&lt;s&gt;</u>
              Assert  that  the input <u>&lt;seqfile&gt;</u> is in format <u>&lt;s&gt;.</u>  Do not run Babelfish format autodection. This
              increases the reliability of the program  somewhat,  because  the  Babelfish  can  make  mistakes;
              particularly  recommended  for  unattended,  high-throughput runs of Infernal.  Acceptable formats
              are: FASTA, GENBANK, and DDBJ.  <u>&lt;s&gt;</u> is case-insensitive.

       <b>--outformat</b> <u>&lt;s&gt;</u>
              Specify the output alignment format as <u>&lt;s&gt;.</u>  Acceptable formats are: Pfam, AFA, A2M, Clustal,  and
              Phylip.   AFA  is  aligned fasta. Only Pfam and Stockholm alignment formats will include consensus
              structure annotation and posterior probability annotation of aligned residues.

       <b>--dnaout</b>
              Output the alignments as DNA sequence alignments, instead of RNA ones.

       <b>--noprob</b>
              Do not annotate the output alignment with posterior probabilities.

       <b>--matchonly</b>
              Only include match columns in the output alignment, do not include any insertions relative to  the
              consensus  model. This option may be useful when creating very large alignments that require a lot
              of memory and disk space, most of which is necessary only to deal with  insert  columns  that  are
              gaps in most sequences.

       <b>--miss</b> In  the output alignment, use missing data characters ('~') before the first residue and after the
              final residue of each sequence to indicate the sequence was aligned  with  a  truncated  alignment
              algorithm.  The  aligned  sequences  would  be  considered  fragments  if  the  alignment was used
              subsequently as input to <b>cmbuild</b> with the  <b>--fraggiven</b>  option.  This  option  has  no  effect  if
              <b>--notrunc</b> is also used.

       <b>--ileaved</b>
              Output  the alignment in interleaved Stockholm format of a fixed width that may be more convenient
              for examination. This was the default output alignment format of  previous  versions  of  <b>cmalign.</b>
              Note  that <b>cmalign</b> requires more memory when this option is used.  For this reason, <b>--ileaved</b> will
              only work for alignments of up to 100,000 sequences or a total of 100,000,000 aligned nucleotides.

       <b>--flanktoins</b> <u>&lt;x1&gt;</u>
              Change the transition probabilities from the ROOT_S state to the ROOT_IL and ROOT_IR  states,  and
              from  the  ROOT_IL  to  the  ROOT_IR  state to <u>&lt;x1&gt;.</u>  This option is meant to be helpful only when
              aligning sequences that include extra sequence at the 5' and/or 3' ends. Without using this option
              <b>cmalign</b> tends to mess up alignments at the end, especially for models with  zero  basepairs.  This
              option  should  not  be necessary when aligning sequences identified by <b>cmsearch</b> or <b>cmscan</b> because
              they should not include extra sequence at the ends.  This option must be used in combination  with
              the  <b>--flankselfins</b> <u>&lt;x2&gt;</u> option. Recommended values to use are 0.1 for <u>&lt;x1&gt;</u> and 0.8 for <u>&lt;x2&gt;</u> , but
              the best performing pair of values may vary for different models.  <u>&lt;x1&gt;</u> must be greater  than  0.0
              and less than 0.4, and the sum of <u>&lt;x1&gt;</u> and <u>&lt;x2&gt;</u> must be less than 0.95.

       <b>--flankselfins</b> <u>&lt;x2&gt;</u>
              Change  the self-transition probabilities for the ROOT_IL and ROOT_IR states to <u>&lt;x2&gt;.</u>  This option
              must be used in combination with the <b>--flanktoins</b> <u>&lt;x1&gt;</u> option. See the explanation of that  option
              above for more information.

       <b>--regress</b> <u>&lt;s&gt;</u>
              Save an additional copy of the output alignment with no author information to file <u>&lt;s&gt;.</u>

       <b>--verbose</b>
              Output  additional information in the tabular scores output (output to stdout if <b>-o</b> is used, or to
              <u>&lt;f&gt;</u> if <b>--sfile</b> <u>&lt;f&gt;</u> is used). These are mainly useful for testing and debugging.

       <b>--cpu</b> <u>&lt;n&gt;</u>
              Set the number of parallel worker threads to <u>&lt;n&gt;</u>.  On multicore machines, the default is  4.   You
              can  also  control this number by setting an environment variable, <u>INFERNAL_NCPU</u>.  There is also a
              master thread, so the actual number of threads that Infernal spawns is <u>&lt;n&gt;</u>+1.  This option is  not
              available if Infernal was compiled with POSIX threads support turned off.

       <b>--mpi</b>  Run as an MPI parallel program. This option will only be available if Infernal has been configured
              and  built  with  the "--enable-mpi" flag (see the Installation section of the user guide for more
              information).

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       See <b><a href="../man1/infernal.1.html">infernal</a>(1)</b> for a master man page with a list of all the individual man pages  for  programs  in  the
       Infernal package.

       For complete documentation, see the user guide that came with your Infernal distribution (Userguide.pdf);
       or see the Infernal web page (<a href="http://eddylab.org/infernal/">http://eddylab.org/infernal/</a>).

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright (C) 2023 Howard Hughes Medical Institute.
       Freely distributed under the BSD open source license.

       For  additional  information  on  copyright and licensing, see the file called COPYRIGHT in your Infernal
       source distribution, or see the Infernal web page (<a href="http://eddylab.org/infernal/">http://eddylab.org/infernal/</a>).

</pre><h4><b>AUTHOR</b></h4><pre>
       <a href="http://eddylab.org">http://eddylab.org</a>

Infernal 1.1.5                                      Sep 2023                                          <u><a href="../man1/cmalign.1.html">cmalign</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>