<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>cmemit - sample sequences from a covariance model</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/infernal">infernal_1.1.5-3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       cmemit - sample sequences from a covariance model

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>cmemit</b> <u>[options]</u> <u>&lt;cmfile&gt;</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The <b>cmemit</b> program samples (emits) sequences from the covariance model(s) in <u>&lt;cmfile&gt;,</u> and writes them to
       output.   Sampling  sequences  may be useful for a variety of purposes, including creating synthetic true
       positives for benchmarks or tests.

       The default is to sample ten unaligned sequence from each CM. Alternatively, with the <b>-c</b> option, you  can
       emit a single majority-rule consensus sequence; or with the <b>-a</b> option, you can emit an alignment.

       The <u>&lt;cmfile&gt;</u> may contain a library of CMs, in which case each CM will be used in turn.

       <u>&lt;cmfile&gt;</u> may be '-' (dash), which means reading this input from <u>stdin</u> rather than a file.

       For  models  with  zero  basepairs,  sequences are sampled from the profile HMM filter instead of the CM.
       However, since these models will be nearly identical (unless special options  were  used  in  <b>cmbuild</b>  to
       prevent  this),  using  the HMM instead of the CM will not change the output in a significant way, unless
       the <b>-l</b> option is used. With <b>-l,</b> the  HMM  will  be  configured  for  equiprobable  model  begin  and  end
       positions,  while the CM will not. You can force <b>cmemit</b> to always sample from the CM with the <b>--nohmmonly</b>
       option.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>     Help; print a brief reminder of command line usage and available options.

       <b>-o</b> <u>&lt;f&gt;</u> Save the synthetic sequences to file <u>&lt;f&gt;</u> rather than writing them to stdout.

       <b>-N</b> <u>&lt;n&gt;</u> Generate <u>&lt;n&gt;</u> sequences. The default value for <u>&lt;n&gt;</u> is 10.

       <b>-u</b>     Write the generated sequences in unaligned format (FASTA). This is the default behavior.

       <b>-a</b>     Write the generated sequences in an aligned format (STOCKHOLM) with consensus structure annotation
              rather than FASTA. Other output formats are possible with the <b>--outformat</b> option.

       <b>-c</b>     Predict a single majority-rule consensus sequence instead of  sampling  sequences  from  the  CMÂ´s
              probability  distribution.  Highly conserved residues (base paired residues that score higher than
              3.0 bits, or single stranded residues that score higher than 1.0 bits) are shown  in  upper  case;
              others are shown in lower case.

       <b>-e</b> <u>&lt;n&gt;</u> Embed  the  CM  emitted  sequences in a larger randomly generated sequence of length <u>&lt;n&gt;</u> generated
              from an HMM that was trained on real genomic sequences with various GC contents (the same HMM used
              by <b>cmcalibrate).</b>  You can use the <b>--iid</b> option to generate 25% A, C, G, and  U  sequence  instead.
              The  CM  emitted  sequence  will begin at a random position within the larger sequence and will be
              included in its entirety unless the <b>--u5p</b>  or  <b>--u3p</b>  options  are  used.   When  <b>-e</b>  is  used  in
              combination  with  <b>--u5p,</b>  the  CM  emitted sequence will always begin at position 1 of the larger
              sequence and will be truncated 5'. When used in combination <b>--u3p</b> the  CM  emitted  sequence  will
              always end at position <u>&lt;n&gt;</u> of the larger sequence and will be truncated 3'.

       <b>-l</b>     Configure  the  CMs  into  local  mode  before emitting sequences. By default the model will be in
              global mode. In local mode, large insertions and deletions are more common than in global mode.

</pre><h4><b>OPTIONS</b> <b>FOR</b> <b>TRUNCATING</b> <b>EMITTED</b> <b>SEQUENCES</b></h4><pre>
       <b>--u5p</b>  Truncate all emitted sequences at a  randomly  chosen  start  position  <b>&lt;n&gt;,</b>  by  only  outputting
              residues beginning at <b>&lt;n&gt;.</b>  A different start point is randomly chosen for each sequence.

       <b>--u3p</b>  Truncate  all emitted sequences at a randomly chosen end position <b>&lt;n&gt;,</b> by only outputting residues
              up to position <b>&lt;n&gt;.</b>  A different end point is randomly chosen for each sequence.

       <b>--a5p</b> <u>&lt;n&gt;</u>
              In combination with the <b>-a</b> option, truncate the emitted alignment at a randomly chosen start match
              position <u>&lt;n&gt;,</u> by only outputting alignment columns for positions after match state <u>&lt;n&gt;</u> -  1.   <u>&lt;n&gt;</u>
              must  be an integer between 0 and the consensus length of the model (which can be determined using
              the <b>cmstat</b> program. As a special case, using 0 as <u>&lt;n&gt;</u> will  result  in  a  randomly  chosen  start
              position.

       <b>--a3p</b> <u>&lt;n&gt;</u>
              In  combination  with the <b>-a</b> option, truncate the emitted alignment at a randomly chosen end match
              position <u>&lt;n&gt;,</u> by only outputting alignment columns for positions before match state <u>&lt;n&gt;</u> + 1.   <u>&lt;n&gt;</u>
              must  be an integer between 1 and the consensus length of the model (which can be determined using
              the <b>cmstat</b> program). As a special case, using 0 as <u>&lt;n&gt;</u>  will  result  in  a  randomly  chosen  end
              position.

</pre><h4><b>OTHER</b> <b>OPTIONS</b></h4><pre>
       <b>--seed</b> <u>&lt;n&gt;</u>
              Seed the random number generator with <u>&lt;n&gt;,</u> an integer &gt;= 0. If <u>&lt;n&gt;</u> is nonzero, stochastic sampling
              of  sequences will be reproducible; the same command will give the same results.  If <u>&lt;n&gt;</u> is 0, the
              random number generator is seeded arbitrarily, and stochastic samplings will vary from run to  run
              of the same command.  The default seed is 0.

       <b>--iid</b>  With <b>-e,</b> generate the larger sequences as 25% each A, C, G and U.

       <b>--rna</b>  Specify that the emitted sequences be output as RNA sequences. This is true by default.

       <b>--dna</b>  Specify  that the emitted sequences be output as DNA sequences. By default, the output alphabet is
              RNA.

       <b>--idx</b> <u>&lt;n&gt;</u>
              Specify that the emitted sequences be named starting with <u>&lt;modelname&gt;.&lt;n&gt;.</u>  By default <u>&lt;n&gt;</u> is 1.

       <b>--outformat</b> <u>&lt;s&gt;</u>
              With <b>-a,</b> specify the output alignment format as <u>&lt;s&gt;.</u>  Acceptable  formats  are:  Pfam,  AFA,  A2M,
              Clustal, and Phylip.  AFA is aligned fasta. Only Pfam and Stockholm alignment formats will include
              consensus structure annotation.

       <b>--tfile</b> <u>&lt;f&gt;</u>
              Dump  tabular  sequence  parsetrees (tracebacks) for each emitted sequence to file <u>&lt;f&gt;.</u>  Primarily
              useful for debugging.

       <b>--exp</b> <u>&lt;x&gt;</u>
              Exponentiate the emission and transition probabilities of the CM by <u>&lt;x&gt;</u> and then renormalize those
              distributions before emitting sequences. This option changes the CM  probability  distribution  of
              parsetrees  relative  to  default.  With <u>&lt;x&gt;</u> less than 1.0 the emitted sequences will tend to have
              lower bit scores upon alignment to the CM.  With &lt;x&gt; greater than 1.0, the emitted sequences  will
              tend  to  have higher bit scores upon alignment to the CM. This bit score difference will increase
              as &lt;x&gt; moves further away from 1.0 in either direction.  If &lt;x&gt; equals 1.0,  this  option  has  no
              effect  relative  to default.  This option is useful for generating sequences that are either more
              difficult ( <u>&lt;x&gt;</u> &lt; 1.0) or easier ( <u>&lt;x&gt;</u> &gt; 1.0)  for  the  CM  to  distinguish  as  homologous  from
              background, random sequence.

       <b>--hmmonly</b>
              Emit from the filter profile HMM instead of the CM.

       <b>--nohmmonly</b>
              Never emit from the filter profile HMM, always use the CM, even for models with zero basepairs.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       See  <b><a href="../man1/infernal.1.html">infernal</a>(1)</b>  for  a  master man page with a list of all the individual man pages for programs in the
       Infernal package.

       For complete documentation, see the user guide that came with your Infernal distribution (Userguide.pdf);
       or see the Infernal web page (<a href="http://eddylab.org/infernal/">http://eddylab.org/infernal/</a>).

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright (C) 2023 Howard Hughes Medical Institute.
       Freely distributed under the BSD open source license.

       For additional information on copyright and licensing, see the file called  COPYRIGHT  in  your  Infernal
       source distribution, or see the Infernal web page (<a href="http://eddylab.org/infernal/">http://eddylab.org/infernal/</a>).

</pre><h4><b>AUTHOR</b></h4><pre>
       <a href="http://eddylab.org">http://eddylab.org</a>

Infernal 1.1.5                                      Sep 2023                                           <u><a href="../man1/cmemit.1.html">cmemit</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>