<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>dotter - Sequence dotplots with image enhancement tools</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/dotter">dotter_4.44.1+dfsg-7.1build3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       dotter - Sequence dotplots with image enhancement tools

</pre><h4><b>DESCRIPTION</b></h4><pre>
              Dotter - Sequence dotplots with image enhancement tools.

       Usage: dotter [options] &lt;horizontal_sequence&gt; &lt;vertical_sequence&gt;
              [X options]

              Where  &lt;horizontal_sequence&gt; and &lt;vertical_sequence&gt; are file names for FASTA files containing the
              two sequences.

       Allowed sequence types:
              Protein  -   Protein DNA      -   DNA DNA      -   Protein

              Options:

       <b>-h</b>, <b>--help</b>

              Show this usage information

       <b>-b</b> &lt;file&gt;, <b>--batch-save=</b>&lt;file&gt;

              Batch mode; save dot matrix to &lt;file&gt;

       <b>-e</b> &lt;file&gt;, <b>--batch-export=</b>&lt;file&gt;

              Batch mode; export plot to PDF file &lt;file&gt;

       <b>-l</b> &lt;file&gt;, <b>--load</b>

              Load dot matrix from &lt;file&gt;

       <b>-m</b> &lt;float&gt;, <b>--memory-limit=</b>&lt;float&gt;

              Memory usage limit in Mb (default 0.5)

       <b>-z</b> &lt;int&gt;, <b>--zoom</b>

              Set zoom (compression) factor

       <b>-p</b> &lt;int&gt;, <b>--pixel-factor</b>

              Set pixel factor manually (ratio pixelvalue/score)

       <b>-W</b> &lt;int&gt;, <b>--window-size</b>

              Set sliding window size. (K =&gt; Karlin/Altschul estimate)

       <b>-M</b> &lt;file&gt;, <b>--matrix-file=</b>&lt;file&gt;

              Read in score matrix from &lt;file&gt; (Blast format; Default: Blosum62).

       <b>-F</b> &lt;file&gt;, <b>--sequence-file</b>

              Read in sequences and data from &lt;file&gt; (replaces sequencefiles).

       <b>-f</b> &lt;file&gt;, <b>--feature-file</b>

              Read feature segments from &lt;file&gt;

       <b>-H</b>, <b>--hsp-mode</b>

              Do not calculate dotplot at startup.

       <b>-R</b>, <b>--reverse-greyramp</b>

              Reversed Greyramp tool at start.

       <b>-r</b>, <b>--reverse-horizontal</b>

              Reverse and complement horizontal_sequence (if DNA)

       <b>-v</b>, <b>--reverse-vertical</b>

              Reverse and complement vertical_sequence (if DNA)

       <b>-D</b>, <b>--disable-mirror</b>

              Don't display mirror image in self comparisons

       <b>-w</b>, <b>--watson-only</b>

              For DNA: horizontal_sequence top strand only (Watson)

       <b>-c</b>, <b>--crick-only</b>

              For DNA: horizontal_sequence bottom strand only (Crick)

       <b>-q</b> &lt;int&gt;, <b>--horizontal-offset=</b>&lt;int&gt;

              Horizontal_sequence offset

       <b>-s</b> &lt;int&gt;, <b>--vertical-offset=</b>&lt;int&gt;

              Vertical_sequence offset

       <b>--horizontal-type</b>=<u>p</u>|d

              Horizontal_sequence type ('p' for peptide or 'd' for DNA)

       <b>--vertical-type</b>=<u>p</u>|d

              Vertical_sequence type ('p' for peptide or 'd' for DNA)

       <b>--abbrev-title-on</b>

              Abbreviate window title prefixes

       <b>--abbrev-title-off</b>

              Do not abbreviate window title prefixes

       <b>--session_colour=</b>&lt;colour_str&gt;

              Set the background colour of the dotter window

       <b>--compiled</b>

              Show package compile date

       <b>--version</b>

              Show package version

       <b>-----</b>

              Written  by  Gemma  Barson  &lt;<a href="mailto:gb10@sanger.ac.uk">gb10@sanger.ac.uk</a>&gt;  Based  on  original  code  by   Erik   Sonnhammer
              &lt;<a href="mailto:Erik.Sonnhammer@sbc.su.se">Erik.Sonnhammer@sbc.su.se</a>&gt;

              Reference: Sonnhammer ELL &amp; Durbin R (1995). A dot-matrix program

              with  dynamic  threshold  control  suited  for  genomic  DNA  and  protein sequence analysis. Gene
              <a href="../man2/167.2.html">167</a>(2):GC1-10.

              See <a href="http://www.sanger.ac.uk/resources/software/seqtools/">http://www.sanger.ac.uk/resources/software/seqtools/</a> for more info.

              Copyright (c) 2010-2015: Genome Research Ltd.  Dotter is distributed under the GNU General  Public
              License; see <a href="http://www.gnu.org/copyleft/gpl.txt">http://www.gnu.org/copyleft/gpl.txt</a>

       Version 4.44.1
              14:27:56 Oct 19 2017

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

dotter 4.44.1                                     October 2017                                         <u><a href="../man1/DOTTER.1.html">DOTTER</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>