<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>est2genome - Align EST sequences to genomic DNA sequence</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/emboss">emboss_6.6.0+dfsg-15ubuntu2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       est2genome - Align EST sequences to genomic DNA sequence

</pre><h4><b>SYNOPSIS</b></h4><pre>

       <b>est2genome</b> <b>-estsequence</b> <u>seqall</u> <b>-genomesequence</b> <u>sequence</u> [<b>-match</b> <u>integer</u>] [<b>-mismatch</b> <u>integer</u>]
                  [<b>-gappenalty</b> <u>integer</u>] [<b>-intronpenalty</b> <u>integer</u>] [<b>-splicepenalty</b> <u>integer</u>] [<b>-minscore</b> <u>integer</u>]
                  <b>-reverse</b> <u>boolean</u> <b>-usesplice</b> <u>boolean</u> <b>-mode</b> <u>list</u> <b>-best</b> <u>boolean</u> <b>-space</b> <u>float</u> <b>-shuffle</b> <u>integer</u>
                  <b>-seed</b> <u>integer</u> <b>-outfile</b> <u>outfile</u> <b>-align</b> <u>boolean</u> <b>-width</b> <u>integer</u>

       <b>est2genome</b> <b>-help</b>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>est2genome</b> is a command line program from EMBOSS (“the European Molecular Biology Open Software Suite”).
       It is part of the "Alignment:Global" command group(s).

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Input</b> <b>section</b>
       <b>-estsequence</b> <u>seqall</u>

       <b>-genomesequence</b> <u>sequence</u>

   <b>Additional</b> <b>section</b>
       <b>-match</b> <u>integer</u>
           Default value: 1

       <b>-mismatch</b> <u>integer</u>
           Default value: 1

       <b>-gappenalty</b> <u>integer</u>
           Cost for deleting a single base in either sequence, excluding introns Default value: 2

       <b>-intronpenalty</b> <u>integer</u>
           Cost for an intron, independent of length. Default value: 40

       <b>-splicepenalty</b> <u>integer</u>
           Cost for an intron, independent of length and starting/ending on donor-acceptor sites Default value:
           20

       <b>-minscore</b> <u>integer</u>
           Exclude alignments with scores below this threshold score. Default value: 30

   <b>Advanced</b> <b>section</b>
       <b>-reverse</b> <u>boolean</u>
           Reverse the orientation of the EST sequence

       <b>-usesplice</b> <u>boolean</u>
           Use donor and acceptor splice sites. If you want to ignore donor-acceptor sites then set this to be
           false. Default value: Y

       <b>-mode</b> <u>list</u>
           This determines the comparison mode. The default value is 'both', in which case both strands of the
           est are compared assuming a forward gene direction (ie GT/AG splice sites), and the best comparison
           redone assuming a reversed (CT/AC) gene splicing direction. The other allowed modes are 'forward',
           when just the forward strand is searched, and 'reverse', ditto for the reverse strand. Default value:
           both

       <b>-best</b> <u>boolean</u>
           You can print out all comparisons instead of just the best one by setting this to be false. Default
           value: Y

       <b>-space</b> <u>float</u>
           For linear-space recursion. If product of sequence lengths divided by 4 exceeds this then a
           divide-and-conquer strategy is used to control the memory requirements. In this way very long
           sequences can be aligned. If you have a machine with plenty of memory you can raise this parameter
           (but do not exceed the machine's physical RAM) Default value: 10.0

       <b>-shuffle</b> <u>integer</u>

       <b>-seed</b> <u>integer</u>
           Default value: 20825

   <b>Output</b> <b>section</b>
       <b>-outfile</b> <u>outfile</u>

       <b>-align</b> <u>boolean</u>
           Show the alignment. The alignment includes the first and last 5 bases of each intron, together with
           the intron width. The direction of splicing is indicated by angle brackets (forward or reverse) or
           ???? (unknown).

       <b>-width</b> <u>integer</u>
           Default value: 50

</pre><h4><b>BUGS</b></h4><pre>
       Bugs can be reported to the Debian Bug Tracking system (<a href="http://bugs.debian.org/emboss">http://bugs.debian.org/emboss</a>), or directly to
       the EMBOSS developers (<a href="http://sourceforge.net/tracker/">http://sourceforge.net/tracker/</a>?group_id=93650&amp;atid=605031).

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       est2genome is fully documented via the <b><a href="../man1/tfm.1.html">tfm</a></b>(1) system.

</pre><h4><b>AUTHOR</b></h4><pre>
       <b>Debian</b> <b>Med</b> <b>Packaging</b> <b>Team</b> &lt;<a href="mailto:debian-med-packaging@lists.alioth.debian.org">debian-med-packaging@lists.alioth.debian.org</a>&gt;
           Wrote the script used to autogenerate this manual page.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       This manual page was autogenerated from an Ajax Control Definition of the EMBOSS package. It can be
       redistributed under the same terms as EMBOSS itself.

EMBOSS 6.4.0                                       05/11/2012                                     <u><a href="../man1e/EST2GENOME.1e.html">EST2GENOME</a></u>(1e)
</pre>
 </div>
</div></section>
</div>
</body>
</html>