<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>exonerate - a generic tool for sequence comparison</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/exonerate">exonerate_2.4.0-5build2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       exonerate - a generic tool for sequence comparison

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>exonerate</b> <b>[</b> <b>options</b> <b>]</b> <b>&lt;query</b> <b>path&gt;</b> <b>&lt;target</b> <b>path&gt;</b>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>exonerate</b> is a general tool for sequence comparison.

       It  uses the <b>C4</b> dynamic programming library.  It is designed to be both general and fast.  It can produce
       either gapped or ungapped alignments, according to a variety  of  different  alignment  models.   The  C4
       library allows sequence alignment using a reduced space full dynamic programming implementation, but also
       allows  automated  generation  of  heuristics  from  the  alignment  models, using bounded sparse dynamic
       programming, so that these alignments may also be rapidly generated.  Alignments  generated  using  these
       heuristics  will  represent  a  valid  path  through  the  alignment  model,  yet  (unlike the exhaustive
       alignments), the results are not guaranteed to be optimal.

</pre><h4><b>CONVENTIONS</b></h4><pre>
       A number of conventions (and idiosyncracies)  are  used  within  exonerate.   An  understanding  of  them
       facilitates interpretation of the output.

       <b>Coordinates</b>
              An  in-between  coordinate  system  is  used, where the positions are counted between the symbols,
              rather than on the symbols.  This numbering scheme starts from  zero.   This  numbering  is  shown
              below for the sequence "ACGT":

               A C G T
              0 1 2 3 4

              Hence  the  subsequence  "CG"  would have start=1, end=3, and length=2.  This coordinate system is
              used internally in exonerate, and for all the output formats produced with the  exception  of  the
              "human  readable"  alignment  display  and  the  GFF output where convention and standards dictate
              otherwise.

       <b>Reverse</b> <b>Complements</b>
              When an alignment is reported on the reverse complement of a sequence, the coordinates are  simply
              given  on the reverse complement copy of the sequence.  Hence positions on the sequences are never
              negative.  Generally, the forward strand is indicated by '+', the reverse strand by  '-',  and  an
              unknown or not-applicable strand (as in the case of a protein sequence) is indicated by '.'

       <b>Alignment</b> <b>Scores</b>
              Currently, only the raw alignment scores are displayed.  This score just is the sum of transistion
              scores  used  in the dynamic programming.  For example, in the case of a Smith-Waterman alignment,
              this will be the sum of the substitution matrix scores and the gap penalties.

</pre><h4><b>GENERAL</b> <b>OPTIONS</b></h4><pre>
       Most arguments have short and long forms.  The long forms
              are more likely to be stable over time, and hence should be used in scripts which call exonerate.

       <b>-h</b> <b>|</b> <b>--shorthelp</b> <b>&lt;boolean&gt;</b>
              Show help.  This will display a concise summary of the  available  options,  defaults  and  values
              currently set.

       <b>--help</b> <b>&lt;boolean&gt;</b>
              This  shows  all  the  help  options  including  the  defaults,  the  value currently set, and the
              environment variable which may be used to set each parameter.  There  will  be  an  indication  of
              which  options  are  mandatory.  Mandatory options have no default, and must have a value supplied
              for exonerate to run.  If mandatory options are used in order, their flags may be skipped from the
              command line (see examples below).  Unlike this man page, the information from  this  option  will
              always be up to date with the latest version of the program.

       <b>-v</b> <b>|</b> <b>--version</b> <b>&lt;boolean&gt;</b>
              Display  the  version  number.   Also  displays  other information such as the build date and glib
              version used.

</pre><h4><b>SEQUENCE</b> <b>INPUT</b> <b>OPTIONS</b></h4><pre>
       Pairwise comparisons will be performed between all query sequences and all target sequences.   Generally,
       for  the  best  performance,  shorter sequences (eg. ESTs, shotgun reads, proteins) should be used as the
       query sequences, and longer sequences (eg. genomic sequences) should be used as the target sequences.

       <b>-q</b> <b>|</b> <b>--query</b>  <b>&lt;paths&gt;</b>
              Specify the query sequences required.  These must be in a FASTA format file.  Single or  muiltiple
              query  sequences  may be supplied.  Additionally multiple copies of the fasta file may be supplied
              following a --query flag, or by using with multiple --query flags.

       <b>-t</b> <b>|</b> <b>--target</b> <b>&lt;paths&gt;</b>
              Specify the target sequences required.  Also, must be in a FASTA format file.  As with  the  query
              sequences, single or multiple target sequences and files may be supplied.  The target filename may
              by  replace  by  a  server name and port number in the form of <b>hostname:port</b> when using <b>exonerate-</b>
              <b>server.</b>  See the man page for <b>exonerate-server</b>  for  more  information  on  running  exonerate  in
              client:server  mode.   <b>NEW(v2.4.0):</b>  multiple  servers  may now be used.  These will be queried in
              parallel if you have set the <b>--cores</b> option.  <b>NEW(v2.4.0):</b> If an input file is not a FASTA  format
              file, it is assumed to contain a list of other fasta files, directories or servers (one per line).

       <b>-Q</b> <b>|</b> <b>--querytype</b> <b>&lt;dna</b> <b>|</b> <b>protein&gt;</b>
              Specify  the query type to use.  If this is not supplied, the query type is assumed to be DNA when
              the first sequence in the file contains more than 85% [ACGTN] bases.  Otherwise, it is assumed  to
              be  peptide.   This option forces the query type as some nucleotide and peptide sequences can fall
              either side of this threshold.

       <b>-T</b> <b>|</b> <b>--targettype</b> <b>&lt;dna</b> <b>|</b> <b>protein&gt;</b>
              Specify the target type to use.  The same as <b>--querytype</b> (above), except that it  applies  to  the
              target.  Specifying the sequence type will avoid the overhead of having to read the first sequence
              in the database twice (which may be significant with chromosome-sized sequences)

       <b>--querychunkid</b> <b>&lt;id&gt;</b>

       <b>--querychunktotal</b> <b>&lt;total&gt;</b>

       <b>--targetchunkid</b> <b>&lt;id&gt;</b>

       <b>--targetchunktotal</b> <b>&lt;total&gt;</b>
              These  options  to  facilitate  running  exonerate  on compute farms, and avoid having to split up
              sequence databases into small chunks to run on different nodes.  If, for example,  you  wished  to
              split  the target database into three parts, you would run three exonerate jobs on different nodes
              including the options:

              --targetchunkid 1 --targetchunktotal 3
              --targetchunkid 2 --targetchunktotal 3
              --targetchunkid 3 --targetchunktotal 3
              NB. The granularity offered by this option only goes down to a single sequence, so when there  are
              more chunks than sequences in the database, some processes will do nothing.

       <b>-V</b> <b>|</b> <b>--verbose</b> <b>&lt;int&gt;</b>
              Be  verbose  -  show  information  about  what  is going on during the analysis.  The default is 1
              (little information), the higher the number given, the more information is  printed.   To  silence
              all the default output from exonerate, use --verbose 0 --showalignment no --showvulgar no

</pre><h4><b>ANALYSIS</b> <b>OPTIONS</b></h4><pre>
       <b>-E</b> <b>|</b> <b>--exhaustive</b> <b>&lt;boolean&gt;</b>
              Specify  whether  or  not  exhaustive  alignment  should  be used.  By default, this is FALSE, and
              alignment heuristics will be used.  If it  is  set  to  TRUE,  an  exhaustive  alignment  will  be
              calculated.   This  requires  quadratic  time, and will be much, much slower, but will provide the
              optimal result for the given model.
       <b>-B</b> <b>|</b> <b>--bigseq</b> <b>&lt;int&gt;</b>
              Perform alignment of large (multi-megabase) sequences.  This is very  memory  efficient  and  fast
              when  both  sequences  are  chromosome-sized, but currently does not currently permit the use of a
              word neighbourhood (ie. exactly matching seeds only).
       <b>--revcomp</b> <b>&lt;boolean&gt;</b>
              Include comparison of the reverse complement of the query and target where possible.  By  default,
              this  option  is  enabled,  but  when you know the gene is definitely on the forward strand of the
              query and target, this option can halve the time taken to compute alignments.
       <b>--forcescan</b> <b>&lt;none</b> <b>|</b> <b>query</b> <b>|</b> <b>target&gt;</b>
              Force the FSM to scan the query sequence rather than the  target.   This  option  is  useful,  for
              example, if you have a single piece of genomic sequence and you with to compare it to the whole of
              dbEST.   By scanning the database, rather than the query, the analysis will be completed much more
              quickly, as the overheads of multiple query FSM construction, multiple target reading  and  splice
              site  predictions will be removed.  By default, exonerate will guess the optimal strategy based on
              database sequence sizes.
       <b>--saturatethreshold</b> <b>&lt;number&gt;</b>
              When set to zero, this option does nothing.  Otherwise, once more than this number  of  words  (in
              addition  to  the  expected  number  of words by chance) have matched a position on the query, the
              position on the query  will  be  'numbed'  (ignore  further  matches)  for  the  current  pairwise
              comparison.
       <b>--customserver</b> <b>&lt;command&gt;</b>
              When  using exonerate in client:server mode with a non-standard server, this command allows you to
              send a custom command to the server.  This command is sent by the client  (exonerate)  before  any
              other  commands,  and is provided as a way of passing parameters or other commands specific to the
              custom server.  See the <b>exonerate-server</b> man page for more information  on  running  exonerate  in
              client:server mode.
       <b>--cores</b> <b>&lt;number&gt;</b>
              The  number  of  cores/CPUs/threads  that  should  be used.  On a multi-core or multi-CPU machine,
              increasing this ammount allows alignment computations to run in parallel on  separate  CPUs/cores.
              NB.   Generally,  it  is better to parallelise the analysis by splitting it up into separate jobs,
              but this option may prove useful for problems such as interactive single-gene queries.

</pre><h4><b>FASTA</b> <b>DATABASE</b> <b>OPTIONS</b></h4><pre>
       <b>--fastasuffix</b> <b>&lt;extension&gt;</b>
              If any of the inputs  given  with  <b>--query</b>  or  <b>--target</b>  are  directories,  then  exonerate  will
              recursively  descent  these directories, reading all files ending with this suffix as fasta format
              input.

</pre><h4><b>GAPPED</b> <b>ALIGNMENT</b> <b>OPTIONS</b></h4><pre>
       <b>-m</b> <b>|</b> <b>--model</b> <b>&lt;alignment</b> <b>model&gt;</b>
              Specify the alignment model to use.  The models currently supported are:
              <b>ungapped</b>
                     The simplest type of model, used  by  default.   An  appropriate  model  with  be  selected
                     automatically for the type of input sequences provided.
              <b>ungapped:trans</b>
                     This  ungapped  model  includes  translation  of  all  frames  of both the query and target
                     sequences.  This is similar to an ungapped tblastx type search.
              <b>affine:global</b>
                     This performs gapped global alignment, similar to the  Needleman-Wunsch  algorithm,  except
                     with  affine gaps.  Global alignment requires that both the sequences in their entirety are
                     included in the alignment.
              <b>affine:bestfit</b>
                     This performs a best fit or best location alignment of the query onto the target  sequence.
                     The entire query sequence will be included in the alignment, but only the best location for
                     its alignment on the target sequence.
              <b>affine:local</b>
                     This is local alignment with affine gaps, similar to the Smith-Waterman-Gotoh algorithm.  A
                     general-purpose  alignment  algorithm.   As this is local alignment, any subsequence of the
                     query and target sequence may appear in the alignment.
              <b>affine:overlap</b>
                     This type of alignment finds the best overlap between the query and  target.   The  overlap
                     alignment  must  include  the  start of the query or target and the end of the query or the
                     target sequence, to align sequences which overlap at the ends, or in the mid-section  of  a
                     longer sequence..  This is the type of alignment frequently used in assembly algorithms.
              <b>est2genome</b>
                     This  model  is similar to the affine:local model, but it also includes intron modelling on
                     the target sequence to allow alignment of spliced to unspliced coding  sequences  for  both
                     forward  and reversed genes.  This is similar to the alignment models used in programs such
                     as EST_GENOME and sim4.
              <b>ner</b>    NERs are non-equivalenced regions - large regions in both the query and  target  which  are
                     not  aligned.  This model can be used for protein alignments where strongly conserved helix
                     regions will be aligned, but weakly conserved loop regions are not.  Similarly, this  model
                     could be used to look for co-linearly conserved regions in comparison of genomic sequences.
              <b>protein2dna</b>
                     This model compares a protein sequence to a DNA sequence, incorporating all the appropriate
                     gaps and frameshifts.
              <b>protein2dna:bestfit</b>
                     This  is  a  bestfit  version  of  the  protein2dna model, with which the entire protein is
                     included in the alignment.  It is currently only available when using exhaustive alignment.
              <b>protein2genome</b>
                     This model allows alignment of a protein sequence to genomic DNA.   This is similar to  the
                     protein2dna model, with the addition of modelling of introns and intron phases.  This model
                     is similar to those used by genewise.
              <b>protein2genome:bestfit</b>
                     This  is  a  bestfit  version of the protein2genome model, with which the entire protein is
                     included in the alignment.  It is currently only available when using exhaustive alignment.
              <b>coding2coding</b>
                     This model is similar to the ungapped:trans model, except that  gaps  and  frameshifts  are
                     allowed.  It is similar to a gapped tblastx search.
              <b>coding2genome</b>
                     This  is  similar  to  the  est2genome  model, except that the query sequence is translated
                     during comparison, allowing a more sensitive comparison.
              <b>cdna2genome</b>
                     This combines properties of the est2genome and coding2genome models, to allow  modeling  of
                     an  whole  cDNA  where a central coding region can be flanked by non-coding UTRs.  When the
                     CDS start and end is known it may be specified using the --annotation option (see below) to
                     permit only the correct coding region to appear in the alignemnt.
              <b>genome2genome</b>
                     This model is similar to the coding2coding model,  except  introns  are  modelled  on  both
                     sequences.  (not working well yet)

       The short names u, u:t, a:g, a:b, a:l, a:o, e2g, ner,
              p2d, p2d:b p2g, p2g:b, c2c, c2g cd2g and g2g can also be used for specifying models.

       <b>-s</b> <b>|</b> <b>--score</b> <b>&lt;threshold&gt;</b>
              This  is  the overall score threshold.  Alignments will not be reported below this threshold.  For
              heuristic alignments, the higher this threshold, the less time the analysis will take.

       <b>--percent</b> <b>&lt;percentage&gt;</b>
              Report only alignments scoring at least this percentage of the maximal score for each query.   eg.
              use  <b>--percent</b>  <b>90</b>  to  report alignments with 90% of the maximal score optainable for that query.
              This option is useful not only because it reduces the spurious matches in the output, but  because
              it  generates  query-specific  thresholds  (unlike  <b>--score</b>  )  for  a set of queries of differing
              lengths, and will also speed up the search considerably.  <b>NB.</b>  with this option, it is possible to
              have a cDNA match its corresponding gene exactly, yet still score  less  than  100%,  due  to  the
              addition of the intron penalty scores, hence this option must be used with caution.

       <b>--showalignment</b> <b>&lt;boolean&gt;</b>
              Show the alignments in an human readable form.

       <b>--showsugar</b> <b>&lt;boolean&gt;</b>
              Display  "sugar" output for ungapped alignments.  Sugar is Simple UnGapped Alignment Report, which
              displays ungapped alignments one-per-line.  The sugar line starts with  the  string  "sugar:"  for
              easy extraction from the output, and is followed by the the following 9 fields in the order below:

              <b>query_id</b>        Query identifier
              <b>query_start</b>     Query position at alignment start
              <b>query_end</b>       Query position alignment end
              <b>query_strand</b>    Strand of query matched
              <b>target_id</b>       |
              <b>target_start</b>    | the same 4 fields
              <b>target_end</b>      | for the target sequence
              <b>target_strand</b>   |
              <b>score</b>           The raw alignment score

       <b>--showcigar</b> <b>&lt;boolean&gt;</b>
              Show  the alignments in "cigar" format.  Cigar is a Compact Idiosyncratic Gapped Alignment Report,
              which displays gapped alignments one-per-line.  The format starts with the same 9 fields as  sugar
              output  (see  above),  and is followed by a series of &lt;operation, length&gt; pairs where operation is
              one of match, insert or delete, and the length describes the number of  times  this  operation  is
              repeated.

       <b>--showvulgar</b> <b>&lt;boolean&gt;</b>
              Shows  the  alignments  in  "vulgar"  format.   Vulgar is Verbose Useful Labelled Gapped Alignment
              Report, This format also starts with the same 9  fields  as  sugar  output  (see  above),  and  is
              followed  by  a  series of &lt;label, query_length, target_length&gt; triplets.  The label may be one of
              the following:

              <b>M</b>      Match
              <b>C</b>      Codon
              <b>G</b>      Gap
              <b>N</b>      Non-equivalenced region
              <b>5</b>      5' splice site
              <b>3</b>      3' splice site
              <b>I</b>      Intron
              <b>S</b>      Split codon
              <b>F</b>      Frameshift

       <b>--showquerygff</b> <b>&lt;boolean&gt;</b>
              Report     GFF     output     for     features     on      the      query      sequence.       See
              <a href="http://www.sanger.ac.uk/Software/formats/GFF">http://www.sanger.ac.uk/Software/formats/GFF</a> for more information.

       <b>--showtargetgff</b> <b>&lt;boolean&gt;</b>
              Report GFF output for features on the target sequence.

       <b>--ryo</b> <b>&lt;format&gt;</b>
              Roll-your-own  output  format.   This  allows specification of a printf-esque format line which is
              used to specify which information to include in the output, and how it is to be shown.  The format
              field may contain the following fields:

              <b>%[qt][idlsSt]</b>
                     For either {query,target}, report the {id,definition,length,sequence,Strand,type} Sequences
                     are reported in a fasta-format like block (no headers).
              <b>%[qt]a[bels]</b>
                     For  either  {query,target}  region   which   occurs   <b>in</b>   <b>the</b>   <b>alignment,</b>   report   the
                     {begin,end,length,sequence}
              <b>%[qt]c[bels]</b>
                     For  either  {query,target}  region  which  occurs in the <b>coding</b> <b>sequence</b> in the alignment,
                     report the {begin,end,length,sequence}
              <b>%s</b>     The raw score
              <b>%r</b>     The rank (in results from a bestn search)
              <b>%m</b>     Model name
              <b>%e[tism]</b>
                     Equivalenced {total,id,similarity,mismatches} (ie. %em == (%et - %ei))
              <b>%p[isS]</b>
                     Percent {id,similarity,Self} over the equivalenced portions of the alignment.  (ie. %pi  ==
                     100*(%ei  /  %et)).   Percent  Self  is  the  score  over  the equivalenced portions of the
                     alignment as a percentage of the self comparison score of the query sequence.
              <b>%g</b>     Gene orientation ('+' = forward, '-' = reverse, '.' = unknown)
              <b>%S</b>     Sugar block (the 9 fields used in sugar output (see above)
              <b>%C</b>     Cigar block (the fields of a cigar line after the sugar portion)
              <b>%V</b>     Vulgar block (the fields of a vulgar line after the sugar portion)
              <b>%%</b>     Expands to a percentage sign (%)
              <b>\n</b>     Newline
              <b>\t</b>     Tab
              <b>\\</b>     Expands to a backslash (\)
              <b>\{</b>     Open curly brace
              <b>\}</b>     Close curly brace
              <b>{</b>      Begin per-transition output section
              <b>}</b>      End per-transition output section
              <b>%P[qt][sabe]</b>
                     Per-transition output for {query,target} {sequence,advance,begin,end}
              <b>%P[nsl]</b>
                     Per-transition output for {name,score,label}

       This option is very useful and flexible.  For example, to report all  the  sections  of  query  sequences
       which feature in alignments in fasta format, use:

       --ryo <b>"&gt;%qi</b> <b>%qd\n%qas\n"</b>

       To output all the symbols and scores in an alignment, try something like:

       --ryo <b>"%V{%Pqs</b> <b>%Pts</b> <b>%Ps\n}"</b>

       <b>-n</b> <b>|</b> <b>--bestn</b> <b>&lt;number&gt;</b>
              Report the best N results for each query.  (Only results scoring better than the score threshold
               will  be reported).  The option reduces the amount of output generated, and also allows exonerate
              to speed up the search.

       <b>-S</b> <b>|</b> <b>--subopt</b> <b>&lt;boolean&gt;</b>
              This option allows for the reporting of (Waterman-Eggert style) suboptimal alignments.  (It is  on
              by  default.)   All suboptimal (ie. non-intersecting) alignments will be reported for each pair of
              sequences scoring at least the threshold provided by <b>--score.</b>

              When this option is used with exhaustive alignments, several full quadratic time  passes  will  be
              required, so the running time will be considerably increased.

       <b>-g</b> <b>|</b> <b>--gappedextension</b> <b>&lt;boolean&gt;</b>
              Causes  a gapped extension stage to be performed ie. dynamic programming is applied in arbitrarily
              shaped and dynamically sized regions surrounding HSP seeds.  The extension threshold is controlled
              by the --extensionthreshold option.

              Although sometimes slower than BSDP, gapped extension improves  sensitivity  with  weak,  gap-rich
              alignments such as during cross-species comparison.

              <b>NB.</b>  <b>This</b>  <b>option</b>  <b>is</b> <b>now</b> <b>the</b> <b>default.</b> <b>Set</b> <b>it</b> <b>to</b> <b>false</b> to reverse to the old BSDP type alignments.
              This option may be slower than BSDP for some large scale analyses with simple alignment models.

       <b>--refine</b> <b>&lt;strategy&gt;</b>
              Force exonerate to refine alignments generated by heuristics using dynamic programming over larger
              regions.  This takes more time, but improves the quality of the final alignments.

              The strategies available for refinement are:

              <b>none</b>   The default - no refinement is used.
              <b>full</b>   An exhaustive alignment is calculated from the pair of sequences in their entirety.
              <b>region</b> DP is applied just to the region of the sequences covered by the heuristic alignment.

       <b>--refineboundary</b> <b>&lt;size&gt;</b>
              Specify an extra boundary to be included in the region subject to alignment during  refinement  by
              region.

</pre><h4><b>VITERBI</b> <b>ALGORITHM</b> <b>OPTIONS</b></h4><pre>
       <b>-D</b> <b>|</b> <b>--dpmemory</b> <b>&lt;Mb&gt;</b>
              The  exhaustive  alignment  traceback  routines use a Hughey-style reduced memory technique.  This
              option specifies how much memory will be used for this.  Generally, the more memory  is  permitted
              here, the faster the alignments will be produced.

</pre><h4><b>CODE</b> <b>GENERATION</b> <b>OPTIONS</b></h4><pre>
       <b>-C</b> <b>|</b> <b>--compiled</b> <b>&lt;boolean&gt;</b>
              This  option allows disabling of generated code for dynamic programming.  It is mainly used during
              development of exonerate.  When set to FALSE, an "interpreted" version of the dynamic  programming
              implementation is used, which is much slower.

</pre><h4><b>HEURISTIC</b> <b>OPTIONS</b></h4><pre>
       <b>--terminalrangeint</b>
       <b>--terminalrangeext</b>
       <b>--joinrangeint</b>
       <b>--joinrangeext</b>
       <b>--spanrangeint</b>
       <b>--spanrangeext</b>
              These  options  are  used  to specify the size of the sub-alignment regions to which DP is applied
              around the ends of the HSPs.  This can be at the HSP ends (terminal  range),  between  HSPs  (join
              range),  or  between  HSPs  which  may  be  connected  by a large region such as an intron or non-
              equivalenced region (span range).  These ranges can be specified for a number of matches back onto
              the HSP (internal range) or out from the HSP (external range).

</pre><h4><b>SEEDED</b> <b>DYNAMIC</b> <b>PROGRAMMING</b> <b>OPTIONS</b></h4><pre>
       <b>-x</b> <b>|</b> <b>--extensionthreshold</b> <b>&lt;score&gt;</b>
              This is the amount by which the score will  be  allowed  to  degrade  during  SDP.   This  is  the
              equivalent  of  the hspdropoff penalties, except it is applied during dynamic programming, not HSP
              extension.  Decreasing this parameter will increase the speed of the SDP, and increasing  it  will
              increase the sensitivity.

       <b>--singlepass</b>  <b>&lt;boolean&gt;</b>
              By default the suboptimal SDP alignments are reported by a singlepass algorithm, but may miss some
              suboptimal  alignments  that  are  close  together.  This option can be used to force the use of a
              multipass  suboptimal  alignment  algorithm  for  SDP,  resulting  in  higher  quality  suboptimal
              alignments.

</pre><h4><b>BSDP</b> <b>OPTIONS</b></h4><pre>
       <b>--joinfilter</b> <b>&lt;limit&gt;</b>
              (experimental)

              Only  allow  consider  this  number  of SARs for joining HSPs together.  The SARs with the highest
              potential for appearing in a high-scoring  alignment  are  considered.   This  option  useful  for
              limiting  time and memory usage when searching unmasked data with repetitive sequences, but should
              not be set too low, as valid matches may be ignored.  Something like <b>--joinfilter</b> <b>32</b> seems to work
              well.

</pre><h4><b>SEQUENCE</b> <b>OPTIONS</b></h4><pre>
       <b>--annotation</b> <b>&lt;path&gt;</b>
              Specify basic sequence annotation information.  This is most useful with  the  cdna2genome  model,
              but will work with other models.  The annotation file contains four fields per line:

              &lt;id&gt; &lt;strand&gt; &lt;cds_start&gt; &lt;cds_length&gt;

              Here is a simple example of such a file for 4 cDNAs:

              dhh.human.cdna + 308 1191
              dhh.mouse.cdna + 250 1191
              csn7a.human.cdna + 178 828
              csn7a.mouse.cdna + 126 828
              These  annotation  lines will also work when only the first two fields are used.  This can be used
              when specifying which strand of a specific sequence should be included in a comparison.

</pre><h4><b>SYMBOL</b> <b>COMPARISON</b> <b>OPTIONS</b></h4><pre>
       <b>--softmaskquery</b> <b>&lt;boolean&gt;</b>
              Indicate that the query is softmasked.  See description below for <b>--softmasktarget</b>
       <b>--softmasktarget</b> <b>&lt;boolean&gt;</b>
              Indicate that the target is softmasked.  In a softmasked sequence file, instead of masking regions
              by Ns or Xs they are masked by putting those regions in lower case (and with unmasked  regions  in
              upper case).  This option allows the masking to be ignored by some parts of the program, combining
              the  speed  of  searching  masked  data  with sensitivity of searching unmasked data.  The utility
              <b>fastasoftmask</b> supplied which is supplied with exonerate  can  be  used  for  producing  softmasked
              sequence from conventionally masked sequence.
       <b>-d</b> <b>|</b> <b>--dnasubmat</b> <b>&lt;name&gt;</b>
              Specify  the  the  substitution  matrix to be used for DNA comparison.  This should be a path to a
              substitution matrix in same format as that which is used by blast.
       <b>-p</b> <b>|</b> <b>--proteinsubmat</b> <b>&lt;name&gt;</b>
              Specify the the substitution matrix to be used for protein  comparison.   (Both  DNA  and  protein
              substitution  matrices  are  required  for some types of analysis).  The use of the special names,
              <b>nucleic,</b> <b>blosum62,</b> <b>pam250,</b> <b>edit</b> or <b>identity</b> will cause built-in substitution matrices to be used.
</pre><h4><b>ALIGNMENT</b> <b>SEEDING</b> <b>OPTIONS</b></h4><pre>
       <b>-M</b> <b>|</b> <b>--fsmmemory</b> <b>&lt;Mb&gt;</b>
              Specify the amount of memory to use for the FSM in heuristic analyses.  exonerate multiplexes  the
              query to accelerate large-throughput database queries.  This figure should always be less than the
              physical  memory on the machine, but when searching large databases, generally, the more memory it
              is allowed to use, the faster it will go.
       <b>--forcefsm</b> <b>&lt;none</b> <b>|</b> <b>normal</b> <b>|</b> <b>compact&gt;</b>
              Force the use of more compact finite state machines for analyses involving big sequences and large
              word neighbourhoods.  By default, exonerate will pick a sensible strategy,  so  this  option  will
              rarely need to be set.
       <b>--wordjump</b> <b>&lt;int&gt;</b>
              The  jump  between  query  words used to yield the word neighbourhood.  If set to 1, every word is
              used, if set to 2, every other word is used, and if set to the  wordlength,  only  non-overlapping
              words  will  be  used.   This  option  reduces the memory requirements when using very large query
              sequences, and makes the search run faster, but it  also  damages  search  sensitivity  when  high
              values are set.
       <b>--wordambiguity</b> <b>&lt;limit&gt;</b>
              This option may be used to allow alignment seeds containing IUPAC ambiguity symbols.  The limit is
              the maximum number of ambiguous words allowed at a single position.  If this limit is reached then
              the  position  is  not used for alignment seeding.  Using this option may slow down a search.  For
              large datasets, it is recommended to use  <b>esd2esi</b>  <b>--wordambiguity</b>  instead,  as  then  the  speed
              overhead  is only incurred during indexing, rather than during the database searching itself.  NB.
              This option only works for IUPAC symbols in the target sequence.   Query  words  containing  IUPAC
              symbols are (currently) excluded from seeding.
</pre><h4><b>AFFINE</b> <b>MODEL</b> <b>OPTIONS</b></h4><pre>
       <b>-o</b> <b>|</b> <b>--gapopen</b> <b>&lt;penalty&gt;</b>
              This is the gap open penalty.
       <b>-e</b> <b>|</b> <b>--gapextend</b> <b>&lt;penalty&gt;</b>
              This is the gap extension penalty.
       <b>--codongapopen</b> <b>&lt;penalty&gt;</b>
              This is the codon gap open penalty.
       <b>--codongapextend</b> <b>&lt;penalty&gt;</b>
              This is the codon gap extension penalty.
</pre><h4><b>NER</b> <b>OPTIONS</b></h4><pre>
       <b>--minner</b> <b>&lt;boolean&gt;</b>
              Minimum NER length allowed.
       <b>--maxner</b> <b>&lt;length&gt;</b>
              Maximum NER length allowed.  NB. this option only affects heuristic alignments.
       <b>--neropen</b> <b>&lt;penalty&gt;</b>
              Penalty for opening a non-equivalenced region.
</pre><h4><b>INTRON</b> <b>MODELLING</b> <b>OPTIONS</b></h4><pre>
       <b>--minintron</b> <b>&lt;length&gt;</b>
              Minimum  intron  length  limit.  NB. this option only affects heuristic alignments.  This is not a
              hard limit - it only affects size of introns which are sought during heuristic alignment.
       <b>--maxintron</b> <b>&lt;length&gt;</b>
              Maximum intron length limit.  See notes above for <b>--minintron</b>
       <b>-i</b> <b>|</b> <b>--intronpenalty</b> <b>&lt;penalty&gt;</b>
              Penalty for introduction of an intron.
</pre><h4><b>FRAMESHIFT</b> <b>MODELLING</b> <b>OPTIONS</b></h4><pre>
       <b>-f</b> <b>|</b> <b>--frameshift</b> <b>&lt;penalty&gt;</b>
              The penalty for the inclusion of a frameshift in an alignment.
</pre><h4><b>ALPHABET</b> <b>OPTIONS</b></h4><pre>
       <b>--useaatla</b> <b>&lt;boolean&gt;</b>
              Use three-letter abbreviations for AA names.  ie. when displaying alignment "Met" is used  instead
              of " M "
</pre><h4><b>TRANSLATION</b> <b>OPTIONS</b></h4><pre>
       <b>--geneticcode</b> <b>&lt;code&gt;</b>
              Specify  an  alternative  genetic code.  The default code (1) is the standard genetic code.  Other
              genetic codes may be specified by in shorthand or longhand form.

              In shorthand form, a number between 1 and 23 is used to specify one of 17  built-in  genetic  code
              variants.  These are genetic code variants taken from:

              <b><a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi">http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi</a></b>

              These are:
              <b>1</b>      The Standard Code
              <b>2</b>      The Vertebrate Mitochondrial Code
              <b>3</b>      The Yeast Mitochondrial Code
              <b>4</b>      The  Mold,  Protozoan,  and  Coelenterate Mitochondrial Code and the Mycoplasma/Spiroplasma
                     Code
              <b>5</b>      The Invertebrate Mitochondrial Code
              <b>6</b>      The Ciliate, Dasycladacean and Hexamita Nuclear Code
              <b>9</b>      The Echinoderm and Flatworm Mitochondrial Code
              <b>10</b>     The Euplotid Nuclear Code
              <b>11</b>     The Bacterial and Plant Plastid Code
              <b>12</b>     The Alternative Yeast Nuclear Code
              <b>13</b>     The Ascidian Mitochondrial Code
              <b>14</b>     The Alternative Flatworm Mitochondrial Code
              <b>15</b>     Blepharisma Nuclear Code
              <b>16</b>     Chlorophycean Mitochondrial Code
              <b>21</b>     Trematode Mitochondrial Code
              <b>22</b>     Scenedesmus obliquus mitochondrial Code
              <b>23</b>     Thraustochytrium Mitochondrial Code",
              In longhand form, a genetic code variant may be provided as a 64 byte string in  TCAG  order,  eg.
              the standard genetic code in this form would be:

              FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG

</pre><h4><b>HSP</b> <b>CREATION</b> <b>OPTIONS</b></h4><pre>
       <b>--hspfilter</b> <b>&lt;threshold&gt;</b>
              Use  aggressive  HSP filtering to speed up heuristic searches.  The threshold specifies the number
              of HSPs centred about a point in the query which will be stored.  Any lower scoring HSPs  will  be
              discarded.   This  is an experimental option to handle speed problems caused by some sequences.  A
              value of about 100 seems to work well.
       <b>--useworddropoff</b> <b>&lt;boolean&gt;</b>
              When this is TRUE, the score threshold for admitting words into the word neighbourhood is  set  to
              be  the  initial  word  score  minus the word threshold (see below).  This strategy is designed to
              prevent restricting the word SSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG When this
              is FALSE, the word threshold is taken to be an absolute value.
       <b>--seedrepeat</b> <b>&lt;count&gt;</b>
              The seedrepeat parameter sets the number of seeds which must be found  on  the  same  diagonal  or
              reading  frame  before HSP extension will occur.  Increasing the value for <b>--seedrepeat</b> will speed
              up searches, and is usually a better option than using  longer  word  lengths,  particularly  when
              using  the  <b>exonerate-server</b>  where  increasing  word  lengths requires recomputing the index, and
              greater increases memory requirements.
       <b>-w</b> <b>--dnawordlen</b> <b>&lt;bases&gt;</b>
       <b>-W</b> <b>--proteinwordlen</b> <b>&lt;residues&gt;</b>
       <b>-W</b> <b>--codonnwordlen</b> <b>&lt;bases&gt;</b>
              The word length used for DNA, protein or codon words.  When performing DNA vs protein comparisons,
              a the DNA wordlength will always (automatically) be triple the protein wordlength.
       <b>--dnahspdropoff</b> <b>&lt;score&gt;</b>
       <b>--proteinhspdropoff</b> <b>&lt;score&gt;</b>
       <b>--codonhspdropoff</b> <b>&lt;score&gt;</b>
              The amount by which an HSP score will be  allowed  to  degrade  during  HSP  extension.   Separate
              threshold can be set for dna or protein comparisons.
       <b>--dnahspthreshold</b> <b>&lt;score&gt;</b>
       <b>--proteinhspthreshold</b> <b>&lt;score&gt;</b>
       <b>--codonhspthreshold</b> <b>&lt;score&gt;</b>
              The  HSP  score thresholds.  An HSP must score at least this much before it will be reported or be
              used in preparation of a heuristic alignment.
       <b>--dnawordlimit</b>  <b>&lt;score&gt;</b>
       <b>--proteinwordlimit</b>  <b>&lt;score&gt;</b>
       <b>--codonwordlimit</b>  <b>&lt;score&gt;</b>
              The threshold for admitting DNA or protein words into the word neighbourhood.   The  behaviour  of
              this option is altered by the <b>--useworddropoff</b> option (see above).

       <b>--geneseed</b> <b>&lt;threshold&gt;</b>
              Exclude  HSPs  from gapped alignment computation which cannot feature in a alignment containing at
              least one HSP scoring at least this threshold.

              This option provides considerable speed up for gapped alignment computation, but  may  cause  some
              very gap-rich alignments to be missed.

              It is useful when aligning similar sequences back onto genome quickly, eg. try --geneseed 250
       <b>--geneseedrepeat</b> <b>&lt;count&gt;</b>
              The  geneseedrepeat  parameter  is like the seedrepeat parameter, but is only applied when looking
              for the geneseed hsps.  Using a larger value for <b>--geneseedrepeat</b> will speed up searches when  the
              <b>--geneseed</b> parameter is also used.  (experimental, implementation incomplete)
</pre><h4><b>ALIGNMENT</b> <b>OPTIONS</b></h4><pre>
       <b>--alignmentwidth</b> <b>&lt;width&gt;</b>
              Width of alignment display.  The default is 80.
       <b>--forwardcoordinates</b> <b>&lt;boolean&gt;</b>
              By  default,  all  coordinates  are  reported on the forward strand.  Setting this option to false
              reverts to the old behaviour (pre-0.8.3)  whereby  alignments  on  the  reverse  complement  of  a
              sequence are reported using coordinates on the reverse complement.
</pre><h4><b>SUB-ALIGNMENT</b> <b>REGION</b> <b>OPTIONS</b></h4><pre>
       <b>--quality</b> <b>&lt;percent&gt;</b>
              This  option  excludes  HSPs  from  BSDP when their components outside of the SARs fall below this
              quality threshold.
</pre><h4><b>SPLICE</b> <b>SITE</b> <b>PREDICTION</b> <b>OPTIONS</b></h4><pre>
       <b>--splice3</b> <b>&lt;path&gt;</b>
       <b>--splice5</b> <b>&lt;path&gt;</b>
              Provide a file containing a custom PSSM (position specific score matrix)  for  prediction  of  the
              intron splice sites.

              The  file  format  for  splice  data  is  simple:  lines  beginning  with ´#´ are comments, a line
              containing just the word ´splice´ denotes the position of the splice site,  and  the  other  lines
              show  the  observed  relative  frequencies  of  the  bases flanking the splice sites in the chosen
              organism (in ACGT order).

              <b>Example</b> <b>5'</b> <b>splice</b> <b>data</b> <b>file:</b>

               # start of example 5' splice data
               # A C G T
               28 40  17  14
               59 14  13  14
                8  5  81   6
               splice
                0  0 100   0
                0  0   0 100
               54  2  42   2
               74  8  11   8
                5  6  85   4
               16 18  21  45
               # end of test 5' splice data

              <b>Example</b> <b>3'</b> <b>splice</b> <b>data</b> <b>file:</b>

               # start of example 3' splice data
               # A C G T
                10  31  14  44
                 8  36  14  43
                 6  34  12  48
                 6  34   8  52
                 9  37   9  45
                 9  38  10  44
                 8  44   9  40
                 9  41   8  41
                 6  44   6  45
                 6  40   6  48
                23  28  26  23
                 2  79   1  18
               100   0   0   0
                 0   0 100   0
               splice
                28  14  47  11
               # end of example 3' splice data

       <b>--forcegtag</b> <b>&lt;boolean&gt;</b>
              Only allow splice sites at gt....ag sites (or ct....ac sites when the gene is reversed) With  this
              restriction in place, the splice site prediction scores are still used and allow tie breaking when
              there is more than one possible splice site.

</pre><h4><b>STRATEGIES</b> <b>FOR</b> <b>SPEED</b></h4><pre>
       Keep all data on local disks.

       Apply the highest acceptable score thresholds using a combination of --score, --percent and --bestn.

       Repeat mask and dust the genomic (target) sequence.  (Softmask these sequences and use --softmasktarget).

       Increase the --fsmmemory option to allow more query multiplexing.

       Increase the value for --seedrepeat

       When using an alignment model containing introns, set --geneseed as high as possible.

       If you are compiling exonerate yourself, see the README file supplied with the source code for details of
       compile-time optimisations.

</pre><h4><b>STRATEGIES</b> <b>FOR</b> <b>SENSITIVITY</b></h4><pre>
       Not documented yet.

       Increase   the  word  neighbourhood.   Decrease  the  HSP  threshold.   Increase  the  SAR  ranges.   Run
       exhaustively.

</pre><h4><b>ENVIRONMENT</b></h4><pre>
       Not documented yet.

</pre><h4><b>EXAMPLES</b></h4><pre>
       <b>exonerate</b> <b>cdna.fasta</b> <b>genomic.fasta</b>
              This simplest way in which exonerate may be used.  By default, an ungapped alignment model will be
              used.

       <b>exonerate</b> <b>--exhaustive</b> <b>y</b> <b>--model</b> <b>est2genome</b> <b>cdna.fasta</b> <b>genomic.masked.fasta</b>
              Exhaustively align cdnas to genomic sequence.  This will be much, much slower, but more  accurate.
              This option causes exonerate to behave like EST_GENOME.

       <b>exonerate</b> <b>--exhaustive</b> <b>--model</b> <b>affine:local</b> <b>query.fasta</b> <b>target.fasta</b>
              If  the  affine:local  model  is  used  with  exhaustive  alignment,  you  have the Smith-Waterman
              algorithm.

       <b>exonerate</b> <b>--exhaustive</b> <b>--model</b> <b>affine:global</b> <b>protein.fasta</b> <b>protein.fasta</b>
              Switch to a global model, and you have Needleman-Wunsch.

       <b>exonerate</b> <b>--wordthreshold</b> <b>1</b> <b>--gapped</b> <b>no</b> <b>--showhsp</b> <b>yes</b> <b>protein.fasta</b> <b>genome.fasta</b>
              Generate ungapped Protein:DNA alignments

       <b>exonerate</b> <b>--model</b> <b>coding2coding</b> <b>--score</b> <b>1000</b> <b>--bigseq</b> <b>yes</b> <b>--proteinhspthreshold</b> <b>90</b> <b>chr21.fa</b> <b>chr22.fa</b>
              Perform quick-and-dirty translated pairwise alignment of two very large DNA sequences.

       Many similar combinations should work.  Try them out.

</pre><h4><b>VERSION</b></h4><pre>
       This documentation accompanies version 2.2.0 of the exonerate package.
</pre><h4><b>AUTHOR</b></h4><pre>
       Guy St.C. Slater.  &lt;<a href="mailto:guy@ebi.ac.uk">guy@ebi.ac.uk</a>&gt;.
       See the AUTHORS file accompanying the source code for a list of contributors.
</pre><h4><b>AVAILABILITY</b></h4><pre>
       This source code for the exonerate package is available  under  the  terms  of  the  GNU  <u>general</u>  public
       licence.

       Please     see    the    file    COPYING    which    was    distrubuted    with    this    package,    or
       <a href="http://www.gnu.org/licenses/gpl.txt">http://www.gnu.org/licenses/gpl.txt</a> for details.

       This package has been developed as part of the ensembl project.  Please see  <a href="http://www.ensembl.org/">http://www.ensembl.org/</a>  for
       more information.
</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/exonerate-server.1.html">exonerate-server</a></b>(1), <b><a href="../man1/ipcress.1.html">ipcress</a></b>(1), <b><a href="../man1L/blast.1L.html">blast</a></b>(1L).

exonerate                                         November 2002                                     <u><a href="../man1/exonerate.1.html">exonerate</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>