<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>glam2-purge - Removes redundant sequences from a FASTA file</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/glam2">glam2_1064-10_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       glam2-purge - Removes redundant sequences from a FASTA file

</pre><h4><b>SYNOPSIS</b></h4><pre>

       <b>glam2-purge</b> <u>file</u> <b>score</b> [<b>options</b>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>glam2-purge</b> is a modified version of Andrew Neuwald´s <b>purge</b> program that removes redundant sequences from
       a FASTA file. This is recommended in order to prevent highly similar sequences distorting the search for
       motifs. Purge works with either DNA or protein sequences and creates an output file such that no two
       sequences have a (gapless) local alignment score greater than a threshold specified by the user. The
       output file is named &lt;file&gt;.&lt;score&gt;. The alignment score is based on the BLOSUM62 matrix for proteins,
       and on a +5/-1 scoring scheme for DNA. Purge can also be used to mask tandem repeats. It uses the XNU
       program for this purpose.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-n</b>
           Sequences are DNA (default: protein).

       <b>-b</b>
           Use blast heuristic method (default for protein).

       <b>-e</b>
           Use an exhaustive method (default for DNA).

       <b>-q</b>
           Keep first sequence in the set.

       <b>-x</b>
           Use xnu to mask protein tandem repeats.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/glam2.1.html">glam2</a></b>(1), <b><a href="../man1/glam2format.1.html">glam2format</a></b>(1), <b><a href="../man1/glam2mask.1.html">glam2mask</a></b>(1), <b><a href="../man1/glam2scan.1.html">glam2scan</a></b>(1), <b><a href="../man1/xnu.1.html">xnu</a></b>(1)

       The full Hypertext documentation of GLAM2 is available online at <u><a href="http://bioinformatics.org.au/glam2/">http://bioinformatics.org.au/glam2/</a></u> or
       on this computer in <u>/usr/share/doc/glam2/</u>.

</pre><h4><b>REFERENCES</b></h4><pre>
       Purge was written by Andy Neuwald and is described in more detail in Neuwald et al., "Gibbs motif
       sampling: detection of bacterial outer membrane protein repeats", Protein Science, 4:1618–1632, 1995.
       Please cite it if you use Purge.

       If you use GLAM2, please cite: MC Frith, NFW Saunders, B Kobe, TL Bailey (2008) Discovering sequence
       motifs with arbitrary insertions and deletions, PLoS Computational Biology (in press).

</pre><h4><b>AUTHORS</b></h4><pre>
       <b>Andrew</b> <b>Neuwald</b>
           Author of purge, renamed glam2-purge in Debian.

       <b>Martin</b> <b>Frith</b>
           Modified purge to be ANSI standard C and improved the user interface.

       <b>Timothy</b> <b>Bailey</b>
           Modified purge to be ANSI standard C and improved the user interface.

       <b>Charles</b> <b>Plessy</b> &lt;<a href="mailto:plessy@debian.org">plessy@debian.org</a>&gt;
           Formatted this manpage in DocBook XML for the Debian distribution.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       The source code and the documentation of Purge and GLAM2 are released in the public domain.

GLAM2 1056                                         05/19/2008                                     <u><a href="../man1/GLAM2-PURGE.1.html">GLAM2-PURGE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>