<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>gt-encseq-encode - Encode sequence files (FASTA/FASTQ, GenBank, EMBL) efficiently.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/genometools">genometools_1.6.5+ds-2.2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       gt-encseq-encode - Encode sequence files (FASTA/FASTQ, GenBank, EMBL) efficiently.

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>gt</b> <b>encseq</b> <b>encode</b> sequence_file [sequence_file [sequence_file ...]]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>-showstats</b> [<u>yes|no</u>]
           show compression results (default: no)

       <b>-ssp</b> [<u>yes|no</u>]
           output sequence separator positions to file (default: yes)

       <b>-des</b> [<u>yes|no</u>]
           output sequence descriptions to file (default: yes)

       <b>-sds</b> [<u>yes|no</u>]
           output sequence description separator positions to file (default: yes)

       <b>-md5</b> [<u>yes|no</u>]
           output MD5 sums to file (default: yes)

       <b>-clipdesc</b> [<u>yes|no</u>]
           clip descriptions after first whitespace (default: no)

       <b>-sat</b> [<u>string</u>]
           specify kind of sequence representation by one of the keywords direct, bytecompress, eqlen, bit,
           uchar, ushort, uint32 (default: undefined)

       <b>-dna</b> [<u>yes|no</u>]
           input is DNA sequence (default: no)

       <b>-protein</b> [<u>yes|no</u>]
           input is protein sequence (default: no)

       <b>-plain</b> [<u>yes|no</u>]
           process as plain text (default: no)

       <b>-dust</b> [<u>yes|no</u>]
           mask low-complexity regions using the dust algorithm (default: no)

       <b>-dustwindow</b> [<u>value</u>]
           windowsize for the dust algorithm (default: 64)

       <b>-dustthreshold</b> [<u>value</u>]
           threshold for the dust algorithm (default: 2.000000)

       <b>-dustlink</b> [<u>value</u>]
           Max. distance between regions masked by dust before merging. (default: 1)

       <b>-indexname</b> [<u>string</u>]
           specify name for index to be generated (default: undefined)

       <b>-smap</b> [<u>string</u>]
           specify file containing a symbol mapping (default: undefined)

       <b>-lossless</b> [<u>yes|no</u>]
           allow lossless original sequence retrieval (default: no)

       <b>-v</b> [<u>yes|no</u>]
           be verbose (default: no)

       <b>-help</b>
           display help for basic options and exit

       <b>-help+</b>
           display help for all options and exit

       <b>-version</b>
           display version information and exit

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       Report bugs to https://github.com/genometools/genometools/issues.

GenomeTools 1.6.5                                  04/27/2024                                <u><a href="../man1/GT-ENCSEQ-ENCODE.1.html">GT-ENCSEQ-ENCODE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>