<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>gt-hop - Cognate sequence-based homopolymer error correction.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/genometools">genometools_1.6.5+ds-2.2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       gt-hop - Cognate sequence-based homopolymer error correction.

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>gt</b> <b>hop</b> -&lt;mode&gt; -c &lt;encseq&gt; -map &lt;sam/bam&gt; -reads &lt;fastq&gt; [options...]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>-c</b> [<u>string</u>]
           cognate sequence (encoded using gt encseq encode)

       <b>-map</b> [<u>string</u>]
           mapping of reads to the cognate sequence it must be in SAM/BAM format, and sorted by coordinate (can
           be prepared e.g. using: samtools sort)

       <b>-sam</b> [<u>yes|no</u>]
           mapping file is SAM default: BAM

       <b>-aggressive</b> [<u>yes|no</u>]
           correct as much as possible

       <b>-moderate</b> [<u>yes|no</u>]
           mediate between sensitivity and precision

       <b>-conservative</b> [<u>yes|no</u>]
           correct only most likely errors

       <b>-expert</b> [<u>yes|no</u>]
           manually select correction criteria

       <b>-reads</b>
           uncorrected read file(s) in FastQ format; the corrected reads are output in the currect working
           directory in files which are named as the input files, each prepended by a prefix (see -outprefix
           option) -reads allows one to output the reads in the same order as in the input and is mandatory if
           the SAM contains more than a single primary alignment for each read (e.g. output of bwasw) see also
           -o option as an alternative

       <b>-outprefix</b> [<u>string</u>]
           prefix for output filenames (corrected reads)when -reads is specified the prefix is prepended to each
           input filename (default: hop_)

       <b>-o</b> [<u>string</u>]
           output file for corrected reads (see also -reads/-outprefix) if -o is used, reads are output in a
           single file in the order they are found in the SAM file (which usually differ from the original
           order) this will only work if the reads were aligned with a software which only includes 1 alignment
           for each read (e.g. bwa) (default: undefined)

       <b>-hmin</b> [<u>value</u>]
           minimal homopolymer length in cognate sequence (default: 3)

       <b>-read-hmin</b> [<u>value</u>]
           minimal homopolymer length in reads (default: 2)

       <b>-qmax</b> [<u>value</u>]
           maximal average quality of homopolymer in a read (default: 120)

       <b>-altmax</b> [<u>value</u>]
           max support of alternate homopol. length; e.g. 0.8 means: do not correct any read if homop. length in
           more than 80%% of the reads has the same value, different from the cognate if altmax is set to 1.0
           reads are always corrected (default: 0.800000)

       <b>-cogmin</b> [<u>value</u>]
           min support of cognate sequence homopol. length; e.g. 0.1 means: do not correct any read if cognate
           homop. length is not present in at least 10%% of the reads if cogmin is set to 0.0 reads are always
           corrected

       <b>-mapqmin</b> [<u>value</u>]
           minimal mapping quality (default: 21)

       <b>-covmin</b> [<u>value</u>]
           minimal coverage; e.g. 5 means: do not correct any read if coverage (number of reads mapped over
           whole homopolymer) is less than 5 if covmin is set to 1 reads are always corrected (default: 1)

       <b>-allow-muliple</b> [<u>yes|no</u>]
           allow multiple corrections in a read (default: no)

       <b>-clenmax</b> [<u>value</u>]
           maximal correction length default: unlimited

       <b>-ann</b> [<u>string</u>]
           annotation of cognate sequence it must be sorted by coordinates on the cognate sequence (this can be
           e.g. done using: gt gff3 -sort) if -ann is used, corrections will be limited to homopolymers
           startingor ending inside the feature type indicated by -ft optionformat: sorted GFF3 (default:
           undefined)

       <b>-ft</b> [<u>string</u>]
           feature type to use when -ann option is specified (default: CDS)

       <b>-v</b> [<u>yes|no</u>]
           be verbose (default: no)

       <b>-help</b>
           display help for basic options and exit

       <b>-help+</b>
           display help for all options and exit

       <b>-version</b>
           display version information and exit

       Correction mode:

       One of the options <u>-aggressive</u>, <u>-moderate</u>, <u>-conservative</u> or <u>-expert</u> must be selected.

       The <u>-aggressive</u>, <u>-moderate</u> and <u>-conservative</u> modes are presets of the criteria by which it is decided if
       an observed discrepancy in homopolymer length between cognate sequence and a read shall be corrected or
       not. A description of the single criteria is provided by using the <u>-help+</u>' option. The presets are
       equivalent to the following settings:

                               -aggressive    -moderate      -conservative
           -hmin               3              3              3
           -read-hmin          1              1              2
           -altmax             1.00           0.99           0.80
           -refmin             0.00           0.00           0.10
           -mapqmin            0              10             21
           -covmin             1              1              1
           -clenmax            unlimited      unlimited      unlimited
           -allow-multiple     yes            yes            no

       The aggressive mode tries to maximize the sensitivity, the conservative mode to minimize the false
       positives. An even more conservative set of corrections can be achieved using the <u>-ann</u> option (see
       <u>-help+</u>).

       The <u>-expert</u> mode allows one to manually set each parameter; the default values are the same as in the
       <u>-conservative</u> mode.

       (Finally, for evaluation purposes only, the <u>-state-of-truth</u> mode can be used: this mode assumes that the
       sequenced genome has been specified as cognate sequence and outputs an ideal list of corrections.)

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       Report bugs to https://github.com/genometools/genometools/issues.

GenomeTools 1.6.5                                  04/27/2024                                          <u><a href="../man1/GT-HOP.1.html">GT-HOP</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>