<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>gt-ltrdigest - Identifies and annotates sequence features in LTR retrotransposon candidates.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/genometools">genometools_1.6.5+ds-2.2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       gt-ltrdigest - Identifies and annotates sequence features in LTR retrotransposon candidates.

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>gt</b> <b>ltrdigest</b> [option ...] gff3_file

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>-outfileprefix</b> [<u>string</u>]
           prefix for output files (e.g.  <u>foo</u> will create files called <u>foo_*.csv</u> and <u>foo_*.fas</u>) Omit this option
           for GFF3 output only.

       <b>-metadata</b> [<u>yes|no</u>]
           output metadata (run conditions) to separate file (default: yes)

       <b>-seqnamelen</b> [<u>value</u>]
           set maximal length of sequence names in FASTA headers (e.g. for clustalw or similar tools) (default:
           20)

       <b>-pptlen</b> [<u>start</u> <u>end</u>]
           required PPT length range (default: [8..30])

       <b>-uboxlen</b> [<u>start</u> <u>end</u>]
           required U-box length range (default: [3..30])

       <b>-uboxdist</b> [<u>value</u>]
           allowed U-box distance range from PPT (default: 0)

       <b>-pptradius</b> [<u>value</u>]
           radius around beginning of 3' LTR to search for PPT (default: 30)

       <b>-pptrprob</b> [<u>value</u>]
           purine emission probability inside PPT (default: 0.970000)

       <b>-pptyprob</b> [<u>value</u>]
           pyrimidine emission probability inside PPT (default: 0.030000)

       <b>-pptgprob</b> [<u>value</u>]
           background G emission probability outside PPT (default: 0.250000)

       <b>-pptcprob</b> [<u>value</u>]
           background C emission probability outside PPT (default: 0.250000)

       <b>-pptaprob</b> [<u>value</u>]
           background A emission probability outside PPT (default: 0.250000)

       <b>-ppttprob</b> [<u>value</u>]
           background T emission probability outside PPT (default: 0.250000)

       <b>-pptuprob</b> [<u>value</u>]
           U/T emission probability inside U-box (default: 0.910000)

       <b>-trnas</b> [<u>filename</u>]
           tRNA library in multiple FASTA format for PBS detection Omit this option to disable PBS search.

       <b>-pbsalilen</b> [<u>start</u> <u>end</u>]
           required PBS/tRNA alignment length range (default: [11..30])

       <b>-pbsoffset</b> [<u>start</u> <u>end</u>]
           allowed PBS offset from LTR boundary range (default: [0..5])

       <b>-pbstrnaoffset</b> [<u>start</u> <u>end</u>]
           allowed PBS/tRNA 3' end alignment offset range (default: [0..5])

       <b>-pbsmaxedist</b> [<u>value</u>]
           maximal allowed PBS/tRNA alignment unit edit distance (default: 1)

       <b>-pbsradius</b> [<u>value</u>]
           radius around end of 5' LTR to search for PBS (default: 30)

       <b>-hmms</b>
           profile HMM models for domain detection (separate by spaces, finish with --) in HMMER3 format Omit
           this option to disable pHMM search.

       <b>-pdomevalcutoff</b> [<u>value</u>]
           global E-value cutoff for pHMM search default 1E-6

       <b>-pdomcutoff</b> [<u>...</u>]
           model-specific score cutoff choose from TC (trusted cutoff) | GA (gathering cutoff) | NONE (no
           cutoffs) (default: NONE)

       <b>-aliout</b> [<u>yes|no</u>]
           output pHMM to amino acid sequence alignments (default: no)

       <b>-aaout</b> [<u>yes|no</u>]
           output amino acid sequences for protein domain hits (default: no)

       <b>-allchains</b> [<u>yes|no</u>]
           output features from all chains and unchained features, labeled with chain numbers (default: no)

       <b>-maxgaplen</b> [<u>value</u>]
           maximal allowed gap size between fragments (in amino acids) when chaining pHMM hits for a protein
           domain (default: 50)

       <b>-force_recreate</b> [<u>yes|no</u>]
           force recreation of hmmpressed profiles (default: no)

       <b>-pbsmatchscore</b> [<u>value</u>]
           match score for PBS/tRNA alignments (default: 5)

       <b>-pbsmismatchscore</b> [<u>value</u>]
           mismatch score for PBS/tRNA alignments (default: -10)

       <b>-pbsinsertionscore</b> [<u>value</u>]
           insertion score for PBS/tRNA alignments (default: -20)

       <b>-pbsdeletionscore</b> [<u>value</u>]
           deletion score for PBS/tRNA alignments (default: -20)

       <b>-v</b> [<u>yes|no</u>]
           be verbose (default: no)

       <b>-o</b> [<u>filename</u>]
           redirect output to specified file (default: undefined)

       <b>-gzip</b> [<u>yes|no</u>]
           write gzip compressed output file (default: no)

       <b>-bzip2</b> [<u>yes|no</u>]
           write bzip2 compressed output file (default: no)

       <b>-force</b> [<u>yes|no</u>]
           force writing to output file (default: no)

       <b>-seqfile</b> [<u>filename</u>]
           set the sequence file from which to take the sequences (default: undefined)

       <b>-encseq</b> [<u>filename</u>]
           set the encoded sequence indexname from which to take the sequences (default: undefined)

       <b>-seqfiles</b>
           set the sequence files from which to extract the features use <u>--</u> to terminate the list of sequence
           files

       <b>-matchdesc</b> [<u>yes|no</u>]
           search the sequence descriptions from the input files for the desired sequence IDs (in GFF3),
           reporting the first match (default: no)

       <b>-matchdescstart</b> [<u>yes|no</u>]
           exactly match the sequence descriptions from the input files for the desired sequence IDs (in GFF3)
           from the beginning to the first whitespace (default: no)

       <b>-usedesc</b> [<u>yes|no</u>]
           use sequence descriptions to map the sequence IDs (in GFF3) to actual sequence entries. If a
           description contains a sequence range (e.g., III:1000001..2000000), the first part is used as
           sequence ID (<u>III</u>) and the first range position as offset (<u>1000001</u>) (default: no)

       <b>-regionmapping</b> [<u>string</u>]
           set file containing sequence-region to sequence file mapping (default: undefined)

       <b>-help</b>
           display help for basic options and exit

       <b>-help+</b>
           display help for all options and exit

       <b>-version</b>
           display version information and exit

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       Report bugs to https://github.com/genometools/genometools/issues.

GenomeTools 1.6.5                                  04/27/2024                                    <u><a href="../man1/GT-LTRDIGEST.1.html">GT-LTRDIGEST</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>