<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>hmm2align - align sequences to an HMM profile</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/hmmer2">hmmer2_2.3.2+dfsg-12_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       hmm2align - align sequences to an HMM profile

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>hmm2align</b> <u>[options]</u> <u>hmm2file</u> <u>seqfile</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>hmm2align</b> reads an HMM file from <u>hmmfile</u> and a set of sequences from <u>seqfile,</u> aligns the sequences to the
       profile HMM, and outputs a multiple sequence alignment.

       <u>seqfile</u>  may  be  in  any  unaligned  or  aligned  file  format accepted by HMMER. If it is in a multiple
       alignment format (e.g. Stockholm, MSF, SELEX, ClustalW), the existing alignment  is  ignored  (i.e.,  the
       sequences are read as if they were unaligned - hmm2align will align them the way it wants).

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>     Print brief help; includes version number and summary of all options, including expert options.

       <b>-m</b>     Include in the alignment only those symbols aligned to match states.  Do not show symbols assigned
              to insert states.

       <b>-o</b> <u>&lt;f&gt;</u> Save alignment to file <u>&lt;f&gt;</u> instead of to standard output.

       <b>-q</b>     quiet;  suppress  all  output  except  the alignment itself.  Useful for piping or redirecting the
              output.

</pre><h4><b>EXPERT</b> <b>OPTIONS</b></h4><pre>
       <b>--informat</b> <u>&lt;s&gt;</u>
              Assert that the input <u>seqfile</u> is in format <u>&lt;s&gt;;</u> do not  run  Babelfish  format  autodection.  This
              increases  the  reliability  of  the  program  somewhat,  because the Babelfish can make mistakes;
              particularly recommended for unattended, high-throughput  runs  of  HMMER.  Valid  format  strings
              include FASTA, GENBANK, EMBL, GCG, PIR, STOCKHOLM, SELEX, MSF, CLUSTAL, and PHYLIP. See the User's
              Guide for a complete list.

       <b>--mapali</b> <u>&lt;f&gt;</u>
              Reads  an  alignment from file <u>&lt;f&gt;</u> and aligns it as a single object to the HMM; e.g. the alignment
              in <u>&lt;f&gt;</u> is held fixed.  This allows you to align sequences to a model with <b>hmm2align</b> and view  them
              in  the  context  of  an  existing  trusted multiple alignment.  The alignment to the alignment is
              defined by a "map" kept in the HMM, and so is fast and guaranteed to be consistent  with  the  way
              the  HMM  was  constructed  from the alignment.  The alignment in the file <u>&lt;f&gt;</u> must be exactly the
              alignment that the HMM was built from.  Compare the <b>--withali</b> option.

       <b>--oneline</b>
              Output the alignment with one line per sequence, rather than interleaving the  sequence  alignment
              blocks. Only affects Stockholm format output.

       <b>--outformat</b> <u>&lt;s&gt;</u>
              Output  the  alignment  in  format  <u>&lt;s&gt;.</u>   The  default is Stockholm format. Valid formats include
              Stockholm, SELEX, MSF, Clustal, Phylip, and A2M.

       <b>--withali</b> <u>&lt;f&gt;</u>
              Reads an alignment from file <u>&lt;f&gt;</u> and aligns it as a single object to the HMM; e.g.  the  alignment
              in  <u>&lt;f&gt;</u> is held fixed.  This allows you to align sequences to a model with <b>hmm2align</b> and view them
              in the context of an existing trusted multiple alignment. The alignment to the alignment  is  done
              with a heuristic (nonoptimal) dynamic programming procedure, which may be somewhat slow and is not
              guaranteed  to  be completely consistent with the way the HMM was constructed (though it should be
              quite close).  However, any alignment can be used, not just the alignment that the HMM  was  built
              from. Compare the <b>--mapali</b> option.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Master man page, with full list of and guide to the individual man pages: see <b><a href="../man1/hmmer2.1.html">hmmer2</a>(1).</b>

       For           complete           documentation,           see           the           user          guide
       (<a href="ftp://selab.janelia.org/pub/software/hmmer/2.3.2/Userguide.pdf">ftp://selab.janelia.org/pub/software/hmmer/2.3.2/Userguide.pdf</a>);   or   see   the   HMMER   web    page,
       <a href="http://hmmer.janelia.org/">http://hmmer.janelia.org/</a>.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright (C) 1992-2003 HHMI/Washington University School of Medicine.
       Freely distributed under the GNU General Public License (GPL).
       See the file COPYING in your distribution for details on redistribution conditions.

</pre><h4><b>AUTHOR</b></h4><pre>
       Sean Eddy
       HHMI/Dept. of Genetics
       Washington Univ. School of Medicine
       4566 Scott Ave.
       St Louis, MO 63110 USA
       <a href="http://www.genetics.wustl.edu/eddy/">http://www.genetics.wustl.edu/eddy/</a>

HMMER 2.3.2                                         Oct 2003                                        <u><a href="../man1/hmm2align.1.html">hmm2align</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>