<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>hmm2pfam - search one or more sequences against an HMM database</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/hmmer2">hmmer2_2.3.2+dfsg-12_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       hmm2pfam - search one or more sequences against an HMM database

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>hmm2pfam</b> <u>[options]</u> <u>hmmfile</u> <u>seqfile</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>hmm2pfam</b>  reads  a sequence file <u>seqfile</u> and compares each sequence in it, one at a time, against all the
       HMMs in <u>hmmfile</u> looking for significantly similar sequence matches.

       <u>hmmfile</u> will be looked for first in the current working directory, then  in  a  directory  named  by  the
       environment  variable  <u>HMMERDB.</u>  This lets administrators install HMM library(s) such as Pfam in a common
       location.

       There is a separate output report for each sequence in <u>seqfile.</u>  This report consists of three  sections:
       a  ranked  list of the best scoring HMMs, a list of the best scoring domains in order of their occurrence
       in the sequence, and alignments for all the best scoring domains.  A sequence score may be higher than  a
       domain  score  for the same sequence if there is more than one domain in the sequence; the sequence score
       takes into account all the domains.  All sequences scoring above the <u>-E</u> and <u>-T</u> cutoffs are shown  in  the
       first list, then <u>every</u> domain found in this list is shown in the second list of domain hits.  If desired,
       E-value  and  bit  score  thresholds  may  also be applied to the domain list using the <u>--domE</u> and <u>--domT</u>
       options.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>     Print brief help; includes version number and summary of all options, including expert options.

       <b>-n</b>     Specify that models and sequence are nucleic acid, not protein.  Other HMMER  programs  autodetect
              this;  but  because  of the order in which <b>hmm2pfam</b> accesses data, it can't reliably determine the
              correct "alphabet" by itself.

       <b>-A</b> <u>&lt;n&gt;</u> Limits the alignment output to the <u>&lt;n&gt;</u> best scoring domains.  <b>-A0</b> shuts off the  alignment  output
              and can be used to reduce the size of output files.

       <b>-E</b> <u>&lt;x&gt;</u> Set  the  E-value cutoff for the per-sequence ranked hit list to <u>&lt;x&gt;,</u> where <u>&lt;x&gt;</u> is a positive real
              number. The default is 10.0. Hits with E-values better than (less than)  this  threshold  will  be
              shown.

       <b>-T</b> <u>&lt;x&gt;</u> Set  the bit score cutoff for the per-sequence ranked hit list to <u>&lt;x&gt;,</u> where <u>&lt;x&gt;</u> is a real number.
              The default is negative infinity; by default, the threshold is controlled by E-value  and  not  by
              bit score.  Hits with bit scores better than (greater than) this threshold will be shown.

       <b>-Z</b> <u>&lt;n&gt;</u> Calculate  the  E-value scores as if we had seen a sequence database of <u>&lt;n&gt;</u> sequences. The default
              is arbitrarily set to 59021, the size of Swissprot 34.

</pre><h4><b>EXPERT</b> <b>OPTIONS</b></h4><pre>
       <b>--acc</b>  Report HMM accessions instead  of  names  in  the  output  reports.   Useful  for  high-throughput
              annotation, where the data are being parsed for storage in a relational database.

       <b>--compat</b>
              Use  the  output  format  of  HMMER 2.1.1, the 1998-2001 public release; provided so 2.1.1 parsers
              don't have to be rewritten.

       <b>--cpu</b> <u>&lt;n&gt;</u>
              Sets the maximum number of CPUs that the program will run on. The default is to use  all  CPUs  in
              the  machine.  Overrides  the  HMMER_NCPU  environment variable. Only affects threaded versions of
              HMMER (the default on most systems).

       <b>--cut_ga</b>
              Use Pfam GA (gathering threshold) score cutoffs.  Equivalent to --globT &lt;GA1&gt;  --domT  &lt;GA2&gt;,  but
              the  GA1  and  GA2  cutoffs  are  read from each HMM in <u>hmmfile</u> individually. hmm2build puts these
              cutoffs there if the alignment file was annotated in a Pfam-friendly  alignment  format  (extended
              SELEX  or  Stockholm format) and the optional GA annotation line was present. If these cutoffs are
              not set in the HMM file, <b>--cut_ga</b> doesn't work.

       <b>--cut_tc</b>
              Use Pfam TC (trusted cutoff) score cutoffs. Equivalent to --globT &lt;TC1&gt; --domT &lt;TC2&gt;, but the  TC1
              and TC2 cutoffs are read from each HMM in <u>hmmfile</u> individually. hmm2build puts these cutoffs there
              if  the  alignment  file  was  annotated  in  a  Pfam-friendly alignment format (extended SELEX or
              Stockholm format) and the optional TC annotation line was present. If these cutoffs are not set in
              the HMM file, <b>--cut_tc</b> doesn't work.

       <b>--cut_nc</b>
              Use Pfam NC (noise cutoff) score cutoffs. Equivalent to --globT &lt;NC1&gt; --domT &lt;NC2&gt;,  but  the  NC1
              and NC2 cutoffs are read from each HMM in <u>hmmfile</u> individually. hmm2build puts these cutoffs there
              if  the  alignment  file  was  annotated  in  a  Pfam-friendly alignment format (extended SELEX or
              Stockholm format) and the optional NC annotation line was present. If these cutoffs are not set in
              the HMM file, <b>--cut_nc</b> doesn't work.

       <b>--domE</b> <u>&lt;x&gt;</u>
              Set the E-value cutoff for the per-domain ranked hit list to <u>&lt;x&gt;,</u> where <u>&lt;x&gt;</u>  is  a  positive  real
              number.   The  default is infinity; by default, all domains in the sequences that passed the first
              threshold will be reported in the second list, so that the number of domains reported in the  per-
              sequence list is consistent with the number that appear in the per-domain list.

       <b>--domT</b> <u>&lt;x&gt;</u>
              Set  the  bit  score cutoff for the per-domain ranked hit list to <u>&lt;x&gt;,</u> where <u>&lt;x&gt;</u> is a real number.
              The default is negative infinity; by default, all domains in the sequences that passed  the  first
              threshold  will be reported in the second list, so that the number of domains reported in the per-
              sequence list is consistent with the number that appear in the per-domain list.   <u>Important</u>  <u>note:</u>
              only  one  domain  in  a  sequence  is absolutely controlled by this parameter, or by <b>--domT.</b>  The
              second and subsequent domains in a sequence have a de facto bit score threshold of  0  because  of
              the  details  of  how  HMMER  works.  HMMER  requires at least one pass through the main model per
              sequence; to do more than one pass (more than one domain) the multidomain alignment  must  have  a
              better  score  than  the  single  domain  alignment,  and  hence the extra domains must contribute
              positive score. See the Users' Guide for more detail.

       <b>--forward</b>
              Use the Forward algorithm instead of the Viterbi algorithm to determine the  per-sequence  scores.
              Per-domain  scores are still determined by the Viterbi algorithm. Some have argued that Forward is
              a more sensitive algorithm for detecting remote sequence homologues;  my  experiments  with  HMMER
              have not confirmed this, however.

       <b>--informat</b> <u>&lt;s&gt;</u>
              Assert  that  the  input  <u>seqfile</u>  is in format <u>&lt;s&gt;;</u> do not run Babelfish format autodection. This
              increases the reliability of the program  somewhat,  because  the  Babelfish  can  make  mistakes;
              particularly  recommended  for  unattended,  high-throughput  runs  of HMMER. Valid format strings
              include FASTA, GENBANK, EMBL, GCG, PIR, STOCKHOLM, SELEX, MSF, CLUSTAL, and PHYLIP. See the User's
              Guide for a complete list.

       <b>--null2</b>
              Turn off  the  post  hoc  second  null  model.  By  default,  each  alignment  is  rescored  by  a
              postprocessing  step  that takes into account possible biased composition in either the HMM or the
              target sequence.  This is almost essential in database searches, especially with  local  alignment
              models.  There  is  a very small chance that this postprocessing might remove real matches, and in
              these cases <b>--null2</b> may improve sensitivity at the expense  of  reducing  specificity  by  letting
              biased composition hits through.

       <b>--pvm</b>  Run  on  a  Parallel  Virtual  Machine  (PVM). The PVM must already be running. The client program
              <b>hmm2pfam-pvm</b> must be installed on all the PVM nodes.  The HMM database <u>hmmfile</u> and  an  associated
              GSI  index  file  <u>hmmfile.</u>gsi  must  also  be  installed  on all the PVM nodes.  (The GSI index is
              produced by the program <b>hmm2index</b>.)  Because the PVM implementation is I/O  bound,  it  is  highly
              recommended  that  each  node have a local copy of <u>hmmfile</u> rather than NFS mounting a shared copy.
              Optional PVM support must have been compiled into HMMER for <b>--pvm</b> to function.

       <b>--xnu</b>  Turn on XNU filtering of target protein sequences. Has no effect on  nucleic  acid  sequences.  In
              trial experiments, <b>--xnu</b> appears to perform less well than the default post hoc null2 model.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Master man page, with full list of and guide to the individual man pages: see <b><a href="../man1/hmmer2.1.html">hmmer2</a>(1).</b>

       For           complete           documentation,           see           the           user          guide
       (<a href="ftp://selab.janelia.org/pub/software/hmmer/2.3.2/Userguide.pdf">ftp://selab.janelia.org/pub/software/hmmer/2.3.2/Userguide.pdf</a>);   or   see   the   HMMER   web    page,
       <a href="http://hmmer.janelia.org/">http://hmmer.janelia.org/</a>.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright (C) 1992-2003 HHMI/Washington University School of Medicine.
       Freely distributed under the GNU General Public License (GPL).
       See the file COPYING in your distribution for details on redistribution conditions.

</pre><h4><b>AUTHOR</b></h4><pre>
       Sean Eddy
       HHMI/Dept. of Genetics
       Washington Univ. School of Medicine
       4566 Scott Ave.
       St Louis, MO 63110 USA
       <a href="http://www.genetics.wustl.edu/eddy/">http://www.genetics.wustl.edu/eddy/</a>

HMMER 2.3.2                                         Oct 2003                                         <u><a href="../man1/hmm2pfam.1.html">hmm2pfam</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>