<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Infernal - sequence analysis using profiles of RNA sequence and secondary structure consensus</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/infernal">infernal_1.1.5-3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Infernal - sequence analysis using profiles of RNA sequence and secondary structure consensus

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>cmalign</b>
         Align sequences to a covariance model

       <b>cmbuild</b>
         Construct covariance model(s) from structurally annotated RNA multiple sequence alignment(s)

       <b>cmcalibrate</b>
         Fit exponential tails for covariance model E-value determination

       <b>cmconvert</b>
         Convert Infernal covariance model files

       <b>cmemit</b>
         Sample sequences from a covariance model

       <b>cmfetch</b>
         Retrieve covariance model(s) from a file

       <b>cmpress</b>
         Prepare a covariance model database for cmscan

       <b>cmscan</b>
         Search sequence(s) against a covariance model database

       <b>cmsearch</b>
         Search covariance model(s) against a sequence database

       <b>cmstat</b>
         summary statistics for a covariance model file

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Infernal  is  a  suite  of  several  programs for structural RNA sequence alignment and database homology
       search.  It  uses  probabilistic  models  called  "covariance  models"  (CMs)  to  represent  the  likely
       evolutionary homologs of a multiple alignment (or single sequence) of a structural RNA sequence family.

       Along  with the Rfam database of RNA families and associated CMs (<a href="http://rfam.sanger.ac.uk">http://rfam.sanger.ac.uk</a>), Infernal can
       be used to annotate homologs of known structural RNA families in genomes.

       Infernal is closely related to the HMMER software suite for sequence family analysis using  profile  HMMs
       (<a href="http://hmmer.org">http://hmmer.org</a>),  but  is  designed specifically for structural RNA sequence families.  In addition to
       modeling the conserved sequence of a family as profile HMMs do, CMs model the family's  conserved,  well-
       nested (non-pseudoknotted) secondary structure as well. Consequently, CM search and alignment methods are
       relatively computationally expensive.  Infernal uses profile HMMs as filters and for deriving constraints
       to make the CM methods more practical.

       Infernal is used in three main modes: to search a sequence database for new homologs of an RNA family (or
       annotate  homologs  in  a  genome); to search a CM database (like Rfam) to find what known family a query
       sequence belongs to; and to automatically construct large multiple alignments (i.e. with  an  effectively
       unlimited number of sequences) using a CM representative of a sequence family.

       Suppose  you  have  a  structurally  annotated  multiple  sequence  alignment of a RNA sequence family of
       interest, and you want to search a sequence database for additional homologs. The <b>cmbuild</b> program  builds
       covariance  model(s)  from  multiple  alignment(s)  and  <b>cmcalibrate</b>  determines important parameters for
       estimating the statistical significance of database hits to the model in subsequent searches.

       The <b>cmsearch</b> program searches CM(s) against a sequence database.

       Suppose you have sequence(s) that you want to analyze  using  a  Infernal-based  CM  database  like  Rfam
       (<a href="http://rfam.sanger.ac.uk">http://rfam.sanger.ac.uk</a>).   The  <b>cmpress</b> program formats a covariance model (such as the file you would
       download from Rfam) into a Infernal binary database.  The <b>cmscan</b>  program  searches  sequence(s)  against
       that database.

       Suppose you want to align lots of sequences. You can construct a manageably small structural alignment of
       a  representative  set  of  sequences,  build a CM with <b>cmbuild,</b> and use the <b>cmalign</b> program to align any
       number of sequences to that CM.

       Infernal also includes some auxiliary tools for working with large CM databases.  <b>cmfetch</b> fetches one  or
       more CMs from a database.  <b>cmstat</b> prints summary statistics about a CM file.

       For  compatibility  with  previous  versions  of  Infernal,  as well as with HMMER, the <b>cmconvert</b> program
       converts CM files to a few other formats.

       The <b>cmemit</b> program generates (simulates) "homologous" sequences by  sampling  from  a  CM.  It  can  also
       generate a "consensus" sequence.

       Each program has its own man page.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       This  is  a  summary man page for the entire Infernal package.  See individual man pages <b>[<a href="../man1/cmbuild.1.html">cmbuild</a>(1),</b> for
       example] for usage, options, and description of each program in the package.

       For complete documentation, see the user guide that came with your Infernal distribution (Userguide.pdf);
       or see the Infernal web page (<a href="http://eddylab.org/infernal/">http://eddylab.org/infernal/</a>).

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright (C) 2023 Howard Hughes Medical Institute.
       Freely distributed under the BSD open source license.

       For additional information on copyright and licensing, see the file called  COPYRIGHT  in  your  Infernal
       source distribution, or see the Infernal web page (<a href="http://eddylab.org/infernal/">http://eddylab.org/infernal/</a>).

</pre><h4><b>AUTHOR</b></h4><pre>
       <a href="http://eddylab.org">http://eddylab.org</a>

Infernal 1.1.5                                      Sep 2023                                         <u><a href="../man1/Infernal.1.html">Infernal</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>