<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>lastal5 - genome-scale comparison of biological sequences</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/last-align">last-align_1260-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       lastal5 - genome-scale comparison of biological sequences

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>lastal5-plain</b> [<u>options</u>] <u>lastdb-name</u> <u>fasta-sequence-file(s)</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Find and align similar sequences.

       Cosmetic  options:  <b>-h</b>, <b>--help</b>: show all options and their default settings, and exit <b>-V</b>, <b>--version</b>: show
       version information, and exit <b>-v</b>: be verbose: write messages  about  what  lastal  is  doing  <b>-f</b>:  output
       format: TAB, MAF, BlastTab, BlastTab+ (default=MAF)

       E-value  options  (default settings): <b>-D</b>: query letters per random alignment (1e+06) <b>-E</b>: maximum expected
       alignments per square giga (1e+18/D/refSize/numOfStrands)

       Score options (default settings): <b>-r</b>: match score   (2 if <b>-M</b>, else  6 if 1&lt;=Q&lt;=4,  else  1  if  DNA)  <b>-q</b>:
       mismatch  cost  (3  if  <b>-M</b>,  else  18  if  1&lt;=Q&lt;=4,  else  1  if  DNA)  <b>-p</b>:  match/mismatch  score matrix
       (protein-protein: BL62, DNA-protein: BL80) <b>-X</b>: N/X is  ambiguous  in:  0=neither  sequence,  1=reference,
       2=query,  3=both  (0)  <b>-a</b>:  gap  existence cost (DNA: 7, protein: 11, 1&lt;=Q&lt;=4: 21) <b>-b</b>: gap extension cost
       (DNA: 1, protein:  2, 1&lt;=Q&lt;=4:  9) <b>-A</b>: insertion existence cost (a) <b>-B</b>: insertion extension cost (b)  <b>-c</b>:
       unaligned  residue  pair  cost  (off) <b>-F</b>: frameshift cost(s) (off) <b>-x</b>: maximum score drop for preliminary
       gapped alignments (z) <b>-y</b>: maximum score drop for gapless alignments (min[t*10, x]) <b>-z</b>: maximum score drop
       for final gapped alignments (e-1) <b>-d</b>: minimum score for gapless alignments (min[e, 2500/n  query  letters
       per hit]) <b>-e</b>: minimum score for gapped alignments

       Initial-match options (default settings): <b>-m</b>: maximum initial matches per query position (10) <b>-l</b>: minimum
       length  for initial matches (1) <b>-L</b>: maximum length for initial matches (infinity) <b>-k</b>: use initial matches
       starting at every k-th position in each query (1) <b>-W</b>: use "minimum" positions in  sliding  windows  of  W
       consecutive positions

       Miscellaneous  options  (default  settings):  <b>-s</b>:  strand: 0=reverse, 1=forward, 2=both (2 for DNA, 1 for
       protein) <b>-S</b>: score matrix applies to forward strand of: 0=reference,  1=query  (0)  <b>-K</b>:  omit  alignments
       whose  query range lies in &gt;= K others with &gt; score (off) <b>-C</b>: omit gapless alignments in &gt;= C others with
       &gt; score-per-length (off) <b>-P</b>: number of parallel threads (1) <b>-i</b>: query batch size (8 KiB, unless there  is
       &gt;  1  thread  or  lastdb  volume)  <b>-M</b>: find minimum-difference alignments (faster but cruder) <b>-T</b>: type of
       alignment: 0=local, 1=overlap (0) <b>-n</b>: maximum gapless alignments per query  position  (infinity  if  m=0,
       else  m)  <b>-N</b>:  stop after the first N alignments per query strand <b>-R</b>: lowercase &amp; simple-sequence options
       (the same as was used by lastdb) <b>-u</b>: mask lowercase during extensions: 0=never, 1=gapless,

              2=gapless+postmask, 3=always (2 if lastdb <b>-c</b> and Q!=pssm, else 0)

       <b>-w</b>: suppress repeats inside exact matches, offset by &lt;= this distance (1000) <b>-G</b>:  genetic  code  (1)  <b>-t</b>:
       'temperature'  for calculating probabilities (1/lambda) <b>-g</b>: 'gamma' parameter for gamma-centroid and LAMA
       (1) <b>-j</b>: output type: 0=match counts, 1=gapless, 2=redundant gapped, 3=gapped,

              4=column ambiguity estimates, 5=gamma-centroid, 6=LAMA, 7=expected counts (3)

       <b>-J</b>: score type: 0=ordinary, 1=full (1 for new-style frameshifts, else 0) <b>-Q</b>: input format:  fastx,  keep,
       sanger, solexa, illumina, prb, pssm

              (default=fasta)

lastal5 1260                                      February 2022                                       <u><a href="../man1/LASTAL5.1.html">LASTAL5</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>