<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>micro_razers - Map small RNA reads possibly containing 3' adapter sequence</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/seqan-apps">seqan-apps_2.4.0+dfsg-16_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       micro_razers - Map small RNA reads possibly containing 3' adapter sequence

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>micro_razers</b> [<u>OPTIONS</u>] &lt;<u>GENOME</u> <u>FILE</u>&gt; &lt;<u>READS</u> <u>FILE</u>&gt;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       MicroRazerS  uses  a  prefix-based mapping strategy to map small RNA reads possibly containing 3' adapter
       sequence.

       Input to MicroRazerS is a reference genome file and a file with single-end reads. Use - to read the reads
       from stdin.

       (c) Copyright 2009 by Anne-Katrin Emde.

</pre><h4><b>REQUIRED</b> <b>ARGUMENTS</b></h4><pre>
       <b>ARGUMENT</b> <b>0</b> <u>INPUT_FILE</u>
              A reference genome file. Valid filetypes are: <u>.sam[.*]</u>,  <u>.raw[.*]</u>,  <u>.gbk[.*]</u>,  <u>.frn[.*]</u>,  <u>.fq[.*]</u>,
              <u>.fna[.*]</u>, <u>.ffn[.*]</u>, <u>.fastq[.*]</u>, <u>.fasta[.*]</u>, <u>.faa[.*]</u>, <u>.fa[.*]</u>, <u>.embl[.*]</u>, and <u>.bam</u>, where * is any
              of the following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent (de)compression.

       <b>READS</b> List of <u>INPUT_FILE</u>'s
              Either  one  (single-end) or two (paired-end) read files. Valid filetypes are: <u>.sam[.*]</u>, <u>.raw[.*]</u>,
              <u>.gbk[.*]</u>, <u>.frn[.*]</u>,  <u>.fq[.*]</u>,  <u>.fna[.*]</u>,  <u>.ffn[.*]</u>,  <u>.fastq[.*]</u>,  <u>.fasta[.*]</u>,  <u>.faa[.*]</u>,  <u>.fa[.*]</u>,
              <u>.embl[.*]</u>, and <u>.bam</u>, where * is any of the following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent
              (de)compression.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              Display the help message.

       <b>--version</b>
              Display version information.

   <b>Main</b> <b>Options::</b>
       <b>-o</b>, <b>--output</b> <u>OUTPUT_FILE</u>
              Change  output  filename. (use - to dump to stdout in razers format) Default: &lt;<u>READS</u> <u>FILE</u>&gt;.razers.
              Valid filetypes are: <u>.sam</u> and <u>.razers</u>.

       <b>-rr</b>, <b>--recognition-rate</b> <u>DOUBLE</u>
              set the percent recognition rate In range [80..100]. Default: <u>100</u>.

       <b>-sL</b>, <b>--seed-length</b> <u>INTEGER</u>
              seed length In range [10..inf]. Default: <u>16</u>.

       <b>-sE</b>, <b>--seed-error</b>
              allow for one error in the seed

       <b>-f</b>, <b>--forward</b>
              map reads only to forward strands.

       <b>-r</b>, <b>--reverse</b>
              map reads only to reverse strands.

       <b>-mN</b>, <b>--match-N</b>
              'N' matches with all other characters

       <b>-m</b>, <b>--max-hits</b> <u>INTEGER</u>
              output only NUM of the best hits In range [1..inf]. Default: <u>100</u>.

       <b>-pa</b>, <b>--purge-ambiguous</b>
              purge reads with more than max-hits best matches

       <b>-lm</b>, <b>--low-memory</b>
              decrease memory usage at the expense of runtime

       <b>-v</b>, <b>--verbose</b>
              verbose mode

       <b>-vv</b>, <b>--vverbose</b>
              very verbose mode

   <b>Output</b> <b>Options::</b>
       <b>-a</b>, <b>--alignment</b>
              dump the alignment for each match

       <b>-gn</b>, <b>--genome-naming</b> <u>INTEGER</u>
              Select how genomes are named. 0 = use Fasta id, 1 = enumerate beginning with 1. In  range  [0..1].
              Default: <u>0</u>.

       <b>-rn</b>, <b>--read-naming</b> <u>INTEGER</u>
              Select  how  reads  are  named. 0 = use Fasta id, 1 = enumerate beginning with 1. In range [0..1].
              Default: <u>0</u>.

       <b>-so</b>, <b>--sort-order</b> <u>INTEGER</u>
              Select how matches are sorted. 0 = read number, 1 = genome position. In range [0..1]. Default: <u>0</u>.

       <b>-pf</b>, <b>--position-format</b> <u>INTEGER</u>
              Select begin/end position numbering (see Coordinate section below). 0 = gap space,  1  =  position
              space. In range [0..1]. Default: <u>0</u>.

micro_razers 1.0.11 [tarball]                                                                    <u><a href="../man1/MICRO_RAZERS.1.html">MICRO_RAZERS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>