<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>ncbi_search - visually comparing bacterial, plasmid, chloroplast, or mitochondrial sequences</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/cct">cct_1.0.3-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       ncbi_search - visually comparing bacterial, plasmid, chloroplast, or mitochondrial sequences

</pre><h4><b>DESCRIPTION</b></h4><pre>
   <b>USAGE:</b>
              ncbi_search <b>-q</b> STRING <b>-o</b> FILE <b>-d</b> STRING <b>-r</b> STRING [Options]

   <b>DESCRIPTION:</b>
              Uses NCBI's eSearch to download collections of sequences.

   <b>REQUIRED</b> <b>ARGUMENTS:</b>

       <b>-q</b>, <b>--query</b> [STRING]

              Raw query text.

       <b>-o</b>, <b>--output</b> [FILE]

              Output  file  to  create.  If  the  split  option  is given, this should be a directory, where the
              returned records will be written. If the directory does not exist it will be created.

       <b>-d</b>, <b>--database</b> [STRING]

              Name of the NCBI database to search, such as 'nucleotide', 'protein', or 'gene'.

       <b>-r</b>, <b>--return_type</b> [STRING]

              The type of information requested. For sequences 'fasta' is often used.  The accepted formats vary
              depending on the database being queried.

       <b>-s</b>, <b>--split</b>

              Return each record as a separate file where the file name will will be the accesssion  id  of  the
              record. This option only works if the return_type is 'gb' or 'gbwithparts'.

       <b>-m</b>, <b>--max_records</b> [INTEGER]

              The maximum number of records to return (default is to return all matches satisfying the query).

       <b>-v</b>, <b>--verbose</b>

              Provide progress messages.

       <b>-h</b>, <b>--help</b>

              Show this message.

   <b>EXAMPLE:</b>
              ncbi_search <b>-q</b> 'dysphagia AND homo sapiens[ORGN]' \

       <b>-o</b> results.txt <b>-d</b> pubmed <b>-r</b> uilist <b>-m</b> 100

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

ncbi_search 20170919                               August 2018                                    <u><a href="../man1/NCBI_SEARCH.1.html">NCBI_SEARCH</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>