<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>probalign - Align sequences in MFAFILE(s) and print result to standard output</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/probalign">probalign_1.4-10_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       probalign - Align sequences in MFAFILE(s) and print result to standard output

</pre><h4><b>DESCRIPTION</b></h4><pre>
       PROBALIGN  Version  1.4  (Nov  2010) aligns multiple protein sequences and prints to the standard output.
       Written by Satish Chikkagoudar and Usman Roshan using code from PROBCONS version 1.1 (written  by  Chuong
       Do) and based upon probA (written by Ulrike Muckstein).

       PROBALIGN  1.4  comes  with  ABSOLUTELY  NO  WARRANTY.   This  is  free  software, and you are welcome to
       redistribute it under certain conditions.  See the README file for details.

   <b>Usage:</b>
              probalign [OPTION]... [MFAFILE]...

   <b>Description:</b>
              Align sequences in MFAFILE(s) and print result to standard output

       <b>-clustalw</b>

              use CLUSTALW output format instead of MFA

       <b>-v</b>, <b>--verbose</b>

              report progress while aligning (default: off)

       <b>-a</b>, <b>--alignment-order</b>

              print sequences in alignment order rather than input order (default: off)

       <b>-T</b>, <b>-temperature</b>

              Sets the thermodynamic temperature parameter

              (default: 5 (for protein data mode), 1 ( for nucleotide data mode)).

       <b>-score_matrix</b>, <b>--score_matrix</b>

              Sets the  type  of  scoring  matrix  used  to  calculate  the  posterior  probabilities  (default:
              gonnet_160, representing gonnet 160, refer README for details)

       <b>-go</b>, <b>--gap-open</b>

              This option can be used to specify the gap open parameter. The default for Gonnet 160 (protein) is
              22 and nucleotide (simple matrix) is 4.

       <b>-ge</b>, <b>--gap-extension</b>

              This  option  can  be  used  to  specify  the  gap extension parameter. The default for Gonnet 160
              (protein) is 1 and nucleotide (simple matrix) is 0.25.

       <b>-nuc</b>

              Specify this option to indicate that inputted sequences are nucleotide sequences

       <b>-prot</b>

              Specify this option to indicate that inputted sequences are protein sequences [DEFAULT]

       <b>-showPP</b>

              Outputs the posterior probabilities of  alignment  columns  as  a  new  sequence  named  Posterior
              Probabilities (The probability values are scaled to be between integers between 0 and 9).

probalign 1.4                                     January 2023                                      <u><a href="../man1/PROBALIGN.1.html">PROBALIGN</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>