<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>seq-gen - Sequence Generator</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/seq-gen">seq-gen_1.3.4-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       seq-gen - Sequence Generator

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>seq-gen</b>  [<u>-m</u> <u>MODEL</u>] [<u>-l</u> <u>#</u>] [<u>-n</u> <u>#</u>] [<u>-p</u> <u>#</u>] [<u>-s</u> <u>#</u> <u>|</u> <u>-d</u> <u>#</u>] [<u>-k</u> <u>#</u>][-c #1 #2 #3 | -a # [-g #]] [-f e | #] [-t #
       | -r #][-z #] [-o[p][r][n]] [-w[a][r]] [-x NAME] [-q] [-h] [treefile]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Sequence Generator - seq-gen Version 1.3.3

       <b>-l</b>: # = sequence length [default = 1000].

       <b>-n</b>: # = simulated datasets per tree [default = 1].

       <b>-p</b>: # = number of partitions (and trees) per sequence [default = 1].

       <b>-s</b>: # = branch length scaling factor [default = 1.0].

       <b>-d</b>: # = total tree scale [default = use branch lengths].

       <b>-k</b>: # = use sequence k as ancestral (needs alignment) [default = random].

              Substitution model options:

       <b>-m</b>: MODEL = HKY, F84, GTR, JTT, WAG, PAM, BLOSUM, MTREV, GENERAL

              HKY, F84 &amp; GTR are for nucleotides the rest are for amino acids

       <b>-a</b>: # = shape (alpha) for gamma rate heterogeneity [default = none].

       <b>-g</b>: # = number of gamma rate categories [default = continuous].

       <b>-i</b>: # = proportion of invariable sites [default = 0.0].

              Nucleotid model specific options:

       <b>-c</b>: #1 #2 #3 = rates for codon position heterogeneity [default = none].

       <b>-t</b>: # = transition-transversion ratio [default = equal rate].

       <b>-r</b>: #1 #2 #3 #4 #5 #6= general rate matrix [default = all 1.0].

       <b>-f</b>: #A #C #G #T = nucleotide frequencies [default = all equal].

              Amino Acid model specific options:

              specify using the order ARNDCQEGHILKMFPSTWYV

       <b>-r</b>: #1 .. #190 = general rate matrix [default = all 1.0].

       <b>-f</b>: #1 .. #20 = amino acid frequencies e=equal [default = matrix freqs].

              Miscellaneous options:

       <b>-z</b>: # = seed for random number generator [default = system generated].

       <b>-o</b>: Output file format [default = PHYLIP]

              p PHYLIP format r relaxed PHYLIP format n NEXUS format

       <b>-w</b>: Write additional information [default = none]

              a Write ancestral sequences for each node r Write rate for each site

       <b>-x</b>: NAME = a text file to insert after every dataset [default = none].

       <b>-h</b>: Give this help message

       <b>-q</b>: Quiet

              treefile: name of tree file [default = trees on stdin]

</pre><h4><b>AUTHOR</b></h4><pre>
       (c) Copyright, 1996-2004 Andrew Rambaut and Nick Grassly Department  of  Zoology,  University  of  Oxford
       South Parks Road, Oxford OX1 3PS, U.K.

seq-gen 1.3.3                                      March 2011                                         <u><a href="../man1/SEQ-GEN.1.html">SEQ-GEN</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>