<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>shuffle - randomize the sequences in a sequence file</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/biosquid">biosquid_1.9g+cvs20050121-15.1build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       shuffle - randomize the sequences in a sequence file

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>shuffle</b> <u>[options]</u> <u>seqfile</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>shuffle</b>  reads  a sequence file <u>seqfile,</u> randomizes each sequence, and prints the randomized sequences in
       FASTA format on standard output. The sequence names are unchanged; this allows  you  to  track  down  the
       source of each randomized sequence if necessary.

       The  default  is  to  simply  shuffle  each input sequence, preserving monosymbol composition exactly. To
       shuffle each sequence while preserving both its monosymbol and disymbol composition exactly, use  the  <u>-d</u>
       option.

       The  <u>-0</u>  and  <u>-1</u>  options  allow  you to generate sequences with the same Markov properties as each input
       sequence. With <u>-0,</u> for each input sequence, 0th  order  Markov  statistics  are  collected  (e.g.  symbol
       composition), and a new sequence is generated with the same composition.  With <u>-1,</u> the generated sequence
       has the same 1st order Markov properties as the input sequence (e.g.  the same disymbol frequencies).

       Note  that  the  default and <u>-0,</u> or <u>-d</u> and <u>-1,</u> are similar; the shuffling algorithms preserve composition
       exactly, while the Markov algorithms only expect  to  generate  a  sequence  of  similar  composition  on
       average.

       Other shuffling algorithms are also available, as documented below in the options.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-0</b>     Calculate 0th order Markov frequencies of each input sequence (e.g. residue composition); generate
              output sequence using the same 0th order Markov frequencies.

       <b>-1</b>     Calculate  1st  order  Markov  frequencies  for  each input sequence (e.g. diresidue composition);
              generate output sequence using the same 1st order Markov frequencies.  The first  residue  of  the
              output sequence is always the same as the first residue of the input sequence.

       <b>-d</b>     Shuffle the input sequence while preserving both monosymbol and disymbol composition exactly. Uses
              an algorithm published by  S.F. Altschul and B.W. Erickson, Mol. Biol. Evol. 2:526-538, 1985.

       <b>-h</b>     Print brief help; includes version number and summary of all options, including expert options.

       <b>-l</b>     Look only at the length of each input sequence; generate an i.i.d. output protein sequence of that
              length, using monoresidue frequencies typical of proteins (taken from Swissprot 35).

       <b>-n</b> <u>&lt;n&gt;</u> Make  <u>&lt;n&gt;</u>  different  randomizations of each input sequence in <u>seqfile,</u> rather than the default of
              one.

       <b>-r</b>     Generate the output sequence by reversing the input sequence. (Therefore only one  "randomization"
              per input sequence is possible, so it's not worth using <u>-n</u> if you use reversal.)

       <b>-t</b> <u>&lt;n&gt;</u> Truncate  each  input sequence to a fixed length of exactly <u>&lt;n&gt;</u> residues. If the input sequence is
              shorter than <u>&lt;n&gt;</u> it is discarded (therefore the output file may contain fewer sequences  than  the
              input  file).   If  the  input  sequence  is  longer than <u>&lt;n&gt;</u> a contiguous subsequence is randomly
              chosen.

       <b>-w</b> <u>&lt;n&gt;</u> Regionally shuffle  each  input  sequence  in  window  sizes  of  <u>&lt;n&gt;,</u>  preserving  local  residue
              composition  in  each  window.   Probably  a  better  shuffling  algorithm  for  biosequences with
              nonstationary residue composition (e.g. composition that is varying along the  sequence,  such  as
              between different isochores in human genome sequence).

       <b>-B</b>     (Babelfish). Autodetect and read a sequence file format other than the default (FASTA). Almost any
              common  sequence  file  format  is  recognized  (including Genbank, EMBL, SWISS-PROT, PIR, and GCG
              unaligned sequence formats, and Stockholm, GCG  MSF,  and  Clustal  alignment  formats).  See  the
              printed documentation for a complete list of supported formats.

</pre><h4><b>EXPERT</b> <b>OPTIONS</b></h4><pre>
       <b>--informat</b> <u>&lt;s&gt;</u>
              Specify  that  the  sequence  file is in format <u>&lt;s&gt;,</u> rather than the default FASTA format.  Common
              examples include Genbank, EMBL, GCG, PIR, Stockholm, Clustal, MSF,  or  PHYLIP;  see  the  printed
              documentation  for  a  complete  list of accepted format names.  This option overrides the default
              expected format (FASTA) and the <u>-B</u> Babelfish autodetection option.

       <b>--nodesc</b>
              Do not output any sequence description in the output file, only the sequence names.

       <b>--seed</b> <u>&lt;s&gt;</u>
              Set the random number seed to <u>&lt;s&gt;.</u>  If you want reproducible results, use the same seed each time.
              By default, <b>shuffle</b> uses a different seed each time, so does  not  generate  the  same  output  in
              subsequent runs with the same input.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/afetch.1.html">afetch</a></b>(1),  <b><a href="../man1/alistat.1.html">alistat</a></b>(1),  <b><a href="../man1/compalign.1.html">compalign</a></b>(1),  <b><a href="../man1/compstruct.1.html">compstruct</a></b>(1),  <b><a href="../man1/revcomp.1.html">revcomp</a></b>(1),  <b><a href="../man1/seqsplit.1.html">seqsplit</a></b>(1),  <b><a href="../man1/seqstat.1.html">seqstat</a></b>(1), <b><a href="../man1/sfetch.1.html">sfetch</a></b>(1),
       <b><a href="../man1/sindex.1.html">sindex</a></b>(1), <b><a href="../man1/sreformat.1.html">sreformat</a></b>(1), <b><a href="../man1/stranslate.1.html">stranslate</a></b>(1), <b><a href="../man1/weight.1.html">weight</a></b>(1).

</pre><h4><b>AUTHOR</b></h4><pre>
       Biosquid and its documentation are Copyright (C) 1992-2003 HHMI/Washington University School of  Medicine
       Freely distributed under the GNU General Public License (GPL) See COPYING in the source code distribution
       for more details, or contact me.

       Sean Eddy
       HHMI/Department of Genetics
       Washington University School of Medicine
       4444 Forest Park Blvd., Box 8510
       St Louis, MO 63108 USA
       Phone: 1-314-362-7666
       FAX  : 1-314-362-2157
       Email: <a href="mailto:eddy@genetics.wustl.edu">eddy@genetics.wustl.edu</a>

Biosquid 1.9g                                     January 2003                                        <u><a href="../man1/shuffle.1.html">shuffle</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>