<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>SIBsim4 - align RNA sequences with a DNA sequence, allowing for introns</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/sibsim4">sibsim4_0.20-5_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       SIBsim4 - align RNA sequences with a DNA sequence, allowing for introns

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>SIBsim4</b> <b>[</b> <u>options</u> <b>]</b> <u>dna</u> <u>rna_db</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>SIBsim4</b>  is  a  similarity-based tool for aligning a collection of expressed sequences (EST, mRNA) with a
       genomic DNA sequence.

       Launching <b>SIBsim4</b> without any arguments will print the options list, along with their default values.

       <b>SIBsim4</b> employs a blast-based technique to first determine the basic  matching  blocks  representing  the
       "exon  cores".   In this first stage, it detects all possible exact matches of W-mers (i.e., DNA words of
       size W) between the two sequences and extends them to maximal scoring gap-free segments.  In  the  second
       stage,  the  exon  cores are extended into the adjacent as-yet-unmatched fragments using greedy alignment
       algorithms, and heuristics are used to favor configurations that conform to the  splice-site  recognition
       signals  (e.g.,  GT-AG).  If  necessary,  the  process  is repeated with less stringent parameters on the
       unmatched fragments.

       By default, <b>SIBsim4</b> searches both strands and reports  the  best  matches,  measured  by  the  number  of
       matching  nucleotides  found  in  the  alignment.   The <b>R</b> command line option can be used to restrict the
       search to one orientation (strand) only.

       Currently, four major alignment display options are supported, controlled by the <b>A</b>  option.  By  default,
       only  the endpoints, overall similarity, and orientation of the introns are reported. An arrow sign ('-&gt;'
       or '&lt;-') indicates the orientation of the intron.  The sign `==' marks the absence from the alignment  of
       a cDNA fragment starting at that position.

       In  the description below, the term <b>MSP</b> denotes a maximal scoring pair, that is, a pair of highly similar
       fragments in the two sequences, obtained during the blast-like procedure by  extending  a  W-mer  hit  by
       matches and perhaps a few mismatches.

</pre><h4><b>OPTIONS</b></h4><pre>
       -A &lt;int&gt;
              output format
                0: exon endpoints only
                1: alignment text
                3: both exon endpoints and alignment text
                4: both exon endpoints and alignment text with polyA info

              Note that 2 is unimplemented.

              Default value is 0.

       -C &lt;int&gt;
              MSP score threshold for the second pass.

              Default value is 12.

       -c &lt;int&gt;
              minimum score cutoff value.  Alignments which have scores below this value are not reported.

              Default value is 50.

       -E &lt;int&gt;
              cutoff value.

              Default value is 3.

       -f &lt;int&gt;
              score  filter  in  percent.   When multiple hits are detected for the same RNA element, only those
              having a score within this percentage of the maximal score for  that  RNA  element  are  reported.
              Setting  this value to 0 disables filtering and all hits will be reported, provided their score is
              above the cutoff value specified through the <b>c</b> option.

              Default value is 75.

       -g &lt;int&gt;
              join exons when gap on genomic and RNA have lengths which differ at most by this percentage.

              Default value is 10.

       -H &lt;int&gt;
              report chimeric transcripts when the best score is lower than this percentage of the  overall  RNA
              coverage  and the chimera score is greater than this percentage of the RNA length (0 disables this
              report)

              Default value is 75.

       -I &lt;int&gt;
              window width in which to search for intron splicing.

              Default value is 6.

       -K &lt;int&gt;
              MSP score threshold for the first pass.

              Default value is 16.

       -L &lt;str&gt;
              a comma separated list of forward splice-types.

              Default value is "GTAG,GCAG,GTAC,ATAC".

       -M &lt;int&gt;
              scoring splice sites, evaluate match within M nucleotides.

              Default value is 10.

       -o &lt;int&gt;
              when printing results, offset nt positions in dna sequence by this amount.

              Default value is 0.

       -q &lt;int&gt;
              penalty for a nucleotide mismatch.

              Default value is -5.

       -R &lt;int&gt;
              direction of search
                0: search the '+' (direct) strand only
                1: search the '-' strand only
                2: search both strands

              Default value is 2.

       -r &lt;int&gt;
              reward for a nucleotide match.

              Default value is 1.

       -S &lt;int&gt;
              splice site indels search breadth.  While determining the best position of a splice site,  <b>SIBsim4</b>
              will  evaluate  adding  at  most this number of insertions and deletions on the DNA strand on each
              side of the splice junction.

              Default value is 2.

       -s &lt;int&gt;
              split score in percent.  While linking MSP, if two consecutive group of  exons  appear  like  they
              could be part of two different copies of the same gene, they will be tested to see if the score of
              each  individual  group  relative  to  the best overall score is greater than this value.  If both
              groups have a relative score above this threshold they will be split.

              Default value is 75.

       -W &lt;int&gt;
              word size.

              Default value is 12.

       -X &lt;int&gt;
              value for terminating word extensions.

              Default value is 12.

Bioinformatics                                     April 2007                                         <u><a href="../man1/SIBsim4.1.html">SIBsim4</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>