<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>sigma - Simple greedy multiple alignment of non-coding DNA sequences</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/sigma-align">sigma-align_1.1.3-9build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       sigma - Simple greedy multiple alignment of non-coding DNA sequences

</pre><h4><b>SYNOPSIS</b></h4><pre>

       <b>sigma</b> [options] [<u>inputfile.fasta</u>] [<u>inputfile2.fasta</u> <u>...</u>]

       Each fasta file may contain a single sequence or multiple sequences; all sequences will be aligned
       together.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>Sigma</b> ("Simple greedy multiple alignment") is an alignment program with a new algorithm and scoring
       scheme designed specifically for non-coding DNA sequence. It uses a strategy of seeking the best possible
       gapless local alignments, at each step making the best possible alignment consistent with existing
       alignments, and scores the significance of the alignment based on the lengths of the aligned fragments
       and a background model which may be supplied or estimated from an auxiliary file of intergenic DNA. With
       real data, while "correctness" can't be directly quantified for the alignment, running the PhyloGibbs
       motif finder on pre-aligned sequence suggests that Sigma's alignments are superior.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-A</b> <b>--aligned_output</b>
          Aligned, pretty-printed output (compare with <b>-F</b> option) (default: only this). See also <b>-C</b>.

       <b>-b</b> <b>--bgprobfile</b> <u>filename</u>
          Auxiliary file (in fasta format) from which to read background sequences (overridden by <b>-B</b>). Typically
          this is a file containing large quantities of similar non-coding sequence, from which background
          probabilities of single- and di-nucleotides may be estimated.

       <b>-B</b> <b>--bgseqfile</b> <u>filename</u>
          File containing background probabilities. The format is described further below.

       <b>-C</b> <b>--caps_only</b>
          Use only upper-case letters in output sequence, for compatibility with output of some other programs
          like ClustalW and MLagan. By default, output is mixed-case (as in Dialign), and lower-case bases are
          treated as not aligned.

       <b>-F</b> <b>--fasta_output</b>
          Multi-fasta output (can use both <b>-A</b> and <b>-F</b> in either order). See also <b>-C</b>.

       <b>-n</b> <b>--ncorrel</b> <u>number</u>
          Background correlation (default <u>2</u>=dinucleotide; <u>1</u>=single-site basecounts, <u>0</u>=0.25 per base).

       <b>-x</b>, <b>--significance</b> <u>number</u>
          Set limit for how probable the match is by chance (default 0.002, smaller=more stringent).

       <b>-h</b>, <b>--help</b>
          Displays this list of options.

</pre><h4><b>MORE</b> <b>HELP</b></h4><pre>
       The "significance" parameter (<b>-x</b>) determines whether local alignments are accepted or rejected. The
       default at present is 0.002. Experiments on synthetic data (described in the paper) suggest that 0.002 is
       about the threshold where sigma fails to align phylogenetically-unrelated data that has moderate
       (yeast-like) dinucleotide correlation.

       Using a “background model” appropriate to the sequences being aligned greatly reduces spurious alignments
       on synthetic data (and, one hopes, on real data too). The simplest way to ensure this is to supply, via
       the <b>-b</b> parameter, a FASTA-format file containing large quantities of similar sequence data (eg, if one is
       aligning yeast sequences, supply a file containing all intergenic yeast sequence).

       Instead of this, if the single-site and dinucleotide frequencies are known already, they may be supplied
       in a file via the <b>-B</b> option. The file format should be: one entry per line, with the mononucleotide or
       dinucleotide (case-insensitive) followed by the frequency. (eg, "A 0.3", "AT 0.16", etc on successive
       lines.) A sample file is in the "Background" subdirectory of the source distribution (on Debian systems,
       this file can be found in the <u>/usr/share/doc/sigma-align/Background</u> directory). A file like
       "yeast.nc.3.freq" in the "tests" subdirectory of the MEME source distribution works fine (trinucleotide
       counts are ignored).

</pre><h4><b>REFERENCE</b></h4><pre>
       Please cite Sigma: Rahul Siddharthan (2006) Multiple alignment of weakly-conserved non-coding DNA
       sequence BMC Bioinformatics 2006, 7:143 doi:10.1186/1471-2105-7-143 Published 16 March 2006, available
       online at <a href="http://www.biomedcentral.com/1471-2105/7/143/">http://www.biomedcentral.com/1471-2105/7/143/</a>

</pre><h4><b>AUTHORS</b></h4><pre>
       <b>Rahul</b> <b>Siddharthan</b> &lt;<a href="mailto:rsidd@imsc.res.in">rsidd@imsc.res.in</a>&gt;
          Wrote  sigma.  If  you're  using  Sigma for actual research, please let the author know so that he can
          alert you of bugfixes or new releases.

       <b>Charles</b> <b>Plessy</b> &lt;<a href="mailto:charles-debian-nospam@plessy.org">charles-debian-nospam@plessy.org</a>&gt;
          Wrote the manpage in DocBook XML for the Debian distribution.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright © 2006-2007 Rahul Siddharthan
       Copyright © 2006-2007 Charles Plessy

       Sigma is free software. You can redistribute it and/or modify it under  the  terms  of  the  GNU  General
       Public License as published by the Free Software Foundation.

       On   Debian   systems,   the   complete  text  of  the  GNU  General  Public  License  can  be  found  in
       /usr/share/common-licenses/GPL.

sigma 1.1                                          2007-04-07                                           <u><a href="../man1/SIGMA.1.html">SIGMA</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>