<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>sim4 - align an expressed DNA sequence with a genomic sequence</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/sim4">sim4_0.0.20121010-8_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       sim4 - align an expressed DNA sequence with a genomic sequence

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>sim4</b> <u>seqfile1</u> <u>seqfile2</u> {[WXKCRDAPNB]=<u>value</u>}

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>sim4</b>  is  a similarity-based tool for aligning an expressed DNA sequence (EST, cDNA, mRNA) with a genomic
       sequence for the gene. It also detects end matches when the two input sequences overlap at one end (i.e.,
       the start of one sequence overlaps the end of the other). If <u>seqfile2</u> is a  database  of  sequences,  the
       sequence in <u>seqfile1</u> will be aligned with each of the sequences in <u>seqfile2</u>.

       <b>sim4</b>  employs a blast-based technique to first determine the basic matching blocks representing the "exon
       cores". In this first stage, it detects all possible exact matches of W-mers (i.e., DNA words of size  W)
       between the two sequences and extends them to maximal scoring gap-free segments. In the second stage, the
       exon  cores  are extended into the adjacent as-yet-unmatched fragments using greedy alignment algorithms,
       and heuristics are used to favor configurations that conform to the splice-site recognition signals  (GT-
       AG,  CT-AC).  If  necessary,  the  process  is  repeated  with less stringent parameters on the unmatched
       fragments.

       By default, <b>sim4</b> searches both strands and reports the best match, measured by  the  number  of  matching
       nucleotides  found  in the alignment. The R command line option can be used to restrict the search to one
       orientation (strand) only.

       Currently, five major alignment display options are supported, controlled by the  A  option.  By  default
       (A=0), only the endpoints, overall similarity, and orientation of the introns are reported. An arrow sign
       (`-&gt;' or `&lt;-') indicates the orientation of the intron (`+' or `-' strand), when the signals flanking the
       intron have three or more position matches with either the GT-AG or the CT-AC splice recognition signals.
       When  the same number of matches is found for both orientations, the intron is reported as ambiguous, and
       represented by `--'. The sign `==' marks the absence from the alignment of a cDNA  fragment  starting  at
       that  position.  Alternative  formats  (lav-block  format,  text,  PipMaker-type `exons file', or certain
       combinations of these options) can be requested by specifying a different value for A.

       If the P option is specified with a non-zero value, <b>sim4</b> will remove any  3'-end  poly-A  tails  that  it
       detects in the alignment.

       Occasionally, <b>sim4</b> may miss an internal exon when surrounded by very large introns, typically longer than
       100 Kb. When this is suspected, the H option can be used to reset the exons' weight to compensate for the
       intron gap penalty.

       Ambiguity  codes  are  by  default  allowed in sequence data, but <b>sim4</b> treats them non-differentially. If
       desired, the B command option can restrict the set of acceptable characters to A,C,G,T,N and X only.

       <b>sim4</b> compares the lengths of the input sequences to  distinguish  between  the  cDNA  (`short')  and  the
       genomic  (`long')  components  in  the  comparison. When <u>seqfile2</u> contains a collection of sequences, the
       first entry in the file will be used to determine the type of this and all subsequent comparisons.

       In the description below, the term MSP denotes a <u>M</u>aximal <u>S</u>egment <u>P</u>air, that is, a pair of highly  similar
       fragments  in  the  two  sequences,  obtained during the blast-like procedure by extending a W-mer hit by
       matches and perhaps a few mismatches.

</pre><h4><b>OPTIONS</b></h4><pre>
       The algorithm parameters (included in the first two sections below) have already been tuned  and  do  not
       normally require adjustment by the user.

       Parameters internal to the blast-like procedure:

       <b>W</b>      Sets  the  word  size for blast hits in the first stage of the algorithm. The default value is 12,
              but it can be increased for a more stringent search or decreased to find weaker matches.

       <b>X</b>      Controls the limits for terminating word extensions in the blast-like stage of the algorithm.  The
              default value is 12.

       <b>K</b>      Sets  the  threshold  for the MSP scores when determining the basic `exon cores', during the first
              stage of the algorithm. (If this option is not specified,  the  threshold  is  computed  from  the
              lengths  of  the  sequences,  using  statistical  criteria.) For example, a good value for genomic
              sequences in the range of a few hundred Kb is 16. To avoid spurious  matches,  however,  a  larger
              value may be needed for longer sequences.

       <b>C</b>      Sets  the  threshold  for  the MSP scores when aligning the as-yet-unmatched fragments, during the
              second stage of the algorithm. By default, the smaller of the constant 12 and  a  statistics-based
              threshold is chosen.

       Additional algorithm parameters:

       <b>D</b>      Sets  the  bound for the "diagonal" distance within consecutive MSPs in an exon. The default value
              is 10.

       Context parameters:

       <b>R</b>      Specifies the direction of the search. If R=0, only the "+" (direct) strand is searched.  If  R=1,
              only the "-" (reverse complement) matches are sought. By default (R=2), sim4 searches both strands
              and reports the best match, measured by the number of matching pairs in the alignment.

       <b>A</b>      Specifies  the  format  of the output: exon endpoints only (A=0), exon endpoints and boundaries of
              the coding region (CDS) in the  genomic  sequence,  when  specified  for  the  input  mRNA  (A=5),
              alignment  text  (A=1),  alignment in lav-block format (A=2), or both exon endpoints and alignment
              text (A=3 or A=4). If a reverse complement match is found, A=0,1,2,3,5 will give its  position  in
              the  "+"  strand  of the longer sequence and the "-" strand of the shorter sequence. A=4 will give
              its position in the "+" strand of the first sequence (seqfile1) and the "-" strand of  the  second
              sequence (seqfile2), regardless of which sequence is longer. The A=5 option can be used with the S
              command  line  option  to specify the endpoints of the CDS in the mRNA, and produces output in the
              `exons file' format required by PipMaker.

       <b>P</b>      Specifies whether or not the program should report the fragment of the  alignment  containing  the
              poly-A  tail (if found). By default (P=0) the alignment is displayed as computed, but specifying a
              non-zero value will request sim4 to remove the poly-A tails. When this  feature  is  enabled,  all
              display options produce additional lav alignment headers.

       <b>H</b>      Resets the MSPs' weight to compensate for very large introns. The default value is H=500, but some
              introns larger than 100 Kb may require higher values, typically between 1000 and 2500. This option
              should  be used cautiously, generally in cases where an unmatched internal portion of the cDNA may
              disguise a missed exon within a very large intron. It is not recommended for ESTs, where they  may
              produce spurious exons.

       <b>N</b>      Requests an additional search for small marginal exons (N=1) guided by the splice-site recognition
              signals. This option can be used when a high accuracy match is expected. The default value is N=0,
              specifying no additional search.

       <b>B</b>      Controls  the  set  of  characters  allowed  in  the  input sequences. By default (B=1), ambiguity
              characters (ABCDGHKMNRSTVWXY) are allowed. By specifying B=0, the set of acceptable characters  is
              restricted to A,C,G,T,N and X only.

       <b>S</b>      Allows  the user to specify the endpoints of the CDS in the input mRNA, with the syntax: S=n1..n2.
              This option is only available with the A=5 flag, which produces output in the format  required  by
              PipMaker.  Alternatively,  the CDS coordinates could appear in a construct CDS=n1..n2 in the FastA
              header of the mRNA sequence.  When  the  second  file  is  an  mRNA  database,  the  command  line
              specification for the CDS will apply to the first sequence in the file only.

</pre><h4><b>EXAMPLES</b></h4><pre>
       sim4 est genomic

       sim4 genomic estdb

       sim4 est genomic A=1 P=1

       sim4 est1 est2 R=1

       sim4 mRNA genomic A=5 S=123..1020

       sim4 mouse_cDNA human_genomic K=15 C=11 A=3 W=10

</pre><h4><b>AUTHORS</b></h4><pre>
       sim4 was written by Liliana Florea &lt;<a href="mailto:florea@gwu.edu">florea@gwu.edu</a>&gt; and Scott Schwartz.

       This  manual  page  was  written  by  Nelson  A.  de  Oliveira  &lt;<a href="mailto:naoliv@gmail.com">naoliv@gmail.com</a>&gt;,  based  on the online
       documentation at <a href="http://globin.cse.psu.edu/html/docs/sim4.html">http://globin.cse.psu.edu/html/docs/sim4.html</a>, for the Debian project (but may  be  used
       by others).

                                         Wed, 03 Aug 2005 18:40:58 -0300                                 <u><a href="../man1/SIM4.1.html">SIM4</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>