<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>smithwaterman - determine similar regions between two strings or nucleotide or protein sequences</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/libsmithwaterman0">libsmithwaterman0_0.0+git20160702.2610e25-12_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       smithwaterman - determine similar regions between two strings or nucleotide or protein sequences

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>smithwaterman</b> [options] &lt;reference sequence&gt; &lt;query sequence&gt;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The  Smith–Waterman algorithm performs local sequence alignment; that is, for determining similar regions
       between two strings or nucleotide or protein sequences. Instead of looking at  the  total  sequence,  the
       Smith–Waterman algorithm compares segments of all possible lengths and optimizes the similarity measure.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-m</b>, <b>--match-score</b>
              the match score (default 10.0)

       <b>-n</b>, <b>--mismatch-score</b>
              the mismatch score (default <b>-9</b>.0)

       <b>-g</b>, <b>--gap-open-penalty</b>
              the gap open penalty (default 15.0)

       <b>-z</b>, <b>--entropy-gap-open-penalty</b>
              enable entropy scaling of the gap open penalty

       <b>-e</b>, <b>--gap-extend-penalty</b>
              the gap extend penalty (default 6.66)

       <b>-r</b>, <b>--repeat-gap-extend-penalty</b>
              use repeat information when generating gap extension penalties

       <b>-b</b>, <b>--bandwidth</b>
              bandwidth to use (default 0, or non-banded algorithm)

       <b>-p</b>, <b>--print-alignment</b>
              print out the alignment

       <b>-R</b>, <b>--reverse-complement</b>
              report the reverse-complement alignment if it scores better

       When  called  with literal reference and query sequences, smithwaterman prints the cigar match positional
       string and the match position for the query sequence against the reference sequence.

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

smithwaterman 0.0+20151117                          June 2016                                   <u><a href="../man1/SMITHWATERMAN.1.html">SMITHWATERMAN</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>