<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>spaced - alignment-free sequence comparison</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/spaced">spaced_1.2.0-201605+dfsg-4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       spaced - alignment-free sequence comparison

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>spaced</b> [<u>-r</u>] [<u>-k</u> <u>INT</u>] [<u>-l</u> <u>INT</u>] [<u>-n</u> <u>INT</u>] [<u>-t</u> <u>INT</u>] [<u>-d</u> <u>TYPE</u>] [<u>-f</u> <u>FILE</u>] <u>FILES</u>...

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Spaced Words is a new approach to alignment-free sequence comparison. While most alignment-free
       algorithms compare the word-composition of sequences, Spaced Words uses a pattern of care and don't care
       positions. The occurrence of a spaced word in a sequence is then defined by the characters at the match
       positions only, while the characters at the don't care positions are ignored (this was originally
       inspired by the PatternHunter algorithm for homology search in databases). Instead of comparing the
       frequencies of contiguous words in the input sequences, our new approach compares the frequencies of the
       spaced words according to the pre-defined pattern. An information-theoretic distance measure is then used
       to define pairwise distances on the set of input sequences based on their spaced-word frequencies. The
       original version of our spaced-words approach was published in Boden et <a href="../man2013/al..2013.html">al.</a>(2013).

</pre><h4><b>OUTPUT</b></h4><pre>
       The  output  is  a  symmetrical  distance  matrix  similar to <u>PHYLIP</u> format, with each entry representing
       divergence with a positive real number. A distance of  zero  means  that  two  sequences  are  identical,
       whereas other values are estimates for the nucleotide substitution rate (Jukes-Cantor corrected).

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-o</b> &lt;file&gt;
              Print the distance matrix to the given <u>file</u>. Default is <u>DMat</u>.

       <b>-k</b> &lt;int&gt;
              Set the patterns weight. Default: 14.

       <b>-l</b> &lt;int&gt;
              Set <u>don't</u> <u>care</u> positions for the used patterns. Default: 15.

       <b>-n</b> &lt;int&gt;
              Set the number of patterns. Default: 5.

       <b>-f</b> &lt;file&gt;
              Instead of generating new patterns, use read them from the given file.

       <b>-t</b> &lt;INT&gt;
              The number of threads to be used; by default, 25 threads are used.
              Multithreading is only available if <b>spaced</b> was compiled with OpenMP support.

       <b>-r</b>     Skip comparison with the reverse complement.

       <b>-d</b> &lt;type&gt;
              The  distances  can  be  compute  with  different  measures. Available options are Euclidean (<u>EU</u>),
              Jensen-Shannon (<u>JS</u>), and evolutionary distance (<u>EV</u>). Default: EV.

       <b>-h</b>     Prints the synopsis and an explanation of available options.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright Â© 2016 Chris  Leimeister  &lt;<a href="mailto:chris.leimeister@stud.uni-goettingen.de">chris.leimeister@stud.uni-goettingen.de</a>&gt;  License  GPLv3+:  GNU  GPL
       version 3 or later.
       This  is  free software: you are free to change and redistribute it.  There is NO WARRANTY, to the extent
       permitted by law.  The full license text is available at &lt;<a href="http://gnu.org/licenses/gpl.html">http://gnu.org/licenses/gpl.html</a>&gt;.

</pre><h4><b>REFERENCES</b></h4><pre>
       1) C.-A. Leimeister, M. Boden, S.  Horwege,  S.  Lindner,  B.  Morgenstern  (2014).  Fast  alignment-free
       sequence         comparison         using         spaced-word         frequencies,         Bioinformatics
       &lt;<a href="http://bioinformatics.oxfordjournals.org/content/early/2014/04/03/bioinformatics.btu177">http://bioinformatics.oxfordjournals.org/content/early/2014/04/03/bioinformatics.btu177</a>&gt;
       2) S. Horwege, S. Linder, M. Boden, K. Hatje, M. Kollmar, C.-A. Leimeister, B. Morgenstern (2014). Spaced
       words and kmacs: fast alignment-free sequence comparison based on inexact  word  matches,  Nucleic  Acids
       Research 42, W7-W11 &lt;<a href="http://nar.oxfordjournals.org/content/42/W1/W7.abstract">http://nar.oxfordjournals.org/content/42/W1/W7.abstract</a>&gt;
       3)  B. Morgenstern, B. Zhu, S. Horwege, C.-A Leimeister (2015). Estimating evolutionary distances between
       genomic sequences from spaced-word matches, Algorithms for Molecular Biology 10,5

</pre><h4><b>BUGS</b></h4><pre>
   <b>Reporting</b> <b>Bugs</b>
       Please report bugs to &lt;<a href="mailto:kloetzl@evolbio.mpg.de">kloetzl@evolbio.mpg.de</a>&gt; or &lt;<a href="mailto:chris.leimeister@stud.uni-goettingen.de">chris.leimeister@stud.uni-goettingen.de</a>&gt;.

1.0.2                                             February 2016                                        <u><a href="../man1/SPACED.1.html">SPACED</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>