<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>spaln  -  mapping  and alignment of cDNA/protein sequences onto genomic sequence or rapid homology search</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/spaln">spaln_3.0.2+dfsg-2ubuntu2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       spaln  -  mapping  and alignment of cDNA/protein sequences onto genomic sequence or rapid homology search
       against protein sequence database and (semi)global alignment

</pre><h4><b>SYNOPSIS</b></h4><pre>
       spaln -WGenome.bk[n|p] -K[D|P] [W_Options] Genome.mfa(.gz)
       spaln -WProsDb.bka -KA [W_Options] ProteinSeqenceDB.faa(.gz)
       spaln [-Q[0|1|2|3]] [R_options] Genome_segment [cDNA|protein]_queries
       spaln -Q[4|5|6|7] -dGenome [R_options] [cDNA|protein]_queries
       spaln -Q[4|5|6|7] -aProsDb [R_options] Genomic_segment
       spaln -Q[4|5|6|7] -aProsDb [R_options] protein_queries
       spaln -Q[4|5|6|7] [R_options] Genome.mfa [cDNA|protein]_queries
       spaln -Q[4|5|6|7] [R_options] ProsDb.faa protein_queries

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>spaln</b> is a stand-alone program that maps and aligns a  set  of  cDNA/protein  sequences  onto  a  genomic
       sequence.   From  version  1.4,  <b>spaln</b>  supports  a  combination of protein sequence database and a given
       genomic segment. From version 2.2, it also supports rapid similarity search of protein sequences  against
       a protein sequence database followd by ordinary alignment.

       <b>Spaln</b>  runs  with relatively small internal memory, making it feasible to search a whole mammalian genome
       in a single job on a conventional computer.

</pre><h4><b>EXAMPLES</b></h4><pre>
       <b>spaln</b> <b>-Wddignm.bkn</b> <b>-KD</b> <b>-Xk10</b> <b>-Xb8192</b> <b>ddignm.mfa</b>
              Make the block index table 'ddignm.bkn' of the genomic sequence 'ddignm.mfa' in multi-fasta format
              with the word size of 10 nucleotides and the block size of 8192.

       <b>spaln</b> <b>-O1</b> <b>'chr1.fa</b> <b>10001</b> <b>40000</b> <b>&lt;'</b> <b>cdna.fa</b>
              Align the cDNA sequence 'cdna.fa' onto the genomic segment of 'chr1.fa' within the range from  the
              10001-th to the 40000-th nucleotide of the complementary strand in the semi-global mode.

       <b>spaln</b> <b>-Q7</b> <b>-LS</b> <b>-t10</b> <b>-dddignm</b> <b>ddicdna.mfa</b> <b>&gt;</b> <b>ddi.exon</b>
              Report the gene structures corresponding to individual cDNAs in 'ddicdna.mfa'. The local alignment
              mode  is  used  and  the  results  are  written  in  'ddi.exon'  in the exon-oriented format.  The
              computation proceeds with 10 threads in parallel.

       <b>spaln</b> <b>-Q7</b> <b>-O5</b> <b>-XG1M</b> <b>-yX</b> <b>-TMus</b> <b>-ommu.intron</b> <b>-dmmugnm</b> <b>'ratcdna.mfa</b> <b>(1</b> <b>20)'</b>
              Cross-species comparison between the genome 'mmugnm' and the cDNAs in 'ratcdna.mfa' from the first
              to the 20-th entries.  The maximal expected gene size is reset  to  1  Mbp.   The  outputs  go  to
              'mmu.intron' in the intron-oriented format. The tetrapod-specific parameter set is used.

       <b>spaln</b> <b>-Q7</b> <b>-O0</b> <b>-Tarabthal</b> <b>-aSwiss</b> <b>-oyour_genes.gff3</b> <b>your_genomic_segment</b>
              A  set  of  ORFs  in  the  genomic segment are translated and rapidly searched against the protein
              database, and then the best-hit sequence is used as the template of the spliced alignment to infer
              the organization of the gene located in the relevant genomic region. The output is shown  in  Gff3
              gene format.  The dicot-specific parameter set is used.

       <b>spaln</b> <b>-Q4</b> <b>-O0</b> <b>-aSwiss</b> <b>-M4</b> <b>-t10</b> <b>aa_queries</b> <b>&gt;</b> <b>output</b>
              Find  up  to  four  SwissProt  sequences  most  similar  to each query sequence, and report global
              alignment statistics. To show  alignment  themselves,  use  -O1  option  in  stead  of  -O0.   The
              computation proceeds with 10 threads in parallel.

</pre><h4><b>OPTIONS</b></h4><pre>
       Conventions: #: number; $: string; default values in ()
        (default for DNA, with -yX option, default for protein)

</pre><h4><b>FORMATING</b> <b>OPTIONS</b></h4><pre>
       <b>-K$</b>    Format the genomic sequence for DNA ($=D) or protein ($=P) queries, or format the protein database
              sequences ($=A) for rapid search of template.

       <b>-Xk#</b>   Word size (11 for DNA, 5 for protein)

       <b>-Xb#</b>   Block size (4096)

       <b>-XG#</b>   Maximum gene size (262144)

       <b>-Xa#</b>   Abundance factor (10)

       <b>-Xs#</b>   Shifts between adjacent seeds (k)

</pre><h4><b>RUN</b> <b>TIME</b> <b>OPTIONS</b></h4><pre>
       <b>-C#</b>    NCBI transl_table number indicating the genetic code (1)

       <b>-H#</b>    Minimum alignment score for report (35)

       <b>-LS</b>    Smith-Waterman-type local alignment

       <b>-M[#]</b>  Multiple loci (0)
              #=empty: Multiple loci maximally up to 4
              #=0: Single locus
              #=1: Re-search unaligned parts
              #&gt;1: Multiple loci maximally up to #

       <b>-O[#]</b>  Output format (4)
              #=0: GFF3 format (gene) in genome vs [cDNA|protein] mode
              #=0: Alignment statistics in protein vs protein mode
              #=1: Alignment
              #=2: GFF3 format (match)
              #=3: Bed format
              #=4: Exon-oriented format similar to output of megablast -D 3
              #=5: Intron-oriented output
              #=6: Concatenated exon sequence
              #=7: Translated amino-acid sequence
              #=8: Mapping (block) information only. Use with -Q4
              #=12: Output the same information as -O4 in binary formats

       <b>-Q[#]</b>  Select algorithm (3)
              0&lt;=#&lt;=3: Genomic segment in the fasta format given by the first argument vs. cdna/protein given by
              the second argument
              4&lt;=#&lt;=7: Genome mapping and alignment
              #=0,4: DP procedure without HSP search
              #=1-3,5-7: Recursive HSP searches up to the level of (# % 4)

       <b>-R$</b>    Read block index table from the file $

       <b>-S[#]</b>  Specify the orientation of query (0)
              0: depend on the query annotation
              1: forward direction only
              2: reverse direction only
              3: both directions;

       <b>-T$</b>    Specify the species (in combination with the -yS option for cDNA query)

       <b>-U</b>     Map/align without splicing

       <b>-V#</b>    Minimum space to induce the Hirschberg's algorithm (16M)

       <b>-W$</b>    Write block index table to file $

       <b>-i[a|p]</b>
              Paired-end reads as the queries from a single or two files
              -ia: 5' and 3' matching pairs must appear alternatively in a file
              -ip: 5' and 3' matching pairs must appear in the same order in the two files

       <b>-o$</b>    Destination of output file (stdout)

       <b>-pa</b>    Suppress trimming of terminal polyA or polyT sequence

       <b>-pq</b>    Suppress some outputs to stderr

       <b>-pw</b>    Report the result irrespective of the alignment score

       <b>-u#</b>    Gap-extension penalty (3, 2, 2)

       <b>-v#</b>    Gap-opening penalty (8, 6, 9)

       <b>-xB$</b>   Bit pattern of the seeds used for HSP search at level 1

       <b>-xb$</b>   Bit pattern of the seeds used for HSP search at level 3

       <b>-ya#</b>   Dinucleotide pairs at the ends of an intron (0)
              0: canonical only (GT..AG, GC..AG, AT..AC)
              1: relaxed to (GT..AG, GC..AG, AT..AN)
              2: #=1 + allow 1 mismatch from GT..AG
              3: any;

       <b>-yi#</b>   Intron penalty (11, 8, 11)

       <b>-yj#</b>   Incline of long gap penalty (0.6)

       <b>-yk#</b>   Flex point where the incline of gap penalty changes (7)

       <b>-yl#</b>   Double affine gap penalty if #=3; affine penalty otherwise

       <b>-ym#</b>   Score for a nucleotide match (2, 2)

       <b>-yn#</b>   Penalty for a nucleotide mismatch (6, 2)

       <b>-yo#</b>   Penalty for in-frame termination codon (100)

       <b>-yp#</b>   PAM level used in the alignment (third) phase (150)

       <b>-yq#</b>   PAM level used in the second phase (50)

       <b>-yx#</b>   Penalty for a frame shift (100)

       <b>-yy#</b>   Relative contribution of splicing signal (8)

       <b>-yz#</b>   Relative contribution of coding potential (2)

       <b>-yA#</b>   Relative contribution of the translational initiation or termination signal (8)

       <b>-yB#</b>   Relative contribution of branch point signal (0)

       <b>-yE#</b>   Minimum exon length (2)

       <b>-yI$</b>   Intron distribution parameters

       <b>-yJ#</b>   Relative contribution of the bonus given to a conserved intron position

       <b>-yL#</b>   The minimum intron length (20)

       <b>-yS#</b>   Percent  contribution  of  the  species-specific splice signal. if #=0, only the ubiquitous signal
              given to the dinucleotide pair at the ends of an intron is used. By default #=0 for DNA and  #=100
              for  protein  queries.  -yX option automatically sets #=100 for DNA and #=30 for protein.  <b>-yS</b> For
              cDNA queries, use species-specific exon-intron  boundary  signals.  For  protein  queries,  invoke
              'salvage' procedure in phase 1

       <b>-yX</b>    For  a DNA query, this option sets parameter values for cross-species comparison. Conversely, this
              option specifies an intra-species mode for a protein query.

       <b>-yY#</b>   Relative contribution of length-dependent part of intron penalty (8)

       <b>-yZ#</b>   Relative contribution of oligomer composition within an intron (0)

</pre><h4><b>REFERENCES</b></h4><pre>
       (1) "A Space-Efficient and  Accurate  Method  for  Mapping  and  Aligning  cDNA  Sequences  onto  Genomic
       Sequence", Osamu Gotoh, Nucleic Acid Res., 36 (8), 2630-2638 (2008).
       (2)   "Direct   Mapping  and  Alignment  of  Protein  Sequences  onto  Genomic  Sequence",  Osamu  Gotoh,
       Bioinformatics, 24 (21) 2438-2444 (2008).
       (3) "Benchmarking spliced alignment programs  including   Spaln2,  an  extended  version  of  Spaln  that
       incorporates additional species-specific features", Iwata, H. and Gotoh, O.", Nucleic Acids Res., 40 (20)
       e161 (2012).

</pre><h4><b>AUTHOR</b></h4><pre>
       Osamu Gotoh &lt;<a href="mailto:o.gotoh@aist.go.jp">o.gotoh@aist.go.jp</a>&gt;

                                                   2018-09-06                                           <u><a href="../man1/spaln.1.html">spaln</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>