<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>spidey - align mRNA sequences to a genome</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/ncbi-tools-bin">ncbi-tools-bin_6.1.20170106+dfsg2-5_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       spidey - align mRNA sequences to a genome

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>spidey</b>   [<b>-</b>]   [<b>-F</b> <u>N</u>]   [<b>-G</b>]  [<b>-L</b> <u>N</u>]  [<b>-M</b> <u>filename</u>]  [<b>-N</b> <u>filename</u>]  [<b>-R</b> <u>filename</u>]  [<b>-S</b> <u>p/m</u>]  [<b>-T</b> <u>N</u>]  [<b>-X</b>]
       [<b>-a</b> <u>filename</u>] [<b>-c</b> <u>N</u>] [<b>-d</b>] [<b>-e</b> <u>X</u>] [<b>-f</b> <u>X</u>] [<b>-g</b> <u>X</u>] <b>-i</b> <u>filename</u> [<b>-j</b>] [<b>-k</b> <u>filename</u>] [<b>-l</b> <u>N</u>]  <b>-m</b> <u>filename</u>  [<b>-n</b> <u>N</u>]
       [<b>-o</b> <u>str</u>] [<b>-p</b> <u>N</u>] [<b>-r</b> <u>c/d/m/p/v</u>] [<b>-s</b>] [<b>-t</b> <u>filename</u>] [<b>-u</b>] [<b>-w</b>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>spidey</b> is a tool for aligning one or more mRNA sequences to a given genomic sequence.  <b>spidey</b> was written
       with  two  main goals in mind: find good alignments regardless of intron size; and avoid getting confused
       by nearby pseudogenes and paralogs.  Towards the first goal, <b>spidey</b> uses  BLAST  and  Dot  View  (another
       local alignment tool) to find its alignments; since these are both local alignment tools, <b>spidey</b> does not
       intrinsically  favor  shorter  or  longer  introns  and  has no maximum intron size.  To avoid mistakenly
       including exons from paralogs and pseudogenes, <b>spidey</b> first defines windows on the genomic  sequence  and
       then  performs  the  mRNA-to-genomic  alignment  separately  within  each window.  Because of the way the
       windows are constructed, neighboring paralogs or pseudogenes should be in separate windows and should not
       be included in the final spliced alignment.

   <b>Initial</b> <b>alignments</b> <b>and</b> <b>construction</b> <b>of</b> <b>genomic</b> <b>windows</b>
       <b>spidey</b> takes as input a single genomic sequence and a set of mRNA accessions  or  FASTA  sequences.   All
       processing  is  done  one  mRNA  sequence  at  a  time.  The first step for each mRNA sequence is a high-
       stringency BLAST against the genomic sequence.  The resulting hits  are  analyzed  to  find  the  genomic
       windows.

       The  BLAST  alignments  are  sorted by score and then assigned into windows by a recursive function which
       takes the first alignment and then goes  down  the  alignment  list  to  find  all  alignments  that  are
       consistent  with the first (same strand of mRNA, both the mRNA and genomic coordinates are nonoverlapping
       and linearly consistent).  On subsequent passes, the remaining alignments are examined and are  put  into
       their  own  nonoverlapping, consistent windows, until no alignments are left.  Depending on how many gene
       models are desired, the top <u>n</u> windows are chosen to go on to the next step and the others are deleted.

   <b>Aligning</b> <b>in</b> <b>each</b> <b>window</b>
       Once the genomic windows are constructed, the initial BLAST alignments are freed and another BLAST search
       is performed, this time with the entire mRNA against the genomic region defined by the window, and  at  a
       lower  stringency  than  the  initial  search.   <b>spidey</b>  then uses a greedy algorithm to generate a high-
       scoring, nonoverlapping subset of the alignments from the second BLAST search.  This  consistent  set  is
       analyzed  carefully  to  make sure that the entire mRNA sequence is covered by the alignments.  When gaps
       are found between the alignments, the appropriate region of genomic  sequence  is  searched  against  the
       missing  mRNA,  first  using  a  very  low-stringency  BLAST and, if the BLAST fails to find a hit, using
       DotView functions to locate the alignment.  When gaps are found at the ends of the alignments, the  BLAST
       and  DotView searches are actually allowed to extend past the boundaries of the window.  If the 3' end of
       the mRNA does not align completely, it is first examined for the presence of a poly(A) tail.  No  attempt
       is  made  to  align the portion of the mRNA that seems to be a poly(A) tail; sometimes there is a poly(A)
       tail that does align to the genomic sequence, and these are noted because they indicate  the  possibility
       of a pseudogene.

       Now that the mRNA is completely covered by the set of alignments, the boundaries of the alignments (there
       should  be  one alignment per exon now) are adjusted so that the alignments abut each other precisely and
       so that they are adjacent to good splice donor and acceptor sites.  Most commonly,  two  adjacent  exons'
       alignments  overlap  by  as much as 20 or 30 base pairs on the mRNA sequence.  The true exon boundary may
       lie anywhere within this overlap, or (as we have seen empirically) even a  few  base  pairs  outside  the
       overlap.  To position the exon boundaries, the overlap plus a few base pairs on each side is examined for
       splice donor sites, using functions that have different splice matrices depending on the organism chosen.
       The  top  few  splice  donor  sites (by score) are then evaluated as to how much they affect the original
       alignment boundaries.  The site that affects the boundaries the least is chosen, and is evaluated  as  to
       the  presence  of  an  acceptor site.  The alignments are truncated or extended as necessary so that they
       terminate at the splice donor site and so that they do not overlap.

   <b>Final</b> <b>result</b>
       The windows are examined carefully to get the percent identity per exon, the number of gaps per exon, the
       overall percent identity, the percent coverage of the mRNA,  presence  of  an  aligning  or  non-aligning
       poly(A) tail, number of splice donor sites and the presence or absence of splice donor and acceptor sites
       for each exon, and the occurrence of an mRNA that has a 5' or 3' end (or both) that does not align to the
       genomic sequence.  If the overall percent identity and percent length coverage are above the user-defined
       cutoffs,  a  summary  report  is  printed,  and,  if  requested,  a text alignment showing identities and
       mismatches is also printed.

   <b>Interspecies</b> <b>alignments</b>
       <b>spidey</b> is capable of performing interspecies alignments.  The major difference in interspecies alignments
       is that the mRNA-genomic identity will not be close to 100% as it is in  intraspecies  alignments;  also,
       the  alignments  have numerous and lengthy gaps.  If <b>spidey</b> is used in its normal mode to do interspecies
       alignments, it produces gene models with many, many short exons.  When  the  interspecies  flag  is  set,
       <b>spidey</b>  uses  different BLAST parameters to encourage longer and more gaps and to not penalize as heavily
       for mismatches.  This way, the alignments for the exons are much longer and more closely approximate  the
       actual gene structure.

   <b>Extracting</b> <b>CDS</b> <b>alignments</b>
       When <b>spidey</b> is run in network-aware mode or when ASN.1 files are used for the mRNA records, it is capable
       of  extracting  a  CDS alignment from an mRNA alignment and printing the CDS information also.  Since the
       CDS alignment is just a subset of the mRNA alignment, it is relatively straightforward  to  truncate  the
       exon  alignments as necessary and to generate a CDS alignment.  Furthermore, the untranslated regions are
       now defined, so the percent identity for the 5' and 3' untranslated regions is also calculated.

</pre><h4><b>OPTIONS</b></h4><pre>
       A summary of options is included below.

       <b>-</b>      Print usage message.

       <b>-F</b> <u>N</u>   Start of genomic interval desired (from; 0-based).

       <b>-G</b>     Input file is a GI list.

       <b>-L</b> <u>N</u>   The extra-large intron size to use (default = 220000).

       <b>-M</b> <u>filename</u>
              File with donor splice matrix.

       <b>-N</b> <u>filename</u>
              File with acceptor splice matrix.

       <b>-R</b> <u>filename</u>
              File (including path) to repeat blast database for filtering.

       <b>-S</b> <u>p/m</u> Restrict to plus (p) or minus (m) strand of genomic sequence.

       <b>-T</b> <u>N</u>   Stop of genomic interval desired (to; 0-based).

       <b>-X</b>     Use extra-large intron sizes (increases the limit for initial and terminal introns from  100kb  to
              240kb and for all others from 35kb to 120kb); may result in significantly longer compute times.

       <b>-a</b> <u>filename</u>
              Output file for alignments when directed to a separate file with <b>-p</b> <b>3</b> (default = spidey.aln).

       <b>-c</b> <u>N</u>   Identity cutoff, in percent, for quality control purposes.

       <b>-d</b>     Also  try  to  align coding sequences corresponding to the given mRNA records (may require network
              access).

       <b>-e</b> <u>X</u>   First-pass e-value (default = 1.0e-10).  Higher values increase speed at the cost of sensitivity.

       <b>-f</b> <u>X</u>   Second-pass e-value (default = 0.001).

       <b>-g</b> <u>X</u>   Third-pass e-value (default = 10).

       <b>-i</b> <u>filename</u>
              Input file containing the genomic sequence in ASN.1 or FASTA format.  If your computer is  running
              on  a  network  that  can  access GenBank, you can substitute the desired accession number for the
              filename.

       <b>-j</b>     Print ASN.1 alignment?

       <b>-k</b> <u>filename</u>
              File for ASN.1 output with <b>-k</b> (default = spidey.asn).

       <b>-l</b> <u>N</u>   Length coverage cutoff, in percent.

       <b>-m</b> <u>filename</u>
              Input file containing the mRNA sequence(s) in ASN.1 or FASTA format, or a list of their accessions
              (with <b>-G</b>).  If your computer is running on a network that can access GenBank, you can substitute a
              single accession number for the filename.

       <b>-n</b> <u>N</u>   Number of gene models to return per input mRNA (default = 1).

       <b>-o</b> <u>str</u> Main output file (default = stdout; contents controlled by <b>-p</b>).

       <b>-p</b> <u>N</u>   Print alignment?
              <b>0</b>      summary and alignments together (default)
              <b>1</b>      just the summary
              <b>2</b>      just the alignments
              <b>3</b>      summary and alignments in different files

       <b>-r</b> <u>c/d/m/p/v</u>
              Organism of genomic sequence, used to determine splice matrices.
              <b>c</b>      C. elegans
              <b>d</b>      Drosophila
              <b>m</b>      Dictyostelium discoideum
              <b>p</b>      plant
              <b>v</b>      vertebrate (default)

       <b>-s</b>     Tune for interspecies alignments.

       <b>-t</b> <u>filename</u>
              File with feature table, in 4 tab-delimited columns:
              <u>seqid</u>  (e.g., <b>NM_04377.1</b>)
              <u>name</u>   (only <b>repetitive_region</b> is currently supported)
              <u>start</u>  (0-based)
              <u>stop</u>   (0-based)

       <b>-u</b>     Make a multiple alignment of all input mRNAs (which must overlap on the genomic sequence).

       <b>-w</b>     Consider lowercase characters in input FASTA sequences to be masked.

</pre><h4><b>AUTHOR</b></h4><pre>
       Sarah Wheelan  and  others  at  the  National  Center  for  Biotechnology  Information;  Steffen  Moeller
       contributed to this documentation.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       &lt;<a href="http://www.ncbi.nlm.nih.gov/spidey">http://www.ncbi.nlm.nih.gov/spidey</a>&gt;

NCBI                                               2005-01-25                                          <u><a href="../man1/SPIDEY.1.html">SPIDEY</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>