<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>srst2 - Short Read Sequence Typer</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/srst2">srst2_0.2.0-13_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       srst2 - Short Read Sequence Typer

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>srst2</b>  [<b>-h</b>]  [<b>--version</b>]  [<b>--input_se</b>  INPUT_SE  [INPUT_SE  ...]]   [<b>--input_pe</b>  INPUT_PE [INPUT_PE ...]]
       [<b>--merge_paired</b>] [<b>--forward</b> FORWARD] [<b>--reverse</b> REVERSE]  [<b>--read_type</b>  {q,qseq,f}]  [<b>--mlst_db</b>  MLST_DB]
       [<b>--mlst_delimiter</b>     MLST_DELIMITER]    [<b>--mlst_definitions</b>    MLST_DEFINITIONS]    [<b>--mlst_max_mismatch</b>
       MLST_MAX_MISMATCH]   [<b>--gene_db</b>   GENE_DB   [GENE_DB   ...]]   [<b>--no_gene_details</b>]   [<b>--gene_max_mismatch</b>
       GENE_MAX_MISMATCH]   [<b>--min_coverage</b>   MIN_COVERAGE]   [<b>--max_divergence</b>   MAX_DIVERGENCE]   [<b>--min_depth</b>
       MIN_DEPTH] [<b>--min_edge_depth</b> MIN_EDGE_DEPTH] [<b>--prob_err</b>  PROB_ERR]  [<b>--stop_after</b>  STOP_AFTER]  [<b>--other</b>
       OTHER]   [<b>--mapq</b>   MAPQ]   [<b>--baseq</b>   BASEQ]  [<b>--samtools_args</b>  SAMTOOLS_ARGS]  <b>--output</b>  OUTPUT  [<b>--log</b>]
       [<b>--save_scores</b>]      [<b>--report_new_consensus</b>]      [<b>--report_all_consensus</b>]       [<b>--use_existing_pileup</b>]
       [<b>--use_existing_scores</b>] [<b>--keep_interim_alignment</b>] [<b>--prev_output</b> PREV_OUTPUT [PREV_OUTPUT ...]]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       SRST2 - Short Read Sequence Typer (v2)

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>--version</b>
              show program's version number and exit

       <b>--input_se</b> INPUT_SE [INPUT_SE ...]
              Single end read file(s) for analysing (may be gzipped)

       <b>--input_pe</b> INPUT_PE [INPUT_PE ...]
              Paired end read files for analysing (may be gzipped)

       <b>--merge_paired</b>
              Switch  on  if all the input read sets belong to a single sample, and you want to merge their data
              to get a single result

       <b>--forward</b> FORWARD
              Designator for forward reads (only used if NOT  in  MiSeq  format  sample_S1_L001_R1_001.fastq.gz;
              otherwise default is _1, i.e. expect forward reads as sample_1.fastq.gz)

       <b>--reverse</b> REVERSE
              Designator  for  reverse  reads  (only used if NOT in MiSeq format sample_S1_L001_R2_001.fastq.gz;
              otherwise default is _2, i.e. expect forward reads as sample_2.fastq.gz

       <b>--read_type</b> {q,qseq,f}
              Read file type (for bowtie2; default is q=fastq; other options: qseq=solexa, f=fasta).

       <b>--mlst_db</b> MLST_DB
              Fasta file of MLST alleles (optional)

       <b>--mlst_delimiter</b> MLST_DELIMITER
              Character(s) separating gene name from allele number in MLST database (default "-", as in arcc-1)

       <b>--mlst_definitions</b> MLST_DEFINITIONS
              ST definitions for MLST scheme (required if mlst_db supplied and you want to calculate STs)

       <b>--mlst_max_mismatch</b> MLST_MAX_MISMATCH
              Maximum number of mismatches per read for MLST allele calling (default 10)

       <b>--gene_db</b> GENE_DB [GENE_DB ...]
              Fasta file/s for gene databases (optional)

       <b>--no_gene_details</b>
              Switch OFF verbose reporting of gene typing

       <b>--gene_max_mismatch</b> GENE_MAX_MISMATCH
              Maximum number of mismatches per read for gene detection and allele calling (default 10)

       <b>--min_coverage</b> MIN_COVERAGE
              Minimum %coverage cutoff for gene reporting (default 90)

       <b>--max_divergence</b> MAX_DIVERGENCE
              Maximum %divergence cutoff for gene reporting (default 10)

       <b>--min_depth</b> MIN_DEPTH
              Minimum mean depth to flag as dubious allele call (default 5)

       <b>--min_edge_depth</b> MIN_EDGE_DEPTH
              Minimum edge depth to flag as dubious allele call (default 2)

       <b>--prob_err</b> PROB_ERR
              Probability of sequencing error (default 0.01)

       <b>--stop_after</b> STOP_AFTER
              Stop mapping after this number of reads have been mapped (otherwise map all)

       <b>--other</b> OTHER
              Other arguments to pass to bowtie2 (must be escaped, e.g. "\--no-mixed".

       <b>--mapq</b> MAPQ
              Samtools <b>-q</b> parameter (default 1)

       <b>--baseq</b> BASEQ
              Samtools <b>-Q</b> parameter (default 20)

       <b>--samtools_args</b> SAMTOOLS_ARGS
              Other arguments to pass to samtools mpileup (must be escaped, e.g. "\-A").

       <b>--output</b> OUTPUT
              Prefix for srst2 output files

       <b>--log</b>  Switch ON logging to file (otherwise log to stdout)

       <b>--save_scores</b>
              Switch ON verbose reporting of all scores

       <b>--report_new_consensus</b>
              If a matching alleles is not found, report the consensus allele. Note, only  SNP  differences  are
              considered, not indels.

       <b>--report_all_consensus</b>
              Report the consensus allele for the most likely allele. Note, only SNP differences are considered,
              not indels.

       <b>--use_existing_pileup</b>
              Use existing pileups if available, otherwise they will be generated

       <b>--use_existing_scores</b>
              Use existing scores files if available, otherwise they will be generated

       <b>--keep_interim_alignment</b>
              Keep  interim  files  (sam  &amp;  unsorted  bam),  otherwise they will be deleted after sorted bam is
              created

       <b>--prev_output</b> PREV_OUTPUT [PREV_OUTPUT ...]
              SRST2 results files to compile (any new results from this run will also be incorporated)

</pre><h4><b>EXAMPLE</b></h4><pre>
       Assume you have a database downloaded by <a href="../man1/getmlst.1.html">getmlst</a>(1) by using

              getmlst --species "Escherichia coli#1"

       For SRST2, remember to check what separator is being used in this allele database

       Looks like --mlst_delimiter '-'

                &gt;adk-1  --&gt; --&gt;   ('adk', '-', '1')

       Suggested srst2 command for use with this MLST database:

              srst2 --output test --input_pe *.fastq.gz  --mlst_db  Escherichia_coli#1.fasta  --mlst_definitions
              ecoli.txt --mlst_delimiter '-'

       Note,  this is correctly guessing that we should use the default --mlst_delimiter '-' with this database.
       The log file will tell you exactly what files were downloaded.

       More verbose example usage is described in /usr/share/doc/srst2/example.txt.gz

</pre><h4><b>AUTHOR</b></h4><pre>
       Michael  Inouye   (<a href="mailto:minouye@unimelb.edu.au">minouye@unimelb.edu.au</a>),   Harriet   Dashnow   (<a href="mailto:h.dashnow@gmail.com">h.dashnow@gmail.com</a>),   Kathryn   Holt
       (<a href="mailto:kholt@unimelb.edu.au">kholt@unimelb.edu.au</a>), Bernie Pope (<a href="mailto:bjpope@unimelb.edu.au">bjpope@unimelb.edu.au</a>)

       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

srst2 0.1.6                                       December 2015                                         <u><a href="../man1/SRST2.1.html">SRST2</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>