<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>ssake - assembling millions of very short DNA sequences</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/ssake">ssake_4.0.1-2_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       ssake - assembling millions of very short DNA sequences

</pre><h4><b>SYNOPSIS</b></h4><pre>
       Progressive  assembly  of  millions  of  short DNA sequences by k-mer search through a prefix tree and 3'
       extension.

</pre><h4><b>OPTIONS</b></h4><pre>
       -f     Fasta file containing all the [paired (-p 1) / unpaired (-p 0)] reads (required) <b>paired</b> <b>reads</b> <b>must</b>
              <b>now</b> <b>be</b> <b>separated</b> <b>by</b> <b>":"</b>

       -s     Fasta file containing sequences to use as seeds exclusively (specify only if different  from  read
              set, optional)

       -m     Minimum  number  of  overlapping  bases  with  the  seed/contig during overhang consensus build up
              (default -m 16)

       -o     Minimum number of reads needed to call a base during an extension (default -o 3)

       -r     Minimum base ratio used to accept a overhang consensus base (default -r 0.7)

       -t     Trim up to -t base(s) on the contig  end  when  all  possibilities  have  been  exhausted  for  an
              extension (default -t 0)&gt;

       -p     Paired-end reads used? (-p 1=yes, -p 0=no, default -p 0)

       -v     Runs in verbose mode (-v 1=yes, -v 0=no, default -v 0, optional)

       -b     Base name for your output files (optional)

       ============ Options below only considered with -p 1 ============

       -d     Mean distance expected/observed between paired-end reads (default -d 200, optional)

       -e     Error (%) allowed on mean distance   e.g. -e 0.75  == distance +/- 75% (default -e 0.75, optional)

       -k     Minimum number of links (read pairs) to compute scaffold (default -k 2, optional)

       -a     Maximum  link  ratio  between  two  best  contig  pairs  *higher  values  lead  to  least accurate
              scaffolding* (default -a 0.70, optional)

       -z     Minimum contig size to track paired-end reads (default -z 50, optional)

       -g     Fasta file containing unpaired sequence reads (optional)

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       /usr/share/doc/ssake/SSAKE.readme between

</pre><h4><b>AUTHORS</b></h4><pre>
       This manual page was written by Andreas Tille &lt;<a href="mailto:tille@debian.org">tille@debian.org</a>&gt; for the <b>Debian</b> system (but may  be  used
       by others).  Permission is granted to copy, distribute and/or modify this document under the terms of the
       GNU General Public License, Version 2 any later version published by the Free Software Foundation.

       On Debian systems, the complete text of the GNU General Public License can be found in /usr/share/common-
       licenses/GPL.

                                                  January 2008                                          <u><a href="../man1/SSAKE.1.html">SSAKE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>