<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>tigr-glimmer — Fine start/stop positions of genes in genome sequence</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/tigr-glimmer">tigr-glimmer_3.02b-6_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       tigr-glimmer — Fine start/stop positions of genes in genome sequence

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>tigr-extract</b> [genome-file <u>options]</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Program extract takes a FASTA format sequence file and a file with a list of start/stop positions in that
       file  (e.g., as produced by the  long-orfs  program) and extracts and outputs the specified sequences.

       The first command-line argument is the name of the sequence file, which must be in FASTA format.

       The  second command-line argument is the name of the coordinate file.  It must contain a list of pairs of
       positions in the first file, one per line.  The format of each entry is:

       &lt;IDstring&gt;&gt;  &lt;start position&gt;  &lt;stop position&gt;

       This file should contain no other information, so if you're using the output of  glimmer  or  long-orfs ,
       you'll have to cut off header lines.

       The output of the program goes to the standard output and has one line for each line  in  the  coordinate
       file.   Each  line  contains  the   IDstring  , followed by white space, followed by the substring of the
       sequence file specified by the coordinate pair.  Specifically, the substring starts at the first position
       of the pair and ends at the second position (inclusive).  If  the  first  position  is  bigger  than  the
       second,  then  the  DNA  reverse  complement  of each position is generated.  Start/stop pairs that "wrap
       around" the end of the genome are allowed.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-skip</b>     makes the output omit the first 3 characters of each sequence, i.e., it skips  over  the  start
                 codon.  This was the behaviour of the previous version of the program.

       <b>-l</b>        makes  the  output  omit  an  sequences shorter than  n  characters.  n  includes the 3 skipped
                 characters if the  -skip  switch is one.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       tigr-glimmer3 (1), tigr-long-orfs (1), tigr-adjust (1),  tigr-anomaly   (1),  tigr-build-icm  (1),  tigr-
       check  (1),  tigr-codon-usage (1), tigr-compare-lists (1), tigr-extract (1), tigr-generate (1), tigr-get-
       len (1), tigr-get-putative (1),

       <a href="http://www.tigr.org/software/glimmer/">http://www.tigr.org/software/glimmer/</a>

       Please see the readme in /usr/share/doc/tigr-glimmer for a description on how to use Glimmer3.

</pre><h4><b>AUTHOR</b></h4><pre>
       This manual page was quickly copied from the glimmer web site by Steffen Moeller  <a href="mailto:moeller@debian.org">moeller@debian.org</a>  for
       the <b>Debian</b> system.

                                                                                                 <u><a href="../man1/TIGR-GLIMMER.1.html">TIGR-GLIMMER</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>