<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>tRNAscan-SE - detection of transfer RNA genes in genomic sequence</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/trnascan-se">trnascan-se_2.0.12+ds-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       tRNAscan-SE - detection of transfer RNA genes in genomic sequence

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>tRNAscan-SE</b> [<u>-options</u>] <u>&lt;FASTA</u> <u>file(s)&gt;</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       tRNAscan-SE 2.0.9 (July 2021) Copyright Â© 2020 Patricia Chan and Todd Lowe

              University of California Santa Cruz

       Freely distributed under the GNU General Public License (GPLv3)

              Scan a sequence file for tRNAs

       <b>--</b> default: use Infernal &amp; tRNA covariance models

              with eukaryotic sequences (use 'Search Mode Options' below to scan other types of sequences)

       Search Mode Options:

       <b>-E</b>     : search for eukaryotic tRNAs (default)

       <b>-B</b>     : search for bacterial tRNAs

       <b>-A</b>     : search for archaeal tRNAs

       <b>-M</b> &lt;model&gt;
              : search for mitochondrial tRNAs options: mammal, vert

       <b>-O</b>     : search for other organellar tRNAs

       <b>-G</b>     : use general tRNA model (cytoslic tRNAs from all 3 domains included)

       <b>--mt</b> &lt;model&gt;
              :  use  mito  tRNA  models  for  cytosolic/mito  detemination  (if  not  specified, only cytosolic
              isotype-specific model scan will be performed)

       <b>-I</b>     : search using Infernal default use with <b>-E</b>, <b>-B</b>, <b>-A</b>, or <b>-G</b>; optional for <b>-O</b>

       <b>--max</b>  : maximum sensitivity mode - search using Infernal without hmm filter (very slow)

       <b>-L</b>     : search using the legacy method (tRNAscan, EufindtRNA, and COVE) use with <b>-E</b>, <b>-B</b>, <b>-A</b> or <b>-G</b>

       <b>-C</b>  <b>--cove</b>
              : search using COVE analysis only (legacy, extremely slow) default use with <b>-O</b>

       <b>-H</b>  <b>--breakdown</b>
              : show breakdown of primary and secondary structure components to covariance model bit scores

       <b>-D</b>  <b>--nopseudo</b>
              : disable pseudogene checking

       Output options:

       <b>-o</b>  <b>--output</b> &lt;file&gt;
              : save final results in &lt;file&gt;

       <b>-f</b>  <b>--struct</b> &lt;file&gt;
              : save tRNA secondary structures to &lt;file&gt;

       <b>-s</b>  <b>--isospecific</b> &lt;file&gt;
              : save results using isotype-specific models in &lt;file&gt;

       <b>-m</b>  <b>--stats</b> &lt;file&gt;
              : save statistics summary for run in &lt;file&gt; (speed, # tRNAs found in each part of search, etc)

       <b>-b</b>  <b>--bed</b> &lt;file&gt;
              : save results in BED file format of &lt;file&gt;

       <b>-j</b>  <b>--gff</b> &lt;file&gt;
              : save results in GFF3 file format of &lt;file&gt;

       <b>-a</b>  <b>--fasta</b> &lt;file&gt;
              : save predicted tRNA sequences in FASTA file format of &lt;file&gt;

       <b>-l</b>  <b>--log</b> &lt;file&gt;
              : save log of program progress in &lt;file&gt;

       <b>--detail</b>
              : display prediction outputs in detailed view

       <b>--brief</b>
              : brief output format (no column headers)

       -? #   : '#' in place of &lt;file&gt; chooses default name for output files

       <b>-p</b>  <b>--prefix</b> &lt;label&gt;
              : use &lt;label&gt; prefix for all default output file names

       <b>-d</b>  <b>--progress</b>
              : display program progress messages

       <b>-q</b>  <b>--quiet</b>
              : quiet mode (credits &amp; run option selections suppressed)

       <b>-y</b>  <b>--hitsrc</b>
              : show origin of hits (Ts=tRNAscan 1.4, Eu=EufindtRNA, Bo=Both Ts and Eu, Inf=Infernal)

       Specify Alternate Cutoffs / Data Files:

       <b>-X</b>  <b>--score</b> &lt;score&gt;
              : set cutoff score (in bits) for reporting tRNAs (default=20)

       <b>-g</b>  <b>--gencode</b> &lt;file&gt;
              : use alternate genetic codes specified in &lt;file&gt; for determining tRNA type

       <b>-z</b>  <b>--pad</b> &lt;number&gt;
              : use &lt;number&gt; nucleotides padding when passing first-pass tRNA bounds predictions to CM  analysis
              (default=8)

       <b>--len</b> &lt;length&gt;
              : set max length of tRNA intron+variable region for legacy search mode (default=116bp)

       Misc Options:

       <b>-h</b>  <b>--help</b>
              : print this help message

       <b>-c</b>  <b>--conf</b> &lt;file&gt;
              : tRNAscan-SE configuration file (default: tRNAscan-SE.conf)

       <b>-Q</b>  <b>--forceow</b>
              : do not prompt user before overwriting pre-existing result files  (for batch processing)

       <b>--match</b> &lt;EXPR&gt;
              :  search  only  sequences  with  names matching &lt;EXPR&gt; string (&lt;EXPR&gt; may contain * or ? wildcard
              chars)

       <b>--search</b> &lt;EXPR&gt;
              : start search at sequence with name matching &lt;EXPR&gt; string and continue to end of input  sequence
              file(s)

       Special Advanced Options (for testing &amp; special purposes)

       <b>-U</b>     : search for tRNAs with alternate models defined in configuration file

       <b>-t</b>  <b>--tscan</b>
              : search using tRNAscan only (defaults to strict params)

       <b>--tmode</b> &lt;mode&gt;
              : explicitly set tRNAscan params, where &lt;mode&gt;=R or S (R=relaxed, S=strict tRNAscan v1.3 params)

       <b>-v</b>  <b>--verbose</b> &lt;file&gt;
              : save verbose tRNAscan 1.3 output to &lt;file&gt;

       <b>--nomerge</b>
              : Keep redundant tRNAscan 1.3 hits (don't filter out multiple predictions per tRNA identification)

       <b>-e</b>  <b>--eufind</b>
              :  search using Eukaryotic tRNA finder (EufindtRNA) only (defaults to Normal seach parameters when
              run alone, or to Relaxed search params when run with Cove)

       <b>--emode</b> &lt;mode&gt;
              : explicitly set EufindtRNA params, where &lt;mode&gt;=R, N, or S (relaxed, normal, or strict)

       <b>--iscore</b> &lt;score&gt;
              : manually set "intermediate" cutoff score for EufindtRNA

       <b>-r</b>  <b>--fsres</b> &lt;file&gt;
              : save first-pass scan results from EufindtRNA, tRNAscan, or Infernal hmm  in  &lt;file&gt;  in  tabular
              results format

       <b>--mid</b>  : fast scan mode - search using Infernal with mid-level strictness of hmm filter

       <b>-F</b>  <b>--falsepos</b> &lt;file&gt;
              :  save  first-pass  candidate  tRNAs  in  &lt;file&gt;  that  were  then found to be false positives by
              second-pass analysis

       <b>--missed</b> &lt;file&gt;
              : save all seqs that do NOT have at least one tRNA prediction in them (aka "missed" seqs)

       <b>--thread</b> &lt;number&gt;
              : number of threads used for running infernal (default is to use available threads)

</pre><h4><b>AUTHOR</b></h4><pre>
        This manpage was written by Nilesh Patra for the Debian distribution and
        can be used for any other usage of the program.

tRNAscan-SE 2.0.9+ds                                July 2021                                     <u><a href="../man1/TRNASCAN-SE.1.html">TRNASCAN-SE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>