<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>psa - biological sequence alignment file format</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/pftools">pftools_3.2.12-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       psa - biological sequence alignment file format

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>psa</b>  is  an output format used by the <b>pftools</b> package to describe alignments between biological sequences
       (DNA or protein) and <u>PROSITE</u> profiles.

       <b>psa</b> is apparented to the widely used biological sequence file format <u>fasta</u>.   Nevertheless  it  does  not
       only describe a biological sequence, it is especially used to include information of alignments between a
       motif  descriptor like a <u>PROSITE</u> profile and a given sequence. This information is included in the header
       and reflected in the structure of the sequence following the header line.

</pre><h4><b>SYNTAX</b></h4><pre>
       Each sequence in a <b>psa</b> alignment file or output must be preceded by a <u>fasta</u> header line.
       The general syntax of such a <u>fasta</u> header line is as follows:

              <b>&gt;</b><u>seq_id</u> [ <u>free_text</u> ]

       The header must start with a '<b>&gt;</b>' character which is directly followed by the <u>seq_id</u> field. This field  is
       interpreted  by  most programs as the sequence's <u>identifier</u> and/or <u>accession</u> number. It ends at the first
       encountered whitespace character.
       The <b>pftools</b> programs will use the <u>free_text</u> to add  information  about  the  match  score,  position  and
       description  of  the  sequence  or motif.  Please refer to the man page of the corresponding programs for
       further information about the output formats.
       The header can only extend over one line. The following lines up to  a  new  line  starting  with  a  '<b>&gt;</b>'
       character or the end of the file are interpreted as sequence data.

       The  line  following the header, starts the alignment data between a sequence and a <u>PROSITE</u> profile. This
       data can span over several lines of different length.
       The data is formed by <u>upper</u> or <u>lower</u>-case characters of  the  corresponding  sequence  alphabet  (DNA  or
       protein).  The gap characters '<b>.</b>' and '<b>-</b>' are also supported.
       The  alignment  always  has at least the length of the matching profile. Insertions or deletions detected
       during the motif/sequence alignment step will vary the length of the data reported, and can be identified
       using the following conventions:

              <u>upper-case</u> <u>character</u>
                     Any upper-case character of the sequence alphabet identifies a <u>match</u> position  between  the
                     sequence and the motif descriptor.

              <u>lower-case</u> <u>character</u>
                     A  lower-case  character  of the sequence alphabet is used to symbolize an <u>insertion</u> in the
                     sequence compared to the motif descriptor.

              <u>'-'</u> <u>(dash)</u> <u>character</u>
                     A '<b>-</b>' character in the output identifies  the  presence  of  a  <u>deletion</u>  in  the  sequence
                     compared to the motif descriptor.

</pre><h4><b>EXAMPLES</b></h4><pre>
       (1)    &gt;YD28_SCHPO 556 pos. 291 - 332 sp|Q10256|YD28_SCHPO

              PTDPGlnsKIAQLVSMGFDPLEAAQALDAANGDLDVAASFLL--
              This is an example of the output produced by <b><a href="../man1/pfsearch.1.html">pfsearch</a></b>(1) using the '-x' (i.e.  <b>psa</b> output) option.
              The  first  line starting with the '<b>&gt;</b>' character is the <u>fasta</u> header. It also contains information
              about the raw score of the alignment as well as its position in the input sequence.
              On the next line you find the alignment proper. Starting at position 6, we can find  an  <u>insertion</u>
              of  the  '<u>lns</u>' residues in the sequence compared to the motif. The last two positions of the motif
              are not present in the sequence (i.e. they are <u>deleted</u>).  This is indicated by the presence of two
              '<b>-</b>' (dash) characters at the end of the alignment.

</pre><h4><b>NOTES</b></h4><pre>
       (1)    The <b><a href="../man5/xpsa.5.html">xpsa</a></b>(5) format defines a more strict syntax of the  header  line,  allowing  the  exchange  of
              information between different sequence analysis tools. It uses <u>keyword</u>=<u>value</u> pairs to annotate the
              current  match  between  a  sequence  and a motif descriptor. This syntax can be easily parsed and
              extended, according to the needs of bioinformatic tools.

       (2)    The current implementation of the <b>pftools</b> package does not use the '<b>.</b>'  (dot) character in the <b>psa</b>
              output. Nevertheless <b><a href="../man1/psa2msa.1.html">psa2msa</a></b>(1) will read it and interpret it in the same manner as the '<b>-</b>' (dash)
              character.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man5/xpsa.5.html">xpsa</a></b>(5), <b><a href="../man1/pfsearch.1.html">pfsearch</a></b>(1), <b><a href="../man1/pfscan.1.html">pfscan</a></b>(1), <b><a href="../man1/pfw.1.html">pfw</a></b>(1), <b><a href="../man1/pfmake.1.html">pfmake</a></b>(1), <b><a href="../man1/psa2msa.1.html">psa2msa</a></b>(1)

</pre><h4><b>AUTHOR</b></h4><pre>
       This manual page was originally written by Volker Flegel.
       The <b>pftools</b> package was developed by Philipp Bucher.
       Any comments or suggestions should be addressed to &lt;<a href="mailto:pftools@sib.swiss">pftools@sib.swiss</a>&gt;.

pftools 2.3                                        April 2003                                             <u><a href="../man5/PSA.5.html">PSA</a></u>(5)
</pre>
 </div>
</div></section>
</div>
</body>
</html>