<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::DasI - DAS-style access to a feature database</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::DasI - DAS-style access to a feature database

</pre><h4><b>SYNOPSIS</b></h4><pre>
         # Open up a feature database somehow...
         $db = Bio::DasI-&gt;new(@args);

         @segments = $db-&gt;segment(-name  =&gt; 'NT_29921.4',
                                  -start =&gt; 1,
                                  -end   =&gt; 1000000);

         # segments are Bio::Das::SegmentI - compliant objects

         # fetch a list of features
         @features = $db-&gt;features(-type=&gt;['type1','type2','type3']);

         # invoke a callback over features
         $db-&gt;features(-type=&gt;['type1','type2','type3'],
                       -callback =&gt; sub { ... }
                       );

         $stream   = $db-&gt;get_seq_stream(-type=&gt;['type1','type2','type3']);
         while (my $feature = $stream-&gt;next_seq) {
            # each feature is a Bio::SeqFeatureI-compliant object
         }

         # get all feature types
         @types   = $db-&gt;types;

         # count types
         %types   = $db-&gt;types(-enumerate=&gt;1);

         @feature = $db-&gt;get_feature_by_name($class=&gt;$name);
         @feature = $db-&gt;get_feature_by_target($target_name);
         @feature = $db-&gt;get_feature_by_attribute($att1=&gt;$value1,$att2=&gt;$value2);
         $feature = $db-&gt;get_feature_by_id($id);

         $error = $db-&gt;error;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::DasI is a simplified alternative interface to sequence annotation databases used by the distributed
       annotation system (see Bio::Das). In this scheme, the genome is represented as a series of features, a
       subset of which are named.  Named features can be used as reference points for retrieving "segments" (see
       Bio::Das::SegmentI), and these can, in turn, be used as the basis for exploring the genome further.

       In addition to a name, each feature has a "class", which is essentially a namespace qualifier and a
       "type", which describes what type of feature it is.  Das uses the GO consortium's ontology of feature
       types, and so the type is actually an object of class Bio::Das::FeatureTypeI (see
       Bio::Das::FeatureTypeI). Bio::DasI provides methods for interrogating the database for the types it
       contains and the counts of each type.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Lincoln</b> <b>Stein</b></h4><pre>
       Email <a href="mailto:lstein@cshl.org">lstein@cshl.org</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : Bio::DasI-&gt;new(@args)
        Function: Create new Bio::DasI object
        Returns : a Bio::DasI object
        Args    : see below

       The <b>new()</b> method creates a new object.  The argument list is either a single argument consisting of a
       connection string, or the following list of -name=&gt;value arguments:

          Argument        Description
          --------        -----------

          -dsn            Connection string for database
          -adaptor        Name of an adaptor class to use when connecting
          -aggregator     Array ref containing list of aggregators
                            "semantic mappers" to apply to database
          -user           Authentication username
          -pass           Authentication password

       Implementors of DasI may add other arguments.

   <b>types</b>
        Title   : types
        Usage   : $db-&gt;types(@args)
        Function: return list of feature types in database
        Returns : a list of Bio::Das::FeatureTypeI objects
        Args    : see below

       This routine returns a list of feature types known to the database. It is also possible to find out how
       many times each feature occurs.

       Arguments are -option=&gt;value pairs as follows:

         -enumerate  if true, count the features

       The returned value will be a list of Bio::Das::FeatureTypeI objects (see Bio::Das::FeatureTypeI.

       If -enumerate is true, then the function returns a hash (not a hash reference) in which the keys are the
       stringified versions of Bio::Das::FeatureTypeI and the values are the number of times each feature
       appears in the database.

   <b>parse_types</b>
        Title   : parse_types
        Usage   : $db-&gt;parse_types(@args)
        Function: parses list of types
        Returns : an array ref containing ['method','source'] pairs
        Args    : a list of types in 'method:source' form
        Status  : internal

       This method takes an array of type names in the format "method:source" and returns an array reference of
       ['method','source'] pairs.  It will also accept a single argument consisting of an array reference with
       the list of type names.

   <b>segment</b>
        Title   : segment
        Usage   : $db-&gt;segment(@args);
        Function: create a segment object
        Returns : segment object(s)
        Args    : see below

       This method generates a Bio::Das::SegmentI object (see Bio::Das::SegmentI).  The segment can be used to
       find overlapping features and the raw sequence.

       When making the <b>segment()</b> call, you specify the ID of a sequence landmark (e.g. an accession number, a
       clone or contig), and a positional range relative to the landmark.  If no range is specified, then the
       entire region spanned by the landmark is used to generate the segment.

       Arguments are -option=&gt;value pairs as follows:

        -name         ID of the landmark sequence.

        -class        A namespace qualifier.  It is not necessary for the
                      database to honor namespace qualifiers, but if it
                      does, this is where the qualifier is indicated.

        -version      Version number of the landmark.  It is not necessary for
                      the database to honor versions, but if it does, this is
                      where the version is indicated.

        -start        Start of the segment relative to landmark.  Positions
                      follow standard 1-based sequence rules.  If not specified,
                      defaults to the beginning of the landmark.

        -end          End of the segment relative to the landmark.  If not specified,
                      defaults to the end of the landmark.

       The return value is a list of Bio::Das::SegmentI objects.  If the method is called in a scalar context
       and there are no more than one segments that satisfy the request, then it is allowed to return the
       segment.  Otherwise, the method must throw a "multiple segment exception".

   <b>features</b>
        Title   : features
        Usage   : $db-&gt;features(@args)
        Function: get all features, possibly filtered by type
        Returns : a list of Bio::SeqFeatureI objects
        Args    : see below
        Status  : public

       This routine will retrieve features in the database regardless of position.  It can be used to return all
       features, or a subset based on their type

       Arguments are -option=&gt;value pairs as follows:

         -types     List of feature types to return.  Argument is an array
                    of Bio::Das::FeatureTypeI objects or a set of strings
                    that can be converted into FeatureTypeI objects.

         -callback   A callback to invoke on each feature.  The subroutine
                     will be passed each Bio::SeqFeatureI object in turn.

         -attributes A hash reference containing attributes to match.

       The -attributes argument is a hashref containing one or more attributes to match against:

         -attributes =&gt; { Gene =&gt; 'abc-1',
                          Note =&gt; 'confirmed' }

       Attribute matching is simple exact string matching, and multiple attributes are ANDed together.  See
       Bio::DB::ConstraintsI for a more sophisticated take on this.

       If one provides a callback, it will be invoked on each feature in turn.  If the callback returns a false
       value, iteration will be interrupted.  When a callback is provided, the method returns undef.

   <b>get_feature_by_name</b>
        Title   : get_feature_by_name
        Usage   : $db-&gt;get_feature_by_name(-class=&gt;$class,-name=&gt;$name)
        Function: fetch features by their name
        Returns : a list of Bio::SeqFeatureI objects
        Args    : the class and name of the desired feature
        Status  : public

       This method can be used to fetch named feature(s) from the database.  The -class and -name arguments have
       the same meaning as in <b>segment()</b>, and the method also accepts the following short-cut forms:

         1) one argument: the argument is treated as the feature name
         2) two arguments: the arguments are treated as the class and name
            (note: this uses _rearrange() so the first argument must not
            begin with a hyphen or it will be interpreted as a named
            argument).

       This method may return zero, one, or several Bio::SeqFeatureI objects.  The implementor may allow the
       name to contain wildcards, in which case standard C-shell glob semantics are expected.

   <b>get_feature_by_target</b>
        Title   : get_feature_by_target
        Usage   : $db-&gt;get_feature_by_target($class =&gt; $name)
        Function: fetch features by their similarity target
        Returns : a list of Bio::SeqFeatureI objects
        Args    : the class and name of the desired feature
        Status  : public

       This method can be used to fetch a named feature from the database based on its similarity hit.  The
       arguments are the same as <b>get_feature_by_name()</b>.  If this is not implemented, the interface defaults to
       using <b>get_feature_by_name()</b>.

   <b>get_feature_by_id</b>
        Title   : get_feature_by_id
        Usage   : $db-&gt;get_feature_by_target($id)
        Function: fetch a feature by its ID
        Returns : a Bio::SeqFeatureI objects
        Args    : the ID of the feature
        Status  : public

       If the database provides unique feature IDs, this can be used to retrieve a single feature from the
       database.  If not overridden, this interface calls <b>get_feature_by_name()</b> and returns the first element.

   <b>get_feature_by_attribute</b>
        Title   : get_feature_by_attribute
        Usage   : $db-&gt;get_feature_by_attribute(attribute1=&gt;value1,attribute2=&gt;value2)
        Function: fetch features by combinations of attribute values
        Returns : a list of Bio::SeqFeatureI objects
        Args    : the class and name of the desired feature
        Status  : public

       This method can be used to fetch a set of features from the database.  Attributes are a list of
       name=&gt;value pairs.  They will be logically ANDed together.  If an attribute value is an array reference,
       the list of values in the array is treated as an alternative set of values to be ORed together.

   <b>search_notes</b>
        Title   : search_notes
        Usage   : $db-&gt;search_notes($search_term,$max_results)
        Function: full-text search on features, ENSEMBL-style
        Returns : an array of [$name,$description,$score]
        Args    : see below
        Status  : public

       This routine performs a full-text search on feature attributes (which attributes depend on
       implementation) and returns a list of [$name,$description,$score], where $name is the feature ID,
       $description is a human-readable description such as a locus line, and $score is the match strength.

       Since this is a decidedly non-standard thing to do (but the generic genome browser uses it), the default
       method returns an empty list.  You do not have to implement it.

   <b>get_seq_stream</b>
        Title   : get_seq_stream
        Usage   : $seqio = $db-&gt;get_seq_stream(@args)
        Function: Performs a query and returns an iterator over it
        Returns : a Bio::SeqIO stream capable of returning Bio::SeqFeatureI objects
        Args    : As in features()
        Status  : public

       This routine takes the same arguments as <b>features()</b>, but returns a Bio::SeqIO::Stream-compliant object.
       Use it like this:

         $stream = $db-&gt;get_seq_stream('exon');
         while (my $exon = $stream-&gt;next_seq) {
            print $exon,"\n";
         }

       NOTE: In the interface this method is aliased to <b>get_feature_stream()</b>, as the name is more descriptive.

   <b>refclass</b>
        Title   : refclass
        Usage   : $class = $db-&gt;refclass
        Function: returns the default class to use for segment() calls
        Returns : a string
        Args    : none
        Status  : public

       For data sources which use namespaces to distinguish reference sequence accessions, this returns the
       default namespace (or "class") to use.  This interface defines a default of "Accession".

perl v5.32.1                                       2021-08-15                                     <u>Bio::<a href="../man3pm/DasI.3pm.html">DasI</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>