<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>axe - axe Documentation</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/axe-demultiplexer">axe-demultiplexer_0.3.3+dfsg-4build2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       axe - axe Documentation

       Axe is a read de-multiplexer, useful in situations where sequence reads contain the indexes that uniquely
       distinguish  samples.  Axe  uses  a  rapid  and  accurate  algorithm  based  on hamming mismatch tries to
       competitively match the prefix of a sequencing read against a set of indexes. Axe supports  combinatorial
       indexing schemes.

       Contents:

</pre><h4><b>AXE</b> <b>TUTORIAL</b></h4><pre>
       In   this   tutorial,   we'll   use   Axe   to   demultiplex   some   paired-end,   combinatorially-index
       Genotyping-by-Sequencing  reads.  The  data  for  this   tutorial   is   available   from   figshare:   ‐
       <u>https://figshare.com/articles/axe-tutorial_tar/6143720</u> .

       Axe should be run as the initial step of any analysis: don't use sequence QC tools like AdapterRemoval or
       Trimmomatic before using axe, as indexes may be trimmed away, or pairing information removed.

   <b>Step</b> <b>0:</b> <b>Download</b> <b>the</b> <b>trial</b> <b>data</b>
       This will download the trial data, and extract it on the fly:

          curl -LS https://ndownloader.figshare.com/files/11094782 | tar xv

   <b>Step</b> <b>1:</b> <b>prepare</b> <b>a</b> <b>key</b> <b>file</b>
       The  key  file  associates index sequences with sample names. A key file can be prepared in a spreadsheet
       editor, like LibreOffice Calc, or Excel. The format is quite strict, and is described in  detail  in  the
       online usage documentation.

       Let's now inspect the keyfile I have provided for the tutorial.

          head axe-keyfile.tsv

   <b>Step</b> <b>2:</b> <b>Demultiplex</b> <b>with</b> <b>Axe</b>
       In  this  step, we will demultiplex our interleaved input file to per-sample interleaved output files. To
       see a full range of Axe's options, please run <b>axe-demux</b> <b>-h</b>, or inspect the online usage documentation.

       First, let's inspect the input.

          zcat axe-tutorial.fastq.gz | head -n 8

       Then, we need to ensure that axe has somewhere to put the demultiplexed reads.  Axe outputs one file  (or
       more, depending on pairing) per sample. Axe does so by appending the sample name to some prefix (as given
       by the <b>-I</b>, <b>-F</b>, and/or <b>-R</b> options). If this prefix is a directory, then sample fastq files will be created
       in that sub-directory, but the directory must exist. Let's make an output directory:

          mkdir -p output

       Now, let's demultiplex the reads!

          axe-demux -i axe-tutorial.fastq.gz -I output/ \
             -c -b axe-keyfile.tsv -t demux-stats.tsv -z 1

       The  command above demultiplexes reads from <b>axe-tutorial.fastq.gz</b> into separate files under <b>output</b>, based
       on the combinatorial (<b>-c</b>) sample-to-index-sequence mapping described in <b>axe-keyfile.tsv</b>, and saves a file
       of statistics as <b>demux-stats.tsv</b>. Note that we have enabled compression of  output  files  using  the  <b>-z</b>
       option, in case you don't have much disk space available. This will make Axe slightly slower.

</pre><h4><b>AXE</b> <b>USAGE</b></h4><pre>
       <b>NOTE:</b>
          For  arcane reasons, the name of the <b>axe</b> binary changed to <b>axe-demux</b> with version 0.3.0. Apologies for
          the inconvenience, this  was  required  to  make  <b>axe</b>  installable  in  Debian  and  its  derivatives.
          Command-line usage did not change.

       Axe  has  several  usage  modes.  The  primary distinction is between the two alternate indexing schemes,
       single and combinatorial indexing. Single index matching is used when only the first read contains  index
       sequences.   Combinatorial indexing is used when both reads in a read pair contain independent (typically
       different) index sequences.

       For concise reference, the command-line usage of <b>axe-demux</b> is reproduced below:

          USAGE:
          axe-demux [-mzc2pt] -b (-f [-r] | -i) (-F [-R] | -I)
          axe-demux -h
          axe-demux -v

          OPTIONS:
              -m, --mismatch Maximum hamming distance mismatch. [int, default 1]
              -z, --ziplevel Gzip compression level, or 0 for plain text [int, default 0]
              -c, --combinatorial Use combinatorial barcode matching. [flag, default OFF]
              -p, --permissive    Don't error on barcode mismatch confict, matching only
                                   exactly for conficting barcodes. [flag, default OFF]
              -2, --trim-r2  Trim barcode from R2 read as well as R1. [flag, default OFF]
              -b, --barcodes Barcode file. See --help for example. [file]
              -f, --fwd-in   Input forward read. [file]
              -F, --fwd-out  Output forward read prefix. [file]
              -r, --rev-in   Input reverse read. [file]
              -R, --rev-out  Output reverse read prefix. [file]
              -i, --ilfq-in  Input interleaved paired reads. [file]
              -I, --ilfq-out Output interleaved paired reads prefix. [file]
              -t, --table-file    Output a summary table of demultiplexing statistics to file. [file]
              -h, --help          Print this usage plus additional help.
              -V, --version  Print version string.
              -v, --verbose  Be more verbose. Additive, -vv is more vebose than -v.
              -q, --quiet         Be very quiet.

   <b>Inputs</b> <b>and</b> <b>Outputs</b>
       Regardless of read mode, three input and output schemes are supported:  single-end  reads,  paired  reads
       (separate  R1 and R2 files) and interleaved paired reads (one file, with R1 and R2 as consecutive reads).
       If single end reads are inputted, they  must  be  output  as  single  end  reads.  If  either  paired  or
       interleaved paired reads are read, they can be output as either paired reads or interleaved paired reads.
       This applies to both successfully de-multiplexed reads and reads that could not be de-multiplexed.

       The  <b>-z</b> flag can be used to specify that outputs should be compressed using gzip compression. The <b>-z</b> flag
       takes an integer argument between 0 (the default) and 9, where 0 indicates plain text output (<b>gzopen</b> mode
       "wT"), and 1-9 indicate that the respective compression level should be used, where 1 is fastest and 9 is
       most compact.

       The output flags should be prefixes that are used to generate the output file name based on  the  index's
       (or  index  pair's) ID. The names are generated as: <b>prefix</b> + <b>_</b> + <b>index</b> <b>ID</b> + <b>_</b> + <b>read</b> <b>number</b> + <b>.extension</b>.
       The output file for reads that could not be demultiplexed is <b>prefix</b> + <b>_</b> + <b>unknown</b> + <b>_</b>  +  <b>read</b>  <b>number</b>  +
       <b>.extension</b>.   The  read  number  is  omitted  unless the paired read file scheme is used, and is "il" for
       interleaved output. The extension is "fastq"; ".gz" is appended to the extension if the <b>-z</b> flag is used.

       <b>The</b> <b>corresponding</b> <b>CLI</b> <b>flags</b> <b>are:</b>

              • <b>-f</b> and <b>-F</b>: Single end or paired R1 file input and output respectively.

              • <b>-r</b> and <b>-R</b>: Paired R2 file input and output.

              • <b>-i</b> and <b>-I</b>: Interleaved paired input and output.

   <b>The</b> <b>index</b> <b>file</b>
       The index file is a tab-separated file with an optional header. It is mandatory, and is  always  supplied
       using  the <b>-b</b> command line flag. The exact format is dependent on indexing mode, and is described further
       in the sections below. If a header is present, the header line must start with either <u>Barcode</u>  or  <b>index</b>,
       or  it will be interpreted as a index line, leading to a parsing error. Any line starting with ';' or '#'
       is ignored, allowing comments to be added in line with indexes. Please ensure that the software  used  to
       produce  the  index uses ASCII encoding, and does not insert a Byte-order Mark (BoM) as many text editors
       can silently use Unicode-based encoding schemes. I recommend the use of <u>LibreOffice</u> <u>Calc</u> (part of a  free
       and open source office suite) to generate index tables; Microsoft Excel can also be used.

   <b>Mismatch</b> <b>level</b> <b>selection</b>
       Independent of index mode, the <b>-m</b> flag is used to select the maximum allowable hamming distance between a
       read's  prefix  and  a  index to be considered as a match. As "mutated" indexes must be unique, a hamming
       distance of one is the default as typically indexes are designed to differ by a hamming  distance  of  at
       least  two.  Optionally, (using the <b>-p</b> flag), axe will allow selective mismatch levels, where, if clashes
       are observed, the index will only be matched exactly. This allows one to process  datasets  with  indexes
       that don't have a sufficiently high distance between them.

   <b>Single</b> <b>index</b> <b>mode</b>
       Single  index mode is the default mode of operation. Barcodes are matched against read one (hereafter the
       forward read), and the index is trimmed from only the forward read, unless the <b>-2</b> command  line  flag  is
       given,  in  which  case  a prefix the same length as the matched index is also trimmed from the second or
       reverse read. Note that sequence of this second read is not checked before trimming.

       In single index mode, the index file has two columns: <b>Barcode</b> and <b>ID</b>.

   <b>Combinatorial</b> <b>index</b> <b>mode</b>
       Combinatorial index mode is activated by giving the <b>-c</b> flag on the command line. Forward read indexes are
       matched against the forward read, and reverse read indexes are matched  against  the  reverse  read.  The
       optimal  indexes  are  selected independently, and the index pair is selected from these two indexes. The
       respective  indexes are trimmed from both reads; the <b>-2</b> command line flag has no effect in  combinatorial
       index mode.

       In  combinatorial  index  mode,  the  index file has three columns: <b>Barcode1</b>, <b>Barcode2</b> and <b>ID</b>. Individual
       indexes can occur many times within the forward and reverse indexes,  but  index  pairs  must  be  unique
       combinations.

   <b>The</b> <b>Demultiplexing</b> <b>Statistics</b> <b>File</b>
       The  <b>-t</b>  option allows the output of per-sample read counts to a tab-separated file. The file will have a
       header describing its format, and includes a line for reads which could not be demultiplexed.

</pre><h4><b>AXE'S</b> <b>MATCHING</b> <b>ALGORITHM</b></h4><pre>
       Axe uses an algorithm based on longest-prefix-in-trie matching to match a variable length from the  start
       of each read against a set of 'mutated' indexes.

   <b>Hamming</b> <b>distance</b> <b>matching</b>
       While for most applications in high-throughput sequencing hamming distances are a frowned-upon metric, it
       is  typical  for HTS read indexes to be designed to tolerate a certain level of hamming mismatches. Given
       these sequences are short and typically occur at the 5' end of reads,  insertions  and  deletions  rarely
       need  be considered, and the increased rate of assignment of reads with many errors is offset by the risk
       of falsely assigning indexes to an incorrect sample. In any case, reads with  more  than  1-2  sequencing
       errors in their first several bases are likely to be poor quality, and will simply be filtered out during
       downstream quality control.

   <b>Hamming</b> <b>mismatch</b> <b>tries</b>
       Typically,  reads  are matched to a set of indexes by calculating the hamming distance between the index,
       and the first l bases of a read for a index of  length  l.  The  "correct"  index  is  then  selected  by
       recording  either  the  index  with  the lowest hamming distance to the read (competitive matching) or by
       simply accepting the first index with a hamming distance below a certain threshold.  These approaches are
       both very computationally  expensive,  and  can  have  lower  accuracy  than  the  algorithm  I  propose.
       Additionally,   implementations   of  these  methods  rarely  handle  indexes  of  differing  length  and
       combinatorial indexing well, if at all.

       Central to Axe's algorithm is the concept of hamming-mismatch tries. A trie is a  N-ary  tree  for  an  N
       letter  alphabet.  In  the  case  of  high-throughput  sequencing  reads,  we  have  the  alphabet  <b>AGCT</b>,
       corresponding to the four nucleotides of DNA, plus <b>N</b>, used to represent ambiguous base calls. Instead  of
       matching  each  index  to each read, we pre-calculate all allowable sequences at each mismatch level, and
       store these in level-wise tries. For  example, to match to a hamming  distance  of  2,  we  create  three
       tries: One containing all indexes, verbatim, and two tries where every sequence within a hamming distance
       of  1  and  2  of  each  index respectively. Hereafter, these tries are referred to  as the 0, 1 and 2-mm
       tries, for a hamming distance (mismatch) of 0, 1 and 2. Then, we find the longest prefix in each sequence
       read in the 0mm trie. If this prefix is not a valid leaf in the 0mm trie, we find the longest  prefix  in
       the 1mm trie, and so on for all tries in ascending order. If no prefix of the read is a complete sequence
       in any trie, the read is assigned to an "non-indexed" output file.

       This  algorithm  ensures  optimal  index matching in many ways, but is also extremely fast. In situations
       with indexes of differing length, we ensure that the <u>longest</u> acceptable index at a given hamming distance
       is chosen; assuming that sequence is random after the index, the probability of false  assignments  using
       this method is low. We also ensure that short perfect matches are preferred to longer inexact matches, as
       we  firstly  only  consider  indexes with no error, then 1 error, and so on. This ensures that reads with
       indexes that are followed by random sequence that happens to inexactly match a longer index  in  the  set
       are not falsely assigned to this longer index.

       The  speed  of  this algorithm is largely due to the constant time matching algorithm with respect to the
       number of indexes to match. The time taken to match each read is proportional instead to  the  length  of
       the  indexes, as for a index of length l, at most l + 1 trie level descents are required to find an entry
       in the trie. As this length is more-or-less constant and small, the overall complexity of axe's algorithm
       is O(n) for n reads, as opposed to O(nm) for n reads and m indexes as is typical for traditional matching
       algorithms

       • <u>Index</u>

</pre><h4><b>AUTHOR</b></h4><pre>
       Kevin Murray

</pre><h4><b>COPYRIGHT</b></h4><pre>
       2024, Kevin Murray

0.3.3                                             Jul 30, 2024                                            <u><a href="../man1/AXE.1.html">AXE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>