<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Phylo::PAML::Result - A PAML result set object</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-tools-phylo-paml-perl">libbio-tools-phylo-paml-perl_1.7.3-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Phylo::PAML::Result - A PAML result set object

</pre><h4><b>VERSION</b></h4><pre>
       version 1.7.3

</pre><h4><b>SYNOPSIS</b></h4><pre>
         # see Bio::Tools::Phylo::PAML for example usage
         use Bio::Tools::Phylo::PAML;
         my $parser = Bio::Tools::Phylo::PAML-&gt;new
           (-file =&gt; "./results/mlc", -dir =&gt; "./results/");

         # get the first/next result; a Bio::Tools::Phylo::PAML::Result object,
         # which isa Bio::SeqAnalysisResultI object.
         my $result = $parser-&gt;next_result();

         my @seqs         = $result-&gt;get_seqs;
         my %input_params = $result-&gt;get_input_parameters;
         my @basfreq      = $result-&gt;get_codon_pos_basefreq;
         my $MLmatrix     = $result-&gt;get_MLmatrix; # get MaxLikelihood Matrix
         my $NGmatrix     = $result-&gt;get_NGmatrix; # get Nei-Gojoburi Matrix

         # for AAML runs
         my $AAmatrix   = $result-&gt;get_AADistMatrix;
         my $AAMLmatrix   = $result-&gt;get_AAMLDistMatrix;

         # if -dir contains an rst file get list of
         # Bio::PrimarySeq ancestral state reconstructions of the sequences
         my @rsts          = $result-&gt;get_rst_seqs;

         # if you want to print the changes on the tree
         # this will print out the
         # anc_aa       =&gt; ANCESTRAL AMINO ACID
         # anc_prob     =&gt; ANCESTRAL AA PROBABILITY
         # derived_aa   =&gt; DERIVED AA
         # derived_prob =&gt; DERIVE AA PROBABILITY (where appropriate - NA for extant/tip taxas)
         # site         =&gt; which codon site this in the alignment
           @trees = $result-&gt;get_rst_trees;
           for my $t ( @trees ) {
           for my $node ( $t-&gt;get_nodes ) {
               next unless $node-&gt;ancestor; # skip root node
               my @changes = $node-&gt;get_tag_values('changes');
               my $chgstr = '';
               for my $c ( @changes ) {
               for my $k ( sort keys %$c ) {
                   $chgstr .= "$k =&gt; $c-&gt;{$k} ";
               }
               $chgstr .= "\n\t";
               }

               printf "node:%s n=%s s=%s\n\t%s\n",
               $node-&gt;id,
               $node-&gt;get_tag_values('n'),
               $node-&gt;get_tag_values('s'),
               $chgstr;
           }
           }

         # Persite probabilities
         my $persite = $result-&gt;get_rst_persite;
         # let's score site 1
         $site = $persite-&gt;[2];
         # so site 2, node 2 (extant node, node 2)
         print $site-&gt;[2]-&gt;{'codon'}, ' ',$site-&gt;[2]-&gt;{'aa'},"\n";
         # site 2, node 3
         print $site-&gt;[3]-&gt;{'codon'}, ' ',$site-&gt;[3]-&gt;{'aa'}, "\n";

         # ancestral node 9, codon, aa, marginal probabilities; Yang95 is listed as
         #  (eqn. 4 in Yang et al. 1995 Genetics 141:1641-1650) in PAML rst file.
         print $site-&gt;[9]-&gt;{'codon'}, ' ',$site-&gt;[9]-&gt;{'aa'}, ' ', $site-&gt;[9]-&gt;{'prob'}, ' ',
               $site-&gt;[9]-&gt;{'Yang95_aa'},' ', $site-&gt;[9]-&gt;{'Yang95_aa_prob'},"\n";

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This is a container object for PAML Results.

</pre><h4><b>METHODS</b></h4><pre>
   <b>new</b>
        Title   : new
        Usage   : my $obj = Bio::Tools::Phylo::PAML::Result-&gt;new(%data);
        Function: Builds a new Bio::Tools::Phylo::PAML::Result object
        Returns : Bio::Tools::Phylo::PAML::Result
        Args    : -trees     =&gt; array reference of Bio::Tree::TreeI objects
                  -MLmatrix  =&gt; ML matrix
                  -seqs      =&gt; array reference of Bio::PrimarySeqI objects
                  -codonpos  =&gt; array reference of codon positions
                  -codonfreq =&gt; array reference of codon frequencies
                  -version   =&gt; version string
                  -model     =&gt; model string
                  -patterns  =&gt; hashref with the fields '-patterns', '-ns', '-ls'
                  -stats     =&gt; array ref of misc stats   (optional)
                  -aafreq    =&gt; Hashref of AA frequencies (only for AAML)
                  -aadistmat =&gt; Bio::Matrix::PhylipDist   (only for AAML)
                  -aamldistmat =&gt; Bio::Matrix::PhylipDist   (only for pairwise AAML)
                  -ntfreq    =&gt; array ref of NT frequencies (only for BASEML)
                  -seqfile    =&gt; seqfile used
                  -kappa_mat =&gt; Bio::Matrix::PhylipDist of kappa values (only for BASEML)
                  -alpha_mat =&gt; Bio::Matrix::PhylipDist of alpha values (only for BASEML)
                  -NSSitesresult =&gt; arrayref of PAML::ModelResult
                  -input_params  =&gt; input params from .ctl file
                  -rst       =&gt; array reference of Bio::PrimarySeqI objects
                                of ancestral state reconstruction
                  -rst_persite=&gt; arrayref of persite data, this is a complicated set of AoH
                  -rst_trees  =&gt; rst trees with changes coded on the tree

       See Also: Bio::Tree::TreeI, Bio::PrimarySeqI, Bio::Matrix::PhylipDist, Bio::Tools::Phylo::PAML

   <b>next_tree</b>
        Title   : next_tree
        Usage   : my $tree = $factory-&gt;next_tree;
        Function: Get the next tree from the factory
        Returns : L&lt;Bio::Tree::TreeI&gt;
        Args    : none

   <b>get_trees</b>
        Title   : get_trees
        Usage   : my @trees = $result-&gt;get_trees;
        Function: Get all the parsed trees as an array
        Returns : Array of trees
        Args    : none

   <b>rewind_tree_iterator</b>
        Title   : rewind_tree_iterator
        Usage   : $result-&gt;rewind_tree_iterator()
        Function: Rewinds the tree iterator so that next_tree can be
                  called again from the beginning
        Returns : none
        Args    : none

   <b>add_tree</b>
        Title   : add_tree
        Usage   : $result-&gt;add_tree($tree);
        Function: Adds a tree
        Returns : integer which is the number of trees stored
        Args    : L&lt;Bio::Tree::TreeI&gt;

   <b>set_MLmatrix</b>
        Title   : set_MLmatrix
        Usage   : $result-&gt;set_MLmatrix($mat)
        Function: Set the ML Matrix
        Returns : none
        Args    : Arrayref to MLmatrix (must be arrayref to 2D matrix whic is
              lower triangle pairwise)

   <b>get_MLmatrix</b>
        Title   : get_MLmatrix
        Usage   : my $mat = $result-&gt;get_MLmatrix()
        Function: Get the ML matrix
        Returns : 2D Array reference
        Args    : none

   <b>set_NGmatrix</b>
        Title   : set_NGmatrix
        Usage   : $result-&gt;set_NGmatrix($mat)
        Function: Set the Nei &amp; Gojobori Matrix
        Returns : none
        Args    : Arrayref to NGmatrix (must be arrayref to 2D matrix whic is
              lower triangle pairwise)

   <b>get_NGmatrix</b>
        Title   : get_NGmatrix
        Usage   : my $mat = $result-&gt;get_NGmatrix()
        Function: Get the Nei &amp; Gojobori matrix
        Returns : 2D Array reference
        Args    : none

   <b>add_seq</b>
        Title   : add_seq
        Usage   : $obj-&gt;add_seq($seq)
        Function: Add a Bio::PrimarySeq to the Result
        Returns : none
        Args    : Bio::PrimarySeqI
       See also : L&lt;Bio::PrimarySeqI&gt;

   <b>reset_seqs</b>
        Title   : reset_seqs
        Usage   : $result-&gt;reset_seqs
        Function: Reset the OTU seqs stored
        Returns : none
        Args    : none

   <b>get_seqs</b>
        Title   : get_seqs
        Usage   : my @otus = $result-&gt;get_seqs
        Function: Get the seqs Bio::PrimarySeq (OTU = Operational Taxonomic Unit)
        Returns : Array of Bio::PrimarySeq
        Args    : None
       See also : L&lt;Bio::PrimarySeq&gt;

   <b>set_codon_pos_basefreq</b>
        Title   : set_codon_pos_basefreq
        Usage   : $result-&gt;set_codon_pos_basefreq(@freqs)
        Function: Set the codon position base frequencies
        Returns : none
        Args    : Array of length 3 where each slot has a hashref
                  keyed on DNA base

   <b>get_codon_pos_basefreq</b>
        Title   : get_codon_pos_basefreq
        Usage   : my @basepos = $result-&gt;get_codon_pos_basefreq;
        Function: Get the codon position base frequencies
        Returns : Array of length 3 (each codon position), each
                  slot is a hashref keyed on DNA bases, the values are
                  the frequency of the base at that position for all sequences
        Args    : none
        Note    : The array starts at 0 so position '1' is in position '0'
                  of the array

   <b>version</b>
        Title   : version
        Usage   : $obj-&gt;version($newval)
        Function: Get/Set version
        Returns : value of version
        Args    : newvalue (optional)

   <b>seqfile</b>
        Title   : seqfile
        Usage   : $obj-&gt;seqfile($newval)
        Function: Get/Set seqfile
        Returns : value of seqfile
        Args    : newvalue (optional)

   <b>model</b>
        Title   : model
        Usage   : $obj-&gt;model($newval)
        Function: Get/Set model
        Returns : value of model
        Args    : on set, new value (a scalar or undef, optional)

   <b>patterns</b>
        Title   : patterns
        Usage   : $obj-&gt;patterns($newval)
        Function: Get/Set Patterns hash
        Returns : Hashref of pattern data
        Args    : [optional] Hashref of patterns
                : The hashref is typically
                : { -patterns =&gt; \@arrayref
                :   -ns       =&gt; $ns
                :   -ls       =&gt; $ls
                : }

   <b>set_AAFreqs</b>
        Title   : set_AAFreqs
        Usage   : $result-&gt;set_AAFreqs(\%aafreqs);
        Function: Get/Set AA freqs
        Returns : none
        Args    : Hashref, keys are the sequence names, each points to a hashref
                  which in turn has keys which are the amino acids

   <b>get_AAFreqs</b>
        Title   : get_AAFreqs
        Usage   : my %all_aa_freqs = $result-&gt;get_AAFreqs()
                   OR
                  my %seq_aa_freqs = $result-&gt;get_AAFreqs($seqname)
        Function: Get the AA freqs, either for every sequence or just
                  for a specific sequence
                  The average aa freqs for the entire set are also available
                  for the sequence named 'Average'
        Returns : Hashref
        Args    : (optional) sequence name to retrieve aa freqs for

   <b>set_NTFreqs</b>
        Title   : set_NTFreqs
        Usage   : $result-&gt;set_NTFreqs(\%aafreqs);
        Function: Get/Set NT freqs
        Returns : none
        Args    : Hashref, keys are the sequence names, each points to a hashref
                  which in turn has keys which are the amino acids

   <b>get_NTFreqs</b>
        Title   : get_NTFreqs
        Usage   : my %all_nt_freqs = $result-&gt;get_NTFreqs()
                   OR
                  my %seq_nt_freqs = $result-&gt;get_NTFreqs($seqname)
        Function: Get the NT freqs, either for every sequence or just
                  for a specific sequence
                  The average nt freqs for the entire set are also available
                  for the sequence named 'Average'
        Returns : Hashref
        Args    : (optional) sequence name to retrieve nt freqs for

   <b>add_stat</b>
        Title   : add_stat
        Usage   : $result-&gt;add_stat($stat,$value);
        Function: Add some misc stat valuess (key/value pairs)
        Returns : none
        Args    : $stat  stat name
                  $value stat value

   <b>get_stat</b>
        Title   : get_stat
        Usage   : my $value = $result-&gt;get_stat($name);
        Function: Get the value for a stat of a given name
        Returns : scalar value
        Args    : name of the stat

   <b>get_stat_names</b>
        Title   : get_stat_names
        Usage   : my @names = $result-&gt;get_stat_names;
        Function: Get the stat names stored for the result
        Returns : array of names
        Args    : none

   <b>get_AADistMatrix</b>
        Title   : get_AADistMatrix
        Usage   : my $mat = $obj-&gt;get_AADistMatrix()
        Function: Get AADistance Matrix
        Returns : value of AADistMatrix (Bio::Matrix::PhylipDist)
        Args    : none

   <b>set_AADistMatrix</b>
        Title   : set_AADistMatrix
        Usage   : $obj-&gt;set_AADistMatrix($mat);
        Function: Set the AADistrance Matrix (Bio::Matrix::PhylipDist)
        Returns : none
        Args    : AADistrance Matrix (Bio::Matrix::PhylipDist)

   <b>get_AAMLDistMatrix</b>
        Title   : get_AAMLDistMatrix
        Usage   : my $mat = $obj-&gt;get_AAMLDistMatrix()
        Function: Get AAMLDistance Matrix
        Returns : value of AAMLDistMatrix (Bio::Matrix::PhylipDist)
        Args    : none

   <b>set_AAMLDistMatrix</b>
        Title   : set_AAMLDistMatrix
        Usage   : $obj-&gt;set_AAMLDistMatrix($mat);
        Function: Set the AA ML Distrance Matrix (Bio::Matrix::PhylipDist)
        Returns : none
        Args    : AAMLDistrance Matrix (Bio::Matrix::PhylipDist)

   <b>add_NSSite_result</b>
        Title   : add_NSSite_result
        Usage   : $result-&gt;add_NSSite_result($model)
        Function: Add a NSsite result (PAML::ModelResult)
        Returns : none
        Args    : Bio::Tools::Phylo::PAML::ModelResult

   <b>get_NSSite_results</b>
        Title   : get_NSSite_results
        Usage   : my @results = @{$self-&gt;get_NSSite_results};
        Function: Get the reference to the array of NSSite_results
        Returns : Array of PAML::ModelResult results
        Args    : none

   <b>set_CodonFreqs</b>
        Title   : set_CodonFreqs
        Usage   : $obj-&gt;set_CodonFreqs($newval)
        Function: Get/Set the Codon Frequence table
        Returns : value of set_CodonFreqs (a scalar)
        Args    : on set, new value (a scalar or undef, optional)

   <b>get_CodonFreqs</b>
        Title   : get_CodonFreqs
        Usage   : my @codon_freqs = $result-&gt;get_CodonFreqs()
        Function: Get the Codon freqs
        Returns : Array
        Args    : none

   <b>get_KappaMatrix</b>
        Title   : get_KappaMatrix
        Usage   : my $mat = $obj-&gt;get_KappaMatrix()
        Function: Get KappaDistance Matrix
        Returns : value of KappaMatrix (Bio::Matrix::PhylipDist)
        Args    : none

   <b>set_KappaMatrix</b>
        Title   : set_KappaMatrix
        Usage   : $obj-&gt;set_KappaMatrix($mat);
        Function: Set the KappaDistrance Matrix (Bio::Matrix::PhylipDist)
        Returns : none
        Args    : KappaDistrance Matrix (Bio::Matrix::PhylipDist)

   <b>get_AlphaMatrix</b>
        Title   : get_AlphaMatrix
        Usage   : my $mat = $obj-&gt;get_AlphaMatrix()
        Function: Get AlphaDistance Matrix
        Returns : value of AlphaMatrix (Bio::Matrix::PhylipDist)
        Args    : none

   <b>set_AlphaMatrix</b>
        Title   : set_AlphaMatrix
        Usage   : $obj-&gt;set_AlphaMatrix($mat);
        Function: Set the AlphaDistrance Matrix (Bio::Matrix::PhylipDist)
        Returns : none
        Args    : AlphaDistrance Matrix (Bio::Matrix::PhylipDist)

   <b>set_input_parameter</b>
        Title   : set_input_parameter
        Usage   : $obj-&gt;set_input_parameter($p,$vl);
        Function: Set an Input Parameter
        Returns : none
        Args    : $parameter and $value

   <b>get_input_parameters</b>
        Title   : get_input_parameters
        Usage   : $obj-&gt;get_input_parameters;
        Function: Get Input Parameters
        Returns : Hash of key/value pairs
        Args    : none

   <b>reset_input_parameters</b>
        Title   : reset_input_parameters
        Usage   : $obj-&gt;reset_input_parameters;
        Function: Reset the Input Parameters hash
        Returns : none
        Args    : none

   <b>add_rst_seq</b>
        Title   : add_rst_seq
        Usage   : $obj-&gt;add_rst_seq($seq)
        Function: Add a Bio::PrimarySeq to the RST Result
        Returns : none
        Args    : Bio::PrimarySeqI
       See also : L&lt;Bio::PrimarySeqI&gt;

   <b>reset_rst_seqs</b>
        Title   : reset_rst_seqs
        Usage   : $result-&gt;reset_rst_seqs
        Function: Reset the RST seqs stored
        Returns : none
        Args    : none

   <b>get_rst_seqs</b>
        Title   : get_rst_seqs
        Usage   : my @otus = $result-&gt;get_rst_seqs
        Function: Get the seqs Bio::PrimarySeq
        Returns : Array of Bio::PrimarySeqI objects
        Args    : None
       See also : L&lt;Bio::PrimarySeq&gt;

   <b>add_rst_tree</b>
        Title   : add_rst_tree
        Usage   : $obj-&gt;add_rst_tree($tree)
        Function: Add a Bio::Tree::TreeI to the RST Result
        Returns : none
        Args    : Bio::Tree::TreeI
       See also : L&lt;Bio::Tree::TreeI&gt;

   <b>reset_rst_trees</b>
        Title   : reset_rst_trees
        Usage   : $result-&gt;reset_rst_trees
        Function: Reset the RST trees stored
        Returns : none
        Args    : none

   <b>get_rst_trees</b>
        Title   : get_rst_trees
        Usage   : my @otus = $result-&gt;get_rst_trees
        Function: Get the trees Bio::Tree::TreeI
        Returns : Array of Bio::Tree::TreeI objects
        Args    : None
       See also : L&lt;Bio::Tree::TreeI&gt;

   <b>set_rst_persite</b>
        Title   : set_rst_persite
        Usage   : $obj-&gt;set_rst_persite($newval)
        Function: Get/Set the per-site RST values
        Returns : value of set_rst_persite (a scalar)
        Args    : on set, new value (a scalar or undef, optional)

   <b>get_rst_persite</b>
        Title   : get_rst_persite
        Usage   : my @rst_persite = @{$result-&gt;get_rst_persite()}
        Function: Get the per-site RST values
        Returns : Array
        Args    : none

</pre><h4><b>Reconstructed</b> <b>Ancestral</b> <b>State</b> <b>relevant</b> <b>options</b></h4><pre>
</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>              - General discussion
         <a href="http://bioperl.org/Support.html">http://bioperl.org/Support.html</a>    - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list: <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bio-tools-phylo-paml/issues

</pre><h4><b>AUTHORS</b></h4><pre>
       Jason Stajich &lt;<a href="mailto:jason@bioperl.org">jason@bioperl.org</a>&gt;

       Aaron Mackey &lt;<a href="mailto:amackey@virginia.edu">amackey@virginia.edu</a>&gt;

       Albert Vilella &lt;<a href="mailto:avilella@gmail.com">avilella@gmail.com</a>&gt;

</pre><h4><b>COPYRIGHT</b></h4><pre>
       This software is copyright (c) by Jason Stajich &lt;<a href="mailto:jason@bioperl.org">jason@bioperl.org</a>&gt;, and by Aaron Mackey
       &lt;<a href="mailto:amackey@virginia.edu">amackey@virginia.edu</a>&gt;.

       This software is available under the same terms as the perl 5 programming language system itself.

perl v5.32.0                                       2020-11-10               <u>Bio::Tools::Phylo::PAML::<a href="../man3pm/Result.3pm.html">Result</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>