<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>andi - estimates evolutionary distances</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/andi">andi_0.14-3build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       andi - estimates evolutionary distances

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>andi</b> [<u>OPTIONS...</u>] <u>FILES</u>...

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>andi</b>  estimates  the evolutionary distance between closely related genomes. For this <b>andi</b> reads the input
       sequences from <u>FASTA</u> files and computes the pairwise anchor distance. The idea behind this  is  explained
       in a paper by Haubold et al. (2015).

</pre><h4><b>OUTPUT</b></h4><pre>
       The  output  is  a  symmetrical distance matrix in <u>PHYLIP</u> format, with each entry representing divergence
       with a positive real number. A distance of zero means that two sequences  are  identical,  whereas  other
       values are estimates for the nucleotide substitution rate (Jukes-Cantor corrected). For technical reasons
       the  comparison  might  fail  and  no estimate can be computed. In such cases <u>nan</u> is printed. This either
       means that the input sequences were too short (&lt;200bp) or too diverse (K&gt;0.5)  for  our  method  to  work
       properly.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-b</b> <u>INT</u>, <b>--bootstrap</b>=<u>INT</u>
              Compute  multiple  distance matrices, with <u>n-1</u> bootstrapped from the first. See the paper Klötzl &amp;
              Haubold (2016) for a detailed explanation.

       <b>--file-of-filenames</b>=<u>FILE</u>
              Usually, <b>andi</b> is called with  the  filenames  as  commandline  arguments.  With  this  option  the
              filenames  may also be read from a file itself, with one name per line. Use a single dash (<b>'-'</b>) to
              read from stdin.

       <b>-j</b>, <b>--join</b>
              Use this mode if each of your <u>FASTA</u> files represents one assembly with numerous contigs. <b>andi</b> will
              then treat all of the contained sequences per file as a single genome. In this mode at  least  one
              filename  must  be  provided  via  command  line arguments. For the output the filename is used to
              identify each sequence.

       <b>-l</b>, <b>--low-memory</b>
              In multithreaded mode, <b>andi</b> requires memory linear to the amount of threads. The low  memory  mode
              changes this to a constant demand independent from the used number of threads. Unfortunately, this
              comes at a significant runtime cost.

       <b>-m</b> <u>MODEL</u>, <b>--model</b>=<u>MODEL</u>
              Set  the  nucleotide  evolution model to one of 'Raw', 'JC', 'Kimura', or 'LogDet'. By default the
              Jukes-Cantor correction is used.

       <b>-p</b> <u>FLOAT</u>
              Significance of an anchor; default: 0.025.

       <b>--progress</b>[=<u>WHEN</u>]
              Print a progress bar. <u>WHEN</u> can be 'auto' (default if omitted), 'always', or 'never'.

       <b>-t</b> <u>INT</u>, <b>--threads</b>=<u>INT</u>
              The number of threads to be used; by default, all available processors are used.
              Multithreading is only available if <b>andi</b> was compiled with OpenMP support.

       <b>--truncate-names</b>
              By default <b>andi</b> outputs the full names of sequences, optionally padded with spaces,  if  they  are
              shorter than ten characters. Names longer than ten characters may lead to problems with downstream
              tools. With this switch names will be truncated.

       <b>-v</b>, <b>--verbose</b>
              Prints  additional  information,  including the amount of found homology. Apply multiple times for
              extra verboseness.

       <b>-h</b>, <b>--help</b>
              Prints the synopsis and an explanation of available options.

       <b>--version</b>
              Outputs version information and acknowledgments.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright © 2014 - 2021 Fabian Klötzl License GPLv3+: GNU GPL version 3 or later.
       This is free software: you are free to change and redistribute it.  There is NO WARRANTY, to  the  extent
       permitted by law.  The full license text is available at &lt;<a href="http://gnu.org/licenses/gpl.html">http://gnu.org/licenses/gpl.html</a>&gt;.

</pre><h4><b>ACKNOWLEDGMENTS</b></h4><pre>
       1)  andi:  Haubold,  B.  Klötzl,  F.  and  Pfaffelhuber, P. (2015). andi: Fast and accurate estimation of
       evolutionary distances between closely related genomes, Bioinformatics 31.8.
       2) Algorithms: Ohlebusch, E. (2013). Bioinformatics Algorithms. Sequence Analysis, Genome Rearrangements,
       and Phylogenetic Reconstruction. pp 118f.
       3) SA construction: Mori, Y. (2005). libdivsufsort, unpublished.
       4) Bootstrapping: Klötzl, F. and Haubold, B. (2016). Support Values for Genome Phylogenies, Life 6.1.

</pre><h4><b>BUGS</b></h4><pre>
   <b>Reporting</b> <b>Bugs</b>
       Please report bugs to &lt;<a href="mailto:kloetzl@evolbio.mpg.de">kloetzl@evolbio.mpg.de</a>&gt; or at &lt;https://github.com/EvolBioInf/andi&gt;.

0.14                                               2020-01-09                                            <u><a href="../man1/ANDI.1.html">ANDI</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>