<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>simkaMinCore - comparative metagenomics method dedicated to NGS datasets</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/simka">simka_1.5.3-8build3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       simkaMinCore - comparative metagenomics method dedicated to NGS datasets

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>simkaMin</b> [<u>option</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       [Distance computation options]

       sketch : transform datasets in small sketches of k-mers and their abundance

       distance
              : compute Jaccard and Bray-Curtis distances between sketches

       [Distance matrix manipulation options]

       export : export distance matrices stored in binary format

       [Sketch options]

       append : merge multiple sketch files into a single one

       info   : list datasets contained in a sketch file

</pre><h4><b>AUTHOR</b></h4><pre>
       This  manpage was written by Shayan Doust for the Debian distribution and can be used for any other usage
       of the program.

simkaMinCore 1.5.1                               September 2019                                  <u><a href="../man1/SIMKAMINCORE.1.html">SIMKAMINCORE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>