<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>ttuner - interpretation of DNA Sanger sequencing data</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/tracetuner">tracetuner_3.0.6~beta+dfsg-3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       ttuner - interpretation of DNA Sanger sequencing data

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>-h</b>     (Help) This message

       <b>-Q</b>     (Quiet) Turn off status messages

       <b>-V</b>     (Verbose) Output more status messages

       <b>-nocall</b>
              Disable base recalling and just use the original called bases read from the input sample file

       <b>-recalln</b>
              Disable adding bases to or deleting from the original called sequence. Only recall Ns

       <b>-het</b>   Call call hetezygotes

       <b>-mix</b>   Call mixed bases

       <b>-min_ratio</b> &lt;ratio&gt;
              Override the default threshold ratio of heights of

       <b>-trim_window</b> &lt;size&gt;
              Set the trimming window size for averaging quality to the specified value. The default is 10.

       <b>-trim_threshold</b> &lt;qv&gt; Set the average quality value used in trimming to

       <b>-C</b> &lt;consensusfile&gt;
              Specify the name of the FASTA file which contains the consensus sequence

       <b>-edited_bases</b>
              Start base recalling from the ABI's edited bases

       <b>-t</b> &lt;table&gt;
              Use  specified  lookup  table.  This  option overrides the default (automatic choice of the lookup
              table) as well as the options <b>-3700pop5</b>, <b>-3700pop6</b>, <b>-3100</b>, and <b>-mbace</b>. To get  a  message  showing
              which table was used, specify <b>-V</b> option

       <b>-3730</b>  Use the built-in ABI 3730-pop7 lookup table

       <b>-3700pop5</b>
              Use the built-in ABI 3700-pop5 lookup table

       <b>-3700pop6</b>
              Use the built-in ABI 3700-pop6 lookup table

       <b>-3100</b>  Use the built-in ABI 3100-pop6 lookup table

       <b>-mbace</b> Use the built-in MegaBACE lookup table

       <b>-c</b>     Output SCF file(s), in the current directory

       <b>-cd</b> &lt;dir&gt;
              Output SCF file(s), in the specified directory

       <b>-cv3</b>   Use version 3 for output SCF file. Default is version 2.

       <b>-o</b> &lt;dir&gt;
              Output multi-fasta files of bases (tt.seq), their locations (tt.pos), quality values (tt.qual) and
              status reports (tt.status) to directory &lt;dir&gt;

       <b>-p</b>     Output .phd.1 file(s), in the current directory

       <b>-pd</b> &lt;dir&gt;
              Output .phd.1 file(s), in the specified directory

       <b>-q</b>     Output .qual file(s), in the current directory

       <b>-qa</b> &lt;file&gt;
              Append .qual file(s) to &lt;file&gt;

       <b>-qd</b> &lt;dir&gt;
              Output .qual file(s), in the specified directory

       <b>-s</b>     Output .seq file(s) in FASTA format, in the current directory

       <b>-sa</b> &lt;file&gt;
              Append .seq file(s) in FASTA format to &lt;file&gt;

       <b>-sd</b> &lt;dir&gt;
              Output .seq file(s) in FASTA format, in the specified directory

       <b>-qr</b> &lt;file&gt;
              Output a quality report that gives data for a histogram on the number of reads with quality values
              &gt;= 20, to the specified file

       <b>-if</b> &lt;file&gt;
              Read the input sample filenames from the specified file

       <b>-id</b> &lt;dir&gt;
              Read the input sample files from specified directory

       <b>-tab</b>   Call heterozygotes or mixed bases and output .tab file(s) in the  current directory

       <b>-tabd</b> &lt;dir&gt;
              Call mixed bases and output .tab file(s), in the specified directory

       <b>-tal</b>   Output .tal file(s),in the current directory

       <b>-tald</b> &lt;dir&gt;
              Output .tal file(s),in the specified directory

       <b>-hpr</b>   Output a homopolymer runs file in current directory

       <b>-hprd</b> &lt;dir&gt;
              Output a homopolymer runs file(s),in the specified directory

       <b>-d</b>     Output .poly file(s),in the current directory

       <b>-dd</b>    &lt;dir&gt;           Output .poly file(s),in the specified directory

       <b>-ipd</b> &lt;dir&gt;
              Input the original bases and peak locations from a .phd file in the specified directory.

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

ttuner 3.0.6~beta                                 October 2018                                         <u><a href="../man1/TTUNER.1.html">TTUNER</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>