<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>phylonium - rapidly estimate evolutionary distances</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/phylonium">phylonium_1.7-3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       phylonium - rapidly estimate evolutionary distances

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>phylonium</b> [<u>OPTIONS...</u>] <u>FILES</u>...

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>phylonium</b>  estimates  the evolutionary distance between closely related genomes. For this <b>phylonium</b> reads
       the input sequences from <u>FASTA</u> files and computes the pairwise evolutionary distance.  All  sequences  in
       one file are considered to be contigs of the same genome.

</pre><h4><b>OUTPUT</b></h4><pre>
       The  output  is  a  symmetrical distance matrix in <u>PHYLIP</u> format, with each entry representing divergence
       with a positive, real number. A distance of zero means that two sequences are  identical,  whereas  other
       values are estimates for the nucleotide substitution rate (Jukes-Cantor corrected). For technical reasons
       the  comparison  might  fail  and  no estimate can be computed. In such cases <u>nan</u> is printed. This either
       means that the input sequences were too short (&lt;200bp) or too diverse (K&gt;0.5)  for  our  method  to  work
       properly.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-2</b>, <b>--2pass</b>
              After  a  first run a 'central' sequence is chosen as reference for the second pass. This improves
              the accuracy when the first reference is an outgroup.

       <b>-b</b> <u>INT</u>, <b>--bootstrap</b>=<u>INT</u>
              Compute multiple distance matrices, with <u>n-1</u> bootstrapped from the first. See the paper  Klötzl  &amp;
              Haubold (2016) for a detailed explanation.

       <b>-r</b> <u>FILE</u>
              Define the reference sequence.

       <b>-t</b> <u>INT</u>, <b>--threads</b>=<u>INT</u>
              The number of threads to be used; by default, all available processors are used.
              Multithreading is only available if <b>phylonium</b> was compiled with OpenMP support.

       <b>-p</b> <u>FILE</u>
              Print the reference positions to the given path. Also sets complete deletion.

       <b>--progress</b>[=<u>WHEN</u>]
              Print a progress bar. <u>WHEN</u> can be 'auto' (default if omitted), 'always', or 'never'.

       <b>-v</b>, <b>--verbose</b>
              Prints  additional  information,  including the amount of found homology. Apply multiple times for
              extra verboseness.

       <b>-h</b>, <b>--help</b>
              Prints the synopsis and an explanation of available options.

       <b>--version</b>
              Outputs version information and acknowledgments.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright © 2017 - 2023 Fabian Klötzl License GPLv3+: GNU GPL version 3 or later.
       This is free software: you are free to change and redistribute it.  There is NO WARRANTY, to  the  extent
       permitted by law.  The full license text is available at &lt;<a href="http://gnu.org/licenses/gpl.html">http://gnu.org/licenses/gpl.html</a>&gt;.

</pre><h4><b>ACKNOWLEDGMENTS</b></h4><pre>
       1) Algorithms: Ohlebusch, E. (2013). Bioinformatics Algorithms. Sequence Analysis, Genome Rearrangements,
       and Phylogenetic Reconstruction. pp 118f.
       2) SA construction: Mori, Y. (2005). Short description of improved two-stage suffix sorting algorithm.
       3)  Bootstrapping: Klötzl, F. and Haubold, B. (2016). Support Values for Genome Phylogenies 4) Phylonium:
       Klötzl, F. and Haubold, B. (2020). Phylonium:  Fast  Estimation  of  Evolutionary  Distances  from  Large
       Samples of Similar Genomes

</pre><h4><b>BUGS</b></h4><pre>
   <b>Reporting</b> <b>Bugs</b>
       Please report bugs to &lt;<a href="mailto:fabian-phylonium@kloetzl.info">fabian-phylonium@kloetzl.info</a>&gt;.

1.7                                                2023-03-16                                       <u><a href="../man1/PHYLONIUM.1.html">PHYLONIUM</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>