<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>ClonalFrameML - Efficient Inference of Recombination in Whole Bacterial Genomes</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/clonalframeml">clonalframeml_1.13-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       ClonalFrameML - Efficient Inference of Recombination in Whole Bacterial Genomes

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>ClonalFrameML</b> newick_file fasta_file output_file <b>[OPTIONS]</b>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       ClonalFrameML  is  a  software  package  that  performs efficient inference of recombination in bacterial
       genomes. ClonalFrameML was created by Xavier Didelot and Daniel Wilson. ClonalFrameML can be  applied  to
       any  type  of aligned sequence data, but is especially aimed at analysis of whole genome sequences. It is
       able to compare hundreds of whole genomes in a matter of hours on a standard Desktop computer. There  are
       three  main outputs from a run of ClonalFrameML: a phylogeny with branch lengths corrected to account for
       recombination, an estimation of the key parameters of the recombination process, and  a  genomic  map  of
       where recombination took place for each branch of the phylogeny.

       ClonalFrameML  is  a  maximum  likelihood  implementation  of the Bayesian software ClonalFrame which was
       previously described by Didelot and Falush (2007). The recombination model underpinning ClonalFrameML  is
       exactly  the  same  as for ClonalFrame, but this new implementation is a lot faster, is able to deal with
       much larger genomic dataset, and does not suffer from MCMC convergence issues

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Options</b> <b>specifying</b> <b>the</b> <b>analysis</b> <b>type</b>
       <b>-em</b>    true (default) or false   Estimate parameters by a Baum-Welch expectation maximization algorithm.

       <b>-embranch</b>
              true or false (default)   Estimate parameters for each branch using the EM algorithm.

       <b>-rescale_no_recombination</b>
              true or false (default)   Rescale branch lengths for given sites with no recombination model.

       <b>-imputation_only</b>
              true or false (default)   Perform only ancestral state reconstruction and imputation.

   <b>Options</b> <b>affecting</b> <b>all</b> <b>analyses</b>
       <b>-kappa</b> value &gt; 0 (default 2.0)   Relative rate of transitions vs transversions in substitution model

       <b>-fasta_file_list</b>
              true or false (default)   Take fasta_file to be a white-space separated file list.

       <b>-xmfa_file</b>
              true or false (default)   Take fasta_file to be an XMFA file.

       <b>-ignore_user_sites</b>
              sites_file                Ignore sites listed in whitespace-separated sites_file.

       <b>-ignore_incomplete_sites</b>
              true or false (default)   Ignore sites with any ambiguous bases.

       <b>-use_incompatible_sites</b>
              true (default) or false   Use homoplasious and multiallelic sites to correct branch lengths.

       <b>-show_progress</b>
              true or false (default)   Output the progress of the maximum likelihood routines.

       <b>-chromosome_name</b>
              name, eg "chr"            Output importation status file in  BED  format  using  given  chromosome
              name.

       <b>-min_branch_length</b>
              value &gt; 0 (default 1e-7)  Minimum branch length.

       <b>-reconstruct_invariant_sites</b>
              true or false (default)   Reconstruct the ancestral states at invariant sites.

       <b>-label_uncorrected_tree</b>
              true or false (default)   Regurgitate the uncorrected Newick tree with internal nodes labelled.

   <b>Options</b> <b>affecting</b> <b>-em</b> and <b>-embranch</b>:
       <b>-prior_mean</b>
              df "0.1 0.001 0.1 0.0001" Prior mean for R/theta, 1/delta, nu and M.

       <b>-prior_sd</b>
              df "0.1 0.001 0.1 0.0001" Prior standard deviation for R/theta, 1/delta, nu and M.

       <b>-initial_values</b>
              default "0.1 0.001 0.05"  Initial values for R/theta, 1/delta and nu.

       <b>-guess_initial_m</b>
              true (default) or false   Initialize M and nu jointly in the EM algorithms.

       <b>-emsim</b> value &gt;= 0  (default 0)   Number of simulations to estimate uncertainty in the EM results.

       <b>-embranch_dispersion</b>
              value &gt; 0 (default .01)   Dispersion in parameters among branches in the <b>-embranch</b> model.

   <b>Options</b> <b>affecting</b> <b>-rescale_no_recombination</b>:
       <b>-brent_tolerance</b>
              tolerance (default .001)  Set the tolerance of the Brent routine for <b>-rescale_no_recombination</b>.

       <b>-powell_tolerance</b>
              tolerance (default .001)  Set the tolerance of the Powell routine for <b>-rescale_no_recombination</b>.

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

ClonalFrameML 1.11                               September 2017                                 <u><a href="../man1/CLONALFRAMEML.1.html">CLONALFRAMEML</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>