<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>sam - Sequence Alignment/Map file format</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libhts3t64">libhts3t64_1.21+ds-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       sam - Sequence Alignment/Map file format

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Sequence Alignment/Map (SAM) format is TAB-delimited. Apart from the header lines, which are started with
       the `@' symbol, each alignment line consists of:
        1    <b>QNAME</b>   Query template/pair NAME
        2    <b>FLAG</b>    bitwise FLAG
        3    <b>RNAME</b>   Reference sequence NAME
        4    <b>POS</b>     1-based leftmost POSition/coordinate of clipped sequence
        5    <b>MAPQ</b>    MAPping Quality (Phred-scaled)
        6    <b>CIGAR</b>   extended CIGAR string
        7    <b>MRNM</b>    Mate Reference sequence NaMe (`=' if same as RNAME)
        8    <b>MPOS</b>    1-based Mate POSition
        9    <b>TLEN</b>    inferred Template LENgth (insert size)
       10    <b>SEQ</b>     query SEQuence on the same strand as the reference
       11    <b>QUAL</b>    query QUALity (ASCII-33 gives the Phred base quality)
       12+   <b>OPT</b>     variable OPTional fields in the format TAG:VTYPE:VALUE

       Each bit in the FLAG field is defined as:
       0x0001   <b>p</b>   the read is paired in sequencing
       0x0002   <b>P</b>   the read is mapped in a proper pair
       0x0004   <b>u</b>   the query sequence itself is unmapped
       0x0008   <b>U</b>   the mate is unmapped
       0x0010   <b>r</b>   strand of the query (1 for reverse)
       0x0020   <b>R</b>   strand of the mate
       0x0040   <b>1</b>   the read is the first read in a pair
       0x0080   <b>2</b>   the read is the second read in a pair
       0x0100   <b>s</b>   the alignment is not primary
       0x0200   <b>f</b>   the read fails platform/vendor quality checks
       0x0400   <b>d</b>   the read is either a PCR or an optical duplicate
       0x0800   <b>S</b>   the alignment is supplementary

       where the second column gives the string representation of the FLAG field.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       https://github.com/samtools/hts-specs
              The full SAM/BAM file format specification

htslib                                             August 2013                                            <u><a href="../man5/sam.5.html">sam</a></u>(5)
</pre>
 </div>
</div></section>
</div>
</body>
</html>