<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>mash-screen - determine whether query sequences are within a larger pool of sequences</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/mash">mash_2.3+dfsg-6build2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       mash-screen - determine whether query sequences are within a larger pool of sequences

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>mash</b> <b>screen</b> [options] &lt;queries&gt;.msh &lt;pool&gt; [&lt;pool&gt;] ...

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Determine how well query sequences are contained within a pool of sequences. The queries must be
       formatted as a single Mash sketch file (.msh), created with the mash sketch command. The &lt;pool&gt; files can
       be contigs or reads, in fasta or fastq, gzipped or not, and "-" can be given for &lt;pool&gt; to read from
       standard input. The &lt;pool&gt; sequences are assumed to be nucleotides, and will be 6-frame translated if the
       &lt;queries&gt; are amino acids. The output fields are [identity, shared-hashes, median-multiplicity, p-value,
       query-ID, query-comment], where median-multiplicity is computed for shared hashes, based on the number of
       observations of those hashes within the pool.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>
           Help

       <b>-p</b> &lt;int&gt;
           Parallelism. This many threads will be spawned for processing.

       <b>-w</b>
           Winner-takes-all strategy for identity estimates. After counting hashes for each query, hashes that
           appear in multiple queries will be removed from all except the one with the best identity (ties
           broken by larger query), and other identities will be reduced. This removes output redundancy,
           providing a rough compositional outline.

   <b>Output</b>
       <b>-i</b> &lt;num&gt;
           Minimum identity to report. Inclusive unless set to zero, in which case only identities greater than
           zero (i.e. with at least one shared hash) will be reported. Set to -1 to output everything.

       <b>-v</b> &lt;num&gt;
           Maximum p-value to report.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <a href="../man1/mash.1.html">mash</a>(1)

                                                   2019-12-13                                     <u><a href="../man1/MASH-SCREEN.1.html">MASH-SCREEN</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>