<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>QTLtools fdensity - Functional density around molecular QTLs</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/qtltools">qtltools_1.3.1+dfsg-4build4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       QTLtools fdensity - Functional density around molecular QTLs

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>QTLtools</b> <b>fdensity</b> <b>--qtl</b> <u>significant_genes.bed</u> <b>--bed</b> <u>TFs.encode.bed.gz</u> <b>--out</b> <u>output.txt</u> <u>[OPTIONS]</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This  mode measures the density of functional annotations around the genomic positions of molecular QTLs.
       The method is detailed in &lt;https://www.nature.com/articles/ncomms15452&gt;.  In brief,  we  first  enumerate
       all  annotations  within a given window around the molecular QTLs (by default 1 Mb).  Then, we split this
       window into small bins (default 1 kb) and count the number of  functional  annotations  overlapping  each
       bin.   This  produces an annotation count per bin that can be then plotted to see if there is any peak or
       depletion around the molQTLs.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>--qtl</b> <u>in.bed</u>
              List of QTLs of interest in BED format.  REQUIRED.

       <b>--bed</b> <u>functional_annotation.bed.gz</u>
              Functional annotations in BED format.  REQUIRED.

       <b>--out</b> <u>output.txt</u>
              Output file.  REQUIRED.

       <b>--window</b> <u>integer</u>
              Window size around the molecular QTL position.  DEFAULT=1000000

       <b>--bin</b> <u>integer</u>
              Bin size in base pairs.  DEFAULT=1000

</pre><h4><b>INPUT</b> <b>FILES</b></h4><pre>
       <b>--qtl</b> <b>file</b>
        List of QTLs of interest.  An example:

        1    15210     15211     1_15211   ENSG00000227232.4   -
        1    735984    735985    1_735985  ENSG00000177757.1   +
        1    735984    735985    1_735985  ENSG00000240453.1   -
        1    739527    739528    1_739528  ENSG00000237491.4   +

        The column definitions are:
        1   The variant chromosome
        2   The variant's start position (<b>0-based</b>)
        3   The variant's end position (<b>1-based</b>)
        4   The variant ID
        5   The phenotype ID (<b>not</b> <b>used</b>)
        6   The phenotype's strand.

       <b>--bed</b> <b>file</b>
        List of annotations in BED format.  An example:

        1    254874    265487
        1    730984    735985
        1    734984    736585
        1    739527    748528

        The column definitions are:
        1   Chromosome
        2   Start position (<b>0-based</b>)
        3   End position (<b>1-based</b>)

</pre><h4><b>OUTPUT</b> <b>FILE</b></h4><pre>
       <b>--out</b> file
        Space separated results output file detailing the enrichment with the following columns:
        1   The start position of the bin
        2   The end position of the bin
        3   The number of associations in this bin

</pre><h4><b>EXAMPLE</b></h4><pre>
       1 You need to prepare a BED file containing the positions of the QTLs of interest.  To do so, extract all
         significant hits at a given FDR threshold (e.g. 5%), and then transform the significant QTL list into a
         BED file:

         Rscript ./script/qtltools_runFDR_cis.R results.genes.full.txt.gz 0.05 results.genes
         cat results.genes.significant.txt | awk '{ print $9, $10-1, $11, $8, $1, $5 }' | tr '  '  '\t'  |  sort
         -k1,1V -k2,2g &gt; results.genes.significant.bed

       2 Measure the density using the following command:

         QTLtools     fdensity     --qtl    results.genes.significant.bed    --bed    TFs.encode.bed.gz    --out
         density.TF.around.QTL.txt

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/QTLtools.1.html">QTLtools</a></u>(1)

       QTLtools website: &lt;https://qtltools.github.io/qtltools&gt;

</pre><h4><b>BUGS</b></h4><pre>
       Please submit bugs to &lt;https://github.com/qtltools/qtltools&gt;

</pre><h4><b>CITATION</b></h4><pre>
       Delaneau, O., Ongen, H., Brown, A. et al. A complete tool set for molecular QTL discovery  and  analysis.
       <u>Nat</u> <u>Commun</u> <b>8</b>, 15452 (2017).  &lt;https://doi.org/10.1038/ncomms15452&gt;

</pre><h4><b>AUTHORS</b></h4><pre>
       Olivier Delaneau (<a href="mailto:olivier.delaneau@gmail.com">olivier.delaneau@gmail.com</a>), Halit Ongen (<a href="mailto:halitongen@gmail.com">halitongen@gmail.com</a>)

QTLtools-v1.3                                      06 May 2020                              <u><a href="../man1/QTLtools-fdensity.1.html">QTLtools-fdensity</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>