<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>augur - pipeline components for real-time virus analysis</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/augur">augur_24.4.0-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       augur - pipeline components for real-time virus analysis

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: augur [-h]

              {parse,  index,  filter,  mask,  align, tree, refine, ancestral, translate, reconstruct-sequences,
              clades, traits, sequence-traits, lbi, distance, titers, frequencies,  export,  validate,  version,
              import, measurements} ...

       Augur: A bioinformatics toolkit for phylogenetic analysis.

   <b>positional</b> <b>arguments:</b>
              {parse,  index,  filter,  mask,  align, tree, refine, ancestral, translate, reconstruct-sequences,
              clades, traits, sequence-traits, lbi, distance, titers, frequencies,  export,  validate,  version,
              import, measurements}

       parse  Parse delimited fields from FASTA sequence names into a TSV and FASTA file.

       index  Count occurrence of bases in a set of sequences.

       filter Filter and subsample a sequence set.

       mask   Mask specified sites from a VCF or FASTA file.

       align  Align multiple sequences from FASTA.

       tree   Build a tree using a variety of methods.

       refine Refine an initial tree using sequence metadata.

       ancestral
              Infer ancestral sequences based on a tree.

       translate
              Translate gene regions from nucleotides to amino acids.

       reconstruct-sequences
              Reconstruct alignments from mutations inferred on the tree

       clades Assign clades to nodes in a tree based on amino-acid or nucleotide signatures.

       traits Infer ancestral traits based on a tree.

       sequence-traits
              Annotate sequences based on amino-acid or nucleotide signatures.

       lbi    Calculate LBI for a given tree and one or more sets of parameters.

       distance
              Calculate the distance between sequences across entire genes or at a predefined subset of sites.

       titers Annotate a tree with actual and inferred titer measurements.

       frequencies
              infer frequencies of mutations or clades

       export Export JSON files suitable for visualization with auspice.

       validate
              Validate files related to augur consumption or export.

       version
              Print the version of augur.

       import Import analyses into augur pipeline from other systems

       measurements
              Create JSON files suitable for visualization within the measurements panel of Auspice.

   <b>options:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

</pre><h4><b>AUTHOR</b></h4><pre>
        This  manpage  was  written by Ã‰tienne Mollier for the Debian distribution and can be used for any other
       usage of the program.

augur 16.0.1                                        June 2022                                           <u><a href="../man1/AUGUR.1.html">AUGUR</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>