<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>cnvkit - a command-line toolkit for copy number analysis.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/cnvkit">cnvkit_0.9.9-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       cnvkit - a command-line toolkit for copy number analysis.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: cnvkit [-h]

              {batch,target,access,antitarget,autobin,coverage,reference,fix,segment,call,diagram,scatter,heatmap,breaks,genemetrics,gainloss,sex,gender,metrics,segmetrics,bintest,import-picard,import-seg,import-theta,import-rna,export,version}
              ...

       CNVkit, a command-line toolkit for copy number analysis.

   <b>positional</b> <b>arguments:</b>
       {batch,target,access,antitarget,autobin,coverage,reference,fix,segment,call,diagram,scatter,heatmap,breaks,genemetrics,gainloss,sex,gender,metrics,segmetrics,bintest,import-picard,import-seg,import-theta,import-rna,export,version}
              Sub-commands (use with <b>-h</b> for more info)

       batch  Run the complete CNVkit pipeline on one or more BAM files.

       target Transform bait intervals into targets more suitable for CNVkit.

       access List the locations of accessible sequence regions in a FASTA file.

       antitarget
              Derive off-target ("antitarget") bins from target regions.

       autobin
              Quickly calculate reasonable bin sizes from BAM read counts.

       coverage
              Calculate coverage in the given regions from BAM read depths.

       reference
              Compile a coverage reference from the given files (normal samples).

       fix    Combine target and antitarget coverages and correct for biases. Adjust raw coverage data according
              to the given reference, correct potential biases and recenter.

       segment
              Infer copy number segments from the given coverage table.

       call   Call copy number variants from segmented log2 ratios.

       diagram
              Draw  copy  number  (log2 coverages, segments) on chromosomes as a diagram. If both the raw probes
              and segments are given, show them side-by-side on each chromosome  (segments  on  the  left  side,
              probes on the right side).

       scatter
              Plot probe log2 coverages and segmentation calls together.

       heatmap
              Plot copy number for multiple samples as a heatmap.

       breaks List the targeted genes in which a copy number breakpoint occurs.

       genemetrics
              Identify targeted genes with copy number gain or loss.

       sex    Guess samples' sex from the relative coverage of chromosomes X and Y.

       metrics
              Compute coverage deviations and other metrics for self-evaluation.

       segmetrics
              Compute segment-level metrics from bin-level log2 ratios.

       bintest
              Test for single-bin copy number alterations.

       import-picard
              Convert Picard CalculateHsMetrics tabular output to CNVkit .cnn files. The input file is generated
              by   the  PER_TARGET_COVERAGE  option  in  the  CalculateHsMetrics  script  in  Picard  tools.  If
              'antitarget' is in the input  filename,  the  generated  output  filename  will  have  the  suffix
              '.antitargetcoverage.cnn', otherwise '.targetcoverage.cnn'.

       import-seg
              Convert a SEG file to CNVkit .cns files.

       import-theta
              Convert  THetA  output  to  a  BED-like,  CNVkit-like  tabular format. Equivalently, use the THetA
              results file to convert CNVkit .cns segments to integer copy number calls.

       import-rna
              Convert a cohort of per-gene log2 ratios to CNVkit .cnr format.

       export Convert CNVkit output files to another format.

       version
              Display this program's version.

   <b>optional</b> <b>arguments:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       See the online manual for details: https://cnvkit.readthedocs.io

cnvkit 0.9.9                                      October 2021                                         <u><a href="../man1/CNVKIT.1.html">CNVKIT</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>