<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>GWAMA - Genome-Wide Association Meta Analysis</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/gwama">gwama_2.2.2+dfsg-6_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       GWAMA - Genome-Wide Association Meta Analysis

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>GWAMA</b> [<u>options</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       GWAMA  (Genome-Wide  Association  Meta  Analysis)  software  performs meta-analysis of the results of GWA
       studies of binary or quantitative phenotypes. Fixed- and random-effect meta-analyses  are  performed  for
       both  directly  genotyped  and  imputed SNPs using estimates of the allelic odds ratio and 95% confidence
       interval for binary traits, and estimates of the allelic effect size and standard error for  quantitative
       phenotypes.  GWAMA can be used for analysing the results of all different genetic models (multiplicative,
       additive, dominant, recessive). The software incorporates error trapping facilities  to  identify  strand
       alignment errors and allele flipping, and performs tests of heterogeneity of effects between studies.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-i</b>, <b>--filelist</b>=<u>filename</u>
              Specify studies' result files. Default = gwama.in

       <b>-o</b>, <b>--output</b>=<u>fileroot</u>
              Specify file root for output of analysis. Default = gwama (gwama.out, gwama.gc.out)

       <b>-r</b>, <b>--random</b>
              Use random effect correction. Default = disabled

       <b>-gc</b>, <b>--genomic_control</b>
              Use genomic control for adjusting studies' result files. Default = disabled

       <b>-gco</b>, <b>--genomic_control_output</b>
              Use  genomic  control  on  meta-analysis summary (i.e. results of meta- analysis are corrected for
              gc). Default = disabled

       <b>-qt</b>, <b>--quantitative</b>
              Select quantitative trait version (BETA and SE columns). Default = binary trait

       <b>-m</b>, <b>--map</b>=<u>filename</u>
              Select file name for marker map.

       <b>-t</b>, <b>--threshold</b>=<u>0-1</u>
              The p-value threshold for showing direction in summary effect directions. Default = 1

       <b>--no_alleles</b>
              No allele information has been given. Expecting always the same EA.

       <b>--indel_alleles</b>
              Allele labes might contain more than single letter. No strand checks.

       <b>--sex</b>  Run gender-differentiated and gender- heterogeneity analysis  (method  described  in  paper  Magi,
              Lindgren  &amp; Morris 2010). Gender info must be provided in filelist file. (second column after file
              names is either M or F).

       <b>--filter</b>
              Set  a  filtering  based  on  column  header.  It  needs  3  arguments:  column   name,   equation
              [&gt;,&lt;,&gt;=,&lt;=,==,!=], filter value. Multiple filters can be set.

       <b>--name_marker</b>
              Alternative header to marker name col.

       <b>--name_strand</b>
              Alternative header to strand column.

       <b>--name_n</b>
              Alternative header to sample size col.

       <b>--name_ea</b>
              Alternative header to effect allele column.

       <b>--name_nea</b>
              Alternative header to non-effect allele column.

       <b>--name_eaf</b>
              Alternative header to effect allele frequency column.

       <b>--name_beta</b>
              Alternative header to beta column.

       <b>--name_se</b>
              Alternative header to std. err. col.

       <b>--name_or</b>
              Alternative header to OR column.

       <b>--name_or_95l</b>
              Alternative header to OR 95L column.

       <b>--name_or_95u</b>
              Alternative header to OR 95U column.

       <b>-h</b>, <b>--help</b>
              Print this help.

       <b>-v</b>, <b>--version</b>
              Print GWAMA version number.  This manual page must be insufficient. Please use the help option for
              a quick reminder of the options of gwama or turn to its home page with the online documentation or
              the downloadable manual.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="http://www.geenivaramu.ee/en/tools/gwama">http://www.geenivaramu.ee/en/tools/gwama</a></b>

</pre><h4><b>AUTHOR</b></h4><pre>
       GWAMA was written by Reedik Magi and Andrew P Morris.

2.2.2                                              August 2016                                          <u><a href="../man1/GWAMA.1.html">GWAMA</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>