<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>mac3_filterdup - Model-based Analysis for ChIP-Sequencing</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/macs">macs_3.0.2-1build3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       mac3_filterdup - Model-based Analysis for ChIP-Sequencing

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: macs3 filterdup [-h] <b>-i</b> IFILE [IFILE ...]

       [-f {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE}]
              [-g  GSIZE]  [-s  TSIZE]  [-p  PVALUE]  [--keep-dup  KEEPDUPLICATES]  [--buffer-size  BUFFER_SIZE]
              [--verbose VERBOSE] [--outdir OUTDIR] [-o OUTPUTFILE] [-d]

   <b>options:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>-i</b>, <b>--ifile</b> IFILE [IFILE ...]
              Alignment file. If multiple files are given as '-t A  B  C',  then  they  will  all  be  read  and
              combined. REQUIRED.

       <b>-f</b>, <b>--format</b> {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE}
              Format of tag file, "AUTO", "BED" or "ELAND" or "ELANDMULTI" or "ELANDEXPORT" or "SAM" or "BAM" or
              "BOWTIE"  or  "BAMPE"  or "BEDPE". The default AUTO option will let 'macs3 filterdup' decide which
              format  the  file  is.  Please   check   the   definition   in   README   file   if   you   choose
              ELAND/ELANDMULTI/ELANDEXPORT/SAM/BAM/BOWTIE or BAMPE/BEDPE. DEFAULT: "AUTO"

       <b>-g</b>, <b>--gsize</b> GSIZE
              Effective   genome   size.   It   can  be  1.0e+9  or  1000000000,  or  shortcuts:'hs'  for  human
              (2,913,022,398), 'mm' for mouse (2,652,783,500), 'ce' for C. elegans (100,286,401)  and  'dm'  for
              fruitfly  (142,573,017),  Default:hs. The effective genome size numbers for the above four species
              are  collected  from   Deeptools   https:   //deeptools.readthedocs.io/en/develop/content/feature/
              effectiveGenomeSize.html Please refer to deeptools to define the best genome size you plan to use.

       <b>-s</b>, <b>--tsize</b> TSIZE
              Tag size. This will override the auto detected tag size. DEFAULT: Not set

       <b>-p</b>, <b>--pvalue</b> PVALUE
              Pvalue cutoff for binomial distribution test.  DEFAULT:1e-5

       <b>--keep-dup</b> KEEPDUPLICATES
              It controls the 'macs3 filterdup' behavior towards duplicate tags/pairs at the exact same location
              <b>--</b>  the same coordination and the same strand. The 'auto' option makes 'macs3 filterdup' calculate
              the maximum tags at the exact same location based on binomal distribution using given <b>-p</b> as pvalue
              cutoff; and the 'all' option keeps every tags (useful if you only want to convert formats). If  an
              integer  is  given,  at  most  this number of tags will be kept at the same location.  Note, MACS3
              callpeak function uses KEEPDUPLICATES=1 as default. Note, if you've used  samtools  or  picard  to
              flag  reads  as 'PCR/Optical duplicate' in bit 1024, MACS3 will still read them although the reads
              may be decided by MACS3 as duplicate later. Default: auto

       <b>--buffer-size</b> BUFFER_SIZE
              Buffer size for incrementally increasing internal array size to store reads alignment information.
              In most cases, you don't have to change this parameter.  However, if there  are  large  number  of
              chromosomes/contigs/scaffolds in your alignment, it's recommended to specify a smaller buffer size
              in  order to decrease memory usage (but it will take longer time to read alignment files). Minimum
              memory requested for reading an alignment file is about # of CHROMOSOME * BUFFER_SIZE *  8  Bytes.
              DEFAULT: 100000

       <b>--verbose</b> VERBOSE
              Set  verbose  level.  0:  only  show critical message, 1: show additional warning message, 2: show
              process information, 3: show debug messages. If you want to know where are  the  duplicate  reads,
              use 3. DEFAULT:2

       <b>--outdir</b> OUTDIR
              If  specified  all  output  files  will be written to that directory. Default: the current working
              directory

       <b>-o</b>, <b>--ofile</b> OUTPUTFILE
              Output BED file name. If not specified, will write to standard output. Note, if the  input  format
              is BAMPE or BEDPE, the output will be in BEDPE format. DEFAULT: stdout

       <b>-d</b>, <b>--dry-run</b>
              When  set,  filterdup  will only output numbers instead of writing output files, including maximum
              allowable duplicates, total number  of  reads  before  filtering,  total  number  of  reads  after
              filtering, and redundant rate. Default: not set

macs3 filterdup 3.0.2                              March 2025                                  <u><a href="../man1/MAC3_FILTERDUP.1.html">MAC3_FILTERDUP</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>