<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>mac3_callpeak - Model-based Analysis for ChIP-Sequencing</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/macs">macs_3.0.2-1build3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       mac3_callpeak - Model-based Analysis for ChIP-Sequencing

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: macs3 callpeak [-h] <b>-t</b> TFILE [TFILE ...] [-c [CFILE ...]]

       [-f {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE}]
              [-g  GSIZE]  [-s  TSIZE]  [--keep-dup  KEEPDUPLICATES] [--outdir OUTDIR] [-n NAME] [-B] [--verbose
              VERBOSE] [--trackline] [--SPMR] [--nomodel] [--shift SHIFT] [--extsize EXTSIZE] [--bw BW] [--d-min
              D_MIN] [-m MFOLD MFOLD]  [--fix-bimodal]  [-q  QVALUE  |  <b>-p</b>  PVALUE]  [--scale-to  {large,small}]
              [--down-sample]  [--seed  SEED]  [--tempdir  TEMPDIR] [--nolambda] [--slocal SMALLLOCAL] [--llocal
              LARGELOCAL]  [--max-gap  MAXGAP]  [--min-length  MINLEN]  [--broad]  [--broad-cutoff  BROADCUTOFF]
              [--cutoff-analysis]   [--call-summits]   [--fe-cutoff   FECUTOFF]   [--to-large]  [--ratio  RATIO]
              [--buffer-size BUFFER_SIZE]

   <b>options:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

   <b>Input</b> <b>files</b> <b>arguments:</b>
       <b>-t</b>, <b>--treatment</b> TFILE [TFILE ...]
              ChIP-seq treatment file. If multiple files are given as '-t A B C', then they will all be read and
              pooled together. REQUIRED.

       <b>-c</b>, <b>--control</b> [CFILE ...]
              Control file. If multiple files are given as '-c A B C', they will be pooled to estimate  ChIP-seq
              background noise.

       <b>-f</b>, <b>--format</b> {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE}
              Format of tag file, "AUTO", "BED" or "ELAND" or "ELANDMULTI" or "ELANDEXPORT" or "SAM" or "BAM" or
              "BOWTIE"  or "BAMPE" or "BEDPE". The default AUTO option will let MACS decide which format (except
              for BAMPE and BEDPE which should be implicitly set) the file is. Please check  the  definition  in
              README.  Please  note  that  if  the  format is set as BAMPE or BEDPE, MACS3 will call its special
              Paired-end mode to call peaks by piling up the actual ChIPed fragments  defined  by  both  aligned
              ends, instead of predicting the fragment size first and extending reads. Also please note that the
              BEDPE  only  contains  three columns, and is NOT the same BEDPE format used by BEDTOOLS.  DEFAULT:
              "AUTO"

       <b>-g</b>, <b>--gsize</b> GSIZE
              Effective  genome  size.  It  can  be  1.0e+9  or  1000000000,   or   shortcuts:'hs'   for   human
              (2,913,022,398),  'mm'  for  mouse (2,652,783,500), 'ce' for C. elegans (100,286,401) and 'dm' for
              fruitfly (142,573,017), Default:hs. The effective genome size numbers for the above  four  species
              are   collected   from   Deeptools  https:  //deeptools.readthedocs.io/en/develop/content/feature/
              effectiveGenomeSize.html Please refer to deeptools to define the best genome size you plan to use.

       <b>-s</b>, <b>--tsize</b> TSIZE
              Tag size/read length. This will override the auto detected tag size. DEFAULT: Not set

       <b>--keep-dup</b> KEEPDUPLICATES
              It controls the  behavior  towards  duplicate  tags  at  the  exact  same  location  <b>--</b>  the  same
              coordination  and  the same strand. The 'auto' option makes MACS calculate the maximum tags at the
              exact same location based on binomal distribution using 1e-5  as  pvalue  cutoff;  and  the  'all'
              option  keeps  every tags. If an integer is given, at most this number of tags will be kept at the
              same location. Note, if you've used samtools or picard to flag reads as 'PCR/Optical duplicate' in
              bit 1024, MACS3 will still read them although the reads may  be  decided  by  MACS3  as  duplicate
              later.  If  you plan to rely on samtools/picard/any other tool to filter duplicates, please remove
              those duplicate reads and save a new alignment file then ask MACS3  to  keep  all  by  '--keep-dup
              all'. The default is to keep one tag at the same location. Default: 1

   <b>Output</b> <b>arguments:</b>
       <b>--outdir</b> OUTDIR
              If  specified  all  output  files  will be written to that directory. Default: the current working
              directory

       <b>-n</b>, <b>--name</b> NAME
              Experiment name, which will be used to generate output file names. DEFAULT: "NA"

       <b>-B</b>, <b>--bdg</b>
              Whether or not to save extended fragment pileup, and local lambda tracks (two files) at  every  bp
              into a bedGraph file. DEFAULT: False

       <b>--verbose</b> VERBOSE
              Set  verbose  level  of runtime message. 0: only show critical message, 1: show additional warning
              message, 2: show process information, 3: show debug messages.  DEFAULT:2

       <b>--trackline</b>
              Instruct MACS to include trackline  in  the  header  of  output  files,  including  the  bedGraph,
              narrowPeak,  gappedPeak,  BED format files. To include this trackline is necessary while uploading
              them to the UCSC genome browser. You can also  mannually  add  these  trackline  to  corresponding
              output  files. For example, in order to upload narrowPeak file to UCSC browser, add this to as the
              first line <b>--</b> `track type=narrowPeak name="my_peaks"  description="my  peaks"`.  Default:  Not  to
              include trackline.

       <b>--SPMR</b> If  True, MACS will SAVE signal per million reads for fragment pileup profiles. It won't interfere
              with computing pvalue/qvalue during peak calling, since  internally  MACS3  keeps  using  the  raw
              pileup   and   scaling  factors  between  larger  and  smaller  dataset  to  calculate  statistics
              measurements. If you plan to use the signal output in bedGraph to  call  peaks  using  bdgcmp  and
              bdgpeakcall,  you  shouldn't  use  this  option  because  you  will end up with different results.
              However, this option is recommended for displaying normalized pileup tracks across many  datasets.
              Require <b>-B</b> to be set. Default: False

   <b>Shifting</b> <b>model</b> <b>arguments:</b>
       <b>--nomodel</b>
              Whether  or  not  to  build  the shifting model. If True, MACS will not build model. by default it
              means shifting size = 100, try to set extsize to change it.  It's highly  recommended  that  while
              you  have  many datasets to process and you plan to compare different conditions, aka differential
              calling, use both 'nomodel' and 'extsize' to make signal files from different datasets comparable.
              DEFAULT: False

       <b>--shift</b> SHIFT
              (NOT the legacy <b>--shiftsize</b> option!) The arbitrary shift in bp. Use discretion  while  setting  it
              other  than default value. When NOMODEL is set, MACS will use this value to move cutting ends (5')
              towards 5'-&gt;3' direction then apply EXTSIZE to extend  them  to  fragments.  When  this  value  is
              negative,  ends  will  be  moved  toward 3'-&gt;5' direction. Recommended to keep it as default 0 for
              ChIP-Seq datasets, or <b>-1</b> * half of EXTSIZE together with EXTSIZE  option  for  detecting  enriched
              cutting  loci  such  as  certain  DNAseI-Seq  datasets. Note, you can't set values other than 0 if
              format is BAMPE or BEDPE for paired-end data. DEFAULT: 0.

       <b>--extsize</b> EXTSIZE
              The arbitrary extension size in bp. When nomodel is true, MACS will use  this  value  as  fragment
              size  to  extend  each  read  towards 3' end, then pile them up.  It's exactly twice the number of
              obsolete SHIFTSIZE.  In previous language, each read  is  moved  5'-&gt;3'  direction  to  middle  of
              fragment by 1/2 d, then extended to both direction with 1/2 d. This is equivalent to say each read
              is extended towards 5'-&gt;3' into a d size fragment. DEFAULT: 200. EXTSIZE and SHIFT can be combined
              when necessary. Check SHIFT option.

       <b>--bw</b> BW
              Band  width  for  picking regions to compute fragment size. This value is only used while building
              the shifting model. Tweaking this is not recommended.  DEFAULT: 300

       <b>--d-min</b> D_MIN
              Minimum fragment size in basepair. Any predicted fragment size less than this  will  be  excluded.
              DEFAULT: 20

       <b>-m</b>, <b>--mfold</b> MFOLD MFOLD
              Select  the  regions  within  MFOLD range of highconfidence enrichment ratio against background to
              build model. Fold-enrichment in regions must be lower than upper limit, and higher than the  lower
              limit. Use as "-m 10 30". This setting is only used while building the shifting model. Tweaking it
              is not recommended. DEFAULT:5 50

       <b>--fix-bimodal</b>
              Whether  turn  on  the auto pair model process. If set, when MACS failed to build paired model, it
              will use the nomodel settings, the <b>--exsize</b> parameter to extend each tags  towards  3'  direction.
              Not to use this automate fixation is a default behavior now. DEFAULT: False

   <b>Peak</b> <b>calling</b> <b>arguments:</b>
       <b>-q</b>, <b>--qvalue</b> QVALUE
              Minimum  FDR  (q-value)  cutoff  for  peak  detection.   DEFAULT:  0.05.  <b>-q</b>,  and <b>-p</b> are mutually
              exclusive.

       <b>-p</b>, <b>--pvalue</b> PVALUE
              Pvalue cutoff for peak detection. DEFAULT: not set.  <b>-q</b>, and <b>-p</b> are mutually exclusive. If  pvalue
              cutoff is set, qvalue will not be calculated and reported as <b>-1</b> in the final .xls file.

       <b>--scale-to</b> {large,small}
              When set to 'small', scale the larger sample up to the smaller sample. When set to 'larger', scale
              the smaller sample up to the bigger sample. By default, scale to 'small'. This option replaces the
              obsolete  '  <b>--to-large</b>'  option.  The  default behavior is recommended since it will lead to less
              significant p/q-values in general but more specific results. Keep in mind that scaling  down  will
              influence control/input sample more. DEFAULT: 'small', the choice is either 'small' or 'large'.

       <b>--down-sample</b>
              When  set,  random sampling method will scale down the bigger sample. By default, MACS uses linear
              scaling.  Warning: This option will make your result unstable and irreproducible since each  time,
              random  reads  would  be selected. Consider to use 'randsample' script instead. &lt;not implmented&gt;If
              used together with <b>--SPMR</b>, 1 million unique  reads  will  be  randomly  picked.&lt;/not  implemented&gt;
              Caution: due to the implementation, the final number of selected reads may not be as you expected!
              DEFAULT: False

       <b>--seed</b> SEED
              Set  the  random  seed  while  down  sampling  data. Must be a non-negative integer in order to be
              effective.  DEFAULT: not set

       <b>--tempdir</b> TEMPDIR
              Optional directory to store temp files. DEFAULT: <u><a href="file:/tmp">/tmp</a></u>

       <b>--nolambda</b>
              If True, MACS will use fixed background lambda as local lambda for every  peak  region.  Normally,
              MACS  calculates  a  dynamic local lambda to reflect the local bias due to the potential chromatin
              accessibility.

       <b>--slocal</b> SMALLLOCAL
              The small nearby region in basepairs to calculate dynamic lambda. This is used to capture the bias
              near the peak summit region. Invalid if there is no control data. If you set this to 0, MACS  will
              skip  slocal  lambda  calculation.  *Note*  that  MACS  will  always perform a d-size local lambda
              calculation while the control data is available. The final local bias would be the maximum of  the
              lambda value from d, slocal, and llocal size windows. While control is not available, d and slocal
              lambda won't be considered. DEFAULT: 1000

       <b>--llocal</b> LARGELOCAL
              The  large  nearby  region  in  basepairs to calculate dynamic lambda. This is used to capture the
              surround bias. If you set this to 0, MACS will skip llocal lambda calculation.  *Note*  that  MACS
              will  always  perform  a  d-size local lambda calculation while the control data is available. The
              final local bias would be the maximum of the lambda value from d, slocal, and llocal size windows.
              While control is not available, d and slocal lambda won't be considered. DEFAULT: 10000.

       <b>--max-gap</b> MAXGAP
              Maximum gap between significant sites to cluster them together. The DEFAULT value is the  detected
              read length/tag size.

       <b>--min-length</b> MINLEN
              Minimum  length  of a peak. The DEFAULT value is the predicted fragment size d. Note, if you set a
              value smaller than the fragment size, it may have NO effect on the result. For BROAD peak calling,
              try to set a large value such as 500bps. You can also use '-- cutoff-analysis' option with default
              setting, and check the column 'avelpeak' under different cutoff  values  to  decide  a  reasonable
              minlen value.

       <b>--broad</b>
              If  set,  MACS  will  try  to  call  broad  peaks  using  the <b>--broad-cutoff</b> setting. Please tweak
              '--broad-cutoff' setting to control the peak calling behavior. At the meantime, either  <b>-q</b>  or  <b>-p</b>
              cutoff  will  be  used  to  define  regions  with 'stronger enrichment' inside of broad peaks. The
              maximum gap is expanded to 4 * MAXGAP (<b>--max-gap</b> parameter). As  a  result,  MACS  will  output  a
              'gappedPeak'  and  a  'broadPeak'  file  instead  of 'narrowPeak' file. Note, a broad peak will be
              reported even if there is no 'stronger enrichment' inside.  DEFAULT: False

       <b>--broad-cutoff</b> BROADCUTOFF
              Cutoff for broad region. This option is not available unless <b>--broad</b> is set. If <b>-p</b> is set, this is
              a pvalue cutoff, otherwise, it's a qvalue cutoff. Please note  that  in  broad  peakcalling  mode,
              MACS3  uses  this setting to control the overall peak calling behavior, then uses <b>-q</b> or <b>-p</b> setting
              to define regions inside broad region as 'stronger' enrichment. DEFAULT: 0.1

       <b>--cutoff-analysis</b>
              While set, MACS3 will analyze number or total length of peaks that  can  be  called  by  different
              p-value  cutoff then output a summary table to help user decide a better cutoff. The table will be
              saved in NAME_cutoff_analysis.txt file. Note, minlen and maxgap may affect the  results.  WARNING:
              May  take  ~30  folds  longer  time  to  finish.  The  result  can be useful for users to decide a
              reasonable cutoff value. DEFAULT: False

   <b>Post-processing</b> <b>options:</b>
       <b>--call-summits</b>
              If set, MACS will use a more sophisticated signal processing approach to find subpeak  summits  in
              each enriched peak region. DEFAULT: False

       <b>--fe-cutoff</b> FECUTOFF
              When  set,  the  value  will  be  used  as  the  minimum  requirement to filter out peaks with low
              foldenrichment. Note, MACS3 adds one as pseudocount while calculating fold-enrichment. By default,
              it is set as 1 so there is no filtering. DEFAULT: 1.0

   <b>Obsolete</b> <b>options:</b>
       <b>--to-large</b>
              Obsolete option. Please use '--scale-to large' instead.

       <b>--ratio</b> RATIO
              Obsolete option. Originally designed to normalize treatment and control with customized ratio, now
              it won't have any effect.

   <b>Other</b> <b>options:</b>
       <b>--buffer-size</b> BUFFER_SIZE
              Buffer size for incrementally increasing internal array size to store reads alignment information.
              In most cases, you don't have to change this parameter.  However, if there  are  large  number  of
              chromosomes/contigs/scaffolds in your alignment, it's recommended to specify a smaller buffer size
              in  order to decrease memory usage (but it will take longer time to read alignment files). Minimum
              memory requested for reading an alignment file is about # of CHROMOSOME * BUFFER_SIZE *  8  Bytes.
              DEFAULT: 100000

</pre><h4><b>EXAMPLES</b></h4><pre>
   <b>1.</b> <b>Peak</b> <b>calling</b> <b>for</b> <b>regular</b> <b>TF</b> <b>ChIP-seq:</b>
              $ macs3 callpeak -t ChIP.bam -c Control.bam -f BAM -g hs -n test -B -q 0.01

   <b>2.</b> <b>Broad</b> <b>peak</b> <b>calling</b> <b>on</b> <b>Histone</b> <b>Mark</b> <b>ChIP-seq:</b>
              $ macs3 callpeak -t ChIP.bam -c Control.bam --broad -g hs --broad-cutoff 0.1

   <b>3.</b> <b>Peak</b> <b>calling</b> <b>on</b> <b>ATAC-seq</b> <b>(paired-end</b> <b>mode):</b>
              $ macs3 callpeak -f BAMPE -t ATAC.bam -g hs -n test -B -q 0.01

   <b>4.</b> <b>Peak</b> <b>calling</b> <b>on</b> <b>ATAC-seq</b> <b>(</b> <b>focusing</b> <b>on</b> <b>insertion</b> <b>sites,</b> <b>and</b> <b>using</b> <b>single-end</b> <b>mode):</b>
              $ macs3 callpeak -f BAM -t ATAC.bam -g hs -n test -B -q 0.01 --shift -50 --extension 100

macs3 callpeak 3.0.2                               March 2025                                   <u><a href="../man1/MAC3_CALLPEAK.1.html">MAC3_CALLPEAK</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>