<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>PLINK - whole genome SNP analysis</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/plink1.9">plink1.9_1.90~b7.7-241022-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       PLINK - whole genome SNP analysis

</pre><h4><b>DESCRIPTION</b></h4><pre>
       PLINK  v1.90b6.16  64-bit  (19  Feb  2020)           www.cog-genomics.org/plink/1.9/  (C) 2005-2020 Shaun
       Purcell, Christopher Chang   GNU General Public License v3

       In the command line flag definitions that follow,

              * &lt;angle brackets&gt; denote a required parameter, where the text between the

              angle brackets describes its nature.

       * ['square brackets + single-quotes'] denotes an optional modifier.
              Use the

              EXACT text in the quotes.

              * [{bar|separated|braced|bracketed|values}] denotes a collection of mutually

       exclusive optional modifiers (again, the exact text must be used).
              When

              there are no outer square brackets, one of the choices must be selected.

              * ['quoted_text='&lt;description of value&gt;] denotes an optional modifier that

              must begin with the quoted text, and be followed by a value with no whitespace  in  between.   '|'
              may also be used here to indicate mutually exclusive options.

              * [square brackets without quotes or braces] denote an optional parameter,

              where the text between the brackets describes its nature.

              * An ellipsis (...) indicates that you may enter multiple parameters of the

              specified type.

              plink &lt;input flag(s)...&gt; [command flag(s)...] [other flag(s)...]  plink <b>--help</b> [flag name(s)...]

       Most  PLINK  runs require exactly one main input fileset.  The following flags are available for defining
       its form and location:

       <b>--bfile</b> [prefix] : Specify .bed + .bim + .fam prefix (default 'plink').

       <b>--bed</b> &lt;filename&gt; : Specify full name of .bed file.

       <b>--bim</b> &lt;filename&gt; : Specify full name of .bim file.

       <b>--fam</b> &lt;filename&gt; : Specify full name of .fam file.

       <b>--keep-autoconv</b>
              :  With  <b>--file</b>/--tfile/--lfile/--vcf/--bcf/--data/--23file,  don't  delete  autogenerated  binary
              fileset at end of run.

       <b>--file</b> [prefix]
              : Specify .ped + .map filename prefix (default 'plink').

       <b>--ped</b> &lt;filename&gt; : Specify full name of .ped file.

       <b>--map</b> &lt;filename&gt; : Specify full name of .map file.

       <b>--no-fid</b>
              : .fam/.ped file does not contain column 1 (family ID).

       <b>--no-parents</b>
              : .fam/.ped file does not contain columns 3-4 (parents).

       <b>--no-sex</b>
              : .fam/.ped file does not contain column 5 (sex).

       <b>--no-pheno</b>
              : .fam/.ped file does not contain column 6 (phenotype).

       <b>--tfile</b> [prefix] : Specify .tped + .tfam filename prefix (default 'plink').

       <b>--tped</b> &lt;fname&gt;
              : Specify full name of .tped file.

       <b>--tfam</b> &lt;fname&gt;
              : Specify full name of .tfam file.

       <b>--lfile</b> [prefix] : Specify .lgen + .map + .fam (long-format fileset) prefix.

       <b>--lgen</b> &lt;fname&gt;
              : Specify full name of .lgen file.

       <b>--reference</b> &lt;fn&gt; : Specify default allele file accompanying .lgen input.

       <b>--allele-count</b>
              :  When  used  with <b>--lfile</b>/--lgen + <b>--reference</b>, specifies that the .lgen file contains reference
              allele counts.

       <b>--vcf</b> &lt;filename&gt; : Specify full name of .vcf or .vcf.gz file.

       <b>--bcf</b> &lt;filename&gt; : Specify full name of BCF2 file.

       <b>--data</b> [prefix]
              : Specify Oxford .gen + .sample prefix (default 'plink').

       <b>--gen</b> &lt;filename&gt; : Specify full name of .gen or .gen.gz file.

       <b>--bgen</b> &lt;f&gt; ['snpid-chr'] : Specify full name of .bgen file.

       <b>--sample</b> &lt;fname&gt; : Specify full name of .sample file.

       <b>--23file</b> &lt;fname&gt; [FID] [IID] [sex] [pheno] [pat. ID] [mat. ID] :

              Specify 23andMe input file.

       <b>--grm-gz</b> [prfx]
              : Specify .grm.gz + .grm.id (GCTA rel. matrix) prefix.

       <b>--grm-bin</b> [prfx] : Specify .grm.bin + .grm.N.bin + .grm.id (GCTA triangular
              binary relationship matrix) filename prefix.

       <b>--dummy</b> &lt;sample ct&gt; &lt;SNP ct&gt; [missing geno freq] [missing pheno freq]

              [{acgt | 1234 | 12}] ['scalar-pheno']

              This generates a fake input dataset with the specified number of samples and  SNPs.   By  default,
              the  missing  genotype and phenotype frequencies are zero, and genotypes are As and Bs (change the
              latter with 'acgt'/'1234'/'12').  The 'scalar-pheno' modifier causes a normally distributed scalar
              phenotype to be generated instead of a binary one.

       <b>--simulate</b> &lt;simulation parameter file&gt; [{tags | haps}] [{acgt | 1234 | 12}]

       <b>--simulate-qt</b> &lt;sim. parameter file&gt; [{tags | haps}] [{acgt | 1234 | 12}]

       <b>--simulate</b> generates a fake input dataset with disease-associated SNPs,

              while <b>--simulate-qt</b> generates a dataset with quantitative trait loci.

       Output files have names of the form 'plink.&lt;extension&gt;' by default.  You can change  the  'plink'  prefix
       with

       <b>--out</b> &lt;prefix&gt;
              : Specify prefix for output files.

       Most runs also require at least one of the following commands:

       <b>--make-bed</b>

       Create a new binary fileset.
              Unlike the automatic text-to-binary

              converters (which only heed chromosome filters), this supports all of PLINK's filtering flags.

       <b>--make-just-bim</b>

       <b>--make-just-fam</b>

       Variants of <b>--make-bed</b> which only write a new .bim or .fam file.
              Can be

              used  with  only  .bim/.fam  input.   USE THESE CAUTIOUSLY.  It is very easy to desynchronize your
              binary genotype data and your .bim/.fam indexes if you use these commands improperly.  If you have
              any doubt, stick with <b>--make-bed</b>.

       <b>--recode</b> &lt;output format&gt; [{01 | 12}] [{tab | tabx | spacex | bgz | gen-gz}]

              ['include-alt'] ['omit-nonmale-y']

       Create a new text fileset with all filters applied.
              The following output

              formats are supported: * '23': 23andMe 4-column format.  This can only be used on a single

              sample's data (<b>--keep</b> may be handy), and does not support multicharacter allele codes.

              * 'A': Sample-major additive (0/1/2) coding, suitable for loading from R.

              If you need uncounted alleles to be named in the header line, add the 'include-alt' modifier.

              * 'AD': Sample-major additive (0/1/2) + dominant (het=1/hom=0) coding.

              Also supports 'include-alt'.

              * 'A-transpose': Variant-major 0/1/2.  * 'beagle': Unphased  per-autosome  .dat  and  .map  files,
              readable by early

              BEAGLE versions.

              *  'beagle-nomap':  Single .beagle.dat file.  * 'bimbam': Regular BIMBAM format.  * 'bimbam-1chr':
              BIMBAM format, with a two-column .pos.txt file.  Does not

              support multiple chromosomes.

              * 'fastphase': Per-chromosome fastPHASE files, with

              .chr-&lt;chr #&gt;.recode.phase.inp filename extensions.

       * 'fastphase-1chr': Single .recode.phase.inp file.
              Does not support

              multiple chromosomes.

              * 'HV': Per-chromosome Haploview files, with .chr-&lt;chr #&gt;{.ped,.info}

              filename extensions.

       * 'HV-1chr': Single Haploview .ped + .info file pair.
              Does not support

              multiple chromosomes.

              * 'lgen': PLINK 1 long-format (.lgen + .fam + .map), loadable with <b>--lfile</b>.  * 'lgen-ref': .lgen +
              .fam + .map + .ref, loadable with <b>--lfile</b> +

       <b>--reference</b>.

       * 'list': Single genotype-based list, up to 4 lines per variant.
              To omit

              nonmale genotypes on the Y chromosome, add the 'omit-nonmale-y' modifier.

              * 'rlist': .rlist + .fam + .map fileset, where the .rlist file is a

              genotype-based list which omits  the  most  common  genotype  for  each  variant.   Also  supports
              'omit-nonmale-y'.

       * 'oxford': Oxford-format .gen + .sample.
              With the 'gen-gz' modifier, the

              .gen file is gzipped.

              *  'ped':  PLINK 1 sample-major (.ped + .map), loadable with <b>--file</b>.  * 'compound-genotypes': Same
              as 'ped', except that the space between each

              pair of same-variant allele codes is removed.

              * 'structure': Structure-format.  * 'transpose': PLINK 1 variant-major (.tped +  .tfam),  loadable
              with

       <b>--tfile</b>.

       * 'vcf', 'vcf-fid', 'vcf-iid': VCFv4.2.
              'vcf-fid' and 'vcf-iid' cause

              family IDs or within-family IDs respectively to be used for the sample IDs in the last header row,
              while  'vcf' merges both IDs and puts an underscore between them.  If the 'bgz' modifier is added,
              the VCF file is block-gzipped.  The A2 allele is saved as the reference and  normally  flagged  as
              not  based on a real reference genome (INFO:PR).  When it is important for reference alleles to be
              correct, you'll also want to include <b>--a2-allele</b> and <b>--real-ref-alleles</b> in your command.

              In addition, * The '12' modifier causes A1 (usually minor) alleles to be coded as '1'

              and A2 alleles to be coded as '2', while '01' maps A1 -&gt; 0 and A2 -&gt; 1.

              * The 'tab' modifier makes the output mostly tab-delimited instead of

       mostly space-delimited.
              'tabx' and 'spacex' force all tabs and all

              spaces, respectively.

       <b>--flip-scan</b> ['verbose']

              (alias: <b>--flipscan</b>) LD-based scan for case/control strand inconsistency.

       <b>--write-covar</b>

              If a <b>--covar</b> file is loaded, <b>--make-bed</b>/--make-just-fam and  <b>--recode</b>  automatically  generate  an
              updated  version  (with  all  filters  applied).   However,  if  you do not wish to simultaneously
              generate a new genotype file, you can use <b>--write-covar</b> to just produce a pruned covariate file.

       <b>--write-cluster</b> ['omit-unassigned']

              If clusters are specified with <b>--within</b>/--family, this generates a  new  cluster  file  (with  all
              filters  applied).   The  'omit-unassigned' modifier causes unclustered samples to be omitted from
              the file; otherwise their cluster is 'NA'.

       <b>--write-set</b>

       <b>--set-table</b>

              If sets have been defined, <b>--write-set</b> dumps 'END'-terminated  set  membership  lists  to  &lt;output
              prefix&gt;.set,   while   <b>--set-table</b>   writes   a   variant-by-set   membership   table  to  &lt;output
              prefix&gt;.set.table.

       <b>--merge</b> &lt;.ped filename&gt; &lt;.map filename&gt;

       <b>--merge</b> &lt;text fileset prefix&gt;

       <b>--bmerge</b> &lt;.bed filename&gt; &lt;.bim filename&gt; &lt;.fam filename&gt;

       <b>--bmerge</b> &lt;binary fileset prefix&gt;

              Merge the given fileset  with  the  initially  loaded  fileset,  writing  the  result  to  &lt;output
              prefix&gt;.bed + .bim + .fam.  (It is no longer necessary to simultaneously specify <b>--make-bed</b>.)

       <b>--merge-list</b> &lt;filename&gt;

              Merge  all  filesets  named  in  the  text  file with the reference fileset, if one was specified.
              (However, this can also be used *without* a reference; in that case, the newly created fileset  is
              then  treated  as  the reference by most other PLINK operations.)  The text file is interpreted as
              follows: * If a line contains only one name, it is assumed to be the prefix for a

              binary fileset.

              * If a line contains exactly two names, they are assumed to be the full

              filenames for a text fileset (.ped first, then .map).

              * If a line contains exactly three names, they are assumed to be the full

              filenames for a binary fileset (.bed, then .bim, then .fam).

       <b>--write-snplist</b>

       <b>--list-23-indels</b>

       <b>--write-snplist</b> writes a .snplist file listing the names of all variants

              which pass the filters and inclusion thresholds you've specified,  while  <b>--list-23-indels</b>  writes
              the subset with 23andMe-style indel calls (D/I allele codes).

       <b>--list-duplicate-vars</b> ['require-same-ref'] ['ids-only'] ['suppress-first']

       <b>--list-duplicate-vars</b> writes a .dupvar file describing all groups of

              variants  with  matching  positions  and allele codes.  * By default, A1/A2 allele assignments are
              ignored; use 'require-same-ref'

              to override this.

       * Normally, the report contains position and allele codes.
              To remove them

              (and produce a file directly usable with e.g. <b>--extract</b>/--exclude),  use  'ids-only'.   Note  that
              this command will fail in 'ids-only' mode if any of the reported IDs are not unique.

              * 'suppress-first' causes the first variant ID in each group to be omitted

              from the report.

       <b>--freq</b> [{counts | case-control}] ['gz']

       <b>--freqx</b> ['gz']

       <b>--freq</b> generates a basic allele frequency (or count, if the 'counts'

       modifier is present) report.
              This can be combined with <b>--within</b>/--family

              to  produce  a  cluster-stratified  allele  frequency/count  report instead, or the 'case-control'
              modifier to report case and control allele  frequencies  separately.   <b>--freqx</b>  generates  a  more
              detailed genotype count report, designed for use with <b>--read-freq</b>.

       <b>--missing</b> ['gz']

       Generate sample- and variant-based missing data reports.
              If clusters are

       defined, the variant-based report is cluster-stratified.
              'gz' causes the

              output  files  to  be  gzipped.   Unlike  most other commands, this doesn't treat het. haploids as
              missing.

       <b>--test-mishap</b>

              Check for association between missing calls and flanking haplotypes.

       <b>--hardy</b> ['midp'] ['gz']

       Generate a Hardy-Weinberg exact test p-value report.
              (This does NOT

       simultaneously filter on the p-value any more; use <b>--hwe</b> for that.)
              With

              the 'midp' modifier, the test applies the mid-p adjustment described in  Graffelman  J,  Moreno  V
              (2013) The mid p-value in exact tests for Hardy-Weinberg Equilibrium.

       <b>--mendel</b> ['summaries-only']

       Generate a Mendel error report.
              The 'summaries-only' modifier causes the

              .mendel file (listing every single error) to be skipped.

       <b>--het</b> ['small-sample'] ['gz']

       <b>--ibc</b>

       Estimate inbreeding coefficients.
              <b>--het</b> reports method-of-moments

              estimates,  while  <b>--ibc</b>  calculates  all three values described in Yang J, Lee SH, Goddard ME and
              Visscher PM (2011) GCTA: A  Tool  for  Genome-wide  Complex  Trait  Analysis.   (That  paper  also
              describes  the  relationship matrix computation we reimplement.)  * These functions require decent
              MAF estimates.  If there are very few

              samples in your immediate fileset, <b>--read-freq</b> is practically mandatory  since  imputed  MAFs  are
              wildly inaccurate in that case.

              *  They  also  assume  the  marker set is in approximate linkage equilibrium.  * By default, <b>--het</b>
              omits the n/(n-1) multiplier in Nei's expected

       homozygosity formula.
              The 'small-sample' modifier causes it to be

              included, while forcing <b>--het</b> to use MAFs imputed from founders in the immediate dataset.

       <b>--check-sex</b> [female max F] [male min F]

       <b>--check-sex</b> ycount [female max F] [male min F] [female max Y obs]
              [male min Y obs]

       <b>--check-sex</b> y-only [female max Y obs] [male min Y obs]

       <b>--impute-sex</b> [female max F] [male min F]

       <b>--impute-sex</b> ycount [female max F] [male min F] [female max Y obs]
              [male min Y obs]

       <b>--impute-sex</b> y-only [female max Y obs] [male min Y obs]

       <b>--check-sex</b> normally compares sex assignments in the input dataset with

              those imputed from X chromosome inbreeding coefficients.   *  Make  sure  that  the  X  chromosome
              pseudo-autosomal region has been split

              off (with e.g. <b>--split-x</b>) before using this.

              * You also need decent MAF estimates (so, with very few samples in your

              immediate  fileset,  use  <b>--read-freq</b>),  and  your  marker  set  should  be in approximate linkage
              equilibrium.

              * By default, F estimates smaller than 0.2 yield female calls, and values

       larger than 0.8 yield male calls.
              If you pass numeric parameter(s) to

       <b>--check-sex</b>, the first two control these thresholds.

              There are now two modes which consider Y chromosome data.  * In 'ycount'  mode,  gender  is  still
              imputed from the X chromosome, but

              female  calls are downgraded to ambiguous whenever more than 0 nonmissing Y genotypes are present,
              and male calls are downgraded when fewer than 0 are present.  (Note that  these  are  counts,  not
              rates.)   These thresholds are controllable with <b>--check-sex</b> ycount's optional 3rd and 4th numeric
              parameters.

              * In 'y-only' mode, gender is imputed from nonmissing Y genotype counts.

              The male minimum threshold defaults to 1 instead of zero in this case.

       <b>--impute-sex</b> changes sex assignments to the imputed values, and is

       otherwise identical to <b>--check-sex</b>.
              It must be used with

       <b>--make-bed</b>/--recode/--write-covar.

       <b>--fst</b> ['case-control']

              (alias: <b>--Fst</b>) Estimate  Wright's  Fst  for  each  autosomal  diploid  variant  using  the  method
              introduced  in Weir BS, Cockerham CC (1984) Estimating F-statistics for the analysis of population
              structure, given a set of subpopulations defined via <b>--within</b>.  Raw and weighted global means  are
              also reported.  * If you're interested in the global means, it is usually best to perform

              this calculation on a marker set in approximate linkage equilibrium.

              * If you have only two subpopulations, you can represent them with

              case/control status and use the 'case-control' modifier.

       <b>--indep</b> &lt;window size&gt;['kb'] &lt;step size (variant ct)&gt; &lt;VIF threshold&gt;

       <b>--indep-pairwise</b> &lt;window size&gt;['kb'] &lt;step size (variant ct)&gt; &lt;r^2 threshold&gt;

       <b>--indep-pairphase</b> &lt;window size&gt;['kb'] &lt;step size (variant ct)&gt; &lt;r^2 thresh&gt;

       Generate a list of markers in approximate linkage equilibrium.
              With the

              'kb'  modifier, the window size is in kilobase instead of variant count units.  (Pre-'kb' space is
              optional, i.e. "--indep-pairwise 500 kb 5 0.5" and "--indep-pairwise 500kb 5 0.5"  have  the  same
              effect.)    Note   that   you   need   to   rerun  PLINK  using  <b>--extract</b>  or  <b>--exclude</b>  on  the
              .prune.in/.prune.out file to apply the list to another computation.

       <b>--r</b> [{square | square0 | triangle | inter-chr}] [{gz | bin | bin4}]

              ['spaces'] ['in-phase'] [{d | dprime | dprime-signed}] ['with-freqs'] ['yes-really']

       <b>--r2</b> [{square | square0 | triangle | inter-chr}] [{gz | bin | bin4}]

              ['spaces'] ['in-phase'] [{d | dprime | dprime-signed}] ['with-freqs'] ['yes-really']

       LD statistic reports.
              <b>--r</b> yields raw inter-variant correlations, while

       <b>--r2</b> reports their squares.
              You can request results for all pairs in

              matrix format (if you specify 'bin' or one of the shape modifiers),  all  pairs  in  table  format
              ('inter-chr'),  or  a  limited  window  in table format (default).  * The 'gz' modifier causes the
              output  text  file  to  be  gzipped.   *  'bin'  causes  the  output  matrix  to  be  written   in
              double-precision binary

       format, while 'bin4' specifics single-precision binary.
              The matrix is

              square if no shape is explicitly specified.

              * By default, text matrices are tab-delimited; 'spaces' switches this.  * 'in-phase' adds a column
              with in-phase allele pairs to table-formatted

       reports.
              (This cannot be used with very long allele codes.)

              * 'dprime' adds the absolute value of Lewontin's D-prime statistic to

              table-formatted  reports,  and forces both r/r^2 and D-prime to be based on the maximum likelihood
              solution to the cubic equation discussed in Gaunt  T,  Rodriguez  S,  Day  I  (2007)  Cubic  exact
              solutions  for  the estimation of pairwise haplotype frequencies.  'dprime-signed' keeps the sign,
              while 'd' skips division by D_{max}.

              * 'with-freqs' adds MAF columns to table-formatted reports.  * Since the resulting file can easily
              be huge, you're required to add the

              'yes-really' modifier when requesting an unfiltered, non-distributed all pairs computation on more
              than 400k variants.

              * These computations can be subdivided with <b>--parallel</b> (even when the

              'square' modifier is active).

       <b>--ld</b> &lt;variant ID&gt; &lt;variant ID&gt; ['hwe-midp']

              This displays haplotype frequencies, r^2, and D' for a single pair of variants.   When  there  are
              multiple  biologically  possible  solutions  to  the  haplotype  frequency cubic equation, all are
              displayed (instead of just the maximum likelihood solution identified by <b>--r</b>/--r2), along with HWE
              exact test statistics.

       <b>--show-tags</b> &lt;filename&gt;

       <b>--show-tags</b> all

              * If a file is specified, list all variants which tag at least one variant

       named in the file.
              (This will normally be a superset of the original

              list, since a variant is considered to tag itself here.)

              * If 'all' mode is specified, for each variant, each *other* variant which

              tags it is reported.

       <b>--blocks</b> ['no-pheno-req'] ['no-small-max-span']

              Estimate haplotype blocks, via Haploview's interpretation of the  block  definition  suggested  by
              Gabriel  S  et  al.  (2002)  The  Structure  of Haplotype Blocks in the Human Genome.  * Normally,
              samples with missing phenotypes are not considered by this

              computation; the 'no-pheno-req' modifier lifts this restriction.

              * Normally, size-2 blocks may not span more than 20kb, and size-3 blocks

       are limited to 30kb.
              The 'no-small-max-span' modifier removes these

              limits.

       The .blocks file is valid input for PLINK 1.07's <b>--hap</b> command.
              However,

              the <b>--hap</b>... family of flags has not been reimplemented in PLINK 1.9 due to poor phasing  accuracy
              relative  to  other software; for now, we recommend using BEAGLE instead of PLINK for case/control
              haplotype association analysis.  (You can use "--recode beagle" to export data to BEAGLE 3.3.)  We
              apologize for the inconvenience, and plan to develop variants of the <b>--hap</b>... flags  which  handle
              pre-phased data effectively.

       <b>--distance</b> [{square | square0 | triangle}] [{gz | bin | bin4}] ['ibs']

              ['1-ibs'] ['allele-ct'] ['flat-missing']

              Write a lower-triangular tab-delimited table of (weighted) genomic distances in allele count units
              to  &lt;output  prefix&gt;.dist,  and a list of the corresponding sample IDs to &lt;output prefix&gt;.dist.id.
              The first row of the .dist file contains a single &lt;genome 1-genome 2&gt; distance, the second row has
              the &lt;genome 1-genome 3&gt; and &lt;genome 2-genome 3&gt; distances in that order, etc.   *  It  is  usually
              best to perform this calculation on a marker set in

              approximate linkage equilibrium.

              * If the 'square' or 'square0' modifier is present, a square matrix is

              written instead; 'square0' fills the upper right triangle with zeroes.

              * If the 'gz' modifier is present, a compressed .dist.gz file is written

              instead of a plain text file.

              * If the 'bin' modifier is present, a binary (square) matrix of

              double-precision floating point values, suitable for loading from R, is instead written to &lt;output
              prefix&gt;.dist.bin.  ('bin4' specifies single-precision numbers instead.)  This can be combined with
              'square0' if you still want the upper right zeroed out, or 'triangle' if you don't want to pad the
              upper right at all.

              * If the 'ibs' modifier is present, an identity-by-state matrix is written

       to &lt;output prefix&gt;.mibs.
              '1-ibs' causes distances expressed as genomic

              proportions  (i.e.  1  - IBS) to be written to &lt;output prefix&gt;.mdist.  Combine with 'allele-ct' if
              you want to generate the usual .dist file as well.

              * By default, distance rescaling in the presence of missing genotype calls

              is sensitive to allele count distributions: if variant A contributes, on average, twice as much to
              other pairwise distances as variant B, a missing call at variant A will result in twice  as  large
              of  a  missingness  correction.   To  turn  this  off  (because e.g. your missing calls are highly
              nonrandom), use the 'flat-missing' modifier.

              * The computation can be subdivided with <b>--parallel</b>.

       <b>--distance-matrix</b>

       <b>--ibs-matrix</b>

              These deprecated commands are equivalent to "--distance 1-ibs flat-missing square" and "--distance
              ibs flat-missing square", respectively, except that they generate space- instead of  tab-delimited
              text matrices.

       <b>--make-rel</b> [{square | square0 | triangle}] [{gz | bin | bin4}]

              [{cov | ibc2 | ibc3}]

              Write   a   lower-triangular   variance-standardized   realized  relationship  matrix  to  &lt;output
              prefix&gt;.rel, and corresponding IDs to &lt;output prefix&gt;.rel.id.  * It is  usually  best  to  perform
              this calculation on a marker set in

              approximate linkage equilibrium.

              * 'square', 'square0', 'triangle', 'gz', 'bin', and 'bin4' act as they do

              on <b>--distance</b>.

              * The 'cov' modifier removes the variance standardization step, causing a

              covariance matrix to be calculated instead.

              * By default, the diagonal elements in the relationship matrix are based on

       <b>--ibc</b>'s Fhat1; use the 'ibc2' or 'ibc3' modifiers to base them on Fhat2

              or Fhat3 instead.

              * The computation can be subdivided with <b>--parallel</b>.

       <b>--make-grm-gz</b> ['no-gz'] [{cov | ibc2 | ibc3}]

       <b>--make-grm-bin</b> [{cov | ibc2 | ibc3}]

       <b>--make-grm-gz</b> writes the relationships in GCTA's original gzipped list

              format,  which  describes  one  pair  per  line,  while  <b>--make-grm-bin</b> writes them in GCTA 1.1+'s
              single-precision triangular binary format.  Note that these formats explicitly report  the  number
              of  valid  observations  (where  neither sample has a missing call) for each pair, which is useful
              input for some scripts.  These computations can be subdivided with <b>--parallel</b>.

       <b>--rel-cutoff</b> [val]

              (alias: <b>--grm-cutoff</b>) Exclude one member of each pair of samples with relatedness greater than the
              given cutoff value (default 0.025).  If no later  operation  will  cause  the  list  of  remaining
              samples  to be written to disk, this will save it to &lt;output prefix&gt;.rel.id.  Note that maximizing
              the remaining sample size is equivalent to the NP-hard maximum independent set problem, so we  use
              a  greedy  algorithm  instead of guaranteeing optimality.  (Use the <b>--make-rel</b> and <b>--keep</b>/--remove
              flags if you want to try to do better.)

       <b>--ibs-test</b> [permutation count]

       <b>--groupdist</b> [iters] [d]

              Given case/control phenotype data, these commands consider three subsets of the  distance  matrix:
              pairs  of  affected  samples, affected-unaffected pairs, and pairs of unaffected samples.  Each of
              these subsets has a distribution of pairwise genomic distances;  <b>--ibs-test</b>  uses  permutation  to
              estimate  p-values  re:  which  types  of pairs are most similar, while <b>--groupdist</b> focuses on the
              differences between the centers of these distributions and estimates standard errors via  delete-d
              jackknife.

       <b>--regress-distance</b> [iters] [d]

              Linear  regression  of  pairwise  genomic distances on pairwise average phenotypes and vice versa,
              using delete-d jackknife for standard errors.  A scalar phenotype is required.  * With  less  than
              two parameters, d is set to &lt;number of people&gt;^0.6 rounded

       down.
              With no parameters, 100k iterations are run.

       <b>--regress-rel</b> [iters] [d]

              Linear  regression  of  pairwise  genomic  relationships  on pairwise average phenotypes, and vice
              versa.  Defaults for iters and d are the same as for <b>--regress-distance</b>.

       <b>--genome</b> ['gz'] ['rel-check'] ['full'] ['unbounded'] ['nudge']

              Generate an identity-by-descent report.  * It is usually best to perform  this  calculation  on  a
              marker set in

              approximate linkage equilibrium.

              * The 'rel-check' modifier excludes pairs of samples with different FIDs

              from the final report.

              *  'full'  adds raw pairwise comparison data to the report.  * The P(IBD=0/1/2) estimator employed
              by this command sometimes yields

       numbers outside the range [0,1]; by default, these are clipped.
              The

              'unbounded' modifier turns off this clipping.

              * Then, when PI_HAT^2 &lt; P(IBD=2), 'nudge' adjusts the final P(IBD=0/1/2)

              estimates to a theoretically possible configuration.

              * The computation can be subdivided with <b>--parallel</b>.

       <b>--homozyg</b> [{group | group-verbose}] ['consensus-match'] ['extend']

              ['subtract-1-from-lengths']

       <b>--homozyg-snp</b> &lt;min var count&gt;

       <b>--homozyg-kb</b> &lt;min length&gt;

       <b>--homozyg-density</b> &lt;max inverse density (kb/var)&gt;

       <b>--homozyg-gap</b> &lt;max internal gap kb length&gt;

       <b>--homozyg-het</b> &lt;max hets&gt;

       <b>--homozyg-window-snp</b> &lt;scanning window size&gt;

       <b>--homozyg-window-het</b> &lt;max hets in scanning window hit&gt;

       <b>--homozyg-window-missing</b> &lt;max missing calls in scanning window hit&gt;

       <b>--homozyg-window-threshold</b> &lt;min scanning window hit rate&gt;

              These commands request a set of run-of-homozygosity reports, and allow you to customize  how  they
              are generated.  * If you're satisfied with all the default settings described below, just

       use <b>--homozyg</b> with no modifiers.
              Otherwise, <b>--homozyg</b> lets you change a

              few binary settings: * 'group[-verbose]' adds a report on pools of overlapping runs of

       homozygosity.
              (Automatically set when <b>--homozyg-match</b> is present.)

              * With 'group[-verbose]', 'consensus-match' causes pairwise segmental

              matches  to  be  called  based  on  the  variants in the pool's consensus segment, rather than the
              variants in the pairwise intersection.

              * Due to how the scanning window algorithm works, it is possible for a

       reported ROH to be adjacent to a few homozygous variants.
              The 'extend'

              modifier causes them to be included in the reported ROH if that wouldn't cause a violation of  the
              <b>--homozyg-density</b> bound.

              * By default, segment bp lengths are calculated as &lt;end bp position&gt; -

       &lt;start bp position&gt; + 1.
              Therefore, reports normally differ slightly

       from PLINK 1.07, which does not add 1 at the end.
              For testing

              purposes, you can use the 'subtract-1-from-lengths' modifier to apply the old formula.

              * By default, only runs of homozygosity containing at least 100 variants,

       and of total length &gt;= 1000 kilobases, are noted.
              You can change these

              minimums with <b>--homozyg-snp</b> and <b>--homozyg-kb</b>, respectively.

              * By default, a ROH must have at least one variant per 50 kb on average;

              change this bound with <b>--homozyg-density</b>.

              * By default, if two consecutive variants are more than 1000 kb apart, they

              cannot be in the same ROH; change this bound with <b>--homozyg-gap</b>.

              * By default, a ROH can contain an unlimited number of heterozygous calls;

              you can impose a limit with <b>--homozyg-het</b>.

              * By default, the scanning window contains 50 variants; change this with

       <b>--homozyg-window-snp</b>.

              * By default, a scanning window hit can contain at most 1 heterozygous

              call    and    5    missing   calls;   change   these   limits   with   <b>--homozyg-window-het</b>   and
              <b>--homozyg-window-missing</b>, respectively.

              * By default, for a variant to be eligible for inclusion in a ROH, the hit

              rate of all scanning windows containing the variant must be at least 0.05; change  this  threshold
              with <b>--homozyg-window-threshold</b>.

       <b>--cluster</b> ['cc'] [{group-avg | old-tiebreaks}] ['missing'] ['only2']

              Cluster  samples using a pairwise similarity statistic (normally IBS).  * The 'cc' modifier forces
              every cluster to have at least one case and one

              control.

              * The 'group-avg' modifier causes clusters to be joined based on average

              instead of minimum pairwise similarity.

              * The 'missing' modifier causes clustering to be based on

              identity-by-missingness   instead   of   identity-by-state,   and   writes    a    space-delimited
              identity-by-missingness matrix to disk.

              * The 'only2' modifier causes only a .cluster2 file (which is valid input

              for <b>--within</b>) to be written; otherwise 2 other files will be produced.

              * By default, IBS ties are not broken in the same manner as PLINK 1.07, so

       final cluster solutions tend to differ.
              This is generally harmless.

              However,  to  simplify testing, you can use the 'old-tiebreaks' modifier to force emulation of the
              old algorithm.

       <b>--pca</b> [count] ['header'] ['tabs'] ['var-wts']

              Calculates        a         variance-standardized         relationship         matrix         (use
              <b>--make-rel</b>/--make-grm-gz/--make-grm-bin to dump it), and extracts the top 20 principal components.
              * It is usually best to perform this calculation on a marker set in

              approximate linkage equilibrium.

              * You can change the number of PCs by passing a numeric parameter.  * The 'header' modifier adds a
              header line to the .eigenvec output file.

              (For compatibility with the GCTA flag of the same name, the default is no header line.)

              *  The 'tabs' modifier causes the .eigenvec file(s) to be tab-delimited.  * The 'var-wts' modifier
              requests an additional .eigenvec.var file with PCs

              expressed as variant weights instead of sample weights.

       <b>--neighbour</b> &lt;n1&gt; &lt;n2&gt;

              (alias: <b>--neighbor</b>) Report  IBS  distances  from  each  sample  to  their  n1th-  to  n2th-nearest
              neighbors,  associated  Z-scores,  and  the  identities  of  those  neighbors.  Useful for outlier
              detection.

       <b>--assoc</b> ['perm' | 'mperm='&lt;value&gt;] ['perm-count'] [{fisher | fisher-midp}]

              ['counts'] ['set-test']

       <b>--assoc</b> ['perm' | 'mperm='&lt;value&gt;] ['perm-count'] ['qt-means'] ['lin']

              ['set-test']

       <b>--model</b> ['perm' | 'mperm='&lt;value&gt;] ['perm-count']

              [{fisher | fisher-midp | trend-only}] ['set-test'] [{dom | rec | gen | trend}]

              Basic association analysis report.   Given  a  case/control  phenotype,  <b>--assoc</b>  performs  a  1df
              chi-square  allelic  test, while <b>--model</b> performs 4 other tests as well (1df dominant gene action,
              1df recessive gene action, 2df genotypic, Cochran-Armitage trend).  * With 'fisher'/'fisher-midp',
              Fisher's exact test is used to generate

       p-values.
              'fisher-midp' also applies Lancaster's mid-p adjustment.

              * 'perm' causes an adaptive permutation test to be performed.  * 'mperm='&lt;value&gt; causes  a  max(T)
              permutation test with the specified

              number of replications to be performed.

              * 'perm-count' causes the permutation test report to include counts instead

              of frequencies.

              *  'counts' causes <b>--assoc</b> to report allele counts instead of frequencies.  * 'set-test' tests the
              significance of variant sets.  Requires permutation;

              can be customized with <b>--set-p</b>/--set-r2/--set-max.

              * 'dom', 'rec', 'gen', and 'trend' force the corresponding test to be used

       as the basis for <b>--model</b> permutation.
              (By default, the most significant

              result among the allelic, dominant, and recessive tests is used.)

              * 'trend-only' causes only the trend test  to  be  performed.   Given  a  quantitative  phenotype,
              <b>--assoc</b> normally performs a Wald test.  * In this case, the 'qt-means' modifier causes trait means
              and standard

              deviations stratified by genotype to be reported as well.

              * 'lin' causes the Lin statistic to be computed, and makes it the basis for

              multiple-testing corrections and permutation tests.

              Several other flags (most notably, <b>--aperm</b>) can be used to customize the permutation test.

       <b>--mh</b> ['perm' | 'mperm='&lt;value&gt;] ['perm-count'] ['set-test']

              (alias: <b>--cmh</b>)

       <b>--bd</b> ['perm' | 'perm-bd' | 'mperm='&lt;value&gt;] ['perm-count'] ['set-test']

       <b>--mh2</b>

       <b>--homog</b>

              Given  a case/control phenotype and a set of clusters, <b>--mh</b> computes 2x2xK Cochran-Mantel-Haenszel
              statistics for each variant, while  <b>--bd</b>  also  performs  the  Breslow-Day  test  for  odds  ratio
              homogeneity.   Permutation and variant set testing based on the CMH (default) or Breslow-Day (when
              'perm-bd' is present) statistic are supported.  The following similar analyses are also available:
              * <b>--mh2</b> swaps the roles of case/control status and cluster membership,

              performing a  phenotype-stratified  IxJxK  Cochran-Mantel-Haenszel  test  on  association  between
              cluster assignments and genotypes.

              * <b>--homog</b> executes an alternative to the Breslow-Day test, based on

              partitioning of the chi-square statistic.

       <b>--gxe</b> [covariate index]

              Given  both a quantitative phenotype and a case/control covariate loaded with <b>--covar</b> defining two
              groups, <b>--gxe</b> compares the regression coefficient derived from considering  only  members  of  one
              group  to  the  regression  coefficient  derived  from  considering only members of the other.  By
              default, the first covariate in the <b>--covar</b> file defines the groups; use e.g. "--gxe  3"  to  base
              them on the third covariate instead.

       <b>--linear</b> ['perm' | 'mperm='&lt;value&gt;] ['perm-count'] ['set-test']

              [{genotypic  |  hethom  |  dominant  |  recessive  |  no-snp}]  ['hide-covar']  [{sex | no-x-sex}]
              ['interaction'] ['beta'] ['standard-beta'] ['intercept']

       <b>--logistic</b> ['perm' | 'mperm='&lt;value&gt;] ['perm-count'] ['set-test']

              [{genotypic | hethom  |  dominant  |  recessive  |  no-snp}]  ['hide-covar']  [{sex  |  no-x-sex}]
              ['interaction'] ['beta'] ['intercept']

              Multi-covariate  association  analysis  on  a quantitative (<b>--linear</b>) or case/control (<b>--logistic</b>)
              phenotype.  Normally used with <b>--covar</b>.  * 'perm' normally causes an adaptive permutation test  to
              be performed on

              the main effect, while 'mperm='&lt;value&gt; starts a max(T) permutation test.

              * 'perm-count' causes the permutation test report to include counts instead

              of frequencies.

       * 'set-test' tests the significance of variant sets.
              Requires permutation;

              can be customized with <b>--set-p</b>/--set-r2/--set-max.

              * The 'genotypic' modifier adds an additive effect/dominance deviation 2df

              joint  test  (0/1/2  and  0/1/0  coding),  while 'hethom' uses 0/0/1 and 0/1/0 coding instead.  If
              permutation is also requested, these modifiers cause permutation to be based on the joint test.

              * 'dominant' and 'recessive' specify a model assuming full dominance or

              recessiveness, respectively, for the A1 allele.

              * 'no-snp' causes regression to be performed only on the phenotype and the

       covariates, without reference to genomic data.
              If permutation is also

              requested, results are reported for all covariates.

              * 'hide-covar' removes covariate-specific lines from the report.  * By default,  sex  (male  =  1,
              female = 0) is automatically added as a

       covariate on X chromosome variants, and nowhere else.
              The 'sex' modifier

              causes it to be added everywhere, while 'no-x-sex' excludes it.

       * 'interaction' adds genotype x covariate interactions to the model.
              This

              cannot  be  used  with  the  usual  permutation  tests; use <b>--tests</b> to define the permutation test
              statistic instead.

              * 'intercept' causes intercepts to be included in the main report.  *  For  logistic  regressions,
              the 'beta' modifier causes regression

              coefficients instead of odds ratios to be reported.

              * With <b>--linear</b>, the 'standard-beta' modifier standardizes the phenotype

              and all predictors to zero mean and unit variance before regression.

       <b>--dosage</b> &lt;allele dosage file&gt; ['noheader'] ['skip0='&lt;i&gt;] ['skip1='&lt;j&gt;]

              ['skip2='&lt;k&gt;]    ['dose1']    ['format='&lt;m&gt;]    ['Zout']   [{occur   |   standard-beta}]   ['sex']
              ['case-control-freqs']

       <b>--dosage</b> &lt;list file&gt; list [{sepheader | noheader}] ['skip0='&lt;i&gt;]

              ['skip1='&lt;j&gt;] ['skip2='&lt;k&gt;] ['dose1'] ['format='&lt;m&gt;] ['Zout'] [{occur  |  standard-beta}]  ['sex']
              ['case-control-freqs']

       <b>--write-dosage</b>

              Process (possibly gzipped) text files with variant-major allelic dosage data.  This cannot be used
              with  a  regular  input fileset; instead, you must *only* specify a .fam and possibly a .map file,
              and you can't specify any other commands.  * PLINK 2.0 will have first-class support for  genotype
              probabilities.  An

              equivalent data import flag will be provided then, and <b>--dosage</b> will be retired.

              * By default, <b>--dosage</b> assumes that only one allelic dosage file should be

       loaded.
              To specify multiple files,

       1. create a master list with one entry per line.
              There are normally two

              supported  formats  for this list: just a filename per line, or variant batch numbers in the first
              column and filenames in the second.

              2. Provide the name of that list as the first <b>--dosage</b> parameter.  3. Add the 'list' modifier.

              * By default, <b>--dosage</b> assumes the allelic dosage file(s) contain a header

              line, which has 'SNP' in column i+1, 'A1' in column  i+j+2,  'A2'  in  column  i+j+3,  and  sample
              FID/IIDs starting from column i+j+k+4.  (i/j/k are normally zero, but can be changed with 'skip0',
              'skip1',  and  'skip2'  respectively.)   If  such a header line is not present, * when all samples
              appear in the same order as they do in the .fam file,

              you can use the 'noheader' modifier.

              * Otherwise, use the 'sepheader' modifier, and append sample ID filenames

              to your 'list' file entries.

              * The 'format=' modifier lets you specify the number of values used to

       represent each dosage.
              'format=1' normally indicates a single 0..2 A1

       expected count; 'dose1' modifies this to a 0..1 frequency.
              'format=2'

              (the default) indicates a 0..1 homozygous A1 likelihood followed by a 0..1 het  likelihood,  while
              'format=3' indicates 0..1 hom A1, 0..1 het, 0..1 hom A2.

              *  'Zout' causes the output file to be gzipped.  * Normally, an association analysis is performed.
              'standard-beta' and

              'sex' behave as they are supposed to with <b>--linear</b>/--logistic.  'case-control-freqs'  causes  case
              and control allele frequencies to be reported separately.

              * There are three alternate modes which cause the association analysis to

              be  skipped.   *  'occur'  requests a simple variant occurrence report.  * <b>--write-dosage</b> causes a
              simple merged file matching the 'format'

              specification (not including 'dose1') to be generated.

              * <b>--score</b> applies a linear scoring system to the dosages.

       <b>--lasso</b> &lt;h2 estimate&gt; [min lambda] ['report-zeroes']

       Estimate variant effect sizes via LASSO regression.
              You must provide an

              additive heritability estimate to calibrate the regression.  Note that this method may  require  a
              very large sample size (e.g. hundreds of thousands) to be effective on complex polygenic traits.

       <b>--test-missing</b> ['perm' | 'mperm='&lt;value&gt;] ['perm-count'] ['midp']

              Check  for  association  between  missingness  and case/control status, using Fisher's exact test.
              (Het. haploids are treated as missing.)  The 'midp' modifier causes Lancaster's  mid-p  adjustment
              to be applied.

       <b>--make-perm-pheno</b> &lt;ct&gt;

              Generate phenotype permutations and write them to disk, without invoking an association test.

       <b>--tdt</b> [{exact | exact-midp | poo}] ['perm' | 'mperm='&lt;value&gt;] ['perm-count']

              [{parentdt1 | parentdt2 | pat | mat}] ['set-test']

              Report  transmission  disequilibrium  test  statistics, given case/control phenotypes and pedigree
              information.  * A Mendel error check is performed before the main tests; offending

              genotypes are treated as missing by this analysis.

              * By default, the basic TDT p-value is based on a chi-square test unless

              you request the exact binomial test with 'exact' or 'exact-midp'.

              * 'perm'/'mperm=' requests a family-based adaptive or max(T) permutation

       test.
              By default, the permutation test statistic is the basic TDT

              p-value; 'parentdt1'/'parentdt2' cause parenTDT or combined test  p-values,  respectively,  to  be
              considered instead.

       * 'set-test' tests the significance of variant sets.
              This cannot be used

              with exact tests for now.

              The  'poo' modifier causes a parent-of-origin analysis to be performed instead, with transmissions
              from heterozygous fathers and heterozygous mothers considered separately.  * The  parent-of-origin
              analysis  does not currently support exact tests.  * By default, the permutation test statistic is
              the absolute

              parent-of-origin test Z score; 'pat'/'mat' cause paternal or maternal TDT  chi-square  statistics,
              respectively, to be considered instead.

       <b>--qfam</b> ['perm' | 'mperm='&lt;value&gt;] ['perm-count'] ['emp-se']

       <b>--qfam-parents</b> ['perm' | 'mperm='&lt;value&gt;] ['perm-count'] ['emp-se']

       <b>--qfam-between</b> ['perm' | 'mperm='&lt;value&gt;] ['perm-count'] ['emp-se']

       <b>--qfam-total</b> ['perm' | 'mperm='&lt;value&gt;] ['perm-count'] ['emp-se']

              QFAM  family-based  association test for quantitative traits.  * A Mendel error check is performed
              before the main tests; offending

              genotypes are treated as missing by this analysis.

       * This procedure requires permutation.
              'perm' and 'perm-count' have the

       usual meanings.
              However, 'mperm='&lt;value&gt; just specifies a fixed number

              of permutations; the method does not support a proper max(T) test.

              * The 'emp-se' modifier adds BETA and EMP_SE (empirical standard error for

              beta) fields to the .perm output file.

       <b>--annotate</b> &lt;PLINK report&gt; ['attrib='&lt;file&gt;] ['ranges='&lt;file&gt;]

              ['filter='&lt;file&gt;]  ['snps='&lt;file&gt;]  [{NA  |  prune}]   ['block']   ['subset='&lt;file&gt;]   ['minimal']
              ['distance']

       Add annotations to a variant-based PLINK report.
              This requires an

              annotation   source:   *  'attrib='&lt;file&gt;  specifies  a  (possibly  gzipped)  attribute  file.   *
              'ranges='&lt;file&gt;  specifies  a  gene/range  list  file.   (Both  source  types  can  be   specified
              simultaneously.)  The following options are also supported: * 'filter='&lt;file&gt; causes only variants
              within one of the ranges in the file

              to be included in the new report.

              * 'snps='&lt;file&gt; causes only variants named in the file to be included in

              the new report.

              * The 'NA' modifier causes unannotated variants to have 'NA' instead of '.'

              in the new report's ANNOT column, while the 'prune' modifier excludes them entirely.

              * The 'block' modifier replaces the single ANNOT column with a 0/1-coded

              column for each possible annotation.

              * With 'ranges',

              * 'subset='&lt;file&gt; causes only intervals named in the subset file to be

              loaded from the ranges file.

              * interval annotations normally come with a parenthesized signed distance

              to  the  interval  boundary  (0 if the variant is located inside the interval; this is always true
              without <b>--border</b>).  They can be excluded with the 'minimal' modifier.

              * the 'distance' modifier adds 'DIST' and 'SGN' columns describing signed

              distance to the nearest interval.

              * When <b>--pfilter</b> is present, high p-values are filtered out.

       <b>--clump</b> &lt;PLINK report filename(s)...&gt;

              Process association analysis report(s) with 'SNP'  and  p-value  columns,  organizing  results  by
              LD-based clumps.  Multiple filenames can be separated by spaces or commas.

       <b>--gene-report</b> &lt;PLINK report&gt; &lt;gene range file&gt;

              Generate  a  gene-based  report  from  a  variant-based report.  * When <b>--pfilter</b> is present, high
              p-values are filtered out.  * When <b>--extract</b> (without 'range') is present, only variants named  in
              the

       <b>--extract</b> file are considered.

       <b>--meta-analysis</b> &lt;PLINK report filenames...&gt;

       <b>--meta-analysis</b> &lt;PLINK report filenames...&gt; + [{logscale | qt}]

              [{no-map | no-allele}] ['study'] ['report-all'] ['weighted-z']

              Perform  a meta-analysis on several variant-based reports with 'SNP' and 'SE' fields.  * Normally,
              an 'OR' odds ratio field must also be present in each input

       file.
              With 'logscale', 'BETA' log-odds values/regression coefficients

              are expected instead, but the generated report will still  contain  odds  ratio  estimates.   With
              'qt', both input and output values are regression betas.

       * 'CHR', 'BP', and 'A1' fields are also normally required.
              'no-map' causes

              them to all be ignored, while 'no-allele' causes just 'A1' to be ignored.

              * If 'A2' fields are present, and neither 'no-map' nor 'no-allele' was

       specified, A1/A2 allele flips are handled properly.
              Otherwise, A1

              mismatches are thrown out.

              * 'study' causes study-specific effect estimates to be collated in the

              meta-analysis report.

              * 'report-all' causes variants present in only a single input file to be

              included in the meta-analysis report.

              * 'weighted-z' requests weighted Z-score-based p-values (as computed by the

              Abecasis  Lab's  METAL  software)  in addition to the usual inverse variance-based analysis.  This
              requires P and effective sample size fields.

              * When <b>--extract</b> (without 'range') is present, only variants named in the

       <b>--extract</b> file are considered.

              * Unless 'no-map' is specified, chromosome filters are also respected.

       <b>--fast-epistasis</b> [{boost | joint-effects | no-ueki}] ['case-only']

              [{set-by-set | set-by-all}] ['nop']

       <b>--epistasis</b> [{set-by-set | set-by-all}]

       Scan for epistatic interactions.
              <b>--fast-epistasis</b> inspects 3x3 joint

              genotype count tables and only applies to  case/control  phenotypes,  while  <b>--epistasis</b>  performs
              linear  or  logistic  regression.  * By default, <b>--fast-epistasis</b> uses the PLINK 1.07 allele-based
              test.  Two

              newer tests are now supported: 'boost' invokes the likelihood ratio test introduced by  Wan  X  et
              al.  (2010) BOOST: A Fast Approach to Detecting Gene-Gene Interactions in Genome-wide Case-Control
              Studies, while 'joint-effects' applies the joint effects test introduced in  Ueki  M,  Cordell  HJ
              (2012) Improved statistics for genome-wide interaction analysis.

              * The original <b>--fast-epistasis</b> test normally applies the variance and

       empty cell corrections suggested by Ueki and Cordell's paper.
              To disable

              them, use the 'no-ueki' modifier.

              *  'case-only'  requests  a  case-only instead of a case/control test.  * By default, all pairs of
              variants across the entire genome are tested.

              To just test pairs of variants within a single set, add the 'set-by-set' modifier and load exactly
              one set with <b>--set</b>/--make-set; with exactly two sets loaded, all variants in one  set  are  tested
              against  all variants in the other.  'set-by-all' tests all variants in one set against the entire
              genome instead.

              * 'nop' strips p-values from the main  report.   *  These  computations  can  be  subdivided  with
              <b>--parallel</b>; however...

       <b>--epistasis-summary-merge</b> &lt;common file prefix&gt; &lt;ct&gt;

              When  a  <b>--[fast-]epistasis</b> job is subdivided with <b>--parallel</b>, the main report can be assembled at
              the end by applying Unix 'cat' in the usual manner, but the .summary.1, .summary.2, ... files  may
              require a specialized merge.  <b>--epistasis-summary-merge</b> takes care of the latter.

       <b>--twolocus</b> &lt;variant ID&gt; &lt;variant ID&gt;

              Two-locus joint genotype count report.

       <b>--score</b> &lt;filename&gt; [i] [j] [k] ['header'] [{sum | no-sum}]

              [{no-mean-imputation | center}] ['include-cnt'] ['double-dosage']

              Apply  a  linear  scoring  system  to each sample.  The input file should have one line per scored
              variant.  Variant IDs are read from column #i, allele codes are read from column  #j,  and  scores
              are  read  from column #k, where i defaults to 1, j defaults to i+1, and k defaults to j+1.  * The
              'header' modifier causes the first nonempty line of the input file to

              be ignored; otherwise, <b>--score</b> assumes there is no header line.

              * By default, final scores are averages of the valid per-variant scores.

       The 'sum' modifier causes sums to be reported instead.
              (This cannot be

       used with 'no-mean-imputation'.
              And for backward compatibility, 'sum' is

              automatically on with dosage data unless 'no-sum' is specified.)

              * By default, copies of the unnamed allele contribute zero to score, while

              missing genotypes contribute an amount proportional to the loaded  (via  <b>--read-freq</b>)  or  imputed
              allele  frequency.   To  throw out missing observations instead (decreasing the denominator in the
              final average when this happens), use the 'no-mean-imputation' modifier.

              * Alternatively, you can use the 'center' modifier to shift all scores to

              mean zero.

       * This command can be used with dosage data.
              By default, the 'CNT' column

              is omitted from the output file in this case; use 'include-cnt'  to  keep  it.   Also,  note  that
              scores  are multiplied by 0..1 dosages, not 0..2 diploid allele counts, unless the 'double-dosage'
              modifier is present.

       <b>--R</b> &lt;R script file&gt; ['debug']

              Connect to a Rserve (preferably version 1.7 or later) background process, and execute  the  Rplink
              function  defined in the input file.  (Unless the 'debug' modifier is present; in that case, the R
              commands that PLINK would have tried to execute are logged to a file.)

       <b>--write-var-ranges</b> &lt;block ct&gt;

       Divide the set of variants into equal-size blocks.
              (Can be used with

       <b>--snps</b> to split a job across multiple machines.)

       The   following   other   flags   are   supported.    (Order    of    operations    is    described    at
       https://www.cog-genomics.org/plink/1.9/order .)

       <b>--script</b> &lt;fname&gt; : Include command-line options from file.

       <b>--rerun</b> [log]
              : Rerun commands in log (default 'plink.log').

       <b>--version</b>
              : Display only version number before exiting.

       <b>--silent</b>
              : Suppress output to console.

       <b>--gplink</b>
              : Reserved for interoperation with gPLINK.

       <b>--missing-genotype</b> &lt;char&gt; : Set missing genotype code (normally '0').

       <b>--double-id</b>
              : Set both FIDs and IIDs to the VCF/BCF sample ID.

       <b>--const-fid</b> [ID]
              : Set all FIDs to the given constant (default '0').

       <b>--id-delim</b> [d]
              : Parse sample IDs as &lt;FID&gt;&lt;d&gt;&lt;IID&gt; (default delim '_').

       <b>--vcf-idspace-to</b> &lt;c&gt; : Convert spaces in sample IDs to the given character.

       <b>--biallelic-only</b> ['strict'] ['list'] : Skip VCF variants with 2+ ALT alleles.

       <b>--vcf-min-qual</b> &lt;val&gt;
              : Skip VCF variants with low/missing QUAL.

       <b>--vcf-filter</b> [exception(s)...] : Skip variants which have FILTER failures.

       <b>--vcf-require-gt</b>
              : Skip variants with no GT field.

       <b>--vcf-min-gq</b> &lt;val&gt;
              : No-call a genotype when GQ is below the given threshold.

       <b>--vcf-min-gp</b> &lt;val&gt;
              : No-call a genotype when 0-1 scaled GP is below the given threshold.

       <b>--vcf-half-call</b> &lt;m&gt;
              :  Specify  how  '0/.'  and similar VCF GT values should be handled.  The following four modes are
              supported: * 'error'/'e' (default) errors out and reports line #.  * 'haploid'/'h' treats them  as
              haploid  calls.   *  'missing'/'m'  treats  them as missing.  * 'reference'/'r' treats the missing
              value as 0.

       <b>--oxford-single-chr</b> &lt;chr nm&gt; : Specify single-chromosome .gen file with
              ignorable first column.

       <b>--oxford-pheno-name</b> &lt;col nm&gt; : Import named phenotype from the .sample file.

       <b>--hard-call-threshold</b> &lt;val&gt;
              : When an Oxford-format fileset is loaded, calls

       <b>--hard-call-threshold</b> random
              with uncertainty level greater than 0.1 are normally treated as  missing.   You  can  adjust  this
              threshold by providing a numeric parameter, or randomize all calls with 'random'.

       <b>--missing-code</b> [string list] : Comma-delimited list of missing phenotype

       (alias: <b>--missing_code</b>)
              values for Oxford-format filesets (def. 'NA').

       <b>--simulate-ncases</b> &lt;num&gt;
              : Set <b>--simulate</b> case count (default 1000).

       <b>--simulate-ncontrols</b> &lt;n&gt;
              : Set <b>--simulate</b> control count (default 1000).

       <b>--simulate-prevalence</b> &lt;p&gt; : Set <b>--simulate</b> disease prevalence (default 0.01).

       <b>--simulate-n</b> &lt;num&gt;
              : Set <b>--simulate-qt</b> sample count (default 1000).

       <b>--simulate-label</b> &lt;prefix&gt; : Set <b>--simulate[-qt]</b> FID/IID name prefix.

       <b>--simulate-missing</b> &lt;freq&gt; : Set <b>--simulate[-qt]</b> missing genotype frequency.

       <b>--allow-extra-chr</b> ['0']
              : Permit unrecognized chromosome codes.  The '0'

       (alias: <b>--aec</b>)
              modifier causes them to be treated as if they had been set to zero.

       <b>--chr-set</b> &lt;autosome ct&gt; ['no-x'] ['no-y'] ['no-xy'] ['no-mt'] :

       Specify a nonhuman chromosome set.
              The first parameter sets the number of

              diploid  autosome pairs if positive, or haploid chromosomes if negative.  Given diploid autosomes,
              the remaining modifiers indicate the absence of the named non-autosomal chromosomes.

       <b>--cow</b>/--dog/--horse/--mouse/--rice/--sheep : Shortcuts for those species.

       <b>--autosome-num</b> &lt;value&gt;
              : Alias for "--chr-set &lt;value&gt; no-y no-xy no-mt".

       <b>--cm-map</b> &lt;fname pattern&gt; [chr] : Use SHAPEIT-format recombination maps to set
              centimorgan positions.  To process more than one chromosome, include a '@' in the first  parameter
              where the chrom. number belongs, e.g. '<a href="mailto:genetic_map_chr@_combined_b37.txt">genetic_map_chr@_combined_b37.txt</a>'.

       <b>--zero-cms</b>
              : Zero out centimorgan positions.

       <b>--allow-no-samples</b> : Allow the input fileset to contain no samples.

       <b>--allow-no-vars</b>
              : Allow the input fileset to contain no variants.

       <b>--pheno</b> &lt;fname&gt;
              :  Load  phenotype  data  from  the  specified file, instead of using the values in the main input
              fileset.

       <b>--all-pheno</b>
              : For basic association tests, loop through all phenotypes in <b>--pheno</b> file.

       <b>--mpheno</b> &lt;n&gt;
              : Load phenotype from column (n+2) in <b>--pheno</b> file.

       <b>--pheno-name</b> &lt;c&gt; : If <b>--pheno</b> file has a header row, use column with the
              given name.

       <b>--pheno-merge</b>
              : When the main input fileset contains an phenotype value for a sample, but the <b>--pheno</b> file  does
              not, use the original value instead of treating the phenotype as missing.

       <b>--missing-phenotype</b> &lt;v&gt; : Set missing phenotype value (normally <b>-9</b>).

       <b>--1</b>    :  Expect  case/control  phenotypes to be coded as 0 = control, 1 = case, instead of the usual 0 =
              missing, 1 = control, 2 = case.  This also forces phenotypes to be interpreted as case/ctrl.

       <b>--make-pheno</b> &lt;fn&gt; &lt;val&gt; : Define a new case/control phenotype.
              If the val parameter is '*', all samples listed in the given file are cases, and everyone else  is
              a  control.   (Note  that,  in  some  shells,  it  is  necessary  to  surround the * with quotes.)
              Otherwise, all samples with third column entry equal to the val parameter are cases, and all other
              samples mentioned in the file are controls.

       <b>--tail-pheno</b> &lt;Lt&gt; [Hbt] : Downcode a scalar phenotype to a case/control
              phenotype.  All samples with phenotype values greater than Hbt are cases, and all with values less
              than or equal to Lt are controls.  If Hbt is unspecified, it is equal to Lt; otherwise, in-between
              phenotype values are set to missing.

       <b>--covar</b> &lt;filename&gt; ['keep-pheno-on-missing-cov'] : Specify covariate file.

       <b>--covar-name</b> &lt;...&gt;
              : Specify covariate(s) in <b>--covar</b> file by name.  Separate multiple names with  spaces  or  commas,
              and use dashes to designate ranges.

       <b>--covar-number</b> &lt;...&gt;
              : Specify covariate(s) in <b>--covar</b> file by index.

       <b>--no-const-covar</b>
              : Exclude constant covariates.

       <b>--allow-no-covars</b>
              : Allow no covariates to be loaded from <b>--covar</b> file.

       <b>--within</b> &lt;f&gt; ['keep-NA'] : Specify initial cluster assignments.

       <b>--mwithin</b> &lt;n&gt;
              : Load cluster assignments from column n+2.

       <b>--family</b>
              : Create a cluster for each family ID.

       <b>--loop-assoc</b> &lt;f&gt; ['keep-NA']
              : Run specified case/control association commands once for each cluster in the file, using cluster
              membership as the phenotype.

       <b>--set</b> &lt;filename&gt;
              : Load sets from a .set file.

       <b>--set-names</b> &lt;name(s)...&gt;
              : Load only sets named on the command line.  Use spaces to separate multiple names.

       <b>--subset</b> &lt;filename&gt;
              : Load only sets named in the given text file.

       <b>--set-collapse-all</b> &lt;set name&gt; : Merge all sets.

       <b>--complement-sets</b>
              : Invert all sets.  (Names gain 'C_' prefixes.)

       <b>--make-set-complement-all</b> &lt;s&gt; : <b>--set-collapse-all</b> + inversion.

       <b>--make-set</b> &lt;filename&gt;
              : Define sets from a list of named bp ranges.

       <b>--make-set-border</b> &lt;kbs&gt;
              : Stretch regions in <b>--make-set</b> file.

       <b>--make-set-collapse-group</b>
              : Define sets from groups instead of sets in <b>--make-set</b> file.

       <b>--keep</b> &lt;filename&gt;
              : Exclude all samples not named in the file.

       <b>--remove</b> &lt;filename&gt;
              : Exclude all samples named in the file.

       <b>--keep-fam</b> &lt;filename&gt;
              : Exclude all families not named in the file.

       <b>--remove-fam</b> &lt;filename&gt; : Exclude all families named in the file.

       <b>--extract</b> ['range'] &lt;f&gt; : Exclude all variants not named in the file.

       <b>--exclude</b> ['range'] &lt;f&gt; : Exclude all variants named in the file.

       <b>--keep-clusters</b> &lt;filename&gt;
              : These can be used individually or in

       <b>--keep-cluster-names</b> &lt;name(s)...&gt;
              combination  to  define  a list of clusters to keep; all samples not in a cluster in that list are
              then excluded.  Use spaces to separate cluster names for <b>--keep-cluster-names</b>.

       <b>--remove-clusters</b> &lt;filename&gt;
              : Exclude all clusters named in the file.

       <b>--remove-cluster-names</b> &lt;name(s)...&gt; : Exclude the named clusters.

       <b>--gene</b> &lt;sets...&gt; : Exclude variants not in a set named on the command line.
              (Separate multiple set names with spaces.)

       <b>--gene-all</b>
              : Exclude variants which aren't a member of any set.  (PLINK 1.07  automatically  did  this  under
              some circumstances.)

       <b>--attrib</b> &lt;f&gt; [att lst] : Given a file assigning attributes to variants, and a

       <b>--attrib-indiv</b> &lt;f&gt; [a]
              comma-delimited  list  (with  no whitespace) of attribute names, remove variants/samples which are
              either missing from the file or don't have any of the listed attributes.  If some attribute  names
              in the list are preceded by '-', they are treated as "negative match conditions" instead: variants
              with at least one negative match attribute are removed.  The first character in the list cannot be
              a '-', due to how command-line parsing works; add a comma in front to get around this.

       <b>--chr</b> &lt;chrs...&gt;
              : Exclude all variants not on the given chromosome(s).  Valid choices for humans are 0 (unplaced),
              1-22,  X,  Y, XY, and MT.  Separate multiple chromosomes with spaces and/or commas, and use a dash
              (no adjacent spaces permitted) to denote a range, e.g. "--chr 1-4, 22, xy".

       <b>--not-chr</b> &lt;...&gt;
              : Reverse of <b>--chr</b> (exclude variants on listed chromosomes).

       <b>--autosome</b>
              : Exclude all non-autosomal variants.

       <b>--autosome-xy</b>
              : Exclude all non-autosomal variants, except  those  with  chromosome  code  XY  (pseudo-autosomal
              region of X).

       <b>--snps-only</b> ['just-acgt'] : Exclude non-SNP variants.
              By  default,  SNP  =  both  allele  codes  are  single-character;  'just-acgt'  restricts codes to
              {A,C,G,T,a,c,g,t,&lt;missing&gt;}.

       <b>--from</b> &lt;var ID&gt;
              : Use ID(s) to specify a variant range to load.  When used

       <b>--to</b>   &lt;var ID&gt;    together, both variants must be on the same chromosome.

       <b>--snp</b>  &lt;var ID&gt;  : Specify a single variant to load.

       <b>--exclude-snp</b> &lt;&gt; : Specify a single variant to exclude.

       <b>--window</b>
              &lt;kbs&gt;  : With <b>--snp</b> or <b>--exclude-snp</b>, loads/excludes all variants within  half  the  specified  kb
              distance of the named one.

       <b>--from-bp</b> &lt;pos&gt;
              : Use physical position(s) to define a variant range to

       <b>--to-bp</b>
              &lt;pos&gt;    load.  <b>--from-kb</b>/--to-kb/--from-mb/--to-mb allow decimal

       <b>--from-kb</b> &lt;pos&gt;
              values.  You must also specify a single chromosome (using

       <b>--to-kb</b>
              &lt;pos&gt;    e.g. <b>--chr</b>) when using these flags.

       <b>--from-mb</b> &lt;pos&gt;

       <b>--to-mb</b>
              &lt;pos&gt;

       <b>--snps</b> &lt;var IDs...&gt;
              : Use IDs to specify variant range(s) to load or

       <b>--exclude-snps</b> &lt;...&gt;
              exclude.  E.g. "--snps rs1111-rs2222, rs3333, rs4444".

       <b>--thin</b> &lt;p&gt;
              : Randomly remove variants, retaining each with prob. p.

       <b>--thin-count</b> &lt;n&gt; : Randomly remove variants until n of them remain.

       <b>--bp-space</b> &lt;bps&gt; : Remove variants so that each pair is no closer than the
              given bp distance.  (Equivalent to VCFtools <b>--thin</b>.)

       <b>--thin-indiv</b> &lt;p&gt;
              : Randomly remove samples, retaining with prob. p.

       <b>--thin-indiv-count</b> &lt;n&gt;
              : Randomly remove samples until n of them remain.

       <b>--filter</b> &lt;f&gt; &lt;val(s)...&gt; : Exclude all samples without a 3rd column entry in
              the given file matching one of the given space-separated value(s).

       <b>--mfilter</b> &lt;n&gt;
              : Match against (n+2)th column instead.

       <b>--geno</b> [val]
              :  Exclude  variants  with missing call frequencies greater than a threshold (default 0.1).  (Note
              that the default threshold is only applied if <b>--geno</b> is invoked without a parameter;  when  <b>--geno</b>
              is  not  invoked,  no  per-variant  missing  call  frequency  ceiling  is  enforced at all.  Other
              inclusion/exclusion default thresholds work the same way.)

       <b>--mind</b> [val]
              : Exclude samples with missing call frequencies greater than a threshold (default 0.1).

       <b>--oblig-missing</b> &lt;f1&gt; &lt;f2&gt; : Specify blocks of missing genotype calls for
              <b>--geno</b>/--mind to ignore.  The first file should have variant IDs in the first column and block IDs
              in the second, while the second file should have FIDs in the first column, IIDs in the second, and
              block IDs in the third.

       <b>--prune</b>
              : Remove samples with missing phenotypes.

       <b>--maf</b> [freq]
              : Exclude variants with minor allele frequency lower than a threshold (default 0.01).

       <b>--max-maf</b> &lt;freq&gt;
              : Exclude variants with MAF greater than the threshold.

       <b>--mac</b> &lt;ct&gt;
              : Exclude variants with minor allele count lower than the

       (alias: <b>--min-ac</b>)
              given threshold.

       <b>--max-mac</b> &lt;ct&gt;
              : Exclude variants with minor allele count greater than

       (alias: <b>--max-ac</b>)
              the given threshold.

       <b>--maf-succ</b>
              : Rule of succession MAF estimation (used in EIGENSOFT).  Given j observations of one allele and k
              &gt;= j observations of the other, infer a MAF of (j+1) / (j+k+2), rather than the default j / (j+k).

       <b>--read-freq</b> &lt;fn&gt; : Estimate MAFs and heterozygote frequencies from the given
              <b>--freq[x]</b> report, instead of the input fileset.

       <b>--hwe</b> &lt;p&gt; ['midp'] ['include-nonctrl'] : Exclude variants with Hardy-Weinberg
              equilibrium exact test p-values below a threshold.

       <b>--me</b> &lt;t&gt; &lt;v&gt; ['var-first'] : Filter out trios and variants with Mendel error
              rates exceeding the given thresholds.

       <b>--me-exclude-one</b> [ratio]
              : Make <b>--me</b> exclude only one sample per trio.

       <b>--qual-scores</b> &lt;f&gt; [qcol] [IDcol] [skip] : Filter out variants with
              out-of-range quality scores.  Default range is now [0, \infty ).

       <b>--qual-threshold</b> &lt;min qual score&gt;
              : Set <b>--qual-scores</b> range floor.

       <b>--qual-max-threshold</b> &lt;max qual score&gt;
              : Set <b>--qual-scores</b> range ceiling.

       <b>--allow-no-sex</b>
              : Do  not  treat  ambiguous-sex  samples  as  having  missing  phenotypes  in  analysis  commands.
              (Automatic <u>/w</u> <b>--no-sex</b>.)

       <b>--must-have-sex</b>
              : Force ambiguous-sex phenotypes to missing on <b>--make-bed</b>/--make-just-fam/--recode/--write-covar.

       <b>--filter-cases</b>
              : Include only cases in the current analysis.

       <b>--filter-controls</b>
              : Include only controls.

       <b>--filter-males</b>
              : Include only males.

       <b>--filter-females</b>
              : Include only females.

       <b>--filter-founders</b>
              : Include only founders.

       <b>--filter-nonfounders</b> : Include only nonfounders.

       <b>--nonfounders</b>
              : Include nonfounders in allele freq/HWE calculations.

       <b>--make-founders</b> ['require-2-missing'] ['first'] :

              Clear parental IDs for those with 1+ missing parent(s).

       <b>--recode-allele</b> &lt;fn&gt; : With <b>--recode</b> A/A-transpose/AD, count alleles named in
              the file (otherwise A1 alleles are always counted).

       <b>--output-chr</b> &lt;MT code&gt; : Set chromosome coding scheme in output files by
              providing  the  desired  human  mitochondrial  code.  (Options are '26', 'M', 'MT', '0M', 'chr26',
              'chrM', and 'chrMT'.)

       <b>--output-missing-genotype</b> &lt;ch&gt; : Set the code used to represent missing
              genotypes in output files (normally the <b>--missing-genotype</b> value).

       <b>--output-missing-phenotype</b> &lt;s&gt; : Set the string used to represent missing
              phenotypes in output files (normally the <b>--missing-phenotype</b> value).

       <b>--zero-cluster</b> &lt;f&gt; : In combination with <b>--within</b>/--family, set blocks of
              genotype calls to missing.  The input file should have variant IDs in the first column and cluster
              IDs in the second.  This must now be used with <b>--make-bed</b> and no other output commands.

       <b>--set-hh-missing</b>
              : Cause <b>--make-bed</b> and <b>--recode</b> to set heterozygous haploid genotypes to missing.

       <b>--set-mixed-mt-missing</b> : Cause <b>--make-bed</b> and <b>--recode</b> to set mixed MT
              genotypes to missing.

       <b>--split-x</b> &lt;bp1&gt; &lt;bp2&gt; ['no-fail']

       <b>--split-x</b> &lt;build&gt; ['no-fail'] :

              Changes chromosome code of all chrX variants with bp position  &lt;=  bp1  or  &gt;=  bp2  to  XY.   The
              following  build  codes  are supported as shorthand: * 'b36'/'hg18' = NCBI 36, 2709521/154584237 *
              'b37'/'hg19' = GRCh37, 2699520/154931044 * 'b38'/'hg38' = GRCh38,  2781479/155701383  By  default,
              PLINK  errors  out when no variants would be affected by <b>--split-x</b>; the 'no-fail' modifier (useful
              in scripts) overrides this.

       <b>--merge-x</b> ['no-fail'] : Merge XY chromosome back with X.

       <b>--set-me-missing</b>
              : Cause <b>--make-bed</b> to set Mendel errors to missing.

       <b>--fill-missing-a2</b>
              : Cause <b>--make-bed</b> to replace all missing calls with homozygous A2 calls.

       <b>--set-missing-var-ids</b> &lt;t&gt;
              : Given a template string with a '@' where the chromosome code should go  and  '#'  where  the  bp
              coordinate belongs, <b>--set-missing-var-ids</b> assigns chromosome-and-bp-based IDs to unnamed variants.
              You  may  also use '$1' and '$2' to refer to allele names in the template string, and in fact this
              becomes essential when multiple variants share the same coordinate.

       <b>--new-id-max-allele-len</b> &lt;n&gt; : Specify maximum number of leading characters
              from allele names to include in new variant IDs (default 23).

       <b>--missing-var-code</b> &lt;string&gt; : Change unnamed variant code (default '.').

       <b>--update-chr</b>
              &lt;f&gt; [chrcol] [IDcol]  [skip] : Update variant chromosome codes.

       <b>--update-cm</b>
              &lt;f&gt; [cmcol]  [IDcol]  [skip] : Update centimorgan positions.

       <b>--update-map</b>
              &lt;f&gt; [bpcol]  [IDcol]  [skip] : Update variant bp positions.

       <b>--update-name</b> &lt;f&gt; [newcol] [oldcol] [skip] : Update variant IDs.

       <b>--update-alleles</b> &lt;fname&gt; : Update variant allele codes.

       <b>--allele1234</b> ['multichar'] : Interpret/recode A/C/G/T alleles as 1/2/3/4.
              With 'multichar', converts all A/C/G/Ts in allele names to 1/2/3/4s.

       <b>--alleleACGT</b> ['multichar'] : Reverse of <b>--allele1234</b>.

       <b>--update-ids</b> &lt;f&gt;
              : Update sample IDs.

       <b>--update-parents</b> &lt;f&gt; : Update parental IDs.

       <b>--update-sex</b> &lt;f&gt; [n] : Update sexes.
              Sex (1 or M = male, 2 or F = female, 0 = missing) is loaded from column n+2 (default n is 1).

       <b>--flip</b> &lt;filename&gt;
              : Flip alleles (A&lt;-&gt;T, C&lt;-&gt;G) for SNP IDs in the file.

       <b>--flip-subset</b> &lt;fn&gt;
              : Only apply <b>--flip</b> to samples in <b>--flip-subset</b> file.

       <b>--flip-scan-window</b> &lt;ct+1&gt; : Set <b>--flip-scan</b> max variant ct dist. (def. 10).

       <b>--flip-scan-window-kb</b> &lt;x&gt; : Set <b>--flip-scan</b> max kb distance (default 1000).

       <b>--flip-scan-threshold</b> &lt;x&gt; : Set <b>--flip-scan</b> min correlation (default 0.5).

       <b>--keep-allele-order</b>
              : Keep the allele order defined in the .bim file,

       <b>--real-ref-alleles</b>
              instead of forcing A2 to be the major allele.  <b>--real-ref-alleles</b> also removes 'PR' from the  INFO
              values emitted by <b>--recode</b> vcf{,-fid,-iid}.

       <b>--a1-allele</b> &lt;f&gt; [a1col] [IDcol] [skip] : Force alleles in the file to A1.

       <b>--a2-allele</b> &lt;filename&gt; [a2col] [IDcol] [skip] :

       Force alleles in the file to A2.
              ("--a2-allele &lt;VCF filename&gt; 4 3 '#'",

              which scrapes reference allele assignments from a VCF file, is especially useful.)

       <b>--indiv-sort</b> &lt;m&gt; [f] : Specify FID/IID sort order.
              The following four modes are supported: * 'none'/'0' keeps samples in the order they were

       loaded.
              Default for non-merge operations.

       * 'natural'/'n' invokes 'natural sort', e.g.
              'id2' &lt; 'ID3' &lt; 'id10'.  Default when merging.

       * 'ascii'/'a' sorts in ASCII order, e.g.
              'ID3' &lt; 'id10' &lt; 'id2'.

       * 'file'/'f' uses the order in the given file (named
              in the second parameter).

       For now, only <b>--merge</b>/--bmerge/--merge-list and
              <b>--make-bed</b>/--make-just-fam respect this flag.

       <b>--with-phenotype</b> ['no-parents'] [{no-sex | female-2}] :

              Include more sample info in new .cov file.

       <b>--dummy-coding</b> [N] ['no-round'] : Split categorical variables (n categories,
              2 &lt; n &lt;= N, default N is 49) into n-1 binary dummy variables when writing covariate file.

       <b>--merge-mode</b> &lt;n&gt;
              :  Adjust  <b>--[b]merge</b>/--merge-list behavior based on a numeric code.  1 (default) = ignore missing
              calls, otherwise difference

       -&gt; missing
              2 = only overwrite originally missing calls 3 = only overwrite when nonmissing in new file  4/5  =
              never  overwrite  and  always  overwrite,  respectively  6  = report all mismatching calls without
              merging 7 = report mismatching nonmissing calls without merging

       <b>--merge-equal-pos</b>
              : With <b>--merge</b>/--bmerge/--merge-list, merge variants with different names but identical positions.
              (Exception: same-position chromosome code 0 variants aren't merged.)

       <b>--mendel-duos</b>
              : Make Mendel error checks consider samples with only one parent in the dataset.

       <b>--mendel-multigen</b>
              : Make Mendel error checks consider (great-)grandparental genotypes when parental genotype data is
              missing.

       <b>--ld-window</b> &lt;ct+1&gt; : Set <b>--r</b>/--r2 max variant ct pairwise distance (usu. 10).

       <b>--ld-window-kb</b> &lt;x&gt; : Set <b>--r</b>/--r2 max kb pairwise distance (usually 1000).

       <b>--ld-window-cm</b> &lt;x&gt; : Set <b>--r</b>/--r2 max centimorgan pairwise distance.

       <b>--ld-window-r2</b> &lt;x&gt; : Set threshold for <b>--r2</b> report inclusion (usually 0.2).

       <b>--ld-snp</b> &lt;var ID&gt;
              : Set first variant in all <b>--r</b>/--r2 pairs.

       <b>--ld-snps</b> &lt;vID...&gt; : Restrict first <b>--r</b>/--r2 variant to the given ranges.

       <b>--ld-snp-list</b> &lt;f&gt;
              : Restrict first <b>--r</b>/--r2 var. to those named in the file.

       <b>--list-all</b>
              : Generate the 'all' mode report when using <b>--show-tags</b> in file mode.

       <b>--tag-kb</b> &lt;kbs&gt;
              : Set <b>--show-tags</b> max tag kb distance (default 250).

       <b>--tag-r2</b> &lt;val&gt;
              : Set <b>--show-tags</b> min tag r-squared (default 0.8)

       <b>--tag-mode2</b>
              : Use two-column <b>--show-tags</b> (file mode) I/O format.

       <b>--ld-xchr</b> &lt;code&gt;
              : Set chrX model for <b>--indep[-pairwise]</b>, <b>--r</b>/--r2, <b>--flip-scan</b>, and <b>--show-tags</b>.   1  (default)  =
              males  coded  0/1,  females 0/1/2 (A1 dosage) 2 = males coded 0/2 3 = males coded 0/2, but females
              given double weighting

       <b>--blocks-max-kb</b> &lt;kbs&gt;
              : Set <b>--blocks</b> maximum haploblock span (def. 200).

       <b>--blocks-min-maf</b> &lt;cutoff&gt;
              : Adjust <b>--blocks</b> MAF minimum (default 0.05).

       <b>--blocks-strong-lowci</b> &lt;x&gt;
              : Set <b>--blocks</b> "strong LD" CI thresholds (defaults

       <b>--blocks-strong-highci</b> &lt;x&gt;
              0.70 and 0.98).

       <b>--blocks-recomb-highci</b> &lt;x&gt; : Set 'recombination' CI threshold (default 0.90).

       <b>--blocks-inform-frac</b> &lt;x&gt;
              : Force haploblock &lt;strong LD pairs&gt;:&lt;total informative pairs&gt; ratios to be larger than this value
              (default 0.95).

       <b>--distance-wts</b> exp=&lt;x&gt;
              : When computing genomic distances, assign each variant a weight of (2q(1-q))^{-x}, where q is the
              loaded or inferred MAF.

       <b>--read-dists</b> &lt;dist file&gt; [id file] : Load a triangular binary distance matrix
              instead of recalculating from scratch.

       <b>--ppc-gap</b> &lt;val&gt;
              : Minimum number of base pairs, in  thousands,  between  informative  pairs  of  markers  used  in
              <b>--genome</b> PPC test.  500 if unspecified.

       <b>--min</b> &lt;cutoff&gt;
              : Specify minimum PI_HAT for inclusion in <b>--genome</b> report.

       <b>--max</b> &lt;cutoff&gt;
              : Specify maximum PI_HAT for inclusion in <b>--genome</b> report.

       <b>--homozyg-match</b> &lt;&gt; : Set minimum concordance across jointly homozygous
              variants for a pairwise allelic match to be declared.

       <b>--pool-size</b> &lt;ct&gt;
              : Set minimum size of pools in "--homozyg group" report.

       <b>--read-genome</b> &lt;fn&gt; : Load <b>--genome</b> report for <b>--cluster</b>/--neighbour, instead
              of recalculating IBS and PPC test p-values from scratch.

       <b>--ppc</b> &lt;p-val&gt;
              : Specify minimum PPC test p-value within a cluster.

       <b>--mc</b> &lt;max size&gt;
              : Specify maximum cluster size.

       <b>--mcc</b> &lt;c1&gt; &lt;c2&gt;
              : Specify maximum case and control counts per cluster.

       <b>--K</b> &lt;min count&gt;
              : Specify minimum cluster count.

       <b>--ibm</b> &lt;val&gt;
              : Specify minimum identity-by-missingness.

       <b>--match</b> &lt;f&gt; [mv] : Use covariate values to restrict clustering.
              Without  <b>--match-type</b>,  two  samples can only be in the same cluster if all covariates match.  The
              optional second parameter specifies a covariate value to treat as missing.

       <b>--match-type</b> &lt;f&gt; : Refine interpretation of <b>--match</b> file.
              The <b>--match-type</b> file is expected to be a single line with as many entries as the <b>--match</b> file has
              covariates; '0' entries specify "negative matches"  (i.e.  samples  with  equal  covariate  values
              cannot  be in the same cluster), '1' entries specify "positive matches" (default), and '-1' causes
              the corresponding covariate to be ignored.

       <b>--qmatch</b> &lt;f&gt; [m] : Force all members of a cluster to have similar

       <b>--qt</b> &lt;fname&gt;
              quantitative covariate values.  The <b>--qmatch</b> file contains the covariate values,  while  the  <b>--qt</b>
              file is a list of nonnegative tolerances (and '-1's marking covariates to skip).

       <b>--pca-cluster-names</b> &lt;...&gt; : These can be used individually or in combination

       <b>--pca-clusters</b> &lt;fname&gt;
              to  define a list of clusters to use in the basic <b>--pca</b> computation.  (<b>--pca-cluster-names</b> expects
              a space-delimited sequence of cluster names, while <b>--pca-clusters</b> expects a file with one  cluster
              name  per  line.)   All samples outside those clusters will then be projected on to the calculated
              PCs.

       <b>--mds-plot</b> &lt;dims&gt; ['by-cluster'] ['eigendecomp'] ['eigvals'] :

       Multidimensional scaling analysis.
              Requires <b>--cluster</b>.

       <b>--cell</b> &lt;thresh&gt;
              : Skip some <b>--model</b> tests when a contingency table entry is smaller than the given threshold.

       <b>--condition</b> &lt;var ID&gt; [{dominant | recessive}] : Add one variant as a <b>--linear</b>
              or <b>--logistic</b> covariate.

       <b>--condition-list</b> &lt;f&gt; [{dominant | recessive}] : Add variants named in the
              file as <b>--linear</b>/--logistic covariates.

       <b>--parameters</b> &lt;...&gt;
              : Include only the given covariates/interactions in the <b>--linear</b>/--logistic models, identified  by
              a list of 1-based indices and/or ranges of them.

       <b>--tests</b> &lt;all&gt; [...] : Perform a (joint) test on the specified term(s) in the
              <b>--linear</b>/--logistic  model,  identified  by 1-based indices and/or ranges of them.  If permutation
              was requested, it is based on this test.  * Note that, when <b>--parameters</b> is also present, the

       indices refer to the terms remaining AFTER pruning by
              <b>--parameters</b>.

              * You can use "--tests all" to include all terms.

       <b>--vif</b> &lt;max VIF&gt;
              : Set VIF threshold for <b>--linear</b> multicollinearity check (default 50).

       <b>--xchr-model</b> &lt;code&gt; : Set the X chromosome <b>--linear</b>/--logistic model.
              0 = skip sex and haploid chromosomes 1 (default) = add sex as a covariate on X chromosome 2 = code
              male genotypes 0/2 instead of 0/1 3 = test for interaction between genotype and sex

       <b>--lasso-select-covars</b> [cov(s)...] : Subject some or all covariates to LASSO
              model selection.

       <b>--adjust</b> ['gc'] ['log10'] ['qq-plot'] : Report some multiple-testing
              corrections.

       <b>--lambda</b> &lt;val&gt;
              : Set genomic control lambda for <b>--adjust</b>.

       <b>--ci</b> &lt;size&gt;
              : Report confidence intervals for odds ratios.

       <b>--pfilter</b> &lt;val&gt;
              : Filter out association test results with higher p-values.

       <b>--aperm</b> &lt;min perms - 1&gt; [max perms] [alpha] [beta] [init interval] [slope] :

              Set up to six parameters controlling adaptive permutation tests.  *  The  first  two  control  the
              minimum and maximum number of permutations that

              may be run for each variant; default values are 5 and 1000000.

       * The next two control the early termination condition.
              A

              100%  * (1 - beta/2T) confidence interval is calculated for each empirical p-value, where T is the
              total number of variants; whenever this confidence interval doesn't contain alpha, the variant  is
              exempted from further permutation testing.  Default values are 0 and 1e-4.

       * The last two control when the early termination condition is checked.
              If

              a  check  occurs  at  permutation  #p, the next check occurs after &lt;slope&gt;p + &lt;init interval&gt; more
              permutations (rounded down).  Default initial interval is 1, and default slope is 0.001.

       <b>--mperm-save</b>
              : Save best max(T) permutation test statistics.

       <b>--mperm-save-all</b> : Save all max(T) permutation test statistics.

       <b>--set-p</b> &lt;p-val&gt;
              : Adjust set test significant variant p-value ceiling (default 0.05).

       <b>--set-r2</b> [v] ['write'] : Adjust set test significant variant pairwise r^2
              ceiling (default 0.5).  'write' causes violating

              pairs to be dumped to &lt;output prefix&gt;.ldset.

       <b>--set-max</b> &lt;ct&gt;
              : Adjust set test maximum # of significant variants considered per set (default 5).

       <b>--set-test-lambda</b> &lt;v&gt;
              : Specify genomic control correction for set test.

       <b>--border</b> &lt;kbs&gt;
              : Extend <b>--annotate</b> range intervals by given # kbs.

       <b>--annotate-snp-field</b> &lt;nm&gt; : Set <b>--annotate</b> variant ID field name.

       <b>--clump-p1</b> &lt;pval&gt; : Set <b>--clump</b> index var. p-value ceiling (default 1e-4).

       <b>--clump-p2</b> &lt;pval&gt; : Set <b>--clump</b> secondary p-value threshold (default 0.01).

       <b>--clump-r2</b> &lt;r^2&gt;
              : Set <b>--clump</b> r^2 threshold (default 0.5).

       <b>--clump-kb</b> &lt;kbs&gt;
              : Set <b>--clump</b> kb radius (default 250).

       <b>--clump-snp-field</b> &lt;n...&gt;
              : Set <b>--clump</b> variant ID field name (default 'SNP').  With multiple  field  names,  earlier  names
              take precedence over later ones.

       <b>--clump-field</b> &lt;name...&gt;
              : Set <b>--clump</b> p-value field name (default 'P').

       <b>--clump-allow-overlap</b>
              : Let <b>--clump</b> non-index vars. join multiple clumps.

       <b>--clump-verbose</b>
              : Request extended <b>--clump</b> report.

       <b>--clump-annotate</b> &lt;hdr...&gt; : Include named extra fields in <b>--clump-verbose</b> and
              <b>--clump-best</b> reports.  (Field names can be separated with spaces or commas.)

       <b>--clump-range</b> &lt;filename&gt;
              : Report overlaps between clumps and regions.

       <b>--clump-range-border</b> &lt;kb&gt; : Stretch regions in <b>--clump-range</b> file.

       <b>--clump-index-first</b>
              : Extract <b>--clump</b> index vars. from only first file.

       <b>--clump-replicate</b>
              : Exclude clumps which contain secondary results from only one file.

       <b>--clump-best</b>
              : Report best proxy for each <b>--clump</b> index var.

       <b>--meta-analysis-chr-field</b> &lt;n...&gt; : Set <b>--meta-analysis</b> chromosome, variant

       <b>--meta-analysis-snp-field</b> &lt;n...&gt;
              ID, position, A1/A2 allele, p-value,

       <b>--meta-analysis-bp-field</b> &lt;n...&gt;
              standard error, and/or effective sample

       <b>--meta-analysis-a1-field</b> &lt;n...&gt;
              size field names.

       <b>--meta-analysis-a2-field</b> &lt;n...&gt;
              Defaults are 'CHR', 'SNP', 'BP', 'A1',

       <b>--meta-analysis-p-field</b> &lt;n...&gt;
              'A2', 'P', 'SE', and 'NMISS',

       <b>--meta-analysis-se-field</b> &lt;n...&gt;
              respectively.  When multiple parameters

       <b>--meta-analysis-ess-field</b> &lt;n...&gt;
              are  given  to  these  flags,  earlier  names  take precedence over later ones.  Note that, if the
              numbers of cases and controls are unequal, effective sample size should be

              4 / (1/&lt;# cases&gt; + 1/&lt;# controls&gt;).

       <b>--meta-analysis-report-dups</b>
              : When a variant appears multiple times in in the same file, report that.

       <b>--gene-list-border</b> &lt;kbs&gt;
              : Extend <b>--gene-report</b> regions by given # of kbs.

       <b>--gene-subset</b> &lt;filename&gt;
              : Specify gene name subset for <b>--gene-report</b>.

       <b>--gene-report-snp-field</b> &lt;&gt; : Set <b>--gene-report</b> variant ID field name (default
              'SNP').  Only relevant with <b>--extract</b>.

       <b>--gap</b> &lt;kbs&gt;
              : Set "--fast-epistasis case-only" min. gap (default 1000).

       <b>--epi1</b> &lt;p-value&gt; : Set <b>--[fast-]epistasis</b> reporting threshold (default
              5e-6 for 'boost', 1e-4 otherwise).

       <b>--epi2</b> &lt;p-value&gt; : Set threshold for contributing to SIG_E count (def. 0.01).

       <b>--je-cellmin</b> &lt;n&gt; : Set required number of observations per 3x3x2 contingency
              table cell for joint-effects test (default 5).

       <b>--q-score-range</b> &lt;range file&gt; &lt;data file&gt; [i] [j] ['header'] :

              Apply <b>--score</b> to subset(s) of variants in the primary score list based on e.g. p-value ranges.   *
              The first file should have range labels in the first column, p-value

              lower  bounds  in  the  second  column,  and upper bounds in the third column.  Lines with too few
              entries, or nonnumeric values in the second or third column, are ignored.

              * The second file should contain a variant ID and a p-value on each

       nonempty line (except possibly the first).
              Variant IDs are read from

              column #i and p-values are read from column #j, where i defaults to 1 and j defaults to i+1.   The
              'header' modifier causes the first nonempty line of this file to be skipped.

       <b>--R-port</b> &lt;port #&gt;
              : Connect to Rserve on a port other than 6311.

       <b>--R-host</b> &lt;host&gt;
              : Connect to Rserve host.

       <b>--R-socket</b> &lt;sock&gt;
              : Connect to Rserve socket.

       <b>--parallel</b> &lt;k&gt; &lt;n&gt; : Divide the output matrix into n pieces, and only compute
              the  kth  piece.   The  primary  output file will have the piece number included in its name, e.g.
              plink.rel.13 or plink.rel.13.gz if k is 13.  Concatenating these files in  order  will  yield  the
              full matrix of interest.  (Yes, this can be done before unzipping.)  N.B. This generally cannot be
              used  to  directly write a symmetric square matrix.  Choose square0 or triangle shape instead, and
              postprocess as necessary.

       <b>--memory</b> &lt;val&gt;
              : Set size, in MB, of initial workspace malloc attempt.  (Practically  mandatory  when  using  GNU
              parallel.)

       <b>--threads</b> &lt;val&gt;
              :  Set  maximum  number  of  concurrent  threads.  This has one known limitation: some BLAS/LAPACK
              linear algebra operations are multithreaded in a way  that  PLINK  cannot  control.   If  this  is
              problematic, you should recompile against single-threaded BLAS/LAPACK.

       <b>--d</b> &lt;char&gt;
              : Change variant/covariate range delimiter (normally '-').

       <b>--seed</b> &lt;val...&gt;
              : Set random number seed(s).  Each value must be an integer between 0 and 4294967295 inclusive.

       <b>--perm-batch-size</b> &lt;val&gt; : Set number of permutations per batch for some
              permutation tests.

       <b>--output-min-p</b> &lt;p&gt; : Specify minimum p-value to write to reports.

       <b>--debug</b>
              : Use slower, more crash-resistant logging method.

       Primary  methods  paper:  Chang  CC,  Chow  CC,  Tellier  LCAM,  Vattikuti  S,  Purcell SM, Lee JJ (2015)
       Second-generation PLINK: rising to the challenge of larger and richer datasets.  GigaScience, 4.

       For further documentation and support, consult the main webpage (https://www.cog-genomics.org/plink/1.9 )
       and/or the mailing list (https://groups.google.com/d/forum/plink2-users ).

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       The full documentation for <b>PLINK</b> <b>v1.90b6.16</b> <b>64-bit</b> <b>(19</b> <b>Feb</b> is maintained as a  Texinfo  manual.   If  the
       <b>info</b> and <b>PLINK</b> <b>v1.90b6.16</b> <b>64-bit</b> <b>(19</b> <b>Feb</b> programs are properly installed at your site, the command

              <b>info</b> <b>PLINK</b> <b>v1.90b6.16</b> <b>64-bit</b> <b>(19</b> <b>Feb</b>

       should give you access to the complete manual.

PLINK v1.90b6.16 64-bit (19 Feb 2020)              March 2020                                           <u><a href="../man1/PLINK.1.html">PLINK</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>