<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>PLINK - whole genome SNP analysis</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/plink2">plink2_2.00~a5.8-231123+dfsg-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       PLINK - whole genome SNP analysis

</pre><h4><b>DESCRIPTION</b></h4><pre>
       PLINK  v2.00a3  64-bit  (9  Apr  2020)               www.cog-genomics.org/plink/2.0/  (C) 2005-2020 Shaun
       Purcell, Christopher Chang   GNU General Public License v3

       In the command line flag definitions that follow,

              * &lt;angle brackets&gt; denote a required parameter, where the text between the

              angle brackets describes its nature.

       * ['square brackets + single-quotes'] denotes an optional modifier.
              Use the

              EXACT text in the quotes.

              * [{bar|separated|braced|bracketed|values}] denotes a collection of mutually

       exclusive optional modifiers (again, the exact text must be used).
              When

              there are no outer square brackets, one of the choices must be selected.

              * ['quoted_text='&lt;description of value&gt;] denotes an optional modifier that

              must begin with the quoted text, and be followed by a value with no whitespace  in  between.   '|'
              may also be used here to indicate mutually exclusive options.

              * [square brackets without quotes or braces] denote an optional parameter,

              where the text between the brackets describes its nature.

              * An ellipsis (...) indicates that you may enter multiple arguments of the

              specified type.

              * A "column set descriptor" is either

              1. a comma-separated sequence of column set names; this is interpreted as

              the full list of column sets to include.

              2. a comma-separated sequence of column set names, all preceded by '+' or

              '-'; this is interpreted as a list of changes to the default.

              plink2 &lt;input flag(s)...&gt; [command flag(s)...] [other flag(s)...]  plink2 <b>--help</b> [flag name(s)...]

       Most  PLINK  runs require exactly one main input fileset.  The following flags are available for defining
       its form and location:

       <b>--pfile</b> &lt;prefix&gt; ['vzs']
              : Specify .pgen + .pvar[.zst] + .psam prefix.

       <b>--pgen</b> &lt;filename&gt;
              : Specify full name of .pgen/.bed file.

       <b>--pvar</b> &lt;filename&gt;
              : Specify full name of .pvar/.bim file.

       <b>--psam</b> &lt;filename&gt;
              : Specify full name of .psam/.fam file.

       <b>--bfile</b>
              &lt;prefix&gt; ['vzs'] : Specify .bed + .bim[.zst] + .fam prefix.

       <b>--bpfile</b> &lt;prefix&gt; ['vzs'] : Specify .pgen + .bim[.zst] + .fam prefix.

       <b>--keep-autoconv</b> ['vzs']
              : When importing non-PLINK-binary data, don't delete autogenerated fileset at end of run.

       <b>--no-fid</b>
              : .fam file does not contain column 1 (family ID).

       <b>--no-parents</b>
              : .fam file does not contain columns 3-4 (parents).

       <b>--no-sex</b>
              : .fam file does not contain column 5 (sex).

       <b>--vcf</b> &lt;filename&gt; ['dosage='&lt;field&gt;]

       <b>--bcf</b> &lt;filename&gt; ['dosage='&lt;field&gt;] :

              Specify full name of  .vcf{|.gz|.zst}  or  BCF2  file  to  import.   *  These  can  be  used  with
              <b>--psam</b>/--fam.  * By default, dosage information is not imported.  To import the GP field

              (must  be  VCFv4.3-style  0..1,  one  probability  per  possible  genotype),  add  'dosage=GP' (or
              'dosage=GP-force', see below).  To import Minimac3-style DS+HDS phased dosage,  add  'dosage=HDS'.
              'dosage=DS'  (or  anything  else  for  now)  causes  the  named  field  to  be  interpreted  as  a
              Minimac3-style dosage.  Note that, in the dosage=GP case, PLINK 2 collapses the probabilities down
              to dosages; you cannot use PLINK 2 to losslessly convert VCF FORMAT:GP data to e.g.  BGEN  format.
              To  make  this  more  obvious,  PLINK  2  now  errors  out when dosage=GP is used on a file with a
              FORMAT:DS header line and <b>--import-dosage-certainty</b> wasn't specified, since dosage=DS extracts the
              same  information  more  quickly  in  this  situation.   You  can   suppress   this   error   with
              'dosage=GP-force'.   In  all  of  these  cases,  hardcalls  are  regenerated from scratch from the
              dosages.  As a consequence, variants with no GT field can now be imported; they will be assumed to
              contain only diploid calls when HDS is also absent.

       <b>--data</b> &lt;filename prefix&gt; &lt;REF/ALT mode&gt; ['gzs']

       <b>--bgen</b> &lt;filename&gt; &lt;REF/ALT mode&gt; ['snpid-chr']

       <b>--gen</b> &lt;filename&gt; &lt;REF/ALT mode&gt;

       <b>--sample</b> &lt;filename&gt; :

       Specify an Oxford-format dataset to import.
              <b>--data</b> specifies a .gen[.zst]

              + .sample pair, while <b>--bgen</b> specifies a BGEN v1.1+ file.  * If a BGEN v1.2+ file contains  sample
              IDs, it may be imported without a

              companion .sample file.

              * With 'snpid-chr', chromosome codes are read from the 'SNP ID' field

              instead of the usual chromosome field.

              * The following REF/ALT modes are supported:

              'ref-first':  The  first  allele  for  each  variant is REF.  'ref-last': The last allele for each
              variant is REF.  'ref-unknown': The last allele for each variant is treated as

              provisional-REF.

       <b>--haps</b> &lt;filename&gt; [{ref-first | ref-last}]

       <b>--legend</b> &lt;filename&gt; &lt;chr code&gt; :

              Specify .haps [+ .legend] file(s) to import.  * When <b>--legend</b> is specified, it's assumed that  the
              <b>--haps</b> file doesn't

              contain header columns.

       * On chrX, the second male column may contain dummy '-' entries.
              (However,

              PLINK 2 currently cannot handle omitted male columns.)

              * If not used with <b>--sample</b>, new sample IDs are of the form 'per#/per#'.

       <b>--map</b> &lt;filename&gt;
              : Specify full name of .map file.

       <b>--import-dosage</b> &lt;allele dosage file&gt; ['noheader'] ['id-delim='&lt;char&gt;]

              ['skip0='&lt;i&gt;]  ['skip1='&lt;j&gt;]  ['skip2='&lt;k&gt;]  ['dose1']  ['format='&lt;m&gt;]  [{ref-first  |  ref-last}]
              ['single-chr='&lt;code&gt;] ['chr-col-num='&lt;#&gt;] ['pos-col-num='&lt;#&gt;] :

              Specify PLINK 1.x-style dosage file to import.  * You must also  specify  a  companion  .psam/.fam
              file.  * By default, PLINK assumes that the file contains a header line, which has

              'SNP'  in  (1-based)  column  i+1, 'A1' in column i+j+2, 'A2' in column i+j+3, and sample FID/IIDs
              starting from column i+j+k+4.  (i/j/k are normally zero, but can be changed with 'skip0', 'skip1',
              and 'skip2' respectively.  FID/IID are normally assumed to be  separate  tokens,  but  if  they're
              merged into a single token you can specify the delimiter with 'id-delim='.)  If such a header line
              is  not  present,  use the 'noheader' modifier; samples will then be assumed to appear in the same
              order as they do in the .psam/.fam file.

       * You may specify a companion .map file.
              If you do not,

              * 'single-chr=' can be used to specify that all variants are on the named

       chromosome.
              Otherwise, you can use 'chr-col-num=' to read chromosome

              codes from the given (1-based) column number.

              * 'pos-col-num=' causes bp coordinates to be read from the given column

              number.

              * The 'format=' modifier lets you specify the number of values used to

       represent each dosage.
              'format=1' normally indicates a single 0..2 A1

       expected count; 'dose1' modifies this to a 0..1 frequency.
              'format=2'

              indicates a 0..1 homozygous A1 likelihood followed by a 0..1 het likelihood.  'format=3' indicates
              0..1 hom A1, 0..1 het, 0..1 hom A2.  'format=infer' (the  default)  infers  the  format  from  the
              number of columns in the first nonheader line.

       <b>--dummy</b> &lt;sample ct&gt; &lt;SNP ct&gt; [missing dosage freq] [missing pheno freq]

              [{acgt | 1234 | 12}] ['pheno-ct='&lt;count&gt;] ['scalar-pheno'] ['dosage-freq='&lt;rate&gt;]

              This  generates a fake input dataset with the specified number of samples and SNPs.  * By default,
              the missing dosage and phenotype frequencies are zero.

              These can be changed by providing 3rd and 4th numeric arguments.

              * By default, allele codes are As and Bs; this can be changed with the

              'acgt', '1234', or '12' modifier.

       * By default, one binary phenotype is generated.
              'pheno-ct=' can be used

              to change the number of phenotypes, and 'scalar-pheno' causes  these  phenotypes  to  be  normally
              distributed scalars.

       * By default, all (nonmissing) dosages are in {0,1,2}.
              To make some of

       them take on decimal values, use 'dosage-freq='.
              (These dosages are

              affected by <b>--hard-call-threshold</b> and <b>--dosage-erase-threshold</b>.)

       <b>--fa</b> &lt;filename&gt;
              : Specify full name of reference FASTA file.

       Output  files have names of the form 'plink2.&lt;extension&gt;' by default.  You can change the 'plink2' prefix
       with

       <b>--out</b> &lt;prefix&gt;
              : Specify prefix for output files.

       Most runs also require at least one of the following commands:

       <b>--rm-dup</b> [mode] ['list']

              Remove all but one instance of each duplicate-ID variant (ignoring the missing ID), and (with  the
              'list'  modifier)  write  a  list  of duplicated IDs to &lt;output prefix&gt;.rmdup.list.  The following
              modes of operation are supported: * 'error' (default) causes this to  error  out  when  there's  a
              genotype data

       or other mismatch between the records.
              A list of affected IDs is written

              to &lt;output prefix&gt;.rmdup.mismatch.

              * 'retain-mismatch' causes all instances of a duplicate-ID variant to be

              retained  when  there's  a genotype data or variant info mismatch; otherwise one instance is kept.
              The .rmdup.mismatch file is also written.

              * 'exclude-mismatch' removes all instances of duplicate-ID mismatched

              variants instead.

              * 'exclude-all' causes all instances of duplicate-ID variants to be

              removed, even when the actual records are identical.

              * 'force-first' causes only the first instance of duplicate-ID variants to

              be kept, under all circumstances.

       <b>--make-pgen</b> ['vzs'] ['format='&lt;code&gt;] ['trim-alts'] ['erase-phase']

              ['erase-dosage'] ['pvar-cols='&lt;col set descriptor&gt;] ['psam-cols='&lt;col set descriptor&gt;]

       <b>--make-bpgen</b> ['vzs'] ['format='&lt;code&gt;] ['trim-alts'] ['erase-phase']

              ['erase-dosage']

       <b>--make-bed</b> ['vzs'] ['trim-alts']

              Create a new PLINK 2 binary fileset (<b>--make-pgen</b> = .pgen + .pvar[.zst]  +  .psam,  <b>--make-bpgen</b>  =
              .pgen + .bim[.zst] + .fam).  * Unlike the automatic text-to-binary converters (which only heed

              chromosome filters), this supports all of PLINK's filtering flags.

              * The 'vzs' modifier causes the variant file (.pvar/.bim) to be

              Zstd-compressed.

              * The 'format' modifier requests an uncompressed fixed-variant-width .pgen

       file.
              (These do not directly support multiallelic variants.)  The

              following format code is currently supported:

              2: just like .bed, except with an extended (12-byte instead of 3-byte)

              header  containing variant/sample counts, and rotated genotype codes (00 = hom ref, 01 = het, 10 =
              hom alt, 11 = missing).

              * The 'erase-phase' and 'erase-dosage' modifiers prevent phase and dosage

              information from being written to the new .pgen.

              * The first five columns of a .pvar file are always #CHROM/POS/ID/REF/ALT.

              Supported optional .pvar column sets are:

              xheader: All ## header lines (yeah, this is technically not a column),

              except for possibly FILTER/INFO definitions when those column(s) have been removed.  Without this,
              only the #CHROM header line is kept.

              vcfheader: xheader, with additions to make it a valid VCF header.  maybequal:  QUAL.   Omitted  if
              all  remaining  values  are  missing.   qual:  Force  QUAL  column  to be written even when empty.
              maybefilter: FILTER.  Omitted if all remaining values are missing.  filter: Force FILTER column to
              be written even when empty.  maybeinfo: INFO.  Omitted if all remaining values are missing, or if

              INFO:PR is the only subfield.

              info: Force INFO column to be written.  maybecm: Centimorgan coordinate.  Omitted if all remaining
              values = 0.  cm: Force CM column to be written even when empty.

              The default is xheader,maybequal,maybefilter,maybeinfo,maybecm.

              * Supported column sets for the .psam file are:

       maybefid: Family ID, '0' = missing.
              Omitted if all values missing.

              fid: Force FID column to be written even when empty.  maybesid: Source ID, '0' = missing.  Omitted
              if all values missing.  sid: Force SID column to be written even when empty.  maybeparents: Father
              and mother IIDs.  Omitted if all values missing.  parents: Force PAT and MAT columns to be written
              even when empty.  sex: '1' = male, '2' = female, 'NA' = missing.  pheno1: First active  phenotype.
              If none, all column entries are set to

              the <b>--output-missing-phenotype</b> string.

       phenos: All active phenotypes, if any.
              (Can be combined with pheno1 to

              force at least one phenotype column to be written.)

              The default is maybefid,maybesid,maybeparents,sex,phenos.

       <b>--make-just-pvar</b> ['zs'] ['cols='&lt;column set descriptor&gt;]

       <b>--make-just-psam</b> ['cols='&lt;column set descriptor&gt;]

       <b>--make-just-bim</b> ['zs']

       <b>--make-just-fam</b>

              Variants  of  <b>--make-pgen</b>/--make-bed  which  only write a new variant or sample file.  These don't
              always require an input genotype file.  USE THESE CAUTIOUSLY.  It is very  easy  to  desynchronize
              your  binary  genotype  data and your sample/variant indexes if you use these commands improperly.
              If you have any doubt, stick with <b>--make-[b]pgen</b>/--make-bed.

       <b>--export</b> &lt;output format(s)...&gt; [{01 | 12}] ['bgz'] ['id-delim='&lt;char&gt;]

              ['id-paste='&lt;column    set    descriptor&gt;]    ['include-alt']    ['omit-nonmale-y']     ['spaces']
              ['vcf-dosage='&lt;field&gt;] ['ref-first'] ['bits='&lt;#&gt;]

       Create a new fileset with all filters applied.
              The following output

              formats  are  supported:  (actually,  only  A,  AD,  A-transpose, bcf, bgen-1.x, haps, hapslegend,
              ind-major-bed, oxford, and vcf are implemented for now) * '23': 23andMe 4-column format.  This can
              only be used on a single

              sample's data (<b>--keep</b> may be handy), and does not support multicharacter allele codes.

              * 'A': Sample-major additive (0/1/2) coding, suitable for loading from R.

              If you need uncounted alleles to be named in the header line, add the 'include-alt' modifier.

              * 'AD': Sample-major additive (0/1/2) + dominant (het=1/hom=0) coding.

              Also supports 'include-alt'.

              * 'A-transpose': Variant-major 0/1/2.  * 'beagle': Unphased  per-autosome  .dat  and  .map  files,
              readable by early

              BEAGLE versions.

              *  'beagle-nomap':  Single  .beagle.dat  file.   * 'bgen-1.x': Oxford-format .bgen + .sample.  For
              v1.2/v1.3, sample

              identifiers are stored in the .bgen (with id-delim and id-paste  settings  applied),  and  default
              precision is 16-bit (use the 'bits' modifier to reduce this).

              *  'bimbam':  Regular  BIMBAM  format.  * 'bimbam-1chr': BIMBAM format, with a two-column .pos.txt
              file.  Does not

              support multiple chromosomes.

              * 'fastphase': Per-chromosome fastPHASE files, with

              .chr-&lt;chr #&gt;.phase.inp filename extensions.

       * 'fastphase-1chr': Single .phase.inp file.
              Does not support

              multiple chromosomes.

       * 'haps', 'hapslegend': Oxford-format .haps + .sample[ + .legend].
              All

       data must be biallelic and phased.
              When the 'bgz'

       modifier is present, the .haps file is
              block-gzipped.

              * 'HV': Per-chromosome Haploview files, with .chr-&lt;chr #&gt;{.ped,.info}

              filename extensions.

       * 'HV-1chr': Single Haploview .ped + .info file pair.
              Does not support

              multiple chromosomes.

              * 'ind-major-bed': PLINK 1 sample-major .bed (+ .bim +  .fam).   *  'lgen':  PLINK  1  long-format
              (.lgen  + .fam + .map), loadable with <b>--lfile</b>.  * 'lgen-ref': .lgen + .fam + .map + .ref, loadable
              with <b>--lfile</b> +

              <b>--reference</b>.

       * 'list': Single genotype-based list, up to 4 lines per variant.
              To omit

              nonmale genotypes on the Y chromosome, add the 'omit-nonmale-y' modifier.

              * 'rlist': .rlist + .fam + .map fileset, where the .rlist file is a

              genotype-based list which omits  the  most  common  genotype  for  each  variant.   Also  supports
              'omit-nonmale-y'.

       * 'oxford': Oxford-format .gen + .sample.
              When the 'bgz' modifier is

              present, the .gen file is block-gzipped.

              *  'ped':  PLINK 1 sample-major (.ped + .map), loadable with <b>--file</b>.  * 'compound-genotypes': Same
              as 'ped', except that the space between each

              pair of same-variant allele codes is removed.

              * 'structure': Structure-format.  * 'transpose': PLINK 1 variant-major (.tped +  .tfam),  loadable
              with

              <b>--tfile</b>.

       * 'vcf',
              : VCF (default version 4.3).  If PAR1 and PAR2 are present,

       'vcf-4.2',
              they are automatically merged with chrX, with proper

       'bcf', handling of chromosome codes and male ploidy.

       'bcf-4.2'
              When  the 'bgz' modifier is present, the VCF file is block-gzipped.  (This always happens with BCF
              output.)  The 'id-paste' modifier controls which .psam columns are used to  construct  sample  IDs
              (choices  are  maybefid, fid, iid, maybesid, and sid; default is maybefid,iid,maybesid), while the
              'id-delim' modifier sets the character between the ID pieces (default '_').  Genotypes are  always
              exported.   If  you  want  to  export a sites-only VCF instead, see <b>--make-pgen</b>/--make-just-pvar's
              'vcfheader' column set.  Dosages are not exported unless the 'vcf-dosage='  modifier  is  present.
              The following five dosage export modes are supported: 'GP': genotype posterior probabilities (v4.3
              only).   'DS': Minimac3-style dosages, omitted for hardcalls.  'DS-force': Minimac3-style dosages,
              never omit.  'HDS': Minimac3-style phased dosages, omitted for hardcalls

       and unphased calls.
              Also includes 'DS' output.

              'HDS-force': Always report DS and HDS.

              In addition, * The '12' modifier causes alt1 alleles to be coded as '1' and ref alleles

              to be coded as '2', while '01' maps alt1 -&gt; 0 and ref -&gt; 1.

              * The 'spaces' modifier makes the output space-delimited instead of

              tab-delimited, whenever both are permitted.

              * For biallelic formats where it's unspecified whether the reference/major

              allele should appear first or second, <b>--export</b> defaults to second  for  compatibility  with  PLINK
              1.9.   Use  'ref-first'  to  change  this.   (Note  that  this doesn't apply to the 'A', 'AD', and
              'A-transpose' formats; use <b>--export-allele</b> to control which alleles are counted there.)

       <b>--freq</b> ['zs'] ['counts'] ['cols='&lt;column set descriptor&gt;] ['bins-only']

              ['refbins='&lt;comma-separated bin boundaries&gt; | 'refbins-file='&lt;file&gt;]  ['alt1bins='&lt;comma-separated
              bin boundaries&gt; | 'alt1bins-file='&lt;file&gt;]

       Empirical allele frequency report.
              By default, only founders are

       considered.
              Dosages are taken into account (e.g. heterozygous haploid

              calls count as 0.5).  Supported column sets are:

              chrom:  Chromosome  ID.   pos: Base-pair coordinate.  (ID is always present, and positioned here.)
              ref: Reference allele.  alt1: Alternate allele 1.  alt: All  alternate  alleles,  comma-separated.
              reffreq:   Reference   allele   frequency/dosage.    alt1freq:  Alt1  frequency/dosage.   altfreq:
              Comma-separated frequencies/dosages for all alternate alleles.  freq: Similar to  altfreq,  except
              ref  is  also included at the start.  eq: Comma-separated &lt;allele&gt;=&lt;freq&gt; for all present alleles.
              (If no

              alleles are present, the column contains a single '.'.)

              eqz: Same as eq, except zero-counts are included.  alteq/alteqz: Same as eq/eqz, except  reference
              allele  is  omitted.   numeq: 0=&lt;freq&gt;,1=&lt;freq&gt;, etc.  Zero-counts are omitted.  altnumeq: Same as
              numeq, except reference allele is omitted.   machr2:  Unphased  MaCH  imputation  quality  metric.
              minimac3r2: Phased Minimac3 imputation quality.  nobs: Number of allele observations.

              The    default    is    chrom,ref,alt,altfreq,nobs.     Additional    .afreq.{ref,alt1}.bins   (or
              .acount.{ref,alt1}.bins with 'counts') file(s) are generated  when  'refbins='/'refbins-file='  or
              'alt1bins='/'alt1bins-file=' is present; these report the total number of frequencies or counts in
              each  left-closed,  right-open  interval.   (If you only want these histogram(s), and not the main
              report, add 'bins-only'.)

       <b>--geno-counts</b> ['zs'] ['cols='&lt;column set descriptor&gt;]

              Variant-based hardcall genotype count report  (considering  both  alleles  simultaneously  in  the
              diploid  case).   Nonfounders  are  now  included;  use  <b>--keep-founders</b>  if  this  is  a problem.
              Heterozygous haploid calls are treated as missing.  Supported column sets are:

              chrom: Chromosome ID.  pos: Base-pair coordinate.  (ID is always present,  and  positioned  here.)
              ref:  Reference  allele.   alt1: Alternate allele 1.  alt: All alternate alleles, comma-separated.
              homref: Homozygous-ref count.  refalt1: Heterozygous ref-alt1 count.  refalt: Comma-separated  het
              ref-altx  counts.   homalt1:  Homozygous-alt1  count.  altxy: Comma-separated altx-alty counts, in
              (1/1)-(1/2)-(2/2)-(1/3)-...

              order.

              xy: Similar to altxy, except the reference allele is treated as alt0,

              and the sequence starts (0/0)-(0/1)-(1/1)-(0/2)-...

              hapref: Haploid-ref count.  hapalt1: Haploid-alt1  count.   hapalt:  Comma-separated  haploid-altx
              counts.   hap:  Similar to hapalts, except ref is also included at the start.  numeq: 0/0=&lt;hom ref
              ct&gt;,0/1=&lt;het ref-alt1&gt;,1/1=&lt;hom alt1&gt;,...,0=&lt;hap ref&gt;

       etc.   Zero-counts are omitted.  (If all genotypes are missing, the

              column contains a single '.'.)

              missing: Number of missing genotypes.  nobs: Number of (nonmissing) genotype observations.

              The default is chrom,ref,alt,homref,refalt,altxy,hapref,hapalt,missing.

       <b>--sample-counts</b> ['zs'] ['cols='&lt;column set descriptor&gt;]

              Sample-based hardcall genotype count report.  * Unknown-sex samples  are  treated  as  female.   *
              Heterozygous  haploid  calls  (MT  included)  are  treated  as  missing.   * As with other PLINK 2
              commands, SNPs that have not been left-normalized

       are counted as non-SNP non-symbolic.
              (Use e.g. <b>--normalize</b> when that's a

              problem.)

              * Supported column sets are:

              maybefid: FID, if that column was present in the input.  fid: Force FID column to be written  even
              when  absent  in the input.  (IID is always present, and positioned here.)  maybesid: SID, if that
              column was present in the input.  sid: Force SID column to be written  even  when  absent  in  the
              input.   sex:  '1' = male, '2' = female, 'NA' = missing.  hom: Homozygous genotype count.  homref:
              Homozygous-ref genotype count.  homalt: Homozygous-alt genotype count.  homaltsnp:  Homozygous-alt
              SNP  count.   het:  Heterozygous  genotype  count.   refalt: Heterozygous ref-altx genotype count.
              het2alt: Heterozygous altx-alty genotype count.  hetsnp: Heterozygous SNP count.   dipts:  Diploid
              SNP  transition count.  ts: SNP transition count (excluding chrY for females).  diptv: Diploid SNP
              transversion count.  tv: SNP transversion count.   dipnonsnpsymb:  Diploid  non-SNP,  non-symbolic
              count.   nonsnpsymb:  Non-SNP,  non-symbolic  count.   symbolic:  Symbolic variant count.  nonsnp:
              Non-SNP count.  dipsingle: Number of singletons relative to this dataset, across just

       diploid calls.
              (Note that if the ALT allele in a chrX

              biallelic variant appears in exactly one female and one male, that counts as a singleton for  just
              the female.)

              single:  Number  of  singletons  relative  to  this  dataset.   haprefwfemaley: Haploid-ref count,
              counting  chrY  for  everyone.   hapref:  Haploid-ref   count,   excluding   chrY   for   females.
              hapaltwfemaley:  Haploid-alt  count,  counting  chrY  for  everyone.   hapalt:  Haploid-alt count,
              excluding chrY for females.  missingwfemaley: Missing call  count,  counting  chrY  for  everyone.
              missing: Missing call count, excluding chrY for females.

              The           default           is          maybefid,maybesid,homref,homaltsnp,hetsnp,dipts,diptv,
              dipnonsnpsymb,dipsingle,haprefwfemaley,hapaltwfemaley,missingwfemaley.

              * The 'hetsnp', 'dipts'/'ts'/'diptv'/'tv', 'dipnonsnpsymb'/'nonsnpsymb',

              'symbolic', and  'nonsnp'  columns  count  each  ALT  allele  in  a  heterozygous  altx-alty  call
              separately,  since  they can be of different subtypes.  (I.e. if they are of the same subtype, the
              corresponding count is incremented by 2.)  As a  consequence,  these  columns  are  unaffected  by
              variant split/join.

       <b>--missing</b> ['zs'] [{sample-only | variant-only}]

              ['scols='&lt;column set descriptor&gt;] ['vcols='&lt;column set descriptor&gt;]

              Generate   sample-   and   variant-based   missing   data   reports   (or   just   one  report  if
              'sample-only'/'variant-only' is specified).  As of alpha 2,  mixed  MT  hardcalls  appear  in  the
              heterozygous haploid stats.  Supported column sets in the sample-based report are:

              maybefid:  FID, if that column was present in the input.  fid: Force FID column to be written even
              when absent in the input.  (IID is always present, and positioned here.)  maybesid: SID,  if  that
              column  was  present  in  the  input.  sid: Force SID column to be written even when absent in the
              input.  misspheno1: First active phenotype missing (Y/N)?  Always 'Y' if no

              phenotypes are loaded.

       missphenos: A Y/N column for each loaded phenotype.
              (Can be combined

              with misspheno1 to force at least one such column.)

              nmissdosage: Number of missing dosages.  nmiss: Number of  missing  hardcalls,  not  counting  het
              haploids.   nmisshh:  Number  of  missing  hardcalls,  counting  het  haploids.  hethap: Number of
              heterozygous haploid hardcalls.  nobs: Denominator (male count on  chrY,  otherwise  total  sample
              count).   fmissdosage:  Missing  dosage  rate.   fmiss:  Missing  hardcall  rate, not counting het
              haploids.  fmisshh: Missing hardcall rate, counting het haploids.

              The default  is  maybefid,maybesid,missphenos,nmiss,nobs,fmiss.   Supported  column  sets  in  the
              variant-based report are:

              chrom:  Chromosome  ID.   pos: Base-pair coordinate.  (ID is always present, and positioned here.)
              ref: Reference allele.  alt1: Alternate allele 1.  alt: All  alternate  alleles,  comma-separated.
              nmissdosage:  Number  of  missing  dosages.   nmiss: Number of missing hardcalls, not counting het
              haploids.  nmisshh: Number of  missing  hardcalls,  counting  het  haploids.   hethap:  Number  of
              heterozygous haploid calls.  nobs: Number of potentially valid calls.  fmissdosage: Missing dosage
              rate.   fmiss:  Missing hardcall rate, not counting het haploids.  fmisshh: Missing hardcall rate,
              counting het haploids.  fhethap: Heterozygous haploid rate.

              The default is chrom,nmiss,nobs,fmiss.

       <b>--hardy</b> ['zs'] ['midp'] ['redundant'] ['cols='&lt;column set descriptor&gt;]

              Hardy-Weinberg exact test p-value report(s).  * By default, only founders are  considered;  change
              this  with  <b>--nonfounders</b>.   *  chrX  is  now  omitted from the main &lt;output prefix&gt;.hardy report.
              Instead,

              (if present) it gets its own &lt;output prefix&gt;.hardy.x report  based  on  the  method  described  in
              Graffelman J, Weir BS (2016) Hardy-Weinberg equilibrium and the X chromosome.

              * For variants with k alleles where k&gt;2, k separate 'biallelic' tests are

       performed, each reported on its own line.
              However, biallelic variants

              are  normally  reported  on a single line, since the counts/frequencies would be mirror-images and
              the p-values would be the same.  You can add the 'redundant' modifier to force  biallelic  variant
              results to be reported on two lines for parsing convenience.

              * There is currently no special handling of case/control phenotypes.  Supported column sets are:

              chrom:  Chromosome  ID.   pos: Base-pair coordinate.  (ID is always present, and positioned here.)
              ref: Reference allele.  alt1: Alternate allele 1.  alt: All  alternate  alleles,  comma-separated.
              (A1  is  always  present,  and positioned here.)  ax: Non-A1 allele(s), comma-separated.  gcounts:
              Hom-A1 count, total number of het-A1 calls, and total number of

       nonmissing calls with no copies of A1.
              On chrX, these are

              followed by male A1 and male non-A1 counts.

              gcount1col: gcounts values in a single comma-separated column.   hetfreq:  Observed  and  expected
              het-A1  frequencies.  sexaf: Female and male A1 observed allele frequencies (chrX only).  femalep:
              Female-only p/midp-value (chrX only).  p: Hardy-Weinberg equilibrium exact test p/midp-value.

              The default is chrom,ax,gcounts,hetfreq,sexaf,p.

       <b>--indep-pairwise</b> &lt;window size&gt;['kb'] [step size (variant ct)]

              &lt;unphased-hardcall-r^2 threshold&gt;

              Generate a list of variants in approximate linkage  equilibrium.   *  For  multiallelic  variants,
              major allele counts are used in the r^2

              computation.

              * With the 'kb' modifier, the window size is in kilobase instead of variant

       count units.
              (Pre-'kb' space is optional, i.e.

              "--indep-pairwise 500 kb 0.5" and "--indep-pairwise 500kb 0.5" have the same effect.)

              * The step size now defaults to 1 if it's unspecified, and *must* be 1 if

              the window is in kilobase units.

              * Note that you need to rerun PLINK using <b>--extract</b> or <b>--exclude</b> on the

              .prune.in/.prune.out  file  to  apply  the  list  to  another  computation...  and  as  with other
              applications of <b>--extract</b>/--exclude, duplicate variant IDs are a problem.  <b>--indep-pairwise</b>  still
              runs  to  completion for now when duplicate variant IDs are present, but that will become an error
              in alpha 3.

       <b>--ld</b> &lt;variant ID&gt; &lt;variant ID&gt; ['dosage'] ['hwe-midp']

              This displays diplotype  frequencies,  r^2,  and  D'  for  a  single  pair  of  variants.   *  For
              multiallelic  variants,  major  allele  counts/dosages are used.  * Phase information is used when
              both variants are on the same chromosome.  * When there is at least one sample with  unphased  het
              calls for both

              variants,  diplotype  frequencies  are  estimated  using the Hill equation.  If there are multiple
              biologically possible local maxima, all are displayed, along with HWE exact test statistics.

       * By default, only hardcalls are considered.
              Add the 'dosage' modifier if

       you want dosages to be taken into account.
              (In the diploid case, an

              unphased dosage of x is interpreted as P(0/0) = 1 - x, P(0/1) = x when x is in 0..1.)

       <b>--sample-diff</b> ['id-delim='&lt;char&gt;] ['dosage' | 'dosage='&lt;tolerance&gt;]

              ['include-missing'] [{pairwise | counts-only}] ['fname-id-delim='&lt;c&gt;] ['zs']  ['cols='&lt;column  set
              descriptor&gt;]  ['counts-cols='&lt;column  set  descriptor&gt;]  {base=  |  ids=}&lt;sample ID&gt; [other sample
              ID(s)...]

       <b>--sample-diff</b> ['id-delim='&lt;char&gt;] ['dosage' | 'dosage='&lt;tolerance&gt;]

              ['include-missing'] [{pairwise | counts-only}] ['fname-id-delim='&lt;c&gt;] ['zs']  ['cols='&lt;column  set
              descriptor&gt;] ['counts-cols='&lt;column set descriptor&gt;] file=&lt;ID-pair file&gt;

              (alias:  <b>--sdiff</b>) Report discordances and discordance-counts between pairs of samples.  If chrX or
              chrY is present, sex must be defined and consistent.  * There are  three  ways  to  specify  which
              sample pairs to compare.  To

              compare  a  single baseline sample against some others, start the (space-delimited) sample ID list
              with 'base='.  To perform an all-vs.-all comparison, start it with  'ids='  instead.   To  compare
              sample  pairs  listed in a file, use 'file='.  Note that 'base='/'ids='/'file=' must be positioned
              after all modifiers.

              * Sample IDs are interpreted as if they were in a VCF header line, with

              'id-delim=' having the usual effect.

       * By default, comparisons are based on hardcalls.
              Use 'dosage' to compare

              dosages instead; you can combine this with a tolerance in [0, 0.5).

              * By default, if one genotype is missing and the other isn't, that doesn't

              count as a difference; this can be changed with 'include-missing'.

              * By default, a single main report is written to

       &lt;output prefix&gt;[.&lt;base ID&gt;].sdiff.
              To write separate pairwise

              &lt;output prefix&gt;.&lt;ID1&gt;.&lt;ID2&gt;.sdiff reports for each compared ID pair, add the 'pairwise'  modifier.
              To omit the main report, add the 'counts-only' modifier.  (Note that, if you're only interested in
              nonmissing  autosomal  biallelic  hardcalls,  <b>--make-king-table</b>  provides  a more efficient way to
              compute just counts.)

              * By default, if an output filename has a multipart sample ID, the parts

              will be delimited by '_'; use 'fname-id-delim=' to change this.

              Supported main-report column sets are:

              chrom: Chromosome ID.  pos: Base-pair coordinate.  (Variant ID is always present,  and  positioned
              here.)  ref: Reference allele.  alt: All alternate alleles, comma-separated.  maybefid: FID1/FID2,
              if that column was in the input.  Requires 'id'.  fid: Force FID1/FID2 even when FID was absent in
              the  input.  id: IID1/IID2.  maybesid: SID1/SID2, if that column was in the input.  Requires 'id'.
              sid: Force SID1/SID2 even when SID was absent in the input.  geno: Unphased GT or DS for  the  two
              samples.

              The  default  is  usually  chrom,pos,ref,alt,maybefid,id,maybesid,geno; the sample IDs are removed
              from the default in 'pairwise' mode.  Supported discordance-count-summary column sets are:

              maybefid: FID1/FID2, if that column was in the input.  fid: Force  FID1/FID2  even  when  FID  was
              absent  in the input.  (IID1/IID2 are always present.)  maybesid: SID1/SID2, if that column was in
              the input.  sid: Force SID1/SID2 even when SID was absent in the input.  nobs: Number of  variants
              considered.  This includes variants where one or

              both variants are missing iff 'include-missing' was specified.

              nobsibs: ibs0+ibs1+ibs2.  ibs0: Number of diploid variants with no common hardcall alleles.  ibs1:
              Number  of  diploid  variants  with  exactly  1  common  hardcall allele.  ibs2: Number of diploid
              variants with both hardcall alleles matching.  halfmiss: Number of variants with exactly 1 missing
              genotype/dosage.

              Ignored without 'include-missing'.

              diff: Total number of differences.

              The default is maybefid,maybesid,nobs,halfmiss,diff.

       <b>--make-king</b> [{square | square0 | triangle}] [{zs | bin | bin4}]

              KING-robust kinship estimator, described by Manichaikul A, Mychaleckyj JC, Rich SS, Daly  K,  Sale
              M,  Chen  WM (2010) Robust relationship inference in genome-wide association studies.  By default,
              this  writes  a  lower-triangular  tab-delimited  table  of  kinship   coefficients   to   &lt;output
              prefix&gt;.king,  and  a  list of the corresponding sample IDs to &lt;output prefix&gt;.king.id.  The first
              row of the .king file contains a single &lt;genome 1-genome 2&gt; kinship coefficient,  the  second  row
              has  the  &lt;genome  1-genome  3&gt; and &lt;genome 2-genome 3&gt; kinship values in that order, etc.  * Only
              autosomes  are  currently  considered.   *  Pedigree  information  is   currently   ignored;   the
              between-family estimator

              is used for all pairs.

              *  For multiallelic variants, REF allele counts are used.  * If the 'square' or 'square0' modifier
              is present, a square matrix is

              written instead; 'square0' fills the upper right triangle with zeroes.

              * If the 'zs' modifier is present, the .king file is Zstd-compressed.  * If the 'bin' modifier  is
              present, a binary (square) matrix of

              double-precision floating point values, suitable for loading from R, is instead written to &lt;output
              prefix&gt;.king.bin.  ('bin4' specifies single-precision numbers instead.)  This can be combined with
              'square0' if you still want the upper right zeroed out, or 'triangle' if you don't want to pad the
              upper right at all.

              * The computation can be subdivided with <b>--parallel</b>.

       <b>--make-king-table</b> ['zs'] ['counts'] ['rel-check'] ['cols='&lt;col set descrip.&gt;]

              Similar  to  <b>--make-king</b>,  except  results are reported in KING's original .kin0 text table format
              (with minor changes, e.g. row  order  is  more  friendly  to  incremental  addition  of  samples),
              <b>--king-table-filter</b> can be used to restrict the report to high kinship values, and the 'rel-check'
              modifier can be used to restrict to same-FID pairs.  Supported column sets are:

       maybefid: FID1/FID2, if that column was in the input.
              Requires 'id'.

              fid:  Force FID1/FID2 even when FID was absent in the input.  id: IID1/IID2.  maybesid: SID1/SID2,
              if that column was in the input.  Requires 'id'.  sid: Force SID1/SID2 even when SID was absent in
              the input.  nsnp: Number of  variants  considered  (autosomal,  neither  call  missing).   hethet:
              Proportion/count of considered call pairs which are het-het.  ibs0: Proportion/count of considered
              call  pairs  which are opposite homs.  ibs1: HET1_HOM2 and HET2_HOM1 proportions/counts.  kinship:
              KING-robust between-family kinship estimator.

              The  default  is  maybefid,id,maybesid,nsnp,hethet,ibs0,kinship.   hethet/ibs0/ibs1   values   are
              proportions  unless  the  'counts' modifier is present.  If id is omitted, a .kin0.id file is also
              written.

       <b>--make-rel</b> ['cov'] ['meanimpute'] [{square | square0 | triangle}]

              [{zs | bin | bin4}]

              Write a lower-triangular variance-standardized relationship matrix  to  &lt;output  prefix&gt;.rel,  and
              corresponding IDs to &lt;output prefix&gt;.rel.id.  * This computation assumes that variants do not have
              very low MAF, or

              deviate greatly from Hardy-Weinberg equilibrium.

              * Also, it's usually best to perform this calculation on a variant set in

              approximate linkage equilibrium.

              * The 'cov' modifier replaces the variance-standardization step with basic

              mean-centering, causing a covariance matrix to be calculated instead.

              * The computation can be subdivided with <b>--parallel</b>.

       <b>--make-grm-list</b> ['cov'] ['meanimpute'] ['zs'] [{id-header | iid-only}]

       <b>--make-grm-bin</b> ['cov'] ['meanimpute'] [{id-header | iid-only}]

       <b>--make-grm-list</b> causes the relationships to be written to GCTA's original

              list  format,  which  describes one pair per line, while <b>--make-grm-bin</b> writes them in GCTA 1.1+'s
              single-precision triangular binary format.  Note that these formats explicitly report  the  number
              of  valid  observations  (where  neither sample has a missing call) for each pair, which is useful
              input for some scripts.

       <b>--pca</b> [count] [{approx | meanimpute}] ['scols='&lt;col set descriptor&gt;]

       <b>--pca</b> [{allele-wts | biallelic-var-wts}] [count] [{approx | meanimpute}]

              ['vzs'] ['scols='&lt;col set descriptor&gt;] ['vcols='&lt;col set descriptor&gt;]

              Extracts top principal components from the variance-standardized relationship  matrix.   *  It  is
              usually best to perform this calculation on a variant set in

              approximate linkage equilibrium, with no very-low-MAF variants.

              * By default, 10 PCs are extracted; you can adjust this by passing a

       numeric argument.
              (Note that 10 is lower than the PLINK 1.9 default of

              20; this is due to the randomized algorithm's memory footprint growing quadratically w.r.t. the PC
              count.)

              * The 'approx' modifier causes the standard deterministic computation to be

              replaced  with  the randomized algorithm originally implemented for Galinsky KJ, Bhatia G, Loh PR,
              Georgiev S, Mukherjee S, Patterson NJ, Price AL (2016) Fast Principal-Component  Analysis  Reveals
              Convergent  Evolution of ADH1B in Europe and East Asia.  This can be a good idea when you have &gt;5k
              samples, and is almost required with &gt;50k.

              * The randomized algorithm always uses mean imputation for missing genotype

       calls. For comparison purposes, you can use the 'meanimpute' modifier to

              request this behavior for the standard computation.

              * 'scols=' can be used to customize how sample IDs appear in the .eigenvec

       file.  (maybefid, fid, maybesid, and sid supported; default is

              maybefid,maybesid.)

              * The 'allele-wts' modifier requests an additional one-line-per-allele

              .eigenvec.allele file with PCs expressed as allele weights instead of sample weights.   When  it's
              present,  'vzs'  causes  the .eigenvec.allele file to be Zstd-compressed.  'vcols=' can be used to
              customize the report columns; supported column sets are:

              chrom: Chromosome ID.  pos: Base-pair coordinate.  (ID is always present,  and  positioned  here.)
              ref:  Reference  allele.   alt1: Alternate allele 1.  alt: All alternate alleles, comma-separated.
              (A1 is always present, and positioned here.)   ax:  Non-A1  alleles,  comma-separated.   (PCs  are
              always present, and positioned here.)

              Default is chrom,ref,alt.

              * For datasets with no multiallelic variants, the 'biallelic-var-wts'

              modifier  requests  the  old  .eigenvec.var  format, which only reports weights for major alleles.
              (These weights are 2x the corresponding .eigenvec.allele weights.)  Supported column sets are:

              chrom: Chromosome ID.  pos: Base-pair coordinate.  (ID is always present,  and  positioned  here.)
              ref:  Reference  allele.   alt1: Alternate allele 1.  alt: All alternate alleles, comma-separated.
              maj: Major allele.  nonmaj: Minor allele.  (PCs are always present, and  positioned  here.   Signs
              are w.r.t. the major, not necessarily reference, allele.)

              Default is chrom,maj,nonmaj.

       <b>--king-cutoff</b> [.king.bin + .king.id fileset prefix] &lt;threshold&gt;

              Exclude  one  member  of  each  pair  of  samples  with KING-robust kinship greater than the given
              threshold.  Remaining/excluded sample IDs  are  written  to  &lt;output  prefix&gt;.king.cutoff.in.id  +
              .king.cutoff.out.id.   If present, the .king.bin file must be triangular (either precision is ok).

       <b>--write-covar</b> ['cols='&lt;column set descriptor&gt;]

              If covariates are defined, an updated version (with all filters applied) is automatically  written
              to  &lt;output  prefix&gt;.cov whenever <b>--make-pgen</b>, <b>--make-just-psam</b>, <b>--export</b>, or a similar command is
              present.  However, if you do not wish to simultaneously generate a new sample file,  you  can  use
              <b>--write-covar</b> to just produce a pruned covariate file.  Supported column sets are:

              maybefid:  FID,  if  that  column was in the input.  fid: Force FID column to be written even when
              absent in the input.  maybesid: SID, if that column was in the input.  sid: Force SID column to be
              written even when absent in the input.  maybeparents:  Father/mother  IIDs  ('0'  =  missing),  if
              columns  in  input.  parents: Force PAT/MAT columns to be written even when absent in input.  sex:
              '1' = male, '2' = female, 'NA' = missing.  pheno1: First active phenotype.  If  none,  all  column
              entries are set to

              the <b>--output-missing-phenotype</b> string.

       phenos: All active phenotypes, if any.
              (Can be combined with pheno1 to

              force at least one phenotype column to be written.)

              (Covariates are always present, and positioned here.)

              The default is maybefid,maybesid.

       <b>--write-samples</b>

              Report IDs of all samples which pass your filters/inclusion thresholds.

       <b>--write-snplist</b> ['zs'] ['allow-dups']

       List all variants which pass your filters/inclusion thresholds.
              Unless the

              'allow-dups' modifier is provided, this now errors out when duplicate variant ID(s) remain.

       <b>--glm</b> ['zs'] ['omit-ref'] [{sex | no-x-sex}] ['log10'] ['pheno-ids']

              [{genotypic | hethom | dominant | recessive}] ['interaction'] ['skip'] ['hide-covar'] [{no-firth |
              firth-fallback   |   firth}]   ['intercept']   ['cols='&lt;col   set   desc&gt;]  ['local-covar='&lt;file&gt;]
              ['local-psam='&lt;file&gt;]  ['local-pos-cols='&lt;key  col  #s&gt;  |   'local-pvar='&lt;file&gt;]   ['local-haps']
              ['local-omit-last' | 'local-cats[0]='&lt;category ct&gt;] ['allow-no-covars']

              Basic  association  analysis  on quantitative and/or case/control phenotypes.  For each variant, a
              linear (for quantitative traits) or  logistic  (for  case/control)  regression  is  run  with  the
              phenotype  as  the  dependent  variable,  and nonmajor allele dosage(s) and a constant-1 column as
              predictors.  * There is usually an additive effect line for every nonmajor allele, and

       no such line for the major allele.
              To omit REF alleles instead of major

       alleles, add the 'omit-ref' modifier.
              (When performing interaction

              testing, this tends to cause the multicollinearity check to fail for low-ref-frequency variants.)

              * By default, sex (male = 1, female = 2; note that this is a change from

              PLINK 1.x) is automatically added as a predictor for X chromosome variants, and  no  others.   The
              'sex'  modifier  causes  it  to  be  added  everywhere (except chrY), while 'no-x-sex' excludes it
              entirely.

              * The 'log10' modifier causes p-values to be reported in <b>-log10</b>(p) form.  * 'pheno-ids' causes the
              samples used in each set of regressions to be

       written to an .id file.
              (When the samples differ on chrX or chrY, .x.id

              and/or .y.id files are also written.)

              * The 'genotypic' modifier adds an additive effect/dominance deviation 2df

              joint test (0-2 and 0..1..0 coding), while 'hethom' uses 0..0..1 and 0..1..0 coding instead.

              * 'dominant' and 'recessive' specify a model assuming full dominance or

       recessiveness, respectively, for the ref allele.
              I.e. the genotype

              column is recoded as 0..1..1 or 0..0..1, respectively.

       * 'interaction' adds genotype x covariate interactions to the model.
              Note

              that this tends to produce 'NA' results (due to the multicollinearity check)  when  the  reference
              allele is 'wrong'; <b>--maj-ref</b> can be used to enable analysis of those variants.

       * Additional predictors can be added with <b>--covar</b>.
              By default, association

              statistics  are  reported  for  all nonconstant predictors; 'hide-covar' suppresses covariate-only
              results, while 'intercept' causes intercepts to be reported.  Since running <b>--glm</b> without at least
              e.g. principal component covariates  is  usually  an  analytical  mistake,  the  'allow-no-covars'
              modifier is now required when you're intentionally running <b>--glm</b> without a covariate file.

              * By default, if the current phenotype and covariates are such that every

       regression on a chromosome will fail, PLINK 2 errors out.
              To just skip

              the phenotype or chromosome instead, add the 'skip' modifier.

              * There are now three regression modes for case/control phenotypes:

              * 'no-firth' requests PLINK 1.x's behavior, where a NA result is reported

              when basic logistic regression fails to converge.

              * 'firth-fallback' requests logistic regression, followed by Firth

       regression whenever the logistic regression fails to converge.
              This is

              now the default.

              * 'firth' requests Firth regression all the time.

              * To add covariates which are not constant across all variants, add the

              'local-covar='  and  'local-psam='  modifiers,  use  full  filenames  for  each,  and  use  either
              'local-pvar=' or 'local-pos-cols=' to provide variant ID or position information.   Normally,  the
              local-covar  file  should  have c * n real-valued columns, where the first c columns correspond to
              the first sample in the local-psam file, columns (c+1) to 2c  correspond  to  the  second  sample,
              etc.;  and the mth line corresponds to the mth nonheader line of the local-pvar file when there is
              one.  (Variants outside of the local-pvar file are  excluded  from  the  regression.)   The  local
              covariates  are  assigned the names LOCAL1, LOCAL2, etc.; to exclude the last local covariate from
              the regression (necessary if they are e.g. local  ancestry  coefficients  which  sum  to  1),  add
              'local-omit-last'.  Alternatively, with 'local-cats='&lt;k&gt;, the local-covar file is expected to have
              n  columns  with  integer-valued entries in [1, k].  (This range is [0, k-1] with 'local-cats0='.)
              These category assignments are expanded into (k-1) local covariates in  the  usual  manner.   When
              position   information   is   in   the   local-covar   file,   this   should   be   indicated   by
              'local-pos-cols='&lt;number of header rows&gt;,&lt;chrom col #&gt;,&lt;pos start col #&gt;,&lt;first covariate col  #&gt;.
              'local-haps'  indicates  that  there's  one  column  or  column-group per haplotype instead of per
              sample; they are averaged by <b>--glm</b>.

              The main report supports the following column sets:

              chrom: Chromosome ID.  pos: Base-pair coordinate.  (ID is always present,  and  positioned  here.)
              ref:  Reference  allele.   alt1: Alternate allele 1.  alt: All alternate alleles, comma-separated.
              (A1 is always present, and positioned here.  For multiallelic variants, this  column  may  contain
              multiple  comma-separated  alleles  when  the  result  doesn't depend on which allele is A1.)  ax:
              Non-A1 alleles, comma-separated.  a1count: A1 allele count (can  be  decimal  with  dosage  data).
              totallele: Allele observation count (can be higher than <b>--freq</b> value, due

              to inclusion of het haploids and chrX model).

              a1countcc:  A1  count  in cases, then controls (case/control only).  totallelecc: Case and control
              allele observation counts.  gcountcc: Genotype hardcall  counts  (neither-A1,  het-A1,  A1-A1)  in
              cases,

              then controls (case/control only).

              a1freq:  A1 allele frequency.  a1freqcc: A1 frequency in cases, then controls (case/control only).
              machr2: Unphased MaCH imputation quality (frequently  labeled  'INFO').   firth:  Reports  whether
              Firth  regression  was  used (firth-fallback only).  test: Test identifier.  (Required unless only
              one test is run.)  nobs: Number of samples in the regression.  beta: Regression  coefficient  (for
              A1 if additive test).  orbeta: Odds ratio for case/control, beta for quantitative traits.

              (Ignored if 'beta' column set included.)

              se:  Standard  error  of  beta.  ci: Bounds of symmetric approximate confidence interval (requires
              <b>--ci</b>).  tz: T-statistic for linear regression, Wald Z-score  for  logistic/Firth.   p:  Asymptotic
              p-value (or <b>-log10</b>(p)) for T/Z-statistic.  err: Error code for NA results.

              The default is chrom,pos,ref,alt,firth,test,nobs,orbeta,se,ci,tz,p,err.

       <b>--score</b> &lt;filename&gt; [i] [j] [k] [{header | header-read}]

              [{center  |  variance-standardize  |  dominant  | recessive}] ['no-mean-imputation'] ['se'] ['zs']
              ['ignore-dup-ids'] [{list-variants | list-variants-zs}] ['cols='&lt;col set descriptor&gt;]

              Apply linear scoring system(s) to each sample.  The input file should have  one  line  per  scored
              (variant, allele) pair.  Variant IDs are read from column #i and allele codes are read from column
              #j,  where  i  defaults  to  1  and  j  defaults  to  i+1.  * By default, a single column of input
              coefficients is read from column #k,

       where k defaults to j+1.
              (<b>--score-col-nums</b> can be used to specify

              multiple columns.)

              * 'header-read' causes the first line of the input file to be treated as a

       header line containing score names.
              Otherwise, score(s) are assigned the

              names 'SCORE1', 'SCORE2', etc.; and 'header' just causes the first line to be entirely ignored.

              * By default, copies of unnamed alleles contribute zero to score, while

              missing genotypes contribute an amount proportional to the loaded  (via  <b>--read-freq</b>)  or  imputed
              allele  frequency.   To  throw out missing observations instead (decreasing the denominator in the
              final average when this happens), use the 'no-mean-imputation' modifier.

              * You can use the 'center' modifier to shift all genotypes to mean zero, or

              'variance-standardize' to linearly transform the genotypes to mean-0, variance-1.

              * The 'dominant' modifier causes dosages greater than 1 to be treated as 1,

              while 'recessive' uses max(dosage - 1, 0) on diploid chromosomes.  ('dominant',  'recessive',  and
              'variance-standardize' cannot be used with chrX.)

              * The 'se' modifier causes the input coefficients to be treated as

              independent standard errors; in this case, standard errors for the score average/sum are reported.
              (Note that this will systematically underestimate standard errors when scored variants are in LD.)

              * By default, <b>--score</b> errors out if a variant ID in the input file appears

       multiple times in the main dataset.
              Use the 'ignore-dup-ids' modifier to

              skip them instead (a warning is still printed if such variants are present).

              * The 'list-variants[-zs]' modifier causes variant IDs used for scoring to

              be written to &lt;output prefix&gt;.sscore.vars[.zst].

              The main report supports the following column sets:

              maybefid:  FID,  if  that  column was in the input.  fid: Force FID column to be written even when
              absent in the input.  (IID is always present, and positioned here.)  maybesid: SID, if that column
              was in the input.  sid: Force SID column to be written even when absent  in  the  input.   pheno1:
              First  active  phenotype.   phenos:  All active phenotypes, if any.  nallele: Number of nonmissing
              alleles.  denom: Denominator of score average (equal to nallele value when

              'no-mean-imputation' specified).

              dosagesum: Sum of named allele dosages.  scoreavgs: Score averages.  scoresums: Score sums.

              The  default  is  maybefid,maybesid,phenos,nallele,dosagesum,scoreavgs.   For  more  sophisticated
              polygenic  risk  scoring, we recommend looking at the LDpred (https://github.com/bvilhjal/ldpred )
              and PRSice-2 (https://www.prsice.info/ ) software packages.

       <b>--variant-score</b> &lt;filename&gt; ['zs'] ['bin' | 'cols='&lt;col set descriptor&gt;]

              (alias: <b>--vscore</b>) Apply linear scoring system(s) to each variant.  Each reported variant score  is
              the  dot  product  of a sample-weight vector with the total-ALT-dosage vector, with MAF-based mean
              imputation applied to missing dosages.  Input file format: one line per sample, each starting with
              an ID and followed by scoring weight(s); it can also  have  a  header  line  with  the  sample  ID
              representation and the score name(s).  The usual .vscore text report supports the following column
              sets:

              chrom:  Chromosome  ID.   pos: Base-pair coordinate.  (ID is always present, and positioned here.)
              ref: Reference allele.  alt1: Alternate allele 1.  alt: All  alternate  alleles,  comma-separated.
              altfreq:  ALT  allele  frequency  used  for  mean-imputation.   nmiss: Number of missing (and thus
              mean-imputed) dosages.  nobs: Number of (nonmissing) sample  observations.   (Variant  scores  are
              always present, and positioned here.)

              Default  is chrom,pos,ref,alt.  If binary output is requested instead, the main .vscore.bin matrix
              contains double-precision floating-point  values,  column  (score)  ID(s)  are  saved  to  &lt;output
              prefix&gt;.vscore.cols, and variant IDs are saved to &lt;output prefix&gt;.vscore.vars[.zst].

       <b>--adjust-file</b> &lt;filename&gt; ['zs'] ['gc'] ['cols='&lt;column set descriptor&gt;]

              ['log10'] ['input-log10'] ['test='&lt;test name, case-sensitive&gt;]

              Given  a  file  with  unfiltered  association  test  results,  report  some basic multiple-testing
              corrections, sorted in increasing-p-value order.  * 'gc' causes genomic-controlled p-values to  be
              used in the formulas.

       (This tends to be overly conservative.
              We note that LD Score regression

              usually  does  a  better job of calibrating lambda; see Lee JJ, McGue M, Iacono WG, Chow CC (2018)
              The accuracy of LD Score regression as an estimator of confounding  and  genetic  correlations  in
              genome-wide association studies.)

              * 'log10' causes negative base 10 logs of p-values to be reported, instead

       of raw p-values.
              'input-log10' specifies that the input file contains

       <b>-log10</b>(p) values.

              * If the input file contains multiple tests per variant which are

              distinguished  by  a  'TEST'  column (true for <b>--linear</b>/--logistic/--glm), you must use 'test=' to
              select the test to process.

              The following column sets are supported:

              chrom: Chromosome ID.  pos: Base-pair coordinate.  (ID is always present,  and  positioned  here.)
              ref:  Reference  allele.   alt1: Alternate allele 1.  alt: All alternate alleles, comma-separated.
              a1: Tested allele.  (Omitted if missing from input file.)  unadj: Unadjusted p-value.  gc:  Devlin
              &amp; Roeder (1999) genomic control corrected p-value (additive

              models only).

              qq:  P-value  quantile.   bonf:  Bonferroni  correction.   holm:  Holm-Bonferroni  (1979) adjusted
              p-value.  sidakss: Sidak single-step adjusted p-value.  sidaksd: Sidak step-down adjusted p-value.
              fdrbh: Benjamini &amp; Hochberg (1995) step-up false discovery control.  fdrby: Benjamini &amp;  Yekutieli
              (2001) step-up false discovery control.

              Default set is chrom,a1,unadj,gc,bonf,holm,sidakss,sidaksd,fdrbh,fdrby.

       <b>--genotyping-rate</b> ['dosage']

              Report genotyping rate in log (this was automatic in PLINK 1.x).

       <b>--pgen-info</b>

              Reports basic information about a .pgen file.

       <b>--validate</b>

              Validates all variant records in a .pgen file.

       <b>--zst-decompress</b> &lt;.zst file&gt; [output filename]

              (alias:  <b>--zd</b>) Decompress a Zstd-compressed file.  If no output filename is specified, the file is
              decompressed to standard output.  This cannot be used with any other flags, and does not  cause  a
              log file to be generated.

       The following other flags are supported.

       <b>--script</b> &lt;fname&gt;
              : Include command-line options from file.

       <b>--rerun</b> [log]
              : Rerun commands in log (default 'plink2.log').

       <b>--version</b>
              : Display only version number before exiting.

       <b>--silent</b>
              : Suppress regular output to console.  (Error-output is not suppressed.)

       <b>--double-id</b>
              : Set both FIDs and IIDs to the VCF/.bgen sample ID.

       <b>--const-fid</b> [ID]
              : Set all FIDs to the given constant.  If '0' (the default), no FID column is created.

       <b>--id-delim</b> [d]
              :  Normally  parses  single-delimiter  sample  IDs  as  &lt;FID&gt;&lt;d&gt;&lt;IID&gt;, and double-delimiter IDs as
              &lt;FID&gt;&lt;d&gt;&lt;IID&gt;&lt;d&gt;&lt;SID&gt;;  default  delimiter  is  '_'.   <b>--id-delim</b>  can  no  longer  be  used  with
              <b>--double-id</b>/--const-fid; it will error out if any ID lacks the delimiter.

       <b>--idspace-to</b> &lt;c&gt;
              : Convert spaces in VCF/.bgen sample IDs to the given character.

       <b>--iid-sid</b>
              : Make <b>--id-delim</b> and <b>--sample-diff</b> interpret two-token sample IDs as IID-SID instead of FID-IID.

       <b>--vcf-require-gt</b>
              : Skip variants with no GT field.

       <b>--vcf-min-gq</b> &lt;val&gt;
              : No-call genotypes when GQ is present and below the threshold.

       <b>--vcf-max-dp</b> &lt;val&gt;
              : No-call genotypes when DP is present and above/below

       <b>--vcf-min-dp</b> &lt;val&gt;
              the threshold.

       <b>--vcf-half-call</b> &lt;m&gt; : Specify how '0/.' and similar VCF GT values should be
              handled.   The  following four modes are supported: * 'error'/'e' (default) errors out and reports
              line #.  * 'haploid'/'h' treats them as haploid calls.  * 'missing'/'m' treats them as missing.  *
              'reference'/'r' treats the missing value as 0.

       <b>--oxford-single-chr</b> &lt;chr name&gt;
              : Specify single-chromosome .gen/.bgen file with no useful chromosome info inside.

       <b>--missing-code</b> [string list]
              : Comma-delimited list of missing phenotype

       (alias: <b>--missing_code</b>)
              values for Oxford-format import (default 'NA').

       <b>--hard-call-threshold</b> &lt;val&gt;
              : When importing dosage data, a hardcall is normally saved when  the  distance  from  the  nearest
              hardcall, defined as

       0.5 * sum_i |x_i - round(x_i)|
              (where the x_i's are 0..2 allele dosages), is not greater than 0.1.  You can adjust this threshold
              by   providing  a  numeric  argument  to  <b>--hard-call-threshold</b>.   You  can  also  use  this  with
              <b>--make-[b]pgen</b> to alter the saved hardcalls while leaving the dosages untouched, or <b>--make-bed</b>  to
              tweak hardcall export.

       <b>--dosage-erase-threshold</b> &lt;val&gt;
              :  <b>--hard-call-threshold</b> normally preserves the original dosages, and several PLINK 2 commands use
              them when they're available.  Use <b>--dosage-erase-threshold</b> to make PLINK 2 erase dosages and  keep
              only hardcalls when distance-from-hardcall &lt;= the given level.

       <b>--import-dosage-certainty</b> &lt;val&gt; : The PLINK 2 file format currently supports
              a  single  dosage  for each allele.  Some other dosage file formats include a separate probability
              for every possible genotype, e.g. {P(0/0)=0.2, P(0/1)=0.52, P(1/1)=0.28}, a highly uncertain  call
              that  is  nevertheless  treated  as a hardcall under '--hard-call-threshold 0.1'.  To make PLINK 2
              treat a dosage as missing  whenever  the  largest  probability  is  less  than  a  threshold,  use
              <b>--import-dosage-certainty</b>.

       <b>--input-missing-genotype</b> &lt;c&gt; : '.' is always interpreted as a missing
              genotype  code  in  input files.  By default, '0' also is; you can change this second missing code
              with <b>--input-missing-genotype</b>.

       <b>--allow-extra-chr</b>
              : Permit unrecognized chromosome codes (alias <b>--aec</b>).

       <b>--chr-set</b> &lt;autosome ct&gt; ['no-x'] ['no-y'] ['no-xy'] ['no-mt'] :

       Specify a nonhuman chromosome set.
              The first parameter sets the number of

              diploid autosome pairs if positive, or haploid chromosomes if negative.  Given diploid  autosomes,
              the remaining modifiers indicate the absence of the named non-autosomal chromosomes.

       <b>--cow</b>/--dog/--horse/--mouse/--rice/--sheep : Shortcuts for those species.

       <b>--autosome-num</b> &lt;val&gt;
              : Alias for '--chr-set &lt;value&gt; no-y no-xy no-mt'.

       <b>--human</b>
              :  Explicitly  specify  human  chromosome  set, and make output .pvar/VCF files include a ##chrSet
              header line.  (.pvar/VCF output files automatically  include  ##chrSet  when  a  nonhuman  set  is
              specified.)

       <b>--chr-override</b> ['file'] : By default, if <b>--chr-set</b>/--autosome-num/--cow/etc.
              conflicts with an input file ##chrSet header line, PLINK 2 will error out.  <b>--chr-override</b> with no
              argument causes the command line to take precedence; '--chr-override file' defers to the file.

       <b>--var-min-qual</b> &lt;val&gt;
              : Skip variants with low/missing QUAL.

       <b>--var-filter</b> [exception(s)...]
              : Skip variants which have FILTER failures.

       <b>--extract-if-info</b> &lt;key&gt; &lt;op&gt; &lt;val&gt; : Exclude variants which don't/do satisfy

       <b>--exclude-if-info</b> &lt;key&gt; &lt;op&gt; &lt;val&gt;
              a comparison predicate on an INFO key,

       (aliases: <b>--extract-if</b>,
              e.g.

       <b>--exclude-if</b>)
              <b>--extract-if-info</b> "VT == SNP"

       Unless the operator is !=, the predicate
              always evaluates to false when the key is missing.

       <b>--require-info</b> &lt;key(s)...&gt;
              : Exclude variants based on nonexistence

       <b>--require-no-info</b> &lt;key(s)...&gt;
              or existence of an INFO key.  "&lt;key&gt;=."  is treated as nonexistence.

       <b>--extract-col-cond</b> &lt;f&gt; [valcol] [IDcol] [skip] :

       <b>--extract-col-cond-match</b> &lt;(sub)string(s)...&gt;

       <b>--extract-col-cond-mismatch</b> &lt;(sub)string(s)...&gt;

       <b>--extract-col-cond-substr</b>

       <b>--extract-col-cond-min</b> &lt;min&gt;

       <b>--extract-col-cond-max</b> &lt;max&gt; :

              Exclude  all  variants  without a value-column entry satisfying a condition.  * By default, values
              are read from column 2 of the file, and variant IDs

              are read from column 1.

              * Three types of conditions are supported:

              * When <b>--extract-col-cond-match</b> is specified without

       <b>--extract-col-cond-substr</b>, the value is checked for equality with the

       given strings, and kept iff one of them matches.
              Similarly,

       <b>--extract-col-cond-mismatch</b> without <b>--extract-col-cond-substr</b> causes

              the variant to be kept iff the value matches none of the given strings.

              * When <b>--extract-col-cond-match</b> and/or <b>-mismatch</b> are specified with

       <b>--extract-col-cond-substr</b>, the variant is kept iff none of the

       <b>--extract-col-cond-mismatch</b> substrings are contained in the value, and

              either <b>--extract-col-cond-match</b> was unspecified or at least one of its substrings is contained.

              * Otherwise, the value is interpreted as a number, and the variant is

              kept if the number is in [&lt;min&gt;, &lt;max&gt;] (default min=0, max=DBL_MAX).

       <b>--pheno</b> ['iid-only'] &lt;f&gt; : Specify additional phenotype/covariate file.
              Comma-delimited files with a header line are now permitted.

       <b>--pheno-name</b> &lt;name...&gt;
              : Only load the designated phenotype(s) from the <b>--pheno</b> (if one was specified) or  .psam  (if  no
              <b>--pheno</b>) file.  Separate multiple names with spaces or commas, and use dashes to designate ranges.

       <b>--pheno-col-nums</b> &lt;#...&gt;
              : Only load the phenotype(s) in the designated column number(s) from the <b>--pheno</b> file.

       <b>--no-psam-pheno</b>
              : Ignore phenotype(s) in .psam/.fam file.

       <b>--strict-sid0</b>
              : By default, if there is no SID column in the .psam/.fam (or <b>--update-ids</b>) file, but there is one
              in another input file (for e.g. <b>--keep</b>/--remove), the latter SID column is ignored; sample IDs are
              considered  matching  as  long as FID and IID are equal (with missing FID treated as '0').  If you
              also want to require SID = '0' for a sample ID match in this situation, add <b>--strict-sid0</b>.

       <b>--input-missing-phenotype</b> &lt;v&gt; : Set nonzero number to treat as a missing
              pheno/covar in input files (default <b>-9</b>).

       <b>--no-input-missing-phenotype</b>
              : Don't treat any nonzero number as a missing  pheno/covar.   ('NA'/'nan'  are  still  treated  as
              missing.)

       <b>--1</b>    :  Expect  case/control phenotypes in input files to be coded as 0 = control, 1 = case, instead of
              the usual 0 = missing, 1 = ctrl, 2 = case.  (Unlike PLINK 1.x, this does not force all  phenotypes
              to be interpreted as case/ctrl.)

       <b>--missing-catname</b> &lt;str&gt;
              : Set missing-categorical-phenotype string (case-sensitive, default 'NONE').

       <b>--covar</b> ['iid-only'] &lt;f&gt; : Specify additional covariate file.
              Comma-delimited files with a header line are now permitted.

       <b>--covar-name</b> &lt;name...&gt;
              :  Only  load  the designated covariate(s) from the <b>--covar</b> (if one was specified), <b>--pheno</b> (if no
              <b>--covar</b>), or .psam (if no <b>--covar</b> or <b>--pheno</b>) file.

       <b>--covar-col-nums</b> &lt;#...&gt;
              : Only load the covariate(s) in the designated column number(s)  from  the  <b>--covar</b>  (if  one  was
              specified) or <b>--pheno</b> (if no <b>--covar</b>) file.

       <b>--within</b> &lt;f&gt; [new pheno name] : Import a PLINK 1.x categorical phenotype.
              (Phenotype name defaults to 'CATPHENO'.)  * If any numeric values are present, ALL

       values must be numeric.
              In that case, 'C'

       is added in front of all category names.
              * 'NA' is treated as a missing value.

       <b>--mwithin</b> &lt;n&gt;
              : Load <b>--within</b> categories from column n+2.

       <b>--family</b> [new pheno name]
              : Create a categorical phenotype from FID.  Restrictions on and handling of numeric values are the
              same as for <b>--within</b>.

       <b>--family-missing-catname</b> &lt;nm&gt; : Make <b>--family</b> treat the specified FID as
              missing.

       <b>--keep</b> &lt;fname...&gt;
              : Exclude all samples not named in a file.

       <b>--remove</b> &lt;fname...&gt;
              : Exclude all samples named in a file.

       <b>--keep-fam</b> &lt;fn...&gt;
              : Exclude all families not named in a file.

       <b>--remove-fam</b> &lt;f...&gt;
              : Exclude all families named in a file.

       <b>--extract</b> [{bed0 | bed1}] &lt;f...&gt; : Usually excludes all variants (not) named

       <b>--exclude</b> [{bed0 | bed1}] &lt;f...&gt;
              in  the  given file(s).  When multiple files are named, they are concatenated.  With the 'bed0' or
              'bed1' modifier, variants outside/inside the positional ranges in  the  interval-BED  file(s)  are
              excluded  instead.   'bed0'  tells  PLINK  2 to assume the interval bounds follow the UCSC 0-based
              half-open  convention,  while  'bed1'  (equivalent  to  PLINK  1.9  'range')   specifies   1-based
              fully-closed.

       <b>--extract-intersect</b> [{bed0 | bed1}] &lt;f...&gt; : Just like <b>--extract</b>, except that
              a variant must be in the intersection, rather than just the union, of the files to remain.

       <b>--bed-border-bp</b> &lt;n&gt;
              : Stretch BED intervals by the given amount on each

       <b>--bed-border-kb</b> &lt;n&gt;
              side.

       <b>--keep-cats</b> &lt;filename&gt;
              : These can be used individually or in combination

       <b>--keep-cat-names</b> &lt;nm...&gt;
              to  define  a  list  of  categories  to  keep;  all samples not in one of the named categories are
              excluded.  Use spaces to separate category names for <b>--keep-cat-names</b>.  Use the  <b>--missing-catname</b>
              value (default 'NONE') to refer to the group of uncategorized samples.

       <b>--keep-cat-pheno</b> &lt;pheno&gt; : If more than one categorical phenotype is loaded,
              or  you  wish to filter on a categorical covariate, <b>--keep-cat-pheno</b> must be used to specify which
              phenotype/covariate <b>--keep-cats</b> and <b>--keep-cat-names</b> apply to.

       <b>--remove-cats</b> &lt;filename&gt; : Exclude all categories named in the file.

       <b>--remove-cat-names</b> &lt;...&gt; : Exclude named categories.

       <b>--remove-cat-pheno</b> &lt;phe&gt; : Specify pheno for <b>--remove-cats</b>/remove-cat-names.

       <b>--split-cat-pheno</b> [{omit-most | omit-last}] ['covar-01']
              [cat. pheno/covar name(s)...] :

              Split n-category phenotype(s) into n (or n-1,  with  'omit-most'/'omit-last')  binary  phenotypes,
              with names of the form &lt;orig. pheno name&gt;=&lt;cat. name&gt;.  (As a consequence, affected phenotypes and
              categories  are  not  permitted  to  contain  the '=' character.)  * This happens after all sample
              filters.  * If no phenotype or covariate names are provided, all categorical

              phenotypes (but not covariates) are processed.

       * By default, generated covariates are coded as 1=false, 2=true.
              To code

              them as 0=false, 1=true instead, add the 'covar-01' modifier.

       <b>--loop-cats</b> &lt;pheno/covar&gt;
              : Run variant filters and subsequent operations on just the samples in the  first  category;  then
              just  the  samples  in the second category; and so on, for all categories in the named categorical
              phenotype.

       <b>--no-id-header</b> ['iid-only'] : Don't include a header line in .id output
              files.  This normally forces two-column FID/IID output; add 'iid-only' to force just single-column
              IID.

       <b>--variance-standardize</b> [pheno/covar name(s)...]

       <b>--covar-variance-standardize</b> [covar name(s)...] :

              Linearly transform named covariates (and quantitative phenotypes,  if  <b>--variance-standardize</b>)  to
              mean-zero,  variance  1.   If  no  arguments  are provided, all possible phenotypes/covariates are
              affected.  This is frequently necessary to prevent multicollinearity when dealing with  covariates
              where <a href="../manmean/abs.mean.html">abs</a>(mean) is much larger than abs(standard deviation), such as year of birth.

       <b>--quantile-normalize</b> [...]
              : Force named covariates and quantitative

       <b>--pheno-quantile-normalize</b> [...]
              phenotypes to a N(0,1) distribution,

       <b>--covar-quantile-normalize</b> [...]
              preserving only the original rank orders.

       <b>--chr</b> &lt;chr(s)...&gt;
              : Exclude all variants not on the given chromosome(s).  Valid choices for humans are 0 (unplaced),
              1-22,  X, Y, XY, MT, PAR1, and PAR2.  Separate multiple chromosomes with spaces and/or commas, and
              use a dash (no adjacent spaces permitted) to denote a range, e.g.  '--chr 1-4, 22, par1, x, par2'.

       <b>--not-chr</b> &lt;...&gt;
              : Reverse of <b>--chr</b> (exclude variants on listed chromosomes).

       <b>--autosome</b>
              : Exclude all non-autosomal variants.

       <b>--autosome-par</b>
              : Exclude all non-autosomal variants, except those in a pseudo-autosomal region.

       <b>--snps-only</b> ['just-acgt'] : Exclude non-SNP variants.
              By default, SNP = all allele codes are single-character (so multiallelic variants with  a  mix  of
              SNPs  and  non-SNPs are excluded; split your variants first if that's a problem).  The 'just-acgt'
              modifier restricts SNP codes to {A,C,G,T,a,c,g,t,&lt;missing&gt;}.

       <b>--from</b> &lt;var ID&gt;
              : Use ID(s) to specify a variant range to load.  When used

       <b>--to</b>   &lt;var ID&gt;      together, both variants must be on the same chromosome.   (<b>--snps</b>  can  be  used  to
              specify intervals which cross chromosome boundaries.)

       <b>--snp</b>  &lt;var ID&gt;    : Specify a single variant to load.

       <b>--exclude-snp</b> &lt;ID&gt; : Specify a single variant to exclude.

       <b>--window</b>
              &lt;kbs&gt;     :  With  <b>--snp</b>/--exclude-snp,  loads/excludes  all variants within half the specified kb
              distance of the named one.

       <b>--from-bp</b> &lt;pos&gt;
              : Use base-pair coordinates to define a variant range to

       <b>--to-bp</b>
              &lt;pos&gt;      load.

       <b>--from-kb</b> &lt;pos&gt;
              * You must use these with <b>--chr</b>, specifying a single

       <b>--to-kb</b>
              &lt;pos&gt;        chromosome.

       <b>--from-mb</b> &lt;pos&gt;
              * Decimals and negative numbers are permitted.

       <b>--to-mb</b>
              &lt;pos&gt;      * The <b>--to-bp</b>(/-kb/-mb) position  is  no  longer  permitted  to  be  smaller  than  the
              <b>--from-bp</b> position.

       <b>--snps</b> &lt;var IDs...&gt;
              : Use IDs to specify variant range(s) to load or

       <b>--exclude-snps</b> &lt;...&gt;
              exclude.  E.g. '--snps rs1111-rs2222, rs3333, rs4444'.

       <b>--force-intersect</b>
              :   PLINK   2   normally   errors   out   when  multiple  variant  inclusion  filters  (<b>--extract</b>,
              <b>--extract-col-cond</b>,  <b>--extract-intersect</b>,  <b>--from</b>/--to,  <b>--from-bp</b>/--to-bp,  <b>--snp</b>,  <b>--snps</b>)   are
              specified.  <b>--force-intersect</b> allows the run to proceed; the set intersection will be taken.

       <b>--thin</b> &lt;p&gt;
              : Randomly remove variants, retaining each with prob. p.

       <b>--thin-count</b> &lt;n&gt;
              : Randomly remove variants until n of them remain.

       <b>--bp-space</b> &lt;bps&gt;
              : Remove variants so that each pair is no closer than the given bp distance.

       <b>--thin-indiv</b> &lt;p&gt;
              : Randomly remove samples, retaining with prob. p.

       <b>--thin-indiv-count</b> &lt;n&gt; : Randomly remove samples until n of them remain.

       <b>--keep-col-match</b> &lt;f&gt; &lt;val(s)...&gt; : Exclude all samples without a 3rd column
              entry  in  the  given  file exactly matching one of the given strings.  (Separate multiple strings
              with spaces.)

       <b>--keep-col-match-name</b> &lt;col name&gt; : Check column with given name instead.

       <b>--keep-col-match-num</b> &lt;n&gt;
              : Check nth column instead.

       <b>--geno</b> [val] [{dosage | hh-missing}]

       <b>--mind</b> [val] [{dosage | hh-missing}] :

              Exclude variants (<b>--geno</b>) and/or samples (<b>--mind</b>) with missing call  frequencies  greater  than  a
              threshold  (default  0.1).   (Note  that the default threshold is only applied if <b>--geno</b>/--mind is
              invoked without an argument; when <b>--geno</b>/--mind is not invoked, no missing call frequency  ceiling
              is enforced at all.  Other inclusion/exclusion default thresholds work the same way.)  By default,
              when  a  dosage  is  present  but  a  hardcall is not, the genotype is treated as missing; add the
              'dosage' modifier to treat this case as nonmissing.  Alternatively, you can  use  'hh-missing'  to
              also treat heterozygous haploid calls as missing.

       <b>--require-pheno</b> [name(s)...] : Remove samples missing any of the named

       <b>--require-covar</b> [name(s)...]
              phenotype(s)/covariate(s).   If  no  arguments are provided, all phenotype(s)/covariate(s) must be
              present.

       <b>--maf</b> [freq] [mode]
              : Exclude variants with allele frequency lower than a

       (alias: <b>--min-af</b>)
              threshold (default 0.01).  By default, the nonmajor allele frequency is used; the other  supported
              modes  are  'nref'  (non-reference),  'alt1',  and  'minor'  (least frequent).  bcftools freq:mode
              notation is permitted.

       <b>--max-maf</b> &lt;freq&gt; [mode] : Exclude variants with MAF greater than the

       (alias: <b>--max-af</b>)
              threshold.

       <b>--mac</b> &lt;ct&gt; [mode]
              : Exclude variants with allele dosage lower than the

       (alias: <b>--min-ac</b>)
              given threshold.

       <b>--max-mac</b> &lt;ct&gt; [mode]
              : Exclude variants with allele dosage greater than

       (alias: <b>--max-ac</b>)
              the given threshold.

       <b>--maf-succ</b>
              : Rule of succession allele frequency estimation (used in EIGENSOFT).  Given j observations of one
              allele and k observations of the other for a biallelic variant, infer allele frequencies of  (j+1)
              /  (j+k+2)  and  (k+1) / (j+k+2), rather than the default j / (j+k) and k / (j+k).  Note that this
              does not affect <b>--freq</b>'s output.

       <b>--min-alleles</b> &lt;ct&gt; : Exclude variants with fewer than the given # of alleles.
              (When a  variant  has  exactly  one  ALT  allele,  and  it's  a  missing-code,  it's  excluded  by
              "--min-alleles 2".)

       <b>--max-alleles</b> &lt;ct&gt; : Exclude variants with more than the given # of alleles.

       <b>--read-freq</b> &lt;file&gt; : Load allele frequency estimates from the given <b>--freq</b> or
              <b>--geno-counts</b>  (or PLINK 1.9 <b>--freqx</b>) report, instead of imputing them from the immediate dataset.

       <b>--hwe</b> &lt;p&gt; ['midp'] ['keep-fewhet'] :

              Exclude variants with Hardy-Weinberg equilibrium exact test p-values  below  a  threshold.   *  By
              default,  only  founders  are  considered.   * chrX p-values are now computed using Graffelman and
              Weir's method.  * For variants with k alleles with k&gt;2, k separate 'biallelic' tests are

              performed, and the variant is filtered out if any of them fail.

              * With 'keep-fewhet', variants which fail the test in the too-few-hets

       direction are not excluded.
              On chrX, this uses the ratio between the

              Graffelman/Weir p-value and the female-only p-value.

              * There is currently no special handling of case/control phenotypes.

       <b>--mach-r2-filter</b> [min] [max] : Exclude variants with MaCH imputation quality
              metric less than min or greater than max (defaults 0.1 and 2.0).  (Monomorphic variants, with r2 =
              nan, are not excluded.)  * This is NOT identical to the R2 metric

       reported by Minimac3 0.1.13+; see below.
              * If a single argument is provided, it is

       treated as the minimum.
              * The metric is not computed on chrX and MT.

       <b>--minimac3-r2-filter</b> &lt;min&gt; [max] : Compute Minimac3 R2 values from scratch,
              and exclude variants with R2 less than min or (if max is provided) greater than max.  * Note  that
              this requires phased-dosage

       data for all samples and variants;
              otherwise   this   will   systematically   underestimate   imputation   quality,   since  unphased
              hardcalls/dosages   are    treated    as    if    they    were    maximally    uncertain.     (Use
              <b>--extract-if-info</b>/--exclude-if-info  to  filter  on  precomputed  Minimac3  R2 in a VCF/.pvar INFO
              column.)

       <b>--keep-females</b>
              : Exclude male and unknown-sex samples.

       <b>--keep-males</b>
              : Exclude female and unknown-sex samples.

       <b>--keep-nosex</b>
              : Exclude all known-sex samples.

       <b>--remove-females</b>
              : Exclude female samples.

       <b>--remove-males</b>
              : Exclude male samples.

       <b>--remove-nosex</b>
              : Exclude unknown-sex samples.

       <b>--keep-founders</b>
              : Exclude nonfounder samples.

       <b>--keep-nonfounders</b> : Exclude founder samples.

       <b>--keep-if</b> &lt;pheno/covar&gt; &lt;op&gt; &lt;val&gt; : Exclude samples which don't/do satisfy a

       <b>--remove-if</b> &lt;pheno/covar&gt; &lt;op&gt; &lt;v&gt;
              comparison predicate, e.g.  <b>--keep-if</b> "PHENO1 == case"

       Unless the operator is !=, the predicate
              always evaluates to false when the phenotype/covariate is missing.

       <b>--nonfounders</b>
              : Include nonfounders in allele freq/HWE calculations.

       <b>--bad-freqs</b>
              : When PLINK 2 needs decent allele frequencies, it normally errors out if they aren't provided  by
              <b>--read-freq</b> and less than 50 founders are available to impute them from.  Use <b>--bad-freqs</b> to force
              PLINK 2 to proceed in this case.

       <b>--bad-ld</b>
              :  PLINK  2  normally errors out when it needs to estimate LD between variants, but there are less
              than 50 founders to estimate from.  Use <b>--bad-ld</b> to force PLINK 2 to proceed.

       <b>--export-allele</b> &lt;file&gt; : With <b>--export</b> A/A-transpose/AD, count alleles named
              in the file, instead of REF alleles.

       <b>--output-chr</b> &lt;MT code&gt; : Set chromosome coding scheme in output files by
              providing the desired human mitochondrial code.  Options  are  '26',  'M',  'MT',  '0M',  'chr26',
              'chrM',  and  'chrMT';  default  is  now  'MT'  (note  that this is a change from PLINK 1.x, which
              defaulted to '26').

       <b>--output-missing-genotype</b> &lt;ch&gt; : Set the code used to represent missing
              genotypes in output files (default '.').

       <b>--output-missing-phenotype</b> &lt;s&gt; : Set the string used to represent missing
              phenotypes in output files (default 'NA').

       <b>--sort-vars</b> [mode]
              : Sort variants by chromosome, then position, then ID.  The following string orders are supported:
              * 'natural'/'n':  Natural  sort  (default).   *  'ascii'/'a':  ASCII.   This  must  be  used  with
              <b>--make-[b]pgen</b>/--make-bed.

       <b>--set-hh-missing</b> ['keep-dosage'] : Make <b>--make-[b]pgen</b>/--make-bed set non-MT
              heterozygous  haploid  hardcalls,  and all female chrY calls, to missing.  (Unlike PLINK 1.x, this
              treats unknown-sex chrY genotypes like males, not females.)  By default,  all  associated  dosages
              are also erased; use 'keep-dosage' to keep them all.

       <b>--set-mixed-mt-missing</b> ['keep-dosage'] : Make <b>--make-[b]pgen</b>/--make-bed set
              mixed MT hardcalls to missing.

       <b>--split-par</b> &lt;bp1&gt; &lt;bp2&gt; : Changes chromosome code of all X chromosome

       <b>--split-par</b> &lt;build&gt;
              variants  with  bp position &lt;= bp1 to PAR1, and those with position &gt;= bp2 to PAR2.  The following
              build codes are supported as shorthand: * 'b36'/'hg18' = NCBI 36, 2709521/154584237 * 'b37'/'hg19'
              = GRCh37, 2699520/154931044 * 'b38'/'hg38' = GRCh38, 2781479/155701383

       <b>--merge-par</b>
              : Merge PAR1/PAR2 back with X.  Requires PAR1 to be positioned immediately before X, and  PAR2  to
              be  immediately  after  X.   (Should  *not*  be  used  with  "--export  vcf", since it causes male
              homozygous/missing calls in PAR1/PAR2 to be reported as haploid.)

       <b>--merge-x</b>
              : Merge XY back with X.  This usually has to be combined with <b>--sort-vars</b>.

       <b>--set-missing-var-ids</b> &lt;t&gt;
              : Given a template string with a '@' where the

       <b>--set-all-var-ids</b> &lt;t&gt;
              chromosome code should go and '#' where the bp coordinate belongs,  <b>--set-missing-var-ids</b>  assigns
              chromosome-and-bp-based  IDs to unnamed variants, while <b>--set-all-var-ids</b> resets all IDs.  You may
              also use '$r'/'$a' to refer to the ref and  alt1  alleles,  or  '$1'/'$2'  to  refer  to  them  in
              alphabetical order.

       <b>--var-id-multi</b> &lt;t&gt;
              : Specify alternative templates for multiallelic

       <b>--var-id-multi-nonsnp</b> &lt;t&gt;
              variants.   ('$a' and '$1'/'$2' should be avoided here, though they're technically still allowed.)

       <b>--new-id-max-allele-len</b> &lt;len&gt; [{error | missing | truncate}] :

              Specify maximum number of leading characters from allele codes to include in new variant IDs,  and
              behavior on longer codes (defaults 23, error).

       <b>--missing-var-code</b> &lt;str&gt;
              :  Change  unnamed  variant  code for <b>--rm-dup</b>, <b>--set-</b>{missing|all}-var-ids, and <b>--recover-var-ids</b>
              (default '.').

       <b>--update-map</b>
              &lt;f&gt; [bpcol]  [IDcol]  [skip] : Update variant bp positions.

       <b>--update-name</b> &lt;f&gt; [newcol] [oldcol] [skip] : Update variant IDs.

       <b>--recover-var-ids</b> &lt;file&gt; ['strict-bim-order'] [{rigid | force}] ['partial'] :

       Undo <b>--set-all-var-ids</b>, given the original .pvar/VCF/.bim file.
              Original

              IDs are looked up by position and allele codes.  * By default, if the original-ID file is a  .bim,
              allele order is ignored.

              Use 'strict-bim-order' to force A1=ALT, A2=REF.

              * If any variant has multiple matching records in the original-ID file, and

              the   IDs   conflict,   <b>--recover-var-ids</b>   writes   the   affected  (current)  ID(s)  to  &lt;output
              prefix&gt;.recoverid.dup, and normally errors out.  If the original-ID file has the  same  number  of
              variants  in  the  same order, you can still recover the old IDs with the 'rigid' modifier in this
              case.  Alternatively, to proceed and assign the missing-ID code to these variants, add the 'force'
              modifier.  (The .recoverid.dup file is still written when 'rigid' or 'force' is specified.)

              * <b>--recover-var-ids</b> normally expects to replace all variant IDs, and errors

       out if any are left untouched.
              Add the 'partial' modifier when you

              actually want to update just a proper subset.

       <b>--update-alleles</b> &lt;fname&gt;
              : Update variant allele codes.

       <b>--update-ids</b> &lt;fname&gt;
              : Update sample IDs.

       <b>--update-parents</b> &lt;fname&gt;
              : Update parental IDs.

       <b>--update-sex</b> &lt;filename&gt; ['col-num='&lt;n&gt;] ['male0'] :

              Update sex information.  * By default, if there is a header line starting with '#FID'/'#IID',  sex
              is

              loaded  from  the  first column titled 'SEX' (any capitalization); otherwise, column 3 is assumed.
              Use 'col-num=' to force a column number.

       * Only the first character in the sex column is processed.
              By default,

              '1'/'M'/'m' is interpreted  as  male,  '2'/'F'/'f'  is  interpreted  as  female,  and  '0'/'N'  is
              interpreted  as unknown-sex.  To change this to '0'/'M'/'m' = male, '1'/'F'/'f' = female, anything
              else other than '2' = unknown-sex, add 'male0'.

       <b>--real-ref-alleles</b>
              : Treat A2 alleles in a PLINK 1.x fileset as actual  REF  alleles;  otherwise  they're  marked  as
              provisional.

       <b>--maj-ref</b> ['force'] : Set major alleles to reference, like PLINK 1.x
              automatically  did.   (Note that this is now opt-in rather than opt-out; <b>--keep-allele-order</b> is no
              longer necessary to prevent allele-swapping.)  * This can only be used in runs with

       <b>--make-bed</b>/--make-[b]pgen/--export and no other
              commands.

       * By default, this only affects variants marked as
              having 'provisional' reference alleles.  Add 'force' to apply this to all variants.

              * All new reference alleles are marked as provisional.

       <b>--ref-allele</b> ['force'] &lt;filename&gt; [refcol] [IDcol] [skip]

       <b>--alt1-allele</b> ['force'] &lt;filename&gt; [alt1col] [IDcol] [skip] :

              These set the alleles specified in the file to ref (<b>--ref-allele</b>) or alt1  (<b>--alt1-allele</b>).   They
              can   be   combined   in   the   same   run.    *   These   can   only   be   used  in  runs  with
              <b>--make-bed</b>/--make-[b]pgen/--export

              and no other commands.

              * "--ref-allele &lt;VCF filename&gt; 4 3 '#'", which scrapes reference allele

              assignments from a VCF file, is especially useful.

              * By default, these error out when asked to change a 'known' reference

       allele.
              Add 'force' to permit that (when e.g. switching to a new

              reference genome).

              * When <b>--alt1-allele</b> changes the previous ref allele to alt1, the previous

              alt1 allele is set to reference and marked as provisional.

       <b>--ref-from-fa</b> ['force'] : This sets reference alleles from the <b>--fa</b> file when
              it can be done unambiguously (note that it's never possible for deletions or some insertions).  By
              default, it errors out when asked to change a 'known' reference allele; add the  'force'  modifier
              to permit that.

       <b>--normalize</b> ['list']
              : Left-normalize all variants, using the <b>--fa</b> file.

       (alias: <b>--norm</b>)
              (Assumes no differences in capitalization.)  The 'list' modifier causes a list of affected variant
              IDs to be written to &lt;output prefix&gt;.normalized.

       <b>--indiv-sort</b> &lt;mode&gt; [f] : Specify sample ID sort order for merge and
              <b>--make-[b]pgen</b>/--make-bed.   The following four modes are supported: * 'none'/'0' keeps samples in
              the order they were

       loaded.
              Default for non-merge.

       * 'natural'/'n' invokes "natural sort", e.g.
              'id2' &lt; 'ID3' &lt; 'id10'.  Default when merging.

       * 'ascii'/'a' sorts in ASCII order, e.g.
              'ID3' &lt; 'id10' &lt; 'id2'.

       * 'file'/'f' uses the order in the given file
              (named in the last argument).

       <b>--king-table-filter</b> &lt;min&gt;
              : Specify minimum kinship coefficient for inclusion in <b>--make-king-table</b> report.

       <b>--king-table-subset</b> &lt;f&gt; [kmin] : Restrict current <b>--make-king-table</b> run to
              sample pairs listed in the given .kin0 file.  If a second argument is provided, only sample  pairs
              with kinship &gt;= that threshold (in the input .kin0) are processed.

       <b>--condition</b> &lt;variant ID&gt; [{dominant | recessive}] ['multiallelic']

       <b>--condition-list</b> &lt;fname&gt; [{dominant | recessive}] ['multiallelic'] :

              Add  the  given variant, or all variants in the given file, as <b>--glm</b> covariates.  By default, this
              errors out if any of the variants  are  multiallelic;  add  the  'multiallelic'  ('m'  for  short)
              modifier  to  allow  them.   They'll effectively be split against the major allele (unless <b>--glm</b>'s
              'omit-ref'  modifier  was  specified),  and  all  induced  covariate  names--even  for   biallelic
              variants--will have an underscore followed by the allele code at the end.

       <b>--parameters</b> &lt;...&gt; : Include only the given covariates/interactions in the
              <b>--glm</b> model, identified by a list of 1-based indices and/or ranges of them.

       <b>--tests</b> &lt;...&gt;
              : Perform a (joint) test on the specified term(s) in the

       <b>--tests</b> all          <b>--glm</b> model, identified by 1-based indices and/or ranges
              of them.  * Note that, when <b>--parameters</b> is also present, the

       indices refer to the terms remaining AFTER pruning by
              <b>--parameters</b>.

              * You can use '--tests all' to include all terms.

       <b>--vif</b> &lt;max VIF&gt;
              :  Set  VIF  threshold for <b>--glm</b> multicollinearity check (default 50).  (This is no longer skipped
              for case/control phenotypes.)

       <b>--max-corr</b> &lt;val&gt;
              : Skip <b>--glm</b> regression when the absolute value of the correlation between two predictors  exceeds
              this value (default 0.999).

       <b>--xchr-model</b> &lt;m&gt;
              :  Set the chrX <b>--glm</b>/--condition[-list]/--[v]score model.  * '0' = skip chrX.  * '1' = add sex as
              a covar on chrX, code males 0..1.  * '2' (default) = chrX sex covar, code males  0..2.   (Use  the
              <b>--glm</b> 'interaction' modifier to test for interaction between genotype and sex.)

       <b>--adjust</b> ['zs'] ['gc'] ['log10'] ['cols='&lt;column set descriptor&gt;] :

              For  each  association test in this run, report some basic multiple-testing corrections, sorted in
              increasing-p-value order.  Modifiers work the same way as they do on <b>--adjust-file</b>.

       <b>--lambda</b>
              : Set genomic control lambda for <b>--adjust[-file]</b>.

       <b>--adjust-chr-field</b> &lt;n...&gt;
              : Set <b>--adjust-file</b> input field names.  When

       <b>--adjust-pos-field</b> &lt;n...&gt;
              multiple arguments are given to these flags,

       <b>--adjust-id-field</b> &lt;n...&gt;
              earlier names take precedence over later ones.

       <b>--adjust-ref-field</b> &lt;n...&gt;

       <b>--adjust-alt-field</b> &lt;n...&gt;

       <b>--adjust-a1-field</b> &lt;n...&gt;

       <b>--adjust-test-field</b> &lt;n...&gt;

       <b>--adjust-p-field</b> &lt;n...&gt;

       <b>--ci</b> &lt;size&gt;
              : Report confidence ratios for odds ratios/betas.

       <b>--pfilter</b> &lt;val&gt;
              : Filter out assoc. test results with higher p-values.

       <b>--score-col-nums</b> &lt;...&gt; : Process all the specified coefficient columns in the
              <b>--score</b> file, identified by 1-based indexes and/or ranges of them.

       <b>--q-score-range</b> &lt;range file&gt; &lt;data file&gt; [i] [j] ['header'] ['min'] :

              Apply <b>--score</b> to subset(s) of variants in the primary score list(s) based on e.g. p-value  ranges.
              * The first file should have range labels in the first column, p-value

              lower  bounds  in  the  second  column,  and upper bounds in the third column.  Lines with too few
              entries, or nonnumeric values in the second or third column, are ignored.

              * The second file should contain a variant ID and a p-value on each line

       (except possibly the first).
              Variant IDs are read from column #i and

              p-values are read from column #j, where i defaults to 1 and  j  defaults  to  i+1.   The  'header'
              modifier causes the first nonempty line of this file to be skipped.

              * By default, <b>--q-score-range</b> errors out when a variant ID appears multiple

       times in the data file (and is also present in the main dataset).
              To use

              the minimum p-value in this case instead, add the 'min' modifier.

       <b>--vscore-col-nums</b> &lt;...&gt; : Process all the specified coefficient columns in
              the <b>--variant-score</b> file, identified by 1-based indexes and/or ranges of them.

       <b>--parallel</b> &lt;k&gt; &lt;n&gt; : Divide the output matrix into n pieces, and only compute
              the  kth  piece.   The  primary  output file will have the piece number included in its name, e.g.
              plink2.king.13 or plink2.king.13.zst if k is 13.  Concatenating these files in  order  will  yield
              the  full  matrix of interest.  (Yes, this can be done before decompression.)  N.B. This generally
              cannot be used to directly write a symmetric square matrix.   Choose  square0  or  triangle  shape
              instead, and postprocess as necessary.

       <b>--memory</b> &lt;val&gt; ['require'] : Set size, in MiB, of initial workspace malloc
              attempt.   To  error  out instead of reducing the request size when the initial attempt fails, add
              the 'require' modifier.

       <b>--threads</b> &lt;val&gt;
              : Set maximum number of compute threads.

       <b>--d</b> &lt;char&gt;
              : Change variant/covariate range delimiter (normally '-').

       <b>--seed</b> &lt;val...&gt;
              : Set random number seed(s).  Each value must be an integer between 0  and  4294967295  inclusive.
              Note that <b>--threads</b> and "--memory require" may also be needed to reproduce some randomized runs.

       <b>--output-min-p</b> &lt;p&gt; : Specify minimum p-value to write to reports.
              (2.23e-308 is useful for preventing underflow in some programs.)

       <b>--debug</b>
              : Use slower, more crash-resistant logging method.

       <b>--randmem</b>
              : Randomize initial workspace memory (helps catch uninitialized-memory bugs).

       <b>--warning-errcode</b>
              : Return a nonzero error code to the OS when a run completes with warning(s).

       <b>--zst-level</b> &lt;lvl&gt;
              : Set the Zstd compression level (1-22, default 3).

       Primary  methods  paper:  Chang  CC,  Chow  CC,  Tellier  LCAM,  Vattikuti  S,  Purcell SM, Lee JJ (2015)
       Second-generation PLINK: rising to the challenge of larger and richer datasets.  GigaScience, 4.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       The full documentation for <b>PLINK</b> <b>v2.00a3</b> <b>64-bit</b> <b>(9</b> <b>Apr</b> is maintained as a Texinfo manual.   If  the  <b>info</b>
       and <b>PLINK</b> <b>v2.00a3</b> <b>64-bit</b> <b>(9</b> <b>Apr</b> programs are properly installed at your site, the command

              <b>info</b> <b>PLINK</b> <b>v2.00a3</b> <b>64-bit</b> <b>(9</b> <b>Apr</b>

       should give you access to the complete manual.

PLINK v2.00a3 64-bit (9 Apr 2020)                  March 2020                                           <u><a href="../man1/PLINK.1.html">PLINK</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>