<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>topp - library for LC/MS data management and analysis - tools</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/topp">topp_2.6.0+cleaned1-4build5_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       topp - library for LC/MS data management and analysis - tools

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This manual page documents briefly the <b>topp</b> package that brings a set of binary tools to build the OpenMS
       proteomics  pipeline  (topp).  OpenMS  offers an infrastructure for the development of mass spectrometry-
       related software and powerful 2D and 3D visualization  solutions.  OpenMS  offers  analyses  for  various
       quantitation protocols, including label-free quantitation, SILAC, iTRAQ, SRM, SWATH… It provides build-in
       algorithms  for de-novo identification and database search, as well as adapters to other state-of-the art
       tools like X!Tandem, Mascot, OMSSA… OpenMS supports the Proteomics Standard Initiative (PSI) formats  for
       MS  data  and supports easy integration of tools into workflow engines like Knime, Galaxy, WS-Pgrade, and
       TOPPAS via the TOPPtools concept and a unified parameter handling.

</pre><h4><b>AVAILABLE</b> <b>MODULES</b></h4><pre>
       <b>AccurateMassSearch</b> -- Match MS signals to molecules from a database by mass.

       <b>AssayGeneratorMetabo</b> -- Assay library generation from DDA data (Metabolomics).

       <b>BaselineFilter</b> -- Removes the baseline from profile spectra using a top-hat filter.

       <b>CVInspector</b> -- A tool for visualization and validation of PSI mapping and CV files.

       <b>ClusterMassTraces</b> -- Creates pseudo spectra.

       <b>ClusterMassTracesByPrecursor</b> -- Correlate precursor masstraces with fragment ion masstraces in SWATH maps
       based on their elution profile.

       <b>CometAdapter</b> -- Annotates MS/MS spectra using Comet.

       <b>CompNovo</b> -- Performs a de novo peptide identification using the CompNovo engine.

       <b>CompNovoCID</b> -- Performs a de novo peptide identification using the CompNovo engine.

       <b>ConsensusID</b> -- Computes a consensus of peptide identifications of several identification engines.

       <b>ConsensusMapNormalizer</b> -- Normalizes maps of one consensusXML file.

       <b>CruxAdapter</b> -- Identifies MS/MS spectra using Crux.

       <b>DTAExtractor</b> -- Extracts spectra of an MS run file to several files in DTA format.

       <b>DatabaseFilter</b> -- Filters a protein database (FASTA format) based on identified proteins.

       <b>DatabaseSuitability</b> -- Computes a suitability  score  for  a  database  which  was  used  for  a  peptide
       identification search. Also reports the quality of LC-MS spectra.

       <b>DeMeanderize</b> -- Orders the spectra of MALDI spotting plates correctly.

       <b>Decharger</b> -- Decharges and merges different feature charge variants of the same peptide.

       <b>DecoyDatabase</b> -- Create decoy sequence database from forward sequence database.

       <b>Digestor</b> -- Digests a protein database in-silico.

       <b>DigestorMotif</b> -- digests a protein database in-silico.

       <b>EICExtractor</b> -- Extracts intensities from dedicates positions in a LC/MS map.

       <b>ERPairFinder</b> -- Util which can be used to evaluate pair ratios on enhanced resolution (zoom) scans.

       <b>Epifany</b> -- Runs a Bayesian protein inference.

       <b>ExecutePipeline</b> -- Executes workflows created by TOPPAS.

       <b>ExternalCalibration</b> -- Applies an external mass recalibration.

       <b>FFEval</b> -- Evaluation tool for feature detection algorithms.

       <b>FalseDiscoveryRate</b>  --  Estimates  the  false  discovery  rate  on  peptide and protein level using decoy
       searches.

       <b>FeatureFinderCentroided</b> -- Detects two-dimensional features in LC-MS data.

       <b>FeatureFinderIdentification</b> -- Detects features in MS1 data based on peptide identifications.

       <b>FeatureFinderIsotopeWavelet</b> -- Detects two-dimensional features in LC-MS data.

       <b>FeatureFinderMRM</b> -- Detects two-dimensional features in LC-MS data.

       <b>FeatureFinderMetabo</b> -- Assembles metabolite features from centroided (LC-)MS data using  the  mass  trace
       approach.

       <b>FeatureFinderMetaboIdent</b> -- Detects features in MS1 data based on metabolite identifications.

       <b>FeatureFinderMultiplex</b> -- Determination of peak ratios in LC-MS data.

       <b>FeatureFinderSuperHirn</b> -- Finds mass spectrometric features in mass spectra.

       <b>FeatureLinkerLabeled</b> -- Groups corresponding isotope-labeled features in a feature map.

       <b>FeatureLinkerUnlabeled</b> -- Groups corresponding features from multiple maps.

       <b>FeatureLinkerUnlabeledKD</b> -- Groups corresponding features from multiple maps.

       <b>FeatureLinkerUnlabeledQT</b> -- Groups corresponding features from multiple maps.

       <b>FidoAdapter</b> -- Runs the protein inference engine Fido.

       <b>FileConverter</b> -- Converts between different MS file formats.

       <b>FileFilter</b> -- Extracts or manipulates portions of data from peak, feature or consensus-feature files.

       <b>FileInfo</b> -- Shows basic information about the file, such as data ranges and file type.

       <b>FileMerger</b> -- Merges several MS files into one file.

       <b>FuzzyDiff</b> -- Compares two files, tolerating numeric differences.

       <b>GNPSExport</b> -- Tool to export consensus features into MGF format.

       <b>GenericWrapper</b> -- Allows the generic wrapping of external tools.

       <b>HighResPrecursorMassCorrector</b>  --  Corrects  the  precursor  mass and charge determined by the instrument
       software.

       <b>IDConflictResolver</b> -- Resolves ambiguous annotations of features with peptide identifications.

       <b>IDDecoyProbability</b> -- Estimates peptide probabilities using a decoy search strategy.

       <b>IDExtractor</b> -- Extracts 'n' peptides randomly or best 'n' from idXML files.

       <b>IDFileConverter</b> -- Converts identification engine file formats.

       <b>IDFilter</b> -- Filters results from protein or peptide identification engines based on different criteria.

       <b>IDMapper</b> -- Assigns protein/peptide identifications to features or consensus features.

       <b>IDMassAccuracy</b> -- Calculates a distribution of the mass error from given mass spectra and IDs.

       <b>IDMerger</b> -- Merges several protein/peptide identification files into one file.

       <b>IDPosteriorErrorProbability</b> -- Estimates probabilities for incorrectly assigned peptide sequences  and  a
       set of search engine scores using a mixture model.

       <b>IDRTCalibration</b> -- Can be used to calibrate RTs of peptide hits linearly to standards.

       <b>IDRipper</b>  --  Split  protein/peptide  identification  file into several files according to annotated file
       origin.

       <b>IDScoreSwitcher</b> -- Switches between different scores of peptide or protein hits in identification data.

       <b>IDSplitter</b> -- Splits protein/peptide identifications off of annotated data files.

       <b>INIFileEditor</b> -- An editor for OpenMS configuration files.

       <b>INIUpdater</b> -- Update INI and TOPPAS files to new OpenMS version.

       <b>ImageCreator</b> -- Transforms an LC-MS map into an image.

       <b>InclusionExclusionListCreator</b> -- Creates inclusion and/or exclusion lists.

       <b>InspectAdapter</b> -- Annotates MS/MS spectra using Inspect.

       <b>InternalCalibration</b> -- Applies an internal mass recalibration.

       <b>IsobaricAnalyzer</b> -- Calculates isobaric quantitative values for peptides.

       <b>LabeledEval</b> -- Evaluation tool for isotope-labeled quantitation experiments.

       <b>LuciphorAdapter</b> -- Modification site localisation using LuciPHOr2.

       <b>MRMMapper</b> -- MRMMapper maps measured chromatograms (mzML) and the transitions used (TraML).

       <b>MRMPairFinder</b> -- Util which can be used to evaluate labeled pair ratios on MRM features.

       <b>MRMTransitionGroupPicker</b> -- Picks peaks in SRM/MRM chromatograms.

       <b>MSFraggerAdapter</b> -- Peptide Identification with MSFragger.

       <b>MSGFPlusAdapter</b> -- MS/MS database search using MS-GF+.

       <b>MSSimulator</b> -- A highly configurable simulator for mass spectrometry experiments.

       <b>MSstatsConverter</b> -- Converter to input for MSstats.

       <b>MaRaClusterAdapter</b> -- Facilitate input to MaRaCluster and reintegrate.

       <b>MapAlignerIdentification</b> -- Corrects retention time distortions between  maps  based  on  common  peptide
       identifications.

       <b>MapAlignerPoseClustering</b>  --  Corrects  retention  time  distortions between maps using a pose clustering
       approach.

       <b>MapAlignerSpectrum</b> -- Corrects retention time distortions between maps by spectrum alignment.

       <b>MapAlignerTreeGuided</b> -- Tree guided correction of retention time distortions between maps.

       <b>MapNormalizer</b> -- Normalizes peak intensities in an MS run.

       <b>MapRTTransformer</b> -- Applies retention time transformations to maps.

       <b>MapStatistics</b> -- Extract extended statistics on the features of a map for quality control.

       <b>MascotAdapter</b> -- Annotates MS/MS spectra using Mascot.

       <b>MascotAdapterOnline</b> -- Annotates MS/MS spectra using Mascot.

       <b>MassCalculator</b> -- Calculates masses and mass-to-charge ratios of peptide sequences.

       <b>MassTraceExtractor</b> -- Detects mass traces in centroided LC-MS data.

       <b>MetaProSIP</b> -- Performs proteinSIP on peptide features for elemental flux analysis.

       <b>MetaboliteAdductDecharger</b> --  Decharges  and  merges  different  feature  charge  variants  of  the  same
       metabolite.

       <b>MetaboliteSpectralMatcher</b> -- Perform a spectral library search.

       <b>MultiplexResolver</b> -- Completes peptide multiplets and resolves conflicts within them.

       <b>MyriMatchAdapter</b> -- Annotates MS/MS spectra using MyriMatch.

       <b>MzMLSplitter</b> -- Splits an mzML file into multiple parts.

       <b>MzTabExporter</b> -- Exports various XML formats to an mzTab file.

       <b>NoiseFilterGaussian</b> -- Removes noise from profile spectra by using Gaussian filter (on uniform as well as
       non-uniform data).

       <b>NoiseFilterSGolay</b>  --  Removes  noise  from  profile  spectra  by using a Savitzky Golay filter. Requires
       uniform (equidistant) data.

       <b>NovorAdapter</b> -- Template for Tool creation.

       <b>NucleicAcidSearchEngine</b> -- Annotate nucleic acid identifications to MS/MS spectra.

       <b>OMSSAAdapter</b> -- Annotates MS/MS spectra using OMSSA.

       <b>OpenPepXL</b> -- Tool for protein-protein cross-linking identification using labeled linkers.

       <b>OpenPepXLLF</b> -- Tool for protein-protein cross linking with label-free linkers.

       <b>OpenSwathAnalyzer</b> -- Picks peaks and finds features in an SWATH-MS or SRM experiment.

       <b>OpenSwathAssayGenerator</b> -- Generates assays according to different models for a specific TraML.

       <b>OpenSwathChromatogramExtractor</b> -- Extract chromatograms (XIC) from a MS2 map file.

       <b>OpenSwathConfidenceScoring</b> -- Compute confidence scores for OpenSwath results.

       <b>OpenSwathDIAPreScoring</b> -- Scoring spectra using the DIA scores.

       <b>OpenSwathDecoyGenerator</b> -- Generates decoys according to different models for a specific TraML.

       <b>OpenSwathFeatureXMLToTSV</b> -- Converts a featureXML to a mProphet tsv.

       <b>OpenSwathFileSplitter</b> -- Splits SWATH files into n files, each containing one window.

       <b>OpenSwathMzMLFileCacher</b> -- This tool caches the spectra and chromatogram data of an mzML to disk.

       <b>OpenSwathRTNormalizer</b> -- This tool will take a description of RT peptides and their normalized  retention
       time to write out a transformation file on how to transform the RT space into the normalized space.

       <b>OpenSwathRewriteToFeatureXML</b> -- Combines featureXML and mProphet tsv to FDR filtered featureXML.

       <b>OpenSwathWorkflow</b> -- Complete workflow to run OpenSWATH.

       <b>PSMFeatureExtractor</b> -- Computes extra features for each input PSM.

       <b>PTModel</b> -- Trains a model for the prediction of proteotypic peptides from a training set.

       <b>PTPredict</b>  --  predicts  the  likelihood  of peptides to be proteotypic via svm_model which is trained by
       PTModel.

       <b>PeakPickerHiRes</b> -- Finds mass spectrometric peaks in profile mass spectra.

       <b>PeakPickerIterative</b> -- Finds mass spectrometric peaks in profile mass spectra.

       <b>PeakPickerWavelet</b> -- Finds mass spectrometric peaks in profile mass spectra.

       <b>PepNovoAdapter</b> -- Adapter to PepNovo supporting all PepNovo command  line  parameters.  The  results  are
       converted from the PepNovo text outfile format into the idXML format.

       <b>PeptideIndexer</b> -- Refreshes the protein references for all peptide hits.

       <b>PercolatorAdapter</b> -- Facilitate input to Percolator and reintegrate.

       <b>PhosphoScoring</b> -- Scores potential phosphorylation sites in order to localize the most probable sites.

       <b>PrecursorIonSelector</b> -- PrecursorIonSelector.

       <b>PrecursorMassCorrector</b> -- Corrects the precursor entries of MS/MS spectra, by using MS1 information.

       <b>ProteinInference</b> -- Protein inference based on an aggregation of the scores of the identified peptides.

       <b>ProteinQuantifier</b> -- Compute peptide and protein abundances.

       <b>ProteinResolver</b> -- protein inference.

       <b>ProteomicsLFQ</b> -- A standard proteomics LFQ pipeline.

       <b>QCCalculator</b>  --  Calculates basic quality parameters from MS experiments and subsequent analysis data as
       identification or feature detection.

       <b>QCEmbedder</b> -- Attaches a table or an image to a given qc parameter.

       <b>QCExporter</b> -- Will extract several qp from several run/sets in a tabular format.

       <b>QCExtractor</b> -- Extracts a table attachment to a given qc parameter.

       <b>QCImporter</b> -- Imports tables with quality control parameters into qcml files.

       <b>QCMerger</b> -- Merges two qcml files together.

       <b>QCShrinker</b> -- This application is used to remove the verbose table attachments from a qcml file that  are
       not needed anymore, e.g. for a final report.

       <b>QualityControl</b>  --  Computes  various QC metrics from many possible input files (only the consensusXML is
       required). The more optional files you provide, the more metrics you get.

       <b>RNADigestor</b> -- Digests an RNA sequence database in-silico.

       <b>RNAMassCalculator</b> -- Calculates masses, mass-to-charge ratios and sum formulas of RNA sequences.

       <b>RNPxlSearch</b> -- Annotate RNA/DNA-peptide cross-links in MS/MS spectra.

       <b>RNPxlXICFilter</b> -- Remove MS2 spectra from  treatment  based  on  the  fold  change  between  control  and
       treatment.

       <b>RTEvaluation</b> -- Application that evaluates TPs (true positives), TNs, FPs, and FNs for an idXML file with
       predicted RTs.

       <b>RTModel</b> -- Trains a model for the retention time prediction of peptides from a training set.

       <b>RTPredict</b> -- Predicts retention times for peptides using a model trained by RTModel.

       <b>Resampler</b> -- Transforms an LC/MS map into a resampled map or a PNG image.

       <b>SeedListGenerator</b> -- Generates seed lists for feature detection.

       <b>SemanticValidator</b> -- SemanticValidator for semantically validating certain XML files.

       <b>SequenceCoverageCalculator</b> -- Prints information about idXML files.

       <b>SimpleSearchEngine</b> -- Annotates MS/MS spectra using SimpleSearchEngine.

       <b>SiriusAdapter</b> -- Tool for metabolite identification using single and tandem mass spectrometry.

       <b>SpecLibCreator</b> -- Creates an MSP formatted spectral library.

       <b>SpecLibSearcher</b>  --  Identifies  peptide  MS/MS  spectra  by spectral matching with a searchable spectral
       library.

       <b>SpectraFilterBernNorm</b> -- Applies thresholdfilter to peak spectra.

       <b>SpectraFilterMarkerMower</b> -- Applies thresholdfilter to peak spectra.

       <b>SpectraFilterNLargest</b> -- Applies thresholdfilter to peak spectra.

       <b>SpectraFilterNormalizer</b> -- Normalizes intensity of peak spectra.

       <b>SpectraFilterParentPeakMower</b> -- Applies thresholdfilter to peak spectra.

       <b>SpectraFilterScaler</b> -- Applies thresholdfilter to peak spectra.

       <b>SpectraFilterSqrtMower</b> -- Applies thresholdfilter to peak spectra.

       <b>SpectraFilterThresholdMower</b> -- Applies thresholdfilter to peak spectra.

       <b>SpectraFilterWindowMower</b> -- Applies thresholdfilter to peak spectra.

       <b>SpectraMerger</b> -- Merges spectra (each MS level separately), increasing S/N ratios.

       <b>SpectraSTSearchAdapter</b> -- Interface to the SEARCH Mode of the SpectraST executable.

       <b>StaticModification</b> -- Applies a set of modifications to all PeptideIDs in an idXML file.

       <b>SvmTheoreticalSpectrumGeneratorTrainer</b>    --    Trainer    for    SVM     models     as     input     for
       SvmTheoreticalSpectrumGenerator.

       <b>SwathWizard</b> -- An assistant for Swath-Analysis.

       <b>TICCalculator</b> -- Calculates the TIC from a mass spectrometric raw file (useful for benchmarking).

       <b>TOFCalibration</b> -- Applies time of flight calibration.

       <b>TOPPAS</b> -- An assistant for GUI-driven TOPP workflow design.

       <b>TOPPView</b> -- A viewer for mass spectrometry data.

       <b>TargetedFileConverter</b>  --  Converts  different  transition  files  for targeted proteomics / metabolomics
       analysis.

       <b>TextExporter</b> -- Exports various XML formats to a text file.

       <b>TransformationEvaluation</b> -- Applies a transformation to a range of values.

       <b>XFDR</b> -- Calculates false discovery rate estimates on crosslink identifications.

       <b>XMLValidator</b> -- Validates XML files against an XSD schema.

       <b>XTandemAdapter</b> -- Annotates MS/MS spectra using X! Tandem.

</pre><h4><b>BIBLIOGRAPHICAL</b> <b>REFERENCE</b> <b>TO</b> <b>BE</b> <b>CITED</b></h4><pre>
       Marc Sturm, Andreas Bertsch, Clemens Gröpl, Andreas Hildebrandt, Rene Hussong, Eva Lange,  Nico  Pfeifer,
       Ole  Schulz-Trieglaff,  Alexandra  Zerck,  Knut  Reinert, and Oliver Kohlbacher (2008)  OpenMS – an Open-
       Source Software Framework for Mass Spectrometry.  <u>BMC</u> <u>Bioinformatics</u>, 9:163. doi:10.1186/1471-2105-9-163.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man3/libopenms.3.html">libopenms</a></b>(3), <b><a href="../man7/openms-common.7.html">openms-common</a></b>(7), <b><a href="../man7/openms.7.html">openms</a></b>(7), <b><a href="../man7/openms-doc.7.html">openms-doc</a></b>(7).

       The program is documented fully in the <u>Tutorials</u>, that are packaged in openms-doc.

</pre><h4><b>AUTHOR</b></h4><pre>
       This manual page was written by Filippo Rusconi &lt;<a href="mailto:lopippo@debian.org">lopippo@debian.org</a>&gt;.  Permission  is  granted  to  copy,
       distribute  and/or  modify  this  document  under the terms of the GNU General Public License, Version 3,
       published by the Free Software Foundation.

        On a Debian system the complete text of the GNU General Public License version 3 can  be  found  in  the
       file `<u><a href="file:/usr/share/common-licenses/GPL-3">/usr/share/common-licenses/GPL-3</a></u>'.

                                                  November 2012                                          <u><a href="../man1/TOPP.1.html">TOPP</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>