<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>alien_hunter - Interpolated Variable Order Motifs for identification of horizontally acquired DNA</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/alien-hunter">alien-hunter_1.7-10_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       alien_hunter - Interpolated Variable Order Motifs for identification of horizontally acquired DNA

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>alien_hunter</b> <u>&lt;sequence.file&gt;</u> <u>&lt;output.file&gt;</u> [<u>Options</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Alien_hunter  is an application for the prediction of putative Horizontal Gene Transfer (HGT) events with
       the implementation of Interpolated Variable Order Motifs (IVOMs). An IVOM approach exploits compositional
       biases using variable order motif distributions and captures more reliably the  local  composition  of  a
       sequence  compared  to  fixed-order  methods. Optionally the predictions can be parsed into a 2-state 2nd
       order Hidden Markov Model (HMM), in a change-point detection framework, to optimize the  localization  of
       the  boundaries of the predicted regions.  The predictions (embl format) can be automatically loaded into
       Artemis genome viewer freely available at: <a href="http://www.sanger.ac.uk/Software/Artemis/">http://www.sanger.ac.uk/Software/Artemis/</a>.

       INPUT: raw genomic sequence PREDICTION: HGT regions based on Interpolated Variable Order  Motifs  (IVOMs)
       arguments:

       &lt;sequence.file&gt;
              raw genomic sequence

       &lt;output.file&gt;
              filename for output data

       Optional arguments:

       -a     to load the prediction file into Artemis

       -c     optimize predicted boundaries with a change-point detection 2 state 2nd order HMM

       Output files:

       &lt;output.file&gt;
              predictions (tab file) in embl format

       &lt;output.file&gt;.opt
              optimized (HMM) predictions (tab file) in embl format

       &lt;output.file&gt;.plot
              predictions in Artemis User Plot format to be loaded manually using Graph -&gt; Add User Plot...

       &lt;output.file&gt;.opt.plot
              optimized  (HMM)  predictions in Artemis User Plot format to be loaded manually using Graph -&gt; Add
              User Plot...

       &lt;output.file&gt;.sco
              the scores over all the sliding windows - for score distribution check

       Note: Predictions that overlap with rRNA operon are mentioned in the note qualifier

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       The manuscript describing the alien_hunter  algorithm  is  available  from  Bioinformatics:  Interpolated
       variable  order  motifs  for  identification  of  horizontally  acquired  DNA:  revisiting the Salmonella
       pathogenicity islands.  Vernikos GS, Parkhill J Bioinformatics. 2006;. PMID: 16837528

</pre><h4><b>AUTHOR</b></h4><pre>
       This manual page was written by Andreas Tille &lt;<a href="mailto:tille@debian.com">tille@debian.com</a>&gt; for the Debian(TM) system  (but  may  be
       used by others). This man page is released under the same conditions as the software, that is GPL.

alien_hunter 1.7                                  October 2009                                   <u><a href="../man1/ALIEN_HUNTER.1.html">ALIEN_HUNTER</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>