<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>mason_materializer - VCF Materialization</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/seqan-apps">seqan-apps_2.4.0+dfsg-16_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       mason_materializer - VCF Materialization

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>mason_materializer</b> [OPTIONS] <b>-ir</b> <u>IN.fa</u> <b>-iv</b> <u>IN.vcf</u> <b>-o</b> <u>OUT.fa</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Apply variants from <u>IN.vcf</u> to <u>IN.fa</u> and write the results to <u>out.fa</u>.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              Display the help message.

       <b>--version</b>
              Display version information.

       <b>-q</b>, <b>--quiet</b>
              Low verbosity.

       <b>-v</b>, <b>--verbose</b>
              Higher verbosity.

       <b>-vv</b>, <b>--very-verbose</b>
              Highest verbosity.

       <b>--seed</b> <u>INTEGER</u>
              Seed for random number generation. Default: <u>0</u>.

       <b>--meth-seed</b> <u>INTEGER</u>
              Seed for methylation simulation random number generation. Default: <u>0</u>.

       <b>-o</b>, <b>--out</b> <u>OUTPUT_FILE</u>
              Output  of  materialized  contigs.  Valid  filetypes  are:  <u>.sam[.*]</u>, <u>.raw[.*]</u>, <u>.frn[.*]</u>, <u>.fq[.*]</u>,
              <u>.fna[.*]</u>, <u>.ffn[.*]</u>, <u>.fastq[.*]</u>, <u>.fasta[.*]</u>, <u>.faa[.*]</u>, <u>.fa[.*]</u>, and <u>.bam</u>, where *  is  any  of  the
              following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent (de)compression.

       <b>--out-breakpoints</b> <u>OUTPUT_FILE</u>
              TSV file to write breakpoints in variants to. Valid filetypes are: <u>.txt</u> and <u>.tsv</u>.

       <b>--haplotype-name-sep</b> <u>STRING</u>
              String separating contig name from haplotype number. Default: <u>/</u>.

       <b>--meth-fasta-in</b> <u>INPUT_FILE</u>
              FASTA  file  with  methylation  levels of the input file. Valid filetypes are: <u>.sam[.*]</u>, <u>.raw[.*]</u>,
              <u>.gbk[.*]</u>, <u>.frn[.*]</u>,  <u>.fq[.*]</u>,  <u>.fna[.*]</u>,  <u>.ffn[.*]</u>,  <u>.fastq[.*]</u>,  <u>.fasta[.*]</u>,  <u>.faa[.*]</u>,  <u>.fa[.*]</u>,
              <u>.embl[.*]</u>, and <u>.bam</u>, where * is any of the following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent
              (de)compression.

       <b>--meth-fasta-out</b> <u>OUTPUT_FILE</u>
              FASTA  file  with  methylation levels of the output file. Valid filetypes are: <u>.sam[.*]</u>, <u>.raw[.*]</u>,
              <u>.frn[.*]</u>, <u>.fq[.*]</u>, <u>.fna[.*]</u>, <u>.ffn[.*]</u>, <u>.fastq[.*]</u>, <u>.fasta[.*]</u>, <u>.faa[.*]</u>, <u>.fa[.*]</u>, and <u>.bam</u>,  where
              * is any of the following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent (de)compression.

   <b>Apply</b> <b>VCF</b> <b>Variants</b> <b>to</b> <b>Reference:</b>
       <b>-ir</b>, <b>--input-reference</b> <u>INPUT_FILE</u>
              Path  to FASTA file to read the reference from. Valid filetypes are: <u>.sam[.*]</u>, <u>.raw[.*]</u>, <u>.gbk[.*]</u>,
              <u>.frn[.*]</u>, <u>.fq[.*]</u>, <u>.fna[.*]</u>, <u>.ffn[.*]</u>, <u>.fastq[.*]</u>, <u>.fasta[.*]</u>, <u>.faa[.*]</u>, <u>.fa[.*]</u>,  <u>.embl[.*]</u>,  and
              <u>.bam</u>,   where  *  is  any  of  the  following  extensions:  <u>gz</u>,  <u>bz2</u>,  and  <u>bgzf</u>  for  transparent
              (de)compression.

       <b>-iv</b>, <b>--input-vcf</b> <u>INPUT_FILE</u>
              Path to the VCF file with variants to apply. Valid filetype is: <u>.vcf[.*]</u>, where * is  any  of  the
              following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent (de)compression.

   <b>Methylation</b> <b>Level</b> <b>Simulation:</b>
       <b>--methylation-levels</b>
              Enable methylation level simulation.

       <b>--meth-cg-mu</b> <u>DOUBLE</u>
              Median of beta distribution for methylation level of CpG loci. In range [0..1]. Default: <u>0.6</u>.

       <b>--meth-cg-sigma</b> <u>DOUBLE</u>
              Standard  deviation  of  beta  distribution  for  methylation  level of CpG loci. In range [0..1].
              Default: <u>0.03</u>.

       <b>--meth-chg-mu</b> <u>DOUBLE</u>
              Median of beta distribution for methylation level of CHG loci. In range [0..1]. Default: <u>0.08</u>.

       <b>--meth-chg-sigma</b> <u>DOUBLE</u>
              Standard deviation of beta distribution for methylation  level  of  CHG  loci.  In  range  [0..1].
              Default: <u>0.008</u>.

       <b>--meth-chh-mu</b> <u>DOUBLE</u>
              Median of beta distribution for methylation level of CHH loci. In range [0..1]. Default: <u>0.05</u>.

       <b>--meth-chh-sigma</b> <u>DOUBLE</u>
              Standard  deviation  of  beta  distribution  for  methylation  level of CHH loci. In range [0..1].
              Default: <u>0.005</u>.

</pre><h4><b>VCF</b> <b>VARIANT</b> <b>NOTES</b></h4><pre>
       If the option <b>--input-vcf</b>/<b>-iv</b> is given then the given VCF file is read and the variants  are  applied  to
       the  input  reference  file.   If  it  is not given then the input reference file is taken verbatimly for
       simulating reads.

       There are some restrictions on the VCF file and the application of the variants  to  the  reference  will
       fail  if  the VCF file is non-conforming.  VCF files from the <u>mason_variator</u> program are guaranteed to be
       read.

       Only the haplotypes of the first individual will be generated.

mason_materializer 2.0.9 [tarball]                                                         <u><a href="../man1/MASON_MATERIALIZER.1.html">MASON_MATERIALIZER</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>