<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>VFDB_cdhit_to_csv - Parse cdhit results</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/srst2">srst2_0.2.0-13_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       VFDB_cdhit_to_csv - Parse cdhit results

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>VFDB_cdhit_to_csv</b> [-h] <b>--cluster_file</b> CLUSTER_FILE <b>--infile</b> INFILE <b>--outfile</b> OUTFILE

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This tool is part of the SRST2 suite.

       Output  a  comma-separated  table  of  alleles,  their inferred gene name (from annotation) and their 90%
       cluster from CD-HIT analysis

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>optional</b> <b>arguments:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>--cluster_file</b> CLUSTER_FILE
              cd hit output file (.clstr)

       <b>--infile</b> INFILE
              raw sequences file that was input to cdhit (fasta)

       <b>--outfile</b> OUTFILE
              output file (csv)

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

VFDB_cdhit_to_csv 0.1.8                             June 2016                               <u><a href="../man1/VFDB_CDHIT_TO_CSV.1.html">VFDB_CDHIT_TO_CSV</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>