<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>fasttreeMP  -  create  phylogenetic  trees  from  alignments  of  nucleotide or protein sequences (openMP</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/fasttree">fasttree_2.1.11-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       fasttreeMP  -  create  phylogenetic  trees  from  alignments  of  nucleotide or protein sequences (openMP
       version)

</pre><h4><b>DESCRIPTION</b></h4><pre>
       fasttreeMP infers approximately-maximum-likelihood phylogenetic trees from alignments  of  nucleotide  or
       protein sequences. It handles alignments with up to a million of sequences in a reasonable amount of time
       and memory.

       fasttreeMP is more accurate than PhyML 3 with default settings, and much more accurate than the distance-
       matrix  methods  that  are  traditionally  used for large alignments. fasttreeMP uses the Jukes-Cantor or
       generalized time-reversible (GTR) models of nucleotide evolution and the JTT (Jones-Taylor-Thornton 1992)
       model of amino acid evolution. To account for the varying rates of  evolution  across  sites,  fasttreeMP
       uses  a  single rate for each site (the "CAT" approximation). To quickly estimate the reliability of each
       split in the tree, fasttreeMP computes local support values with the Shimodaira-Hasegawa test (these  are
       the same as PhyML 3's "SH-like local supports").

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>fasttreeMP</b> <b>protein_alignment</b> <b>&gt;</b> <b>tree</b>

       <b>fasttreeMP</b> <b>-nt</b> nucleotide_alignment &gt; tree

       <b>fasttreeMP</b> <b>-nt</b> <b>-gtr</b> &lt; nucleotide_alignment &gt; tree

       fasttreeMP accepts alignments in fasta or phylip interleaved formats

   <b>Common</b> <b>options</b> <b>(must</b> <b>be</b> <b>before</b> <b>the</b> <b>alignment</b> <b>file):</b>

       <b>-quiet</b> to suppress reporting information

       <b>-nopr</b> to suppress progress indicator

       <b>-log</b> logfile <b>--</b> save intermediate trees, settings, and model details

       <b>-fastest</b> <b>--</b> speed up the neighbor joining phase &amp; reduce memory usage

              (recommended for &gt;50,000 sequences)

       <b>-n</b> &lt;number&gt; to analyze multiple alignments (phylip format only)

              (use for global bootstrap, with seqboot and CompareToBootstrap.pl)

       <b>-nosupport</b> to not compute support values

       <b>-intree</b> newick_file to set the starting tree(s)

       <b>-intree1</b> newick_file to use this starting tree for all the alignments

              (for faster global bootstrap on huge alignments)

       <b>-pseudo</b> to use pseudocounts (recommended for highly gapped sequences)

       <b>-gtr</b> <b>--</b> generalized time-reversible model (nucleotide alignments only)

       <b>-wag</b> <b>--</b> Whelan-And-Goldman 2001 model (amino acid alignments only)

       <b>-quote</b> <b>--</b> allow spaces and other restricted characters (but not ' characters) in

              sequence  names  and quote names in the output tree (fasta input only; fasttreeMP will not be able
              to read these trees back in

       <b>-noml</b> to turn off maximum-likelihood

       <b>-nome</b> to turn off minimum-evolution NNIs and SPRs

              (recommended if running additional ML NNIs with <b>-intree</b>)

       <b>-nome</b> <b>-mllen</b> with <b>-intree</b> to optimize branch lengths for a fixed topology

       <b>-cat</b> # to specify the number of rate categories of sites (default 20)

              or <b>-nocat</b> to use constant rates

       <b>-gamma</b> <b>--</b> after optimizing the tree under the CAT approximation,

              rescale the lengths to optimize the Gamma20 likelihood

       <b>-constraints</b> constraintAlignment to constrain the topology search

              constraintAlignment should have 1s or 0s to indicates splits

       <b>-expert</b> <b>--</b> see more options

   <b>Detailed</b> <b>usage</b> <b>for</b> <b>fasttreeMP</b> <b>2.1.4</b> <b>SSE3:</b>
       fasttreeMP [-nt] [-n 100] [-quote] [-pseudo | <b>-pseudo</b> 1.0]

              [-boot 1000 | <b>-nosupport]</b> [-intree starting_trees_file | <b>-intree1</b>  starting_tree_file]  [-quiet  |
              <b>-nopr]</b>  [-nni  10]  [-spr  2]  [-noml | <b>-mllen</b> | <b>-mlnni</b> 10] [-mlacc 2] [-cat 20 | <b>-nocat]</b> [-gamma]
              [-slow | <b>-fastest]</b> [-2nd | <b>-no2nd]</b> [-slownni] [-seed 1253] [-top  |  <b>-notop]</b>  [-topm  1.0  [-close
              0.75]  [-refresh  0.8]] [-matrix Matrix | <b>-nomatrix]</b> [-nj | <b>-bionj]</b> [-wag] [-nt] [-gtr] [-gtrrates
              ac ag at cg ct gt] [-gtrfreq A C G T] [ <b>-constraints</b> constraintAlignment [ <b>-constraintWeight</b> 100.0
              ] ] [-log logfile]

              [ alignment_file ]

              &gt; newick_tree

       or

       fasttreeMP [-nt] [-matrix Matrix | <b>-nomatrix]</b> [-rawdist] <b>-makematrix</b> [alignment]

              [-n 100] &gt; phylip_distance_matrix

              fasttreeMP supports fasta or phylip interleaved alignments By default fasttreeMP  expects  protein
              alignments,  use <b>-nt</b> for nucleotides fasttreeMP reads standard input if no alignment file is given

   <b>Input/output</b> <b>options:</b>

       <b>-n</b> <b>--</b> read in multiple alignments in. This only

              works  with  phylip  interleaved format. For example, you can use it with the output from phylip's
              seqboot. If you use <b>-n</b>, fasttreeMP will write 1 tree per line to standard output.

       <b>-intree</b> newickfile <b>--</b> read the starting tree in from newickfile.

              Any branch lengths in the starting trees are ignored.

       <b>-intree</b> with <b>-n</b> will read a separate starting tree for each alignment.

       <b>-intree1</b> newickfile <b>--</b> read the same starting tree for each alignment

       <b>-quiet</b> <b>--</b> do not write to standard error during normal operation (no progress

              indicator, no options summary, no likelihood values, etc.)

       <b>-nopr</b> <b>--</b> do not write the progress indicator to stderr

       <b>-log</b> logfile <b>--</b> save intermediate trees so you can extract

              the trees and restart long-running jobs if they crash <b>-log</b> also  reports  the  per-site  rates  (1
              means slowest category)

       <b>-quote</b> <b>--</b> quote sequence names in the output and allow spaces, commas,

              parentheses, and colons in them but not ' characters (fasta files only)

   <b>Distances:</b>
              Default: For protein sequences, log-corrected distances and an

              amino acid dissimilarity matrix derived from BLOSUM45

              or  for  nucleotide  sequences,  Jukes-Cantor distances To specify a different matrix, use <b>-matrix</b>
              FilePrefix or <b>-nomatrix</b> Use <b>-rawdist</b> to turn the log-correction off or to use  %different  instead
              of Jukes-Cantor

       <b>-pseudo</b> [weight] <b>--</b> Use pseudocounts to estimate distances between

              sequences  with little or no overlap. (Off by default.) Recommended if analyzing the alignment has
              sequences with little or no overlap.  If the weight is not specified, it is 1.0

   <b>Topology</b> <b>refinement:</b>
              By default, fasttreeMP tries to improve the tree with up to 4*log2(N) rounds of  minimum-evolution
              nearest-neighbor  interchanges  (NNI),  where  N  is  the  number of unique sequences, 2 rounds of
              subtree-prune-regraft  (SPR)  moves  (also  min.   evo.),   and   up   to   2*log(N)   rounds   of
              maximum-likelihood  NNIs.  Use <b>-nni</b> to set the number of rounds of min. evo. NNIs, and <b>-spr</b> to set
              the rounds of SPRs.  Use <b>-noml</b> to turn off both min-evo NNIs and SPRs (useful if refining

              an approximately maximum-likelihood tree with further NNIs)

              Use <b>-sprlength</b> set the maximum length of a SPR move (default 10) Use <b>-mlnni</b> to set the  number  of
              rounds  of  maximum-likelihood  NNIs Use <b>-mlacc</b> 2 or <b>-mlacc</b> 3 to always optimize all 5 branches at
              each NNI,

              and to optimize all 5 branches in 2 or 3 rounds

              Use <b>-mllen</b> to optimize branch lengths without ML NNIs Use <b>-mllen</b> <b>-nome</b> with  <b>-intree</b>  to  optimize
              branch  lengths on a fixed topology Use <b>-slownni</b> to turn off heuristics to avoid constant subtrees
              (affects both

              ML and ME NNIs)

   <b>Maximum</b> <b>likelihood</b> <b>model</b> <b>options:</b>

       <b>-wag</b> <b>--</b> Whelan-And-Goldman 2001 model instead of (default) Jones-Taylor-Thorton 1992 model (a.a. only)

       <b>-gtr</b> <b>--</b> generalized time-reversible instead of (default) Jukes-Cantor (nt only)

       <b>-cat</b> # <b>--</b> specify the number of rate categories of sites (default 20)

       <b>-nocat</b> <b>--</b> no CAT model (just 1 category)

       <b>-gamma</b> <b>--</b> after the final round of optimizing branch lengths with the CAT model,

              report the likelihood under  the  discrete  gamma  model  with  the  same  number  of  categories.
              fasttreeMP  uses  the same branch lengths but optimizes the gamma shape parameter and the scale of
              the lengths.  The final tree will have rescaled lengths.  Used  with  <b>-log</b>,  this  also  generates
              per-site  likelihoods  for  use  with  CONSEL,  see  GammaLogToPaup.pl  and  documentation  on the
              fasttreeMP web site.

   <b>Support</b> <b>value</b> <b>options:</b>
              By default, fasttreeMP computes local support values by  resampling  the  site  likelihoods  1,000
              times  and  the  Shimodaira Hasegawa test. If you specify <b>-nome</b>, it will compute minimum-evolution
              bootstrap supports instead In either case, the support values are proportions ranging from 0 to 1

              Use <b>-nosupport</b> to turn off support values or <b>-boot</b> 100 to use just  100  resamples  Use  <b>-seed</b>  to
              initialize the random number generator

   <b>Searching</b> <b>for</b> <b>the</b> <b>best</b> <b>join:</b>
              By default, fasttreeMP combines the 'visible set' of fast neighbor-joining with

              local hill-climbing as in relaxed neighbor-joining

       <b>-slow</b> <b>--</b> exhaustive search (like NJ or BIONJ, but different gap handling)

       <b>-slow</b> takes half an hour instead of 8 seconds for 1,250 proteins

       <b>-fastest</b> <b>--</b> search the visible set (the top hit for each node) only

              Unlike  the  original  fast neighbor-joining, <b>-fastest</b> updates visible(C) after joining A and B if
              join(AB,C) is better than join(C,visible(C)) <b>-fastest</b> also updates out-distances in  a  very  lazy
              way, <b>-fastest</b> sets <b>-2nd</b> on as well, use <b>-fastest</b> <b>-no2nd</b> to avoid this

   <b>Top-hit</b> <b>heuristics:</b>
              By  default,  fasttreeMP uses a top-hit list to speed up search Use <b>-notop</b> (or <b>-slow</b>) to turn this
              feature off

              and compare all leaves to each other, and all new joined nodes to each other

       <b>-topm</b> 1.0 <b>--</b> set the top-hit list size to parameter*sqrt(N)

              fasttreeMP estimates the top m hits of a leaf from the top 2*m hits of a 'close'  neighbor,  where
              close  is  defined as d(seed,close) &lt; 0.75 * d(seed, hit of rank 2*m), and updates the top-hits as
              joins proceed

       <b>-close</b> 0.75 <b>--</b> modify the close heuristic, lower is more conservative

       <b>-refresh</b> 0.8 <b>--</b> compare a joined node to all other nodes if its

              top-hit list is less than 80% of the desired length, or if the age of the top-hit list is  <a href="../manm/log2.m.html">log2</a>(m)
              or greater

       <b>-2nd</b> or <b>-no2nd</b> to turn 2nd-level top hits heuristic on or off

              This  reduces  memory  usage and running time but may lead to marginal reductions in tree quality.
              (By default, <b>-fastest</b> turns on <b>-2nd</b>.)

   <b>Join</b> <b>options:</b>

       <b>-nj</b>: regular (unweighted) neighbor-joining (default)

       <b>-bionj</b>: weighted joins as in BIONJ

              fasttreeMP will also weight joins during NNIs

   <b>Constrained</b> <b>topology</b> <b>search</b> <b>options:</b>

       <b>-constraints</b> alignmentfile <b>--</b> an alignment with values of 0, 1, and -

              Not all sequences need be present. A column of  0s  and  1s  defines  a  constrained  split.  Some
              constraints may be violated (see 'violating constraints:' in standard error).

       <b>-constraintWeight</b> <b>--</b> how strongly to weight the constraints. A value of 1

              means a penalty of 1 in tree length for violating a constraint Default: 100.0

       For more information, see <a href="http://www.microbesonline.org/fasttree/">http://www.microbesonline.org/fasttree/</a>

              or the comments in the source code

              fasttreeMP protein_alignment &gt; tree fasttreeMP <b>-nt</b> nucleotide_alignment &gt; tree fasttreeMP <b>-nt</b> <b>-gtr</b>
              &lt; nucleotide_alignment &gt; tree

       fasttreeMP accepts alignments in fasta or phylip interleaved formats

   <b>Common</b> <b>options</b> <b>(must</b> <b>be</b> <b>before</b> <b>the</b> <b>alignment</b> <b>file):</b>

       <b>-quiet</b> to suppress reporting information

       <b>-nopr</b> to suppress progress indicator

       <b>-log</b> logfile <b>--</b> save intermediate trees, settings, and model details

       <b>-fastest</b> <b>--</b> speed up the neighbor joining phase &amp; reduce memory usage

              (recommended for &gt;50,000 sequences)

       <b>-n</b> &lt;number&gt; to analyze multiple alignments (phylip format only)

              (use for global bootstrap, with seqboot and CompareToBootstrap.pl)

       <b>-nosupport</b> to not compute support values

       <b>-intree</b> newick_file to set the starting tree(s)

       <b>-intree1</b> newick_file to use this starting tree for all the alignments

              (for faster global bootstrap on huge alignments)

       <b>-pseudo</b> to use pseudocounts (recommended for highly gapped sequences)

       <b>-gtr</b> <b>--</b> generalized time-reversible model (nucleotide alignments only)

       <b>-wag</b> <b>--</b> Whelan-And-Goldman 2001 model (amino acid alignments only)

       <b>-quote</b> <b>--</b> allow spaces and other restricted characters (but not ' characters) in

              sequence  names  and quote names in the output tree (fasta input only; fasttreeMP will not be able
              to read these trees back in

       <b>-noml</b> to turn off maximum-likelihood

       <b>-nome</b> to turn off minimum-evolution NNIs and SPRs

              (recommended if running additional ML NNIs with <b>-intree</b>)

       <b>-nome</b> <b>-mllen</b> with <b>-intree</b> to optimize branch lengths for a fixed topology

       <b>-cat</b> # to specify the number of rate categories of sites (default 20)

              or <b>-nocat</b> to use constant rates

       <b>-gamma</b> <b>--</b> after optimizing the tree under the CAT approximation,

              rescale the lengths to optimize the Gamma20 likelihood

       <b>-constraints</b> constraintAlignment to constrain the topology search

              constraintAlignment should have 1s or 0s to indicates splits

       <b>-expert</b> <b>--</b> see more options

       For more information, see <a href="http://www.microbesonline.org/fasttree/">http://www.microbesonline.org/fasttree/</a>

Lawrence Berkeley National Lab                      June 2012                                      <u><a href="../man1/fasttreeMP.1.html">fasttreeMP</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>