<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>VeryFastTree   -  Speeding  up  the  estimation  of  phylogenetic  trees  for  large  alignments  through</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/veryfasttree">veryfasttree_4.0.4+dfsg-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       VeryFastTree   -  Speeding  up  the  estimation  of  phylogenetic  trees  for  large  alignments  through
       parallelization and vectorization strategies

</pre><h4><b>DESCRIPTION</b></h4><pre>
       VeryFastTree is a highly efficient implementation inspired by the FastTree-2 tool, designed  to  expedite
       the  inference of approximately-maximum-likelihood phylogenetic trees from nucleotide or protein sequence
       alignments. It is an optimized implementation designed to accelerate the estimation  of  phylogenies  for
       large  alignments. By leveraging parallelization and vectorization strategies, VeryFastTree significantly
       improves the performance and scalability of phylogenetic analysis, allowing it to construct  phylogenetic
       trees in a fraction of the time previously required.

       Maintaining  the  integrity  of FastTree-2, VeryFastTree retains the same phases, methods, and heuristics
       used for estimating phylogenetic trees. This ensures that the topological accuracy of the trees  produced
       by  VeryFastTree  remains  equivalent  to  that  of  FastTree-2. Moreover, unlike the parallel version of
       FastTree-2, VeryFastTree guarantees deterministic results, eliminating any potential  variations  in  the
       output.

       To  facilitate a seamless transition for users, VeryFastTree adopts the exact same command line arguments
       as FastTree-2. This means that by simply substituting FastTree-2 with VeryFastTree, and  using  the  same
       set of options, users can significantly enhance the overall performance of their phylogenetic analyses.

       VeryFastTree  is  more  accurate  than  PhyML  3  with  default settings, and much more accurate than the
       distance-matrix methods that are traditionally used for large alignments. VeryFastTree  uses  the  Jukes-
       Cantor  or  generalized  time-reversible  (GTR) models of nucleotide evolution and the JTT (Jones-Taylor-
       Thornton 1992) model of amino acid evolution. To account for the varying rates of evolution across sites,
       VeryFastTree uses a single rate for  each  site  (the  "CAT"  approximation).  To  quickly  estimate  the
       reliability  of  each  split in the tree, VeryFastTree computes local support values with the Shimodaira-
       Hasegawa test (these are the same as PhyML 3's "SH-like local supports").

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>VeryFastTree</b> <b>protein_alignment</b> <b>&gt;</b> <b>tree</b>

       <b>VeryFastTree</b> <b>-nt</b> nucleotide_alignment &gt; tree

       <b>VeryFastTree</b> <b>-nt</b> <b>-gtr</b> &lt; nucleotide_alignment &gt; tree

       accepts alignments in NEXUS, Fasta, Fastq or Phylip interleaved formats compressed with ZLib and libBZ2.

   <b>Common</b> <b>options:</b>

       <b>-quiet</b> to suppress reporting information

       <b>-nopr</b> to suppress progress indicator

       <b>-log</b> logfile <b>--</b> save intermediate trees, settings, and model details

       <b>-fastest</b> <b>--</b> speed up the neighbor joining phase &amp; reduce memory usage (recommended for &gt;50,000 sequences)

       <b>-n</b> &lt;number&gt; to analyze multiple alignments (phylip format only) (use for global bootstrap,  with  seqboot
              and CompareToBootstrap.pl)

       <b>-nosupport</b> to not compute support values

       <b>-intree</b> newick_file to set the starting tree(s)

       <b>-intree1</b>  newick_file  to  use  this starting tree for all the alignments (for faster global bootstrap on
              huge alignments)

       <b>-pseudo</b> to use pseudocounts (recommended for highly gapped sequences)

       <b>-gtr</b> <b>--</b> generalized time-reversible model (nucleotide alignments only)

       <b>-wag</b> <b>--</b> Whelan-And-Goldman 2001 model (amino acid alignments only)

       sequence names and quote names in the output tree (fasta input only; VeryFastTree will  not  be  able  to
              read these trees back in

       <b>-noml</b> <b>--</b> to turn off maximum-likelihood

       <b>-nome</b>  <b>--</b>  to  turn  off  minimum-evolution NNIs and SPRs (recommended if running additional ML NNIs with
              <b>-intree</b>)

       <b>-nome</b> <b>-mllen</b> with <b>-intree</b> to optimize branch lengths for a fixed topology

       <b>-cat</b> # to specify the number of rate categories of sites (default 20) or <b>-nocat</b> to use constant rates

       <b>-gamma</b> <b>--</b> after optimizing the tree under the CAT approximation, rescale  the  lengths  to  optimize  the
              Gamma20 likelihood

       <b>-constraints</b>  constraintAlignment  to constrain the topology search constraintAlignment should have 1s or
              0s to indicates splits <b>-threads</b> &lt;number&gt; (Env:OMP_NUM_THREADS) number of threads (n) used  in  the
              parallel execution

       <b>-double-precision</b>  <b>--</b>  to  use  double  precision  arithmetic.   Therefore,  it  is equivalent to compile
              FastTree-2 with -DUSE_DOUBLE.

       <b>-ext</b> <b>name</b> to speed up computations enabling the vector extensions. Available: AUTO(default),  NONE,  SSE,
              SSE3 , AVX, AVX2, AVX512 or CUDA

       <b>-expert</b> <b>--</b> see more options

   <b>Detailed</b> <b>usage</b> <b>for</b> <b>VeryFastTree:</b>
       VeryFastTree  [-nt]  [-n  100]  [-quote]  [-pseudo  |  -pseudo  1.0]  [-boot  1000 | -nosupport] [-intree
       starting_trees_file | -intree1 starting_tree_file] [-quiet | -nopr] [-nni 10] [-spr 2] [-noml | -mllen  |
       -mlnni  10]  [-mlacc  2] [-cat 20 | -nocat] [-gamma] [-slow | -fastest] [-2nd | -no2nd] [-slownni] [-seed
       1253] [-top | -notop] [-topm 1.0 [-close 0.75] [-refresh 0.8]] [-gtr] [-gtrrates  ac  ag  at  cg  ct  gt]
       [-gtrfreq  A  C  G  T]  [  -lg | -wag | -trans transitionmatrixfile ] [-matrix Matrix | -nomatrix] [-nj |
       -bionj] [ -constraints constraintAlignment [ -constraintWeight 100.0 ] ] [-log logfile] [  alignment_file
       ]  [  -threads 1 ] [ -threads-level 3 [ -threads-ptw 20 ] [-threads-verbose] [ -double-precision ] [ -ext
       AUTO ] [ -fastexp 0 ] [ -disk-computing ] [ -disk-computing-path ./  ]  [  -disk-dynamic-limit  inf  ]  [
       -relative-progress ] [ -out output_newick_file | &gt; newick_tree]

       or

       VeryFastTree   [-nt]  [-matrix  Matrix  |  <b>-nomatrix]</b>  [-rawdist]  <b>-makematrix</b>  [alignment]  [-n  100]  &gt;
       phylip_distance_matrix

       <b>VeryFastTree</b> <b>supports</b> <b>NEXUS,</b> <b>Fasta,</b> <b>Fastq</b> <b>or</b> <b>Phylip</b> <b>interleaved</b> <b>formats</b>

       <b>VeryFastTree</b> <b>supports</b> <b>files</b> <b>compressed</b> <b>with</b> <b>ZLib</b> <b>and</b> <b>libBZ2</b>

       <b>By</b> <b>default</b> <b>VeryFastTree</b> <b>expects</b> <b>protein</b> <b>alignments,</b> <b>use</b> <b>-nt</b> <b>for</b> <b>nucleotides</b>

       <b>VeryFastTree</b> <b>reads</b> <b>standard</b> <b>input</b> <b>if</b> <b>no</b> <b>alignment</b> <b>file</b> <b>is</b> <b>given</b>

   <b>Input/output</b> <b>options:</b>

       <b>-out</b> file print tree in output file instead of stdout

       <b>-n</b> &lt;number&gt; read in multiple alignments in. This only works with phylip interleaved format. For  example,
              you  can  use  it  with the output from phylip's seqboot. If you use <b>-n</b>, VeryFastTree will write 1
              tree per line to standard output.

       <b>-nt</b> <b>--</b> nucleotides instead of protein alignments

       <b>-intree</b> newickfile read the starting tree in from newickfile.  Any branch lengths in the  starting  trees
              are ignored.

       <b>-intree</b> with <b>-n</b> will read a separate starting tree for each alignment.

       <b>-intree1</b> newickfile read the same starting tree for each alignment

       <b>-verbose</b> lvl level of details during normal operation

       <b>-quiet</b>  <b>--</b>  do  not  write  to  standard error during normal operation (no progress indicator, no options
              summary, no likelihood values, etc.)

       <b>-nopr</b> <b>--</b> do not write the progress indicator to stderr

       <b>-log</b> logfile <b>--</b> save intermediate trees so you can extract the trees and  restart  long-running  jobs  if
              they crash <b>-log</b> also reports the per-site rates (1 means slowest category)

       <b>-quote</b>  <b>--</b>  quote  sequence names in the output and allow spaces, commas, parentheses, and colons in them
              but not ' characters (fasta files only)

   <b>Distances:</b>
              Default: For protein sequences, log-corrected distances and an  amino  acid  dissimilarity  matrix
              derived from BLOSUM45 or for nucleotide sequences, Jukes-Cantor distances

              To specify a different matrix, use -matrix FilePrefix or -nomatrix

              Use  -rawdist  to  turn the log-correction off or to use %different instead of Jukes-Cantor (These
              options affect minimum-evolution computations  only;  use  -trans  to  affect  maximum-likelihoood
              computations)

       <b>-makematrix</b> <b>--</b> print distance matrix

       <b>-rawdist</b> <b>--</b> to turn the log-correction off

       <b>-matrix</b> file to turn the log-correction off

       <b>-nomatrix</b> <b>--</b> to turn the log-correction off

       <b>-pseudo</b> [weight] Use pseudocounts to estimate distances between sequences with little or no overlap. (Off
              by  default.)  Recommended if analyzing the alignment has sequences with little or no overlap.  If
              the weight is not specified, it is 1.0

   <b>Topology</b> <b>refinement:</b>
              By  default,  VeryFastTree  tries  to  improve  the  tree  with  up   to   4*log2(N)   rounds   of
              minimum-evolution  nearest-neighbor interchanges (NNI), where N is the number of unique sequences,
              2 rounds of subtree-prune-regraft (SPR) moves (also min. evo.),  and  up  to  2*log(N)  rounds  of
              maximum-likelihood NNIs.

       <b>-nni</b> &lt;number&gt; to set the number of rounds of min. evo. NNIs

       <b>-spr</b> &lt;number&gt; to set the rounds of SPRs

       <b>-noml</b>  <b>--</b>  to turn off both min-evo NNIs and SPRs (useful if refining an approximately maximum-likelihood
              tree with further NNIs)

       <b>-sprlength</b> &lt;number&gt; set the maximum length of a SPR move (default 10)

       <b>-mlnni</b> &lt;number&gt; to set the number of rounds of maximum-likelihood NNIs

       <b>-mlacc</b> &lt;number&gt; Use -mlacc 2 or -mlacc 3 to always optimize all 5 branches at each NNI, and  to  optimize
              all 5 branches in 2 or 3 rounds

       <b>-mllen</b>  <b>--</b>  to  optimize branch lengths without ML NNIs. Use -mllen -nome with -intree to optimize branch
              lengths on a fixed topology

       <b>-approxml,-mlapprox</b> <b>--</b> approximate posterior distributions for a.a.s

       <b>-slownni</b> <b>--</b> to optimize branch lengths without ML NNIs.

   <b>Maximum</b> <b>likelihood</b> <b>model</b> <b>options:</b>

       <b>-lg</b> <b>--</b> Le-Gascuel 2008 model instead of (default) Jones-Taylor-Thorton 1992 model (a.a. only)

       <b>-wag</b> <b>--</b> Whelan-And-Goldman 2001 model instead of (default) Jones-Taylor-Thorton 1992 model (a.a. only)

       <b>-gtr</b> <b>--</b> generalized time-reversible instead of (default) Jukes-Cantor (nt only)

       <b>-gtrrates</b> rates set the gtr rates

       <b>-gtrfreq</b> seqs set the gtr frequences

       <b>-cat</b> # <b>--</b> specify the number of rate categories of sites (default 20)

       <b>-nocat</b> <b>--</b> no CAT model (just 1 category)

       <b>-trans</b> filename use the transition matrix from filename.This  is  supported  for  amino  acid  alignments
              only.The file must be tab-delimited with columns in the order ARNDCQEGHILKMFPSTWYV*.The additional
              column  named * is for the stationary distribution.Each row must have a row name in the same order
              ARNDCQEGHILKMFPSTWYV

       <b>-gamma</b> <b>--</b> after the final round of optimizing branch lengths with the CAT model,  report  the  likelihood
              under  the  discrete  gamma  model  with the same number of categories. VeryFastTree uses the same
              branch lengths but optimizes the gamma shape parameter and the scale of the  lengths.   The  final
              tree  will have rescaled lengths. Used with <b>-log</b>, this also generates per-site likelihoods for use
              with CONSEL, see GammaLogToPaup.pl and documentation on the VeryFastTree web site.

   <b>Support</b> <b>value</b> <b>options:</b>
              By default, VeryFastTree computes local support values by resampling the  site  likelihoods  1,000
              times  and  the  Shimodaira Hasegawa test. If you specify <b>-nome</b>, it will compute minimum-evolution
              bootstrap supports instead In either case, the support values are proportions ranging from 0 to 1

       <b>-nome</b> <b>--</b> to compute minimum-evolution bootstrap supports

       <b>-nosupport</b> <b>--</b> to turn off support values

       <b>-boot</b> &lt;number&gt; to use just n resamples

       <b>-noboot</b> <b>--</b>  to no use resamples

       <b>-seed</b> &lt;number&gt; to initialize the random number generator

   <b>Searching</b> <b>for</b> <b>the</b> <b>best</b> <b>join:</b>
       By default, VeryFastTree combines the 'visible set' of fast neighbor-joining with local hill-climbing  as
       in relaxed neighbor-joining

       <b>-slow</b>  <b>--</b>  exhaustive  search  (like  NJ  or  BIONJ, but different gap handling) -slow takes half an hour
              instead of 8 seconds for 1,250 proteins

       <b>-fastest</b> <b>--</b> search the  visible  set  (the  top  hit  for  each  node)  only  Unlike  the  original  fast
              neighbor-joining,  <b>-fastest</b>  updates visible(C) after joining A and B if join(AB,C) is better than
              join(C,visible(C)) <b>-fastest</b> also updates out-distances in a very lazy way, <b>-fastest</b> sets  <b>-2nd</b>  on
              as well, use <b>-fastest</b> <b>-no2nd</b> to avoid this

   <b>Top-hit</b> <b>heuristics:</b>
              By default, VeryFastTree uses a top-hit list to speed up search

       <b>-top</b>  0.01  set  the  top-hit  list size to 1.0 if it is less than 0.01 -notop <b>--</b> (or -slow) to turn this
              feature off and compare all leaves to each other, and all new joined nodes to each other

       <b>-topm</b> 1.0 <b>--</b> set the top-hit list size to parameter*sqrt(N) VeryFastTree estimates the top m  hits  of  a
              leaf from the top 2*m hits of a 'close' neighbor, where close is defined as d(seed,close) &lt; 0.75 *
              d(seed, hit of rank 2*m), and updates the top-hits as joins proceed

       <b>-close</b> 0.75 <b>--</b> modify the close heuristic, lower is more conservative

       <b>-refresh</b>  0.8  <b>--</b>  compare  a  joined node to all other nodes if its top-hit list is less than 80% of the
              desired length, or if the age of the top-hit list is <a href="../manm/log2.m.html">log2</a>(m) or greater

       <b>-2nd</b> or <b>-no2nd</b> to turn 2nd-level top hits heuristic on or off This reduces memory usage and running  time
              but may lead to marginal reductions in tree quality.  (By default, <b>-fastest</b> turns on <b>-2nd</b>.)

   <b>Join</b> <b>options:</b>

       <b>-nj</b>: regular (unweighted) neighbor-joining (default)

       <b>-bionj</b>: weighted joins as in BIONJ VeryFastTree will also weight joins during NNIs

   <b>Constrained</b> <b>topology</b> <b>search</b> <b>options:</b>

       <b>-constraints</b>  alignmentfile  an alignment with values of 0, 1, and - Not all sequences need be present. A
              column of 0s and 1s defines a constrained split. Some constraints may be violated (see  'violating
              constraints:' in standard error).

       <b>-constraintWeight</b>  w  how  strongly  to weight the constraints. A value of 1 means a penalty of 1 in tree
              length for violating a constraint Default: 100.0

   <b>Optimizations:</b>

       <b>-threads</b> &lt;number&gt; (Env:OMP_NUM_THREADS) number of threads used in the parallel execution. If this  option
              is   not   set,   the   corresponding  value  will  be  obtained  from  the  environment  variable
              OMP_NUM_THREADS. This is the same approach followed by FastTree-2. If n=1, VeryFastTree behaves in
              the same way than FastTree-2 compiled without the -DOPENMP flag

       <b>-threads-level</b> &lt;number&gt; in [0 - 4] degree of parallelization. If level is 0, VeryFastTree uses  the  same
              parallelization  strategy as FastTree-2 with some new parallel blocks. If level is 1, VeryFastTree
              uses parallel blocks that require additional memory for computation. If level is  2,  VeryFastTree
              accelerates  the  rounds  of  ML NNIs using its tree partitioning method. If level is 3 (default),
              VeryFastTree performs more computations without  preserving  sequential  order.  If  level  is  4,
              VeryFastTree  accelerates  the rounds of SPR steps using its tree partitioning method (it can only
              be used with datasets larger than 2^sprlength + 2).

       Note: Each level includes the previous ones, and computation at level 2  and  above  is  performed  in  a
       different tree traverse order, so the result may change but is still correct

       <b>-threads-mode</b>  &lt;number&gt;  in [0 - 1] changes the mode of parallelization. If level is 0, VeryFastTree uses
              non-deterministic parts, some inspired by FastTree-2  but  improved.  If  level  is  1  (default),
              VeryFastTree only uses deterministic parallelization.  Since version 4.0, deterministic algorithms
              are at least faster than non-deterministic ones, making deterministic the preferred choice

       <b>-threads-ptw</b> &lt;number&gt; (Partitioning Tendency Window) It sets the size of the partitioning tendency window
              used by the tree partitioning algorithm to determine when to stop searching. The window stores the
              last  solutions  and  checks  if  a  better solution can be found. Increasing the value allows the
              algorithm to explore the tree deeper and potentially find better solutions. The default  value  is
              20.

       <b>-threads-verbose</b> <b>--</b> to show subtrees assigned to the threads and theoretical
               speedup, only with verbose &gt; 0

       <b>-double-precision</b>  <b>--</b>  to  use  double  precision  arithmetic.  Therefore,  it  is  equivalent to compile
              FastTree-2 with -DUSE_DOUBLE

       <b>-ext</b> name to speed up computations enabling the vector extensions.  Available: AUTO(default), NONE,  SSE,
              SSE3, AVX, AVX2, AVX512 or CUDA

       <b>-fastexp</b> &lt;number&gt; in [0 - 3] to select an alternative implementation for the exponential function exp(x),
              which  has  a  significant  impact on performance.  Options: 0 - built-in math library with double
              precision (default), 1 - built-in  math  library  with  simple  precision  (not  recommended  with
              -double-precision  option),  2  -  fast implementation to compute an approximation of exp(x) using
              double precision, and 3 - fast implementation to compute an approximation of exp(x)  using  simple
              precision (not recommended with -double-precision option)

       <b>-disk-computing</b>  <b>--</b> if there is not enough available RAM to perform the computation, disk will be used to
              store extra data when it was not needed. Using disk to perform the computation will  substantially
              increase the execution time

       <b>-disk-computing-path</b> path like -disk-computing but using a custom path folder to store data

       <b>-disk-dynamic-computing</b>  <b>--</b>  by default, disk computing only creates files associated with static data in
              RAM, which means that there is no significant impact on performance as long as there is  available
              RAM.  This  option further reduces memory usage by storing dynamic data on disk.  However, even if
              there is enough RAM, it will have a negative impact on performance due to  the  constant  creation
              and deletion of files

       <b>-disk-dynamic-limit</b>  &lt;number&gt;  -disk-dynamic-computing can exceed the limit of memory-mapped file system.
              If 'memory mapping fails' errors occur, setting a limit will solve  the  problem.  In  Linux,  the
              limit  can  be  checked with 'sysctl vm.max_map_count'. It is important not to use the exact value
              and leave a small margin for other operations that require this feature

       <b>-relative-progress</b> <b>--</b> to shows relative time to previous step rather  than  absolute  time  in   progress
              report

   <b>Deprecated:</b>

       <b>-logdist</b> <b>--</b> use logarithmic distances, now on by default and obsolete

       <b>-exactml</b> <b>--</b> Exact  posterior distributions, now on by default and obsolete

       <b>-mlexact</b> <b>--</b> Exact posterior distributions, now on by default and obsolete

       For        more        information,        see        https://github.com/citiususc/veryfasttree        or
       <a href="http://www.microbesonline.org/fasttree/">http://www.microbesonline.org/fasttree/</a>

Univ. of Santiago de Compostela                     June 2023                                    <u><a href="../man1/veryfasttree.1.html">veryfasttree</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>