<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>filterBam - filter BAM file for use with AUGUSTUS tools</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/augustus">augustus_3.5.0+dfsg-4build7_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       filterBam - filter BAM file for use with AUGUSTUS tools

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>filterBam</b> --in in.bam --out out.bam [options]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The input file must be sorted lexicographically by 'queryname', with e.g.

       •   Convert the file into SAM format, sort lexicographically by queryname and convert back again into BAM
           format.
           E.g.
           <b>export</b> <b>LC_ALL=C</b>
           <b>samtools</b> <b>view</b> <b>-H</b> <b>file.bam</b> <b>&gt;</b> <b>file_sorted.sam</b>
           <b>samtools</b> <b>view</b> <b>file.bam</b> <b>|</b> <b>sort</b> <b>-k</b> <b>1</b> <b>&gt;&gt;</b> <b>file_sorted.sam</b>
           <b>samtools</b> <b>view</b> <b>-bS</b> <b>file_sorted.sam</b> <b>&gt;</b> <b>file_sorted.bam</b>
           [be aware: 'export LC_ALL=C' might be used because sort ignores characters like ':']
           Also, please bear in mind that this will require converting your BAM file into SAM.

       •   <b>samtools</b> and <b>bamtools</b> provide facilities to do the sorting, but they are not guaranteed to work
           because of the problem mentioned above.

       •   In the case of <b>samtools</b>, the command to sort by 'queryname' is:
           <b>samtools</b> <b>sort</b> <b>-n</b> <b>-o</b> <b>file_sorted.bam</b> <b>file.bam</b>
           For more information check the man page included in samtools distribution.

       •   bamtools can also sort bam files:
           <b>bamtools</b> <b>sort</b> <b>-byname</b> <b>-in</b> <b>file.bam</b> <b>-out</b> <b>file_sorted.bam</b>
           but only provides the option to do it by queryname.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Mandatory</b> <b>options</b>
       <b>-i</b>, <b>--in=in.bam</b>
           input file in BAM format

       <b>-o</b>, <b>--out=out.bam</b>
           output file in BAM format

   <b>Optional</b> <b>options</b>
       <b>-u</b>, <b>--uniq</b>
           keep only the best match, remove all matches, if the second best is not much worse

       <b>-q</b> <b>f</b>, <b>--uniqThresh=f</b>
           threshold % for uniq, second best must be lower than this fraction of best to keep the best match
           (default 0.96)

       <b>-b</b>, <b>--best</b>
           output all best matches that satisfy minId and minCover

       <b>-e</b> <b>n</b>, <b>--minId=n</b>
           minimal percentage of identity (default 92)

       <b>-c</b> <b>n</b>, <b>--minCover=n</b>
           minimal percentage of coverage of the query read (default 80)

       <b>-n</b>, <b>--noIntrons</b>

           do not allow longer gaps -for RNA-RNA alignments-
       <b>-l</b> <b>n</b>, <b>--insertLimit=n</b>
           maximum assumed size of inserts (default 10)

       <b>-s</b> <b>n</b>, <b>--maxSortesTest=n</b>
           test if input file is sorted by query name for this number of alignments (default 100000)

       <b>-p</b>, <b>--paired</b>
           require that paired reads are on opposite strands of same target. Requires alignment names to contain
           the suffixes /1,/2 or /f,/r.

       <b>-w</b>, <b>--pairwiseAlignments</b>
           use in case alignments were done in pairwise fashion

       <b>-x</b> <b>n</b>, <b>--maxIntronLen=n</b>
           maximal separation of paired reads (default 500000)

       <b>-d</b> <b>file</b>, <b>--pairBedFile=file</b>
           file name of pairedness coverage: a BED format file in which for each position the number of filtered
           read pairs is reported that contain the position in or between the reads

       <b>-g</b> <b>file</b>, <b>--commonGeneFile=file</b>
           file name in which to write cases where one read maps several different genes

       <b>-t</b> <b>n</b>, <b>--threads=n</b>
           use n threads for compression/decompression (default 1); available only if library SeqLib &gt;= 1.2 is
           used

       <b>-v</b>, <b>--verbose</b>
           output debugging info

       <b>-h</b>, <b>--help</b>
           produce help message.

</pre><h4><b>AUTHORS</b></h4><pre>
       AUGUSTUS was written by M. Stanke, O. Keller, S. König, L. Gerischer and L. Romoth.

</pre><h4><b>ADDITIONAL</b> <b>DOCUMENTATION</b></h4><pre>
       An exhaustive documentation can be found in the file /usr/share/doc/augustus/README.md.gz.

                                                                                                    <u><a href="../man1/FILTERBAM.1.html">FILTERBAM</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>