<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>samtools - Utilities for the Sequence Alignment/Map (SAM) format</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/samtools">samtools_1.21-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       samtools - Utilities for the Sequence Alignment/Map (SAM) format

</pre><h4><b>SYNOPSIS</b></h4><pre>
       samtools addreplacerg -r 'ID:fish' -r 'LB:1334' -r 'SM:alpha' -o output.bam input.bam

       samtools ampliconclip -b bed.file input.bam

       samtools ampliconstats primers.bed in.bam

       samtools bedcov aln.sorted.bam

       samtools calmd in.sorted.bam ref.fasta

       samtools cat out.bam in1.bam in2.bam in3.bam

       samtools collate -o aln.name_collated.bam aln.sorted.bam

       samtools consensus -o out.fasta in.bam

       samtools coverage aln.sorted.bam

       samtools cram-size -v -o out.size in.cram

       samtools depad input.bam

       samtools depth aln.sorted.bam

       samtools dict -a GRCh38 -s "Homo sapiens" ref.fasta

       samtools faidx ref.fasta

       samtools fasta input.bam &gt; output.fasta

       samtools fastq input.bam &gt; output.fastq

       samtools fixmate in.namesorted.sam out.bam

       samtools flags PAIRED,UNMAP,MUNMAP

       samtools flagstat aln.sorted.bam

       samtools fqidx ref.fastq

       samtools head in.bam

       samtools idxstats aln.sorted.bam

       samtools import input.fastq &gt; output.bam

       samtools index aln.sorted.bam

       samtools markdup in.algnsorted.bam out.bam

       samtools merge out.bam in1.bam in2.bam in3.bam

       samtools mpileup -f ref.fasta -r chr3:1,000-2,000 in1.bam in2.bam

       samtools phase input.bam

       samtools quickcheck in1.bam in2.cram

       samtools reference -o ref.fa in.cram

       samtools reheader in.header.sam in.bam &gt; out.bam

       samtools reset -o /tmp/reset.bam processed.bam

       samtools samples input.bam

       samtools sort -T /tmp/aln.sorted -o aln.sorted.bam aln.bam

       samtools split merged.bam

       samtools stats aln.sorted.bam

       samtools targetcut input.bam

       samtools tview aln.sorted.bam ref.fasta

       samtools view -bt ref_list.txt -o aln.bam aln.sam.gz

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Samtools  is  a set of utilities that manipulate alignments in the SAM (Sequence Alignment/Map), BAM, and
       CRAM formats.  It converts between the formats, does sorting, merging  and  indexing,  and  can  retrieve
       reads in any regions swiftly.

       Samtools  is designed to work on a stream. It regards an input file `-' as the standard input (stdin) and
       an output file `-' as the standard output (stdout). Several commands  can  thus  be  combined  with  Unix
       pipes. Samtools always output warning and error messages to the standard error output (stderr).

       Samtools  is  also  able  to  open  files  on  remote FTP or HTTP(S) servers if the file name starts with
       `ftp://', `http://', etc.  Samtools checks the current working directory for  the  index  file  and  will
       download  the index upon absence. Samtools does not retrieve the entire alignment file unless it is asked
       to do so.

       If an index is needed, samtools looks for the index suffix appended to the filename, and  if  that  isn't
       found it tries again without the filename suffix (for example <b>in.bam.bai</b> followed by <b>in.bai</b>).  However if
       an  index  is in a completely different location or has a different name, both the main data filename and
       index filename can be pasted together with <b>##idx##</b>.  For  example  <b>/data/in.bam##idx##/indices/in.bam.bai</b>
       may be used to explicitly indicate where the data and index files reside.

</pre><h4><b>COMMANDS</b></h4><pre>
       Each  command  has its own man page which can be viewed using e.g. <b>man</b> <b>samtools-view</b> or with a recent GNU
       man using <b>man</b> <b>samtools</b> <b>view</b>.  Below we have a brief summary of syntax and sub-command description.

       Options common to all sub-commands are documented below in the GLOBAL COMMAND OPTIONS section.

       <b>view</b>      samtools view [<u>options</u>] <u>in.sam</u>|<u>in.bam</u>|<u>in.cram</u> [<u>region</u>...]

                 With no options or regions specified, prints all alignments in the  specified  input  alignment
                 file  (in  SAM,  BAM,  or  CRAM  format)  to  standard  output in SAM format (with no header by
                 default).

                 You may specify one or more space-separated region specifications after the input  filename  to
                 restrict  output  to only those alignments which overlap the specified region(s). Use of region
                 specifications requires a coordinate-sorted and indexed input file.

                 Options exist to change the output format from SAM to BAM or CRAM, so this command also acts as
                 a file format conversion utility.

       <b>tview</b>     samtools tview [<b>-p</b> <u>chr:pos</u>] [<b>-s</b> <u>STR</u>] [<b>-d</b> <u>display</u>] &lt;in.sorted.bam&gt; [ref.fasta]

                 Text alignment viewer (based on the ncurses library). In the viewer, press  `?'  for  help  and
                 press  `g'  to check the alignment start from a region in the format like `chr10:10,000,000' or
                 `=10,000,000' when viewing the same reference sequence.

       <b>quickcheck</b>
                 samtools quickcheck [<u>options</u>] <u>in.sam</u>|<u>in.bam</u>|<u>in.cram</u> [ ... ]

                 Quickly check that input files appear to be intact. Checks that beginning of the file  contains
                 a  valid header (all formats) containing at least one target sequence and then seeks to the end
                 of the file and checks that an end-of-file (EOF) is present and intact (BAM only).

                 Data in the middle of the file is not read since that would be much  more  time  consuming,  so
                 please  note  that  this command will not detect internal corruption, but is useful for testing
                 that files are not truncated before performing more intensive tasks on them.

                 This command will exit with a non-zero exit code if any input files don't have a  valid  header
                 or are missing an EOF block. Otherwise it will exit successfully (with a zero exit code).

       <b>head</b>      samtools head [<u>options</u>] <u>in.sam</u>|<u>in.bam</u>|<u>in.cram</u>

                 Prints  the  input  file's  headers  and  optionally also its first few alignment records. This
                 command always displays the headers as they are in the file, never adding an extra  @PG  header
                 itself.

       <b>index</b>     samtools index [<b>-bc</b>] [<b>-m</b> <u>INT</u>] <u>aln.sam.gz</u>|<u>aln.bam</u>|<u>aln.cram</u> [<u>out.index</u>]

                 Index a coordinate-sorted SAM, BAM or CRAM file for fast random access.  Note for SAM this only
                 works  if  the file has been BGZF compressed first.  (Starting from Samtools 1.16, this command
                 can also be given several alignment filenames, which are indexed individually.)

                 This index is needed when <u>region</u> arguments are used to limit <b>samtools</b> <b>view</b> and similar commands
                 to particular regions of interest.

                 If an output filename is given, the index file will be written to <u>out.index</u>.  Otherwise, for  a
                 CRAM  file  <u>aln.cram</u>,  index file <u>aln.cram</u><b>.crai</b> will be created; for a BAM or SAM file <u>aln.bam</u>,
                 either <u>aln.bam</u><b>.bai</b> or <u>aln.bam</u><b>.csi</b> will be created, depending on the index format selected.

       <b>sort</b>      samtools sort [<b>-l</b> <u>level</u>] [<b>-m</b> <u>maxMem</u>] [<b>-o</b> <u>out.bam</u>] [<b>-O</b> <u>format</u>] [<b>-n</b>] [<b>-t</b> <u>tag</u>] [<b>-T</b> <u>tmpprefix</u>] [<b>-@</b>
                 <u>threads</u>] [<u>in.sam</u>|<u>in.bam</u>|<u>in.cram</u>]

                 Sort alignments by leftmost coordinates, or by read name when <b>-n</b> is used.  An appropriate  <b>@HD-</b>
                 <b>SO</b> sort order header tag will be added or an existing one updated if necessary.

                 The  sorted output is written to standard output by default, or to the specified file (<u>out.bam</u>)
                 when <b>-o</b> is used.  This command will also create temporary files <u>tmpprefix</u><b>.</b><u>%d</u><b>.bam</b> as needed when
                 the entire alignment data cannot fit into memory (as controlled via the <b>-m</b> option).

                 Consider using <b>samtools</b> <b>collate</b>  instead  if  you  need  name  collated  data  without  a  full
                 lexicographical sort.

                 Note  that  if  the  sorted  output  file  is  to  be  indexed with <b>samtools</b> <b>index</b>, the default
                 coordinate sort must be used.  Thus the <b>-n</b> and <b>-t</b> options are incompatible with <b>samtools</b> <b>index</b>.

       <b>collate</b>   samtools collate [<u>options</u>] <u>in.sam</u>|<u>in.bam</u>|<u>in.cram</u> [<u>&lt;prefix&gt;</u>]

                 Shuffles and groups reads together by their names.  A faster alternative to a full  query  name
                 sort,  <b>collate</b>  ensures  that reads of the same name are grouped together in contiguous groups,
                 but doesn't make any guarantees about the order of read names between groups.

                 The output from this command should be suitable for any operation that requires all reads  from
                 the same template to be grouped together.

       <b>cram-size</b> samtools cram-size [<u>options</u>] <u>in.cram</u>

                 Produces  a  summary  of  CRAM block Content ID numbers and their associated Data Series stored
                 within them.  Optionally a more detailed breakdown of how  each  data  series  is  encoded  per
                 container may also be listed using the <b>-e</b> or <b>--encodings</b> option.

       <b>idxstats</b>  samtools idxstats <u>in.sam</u>|<u>in.bam</u>|<u>in.cram</u>

                 Retrieve  and  print  stats  in the index file corresponding to the input file.  Before calling
                 idxstats, the input BAM file should be indexed by samtools index.

                 If run on a SAM or CRAM file or an unindexed BAM file, this command will still produce the same
                 summary statistics, but does so by reading through the entire file.  This is  far  slower  than
                 using the BAM indices.

                 The  output  is  TAB-delimited  with  each line consisting of reference sequence name, sequence
                 length, # mapped reads and # unmapped reads. It is written to stdout.

       <b>flagstat</b>  samtools flagstat <u>in.sam</u>|<u>in.bam</u>|<u>in.cram</u>

                 Does a full pass through the input file to calculate and print statistics to stdout.

                 Provides counts for each of 13 categories based primarily on bit flags in the FLAG field.  Each
                 category  in the output is broken down into QC pass and QC fail, which is presented as "#PASS +
                 #FAIL" followed by a description of the category.

       <b>flags</b>     samtools flags <u>INT</u>|<u>STR</u>[,...]

                 Convert between textual and numeric flag representation.

                 <b>FLAGS:</b>
                   <b>0x1</b>   PAIRED          paired-end (or multiple-segment) sequencing technology
                   <b>0x2</b>   PROPER_PAIR     each segment properly aligned according to the aligner
                   <b>0x4</b>   UNMAP           segment unmapped
                   <b>0x8</b>   MUNMAP          next segment in the template unmapped
                  <b>0x10</b>   REVERSE         SEQ is reverse complemented
                  <b>0x20</b>   MREVERSE        SEQ of the next segment in the template is reverse complemented
                  <b>0x40</b>   READ1           the first segment in the template
                  <b>0x80</b>   READ2           the last segment in the template
                 <b>0x100</b>   SECONDARY       secondary alignment
                 <b>0x200</b>   QCFAIL          not passing quality controls
                 <b>0x400</b>   DUP             PCR or optical duplicate
                 <b>0x800</b>   SUPPLEMENTARY   supplementary alignment

       <b>stats</b>     samtools stats [<u>options</u>] <u>in.sam</u>|<u>in.bam</u>|<u>in.cram</u> [<u>region</u>...]

                 samtools stats collects statistics from BAM files and outputs in a text format.  The output can
                 be visualized graphically using plot-bamstats.

       <b>bedcov</b>    samtools bedcov [<u>options</u>] <u>region.bed</u> <u>in1.sam</u>|<u>in1.bam</u>|<u>in1.cram</u>[...]

                 Reports the total read base count (i.e. the sum of per  base  read  depths)  for  each  genomic
                 region  specified  in  the  supplied BED file. The regions are output as they appear in the BED
                 file and are 0-based.  Counts for  each  alignment  file  supplied  are  reported  in  separate
                 columns.

       <b>depth</b>     samtools depth [<u>options</u>] [<u>in1.sam</u>|<u>in1.bam</u>|<u>in1.cram</u> [<u>in2.sam</u>|<u>in2.bam</u>|<u>in2.cram</u>] [...]]

                 Computes the read depth at each position or region.

       <b>ampliconstats</b>
                 samtools ampliconstats [<u>options</u>] <u>primers.bed</u> <u>in.sam</u>|<u>in.bam</u>|<u>in.cram</u>[...]

                 samtools  ampliconstats collects statistics from one or more input alignment files and produces
                 tables in text format.  The output can be visualized graphically using plot-ampliconstats.

                 The alignment files should have previously been clipped of  primer  sequence,  for  example  by
                 <b>samtools</b>  <b>ampliconclip</b>  and the sites of these primers should be specified as a bed file in the
                 arguments.

       <b>mpileup</b>   samtools mpileup [<b>-EB</b>] [<b>-C</b> <u>capQcoef</u>] [<b>-r</b> <u>reg</u>] [<b>-f</b> <u>in.fa</u>] [<b>-l</b> <u>list</u>] [<b>-Q</b> <u>minBaseQ</u>]  [<b>-q</b>  <u>minMapQ</u>]
                 <u>in.bam</u> [<u>in2.bam</u> [<u>...</u>]]

                 Generate  textual pileup for one or multiple BAM files.  For VCF and BCF output, please use the
                 <b>bcftools</b> <b>mpileup</b> command instead.  Alignment records are grouped by sample (SM) identifiers  in
                 @RG header lines.  If sample identifiers are absent, each input file is regarded as one sample.

                 See the samtools-mpileup man page for a description of the pileup format and options.

       <b>consensus</b> samtools consensus [<b>options</b>] <u>in.bam</u>

                 Generate consensus from a SAM, BAM or CRAM file based on the contents of the alignment records.
                 The consensus is written either as FASTA, FASTQ, or a pileup oriented format.

                 The  default  output for FASTA and FASTQ formats include one base per non-gap consensus.  Hence
                 insertions with respect to the aligned reference will be included and deletions removed.   This
                 behaviour can be adjusted.

                 Two consensus calling algorithms are offered.  The default computes a heterozygous consensus in
                 a  Bayesian  manner,  derived  from  the  "Gap5" consensus algorithm.  A simpler base frequency
                 counting method is also available.

       <b>reference</b> samtools reference [<b>options</b>] <u>in.bam</u>

                 Generate a reference from a SAM, BAM or CRAM file based on the contents of the  SEQuence  field
                 and  the  MD:Z:  auxiliary  tags,  or  from  the  embedded  reference blocks within a CRAM file
                 (provided it was constructed using the <b>embed_ref=1</b> option).

       <b>coverage</b>  samtools coverage [<u>options</u>] [<u>in1.sam</u>|<u>in1.bam</u>|<u>in1.cram</u> [<u>in2.sam</u>|<u>in2.bam</u>|<u>in2.cram</u>] [...]]

                 Produces a histogram or table of coverage per chromosome.

       <b>merge</b>     samtools merge [<b>-nur1f</b>] [<b>-h</b> <u>inh.sam</u>] [<b>-t</b> <u>tag</u>] [<b>-R</b>  <u>reg</u>]  [<b>-b</b>  <u>list</u>]  <u>out.bam</u>  <u>in1.bam</u>  [<u>in2.bam</u>
                 <u>in3.bam</u> ... <u>inN.bam</u>]

                 Merge  multiple sorted alignment files, producing a single sorted output file that contains all
                 the input records and maintains the existing sort order.

                 If <b>-h</b> is specified the @SQ headers of input files will be merged  into  the  specified  header,
                 otherwise  they  will be merged into a composite header created from the input headers.  If the
                 @SQ headers differ in order this may require the output file to be re-sorted after merge.

                 The ordering of the records in the input files must match the usage of the <b>-n</b> and  <b>-t</b>  command-
                 line  options.   If  they do not, the output order will be undefined.  See <b>sort</b> for information
                 about record ordering.

       <b>split</b>     samtools split [<u>options</u>] <u>merged.sam</u>|<u>merged.bam</u>|<u>merged.cram</u>

                 Splits a file by read group, producing one or more output files matching a  common  prefix  (by
                 default based on the input filename) each containing one read-group.

       <b>cat</b>       samtools cat [<b>-b</b> <u>list</u>] [<b>-h</b> <u>header.sam</u>] [<b>-o</b> <u>out.bam</u>] <u>in1.bam</u> <u>in2.bam</u> [ ... ]

                 Concatenate  BAMs  or CRAMs. Although this works on either BAM or CRAM, all input files must be
                 the same format as each other. The sequence dictionary of each input file  must  be  identical,
                 although  this command does not check this. This command uses a similar trick to <b>reheader</b> which
                 enables fast BAM concatenation.

       <b>import</b>    samtools import [<u>options</u>] <u>in.fastq</u> [ ... ]

                 Converts one or more FASTQ files to unaligned SAM, BAM or CRAM.  These formats offer  a  richer
                 capability  of  tracking  sample  meta-data  via  the SAM header and per-read meta-data via the
                 auxiliary tags.  The <b>fastq</b> command may be used to reverse this conversion.

       <b>fastq/a</b>   samtools fastq [<u>options</u>] <u>in.bam</u>
                 samtools fasta [<u>options</u>] <u>in.bam</u>

                 Converts a BAM or CRAM into either FASTQ or FASTA format depending on the command invoked.  The
                 files will be automatically compressed if the file names have a .gz, .bgz, or .bgzf extension.

                 The  input  to this program must be collated by name.  Use <b>samtools</b> <b>collate</b> or <b>samtools</b> <b>sort</b> <b>-n</b>
                 to ensure this.

       <b>faidx</b>     samtools faidx &lt;ref.fasta&gt; [region1 [...]]

                 Index reference sequence in the FASTA format or  extract  subsequence  from  indexed  reference
                 sequence.  If  no  region is specified, <b>faidx</b> will index the file and create <u>&lt;ref.fasta&gt;.fai</u> on
                 the disk. If regions are specified, the subsequences will be retrieved and printed to stdout in
                 the FASTA format.

                 The input file can be compressed in the <b>BGZF</b> format.

                 FASTQ files can be read and indexed by this  command.   Without  using  <b>--fastq</b>  any  extracted
                 subsequence will be in FASTA format.

       <b>fqidx</b>     samtools fqidx &lt;ref.fastq&gt; [region1 [...]]

                 Index  reference  sequence  in  the  FASTQ format or extract subsequence from indexed reference
                 sequence. If no region is specified, <b>fqidx</b> will index the file and  create  <u>&lt;ref.fastq&gt;.fai</u>  on
                 the disk. If regions are specified, the subsequences will be retrieved and printed to stdout in
                 the FASTQ format.

                 The input file can be compressed in the <b>BGZF</b> format.

                 <b>samtools</b>  <b>fqidx</b>  should  only be used on fastq files with a small number of entries.  Trying to
                 use it on a file containing millions of short sequencing reads will produce an  index  that  is
                 almost  as  big  as the original file, and searches using the index will be very slow and use a
                 lot of memory.

       <b>dict</b>      samtools dict <u>ref.fasta</u>|<u>ref.fasta.gz</u>

                 Create a sequence dictionary file from a fasta file.

       <b>calmd</b>     samtools calmd [<b>-Eeubr</b>] [<b>-C</b> <u>capQcoef</u>] <u>aln.bam</u> <u>ref.fasta</u>

                 Generate the MD tag. If the MD tag is already present, this command will give a warning if  the
                 MD tag generated is different from the existing tag. Output SAM by default.

                 Calmd  can  also  read  and  write  CRAM  files  although in most cases it is pointless as CRAM
                 recalculates MD and NM tags on the fly.  The one exception to this case is where both input and
                 output CRAM files have been / are being created with the <u>no_ref</u> option.

       <b>fixmate</b>   samtools fixmate [<b>-rpcm</b>] [<b>-O</b> <u>format</u>] <u>in.nameSrt.bam</u> <u>out.bam</u>

                 Fill in mate coordinates, ISIZE and mate related flags from a name-sorted alignment.

       <b>markdup</b>   samtools markdup [<b>-l</b> <u>length</u>] [<b>-r</b>] [<b>-s</b>] [<b>-T</b>] [<b>-S</b>] <u>in.algsort.bam</u> <u>out.bam</u>

                 Mark duplicate alignments from a coordinate sorted file that  has  been  run  through  <b>samtools</b>
                 <b>fixmate</b> with the <b>-m</b> option.  This program relies on the MC and ms tags that fixmate provides.

       <b>rmdup</b>     samtools rmdup [-sS] &lt;input.srt.bam&gt; &lt;out.bam&gt;

                 <b>This</b> <b>command</b> <b>is</b> <b>obsolete.</b> <b>Use</b> <b>markdup</b> <b>instead.</b>

       <b>addreplacerg</b>
                 samtools addreplacerg [<b>-r</b> <u>rg-line</u> | <b>-R</b> <u>rg-ID</u>] [<b>-m</b> <u>mode</u>] [<b>-l</b> <u>level</u>] [<b>-o</b> <u>out.bam</u>] <u>in.bam</u>

                 Adds or replaces read group tags in a file.

       <b>reheader</b>  samtools reheader [<b>-iP</b>] <u>in.header.sam</u> <u>in.bam</u>

                 Replace  the  header  in  <u>in.bam</u> with the header in <u>in.header.sam</u>.  This command is much faster
                 than replacing the header with a BAM→SAM→BAM conversion.

                 By default this command outputs the BAM or CRAM file to standard output (stdout), but for  CRAM
                 format  files  it  has  the option to perform an in-place edit, both reading and writing to the
                 same file.  No validity checking is performed on the header, nor that it  is  suitable  to  use
                 with the sequence data itself.

       <b>targetcut</b> samtools targetcut [<b>-Q</b> <u>minBaseQ</u>] [<b>-i</b> <u>inPenalty</u>] [<b>-0</b> <u>em0</u>] [<b>-1</b> <u>em1</u>] [<b>-2</b> <u>em2</u>] [<b>-f</b> <u>ref</u>] <u>in.bam</u>

                 This  command  identifies  target  regions  by examining the continuity of read depth, computes
                 haploid consensus sequences of targets and outputs a SAM with each sequence corresponding to  a
                 target.  When  option  <b>-f</b>  is  in  use,  BAQ will be applied. This command is <b>only</b> designed for
                 cutting fosmid clones from fosmid pool sequencing [Ref. Kitzman et al. (2010)].

       <b>phase</b>     samtools phase [<b>-AF</b>] [<b>-k</b> <u>len</u>] [<b>-b</b> <u>prefix</u>] [<b>-q</b> <u>minLOD</u>] [<b>-Q</b> <u>minBaseQ</u>] <u>in.bam</u>

                 Call and phase heterozygous SNPs.

       <b>depad</b>     samtools depad [<b>-SsCu1</b>] [<b>-T</b> <u>ref.fa</u>] [<b>-o</b> <u>output</u>] <u>in.bam</u>

                 Converts a BAM aligned against a padded  reference  to  a  BAM  aligned  against  the  depadded
                 reference.   The  padded reference may contain verbatim "*" bases in it, but "*" bases are also
                 counted in the reference numbering.  This means that a sequence  base-call  aligned  against  a
                 reference "*" is considered to be a cigar match ("M" or "X") operator (if the base-call is "A",
                 "C",  "G"  or  "T").   After  depadding  the  reference  "*" bases are deleted and such aligned
                 sequence base-calls become insertions.   Similarly  transformations  apply  for  deletions  and
                 padding cigar operations.

       <b>ampliconclip</b>
                 samtools  ampliconclip  [<b>-o</b>  <u>out.file</u>] [<b>-f</b> <u>stat.file</u>] [<b>--soft-clip</b>] [<b>--hard-clip</b>] [<b>--both-ends</b>]
                 [<b>--strand</b>] [<b>--clipped</b>] [<b>--fail</b>] [<b>--no-PG</b>] <b>-b</b> <u>bed.file</u> <u>in.file</u>

                 Clip reads in a SAM compatible file based on data from a BED file.

       <b>samples</b>   samtools samples [<b>-o</b> <u>out.file</u>] [<b>-i</b>] [<b>-T</b> <u>TAG</u>] [<b>-f</b> <u>refs.fasta</u>] [<b>-F</b> <u>refs_list</u>] [<b>-X</b>]

                 Prints the samples from alignment files

       <b>reset</b>     samtools reset [<b>-o</b> <u>FILE</u>] [<b>-x</b>/<b>--remove-tag</b> <u>tag_list</u>] [<b>--keep-tag</b>  <u>tag_list</u>]  [<b>--reject-PG</b>  <u>pgid</u>]
                 [<b>--no-RG</b>] [<b>--no-PG</b>] [...]

                 Removes  alignment  information  from  records,  producing  an unaligned SAM, BAM or CRAM file.
                 Flags are reset, header tags are updated or removed as  appropriate,  and  auxiliary  tags  are
                 removed or retained as specified.  Note that the sort order is unchanged.

</pre><h4><b>SAMTOOLS</b> <b>OPTIONS</b></h4><pre>
       These are options that are passed after the <b>samtools</b> command, before any sub-command is specified.

       <b>help</b>, <b>--help</b>
              Display  a  brief usage message listing the samtools commands available.  If the name of a command
              is also given, e.g., <b>samtools</b> <b>help</b> <b>view</b>, the detailed usage message for that particular command is
              displayed.

       <b>--version</b>
              Display the version numbers and copyright information for samtools  and  the  important  libraries
              used by samtools.

       <b>--version-only</b>
              Display the full samtools version number in a machine-readable format.

</pre><h4><b>GLOBAL</b> <b>COMMAND</b> <b>OPTIONS</b></h4><pre>
       Several  long-options are shared between multiple samtools sub-commands: <b>--input-fmt</b>, <b>--input-fmt-option</b>,
       <b>--output-fmt</b>, <b>--output-fmt-option</b>, <b>--reference</b>, <b>--write-index</b>, and  <b>--verbosity</b>.   The  input  format  is
       auto-detected  and specifying the format is unnecessary, so this option is rarely offered.  Note that not
       all subcommands have all options.  Consult the subcommand help for more details.

       Format strings recognised are "sam", "sam.gz", "bam" and  "cram".   They  may  be  followed  by  a  comma
       separated list of options as <u>key</u> or <u>key</u>=<u>value</u>. See below for examples.

       The <b>fmt-option</b> arguments accept either a single <u>option</u> or <u>option</u>=<u>value</u>.  Note that some options only work
       on  some  file  formats and only on read or write streams.  If value is unspecified for a boolean option,
       the value is assumed to be 1.  The valid options are as follows.

       <b>level=</b><u>INT</u>
           Output only. Specifies the compression level from 1 to 9, or  0  for  uncompressed.   If  the  output
           format is SAM, this also enables BGZF compression, otherwise SAM defaults to uncompressed.

       <b>nthreads=</b><u>INT</u>
           Specifies  the  number  of  threads  to  use  during  encoding and/or decoding.  For BAM this will be
           encoding only.  In CRAM the threads are dynamically shared between encoder and decoder.

       <b>filter=</b><u>STRING</u>
           Apply filter STRING to all incoming records, rejecting any that do not satisfy the  expression.   See
           the FILTER EXPRESSIONS section below for specifics.

       <b>reference=</b><u>fasta_file</u>
           Specifies  a  FASTA  reference file for use in CRAM encoding or decoding.  It usually is not required
           for decoding except in the situation of the MD5 not being obtainable via the  REF_PATH  or  REF_CACHE
           environment variables.

       <b>decode_md=</b><u>0|1</u>
           CRAM  input  only;  defaults  to 1 (on).  CRAM does not typically store MD and NM tags, preferring to
           generate them on the fly.  When this option is 0 missing MD, NM tags will not be generated.   It  can
           be particularly useful when combined with a file encoded using store_md=1 and store_nm=1.

       <b>store_md=</b><u>0|1</u>
           CRAM  output  only;  defaults  to  0  (off).  CRAM normally only stores MD tags when the reference is
           unknown and lets the decoder generate these values on-the-fly (see decode_md).

       <b>store_nm=</b><u>0|1</u>
           CRAM output only; defaults to 0 (off).  CRAM normally only stores  NM  tags  when  the  reference  is
           unknown and lets the decoder generate these values on-the-fly (see decode_md).

       <b>ignore_md5=</b><u>0|1</u>
           CRAM  input  only;  defaults to 0 (off).  When enabled, md5 checksum errors on the reference sequence
           and block checksum errors within CRAM are ignored.  Use of this option is strongly discouraged.

       <b>required_fields=</b><u>bit-field</u>
           CRAM input only; specifies which SAM columns need to be populated.  By default all fields  are  used.
           Limiting  the  decode to specific columns can have significant performance gains.  The bit-field is a
           numerical value constructed from the following table.
              <b>0x1</b>   SAM_QNAME
              <b>0x2</b>   SAM_FLAG
              <b>0x4</b>   SAM_RNAME
              <b>0x8</b>   SAM_POS
             <b>0x10</b>   SAM_MAPQ
             <b>0x20</b>   SAM_CIGAR
             <b>0x40</b>   SAM_RNEXT
             <b>0x80</b>   SAM_PNEXT
            <b>0x100</b>   SAM_TLEN
            <b>0x200</b>   SAM_SEQ
            <b>0x400</b>   SAM_QUAL
            <b>0x800</b>   SAM_AUX
           <b>0x1000</b>   SAM_RGAUX

       <b>name_prefix=</b><u>string</u>
           CRAM input only; defaults to output filename.  Any sequences with auto-generated read names will  use
           <u>string</u> as the name prefix.

       <b>multi_seq_per_slice=</b><u>0|1</u>
           CRAM  output  only;  defaults  to  0  (off).   By  default CRAM generates one container per reference
           sequence, except in the case of many small references (such as a fragmented assembly).

       <b>version=</b><u>major.minor</u>
           CRAM output only.  Specifies the CRAM version number.  Acceptable values are "2.1", "3.0", and "3.1".

       <b>seqs_per_slice=</b><u>INT</u>
           CRAM output only; defaults to 10000.

       <b>slices_per_container=</b><u>INT</u>
           CRAM output only; defaults to 1.  The effect of having multiple slices per container is to share  the
           compression  header  block  between multiple slices.  This is unlikely to have any significant impact
           unless the number of sequences per slice  is  reduced.   (Together  these  two  options  control  the
           granularity of random access.)

       <b>embed_ref=</b><u>0|1</u>
           CRAM  output  only; defaults to 0 (off).  If 1, this will store portions of the reference sequence in
           each slice, permitting decode without having requiring an external copy of the reference sequence.

       <b>no_ref=</b><u>0|1</u>
           CRAM output only; defaults to 0 (off).  If 1, sequences will be stored  verbatim  with  no  reference
           encoding.  This can be useful if no reference is available for the file.

       <b>use_bzip2=</b><u>0|1</u>
           CRAM output only; defaults to 0 (off).  Permits use of bzip2 in CRAM block compression.

       <b>use_lzma=</b><u>0|1</u>
           CRAM output only; defaults to 0 (off).  Permits use of lzma in CRAM block compression.

       <b>use_arith=</b><u>0|1</u>
           CRAM  ≥ 3.1 output only; enables use of arithmetic entropy coding in CRAM block compression.  This is
           off by default, but enabled for archive mode.  This is significantly  slower  but  sometimes  smaller
           than the standard rANS entropy encoder.

       <b>use_fqz=</b><u>0|1</u>
           CRAM  ≥  3.1 output only; enables and disables the fqzcomp quality compression method.  This is on by
           default for version 3.1 and above only when the small and archive profiles are in use.

       <b>use_tok=</b><u>0|1</u>
           CRAM ≥ 3.1 output only; enables and disables the name tokeniser compression method.  This  is  on  by
           default for version 3.1 and above.

       <b>lossy_names=</b><u>0|1</u>
           CRAM  output  only; defaults to 0 (off).  If 1, templates with all members within the same CRAM slice
           will have their read names removed.  New names will be automatically generated during decoding.  Also
           see the <b>name_prefix</b> option.

       <b>fast,</b> <b>normal,</b> <b>small,</b> <b>archive</b>
           CRAM output only.  Set the CRAM compression profile.  This is a simplified way of setting many output
           options at once.  It changes the following options according to the profile  in  use.   The  "normal"
           profile is the default.

           <b>Option</b>           <b>fast</b>    <b>normal</b>   <b>small</b>   <b>archive</b>
           <b>level</b>            1       5        6       7
           <b>use_bzip2</b>        off     off      on      on
           <b>use_lzma</b>         off     off      off     on if level&gt;7
           <b>use_tok(*)</b>       off     on       on      on
           <b>use_fqz(*)</b>       off     off      on      on
           <b>use_arith(*)</b>     off     off      off     on
           <b>seqs_per_slice</b>   10000   10000    25000   100000

           (*) <b>use_tok</b>, <b>use_fqz</b> and <b>use_arith</b> are only enabled for CRAM version 3.1 and above.

           The  <b>level</b>  listed  is  only the default value, and will not be set if it has been explicitly changed
           already.  Additionally <b>bases_per_slice</b> is set to <b>500*seqs_per_slice</b> unless previously explicitly set.

       For example:

           samtools view --input-fmt-option decode_md=0
               --output-fmt cram,version=3.0 --output-fmt-option embed_ref
               --output-fmt-option seqs_per_slice=2000 -o foo.cram foo.bam

           samtools view -O cram,small -o bar.cram bar.bam

       The <b>--write-index</b> option enables automatic index creation while writing out BAM, CRAM or bgzf SAM  files.
       Note  to  get  compressed  SAM  as  the  output  format you need to manually request a compression level,
       otherwise all SAM files are uncompressed.  By default SAM and BAM will use CSI indices  while  CRAM  will
       use  CRAI indices.  If you need to create BAI indices note that it is possible to specify the name of the
       index being written to, and hence the format, by using the <b>filename##idx##indexname</b> notation.

       For example: to convert a BAM to a compressed SAM with CSI indexing:

           samtools view -h -O sam,level=6 --write-index in.bam -o out.sam.gz

       To convert a SAM to a compressed BAM using BAI indexing:

           samtools view --write-index in.sam -o out.bam##idx##out.bam.bai

       The <b>--verbosity</b> <u>INT</u> option sets  the  verbosity  level  for  samtools  and  HTSlib.   The  default  is  3
       (HTS_LOG_WARNING);  2  reduces warning messages and 0 or 1 also reduces some error messages, while values
       greater than 3 produce increasing numbers of additional warnings and logging messages.

</pre><h4><b>REFERENCE</b> <b>SEQUENCES</b></h4><pre>
       The CRAM format requires use of a reference sequence for both reading and writing.

       When reading a CRAM the <b>@SQ</b> headers are interrogated to identify the reference sequence MD5sum (<b>M5:</b>  tag)
       and  the local reference sequence filename (<b>UR:</b> tag).  Note that <u>http://</u> and <u>ftp://</u> based URLs in the UR:
       field are not used, but local fasta filenames (with or without <u>file://</u>) can be used.

       To create a CRAM the <b>@SQ</b> headers will also be read to identify the reference sequences, but M5:  and  UR:
       tags  may  not be present. In this case the <b>-T</b> and <b>-t</b> options of samtools view may be used to specify the
       fasta or fasta.fai filenames respectively (provided the .fasta.fai file is also backed  up  by  a  .fasta
       file).

       The search order to obtain a reference is:

       1. Use any local file specified by the command line options (eg -T).

       2. Look for MD5 via REF_CACHE environment variable.

       3. Look for MD5 in each element of the REF_PATH environment variable.

       4. Look for a local file listed in the UR: header tag.

</pre><h4><b>FILTER</b> <b>EXPRESSIONS</b></h4><pre>
       Filter  expressions  are  used as an on-the-fly checking of incoming SAM, BAM or CRAM records, discarding
       records that do not match the specified expression.

       The language used is primarily C style, but with a few  differences  in  the  precedence  rules  for  bit
       operators and the inclusion of regular expression matching.

       The operator precedence, from strongest binding to weakest, is:

       Grouping        <b>(,</b> <b>)</b>             E.g. "(1+2)*3"
       Values:         <b>literals,</b> <b>vars</b>   Numbers, strings and variables
       Unary ops:      <b>+,</b> <b>-,</b> <b>!,</b> <b>~</b>       E.g. -10 +10, !10 (not), ~5 (bit not)
       Math ops:       <b>*,</b> <b>/,</b> <b>%</b>          Multiply, division and (integer) modulo
       Math ops:       <b>+,</b> <b>-</b>             Addition / subtraction
       Bit-wise:       <b>&amp;</b>                Integer AND
       Bit-wise        <b>^</b>                Integer XOR
       Bit-wise        <b>|</b>                Integer OR
       Conditionals:   <b>&gt;,</b> <b>&gt;=,</b> <b>&lt;,</b> <b>&lt;=</b>
       Equality:       <b>==,</b> <b>!=,</b> <b>=~,</b> <b>!~</b>   =~ and !~ match regular expressions
       Boolean:        <b>&amp;&amp;,</b> <b>||</b>           Logical AND / OR

       Expressions  are  computed  using floating point mathematics, so "10 / 4" evaluates to 2.5 rather than 2.
       They may be written as integers in decimal or "0x" plus hexadecimal, and floating point with  or  without
       exponents.However  operations that require integers first do an implicit type conversion, so "7.9 % 5" is
       2 and "7.9 &amp; 4.1" is equivalent to "7 &amp; 4", which is  4.   Strings  are  always  specified  using  double
       quotes.  To get a double quote in a string, use backslash.  Similarly a double backslash is used to get a
       literal backslash.  For example <b>ab\"c\\d</b> is the string <b>ab"c\d</b>.

       Comparison  operators  are  evaluated as a match being 1 and a mismatch being 0, thus "(2 &gt; 1) + (3 &lt; 5)"
       evaluates as 2.  All comparisons involving undefined (null) values are deemed to be false.

       The variables are where the file format specifics  are  accessed  from  the  expression.   The  variables
       correspond  to  SAM  fields,  for  example to find paired alignments with high mapping quality and a very
       large insert size, we may use the expression "<b>mapq</b> <b>&gt;=</b> <b>30</b> <b>&amp;&amp;</b> <b>(tlen</b> <b>&gt;=</b> <b>100000</b> <b>||</b> <b>tlen</b> <b>&lt;=</b> <b>-100000)</b>".   Valid
       variable names and their data types are:

       <b>endpos</b>               int            Alignment end position (1-based)
       <b>flag</b>                 int            Combined FLAG field
       <b>flag.paired</b>          int            Single bit, 0 or 1
       <b>flag.proper_pair</b>     int            Single bit, 0 or 2
       <b>flag.unmap</b>           int            Single bit, 0 or 4
       <b>flag.munmap</b>          int            Single bit, 0 or 8
       <b>flag.reverse</b>         int            Single bit, 0 or 16
       <b>flag.mreverse</b>        int            Single bit, 0 or 32
       <b>flag.read1</b>           int            Single bit, 0 or 64
       <b>flag.read2</b>           int            Single bit, 0 or 128
       <b>flag.secondary</b>       int            Single bit, 0 or 256
       <b>flag.qcfail</b>          int            Single bit, 0 or 512
       <b>flag.dup</b>             int            Single bit, 0 or 1024
       <b>flag.supplementary</b>   int            Single bit, 0 or 2048
       <b>hclen</b>                int            Number of hard-clipped bases
       <b>library</b>              string         Library (LB header via RG)
       <b>mapq</b>                 int            Mapping quality
       <b>mpos</b>                 int            Synonym for pnext
       <b>mrefid</b>               int            Mate reference number (0 based)
       <b>mrname</b>               string         Synonym for rnext
       <b>ncigar</b>               int            Number of cigar operations
       <b>pnext</b>                int            Mate's alignment position (1-based)
       <b>pos</b>                  int            Alignment position (1-based)
       <b>qlen</b>                 int            Alignment length: no. query bases
       <b>qname</b>                string         Query name
       <b>qual</b>                 string         Quality values (raw, 0 based)
       <b>refid</b>                int            Integer reference number (0 based)
       <b>rlen</b>                 int            Alignment length: no. reference bases
       <b>rname</b>                string         Reference name
       <b>rnext</b>                string         Mate's reference name
       <b>sclen</b>                int            Number of soft-clipped bases
       <b>seq</b>                  string         Sequence
       <b>tlen</b>                 int            Template length (insert size)
       <b>[XX]</b>                 int / string   XX tag value

       Flags  are  returned  either  as  the whole flag value or by checking for a single bit.  Hence the filter
       expression <b>flag.dup</b> is equivalent to <b>flag</b> <b>&amp;</b> <b>1024</b>.

       "qlen" and "rlen" are measured using the CIGAR string  to  count  the  number  of  query  (sequence)  and
       reference  bases  consumed.   Note  "qlen"  may  not  exactly  match the length of the "seq" field if the
       sequence is "*".

       "sclen" and "hclen" are the number of soft and hard-clipped bases respectively.  The formula "qlen-sclen"
       gives the number of sequence bases used in the alignment, distinguishing  between  global  alignment  and
       local alignment length.

       "endpos"  is the (1-based inclusive) position of the rightmost mapped base of the read, as measured using
       the CIGAR string, and for mapped reads is equivalent to "pos+rlen-1". For unmapped reads, it is the  same
       as "pos".

       Reference names may be matched either by their string forms ("rname" and "mrname") or as the Nth <b>@SQ</b> line
       (counting from zero) as stored in BAM using "tid" and "mtid" respectively.

       Auxiliary  tags  are described in square brackets and these expand to either integer or string as defined
       by the tag itself (<b>XX:Z:</b><u>string</u> or <b>XX:i:</b><u>int</u>).  For example <b>[NM]&gt;=10</b> can be used  to  look  for  alignments
       with many mismatches and <b>[RG]=~"grp[ABC]-"</b> will match the read-group string.

       If  no comparison is used with an auxiliary tag it is taken simply to be a test for the existence of that
       tag.  So <b>[NM]</b> will return any record containing an NM tag, even if that tag is zero (<b>NM:i:0</b>).  In  htslib
       &lt;=  1.15  negating this with <b>![NM]</b> gave misleading results as it was true if the tag did not exist or did
       exist but was zero.  Now this is strictly does-not-exist.  An  explicit  <b>exists([NM])</b>  and  <b>!exists([NM])</b>
       function has also been added to make this intention clear.

       Similarly  in htslib &lt;= 1.15 using <b>[NM]!=0</b> was true both when the tag existed and was not zero as well as
       when the tag did not exist.  From 1.16 onwards all comparison operators  are  only  true  for  tags  that
       exist, so <b>[NM]!=0</b> works as expected.

       Some  simple  functions are available to operate on strings.  These treat the strings as arrays of bytes,
       permitting their length, minimum, maximum and average values  to  be  computed.   These  are  useful  for
       processing Quality Scores.

       <b>length(x)</b>   Length of the string (excluding nul char)
       <b>min(x)</b>      Minimum byte value in the string
       <b>max(x)</b>      Maximum byte value in the string
       <b>avg(x)</b>      Average byte value in the string

       Note  that  "avg"  is  a  floating  point  value  and  it  may be NAN for empty strings.  This means that
       "avg(qual)" does not produce an error for records that have both seq and qual of "*".   NAN  values  will
       fail  any conditional checks, so e.g. "avg(qual) &gt; 20" works and will not report these records.  NAN also
       fails all equality, &lt; and &gt; comparisons, and returns zero  when  given  as  an  argument  to  the  <b>exists</b>
       function.  It can be negated with <b>!x</b> in which case it becomes true.

       Functions that operate on both strings and numerics:

       <b>exists(x)</b>      True if the value exists (or is explicitly true).
       <b>default(x,d)</b>   Value <b>x</b> if it exists or <b>d</b> if not.

       Functions that apply only to numeric values:

       <b>sqrt(x)</b>     Square root of <b>x</b>
       <b>log(x)</b>      Natural logarithm of <b>x</b>
       <b>pow(x,</b> <b>y)</b>   Power function, <b>x</b> to the power of <b>y</b>
       <b>exp(x)</b>      Base-e exponential, equivalent to <b>pow(e,x)</b>

</pre><h4><b>ENVIRONMENT</b> <b>VARIABLES</b></h4><pre>
       <b>HTS_PATH</b>
              A  colon-separated list of directories in which to search for HTSlib plugins.  If $HTS_PATH starts
              or ends with a colon or contains a double colon (<b>::</b>), the built-in list of directories is searched
              at that point in the search.

              If no HTS_PATH variable is defined, the built-in list of directories  specified  when  HTSlib  was
              built is used, which typically includes <b>/usr/local/libexec/htslib</b> and similar directories.

       <b>REF_PATH</b>
              A  colon  separated  (semi-colon  on  Windows)  list  of  locations in which to look for sequences
              identified by their MD5sums.  This can be either a list of directories or URLs. Note that if a URL
              is included then the colon in http:// and ftp:// and the optional port number will be  treated  as
              part  of  the  URL  and not a PATH field separator.  For URLs, the text <b>%s</b> will be replaced by the
              MD5sum being read.

              If no REF_PATH has been specified it will default to <b><a href="http://www.ebi.ac.uk/ena/cram/md5/">http://www.ebi.ac.uk/ena/cram/md5/</a>%s</b>  and  if
              REF_CACHE is also unset, it will be set to <b>$XDG_CACHE_HOME/hts-ref/%2s/%2s/%s</b>.  If <b>$XDG_CACHE_HOME</b>
              is  unset, <b>$HOME/.cache</b> (or a local system temporary directory if no home directory is found) will
              be used similarly.

       <b>REF_CACHE</b>
              This can be defined to a single location housing a local cache of references.  Upon downloading  a
              reference  it  will be stored in the location pointed to by REF_CACHE.  REF_CACHE will be searched
              before attempting to load via the REF_PATH search list.  If no REF_PATH is defined, both  REF_PATH
              and  REF_CACHE will be automatically set (see above), but if REF_PATH is defined and REF_CACHE not
              then no local cache is used.

              To avoid many files being stored in the same directory, REF_CACHE may  be  defined  as  a  pattern
              using  <b>%</b><u>num</u><b>s</b>  to  consume <u>num</u> characters of the MD5sum and <b>%s</b> to consume all remaining characters.
              If REF_CACHE lacks <b>%s</b> then it will get an implicit <b>/%s</b> appended.

              To aid population of the REF_CACHE directory a script <b>misc/seq_cache_populate.pl</b>  is  provided  in
              the Samtools distribution. This takes a fasta file or a directory of fasta files and generates the
              MD5sum named files.

              For  example  if  you  use <b>seq_cache_populate</b> <b>-subdirs</b> <b>2</b> <b>-root</b> <b>/local/ref_cache</b> to create 2 nested
              subdirectories (the default), each consuming 2 characters of the MD5sum, then  REF_CACHE  must  be
              set to <b>/local/ref_cache/%2s/%2s/%s</b>.

</pre><h4><b>EXAMPLES</b></h4><pre>
       o Import SAM to BAM when <b>@SQ</b> lines are present in the header:

           samtools view -b aln.sam &gt; aln.bam

         If <b>@SQ</b> lines are absent:

           samtools faidx ref.fa
           samtools view -bt ref.fa.fai aln.sam &gt; aln.bam

         where <u>ref.fa.fai</u> is generated automatically by the <b>faidx</b> command.

       o Convert a BAM file to a CRAM file using a local reference sequence.

           samtools view -C -T ref.fa aln.bam &gt; aln.cram

</pre><h4><b>AUTHOR</b></h4><pre>
       Heng Li from the Sanger Institute wrote the original C version of samtools.  Bob Handsaker from the Broad
       Institute  implemented  the  BGZF  library.   Petr  Danecek and Heng Li wrote the VCF/BCF implementation.
       James  Bonfield  from  the  Sanger  Institute  developed  the  CRAM  implementation.   Other  large  code
       contributions have been made by John Marshall, Rob Davies, Martin Pollard, Andrew Whitwham, Valeriu Ohan,
       Vasudeva  Sarma  (all  while primarily at the Sanger Institute), with numerous other smaller but valuable
       contributions.  See the per-command manual pages for further authorship.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/samtools-addreplacerg.1.html">samtools-addreplacerg</a></u>(1),   <u><a href="../man1/samtools-ampliconclip.1.html">samtools-ampliconclip</a></u>(1),   <u><a href="../man1/samtools-ampliconstats.1.html">samtools-ampliconstats</a></u>(1),    <u><a href="../man1/samtools-bedcov.1.html">samtools-bedcov</a></u>(1),
       <u><a href="../man1/samtools-calmd.1.html">samtools-calmd</a></u>(1),  <u><a href="../man1/samtools-cat.1.html">samtools-cat</a></u>(1),  <u><a href="../man1/samtools-collate.1.html">samtools-collate</a></u>(1),  <u><a href="../man1/samtools-consensus.1.html">samtools-consensus</a></u>(1),  <u><a href="../man1/samtools-coverage.1.html">samtools-coverage</a></u>(1),
       <u><a href="../man1/samtools-cram-size.1.html">samtools-cram-size</a></u>(1),   <u><a href="../man1/samtools-depad.1.html">samtools-depad</a></u>(1),   <u><a href="../man1/samtools-depth.1.html">samtools-depth</a></u>(1),   <u><a href="../man1/samtools-dict.1.html">samtools-dict</a></u>(1),   <u><a href="../man1/samtools-faidx.1.html">samtools-faidx</a></u>(1),
       <u><a href="../man1/samtools-fasta.1.html">samtools-fasta</a></u>(1),   <u><a href="../man1/samtools-fastq.1.html">samtools-fastq</a></u>(1),   <u><a href="../man1/samtools-fixmate.1.html">samtools-fixmate</a></u>(1),  <u><a href="../man1/samtools-flags.1.html">samtools-flags</a></u>(1),  <u><a href="../man1/samtools-flagstat.1.html">samtools-flagstat</a></u>(1),
       <u><a href="../man1/samtools-fqidx.1.html">samtools-fqidx</a></u>(1),   <u><a href="../man1/samtools-head.1.html">samtools-head</a></u>(1),   <u><a href="../man1/samtools-idxstats.1.html">samtools-idxstats</a></u>(1),   <u><a href="../man1/samtools-import.1.html">samtools-import</a></u>(1),   <u><a href="../man1/samtools-index.1.html">samtools-index</a></u>(1),
       <u><a href="../man1/samtools-markdup.1.html">samtools-markdup</a></u>(1),  <u><a href="../man1/samtools-merge.1.html">samtools-merge</a></u>(1),  <u><a href="../man1/samtools-mpileup.1.html">samtools-mpileup</a></u>(1), <u><a href="../man1/samtools-phase.1.html">samtools-phase</a></u>(1), <u><a href="../man1/samtools-quickcheck.1.html">samtools-quickcheck</a></u>(1),
       <u><a href="../man1/samtools-reference.1.html">samtools-reference</a></u>(1),  <u><a href="../man1/samtools-reheader.1.html">samtools-reheader</a></u>(1),  <u><a href="../man1/samtools-reset.1.html">samtools-reset</a></u>(1),  <u><a href="../man1/samtools-rmdup.1.html">samtools-rmdup</a></u>(1),   <u><a href="../man1/samtools-sort.1.html">samtools-sort</a></u>(1),
       <u><a href="../man1/samtools-split.1.html">samtools-split</a></u>(1),   <u><a href="../man1/samtools-stats.1.html">samtools-stats</a></u>(1),   <u><a href="../man1/samtools-targetcut.1.html">samtools-targetcut</a></u>(1),   <u><a href="../man1/samtools-tview.1.html">samtools-tview</a></u>(1),   <u><a href="../man1/samtools-view.1.html">samtools-view</a></u>(1),
       <u><a href="../man1/bcftools.1.html">bcftools</a></u>(1), <u><a href="../man5/sam.5.html">sam</a></u>(5), <u><a href="../man1/tabix.1.html">tabix</a></u>(1)

       Samtools website: &lt;<a href="http://www.htslib.org/">http://www.htslib.org/</a>&gt;
       File format specification of SAM/BAM,CRAM,VCF/BCF: &lt;<a href="http://samtools.github.io/hts-specs">http://samtools.github.io/hts-specs</a>&gt;
       Samtools latest source: &lt;https://github.com/samtools/samtools&gt;
       HTSlib latest source: &lt;https://github.com/samtools/htslib&gt;
       Bcftools website: &lt;<a href="http://samtools.github.io/bcftools">http://samtools.github.io/bcftools</a>&gt;

samtools-1.21                                   12 September 2024                                    <u><a href="../man1/samtools.1.html">samtools</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>