<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>samtools-merge - merges multiple sorted files into a single file</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/samtools">samtools_1.21-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       samtools-merge - merges multiple sorted files into a single file

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>samtools</b> <b>merge</b> [<u>options</u>] <b>-o</b> <u>out.bam</u> [<u>options</u>] <u>in1.bam</u> ... <u>inN.bam</u>

       <b>samtools</b> <b>merge</b> [<u>options</u>] <u>out.bam</u> <u>in1.bam</u> ... <u>inN.bam</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Merge  multiple sorted alignment files, producing a single sorted output file that contains all the input
       records and maintains the existing sort order.

       The output file can be specified via <b>-o</b> as shown in the first synopsis.  Otherwise the  first  non-option
       filename  argument is taken to be <u>out.bam</u> rather than an input file, as in the second synopsis.  There is
       no default; to write to standard output (or to a pipe), use either “<b>-o</b> <b>-</b>” or the equivalent using “<b>-</b>”  as
       the first filename argument.

       If  <b>-h</b>  is  specified  the @SQ headers of input files will be merged into the specified header, otherwise
       they will be merged into a composite header created from the input headers.  If in the process of merging
       @SQ lines for coordinate sorted input files, a conflict arises as to the order  (for  example  input1.bam
       has @SQ for a,b,c and input2.bam has b,a,c) then the resulting output file will need to be re-sorted back
       into coordinate order.

       Unless the <b>-c</b> or <b>-p</b> flags are specified then when merging @RG and @PG records into the output header then
       any  IDs  found to be duplicates of existing IDs in the output header will have a suffix appended to them
       to differentiate them from similar header records from other files and the read records will  be  updated
       to reflect this.

       The  ordering  of  the  records in the input files must match the usage of the <b>-n</b>, <b>-N</b> and <b>-t</b> command-line
       options.  If they do not, the output order will be undefined.  Note  this  also  extends  to  disallowing
       mixing  of "queryname" files with a combination of natural and lexicographical sort orders.  See <b>sort</b> for
       information about record ordering.

       Problems may arise when attempting to merge thousands of files together.  The operating system may impose
       a limit on the maximum number of  simultaneously  open  files.   See  <b>ulimit</b>  <b>-n</b>  for  more  information.
       Additionally  many  files  being read from simultaneously may cause a certain amount of "disk thrashing".
       To partially alleviate this the merge command will load 1MB of data at a time from each file, but this in
       turn adds to the overall merge program memory usage.  Please take this into account when  setting  memory
       limits.

       In  extreme  cases,  it  may  be  necessary  to reduce the problem to fewer files by successively merging
       subsets before a second round of merging.

       <b>-1</b>      Use Deflate compression level 1 to compress the output.

       <b>-b</b> <u>FILE</u> List of input BAM files, one file per line.

       <b>-f</b>      Force to overwrite the output file if present.

       <b>-h</b> <u>FILE</u> Use the lines of <u>FILE</u> as `@' headers to be copied to <u>out.bam</u>, replacing  any  header  lines  that
               would  otherwise  be  copied from <u>in1.bam</u>.  (<u>FILE</u> is actually in SAM format, though any alignment
               records it may contain are ignored.)

       <b>-n</b>      The input alignments are sorted by read names using an alpha-numeric  ordering,  rather  than  by
               chromosomal coordinates.  The alpha-numeric or “natural” sort order detects runs of digits in the
               strings  and  sorts  these  numerically.   Hence  "a7b"  appears before "a12b".  Note this is not
               suitable where hexadecimal values are in use.

       <b>-N</b>      The input alignments are sorted by read names using a lexicographical ordering,  rather  than  by
               chromosomal  coordinates.   Unlike <b>-n</b> no detection of numeric components is used, instead relying
               purely on the ASCII value of each character.  Hence "x12" comes before "x7" as "1" is before  "7"
               in ASCII.  This is a more appropriate name sort order where all digits in names are already zero-
               padded and/or hexadecimal values are being used.

       <b>-o</b> <u>FILE</u> Write  merged  output  to  <u>FILE</u>,  specifying  the filename via an option rather than as the first
               filename argument.  When <b>-o</b> is used, all non-option filename arguments specify input files to  be
               merged.

       <b>-t</b> <b>TAG</b>  The input alignments have been sorted by the value of TAG, then by either position or name (if <b>-n</b>
               is given).

       <b>-R</b> <u>STR</u>  Merge files in the specified region indicated by <u>STR</u> [null]

       <b>-r</b>      Attach an RG tag to each alignment. The tag value is inferred from file names.

       <b>-u</b>      Uncompressed BAM output

       <b>-c</b>      When several input files contain @RG headers with the same ID, emit only one of them (namely, the
               header  line  from the first file we find that ID in) to the merged output file.  Combining these
               similar headers is usually the right thing to do when the files being merged originated from  the
               same file.

               Without  <b>-c</b>,  all  @RG headers appear in the output file, with random suffixes added to their IDs
               where necessary to differentiate them.

       <b>-p</b>      Similarly, for each @PG ID in the set of files to merge, use the @PG line of the  first  file  we
               find that ID in rather than adding a suffix to differentiate similar IDs.

       <b>-X</b>      If  this  option  is set, it will allows user to specify customized index file location(s) if the
               data folder does not contain any index file. See <b>EXAMPLES</b> section for sample of usage.

       <b>-L</b> <u>FILE</u> BED file for specifying multiple regions on which the  merge  will  be  performed.   This  option
               extends the usage of <b>-R</b> option and cannot be used concurrently with it.

       <b>--no-PG</b> Do not add a @PG line to the header of the output file.

       <b>-@,</b> <b>--threads</b> <u>INT</u>
               Number of input/output compression threads to use in addition to main thread [0].

</pre><h4><b>EXAMPLES</b></h4><pre>
       o Attach the <b>RG</b> tag while merging sorted alignments:

           printf '@RG\tID:ga\tSM:hs\tLB:ga\tPL:ILLUMINA\n@RG\tID:454\tSM:hs\tLB:454\tPL:LS454\n' &gt; rg.txt
           samtools merge -rh rg.txt merged.bam ga.bam 454.bam

         The  value  in a <b>RG</b> tag is determined by the file name the read is coming from. In this example, in the
         <u>merged.bam</u>, reads from <u>ga.bam</u> will be attached <u>RG:Z:ga</u>, while  reads  from  <u>454.bam</u>  will  be  attached
         <u>RG:Z:454</u>.

       o Include customized index file as a part of arguments:

           samtools merge [options] -X &lt;out.bam&gt; &lt;/data_folder/in1.bam&gt; [&lt;/data_folder/in2.bam&gt; ... &lt;/data_folder/inN.bam&gt;] &lt;/index_folder/index1.bai&gt; [&lt;/index_folder/index2.bai&gt; ... &lt;/index_folder/indexN.bai&gt;]

</pre><h4><b>AUTHOR</b></h4><pre>
       Written by Heng Li from the Sanger Institute.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/samtools.1.html">samtools</a></u>(1), <u><a href="../man1/samtools-sort.1.html">samtools-sort</a></u>(1), <u><a href="../man5/sam.5.html">sam</a></u>(5)

       Samtools website: &lt;<a href="http://www.htslib.org/">http://www.htslib.org/</a>&gt;

samtools-1.21                                   12 September 2024                              <u><a href="../man1/samtools-merge.1.html">samtools-merge</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>