<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>pp_simScore - print similarity and alignments for block-profile and protein sequence on the standard</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/augustus">augustus_3.5.0+dfsg-4build7_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       pp_simScore - print similarity and alignments for block-profile and protein sequence on the standard
       output

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>pp_simScore</b> [OPTIONS] --fasta=<u>protein-sequence-file</u> --prfl=<u>protein-profile-file</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Algorithm for calculating the similarity score and the optimal alignments of a block-profile and a
       protein sequence. The algorithm can optional take intron positions into account. Print to standard
       output.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Mandatory</b> <b>options</b>
       <b>-f</b>, <b>--fasta</b>=<u>file</u>
           Protein sequence file in FASTA format.
           It may contain an optional [Intron] section. This section denotes the intron positions in the protein
           sequence, which are specified as list of (j, f), where j is the index of the amino acid after witch
           the intron immediately occurs. The indices range from 0 to m - 1 if the protein sequence has a length
           of m.

               &gt;protein sequence header
               XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
               XXXXXXX protein sequence XXXXXXXXXXX
               XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX

               [Introns]
               # index of the position after which an intron occurs | residual nucleotides before the intron
               2 0
               5 1
               30 2
               104 1

       <b>-p</b>, <b>--prfl</b>= <u>file</u>
           The block profile file has to have following structure:

               [dist]
               min    max
               [block]
               B
               [intron profile]
               w
               inter-block_profile_list
               intra-block_profile_list

           This structure can be repeated in this file. The file has to end either
           in a [dist] section or a [dist] and than [intron profile] section.
           The [intron profile] sections are optional.

               [dist] explanation:
               min, max denote the distance interval of an inter-block section

               [block] explanation:
               B denotes a (20 x t) matrix for a block of t of the block-profile

               [intron profile] explanation:
               an intron profile describes the positions and frequencies of introns in and
               before the associated block
               w: number of protein family members used to build the intron profile
               inter-block_profile_list: list of (h, v),
               where   h denotes the number of introns which occurred within a family member,
                       v the number of family members which have this number of introns
               intra-block_profile_list: list (s, f, v),
               where   s denotes the index of the position in the block after which an intron occurs,
                       f denotes the number of nucleotides which are left before the intron (0,1,2)
                       v the number of family members which have an intron at that position

   <b>Additional</b> <b>options</b>
       <b>-g</b>, <b>--gap_inter</b>=<u>float</u>
           Gap costs for an alignment column that is a gap in an inter-block section. Default Value: -5.0

       <b>-b</b>, <b>--gap_intra</b>=<u>float</u>
           Gap costs for an alignment column that is a gap in a block. Default Value: -50.0

       <b>-r</b>, <b>--gap_intron</b>=<u>float</u>
           Gap costs for an gap in intron positions. Default Value: -5.0

       <b>-e</b>, <b>--epsilon_intron</b>=<u>float</u>
           Pseudocount parameter epsilon1, the pseudocount is added to a relative intron frequency v/w with
           (v+epsilon1)/(w+epsilon1+epsilon2). Default Value: 0.0000001

       <b>-n</b>, <b>--epsilon_noIntron</b>=<u>float</u>
           Pseudocount parameter epsilon2, the pseudocount is added to a relative intron frequency v/w with
           (v+epsilon1)/(w+epsilon1+epsilon2). Default Value: 0.1

       <b>-i</b>, <b>--intron_weight_intra</b>=<u>float</u>
           Value that is added to an intron score for a match of intron positions in a block. Default Value: 5.0

       <b>-t</b>, <b>--intron_weight_inter</b>=<u>float</u>
           Value that is added to an intron score for a match of intron positions in an inter-block. Default
           Value: 5.0

       <b>-a</b>, <b>--alignment</b>=<u>number</u>
           Number of optimal alignments that are computed. Default Value: 1

       <b>-o</b>, <b>--out</b>=<u>format</u>

           Denotes the output format, the following output options are implemented:

               <b>score</b>
                   output is the similarity score

               <b>matrix</b>
                   output are similarity matrix and similarity score

               <b>alignment</b>
                   output are the computed alignments as

                   •   Alignment representation of P as symbols of {AminoAcid, gap symbol or number of amino
                       acids in inter-block}

                   •   Alignment representation of argmax of B as symbols of {argmax AminoAcid for aligned block
                       column, gap symbol or inter-block length}

                   •   Frequency of amino acid of P in aligned block column of B, if alignment type is a match

               <b>matrix+alignment</b>
                   output are similarity matrix, similarity score and the computed alignments in the format
                   described above

               <b>db</b>
                   output are the computed alignment as list of alignment frames, an element of the list
                   consists of:

                   •   starting position of the first amino acid of the protein sequence that is included in the
                       alignment frame

                   •   block number in which the alignment frame is located

                   •   index of the first block column that is included in the alignment frame

                   •   length of the frame (number of alignment columns)

                   •   alignment type: 'm', 's'. 'p' or '-'

               <b>bp</b>
                   output is a list of translations from the index of a block to the number of the block in the
                   .prfl file

               <b>consents</b>
                   output is the average of the argmax of the block columns for the complete profile

               <b>interblock</b>
                   output is a list of all inter-block distance intervals

                   :
                       Default Value: <b>score</b>

       <b>-h</b>, <b>--help</b>
           Produce help message.

</pre><h4><b>EXAMPLE</b></h4><pre>
           pp_simScore --fasta=EDW03868.1.fa --prfl=EOG09150290.prfl --out=alignment

</pre><h4><b>AUTHORS</b></h4><pre>
       AUGUSTUS was written by M. Stanke, O. Keller, S. König, L. Gerischer, L. Romoth and L.Gabriel.

                                                                                                  <u><a href="../man1/PP_SIMSCORE.1.html">PP_SIMSCORE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>