<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>proda - multiple alignment of protein sequences with repeats and rearrangements</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/proda">proda_1.0-14_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       proda - multiple alignment of protein sequences with repeats and rearrangements

</pre><h4><b>SYNOPSIS</b></h4><pre>

       <b>proda</b> [<u>option</u>] [<u>mfafile</u>] [&gt; <u>output</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This manual page documents briefly the <b>proda</b> command.

       <b>proda</b> (Protein Domain Aligner) is public domain software for generating multiple alignments of protein
       sequences with repeats and rearrangements, e.g. proteins with multiple domains.

       Given a set of protein sequences as input, ProDA first finds local pairwise alignments between all pairs
       of sequences, then forms blocks of alignable sequence fragments, and finally generates multiple
       alignments of the blocks. ProDA relies on many techniques used in <b>probcons</b>
       (&lt;<a href="http://probcons.stanford.edu">http://probcons.stanford.edu</a>&gt;), a recent multiple aligner that shows high accuracy in a number of
       popular benchmarks.

</pre><h4><b>INPUT</b> <b>FORMAT</b></h4><pre>
       Proda accepts input files in the MFA format. The MFA format is specified below:

       •   the MFA format consists of multiple sequences;

       •   each sequence in the MFA format begins with a single-line description, followed by lines of sequence
           data;

       •   the description line is distinguished from the sequence data by a greater-than ("&gt;") symbol in the
           first column.

</pre><h4><b>OUTPUT</b> <b>FORMAT</b></h4><pre>
       For a set of input sequences, Proda usually outputs several blocks in turn, each consists of alignable
       sequence fragments. Each block is followed by its multiple alignment.

       A block is specified by listing its sequence fragments. Each fragment is output as
       sequence_name(start-end), where sequence_name is the name of the original sequence and start and end are
       positions at which the fragment begins and ends respectively.

       Proda produces block alignments in the ClustalW (ALN) format described below:

       •   the ClustalW format consists of a single header line followed by sequence data in blocks of 50
           alignment positions;

       •   each block consists of:

           •   one line of data for each of the sequences in the alignment - in particular, the name of the
               sequence;

           •   50 characters of the alignment;

           •   one annotation line indicating fully conserved (*), strongly-conserved (:), or weakly-conserved
               columns (.);

       •   the description line is distinguished from the sequence data by a greater-than ("&gt;") symbol in the
           first column.

   <b>FASTA</b> <b>format</b> <b>for</b> <b>output</b>
       If the -fasta option is specified, then, in addition to regular output, ProDA produces a file containing
       block alignments in the FASTA format. The output file has the same name as the first input file and
       extension ".fasta". Two consecutive block alignments are separated by a line containing character ´#´.

       The FASTA format is described below:

       •   the FASTA format consists of all the sequences given in the input files;

       •   each sequence in the FASTA format begins with a single-line description, followed by lines of
           sequence data;

       •   the description line is distinguished from the sequence data by a greater-than ("&gt;") symbol in the
           first column;

       •   aligned residues are in upper case, unaligned residues are in lower case.

       Since a final alignment contains each sequence only once, this option should be used only if input
       sequences do not contain repeats.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-L</b> [min_length]

               Set minimal alignment length equal to [min_length].

               ProDA finds alignments of length greater than or equal to a threshold L. By default, L = 30. This
               option sets the threshold to [min_length].

       <b>-posterior</b>

               Use posterior decoding when computing local pairwise alignments.

               ProDA computes local pairwise alignments between two sequences using a pair-HMM and either
               Viterbi decoding or posterior decoding. The default option is using Viterbi decoding which is
               faster than posterior decoding but may be less accurate. Turning on this option instructs the
               aligner to use posterior decoding instead. In the example above, the output was generated with
               -posterior option turned on.

       <b>-silent</b>

               Do not report progress while aligning.

               Turning on this option instructs the aligner not to report the progress while aligning. By
               default, ProDA reports the progress on all pairwise alignments, block generation, and on block
               alignment. Since some stages of the algorithm, especially pairwise alignment, may take long time,
               reporting progress makes the program look alive while running.

       <b>-tran</b>

               Use transitivity when forming blocks of alignable sequence fragments.

               Two sequence fragments are directly alignable if they are parts of a local pairwise alignment. By
               default, two fragments are considered alignable if and only if they are directly alignable.
               Turning on this option instructs the aligner to consider two fragments alignable when they are
               directly alignable or when both of them are directly alignable to a third fragment.

       <b>-fasta</b>

               Use FASTA output format in addition to the ClustalW format.

               When this option is turned on, the aligner generates output in the FASTA format and stores in a
               file with the same name as the first input file and extension ".fasta", in addition to the normal
               output to stdout. This option should be used only if input sequences do not contain repeats.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b>probcons</b> (1)

</pre><h4><b>AUTHOR</b></h4><pre>
       This manual page was written by David Paleino &lt;<a href="mailto:d.paleino@gmail.com">d.paleino@gmail.com</a>&gt; for the Debian(TM) system (but may be
       used by others). This man page is released under the same conditions as the software, that is Public
       Domain.

       This software has been released in Public Domain by Phuong T.M., Do C.B., Edgar R.C. and Batzoglou S. in
       "Multiple alignment of protein sequences with repeats and rearrangements", Nucleic Acids Research 2006 -
       <a href="../man20/34.20.html">34</a>(20), 5932-5942

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright © 2007 David Paleino

                                                 april 25, 2007                                         <u><a href="../man1/PRODA.1.html">PRODA</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>