<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>profisis - protein-protein interaction sites identified from sequence</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/profisis">profisis_1.0.11-7_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       profisis - protein-protein interaction sites identified from sequence

</pre><h4><b>SYNOPSIS</b></h4><pre>
       profisis [OPTION]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       profisis (ISIS) is a machine learning-based method that identifies interacting residues from sequence
       alone. Although the method is developed using transient protein-protein interfaces from complexes of
       experimentally known 3D structures, it never explicitly uses 3D infor- mation. Instead, we combine
       predicted structural features with evolutionary information. The strongest predictions of the method
       reached over 90% accuracy in a cross-validation experiment. Our results suggest that despite the
       significant diversity in the nature of protein-protein interactions, they all share common basic
       principles and that these principles are identifiable from sequence alone.

   <b>Conversion</b> <b>of</b> <b>PSI-BLAST</b> <b>alignment</b> <b>to</b> <b>HSSP</b> <b>format</b>
       The most up-to-date procedure can be found at
       &lt;https://www.rostlab.org/owiki/index.php/How_to_generate_an_HSSP_file_from_alignment#Generating_an_HSSP_profile&gt;.

       1. Convert BLAST output to a Single Alignment Format (SAF):
            /usr/share/librg-utils-perl/blast2saf.pl fasta=&lt;query_fasta_file&gt; maxAli=3000 eSaf=1 \
             saf=&lt;saf_formatted_file&gt; &lt;blast_output&gt;

       2. Convert SAF format to HSSP:
            /usr/share/librg-utils-perl/copf.pl &lt;saf_formatted_file&gt; formatIn=saf formatOut=hssp \
             fileOut=&lt;hssp_formatted_file&gt; exeConvertSeq=convert_seq

       3. Filter results to 80% redundancy:
            /usr/share/librg-utils-perl/hssp_filter.pl red=80 &lt;hssp_formatted_file&gt; fileOut=&lt;filtered_hssp_formatted_file&gt;

   <b>Output</b> <b>format</b>
       See description of <b>--outformat</b> option.

</pre><h4><b>REFERENCES</b></h4><pre>
       Ofran, Y. and Rost, B. (2007). ISIS: interaction sites identified from sequence.  Bioinformatics, <a href="../man2/23.2.html">23</a>(2),
       e13-6.

</pre><h4><b>OPTIONS</b></h4><pre>
       Required parameters

       --fastafile
           file that contains your sequence in fasta format

       --hsspfile
           file with hssp data for sequence in --fastafile

       --rdbproffile
           file with prof output for sequence in --fastafile

       --outfile
           output file

       Optional parameters

       --outformat
           output format [pp|prval], default=pp

           pp  PredictProtein format:

                Output ::= Header_Line Binary_Out Raw_Out

                Header_Line ::= '&gt;' Header_String '\n'

                Binary_Out ::= ( Horiz_Sequence '\n' Bin_Pred '\n\n' )+

                Horiz_Sequence ::= Amino_Acid_One_Letter_Code{,40}

                Bin_Pred ::= [P-]{,40}

               'P' marks binding residue.

                Raw_Out ::= ( Amino_Acid_Number ' ' Amino_Acid_One_Letter_Code ' ' Prediction_Score '\n' )+

                Prediction_Score ::= Integer_Value

               See example outputs in <u>/usr/share/doc/profisis/examples</u>.

           prval
               ( 'resn resi predicted_value' )+, e.g.

                '1 M 25'
                '2 R 36'
                ...

       --debug
       --nodebug
           Default: --nodebug

       --succinct
           Succinct output (print no confidence values).

       Parameters  controlling  post  processing  - these parameters affect only the top part of the 'pp' output
       format

       --gap=int
           default=20

       --stretch=int
           default=5

       --crd=int
           default=7

       --crd-restriction
       --nocrd-restriction
           Default: --crd-restriction. Use original ($crd = undef) code (--crd-restriction)  or  use  new  ($cr)
           code (--nocrd-restriction).

</pre><h4><b>EXAMPLES</b></h4><pre>
        profisis --fastafile /usr/share/doc/profisis/examples/3A1P_A.fasta --hsspfile /usr/share/doc/profisis/examples/3A1P_A.hssp --rdbproffile /usr/share/doc/profisis/examples/3A1P_A.rdbProf --outfile /tmp/3A1P_A.profisis

</pre><h4><b>ENVIRONMENT</b></h4><pre>
       PROFISISCONF
           Location of configuration file to use, overriding other configuration files

</pre><h4><b>FILES</b></h4><pre>
       <u>/usr/share/profisis/profisisrc.default</u>
           Default configuration file. See this file for a description of the parameters.

       <u>/etc/profisisrc</u>
           System configuration file overriding values in <u>/usr/share/profisis/profisisrc.default</u>

       <u><a href="file:~/.profisisrc">~/.profisisrc</a></u>
           User configuration file overriding values in <u>/etc/profisisrc</u>

       <u>$PROFISISCONF</u>
           If  this  environment  variable  is set <u><a href="file:~/.profisisrc">~/.profisisrc</a></u> is disregarded and the value of the variable is
           read for configuration options overriding <u>/etc/profisisrc</u>

</pre><h4><b>AUTHOR</b></h4><pre>
       Yanay Ofran and Burkhard Rost

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/prof.1.html">prof</a></b>(1)

1.0.11                                             2022-01-18                                        <u><a href="../man1/PROFISIS.1.html">PROFISIS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>