<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>theseus_align - quick-and-dirty way to superimpose proteins</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/theseus">theseus_3.3.0-14build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       theseus_align - quick-and-dirty way to superimpose proteins

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>theseus_align</b> [<u>theseus</u> <u>options</u>] -f <u>pdbfile1.pdb</u> <u>pdbfile2.pdb</u> ...

</pre><h4><b>OPTIONS</b></h4><pre>
       The  options  given  to  the script will be passed on to <b>theseus</b>. For a complete description, see the man
       page for <b>theseus</b> (1).

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This manual page briefly documents briefly the script <b>theseus_align</b>, designed for a  quick-and-dirty  way
       to  ML  superposition  proteins  with  different  sequences.  It  should  work very well when the protein
       sequences are relatively similar, although the ML method will still give much better results than  least-
       squares when the sequences are moderately divergent.  Technically, this procedure gives a structure-based
       superposition of a sequence-based alignment. It <u>does</u> <u>not</u> perform a structure-based alignment.

       First,  the  script  uses  <b>theseus</b>  to  create  FASTA formatted sequence files corresponding to the exact
       protein sequences found in the pdb files that you supply.

       Second, these sequences are aligned using the multiple sequence alignment program  of  your  choice.  The
       script  can  easily be modified for CLUSTALW, T_COFFEE, KALIGN, DIALIGN2, or MAFFT. Any multiple sequence
       alignment program can be used, as long as it can generate  clustal-formatted  files.  However,  I  highly
       recommend   Bob   Edgar's   MUSCLE  program  for  both  its  speed  and  accuracy.  (For  more  info  see
       <a href="http://www.drive5.com/muscle/">http://www.drive5.com/muscle/</a> .)

       Third, <b>theseus</b> performs a superposition of the structures using the sequence alignment as a guide.

       The installed version of <b>theseus_align</b> uses <b>muscle</b> (1) for doing the multiple sequence alignment. If  you
       wish  to  use  one  of  the  other  programs  mentioned above, you'll have to copy the script to your own
       directory and edit it.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b>theseus</b> (1), <b>muscle</b> (1), <b>clustalw</b> (1), <b>t_coffee</b> (1), <b>kalign</b> (1), <b>dialign2</b> (1), <b>mafft</b> (1).  All  of  these
       programs can be installed on Debian or Ubuntu systems using <b>apt-get</b> (8).

</pre><h4><b>AUTHOR</b></h4><pre>
       <b>theseus_align</b> was written by Douglas L. Theobald, Department of Biochemistry, Brandeis University.

                                                 November, 2008                                 <u><a href="../man1/THESEUS_ALIGN.1.html">THESEUS_ALIGN</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>