<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>run_tipp.py - an identification and phylogenetic profiling tool</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/tipp">tipp_1.0+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       run_tipp.py - an identification and phylogenetic profiling tool

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: run_tipp.py [-h] [-v] [-A N] [-P N] [-F N] [--distance DISTANCE]

              [-M  DIAMETER]  [-S  DECOMP]  [-p DIR] [-rt] [-o OUTPUT] [-d OUTPUT_DIR] [-c CONFIG] [-t TREE] [-r
              RAXML] [-a ALIGN] [-f FRAG] [-m MOLECULE] [--ignore-overlap]  [-x  N]  [-cp  CHCK_FILE]  [-cpi  N]
              [-seed N] [-R N] [-at N] [-D] [-pt N] [-PD N] [-tx TAXONOMY] [-txm MAPPING] [-adt TREE] [-C N]

       This  script  runs the SEPP algorithm on an input tree, alignment, fragment file, and RAxML info file. It
       uses      a      reference      dataset      which      has       to       be       downloaded       from
       <b>https://obj.umiacs.umd.edu/tipp/tipp2-refpkg.tar.gz</b>

       If  the  local administrator has not set the path to this reference dataset in /etc/tipp/tipp.config, you
       should copy this file to <a href="file:~/.tipp/">~/.tipp/</a> and put the path to  the  dataset  in  the  <b>reference</b>  section  of  the
       configuration file, see <b><a href="../man5/tipp.config.5.html">tipp.config</a></b>(5).

   <b>optional</b> <b>arguments:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>-v</b>, <b>--version</b>
              show program's version number and exit

   <b>DECOMPOSITION</b> <b>OPTIONS:</b>
              These  options  determine  the  alignment  decomposition size and taxon insertion size. If None is
              given, then the default is to align/place at 10% of total taxa. The  alignment  decomosition  size
              must be less than the taxon insertion size.

       <b>-A</b> N, <b>--alignmentSize</b> N
              max  alignment  subset size of N [default: 10% of the total number of taxa or the placement subset
              size if given]

       <b>-P</b> N, <b>--placementSize</b> N
              max placement subset size of N [default: 10% of the total number of taxa or the  alignment  length
              (whichever bigger)]

       <b>-F</b> N, <b>--fragmentChunkSize</b> N
              maximum fragment chunk size of N. Helps controlling memory. [default: 20000]

       <b>--distance</b> DISTANCE
              minimum p-distance before stopping the decomposition[default: 1]

       <b>-M</b> DIAMETER, <b>--diameter</b> DIAMETER
              maximum tree diameter before stopping the decomposition[default: None]

       <b>-S</b> DECOMP, <b>--decomp_strategy</b> DECOMP
              decomposition strategy [default: using tree branch length]

   <b>OUTPUT</b> <b>OPTIONS:</b>
              These options control output.

       <b>-p</b> DIR, <b>--tempdir</b> DIR
              Tempfile files will be written to DIR. Full-path required. [default: /tmp/sepp]

       <b>-rt</b>, <b>--remtemp</b>
              Remove template directory. [default: disabled]

       <b>-o</b> OUTPUT, <b>--output</b> OUTPUT
              output files with prefix OUTPUT. [default: output]

       <b>-d</b> OUTPUT_DIR, <b>--outdir</b> OUTPUT_DIR
              output to OUTPUT_DIR directory. full-path required.  [default: .]

   <b>INPUT</b> <b>OPTIONS:</b>
              These  options  control  input.  To run SEPP the following is required. A backbone tree (in newick
              format), a RAxML_info file (this is the file generated by RAxML during estimation of the  backbone
              tree.  Pplacer  uses  this info file to set model parameters), a backbone alignment file (in fasta
              format), and a fasta file including fragments. The input sequences are assumed to  be  DNA  unless
              specified otherwise.

       <b>-c</b> CONFIG, <b>--config</b> <b>CONFIG</b>
              A  config  file,  including  options used to run SEPP.  Options provided as command line arguments
              overwrite config file values for those options. [default: None]

       <b>-t</b> TREE, <b>--tree</b> TREE
              Input tree file (newick format) [default: None]

       <b>-r</b> RAXML, <b>--raxml</b> RAXML
              RAxML_info file including model parameters, generated by RAxML.[default: None]

       <b>-a</b> ALIGN, <b>--alignment</b> ALIGN
              Aligned fasta file [default: None]

       <b>-f</b> FRAG, <b>--fragment</b> FRAG
              fragment file [default: None]

       <b>-m</b> MOLECULE, <b>--molecule</b> MOLECULE
              Molecule type of sequences. Can be amino, dna, or rna [default: dna]

       <b>--ignore-overlap</b>
              When a query sequence has the same name as a backbone sequence, ignore  the  query  sequences  and
              keep the backbone sequence [default: False]

   <b>OTHER</b> <b>OPTIONS:</b>
              These options control how SEPP is run

       <b>-x</b> N, <b>--cpu</b> N
              Use N cpus [default: number of cpus available on the machine]

       <b>-cp</b> CHCK_FILE, <b>--checkpoint</b> CHCK_FILE
              checkpoint file [default: no checkpointing]

       <b>-cpi</b> N, <b>--interval</b> N
              Interval  (in  seconds)  between  checkpoint  writes. Has effect only with <b>-cp</b> provided. [default:
              3600]

       <b>-seed</b> N, <b>--randomseed</b> N
              random seed number. [default: 297834]

   <b>TIPP</b> <b>OPTIONS:</b>
              These arguments set settings specific to TIPP

       <b>-R</b> N, <b>--reference_pkg</b> N
              Use a pre-computed reference package [default: None]

       <b>-at</b> N, <b>--alignmentThreshold</b> N
              Enough alignment subsets are selected to reach a commulative probability of N. This  should  be  a
              number between 0 and 1 [default: 0.95]

       <b>-D</b>, <b>--dist</b>
              Treat fragments as distribution

       <b>-pt</b> N, <b>--placementThreshold</b> N
              Enough  placements  are  selected to reach a commulative probability of N. This should be a number
              between 0 and 1 [default: 0.95]

       <b>-PD</b> N, <b>--push_down</b> N
              Whether to classify based on children below or above insertion point. [default: True]

       <b>-tx</b> TAXONOMY, <b>--taxonomy</b> TAXONOMY
              A file describing the taxonomy. This is a commaseparated text file that has the following  fields:
              taxon_id,parent_id,taxon_name,rank.  If  there are other columns, they are ignored. The first line
              is also ignored.

       <b>-txm</b> MAPPING, <b>--taxonomyNameMapping</b> MAPPING
              A comma-separated text file mapping alignment sequence  names  to  taxonomic  ids.  Formats  (each
              line):  sequence_name,taxon_id.  If  there  are other columns, they are ignored. The first line is
              also ignored.

       <b>-adt</b> TREE, <b>--alignmentDecompositionTree</b> TREE
              A newick tree file used for decomposing taxa into alignment subsets. [default: the backbone tree]

       <b>-C</b> N, <b>--cutoff</b> N
              Placement probability requirement to count toward  the  distribution.  This  should  be  a  number
              between 0 and 1 [default: 0.0]

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/run_sepp.py.1.html">run_sepp.py</a></b>(1), <b><a href="../man5/tipp.config.5.html">tipp.config</a></b>(5)

run_tipp.py                                      September 2021                                   <u><a href="../man1/RUN_TIPP.PY.1.html">RUN_TIPP.PY</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>