<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>iHS</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/libvcflib-tools">libvcflib-tools_1.0.12+dfsg-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       <b>iHS</b>

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>iHS</b> –target 0,1,2,3,4,5,6,7 –file my.phased.vcf  –region chr1:1-1000 &gt; STDOUT 2&gt; STDERR

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>iHS</b> calculates the integrated haplotype score which measures the relative decay of extended haplotype ho‐
       mozygosity  (EHH)  for the reference and alternative alleles at a site (see: voight et al. 2006, Spiech &amp;
       Hernandez 2014).

</pre><h4><b>OPTIONS</b></h4><pre>
              Our code is highly concordant with both implementations mentioned. However, we do not set an upper limit to the allele frequency.  **iHS** can be run without a genetic map, in which case the change in EHH is integrated over a constant.  Human genetic maps for GRCh36 and GRCh37 (hg18 &amp; hg19) can be found at: <a href="http://bochet.gcc.biostat.washington.edu/beagle/genetic_maps/">http://bochet.gcc.biostat.washington.edu/beagle/genetic_maps/</a> . **iHS** by default interpolates SNV positions to genetic position (you don't need a genetic position for every VCF entry in the map file).

              **iHS** analyses requires normalization by allele frequency.  It is important that **iHS** is calculated over large regions so that the normalization does not down weight real signals.  For genome-wide runs it is recommended to run slightly overlapping windows and throwing out values that fail integration (columns 7 &amp; 8 in the output) and then removing duplicates by using the 'sort' and 'uniq' linux commands.  Normalization of the output is as simple as running 'normalize-**iHS**'.

                   **iHS** calculates the integrated ratio of haplotype decay between the reference and non-reference allele.
              Output : 4 columns :
                   1. seqid
                   2. position
                   3. target allele frequency
                   4. integrated EHH (alternative)
                   5. integrated EHH (reference)
                   6. **iHS** ln(iEHHalt/iEHHref)
                   7. != 0 integration failure
                   8. != 0 integration failure

              Params:
                     required: t,target  &lt;STRING&gt;  A zero base comma separated list of target
                                                   individuals corresponding to VCF columns
                     required: r,region  &lt;STRING&gt;  A tabix compliant genomic range
                                                   format: "seqid:start-end" or "seqid"
                     required: f,file    &lt;STRING&gt;  Proper formatted and phased VCF.
                     required: y,type    &lt;STRING&gt;  Genotype likelihood format: GT,PL,GL,GP
                     optional: a,af      &lt;DOUBLE&gt;  Alternative alleles with frquences less
                                                   than [0.05] are skipped.
                     optional: x,threads &lt;INT&gt;     Number of CPUS [1].
                     recommended: g,gen &lt;STRING&gt;   A PLINK formatted map file.

              Type: statistics

</pre><h4><b>EXIT</b> <b>VALUES</b></h4><pre>
       <b>0</b>      Success

       <b>not</b> <b>0</b>  Failure

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/vcflib.1.html">vcflib</a></b>(1)

</pre><h4><b>OTHER</b></h4><pre>
</pre><h4><b>LICENSE</b></h4><pre>
       Copyright 2011-2024 (C) Erik Garrison and vcflib contributors.  MIT licensed.   Copyright  2020-2024  (C)
       Pjotr Prins.

</pre><h4><b>AUTHORS</b></h4><pre>
       Erik Garrison and vcflib contributors.

iHS (vcflib)                                                                                              <u><a href="../man1/IHS.1.html">IHS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>