<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>minimap - fast mapping between long DNA sequences</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/minimap">minimap_0.2-8_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       minimap - fast mapping between long DNA sequences

</pre><h4><b>SYNOPSIS</b></h4><pre>
       minimap  [<b>-lSOV</b>]  [<b>-k</b>  <u>kmer</u>]  [<b>-w</b>  <u>winSize</u>] [<b>-I</b> <u>batchSize</u>] [<b>-d</b> <u>dumpFile</u>] [<b>-f</b> <u>occThres</u>] [<b>-r</b> <u>bandWidth</u>] [<b>-m</b>
       <u>minShared</u>] [<b>-c</b> <u>minCount</u>] [<b>-L</b> <u>minMatch</u>] [<b>-g</b> <u>maxGap</u>] [<b>-T</b> <u>dustThres</u>] [<b>-t</b>  <u>nThreads</u>]  [<b>-x</b>  <u>preset</u>]  <u>target.fa</u>
       <u>query.fa</u> &gt; <u>output.paf</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Minimap  is  a  tool  to efficiently find multiple approximate mapping positions between two sets of long
       sequences, such as between reads and reference genomes, between genomes and  between  long  noisy  reads.
       Minimap has an indexing and a mapping phase. In the indexing phase, it collects all minimizers of a large
       batch  of target sequences in a hash table; in the mapping phase, it identifies good clusters of colinear
       minimizer hits. Minimap does not generate detailed alignments between the target and the query sequences.
       It only outputs the approximate start and the end coordinates of these clusters.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Indexing</b> <b>options</b>
       <b>-k</b> <u>INT</u>    Minimizer k-mer length [15]

       <b>-w</b> <u>INT</u>    Minimizer window size [2/3 of k-mer length]. A minimizer is the smallest k-mer in a window of w
                 consecutive k-mers.

       <b>-I</b> <u>NUM</u>    Load at most <u>NUM</u> target bases into RAM for indexing [4G]. If there are more than <u>NUM</u>  bases  in
                 <u>target.fa</u>, minimap needs to read <u>query.fa</u> multiple times to map it against each batch of target
                 sequences.  <u>NUM</u> may be ending with k/K/m/M/g/G.

       <b>-d</b> <u>FILE</u>   Dump minimizer index to <u>FILE</u> [no dump]

       <b>-l</b>        Indicate  that  <u>target.fa</u>  is  in fact a minimizer index generated by option <b>-d</b>, not a FASTA or
                 FASTQ file.

   <b>Mapping</b> <b>options</b>
       <b>-f</b> <u>FLOAT</u>  Ignore top <u>FLOAT</u> fraction of most occurring minimizers [0.001]

       <b>-r</b> <u>INT</u>    Approximate bandwidth for initial minimizer  hits  clustering  [500].  A  <u>minimizer</u>  <u>hit</u>  is  a
                 minimizer present in both the target and query sequences. A <u>minimizer</u> <u>hit</u> <u>cluster</u> is a group of
                 potentially colinear minimizer hits between a target and a query sequence.

       <b>-m</b> <u>FLOAT</u>  Merge  initial  minimizer  hit  clusters  if  <u>FLOAT</u> or higher fraction of minimizers are shared
                 between the clusters [0.5]

       <b>-c</b> <u>INT</u>    Retain a minimizer hit cluster if it contains <u>INT</u> or more minimizer hits [4]

       <b>-L</b> <u>INT</u>    Discard a minimizer hit cluster if after colinearization, the number of matching bases is below
                 <u>INT</u> [40]. This option mainly reduces the size of output. It has little effect on the speed  and
                 peak memory.

       <b>-g</b> <u>INT</u>    Split  a  minimizer  hit  cluster at a gap <u>INT</u>-bp or longer that does not contain any minimizer
                 hits [10000]

       <b>-T</b> <u>INT</u>    Mask regions on query sequences with SDUST score threshold <u>INT</u>; 0 to disable [0]. SDUST  is  an
                 algorithm  to  identify  low-complexity subsequences. It is not enabled by default. If SDUST is
                 preferred, a value between 20 and 25 is recommended. A higher threshold masks less sequences.

       <b>-S</b>        Perform all-vs-all mapping. In this mode, if  the  query  sequence  name  is  lexicographically
                 larger  than  the  target sequence name, the hits between them will be suppressed; if the query
                 sequence name is the same as the target name, diagonal minimizer hits will also be suppressed.

       <b>-O</b>        Drop a minimizer hit if it is far away from other hits (EXPERIMENTAL). This  option  is  useful
                 for mapping long chromosomes from two diverged species.

       <b>-x</b> <u>STR</u>    Changing  multiple  settings  based  on  <u>STR</u>  [not set]. It is recommended to apply this option
                 before other options, such that the  following  options  may  override  the  multiple  settings
                 modified by this option.

                 <b>ava10k</b>  for PacBio or Oxford Nanopore all-vs-all read mapping (-Sw5 -L100 -m0).

   <b>Input/output</b> <b>options</b>
       <b>-t</b> <u>INT</u>    Number  of threads [3]. Minimap uses at most three threads when collecting minimizers on target
                 sequences, and uses up to <u>INT</u>+1 threads when mapping (the extra thread is  for  I/O,  which  is
                 frequently idle and takes little CPU time).

       <b>-V</b>        Print version number to stdout

</pre><h4><b>OUTPUT</b> <b>FORMAT</b></h4><pre>
       Minimap  outputs  mapping  positions  in  the  Pairwise mApping Format (PAF). PAF is a TAB-delimited text
       format with each line consisting of at least 12 fields as are described in the following table:

                    ┌─────┬────────┬─────────────────────────────────────────────────────────────┐
                    │ <b>Col</b> │  <b>Type</b>  │                         <b>Description</b>                         │
                    ├─────┼────────┼─────────────────────────────────────────────────────────────┤
                    │   1 │ string │ Query sequence name                                         │
                    │   2 │  int   │ Query sequence length                                       │
                    │   3 │  int   │ Query start coordinate (0-based)                            │
                    │   4 │  int   │ Query end coordinate (0-based)                              │
                    │   5 │  char  │ `+' if query and target on the same strand; `-' if opposite │
                    │   6 │ string │ Target sequence name                                        │
                    │   7 │  int   │ Target sequence length                                      │
                    │   8 │  int   │ Target start coordinate on the original strand              │
                    │   9 │  int   │ Target end coordinate on the original strand                │
                    │  10 │  int   │ Number of matching bases in the mapping                     │
                    │  11 │  int   │ Number bases, including gaps, in the mapping                │
                    │  12 │  int   │ Mapping quality (0-255 with 255 for missing)                │
                    └─────┴────────┴─────────────────────────────────────────────────────────────┘

       When the alignment is available, column 11 gives the total number of  sequence  matches,  mismatches  and
       gaps  in  the alignment; column 10 divided by column 11 gives the alignment identity. As minimap does not
       generate detailed alignment, these two columns are approximate. PAF may optionally have additional fields
       in the SAM-like typed key-value format. Minimap writes the number of minimizer hits in a cluster  to  the
       cm tag.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <a href="../man1/miniasm.1.html">miniasm</a>(1)

minimap-0.2                                     06 December 2015                                      <u><a href="../man1/minimap.1.html">minimap</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>