<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>last-train - Try to find suitable score parameters for aligning the given sequences</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/last-align">last-align_1609-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       last-train - Try to find suitable score parameters for aligning the given sequences

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>last-train</b> [<u>options</u>] <u>lastdb-name</u> <u>sequence-file(s)</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Try to find suitable score parameters for aligning the given sequences.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>-v</b>, <b>--verbose</b>
              show more details of intermediate steps

              Training options:

       <b>--revsym</b>
              force reverse-complement symmetry

       <b>--matsym</b>
              force symmetric substitution matrix

       <b>--gapsym</b>
              force insertion/deletion symmetry

       <b>--fixmat</b>=<u>NAME</u>
              fix the substitution probabilities to e.g. BLOSUM62

       <b>--pid</b>=<u>PID</u>
              skip alignments with &gt; PID% identity (default: 100)

       <b>--postmask</b>=<u>NUMBER</u>
              skip mostly-lowercase alignments (default=1)

       <b>--sample-number</b>=<u>N</u>
              number of random sequence samples (default: 20000 if <b>--codon</b> else 500)

       <b>--sample-length</b>=<u>L</u>
              length of each sample (default: 2000)

       <b>--scale</b>=<u>S</u>
              output scores in units of 1/S bits

       <b>--codon</b>
              DNA queries &amp; protein reference, with frameshifts

              Initial parameter options:

       <b>-r</b> SCORE
              match score

       <b>-q</b> COST
              mismatch cost

       <b>-p</b> NAME
              match/mismatch score matrix

       <b>-a</b> COST
              gap existence cost

       <b>-b</b> COST
              gap extension cost

       <b>-A</b> COST
              insertion existence cost

       <b>-B</b> COST
              insertion extension cost

       <b>-F</b> LIST
              frameshift probabilities: del-1,del-2,ins+1,ins+2 (default: 1-b,1-b,1-B,1-B)

              Alignment options:

       <b>-D</b> LENGTH
              query letters per random alignment (default: total sample length)

       <b>-E</b> EG2 maximum expected alignments per square giga

       <b>-s</b> STRAND
              0=reverse, 1=forward, 2=both (default: 2 if DNA and not lastdb <b>-S2</b>, else 1)

       <b>-S</b> NUMBER
              use score matrix: 0=as-is, 1=on query forward strands (default: 1)

       <b>-C</b> COUNT
              omit gapless alignments in COUNT others with &gt; scoreper-length

       <b>-T</b> NUMBER
              type of alignment: 0=local, 1=overlap (default: 0)

       <b>-R</b> DIGITS
              lowercase &amp; simple-sequence options

       <b>-m</b> COUNT
              maximum initial matches per query position (default: 10)

       <b>-k</b> STEP
              use initial matches starting at every STEP-th position in each query (default: 1)

       <b>-P</b> THREADS
              number of parallel threads

       <b>-X</b> NUMBER
              N/X is ambiguous in: 0=neither sequence, 1=reference, 2=query, 3=both (default=0)

       <b>-Q</b> NAME
              input format: fastx, sanger (default=fasta)

last-train 1609                                   February 2025                                    <u><a href="../man1/LAST-TRAIN.1.html">LAST-TRAIN</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>