<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::DB::Fasta - Fast indexed access to fasta files</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::DB::Fasta - Fast indexed access to fasta files

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::DB::Fasta;

         # Create database from a directory of Fasta files
         my $db       = Bio::DB::Fasta-&gt;new('/path/to/fasta/files/');
         my @ids      = $db-&gt;get_all_primary_ids;

         # Simple access
         my $seqstr   = $db-&gt;seq('CHROMOSOME_I', 4_000_000 =&gt; 4_100_000);
         my $revseq   = $db-&gt;seq('CHROMOSOME_I', 4_100_000 =&gt; 4_000_000);
         my $length   = $db-&gt;length('CHROMOSOME_I');
         my $header   = $db-&gt;header('CHROMOSOME_I');
         my $alphabet = $db-&gt;alphabet('CHROMOSOME_I');

         # Access to sequence objects. See Bio::PrimarySeqI.
         my $seq     = $db-&gt;get_Seq_by_id('CHROMOSOME_I');
         my $seqstr  = $seq-&gt;seq;
         my $subseq  = $seq-&gt;subseq(4_000_000 =&gt; 4_100_000);
         my $trunc   = $seq-&gt;trunc(4_000_000 =&gt; 4_100_000);
         my $length  = $seq-&gt;length;

         # Loop through sequence objects
         my $stream  = $db-&gt;get_PrimarySeq_stream;
         while (my $seq = $stream-&gt;next_seq) {
           # Bio::PrimarySeqI stuff
         }

         # Filehandle access
         my $fh = Bio::DB::Fasta-&gt;newFh('/path/to/fasta/files/');
         while (my $seq = &lt;$fh&gt;) {
           # Bio::PrimarySeqI stuff
         }

         # Tied hash access
         tie %sequences,'Bio::DB::Fasta','/path/to/fasta/files/';
         print $sequences{'CHROMOSOME_I:1,20000'};

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::DB::Fasta provides indexed access to a single Fasta file, several files, or a directory of files. It
       provides persistent random access to each sequence entry (either as a Bio::PrimarySeqI-compliant object
       or a string), and to subsequences within each entry, allowing you to retrieve portions of very large
       sequences without bringing the entire sequence into memory. Bio::DB::Fasta is based on
       Bio::DB::IndexedBase. See this module's documentation for details.

       The Fasta files may contain any combination of nucleotide and protein sequences; during indexing the
       module guesses the molecular type. Entries may have any line length up to 65,536 characters, and
       different line lengths are allowed in the same file.  However, within a sequence entry, all lines must be
       the same length except for the last. An error will be thrown if this is not the case.

       The module uses /^&gt;(\S+)/ to extract the primary ID of each sequence from the Fasta header. See -makeid
       in Bio::DB::IndexedBase to pass a callback routine to reversibly modify this primary ID, e.g. if you wish
       to extract a specific portion of the gi|gb|abc|xyz GenBank IDs.

</pre><h4><b>DATABASE</b> <b>CREATION</b> <b>AND</b> <b>INDEXING</b></h4><pre>
       The object-oriented constructor is <b>new()</b>, the filehandle constructor is <b>newFh()</b> and the tied hash
       constructor is <b>tie()</b>. They all allow one to index a single Fasta file, several files, or a directory of
       files. See Bio::DB::IndexedBase.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Bio::DB::IndexedBase

       Bio::DB::Qual

       Bio::PrimarySeqI

</pre><h4><b>AUTHOR</b></h4><pre>
       Lincoln Stein &lt;<a href="mailto:lstein@cshl.org">lstein@cshl.org</a>&gt;.

       Copyright (c) 2001 Cold Spring Harbor Laboratory.

       This library is free software; you can redistribute it and/or modify it under the same terms as Perl
       itself.  See DISCLAIMER.txt for disclaimers of warranty.

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

       For BioPerl-style access, the following methods are provided:

   <b>get_Seq_by_id</b>
        Title   : get_Seq_by_id, get_Seq_by_acc, get_Seq_by_primary_id
        Usage   : my $seq = $db-&gt;get_Seq_by_id($id);
        Function: Given an ID, fetch the corresponding sequence from the database.
        Returns : A Bio::PrimarySeq::Fasta object (Bio::PrimarySeqI compliant)
                  Note that to save resource, Bio::PrimarySeq::Fasta sequence objects
                  only load the sequence string into memory when requested using seq().
                  See L&lt;Bio::PrimarySeqI&gt; for methods provided by the sequence objects
                  returned from get_Seq_by_id() and get_PrimarySeq_stream().
        Args    : ID

   <b>get_PrimarySeq_stream</b>
        Title   : get_PrimarySeq_stream
        Usage   : my $stream = $db-&gt;get_PrimarySeq_stream();
        Function: Get a stream of Bio::PrimarySeq::Fasta objects. The stream supports a
                  single method, next_seq(). Each call to next_seq() returns a new
                  Bio::PrimarySeq::Fasta sequence object, until no more sequences remain.
        Returns : A Bio::DB::Indexed::Stream object
        Args    : None

       For simple access, the following methods are provided:

   <b>new</b>
        Title   : new
        Usage   : my $db = Bio::DB::Fasta-&gt;new( $path, %options);
        Function: Initialize a new database object. When indexing a directory, files
                  ending in .fa,fasta,fast,dna,fna,faa,fsa are indexed by default.
        Returns : A new Bio::DB::Fasta object.
        Args    : A single file, or path to dir, or arrayref of files
                  Optional arguments: see Bio::DB::IndexedBase

   <b>seq</b>
        Title   : seq, sequence, subseq
        Usage   : # Entire sequence string
                  my $seqstr    = $db-&gt;seq($id);
                  # Subsequence
                  my $subseqstr = $db-&gt;seq($id, $start, $stop, $strand);
                  # or...
                  my $subseqstr = $db-&gt;seq($compound_id);
        Function: Get a subseq of a sequence from the database. For your convenience,
                  the sequence to extract can be specified with any of the following
                  compound IDs:
                     $db-&gt;seq("$id:$start,$stop")
                     $db-&gt;seq("$id:$start..$stop")
                     $db-&gt;seq("$id:$start-$stop")
                     $db-&gt;seq("$id:$start,$stop/$strand")
                     $db-&gt;seq("$id:$start..$stop/$strand")
                     $db-&gt;seq("$id:$start-$stop/$strand")
                     $db-&gt;seq("$id/$strand")
                  In the case of DNA or RNA sequence, if $stop is less than $start,
                  then the reverse complement of the sequence is returned. Avoid using
                  it if possible since this goes against Bio::Seq conventions.
        Returns : A string
        Args    : ID of sequence to retrieve
                    or
                  Compound ID of subsequence to fetch
                    or
                  ID, optional start (defaults to 1), optional end (defaults to length
                  of sequence) and optional strand (defaults to 1).

   <b>length</b>
        Title   : length
        Usage   : my $length = $qualdb-&gt;length($id);
        Function: Get the number of residues in the indicated sequence.
        Returns : Number
        Args    : ID of entry

   <b>header</b>
        Title   : header
        Usage   : my $header = $db-&gt;header($id);
        Function: Get the header line (ID and description fields) of the specified
                  sequence.
        Returns : String
        Args    : ID of sequence

   <b>alphabet</b>
        Title   : alphabet
        Usage   : my $alphabet = $db-&gt;alphabet($id);
        Function: Get the molecular type of the indicated sequence: dna, rna or protein
        Returns : String
        Args    : ID of sequence

perl v5.32.1                                       2021-08-15                                <u>Bio::DB::<a href="../man3pm/Fasta.3pm.html">Fasta</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>