<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::GuessSeqFormat - Module for determining the sequence format of the contents of a file, a</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::GuessSeqFormat - Module for determining the sequence format of the contents of a file, a
       string, or through a filehandle.

</pre><h4><b>SYNOPSIS</b></h4><pre>
           # To guess the format of a flat file, given a filename:
           my $guesser = Bio::Tools::GuessSeqFormat-&gt;new( -file =&gt; $filename );
           my $format  = $guesser-&gt;guess;

           # To guess the format from an already open filehandle:
           my $guesser = Bio::Tools::GuessSeqFormat-&gt;new( -fh =&gt; $filehandle );
           my $format  = $guesser-&gt;guess;
           # The filehandle will be returned to its original position. Note that this
           # filehandle can be STDIN.

           # To guess the format of one or several lines of text (with
           # embedded newlines):
           my $guesser = Bio::Tools::GuessSeqFormat-&gt;new( -text =&gt; $linesoftext );
           my $format = $guesser-&gt;guess;

           # To create a Bio::Tools::GuessSeqFormat object and set the
           # filename, filehandle, or line to parse afterwards:
           my $guesser = Bio::Tools::GuessSeqFormat-&gt;new();
           $guesser-&gt;file($filename);
           $guesser-&gt;fh($filehandle);
           $guesser-&gt;text($linesoftext);

           # To guess in one go, given e.g. a filename:
           my $format = Bio::Tools::GuessSeqFormat-&gt;new( -file =&gt; $filename )-&gt;guess;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::Tools::GuessSeqFormat tries to guess the format ("swiss", "pir", "fasta" etc.) of the sequence or
       MSA in a file, in a scalar, or through a filehandle.

       The <b>guess()</b> method of a Bio::Tools::GuessSeqFormat object will examine the data, line by line, until it
       finds a line to which only one format can be assigned.  If no conclusive guess can be made, undef is
       returned.

       If the Bio::Tools::GuessSeqFormat object is given a filehandle, e.g. STDIN, it will be restored to its
       original position on return from the <b>guess()</b> method.

   <b>Formats</b>
       Tests are currently implemented for the following formats:

       •   ACeDB ("ace")

       •   Blast ("blast")

       •   ClustalW ("clustalw")

       •   Codata ("codata")

       •   EMBL ("embl")

       •   FastA sequence ("fasta")

       •   FastQ sequence ("fastq")

       •   FastXY/FastA alignment ("fastxy")

       •   Game XML ("game")

       •   GCG ("gcg")

       •   GCG Blast ("gcgblast")

       •   GCG FastA ("gcgfasta")

       •   GDE ("gde")

       •   Genbank ("genbank")

       •   Genscan ("genscan")

       •   GFF ("gff")

       •   HMMER ("hmmer")

       •   PAUP/NEXUS ("nexus")

       •   Phrap assembly file ("phrap")

       •   NBRF/PIR ("pir")

       •   Mase ("mase")

       •   Mega ("mega")

       •   GCG/MSF ("msf")

       •   Pfam ("pfam")

       •   Phylip ("phylip")

       •   Prodom ("prodom")

       •   Raw ("raw")

       •   RSF ("rsf")

       •   Selex ("selex")

       •   Stockholm ("stockholm")

       •   Swissprot ("swiss")

       •   Tab ("tab")

       •   Variant Call Format ("vcf")

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules.  Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather  than  to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.   Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b></h4><pre>
       Andreas Kähäri, <a href="mailto:andreas.kahari@ebi.ac.uk">andreas.kahari@ebi.ac.uk</a>

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
       Heikki Lehväslaiho, heikki-at-bioperl-dot-org Mark A. Jensen, maj-at-fortinbras-dot-us

</pre><h4><b>METHODS</b></h4><pre>
       Methods  available  to  Bio::Tools::GuessSeqFormat  objects  are  described  below.   Methods  with names
       beginning with an underscore are considered to be internal.

   <b>new</b>
        Title      : new
        Usage      : $guesser = Bio::Tools::GuessSeqFormat-&gt;new( ... );
        Function   : Creates a new object.
        Example    : See SYNOPSIS.
        Returns    : A new object.
        Arguments  : -file The filename of the file whose format is to
                           be guessed, e.g. STDIN, or
                     -fh   An already opened filehandle from which a text
                           stream may be read, or
                     -text A scalar containing one or several lines of
                           text with embedded newlines.

           If more than one of the above arguments are given, they
           are tested in the order -text, -file, -fh, and the first
           available argument will be used.

   <b>file</b>
        Title      : file
        Usage      : $guesser-&gt;file($filename);
                     $filename = $guesser-&gt;file;
        Function   : Gets or sets the current filename associated with
                     an object.
        Returns    : The new filename.
        Arguments  : The filename of the file whose format is to be
                     guessed.

           A call to this method will clear the current filehandle and
           the current lines of text associated with the object.

   <b>fh</b>
        Title      : fh
        Usage      : $guesser-&gt;fh($filehandle);
                     $filehandle = $guesser-&gt;fh;
        Function   : Gets or sets the current filehandle associated with
                     an object.
        Returns    : The new filehandle.
        Arguments  : An already opened filehandle from which a text
                     stream may be read.

           A call to this method will clear the current filename and
           the current lines of text associated with the object.

   <b>text</b>
        Title      : text
        Usage      : $guesser-&gt;text($linesoftext);
                     $linesofext = $guesser-&gt;text;
        Function   : Gets or sets the current text associated with an
                     object.
        Returns    : The new lines of texts.
        Arguments  : A scalar containing one or several lines of text,
                     including embedded newlines.

           A call to this method will clear the current filename and
           the current filehandle associated with the object.

   <b>guess</b>
        Title      : guess
        Usage      : $format = $guesser-&gt;guess;
                     @format = $guesser-&gt;guess; # if given a line of text
        Function   : Guesses the format of the data accociated with the
                     object.
        Returns    : A format string such as "swiss" or "pir".  If a
                     format can not be found, undef is returned.
        Arguments  : None.

           If the object is associated with a filehandle, the position
           of the filehandle will be returned to its original position
           before the method returns.

</pre><h4><b>HELPER</b> <b>SUBROUTINES</b></h4><pre>
       All helper subroutines will, given a line of text and the line number of the same line, return 1  if  the
       line possibly is from a file of the type that they perform a test of.

       A zero return value does not mean that the line is not part of a certain type of file, just that the test
       did not find any characteristics of that type of file in the line.

   <b>_possibly_ace</b>
       From                bioperl                test                data,               and               from
       "<a href="http://www.isrec.isb-sib.ch/DEA/module8/B_Stevenson/Practicals/transcriptome_recon/transcriptome_recon.html">http://www.isrec.isb-sib.ch/DEA/module8/B_Stevenson/Practicals/transcriptome_recon/transcriptome_recon.html</a>".

   <b>_possibly_blast</b>
        From various blast results.

   <b>_possibly_bowtie</b>
       Contributed by kortsch.

   <b>_possibly_clustalw</b>
       From "<a href="http://www.ebi.ac.uk/help/formats.html">http://www.ebi.ac.uk/help/formats.html</a>".

   <b>_possibly_codata</b>
       From "<a href="http://www.ebi.ac.uk/help/formats.html">http://www.ebi.ac.uk/help/formats.html</a>".

   <b>_possibly_embl</b>
       From "<a href="http://www.ebi.ac.uk/embl/Documentation/User_manual/usrman.html">http://www.ebi.ac.uk/embl/Documentation/User_manual/usrman.html</a>#3.3".

   <b>_possibly_fasta</b>
       From "<a href="http://www.ebi.ac.uk/help/formats.html">http://www.ebi.ac.uk/help/formats.html</a>".

   <b>_possibly_fastq</b>
       From bioperl test data.

   <b>_possibly_fastxy</b>
       From bioperl test data.

   <b>_possibly_game</b>
       From bioperl testdata.

   <b>_possibly_gcg</b>
       From bioperl, Bio::SeqIO::gcg.

   <b>_possibly_gcgblast</b>
       From bioperl testdata.

   <b>_possibly_gcgfasta</b>
       From bioperl testdata.

   <b>_possibly_gde</b>
       From "<a href="http://www.ebi.ac.uk/help/formats.html">http://www.ebi.ac.uk/help/formats.html</a>".

   <b>_possibly_genbank</b>
       From  "<a href="http://www.ebi.ac.uk/help/formats.html">http://www.ebi.ac.uk/help/formats.html</a>".   Format  of  [apparantly  optional]  file  header   from
       "<a href="http://www.umdnj.edu/rcompweb/PA/Notes/GenbankFF.htm">http://www.umdnj.edu/rcompweb/PA/Notes/GenbankFF.htm</a>". (TODO: dead link)

   <b>_possibly_genscan</b>
       From bioperl test data.

   <b>_possibly_gff</b>
       From bioperl test data.

   <b>_possibly_hmmer</b>
       From bioperl test data.

   <b>_possibly_nexus</b>
       From "<a href="http://paup.csit.fsu.edu/nfiles.html">http://paup.csit.fsu.edu/nfiles.html</a>".

   <b>_possibly_mase</b>
       From  bioperl  test data.  More detail from "<a href="http://www.umdnj.edu/rcompweb/PA/Notes/GenbankFF.htm">http://www.umdnj.edu/rcompweb/PA/Notes/GenbankFF.htm</a>" (TODO:
       dead link)

   <b>_possibly_mega</b>
       From the ensembl broswer (AlignView data export).

   <b>_possibly_msf</b>
       From "<a href="http://www.ebi.ac.uk/help/formats.html">http://www.ebi.ac.uk/help/formats.html</a>".

   <b>_possibly_phrap</b>
       From      "<a href="http://biodata.ccgb.umn.edu/docs/contigimage.html">http://biodata.ccgb.umn.edu/docs/contigimage.html</a>".      (TODO:      dead      link)      From
       "<a href="http://genetics.gene.cwru.edu/gene508/Lec6.htm">http://genetics.gene.cwru.edu/gene508/Lec6.htm</a>".     (TODO: dead link) From bioperl test data ("*.ace.1"
       files).

   <b>_possibly_pir</b>
       From "<a href="http://www.ebi.ac.uk/help/formats.html">http://www.ebi.ac.uk/help/formats.html</a>".  The ".,()" spotted in bioperl test data.

   <b>_possibly_pfam</b>
       From bioperl test data.

   <b>_possibly_phylip</b>
       From "<a href="http://www.ebi.ac.uk/help/formats.html">http://www.ebi.ac.uk/help/formats.html</a>".  Initial space allowed on first line (spotted  in  ensembl
       AlignView exported data).

   <b>_possibly_prodom</b>
       From "<a href="http://prodom.prabi.fr/prodom/current/documentation/data.php">http://prodom.prabi.fr/prodom/current/documentation/data.php</a>".

   <b>_possibly_raw</b>
       From "<a href="http://www.ebi.ac.uk/help/formats.html">http://www.ebi.ac.uk/help/formats.html</a>".

   <b>_possibly_rsf</b>
       From "<a href="http://www.ebi.ac.uk/help/formats.html">http://www.ebi.ac.uk/help/formats.html</a>".

   <b>_possibly_selex</b>
       From "<a href="http://www.ebc.ee/WWW/hmmer2-html/node27.html">http://www.ebc.ee/WWW/hmmer2-html/node27.html</a>".

       Assuming  presence  of  Selex  file  header.   Data  exported  by  Bioperl  on Pfam and Selex formats are
       identical, but Pfam file only holds one alignment.

   <b>_possibly_stockholm</b>
       From bioperl test data.

   <b>_possibly_swiss</b>
       From "<a href="http://ca.expasy.org/sprot/userman.html">http://ca.expasy.org/sprot/userman.html</a>#entrystruc".

   <b>_possibly_tab</b>
       Contributed by Heikki.

   <b>_possibly_vcf</b>
       From "<a href="http://www.1000genomes.org/wiki/analysis/vcf4.0">http://www.1000genomes.org/wiki/analysis/vcf4.0</a>".

       Assumptions made about sanity - format and date lines  are  line  1  and  2  respectively.  This  is  not
       specified in the format document.

perl v5.32.1                                       2021-08-15                    <u>Bio::Tools::<a href="../man3pm/GuessSeqFormat.3pm.html">GuessSeqFormat</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>