<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>tandem - X!Tandem mass spectrometry software for peptide sequencing</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/tandem-mass">tandem-mass_201702011-1build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       tandem - X!Tandem mass spectrometry software for peptide sequencing

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>tandem</b> &lt;input.xml&gt; &lt;output.xml&gt;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This  manual  page  documents  briefly  the <b>tandem-mass</b> package that brings a mass spectrometric software
       software for peptide sequencing and protein identification.

       This software has a very simple, unsophisticated application programming interface (API): it simply takes
       an XML file of instructions on its command line, and outputs the results into an XML file, which has been
       specified   in   the    input    XML    file.    The    output    file    format    is    described    at
       <u>`<a href="http://www.thegpm.org/docs/X_series_output_form.pdf">http://www.thegpm.org/docs/X_series_output_form.pdf</a>'</u>.

       Unlike  some earlier generation search engines, all of the X! Series search engines calculate statistical
       confidence (expectation values) for all of the individual  spectrum-to-sequence  assignments.  They  also
       reassemble  all  of the peptide assignments in a data set onto the known protein sequences and assign the
       statistical confidence that this assembly and alignment is non-random. The formula for which can be found
       here.  Therefore,  separate  assembly  and  statistical  analysis  software,  e.g.   PeptideProphet   and
       ProteinProphet, do not need to be used.

</pre><h4><b>BIBLIOGRAPHICAL</b> <b>REFERENCE</b> <b>TO</b> <b>BE</b> <b>CITED</b></h4><pre>
       Robertson Craig and Ronald C. Beavis. (2004) TANDEM: matching proteins with mass spectra. <u>Bioinformatics</u>,
       20, 1466-1467.

       See more references pertaining to this software at `<a href="http://www.thegpm.org/GPM/references.html">http://www.thegpm.org/GPM/references.html</a>'.

</pre><h4><b>AUTHOR</b></h4><pre>
       This  manual  page  was  written  by Filippo Rusconi &lt;<a href="mailto:lopippo@debian.org">lopippo@debian.org</a>&gt;. Permission is granted to copy,
       distribute and/or modify this document under the terms of the GNU  General  Public  License,  Version  3,
       published by the Free Software Foundation.

        On  a  Debian  system  the complete text of the GNU General Public License version 3 can be found in the
       file `<u><a href="file:/usr/share/common-licenses/GPL-3">/usr/share/common-licenses/GPL-3</a></u>'.

                                                  November 2012                                        <u><a href="../man1/TANDEM.1.html">TANDEM</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>