<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>fastBlockSearch - search loci matching protein block profiles</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/augustus">augustus_3.5.0+dfsg-4build7_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       fastBlockSearch - search loci matching protein block profiles

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>fastBlockSearch</b> [options] seqs.fa fam.prfl

</pre><h4><b>DESCRIPTION</b></h4><pre>
           Searches hits (matches) of the blocks in the profile given by fam.prfl within the genomic sequences in the file seqs.fa.
           Hits are sorted increasingly by score, so the last displayed hit is the best one found in the region. The format is similar
           to that of the blockscore file (which is optionally generated by msa2prfl.pl): It shows coordinate, strand,
           mean odds-ratio score, and specificity of score, and the motif.
           From the output users can chose regions with matching blocks to perform gene prediction with AUGUSTUS-PPX using the same block profile.

</pre><h4><b>EXAMPLE</b></h4><pre>
           &gt; fastBlockSearch --cutoff=1.1 chr4.103M.fa PF00225_seed.prfl

           Hits found in chr4 103000000 105000000
           Score:207.987
           Mult. score:4.83391
           1081586 unknown_M[5,13] -       2.32574 5.04633 .....YATRLKNI
           1103952 unknown_L       -       4.85363 6.75245 NAKTRIICTITP
           1103991 unknown_K       -       8.38065 9.92928 YRDSKLTRILQNSLG
           1104375 unknown_J       -       3.96065 6.79408 RSLFILGQVIKKL
           1106992 unknown_I       -       9.22487 7.64306 LVDLAGSE
           1115567 unknown_H[5,16] -       2.31869 5.58986 .....ESRHYGETKMN
           1116319 unknown_G       -       7.34282 8.29425 EIYNETITDLL
           1117092 unknown_F       -       5.10694 6.10274 VIPRAIHDIF
           1117146 unknown_E       -       9.43596 9.18891 QTASGKTYTM
           1117176 unknown_D[1,8]  -       5.73796 6.31532 .GTIFAYG
           1117399 unknown_B[1,7]  -       3.59083 5.03059 .CLDRVF
           1119420 unknown_A[0,8]  -       4.64107 6.44285 RVRPLNSR.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>--cutoff=c</b>
           This minimum for the average log score of the motifs found can be used to adjust the sensitivity of
           the block search. The standard cutoff is 0.7, which is very sensitive but can give many false
           positive hits for smaller profiles.

</pre><h4><b>AUTHORS</b></h4><pre>
       Oliver Keller

                                                                                              <u><a href="../man1/FASTBLOCKSEARCH.1.html">FASTBLOCKSEARCH</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>