<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>hmmalign - align sequences to a profile</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/hmmer">hmmer_3.4+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       hmmalign - align sequences to a profile

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>hmmalign</b> [<u>options</u>] <u>hmmfile</u> <u>seqfile</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Perform  a  multiple  sequence alignment of all the sequences in <u>seqfile</u> by aligning them individually to
       the profile HMM in <u>hmmfile.</u>  The new alignment is output to stdout.

       The <u>hmmfile</u> should contain only a single profile. If it contains more, only the first profile in the file
       will be used.

       Either <u>hmmfile</u> or <u>seqfile</u> (but not both) may be '-' (dash), which means reading  this  input  from  stdin
       rather than a file.

       The  sequences  in  <u>seqfile</u> are aligned in unihit local alignment mode.  Therefore they should already be
       known to contain only a single domain (or a fragment of one).  The  optimal  alignment  may  assign  some
       residues  as  nonhomologous  (N  and  C  states),  in which case these residues are still included in the
       resulting alignment, but shoved to the outer edges. To trim these unaligned nonhomologous  residues  from
       the result, see the <b>--trim</b> option.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>     Help; print a brief reminder of command line usage and all available options.

       <b>-o</b> <u>&lt;f&gt;</u> Direct the output alignment to file <u>&lt;f&gt;,</u> rather than to stdout.

       <b>--mapali</b> <u>&lt;f&gt;</u>
              Merge  the existing alignment in file <u>&lt;f&gt;</u> into the result, where <u>&lt;f&gt;</u> is exactly the same alignment
              that was used to build the model in <u>hmmfile.</u>  This is done using a map  of  alignment  columns  to
              consensus  profile positions that is stored in the <u>hmmfile.</u>  The multiple alignment in <u>&lt;f&gt;</u> will be
              exactly reproduced in its consensus columns  (as  defined  by  the  profile),  but  the  displayed
              alignment  in  insert  columns  may  be  altered,  because  insertions  relative  to a profile are
              considered by convention to be unaligned data.

       <b>--trim</b> Trim nonhomologous residues (assigned to N and C  states  in  the  optimal  alignments)  from  the
              resulting multiple alignment output.

       <b>--amino</b>
              Assert that sequences in <u>seqfile</u> are protein, bypassing alphabet autodetection.

       <b>--dna</b>  Assert that sequences in <u>seqfile</u> are DNA, bypassing alphabet autodetection.

       <b>--rna</b>  Assert that sequences in <u>seqfile</u> are RNA, bypassing alphabet autodetection.

       <b>--informat</b> <u>&lt;s&gt;</u>
              Assert  that  input  <u>seqfile</u> is in format <u>&lt;s&gt;</u>, bypassing format autodetection.  Common choices for
              <u>&lt;s&gt;</u> include:  <b>fasta</b>,  <b>embl</b>,  <b>genbank.</b>   Alignment  formats  also  work;  common  choices  include:
              <b>stockholm</b>, <b>a2m</b>, <b>afa</b>, <b>psiblast</b>, <b>clustal</b>, <b>phylip</b>.  For more information, and for codes for some less
              common  formats,  see main documentation.  The string <u>&lt;s&gt;</u> is case-insensitive (<b>fasta</b> or <b>FASTA</b> both
              work).

       <b>--outformat</b> <u>&lt;s&gt;</u>
              Write the output alignment in format <u>&lt;s&gt;</u>.  Common choices for <u>&lt;s&gt;</u> include:  <b>stockholm</b>,  <b>a2m</b>,  <b>afa</b>,
              <b>psiblast</b>, <b>clustal</b>, <b>phylip</b>.  The string <u>&lt;s&gt;</u> is case-insensitive (<b>a2m</b> or <b>A2M</b> both work).  Default is
              <b>stockholm</b>.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       See  <b><a href="../man1/hmmer.1.html">hmmer</a></b>(1) for a master man page with a list of all the individual man pages for programs in the HMMER
       package.

       For complete documentation, see the user guide that came with your HMMER distribution (Userguide.pdf); or
       see the HMMER web page (<a href="http://hmmer.org/">http://hmmer.org/</a>).

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright (C) 2023 Howard Hughes Medical Institute.
       Freely distributed under the BSD open source license.

       For additional information on copyright and licensing, see the file called COPYRIGHT in your HMMER source
       distribution, or see the HMMER web page (<a href="http://hmmer.org/">http://hmmer.org/</a>).

</pre><h4><b>AUTHOR</b></h4><pre>
       <a href="http://eddylab.org">http://eddylab.org</a>

HMMER 3.4                                           Aug 2023                                         <u><a href="../man1/hmmalign.1.html">hmmalign</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>