<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>HMMER - profile HMMs for biological sequence analysis</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/hmmer">hmmer_3.4+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       HMMER - profile HMMs for biological sequence analysis

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>hmmalign</b>
         Align sequences to a profile

       <b>hmmbuild</b>
         Construct profiles from multiple sequence alignments

       <b>hmmconvert</b>
         Convert profile file to various formats

       <b>hmmemit</b>
         Sample sequences from a profile

       <b>hmmfetch</b>
         Retrieve profiles from a file

       <b>hmmlogo</b>
         Produce a conservation logo graphic from a profile

       <b>hmmpgmd</b>
         Daemon for database search web services

       <b>hmmpress</b>
         Prepare a profile database for hmmscan

       <b>hmmscan</b>
         Search sequence(s) against a profile database

       <b>hmmsearch</b>
         Search profile(s) against a sequence database

       <b>hmmsim</b>
         Collect profile score distributions on random sequences

       <b>hmmstat</b>
         Summary statistics for a profile file

       <b>jackhmmer</b>
         Iteratively search sequence(s) against a sequence database

       <b>makehmmerdb</b>
         build nhmmer database from a sequence file

       <b>nhmmer</b>
         Search DNA/RNA queries against a DNA/RNA sequence database

       <b>nhmmscan</b>
         Search DNA/RNA sequence(s) against a DNA/RNA profile database

       <b>phmmer</b>
         Search protein sequence(s) against a protein sequence database

       <b>alimask</b>
         Calculate and add column mask to a multiple sequence alignment

</pre><h4><b>DESCRIPTION</b></h4><pre>
       HMMER  is  a suite of several programs for biological sequence alignment and database homology search. It
       uses probabilistic models called "profile hidden Markov models" (profile HMMs) to  represent  the  likely
       evolutionary homologs of a single sequence or a multiple alignment of a sequence family. A main avenue of
       research is to improve the evolutionary predictive models in HMMER to be able to recognize and accurately
       align increasingly remote homologs, distant in time.

       HMMER  is  also  used  as an organizational tool, to group the exponentially growing number of biological
       sequences into a vastly smaller set of well-annotated sequence families. New sequences can  be  annotated
       by  comparison  against  curated  sequence  family  databases  of prebuilt HMMER profiles, in addition or
       instead of comparison to the entire sequence database. Databases such as Pfam, SMART, and TIGRfams, among
       others, are based on this principle.

       HMMER is used in three main modes: to search a sequence database for new homologs  of  a  sequence  or  a
       sequence  family;  to  search  a  profile database (like Pfam) to find what known family a query sequence
       belongs to, or what domains it has; and to automatically construct large multiple alignments  (i.e.  with
       an effectively unlimited number of sequences) using a profile representative of a sequence family.

       Suppose you have a multiple sequence alignment of a sequence family of interest, and you want to search a
       sequence  database  for  additional  homologs.  The  <b>hmmbuild</b>  program  builds  profile(s)  from multiple
       alignment(s).  The <b>hmmsearch</b> program searches protein profile(s) against a protein sequence database, and
       <b>nhmmer</b> searches nucleotide profile(s) against a nucleotide sequence database.

       Suppose you have a single sequence of interest, and you want to search a sequence database for additional
       homologs. The <b>phmmer</b> program searches a single protein sequence against a protein sequence database.  The
       <b>jackhmmer</b>  program  does  the  same  thing  but  iteratively -- homologs detected in a previous round are
       incorporated into a new profile, and the new profile is searched again.  <b>phmmer</b> is used like BLASTP,  and
       <b>jackhmmer</b>  is  used  like  a  protein PSI-BLAST. The <b>nhmmer</b> program searches a single nucleotide sequence
       against a nucleotide sequence.

       Suppose you have sequence(s) that you want to analyze using a HMMER-based profile HMM database like  Pfam
       (<a href="http://pfam.sanger.ac.uk">http://pfam.sanger.ac.uk</a>).   The  <b>hmmpress</b>  program formats a profile HMM flatfile (such as the file you
       would download from Pfam) into a HMMER binary database.  The <b>hmmscan</b> program searches protein sequence(s)
       against that database.  The <b>nhmmscan</b> program  can  similarly  search  nucleotide  sequence(s)  against  a
       pressed database of nucleotide profiles, such as from Dfam (<a href="http://dfam.janelia.org">http://dfam.janelia.org</a>).

       Suppose  you  want  to  align  lots  of  sequences.  You  can construct a manageably small alignment of a
       representative set of sequences, build a profile with <b>hmmbuild</b>, and use the <b>hmmalign</b> program to align any
       number of sequences to that profile.

       HMMER also includes some auxiliary tools for working with large profile databases.  <b>hmmfetch</b> fetches  one
       or more profiles from a database.  <b>hmmstat</b> prints summary statistics about a profile file.

       For  compatibility  with  other  profile  software and previous versions of HMMER, the <b>hmmconvert</b> program
       converts profiles to a few other formats. We intend to add more support for other formats over time.

       The <b>hmmemit</b> program generates (simulates) "homologous" sequences by sampling from a profile. It can  also
       generate a "consensus" sequence.

       The  <b>hmmsim</b>  program  is  a  simulator used for collecting statistics about score distributions on random
       sequences.

       Each program has its own man page.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       This is a summary man page for the entire HMMER3 package.  See individual  man  pages  [<b><a href="../man1/hmmbuild.1.html">hmmbuild</a></b>(1),  for
       example] for usage, options, and description of each program in the package.

       For complete documentation, see the user guide that came with your HMMER distribution (Userguide.pdf); or
       see the HMMER web page (<a href="http://hmmer.org/">http://hmmer.org/</a>).

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright (C) 2023 Howard Hughes Medical Institute.
       Freely distributed under the BSD open source license.

       For additional information on copyright and licensing, see the file called COPYRIGHT in your HMMER source
       distribution, or see the HMMER web page (<a href="http://hmmer.org/">http://hmmer.org/</a>).

</pre><h4><b>AUTHOR</b></h4><pre>
       <a href="http://eddylab.org">http://eddylab.org</a>

HMMER 3.4                                           Aug 2023                                            <u><a href="../man1/HMMER.1.html">HMMER</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>