<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>HMMER - profile hidden Markov model software</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/hmmer2">hmmer2_2.3.2+dfsg-12_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       HMMER - profile hidden Markov model software

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>hmm2align</b>
              Align multiple sequences to a profile HMM.

       <b>hmm2build</b>
              Build a profile HMM from a given multiple sequence alignment.

       <b>hmm2calibrate</b>
              Determine  appropriate  statistical  significance  parameters  for  a  profile  HMM prior to doing
              database searches.

       <b>hmm2convert</b>
              Convert HMMER profile HMMs to other formats, such as GCG profiles.

       <b>hmm2emit</b>
              Generate sequences probabilistically from a profile HMM.

       <b>hmm2fetch</b>
              Retrieve an HMM from an HMM database

       <b>hmm2index</b>
              Create a binary SSI index for an HMM database

       <b>hmm2pfam</b>
              Search a profile HMM database with a sequence (i.e., annotate various  kinds  of  domains  in  the
              query sequence).

       <b>hmm2search</b>
              Search  a  sequence  database  with  a  profile HMM (i.e., find additional homologues of a modeled
              family).

</pre><h4><b>DESCRIPTION</b></h4><pre>
       These programs use profile hidden Markov models (profile HMMs) to model the primary  structure  consensus
       of a family of protein or nucleic acid sequences.

</pre><h4><b>OPTIONS</b></h4><pre>
       All  <b>HMMER</b>  programs give a brief summary of their command-line syntax and options if invoked without any
       arguments.  When invoked with the single argument, <b>-h</b> (i.e., help), a program will  report  more  verbose
       command-line  usage information, including rarely used, experimental, and expert options.  <b>-h</b> will report
       version numbers which are useful if you need to report a bug or problem to me.

       Each <b>HMMER</b> program has its own man page briefly summarizing command line usage.  There is also  a  user's
       guide  that came with the software distribution, which includes a tutorial introduction and more detailed
       descriptions of the programs.

       See <a href="http://hmmer.janelia.org/">http://hmmer.janelia.org/</a> for on-line documentation and the current HMMER release.

       In general, no command line options should be needed by beginning users.  The defaults  are  set  up  for
       optimum  performance  in  most  situations.   Options  that  are single lowercase letters (e.g.  <b>-a</b> ) are
       "common" options that are expected to be frequently used and will  be  important  in  many  applications.
       Options  that  are  single uppercase letters (e.g.  <b>-B</b> ) are usually less common options, but also may be
       important in some applications.  Options that are full words (e.g.  <b>--verbose</b> ) are either  rarely  used,
       experimental, or expert options.  Some experimental options are only there for my own ongoing experiments
       with HMMER, and may not be supported or documented adequately.

</pre><h4><b>SEQUENCE</b> <b>FILE</b> <b>FORMATS</b></h4><pre>
       In  general,  <b>HMMER</b>  attempts  to  read most common biological sequence file formats.  It autodetects the
       format of the file. It also autodetects whether the sequences are  protein  or  nucleic  acid.   Standard
       IUPAC degeneracy codes are allowed in addition to the usual 4-letter or 20-letter codes.

       <b>Unaligned</b> <b>sequences</b>
              Unaligned sequence files may be in FASTA, Swissprot, EMBL, GenBank, PIR, Intelligenetics, Strider,
              or GCG format.  These formats are documented in the User's Guide.

       <b>Sequence</b> <b>alignments</b>
              Multiple  sequence  alignments  may  be  in  CLUSTALW, SELEX, or GCG MSF format. These formats are
              documented in the User's Guide.

</pre><h4><b>ENVIRONMENT</b> <b>VARIABLES</b></h4><pre>
       For ease of using large stable sequence and HMM databases, <b>HMMER</b> looks for sequence files and  HMM  files
       in the current working directory as well as in system directories specified by environment variables.

       <b>BLASTDB</b>
              Specifies   the  directory  location  of  sequence  databases.  Example:  <b>/seqlibs/blast-db/.</b>   In
              installations that use BLAST software, this environment variable is likely to already be set.

       <b>HMMERDB</b>
              Specifies the directory location of HMM databases. Example: <b>/seqlibs/pfam/.</b>

HMMER 2.3.2                                         Oct 2003                                           <u><a href="../man1/hmmer2.1.html">hmmer2</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>