<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>pfscale - fit parameters of an extreme-value distribution to a profile score list</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/pftools">pftools_3.2.12-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       pfscale - fit parameters of an extreme-value distribution to a profile score list

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>pfscale</b>   [  <b>-hl</b>  ] [ <b>-L</b> <u>log_base</u> ] [ <b>-M</b> <u>mode_nb</u> ] [ <b>-N</b> <u>db_size</u> ] [ <b>-P</b> <u>upper_limit</u> ] [ <b>-Q</b> <u>lower_limit</u> ] [
                 <u>score_list</u> | <b>-</b> ] [ <u>profile</u> ] [ <u>parameters</u> ]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>pfscale</b> fits the two parameters of an extreme-value distribution to a sorted score distribution  obtained
       by searching a sequence database with a profile.  The file '<u>score_list</u>' is a sorted list of profile match
       scores generated by <b>pfsearch</b>.  If '<b>-</b>' is specified instead of a filename, the score list is read from the
       standard input. The result is written to the standard output.

       If  the original profile is given as the second argument, the normalization function with the lowest mode
       number or the lowest priority number specified within the profile will be  updated  such  as  to  produce
       -Log10  per-residue  E-values.   If  the second argument is omitted, the output consists of a header line
       containing the normalization parameters followed by a modified score list, showing <u>score</u>  <u>rank</u>,  <u>original</u>
       <u>raw</u> <u>scores</u>, <u>log-cumulative</u> <u>frequencies</u> and corresponding <u>normalized</u> <u>scores</u> next to each other.

       Note  that  this  program  implements  the  significance  estimation  procedure  for profile match scores
       described in Hofmann &amp; Bucher (1995).  It  has  been  used  for  the  calculation  of  the  normalization
       parameters of all profiles in the PROSITE database.

</pre><h4><b>OPTIONS</b></h4><pre>
       <u>score_list</u>
              Input score list.
              The file must contain a sorted list of scores. The first field of each line is considered as being
              a  score, all other fields on the same line are ignored.  The different fields of each line should
              be delimited by whitespaces.  If the filename is replaced by a '<b>-</b>', <b>pfscale</b> will  read  the  score
              list from <b>stdin</b>.

       <u>profile</u>
              Optional profile file.
              If  a filename is specified, the profile will be parsed and either the lowest priority mode or the
              mode number specified with option <b>-M</b> will be scaled. All cut-off levels which  use  the  specified
              mode number will also be updated.

       <b>-h</b>     Display usage help text.

       <b>-l</b>     Remove output line length limit. Individual lines of the output profile can exceed a length of 132
              characters, removing the need to wrap them over several lines.

       <b>-L</b> <u>log_base</u>
              Logarithmic  base  of  the parameters of the estimated extreme-value distribution.  The parameters
              reported by <b>pfscale</b> are expressed as logarithms and thus can be inserted directly  into  a  linear
              normalization function defined in a generalized profile.
              Default: 10

       <b>-M</b> <u>mode_nb</u>
              Mode number to scale.
              Defines which mode number (and implicitly which cut-off level) of the input PROSITE profile should
              be  scaled.  This  overrides the default behaviour of scaling only the normalization mode with the
              lowest priority (or lowest mode number).  All cut-off levels defined in the profile as using  this
              mode number (via the <u>MODE</u> keyword) will be updated as well.

       <b>-N</b> <u>db_size</u>
              Size  of  the  database  from  which  the  input score list was derived.  The searched database is
              typically a shuffled version of a real protein or nucleotide sequence database.
              Default: 14147368 (size of SWISS-PROT release 30 and shuffled derivatives of it).

       <b>-P</b> <u>upper_limit</u>
              Upper threshold of the probability range to which the extreme-value distribution will  be  fitted.
              For instance: if <u>N</u>=10'000'000 and <u>P</u>=0.0001 then profile match scores below rank 1000 in the sorted
              input list (corresponding to occurrence probabilities &gt; 0.0001) will be ignored.
              Default: 0.0001

       <b>-Q</b> <u>lower_limit</u>
              Lower  threshold  of the probability range to which the extreme-value distribution will be fitted.
              For instance: if <u>N</u>=10'000'000 and <u>Q</u>=0.000001 then profile match scores above rank 10 in the sorted
              input list (corresponding to occurrence probabilities &lt; 0.000001) will be ignored.
              Default: 0.000001

</pre><h4><b>PARAMETERS</b></h4><pre>
       Note:  for backwards compatibility, release 2.3 of the <b>pftools</b> package will parse the version  2.2  style
              parameters,  but  these are <u>deprecated</u> and the corresponding option (refer to the <u>options</u> section)
              should be used instead.

       L=#    Logarithmic base.
              Use option <b>-L</b> instead.

       M=#    Mode number.
              Use option <b>-M</b> instead.

       N=#    Database size.
              Use option <b>-N</b> instead.

       P=#    Upper probability threshold.
              Use option <b>-P</b> instead.

       Q=#    Lower probability threshold.
              Use option <b>-Q</b> instead.

</pre><h4><b>EXAMPLES</b></h4><pre>
       (1)    <b>pfsearch</b> -fr -C 200 sh3.prf shuffle20.seq | <b>sort</b> -nr | <b>pfscale</b> -P 0.0001 -Q 0.000001 -

              derives score-normalization parameters for the SH3 domain profile in  file  '<b>sh3.prf</b>'.   The  file
              '<b>shuffle20.seq</b>'  contains  a  window-shuffled derivative of SWISS-PROT release 30 in Pearson/Fasta
              format (window-size 20).  Note that the implicit default of <u>N</u> corresponds  to  the  size  of  this
              database  and  thus  needs not to be specified on the command line.  The cut-off value 200 for the
              <b><a href="../man1/pfsearch.1.html">pfsearch</a></b>(1) option <b>-C</b> will produce about 2000 matches completely covering the range defined by the
              command line parameters <b>-P</b> and <b>-Q</b> of <b>pfscale</b>.  A suitable cut-off  value  has  to  be  guessed  in
              advance by computing a few optimal alignment scores for random sequences.

</pre><h4><b>EXIT</b> <b>CODE</b></h4><pre>
       On  successful  completion  of  its  task,  <b>pfscale</b>  will return an exit code of 0. If an error occurs, a
       diagnostic message will be output on standard error and the exit code will  be  different  from  0.  When
       conflicting  options  where  passed to the program but the task could nevertheless be completed, warnings
       will be issued on standard error.

</pre><h4><b>NOTES</b></h4><pre>
       (1)    The current version of <b>pfscale</b> does  not  yet  support  the  <b><a href="../man5/xpsa.5.html">xpsa</a></b>(5)  output  format  produced  by
              <b><a href="../man1/pfscan.1.html">pfscan</a></b>(1)  or <b><a href="../man1/pfsearch.1.html">pfsearch</a></b>(1).  The score list should therefore be generated without the <b><a href="../man1/pfscan.1.html">pfscan</a></b>(1) and
              <b><a href="../man1/pfsearch.1.html">pfsearch</a></b>(1) option <b>-k</b>.

</pre><h4><b>REFERENCES</b></h4><pre>
       Hofmann K &amp; Bucher P. (1995).  <u>The</u> <u>FHA-domain:</u> <u>a</u> <u>nuclear</u> <u>signalling</u> <u>domain</u> <u>found</u> <u>in</u> <u>protein</u>  <u>kinases</u>  <u>and</u>
       <u>transcription</u> <u>factors.</u>  Trends Biochem. Sci.  <b>20</b>:47-349.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/pfsearch.1.html">pfsearch</a></b>(1), <b><a href="../man1/pfscan.1.html">pfscan</a></b>(1), <b><a href="../man5/xpsa.5.html">xpsa</a></b>(5)

</pre><h4><b>AUTHOR</b></h4><pre>
       The <b>pftools</b> package was developed by Philipp Bucher.
       Any comments or suggestions should be addressed to &lt;<a href="mailto:pftools@sib.swiss">pftools@sib.swiss</a>&gt;.

pftools 2.3                                        August 2003                                        <u><a href="../man1/PFSCALE.1.html">PFSCALE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>