<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>pfsearch - search a protein or DNA sequence library for sequence segments matching a profile</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/pftools">pftools_3.2.12-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       pfsearch - search a protein or DNA sequence library for sequence segments matching a profile

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>pfsearch</b>  [  <b>-abdfhlLmruksvxyz</b>  ]  [  <b>-C</b>  <u>cut_off</u>  ]  [  <b>-M</b>  <u>mode_nb</u>  ]  [  <b>-W</b>  <u>width</u> ] [ <u>profile</u> | <b>-</b> ] [
                 <u>sequence_library</u> | <b>-</b> ] [ <u>parameters</u> ]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>pfsearch</b> compares a query profile against a DNA or protein sequence library.  The result is  an  unsorted
       list  of profile-sequence matches written to the standard output.  A variety of output formats containing
       different information can be specified via the  options  <b>-l</b>, <b>-L</b>, <b>-r</b>, <b>-k</b>, <b>-s</b>, <b>-x</b>, <b>-y</b>  and  <b>-z</b>.   The  file
       '<u>profile</u>'  contains  a profile in PROSITE format. The '<u>sequence_library</u>' file contains a sequence library
       in EMBL/SWISS-PROT format (assumed by default) or in Pearson/Fasta format (indicated by option  <b>-f</b>).   If
       '<b>-</b>' is specified instead of one of the filenames, the corresponding data is read from the standard input.

</pre><h4><b>OPTIONS</b></h4><pre>
       <u>profile</u>
              Input query profile.
              The  PROSITE profile contained in this file will be used to search for profile to sequence matches
              in a biological sequence library. If the filename is replaced by a '<b>-</b>',  <b>pfsearch</b>  will  read  the
              profile from <b>stdin</b>.

       <u>sequence_library</u>
              Library of DNA or protein sequences.
              This  file  should  contain  one or several EMBL/SWISS-PROT (default) or Pearson/Fasta (option <b>-f</b>)
              formatted DNA or protein sequences. The program <b>pfsearch</b> tries to  identify  matches  between  the
              input  profile and all individual sequences of this library. If the filename is replaced by a '<b>-</b>',
              <b>pfsearch</b> will read the sequence library from <b>stdin</b>.

       <b>-a</b>     Report optimal alignment scores for all sequences regardless of the cut-off  value.   This  option
              simultaneously forces <u>DISJOINT</u>=<u>UNIQUE</u>.

       <b>-b</b>     Search the complementary strands of DNA sequences as well.

       <b>-f</b>     Input sequence library is in Pearson/Fasta format.

       <b>-h</b>     Display usage help text.

       <b>-l</b>     Indicate the value of the highest cut-off level exceeded by the match score in the output list.

       <b>-L</b>     Indicate  by  character string the highest cut-off level exceeded by the match score in the output
              list.

              Note:  The generalized profile format includes a text string field to specify a name for a cut-off
                     level. The <b>-L</b> option causes the program to display the first two characters  of  this  text
                     string  (usually  something  like  '<u>!</u>', '<u>?</u>', '<u>??</u>',  etc.)  at  the  beginning of each match
                     description.

       <b>-m</b>     Report individual matches for circular profiles.
              If the profile is circular, each match between a sequence and a  profile  can  be  composed  of  a
              stretch  of individual matches of the profile. By default, <b>pfsearch</b> reports only the total matched
              region. When this option is set, detailed information for each individual match will be output  as
              well.

              Note:  The  scoring  system  for  most circular profiles has been optimized to find total matches,
                     therefore the normalized scores of individual matches of a circular profile to  a  sequence
                     should be considered with caution.

       <b>-r</b>     Use  raw  scores  rather  than  normalized scores for match selection. The normalized score is not
              printed, this option is useful to create raw score lists to be used with <b><a href="../man1/pfscale.1.html">pfscale</a></b>(1).

       <b>-u</b>     Forces <u>DISJOINT</u>=<u>UNIQUE</u>.

       <b>-C</b> <u>cut_off</u>
              Cut-off value to be used for match selection.
              The value of '<u>cut_off</u>' overwrites the level zero cut-off value specified in the profile.
              An integer argument is interpreted as a raw score value, a decimal argument as a normalized  score
              value.
              Default: profile level 0 cut-off value (normalized score if present)

              Note:  Compared to release 2.2, an integer value does <u>not</u> force option <b>-r</b>.  Normalized scores will
                     still  be  listed in output if an integer cut-off is specified on the command line but cut-
                     off level computation will be based on raw score.

       <b>-M</b> <u>mode_nb</u>
              Normalization mode to use for score computation.
              The '<u>mode_nb</u>' specifies which normalization mode defined in the profile should be used to  compute
              the  normalized  scores  for  profile to sequence matches. This option will override the profile's
              <u>PRIORITY</u> parameter.
              If the specified normalization mode does not exist in the profile, an error message will be output
              to standard error and the search is interrupted.
              Type: integer
              Default: lowest priority mode defined in the profile

   <b>Output</b> <b>modifiers</b>
       <b>-d</b>     Limit sequence description length.
              If this option is set, the description of the sequence on the  header  line  will  be  limited  in
              length.  If  the  match information is longer than the output width specified using option <b>-W</b>, the
              sequence description will not be printed. Else the description will be truncated  to  fit  the  <b>-W</b>
              value.
              By  default, the sequence description is not truncated. This option can not be used when option <b>-k</b>
              is set.

       <b>-k</b>     Use <b><a href="../man5/xpsa.5.html">xpsa</a></b>(5) headers for output.
              When this option is set, all output types (<u>see</u> below) will use an <b><a href="../man5/xpsa.5.html">xpsa</a></b>(5) style header line.  This
              format  uses  <u>keyword</u>=<u>value</u>  pairs  to  output  alignment  parameters.  It  is  useful to transfer
              information between different sequence alignment tools.

       <b>-s</b>     List the sequences of the matched regions as well.  The output will be  a  Pearson/Fasta-formatted
              sequence library.

       <b>-v</b>     Suppress  sequence/profile  parsing  warnings.   If this option is set no warning messages will be
              printed on <u>stderr</u>.  Only fatal errors will be reported. This option should be used with caution.

       <b>-x</b>     List profile-sequence alignments in <b><a href="../man5/psa.5.html">psa</a></b>(5) format. Please refer to the corresponding man page  for
              more information.

       <b>-y</b>     Display  alignments  between  the  profile  and  the  matched sequence regions in a human-friendly
              pairwise alignment format.

       <b>-z</b>     Indicate starting and ending position of the matched profile range. The latter  position  will  be
              given as a negative offset from the end of the profile. Thus the range [    1,    -1] means entire
              profile.

       <b>-W</b> <u>width</u>
              Set alignment output width.
              The  value  of  '<u>width</u>'  specifies  how  many  residues will be output on one line when any of the
              <b>-s</b>, <b>-x</b> or <b>-y</b> options is set.
              Type: integer
              Default: 60

</pre><h4><b>PARAMETERS</b></h4><pre>
       Note:  for backwards compatibility, release 2.3 of the <b>pftools</b> package will parse the version  2.2  style
              parameters,  but  these are <u>deprecated</u> and the corresponding option (refer to the <u>options</u> section)
              should be used instead.

       C=#    Cut-off value.
              Use option <b>-C</b> instead.

       W=#    Output width.
              Use option <b>-W</b> instead.

</pre><h4><b>EXAMPLES</b></h4><pre>
       (1)    <b>pfsearch</b> -f -C 6.0 sh3.prf sh3.seq

              Searches the Pearson/Fasta-formatted protein sequence library '<u>sh3.seq</u>' for  SH3  domains  with  a
              cut-off value of 6.0 normalized score units.  The file '<u>sh3.seq</u>' contains 20 SH3 domain-containing
              protein  sequences  from  SWISS-PROT  release  32.   The file '<u>sh3.prf</u>' contains the PROSITE entry
              SH3/PS50002.

       (2)    <b>pfsearch</b> -bx ecp.prf CVPBR322 | <b>psa2msa</b> -du | <b>readseq</b> -p -fMSF &gt; ecp.msf

              Generates a multiple sequence alignment  of potential <u>E.</u> <u>coli</u> promoters on both strands of plasmid
              pBR322.  The file '<u>ecp.prf</u>' contains a profile for <u>E.</u> <u>coli</u> promoters.  The file '<u>CVPBR322</u>'
               contains EMBL entry J01749|CVPBR322.  The result file '<u>ecp.msf</u>' can further be processed  by  GCG
              programs accepting MSF files as input.

</pre><h4><b>EXIT</b> <b>CODE</b></h4><pre>
       On  successful  completion  of  its  task,  <b>pfsearch</b> will return an exit code of 0. If an error occurs, a
       diagnostic message will be output on standard error and the exit code will  be  different  from  0.  When
       conflicting  options  where  passed to the program but the task could nevertheless be completed, warnings
       will be issued on standard error.

</pre><h4><b>BUGS</b></h4><pre>
       The use of normalized scores as cut-off values can lead to a different behaviour  when  compared  to  raw
       scores.  This  is due to the inherent rounding inaccuracy of real numbers.  The normalized scores used as
       cut-offs should be rounded down in order to circumvent this problem.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/pfscan.1.html">pfscan</a></b>(1), <b><a href="../man1/pfmake.1.html">pfmake</a></b>(1), <b><a href="../man1/psa2msa.1.html">psa2msa</a></b>(1), <b><a href="../man5/psa.5.html">psa</a></b>(5), <b><a href="../man5/xpsa.5.html">xpsa</a></b>(5)

</pre><h4><b>AUTHOR</b></h4><pre>
       The <b>pftools</b> package was developed by Philipp Bucher.
       Any comments or suggestions should be addressed to &lt;<a href="mailto:pftools@sib.swiss">pftools@sib.swiss</a>&gt;.

pftools 2.3                                         July 2003                                        <u><a href="../man1/PFSEARCH.1.html">PFSEARCH</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>