<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>pfsearchV2 - search a protein or DNA sequence library for sequence segments matching a profile</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/pftools">pftools_3.2.12-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       pfsearchV2 - search a protein or DNA sequence library for sequence segments matching a profile

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>pfsearchV2</b> [ -abflLrsuxyz ] [ profile-file | - ]
                   [ seq-library-file | - ]    [C=#] [W=#]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>pfsearchV2</b> compares a query profile against a DNA or protein sequence library.  The result is an unsorted
       list  of profile-sequence matches written to the standard output.  A variety of output formats containing
       different information can be specified via the options <u>-a,</u> <u>-l,</u> <u>-L,</u> <u>-r,</u>  <u>-u,</u>  <u>-s,</u>  <u>-x,</u>  <u>-y,</u>  <u>-z,</u>  and  the
       command-line  parameter  <u>C=#.</u>   <u>profile-file</u>  contains  a  profile  in  PROSITE format.  <u>seq-library-file</u>
       contains a sequence library in EMBL/SWISS-PROT format (assumed by default)  or  in  Pearson/Fasta  format
       (indicated  by  option  <u>-f).</u>  <b>pfsearchV2</b> can be used as a filter if - is used instead of one of the input
       filenames.

</pre><h4><b>OPTIONS</b></h4><pre>
       -a     Report optimal alignment scores for all sequences regardless of the cut-off  value.   This  option
              simultaneously forces DISJOINT=UNIQUE.

       -b     Search the complementary strands of DNA sequences as well.

       -f     Input sequence-library is in Pearson/Fasta format.

       -l     Indicate by number the highest cut-off level exceeded by the match score in the output list.

       -L     Indicate  by  character string the highest cut-off level exceeded by the match score in the output
              list. Note that the generalized profile format includes a text string field to specify a name  for
              a cut-off level. The -L option causes the program to display the first two characters of this text
              string (usually something like "!", "?", "??", etc.) at the beginning of each match description.

       -r     Use  raw  scores  rather than normalized scores for match selection. Normalized scores will not be
              listed in the output.

       -s     List the sequences of the matched regions as well.  The output will be  a  Pearson/Fasta-formatted
              sequence library.

       -u     Forces DISJOINT=UNIQUE.

       -x     List profile-sequence alignments in pftools PSA format.

       -y     Display  alignments  between  the  profile  and  the  matched sequence regions in a human-friendly
              format.

       -z     Indicate starting and ending position of the matched profile range. The latter  position  will  be
              given as a negative offset from the end of the profile. Thus the range [    1,    -1] means entire
              profile.

</pre><h4><b>PARAMETERS</b></h4><pre>
       C=#    Cut-off  value.   Over-writes  the  level zero cut-off value specified in the profile.  An integer
              argument is interpreted as a raw score value, a decimal argument as a normalized score  value.  An
              integer value forces option -r.

       W=#    Output width.  Output lines will be truncated after <u>W</u> characters.  Default: W=132.

</pre><h4><b>EXAMPLES</b></h4><pre>
       (1)    <b>pfsearchV2</b> -f sh3.prf sh3.seq C=6.0

              Searches  the Pearson/Fasta-formatted protein sequence library sh3.seq for SH3 domains with a cut-
              off value of 6.0 normalized score  units.   sh3.seq  contains  20  SH3  domain-containing  protein
              sequences from SWISS-PROT release 32.  sh3.prf contains the PROSITE entry SH3/PS50002.

       (2)    <b>pfsearchV2</b> -bx ecp.prf CVPBR322 | <b>psa2msa</b> -du | <b>readseq</b> -p -fMSF &gt; ecp.msf

              Generates a multiple sequence alignment  of potential <u>E.</u> <u>coli</u> promoters on both strands of plasmid
              pBR322.   ecp.prf  contains  a  profile  for  <u>E.</u>  <u>coli</u>  promoters.   CVPBR322  contains EMBL entry
              J01749|CVPBR322.  The result file <u>ecp.msf</u> can further be processed by GCG programs  accepting  MSF
              files as input.

              <u>See</u> also manual pages of <b>psa2msa.</b>

</pre><h4><b>AUTHOR</b></h4><pre>
       Philipp Bucher
       <a href="mailto:Philipp.Bucher@isrec.unil.ch">Philipp.Bucher@isrec.unil.ch</a>

pftools 2.1                                       February 1998                                    <u><a href="../man1/PFSEARCHV2.1.html">PFSEARCHV2</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>