<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>ptof - convert a protein profile into a frame-search profile</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/pftools">pftools_3.2.12-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       ptof - convert a protein profile into a frame-search profile

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>ptof</b>      [  <b>-hlr</b> ] [ <b>-B</b> <u>init/term_score</u> ] [ <b>-F</b> <u>frameshift</u> ] [ <b>-I</b> <u>insert_multiplier</u> ] [ <b>-X</b> <u>stop_codon</u> ] [
                 <b>-Y</b> <u>intron_open</u> ] [ <b>-Z</b> <u>intron_extend</u> ] [ <u>protein_profile</u> | <b>-</b> ] [ <u>parameters</u> ]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>ptof</b> converts a protein profile (generated  for  instance  by  pftools  programs  <b><a href="../man1/pfmake.1.html">pfmake</a></b>(1),  <b><a href="../man1/gtop.1.html">gtop</a></b>(1)  or
       <b><a href="../man1/htop.1.html">htop</a></b>(1))  into  a  so-called  "<u>frame-search</u> <u>profile</u>".   A  frame-search  profile  is  used  to  search an
       "<u>interleaved</u> <u>frame-translated</u>" DNA sequence (generated by pftools program <b><a href="../man1/2ft.1.html">2ft</a></b>(1)) for  occurrences  of  a
       protein  sequence  motif.   An  "<u>interleaved</u> <u>frame-translated</u>"  DNA  sequence  is  an amino acid sequence
       corresponding to the <u>N-2</u> overlapping codons of a DNAsequence of length <u>N</u>.  Note that in such a  sequence,
       the character '<b>O</b>' is used to represent stop codons.

       The conversion procedure works as follows: The protein profile is expanded in length by a factor of three
       to accommodate three translated codons per original match position.  Two dummy match positions are placed
       between  two  consecutive  significant  match positions imported from the original profile.  The original
       insert  positions  are  placed  between  pairs  of  adjacent  dummy  match  positions.   The  initiation,
       termination,  and  transition  scores  of  the  original  insert positions are left unchanged; the insert
       extension scores are divided by a factor of 3, or by the value of the command-line option  <b>-I</b>.   The  two
       insert  positions  flanking  the  significant match positions serve to accommodate frame-shift errors and
       introns, respectively.  The frame-shift insert position allows free insertion  opening  combined  with  a
       high  insert  extension  penalty  (command-line option <b>-F</b>) whereas the intron insertion position has high
       opening but low extension penalties (command line options <b>-Y</b> and <b>-Z</b>).  The deletion opening  and  closing
       penalties  next to the significant match positions are set to values that ensure that the total cost of a
       single-base deletion is the same as the cost of a single base-insertion at a frame-shift insert position.
       Furthermore, the alphabet of the original profile is extended by the  stop  codon  symbol  '<b>O</b>'  which  is
       assigned  a  constant negative value (command-line option <b>-X</b>) at significant match positions, and zero at
       dummy match positions. At insert positions, it is set to  the  average  of  the  other  insert  extension
       scores.

</pre><h4><b>OPTIONS</b></h4><pre>
       <u>protein_profile</u>
              Input protein profile.
              The  protein profile contained in this file will be converted into a frame-search profile.  If the
              filename is replaced by a '<b>-</b>', <b>ptof</b> will read the profile from <b>stdin</b>.

       <b>-h</b>     Display usage help text.

       <b>-l</b>     Remove output line length limit. Individual lines of the output profile can exceed a length of 132
              characters, removing the need to wrap them over several lines.

       <b>-r</b>     Frame-search parameters are given in normalized score units. This option will only  be  considered
              if  a  linear  normalization  function  with  priority  over  all other normalization functions is
              specified in the profile.  In this case, the frame-search scores specified  on  the  command  line
              will  be  divided  by  the  slope  (<u>R2</u>  parameter)  of the normalization function.  This option is
              particularly useful for profiles which are already scaled in units  that  can  be  interpreted  as
              âˆ’Log(P)-values, <u>e.g.</u>  bits.

       <b>-B</b> <u>init/term_score</u>
              Minimal initiation and termination score.
              All  internal  and  external  initiation  and  termination scores will be set to this value if the
              corresponding value in the original profile is lower than this value.  This parameter is  used  to
              impose  a  more  local  alignment  behavior  on  the  frame-search  profile  in order to deal with
              discontinuities in DNA sequences (long introns, alternative splicing, chimeric clones, <u>etc.</u>)
              Default: -50 (-0.5 with option <b>-r</b>)

       <b>-F</b> <u>frameshift</u>
              Frame-shift error penalty.
              Default: -100 (-1.0 with option <b>-r</b>)

       <b>-I</b> <u>insert_multiplier</u>
              Insert score multiplier.
              The values of the original insert extension scores will be multiplied by this factor in  order  to
              compensate  for the fact that a single amino acid corresponds to three overlapping codon positions
              in the target sequence.
              Default: 1/3

       <b>-X</b> <u>stop_codon</u>
              Stop codon penalty.
              Default: -100 (-1.0 with option <b>-r</b>)

       <b>-Y</b> <u>intron_open</u>
              Intron opening penalty.
              Default: -300 (-3.0 with option <b>-r</b>)

       <b>-Z</b> <u>intron_extend</u>
              Intron extension penalty.
              Default: -1 (-0.01 with option <b>-r</b>)

</pre><h4><b>PARAMETERS</b></h4><pre>
       Note:  for backwards compatibility, release 2.3 of the <b>pftools</b> package will parse the version  2.2  style
              parameters,  but  these are <u>deprecated</u> and the corresponding option (refer to the <u>options</u> section)
              should be used instead.

       B=#    Minimal initiation and termination score.
              Use option <b>-B</b> instead.

       F=#    Frame-shift error penalty.
              Use option <b>-F</b> instead.

       I=#    Insert score multiplier.
              Use option <b>-I</b> instead.

       X=#    Stop codon penalty.
              Use option <b>-X</b> instead.

       Y=#    Intron opening penalty.
              Use option <b>-Y</b> instead.

       Z=#    Intron extension penalty.
              Use option <b>-Z</b> instead.

</pre><h4><b>EXAMPLES</b></h4><pre>
       (1)    <b>ptof</b> -r -F -1.2 -I 0.6 -X -1.5 -B -0.5 sh3.prf &gt; sh3.fsp
              <b>2ft</b> <b>-</b> &lt; R76849.seq | <b>pfsearch</b> -fy -C 5.0 sh3.fsp -

              The protein domain profile in '<u>sh3.prf</u>' is first converted into a frame-search profile  '<u>sh3.fsp</u>'.
              Then  both  strands  of  the Fasta-formatted EST sequence in '<u>R76849.seq</u>' (GenBank/EMBL-accession:
              R76849) are converted into interleaved frame-translated protein sequences  and  searched  for  SH3
              domains with the frame-search profile generated in the preceding step.

              The  output  may  be compared to the result of a more conventional search strategy using a protein
              profile in conjunction with a six-frame translation of the same DNA sequence:

              <b>6ft</b> <b>-</b> &lt; R76849.seq | <b>pfsearch</b> -fy -C 5.0 sh3.prf -

</pre><h4><b>EXIT</b> <b>CODE</b></h4><pre>
       On successful completion of its task, <b>ptof</b> will return  an  exit  code  of  0.  If  an  error  occurs,  a
       diagnostic  message  will  be  output  on standard error and the exit code will be different from 0. When
       conflicting options where passed to the program but the task could nevertheless  be  completed,  warnings
       will be issued on standard error.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/pfscan.1.html">pfscan</a></b>(1), <b><a href="../man1/pfsearch.1.html">pfsearch</a></b>(1), <b><a href="../man1/2ft.1.html">2ft</a></b>(1), <b><a href="../man1/6ft.1.html">6ft</a></b>(1)

</pre><h4><b>AUTHOR</b></h4><pre>
       The <b>pftools</b> package was developed by Philipp Bucher.
       Any comments or suggestions should be addressed to &lt;<a href="mailto:pftools@sib.swiss">pftools@sib.swiss</a>&gt;.

pftools 2.3                                        August 2003                                           <u><a href="../man1/PTOF.1.html">PTOF</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>