<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>train - interpretation of DNA Sanger sequencing data</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/tracetuner">tracetuner_3.0.6~beta+dfsg-3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       train - interpretation of DNA Sanger sequencing data

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>-h</b>     (Help) This message

       <b>-nocall</b>
              Disable base recalling and just use the original called bases read from the input sample file

       <b>-recalln</b>
              Disable adding bases to or deleting from the original called sequence. Only recall Ns

       <b>-recallndb</b>
              Disable  adding  bases  to  or  deleting from the original called sequence. Only recall Ns and dye
              blobs

       <b>-ladder</b>
              Similar to <b>-recalln</b>, but all bases, not only Ns will be recalled from original locations

       <b>-edited_bases</b>
              Start base recalling from the ABI's edited bases

       <b>-het</b>   Call heterozygous bases

       <b>-mix</b>   Call mixed        bases

       <b>-min_ratio</b> &lt;ratio&gt;
              Override the default threshold ratio of heights of the lowest peak to the highest peak at a  given
              position

       <b>-fr</b> &lt;repeat_fraction&gt; Specify the repeat_fraction
              (default is 0.85)

       <b>-fe</b> &lt;max_frac_of_err&gt; Specify the allowable fraction of errors within the
              best  alignment  region.  Default  is  0.1.  If  actual  fraction of errors exceeds this vale, the
              fragment will be rejected (=not used in training process)

       <b>-o</b> &lt;output_file&gt;
              Specify the name of the output file. By default, the output will be made to stdout

       <b>-C</b> &lt;consensusfile&gt;
              Specify the name of the FASTA file which contains the consensus sequence

       <b>-V</b> &lt;vector&gt;
              Specify the name of the FASTA file which contains the vector sequence

       <b>-P</b> &lt;primer&gt;
              Specify the name of the FASTA file which contains the primer sequence

       <b>-S</b> &lt;site&gt;
              Specify the name of the FASTA file which contains the restriction site sequence

       <b>-M</b> &lt;match&gt;
              Specify the match premium     (default is 10)

       <b>-X</b> &lt;mismatch&gt;
              Specify the mismatch penalty  (default is 20)

       <b>-G</b> &lt;gap_penalty &gt;
              Specify the gap initiation or extension penalty (default is 40)

       <b>-d</b> &lt;dir&gt;
              Read the input sample files from specified directory

       <b>-p</b> &lt;projectfile&gt;
              Specify the name of the projectfile which consists of two columns. Each line in this file contains
              the full path to the sample file to be processed and  the  full  path  to  the  FASTA  file  which
              contains the consensus sequence that should be used with this sample file

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

train 3.0.6~beta                                  October 2018                                          <u><a href="../man1/TRAIN.1.html">TRAIN</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>