<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>vcf_filter - Filter a VCF file</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/pyvcf">pyvcf_0.6.8+git20170215.476169c-11build1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       vcf_filter - Filter a VCF file

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>vcf_filter</b>  [-h]  [--no-short-circuit]  [--no-filtered]  [--output  OUTPUT] [--local-script LOCAL_SCRIPT]
       input filter [filter_args] [filter [filter_args]] ...

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This script is part of PyVCF.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>positional</b> <b>arguments:</b>
       input  File to process (use - for STDIN) (default: None)

   <b>optional</b> <b>arguments:</b>
       <b>-h</b>, <b>--help</b>
              Show this help message and exit. (default: False)

       <b>--no-short-circuit</b>
              Do not stop filter processing on a site if any filter is triggered (default: False)

       <b>--output</b> OUTPUT
              Filename   to   output   [STDOUT]   (default:    &lt;_io.TextIOWrapper    name='&lt;stdout&gt;'    mode='w'
              encoding='ANSI_X3.4-1968'&gt;)

       <b>--no-filtered</b>
              Output only sites passing the filters (default: False)

       <b>--local-script</b> LOCAL_SCRIPT
              Python file in current working directory with the filter classes (default: None)

   <b>mgq:</b>
              Filters sites with only low quality variants. It is possible to have a high site quality with many
              low quality calls. This filter demands at least one call be above a threshold quality.

       <b>--genotype-quality</b> GENOTYPE_QUALITY
              Filter sites with no genotypes above this quality (default: 50)

   <b>snp-only:</b>
              Choose only SNP variants

   <b>dps:</b>
              Threshold read depth per sample

       <b>--depth-per-sample</b> DEPTH_PER_SAMPLE
              Minimum required coverage in each sample (default: 5)

   <b>avg-dps:</b>
              Threshold average read depth per sample (read_depth / sample_count)

       <b>--avg-depth-per-sample</b> AVG_DEPTH_PER_SAMPLE
              Minimum required average coverage per sample (default: 3)

   <b>eb:</b>
              Filter  sites  that  look like correlated sequencing errors. Some sequencing technologies, notably
              pyrosequencing, produce mutation hotspots where there is a constant level of noise, producing some
              reference and some heterozygote calls. This filter computes  a  Bayes  Factor  for  each  site  by
              comparing  the  binomial  likelihood of the observed allelic depths under: * A model with constant
              error equal to the MAF. * A model where each sample is the ploidy reported by the caller. The test
              value is the log of the bayes factor. Higher values are more  likely  to  be  errors.  Note:  this
              filter requires rpy2

       <b>--eblr</b> EBLR
              Filter sites above this error log odds ratio (default: <b>-10</b>)

   <b>sq:</b>
              Filter low quailty sites

       <b>--site-quality</b> SITE_QUALITY
              Filter sites below this quality (default: 30)

0.6.7                                             October 2015                                     <u><a href="../man1/VCF_FILTER.1.html">VCF_FILTER</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>