<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>bali-phy - Bayesian Inference of Alignment and Phylogeny</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/bali-phy">bali-phy_3.6.1+dfsg-3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       <b>bali-phy</b> - Bayesian Inference of Alignment and Phylogeny

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>bali-phy</b> <u>sequence-file1</u> [<u>sequence-file2</u> ...]  [<u>OPTIONS</u>]

       <b>bali-phy</b> help <u>topic</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>bali-phy</b> estimates multiple sequence alignments and evolutionary trees from DNA, amino acid, or codon se‐
       quences.  BAli-Phy uses MCMC and Bayesian methods to estimate evolutionary trees, positive selection, and
       branch lengths while averaging over alternative alignments.

       BAli-Phy can also estimate phylogenies from a fixed alignment (like MrBayes and BEAST) using substitution
       models like GTR+gamma.  BAli-Phy automatically estimates relative rates for each gene.

</pre><h4><b>GENERAL</b> <b>OPTIONS</b></h4><pre>
       For  each option below, more information is available by specifying the long form of the option as a help
       topic.  For example: <b>bali-phy</b> <b>help</b> <b>alphabet</b>

       <b>-h</b>, <b>–help</b>, <b>–help</b>=<u>topic</u>
              Display a friendly help message.  Specify <b>–help=advanced</b> or <b>–help=expert</b> to display more  advanced
              options.

       <b>-v</b>, <b>–version</b>
              Print version information.

       <b>-t</b>, <b>–test</b>
              Analyze the initial values and exit.

       <b>-V</b>, <b>–verbose</b>, <b>–verbose</b> <u>NUM</u>
              Print  extra output to aid in trouble-shooting.  If <u>NUM</u> is not specified the default is 1.  Values
              from 2 to 4 increase the amount of information displayed.

       <b>-c</b> <u>filename</u>, <b>–config</b> <u>filename</u>
              Read commands from <u>filename</u> before command line.

</pre><h4><b>MCMC</b> <b>OPTIONS</b></h4><pre>
       <b>-i</b> <u>NUM</u>, <b>–iterations</b> <u>NUM</u>
              The number of iterations to run.

       <b>-n</b> <u>STRING</u>, <b>–name</b> <u>STRING</u>
              Name for the output directory to create.

       <b>-x</b> <u>NUM</u>, <b>–subsample</b> <u>NUM</u>
              Factor by which to subsample.  This option should usually not be used.

       <b>-s</b> <u>NUM</u>, <b>–seed</b> <u>NUM</u>
              Random seed.  Useful for replaying specific runs when trouble-shooting.

</pre><h4><b>PARAMETER</b> <b>OPTIONS</b></h4><pre>
       <b>-T</b> <u>filename</u>, <b>–tree</b> <u>filename</u>
              File with initial tree in Newick format or NEXUS format.

       <b>-U</b>, <b>–unalign</b>
              Unalign all variable-alignment partitions before starting MCMC instead using the  supplied  align‐
              ment as a starting value.

</pre><h4><b>MODEL</b> <b>OPTIONS</b></h4><pre>
       <b>-A</b> <u>alphabet</u>, <b>–alphabet</b> <u>alphabet</u>
              The alphabet.

       <b>-S</b> <u>model</u>, <b>–smodel</b> <u>model</u>
              The substitution model.

       <b>-I</b> <u>model</u>, <b>–imodel</b> <u>model</u>
              The insertion-deletion model.

       <b>-B</b> <u>prior</u>, <b>–branch-lengths</b> <u>prior</u>
              Prior on branch lengths.

       <b>-R</b> <u>prior</u>, <b>–scale</b> <u>prior</u>
              Prior on the scale.

       <b>-L</b> <u>NUMS</u>, <b>–link</b> <u>NUMS</u>
              Link  partitions.   Takes  a  comma-separated  list of numbers indicating partitions.  For example
              <b>--link</b> <b>1,2,3</b>.

</pre><h4><b>EXAMPLES</b></h4><pre>
       <b>bali-phy</b> <b>dna.fasta</b> <b>--smodel</b> <b>gtr</b>
              Analyze sequences in <u>dna.fasta</u> under the GTR model.

       <b>bali-phy</b> <b>dna.fasta</b> <b>-S</b> <b>gtr</b> <b>-I</b> <b>none</b>
              Perform a traditional fixed-alignment analysis with gaps treated as missing data.

       <b>bali-phy</b> <b>dna.fasta</b> <b>amino.fasta</b> <b>codons.fasta</b> <b>-S</b> <b>1:gtr</b> <b>-S</b> <b>2:lg08</b> <b>-S</b> <b>3:gy94</b>
              Perform an analysis of 3 genes where each gene has a different substitution  mode.   The  sequence
              names in all three files must be the same.

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       BAli-Phy online help: &lt;<a href="http://www.bali-phy.org/docs.php">http://www.bali-phy.org/docs.php</a>&gt;.

       Please send bug reports to &lt;<a href="mailto:bali-phy-users@googlegroups.com">bali-phy-users@googlegroups.com</a>&gt;.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       bp-analyze

</pre><h4><b>AUTHORS</b></h4><pre>
       Benjamin Redelings.

                                                    Feb 2018                                         <u><a href="../man1/bali-phy.1.html">bali-phy</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>