<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>samblaster: -  post processing on an aligner (eg. bwa mem):</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/samblaster">samblaster_0.1.26-4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       samblaster: -  post processing on an aligner (eg. bwa mem):

</pre><h4><b>SYNOPSIS</b></h4><pre>
       bwa  mem  &lt;idxbase&gt;  samp.r1.fq  samp.r2.fq  |  samblaster  [-e] [-d samp.disc.sam] [-s samp.split.sam] |
       samtools view <b>-Sb</b> - &gt; samp.out.bam bwa mem <b>-M</b> &lt;idxbase&gt; samp.r1.fq samp.r2.fq | samblaster  <b>-M</b>  [-e]  [-d
       samp.disc.sam] [-s samp.split.sam] | samtools view <b>-Sb</b> - &gt; samp.out.bam

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Tool  to  mark duplicates and optionally output split reads and/or discordant pairs.  Input sam file must
       contain paired end data, contain sequence header  and  be  sorted  by  read  ids.   Output  will  be  all
       alignments in the same order as input, with duplicates marked with FLAG 0x400.

   <b>For</b> <b>use</b> <b>with</b> <b>a</b> <b>pre-existing</b> <b>bam</b> <b>file</b> <b>to</b> <b>pull</b> <b>split,</b> <b>discordant</b> <b>and/or</b> <b>unmapped</b> <b>reads:</b>
              samtools  view  <b>-h</b>  samp.bam  |  samblaster  [-a]  [-e] [-d samp.disc.sam] [-s samp.split.sam] [-u
              samp.umc.fasta] <b>-o</b> <u>/dev/null</u>

</pre><h4><b>OPTIONS</b></h4><pre>
       Input/Output Options: <b>-i</b> <b>--input</b>           FILE Input sam file [stdin].  <b>-o</b> <b>--output</b>          FILE Output
       sam file for all input alignments [stdout].  <b>-d</b> <b>--discordantFile</b>  FILE Output discordant  read  pairs  to
       this  file. [no discordant file output] <b>-s</b> <b>--splitterFile</b>    FILE Output split reads to this file abiding
       by parameters below. [no splitter file output] <b>-u</b> <b>--unmappedFile</b>    FILE Output unmapped/clipped reads as
       FASTQ to this file abiding by parameters below. [no unmapped file output].

       Requires soft clipping in input file.
              Will output FASTQ if QUAL information available, otherwise FASTA.

       Other Options: <b>-a</b> <b>--acceptDupMarks</b>       Accept duplicate marks already in input file instead of  looking
       for  duplicates  in  the  input.   <b>-e</b>  <b>--excludeDups</b>           Exclude  reads  marked  as duplicates from
       discordant, splitter, and/or unmapped file.  <b>-r</b> <b>--removeDups</b>           Remove duplicates reads  from  all
       output files. (Implies <b>--excludeDups</b>).

       <b>--addMateTags</b>
              Add MC and MQ tags to all output paired-end SAM lines.

       <b>--ignoreUnmated</b>
              Suppress  abort  on unmated alignments. Use only when sure input is read-id grouped and alignments
              have been filtered.

       <b>-M</b>                        Run in compatibility mode;  both  0x100  and  0x800  are  considered  chimeric.
       Similar to BWA MEM <b>-M</b> option.

       <b>--maxSplitCount</b>
              INT Maximum number of split alignments for a read to be included in splitter file. [2]

       <b>--maxUnmappedBases</b>  INT  Maximum  number  of  un-aligned  bases  between two alignments to be included in
              splitter file. [50]

       <b>--minIndelSize</b>
              INT Minimum structural variant feature size for split alignments to be included in splitter  file.
              [50]

       <b>--minNonOverlap</b>
              INT Minimum non-overlaping base pairs between two alignments for a read to be included in splitter
              file. [20]

       <b>--minClipSize</b>
              INT Minimum number of bases a mapped read must be clipped to be included in unmapped file. [20]

       <b>-q</b> <b>--quiet</b>                Output fewer statistics.

</pre><h4><b>VERSION</b></h4><pre>
       0.1.24

</pre><h4><b>AUTHOR</b></h4><pre>
       Greg Faust (<a href="mailto:gf4ea@virginia.edu">gf4ea@virginia.edu</a>)

samblaster: Version 0.1.24                          July 2018                                     <u>SAMBLASTER:</u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>