<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>rambo-k - Read Assignment Method Based On K-mers</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/rambo-k">rambo-k_1.21+dfsg-5_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       rambo-k - Read Assignment Method Based On K-mers

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>rambo-k</b> [-h] <b>-r</b> REFFILE1 <b>-R</b> REFFILE2 [-n NAME1] [-N NAME2] <b>-1</b> UNASSIGNED1 [-2 UNASSIGNED2] [-k KMERSIZES]
       [-o  OUTPATH]  [-a  AMOUNT] [-t THREADS] [-x CHUNKSIZE] [-g GAPSIZE] [-c CUTOFF_LOWER] [-C CUTOFF_HIGHER]
       [-d] [-f FILETYPE]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       RAMBO-K is a reference-based tool for rapid and sensitive extraction of one organisms reads from a  mixed
       dataset.  It  is  based  on  a  Markov  chain  implementation, which uses genomic characteristics of each
       reference to assign reads to the associated set.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>-r</b> REFFILE1, <b>--reffile1</b> REFFILE1
              Reference file of species 1 in fasta-format. Should pair with name1

       <b>-R</b> REFFILE2, <b>--reffile2</b> REFFILE2
              Reference file of species 2 in fasta-format. Should pair with name2

       <b>-n</b> NAME1, <b>--name1</b> NAME1
              Name of species 1

       <b>-N</b> NAME2, <b>--name2</b> NAME2
              Name of species 2

       <b>-1</b> UNASSIGNED1, <b>--unassigned1</b> UNASSIGNED1
              Fasta- or fastq-file containing mixed reads.

       <b>-2</b> UNASSIGNED2, <b>--unassigned2</b> UNASSIGNED2
              Fasta- or fastq-file containing mixed reads, only required in paired end mode.

       <b>-k</b> KMERSIZES, <b>--kmersizes</b> KMERSIZES
              Order of Markov-Chain/kmer length. Enter as range (e.g. 4:8) or list (e.g. 4,6,8) or integer (e.g.
              8).  Default = 8

       <b>-o</b> OUTPATH, <b>--outpath</b> OUTPATH
              Folder to write results to. Default = $name1_$name2/ in your working directory

       <b>-a</b> AMOUNT, <b>--amount</b> AMOUNT
              Number of reads to be simulated, default = 50000

       <b>-t</b> THREADS, <b>--threads</b> THREADS
              Number of Threads to use

       <b>-x</b> CHUNKSIZE, <b>--chunksize</b> CHUNKSIZE
              Size of chunks created at a time for simulation, default = 100000. Only change if  you  know  what
              you are doing!

       <b>-g</b> GAPSIZE, <b>--gapsize</b> GAPSIZE
              Estimated size of gapsize in case of paired end reads, default = 1

       <b>-c</b> CUTOFF_LOWER, <b>--cutoff_lower</b> CUTOFF_LOWER
              Lower cutoff: Output only reads with a score lower than or equal to this value, use m1 for <b>-1</b>

       <b>-C</b> CUTOFF_HIGHER, <b>--cutoff_higher</b> CUTOFF_HIGHER
              Higher cutoff: Output only reads with a score higher than or equal to this value, use m1 for <b>-1</b>

       <b>-d</b>, <b>--delete_temp</b>
              \Delete temporary files. Calculations will start from beginning next time.

       <b>-f</b> FILETYPE, <b>--filetype</b> FILETYPE
              Type of your input reads. fasta or fastq, default = fastq

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

rambo-k 1.21                                      November 2016                                       <u><a href="../man1/RAMBO-K.1.html">RAMBO-K</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>