<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Graphics::Glyph::wiggle_box - A generic box glyph compatible with dense "wig"data</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-graphics-perl">libbio-graphics-perl_2.40-6_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Graphics::Glyph::wiggle_box - A generic box glyph compatible with dense "wig"data

</pre><h4><b>SYNOPSIS</b></h4><pre>
         See &lt;Bio::Graphics::Panel&gt; and &lt;Bio::Graphics::Glyph&gt;.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This glyph works like the regular 'box' glyph but takes value data in Bio::Graphics::Wiggle file format:

        reference = chr1
        ChipCHIP Feature1 1..10000 wigfile=./test.wig;wigstart=0
        ChipCHIP Feature2 10001..20000 wigfile=./test.wig;wigstart=656
        ChipCHIP Feature3 25001..35000 wigfile=./test.wig;wigstart=1312

       The "wigfile" attribute gives a relative or absolute pathname to a Bio::Graphics::Wiggle format file. The
       optional "wigstart" option gives the offset to the start of the data. If not specified, a linear search
       will be used to find the data. The data consist of a packed binary representation of the values in the
       feature, using a constant step such as present in tiling array data.

       This glyph is intended for dense, qualitative feature data.  Any score data for each data point is only
       evaluated for true/false, when true, a box of the specified bgcolor is drawn, when false, nothing is
       drawn.  No data smoothing is used.

       Two primary benefits of using this glyph (with wiggle data) are:

        1) For large, genome-wide data sets, the speed of panel rendering is
           greatly improved.
        2) Large sets of related features can be rendered as a UCSC-style subtrack
           without the need for  aggregation or a GFF3 containment hierarchy.

       A disadvantage to this approach is that individual features will have no attributes associated with them
       and will appear as anonymous blocks within a sub-track.

       An example use for this glyph is annotated transcribed regions from microarray experiments.  Such regions
       are identified based on raw microarray data but do not necessarily have a score associated with them.  In
       this case, using the wiggle_box glyph provides a graphical summary of an expression array experiment.

   <b>DATA</b>
       The wiggle data used for this glyph should be loaded using the 'BED' format in order to allow features of
       variable width.  The fourth column should be a true value, with numeric or ".".  An example is shown
       below:

        track type=wiggle_0 name="transfrags" description="D. melanogaster transcribed fragments 0-2hrs"
        2L      9309    9451    1
        2L      10697   11021   1
        2L      11101   11345   1
        2L      11410   11521   1
        2L      11771   12243   1
        2L      12314   12954   1
        2L      13516   15746   1
        2L      17033   17191   1
        2L      18232   18580   1
        2L      19860   19999   1

   <b>OPTIONS</b>
       This glyph accepts the standard generic option set.  It differs in that the label and description and
       title/mouseover labels apply to the whole, panel-wide sub-track feature rather than to individual boxes.

       See Bio::Graphics::Glyph::wiggle_xyplot for a description of the wiggle-specific options and data
       formats.

</pre><h4><b>BUGS</b></h4><pre>
       Please report them.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Bio::Graphics::Panel, Bio::Graphics::Glyph, Bio::Graphics::Glyph::arrow, Bio::Graphics::Glyph::cds,
       Bio::Graphics::Glyph::crossbox, Bio::Graphics::Glyph::diamond, Bio::Graphics::Glyph::dna,
       Bio::Graphics::Glyph::dot, Bio::Graphics::Glyph::ellipse, Bio::Graphics::Glyph::extending_arrow,
       Bio::Graphics::Glyph::generic, Bio::Graphics::Glyph::graded_segments,
       Bio::Graphics::Glyph::heterogeneous_segments, Bio::Graphics::Glyph::line,
       Bio::Graphics::Glyph::pinsertion, Bio::Graphics::Glyph::primers, Bio::Graphics::Glyph::rndrect,
       Bio::Graphics::Glyph::segments, Bio::Graphics::Glyph::ruler_arrow, Bio::Graphics::Glyph::toomany,
       Bio::Graphics::Glyph::transcript, Bio::Graphics::Glyph::transcript2, Bio::Graphics::Glyph::translation,
       Bio::Graphics::Glyph::allele_tower, Bio::DB::GFF, Bio::SeqI, Bio::SeqFeatureI, Bio::Das, GD

</pre><h4><b>AUTHOR</b></h4><pre>
       Sheldon McKay  &lt;<a href="mailto:mckays@cshl.edu">mckays@cshl.edu</a>&gt;.

       Copyright (c) 2008 Cold Spring Harbor Laboratory

       This package and its accompanying libraries is free software; you can redistribute it and/or modify it
       under the terms of the GPL (either version 1, or at your option, any later version) or the Artistic
       License 2.0.  Refer to LICENSE for the full license text. In addition, please see DISCLAIMER.txt for
       disclaimers of warranty.

perl v5.30.0                                       2019-11-25              <u>Bio::Graphics::Glyph::<a href="../man3pm/wiggle_box.3pm.html">wiggle_box</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>