<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Graphics::Glyph::vista_plot - The "vista_plot" glyph</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-graphics-perl">libbio-graphics-perl_2.40-6_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Graphics::Glyph::vista_plot - The "vista_plot" glyph

</pre><h4><b>SYNOPSIS</b></h4><pre>
       See Bio::Graphics::Glyph, Bio::Graphics::Glyph::wiggle_xyplot and Bio::Graphics::Glyph::heat_map.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This glyph draws peak calls (features with discreet boundaries, i.e. putative transcription sites, over
       signal graph (wiggle_xyplot) requires a special load gff file that uses attributes 'wigfile' and
       'peak_type'

       <b>Example:</b>

       2L  chip_seq  vista  5407   23011573  .  .  .  Name=ChipSeq Exp
       1;wigfile=SomeWigFile.wigdb;peak_type=binding_site:exp1

       The glyph will draw the wiggle file first, than overlay the peaks (if there are any) over signal graph.
       Elsewhere in the GFF3 file, there should be one or more features of type "binding_site:exp1", e.g.:

       2L  exp1  binding_site  91934  92005  .  .  .

       Options like 'balloon hover' and 'link' are available to customize interaction with peaks in detail view.

       <b>BigWig</b> <b>support:</b>

       Supported bigwig format also requires another attribute to be supplied in load gff file (fasta) which
       specifies sequence index file for the organism in use. The data file should have the 'bw' extension - it
       is used to detect the BigWig format by vista_plot

       3L  chip_seq  vista   1    24543530  .  .  .   Name=ChipSeq Exp
       2;wigfile=SomeBigWigFile.bw;peak_type=binding_site:exp2;fasta=YourOrganism.fasta

       Note that all attributes should be present in load gff, as the code currently does not handle situation
       when only some of the attributes are in gff. To omit peak or signal drawing use "" (i.e. peak_type="") In
       both cases, the stanza code will look the same (only essential parameters shown):

        [VISTA_PLOT]
        feature         = vista:chip_seq
        glyph           = vista_plot
        label           = 1
        smoothing       = mean
        smoothing_window = 10
        bump density    = 250
        autoscale       = local
        variance_band   = 1
        max_peak        = 255
        min_peak        = 1
        peakwidth       = 3
        start_color     = lightgray
        end_color       = black
        pos_color       = blue
        neg_color       = orange
        bgcolor         = orange
        alpha           = 80
        fgcolor         = black
        database        = database_with_load_gff_data
        box_subparts    = 1
        bicolor_pivot   = min
        key             = VISTA plot

</pre><h4><b>OPTIONS</b></h4><pre>
       Options are the same as for wiggle_xyplot and heat_map

       <b>Additional</b> <b>parameters:</b>

       <b>alpha</b> set transparency for peak area.

       <b>glyph_subtype</b> Display only 'peaks', 'signal', 'density' or 'peaks+signal'.  Aliases for 'peaks+signal'
       include "both" and "vista".

       <b>Recommended</b> <b>global</b> <b>settings:</b>

       for proper peak drawing transparency should be enabled by setting <b>truecolors=1</b> in <u>GBrowse.conf</u> file

</pre><h4><b>BUGS</b></h4><pre>
       Please report them.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Bio::Graphics::Panel Bio::Graphics::Glyph Bio::Graphics::Glyph::wiggle_xyplot
       Bio::Graphics::Glyph::heat_map GD

</pre><h4><b>AUTHOR</b></h4><pre>
       Peter Ruzanov <a href="mailto:pruzanov@oicr.on.ca">pruzanov@oicr.on.ca</a>

       Copyright (c) 2010 Ontario Institute for Cancer Research

        This package and its accompanying libraries is free software; you can
        redistribute it and/or modify it under the terms of the GPL (either
        version 1, or at your option, any later version) or the Artistic
        License 2.0.  Refer to LICENSE for the full license text. In addition,
        please see DISCLAIMER.txt for disclaimers of warranty.

perl v5.30.0                                       2019-11-25              <u>Bio::Graphics::Glyph::<a href="../man3pm/vista_plot.3pm.html">vista_plot</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>