<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Graphics::Glyph::allele_tower - The "allele_tower" glyph</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-graphics-perl">libbio-graphics-perl_2.40-6_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Graphics::Glyph::allele_tower - The "allele_tower" glyph

</pre><h4><b>SYNOPSIS</b></h4><pre>
         See &lt;Bio::Graphics::Panel&gt; and &lt;Bio::Graphics::Glyph&gt;.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This glyph draws a letter for each allele found at a SNP position, one above the other (i.e. in a
       column). For example:
           A
           G

       See also <a href="http://www.hapmap.org/cgi-perl/gbrowse/gbrowse">http://www.hapmap.org/cgi-perl/gbrowse/gbrowse</a> 'genotyped SNPs' for an example.

       The common options are available (except height which is calculated based on the number of alleles).  In
       addition, if you give the glyph the minor allele frequency (MAF) and indicate which is the minor allele,
       the glyph will display these differences.

   <b>GETTING</b> <b>THE</b> <b>ALLELES</b>
       To specify the alleles, create an "Alleles" attribute for the feature.  There should be two such
       attributes.  For example, for a T/G polymorphism, the GFF load file should look like:

        Chr3  .  SNP   12345 12345 . . . SNP ABC123; Alleles T ; Alleles G

       Alternatively, you can pass an "alleles" callback to the appropriate section of the config file.  This
       option should return the two alleles separated by a slash:

         alleles = sub {
               my $snp = shift;
               my @d   = $snp-&gt;get_tag_values('AllelePair');
               return join "/",@d;
           }

   <b>OPTIONS</b>
        . Glyph Colour
        . Different colour for alleles on the reverse strand
        . Print out the complement for alleles on the reverse strand
        . Major allele shown in bold
        . Horizontal histogram to show allele frequency

       <u>GLYPH</u> <u>COLOR</u>

       The glyph color can be configured to be different if the feature is on the plus or minus strand.  Use
       fgcolor to define the glyph color for the plus strand and bgcolor for the minus strand.  For example:

          fgcolor     = blue
          bgcolor     = red

       For this option to work, you must also set ref_strand to return the strand of the feature:
          ref_strand        = sub {shift-&gt;strand}

       <u>REVERSE</u> <u>STRAND</u> <u>ALLELES</u>

       If the alleles on the negative strand need to be the complement of what is listed in the GFF files, (e.g.
       A/G becomes T/C), set the complement option to have value 1

       complement   = 1

       For this option to work, you must also set ref_strand to return the strand of the feature:

       ref_strand        = sub {shift-&gt;strand}

       <u>MAJOR/MINOR</u> <u>ALLELE</u>

       Use the 'minor_allele' option to return the minor allele for the SNP.  If you use this option, the major
       allele will appear in bold type.

       <u>ALLELE</u> <u>FREQUENCY</u> <u>HISTOGRAMS</u>

       Use the 'maf' option to return the minor allele frequency for the SNP.  If you use this option, a
       horizontal histogram will be drawn next to the alleles, to indicate their relative frequencies. e.g.

        A______
        C__

       Note: The 'label' option must be set to 1 (i.e. on) and the 'minor_allele' option must return a valid
       allele for this to work.

</pre><h4><b>BUGS</b></h4><pre>
       Please report them.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Bio::Graphics::Panel, Bio::Graphics::Glyph, Bio::Graphics::Glyph::arrow, Bio::Graphics::Glyph::cds,
       Bio::Graphics::Glyph::crossbox, Bio::Graphics::Glyph::diamond, Bio::Graphics::Glyph::dna,
       Bio::Graphics::Glyph::dot, Bio::Graphics::Glyph::ellipse, Bio::Graphics::Glyph::extending_arrow,
       Bio::Graphics::Glyph::generic, Bio::Graphics::Glyph::graded_segments,
       Bio::Graphics::Glyph::heterogeneous_segments, Bio::Graphics::Glyph::line,
       Bio::Graphics::Glyph::pinsertion, Bio::Graphics::Glyph::primers, Bio::Graphics::Glyph::rndrect,
       Bio::Graphics::Glyph::segments, Bio::Graphics::Glyph::ruler_arrow, Bio::Graphics::Glyph::toomany,
       Bio::Graphics::Glyph::transcript, Bio::Graphics::Glyph::transcript2, Bio::Graphics::Glyph::translation,
       Bio::Graphics::Glyph::allele_tower, Bio::DB::GFF, Bio::SeqI, Bio::SeqFeatureI, Bio::Das, GD

</pre><h4><b>AUTHOR</b></h4><pre>
       Fiona Cunningham &lt;<a href="mailto:cunningh@cshl.edu">cunningh@cshl.edu</a>&gt; in Lincoln Stein's lab &lt;<a href="mailto:steinl@cshl.edu">steinl@cshl.edu</a>&gt;.

       Copyright (c) 2003 Cold Spring Harbor Laboratory

       This package and its accompanying libraries is free software; you can redistribute it and/or modify it
       under the terms of the GPL (either version 1, or at your option, any later version) or the Artistic
       License 2.0.  Refer to LICENSE for the full license text. In addition, please see DISCLAIMER.txt for
       disclaimers of warranty.

perl v5.30.0                                       2019-11-25             <u>Bio::Graphics::...h::<a href="../man3pm/allele_tower.3pm.html">allele_tower</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>