<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Graphics::Glyph::hybrid_plot - An xyplot plot drawing dual graph using data from two or more wiggle</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-graphics-perl">libbio-graphics-perl_2.40-6_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Graphics::Glyph::hybrid_plot - An xyplot plot drawing dual graph using data from two or more wiggle
       files per track

</pre><h4><b>SYNOPSIS</b></h4><pre>
       See &lt;Bio::Graphics::Panel&gt; &lt;Bio::Graphics::Glyph&gt; and &lt;Bio::Graphics::Glyph::wiggle_xyplot&gt;.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Note that for full functionality this glyph requires Bio::Graphics::Glyph::generic (generic glyph is used
       for drawing individual matches for small RNA alignments at a high zoom level, specified by semantic
       zooming in GBrowse conf file) Unlike the regular xyplot, this glyph draws two overlapping graphs using
       value data in Bio::Graphics::Wiggle file format:

       track type=wiggle_0 name="Experiment" description="snRNA seq data" visibility=pack viewLimits=-2:2
       color=255,0,0 altColor=0,0,255 windowingFunction=mean smoothingWindow=16

        2L 400 500 0.5
        2L 501 600 0.5
        2L 601 700 0.4
        2L 701 800 0.1
        2L 800 900 0.1

       ##gff-version 3

       2L      Sample_rnaseq  rnaseq_wiggle 41   3009 . . .
       ID=Samlpe_2L;Name=Sample;Note=YourNoteHere;wigfileA=/datadir/track_001.2L.wig;wigfileB=/datadir/track_002.2L.wig

       The "wigfileA" and "wigfileB" attributes give a relative or absolute pathname to Bio::Graphics::Wiggle
       format files for two concurrent sets of data. Basically, these wigfiles contain the data on signal
       intensity (counts) for sequences aligned with genomic regions. In wigfileA these data are additive, so
       for each sequence region the signal is calculated as a sum of signals from overlapping matches (signal).
       In wigfileB the signal represents the maximum value among all sequences (signal quality) aligned with the
       current region so the user can see the difference between accumulated signal from overlapping multiple
       matches (which may likely be just a noise from products of degradation) and high-quality signal from
       unique sequences.

       For a third wiggle file use the attribute "wigfileC" and so forth.

       It is essential that wigfile entries in gff file do not have score, because score used to differentiate
       between data for dual graph and data for matches (individual features visible at higher magnification).
       After an update to wiggle_xyplot code colors for dual plot are now hard-coded (blue for signal and orange
       for signal quality). Alpha channel is also handled by wiggle_xyplot code now.

   <b>OPTIONS</b>
       In addition to some of the wiggle_xyplot glyph options, the following options are recognized:

        Name        Value        Description
        ----        -----        -----------

        wigfileA    path name    Path to a Bio::Graphics::Wiggle file for accumulated vales in 10-base bins

        wigfileB    path name    Path to a Bio::Graphics::Wiggle file for max values in 10-base bins

        fasta       path name    Path to fasta file to enable BigWig drawing

        u_method    method name  Use method of [method name] to identify individual features (like alignment matches)
                                 to show at high zoom level. By default it is set to 'match'

</pre><h4><b>BUGS</b></h4><pre>
        Please report them.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Bio::Graphics::Panel, Bio::Graphics::Glyph, Bio::Graphics::Glyph::arrow, Bio::Graphics::Glyph::cds,
       Bio::Graphics::Glyph::crossbox, Bio::Graphics::Glyph::diamond, Bio::Graphics::Glyph::dna,
       Bio::Graphics::Glyph::dot, Bio::Graphics::Glyph::ellipse, Bio::Graphics::Glyph::extending_arrow,
       Bio::Graphics::Glyph::generic, Bio::Graphics::Glyph::graded_segments,
       Bio::Graphics::Glyph::heterogeneous_segments, Bio::Graphics::Glyph::line,
       Bio::Graphics::Glyph::pinsertion, Bio::Graphics::Glyph::primers, Bio::Graphics::Glyph::rndrect,
       Bio::Graphics::Glyph::segments, Bio::Graphics::Glyph::ruler_arrow, Bio::Graphics::Glyph::toomany,
       Bio::Graphics::Glyph::transcript, Bio::Graphics::Glyph::transcript2, Bio::Graphics::Glyph::translation,
       Bio::Graphics::Glyph::allele_tower, Bio::DB::GFF, Bio::SeqI, Bio::SeqFeatureI, Bio::Das, GD

</pre><h4><b>AUTHOR</b></h4><pre>
       Peter Ruzanov &lt;<a href="mailto:pruzanov@oicr.on.ca">pruzanov@oicr.on.ca</a>&gt;.

       Copyright (c) 2008 Ontario Institute for Cancer Research

       This package and its accompanying libraries is free software; you can redistribute it and/or modify it
       under the terms of the GPL (either version 1, or at your option, any later version) or the Artistic
       License 2.0.  Refer to LICENSE for the full license text. In addition, please see DISCLAIMER.txt for
       disclaimers of warranty.

perl v5.30.0                                       2019-11-25             <u>Bio::Graphics::Glyph::<a href="../man3pm/hybrid_plot.3pm.html">hybrid_plot</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>