<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Graphics::Glyph::pairplot - The "pairwise plot" glyph</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-graphics-perl">libbio-graphics-perl_2.40-6_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Graphics::Glyph::pairplot - The "pairwise plot" glyph

</pre><h4><b>SYNOPSIS</b></h4><pre>
        use Bio::Graphics;

        # create the panel, etc.  See Bio::Graphics::Panel
        # for the synopsis

        # Create one big feature using the PairFeature
        # glyph (see end of synopsis for an implementation)
        my $block = PairFeature-&gt;new(-start=&gt;  2001,
                                     -end  =&gt; 10000);

        # It will contain a series of subfeatures.
        my $start = 2001;
        while ($start &lt; 10000) {
          my $end = $start+120;
          $block-&gt;add_SeqFeature($bsg-&gt;new(-start=&gt;$start,
                                           -end  =&gt;$end
                                          ),'EXPAND');
          $start += 200;
        }

        $panel-&gt;add_track($block,
                          -glyph =&gt; 'pairplot',
                          -angle =&gt; 45,
                          -bgcolor =&gt; 'red',
                          -point =&gt; 1,
                         );

        print $panel-&gt;png;

        package PairFeature;
        use base 'Bio::SeqFeature::Generic';

        sub pair_score {
          my $self = shift;
          my ($sf1,$sf2) = @_;
          # simple distance function
          my $dist  = $sf2-&gt;end    - $sf1-&gt;start;
          my $total = $self-&gt;end   - $self-&gt;start;
          return sprintf('%2.2f',1-$dist/$total);
        }

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This glyph draws a "triangle plot" similar to the ones used to show linkage disequilibrium between a
       series of genetic markers.  It is basically a dotplot drawn at a 45 degree angle, with each diamond-
       shaped region colored with an intensity proportional to an arbitrary scoring value relating one feature
       to another (typically a D' value in LD studies).

       This glyph requires more preparation than other glyphs.  First, you must create a subclass of
       Bio::SeqFeature::Generic (or Bio::Graphics::Feature, if you prefer) that has a <b>pair_score()</b> method.  The
       <b>pair_score()</b> method will take two features and return a numeric value between 0.0 and 1.0, where higher
       values mean more intense.

       You should then create a feature of this new type and use <b>add_SeqFeature()</b> to add to it all the genomic
       features that you wish to compare.

       Then add this feature to a track using the pairplot glyph.  When the glyph renders the feature, it will
       interrogate the <b>pair_score()</b> method for each pair of subfeatures.

   <b>OPTIONS</b>
       In addition to the common options, the following glyph-specific options are recognized:

         Option      Description                  Default
         ------      -----------                  -------

         -point      If true, the plot will be         0
                     drawn relative to the
                     midpoint between each adjacent
                     subfeature.  This is appropriate
                     for point-like subfeatures, such
                     as SNPs.

         -angle      Angle to draw the plot.  Values   45
                     between 1 degree and 89 degrees
                     are valid.  Higher angles give
                     a more vertical plot.

         -bgcolor    The color of the plot.            cyan

</pre><h4><b>BUGS</b></h4><pre>
       Please report them.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Bio::Graphics::Panel, Bio::Graphics::Glyph, Bio::Graphics::Glyph::arrow, Bio::Graphics::Glyph::cds,
       Bio::Graphics::Glyph::crossbox, Bio::Graphics::Glyph::diamond, Bio::Graphics::Glyph::dna,
       Bio::Graphics::Glyph::dot, Bio::Graphics::Glyph::ellipse, Bio::Graphics::Glyph::extending_arrow,
       Bio::Graphics::Glyph::generic, Bio::Graphics::Glyph::graded_segments,
       Bio::Graphics::Glyph::heterogeneous_segments, Bio::Graphics::Glyph::line,
       Bio::Graphics::Glyph::pinsertion, Bio::Graphics::Glyph::primers, Bio::Graphics::Glyph::rndrect,
       Bio::Graphics::Glyph::segments, Bio::Graphics::Glyph::ruler_arrow, Bio::Graphics::Glyph::toomany,
       Bio::Graphics::Glyph::transcript, Bio::Graphics::Glyph::transcript2, Bio::Graphics::Glyph::translation,
       Bio::Graphics::Glyph::triangle, Bio::Graphics::Glyph::xyplot, Bio::DB::GFF, Bio::SeqI, Bio::SeqFeatureI,
       Bio::Das, GD

</pre><h4><b>AUTHOR</b></h4><pre>
       Lincoln Stein &lt;<a href="mailto:lstein@cshl.edu">lstein@cshl.edu</a>&lt;gt&gt;.

       Copyright (c) 2004 Cold Spring Harbor Laboratory

       This package and its accompanying libraries is free software; you can redistribute it and/or modify it
       under the terms of the GPL (either version 1, or at your option, any later version) or the Artistic
       License 2.0.  Refer to LICENSE for the full license text. In addition, please see DISCLAIMER.txt for
       disclaimers of warranty.

perl v5.30.0                                       2019-11-25                <u>Bio::Graphics::Glyph::<a href="../man3pm/pairplot.3pm.html">pairplot</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>