<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Raptor-layout  -  A fast and space-efficient pre-filter for querying very large collections of nucleotide</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/seqan-raptor">seqan-raptor_3.0.1+ds-9_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Raptor-layout  -  A fast and space-efficient pre-filter for querying very large collections of nucleotide
       sequences.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Computes an HIBF layout that tries to minimize the disk space consumption of  the  resulting  index.  The
       space  is  estimated  using  a  k-mer  count  per  user  bin which represents the potential denisity in a
       technical  bin  in  an  interleaved  Bloom  filter.   You  can  pass  the  resulting  layout  to   raptor
       (https://github.com/seqan/raptor) to build the index and conduct queries.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Main</b> <b>options:</b>
       <b>--input-file</b> (<u>std::filesystem::path</u>)
              The  input must be a file containing paths to sequence data you wish to estimate; one filepath per
              line. If your file contains auxiliary information (e.g. species  IDs),  your  file  must  be  tab-
              separated.

       Example file:

       ```

       /absolute/path/to/file1.fasta

       /absolute/path/to/file2.fa.gz

       ```

       <b>--kmer-size</b> (<u>unsigned</u> <u>8</u> <u>bit</u> <u>integer</u>)
              The k-mer size influences the size estimates of the input. Choosing a k-mer size that is too small
              for  your data will result in files appearing more similar than they really are. Likewise, a large
              k-mer size might miss out on certain similarities. For DNA sequences, a k-mer size between [16,32]
              has proven to work well. Default: 19.

       <b>--num-hash-functions</b> (<u>unsigned</u> <u>64</u> <u>bit</u> <u>integer</u>)
              The number of hash functions to use when  building  the  HIBF  from  the  resulting  layout.  This
              parameter is needed to correctly estimate the index size when computing the layout. Default: 2.

       <b>--false-positive-rate</b> (<u>double</u>)
              The  false  positive  rate  you  aim  for  when  building the HIBF from the resulting layout. This
              parameter is needed to correctly estimate the index size when computing the layout. Default: 0.05.

       <b>--output-filename</b> (<u>std::filesystem::path</u>)
              A file name for the resulting layout. Default: "binning.out".

       <b>--threads</b> (<u>unsigned</u> <u>64</u> <u>bit</u> <u>integer</u>)
              The number of threads to use. Currently, only merging of sketches is parallelized, so if the  flag
              --disable-rearrangement  is set, --threads will have no effect. Default: 1. Value must be in range
              [1,18446744073709551615].

   <b>HyperLogLog</b> <b>Sketches:</b>
       To improve the layout, you can estimate the sequence similarities using HyperLogLog sketches.

       <b>--disable-estimate-union</b>
              The sketches are used to estimate the sequence similarity among a set  of  user  bins.  This  will
              improve  the layout computation as merging user bins that do not increase technical bin sizes will
              be preferred. This may use more RAM and can be disabled in RAM-critical  environments.  Attention:
              Also disables rearrangement which depends on union estimations.

       <b>--disable-rearrangement</b>
              As  a  preprocessing  step,  rearranging  the order of the given user bins based on their sequence
              similarity may lead to favourable small unions and thus a smaller index. Depending on  the  number
              of  input samples (user bins), this may be time-consuming and can thus be disabled if a suboptimal
              layout is sufficient.

   <b>Parameter</b> <b>Tweaking:</b>
   <b>Special</b> <b>options</b>
</pre><h4><b>REFERENCES</b></h4><pre>
       [1] Philippe Flajolet, Éric Fusy, Olivier Gandouet, Frédéric Meunier.  HyperLogLog:  the  analysis  of  a
       near-optimal  cardinality  estimation  algorithm.  AofA: Analysis of Algorithms, Jun 2007, Juan les Pins,
       France. pp.137-156. hal-00406166v2, https://doi.org/10.46298/dmtcs.3545

   <b>Common</b> <b>options</b>
       <b>-h</b>, <b>--help</b>
              Prints the help page.

       <b>-hh</b>, <b>--advanced-help</b>
              Prints the help page including advanced options.

       <b>--version</b>
              Prints the version information.

       <b>--copyright</b>
              Prints the copyright/license information.

       <b>--export-help</b> (std::string)
              Export the help page information. Value must be one of [html, man, ctd, cwl].

</pre><h4><b>VERSION</b></h4><pre>
       <b>Last</b> <b>update:</b> Unavailable
       <b>Raptor-layout</b> <b>version:</b> 3.0.1 (commit unavailable)
       <b>Sharg</b> <b>version:</b> 1.1.1
       <b>SeqAn</b> <b>version:</b> 3.4.0-rc.3

</pre><h4><b>URL</b></h4><pre>
       https://github.com/seqan/raptor

</pre><h4><b>LEGAL</b></h4><pre>
       <b>Raptor-layout</b> <b>Copyright:</b> BSD 3-Clause License
       <b>Author:</b> Svenja Mehringer
       <b>Contact:</b> <a href="mailto:svenja.mehringer@fu-berlin.de">svenja.mehringer@fu-berlin.de</a>
       <b>SeqAn</b> <b>Copyright:</b> 2006-2023 Knut Reinert, FU-Berlin; released under the 3-clause BSDL.
       <b>In</b> <b>your</b> <b>academic</b> <b>works</b> <b>please</b> <b>cite:</b> Raptor: A fast and space-efficient pre-filter for querying very large
       collections of nucleotide sequences; Enrico Seiler, Svenja Mehringer, Mitra Darvish,  Etienne  Turc,  and
       Knut Reinert; iScience 2021 24 (7): 102782. doi: https://doi.org/10.1016/j.isci.2021.102782
       For full copyright and/or warranty information see <b>--copyright</b>.

raptor-layout 3.0.1 (commit unavailable)           Unavailable                                  <u><a href="../man1/RAPTOR-LAYOUT.1.html">RAPTOR-LAYOUT</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>