<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Copyright [2015-2018] EMBL-European Bioinformatics Institute</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-db-hts-perl">libbio-db-hts-perl_3.01-5_amd64</a> <br><br><pre>
</pre><h4><b>LICENSE</b></h4><pre>
       Copyright [2015-2018] EMBL-European Bioinformatics Institute

       Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
       compliance with the License.  You may obtain a copy of the License at

            <a href="http://www.apache.org/licenses/LICENSE-2.0">http://www.apache.org/licenses/LICENSE-2.0</a>

       Unless required by applicable law or agreed to in writing, software distributed under the License is
       distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
       See the License for the specific language governing permissions and limitations under the License.

</pre><h4><b>AUTHOR</b></h4><pre>
       Rishi Nag &lt;<a href="mailto:rishi@ebi.ac.uk">rishi@ebi.ac.uk</a>&lt;gt&gt;

</pre><h4><b>NAME</b></h4><pre>
       Bio::DB::HTS::Query -- Object representing the query portion of a BAM/SAM alignment

</pre><h4><b>SYNOPSIS</b></h4><pre>
       Given an alignment retrieved from a Bio::DB::HTS database,

        my $query = $alignment-&gt;query;

        my $name   = $query-&gt;display_name;
        my $start  = $query-&gt;start;
        my $end    = $query-&gt;end;
        my $dna    = $query-&gt;dna;    # dna string
        my $seq    = $query-&gt;seq;    # Bio::PrimarySeq object
        my @scores = $query-&gt;qscore; # quality score

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This is a simple Bio::SeqFeatureI object that represents the query part of a SAM alignment.

   <b>Methods</b>
       $seqid = $query-&gt;seq_id
           The name of the read.

       $name = $query-&gt;name
           The read name (same as seq_id in this case).

       $name = $query-&gt;display_name
           The read display_name (same as seq_id in this case).

       $tag = $query-&gt;primary_tag
           The string "match".

       $tag = $query-&gt;source_tag
           The string "sam/bam".

       $start = $query-&gt;start
           The start of the match in read coordinates.

       $end = $query-&gt;end
           The end of the match in read coordinates;

       $len = $query-&gt;length
           The length of the read.

       $seq = $query-&gt;seq
           A Bio::PrimarySeq representing the read sequence in REFERENCE orientation.

       $scores = $query-&gt;qscore
           The  read  quality scores. In a list context, a list of integers equal in length to the read sequence
           length. In a scalar context, an array ref. The qscores are in REFERENCE sequence orientation.

       $dna = $query-&gt;dna
           The DNA string in reference sequence orientation.

       $strand = $query-&gt;strand
           If the query was reversed to align it, -1. Otherwise +1.

       $seq = $query-&gt;subseq($start,$end)
           Return a Bio::PrimarySeq object representing the requested subsequence on the read.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Bio::Perl, Bio::DB::HTS, Bio::DB::HTS::Alignment, Bio::DB::HTS::Constants

perl v5.40.0                                       2024-10-31                           <u>Bio::DB::HTS::<a href="../man3pm/Query.3pm.html">Query</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>