<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>glam2scan - finds a GLAM2 motif in a database</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/glam2">glam2_1064-10_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       glam2scan - finds a GLAM2 motif in a database

</pre><h4><b>SYNOPSIS</b></h4><pre>

       <b>glam2scan</b> [<b>options</b>] <b>alphabet</b> <u>my_motif.glam2</u> <u>my_seqs.fa</u>

       An alphabet other than <u>p</u> or <u>n</u> is interpreted as the name of an alphabet file.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>glam2scan</b> finds matches, in a sequence database, to a motif discovered by <b>glam2</b>. Each match receives a
       score, indicating how well it fits the motif.

</pre><h4><b>OPTIONS</b> <b>(DEFAULT</b> <b>SETTINGS)</b></h4><pre>
       <b>-h</b>
           Show all options and their default settings.

       <b>-o</b>
           Output file (<u>stdout</u>).

       <b>-n</b>
           Number of alignments to report (<u>25</u>).

       <b>-2</b>
           Examine both strands - forward and reverse complement.

       <b>-D</b>
           Deletion pseudocount (<u>0.1</u>).

       <b>-E</b>
           No-deletion pseudocount (<u>2.0</u>).

       <b>-I</b>
           Insertion pseudocount (<u>0.02</u>).

       <b>-J</b>
           No-insertion pseudocount (<u>1.0</u>).

       <b>-d</b>
           Dirichlet mixture file.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/glam2format.1.html">glam2format</a></b>(1), <b><a href="../man1/glam2mask.1.html">glam2mask</a></b>(1), <b><a href="../man1/glam2-purge.1.html">glam2-purge</a></b>(1), <b><a href="../man1/glam2.1.html">glam2</a></b>(1)

       The full Hypertext documentation of GLAM2 is available online at <u><a href="http://bioinformatics.org.au/glam2/">http://bioinformatics.org.au/glam2/</a></u> or
       on this computer in <u>/usr/share/doc/glam2/</u>.

</pre><h4><b>REFERENCE</b></h4><pre>
       If you use GLAM2, please cite: MC Frith, NFW Saunders, B Kobe, TL Bailey (2008) Discovering sequence
       motifs with arbitrary insertions and deletions, PLoS Computational Biology (in press).

</pre><h4><b>AUTHORS</b></h4><pre>
       <b>Martin</b> <b>Frith</b>
           Author of GLAM2.

       <b>Timothy</b> <b>Bailey</b>
           Author of GLAM2.

       <b>Charles</b> <b>Plessy</b> &lt;<a href="mailto:plessy@debian.org">plessy@debian.org</a>&gt;
           Formatted this manpage in DocBook XML for the Debian distribution.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       The source code and the documentation of GLAM2 are released in the public domain.

GLAM2 1056                                         05/19/2008                                       <u><a href="../man1/GLAM2SCAN.1.html">GLAM2SCAN</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>