<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>glam2format - converts GLAM2 motifs to FASTA or MSF format</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/glam2">glam2_1064-10_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       glam2format - converts GLAM2 motifs to FASTA or MSF format

</pre><h4><b>SYNOPSIS</b></h4><pre>

       <b>glam2format</b> [<b>options</b>] <b>my_format</b> <u>my_motif.glam2</u>

       Formats: fasta, msf.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>glam2format</b> reads in a motif found by <b>glam2</b>, and writes it in a standard alignment format
       (FASTA-with-gaps or MSF). This enables the alignment to be passed to third-party software, including
       graphical visualization tools such as Kalignvu, Boxshade, and WebLogo. On the other hand, not all the
       motif information is preserved: in particular, the key positions are lost. Only the top motif in <b>glam2</b>
       output is converted.

</pre><h4><b>OPTIONS</b> <b>(DEFAULT</b> <b>SETTINGS)</b></h4><pre>
       <b>-o</b>
           Output file (<u>stdout</u>).

       <b>-c</b>
           Make a compact alignment. By default, residues that are inserted between key positions are written as
           unaligned with each other. This best reflects <b>glam2</b>´s intention, but it can make the alignment large
           and full of gaps. With <u>-c</u>, inserted residues are written as arbitrarily aligned with each other, just
           as they appear in the <b>glam2</b> output.

       <b>-f</b>
           Sequence file to make a “global” alignment by adding flanking sequences from the original
           FASTA-format sequence file. The flanking sequences will be written as either unaligned with each
           other or arbitrarily aligned, depending on the <u>-c</u> option. The sequences should have unique names and
           their order should be unchanged.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/boxshade.1.html">boxshade</a></b>(1), <b><a href="../man1/glam2.1.html">glam2</a></b>(1), <b><a href="../man1/glam2mask.1.html">glam2mask</a></b>(1), <b><a href="../man1/glam2-purge.1.html">glam2-purge</a></b>(1), <b><a href="../man1/glam2scan.1.html">glam2scan</a></b>(1)

       The full Hypertext documentation of GLAM2 is available online at <u><a href="http://bioinformatics.org.au/glam2/">http://bioinformatics.org.au/glam2/</a></u> or
       on this computer in <u>/usr/share/doc/glam2/</u>.

</pre><h4><b>REFERENCE</b></h4><pre>
       If you use GLAM2, please cite: MC Frith, NFW Saunders, B Kobe, TL Bailey (2008) Discovering sequence
       motifs with arbitrary insertions and deletions, PLoS Computational Biology (in press).

</pre><h4><b>AUTHORS</b></h4><pre>
       <b>Martin</b> <b>Frith</b>
           Author of GLAM2.

       <b>Timothy</b> <b>Bailey</b>
           Author of GLAM2.

       <b>Charles</b> <b>Plessy</b> &lt;<a href="mailto:plessy@debian.org">plessy@debian.org</a>&gt;
           Formatted this manpage in DocBook XML for the Debian distribution.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       The source code and the documentation of GLAM2 are released in the public domain.

GLAM2 1056                                         05/19/2008                                     <u><a href="../man1/GLAM2FORMAT.1.html">GLAM2FORMAT</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>