<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>vcf2genomicsdb_init - workspace initializer for GenomicsDB</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/genomicsdb-tools">genomicsdb-tools_1.5.4-5_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       vcf2genomicsdb_init - workspace initializer for GenomicsDB

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>vcf2genomicsdb_init</b> [<u>options</u>]

</pre><h4><b>OPTIONS</b></h4><pre>
              --help, -h

              Print a usage message summarizing options available and exit

              --workspace=&lt;GenomicsDB workspace URI&gt;, -w &lt;GenomicsDB workspace URI&gt;

              If  workspace does not exist, it is created first exits if workspace exists and is invoked without
              the overwrite-workspace option

              --overwrite-workspace, -o

              Allow for workspace json artifacts to be overwritten

              --sample-list=&lt;sample list&gt;, -s &lt;sample list file&gt;

              Specify sample URIs for import, one line per sample path

              --samples-dir=&lt;folder to samples&gt;, -S &lt;folder to samples&gt;

              Specify Folder URI containing samples. Only vcf.gz/bcf.gz compressed samples are considered

              --interval-list=&lt;genomic interval list&gt;, -i &lt;genomic interval list file&gt;

              Optional, create array partitions from intervals in interval list, one line per interval,  default
              is     partition     by     chromosome/contig,    overrides    <b>--number-of-array-partitions</b>    and
              <b>--size-of-array-partitions</b>

              --number-of-array-partitions=&lt;number&gt;, -n &lt;number&gt;

              Optional, suggested number of array partitions. Usually, the partitioning is per  contig  But,  if
              this  is 0, only a single array is created for the entire genomic space and it overrides all other
              partition specific command arguments

              --size-of-array-partitions=&lt;size&gt;, -z &lt;size&gt;

              Optional, suggested size of arrays partitions, overrides <b>--number-of-array-partitions</b>

              --merge-small-contigs, -m

              Optional, default is false and any contig smaller than ~1M will be merged into scaffolds

              --include-fields=&lt;fields&gt;, -f &lt;fields&gt;

              Optional, Include only fields(comma-separated) listed in this argument while generating the vidmap
              default is to include all fields found in the vcf headers

              --template-loader-json=&lt;template file&gt;, -t &lt;template file&gt;

              Optional, specify a template loader json file to use as a basis with loader json files

              --append-samples, -a

              Optional, if specified, callsets will be appended with the new samples and lb_row_idx set  to  the
              new starting row. Note that the workspace and vidmap/callset/loader jsons should already exist and
              that  the interval list, number and size of partitions and merge small contigs options are ignored
              with append samples

              --verbose, -v

              Allow verbose messages to be logged

              --version Print version and exit

vcf2genomicsdb_init                                 July 2022                             <u><a href="../man1/VCF2GENOMICSDB_INIT.1.html">VCF2GENOMICSDB_INIT</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>