<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>samtools-view - views and converts SAM/BAM/CRAM files</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/samtools">samtools_1.21-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       samtools-view - views and converts SAM/BAM/CRAM files

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>samtools</b> <b>view</b> [<u>options</u>] <u>in.sam</u>|<u>in.bam</u>|<u>in.cram</u> [<u>region</u>...]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       With  no  options  or  regions specified, prints all alignments in the specified input alignment file (in
       SAM, BAM, or CRAM format) to standard output in SAM format (with no header).

       You may specify one or more space-separated region specifications after the input  filename  to  restrict
       output  to  only  those  alignments  which  overlap the specified region(s). Use of region specifications
       requires a coordinate-sorted and indexed input file (in BAM or CRAM format).

       The <b>-b</b>, <b>-C</b>, <b>-1</b>, <b>-u</b>, <b>-h</b>, <b>-H</b>, and <b>-c</b> options change the output format from the default of  headerless  SAM,
       and the <b>-o</b> and <b>-U</b> options set the output file name(s).

       The  <b>-t</b>  and  <b>-T</b> options provide additional reference data. One of these two options is required when SAM
       input does not contain @SQ headers, and the <b>-T</b> option is required whenever writing CRAM output.

       The <b>-L</b>, <b>-M</b>, <b>-N</b>, <b>-r</b>, <b>-R</b>, <b>-d</b>, <b>-D</b>, <b>-s</b>, <b>-q</b>, <b>-l</b>, <b>-m</b>, <b>-f</b>, <b>-F</b>, <b>-G</b>, and <b>--rf</b> options filter the  alignments  that
       will be included in the output to only those alignments that match certain criteria.

       The <b>-p</b>, option sets the UNMAP flag on filtered alignments then writes them to the output file.

       The <b>-x</b>, <b>-B</b>, <b>--add-flags</b>, and <b>--remove-flags</b> options modify the data which is contained in each alignment.

       The  <b>-X</b>  option can be used to allow user to specify customized index file location(s) if the data folder
       does not contain any index file. See <b>EXAMPLES</b> section for sample of usage.

       Finally, the <b>-@</b> option can be used to allocate additional threads to be used for compression, and the  <b>-?</b>
       option requests a long help message.

       <b>REGIONS:</b>
              Regions can be specified as: RNAME[:STARTPOS[-ENDPOS]] and all position coordinates are 1-based.

              Important  note:  when multiple regions are given, some alignments may be output multiple times if
              they overlap more than one of the specified regions.

              Examples of region specifications:

              <b>chr1</b>      Output all alignments mapped to the reference sequence named `chr1' (i.e. @SQ SN:chr1).

              <b>chr2:1000000</b>
                        The region on chr2 beginning at base position 1,000,000 and ending at  the  end  of  the
                        chromosome.

              <b>chr3:1000-2000</b>
                        The  1001bp  region on chr3 beginning at base position 1,000 and ending at base position
                        2,000 (including both end positions).

              <b>'*'</b>       Output the unmapped reads at the end of the file.  (This does not include  any  unmapped
                        reads placed on a reference sequence alongside their mapped mates.)

              <b>.</b>         Output  all  alignments.   (Mostly unnecessary as not specifying a region at all has the
                        same effect.)

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-b</b>, <b>--bam</b> Output in the BAM format.

       <b>-C</b>, <b>--cram</b>
                 Output in the CRAM format (requires -T).

       <b>-1</b>, <b>--fast</b>
                 Enable fast compression.  This also changes the default output format to BAM, but this  can  be
                 overridden by the explicit format options or using a filename with a known suffix.

       <b>-u</b>, <b>--uncompressed</b>
                 Output  uncompressed  data. This also changes the default output format to BAM, but this can be
                 overridden by the explicit format options or using a filename with a known suffix.

                 This option saves time spent on compression/decompression and is thus preferred when the output
                 is piped to another samtools command.

       <b>-h</b>, <b>--with-header</b>
                 Include the header in the output.

       <b>-H</b>, <b>--header-only</b>
                 Output the header only.

       <b>--no-header</b>
                 When producing SAM format, output alignment records but not headers.  This is the default;  the
                 option can be used to reset the effect of <b>-h</b>/<b>-H</b>.

       <b>-c</b>, <b>--count</b>
                 Instead  of  printing  the  alignments,  only count them and print the total number. All filter
                 options, such as <b>-f</b>, <b>-F</b>, and <b>-q</b>, are taken into account.  The <b>-p</b>  option  is  ignored  in  this
                 mode.

       <b>-?</b>, <b>--help</b>
                 Output long help and exit immediately.

       <b>-o</b> <u>FILE</u><b>,</b> <b>--output</b> <u>FILE</u>
                 Output to <u>FILE</u> <u>[stdout].</u>

       <b>-U</b> <u>FILE</u><b>,</b> <b>--unoutput</b> <u>FILE</u><b>,</b> <b>--output-unselected</b> <u>FILE</u>
                 Write alignments that are <u>not</u> selected by the various filter options to <u>FILE</u>.  When this option
                 is  used,  all alignments (or all alignments intersecting the <u>regions</u> specified) are written to
                 either the output file or this file, but never both.

       <b>-p</b>, <b>--unmap</b>
                 Set the UNMAP flag on alignments that are not selected by the filter options.  These alignments
                 are then written to the normal output.  This is not compatible with <b>-U</b>.

       <b>-t</b> <u>FILE</u><b>,</b> <b>--fai-reference</b> <u>FILE</u>
                 A tab-delimited <u>FILE</u>.  Each line must contain the reference name in the first  column  and  the
                 length  of  the reference in the second column, with one line for each distinct reference.  Any
                 additional fields beyond the second column are ignored. This file also defines the order of the
                 reference sequences in sorting. If you run: `samtools faidx &lt;ref.fa&gt;', the resulting index file
                 <u>&lt;ref.fa&gt;.fai</u> can be used as this <u>FILE</u>.

       <b>-T</b> <u>FILE</u><b>,</b> <b>--reference</b> <u>FILE</u>
                 A FASTA format reference <u>FILE</u>, optionally compressed by <b>bgzip</b> and ideally indexed  by  <b>samtools</b>
                 <b>faidx</b>.   If  an  index  is  not present one will be generated for you, if the reference file is
                 local.

                 If the reference file is not local, but is accessed instead via an  https://,  s3://  or  other
                 URL,  the  index  file  will  need to be supplied by the server alongside the reference.  It is
                 possible to have the reference and index files in different locations by supplying both to this
                 option separated by the string "##idx##", for example:

                 <b>-T</b> <b><a href="ftp://x.com/ref.fa">ftp://x.com/ref.fa</a>##idx##<a href="ftp://y.com/index.fa.fai">ftp://y.com/index.fa.fai</a></b>

                 However, note that only the location of the reference will be stored in the output file header.
                 If this method is used to make CRAM files, the cram reader may not be able to find  the  index,
                 and  may  not  be  able  to  decode  the file unless it can get the references it needs using a
                 different method.

       <b>-L</b> <u>FILE</u><b>,</b> <b>--target-file</b> <u>FILE</u><b>,</b> <b>--targets-file</b> <u>FILE</u>
                 Only output alignments overlapping the input BED <u>FILE</u> [null].

       <b>-M</b>, <b>--use-index</b>
                 Use the multi-region iterator on the union of a BED file  and  command-line  region  arguments.
                 This  avoids  re-reading  the same regions of files so can sometimes be much faster.  Note this
                 also removes duplicate sequences.  Without this  a  sequence  that  overlaps  multiple  regions
                 specified  on  the  command  line  will be reported multiple times.  The usage of a BED file is
                 optional and its path has to be preceded by <b>-L</b> option.

       <b>--region-file</b> <u>FILE</u><b>,</b> <b>--regions-file</b> <u>FILE</u>
                 Use an index and multi-region iterator to only output  alignments  overlapping  the  input  BED
                 <u>FILE</u>.  Equivalent to <b>-M</b> <b>-L</b> <u>FILE</u> or <b>--use-index</b> <b>--target-file</b> <u>FILE</u>.

       <b>-N</b> <u>FILE</u><b>,</b> <b>--qname-file</b> <u>FILE</u>
                 Output  only  alignments  with  read  names  listed  in  <u>FILE</u>.   If <u>FILE</u> starts with <b>^</b> then the
                 operation is negated and only outputs alignment with read groups not listed in <u>FILE</u>.  It is not
                 permissible to mix both the filter-in and filter-out style syntax in the same command.

       <b>-r</b> <u>STR</u><b>,</b> <b>--read-group</b> <u>STR</u>
                 Output alignments in read group <u>STR</u> [null].  Note that records with no  <b>RG</b>  tag  will  also  be
                 output when using this option.  This behaviour may change in a future release.

       <b>-R</b> <u>FILE</u><b>,</b> <b>--read-group-file</b> <u>FILE</u>
                 Output  alignments  in  read  groups  listed  in  <u>FILE</u>  [null].  If <u>FILE</u> starts with <b>^</b> then the
                 operation is negated and only outputs alignment with read names not listed in <u>FILE</u>.  It is  not
                 permissible  to  mix  both the filter-in and filter-out style syntax in the same command.  Note
                 that records with no <b>RG</b> tag will also be output when using this  option.   This  behaviour  may
                 change in a future release.

       <b>-d</b> <u>STR1[:STR2]</u><b>,</b> <b>--tag</b> <u>STR1[:STR2]</u>
                 Only  output  alignments  with  tag <u>STR1</u> and associated value <u>STR2</u>, which can be a string or an
                 integer [null].  The value can be omitted, in which case only the tag is considered.

                 Note that this option does not specify a tag  type.   For  example,  use  <b>-d</b>  <b>XX:42</b>  to  select
                 alignments with an <b>XX:i:42</b> field, not <b>-d</b> <b>XX:i:42</b>.

       <b>-D</b> <u>STR:FILE</u><b>,</b> <b>--tag-file</b> <u>STR:FILE</u>
                 Only output alignments with tag <u>STR</u> and associated values listed in <u>FILE</u> [null].

       <b>-q</b> <u>INT</u><b>,</b> <b>--min-MQ</b> <u>INT</u>
                 Skip alignments with MAPQ smaller than <u>INT</u> [0].

       <b>-l</b> <u>STR</u><b>,</b> <b>--library</b> <u>STR</u>
                 Only output alignments in library <u>STR</u> [null].

       <b>-m</b> <u>INT</u><b>,</b> <b>--min-qlen</b> <u>INT</u>
                 Only output alignments with number of CIGAR bases consuming query sequence ≥ <u>INT</u> [0]

       <b>-e</b> <u>STR</u><b>,</b> <b>--expr</b> <u>STR</u>
                 Only include alignments that match the filter expression <u>STR</u>.  The syntax for these expressions
                 is described in the main <a href="../man1/samtools.1.html">samtools</a>(1) man page under the FILTER EXPRESSIONS heading.

       <b>-f</b> <u>FLAG</u><b>,</b> <b>--require-flags</b> <u>FLAG</u>
                 Only  output  alignments  with  all  bits  set  in <u>FLAG</u> present in the FLAG field.  <u>FLAG</u> can be
                 specified in hex by beginning with `0x' (i.e. /^0x[0-9A-F]+/), in octal by beginning  with  `0'
                 (i.e.  /^0[0-7]+/),  as a decimal number not beginning with '0' or as a comma-separated list of
                 flag names.

                 For a list of flag names see <u><a href="../man1/samtools-flags.1.html">samtools-flags</a></u>(1).

       <b>-F</b> <u>FLAG</u><b>,</b> <b>--excl-flags</b> <u>FLAG</u><b>,</b> <b>--exclude-flags</b> <u>FLAG</u>
                 Do not output alignments with any bits set in <u>FLAG</u> present in the  FLAG  field.   <u>FLAG</u>  can  be
                 specified  in  hex by beginning with `0x' (i.e. /^0x[0-9A-F]+/), in octal by beginning with `0'
                 (i.e. /^0[0-7]+/), as a decimal number not beginning with '0' or as a comma-separated  list  of
                 flag names.

       <b>--rf</b> <u>FLAG</u> <b>,</b> <b>--incl-flags</b> <u>FLAG</u><b>,</b> <b>--include-flags</b> <u>FLAG</u>
                 Only  output  alignments  with  any  bit  set  in  <u>FLAG</u> present in the FLAG field.  <u>FLAG</u> can be
                 specified in hex by beginning with `0x' (i.e. /^0x[0-9A-F]+/), in octal by beginning  with  `0'
                 (i.e.  /^0[0-7]+/),  as a decimal number not beginning with '0' or as a comma-separated list of
                 flag names.

       <b>-G</b> <u>FLAG</u>   Do not output alignments with all bits set in <u>INT</u> present in  the  FLAG  field.   This  is  the
                 opposite  of  <u>-f</u> such that <u>-f12</u> <u>-G12</u> is the same as no filtering at all.  <u>FLAG</u> can be specified
                 in hex by beginning with `0x' (i.e. /^0x[0-9A-F]+/), in  octal  by  beginning  with  `0'  (i.e.
                 /^0[0-7]+/),  as  a  decimal number not beginning with '0' or as a comma-separated list of flag
                 names.

       <b>-x</b> <u>STR</u><b>,</b> <b>--remove-tag</b> <u>STR</u>
                 Read tag(s) to exclude from output (repeatable) [null].  This can be a single tag  or  a  comma
                 separated list.  Alternatively the option itself can be repeated multiple times.

                 If  the  list  starts with a `^' then it is negated and treated as a request to remove all tags
                 except those in <u>STR</u>. The list may be empty, so <b>-x</b> <b>^</b> will remove all tags.

                 Note that tags will only be removed from reads that pass filtering.

       <b>--keep-tag</b> <u>STR</u>
                 This keeps <u>only</u> tags listed in <u>STR</u> and is directly equivalent to <b>--remove-tag</b> <b>^</b><u>STR</u>.  Specifying
                 an empty list will remove all tags.  If both <b>--keep-tag</b> and  <b>--remove-tag</b>  are  specified  then
                 <b>--keep-tag</b> has precedence.

                 Note that tags will only be removed from reads that pass filtering.

       <b>-B</b>, <b>--remove-B</b>
                 Collapse the backward CIGAR operation.

       <b>--add-flags</b> <u>FLAG</u>
                 Adds  flag(s)  to  read.   <u>FLAG</u> can be specified in hex by beginning with `0x' (i.e. /^0x[0-9A-
                 F]+/), in octal by beginning with `0' (i.e. /^0[0-7]+/), as a decimal number not beginning with
                 '0' or as a comma-separated list of flag names.

       <b>--remove-flags</b> <u>FLAG</u>
                 Remove flag(s) from read.  <u>FLAG</u> is specified in the same way as with the <b>--add-flags</b> option.

       <b>--subsample</b> <u>FLOAT</u>
                 Output only a proportion of the input alignments, as specified by 0.0  ≤  <u>FLOAT</u>  ≤  1.0,  which
                 gives the fraction of templates/pairs to be kept.  This subsampling acts in the same way on all
                 of  the  alignment  records in the same template or read pair, so it never keeps a read but not
                 its mate.

       <b>--subsample-seed</b> <u>INT</u>
                 Subsampling seed used to influence <u>which</u> subset of reads is kept.  When subsampling  data  that
                 has  previously  been  subsampled,  be  sure  to  use  a  different  seed value from those used
                 previously; otherwise more reads will be retained than expected.  [0]

       <b>-s</b> <u>FLOAT</u>  Subsampling shorthand option: <b>-s</b> <u>INT</u><b>.</b><u>FRAC</u> is equivalent  to  <b>--subsample-seed</b>  <u>INT</u>  <b>--subsample</b>
                 0.<u>FRAC</u>.

       <b>-@</b> <u>INT</u><b>,</b> <b>--threads</b> <u>INT</u>
                 Number of BAM compression threads to use in addition to main thread [0].

       <b>-P</b>, <b>--fetch-pairs</b>
                 Retrieve  pairs  even  when  the mate is outside of the requested region.  Enabling this option
                 also turns on the multi-region iterator (<b>-M</b>).  A region to search must be specified, either  on
                 the command-line, or using the <b>-L</b> option.  The input file must be an indexed regular file.

                 This  option  first scans the requested region, using the <b>RNEXT</b> and <b>PNEXT</b> fields of the records
                 that have the PAIRED flag set and pass other filtering options to find where paired  reads  are
                 located.   These  locations  are  used to build an expanded region list, and a set of <b>QNAME</b>s to
                 allow from the new regions.  It will then make a second pass, collecting  all  reads  from  the
                 originally-specified  region  list together with reads from additional locations that match the
                 allowed set of <b>QNAME</b>s.  Any other filtering options used will be applied  to  all  reads  found
                 during this second pass.

                 As  this  option  links  reads using <b>RNEXT</b> and <b>PNEXT</b>, it is important that these fields are set
                 accurately.  Use 'samtools fixmate' to correct them if necessary.

                 Note that this option does not work with the  <b>-c,</b>  <b>--count</b>;  <b>-U,</b>  <b>--output-unselected</b>;  or  <b>-p,</b>
                 <b>--unmap</b> options.

       <b>-S</b>        Ignored for compatibility with previous samtools versions.  Previously this option was required
                 if  input  was in SAM format, but now the correct format is automatically detected by examining
                 the first few characters of input.

       <b>-X</b>, <b>--customized-index</b>
                 Include customized index file as a part of arguments. See <b>EXAMPLES</b> section for sample of usage.

       <b>-z</b> <u>FLAGs</u><b>,</b> <b>--sanitize</b> <u>FLAGs</u>
                 Perform some sanity checks on the state of SAM record fields, fixing up common mistakes made by
                 aligners.  These include soft-clipping alignments when  they  extend  beyond  the  end  of  the
                 reference,  marking  records as unmapped when they have reference * or position 0, and ensuring
                 unmapped alignments have no CIGAR or mapping quality for unmapped alignments and no MD, NM,  CG
                 or SM tags.

                 <u>FLAGs</u> is a comma-separated list of keywords chosen from the following list.

                 unmap  The  UNMAPPED  BAM flag. This is set for reads with position &lt;= 0, reference name "*" or
                        reads starting beyond the end of the reference. Note CIGAR "*" is permitted  for  mapped
                        data so does not trigger this.

                 pos    Position  and  reference  name fields.  These may be cleared when a sequence is unmapped
                        due to the coordinates being beyond the end of the reference.  Selecting this may change
                        the sort order of the file, so it is not a part of the <b>on</b> compound argument.

                 mqual  Mapping quality.  This is set to zero for unmapped reads.

                 cigar  Modifies CIGAR fields, either by adding soft-clips for reads that overlap the end of the
                        reference or by clearing it for unmapped reads.

                 aux    For unmapped data, some auxiliary fields are meaningless and  will  be  removed.   These
                        include NM, MD, CG and SM.

                 off    Perform no sanity fixing.  This is the default

                 on     Sanitize  data  in a way that guarantees the same sort order.  This is everything except
                        for <b>pos</b>.

                 all    All sanitizing options, including <b>pos</b>.

       <b>--no-PG</b>   Do not add a @PG line to the header of the output file.

</pre><h4><b>EXAMPLES</b></h4><pre>
       o Import SAM to BAM when <b>@SQ</b> lines are present in the header:

           samtools view -bo aln.bam aln.sam

         If <b>@SQ</b> lines are absent:

           samtools faidx ref.fa
           samtools view -bt ref.fa.fai -o aln.bam aln.sam

         where <u>ref.fa.fai</u> is generated automatically by the <b>faidx</b> command.

       o Convert a BAM file to a CRAM file using a local reference sequence.

           samtools view -C -T ref.fa -o aln.cram aln.bam

       o Convert a BAM file to a CRAM with NM and MD tags stored verbatim rather than  calculating  on  the  fly
         during  CRAM  decode,  so  that mixed data sets with MD/NM only on some records, or NM calculated using
         different definitions of mismatch, can be decoded without change.  The second command demonstrates  how
         to decode such a file.  The request to not decode MD here is turning off auto-generation of both MD and
         NM; it will still emit the MD/NM tags on records that had these stored verbatim.

           samtools view -C --output-fmt-option store_md=1 --output-fmt-option store_nm=1 -o aln.cram aln.bam
           samtools view --input-fmt-option decode_md=0 -o aln.new.bam aln.cram

       o An  alternative  way  of  achieving  the above is listing multiple options after the <b>--output-fmt</b> or <b>-O</b>
         option.  The commands below are equivalent to the two above.

           samtools view -O cram,store_md=1,store_nm=1 -o aln.cram aln.bam
           samtools view --input-fmt cram,decode_md=0 -o aln.new.bam aln.cram

       o Include customized index file as a part of arguments.

           samtools view [options] -X /data_folder/data.bam /index_folder/data.bai chrM:1-10

       o Output alignments in read group <b>grp2</b> (records with no <b>RG</b> tag will also be in the output).

           samtools view -r grp2 -o /data_folder/data.rg2.bam /data_folder/data.bam

       o Only keep reads with tag <b>BC</b> and were the barcode matches the barcodes listed in the barcode file.

           samtools view -D BC:barcodes.txt -o /data_folder/data.barcodes.bam /data_folder/data.bam

       o Only keep reads with tag <b>RG</b> and read group <b>grp2</b>.  This does almost the same than <b>-r</b> <b>grp2</b> but  will  not
         keep records without the <b>RG</b> tag.

           samtools view -d RG:grp2 -o /data_folder/data.rg2_only.bam /data_folder/data.bam

       o Remove  the  actions  of  samtools  markdup.   Clear the duplicate flag and remove the <b>dt</b> tag, keep the
         header.

           samtools view -h --remove-flags DUP -x dt -o /data_folder/dat.no_dup_markings.bam /data_folder/data.bam

</pre><h4><b>AUTHOR</b></h4><pre>
       Written by Heng Li from the Sanger Institute.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/samtools.1.html">samtools</a></u>(1), <u><a href="../man1/samtools-tview.1.html">samtools-tview</a></u>(1), <u><a href="../man5/sam.5.html">sam</a></u>(5)

       Samtools website: &lt;<a href="http://www.htslib.org/">http://www.htslib.org/</a>&gt;

samtools-1.21                                   12 September 2024                               <u><a href="../man1/samtools-view.1.html">samtools-view</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>