<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>run_resfinder.py - identify acquired antimicrobial resistance genes</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/resfinder">resfinder_4.4.2-2_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       run_resfinder.py - identify acquired antimicrobial resistance genes

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: run_resfinder.py [-h] [-ifa INPUTFASTA]

       [-ifq INPUTFASTQ [INPUTFASTQ ...]] [-o OUT_PATH]
              [-b BLAST_PATH] [-k KMA_PATH] [-s SPECIES] [-db_res DB_PATH_RES] [-db_res_kma DB_PATH_RES_KMA] [-d
              DATABASES]  [-acq]  [-ao  ACQ_OVERLAP]  [-l MIN_COV] [-t THRESHOLD] [-c] [-db_point DB_PATH_POINT]
              [-db_point_kma DB_PATH_POINT_KMA] [-g SPECIFIC_GENE [SPECIFIC_GENE ...]] [-u] [-l_p MIN_COV_POINT]
              [-t_p THRESHOLD_POINT] [--pickle]

   <b>optional</b> <b>arguments:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>-ifa</b> INPUTFASTA, <b>--inputfasta</b> INPUTFASTA
              Input fasta file.

       <b>-ifq</b> INPUTFASTQ [INPUTFASTQ ...], <b>--inputfastq</b> INPUTFASTQ [INPUTFASTQ ...]
              Input fastq file(s). Assumed to be single-end fastq if only one file is provided, and  assumed  to
              be pairedend data if two files are provided.

       <b>-o</b> OUT_PATH, <b>--outputPath</b> OUT_PATH
              Path to blast output

       <b>-b</b> BLAST_PATH, <b>--blastPath</b> BLAST_PATH
              Path to blastn

       <b>-k</b> KMA_PATH, <b>--kmaPath</b> KMA_PATH
              Path to KMA

       <b>-s</b> SPECIES, <b>--species</b> SPECIES
              Species in the sample

       <b>-db_res</b> DB_PATH_RES, <b>--db_path_res</b> DB_PATH_RES
              Path to the databases for ResFinder

       <b>-db_res_kma</b> DB_PATH_RES_KMA, <b>--db_path_res_kma</b> DB_PATH_RES_KMA
              Path to the ResFinder databases indexed with KMA.  Defaults to the 'kma_indexing' directory inside
              the given database directory.

       <b>-d</b> DATABASES, <b>--databases</b> DATABASES
              Databases chosen to search in - if none is specified all is used

       <b>-acq</b>, <b>--acquired</b>
              Run resfinder for acquired resistance genes

       <b>-ao</b> ACQ_OVERLAP, <b>--acq_overlap</b> ACQ_OVERLAP
              Genes are allowed to overlap this number of nucleotides. Default: 30.

       <b>-l</b> MIN_COV, <b>--min_cov</b> MIN_COV
              Minimum (breadth-of) coverage of ResFinder

       <b>-t</b> THRESHOLD, <b>--threshold</b> THRESHOLD
              Threshold for identity of ResFinder

       <b>-c</b>, <b>--point</b>
              Run pointfinder for chromosomal mutations

       <b>-db_point</b> DB_PATH_POINT, <b>--db_path_point</b> DB_PATH_POINT
              Path to the databases for PointFinder

       <b>-db_point_kma</b> DB_PATH_POINT_KMA, <b>--db_path_point_kma</b> DB_PATH_POINT_KMA
              Path  to  the  PointFinder  databases  indexed with KMA.  Defaults to the 'kma_indexing' directory
              inside the given database directory.

       <b>-g</b> SPECIFIC_GENE [SPECIFIC_GENE ...]
              Specify genes existing in the database to search for - if none is specified all genes are included
              in the search.

       <b>-u</b>, <b>--unknown_mut</b>
              Show all mutations found even if in unknown to the resistance database

       <b>-l_p</b> MIN_COV_POINT, <b>--min_cov_point</b> MIN_COV_POINT
              Minimum (breadth-of) coverage of Pointfinder.  If  None  is  selected,  the  minimum  coverage  of
              ResFinder will be used.

       <b>-t_p</b> THRESHOLD_POINT, <b>--threshold_point</b> THRESHOLD_POINT
              Threshold for identity of Pointfinder. If None is selected, the minimum coverage of ResFinder will
              be used.

       <b>--pickle</b>
              Create  a pickle dump of the Isolate object. Currently needed in the CGE webserver. Dependency and
              this option is being removed.

</pre><h4><b>AUTHOR</b></h4><pre>
        This manpage was written by Nilesh Patra for the Debian distribution and
        can be used for any other usage of the program.

run_resfinder.py 4.1.5                              July 2021                                <u><a href="../man1/RUN_RESFINDER.PY.1.html">RUN_RESFINDER.PY</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>