<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Transdecoder - Transcriptome Protein Prediction</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/transdecoder">transdecoder_5.0.1-5_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Transdecoder - Transcriptome Protein Prediction

</pre><h4><b>USAGE</b></h4><pre>
       Required:

        -t &lt;string&gt;                            transcripts.fasta

       Common options:

        --retain_long_orfs_mode &lt;string&gt;        'dynamic' or 'strict' (default: dynamic)
                                               In dynamic mode, sets range according to 1%FDR in random sequence of same GC content.

        --retain_long_orfs_length &lt;int&gt;         under 'strict' mode, retain all ORFs found that are equal or longer than these many nucleotides even if no other evidence
                                                marks it as coding (default: 1000000) so essentially turned off by default.)

        --retain_pfam_hits &lt;string&gt;            domain table output file from running hmmscan to search Pfam (see transdecoder.github.io for info)
                                               Any ORF with a pfam domain hit will be retained in the final output.

        --retain_blastp_hits &lt;string&gt;          blastp output in '-outfmt 6' format.
                                               Any ORF with a blast match will be retained in the final output.

        --single_best_only                     Retain only the single best orf per transcript (prioritized by homology then orf length)

        -G &lt;string&gt;                            genetic code (default: universal; see PerlDoc; options: Euplotes, Tetrahymena, Candida, Acetabularia, ...)

        --no_refine_starts                     start refinement identifies potential start codons for 5' partial ORFs using a PWM, process on by default.

       Advanced options

        -T &lt;int&gt;                               If no --train, top longest ORFs to train Markov Model (hexamer stats) (default: 500)
                                               Note, 10x this value are first selected for removing redundancies,
                                               and then this -T value of longest ORFs are selected from the non-redundant set.

</pre><h4><b>Genetic</b> <b>Codes</b></h4><pre>
       See &lt;<a href="http://golgi.harvard.edu/biolinks/gencode.html">http://golgi.harvard.edu/biolinks/gencode.html</a>&gt;. These are currently supported:

        universal (default)
        Euplotes
        Tetrahymena
        Candida
        Acetabularia
        Mitochondrial-Canonical
        Mitochondrial-Vertebrates
        Mitochondrial-Arthropods
        Mitochondrial-Echinoderms
        Mitochondrial-Molluscs
        Mitochondrial-Ascidians
        Mitochondrial-Nematodes
        Mitochondrial-Platyhelminths
        Mitochondrial-Yeasts
        Mitochondrial-Euascomycetes
        Mitochondrial-Protozoans

5.0.1                                              2021-12-13                            <u><a href="../man1/TRANSDECODER.PREDICT.1.html">TRANSDECODER.PREDICT</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>