<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>bamauxmerge2 - merge information in unmapped and mapped BAM files</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/biobambam2">biobambam2_2.0.185+ds-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       bamauxmerge2 - merge information in unmapped and mapped BAM files

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>bamauxmerge2</b> [options] in_unmapped in_mapped

</pre><h4><b>DESCRIPTION</b></h4><pre>
       bamauxmerge2 reads and merges two BAM files which are expected to have the following properties

       *      the first file contains only unmapped reads and it's header contains no SQ lines

       *      the second file was produced by an aligner based on the content of the first file.

       *      both files are sorted in query name order into a single alignment file.

       The headers of the two files are merged in the following file:

       *      the  SQ  lines  contained in the header of the second file are appended to the header of the first
              file to obtain the header of the output file

       *      all other header information from the second file is discarded

       The output records are constructed in the following way:

       1.     Take a record from the second file

       2.     Copy all aux fields from the corresponding record in the first file which are not already present.

       3.     Reinsert clipped adapter bases/quality values stored in the qs/qq by aux fields by fastqtobam2 and
              remove the qs/qq aux fields while inserting appropriate soft clipping CIGAR operations.

       4.     Fix mate information like bamfixmateinformation.

       5.     Insert the mate CIGAR information fields MC and MS if the mate is aligned.

       6.     Insert the MQ (mate quality) aux field.

       The following key=value pairs can be given:

       <b>zz=&lt;0|1&gt;:</b> replace read name by content of nn aux field. Valid values are

       1:     replace read name

       0:     do not replace read name

       <b>calmdnm=&lt;0|1&gt;:</b> recompute MD and NM aux fields. Valid values are

       1:     recompute MD and NM aux fields. This requires the calmdnmreference key to be set to the name of an
              appropriate FastA file.

       0:     do not recompute MD and NM aux fields

       <b>calmdnmreference=&lt;&gt;:</b> reference FastA file.

       <b>replacecigar=&lt;0|1&gt;:</b> replace M cigar operations by the appropriate = and X operations. Valid values are

       1:     replace cigar operations. This requires the calmdnmreference key to be set on invocation.

       0:     do not replace cigar operations

       <b>hash=&lt;crc32prod&gt;:</b> hash used for computing sequence checksums. See bamseqchksum for further information.

       <b>filehash=&lt;md5&gt;:</b> hash used for computing output file checksums.

       <b>chksumfn=&lt;&gt;:</b> file name used for storing sequence checksum information. By default this information is not
       saved.

       <b>filehashfn=&lt;&gt;:</b> file name used for storing file checksum information. By default this information  is  not
       saved.

       <b>level=&lt;-1|0|1|9&gt;:</b> set compression level of the output BAM file. Valid values are

       -1:    zlib/gzip default compression level

       0:     uncompressed

       1:     zlib/gzip level 1 (fast) compression

       9:     zlib/gzip level 9 (best) compression

       <b>verbose=&lt;1&gt;:</b> Valid values are

       1:     print progress report on standard error

       0:     do not print progress report

       <b>tmpfile=&lt;filename&gt;:</b> prefix for temporary files. By default the temporary files are created in the current
       directory

       <b>md5=&lt;0|1&gt;:</b> md5 checksum creation for output file. Valid values are

       0:     do not compute checksum. This is the default.

       1:     compute  checksum.  If the md5filename key is set, then the checksum is written to the given file.
              If md5filename is unset, then no checksum will be computed.

       <b>md5filename</b> file name for md5 checksum if md5=1.

       <b>threads=&lt;[1]&gt;:</b> number of threads used for processing. By default 1 thread is used. Set to 0 for using  as
       many threads as CPU cores detected.

</pre><h4><b>AUTHOR</b></h4><pre>
       Written by German Tischler.

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       Report bugs to &lt;<a href="mailto:germant@miltenyibiotec.de">germant@miltenyibiotec.de</a>&gt;

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright  ©  2009-2019  German  Tischler,  © 2011-2013 Genome Research Limited.  License GPLv3+: GNU GPL
       version 3 &lt;<a href="http://gnu.org/licenses/gpl.html">http://gnu.org/licenses/gpl.html</a>&gt;
       This is free software: you are free to change and redistribute it.  There is NO WARRANTY, to  the  extent
       permitted by law.

BIOBAMBAM                                           July 2019                                    <u><a href="../man1/BAMAUXMERGE2.1.html">BAMAUXMERGE2</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>