<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>TFBS::PatternGen::Elph - a pattern factory that uses the Elph program</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libtfbs-perl">libtfbs-perl_0.7.1+ds-4build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       TFBS::PatternGen::Elph - a pattern factory that uses the Elph program

</pre><h4><b>SYNOPSIS</b></h4><pre>
           my $patterngen =
                   TFBS::PatternGen::Elph-&gt;new(-seq_file=&gt;'sequences.fa',
                                                -binary =&gt; '/Elph/elph'
                                                -motif_length =&gt; [8, 9, 10],
                                                -additional_params =&gt; '-x -r -e');

           my $pfm = $patterngen-&gt;pattern(); # $pfm is now a TFBS::Matrix::PFM object

</pre><h4><b>DESCRIPTION</b></h4><pre>
       TFBS::PatternGen::Gibbs builds position frequency matrices using an advanced Gibbs sampling algorithm
       implemented in external <u>Gibbs</u> program by Chip Lawrence. The algorithm can produce multiple patterns from
       a single set of sequences.

   <b>new</b>
        Title   : new
        Usage   : my $db = TFBS::PatternGen::Gibbs-&gt;new(%args);
        Function: the constructor for the TFBS::PatternGen::Gibbs object
        Returns : a TFBS::PatternGen::Gibbs object
        Args    : This method takes named arguments;
                   you must specify one of the following three
                   -seq_list     # a reference to an array of strings
                                 #   and/or Bio::Seq objects
                     # or
                   -seq_stream   # A Bio::SeqIO object
                     # or
                   -seq_file     # the name of the fasta file containing
                                 #   all the sequences
                  Other arguments are:
                   -binary       # a fully qualified path to Gibbs executable
                                 #  OPTIONAL: default 'Gibbs'
                   -nr_hits      # a presumed number of pattern occurrences in the
                                 #   sequence set: it can be a single integer, e.g.
                                 #   -nr_hits =&gt; 24 , or a reference to an array of
                                 #   integers, e.g -nr_hits =&gt; [12, 24, 36]
                   -motif_length # an expected length of motif in nucleotides:
                                 #   it can be a single integer, e.g.
                                 #   -motif_length =&gt; 8 , or a reference to an
                                 #   array ofintegers, e.g -motif_length =&gt; [8..12]
                   -additional_params  # a string containing additional
                                       #   command-line switches for the
                                       #   Gibbs program

   <b>pattern</b>
   <b>all_patterns</b>
   <b>patternSet</b>
       The three methods listed above are used for the retrieval of patterns, and are common to all
       TFBS::PatternGen::* classes. Please see TFBS::PatternGen for details.

perl v5.40.0                                       2024-10-20                        <u>TFBS::PatternGen::<a href="../man3pm/Elph.3pm.html">Elph</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>