<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>racon - consensus module for raw de novo DNA assembly of long uncorrected reads</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/racon">racon_1.5.0-3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       racon - consensus module for raw de novo DNA assembly of long uncorrected reads

</pre><h4><b>SYNPOPSIS</b></h4><pre>
       <b>racon</b> [options ...] &lt;sequences&gt; &lt;overlaps&gt; &lt;target sequences&gt;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Racon  is  intended  as  a standalone consensus module to correct raw contigs generated by rapid assembly
       methods which do not include a consensus step. The goal of Racon is to generate genomic  consensus  which
       is  of  similar  or better quality compared to the output generated by assembly methods which employ both
       error correction and consensus steps, while providing a  speedup  of  several  times  compared  to  those
       methods. It supports data produced by both Pacific Biosciences and Oxford Nanopore Technologies.

       Racon  can  be  used as a polishing tool after the assembly with either Illumina data or data produced by
       third generation of sequencing. The type of data inputed is automatically detected.

       Racon takes as input only three files: contigs in FASTA/FASTQ format, reads  in  FASTA/FASTQ  format  and
       overlaps/alignments between the reads and the contigs in MHAP/PAF/SAM format. Output is a set of polished
       contigs in FASTA format printed to stdout. All input files can be compressed with gzip.

       Racon  can also be used as a read error-correction tool. In this scenario, the MHAP/PAF/SAM file needs to
       contain pairwise overlaps between reads including dual overlaps.

       A wrapper script is also available to enable easier usage to the end- user for large datasets. It has the
       same interface as racon but adds two additional features from the outside. Sequences can be subsampled to
       decrease the total execution time (accuracy might be lower) while target  sequences  can  be  split  into
       smaller  chunks and run sequentially to decrease memory consumption. Both features can be run at the same
       time as well.

</pre><h4><b>OPTIONS</b></h4><pre>

       &lt;sequences&gt;

              input file in FASTA/FASTQ format (can be compressed  with  gzip)  containing  sequences  used  for
              correction

       &lt;overlaps&gt;

              input  file  in  MHAP/PAF/SAM  format  (can  be  compressed with gzip) containing overlaps between
              sequences and target sequences

       &lt;target sequences&gt;

              input file in FASTA/FASTQ format (can be compressed with gzip) containing sequences which will  be
              corrected

   <b>optional</b>

       <b>-u</b>, <b>--include-unpolished</b>

              output unpolished target sequences

       <b>-f</b>, <b>--fragment-correction</b>

              perform  fragment  correction  instead of contig polishing (overlaps file should contain dual/self
              overlaps!)

       <b>-w</b>, <b>--window-length</b> &lt;int&gt;

              default: 500 size of window on which POA is performed

       <b>-q</b>, <b>--quality-threshold</b> &lt;float&gt;

              default: 10.0 threshold for average base quality of windows used in POA

       <b>-e</b>, <b>--error-threshold</b> &lt;float&gt;

              default: 0.3 maximum allowed error rate used for filtering overlaps

       <b>-m</b>, <b>--match</b> &lt;int&gt;

              default: 5 score for matching bases

       <b>-x</b>, <b>--mismatch</b> &lt;int&gt;

              default: <b>-4</b> score for mismatching bases

       <b>-g</b>, <b>--gap</b> &lt;int&gt;

              default: <b>-8</b> gap penalty (must be negative)

       <b>-t</b>, <b>--threads</b> &lt;int&gt;

              default: 1 number of threads

       <b>--version</b>

              prints the version number

       <b>-h</b>, <b>--help</b>

              prints the usage

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

racon 1.3.1                                         June 2018                                           <u><a href="../man1/RACON.1.html">RACON</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>