<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>samtools-consensus - produces a consensus FASTA/FASTQ/PILEUP</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/samtools">samtools_1.21-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       samtools-consensus - produces a consensus FASTA/FASTQ/PILEUP

</pre><h4><b>SYNOPSIS</b></h4><pre>
       samtools  consensus  [<b>-saAMq</b>]  [<b>-r</b> <u>region</u>] [<b>-f</b> <u>format</u>] [<b>-l</b> <u>line-len</u>] [<b>-d</b> <u>min-depth</u>] [<b>-C</b> <u>cutoff</u>] [<b>-c</b> <u>call-</u>
       <u>fract</u>] [<b>-H</b> <u>het-fract</u>] <u>in.bam</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Generate consensus from a SAM, BAM or CRAM file based on the contents  of  the  alignment  records.   The
       consensus  is written either as FASTA, FASTQ, or a pileup oriented format.  This is selected using the <b>-f</b>
       <u>FORMAT</u> option.

       The default output for FASTA and FASTQ formats include one base per non-gap consensus.  Hence  insertions
       with  respect  to  the  aligned  reference will be included and deletions removed.  This behaviour can be
       controlled with the <b>--show-ins</b> and <b>--show-del</b> options.  This could be used to  compute  a  new  reference
       from sequences assemblies to realign against.

       The  pileup-style  format  strictly adheres to one row per consensus location, differing from the one row
       per reference based used in the related "samtools mpileup" command.  This means the base  quality  values
       for  inserted  columns  are  reported.   The  base  quality  value of gaps (either within an insertion or
       otherwise) are determined as the average of the surrounding non-gap bases.  The  columns  shown  are  the
       reference name, position, nth base at that position (zero if not an insertion), consensus call, consensus
       confidence, sequences and quality values.

       Two  consensus  calling  algorithms  are  offered.   The  default  computes a heterozygous consensus in a
       Bayesian manner, derived from the "Gap5" consensus algorithm.  Quality values are also  tweaked  to  take
       into account other nearby low quality values.  This can also be disabled, using the <b>--no-adj-qual</b> option.

       This  method  also  utilises  the  mapping  qualities,  unless  the  <b>--no-use-MQ</b> option is used.  Mapping
       qualities are also auto-scaled to take into account the local reference variation by processing the  MD:Z
       tag,  unless  <b>--no-adj-MQ</b>  is  used.   Mapping  qualities  can be capped between a minimum (<b>--low-MQ</b>) and
       maximum (<b>--high-MQ</b>), although the defaults are liberal and trust the data to be true.  Finally an overall
       scale on the resulting mapping quality can be supplied (<b>--scale-MQ</b>, defaulting to  1.0).   This  has  the
       effect  of favouring more calls with a higher false positive rate (values greater than 1.0) or being more
       cautious with higher false negative rates and lower false positive (values less than 1.0).

       The second method is a simple frequency counting algorithm, summing either +1 for each base type or +<u>qual</u>
       if the <b>--use-qual</b> option is specified.  This is enabled with the <b>--mode</b> <b>simple</b> option.

       The summed share of a specific base type is then compared against the total possible and if this is above
       the <b>--call-fract</b> <u>fraction</u> parameter then the most likely base type is called, or "N" otherwise (or absent
       if it is a gap).  The <b>--ambig</b> option permits generation of ambiguity codes instead of "N",  provided  the
       minimum  fraction  of  the  second  most  common  base  type  to the most common is above the <b>--het-fract</b>
       <u>fraction</u><b>.</b>

</pre><h4><b>OPTIONS</b></h4><pre>
       General options that apply to both algorithms:

       <b>-r</b> <u>REG</u><b>,</b> <b>--region</b> <u>REG</u>
                 Limit the query to region <u>REG</u>.  This requires an index.

       <b>-f</b> <u>FMT</u><b>,</b> <b>--format</b> <u>FMT</u>
                 Produce format <u>FMT</u>, with "fastq", "fasta" and "pileup" as permitted options.

       <b>-l</b> <u>N</u><b>,</b> <b>--line-len</b> <u>N</u>
                 Sets the maximum line length of line-wrapped fasta and fastq formats to <u>N</u>.

       <b>-o</b> <u>FILE</u><b>,</b> <b>--output</b> <u>FILE</u>
                 Output consensus to FILE instead of stdout.

       <b>-m</b> <u>STR</u><b>,</b> <b>--mode</b> <u>STR</u>
                 Select the consensus algorithm.  Valid modes are "simple" frequency counting and the "bayesian"
                 (Gap5) methods, with Bayesian being the default.  (Note case does not matter, so "Bayesian"  is
                 accepted  too.)   There are a variety of bayesian methods.  Straight "bayesian" is the best set
                 suitable for the other parameters selected.  The  choice  of  internal  parameters  may  change
                 depending  on  the "--P-indel" score.  This method distinguishes between substitution and indel
                 error rates.  The old Samtools consensus in version 1.16 did not distinguish types  of  errors,
                 but for compatibility the "bayesian_116" mode may be selected to replicate this.

       <b>-a</b>        Outputs  all  bases, from start to end of reference, even when the aligned data does not extend
                 to the ends.  This is most useful for construction of a full length reference sequence.

       <b>-a</b> <b>-a,</b> <b>-aa</b>
                 Output absolutely all positions, including references with no data aligned against them.

       <b>--rf</b>, <b>--incl-flags</b> <u>STR</u>|<u>INT</u>
                 Only include reads with at least one FLAG bit set.  Defaults to zero, which filters no reads.

       <b>--ff</b>, <b>--excl-flags</b> <u>STR</u>|<u>INT</u>
                 Exclude reads with any FLAG bit set.  Defaults to "UNMAP,SECONDARY,QCFAIL,DUP".

       <b>--min-MQ</b> <u>INT</u>
                 Filters out reads with a mapping quality below <u>INT</u>.  This defaults to zero.

       <b>--min-BQ</b> <u>INT</u>
                 Filters out bases with a base quality below <u>INT</u>.  This defaults to zero.

       <b>--show-del</b> <u>yes</u><b>/</b><u>no</u>
                 Whether to show deletions as "*" (yes) or to omit from the output (no).  Defaults to no.

       <b>--show-ins</b> <u>yes</u><b>/</b><u>no</u>
                 Whether to show insertions in the consensus.  Defaults to yes.

       <b>--mark-ins</b>
                 Insertions, when shown, are normally recorded in the consensus with plain 7-bit ASCII (ACGT, or
                 acgt if heterozygous).  However this makes  it  impossible  to  identify  the  mapping  between
                 consensus coordinates and the original reference coordinates.  If fasta output is selected then
                 the option adds an underscore before every inserted base, plus a corresponding character in the
                 quality  for  fastq  format.   When used in conjunction with <b>-a</b> <b>--show-del</b> <b>yes</b>, this permits an
                 easy derivation of the consensus to reference coordinate mapping.

       <b>-A</b>, <b>--ambig</b>
                 Enables IUPAC ambiguity codes in the consensus output.  Without this the output will be limited
                 to A, C, G, T, N and *.

       <b>-d</b> <u>D</u><b>,</b> <b>--min-depth</b> <u>D</u>
                 The minimum depth required to make a call.  Defaults to  1.   Failing  this  depth  check  will
                 produce  consensus  "N",  or  absent if it is an insertion.  Note this check is performed after
                 filtering by flags and mapping/base quality.

       The following options apply only to the simple consensus mode:

       <b>-q</b>, <b>--use-qual</b>
                 For the simple consensus algorithm, this enables  use  of  base  quality  values.   Instead  of
                 summing  1  per base called, it sums the base quality instead.  These sums are also used in the
                 <b>--call-fract</b> and <b>--het-fract</b> parameters too.  Quality values are always  used  for  the  "Gap5"
                 consensus  method  and  this  option has no affect.  Note currently  quality values only affect
                 SNPs and not inserted sequences, which  still  get  scores  with  a  fixed  +1  per  base  type
                 occurrence.

       <b>-H</b> <u>H</u><b>,</b> <b>--het-fract</b> <u>H</u>
                 For  consensus  columns  containing multiple base types, if the second most frequent type is at
                 least <u>H</u> fraction of the most common type then a heterozygous base type will be reported in  the
                 consensus.   Otherwise  the  most  common  base  is  used,  provided  it meets the <u>--call-fract</u>
                 parameter (otherwise "N").  The fractions computed may be modified by the use of quality values
                 if the <b>-q</b> option is enabled.  Note although IUPAC has ambiguity codes for A,C,G,T vs any  other
                 A,C,G,T  it does not have codes for A,C,G,T vs gap (such as in a heterozygous deletion).  Given
                 the lack of any official code, we use lower-case letter to symbolise a half-present base type.

       <b>-c</b> <u>C</u><b>,</b> <b>--call-fract</b> <u>C</u>
                 Only used for the simple consensus algorithm.  Require at least <u>C</u> fraction  of  bases  agreeing
                 with  the  most  likely consensus call to emit that base type.  This defaults to 0.75.  Failing
                 this check will output "N".

       The following options apply only to Bayesian consensus mode enabled
       (default on).

       <b>-C</b> <u>C</u><b>,</b> <b>--cutoff</b> <u>C</u>
                 Only used for the Gap5 consensus mode, which  produces  a  Phred  style  score  for  the  final
                 consensus quality.  If this is below <u>C</u> then the consensus is called as "N".

       <b>--use-MQ</b>, <b>--no-use-MQ</b>
                 Enable or disable the use of mapping qualities.  Defaults to on.

       <b>--adj-MQ</b>, <b>--no-adj-MQ</b>
                 If  mapping  qualities  are  used, this controls whether they are scaled by the local number of
                 mismatches to the reference.  The reference is unknown by this tool, so this data  is  obtained
                 from the MD:Z auxiliary tag (or ignored if not present).  Defaults to on.

       <b>--NM-halo</b> <u>INT</u>
                 Specifies  the  distance either side of the base call being considered for computing the number
                 of local mismatches.

       <b>--low-MQ</b> <u>MIN</u>, <b>--high-MQ</b> <u>MAX</u>
                 Specifies a minimum and maximum value of the  mapping  quality.   These  are  not  filters  and
                 instead simply put upper and lower caps on the values.  The defaults are 0 and 60.

       <b>--scale-MQ</b> <u>FLOAT</u>
                 This  is  a  general multiplicative  mapping quality scaling factor.  The effect is to globally
                 raise or lower the quality values used in the consensus  algorithm.   Defaults  to  1.0,  which
                 leaves the values unchanged.

       <b>--P-het</b> <u>FLOAT</u>
                 Controls  the likelihood of any position being a heterozygous site.  This is used in the priors
                 for the Bayesian calculations, and has little difference  on  deep  data.   Defaults  to  1e-3.
                 Smaller  numbers  makes the algorithm more likely to call a pure base type.  Note the algorithm
                 will always compute the probability of the base being homozygous vs heterozygous,  irrespective
                 of  whether  the  output  is reported as ambiguous (it will be "N" if deemed to be heterozygous
                 without <b>--ambig</b> mode enabled).

       <b>--P-indel</b> <u>FLOAT</u>
                 Controls the likelihood of small  indels.   This  is  used  in  the  priors  for  the  Bayesian
                 calculations, and has little difference on deep data.  Defaults to 2e-4.

       <b>--het-scale</b> <u>FLOAT</u>
                 This  is a multiplicative correction applied per base quality before adding to the heterozygous
                 hypotheses.  Reducing it means  fewer  heterozygous  calls  are  made.   This  oftens  leads  a
                 significant  reduction  in  false  positive  het  calls,  for  some increase in false negatives
                 (mislabelling real heterozygous sites as homozygous).  It is usually beneficial to reduce  this
                 on  instruments  where a significant proportion of bases may be aligned in the wrong column due
                 to insertions and deletions leading  to  alignment  errors  and  reference  bias.   It  can  be
                 considered as a het sensitivity tuning parameter.  Defaults to 1.0 (nop).

       <b>-p</b>, <b>--homopoly-fix</b>
                 Some  technologies  that call runs of the same base type together always put the lowest quality
                 calls at one end.  This can cause problems when reverse complementing and comparing  alignments
                 with  indels.   This  option  averages  the qualities at both ends to avoid orientation biases.
                 Recommended for old 454 or PacBio HiFi data sets.

       <b>--homopoly-score</b> <u>FLOAT</u>
                 The <b>-p</b> option  also  reduces  confidence  values  within  homopolymers  due  to  an  additional
                 likelihood  of  sequence  specific  errors.   The quality values are multiplied by <u>FLOAT</u>.  This
                 defaults to 0.5, but is  not  used  if  <b>-p</b>  was  not  specified.   Adjusting  this  score  also
                 automatically enables <b>-p</b>.

       <b>-t</b>, <b>--qual-calibration</b> <u>FILE</u>
                 Loads  a quality calibration table from <u>FILE</u>.  The format of this is a series of lines with the
                 following fields, each starting with the literal text "QUAL":

                     <b>QUAL</b> <u>value</u> <u>substitution</u> <u>undercall</u> <u>overcall</u>

                 Lines starting with a "#" are ignored.  Each line maps a recorded quality value  to  the  Phred
                 equivalent  score for substitution, undercall and overcall errors.  Quality <u>value</u>s are expected
                 to be sorted in increasing numerical order, but may skip values.   This  allows  the  consensus
                 algorithm  to know the most likely cause of an error, and whether the instrument is more likely
                 to have indel errors (more common in some long read technologies) or substitution errors  (more
                 common in clocked short-read instruments).

                 Some  pre-defined  calibration  tables  are built in.  These are specified with a fake filename
                 starting with a colon.  See the <b>-X</b> option for more details.

                 Note due to the additional heuristics applied by the consensus algorithm,  these  recalibration
                 tables are not a true reflection of the instrument error rates and are a work in progress.

       <b>-X</b>, <b>--config</b> <u>STR</u>
                 Specifies  predefined  sets of configuration parameters.  Acceptable values for <u>STR</u> are defined
                 below, along with the list of parameters they are equivalent to.

                 <b>hiseq</b>     --qual-calibration :hiseq

                 <b>hifi</b>      --qual-calibration :hifi --homopoly-fix 0.3 --low-MQ  5  --scale-MQ  1.5  --het-scale
                           0.37

                 <b>r10.4_sup</b> --qual-calibration  :r10.4_sup  --homopoly-fix  0.3  --low-MQ 5 --scale-MQ 1.5 --het-
                           scale 0.37

                 <b>r10.4_dup</b> --qual-calibration :r10.4_dup --homopoly-fix 0.3 --low-MQ  5  --scale-MQ  1.5  --het-
                           scale 0.37

                 <b>ultima</b>    --qual-calibration  :ultima  --homopoly-fix  0.3 --low-MQ 10 --scale-MQ 2 --het-scale
                           0.37

</pre><h4><b>EXAMPLES</b></h4><pre>
       -      Create a modified FASTA reference that has a  1:1  coordinate  correspondence  with  the  original
              reference used in alignment.

                samtools consensus -a --show-ins no in.bam -o ref.fa

       -      Create a FASTQ file for the contigs with aligned data, including insertions.

                samtools consensus -f fastq in.bam -o cons.fq

</pre><h4><b>AUTHOR</b></h4><pre>
       Written by James Bonfield from the Sanger Institute.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/samtools.1.html">samtools</a></u>(1), <u><a href="../man1/samtools-mpileup.1.html">samtools-mpileup</a></u>(1),

       Samtools website: &lt;<a href="http://www.htslib.org/">http://www.htslib.org/</a>&gt;

samtools-1.21                                   12 September 2024                          <u><a href="../man1/samtools-consensus.1.html">samtools-consensus</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>