<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Variation::VariantI - Sequence Change SeqFeature abstract class</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-variation-perl">libbio-variation-perl_1.7.5-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Variation::VariantI - Sequence Change SeqFeature abstract class

</pre><h4><b>SYNOPSIS</b></h4><pre>
         #get Bio::Variant::VariantI somehow
         print $var-&gt;restriction_changes, "\n";
         foreach $allele ($var-&gt;each_Allele) {
             #work on Bio::Variation::Allele objects
         }

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This superclass defines common methods to basic sequence changes.  The instantiable classes
       Bio::Variation::DNAMutation, Bio::Variation::RNAChange and Bio::Variation::AAChange use them.  See
       Bio::Variation::DNAMutation, Bio::Variation::RNAChange, and Bio::Variation::AAChange for more
       information.

       These classes store information, heavy computation to determine allele sequences is done elsewhere.

       The database cross-references are implemented as Bio::Annotation::DBLink objects. The methods to access
       them are defined in Bio::DBLinkContainerI. See Bio::Annotation::DBLink and Bio::DBLinkContainerI for
       details.

       Bio::Variation::VariantI redifines and extends Bio::SeqFeature::Generic for sequence variations. This
       class describes specific sequence change events. These events are always from a specific reference
       sequence to something different. See Bio::SeqFeature::Generic for more information.

       IMPORTANT: The notion of reference sequence permeates all Bio::Variation classes. This is especially
       important to remember when dealing with Alleles. In a polymorphic site, there can be a large number of
       alleles. One of then has to be selected to be the reference allele (allele_ori). ALL the rest has to be
       passed to the Variant using the method add_Allele, including the mutated allele in a canonical mutation.
       The IO modules and generated attributes depend on it. They ignore the allele linked to using allele_mut
       and circulate each Allele returned by each_Allele into allele_mut and calculate the changes between that
       and allele_ori.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing lists  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Heikki</b> <b>Lehvaslaiho</b></h4><pre>
       Email:  heikki-at-bioperl-dot-org

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>id</b>
        Title   : id
        Usage   : $obj-&gt;id
        Function:

                  Read only method. Returns the id of the variation object.
                  The id is the id of the first DBLink object attached to this object.

        Example :
        Returns : scalar
        Args    : none

   <b>add_Allele</b>
        Title   : add_Allele
        Usage   : $self-&gt;add_Allele($allele)
        Function:

                   Adds one Bio::Variation::Allele into the list of alleles.
                   Note that the method forces the convention that nucleotide
                   sequence is in lower case and amino acds are in upper
                   case.

        Example :
        Returns : 1 when succeeds, 0 for failure.
        Args    : Allele object

   <b>each_Allele</b>
        Title   : alleles
        Usage   : $obj-&gt;each_Allele();
        Function:

                    Returns a list of Bio::Variation::Allele objects

        Example :
        Returns : list of Alleles
        Args    : none

   <b>isMutation</b>
        Title   : isMutation
        Usage   : print join('/', $obj-&gt;each_Allele) if not $obj-&gt;isMutation;
        Function:

                  Returns or sets the boolean value indicating that the
                  variant described is a canonical mutation with two alleles
                  assinged to be the original (wild type) allele and mutated
                  allele, respectively. If this value is not set, it is
                  assumed that the Variant describes polymorphisms.

        Returns : a boolean

   <b>allele_ori</b>
        Title   : allele_ori
        Usage   : $obj-&gt;allele_ori();
        Function:

                   Links to and returns the Bio::Variation::Allele object.
                   If value is not set, returns false. All other Alleles are
                   compared to this.

                   Amino acid sequences are stored in upper case characters,
                   others in lower case.

        Example :
        Returns : string
        Args    : string

       See Bio::Variation::Allele for more.

   <b>allele_mut</b>
        Title   : allele_mut
        Usage   : $obj-&gt;allele_mut();
        Function:

                    Links to and returns the Bio::Variation::Allele
                    object.  Sets and returns the mutated allele sequence.
                    If value is not set, returns false.

                    Amino acid sequences are stored in upper case characters,
                    others in lower case.

        Example :
        Returns : string
        Args    : string

       See Bio::Variation::Allele for more.

   <b>length</b>
        Title   : length
        Usage   : $obj-&gt;length();
        Function:

                   Sets and returns the length of the affected original
                   allele sequence.  If value is not set, returns false == 0.

                   Value 0 means that the variant position is before the
                   start=end sequence position. (Value 1 would denote a point
                   mutation). This follows the convension to report an
                   insertion (2insT) in equivalent way to a corresponding
                   deletion (2delT) (Think about indel polymorpism ATC &lt;=&gt; AC
                   where the origianal state is not known ).

        Example :
        Returns : string
        Args    : string

   <b>upStreamSeq</b>
        Title   : upStreamSeq
        Usage   : $obj-&gt;upStreamSeq();
        Function:

                   Sets and returns upstream flanking sequence string.  If
                   value is not set, returns false. The sequence should be
                   &gt;=25 characters long, if possible.

        Example :
        Returns : string or false
        Args    : string

   <b>dnStreamSeq</b>
        Title   : dnStreamSeq
        Usage   : $obj-&gt;dnStreamSeq();
        Function:

                   Sets and returns dnstream flanking sequence string.  If
                   value is not set, returns false. The sequence should be
                   &gt;=25 characters long, if possible.

        Example :
        Returns : string or false
        Args    : string

   <b>label</b>
        Title   : label
        Usage   : $obj-&gt;label();
        Function:

                   Sets and returns mutation event label(s).  If value is not
                   set, or no argument is given returns false.  Each
                   instantiable class needs to implement this method. Valid
                   values are listed in 'Mutation event controlled vocabulary' in
                   <a href="http://www.ebi.ac.uk/mutations/recommendations/mutevent.html">http://www.ebi.ac.uk/mutations/recommendations/mutevent.html</a>.

        Example :
        Returns : string
        Args    : string

   <b>status</b>
        Title   : status
        Usage   : $obj-&gt;status()
        Function:

                  Returns the status of the sequence change object.
                  Valid values are: 'suspected' and 'proven'

        Example : $obj-&gt;status('proven');
        Returns : scalar
        Args    : valid string (optional, for setting)

   <b>proof</b>
        Title   : proof
        Usage   : $obj-&gt;proof()
        Function:

                  Returns the proof of the sequence change object.
                  Valid values are: 'computed' and 'experimental'.

        Example : $obj-&gt;proof('computed');
        Returns : scalar
        Args    : valid string (optional, for setting)

   <b>region</b>
        Title   : region
        Usage   : $obj-&gt;region();
        Function:

                   Sets and returns the name of the sequence region type or
                   protein domain at this location.  If value is not set,
                   returns false.

        Example :
        Returns : string
        Args    : string

   <b>region_value</b>
        Title   : region_value
        Usage   : $obj-&gt;region_value();
        Function:

                   Sets and returns the name of the sequence region_value or
                   protein domain at this location.  If value is not set,
                   returns false.

        Example :
        Returns : string
        Args    : string

   <b>region_dist</b>
        Title   : region_dist
        Usage   : $obj-&gt;region_dist();
        Function:

                   Sets and returns the distance tot the closest region
                   (i.e. intro/exon or domain) boundary. If distance is not
                   set, returns false.

        Example :
        Returns : integer
        Args    : integer

   <b>numbering</b>
        Title   : numbering
        Usage   : $obj-&gt;numbering()
        Function:

                  Returns the numbering chema used locating sequnce features.
                  Valid values are: 'entry' and 'coding'

        Example : $obj-&gt;numbering('coding');
        Returns : scalar
        Args    : valid string (optional, for setting)

   <b>mut_number</b>
        Title   : mut_number
        Usage   : $num = $obj-&gt;mut_number;
                : $num = $obj-&gt;mut_number($number);
        Function:

                  Returns or sets the number identifying the order in which the
                  mutation has been issued. Numbers shouldstart from 1.
                  If the number has never been set, the method will return ''

                  If you want the output from IO modules look nice and, for
                  multivariant/allele variations, make sense you better set
                  this attribute.

        Returns : an integer

   <b>SeqDiff</b>
        Title   : SeqDiff
        Usage   : $mutobj = $obj-&gt;SeqDiff;
                : $mutobj = $obj-&gt;SeqDiff($objref);
        Function:

                  Returns or sets the link-reference to the umbrella
                  Bio::Variation::SeqDiff object.  If there is no link,
                  it will return undef

                  Note: Adding a variant into a SeqDiff object will
                  automatically set this value.

        Returns : an obj_ref or undef

       See Bio::Variation::SeqDiff for more information.

   <b>add_DBLink</b>
        Title   : add_DBLink
        Usage   : $self-&gt;add_DBLink($ref)
        Function: adds a link object
        Example :
        Returns :
        Args    :

   <b>each_DBLink</b>
        Title   : each_DBLink
        Usage   : foreach $ref ( $self-&gt;each_DBlink() )
        Function: gets an array of DBlink of objects
        Example :
        Returns :
        Args    :

   <b>restriction_changes</b>
        Title   : restriction_changes
        Usage   : $obj-&gt;restriction_changes();
        Function:

                   Returns a string containing a list of restriction
                   enzyme changes of form +EcoRI, separated by
                   commas. Strings need to be valid restriction enzyme names
                   as stored in REBASE. allele_ori and allele_mut need to be assigned.

        Example :
        Returns : string
        Args    : string

perl v5.32.1                                       2021-12-02                      <u>Bio::Variation::<a href="../man3pm/VariantI.3pm.html">VariantI</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>