<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::SeqIO::game::gameWriter -- a class for writing game-XML</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::SeqIO::game::gameWriter -- a class for writing game-XML

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::SeqIO;

         my $in  = Bio::SeqIO-&gt;new( -format =&gt; 'genbank',
                                    -file =&gt; 'myfile.gbk' );
         my $out = Bio::SeqIO-&gt;new( -format =&gt; 'game',
                                    -file =&gt; 'myfile.xml' );

         # get a sequence object
         my $seq = $in-&gt;next_seq;

         #write it in GAME format
         $out-&gt;write_seq($seq);

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::SeqIO::game::gameWriter writes GAME-XML (v. 1.2) that is readable by Apollo.  It is best not used
       directly.  It is accessed via Bio::SeqIO.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.

       Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Sheldon</b> <b>McKay</b></h4><pre>
       Email <a href="mailto:mckays@cshl.edu">mckays@cshl.edu</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : my $writer = Bio::SeqIO::game::gameWriter-&gt;new($seq);
        Function: constructor method for gameWriter
        Returns : a game writer object
        Args    : a Bio::SeqI implementing object
                  optionally, an argument to set map_position to on.
                  ( map =&gt; 1 ).  This will create a map_position elemant
                  that will cause the feature coordinates to be remapped to
                  a parent seqeunce.  A sequence name in the format seq:xxx-xxx
                  is expected to determine the offset for the map_position.
                  The default behavior is to have features mapped relative to
                  the sequence contained in the GAME-XML file

   <b>write_to_game</b>
        Title   : write_to_game
        Usage   : $writer-&gt;write_to_game
        Function: writes the sequence object to game-XML
        Returns : xml as a multiline string
        Args    : none

   <b>_rearrange_hierarchies</b>
        Title   : _rearrange_hierarchies
        Usage   : $self-&gt;_rearrange_hierarchies($seq)
        Function: internal method to rearrange gene containment hierarchies
                  so that snRNA or transposon features contain their genes
                  rather than the other way around
        Returns : nothing
        Args    : a Bio::RichSeq object
        Note    : Not currently used, may be removed

   <b>_write_feature</b>
        Title   : _write_feature
        Usage   : $seld-&gt;_write_feature($feat, 1)
        Function: internal method for writing generic features as &lt;annotation&gt; elements
        Returns : nothing
        Args    : a Bio::SeqFeature::Generic object and an optional flag to write a
                  bare feature set with no annotation wrapper

   <b>_write_gene</b>
        Title   : _write_gene
        Usage   : $self-&gt;_write_gene($feature)
        Function: internal method for rendering gene containment hierarchies into
                  a nested &lt;annotation&gt; element
        Returns : nothing
        Args    : a nested Bio::SeqFeature::Generic gene feature
        Note    : A nested gene hierarchy (gene-&gt;mRNA-&gt;CDS-&gt;exon) is expected.  If other gene
                  subfeatures occur as level one subfeatures (same level as mRNA subfeats)
                  an attempt will be made to link them to transcripts via the 'standard_name'
                  qualifier

   <b>_check_cds</b>
        Title   : _check_cds
        Usage   : $self-&gt;_check_cds($cds, $name)
        Function: internal method to check if the CDS associated with an mRNA is
                  the correct alternative splice variant
        Returns : a Bio::SeqFeature::Generic CDS object
        Args    : the CDS object plus the transcript\'s 'standard_name'
        Note    : this method only works if alternatively spliced transcripts are bound
                  together by a 'standard_name' or 'mRNA' qualifier.  If none is present,
                  we will hope that the exons were derived from a segmented RNA or a CDS
                  with no associated mRNA feature.  Neither of these two cases would be
                  confounded by alternative splice variants.

   <b>_comp_analysis</b>
         Usage:
         Desc :
         Ret  :
         Args :
         Side Effects:
         Example:

   <b>_comp_result</b>
         Usage:
         Desc : recursively render a feature and its subfeatures as
                &lt;result_set&gt; and &lt;result_span&gt; elements
         Ret  : nothing meaningful
         Args : a feature

   <b>_comp_result_span</b>
         Usage: _comp_result_span('foo12',$feature);
         Desc : write GAME XML for a Bio::SeqFeature::Computation feature
                that has no subfeatures
         Ret  : nothing meaningful
         Args : name for this span (some kind of identifier),
                SeqFeature object to put into this span
         Side Effects:
         Example:

   <b>_render_tags</b>
         Usage:
         Desc :
         Ret  :
         Args :
         Side Effects:
         Example:

   <b>_render_output_tags</b>
         Usage:
         Desc : print out &lt;output&gt; elements, with contents
                taken from the SeqFeature::Computation's 'output' tag
         Ret  : array of tag names this did not render
         Args : feature object, list of tag names to maybe render

         In game xml, only &lt;result_span&gt; and &lt;result_set&gt; elements can
         have &lt;output&gt; elements.

   <b>_render_tags_as_properties</b>
         Usage:
         Desc :
         Ret  : empty array
         Args : feature object, array of tag names
         Side Effects:
         Example:

         In game xml, &lt;annotation&gt;, &lt;computational_analysis&gt;,
         and &lt;feature_set&gt; elements can have properties.

   <b>_render_comment_tags</b>
         Usage:
         Desc :
         Ret  : names of tags that were not comment tags
         Args : feature object, tag names available for us to render
         Side Effects: writes XML
         Example:

         In game xml, &lt;annotation&gt; and &lt;feature_set&gt; elements can
         have comments.

   <b>_render_date_tags</b>
         Usage:
         Desc :
         Ret  : names of tags that were not date tags
         Args : feature, list of tag names available for us to render
         Side Effects: writes XML for &lt;date&gt; elements
         Example:

         In game xml, &lt;annotation&gt;, &lt;computational_analysis&gt;,
         &lt;transaction&gt;, &lt;comment&gt;, and &lt;feature_set&gt; elements
         can have &lt;date&gt;s.

   <b>_render_dbxref_tags</b>
         Desc : look for xref tags and render them if they are there
         Ret  : tag names that we didn't render
         Args : feature object, list of tag names to render
         Side Effects: writes a &lt;dbxref&gt; element if a tag with name
                       matching /xref$/i is present

         In game xml, &lt;annotation&gt; and &lt;seq&gt; elements can have dbxrefs.

   <b>_render_target_tags</b>
         Usage:
         Desc : process any 'Target' tags that would indicate a sequence alignment subject
         Ret  : array of tag names that we didn't render
         Args : feature object
         Side Effects: writes a &lt;seq_relationship&gt; of type 'subject' if it finds
                       any properly formed tags named 'Target'
         Example:

         In game xml, &lt;result_span&gt;, &lt;feature_span&gt;, and &lt;result_set&gt; can have
         &lt;seq_relationship&gt;s.  &lt;result_set&gt; can only have one, a 'query' relation.

   <b>_property</b>
        Title   : _property
        Usage   : $self-&gt;_property($tag =&gt; $value);
        Function: an internal method to write property XML elements
        Returns : nothing
        Args    : a tag/value pair

   <b>_unflatten_attribute</b>
        Title   : _unflatten_attribute
        Usage   : $self-&gt;_unflatten_attribute($name, $value)
        Function: an internal method to unflatten and write comment or evidence elements
        Returns : nothing
        Args    : a list of strings

   <b>_xref</b>
        Title   : _xref
        Usage   : $self-&gt;_xref($value)
        Function: an internal method to write db_xref elements
        Returns : nothing
        Args    : a list of strings

   <b>_feature_span</b>
        Title   : _feature_span
        Usage   : $self-&gt;_feature_span($name, $type, $loc)
        Function: an internal method to write a feature_span element
                 (the actual feature with coordinates)
        Returns : nothing
        Args    : a feature name and Bio::SeqFeatureI-compliant object

   <b>_seq_relationship</b>
        Title   : _seq_relationship
        Usage   : $self-&gt;_seq_relationship($type, $loc)
        Function: an internal method to handle feature_span sequence relationships
        Returns : nothing
        Args    : feature type, a Bio::LocationI-compliant object,
                  (optional) sequence name (defaults to the query seq)
                  and (optional) alignment string

   <b>_element</b>
        Title   : _element
        Usage   : $self-&gt;_element($name, $chars, $atts)
        Function: an internal method to generate 'generic' XML elements
        Example :
        my $name = 'foo';
        my $content = 'bar';
        my $attributes = { baz =&gt; 1 };
        # print the element
        $self-&gt;_element($name, $content, $attributes);
        Returns : nothing
        Args    : the element name and content plus a ref to an attribute hash

   <b>_span</b>
        Title   : _span
        Usage   : $self-&gt;_span($loc)
        Function: an internal method to write the 'span' element
        Returns : nothing
        Args    : a Bio::LocationI-compliant object

   <b>_seq</b>
        Title   : _seq
        Usage   : $self-&gt;_seq($seq, $dna)
        Function: an internal method to print the 'sequence' element
        Returns : nothing
        Args    : and Bio::SeqI-compliant object and a reference to an attribute  hash

   <b>_find_name</b>
        Title   : _find_name
        Usage   : my $name = $self-&gt;_find_name($feature)
        Function: an internal method to look for a gene name
        Returns : a string
        Args    : a Bio::SeqFeatureI-compliant object

perl v5.32.1                                       2021-08-15                  <u>Bio::SeqIO::game::<a href="../man3pm/gameWriter.3pm.html">gameWriter</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>