<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::AlignIO - Handler for AlignIO Formats</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::AlignIO - Handler for AlignIO Formats

</pre><h4><b>SYNOPSIS</b></h4><pre>
           use Bio::AlignIO;

           $inputfilename = "testaln.fasta";
           $in  = Bio::AlignIO-&gt;new(-file   =&gt; $inputfilename ,
                                    -format =&gt; 'fasta');
           $out = Bio::AlignIO-&gt;new(-file   =&gt; "&gt;out.aln.pfam" ,
                                    -format =&gt; 'pfam');

           while ( my $aln = $in-&gt;next_aln() ) {
               $out-&gt;write_aln($aln);
           }

           # OR

           use Bio::AlignIO;

           open MYIN, '&lt;', 'testaln.fasta' or die "Could not read file 'testaln.fasta': $!\n";
           $in  = Bio::AlignIO-&gt;newFh(-fh     =&gt; \*MYIN,
                                      -format =&gt; 'fasta');
           open my $MYOUT, '&gt;', 'testaln.pfam' or die "Could not write file 'testaln.pfam': $!\n";
           $out = Bio::AlignIO-&gt;newFh(-fh     =&gt;  $MYOUT,
                                      -format =&gt; 'pfam');

           # World's smallest Fasta&lt;-&gt;pfam format converter:
           print $out $_ while &lt;$in&gt;;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::AlignIO is a handler module for the formats in the AlignIO set, for example, Bio::AlignIO::fasta. It
       is the officially sanctioned way of getting at the alignment objects. The resulting alignment is a
       Bio::Align::AlignI-compliant object.

       The idea is that you request an object for a particular format.  All the objects have a notion of an
       internal file that is read from or written to. A particular AlignIO object instance is configured for
       either input or output, you can think of it as a stream object.

       Each object has functions:

          $stream-&gt;next_aln();

       And:

          $stream-&gt;write_aln($aln);

       Also:

          $stream-&gt;type() # returns 'INPUT' or 'OUTPUT'

       As an added bonus, you can recover a filehandle that is tied to the AlignIO object, allowing you to use
       the standard &lt;&gt; and print operations to read and write alignment objects:

           use Bio::AlignIO;

           # read from standard input
           $stream = Bio::AlignIO-&gt;newFh(-format =&gt; 'Fasta');

           while ( $aln = &lt;$stream&gt; ) {
                    # do something with $aln
           }

       And:

           print $stream $aln; # when stream is in output mode

       Bio::AlignIO is patterned on the Bio::SeqIO module and shares most of its features.  One significant
       difference is that Bio::AlignIO usually handles IO for only a single alignment at a time, whereas
       Bio::SeqIO handles IO for multiple sequences in a single stream.  The principal reason for this is that
       whereas simultaneously handling multiple sequences is a common requirement, simultaneous handling of
       multiple alignments is not. The only current exception is format "bl2seq" which parses results of the
       BLAST "bl2seq" program and which may produce several alignment pairs.  This set of alignment pairs can be
       read using multiple calls to next_aln.

</pre><h4><b>CONSTRUCTORS</b></h4><pre>
   <b>Bio::AlignIO-&gt;new()</b>
          $seqIO = Bio::AlignIO-&gt;new(-file =&gt; 'filename',   -format=&gt;$format);
          $seqIO = Bio::AlignIO-&gt;new(-fh   =&gt; \*FILEHANDLE, -format=&gt;$format);
          $seqIO = Bio::AlignIO-&gt;new(-format =&gt; $format);
          $seqIO = Bio::AlignIO-&gt;new(-fh =&gt; \*STDOUT, -format =&gt; $format);

       The new class method constructs a new Bio::AlignIO object.  The returned object can be used to retrieve
       or print alignment objects. new accepts the following parameters:

       -file
           A file path to be opened for reading or writing.  The usual Perl conventions apply:

              'file'       # open file for reading
              '&gt;file'      # open file for writing
              '&gt;&gt;file'     # open file for appending
              '+&lt;file'     # open file read/write
              'command |'  # open a pipe from the command
              '| command'  # open a pipe to the command

       -fh You may provide <b>new()</b> with a previously-opened filehandle.  For example, to read from STDIN:

              $seqIO = Bio::AlignIO-&gt;new(-fh =&gt; \*STDIN);

           Note that you must pass filehandles as references to globs.

           If  neither  a filehandle nor a filename is specified, then the module will read from the @ARGV array
           or STDIN, using the familiar &lt;&gt; semantics.

       -format
           Specify the format of the file.  Supported formats include:

              bl2seq      Bl2seq Blast output
              clustalw    clustalw (.aln) format
              emboss      EMBOSS water and needle format
              fasta       FASTA format
              maf         Multiple Alignment Format
              mase        mase (seaview) format
              mega        MEGA format
              meme        MEME format
              msf         msf (GCG) format
              nexus       Swofford et al NEXUS format
              pfam        Pfam sequence alignment format
              phylip      Felsenstein PHYLIP format
              prodom      prodom (protein domain) format
              psi         PSI-BLAST format
              selex       selex (hmmer) format
              stockholm   stockholm format

           Currently only those formats which were implemented in Bio::SimpleAlign have been  incorporated  into
           Bio::AlignIO.   Specifically,  "mase", "stockholm" and "prodom" have only been implemented for input.
           See the specific module (e.g. Bio::AlignIO::prodom) for notes on supported versions.

           If no format is specified and a filename is given, then the module will attempt to deduce it from the
           filename suffix.  If this is unsuccessful, "fasta" format is assumed.

           The format name is case insensitive; "FASTA", "Fasta" and "fasta" are all treated equivalently.

   <b>Bio::AlignIO-&gt;newFh()</b>
          $fh = Bio::AlignIO-&gt;newFh(-fh   =&gt; \*FILEHANDLE, -format=&gt;$format);
          # read from STDIN or use @ARGV:
          $fh = Bio::AlignIO-&gt;newFh(-format =&gt; $format);

       This constructor behaves like new, but returns a tied filehandle rather than a Bio::AlignIO object.   You
       can  read  sequences  from  this  object using the familiar &lt;&gt; operator, and write to it using print. The
       usual array and $_ semantics work.  For example, you can read all sequence objects  into  an  array  like
       this:

         @sequences = &lt;$fh&gt;;

       Other operations, such as <b>read()</b>, <b>sysread()</b>, <b>write()</b>, <b>close()</b>, and <b>printf()</b> are not supported.

       -flush
        By  default,  all  files  (or  filehandles)  opened  for  writing  alignments will be flushed after each
        <b>write_aln()</b> making the file immediately usable.  If you do not need this  facility  and  would  like  to
        marginally  improve  the efficiency of writing multiple sequences to the same file (or filehandle), pass
        the -flush option '0' or any other value that evaluates as defined but false:

          my $clustal = Bio::AlignIO-&gt;new( -file   =&gt; "&lt;prot.aln",
                                           -format =&gt; "clustalw" );
          my $msf = Bio::AlignIO-&gt;new(-file   =&gt; "&gt;prot.msf",
                                      -format =&gt; "msf",
                                      -flush  =&gt; 0 ); # go as fast as we can!
          while($seq = $clustal-&gt;next_aln) { $msf-&gt;write_aln($seq) }

</pre><h4><b>OBJECT</b> <b>METHODS</b></h4><pre>
       See below for more detailed summaries.  The main methods are:

   <b>$alignment</b> <b>=</b> <b>$AlignIO-&gt;next_aln()</b>
       Fetch an alignment from a formatted file.

   <b>$AlignIO-&gt;write_aln($aln)</b>
       Write the specified alignment to a file..

   <b>TIEHANDLE(),</b> <b>READLINE(),</b> <b>PRINT()</b>
       These provide the tie interface.  See perltie for more details.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your  comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather  than  to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.   Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Peter</b> <b>Schattner</b></h4><pre>
       Email: <a href="mailto:schattner@alum.mit.edu">schattner@alum.mit.edu</a>

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
       Jason Stajich, <a href="mailto:jason@bioperl.org">jason@bioperl.org</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The  rest  of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : $stream = Bio::AlignIO-&gt;new(-file =&gt; $filename,
                                              -format =&gt; 'Format')
        Function: Returns a new seqstream
        Returns : A Bio::AlignIO::Handler initialised with
                  the appropriate format
        Args    : -file =&gt; $filename
                  -format =&gt; format
                  -fh =&gt; filehandle to attach to
                  -displayname_flat =&gt; 1 [optional]
                                       to force the displayname to not show start/end
                                       information

   <b>newFh</b>
        Title   : newFh
        Usage   : $fh = Bio::AlignIO-&gt;newFh(-file=&gt;$filename,-format=&gt;'Format')
        Function: does a new() followed by an fh()
        Example : $fh = Bio::AlignIO-&gt;newFh(-file=&gt;$filename,-format=&gt;'Format')
                  $sequence = &lt;$fh&gt;;   # read a sequence object
                  print $fh $sequence; # write a sequence object
        Returns : filehandle tied to the Bio::AlignIO::Fh class
        Args    :

   <b>fh</b>
        Title   : fh
        Usage   : $obj-&gt;fh
        Function:
        Example : $fh = $obj-&gt;fh;      # make a tied filehandle
                  $sequence = &lt;$fh&gt;;   # read a sequence object
                  print $fh $sequence; # write a sequence object
        Returns : filehandle tied to the Bio::AlignIO::Fh class
        Args    :

   <b>format</b>
        Title   : format
        Usage   : $format = $stream-&gt;format()
        Function: Get the alignment format
        Returns : alignment format
        Args    : none

   <b>_load_format_module</b>
        Title   : _load_format_module
        Usage   : *INTERNAL AlignIO stuff*
        Function: Loads up (like use) a module at run time on demand
        Example :
        Returns :
        Args    :

   <b>next_aln</b>
        Title   : next_aln
        Usage   : $aln = stream-&gt;next_aln
        Function: reads the next $aln object from the stream
        Returns : a Bio::Align::AlignI compliant object
        Args    :

   <b>write_aln</b>
        Title   : write_aln
        Usage   : $stream-&gt;write_aln($aln)
        Function: writes the $aln object into the stream
        Returns : 1 for success and 0 for error
        Args    : Bio::Seq object

   <b>_guess_format</b>
        Title   : _guess_format
        Usage   : $obj-&gt;_guess_format($filename)
        Function:
        Example :
        Returns : guessed format of filename (lower case)
        Args    :

   <b>force_displayname_flat</b>
        Title   : force_displayname_flat
        Usage   : $obj-&gt;force_displayname_flat($newval)
        Function:
        Example :
        Returns : value of force_displayname_flat (a scalar)
        Args    : on set, new value (a scalar or undef, optional)

   <b>alphabet</b>
        Title   : alphabet
        Usage   : $obj-&gt;alphabet($newval)
        Function: Get/Set alphabet for purpose of passing to Bio::LocatableSeq creation
        Example : $obj-&gt;alphabet('dna');
        Returns : value of alphabet (a scalar)
        Args    : on set, new value (a scalar or undef, optional)

perl v5.32.1                                       2021-08-15                                  <u>Bio::<a href="../man3pm/AlignIO.3pm.html">AlignIO</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>