<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::UpdateableSeqI - Descendant of Bio::SeqI that allows updates</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::UpdateableSeqI - Descendant of Bio::SeqI that allows updates

</pre><h4><b>SYNOPSIS</b></h4><pre>
       See Bio::SeqI for most of the documentation.  See the documentation of the methods for further details.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::UpdateableSeqI is an interface for Sequence objects which are expected to allow users to perform
       basic editing functions (update/delete) on their component SeqFeatures.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>David</b> <b>Block</b></h4><pre>
       Email <a href="mailto:dblock@gene.pbi.nrc.ca">dblock@gene.pbi.nrc.ca</a>

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
       Ewan Birney forced me to this...

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>delete_feature</b>
        Title   : delete_feature
        Usage   : my $orphanlist=$self-&gt;delete_feature($feature,$transcript,$gene);
        Function: deletes the specified $feature from the given transcript, if $transcript is sent and exists and $feature is a feature of $transcript,
                  or from $gene if the $feature is a feature of $gene, or from $self if $transcript and $gene are not sent.  Keeps track of the features
                  of the $gene object that may be left as orphans and returns them as a listref.
        Example : I want to delete transcript 'abc' of gene 'def', with three exons, leaving only transcript 'ghi' with two exons.
                  This will leave exons 1 and 3 part of 'ghi', but exon 2 will become an orphan.
                  my $orphanlist=$seq-&gt;delete_feature($transcript{'abc'},undef,$gene{'def'});
                  $orphanlist is a reference to a list containing $exon{'2'};
        Returns : a listref of orphaned features after the deletion of $feature (optional)
        Args    : $feature - the feature to be deleted
                  $transcript - the transcript containing the $feature, so that a $feature can be removed from only one transcript when there are multiple
                                transcripts in a gene.
                  $gene - the gene containing the $transcript and/or the $feature

perl v5.32.1                                       2021-08-15                           <u>Bio::<a href="../man3pm/UpdateableSeqI.3pm.html">UpdateableSeqI</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>