<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::DB::GFF::Aggregator::waba_alignment -- A WABA alignment</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/gbrowse">gbrowse_2.56+dfsg-12build4_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::DB::GFF::Aggregator::waba_alignment -- A WABA alignment

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::DB::GFF;

         # Open the sequence database
         my $db      = Bio::DB::GFF-&gt;new( -adaptor =&gt; 'dbi:mysql',
                                          -dsn     =&gt; 'dbi:mysql:elegans42',
                                          -aggregator =&gt; ['waba_alignment'],
                                        );
         # fetch the synthetic feature type "waba_alignment"
         my @waba    = $db-&gt;features('waba_alignment');

        -------------------------------------------------------------------------------------
        Aggregator method: waba_alignment
        Main method:       -none
        Sub methods:       nucleotide_match:waba_weak nucleotide_match:waba_strong
                           nucleotide_match::waba_coding
        -------------------------------------------------------------------------------------

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::DB::GFF::Aggregator::waba_alignment handles the type of alignments produced by Jim Kent's WABA
       program, and was written to be compatible with the C elegans GFF files.  It aggregates the following
       feature types into an aggregate type of "waba_alignment":

          nucleotide_match:waba_weak
          nucleotide_match:waba_strong
          nucleotide_match:waba_coding

   <b>method</b>
        Title   : method
        Usage   : $aggregator-&gt;method
        Function: return the method for the composite object
        Returns : the string "waba_alignment"
        Args    : none
        Status  : Public

   <b>part_names</b>
        Title   : part_names
        Usage   : $aggregator-&gt;part_names
        Function: return the methods for the sub-parts
        Returns : the list "nucleotide_match:waba_weak", "nucleotide_match:waba_strong" and "nucleotide_match:waba_coding"
        Args    : none
        Status  : Public

</pre><h4><b>BUGS</b></h4><pre>
       None reported.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Bio::DB::GFF, Bio::DB::GFF::Aggregator

</pre><h4><b>AUTHOR</b></h4><pre>
       Lincoln Stein &lt;<a href="mailto:lstein@cshl.org">lstein@cshl.org</a>&gt;.

       Copyright (c) 2001 Cold Spring Harbor Laboratory.

       This library is free software; you can redistribute it and/or modify it under the same terms as Perl
       itself.

perl v5.40.0                                       2024-10-20             <u>Bio::DB::GFF::A...:<a href="../man3pm/waba_alignment.3pm.html">waba_alignment</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>