<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Run::EMBOSSacd - class for EMBOSS Application qualifiers</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-perl-run-perl">libbio-perl-run-perl_1.7.3-8_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Run::EMBOSSacd - class for EMBOSS Application qualifiers

</pre><h4><b>SYNOPSIS</b></h4><pre>
         # Get an EMBOSS factory
         use Bio::Factory::EMBOSS;
         $f = Bio::Factory::EMBOSS -&gt; new();
         # Get an EMBOSS application  object from the factory
         $water = $f-&gt;program('water') || die "Program not found!\n";

         # Here is an example of running the application - water can
         # compare 1 sequence against 1 or more sequences using Smith-Waterman.
         # Pass a Sequence object and a reference to an array of objects.

         my $wateroutfile = 'out.water';
         $water-&gt;run({-asequence =&gt; $seq_object,
                      -bsequence =&gt; \@seq_objects,
                      -gapopen   =&gt; '10.0',
                      -gapextend =&gt; '0.5',
                      -outfile   =&gt; $wateroutfile});

         # Now you might want to get the alignment
         use Bio::AlignIO;
         my $alnin = Bio::AlignIO-&gt;new(-format =&gt; 'emboss',
                                       -file   =&gt; $wateroutfile);

         while ( my $aln = $alnin-&gt;next_aln ) {
             # process the alignment -- these will be Bio::SimpleAlign objects
         }

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The EMBOSSacd represents all the possible command line arguments that can be given to an EMBOSS
       application.

       Do not create this object directly. It will be created and attached to its corresponding
       Bio::Tools::Run::EMBOSSApplication if you set

         $application-&gt;verbose &gt; 0

       Call

         $application-&gt;acd

       to retrive the Bio::Tools::Run::EMBOSSApplication::EMBOSSacd object.

       See also Bio::Tools::Run::EMBOSSApplication and Bio::Factory::EMBOSS.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing lists  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         <a href="http://redmine.open-bio.org/projects/bioperl/">http://redmine.open-bio.org/projects/bioperl/</a>

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Heikki</b> <b>Lehvaslaiho</b></h4><pre>
       Email:  heikki-at-bioperl-dot-org Address:

            EMBL Outstation, European Bioinformatics Institute
            Wellcome Trust Genome Campus, Hinxton
            Cambs. CB10 1SD, United Kingdom

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : $emboss_prog-&gt;acd($prog_name);
        Function: Constructor for the class.
                  Calls EMBOSS program 'acdc', converts the
                  HTML output into XML and uses XML::Twig XML
                  parser to write out a hash of qualifiers which is
                  then blessed.
        Throws  : without program name
        Returns : new object
        Args    : EMBOSS program name

   <b>name</b>
        Title   : name
        Usage   : $embossacd-&gt;name
        Function: sets/gets the name of the EMBOSS program
                  Setting is done by the EMBOSSApplication object,
                  you should only get it.
        Throws  :
        Returns : name string
        Args    : None

   <b>print</b>
        Title   : print
        Usage   : $embossacd-&gt;print; $embossacd-&gt;print('-word');
        Function: Print out the qualifiers.
                  Uses Data::Dumper to print the qualifiers into STDOUT.
                  A valid qualifier name given as an argment limits the output.
        Throws  :
        Returns : print string
        Args    : optional qualifier name

   <b>mandatory</b>
        Title   : mandatory
        Usage   : $acd-&gt;mandatory
        Function: gets a  mandatory subset of qualifiers
        Throws  :
        Returns : Bio::Tools::Run::EMBOSSacd object
        Args    : none

   <b>Qualifier</b> <b>queries</b>
       These methods can be used test qualifier names and read values.

   <b>qualifier</b>
        Title   : qualifier
        Usage   : $acd-&gt;qualifier($string)
        Function: tests for the existence of the qualifier
        Throws  :
        Returns : boolean
        Args    : string, name of the qualifier

   <b>category</b>
        Title   : category
        Usage   : $acd-&gt;category($qual_name)
        Function: Return the category of the qualifier
        Throws  :
        Returns : 'mandatory' or 'optional' or 'advanced' or
                   'associated' or 'general'
        Args    : string, name of the qualifier

   <b>values</b>
        Title   : values
        Usage   : $acd-&gt;values($qual_name)
        Function: Return the possible values for the qualifier
        Throws  :
        Returns : string
        Args    : string, name of the qualifier

   <b>descr</b>
        Title   : descr
        Usage   : $acd-&gt;descr($qual_name)
        Function: Return the description of the qualifier
        Throws  :
        Returns : boolean
        Args    : string, name of the qualifier

   <b>unnamed</b>
        Title   : unnamed
        Usage   : $acd-&gt;unnamed($qual_name)
        Function: Find if the qualifier can be left unnamed
        Throws  :
        Returns : 0 if needs to be named, order number otherwise
        Args    : string, name of the qualifier

   <b>default</b>
        Title   : default
        Usage   : $acd-&gt;default($qual_name)
        Function: Return the default value for the qualifier
        Throws  :
        Returns : scalar
        Args    : string, name of the qualifier

perl v5.34.0                                       2022-02-10                    <u>Bio::Tools::Run::<a href="../man3pm/EMBOSSacd.3pm.html">EMBOSSacd</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>