<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::FeatureIO - Handler for FeatureIO</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-featureio-perl">libbio-featureio-perl_1.6.905-2_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::FeatureIO - Handler for FeatureIO

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::FeatureIO;

         #read from a file
         $in  = Bio::FeatureIO-&gt;new(-file =&gt; "my.gff" , -format =&gt; 'GFF');

         #read from a filehandle
         $in  = Bio::FeatureIO-&gt;new(-fh =&gt; \*GFF , -format =&gt; 'GFF');

         #read features already attached to a sequence
         my $feat = Bio::FeatureIO-&gt;new(-seq =&gt; $seq , -format =&gt; 'features');

         #read new features for existing sequence
         my $seq = Bio::FeatureIO-&gt;new(-seq =&gt; $seq , -format =&gt; 'Das');

         #write out features
         $out = Bio::FeatureIO-&gt;new(-file    =&gt; "&gt;outputfilename" ,
                                    -format  =&gt; 'GFF' ,
                                    -version =&gt; 3);

         while ( my $feature = $in-&gt;next_feature() ) {
           $out-&gt;write_feature($feature);
         }

</pre><h4><b>DESCRIPTION</b></h4><pre>
       An I/O iterator subsystem for genomic sequence features.

       Bio::FeatureIO is a handler module for the formats in the FeatureIO set (eg, Bio::FeatureIO::GFF). It is
       the officially sanctioned way of getting at the format objects, which most people should use.

       The Bio::FeatureIO system can be thought of like biological file handles. They are attached to
       filehandles with smart formatting rules (eg, GFF format, or BED format) and can either read or write
       feature objects (Bio::SeqFeature objects, or more correctly, Bio::FeatureHolderI implementing objects, of
       which Bio::SeqFeature is one such object). If you want to know what to do with a Bio::SeqFeatureI object,
       read Bio::SeqFeatureI.

       The idea is that you request a stream object for a particular format. All the stream objects have a
       notion of an internal file that is read from or written to. A particular FeatureIO object instance is
       configured for either input or output. A specific example of a stream object is the Bio::FeatureIO::gff
       object.

       Each stream object has functions:

         $stream-&gt;next_feature();
         $stream-&gt;write_feature($feature);

</pre><h4><b>SUPPORTED</b> <b>FORMATS</b></h4><pre>
        name                         module
        -----------------------------------
        BED                          bed.pm
        GFF                          gff.pm
        GTF                          gtf.pm
        InterPro (IPRScan 4.0)  interpro.pm
        PTT (NCBI protein table)     ptt.pm

</pre><h4><b>CONSTRUCTORS</b></h4><pre>
   <b>Bio::FeatureIO-&gt;new()</b>
          $featureIO = Bio::FeatureIO-&gt;new(-file =&gt; 'filename',   -format=&gt;$format);
          $featureIO = Bio::FeatureIO-&gt;new(-fh   =&gt; \*FILEHANDLE, -format=&gt;$format);
          $featureIO = Bio::FeatureIO-&gt;new(-seq  =&gt; $seq,         -format=&gt;$format);

       The <b>new()</b> class method constructs a new Bio::FeatureIO object.  The returned object can be used to
       retrieve or print Seq objects. <b>new()</b> accepts the following parameters:

       -file
           A file path to be opened for reading or writing.  The usual Perl conventions apply:

              'file'       # open file for reading
              '&gt;file'      # open file for writing
              '&gt;&gt;file'     # open file for appending
              '+&lt;file'     # open file read/write
              'command |'  # open a pipe from the command
              '| command'  # open a pipe to the command

       -fh You may provide <b>new()</b> with a previously-opened filehandle.  For example, to read from STDIN:

              $featio = Bio::FeatureIO-&gt;new(-fh =&gt; \*STDIN);

           Note that you must pass filehandles as references to globs.

           If  neither  a filehandle nor a filename is specified, then the module will read from the @ARGV array
           or STDIN, using the familiar &lt;&gt; semantics.

           A string filehandle is handy if you want to modify the output in the memory, before printing it  out.
           The following program reads in EMBL formatted entries from a file and prints them out in fasta format
           with some HTML tags:

             use Bio::FeatureIO;
             use IO::String;
             my $in  = Bio::FeatureIO-&gt;new('-file' =&gt; "my.gff" ,
                                       '-format' =&gt; 'EMBL');
             while ( my $f = $in-&gt;next_feature() ) {
                 # the output handle is reset for every file
                 my $stringio = IO::String-&gt;new($string);
                 my $out = Bio::FeatureIO-&gt;new('-fh' =&gt; $stringio,
                                           '-format' =&gt; 'gtf');
                 # output goes into $string
                 $out-&gt;write_feature($f);
                 # modify $string
                 $string =~ s|(&gt;)(\w+)|$1&lt;font color="Red"&gt;$2&lt;/font&gt;|g;
                 # print into STDOUT
                 print $string;
             }

       -format
           Specify the format of the file.  See above for list of supported formats

       -flush
           By  default,  all  files  (or  filehandles)  opened  for writing sequences will be flushed after each
           <b>write_seq()</b> (making the file immediately usable). If you don't need this facility and would  like  to
           marginally  improve  the  efficiency  of writing multiple sequences to the same file (or filehandle),
           pass the -flush option '0' or any other value that evaluates as defined but false:

             my $f1 = Bio::FeatureIO-&gt;new -file   =&gt; "&lt;a.f1",
                                         -format =&gt; "f1";
             my $f2 = Bio::FeatureIO-&gt;new -file   =&gt; "&gt;a.f2",
                                         -format =&gt; "f2",
                                         -flush  =&gt; 0; # go as fast as we can!

             while($feature = $f1-&gt;next_feature) { $f2-&gt;write_feature($feature) }

   <b>Bio::FeatureIO-&gt;newFh()</b>
          $fh = Bio::FeatureIO-&gt;newFh(-fh   =&gt; \*FILEHANDLE, -format=&gt;$format);
          $fh = Bio::FeatureIO-&gt;newFh(-format =&gt; $format);
          # etc.

       This constructor behaves like <b>new()</b>, but returns a tied filehandle rather than a  Bio::FeatureIO  object.
       You  can  read  sequences from this object using the familiar &lt;&gt; operator, and write to it using <b>print()</b>.
       The usual array and $_ semantics work. For example, you can read all sequence objects into an array  like
       this:

         @features = &lt;$fh&gt;;

       Other operations, such as <b>read()</b>, <b>sysread()</b>, <b>write()</b>, <b>close()</b>, and <b>printf()</b> are not supported.

</pre><h4><b>OBJECT</b> <b>METHODS</b></h4><pre>
       See below for more detailed summaries.  The main methods are:

       next_feature
          Fetch the next feature from the stream.

       write_feature
          Write the specified feature(s) to the stream.

       feature_factory
          This gets/sets the specific Bio::Factory::FeatureFactoryI

       The following methods delegate to the inter

       feature_class
          Set the specific Bio::SeqFeatureI class to return

       type_features
          Boolean flag, ensures the returned features are typed

       unflatten_features
          Ensure the returned features are unflattened

   <b>TIEHANDLE(),</b> <b>READLINE(),</b> <b>PRINT()</b>
       These provide the tie interface.  See perltie for more details.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User  feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.

       Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be  able  look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report  bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         <a href="http://bugzilla.open-bio.org/">http://bugzilla.open-bio.org/</a>

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Allen</b> <b>Day</b></h4><pre>
       Email <a href="mailto:allenday@ucla.edu">allenday@ucla.edu</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are  usually  preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : $stream = Bio::FeatureIO-&gt;new(-file =&gt; $filename, -format =&gt; 'Format')
        Function: Returns a new feature stream
        Returns : A Bio::FeatureIO stream initialised with the appropriate format
        Args    : Named parameters:
                    -file =&gt; $filename
                    -fh =&gt; filehandle to attach to
                    -format =&gt; format

   <b>newFh</b>
        Title   : newFh
        Usage   : $fh = Bio::FeatureIO-&gt;newFh(-file=&gt;$filename,-format=&gt;'Format')
        Function: does a new() followed by an fh()
        Example : $fh = Bio::FeatureIO-&gt;newFh(-file=&gt;$filename,-format=&gt;'Format')
                  $feature = &lt;$fh&gt;;   # read a feature object
                  print $fh $feature; # write a feature object
        Returns : filehandle tied to the Bio::FeatureIO::Fh class
        Args    :

       See Bio::FeatureIO::Fh

   <b>fh</b>
        Title   : fh
        Usage   : $obj-&gt;fh
        Function:
        Example : $fh = $obj-&gt;fh;      # make a tied filehandle
                  $feature = &lt;$fh&gt;;   # read a feature object
                  print $fh $feature; # write a feature object
        Returns : filehandle tied to Bio::FeatureIO class
        Args    : none

   <b>next_feature</b>
        Title   : next_feature
        Usage   : $feature = stream-&gt;next_feature
        Function: Reads the next feature object from the stream and returns it.

                  Certain driver modules may encounter entries in the stream
                  that are either misformatted or that use syntax not yet
                  understood by the driver. If such an incident is
                  recoverable, e.g., by dismissing a feature of a feature
                  table or some other non-mandatory part of an entry, the
                  driver will issue a warning. In the case of a
                  non-recoverable situation an exception will be thrown.  Do
                  not assume that you can resume parsing the same stream
                  after catching the exception. Note that you can always turn
                  recoverable errors into exceptions by calling
                  $stream-&gt;<a href="../man2/verbose.2.html">verbose</a>(2).

        Returns : a Bio::SeqFeatureI feature object
        Args    : none

       See Bio::Root::RootI, Bio::SeqFeatureI

   <b>write_feature</b>
        Title   : write_feature
        Usage   : $stream-&gt;write_feature($feature)
        Function: writes the $feature object into the stream
        Returns : 1 for success and 0 for error
        Args    : Bio::SeqFeature object

   <b>_load_format_module</b>
        Title   : _load_format_module
        Usage   : *INTERNAL FeatureIO stuff*
        Function: Loads up (like use) a module at run time on demand
        Example :
        Returns :
        Args    :

   <b>seq</b>
        Title   : seq
        Usage   : $obj-&gt;seq() OR $obj-&gt;seq($newSeq)
        Example :
        Returns : Bio::SeqI object
        Args    : newSeq (optional)

   <b>_filehandle</b>
        Title   : _filehandle
        Usage   : $obj-&gt;_filehandle($newval)
        Function: This method is deprecated. Call _fh() instead.
        Example :
        Returns : value of _filehandle
        Args    : newvalue (optional)

   <b>_guess_format</b>
        Title   : _guess_format
        Usage   : $obj-&gt;_guess_format($filename)
        Function: guess format based on file suffix
        Example :
        Returns : guessed format of filename (lower case)
        Args    :
        Notes   : See "SUPPORTED FORMATS"

perl v5.30.0                                       2020-01-13                                <u>Bio::<a href="../man3pm/FeatureIO.3pm.html">FeatureIO</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>