<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::AlignIO::arp - ARP MSA Sequence input/output stream</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::AlignIO::arp - ARP MSA Sequence input/output stream

</pre><h4><b>SYNOPSIS</b></h4><pre>
       Do not use this module directly.  Use it via the Bio::AlignIO class.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This object can create Bio::SimpleAlign objects from ARP flat files.  These are typically configuration-
       like data files for the program Arlequin.  For more information, see:

         <a href="http://lgb.unige.ch/arlequin/">http://lgb.unige.ch/arlequin/</a>

       For the moment, this retains the allele sequence data in the DATA section and inserts them into
       SimpleAlign objects. ARP files that contain other data (RFLP, etc.) are not expected to parse properly.
       Also, if the DNA data is actually SNP data, then the LocatableSeq object instantiation will throw an
       error.

       This is now set up as a generic parser (i.e. it parses everything) and collects as much data as possible
       into the SimpleAlign object.  The following in a general mapping of where data can be found:

           Tag        SimpleAlign
                      Method
           ----------------------------------------------------------------------
           Title      description
           SampleName id
           ----------------------------------------------------------------------

           Tag        Bio::Annotation   TagName                    Bio::Annotation
                      Class                                        Parameters
           ----------------------------------------------------------------------
            NE        SimpleValue       pfam_family_accession      value
            NL        SimpleValue       sequence_start_stop        value
            SS        SimpleValue       sec_structure_source       value
            BM        SimpleValue       build_model                value
            RN        Reference         reference                  *
           ----------------------------------------------------------------------
         * RN is generated based on the number of Bio::Annotation::Reference objects

       In addition, the number of samples found in the alignment is retained in a Bio::Annotation::TagTree
       object in the annotation collection and is accessible via:

         ($samples) = $aln-&gt;annotation-&gt;get_Annotations('Samples');
         say $samples-&gt;display_text;
         # or use other relevant TagTree methods to retrieve data

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHORS</b></h4><pre>
       Chris Fields (cjfields)

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>next_aln</b>
        Title   : next_aln
        Usage   : $aln = $stream-&gt;next_aln
        Function: returns the next alignment in the stream.
        Returns : Bio::Align::AlignI object - returns 0 on end of file
                   or on error
        Args    : -width =&gt; optional argument to specify the width sequence
                  will be written (60 chars by default)

       See Bio::Align::AlignI

   <b>write_aln</b>
        Title   : write_aln
        Usage   : $stream-&gt;write_aln(@aln)
        Function: writes the $aln object into the stream in xmfa format
        Returns : 1 for success and 0 for error
        Args    : L&lt;Bio::Align::AlignI&gt; object

       See Bio::Align::AlignI

perl v5.32.1                                       2021-08-15                             <u>Bio::AlignIO::<a href="../man3pm/arp.3pm.html">arp</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>