<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::PrimarySeqI - Interface definition for a Bio::PrimarySeq</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::PrimarySeqI - Interface definition for a Bio::PrimarySeq

</pre><h4><b>SYNOPSIS</b></h4><pre>
           # Bio::PrimarySeqI is the interface class for sequences.
           # If you are a newcomer to bioperl, you might want to start with
           # Bio::Seq documentation.

           # Test if this is a seq object
           $obj-&gt;isa("Bio::PrimarySeqI") ||
             $obj-&gt;throw("$obj does not implement the Bio::PrimarySeqI interface");

           # Accessors
           $string    = $obj-&gt;seq();
           $substring = $obj-&gt;subseq(12,50);
           $display   = $obj-&gt;display_id();       # for human display
           $id        = $obj-&gt;primary_id();       # unique id for this object,
                                                  # implementation defined
           $unique_key= $obj-&gt;accession_number(); # unique biological id

           # Object manipulation
           eval {
              $rev = $obj-&gt;revcom();
           };
           if( $@ ) {
              $obj-&gt;throw( "Could not reverse complement. ".
                           "Probably not DNA. Actual exception\n$@\n" );
           }

           $trunc = $obj-&gt;trunc(12,50);
           # $rev and $trunc are Bio::PrimarySeqI compliant objects

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This object defines an abstract interface to basic sequence information - for most users of the package
       the documentation (and methods) in this class are not useful - this is a developers-only class which
       defines what methods have to be implemented by other Perl objects to comply to the Bio::PrimarySeqI
       interface. Go "perldoc Bio::Seq" or "man Bio::Seq" for more information on the main class for sequences.

       PrimarySeq is an object just for the sequence and its name(s), nothing more. Seq is the larger object
       complete with features. There is a pure perl implementation of this in Bio::PrimarySeq. If you just want
       to use Bio::PrimarySeq objects, then please read that module first. This module defines the interface,
       and is of more interest to people who want to wrap their own Perl Objects/RDBs/FileSystems etc in way
       that they "are" bioperl sequence objects, even though it is not using Perl to store the sequence etc.

       This interface defines what bioperl considers necessary to "be" a sequence, without providing an
       implementation of this, an implementation is provided in Bio::PrimarySeq. If you want to provide a
       Bio::PrimarySeq-compliant object which in fact wraps another object/database/out-of-perl experience, then
       this is the correct thing to wrap, generally by providing a wrapper class which would inherit from your
       object and this Bio::PrimarySeqI interface. The wrapper class then would have methods lists in the
       "Implementation Specific Functions" which would provide these methods for your object.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Ewan</b> <b>Birney</b></h4><pre>
       Email <a href="mailto:birney@ebi.ac.uk">birney@ebi.ac.uk</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

</pre><h4><b>Implementation-specific</b> <b>Functions</b></h4><pre>
       These functions are the ones that a specific implementation must define.

   <b>seq</b>
        Title   : seq
        Usage   : $string = $obj-&gt;seq()
        Function: Returns the sequence as a string of letters. The
                  case of the letters is left up to the implementer.
                  Suggested cases are upper case for proteins and lower case for
                  DNA sequence (IUPAC standard), but implementations are suggested to
                  keep an open mind about case (some users... want mixed case!)
        Returns : A scalar
        Status  : Virtual

   <b>subseq</b>
        Title   : subseq
        Usage   : $substring = $obj-&gt;subseq(10,40);
        Function: Returns the subseq from start to end, where the first base
                  is 1 and the number is inclusive, i.e. 1-2 are the first two
                  bases of the sequence.

                  Start cannot be larger than end but can be equal.

        Returns : A string
        Args    :
        Status  : Virtual

   <b>display_id</b>
        Title   : display_id
        Usage   : $id_string = $obj-&gt;display_id();
        Function: Returns the display id, also known as the common name of the Sequence
                  object.

                  The semantics of this is that it is the most likely string
                  to be used as an identifier of the sequence, and likely to
                  have "human" readability.  The id is equivalent to the ID
                  field of the GenBank/EMBL databanks and the id field of the
                  Swissprot/sptrembl database. In fasta format, the &gt;(\S+) is
                  presumed to be the id, though some people overload the id
                  to embed other information. Bioperl does not use any
                  embedded information in the ID field, and people are
                  encouraged to use other mechanisms (accession field for
                  example, or extending the sequence object) to solve this.

                  Notice that $seq-&gt;id() maps to this function, mainly for
                  legacy/convenience reasons.
        Returns : A string
        Args    : None
        Status  : Virtual

   <b>accession_number</b>
        Title   : accession_number
        Usage   : $unique_biological_key = $obj-&gt;accession_number;
        Function: Returns the unique biological id for a sequence, commonly
                  called the accession_number. For sequences from established
                  databases, the implementors should try to use the correct
                  accession number. Notice that primary_id() provides the
                  unique id for the implementation, allowing multiple objects
                  to have the same accession number in a particular implementation.

                  For sequences with no accession number, this method should return
                  "unknown".
        Returns : A string
        Args    : None
        Status  : Virtual

   <b>primary_id</b>
        Title   : primary_id
        Usage   : $unique_implementation_key = $obj-&gt;primary_id;
        Function: Returns the unique id for this object in this
                  implementation. This allows implementations to manage their
                  own object ids in a way the implementation can control
                  clients can expect one id to map to one object.

                  For sequences with no accession number, this method should
                  return a stringified memory location.

        Returns : A string
        Args    : None
        Status  : Virtual

   <b>can_call_new</b>
        Title   : can_call_new
        Usage   : if( $obj-&gt;can_call_new ) {
                    $newobj = $obj-&gt;new( %param );
                }
        Function: Can_call_new returns 1 or 0 depending
                  on whether an implementation allows new
                  constructor to be called. If a new constructor
                  is allowed, then it should take the followed hashed
                  constructor list.

                  $myobject-&gt;new( -seq =&gt; $sequence_as_string,
                                  -display_id  =&gt; $id
                                  -accession_number =&gt; $accession
                                  -alphabet =&gt; 'dna',
                                  );
        Returns : 1 or 0
        Args    :

   <b>alphabet</b>
        Title   : alphabet
        Usage   : if( $obj-&gt;alphabet eq 'dna' ) { /Do Something/ }
        Function: Returns the type of sequence being one of
                  'dna', 'rna' or 'protein'. This is case sensitive.

                  This is not called "type" because this would cause
                  upgrade problems from the 0.5 and earlier Seq objects.

        Returns : A string either 'dna','rna','protein'. NB - the object must
                  make a call of the alphabet, if there is no alphabet specified it
                  has to guess.
        Args    : None
        Status  : Virtual

   <b>moltype</b>
        Title   : moltype
        Usage   : Deprecated. Use alphabet() instead.

</pre><h4><b>Implementation-optional</b> <b>Functions</b></h4><pre>
       The following functions rely on the above functions. An implementing class does not need to provide these
       functions, as they will be provided by this class, but is free to override these functions.

       The <b>revcom()</b>, <b>trunc()</b>, and <b>translate()</b> methods create new sequence objects. They will call <b>new()</b> on the
       class of the sequence object instance passed as argument, unless <b>can_call_new()</b> returns FALSE. In the
       latter case a Bio::PrimarySeq object will be created. Implementors which really want to control how
       objects are created (eg, for object persistence over a database, or objects in a CORBA framework), they
       are encouraged to override these methods

   <b>revcom</b>
        Title   : revcom
        Usage   : $rev = $seq-&gt;revcom()
        Function: Produces a new Bio::PrimarySeqI implementing object which
                  is the reversed complement of the sequence. For protein
                  sequences this throws an exception of "Sequence is a
                  protein. Cannot revcom".

                  The id is the same id as the original sequence, and the
                  accession number is also identical. If someone wants to
                  track that this sequence has be reversed, it needs to
                  define its own extensions.

                  To do an inplace edit of an object you can go:

                  $seq = $seq-&gt;revcom();

                  This of course, causes Perl to handle the garbage
                  collection of the old object, but it is roughly speaking as
                  efficient as an inplace edit.

        Returns : A new (fresh) Bio::PrimarySeqI object
        Args    : None

   <b>trunc</b>
        Title   : trunc
        Usage   : $subseq = $myseq-&gt;trunc(10,100);
        Function: Provides a truncation of a sequence.
        Returns : A fresh Bio::PrimarySeqI implementing object.
        Args    : Two integers denoting first and last base of the sub-sequence.

   <b>translate</b>
        Title   : translate
        Usage   : $protein_seq_obj = $dna_seq_obj-&gt;translate

                  Or if you expect a complete coding sequence (CDS) translation,
                  with initiator at the beginning and terminator at the end:

                  $protein_seq_obj = $cds_seq_obj-&gt;translate(-complete =&gt; 1);

                  Or if you want translate() to find the first initiation
                  codon and return the corresponding protein:

                  $protein_seq_obj = $cds_seq_obj-&gt;translate(-orf =&gt; 1);

        Function: Provides the translation of the DNA sequence using full
                  IUPAC ambiguities in DNA/RNA and amino acid codes.

                  The complete CDS translation is identical to EMBL/TREMBL
                  database translation. Note that the trailing terminator
                  character is removed before returning the translated protein
                  object.

                  Note: if you set $dna_seq_obj-&gt;<a href="../man1/verbose.1.html">verbose</a>(1) you will get a
                  warning if the first codon is not a valid initiator.

        Returns : A Bio::PrimarySeqI implementing object
        Args    : -terminator
                      character for terminator, default '*'
                  -unknown
                      character for unknown, default 'X'
                  -frame
                      positive integer frame shift (in bases), default 0
                  -codontable_id
                      integer codon table id, default 1
                  -complete
                      boolean, if true, complete CDS is expected. default false
                  -complete_codons
                      boolean, if true, codons which are incomplete are translated if a
                      suitable amino acid is found. For instance, if the incomplete
                      codon is 'GG', the completed codon is 'GGN', which is glycine
                      (G). Defaults to 'false'; setting '-complete' also makes this
                      true.
                  -throw
                      boolean, throw exception if ORF not complete, default false
                  -orf
                      if 'longest', find longest ORF.  other true value, find
                      first ORF.  default 0
                  -codontable
                      optional L&lt;Bio::Tools::CodonTable&gt; object to use for
                      translation
                  -start
                      optional three-character string to force as initiation
                      codon (e.g. 'atg'). If unset, start codons are
                      determined by the CodonTable.  Case insensitive.
                  -offset
                      optional positive integer offset for fuzzy locations.
                      if set, must be either 1, 2, or 3

       <u>Notes</u>

       The -start argument only applies when -orf is set to 1. By default all initiation codons found in the
       given codon table are used but when "start" is set to some codon this codon will be used exclusively as
       the initiation codon. Note that the default codon table (NCBI "Standard") has 3 initiation codons!

       By default <b>translate()</b> translates termination codons to the some character (default is *), both internal
       and trailing codons. Setting "-complete" to 1 tells <b>translate()</b> to remove the trailing character.

       -offset is used for seqfeatures which contain the the \codon_start tag and can be set to 1, 2, or 3.
       This is the offset by which the sequence translation starts relative to the first base of the feature

       For details on codon tables used by <b>translate()</b> see Bio::Tools::CodonTable.

       Deprecated argument set (v. 1.5.1 and prior versions) where each argument is an element in an array:

         1: character for terminator (optional), defaults to '*'.
         2: character for unknown amino acid (optional), defaults to 'X'.
         3: frame (optional), valid values are 0, 1, 2, defaults to 0.
         4: codon table id (optional), defaults to 1.
         5: complete coding sequence expected, defaults to 0 (false).
         6: boolean, throw exception if not complete coding sequence
            (true), defaults to warning (false)
         7: codontable, a custom Bio::Tools::CodonTable object (optional).

   <b>transcribe()</b>
        Title   : transcribe
        Usage   : $xseq = $seq-&gt;transcribe
        Function: Convert base T to base U
        Returns : PrimarySeqI object of alphabet 'rna' or
                  undef if $seq-&gt;alphabet ne 'dna'
        Args    :

   <b>rev_transcribe()</b>
        Title   : rev_transcribe
        Usage   : $rtseq = $seq-&gt;rev_transcribe
        Function: Convert base U to base T
        Returns : PrimarySeqI object of alphabet 'dna' or
                  undef if $seq-&gt;alphabet ne 'rna'
        Args    :

   <b>id</b>
        Title   : id
        Usage   : $id = $seq-&gt;id()
        Function: ID of the sequence. This should normally be (and actually is in
                  the implementation provided here) just a synonym for display_id().
        Returns : A string.
        Args    :

   <b>length</b>
        Title   : length
        Usage   : $len = $seq-&gt;length()
        Function:
        Returns : Integer representing the length of the sequence.
        Args    :

   <b>desc</b>
        Title   : desc
        Usage   : $seq-&gt;desc($newval);
                  $description = $seq-&gt;desc();
        Function: Get/set description text for a seq object
        Returns : Value of desc
        Args    : newvalue (optional)

   <b>is_circular</b>
        Title   : is_circular
        Usage   : if( $obj-&gt;is_circular) { # Do something }
        Function: Returns true if the molecule is circular
        Returns : Boolean value
        Args    : none

</pre><h4><b>Private</b> <b>functions</b></h4><pre>
       These are some private functions for the PrimarySeqI interface. You do not need to implement these
       functions

   <b>_find_orfs_nucleotide</b>
        Title   : _find_orfs_nucleotide
        Usage   :
        Function: Finds ORF starting at 1st initiation codon in nucleotide sequence.
                  The ORF is not required to have a termination codon.
        Example :
        Returns : a list of string coordinates of ORF locations (0-based half-open),
                  sorted descending by length (so that the longest is first)
                  as: [ start, end, frame, length ], [ start, end, frame, length ], ...
        Args    : Nucleotide sequence,
                  CodonTable object,
                  (optional) alternative initiation codon (e.g. 'ATA'),
                  (optional) boolean that, if true, stops after finding the
                             first available ORF

   <b>_attempt_to_load_Seq</b>
        Title   : _attempt_to_load_Seq
        Usage   :
        Function:
        Example :
        Returns :
        Args    :

perl v5.32.1                                       2021-08-15                              <u>Bio::<a href="../man3pm/PrimarySeqI.3pm.html">PrimarySeqI</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>