<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Run::Primer3 - Create input for and work with the output from the program primer3</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-perl-run-perl">libbio-perl-run-perl_1.7.3-8_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Run::Primer3 - Create input for and work with the output from the program primer3

</pre><h4><b>SYNOPSIS</b></h4><pre>
       Bio::Tools::Primer3 creates the input files needed to design primers using primer3 and provides
       mechanisms to access data in the primer3 output files.

       This module provides a bioperl interface to the program primer3. See
       <a href="http://frodo.wi.mit.edu/primer3/primer3_code.html">http://frodo.wi.mit.edu/primer3/primer3_code.html</a> for details and to download the software. This module
       only works for primer3 release 1 but is not guaranteed to work with earlier versions.

         # design some primers.
         # the output will be put into temp.out
         use Bio::Tools::Run::Primer3;
         use Bio::SeqIO;

         my $seqio = Bio::SeqIO-&gt;new(-file=&gt;'data/dna1.fa');
         my $seq = $seqio-&gt;next_seq;
         my $primer3 = Bio::Tools::Run::Primer3-&gt;new(-seq =&gt; $seq,
                                                     -outfile =&gt; "temp.out",
                                                     -path =&gt; "/usr/bin/primer3_core");

         # or after the fact you can change the program_name
         $primer3-&gt;program_name('my_suprefast_primer3');

         unless ($primer3-&gt;executable) {
           print STDERR "primer3 can not be found. Is it installed?\n";
           exit(-1)
         }

         # what are the arguments, and what do they mean?
         my $args = $primer3-&gt;arguments;

         print "ARGUMENT\tMEANING\n";
         foreach my $key (keys %{$args}) {print "$key\t", $$args{$key}, "\n"}

         # set the maximum and minimum Tm of the primer
         $primer3-&gt;add_targets('PRIMER_MIN_TM'=&gt;56, 'PRIMER_MAX_TM'=&gt;90);

         # design the primers. This runs primer3 and returns a
         # Bio::Tools::Run::Primer3 object with the results
         $results = $primer3-&gt;run;

         # see the Bio::Tools::Run::Primer3 pod for
         # things that you can get from this. For example:

         print "There were ", $results-&gt;number_of_results, " primers\n";

       Bio::Tools::Run::Primer3 creates the input files needed to design primers using primer3 and provides
       mechanisms to access data in the primer3 output files.

       This module provides a bioperl interface to the program primer3. See
       <a href="http://www-genome.wi.mit.edu/genome_software/other/primer3.html">http://www-genome.wi.mit.edu/genome_software/other/primer3.html</a> for details and to download the software.

       This module is based on one written by Chad Matsalla (<a href="mailto:bioinformatics1@dieselwurks.com">bioinformatics1@dieselwurks.com</a>). I have ripped
       some of his code, and added a lot of my own. I hope he is not mad at me!

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://www.bioperl.org/MailList.html">http://www.bioperl.org/MailList.html</a>             - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         <a href="http://redmine.open-bio.org/projects/bioperl/">http://redmine.open-bio.org/projects/bioperl/</a>

</pre><h4><b>AUTHOR</b></h4><pre>
       Rob Edwards

       <a href="mailto:redwards@utmem.edu">redwards@utmem.edu</a>

       Based heavily on work of Chad Matsalla

       <a href="mailto:bioinformatics1@dieselwurks.com">bioinformatics1@dieselwurks.com</a>

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
       Shawn Hoon shawnh-at-stanford.edu Jason Stajich jason-at-bioperl.org Brian Osborne
       osborne1-at-optonline.net

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Bio::Tools::Primer3

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>new()</b>
        Title   : new()
        Usage   : my $primer3 = Bio::Tools::Run::Primer3-&gt;new(-file=&gt;$file) to read
                  a primer3 output file.
                  my $primer3 = Bio::Tools::Run::Primer3-&gt;new(-seq=&gt;sequence object)
                  design primers against sequence
        Function: Start primer3 working and adds a sequence. At the moment it
                  will not clear out the old sequence, but I suppose it should.
        Returns : Does not return anything. If called with a filename will allow
                  you to retrieve the results
        Args    : -seq (optional) Bio::Seq object of sequence. This is required
                  to run primer3 but can be added later with add_targets()
                       -outfile file name to output results to (can also be added
                  with $primer3-&gt;outfile_name
                       -path path to primer3 executable, including program name, e.g.
                  "/usr/bin/primer3_core". This can also be set with program_name
                  and program_dir
                       -verbose (optional) set verbose output
        Notes   :

   <b>program_name</b>
        Title   : program_name
        Usage   : $primer3-&gt;program_name()
        Function: holds the program name
        Returns:  string
        Args    : None

   <b>program_dir</b>
        Title   : program_dir
        Usage   : $primer3-&gt;program_dir($dir)
        Function: returns the program directory, which may also be obtained from ENV variable.
        Returns :  string
        Args    :

   <b>add_targets()</b>
        Title   : add_targets()
        Usage   : $primer3-&gt;add_targets(key=&gt;value)
        Function: Add any legal value to the input command line.
        Returns : Returns the number of arguments added.
        Args    : Use $primer3-&gt;arguments to find a list of all the values
                  that are allowed, or see the primer3 docs.
        Notes   : This will only do limited error checking at the moment,
                  but it should work.

   <b>run()</b>
        Title   : run()
        Usage   : $primer3-&gt;run();
        Function: Run the primer3 program with the arguments that you have supplied.
        Returns : A Bio::Tools::Primer3 object containing the results.
        Args    : None.
        Note    : See the Bio::Tools::Primer3 documentation for those functions.

   <b>arguments()</b>
        Title   : arguments()
        Usage   : $hashref = $primer3-&gt;arguments();
        Function: Describes the options that you can set through Bio::Tools::Run::Primer3,
                  with a brief (one line) description of what they are and their
                  default values
        Returns : A string (if an argument is supplied) or a reference to a hash.
        Args    : If supplied with an argument will return a string of its
                  description.
                  If no arguments are supplied, will return all the arguments as a
                  reference to a hash
        Notes   : Much of this is taken from the primer3 README file, and you should
                  read that file for a more detailed description.

   <b>version</b>
        Title   : version
        Usage   : $v = $prog-&gt;version();
        Function: Determine the version number of the program
        Example :
        Returns : float or undef
        Args    : none

   <b>_input_args()</b>
        Title   : _input_args()
        Usage   : an internal method to set the input arguments for Primer3
        Function: Define a hash with keys for each of the input arguments and values
                  as a short one line description
        Returns : A reference to a hash.
        Args    : None.
        Notes   : Much of this is taken from the primer3 README file, and you should
                  read that file for a more detailed description.

perl v5.34.0                                       2022-02-10                      <u>Bio::Tools::Run::<a href="../man3pm/Primer3.3pm.html">Primer3</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>