<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::SearchIO::exonerate - parser for Exonerate</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::SearchIO::exonerate - parser for Exonerate

</pre><h4><b>SYNOPSIS</b></h4><pre>
         # do not use this module directly, it is a driver for SearchIO

         use Bio::SearchIO;
         my $searchio = Bio::SearchIO-&gt;new(-file =&gt; 'file.exonerate',
                                          -format =&gt; 'exonerate');

         while( my $r = $searchio-&gt;next_result ) {
           print $r-&gt;query_name, "\n";
         }

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This is a driver for the SearchIO system for parsing Exonerate (Guy Slater) output.  You can get
       Exonerate at <a href="http://www.ebi.ac.uk/~guy/exonerate/">http://www.ebi.ac.uk/~guy/exonerate/</a> [until Guy puts up a Web reference,publication for
       it.]).

       An optional parameter -min_intron is supported by the new initialization method.  This is if you run
       Exonerate with a different minimum intron length (default is 30) the parser will be able to detect the
       difference between standard deletions and an intron.  Still some room to play with there that might cause
       this to get misinterpreted that has not been fully tested or explored.

       The VULGAR and CIGAR formats should be parsed okay now creating HSPs where appropriate (so merging match
       states where appropriate rather than breaking an HSP at each indel as it may have done in the past).  The
       GFF that comes from exonerate is still probably a better way to go if you are doing protein2genome or
       est2genome mapping.  For example you can see this script:

       ### TODO: Jason, this link is dead, do we have an updated one?
       <a href="http://fungal.genome.duke.edu/~jes12/software/scripts/process_exonerate_gff3.perl.txt">http://fungal.genome.duke.edu/~jes12/software/scripts/process_exonerate_gff3.perl.txt</a>

       If your report contains both CIGAR and VULGAR lines only the first one will processed for a given
       Query/Target pair.  If you preferentially want to use VULGAR or CIGAR add one of these options when
       initializing the SearchIO object.

           -cigar  =&gt; 1
       OR
           -vulgar =&gt; 1

       Or set them via these methods.

           $parser-&gt;<a href="../man1/cigar.1.html">cigar</a>(1)
       OR
           $parser-&gt;<a href="../man1/vulgar.1.html">vulgar</a>(1)

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Jason</b> <b>Stajich</b></h4><pre>
       Email jason-at-bioperl.org

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : my $obj = Bio::SearchIO::exonerate-&gt;new();
        Function: Builds a new Bio::SearchIO::exonerate object
        Returns : an instance of Bio::SearchIO::exonerate
        Args    : -min_intron =&gt; somewhat obsolete option, how to determine if a
                                 an indel is an intron or a local gap.  Use VULGAR
                                 rather than CIGAR to avoid this heuristic,default 30.
                  -cigar       =&gt; 1   set this to 1 if you want to parse
                                      CIGAR exclusively.
                  -vulgar      =&gt; 1   set this to 1 if you want to parse VULGAR
                                      exclusively, setting both to 1 will revert
                                      to the default behavior of just parsing the
                                      first line that it sees.

   <b>next_result</b>
        Title   : next_result
        Usage   : my $hit = $searchio-&gt;next_result;
        Function: Returns the next Result from a search
        Returns : Bio::Search::Result::ResultI object
        Args    : none

   <b>start_element</b>
        Title   : start_element
        Usage   : $eventgenerator-&gt;start_element
        Function: Handles a start element event
        Returns : none
        Args    : hashref with at least 2 keys 'Data' and 'Name'

   <b>end_element</b>
        Title   : start_element
        Usage   : $eventgenerator-&gt;end_element
        Function: Handles an end element event
        Returns : none
        Args    : hashref with at least 2 keys 'Data' and 'Name'

   <b>element</b>
        Title   : element
        Usage   : $eventhandler-&gt;element({'Name' =&gt; $name, 'Data' =&gt; $str});
        Function: Convience method that calls start_element, characters, end_element
        Returns : none
        Args    : Hash ref with the keys 'Name' and 'Data'

   <b>characters</b>
        Title   : characters
        Usage   : $eventgenerator-&gt;characters($str)
        Function: Send a character events
        Returns : none
        Args    : string

   <b>within_element</b>
        Title   : within_element
        Usage   : if( $eventgenerator-&gt;within_element($element) ) {}
        Function: Test if we are within a particular element
                  This is different than 'in' because within can be tested
                  for a whole block.
        Returns : boolean
        Args    : string element name

   <b>in_element</b>
        Title   : in_element
        Usage   : if( $eventgenerator-&gt;in_element($element) ) {}
        Function: Test if we are in a particular element
                  This is different than 'in' because within can be tested
                  for a whole block.
        Returns : boolean
        Args    : string element name

   <b>start_document</b>
        Title   : start_document
        Usage   : $eventgenerator-&gt;start_document
        Function: Handle a start document event
        Returns : none
        Args    : none

   <b>end_document</b>
        Title   : end_document
        Usage   : $eventgenerator-&gt;end_document
        Function: Handles an end document event
        Returns : Bio::Search::Result::ResultI object
        Args    : none

   <b>vulgar</b>
        Title   : vulgar
        Usage   : $obj-&gt;vulgar($newval)
        Function: Get/Set flag, do you want to build HSPs from VULGAR string?
        Returns : value of vulgar (a scalar)
        Args    : on set, new value (a scalar or undef, optional)

   <b>cigar</b>
        Title   : cigar
        Usage   : $obj-&gt;cigar($newval)
        Function: Get/Set boolean flag do you want to build HSPs from CIGAR strings?
        Returns : value of cigar (a scalar)
        Args    : on set, new value (a scalar or undef, optional)

perl v5.32.1                                       2021-08-15                      <u>Bio::SearchIO::<a href="../man3pm/exonerate.3pm.html">exonerate</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>