<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::SearchIO::Writer::HTMLResultWriter - write a Bio::Search::ResultI in HTML</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::SearchIO::Writer::HTMLResultWriter - write a Bio::Search::ResultI in HTML

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::SearchIO;
         use Bio::SearchIO::Writer::HTMLResultWriter;

         my $in = Bio::SearchIO-&gt;new(-format =&gt; 'blast',
                                    -file   =&gt; shift @ARGV);

         my $writer = Bio::SearchIO::Writer::HTMLResultWriter-&gt;new();
         my $out = Bio::SearchIO-&gt;new(-writer =&gt; $writer);
         $out-&gt;write_result($in-&gt;next_result);

         # to filter your output
         my $MinLength = 100; # need a variable with scope outside the method
         sub hsp_filter {
             my $hsp = shift;
             return 1 if $hsp-&gt;length('total') &gt; $MinLength;
         }
         sub result_filter {
             my $result = shift;
             return $hsp-&gt;num_hits &gt; 0;
         }

         my $writer = Bio::SearchIO::Writer::HTMLResultWriter-&gt;new
                            (-filters =&gt; { 'HSP' =&gt; \&amp;hsp_filter} );
         my $out = Bio::SearchIO-&gt;new(-writer =&gt; $writer);
         $out-&gt;write_result($in-&gt;next_result);

         # can also set the filter via the writer object
         $writer-&gt;filter('RESULT', \&amp;result_filter);

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This object implements the SearchWriterI interface which will produce a set of HTML for a specific
       Bio::Search::Report::ReportI interface.

       See Bio::SearchIO::SearchWriterI for more info on the filter method.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Jason</b> <b>Stajich</b></h4><pre>
       Email jason-at-bioperl-dot-org

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
       Gary Williams <a href="mailto:G.Williams@hgmp.mrc.ac.uk">G.Williams@hgmp.mrc.ac.uk</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : my $obj = Bio::SearchIO::Writer::HTMLResultWriter-&gt;new();
        Function: Builds a new Bio::SearchIO::Writer::HTMLResultWriter object
        Returns : Bio::SearchIO::Writer::HTMLResultWriter
        Args    : -filters =&gt; hashref with any or all of the keys (HSP HIT RESULT)
                  which have values pointing to a subroutine reference
                  which will expect to get a
                  -nucleotide_url =&gt; URL sprintf string base for the nt sequences
                  -protein_url =&gt; URL sprintf string base for the aa sequences
                  -no_wublastlinks =&gt; boolean. Do not display WU-BLAST lines
                                      even if they are parsed out.
                                      Links = (1)

   <b>remote_database_url</b>
        Title   : remote_database_url
        Usage   : $obj-&gt;remote_database_url($type,$newval)
        Function: This should return or set a string that contains a %s which can be
                  filled in with sprintf.
        Returns : value of remote_database_url
        Args    : $type - 'PROTEIN' or 'P' for protein URLS
                          'NUCLEOTIDE' or 'N' for nucleotide URLS
                  $value - new value to set [optional]

   <b>to_string</b>
        Purpose   : Produces data for each Search::Result::ResultI in a string.
                  : This is an abstract method. For some useful implementations,
                  : see ResultTableWriter.pm, HitTableWriter.pm,
                  : and HSPTableWriter.pm.
        Usage     : print $writer-&gt;to_string( $result_obj, @args );
        Argument  : $result_obj = A Bio::Search::Result::ResultI object
                  : @args = any additional arguments used by your implementation.
        Returns   : String containing data for each search Result or any of its
                  : sub-objects (Hits and HSPs).
        Throws    : n/a

   <b>hit_link_desc</b>
        Title   : hit_link_desc
        Usage   : $self-&gt;hit_link_desc(\&amp;link_function);
        Function: Get/Set the function which provides an HTML
                  link(s) for the given hit to be used
                  within the description section at the top of the BLAST report.
                  This allows a person reading the report within
                  a web browser to go to one or more database entries for
                  the given hit from the description section.
        Returns : Function reference
        Args    : Function reference
        See Also: L&lt;default_hit_link_desc()&gt;

   <b>default_hit_link_desc</b>
        Title   : default_hit_link_desc
        Usage   : $self-&gt;default_hit_link_desc($hit, $result)
        Function: Provides an HTML link(s) for the given hit to be used
                  within the description section at the top of the BLAST report.
                  This allows a person reading the report within
                  a web browser to go to one or more database entries for
                  the given hit from the description section.
        Returns : string containing HTML markup "&lt;a href...")

                  The default implementation returns an HTML link to the
                  URL supplied by the remote_database_url() method
                  and using the identifier supplied by the id_parser() method.
                  It will use the NCBI GI if present, and the accession if not.

        Args    : First argument is a Bio::Search::Hit::HitI
                  Second argument is a Bio::Search::Result::ResultI

       See Also: hit_link_align, remote_database, id_parser

   <b>hit_link_align</b>
        Title   : hit_link_align
        Usage   : $self-&gt;hit_link_align(\&amp;link_function);
        Function: Get/Set the function which provides an HTML link(s)
                  for the given hit to be used
                  within the HSP alignment section of the BLAST report.
                  This allows a person reading the report within
                  a web browser to go to one or more database entries for
                  the given hit from the alignment section.
        Returns : string containing HTML markup "&lt;a href...")

                  The default implementation delegates to hit_link_desc().

        Args    : First argument is a Bio::Search::Hit::HitI
                  Second argument is a Bio::Search::Result::ResultI

       See Also: hit_link_desc, remote_database, id_parser

   <b>hit_desc_line</b>
        Title   : hit_desc_line
        Usage   : $self-&gt;hit_desc_line(\&amp;link_function);
        Function: Get/Set the function which provides HTML for the description
                  information from a hit. This allows one to parse
                  the rest of the description and split up lines, add links, etc.
        Returns : Function reference
        Args    : Function reference
        See Also: L&lt;default_hit_link_desc()&gt;

   <b>default_hit_desc_line</b>
        Title   : default_hit_desc_line
        Usage   : $self-&gt;default_hit_desc_line($hit, $result)
        Function: Parses the description line information, splits based on the
                  hidden \x01 between independent descriptions, checks the lines for
                  possible web links, and adds HTML link(s) for the given hit to be
                  used.

        Returns : string containing HTML markup "&lt;a href...")
                  The default implementation returns an HTML link to the
                  URL supplied by the remote_database_url() method
                  and using the identifier supplied by the id_parser() method.
                  It will use the NCBI GI if present, and the accession if not.

        Args    : First argument is a Bio::Search::Hit::HitI
                  Second argument is a Bio::Search::Result::ResultI

       See Also: hit_link_align, remote_database, id_parser

   <b>start_report</b>
         Title   : start_report
         Usage   : $index-&gt;start_report( CODE )
         Function: Stores or returns the code to
                   write the start of the &lt;HTML&gt; block, the &lt;TITLE&gt; block
                   and the start of the &lt;BODY&gt; block of HTML.   Useful
                   for (for instance) specifying alternative
                   HTML if you are embedding the output in
                   an HTML page which you have already started.
                   (For example a routine returning a null string).
                   Returns \&amp;default_start_report (see below) if not
                   set.
         Example : $index-&gt;start_report( \&amp;my_start_report )
         Returns : ref to CODE if called without arguments
         Args    : CODE

   <b>default_start_report</b>
        Title   : default_start_report
        Usage   : $self-&gt;default_start_report($result)
        Function: The default method to call when starting a report.
        Returns : sting
        Args    : First argument is a Bio::Search::Result::ResultI

   <b>title</b>
        Title   : title
        Usage   : $self-&gt;title($CODE)

         Function: Stores or returns the code to provide HTML for the given
                   BLAST report that will appear at the top of the BLAST report
                   HTML output.  Useful for (for instance) specifying
                   alternative routines to write your own titles.
                   Returns \&amp;default_title (see below) if not
                   set.
         Example : $index-&gt;title( \&amp;my_title )
         Returns : ref to CODE if called without arguments
         Args    : CODE

   <b>default_title</b>
        Title   : default_title
        Usage   : $self-&gt;default_title($result)
        Function: Provides HTML for the given BLAST report that will appear
                  at the top of the BLAST report HTML output.
        Returns : string containing HTML markup
                  The default implementation returns &lt;CENTER&gt; &lt;H1&gt; HTML
                  containing text such as:
                  "Bioperl Reformatted HTML of BLASTP Search Report
                            for gi|1786183|gb|AAC73113.1|"
        Args    : First argument is a Bio::Search::Result::ResultI

   <b>introduction</b>
        Title   : introduction
        Usage   : $self-&gt;introduction($CODE)

         Function: Stores or returns the code to provide HTML for the given
                   BLAST report detailing the query and the
                   database information.
                   Useful for (for instance) specifying
                   routines returning alternative introductions.
                   Returns \&amp;default_introduction (see below) if not
                   set.
         Example : $index-&gt;introduction( \&amp;my_introduction )
         Returns : ref to CODE if called without arguments
         Args    : CODE

   <b>default_introduction</b>
        Title   : default_introduction
        Usage   : $self-&gt;default_introduction($result)
        Function: Outputs HTML to provide the query
                  and the database information
        Returns : string containing HTML
        Args    : First argument is a Bio::Search::Result::ResultI
                  Second argument is string holding literature citation

   <b>end_report</b>
        Title   : end_report
        Usage   : $self-&gt;end_report()
        Function: The method to call when ending a report, this is
                  mostly for cleanup for formats which require you to
                  have something at the end of the document (&lt;/BODY&gt;&lt;/HTML&gt;)
                  for HTML
        Returns : string
        Args    : none

   <b>id_parser</b>
         Title   : id_parser
         Usage   : $index-&gt;id_parser( CODE )
         Function: Stores or returns the code used by record_id to
                   parse the ID for record from a string.  Useful
                   for (for instance) specifying a different
                   parser for different flavours of FASTA file.
                   Returns \&amp;default_id_parser (see below) if not
                   set. If you supply your own id_parser
                   subroutine, then it should expect a fasta
                   description line.  An entry will be added to
                   the index for each string in the list returned.
         Example : $index-&gt;id_parser( \&amp;my_id_parser )
         Returns : ref to CODE if called without arguments
         Args    : CODE

   <b>default_id_parser</b>
         Title   : default_id_parser
         Usage   : $id = default_id_parser( $header )
         Function: The default Fasta ID parser for Fasta.pm
                   Returns $1 from applying the regexp /^&gt;\s*(\S+)/
                   to $header.
         Returns : ID string
                   The default implementation checks for NCBI-style
                   identifiers in the given string ('gi|12345|AA54321').
                   For these IDs, it extracts the GI and accession and
                   returns a two-element list of strings (GI, acc).
         Args    : a fasta header line string

   <b>algorithm_reference</b>
        Title   : algorithm_reference
        Usage   : my $reference = $writer-&gt;algorithm_reference($result);
        Function: Returns the appropriate Bibliographic reference for the
                  algorithm format being produced
        Returns : String
        Args    : L&lt;Bio::Search::Result::ResultI&gt; to reference

   <b>Methods</b> <b>Bio::SearchIO::SearchWriterI</b>
       Bio::SearchIO::SearchWriterI inherited methods.

   <b>filter</b>
        Title   : filter
        Usage   : $writer-&gt;filter('hsp', \&amp;hsp_filter);
        Function: Filter out either at HSP,Hit,or Result level
        Returns : none
        Args    : string =&gt; data type,
                  CODE reference

   <b>no_wublastlinks</b>
        Title   : no_wublastlinks
        Usage   : $obj-&gt;no_wublastlinks($newval)
        Function: Get/Set boolean value regarding whether or not to display
                  Link = (1)
                  type output in the report output (WU-BLAST only)
        Returns : boolean
        Args    : on set, new boolean value (a scalar or undef, optional)

perl v5.32.1                                       2021-08-15              <u>Bio::SearchIO:...<a href="../man3pm/TMLResultWriter.3pm.html">TMLResultWriter</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>