<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::SeqPattern - represent a sequence pattern or motif</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::SeqPattern - represent a sequence pattern or motif

</pre><h4><b>SYNOPSIS</b></h4><pre>
        use Bio::Tools::SeqPattern;

        my $pat1     = 'T[GA]AA...TAAT';
        my $pattern1 = Bio::Tools::SeqPattern-&gt;new(-SEQ =&gt;$pat1, -TYPE =&gt;'Dna');

        my $pat2     = '[VILM]R(GXX){3,2}...[^PG]';
        my $pattern2 = Bio::Tools::SeqPattern-&gt;new(-SEQ =&gt;$pat2, -TYPE =&gt;'Amino');

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::Tools::SeqPattern module encapsulates generic data and methods for manipulating regular expressions
       describing nucleic or amino acid sequence patterns (a.k.a, "motifs"), such as the ones produced by
       Bio::Tools::IUPAC.

       Bio::Tools::SeqPattern is a concrete class that inherits from Bio::Seq.

       This class grew out of a need to have a standard module for doing routine tasks with sequence patterns
       such as:

         -- Forming a reverse-complement version of a nucleotide sequence pattern
         -- Expanding patterns containing ambiguity codes
         -- Checking for invalid regexp characters
         -- Untainting yet preserving special characters in the pattern

       Other features to look for in the future:

         -- Full pattern syntax checking
         -- Conversion between expanded and condensed forms of the pattern

</pre><h4><b>MOTIVATIONS</b></h4><pre>
       A key motivation for Bio::Tools::SeqPattern is to have a way to generate a reverse complement of a
       nucleotide sequence pattern.  This makes possible simultaneous pattern matching on both sense and anti-
       sense strands of a query sequence.

       In principle, one could do such a search more inefficiently by testing against both sense and anti-sense
       versions of a sequence.  It is entirely equivalent to test a regexp containing both sense and anti-sense
       versions of the *pattern* against one copy of the sequence.  The latter approach is much more efficient
       since:

          1) You need only one copy of the sequence.
          2) Only one regexp is executed.
          3) Regexp patterns are typically much smaller than sequences.

       Patterns can be quite complex and it is often difficult to generate the reverse complement pattern. The
       Bioperl SeqPattern.pm addresses this problem, providing a convenient set of tools for working with
       biological sequence regular expressions.

       Not all patterns have been tested. If you discover a pattern that is not handled properly by
       Bio::Tools::SeqPattern.pm, please send me some email (<a href="mailto:sac@bioperl.org">sac@bioperl.org</a>). Thanks.

</pre><h4><b>OTHER</b> <b>FEATURES</b></h4><pre>
   <b>Extended</b> <b>Alphabet</b> <b>Support</b>
       This module supports the same set of ambiguity codes for nucleotide sequences as supported by Bio::Seq.
       These ambiguity codes define the behavior or the expand method.

        ------------------------------------------
        Symbol       Meaning      Nucleic Acid
        ------------------------------------------
         A            A           (A)denine
         C            C           (C)ytosine
         G            G           (G)uanine
         T            T           (T)hymine
         U            U           (U)racil
         M          A or C        a(M)ino group
         R          A or G        pu(R)ine
         W          A or T        (W)eak bond
         S          C or G        (S)trong bond
         Y          C or T        p(Y)rimidine
         K          G or T        (K)eto group
         V        A or C or G
         H        A or C or T
         D        A or G or T
         B        C or G or T
         X      G or A or T or C
         N      G or A or T or C
         .      G or A or T or C

        ------------------------------------------
        Symbol           Meaning
        ------------------------------------------
        A        Alanine
        C        Cysteine
        D        Aspartic Acid
        E        Glutamic Acid
        F        Phenylalanine
        G        Glycine
        H        Histidine
        I        Isoleucine
        K        Lysine
        L        Leucine
        M        Methionine
        N        Asparagine
        P        Proline
        Q        Glutamine
        R        Arginine
        S        Serine
        T        Threonine
        V        Valine
        W        Tryptophan
        Y        Tyrosine

        B        Aspartic Acid, Asparagine
        Z        Glutamic Acid, Glutamine
        X        Any amino acid
        .        Any amino acid

   <b>Multiple</b> <b>Format</b> <b>Support</b>
       Ultimately, this module should be able to build SeqPattern.pm objects using a variety of pattern formats
       such as ProSite, Blocks, Prints, GCG, etc.  Currently, this module only supports patterns using a grep-
       like syntax.

</pre><h4><b>USAGE</b></h4><pre>
       A simple demo script called seq_pattern.pl is included in the examples/ directory of the central Bioperl
       distribution.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Bio::Seq - Lightweight sequence object.

       Bio::Tools::IUPAC - The IUPAC code for degenerate residues and their conversion to a regular expression.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules.  Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution. Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b></h4><pre>
       Steve Chervitz, sac-at-bioperl.org

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright (c) 1997-8 Steve Chervitz. All Rights Reserved.  This module is free software; you can
       redistribute it and/or modify it under the same terms as Perl itself.

</pre><h4><b>new</b></h4><pre>
        Title     : new
        Usage     : my $seqpat = Bio::Tools::SeqPattern-&gt;new();
        Purpose   : Verifies that the type is correct for superclass (Bio::Seq.pm)
                  : and calls superclass constructor last.
        Returns   : n/a
        Argument  : Parameters passed to new()
        Throws    : Exception if the pattern string (seq) is empty.
        Comments  : The process of creating a new SeqPattern.pm object
                  : ensures that the pattern string is untained.

       See Also   : Bio::Root::Root::new,
                    Bio::Seq::_initialize

</pre><h4><b>alphabet_ok</b></h4><pre>
        Title     : alphabet_ok
        Usage     : $mypat-&gt;alphabet_ok;
        Purpose   : Checks for invalid regexp characters.
                  : Overrides Bio::Seq::alphabet_ok() to allow
                  : additional regexp characters ,.*()[]&lt;&gt;{}^$
                  : in addition to the standard genetic alphabet.
                  : Also untaints the pattern and sets the sequence
                  : object's sequence to the untained string.
        Returns   : Boolean (1 | 0)
        Argument  : n/a
        Throws    : Exception if the pattern contains invalid characters.
        Comments  : Does not call the superclass method.
                  : Actually permits any alphanumeric, not just the
                  : standard genetic alphabet.

</pre><h4><b>expand</b></h4><pre>
        Title     : expand
        Usage     : $seqpat_object-&gt;expand();
        Purpose   : Expands the sequence pattern using special ambiguity codes.
        Example   : $pat = $seq_pat-&gt;expand();
        Returns   : String containing fully expanded sequence pattern
        Argument  : n/a
        Throws    : Exception if sequence type is not recognized
                  : (i.e., is not one of [DR]NA, Amino)

       See Also   : "Extended Alphabet Support", _expand_pep(), _expand_nuc()

</pre><h4><b>_expand_pep</b></h4><pre>
        Title     : _expand_pep
        Usage     : n/a; automatically called by expand()
        Purpose   : Expands peptide patterns
        Returns   : String (the expanded pattern)
        Argument  : String (the unexpanded pattern)
        Throws    : n/a

       See Also   : expand(), _expand_nuc()

</pre><h4><b>_expand_nuc</b></h4><pre>
        Title     : _expand_nuc
        Purpose   : Expands nucleotide patterns
        Returns   : String (the expanded pattern)
        Argument  : String (the unexpanded pattern)
        Throws    : n/a

       See Also   : expand(), _expand_pep()

</pre><h4><b>revcom</b></h4><pre>
        Title     : revcom
        Usage     : revcom([1]);
        Purpose   : Forms a pattern capable of recognizing the reverse complement
                  : version of a nucleotide sequence pattern.
        Example   : $pattern_object-&gt;revcom();
                  : $pattern_object-&gt;<a href="../man1/revcom.1.html">revcom</a>(1); ## returns expanded rev complement pattern.
        Returns   : Object reference for a new Bio::Tools::SeqPattern containing
                  : the revcom of the current pattern as its sequence.
        Argument  : (1) boolean (optional) (default= false)
                  :     true : expand the pattern before rev-complementing.
                  :     false: don't expand pattern before or after rev-complementing.
        Throws    : Exception if called for amino acid sequence pattern.
        Comments  : This method permits the simultaneous searching of both
                  : sense and anti-sense versions of a nucleotide pattern
                  : by means of a grep-type of functionality in which any
                  : number of patterns may be or-ed into the recognition
                  : pattern.
                  : Overrides Bio::Seq::revcom() and calls it first thing.
                  : The order of _fixpat() calls is critical.

       See Also   : Bio::Seq::revcom, "_fixpat_1", "_fixpat_2", "_fixpat_3", "_fixpat_4", "_fixpat_5"

</pre><h4><b>backtranslate</b></h4><pre>
        Title     : backtranslate
        Usage     : backtranslate();
        Purpose   : Produce a degenerate oligonucleotide whose translation would produce
                  : the original protein motif.
        Example   : $pattern_object-&gt;backtranslate();
        Returns   : Object reference for a new Bio::Tools::SeqPattern containing
                  : the reverse translation of the current pattern as its sequence.
        Throws    : Exception if called for nucleotide sequence pattern.

</pre><h4><b>_fixpat_1</b></h4><pre>
        Title     : _fixpat_1
        Usage     : n/a; called automatically by revcom()
        Purpose   : Utility method for revcom()
                  : Converts all {7,5} --&gt; {5,7}     (Part I)
                  :           and [T^] --&gt; [^T]      (Part II)
                  :           and *N   --&gt; N*        (Part III)
        Returns   : String (the new, partially reversed pattern)
        Argument  : String (the expanded pattern)
        Throws    : n/a

       See Also   : revcom()

</pre><h4><b>_fixpat_2</b></h4><pre>
        Title     : _fixpat_2
        Usage     : n/a; called automatically by revcom()
        Purpose   : Utility method for revcom()
                  : Converts all {5,7}Y ---&gt; Y{5,7}
                  :          and {10,}. ---&gt; .{10,}
        Returns   : String (the new, partially reversed pattern)
        Argument  : String (the expanded, partially reversed pattern)
        Throws    : n/a

       See Also   : revcom()

</pre><h4><b>_fixpat_3</b></h4><pre>
        Title     : _fixpat_3
        Usage     : n/a; called automatically by revcom()
        Purpose   : Utility method for revcom()
                  : Converts all {5,7}(XXX) ---&gt; (XXX){5,7}
        Returns   : String (the new, partially reversed pattern)
        Argument  : String (the expanded, partially reversed pattern)
        Throws    : n/a

       See Also   : revcom()

</pre><h4><b>_fixpat_4</b></h4><pre>
        Title     : _fixpat_4
        Usage     : n/a; called automatically by revcom()
        Purpose   : Utility method for revcom()
                  : Converts all {5,7}[XXX] ---&gt; [XXX]{5,7}
        Returns   : String (the new, partially reversed pattern)
        Argument  : String (the expanded, partially reversed  pattern)
        Throws    : n/a

       See Also   : revcom()

</pre><h4><b>_fixpat_5</b></h4><pre>
        Title     : _fixpat_5
        Usage     : n/a; called automatically by revcom()
        Purpose   : Utility method for revcom()
                  : Converts all *[XXX]  ---&gt; [XXX]*
                  :          and *(XXX)  ---&gt; (XXX)*
        Returns   : String (the new, partially reversed pattern)
        Argument  : String (the expanded, partially reversed pattern)
        Throws    : n/a

       See Also   : revcom()

</pre><h4><b>_fixpat_6</b></h4><pre>
        Title     : _fixpat_6
        Usage     : n/a; called automatically by revcom()
        Purpose   : Utility method for revcom()
                  : Converts all ?Y{5,7}  ---&gt; Y{5,7}?
                  :          and ?(XXX){5,7}  ---&gt; (XXX){5,7}?
                  :          and ?[XYZ]{5,7}  ---&gt; [XYZ]{5,7}?
        Returns   : String (the new, partially reversed pattern)
        Argument  : String (the expanded, partially reversed pattern)
        Throws    : n/a

       See Also   : revcom()

   <b>str</b>
        Title   : str
        Usage   : $obj-&gt;str($newval)
        Function:
        Returns : value of str
        Args    : newvalue (optional)

   <b>type</b>
        Title   : type
        Usage   : $obj-&gt;type($newval)
        Function:
        Returns : value of type
        Args    : newvalue (optional)

</pre><h4><b>FOR</b> <b>DEVELOPERS</b> <b>ONLY</b></h4><pre>
   <b>Data</b> <b>Members</b>
       Information about the various data members of this module is provided for those wishing to modify or
       understand the code. Two things to bear in mind:

       1 Do NOT rely on these in any code outside of this module.
         All data members are prefixed with an underscore to signify that they are private.  Always use accessor
         methods. If the accessor doesn't exist or is inadequate, create or modify an accessor (and let me know,
         too!).

       2 This documentation may be incomplete and out of date.
         It  is  easy for this documentation to become obsolete as this module is still evolving.  Always double
         check this info and search for members not described here.

       An instance of Bio::Tools::RestrictionEnzyme.pm is a blessed reference to a hash containing all  or  some
       of the following fields:

        FIELD          VALUE
        ------------------------------------------------------------------------
        _rev     : The corrected reverse complement of the fully expanded pattern.

        INHERITED DATA MEMBERS:

        _seq     : (From Bio::Seq.pm) The original, unexpanded input sequence after untainting.
        _type    : (From Bio::Seq.pm) 'Dna' or 'Amino'

perl v5.32.1                                       2021-08-15                        <u>Bio::Tools::<a href="../man3pm/SeqPattern.3pm.html">SeqPattern</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>