<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::DB::Biblio::eutils - access to PubMed's bibliographic query service</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-biblio-perl">libbio-biblio-perl_1.70-2_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::DB::Biblio::eutils - access to PubMed's bibliographic query service

</pre><h4><b>VERSION</b></h4><pre>
       version 1.70

</pre><h4><b>SYNOPSIS</b></h4><pre>
       Do not use this object directly, it is recommended to access it and use it through the <u>Bio::Biblio</u>
       module:

         use Bio::Biblio;
         use Bio::Biblio::IO;

         my $biblio = Bio::Biblio-&gt;new( -access =&gt; 'eutils' );
         $biblio-&gt;find("10336996");
         my $xml = $biblio-&gt;get_next;
         my $io = Bio::Biblio::IO-&gt;new( -data =&gt; $xml,
                                        -format =&gt; 'medlinexml' );
         my $article = $io-&gt;next_bibref();

       The main documentation details are to be found in Bio::DB::BiblioI.

</pre><h4><b>ATTRIBUTES</b></h4><pre>
   <b>db</b>
        Title   : db
        Usage   : $obj-&gt;db($newval)
        Function: specifies the database to search.  valid values are:

                  pubmed, pmc, journals

                  it is also possible to add the following, and i will do
                  so on request:

                  genome, nucleotide, protein, popset, snp, sequence, taxonomy

                  pubmed is default.

        Returns : value of db (a scalar)
        Args    : on set, new value (a scalar or undef, optional)

   <b>has_next</b>
         Title   : has_next
         Usage   : $has_next = $biblio-&gt;has_next();
         Function: check to see if there are more items to be retrieved
         Returns : 1 on true, undef on false
         Args    : none

</pre><h4><b>METHODS</b></h4><pre>
   <b>get_collection_id</b>
         Title   : get_collection_id
         Usage   : $id = $biblio-&gt;get_collection_id();
         Function: returns WebEnv value from ESearch
         Returns : ESearch WebEnv value as a string
         Args    : none

   <b>reset_retrieval</b>
         Title   : reset_retrieval
         Usage   : $biblio-&gt;reset_retrieval();
         Function: reset cursor in id list, see cursor()
         Returns : 1
         Args    : none

   <b>get_next</b>
         Title   : get_next
         Usage   : $xml = $biblio-&gt;get_next();
         Function: return next record as xml
         Returns : an xml string
         Args    : none

   <b>get_more</b>
         Title   : get_more
         Usage   : $xml = $biblio-&gt;get_more($more);
         Function: returns next $more records concatenated
         Returns : a string containing multiple xml documents
         Args    : an integer representing how many records to retrieve

   <b>find</b>
         Title   : find
         Usage   : $biblio = $biblio-&gt;find($pubmed_query_phrase);
         Function: perform a PubMed query using Entrez ESearch
         Returns : a reference to the object on which the method was called
         Args    : a PubMed query phrase.  See
                   <a href="http://eutils.ncbi.nlm.nih.gov/entrez/query/static/help/pmhelp.html">http://eutils.ncbi.nlm.nih.gov/entrez/query/static/help/pmhelp.html</a>
                   for help on how to construct a query.

   <b>get_all_ids</b>
         Title   : get_all_ids
         Usage   : @ids = $biblio-&gt;get_all_ids();
         Function: return a list of PubMed ids resulting from call to find()
         Returns : a list of PubMed ids, or an empty list
         Args    : none

   <b>get_all</b>
         Title   : get_all
         Usage   : $xml = $biblio-&gt;get_all();
         Function: retrieve all records from query
         Returns : return a large concatenated string of PubMed xml documents
         Args    : none

   <b>get_vocabulary_names</b>
         Title   : get_vocabulary_names
         Usage   : do not use
         Function: no-op.  this is here only for interface compatibility
         Returns : empty arrayref
         Args    : none

   <b>get_entry_description</b>
         Title   : get_entry_description
         Usage   : do not use
         Function: no-op.  this is here only for interface compatibility
         Returns : undef
         Args    : none

   <b>get_all_values</b>
         Title   : get_all_values
         Usage   : do not use
         Function: no-op.  this is here only for interface compatibility
         Returns : undef
         Args    : none

   <b>get_all_entries</b>
         Title   : get_all_entries
         Usage   : do not use
         Function: no-op.  this is here only for interface compatibility
         Returns : undef
         Args    : none

</pre><h4><b>INTERNAL</b> <b>METHODS</b></h4><pre>
   <b>_initialize</b>
        Usage   : my $obj = Bio::Biblio-&gt;new(-access =&gt; 'eutils' ...);
                  (_initialize is internally called from this constructor)
        Returns : 1 on success
        Args    : none

       This is an actual <b>new()</b> method (except for the real object creation and its blessing which is done in the
       parent class Bio::Root::Root in method _create_object).

       Note that this method is called always as an <u>object</u> method (never as a <u>class</u> method) - and that the
       object who calls this method may already be partly initiated (from Bio::Biblio::new method); so if you
       need to do some tricks with the 'class invocation' you need to change Bio::Biblio::new method, not this
       one.

   <b>exists</b>
         Title   : exists
         Usage   : do not use
         Function: no-op.  this is here only for interface compatibility
         Returns : undef
         Args    : none

   <b>destroy</b>
         Title   : destroy
         Usage   : do not use
         Function: no-op.  this is here only for interface compatibility
         Returns : undef
         Args    : none

   <b>contains</b>
         Title   : contains
         Usage   : do not use
         Function: no-op.  this is here only for interface compatibility
         Returns : undef
         Args    : none

   <b>cursor</b>
         Title   : cursor
         Usage   : $obj-&gt;cursor($newval)
         Function: holds position in reference collection
         Returns : value of cursor (a scalar)
         Args    : on set, new value (a scalar or undef, optional)

   <b>twig</b>
         Title   : twig
         Usage   : $obj-&gt;twig($newval)
         Function: holds an XML::Twig instance.
         Returns : value of twig (a scalar)
         Args    : on set, new value (a scalar or undef, optional)

   <b>ids</b>
         Title   : ids
         Usage   : $obj-&gt;ids($newval)
         Function: store pubmed ids resulting from find() query
         Returns : value of ids (a scalar)
         Args    : on set, new value (a scalar or undef, optional)

   <b>collection_id</b>
         Title   : collection_id
         Usage   : $obj-&gt;collection_id($newval)
         Function:
         Returns : value of collection_id (a scalar)
         Args    : on set, new value (a scalar or undef, optional)

   <b>count</b>
         Title   : count
         Usage   : $obj-&gt;count($newval)
         Function:
         Returns : value of count (a scalar)
         Args    : on set, new value (a scalar or undef, optional)

   <b>query_key</b>
         Title   : query_key
         Usage   : $obj-&gt;query_key($newval)
         Function: holds query_key from ESearch document
         Returns : value of query_key (a scalar)
         Args    : on set, new value (a scalar or undef, optional)

</pre><h4><b>BUGS</b> <b>AND</b> <b>LIMITATIONS</b></h4><pre>
           * More testing and debugging needed to ensure that returned citations are properly transferred even
           if they contain foreign characters.  * Maximum record count (MAX_RECORDS) returned currently hard
           coded to 100K.  * Biblio retrieval methods should be more tightly integrated with Bio::Biblio::Ref
           and Bio::DB::MeSH.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       •   Pub Med Help

           <a href="http://eutils.ncbi.nlm.nih.gov/entrez/query/static/help/pmhelp.html">http://eutils.ncbi.nlm.nih.gov/entrez/query/static/help/pmhelp.html</a>

       •   Entrez Utilities

           <a href="http://eutils.ncbi.nlm.nih.gov/entrez/query/static/eutils_help.html">http://eutils.ncbi.nlm.nih.gov/entrez/query/static/eutils_help.html</a>

       •   Example code

           <u>eg/biblio-eutils.pl</u>

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>lists</b>
       User  feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list: <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be  able  look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>bugs</b>
       Report  bugs  to  the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://redmine.open-bio.org/projects/bioperl/

</pre><h4><b>LEGAL</b></h4><pre>
   <b>Authors</b>
       Allen Day &lt;<a href="mailto:allenday@ucla.edu">allenday@ucla.edu</a>&gt;

   <b>Copyright</b> <b>and</b> <b>License</b>
       This software is Copyright (c) by 2004 Allen Day, University of  California,  Los  Angeles  and  released
       under the license of the same terms as the perl 5 programming language system itself

perl v5.36.0                                       2023-02-14                       <u>Bio::DB::Biblio::<a href="../man3pm/eutils.3pm.html">eutils</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>