<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Perl - Functional access to BioPerl for people who don't know objects</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-procedural-perl">libbio-procedural-perl_1.7.4-5_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Perl - Functional access to BioPerl for people who don't know objects

</pre><h4><b>VERSION</b></h4><pre>
       version 1.7.4

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::Perl;

         # will guess file format from extension
         $seq_object = read_sequence($filename);

         # forces genbank format
         $seq_object = read_sequence($filename,'genbank');

         # reads an array of sequences
         @seq_object_array = read_all_sequences($filename,'fasta');

         # sequences are Bio::Seq objects, so the following methods work
         # for more info see Bio::Seq, or do 'perldoc Bio/Seq.pm'

         print "Sequence name is ",$seq_object-&gt;display_id,"\n";
         print "Sequence acc  is ",$seq_object-&gt;accession_number,"\n";
         print "First 5 bases is ",$seq_object-&gt;subseq(1,5),"\n";

         # get the whole sequence as a single string

         $sequence_as_a_string = $seq_object-&gt;seq();

         # writing sequences

         write_sequence("&gt;$filename",'genbank',$seq_object);

         write_sequence("&gt;$filename",'genbank',@seq_object_array);

         # making a new sequence from just a string

         $seq_object = new_sequence("ATTGGTTTGGGGACCCAATTTGTGTGTTATATGTA",
             "myname","AL12232");

         # getting a sequence from a database (assumes internet connection)

         $seq_object = get_sequence('swissprot',"ROA1_HUMAN");

         $seq_object = get_sequence('embl',"AI129902");

         $seq_object = get_sequence('genbank',"AI129902");

         # BLAST a sequence (assumes an internet connection)

         $blast_report = blast_sequence($seq_object);

         write_blast("&gt;blast.out",$blast_report);

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Easy first time access to BioPerl via functions.

   <b>read_sequence</b>
        Title   : read_sequence
        Usage   : $seq = read_sequence('sequences.fa')
                  $seq = read_sequence($filename,'genbank');

                  # pipes are fine
                  $seq = read_sequence("my_fetching_program $id |",'fasta');

        Function: Reads the top sequence from the file. If no format is given, it will
                  try to guess the format from the filename. If a format is given, it
                  forces that format. The filename can be any valid perl open() string
                  - in particular, you can put in pipes

        Returns : A Bio::Seq object. A quick synopsis:
                  $seq_object-&gt;display_id - name of the sequence
                  $seq_object-&gt;seq        - sequence as a string

        Args    : Two strings, first the filename - any Perl open() string is ok
                  Second string is the format, which is optional

       For more information on Seq objects see Bio::Seq.

   <b>read_all_sequences</b>
        Title   : read_all_sequences
        Usage   : @seq_object_array = read_all_sequences($filename);
                  @seq_object_array = read_all_sequences($filename,'genbank');

        Function: Just as the function above, but reads all the sequences in the
                  file and loads them into an array.

                  For very large files, you will run out of memory. When this
                  happens, you've got to use the SeqIO system directly (this is
                  not so hard! Don't worry about it!).

        Returns : array of Bio::Seq objects

        Args    : two strings, first the filename (any open() string is ok)
                  second the format (which is optional)

       See Bio::SeqIO and Bio::Seq for more information

   <b>write_sequence</b>
        Title   : write_sequence
        Usage   : write_sequence("&gt;new_file.gb",'genbank',$seq)
                  write_sequence("&gt;new_file.gb",'genbank',@array_of_sequence_objects)

        Function: writes sequences in the specified format

        Returns : true

        Args    : filename as a string, must provide an open() output file
                  format as a string
                  one or more sequence objects

   <b>new_sequence</b>
        Title   : new_sequence
        Usage   : $seq_obj = new_sequence("GATTACA", "kino-enzyme");

        Function: Construct a sequency object from sequence string
        Returns : A Bio::Seq object

        Args    : sequence string
                  name string (optional, default "no-name-for-sequence")
                  accession - accession number (optional, no default)

   <b>blast_sequence</b>
        Title   : blast_sequence
        Usage   : $blast_result = blast_sequence($seq)
                  $blast_result = blast_sequence('MFVEGGTFASEDDDSASAEDE');

        Function: If the computer has Internet accessibility, blasts
                  the sequence using the NCBI BLAST server against nrdb.

                  It chooses the flavour of BLAST on the basis of the sequence.

                  This function uses Bio::Tools::Run::RemoteBlast, which itself
                  use Bio::SearchIO - as soon as you want to know more, check out
                  these modules
        Returns : Bio::Search::Result::GenericResult.pm

        Args    : Either a string of protein letters or nucleotides, or a
                  Bio::Seq object

   <b>write_blast</b>
        Title   : write_blast
        Usage   : write_blast($filename,$blast_report);

        Function: Writes a BLAST result object (or more formally
                  a SearchIO result object) out to a filename
                  in BLAST-like format

        Returns : none

        Args    : filename as a string
                  Bio::SearchIO::Results object

   <b>get_sequence</b>
        Title   : get_sequence
        Usage   : $seq_object = get_sequence('swiss',"ROA1_HUMAN");

        Function: If the computer has Internet access this method gets
                  the sequence from Internet accessible databases. Currently
                  this supports Swissprot ('swiss'), EMBL ('embl'), GenBank
                  ('genbank'), GenPept ('genpept'), and RefSeq ('refseq').

                  Swissprot and EMBL are more robust than GenBank fetching.

                  If the user is trying to retrieve a RefSeq entry from
                  GenBank/EMBL, the query is silently redirected.

        Returns : A Bio::Seq object

        Args    : database type - one of swiss, embl, genbank, genpept, or
                  refseq

   <b>translate</b>
        Title   : translate
        Usage   : $seqobj = translate($seq_or_string_scalar)

        Function: translates a DNA sequence object OR just a plain
                  string of DNA to amino acids
        Returns : A Bio::Seq object

        Args    : Either a sequence object or a string of
                  just DNA sequence characters

   <b>translate_as_string</b>
        Title   : translate_as_string
        Usage   : $seqstring = translate_as_string($seq_or_string_scalar)

        Function: translates a DNA sequence object OR just a plain
                  string of DNA to amino acids
        Returns : A string of just amino acids

        Args    : Either a sequence object or a string of
                  just DNA sequence characters

   <b>reverse_complement</b>
        Title   : reverse_complement
        Usage   : $seqobj = reverse_complement($seq_or_string_scalar)

        Function: reverse complements a string or sequence argument
                  producing a Bio::Seq - if you want a string, you
                  can use reverse_complement_as_string
        Returns : A Bio::Seq object

        Args    : Either a sequence object or a string of
                  just DNA sequence characters

   <b>revcom</b>
        Title   : revcom
        Usage   : $seqobj = revcom($seq_or_string_scalar)

        Function: reverse complements a string or sequence argument
                  producing a Bio::Seq - if you want a string, you
                  can use reverse_complement_as_string

                  This is an alias for reverse_complement
        Returns : A Bio::Seq object

        Args    : Either a sequence object or a string of
                  just DNA sequence characters

   <b>reverse_complement_as_string</b>
        Title   : reverse_complement_as_string
        Usage   : $string = reverse_complement_as_string($seq_or_string_scalar)

        Function: reverse complements a string or sequence argument
                  producing a string
        Returns : A string of DNA letters

        Args    : Either a sequence object or a string of
                  just DNA sequence characters

   <b>revcom_as_string</b>
        Title   : revcom_as_string
        Usage   : $string = revcom_as_string($seq_or_string_scalar)

        Function: reverse complements a string or sequence argument
                  producing a string
        Returns : A string of DNA letters

        Args    : Either a sequence object or a string of
                  just DNA sequence characters

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>               - General discussion
         https://bioperl.org/Support.html    - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list: <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u> rather than to
       the module maintainer directly. Many experienced and reponsive experts will be able look at the problem
       and quickly address it. Please include a thorough description of the problem with code and data examples
       if at all possible.

   <b>Reporting</b> <b>bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bio-procedural/issues

</pre><h4><b>AUTHOR</b></h4><pre>
       Ewan Birney &lt;<a href="mailto:birney@ebi.ac.uk">birney@ebi.ac.uk</a>&gt;

</pre><h4><b>COPYRIGHT</b></h4><pre>
       This software is copyright (c) by many people (see the individual modules for their copyright holders).

       This software is available under the same terms as the perl 5 programming language system itself.

perl v5.40.1                                       2025-04-09                                     <u>Bio::<a href="../man3pm/Perl.3pm.html">Perl</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>