<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Variation::AAChange - Sequence change class for polypeptides</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-variation-perl">libbio-variation-perl_1.7.5-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Variation::AAChange - Sequence change class for polypeptides

</pre><h4><b>SYNOPSIS</b></h4><pre>
          $aamut = Bio::Variation::AAChange-&gt;new
              ('-start'         =&gt; $start,
               '-end'           =&gt; $end,
               '-length'        =&gt; $len,
               '-proof'         =&gt; $proof,
               '-isMutation'    =&gt; 1,
               '-mut_number'    =&gt; $mut_number
               );

          my $a1 = Bio::Variation::Allele-&gt;new;
          $a1-&gt;seq($ori) if $ori;
          $aamut-&gt;allele_ori($a1);
          my $a2 = Bio::Variation::Allele-&gt;new;
          $a2-&gt;seq($mut) if $mut;
          $aachange-&gt;add_Allele($a2);
          $aachange-&gt;allele_mut($a2);

          print  "\n";

          # add it to a SeqDiff container object
          $seqdiff-&gt;add_Variant($rnachange);

          # and create links to and from RNA level variant objects
          $aamut-&gt;RNAChange($rnachange);
          $rnachange-&gt;AAChange($rnachange);

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The instantiable class Bio::Variation::RNAChange describes basic sequence changes at polypeptide  level.
       It uses methods defined in superclass Bio::Variation::VariantI, see Bio::Variation::VariantI for details.

       If the variation described by a AAChange object has a known Bio::Variation::RNAAChange object, create the
       link with method <b>AAChange()</b>. See Bio::Variation::AAChange for more information.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing lists  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Heikki</b> <b>Lehvaslaiho</b></h4><pre>
       Email:  heikki-at-bioperl-dot-org

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>RNAChange</b>
        Title   : RNAChange
        Usage   : $mutobj = $self-&gt;RNAChange;
                : $mutobj = $self-&gt;RNAChange($objref);
        Function: Returns or sets the link-reference to a mutation/change object.
                  If there is no link, it will return undef
        Returns : an obj_ref or undef

   <b>label</b>
        Title   : label
        Usage   : $obj-&gt;label();
        Function:

                   Sets and returns mutation event label(s).  If value is not
                   set, or no argument is given returns false.  Each
                   instantiable subclass of L&lt;Bio::Variation::VariantI&gt; needs
                   to implement this method. Valid values are listed in
                   'Mutation event controlled vocabulary' in
                   <a href="http://www.ebi.ac.uk/mutations/recommendations/mutevent.html">http://www.ebi.ac.uk/mutations/recommendations/mutevent.html</a>.

        Example :
        Returns : string
        Args    : string

   <b>similarity_score</b>
        Title   : similarity_score
        Usage   : $self-&gt;similarity_score
        Function: Measure for evolutionary conservativeness
                  of single amino substitutions. Uses BLOSUM62.
                  Negative numbers are noncoservative changes.
        Returns : integer, undef if not single amino acid change

   <b>trivname</b>
        Title   : trivname
        Usage   : $self-&gt;trivname
        Function:

                  Given a Bio::Variation::AAChange object with linked
                  Bio::Variation::RNAChange and Bio::Variation::DNAMutation
                  objects, this subroutine creates a string corresponding to
                  the 'trivial name' of the mutation. Trivial name is
                  specified in Antonorakis &amp; MDI Nomenclature Working Group:
                  Human Mutation 11:1-3, 1998.

        Returns : string

perl v5.32.1                                       2021-12-02                      <u>Bio::Variation::<a href="../man3pm/AAChange.3pm.html">AAChange</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>