<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tree::DistanceFactory - Construct a tree using distance based methods</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tree::DistanceFactory - Construct a tree using distance based methods

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::Tree::DistanceFactory;
         use Bio::AlignIO;
         use Bio::Align::DNAStatistics;
         my $tfactory = Bio::Tree::DistanceFactory-&gt;new(-method =&gt; "NJ");
         my $stats    = Bio::Align::DNAStatistics-&gt;new();

         my $alnin    = Bio::AlignIO-&gt;new(-format =&gt; 'clustalw',
                                          -file   =&gt; 'file.aln');
         my $aln = $alnin-&gt;next_aln;
         # Of course matrix can come from a different place
         # like PHYLIP if you prefer, Bio::Matrix::IO should be able
         # to parse many things
         my $jcmatrix = $stats-&gt;distance(-align =&gt; $aln,
                                         -method =&gt; 'Jukes-Cantor');
         my $tree = $tfactory-&gt;make_tree($jcmatrix);

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This is a factory which will construct a phylogenetic tree based on the pairwise sequence distances for a
       set of sequences.  Currently UPGMA (Sokal and Michener 1958) and NJ (Saitou and Nei 1987) tree
       construction methods are implemented.

</pre><h4><b>REFERENCES</b></h4><pre>
       Eddy SR, Durbin R, Krogh A, Mitchison G, (1998) "Biological Sequence Analysis", Cambridge Univ Press,
       Cambridge, UK.

       Howe K, Bateman A, Durbin R, (2002) "QuickTree: building huge Neighbour-Joining trees of protein
       sequences." Bioinformatics <a href="../man11/18.11.html">18</a>(11):1546-1547.

       Saitou N and Nei M, (1987) "The neighbor-joining method: a new method for reconstructing phylogenetic
       trees." Mol Biol Evol <a href="../man4/4.4.html">4</a>(4):406-25.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Jason</b> <b>Stajich</b></h4><pre>
       Email jason-at-bioperl.org

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : my $obj = Bio::Tree::DistanceFactory-&gt;new();
        Function: Builds a new Bio::Tree::DistanceFactory object
        Returns : an instance of Bio::Tree::DistanceFactory
        Args    : -method =&gt; 'NJ' or 'UPGMA'

   <b>make_tree</b>
        Title   : make_tree
        Usage   : my $tree = $disttreefact-&gt;make_tree($matrix);
        Function: Build a Tree based on a distance matrix
        Returns : L&lt;Bio::Tree::TreeI&gt;
        Args    : L&lt;Bio::Matrix::MatrixI&gt; object

   <b>_nj</b>
        Title   : _nj
        Usage   : my $tree = $disttreefact-&gt;_nj($matrix);
        Function: Construct a tree based on distance matrix using the
                  Neighbor Joining algorithm (Saitou and Nei, 1987)
                  Implementation based on Kevin Howe's Quicktree implementation
                  and uses his tricks (some based on Bill Bruno's work) to eliminate
                  negative branch lengths
        Returns : L&lt;Bio::Tree::TreeI&gt;
        Args    : L&lt;Bio::Matrix::MatrixI&gt; object

   <b>_upgma</b>
        Title   : _upgma
        Usage   : my $tree = $disttreefact-&gt;_upgma($matrix);
        Function: Construct a tree based on alignment using UPGMA
        Returns : L&lt;Bio::Tree::TreeI&gt;
        Args    : L&lt;Bio::Matrix::MatrixI&gt; object

   <b>method</b>
        Title   : method
        Usage   : $obj-&gt;method($newval)
        Function:
        Example :
        Returns : value of method (a scalar)
        Args    : on set, new value (a scalar or undef, optional)

   <b>check_additivity</b>
        Title     : check_additivity
        Usage     : if( $distance-&gt;check_additivity($matrix) ) {
                    }
        Function  : See if matrix obeys additivity principal
        Returns   : boolean
        Args      : Bio::Matrix::MatrixI
        References: Based on a Java implementation by
                    Peter Sestoft, <a href="mailto:sestoft@dina.kvl.dk">sestoft@dina.kvl.dk</a> 1999-12-07 version 0.3
                    <a href="http://www.dina.kvl.dk/~sestoft/bsa.html">http://www.dina.kvl.dk/~sestoft/bsa.html</a>
                    which in turn is based on algorithms described in
                    R. Durbin, S. Eddy, A. Krogh, G. Mitchison.
                    Biological Sequence Analysis CUP 1998, Chapter 7.

perl v5.32.1                                       2021-08-15                    <u>Bio::Tree::<a href="../man3pm/DistanceFactory.3pm.html">DistanceFactory</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>