<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Run::Phylo::PAML::Codeml - Wrapper aroud the PAML program codeml</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-tools-phylo-paml-perl">libbio-tools-phylo-paml-perl_1.7.3-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Run::Phylo::PAML::Codeml - Wrapper aroud the PAML program codeml

</pre><h4><b>VERSION</b></h4><pre>
       version 1.7.3

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::Tools::Run::Phylo::PAML::Codeml;
         use Bio::AlignIO;

         my $alignio = Bio::AlignIO-&gt;new(-format =&gt; 'phylip',
                              -file   =&gt; 't/data/gf-s85.phylip');

         my $aln = $alignio-&gt;next_aln;

         my $codeml = Bio::Tools::Run::Phylo::PAML::Codeml-&gt;new();
         $codeml-&gt;alignment($aln);
         my ($rc,$parser) = $codeml-&gt;run();
         my $result = $parser-&gt;next_result;
         my $MLmatrix = $result-&gt;get_MLmatrix();
         print "Ka = ", $MLmatrix-&gt;[0]-&gt;[1]-&gt;{'dN'},"\n";
         print "Ks = ", $MLmatrix-&gt;[0]-&gt;[1]-&gt;{'dS'},"\n";
         print "Ka/Ks = ", $MLmatrix-&gt;[0]-&gt;[1]-&gt;{'omega'},"\n";

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This is a wrapper around the codeml program of PAML (Phylogenetic Analysis by Maximum Likelihood) package
       of Ziheng Yang.  See <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">http://abacus.gene.ucl.ac.uk/software/paml.html</a> for more information.

       This module is more about generating the properl codeml.ctl file and will run the program in a separate
       temporary directory to avoid creating temp files all over the place.

   <b>Default</b> <b>Values</b>
       Valid and default values for codeml programs are listed below.  The default values are always the first
       one listed.  These descriptions are essentially lifted from the example codeml.ctl file and pamlDOC
       documentation provided by the author.

       <b>CodonFreq</b> specifies the equilibrium codon frequencies in codon substitution model. These frequencies can
       be assumed to be equal (1/61 each for the standard genetic code, <b>CodonFreq</b> = 0), calculated from the
       average nucleotide frequencies (<b>CodonFreq</b> = 1), from the average nucleotide frequencies at the three
       codon positions (<b>CodonFreq</b> = 2), or used as free parameters (<b>CodonFreq</b> = 3). The number of parameters
       involved in those models of codon frequencies is 0, 3, 9, and 60 (under the universal code), for
       <b>CodonFreq</b> = 0, 1, 2, and 3 respectively.

       <b>aaDist</b> specifies whether equal amino acid distances are assumed (= 0) or Grantham's matrix is used (= 1)
       (Yang et al. 1998).

       <b>runmode</b> = -2 performs ML estimation of dS and dN in pairwise comparisons. The program will collect
       estimates of dS and dN into the files 2ML.dS and 2ML.dN. Since many users seem interested in looking at
       dN /dS ratios among lineages, examination of the tree shapes indicated by branch lengths calculated from
       the two rates may be interesting although the analysis is ad hoc. If your species names have no more than
       10 characters, you can use the output distance matrices as input to Phylip programs such as neighbor
       without change. Otherwise you need to edit the files to cut the names short.

       <b>model</b> concerns assumptions about the dN/dS rate ratios among branches (Yang 1998; Yang and Nielsen 1998).
       <b>model</b> =0 means a single dN/dS ratio for all lineages (branches), 1 means one ratio for each branch (free
       ratio model), and 2 means arbitrary number of rations (such as the 2-ratios or 3-ratios models. with
       <b>model</b> =2, you may specify the omega ratios for the branches using branch labels (read about the tree
       structure file in the document).  This option seems rather easy to use. Otherwise, the program will ask
       the user to input a branch mark for the dN/dS ratio assumed for each branch. This should be an integral
       number between 0 to k - 1 if k different dN/dS ratios (omega_0 - omega_k - 1) are assumed for the
       branches of the tree. <b>Bioperl</b> note basically, doing this interactively is not going to work very well, so
       this module is really focused around using the 0 or 1 parameters.  Read the program documentation if
       you'd like some more detailed instructions.

       <b>NSsites</b> specifies models that allow the dN/dS ratio (omega) to vary among sites (Nielsen and Yang 1998,
       Yang et al. 2000) <b>Nssites</b> = m corresponds to model Mm in Yang et al (2000).  The variable <b>ncatG</b> is used
       to specify the number of categories in the omega distribution under some models.  The values of <b>ncatG()</b>
       used to perform our analyses are 3 for M3 (discrete), 5 for M4 (freq), 10 for the continuous
       distributions (M5: gamma, M6: 2gamma, M7: beta, M8:beta&amp;amp;w, M9:beta&amp;amp;gamma, M10: beta&amp;gamma+1,
       M11:beta&amp;amp;normal&amp;gt;1, and M12:0&amp;amp;2normal&amp;gt;1, M13:3normal&amp;gt;0). This means M8 will have 11 site
       classes (10 from the beta distribution plus 1 additional class). The posterior probabilities for site
       classes as well as the expected omega values for sites are listed in the file rst, which may be useful to
       pinpoint sites under positive selection, if they exist.

       To make it easy to run several <b>Nssites</b> models in one go, the executable
       Bio::Tools::Run::Phylo::PAML::Codemlsites can be used, which asks you how many and which models to run at
       the start of the program. The number of categories used will then match those used in Yang et <a href="../man2000/al.2000.html">al</a>(2000).

       As noted in that paper, some of the models are hard to use, in particular, M12 and M13. Recommended
       models are 0 (one-ratio), 1 (neutral), 2 (selection), 3 (discrete), 7 (beta), and 8 (beta&amp;amp;omega ).
       Some of the models like M2 and M8 are noted to be prone to the problem of multiple local optima. You are
       advised to run the program at least twice, once with a starting omega value &lt;1 and a second time with a
       value &gt;1, and use the results corresponding to the highest likelihood. The continuous neutral and
       selection models of Nielsen and Yang (1998) are not implemented in the program.

       <b>icode</b> for genetic code and these correspond to 1-11 in the genbank transl table.
         0:universal code
         1:mamalian mt
         2:yeast mt
         3:mold mt,
         4:invertebrate mt
         5:ciliate nuclear
         6:echinoderm mt
         7:euplotid mt
         8:alternative yeast nu.
         9:ascidian mt
         10:blepharisma nu

       <b>RateAncestor</b> For codon sequences, ancestral reconstruction is not implemented for the models of variable
       dN/dS ratios among sites. The output under codon-based models usually shows the encoded amino acid for
       each codon. The output under "Prob of best character at each node, listed by site" has two posterior
       probabilities for each node at each codon (amino acid) site. The first is for the best codon. The second,
       in parentheses, is for the most likely amino acid under the codon substitution model. This is a sum of
       posterior probabilities across synonymous codons. In theory it is possible although rare for the most
       likely amino acid not to match the most likely codon.

       <b>Output</b> for codon sequences (seqtype = 1): The codon frequencies in each sequence are counted and listed
       in a genetic code table, together with their sums across species. Each table contains six or fewer
       species. For data of multiple genes (option G in the sequence file), codon frequencies in each gene
       (summed over species) are also listed. The nucleotide distributions at the three codon positions are also
       listed. The method of Nei and Gojobori (1986) is used to calculate the number of synonymous substitutions
       per synonymous site (dS ) and the number of nonsynonymous substitutions per nonsynonymous site (dN ) and
       their ratio (dN /dS ). These are used to construct initial estimates of branch lengths for the likelihood
       analysis but are not MLEs themselves. Note that the estimates of these quantities for the a- and b-globin
       genes shown in Table 2 of Goldman and Yang (1994), calculated using the MEGA package (Kumar et al.,
       1993), are not accurate.

       Results of ancestral reconstructions (<b>RateAncestor</b> = 1) are collected in the file rst. Under models of
       variable dN/dS ratios among sites (NSsites models), the posterior probabilities for site classes as well
       as positively selected sites are listed in rst.

       INCOMPLETE DOCUMENTATION OF ALL METHODS

</pre><h4><b>ATTRIBUTES</b></h4><pre>
   <b>program_name</b>
        Title   : program_name
        Usage   : $factory-&gt;program_name()
        Function: holds the program name
        Returns:  string
        Args    : None

   <b>program_dir</b>
        Title   : program_dir
        Usage   : -&gt;program_dir()
        Function: returns the program directory, obtained from ENV variable.
        Returns:  string
        Args    :

   <b>error_string</b>
        Title   : error_string
        Usage   : $obj-&gt;error_string($newval)
        Function: Where the output from the last analysus run is stored.
        Returns : value of error_string
        Args    : newvalue (optional)

   <b>alignment</b>
        Title   : alignment
        Usage   : $codeml-&gt;align($aln);
        Function: Get/Set the L&lt;Bio::Align::AlignI&gt; object
        Returns : L&lt;Bio::Align::AlignI&gt; object
        Args    : [optional] L&lt;Bio::Align::AlignI&gt;
        Comment : We could potentially add support for running directly on a file
                  but we shall keep it simple
        See also: L&lt;Bio::SimpleAlign&gt;

   <b>tree</b>
        Title   : tree
        Usage   : $codeml-&gt;tree($tree, %params);
        Function: Get/Set the L&lt;Bio::Tree::TreeI&gt; object
        Returns : L&lt;Bio::Tree::TreeI&gt;
        Args    : [optional] $tree =&gt; L&lt;Bio::Tree::TreeI&gt;,
                  [optional] %parameters =&gt; hash of tree-specific parameters:
                         branchLengths: 0, 1 or 2
                         out

        Comment : We could potentially add support for running directly on a file
                  but we shall keep it simple
        See also: L&lt;Bio::Tree::Tree&gt;

   <b>no_param_checks</b>
        Title   : no_param_checks
        Usage   : $obj-&gt;no_param_checks($newval)
        Function: Boolean flag as to whether or not we should
                  trust the sanity checks for parameter values
        Returns : value of no_param_checks
        Args    : newvalue (optional)

   <b>save_tempfiles</b>
        Title   : save_tempfiles
        Usage   : $obj-&gt;save_tempfiles($newval)
        Function:
        Returns : value of save_tempfiles
        Args    : newvalue (optional)

   <b>outfile_name</b>
        Title   : outfile_name
        Usage   : my $outfile = $codeml-&gt;outfile_name();
        Function: Get/Set the name of the output file for this run
                  (if you wanted to do something special)
        Returns : string
        Args    : [optional] string to set value to

   <b>tempdir</b>
        Title   : tempdir
        Usage   : my $tmpdir = $self-&gt;tempdir();
        Function: Retrieve a temporary directory name (which is created)
        Returns : string which is the name of the temporary directory
        Args    : none

</pre><h4><b>METHODS</b></h4><pre>
   <b>new</b>
        Title   : new
        Usage   : my $obj = Bio::Tools::Run::Phylo::PAML::Codeml-&gt;new();
        Function: Builds a new Bio::Tools::Run::Phylo::PAML::Codeml object
        Returns : Bio::Tools::Run::Phylo::PAML::Codeml
        Args    : -alignment =&gt; the Bio::Align::AlignI object
                  -save_tempfiles =&gt; boolean to save the generated tempfiles and
                                     NOT cleanup after onesself (default FALSE)
                  -tree =&gt; the Bio::Tree::TreeI object
                  -branchlengths =&gt; 0: ignore any branch lengths found on the tree
                                    1: use as initial values
                                    2: fix branch lengths
                  -params =&gt; a hashref of PAML parameters (all passed to set_parameter)
                  -executable =&gt; where the codeml executable resides

       See also: Bio::Tree::TreeI, Bio::Align::AlignI

   <b>prepare</b>
        Title   : prepare
        Usage   : my $rundir = $codeml-&gt;prepare($aln);
        Function: prepare the codeml analysis using the default or updated parameters
                  the alignment parameter must have been set
        Returns : value of rundir
        Args    : L&lt;Bio::Align::AlignI&gt; object,
              L&lt;Bio::Tree::TreeI&gt; object [optional]

   <b>run</b>
        Title   : run
        Usage   : my ($rc,$parser) = $codeml-&gt;run($aln,$tree);
        Function: run the codeml analysis using the default or updated parameters
                  the alignment parameter must have been set
        Returns : Return code, L&lt;Bio::Tools::Phylo::PAML&gt;
        Args    : L&lt;Bio::Align::AlignI&gt; object,
              L&lt;Bio::Tree::TreeI&gt; object [optional]

   <b>get_parameters</b>
        Title   : get_parameters
        Usage   : my %params = $self-&gt;get_parameters();
        Function: returns the list of parameters as a hash
        Returns : associative array keyed on parameter names
        Args    : none

   <b>set_parameter</b>
        Title   : set_parameter
        Usage   : $codeml-&gt;set_parameter($param,$val);
        Function: Sets a codeml parameter, will be validated against
                  the valid values as set in the %VALIDVALUES class variable.
                  The checks can be ignored if one turns off param checks like this:
                    $codeml-&gt;<a href="../man1/no_param_checks.1.html">no_param_checks</a>(1)
        Returns : boolean if set was success, if verbose is set to -1
                  then no warning will be reported
        Args    : $param =&gt; name of the parameter
                  $value =&gt; value to set the parameter to
        See also: L&lt;no_param_checks()&gt;

   <b>set_default_parameters</b>
        Title   : set_default_parameters
        Usage   : $codeml-&gt;<a href="../man0/set_default_parameters.0.html">set_default_parameters</a>(0);
        Function: (Re)set the default parameters from the defaults
                  (the first value in each array in the
               %VALIDVALUES class variable)
        Returns : none
        Args    : boolean: keep existing parameter values

   <b>cleanup</b>
        Title   : cleanup
        Usage   : $codeml-&gt;cleanup();
        Function: Will cleanup the tempdir directory after a PAML run
        Returns : none
        Args    : none

   <b>io</b>
        Title   : io
        Usage   : $obj-&gt;io($newval)
        Function:  Gets a L&lt;Bio::Root::IO&gt; object
        Returns : L&lt;Bio::Root::IO&gt;
        Args    : none

</pre><h4><b>Bio::Tools::Run::WrapperBase</b> <b>methods</b></h4><pre>
</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>              - General discussion
         <a href="http://bioperl.org/Support.html">http://bioperl.org/Support.html</a>    - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list: <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bio-tools-phylo-paml/issues

</pre><h4><b>AUTHOR</b></h4><pre>
       Jason Stajich &lt;<a href="mailto:jason@bioperl.org">jason@bioperl.org</a>&gt;

</pre><h4><b>COPYRIGHT</b></h4><pre>
       This software is copyright (c) by Jason Stajich &lt;<a href="mailto:jason@bioperl.org">jason@bioperl.org</a>&gt;.

       This software is available under the same terms as the perl 5 programming language system itself.

perl v5.32.0                                       2020-11-10             <u>Bio::Tools::Run...o::PAML::<a href="../man3pm/Codeml.3pm.html">Codeml</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>