<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Run::Phylo::PAML::Yn00 - Wrapper aroud the PAML program yn00</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-tools-phylo-paml-perl">libbio-tools-phylo-paml-perl_1.7.3-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Run::Phylo::PAML::Yn00 - Wrapper aroud the PAML program yn00

</pre><h4><b>VERSION</b></h4><pre>
       version 1.7.3

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::Tools::Run::Phylo::PAML::Yn00;
         use Bio::AlignIO;
         my $alignio = Bio::AlignIO-&gt;new(-format =&gt; 'phylip',
                               -file   =&gt; 't/data/gf-s85.phylip');
         my $aln = $alignio-&gt;next_aln;

         my $yn = Bio::Tools::Run::Phylo::PAML::Yn00-&gt;new();
         $yn-&gt;alignment($aln);
         my ($rc,$parser) = $yn-&gt;run();
         while( my $result = $parser-&gt;next_result ) {
           my @otus = $result-&gt;get_seqs();
           my $MLmatrix = $result-&gt;get_MLmatrix();
           # 0 and 1 correspond to the 1st and 2nd entry in the @otus array
           my $dN = $MLmatrix-&gt;[0]-&gt;[1]-&gt;{dN};
           my $dS = $MLmatrix-&gt;[0]-&gt;[1]-&gt;{dS};
           my $kaks =$MLmatrix-&gt;[0]-&gt;[1]-&gt;{omega};
           print "Ka = $dN Ks = $dS Ka/Ks = $kaks\n";
         }

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This is a wrapper around the yn00 (method of Yang and Nielsen, 2000) program of PAML (Phylogenetic
       Analysis by Maximum Likelihood) package of Ziheng Yang.  See
       <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">http://abacus.gene.ucl.ac.uk/software/paml.html</a> for more information.

       This module will generate a proper yn00.ctl file and will run the program in a separate temporary
       directory to avoid creating temp files all over the place and will cleanup after itself.

   <b>Default</b> <b>Values</b>
       See the Bio::Tools::Run::Phylo::PAML::Codeml module for documentation of the default values.

</pre><h4><b>ATTRIBUTES</b></h4><pre>
   <b>program_name</b>
        Title   : program_name
        Usage   : $yn00-&gt;program_name()
        Function: holds the program name
        Returns:  string
        Args    : None

   <b>program_dir</b>
        Title   : program_dir
        Usage   : $yn00-&gt;program_dir()
        Function: returns the program directory, obtained from ENV variable.
        Returns:  string
        Args    :

   <b>error_string</b>
        Title   : error_string
        Usage   : $obj-&gt;error_string($newval)
        Function: Where the output from the last analysus run is stored.
        Returns : value of error_string
        Args    : newvalue (optional)

   <b>alignment</b>
        Title   : alignment
        Usage   : $codeml-&gt;align($aln);
        Function: Get/Set the L&lt;Bio::Align::AlignI&gt; object
        Returns : L&lt;Bio::Align::AlignI&gt; object
        Args    : [optional] L&lt;Bio::Align::AlignI&gt;
        Comment : We could potentially add support for running directly on a file
                  but we shall keep it simple
        See also: L&lt;Bio::SimpleAlign&gt;

   <b>no_param_checks</b>
        Title   : no_param_checks
        Usage   : $obj-&gt;no_param_checks($newval)
        Function: Boolean flag as to whether or not we should
                  trust the sanity checks for parameter values
        Returns : value of no_param_checks
        Args    : newvalue (optional)

   <b>save_tempfiles</b>
        Title   : save_tempfiles
        Usage   : $obj-&gt;save_tempfiles($newval)
        Function:
        Returns : value of save_tempfiles
        Args    : newvalue (optional)

   <b>outfile_name</b>
        Title   : outfile_name
        Usage   : my $outfile = $codeml-&gt;outfile_name();
        Function: Get/Set the name of the output file for this run
                  (if you wanted to do something special)
        Returns : string
        Args    : [optional] string to set value to

   <b>tempdir</b>
        Title   : tempdir
        Usage   : my $tmpdir = $self-&gt;tempdir();
        Function: Retrieve a temporary directory name (which is created)
        Returns : string which is the name of the temporary directory
        Args    : none

</pre><h4><b>METHODS</b></h4><pre>
   <b>new</b>
        Title   : new
        Usage   : my $obj = Bio::Tools::Run::Phylo::PAML::Yn00-&gt;new();
        Function: Builds a new Bio::Tools::Run::Phylo::PAML::Yn00 object
        Returns : Bio::Tools::Run::Phylo::PAML::Yn00
        Args    : -alignment =&gt; the L&lt;Bio::Align::AlignI&gt; object
                  -save_tempfiles =&gt; boolean to save the generated tempfiles and
                                     NOT cleanup after onesself (default FALSE)

   <b>run</b>
        Title   : run
        Usage   : $yn-&gt;run();
        Function: run the yn00 analysis using the default or updated parameters
                  the alignment parameter must have been set
        Returns : 3 values,
                  $rc = 1 for success, 0 for errors
                  hash reference of the Yang calculated Ka/Ks values
                           this is a set of pairwise observations keyed as
                           sequencenameA-&gt;sequencenameB-&gt;datatype
                  hash reference same as the previous one except it for the
                  Nei and Gojobori calculated Ka,Ks,omega values
        Args    : none

   <b>get_parameters</b>
        Title   : get_parameters
        Usage   : my %params = $self-&gt;get_parameters();
        Function: returns the list of parameters as a hash
        Returns : associative array keyed on parameter names
        Args    : none

   <b>set_parameter</b>
        Title   : set_parameter
        Usage   : $codeml-&gt;set_parameter($param,$val);
        Function: Sets a codeml parameter, will be validated against
                  the valid values as set in the %VALIDVALUES class variable.
                  The checks can be ignored if on turns of param checks like this:
                    $codeml-&gt;<a href="../man1/no_param_checks.1.html">no_param_checks</a>(1)
        Returns : boolean if set was success, if verbose is set to -1
                  then no warning will be reported
        Args    : $paramname =&gt; name of the parameter
                  $value     =&gt; value to set the parameter to
        See also: L&lt;no_param_checks()&gt;

   <b>set_default_parameters</b>
        Title   : set_default_parameters
        Usage   : $codeml-&gt;<a href="../man0/set_default_parameters.0.html">set_default_parameters</a>(0);
        Function: (Re)set the default parameters from the defaults
                  (the first value in each array in the
               %VALIDVALUES class variable)
        Returns : none
        Args    : boolean: keep existing parameter values

   <b>cleanup</b>
        Title   : cleanup
        Usage   : $codeml-&gt;cleanup();
        Function: Will cleanup the tempdir directory after a PAML run
        Returns : none
        Args    : none

   <b>io</b>
        Title   : io
        Usage   : $obj-&gt;io($newval)
        Function:  Gets a L&lt;Bio::Root::IO&gt; object
        Returns : L&lt;Bio::Root::IO&gt;
        Args    : none

</pre><h4><b>Bio::Tools::Run::Wrapper</b> <b>methods</b></h4><pre>
</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>              - General discussion
         <a href="http://bioperl.org/Support.html">http://bioperl.org/Support.html</a>    - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list: <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bio-tools-phylo-paml/issues

</pre><h4><b>AUTHOR</b></h4><pre>
       Jason Stajich &lt;<a href="mailto:jason@bioperl.org">jason@bioperl.org</a>&gt;

</pre><h4><b>COPYRIGHT</b></h4><pre>
       This software is copyright (c) by Jason Stajich &lt;<a href="mailto:jason@bioperl.org">jason@bioperl.org</a>&gt;.

       This software is available under the same terms as the perl 5 programming language system itself.

perl v5.32.0                                       2020-11-10             <u>Bio::Tools::Run...ylo::PAML::<a href="../man3pm/Yn00.3pm.html">Yn00</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>