<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Seq::EncodedSeq - subtype of Bio::LocatableSeq to store DNA that encodes a protein</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Seq::EncodedSeq - subtype of Bio::LocatableSeq to store DNA that encodes a protein

</pre><h4><b>SYNOPSIS</b></h4><pre>
         $obj = Bio::Seq::EncodedSeq-&gt;new( -seq      =&gt; $dna,
                                           -encoding =&gt; "CCCCCCCIIIIICCCCC",
                                           -start    =&gt; 1,
                                           -strand   =&gt; 1,
                                           -length   =&gt; 17 );

         # splice out (and possibly revcomp) the coding sequence
         $cds = obj-&gt;cds;

         # obtain the protein translation of the sequence
         $prot = $obj-&gt;translate;

         # other access/inspection routines as with Bio::LocatableSeq and
         # Bio::SeqI; note that coordinates are relative only to the DNA
         # sequence, not it's implicit encoded protein sequence.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::Seq::EncodedSeq is a Bio::LocatableSeq object that holds a DNA sequence as well as information about
       the coding potential of that DNA sequence.  It is meant to be useful in an alignment context, where the
       DNA may contain frameshifts, gaps and/or introns, or in describing the transcript of a gene.  An
       EncodedSeq provides the ability to access the "spliced" coding sequence, meaning that all introns and
       gaps are removed, and any frameshifts are adjusted to provide a "clean" CDS.

       In order to make simultaneous use of either the DNA or the implicit encoded protein sequence coordinates,
       please see Bio::Coordinate::EncodingPair.

</pre><h4><b>ENCODING</b></h4><pre>
       We use the term "encoding" here to refer to the series of symbols that we use to identify which residues
       of a DNA sequence are protein-coding (i.e. part of a codon), intronic, part of a 5' or 3', frameshift
       "mutations", etc.  From this information, a Bio::Seq::EncodedSeq is able to "figure out" its
       translational CDS.  There are two sets of coding characters, one termed "implicit" and one termed
       "explicit".

       The "implicit" encoding is a bit simpler than the "explicit" encoding: 'C' is used for any nucleotide
       that's part of a codon, 'U' for any UTR, etc.  The full list is shown below:

        Code  Meaning
        ----  -------
         C    coding
         I    intronic
         U    untranslated
         G    gapped (for use in alignments)
         F    a "forward", +1 frameshift
         B    a "backward", -1 frameshift

       The "explicit" encoding is just an expansion of the "implicit" encoding, to denote phase:

        Code  Meaning
        ----  -------
         C    coding, 1st codon position
         D    coding, 2nd codon position
         E    coding, 3rd codon position

         I    intronic, phase 0 (relative to intron begin)
         J    intronic, phase 1
         K    intronic, phase 2

         U    untranslated 3'UTR
         V    untranslated 5'UTR

         G    gapped (for use in alignments)
         F    a "forward", +1 frameshift
         B    a "backward", -1 frameshift

       Note that the explicit coding is meant to provide easy access to position/phase specific nucleotides:

         $obj = Bio::Seq::EncodedSeq-&gt;new(-seq =&gt; "ACAATCAGACTACG...",
                                          -encoding =&gt; "CCCCCCIII..."
                                         );

         # fetch arrays of nucleotides at each codon position:
         my @pos1 = $obj-&gt;dnaseq(encoding =&gt; 'C', explicit =&gt; 1);
         my @pos2 = $obj-&gt;dnaseq(encoding =&gt; 'D');
         my @pos3 = $obj-&gt;dnaseq(encoding =&gt; 'E');

         # fetch arrays of "3-1" codon dinucleotides, useful for genomic
         # signature analyses without compounding influences of codon bias:
         my @pairs = $obj-&gt;dnaseq(encoding =&gt; 'EC');

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Aaron</b> <b>Mackey</b></h4><pre>
       Email <a href="mailto:amackey@virginia.edu">amackey@virginia.edu</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : $obj = Bio::Seq::EncodedSeq-&gt;new(-seq      =&gt; "AGTACGTGTCATG",
                                                   -encoding =&gt; "CCCCCCFCCCCCC",
                                                   -id       =&gt; "myseq",
                                                   -start    =&gt; 1,
                                                   -end      =&gt; 13,
                                                   -strand   =&gt; 1
                                             );
        Function: creates a new Bio::Seq::EncodedSeq object from a supplied DNA
                  sequence
        Returns : a new Bio::Seq::EncodedSeq object

        Args    : seq      - primary nucleotide sequence used to encode the
                             protein; note that any positions involved in a
                             gap ('G') or backward frameshift ('B') should
                             have one or more gap characters; if the encoding
                             specifies G or B, but no (or not enough) gap
                             characters exist, *they'll be added*; similarly,
                             if there are gap characters without a
                             corresponding G or B encoding, G's will be
                             inserted into the encoding.  This allows some
                             flexibility in specifying your sequence and
                             coding without having to calculate a lot of the
                             encoding for yourself.

                  encoding - a string of characters (see Encoding Table)
                             describing backwards frameshifts implied by the
                             encoding but not present in the sequence will be
                             added (as '-'s) to the sequence.  If not
                             supplied, it will be assumed that all positions
                             are coding (C).  Encoding may include either
                             implicit phase encoding characters (i.e. "CCC")
                             and/or explicit encoding characters (i.e. "CDE").
                             Additionally, prefixed numbers may be used to
                             denote repetition (i.e. "27C3I28C").

                             Alternatively, encoding may be a hashref
                             datastructure, with encoding characters as keys
                             and Bio::LocationI objects (or arrayrefs of
                             Bio::LocationI objects) as values, e.g.:

                             { C =&gt; [ Bio::Location::Simple-&gt;new(1,9),
                                      Bio::Location::Simple-&gt;new(11,13) ],
                               F =&gt; Bio::Location::Simple-&gt;new(10,10),
                             } # same as "CCCCCCCCCFCCC"

                             Note that if the location ranges overlap, the
                             behavior of the encoding will be undefined
                             (well, it will be defined, but only according to
                             the order in which the hash keys are read, which
                             is basically undefined ... just don't do that).

                  id, start, end, strand - as with Bio::LocatableSeq; note
                             that the coordinates are relative to the
                             encoding DNA sequence, not the implicit protein
                             sequence.  If strand is reversed, then the
                             encoding is assumed to be relative to the
                             reverse strand as well.

   <b>encoding</b>
        Title   : encoding
        Usage   : $obj-&gt;encoding("CCCCCC");
                  $obj-&gt;encoding( -encoding =&gt; { I =&gt; $location } );
                  $enc = $obj-&gt;encoding(-explicit =&gt; 1);
                  $enc = $obj-&gt;encoding("CCCCCC", -explicit =&gt; 1);
                  $enc = $obj-&gt;encoding(-location =&gt; $location,
                                        -explicit =&gt; 1,
                                        -absolute =&gt; 1 );
        Function: get/set the objects encoding, either globally or by location(s).
        Returns : the (possibly new) encoding string.
        Args    : encoding - see the encoding argument to the new() function.

                  explicit - whether or not to return explicit phase
                             information in the coding (i.e. "CCC" becomes
                             "CDE", "III" becomes "IJK", etc); defaults to 0.

                  location - optional; location to get/set the encoding.
                             Defaults to the entire sequence.

                  absolute - whether or not the locational elements (either
                             in the encoding hashref or the location
                             argument) are relative to the absolute start/end
                             of the Bio::LocatableSeq, or to the internal,
                             relative coordinate system (beginning at 1);
                             defaults to 0 (i.e. relative)

   <b>cds</b>
        Title   : cds
        Usage   : $cds = $obj-&gt;cds(-nogaps =&gt; 1);
        Function: obtain the "spliced" DNA sequence, by removing any
                  nucleotides that participate in an UTR, forward frameshift
                  or intron, and replacing any unknown nucleotide implied by
                  a backward frameshift or gap with N's.
        Returns : a Bio::Seq::EncodedSeq object, with an encoding consisting only
                  of "CCCC..".
        Args    : nogaps - strip any gap characters (resulting from 'G' or 'B'
                  encodings), rather than replacing them with N's.

   <b>translate</b>
        Title   : translate
        Usage   : $prot = $obj-&gt;translate(@args);
        Function: obtain the protein sequence encoded by the underlying DNA
                  sequence; same as $obj-&gt;cds()-&gt;translate(@args).
        Returns : a Bio::PrimarySeq object.
        Args    : same as the translate() function of Bio::PrimarySeqI

   <b>protseq</b>
        Title   : seq
        Usage   : $protseq = $obj-&gt;protseq();
        Function: obtain the raw protein sequence encoded by the underlying
                  DNA sequence; This is the same as calling
                  $obj-&gt;translate()-&gt;seq();
        Returns : a string of single-letter amino acid codes
        Args :    same as the seq() function of Bio::PrimarySeq; note that this
                  function may not be used to set the protein sequence; see
                  the dnaseq() function for that.

   <b>dnaseq</b>
        Title   : dnaseq
        Usage   : $dnaseq = $obj-&gt;dnaseq();
                  $obj-&gt;dnaseq("ACGTGTCGT", "CCCCCCCCC");
                  $obj-&gt;dnaseq(-seq      =&gt; "ATG",
                               -encoding =&gt; "CCC",
                               -location =&gt; $loc );
                  @introns = $obj-&gt;$dnaseq(-encoding =&gt; 'I')
        Function: get/set the underlying DNA sequence; will overwrite any
                  current DNA and/or encoding information present.
        Returns : a string of single-letter nucleotide codes, including any
                  gaps implied by the encoding.
        Args    : seq      - the DNA sequence to be used as a replacement
                  encoding - the encoding of the DNA sequence (see the new()
                             constructor); defaults to all 'C' if setting a
                             new DNA sequence.  If no new DNA sequence is
                             being provided, then the encoding is used as a
                             "filter" for which to return fragments of
                             non-overlapping DNA that match the encoding.
                  location - optional, the location of the DNA sequence to
                             get/set; defaults to the entire sequence.

perl v5.32.1                                       2021-08-15                          <u>Bio::Seq::<a href="../man3pm/EncodedSeq.3pm.html">EncodedSeq</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>