<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::SeqFeature::FeaturePair - hold pair feature information e.g. blast hits</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::SeqFeature::FeaturePair - hold pair feature information e.g. blast hits

</pre><h4><b>SYNOPSIS</b></h4><pre>
           my $feat  = Bio::SeqFeature::FeaturePair-&gt;new(
               -feature1 =&gt; $f1,
               -feature2 =&gt; $f2,
           );

           # Bio::SeqFeatureI methods can be used

           my $start = $feat-&gt;start;
           my $end   = $feat-&gt;end;

           # Bio::FeaturePair methods can be used
           my $hstart = $feat-&gt;hstart;
           my $hend   = $feat-&gt;hend;

          my $feature1 = $feat-&gt;feature1;  # returns feature1 object

</pre><h4><b>DESCRIPTION</b></h4><pre>
       A sequence feature object where the feature is itself a feature on another sequence - e.g. a blast hit
       where residues 1-40 of a protein sequence SW:HBA_HUMAN has hit to bases 100 - 220 on a genomic sequence
       HS120G22.  The genomic sequence coordinates are used to create one sequence feature $f1 and the protein
       coordinates are used to create feature $f2.  A FeaturePair object can then be made

           my $fp = Bio::SeqFeature::FeaturePair-&gt;new(
               -feature1 =&gt; $f1,   # genomic
               -feature2 =&gt; $f2,   # protein
           );

       This object can be used as a standard Bio::SeqFeatureI in which case

           my $gstart = $fp-&gt;start  # returns start coord on feature1 - genomic seq.
           my $gend   = $fp-&gt;end    # returns end coord on feature1.

       In general standard Bio::SeqFeatureI method calls return information in feature1.

       Data in the feature 2 object are generally obtained using the standard methods prefixed by h (for hit!)

           my $pstart = $fp-&gt;hstart # returns start coord on feature2 = protein seq.
           my $pend   = $fp-&gt;hend   # returns end coord on feature2.

       If you wish to swap feature1 and feature2 around :

           $feat-&gt;invert

       so...

           $feat-&gt;start # etc. returns data in $feature2 object

       No sub_SeqFeatures or tags can be stored in this object directly.  Any features or tags are expected to
       be stored in the contained objects feature1, and feature2.

</pre><h4><b>CONTACT</b></h4><pre>
       Ewan Birney &lt;<a href="mailto:birney@sanger.ac.uk">birney@sanger.ac.uk</a>&gt;

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   :
        Function: Constructor for this module. Accepts the following parameters:

                    -feature1   Bio::SeqFeatureI-compliant object
                    -feature2   Bio::SeqFeatureI-compliant object
                    -feature_factory  Bio::Factory::ObjectFactoryI compliant
                                object to be used when feature1 and/or feature2
                                are accessed without explicitly set before. This
                                is mostly useful for derived classes who want to
                                set their preferred class for feature objects.

        Example :
        Returns :
        Args    : see above

   <b>feature1</b>
        Title   : feature1
        Usage   : $f = $featpair-&gt;feature1
                  $featpair-&gt;feature1($feature)
        Function: Get/set for the query feature
        Returns : Bio::SeqFeatureI
        Args    : Bio::SeqFeatureI

   <b>feature2</b>
        Title   : feature2
        Usage   : $f = $featpair-&gt;feature2
                  $featpair-&gt;feature2($feature)
        Function: Get/set for the hit feature
        Returns : Bio::SeqFeatureI
        Args    : Bio::SeqFeatureI

   <b>start</b>
        Title   : start
        Usage   : $start = $featpair-&gt;start
                  $featpair-&gt;<a href="../man20/start.20.html">start</a>(20)
        Function: Get/set on the start coordinate of feature1
        Returns : integer
        Args    : [optional] beginning of feature

   <b>end</b>
        Title   : end
        Usage   : $end = $featpair-&gt;end
                  $featpair-&gt;end($end)
        Function: get/set on the end coordinate of feature1
        Returns : integer
        Args    : [optional] ending point of feature

   <b>strand</b>
        Title   : strand
        Usage   : $strand = $feat-&gt;strand()
                  $feat-&gt;strand($strand)
        Function: get/set on strand information, being 1,-1 or 0
        Returns : -1,1 or 0
        Args    : [optional] strand information to set

   <b>location</b>
        Title   : location
        Usage   : $location = $featpair-&gt;location
                  $featpair-&gt;location($location)
        Function: Get/set location object (using feature1)
        Returns : Bio::LocationI object
        Args    : [optional] LocationI to store

   <b>score</b>
        Title   : score
        Usage   : $score = $feat-&gt;score()
                  $feat-&gt;score($score)
        Function: get/set on score information
        Returns : float
        Args    : none if get, the new value if set

   <b>frame</b>
        Title   : frame
        Usage   : $frame = $feat-&gt;frame()
                  $feat-&gt;frame($frame)
        Function: get/set on frame information
        Returns : 0,1,2
        Args    : none if get, the new value if set

   <b>primary_tag</b>
        Title   : primary_tag
        Usage   : $ptag = $featpair-&gt;primary_tag
        Function: get/set on the primary_tag of feature1
        Returns : 0,1,2
        Args    : none if get, the new value if set

   <b>source_tag</b>
        Title   : source_tag
        Usage   : $tag = $feat-&gt;source_tag()
                  $feat-&gt;source_tag('genscan');
        Function: Returns the source tag for a feature,
                  eg, 'genscan'
        Returns : a string
        Args    : none

   <b>seqname</b>
        Title   : seqname
        Usage   : $obj-&gt;seq_id($newval)
        Function: There are many cases when you make a feature that you
                  do know the sequence name, but do not know its actual
                  sequence. This is an attribute such that you can store
                  the seqname.

                  This attribute should *not* be used in GFF dumping, as
                  that should come from the collection in which the seq
                  feature was found.
        Returns : value of seqname
        Args    : newvalue (optional)

   <b>hseqname</b>
        Title   : hseqname
        Usage   : $featpair-&gt;hseqname($newval)
        Function: Get/set method for the name of
                  feature2.
        Returns : value of $feature2-&gt;seq_id
        Args    : newvalue (optional)

   <b>hstart</b>
        Title   : hstart
        Usage   : $start = $featpair-&gt;hstart
                  $featpair-&gt;<a href="../man20/hstart.20.html">hstart</a>(20)
        Function: Get/set on the start coordinate of feature2
        Returns : integer
        Args    : none

   <b>hend</b>
        Title   : hend
        Usage   : $end = $featpair-&gt;hend
                  $featpair-&gt;hend($end)
        Function: get/set on the end coordinate of feature2
        Returns : integer
        Args    : none

   <b>hstrand</b>
        Title   : hstrand
        Usage   : $strand = $feat-&gt;strand()
                  $feat-&gt;strand($strand)
        Function: get/set on strand information, being 1,-1 or 0
        Returns : -1,1 or 0
        Args    : none

   <b>hscore</b>
        Title   : hscore
        Usage   : $score = $feat-&gt;score()
                  $feat-&gt;score($score)
        Function: get/set on score information
        Returns : float
        Args    : none if get, the new value if set

   <b>hframe</b>
        Title   : hframe
        Usage   : $frame = $feat-&gt;frame()
                  $feat-&gt;frame($frame)
        Function: get/set on frame information
        Returns : 0,1,2
        Args    : none if get, the new value if set

   <b>hprimary_tag</b>
        Title   : hprimary_tag
        Usage   : $ptag = $featpair-&gt;hprimary_tag
        Function: Get/set on the primary_tag of feature2
        Returns : 0,1,2
        Args    : none if get, the new value if set

   <b>hsource_tag</b>
        Title   : hsource_tag
        Usage   : $tag = $feat-&gt;hsource_tag()
                  $feat-&gt;source_tag('genscan');
        Function: Returns the source tag for a feature,
                  eg, 'genscan'
        Returns : a string
        Args    : none

   <b>invert</b>
        Title   : invert
        Usage   : $tag = $feat-&gt;invert
        Function: Swaps feature1 and feature2 around
        Returns : Nothing
        Args    : none

   <b>feature_factory</b>
        Title   : feature_factory
        Usage   : $obj-&gt;feature_factory($newval)
        Function: Get/set the feature object factory for this feature pair.

                  The feature object factory will be used to create a feature
                  object if feature1() or feature2() is called in get mode
                  without having been set before.

                  The default is an instance of Bio::Factory::ObjectFactory
                  and hence allows the type to be changed dynamically at any
                  time.

        Example :
        Returns : The feature object factory in use (a
                  Bio::Factory::ObjectFactoryI compliant object)
        Args    : on set, a Bio::Factory::ObjectFactoryI compliant object

perl v5.32.1                                       2021-08-15                  <u>Bio::SeqFeature::<a href="../man3pm/FeaturePair.3pm.html">FeaturePair</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>