<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Graphics::Pictogram - generate SVG output of Pictogram display for consensus motifs</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-graphics-perl">libbio-graphics-perl_2.40-6_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Graphics::Pictogram - generate SVG output of Pictogram display for consensus motifs

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::Graphics::Pictogram;
         use Bio::SeqIO;

         my $sio = Bio::SeqIO-&gt;new(-file=&gt;$ARGV[0],-format=&gt;'fasta');
         my @seq;
         while(my $seq = $sio-&gt;next_seq){
           push @seq, $seq;
         }

         my $picto = Bio::Graphics::Pictogram-&gt;new(-width=&gt;"800",
                                                   -height=&gt;"500",
                                                   -fontsize=&gt;"60",
                                                   -plot_bits=&gt;1,
                                                   -background=&gt;{
                                                                 'A'=&gt;0.25,
                                                                 'C'=&gt;0.18,
                                                                 'T'=&gt;0.32,
                                                                 'G'=&gt;0.25},
                                                   -color=&gt;{'A'=&gt;'red',
                                                            'G'=&gt;'blue',
                                                            'C'=&gt;'green',
                                                            'T'=&gt;'magenta'});

         my $svg = $picto-&gt;make_svg(\@seq);

         print $svg-&gt;xmlify."\n";

         #Support for Bio::Matrix::PSM::SiteMatrix now included

          use Bio::Matrix::PSM::IO;

          my $picto = Bio::Graphics::Pictogram-&gt;new(-width=&gt;"800",
                                                   -height=&gt;"500",
                                                   -fontsize=&gt;"60",
                                                   -plot_bits=&gt;1,
                                                   -background=&gt;{
                                                                 'A'=&gt;0.25,
                                                                 'C'=&gt;0.18,
                                                                 'T'=&gt;0.32,
                                                                 'G'=&gt;0.25},
                                                   -color=&gt;{'A'=&gt;'red',
                                                            'G'=&gt;'blue',
                                                            'C'=&gt;'green',
                                                            'T'=&gt;'magenta'});

         my $psm = $psmIO-&gt;next_psm;
         my $svg = $picto-&gt;make_svg($psm);
         print $svg-&gt;xmlify;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       A module for generating SVG output of Pictogram display for consensus motifs.  This method of
       representation was describe by Burge and colleagues: (Burge, C.B.,Tuschl, T., Sharp, P.A. in The RNA
       world II, 525-560, CSHL press, 1999)

       This is a simple module that takes in an array of sequences (assuming equal lengths) and calculates
       relative base frequencies where the height of each letter reflects the frequency of each nucleotide at a
       given position. It can also plot the information content at each position scaled by the background
       frequencies of each nucleotide.

       It requires the SVG-2.26 or later module by Ronan Oger available at <a href="http://www.cpan.org">http://www.cpan.org</a>

       Recommended viewing of the SVG is the plugin available at Adobe: <a href="http://www.adobe.com/svg">http://www.adobe.com/svg</a>

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists. Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         <a href="http://bugzilla.open-bio.org/">http://bugzilla.open-bio.org/</a>

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Shawn</b> <b>Hoon</b></h4><pre>
       Email <a href="mailto:shawnh@fugu-sg.org">shawnh@fugu-sg.org</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a "_".

   <b>new</b>
        Title   : new
        Usage   : my $picto = Bio::Graphics::Pictogram-&gt;new(-width=&gt;"800",
                                                   -height=&gt;"500",
                                                   -fontsize=&gt;"60",
                                                   -plot_bits=&gt;1,
                                                   -background=&gt;{
                                                                 'A'=&gt;0.25,
                                                                 'C'=&gt;0.18,
                                                                 'T'=&gt;0.32,
                                                                 'G'=&gt;0.25},
                                                   -color=&gt;{'A'=&gt;'red',
                                                             'G'=&gt;'blue',
                                                             'C'=&gt;'green',
                                                             'T'=&gt;'magenta'});
        Function: Constructor for Pictogram Object
        Returns : L&lt;Bio::Graphics::Pictogram&gt;

   <b>make_svg</b>
        Title   : make_svg
        Usage   : $picto-&gt;make_svg();
        Function: make the SVG object
        Returns : L&lt;SVG&gt;
        Arguments: A fasta file or array ref of L&lt;Bio::Seq&gt; objects or a L&lt;Bio::Matrix::PSM::SiteMatrixI&gt;

   <b>fontsize</b>
        Title   : fontsize
        Usage   : $picto-&gt;fontsize();
        Function: get/set for fontsize
        Returns : int
        Arguments: int

   <b>color</b>
        Title   : color
        Usage   : $picto-&gt;color();
        Function: get/set for color
        Returns : a hash reference
        Arguments: a hash  reference

   <b>svg_obj</b>
        Title   : svg_obj
        Usage   : $picto-&gt;svg_obj();
        Function: get/set for svg_obj
        Returns : L&lt;SVG&gt;
        Arguments: L&lt;SVG&gt;

   <b>plot_bits</b>
        Title   : plot_bits
        Usage   : $picto-&gt;plot_bits();
        Function: get/set for plot_bits to indicate whether to plot
                  information content at each base position
        Returns :1/0
        Arguments: 1/0

   <b>normalize</b>
        Title   : normalize
        Usage   : $picto-&gt;normalize($newval)
        Function: get/set to make all columns the same height.
                  default is to scale height with information
                  content.
        Returns : value of normalize (a scalar)
        Args    : on set, new value (a scalar or undef, optional)

   <b>background</b>
        Title   : background
        Usage   : $picto-&gt;background();
        Function: get/set for hash reference of nucleodtide bgd frequencies
        Returns : hash reference
        Arguments: hash reference

   <b>pwm</b>
        Title   : pwm
        Usage   : $picto-&gt;pwm();
        Function: get/set for pwm
        Returns : int
        Arguments: int

perl v5.30.0                                       2019-11-25                      <u>Bio::Graphics::<a href="../man3pm/Pictogram.3pm.html">Pictogram</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>