<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::DB::GFF::RelSegment -- Sequence segment with relative coordinate support</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-db-gff-perl">libbio-db-gff-perl_1.7.4-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::DB::GFF::RelSegment -- Sequence segment with relative coordinate support

</pre><h4><b>SYNOPSIS</b></h4><pre>
       See Bio::DB::GFF.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::DB::GFF::RelSegment is a stretch of sequence that can handle relative coordinate addressing.  It
       inherits from Bio::DB::GFF::Segment, and is the base class for Bio::DB::GFF::Feature.

       In addition to the source sequence, a relative segment has a "reference sequence", which is used as the
       basis for its coordinate system.  The reference sequence can be changed at will, allowing you freedom to
       change the "frame of reference" for features contained within the segment.  For example, by setting a
       segment's reference sequence to the beginning of a gene, you can view all other features in gene-relative
       coordinates.

       The reference sequence and the source sequence must be on the same physical stretch of DNA, naturally.
       However, they do not have to be on the same strand.  The strandedness of the reference sequence
       determines whether coordinates increase to the right or the left.

       Generally, you will not create or manipulate Bio::DB::GFF::RelSeg0ment objects directly, but use those
       that are returned by the Bio::DB::GFF module.

   <b>An</b> <b>Example</b>
       To understand how relative coordinates work, consider the following example from the C. elegans database.
       First we create the appropriate GFF accessor object (the factory):

          my $db = Bio::DB::GFF-&gt;new(-dsn =&gt; 'dbi:mysql:elegans',
                                     -adaptor=&gt;'dbi:mysqlopt');

       Now we fetch out a segment based on cosmid clone ZK909:

         my $seg = $db-&gt;segment('ZK909');

       If we call the segment's <b>refseq()</b> method, we see that the base of the coordinate system is the sequence
       "ZK154", and that its start and stop positions are 1 and the length of the cosmid:

         print $seg-&gt;refseq;
         =&gt; ZK909

         print $seg-&gt;start,' - ',$seg-&gt;stop;
         =&gt; 1 - 33782

       As a convenience, the "" operator is overloaded in this class, to give the reference sequence, and start
       and stop positions:

         print $seg;
         =&gt; ZK909:1,33782

       Internally, Bio::DB::GFF::RelSegment has looked up the absolute coordinates of this segment and maintains
       the source sequence and the absolute coordinates relative to the source sequence.  We can see this
       information using <b>sourceseq()</b> (inherited from Bio::DB::GFF::Segment) and the <b>abs_start()</b> and <b>abs_end()</b>
       methods:

         print $seg-&gt;sourceseq;
         =&gt; CHROMOSOME_I

         print $seg-&gt;abs_start,' - ',$seg-&gt;abs_end;
         =&gt; 14839545 - 14873326

       We can also put the segment into absolute mode, so that it behaves like Bio::DB::Segment, and always
       represents coordinates on the source sequence.  This is done by passing a true value to the <b>absolute()</b>
       method:

         $seq-&gt;<a href="../man1/absolute.1.html">absolute</a>(1);
         print $seg;
         =&gt; CHROMOSOME_I:14839545,14873326

       We can change the reference sequence at any time.  One way is to call the segment's <b>ref()</b> method, giving
       it the ID (and optionally the class) of another landmark on the genome.  For example, if we know that
       cosmid ZK337 is adjacent to ZK909, then we can view ZK909 in ZK337-relative coordinates:

         $seg-&gt;refseq('ZK337');
         print $seg;
         =&gt; ZK337:-33670,111

       We can call the segment's <b>features()</b> method in order to get the list of contigs that overlap this segment
       (in the C. elegans database, contigs have feature type "Sequence:Link"):

         @links = $seg-&gt;features('Sequence:Link');

       We can now set the reference sequence to the first of these contigs like so:

         $seg-&gt;refseq($links[0]);
         print $seg;
         =&gt; Sequence:Link(LINK_Y95D11A):3997326,4031107

</pre><h4><b>API</b></h4><pre>
       The remainder of this document describes the API for Bio::DB::GFF::Segment.

   <b>new</b>
        Title   : new
        Usage   : $s = Bio::DB::GFF::RelSegment-&gt;new(@args)
        Function: create a new relative segment
        Returns : a new Bio::DB::GFF::RelSegment object
        Args    : see below
        Status  : Public

       This method creates a new Bio::DB::GFF::RelSegment object.  Generally this is called automatically by the
       Bio::DB::GFF module and derivatives.

       This function uses a named-argument style:

        -factory      a Bio::DB::GFF::Adaptor to use for database access
        -seq          ID of the source sequence
        -class        class of the source sequence
        -start        start of the desired segment relative to source sequence
        -stop         stop of the desired segment relative to source sequence
        -ref          ID of the reference sequence
        -refclass     class of the reference sequence
        -offset       0-based offset from source sequence to start of segment
        -length       length of desired segment
        -absolute, -force_absolute
                      use absolute coordinates, rather than coordinates relative
                      to the start of self or the reference sequence

       The -seq argument accepts the ID of any landmark in the database.  The stored source sequence becomes
       whatever the GFF file indicates is the proper sequence for this landmark.  A class of "Sequence" is
       assumed unless otherwise specified in the -class argument.

       If the argument to -seq is a Bio::GFF::Featname object (such as returned by the <b>group()</b> method), then the
       class is taken from that.

       The optional -start and -stop arguments specify the end points for the retrieved segment.  For those who
       do not like 1-based indexing, -offset and -length are provided.  If both -start/-stop and -offset/-length
       are provided, the latter overrides the former.  Generally it is not a good idea to mix metaphors.

       -ref and -refclass together indicate a sequence to be used for relative coordinates.  If not provided,
       the source sequence indicated by -seq is used as the reference sequence.  If the argument to -ref is a
       Bio::GFF::Featname object (such as returned by the <b>group()</b> method), then the class is taken from that.

       -force_absolute should be used if you wish to skip the lookup of the absolute position of the source
       sequence that ordinarily occurs when you create a relative segment.  In this case, the source sequence
       must be a sequence that has been specified as the "source" in the GFF file.

   <b>refseq</b>
        Title   : refseq
        Usage   : $ref = $s-&gt;refseq([$newseq] [,$newseqclass])
        Function: get/set reference sequence
        Returns : current reference sequence
        Args    : new reference sequence and class (optional)
        Status  : Public

       This method will get or set the reference sequence.  Called with no arguments, it returns the current
       reference sequence.  Called with either a sequence ID and class, a Bio::DB::GFF::Segment object (or
       subclass) or a Bio::DB::GFF::Featname object, it will set the current reference sequence and return the
       previous one.

       The method will generate an exception if you attempt to set the reference sequence to a sequence that
       isn't contained in the database, or one that has a different source sequence from the segment.

   <b>abs_low</b>
        Title   : abs_low
        Usage   : $s-&gt;abs_low
        Function: the absolute lowest coordinate of the segment
        Returns : an integer
        Args    : none
        Status  : Public

       This is for GadFly compatibility, and returns the low coordinate in absolute coordinates;

   <b>abs_high</b>
        Title   : abs_high
        Usage   : $s-&gt;abs_high
        Function: the absolute highest coordinate of the segment
        Returns : an integer
        Args    : none
        Status  : Public

       This is for GadFly compatibility, and returns the high coordinate in absolute coordinates;

   <b>asString</b>
        Title   : asString
        Usage   : $s-&gt;asString
        Function: human-readable representation of the segment
        Returns : a string
        Args    : none
        Status  : Public

       This method will return a human-readable representation of the segment.  It is the overloaded method call
       for the "" operator.

       Currently the format is:

         refseq:start,stop

   <b>name</b>
        Title   : name
        Usage   : Alias for asString()

   <b>absolute</b>
        Title   : absolute
        Usage   : $abs = $s-&gt;absolute([$abs])
        Function: get/set absolute coordinates
        Returns : a boolean flag
        Args    : new setting for flag (optional)
        Status  : Public

       Called with a boolean flag, this method controls whether to display relative coordinates (relative to the
       reference sequence) or absolute coordinates (relative to the source sequence).  It will return the
       previous value of the setting.

   <b>features</b>
        Title   : features
        Usage   : @features = $s-&gt;features(@args)
        Function: get features that overlap this segment
        Returns : a list of Bio::DB::GFF::Feature objects
        Args    : see below
        Status  : Public

       This method will find all features that overlap the segment and return a list of Bio::DB::GFF::Feature
       objects.  The features will use coordinates relative to the reference sequence in effect at the time that
       <b>features()</b> was called.

       The returned list can be limited to certain types of feature by filtering on their method and/or source.
       In addition, it is possible to obtain an iterator that will step through a large number of features
       sequentially.

       Arguments can be provided positionally or using the named arguments format.  In the former case, the
       arguments are a list of feature types in the format "method:source".  Either method or source can be
       omitted, in which case the missing component is treated as a wildcard.  If no colon is present, then the
       type is treated as a method name.  Multiple arguments are ORed together.

       Examples:

        @f = $s-&gt;features('exon:curated');           # all curated exons
        @f = $s-&gt;features('exon:curated','intron');  # curated exons and all introns
        @f = $s-&gt;features('similarity:.*EST.*');     # all similarities
                                                     # having something to do
                                                     # with ESTs

       The named parameter form gives you control over a few options:

         -types      an array reference to type names in the format
                     "method:source"

         -merge     Whether to apply aggregators to the generated features (default yes)

         -rare      Turn on an optimization suitable for a relatively rare feature type,
                    where it will be faster to filter by feature type first
                    and then by position, rather than vice versa.

         -attributes a hashref containing a set of attributes to match

         -range_type One of 'overlapping', 'contains', or 'contained_in'

         -iterator  Whether to return an iterator across the features.

         -binsize   A true value will create a set of artificial features whose
                    start and stop positions indicate bins of the given size, and
                    whose scores are the number of features in the bin.  The
                    class and method of the feature will be set to "bin",
                    its source to "method:source", and its group to "bin:method:source".
                    This is a handy way of generating histograms of feature density.

       -merge is a boolean flag that controls whether the adaptor's aggregators wll be applied to the features
       returned by this method.

       If -iterator is true, then the method returns a single scalar value consisting of a Bio::SeqIO object.
       You can call <b>next_seq()</b> repeatedly on this object to fetch each of the features in turn.  If iterator is
       false or absent, then all the features are returned as a list.

       The -attributes argument is a hashref containing one or more attributes to match against:

         -attributes =&gt; { Gene =&gt; 'abc-1',
                          Note =&gt; 'confirmed' }

       Attribute matching is simple string matching, and multiple attributes are ANDed together.

   <b>get_SeqFeatures</b>
        Title   : get_SeqFeatures
        Usage   :
        Function: returns the top level sequence features
        Returns : L&lt;Bio::SeqFeatureI&gt; objects
        Args    : none

       Segments do not ordinarily return any subfeatures.

   <b>feature_count</b>
        Title   : feature_count
        Usage   : $seq-&gt;feature_count()
        Function: Return the number of SeqFeatures attached to a sequence
        Returns : integer representing the number of SeqFeatures
        Args    : none

       This method comes through extension of Bio::FeatureHolderI. See Bio::FeatureHolderI for more information.

   <b>get_feature_stream</b>
        Title   : features
        Usage   : $stream = $s-&gt;get_feature_stream(@args)
        Function: get a stream of features that overlap this segment
        Returns : a Bio::SeqIO::Stream-compliant stream
        Args    : see below
        Status  : Public

       This is the same as <b>features()</b>, but returns a stream.  Use like this:

        $stream = $s-&gt;get_feature_stream('exon');
        while (my $exon = $stream-&gt;next_seq) {
           print $exon-&gt;start,"\n";
        }

   <b>get_seq_stream</b>
        Title   : get_seq_stream
        Usage   : $stream = $s-&gt;get_seq_stream(@args)
        Function: get a stream of features that overlap this segment
        Returns : a Bio::SeqIO::Stream-compliant stream
        Args    : see below
        Status  : Public

       This is the same as <b>feature_stream()</b>, and is provided for Bioperl compatibility.  Use like this:

        $stream = $s-&gt;get_seq_stream('exon');
        while (my $exon = $stream-&gt;next_seq) {
           print $exon-&gt;start,"\n";
        }

   <b>overlapping_features</b>
        Title   : overlapping_features
        Usage   : @features = $s-&gt;overlapping_features(@args)
        Function: get features that overlap this segment
        Returns : a list of Bio::DB::GFF::Feature objects
        Args    : see features()
        Status  : Public

       This is an alias for the <b>features()</b> method, and takes the same arguments.

   <b>contained_features</b>
        Title   : contained_features
        Usage   : @features = $s-&gt;contained_features(@args)
        Function: get features that are contained by this segment
        Returns : a list of Bio::DB::GFF::Feature objects
        Args    : see features()
        Status  : Public

       This is identical in behavior to <b>features()</b> except that it returns only those features that are
       completely contained within the segment, rather than any that overlap.

   <b>contained_in</b>
        Title   : contained_in
        Usage   : @features = $s-&gt;contained_in(@args)
        Function: get features that contain this segment
        Returns : a list of Bio::DB::GFF::Feature objects
        Args    : see features()
        Status  : Public

       This is identical in behavior to <b>features()</b> except that it returns only those features that completely
       contain the segment.

   <b>delete</b>
        Title   : delete
        Usage   : $db-&gt;delete(@args)
        Function: delete features
        Returns : count of features deleted -- if available
        Args    : numerous, see below
        Status  : public

       This method deletes all features that overlap the specified region or are of a particular type.  If no
       arguments are provided and the -force argument is true, then deletes ALL features.

       Arguments:

        -type,-types  Either a single scalar type to be deleted, or an
                      reference to an array of types.

        -range_type   Control the range type of the deletion.  One of "overlaps" (default)
                      "contains" or "contained_in"

       Examples:

         $segment-&gt;delete(-type=&gt;['intron','repeat:repeatMasker']);  # remove all introns &amp; repeats
         $segment-&gt;delete(-type=&gt;['intron','repeat:repeatMasker']
                          -range_type =&gt; 'contains');                # remove all introns &amp; repeats
                                                                     # strictly contained in segment

       IMPORTANT NOTE: This method only deletes features.  It does *NOT* delete the names of groups that contain
       the deleted features.  Group IDs will be reused if you later load a feature with the same group name as
       one that was previously deleted.

       NOTE ON FEATURE COUNTS: The DBI-based versions of this call return the result code from the SQL DELETE
       operation.  Some dbd drivers return the count of rows deleted, while others return 0E0.  Caveat emptor.

   <b>_process_feature_args</b>
        Title   : _process_feature_args
        Usage   : @args = $s-&gt;_process_feature_args(@args)
        Function: preprocess arguments passed to features,
                  contained_features, and overlapping_features
        Returns : a list of parsed arguents
        Args    : see feature()
        Status  : Internal

       This is an internal method that is used to check and format the arguments to <b>features()</b> before passing
       them on to the adaptor.

   <b>types</b>
        Title   : types
        Usage   : @types = $s-&gt;types([-enumerate=&gt;1])
        Function: list feature types that overlap this segment
        Returns : a list of Bio::DB::GFF::Typename objects or a hash
        Args    : see below
        Status  : Public

       The <b>types()</b> method will return a list of Bio::DB::GFF::Typename objects, each corresponding to a feature
       that overlaps the segment.  If the optional -enumerate parameter is set to a true value, then the method
       will return a hash in which the keys are the type names and the values are the number of times a feature
       of that type is present on the segment.  For example:

         %count = $s-&gt;types(-enumerate=&gt;1);

</pre><h4><b>Internal</b> <b>Methods</b></h4><pre>
       The following are internal methods and should not be called directly.

   <b>new_from_segment</b>
        Title   : new_from_segment
        Usage   : $s = $segment-&gt;new_from_segment(@args)
        Function: create a new relative segment
        Returns : a new Bio::DB::GFF::RelSegment object
        Args    : see below
        Status  : Internal

       This constructor is used internally by the <b>subseq()</b> method.  It forces the new segment into the
       Bio::DB::GFF::RelSegment package, regardless of the package that it is called from.  This causes
       subclass-specfic information, such as feature types, to be dropped when a subsequence is created.

   <b>_abs2rel</b>
        Title   : _abs2rel
        Usage   : @coords = $s-&gt;_abs2rel(@coords)
        Function: convert absolute coordinates into relative coordinates
        Returns : a list of relative coordinates
        Args    : a list of absolute coordinates
        Status  : Internal

       This is used internally to map from absolute to relative coordinates. It does not take the offset of the
       reference sequence into account, so please use <b>abs2rel()</b> instead.

   <b>rel2abs</b>
        Title   : rel2abs
        Usage   : @coords = $s-&gt;rel2abs(@coords)
        Function: convert relative coordinates into absolute coordinates
        Returns : a list of absolute coordinates
        Args    : a list of relative coordinates
        Status  : Public

       This function takes a list of positions in relative coordinates to the segment, and converts them into
       absolute coordinates.

   <b>abs2rel</b>
        Title   : abs2rel
        Usage   : @rel_coords = $s-&gt;abs2rel(@abs_coords)
        Function: convert absolute coordinates into relative coordinates
        Returns : a list of relative coordinates
        Args    : a list of absolute coordinates
        Status  : Public

       This function takes a list of positions in absolute coordinates and returns a list expressed in relative
       coordinates.

   <b>Bio::RangeI</b> <b>Methods</b>
       The following Bio::RangeI methods are supported:

       <b>overlaps()</b>, <b>contains()</b>, <b>equals()</b>,<b>intersection()</b>,<b>union()</b>,<b>overlap_extent()</b>

</pre><h4><b>BUGS</b></h4><pre>
       Schemas need some work.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       bioperl

</pre><h4><b>AUTHOR</b></h4><pre>
       Lincoln Stein &lt;<a href="mailto:lstein@cshl.org">lstein@cshl.org</a>&gt;.

       Copyright (c) 2001 Cold Spring Harbor Laboratory.

       This library is free software; you can redistribute it and/or modify it under the same terms as Perl
       itself.

perl v5.30.0                                       2020-01-13                      <u>Bio::DB::GFF::<a href="../man3pm/RelSegment.3pm.html">RelSegment</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>