<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Run::Phylo::Phast::PhastCons - Wrapper for footprinting using</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-perl-run-perl">libbio-perl-run-perl_1.7.3-8_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Run::Phylo::Phast::PhastCons - Wrapper for footprinting using
                                                  phastCons

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::Tools::Run::Phylo::Phast::PhastCons;

         # Make a PhastCons factory
         $factory = Bio::Tools::Run::Phylo::Phast::PhastCons-&gt;new();

         # Pass the factory an alignment and the corresponding species tree
         $align_filename = 't/data/apes.multi_fasta';
         $species_tree_filename = 't/data/apes.newick';
         @features = $factory-&gt;run($align_filename, $species_tree_filename);

         # or get a Bio::Align::AlignI (SimpleAlign) object from somewhere, and
         # generate the species tree automatically using a Bio::DB::Taxonomy database
         $tdb = Bio::DB::Taxonomy-&gt;new(-source =&gt; 'entrez');
         @features = $factory-&gt;run($aln_obj, $tdb);

         # @features is an array of Bio::SeqFeature::Annotated, one feature per
         # alignment sequence and prediction

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This is a wrapper for running the phastCons application by Adam Siepel. You can get details here:
       <a href="http://compgen.bscb.cornell.edu/~acs/software.html">http://compgen.bscb.cornell.edu/~acs/software.html</a> phastCons is used for phylogenetic footprinting/
       shadowing.

       Currently the interface is extremely simplified, allowing only one analysis method. The focus here is on
       ease of use, allowing phastCons to estimate as many parameters as possible and having it output just the
       'most conserved' blocks it detects. You can, however, try supplying normal phastCons arguments to <b>new()</b>,
       or calling arg-named methods (excluding initial hyphens and converting others to underscores, eg.
       $factory-&gt;<b><a href="../man1/indels_only.1.html">indels_only</a></b>(1) to set the --indels-only arg).

       The particular analysis carried out here is to:

        1. Use phyloFit to generate a tree model for initialization of the nonconserved
           model from the supplied alignment (all data) and species tree
        2. Run phastCons in 'training' mode for parameter estimation using all the
           alignment data and the model from step 1
        3. Run phastCons with the trees from step 2 to discover the most conserved
           regions

       See the 'HowTo' at <a href="http://compgen.bscb.cornell.edu/~acs/phastCons-HOWTO.html">http://compgen.bscb.cornell.edu/~acs/phastCons-HOWTO.html</a> for details on how to
       improve results.

       WARNING: the API is likely to change in the future to allow for alternative analysis types.

       You will need to enable this phastCons wrapper to find the phast programs (at least phastCons and
       phyloFit).  This can be done in (at least) three ways:

        1. Make sure the phastCons and phyloFit executables are in your path.
        2. Define an environmental variable PHASTDIR which is a
           directory which contains the phastCons and phyloFit applications:
           In bash:

           export PHASTDIR=/home/username/phast/bin

           In csh/tcsh:

           setenv PHASTDIR /home/username/phast/bin

        3. Include a definition of an environmental variable PHASTDIR in
           every script that will use this PhastCons wrapper module, e.g.:

           BEGIN { $ENV{PHASTDIR} = '/home/username/phast/bin' }
           use Bio::Tools::Run::Phylo::Phast::PhastCons;

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         <a href="http://redmine.open-bio.org/projects/bioperl/">http://redmine.open-bio.org/projects/bioperl/</a>

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Sendu</b> <b>Bala</b></h4><pre>
       Email <a href="mailto:bix@sendu.me.uk">bix@sendu.me.uk</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>program_name</b>
        Title   : program_name
        Usage   : $factory&gt;program_name()
        Function: holds the program name
        Returns : string
        Args    : None

   <b>program_dir</b>
        Title   : program_dir
        Usage   : $factory-&gt;program_dir(@params)
        Function: returns the program directory, obtained from ENV variable.
        Returns : string
        Args    : None

   <b>new</b>
        Title   : new
        Usage   : $factory = Bio::Tools::Run::Phylo::Phast::PhastCons-&gt;new(@params)
        Function: Creates a new PhastCons factory
        Returns : Bio::Tools::Run::Phylo::Phast::PhastCons
        Args    : Optionally, provide any of the following (defaults are not to use,
                  see the same-named methods for information on what each option does):
                  {
                   -target_coverage  =&gt; number between 0 and 1
                   AND
                   -expected_length  =&gt; int
                  }
                  -rho =&gt; number between 0 and 1
                  -quiet =&gt; boolean (turn on or off program output to console)

                  Most other options understood by phastCons can be supplied as key =&gt;
                  value pairs in this way. Options that don't normally take a value
                  should be given a value of 1. You can type the keys as you would on
                  the command line (eg. '--indels-only' =&gt; 1) or with only a single
                  hyphen to start and internal hyphens converted to underscores (eg.
                  -indels_only =&gt; 1) to avoid having to quote the key.

                  These options can NOT be used with this wrapper currently:
                  estimate_trees / T
                  estimate_rho / O
                  gc / G
                  msa_format / i
                  score / s
                  no_post_probs / n
                  seqname / N
                  idpref / P
                  help / h
                  alias / A
                  most_conserved / V / viterbi
                  refidx / r

   <b>target_coverage</b>
        Title   : target_coverage
        Usage   : $factory-&gt;target_coverage(0.25);
        Function: Constrain transition parameters such that the expected fraction of
                  sites in conserved elements is the supplied value.
        Returns : number (default undef)
        Args    : None to get, number (between 0 and 1) to set

   <b>expected_length</b>
        Title   : expected_length
        Usage   : $factory-&gt;<a href="../man5/expected_length.5.html">expected_length</a>(5);
        Function: Set transition probabilities such that the expected length of a
                  conserved element is the supplied value. target_coverage() must also
                  be set.
        Returns : int (default undef)
        Args    : None to get, int to set

   <b>rho</b>
        Title   : rho
        Usage   : $factory-&gt;rho(0.3);
        Function: Set the *scale* (overall evolutionary rate) of the model for the
                  conserved state to be the supplied number times that of the model for
                  the non-conserved state (default 0.3).
        Returns : number (default undef)
        Args    : None to get, number (between 0 and 1) to set

   <b>run</b>
        Title   : run
        Usage   : $result = $factory-&gt;run($fasta_align_file, $newick_tree_file);
                  -or-
                  $result = $factory-&gt;run($align_object, $tree_object);
                  -or-
                  $result = $factory-&gt;run($align_object, $db_taxonomy_object);
        Function: Runs phastCons on an alignment to find the most conserved regions
                  ('footprinting').
        Returns : array of Bio::SeqFeature::Annotated (one feature per alignment
                  sequence and prediction)
        Args    : The first argument represents an alignment, the second argument
                  a species tree.
                  The alignment can be provided as a multi-fasta format alignment
                  filename, or a Bio::Align::AlignI compliant object (eg. a
                  Bio::SimpleAlign).
                  The species tree can be provided as a newick format tree filename
                  or a Bio::Tree::TreeI compliant object. Alternatively a
                  Bio::DB::Taxonomy object can be supplied, in which case the species
                  tree will be generated by using the alignment sequence names as
                  species names and looking for those in the supplied database.

                  In all cases, the alignment sequence names must correspond to node
                  ids in the species tree. Multi-word species names should be joined
                  with underscores to form the sequence names, eg. Homo_sapiens

   <b>_setparams</b>
        Title   : _setparams
        Usage   : Internal function, not to be called directly
        Function: Creates a string of params to be used in the command string
        Returns : string of params
        Args    : alignment file name for result production, AND filename of phyloFit
                  generated init.mod file to estimate trees

perl v5.34.0                                       2022-02-10             <u>Bio::Tools::Run...hast::<a href="../man3pm/PhastCons.3pm.html">PhastCons</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>