<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::SeqWords - Object holding n-mer statistics for a sequence</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::SeqWords - Object holding n-mer statistics for a sequence

</pre><h4><b>SYNOPSIS</b></h4><pre>
         # Create the SeqWords object, e.g.:

         my $inputstream = Bio::SeqIO-&gt;new(-file =&gt; "seqfile",
                                                -format =&gt; 'Fasta');
         my $seqobj = $inputstream-&gt;next_seq();
         my $seq_word = Bio::Tools::SeqWords-&gt;new(-seq =&gt; $seqobj);

         # Or:
         my $seqobj = Bio::PrimarySeq-&gt;new(-seq =&gt; "agggtttccc",
                                           -alphabet =&gt; 'dna',
                                           -id =&gt; 'test');
         my $seq_word  =  Bio::Tools::SeqWords-&gt;new(-seq =&gt; $seqobj);

         # obtain a hash of word counts, eg:
         my $hash_ref = $seq_stats-&gt;count_words($word_length);

         # display hash table, eg:
         my %hash = %$hash_ref;
         foreach my $key(sort keys %hash)
         {
           print "\n$key\t$hash{$key}";
         }

         # Or:

         my $hash_ref =
            Bio::Tools::SeqWords-&gt;count_words($seqobj,$word_length);

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::Tools::SeqWords is a featherweight object for the calculation of n-mer word occurrences in a single
       sequence.  It is envisaged that the object will be useful for construction of scripts which use n-mer
       word tables as the raw material for statistical calculations; for instance, hexamer frequency for the
       calculation of coding protential, or the calculation of periodicity in repetitive DNA.  Triplet frequency
       is already handled by Bio::Tools::SeqStats (author: Peter Schattner).

       There are a few possible applications for protein, e.g. hypothesised amino acid 7-mers in heat shock
       proteins, or proteins with multiple simple motifs.  Sometimes these protein periodicities are best seen
       when the amino acid alphabet is truncated, e.g. Shulman alphabet.  Since there are quite a few of these
       shortened alphabets, this module does not specify any particular alphabet.

       See Synopsis above for object creation code.

   <b>Rationale</b>
       Take a sequence object and create an object for the purposes of holding n-mer word statistics about that
       sequence. The sequence can be nucleic acid or protein.

       In <b>count_words()</b> the words are counted in a non-overlapping manner, ie. in the style of a codon table,
       but with any word length.

       In <b>count_overlap_words()</b> the words are counted in an overlapping manner.

       For counts on opposite strand (DNA/RNA), a reverse complement method should be performed, and then the
       count repeated.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b></h4><pre>
       Derek Gatherer, in the loosest sense of the word 'author'.  The general shape of the module is lifted
       directly from the SeqStat module of Peter Schattner. The central subroutine to count the words is adapted
       from original code provided by Dave Shivak, in response to a query on the bioperl mailing list.  At least
       2 other people provided alternative means (equally good but not used in the end) of performing the same
       calculation.  Thanks to all for your assistance.

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
       Jason Stajich, jason-at-bioperl.org

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>count_words</b>
        Title   : count_words
        Usage   : $word_count = $seq_stats-&gt;count_words($word_length)
                       or
                  $word_count = $seq_stats-&gt;Bio::Tools::SeqWords-&gt;($seqobj,$word_length);
        Function: Counts non-overlapping words within a string, any alphabet is
                  used
        Example : a sequence ACCGTCCGT, counted at word length 4, will give the hash
                  {ACCG =&gt; 1, TCCG =&gt; 1}
        Returns : Reference to a hash in which keys are words (any length) of the
                  alphabet used and values are number of occurrences of the word
                  in the sequence.
        Args    : Word length as scalar and, reference to sequence object if
                  required

                  Throws an exception word length is not a positive integer
                  or if word length is longer than the sequence.

   <b>count_overlap_words</b>
        Title   : count_overlap_words
        Usage   : $word_count = $word_obj-&gt;count_overlap_words($word_length);
        Function: Counts overlapping words within a string, any alphabet is used
        Example : A sequence ACCAACCA, counted at word length 4, will give the hash
                       {ACCA=&gt;2, CCAA=&gt;1, CAAC=&gt;1, AACC=&gt;1}
        Returns : Reference to a hash in which keys are words (any length) of the
                  alphabet used and values are number of occurrences of the word in
                  the sequence.
        Args    : Word length as scalar

                  Throws an exception if word length is not a positive integer
                  or if word length is longer than the sequence.

perl v5.32.1                                       2021-08-15                          <u>Bio::Tools::<a href="../man3pm/SeqWords.3pm.html">SeqWords</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>