<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Run::Phylo::Njtree::Best - Wrapper around the Njtree (Njtree/phyml) best program.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-perl-run-perl">libbio-perl-run-perl_1.7.3-8_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Run::Phylo::Njtree::Best - Wrapper around the Njtree (Njtree/phyml) best program.

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::Tools::Run::Phylo::Njtree::Best;
         use Bio::AlignIO;
         use Bio::TreeIO;

         my $alignio = Bio::AlignIO-&gt;new(-format =&gt; 'fasta',
                                        -file   =&gt; 't/data/njtree_aln2.nucl.mfa');

         my $aln = $alignio-&gt;next_aln;

         my $treeio = Bio::TreeIO-&gt;new(
             -format =&gt; 'nhx', -file =&gt; 't/data/species_tree_njtree.nh');

         my $tree = $treeio-&gt;next_tree;

         my $njtree_best = Bio::Tools::Run::Phylo::Njtree::Best-&gt;new();
         $njtree_best-&gt;alignment($aln);
         $njtree_best-&gt;tree($tree);
         my $nhx_tree = $njtree_best-&gt;run();

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This is a wrapper around the best program of Njtree by Li Heng.  See
       <a href="http://treesoft.sourceforge.net/njtree.shtml">http://treesoft.sourceforge.net/njtree.shtml</a> for more information.

       Wrapper for the calculation of a reconciled phylogenetic tree with inferred duplication tags from
       amultiple sequence alignment and a species tree using NJTREE.

   <b>Helping</b> <b>the</b> <b>module</b> <b>find</b> <b>your</b> <b>executable</b>
       You will need to enable NJTREEDIR to find the njtree program. This can be done in (at least) three ways:

         1. Make sure the njtree executable is in your path (i.e.
            'which njtree' returns a valid program
         2. define an environmental variable NJTREEDIR which points to a
            directory containing the 'njtree' app:
          In bash
               export NJTREEDIR=/home/progs/treesoft/njtree   or
          In csh/tcsh
               setenv NJTREEDIR /home/progs/treesoft/njtree

         3. include a definition of an environmental variable NJTREEDIR
             in every script that will
            BEGIN {$ENV{NJTREEDIR} = '/home/progs/treesoft/njtree'; }
            use Bio::Tools::Run::Phylo::Njtree::Best;

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

        <a href="http://redmine.open-bio.org/projects/bioperl/">http://redmine.open-bio.org/projects/bioperl/</a>

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Albert</b> <b>Vilella</b></h4><pre>
       Email avilella-at-gmail-dot-com

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>program_name</b>
        Title   : program_name
        Usage   : $factory-&gt;program_name()
        Function: holds the program name
        Returns:  string
        Args    : None

   <b>program_dir</b>
        Title   : program_dir
        Usage   : $factory-&gt;program_dir(@params)
        Function: returns the program directory, obtained from ENV variable.
        Returns:  string
        Args    :

   <b>new</b>
        Title   : new
        Usage   : my $njtree_best = Bio::Tools::Run::Phylo::Njtree::Best-&gt;new();
        Function: Builds a new Bio::Tools::Run::Phylo::Njtree::Best
        Returns : Bio::Tools::Run::Phylo::Njtree::Best
        Args    : -alignment =&gt; the Bio::Align::AlignI object
                  -tree =&gt; the Bio::Tree::TreeI object
                  -save_tempfiles =&gt; boolean to save the generated tempfiles and
                                     NOT cleanup after onesself (default FALSE)
                  -executable =&gt; where the njtree executable resides

       See also: Bio::Tree::TreeI, Bio::Align::AlignI

   <b>prepare</b>
        Title   : prepare
        Usage   : my $rundir = $njtree_best-&gt;prepare();
        Function: prepare the njtree_best analysis using the default or updated parameters
                  the alignment parameter and species tree must have been set
        Returns : value of rundir
        Args    : L&lt;Bio::Align::AlignI&gt; object,
                  L&lt;Bio::Tree::TreeI&gt; object [optional]

   <b>run</b>
        Title   : run
        Usage   : my $nhx_tree = $njtree_best-&gt;run();
        Function: run the njtree_best analysis using the default or updated parameters
                  the alignment parameter must have been set
        Returns : L&lt;Bio::Tree::TreeI&gt; object [optional]
        Args    : L&lt;Bio::Align::AlignI&gt; object
                  L&lt;Bio::Tree::TreeI&gt; object

   <b>error_string</b>
        Title   : error_string
        Usage   : $obj-&gt;error_string($newval)
        Function: Where the output from the last analysus run is stored.
        Returns : value of error_string
        Args    : newvalue (optional)

   <b>version</b>
        Title   : version
        Usage   : exit if $prog-&gt;version() &lt; 1.8
        Function: Determine the version number of the program
        Example :
        Returns : float or undef
        Args    : none

   <b>alignment</b>
        Title   : alignment
        Usage   : $njtree_best-&gt;align($aln);
        Function: Get/Set the L&lt;Bio::Align::AlignI&gt; object
        Returns : L&lt;Bio::Align::AlignI&gt; object
        Args    : [optional] L&lt;Bio::Align::AlignI&gt;
        Comment : We could potentially add support for running directly on a file
                  but we shall keep it simple
        See also: L&lt;Bio::SimpleAlign&gt;

   <b>tree</b>
        Title   : tree
        Usage   : $njtree_best-&gt;tree($tree, %params);
        Function: Get/Set the L&lt;Bio::Tree::TreeI&gt; object
        Returns : L&lt;Bio::Tree::TreeI&gt;
        Args    : [optional] $tree =&gt; L&lt;Bio::Tree::TreeI&gt;,
                  [optional] %parameters =&gt; hash of tree-specific parameters

        Comment : We could potentially add support for running directly on a file
                  but we shall keep it simple
        See also: L&lt;Bio::Tree::Tree&gt;

   <b>check_names</b>
        Title   : check_names
        Usage   :
        Function:
        Example :
        Returns :
        Args    :

   <b>_setparams</b>
        Title   :  _setparams
        Usage   :  Internal function, not to be called directly
        Function:  Create parameter inputs for njtree_best program
        Example :
        Returns : parameter string to be passed to njtree_best
                  during align or profile_align
        Args    : name of calling object

</pre><h4><b>Bio::Tools::Run::BaseWrapper</b> <b>methods</b></h4><pre>
   <b>save_tempfiles</b>
        Title   : save_tempfiles
        Usage   : $obj-&gt;save_tempfiles($newval)
        Function:
        Returns : value of save_tempfiles
        Args    : newvalue (optional)

   <b>outfile_name</b>
        Title   : outfile_name
        Usage   : my $outfile = $njtree_best-&gt;outfile_name();
        Function: Get/Set the name of the output file for this run
                  (if you wanted to do something special)
        Returns : string
        Args    : [optional] string to set value to

   <b>tempdir</b>
        Title   : tempdir
        Usage   : my $tmpdir = $self-&gt;tempdir();
        Function: Retrieve a temporary directory name (which is created)
        Returns : string which is the name of the temporary directory
        Args    : none

   <b>cleanup</b>
        Title   : cleanup
        Usage   : $njtree_best-&gt;cleanup();
        Function: Will cleanup the tempdir directory
        Returns : none
        Args    : none

   <b>io</b>
        Title   : io
        Usage   : $obj-&gt;io($newval)
        Function:  Gets a L&lt;Bio::Root::IO&gt; object
        Returns : L&lt;Bio::Root::IO&gt;
        Args    : none

perl v5.34.0                                       2022-02-10             <u>Bio::Tools::Run...o::Njtree::<a href="../man3pm/Best.3pm.html">Best</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>