<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Est2Genome - Parse est2genome output, makes simple Bio::SeqFeature::Generic objects</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Est2Genome - Parse est2genome output, makes simple Bio::SeqFeature::Generic objects

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::Tools::Est2Genome;

         my $featureiter = Bio::Tools::Est2Genome-&gt;new(-file =&gt; 'output.est2genome');

         # This is going to be fixed to use the SeqAnalysisI next_feature
         # Method eventually when we have the objects to put the data in
         # properly
         while( my $f = $featureiter-&gt;parse_next_gene ) {
          # process Bio::SeqFeature::Generic objects here
         }

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This module is a parser for "est2genome" [EMBOSS] alignments of est/cdna sequence to genomic DNA.  This
       is generally accepted as the best program for predicting splice sites based on est/dnas (as far as I
       know).

       This module currently does not try pull out the ungapped alignments (Segment) but may in the future.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Jason</b> <b>Stajich</b></h4><pre>
       Email jason-at-bioperl.org

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : my $obj = Bio::Tools::Est2Genome-&gt;new();
        Function: Builds a new Bio::Tools::Est2Genome object
        Returns : an instance of Bio::Tools::Est2Genome
        Args    : -file =&gt; 'output.est2genome' or
                  -fh   =&gt; \*EST2GENOMEOUTPUT
                  -genomefirst =&gt; 1  # genome was the first input (not standard)

   <b>analysis_method</b>
        Usage     : $sim4-&gt;analysis_method();
        Purpose   : Inherited method. Overridden to ensure that the name matches
                    /est2genome/i.
        Returns   : String
        Argument  : n/a

   <b>parse_next_gene</b>
        Title   : parse_next_gene
        Usage   : @gene = $est2genome_result-&gt;parse_next_gene;
                  foreach $exon (@exons) {
                      # do something
                  }

        Function: Parses the next alignments of the est2genome result file and
                  returns the found exons as an array of
                  Bio::SeqFeature::SimilarityPair objects. Call
                  this method repeatedly until an empty array is returned to get the
                  results for all alignments.

                  The $exon-&gt;seq_id() attribute will be set to the identifier of the
                  respective sequence for both sequences.
                  The length is accessible via the seqlength()
                  attribute of $exon-&gt;query() and
                  $exon-&gt;est_hit().
        Returns : An array (or array reference) of Bio::SeqFeature::SimilarityPair and Bio::SeqFeature::Generic objects
                  or Bio::SeqFeature::Gene::GeneStructure
        Args    : flag(1/0) indicating to return Bio::SeqFeature::Gene::GeneStructure or Bio::SeqFeature::SimilarityPair
                  defaults to 0

   <b>next_feature</b>
        Title   : next_feature
        Usage   : $seqfeature = $obj-&gt;next_feature();
        Function: Returns the next feature available in the analysis result, or
                  undef if there are no more features.
        Example :
        Returns : A Bio::SeqFeatureI implementing object, or undef if there are no
                  more features.
        Args    : none

perl v5.32.1                                       2021-08-15                        <u>Bio::Tools::<a href="../man3pm/Est2Genome.3pm.html">Est2Genome</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>