<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Align::AlignI - An interface for describing sequence alignments.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Align::AlignI - An interface for describing sequence alignments.

</pre><h4><b>SYNOPSIS</b></h4><pre>
         # get a Bio::Align::AlignI somehow - typically using Bio::AlignIO system
         # some descriptors
         print $aln-&gt;length, "\n";
         print $aln-&gt;num_residues, "\n";
         print $aln-&gt;is_flush, "\n";
         print $aln-&gt;num_sequences, "\n";
         print $aln-&gt;percentage_identity, "\n";
         print $aln-&gt;<a href="../man50/consensus_string.50.html">consensus_string</a>(50), "\n";

         # find the position in the alignment for a sequence location
         $pos = $aln-&gt;column_from_residue_number('1433_LYCES', 14); # = 6;

         # extract sequences and check values for the alignment column $pos
         foreach $seq ($aln-&gt;each_seq) {
             $res = $seq-&gt;subseq($pos, $pos);
             $count{$res}++;
         }
         foreach $res (keys %count) {
             printf "Res: %s  Count: %2d\n", $res, $count{$res};
         }

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This interface describes the basis for alignment objects.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Jason</b> <b>Stajich</b></h4><pre>
       Email <a href="mailto:jason@bioperl.org">jason@bioperl.org</a>

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
       Ewan Birney, <a href="mailto:birney@ebi.ac.uk">birney@ebi.ac.uk</a> Heikki Lehvaslaiho, heikki-at-bioperl-dot-org

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

</pre><h4><b>Modifier</b> <b>methods</b></h4><pre>
       These methods modify the MSE by adding, removing or shuffling complete sequences.

   <b>add_seq</b>
        Title     : add_seq
        Usage     : $myalign-&gt;add_seq($newseq);
        Function  : Adds another sequence to the alignment. *Does not* align
                    it - just adds it to the hashes.
        Returns   : None
        Argument  : a Bio::LocatableSeq object
                    order (optional)

       See Bio::LocatableSeq for more information.

   <b>remove_seq</b>
        Title     : remove_seq
        Usage     : $aln-&gt;remove_seq($seq);
        Function  : Removes a single sequence from an alignment
        Returns   :
        Argument  : a Bio::LocatableSeq object

   <b>purge</b>
        Title   : purge
        Usage   : $aln-&gt;purge(0.7);
        Function:

                  Removes sequences above whatever %id.

                  This function will grind on large alignments. Beware!
                  (perhaps not ideally implemented)

        Example :
        Returns : An array of the removed sequences
        Argument:

   <b>sort_alphabetically</b>
        Title     : sort_alphabetically
        Usage     : $ali-&gt;sort_alphabetically
        Function  :

                    Changes the order of the alignment to alphabetical on name
                    followed by numerical by number.

        Returns   : an array
        Argument  :

</pre><h4><b>Sequence</b> <b>selection</b> <b>methods</b></h4><pre>
       Methods returning one or more sequences objects.

   <b>each_seq</b>
        Title     : each_seq
        Usage     : foreach $seq ( $align-&gt;each_seq() )
        Function  : Gets an array of Seq objects from the alignment
        Returns   : an array
        Argument  :

   <b>each_alphabetically</b>
        Title     : each_alphabetically
        Usage     : foreach $seq ( $ali-&gt;each_alphabetically() )
        Function  :

                    Returns an array of sequence object sorted alphabetically
                    by name and then by start point.
                    Does not change the order of the alignment

        Returns   :
        Argument  :

   <b>each_seq_with_id</b>
        Title     : each_seq_with_id
        Usage     : foreach $seq ( $align-&gt;each_seq_with_id() )
        Function  :

                    Gets an array of Seq objects from the
                    alignment, the contents being those sequences
                    with the given name (there may be more than one)

        Returns   : an array
        Argument  : a seq name

   <b>get_seq_by_pos</b>
        Title     : get_seq_by_pos
        Usage     : $seq = $aln-&gt;<a href="../man3/get_seq_by_pos.3.html">get_seq_by_pos</a>(3) # third sequence from the alignment
        Function  :

                    Gets a sequence based on its position in the alignment.
                    Numbering starts from 1.  Sequence positions larger than
                    num_sequences() will throw an error.

        Returns   : a Bio::LocatableSeq object
        Argument  : positive integer for the sequence position

</pre><h4><b>Create</b> <b>new</b> <b>alignments</b></h4><pre>
       The result of these methods are horizontal or vertical subsets of the current MSE.

   <b>select</b>
        Title     : select
        Usage     : $aln2 = $aln-&gt;select(1, 3) # three first sequences
        Function  :

                    Creates a new alignment from a continuous subset of
                    sequences.  Numbering starts from 1.  Sequence positions
                    larger than num_sequences() will throw an error.

        Returns   : a Bio::SimpleAlign object
        Argument  : positive integer for the first sequence
                    positive integer for the last sequence to include (optional)

   <b>select_noncont</b>
        Title     : select_noncont
        Usage     : $aln2 = $aln-&gt;select_noncont(1, 3) # first and 3rd sequences
        Function  :

                    Creates a new alignment from a subset of
                    sequences.  Numbering starts from 1.  Sequence positions
                    larger than num_sequences() will throw an error.

        Returns   : a Bio::SimpleAlign object
        Args      : array of integers for the sequences

   <b>slice</b>
        Title     : slice
        Usage     : $aln2 = $aln-&gt;slice(20, 30)
        Function  :

                    Creates a slice from the alignment inclusive of start and
                    end columns.  Sequences with no residues in the slice are
                    excluded from the new alignment and a warning is printed.
                    Slice beyond the length of the sequence does not do
                    padding.

        Returns   : a Bio::SimpleAlign object
        Argument  : positive integer for start column
                    positive integer for end column

</pre><h4><b>Change</b> <b>sequences</b> <b>within</b> <b>the</b> <b>MSE</b></h4><pre>
       These methods affect characters in all sequences without changing the alignment.

   <b>map_chars</b>
        Title     : map_chars
        Usage     : $ali-&gt;map_chars('\.','-')
        Function  :

                    Does a s/$arg1/$arg2/ on the sequences. Useful for gap
                    characters

                    Notice that the "from" (arg1) is interpreted as a regex,
                    so be careful about quoting meta characters (e.g.
                    $ali-&gt;map_chars('.', '-') won't do what you want)

        Returns   : None
        Argument  : 'from' rexexp
                    'to' string

   <b>uppercase</b>
        Title     : uppercase()
        Usage     : $ali-&gt;uppercase()
        Function  : Sets all the sequences to uppercase
        Returns   :
        Argument  :

   <b>match_line</b>
        Title    : match_line()
        Usage    : $align-&gt;match_line()
        Function : Generates a match line - much like consensus string
                   except that a line indicating the '*' for a match.
        Argument : (optional) Match line characters ('*' by default)
                   (optional) Strong match char (':' by default)
                   (optional) Weak match char ('.' by default)

   <b>match</b>
        Title     : match()
        Usage     : $ali-&gt;match()
        Function  :

                    Goes through all columns and changes residues that are
                    identical to residue in first sequence to match '.'
                    character. Sets match_char.

                    USE WITH CARE: Most MSE formats do not support match
                    characters in sequences, so this is mostly for output
                    only. NEXUS format (Bio::AlignIO::nexus) can handle
                    it.

        Returns   : 1
        Argument  : a match character, optional, defaults to '.'

   <b>unmatch</b>
        Title     : unmatch()
        Usage     : $ali-&gt;unmatch()
        Function  :

                    Undoes the effect of method match. Unsets match_char.

        Returns   : 1
        Argument  : a match character, optional, defaults to '.'

</pre><h4><b>MSE</b> <b>attibutes</b></h4><pre>
       Methods for setting and reading the MSE attributes.

       Note that the methods defining character semantics depend on the user to set them sensibly.  They are
       needed only by certain input/output methods. Unset them by setting to an empty string ('').

   <b>id</b>
        Title     : id
        Usage     : $myalign-&gt;id("Ig")
        Function  : Gets/sets the id field of the alignment
        Returns   : An id string
        Argument  : An id string (optional)

   <b>missing_char</b>
        Title     : missing_char
        Usage     : $myalign-&gt;missing_char("?")
        Function  : Gets/sets the missing_char attribute of the alignment
                    It is generally recommended to set it to 'n' or 'N'
                    for nucleotides and to 'X' for protein.
        Returns   : An missing_char string,
        Argument  : An missing_char string (optional)

   <b>match_char</b>
        Title     : match_char
        Usage     : $myalign-&gt;match_char('.')
        Function  : Gets/sets the match_char attribute of the alignment
        Returns   : An match_char string,
        Argument  : An match_char string (optional)

   <b>gap_char</b>
        Title     : gap_char
        Usage     : $myalign-&gt;gap_char('-')
        Function  : Gets/sets the gap_char attribute of the alignment
        Returns   : An gap_char string, defaults to '-'
        Argument  : An gap_char string (optional)

   <b>symbol_chars</b>
        Title   : symbol_chars
        Usage   : my @symbolchars = $aln-&gt;symbol_chars;
        Function: Returns all the seen symbols (other than gaps)
        Returns : array of characters that are the seen symbols
        Argument: boolean to include the gap/missing/match characters

</pre><h4><b>Alignment</b> <b>descriptors</b></h4><pre>
       These read only methods describe the MSE in various ways.

   <b>consensus_string</b>
        Title     : consensus_string
        Usage     : $str = $ali-&gt;consensus_string($threshold_percent)
        Function  : Makes a strict consensus
        Returns   : consensus string
        Argument  : Optional threshold ranging from 0 to 100.
                    The consensus residue has to appear at least threshold %
                    of the sequences at a given location, otherwise a '?'
                    character will be placed at that location.
                    (Default value = 0%)

   <b>consensus_iupac</b>
        Title     : consensus_iupac
        Usage     : $str = $ali-&gt;consensus_iupac()
        Function  :

                    Makes a consensus using IUPAC ambiguity codes from DNA
                    and RNA. The output is in upper case except when gaps in
                    a column force output to be in lower case.

                    Note that if your alignment sequences contain a lot of
                    IUPAC ambiquity codes you often have to manually set
                    alphabet.  Bio::PrimarySeq::_guess_type thinks they
                    indicate a protein sequence.

        Returns   : consensus string
        Argument  : none
        Throws    : on protein sequences

   <b>is_flush</b>
        Title     : is_flush
        Usage     : if( $ali-&gt;is_flush() )
                  :
                  :
        Function  : Tells you whether the alignment
                  : is flush, ie all of the same length
                  :
                  :
        Returns   : 1 or 0
        Argument  :

   <b>length</b>
        Title     : length()
        Usage     : $len = $ali-&gt;length()
        Function  : Returns the maximum length of the alignment.
                    To be sure the alignment is a block, use is_flush
        Returns   : integer
        Argument  :

   <b>maxname_length</b>
        Title     : maxname_length
        Usage     : $ali-&gt;maxname_length()
        Function  :

                    Gets the maximum length of the displayname in the
                    alignment. Used in writing out various MSE formats.

        Returns   : integer
        Argument  :

   <b>num_residues</b>
        Title     : num_residues
        Usage     : $no = $ali-&gt;num_residues
        Function  : number of residues in total in the alignment
        Returns   : integer
        Argument  :
        Note      : replaces no_residues

   <b>num_sequences</b>
        Title     : num_sequences
        Usage     : $depth = $ali-&gt;num_sequences
        Function  : number of sequence in the sequence alignment
        Returns   : integer
        Argument  : None
        Note      : replaces no_sequences

   <b>percentage_identity</b>
        Title   : percentage_identity
        Usage   : $id = $align-&gt;percentage_identity
        Function: The function calculates the percentage identity of the alignment
        Returns : The percentage identity of the alignment (as defined by the
                  implementation)
        Argument: None

   <b>overall_percentage_identity</b>
        Title   : overall_percentage_identity
        Usage   : $id = $align-&gt;overall_percentage_identity
        Function: The function calculates the percentage identity of
                  the conserved columns
        Returns : The percentage identity of the conserved columns
        Args    : None

   <b>average_percentage_identity</b>
        Title   : average_percentage_identity
        Usage   : $id = $align-&gt;average_percentage_identity
        Function: The function uses a fast method to calculate the average
                  percentage identity of the alignment
        Returns : The average percentage identity of the alignment
        Args    : None

</pre><h4><b>Alignment</b> <b>positions</b></h4><pre>
       Methods to map a sequence position into an alignment column and back.  <b>column_from_residue_number()</b> does
       the former. The latter is really a property of the sequence object and can done using
       Bio::LocatableSeq::location_from_column:

           # select somehow a sequence from the alignment, e.g.
           my $seq = $aln-&gt;<a href="../man1/get_seq_by_pos.1.html">get_seq_by_pos</a>(1);
           #$loc is undef or Bio::LocationI object
           my $loc = $seq-&gt;<a href="../man5/location_from_column.5.html">location_from_column</a>(5);

   <b>column_from_residue_number</b>
        Title   : column_from_residue_number
        Usage   : $col = $ali-&gt;column_from_residue_number( $seqname, $resnumber)
        Function:

                  This function gives the position in the alignment
                  (i.e. column number) of the given residue number in the
                  sequence with the given name. For example, for the
                  alignment

                    Seq1/91-97 AC..DEF.GH
                    Seq2/24-30 ACGG.RTY..
                    Seq3/43-51 AC.DDEFGHI

                  column_from_residue_number( "Seq1", 94 ) returns 6.
                  column_from_residue_number( "Seq2", 25 ) returns 2.
                  column_from_residue_number( "Seq3", 50 ) returns 9.

                  An exception is thrown if the residue number would lie
                  outside the length of the alignment
                  (e.g. column_from_residue_number( "Seq2", 22 )

                 Note: If the parent sequence is represented by more than one
                 alignment sequence and the residue number is present in
                 them, this method finds only the first one.

        Returns : A column number for the position in the alignment of the
                  given residue in the given sequence (1 = first column)
        Args    : A sequence id/name (not a name/start-end)
                  A residue number in the whole sequence (not just that
                  segment of it in the alignment)

</pre><h4><b>Sequence</b> <b>names</b></h4><pre>
       Methods to manipulate the display name. The default name based on the sequence id and subsequence
       positions can be overridden in various ways.

   <b>displayname</b>
        Title     : displayname
        Usage     : $myalign-&gt;displayname("Ig", "IgA")
        Function  : Gets/sets the display name of a sequence in the alignment
                  :
        Returns   : A display name string
        Argument  : name of the sequence
                    displayname of the sequence (optional)

   <b>set_displayname_count</b>
        Title     : set_displayname_count
        Usage     : $ali-&gt;set_displayname_count
        Function  :

                    Sets the names to be name_# where # is the number of
                    times this name has been used.

        Returns   : None
        Argument  : None

   <b>set_displayname_flat</b>
        Title     : set_displayname_flat
        Usage     : $ali-&gt;set_displayname_flat()
        Function  : Makes all the sequences be displayed as just their name,
                    not name/start-end
        Returns   : 1
        Argument  : None

   <b>set_displayname_normal</b>
        Title     : set_displayname_normal
        Usage     : $ali-&gt;set_displayname_normal()
        Function  : Makes all the sequences be displayed as name/start-end
        Returns   : None
        Argument  : None

perl v5.32.1                                       2021-08-15                            <u>Bio::Align::<a href="../man3pm/AlignI.3pm.html">AlignI</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>