<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Phylo::PAML - Parses output from the PAML programs codeml, baseml, basemlg, codemlsites and</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-tools-phylo-paml-perl">libbio-tools-phylo-paml-perl_1.7.3-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Phylo::PAML - Parses output from the PAML programs codeml, baseml, basemlg, codemlsites and
       yn00

</pre><h4><b>VERSION</b></h4><pre>
       version 1.7.3

</pre><h4><b>SYNOPSIS</b></h4><pre>
         #!<a href="file:///usr/lib/w3m/cgi-bin/w3mman2html.cgi?perl">/usr/bin/perl</a> -Tw
         use strict;

         use Bio::Tools::Phylo::PAML;

         # need to specify the output file name (or a fh) (defaults to
         # -file =&gt; "codeml.mlc"); also, optionally, the directory in which
         # the other result files (rst, 2ML.dS, etc) may be found (defaults
         # to "./")
         my $parser = Bio::Tools::Phylo::PAML-&gt;new
           (-file =&gt; "./results/mlc", -dir =&gt; "./results/");

         # get the first/next result; a Bio::Tools::Phylo::PAML::Result object,
         # which isa Bio::SeqAnalysisResultI object.
         my $result = $parser-&gt;next_result();

         # get the sequences used in the analysis; returns Bio::PrimarySeq
         # objects (OTU = Operational Taxonomic Unit).
         my @otus = $result-&gt;get_seqs();

         # codon summary: codon usage of each sequence [ arrayref of {
         # hashref of counts for each codon } for each sequence and the
         # overall sum ], and positional nucleotide distribution [ arrayref
         # of { hashref of frequencies for each nucleotide } for each
         # sequence and overall frequencies ]:
         my ($codonusage, $ntdist) = $result-&gt;get_codon_summary();

         # example manipulations of $codonusage and $ntdist:
         printf "There were %d %s codons in the first seq (%s)\n",
           $codonusage-&gt;[0]-&gt;{AAA}, 'AAA', $otus[0]-&gt;id();
         printf "There were %d %s codons used in all the sequences\n",
           $codonusage-&gt;[$#{$codonusage}]-&gt;{AAA}, 'AAA';
         printf "Nucleotide %c was present %g of the time in seq %s\n",
           'A', $ntdist-&gt;[1]-&gt;{A}, $otus[1]-&gt;id();

         # get Nei &amp; Gojobori dN/dS matrix:
         my $NGmatrix = $result-&gt;get_NGmatrix();

         # get ML-estimated dN/dS matrix, if calculated; this corresponds to
         # the runmode = -2, pairwise comparison usage of codeml
         my $MLmatrix = $result-&gt;get_MLmatrix();

         # These matrices are length(@otu) x length(@otu) "strict lower
         # triangle" 2D-matrices, which means that the diagonal and
         # everything above it is undefined.  Each of the defined cells is a
         # hashref of estimates for "dN", "dS", "omega" (dN/dS ratio), "t",
         # "S" and "N".  If a ML matrix, "lnL" and "kappa" will also be defined.
         printf "The omega ratio for sequences %s vs %s was: %g\n",
           $otus[0]-&gt;id, $otus[1]-&gt;id, $MLmatrix-&gt;[0]-&gt;[1]-&gt;{omega};

         # with a little work, these matrices could also be passed to
         # Bio::Tools::Run::Phylip::Neighbor, or other similar tree-building
         # method that accepts a matrix of "distances" (using the LOWTRI
         # option):
         my $distmat = [ map { [ map { $$_{omega} } @$_ ] } @$MLmatrix ];

         # for runmode's other than -2, get tree topology with estimated
         # branch lengths; returns a Bio::Tree::TreeI-based tree object with
         # added PAML parameters at each node
         my ($tree) = $result-&gt;get_trees();
         for my $node ($tree-&gt;get_nodes()) {
            # inspect the tree: the "t" (time) parameter is available via
            # $node-&gt;branch_length(); all other branch-specific parameters
            # ("omega", "dN", etc.) are available via
            # ($omega) = $node-&gt;get_tag_values('omega');
         }

         # if you are using model based Codeml then trees are stored in each
         # modelresult object
         for my $modelresult ( $result-&gt;get_NSSite_results ) {
           # model M0, M1, etc
           print "model is ", $modelresult-&gt;model_num, "\n";
           my ($tree) = $modelresult-&gt;get_trees();
           for my $node ($tree-&gt;get_nodes()) {
            # inspect the tree: the "t" (time) parameter is available via
            # $node-&gt;branch_length(); all other branch-specific parameters
            # ("omega", "dN", etc.) are available via
            # ($omega) = $node-&gt;get_tag_values('omega');
          }
         }

         # get any general model parameters: kappa (the
         # transition/transversion ratio), NSsites model parameters ("p0",
         # "p1", "w0", "w1", etc.), etc.
         my $params = $result-&gt;get_model_params();
         printf "M1 params: p0 = %g\tp1 = %g\n", $params-&gt;{p0}, $params-&gt;{p1};

         # parse AAML result files
         my $aamat = $result-&gt;get_AADistMatrix();
         my $aaMLmat = $result-&gt;get_AAMLDistMatrix();

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This module is used to parse the output from the PAML programs codeml, baseml, basemlg, codemlsites and
       yn00.  You can use the Bio::Tools::Run::Phylo::PAML::* modules to actually run some of the PAML programs,
       but this module is only useful to parse the output.

       This module has fledgling support for PAML version 4.3a (October 2009).  Please report any problems to
       the mailing list (see FEEDBACK below).

   <b>parse_codeml</b> <b>example</b>
         use Bio::Tools::Phylo::PAML;

         my $parser = new Bio::Tools::Phylo::PAML(-file    =&gt; shift,
                                                  -verbose =&gt; shift);

         my $result = $parser-&gt;next_result;
         my @otus = $result-&gt;get_seqs();
         my $MLmatrix = $result-&gt;get_MLmatrix();
         my $NGmatrix = $result-&gt;get_NGmatrix();

         # These matrices are length(@otu) x length(@otu) "strict lower
         # triangle" 2D-matrices, which means that the diagonal and
         # everything above it is undefined.  Each of the defined cells is a
         # hashref of estimates for "dN", "dS", "omega" (dN/dS ratio), "t",
         # "S" and "N".  If a ML matrix, "lnL" will also be defined.

         @otus = $result-&gt;get_seqs();
         $MLmatrix = $result-&gt;get_MLmatrix();
         $NGmatrix = $result-&gt;get_NGmatrix();
         for( my $i=0;$i&lt;scalar @$MLmatrix;$i++) {
             for( my $j = $i+1; $j &lt; scalar @{$MLmatrix-&gt;[$i]}; $j++ ) {
                 printf "The ML omega ratio for sequences %s vs %s was: %g\n",
                   $otus[$i]-&gt;id, $otus[$j]-&gt;id, $MLmatrix-&gt;[$i]-&gt;[$j]-&gt;{omega};
             }
         }

         for( my $i=0;$i&lt;scalar @$MLmatrix;$i++) {
             for( my $j = $i+1; $j &lt; scalar @{$MLmatrix-&gt;[$i]}; $j++ ) {
                 printf "The NG omega ratio for sequences %s vs %s was: %g\n",
                   $otus[$i]-&gt;id, $otus[$j]-&gt;id, $NGmatrix-&gt;[$i]-&gt;[$j]-&gt;{'omega'};
             }
         }

</pre><h4><b>METHODS</b></h4><pre>
   <b>new</b>
        Title   : new
        Usage   : my $obj = Bio::Tools::Phylo::PAML-&gt;new(%args);
        Function: Builds a new Bio::Tools::Phylo::PAML object
        Returns : Bio::Tools::Phylo::PAML
        Args    : Hash of options: -file, -fh, -dir
                  -file (or -fh) should contain the contents of the PAML
                        outfile;
                  -dir is the (optional) name of the directory in
                       which the PAML program was run (and includes other
                       PAML-generated files from which we can try to gather data)

   <b>Implement</b> <b>Bio::AnalysisParserI</b> <b>interface</b>
   <b>next_result</b>
        Title   : next_result
        Usage   : $result = $obj-&gt;next_result();
        Function: Returns the next result available from the input, or
                  undef if there are no more results.
        Example :
        Returns : a Bio::Tools::Phylo::PAML::Result object
        Args    : none

</pre><h4><b>TO</b> <b>DO</b></h4><pre>
       Implement <b>get_posteriors()</b>. For NSsites models, obtain arrayrefs of posterior probabilities for
       membership in each class for every position; probabilities correspond to classes w0, w1, ... etc.

         my @probs = $result-&gt;get_posteriors();

         # find, say, positively selected sites!
         if ($params-&gt;{w2} &gt; 1) {
           for (my $i = 0; $i &lt; @probs ; $i++) {
             if ($probs[$i]-&gt;[2] &gt; 0.5) {
                # assumes model M1: three w's, w0, w1 and w2 (positive selection)
                printf "position %d: (%g prob, %g omega, %g mean w)\n",
                  $i, $probs[$i]-&gt;[2], $params-&gt;{w2}, $probs[$i]-&gt;[3];
             }
           }
         } else { print "No positive selection found!\n"; }

</pre><h4><b>TODO</b></h4><pre>
       RST parsing -- done, Avilella contributions bug#1506, added by jason 1.29
                   -- still need to parse in joint probability and non-syn changes
                      at site table

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>              - General discussion
         <a href="http://bioperl.org/Support.html">http://bioperl.org/Support.html</a>    - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list: <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bio-tools-phylo-paml/issues

</pre><h4><b>AUTHORS</b></h4><pre>
       Jason Stajich &lt;<a href="mailto:jason@bioperl.org">jason@bioperl.org</a>&gt;

       Aaron Mackey &lt;<a href="mailto:amackey@virginia.edu">amackey@virginia.edu</a>&gt;

       Albert Vilella &lt;<a href="mailto:avilella@gmail.com">avilella@gmail.com</a>&gt;

       Sendu Bala &lt;<a href="mailto:bix@sendu.me.uk">bix@sendu.me.uk</a>&gt;

       Dave Messina &lt;<a href="mailto:dmessina@cpan.org">dmessina@cpan.org</a>&gt;

</pre><h4><b>COPYRIGHT</b></h4><pre>
       This software is copyright (c) by Jason Stajich &lt;<a href="mailto:jason@bioperl.org">jason@bioperl.org</a>&gt;, and by Aaron Mackey
       &lt;<a href="mailto:amackey@virginia.edu">amackey@virginia.edu</a>&gt;.

       This software is available under the same terms as the perl 5 programming language system itself.

perl v5.32.0                                       2020-11-10                       <u>Bio::Tools::Phylo::<a href="../man3pm/PAML.3pm.html">PAML</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>