<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Run::Pseudowise - Object for prediting pseudogenes in a given sequence given a protein and a</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-perl-run-perl">libbio-perl-run-perl_1.7.3-8_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Run::Pseudowise - Object for prediting pseudogenes in a given sequence given a protein and a
       cdna sequence

</pre><h4><b>SYNOPSIS</b></h4><pre>
         # Build a pseudowise alignment factory
         my $factory = Bio::Tools::Run::Pseudowise-&gt;new();

         # Pass the factory 3 Bio:SeqI objects (in the order of query
         # peptide and cdna and target_genomic)
         # @genes is an array of GenericSeqFeature objects
         my @genes = $factory-&gt;run($seq1, $seq2, $seq3);

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Pseudowise is a pseudogene predition program developed by Ewan Birney
       <a href="http://www.sanger.ac.uk/software/wise2">http://www.sanger.ac.uk/software/wise2</a>.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         <a href="http://redmine.open-bio.org/projects/bioperl/">http://redmine.open-bio.org/projects/bioperl/</a>

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Kiran</b></h4><pre>
       Email <a href="mailto:kiran@fugu-sg.org">kiran@fugu-sg.org</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>program_name</b>
        Title   : program_name
        Usage   : $factory&gt;program_name()
        Function: holds the program name
        Returns:  string
        Args    : None

   <b>program_dir</b>
        Title   : program_dir
        Usage   : $factory-&gt;program_dir(@params)
        Function: returns the program directory, obtained from ENV variable.
        Returns:  string
        Args    :

   <b>version</b>
        Title   : version
        Usage   : exit if $prog-&gt;version() &lt; 1.8
        Function: Determine the version number of the program
        Example :
        Returns : float or undef
        Args    : none

   <b>predict_genes</b>
        Title   : predict_genes
        Usage   : DEPRECATED. Use $factory-&gt;run instead
        Function: Predict pseudogenes
        Returns : An array of Bio::Seqfeature::Generic objects
        Args    : Name of a file containing a set of 3 fasta sequences in the order of
                  peptide, cdna and genomic sequences
                  or else 3  Bio::Seq objects.

       Throws an exception if argument is not either a string (eg a filename) or 3 Bio::Seq objects.  If
       arguments are strings, throws exception if file corresponding to string name can not be found.

   <b>run</b>
        Title   : run
        Usage   : my @feats = $factory-&gt;run($seq1, $seq2, $seq3);
        Function: Executes pseudogene binary
        Returns : An array of Bio::Seqfeature::Generic objects
        Args    : Name of a file containing a set of 3 fasta sequences in the order of
                  peptide, cdna and genomic sequences
                  or else 3  Bio::Seq objects.

       Throws an exception if argument is not either a string (eg a filename) or 3 Bio::Seq objects.  If
       arguments are strings, throws exception if file corresponding to string name can not be found.

   <b>_run</b>
        Title   :  _run
        Usage   :  Internal function, not to be called directly
        Function:   makes actual system call to a pseudowise program
        Example :
        Returns : nothing; pseudowise  output is written to a
                  temporary file $TMPOUTFILE
        Args    : Name of a files containing 3 sequences in the order of peptide, cdna and genomic

   <b>_parse_results</b>
        Title   :  __parse_results
        Usage   :  Internal function, not to be called directly
        Function:  Parses pseudowise output
        Example :
        Returns : an reference to an array of Seqfeatures
        Args    : the name of the output file

   <b>_setinput()</b>
        Title   :  _setinput
        Usage   :  Internal function, not to be called directly
        Function:   Create input files for pseudowise program
        Example :
        Returns : name of file containing dba data input
        Args    : Seq objects in the order of query protein and cdna and target genomic sequence

   <b>_query_pep_seq()</b>
        Title   :  _query_pep_seq
        Usage   :  Internal function, not to be called directly
        Function:  get/set for the query sequence
        Example :
        Returns :
        Args    :

   <b>_query_cdna_seq()</b>
        Title   :  _query_cdna_seq
        Usage   :  Internal function, not to be called directly
        Function:  get/set for the query sequence
        Example :
        Returns :
        Args    :

   <b>_subject_dna_seq()</b>
        Title   :  _subject_dna_seq
        Usage   :  Internal function, not to be called directly
        Function:  get/set for the subject sequence
        Example :
        Returns :

        Args    :

perl v5.34.0                                       2022-02-10                   <u>Bio::Tools::Run::<a href="../man3pm/Pseudowise.3pm.html">Pseudowise</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>