<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::SeqFeature::Tools::FeatureNamer - generates unique persistent names for features</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::SeqFeature::Tools::FeatureNamer - generates unique persistent names for features

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::SeqIO;
         use Bio::SeqFeature::Tools::FeatureNamer;

         # first fetch a genbank SeqI object
         $seqio =
           Bio::SeqIO-&gt;new(-file=&gt;'AE003644.gbk',
                           -format=&gt;'GenBank');
         $seq = $seqio-&gt;next_seq();

         $namer = Bio::SeqFeature::Tools::FeatureNamer-&gt;new;
         my @features = $seq-&gt;get_SeqFeatures;
         foreach my $feature (@features) {
           $namer-&gt;name_feature($feature) unless $feature-&gt;display_name;
         }

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This is a helper class for providing names for SeqFeatures

       The Bio::SeqFeatureI class provides a display_name method. Typically the display_name is not set when
       parsing formats such as genbank - instead properties such as <b>label</b>, <b>product</b> or <b>gene</b> are set in a somewhat
       inconsistent manner.

       In addition, when generating subfeatures (for example, exons that are subfeatures of a transcript
       feature), it is often desirable to name these subfeatures before either exporting to another format or
       reporting to the user.

       This module is intended to help given uniform display_names to features and their subfeatures.

</pre><h4><b>TODO</b></h4><pre>
       Currently the naming policy is hardcoded. It may be desirable to allow plugging in variations on naming
       policies; this could be done either by subclassing, anonymous subroutines (closures) or parameterization.
       Contact the author if you feel you have need for a different naming policy

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing lists  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Chris</b> <b>Mungall</b></h4><pre>
       Email:  cjm AT fruitfly DOT org

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : $unflattener = Bio::SeqFeature::Tools::FeatureNamer-&gt;new();
        Function: constructor
        Example :
        Returns : a new Bio::SeqFeature::Tools::FeatureNamer
        Args    : see below

   <b>name_feature</b>
        Title   : name_feature
        Usage   : $namer-&gt;name_feature($sf);
        Function: sets display_name
        Example :
        Returns :
        Args    : L&lt;Bio::SeqFeatureI&gt;

       This method calls <b>generate_feature_name()</b> and uses the returned value to set the display_name of the
       feature

   <b>name_contained_features</b>
        Title   : name_contained_features
        Usage   : $namer-&gt;name_contained_features($sf);
        Function: sets display_name for all features contained by sf
        Example :
        Returns :
        Args    : L&lt;Bio::SeqFeatureI&gt;

       iterates through all subfeatures of a certain feature (using get_all_SeqFeatures) and names each
       subfeatures, based on the generated name for the holder feature

       A subfeature is named by concatenating the generated name of the container feature with the type and a
       number.

       For example, if the containing feature is a gene with display name <b>dpp</b>, subfeatures will be named
       dpp-mRNA-1 dpp-mRNA2 dpp-exon1 dpp-exon2 etc

   <b>generate_feature_name</b>
        Title   : generate_feature_name
        Usage   : $name = $namer-&gt;generate_feature_name($sf);
        Function: derives a sensible human readable name for a $sf
        Example :
        Returns : str
        Args    : L&lt;Bio::SeqFeatureI&gt;

       returns a generated name (but does not actually set display_name).

       If display_name is already set, the method will return this

       Otherwise, the name will depend on the property:

       label
       product
       gene
       locus_tag

       (in order of priority)

perl v5.32.1                                       2021-08-15             <u>Bio::SeqFeature...s::<a href="../man3pm/FeatureNamer.3pm.html">FeatureNamer</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>