<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::DB::GFF::Segment -- Simple DNA segment object</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-db-gff-perl">libbio-db-gff-perl_1.7.4-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::DB::GFF::Segment -- Simple DNA segment object

</pre><h4><b>SYNOPSIS</b></h4><pre>
       See Bio::DB::GFF.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::DB::GFF::Segment provides the basic representation of a range of DNA contained in a GFF database.
       It is the base class from which the Bio::DB::GFF::RelSegment and Bio::DB::GFF::Feature classes are
       derived.

       Generally, you will not create or manipulate Bio::DB::GFF::Segment objects directly, but use those that
       are returned by the Bio::DB::GFF module.

</pre><h4><b>API</b></h4><pre>
       The remainder of this document describes the API for Bio::DB::GFF::Segment.

   <b>new</b>
        Title   : new
        Usage   : $s = Bio::DB::GFF::Segment-&gt;new(@args)
        Function: create a new segment
        Returns : a new Bio::DB::GFF::Segment object
        Args    : see below
        Status  : Public

       This method creates a new Bio::DB::GFF::Segment object.  Generally this is called automatically by the
       Bio::DB::GFF module and derivatives.

       There are five positional arguments:

        $factory      a Bio::DB::GFF::Adaptor to use for database access
        $sourceseq    ID of the source sequence
        $sourceclass  class of the source sequence
        $start        start of the desired segment relative to source sequence
        $stop         stop of the desired segment relative to source sequence

   <b>factory</b>
        Title   : factory
        Usage   : $s-&gt;factory
        Function: get the factory object
        Returns : a Bio::DB::GFF::Adaptor
        Args    : none
        Status  : Public

       This is a read-only accessor for the Bio::DB::GFF::Adaptor object used to create the segment.

   <b>start</b>
        Title   : start
        Usage   : $s-&gt;start
        Function: start of segment
        Returns : integer
        Args    : none
        Status  : Public

       This is a read-only accessor for the start of the segment.

   <b>end</b>
        Title   : end
        Usage   : $s-&gt;end
        Function: end of segment
        Returns : integer
        Args    : none
        Status  : Public

       This is a read-only accessor for the end of the segment.

   <b>stop</b>
        Title   : stop
        Usage   : $s-&gt;stop
        Function: stop of segment
        Returns : integer
        Args    : none
        Status  : Public

       This is an alias for <b>end()</b>, provided for AcePerl compatibility.

   <b>length</b>
        Title   : length
        Usage   : $s-&gt;length
        Function: length of segment
        Returns : integer
        Args    : none
        Status  : Public

       Returns the length of the segment.  Always a positive number.

   <b>strand</b>
        Title   : strand
        Usage   : $s-&gt;strand
        Function: strand of segment
        Returns : +1,0,-1
        Args    : none
        Status  : Public

       Returns the strand on which the segment resides, either +1, 0 or -1.

   <b>low</b>
        Title   : low
        Usage   : $s-&gt;low
        Function: return lower coordinate
        Returns : lower coordinate
        Args    : none
        Status  : Public

       Returns the lower coordinate, either start or end.

   <b>high</b>
        Title   : high
        Usage   : $s-&gt;high
        Function: return higher coordinate
        Returns : higher coordinate
        Args    : none
        Status  : Public

       Returns the higher coordinate, either start or end.

   <b>sourceseq</b>
        Title   : sourceseq
        Usage   : $s-&gt;sourceseq
        Function: get the segment source
        Returns : a string
        Args    : none
        Status  : Public

       Returns the name of the source sequence for this segment.

   <b>class</b>
        Title   : class
        Usage   : $s-&gt;class([$newclass])
        Function: get the source sequence class
        Returns : a string
        Args    : new class (optional)
        Status  : Public

       Gets or sets the class for the source sequence for this segment.

   <b>subseq</b>
        Title   : subseq
        Usage   : $s-&gt;subseq($start,$stop)
        Function: generate a subsequence
        Returns : a Bio::DB::GFF::Segment object
        Args    : start and end of subsequence
        Status  : Public

       This method generates a new segment from the start and end positions given in the arguments.  If stop &lt;
       start, then the strand is reversed.

   <b>seq</b>
        Title   : seq
        Usage   : $s-&gt;seq
        Function: get the sequence string for this segment
        Returns : a Bio::PrimarySeq
        Args    : none
        Status  : Public

       Returns the sequence for this segment as a Bio::PrimarySeq.  (-) strand segments are automatically
       reverse complemented

       The method is called <b>dna()</b> return the data as a simple sequence string.

   <b>dna</b>
        Title   : dna
        Usage   : $s-&gt;dna
        Function: get the DNA string for this segment
        Returns : a string
        Args    : none
        Status  : Public

       Returns the sequence for this segment as a simple string. (-) strand segments are automatically reverse
       complemented

       The method is also called <b>protein()</b>.

   <b>primary_seq</b>
        Title   : primary_seq
        Usage   : $s-&gt;primary_seq
        Function: returns a Bio::PrimarySeqI compatible object
        Returns : a Bio::PrimarySeqI object
        Args    : none
        Status  : Public

       This is for compatibility with BioPerl's separation of SeqI from PrimarySeqI.  It just returns itself.

   <b>type</b>
        Title   : type
        Usage   : $s-&gt;type
        Function: return the string "feature"
        Returns : the string "feature"
        Args    : none
        Status  : Public

       This is for future sequence ontology-compatibility and represents the default type of a feature on the
       genome

   <b>equals</b>
        Title   : equals
        Usage   : $s-&gt;equals($d)
        Function: segment equality
        Returns : true, if two segments are equal
        Args    : another segment
        Status  : Public

       Returns true if the two segments have the same source sequence, start and stop.

   <b>asString</b>
        Title   : asString
        Usage   : $s-&gt;asString
        Function: human-readable string for segment
        Returns : a string
        Args    : none
        Status  : Public

       Returns a human-readable string representing this sequence.  Format is:

          sourceseq/start,stop

   <b>clone</b>
        Title   : clone
        Usage   : $copy = $s-&gt;clone
        Function: make a copy of this segment
        Returns : a Bio::DB::GFF::Segment object
        Args    : none
        Status  : Public

       This method creates a copy of the segment and returns it.

   <b>error</b>
        Title   : error
        Usage   : $error = $s-&gt;error([$new_error])
        Function: get or set the last error
        Returns : a string
        Args    : an error message (optional)
        Status  : Public

       In case of a fault, this method can be used to obtain the last error message.  Internally it is called to
       set the error message.

</pre><h4><b>Relative</b> <b>Addressing</b> <b>Methods</b></h4><pre>
       The following methods are provided for compatibility with Bio::DB::GFF::RelSegment, which provides
       relative addressing functions.

   <b>abs_start</b>
        Title   : abs_start
        Usage   : $s-&gt;abs_start
        Function: the absolute start of the segment
        Returns : an integer
        Args    : none
        Status  : Public

       This is an alias to <b>start()</b>, and provided for API compatibility with Bio::DB::GFF::RelSegment.

   <b>abs_end</b>
        Title   : abs_end
        Usage   : $s-&gt;abs_end
        Function: the absolute stop of the segment
        Returns : an integer
        Args    : none
        Status  : Public

       This is an alias to <b>stop()</b>, and provided for API compatibility with Bio::DB::GFF::RelSegment.

   <b>abs_strand</b>
        Title   : abs_strand
        Usage   : $s-&gt;abs_strand
        Function: the absolute strand of the segment
        Returns : +1,0,-1
        Args    : none
        Status  : Public

       This is an alias to <b>strand()</b>, and provided for API compatibility with Bio::DB::GFF::RelSegment.

   <b>abs_ref</b>
        Title   : abs_ref
        Usage   : $s-&gt;abs_ref
        Function: the reference sequence for this segment
        Returns : a string
        Args    : none
        Status  : Public

       This is an alias to <b>sourceseq()</b>, and is here to provide API compatibility with Bio::DB::GFF::RelSegment.

   <b>refseq</b>
        Title   : refseq
        Usage   : $s-&gt;refseq
        Function: get or set the reference sequence
        Returns : a string
        Args    : none
        Status  : Public

       Examine or change the reference sequence. This is an alias to <b>sourceseq()</b>, provided here for API
       compatibility with Bio::DB::GFF::RelSegment.

   <b>ref</b>
        Title   : ref
        Usage   : $s-&gt;refseq
        Function: get or set the reference sequence
        Returns : a string
        Args    : none
        Status  : Public

       An alias for <b>refseq()</b>

   <b>seq_id</b>
        Title   : seq_id
        Usage   : $ref = $s-&gt;seq_id
        Function: get the reference sequence in a LocationI-compatible way
        Returns : a string
        Args    : none
        Status  : Public

       An alias for <b>refseq()</b> but only allows reading.

   <b>truncated</b>
        Title   : truncated
        Usage   : $truncated = $s-&gt;truncated
        Function: Flag indicating that the segment was truncated during creation
        Returns : A boolean flag
        Args    : none
        Status  : Public

       This indicates that the sequence was truncated during creation.  The returned flag is undef if no
       truncation occurred.  If truncation did occur, the flag is actually an array ref in which the first
       element is true if truncation occurred on the left, and the second element occurred if truncation
       occurred on the right.

   <b>Bio::RangeI</b> <b>Methods</b>
       The following Bio::RangeI methods are supported:

       <b>overlaps()</b>, <b>contains()</b>, <b>equals()</b>,<b>intersection()</b>,<b>union()</b>,<b>overlap_extent()</b>

   <b>Bio::SeqI</b> <b>implementation</b>
   <b>primary_id</b>
        Title   : primary_id
        Usage   : $unique_implementation_key = $obj-&gt;primary_id;
        Function: Returns the unique id for this object in this
                  implementation. This allows implementations to manage their
                  own object ids in a way the implementation can control
                  clients can expect one id to map to one object.

                  For sequences with no accession number, this method should
                  return a stringified memory location.

        Returns : A string
        Args    : None
        Status  : Virtual

   <b>display_name</b>
        Title   : display_name
        Usage   : $id = $obj-&gt;display_name or $obj-&gt;display_name($newid);
        Function: Gets or sets the display id, also known as the common name of
                  the Seq object.

                  The semantics of this is that it is the most likely string
                  to be used as an identifier of the sequence, and likely to
                  have "human" readability.  The id is equivalent to the LOCUS
                  field of the GenBank/EMBL databanks and the ID field of the
                  Swissprot/sptrembl database. In fasta format, the &gt;(\S+) is
                  presumed to be the id, though some people overload the id
                  to embed other information. Bioperl does not use any
                  embedded information in the ID field, and people are
                  encouraged to use other mechanisms (accession field for
                  example, or extending the sequence object) to solve this.

                  Notice that $seq-&gt;id() maps to this function, mainly for
                  legacy/convenience issues.
        Returns : A string
        Args    : None or a new id

       Note, this used to be called <b>display_id()</b>, and this name is preserved for backward compatibility.  The
       default is to return the <b>seq_id()</b>.

   <b>accession_number</b>
        Title   : accession_number
        Usage   : $unique_biological_key = $obj-&gt;accession_number;
        Function: Returns the unique biological id for a sequence, commonly
                  called the accession_number. For sequences from established
                  databases, the implementors should try to use the correct
                  accession number. Notice that primary_id() provides the
                  unique id for the implementation, allowing multiple objects
                  to have the same accession number in a particular implementation.

                  For sequences with no accession number, this method should return
                  "unknown".
        Returns : A string
        Args    : None

   <b>alphabet</b>
        Title   : alphabet
        Usage   : if( $obj-&gt;alphabet eq 'dna' ) { /Do Something/ }
        Function: Returns the type of sequence being one of
                  'dna', 'rna' or 'protein'. This is case sensitive.

                  This is not called &lt;type&gt; because this would cause
                  upgrade problems from the 0.5 and earlier Seq objects.

        Returns : a string either 'dna','rna','protein'. NB - the object must
                  make a call of the type - if there is no type specified it
                  has to guess.
        Args    : none
        Status  : Virtual

   <b>desc</b>
        Title   : desc
        Usage   : $seqobj-&gt;desc($string) or $seqobj-&gt;desc()
        Function: Sets or gets the description of the sequence
        Example :
        Returns : The description
        Args    : The description or none

   <b>species</b>
        Title   : species
        Usage   : $species = $seq-&gt;species() or $seq-&gt;species($species)
        Function: Gets or sets the species
        Example :
        Returns : Bio::Species object
        Args    : None or Bio::Species object

       See Bio::Species for more information

   <b>annotation</b>
        Title   : annotation
        Usage   : $ann = $seq-&gt;annotation or $seq-&gt;annotation($annotation)
        Function: Gets or sets the annotation
        Example :
        Returns : Bio::Annotation object
        Args    : None or Bio::Annotation object

       See Bio::Annotation for more information

   <b>is_circular</b>
        Title   : is_circular
        Usage   : if( $obj-&gt;is_circular) { /Do Something/ }
        Function: Returns true if the molecule is circular
        Returns : Boolean value
        Args    : none

</pre><h4><b>BUGS</b></h4><pre>
       Report them please.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       bioperl

</pre><h4><b>AUTHOR</b></h4><pre>
       Lincoln Stein &lt;<a href="mailto:lstein@cshl.org">lstein@cshl.org</a>&gt;.

       Copyright (c) 2001 Cold Spring Harbor Laboratory.

       This library is free software; you can redistribute it and/or modify it under the same terms as Perl
       itself.

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
       Jason Stajich &lt;<a href="mailto:jason@bioperl.org">jason@bioperl.org</a>&gt;.

perl v5.30.0                                       2020-01-13                         <u>Bio::DB::GFF::<a href="../man3pm/Segment.3pm.html">Segment</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>