<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Run::Vista - Wrapper for Vista</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-perl-run-perl">libbio-perl-run-perl_1.7.3-8_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Run::Vista - Wrapper for Vista

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::Tools::Run::Vista;
         use Bio::Tools::Run::Alignment::Lagan;
         use Bio::AlignIO;

         my $sio = Bio::SeqIO-&gt;new(-file=&gt;$ARGV[0],-format=&gt;'genbank');
         my @seq;
         my $reference = $sio-&gt;next_seq;
         push @seq, $reference;
         while(my $seq = $sio-&gt;next_seq){
           push @seq,$seq;
         }
         my @features = grep{$_-&gt;primary_tag eq 'CDS'} $reference-&gt;get_SeqFeatures;

         my $lagan = Bio::Tools::Run::Alignment::Lagan-&gt;new;

         my $aln = $lagan-&gt;mlagan(\@seq,'(fugu (mouse human))');

         my $vis = Bio::Tools::Run::Vista-&gt;new('outfile'=&gt;"outfile.pdf",
                                               'title' =&gt; "My Vista Plot",
                                               'annotation'=&gt;\@features,
                                               'annotation_format'=&gt;'GFF',
                                               'min_perc_id'=&gt;75,
                                               'min_length'=&gt;100,
                                               'plotmin'   =&gt; 50,
                                               'tickdist' =&gt; 2000,
                                               'window'=&gt;40,
                                               'numwindows'=&gt;4,
                                               'start'=&gt;50,
                                               'end'=&gt;1500,
                                               'tickdist'=&gt;100,
                                               'bases'=&gt;1000,
                                               'java_param'=&gt;"-Xmx128m",
                                               'num_pages'=&gt;1,
                                               'color'=&gt; {'EXON'=&gt;'100 0 0',
                                                          'CNS'=&gt;'0 0 100'},
                                               'quiet'=&gt;1);

         my $referenceid= 'human';
         $vis-&gt;run($aln,$referenceid);

         #alternative one can choose pairwise alignments to plot
         #where the second id in each pair is the reference sequence
         $vis-&gt;run($aln,([mouse,human],[fugu,human],[mouse,fugu]));

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Pls see Vista documentation for plotfile options

       Wrapper for Vista :

       C. Mayor, M. Brudno, J. R. Schwartz, A. Poliakov, E. M. Rubin, K. A.  Frazer, L. S. Pachter, I. Dubchak.
       VISTA: Visualizing global DNA  sequence alignments of arbitrary length.  Bioinformatics, 2000
       Nov;<a href="../man11/16.11.html">16</a>(11):1046-1047.  Get it here: <a href="http://www-gsd.lbl.gov/vista/VISTAdownload2.html">http://www-gsd.lbl.gov/vista/VISTAdownload2.html</a>

       On the command line, it is assumed that this can be executed:

       java Vista plotfile

       Some of the code was adapted from MLAGAN toolkit

       M. Brudno,  C.B. Do,  G. Cooper,  M.F. Kim,  E. Davydov,  NISC Sequencing Consortium, E.D. Green,  A.
       Sidow and S. Batzoglou LAGAN and Multi-LAGAN: Efficient Tools for Large-Scale Multiple  Alignment of
       Genomic DNA, Genome Research, in press

       get lagan here:

       <a href="http://lagan.stanford.edu/">http://lagan.stanford.edu/</a>

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         <a href="http://redmine.open-bio.org/projects/bioperl/">http://redmine.open-bio.org/projects/bioperl/</a>

</pre><h4><b>AUTHOR</b></h4><pre>
       Shawn Hoon Email <a href="mailto:shawnh@fugu-sg.org">shawnh@fugu-sg.org</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>new</b>
           Title   :   new
           Usage   :   my $vis = Bio::Tools::Run::Vista-&gt;new('outfile'=&gt;$out,
                                               'title' =&gt; "My Vista Plot",
                                               'annotation'=&gt;\@features,
                                               'annotation_format'=&gt;'GFF',
                                               'min_perc_id'=&gt;75,
                                               'min_length'=&gt;100,
                                               'plotmin'   =&gt; 50,
                                               'tickdist' =&gt; 2000,
                                               'window'=&gt;40,
                                               'numwindows'=&gt;4,
                                               'start'=&gt;50,
                                               'end'=&gt;1500,
                                               'tickdist'=&gt;100,
                                               'bases'=&gt;1000,
                                               'color'=&gt; {'EXON'=&gt;'100 0 0',
                                                          'CNS'=&gt;'0 0 100'},
                                               'quiet'=&gt;1);
           Function:   Construtor for Vista wrapper
           Args    :   outfile - location of the pdf generated
                       annotation - either a file or and array ref of Bio::SeqFeatureI
                                    indicating the exons
                       regmin     -region min

   <b>java</b>
           Title   :   java
           Usage   :   $obj-&gt;java('/usr/opt/java130/bin/java');
           Function:   Get/set method for the location of java VM
           Args    :   File path (optional)

   <b>run</b>
        Title   : run
        Usage   : my @genes = $self-&gt;run($seq)
        Function: runs Vista
        Returns : A boolean 1 if no errors
        Args    : Argument 1: Bio::Align::Align required
                  Argument 2: a string or number, which is the sequence id of the
                              reference sequence or the rank of the sequence
                              in the alignment

   <b>_setinput</b>
        Title   : _setinput
        Usage   : Internal function, not to be called directly
        Function: writes input sequence to file and return the file name
        Example :
        Returns : string
        Args    :

   <b>outfile</b>
         Title    : outfile
         Usage    : $obj-&gt;outfile
         Function : Get/Set method outfile
         Args     :

   <b>min_perc_id</b>
         Title    : min_perc_id
         Usage    : $obj-&gt;min_perc_id
         Function : Get/Set method min_perc_id
         Args     :

   <b>quiet</b>
         Title    : quiet
         Usage    : $obj-&gt;quiet
         Function : Get/Set method quiet
         Args     :

   <b>verbose</b>
         Title    : verbose
         Usage    : $obj-&gt;verbose
         Function : Get/Set method verbose
         Args     :

   <b>annotation_format</b>
         Title    : annotation_format
         Usage    : $obj-&gt;annotation_format
         Function : Get/Set method annotation_format
         Args     :

   <b>region_file</b>
         Title    : region_file
         Usage    : $obj-&gt;region_file
         Function : Get/Set method region_file
         Args     :

   <b>score_file</b>
         Title    : score_file
         Usage    : $obj-&gt;score_file
         Function : Get/Set method score_file
         Args     :

   <b>alignment_file</b>
         Title    : alignment_file
         Usage    : $obj-&gt;alignment_file
         Function : Get/Set method alignment_file
         Args     :

   <b>contigs_file</b>
         Title    : contigs_file
         Usage    : $obj-&gt;contigs_file
         Function : Get/Set method contigs_file
         Args     :

   <b>diffs</b>
         Title    : diffs
         Usage    : $obj-&gt;diffs
         Function : Get/Set method diffs
         Args     :

   <b>plotfile</b>
         Title    : plotfile
         Usage    : $obj-&gt;plotfile
         Function : Get/Set method plotfile
         Args     :

   <b>min_length</b>
         Title    : min_length
         Usage    : $obj-&gt;min_length
         Function : Get/Set method min_length
         Args     :

   <b>plotmin</b>
         Title    : plotmin
         Usage    : $obj-&gt;plotmin
         Function : Get/Set method plotmin
         Args     :

   <b>annotation</b>
         Title    : annotation
         Usage    : $obj-&gt;annotation
         Function : Get/Set method annotation
         Args     :

   <b>bases</b>
         Title    : bases
         Usage    : $obj-&gt;bases
         Function : Get/Set method bases
         Args     :

   <b>tickdist</b>
         Title    : tickdist
         Usage    : $obj-&gt;tickdist
         Function : Get/Set method tickdist
         Args     :

   <b>resolution</b>
         Title    : resolution
         Usage    : $obj-&gt;resolution
         Function : Get/Set method resolution
         Args     :

   <b>title</b>
         Title    : title
         Usage    : $obj-&gt;title
         Function : Get/Set method title
         Args     :

   <b>window</b>
         Title    : window
         Usage    : $obj-&gt;window
         Function : Get/Set method window
         Args     :

   <b>numwindows</b>
         Title    : numwindows
         Usage    : $obj-&gt;numwindows
         Function : Get/Set method numwindows
         Args     :

   <b>start</b>
         Title    : start
         Usage    : $obj-&gt;start
         Function : Get/Set method start
         Args     :

   <b>end</b>
         Title    : end
         Usage    : $obj-&gt;end
         Function : Get/Set method end
         Args     :

   <b>num_plot_lines</b>
         Title    : num_plot_lines
         Usage    : $obj-&gt;num_plot_lines
         Function : Get/Set method num_plot_lines
         Args     :

   <b>legend</b>
         Title    : legend
         Usage    : $obj-&gt;legend
         Function : Get/Set method legend
         Args     :

   <b>filename</b>
         Title    : filename
         Usage    : $obj-&gt;filename
         Function : Get/Set method filename
         Args     :

   <b>axis_label</b>
         Title    : axis_label
         Usage    : $obj-&gt;axis_label
         Function : Get/Set method axis_label
         Args     :

   <b>ticks_file</b>
         Title    : ticks_file
         Usage    : $obj-&gt;ticks_file
         Function : Get/Set method ticks_file
         Args     :

   <b>color</b>
         Title    : color
         Usage    : $obj-&gt;color
         Function : Get/Set method color
         Args     :

   <b>use_order</b>
         Title    : use_order
         Usage    : $obj-&gt;use_order
         Function : Get/Set method use_order
         Args     :

   <b>gaps</b>
         Title    : gaps
         Usage    : $obj-&gt;gaps
         Function : Get/Set method gaps
         Args     :

   <b>snps_file</b>
         Title    : snps_file
         Usage    : $obj-&gt;snps_file
         Function : Get/Set method snps_file
         Args     :

   <b>repeats_file</b>
         Title    : repeats_file
         Usage    : $obj-&gt;repeats_file
         Function : Get/Set method repeats_file
         Args     :

   <b>filter_repeats</b>
         Title    : filter_repeats
         Usage    : $obj-&gt;filter_repeats
         Function : Get/Set method filter_repeats
         Args     :

perl v5.34.0                                       2022-02-10                        <u>Bio::Tools::Run::<a href="../man3pm/Vista.3pm.html">Vista</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>