<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Seq::PrimedSeq - A sequence and a pair of primers matching on it</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Seq::PrimedSeq - A sequence and a pair of primers matching on it

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::Seq;
         use Bio::Seq::PrimedSeq;

         my $template = Bio::Seq-&gt;new( -seq =&gt; 'AGCTTTTCATTCTGACTGCAAC' );
         my $left     = Bio::Seq-&gt;new( -seq =&gt; 'AGCT' );
         my $right    = Bio::Seq-&gt;new( -seq =&gt; 'GTTGC' );

         my $primedseq = Bio::Seq::PrimedSeq-&gt;new(
                 -seq          =&gt; $template,  # sequence object
                 -left_primer  =&gt; $left,      # sequence or primer object
                 -right_primer =&gt; $right      # sequence or primer object
         );

         # Get the primers as Bio::SeqFeature::Primer objects
         my @primer_objects = $primedseq-&gt;get_primer();

         # Sequence object representing the PCR product, i.e. the section of the target
         # sequence contained between the primers (primers included)
         my $amplicon_seq = $primedseq-&gt;amplicon();

         print 'Got amplicon sequence: '.$amplicon_seq-&gt;seq."\n";
         # Amplicon should be: agctTTTCATTCTGACTgcaac

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This module was created to address the fact that a primer is more than a SeqFeature and that there is a
       need to represent the primer-sequence complex and the attributes that are associated with the complex.

       A PrimedSeq object is initialized with a target sequence and two primers. The location of the primers on
       the target sequence is calculated if needed so that one can then get the PCR product, or amplicon
       sequence. From the PrimedSeq object you can also retrieve information about melting temperatures and what
       not on each of the primers and the amplicon. The Bio::Tools::Primer3 module uses PrimedSeq objects
       extensively.

       Note that this module does not simulate PCR. It assumes that the primers will match in a single location
       on the target sequence and does not understand degenerate primers.

       •   Providing primers as sequence objects

           If  the  primers  are specified as sequence objects, e.g. Bio::PrimarySeq or Bio::Seq, they are first
           converted to Bio::SeqFeature::Primer  objects.   Their  location  on  the  target  sequence  is  then
           calculated  and  added  to  the  Bio::SeqFeature::Primer  objects,  which  you can retrieve using the
           <b>get_primer()</b> method.

       •   Providing primers as primer objects

           Because of the limitations of specifying primers as sequence objects, the recommended  method  is  to
           provide  Bio::SeqFeature::Primer  objects.  If  you did not record the location of the primers in the
           primer object, it will be calculated.

       Bio::Seq::PrimedSeq was initially started by Chad Matsalla, and later improved on by Rob Edwards.

</pre><h4><b>RECIPES</b></h4><pre>
       1.  Run Primer3 to get PrimedSeq objects:

             use Bio::SeqIO;
             use Bio::Tools::Run::Primer3;

             # Read a target sequences from a FASTA file
             my $file = shift || die "Need a file to read";
             my $seqin = Bio::SeqIO-&gt;new(-file =&gt; $file);
             my $seq = $seqin-&gt;next_seq;

             # Use Primer3 to design some primers
             my $primer3 = Bio::Tools::Run::Primer3-&gt;new(-seq =&gt; $seq);
             my $results = $primer3-&gt;run; # default parameters

             # Write all the results in a Genbank file
             my $seqout = Bio::SeqIO-&gt;new(-file =&gt; "&gt;primed_sequence.gbk",
                                          -format =&gt; 'genbank');
             while (my $primedseq = $results-&gt;next_primer) {
                $seqout-&gt;write_seq( $primedseq-&gt;annotated_seq );
             }

       2.  Create a genbank file for a sequence and its cognate primers:

             use Bio::SeqIO;
             use Bio::Seq::PrimedSeq;

             # Read a FASTA file that contains the target sequence and its two primers
             my $file = shift || die "$0 &lt;file&gt;";
             my $seqin = Bio::SeqIO-&gt;new(-file =&gt; $file);
             my ($template, $leftprimer, $rightprimer) =
                   ($seqin-&gt;next_seq, $seqin-&gt;next_seq, $seqin-&gt;next_seq);

             # Set up a PrimedSeq object
             my $primedseq = Bio::Seq::PrimedSeq-&gt;new(-seq =&gt; $template,
                                                      -left_primer =&gt; $leftprimer,
                                                      -right_primer =&gt; $rightprimer);

             # Write the sequences in an output Genbank file
             my $seqout = Bio::SeqIO-&gt;new(-file =&gt; "&gt;primed_sequence.gbk",
                                          -format =&gt; 'genbank');
             $seqout-&gt;write_seq($primedseq-&gt;annotated_sequence);

</pre><h4><b>FEEDBACK</b></h4><pre>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your  comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather  than  to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.   Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b></h4><pre>
       Rob Edwards, <a href="mailto:redwards@utmem.edu">redwards@utmem.edu</a>

       Based on a module written by Chad Matsalla, <a href="mailto:bioinformatics1@dieselwurks.com">bioinformatics1@dieselwurks.com</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The  rest  of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new()
        Usage   : my $primedseq = Bio::SeqFeature::Primer-&gt;new(
                                   -seq =&gt; $sequence,
                                   -left_primer =&gt; $left_primer,
                                   -right_primer =&gt; $right_primer
                  );
        Function: Construct a primed sequence.
        Returns : A Bio::Seq::PrimedSeq object
        Args    :  -seq =&gt; a Bio::Seq object (required)
                   -left_primer =&gt; a Bio::SeqFeature::Primer or sequence object (required)
                   -right_primer =&gt; a Bio::SeqFeature::Primer or sequence object (required)

                  If you pass a sequence object to specify a primer, it will be used to
                  construct a Bio::SeqFeature::Primer that you can retrieve with the
                  L&lt;get_primer&gt; method.

                  Many other parameters can be included including all of the output
                  parameters from the primer3 program (see L&lt;Bio::Tools::Primer3&gt;). At
                  the moment these parameters will simply be stored and do anything.

   <b>get_primer</b>
        Title   : get_primer();
        Usage   :  my @primers = $primedseq-&gt;get_primer();
                     or
                   my $primer = $primedseq-&gt;get_primer('-left_primer');
        Function: Get the primers associated with the PrimedSeq object.
        Returns : A Bio::SeqFeature::Primer object
        Args    : For the left primer, use: l, left, left_primer or -left_primer
                  For the right primer, use: r, right, right_primer or -right_primer
                  For both primers [default], use: b, both, both_primers or -both_primers

   <b>annotated_sequence</b>
        Title   : annotated_sequence
        Usage   : my $annotated_sequence_object = $primedseq-&gt;annotate_sequence();
        Function: Get an annotated sequence object containing the left and right
                  primers
        Returns : An annotated sequence object or 0 if not defined.
        Args    :
        Note    : Use this method to return a sequence object that you can write
                  out (e.g. in GenBank format). See the example above.

   <b>amplicon</b>
        Title   : amplicon
        Usage   : my $amplicon = $primedseq-&gt;amplicon();
        Function: Retrieve the amplicon as a sequence object. The amplicon sequnce is
                  only the section of the target sequence between the primer matches
                  (primers included).
        Returns : A Bio::Seq object. To get the sequence use $amplicon-&gt;seq
        Args    : None
        Note    :

   <b>seq</b>
        Title   : seq
        Usage   : my $seqobj = $primedseq-&gt;seq();
        Function: Retrieve the target sequence as a sequence object
        Returns : A seq object. To get the sequence use $seqobj-&gt;seq
        Args    : None
        Note    :

   <b>_place_primers</b>
        Title   : _place_primers
        Usage   : $self-&gt;_place_primers();
        Function: An internal method to place the primers on the sequence and
                  set up the ranges of the sequences
        Returns : Nothing
        Args    : None
        Note    : Internal use only

perl v5.32.1                                       2021-08-15                           <u>Bio::Seq::<a href="../man3pm/PrimedSeq.3pm.html">PrimedSeq</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>