<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Seq::Meta - Generic superclass for sequence objects with residue-based meta information</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Seq::Meta - Generic superclass for sequence objects with residue-based meta information

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::LocatableSeq;
         use Bio::Seq::Meta;
         use Bio::Tools::OddCodes;
         use Bio::SeqIO;

         my $seq = Bio::Seq::Meta-&gt;new(-id=&gt;'test',
                                          -seq=&gt;'ACTGCTAGCT',
                                          -start=&gt;2434,
                                          -end=&gt;2443,
                                          -strand=&gt;1,
                                          -verbose=&gt;1, # to see warnings
                                         );

         # the existing sequence object can be a Bio::PrimarySeq, too

         # to test this is a meta seq object
         $seq-&gt;isa("Bio::Seq::Meta")
             || $seq-&gt;throw("$seq is not a Bio::Seq::Meta");

         $seq-&gt;meta('1234567890');
         $seq = Bio::Seq::Meta-&gt;new(-id=&gt;'test',
                                    -seq=&gt;'HACILMIFGT',
                                    -start=&gt;2434,
                                    -end=&gt;2443,
                                    -strand=&gt;1,
                                    -meta=&gt;'1234567890',
                                    -verbose=&gt;1, # to see warnings
                                   );

         # accessors
         $string     = $seq-&gt;meta_text();
         $substring  = $seq-&gt;submeta_text(2,5);
         $unique_key = $seq-&gt;accession_number();

         # storing output from Bio::Tools::OddCodes as meta data
         my $protcodes = Bio::Tools::OddCodes-&gt;new(-seq =&gt; $seq);
         my @codes = qw(structural chemical functional charge hydrophobic);
         map { $seq-&gt;named_meta($_, ${$protcodes-&gt;$_($seq) } )} @codes;

         my $out = Bio::SeqIO-&gt;new(-format=&gt;'metafasta');
         $out-&gt;write_seq($seq);

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This class implements generic methods for sequences with residue-based meta information. Meta sequences
       with meta data are Bio::LocatableSeq objects with additional methods to store that meta information. See
       Bio::LocatableSeq and Bio::Seq::MetaI.

       The meta information in this class is always one character per residue long and blank values are space
       characters (ASCII 32).

       After the latest rewrite, the meta information no longer covers all the residues automatically. Methods
       to check the length of meta information (meta_length)and to see if the ends are flushed to the sequence
       have been added (is_flush). To force the old functionality, set force_flush to true.

       It is assumed that meta data values do not depend on the nucleotide sequence strand value.

       Application specific implementations should inherit from this class to override and add to these methods.

       Bio::Seq::Meta::Array allows for more complex meta values (scalars or objects) to be used.

   <b>Method</b> <b>naming</b>
       Character based meta data is read and set by method <b>meta()</b> and its variants. These are the suffixes and
       prefixes used in the variants:

           [named_] [sub] meta [_text]

       _text
          Suffix <b>_text</b> guaranties that output is a string. Note that it does not limit the input.

          In this implementation, the output is always text, so these methods are redundant.

       sub
          Prefix  <b>sub</b>,  like  in <b>subseq()</b>, means that the method applies to sub region of the sequence range and
          takes start and end as arguments.  Unlike <b>subseq()</b>, these methods are able  to  set  values.   If  the
          range is not defined, it defaults to the complete sequence.

       named
          Prefix  <b>named_</b>  in  method  names  allows  the used to attach multiple meta strings to one sequence by
          explicitly naming them. The name is always the first argument to the method. The "unnamed" methods use
          the class wide default name for the meta data and are thus special cases "named" methods.

          Note that internally names are keys in a hash and any misspelling of a name will  silently  store  the
          data  under  a  wrong  name.  The  used  names  (keys) can be retrieved using method <b>meta_names()</b>. See
          meta_names.

</pre><h4><b>NOTE</b></h4><pre>
       This  Bio::Seq::MetaI  implementation  inherits  from  Bio::LocatableSeq,  which  itself  inherits   from
       Bio::PrimarySeq. It is not a Bio::SeqI, so bless-ing objects of this class into a Bio::SeqI or vice versa
       and  will  not  work  as  expected  (see  bug  2262).  This  may  be  addressed  in  a future refactor of
       Bio::LocatableSeq.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Bio::LocatableSeq, Bio::Seq::MetaI, Bio::Seq::Meta::Array

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your  comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather  than  to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.   Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Heikki</b> <b>Lehvaslaiho</b></h4><pre>
       Email heikki-at-bioperl-dot-org

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
       Chad Matsalla, <a href="mailto:bioinformatics@dieselwurks.com">bioinformatics@dieselwurks.com</a>

       Aaron Mackey, <a href="mailto:amackey@virginia.edu">amackey@virginia.edu</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The  rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : $metaseq = Bio::Seq::Meta-&gt;new
                       ( -meta =&gt; 'aaaaaaaabbbbbbbb',
                         -seq =&gt;  'TKLMILVSHIVILSRM'
                         -id  =&gt; 'human_id',
                         -accession_number =&gt; 'S000012',
                       );
        Function: Constructor for Bio::Seq::Meta class, meta data being in a
                  string. Note that you can provide an empty quality string.
        Returns : a new Bio::Seq::Meta object

   <b>meta</b>
        Title   : meta
        Usage   : $meta_values  = $obj-&gt;meta($values_string);
        Function:

                  Get and set method for the meta data starting from residue
                  position one. Since it is dependent on the length of the
                  sequence, it needs to be manipulated after the sequence.

                  The length of the returned value always matches the length
                  of the sequence, if force_flush() is set. See L&lt;force_flush&gt;.

        Returns : meta data in a string
        Args    : new value, string, optional

   <b>meta_text</b>
        Title   : meta_text
        Usage   : $meta_values  = $obj-&gt;meta_text($values_arrayref);
        Function: Variant of meta() guarantied to return a textual
                  representation  of meta data. For details, see L&lt;meta&gt;.
        Returns : a string
        Args    : new value, optional

   <b>named_meta</b>
        Title   : named_meta()
        Usage   : $meta_values  = $obj-&gt;named_meta($name, $values_arrayref);
        Function: A more general version of meta(). Each meta data set needs
                  to be named. See also L&lt;meta_names&gt;.
        Returns : a string
        Args    : scalar, name of the meta data set
                  new value, optional

   <b>_test_gap_positions</b>
        Title   : _test_gap_positions
        Usage   : $meta_values  = $obj-&gt;_test_gap_positions($name);
        Function: Internal test for correct position of gap characters.
                  Gap being only '-' this time.

                  This method is called from named_meta() when setting meta
                  data but only if verbose is positive as this can be an
                  expensive process on very long sequences. Set <a href="../man1/verbose.1.html">verbose</a>(1) to
                  see warnings when gaps do not align in sequence and meta
                  data and turn them into errors by setting <a href="../man2/verbose.2.html">verbose</a>(2).

        Returns : true on success, prints warnings
        Args    : none

   <b>named_meta_text</b>
        Title   : named_meta_text()
        Usage   : $meta_values  = $obj-&gt;named_meta_text($name, $values_arrayref);
        Function: Variant of named_meta() guarantied to return a textual
                  representation  of the named meta data.
                  For details, see L&lt;meta&gt;.
        Returns : a string
        Args    : scalar, name of the meta data set
                  new value, optional

   <b>submeta</b>
        Title   : submeta
        Usage   : $subset_of_meta_values = $obj-&gt;submeta(10, 20, $value_string);
                  $subset_of_meta_values = $obj-&gt;submeta(10, undef, $value_string);
        Function:

                  Get and set method for meta data for subsequences.

                  Numbering starts from 1 and the number is inclusive, ie 1-2
                  are the first two residue of the sequence. Start cannot be
                  larger than end but can be equal.

                  If the second argument is missing the returned values
                  should extend to the end of the sequence.

                  The return value may be a string or an array reference,
                  depending on the implementation. If in doubt, use
                  submeta_text() which is a variant guarantied to return a
                  string.  See L&lt;submeta_text&gt;.

        Returns : A reference to an array or a string
        Args    : integer, start position
                  integer, end position, optional when a third argument present
                  new value, optional

   <b>submeta_text</b>
        Title   : submeta_text
        Usage   : $meta_values  = $obj-&gt;submeta_text(20, $value_string);
        Function: Variant of submeta() guarantied to return a textual
                  representation  of meta data. For details, see L&lt;meta&gt;.
        Returns : a string
        Args    : new value, optional

   <b>named_submeta</b>
        Title   : named_submeta
        Usage   : $subset_of_meta_values = $obj-&gt;named_submeta($name, 10, 20, $value_string);
                  $subset_of_meta_values = $obj-&gt;named_submeta($name, 10);
        Function: Variant of submeta() guarantied to return a textual
                  representation  of meta data. For details, see L&lt;meta&gt;.
        Returns : A reference to an array or a string
        Args    : scalar, name of the meta data set
                  integer, start position
                  integer, end position, optional when a third argument present
                  new value, optional

   <b>named_submeta_text</b>
        Title   : named_submeta_text
        Usage   : $meta_values  = $obj-&gt;named_submeta_text($name, 20, $value_string);
        Function: Variant of submeta() guarantied to return a textual
                  representation  of meta data. For details, see L&lt;meta&gt;.
        Returns : a string
        Args    : scalar, name of the meta data
        Args    : integer, start position, optional
                  integer, end position, optional
                  new value, optional

   <b>meta_names</b>
        Title   : meta_names
        Usage   : @meta_names  = $obj-&gt;meta_names()
        Function: Retrieves an array of meta data set names. The default
                  (unnamed) set name is guarantied to be the first name.
        Returns : an array of names
        Args    : none

   <b>meta_length</b>
        Title   : meta_length()
        Usage   : $meeta_len  = $obj-&gt;meta_length();
        Function: return the number of elements in the meta set
        Returns : integer
        Args    : -

   <b>named_meta_length</b>
        Title   : named_meta_length()
        Usage   : $meta_len  = $obj-&gt;named_meta_length($name);
        Function: return the number of elements in the named meta set
        Returns : integer
        Args    : -

   <b>force_flush</b>
        Title   : force_flush()
        Usage   : $force_flush = $obj-&gt;<a href="../man1/force_flush.1.html">force_flush</a>(1);
        Function: Automatically pad with empty values or truncate meta values
                  to sequence length. Not done by default.
        Returns : boolean 1 or 0
        Args    : optional boolean value

       Note that if you turn this forced padding off, the previously padded values are not changed.

   <b>_do_flush</b>
        Title   : _do_flush
        Usage   :
        Function: internal method to do the force that meta values are same
                  length as the sequence . Called from L&lt;force_flush&gt;
        Returns :
        Args    :

   <b>is_flush</b>
        Title   : is_flush
        Usage   : $is_flush  = $obj-&gt;is_flush()
                  or  $is_flush = $obj-&gt;is_flush($my_meta_name)
        Function: Boolean to tell if all meta values are in
                  flush with the sequence length.
                  Returns true if force_flush() is set
                  Set verbosity to a positive value to see failed meta sets
        Returns : boolean 1 or 0
        Args    : optional name of the meta set

</pre><h4><b>Bio::PrimarySeqI</b> <b>methods</b></h4><pre>
   <b>revcom</b>
        Title   : revcom
        Usage   : $newseq = $seq-&gt;revcom();
        Function: Produces a new Bio::Seq::MetaI implementing object where
                  the order of residues and their meta information is reversed.
        Returns : A new (fresh) Bio::Seq::Meta object
        Args    : none
        Throws  : if the object returns false on is_flush()

       Note: The method does nothing to meta values, it reorders them, only.

   <b>trunc</b>
        Title   : trunc
        Usage   : $subseq = $seq-&gt;trunc(10,100);
        Function: Provides a truncation of a sequence together with meta data
        Returns : a fresh Bio::Seq::Meta implementing object
        Args    : Two integers denoting first and last residue of the sub-sequence.

perl v5.32.1                                       2021-08-15                                <u>Bio::Seq::<a href="../man3pm/Meta.3pm.html">Meta</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>