<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Das::Lite - Perl extension for the DAS (HTTP+XML) Protocol (http://biodas.org/)</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-das-lite-perl">libbio-das-lite-perl_2.11-9_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Das::Lite - Perl extension for the DAS (HTTP+XML) Protocol (<a href="http://biodas.org/">http://biodas.org/</a>)

</pre><h4><b>VERSION</b></h4><pre>
         See $Bio::Das::Lite::VERSION

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::Das::Lite;
         my $bdl     = Bio::Das::Lite-&gt;new_from_registry({'category' =&gt; 'GRCh_37,Chromosome,Homo sapiens'});
         my $results = $bdl-&gt;features('22');

</pre><h4><b>SUBROUTINES/METHODS</b></h4><pre>
   <b>new</b> <b>:</b> <b>Constructor</b>
         my $das = Bio::Das::Lite-&gt;new('<a href="http://das.ensembl.org/das/ensembl1834">http://das.ensembl.org/das/ensembl1834</a>');

         my $das = Bio::Das::Lite-&gt;new({
                                      'timeout'    =&gt; 60,
                                      'dsn'        =&gt; '<a href="http://user">http://user</a>:<a href="mailto:pass@das.ensembl.org">pass@das.ensembl.org</a>/das/ensembl1834',
                                      'http_proxy' =&gt; '<a href="http://user">http://user</a>:<a href="mailto:pass@webcache.local.com">pass@webcache.local.com</a>:3128/',
                                     });

        Options can be: dsn        (optional scalar or array ref, URLs of DAS services)
                        timeout    (optional int,      HTTP fetch timeout in seconds)
                        http_proxy (optional scalar,   web cache or proxy if not set in %ENV)
                        no_proxy   (optional list/ref, non-proxiable domains if not set in %ENV)
                        caching    (optional bool,     primitive caching on/off)
                        callback   (optional code ref, callback for processed XML blocks)
                        registry   (optional array ref containing DAS registry service URLs
                                    defaults to '<a href="http://das.sanger.ac.uk/registry/services/das">http://das.sanger.ac.uk/registry/services/das</a>')
                        proxy_user (optional scalar,   username for authenticating forward-proxy)
                        proxy_pass (optional scalar,   password for authenticating forward-proxy)
                        user_agent (optional scalar,   User-Agent HTTP request header value)

   <b>new_from_registry</b> <b>:</b> <b>Constructor</b>
         Similar to 'new' above but supports 'capability' and 'category'
         in the given hashref, using them to query the DAS registry and
         configuring the DSNs accordingly.

         my $das = Bio::Das::Lite-&gt;new_from_registry({
                                                    'capability' =&gt; ['features'],
                                                    'category'   =&gt; ['Protein Sequence'],
                                                   });

        Options are as above, plus
                        capability OR capabilities   (optional arrayref of capabilities)
                        category                     (optional arrayref of categories)

         For a complete list of capabilities and categories, see:

           <a href="http://das.sanger.ac.uk/registry/">http://das.sanger.ac.uk/registry/</a>

         The category can optionally be a full coordinate system name,
         allowing further restriction by authority, version and species.
         For example:
             'Protein Sequence' OR
             'UniProt,Protein Sequence' OR
             'GRCh_37,Chromosome,Homo sapiens'

   <b>http_proxy</b> <b>:</b> <b>Get/Set</b> <b>http_proxy</b>
           $das-&gt;http_proxy('<a href="http://user">http://user</a>:<a href="mailto:pass@squid.myco.com">pass@squid.myco.com</a>:3128/');

   <b>proxy_user</b> <b>:</b> <b>Get/Set</b> <b>proxy</b> <b>username</b> <b>for</b> <b>authenticating</b> <b>forward-proxies</b>
         This is only required if the username wasn't specified when setting http_proxy

           $das-&gt;proxy_user('myusername');

   <b>proxy_pass</b> <b>:</b> <b>Get/Set</b> <b>proxy</b> <b>password</b> <b>for</b> <b>authenticating</b> <b>forward-proxies</b>
         This is only required if the password wasn't specified when setting http_proxy

           $das-&gt;proxy_pass('secretpassword');

   <b>no_proxy</b> <b>:</b> <b>Get/Set</b> <b>domains</b> <b>to</b> <b>not</b> <b>use</b> <b>proxy</b> <b>for</b>
           $das-&gt;no_proxy('ebi.ac.uk', 'localhost');
           OR
           $das-&gt;no_proxy( ['ebi.ac.uk', 'localhost'] );

           Always returns an arrayref

   <b>user_agent</b> <b>:</b> <b>Get/Set</b> <b>user-agent</b> <b>for</b> <b>request</b> <b>headers</b>
           $das-&gt;user_agent('GroovyDAS/1.0');

   <b>timeout</b> <b>:</b> <b>Get/Set</b> <b>timeout</b>
           $das-&gt;<a href="../man30/timeout.30.html">timeout</a>(30);

   <b>caching</b> <b>:</b> <b>Get/Set</b> <b>caching</b>
           $das-&gt;<a href="../man1/caching.1.html">caching</a>(1);

   <b>callback</b> <b>:</b> <b>Get/Set</b> <b>callback</b> <b>code</b> <b>ref</b>
           $das-&gt;callback(sub { });

   <b>basename</b> <b>:</b> <b>Get</b> <b>base</b> <b>URL(s)</b> <b>of</b> <b>service</b>
           $das-&gt;basename(optional $dsn);

   <b>dsn</b> <b>:</b> <b>Get/Set</b> <b>DSN</b>
         $das-&gt;dsn('<a href="http://das.ensembl.org/das/ensembl1834/">http://das.ensembl.org/das/ensembl1834/</a>'); # give dsn (scalar or arrayref) here if not specified in new()

         Or, if you want to add to the existing dsn list and you're feeling sneaky...

         push @{$das-&gt;dsn}, '<a href="http://my.server/das/additionalsource">http://my.server/das/additionalsource</a>';

   <b>dsns</b> <b>:</b> <b>Retrieve</b> <b>information</b> <b>about</b> <b>other</b> <b>sources</b> <b>served</b> <b>from</b> <b>this</b> <b>server.</b>
        Note this call is 'dsns', as differentiated from 'dsn' which is the current configured source

         my $src_data = $das-&gt;dsns();

   <b>entry_points</b> <b>:</b> <b>Retrieve</b> <b>the</b> <b>list</b> <b>of</b> <b>entry_points</b> <b>for</b> <b>this</b> <b>source</b>
         e.g. chromosomes and associated information (e.g. sequence length and version)

         my $entry_points  = $das-&gt;entry_points();

   <b>Types</b> <b>of</b> <b>argument</b> <b>for</b> <b>'types',</b> <b>'features',</b> <b>'sequence'</b> <b>calls:</b>
         Segment Id:
         '1'

         Segment Id with range:
         '1:1,1000'

         Segment Id with range and type:
         {
           'segment' =&gt; '1:1,1000',
           'type'    =&gt; 'exon',
         }

         Multiple Ids with ranges and types:
         [
           {
             'segment' =&gt; '1:1,1000',
             'type'    =&gt; 'exon',
           },
           {
             'segment' =&gt; '2:1,1000',
             'type'    =&gt; 'exon',
           },
         ]

         See DAS specifications for other parameters

   <b>types</b> <b>:</b> <b>Find</b> <b>out</b> <b>about</b> <b>different</b> <b>data</b> <b>types</b> <b>available</b> <b>from</b> <b>this</b> <b>source</b>
         my $types         = $das-&gt;types(); # takes optional args - see DAS specs

        Retrieve the types of data available for this source
        e.g. 32k_cloneset, karyotype, swissprot

   <b>features</b> <b>:</b> <b>Retrieve</b> <b>features</b> <b>from</b> <b>a</b> <b>segment</b>
          e.g. clones on a chromosome

         #########
         # Different ways to fetch features -
         #
         my $feature_data1 = $das-&gt;features('1:1,100000');
         my $feature_data2 = $das-&gt;features(['1:1,100000', '2:20435000,21435000']);
         my $feature_data3 = $das-&gt;features({
                                             'segment' =&gt; '1:1,1000',
                                             'type'    =&gt; 'karyotype',
                                             # optional args - see DAS Spec
                                            });
         my $feature_data4 = $das-&gt;features([
                                             {'segment'  =&gt; '1:1,1000000','type' =&gt; 'karyotype',},
                                             {'segment'  =&gt; '2:1,1000000',},
                                             {'group_id' =&gt; 'OTTHUMG00000036084',},
                                            ]);

         #########
         # Feature fetch with callback
         #
         my $callback = sub {
                             my $struct = shift;
                             print {*STDERR} Dumper($struct);
                            };
         # then:
         $das-&gt;callback($callback);
         $das-&gt;features('1:1,1000000');

         # or:
         $das-&gt;features('1:1,1000000', $callback);

         # or:
         $das-&gt;features(['1:1,1000000', '2:1,1000000', '3:1,1000000'], $callback);

         # or:
         $das-&gt;features([{'group_id' =&gt; 'OTTHUMG00000036084'}, '2:1,1000000', '3:1,1000000'], $callback);

   <b>alignment</b> <b>:</b> <b>Retrieve</b> <b>protein</b> <b>alignment</b> <b>data</b> <b>for</b> <b>a</b> <b>query.</b>  <b>This</b> <b>can</b> <b>be</b> <b>a</b> <b>multiple</b> <b>sequence</b> <b>alignment</b> <b>or</b>
       <b>pairwise</b> <b>alignment.</b>  <b>Note</b> <b>-</b> <b>this</b> <b>has</b> <b>not</b> <b>been</b> <b>tested</b> <b>for</b> <b>structural</b> <b>alignments</b> <b>as</b> <b>there</b> <b>is</b> <b>currently</b> <b>no</b>
       <b>Das</b> <b>source</b> <b>avialable.</b>
         my $alignment = $das-&gt;alignment({query =&gt; 'Q01234'});

   <b>structure</b> <b>:</b> <b>Retrieve</b> <b>known</b> <b>structure</b> <b>(i.e.</b> <b>PDB)</b> <b>for</b> <b>a</b> <b>query</b>
         my $structure = $das-&gt;structure({ query =&gt; 'pdb_id'});

   <b>sources</b> <b>:</b> <b>Retrieves</b> <b>the</b> <b>list</b> <b>of</b> <b>sources</b> <b>form</b> <b>the</b> <b>DAS</b> <b>registry,</b> <b>via</b> <b>a</b> <b>DAS</b> <b>call.</b>
         my $sources = $das-&gt;source;

   <b>sequence</b> <b>:</b> <b>Retrieve</b> <b>sequence</b> <b>data</b> <b>for</b> <b>a</b> <b>segment</b> <b>(probably</b> <b>dna</b> <b>or</b> <b>protein)</b>
         my $sequence      = $das-&gt;sequence('2:1,1000'); # segment:start,stop (e.g. chromosome 2, bases 1 to 1000)

   <b>stylesheet</b> <b>:</b> <b>Retrieve</b> <b>stylesheet</b> <b>data</b>
         my $style_data    = $das-&gt;stylesheet();
         my $style_data2   = $das-&gt;stylesheet($callback);

   <b>statuscodes</b> <b>:</b> <b>Retrieve</b> <b>HTTP</b> <b>status</b> <b>codes</b> <b>for</b> <b>request</b> <b>URLs</b>
         my $code         = $das-&gt;statuscodes($url);
         my $code_hashref = $das-&gt;statuscodes();

   <b>specversions</b> <b>:</b> <b>Retrieve</b> <b>a</b> <b>server's</b> <b>DAS</b> <b>specification</b> <b>version</b> <b>for</b> <b>a</b> <b>request</b> <b>URL</b>
         my $version         = $das-&gt;specversions($url);  # e.g. 1.53, 1.6, 1.6E
         my $version_hashref = $das-&gt;specversions();

   <b>max_hosts</b> <b>set</b> <b>number</b> <b>of</b> <b>running</b> <b>concurrent</b> <b>host</b> <b>connections</b>
         THIS METHOD IS NOW DEPRECATED AND HAS NO EFFECT

         $das-&gt;<a href="../man7/max_hosts.7.html">max_hosts</a>(7);
         print $das-&gt;max_hosts();

   <b>max_req</b> <b>set</b> <b>number</b> <b>of</b> <b>running</b> <b>concurrent</b> <b>requests</b> <b>per</b> <b>host</b>
         THIS METHOD IS NOW DEPRECATED AND HAS NO EFFECT

         $das-&gt;<a href="../man5/max_req.5.html">max_req</a>(5);
         print $das-&gt;max_req();

   <b>registry</b> <b>:</b> <b>Get/Set</b> <b>accessor</b> <b>for</b> <b>DAS-Registry</b> <b>service</b> <b>URLs</b>
         $biodaslite-&gt;registry('<a href="http://www.dasregistry.org/das">http://www.dasregistry.org/das</a>');

         my $registry_arrayref = $biodaslite-&gt;registry();

   <b>registry_sources</b> <b>:</b> <b>Arrayref</b> <b>of</b> <b>dassource</b> <b>objects</b> <b>from</b> <b>the</b> <b>configured</b> <b>registry</b> <b>services</b>
         my $sources_ref = $biodaslite-&gt;registry_sources();

         my $sources_ref = $biodaslite-&gt;registry_sources({
           'capability' =&gt; ['features','stylesheet'],
         });

         my $sources_ref = $biodaslite-&gt;registry_sources({
           'category' =&gt; ['Protein Sequence'],
         });

   <b>build_queries</b>
       Constructs an arrayref of DAS requests including parameters for each call

   <b>build_requests</b>
       Constructs the WWW::Curl callbacks

   <b>postprocess</b>
       Applies processing to the result set, e.g. removal of whitespace from sequence responses.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This module is an implementation of a client for the DAS protocol (XML over HTTP primarily for
       biological-data).

</pre><h4><b>DEPENDENCIES</b></h4><pre>
       strict
       warnings
       WWW::Curl
       HTTP::Response
       Carp
       English
       Readonly

</pre><h4><b>DIAGNOSTICS</b></h4><pre>
         Set $Bio::Das::Lite::DEBUG = 1;

</pre><h4><b>CONFIGURATION</b> <b>AND</b> <b>ENVIRONMENT</b></h4><pre>
</pre><h4><b>INCOMPATIBILITIES</b></h4><pre>
</pre><h4><b>BUGS</b> <b>AND</b> <b>LIMITATIONS</b></h4><pre>
         The max_req and max_hosts methods are now deprecated and have no effect.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       DAS Specifications at: <a href="http://biodas.org/documents/spec.html">http://biodas.org/documents/spec.html</a>

       ProServer (A DAS Server implementation also by the author) at:
          <a href="http://www.sanger.ac.uk/proserver/">http://www.sanger.ac.uk/proserver/</a>

       The venerable Bio::Das suite (CPAN and <a href="http://www.biodas.org/download/Bio">http://www.biodas.org/download/Bio</a>::Das/).

       The DAS Registry at:
          <a href="http://das.sanger.ac.uk/registry/">http://das.sanger.ac.uk/registry/</a>

</pre><h4><b>AUTHOR</b></h4><pre>
       Roger Pettett, &lt;<a href="mailto:rpettett@cpan.org">rpettett@cpan.org</a>&gt;

</pre><h4><b>LICENSE</b> <b>AND</b> <b>COPYRIGHT</b></h4><pre>
       Copyright (C) 2007 GRL, by Roger Pettett

       This  library  is  free  software;  you can redistribute it and/or modify it under the same terms as Perl
       itself, either Perl version 5.8.4 or, at your option, any later version of Perl 5 you may have available.

perl v5.40.0                                       2024-12-01                                <u>Bio::Das::<a href="../man3pm/Lite.3pm.html">Lite</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>