<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Run::Alignment::StandAloneFasta - Object for the local execution of the Fasta3 programs</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-perl-run-perl">libbio-perl-run-perl_1.7.3-8_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Run::Alignment::StandAloneFasta - Object for the local execution of the Fasta3 programs
       ((t)fasta3, (t)fastx3, (t)fasty3 ssearch3)

</pre><h4><b>SYNOPSIS</b></h4><pre>
         #!<a href="file:///usr/lib/w3m/cgi-bin/w3mman2html.cgi?perl">/usr/bin/perl</a>
         use Bio::Tools::Run::Alignment::StandAloneFasta;
         use Bio::SeqIO;
         use strict;
         my @arg=(
         'b' =&gt;'15',
         'O' =&gt;'resultfile',
         'H'=&gt;'',
         'program'=&gt;'fasta34'
         );

         my $factory=Bio::Tools::Run::Alignment::StandAloneFasta-&gt;new(@arg);
         $factory-&gt;<a href="../man1/ktup.1.html">ktup</a>(1);

         $factory-&gt;library('p');

         #print result file name
         print $factory-&gt;O;

         my @fastreport=$factory-&gt;run($ARGV[0]);

         foreach  (@fastreport) {
               print "Parsed fasta report:\n";
           my $result = $_-&gt;next_result;
           while( my $hit = $result-&gt;next_hit()) {
              print "\thit name: ", $hit-&gt;name(), "\n";
                while( my $hsp = $hit-&gt;next_hsp()) {
                print "E: ", $hsp-&gt;evalue(), "frac_identical: ",
               $hsp-&gt;frac_identical(), "\n";
                }
             }
           }

          #pass in seq objects
          my $sio = Bio::SeqIO-&gt;new(-file=&gt;$ARGV[0],-format=&gt;"fasta");
          my $seq = $sio-&gt;next_seq;
          my @fastreport=$factory-&gt;run($ARGV[0]);

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This wrapper works with version 3 of the FASTA program package (see W. R. Pearson and D. J. Lipman
       (1988), "Improved Tools for Biological Sequence Analysis", PNAS 85:2444-2448 (Pearson and Lipman, 1988);
       W. R.  Pearson (1996) "Effective protein sequence comparison" Meth. Enzymol.  266:227-258 (Pearson,
       1996); Pearson et. al. (1997) Genomics 46:24-36 (Zhang et al., 1997); Pearson, (1999) Meth. in Molecular
       Biology 132:185-219 (Pearson, 1999).  Version 3 of the FASTA packages contains many programs for
       searching DNA and protein databases and one program (prss3) for evaluating statistical significance from
       randomly shuffled sequences.

       Fasta is available at <a href="ftp://ftp.virginia.edu/pub/fasta">ftp://ftp.virginia.edu/pub/fasta</a>

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         <a href="http://redmine.open-bio.org/projects/bioperl/">http://redmine.open-bio.org/projects/bioperl/</a>

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Tiequan</b> <b>Zhang</b></h4><pre>
              Adapted for bioperl by Shawn Hoon
              Enhanced by Jason Stajich

       Email <a href="mailto:tqzhang1973@yahoo.com">tqzhang1973@yahoo.com</a>
             <a href="mailto:shawnh@fugu-sg.org">shawnh@fugu-sg.org</a>
             jason-at-bioperl.org

</pre><h4><b>Appendix</b></h4><pre>
       The rest of the documendation details each of the object methods. Internal methods are preceded with a
       underscore

   <b>program_name</b>
        Title   : program_name
        Usage   : $factory-&gt;program_name()
        Function: holds the program name
        Returns:  string
        Args    : None

   <b>executable</b>
        Title   : executable
        Usage   : my $exe = $blastfactory-&gt;executable('blastall');
        Function: Finds the full path to the 'codeml' executable
        Returns : string representing the full path to the exe
        Args    : [optional] name of executable to set path to
                  [optional] boolean flag whether or not warn when exe is not found

   <b>program_dir</b>
        Title   : program_dir
        Usage   : $factory-&gt;program_dir(@params)
        Function: returns the program directory, obtained from ENV variable.
        Returns:  string
        Args    :

   <b>run</b>
        Title   : run

        Usage   : my @fasta_object = $factory-&gt;($input,$onefile);
                  where $factory is the name of executable FASTA program;
                  $input is file name containing the sequences in the format
                  of fasta  or Bio::Seq object or array of Bio::Seq object;
                  $onefile is 0 if you want to save the outputs to different files
                  default: outputs are saved in one file

        Function: Attempts to run an executable FASTA program
                  and return array of  fasta objects containing the fasta report
        Returns : aray of fasta report object
                  If the user specify the output file(s),
                  the raw fasta report will be saved
        Args    : sequence object OR array reference of sequence objects
                  filename of file containing fasta formatted sequences

   <b>library</b>
        Title   : library
        Usage   : my $lb = $self-&gt;library
        Function: Fetch or set the name of the library to search against
        Returns : The name of the library
        Args    : No argument if user wants to fetch the name of library file;
                  A letter or a string of letter preceded by %;
                  (e.g. P or %pn, the letter is  the character in the third field
                  of any line of fastlibs file  ) or the name of library file
                  (if environmental variable FASTLIBS is not set);
                   if user wants to set the name of library file to search against

   <b>output</b>
        Title   : output
        Usage   : $obj-&gt;output($newval)
        Function: The output directory if we want to use this
        Example :
        Returns : value of output (a scalar)
        Args    : on set, new value (a scalar or undef, optional)

   <b>ktup</b>
        Title   :  ktup
        Usage   :  my $ktup = $self-&gt;ktup
        Function:  Fetch or set the ktup value for executable FASTA programs
        Example :
        Returns :  The value of ktup  if defined, else undef is returned
        Args    :  No argument if user want to fetch ktup value; A integer value between 1-6 if user want to set the
                  ktup value

   <b>_setinput</b>
        Title   :  _setinput
        Usage   :  Internal function, not to be called directly
        Function:   Create input file(s) for Blast executable
        Example :
        Returns : array of Bio::Seq object reference
        Args    : Seq object reference or input file name

   <b>_exist</b>
        Title   : _exist
        Usage   : Internal function, not to be called directly
        Function: Determine whether a executable FASTA program can be found
                  Cf. the DESCRIPTION section of this POD for how to make sure
                  for your FASTA installation to be found. This method checks for
                  existence of the blastall executable in the path.
        Returns : 1 if FASTA program found at expected location, 0 otherwise.
        Args    :  none

   <b>_setparams</b>
        Title   :  _setparams
        Usage   :  Internal function, not to be called directly
        Function:  Create parameter inputs for FASTA executable
        Returns : part of parameter string to be passed to FASTA program
        Args    : none

perl v5.34.0                                       2022-02-10              <u>Bio::Tools::<a href="../man3pm/Ru...StandAloneFasta.3pm.html">Ru...StandAloneFasta</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>