<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::DB::GFF::Adaptor::biofetch_oracle -- Cache BioFetch objects in a Bio::DB::GFF database</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-db-gff-perl">libbio-db-gff-perl_1.7.4-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::DB::GFF::Adaptor::biofetch_oracle -- Cache BioFetch objects in a Bio::DB::GFF database

</pre><h4><b>SYNOPSIS</b></h4><pre>
       Proof of principle.  Not for production use.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This adaptor is a proof-of-principle.  It is used to fetch BioFetch sequences into a Bio::DB::GFF
       database (currently uses a hard-coded EMBL database) as needed.  This allows the Generic Genome Browser
       to be used as a Genbank/EMBL browser.

</pre><h4><b>AUTHOR</b></h4><pre>
       Lincoln Stein &lt;<a href="mailto:lstein@cshl.org">lstein@cshl.org</a>&gt;.

       Copyright 2002 Cold Spring Harbor Laboratory.

       This library is free software; you can redistribute it and/or modify it under the same terms as Perl
       itself.

   <b>new</b>
        Title   : new
        Usage   : $db = Bio::DB::GFF-&gt;new(-adaptor=&gt;'biofetch_oracle', -preferred_tags =&gt; \%preferred, @args)
        Function: create a new adaptor
        Returns : a Bio::DB::GFF object
        Args    :   -adaptor : required.  Which adaptor to use; biofetch for mysql, biofetch_oracle for Oracle
                    -preferred_tags : optional.  A hash of {classname =&gt; weight,...}
                                      used to determine the class and name of the feature
                                      when a choice of possible feature classes is available
                                      (e.g. a feature has both a 'gene' and a 'locus' tag).
                                      Common defaults are provided that work well for eukaryotic
                                      features (but not well for viral/prokaryotic)
                     see below for additional arguments.
        Status  : Public

       This is the constructor for the adaptor.  It is called automatically by Bio::DB::GFF-&gt;new.  In addition
       to arguments that are common among all adaptors, the following class-specific arguments are recgonized:

         Argument       Description
         --------       -----------

         -dsn           the DBI data source, e.g. 'dbi:mysql:ens0040'

         -user          username for authentication

         -pass          the password for authentication

         -proxy         [['http','ftp'],'<a href="http://proxy">http://proxy</a>:8080']

         -create        initialize the database

       -dsn,-user and -pass indicate the local database to cache results in, and as are per
       Bio::DB::GFF::Adaptor::dbi.  The -proxy argument allows you to set the biofetch web proxy, and uses the
       same syntax described for the <b>proxy()</b> method of Bio::DB::WebDBSeqI, except that the argument must be
       passed as an array reference.

perl v5.30.0                                       2020-01-13             <u>Bio::DB::GFF::<a href="../man3pm/A...biofetch_oracle.3pm.html">A...biofetch_oracle</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>