<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::CodonUsage::Table - for access to the Codon usage Database at http://www.kazusa.or.jp/codon.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::CodonUsage::Table - for access to the Codon usage Database at <a href="http://www.kazusa.or.jp/codon">http://www.kazusa.or.jp/codon</a>.

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::CodonUsage::Table;
         use Bio::DB::CUTG;
         use Bio::CodonUsage::IO;
         use Bio::Tools::SeqStats;

         # Get  a codon usage table from web database
         my $cdtable = Bio::DB::CUTG-&gt;new(-sp =&gt; 'Mus musculus',
                                          -gc =&gt; 1);

         # Or from local file
         my $io      = Bio::CodonUsage::IO-&gt;new(-file =&gt; "file");
         my $cdtable = $io-&gt;next_data();

         # Or create your own from a Bio::PrimarySeq compliant object,
         # $codonstats is a ref to a hash of codon name /count key-value pairs
         my $codonstats = Bio::Tools::SeqStats-&gt;count_codons($Seq_objct);

         # '-data' must be specified, '-species' and 'genetic_code' are optional
         my $CUT = Bio::CodonUsage::Table-&gt;new(-data    =&gt; $codonstats,
                                               -species =&gt; 'Hsapiens_kinase');

         print "leu frequency is ", $cdtable-&gt;aa_frequency('LEU'), "\n";
         print "freq of ATG is ", $cdtable-&gt;codon_rel_frequency('ttc'), "\n";
         print "abs freq of ATG is ", $cdtable-&gt;codon_abs_frequency('ATG'), "\n";
         print "number of ATG codons is ", $cdtable-&gt;codon_count('ATG'), "\n";
         print "GC content at position 1 is ", $cdtable-&gt;get_coding_gc('1'), "\n";
         print "total CDSs for Mus musculus  is ", $cdtable-&gt;cds_count(), "\n";

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This class provides methods for accessing codon usage table data.

       All of the methods at present are simple look-ups of the table or are derived from simple calculations
       from the table. Future methods could include measuring the codon usage of a sequence , for example, or
       provide methods for examining codon usage in alignments.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Bio::Tools::CodonTable, Bio::WebAgent, Bio::CodonUsage::IO, Bio::DB::CUTG

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHORS</b></h4><pre>
       Richard Adams, <a href="mailto:Richard.Adams@ed.ac.uk">Richard.Adams@ed.ac.uk</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : my $cut = Bio::CodonUsage::Table-&gt;new(-data =&gt; $cut_hash_ref,
                                                        -species =&gt; 'H.sapiens_kinase'
                                                        -genetic_code =&gt;1);
        Returns : a reference to a new  Bio::CodonUsage::Table object
        Args    : none or a reference to a hash of codon counts. This constructor is
                  designed to be compatible with the output of
                  Bio::Tools::SeqUtils::count_codons()
                  Species and genetic code parameters can be entered here or via the
                  species() and genetic_code() methods separately.

   <b>all_aa_frequencies</b>
        Title   : all_aa_frequencies
        Usage   : my $freq = $cdtable-&gt;all_aa_frequencies();
        Returns : a reference to a hash where each key is an amino acid
                  and each value is its frequency in all proteins in that
                  species.
        Args    : none

   <b>codon_abs_frequency</b>
        Title   : codon_abs_frequency
        Usage   : my $freq = $cdtable-&gt;codon_abs_frequency('CTG');
        Purpose : To return the frequency of that codon as a percentage
                  of all codons in the organism.
        Returns : a percentage frequency
        Args    : a non-ambiguous codon string

   <b>codon_rel_frequency</b>
        Title   : codon_rel_frequency
        Usage   : my $freq = $cdtable-&gt;codon_rel_frequency('CTG');
        Purpose : To return the frequency of that codon as a percentage
                  of codons coding for the same amino acid. E.g., ATG and TGG
                  would return 100 as those codons are unique.
        Returns : a percentage frequency
        Args    : a non-ambiguous codon string

   <b>probable_codons</b>
        Title    : probable_codons
        Usage    : my $prob_codons = $cd_table-&gt;<a href="../man10/probable_codons.10.html">probable_codons</a>(10);
        Purpose  : to obtain a list of codons for the amino acid above a given
                   threshold % relative frequency
        Returns  : A reference to a hash where keys are 1 letter amino acid  codes
                   and values are references to arrays of codons whose frequency
                   is above the threshold.
        Arguments: a minimum threshold frequency

   <b>most_common_codons</b>
        Title    : most_common_codons
        Usage    : my $common_codons = $cd_table-&gt;most_common_codons();
        Purpose  : To obtain the most common codon for a given amino acid
        Returns  : A reference to a hash where keys are 1 letter amino acid codes
                   and the values are the single most common codons for those amino acids
        Arguments: None

   <b>codon_count</b>
        Title   : codon_count
        Usage   : my $count = $cdtable-&gt;codon_count('CTG');
        Purpose : To obtain the absolute number of the codons in the
                  organism.
        Returns : an integer
        Args    : a non-ambiguous codon string

   <b>get_coding_gc</b>
        Title   : get_coding_gc
        Usage   : my $count = $cdtable-&gt;<a href="../man1/get_coding_gc.1.html">get_coding_gc</a>(1);
        Purpose : To return the percentage GC composition for the organism at
                  codon positions 1,2 or 3, or an average for all coding sequence
                 ('all').
        Returns : a number (%-age GC content) or 0 if these fields are undefined
        Args    : 1,2,3 or 'all'.

   <b>set_coding_gc</b>
        Title   : set_coding_gc
        Usage   : my $count = $cdtable-&gt;set_coding_gc(-1=&gt;55.78);
        Purpose : To set the percentage GC composition for the organism at
                  codon positions 1,2 or 3, or an average for all coding sequence
                  ('all').
        Returns : void
        Args    : a hash where the key must be 1,2,3 or 'all' and the value the %age GC
                  at that codon position..

   <b>species</b>
        Title     : species
        Usage     : my $sp = $cut-&gt;species();
        Purpose   : Get/setter for species name of codon table
        Returns   : Void or species name string
        Args      : None or species name string

   <b>genetic_code</b>
        Title     : genetic_code
        Usage     : my $sp = $cut-&gt;genetic_code();
        Purpose   : Get/setter for genetic_code name of codon table
        Returns   : Void or genetic_code id, 1 by default
        Args      : None or genetic_code id, 1 by default if invalid argument.

   <b>cds_count</b>
        Title   : cds_count
        Usage   : my $count = $cdtable-&gt;cds_count();
        Purpose : To retrieve the total number of CDSs used to generate the Codon Table
                  for that organism.
        Returns : an integer
        Args    : none (if retrieving the value) or an integer( if setting ).

   <b>aa_frequency</b>
        Title   : aa_frequency
        Usage   : my $freq = $cdtable-&gt;aa_frequency('Leu');
        Purpose : To retrieve the frequency of an amino acid in the organism
        Returns : a percentage
        Args    : a 1 letter or 3 letter string representing the amino acid

   <b>common_codon</b>
        Title   : common_codon
        Usage   : my $freq = $cdtable-&gt;common_codon('Leu');
        Purpose : To retrieve the frequency of the most common codon of that aa
        Returns : a percentage
        Args    : a 1 letter or 3 letter string representing the amino acid

   <b>rare_codon</b>
        Title   : rare_codon
        Usage   : my $freq = $cdtable-&gt;rare_codon('Leu');
        Purpose : To retrieve the frequency of the least common codon of that aa
        Returns : a percentage
        Args    : a 1 letter or 3 letter string representing the amino acid

perl v5.32.1                                       2021-08-15                        <u>Bio::CodonUsage::<a href="../man3pm/Table.3pm.html">Table</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>