<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Glimmer - parser for Glimmer 2.X/3.X prokaryotic and GlimmerM/GlimmerHMM eukaryotic gene</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Glimmer - parser for Glimmer 2.X/3.X prokaryotic and GlimmerM/GlimmerHMM eukaryotic gene
       predictions

</pre><h4><b>SYNOPSIS</b></h4><pre>
          use Bio::Tools::Glimmer;

          # file
          my $parser = Bio::Tools::Glimmer-&gt;new(-file =&gt; $file);
          # filehandle:
          $parser = Bio::Tools::Glimmer-&gt;new( -fh  =&gt; \*INPUT );
          # provide a sequence identifier (Glimmer 2.X)
          my $parser = Bio::Tools::Glimmer-&gt;new(-file =&gt; $file, -seqname =&gt; seqname);
          # force format (override automatic detection)
          my $parser = Bio::Tools::Glimmer-&gt;new(-file =&gt; $file, -format =&gt; 'GlimmerM');

          # parse the results
          # note: this class is-a Bio::Tools::AnalysisResult which implements
          # Bio::SeqAnalysisParserI, i.e., $glimmer-&gt;next_feature() is the same

          while(my $gene = $parser-&gt;next_prediction()) {
              # For eukaryotic input (GlimmerM/GlimmerHMM), $gene will be an instance
              # of Bio::Tools::Prediction::Gene, which inherits off
              # Bio::SeqFeature::Gene::Transcript, and $gene-&gt;exons() will return an
              # array of Bio::Tools::Prediction::Exon objects.
              # For prokaryotic input (Glimmer2.X/Glimmer3.X), $gene will be an
              # instance of Bio::SeqFeature::Generic

              # all exons (eukaryotic only):
              @exon_arr = $gene-&gt;exons();
              # initial exons only
              @init_exons = $gene-&gt;exons('Initial');
              # internal exons only
              @intrl_exons = $gene-&gt;exons('Internal');
              # terminal exons only
              @term_exons = $gene-&gt;exons('Terminal');
          }

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This is a module for parsing Glimmer, GlimmerM and GlimmerHMM predictions.  It will create gene objects
       from the prediction report which can be attached to a sequence using Bioperl objects, or output as GFF
       suitable for loading into Bio::DB::GFF for use with Gbrowse.

       Glimmer is open source and available at &lt;<a href="http://www.cbcb.umd.edu/software/glimmer/">http://www.cbcb.umd.edu/software/glimmer/</a>&gt;.

       GlimmerM is open source and available at &lt;<a href="http://www.tigr.org/software/glimmerm/">http://www.tigr.org/software/glimmerm/</a>&gt;.

       GlimmerHMM is open source and available at &lt;<a href="http://www.cbcb.umd.edu/software/GlimmerHMM/">http://www.cbcb.umd.edu/software/GlimmerHMM/</a>&gt;.

       Note that Glimmer 2.X will only process the first sequence in a fasta file, and the prediction report
       does not contain any sort of sequence identifier

       Note that Glimmer 3.X produces two output files.  This module only parses the .predict file.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via email or the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Jason</b> <b>Stajich</b></h4><pre>
       Email jason-at-bioperl-dot-org

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
       Torsten Seemann

       Mark Johnson

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : my $obj = Bio::Tools::Glimmer-&gt;new();
        Function: Builds a new Bio::Tools::Glimmer object
        Returns : an instance of Bio::Tools::Glimmer
        Args    : format ('Glimmer', 'GlimmerM', 'GlimmerHMM'), seqname

   <b>analysis_method</b>
        Usage     : $glimmer-&gt;analysis_method();
        Purpose   : Inherited method. Overridden to ensure that the name matches
                    /glimmer/i.
        Returns   : String
        Argument  : n/a

   <b>next_feature</b>
        Title   : next_feature
        Usage   : while($gene = $glimmer-&gt;next_feature()) {
                         # do something
                  }
        Function: Returns the next gene structure prediction of the Glimmer result
                  file. Call this method repeatedly until FALSE is returned.

                  The returned object is actually a SeqFeatureI implementing object.
                  This method is required for classes implementing the
                  SeqAnalysisParserI interface, and is merely an alias for
                  next_prediction() at present.

        Example :
        Returns : A Bio::Tools::Prediction::Gene object.
        Args    :

   <b>next_prediction</b>
        Title   : next_prediction
        Usage   : while($gene = $glimmer-&gt;next_prediction()) {
                         # do something
                  }
        Function: Returns the next gene structure prediction of the Glimmer result
                  file. Call this method repeatedly until FALSE is returned.

        Example :
        Returns : A Bio::Tools::Prediction::Gene object.
        Args    :

   <b>_parse_predictions</b>
        Title   : _parse_predictions()
        Usage   : $obj-&gt;_parse_predictions()
        Function: Parses the prediction section. Automatically called by
                  next_prediction() if not yet done.
        Example :
        Returns :

   <b>_parse_eukaryotic</b>
        Title   : _parse_eukaryotic()
        Usage   : $obj-&gt;_parse_eukaryotic()
        Function: Parses the prediction section. Automatically called by
                  next_prediction() if not yet done.
        Example :
        Returns :

   <b>_parse_prokaryotic</b>
        Title   : _parse_prokaryotic()
        Usage   : $obj-&gt;_parse_prokaryotic()
        Function: Parses the prediction section. Automatically called by
                  next_prediction() if not yet done.
        Example :
        Returns :

   <b>_prediction</b>
        Title   : _prediction()
        Usage   : $gene = $obj-&gt;_prediction()
        Function: internal
        Example :
        Returns :

   <b>_add_prediction</b>
        Title   : _add_prediction()
        Usage   : $obj-&gt;_add_prediction($gene)
        Function: internal
        Example :
        Returns :

   <b>_predictions_parsed</b>
        Title   : _predictions_parsed
        Usage   : $obj-&gt;_predictions_parsed
        Function: internal
        Example :
        Returns : TRUE or FALSE

   <b>_seqname</b>
        Title   : _seqname
        Usage   : $obj-&gt;_seqname($seqname)
        Function: internal (for Glimmer 2.X)
        Example :
        Returns : String

   <b>_seqlength</b>
        Title   : _seqlength
        Usage   : $obj-&gt;_seqlength($seqlength)
        Function: internal (for Glimmer 2.X)
        Example :
        Returns : String

   <b>_format</b>
        Title   : _format
        Usage   : $obj-&gt;_format($format)
        Function: internal
        Example :
        Returns : String

   <b>_detail_file</b>
        Title   : _detail_file
        Usage   : $obj-&gt;_detail_file($filename)
        Function: internal (for Glimmer 3.X)
        Example :
        Returns : String

perl v5.32.1                                       2021-08-15                           <u>Bio::Tools::<a href="../man3pm/Glimmer.3pm.html">Glimmer</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>