<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::PrimarySeq - Bioperl lightweight sequence object</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::PrimarySeq - Bioperl lightweight sequence object

</pre><h4><b>SYNOPSIS</b></h4><pre>
         # Bio::SeqIO for file reading, Bio::DB::GenBank for
         # database reading

         use Bio::Seq;
         use Bio::SeqIO;
         use Bio::DB::GenBank;

         # make from memory

         $seqobj = Bio::PrimarySeq-&gt;new (
             -seq              =&gt; 'ATGGGGTGGGCGGTGGGTGGTTTG',
             -id               =&gt; 'GeneFragment-12',
             -accession_number =&gt; 'X78121',
             -alphabet         =&gt; 'dna',
             -is_circular      =&gt; 1,
         );
         print "Sequence ", $seqobj-&gt;id(), " with accession ",
           $seqobj-&gt;accession_number, "\n";

         # read from file

         $inputstream = Bio::SeqIO-&gt;new(
             -file   =&gt; "myseq.fa",
             -format =&gt; 'Fasta',
         );
         $seqobj = $inputstream-&gt;next_seq();
         print "Sequence ", $seqobj-&gt;id(), " and desc ", $seqobj-&gt;desc, "\n";

         # to get out parts of the sequence.

         print "Sequence ", $seqobj-&gt;id(), " with accession ",
           $seqobj-&gt;accession_number, " and desc ", $seqobj-&gt;desc, "\n";

         $string  = $seqobj-&gt;seq();
         $string2 = $seqobj-&gt;subseq(1,40);

</pre><h4><b>DESCRIPTION</b></h4><pre>
       PrimarySeq is a lightweight sequence object, storing the sequence, its name, a computer-useful unique
       name, and other fundamental attributes.  It does not contain sequence features or other information.  To
       have a sequence with sequence features you should use the Seq object which uses this object.

       Although new users will use Bio::PrimarySeq a lot, in general you will be using it from the Bio::Seq
       object. For more information on Bio::Seq see Bio::Seq. For interest you might like to know that Bio::Seq
       has-a Bio::PrimarySeq and forwards most of the function calls to do with sequence to it (the has-a
       relationship lets us get out of a otherwise nasty cyclical reference in Perl which would leak memory).

       Sequence objects are defined by the Bio::PrimarySeqI interface, and this object is a pure Perl
       implementation of the interface. If that's gibberish to you, don't worry. The take home message is that
       this object is the bioperl default sequence object, but other people can use their own objects as
       sequences if they so wish. If you are interested in wrapping your own objects as compliant Bioperl
       sequence objects, then you should read the Bio::PrimarySeqI documentation

       The documentation of this object is a merge of the Bio::PrimarySeq and Bio::PrimarySeqI documentation.
       This allows all the methods which you can call on sequence objects here.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Ewan</b> <b>Birney</b></h4><pre>
       Email <a href="mailto:birney@ebi.ac.uk">birney@ebi.ac.uk</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : $seqobj = Bio::PrimarySeq-&gt;new( -seq =&gt; 'ATGGGGGTGGTGGTACCCT',
                                                  -id  =&gt; 'human_id',
                                                  -accession_number =&gt; 'AL000012',
                                                  );
        Function: Returns a new primary seq object from
                  basic constructors, being a string for the sequence
                  and strings for id and accession_number.

                  Note that you can provide an empty sequence string. However, in
                  this case you MUST specify the type of sequence you wish to
                  initialize by the parameter -alphabet. See alphabet() for possible
                  values.
        Returns : a new Bio::PrimarySeq object
        Args    : -seq              =&gt; sequence string
                  -ref_to_seq       =&gt; ... or reference to a sequence string
                  -display_id       =&gt; display id of the sequence (locus name)
                  -accession_number =&gt; accession number
                  -primary_id       =&gt; primary id (Genbank id)
                  -version          =&gt; version number
                  -namespace        =&gt; the namespace for the accession
                  -authority        =&gt; the authority for the namespace
                  -description      =&gt; description text
                  -desc             =&gt; alias for description
                  -alphabet         =&gt; skip alphabet guess and set it to dna, rna or protein
                  -id               =&gt; alias for display id
                  -is_circular      =&gt; boolean to indicate that sequence is circular
                  -direct           =&gt; boolean to directly set sequences. The next time -seq,
                                       seq() or -ref_to_seq is use, the sequence will not be
                                       validated. Be careful with this...
                  -nowarnonempty    =&gt; boolean to avoid warning when sequence is empty

   <b>seq</b>
        Title   : seq
        Usage   : $string = $seqobj-&gt;seq();
        Function: Get or set  the sequence as a string of letters. The case of
                  the letters is left up to the implementer. Suggested cases are
                  upper case for proteins and lower case for DNA sequence (IUPAC
                  standard), but you should not rely on this. An error is thrown if
                  the sequence contains invalid characters: see validate_seq().
        Returns : A scalar
        Args    : - Optional new sequence value (a string) to set
                  - Optional alphabet (it is guessed by default)

   <b>validate_seq</b>
        Title   : validate_seq
        Usage   : if(! $seqobj-&gt;validate_seq($seq_str) ) {
                       print "sequence $seq_str is not valid for an object of
                       alphabet ",$seqobj-&gt;alphabet, "\n";
                  }
        Function: Test that the given sequence is valid, i.e. contains only valid
                  characters. The allowed characters are all letters (A-Z) and '-','.',
                  '*','?','=' and '~'. Spaces are not valid. Note that this
                  implementation does not take alphabet() into account and that empty
                  sequences are considered valid.
        Returns : 1 if the supplied sequence string is valid, 0 otherwise.
        Args    : - Sequence string to be validated
                  - Boolean to optionally throw an error if the sequence is invalid

   <b>subseq</b>
        Title   : subseq
        Usage   : $substring = $seqobj-&gt;subseq(10,40);
                  $substring = $seqobj-&gt;subseq(10,40,'nogap');
                  $substring = $seqobj-&gt;subseq(-start=&gt;10, -end=&gt;40, -replace_with=&gt;'tga');
                  $substring = $seqobj-&gt;subseq($location_obj);
                  $substring = $seqobj-&gt;subseq($location_obj, -nogap =&gt; 1);
        Function: Return the subseq from start to end, where the first sequence
                  character has coordinate 1 number is inclusive, ie 1-2 are the
                  first two characters of the sequence. The given start coordinate
                  has to be larger than the end, even if the sequence is circular.
        Returns : a string
        Args    : integer for start position
                  integer for end position
                        OR
                  Bio::LocationI location for subseq (strand honored)
                  Specify -NOGAP=&gt;1 to return subseq with gap characters removed
                  Specify -REPLACE_WITH=&gt;$new_subseq to replace the subseq returned
                  with $new_subseq in the sequence object

   <b>length</b>
        Title   : length
        Usage   : $len = $seqobj-&gt;length();
        Function: Get the stored length of the sequence in number of symbols (bases
                  or amino acids). In some circumstances, you can also set this attribute:

                  1. For empty sequences, you can set the length to anything you want:
                     my $seqobj = Bio::PrimarySeq-&gt;new( -length =&gt; 123 );
                     my $len = $seqobj-&gt;len; # 123
                  2. To save memory when using very long sequences, you can set the
                     length of the sequence to the length of the sequence (and nothing
                     else):
                     my $seqobj = Bio::PrimarySeq-&gt;new( -seq =&gt; 'ACGT...' ); # 1 Mbp sequence
                     # process $seqobj... then after you're done with it
                     $seqobj-&gt;length($seqobj-&gt;length);
                     $seqobj-&gt;seq(undef); # free memory!
                     my $len = $seqobj-&gt;len; # 1 Mbp

                  Note that if you set seq() to a value other than undef at any time,
                  the length attribute will be reset.
        Returns : integer representing the length of the sequence.
        Args    : Optionally, the value on set

   <b>display_id</b>
        Title   : display_id or display_name
        Usage   : $id_string = $seqobj-&gt;display_id();
        Function: Get or set the display id, aka the common name of the sequence object.

                  The semantics of this is that it is the most likely string to
                  be used as an identifier of the sequence, and likely to have
                  "human" readability.  The id is equivalent to the ID field of
                  the GenBank/EMBL databanks and the id field of the
                  Swissprot/sptrembl database. In fasta format, the &gt;(\S+) is
                  presumed to be the id, though some people overload the id to
                  embed other information. Bioperl does not use any embedded
                  information in the ID field, and people are encouraged to use
                  other mechanisms (accession field for example, or extending
                  the sequence object) to solve this.

                  With the new Bio::DescribeableI interface, display_name aliases
                  to this method.
        Returns : A string for the display ID
        Args    : Optional string for the display ID to set

   <b>accession_number</b>
        Title   : accession_number or object_id
        Usage   : $unique_key = $seqobj-&gt;accession_number;
        Function: Returns the unique biological id for a sequence, commonly
                  called the accession_number. For sequences from established
                  databases, the implementors should try to use the correct
                  accession number. Notice that primary_id() provides the
                  unique id for the implementation, allowing multiple objects
                  to have the same accession number in a particular implementation.

                  For sequences with no accession number, this method should
                  return "unknown".

                  [Note this method name is likely to change in 1.3]

                  With the new Bio::IdentifiableI interface, this is aliased
                  to object_id
        Returns : A string
        Args    : A string (optional) for setting

   <b>primary_id</b>
        Title   : primary_id
        Usage   : $unique_key = $seqobj-&gt;primary_id;
        Function: Returns the unique id for this object in this
                  implementation. This allows implementations to manage their
                  own object ids in a way the implementation can control
                  clients can expect one id to map to one object.

                  For sequences with no natural primary id, this method
                  should return a stringified memory location.
        Returns : A string
        Args    : A string (optional, for setting)

   <b>alphabet</b>
        Title   : alphabet
        Usage   : if( $seqobj-&gt;alphabet eq 'dna' ) { # Do something }
        Function: Get/set the alphabet of sequence, one of
                  'dna', 'rna' or 'protein'. This is case sensitive.

                  This is not called &lt;type&gt; because this would cause
                  upgrade problems from the 0.5 and earlier Seq objects.
        Returns : a string either 'dna','rna','protein'. NB - the object must
                  make a call of the type - if there is no alphabet specified it
                  has to guess.
        Args    : optional string to set : 'dna' | 'rna' | 'protein'

   <b>desc</b>
        Title   : desc or description
        Usage   : $seqobj-&gt;desc($newval);
        Function: Get/set description of the sequence.

                  'description' is an alias for this for compliance with the
                  Bio::DescribeableI interface.
        Returns : value of desc (a string)
        Args    : newvalue (a string or undef, optional)

   <b>can_call_new</b>
        Title   : can_call_new
        Usage   :
        Function:
        Example :
        Returns : true
        Args    :

   <b>id</b>
        Title   : id
        Usage   : $id = $seqobj-&gt;id();
        Function: This is mapped on display_id
        Example :
        Returns :
        Args    :

   <b>is_circular</b>
        Title   : is_circular
        Usage   : if( $seqobj-&gt;is_circular) { # Do something }
        Function: Returns true if the molecule is circular
        Returns : Boolean value
        Args    : none

</pre><h4><b>Methods</b> <b>for</b> <b>Bio::IdentifiableI</b> <b>compliance</b></h4><pre>
   <b>object_id</b>
        Title   : object_id
        Usage   : $string = $seqobj-&gt;object_id();
        Function: Get or set a string which represents the stable primary identifier
                  in this namespace of this object. For DNA sequences this
                  is its accession_number, similarly for protein sequences.

                  This is aliased to accession_number().
        Returns : A scalar
        Args    : Optional object ID to set.

   <b>version</b>
        Title   : version
        Usage   : $version = $seqobj-&gt;version();
        Function: Get or set a number which differentiates between versions of
                  the same object. Higher numbers are considered to be
                  later and more relevant, but a single object described
                  the same identifier should represent the same concept.
        Returns : A number
        Args    : Optional version to set.

   <b>authority</b>
        Title   : authority
        Usage   : $authority = $seqobj-&gt;authority();
        Function: Get or set a string which represents the organisation which
                  granted the namespace, written as the DNS name of the
                  organisation (eg, wormbase.org).
        Returns : A scalar
        Args    : Optional authority to set.

   <b>namespace</b>
        Title   : namespace
        Usage   : $string = $seqobj-&gt;namespace();
        Function: Get or set a string representing the name space this identifier
                  is valid in, often the database name or the name describing the
                  collection.
        Returns : A scalar
        Args    : Optional namespace to set.

</pre><h4><b>Methods</b> <b>for</b> <b>Bio::DescribableI</b> <b>compliance</b></h4><pre>
       This comprises of display_name and description.

   <b>display_name</b>
        Title   : display_name
        Usage   : $string = $seqobj-&gt;display_name();
        Function: Get or set a string which is what should be displayed to the user.
                  The string should have no spaces (ideally, though a cautious
                  user of this interface would not assume this) and should be
                  less than thirty characters (though again, double checking
                  this is a good idea).

                  This is aliased to display_id().
        Returns : A string for the display name
        Args    : Optional string for the display name to set.

   <b>description</b>
        Title   : description
        Usage   : $string = $seqobj-&gt;description();
        Function: Get or set a text string suitable for displaying to the user a
                  description. This string is likely to have spaces, but
                  should not have any newlines or formatting - just plain
                  text. The string should not be greater than 255 characters
                  and clients can feel justified at truncating strings at 255
                  characters for the purposes of display.

                  This is aliased to desc().
        Returns : A string for the description
        Args    : Optional string for the description to set.

</pre><h4><b>Methods</b> <b>Inherited</b> <b>from</b> <b>Bio::PrimarySeqI</b></h4><pre>
       These methods are available on Bio::PrimarySeq, although they are actually implemented on
       Bio::PrimarySeqI

   <b>revcom</b>
        Title   : revcom
        Usage   : $rev = $seqobj-&gt;revcom();
        Function: Produces a new Bio::SeqI implementing object which
                  is the reversed complement of the sequence. For protein
                  sequences this throws an exception of
                  "Sequence is a protein. Cannot revcom".

                  The id is the same id as the original sequence, and the
                  accession number is also identical. If someone wants to
                  track that this sequence has be reversed, it needs to
                  define its own extensions.

                  To do an inplace edit of an object you can go:

                  $seqobj = $seqobj-&gt;revcom();

                  This of course, causes Perl to handle the garbage
                  collection of the old object, but it is roughly speaking as
                  efficient as an inplace edit.
        Returns : A new (fresh) Bio::SeqI object
        Args    : none

   <b>trunc</b>
        Title   : trunc
        Usage   : $subseq = $myseq-&gt;trunc(10,100);
        Function: Provides a truncation of a sequence,
        Returns : A fresh Bio::SeqI implementing object.
        Args    : Numbers for the start and end positions

</pre><h4><b>Internal</b> <b>methods</b></h4><pre>
       These are internal methods to PrimarySeq

   <b>_guess_alphabet</b>
        Title   : _guess_alphabet
        Usage   :
        Function: Automatically guess and set the type of sequence: dna, rna, protein
                  or '' if the sequence was empty. This method first removes dots (.),
                  dashes (-) and question marks (?) before guessing the alphabet
                  using the IUPAC conventions for ambiguous residues. Since the DNA and
                  RNA characters are also valid characters for proteins, there is
                  no foolproof way of determining the right alphabet. This is our best
                  guess only!
        Returns : string 'dna', 'rna', 'protein' or ''.
        Args    : none

perl v5.32.1                                       2021-08-15                               <u>Bio::<a href="../man3pm/PrimarySeq.3pm.html">PrimarySeq</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>