<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Search::Hit::PsiBlastHit - Bioperl BLAST Hit object</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Search::Hit::PsiBlastHit - Bioperl BLAST Hit object

</pre><h4><b>SYNOPSIS</b></h4><pre>
       See Bio::Search::Result::BlastResult.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The Bio::Search::Hit::PsiBlastHit.pm module encapsulates data and methods for manipulating "hits" from a
       BLAST report. A BLAST hit is a collection of HSPs along with other metadata such as sequence name and
       score information. Hit objects are accessed via Bio::Search::Result::BlastResult objects after parsing a
       BLAST report using the Bio::SearchIO system.

       In Blast lingo, the "sbjct" sequences are all the sequences in a target database which were compared
       against a "query" sequence.  The terms "sbjct" and "hit" will be used interchangeably in this module.
       All methods that take 'sbjct' as an argument also support 'hit' as a synonym.

       This module supports BLAST versions 1.x and 2.x, gapped and ungapped, and PSI-BLAST.

       The construction of PsiBlastHit objects is performed by Bio::SearchIO::blast::PsiBlastHitFactory in a
       process that is orchestrated by the Blast parser (Bio::SearchIO::blast).  The resulting PsiBlastHits are
       then accessed via Bio::Search::Result::BlastResult). Therefore, you do not need to use
       Bio::Search::Hit::PsiBlastHit) directly. If you need to construct PsiBlastHits directly, see the "new()"
       function for details.

       For Bio::SearchIO BLAST parsing usage examples, see the "examples/search-blast" directory of the Bioperl
       distribution.

   <b>HSP</b> <b>Tiling</b> <b>and</b> <b>Ambiguous</b> <b>Alignments</b>
       If a Blast hit has more than one HSP, the Bio::Search::Hit::PsiBlastHit.pm object has the ability to
       merge overlapping HSPs into contiguous blocks. This permits the PsiBlastHit object to sum data across all
       HSPs without counting data in the overlapping regions multiple times, which would happen if data from
       each overlapping HSP are simply summed.  HSP tiling is performed automatically when methods of the
       PsiBlastHit object that rely on tiled data are invoked. These include <b>frac_identical()</b>, <b>frac_conserved()</b>,
       <b>gaps()</b>, <b>frac_aligned_query()</b>, <b>frac_aligned_hit()</b>, <b>num_unaligned_query()</b>, <b>num_unaligned_hit()</b>.

       It also permits the assessment of an "ambiguous alignment" if the query (or sbjct) sequences from
       different HSPs overlap (see <b>ambiguous_aln()</b>). The existence of an overlap could indicate a biologically
       interesting region in the sequence, such as a repeated domain.  The PsiBlastHit object uses the
       "-OVERLAP" parameter to determine when two sequences overlap; if this is set to 2 -- the default -- then
       any two sbjct or query HSP sequences must overlap by more than two residues to get merged into the same
       contig and counted as an overlap. See the BUGS  section below for "issues" with HSP tiling.

       The results of the HSP tiling is reported with the following ambiguity codes:

          'q' = Query sequence contains multiple sub-sequences matching
                a single region in the sbjct sequence.

          's' = Subject (PsiBlastHit) sequence contains multiple sub-sequences matching
                a single region in the query sequence.

          'qs' = Both query and sbjct sequences contain more than one
                 sub-sequence with similarity to the other sequence.

       For addition information about ambiguous BLAST alignments, see <b>Bio::Search::BlastUtils::tile_hsps()</b>

</pre><h4><b>DEPENDENCIES</b></h4><pre>
       Bio::Search::Hit::PsiBlastHit.pm is a concrete class that inherits from Bio::Root::Root and
       Bio::Search::Hit::HitI.  and relies on Bio::Search::HSP::BlastHSP.

</pre><h4><b>BUGS</b></h4><pre>
       One consequence of the HSP tiling is that methods that rely on HSP tiling such as <b>frac_identical()</b>,
       <b>frac_conserved()</b>, <b>gaps()</b> etc. may report misleading numbers when "-OVERLAP" is set to a large number.
       For example, say we have two HSPs and the query sequence tile as follows:

                   1      8             22      30        40             60
        Full seq:  ------------------------------------------------------------
                           *  ** *   **
        HSP1:             ---------------                    (6 identical matches)
                                     **   **  **
        HSP2:                        -------------           (6 identical matches)

       If "-OVERLAP" is set to some number over 4, HSP1 and HSP2 will not be tiled into a single contig and
       their numbers of identical matches will be added, giving a total of 12, not 10 if they had be combined
       into one contig. This can lead to number greater than 1.0 for methods <b>frac_identical()</b> and
       <b>frac_conserved()</b>. This is less of an issue with gapped Blast since it tends to combine HSPs that would be
       listed separately without gapping.  (Fractions &gt;1.0 can be viewed as a signal for an interesting
       alignment that warrants further inspection, thus turning this bug into a feature :-).

       Using large values for "-OVERLAP" can lead to incorrect numbers reported by methods that rely on HSP
       tiling but can be useful if you care more about detecting ambiguous alignments.  Setting "-OVERLAP" to
       zero will lead to the most accurate numbers for the tiling-dependent methods but will be useless for
       detecting overlapping HSPs since all HSPs will appear to overlap.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
        Bio::Search::HSP::BlastHSP.pm         - Blast HSP object.
        Bio::Search::Result::BlastResult.pm   - Blast Result object.
        Bio::Search::Hit::HitI.pm             - Interface implemented by PsiBlastHit.pm
        Bio::Root::Root.pm                    - Base class for PsiBlastHit.pm

       Links:

        <a href="http://bio.perl.org/">http://bio.perl.org/</a>                       - Bioperl Project Homepage

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules.  Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution. Bug
       reports can be submitted via the web:

           https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b></h4><pre>
       Steve Chervitz &lt;<a href="mailto:sac@bioperl.org">sac@bioperl.org</a>&gt;

       See the FEEDBACK section  for where to send bug reports and comments.

</pre><h4><b>ACKNOWLEDGEMENTS</b></h4><pre>
       This software was originally developed in the Department of Genetics at Stanford University. I would also
       like to acknowledge my colleagues at Affymetrix for useful feedback.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright (c) 1996-2001 Steve Chervitz. All Rights Reserved.

</pre><h4><b>DISCLAIMER</b></h4><pre>
       This software is provided "as is" without warranty of any kind.

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>new</b>
        Usage     : $hit = Bio::Search::Hit::PsiBlastHit-&gt;new( %named_params );
                  : Bio::Search::Hit::PsiBlastHit.pm objects are constructed
                  : automatically by Bio::SearchIO::PsiBlastHitFactory.pm,
                  : so there is no need for direct instantiation.
        Purpose   : Constructs a new PsiBlastHit object and Initializes key variables
                  : for the hit.
        Returns   : A Bio::Search::Hit::PsiBlastHit object
        Argument  : Named Parameters:
                  : Parameter keys are case-insensitive.
                  :     -RAW_DATA   =&gt; array reference holding raw BLAST report data
                  :                    for a single hit. This includes all lines
                  :                    within the HSP alignment listing section of a
                  :                    traditional BLAST or PSI-BLAST (non-XML) report,
                  :                    starting at (or just after) the leading '&gt;'.
                  :         -HOLD_RAW_DATA =&gt; boolean, should -RAW_DATA be saved within the object.
                  :         -QUERY_LEN  =&gt; Length of the query sequence
                  :         -ITERATION  =&gt; integer (PSI-BLAST iteration number in which hit was found)
                  :         -OVERLAP    =&gt; integer (maximum overlap between adjacent
                  :                    HSPs when tiling)
                  :         -PROGRAM    =&gt; string (type of Blast: BLASTP, BLASTN, etc)
                  :         -SIGNIF     =&gt; significance
                  :         -IS_PVAL    =&gt; boolean, true if -SIGNIF contains a P-value
                  :         -SCORE      =&gt; raw BLAST score
                  :         -FOUND_AGAIN   =&gt; boolean, true if this was a hit from the
                  :                       section of a PSI-BLAST with iteration &gt; 1
                  :                       containing sequences that were also found
                  :                       in iteration 1.
        Comments  : This object accepts raw Blast report data not because it
                  : is required for parsing, but in order to retrieve it
                  : (only available if -HOLD_RAW_DATA is set to true).

       See Also   : <b>Bio::Search::BlastUtils::tile_hsps()</b>, <b>Bio::Root::Root::new()</b>

   <b>algorithm</b>
        Title   : algorithm
        Usage   : $alg = $hit-&gt;algorithm();
        Function: Gets the algorithm specification that was used to obtain the hit
                  For BLAST, the algorithm denotes what type of sequence was aligned
                  against what (BLASTN: dna-dna, BLASTP prt-prt, BLASTX translated
                  dna-prt, TBLASTN prt-translated dna, TBLASTX translated
                  dna-translated dna).
        Returns : a scalar string
        Args    : none

   <b>name</b>
        Usage     : $hit-&gt;name([string]);
        Purpose   : Set/Get a string to identify the hit.
        Example   : $name = $hit-&gt;name;
                  : $hit-&gt;name('M81707');
        Returns   : String consisting of the hit's name or undef if not set.
        Comments  : The name is parsed out of the "Query=" line as the first chunk of
                    non-whitespace text. If you want the rest of the line, use
                    $hit-&gt;description().

       See Also: <b>accession()</b>

   <b>description</b>
        Usage     : $hit_object-&gt;description( [integer] );
        Purpose   : Set/Get a description string for the hit.
                    This is parsed out of the "Query=" line as everything after
                    the first chunk of non-whitespace text. Use $hit-&gt;name()
                    to get the first chunk (the ID of the sequence).
        Example   : $description = $hit-&gt;description;
                  : $desc_60char = $hit-&gt;<a href="../man60/description.60.html">description</a>(60);
        Argument  : Integer (optional) indicating the desired length of the
                  : description string to be returned.
        Returns   : String consisting of the hit's description or undef if not set.

   <b>accession</b>
        Title   : accession
        Usage   : $acc = $hit-&gt;accession();
        Function: Retrieve the accession (if available) for the hit
        Returns : a scalar string (empty string if not set)
        Args    : none
        Comments: Accession numbers are extracted based on the assumption that they
                  are delimited by | characters (NCBI-style). If this is not the case,
                  use the name() method and parse it as necessary.

       See Also: <b>name()</b>

   <b>raw_score</b>
        Usage     : $hit_object-&gt;raw_score();
        Purpose   : Gets the BLAST score of the best HSP for the current Blast hit.
        Example   : $score = $hit_object-&gt;raw_score();
        Returns   : Integer
        Argument  : n/a
        Throws    : n/a

       See Also   : <b>bits()</b>

   <b>length</b>
        Usage     : $hit_object-&gt;length();
        Purpose   : Get the total length of the hit sequence.
        Example   : $len = $hit_object-&gt;length();
        Returns   : Integer
        Argument  : n/a
        Throws    : n/a
        Comments  : Developer note: when using the built-in length function within
                  : this module, call it as CORE::length().

       See Also   : <b>logical_length()</b>,  <b>length_aln()</b>

   <b>significance</b>
       Equivalent to <b>signif()</b>

   <b>next_hsp</b>
        Title    : next_hsp
        Usage    : $hsp = $obj-&gt;next_hsp();
        Function : returns the next available High Scoring Pair object
        Example  :
        Returns  : Bio::Search::HSP::BlastHSP or undef if finished
        Args     : none

   <b>score</b>
       Equivalent to <b>raw_score()</b>

   <b>hit_length</b>
       Equivalent to <b>length()</b>

   <b>signif</b>
        Usage     : $hit_object-&gt;signif( [format] );
        Purpose   : Get the P or Expect value for the best HSP of the given BLAST hit.
                  : The value returned is the one which is reported in the description
                  : section of the Blast report. For Blast1 and WU-Blast2, this
                  : is a P-value, for Blast2, it is an Expect value.
        Example   : $obj-&gt;signif()        # returns 1.3e-34
                  : $obj-&gt;signif('exp')   # returns -34
                  : $obj-&gt;signif('parts') # returns (1.3, -34)
        Returns   : Float or scientific notation number (the raw P/Expect value, DEFAULT).
                  : Integer if format == 'exp' (the magnitude of the base 10 exponent).
                  : 2-element list (float, int) if format == 'parts' and P/Expect value
                  :                is in scientific notation (see Comments).
        Argument  : format: string of 'raw' | 'exp' | 'parts'
                  :    'raw' returns value given in report. Default. (1.2e-34)
                  :    'exp' returns exponent value only (34)
                  :    'parts' returns the decimal and exponent as a
                  :            2-element list (1.2, -34)  (see Comments).
        Throws    : n/a
        Comments  : The signif() method provides a way to deal with the fact that
                  : Blast1 and Blast2 formats (and WU- vs. NCBI-BLAST) differ in
                  : what is reported in the description lines of each hit in the
                  : Blast report. The signif() method frees any client code from
                  : having to know if this is a P-value or an Expect value,
                  : making it easier to write code that can process both
                  : Blast1 and Blast2 reports. This is not necessarily a good thing,
                  : since one should always know when one is working with P-values or
                  : Expect values (hence the deprecated status).
                  : Use of expect() is recommended since all hits will have an Expect value.
                  :
                  : Using the 'parts' argument is not recommended since it will not
                  : work as expected if the expect value is not in scientific notation.
                  : That is, floats are not converted into sci notation before
                  : splitting into parts.

       See Also   : p(), <b>expect()</b>, <b>Bio::Search::BlastUtils::get_exponent()</b>

   <b>to_string</b>
        Title   : to_string
        Usage   : print $hit-&gt;to_string;
        Function: Returns a string representation for the Blast Hit.
                  Primarily intended for debugging purposes.
        Example : see usage
        Returns : A string of the form:
                  [PsiBlastHit] &lt;name&gt; &lt;description&gt;
                  e.g.:
                  [PsiBlastHit] emb|Z46660|SC9725 S.cerevisiae chromosome XIII cosmid
        Args    : None

   <b>ambiguous_aln</b>
        Usage     : $ambig_code = $hit_object-&gt;ambiguous_aln();
        Purpose   : Sets/Gets ambiguity code data member.
        Example   : (see usage)
        Returns   : String = 'q', 's', 'qs', '-'
                  :   'q'  = query sequence contains overlapping sub-sequences
                  :          while sbjct does not.
                  :   's'  = sbjct sequence contains overlapping sub-sequences
                  :          while query does not.
                  :   'qs' = query and sbjct sequence contains overlapping sub-sequences
                  :          relative to each other.
                  :   '-'  = query and sbjct sequence do not contains multiple domains
                  :          relative to each other OR both contain the same distribution
                  :          of similar domains.
        Argument  : n/a
        Throws    : n/a
        Status    : Experimental

       See Also   : Bio::Search::BlastUtils::tile_hsps, "HSP Tiling and Ambiguous Alignments"

   <b>overlap</b>
        Usage     : $blast_object-&gt;overlap( [integer] );
        Purpose   : Gets/Sets the allowable amount overlap between different HSP sequences.
        Example   : $blast_object-&gt;<a href="../man5/overlap.5.html">overlap</a>(5);
                  : $overlap = $blast_object-&gt;overlap;
        Returns   : Integer.
        Argument  : integer.
        Throws    : n/a
        Status    : Experimental
        Comments  : Any two HSPs whose sequences overlap by less than or equal
                  : to the overlap() number of resides will be considered separate HSPs
                  : and will not get tiled by Bio::Search::BlastUtils::_adjust_contigs().

       See Also   : <b>Bio::Search::BlastUtils::_adjust_contigs()</b>, BUGS

   <b>bits</b>
        Usage     : $hit_object-&gt;bits();
        Purpose   : Gets the BLAST bit score of the best HSP for the current Blast hit.
        Example   : $bits = $hit_object-&gt;bits();
        Returns   : Integer
        Argument  : n/a
        Throws    : Exception if bit score is not set.
        Comments  : For BLAST1, the non-bit score is listed in the summary line.

       See Also   : <b>score()</b>

   <b>n</b>
        Usage     : $hit_object-&gt;n();
        Purpose   : Gets the N number for the current Blast hit.
                  : This is the number of HSPs in the set which was ascribed
                  : the lowest P-value (listed on the description line).
                  : This number is not the same as the total number of HSPs.
                  : To get the total number of HSPs, use num_hsps().
        Example   : $n = $hit_object-&gt;n();
        Returns   : Integer
        Argument  : n/a
        Throws    : Exception if HSPs have not been set (BLAST2 reports).
        Comments  : Note that the N parameter is not reported in gapped BLAST2.
                  : Calling n() on such reports will result in a call to num_hsps().
                  : The num_hsps() method will count the actual number of
                  : HSPs in the alignment listing, which may exceed N in
                  : some cases.

       See Also   : <b>num_hsps()</b>

   <b>frame</b>
        Usage     : $hit_object-&gt;frame();
        Purpose   : Gets the reading frame for the best HSP after HSP tiling.
                  : This is only valid for BLASTX and TBLASTN/X reports.
        Example   : $frame = $hit_object-&gt;frame();
        Returns   : Integer (-2 .. +2)
        Argument  : n/a
        Throws    : Exception if HSPs have not been set (BLAST2 reports).
        Comments  : This method requires that all HSPs be tiled. If they have not
                  : already been tiled, they will be tiled first automatically..
                  : If you don't want the tiled data, iterate through each HSP
                  : calling frame() on each (use hsps() to get all HSPs).

       See Also   : <b>hsps()</b>

   <b>p</b>
        Usage     : $hit_object-&gt;p( [format] );
        Purpose   : Get the P-value for the best HSP of the given BLAST hit.
                  : (Note that P-values are not provided with NCBI Blast2 reports).
        Example   : $p =  $sbjct-&gt;p;
                  : $p =  $sbjct-&gt;p('exp');  # get exponent only.
                  : ($num, $exp) =  $sbjct-&gt;p('parts');  # split sci notation into parts
        Returns   : Float or scientific notation number (the raw P-value, DEFAULT).
                  : Integer if format == 'exp' (the magnitude of the base 10 exponent).
                  : 2-element list (float, int) if format == 'parts' and P-value
                  :                is in scientific notation (See Comments).
        Argument  : format: string of 'raw' | 'exp' | 'parts'
                  :    'raw' returns value given in report. Default. (1.2e-34)
                  :    'exp' returns exponent value only (34)
                  :    'parts' returns the decimal and exponent as a
                  :            2-element list (1.2, -34) (See Comments).
        Throws    : Warns if no P-value is defined. Uses expect instead.
        Comments  : Using the 'parts' argument is not recommended since it will not
                  : work as expected if the P-value is not in scientific notation.
                  : That is, floats are not converted into sci notation before
                  : splitting into parts.

       See Also   : <b>expect()</b>, <b>signif()</b>, <b>Bio::Search::BlastUtils::get_exponent()</b>

   <b>expect</b>
        Usage     : $hit_object-&gt;expect( [format] );
        Purpose   : Get the Expect value for the best HSP of the given BLAST hit.
        Example   : $e =  $sbjct-&gt;expect;
                  : $e =  $sbjct-&gt;expect('exp');  # get exponent only.
                  : ($num, $exp) = $sbjct-&gt;expect('parts');  # split sci notation into parts
        Returns   : Float or scientific notation number (the raw expect value, DEFAULT).
                  : Integer if format == 'exp' (the magnitude of the base 10 exponent).
                  : 2-element list (float, int) if format == 'parts' and Expect
                  :                is in scientific notation (see Comments).
        Argument  : format: string of 'raw' | 'exp' | 'parts'
                  :    'raw' returns value given in report. Default. (1.2e-34)
                  :    'exp' returns exponent value only (34)
                  :    'parts' returns the decimal and exponent as a
                  :            2-element list (1.2, -34)  (see Comments).
        Throws    : Exception if the Expect value is not defined.
        Comments  : Using the 'parts' argument is not recommended since it will not
                  : work as expected if the expect value is not in scientific notation.
                  : That is, floats are not converted into sci notation before
                  : splitting into parts.

       See Also   : p(), <b>signif()</b>, <b>Bio::Search::BlastUtils::get_exponent()</b>

   <b>hsps</b>
        Usage     : $hit_object-&gt;hsps();
        Purpose   : Get a list containing all HSP objects.
                  : Get the numbers of HSPs for the current hit.
        Example   : @hsps = $hit_object-&gt;hsps();
                  : $num  = $hit_object-&gt;hsps();  # alternatively, use num_hsps()
        Returns   : Array context : list of Bio::Search::HSP::BlastHSP.pm objects.
                  : Scalar context: integer (number of HSPs).
                  :                 (Equivalent to num_hsps()).
        Argument  : n/a. Relies on wantarray
        Throws    : Exception if the HSPs have not been collected.

       See Also   : <b>hsp()</b>, <b>num_hsps()</b>

   <b>hsp</b>
        Usage     : $hit_object-&gt;hsp( [string] );
        Purpose   : Get a single BlastHSP.pm object for the present PsiBlastHit.pm object.
        Example   : $hspObj  = $hit_object-&gt;hsp;  # same as 'best'
                  : $hspObj  = $hit_object-&gt;hsp('best');
                  : $hspObj  = $hit_object-&gt;hsp('worst');
        Returns   : Object reference for a Bio::Search::HSP::BlastHSP.pm object.
        Argument  : String (or no argument).
                  :   No argument (default) = highest scoring HSP (same as 'best').
                  :   'best' or 'first' = highest scoring HSP.
                  :   'worst' or 'last' = lowest scoring HSP.
        Throws    : Exception if the HSPs have not been collected.
                  : Exception if an unrecognized argument is used.

       See Also   : <b>hsps()</b>, num_hsps()

   <b>num_hsps</b>
        Usage     : $hit_object-&gt;num_hsps();
        Purpose   : Get the number of HSPs for the present Blast hit.
        Example   : $nhsps = $hit_object-&gt;num_hsps();
        Returns   : Integer
        Argument  : n/a
        Throws    : Exception if the HSPs have not been collected.

       See Also   : <b>hsps()</b>

   <b>logical_length</b>
        Usage     : $hit_object-&gt;logical_length( [seq_type] );
                  : (mostly intended for internal use).
        Purpose   : Get the logical length of the hit sequence.
                  : For query sequence of BLASTX and TBLASTX reports and the hit
                  : sequence of TBLASTN and TBLASTX reports, the returned length
                  : is the length of the would-be amino acid sequence (length/3).
                  : For all other BLAST flavors, this function is the same as length().
        Example   : $len = $hit_object-&gt;logical_length();
        Returns   : Integer
        Argument  : seq_type = 'query' or 'hit' or 'sbjct' (default = 'query')
                    ('sbjct' is synonymous with 'hit')
        Throws    : n/a
        Comments  : This is important for functions like frac_aligned_query()
                  : which need to operate in amino acid coordinate space when dealing
                  : with T?BLASTX type reports.

       See Also   : <b>length()</b>, <b>frac_aligned_query()</b>, <b>frac_aligned_hit()</b>

   <b>length_aln</b>
        Usage     : $hit_object-&gt;length_aln( [seq_type] );
        Purpose   : Get the total length of the aligned region for query or sbjct seq.
                  : This number will include all HSPs
        Example   : $len    = $hit_object-&gt;length_aln(); # default = query
                  : $lenAln = $hit_object-&gt;length_aln('query');
        Returns   : Integer
        Argument  : seq_Type = 'query' or 'hit' or 'sbjct' (Default = 'query')
                    ('sbjct' is synonymous with 'hit')
        Throws    : Exception if the argument is not recognized.
        Comments  : This method will report the logical length of the alignment,
                  : meaning that for TBLAST[NX] reports, the length is reported
                  : using amino acid coordinate space (i.e., nucleotides / 3).
                  :
                  : This method requires that all HSPs be tiled. If they have not
                  : already been tiled, they will be tiled first automatically..
                  : If you don't want the tiled data, iterate through each HSP
                  : calling length() on each (use hsps() to get all HSPs).

       See Also   : <b>length()</b>, <b>frac_aligned_query()</b>, <b>frac_aligned_hit()</b>, <b>gaps()</b>,
       <b>Bio::Search::BlastUtils::tile_hsps()</b>, <b>Bio::Search::HSP::BlastHSP::length()</b>

   <b>gaps</b>
        Usage     : $hit_object-&gt;gaps( [seq_type] );
        Purpose   : Get the number of gaps in the aligned query, sbjct, or both sequences.
                  : Data is summed across all HSPs.
        Example   : $qgaps = $hit_object-&gt;gaps('query');
                  : $hgaps = $hit_object-&gt;gaps('hit');
                  : $tgaps = $hit_object-&gt;gaps();    # default = total (query + hit)
        Returns   : scalar context: integer
                  : array context without args: two-element list of integers
                  :    (queryGaps, sbjctGaps)
                  : Array context can be forced by providing an argument of 'list' or 'array'.
                  :
                  : CAUTION: Calling this method within printf or sprintf is arrray context.
                  : So this function may not give you what you expect. For example:
                  :          printf "Total gaps: %d", $hit-&gt;gaps();
                  : Actually returns a two-element array, so what gets printed
                  : is the number of gaps in the query, not the total
                  :
        Argument  : seq_type: 'query' | 'hit' or 'sbjct' | 'total' | 'list'  (default = 'total')
                    ('sbjct' is synonymous with 'hit')
        Throws    : n/a
        Comments  : If you need data for each HSP, use hsps() and then interate
                  : through each HSP object.
                  : This method requires that all HSPs be tiled. If they have not
                  : already been tiled, they will be tiled first automatically..
                  : Not relying on wantarray since that will fail in situations
                  : such as printf "%d", $hit-&gt;gaps() in which you might expect to
                  : be printing the total gaps, but evaluates to array context.

       See Also   : <b>length_aln()</b>

   <b>matches</b>
        Usage     : $hit_object-&gt;matches( [class] );
        Purpose   : Get the total number of identical or conserved matches
                  : (or both) across all HSPs.
                  : (Note: 'conservative' matches are indicated as 'positives'
                  :         in the Blast report.)
        Example   : ($id,$cons) = $hit_object-&gt;matches(); # no argument
                  : $id = $hit_object-&gt;matches('id');
                  : $cons = $hit_object-&gt;matches('cons');
        Returns   : Integer or a 2-element array of integers
        Argument  : class = 'id' | 'cons' OR none.
                  : If no argument is provided, both identical and conservative
                  : numbers are returned in a two element list.
                  : (Other terms can be used to refer to the conservative
                  :  matches, e.g., 'positive'. All that is checked is whether or
                  :  not the supplied string starts with 'id'. If not, the
                  : conservative matches are returned.)
        Throws    : Exception if the requested data cannot be obtained.
        Comments  : If you need data for each HSP, use hsps() and then interate
                  : through the HSP objects.
                  : Does not rely on wantarray to return a list. Only checks for
                  : the presence of an argument (no arg = return list).

       See Also   : <b>Bio::Search::HSP::BlastHSP::matches()</b>, <b>hsps()</b>

   <b>start</b>
        Usage     : $sbjct-&gt;start( [seq_type] );
        Purpose   : Gets the start coordinate for the query, sbjct, or both sequences
                  : in the PsiBlastHit object. If there is more than one HSP, the lowest start
                  : value of all HSPs is returned.
        Example   : $qbeg = $sbjct-&gt;start('query');
                  : $sbeg = $sbjct-&gt;start('hit');
                  : ($qbeg, $sbeg) = $sbjct-&gt;start();
        Returns   : scalar context: integer
                  : array context without args: list of two integers (queryStart, sbjctStart)
                  : Array context can be "induced" by providing an argument of 'list' or 'array'.
        Argument  : In scalar context: seq_type = 'query' or 'hit' or 'sbjct' (default = 'query')
                    ('sbjct' is synonymous with 'hit')
        Throws    : n/a
        Comments  : This method requires that all HSPs be tiled. If there is more than one
                  : HSP and they have not already been tiled, they will be tiled first automatically..
                  : Remember that the start and end coordinates of all HSPs are
                  : normalized so that start &lt; end. Strand information can be
                  : obtained by calling $hit-&gt;strand().

       See Also   : <b>end()</b>, <b>range()</b>, <b>strand()</b>, "HSP Tiling and Ambiguous Alignments",
       Bio::Search::HSP::BlastHSP::start

   <b>end</b>
        Usage     : $sbjct-&gt;end( [seq_type] );
        Purpose   : Gets the end coordinate for the query, sbjct, or both sequences
                  : in the PsiBlastHit object. If there is more than one HSP, the largest end
                  : value of all HSPs is returned.
        Example   : $qend = $sbjct-&gt;end('query');
                  : $send = $sbjct-&gt;end('hit');
                  : ($qend, $send) = $sbjct-&gt;end();
        Returns   : scalar context: integer
                  : array context without args: list of two integers (queryEnd, sbjctEnd)
                  : Array context can be "induced" by providing an argument of 'list' or 'array'.
        Argument  : In scalar context: seq_type = 'query' or 'sbjct'
                  :  (case insensitive). If not supplied, 'query' is used.
        Throws    : n/a
        Comments  : This method requires that all HSPs be tiled. If there is more than one
                  : HSP and they have not already been tiled, they will be tiled first automatically..
                  : Remember that the start and end coordinates of all HSPs are
                  : normalized so that start &lt; end. Strand information can be
                  : obtained by calling $hit-&gt;strand().

       See Also   : <b>start()</b>, <b>range()</b>, <b>strand()</b>, "HSP Tiling and Ambiguous Alignments",
       Bio::Search::HSP::BlastHSP::end

   <b>range</b>
        Usage     : $sbjct-&gt;range( [seq_type] );
        Purpose   : Gets the (start, end) coordinates for the query or sbjct sequence
                  : in the HSP alignment.
        Example   : ($qbeg, $qend) = $sbjct-&gt;range('query');
                  : ($sbeg, $send) = $sbjct-&gt;range('hit');
        Returns   : Two-element array of integers
        Argument  : seq_type = string, 'query' or 'hit' or 'sbjct'  (default = 'query')
                    ('sbjct' is synonymous with 'hit')
        Throws    : n/a

       See Also   : <b>start()</b>, <b>end()</b>

   <b>frac_identical</b>
        Usage     : $hit_object-&gt;frac_identical( [seq_type] );
        Purpose   : Get the overall fraction of identical positions across all HSPs.
                  : The number refers to only the aligned regions and does not
                  : account for unaligned regions in between the HSPs, if any.
        Example   : $frac_iden = $hit_object-&gt;frac_identical('query');
        Returns   : Float (2-decimal precision, e.g., 0.75).
        Argument  : seq_type: 'query' | 'hit' or 'sbjct' | 'total'
                  : default = 'query' (but see comments below).
                  : ('sbjct' is synonymous with 'hit')
        Throws    : n/a
        Comments  : Different versions of Blast report different values for the total
                  : length of the alignment. This is the number reported in the
                  : denominators in the stats section:
                  : "Identical = 34/120 Positives = 67/120".
                  : NCBI BLAST uses the total length of the alignment (with gaps)
                  : WU-BLAST uses the length of the query sequence (without gaps).
                  :
                  : Therefore, when called with an argument of 'total',
                  : this method will report different values depending on the
                  : version of BLAST used. Total does NOT take into account HSP
                  : tiling, so it should not be used.
                  :
                  : To get the fraction identical among only the aligned residues,
                  : ignoring the gaps, call this method without an argument or
                  : with an argument of 'query' or 'hit'.
                  :
                  : If you need data for each HSP, use hsps() and then iterate
                  : through the HSP objects.
                  : This method requires that all HSPs be tiled. If they have not
                  : already been tiled, they will be tiled first automatically.

       See Also   : <b>frac_conserved()</b>, <b>frac_aligned_query()</b>, <b>matches()</b>, <b>Bio::Search::BlastUtils::tile_hsps()</b>

   <b>frac_conserved</b>
        Usage     : $hit_object-&gt;frac_conserved( [seq_type] );
        Purpose   : Get the overall fraction of conserved positions across all HSPs.
                  : The number refers to only the aligned regions and does not
                  : account for unaligned regions in between the HSPs, if any.
        Example   : $frac_cons = $hit_object-&gt;frac_conserved('hit');
        Returns   : Float (2-decimal precision, e.g., 0.75).
        Argument  : seq_type: 'query' | 'hit' or 'sbjct' | 'total'
                  : default = 'query' (but see comments below).
                  : ('sbjct' is synonymous with 'hit')
        Throws    : n/a
        Comments  : Different versions of Blast report different values for the total
                  : length of the alignment. This is the number reported in the
                  : denominators in the stats section:
                  : "Positives = 34/120 Positives = 67/120".
                  : NCBI BLAST uses the total length of the alignment (with gaps)
                  : WU-BLAST uses the length of the query sequence (without gaps).
                  :
                  : Therefore, when called with an argument of 'total',
                  : this method will report different values depending on the
                  : version of BLAST used. Total does NOT take into account HSP
                  : tiling, so it should not be used.
                  :
                  : To get the fraction conserved among only the aligned residues,
                  : ignoring the gaps, call this method without an argument or
                  : with an argument of 'query' or 'hit'.
                  :
                  : If you need data for each HSP, use hsps() and then interate
                  : through the HSP objects.
                  : This method requires that all HSPs be tiled. If they have not
                  : already been tiled, they will be tiled first automatically.

       See Also   : <b>frac_identical()</b>, <b>matches()</b>, <b>Bio::Search::BlastUtils::tile_hsps()</b>

   <b>frac_aligned_query</b>
        Usage     : $hit_object-&gt;frac_aligned_query();
        Purpose   : Get the fraction of the query sequence which has been aligned
                  : across all HSPs (not including intervals between non-overlapping
                  : HSPs).
        Example   : $frac_alnq = $hit_object-&gt;frac_aligned_query();
        Returns   : Float (2-decimal precision, e.g., 0.75).
        Argument  : n/a
        Throws    : n/a
        Comments  : If you need data for each HSP, use hsps() and then interate
                  : through the HSP objects.
                  : To compute the fraction aligned, the logical length of the query
                  : sequence is used, meaning that for [T]BLASTX reports, the
                  : full length of the query sequence is converted into amino acids
                  : by dividing by 3. This is necessary because of the way
                  : the lengths of aligned sequences are computed.
                  : This method requires that all HSPs be tiled. If they have not
                  : already been tiled, they will be tiled first automatically.

       See Also   : <b>frac_aligned_hit()</b>, <b>logical_length()</b>, <b>length_aln()</b>,  <b>Bio::Search::BlastUtils::tile_hsps()</b>

   <b>frac_aligned_hit</b>
        Usage     : $hit_object-&gt;frac_aligned_hit();
        Purpose   : Get the fraction of the hit (sbjct) sequence which has been aligned
                  : across all HSPs (not including intervals between non-overlapping
                  : HSPs).
        Example   : $frac_alnq = $hit_object-&gt;frac_aligned_hit();
        Returns   : Float (2-decimal precision, e.g., 0.75).
        Argument  : n/a
        Throws    : n/a
        Comments  : If you need data for each HSP, use hsps() and then interate
                  : through the HSP objects.
                  : To compute the fraction aligned, the logical length of the sbjct
                  : sequence is used, meaning that for TBLAST[NX] reports, the
                  : full length of the sbjct sequence is converted into amino acids
                  : by dividing by 3. This is necessary because of the way
                  : the lengths of aligned sequences are computed.
                  : This method requires that all HSPs be tiled. If they have not
                  : already been tiled, they will be tiled first automatically.

       See Also   : <b>frac_aligned_query()</b>, <b>matches()</b>, , <b>logical_length()</b>, <b>length_aln()</b>,
       <b>Bio::Search::BlastUtils::tile_hsps()</b>

   <b>frac_aligned_sbjct</b>
       Same as <b>frac_aligned_hit()</b>

   <b>num_unaligned_sbjct</b>
       Same as <b>num_unaligned_hit()</b>

   <b>num_unaligned_hit</b>
        Usage     : $hit_object-&gt;num_unaligned_hit();
        Purpose   : Get the number of the unaligned residues in the hit sequence.
                  : Sums across all all HSPs.
        Example   : $num_unaln = $hit_object-&gt;num_unaligned_hit();
        Returns   : Integer
        Argument  : n/a
        Throws    : n/a
        Comments  : See notes regarding logical lengths in the comments for frac_aligned_hit().
                  : They apply here as well.
                  : If you need data for each HSP, use hsps() and then interate
                  : through the HSP objects.
                  : This method requires that all HSPs be tiled. If they have not
                  : already been tiled, they will be tiled first automatically..

       See Also   : <b>num_unaligned_query()</b>,  <b>Bio::Search::BlastUtils::tile_hsps()</b>, <b>frac_aligned_hit()</b>

   <b>num_unaligned_query</b>
        Usage     : $hit_object-&gt;num_unaligned_query();
        Purpose   : Get the number of the unaligned residues in the query sequence.
                  : Sums across all all HSPs.
        Example   : $num_unaln = $hit_object-&gt;num_unaligned_query();
        Returns   : Integer
        Argument  : n/a
        Throws    : n/a
        Comments  : See notes regarding logical lengths in the comments for frac_aligned_query().
                  : They apply here as well.
                  : If you need data for each HSP, use hsps() and then interate
                  : through the HSP objects.
                  : This method requires that all HSPs be tiled. If they have not
                  : already been tiled, they will be tiled first automatically..

       See Also   : <b>num_unaligned_hit()</b>, <b>frac_aligned_query()</b>,  <b>Bio::Search::BlastUtils::tile_hsps()</b>

   <b>seq_inds</b>
        Usage     : $hit-&gt;seq_inds( seq_type, class, collapse );
        Purpose   : Get a list of residue positions (indices) across all HSPs
                  : for identical or conserved residues in the query or sbjct sequence.
        Example   : @s_ind = $hit-&gt;seq_inds('query', 'identical');
                  : @h_ind = $hit-&gt;seq_inds('hit', 'conserved');
                  : @h_ind = $hit-&gt;seq_inds('hit', 'conserved', 1);
        Returns   : Array of integers
                  : May include ranges if collapse is non-zero.
        Argument  : [0] seq_type  = 'query' or 'hit' or 'sbjct'  (default = 'query')
                  :                 ('sbjct' is synonymous with 'hit')
                  : [1] class     = 'identical' or 'conserved' (default = 'identical')
                  :              (can be shortened to 'id' or 'cons')
                  :              (actually, anything not 'id' will evaluate to 'conserved').
                  : [2] collapse  = boolean, if non-zero, consecutive positions are merged
                  :             using a range notation, e.g., "1 2 3 4 5 7 9 10 11"
                  :             collapses to "1-5 7 9-11". This is useful for
                  :             consolidating long lists. Default = no collapse.
        Throws    : n/a.
        Comments  : Note that HSPs are not tiled for this. This could be a problem
                  : for hits containing mutually exclusive HSPs.
                  : TODO: Consider tiling and then reporting seq_inds for the
                  : best HSP contig.

       See Also   : <b>Bio::Search::HSP::BlastHSP::seq_inds()</b>

   <b>iteration</b>
        Usage     : $sbjct-&gt;iteration( );
        Purpose   : Gets the iteration number in which the Hit was found.
        Example   : $iteration_num = $sbjct-&gt;iteration();
        Returns   : Integer greater than or equal to 1
                    Non-PSI-BLAST reports will report iteration as 1, but this number
                    is only meaningful for PSI-BLAST reports.
        Argument  : none
        Throws    : none

       See Also   : <b>found_again()</b>

   <b>found_again</b>
        Usage     : $sbjct-&gt;found_again;
        Purpose   : Gets a boolean indicator whether or not the hit has
                    been found in a previous iteration.
                    This is only applicable to PSI-BLAST reports.

                     This method indicates if the hit was reported in the
                     "Sequences used in model and found again" section of the
                     PSI-BLAST report or if it was reported in the
                     "Sequences not found previously or not previously below threshold"
                     section of the PSI-BLAST report. Only for hits in iteration &gt; 1.

        Example   : if( $sbjct-&gt;found_again()) { ... };
        Returns   : Boolean (1 or 0) for PSI-BLAST report iterations greater than 1.
                    Returns undef for PSI-BLAST report iteration 1 and non PSI_BLAST
                    reports.
        Argument  : none
        Throws    : none

       See Also   : <b>found_again()</b>

   <b>strand</b>
        Usage     : $sbjct-&gt;strand( [seq_type] );
        Purpose   : Gets the strand(s) for the query, sbjct, or both sequences
                  : in the best HSP of the PsiBlastHit object after HSP tiling.
                  : Only valid for BLASTN, TBLASTX, BLASTX-query, TBLASTN-hit.
        Example   : $qstrand = $sbjct-&gt;strand('query');
                  : $sstrand = $sbjct-&gt;strand('hit');
                  : ($qstrand, $sstrand) = $sbjct-&gt;strand();
        Returns   : scalar context: integer '1', '-1', or '0'
                  : array context without args: list of two strings (queryStrand, sbjctStrand)
                  : Array context can be "induced" by providing an argument of 'list' or 'array'.
        Argument  : In scalar context: seq_type = 'query' or 'hit' or 'sbjct' (default = 'query')
                    ('sbjct' is synonymous with 'hit')
        Throws    : n/a
        Comments  : This method requires that all HSPs be tiled. If they have not
                  : already been tiled, they will be tiled first automatically..
                  : If you don't want the tiled data, iterate through each HSP
                  : calling strand() on each (use hsps() to get all HSPs).
                  :
                  : Formerly (prior to 10/21/02), this method would return the
                  : string "-1/1" for hits with HSPs on both strands.
                  : However, now that strand and frame is properly being accounted
                  : for during HSP tiling, it makes more sense for strand()
                  : to return the strand data for the best HSP after tiling.
                  :
                  : If you really want to know about hits on opposite strands,
                  : you should be iterating through the HSPs using methods on the
                  : HSP objects.
                  :
                  : A possible use case where knowing whether a hit has HSPs
                  : on both strands would be when filtering via SearchIO for hits with
                  : this property. However, in this case it would be better to have a
                  : dedicated method such as $hit-&gt;hsps_on_both_strands(). Similarly
                  : for frame. This could be provided if there is interest.

       See Also   : Bio::Search::HSP::BlastHSP::strand()

</pre><h4><b>FOR</b> <b>DEVELOPERS</b> <b>ONLY</b></h4><pre>
   <b>Data</b> <b>Members</b>
       Information about the various data members of this module is provided for those wishing to modify or
       understand the code. Two things to bear in mind:

       1 Do NOT rely on these in any code outside of this module.
           All  data  members  are  prefixed  with  an  underscore to signify that they are private.  Always use
           accessor methods. If the accessor doesn't exist or is inadequate, create or modify an  accessor  (and
           let  me  know, too!). (An exception to this might be for BlastHSP.pm which is more tightly coupled to
           PsiBlastHit.pm and may access PsiBlastHit data members directly for efficiency purposes, but probably
           should not).

       2 This documentation may be incomplete and out of date.
           It is easy for these data member descriptions to become obsolete as this module  is  still  evolving.
           Always double check this info and search for members not described here.

       An  instance  of Bio::Search::Hit::PsiBlastHit.pm is a blessed reference to a hash containing all or some
       of the following fields:

        FIELD           VALUE
        --------------------------------------------------------------
        _hsps          : Array ref for a list of Bio::Search::HSP::BlastHSP.pm objects.
                       :
        _db            : Database identifier from the summary line.
                       :
        _desc          : Description data for the hit from the summary line.
                       :
        _length        : Total length of the hit sequence.
                       :
        _score         : BLAST score.
                       :
        _bits          : BLAST score (in bits). Matrix-independent.
                       :
        _p             : BLAST P value. Obtained from summary section. (Blast1/WU-Blast only)
                       :
        _expect        : BLAST Expect value. Obtained from summary section.
                       :
        _n             : BLAST N value (number of HSPs) (Blast1/WU-Blast2 only)
                       :
        _frame         : Reading frame for TBLASTN and TBLASTX analyses.
                       :
        _totalIdentical: Total number of identical aligned monomers.
                       :
        _totalConserved: Total number of conserved aligned monomers (a.k.a. "positives").
                       :
        _overlap       : Maximum number of overlapping residues between adjacent HSPs
                       : before considering the alignment to be ambiguous.
                       :
        _ambiguous_aln : Boolean. True if the alignment of all HSPs is ambiguous.
                       :
        _length_aln_query : Length of the aligned region of the query sequence.
                          :
        _length_aln_sbjct : Length of the aligned region of the sbjct sequence.

perl v5.32.1                                       2021-08-15                 <u>Bio::Search::Hit::<a href="../man3pm/PsiBlastHit.3pm.html">PsiBlastHit</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>