<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>bio-rainbow - clustering and assembling short reads for bioinformatics</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/bio-rainbow">bio-rainbow_2.0.4+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       bio-rainbow - clustering and assembling short reads for bioinformatics

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>rainbow</b> <u>&lt;cmd&gt;</u> [<u>options</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       rainbow 2.0.4 <b>--</b> &lt;<a href="mailto:ruanjue@gmail.com">ruanjue@gmail.com</a>, <a href="mailto:chongzechen@gmail.com">chongzechen@gmail.com</a>&gt;

              cluster

       Input       File      Format:      paired      fasta/fastq      file(s)      Output      File     Format:
       &lt;seqid:int&gt;\t&lt;cluster_id:int&gt;\t&lt;read1:string&gt;\t&lt;read2:string&gt;

       <b>-1</b> &lt;string&gt; Input fasta/fastq file, supports multiple '-1'

       <b>-2</b> &lt;string&gt; Input fasta/fastq file, supports multiple '-2' [null]

       <b>-l</b> &lt;int&gt;
              Read length, default: 0 variable

       <b>-m</b> &lt;int&gt;
              Maximum mismatches [4]

       <b>-e</b> &lt;int&gt;
              Exactly matching threshold [2000]

       <b>-L</b>     Low level of polymorphism

              div

       Input File  Format:  &lt;seqid:int&gt;\t&lt;cluster_id:int&gt;\t&lt;read1:string&gt;\t&lt;read2:string&gt;  Output  File  Format:
       &lt;seqid:int&gt;\t&lt;cluster_id:int&gt;\t&lt;read1:string&gt;\t&lt;read2:string&gt;[\t&lt;pre_cluster_id:int&gt;]

       <b>-i</b> &lt;string&gt; Input file [stdin]

       <b>-o</b> &lt;string&gt; Output file [stdout]

       <b>-k</b> &lt;int&gt;
              K_allele, min variants to create a new group [2]

       <b>-K</b> &lt;int&gt;
              K_allele, divide regardless of frequency when num of variants exceed this value [50]

       <b>-f</b> &lt;float&gt;
              Frequency, min variant frequency to create a new group [0.2]

              merge

       Input File Format: &lt;seqid:int&gt;\t&lt;cluster_id:int&gt;\t&lt;read1:string&gt;\t&lt;read2:string&gt;[\t&lt;pre_cluster_id:int&gt;]

       <b>-i</b> &lt;string&gt; Input rbasm output file [stdin]

       <b>-a</b>     output assembly

       <b>-o</b> &lt;string&gt; Output file for merged contigs, one line per cluster [stdout]

       <b>-N</b> &lt;int&gt;
              Maximum number of divided clusters to merge [300]

       <b>-l</b> &lt;int&gt;
              Minimum overlap when assemble two reads (valid only when '-a' is opened) [5]

       <b>-f</b> &lt;float&gt;
              Minimum fraction of similarity when assembly (valid only when '-a' is opened) [0.90]

       <b>-r</b> &lt;int&gt;
              Minimum number of reads to assemble (valid only when '-a' is opened) [5]

       <b>-R</b> &lt;int&gt;
              Maximum number of reads to assemble (valid only when '-a' is opened) [300]

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

bio-rainbow 2.0.4                                   July 2018                                     <u><a href="../man1/BIO-RAINBOW.1.html">BIO-RAINBOW</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>