<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Species - Generic species object.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Species - Generic species object.

</pre><h4><b>SYNOPSIS</b></h4><pre>
           $species = Bio::Species-&gt;new(-classification =&gt; [@classification]);
                                           # Can also pass classification
                                           # array to new as below

           $species-&gt;classification(qw( sapiens Homo Hominidae
                                        Catarrhini Primates Eutheria
                                        Mammalia Vertebrata Chordata
                                        Metazoa Eukaryota ));

           $genus = $species-&gt;genus();

           $bi = $species-&gt;binomial();     # $bi is now "Homo sapiens"

           # For storing common name
           $species-&gt;common_name("human");

           # For storing subspecies
           $species-&gt;sub_species("accountant");

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>NOTE:</b> <b>This</b> <b>class</b> <b>is</b> <b>planned</b> <b>for</b> <b>deprecation</b> <b>in</b> <b>favor</b> <b>of</b> <b>the</b> <b>simpler</b> <b>Bio::Taxon.</b>  <b>Please</b> <b>use</b> <b>that</b> <b>class</b>
       <b>instead.</b>

       Provides a very simple object for storing phylogenetic information. The classification is stored in an
       array, which is a list of nodes in a phylogenetic tree. Access to getting and setting species and genus
       is provided, but not to any of the other node types (eg: "phylum", "class", "order", "family"). There's
       plenty of scope for making the model more sophisticated, if this is ever needed.

       A methods are also provided for storing common names, and subspecies.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b></h4><pre>
       James Gilbert email <b><a href="mailto:jgrg@sanger.ac.uk">jgrg@sanger.ac.uk</a></b>

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
       Sendu Bala, <a href="mailto:bix@sendu.me.uk">bix@sendu.me.uk</a> Chris Fields, cjfields at bioperl dot org

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : my $obj = Bio::Species-&gt;new(-classification =&gt; \@class)
        Function: Build a new Species object
        Returns : Bio::Species object
        Args    : -ncbi_taxid     =&gt; NCBI taxonomic ID (optional)
                  -classification =&gt; arrayref of classification

   <b>classification</b>
        Title   : classification
        Usage   : $self-&gt;classification(@class_array);
                  @classification = $self-&gt;classification();
        Function: Get/set the lineage of this species. The array provided must be in
                  the order ... ---&gt; SPECIES, GENUS ---&gt; KINGDOM ---&gt; etc.
        Example : $obj-&gt;classification(qw( 'Homo sapiens' Homo Hominidae
                  Catarrhini Primates Eutheria Mammalia Vertebrata
                  Chordata Metazoa Eukaryota));
        Returns : Classification array
        Args    : Classification array
                        OR
                  A reference to the classification array. In the latter case
                  if there is a second argument and it evaluates to true,
                  names will not be validated. NB: in any case, names are never
                  validated anyway.

   <b>ncbi_taxid</b>
        Title   : ncbi_taxid
        Usage   : $obj-&gt;ncbi_taxid($newval)
        Function: Get/set the NCBI Taxon ID
        Returns : the NCBI Taxon ID as a string
        Args    : newvalue to set or undef to unset (optional)

   <b>common_name</b>
        Title   : common_name
        Usage   : $self-&gt;common_name( $common_name );
                  $common_name = $self-&gt;common_name();
        Function: Get or set the common name of the species
        Example : $self-&gt;common_name('human')
        Returns : The common name in a string
        Args    : String, which is the common name (optional)

   <b>division</b>
        Title   : division
        Usage   : $obj-&gt;division($newval)
        Function: Genbank Division for a species
        Returns : value of division (a scalar)
        Args    : value of division (a scalar)

   <b>species</b>
        Title   : species
        Usage   : $self-&gt;species( $species );
                  $species = $self-&gt;species();
        Function: Get or set the species name.
                  Note that this is  NOT genus and species
                  -- use $self-&gt;binomial() for that.
        Example : $self-&gt;species('sapiens');
        Returns : species name as string (NOT genus and species)
        Args    : species name as string (NOT genus and species)

   <b>genus</b>
        Title   : genus
        Usage   : $self-&gt;genus( $genus );
                  $genus = $self-&gt;genus();
        Function: Get or set the scientific genus name.
        Example : $self-&gt;genus('Homo');
        Returns : Scientific genus name as string
        Args    : Scientific genus name as string

   <b>sub_species</b>
        Title   : sub_species
        Usage   : $obj-&gt;sub_species($newval)
        Function: Get or set the scientific subspecies name.
        Returns : value of sub_species
        Args    : newvalue (optional)

   <b>variant</b>
        Title   : variant
        Usage   : $obj-&gt;variant($newval)
        Function: Get/set variant information for this species object (strain,
                  isolate, etc).
        Example :
        Returns : value of variant (a scalar)
        Args    : new value (a scalar or undef, optional)

   <b>binomial</b>
        Title   : binomial
        Usage   : $binomial = $self-&gt;binomial();
                  $binomial = $self-&gt;binomial('FULL');
        Function: Returns a string "Genus species", or "Genus species subspecies",
                  if the first argument is 'FULL' (and the species has a subspecies).
        Args    : Optionally the string 'FULL' to get the full name including
                  the subspecies.
        Note    : This is just munged from the taxon() name

   <b>validate_species_name</b>
        Title   : validate_species_name
        Usage   : $result = $self-&gt;validate_species_name($string);
        Function: Validate the species portion of the binomial
        Args    : string
        Notes   : The string following the "genus name" in the NCBI binomial is so
                  variable that it's not clear that this is a useful function. Consider
                  the binomials "Simian 11 rotavirus (serotype 3 / strain
                  SA11-Patton)", or "St. Thomas 3 rotavirus", straight from GenBank.
                  This is particularly problematic in microbes and viruses. As such,
                  this isn't actually used automatically by any Bio::Species method.

   <b>organelle</b>
        Title   : organelle
        Usage   : $self-&gt;organelle( $organelle );
                  $organelle = $self-&gt;organelle();
        Function: Get or set the organelle name
        Example : $self-&gt;organelle('Chloroplast')
        Returns : The organelle name in a string
        Args    : String, which is the organelle name
        Note    : TODO: We currently do not know where the organelle definition will
                  eventually go.  This is stored in the source seqfeature, though,
                  so the information isn't lost.

   <b>Delegation</b>
       The following methods delegate to the internal Bio::Taxon instance. This is mainly to allow code continue
       using older methods, with the mind to migrate to using Bio::Taxon and related methods when this class is
       deprecated.

   <b>taxon</b>
        Title    : taxon
        Usage    : $obj-&gt;taxon
        Function : retrieve the internal Bio::Taxon instance
        Returns  : A Bio::Taxon. If one is not previously set,
                   an instance is created lazily
        Args     : Bio::Taxon (optional)

   <b>tree</b>
        Title    : tree
        Usage    : $obj-&gt;tree
        Function : Returns a Bio::Tree::Tree object
        Returns  : A Bio::Tree::Tree. If one is not previously set,
                   an instance is created lazily
        Args     : Bio::Tree::Tree (optional)

perl v5.32.1                                       2021-08-15                                  <u>Bio::<a href="../man3pm/Species.3pm.html">Species</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>