<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Matrix::PSM::IO::masta - motif fasta format parser</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Matrix::PSM::IO::masta - motif fasta format parser

</pre><h4><b>SYNOPSIS</b></h4><pre>
       MASTA is a position frequency matrix format similar to fasta. It contains one ID row just like fasta and
       then the actual data, which is tab delimited:

         0.1   0.62    .017    0.11
         0.22  0.13    0.54    0.11

       Or A,C,G and T could be horizontally positioned (positioning is automatically detected).  Please note
       masta will parse only DNA at the moment.

       It will also convert a set of aligned sequences: ACATGCAT ACAGGGAT ACAGGCAT ACCGGCAT

       to a PFM (SiteMatrix object). When writing if you supply SEQ it will write 10 random instances, which
       represent correctly the frequency and can be used as an input for weblogo creation purposes.

       See Bio::Matrix::PSM::IO for detailed documentation on how to use masta parser

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Parser for meme.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably
        to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Stefan</b> <b>Kirov</b></h4><pre>
       Email <a href="mailto:skirov@utk.edu">skirov@utk.edu</a>

</pre><h4><b>APPENDIX</b></h4><pre>
   <b>new</b>
        Title   : new
        Usage   : my $psmIO =  Bio::Matrix::PSM::IO-&gt;new(-format=&gt; 'masta',
                                                        -file  =&gt; $file,
                                                        -mtype =&gt; 'PWM');
        Function: Associates a file with the appropriate parser
        Throws  :
        Example :
        Args    : hash
        Returns : "Bio::Matrix::PSM::$format"-&gt;new(@args);

   <b>write_psm</b>
        Title   : write_psm
        Usage   :
        Function: writes a pfm/pwm/raw sequence in a simple masta format
        Throws  :
        Example :
        Args    : SiteMatrix object, type (optional string: PWM, SEQ or PFM)
        Returns :

   <b>next_matrix</b>
         Title   : next_matrix
         Usage   : my $matrix = $psmio-&gt;next_matrix;
         Function: Alias of next_psm function

   <b>next_psm</b>
        Title   : next_psm
        Usage   : my $matrix=$psmio-&gt;next_psm;
        Function: returns the next matrix in the stream
        Throws  : If there is you mix different types, for example weights and
                  frequencies occur in the same entry You can mix weights, but these
                  should be designated by different ID lines
        Example :
        Args    :
        Returns : Bio::Matrix::PSM::SiteMatrix

perl v5.32.1                                       2021-08-15                   <u>Bio::Matrix::PSM::IO::<a href="../man3pm/masta.3pm.html">masta</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>