<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::AlignIO::Handler::GenericAlignHandler - Bio::HandlerI-based generic data handler class for</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::AlignIO::Handler::GenericAlignHandler - Bio::HandlerI-based generic data handler class for
       alignment-based data

</pre><h4><b>SYNOPSIS</b></h4><pre>
         # MyHandler is a GenericAlignHandler object.
         # inside a parser (driver) constructor....

         $self-&gt;alignhandler($handler || MyHandler-&gt;new(-format =&gt; 'stockholm'));

         # in next_aln() in driver...

         $hobj = $self-&gt;alignhandler();

         # roll data up into hashref chunks, pass off into Handler for processing...

         $hobj-&gt;data_handler($data);

         # or retrieve Handler methods and pass data directly to Handler methods...

         my $hmeth = $hobj-&gt;handler_methods;

         if ($hmeth-&gt;{ $data-&gt;{NAME} }) {
             my $mth = $hmeth-&gt;{ $data-&gt;{NAME} };
             $hobj-&gt;$mth($data);
         }

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This is an experimental implementation of a alignment-based HandlerBaseI parser and may change over time.
       It is possible that the way handler methods are set up will change over development to allow more
       flexibility.

       Standard Developer caveats:

       Here thar be dragoons...

       Consider yourself warned!

   <b>NOTES</b>
       As in the SeqIO Handler object (still in development), data is passed in as chunks. The Annotation and
       SeqFeatures are essentially the same as the SeqIO parser; the significant difference is that data hash
       being passed could pertain to either the alignment or to a specific sequence, so an extra tag may be
       needed to disambiguate between the two in some cases. Here I use the ALIGNMENT tag as a boolean flag: it
       must be present and set to 0 for the data to be tagged for Bio::LocatableSeq or similar (in all other
       cases it is assumed to be for the alignment). In some cases this will not matter (the actual sequence
       data, for instance) but it is highly recommended adding this tag in to prevent possible ambiguities.

       This is the current Annotation data chunk (via Data::Dumper):

         $VAR1 = {
                   'NAME' =&gt; 'REFERENCE',
                   'DATA' =&gt; '1  (bases 1 to 10001)'
                   'AUTHORS' =&gt; 'International Human Genome Sequencing Consortium.'
                   'TITLE' =&gt; 'The DNA sequence of Homo sapiens'
                   'JOURNAL' =&gt; 'Unpublished (2003)'
                   'ALIGNMENT' =&gt; 1,
                 };

       In the case of LocatableSeqs, one can pass them in as follows for simplicity (note the block line):

         $VAR1 = {
                   'NAME' =&gt; 'SEQUENCE',
                   'BLOCK_LINE' =&gt; 0,
                   'NSE' =&gt; 'Q7WNI7_BORBR/113-292',
                   'ALPHABET' =&gt; 'protein',
                   'DATA' =&gt; 'VALILGVYRRL...CYVNREM..RAG....QW',
                   'ALIGNMENT' =&gt; 0
                 };

       This can be done as the parser parses each block instead of parsing all the blocks and then passing them
       in one at a time; the handler will store the sequence data by the block line in an internal hash,
       concatenating them along the way.  This behaviour is b/c the alignment building step requires that the
       sequence be checked for start/end/strand, possible meta sequence, optional accession, etc.

       Similarly, a Meta sequence line can be passed in as follows:

         $VAR1 = {
                   'NAME' =&gt; 'NAMED_META',
                   'BLOCK_LINE' =&gt; 0,
                   'NSE' =&gt; 'Q7WNI7_BORBR/113-292',
                   'META_KEY' =&gt; 'pAS',
                   'DATA' =&gt; '................................',
                   'ALIGNMENT' =&gt; 0
                 };

       The meta sequence will be checked against the NSE for the block position and stored based on the meta
       tag. A meta sequence does not have to correspond to a real sequence. At this time, unique meta sequence
       tags must be used for each sequence or they will be overwritten (this may change).

       An alignment consensus string:

         $VAR1 = {
                   'NAME' =&gt; 'CONSENSUS',
                   'DATA' =&gt; 'VALILGVYRRL...CYVNREM..RAG....QW',
                   'ALIGNMENT' =&gt; 1
                 };

       A consensus meta sequence:

         $VAR1 = {
                   'NAME' =&gt; 'CONSENSUS_META',
                   'META_KEY' =&gt; 'pAS',
                   'DATA' =&gt; '................................',
                   'ALIGNMENT' =&gt; 1
                 };

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Chris</b> <b>Fields</b></h4><pre>
       Email cjfields at bioperl dot org

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   :  new
        Usage   :
        Function:
        Returns :
        Args    :  -format    Sequence format to be mapped for handler methods
                   -builder   Bio::Seq::SeqBuilder object (normally defined in
                              SequenceStreamI object implementation constructor)
        Throws  :  On undefined '-format' sequence format parameter
        Note    :  Still under heavy development

</pre><h4><b>Bio::HandlerBaseI</b> <b>implementing</b> <b>methods</b></h4><pre>
   <b>handler_methods</b>
        Title   :  handler_methods
        Usage   :  $handler-&gt;handler_methods('GenBank')
                   %handlers = $handler-&gt;handler_methods();
        Function:  Retrieve the handler methods used for the current format() in
                   the handler.  This assumes the handler methods are already
                   described in the HandlerI-implementing class.
        Returns :  a hash reference with the data type handled and the code ref
                   associated with it.
        Args    :  [optional] String representing the sequence format.  If set here
                   this will also set sequence_format()
        Throws  :  On unimplemented sequence format in %HANDLERS

   <b>data_handler</b>
        Title   :  data_handler
        Usage   :  $handler-&gt;data_handler($data)
        Function:  Centralized method which accepts all data chunks, then distributes
                   to the appropriate methods for processing based on the chunk name
                   from within the HandlerBaseI object.

                   One can also use
        Returns :  None
        Args    :  an hash ref containing a data chunk.

   <b>reset_parameters</b>
        Title   :  reset_parameters
        Usage   :  $handler-&gt;reset_parameters()
        Function:  Resets the internal cache of data (normally object parameters for
                   a builder or factory)
        Returns :  None
        Args    :  None

   <b>format</b>
        Title   :  format
        Usage   :  $handler-&gt;format('GenBank')
        Function:  Get/Set the format for the report/record being parsed. This can be
                   used to set handlers in classes which are capable of processing
                   similar data chunks from multiple driver modules.
        Returns :  String with the sequence format
        Args    :  [optional] String with the sequence format
        Note    :  The format may be used to set the handlers (as in the
                   current GenericRichSeqHandler implementation)

   <b>get_params</b>
        Title   :  get_params
        Usage   :  $handler-&gt;get_params('-species')
        Function:  Convenience method used to retrieve the specified
                   parameters from the internal parameter cache
        Returns :  Hash ref containing parameters requested and data as
                   key-value pairs.  Note that some parameter values may be
                   objects, arrays, etc.
        Args    :  List (array) representing the parameters requested

   <b>set_params</b>
        Title   :  set_params
        Usage   :  $handler-&gt;set_param({'-seqs' =&gt; $seqs})
        Function:  Convenience method used to set specific parameters
        Returns :  None
        Args    :  Hash ref containing the data to be passed as key-value pairs

</pre><h4><b>Methods</b> <b>unique</b> <b>to</b> <b>this</b> <b>implementation</b></h4><pre>
   <b>build_alignment</b>
        Title   :  build_alignment
        Usage   :
        Function:
        Returns :  a Bio::SimpleAlign
        Args    :
        Throws  :
        Note    :  This may be replaced by a Builder object at some point

   <b>annotation_collection</b>
        Title   :  annotation_collection
        Usage   :
        Function:
        Returns :
        Args    :
        Throws  :
        Note    :

   <b>seq_annotation_collection</b>
        Title   :  seq_annotation_collection
        Usage   :
        Function:
        Returns :
        Args    :
        Throws  :
        Note    :

   <b>process_seqs</b>
        Title   :  process_seqs
        Usage   :  $handler-&gt;process_seqs;
        Function:  checks internal sequences to ensure they are converted over
                   to the proper Bio::AlignI-compatible sequence class
        Returns :  1 if successful
        Args    :  none

perl v5.32.1                                       2021-08-15             <u>Bio::AlignIO::<a href="../man3pm/H...ricAlignHandler.3pm.html">H...ricAlignHandler</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>