<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>crac v. 2.5.2 - Crac is a tool to analyse RNA-Seq data provided by NGS.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/crac">crac_2.5.2+dfsg-6build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       crac v. 2.5.2 - Crac is a tool to analyse RNA-Seq data provided by NGS.

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>crac</b> <b>[</b> <b>options</b> <b>]</b> <b>-i</b> &lt;index_file&gt; <b>-r</b> &lt;reads_file1&gt; [reads_file2] <b>-k</b> &lt;int&gt; <b>-o</b> &lt;output_file&gt;

       <b>crac</b> <b>-h</b>|<b>--help</b>
       <b>crac</b> <b>-f</b>|<b>--full-help</b>
       <b>crac</b> <b>-v</b>|<b>--version</b>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>crac</b> CRAC: an integrated approach to the analysis of RNA-seq reads

       Whatever the biological questions it addresses, each RNA-seq analysis requires a computational prediction
       of  either  small  scale mutations, indels, splice junctions or fusion RNAs. This prediction is currently
       performed using complex pipelines  involving  multiple  tools  for  mapping,  coverage  computation,  and
       prediction  at  distinct  steps.   We  propose  a  novel  way  of analyzing reads that integrates genomic
       locations and local coverage, and delivers all above mentioned predictions in a single step. Our program,
       CRAC, uses a double k-mer profiling approach to detect candidate  mutations,  indels,  splice  or  fusion
       junctions  in  each  single read.  Compared to existing tools, CRAC provides state of the art sensitivity
       and improved precision for all types of predictions, yielding high  rates  of  true  positive  candidates
       (99.5%  for  splice  junctions).  When  applied  to  four  breast cancer libraries, CRAC recovered 74% of
       validated fusion RNAs and predicted reccurrent fusion junctions that were overseen in  previous  studies.
       Importantly, CRAC improves its predictive performance when supplied with e.g. 200 nt reads and should fit
       future needs of read analyses.

</pre><h4><b>SPECIAL</b> <b>OPTIONS</b></h4><pre>
       <b>crac</b> As a lot of softwares, there are many optional parameters in CRAC but only three are mandatory. This
       document  is intended to guide users of CRAC to choose the more appropriate parameters according to their
       needs:

       <b>-h</b>, <b>--help</b>
              to print the principal help page of CRAC

       <b>-f</b>, <b>--full-help</b>
              to print the complete help page of CRAC

       <b>-v</b>, <b>--version</b>
              to print version of CRAC

</pre><h4><b>USUAL</b> <b>ARGUMENTS</b></h4><pre>
   <b>Mandatory</b> <b>options</b>
       All these flags must be set.

       <b>-i</b> <u>&lt;index_file&gt;</u>
              is the name of the index previously built with the crac-index binary file.  Note  that  crac-index
              construct  the  structure  &lt;index_file.ssa&gt;  with  its configuration &lt;index_file.conf&gt; so only the
              prefix &lt;index_file&gt; must be specified (without  extension)  to  consider  the  structure  and  the
              configuration files both in CRAC

       <b>-r</b> <u>&lt;reads_file1&gt;</u> <u>[reads_file2]</u>
              is the source(s) of the FASTA or FASTQ file(s) containing the reads. Note that the number of files
              depends if single or paired-reads. The input file may also be compressed using gzip

       <b>-k</b> <u>&lt;int&gt;</u>
              is  the  length  of the k-mer to be used to map the reads on the reference &lt;index_file&gt;. Note that
              the condition (k &lt; m) is necessary and reads (or both paired reads) are ignored if m &lt; k. It  must
              be  chosen  to  ensure  (as  much as possible) that a k-mer has a very high probability to occur a
              single time on the genome

       <b>-o</b>, <b>--sam</b> <u>&lt;output_file&gt;</u>
              is the output file in SAM format (see the Documentation of SAM format in CRAC for more details) or
              print on STDOUT with "-o -" argument

   <b>Optional</b> <b>parameters</b>
       <b>--stranded</b>
              must be specificied if reads are produced by a stranded  protocol  of  RNA-Seq  (not  stranded  by
              default)

       <b>--fr/--rf/--ff</b>
              set the mates alignement orientation (--rf by default)

       <b>-m</b>, <b>--reads-length</b>, <b>-m</b> <u>&lt;int&gt;</u>
              must  be specified for reads of fixed length. If the read length is fixed, we deeply recommend you
              to specify the read length, by using the  -m  parameter.  CRAC  will  therefore  be  much  faster.
              --reads-length  &lt;int&gt;  is  specified  for  variable or longer reads, reads shorter are ignored and
              reads longer are trimmed

       <b>--treat-multiple</b> <u>&lt;int&gt;</u>
              display alignments with multiple locations (with a fixed limit) rather than a single alignment per
              read in the SAM file

       <b>--nb-threads</b> <u>&lt;int&gt;</u>
              is the number of threads to run crac, computational time  is  almost  divided  by  the  number  of
              threads (one thread by default)

       <b>--max-locs</b> <u>&lt;int&gt;</u>
              corresponds  to the max number of occurrences retrieved in the index for a given k-mer: smaller is
              faster, but with a small value, you may miss some locations that would  help  CRAC  detecting  the
              right cause

       <b>--no-ambiguity</b> <u>&lt;none&gt;</u>
              discard  biological  events  (splice,  svn,  indel,  chimera)  which  have  several matches on the
              reference index.  Indeed, if crac has identified a biological cause in the read that can match  in
              differents places of the genome we classify this cause as a biological undetermined event.

   <b>Optional</b> <b>output</b> <b>arguments</b>
       <b>--gz</b> <u>&lt;none&gt;</u>
              all  output files specified after this argument are gzipped (included for the sam file if -o/--sam
              argument is specified after)

       <b>--bam</b> <u>&lt;none&gt;</u>
              sam output is encode in binary format(BAM)

       <b>--summary</b> <u>&lt;output_file&gt;</u>
              save some statistics about mapping and classification

       <b>--show-progressbar</b> <u>&lt;none&gt;</u>
              show a progress bar for the process times on STDERR

       <b>--use-x-in-cigar</b> <u>&lt;none&gt;</u>
              use X cigar operator when CRAC identifies a mismatch

   <b>Optional</b> <b>output</b> <b>homemade</b> <b>file</b> <b>formats</b>
       <b>--all</b> <u>&lt;base_filename&gt;</u>
              set output base filename for all  causes  following.  Note  that  only  a  base_filename  must  be
              specified.  Then,  the  appropriate  file extension is added for each cause (SNP, chimera, splice,
              etc) set output base filename for all causes following

       <b>--normal</b> <u>&lt;output_file&gt;</u>
              save reads that do not contain any break

       <b>--almost-normal</b> <u>&lt;output_file&gt;</u>
              save reads that do not contain any break but with a variable support

       <b>--single</b> <u>&lt;output_file&gt;</u>
              save reads which are located in this way: at least --min-percent-single-loc &lt;float&gt; of k-mers  are
              once located on the reference index

       <b>--duplicate</b> <u>&lt;output_file&gt;</u>
              save reads which are located in this way: at least --min-percent-duplication-loc &lt;float&gt; of k-mers
              are  a few times on the reference index (ie. between --min-duplication &lt;int&gt; and --max-duplication
              &lt;int&gt; of locations)

       <b>--multiple</b> <u>&lt;output_file&gt;</u>
              save reads which are located in this way: at least --min-percent-multiple-loc  &lt;float&gt;  of  k-mers
              are a many times on the reference index (ie. more than --max-duplication &lt;int&gt; of locations)

       <b>--none</b> <u>&lt;output_file&gt;</u>
               save reads which are not located on the reference index

       <b>--snv</b> <u>&lt;output_file&gt;</u>
              save reads that contain at least a snv

       <b>--indel</b> <u>&lt;output_file&gt;</u>
              save reads that contain at least a biological indel

       <b>--splice</b> <u>&lt;output_file&gt;</u>
              save reads that contain at least a splicing junction

       <b>--weak-splice</b> <u>&lt;output_file&gt;</u>
              save reads that contain at least a low coverage splicing junction

       <b>--chimera</b> <u>&lt;output_file&gt;</u>
              save reads that contain at least a chimera junction (junction on different chromosomes, strands or
              genes)

       <b>--paired-end-chimera</b> <u>&lt;output_file&gt;</u>
              paired-end-chimera  &lt;output_file&gt;=  save  paired-end  reads  that  contains  a chimera in the non-
              sequenced part of the original fragment.

       <b>--biological</b> <u>&lt;output_file&gt;</u>
              save reads that contain a biological cause but for which there is not enough  informations  to  be
              more specific. Note that the biological cause is described for each read

       <b>--errors</b> <u>&lt;output_file&gt;</u>
              save reads that contain at least a sequence error

       <b>--repeat</b> <u>&lt;output_file&gt;</u>
               save reads that contain a repeated sequence: at least
               --min-percent-repetition-loc &lt;float&gt; percent of k-mers of a given
               read are located at least --min-repetition &lt;int&gt; occurrences on the
               reference index

       <b>--undetermined</b> <u>&lt;output_file&gt;</u>
              save  reads that contain an undetermined error: some k-mers are not located on the genome, but the
              reason for that could not be determined. Note that the error is described for each read

       <b>--nothing</b> <u>&lt;output_file&gt;</u>
              save reads that are unclassified

   <b>Optional</b> <b>process</b> <b>for</b> <b>specific</b> <b>research</b>
       <b>--deep-snv</b>
              must be specified to increase sensitivity to find SNVs at the  cost  of  more  computations  (only
              substitution,  no  indels  YET).  That process searches for SNV in border cases reads. Those reads
              would otherwise be classified in bioundetermined

       <b>--stringent-chimera</b>
              must be specified to increase accuracy to find chimera junctions in exchange  of  sensitivity  and
              computational times

   <b>Optional</b> <b>process</b> <b>launcher</b> <b>(once</b> <b>must</b> <b>be</b> <b>selected)</b>
       <b>--emt</b>  launch  an  exact matching processing of reads on the index. Either the argument specified with -k
              is equal to 0 which means that the entire read is perfectly mapped on the genome or only a  factor
              of  length  k  per read is mapped (the first one with a location) and the rest is sofclipped. With
              this process, reads are not indexed and it provides a low memory consumption. Note  this  kind  of
              method is very useful for DGE reads mapping.

       <b>--server</b>
              launch  a  server to query a given read more precisely. That process is useful for debugging. Note
              that the output arguments will not be taken into account. Give an --input-name-server &lt;string&gt;  to
              set  the  input  fifo name (classify.fifo by default) and give an --output-name-server &lt;string&gt; to
              set the output fifo name (classify.out.fifo by default). The server can then  be  used  through  a
              client crac-client

   <b>Additional</b> <b>settings</b> <b>for</b> <b>users</b>
       <b>--detailed-sam</b>
              more  informations  are  added in SAM output file. See the Documentation of SAM format in CRAC for
              more details

       <b>--min-percent-single-loc</b> <u>&lt;float&gt;</u>
              is, to consider a given read as uniquely  mapped,  the  minimum  proportion  of  k-mers  that  are
              uniquely mapped on the index (0.15 by default)

       <b>--min-duplication</b> <u>&lt;int&gt;</u>
              is the minimum number of location to consider a duplicated k-mer (2 by default)

       <b>--max-duplication</b> <u>&lt;int&gt;</u>
              is the maximum number of location to consider a duplicated k-mer (9 by default)

       <b>--min-percent-duplication-loc</b> <u>&lt;float&gt;</u>
              is,  to  consider a given read as duplicated, the minimum proportion of k-mers that are duplicated
              on the index (0.15 by default)

       <b>--min-percent-multiple-loc</b> <u>&lt;float&gt;</u>
              is, to consider a given read as “multiple”, the minimum proportion of  k-mers  that  are  multiple
              mapped on the index (0.50 by default)

       <b>--min-repetition</b> <u>&lt;int&gt;</u>
              is the minimum number of locations to consider a repeated k-mer (20 by default)

       <b>--max-percent-repetition-loc</b> <u>&lt;float&gt;</u>
              is,  for a given read, the minimum proportion of k-mers that are repeated on the index to consider
              a repetition (0.20 by default)

       <b>--max-splice-length</b> <u>&lt;int&gt;</u>
              is the threshold to consider a splice, ie. a splice is reported if the junction  length  is  below
              max-splice-length &lt;int&gt;, a chimera is considered otherwise (distance by default is 300Kb)

       <b>--max-bio-indel</b> <u>&lt;int&gt;</u>
              is  the  threshold  to  consider a biological indel, ie. an indel is reported if the gap length is
              below max-bio-indel, a splice is considered otherwise (distance by default is 15)

       <b>--max-bases-retrieved</b> <u>&lt;int&gt;</u>
              is the number of nucleotides to display in outputfile in case of insertion (15 by default)

       <b>--min-support-no-cover</b> <u>&lt;float&gt;</u>
              is the minimum coverage to be able to report a biological  cause.  Note  that  if  a  single  read
              contains a given substitution, it is difficult (if not impossible) to distinguish a sequence error
              and a biological cause (1.30 by default)

   <b>Additional</b> <b>settings</b> <b>for</b> <b>advanced</b> <b>users</b>
       <b>--min-break-length</b> <u>&lt;int&gt;</u>
              is  the  minimal  break  length  (as the percentage of k, the k-mer length) so that a cause can be
              reported. Theoretically, for a given cause, the break length  is  always  &gt;=  (kmer_length  -  1).
              Otherwise,  the  break may be merged with a close enough break, or the break will be considered as
              undetermined. (0.5 by default)

       <b>--max-bases-randomly-matched</b> <u>&lt;int&gt;</u>
              A k-mer overlapping an exon-exon junction, for example, may still  match  on  the  genome  if  the
              overlap  is  at the end of the read (without loss of generality). This is due to the fact that the
              nucleotides starting the second exon may be the same  as  the  nucleotides  starting  the  intron.
              Theoretically,  there is a 0.25 probability that we have the same nucleotide at the first position
              of the intron and the exon. This option specifies how many nucleotides may be matched randomly  at
              most

       <b>--max-extension-length</b> <u>&lt;int&gt;</u>
              is the maximum number of k-mers extended at each side of a read break. In fact, for a given break,
              k-mers  with  false  locations can generate false biological causes, so the consistency is checked
              for each side of the break to discard false k-mers and readjust the good boundaries of  the  break
              (10 by default)

       <b>--nb-tags-info-stored</b> <u>&lt;int&gt;</u>
              is  a  buffer  to store informations for each thread during the computing phase (1000 by default).
              This value must be increased if threads work below their real capabilities. With --nb-threads  15,
              CPU usage must be about 1400%

       <b>--reads-index</b> <u>&lt;string&gt;</u>
              the  reads  index  data-structure uses by CRAC. Available reads index are: JELLYFISH and GKARRAYS.
              (JELLYFISH by default).

       <b>--nb-nucleotides-snp-comparison</b> <u>&lt;int&gt;</u>
               is the minimum k-mer length tolerated for the deep SNVs search (8 by
               default)

       <b>--max-number-of-merges</b> <u>&lt;int&gt;</u>
               is the maximum number of merges tolerated during the break merge process
               for the chimera detection (4 by default)

       <b>--min-score-chimera-stringent</b> <u>&lt;float&gt;</u>
               is the mimimal score to consider a chimera event
               otherwise it is classify as a bioundetermined event (0.6 by default)

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       The full documentation for <b>crac</b> is maintained as a org  manual.   If  the  <b>info</b>  and  <b>crac</b>  programs  are
       properly installed at your site, the command

              <b>info</b> <b>crac</b>

       should give you access to the complete manual.

</pre><h4><b>AUTHOR</b></h4><pre>
   <b>About</b> <b>the</b> <b>crac</b> <b>package.</b>
       You  can  contact Nicolas PHILIPPE, Mikael SALSON, Jerome AUDOUX and Alban MANCHERON by sending an e-mail
       to &lt;<a href="mailto:crac-bugs@lists.gforge.inria.fr">crac-bugs@lists.gforge.inria.fr</a>&gt;.

       Programming:
               Nicolas PHILIPPE &lt;<a href="mailto:nphilippe.research@gmail.com">nphilippe.research@gmail.com</a>&gt;
               Mikaël SALSON    &lt;<a href="mailto:mikael.salson@lifl.fr">mikael.salson@lifl.fr</a>&gt;      Jérome AUDOUX    &lt;<a href="mailto:jerome.audoux@gmail.com">jerome.audoux@gmail.com</a>&gt;
       with additional contribution for the packaging of:
               Alban MANCHERON  &lt;<a href="mailto:alban.mancheron@lirmm.fr">alban.mancheron@lirmm.fr</a>&gt;

   <b>About</b> <b>the</b> <b>crac</b> <b>publication.</b>
       You may cite the following paper if you use our tool:

       Gk-arrays: Querying large read collections in main memory: a versatile
       data structure
       <b>Philippe</b> <b>N.,</b> <b>Salson</b> <b>M.,</b> <b>Lecroq</b> <b>T.,</b> <b>Leonard</b> <b>M.,</b> <b>Commes</b> <b>T.,</b> <b>Rivals</b> <b>E.</b>
       BMC Bioinformatics 2011, 12:242.

       Crac: An integrated RNA-Seq read analysis
       <b>Philippe</b> <b>N.,</b> <b>Salson</b> <b>M.,</b> <b>Commes</b> <b>T.,</b> <b>Rivals</b> <b>E.</b>
       Genome Biology 2013; 14:R30.

                                                   2024-04-01                                            <u><a href="../man1/crac.1.html">crac</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>