<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Ace::Sequence::Gene - Simple "Gene" Object</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libace-perl">libace-perl_1.92-12build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Ace::Sequence::Gene - Simple "Gene" Object

</pre><h4><b>SYNOPSIS</b></h4><pre>
           # open database connection and get an Ace::Object sequence
           use Ace::Sequence;

           # get a megabase from the middle of chromosome I
           $seq = Ace::Sequence-&gt;new(-name   =&gt; 'CHROMOSOME_I,
                                     -db     =&gt; $db,
                                     -offset =&gt; 3_000_000,
                                     -length =&gt; 1_000_000);

           # get all the genes
           @genes = $seq-&gt;genes;

           # get the exons from the first one
           @exons = $genes[0]-&gt;exons;

           # get the introns
           @introns = $genes[0]-&gt;introns

           # get the CDSs (NOT IMPLEMENTED YET!)
           @cds = $genes[0]-&gt;cds;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Ace::Sequence::Gene is a subclass of Ace::Sequence::Feature.  It inherits all the methods of
       Ace::Sequence::Feature, but adds the ability to retrieve the annotated introns and exons of the gene.

</pre><h4><b>OBJECT</b> <b>CREATION</b></h4><pre>
       You will not ordinarily create an <u>Ace::Sequence::Gene</u> object directly.  Instead, objects will be created
       in response to a <b>genes()</b> call to an <u>Ace::Sequence</u> object.

</pre><h4><b>OBJECT</b> <b>METHODS</b></h4><pre>
       Most methods are inherited from <u>Ace::Sequence::Feature</u>.  The following methods are also supported:

       <b>exons()</b>
             @exons = $gene-&gt;exons;

           Return a list of Ace::Sequence::Feature objects corresponding to annotated exons.

       <b>introns()</b>
             @introns = $gene-&gt;introns;

           Return a list of Ace::Sequence::Feature objects corresponding to annotated introns.

       <b>cds()</b>
             @cds = $gene-&gt;cds;

           Return  a  list  of Ace::Sequence::Feature objects corresponding to coding sequence.  THIS IS NOT YET
           IMPLEMENTED.

       <b>relative()</b>
             $relative = $gene-&gt;relative;
             $gene-&gt;<a href="../man1/relative.1.html">relative</a>(1);

           This turns on and off relative coordinates.  By default,  the  exons  and  intron  features  will  be
           returned  in  the  coordinate  system  used  by the gene.  If <b>relative()</b> is set to a true value, then
           coordinates will be expressed as relative to the start of the gene.  The first exon will (usually) be
           1.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Ace, Ace::Object, Ace::Sequence,Ace::Sequence::Homol, Ace::Sequence::Feature, Ace::Sequence::FeatureList,
       GFF

</pre><h4><b>AUTHOR</b></h4><pre>
       Lincoln Stein &lt;<a href="mailto:lstein@cshl.org">lstein@cshl.org</a>&gt; with extensive help from Jean Thierry-Mieg &lt;<a href="mailto:mieg@kaa.crbm.cnrs-mop.fr">mieg@kaa.crbm.cnrs-mop.fr</a>&gt;

       Copyright (c) 1999, Lincoln D. Stein

       This library is free software; you can redistribute it and/or modify it under  the  same  terms  as  Perl
       itself.  See DISCLAIMER.txt for disclaimers of warranty.

</pre><h4><b>POD</b> <b>ERRORS</b></h4><pre>
       Hey! <b>The</b> <b>above</b> <b>document</b> <b>had</b> <b>some</b> <b>coding</b> <b>errors,</b> <b>which</b> <b>are</b> <b>explained</b> <b>below:</b>

       Around line 148:
           You forgot a '=back' before '=head1'

perl v5.40.0                                       2024-10-20                           <u>Ace::Sequence::<a href="../man3pm/Gene.3pm.html">Gene</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>