<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::SeqIO::entrezgene - Entrez Gene ASN1 parser</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-asn1-entrezgene-perl">libbio-asn1-entrezgene-perl_1.730-3_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::SeqIO::entrezgene - Entrez Gene ASN1 parser

</pre><h4><b>VERSION</b></h4><pre>
       version 1.73

</pre><h4><b>SYNOPSIS</b></h4><pre>
          use Bio::SeqIO;

          # don't instantiate directly - instead do
          my $seqio = Bio::SeqIO-&gt;new(-format =&gt; 'entrezgene',
                                      -file =&gt; $file);
          my $gene = $seqio-&gt;next_seq;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This is EntrezGene ASN bioperl parser. It is built on top of Bio::ASN1::EntrezGene, a low level ASN
       parser built by Mingyi Liu (&lt;<a href="http://sourceforge.net/projects/egparser">http://sourceforge.net/projects/egparser</a>&gt;). The easiest way to use it is
       shown above.

       You will get most of the Entrez Gene annotation such as gene symbol, gene name and description, accession
       numbers associated with the gene, etc. Almost all of these are given as  Bio::AnnotationI objects.

       If you need all the data do:

          my $seqio = Bio::SeqIO-&gt;new(-format =&gt; 'entrezgene',
                                      -file =&gt; $file,
                                      -debug =&gt; 'on' );
          my ($gene,$genestructure,$uncaptured) = $seqio-&gt;next_seq;

       The second variable returned, $genestructure, is a Bio::Cluster::SequenceFamily object. It contains all
       Refseqs and the genomic contigs that are associated with the particular gene. The third variable,
       $uncaptured, is a reference to a plain array.

       You can also modify the output to allow back compatibility with the old LocusLink parser:

          my $seqio = Bio::SeqIO-&gt;new(-format =&gt; 'entrezgene',
                                      -file =&gt; $file,
                                      -locuslink =&gt; 'convert');

       The "-debug" and "-locuslink" options slow down the parser.

       Example code which looks for ontology terms:

         my $eio = new Bio::SeqIO(-file =&gt; $file,
                                  -format =&gt; 'entrezgene',
                                  -service_record =&gt; 'yes');

         while (my $seq = $eio-&gt;next_seq) {
           my $gid = $seq-&gt;accession_number;
           foreach my $ot ($ann-&gt;get_Annotations('OntologyTerm')) {
             next if ($ot-&gt;term-&gt;authority eq 'STS marker'); # No STS markers
             my $evid = $ot-&gt;comment;
             $evid =~ s/evidence: //i;
             my @ref = $ot-&gt;term-&gt;get_references;
             my $id = $ot-&gt;identifier;
             my $fid = 'GO:' . sprintf("%07u",$id);
             print join("\t",$gid, $ot-&gt;ontology-&gt;name, $ot-&gt;name, $evid,
               $fid, @ref?$ref[0]-&gt;medline:''), "\n";
           }
         }

</pre><h4><b>_process_products_coordinates</b></h4><pre>
       To do:

</pre><h4><b>_process_prop</b></h4><pre>
       To do: process GO

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>              - General discussion
         <a href="http://bioperl.org/Support.html">http://bioperl.org/Support.html</a>    - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list: <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bio-asn1-entrezgene/issues

</pre><h4><b>AUTHOR</b></h4><pre>
       Stefan Kirov &lt;<a href="mailto:skirov@utk.edu">skirov@utk.edu</a>&gt;

</pre><h4><b>COPYRIGHT</b></h4><pre>
       This software is copyright (c) by Stefan Kirov &lt;<a href="mailto:skirov@utk.edu">skirov@utk.edu</a>&gt;.

       This software is available under the same terms as the perl 5 programming language system itself.

perl v5.38.2                                       2024-03-04                        <u>Bio::SeqIO::<a href="../man3pm/entrezgene.3pm.html">entrezgene</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>