<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::SeqFeature::Gene::GeneStructure - A feature representing an arbitrarily complex structure of a gene</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::SeqFeature::Gene::GeneStructure - A feature representing an arbitrarily complex structure of a gene

</pre><h4><b>SYNOPSIS</b></h4><pre>
         # See documentation of methods.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       A feature representing a gene structure. As of now, a gene structure really is only a collection of
       transcripts. See Bio::SeqFeature::Gene::TranscriptI (interface) and Bio::SeqFeature::Gene::Transcript
       (implementation) for the features of such objects.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track
        the bugs and their resolution. Bug reports can be submitted via the
        web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Hilmar</b> <b>Lapp</b></h4><pre>
       Email hlapp-at-gmx.net

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>transcripts</b>
        Title   : transcripts
        Usage   : @transcripts = $gene-&gt;transcripts();
        Function: Get the transcripts of this gene structure. Many gene structures
                  will have only one transcript.

        Returns : An array of Bio::SeqFeature::Gene::TranscriptI implementing objects.
        Args    :

   <b>add_transcript</b>
        Title   : add_transcript()
        Usage   : $gene-&gt;add_transcript($transcript);
        Function: Add a transcript to this gene structure.
        Returns :
        Args    : A Bio::SeqFeature::Gene::TranscriptI implementing object.

   <b>flush_transcripts</b>
        Title   : flush_transcripts()
        Usage   : $gene-&gt;flush_transcripts();
        Function: Remove all transcripts from this gene structure.
        Returns :
        Args    :

   <b>add_transcript_as_features</b>
        Title   : add_transcript_as_features
        Usage   : $gene-&gt;add_transcript_as_features(@featurelist);
        Function: take a list of Bio::SeqFeatureI objects and turn them into a
                  Bio::SeqFeature::Gene::Transcript object.  Add that transcript to the gene.
        Returns : nothing
        Args    : a list of Bio::SeqFeatureI compliant objects

   <b>promoters</b>
        Title   : promoters
        Usage   : @prom_sites = $gene-&gt;promoters();
        Function: Get the promoter features of this gene structure.

                  This method basically merges the promoters returned by transcripts.

                  Note that OO-modeling of regulatory elements is not stable yet.
                  This means that this method might change or even disappear in a
                  future release. Be aware of this if you use it.

        Returns : An array of Bio::SeqFeatureI implementing objects.
        Args    :

   <b>exons</b>
        Title   : exons()
        Usage   : @exons = $gene-&gt;exons();
                  @inital_exons = $gene-&gt;exons('Initial');
        Function: Get all exon features or all exons of a specified type of this gene
                  structure.

                  Exon type is treated as a case-insensitive regular expression and
                  optional. For consistency, use only the following types:
                  initial, internal, terminal, utr, utr5prime, and utr3prime.
                  A special and virtual type is 'coding', which refers to all types
                  except utr.

                  This method basically merges the exons returned by transcripts.

        Returns : An array of Bio::SeqFeature::Gene::ExonI implementing objects.
        Args    : An optional string specifying the type of exon.

   <b>introns</b>
        Title   : introns()
        Usage   : @introns = $gene-&gt;introns();
        Function: Get all introns of this gene structure.

                  Note that this class currently generates these features on-the-fly,
                  that is, it simply treats all regions between exons as introns.
                  It assumes that the exons in the transcripts do not overlap.

                  This method basically merges the introns returned by transcripts.

        Returns : An array of Bio::SeqFeatureI implementing objects.
        Args    :

   <b>poly_A_sites</b>
        Title   : poly_A_sites()
        Usage   : @polyAsites = $gene-&gt;poly_A_sites();
        Function: Get the poly-adenylation sites of this gene structure.

                  This method basically merges the poly-adenylation sites returned by
                  transcripts.

        Returns : An array of Bio::SeqFeatureI implementing objects.
        Args    :

   <b>utrs</b>
        Title   : utrs()
        Usage   : @utr_sites = $gene-&gt;utrs('3prime');
                  @utr_sites = $gene-&gt;utrs('5prime');
                  @utr_sites = $gene-&gt;utrs();
        Function: Get the features representing untranslated regions (UTR) of this
                  gene structure.

                  You may provide an argument specifying the type of UTR. Currently
                  the following types are recognized: 5prime 3prime for UTR on the
                  5' and 3' end of the CDS, respectively.

                  This method basically merges the UTRs returned by transcripts.

        Returns : An array of Bio::SeqFeature::Gene::ExonI implementing objects
                  representing the UTR regions or sites.
        Args    : Optionally, either 3prime, or 5prime for the the type of UTR
                  feature.

   <b>sub_SeqFeature</b>
        Title   : sub_SeqFeature
        Usage   : @feats = $gene-&gt;sub_SeqFeature();
        Function: Returns an array of all subfeatures.

                  This method is defined in Bio::SeqFeatureI. We override this here
                  to include the transcripts.

        Returns : An array Bio::SeqFeatureI implementing objects.
        Args    : none

   <b>flush_sub_SeqFeature</b>
        Title   : flush_sub_SeqFeature
        Usage   : $gene-&gt;flush_sub_SeqFeature();
                  $gene-&gt;<a href="../man1/flush_sub_SeqFeature.1.html">flush_sub_SeqFeature</a>(1);
        Function: Removes all subfeatures.

                  This method is overridden from Bio::SeqFeature::Generic to flush
                  all additional subfeatures, i.e., transcripts, which is
                  almost certainly not what you want. To remove only features added
                  through $gene-&gt;add_sub_SeqFeature($feature) pass any
                  argument evaluating to TRUE.

        Example :
        Returns : none
        Args    : Optionally, an argument evaluating to TRUE will suppress flushing
                  of all gene structure-specific subfeatures (transcripts).

perl v5.32.1                                       2021-08-15             <u>Bio::SeqFeature...::<a href="../man3pm/GeneStructure.3pm.html">GeneStructure</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>