<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::SeqFeature::Gene::GeneStructureI - A feature representing an arbitrarily</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::SeqFeature::Gene::GeneStructureI - A feature representing an arbitrarily
                  complex structure of a gene

</pre><h4><b>SYNOPSIS</b></h4><pre>
         #documentation needed

</pre><h4><b>DESCRIPTION</b></h4><pre>
       A feature representing a gene structure.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists. Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Hilmar</b> <b>Lapp</b></h4><pre>
       Email <a href="mailto:hlapp@gmx.net">hlapp@gmx.net</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>transcripts</b>
        Title   : transcripts()
        Usage   : @transcripts = $gene-&gt;transcripts();
        Function: Get the transcript features/sites of this gene structure.

                  See Bio::SeqFeature::Gene::TranscriptI for properties of the
                  returned objects.

        Returns : An array of Bio::SeqFeature::Gene::TranscriptI implementing objects
                  representing the promoter regions or sites.
        Args    :

   <b>promoters</b>
        Title   : promoters()
        Usage   : @prom_sites = $gene-&gt;promoters();
        Function: Get the promoter features/sites of this gene structure.

                  Note that OO-modeling of regulatory elements is not stable yet.
                  This means that this method might change or even disappear in a
                  future release. Be aware of this if you use it.

        Returns : An array of Bio::SeqFeatureI implementing objects representing the
                  promoter regions or sites.
        Args    :

   <b>exons</b>
        Title   : exons()
        Usage   : @exons = $gene-&gt;exons();
                  @inital = $gene-&gt;exons('Initial');
        Function: Get all exon features or all exons of specified type of this gene
                  structure.

                  Refer to the documentation of the class that produced this gene
                  structure object for information about the possible types.

                  See Bio::SeqFeature::Gene::ExonI for properties of the
                  returned objects.

        Returns : An array of Bio::SeqFeature::Gene::ExonI implementing objects
                  representing the exon regions.
        Args    : An optional string specifying the type of the exon.

   <b>introns</b>
        Title   : introns()
        Usage   : @introns = $gene-&gt;introns();
        Function: Get all introns of this gene structure.
        Returns : An array of Bio::SeqFeatureI implementing objects representing the
                  introns.
        Args    :

   <b>poly_A_sites</b>
        Title   : poly_A_sites()
        Usage   : @polyAsites = $gene-&gt;poly_A_sites();
        Function: Get the poly-adenylation features/sites of this gene structure.
        Returns : An array of Bio::SeqFeatureI implementing objects representing the
                  poly-adenylation regions or sites.
        Args    :

   <b>utrs</b>
        Title   : utrs()
        Usage   : @utr_sites = $gene-&gt;utrs();
        Function: Get the UTR features/sites of this gene structure.

                  See Bio::SeqFeature::Gene::ExonI for properties of the
                  returned objects.

        Returns : An array of Bio::SeqFeature::Gene::ExonI implementing objects
                  representing the UTR regions or sites.
        Args    :

perl v5.32.1                                       2021-08-15              <u>Bio::SeqFeatur...:<a href="../man3pm/GeneStructureI.3pm.html">GeneStructureI</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>