<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Run::Genewise - Object for predicting genes in a given sequence given a protein</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-perl-run-perl">libbio-perl-run-perl_1.7.3-8_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Run::Genewise - Object for predicting genes in a given sequence given a protein

</pre><h4><b>SYNOPSIS</b></h4><pre>
         # Build a Genewise alignment factory
         my $factory = Bio::Tools::Run::Genewise-&gt;new();

         # Pass the factory 2 Bio:SeqI objects (in the order of query peptide
         # and target_genomic).

         # @genes is an array of Bio::SeqFeature::Gene::GeneStructure objects
         my @genes = $factory-&gt;run($protein_seq, $genomic_seq);

         # Alternatively pass the factory a profile HMM filename and a
         # Bio:SeqI object (in the order of query HMM and target_genomic).

         # Set hmmer switch first to tell genewise to expect an HMM
         $factory-&gt;<a href="../man1/hmmer.1.html">hmmer</a>(1);
         my @genes = $factory-&gt;run($hmmfile, $genomic_seq);

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Genewise is a gene prediction program developed by Ewan Birney <a href="http://www.sanger.ac.uk/software/wise2">http://www.sanger.ac.uk/software/wise2</a>.

   <b>Available</b> <b>Params:</b>
       NB: These should be passed without the '-' or they will be ignored, except switches such as 'hmmer'
       (which have no corresponding value) which should be set on the factory object using the AUTOLOADed
       methods of the same name.

         Model    [-codon,-gene,-cfreq,-splice,-subs,-indel,-intron,-null]
         Alg      [-kbyte,-alg]
         HMM      [-hmmer]
         Output   [-gff,-gener,-alb,-pal,-block,-divide]
         Standard [-help,-version,-silent,-quiet,-errorlog]

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track
        the bugs and their resolution.  Bug reports can be submitted via
        the web:

         <a href="http://redmine.open-bio.org/projects/bioperl/">http://redmine.open-bio.org/projects/bioperl/</a>

</pre><h4><b>AUTHOR</b> <b>-</b> <b>FUGU</b> <b>Student</b> <b>Intern</b></h4><pre>
       Email: <a href="mailto:fugui@worf.fugu-sg.org">fugui@worf.fugu-sg.org</a>

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
       Jason Stajich jason-AT-bioperl_DOT_org Keith James <a href="mailto:kdj@sanger.ac.uk">kdj@sanger.ac.uk</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>program_name</b>
        Title   : program_name
        Usage   : $factory&gt;program_name()
        Function: holds the program name
        Returns:  string
        Args    : None

   <b>program_dir</b>
        Title   : program_dir
        Usage   : $factory-&gt;program_dir(@params)
        Function: returns the program directory, obtained from ENV variable.
        Returns:  string
        Args    :

   <b>version</b>
        Title   : version
        Usage   : exit if $prog-&gt;version() &lt; 1.8
        Function: Determine the version number of the program
        Example :
        Returns : float or undef
        Args    : none

   <b>predict_genes</b>
        Title   : predict_genes
        Usage   : DEPRECATED. Use $factory-&gt;run($seq1,$seq2)
        Function: Predict genes
        Returns : A Bio::Seqfeature::Gene:GeneStructure object
        Args    : Name of a file containing a set of 2 fasta sequences in the order of
                  peptide and genomic sequences
                  or else 2  Bio::Seq objects.

       Throws an exception if argument is not either a string (eg a filename) or 2 Bio::Seq objects.  If
       arguments are strings, throws exception if file corresponding to string name can not be found.

   <b>run</b>
        Title   : run
        Usage   : 2 sequence objects
                  $genes = $factory-&gt;run($seq1, $seq2);
        Function: run
        Returns : A Bio::Seqfeature::Gene:GeneStructure object
        Args    : Names of a files each containing a fasta sequence in the order
                  of either (peptide sequence, genomic sequence) or (profile HMM,
                  genomic sequence). Alternatively any of the fasta sequence
                  filenames may be substituted with a Bio::Seq object.

       Throws an exception if argument is not either a string (eg a filename) or Bio::Seq objects. If arguments
       are strings, throws exception if file corresponding to string name can not be found. Also throws an
       exception if a profile HMM is expected (the -hmmer genewise switch has been set).

   <b>_run</b>
        Title   : _run
        Usage   : Internal function, not to be called directly
        Function: Makes actual system call to a genewise program
        Example :
        Returns : L&lt;Bio::SeqFeature::Gene::GeneStructure&gt;
        Args    : Name of a files containing 2 sequences in the order of peptide and genomic

   <b>_setparams</b>
        Title   :  _setparams
        Usage   :  Internal function, not to be called directly
        Function:  creates a string of params to be used in the command string
        Example :
        Returns :  string of params
        Args    :

   <b>_query_pep_seq</b>
        Title   :  _query_pep_seq
        Usage   :  Internal function, not to be called directly
        Function:  get/set for the query sequence
        Example :
        Returns :
        Args    :

   <b>_subject_dna_seq</b>
        Title   :  _subject_dna_seq
        Usage   :  Internal function, not to be called directly
        Function:  get/set for the subject sequence
        Example :
        Returns :

        Args    :

perl v5.34.0                                       2022-02-10                     <u>Bio::Tools::Run::<a href="../man3pm/Genewise.3pm.html">Genewise</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>