<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>score_conservation - score protein sequence conservation</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/conservation-code">conservation-code_20110309.0-8_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       score_conservation - score protein sequence conservation

</pre><h4><b>SYNOPSIS</b></h4><pre>
       score_conservation [options] ALIGNFILE

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Score protein sequence conservation in <b>ALIGNFILE</b>.  <b>ALIGNFILE</b> must be in FASTA, CLUSTAL or Stockholm
       format.

       The following conservation scoring methods are implemented:
        * sum of pairs
        * weighted sum of pairs
        * Shannon entropy
        * Shannon entropy with property groupings (Mirny and Shakhnovich 1995,
          Valdar and Thornton 2001)
        * relative entropy with property groupings (Williamson 1995)
        * von Neumann entropy (Caffrey et al 2004)
        * relative entropy (Samudrala and Wang 2006)
        * Jensen-Shannon divergence (Capra and Singh 2007)

       A window-based extension that incorporates the estimated conservation of sequentially adjacent residues
       into the score for each column is also given.  This window approach can be applied to any of the
       conservation scoring methods.

       With default parameters <b><a href="../man1/score_conservation.1.html">score_conservation</a></b>(1) computes the conservation scores for the alignment using
       the Jensen-Shannon divergence and a window <b>-w</b> of <u>3</u>.

       The sequence-specific output can be used as the conservation input for <b><a href="../man1/concavity.1.html">concavity</a></b>(1).

       Conservation is highly predictive in identifying catalytic sites and residues near bound ligands.

</pre><h4><b>REFERENCES</b></h4><pre>
       Capra JA and Singh M. Predicting functionally important residues from sequence conservation.
       Bioinformatics, <a href="../man15/23.15.html">23</a>(15):1875-82, 2007.

</pre><h4><b>OPTIONS</b></h4><pre>
       -a [NAME]
           Reference  sequence.  Print scores in reference to the named sequence (ignoring gaps). Default prints
           the entire column.

       -b [0-1]
           Lambda for window heuristic linear combination. Default=<u>.5</u>.

           Equation:

           "score = (1 - lambda) * average_score_over_window_around_middle + lambda * score_of_middle"

       -d [FILE]
           Background distribution file, e.g. <u>distributions/swissprot.distribution</u>. Default=built-in BLOSUM62.

       -g [0-1)]
           Gap cutoff. Do not score columns that contain more than gap cutoff fraction gaps. Default=<u>.3</u>.

       -h  Print help.

       -l [true|false]
           Use sequence weighting. Default=<u>true</u>.

       -m [FILE]
           Similarity matrix file, e.g. <u>matrix/blosum62.bla</u> or .qij. Default=<u>matrix/blosum62.bla</u>.

           Some methods, e.g. <u>js_divergence</u>, do not use this.

       -n [true|false]
           Normalize scores. Print the z-score (over the alignment) of each column raw score. Default=<u>false</u>.

       -o FILE
           Output file. Default: standard output stream.

       -p [true|false]
           Use gap penalty. Lower the score of columns that contain gaps, proportionally to the  sum  weight  of
           the gapped sequences. Default=<u>true</u>.

       -s [METHOD]
           Conservation  estimation  method,  one  of <u>shannon_entropy</u> <u>property_entropy</u> <u>property_relative_entropy</u>
           <u>vn_entropy</u> <u>relative_entropy</u> <u>js_divergence</u> <u>sum_of_pairs</u>. Default=<u>js_divergence</u>.

       -w [0-INT]
           Window size. Number of residues on either side included in the window. Default=<u>3</u>.

</pre><h4><b>EXAMPLES</b></h4><pre>
       Note: you may have to copy and uncompress the example data files before running the following examples.

       Compute conservation scores for the alignment using the Jensen-Shannon divergence with default settings
       and print out the scores:
            score_conservation /usr/share/doc/conservation-code/examples/2plc__hssp-filtered.aln

       Score an alignment using Jensen-Shannon divergence, a window of size 3 (on either side of the residue),
       and the swissprot background distribution:
            score_conservation -s js_divergence -w 3 -d \
             /usr/share/conservation-code/distributions/swissprot.distribution \
             /usr/share/doc/conservation-code/examples/2plc__hssp-filtered.aln

</pre><h4><b>FILES</b></h4><pre>
       Distributions
           <u>/usr/share/conservation-code/distributions</u>

       Matrices
           <u>/usr/share/conservation-code/matrix</u>

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Homepage &lt;<a href="http://compbio.cs.princeton.edu/conservation/">http://compbio.cs.princeton.edu/conservation/</a>&gt;
       Publication &lt;<a href="http://bioinformatics.oxfordjournals.org/cgi/content/full/23/15/1875">http://bioinformatics.oxfordjournals.org/cgi/content/full/23/15/1875</a>&gt;
       <b><a href="../man1/concavity.1.html">concavity</a></b>(1)

20110309.0                                         2019-12-15                       <u>DEBIAN/<a href="../man1/SCORE_CONSERVATION.1.html">SCORE_CONSERVATION</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>