<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>TIDDIT.py - structural variant calling</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/tiddit">tiddit_2.12.2+dfsg-1build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       TIDDIT.py - structural variant calling

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>TIDDIT.py</b> [--sv] [--cov]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       TIDDIT  is a tool to used to identify chromosomal rearrangements using Mate Pair or Paired End sequencing
       data. TIDDIT identifies intra and inter- chromosomal translocations, deletions,  tandem-duplications  and
       inversions, using supplementary alignments as well as discordant pairs.

       TIDDIT  has  two  analysis modules. The sv mode, which is used to search for structural variants. And the
       cov mode that analyse the read depth of a bam file and generates a coverage report.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>--sv</b>   call structural variation

       <b>--cov</b>  generate a coverage bed file

</pre><h4><b>AUTHOR</b></h4><pre>
        This manpage was written by Andreas Tille for the Debian distribution and
        can be used for any other usage of the program.

TIDDIT.py 2.11.0                                    May 2020                                        <u><a href="../man1/TIDDIT.PY.1.html">TIDDIT.PY</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>