<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Run::Phylo::LVB - Object for using the LVB program to create an array of Bio::Tree objects</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-perl-run-perl">libbio-perl-run-perl_1.7.3-8_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Run::Phylo::LVB - Object for using the LVB program to create an array of Bio::Tree objects
       from a nucleotide multiple alignment file or a nucleotide SimpleAlign object. Works with LVB version 2.1.

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::Tools::Run::Phylo::LVB;

         # Create a SimpleAlign object.
         # NOTE. Aligning nucleotide sequence directly, as below, makes
         # sense for non-coding nucleotide sequence (e.g., structural RNA
         # genes, introns, ITS). For protein-coding genes, to prevent
         # Clustal intronducing frameshifts one should instead align the
         # translations of the genes, then convert the multiple alignment
         # to nucleotide by referring to the corresponding transcript
         # sequences (e.g., using EMBOSS tranalign).
         use Bio::Tools::Run::Alignment::Clustalw;
         $aln_factory = Bio::Tools::Run::Alignment::Clustalw-&gt;new(quiet =&gt; 1);
         $inputfilename = "/Users/daniel/nuc.fa";
         $aln = $aln_factory-&gt;align($inputfilename);

         # Create the tree or trees.
         $tree_factory = Bio::Tools::Run::Phylo::LVB-&gt;new(quiet =&gt; 1);
         @trees = $tree_factory-&gt;run($aln);

         # Or one can pass in a file name containing a nucleotide multiple
         # alignment in Phylip 3.6 format:
         $tree_factory = Bio::Tools::Run::Phylo::LVB-&gt;new(quiet =&gt; 1);
         $tree = $tree_factory-&gt;run("/Users/daniel/nuc.phy");

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Wrapper for LVB, which uses a simulated annealing heuristic search to seek parsimonious trees from a
       nucleotide multiple alignment.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         <a href="http://redmine.open-bio.org/projects/bioperl/">http://redmine.open-bio.org/projects/bioperl/</a>

</pre><h4><b>PARAMETERS</b> <b>FOR</b> <b>LVB</b> <b>COMPUTATION</b></h4><pre>
   <b>FORMAT</b>
         Title       : FORMAT
         Description : (optional)
                       When running LVB from a Phylip 3.6-format
                       multiple alignment file, this specifies
                       the layout of the file. It may be
                       "interleaved" or "sequential". FORMAT is
                       automatically set to "interleaved" if
                       running from a SimpleAlign object.
                       Defaults to "interleaved".

   <b>GAPS</b>
         Title       : GAPS
         Description : (optional)
                       LVB can treat gaps represented in the
                       multiple alignment by "-" as either
                       "fifthstate" or "unknown". "fifthstate"
                       regards "-" as equivalent to "O", which
                       is an unambiguous character state
                       distinct from all nucleotides. "unknown"
                       regards "-" as equivalent to "?", which
                       is as an ambiguous site that may contain
                       "A" or "C" or "G" or "T" or "O".
                       Defaults to "unknown".

   <b>SEED</b>
         Title       : SEED
         Description : (optional)
                       This specifies the random number seed
                       for LVB. SEED must be an integer in the
                       range 0 to 900000000 inclusive. If no
                       seed is specified, LVB takes a seed from
                       the system clock. By default, no seed is
                       specified.

   <b>DURATION</b>
         Title       : DURATION
         Description : (optional)
                       This specifies the duration of the
                       analysis, which may be "fast" or "slow".
                       "slow" causes LVB to perform a more
                       thorough and more time-consuming search
                       than "fast". Defaults to "slow".

   <b>BOOTSTRAPS</b>
         Title       : BOOTSTRAPS
         Description : (optional)
                       This specifies the number of bootstrap
                       replicates to use, which must be a
                       positive integer. Set bootstraps to 0 for
                       no bootstrapping. Defaults to 0.

</pre><h4><b>AUTHOR</b></h4><pre>
       Daniel Barker

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
       Email jason-AT-bioperl_DOT_org

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>program_name</b>
        Title   : program_name
        Usage   : -&gt;program_name()
        Function: holds the program name
        Returns:  string
        Args    : None

   <b>program_dir</b>
        Title   : program_dir
        Usage   : -&gt;program_dir()
        Function: returns undef
        Args    :

   <b>run</b>
        Title   : run
        Usage   :
               $inputfilename = '/Users/daniel/nuc.phy';
               @trees = $factory-&gt;run($inputfilename);
        Function: Create one or more LVB trees from a SimpleAlign
                  object or a file containing a Phylip 3.6-format
                  nucleotide multiple alignment.
        Example :
        Returns : Array of L&lt;Bio::Tree&gt; objects
        Args    : Name of a file containing a nucleotide multiple
                  alignment in Phylip 3.6 format, or a SimpleAlign
                  object

   <b>create_tree</b>
        Title   : create_tree
        Usage   :
               $inputfilename = '/Users/daniel/nuc.phy';
               @trees = $factory-&gt;create_tree($inputfilename);
        Function: Create one or more LVB trees from a SimpleAlign
                  object or a file containing a Phylip 3.6-format
                  nucleotide multiple alignment.
        Example :
        Returns : Array of L&lt;Bio::Tree&gt; objects
        Args    : Name of a file containing a nucleotide multiple
                  alignment in Phylip 3.6 format, or a SimpleAlign
                  object

   <b>_run</b>
        Title   : _run
        Usage   : Internal function, not to be called directly
        Function:  makes actual system call to lvb program
        Example :
        Returns : Array of Bio::Tree objects
        Args    : Name of a file containing a multiple alignment
                  in Phylip 3.6 format and a parameter string to be
                  passed to LVB

   <b>_setinput</b>
        Title   :  _setinput
        Usage   :  Internal function, not to be called directly
        Function:   Create input file for lvb program
        Example :
        Returns : name of file containing a multiple alignment in
                  Phylip 3.6 format
        Args    : SimpleAlign object reference or input file name

   <b>_setparams</b>
        Title   :  _setparams
        Usage   :  Internal function, not to be called directly
        Function:   Create parameter inputs for lvb program
        Example :
        Returns : parameter string to be passed to LVB
        Args    : name of calling object

perl v5.34.0                                       2022-02-10                   <u>Bio::Tools::Run::Phylo::<a href="../man3pm/LVB.3pm.html">LVB</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>