<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Run::BEDTools - Run wrapper for the BEDTools suite of programs *BETA*</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-perl-run-perl">libbio-perl-run-perl_1.7.3-8_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Run::BEDTools - Run wrapper for the BEDTools suite of programs *BETA*

</pre><h4><b>SYNOPSIS</b></h4><pre>
        # use a BEDTools program
        $bedtools_fac = Bio::Tools::Run::BEDTools-&gt;new( -command =&gt; 'subtract' );
        $result_file = $bedtools_fac-&gt;run( -bed1 =&gt; 'genes.bed', -bed2 =&gt; 'mask.bed' );

        # if IO::Uncompress::Gunzip is available...
        $result_file = $bedtools_fac-&gt;run( -bed1 =&gt; 'genes.bed.gz', -bed2 =&gt; 'mask.bed.gz' );

        # be more strict
        $bedtools_fac-&gt;set_parameters( -strandedness =&gt; 1 );

        # and even more...
        $bedtools_fac-&gt;set_parameters( -minimum_overlap =&gt; 1e-6 );

        # create a Bio::SeqFeature::Collection object
        $features = $bedtools_fac-&gt;result( -want =&gt; 'Bio::SeqFeature::Collection' );

</pre><h4><b>DEPRECATION</b> <b>WARNING</b></h4><pre>
       Most executables from BEDTools v&gt;=2.10.1 can read GFF and VCF formats in addition to BED format. This
       requires the use of a new input file param, shown in the following documentation, '-bgv', in place of
       '-bed' for the executables that can do this.

       This behaviour breaks existing scripts.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This module provides a wrapper interface for Aaron R. Quinlan and Ira M. Hall's utilities "BEDTools" that
       allow for (among other things):

       •   Intersecting two BED files in search of overlapping features.

       •   Merging overlapping features.

       •   Screening for paired-end (PE) overlaps between PE sequences and existing genomic features.

       •   Calculating the depth and breadth of sequence coverage across defined "windows" in a genome.

       (see &lt;<a href="http://code.google.com/p/bedtools/">http://code.google.com/p/bedtools/</a>&gt; for manuals and downloads).

</pre><h4><b>OPTIONS</b></h4><pre>
       "BEDTools"  is  a  suite  of  17  commandline  executable.  This  module  attempts to provide and options
       comprehensively. You can browse the choices like so:

        $bedtools_fac = Bio::Tools::Run::BEDTools-&gt;new;

        # all bowtie commands
        @all_commands = $bedtools_fac-&gt;available_parameters('commands');
        @all_commands = $bedtools_fac-&gt;available_commands; # alias

        # just for default command ('bam_to_bed')
        @btb_params = $bedtools_fac-&gt;available_parameters('params');
        @btb_switches = $bedtools_fac-&gt;available_parameters('switches');
        @btb_all_options = $bedtools_fac-&gt;available_parameters();

       Reasonably mnemonic names have been assigned to the single-letter command line  options.  These  are  the
       names returned by "available_parameters", and can be used in the factory constructor like typical BioPerl
       named parameters.

       As  a  number  of  options are mutually exclusive, and the interpretation of intent is based on last-pass
       option reaching bowtie with potentially  unpredicted  results.  This  module  will  prevent  inconsistent
       switches and parameters from being passed.

       See &lt;<a href="http://code.google.com/p/bedtools/">http://code.google.com/p/bedtools/</a>&gt; for details of BEDTools options.

</pre><h4><b>FILES</b></h4><pre>
       When a command requires filenames, these are provided to the "run" method, not the constructor ("new()").
       To  see  the  set  of  files  required  by a command, use "available_parameters('filespec')" or the alias
       "filespec()":

         $bedtools_fac = Bio::Tools::Run::BEDTools-&gt;new( -command =&gt; 'pair_to_bed' );
         @filespec = $bedtools_fac-&gt;filespec;

       This example returns the following array:

        #bedpe
        #bam
        bed
        #out

       This indicates that the bed ("BEDTools" BED format) file MUST be  specified,  and  that  the  out,  bedpe
       ("BEDTools" BEDPE format) and bam ("SAM" binary format) file MAY be specified (Note that in this case you
       MUST  provide  ONE  of  bedpe  OR  bam,  the  module  at this stage does not allow this information to be
       queried). Use these in the "run" call like so:

        $bedtools_fac-&gt;run( -bedpe =&gt; 'paired.bedpe',
                            -bgv =&gt; 'genes.bed',
                            -out =&gt; 'overlap' );

       The object will store the programs STDERR output for you in the "stderr()" attribute:

        handle_bed_warning($bedtools_fac) if ($bedtools_fac-&gt;stderr =~ /Usage:/);

       For the commands  'fasta_from_bed'  and  'mask_fasta_from_bed'  STDOUT  will  also  be  captured  in  the
       "stdout()"  attribute by default and all other commands can be forced to capture program output in STDOUT
       by setting the -out filespec parameter to '-'.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your  comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>

       Rather  than  to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs  and  their  resolution.
       Bug reports can be submitted via the web:

         <a href="http://redmine.open-bio.org/projects/bioperl/">http://redmine.open-bio.org/projects/bioperl/</a>

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Dan</b> <b>Kortschak</b></h4><pre>
        Email dan.kortschak adelaide.edu.au

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
       Additional contributors names and emails here

</pre><h4><b>APPENDIX</b></h4><pre>
       The  rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>new()</b>
        Title   : new
        Usage   : my $obj = new Bio::Tools::Run::BEDTools();
        Function: Builds a new Bio::Tools::Run::BEDTools object
        Returns : an instance of Bio::Tools::Run::BEDTools
        Args    :

   <b>run()</b>
        Title   : run
        Usage   : $result = $bedtools_fac-&gt;run(%params);
        Function: Run a BEDTools command.
        Returns : Command results (file, IO object or Bio object)
        Args    : Dependent on filespec for command.
                  See $bedtools_fac-&gt;filespec and BEDTools Manual.
                  Also accepts -want =&gt; '(raw|format|&lt;object_class&gt;)' - see want().
        Note    : gzipped inputs are allowed if IO::Uncompress::Gunzip
                  is available

                  Command                      &lt;in&gt;                    &lt;out&gt;

                  annotate             bgv             ann(s)          #out

                  graph_union          bg_files                        #out

                  fasta_from_bed       seq             bgv             #out

                  mask_fasta_from_bed  seq             bgv             #out

                  bam_to_bed           bam                             #out

                  bed_to_IGV           bgv                             #out

                  merge                bgv                             #out

                  sort                 bgv                             #out

                  links                bgv                             #out

                  b12_to_b6            bed                             #out

                  overlap              bed                             #out

                  group_by             bed                             #out

                  bed_to_bam           bgv                             #out

                  shuffle              bgv             genome          #out

                  slop                 bgv             genome          #out

                  complement           bgv             genome          #out

                  genome_coverage      bed             genome          #out

                  window               bgv1            bgv2            #out

                  closest              bgv1            bgv2            #out

                  coverage             bgv1            bgv2            #out

                  subtract             bgv1            bgv2            #out

                  pair_to_pair         bedpe1          bedpe2          #out

                  intersect            bgv1|bam        bgv2            #out

                  pair_to_bed          bedpe|bam       bgv             #out

                  bgv* signifies any of BED, GFF or VCF. ann is a bgv.

                  NOTE: Replace 'bgv' with 'bed' unless $use_bgv is set.

   <b>want()</b>
        Title   : want
        Usage   : $bowtiefac-&gt;want( $class )
        Function: make factory return $class, or 'raw' results in file
                  or 'format' for result format
                  All commands can return Bio::Root::IO
                  commands returning:       can return object:
                  - BED or BEDPE            - Bio::SeqFeature::Collection
                  - sequence                - Bio::SeqIO
        Returns : return wanted type
        Args    : [optional] string indicating class or raw of wanted result

   <b>result()</b>
        Title   : result
        Usage   : $bedtoolsfac-&gt;result( [-want =&gt; $type|$format] )
        Function: return result in wanted format
        Returns : results
        Args    : [optional] hashref of wanted type
        Note    : -want arg does not persist between result() call when
                  specified in result(), for persistence, use want()

   <b>_determine_format()</b>
        Title   : _determine_format( $has_run )
        Usage   : $bedtools-fac-&gt;_determine_format
        Function: determine the format of output for current options
        Returns : format of bowtie output
        Args    : [optional] boolean to indicate result exists

   <b>_read_bed()</b>
        Title   : _read_bed()
        Usage   : $bedtools_fac-&gt;_read_bed
        Function: return a Bio::SeqFeature::Collection object from a BED file
        Returns : Bio::SeqFeature::Collection
        Args    :

   <b>_read_bedpe()</b>
        Title   : _read_bedpe()
        Usage   : $bedtools_fac-&gt;_read_bedpe
        Function: return a Bio::SeqFeature::Collection object from a BEDPE file
        Returns : Bio::SeqFeature::Collection
        Args    :

   <b>_validate_file_input()</b>
        Title   : _validate_file_input
        Usage   : $bedtools_fac-&gt;_validate_file_input( -type =&gt; $file )
        Function: validate file type for file spec
        Returns : file type if valid type for file spec
        Args    : hash of filespec =&gt; file_name

   <b>version()</b>
        Title   : version
        Usage   : $version = $bedtools_fac-&gt;version()
        Function: Returns the program version (if available)
        Returns : string representing location and version of the program

perl v5.34.0                                       2022-02-10                     <u>Bio::Tools::Run::<a href="../man3pm/BEDTools.3pm.html">BEDTools</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>