<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Run::MCS - Wrapper for MCS</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/jammy/+package/libbio-perl-run-perl">libbio-perl-run-perl_1.7.3-8_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Run::MCS - Wrapper for MCS

</pre><h4><b>SYNOPSIS</b></h4><pre>
         use Bio::Tools::Run::MCS;

         # Make a MCS factory
         $factory = Bio::Tools::Run::MCS-&gt;new();

         # Run MCS on an alignment
         my @results = $factory-&gt;run($alignfilename);

         # or with alignment object
         @results = $factory-&gt;run($bio_simplalign);

         # look at the results
         foreach my $feat (@results) {
           my $seq_id = $feat-&gt;seq_id;
           my $start = $feat-&gt;start;
           my $end = $feat-&gt;end;
           my $score = $feat-&gt;score;
           my ($pvalue) = $feat-&gt;get_tag_values('pvalue');
           my ($kind) = $feat-&gt;get_tag_values('kind'); # 'all', 'exon' or 'nonexon'
         }

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This is a wrapper for running the MCS (binCons) scripts by Elliott H Margulies.  You can get details
       here: <a href="http://zoo.nhgri.nih.gov/elliott/mcs_doc/">http://zoo.nhgri.nih.gov/elliott/mcs_doc/</a>. MCS is used for the prediciton of transcription factor
       binding sites and other regions of the genome conserved amongst different species.

       Note that this wrapper assumes you already have alignments, so only uses MCS for the latter stages (the
       stages involving align2binomial.pl, generate_phyloMAX_score.pl and generate_mcs_beta.pl).

       You can try supplying normal MCS command-line arguments to <b>new()</b>, eg.

         $factory-&gt;new(-percentile =&gt; 95)

       or calling arg-named methods (excluding the initial hyphens, eg.

         $factory-&gt;<a href="../man95/percentile.95.html">percentile</a>(95)

        to set the --percentile arg).

       You will need to enable this MCS wrapper to find the MCS scripts.  This can be done in (at least) three
       ways:

        1. Make sure the MCS scripts are in your path.
        2. Define an environmental variable MCSDIR which is a
           directory which contains the MCS scripts:
           In bash:

           export MCSDIR=/home/username/mcs/

           In csh/tcsh:

           setenv MCSDIR /home/username/mcs

        3. Include a definition of an environmental variable MCSDIR in
           every script that will use this MCS wrapper module, e.g.:

           BEGIN { $ENV{MCSDIR} = '/home/username/mcs/' }
           use Bio::Tools::Run::MCS;

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         <a href="http://redmine.open-bio.org/projects/bioperl/">http://redmine.open-bio.org/projects/bioperl/</a>

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Sendu</b> <b>Bala</b></h4><pre>
       Email <a href="mailto:bix@sendu.me.uk">bix@sendu.me.uk</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>program_name</b>
        Title   : program_name
        Usage   : $factory&gt;program_name()
        Function: holds the program name
        Returns : string
        Args    : None

   <b>program_dir</b>
        Title   : program_dir
        Usage   : $factory-&gt;program_dir(@params)
        Function: returns the program directory, obtained from ENV variable.
        Returns : string
        Args    : None

   <b>new</b>
        Title   : new
        Usage   : $factory = Bio::Tools::Run::MCS-&gt;new()
        Function: creates a new MCS factory
        Returns : Bio::Tools::Run::MCS
        Args    : Many options understood by MCS can be supplied as key =&gt;
                  value pairs.

                  These options can NOT be used with this wrapper:
                  ucsc gtf neutral-only fourd-align align-only ar

   <b>run</b>
        Title   : run
        Usage   : $result = $factory-&gt;run($align_file_or_object, Bio::Location::Atomic,
                                          [Bio::SeqFeatureI]);
        Function: Runs the MCS scripts on an alignment.
        Returns : list of Bio::SeqFeatureI feature objects (with coordinates corrected
                  according to the supplied offset, if any)
        Args    : The first argument represents an alignment, the optional second
                  argument represents the chromosome, stand and end and the optional
                  third argument represents annotation of the exons in the alignment.

                  The alignment can be provided as a multi-fasta format alignment
                  filename, or a Bio::Align::AlignI compliant object (eg. a
                  Bio::SimpleAlign).

                  The position in the genome can be provided as a Bio::Location::Atomic
                  with start, end and seq_id set.

                  The annnotation can be provided as an array of Bio::SeqFeatureI
                  objects.

   <b>_setparams</b>
        Title   : _setparams
        Usage   : Internal function, not to be called directly
        Function: Creates a string of params to be used in the command string
        Returns : string of params
        Args    : none

perl v5.34.0                                       2022-02-10                          <u>Bio::Tools::Run::<a href="../man3pm/MCS.3pm.html">MCS</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>