<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>nanook  -  flexible, multi-reference software for pre- and post-alignment analysis of nanopore sequencing</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/noble/+package/nanook">nanook_1.33+dfsg-5_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       nanook  -  flexible, multi-reference software for pre- and post-alignment analysis of nanopore sequencing
       data, quality and error profiles

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>nanook</b> <u>&lt;extract|align|analyse|compare|process&gt;</u> [options]

</pre><h4><b>DESCRIPTION</b></h4><pre>
        NanoOK (pronounced na-nook) is a tool for extraction, alignment and
        analysis of Nanopore reads. NanoOK will extract reads as FASTA or FASTQ
        files, align them (with a choice of alignment tools), then generate a
        comprehensive multi-page PDF report containing yield, accuracy and
        quality analysis. Along the way, it generates plain text files which can
        be used for further analysis, as well as graphs suitable for inclusion
        in presentations and papers.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>extract</b> <b>options:</b>

       <b>-s</b>|-sample &lt;dir&gt; specifies sample directory

       <b>-f</b>|-reads specifies alternative dir for FAST5 files (default fast5)

              Can be absolute (beginning with /) or relative e.g. <b>-f</b> reads/downloads  if  replicating  Metrichor
              file structure

       <b>-a</b>|-fasta specifies FASTA file extraction (default)

       <b>-q</b>|-fastq specifies FASTQ file extraction

       <b>-basecallindex</b> specifies the index of the analysis (default: latest)

       <b>-mergereads</b> to generate merged FASTA files in addition to single read files

       <b>-minquality</b> &lt;value&gt; to set the minimum quality for a 'pass' read

   <b>align</b> <b>options:</b>

       <b>-s</b>|-sample &lt;dir&gt; specifies sample directory

       <b>-r</b>|-reference &lt;path&gt; specifies path to reference database

       <b>-aligner</b> &lt;name&gt; specifies the aligner (default last)

       <b>-alignerparams</b> &lt;params&gt; specifies parameters to the aligner

       <b>-showaligns</b> echoes aligner commands to screen

   <b>analyse</b> <b>options:</b>

       <b>-s</b>|-sample &lt;dir&gt; specifies sample directory

       <b>-r</b>|-reference &lt;path&gt; specifies path to reference database

       <b>-aligner</b> &lt;name&gt; specifies the aligner (default last)

       <b>-coveragebin</b> &lt;int&gt; specifies coverage bin size (default 100)

       <b>-bitmaps</b> to output bitmap PNG graphs instead of PDF

   <b>compare</b> <b>options:</b>

       <b>-l</b>|-samplelist &lt;file&gt; specifies a sample list file

       <b>-o</b>|-outputdir &lt;directory&gt; specifies an output directory

       <b>-type</b> &lt;2d|template|complement&gt; specifies an output directory

   <b>process</b> <b>options:</b>

       <b>-process</b> &lt;file&gt; specifies a process file

   <b>Read</b> <b>type</b> <b>options:</b>

       <b>-passonly</b> to analyse only pass reads

       <b>-failonly</b> to analyse only fail reads

       <b>-2donly</b> to analyse only 2D reads

       <b>-templateonly</b> to analyse just Template reads

       <b>-complementonly</b> to analyse just Complement reads

   <b>Other</b> <b>options:</b>

       <b>-t</b>|-numthreads &lt;number&gt; specifies the number of threads to use (default 1)

       <b>-log</b> &lt;filename&gt; enables debug logging to file

       <b>-force</b> to force NanoOK to ignore warnings

       <b>-timeout</b> to set the number of seconds before giving up waiting for new reads (default 2)

       Valid aligners: last, bwa, blasr, marginalign, graphmap

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

nanook 1.26                                      September 2017                                        <u><a href="../man1/NANOOK.1.html">NANOOK</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>