<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>altree - Association and Localisation tests using  phylogenetic Trees</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/altree">altree_1.3.2-2build5_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       altree - Association and Localisation tests using  phylogenetic Trees

</pre><h4><b>SYNOPSIS</b></h4><pre>
       altree [options]

        Options:
           --version        program version
           --short-help|h   brief help message
           --help           help message with options descriptions
           --man            full documentation
           --association|a  perform the association test
           --s-localisation|l   perform the localisation using the S character
           --first-input-file|i result_file from phylogeny reconstruction programs
           --second-input-file|j file containing the nb of cases/controls carrying an haplotype
           --output-file|o output_file
           --data-type|t DNA|SNP
           --data-qual|q qualitative|quantitative
           --outgroup outgroup_name
           --remove-outgroup
           --tree-building-program|p phylip|paup|paml
           --splitmode|s nosplit|chi2split
           --no-prolongation
           --chi2-threshold|n value
           --permutations|r number
           --number-of-trees-to-analyse number
           --tree-to-analyse number
           --s-site-number number
           --s-site-characters ancestral state -&gt; derived state
           --co-evo|e simple|double
           --print-tree
           --anc-seq ancestral sequence (only with phylip)
           --nb-files number of input files to analyse (only for association test)

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>--version</b>
               Print the program version and exits.

       <b>--short-help</b>
               Print a brief help message and exits.

       <b>--help</b>  Print a help message with options descriptions and exits.

       <b>--man</b>   Prints the manual page and exits.

       <b>--association|a</b>
               Perform the association test

       <b>--s-localisation|l</b>
               Localise the susceptibility locus using the "S-character method"

       <b>--first-input-file|i</b> <u>result_file</u>
               Input  file  1  (paup,  phylip or paml results file).  If several input files are analysed, their
               names must be separated by colons.  Example: input1:input2 etc

       <b>--second-input-file|j</b> <u>correspond_file</u>
               Input file 2, default <u>correspond.txt</u>.  The number of input file 2 must be the same as the  number
               of input file 1.  The name of the different input file 2 must be separated by colons

       <b>--output-file|o</b> <u>outfile</u>
               Output file

       <b>--data-type|t</b> "DNA"|"SNP"
               Type of data: DNA (ATGCU) or SNP (0-1)

       <b>--data-qual|q</b> "qualitative"|"quantitative"
               Analyse qualitative (case/control) or quantitative data

       <b>--outgroup</b>  outgroup
               Root the tree with this outgroup

       <b>--remove-outgroup</b>
               Remove the outgroup of the tree before performing the tests

       <b>--tree-building-program|p</b> "phylip"|"paup"|"paml"
               Phylogeny reconstruction program

       <b>--splitmode|s</b> "nosplit"|"chi2split"
               how tests are performed from a level to another

       <b>--no-prolongation</b>
               No prolongation of branches in the tree

       <b>--chi2-threshold|n</b> value
               Significance threshold for chi2 (default value 0.01)

       <b>--permutations|r</b> number
               Number of permutations used to calculate exact p_values (Only for association test)

       <b>--number-of-trees-to-analyse</b> number
               Number  of  trees  to  analyse  in  the localisation analysis (only for localisation method using
               S-character)

       <b>--tree-to-analyse</b> number
               With this option, you can specify the tree to use (instead of random). Can be used several  times
               to specify multiple trees.

       <b>--s-site-number</b> number
               Number of the S character site in the sequence (only for localisation method using S-character)

       <b>--s-site-characters</b> transition
               Character  states  for  the S character: ancestral state -&gt; derived state ex: G-&gt;C or 0-&gt;1  (only
               for localisation method using S-character)

       <b>--co-evo|e</b> "simple"|"double"
               Type of co-evolution indice
                 simple: only the anc -&gt; der transition of S is used
                 double: the two possible transitions are used

       <b>--print-tree</b>
               If this option is selected, the tree will be printed to the output

       <b>--anc-seq</b> anc_seq
               With this option, you can specify the ancestral sequence.  This option is only  useful  when  the
               tree  is reconstructed using the mix program of phylip with the ancestral states specified in the
               file "ancestors"

       <b>--nb-files</b> number
               With this option, you specify the number of input files (1 and 2)  to analyse  This  option  only
               works  for  the  association  test.  Be  careful  if  the number of trees is not the same for the
               different input files: if the chosen tree doesn't exist in one file, the  program  wil  not  work
               correctly

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>This</b> <b>program</b> performs

       (a) an association test between a candidate gene and disease or a quantitative trait

       (b)  a  localsation tests: it allows to detect which SNP is involved in the determinism of the disease or
       the quantitative trait

       These two tests are based on the analysis of haplotype phylogenetic trees.

perl v5.40.0                                       2024-10-20                                         <u><a href="../man1p/ALTREE.1p.html">ALTREE</a></u>(1p)
</pre>
 </div>
</div></section>
</div>
</body>
</html>