<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>fastp - Ultra-fast all-in-one FASTQ preprocessor</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/fastp">fastp_0.24.0+dfsg-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       fastp - Ultra-fast all-in-one FASTQ preprocessor

</pre><h4><b>DESCRIPTION</b></h4><pre>
       fastp: an ultra-fast all-in-one FASTQ preprocessor version 0.23.0 usage: fastp [options] ...  options:

       <b>-i</b>, <b>--in1</b>
              read1 input file name (string [=])

       <b>-o</b>, <b>--out1</b>
              read1 output file name (string [=])

       <b>-I</b>, <b>--in2</b>
              read2 input file name (string [=])

       <b>-O</b>, <b>--out2</b>
              read2 output file name (string [=])

       <b>--unpaired1</b>
              for  PE  input,  if  read1 passed QC but read2 not, it will be written to unpaired1. Default is to
              discard it. (string [=])

       <b>--unpaired2</b>
              for PE input, if read2 passed QC but read1 not, it will be written to unpaired2. If <b>--unpaired2</b> is
              same as <b>--unpaired1</b> (default mode), both unpaired reads will be written to this same file. (string
              [=])

       <b>--overlapped_out</b>
              for each read pair, output the overlapped region if it has no any mismatched base. (string [=])

       <b>--failed_out</b>
              specify the file to store reads that cannot pass the filters. (string [=])

       <b>-m</b>, <b>--merge</b>
              for paired-end input, merge each pair of reads into a single read  if  they  are  overlapped.  The
              merged reads will be written to the file given by <b>--merged_out</b>, the unmerged reads will be written
              to the files specified by <b>--out1</b> and <b>--out2</b>. The merging mode is disabled by default.

       <b>--merged_out</b>
              in  the  merging mode, specify the file name to store merged output, or specify <b>--stdout</b> to stream
              the merged output (string [=])

       <b>--include_unmerged</b>
              in the merging mode, write the unmerged or unpaired  reads  to  the  file  specified  by  <b>--merge</b>.
              Disabled by default.

       <b>-6</b>, <b>--phred64</b>
              indicate  the  input  is  using phred64 scoring (it'll be converted to phred33, so the output will
              still be phred33)

       <b>-z</b>, <b>--compression</b>
              compression level for gzip output (1 ~ 9). 1 is fastest, 9 is smallest, default is 4. (int [=4])

       <b>--stdin</b>
              input from STDIN. If the STDIN is interleaved paired-end FASTQ, please also add <b>--interleaved_in</b>.

       <b>--stdout</b>
              stream passing-filters reads to STDOUT. This option will result in interleaved  FASTQ  output  for
              paired-end output. Disabled by default.

       <b>--interleaved_in</b>
              indicate  that  &lt;in1&gt;  is  an  interleaved  FASTQ which contains both read1 and read2. Disabled by
              default.

       <b>--reads_to_process</b>
              specify how many reads/pairs to be processed. Default 0 means process all reads. (int [=0])

       <b>--dont_overwrite</b>
              don't overwrite existing files. Overwritting is allowed by default.

       <b>--fix_mgi_id</b>
              the MGI FASTQ ID format is not compatible with many BAM operation tools, enable this option to fix
              it.

       <b>-V</b>, <b>--verbose</b>
              output verbose log information (i.e. when every 1M reads are processed).

       <b>-A</b>, <b>--disable_adapter_trimming</b>
              adapter trimming is enabled by default. If this option is specified, adapter trimming is disabled

       <b>-a</b>, <b>--adapter_sequence</b>
              the adapter for read1. For SE data, if not specified, the adapter will be  auto-detected.  For  PE
              data, this is used if R1/R2 are found not overlapped. (string [=auto])

       <b>--adapter_sequence_r2</b>
              the  adapter  for  read2  (PE  data  only). This is used if R1/R2 are found not overlapped. If not
              specified, it will be the same as &lt;adapter_sequence&gt; (string [=auto])

       <b>--adapter_fasta</b>
              specify a FASTA file to trim both read1 and read2 (if PE) by all the sequences in this FASTA  file
              (string [=])

       <b>--detect_adapter_for_pe</b>
              by  default,  the  auto-detection  for  adapter  is for SE data input only, turn on this option to
              enable it for PE data.

       <b>-f</b>, <b>--trim_front1</b>
              trimming how many bases in front for read1, default is 0 (int [=0])

       <b>-t</b>, <b>--trim_tail1</b>
              trimming how many bases in tail for read1, default is 0 (int [=0])

       <b>-b</b>, <b>--max_len1</b>
              if read1 is longer than max_len1, then trim read1 at its tail to make  it  as  long  as  max_len1.
              Default 0 means no limitation (int [=0])

       <b>-F</b>, <b>--trim_front2</b>
              trimming how many bases in front for read2. If it's not specified, it will follow read1's settings
              (int [=0])

       <b>-T</b>, <b>--trim_tail2</b>
              trimming  how many bases in tail for read2. If it's not specified, it will follow read1's settings
              (int [=0])

       <b>-B</b>, <b>--max_len2</b>
              if read2 is longer than max_len2, then trim read2 at its tail to make  it  as  long  as  max_len2.
              Default 0 means no limitation. If it's not specified, it will follow read1's settings (int [=0])

       <b>-D</b>, <b>--dedup</b>
              enable deduplication to drop the duplicated reads/pairs

       <b>--dup_calc_accuracy</b>
              accuracy level to calculate duplication (1~6), higher level uses more memory (1G, 2G, 4G, 8G, 16G,
              24G). Default 1 for no-dedup mode, and 3 for dedup mode. (int [=0])

       <b>--dont_eval_duplication</b>
              don't evaluate duplication rate to save time and use less memory.

       <b>-g</b>, <b>--trim_poly_g</b>
              force   polyG   tail   trimming,  by  default  trimming  is  automatically  enabled  for  Illumina
              NextSeq/NovaSeq data

       <b>--poly_g_min_len</b>
              the minimum length to detect polyG in the read tail. 10 by default. (int [=10])

       <b>-G</b>, <b>--disable_trim_poly_g</b>
              disable  polyG  tail  trimming,  by  default  trimming  is  automatically  enabled  for   Illumina
              NextSeq/NovaSeq data

       <b>-x</b>, <b>--trim_poly_x</b>
              enable polyX trimming in 3' ends.

       <b>--poly_x_min_len</b>
              the minimum length to detect polyX in the read tail. 10 by default. (int [=10])

       <b>-5</b>, <b>--cut_front</b>
              move  a sliding window from front (5') to tail, drop the bases in the window if its mean quality &lt;
              threshold, stop otherwise.

       <b>-3</b>, <b>--cut_tail</b>
              move a sliding window from tail (3') to front, drop the bases in the window if its mean quality  &lt;
              threshold, stop otherwise.

       <b>-r</b>, <b>--cut_right</b>
              move  a  sliding window from front to tail, if meet one window with mean quality &lt; threshold, drop
              the bases in the window and the right part, and then stop.

       <b>-W</b>, <b>--cut_window_size</b>
              the window size option shared by cut_front, cut_tail or cut_sliding.  Range:  1~1000,  default:  4
              (int [=4])

       <b>-M</b>, <b>--cut_mean_quality</b>
              the  mean  quality  requirement  option  shared by cut_front, cut_tail or cut_sliding. Range: 1~36
              default: 20 (Q20) (int [=20])

       <b>--cut_front_window_size</b>
              the window size option of cut_front, default to cut_window_size if not specified (int [=4])

       <b>--cut_front_mean_quality</b>
              the mean quality requirement option for cut_front, default to cut_mean_quality  if  not  specified
              (int [=20])

       <b>--cut_tail_window_size</b>
              the window size option of cut_tail, default to cut_window_size if not specified (int [=4])

       <b>--cut_tail_mean_quality</b>
              the  mean  quality  requirement  option for cut_tail, default to cut_mean_quality if not specified
              (int [=20])

       <b>--cut_right_window_size</b>
              the window size option of cut_right, default to cut_window_size if not specified (int [=4])

       <b>--cut_right_mean_quality</b>
              the mean quality requirement option for cut_right, default to cut_mean_quality  if  not  specified
              (int [=20])

       <b>-Q</b>, <b>--disable_quality_filtering</b>
              quality  filtering  is  enabled  by  default.  If  this  option is specified, quality filtering is
              disabled

       <b>-q</b>, <b>--qualified_quality_phred</b>
              the quality value that a base is qualified. Default 15 means phred  quality  &gt;=Q15  is  qualified.
              (int [=15])

       <b>-u</b>, <b>--unqualified_percent_limit</b>
              how many percents of bases are allowed to be unqualified (0~100). Default 40 means 40% (int [=40])

       <b>-n</b>, <b>--n_base_limit</b>
              if  one  read's  number of N base is &gt;n_base_limit, then this read/pair is discarded. Default is 5
              (int [=5])

       <b>-e</b>, <b>--average_qual</b>
              if one read's average quality score &lt;avg_qual, then this read/pair is discarded. Default  0  means
              no requirement (int [=0])

       <b>-L</b>, <b>--disable_length_filtering</b>
              length filtering is enabled by default. If this option is specified, length filtering is disabled

       <b>-l</b>, <b>--length_required</b>
              reads shorter than length_required will be discarded, default is 15. (int [=15])

       <b>--length_limit</b>
              reads longer than length_limit will be discarded, default 0 means no limitation. (int [=0])

       <b>-y</b>, <b>--low_complexity_filter</b>
              enable  low  complexity  filter.  The  complexity  is  defined  as  the percentage of base that is
              different from its next base (base[i] != base[i+1]).

       <b>-Y</b>, <b>--complexity_threshold</b>
              the threshold for low complexity filter (0~100). Default is 30,  which  means  30%  complexity  is
              required. (int [=30])

       <b>--filter_by_index1</b>
              specify  a  file  contains  a  list of barcodes of index1 to be filtered out, one barcode per line
              (string [=])

       <b>--filter_by_index2</b>
              specify a file contains a list of barcodes of index2 to be filtered  out,  one  barcode  per  line
              (string [=])

       <b>--filter_by_index_threshold</b>
              the allowed difference of index barcode for index filtering, default 0 means completely identical.
              (int [=0])

       <b>-c</b>, <b>--correction</b>
              enable base correction in overlapped regions (only for PE data), default is disabled

       <b>--overlap_len_require</b>
              the  minimum  length  to  detect  overlapped region of PE reads. This will affect overlap analysis
              based PE merge, adapter trimming and correction. 30 by default. (int [=30])

       <b>--overlap_diff_limit</b>
              the maximum number of mismatched bases to detect overlapped region of PE reads. This  will  affect
              overlap analysis based PE merge, adapter trimming and correction. 5 by default. (int [=5])

       <b>--overlap_diff_percent_limit</b>
              the  maximum  percentage  of  mismatched  bases to detect overlapped region of PE reads. This will
              affect overlap analysis based PE merge, adapter trimming and correction.  Default  20  means  20%.
              (int [=20])

       <b>-U</b>, <b>--umi</b>
              enable unique molecular identifier (UMI) preprocessing

       <b>--umi_loc</b>
              specify the location of UMI, can be (index1/index2/read1/read2/per_index/per_read, default is none
              (string [=])

       <b>--umi_len</b>
              if the UMI is in read1/read2, its length should be provided (int [=0])

       <b>--umi_prefix</b>
              if  specified,  an  underline will be used to connect prefix and UMI (i.e. prefix=UMI, UMI=AATTCG,
              final=UMI_AATTCG). No prefix by default (string [=])

       <b>--umi_skip</b>
              if the UMI is in read1/read2, fastp can skip several bases following UMI, default is 0 (int [=0])

       <b>-p</b>, <b>--overrepresentation_analysis</b>
              enable overrepresented sequence analysis.

       <b>-P</b>, <b>--overrepresentation_sampling</b>
              one in (<b>--overrepresentation_sampling</b>) reads will  be  computed  for  overrepresentation  analysis
              (1~10000), smaller is slower, default is 20. (int [=20])

       <b>-j</b>, <b>--json</b>
              the json format report file name (string [=fastp.json])

       <b>-h</b>, <b>--html</b>
              the html format report file name (string [=fastp.html])

       <b>-R</b>, <b>--report_title</b>
              should be quoted with ' or ", default is "fastp report" (string [=fastp report])

       <b>-w</b>, <b>--thread</b>
              worker thread number, default is 3 (int [=3])

       <b>-s</b>, <b>--split</b>
              split  output  by  limiting  total split file number with this option (2~999), a sequential number
              prefix will be added to output name ( 0001.out.fq, 0002.out.fq...), disabled by default (int [=0])

       <b>-S</b>, <b>--split_by_lines</b>
              split output by limiting lines of each file with this option(&gt;=1000), a sequential  number  prefix
              will be added to output name ( 0001.out.fq, 0002.out.fq...), disabled by default (long [=0])

       <b>-d</b>, <b>--split_prefix_digits</b>
              the  digits for the sequential number padding (1~10), default is 4, so the filename will be padded
              as 0001.xxx, 0 to disable padding (int [=4])

       <b>--cut_by_quality5</b>
              DEPRECATED, use <b>--cut_front</b> instead.

       <b>--cut_by_quality3</b>
              DEPRECATED, use <b>--cut_tail</b> instead.

       <b>--cut_by_quality_aggressive</b>
              DEPRECATED, use <b>--cut_right</b> instead.

       <b>--discard_unmerged</b>
              DEPRECATED, no effect now, see the introduction for merging.

       -?, <b>--help</b>
              print this message

</pre><h4><b>AUTHOR</b></h4><pre>
        This manpage was written by Nilesh Patra for the Debian distribution and
        can be used for any other usage of the program.

fastp 0.23.0                                      October 2021                                          <u><a href="../man1/FASTP.1.html">FASTP</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>