<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>razers - Fast Read Mapping with Sensitivity Control</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/seqan-apps">seqan-apps_2.4.0+dfsg-16_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       razers - Fast Read Mapping with Sensitivity Control

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>razers</b> [<u>OPTIONS</u>] &lt;<u>GENOME</u> <u>FILE</u>&gt; &lt;<u>READS</u> <u>FILE</u>&gt;
       <b>razers</b> [<u>OPTIONS</u>] &lt;<u>GENOME</u> <u>FILE</u>&gt; &lt;<u>MP-READS</u> <u>FILE1</u>&gt; &lt;<u>MP-READS</u> <u>FILE2</u>&gt;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       RazerS is a versatile full-sensitive read mapper based on a k-mer counting filter. It supports single and
       paired-end  mapping,  and  optimally parametrizes the filter based on a user-defined minimal sensitivity.
       See <u><a href="http://www.seqan.de/projects/razers">http://www.seqan.de/projects/razers</a></u> for more information.

       Input to RazerS is a reference genome file and either  one  file  with  single-end  reads  or  two  files
       containing left or right mates of paired-end reads. Use - to read single-end reads from stdin.

       (c) Copyright 2009 by David Weese.

</pre><h4><b>REQUIRED</b> <b>ARGUMENTS</b></h4><pre>
       <b>ARGUMENT</b> <b>0</b> <u>INPUT_FILE</u>
              A  reference  genome  file.  Valid filetypes are: <u>.sam[.*]</u>, <u>.raw[.*]</u>, <u>.gbk[.*]</u>, <u>.frn[.*]</u>, <u>.fq[.*]</u>,
              <u>.fna[.*]</u>, <u>.ffn[.*]</u>, <u>.fastq[.*]</u>, <u>.fasta[.*]</u>, <u>.faa[.*]</u>, <u>.fa[.*]</u>, <u>.embl[.*]</u>, and <u>.bam</u>, where * is any
              of the following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent (de)compression.

       <b>READS</b> List of <u>INPUT_FILE</u>'s
              Either one (single-end) or two (paired-end) read files. Valid filetypes are:  <u>.sam[.*]</u>,  <u>.raw[.*]</u>,
              <u>.gbk[.*]</u>,  <u>.frn[.*]</u>,  <u>.fq[.*]</u>,  <u>.fna[.*]</u>,  <u>.ffn[.*]</u>,  <u>.fastq[.*]</u>,  <u>.fasta[.*]</u>,  <u>.faa[.*]</u>, <u>.fa[.*]</u>,
              <u>.embl[.*]</u>, and <u>.bam</u>, where * is any of the following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent
              (de)compression.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              Display the help message.

       <b>--version</b>
              Display version information.

   <b>Main</b> <b>Options:</b>
       <b>-f</b>, <b>--forward</b>
              Map reads only to forward strands.

       <b>-r</b>, <b>--reverse</b>
              Map reads only to reverse strands.

       <b>-i</b>, <b>--percent-identity</b> <u>DOUBLE</u>
              Percent identity threshold. In range [50..100]. Default: <u>92</u>.

       <b>-rr</b>, <b>--recognition-rate</b> <u>DOUBLE</u>
              Percent recognition rate. In range [80..100]. Default: <u>99</u>.

       <b>-pd</b>, <b>--param-dir</b> <u>STRING</u>
              Read user-computed parameter files in the directory &lt;<u>DIR</u>&gt;.

       <b>-id</b>, <b>--indels</b>
              Allow indels. Default: mismatches only.

       <b>-ll</b>, <b>--library-length</b> <u>INTEGER</u>
              Paired-end library length. In range [1..inf]. Default: <u>220</u>.

       <b>-le</b>, <b>--library-error</b> <u>INTEGER</u>
              Paired-end library length tolerance. In range [0..inf]. Default: <u>50</u>.

       <b>-m</b>, <b>--max-hits</b> <u>INTEGER</u>
              Output only &lt;<u>NUM</u>&gt; of the best hits. In range [1..inf]. Default: <u>100</u>.

       <b>--unique</b>
              Output only unique best matches (-m 1 -dr 0 -pa).

       <b>-tr</b>, <b>--trim-reads</b> <u>INTEGER</u>
              Trim reads to given length. Default: off. In range [14..inf].

       <b>-o</b>, <b>--output</b> <u>OUTPUT_FILE</u>
              Change output filename (use - to dump to stdout in razers format). Default:  &lt;<u>READS</u>  <u>FILE</u>&gt;.razers.
              Valid filetypes are: <u>.razers</u>, <u>.gff</u>, <u>.fasta</u>, <u>.fa</u>, and <u>.eland</u>.

       <b>-v</b>, <b>--verbose</b>
              Verbose mode.

       <b>-vv</b>, <b>--vverbose</b>
              Very verbose mode.

   <b>Output</b> <b>Format</b> <b>Options:</b>
       <b>-a</b>, <b>--alignment</b>
              Dump the alignment for each match (only <u>razer</u> or <u>fasta</u> format).

       <b>-pa</b>, <b>--purge-ambiguous</b>
              Purge reads with more than &lt;<u>max-hits</u>&gt; best matches.

       <b>-dr</b>, <b>--distance-range</b> <u>INTEGER</u>
              Only consider matches with at most NUM more errors compared to the best. Default: output all.

       <b>-gn</b>, <b>--genome-naming</b> <u>INTEGER</u>
              Select how genomes are named (see Naming section below). In range [0..1]. Default: <u>0</u>.

       <b>-rn</b>, <b>--read-naming</b> <u>INTEGER</u>
              Select how reads are named (see Naming section below). In range [0..2]. Default: <u>0</u>.

       <b>-so</b>, <b>--sort-order</b> <u>INTEGER</u>
              Select how matches are sorted (see Sorting section below). In range [0..1]. Default: <u>0</u>.

       <b>-pf</b>, <b>--position-format</b> <u>INTEGER</u>
              Select begin/end position numbering (see Coordinate section below). In range [0..1]. Default: <u>0</u>.

   <b>Filtration</b> <b>Options:</b>
       <b>-s</b>, <b>--shape</b> <u>STRING</u>
              Manually set k-mer shape. Default: <u>11111111111</u>.

       <b>-t</b>, <b>--threshold</b> <u>INTEGER</u>
              Manually set minimum k-mer count threshold. In range [1..inf].

       <b>-oc</b>, <b>--overabundance-cut</b> <u>INTEGER</u>
              Set k-mer overabundance cut ratio. In range [0..1].

       <b>-rl</b>, <b>--repeat-length</b> <u>INTEGER</u>
              Skip simple-repeats of length &lt;<u>NUM</u>&gt;. In range [1..inf]. Default: <u>1000</u>.

       <b>-tl</b>, <b>--taboo-length</b> <u>INTEGER</u>
              Set taboo length. In range [1..inf]. Default: <u>1</u>.

       <b>-lm</b>, <b>--low-memory</b>
              Decrease memory usage at the expense of runtime.

   <b>Verification</b> <b>Options:</b>
       <b>-mN</b>, <b>--match-N</b>
              N matches all other characters. Default: N matches nothing.

       <b>-ed</b>, <b>--error-distr</b> <u>STRING</u>
              Write error distribution to <u>FILE</u>.

       <b>-mcl</b>, <b>--min-clipped-len</b> <u>INTEGER</u>
              Set minimal read length for read clipping. In range [0..inf]. Default: <u>0</u>.

       <b>-qih</b>, <b>--quality-in-header</b>
              Quality string in fasta header.

</pre><h4><b>FORMATS,</b> <b>NAMING,</b> <b>SORTING,</b> <b>AND</b> <b>COORDINATE</b> <b>SCHEMES</b></h4><pre>
       RazerS  supports  various  output formats. The output format is detected automatically from the file name
       suffix.

       .razers
              Razer format

       .fa, .fasta
              Enhanced Fasta format

       .eland Eland format

       .gff   GFF format

              By default, reads and contigs are referred by their Fasta ids given in the input files.  With  the
              <b>-gn</b> and <b>-rn</b> options this behaviour can be changed:

       0      Use Fasta id.

       1      Enumerate beginning with 1.

       2      Use the read sequence (only for short reads!).

              The way matches are sorted in the output file can be changed with the <b>-so</b> option for the following
              formats: <b>razer</b>, <b>fasta</b>, <b>sam</b>, and <b>amos</b>. Primary and secondary sort keys are:

       0      1. read number, 2. genome position

       1      1. genome position, 2. read number

              The  coordinate  space used for begin and end positions can be changed with the <b>-pf</b> option for the
              <b>razer</b> and <b>fasta</b> formats:

       0      Gap space. Gaps between characters are counted from 0.

       1      Position space. Characters are counted from 1.

</pre><h4><b>EXAMPLES</b></h4><pre>
       <b>razers</b> <b>example/genome.fa</b> <b>example/reads.fa</b> <b>-id</b> <b>-a</b> <b>-mN</b> <b>-v</b>
              Map single-end reads with 4% error rate, indels, and output the alignments. Ns are  considered  to
              match everything.

       <b>razers</b> <b>example/genome.fa</b> <b>example/reads.fa</b> <b>example/reads2.fa</b> <b>-id</b> <b>-mN</b>
              Map  paired-end  reads  with  up to 4% errors, indels, and output concordantly mapped pairs within
              default library size. Ns are considered to match everything.

razers 1.5.8 [tarball]                                                                                 <u><a href="../man1/RAZERS.1.html">RAZERS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>