<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Trinity - RNA-Seq De novo Assembly</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/trinityrnaseq">trinityrnaseq_2.15.2+dfsg-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Trinity - RNA-Seq De novo Assembly

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Trinity  represents  a novel method for the efficient and robust de novo reconstruction of transcriptomes
       from RNA-seq data.  Trinity  combines  three  independent  software  modules:  Inchworm,  Chrysalis,  and
       Butterfly,  applied  sequentially  to  process  large  volumes  of  RNA-seq reads. Trinity partitions the
       sequence data into many individual de Bruijn graphs, each representing the transcriptional complexity  at
       a  given  gene  or  locus,  and  then  processes each graph independently to extract full-length splicing
       isoforms and to tease apart transcripts derived from paralogous genes.

</pre><h4><b>OPTIONS</b></h4><pre>
       Required:

              <b>--seqType</b> &lt;string&gt; type of reads: ( fa, or fq )

              <b>--max_memory</b> &lt;string&gt; suggested max memory to use  by  Trinity  where  limiting  can  be  enabled.
              (jellyfish, sorting, etc) provied in Gb of RAM, ie.  '--max_memory 10G'

       If paired reads:

              <b>--left</b>  &lt;string&gt; left reads, one or more (separated by space)

              <b>--right</b> &lt;string&gt; right reads, one or more (separated by space)

       Or, if unpaired reads:

              <b>--single</b>  &lt;string&gt;  single  reads, one or more (note, if single file contains pairs, can use flag:
              <b>--run_as_paired</b> )

       Misc:

              <b>--SS_lib_type</b> &lt;string&gt; Strand-specific RNA-Seq read orientation.  if paired: RF or FR, if  single:
              F or R.   (dUTP method = RF) See web documentation.

              <b>--CPU</b> &lt;int&gt; number of CPUs to use, default: 2

              <b>--min_contig_length</b> &lt;int&gt; minimum assembled contig length to report (def=200)

              <b>--long_reads</b>  &lt;string&gt;  fasta  file containing error-corrected or circular consensus (CCS) pac bio
              reads

              <b>--genome_guided_bam</b> &lt;string&gt; genome guided mode, provide path to coordinate-sorted bam file.  (see
              genome-guided param section under <b>--show_full_usage_info</b>)

              <b>--jaccard_clip</b> option, set if you have paired reads and you expect  high  gene  density  with  UTR
              overlap  (use  FASTQ input file format for reads).  (note: jaccard_clip is an expensive operation,
              so avoid using it unless necessary due to finding excessive fusion transcripts w/o it.)

              <b>--trimmomatic</b> run Trimmomatic to quality trim reads  see  '--quality_trimming_params'  under  full
              usage info for tailored settings.

              <b>--normalize_reads</b> run in silico normalization of reads. Defaults to max. read coverage of 50.  see
              '--normalize_max_read_cov' under full usage info for tailored settings.

              <b>--no_distributed_trinity_exec</b> do not run Trinity phase 2 (assembly of partitioned reads), and stop
              after generating command list.

              <b>--output</b>  &lt;string&gt;  name  of  directory  for  output (will be created if it doesn't already exist)
              default(your current working directory)

              <b>--full_cleanup</b> only retain the Trinity fasta file, rename as ${output_dir}.Trinity.fasta

              <b>--cite</b> show the Trinity literature citation

              <b>--version</b> reports Trinity version (Trinity_v2.0.2) and exits.

              <b>--show_full_usage_info</b> show the many many more  options  available  for  running  Trinity  (expert
              usage).

</pre><h4><b>EXAMPLES</b></h4><pre>
       A typical Trinity command might be:

              Trinity <b>--seqType</b> fq <b>--max_memory</b> 50G <b>--left</b> reads_1.fq  <b>--right</b> reads_2.fq <b>--CPU</b> 6

       and for Genome-guided Trinity:

              Trinity <b>--genome_guided_bam</b> rnaseq_alignments.csorted.bam <b>--max_memory</b> 50G
                      <b>--genome_guided_max_intron</b> 10000 <b>--CPU</b> 6

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       see: /usr/lib/trinityrnaseq/sample_data/test_Trinity_Assembly/ for sample data and 'runMe.sh' for example
       Trinity execution

       For more details, visit: <a href="http://trinityrnaseq.github.io">http://trinityrnaseq.github.io</a>

Trinity version: Trinity_v2.0.2                   February 2015                                       <u><a href="../man1/TRINITY.1.html">TRINITY</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>