<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Cluster::UniGene - UniGene object</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-cluster-perl">libbio-cluster-perl_1.7.3-6_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Cluster::UniGene - UniGene object

</pre><h4><b>SYNOPSIS</b></h4><pre>
               use Bio::Cluster::UniGene;
               use Bio::ClusterIO;

               $stream  = Bio::ClusterIO-&gt;new('-file' =&gt; "Hs.data",
                                              '-format' =&gt; "unigene");
               # note: we quote -format to keep older perl's from complaining.

               while ( my $in = $stream-&gt;next_cluster() ) {
                       print $in-&gt;unigene_id() . "\n";
                       while ( my $sequence = $in-&gt;next_seq() ) {
                               print $sequence-&gt;accession_number() . "\n";
                       }
              }

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This UniGene object implements the Bio::Cluster::UniGeneI interface for the representation if UniGene
       clusters in Bioperl. It is returned by the Bio::ClusterIO parser for unigene format and contains all the
       data associated with one UniGene record.

       This class implements several interfaces and hence can be used wherever instances of such interfaces are
       expected. In particular, the interfaces are Bio::ClusterI as the base interface for all cluster
       representations, and in addition Bio::IdentifiableI and Bio::DescribableI.

       The following lists the UniGene specific methods that are available (see below for details). Be aware
       next_XXX iterators take a snapshot of the array property when they are first called, and this snapshot is
       not reset until the iterator is exhausted. Hence, once called you need to exhaust the iterator to see any
       changes that have been made to the property in the meantime. You will usually want to use the non-
       iterator equivalents and loop over the elements yourself.

       <b>new()</b> - standard new call

       <b>unigene_id()</b> - set/get unigene_id

       <b>title()</b> - set/get title (description)

       <b>gene()</b> - set/get gene

       <b>cytoband()</b> - set/get cytoband

       <b>mgi()</b> - set/get mgi

       <b>locuslink()</b> - set/get locuslink

       <b>homol()</b> - set/get homologene

       <b>gnm_terminus()</b> - set/get gnm_terminus

       <b>scount()</b> - set/get scount

       <b>express()</b> - set/get express, currently takes/returns a reference to an array of expressed tissues

       <b>next_express()</b> - returns the next tissue expression from the expressed tissue array

       <b>chromosome()</b> - set/get chromosome, currently takes/returns a reference to an array of chromosome lines

       <b>next_chromosome()</b> - returns the next chromosome line from the array of chromosome lines

       <b>sts()</b> - set/get sts, currently takes/returns a reference to an array of sts lines

       <b>next_sts()</b> - returns the next sts line from the array of sts lines

       <b>txmap()</b> - set/get txmap, currently takes/returns a reference to an array of txmap lines

       <b>next_txmap()</b> - returns the next txmap line from the array of txmap lines

       <b>protsim()</b> - set/get protsim, currently takes/returns a reference to an array of protsim lines

       <b>next_protsim()</b> - returns the next protsim line from the array of protsim lines

       <b>sequences()</b> - set/get sequence, currently takes/returns a reference to an array of references to seq info

       <b>next_seq()</b> - returns a Seq object that currently only contains an accession number

</pre><h4><b>Implemented</b> <b>Interfaces</b></h4><pre>
       This class implementes the following interfaces.

       Bio::Cluster::UniGeneI
           This includes implementing Bio::ClusterI.

       Bio::IdentifiableI
       Bio::DescribableI
       Bio::AnnotatableI
       Bio::Factory::SequenceStreamI

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User  feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists. Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be  able  look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report  bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Andrew</b> <b>Macgregor</b></h4><pre>
       Email andrew at cbbc.murdoch.edu.au

</pre><h4><b>CONTRIBUTORS</b></h4><pre>
       Hilmar Lapp, hlapp at gmx.net

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are  usually  preceded
       with a "_".

   <b>new</b>
        Title   : new
        Usage   : used by ClusterIO
        Returns : a new Bio::Cluster::Unigene object

</pre><h4><b>Bio::Cluster::UniGeneI</b> <b>methods</b></h4><pre>
   <b>unigene_id</b>
        Title   : unigene_id
        Usage   : unigene_id();
        Function: Returns the unigene_id associated with the object.
        Example : $id = $unigene-&gt;unigene_id or $unigene-&gt;unigene_id($id)
        Returns : A string
        Args    : None or an id

   <b>title</b>
        Title   : title
        Usage   : title();
        Function: Returns the title associated with the object.
        Example : $title = $unigene-&gt;title or $unigene-&gt;title($title)
        Returns : A string
        Args    : None or a title

   <b>gene</b>
        Title   : gene
        Usage   : gene();
        Function: Returns the gene associated with the object.
        Example : $gene = $unigene-&gt;gene or $unigene-&gt;gene($gene)
        Returns : A string
        Args    : None or a gene

   <b>cytoband</b>
        Title   : cytoband
        Usage   : cytoband();
        Function: Returns the cytoband associated with the object.
        Example : $cytoband = $unigene-&gt;cytoband or $unigene-&gt;cytoband($cytoband)
        Returns : A string
        Args    : None or a cytoband

   <b>mgi</b>
        Title   : mgi
        Usage   : mgi();
        Function: Returns the mgi associated with the object.
        Example : $mgi = $unigene-&gt;mgi or $unigene-&gt;mgi($mgi)
        Returns : A string
        Args    : None or a mgi

   <b>locuslink</b>
        Title   : locuslink
        Usage   : locuslink();
        Function: Returns or stores a reference to an array containing locuslink data.
        Returns : An array reference
        Args    : None or an array reference

   <b>homol</b>
        Title   : homol
        Usage   : homol();
        Function: Returns the homol entry associated with the object.
        Example : $homol = $unigene-&gt;homol or $unigene-&gt;homol($homol)
        Returns : A string
        Args    : None or a homol entry

   <b>restr_expr</b>
        Title   : restr_expr
        Usage   : restr_expr();
        Function: Returns the restr_expr entry associated with the object.
        Example : $restr_expr = $unigene-&gt;restr_expr or $unigene-&gt;restr_expr($restr_expr)
        Returns : A string
        Args    : None or a restr_expr entry

   <b>gnm_terminus</b>
        Title   : gnm_terminus
        Usage   : gnm_terminus();
        Function: Returns the gnm_terminus associated with the object.
        Example : $gnm_terminus = $unigene-&gt;gnm_terminus or
                  $unigene-&gt;gnm_terminus($gnm_terminus)
        Returns : A string
        Args    : None or a gnm_terminus

   <b>scount</b>
        Title   : scount
        Usage   : scount();
        Function: Returns the scount associated with the object.
        Example : $scount = $unigene-&gt;scount or $unigene-&gt;scount($scount)
        Returns : A string
        Args    : None or a scount

   <b>express</b>
        Title   : express
        Usage   : express();
        Function: Returns or stores a reference to an array containing
                  tissue expression data
        Returns : An array reference
        Args    : None or an array reference

   <b>chromosome</b>
        Title   : chromosome
        Usage   : chromosome();
        Function: Returns or stores a reference to an array containing
                  chromosome lines
        Returns : An array reference
        Args    : None or an array reference

   <b>sts</b>
        Title   : sts
        Usage   : sts();
        Function: Returns or stores a reference to an array containing sts lines

        Returns : An array reference
        Args    : None or an array reference

   <b>txmap</b>
        Title   : txmap
        Usage   : txmap();
        Function: Returns or stores a reference to an array containing txmap lines

        Returns : An array reference
        Args    : None or an array reference

   <b>protsim</b>
        Title   : protsim
        Usage   : protsim();
        Function: Returns or stores a reference to an array containing protsim lines
                  This should really only be used by ClusterIO, not directly
        Returns : An array reference
        Args    : None or an array reference

   <b>sequences</b>
        Title   : sequences
        Usage   : sequences();
        Function: Returns or stores a reference to an array containing
                  sequence data.

                  This is mostly reserved for ClusterIO parsers. You should
                  use get_members() for get and add_member()/remove_members()
                  for set.

        Returns : An array reference, or undef
        Args    : None or an array reference or undef

   <b>species</b>
        Title   : species
        Usage   : $obj-&gt;species($newval)
        Function: Get/set the species object for this Unigene cluster.
        Example :
        Returns : value of species (a L&lt;Bio::Species&gt; object)
        Args    : on set, new value (a L&lt;Bio::Species&gt; object or
                  the binomial name, or undef, optional)

</pre><h4><b>Bio::ClusterI</b> <b>methods</b></h4><pre>
   <b>display_id</b>
        Title   : display_id
        Usage   :
        Function: Get/set the display name or identifier for the cluster

                  This is aliased to unigene_id().

        Returns : a string
        Args    : optional, on set the display ID ( a string)

   <b>description</b>
        Title   : description
        Usage   : Bio::ClusterI-&gt;description("POLYUBIQUITIN")
        Function: get/set for the consensus description of the cluster

                  This is aliased to title().

        Returns : the description string
        Args    : Optional the description string

   <b>size</b>
        Title   : size
        Usage   : Bio::ClusterI-&gt;size();
        Function: get for the size of the family,
                  calculated from the number of members

                  This is aliased to scount().

        Returns : the size of the cluster
        Args    :

   <b>cluster_score</b>
        Title   : cluster_score
        Usage   : $cluster -&gt;<a href="../man100/cluster_score.100.html">cluster_score</a>(100);
        Function: get/set for cluster_score which
                  represent the score in which the clustering
                  algorithm assigns to this cluster.

                  For UniGene clusters, there really is no cluster score that
                  would come with the data. However, we provide an
                  implementation here so that you can score UniGene clusters
                  if you want to.

        Returns : a number
        Args    : optionally, on set a number

   <b>get_members</b>
        Title   : get_members
        Usage   : Bio::ClusterI-&gt;get_members(($seq1, $seq2));
        Function: retrieve the members of the family by some criteria

                  Will return all members if no criteria are provided.

                  At this time this implementation does not support
                  specifying criteria and will always return all members.

        Returns : the array of members
        Args    :

</pre><h4><b>Annotatable</b> <b>view</b> <b>at</b> <b>the</b> <b>object</b> <b>properties</b></h4><pre>
   <b>annotation</b>
        Title   : annotation
        Usage   : $obj-&gt;annotation($newval)
        Function: Get/set the L&lt;Bio::AnnotationCollectionI&gt; object for
                  this UniGene cluster.

                  Many attributes of this class are actually stored within
                  the annotation collection object as L&lt;Bio::AnnotationI&gt;
                  compliant objects, so you can conveniently access them
                  through the same interface as you would e.g. access
                  L&lt;Bio::SeqI&gt; annotation properties.

                  If you call this method in set mode and replace the
                  annotation collection with another one you should know
                  exactly what you are doing.

        Example :
        Returns : a L&lt;Bio::AnnotationCollectionI&gt; compliant object
        Args    : on set, new value (a L&lt;Bio::AnnotationCollectionI&gt;
                  compliant object or undef, optional)

</pre><h4><b>Implementation</b> <b>specific</b> <b>methods</b></h4><pre>
        These are mostly for adding/removing to array properties, and for
        methods with special functionality.

   <b>add_member</b>
        Title   : add_member
        Usage   :
        Function: Adds a member object to the list of members.
        Example :
        Returns : TRUE if the new member was successfully added, and FALSE
                  otherwise.
        Args    : The member to add.

   <b>remove_members</b>
        Title   : remove_members
        Usage   :
        Function: Remove the list of members for this cluster such that the
                  member list is undefined afterwards (as opposed to zero members).
        Example :
        Returns : the previous list of members
        Args    : none

   <b>next_locuslink</b>
        Title   : next_locuslink
        Usage   : next_locuslink();
        Function: Returns the next locuslink from an array referred
                  to using $obj-&gt;{'locuslink'}

                  If you call this iterator again after it returned undef, it
                  will re-cycle through the list of elements. Changes in the
                  underlying array property while you loop over this iterator
                  will not be reflected until you exhaust the iterator.

        Example :      while ( my $locuslink = $in-&gt;next_locuslink() ) {
                                       print "$locuslink\n";
                               }
        Returns : String
        Args    : None

   <b>next_express</b>
        Title   : next_express
        Usage   : next_express();
        Function: Returns the next tissue from an array referred
                  to using $obj-&gt;{'express'}

                  If you call this iterator again after it returned undef, it
                  will re-cycle through the list of elements. Changes in the
                  underlying array property while you loop over this iterator
                  will not be reflected until you exhaust the iterator.

        Example :      while ( my $express = $in-&gt;next_express() ) {
                                       print "$express\n";
                               }
        Returns : String
        Args    : None

   <b>next_chromosome</b>
        Title   : next_chromosome
        Usage   : next_chromosome();
        Function: Returns the next chromosome line from an array referred
                  to using $obj-&gt;{'chromosome'}

                  If you call this iterator again after it returned undef, it
                  will re-cycle through the list of elements. Changes in the
                  underlying array property while you loop over this iterator
                  will not be reflected until you exhaust the iterator.

        Example :      while ( my $chromosome = $in-&gt;next_chromosome() ) {
                                       print "$chromosome\n";
                               }
        Returns : String
        Args    : None

   <b>next_protsim</b>
        Title   : next_protsim
        Usage   : next_protsim();
        Function: Returns the next protsim line from an array referred
                  to using $obj-&gt;{'protsim'}

                  If you call this iterator again after it returned undef, it
                  will re-cycle through the list of elements. Changes in the
                  underlying array property while you loop over this iterator
                  will not be reflected until you exhaust the iterator.

        Example :      while ( my $protsim = $in-&gt;next_protsim() ) {
                                       print "$protsim\n";
                               }
        Returns : String
        Args    : None

   <b>next_sts</b>
        Title   : next_sts
        Usage   : next_sts();
        Function: Returns the next sts line from an array referred
                  to using $obj-&gt;{'sts'}

                  If you call this iterator again after it returned undef, it
                  will re-cycle through the list of elements. Changes in the
                  underlying array property while you loop over this iterator
                  will not be reflected until you exhaust the iterator.

        Example :      while ( my $sts = $in-&gt;next_sts() ) {
                                       print "$sts\n";
                               }
        Returns : String
        Args    : None

   <b>next_txmap</b>
        Title   : next_txmap
        Usage   : next_txmap();
        Function: Returns the next txmap line from an array
                  referred to using $obj-&gt;{'txmap'}

                  If you call this iterator again after it returned undef, it
                  will re-cycle through the list of elements. Changes in the
                  underlying array property while you loop over this iterator
                  will not be reflected until you exhaust the iterator.

        Example :      while ( my $tsmap = $in-&gt;next_txmap() ) {
                                       print "$txmap\n";
                               }
        Returns : String
        Args    : None

</pre><h4><b>Bio::IdentifiableI</b> <b>methods</b></h4><pre>
   <b>object_id</b>
        Title   : object_id
        Usage   : $string    = $obj-&gt;object_id()
        Function: a string which represents the stable primary identifier
                  in this namespace of this object. For DNA sequences this
                  is its accession_number, similarly for protein sequences

                  This is aliased to unigene_id().

        Returns : A scalar

   <b>version</b>
        Title   : version
        Usage   : $version    = $obj-&gt;version()
        Function: a number which differentiates between versions of
                  the same object. Higher numbers are considered to be
                  later and more relevant, but a single object described
                  the same identifier should represent the same concept

                  Unigene clusters usually won't have a version, so this
                  will be mostly undefined.

        Returns : A number
        Args    : on set, new value (a scalar or undef, optional)

   <b>authority</b>
        Title   : authority
        Usage   : $authority    = $obj-&gt;authority()
        Function: a string which represents the organisation which
                  granted the namespace, written as the DNS name for
                  organisation (eg, wormbase.org)

        Returns : A scalar
        Args    : on set, new value (a scalar or undef, optional)

   <b>namespace</b>
        Title   : namespace
        Usage   : $string    = $obj-&gt;namespace()
        Function: A string representing the name space this identifier
                  is valid in, often the database name or the name
                  describing the collection

        Returns : A scalar
        Args    : on set, new value (a scalar or undef, optional)

</pre><h4><b>Bio::DescribableI</b> <b>methods</b></h4><pre>
   <b>display_name</b>
        Title   : display_name
        Usage   : $string    = $obj-&gt;display_name()
        Function: A string which is what should be displayed to the user
                  the string should have no spaces (ideally, though a cautious
                  user of this interface would not assume this) and should be
                  less than thirty characters (though again, double checking
                  this is a good idea)

                  This is aliased to unigene_id().

        Returns : A scalar
        Status  : Virtual

   <b>description()</b>
        Title   : description
        Usage   : $string    = $obj-&gt;description()
        Function: A text string suitable for displaying to the user a
                  description. This string is likely to have spaces, but
                  should not have any newlines or formatting - just plain
                  text. The string should not be greater than 255 characters
                  and clients can feel justified at truncating strings at 255
                  characters for the purposes of display

                  This is already demanded by Bio::ClusterI and hence is
                  present anyway.

        Returns : A scalar

</pre><h4><b>Bio::Factory::SequenceStreamI</b> <b>methods</b></h4><pre>
   <b>next_seq</b>
        Title   : next_seq
        Usage   : next_seq();
        Function: Returns the next seq as a Seq object as defined by
                  $seq-&gt;sequence_factory(),
                  at present an empty Bio::Seq::RichSeq object with
                  just the accession_number() and pid() set

                  This iterator will not exhaust the array of member
                  sequences. If you call next_seq() again after it returned
                  undef, it will re-cycle through the list of member
                  sequences.

        Example :  while ( my $sequence = $in-&gt;next_seq() ) {
                    print $sequence-&gt;accession_number() . "\n";
                   }
        Returns : Bio::PrimarySeqI object
        Args    : None

   <b>sequence_factory</b>
        Title   : sequence_factory
        Usage   : $seqio-&gt;sequence_factory($seqfactory)
        Function: Get/Set the Bio::Factory::SequenceFactoryI
        Returns : Bio::Factory::SequenceFactoryI
        Args    : [optional] Bio::Factory::SequenceFactoryI

</pre><h4><b>Private</b> <b>methods</b></h4><pre>
   <b>_annotation_value</b>
        Title   : _annotation_value
        Usage   :
        Function: Private method.
        Example :
        Returns : the value (a string)
        Args    : annotation key (a string)
                  on set, annotation value (a string)

   <b>_annotation_value_ary</b>
        Title   : _annotation_value_ary
        Usage   :
        Function: Private method.
        Example :
        Returns : reference to the array of values
        Args    : annotation key (a string)
                  on set, reference to an array holding the values

   <b>_annotation_dblink</b>
        Title   : _annotation_dblink
        Usage   :
        Function: Private method.
        Example :
        Returns : array of accessions for the given database (namespace)
        Args    : annotation key (a string)
                  dbname (a string) (optional on get, mandatory on set)
                  on set, accession or ID (a string), and version

   <b>_remove_dblink</b>
        Title   : _remove_dblink
        Usage   :
        Function: Private method.
        Example :
        Returns : array of accessions for the given database (namespace)
        Args    : annotation key (a string)
                  dbname (a string) (optional)

perl v5.34.0                                       2022-03-27                         <u>Bio::Cluster::<a href="../man3pm/UniGene.3pm.html">UniGene</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>