<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>samtools-ampliconclip - clip reads using a BED file</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/samtools">samtools_1.21-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       samtools-ampliconclip - clip reads using a BED file

</pre><h4><b>SYNOPSIS</b></h4><pre>
       samtools  ampliconclip  [<b>-o</b> <u>out.file</u>] [<b>-f</b> <u>stat.file</u>] [<b>--soft-clip</b>] [<b>--hard-clip</b>] [<b>--both-ends</b>] [<b>--strand</b>]
       [<b>--clipped</b>] [<b>--fail</b>] [<b>--filter-len</b> <u>INT</u>] [<b>--fail-len</b> <u>INT</u>] [<b>--unmap-len</b>  <u>INT</u>]  [<b>--no-excluded</b>]  [<b>--rejects-</b>
       <b>file</b> <u>rejects.file</u>] [<b>--original</b>] [<b>--keep-tag</b>] [<b>--tolerance</b>] [<b>--no-PG</b>] [<b>-u</b>] <b>-b</b> <u>bed.file</u> <u>in.file</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Clips  the ends of read alignments if they intersect with regions defined in a BED file.  While this tool
       was originally written for  clipping  read  alignment  positions  which  correspond  to  amplicon  primer
       locations it can also be used in other contexts.

       BED  file entries used are chrom, chromStart, chromEnd and, optionally, strand.  Standard BED file format
       must be used, so if strand is needed then the name and score fields must also  be  present  (even  though
       ampliconclip  does  not read them).  There is a default tolerance of 5 bases when matching chromStart and
       chromEnd to alignments.

       By default the reads are soft clipped and clip is only done from the 5' end.

       Some things to be aware of.  While ordering is not significant, adjustments  to  the  left  most  mapping
       position  (<u>POS</u>)  will  mean  that coordinate sorted files will need resorting.  In such cases the sorting
       order in the header is set to unknown.  Clipping  of  reads  results  in  template  length  (<u>TLEN</u>)  being
       incorrect.  This  can  be corrected by <b>samtools</b> <b>fixmates</b>.  Any <u>MD</u> and <u>NM</u> aux tags will also be incorrect,
       which can be fixed by <b>samtools</b> <b>calmd</b>.  By default <u>MD</u> and <u>NM</u> tags are removed though if the output  is  in
       CRAM format these tags will be automatically regenerated.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-b</b> <u>FILE</u>    BED file of regions (e.g. amplicon primers) to be removed.

       <b>-o</b> <u>FILE</u>    Output file name (defaults to stdout).

       <b>-f</b> <u>FILE</u>    File to write stats to (defaults to stderr).

       <b>-u</b>         Output uncompressed SAM, BAM or CRAM.

       <b>--soft-clip</b>
                  Soft clip reads (default).

       <b>--hard-clip</b>
                  Hard clip reads.

       <b>--both-ends</b>
                  Clip  at both the 5' and the 3' ends where regions match.  When using this option the <b>--strand</b>
                  option is ignored.

       <b>--strand</b>   Use strand entry from the BED file to clip on the matching forward or reverse alignment.

       <b>--clipped</b>  Only output clipped reads.  Filter all others.

       <b>--fail</b>     Mark unclipped reads as QC fail.

       <b>--filter-len</b> <u>INT</u>
                  Filter out reads of INT size or shorter.  In this case soft clips are not counted toward  read
                  length.  An INT of 0 will filter out reads with no matching bases.

       <b>--fail-len</b> <u>INT</u>
                  As <b>--filter-len</b> but mark as QC fail rather then filter out.

       <b>--unmap-len</b> <u>INT</u>
                  As <b>--filter-len</b> but mark as unmapped. Default is 0 (no matching reads).  -1 will disable.

       <b>--no-excluded</b>
                  Filter  out  any  reads that are marked as QCFAIL or are unmapped.  This works on the state of
                  the reads before clipping takes place.

       <b>--rejects-file</b> <u>FILE</u>
                  Write any filtered reads out to a file.

       <b>--primer-counts</b> <u>FILE</u>
                  File to write with read counts per bed entry (bedgraph format).

       <b>--original</b> Add an OA tag with the original data for clipped files.

       <b>--keep-tag</b> In clipped reads, keep the possibly invalid NM  and  MD  tags.   By  default  these  tags  are
                  deleted.

       <b>--tolerance</b> <u>INT</u>
                  The amount of latitude given in matching regions to alignments.  Default 5 bases.

       <b>--no-PG</b>    Do not at a PG line to the header.

</pre><h4><b>AUTHOR</b></h4><pre>
       Written by Andrew Whitwham and Rob Davies, both from the Sanger Institute.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/samtools.1.html">samtools</a></u>(1), <u><a href="../man1/samtools-sort.1.html">samtools-sort</a></u>(1), <u><a href="../man1/samtools-fixmate.1.html">samtools-fixmate</a></u>(1), <u><a href="../man1/samtools-calmd.1.html">samtools-calmd</a></u>(1)

       Samtools website: &lt;<a href="http://www.htslib.org/">http://www.htslib.org/</a>&gt;

samtools-1.21                                   12 September 2024                       <u><a href="../man1/samtools-ampliconclip.1.html">samtools-ampliconclip</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>