<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>gustaf  -  Gustaf  -  Generic mUlti-SpliT Alignment Finder: Tool for split-read mapping allowing multiple</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/seqan-apps">seqan-apps_2.4.0+dfsg-16_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       gustaf  -  Gustaf  -  Generic mUlti-SpliT Alignment Finder: Tool for split-read mapping allowing multiple
       splits.

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>gustaf</b> [<u>OPTIONS</u>] &lt;<u>GENOME</u> <u>FASTA</u> <u>FILE</u>&gt; &lt;<u>READ</u> <u>FASTA</u> <u>FILE</u>&gt;

       <b>gustaf</b> [<u>OPTIONS</u>] &lt;<u>GENOME</u> <u>FASTA</u> <u>FILE</u>&gt; &lt;<u>READ</u> <u>FASTA</u> <u>FILE</u>&gt; &lt;<u>READ</u> <u>FASTA</u> <u>FILE</u> <u>2</u>&gt;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       GUSTAF uses SeqAns STELLAR to find splits as local matches on different strands or chromosomes.  Criteria
       and  penalties  to  chain  these matches can be specified. Output file contains the breakpoints along the
       best chain.

       The genome file is used as database input, the read file as query input.

       All STELLAR options are supported. See STELLAR documentation for STELLAR parameters and options.

       (c) 2011-2012 by Kathrin Trappe

</pre><h4><b>REQUIRED</b> <b>ARGUMENTS</b></h4><pre>
       <b>FASTA_FILE_1</b> <u>INPUT_FILE</u>
               Valid filetypes are: <u>.fq</u>, <u>.fastq</u>, <u>.fasta</u>, and <u>.fa</u>.

       <b>FASTA_FILE_2</b> List of <u>INPUT_FILE</u>'s
              Either one (single-end) or two (paired-end) read files. Valid filetypes are: <u>.fq</u>, <u>.fastq</u>,  <u>.fasta</u>,
              and <u>.fa</u>.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              Display the help message.

       <b>--version</b>
              Display version information.

   <b>Main</b> <b>Options:</b>
       <b>-tp</b>, <b>--transPen</b> <u>INTEGER</u>
              Interchromosomal translocation penalty Default: <u>5</u>.

       <b>-ip</b>, <b>--invPen</b> <u>INTEGER</u>
              Inversion penalty Default: <u>5</u>.

       <b>-op</b>, <b>--orderPen</b> <u>INTEGER</u>
              Intrachromosomal order change penalty Default: <u>0</u>.

       <b>-oth</b>, <b>--overlapThresh</b> <u>DOUBLE</u>
              Allowed overlap between matches Default: <u>0.5</u>.

       <b>-gth</b>, <b>--gapThresh</b> <u>INTEGER</u>
              Allowed  gap  length between matches, default value corresponse to expected size of microindels (5
              bp) Default: <u>5</u>.

       <b>-ith</b>, <b>--initGapThresh</b> <u>INTEGER</u>
              Allowed initial or ending gap length at begin and end of read with  no  breakpoint  (e.g.  due  to
              sequencing errors at the end) Default: <u>15</u>.

       <b>-bth</b>, <b>--breakendThresh</b> <u>INTEGER</u>
              Allowed  initial  or ending gap length at begin and end of read that creates a breakend/breakpoint
              (e.g. for reads extending into insertions) Default: <u>30</u>.

       <b>-tth</b>, <b>--tandemThresh</b> <u>INTEGER</u>
              Minimal length of (small) insertion/duplication with double overlap to be considered tandem repeat
              Default: <u>50</u>.

       <b>-pth</b>, <b>--breakpoint-pos-range</b> <u>INTEGER</u>
              Allowed difference in breakpoint position Default: <u>5</u>.

       <b>-cbp</b>, <b>--complex-breakpoints</b>
              Disable inferring complex SVs

       <b>-st</b>, <b>--support</b> <u>INTEGER</u>
              Number of supporting reads Default: <u>2</u>.

       <b>-mst</b>, <b>--mate-support</b> <u>INTEGER</u>
              Number of supporting concordant mates Default: <u>2</u>.

       <b>-ll</b>, <b>--library-size</b> <u>INTEGER</u>
              Library size of paired-end reads

       <b>-le</b>, <b>--library-error</b> <u>INTEGER</u>
              Library error (sd) of paired-end reads

       <b>-rc</b>, <b>--revcompl</b>
              Disable reverse complementing second mate pair input file.

   <b>Input</b> <b>Options:</b>
       <b>-m</b>, <b>--matchfile</b> <u>INPUT_FILE</u>
              File of (stellar) matches Valid filetypes are: <u>.gff</u> and <u>.GFF</u>.

   <b>Output</b> <b>Options:</b>
       <b>-gff</b>, <b>--gffOut</b> <u>OUTPUT_FILE</u>
              Name of gff breakpoint output file. Valid filetypes are: <u>.txt</u> and <u>.gff</u>. Default: <u>breakpoints.gff</u>.

       <b>-vcf</b>, <b>--vcfOut</b> <u>OUTPUT_FILE</u>
              Name of vcf breakpoint output file. Valid filetypes are: <u>.vcf</u> and <u>.txt</u>. Default: <u>breakpoints.vcf</u>.

       <b>-j</b>, <b>--jobName</b> <u>STRING</u>
              Job/Queue name Default: .

       <b>-do</b>, <b>--dots</b>
              Enable graph output in dot format

   <b>Parallelization</b> <b>Options:</b>
       <b>-nth</b>, <b>--numThreads</b> <u>INTEGER</u>
              Number of threads for parallelization of I/O. Default: <u>1</u>.

   <b>Main</b> <b>Options:</b>
       <b>-e</b>, <b>--epsilon</b> <u>DOUBLE</u>
              Maximal error rate (max 0.25). In range [0.0000001..0.25]. Default: <u>0.05</u>.

       <b>-l</b>, <b>--minLength</b> <u>INTEGER</u>
              Minimal length of epsilon-matches. In range [0..inf]. Default: <u>100</u>.

       <b>-f</b>, <b>--forward</b>
              Search only in forward strand of database.

       <b>-r</b>, <b>--reverse</b>
              Search only in reverse complement of database.

       <b>-a</b>, <b>--alphabet</b> <u>STRING</u>
              Alphabet type of input sequences (dna, rna, dna5, rna5, protein, char). One  of  <u>dna</u>,  <u>dna5</u>,  <u>rna</u>,
              <u>rna5</u>, <u>protein</u>, and <u>char</u>.

       <b>-v</b>, <b>--verbose</b>
              Set verbosity mode.

   <b>Filtering</b> <b>Options:</b>
       <b>-k</b>, <b>--kmer</b> <u>INTEGER</u>
              Length of the q-grams (max 32). In range [1..32].

       <b>-rp</b>, <b>--repeatPeriod</b> <u>INTEGER</u>
              Maximal period of low complexity repeats to be filtered. Default: <u>1</u>.

       <b>-rl</b>, <b>--repeatLength</b> <u>INTEGER</u>
              Minimal length of low complexity repeats to be filtered. Default: <u>1000</u>.

       <b>-c</b>, <b>--abundanceCut</b> <u>DOUBLE</u>
              k-mer overabundance cut ratio. In range [0..1]. Default: <u>1</u>.

   <b>Verification</b> <b>Options:</b>
       <b>-x</b>, <b>--xDrop</b> <u>DOUBLE</u>
              Maximal x-drop for extension. Default: <u>5</u>.

       <b>-vs</b>, <b>--verification</b> <u>STRING</u>
              Verification   strategy:  exact  or  bestLocal  or  bandedGlobal  One  of  <u>exact</u>,  <u>bestLocal</u>,  and
              <u>bandedGlobal</u>. Default: <u>exact</u>.

       <b>-dt</b>, <b>--disableThresh</b> <u>INTEGER</u>
              Maximal number of verified matches before disabling verification for one query  sequence  (default
              infinity). In range [0..inf].

       <b>-n</b>, <b>--numMatches</b> <u>INTEGER</u>
              Maximal  number  of  kept  matches per query and database. If STELLAR finds more matches, only the
              longest ones are kept. Default: <u>50</u>.

       <b>-s</b>, <b>--sortThresh</b> <u>INTEGER</u>
              Number of matches triggering removal of duplicates. Choose  a  smaller  value  for  saving  space.
              Default: <u>500</u>.

gustaf 1.0.0                                       August 2014                                         <u><a href="../man1/GUSTAF.1.html">GUSTAF</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>