<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>centrifuge - rapid and memory-efficient system for classification of DNA sequences</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/centrifuge">centrifuge_1.0.4.2-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       centrifuge - rapid and memory-efficient system for classification of DNA sequences

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Centrifuge version 1.0.3-beta by the Centrifuge developer team (<a href="mailto:centrifuge.metagenomics@gmail.com">centrifuge.metagenomics@gmail.com</a>) Usage:

              centrifuge  [options]*  <b>-x</b>  &lt;cf-idx&gt;  {-1  &lt;m1&gt;  <b>-2</b>  &lt;m2&gt; | <b>-U</b> &lt;r&gt;} [-S &lt;filename&gt;] [--report-file
              &lt;report&gt;]

       &lt;cf-idx&gt;
              Index filename prefix (minus trailing .X.cf).

       &lt;m1&gt;   Files with #1 mates, paired with files in &lt;m2&gt;.  Could be gzip'ed  (extension:  .gz)  or  bzip2'ed
              (extension: .bz2).

       &lt;m2&gt;   Files  with  #2  mates,  paired with files in &lt;m1&gt;.  Could be gzip'ed (extension: .gz) or bzip2'ed
              (extension: .bz2).

       &lt;r&gt;    Files with unpaired reads.  Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).

       &lt;filename&gt;
              File for classification output (default: stdout)

       &lt;report&gt;
              File for tabular report output (default: centrifuge_report.tsv)

              &lt;m1&gt;, &lt;m2&gt;, &lt;r&gt; can be comma-separated lists (no whitespace) and  can  be  specified  many  times.
              E.g. '-U file1.fq,file2.fq <b>-U</b> file3.fq'.

       Options (defaults in parentheses):

              Input:

       <b>-q</b>     query input files are FASTQ .fq/.fastq (default)

       <b>--qseq</b> query input files are in Illumina's qseq format

       <b>-f</b>     query input files are (multi-)FASTA .fa/.mfa

       <b>-r</b>     query input files are raw one-sequence-per-line

       <b>-c</b>     &lt;m1&gt;, &lt;m2&gt;, &lt;r&gt; are sequences themselves, not files

       <b>-s</b>/--skip &lt;int&gt;
              skip the first &lt;int&gt; reads/pairs in the input (none)

       <b>-u</b>/--upto &lt;int&gt;
              stop after first &lt;int&gt; reads/pairs (no limit)

       <b>-5</b>/--trim5 &lt;int&gt;
              trim &lt;int&gt; bases from 5'/left end of reads (0)

       <b>-3</b>/--trim3 &lt;int&gt;
              trim &lt;int&gt; bases from 3'/right end of reads (0)

       <b>--phred33</b>
              qualities are Phred+33 (default)

       <b>--phred64</b>
              qualities are Phred+64

       <b>--int-quals</b>
              qualities encoded as space-delimited integers

       <b>--ignore-quals</b>
              treat all quality values as 30 on Phred scale (off)

       <b>--nofw</b> do not align forward (original) version of read (off)

       <b>--norc</b> do not align reverse-complement version of read (off)

   <b>Classification:</b>
       <b>--min-hitlen</b> &lt;int&gt;
              minimum length of partial hits (default 22, must be greater than 15)

       <b>--min-totallen</b> &lt;int&gt;
              minimum summed length of partial hits per read (default 0)

       <b>--host-taxids</b> &lt;taxids&gt; comma-separated list of taxonomic IDs that will be preferred in classification

       <b>--exclude-taxids</b> &lt;taxids&gt; comma-separated list of taxonomic IDs that will be excluded in classification

              Output:

       <b>--out-fmt</b> &lt;str&gt;
              define output format, either 'tab' or 'sam' (tab)

       <b>--tab-fmt-cols</b> &lt;str&gt;
              columns         in         tabular         format,         comma         separated        default:
              readID,seqID,taxID,score,2ndBestScore,hitLength,queryLength,numMatches

       <b>-t</b>/--time
              print wall-clock time taken by search phases

       <b>--un</b> &lt;path&gt;
              write unpaired reads that didn't align to &lt;path&gt;

       <b>--al</b> &lt;path&gt;
              write unpaired reads that aligned at least once to &lt;path&gt;

       <b>--un-conc</b> &lt;path&gt;
              write pairs that didn't align concordantly to &lt;path&gt;

       <b>--al-conc</b> &lt;path&gt;
              write pairs that aligned concordantly at least once to &lt;path&gt;

              (Note: for <b>--un</b>, <b>--al</b>, <b>--un-conc</b>, or <b>--al-conc</b>, add  '-gz'  to  the  option  name,  e.g.   <b>--un-gz</b>
              &lt;path&gt;,   to   gzip   compress   output,  or  add  '-bz2'  to  bzip2  compress  output.)   <b>--quiet</b>
              print nothing to stderr except serious errors <b>--met-file</b> &lt;path&gt;     send metrics to file at &lt;path&gt;
              (off) <b>--met-stderr</b>          send metrics to stderr (off)  <b>--met</b>  &lt;int&gt;            report  internal
              counters &amp; metrics every &lt;int&gt; secs (1)

              Performance:

       <b>-o</b>/--offrate &lt;int&gt; override offrate of index; must be &gt;= index's offrate

       <b>-p</b>/--threads &lt;int&gt; number of alignment threads to launch (1)

       <b>--mm</b>   use memory-mapped I/O for index; many instances can share

              Other:

       <b>--qc-filter</b>
              filter out reads that are bad according to QSEQ filter

       <b>--seed</b> &lt;int&gt;
              seed for random number generator (0)

       <b>--non-deterministic</b> seed rand. gen. arbitrarily instead of using read attributes

       <b>--version</b>
              print version information and quit

       <b>-h</b>/--help
              print this usage message

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

centrifuge 1.0.3                                  October 2021                                     <u><a href="../man1/CENTRIFUGE.1.html">CENTRIFUGE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>