<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>primer3_core - Designs primers for PCR</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/primer3">primer3_2.6.1-5_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       primer3_core - Designs primers for PCR

</pre><h4><b>SYNOPSIS</b></h4><pre>

       <b>primer3_core</b> [-format_output] [-strict_tags] [&lt; <u>input_file</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       primer3_core picks primers for PCR reactions, considering as criteria oligonucleotide melting
       temperature, size, GC content and primer-dimer possibilities, PCR product size, positional constraints
       within the source sequence, and miscellaneous other constraints.

       By default, primer3_core accepts input and produces output in Boulder-io format, a pre-XML text-based
       input/output format for program-to-program data interchange format. The Boulder-io format and the
       commands that primer3_core understands are described in the <u>README</u> file, which on Debian systems can be
       found in <u>/usr/share/doc/primer3/</u>.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-format_output</b>
           Prints a more user-oriented report for each sequence.

       <b>-strict_tags</b>
           primer3_core echoes and ignores any tags it does not recognize, unless the <b>-strict_tags</b> flag is set
           on the command line, in which case primer3_core prints an error in the PRIMER_ERROR output tag, and
           prints additional information on stdout; this option can be useful for debugging systems that
           incorporate primer.

       Note

       The old flag -2x_compat is no longer supported.

</pre><h4><b>EXIT</b> <b>STATUS</b> <b>CODES</b></h4><pre>
       •   0 on normal operation.

       •   -1 under the following conditions: illegal command-line arguments, unable to fflush stdout, unable to
           open (for writing and creating) a .for, .rev or .int file (probably due to a protection problem).

       •   -2 on out-of-memory.

       •   -3 empty input.

       •   -4 error in a "Global" input tag (message in PRIMER_ERROR).

</pre><h4><b>REFERENCE</b></h4><pre>
       Please cite Rozen, S., Skaletsky, H. "Primer3 on the WWW for general users and for biologist
       programmers." In S. Krawetz and S. Misener, eds. Bioinformatics Methods and Protocols in the series
       Methods in Molecular Biology. Humana Press, Totowa, NJ, 2000, pages 365-386.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/ntdpal.1.html">ntdpal</a></b>(1) <b><a href="../man1/oligotm.1.html">oligotm</a></b>(1)

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright © 1996,1997,1998,1999,2000,2001,2004,2006,2007,2008 Whitehead Institute for Biomedical
       Research, Steve Rozen (<a href="http://jura.wi.mit.edu/rozen">http://jura.wi.mit.edu/rozen</a>), Helen Skaletsky

       All rights reserved. On Debian-based systems, please consult <u>/usr/share/doc/primer3/copyright</u> to read the
       licence of primer3_core.

       This manual page was written by Steffen Moeller &lt;<a href="mailto:moeller@debian.org">moeller@debian.org</a>&gt; for the Debian(TM) system (but may
       be used by others). Permission is granted to copy, distribute and/or modify this document under the same
       terms as primer3_core itself.

primer3_core 1.1.4                                 05/09/2008                                    <u><a href="../man1/PRIMER3_CORE.1.html">PRIMER3_CORE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>