<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>NAME</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/libgenome-model-tools-music-perl">libgenome-model-tools-music-perl_0.04-5_all</a> <br><br><pre>
</pre><h4><b>gmt</b> <b>music</b></h4><pre>
</pre><h4><b>NAME</b></h4><pre>
       gmt music - Mutational Significance in Cancer (Cancer Mutation Analysis)

</pre><h4><b>VERSION</b></h4><pre>
       This document describes gmt music version 0.04 (2020-11-06 at 21:54:23)

</pre><h4><b>SYNOPSIS</b></h4><pre>
       gmt music ...

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The MuSiC suite is a set of tools aimed at discovering the significance of somatic mutations found within
       a given cohort of cancer samples, and with respect to a variety of external data sources. The standard
       inputs required are:

       1. mapped reads in BAM format
       2. predicted or validated SNVs or indels in mutation annotation format (MAF)
       3. a list of regions of interest (typically the boundaries of coding exons)
       4. any relevant numeric or categorical clinical data.

       The formats for inputs 3. and 4. are:

       3. Regions of Interest File:
           •   Do not use headers

           •   4 columns, which are [chromosome  start-position(1-based)  stop-position(1-based)  gene_name]

       4. Clinical Data Files:
           •   Headers are required

           •   At  least  1  sample_id  column  and  1  attribute  column,  with  the  format  being  [sample_id
               clinical_data_attribute  clinical_data_attribute  ...]

           •   The sample_id must match the  sample_id  listed  in  the  MAF  under  "Tumor_Sample_Barcode"  for
               relating the mutations of this sample.

           •   The   header  for  each  clinical_data_attribute  will  appear  in  the  output  file  to  denote
               relationships with the mutation data from the MAF.

       Descriptions for the usage of each tool (each sub-command) can be found separately.

       The <b>play</b> command runs all of the sub-commands serially on a selected input set.

</pre><h4><b>LICENSE</b></h4><pre>
       Copyright (C) 2007-2011 Washington University in St. Louis.

       It is released under the Lesser GNU Public License (LGPL) version 3.  See the associated LICENSE file  in
       this distribution.

</pre><h4><b>AUTHORS</b></h4><pre>
       This  software  is  developed by the analysis and engineering teams at The Genome Institute at Washington
       University School of Medicine in St.  Louis,  with  funding  from  the  National  Human  Genome  Research
       Institute.  Richard K. Wilson, P.I.

       The primary authors of the MuSiC suite are:

        Nathan D. Dees, Ph.D.
        Cyriac Kandoth, Ph.D.
        Dan Koboldt, M.S.
        William Schierding, M.S.
        Michael Wendl, Ph.D.
        Qunyuan Zhang, Ph.D.
        Thomas B. Mooney, M.S.

</pre><h4><b>CREDITS</b></h4><pre>
       The MuSiC suite uses tabix, by Heng Li.  See <a href="http://samtools.sourceforge.net/tabix.shtml">http://samtools.sourceforge.net/tabix.shtml</a>.

       MuSiC  depends  on  copies  of  data  from  the following databases, packaged in a form useable for quick
       analysis:

        * KEGG - <a href="http://www.genome.jp/kegg/">http://www.genome.jp/kegg/</a>
        * COSMIC - <a href="http://www.sanger.ac.uk/genetics/CGP/cosmic/">http://www.sanger.ac.uk/genetics/CGP/cosmic/</a>
        * OMIM - <a href="http://www.ncbi.nlm.nih.gov/omim">http://www.ncbi.nlm.nih.gov/omim</a>
        * Pfam - <a href="http://pfam.sanger.ac.uk/">http://pfam.sanger.ac.uk/</a>
        * SMART - <a href="http://smart.embl-heidelberg.de/">http://smart.embl-heidelberg.de/</a>
        * SUPERFAMILY - <a href="http://supfam.cs.bris.ac.uk/SUPERFAMILY/">http://supfam.cs.bris.ac.uk/SUPERFAMILY/</a>
        * PatternScan - <a href="http://www.expasy.ch/prosite/">http://www.expasy.ch/prosite/</a>

</pre><h4><b>BUGS</b></h4><pre>
       For defects with any software in the genome namespace, contact
        genome-dev ~at~ genome.wustl.edu.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/genome.1.html">genome</a></b>(1)

perl v5.30.3                                       2020-11-06                                      <u><a href="../man1p/GMT-MUSIC.1p.html">GMT-MUSIC</a></u>(1p)
</pre>
 </div>
</div></section>
</div>
</body>
</html>