<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>QTLtools rtc-union - Find the union of QTLs from independent datasets</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/qtltools">qtltools_1.3.1+dfsg-4build4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       QTLtools rtc-union - Find the union of QTLs from independent datasets

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>QTLtools</b>  <b>rtc-union</b>  <b>--vcf</b>  [<u>in.vcf</u>|<u>in.vcf.gz</u>|<u>in.bcf</u>|<u>in.bed.gz</u>]  ...   <b>--bed</b>  <u>quantifications.bed.gz</u>  ...
       <b>--hotspots</b> <u>hotspots_b37_hg19.bed</u> <b>--results</b> <u>qtl_results_files.txt</u> ...  <u>[OPTIONS]</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This mode finds the best molQTL (may or may not be genome-wide significant) in  each  region  flanked  by
       recombination  hotspots (coldspot),  if there was a molQTL in the same coldspot in one dataset.  First we
       map all the significant molQTLs in all of the datasets to coldspots. Subsequently if certain datasets  do
       not  have  a  significant  molQTL  in  a  given  coldspot  for  a  given phenotype, we then take the most
       significant variant associated with that phenotype in that coldspot, for all the missing datasets.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>--vcf</b> <b>[</b><u>in.vcf</u>|<u>in.bcf</u>|<u>in.vcf.gz</u><b>|</b><u>in.bed.gz</u><b>]</b> <b>...</b>
              Genotypes in VCF/BCF format, or another molecular phenotype in BED format.  If there is a DS field
              in  the  genotype  FORMAT  of  a  variant  (dosage  of  the  genotype  calculated  from   genotype
              probabilities, e.g. after imputation), then this is used as the genotype.  If there is only the GT
              field  in the genotype FORMAT then this is used and it is converted to a dosage.  If a single file
              is provided then all datasets are assumed to have the same genotypes, and  all  datasets'  samples
              are  all  included in this file.  If multiple files are provided for each dataset, then <b>all</b> <b>--vcf,</b>
              <b>--bed,</b> <b>--cov,</b> <b>and</b> <b>--results</b> <b>files</b> <b>MUST</b> <b>be</b> <b>in</b> <b>the</b> <b>same</b> <b>order</b>.  E.g if the first vcf  file  is  from
              dataset1, then the first bed, cov, and results files must also be from dataset1.  REQUIRED.

       <b>--bed</b> <u>quantifications.bed.gz</u> ...
              Molecular  phenotype  quantifications  in  BED format for each of the datasets.  <b>All</b> <b>--vcf,</b> <b>--bed,</b>
              <b>--cov,</b> <b>and</b> <b>--results</b> <b>files</b> <b>MUST</b> <b>be</b> <b>in</b> <b>the</b> <b>same</b> <b>order</b>.  E.g if the first vcf file is from dataset1,
              then the first bed, cov, and results files must also be from dataset1.  REQUIRED.

       <b>--results</b> <u>significant_qtls.txt</u> ...
              Results file with the QTLs in each of the datasets.  <b>All</b> <b>--vcf,</b> <b>--bed,</b> <b>--cov,</b> <b>and</b> <b>--results</b>  <b>files</b>
              <b>MUST</b>  <b>be</b>  <b>in</b> <b>the</b> <b>same</b> <b>order</b>.  E.g if the first vcf file is from dataset1, then the first bed, cov,
              and results files must also be from dataset1.  REQUIRED.

       <b>--hotspots</b> <u>recombination_hotspots.bed</u>
              Recombination hotspots in BED format.  REQUIRED.

       <b>--out-suffix</b> <u>suffix</u>
              If provided output files will be suffixed with this.

       <b>--cov</b> <u>covariates.txt</u>
              Covariates to correct the phenotype data with for each of the datasets.  <b>All</b> <b>--vcf,</b> <b>--bed,</b>  <b>--cov,</b>
              <b>and</b>  <b>--results</b>  <b>files</b> <b>MUST</b> <b>be</b> <b>in</b> <b>the</b> <b>same</b> <b>order</b>.  E.g if the first vcf file is from dataset1, then
              the first bed, cov, and results files must also be from dataset1.

       <b>--force</b>
              If the output file exists, overwrite it.

       <b>--normal</b>
              Rank normal transform  the  phenotype  data  so  that  each  phenotype  is  normally  distributed.
              RECOMMENDED.

       <b>--conditional</b>
              molQTLs contain independent signals so execute the conditional analysis.

       <b>--window</b> <u>integer</u>
              Size   of   the   cis   window   flanking   each  phenotype's  start  position.   DEFAULT=1000000.
              RECOMMENDED=1000000.

       <b>--pheno-col</b> <u>integer</u>
              1-based phenotype id column number.  DEFAULT=1

       <b>--geno-col</b> <u>integer</u>
              1-based genotype id column number.  DEFAULT=8

       <b>--rank-col</b> <u>integer</u>
              1-based conditional analysis rank column number.  Only relevant if  <b>--conditional</b>  is  in  effect.
              DEFAULT=12

       <b>--best-col</b> <u>integer</u>
              1-based phenotype column number Only relevant if <b>--conditional</b> is in effect.  DEFAULT=21

       <b>--chunk</b> <u>integer1</u> <u>integer2</u>
              For  parallelization.   Divide  the  data  into <u>integer2</u> number of chunks and process chunk number
              <u>integer1</u>.  Chunk 0 will print a header.  Mutually exclusive  with  --region.   <b>Minimum</b>  <b>number</b>  <b>of</b>
              <b>chunks</b> <b>has</b> <b>to</b> <b>be</b> <b>at</b> <b>least</b> <b>the</b> <b>same</b> <b>number</b> <b>of</b> <b>chromosomes</b> <b>in</b> <b>the</b> <b>--bed</b> <b>file.</b>

       <b>--region</b> <u>chr:start-end</u>
              Genomic  region  to  be processed.  E.g. chr4:12334456-16334456, or chr5.  Mutually exclusive with
              --chunk.

</pre><h4><b>OUTPUT</b> <b>FILE</b></h4><pre>
       <b>output</b> <b>file</b>
        Space separated output file with the following columns.
        1   Column showing that this is a rtc-union result.  Always __UNION__
        2   The phenotype ID
        3   The  genotype  ID.   This  can  say   __UNION_FILLER_MAX_INDEP__,   __UNION_FILLER_MISS_GENO__,   or
            __UNION_FILLER_MISS_PHENO__ which are fillers for missing cases in one of the datasets.
        4   The  rank  of  the  best variant in this coldspot.  If this was discovered in the rtc-union run then
            this would be -1, and if there was already a significant variant in this coldspot then  a  different
            value.
        5   Dummy field indicating that this is the best hit per rank
        6   The p-value of the association.  Will be 0 if this was already significant in the dataset
        7   The coldspot ID
        8   The coldspot region

</pre><h4><b>EXAMPLE</b></h4><pre>
       o Find  the  union  of  3 datasets, correcting for technical covariates, and rank normal transforming the
         phenotypes with 20 jobs on a compute cluster (qsub needs to be changed to  the  job  submission  system
         used [bsub, psub, etc...]):

         for j in $(seq 1 20); do
             echo  "QTLtools rtc-union --bed dataset1.bed.gz dataset2.bed.gz dataset3.bed.gz  --vcf dataset1.bcf
             dataset2.bcf      dataset3.bcf      --cov      dataset1.covariates.txt      dataset2.covariates.txt
             dataset3.covariates.txt     --results    dataset1.txt    dataset2.txt    dataset3.txt    --hotspots
             hotspots_b37_hg19.bed --normal --conditional --chunk $j 20 --out-suffix .chunk.$j.20.txt" | qsub
         done

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/QTLtools.1.html">QTLtools</a></u>(1)

       QTLtools website: &lt;https://qtltools.github.io/qtltools&gt;

</pre><h4><b>BUGS</b></h4><pre>
       Versions up to and including 1.2, suffer from a bug in reading missing genotypes in VCF/BCF files.   This
       bug  affects variants with a DS field in their genotype's FORMAT and have a missing genotype (DS field is
       .) in one of the samples, in which case genotypes for all the samples are  set  to  missing,  effectively
       removing this variant from the analyses.

       Please submit bugs to &lt;https://github.com/qtltools/qtltools&gt;

</pre><h4><b>CITATION</b></h4><pre>
       Ongen  H, Brown AA, Delaneau O, et al. Estimating the causal tissues for complex traits and diseases. <u>Nat</u>
       <u>Genet</u>. 2017;<b><a href="../man12/49.12.html">49</a></b>(12):1676-1683. doi:10.1038/ng.3981 &lt;https://doi.org/10.1038/ng.3981&gt;

</pre><h4><b>AUTHORS</b></h4><pre>
       Halit Ongen (<a href="mailto:halitongen@gmail.com">halitongen@gmail.com</a>), Olivier Delaneau (<a href="mailto:olivier.delaneau@gmail.com">olivier.delaneau@gmail.com</a>)

QTLtools-v1.3                                      06 May 2020                             <u><a href="../man1/QTLtools-rtc-union.1.html">QTLtools-rtc-union</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>