<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>samtools-cat - concatenate files together</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/samtools">samtools_1.21-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       samtools-cat - concatenate files together

</pre><h4><b>SYNOPSIS</b></h4><pre>
       samtools cat [<b>-b</b> <u>list</u>] [<b>-h</b> <u>header.sam</u>] [<b>-o</b> <u>out.bam</u>] <u>in1.bam</u> <u>in2.bam</u> [ ... ]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Concatenate  BAMs  or  CRAMs. Although this works on either BAM or CRAM, all input files must be the same
       format as each other. The sequence dictionary of each input file must be identical, although this command
       does not check this. This command uses a similar trick to <b>reheader</b> which enables fast BAM concatenation.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-b</b> <u>FOFN</u> Read the list of input BAM or CRAM files from <u>FOFN</u>.  These are concatenated prior  to  any  files
               specified on the command line.  Multiple <b>-b</b> <u>FOFN</u> options may be specified to concatenate multiple
               lists of BAM/CRAM files.

       <b>-h</b> <u>FILE</u> Uses  the  SAM  header  from  <u>FILE</u>.   By  default  the  header is taken from the first file to be
               concatenated.

       <b>-o</b> <u>FILE</u> Write the concatenated output to <u>FILE</u>.  By default this is sent to stdout.

       <b>-q</b>      [CRAM only] Query the number of containers in the CRAM file.  The output  is  the  filename,  the
               number  of  containers,  and  the first and last container number as an inclusive range, with one
               file per line.

               Note this works in conjunction with the <b>-r</b> <u>RANGE</u> option, in which case the 3rd  and  4th  columns
               become useful for identifying which containers span the requested range.

       <b>-r</b> <u>RANGE</u>
               [CRAM only] Filter the CRAM file to a specific <u>RANGE</u>.  This can be the usual chromosome:start-end
               syntax, or "*" for unmapped records at the end of alignments.

               If  the  range is of the form "#:start-end" then the start and end coordinates are interpreted as
               inclusive CRAM container numbers, starting at 0 and ending 1 less than the number  of  containers
               reported by <b>-q</b>.  For example <b>-r</b> <b>"#:0-9"</b> is the first 10 CRAM containers of data.

               All  range  types  filter data in as fast a manner as possible, using operating system read/write
               loops where appropriate.

       <b>-p</b> <u>A/B</u>  [CRAM only] Filter the  CRAM  file  using  a  specific  fraction.   The  file  is  split  into  B
               approximately  equal  parts  and returns element A where A is between 1 and B inclusive. If there
               are more parts specified than CRAM containers then some of the output will be empty CRAMs.

               This can also be combined with the range option above to operate of parts  of  that  range.   For
               example <b>-r</b> <b>chr2</b> <b>-p</b> <b>1/10</b> returns the first 1/10th of data aligned against chromosome 2.

       <b>-f</b>      [CRAM  only] Enable fast mode.  When filtering by chromosome range with <b>-r</b> we normally do careful
               recoding of any containers that overlap the start and end  of  the  range  so  the  record  count
               precisely  matches  that returned by a <b>samtools</b> <b>view</b> equivalent.  Fast mode does no filtering, so
               may return additional alignments in the same container but outside of the requested region.

       <b>--no-PG</b> Do not add a @PG line to the header of the output file.

       <b>-@,</b> <b>--threads</b> <u>INT</u>
               Number of input/output compression threads to use in addition to main thread [0].

</pre><h4><b>EXAMPLES</b></h4><pre>
       o Extract a specific chromosome from a CRAM file, outputting to a new CRAM.

           samtools cat -o chr10.cram -r chr10 in.cram

       o Split a CRAM file up into separate files, each containing at most 123 containers.

           set -- $(samtools cat -q in.cram); nc=$2; s=0
           while [ $s -lt $nc ]
           do
               e=`expr $s + 123`
               if [ $e -ge $nc ]
               then
                   e=$nc
               fi
               r="$s-`expr $e - 1`"; echo $r
               fn=/tmp/_part-`printf "%08d" $s`.cram
               samtools cat -o $fn in.cram -r "#:$r"
               s=$e
           done

       o Split any unaligned data from a (potentially aligned) CRAM  file  into  10  approximately  equal  sized
         pieces.

           for i in `seq 1 10`
           do
              samtools cat in.cram -r "*" -p $i/10 -o part-`printf "%02d" $i`.cram
           done

</pre><h4><b>AUTHOR</b></h4><pre>
       Written  by  Heng  Li  from  the  Sanger  Institute.   Updated  for  CRAM  by James Bonfield (also Sanger
       Institute).

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/samtools.1.html">samtools</a></u>(1)

       Samtools website: &lt;<a href="http://www.htslib.org/">http://www.htslib.org/</a>&gt;

samtools-1.21                                   12 September 2024                                <u><a href="../man1/samtools-cat.1.html">samtools-cat</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>