<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>pilon - automated genome assembly improvement and variant detection tool</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/pilon">pilon_1.24-4_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       pilon - automated genome assembly improvement and variant detection tool

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>pilon</b>  <b>--genome</b>  genome.fasta [--frags frags.bam] [--jumps jumps.bam] [--unpaired unpaired.bam] [...other
       options...]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Pilon is a software tool which can be used to:

       • Automatically improve draft assemblies

       • Find variation among strains, including large event detection

       Pilon requires as input a FASTA file of the genome along with one or more BAM files of reads  aligned  to
       the  input  FASTA  file. Pilon uses read alignment analysis to identify inconsistencies between the input
       genome and the evidence in the reads. It  then  attempts  to  make  improvements  to  the  input  genome,
       including:

       • Single base differences

       • Small indels

       • Larger indel or block substitution events

       • Gap filling

       • Identification of local misassemblies, including optional opening of new gaps

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>--help</b>

              for option details

   <b>INPUTS</b>
              <b>--genome</b> genome.fasta

              The  input  genome  we  are  trying  to  improve,  which  must  be  the reference used for the bam
              alignments.  At least one of <b>--frags</b> or <b>--jumps</b> must also be given.

              <b>--frags</b> frags.bam

              A bam file consisting of fragment paired-end alignments, aligned to the  <b>--genome</b>  argument  using
              bwa or bowtie2.  This argument may be specified more than once.

              <b>--jumps</b> jumps.bam

              A  bam file consisting of jump (mate pair) paired-end alignments, aligned to the <b>--genome</b> argument
              using bwa or bowtie2.  This argument may be specified more than once.

              <b>--unpaired</b> unpaired.bam

              A bam file consisting of unpaired alignments, aligned  to  the  <b>--genome</b>  argument  using  bwa  or
              bowtie2.  This argument may be specified more than once.

              <b>--bam</b> any.bam

              A  bam file of unknown type; Pilon will scan it and attempt to classify it as one of the above bam
              types.

   <b>OUTPUTS</b>
              <b>--output</b> prefix

              Prefix for output files

              <b>--outdir</b> directory

              Use this directory for all output files.

              <b>--changes</b>

              If specified, a file listing changes in the &lt;output&gt;.fasta will be generated.

              <b>--vcf</b>

              If specified, a vcf file will be generated

              <b>--vcfqe</b>

              If specified, the VCF will contain a QE (quality-weighted evidence) field rather than the  default
              QP (quality-weighted percentage of evidence) field.

              <b>--tracks</b>

              This  options  will cause many track files (*.bed, *.wig) suitable for viewing in a genome browser
              to be written.

   <b>CONTROL</b>
              <b>--variant</b>

              Sets up heuristics for variant calling, as opposed to assembly improvement; equivalent  to  "--vcf
              <b>--fix</b> all,breaks".

              <b>--chunksize</b>

              Input  FASTA elements larger than this will be processed in smaller pieces not to exceed this size
              (default 10000000).

              <b>--diploid</b>

              Sample is from diploid organism; will eventually affect calling of heterozygous SNPs

              <b>--fix</b> fixlist

              A comma-separated list of categories of issues to try to fix:

              "snps": try to fix individual base errors; "indels": try to fix small indels; "gaps": try to  fill
              gaps;  "local":  try  to  detect  and  fix local misassemblies; "all": all of the above (default);
              "bases": shorthand for "snps" and "indels" (for back compatibility); "none": none  of  the  above;
              new fasta file will not be written.

              The following are experimental fix types:

              "amb":  fix  ambiguous  bases  in fasta output (to most likely alternative); "breaks": allow local
              reassembly to open new gaps (with "local"); "circles": try to close  circlar  elements  when  used
              with long corrected reads; "novel": assemble novel sequence from unaligned non-jump reads.

              <b>--dumpreads</b>

              Dump reads for local re-assemblies.

              <b>--duplicates</b>

              Use reads marked as duplicates in the input BAMs (ignored by default).

              <b>--iupac</b>

              Output IUPAC ambiguous base codes in the output FASTA file when appropriate.

              <b>--nonpf</b>

              Use reads which failed sequencer quality filtering (ignored by default).

              <b>--targets</b> targetlist

              Only process the specified target(s).  Targets are comma-separated, and each target

              is    a    fasta    element    name    optionally    followed   by   a   base   range.    Example:
              "scaffold00001,scaffold00002:10000-20000" would result in  processing  all  of  scaffold00001  and
              coordinates  10000-20000  of scaffold00002.  If "targetlist" is the name of a file, each line will
              be treated as a target specification.

              <b>--threads</b>

              Degree of parallelism to use for certain processing (default 1). Experimental.

              <b>--verbose</b>

              More verbose output.

              <b>--debug</b>

              Debugging output (implies verbose).

              <b>--version</b>

              Print version string and exit.

   <b>HEURISTICS:</b>
              <b>--defaultqual</b> qual

              Assumes bases are of this quality if quals are no present in input BAMs (default 15).

              <b>--flank</b> nbases

              Controls how much of the well-aligned reads will be used; this many bases at each end of the  good
              reads will be ignored (default 10).

              <b>--gapmargin</b>

              Closed gaps must be within this number of bases of true size to be closed (100000)

              <b>--K</b>

              Kmer size used by internal assembler (default 47).

              <b>--mindepth</b> depth

              Variants (snps and indels) will only be called if there is coverage of good pairs at this depth or
              more;  if this value is &gt;= 1, it is an absolute depth, if it is a fraction &lt; 1, then minimum depth
              is computed by multiplying this value by the mean coverage for the region, with a minimum value of
              5 (default 0.1: min depth to call is 10% of mean coverage or 5, whichever is greater).

              <b>--mingap</b>

              Minimum size for unclosed gaps (default 10)

              <b>--minmq</b>

              Minimum alignment mapping quality for a read to count in pileups (default 0)

              <b>--minqual</b>

              Minimum base quality to consider for pileups (default 0)

              <b>--nostrays</b>

              Skip making a pass through the input BAM files to identify stray pairs, that is,  those  pairs  in
              which  both  reads  are aligned but not marked valid because they have inconsistent orientation or
              separation. Identifying stray pairs can help fill gaps and assemble larger insertions,  especially
              of repeat content.  However, doing so sometimes consumes considerable memory.

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

pilon 1.22                                          June 2018                                           <u><a href="../man1/PILON.1.html">PILON</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>