<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>chipcor - Feature Extraction Tool for ChIP-seq data analysis</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/chip-seq">chip-seq_1.5.5-3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       chipcor - Feature Extraction Tool for ChIP-seq data analysis

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>chipscore</b> <b>[</b> <u>options</u> <b>]</b> <b>-A</b> <u>&lt;ref</u> <u>feature&gt;</u> <b>-B</b> <u>&lt;tar</u> <u>feature&gt;</u> <b>-b</b> <u>&lt;from&gt;</u> <b>-e</b> <u>&lt;to&gt;</u> <b>[</b> <u>&lt;</u> <b>]</b> <b>[</b> <u>SGA</u> <u>file</u> <b>]</b>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>chipscore</b>  reads  a  ChIP-seq  data file (or from stdin [&lt;]) in SGA format (&lt;SGA file&gt;), and compares two
       features, a reference feature (&lt;feature A&gt;) and a target feature (&lt;feature B&gt;),  with  respect  to  their
       relative  position.  It  then  extract  feature A tags that are enriched in feature B tags according to a
       given Count Output Threshold or Score (-t).

       Launching <b>chipscore</b> without any arguments will print the options list, along with their default values.

       The feature specification must have the following format: &lt;feature&gt; = &lt;name&gt; [&lt;+|-&gt;]

       The strand specification (+|-) being optional.

       The &lt;feature&gt; parameter is a name that corresponds to the second field of the SGA file.  If no feature is
       given then all input tags are processed.

       The SGA input file MUST BE sorted by sequence name (or chromosome id), position, and strand.  One  should
       check the input SGA file with the following command:

       sort -s -c -k1,1 -k3,3n -k4,4 &lt;SGA file&gt;

</pre><h4><b>OPTIONS</b></h4><pre>
       -A &lt;ref feature&gt;
              The feature that is taken as a reference for correlating chIP-seq tag distances.

              This parameter is mandatory.

       -B &lt;tar feature&gt;
              Tags  corresponding to the target feature are positionally correlated to tags corresponding to the
              reference feature.

              This parameter is mandatory.

       -b &lt;from&gt;
              The relative distance between the two features is analysed within a given range: &lt;from&gt;  indicates
              the beginning.

       -e &lt;to&gt;
              The  relative  distance  between the two features is analysed within a given range: &lt;to&gt; indicates
              the end.

       -c &lt;cut-off&gt;
              A value can be optionally specified as a cut-off for the input tag counts.

              This parameter is optional. Its default value is 10.

       -d     Show debug info. The program performs the sorting order check of the input data file.

       -h     Show the usage message.

       -o     Oriented strand option. Requiring oriented strand processing means reverting the  chromosome  axis
              when the reference feature is on the - strand.

       -r     Reverse extraction process. The program extracts sites that are depleted in feature B.

       -q     Report feature B tag counts as 'feature_name=&lt;int&gt;'.

       -t &lt;thres&gt;
              Output threshold or score. The program extracts feature A tags that are enriched in feature B tags
              according to a given count output threshold.

              The default value is 0.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/chipcor.1.html">chipcor</a></b>(1), <b><a href="../man1/chipextract.1.html">chipextract</a></b>(1), <b><a href="../man1/chipcenter.1.html">chipcenter</a></b>(1), <b><a href="../man1/chippeak.1.html">chippeak</a></b>(1) <b><a href="../man1/chippart.1.html">chippart</a></b>(1)

Bioinformatics                                      June 2015                                       <u><a href="../man1/chipscore.1.html">chipscore</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>