<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>chipcenter - Feature Centering Tool for ChIP-seq data analysis</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/chip-seq">chip-seq_1.5.5-3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       chipcenter - Feature Centering Tool for ChIP-seq data analysis

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>chipcenter</b> <b>[</b> <u>options</u> <b>]</b> <b>[</b> <b>-f</b> <u>&lt;feature</u> <u>name&gt;</u> <b>]</b> <b>-s</b> <u>&lt;shift&gt;</u> <b>[</b> <u>&lt;</u> <b>]</b> <b>[</b> <u>SGA</u> <u>file</u> <b>]</b>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>chipcenter</b>  reads  a  ChIP-seq  data  file (or from stdin [&lt;]) in SGA format (&lt;SGA file&gt;), and shifts (by
       &lt;shift&gt;) ChIP-tag positions corresponding to a specific feature (&lt;feature name&gt;) to the estimated center-
       positions of DNA fragments. If no feature specification is set, the program accepts all oriented lines of
       the input SGA. Strandless features are ignored. If -r &lt;new feature&gt; is specified, the  feature  field  is
       replaced with the &lt;new feature&gt; string.

       Launching <b>chipcenter</b> without any arguments will print the options list, along with their default values.

       The  &lt;feature&gt;  parameter  is  a  name  that  corresponds  to the second field of the SGA file.  It might
       optionally include the strand specification (+|-).  If no feature  is  given  then  all  input  tags  are
       processed.

</pre><h4><b>OPTIONS</b></h4><pre>
       -c &lt;cut-off&gt;
              A value can be specified as a cut-off for the input tag counts.

              This parameter is optional. Its default value is 1.

       -d     Show debug info.

       -f &lt;feature name&gt;
              This  parameter  is  used  to select all or a sub-set of chIP-seq input tags.  The feature name is
              specified in the second field of the SGA-formatted input file.

              If no feature name is given, then all features are selected.

       -h     Show the usage message.

       -r &lt;new feature&gt;
              It defines a new feature name for feature replacement.

       -s &lt;shift&gt;
              It defines the relative shift (in bp) of observed ChIP-tags to estimated center-positions  of  DNA
              fragments.

              This parameter is mandatory.

       -z     Set (output) strand to zero.

              This parameter is optional.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/chipcor.1.html">chipcor</a></b>(1), <b><a href="../man1/chipextract.1.html">chipextract</a></b>(1), <b><a href="../man1/chippeak.1.html">chippeak</a></b>(1) <b><a href="../man1/chippart.1.html">chippart</a></b>(1) <b><a href="../man1/chipscore.1.html">chipscore</a></b>(1),

Bioinformatics                                      July 2015                                      <u><a href="../man1/chipcenter.1.html">chipcenter</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>