<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>htseq-count - Count the number of reads in a SAM alignment file that map to GFF features</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/python3-htseq">python3-htseq_2.0.9+dfsg-1build2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       htseq-count - Count the number of reads in a SAM alignment file that map to GFF features

       Given  a file with aligned sequencing reads and a list of genomic features, a common task is to count how
       many reads map to each feature.

       A feature is here an interval (i.e., a range of positions) on a chromosome or a union of such intervals.

       In the case of RNA-Seq, the features are typically genes, where each gene is considered here as the union
       of all its exons. One may also consider each exon as a feature, e.g., in order to check  for  alternative
       splicing.  For comparative ChIP-Seq, the features might be binding region from a pre-determined list.

       Special  care  must  be  taken  to  decide how to deal with reads that overlap more than one feature. The
       <b>htseq-count</b> script allows one to choose between three modes. Of course, if none of these fits your needs,
       you can write your own script with HTSeq. See the chapter tour for a step-by-step guide on how to do  so.
       See also the FAQ at the end, if the following explanation seems to technical.

       The three overlap resolution modes of <b>htseq-count</b> work as follows. For each position <u>i</u> in the read, a set
       <u>S(i)</u>  is  defined  as  the set of all features overlapping position <u>i</u>. Then, consider the set <u>S</u>, which is
       (with <u>i</u> running through all position within the read or a read pair)

       • the union of all the sets <u>S(i)</u> for mode <b>union</b>. This mode is recommended for most use cases.

       • the intersection of all the sets <u>S(i)</u> for mode <b>intersection-strict</b>.

       • the intersection of all non-empty sets <u>S(i)</u> for mode <b>intersection-nonempty</b>.

       If <u>S</u> contains precisely one feature, the read (or read pair) is counted for this feature. If it  contains
       more  than  one  feature,  the  read  (or  read  pair)  is  counted as <b>ambiguous</b> (and not counted for any
       features), and if <b>S</b> is empty, the read (or read pair) is counted as <b>no_feature</b>.

       The following figure illustrates the effect of these three modes: [image]

</pre><h4><b>USAGE</b></h4><pre>
       After you have installed HTSeq (see install), you can run <b>htseq-count</b> from the command line:

          htseq-count [options] &lt;alignment_file&gt; &lt;gff_file&gt;

       If the file <b>htseq-qa</b> is not in your path, you can, alternatively, call the script with

          python -m HTSeq.scripts.count [options] &lt;alignment_file&gt; &lt;gff_file&gt;

       The <b>&lt;alignment_file&gt;</b> contains the aligned reads in the SAM format. (Note that the <u>SAMtools</u>  contain  Perl
       scripts  to  convert  most  alignment formats to SAM.)  Make sure to use a splicing-aware aligner such as
       TopHat. HTSeq-count makes full use of the information in the CIGAR field.

       To read from standard input, use <b>-</b> as <b>&lt;alignment_file&gt;</b>.

       If you have paired-end data, pay attention to the <b>-r</b> option described below.

       The <b>&lt;gff_file&gt;</b> contains the features in the <u>GFF</u> <u>format</u>.

       The script outputs a table with counts for each feature, followed by the special  counters,  which  count
       reads  that  were  not counted for any feature for various reasons. The names of the special counters all
       start with a double underscore, to facilitate filtering. (Note: The  double  unscore  was  absent  up  to
       version 0.5.4). The special counters are:

       • <b>__no_feature</b>:  reads  (or  read  pairs)  which could not be assigned to any feature (set <u>S</u> as described
         above was empty).

       • <b>__ambiguous</b>: reads (or read pairs) which could have been assigned to more than one  feature  and  hence
         were not counted for any of these (set <u>S</u> had more than one element).

       • <b>__too_low_aQual</b>: reads (or read pairs) which were skipped due to the <b>-a</b> option, see below

       • <b>__not_aligned</b>: reads (or read pairs) in the SAM file without alignment

       • <b>__alignment_not_unique</b>:  reads  (or read pairs) with more than one reported alignment.  These reads are
         recognized from the <b>NH</b> optional SAM field tag.  (If the aligner  does  not  set  this  field,  multiply
         aligned reads will be counted multiple times, unless they getv filtered out by due to the <b>-a</b> option.)

       <u>Important:</u>  The  default  for  strandedness  is  <u>yes</u>.  If  your  RNA-Seq  data  has  not been made with a
       strand-specific protocol, this causes half of the reads to be lost.  Hence, make sure to set  the  option
       <b>--stranded=no</b> unless you have strand-specific data!

   <b>Options</b>
       <b>-f</b> <b>&lt;format&gt;,</b> <b>--format=&lt;format&gt;</b>
              Format  of  the  input  data. Possible values are <b>sam</b> (for text SAM files) and <b>bam</b> (for binary BAM
              files). Default is <b>sam</b>.

       <b>-r</b> <b>&lt;order&gt;,</b> <b>--order=&lt;order&gt;</b>
              For paired-end data, the alignment have to be sorted either by read name or by alignment position.
              If your data is not sorted, use the <b>samtools</b> <b>sort</b> function  of  <b>samtools</b>  to  sort  it.  Use  this
              option,  with  <b>name</b> or <b>pos</b> for <b>&lt;order&gt;</b> to indicate how the input data has been sorted. The default
              is <b>name</b>.

              If <b>name</b> is indicated, <b>htseq-count</b> expects all the alignments for the reads of a given read pair to
              appear in adjacent records in the input data.   For  <b>pos</b>,  this  is  not  expected;  rather,  read
              alignments  whose  mate  alignment have not yet been seen are kept in a buffer in memory until the
              mate is found.  While, strictly speaking, the latter will also work with  unsorted  data,  sorting
              ensures  that most alignment mates appear close to each other in the data and hence the  buffer is
              much less likely to overflow.

       <b>-s</b> <b>&lt;yes/no/reverse&gt;,</b> <b>--stranded=&lt;yes/no/reverse&gt;</b>
              whether the data is from a strand-specific assay (default: <b>yes</b>)

              For <b>stranded=no</b>, a read is considered overlapping with a  feature  regardless  of  whether  it  is
              mapped  to the same or the opposite strand as the feature.  For <b>stranded=yes</b> and single-end reads,
              the read has to be mapped to the same strand as the feature. For paired-end reads, the first  read
              has  to  be  on the same strand and the second read on the opposite strand.  For <b>stranded=reverse</b>,
              these rules are reversed.

       <b>-a</b> <b>&lt;minaqual&gt;,</b> <b>--a=&lt;minaqual&gt;</b>
              skip all reads with alignment quality lower than the given minimum value (default: 10 — Note:  the
              default used to be 0 until version 0.5.4.)

       <b>-t</b> <b>&lt;feature</b> <b>type&gt;,</b> <b>--type=&lt;feature</b> <b>type&gt;</b>
              feature type (3rd column in GFF file) to be used, all features of other type are ignored (default,
              suitable for RNA-Seq analysis using an <u>Ensembl</u> <u>GTF</u> file: <b>exon</b>)

       <b>-i</b> <b>&lt;id</b> <b>attribute&gt;,</b> <b>--idattr=&lt;id</b> <b>attribute&gt;</b>
              GFF  attribute  to  be  used  as  feature  ID.  Several GFF lines with the same feature ID will be
              considered as parts of the same feature. The feature ID is used to  identity  the  counts  in  the
              output table. The default, suitable for RNA-Seq analysis using an Ensembl GTF file, is <b>gene_id</b>.

       <b>-m</b> <b>&lt;mode&gt;,</b> <b>--mode=&lt;mode&gt;</b>
              Mode  to  handle  reads  overlapping  more than one feature. Possible values for <u>&lt;mode&gt;</u> are <b>union</b>,
              <b>intersection-strict</b> and <b>intersection-nonempty</b> (default: <b>union</b>)

       <b>-o</b> <b>&lt;samout&gt;,</b> <b>--samout=&lt;samout&gt;</b>
              write out all SAM alignment records into an output SAM file called &lt;samout&gt;, annotating each  line
              with its assignment to a feature or a special counter (as an optional field with tag ‘XF’)

       <b>-q,</b> <b>--quiet</b>
              suppress progress report and warnings

       <b>-h,</b> <b>--help</b>
              Show a usage summary and exit

   <b>Frequenctly</b> <b>asked</b> <b>questions</b>
       <u>My</u> <u>shell</u> <u>reports</u> <u>“command</u> <u>not</u> <u>found”</u> <u>when</u> <u>I</u> <u>try</u> <u>to</u> <u>run</u> <u>“htseq-count”.</u> <u>How</u> <u>can</u> <u>I</u> <u>launch</u> <u>the</u> <u>script?</u>
              The  file “htseq-count” has to be in the system’s <u>search</u> <u>path</u>. By default, Python places it in its
              script directory, which you have to add to your search path. A  maybe  easier  alternative  is  to
              write <b>python</b> <b>-m</b> <b>HTSeq.scripts.count</b> instead of <b>htseq-count</b>, followed by the options and arguments,
              which will launch the htseq-count script as well.

       <u>Why</u> <u>are</u> <u>multi-mapping</u> <u>reads</u> <u>and</u> <u>reads</u> <u>overlapping</u> <u>multiple</u> <u>features</u> <u>discarded</u> <u>rather</u> <u>than</u> <u>counted</u> <u>for</u>
       <u>each</u> <u>feature?</u>
              The  primary  intended  use  case  for  <b>htseq-count</b> is <u>differential</u> expression analysis, where one
              compares the expression of the same gene across samples and not the expression of different  genes
              within a sample. Now, consider two genes, which share a stretch of common sequence such that for a
              read mapping to this stretch, the aligner cannot decide which of the two genes the read originated
              from and hence reports a multiple alignment. If we discard all such reads, we undercount the total
              output  of  the genes, but the <u>ratio</u> of expression strength (the “fold change”) between samples or
              experimental condition will still be correct, because we discard the same fratcion of reads in all
              samples.  On  the  other  hand,  if  we  counted  these  reads  for  both  genes,   a   subsequent
              diffential-expression  analysis  might  find false positives: Even if only one of the gene changes
              increases its expression in reaction to treatment, the additional read caused  by  this  would  be
              counted for both genes, giving the wrong appearance that both genes reacted to the treatment.

       <u>I</u> <u>have</u> <u>used</u> <u>a</u> <u>GTF</u> <u>file</u> <u>generated</u> <u>by</u> <u>the</u> <u>Table</u> <u>Browser</u> <u>function</u> <u>of</u> <u>the</u> <u>UCSC</u> <u>Genome</u> <u>Browser,</u> <u>and</u> <u>most</u> <u>reads</u>
       <u>are</u> <u>counted</u> <u>as</u> <u>ambiguous.</u> <u>Why?</u>
              In  these  files,  the  <b>gene_id</b> attribute incorrectly contains the same value as the <b>transcript_id</b>
              attribute and hence a different value for each transcript of the same gene. Hence, if a read  maps
              to  an exon shared by several transcripts of the same gene, this will appear to <b>htseq-count</b> as and
              overlap with several genes. Therefore, these GTF files cannot be used as is.  Either  correct  the
              incorrect <b>gene_id</b> attributes with a suitable script, or use a GTF file from a different source.

       <u>Can</u> <u>I</u> <u>use</u> <u>htseq-count</u> <u>to</u> <u>count</u> <u>reads</u> <u>mapping</u> <u>to</u> <u>transcripts</u> <u>rather</u> <u>than</u> <u>genes?</u>
              In  principle,  you  could  instruct  htseq-count  to  count  for  each  of  a  gene’s  transcript
              individually, by specifying <b>--idattr</b> <b>transcript_id</b>. However, all reads mapping to exons shared  by
              several  transcripts  will  then be considered ambiguous. (See second question.) Counting them for
              each transcript that contains the exons would be possible but makes little sense for  typical  use
              cases.  (See first question.) If you want to perform differential expression analysis on the level
              of individual transcripts, maybe ahve a look at <u>our</u> <u>paper</u> <u>on</u> <u>DEXSeq</u> for a  discussion  on  why  we
              prefer performing such analyses on the level of exons instead.

       <u>For</u> <u>paired-end</u> <u>data,</u> <u>does</u> <u>htseq-count</u> <u>count</u> <u>reads</u> <u>or</u> <u>read</u> <u>pairs?</u>
              Read pairs. The script is designed to count “units of evidence” for gene expression. If both mates
              map  to  the  same  gene,  this still only shows that one cDNA fragment originated from that gene.
              Hence, it should be counted only once.

       <u>What</u> <u>happens</u> <u>if</u> <u>the</u> <u>two</u> <u>reads</u> <u>in</u> <u>a</u> <u>pair</u> <u>overlap</u> <u>two</u> <u>different</u> <u>features?</u>
              The same as if one read overlaps two features: The read or read pair is counted as ambiguous.

       <u>What</u> <u>happened</u> <u>if</u> <u>the</u> <u>mate</u> <u>of</u> <u>an</u> <u>aligned</u> <u>read</u> <u>is</u> <u>not</u> <u>aligned?</u>
              For the default mode “union”, only the aligned read determines how the read pair is  counted.  For
              the other modes, see their description.

       <u>Most</u> <u>of</u> <u>my</u> <u>RNA-Seq</u> <u>reads</u> <u>are</u> <u>counted</u> <u>as</u> <u>``</u><b>__</b><u>no_feature``.</u> <u>What</u> <u>could</u> <u>have</u> <u>gone</u> <u>wrong?</u>
              Common  causes  include: - The <b>--stranded</b> option was set wrongly. Use a genome browser (e.g., IGV)
              to check.  - The GTF file uses coordinates from another reference assembly as the SAM file.  - The
              chromosome names differ between GTF and SAM file (e.g., <b>chr1</b> in one file and jsut <b>1</b> in the other).

       <u>Which</u> <u>overlap</u> <u>mode</u> <u>should</u> <u>I</u> <u>use?</u>
              When I wrote <b>htseq-count</b>, I was not sure which option is best and  included  three  possibilities.
              Now,  several  years  later,  I  have  seen  very  few  cases where the default <b>union</b> would not be
              appropriate and hence tend to recommend to just stick to <b>union</b>.

       <u>I</u> <u>have</u> <u>a</u> <u>GTF</u> <u>file?</u> <u>How</u> <u>do</u> <u>I</u> <u>convert</u> <u>it</u> <u>to</u> <u>GFF?</u>
              No need to do that, because GTF is a tightening of the GFF format. Hence, all GTF  files  are  GFF
              files, too.  By default, htseq-count expects a GTF file.

       <u>I</u> <u>have</u> <u>a</u> <u>GFF</u> <u>file,</u> <u>not</u> <u>a</u> <u>GTF</u> <u>file.</u> <u>How</u> <u>can</u> <u>I</u> <u>use</u> <u>it</u> <u>to</u> <u>count</u> <u>RNA-Seq</u> <u>reads?</u>
              The  GTF  format specifies, inter alia, that exons are marked by the word <b>exon</b> in the third column
              and that the gene ID is given in an attribute named <b>gene_id</b>, and htseq-count expects  these  words
              to  be  used  by  default. If you GFF file uses a word other than <b>exon</b> in its third column to mark
              lines describing exons, notify <b>htseq-count</b> using the <b>--type</b> option. If the name of  the  attribute
              containing  the  gene  ID  for  exon  lines  is  not <b>gene_id</b>, use the <b>--idattr</b>. Often, its is, for
              example, <b>Parent</b>, <b>GeneID</b> or <b>ID</b>. Make sure it is the gene ID and not the exon ID.

       <u>How</u> <u>can</u> <u>I</u> <u>count</u> <u>overlaps</u> <u>with</u> <u>features</u> <u>other</u> <u>than</u> <u>genes/exons?</u>
              If you have GFF file listing your features, use it together with the <b>--type</b> and <b>--idattr</b>  options.
              If  your  feature  intervals  need  to  be  computed, you are probably better off writing your own
              counting script (provided you have some knowledge of Python). Follow the  tutorial  in  the  other
              pages of this documentation to see how to use HTSeq for this.

       <u>How</u> <u>should</u> <u>I</u> <u>cite</u> <u>htseq-count</u> <u>in</u> <u>a</u> <u>publication?</u>
              Please  cite HTSeq as follows: S Anders, T P Pyl, W Huber: <u>HTSeq</u> <u>—</u> <u>A</u> <u>Python</u> <u>framework</u> <u>to</u> <u>work</u> <u>with</u>
              <u>high-throughput</u> <u>sequencing</u> <u>data</u>. bioRxiv 2014. <u>doi:</u> <u>10.1101/002824</u>.  (This is a preprint currently
              under review. We will replace this  with  the  reference  to  the  final  published  version  once
              available.)

</pre><h4><b>AUTHOR</b></h4><pre>
       Simon Anders

</pre><h4><b>COPYRIGHT</b></h4><pre>
       2017, Simon Anders

0.6.1p1                                           Oct 01, 2017                                    <u><a href="../man1/HTSEQ-COUNT.1.html">HTSEQ-COUNT</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>