<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::Spidey::Results - Results of a Spidey run</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::Spidey::Results - Results of a Spidey run

</pre><h4><b>SYNOPSIS</b></h4><pre>
          use Bio::Tools::Spidey::Results;
               my $spidey = Bio::Tools::Spidey::Results-&gt;new(-file =&gt; 'result.spidey' );

               # or

               my $spidey = Bio::Tools::Spidey::Results-&gt;new( -fh   =&gt; \*INPUT );

               # get the exons before doing anything else
               my $exonset = $spidey-&gt;next_exonset();

               # parse the results
               my @exons = $exonset-&gt;sub_SeqFeature();
               print "Total no of Exons: ", scalar(@exons), "\n";

               print "Genomic sequence length: ", $spidey-&gt;genomic_dna_length(), "\n";

               # $exonset is-a Bio::SeqFeature::Generic with Bio::Tools::Spidey::Exons
               # as sub features
               print "Delimited on sequence ", $exonset-&gt;seq_id(), " from ",
                       $exonset-&gt;start(), " to ", $exonset-&gt;end(), "\n";

               foreach my $exon ( $exonset-&gt;sub_SeqFeature() ) {
                       # $exon is-a Bio::SeqFeature::FeaturePair
                       print "Exon from ", $exon-&gt;start, " to ", $exon-&gt;end,
                               " on strand ", $exon-&gt;strand(), "\n";
                       # you can get out what it matched using the est_hit attribute
                       my $homol = $exon-&gt;est_hit();
                       print "Matched to sequence ", $homol-&gt;seq_id,
                               " at ", $homol-&gt;start," to ", $homol-&gt;end, "\n";
               }

               # essential if you gave a filename at initialization (otherwise
               # the file stays open)
               $spidey-&gt;close();

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The spidey module provides a parser and results object for spidey output. The spidey results are
       specialised types of SeqFeatures, meaning you can add them to AnnSeq objects fine, and manipulate them in
       the "normal" seqfeature manner.

       The spidey Exon objects are Bio::SeqFeature::FeaturePair inherited objects. The $esthit =
       $exon-&gt;<b>est_hit()</b> is the alignment as a feature on the matching object (normally, a cDNA), in which the
       start/end points are where the hit lies.

       To make this module work sensibly you need to run

            spidey -i genomic.fasta -m cDNA.fasta

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Ryan</b> <b>Golhar</b></h4><pre>
       Email <a href="mailto:golharam@umdnj.edu">golharam@umdnj.edu</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>analysis_method</b>
        Usage     : $spidey-&gt;analysis_method();
        Purpose   : Inherited method. Overridden to ensure that the name matches
                    /Spidey/i.
        Returns   : String
        Argument  : n/a

   <b>parse_next_alignment</b>
        Title   : parse_next_alignment
        Usage   : @exons = $spidey_result-&gt;parse_next_alignment;
                  foreach $exon (@exons) {
                      # do something
                  }
        Function: Parses the next alignment of the Spidey result file and returns the
                  found exons as an array of Bio::Tools::Spidey::Exon objects. Call
                  this method repeatedly until an empty array is returned to get the
                  results for all alignments.
        Example :
        Returns : An array of Bio::Tools::Spidey::Exon objects
        Args    :

   <b>next_exonset</b>
         Title   : next_exonset
         Usage   : $exonset = $spidey_result-&gt;parse_next_exonset;
                print "Exons start at ", $exonset-&gt;start(),
               "and end at ", $exonset-&gt;end(), "\n";
                for $exon ($exonset-&gt;sub_SeqFeature()) {
                   # do something
                }
         Function: Parses the next alignment of the Spidey result file and returns the
              set of exons as a container of features. The container is itself
              a Bio::SeqFeature::Generic object, with the Bio::Tools::Spidey::Exon
              objects as sub features. Start, end, and strand of the container
              will represent the total region covered by the exons of this set.

             See the documentation of parse_next_alignment() for further
             reference about parsing and how the information is stored.
        Example :
        Returns : An Bio::SeqFeature::Generic object holding Bio::Tools::Spidey::Exon
                 objects as sub features.
        Args    :

   <b>next_feature</b>
         Title   : next_feature
         Usage   : while($exonset = $spidey-&gt;next_feature()) {
                   # do something
                  }
         Function: Does the same as L&lt;next_exonset()&gt;. See there for documentation of
             the functionality. Call this method repeatedly until FALSE is
             returned.

             The returned object is actually a SeqFeatureI implementing object.
             This method is required for classes implementing the
             SeqAnalysisParserI interface, and is merely an alias for
             next_exonset() at present.

         Example :
         Returns : A Bio::SeqFeature::Generic object.
         Args    :

   <b>genomic_dna_length</b>
           Title   : genomic_dna_length
           Usage   : $spidey-&gt;genomic_dna_length();
           Function: Returns the length of the genomic DNA used in this Spidey result
           Example :
           Returns : An integer value.
           Args    :

   <b>splicesites</b>
           Title   : splicesites
           Usage   : $spidey-&gt;splicesites();
           Function: Returns the number of splice sites found in this Spidey result
           Example :
           Returns : An integer value.
           Args    :

   <b>est_coverage</b>
           Title   : est_coverage
           Usage   : $spidey-&gt;est_coverage();
           Function: Returns the percent of est coverage in this Spidey result
           Example :
           Returns : An integer value.
           Args    :

   <b>overall_percentage_id</b>
           Title   : overall_percentage_id
           Usage   : $spidey-&gt;overall_percentage_id();
           Function: Returns the overall percent id in this Spidey result
           Example :
           Returns : An float value.
           Args    :

   <b>missing_mrna_ends</b>
           Title   : missing_mrna_ends
           Usage   : $spidey-&gt;missing_mrna_ends();
           Function: Returns left/right/neither from Spidey
           Example :
           Returns : A string value.
           Args    :

perl v5.32.1                                       2021-08-15                   <u>Bio::Tools::Spidey::<a href="../man3pm/Results.3pm.html">Results</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>