<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>sickle - windowed adaptive trimming tool for FASTQ files using quality</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/sickle">sickle_1.33+git20150314.f3d6ae3-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       sickle - windowed adaptive trimming tool for FASTQ files using quality

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>sickle</b> <u>&lt;command&gt;</u> [<u>options</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       ## Usage

       Sickle has two modes to work with both paired-end and single-end reads: `sickle se` and `sickle pe`.

       Running sickle by itself will print the help:

           sickle

       Running sickle with either the "se" or "pe" commands will give help specific to those commands:

           sickle se
           sickle pe

       ### Sickle Single End (`sickle se`)

       `sickle se` takes an input fastq file and outputs a trimmed version of that file.  It also has options to
       change  the  length  and  quality  thresholds for trimming, as well as disabling 5'-trimming and enabling
       truncation of sequences with Ns.

       #### Examples

           sickle se -f input_file.fastq -t illumina -o trimmed_output_file.fastq
           sickle se -f input_file.fastq -t illumina -o trimmed_output_file.fastq -q 33 -l 40
           sickle se -f input_file.fastq -t illumina -o trimmed_output_file.fastq -x -n
           sickle se -t sanger -g -f input_file.fastq -o trimmed_output_file.fastq.gz
           sickle se --fastq-file input_file.fastq --qual-type sanger --output-file trimmed_output_file.fastq

       ### Sickle Paired End (`sickle pe`)

       `sickle pe` can operate with two types of input.  First, it can take two paired-end files  as  input  and
       outputs  two  trimmed paired-end files as well as a "singles" file.  The second form starts with a single
       combined input file of reads where you have already interleaved the reads from the  sequencer.   In  this
       form, you also supply a single output file name as well as a "singles" file.  The "singles" file contains
       reads  that  passed filter in either the forward or reverse direction, but not the other.  Finally, there
       is an option (-M) to only produce one interleaved output file where any reads that did  not  pass  filter
       will be output as a FastQ record with a single "N" (whose quality value is the lowest possible based upon
       the  quality  type),  thus  preserving the paired nature of the data.  You can also change the length and
       quality thresholds for trimming, as well as disable 5'-trimming and enable truncation of  sequences  with
       Ns.

       #### Examples

           sickle  pe  -f  input_file1.fastq -r input_file2.fastq -t sanger     -o trimmed_output_file1.fastq -p
       trimmed_output_file2.fastq     -s trimmed_singles_file.fastq

           sickle pe -f input_file1.fastq -r input_file2.fastq -t sanger      -o  trimmed_output_file1.fastq  -p
       trimmed_output_file2.fastq     -s trimmed_singles_file.fastq -q 12 -l 15

           sickle  pe  -f  input_file1.fastq -r input_file2.fastq -t sanger     -o trimmed_output_file1.fastq -p
       trimmed_output_file2.fastq     -s trimmed_singles_file.fastq -n

           sickle pe -c combo.fastq -t sanger -m combo_trimmed.fastq     -s trimmed_singles_file.fastq -n

           sickle pe -t sanger -g -f input_file1.fastq -r input_file2.fastq     -o trimmed_output_file1.fastq.gz
       -p trimmed_output_file2.fastq.gz     -s trimmed_singles_file.fastq.gz

           sickle pe -c combo.fastq -t sanger -M combo_trimmed_all.fastq

           sickle pe --pe-file1 input_file1.fastq --pe-file2 input_file2.fastq --qual-type sanger      --output-
       pe1     trimmed_output_file1.fastq     --output-pe2     trimmed_output_file2.fastq        --output-single
       trimmed_singles_file.fastq

       Command: pe      paired-end sequence trimming se      single-end sequence trimming

       <b>--help</b>, display this help and exit <b>--version</b>, output version information and exit

</pre><h4><b>AUTHOR</b></h4><pre>
       Written by Nikhil Joshi, UC Davis Bioinformatics Core

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright Â© 2011 The Regents of University of California, Davis Campus.   sickle  is  free  software  and
       comes with ABSOLUTELY NO WARRANTY.  Distributed under the MIT License.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       The full documentation for <b>sickle</b> is maintained as a Texinfo manual.  If the <b>info</b> and <b>sickle</b> programs are
       properly installed at your site, the command

              <b>info</b> <b>sickle</b>

       should give you access to the complete manual.

sickle version 1.33                                 June 2020                                          <u><a href="../man1/SICKLE.1.html">SICKLE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>