<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>rabema_build_gold_standard - RABEMA Gold Standard Builder</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/seqan-apps">seqan-apps_2.4.0+dfsg-16_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       rabema_build_gold_standard - RABEMA Gold Standard Builder

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>rabema_build_gold_standard</b> [<u>OPTIONS</u>] <b>--out-gsi</b> <u>OUT.gsi</u> <b>--reference</b> <u>REF.fa</u> <b>--in-bam</b> <u>PERFECT.{sam,bam}</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This  program  allows  one  to  build  a RABEMA gold standard.  The input is a reference FASTA file and a
       perfect SAM/BAM map (e.g. created using RazerS 3 in full-sensitivity mode).

       The input SAM/BAM file must be <u>sorted</u> <u>by</u>  <u>coordinate</u>.   The  program  will  create  a  FASTA  index  file
       <u>REF.fa.fai</u> for fast random access to the reference.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              Display the help message.

       <b>--version</b>
              Display version information.

       <b>-v</b>, <b>--verbose</b>
              Enable verbose output.

       <b>-vv</b>, <b>--very-verbose</b>
              Enable even more verbose output.

   <b>Input</b> <b>/</b> <b>Output:</b>
       <b>-o</b>, <b>--out-gsi</b> <u>OUTPUT_FILE</u>
              Path  to  write  the  resulting  GSI  file  to. Valid filetype is: <u>.gsi[.*]</u>, where * is any of the
              following extensions: <u>gz</u> for transparent (de)compression.

       <b>-r</b>, <b>--reference</b> <u>INPUT_FILE</u>
              Path to load reference FASTA from. Valid filetypes are: <u>.sam[.*]</u>,  <u>.raw[.*]</u>,  <u>.gbk[.*]</u>,  <u>.frn[.*]</u>,
              <u>.fq[.*]</u>, <u>.fna[.*]</u>, <u>.ffn[.*]</u>, <u>.fastq[.*]</u>, <u>.fasta[.*]</u>, <u>.faa[.*]</u>, <u>.fa[.*]</u>, <u>.embl[.*]</u>, and <u>.bam</u>, where
              * is any of the following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent (de)compression.

       <b>-b</b>, <b>--in-bam</b> <u>INPUT_FILE</u>
              Path  to  load the "perfect" SAM/BAM file from. Valid filetypes are: <u>.sam[.*]</u> and <u>.bam</u>, where * is
              any of the following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent (de)compression.

   <b>Gold</b> <b>Standard</b> <b>Parameters:</b>
       <b>--oracle-mode</b>
              Enable oracle mode.  This is used for simulated data when the input SAM/BAM file gives exactly one
              position that is considered as the true sample position.

       <b>--match-N</b>
              When set, N matches all characters without penalty.

       <b>--distance-metric</b> <u>STRING</u>
              Set distance metric.  Valid values: hamming, edit.   Default:  edit.  One  of  <u>hamming</u>  and  <u>edit</u>.
              Default: <u>edit</u>.

       <b>-e</b>, <b>--max-error</b> <u>INTEGER</u>
              Maximal  error  rate  to  build  gold  standard  for in percent.  This parameter is an integer and
              relative to the read length.  In case of oracle mode, the error rate for the read at the  sampling
              position is used and <u>RATE</u> is used as a cutoff threshold. Default: <u>0</u>.

</pre><h4><b>RETURN</b> <b>VALUES</b></h4><pre>
       A return value of 0 indicates success, any other value indicates an error.

</pre><h4><b>EXAMPLES</b></h4><pre>
       <b>rabema_build_gold_standard</b> <b>-e</b> <u>4</u> <b>-o</b> <u>OUT.gsi</u> <b>-s</b> <u>IN.sam</u> <b>-r</b> <u>REF.fa</u>
              Build gold standard from a SAM file <u>IN.sam</u> with all mapping locations and a FASTA reference <u>REF.fa</u>
              to GSI file <u>OUT.gsi</u> with a maximal error rate of <u>4</u>.

       <b>rabema_build_gold_standard</b> <b>--distance-metric</b> <u>edit</u> <b>-e</b> <u>4</u> <b>-o</b> <u>OUT.gsi</u> <b>-b</b> <u>IN.bam</u> <b>-r</b> <u>REF.fa</u>
              Same as above, but using Hamming instead of edit distance and BAM as the input.

       <b>rabema_build_gold_standard</b> <b>--oracle-mode</b> <b>-o</b> <u>OUT.gsi</u> <b>-s</b> <u>IN.sam</u> <b>-r</b> <u>REF.fa</u>
              Build gold standard from a SAM file <u>IN.sam</u> with the original sample position, e.g.  as exported by
              read simulator Mason.

</pre><h4><b>MEMORY</b> <b>REQUIREMENTS</b></h4><pre>
       From  version  1.1,  great  care has been taken to keep the memory requirements as low as possible. There
       memory required is two times the size of the largest chromosome plus some constant memory for each match.

       For example, the memory usage for 100bp human genome reads at 5% error rate was 1.7GB. Of  this,  roughly
       400GB came from the chromosome and 1.3GB from the matches.

</pre><h4><b>REFERENCES</b></h4><pre>
       M.  Holtgrewe,  A.-K.  Emde,  D.  Weese and K. Reinert.  A Novel And Well-Defined Benchmarking Method For
       Second Generation Read Mapping, BMC Bioinformatics 2011, 12:210.

       <u><a href="http://www.seqan.de/rabema">http://www.seqan.de/rabema</a></u>
              RABEMA Homepage

       <u><a href="http://www.seqan.de/mason">http://www.seqan.de/mason</a></u>
              Mason Homepage

rabema_build_gold_standard 1.2.10 [tarball]                                        <u><a href="../man1/RABEMA_BUILD_GOLD_STANDARD.1.html">RABEMA_BUILD_GOLD_STANDARD</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>