<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>eagle - Haplotype phasing within a genotyped cohort or using a phased reference panel</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/bio-eagle">bio-eagle_2.4.1-3build2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       eagle - Haplotype phasing within a genotyped cohort or using a phased reference panel

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>eagle</b> [<u>options</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Eagle  estimates  haplotype phase either within a genotyped cohort or using a phased reference panel. The
       basic idea of the Eagle1 algorithm is to harness identity-by-descent  among  distant  relatives—which  is
       pervasive  at  very  large  sample  sizes but rare among smaller numbers of samples—to rapidly call phase
       using a fast scoring approach. In contrast, the Eagle2 algorithm  analyzes  a  full  probabilistic  model
       similar to the diploid Li-Stephens model used by previous HMM-based methods.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              Print this help

       <b>--geneticMapFile</b> arg
              HapMap genetic map provided with download: tables/genetic_map_hg##.txt.gz

       <b>--outPrefix</b> arg
              prefix for output files

       <b>--numThreads</b> arg (=1)
              number of computational threads

   <b>Input</b> <b>options</b> <b>for</b> <b>phasing</b> <b>without</b> <b>a</b> <b>reference:</b>
       <b>--bfile</b> arg
              prefix of PLINK .fam, .bim, .bed files

       <b>--bfilegz</b> arg
              prefix of PLINK .fam.gz, .bim.gz, .bed.gz files

       <b>--fam</b> arg
              PLINK .fam file (note: file names ending in .gz are auto-decompressed)

       <b>--bim</b> arg
              PLINK .bim file

       <b>--bed</b> arg
              PLINK .bed file

       <b>--vcf</b> arg
              [compressed] VCF/BCF file containing input genotypes

       <b>--remove</b> arg
              file(s) listing individuals to ignore (no header; FID IID must be first two columns)

       <b>--exclude</b> arg
              file(s) listing SNPs to ignore (no header; SNP ID must be first column)

       <b>--maxMissingPerSnp</b> arg (=0.1)
              QC filter: max missing rate per SNP

       <b>--maxMissingPerIndiv</b> arg (=0.1)
              QC filter: max missing rate per person

   <b>Input/output</b> <b>options</b> <b>for</b> <b>phasing</b> <b>using</b> <b>a</b> <b>reference</b> <b>panel:</b>
       <b>--vcfRef</b> arg
              tabix-indexed [compressed] VCF/BCF file for reference haplotypes

       <b>--vcfTarget</b> arg
              tabix-indexed [compressed] VCF/BCF file for target genotypes

       <b>--vcfOutFormat</b> arg (=z)
              b|u|z|v: compressed BCF (b), uncomp BCF (u), compressed VCF (z), uncomp VCF (v)

       <b>--noImpMissing</b>
              disable imputation of missing .<a href="file:/.">/.</a> target genotypes

       <b>--allowRefAltSwap</b>
              allow swapping of REF/ALT in target vs. ref VCF

   <b>Region</b> <b>selection</b> <b>options:</b>
       <b>--chrom</b> arg (=0)
              chromosome to analyze (if input has many)

       <b>--bpStart</b> arg (=0)
              minimum base pair position to analyze

       <b>--bpEnd</b> arg (=1e9)
              maximum base pair position to analyze

       <b>--bpFlanking</b> arg (=0)
              (ref-mode only) flanking region to use during phasing but discard in output

   <b>Algorithm</b> <b>options:</b>
       <b>--Kpbwt</b> arg (=10000)
              number of conditioning haplotypes

       <b>--pbwtIters</b> arg (=0)
              number of PBWT phasing iterations (0=auto)

       <b>--expectIBDcM</b> arg (=2.0)
              expected length of haplotype copying (cM)

       <b>--histFactor</b> arg (=0)
              history length multiplier (0=auto)

       <b>--genoErrProb</b> arg (=0.003)
              estimated genotype error probability

       <b>--pbwtOnly</b>
              in non-ref mode, use only PBWT iters (automatic for sequence data)

       <b>--v1</b>   use Eagle1 phasing algorithm (instead of default Eagle2 algorithm)

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b>https://data.broadinstitute.org/alkesgroup/Eagle/</b>

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright © 2015-2016 Harvard University.  Distributed under the GNU GPLv3+ open source license.

2.3                                              September 2016                                         <u><a href="../man1/EAGLE.1.html">EAGLE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>