<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>FastQTL - Quantitative Trait Loci (QTL) mapper in cis for molecular phenotypes</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/fastqtl">fastqtl_2.184+v7+dfsg-4build4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       FastQTL - Quantitative Trait Loci (QTL) mapper in cis for molecular phenotypes

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>fastQTL</b> <b>[</b><u>options</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The  goal  of  FastQTL  is  to  identify  single-nucleotide  polymorphisms (SNPs) which are significantly
       associated with various molecular phenotypes  (i.e.  expression  of  known  genes,  cytosine  methylation
       levels,  etc).   It performs scans for all possible phenotype-variant pairs in cis (i.e. variants located
       within a specific window around  a  phenotype).   FastQTL  implements  a  new  permutation  scheme  (Beta
       approximation)  to accurately and rapidly correct for multiple-testing at both the genotype and phenotype
       levels.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Basic</b> <b>options</b>
       <b>--help</b>
              Print help about options.

       <b>--silent</b>
              Silent mode on terminal.

       <b>--seed</b> arg (=1459232241)
              Random number generator seed.  Useful to replicate runs of the software.

   <b>Input</b> <b>and</b> <b>Output</b> <b>files</b> <b>options</b>
       <b>-L,</b> <b>--log</b>
              Screen output is copied in this file.

       <b>-V,</b> <b>--vcf</b>
              Genotypes in VCF format.

       <b>-B,</b> <b>--bed</b>
              Phenotypes in BED format.

       <b>-C,</b> <b>--cov</b>
              Covariates in TXT format.

       <b>-G,</b> <b>--grp</b>
              Phenotype groups in TXT format.

       <b>-O,</b> <b>--out</b>
              Output file.

   <b>Exclusion</b> <b>and</b> <b>Inclusion</b> <b>files</b> <b>options</b>
       <b>--exclude-samples</b>
              List of samples to exclude.

       <b>--include-samples</b>
              List of samples to include.

       <b>--exclude-sites</b>
              List of sites to exclude.

       <b>--include-sites</b>
              List of sites to include.

       <b>--exclude-phenotypes</b>
              List of phenotypes to exclude.

       <b>--include-phenotypes</b>
              List of phenotypes to include.

       <b>--exclude-covariates</b>
              List of covariates to exclude.

       <b>--include-covariates</b>
              List of covariates to include.

   <b>Parameters</b>
       <b>--normal</b>
              To perform  quantile  normalization  on  the  phenotype  quantifications  to  make  them  normally
              distributed.  Implemented as the rntransform function of the GenABEL package.

       <b>-W,</b> <b>--window</b>
              Cis-window  size.   Default  values is 1Mb (1e6).  It means that all variants within 1e6 bp of the
              phenotype location (e.g. TSS) is analyzed.

       <b>-T,</b> <b>--threshold</b>
              To filter out all phenotype-variant pairs with a p-value above  the  specified  threshold  in  the
              output of a nominal pass.

   <b>Modes</b>
       <b>-P,</b> <b>--permute</b>
              Permutation pass to calculate corrected p-values for molecular phenotypes.

       <b>--psequence</b>
              Permutation sequence.

       <b>--map</b>  Map best QTL candidates per molecular phenotype.

       <b>--map-full</b>
              Scan full cis-window to discover independent signals.

       <b>--interaction</b>
              Test for interactions with variable specified in file.

   <b>Parallelization</b>
       <b>-K,</b> <b>--chunk</b>
              Specify which chunk needs to be processed.

       <b>--commands</b>
              Generates all commands.

       <b>-R,</b> <b>--region</b>
              Region of interest.

</pre><h4><b>EXAMPLE</b></h4><pre>
       for c in $(seq 1 256); do
            <b>fastQTL</b> <b>--vcf</b> genotypes.vcf.gz
                    <b>--bed</b> phenotypes.bed.gz
                    <b>--chunk</b> $c 256
                    <b>--permute</b> 1000
                    <b>--out</b> results.$c.txt.gz
       done
       zcat results.*.txt.gz | gzip -c &gt; fastqtl_expected_output.txt.gz

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="http://fastqtl.sourceforge.net/">http://fastqtl.sourceforge.net/</a></b>

</pre><h4><b>AUTHOR</b></h4><pre>
       FastQTL was written by Olivier Delaneau, Halit Ongen, Alfonso Buil and Manolis Dermitzakis.

       This  manual  page  was written by Dylan AÃ¯ssi &lt;<a href="mailto:bob.dybian@gmail.com">bob.dybian@gmail.com</a>&gt;, for the Debian project (but may be
       used by others).

2.184                                              March 2016                                         <u><a href="../man1/FastQTL.1.html">FastQTL</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>