<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Graphics::Glyph::wiggle_xyplot - An xyplot plot compatible with dense "wig"data</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-graphics-perl">libbio-graphics-perl_2.40-6_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Graphics::Glyph::wiggle_xyplot - An xyplot plot compatible with dense "wig"data

</pre><h4><b>SYNOPSIS</b></h4><pre>
         See &lt;Bio::Graphics::Panel&gt; and &lt;Bio::Graphics::Glyph&gt;.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This glyph works like the regular xyplot but takes value data in Bio::Graphics::Wiggle file format:

        reference = chr1
        ChipCHIP Feature1 1..10000 wigfile=./test.wig
        ChipCHIP Feature2 10001..20000 wigfile=./test.wig
        ChipCHIP Feature3 25001..35000 wigfile=./test.wig

       The "wigfile" attribute gives a relative or absolute pathname to a Bio::Graphics::Wiggle format file. The
       data consist of a packed binary representation of the values in the feature, using a constant step such
       as present in tiling array data. Wigfiles are created using the Bio::Graphics::Wiggle module or the
       wiggle2gff3.pl script, currently both part of the gbrowse package.

       Alternatively, you can place an array of quantitative data directly in the "wigdata" attribute. This can
       be an arrayref of quantitative data starting at feature start and ending at feature end, or the data
       string returned by Bio::Graphics::Wiggle-&gt;export_to_wif64($start,$end).

       The glyph also supports features which offer a <b>statistical_summary()</b> method, such as those from
       Bio::DB::BigWig. This method returns a hash of values, including validCount, maxVal, minVal, sumData, and
       sumSquares.  For each interval, a statistical score is generated from these values.  The mean, minimum,
       maximum, or sum of the values may be reported.

   <b>OPTIONS</b>
       In addition to all the xyplot glyph options, the following options are recognized:

          Name        Value        Description
          ----        -----        -----------

          basedir     path         Path to be used to resolve "wigfile" and "densefile"
                                       tags giving relative paths. Default is to use the
                                       current working directory. Absolute wigfile &amp;
                                       densefile paths will not be changed.

          autoscale   "local" or "global"
                                    If one or more of min_score and max_score options
                                    are absent, then these values will be calculated
                                    automatically. The "autoscale" option controls how
                                    the calculation is done. The "local" value will
                                    scale values according to the minimum and maximum
                                    values present in the window being graphed. "global"
                                    will use chromosome-wide statistics for the entire
                                    wiggle or dense file to find min and max values.

          smoothing   method name  Smoothing method: one of "mean", "max", "min" or "none"

          smoothing_window
                      integer      Number of values across which data should be smoothed.

          variance_band boolean    If true, draw a grey band across entire plot showing mean
                                      and +/- 1 standard deviation (for wig files only).

          bicolor_pivot
                      name         Where to pivot the two colors when drawing bicolor plots.
                                      Options are "mean" and "zero". A numeric value can
                                      also be provided.

          pos_color   color        When drawing bicolor plots, the fill color to use for values
                                     that are above the pivot point.

          neg_color   color        When drawing bicolor plots, the fill color to use for values
                                     that are below the pivot point.

          interval_method   method  When working with bigWig features that use the
                                    statistical_summary() method, define the method
                                    for reporting one or more scores within each
                                    interval. Options include "mean", "sum", "min",
                                    or "max". Default is "mean".

   <b>SPECIAL</b> <b>FEATURE</b> <b>TAGS</b>
       The glyph expects one or more of the following tags (attributes) in feature it renders:

          Name        Value        Description
          ----        -----        -----------

          wigfile     path name    Path to the Bio::Graphics::Wiggle file or object
                                   for quantitative values.

          wigdata     string       Data exported from a Bio::Graphics::Wiggle in WIF
                                   format using its export_to_wif64() method.

          densefile   path name    Path to a Bio::Graphics::DenseFeature object
                                      (deprecated)

          denseoffset integer      Integer offset to where the data begins in the
                                      Bio::Graphics::DenseFeature file (deprecated)

          densesize   integer      Integer size of the data in the Bio::Graphics::DenseFeature
                                      file (deprecated)

</pre><h4><b>BUGS</b></h4><pre>
       Please report them.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Bio::Graphics::Panel, Bio::Graphics::Glyph, Bio::Graphics::Glyph::arrow, Bio::Graphics::Glyph::cds,
       Bio::Graphics::Glyph::crossbox, Bio::Graphics::Glyph::diamond, Bio::Graphics::Glyph::dna,
       Bio::Graphics::Glyph::dot, Bio::Graphics::Glyph::ellipse, Bio::Graphics::Glyph::extending_arrow,
       Bio::Graphics::Glyph::generic, Bio::Graphics::Glyph::graded_segments,
       Bio::Graphics::Glyph::heterogeneous_segments, Bio::Graphics::Glyph::line,
       Bio::Graphics::Glyph::pinsertion, Bio::Graphics::Glyph::primers, Bio::Graphics::Glyph::rndrect,
       Bio::Graphics::Glyph::segments, Bio::Graphics::Glyph::ruler_arrow, Bio::Graphics::Glyph::toomany,
       Bio::Graphics::Glyph::transcript, Bio::Graphics::Glyph::transcript2, Bio::Graphics::Glyph::translation,
       Bio::Graphics::Glyph::allele_tower, Bio::DB::GFF, Bio::SeqI, Bio::SeqFeatureI, Bio::Das, GD

</pre><h4><b>AUTHOR</b></h4><pre>
       Lincoln Stein &lt;<a href="mailto:steinl@cshl.edu">steinl@cshl.edu</a>&gt;.

       Copyright (c) 2007 Cold Spring Harbor Laboratory

       This package and its accompanying libraries is free software; you can redistribute it and/or modify it
       under the terms of the GPL (either version 1, or at your option, any later version) or the Artistic
       License 2.0.  Refer to LICENSE for the full license text. In addition, please see DISCLAIMER.txt for
       disclaimers of warranty.

perl v5.30.0                                       2019-11-25              <u>Bio::Graphics:...::<a href="../man3pm/wiggle_xyplot.3pm.html">wiggle_xyplot</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>