<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>codonw - Correspondence Analysis of Codon Usage</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/codonw">codonw_1.4.4-8_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       codonw - Correspondence Analysis of Codon Usage

</pre><h4><b>SYNOPSIS</b></h4><pre>
       codonw [<u>inputfile</u>] [<u>outputfile</u>] [<u>bulkoutfile</u>] [<u>options</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       CodonW  is a package for codon usage analysis. It was designed to simplify Multivariate Analysis (MVA) of
       codon usage. The MVA method employed in CodonW is correspondence analysis (COA)  (the  most  popular  MVA
       method  for  codon  usage analysis). CodonW can generate a COA for codon usage, relative synonymous codon
       usage or amino acid usage. Additional analyses of codon usage include investigation  of  optimal  codons,
       codon and dinucleotide bias, and/or base composition.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>     Show help message

       <b>-nomenu</b>
              Prevent the menu interface being displayed

       <b>-nowarn</b>
              Prevent warnings about sequences being displayed

       <b>-silent</b>
              Overwrite files silently

       <b>-totals</b>
              Concatenate all genes in inputfile

       <b>-machine</b>
              Machine readable output

       <b>-human</b> Human readable output

       <b>-code</b> N
              Genetic code as defined under menu 3 option 5

       <b>-f_type</b> N
              Fop/CBI codons as defined by menu 3 option 6

       <b>-c_type</b> N
              Cai fitness values as defined by menu 3 option 7

       <b>-t</b> (char)
              Column separator to be used in output files (comma,tab,space)

       Codon usage indices and Amino acid indices

       <b>-cai</b>   calculate Codon Adaptation Index (CAI)

       <b>-fop</b>   calculate Frequency of OPtimal codons index (FOP)

       <b>-cbi</b>   calculate Codon Bias Index (CBI)

       <b>-enc</b>   Effective Number of Codons (ENc)

       <b>-gc</b>    G+C content of gene (all 3 codon positions)

       <b>-gcs3</b>  GC of synonymous codons 3rd positions

       <b>-sil_base</b>
              Base composition at synonymous third codon positions

       <b>-L_sym</b> Number of synonymous codons

       <b>-L_aa</b>  Total number of synonymous and non-synonymous codons

       <b>-all_indices</b>
              All the above indices

       <b>-aro</b>   Calculate aromaticity of protein

       <b>-hyd</b>   Calculate hydropathicity of protein

       <b>-cai_file</b>
              {file}      User input file of CAI values

       <b>-cbi_file</b>
              {file}      User input file of CBI values

       <b>-fop_file</b>
              {file}      User input file of Fop values

       Correspondence analysis (COA) options

       <b>-coa_cu</b>
              COA of codon usage frequencies

       <b>-coa_rscu</b>
              COA of Relative Synonymous Codon Usage

       <b>-coa_aa</b>
              COA of amino acid usage frequencies

       <b>-coa_expert</b>
              Generate detailed(expert) statistics on COA

       <b>-coa_axes</b> N
              Select number of axis to record

       <b>-coa_num</b> N
              Select  number  of  genes  to  use  to  identify  optimal  codons values can be whole numbers or a
              percentage (5 or 10%)

       Bulk output options | only one can be selected per analysis

       <b>-aau</b>   Amino Acid Usage (AAU)

       <b>-raau</b>  Relative Amino Acid Usage (RAAU)

       <b>-cu</b>    Codon Usage (CU) (default)

       <b>-cutab</b> Tabulation of codon usage

       <b>-cutot</b> Tabulation of dataset's codon usage

       <b>-rscu</b>  Relative Synonymous Codon Usage (RSCU)

       <b>-fasta</b> fasta format

       <b>-tidy</b>  fasta format

       <b>-reader</b>
              Reader format (codons are separated by spaces)

       <b>-transl</b>
              Conceptual translation of DNA to amino acid

       <b>-base</b>  Detailed report of codon G+C composition

       <b>-dinuc</b> Dinucleotide usage of the three codon pos.

       <b>-noblk</b> No bulk output to be written to file

       Some of the codonw subtools can be directly accessed using shortcuts prefixed with 'codonw-'.

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Sascha Steinbiss for the Debian distribution and can be used  for  any  other
       usage of the program.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/codonw-rscu.1.html">codonw-rscu</a></b>(1),  <b><a href="../man1/codonw-cu.1.html">codonw-cu</a></b>(1),  <b><a href="../man1/codonw-aau.1.html">codonw-aau</a></b>(1),  <b><a href="../man1/codonw-fop.1.html">codonw-fop</a></b>(1),  <b><a href="../man1/codonw-raau.1.html">codonw-raau</a></b>(1),  <b><a href="../man1/codonw-tidy.1.html">codonw-tidy</a></b>(1),  <b><a href="../man1/codonwreader.1.html">codonw-</a></b>
       <b><a href="../man1/codonwreader.1.html">reader</a></b>(1), <b><a href="../man1/codonw-cutab.1.html">codonw-cutab</a></b>(1), <b><a href="../man1/codonw-cutot.1.html">codonw-cutot</a></b>(1), <b><a href="../man1/codonw-transl.1.html">codonw-transl</a></b>(1), <b><a href="../man1/codonw-bases.1.html">codonw-bases</a></b>(1), <b><a href="../man1/codonw-base3s.1.html">codonw-base3s</a></b>(1), <b><a href="../man1/codonwdinuc.1.html">codonw-</a></b>
       <b><a href="../man1/codonwdinuc.1.html">dinuc</a></b>(1), <b><a href="../man1/codonw-cai.1.html">codonw-cai</a></b>(1), <b><a href="../man1/codonw-gc3s.1.html">codonw-gc3s</a></b>(1), <b><a href="../man1/codonw-gc.1.html">codonw-gc</a></b>(1), <b><a href="../man1/codonw-cbi.1.html">codonw-cbi</a></b>(1), <b><a href="../man1/codonw-enc.1.html">codonw-enc</a></b>(1)

codonw 1.4.4                                      February 2021                                        <u><a href="../man1/CODONW.1.html">CODONW</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>