<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>alistat - show statistics for a multiple alignment file</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/biosquid">biosquid_1.9g+cvs20050121-15.1build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       alistat - show statistics for a multiple alignment file

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>alistat</b> <u>[options]</u> <u>alignfile</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>alistat</b>  reads  a  multiple sequence alignment from the file <u>alignfile</u> in any supported format (including
       SELEX, GCG MSF, and CLUSTAL), and shows a number of simple statistics about it.  These statistics include
       the name of the format, the number of sequences, the total number of residues, the average and  range  of
       the sequence lengths, the alignment length (e.g. including gap characters).

       Also  shown  are  some  percent identities. A percent pairwise alignment identity is defined as <u>(idents</u> <u>/</u>
       <u>MIN(len1,</u> <u>len2))</u> where <u>idents</u> is the number of exact identities and <u>len1,</u> <u>len2</u> are the unaligned  lengths
       of  the  two sequences. The "average percent identity", "most related pair", and "most unrelated pair" of
       the alignment are the average, maximum, and minimum of all (N)(N-1)/2  pairs,  respectively.   The  "most
       distant  seq" is calculated by finding the maximum pairwise identity (best relative) for all N sequences,
       then finding the minimum of these N numbers (hence, the most outlying sequence).

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-a</b>     Show additional verbose information: a table with one line per sequence showing name, length,  and
              its  highest  and  lowest pairwise identity. These lines are prefixed with a * character to enable
              easily <b>grep'</b>ing them out and sorting them. For example, <u>alistat</u> <u>-a</u> <u>foo.slx</u> <u>|</u> <u>grep</u> <u>*</u> <u>|</u> <u>sort</u>  <u>-n</u>  <u>+3</u>
              gives  a  ranked  list  of  the most distant sequences in the alignment.  Incompatible with the <b>-f</b>
              option.

       <b>-f</b>     Fast; use a sampling method to estimate the average %id.  When  this  option  is  chosen,  <b>alistat</b>
              doesn't  show  the  other  three  pairwise identity numbers.  This option is useful for very large
              alignments, for which the full (N)(N-1) calculation of all pairs would be prohibitive (e.g. Pfam's
              GP120 alignment, with over 10,000 sequences). Incompatible with the <b>-a</b> option.

       <b>-h</b>     Print brief help; includes version number and summary of all options, including expert options.

       <b>-q</b>     be quiet - suppress the verbose header (program name, release number and date, the parameters  and
              options in effect).

       <b>-B</b>     (Babelfish). Autodetect and read a sequence file format other than the default (FASTA). Almost any
              common  sequence  file  format  is  recognized  (including Genbank, EMBL, SWISS-PROT, PIR, and GCG
              unaligned sequence formats, and Stockholm, GCG  MSF,  and  Clustal  alignment  formats).  See  the
              printed documentation for a complete list of supported formats.

</pre><h4><b>EXPERT</b> <b>OPTIONS</b></h4><pre>
       <b>--informat</b> <u>&lt;s&gt;</u>
              Specify  that  the  sequence  file is in format <u>&lt;s&gt;,</u> rather than the default FASTA format.  Common
              examples include Genbank, EMBL, GCG, PIR, Stockholm, Clustal, MSF,  or  PHYLIP;  see  the  printed
              documentation  for  a  complete  list of accepted format names.  This option overrides the default
              format (FASTA) and the <u>-B</u> Babelfish autodetection option.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/afetch.1.html">afetch</a></b>(1), <b><a href="../man1/compalign.1.html">compalign</a></b>(1),  <b><a href="../man1/compstruct.1.html">compstruct</a></b>(1),  <b><a href="../man1/revcomp.1.html">revcomp</a></b>(1),  <b><a href="../man1/seqsplit.1.html">seqsplit</a></b>(1),  <b><a href="../man1/seqstat.1.html">seqstat</a></b>(1),  <b><a href="../man1/sfetch.1.html">sfetch</a></b>(1),  <b><a href="../man1/shuffle.1.html">shuffle</a></b>(1),
       <b><a href="../man1/sindex.1.html">sindex</a></b>(1), <b><a href="../man1/sreformat.1.html">sreformat</a></b>(1), <b><a href="../man1/stranslate.1.html">stranslate</a></b>(1), <b><a href="../man1/weight.1.html">weight</a></b>(1).

</pre><h4><b>AUTHOR</b></h4><pre>
       Biosquid  and its documentation are Copyright (C) 1992-2003 HHMI/Washington University School of Medicine
       Freely distributed under the GNU General Public License (GPL) See COPYING in the source code distribution
       for more details, or contact me.

       Sean Eddy
       HHMI/Department of Genetics
       Washington University School of Medicine
       4444 Forest Park Blvd., Box 8510
       St Louis, MO 63108 USA
       Phone: 1-314-362-7666
       FAX  : 1-314-362-2157
       Email: <a href="mailto:eddy@genetics.wustl.edu">eddy@genetics.wustl.edu</a>

Biosquid 1.9g                                     January 2003                                        <u><a href="../man1/alistat.1.html">alistat</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>