<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::SeqIO::game::featHandler -- a class for handling feature elements</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::SeqIO::game::featHandler -- a class for handling feature elements

</pre><h4><b>SYNOPSIS</b></h4><pre>
       This module is not used directly

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::SeqIO::game::featHandler converts game XML &lt;annotation&gt; elements into flattened
       Bio::SeqFeature::Generic objects to be added to the sequence

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.

       Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Sheldon</b> <b>McKay</b></h4><pre>
       Email <a href="mailto:mckays@cshl.edu">mckays@cshl.edu</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : my $featHandler = Bio::SeqIO::game::featHandler-&gt;new($seq, $seq_h, $ann_l)
        Function: creates an object to deal with sequence features
        Returns : a handler object
        Args    : $seq   -- a Bio::SeqI compliant object
                  $seq_h -- ref. to a hash of other sequences associated
                            with the main sequence (proteins, etc)
                  $ann_l -- ref. to a list of annotations

   <b>add_source</b>
        Title   : add_source
        Usage   : $featHandler-&gt;add_source($seq-&gt;length, \%tags);
        Function: creates a source feature
        Returns : a Bio::SeqFeature::Generic object
        Args    : sequence length and a ref. to a hash of tag/value attributes

   <b>has_gene</b>
        Title   : has_gene
        Usage   : my $gene = $self-&gt;_has_gene($gene, $gname, $id)
        Function: method to get/set the current gene feature
        Returns : a Bio::SeqFeature::Generic object (if there is a gene)
        Args    : (optional)
                  $gene  -- an XML element for the annotation
                  $gname -- gene name
                  $id    -- gene ID (not always the same as the name)

   <b>_has_CDS</b>
        Title   : _has_CDS
        Usage   : my $cds = $self-&gt;_has_CDS
        Function: internal getter/setter for CDS features
        Returns : a Bio::SeqFeature::Generic transcript object (or nothing)
        Args    : a Bio::SeqFeature::Generic transcript feature

   <b>add_annotation</b>
        Title   : add_annotation
        Usage   : $featHandler-&gt;add_annotation($seq, $type, $id, $tags, $feats)
        Function: converts a containment hierarchy into an ordered list of flat features
        Returns : nothing
        Args    : $seq   -- a Bio::SeqI compliant object
                  $type  -- the annotation type
                  $id    -- the anotation ID
                  $tags  -- ref. to a hash of tag/value attributes
                  $feats -- ref to an array of Bio::SeqFeature::Generic objects

   <b>_add_generic_annotation</b>
        Title   : _add_generic_annotation
        Usage   : $self-&gt;_add_generic_annotation($seq, $type, $id, $tags, $feats)
        Function: an internal method to handle non-gene annotations
        Returns : nothing
        Args    : $seq   -- a Bio::SeqI compliant object
                  $type  -- the annotation type
                  $id    -- the anotation ID
                  $tags  -- ref. to a hash of tag/value attributes
                  $feats -- ref to an array of Bio::SeqFeature::Generic objects

   <b>feature_set</b>
        Title   : feature_set
        Usage   : push @feats, $featHandler-&gt;feature_set($id, $gname, $set, $anntype);
        Function: handles &lt;feature_span&gt; hierarchies (usually a transcript)
        Returns : a list of Bio::SeqFeature::Generic objects
        Args    : $id      -- ID of the feature set
                  $gname   -- name of the gene
                  $set     -- the &lt;feature_set&gt; object
                  $anntype -- type of the parent annotation

   <b>_build_feature_set</b>
        Title   : _build_feature_set
        Usage   : $self-&gt;_build_feature_set($set, 1) # 1 flag means retain the exon as a subfeat
        Function: an internal method to process attributes and subfeats of a feature set
        Returns : nothing
        Args    : $set -- a &lt;feature_set&gt; element
                  1    -- optional flag to retain exons as subfeats.  Otherwise, they will
                          be converted to sublocations of a parent CDS feature

   <b>_add_feature_span</b>
        Title   : _add_feature_span
        Usage   : $self-&gt;_add_feature_span($el, 1)
        Function: an internal method to process &lt;feature_span&gt; elements
        Returns : nothing
        Args    : $el -- a &lt;feature_span&gt; element
                  1   -- an optional flag to retain exons as subfeatures

   <b>_add_CDS</b>
        Title   : _add_CDS
        Usage   : my $cds = $self-&gt;_add_CDS($transcript, $tags)
        Function: an internal method to create a CDS feature from a transcript feature
        Returns : a Bio::SeqFeature::Generic object
        Args    : $transcript -- a Bio::SeqFeature::Generic object for a transcript
                  $tags       -- ref. to a hash of tag/value attributes

perl v5.32.1                                       2021-08-15                 <u>Bio::SeqIO::game::<a href="../man3pm/featHandler.3pm.html">featHandler</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>