<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::DB::GenBank - Database object interface to GenBank</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-db-ncbihelper-perl">libbio-db-ncbihelper-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::DB::GenBank - Database object interface to GenBank

</pre><h4><b>SYNOPSIS</b></h4><pre>
           use Bio::DB::GenBank;
           $gb = Bio::DB::GenBank-&gt;new();

           $seq = $gb-&gt;get_Seq_by_id('J00522'); # Unique ID, *not always the LOCUS ID*

           # or ...

           $seq = $gb-&gt;get_Seq_by_acc('J00522'); # Accession Number
           $seq = $gb-&gt;get_Seq_by_version('J00522.1'); # Accession.version
           $seq = $gb-&gt;get_Seq_by_gi('405830'); # GI Number

           # get a stream via a query string
           my $query = Bio::DB::Query::GenBank-&gt;new
               (-query   =&gt;'Oryza sativa[Organism] AND EST',
                -reldate =&gt; '30',
                -db      =&gt; 'nucleotide');
           my $seqio = $gb-&gt;get_Stream_by_query($query);

           while( my $seq =  $seqio-&gt;next_seq ) {
             print "seq length is ", $seq-&gt;length,"\n";
           }

           # or ... best when downloading very large files, prevents
           # keeping all of the file in memory

           # also don't want features, just sequence so let's save bandwidth
           # and request Fasta sequence
           $gb = Bio::DB::GenBank-&gt;new(-retrievaltype =&gt; 'tempfile' ,
                                                     -format =&gt; 'Fasta');
           my $seqio = $gb-&gt;get_Stream_by_acc(['AC013798', 'AC021953'] );
           while( my $clone =  $seqio-&gt;next_seq ) {
             print "cloneid is ", $clone-&gt;display_id, " ",
                    $clone-&gt;accession_number, "\n";
           }
           # note that get_Stream_by_version is not implemented

           # don't want the entire sequence or more options
           my $gb = Bio::DB::GenBank-&gt;new(-format     =&gt; 'Fasta',
                                          -seq_start  =&gt; 100,
                                          -seq_stop   =&gt; 200,
                                          -strand     =&gt; 1,
                                          -complexity =&gt; 4);
           my $seqi = $gb-&gt;get_Stream_by_query($query);

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Allows the dynamic retrieval of Bio::Seq sequence objects from the GenBank database at NCBI, via an
       Entrez query.

       WARNING: Please do <b>NOT</b> spam the Entrez web server with multiple requests.  NCBI offers Batch Entrez for
       this purpose.

       Note that when querying for GenBank accessions starting with 'NT_' you will need to call
       $gb-&gt;request_format('fasta') beforehand, because in GenBank format (the default) the sequence part will
       be left out (the reason is that NT contigs are rather annotation with references to clones).

       Some work has been done to automatically detect and retrieve whole NT_ clones when the data is in that
       format (NCBI RefSeq clones). The former behavior prior to bioperl 1.6 was to retrieve these from EBI, but
       now these are retrieved directly from NCBI. The older behavior can be regained by setting the
       'redirect_refseq' flag to a value evaluating to TRUE.

   <b>Running</b>
       Alternate methods are described at &lt;https://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EFetch&gt;

       NOTE: strand should be 1 for plus or 2 for minus.

       Complexity: gi is often a part of a biological blob, containing other gis

       complexity regulates the display:
         0 - get the whole blob
         1 - get the bioseq for gi of interest (default in Entrez)
         2 - get the minimal bioseq-set containing the gi of interest
         3 - get the minimal nuc-prot containing the gi of interest
         4 - get the minimal pub-set containing the gi of interest

       'seq_start' and 'seq_stop' will not work when setting complexity to any value other than 1.  'strand'
       works for any setting other than a complexity of 0 (whole glob); when you try this with a GenBank return
       format nothing happens, whereas using FASTA works but causes display problems with the other sequences in
       the glob.  As Tao Tao says from NCBI, "Better left it out or set it to 1."

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists. Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Aaron</b> <b>Mackey,</b> <b>Jason</b> <b>Stajich</b></h4><pre>
       Email <a href="mailto:amackey@virginia.edu">amackey@virginia.edu</a> Email <a href="mailto:jason@bioperl.org">jason@bioperl.org</a>

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : $gb = Bio::DB::GenBank-&gt;new(@options)
        Function: Creates a new genbank handle
        Returns : a new Bio::DB::Genbank object
        Args    : -delay   number of seconds to delay between fetches (3s)

       NOTE:  There are other options that are used internally.  By NCBI policy, this module introduces a 3s
       delay between fetches.  If you are fetching multiple genbank ids, it is a good idea to use get

   <b>get_params</b>
        Title   : get_params
        Usage   : my %params = $self-&gt;get_params($mode)
        Function: Returns key,value pairs to be passed to NCBI database
                  for either 'batch' or 'single' sequence retrieval method
        Returns : a key,value pair hash
        Args    : 'single' or 'batch' mode for retrieval

</pre><h4><b>Routines</b> <b>Bio::DB::WebDBSeqI</b> <b>from</b> <b>Bio::DB::RandomAccessI</b></h4><pre>
   <b>get_Seq_by_id</b>
        Title   : get_Seq_by_id
        Usage   : $seq = $db-&gt;get_Seq_by_id('ROA1_HUMAN')
        Function: Gets a Bio::Seq object by its name
        Returns : a Bio::Seq object
        Args    : the id (as a string) of a sequence
        Throws  : "id does not exist" exception

   <b>get_Seq_by_acc</b>
         Title   : get_Seq_by_acc
         Usage   : $seq = $db-&gt;get_Seq_by_acc($acc);
         Function: Gets a Seq object by accession numbers
         Returns : a Bio::Seq object
         Args    : the accession number as a string
         Note    : For GenBank, this just calls the same code for get_Seq_by_id().
                   Caveat: this normally works, but in rare cases simply passing the
                   accession can lead to odd results, possibly due to unsynchronized
                   NCBI ID servers. Using get_Seq_by_version() is slightly better, but
                   using the unique identifier (GI) and get_Seq_by_id is the most
                   consistent
         Throws  : "id does not exist" exception

   <b>get_Seq_by_gi</b>
        Title   : get_Seq_by_gi
        Usage   : $seq = $db-&gt;get_Seq_by_gi('405830');
        Function: Gets a Bio::Seq object by gi number
        Returns : A Bio::Seq object
        Args    : gi number (as a string)
        Throws  : "gi does not exist" exception

   <b>get_Seq_by_version</b>
        Title   : get_Seq_by_version
        Usage   : $seq = $db-&gt;get_Seq_by_version('X77802.1');
        Function: Gets a Bio::Seq object by sequence version
        Returns : A Bio::Seq object
        Args    : accession.version (as a string)
        Note    : Caveat: this normally works, but using the unique identifier (GI) and
                  get_Seq_by_id is the most consistent
        Throws  : "acc.version does not exist" exception

</pre><h4><b>Routines</b> <b>implemented</b> <b>by</b> <b>Bio::DB::NCBIHelper</b></h4><pre>
   <b>get_Stream_by_query</b>
         Title   : get_Stream_by_query
         Usage   : $seq = $db-&gt;get_Stream_by_query($query);
         Function: Retrieves Seq objects from Entrez 'en masse', rather than one
                   at a time.  For large numbers of sequences, this is far superior
                   than get_Stream_by_[id/acc]().
         Example :
         Returns : a Bio::SeqIO stream object
         Args    : $query :   An Entrez query string or a
                   Bio::DB::Query::GenBank object.  It is suggested that you
                   create a Bio::DB::Query::GenBank object and get the entry
                   count before you fetch a potentially large stream.

   <b>get_Stream_by_id</b>
         Title   : get_Stream_by_id
         Usage   : $stream = $db-&gt;get_Stream_by_id( [$uid1, $uid2] );
         Function: Gets a series of Seq objects by unique identifiers
         Returns : a Bio::SeqIO stream object
         Args    : $ref : a reference to an array of unique identifiers for
                          the desired sequence entries

   <b>get_Stream_by_acc</b>
         Title   : get_Stream_by_acc
         Usage   : $seq = $db-&gt;get_Stream_by_acc([$acc1, $acc2]);
         Function: Gets a series of Seq objects by accession numbers
         Returns : a Bio::SeqIO stream object
         Args    : $ref : a reference to an array of accession numbers for
                          the desired sequence entries
         Note    : For GenBank, this just calls the same code for get_Stream_by_id()

   <b>get_Stream_by_gi</b>
         Title   : get_Stream_by_gi
         Usage   : $seq = $db-&gt;get_Seq_by_gi([$gi1, $gi2]);
         Function: Gets a series of Seq objects by gi numbers
         Returns : a Bio::SeqIO stream object
         Args    : $ref : a reference to an array of gi numbers for
                          the desired sequence entries
         Note    : For GenBank, this just calls the same code for get_Stream_by_id()

   <b>get_Stream_by_batch</b>
         Title   : get_Stream_by_batch
         Usage   : $seq = $db-&gt;get_Stream_by_batch($ref);
         Function: Retrieves Seq objects from Entrez 'en masse', rather than one
                   at a time.
         Example :
         Returns : a Bio::SeqIO stream object
         Args    : $ref : either an array reference, a filename, or a filehandle
                   from which to get the list of unique ids/accession numbers.

       NOTE: This method is redundant and deprecated.  Use <b>get_Stream_by_id()</b> instead.

   <b>get_request</b>
        Title   : get_request
        Usage   : my $url = $self-&gt;get_request
        Function: HTTP::Request
        Returns :
        Args    : %qualifiers = a hash of qualifiers (ids, format, etc)

   <b>default_format</b>
        Title   : default_format
        Usage   : my $format = $self-&gt;default_format
        Function: Returns default sequence format for this module
        Returns : string
        Args    : none

perl v5.36.0                                       2023-12-17                              <u>Bio::DB::<a href="../man3pm/GenBank.3pm.html">GenBank</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>