<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>flash - Fast Length Adjustment of SHort reads</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/flash">flash_1.2.11-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       flash - Fast Length Adjustment of SHort reads

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>flash</b> [<u>OPTIONS</u>] <u>MATES_1.FASTQ</u> <u>MATES_2.FASTQ</u>

       <b>flash</b> [<u>OPTIONS</u>] <b>--interleaved-input</b> (MATES.FASTQ | -)

       <b>flash</b> [OPTIONS] <b>--tab-delimited-input</b> (MATES.TAB | -)

</pre><h4><b>DESCRIPTION</b></h4><pre>
       FLASH (Fast Length Adjustment of SHort reads) is an accurate and fast tool to merge paired-end reads that
       were  generated from DNA fragments whose lengths are shorter than twice the length of reads.  Merged read
       pairs result in unpaired longer reads, which are generally more desired in  genome  assembly  and  genome
       analysis processes.

       Briefly,  the  FLASH  algorithm  considers all possible overlaps at or above a minimum length between the
       reads in a pair and chooses the overlap that results  in  the  lowest  mismatch  density  (proportion  of
       mismatched  bases  in  the  overlapped region).  Ties between multiple overlaps are broken by considering
       quality scores at mismatch sites.  When building the merged sequence, FLASH computes a consensus sequence
       in   the   overlapped   region.    More   details   can   be   found   in   the   original    publication
       (<a href="http://bioinformatics.oxfordjournals.org/content/27/21/2957.full">http://bioinformatics.oxfordjournals.org/content/27/21/2957.full</a>).

   <b>Limitations</b> <b>of</b> <b>FLASH</b> <b>include:</b>
              - FLASH cannot merge paired-end reads that do not overlap.

              -  FLASH  is  not  designed for data that has a significant amount of indel errors (such as Sanger
              sequencing data).  It is best suited for Illumina data.

</pre><h4><b>MANDATORY</b> <b>INPUT</b></h4><pre>
       The most common input to FLASH is two FASTQ files containing read  1  and  read  2  of  each  mate  pair,
       respectively, in the same order.

       Alternatively,  you  may provide one FASTQ file, which may be standard input, containing paired-end reads
       in  either  interleaved  FASTQ  (see  the  <b>--interleaved-input</b>  option)   or   tab-delimited   (see   the
       <b>--tab-delimited-input</b> option) format.  In all cases, gzip compressed input is autodetected.  Also, in all
       cases, the PHRED offset is, by default, assumed to be 33; use the <b>--phred-offset</b> option to change it.

</pre><h4><b>OUTPUT</b></h4><pre>
       The default output of FLASH consists of the following files:

       - out.extendedFrags.fastq
              The merged reads.

       - out.notCombined_1.fastq
              Read 1 of mate pairs that were not merged.

       - out.notCombined_2.fastq
              Read 2 of mate pairs that were not merged.

       - out.hist
              Numeric histogram of merged read lengths.

       - out.histogram
              Visual histogram of merged read lengths.

       FLASH also logs informational messages to standard output.  These can also be redirected to a file, as in
       the following example:

              $ flash reads_1.fq reads_2.fq 2&gt;&amp;1 | tee flash.log

       In addition, FLASH supports several features affecting the output:

              - Writing the merged reads directly to standard output (<b>--to-stdout</b>)

              - Writing gzip compressed output files (<b>-z</b>) or using an external
                 compression program (<b>--compress-prog</b>)

              - Writing the uncombined read pairs in interleaved FASTQ format

              (<b>--interleaved-output</b>)

              - Writing all output reads to a single file in tab-delimited format

              (<b>--tab-delimited-output</b>)

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-m</b>, <b>--min-overlap</b>=<u>NUM</u>
              The  minimum  required  overlap length between two reads to provide a confident overlap.  Default:
              10bp.

       <b>-M</b>, <b>--max-overlap</b>=<u>NUM</u>
              Maximum overlap length expected in approximately 90% of read pairs.  It is by default set to 65bp,
              which works well for 100bp reads generated from a 180bp library, assuming a normal distribution of
              fragment lengths.  Overlaps longer than the maximum overlap parameter are still considered as good
              overlaps, but the mismatch density (explained below) is  calculated  over  the  first  max_overlap
              bases  in the overlapped region rather than the entire overlap.  Default: 65bp, or calculated from
              the specified read length, fragment length, and fragment length standard deviation.

       <b>-x</b>, <b>--max-mismatch-density</b>=<u>NUM</u>
              Maximum allowed ratio between the number of mismatched base pairs and  the  overlap  length.   Two
              reads  will  not  be  combined  with  a given overlap if that overlap results in a mismatched base
              density higher than this value.  Note: Any occurence of an 'N' in either read is ignored  and  not
              counted  towards  the  mismatches or overlap length.  Our experimental results suggest that higher
              values of the maximum mismatch density yield larger numbers of correctly merged read pairs but  at
              the expense of higher numbers of incorrectly merged read pairs.  Default: 0.25.

       <b>-O</b>, <b>--allow-outies</b>
              Also try combining read pairs in the "outie" orientation, e.g.

       Read 1: &lt;-----------
              Read 2:       <b>------------</b>&gt;

              as opposed to only the "innie" orientation, e.g.

       Read 1:
              &lt;------------

              Read 2: <b>-----------</b>&gt;

       FLASH uses the same parameters when trying each
              orientation.   If  a  read  pair  can  be  combined  in both "innie" and "outie" orientations, the
              better-fitting one will be chosen using the same scoring algorithm that FLASH normally uses.

       This option also causes extra .innie and .outie
              histogram files to be produced.

       <b>-p</b>, <b>--phred-offset</b>=<u>OFFSET</u>
              The smallest ASCII value of the characters used to represent quality  values  of  bases  in  FASTQ
              files.   It  should  be  set  to  either 33, which corresponds to the later Illumina platforms and
              Sanger platforms, or 64, which corresponds to the earlier Illumina platforms.  Default: 33.

       <b>-r</b>, <b>--read-len</b>=<u>LEN</u>

       <b>-f</b>, <b>--fragment-len</b>=<u>LEN</u>

       <b>-s</b>, <b>--fragment-len-stddev</b>=<u>LEN</u>
              Average read length, fragment length, and fragment  standard  deviation.   These  are  convenience
              parameters  only,  as  they  are  only  used  for  calculating the maximum overlap (<b>--max-overlap</b>)
              parameter.  The maximum overlap is calculated as the  overlap  of  average-length  reads  from  an
              average-size  fragment  plus 2.5 times the fragment length standard deviation.  The default values
              are <b>-r</b> 100, <b>-f</b> 180, and <b>-s</b> 18, so this works out to a maximum overlap of 65 bp.  If  <b>--max-overlap</b>
              is specified, then the specified value overrides the calculated value.

       If you do not know the standard deviation of the
              fragment  library,  you  can  probably  assume  that  the standard deviation is 10% of the average
              fragment length.

       <b>--cap-mismatch-quals</b>
              Cap quality scores assigned at mismatch locations to 2.  This was the default  behavior  in  FLASH
              v1.2.7  and earlier.  Later versions will instead calculate such scores as max(|q1 - q2|, 2); that
              is, the absolute value of the difference in quality scores, but at least 2.  Essentially, the  new
              behavior  prevents  a  low  quality base call that is likely a sequencing error from significantly
              bringing down the quality of a high quality, likely correct base call.

       <b>--interleaved-input</b>
              Instead of requiring files MATES_1.FASTQ and MATES_2.FASTQ, allow a single file  MATES.FASTQ  that
              has the paired-end reads interleaved.  Specify "-" to read from standard input.

       <b>--interleaved-output</b>
              Write the uncombined pairs in interleaved FASTQ format.

       <b>-I</b>, <b>--interleaved</b>
              Equivalent to specifying both <b>--interleaved-input</b> and <b>--interleaved-output</b>.

       <b>-Ti</b>, <b>--tab-delimited-input</b>
              Assume  the  input  is in tab-delimited format rather than FASTQ, in the format described below in
              '--tab-delimited-output'.  In this mode you should provide a single input file, each line of which
              must contain either a read pair (5 fields) or a single read (3 fields).  FLASH will try to combine
              the read  pairs.   Single  reads  will  be  written  to  the  output  file  as-is  if  also  using
              <b>--tab-delimited-output</b>;  otherwise  they  will  be  ignored.  Note that you may specify "-" as the
              input file to read the tab-delimited data from standard input.

       <b>-To</b>, <b>--tab-delimited-output</b>
              Write output in tab-delimited format (not FASTQ).  Each line will contain either a  combined  pair
              in  the  format  'tag  &lt;tab&gt;  seq &lt;tab&gt; qual' or an uncombined pair in the format 'tag &lt;tab&gt; seq_1
              &lt;tab&gt; qual_1 &lt;tab&gt; seq_2 &lt;tab&gt; qual_2'.

       <b>-o</b>, <b>--output-prefix</b>=<u>PREFIX</u>
              Prefix of output files.  Default: "out".

       <b>-d</b>, <b>--output-directory</b>=<u>DIR</u>
              Path to directory for output files.  Default: current working directory.

       <b>-c</b>, <b>--to-stdout</b>
              Write the combined reads to standard output.  In this mode, with FASTQ output  (the  default)  the
              uncombined  reads  are discarded.  With tab-delimited output, uncombined reads are included in the
              tab-delimited data written to standard output.  In both cases, histogram files  are  not  written,
              and informational messages are sent to standard error rather than to standard output.

       <b>-z</b>, <b>--compress</b>
              Compress  the  output  files  directly  with  zlib,  using  the gzip container format.  Similar to
              specifying <b>--compress-prog</b>=<u>gzip</u> and <b>--suffix</b>=<u>gz</u>, but may be slightly faster.

       <b>--compress-prog</b>=<u>PROG</u>
              Pipe the output through the compression program PROG, which will be called as `PROG  <b>-c</b>  -',  plus
              any  arguments  specified by <b>--compress-prog-args</b>.  PROG must read uncompressed data from standard
              input and write compressed data to standard output when invoked as noted above.   Examples:  gzip,
              bzip2, xz, pigz.

       <b>--compress-prog-args</b>=<u>ARGS</u>
              A  string  of  additional  arguments  that  will  be  passed  to the compression program if one is
              specified with <b>--compress-prog</b>=<u>PROG</u>.  (The arguments '-c  -'  are  still  passed  in  addition  to
              explicitly specified arguments.)

       <b>--suffix</b>=<u>SUFFIX</u>, <b>--output-suffix</b>=<u>SUFFIX</u>
              Use  SUFFIX as the suffix of the output files after ".fastq".  A dot before the suffix is assumed,
              unless an empty suffix is provided.  Default: nothing; or 'gz' if <b>-z</b>  is  specified;  or  PROG  if
              <b>--compress-prog</b>=<u>PROG</u> is specified.

       <b>-t</b>, <b>--threads</b>=<u>NTHREADS</u>
              Set  the  number  of  worker threads.  This is in addition to the I/O threads.  Default: number of
              processors.  Note: if you need FLASH's output to appear deterministically or in the same order  as
              the original reads, you must specify <b>-t</b> 1 (<b>--threads</b>=<u>1</u>).

       <b>-q</b>, <b>--quiet</b>
              Do not print informational messages.

       <b>-h</b>, <b>--help</b>
              Display this help and exit.

       <b>-v</b>, <b>--version</b>
              Display version.

</pre><h4><b>AUTHOR</b></h4><pre>
        This manpage was written by Andreas Tille for the Debian distribution and
        can be used for any other usage of the program.

flash 1.2.11                                        May 2020                                            <u><a href="../man1/FLASH.1.html">FLASH</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>