<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>cnvkit_autobin - Quickly calculate reasonable bin sizes from BAM read counts.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/cnvkit">cnvkit_0.9.12-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       cnvkit_autobin - Quickly calculate reasonable bin sizes from BAM read counts.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: cnvkit.py autobin [-h] [-f FILENAME] [-m {hybrid,amplicon,wgs}]

       [-g FILENAME] [-t TARGETS] [-b BP_PER_BIN]
              [--target-max-size     BASES]     [--target-min-size    BASES]    [--antitarget-max-size    BASES]
              [--antitarget-min-size BASES] [--annotate FILENAME] [--short-names] [--target-output-bed FILENAME]
              [--antitarget-output-bed FILENAME] bams [bams ...]

   <b>positional</b> <b>arguments:</b>
       bams   Sample BAM file(s) to test for target coverage

   <b>options:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>-f</b> FILENAME, <b>--fasta</b> FILENAME
              Reference genome, FASTA format (e.g. UCSC hg19.fa)

       <b>-m</b> {hybrid,amplicon,wgs}, <b>--method</b> {hybrid,amplicon,wgs}
              Sequencing protocol: hybridization capture ('hybrid'), targeted amplicon sequencing  ('amplicon'),
              or  whole  genome sequencing ('wgs'). Determines whether and how to use antitarget bins. [Default:
              hybrid]

       <b>-g</b> FILENAME, <b>--access</b> FILENAME
              Sequencing-accessible genomic regions, or exons  to  use  as  possible  targets  (e.g.  output  of
              refFlat2bed.py)

       <b>-t</b> TARGETS, <b>--targets</b> TARGETS
              Potentially  targeted  genomic  regions, e.g. all possible exons for the reference genome. Format:
              BED, interval list, etc.

       <b>-b</b> BP_PER_BIN, <b>--bp-per-bin</b> BP_PER_BIN
              Desired average number of sequencing read bases mapped to each bin. [Default: 100000.0]

       <b>--target-max-size</b> BASES
              Maximum size of target bins. [Default: 20000]

       <b>--target-min-size</b> BASES
              Minimum size of target bins. [Default: 20]

       <b>--antitarget-max-size</b> BASES
              Maximum size of antitarget bins. [Default: 500000]

       <b>--antitarget-min-size</b> BASES
              Minimum size of antitarget bins. [Default: 500]

       <b>--annotate</b> FILENAME
              Use gene models from this file to assign names to the target regions. Format: UCSC refFlat.txt  or
              ensFlat.txt file (preferred), or BED, interval list, GFF, or similar.

       <b>--short-names</b>
              Reduce multi-accession bait labels to be short and consistent.

       <b>--target-output-bed</b> FILENAME
              Filename for target BED output. If not specified, constructed from the input file basename.

       <b>--antitarget-output-bed</b> FILENAME
              Filename for antitarget BED output. If not specified, constructed from the input file basename.

cnvkit.py autobin 0.9.10                            July 2023                                  <u><a href="../man1/CNVKIT_AUTOBIN.1.html">CNVKIT_AUTOBIN</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>