<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>bp_oligo_count - oligo count and frequency</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/bioperl">bioperl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       bp_oligo_count - oligo count and frequency

</pre><h4><b>SYNOPSIS</b></h4><pre>
         Usage:  bp_oligo_count [-h/--help] [-l/--length OLIGOLENGTH]
                 [-f/--format SEQFORMAT] [-i/--in/-s/--sequence SEQFILE]
                 [-o/--out OUTFILE]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This scripts counts occurrence and frequency for all oligonucleotides of given length.

       It can be used to determine what primers are useful for frequent priming of nucleic acid for random
       labeling.

       Note that this script could be run by utilizing the compseq program which is part of EMBOSS.

</pre><h4><b>OPTIONS</b></h4><pre>
       The default sequence format is fasta. If no outfile is given, the results will be printed to standard
       out. All other options can entered interactively.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Charles</b> <b>C.</b> <b>Kim</b></h4><pre>
       Email <a href="mailto:cckim@stanford.edu">cckim@stanford.edu</a>

</pre><h4><b>HISTORY</b></h4><pre>
       Written July 2, 2001

       Submitted to bioperl scripts project 2001/08/06

       &gt;&gt; 100 x speed optimization by Heikki Lehvaslaiho

perl v5.32.1                                       2021-08-15                                 <u><a href="../man1p/BP_OLIGO_COUNT.1p.html">BP_OLIGO_COUNT</a></u>(1p)
</pre>
 </div>
</div></section>
</div>
</body>
</html>