<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Ray - assemble genomes in parallel using the message-passing interface</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/ray">ray_2.3.1-9_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Ray - assemble genomes in parallel using the message-passing interface

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>mpiexec</b> <b>-n</b> NUMBER_OF_RANKS Ray <b>-k</b> KMERLENGTH <b>-p</b> l1_1.fastq l1_2.fastq <b>-p</b> l2_1.fastq l2_2.fastq <b>-o</b> test

       <b>mpiexec</b> <b>-n</b> NUMBER_OF_RANKS Ray Ray.conf # with commands in a file

</pre><h4><b>DESCRIPTION:</b></h4><pre>
       The  Ray  genome  assembler  is  built  on top of the RayPlatform, a generic plugin-based distributed and
       parallel compute engine that uses the message-passing interface for passing messages.

       Ray targets several applications:

              - de novo genome assembly (with Ray vanilla) - de novo meta-genome assembly (with Ray Meta)  -  de
              novo  transcriptome assembly (works, but not tested a lot) - quantification of contig abundances -
              quantification of  microbiome  consortia  members  (with  Ray  Communities)  -  quantification  of
              transcript  expression  -  taxonomy  profiling  of  samples (with Ray Communities) - gene ontology
              profiling of samples (with Ray Ontologies)

       <b>-help</b>

              Displays this help page.

       <b>-version</b>

              Displays Ray version and compilation options.

              Using a configuration file

              Ray can be launched with mpiexec <b>-n</b> 16 Ray Ray.conf The configuration file  can  include  comments
              (starting with #).

              K-mer length

       <b>-k</b> kmerLength

              Selects  the length of k-mers. The default value is 21.  It must be odd because reverse-complement
              vertices are stored together.  The maximum length  is  defined  at  compilation  by  MAXKMERLENGTH
              Larger k-mers utilise more memory.

              Inputs

       <b>-p</b> leftSequenceFile rightSequenceFile [averageOuterDistance standardDeviation]

              Provides  two  files  containing paired-end reads.  averageOuterDistance and standardDeviation are
              automatically computed if not provided.

       <b>-i</b> interleavedSequenceFile [averageOuterDistance standardDeviation]

              Provides  one  file   containing   interleaved   paired-end   reads.    averageOuterDistance   and
              standardDeviation are automatically computed if not provided.

       <b>-s</b> sequenceFile

              Provides a file containing single-end reads.

              Outputs

       <b>-o</b> outputDirectory

              Specifies the directory for outputted files. Default is RayOutput

              Assembly options (defaults work well)

       <b>-disable-recycling</b>

              Disables read recycling during the assembly reads will be set free in 3 cases: 1. the distance did
              not  match for a pair 2. the read has not met its mate 3. the library population indicates a wrong
              placement see Constrained traversal of repeats with paired sequences.  Sebastien Boisvert,  Elenie
              Godzaridis,  Francois  Laviolette  &amp;  Jacques  Corbeil.  First Annual RECOMB Satellite Workshop on
              Massively Parallel Sequencing, March 26-27 2011, Vancouver, BC, Canada.

       <b>-disable-scaffolder</b>

              Disables the scaffolder.

       <b>-minimum-contig-length</b> minimumContigLength

              Changes the minimum contig length, default is 100 nucleotides

       <b>-color-space</b>

              Runs in color-space Needs csfasta files. Activated automatically if csfasta files are provided.

       <b>-use-maximum-seed-coverage</b> maximumSeedCoverageDepth

              Ignores any seed with a coverage depth above this threshold.  The default is 4294967295.

       <b>-use-minimum-seed-coverage</b> minimumSeedCoverageDepth

              Sets the minimum seed coverage depth.  Any path with a coverage depth  lower  than  this  will  be
              discarded. The default is 0.

              Distributed storage engine (all these values are for each MPI rank)

       <b>-bloom-filter-bits</b> bits

              Sets  the number of bits for the Bloom filter Default is 268435456 bits, 0 bits disables the Bloom
              filter.

       <b>-hash-table-buckets</b> buckets

              Sets the initial number of buckets. Must be a power of 2 !  Default value: 268435456

       <b>-hash-table-buckets-per-group</b> buckets

              Sets the number of buckets per group for sparse storage Default value: 64, Must be between &gt;=1 and
              &lt;= 64

       <b>-hash-table-load-factor-threshold</b> threshold

              Sets the load factor threshold for real-time resizing Default value: 0.75, must be &gt;= 0.5 and &lt; 1

       <b>-hash-table-verbosity</b>

              Activates verbosity for the distributed storage engine

              Biological abundances

       <b>-search</b> searchDirectory

              Provides a directory containing fasta files to be searched in the  de  Bruijn  graph.   Biological
              abundances       will       be       written       to      RayOutput/BiologicalAbundances      See
              Documentation/BiologicalAbundances.txt

       <b>-one-color-per-file</b>

              Sets one color per file instead of one per sequence.  By default, each sequence in each file has a
              different color.  For files with large numbers of sequences, using one single color per  file  may
              be more efficient.

              Taxonomic profiling with colored de Bruijn graphs

       <b>-with-taxonomy</b> Genome-to-Taxon.tsv TreeOfLife-Edges.tsv Taxon-Names.tsv

              Provides    a    taxonomy.     Computes    and    writes   detailed   taxonomic   profiles.    See
              Documentation/Taxonomy.txt for details.

       <b>-gene-ontology</b> OntologyTerms.txt
              Annotations.txt

              Provides an ontology and annotations.  OntologyTerms.txt is fetched  from  <a href="http://geneontology.org">http://geneontology.org</a>
              Annotations.txt  is  a  2-column file (EMBL_CDS handle       &amp;       gene ontology identifier) See
              Documentation/GeneOntology.txt

              Other outputs

       <b>-enable-neighbourhoods</b>

              Computes    contig    neighborhoods    in     the     de     Bruijn     graph     Output     file:
              RayOutput/NeighbourhoodRelations.txt

       <b>-amos</b>

              Writes  the  AMOS  file called RayOutput/AMOS.afg An AMOS file contains read positions on contigs.
              Can be opened with software with graphical user interface.

       <b>-write-kmers</b>

              Writes k-mer graph to RayOutput/kmers.txt  The  resulting  file  is  not  utilised  by  Ray.   The
              resulting file is very large.

       <b>-write-read-markers</b>

              Writes read markers to disk.

       <b>-write-seeds</b>

              Writes seed DNA sequences to RayOutput/Rank&lt;rank&gt;.RaySeeds.fasta

       <b>-write-extensions</b>

              Writes extension DNA sequences to RayOutput/Rank&lt;rank&gt;.RayExtensions.fasta

       <b>-write-contig-paths</b>

              Writes contig paths with coverage values to RayOutput/Rank&lt;rank&gt;.RayContigPaths.txt

       <b>-write-marker-summary</b>

              Writes marker statistics.

              Memory usage

       <b>-show-memory-usage</b>

              Shows memory usage. Data is fetched from <a href="file:/proc">/proc</a> on GNU/Linux Needs __linux__

       <b>-show-memory-allocations</b>

              Shows memory allocation events

              Algorithm verbosity

       <b>-show-extension-choice</b>

              Shows the choice made (with other choices) during the extension.

       <b>-show-ending-context</b>

              Shows  the ending context of each extension.  Shows the children of the vertex where extension was
              too difficult.

       <b>-show-distance-summary</b>

              Shows summary of outer distances used for an extension path.

       <b>-show-consensus</b>

              Shows the consensus when a choice is done.

              Checkpointing

       <b>-write-checkpoints</b> checkpointDirectory

              Write checkpoint files

       <b>-read-checkpoints</b> checkpointDirectory

              Read checkpoint files

       <b>-read-write-checkpoints</b> checkpointDirectory

              Read and write checkpoint files

              Message routing for large number of cores

       <b>-route-messages</b>

              Enables the Ray message router. Disabled by default.  Messages will be routed accordingly so  that
              any  rank  can communicate directly with only a few others.  Without <b>-route-messages</b>, any rank can
              communicate  directly  with   any   other   rank.    Files   generated:   Routing/Connections.txt,
              Routing/Routes.txt and Routing/RelayEvents.txt and Routing/Summary.txt

       <b>-connection-type</b> type

              Sets  the  connection  type for routes.  Accepted values are debruijn, hypercube, polytope, group,
              random, kautz and complete. Default is debruijn.

              debruijn: a full de Bruijn graph a given alphabet and diameter hypercube: a hypercube, alphabet is
              {0,1} and the vertices is  a  power  of  2  polytope:  a  convex  regular  polytope,  alphabet  is
              {0,1,...,B-1}  and  the  vertices  is a power of B group: silly model where one representative per
              group can communicate with outsiders random: Erdos-Renyi model kautz: a full de Kautz graph, which
              is a subgraph of a de Bruijn graph complete: a full graph with all the possible connections

              With the type debruijn, the number of ranks must be a power of something.  Examples: 256 =  16*16,
              512=8*8*8,  49=7*7, and so on.  Otherwise, don't use debruijn routing but use another one With the
              type kautz, the number of ranks n must be n=(k+1)*k^(d-1) for some k and d

       <b>-routing-graph-degree</b> degree

              Specifies the outgoing degree for the routing graph.  See Documentation/Routing.txt

              Hardware testing

       <b>-test-network-only</b>

              Tests the network and returns.

       <b>-write-network-test-raw-data</b>

              Writes one additional file per rank detailing the network test.

       <b>-exchanges</b> NumberOfExchanges

              Sets the number of exchanges

       <b>-disable-network-test</b>

              Skips the network test.

              Debugging

       <b>-verify-message-integrity</b>

              Checks message data reliability for  any  non-empty  message.   add  '-D  CONFIG_SSE_4_2'  in  the
              Makefile to use hardware instruction (SSE 4.2)

       <b>-run-profiler</b>

              Runs  the  profiler  as  the  code  runs. By default, only show granularity warnings.  Running the
              profiler increases running times.

       <b>-with-profiler-details</b>

              Shows number of messages sent and received in each methods during in each  time  slices  (epochs).
              Needs <b>-run-profiler</b>.

       <b>-show-communication-events</b>

              Shows all messages sent and received.

       <b>-show-read-placement</b>

              Shows read placement in the graph during the extension.

       <b>-debug-bubbles</b>

              Debugs  bubble  code.   Bubbles  can  be  due  to heterozygous sites or sequencing errors or other
              (unknown) events

       <b>-debug-seeds</b>

              Debugs seed code.  Seeds are paths in the graph that are likely unique.

       <b>-debug-fusions</b>

              Debugs fusion code.

       <b>-debug-scaffolder</b>

              Debug the scaffolder.

       FILES

              Input files

              Note: file format is determined with file extension.

              .fasta  .fasta.gz  (needs  HAVE_LIBZ=y  at  compilation)  .fasta.bz2   (needs   HAVE_LIBBZ2=y   at
              compilation)  .fastq  .fastq.gz (needs HAVE_LIBZ=y at compilation) .fastq.bz2 (needs HAVE_LIBBZ2=y
              at compilation) .sff (paired reads must be extracted manually) .csfasta (color-space reads)

              Outputted files

              Scaffolds

              RayOutput/Scaffolds.fasta

              The scaffold sequences in FASTA format

              RayOutput/ScaffoldComponents.txt

              The components of each scaffold

              RayOutput/ScaffoldLengths.txt

              The length of each scaffold

              RayOutput/ScaffoldLinks.txt

              Scaffold links

              Contigs

              RayOutput/Contigs.fasta

              Contiguous sequences in FASTA format

              RayOutput/ContigLengths.txt

              The lengths of contiguous sequences

              Summary

              RayOutput/OutputNumbers.txt

              Overall numbers for the assembly

              de Bruijn graph

              RayOutput/CoverageDistribution.txt

              The distribution of coverage values

              RayOutput/CoverageDistributionAnalysis.txt

              Analysis of the coverage distribution

              RayOutput/degreeDistribution.txt

              Distribution of ingoing and outgoing degrees

              RayOutput/kmers.txt

              k-mer graph, required option: <b>-write-kmers</b>

              The resulting file is not utilised by Ray.  The resulting file is very large.

              Assembly steps

              RayOutput/SeedLengthDistribution.txt

              Distribution of seed length

              RayOutput/Rank&lt;rank&gt;.OptimalReadMarkers.txt

              Read markers.

              RayOutput/Rank&lt;rank&gt;.RaySeeds.fasta

              Seed DNA sequences, required option: <b>-write-seeds</b>

              RayOutput/Rank&lt;rank&gt;.RayExtensions.fasta

              Extension DNA sequences, required option: <b>-write-extensions</b>

              RayOutput/Rank&lt;rank&gt;.RayContigPaths.txt

              Contig paths with coverage values, required option: <b>-write-contig-paths</b>

              Paired reads

              RayOutput/LibraryStatistics.txt

              Estimation of outer distances for paired reads

              RayOutput/Library&lt;LibraryNumber&gt;.txt

              Frequencies for observed outer distances (insert size + read lengths)

              Partition

              RayOutput/NumberOfSequences.txt

              Number of reads in each file

              RayOutput/SequencePartition.txt

              Sequence partition

              Ray software

              RayOutput/RayVersion.txt

              The version of Ray

              RayOutput/RayCommand.txt

              The exact same command provided

              AMOS

              RayOutput/AMOS.afg

              Assembly representation in AMOS format, required option: <b>-amos</b>

              Communication

              RayOutput/MessagePassingInterface.txt

              Number of messages sent

              RayOutput/NetworkTest.txt

              Latencies in microseconds

              RayOutput/Rank&lt;rank&gt;NetworkTestData.txt

              Network test raw data

       DOCUMENTATION

              - mpiexec <b>-n</b> 1 Ray <b>-help</b>|less (always up-to-date) - This  help  page  (always  up-to-date)  -  The
              directory   Documentation/   -   Manual  (Portable  Document  Format):  InstructionManual.tex  (in
              Documentation)               -                Mailing                list                archives:
              <a href="http://sourceforge.net/mailarchive/forum.php">http://sourceforge.net/mailarchive/forum.php</a>?forum_name=denovoassembler-users

       AUTHOR

              Written by Sebastien Boisvert.

       REPORTING BUGS

              Report       bugs       to       <a href="mailto:denovoassembler-users@lists.sourceforge.net">denovoassembler-users@lists.sourceforge.net</a>       Home      page:
              &lt;<a href="http://denovoassembler.sourceforge.net/">http://denovoassembler.sourceforge.net/</a>&gt;

       COPYRIGHT

              This program is free software: you can redistribute it and/or modify it under the terms of the GNU
              General Public License as published by the Free Software Foundation, version 3 of the License.

              This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;  without
              even  the  implied  warranty  of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
              General Public License for more details.

              You have received a copy of the GNU General Public License along with this program (see LICENSE).

       Ray 2.1.0

       License for Ray: GNU General Public License version 3 RayPlatform version: 1.1.0 License for RayPlatform:
       GNU Lesser General Public License version 3

       MAXKMERLENGTH: 32 KMER_U64_ARRAY_SIZE: 1 Maximum  coverage  depth  stored  by  CoverageDepth:  4294967295
       MAXIMUM_MESSAGE_SIZE_IN_BYTES:  4000  bytes  FORCE_PACKING  =  n ASSERT = n HAVE_LIBZ = y HAVE_LIBBZ2 = y
       CONFIG_PROFILER_COLLECT = n CONFIG_CLOCK_GETTIME = n __linux__ = y _MSC_VER = n __GNUC__ = y  RAY_32_BITS
       = n RAY_64_BITS = y MPI standard version: MPI 2.1 MPI library: Open-MPI 1.4.2 Compiler: GNU gcc/g++ 4.4.5

Ray version 2.1.0                                 November 2012                                           <u><a href="../man1/RAY.1.html">RAY</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>