<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>abacas - Algorithm Based Automatic Contiguation of Assembled Sequences</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/abacas">abacas_1.3.1-9_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       abacas - Algorithm Based Automatic Contiguation of Assembled Sequences

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>abacas</b> <b>-r</b> <u>ref</u> <b>-q</b> <u>qs</u> <b>-p</b> <u>prog</u>  [OPTIONS]

       OR

       <b>abacas</b> <b>-r</b> <u>ref</u> <b>-q</b> <u>psf</u> <b>-e</b>

       <u>ref</u>    reference sequence in a single fasta file

       <u>qs</u>     contigs in multi-fasta format MUMmer program to use 'nucmer' or 'promer'

       <u>psf</u>    pseudomolecule/ordered sequence file in fasta format

       <b>OPTIONS</b>

       <b>-h</b>     print usage

       <b>-d</b>     use default nucmer/promer parameters

       <b>-s</b>     int     minimum length of exact matching word (nucmer default = 12, promer default = 4)

       <b>-m</b>     print ordered contigs to file in multifasta format

       <b>-b</b>     print contigs in bin to file

       <b>-N</b>     print a pseudomolecule without "N"s

       <b>-i</b>     int     minimum percent identity [default 40]

       <b>-v</b>     int     minimum contig coverage [default 40]

       <b>-V</b>     int     minimum contig coverage difference [default 1]

       <b>-l</b>     int     minimum contig length [default 1]

       <b>-t</b>     run tblastx on contigs that are not mapped

       <b>-g</b>     string (file name)      print uncovered regions (gaps) on reference to file name

       <b>-a</b>     append contigs in bin to the pseudomolecule

       <b>-o</b>     prefix  output files will have this prefix

       <b>-P</b>     pick primer sets to close gaps

       <b>-f</b>     int     number of flanking bases on either side of a gap for primer design (default 350)

       <b>-R</b>     int     Run mummer [default 1, use <b>-R</b> 0 to avoid running mummer]

       <b>-e</b>     Escape contig ordering i.e. go to primer design

       <b>-c</b>     Reference sequence is circular

</pre><h4><b>DESCRIPTION</b></h4><pre>
       ABACAS is intended to rapidly contiguate (align, order, orientate), visualize and design primers to close
       gaps on shotgun assembled contigs based on a reference sequence.

       ABACAS  uses  MUMmer  to find alignment positions and identify syntenies of assembled contigs against the
       reference. The output is then processed to generate a pseudomolecule taking overlapping contigs and  gaps
       in  to  account.  ABACAS  generates  a comparison file that can be used to visualize ordered and oriented
       contigs in ACT. Synteny is represented by red bars where colour intensity decreases with lower values  of
       percent  identity  between  comparable  blocks.  Information  on contigs such as the orientation, percent
       identity, coverage and overlap with other contigs can also be visualized by loading the outputted feature
       file on ACT.

</pre><h4><b>AUTHOR</b></h4><pre>
       ABACAS IS Copyright (C) 2008-10 The Wellcome Trust Sanger Institute, Cambridge, UK.

       This manual page was written by Andreas Tille &lt;<a href="mailto:tille@debian.org">tille@debian.org</a>&gt;, for the Debian project (and may be used
       by others).

1.3.1                                              2011-02-11                                          <u><a href="../man1/ABACAS.1.html">ABACAS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>