<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>gt-gff3 - Parse, possibly transform, and output GFF3 files.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/genometools">genometools_1.6.5+ds-2.2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       gt-gff3 - Parse, possibly transform, and output GFF3 files.

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>gt</b> <b>gff3</b> [option ...] [GFF3_file ...]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>-sort</b> [<u>yes|no</u>]
           sort the GFF3 features (memory consumption is proportional to the input file size(s)) (default: no)

       <b>-sortlines</b> [<u>yes|no</u>]
           sort the GFF3 features on a strict line basis (not sorted asdefined by GenomeTools) (default: no)

       <b>-sortnum</b> [<u>yes|no</u>]
           enable natural numeric sorting for sequence regions (not sorted as defined by GenomeTools) (default:
           no)

       <b>-tidy</b> [<u>yes|no</u>]
           try to tidy the GFF3 files up during parsing (default: no)

       <b>-retainids</b> [<u>yes|no</u>]
           when available, use the original IDs provided in the source file (memory consumption is proportional
           to the input file size(s)) (default: no)

       <b>-checkids</b> [<u>yes|no</u>]
           make sure the ID attributes are unique within the scope of each GFF3_file, as required by GFF3
           specification (memory consumption is proportional to the input file size(s)). If features with the
           same Parent attribute are not separated by a <u>#</u> line the GFF3 parser tries to treat them as a
           multi-line feature. This requires at least matching sequence IDs and types. (default: no)

       <b>-addids</b> [<u>yes|no</u>]
           add missing "##sequence-region" lines automatically (default: yes)

       <b>-fixregionboundaries</b> [<u>yes|no</u>]
           automatically adjust "##sequence-region" lines to contain all their features (memory consumption is
           proportional to the input file size(s)) (default: no)

       <b>-addintrons</b> [<u>yes|no</u>]
           add intron features between existing exon features (default: no)

       <b>-offset</b> [<u>value</u>]
           transform all features by the given offset

       <b>-offsetfile</b> [<u>filename</u>]
           transform all features by the offsets given in file (default: undefined)

       <b>-setsource</b> [<u>string</u>]
           set the <u>source</u> value (2nd column) of each feature (default: undefined)

       <b>-typecheck</b> [<u>string</u>]
           use an ontology given in an OBO file to validate parent-child relationships. If no argument is given,
           the sofa.obo file from the gtdata/obo_files directory is used. If an argument is given, it is used as
           an OBO filename. In the case that such a file does not exist <u>.obo</u> is added to the argument and
           loading the resulting filename from the gtdata/obo_files directory is attempted. (default: undefined)

       <b>-xrfcheck</b> [<u>string</u>]
           check Dbxref and Ontology_term attributes for correct syntax according to a abbreviation definition
           file. If no argument is given, the GO.xrf_abbs file from the gtdata/xrf_abbr directory is used. If an
           argument is given, it is used as an specific filename for an abbreviation file. In the case that such
           a file does not exist, <u>.xrf_abbr</u> is added to the argument and loading the resulting filename from the
           gtdata/xrf_abbr directory is attempted. (default: undefined)

       <b>-show</b> [<u>yes|no</u>]
           show GFF3 output (default: yes)

       <b>-v</b> [<u>yes|no</u>]
           be verbose (default: no)

       <b>-width</b> [<u>value</u>]
           set output width for FASTA sequence printing (0 disables formatting) (default: 0)

       <b>-o</b> [<u>filename</u>]
           redirect output to specified file (default: undefined)

       <b>-gzip</b> [<u>yes|no</u>]
           write gzip compressed output file (default: no)

       <b>-bzip2</b> [<u>yes|no</u>]
           write bzip2 compressed output file (default: no)

       <b>-force</b> [<u>yes|no</u>]
           force writing to output file (default: no)

       <b>-help</b>
           display help and exit

       <b>-version</b>
           display version information and exit

       File format for option <u>-offsetfile</u>:

       The file supplied to option <u>-offsetfile</u> defines a mapping table named “offsets”. It maps the
       sequence-region entries given in the GFF3_file to offsets. It can be defined as follows:

           offsets = {
             chr1  = 1000,
             chr2  = 500
           }

       When this example is used, all features with seqid “chr1” will be offset by 1000 and all features with
       seqid “chr2” by 500.

       If <u>-offsetfile</u> is used, offsets for all sequence-regions contained in the given GFF3 files must be
       defined.

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       Report bugs to https://github.com/genometools/genometools/issues.

GenomeTools 1.6.5                                  04/27/2024                                         <u><a href="../man1/GT-GFF3.1.html">GT-GFF3</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>