<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>gt-ltrharvest - Predict LTR retrotransposons.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/genometools">genometools_1.6.5+ds-2.2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       gt-ltrharvest - Predict LTR retrotransposons.

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>gt</b> <b>ltrharvest</b> [option ...] -index &lt;indexname&gt;

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>-index</b> [<u>string</u>]
           specify the name of the enhanced suffix array index (mandatory) (default: undefined)

       <b>-range</b> [<u>start</u> <u>end</u>]
           specify range in the input sequence(s) in which LTR pairs are searched (default: [0..0])

       <b>-seed</b> [<u>value</u>]
           specify minimum seed length for exact repeats (default: 30)

       <b>-minlenltr</b> [<u>value</u>]
           specify minimum length for each LTR (default: 100)

       <b>-maxlenltr</b> [<u>value</u>]
           specify maximum length for each LTR (default: 1000)

       <b>-mindistltr</b> [<u>value</u>]
           specify minimum distance of LTR startpositions (default: 1000)

       <b>-maxdistltr</b> [<u>value</u>]
           specify maximum distance of LTR startpositions (default: 15000)

       <b>-similar</b> [<u>value</u>]
           specify similaritythreshold in range [1..100%] (default: 85.000000)

       <b>-mintsd</b> [<u>value</u>]
           specify minimum length for each TSD (default: 4)

       <b>-maxtsd</b> [<u>value</u>]
           specify maximum length for each TSD (default: 20)

       <b>-motif</b> [<u>string</u>]
           specify 2 nucleotides startmotif + 2 nucleotides endmotif: <b>*</b>* (default: undefined)

       <b>-motifmis</b> [<u>value</u>]
           specify maximum number of mismatches in motif [0,3] (default: 4)

       <b>-vic</b> [<u>value</u>]
           specify the number of nucleotides (to the left and to the right) that will be searched for TSDs
           and/or motifs around 5' and 3' boundary of predicted LTR retrotransposons (default: 60)

       <b>-overlaps</b> [<u>...</u>]
           specify no|best|all (default: best)

       <b>-xdrop</b> [<u>value</u>]
           specify xdropbelowscore for extension-alignment (default: 5)

       <b>-mat</b> [<u>value</u>]
           specify matchscore for extension-alignment (default: 2)

       <b>-mis</b> [<u>value</u>]
           specify mismatchscore for extension-alignment (default: -2)

       <b>-ins</b> [<u>value</u>]
           specify insertionscore for extension-alignment (default: -3)

       <b>-del</b> [<u>value</u>]
           specify deletionscore for extension-alignment (default: -3)

       <b>-v</b> [<u>yes|no</u>]
           verbose mode (default: no)

       <b>-tabout</b> [<u>yes|no</u>]
           show <u>old</u> tabular output instead of GFF3 on stdout (default: yes)

       <b>-seqids</b> [<u>yes|no</u>]
           use sequence descriptions instead of sequence numbers in GFF3 output (default: no)

       <b>-md5</b> [<u>yes|no</u>]
           add MD5 hashes to seqids in GFF3 output (default: no)

       <b>-longoutput</b> [<u>yes|no</u>]
           additional motif/TSD output (default: no)

       <b>-out</b> [<u>string</u>]
           specify FASTA outputfilename (default: undefined)

       <b>-outinner</b> [<u>string</u>]
           specify FASTA outputfilename for inner regions (default: undefined)

       <b>-gff3</b> [<u>string</u>]
           specify GFF3 outputfilename (default: undefined)

       <b>-offset</b> [<u>value</u>]
           offset added to GFF3 coordinates (default: 0)

       <b>-scan</b> [<u>yes|no</u>]
           scan the index sequentially instead of mapping it into memory entirely (default: yes)

       <b>-help</b>
           display help for basic options and exit

       <b>-help+</b>
           display help for all options and exit

       <b>-version</b>
           display version information and exit

</pre><h4><b>ADDITIONAL</b> <b>INFORMATION</b></h4><pre>
       For detailed information, please refer to the manual of ltrharvest.

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       Report bugs to https://github.com/genometools/genometools/issues.

GenomeTools 1.6.5                                  04/27/2024                                   <u><a href="../man1/GT-LTRHARVEST.1.html">GT-LTRHARVEST</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>