<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>gt-tirvish - Identify Terminal Inverted Repeat (TIR) elements,such as DNA transposons.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/genometools">genometools_1.6.5+ds-2.2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       gt-tirvish - Identify Terminal Inverted Repeat (TIR) elements,such as DNA transposons.

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>gt</b> <b>tirvish</b> [option ...] -index INDEXNAME

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>-index</b> [<u>string</u>]
           specify the name of the enhanced suffix array index (mandatory) (default: undefined)

       <b>-seed</b> [<u>value</u>]
           specify minimum seed length for exact repeats (default: 20)

       <b>-mintirlen</b> [<u>value</u>]
           specify minimum length for each TIR (default: 100)

       <b>-maxtirlen</b> [<u>value</u>]
           specify maximum length for each TIR (default: 1000)

       <b>-mintirdist</b> [<u>value</u>]
           specify minimum distance of TIRs (default: 500)

       <b>-maxtirdist</b> [<u>value</u>]
           specify maximum distance of TIRs (default: 10000)

       <b>-mat</b> [<u>value</u>]
           specify matchscore for extension-alignment (default: 2)

       <b>-mis</b> [<u>value</u>]
           specify mismatchscore for extension-alignment (default: -2)

       <b>-ins</b> [<u>value</u>]
           specify insertionscore for extension-alignment (default: -3)

       <b>-del</b> [<u>value</u>]
           specify deletionscore for extension-alignment (default: -3)

       <b>-xdrop</b> [<u>value</u>]
           specify xdropbelowscore for extension-alignment (default: 5)

       <b>-similar</b> [<u>value</u>]
           specify TIR similarity threshold in therange [1..100%] (default: 85.000000)

       <b>-overlaps</b> [<u>...</u>]
           specify no|best|longest|all (default: best)

       <b>-mintsd</b> [<u>value</u>]
           specify minimum length for each TSD (default: 2)

       <b>-maxtsd</b> [<u>value</u>]
           specify maximum length for each TSD (default: 11)

       <b>-vic</b> [<u>value</u>]
           specify the number of nucleotides (to the left and to the right) that will be searched for TSDs
           around 5' and 3' boundary of predicted TIRs (default: 60)

       <b>-hmms</b>
           profile HMM models for domain detection (separate by spaces, finish with --) in HMMER3 format Omit
           this option to disable pHMM search.

       <b>-pdomevalcutoff</b> [<u>value</u>]
           global E-value cutoff for pHMM search default 1E-6

       <b>-pdomcutoff</b> [<u>...</u>]
           model-specific score cutoff choose from TC (trusted cutoff) | GA (gathering cutoff) | NONE (no
           cutoffs) (default: GA)

       <b>-maxgaplen</b> [<u>value</u>]
           maximal allowed gap size between fragments (in amino acids) when chaining pHMM hits for a protein
           domain (default: 50)

       <b>-refseqs</b> [<u>string</u>]
           specify the name of the gene sequences to scan for inside candidates (default: undefined)

       <b>-seqids</b> [<u>yes|no</u>]
           use sequence descriptions instead of sequence numbers in GFF3 output (default: yes)

       <b>-md5</b> [<u>yes|no</u>]
           add MD5 hashes to seqids in GFF3 output (default: no)

       <b>-help</b>
           display help for basic options and exit

       <b>-help+</b>
           display help for all options and exit

       <b>-version</b>
           display version information and exit

</pre><h4><b>REPORTING</b> <b>BUGS</b></h4><pre>
       Report bugs to https://github.com/genometools/genometools/issues.

GenomeTools 1.6.5                                  04/27/2024                                      <u><a href="../man1/GT-TIRVISH.1.html">GT-TIRVISH</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>