<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>shasta - nanopore whole genome assembly tool</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/shasta">shasta_0.12.0-1build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       shasta - nanopore whole genome assembly tool

</pre><h4><b>DESCRIPTION</b></h4><pre>
   <b>Options</b> <b>allowed</b> <b>only</b> <b>on</b> <b>the</b> <b>command</b> <b>line:</b>
       <b>-h</b> [ <b>--help</b> ]
              Write a help message.

       <b>-v</b> [ <b>--version</b> ]
              Identify the Shasta version.

       <b>--config</b> arg
              Configuration file name.

       <b>--input</b> arg
              Names of input files containing reads.  Specify at least one.

       <b>--assemblyDirectory</b> arg (=ShastaRun)
              Name of the output directory. If command is assemble, this directory must not exist.

       <b>--command</b> arg (=assemble)
              Command   to   run.   Must  be  one  of:  assemble,  saveBinaryData,  cleanupBinaryData,  explore,
              createBashCompletionScript

       <b>--memoryMode</b> arg (=anonymous)
              Specify whether allocated memory  is  anonymous  or  backed  by  a  filesystem.   Allowed  values:
              anonymous, filesystem.

       <b>--memoryBacking</b> arg (=4K)
              Specify  the  type  of  pages  used  to  back  memory.   Allowed  values:  disk, 4K , 2M (for best
              performance). All combinations (memoryMode, memoryBacking)  are  allowed  except  for  (anonymous,
              disk).  Some combinations require root privilege, which is obtained using sudo and may result in a
              password prompting depending on your sudo set up.

       <b>--threads</b> arg (=0)
              Number of threads, or 0 to use one thread per virtual processor.

       <b>--exploreAccess</b> arg (=user)
              Specify  allowed  access  for <b>--command</b> explore. Allowed values: user, local, unrestricted. DO NOT
              CHANGE FROM DEFAULT VALUE WITHOUT UNDERSTANDING THE SECURITY IMPLICATIONS.

       <b>--port</b> arg (=17100)
              Port to be used by the http server (command <b>--explore</b>).

   <b>Options</b> <b>allowed</b> <b>on</b> <b>the</b> <b>command</b> <b>line</b> <b>and</b> <b>in</b> <b>the</b> <b>config</b> <b>file:</b>
       <b>--Reads</b>.minReadLength arg (=10000)
              Read length cutoff. Shorter reads are discarded.

       <b>--Reads</b>.noCache
              If set, skip the Linux cache when loading reads. This is done by specifying the O_DIRECT flag when
              opening input files containing reads.

       <b>--Reads</b>.palindromicReads.skipFlagging
              Skip flagging palindromic reads. Oxford Nanopore reads should be flagged for better results.

       <b>--Reads</b>.palindromicReads.maxSkip arg (=100)
              Used for palindromic read detection.

       <b>--Reads</b>.palindromicReads.maxDrift arg (=100)
              Used for palindromic read detection.

       <b>--Reads</b>.palindromicReads.maxMarkerFrequency arg (=10)
              Used for palindromic read detection.

       <b>--Reads</b>.palindromicReads.alignedFractionThreshold arg (=0.1)
              Used for palindromic read detection.

       <b>--Reads</b>.palindromicReads.nearDiagonalFractionThreshold arg (=0.1)
              Used for palindromic read detection.

       <b>--Reads</b>.palindromicReads.deltaThreshold arg (=100)
              Used for palindromic read detection.

       <b>--Kmers</b>.generationMethod arg (=0)
              Method to generate marker k-mers: 0 = random, 1 =  random,  excluding  globally  overenriched,2  =
              random, excluding overenriched even in a single read,3 = read from file.

       <b>--Kmers</b>.k arg (=10)
              Length of marker k-mers (in run-length space).

       <b>--Kmers</b>.probability arg (=0.1)
              Fraction k-mers used as a marker.

       <b>--Kmers</b>.enrichmentThreshold arg (=100.)
              Enrichment threshold for Kmers.generationMethod 1 and 2.

       <b>--Kmers</b>.file arg
              The  absolute path of a file containing the k-mers to be used as markers, one per line. A relative
              path is not accepted. Only used if Kmers.generationMethod is 3.

       <b>--MinHash</b>.version arg (=0)
              Controls the version of the LowHash algorithm to use. Can be 0 (default) or 1.(experimental).

       <b>--MinHash</b>.m arg (=4)
              The number of consecutive markers that define a MinHash/LowHash feature.

       <b>--MinHash</b>.hashFraction arg (=0.01)
              Defines how low a hash has to be to be used with the LowHash algorithm.

       <b>--MinHash</b>.minHashIterationCount arg (=10)
              The number of MinHash/LowHash iterations, or 0 to let <b>--MinHash</b>.alignmentCandidatesPerRead control
              the number of iterations.

       <b>--MinHash</b>.alignmentCandidatesPerRead arg (=20)
              If <b>--MinHash</b>.minHashIterationCount is 0, MinHash iteration is stopped when the average  number  of
              alignment    candidates    that    each   read   is   involved   in   reaches   this   value.   If
              <b>--MinHash</b>.minHashIterationCount is not 0, this is not used.

       <b>--MinHash</b>.minBucketSize arg (=0)
              The minimum bucket size to be used by the LowHash algorithm.

       <b>--MinHash</b>.maxBucketSize arg (=10)
              The maximum bucket size to be used by the LowHash algorithm.

       <b>--MinHash</b>.minFrequency arg (=2)
              The minimum number of times a pair of reads must be found  by  the  MinHash/LowHash  algorithm  in
              order to be considered a candidate alignment.

       <b>--MinHash</b>.allPairs
              Skip  the  MinHash  algorithm  and  mark  all  pairs  of  reads  as  alignmentcandidates with both
              orientation. This should only be used for experimentation on very small runs because  it  is  very
              time consuming.

       <b>--Align</b>.alignMethod arg (=3)
              The  alignment  method  to  be  used  to  create the read graph &amp; the marker graph. 0 = old Shasta
              method, 1 = SeqAn (slow), 3 = banded SeqAn.

       <b>--Align</b>.maxSkip arg (=30)
              The maximum number of markers that an alignment is allowed to skip.

       <b>--Align</b>.maxDrift arg (=30)
              The maximum amount of marker drift that an alignment is allowed  to  tolerate  between  successive
              markers.

       <b>--Align</b>.maxTrim arg (=30)
              The maximum number of unaligned markers tolerated at the beginning and end of an alignment.

       <b>--Align</b>.maxMarkerFrequency arg (=10)
              Marker  frequency threshold. Markers more frequent than this value in either of two oriented reads
              being aligned are discarded and not used to compute the alignment.

       <b>--Align</b>.minAlignedMarkerCount arg (=100)
              The minimum number of aligned markers for an alignment to be used.

       <b>--Align</b>.minAlignedFraction arg (=0)
              The minimum fraction of aligned markers for an alignment to be used.

       <b>--Align</b>.matchScore arg (=6)
              Match score for marker alignments (only used for alignment methods 1 and 3).

       <b>--Align</b>.mismatchScore arg (=-1)
              Mismatch score for marker alignments (only used for alignment methods 1 and 3).

       <b>--Align</b>.gapScore arg (=-1)
              Gap score for marker alignments (only used for alignment methods 1 and 3).

       <b>--Align</b>.downsamplingFactor arg (=0.10000000000000001)
              Downsampling factor (only used for alignment method 3).

       <b>--Align</b>.bandExtend arg (=10)
              Amount to extend the downsampled band (only used for alignment method 3).

       <b>--Align</b>.sameChannelReadAlignment.suppressDeltaThreshold arg (=0)
              If not zero, alignments between reads from the  same  nanopore  channel  and  close  in  time  are
              suppressed.  The  "read"  meta  data  fields  from the FASTA or FASTQ header are checked. If their
              difference, in absolute value, is less than the value of this option, the alignment is suppressed.
              This can help avoid assembly artifact. This check is only done if the  two  reads  have  identical
              meta data fields "runid", "sampleid", and "ch". If any of these meta data fields are missing, this
              check is suppressed and this option has no effect.

       <b>--Align</b>.suppressContainments
              Suppress  containment  alignments,  that  is alignments in which one read is entirely contained in
              another read, except possibly for up to maxTrim markers at the beginning and end.

       <b>--ReadGraph</b>.creationMethod arg (=0)
              The method used to create the read graph (0 = undirected, default, 1 = directed, experimental).

       <b>--ReadGraph</b>.maxAlignmentCount arg (=6)
              The maximum number of alignments to be kept for each read.

       <b>--ReadGraph</b>.minComponentSize arg (=100)
              The minimum size (number of oriented reads) of a connected component of the read graph to be kept.
              This is currently ignored.

       <b>--ReadGraph</b>.maxChimericReadDistance arg (=2)
              Used for chimeric read detection.

       <b>--ReadGraph</b>.crossStrandMaxDistance arg (=6)
              Maximum distance (edges) for flagCrossStrandReadGraphEdges. Set this to zero to entirely  suppress
              flagCrossStrandReadGraphEdges.

       <b>--ReadGraph</b>.containedNeighborCount arg (=6)
              Maximum  number of alignments to be kept for each contained read (only used when creationMethod is
              1).

       <b>--ReadGraph</b>.uncontainedNeighborCountPerDirection arg (=3)
              Maximum number of alignments to be kept in each direction (forward, backward) for each uncontained
              read (only used when creationMethod is 1).

       <b>--ReadGraph</b>.removeConflicts
              Remove conflicts from the read graph.  Experimental - do not use.

       <b>--MarkerGraph</b>.minCoverage arg (=10)
              Minimum number of markers for a marker graph vertex.

       <b>--MarkerGraph</b>.maxCoverage arg (=100)
              Maximum number of markers for a marker graph vertex.

       <b>--MarkerGraph</b>.lowCoverageThreshold arg (=0)
              Used during approximate transitive reduction. Marker graph edges with  coverage  lower  than  this
              value are always marked as removed regardless of reachability.

       <b>--MarkerGraph</b>.highCoverageThreshold arg (=256)
              Used  during  approximate  transitive reduction. Marker graph edges with coverage higher than this
              value are never marked as removed regardless of reachability.

       <b>--MarkerGraph</b>.maxDistance arg (=30)
              Used during approximate transitive reduction.

       <b>--MarkerGraph</b>.edgeMarkerSkipThreshold arg (=100)
              Used during approximate transitive reduction.

       <b>--MarkerGraph</b>.pruneIterationCount arg (=6)
              Number of prune iterations.

       <b>--MarkerGraph</b>.simplifyMaxLength arg (=10,100,1000)
              Maximum lengths (in markers) used at each iteration of simplifyMarkerGraph.

       <b>--MarkerGraph</b>.crossEdgeCoverageThreshold arg (=0)
              Experimental. Cross edge coverage threshold. If this is not zero, assembly graph cross-edges  with
              average  edge  coverage  less  than this value are removed, together with the corresponding marker
              graph edges. A cross edge is defined as an edge v0-&gt;v1 with out-degree(v0)&gt;1, in-degree(v1)&gt;1.

       <b>--MarkerGraph</b>.refineThreshold arg (=0)
              Experimental. Length threshold, in markers, for the marker graph refinement step, or 0 to turn off
              the refinement step.

       <b>--MarkerGraph</b>.reverseTransitiveReduction
              Perform approximate reverse transitive reduction of the marker graph.

       <b>--Assembly</b>.crossEdgeCoverageThreshold arg (=3)
              Maximum average edge coverage for a cross edge of the assembly graph to be removed.

       <b>--Assembly</b>.markerGraphEdgeLengthThresholdForConsensus arg (=1000)
              Controls assembly of long marker graph edges.

       <b>--Assembly</b>.consensusCaller arg (=Bayesian:guppy-2.3.5-a)
              Selects the consensus caller for repeat counts. See the documentation for available choices.

       <b>--Assembly</b>.storeCoverageData
              Used to request storing coverage data in binary format.

       <b>--Assembly</b>.storeCoverageDataCsvLengthThreshold arg (=0)
              Used to specify the minimum length of an assembled segment for which coverage data in  csv  format
              should be stored. If 0, no coverage data in csv format is stored.

       <b>--Assembly</b>.writeReadsByAssembledSegment
              Used to request writing the reads that contributed to assembling each segment.

       <b>--Assembly</b>.detangleMethod arg (=0)
              Experimental. Specify the method used to detangle the assembly graph. 0 = no detangling, 1 = basic
              detangling.

shasta                                             August 2020                                         <u><a href="../man1/SHASTA.1.html">SHASTA</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>