<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>QTLtools fenrich - Functional enrichment of molecular QTLs</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/qtltools">qtltools_1.3.1+dfsg-4build4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       QTLtools fenrich - Functional enrichment of molecular QTLs

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>QTLtools</b> <b>fenrich</b> <b>--qtl</b> <u>significanty_genes.bed</u> <b>--tss</b> <u>gene_tss.bed</u> <b>--bed</b> <u>TFs.encode.bed.gz</u> <b>--out</b> <u>output.txt</u>
       <u>[OPTIONS]</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This  mode allows assessing whether a set of QTLs fall within some functional annotations more often than
       what is expected by chance.  The method  is  detailed  in  &lt;https://www.nature.com/articles/ncomms15452&gt;.
       Here, we mean by chance is what is expected given the non-uniform distributions of molQTLs and functional
       annotations  around  the genomic positions of the molecular phenotypes.  To do so, we first enumerate all
       the functional annotations located nearby a given molecular phenotype.  In  practice,  for  X  phenotypes
       being  quantified,  we  have X lists of annotations.  And, for the subset Y of those having a significant
       molQTL, we count how often the Y molQTLs overlap the annotations in the corresponding lists:  this  gives
       the  observed overlap frequency fobs(Y) between molQTLs and functional annotations.  Then, we permute the
       lists of functional annotations across the phenotypes (e.g, phenotype A  may  be  assigned  the  list  of
       annotations  coming from phenotype B) and for each permuted data set, we count how often the Y molQTLs do
       overlap the newly assigned functional annotations: this gives  the  expected  overlap  frequency  fexp(Y)
       between  molQTLs  and functional annotations.  By doing this permutation scheme, we keep the distribution
       of functional annotations and molQTLs around molecular  phenotypes  unchanged.   Now  that  we  have  the
       observed and expected overlap frequencies, we use a fisher test to assess how fobs(Y) and fexp(Y) differ.
       This  gives  an odd ratio estimate and a two-sided p-value which basically tells us first if there is and
       enrichment or depletion, and second how significant this is.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>--qtl</b> <u>in.bed</u>
              List of QTLs of interest in BED format.  REQUIRED.

       <b>--bed</b> <u>functional_annotation.bed.gz</u>
              Functional annotations in BED format.  REQUIRED.

       <b>--tss</b> <u>genes.bed</u>
              List of positions of all phenotypes you mapped QTLs for, in BED format.  REQUIRED.

       <b>--out</b> <u>output.txt</u>
              Output file.  REQUIRED.

       <b>--permute</b> <u>integer</u>
              Number of permutation to run.  DEFAULT=1000

</pre><h4><b>INPUT</b> <b>FILES</b></h4><pre>
       <b>--qtl</b> <b>file</b>
        List of QTLs of interest.  An example:

        1    15210     15211     1_15211   ENSG00000227232.4   -
        1    735984    735985    1_735985  ENSG00000177757.1   +
        1    735984    735985    1_735985  ENSG00000240453.1   -
        1    739527    739528    1_739528  ENSG00000237491.4   +

        The column definitions are:
        1   The variant chromosome
        2   The variant's start position (<b>0-based</b>)
        3   The variant's end position (<b>1-based</b>)
        4   The variant ID
        5   The phenotype ID
        6   The phenotype's strand. (<b>not</b> <b>used</b>)

       <b>--bed</b> <b>file</b>
        List of annotations in BED format.  An example:

        1    254874    265487
        1    730984    735985
        1    734984    736585
        1    739527    748528

        The column definitions are:
        1   Chromosome
        2   Start position (<b>0-based</b>)
        3   End position (<b>1-based</b>)

       <b>--tss</b> <b>file</b>
        List of positions of all phenotypes you mapped QTLs for.  An example:

        1    29369     29370     ENSG00000227232.4   1_15211   -
        1    135894    135895    ENSG00000268903.1   1_985446  -
        1    137964    137965    ENSG00000269981.1   1_1118728 -
        1    317719    317720    ENSG00000237094.7   1_15211   +

        The column definitions are:
        1   Phenotype's chromosome
        2   The start position of the phenotype (<b>0-based</b>)
        3   The end position of the phenotype (<b>1-based</b>)
        4   The phenotype ID
        5   Top variant (<b>not</b> <b>used</b>)
        6   The phenotype's strand

</pre><h4><b>OUTPUT</b> <b>FILE</b></h4><pre>
       <b>--out</b> file
        Space separated results output file detailing the enrichment with the following columns:
        1   The observed number of QTLs falling within the functional annotations
        2   The total number of QTLs
        3   The mean expected number  of  QTLs  falling  within  the  functional  annotations  (across  multiple
            permutations)
        4   The  standard  deviation  of  the  expected number of QTLs falling within the functional annotations
            (across multiple permutations)
        5   The empirical p-value
        6   Lower bound of the 95% confidence interval of the odds ratio
        7   The odds ratio
        8   Upper bound of the 95% confidence interval of the odds ratio

</pre><h4><b>EXAMPLE</b></h4><pre>
       1 You need to prepare a BED file containing the positions of the QTLs of interest.  To do so, extract all
         significant hits at a given FDR threshold (e.g. 5%), and then transform the significant QTL list into a
         BED file:

         Rscript ./script/qtltools_runFDR_cis.R results.genes.full.txt.gz 0.05 results.genes
         cat results.genes.significant.txt | awk '{ print $9, $10-1, $11, $8, $1, $5 }' | tr '  '  '\t'  |  sort
         -k1,1V -k2,2g &gt; results.genes.significant.bed

       2 Prepare a BED file containing the positions of all phenotypes you mapped QTLs for:

         zcat  results.genes.full.txt.gz  | awk '{ print $2, $3-1, $4, $1, $8, $5 }' | tr ' ' '\t' | sort -k1,1V
         -k2,2g &gt; results.genes.quantified.bed

       3 Run the enrichment analysis:

         QTLtools  fenrich  --qtl   results.genes.significant.bed   --tss   results.genes.quantified.bed   --bed
         TFs.encode.bed.gz --out enrichment.QTL.in.TF.txt

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/QTLtools.1.html">QTLtools</a></u>(1)

       QTLtools website: &lt;https://qtltools.github.io/qtltools&gt;

</pre><h4><b>BUGS</b></h4><pre>
       o Please submit bugs to &lt;https://github.com/qtltools/qtltools&gt;

</pre><h4><b>CITATION</b></h4><pre>
       Delaneau,  O.,  Ongen, H., Brown, A. et al. A complete tool set for molecular QTL discovery and analysis.
       <u>Nat</u> <u>Commun</u> <b>8</b>, 15452 (2017).  &lt;https://doi.org/10.1038/ncomms15452&gt;

</pre><h4><b>AUTHORS</b></h4><pre>
       Olivier Delaneau (<a href="mailto:olivier.delaneau@gmail.com">olivier.delaneau@gmail.com</a>), Halit Ongen (<a href="mailto:halitongen@gmail.com">halitongen@gmail.com</a>)

QTLtools-v1.3                                      06 May 2020                               <u><a href="../man1/QTLtools-fenrich.1.html">QTLtools-fenrich</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>