<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>vcf2tsv</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/libvcflib-tools">libvcflib-tools_1.0.12+dfsg-1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       <b>vcf2tsv</b>

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>vcf2tsv</b> [-n null_string] [-g] [vcf file]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Converts VCF to per-allelle or per-genotype tab-delimited format, using null string to replace empty val‚Äê
       ues  in  the table.  Specifying -g will output one line per sample with genotype information.  When there
       is more than one alt allele there will be multiple rows, one for each allele and, the info will match the
       `A' index

</pre><h4><b>OPTIONS</b></h4><pre>
              Type: transformation

</pre><h4><b>EXIT</b> <b>VALUES</b></h4><pre>
       <b>0</b>      Success

       <b>not</b> <b>0</b>  Failure

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <b><a href="../man1/vcflib.1.html">vcflib</a></b>(1)

</pre><h4><b>OTHER</b></h4><pre>
</pre><h4><b>LICENSE</b></h4><pre>
       Copyright 2011-2024 (C) Erik Garrison and vcflib contributors.  MIT licensed.   Copyright  2020-2024  (C)
       Pjotr Prins.

</pre><h4><b>AUTHORS</b></h4><pre>
       Erik Garrison and vcflib contributors.

vcf2tsv (vcflib)                                                                                      <u><a href="../man1/VCF2TSV.1.html">VCF2TSV</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>