<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::CodonTable - Codon table object</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::CodonTable - Codon table object

</pre><h4><b>SYNOPSIS</b></h4><pre>
         # This is a read-only class for all known codon tables.  The IDs are
         # the ones used by nucleotide sequence databases.  All common IUPAC
         # ambiguity codes for DNA, RNA and amino acids are recognized.

         use Bio::Tools::CodonTable;

         # defaults to ID 1 "Standard"
         $myCodonTable   = Bio::Tools::CodonTable-&gt;new();
         $myCodonTable2  = Bio::Tools::CodonTable-&gt;new( -id =&gt; 3 );

         # change codon table
         $myCodonTable-&gt;<a href="../man5/id.5.html">id</a>(5);

         # examine codon table
         print  join (' ', "The name of the codon table no.", $myCodonTable-&gt;<a href="../man4/id.4.html">id</a>(4),
                  "is:", $myCodonTable-&gt;name(), "\n");

         # print possible codon tables
         $tables = Bio::Tools::CodonTable-&gt;tables;
         while ( ($id,$name) = each %{$tables} ) {
           print "$id = $name\n";
         }

         # translate a codon
         $aa = $myCodonTable-&gt;translate('ACU');
         $aa = $myCodonTable-&gt;translate('act');
         $aa = $myCodonTable-&gt;translate('ytr');

         # reverse translate an amino acid
         @codons = $myCodonTable-&gt;revtranslate('A');
         @codons = $myCodonTable-&gt;revtranslate('Ser');
         @codons = $myCodonTable-&gt;revtranslate('Glx');
         @codons = $myCodonTable-&gt;revtranslate('cYS', 'rna');

         # reverse translate an entire amino acid sequence into a IUPAC
         # nucleotide string

         my $seqobj    = Bio::PrimarySeq-&gt;new(-seq =&gt; 'FHGERHEL');
         my $iupac_str = $myCodonTable-&gt;reverse_translate_all($seqobj);

         # boolean tests
         print "Is a start\n"       if $myCodonTable-&gt;is_start_codon('ATG');
         print "Is a terminator\n"  if $myCodonTable-&gt;is_ter_codon('tar');
         print "Is a unknown\n"     if $myCodonTable-&gt;is_unknown_codon('JTG');

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Codon tables are also called translation tables or genetic codes since that is what they represent. A bit
       more complete picture of the full complexity of codon usage in various taxonomic groups is presented at
       the NCBI Genetic Codes Home page.

       CodonTable is a BioPerl class that knows all current translation tables that are used by primary
       nucleotide sequence databases (GenBank, EMBL and DDBJ). It provides methods to output information about
       tables and relationships between codons and amino acids.

       This class and its methods recognized all common IUPAC ambiguity codes for DNA, RNA and animo acids. The
       translation method follows the conventions in EMBL and TREMBL databases.

       It is a nuisance to separate RNA and cDNA representations of nucleic acid transcripts. The CodonTable
       object accepts codons of both type as input and allows the user to set the mode for output when reverse
       translating. Its default for output is DNA.

       Note:

       This class deals primarily with individual codons and amino acids. However in the interest of speed you
       can translate longer sequence, too. The full complexity of protein translation is tackled by
       Bio::PrimarySeqI::translate.

       The amino acid codes are IUPAC recommendations for common amino acids:

                 A           Ala            Alanine
                 R           Arg            Arginine
                 N           Asn            Asparagine
                 D           Asp            Aspartic acid
                 C           Cys            Cysteine
                 Q           Gln            Glutamine
                 E           Glu            Glutamic acid
                 G           Gly            Glycine
                 H           His            Histidine
                 I           Ile            Isoleucine
                 L           Leu            Leucine
                 K           Lys            Lysine
                 M           Met            Methionine
                 F           Phe            Phenylalanine
                 P           Pro            Proline
                 O           Pyl            Pyrrolysine (22nd amino acid)
                 U           Sec            Selenocysteine (21st amino acid)
                 S           Ser            Serine
                 T           Thr            Threonine
                 W           Trp            Tryptophan
                 Y           Tyr            Tyrosine
                 V           Val            Valine
                 B           Asx            Aspartic acid or Asparagine
                 Z           Glx            Glutamine or Glutamic acid
                 J           Xle            Isoleucine or Valine (mass spec ambiguity)
                 X           Xaa            Any or unknown amino acid

       It is worth noting that, "Bacterial" codon table no. 11 produces an polypeptide that is, confusingly,
       identical to the standard one. The only differences are in available initiator codons.

       NCBI Genetic Codes home page:
            (Last update of the Genetic Codes: Nov. 18, 2016)
            https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=c

       ASN.1 version with ids 1 to 25 is at:
            <a href="ftp://ftp.ncbi.nih.gov/entrez/misc/data/gc.prt">ftp://ftp.ncbi.nih.gov/entrez/misc/data/gc.prt</a>

       Thanks to Matteo diTomasso for the original Perl implementation of these tables.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing lists  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Heikki</b> <b>Lehvaslaiho</b></h4><pre>
       Email:  heikki-at-bioperl-dot-org

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>id</b>
        Title   : id
        Usage   : $obj-&gt;<a href="../man3/id.3.html">id</a>(3); $id_integer = $obj-&gt;id();
        Function: Sets or returns the id of the translation table.  IDs are
                  integers from 0 (special ATG-only start) to 25, excluding
                  7-8 and 17-20 which have been removed. If an invalid ID is
                  given the method returns 1, the standard table.
        Example :
        Returns : value of id, a scalar, warn and fall back to 1 (standard table)
                  if specified id is not valid
        Args    : newvalue (optional)

   <b>name</b>
        Title   : name
        Usage   : $obj-&gt;name()
        Function: returns the descriptive name of the translation table
        Example :
        Returns : A string
        Args    : None

   <b>tables</b>
        Title   : tables
        Usage   : $obj-&gt;tables()  or  Bio::Tools::CodonTable-&gt;tables()
        Function: returns a hash reference where each key is a valid codon
                  table id() number, and each value is the corresponding
                  codon table name() string
        Example :
        Returns : A hashref
        Args    : None

   <b>translate</b>
        Title   : translate
        Usage   : $obj-&gt;translate('YTR')
        Function: Returns a string of one letter amino acid codes from
                  nucleotide sequence input. The imput can be of any length.

                  Returns 'X' for unknown codons and codons that code for
                  more than one amino acid. Returns an empty string if input
                  is not three characters long. Exceptions for these are:

                    - IUPAC amino acid code B for Aspartic Acid and
                      Asparagine, is used.
                    - IUPAC amino acid code Z for Glutamic Acid, Glutamine is
                      used.
                    - if the codon is two nucleotides long and if by adding
                      an a third character 'N', it codes for a single amino
                      acid (with exceptions above), return that, otherwise
                      return empty string.

                  Returns empty string for other input strings that are not
                  three characters long.

        Example :
        Returns : a string of one letter ambiguous IUPAC amino acid codes
        Args    : ambiguous IUPAC nucleotide string

   <b>translate_strict</b>
        Title   : translate_strict
        Usage   : $obj-&gt;translate_strict('ACT')
        Function: returns one letter amino acid code for a codon input

                  Fast and simple translation. User is responsible to resolve
                  ambiguous nucleotide codes before calling this
                  method. Returns 'X' for unknown codons and an empty string
                  for input strings that are not three characters long.

                  It is not recommended to use this method in a production
                  environment. Use method translate, instead.

        Example :
        Returns : A string
        Args    : a codon = a three nucleotide character string

   <b>revtranslate</b>
        Title   : revtranslate
        Usage   : $obj-&gt;revtranslate('G')
        Function: returns codons for an amino acid

                  Returns an empty string for unknown amino acid
                  codes. Ambiguous IUPAC codes Asx,B, (Asp,D; Asn,N) and
                  Glx,Z (Glu,E; Gln,Q) are resolved. Both single and three
                  letter amino acid codes are accepted. '*' and 'Ter' are
                  used for terminator.

                  By default, the output codons are shown in DNA.  If the
                  output is needed in RNA (tr/t/u/), add a second argument
                  'RNA'.

        Example : $obj-&gt;revtranslate('Gly', 'RNA')
        Returns : An array of three lower case letter strings i.e. codons
        Args    : amino acid, 'RNA'

   <b>reverse_translate_all</b>
        Title   : reverse_translate_all
        Usage   : my $iup_str = $cttable-&gt;reverse_translate_all($seq_object)
                  my $iup_str = $cttable-&gt;reverse_translate_all($seq_object,
                                                                $cutable,
                                                                15);
        Function: reverse translates a protein sequence into IUPAC nucleotide
                  sequence. An 'X' in the protein sequence is converted to 'NNN'
                  in the nucleotide sequence.
        Returns : a string
        Args    : a Bio::PrimarySeqI compatible object (mandatory)
                  a Bio::CodonUsage::Table object and a threshold if only
                    codons with a relative frequency above the threshold are
                    to be considered.

   <b>reverse_translate_best</b>
        Title   : reverse_translate_best
        Usage   : my $str = $cttable-&gt;reverse_translate_best($seq_object,$cutable);
        Function: Reverse translates a protein sequence into plain nucleotide
                  sequence (GATC), uses the most common codon for each amino acid
        Returns : A string
        Args    : A Bio::PrimarySeqI compatible object and a Bio::CodonUsage::Table object

   <b>is_start_codon</b>
        Title   : is_start_codon
        Usage   : $obj-&gt;is_start_codon('ATG')
        Function: returns true (1) for all codons that can be used as a
                  translation start, false (0) for others.
        Example : $myCodonTable-&gt;is_start_codon('ATG')
        Returns : boolean
        Args    : codon

   <b>is_ter_codon</b>
        Title   : is_ter_codon
        Usage   : $obj-&gt;is_ter_codon('GAA')
        Function: returns true (1) for all codons that can be used as a
                  translation tarminator, false (0) for others.
        Example : $myCodonTable-&gt;is_ter_codon('ATG')
        Returns : boolean
        Args    : codon

   <b>is_unknown_codon</b>
        Title   : is_unknown_codon
        Usage   : $obj-&gt;is_unknown_codon('GAJ')
        Function: returns false (0) for all codons that are valid,
               true (1) for others.
        Example : $myCodonTable-&gt;is_unknown_codon('NTG')
        Returns : boolean
        Args    : codon

   <b>unambiguous_codons</b>
        Title   : unambiguous_codons
        Usage   : @codons = $self-&gt;unambiguous_codons('ACN')
        Returns : array of strings (one-letter unambiguous amino acid codes)
        Args    : a codon = a three IUPAC nucleotide character string

   <b>_unambiquous_codons</b>
       deprecated, now an alias for unambiguous_codons

   <b>add_table</b>
        Title   : add_table
        Usage   : $newid = $ct-&gt;add_table($name, $table, $starts)
        Function: Add a custom Codon Table into the object.
                  Know what you are doing, only the length of
                  the argument strings is checked!
        Returns : the id of the new codon table
        Args    : name, a string, optional (can be empty)
                  table, a string of 64 characters
                  startcodons, a string of 64 characters, defaults to standard

perl v5.32.1                                       2021-08-15                        <u>Bio::Tools::<a href="../man3pm/CodonTable.3pm.html">CodonTable</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>