<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>phyloBoot - Generate simulated alignment data by parametric or nonparametric</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/phast">phast_1.7+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       phyloBoot - Generate simulated alignment data by parametric or nonparametric

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Generate simulated alignment data by parametric or nonparametric bootstrapping, and/or estimate errors in
       phylogenetic  model  parameters.  When estimating errors in parameters, the tree topology is not inferred
       <b>--</b> estimated errors are conditional on the given topology.

       If a model is given in the form  of  a  .mod  file  (&lt;model_fname&gt;),  then  parametric  bootstrapping  is
       performed  <b>--</b> i.e., synthetic data sets are drawn from the distribution defined by the model.  Otherwise,
       the input file is assumed to be a multiple alignment, and non-parametric bootstrapping  is  performed  <b>--</b>
       i.e., sites are drawn (with replacement) from the empirical distribution defined by the given alignment.

       The default behavior is to produce simulated alignments, estimate model parameters for each one, and then
       write  a  table  to  stdout  with  a  row for each parameter and columns for the mean, standard deviation
       (approximate standard error), median, minimum, and maximum of estimated values, plus  the  boundaries  of
       95%% and 90%% confidence intervals.

       The  <b>--alignments-only</b> option, however, allows the parameter estimation step to be bypassed entirely, and
       the program to be used simply to generate simulated data sets.  See usage  for  phyloFit  for  additional
       details on tree-building options.

</pre><h4><b>EXAMPLE</b></h4><pre>
       (See below for more details on options)

       1. Estimation of parameter errors by parametric bootstrapping.

              phyloBoot <b>--nreps</b> 500 <b>--nsites</b> 10000 mymodel.mod &gt; par_errors

       2. Estimation of parameter errors by nonparametric bootstrapping.

              phyloBoot  <b>--nreps</b>  500  <b>--nsites</b>  10000  <b>--tree</b>  "((human,chimp),(mouse,rat))"  myalignment.fa  &gt;
              nonpar_errors

       3. Parametric generation of simulated data.

              phyloBoot mymodel.mod <b>--alignments-only</b> pardata <b>--nreps</b> 500 <b>--nsites</b> 10000

       4. Nonparametric generation of simulated data.

              phyloBoot myalignment.fa <b>--alignments-only</b> nonpardata <b>--nreps</b> 500 <b>--nsites</b> 10000

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>bootstrapping</b> <b>options</b>

       <b>--nsites</b>, <b>-L</b> &lt;number&gt; Number of sites in sampled alignments.  If an alignment is

              given (non-parametric case), default is number of sites in alignment, otherwise default is 1000.

       <b>--nreps</b>, <b>-n</b> &lt;number&gt; Number of replicates.  Default is 100.

       <b>--msa-format</b>, <b>-i</b> FASTA|PHYLIP|MPM|MAF|SS

       (non-parametric case only)
              Alignment format.  Default is to guess format from file contents.

       <b>--alignments-only</b>, <b>-a</b> &lt;fname_root&gt; Generate alignments and write them to files with given filename  root,
              but do not estimate parameters.

       <b>--dump-mods</b>, <b>-d</b> &lt;fname_root&gt;

              Dump .mod files for individual estimated models (one for each replicate).

       <b>--dump-samples</b>, <b>-m</b> &lt;fname_root&gt;

              Dump  simulated  alignments  to  files with given filename root.  Similar to <b>--alignments-only</b> but
              does not disable parameter estimation.

       <b>--dump-format</b>, <b>-o</b> FASTA|PHYLIP|MPM|SS.

              (For use with <b>--alignments-only</b> or <b>--dump-samples</b>) File format to use when dumping raw alignments.
              Default FASTA.

       <b>--read-mods</b>, <b>-R</b> &lt;fname_list&gt;  Read  estimated  models  from  list  of  filenames  instead  of  generating
              alignments  and  estimating  parameters.   fname_list  can be commadelimited list of files, or, if
              preceded by a '*', the name of a file containing the file names (one per line).  Can  be  used  to
              compute  statistics  for  replicates  that have been processed separately (see <b>--alignments-only</b>).
              When this option is used, the primary argument to the program (&lt;model_fname&gt;|&lt;msa_fname&gt;) will  be
              ignored.

       <b>--output-average</b>,  <b>-A</b>  &lt;fname&gt;  Output  a  tree model representing the average of all input models to the
              specified file.

       <b>--quiet</b>, <b>-q</b>

              Proceed quietly.

       <b>--help</b>, <b>-h</b> Print this help message.

   <b>tree-building</b> <b>options</b>

       <b>--tree</b>, <b>-t</b> &lt;tree_fname&gt;|&lt;tree_string&gt; (Required if non-parametric and more than two species) Name of file
              or literal string defining tree topology.

       <b>--subst-mod</b>, <b>-s</b> JC69|F81|HKY85|REV|SSREV|UNREST|R2|R2S|U2|U2S|R3|R3S|U3|U3S

       (default REV).
              Nucleotide substitution model.

       <b>--nrates</b>, <b>-k</b> &lt;nratecats&gt; (default 1).  Number of rate categories to use.  Specifying a

              value of greater than one causes the discrete gamma model for rate variation to be used.

       <b>--EM</b>, <b>-E</b> Use EM rather than the BFGS quasi-Newton algorithm for parameter estimation.

       <b>--precision</b>, <b>-p</b> HIGH|MED|LOW

              (default HIGH) Level of precision to use in estimating model parameters.

       <b>--init-model</b>, <b>-M</b> &lt;mod_fname&gt;

              Initialize optimization procedure with specified tree model.

       <b>--init-random</b>, <b>-r</b>

              Initialize parameters randomly.

       <b>--scale</b>,-P &lt;rho&gt; Scale input tree by factor rho before doing parametric simulations.

       <b>--subtree</b>,-S &lt;node&gt; For use with <b>--subtree-scale</b> and/or subtree-switch.  Define

              subtree including all children of named node, including branch leading up to node.

       <b>--subtree-scale</b>,-l &lt;lambda&gt; Scale subtree defined with <b>--subtree</b> option by factor lambda.

       <b>--subtree-switch</b>,-w &lt;prob&gt;

              With given probability, randomly switch branches in tree from subtree to supertree and vice versa.
              Randomization is performed independently for each branch in every column of simulated data.

       <b>--scale-file</b>,-F &lt;file&gt; (For use with <b>--subtree</b> in parametric mode).  Instead of using

       <b>--subtree-scale</b> or <b>--scale</b>, read in a tab-delimited file with three columns: numSite,scale,subtree_scale.
              For each row in the file phyloBoot will simulate the given number  of  sites  with  those  scaling
              factors,  and  then  will move on to the next row, so that the total number of sites is the sum of
              the first column.

phyloBoot 1.4                                       May 2016                                        <u><a href="../man1/PHYLOBOOT.1.html">PHYLOBOOT</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>