<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>fastANI - Fast alignment-free computation of whole-genome Average Nucleotide Identity</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/fastani">fastani_1.33-3build1_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       fastANI - Fast alignment-free computation of whole-genome Average Nucleotide Identity

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>-----------------</b>  fastANI  is  a  fast  alignment-free implementation for computing whole-genome Average
       Nucleotide Identity (ANI) between genomes <b>-----------------</b> Example usage: $  fastANI  <b>-q</b>  genome1.fa  <b>-r</b>
       genome2.fa <b>-o</b> output.txt $ fastANI <b>-q</b> genome1.fa <b>--rl</b> genome_list.txt <b>-o</b> output.txt

       Available options <b>-----------------</b> <b>-h</b>, <b>--help</b>

              Print this help page

       <b>-r</b> &lt;value&gt;, <b>--ref</b> &lt;value&gt;

              reference genome (fasta/fastq)[.gz]

       <b>--refList</b> &lt;value&gt;, <b>--rl</b> &lt;value&gt;

              a file containing list of reference genome files, one genome per line

       <b>-q</b> &lt;value&gt;, <b>--query</b> &lt;value&gt;

              query genome (fasta/fastq)[.gz]

       <b>--ql</b> &lt;value&gt;, <b>--queryList</b> &lt;value&gt;

              a file containing list of query genome files, one genome per line

       <b>-k</b> &lt;value&gt;, <b>--kmer</b> &lt;value&gt;

              kmer size &lt;= 16 [default : 16]

       <b>-t</b> &lt;value&gt;, <b>--threads</b> &lt;value&gt;

              thread count for parallel execution [default : 1]

       <b>--fragLen</b> &lt;value&gt;

              fragment length [default : 3,000]

       <b>--minFraction</b> &lt;value&gt;

              minimum  fraction  of  genome  that must be shared for trusting ANI. If reference and query genome
              size differ, smaller one among the two is considered. [default : 0.2]

       <b>--visualize</b>

              output mappings for visualization, can be enabled for single genome to  single  genome  comparison
              only [disabled by default]

       <b>--matrix</b>

              also  output  ANI values as lower triangular matrix (format inspired from phylip). If enabled, you
              should expect an output file with .matrix extension [disabled by default]

       <b>-o</b> &lt;value&gt;, <b>--output</b> &lt;value&gt; [required]

              output file name

       <b>-v</b>, <b>--version</b>

              Show version

</pre><h4><b>AUTHOR</b></h4><pre>
        This manpage was written by Nilesh Patra for the Debian distribution and
        can be used for any other usage of the program.

fastANI 1.32                                       March 2021                                         <u><a href="../man1/FASTANI.1.html">FASTANI</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>