<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>gsnapl - Large Genome Short-read Nucleotide Alignment Program</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/gmap">gmap_2024-11-20+ds-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       gsnapl - Large Genome Short-read Nucleotide Alignment Program

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>gsnap</b> [<u>OPTIONS</u>...] <u>&lt;FASTA</u> <u>file&gt;,</u> <u>or</u> cat &lt;FASTA file&gt; | gmap [OPTIONS...]

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Input</b> <b>options</b> <b>(must</b> <b>include</b> <b>-d</b>)
       <b>-D</b>, <b>--dir</b>=<u>directory</u>
              Genome   directory.   Default  (as  specified  by  <b>--with-gmapdb</b>  to  the  configure  program)  is
              <u>/var/cache/gmap</u>

       <b>-d</b>, <b>--db</b>=<u>STRING</u>
              Genome database

       <b>--two-pass</b>
              Two-pass mode, in which the sequences are processed first to identify splice  sites  and  introns,
              and then aligned using this splicing information

       <b>--use-localdb</b>=<u>INT</u>
              Whether  to  use  the  local  suffix  arrays,  which  help  with finding extensions to the ends of
              alignments in the presence of splicing or indels (0=no, 1=yes if available (default))

       Transcriptome-guided options (optional)

       <b>-C</b>, <b>--transcriptdir</b>=<u>directory</u>
              Transcriptome directory.  Default is the value for <b>--dir</b> above

       <b>-c</b>, <b>--transcriptdb</b>=<u>STRING</u>
              Transcriptome database

       <b>--transcriptome-mode</b>=<u>STRING</u>
              Options: assist, only, annotate (default).  The option assist means to try transcriptome alignment
              first, but then use genomic alignment  if  nothing  is  found.   The  option  only  means  to  try
              transcriptome alignment only.  The option annotate means to try only genomic alignment, to use the
              transcriptome  only  for  annotation;  this  is  the  fastest  option.   In the other two options,
              annotation is also performed

       Computation options

       <b>-k</b>, <b>--kmer</b>=<u>INT</u>
              kmer size to use in genome database (allowed values: 16 or less) If  not  specified,  the  program
              will find the highest available kmer size in the genome database

       <b>--sampling</b>=<u>INT</u>
              Sampling  to  use  in  genome  database.   If  not  specified,  the program will find the smallest
              available sampling value in the genome database within selected k-mer size

       <b>--align-fraction</b>=<u>FLOAT</u>
              Process only the given fraction of reads, selected at random If <b>--align-fraction</b>  and  <b>--part</b>  are
              given, <b>--align-fraction</b> takes precedence

       <b>-q</b>, <b>--part</b>=<u>INT</u>/INT
              Process only the i-th out of every n sequences e.g., 0/100 or 99/100 (useful for distributing jobs
              to a computer farm).

       <b>--input-buffer-size</b>=<u>INT</u>
              Size of input buffer (program reads this many sequences at a time for efficiency) (default 10000)

       <b>--barcode-length</b>=<u>INT</u>
              Amount of barcode to remove from start of every read before alignment (default 0)

       <b>--endtrim-length</b>=<u>INT</u>
              Amount of trim to remove from the end of every read before alignment (default 0)

       <b>--orientation</b>=<u>STRING</u>
              Orientation  of  paired-end  reads  Allowed values: FR (fwd-rev, or typical Illumina; default), RF
              (rev-fwd, for circularized inserts), or FF (fwd-fwd, same strand), or 10X (single-cell where  read
              1 has barcode information; read 2 is rev)

       <b>--10x-whitelist</b>=<u>FILE</u>
              Whitelist  of  10X  Genomics GEM bead barcodes, needed to perform correction of cellular barcodes.
              This file can be obtained  at  cellranger-x.y.z/lib/python/cellranger/barcodes  (for  Cell  Ranger
              version &gt;= 4) cellranger-x.y.z/lib/cellranger-cs/x.y.z/lib/python/cellranger/barcodes (&lt;= 3)

       <b>--10x-well-position</b>=<u>INT</u>
              Position  of well information in the accession, when separated by colons If set to 0, then no well
              information will be printed in the CB field (default: 4)

       <b>--fastq-id-start</b>=<u>INT</u>
              Starting position of identifier in FASTQ header, space-delimited (&gt;= 1)

       <b>--fastq-id-end</b>=<u>INT</u>
              Ending position of identifier in FASTQ header, space-delimited (&gt;= 1)

       Examples:

       @HWUSI-EAS100R:6:73:941:1973#0/1
              start=1, end=1 (default) =&gt; identifier is HWUSI-EAS100R:6:73:941:1973#0

       @SRR001666.1 071112_SLXA-EAS1_s_7:5:1:817:345 length=36
              start=1,   end=1    =&gt;   identifier   is   SRR001666.1   start=2,   end=2    =&gt;   identifier    is
              071112_SLXA-EAS1_s_7:5:1:817:345     start=1,     end=2      =&gt;    identifier    is    SRR001666.1
              071112_SLXA-EAS1_s_7:5:1:817:345

       <b>--force-single-end</b>
              When multiple FASTQ files are provided on the  command  line,  GSNAP  assumes  they  are  matching
              paired-end files.  This flag treats each file as single-end.

       <b>--filter-chastity</b>=<u>STRING</u>
              Skips  reads  marked  by  the  Illumina  chastity program.  Expecting a string after the accession
              having a 'Y' after the first colon, like this:

       @accession 1:Y:0:CTTGTA
              where the 'Y' signifies  filtering  by  chastity.   Values:  off  (default),  either,  both.   For
              'either',  a  'Y'  on  either  end  of  a  paired-end read will be filtered.  For 'both', a 'Y' is
              required on both ends of a paired-end read (or on the only end of a single-end read).

       <b>--allow-pe-name-mismatch</b>
              Allows accession names of reads to mismatch in paired-end files

       <b>--interleaved</b>
              Input is in interleaved format (one read per line, tab-delimited

       <b>--gunzip</b>
              Uncompress gzipped input files

       <b>--bunzip2</b>
              Uncompress bzip2-compressed input files

       Computation options

       <b>-B</b>, <b>--batch</b>=<u>INT</u>
              Batch mode (default = 5) Mode  Hash offsets  Hash positions  Genome           Local  hash  offsets
              Local hash positions  Localdb

       0      allocate      mmap            mmap            allocate            mmap                  mmap

       1      allocate      mmap &amp; preload  mmap            allocate            mmap &amp; preload        mmap

       2      allocate      mmap &amp; preload  mmap &amp; preload  allocate            mmap &amp; preload        mmap

       3      allocate      allocate        mmap &amp; preload  allocate            allocate              mmap

       4      allocate      allocate        allocate        allocate            allocate              mmap

       (default)
              5       allocate          allocate            allocate            allocate                allocate
              allocate

       Note: For a single sequence, all data structures use mmap
              A batch level of 5 means the same as 4, and is kept only for backward compatibility

       <b>--use-shared-memory</b>=<u>INT</u>
              If 1, then allocated memory is shared among all processes on this node If 0 (default),  then  each
              process has private allocated memory

       <b>--preload-shared-memory</b>
              Load  files  indicated by <b>--batch</b> mode into shared memory for use by other GMAP/GSNAP processes on
              this node, and then exit.  Ignore any input files.

       <b>--unload-shared-memory</b>
              Unload files indicated by <b>--batch</b> mode into shared memory, or  allow  them  to  be  unloaded  when
              existing GMAP/GSNAP processes on this node are finished with them.  Ignore any input files.

       <b>-m</b>, <b>--max-mismatches</b>=<u>FLOAT</u>
              Maximum number of mismatches allowed (if not specified, then GSNAP tries to find the best possible
              match  in  the  genome)  If specified between 0.0 and 1.0, then treated as a fraction of each read
              length.  Otherwise, treated as an integral number of  mismatches  (including  indel  and  splicing
              penalties).  Default is 0.3

       <b>--query-unk-mismatch</b>=<u>INT</u>
              Whether to count unknown (N) characters in the query as a mismatch (0=no (default), 1=yes)

       <b>--genome-unk-mismatch</b>=<u>INT</u>
              Whether  to count unknown (N) characters in the genome as a mismatch (0=no, 1=yes).  If <b>--use-mask</b>
              is specified, default is no, otherwise yes.

       <b>--maxsearch</b>=<u>INT</u>
              Maximum number of alignments to find (default 1000).  Should be larger than <b>--npaths</b>, which is the
              number to report.  Keeping this number large  will  allow  for  random  selection  among  multiple
              alignments.  Reducing this number can speed up the program.

       <b>--indel-endlength</b>=<u>INT</u>
              Minimum length at end required for indel alignments (default 4)

       <b>--max-insertions</b>=<u>INT</u>
              Maximum number of insertions allowed (default 9)

       <b>--max-deletions</b>=<u>INT</u>
              Maximum number of deletions allowed (default 15)

       <b>-M</b>, <b>--suboptimal-levels</b>=<u>INT</u>
              Report suboptimal hits beyond best hit (default 0) All hits with best score plus suboptimal-levels
              are reported (Note: Not currently implemented)

       <b>-a</b>, <b>--adapter-strip</b>=<u>STRING</u>
              Method  for  removing  adapters  from  reads.   Currently allowed values: off, paired.  Default is
              "off".  To turn on, specify "paired", which removes adapters from paired-end reads if they  appear
              to be present.

       <b>-e</b>, <b>--use-mask</b>=<u>STRING</u>
              Use genome containing masks (e.g. for non-exons) for scoring preference

       <b>-V</b>, <b>--snpsdir</b>=<u>STRING</u>
              Directory for SNPs index files (created using snpindex) (default is location of genome index files
              specified using <b>-D</b> and <b>-d</b>)

       <b>-v</b>, <b>--use-snps</b>=<u>STRING</u>
              Use  database  containing  known  SNPs  (in  &lt;STRING&gt;.iit,  built  previously  using snpindex) for
              tolerance to SNPs

       <b>--cmetdir</b>=<u>STRING</u>
              Directory for methylcytosine index files (created using cmetindex) (default is location of  genome
              index files specified using <b>-D</b>, <b>-V</b>, and <b>-d</b>)

       <b>--atoidir</b>=<u>STRING</u>
              Directory  for  A-to-I  RNA  editing index files (created using atoiindex) (default is location of
              genome index files specified using <b>-D</b>, <b>-V</b>, and <b>-d</b>)

       <b>--mode</b>=<u>STRING</u>
              Alignment   mode:   standard   (default),    cmet-stranded,    cmet-nonstranded,    atoi-stranded,
              atoi-nonstranded,  ttoc-stranded,  or  ttoc-nonstranded.   Non-standard modes requires you to have
              previously run the cmetindex or atoiindex programs (which also cover the ttoc modes) on the genome

       <b>-t</b>, <b>--nthreads</b>=<u>INT</u>
              Number of worker threads

       Splicing options for DNA-Seq

       <b>--find-dna-chimeras</b>=<u>INT</u>
              Look for distant splicing involving poor splice sites (0=no, 1=yes) If not specified, then default
              is to be on unless only known splicing is desired (<b>--use-splicing</b> is specified and <b>--novelsplicing</b>
              is off)

       Splicing options for RNA-Seq

       <b>-N</b>, <b>--novelsplicing</b>=<u>INT</u>
              Look for novel splicing (0=no (default), 1=yes)

       <b>--splicingdir</b>=<u>STRING</u>
              Directory for splicing involving known  sites  or  known  introns,  as  specified  by  the  <b>-s</b>  or
              <b>--use-splicing</b>  flag  (default  is  directory computed from <b>-D</b> and <b>-d</b> flags).  Note: can just give
              full pathname to the <b>-s</b> flag instead.

       <b>-s</b>, <b>--use-splicing</b>=<u>STRING</u>
              Look for splicing involving known sites or known introns  (in  &lt;STRING&gt;.iit),  at  short  or  long
              distances See README instructions for the distinction between known sites and known introns

       <b>--splices-noeval</b>
              Do not evaluate splices for probability or intron length, but depend only on sequence alignment

       <b>--splices-dump</b>=<u>FILE</u>
              Write  splice  junction  information  to  FILE,  in  the  same  format  as  for  STAR  plus MaxEnt
              probabilities for the two intron positions.  Note that in this dump file, the annotation column is
              reserved strictly for known introns, and not novel introns that passed some criterion from a first
              pass.

       <b>--splices-include-knownp</b>
              In the file for <b>--splices-dump</b>, include all known introns

       <b>--splices-read</b>=<u>FILE</u>
              Read allowable splices from FILE, in the same format as for STAR.  This is useful if some external
              program can evaluate and filter the results from <b>--splices-dump</b> in a  first  alignment  pass,  and
              then GSNAP can use the filtered splices in a second alignment pass

       <b>-w</b>, <b>--localsplicedist</b>=<u>INT</u>
              Definition of local novel splicing event (default 200000)

       <b>--merge-distant-samechr</b>
              Report  distant  splices  on  the same chromosome as a single splice, if possible.  Will produce a
              single SAM line instead of two SAM lines, which is also done for translocations,  inversions,  and
              scramble events

       Options for paired-end reads

       <b>--pairmax-dna</b>=<u>INT</u>
              Max total genomic length for DNA-Seq paired reads, or other reads without splicing (default 2000).
              Used if <b>-N</b> or <b>-s</b> is not specified.  This value is also used for circular chromosomes when splicing
              in linear chromosomes is allowed

       <b>--pairmax-rna</b>=<u>INT</u>
              Max  total  genomic  length  for  RNA-Seq  paired  reads,  or other reads that could have a splice
              (default 200000).  Used if <b>-N</b> or <b>-s</b> is  specified.   Should  probably  match  the  value  for  <b>-w</b>,
              <b>--localsplicedist</b>.

       <b>--resolve-inner</b>=<u>INT</u>
              Whether to resolve soft-clipping on the insides of paired-end reads (default 1)

       <b>--pairexpect</b>=<u>INT</u>
              Expected  paired-end  length, used for resolving soft-clipping on the insides of paired-end reads,
              and for pairing DNA-seq reads (default 200)

       <b>--pairdev</b>=<u>INT</u>
              Allowable deviation from expected paired-end length,  used  for  resolving  soft-clipping  on  the
              insides of paired-end reads (default 100).

       <b>--pass1-min-support</b>=<u>INT</u>
              Threshold read support for learning an intron during pass 1 of <b>--two-pass</b> mode (default 20)

       Options for quality scores

       <b>--quality-protocol</b>=<u>STRING</u>
              Protocol  for  input quality scores.  Allowed values: illumina (ASCII 64-126) (equivalent to <b>-J</b> 64
              <b>-j</b> <b>-31</b>) sanger   (ASCII 33-126) (equivalent to <b>-J</b> 33 <b>-j</b> 0)

       Default is sanger (no quality print shift)
              SAM output files should have quality scores in sanger protocol

              Or you can customize this behavior with these flags:

       <b>-J</b>, <b>--quality-zero-score</b>=<u>INT</u>
              FASTQ quality scores are zero at this  ASCII  value  (default  is  33  for  sanger  protocol;  for
              Illumina, select 64)

       <b>-j</b>, <b>--quality-print-shift</b>=<u>INT</u>
              Shift  FASTQ  quality scores by this amount in output (default is 0 for sanger protocol; to change
              Illumina input to Sanger output, select <b>-31</b>)

       Output options

       <b>-n</b>, <b>--npaths</b>=<u>INT</u>
              Maximum number of paths to print (default 100).

       <b>-Q</b>, <b>--quiet-if-excessive</b>
              If more than maximum number of paths are found, then nothing is printed.

       <b>-O</b>, <b>--ordered</b>
              Print output in same order as input (relevant only if there is more than one worker thread)

       <b>--show-refdiff</b>
              For GSNAP output in SNP-tolerant alignment, shows all differences relative to the reference genome
              as lower case (otherwise, it shows all differences relative to both the  reference  and  alternate
              genome)

       <b>--clip-overlap</b>
              For paired-end reads whose alignments overlap, clip the overlapping region.

       <b>--merge-overlap</b>
              For  paired-end  reads  whose  alignments  overlap,  merge  the  two  ends into a single end (beta
              implementation)

       <b>--print-snps</b>
              Print detailed information about SNPs in reads  (works  only  if  <b>-v</b>  also  selected)  (not  fully
              implemented yet)

       <b>--failsonly</b>
              Print only failed alignments, those with no results

       <b>--nofails</b>
              Exclude printing of failed alignments

       <b>--only-concordant</b>
              Print only concordant alignments (concordant_uniq, concordant_mult, concordant_circular)

       <b>--omit-concordant-uniq</b>
              Do not print any concordant_uniq alignments

       <b>--omit-concordant-mult</b>
              Do not print any concordant_mult alignments

       <b>--omit-softclipped</b>
              Do not allow any alignments with soft clips

       <b>--only-tr-consistent</b>
              Print only alignments with consistent transcripts (XX field present, identical if paired-end)

       <b>-A</b>, <b>--format</b>=<u>STRING</u>
              Another format type, other than default.  Currently implemented: sam, m8 (BLAST tabular format)

       <b>--split-output</b>=<u>STRING</u>
              Basename  for  multiple-file output, separately for nomapping, halfmapping_uniq, halfmapping_mult,
              unpaired_uniq,  unpaired_mult,  paired_uniq,  paired_mult,  concordant_uniq,  and  concordant_mult
              results

       <b>-o</b>, <b>--output-file</b>=<u>STRING</u>
              File name for a single stream of output results.

       <b>--failed-input</b>=<u>STRING</u>
              Print completely failed alignments as input FASTA or FASTQ format, to the given file, appending .1
              or  .2,  for paired-end data.  If the <b>--split-output</b> flag is also given, this file is generated in
              addition to the output in the .nomapping file.

       <b>--append-output</b>
              When <b>--split-output</b> or <b>--failed-input</b> is given, this flag  will  append  output  to  the  existing
              files.  Otherwise, the default is to create new files.

       <b>--order-among-best</b>=<u>STRING</u>
              Among  alignments tied with the best score, order those alignments in this order.  Allowed values:
              genomic, random (default)

       <b>--output-buffer-size</b>=<u>INT</u>
              Buffer size, in queries, for output thread (default 1000).  When  the  number  of  results  to  be
              printed exceeds this size, worker threads wait until the backlog is cleared

       Options for SAM output

       <b>--no-sam-headers</b>
              Do not print headers beginning with '@'

       <b>--add-paired-nomappers</b>
              Add nomapper lines as needed to make all paired-end results alternate between first end and second
              end

       <b>--paired-flag-means-concordant</b>=<u>INT</u>
              Whether  the  paired  bit in the SAM flags means concordant only (1) or paired plus concordant (0,
              default)

       <b>--sam-headers-batch</b>=<u>INT</u>
              Print headers only for this batch, as specified by <b>-q</b>

       <b>--sam-hardclip-use-S</b>
              Use S instead of H for hardclips

       <b>--sam-use-0M</b>=<u>INT</u>
              If 1 (default), then insert 0M in CIGAR between adjacent indels and introns If 0, do not allow 0M.
              Picard disallows 0M, but other tools may require it

       <b>--sam-extended-cigar</b>
              Use extended CIGAR format (using X and = symbols instead of M, to indicate matches and mismatches,
              respectively

       <b>--sam-multiple-primaries</b>
              Allows multiple alignments to be marked as primary if they have equally good mapping scores

       <b>--sam-sparse-secondaries</b>
              For secondary alignments (in multiple mappings), uses '*' for SEQ and QUAL fields, to give smaller
              file sizes.  However, the output will give warnings in Picard to give warnings and  may  not  work
              with downstream tools

       <b>--force-xs-dir</b>
              For  RNA-Seq  alignments,  disallows XS:A:? when the sense direction is unclear, and replaces this
              value arbitrarily with XS:A:+.  May be useful for some programs, such as  Cufflinks,  that  cannot
              handle  XS:A:?.   However,  if you use this flag, the reported value of XS:A:+ in these cases will
              not be meaningful.

       <b>--md-report-snps</b>
              In MD string, when known SNPs are given by the <b>-v</b> flag, prints difference  nucleotides  when  they
              differ from reference but match a known alternate allele

       <b>--no-soft-clips</b>
              Does not allow soft clips at ends.  Mismatches will be counted over the entire query

       <b>--extend-soft-clips</b>
              Extends  alignments  through  soft clipped regions.  CIGAR string and coordinates will be revised,
              but mismatches and the MD string will reflect the clipped CIGAR

       <b>--action-if-cigar-error</b>
              Action to take if there is a disagreement between CIGAR length and sequence length Allowed values:
              ignore, warning (default), noprint, abort Note that the noprint option does not  print  the  CIGAR
              string at all if there is an error, so it may break a SAM parser

       <b>--read-group-id</b>=<u>STRING</u>
              Value to put into read-group id (RG-ID) field

       <b>--read-group-name</b>=<u>STRING</u>
              Value to put into read-group name (RG-SM) field

       <b>--read-group-library</b>=<u>STRING</u>
              Value to put into read-group library (RG-LB) field

       <b>--read-group-platform</b>=<u>STRING</u>
              Value to put into read-group library (RG-PL) field

       Help options

       <b>--check</b>
              Check compiler assumptions

       <b>--version</b>
              Show version

       <b>--help</b> Show this help message

       Other tools of GMAP suite are located in /usr/lib/gmap

gsnapl 2024-11-20+ds-2                            February 2025                                        <u><a href="../man1/GSNAPL.1.html">GSNAPL</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>