<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>snap-aligner_single - scalable nucleotide alignment program</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/snap-aligner">snap-aligner_2.0.3+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       snap-aligner_single - scalable nucleotide alignment program

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Welcome to SNAP version 2.0.3.

       Too  few  parameters  Usage:  snap-aligner  single  &lt;index-dir&gt;  &lt;inputFile(s)&gt;  [&lt;options&gt;] where &lt;input
       file(s)&gt; is a list of files to process.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-o</b>     filename  output alignments to filename in SAM or BAM format, depending on the file  extension  or
              explicit  type  specifier  (see  below).   Use  a dash with an explicit type specifier to write to
              stdout, so for example <b>-o</b> <b>-sam</b> - would write SAM output to stdout

       <b>-d</b>     maximum edit distance allowed per read or pair absent indels (default: 27)

       <b>-i</b>     maximum distance allowed per read for indels (default: 40)

       <b>-n</b>     number of seeds to use per read

       <b>-sc</b>    Seed coverage (i.e., readSize/seedSize).  Floating point.  Exclusive with <b>-n</b>.  (default uses <b>-n</b>)

       <b>-h</b>     maximum hits to consider per seed (default: 300)

       <b>-ms</b>    minimum seed matches per location (default: 1)

       <b>-t</b>     number of threads (default is one per core)

       <b>-b-</b>    Don't bind each thread to its processor (<b>--b</b> (with two dashes) does the smae thing)

       <b>-P</b>     disables cache prefetching in the genome; may be helpful for machines with small caches or lots of
              cores/cache

       <b>-so</b>    sort output file by alignment location

       <b>-sm</b>    memory to use for sorting in Gbytes.  Default is 1 Gbyte/thread.

       <b>-sid</b>   Specifies the sort intermediate directory.  When SNAP is sorting, it aligns the reads in the order
              in which they come in, and writes the aligned reads in batches to  a  temporary  file.   When  the
              aligning  is  done,  it  does a merge sort from the temporary file into the final output file.  By
              default, the intermediate file is in the same directory as the output file, but for performance or
              space reasons, you might want to put it elsewhere.  If so, use this option.

       <b>-x</b>     explore some hits of overly popular seeds (useful for filtering)

       <b>-S</b>     suppress additional processing (sorted BAM output only) i=index, d=duplicate marking

       <b>-f</b>     stop on first match within edit distance limit (filtering mode)

       <b>-F</b>     filter output (a=aligned only, s=single hit only (MAPQ &gt;= 10), u=unaligned only, l=long enough  to
              align (see <b>-mrl</b>))

       <b>-E</b>     an  alternate (and fully general) way to specify filter options.  Emit only these types s = single
              hit (MAPQ &gt;= 10), m = multiple hit (MAPQ &lt; 10), x = not long enough to align, u = unaligned,  b  =
              filter must apply to both ends of a paired-end read.  Combine the letters after <b>-E</b>, so for example
              <b>-E</b>  smu  will emit all reads that aren't too short/have too many Ns (because it leaves off l).  <b>-E</b>
              smx is the same as <b>-F</b> a, <b>-E</b> ux is the same as <b>-F</b> u, and so forth.  When  filtering  in  paired-end
              mode (either with <b>-F</b> or <b>-E</b>) unless you specify the b flag a read will be emitted if it's mate pair
              passes  the  filter  Even if the read itself does not.  If you specify b mode, then a read will be
              emitted only if it and its partner both pass the filter.

       <b>-I</b>     ignore IDs that don't match in the paired-end aligner

       <b>-Cxx</b>   must be followed by two + or - symbols saying whether to clip low-quality  bases  from  front  and
              back of read respectively; default: back only (<b>-C-</b>+)

       <b>-cc</b>    Specifies  the  min  and  max  quality  score to clip in Phred 33 format.  Must be followed by two
              characters.  The default is ##.

       <b>-=</b>     use the new style CIGAR strings with = and X rather than M.  The opposite of <b>-M</b>

       <b>-pf</b>    specify the name of a file to contain the run speed

       <b>-hp</b>    Indicates to use huge pages (this may speed up alignment and slow down index load).

       <b>-D</b>     Specifies the extra search depth (the edit distance beyond the best hit that SNAP uses to  compute
              MAPQ).  Default 1

       <b>-rg</b>    Specify the default read group if it is not specified in the input file

       <b>-R</b>     Specify  the  entire  read group line for the SAM/BAM output.  This must include an ID tag.  If it
              doesn't start with '@RG' SNAP will add that.  Specify tabs by \t.  Two backslashes will generate a
              single backslash.  backslash followed by anything else is  illegal.   So,  '-R  @RG\tID:foo\tDS:my
              data' would generate reads with default tag foo, and an @RG line that also included the DS:my data
              field.

       <b>-sa</b>    Include  reads from SAM or BAM files with the secondary (0x100) or supplementary (0x800) flag set;
              default is to drop them.

       <b>-om</b>    Output multiple alignments.  Takes as a parameter the maximum extra edit distance relative to  the
              best alignment to allow for secondary alignments

       <b>-omax</b> Limit the number of alignments per read generated by <b>-om</b>.
              This means that if <b>-om</b> would generate more

              than  <b>-omax</b>  secondary  alignments,  SNAP will write out only the best <b>-omax</b> of them, where 'best'
              means 'with the lowest edit distance'.  Ties are broken arbitrarily.

       <b>-mpc</b>   Limit the number of alignments generated by <b>-om</b> to this many per contig (chromosome/FASTA  entry);
              'mpc'  means  'max  per  contig;  default  unlimited.  This filter is applied prior to <b>-omax</b>.  The
              primary alignment is counted.

       <b>-pc</b>    Preserve the soft clipping for reads coming from SAM or BAM files

       <b>-xf</b>    Increase expansion factor for BAM and GZ files (default 1.0)

       <b>-hdp</b>   Use Hadoop-style prefixes (reporter:status:...) on error messages, and emit hadoop-style  progress
              messages

       <b>-mrl</b>   Specify the minimum read length to align, reads shorter than this (after clipping) stay unaligned.
              This  should  be  a  good  bit  bigger  than  the  seed  length or you might get some questionable
              alignments.  Default 50

       <b>-map</b>   Use file mapping to load the index rather than reading it.  This might speed up index  loading  in
              cases  where  SNAP  is  run  repatedly on the same index, and the index is larger than half of the
              memory size of the machine.  On some operating systems, loading an index with <b>-map</b> is much  slower
              than  without if the index is not in memory.  You might consider adding <b>-pre</b> to prefetch the index
              into system cache when loading with <b>-map</b> when you don't expect the index to be in cache.  This  is
              the default

       <b>-map-</b> Do not map the index, read it using standard read/write calls.

       <b>-pre</b>   Prefetch  the  index  into system cache.  This is only meaningful with <b>-map</b>, and only helps if the
              index is not already in memory and your operating system is slow at reading  mapped  files  (i.e.,
              some versions of Linux, but not Windows).  This is the default on Linux.

       <b>-pre-</b> Do not prefetch the index into system cache.
              This is the default on Windows.

       <b>-lp</b>    Run SNAP at low scheduling priority (Only implemented on Windows)

       <b>-nu</b>    No Ukkonen: don't reduce edit distance search based on prior candidates. This option is purely for
              evaluating  the  performance  effect  of using Ukkonen's algorithm rather than Smith-Waterman, and
              specifying it will slow down execution without improving the alignments.

       <b>-no</b>    No Ordering: don't order the evalutation of potential alignments  so  as  to  select  more  likely
              candidates  first.   This  option  is  purely  for  evaluating  the performance effect of the read
              evaluation order, and specifying it will slow down execution without improving alignments.

       <b>-nt</b>    Don't truncate searches based on missed seed hits.  This  option  is  purely  for  evaluating  the
              performance  effect  of  candidate  truncation, and specifying it will slow down execution without
              improving alignments.

       <b>-ne</b>    Don't try edit distance scoring before doing affine gap.  This option is to evaluate  the  aligner
              and  isn't  intended  to  be used for ordinary alignments.  It turns off normal affine gap scoring
              (like <b>-G-</b>) and so will have significant effects on the alignment results.

       <b>-nb</b>    Don't use the banded affine gap optimization.  This option is to evaluate  the  aligner  and  will
              just result in slower alignments.

       <b>-wbs</b>   Write  buffer  size in megabytes.  Don't specify this unless you've gotten an error message saying
              to make it bigger.  Default 16.

       <b>-di</b>    Drop the index after aligning and before sorting.  This frees  up  memory  for  the  sort  at  the
              expense of not having the index loaded for your next run.

       <b>-kts</b>   Kill if too slow.  Monitor our progress and kill ourself if we're not moving fast enough.  This is
              intended  for  use on machines with limited memory, where some alignment tasks will push SNAP into
              paging, and take disproportinaltely long.   This  allows  the  script  to  move  on  to  the  next
              alignment.   Only  works when generating output, and not during the sort phase.  If you're running
              out of memory sorting, try using <b>-di</b>.

       <b>-pro</b>   Profile alignment to give you an idea of how much time is spent aligning and how much waiting  for
              IO

       <b>-proAg</b> Profile affine-gap scoring to show how often it forces single-end alignment

       <b>-ae</b>    Apply  the  end-of-contig  soft  clipping  before the <b>-om</b> processing rather than after it.  A read
              that's soft clipped because of hanging off one end or the other  of  a  contig  does  not  have  a
              penalty in its NM tag, but it does in SNAP's internal scoring.  This flag says to use the NM value
              for <b>-om</b> processing rather than SNAP's internal score.

       <b>-is</b>    Write  SNAP's  internal score for an alignment into the output.  The value following <b>-is</b> specifies
              the tag to use, and must be a two letter value starting with X, Y or Z.  So,  <b>-is</b>  ZQ  will  cause
              SNAP to write ZQ:i:3 on a read with internal score 3.  Generally, the internal scores are the same
              as  the  NM  values, except that they contain penalties for soft clipping reads that hang over the
              end of contigs (but not for soft clipping that's due to # quality scores or that  was  present  in
              the input SAM/BAM file and retained due to <b>-pc</b>)

       <b>-G-</b>    disable affine gap scoring Affine gap scoring parameters (works only when <b>-G-</b> is not used):

       <b>-gm</b>    cost for match (default: 1)

       <b>-gs</b>    cost for substitution (default: 4)

       <b>-go</b>    cost for opening a gap (default: 6)

       <b>-ge</b>    cost for extending a gap (default: 1)

       <b>-g5</b>    bonus for alignment reaching 5' end of read (default: 10)

       <b>-g3</b>    bonus for alignment reaching 3' end of read (default: 7)

       <b>-A-</b>    Disable  ALT  awareness.   The  default is to try to map reads to the primary assembly and only to
              choose ALT alignments when they're much better, and to compute MAPQ for non-ALT  alignments  using
              only non-ALT hits. This flag disables that behavior and results in ALT-oblivious behavior.

       <b>-ea</b>    Emit ALT alignments.  When the aligner is ALT aware (i.e., <b>-A-</b> isn't specified) if it finds an ALT
              alignment  that  would have been the primary alignment if <b>-A-</b> had been specified but isn't without
              <b>-A-</b>, SNAP will emit the read with the supplementary alignment flag set and  MAPQ  computed  across
              all potential mappings, both primary and ALT

       <b>-asg</b>   Maximum  score gap to prefer a non-ALT alignment.  If the best non-ALT alignment is more than this
              much worse than the best ALT alignment emit the ALT alignment as the primary result rather than as
              a supplementary result. (default: 64)

       <b>-fmb</b>   Force MAPQ below this value to zero.  By the strict definition of MAPQ a  read  with  two  equally
              good  alignments should have MAPQ 3 Other aligners, however, will score these alignments at MAPQ 0
              and some variant callers depend on that behavior.  Setting this will force any MAPQ  value  at  or
              below the parameter value to zero.  (default:3)

       <b>-hc</b>    Enable SNAP mode optimized for use with GATK HaplotypeCaller.

       <b>-hc-</b>   Turn  off  optimizations  specific  to  GATK  HaplotypeCaller (e.g., when using the DRAGEN variant
              caller on SNAP aligned output) In this mode, when  a  read  (or  pair)  doesn't  align,  try  soft
              clipping the read (or pair) to find an alignment.  This is the default.

       <b>-at</b>    Attach  AT:i:  tags to each read showing the alignment time in microseconds.  For paired-end reads
              this is the time for the pair.

       <b>-pfc</b>   Preserve FASTQ comments.  Anything after the first white space on the FASTQ ID line is appended to
              the SAM/BAM line.  If this is not in valid SAM/BAM format it will produce incorrect output.

       <b>-q</b>     Quiet mode: don't print status messages (other than the welcome message which is printed prior  to
              parsing args).  Error messages are still printed.

       <b>-qq</b>    Super quiet mode: don't print status or error messages.

       You  may  process  more than one alignment without restarting SNAP, and if possible without reloading the
       index.  In order to do this, list on the command line all of the  parameters  for  the  first  alignment,
       followed  by  a comma (separated by a space from the other parameters) followed by the parameters for the
       next alignment (including single or paired).  You may have as many  of  these  as  you  please.   If  two
       consecutive  alignments  use  the  same  index,  it  will not be reloaded.  So, for example, you could do
       'snap-aligner single hg19-20 foo.fq <b>-o</b> foo.sam , paired hg19-20 end1.fq end2.fq  <b>-o</b>  paired.sam'  and  it
       would  not reload the index between the single and paired alignments.  SNAP doesn't parse the options for
       later runs until the earlier ones have completed, so if you make an error in one, it may take a while for
       you to notice.  So, be careful (or check back shortly after you think each run will have completed).

       When specifying an input or output file, you can simply list the filename, in which case SNAP will  infer
       the  type  of  the file from the file extension (.sam or .bam for example), or you can explicitly specify
       the file type by preceding the filename with one of the

              following type specifiers (which are case sensitive):

       <b>-fastq</b>

       <b>-compressedFastq</b>

       <b>-sam</b>

       <b>-bam</b>

       <b>-pairedFastq</b>

       <b>-pairedInterleavedFastq</b>

       <b>-pairedCompressedInterleavedFastq</b>

       So, for example, you could specify <b>-bam</b> input.file to make SNAP treat input.file  as  a  BAM  file,  even
       though  it  would  ordinarily  assume  a  FASTQ  file  for input or a SAM file for output when it doesn't
       recoginize the file extension.  In order to use a file name that begins with a  '-'  and  not  have  SNAP
       treat  it  as  a switch, you must explicitly specify the type.  But really, that's just confusing and you
       shouldn't do it.  Input and output may also be from/to stdin/stdout. To do that, use a - for the input or
       output file name and give an explicit type specifier.   So,  for  example,  snap-aligner  single  myIndex
       <b>-fastq</b> - <b>-o</b> <b>-sam</b> - would read FASTQ from stdin and write SAM to stdout.

snap-aligner single 2.0.3                         January 2024                            <u><a href="../man1/SNAP-ALIGNER_SINGLE.1.html">SNAP-ALIGNER_SINGLE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>