<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Search::HSP::HMMERHSP - A HSP object for HMMER results</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-searchio-hmmer-perl">libbio-searchio-hmmer-perl_1.7.3-2_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Search::HSP::HMMERHSP - A HSP object for HMMER results

</pre><h4><b>SYNOPSIS</b></h4><pre>
           use Bio::Search::HSP::HMMERHSP;
           # use it just like a Bio::Search::HSP::GenericHSP object

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This object is a specialization of Bio::Search::HSP::GenericHSP.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Jason</b> <b>Stajich</b></h4><pre>
       Email jason-at-bioperl.org

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : my $obj = Bio::Search::HSP::HMMERHSP-&gt;new();
        Function: Builds a new Bio::Search::HSP::HMMERHSP object
        Returns : Bio::Search::HSP::HMMERHSP
        Args    :

       Plus Bio::Search::HSP::GenericHSP methods

           -algorithm    =&gt; algorithm used (BLASTP, TBLASTX, FASTX, etc)
           -evalue       =&gt; evalue
           -pvalue       =&gt; pvalue
           -bits         =&gt; bit value for HSP
           -score        =&gt; score value for HSP (typically z-score but depends on
                                                          analysis)
           -hsp_length   =&gt; Length of the HSP (including gaps)
           -identical    =&gt; # of residues that that matched identically
           -conserved    =&gt; # of residues that matched conservatively
                            (only protein comparisons -
                            conserved == identical in nucleotide comparisons)
           -hsp_gaps     =&gt; # of gaps in the HSP
           -query_gaps   =&gt; # of gaps in the query in the alignment
           -hit_gaps     =&gt; # of gaps in the subject in the alignment
           -query_name   =&gt; HSP Query sequence name (if available)
           -query_start  =&gt; HSP Query start (in original query sequence coords)
           -query_end    =&gt; HSP Query end (in original query sequence coords)
           -hit_name     =&gt; HSP Hit sequence name (if available)
           -hit_start    =&gt; HSP Hit start (in original hit sequence coords)
           -hit_end      =&gt; HSP Hit end (in original hit sequence coords)
           -hit_length   =&gt; total length of the hit sequence
           -query_length =&gt; total length of the query sequence
           -query_seq    =&gt; query sequence portion of the HSP
           -hit_seq      =&gt; hit sequence portion of the HSP
           -homology_seq =&gt; homology sequence for the HSP
           -hit_frame    =&gt; hit frame (only if hit is translated protein)
           -query_frame  =&gt; query frame (only if query is translated protein)

   <b>Bio::Search::HSP::HSPI</b> <b>methods</b>
       Implementation of Bio::Search::HSP::HSPI methods follow

   <b>algorithm</b>
        Title   : algorithm
        Usage   : my $r_type = $hsp-&gt;algorithm
        Function: Obtain the name of the algorithm used to obtain the HSP
        Returns : string (e.g., BLASTP)
        Args    : [optional] scalar string to set value

   <b>pvalue</b>
        Title   : pvalue
        Usage   : my $pvalue = $hsp-&gt;pvalue();
        Function: Returns the P-value for this HSP or undef
        Returns : float or exponential (2e-10)
                  P-value is not defined with NCBI Blast2 reports.
        Args    : [optional] numeric to set value

   <b>evalue</b>
        Title   : evalue
        Usage   : my $evalue = $hsp-&gt;evalue();
        Function: Returns the e-value for this HSP
        Returns : float or exponential (2e-10)
        Args    : [optional] numeric to set value

   <b>frac_identical</b>
        Title   : frac_identical
        Usage   : my $frac_id = $hsp-&gt;frac_identical( ['query'|'hit'|'total'] );
        Function: Returns the fraction of identitical positions for this HSP
        Returns : Float in range 0.0 -&gt; 1.0
        Args    : arg 1:  'query' = num identical / length of query seq (without gaps)
                          'hit'   = num identical / length of hit seq (without gaps)
                          'total' = num identical / length of alignment (with gaps)
                          default = 'total'
                  arg 2: [optional] frac identical value to set for the type requested

   <b>frac_conserved</b>
        Title    : frac_conserved
        Usage    : my $frac_cons = $hsp-&gt;frac_conserved( ['query'|'hit'|'total'] );
        Function : Returns the fraction of conserved positions for this HSP.
                   This is the fraction of symbols in the alignment with a
                   positive score.
        Returns : Float in range 0.0 -&gt; 1.0
        Args    : arg 1: 'query' = num conserved / length of query seq (without gaps)
                         'hit'   = num conserved / length of hit seq (without gaps)
                         'total' = num conserved / length of alignment (with gaps)
                         default = 'total'
                  arg 2: [optional] frac conserved value to set for the type requested

   <b>gaps</b>
        Title    : gaps
        Usage    : my $gaps = $hsp-&gt;gaps( ['query'|'hit'|'total'] );
        Function : Get the number of gaps in the query, hit, or total alignment.
        Returns  : Integer, number of gaps or 0 if none
        Args     : arg 1: 'query' = num gaps in query seq
                          'hit'   = num gaps in hit seq
                          'total' = num gaps in whole alignment
                          default = 'total'
                   arg 2: [optional] integer gap value to set for the type requested

   <b>query_string</b>
        Title   : query_string
        Usage   : my $qseq = $hsp-&gt;query_string;
        Function: Retrieves the query sequence of this HSP as a string
        Returns : string
        Args    : [optional] string to set for query sequence

   <b>hit_string</b>
        Title   : hit_string
        Usage   : my $hseq = $hsp-&gt;hit_string;
        Function: Retrieves the hit sequence of this HSP as a string
        Returns : string
        Args    : [optional] string to set for hit sequence

   <b>homology_string</b>
        Title   : homology_string
        Usage   : my $homo_string = $hsp-&gt;homology_string;
        Function: Retrieves the homology sequence for this HSP as a string.
                : The homology sequence is the string of symbols in between the
                : query and hit sequences in the alignment indicating the degree
                : of conservation (e.g., identical, similar, not similar).
        Returns : string
        Args    : [optional] string to set for homology sequence

   <b>length</b>
        Title    : length
        Usage    : my $len = $hsp-&gt;length( ['query'|'hit'|'total'] );
        Function : Returns the length of the query or hit in the alignment
                   (without gaps)
                   or the aggregate length of the HSP (including gaps;
                   this may be greater than either hit or query )
        Returns  : integer
        Args     : arg 1: 'query' = length of query seq (without gaps)
                          'hit'   = length of hit seq (without gaps)
                          'total' = length of alignment (with gaps)
                          default = 'total'
                   arg 2: [optional] integer length value to set for specific type

   <b>percent_identity</b>
        Title   : percent_identity
        Usage   : my $percentid = $hsp-&gt;percent_identity()
        Function: Returns the calculated percent identity for an HSP
        Returns : floating point between 0 and 100
        Args    : none

   <b>frame</b>
        Title   : frame
        Usage   : my ($qframe, $hframe) = $hsp-&gt;frame('list',$queryframe,$subjectframe)
        Function: Set the Frame for both query and subject and insure that
                  they agree.
                  This overrides the frame() method implementation in
                  FeaturePair.
        Returns : array of query and subject frame if return type wants an array
                  or query frame if defined or subject frame if not defined
        Args    : 'hit' or 'subject' or 'sbjct' to retrieve the frame of the subject (default)
                  'query' to retrieve the query frame
                  'list' or 'array' to retrieve both query and hit frames together
        Note    : Frames are stored in the GFF way (0-2) not 1-3
                  as they are in BLAST (negative frames are deduced by checking
                                        the strand of the query or hit)

   <b>get_aln</b>
        Title   : get_aln
        Usage   : my $aln = $hsp-&gt;gel_aln
        Function: Returns a Bio::SimpleAlign representing the HSP alignment
        Returns : Bio::SimpleAlign
        Args    : none

   <b>num_conserved</b>
        Title   : num_conserved
        Usage   : $obj-&gt;num_conserved($newval)
        Function: returns the number of conserved residues in the alignment
        Returns : inetger
        Args    : integer (optional)

   <b>num_identical</b>
        Title   : num_identical
        Usage   : $obj-&gt;num_identical($newval)
        Function: returns the number of identical residues in the alignment
        Returns : integer
        Args    : integer (optional)

   <b>seq_inds</b>
        Title   : seq_inds
        Purpose   : Get a list of residue positions (indices) for all identical
                  : or conserved residues in the query or sbjct sequence.
        Example   : @s_ind = $hsp-&gt;seq_inds('query', 'identical');
                  : @h_ind = $hsp-&gt;seq_inds('hit', 'conserved');
                  : @h_ind = $hsp-&gt;seq_inds('hit', 'conserved', 1);
        Returns   : List of integers
                  : May include ranges if collapse is true.
        Argument  : seq_type  = 'query' or 'hit' or 'sbjct'  (default = query)
                  :  ('sbjct' is synonymous with 'hit')
                  : class     = 'identical' or 'conserved' or 'nomatch' or 'gap'
                  :              (default = identical)
                  :              (can be shortened to 'id' or 'cons')
                  :
                  : collapse  = boolean, if true, consecutive positions are merged
                  :             using a range notation, e.g., "1 2 3 4 5 7 9 10 11"
                  :             collapses to "1-5 7 9-11". This is useful for
                  :             consolidating long lists. Default = no collapse.
        Throws    : n/a.
        Comments  :

       See Also   : <b>Bio::Search::BlastUtils::collapse_nums()</b>, <b>Bio::Search::Hit::HitI::seq_inds()</b>

   <b>Inherited</b> <b>from</b> <b>Bio::SeqFeature::SimilarityPair</b>
       These methods come from Bio::SeqFeature::SimilarityPair

   <b>query</b>
        Title   : query
        Usage   : my $query = $hsp-&gt;query
        Function: Returns a SeqFeature representing the query in the HSP
        Returns : Bio::SeqFeature::Similarity
        Args    : [optional] new value to set

   <b>hit</b>
        Title   : hit
        Usage   : my $hit = $hsp-&gt;hit
        Function: Returns a SeqFeature representing the hit in the HSP
        Returns : Bio::SeqFeature::Similarity
        Args    : [optional] new value to set

   <b>significance</b>
        Title   : significance
        Usage   : $evalue = $obj-&gt;significance();
                  $obj-&gt;significance($evalue);
        Function: Get/Set the significance value
        Returns : numeric
        Args    : [optional] new value to set

   <b>score</b>
        Title   : score
        Usage   : my $score = $hsp-&gt;score();
        Function: Returns the score for this HSP or undef
        Returns : numeric
        Args    : [optional] numeric to set value

   <b>bits</b>
        Title   : bits
        Usage   : my $bits = $hsp-&gt;bits();
        Function: Returns the bit value for this HSP or undef
        Returns : numeric
        Args    : none

perl v5.30.0                                       2019-10-29                    <u>Bio::Search::HSP::<a href="../man3pm/HMMERHSP.3pm.html">HMMERHSP</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>