<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Search::HSP::ModelHSP - A HSP object for model-based searches</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Search::HSP::ModelHSP - A HSP object for model-based searches

</pre><h4><b>SYNOPSIS</b></h4><pre>
           use Bio::Search::HSP::ModelHSP;
           # us it just like a Bio::Search::HSP::ModelHSP object

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This object is a specialization of Bio::Search::HSP::ModelHSP and is used for searches which involve a
       query model, such as a Hidden Markov Model (HMM), covariance model (CM), descriptor, or anything else
       besides a sequence. Note that results from any HSPI class methods which rely on the query being a
       sequence are unreliable and have thus been overridden with warnings indicating they have not been
       implemented at this time.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to the Bioperl mailing list.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track of the bugs and their resolution.
       Bug reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Chris</b> <b>Fields</b></h4><pre>
       Email cjfields at bioperl dot org

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>new</b>
        Title   : new
        Usage   : my $obj = Bio::Search::HSP::ModelHSP-&gt;new();
        Function: Builds a new Bio::Search::HSP::ModelHSP object
        Returns : Bio::Search::HSP::ModelHSP
        Args    :

       Plus Bio::Seach::HSP::ModelHSP methods

                  -algorithm =&gt; algorithm used (Infernal, RNAMotif, ERPIN, etc)
                  -evalue    =&gt; evalue
                  -pvalue    =&gt; pvalue
                  -bits      =&gt; bit value for HSP
                  -score     =&gt; score value for HSP (typically z-score but depends on
                                                     analysis)
                  -hsp_length=&gt; Length of the HSP (including gaps)
                  -identical =&gt; # of residues that that matched identically
                  -conserved =&gt; # of residues that matched conservatively
                                  (only protein comparisons;
                                   conserved == identical in nucleotide comparisons)
                  -hsp_gaps   =&gt; # of gaps in the HSP
                  -query_gaps =&gt; # of gaps in the query in the alignment
                  -hit_gaps   =&gt; # of gaps in the subject in the alignment
                  -query_name  =&gt; HSP Query sequence name (if available)
                  -query_start =&gt; HSP Query start (in original query sequence coords)
                  -query_end   =&gt; HSP Query end (in original query sequence coords)
                  -hit_name    =&gt; HSP Hit sequence name (if available)
                  -hit_start   =&gt; HSP Hit start (in original hit sequence coords)
                  -hit_end     =&gt; HSP Hit end (in original hit sequence coords)
                  -hit_length  =&gt; total length of the hit sequence
                  -query_length=&gt; total length of the query sequence
                  -query_seq   =&gt; query sequence portion of the HSP
                  -hit_seq     =&gt; hit sequence portion of the HSP
                  -homology_seq=&gt; homology sequence for the HSP
                  -hit_frame   =&gt; hit frame (only if hit is translated protein)
                  -query_frame =&gt; query frame (only if query is translated protein)
                  -meta        =&gt; optional meta data (sec structure, markup, etc)
                  -custom_score=&gt; custom score data

   <b>meta</b>
        Title   : meta
        Usage   : my $meta = $hsp-&gt;meta();
        Function: Returns meta data for this HSP or undef
        Returns : string of meta data or undef
        Args    : [optional] string to set value
        Note    : At some point very soon this will likely be a Bio::AnnotationI.
                  Don't get used to a simple string!

   <b>noncanonical_string</b>
        Title   : noncanonical_string
        Usage   : my $nc_seq = $hsp-&gt;noncanonical_string();
        Function: Returns noncanonical string (NC) data for this HSP or undef
        Returns : string of noncanonical data or undef
        Args    : [optional] string to set value

   <b>custom_score</b>
        Title   : custom_score
        Usage   : my $data = $hsp-&gt;custom_score();
        Function: Returns custom_score data for this HSP, or undef
        Returns : custom_score data or undef
        Args    : [optional] custom_score
        Note    : This is a Get/Set used to deal with odd score-like data generated
                  from RNAMotif (and other programs) where the score section
                  can be customized to include non-standard data, including sequence
                  data, user-based scores, and other values.

   <b>Bio::Search::HSP::HSPI</b> <b>methods</b>
       Implementation of Bio::Search::HSP::HSPI methods follow

   <b>algorithm</b>
        Title   : algorithm
        Usage   : my $r_type = $hsp-&gt;algorithm
        Function: Obtain the name of the algorithm used to obtain the HSP
        Returns : string (e.g., BLASTP)
        Args    : [optional] scalar string to set value

   <b>strand</b>
        Title   : strand
        Usage   : $hsp-&gt;strand('hit')
        Function: Retrieves the strand for the HSP component requested
        Returns : +1 or -1 (0 if unknown)
        Args    : 'hit' or 'subject' or 'sbjct' to retrieve the strand of the subject.
                  There is no strand available for 'query', as the query is a model
                  and not a true sequence.

   <b>seq</b>
        Usage     : $hsp-&gt;seq( [seq_type] );
        Purpose   : Get the query or sbjct sequence as a Bio::Seq.pm object.
        Example   : $seqObj = $hsp-&gt;seq('sbjct');
        Returns   : Object reference for a Bio::Seq.pm object.
        Argument  : seq_type = 'query' or 'hit' or 'sbjct' (default = 'sbjct').
                  :  ('sbjct' is synonymous with 'hit')
                  : default is 'sbjct'
                  : Note: if there is no sequence available (eg for a model-based
                  : search), this returns a LocatableSeq object w/o a sequence
        Throws    : Propagates any exception that occurs during construction
                  : of the Bio::Seq.pm object.
        Comments  : The sequence is returned in an array of strings corresponding
                  : to the strings in the original format of the Blast alignment.
                  : (i.e., same spacing).

       See Also   : <b>seq_str()</b>, Bio::Seq

   <b>pvalue</b>
        Title   : pvalue
        Usage   : my $pvalue = $hsp-&gt;pvalue();
        Function: Returns the P-value for this HSP or undef
        Returns : float or exponential (2e-10)
                  P-value is not defined with NCBI Blast2 reports.
        Args    : [optional] numeric to set value

   <b>evalue</b>
        Title   : evalue
        Usage   : my $evalue = $hsp-&gt;evalue();
        Function: Returns the e-value for this HSP
        Returns : float or exponential (2e-10)
        Args    : [optional] numeric to set value

   <b>gaps</b>
        Title    : gaps
        Usage    : my $gaps = $hsp-&gt;gaps( ['query'|'hit'|'total'] );
        Function : Get the number of gaps in the query, hit, or total alignment.
        Returns  : Integer, number of gaps or 0 if none
        Args     : arg 1: 'query' = num gaps in query seq
                          'hit'   = num gaps in hit seq
                          'total' = num gaps in whole alignment
                          default = 'total'
                   arg 2: [optional] integer gap value to set for the type requested

   <b>query_string</b>
        Title   : query_string
        Usage   : my $qseq = $hsp-&gt;query_string;
        Function: Retrieves the query sequence of this HSP as a string
        Returns : string
        Args    : [optional] string to set for query sequence

   <b>hit_string</b>
        Title   : hit_string
        Usage   : my $hseq = $hsp-&gt;hit_string;
        Function: Retrieves the hit sequence of this HSP as a string
        Returns : string
        Args    : [optional] string to set for hit sequence

   <b>homology_string</b>
        Title   : homology_string
        Usage   : my $homo_string = $hsp-&gt;homology_string;
        Function: Retrieves the homology sequence for this HSP as a string.
                : The homology sequence is the string of symbols in between the
                : query and hit sequences in the alignment indicating the degree
                : of conservation (e.g., identical, similar, not similar).
        Returns : string
        Args    : [optional] string to set for homology sequence

   <b>length</b>
        Title    : length
        Usage    : my $len = $hsp-&gt;length( ['query'|'hit'|'total'] );
        Function : Returns the length of the query or hit in the alignment
                   (without gaps)
                   or the aggregate length of the HSP (including gaps;
                   this may be greater than either hit or query )
        Returns  : integer
        Args     : arg 1: 'query' = length of query seq (without gaps)
                          'hit'   = length of hit seq (without gaps)
                          'total' = length of alignment (with gaps)
                          default = 'total'
                   arg 2: [optional] integer length value to set for specific type

   <b>frame</b>
        Title   : frame
        Usage   : my ($qframe, $hframe) = $hsp-&gt;frame('list',$queryframe,$subjectframe)
        Function: Set the Frame for both query and subject and insure that
                  they agree.
                  This overrides the frame() method implementation in
                  FeaturePair.
        Returns : array of query and subject frame if return type wants an array
                  or query frame if defined or subject frame if not defined
        Args    : 'hit' or 'subject' or 'sbjct' to retrieve the frame of the subject (default)
                  'query' to retrieve the query frame
                  'list' or 'array' to retrieve both query and hit frames together
        Note    : Frames are stored in the GFF way (0-2) not 1-3
                  as they are in BLAST (negative frames are deduced by checking
                                        the strand of the query or hit)

   <b>get_aln</b>
        Title   : get_aln
        Usage   : my $aln = $hsp-&gt;gel_aln
        Function: Returns a Bio::SimpleAlign representing the HSP alignment
        Returns : Bio::SimpleAlign
        Args    : none

   <b>Inherited</b> <b>from</b> <b>Bio::SeqFeature::SimilarityPair</b>
       These methods come from Bio::SeqFeature::SimilarityPair

   <b>query</b>
        Title   : query
        Usage   : my $query = $hsp-&gt;query
        Function: Returns a SeqFeature representing the query in the HSP
        Returns : Bio::SeqFeature::Similarity
        Args    : [optional] new value to set

   <b>hit</b>
        Title   : hit
        Usage   : my $hit = $hsp-&gt;hit
        Function: Returns a SeqFeature representing the hit in the HSP
        Returns : Bio::SeqFeature::Similarity
        Args    : [optional] new value to set

   <b>significance</b>
        Title   : significance
        Usage   : $evalue = $obj-&gt;significance();
                  $obj-&gt;significance($evalue);
        Function: Get/Set the significance value
        Returns : numeric
        Args    : [optional] new value to set

   <b>score</b>
        Title   : score
        Usage   : my $score = $hsp-&gt;score();
        Function: Returns the score for this HSP or undef
        Returns : numeric
        Args    : [optional] numeric to set value

   <b>bits</b>
        Title   : bits
        Usage   : my $bits = $hsp-&gt;bits();
        Function: Returns the bit value for this HSP or undef
        Returns : numeric
        Args    : none

</pre><h4><b>ModelHSP</b> <b>methods</b> <b>overridden</b> <b>in</b> <b>ModelHSP</b></h4><pre>
       The following methods have been overridden due to their current reliance on sequence-based queries. They
       may be implemented in future versions of this class.

   <b>seq_inds</b>
   <b>frac_identical</b>
   <b>frac_conserved</b>
   <b>matches</b>
   <b>num_conserved</b>
   <b>num_identical</b>
   <b>cigar_string</b>
   <b>generate_cigar_string</b>
   <b>percent_identity</b>
perl v5.32.1                                       2021-08-15                    <u>Bio::Search::HSP::<a href="../man3pm/ModelHSP.3pm.html">ModelHSP</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>