<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Search::Hit::HitI - Interface for a hit in a similarity search result</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Search::Hit::HitI - Interface for a hit in a similarity search result

</pre><h4><b>SYNOPSIS</b></h4><pre>
       # Bio::Search::Hit::HitI objects should not be instantiated since this # module defines a pure interface.

       # Given an object that implements the Bio::Search::Hit::HitI  interface, # you can do the following
       things with it:

           # Get a HitI object from a SearchIO stream:
           use Bio::SeachIO;
           my $searchio = Bio::SearchIO-&gt;new(-format =&gt; 'blast', -file =&gt; 'result.bls');
           my $result = $searchio-&gt;next_result;
           my $hit    = $result-&gt;next_hit;

           $hit_name = $hit-&gt;name();

           $desc = $hit-&gt;description();

           $len = $hit-&gt;length

           $alg = $hit-&gt;algorithm();

           $score = $hit-&gt;raw_score();

           $significance = $hit-&gt;significance();

           $rank = $hit-&gt;rank(); # the Nth hit for a specific query

           while( $hsp = $obj-&gt;next_hsp()) { ... } # process in iterator fashion

           for my $hsp ( $obj-&gt;hsps()()) { ... } # process in list fashion

</pre><h4><b>DESCRIPTION</b></h4><pre>
           Bio::Search::Hit::* objects are data structures that contain information
       about specific hits obtained during a library search.  Some information will
       be algorithm-specific, but others will be generally defined.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b> <b>-</b> <b>Aaron</b> <b>Mackey,</b> <b>Steve</b> <b>Chervitz</b></h4><pre>
       Email <a href="mailto:amackey@virginia.edu">amackey@virginia.edu</a>  (original author) Email <a href="mailto:sac@bioperl.org">sac@bioperl.org</a>

</pre><h4><b>COPYRIGHT</b></h4><pre>
       Copyright (c) 1999-2001 Aaron Mackey, Steve Chervitz. All Rights Reserved.

</pre><h4><b>DISCLAIMER</b></h4><pre>
       This software is provided "as is" without warranty of any kind.

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods. Internal methods are usually preceded
       with a _

   <b>name</b>
        Title   : name
        Usage   : $hit_name = $hit-&gt;name();
        Function: returns the name of the Hit sequence
        Returns : a scalar string
        Args    : none

       The <b>name</b> of a hit is unique within a Result or within an Iteration.

   <b>description</b>
        Title   : description
        Usage   : $desc = $hit-&gt;description();
        Function: Retrieve the description for the hit
        Returns : a scalar string
        Args    : none

   <b>accession</b>
        Title   : accession
        Usage   : $acc = $hit-&gt;accession();
        Function: Retrieve the accession (if available) for the hit
        Returns : a scalar string (empty string if not set)
        Args    : none

   <b>locus</b>
        Title   : locus
        Usage   : $acc = $hit-&gt;locus();
        Function: Retrieve the locus(if available) for the hit
        Returns : a scalar string (empty string if not set)
        Args    : none

   <b>length</b>
        Title   : length
        Usage   : my $len = $hit-&gt;length
        Function: Returns the length of the hit
        Returns : integer
        Args    : none

   <b>algorithm</b>
        Title   : algorithm
        Usage   : $alg = $hit-&gt;algorithm();
        Function: Gets the algorithm specification that was used to obtain the hit
                  For BLAST, the algorithm denotes what type of sequence was aligned
                  against what (BLASTN: dna-dna, BLASTP prt-prt, BLASTX translated
                  dna-prt, TBLASTN prt-translated dna, TBLASTX translated
                  dna-translated dna).
        Returns : a scalar string
        Args    : none

   <b>raw_score</b>
        Title   : raw_score
        Usage   : $score = $hit-&gt;raw_score();
        Function: Gets the "raw score" generated by the algorithm.  What
                  this score is exactly will vary from algorithm to algorithm,
                  returning undef if unavailable.
        Returns : a scalar value
        Args    : none

   <b>score</b>
       Equivalent to <b>raw_score()</b>

   <b>significance</b>
        Title   : significance
        Usage   : $significance = $hit-&gt;significance();
        Function: Used to obtain the E or P value of a hit, i.e. the probability that
                  this particular hit was obtained purely by random chance.  If
                  information is not available (nor calculatable from other
                  information sources), return undef.
        Returns : a scalar value or undef if unavailable
        Args    : none

   <b>bits</b>
        Usage     : $hit_object-&gt;bits();
        Purpose   : Gets the bit score of the best HSP for the current hit.
        Example   : $bits = $hit_object-&gt;bits();
        Returns   : Integer or double for FASTA reports
        Argument  : n/a
        Comments  : For BLAST1, the non-bit score is listed in the summary line.

       See Also   : <b>score()</b>

   <b>next_hsp</b>
        Title    : next_hsp
        Usage    : while( $hsp = $obj-&gt;next_hsp()) { ... }
        Function : Returns the next available High Scoring Pair
        Example  :
        Returns  : L&lt;Bio::Search::HSP::HSPI&gt; object or null if finished
        Args     : none

   <b>hsps</b>
        Usage     : $hit_object-&gt;hsps();
        Purpose   : Get a list containing all HSP objects.
                  : Get the numbers of HSPs for the current hit.
        Example   : @hsps = $hit_object-&gt;hsps();
                  : $num  = $hit_object-&gt;hsps();  # alternatively, use num_hsps()
        Returns   : Array context : list of L&lt;Bio::Search::HSP::BlastHSP&gt; objects.
                  : Scalar context: integer (number of HSPs).
                  :                 (Equivalent to num_hsps()).
        Argument  : n/a. Relies on wantarray
        Throws    : Exception if the HSPs have not been collected.

       See Also   : <b>hsp()</b>, <b>num_hsps()</b>

   <b>num_hsps</b>
        Usage     : $hit_object-&gt;num_hsps();
        Purpose   : Get the number of HSPs for the present Blast hit.
        Example   : $nhsps = $hit_object-&gt;num_hsps();
        Returns   : Integer
        Argument  : n/a
        Throws    : Exception if the HSPs have not been collected.

       See Also   : <b>hsps()</b>

   <b>seq_inds</b>
        Usage     : $hit-&gt;seq_inds( seq_type, class, collapse );
        Purpose   : Get a list of residue positions (indices) across all HSPs
                  : for identical or conserved residues in the query or sbjct sequence.
        Example   : @s_ind = $hit-&gt;seq_inds('query', 'identical');
                  : @h_ind = $hit-&gt;seq_inds('hit', 'conserved');
                  : @h_ind = $hit-&gt;seq_inds('hit', 'conserved', 1);
        Returns   : Array of integers
                  : May include ranges if collapse is non-zero.
        Argument  : [0] seq_type  = 'query' or 'hit' or 'sbjct'  (default = 'query')
                  :                 ('sbjct' is synonymous with 'hit')
                  : [1] class     = 'identical' or 'conserved' (default = 'identical')
                  :              (can be shortened to 'id' or 'cons')
                  :              (actually, anything not 'id' will evaluate to 'conserved').
                  : [2] collapse  = boolean, if non-zero, consecutive positions are merged
                  :             using a range notation, e.g., "1 2 3 4 5 7 9 10 11"
                  :             collapses to "1-5 7 9-11". This is useful for
                  :             consolidating long lists. Default = no collapse.
        Throws    : n/a.

       See Also   : <b>Bio::Search::HSP::HSPI::seq_inds()</b>

   <b>rewind</b>
        Title   : rewind
        Usage   : $hit-&gt;rewind;
        Function: Allow one to reset the HSP iterator to the beginning
                  if possible
        Returns : none
        Args    : none

   <b>overlap</b>
        Usage     : $hit_object-&gt;overlap( [integer] );
        Purpose   : Gets/Sets the allowable amount overlap between different HSP sequences.
        Example   : $hit_object-&gt;<a href="../man5/overlap.5.html">overlap</a>(5);
                  : $overlap = $hit_object-&gt;overlap;
        Returns   : Integer.
        Argument  : integer.
        Throws    : n/a
        Status    : Experimental
        Comments  : Any two HSPs whose sequences overlap by less than or equal
                  : to the overlap() number of resides will be considered separate HSPs
                  : and will not get tiled by L&lt;Bio::Search::BlastUtils::_adjust_contigs()&gt;.

       See Also   : <b>Bio::Search::BlastUtils::_adjust_contigs()</b>, BUGS

   <b>n</b>
        Usage     : $hit_object-&gt;n();
        Purpose   : Gets the N number for the current Blast hit.
                  : This is the number of HSPs in the set which was ascribed
                  : the lowest P-value (listed on the description line).
                  : This number is not the same as the total number of HSPs.
                  : To get the total number of HSPs, use num_hsps().
        Example   : $n = $hit_object-&gt;n();
        Returns   : Integer
        Argument  : n/a
        Throws    : Exception if HSPs have not been set (BLAST2 reports).
        Comments  : Note that the N parameter is not reported in gapped BLAST2.
                  : Calling n() on such reports will result in a call to num_hsps().
                  : The num_hsps() method will count the actual number of
                  : HSPs in the alignment listing, which may exceed N in
                  : some cases.

       See Also   : <b>num_hsps()</b>

   <b>p</b>
        Usage     : $hit_object-&gt;p( [format] );
        Purpose   : Get the P-value for the best HSP of the given BLAST hit.
                  : (Note that P-values are not provided with NCBI Blast2 reports).
        Example   : $p =  $sbjct-&gt;p;
                  : $p =  $sbjct-&gt;p('exp');  # get exponent only.
                  : ($num, $exp) =  $sbjct-&gt;p('parts');  # split sci notation into parts
        Returns   : Float or scientific notation number (the raw P-value, DEFAULT).
                  : Integer if format == 'exp' (the magnitude of the base 10 exponent).
                  : 2-element list (float, int) if format == 'parts' and P-value
                  :                is in scientific notation (See Comments).
        Argument  : format: string of 'raw' | 'exp' | 'parts'
                  :    'raw' returns value given in report. Default. (1.2e-34)
                  :    'exp' returns exponent value only (34)
                  :    'parts' returns the decimal and exponent as a
                  :            2-element list (1.2, -34) (See Comments).
        Throws    : Warns if no P-value is defined. Uses expect instead.
        Comments  : Using the 'parts' argument is not recommended since it will not
                  : work as expected if the P-value is not in scientific notation.
                  : That is, floats are not converted into sci notation before
                  : splitting into parts.

       See Also   : <b>expect()</b>, <b>signif()</b>, <b>Bio::Search::BlastUtils::get_exponent()</b>

   <b>hsp</b>
        Usage     : $hit_object-&gt;hsp( [string] );
        Purpose   : Get a single HSPI object for the present HitI object.
        Example   : $hspObj  = $hit_object-&gt;hsp;  # same as 'best'
                  : $hspObj  = $hit_object-&gt;hsp('best');
                  : $hspObj  = $hit_object-&gt;hsp('worst');
        Returns   : Object reference for a L&lt;Bio::Search::HSP::HSPI&gt; object.
        Argument  : String (or no argument).
                  :   No argument (default) = highest scoring HSP (same as 'best').
                  :   'best' or 'first' = highest scoring HSP.
                  :   'worst' or 'last' = lowest scoring HSP.
        Throws    : Exception if the HSPs have not been collected.
                  : Exception if an unrecognized argument is used.

       See Also   : <b>hsps()</b>, num_hsps()

   <b>logical_length</b>
        Usage     : $hit_object-&gt;logical_length( [seq_type] );
                  : (mostly intended for internal use).
        Purpose   : Get the logical length of the hit sequence.
                  : If the Blast is a TBLASTN or TBLASTX, the returned length
                  : is the length of the would-be amino acid sequence (length/3).
                  : For all other BLAST flavors, this function is the same as length().
        Example   : $len    = $hit_object-&gt;logical_length();
        Returns   : Integer
        Argument  : seq_type = 'query' or 'hit' or 'sbjct' (default = 'query')
                    ('sbjct' is synonymous with 'hit')
        Throws    : n/a
        Comments  : This is important for functions like frac_aligned_query()
                  : which need to operate in amino acid coordinate space when dealing
                  : with [T]BLAST[NX] type reports.

       See Also   : <b>length()</b>, <b>frac_aligned_query()</b>, <b>frac_aligned_hit()</b>

   <b>rank</b>
        Title   : rank
        Usage   : $obj-&gt;rank($newval)
        Function: Get/Set the rank of this Hit in the Query search list
                  i.e. this is the Nth hit for a specific query
        Returns : value of rank
        Args    : newvalue (optional)

   <b>each_accession_number</b>
        Title   : each_accession_number
        Usage   : $obj-&gt;each_accession_number
        Function: Get each accession number listed in the description of the hit.
                  If there are no alternatives, then only the primary accession will
                  be given
        Returns : list of all accession numbers in the description
        Args    : none

   <b>tiled_hsps</b>
        Usage     : $hit_object-&gt;tiled_hsps( [integer] );
        Purpose   : Gets/Sets an indicator for whether or not the HSPs in this Hit
                  : have been tiled.
                  : Methods that rely on HSPs being tiled should check this
                  : and then call SearchUtils::tile_hsps() if not.
        Example   : $hit_object-&gt;<a href="../man1/tiled_hsps.1.html">tiled_hsps</a>(1);
                  : if( $hit_object-&gt;tiled_hsps ) { # do something }
        Returns   : Boolean (1 or 0)
        Argument  : integer (optional)
        Throws    : n/a

   <b>strand</b>
        Usage     : $sbjct-&gt;strand( [seq_type] );
        Purpose   : Gets the strand(s) for the query, sbjct, or both sequences
                  : in the best HSP of the BlastHit object after HSP tiling.
                  : Only valid for BLASTN, TBLASTX, BLASTX-query, TBLASTN-hit.
        Example   : $qstrand = $sbjct-&gt;strand('query');
                  : $sstrand = $sbjct-&gt;strand('hit');
                  : ($qstrand, $sstrand) = $sbjct-&gt;strand();
        Returns   : scalar context: integer '1', '-1', or '0'
                  : array context without args: list of two strings (queryStrand, sbjctStrand)
                  : Array context can be "induced" by providing an argument of 'list' or 'array'.
        Argument  : In scalar context: seq_type = 'query' or 'hit' or 'sbjct' (default = 'query')
                    ('sbjct' is synonymous with 'hit')
        Throws    : n/a
        Comments  : This method requires that all HSPs be tiled. If they have not
                  : already been tiled, they will be tiled first automatically..
                  : If you don't want the tiled data, iterate through each HSP
                  : calling strand() on each (use hsps() to get all HSPs).
                  :
                  : Formerly (prior to 10/21/02), this method would return the
                  : string "-1/1" for hits with HSPs on both strands.
                  : However, now that strand and frame is properly being accounted
                  : for during HSP tiling, it makes more sense for strand()
                  : to return the strand data for the best HSP after tiling.
                  :
                  : If you really want to know about hits on opposite strands,
                  : you should be iterating through the HSPs using methods on the
                  : HSP objects.
                  :
                  : A possible use case where knowing whether a hit has HSPs
                  : on both strands would be when filtering via SearchIO for hits with
                  : this property. However, in this case it would be better to have a
                  : dedicated method such as $hit-&gt;hsps_on_both_strands(). Similarly
                  : for frame. This could be provided if there is interest.

       See Also   : Bio::Search::HSP::HSPI::strand()

   <b>frame</b>
        Usage     : $hit_object-&gt;frame();
        Purpose   : Gets the reading frame for the best HSP after HSP tiling.
                  : This is only valid for BLASTX and TBLASTN/X type reports.
        Example   : $frame = $hit_object-&gt;frame();
        Returns   : Integer (-2 .. +2)
        Argument  : n/a
        Throws    : Exception if HSPs have not been set.
        Comments  : This method requires that all HSPs be tiled. If they have not
                  : already been tiled, they will be tiled first automatically..
                  : If you don't want the tiled data, iterate through each HSP
                  : calling frame() on each (use hsps() to get all HSPs).

       See Also   : <b>hsps()</b>

   <b>matches</b>
        Usage     : $hit_object-&gt;matches( [class] );
        Purpose   : Get the total number of identical or conserved matches
                  : (or both) across all HSPs.
                  : (Note: 'conservative' matches are indicated as 'positives'
                  :         in BLAST reports.)
        Example   : ($id,$cons) = $hit_object-&gt;matches(); # no argument
                  : $id = $hit_object-&gt;matches('id');
                  : $cons = $hit_object-&gt;matches('cons');
        Returns   : Integer or a 2-element array of integers
        Argument  : class = 'id' | 'cons' OR none.
                  : If no argument is provided, both identical and conservative
                  : numbers are returned in a two element list.
                  : (Other terms can be used to refer to the conservative
                  :  matches, e.g., 'positive'. All that is checked is whether or
                  :  not the supplied string starts with 'id'. If not, the
                  : conservative matches are returned.)
        Throws    : Exception if the requested data cannot be obtained.
        Comments  : This method requires that all HSPs be tiled. If there is more than one
                  : HSP and they have not already been tiled, they will be tiled first automatically..
                  :
                  : If you need data for each HSP, use hsps() and then interate
                  : through the HSP objects.
                  : Does not rely on wantarray to return a list. Only checks for
                  : the presence of an argument (no arg = return list).

       See Also   : <b>Bio::Search::HSP::GenericHSP::matches()</b>, <b>hsps()</b>

   <b>sort_hits</b>
        Title          : sort_hsps
        Usage          : $result-&gt;sort_hsps(\&amp;sort_function)
        Function       : Sorts the available HSP objects by a user-supplied function. Defaults to sort
                         by descending score.
        Returns        : n/a
        Args           : A coderef for the sort function.  See the documentation on the Perl sort()
                         function for guidelines on writing sort functions.
        Note           : To access the special variables $a and $b used by the Perl sort() function
                         the user function must access Bio::Search::Hit::HitI namespace.
                         For example, use :
                         $hit-&gt;sort_hsps( sub{$Bio::Search::Result::HitI::a-&gt;length &lt;=&gt;
                                                 $Bio::Search::Result::HitI::b-&gt;length});
                          NOT $hit-&gt;sort_hsps($a-&gt;length &lt;=&gt; $b-&gt;length);

   <b>_default</b> <b>sort_hsps</b>
         Title : _default_sort_hsps
         Usage : Do not call directly.
         Function : Sort hsps in ascending order by evalue
         Args  : None
         Returns: 1 on success
         Note  : Used by $hit-&gt;sort_hsps()

perl v5.32.1                                       2021-08-15                        <u>Bio::Search::Hit::<a href="../man3pm/HitI.3pm.html">HitI</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>