<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>QTLtools quan - Quantify gene and exon expression from RNA-seq</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/qtltools">qtltools_1.3.1+dfsg-4build4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       QTLtools quan - Quantify gene and exon expression from RNA-seq

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>QTLtools</b> <b>quan</b> <b>--bam</b> <u>[in.sam</u>|<u>in.bam</u>|<u>in.cram]</u> <b>--gtf</b> <u>gene_annotation.gtf</u> <b>--out-prefix</b> <u>output</u> <u>[OPTIONS]</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This mode quantifies the expression of genes and exons in the provided <b>--gtf</b> file using the RNA-seq reads
       in  the  <b>--bam</b>  file.   The  method  counts  the number of reads overlapping the exons in the <b>--gtf</b> file.
       Firstly all exons of a gene are converted into meta-exons where  overlapping  exons  are  merged  into  a
       single  exon  encompassing  all  the  overlapping  exons.   Any  overlap between the read and the exon is
       considered a match, that is a read is not required to be in between start and end positions of an exon to
       count towards that exon's quantification.  Split reads aligning to multiple exons contribute to each exon
       it overlaps with based on the fraction of the read that overlaps with a given  exon.   Thus  split  reads
       contribute  less  than a single count to each of the overlapping exons.  Reads aligning to multiple exons
       (i.e. overlapping exons of multiple genes) count towards the quantification of  all  the  exons  that  it
       overlaps  with.   If  the  <b>--bam</b> file contains paired-end reads and if there are cases where the two mate
       pairs overlap with each other (i.e. have an insert size &lt; 0), then each of these  reads  contribute  less
       then  a  single  count towards the quantifications unless <b>--no-merge</b> is provided.  The following diagram,
       with two genes with overlapping exons and one paired-end read where both mate pairs are split  reads  and
       overlap with each other, illustrates how the quantification works:

                    x             x
                   / \           / \
        +---------+   +---------+   +---------+
        | Exon1|1 |   | Exon1|2 |   | Exon1|3 |      Gene1
        +---------+   +---------+   +---------+
                                  x
                                 / \
                   +------------+   +-------------+
                   |  Exon2|1   |   |   Exon2|2   |  Gene2
                   +------------+   +-------------+
                                     x
                                    / \
                      +------------+   +----+        RNAseq Read Mate1

                      |--a-||-b-||c|   |-d--||--e-|
                                     x
                                    / \
                            +------+   +----------+  RNAseq Read Mate2

         Left Mate1  = ((b * 0.5) + a) / (a + b + d)
         Right Mate1 = (d * 0.5)/(a + b + d)
         Left Mate2  = (b * 0.5)/(b + d + e)
         Right Mate2 = ((d * 0.5) + e)/(b + d + e)

         Exon1|2 = Left Mate1 + Left Mate2
         Exon1|3 = Right Mate1 + Right Mate2
         Exon2|1 = Left Mate1 + Left Mate2
         Exon2|2 = Right Mate2 + Right Mate2

         Gene1 = Exon1|2 + Exon1|3
         Gene2 = Exon2|1 + Exon2|2

       <b>The</b>  <b>quan</b>  <b>mode</b>  <b>in</b>  <b>version</b>  <b>1.2</b>  <b>and</b>  <b>above</b> <b>is</b> <b>not</b> <b>compatible</b> <b>with</b> <b>the</b> <b>quantifications</b> <b>generated</b> <b>by</b> <b>the</b>
       <b>previous</b> <b>versions.</b> <b>This</b> <b>due</b> <b>to</b> <b>bug</b> <b>fixes</b> <b>and</b> <b>slight</b> <b>adjustments</b> <b>to</b>  <b>the</b>  <b>way</b>  <b>we</b>  <b>quantify.</b>  <b>DO</b>  <b>NOT</b>  <b>MIX</b>
       <b>QUANTIFICATIONS</b> <b>GENERATED</b> <b>BY</b> <b>EARLIER</b> <b>VERSIONS</b> <b>OF</b> <b>QTLTOOLS</b> <b>WITH</b> <b>QUANTIFICATIONS</b> <b>FROM</b> <b>VERSION</b> <b>1.2</b> <b>AND</b> <b>ABOVE</b>
       <b>AS</b> <b>THIS</b> <b>WILL</b> <b>CREATE</b> <b>A</b> <b>BIAS</b> <b>IN</b> <b>YOUR</b> <b>DATASET.</b>

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>--gtf</b> <u>gene_annotation.gtf</u>
              Gene  annotations  in  GTF  format.   These  can be obtained from &lt;https://www.gencodegenes.org/&gt;.
              REQUIRED.

       <b>--bam</b> <b>[</b><u>in.bam</u>|<u>in.sam</u>|<u>in.cram</u><b>]</b>
              Sequence data in BAM/SAM/CRAM format sorted by chromosome and then position.  One sample  per  BAM
              file.  REQUIRED.

       <b>--out-prefix</b> <u>output</u>
              Output prefix.  REQUIRED.

       <b>--sample</b> <u>sample_name</u>
              The  sample  name  of the BAM file.  If not provided the sample name will be taken as the BAM file
              path.

       <b>--rpkm</b> Output RPKM values.

       <b>--tpm</b>  Output TPM values.

       <b>--xxhash</b>
              Rather than using the GTF file name to generate unique hash for the options used, use the hash  of
              the GTF file.

       <b>--no-hash</b>
              Do  not include a hash signifying the options used in the quantification in the output file names.
              NOT RECOMMENDED.

       <b>--gene-types</b> <u>gene_type</u> ...
              Only quantify these  gene  types.   Requires  gene_type  attribute  in  GTF.   It  will  also  use
              transcript_type if present.

       <b>--filter-mapping-quality</b> <u>integer</u>
              Minimum  mapping  quality  for  a  read  or  read pair to be considered.  Set this to only include
              uniquely mapped reads.  DEFAULT=10.

       <b>--filter-mismatch</b> <u>integer|float</u>
              Maximum mismatches allowed in a read.  If between 0 and 1 taken as the fraction  of  read  length.
              <b>Requires</b> <b>NM</b> <b>attribute</b> <b>in</b> <b>the</b> <b>BAM</b> <b>file.</b>  DEFAULT=OFF.

       <b>--filter-mismatch-total</b> <u>integer|float</u>
              Maximum  total  mismatches  allowed  in paired-reads.  If between 0 and 1 taken as the fraction of
              combined read length.  <b>Requires</b> <b>NM</b> <b>attribute</b> <b>in</b> <b>the</b> <b>BAM</b> <b>file.</b>  DEFAULT=OFF.

       <b>--filter-min-exon</b> <u>integer</u>
              Minimum length of an exon for it to be quantified.  Exons smaller than this will  not  be  printed
              out in the exon quantifications, but will still count towards gene quantifications.  DEFAULT=0.

       <b>--filter-remove-duplicates</b>
              Remove duplicate sequencing reads,as indicated by the aligner, in the process.  NOT RECOMMENDED.

       <b>--filter-failed-qc</b>
              Remove fastq reads that fail sequencing QC as indicated by the sequencer.

       <b>--check-proper-pairing</b>
              If  provided  only  properly paired reads according to the aligner that are in correct orientation
              will be considered.  Otherwise all pairs in correct orientation will be considered.

       <b>--check-consistency</b>
              If provided checks the consistency of split reads with annotation, rather than pure overlap of one
              of the blocks of the split read.

       <b>--no-merge</b>
              If provided overlapping mate pairs will not be merged.  Default behavior is to  merge  overlapping
              mate pairs based on the amount of overlap, such that each mate pair counts for less than 1 read.

       <b>--legacy-options</b>
              Exactly replicate Dermitzakis lab original quantification script.  DO NOT USE.

       <b>--region</b> <u>chr:start-end</u>
              Genomic region to be processed.  E.g. chr4:12334456-16334456, or chr5.

</pre><h4><b>OUTPUT</b> <b>FILES</b></h4><pre>
       Unless  <b>--no-hash</b> is provided, all output files will include a hash value corresponding to combination of
       the specific options used.  This is given so that one does not merge quantifications  from  samples  that
       were quantified differently, which would create a bias in the dataset.

       <b>.gene.count.bed</b> <b>.exon.count.bed</b> <b>.gene.rpkm.bed</b> <b>.exon.rpkm.bed</b> <b>.gene.tpm.bed</b> <b>.exon.tpm.bed</b>
        These are the quantification results files with the following columns:

        1   chr           Phenotype's chromosome
        2   start         Phenotype's start position (0-based)
        3   end           Phenotype's end position (1-based)
        4   gene|exon     The gene or exon ID.
        5   info|geneID   Information  about the gene or the gene ID of the exon.  The gene info is separated by
                          semicolons, and L=gene length, T=gene type, R=gene positions, N=gene name
        6   strand        Phenotype's strand
        7   sample_name   The sample name of the BAM file

       <b>.stats</b>
        Details the statistics of the quantification, with the following rows:

         1   filtered_secondary_alignments_(does_not_count_towards_total_reads)   Number of secondary alignments
         2   total_reads                                                          Number of  reads  in  the  BAM
                                                                                  file
         3   filtered_unmapped                                                    Number of unmapped reads
         4   filtered_failqc                                                      Number   of   reads  with  the
                                                                                  failed QC tag
         5   filtered_duplicate                                                   Number of duplicate reads
         6   filtered_mapQ_less_than_X                                            Number  of  reads  below   the
                                                                                  mapping quality threshold X
         7   filtered_notpaired                                                   Number  of pairs that were not
                                                                                  in the correct orientation  or
                                                                                  were not properly paired
         8   filtered_mismatches_greater_than_X_Y                                 Number  of  reads  failing the
                                                                                  mismatches per  read,  X,  and
                                                                                  mismatches total filters, Y
         9   filtered_unmatched_mate_pairs                                        Number  of  reads  where there
                                                                                  was  a  paired-read   with   a
                                                                                  missing mate
        10   total_good                                                           Number  of  reads  that passed
                                                                                  all filters
        11   total_exonic                                                         Number of reads  that  aligned
                                                                                  to   exons   and   passed  all
                                                                                  filters
        12   total_exonic_multi_counting                                          Number of reads  that  aligned
                                                                                  to  exons  when we count reads
                                                                                  that align to  multiple  exons
                                                                                  multiple times
        13   total_merged_reads                                                   Number of reads where the mate
                                                                                  pairs   were  overlapping  and
                                                                                  thus were merged
        14   total_exonic_multi_counting_after_merge_(used_for_rpkm)              Number of reads  that  aligned
                                                                                  to   exons   when   we   merge
                                                                                  overlapping mate pairs
        15   good_over_total                                                      Number of good reads over  the
                                                                                  total number of reads
        16   exonic_over_total                                                    Number  of  exonic  reads over
                                                                                  the total number of reads
        17   exonic_over_good                                                     Number of  exonic  reads  over
                                                                                  the number of good reads

</pre><h4><b>EXAMPLE</b></h4><pre>
       o Quantifying  a  sample  mapped with GEM, outputting TPM and RPKM values, and taking the hash of the GTF
         file:

         QTLtools quan --bam HG00381.chr22.bam --gtf  gencode.v19.annotation.chr22.gtf.gz  --out-prefix  HG00381
         --sample HG00381 --rpkm --tpm --xxhash --filter-mismatch-total 8 --filter-mapping-quality 150

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <u><a href="../man1/QTLtools.1.html">QTLtools</a></u>(1)

       QTLtools website: &lt;https://qtltools.github.io/qtltools&gt;

</pre><h4><b>BUGS</b></h4><pre>
       Please submit bugs to &lt;https://github.com/qtltools/qtltools&gt;

</pre><h4><b>CITATION</b></h4><pre>
       Delaneau,  O.,  Ongen, H., Brown, A. et al. A complete tool set for molecular QTL discovery and analysis.
       <u>Nat</u> <u>Commun</u> <b>8</b>, 15452 (2017).  &lt;https://doi.org/10.1038/ncomms15452&gt;

</pre><h4><b>AUTHORS</b></h4><pre>
       Halit Ongen (<a href="mailto:halitongen@gmail.com">halitongen@gmail.com</a>), Olivier Delaneau (<a href="mailto:olivier.delaneau@gmail.com">olivier.delaneau@gmail.com</a>)

QTLtools-v1.3                                      06 May 2020                                  <u><a href="../man1/QTLtools-quan.1.html">QTLtools-quan</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>