<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>emma - Multiple sequence alignment (ClustalW wrapper)</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/emboss">emboss_6.6.0+dfsg-15ubuntu2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       emma - Multiple sequence alignment (ClustalW wrapper)

</pre><h4><b>SYNOPSIS</b></h4><pre>

       <b>emma</b> <b>-sequence</b> <u>seqall</u> [<b>-onlydend</b> <u>toggle</u>] <b>-dend</b> <u>toggle</u> <b>-dendfile</b> <u>infile</u> [<b>-slow</b> <u>toggle</u>] <b>-pwmatrix</b> <u>list</u>
            <b>-pwdnamatrix</b> <u>list</u> <b>-usermatrix</b> <u>variable</u> <b>-pairwisedatafile</b> <u>infile</u> <b>-matrix</b> <u>list</u> <b>-usermamatrix</b> <u>variable</u>
            <b>-dnamatrix</b> <u>list</u> <b>-umamatrix</b> <u>variable</u> <b>-mamatrixfile</b> <u>infile</u> <b>-pwgapopen</b> <u>float</u> <b>-pwgapextend</b> <u>float</u>
            <b>-ktup</b> <u>integer</u> <b>-gapw</b> <u>integer</u> <b>-topdiags</b> <u>integer</u> <b>-window</b> <u>integer</u> <b>-nopercent</b> <u>boolean</u> [<b>-gapopen</b> <u>float</u>]
            [<b>-gapextend</b> <u>float</u>] [<b>-endgaps</b> <u>boolean</u>] [<b>-gapdist</b> <u>integer</u>] <b>-norgap</b> <u>boolean</u> <b>-hgapres</b> <u>string</u>
            <b>-nohgap</b> <u>boolean</u> [<b>-maxdiv</b> <u>integer</u>] <b>-outseq</b> <u>seqoutset</u> <b>-dendoutfile</b> <u>outfile</u>

       <b>emma</b> <b>-help</b>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>emma</b> is a command line program from EMBOSS (“the European Molecular Biology Open Software Suite”). It is
       part of the "Alignment:Multiple" command group(s).

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Input</b> <b>section</b>
       <b>-sequence</b> <u>seqall</u>

       <b>-onlydend</b> <u>toggle</u>
           Default value: N

       <b>-dend</b> <u>toggle</u>
           Default value: N

       <b>-dendfile</b> <u>infile</u>

       <b>-slow</b> <u>toggle</u>
           A distance is calculated between every pair of sequences and these are used to construct the
           dendrogram which guides the final multiple alignment. The scores are calculated from separate
           pairwise alignments. These can be calculated using 2 methods: dynamic programming (slow but accurate)
           or by the method of Wilbur and Lipman (extremely fast but approximate). The slow-accurate method is
           fine for short sequences but will be VERY SLOW for many (e.g. &gt;100) long (e.g. &gt;1000 residue)
           sequences. Default value: Y

   <b>Pairwise</b> <b>align</b> <b>options</b>
       <b>-pwmatrix</b> <u>list</u>
           The scoring table which describes the similarity of each amino acid to each other. There are three
           'in-built' series of weight matrices offered. Each consists of several matrices which work
           differently at different evolutionary distances. To see the exact details, read the documentation.
           Crudely, we store several matrices in memory, spanning the full range of amino acid distance (from
           almost identical sequences to highly divergent ones). For very similar sequences, it is best to use a
           strict weight matrix which only gives a high score to identities and the most favoured conservative
           substitutions. For more divergent sequences, it is appropriate to use 'softer' matrices which give a
           high score to many other frequent substitutions. 1) BLOSUM (Henikoff). These matrices appear to be
           the best available for carrying out data base similarity (homology searches). The matrices used are:
           Blosum80, 62, 45 and 30. 2) PAM (Dayhoff). These have been extremely widely used since the late '70s.
           We use the PAM 120, 160, 250 and 350 matrices. 3) GONNET . These matrices were derived using almost
           the same procedure as the Dayhoff one (above) but are much more up to date and are based on a far
           larger data set. They appear to be more sensitive than the Dayhoff series. We use the GONNET 40, 80,
           120, 160, 250 and 350 matrices. We also supply an identity matrix which gives a score of 1.0 to two
           identical amino acids and a score of zero otherwise. This matrix is not very useful. Default value: b

       <b>-pwdnamatrix</b> <u>list</u>
           The scoring table which describes the scores assigned to matches and mismatches (including IUB
           ambiguity codes). Default value: i

       <b>-usermatrix</b> <u>variable</u>

       <b>-pairwisedatafile</b> <u>infile</u>

   <b>Matrix</b> <b>options</b>
       <b>-matrix</b> <u>list</u>
           This gives a menu where you are offered a choice of weight matrices. The default for proteins is the
           PAM series derived by Gonnet and colleagues. Note, a series is used! The actual matrix that is used
           depends on how similar the sequences to be aligned at this alignment step are. Different matrices
           work differently at each evolutionary distance. There are three 'in-built' series of weight matrices
           offered. Each consists of several matrices which work differently at different evolutionary
           distances. To see the exact details, read the documentation. Crudely, we store several matrices in
           memory, spanning the full range of amino acid distance (from almost identical sequences to highly
           divergent ones). For very similar sequences, it is best to use a strict weight matrix which only
           gives a high score to identities and the most favoured conservative substitutions. For more divergent
           sequences, it is appropriate to use 'softer' matrices which give a high score to many other frequent
           substitutions. 1) BLOSUM (Henikoff). These matrices appear to be the best available for carrying out
           data base similarity (homology searches). The matrices used are: Blosum80, 62, 45 and 30. 2) PAM
           (Dayhoff). These have been extremely widely used since the late '70s. We use the PAM 120, 160, 250
           and 350 matrices. 3) GONNET . These matrices were derived using almost the same procedure as the
           Dayhoff one (above) but are much more up to date and are based on a far larger data set. They appear
           to be more sensitive than the Dayhoff series. We use the GONNET 40, 80, 120, 160, 250 and 350
           matrices. We also supply an identity matrix which gives a score of 1.0 to two identical amino acids
           and a score of zero otherwise. This matrix is not very useful. Alternatively, you can read in your
           own (just one matrix, not a series). Default value: b

       <b>-usermamatrix</b> <u>variable</u>

       <b>-dnamatrix</b> <u>list</u>
           This gives a menu where a single matrix (not a series) can be selected. Default value: i

       <b>-umamatrix</b> <u>variable</u>

       <b>-mamatrixfile</b> <u>infile</u>

   <b>Additional</b> <b>section</b>
   <b>Slow</b> <b>align</b> <b>options</b>
       <b>-pwgapopen</b> <u>float</u>
           The penalty for opening a gap in the pairwise alignments. Default value: 10.0

       <b>-pwgapextend</b> <u>float</u>
           The penalty for extending a gap by 1 residue in the pairwise alignments. Default value: 0.1

   <b>Fast</b> <b>align</b> <b>options</b>
       <b>-ktup</b> <u>integer</u>
           This is the size of exactly matching fragment that is used. INCREASE for speed (max= 2 for proteins;
           4 for DNA), DECREASE for sensitivity. For longer sequences (e.g. &gt;1000 residues) you may need to
           increase the default. Default value: @($(acdprotein)?1:2)

       <b>-gapw</b> <u>integer</u>
           This is a penalty for each gap in the fast alignments. It has little affect on the speed or
           sensitivity except for extreme values. Default value: @($(acdprotein)?3:5)

       <b>-topdiags</b> <u>integer</u>
           The number of k-tuple matches on each diagonal (in an imaginary dot-matrix plot) is calculated. Only
           the best ones (with most matches) are used in the alignment. This parameter specifies how many.
           Decrease for speed; increase for sensitivity. Default value: @($(acdprotein)?5:4)

       <b>-window</b> <u>integer</u>
           This is the number of diagonals around each of the 'best' diagonals that will be used. Decrease for
           speed; increase for sensitivity. Default value: @($(acdprotein)?5:4)

       <b>-nopercent</b> <u>boolean</u>
           Default value: N

   <b>Gap</b> <b>options</b>
       <b>-gapopen</b> <u>float</u>
           The penalty for opening a gap in the alignment. Increasing the gap opening penalty will make gaps
           less frequent. Default value: 10.0

       <b>-gapextend</b> <u>float</u>
           The penalty for extending a gap by 1 residue. Increasing the gap extension penalty will make gaps
           shorter. Terminal gaps are not penalised. Default value: 5.0

       <b>-endgaps</b> <u>boolean</u>
           End gap separation: treats end gaps just like internal gaps for the purposes of avoiding gaps that
           are too close (set by 'gap separation distance'). If you turn this off, end gaps will be ignored for
           this purpose. This is useful when you wish to align fragments where the end gaps are not biologically
           meaningful. Default value: Y

       <b>-gapdist</b> <u>integer</u>
           Gap separation distance: tries to decrease the chances of gaps being too close to each other. Gaps
           that are less than this distance apart are penalised more than other gaps. This does not prevent
           close gaps; it makes them less frequent, promoting a block-like appearance of the alignment. Default
           value: 8

       <b>-norgap</b> <u>boolean</u>
           Residue specific penalties: amino acid specific gap penalties that reduce or increase the gap opening
           penalties at each position in the alignment or sequence. As an example, positions that are rich in
           glycine are more likely to have an adjacent gap than positions that are rich in valine. Default
           value: N

       <b>-hgapres</b> <u>string</u>
           This is a set of the residues 'considered' to be hydrophilic. It is used when introducing Hydrophilic
           gap penalties. Default value: GPSNDQEKR

       <b>-nohgap</b> <u>boolean</u>
           Hydrophilic gap penalties: used to increase the chances of a gap within a run (5 or more residues) of
           hydrophilic amino acids; these are likely to be loop or random coil regions where gaps are more
           common. The residues that are 'considered' to be hydrophilic are set by '-hgapres'. Default value: N

       <b>-maxdiv</b> <u>integer</u>
           This switch, delays the alignment of the most distantly related sequences until after the most
           closely related sequences have been aligned. The setting shows the percent identity level required to
           delay the addition of a sequence; sequences that are less identical than this level to any other
           sequences will be aligned later. Default value: 30

   <b>Output</b> <b>section</b>
       <b>-outseq</b> <u>seqoutset</u>

       <b>-dendoutfile</b> <u>outfile</u>

</pre><h4><b>BUGS</b></h4><pre>
       Bugs can be reported to the Debian Bug Tracking system (<a href="http://bugs.debian.org/emboss">http://bugs.debian.org/emboss</a>), or directly to
       the EMBOSS developers (<a href="http://sourceforge.net/tracker/">http://sourceforge.net/tracker/</a>?group_id=93650&amp;atid=605031).

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       emma is fully documented via the <b><a href="../man1/tfm.1.html">tfm</a></b>(1) system.

</pre><h4><b>AUTHOR</b></h4><pre>
       <b>Debian</b> <b>Med</b> <b>Packaging</b> <b>Team</b> &lt;<a href="mailto:debian-med-packaging@lists.alioth.debian.org">debian-med-packaging@lists.alioth.debian.org</a>&gt;
           Wrote the script used to autogenerate this manual page.

</pre><h4><b>COPYRIGHT</b></h4><pre>
       This manual page was autogenerated from an Ajax Control Definition of the EMBOSS package. It can be
       redistributed under the same terms as EMBOSS itself.

EMBOSS 6.4.0                                       05/11/2012                                           <u><a href="../man1e/EMMA.1e.html">EMMA</a></u>(1e)
</pre>
 </div>
</div></section>
</div>
</body>
</html>