<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>sam-stats - ea-utils: produce digested statistics</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/ea-utils">ea-utils_1.1.2+dfsg-9build2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       sam-stats - ea-utils: produce digested statistics

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>sam-stats</b> [<u>options</u>] [<u>file1</u>] [<u>file2</u>...<u>filen</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Version: 1.38.681

       Produces lots of easily digested statistics for the files listed

       Options (default in parens):

       <b>-D</b>              Keep  track  of multiple alignments <b>-O</b> PREFIX      Output prefix enabling extended output
       (see below) <b>-R</b> FIL         Coverage/RNA output (coverage, 3' bias, etc, implies <b>-A</b>) <b>-A</b>             Report
       all chr sigs, even if there are more than 1000 <b>-b</b> INT         Number of  reads  to  sample  for  per-base
       stats  (1M)  <b>-S</b>  INT          Size  of  ascii-signature  (30)  <b>-x</b> FIL         File extension for handling
       multiple files (stats) <b>-M</b>             Only overwrite  if  newer  (requires  <b>-x</b>,  or  multiple  files)  <b>-B</b>
       Input is bam, don't bother looking at magic <b>-z</b>             Don't fail when zero entries in sam

       OUTPUT:

       If one file is specified, then the output is to standard out.  If multiple files are specified, or if the
       <b>-x</b> option is supplied, the output file is &lt;filename&gt;.&lt;ext&gt;.  Default extension is 'stats'.

       Complete Stats:

       &lt;STATS&gt;
              : mean, max, stdev, median, Q1 (25 percentile), Q3

       reads  : # of entries in the sam file, might not be # reads

       phred  : phred scale used

       bsize  : # reads used for qual stats

       mapped reads
              : number of aligned reads (unique probe id sequences)

       mapped bases
              : total of the lengths of the aligned reads

       forward
              : number of forward-aligned reads

       reverse
              : number of reverse-aligned reads

       snp rate
              : mismatched bases / total bases (snv rate)

       ins rate
              : insert bases / total bases

       del rate
              : deleted bases / total bases

       pct mismatch
              : percent of reads that have mismatches

       pct align
              : percent of reads that aligned

       len &lt;STATS&gt;
              : read length stats, ignored if fixed-length

       mapq &lt;STATS&gt;
              : stats for mapping qualities

       insert &lt;STATS&gt;
              : stats for insert sizes

       %&lt;CHR&gt; : percentage of mapped bases per chr, followed by a signature

   <b>Subsampled</b> <b>stats</b> <b>(1M</b> <b>reads</b> <b>max):</b>
              base qual &lt;STATS&gt; : stats for base qualities %A,%T,%C,%G       : base percentages

   <b>Meaning</b> <b>of</b> <b>the</b> <b>per-chromosome</b> <b>signature:</b>
              A  ascii-histogram  of  mapped  reads  by  chromosome position.  It is only output if the original
              SAM/BAM has a header. The values are the log2 of the # of mapped reads at each  position  +  ascii
              '0'.

   <b>Extended</b> <b>output</b> <b>mode</b> <b>produces</b> <b>a</b> <b>set</b> <b>of</b> <b>files:</b>
       .stats : primary output

       .fastx : fastx-toolkit compatible output

       .rcov  : per-reference counts &amp; coverage

       .xdist : mismatch distribution

       .ldist : length distribution (if applicable)

       .mqdist
              : mapping quality distribution

sam-stats 1.1.2                                     July 2015                                       <u><a href="../man1/SAM-STATS.1.html">SAM-STATS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>