<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>arden-create - generate an artificial reference genome (AR) from a given input</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/arden">arden_1.0-6_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       arden-create - generate an artificial reference genome (AR) from a given input

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>arden-create</b> [<u>options</u>] [<u>OUTPUTFOLDER</u>] [<u>INPUT</u> <u>FASTA</u>]...

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Script  to generate an artificial reference genome (AR) from a given input. The AR fulfills the following
       conditions by default settings:

       1   has a nucleotide distribution equal to the input genome

       2   has an amino acid (aa) distribution equal to the input genome

       3   keeps the aa neighborhood similar to the neighborhood in input genome

       4   protects start and stop codons from mutations

</pre><h4><b>OPTIONS</b></h4><pre>
       Required Arguments:

       <u>OUTPUTFOLDER</u>
              complete path to output destination folder

       <u>INPUT</u> <u>FASTA</u>
              Single sequence Fasta file.

       <b>--version</b>
              show program's version number and exit

       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>-d</b> DISTANCE, <b>--distance</b>=<u>DISTANCE</u>
              distance between 2 mutations on DNA level. The minimum distance will then be d-3. [default: 15]

       <b>-o</b> ORF, <b>--orf</b>=<u>ORF</u>
              1- protect ORF structure from mutations.   0 - allow mutations in start / stop codon [default: 1]

       <b>-r</b> REVSUB, <b>--revsub</b>=<u>REVSUB</u>
              1 - reverse substitution if no suitable counterpart was found (balanced mutation). 0  -  keep  the
              unbalanced mutations.[default: 1]

       <b>-m</b> RANDOM, <b>--random</b>=<u>RANDOM</u>
              1/0 variable. 1 - shuffled starting positions for the

              mutations. 0 - linear mutation [default: 1]

       <b>-s</b> SORF, <b>--saveorfs</b>=<u>SORF</u>
              Save  found start and stop codons in a pickle file (can be loaded if the input fasta is used again
              if a filename is specified). [default: ]

       <b>-l</b> LORF, <b>--loadorfs</b>=<u>LORF</u>
              Specify filename to a previously created pickle file (contains  positions  of  start/stop  codons)
              [default: ]

       <b>-n</b> NAME, <b>--name</b>=<u>NAME</u>
              Specify name which will be used as header. [default: ]

       <b>-p</b> PEXAMPLES, <b>--pexamples</b>=<u>PEXAMPLES</u>
              set to 1 if you want to print examples

arden-create 2013-04-25                           February 2014                                  <u><a href="../man1/ARDEN-CREATE.1.html">ARDEN-CREATE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>