<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>ucsc_genes2gff.pl - Convert UCSC Genome Browser-format gene files into GFF files suitable for loading</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/gbrowse">gbrowse_2.56+dfsg-12build4_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       ucsc_genes2gff.pl - Convert UCSC Genome Browser-format gene files into GFF files suitable for loading
       into gbrowse

</pre><h4><b>SYNOPSIS</b></h4><pre>
         % uscsc_genes2gff.pl [options] ucsc_file1 ucsc_file2...

       Options:

           -src    &lt;string&gt;   Choose a source for the gene, default "UCSC"
           -origin &lt;integer&gt;  Choose a relative position to number from, default
       is "1"

</pre><h4><b>DESCRIPTION</b></h4><pre>
       This script massages the gene files available from the "tables" link of the UCSC genome browser
       (genome.ucsc.edu) into a form suitable for loading of gbrowse.  Warning: it only works with the gene
       tables.  Other tables, such as EST alignments, contours and repeats, have their own formats which will
       require other scripts to parse.

       To use this script, get one or more UCSC tables, either from the "Tables" link on the browser, or from
       the UCSC Genome Browser FTP site.  Give the table file as the argument to this script.  You may want to
       provide an alternative "source" field.  Otherwise this script defaults to "UCSC".

         % pucsc_genes2gff.pl -src RefSeq refseq_data.ucsc &gt; refseq.gff

       The resulting GFF file can then be loaded into a Bio::DB::GFF database using the following command:

         % bulk_load_gff.pl -d &lt;databasename&gt; refseq.gff

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Bio::DB::GFF, bulk_load_gff.pl, load_gff.pl

</pre><h4><b>AUTHOR</b></h4><pre>
       Lincoln Stein &lt;<a href="mailto:lstein@cshl.org">lstein@cshl.org</a>&gt;.

       Copyright (c) 2003 Cold Spring Harbor Laboratory

       This library is free software; you can redistribute it and/or modify it under the same terms as Perl
       itself.  See DISCLAIMER.txt for disclaimers of warranty.

perl v5.40.0                                       2024-10-20                                       <u><a href="../man1p/BED2GFF3.1p.html">BED2GFF3</a></u>(1p)
</pre>
 </div>
</div></section>
</div>
</body>
</html>