<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>clean_genes - Given a GFF describing a set of genes and a corresponding</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/phast">phast_1.7+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       clean_genes - Given a GFF describing a set of genes and a corresponding

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Given  a GFF describing a set of genes and a corresponding multiple alignment, output a new GFF with only
       those genes that meet certain  "cleanliness"  criteria.  The  coordinates  in  the  GFF  are  assumed  to
       correspond  to  the  reference  sequence  in  the alignment, which is assumed to be the first one listed.
       Default behavior is simply to require that all annotated start/stop codons and splice sites are valid  in
       the  reference  sequence  (GT-AG,  GC-AG,  and AT-AC splice sites are allowed).  This can be used with an
       "alignment" consisting of a single sequence to filter out incorrect annotations.  Options  are  available
       to  impose  additional  criteria  as  well, mostly having to do with conservation across species (see the
       '--conserved' option in particular).

</pre><h4><b>SYNOPSIS</b></h4><pre>
       clean_genes [options] &lt;gff_fname&gt; &lt;msa_fname&gt;

</pre><h4><b>OPTIONS</b></h4><pre>

       <b>--start</b>, <b>-s</b>

              Require conserved start codons (all species)

       <b>--stop</b>, <b>-t</b>

              Require conserved stop codons (all species)

       <b>--splice</b>, <b>-l</b>

       Require conserved splice sites (all species).
              By default,

              only GT-AG, GC-AG, and AT-AC splice sites are allowed (see also <b>--splice-strict</b>)

       <b>--fshift</b>, <b>-f</b>

       Require that no frame-shift gap is present in any species.
              Frame

       shifts are evaluated with respect to the reference sequence.
              Gaps

              that have non-multiple-of-three lengths are allowed if compensatory gaps occur nearby (see  source
              code for details).

       <b>--nonsense</b>, <b>-n</b>

              Require that no premature stop codon is present in any species.

       <b>--conserved</b>, <b>-c</b>

              Implies <b>--start</b>, <b>--stop</b>, <b>--splice</b>, <b>--fshift</b>, and <b>--nonsense</b>.  Recommended option for cross-species
              analysis.

       <b>--N-limit</b>, <b>-N</b> &lt;f&gt;

              Maximum  fraction  of  bases aligned to CDSs that are Ns in any species (&lt;f&gt; must be between 0 and
              1).  Default is 0.05.  Set to 1 to allow any number of Ns.

       <b>--clean-gaps</b>, <b>-e</b>

       Require all cds gaps to be multiples of three in length.
              Can be

              used with <b>--conserved</b>.

       <b>--indel-strict</b>, <b>-I</b>

       Implies <b>--clean_gaps</b>, usually used with <b>--conserved</b>.
              Prohibits

              overlapping cds gaps in different sequences, gaps near cds boundaries, and gaps in  the  reference
              sequence  within  and  between  flanking  features  (splice  sites,  etc.;  see code for details).
              Designed for use in training a phylo-HMM with an indel model.

       <b>--splice-strict</b>, <b>-C</b>

       Implies <b>--splice</b>.
              Allow only GT-AG canonical splice sites.  Useful

              when training a gene finder with a simple model for splice sites.

       <b>--groupby</b>, <b>-g</b> &lt;tag&gt;

              Group features according to specified tag (default "transcript_id").   If  any  feature  within  a
              group  fails,  the  entire  group  will  be discarded.  By choosing to group features according to
              different criteria, you can make the program "clean"  the  data  set  at  different  levels.   For
              example,  to  clean  at  the level of individual exons, add a tag like "exon_id" to indicate exons
              (see the program "refeature"), and then invoke clean_genes with "--groupby exon_id".

       <b>--msa-format</b>, <b>-i</b> FASTA|PHYLIP|MPM|MAF|SS

       Alignment file format.
              Default is to guess format from file

              contents.

       <b>--refseq</b>, <b>-r</b> &lt;seqfile.fa&gt;

       (Required with <b>--msa-format</b> MAF)
              Complete reference

              sequence for alignment (FASTA format).

       <b>--offset5</b>, <b>-o</b> &lt;n&gt;

       (Default 0)
              Offset of canonical "GT" with respect to boundary

       on *intron side* of annotated 5' splice sites.
              Useful with

              annotations that describe a window around the canonical splice site.

       <b>--offset3</b>, <b>-p</b> &lt;n&gt;

       (Default 0)
              Offset of canonical "AG" with respect to boundary

              on intron side of annotated 3' splice sites.

       <b>--log</b>, <b>-L</b> &lt;fname&gt;

              Write human-readable log to specified file.

       <b>--machine-log</b>, <b>-M</b> &lt;fname&gt;

              Like <b>--log</b>, but produces more concise, machine-readable log.

       <b>--stats</b>, <b>-S</b> &lt;fname&gt;

              Write statistics on retained and discarded features to specified file.

       <b>--discards</b>, <b>-d</b> &lt;fname&gt;

              Write discarded features to specified file.

       <b>--no-output</b>, <b>-x</b>

       Suppress output of "cleaned" features to stdout.
              Useful if only

              log file and/or stats are of interest.

       <b>--help</b>, <b>-h</b>

              Print this help message.

       NOTES:  Feature types are defined as follows.

       coding exon
              &lt;-&gt; "CDS"

       start codon
              &lt;-&gt; "start_codon"

       stop codon
              &lt;-&gt; "stop_codon"

              5' splice site &lt;-&gt; "5'splice" 3' splice site &lt;-&gt; "3'splice"

              In  addition,  splice  sites  in  UTR  can  be  separately  designated   as   "5'splice_utr"   and
              "3'splice_utr".   Errors in these sites will be given a different code in the log files, which can
              be useful for tracking purposes.

              If evaluation is done at the level of individual exons (see  <b>--groupby</b>),  then  splice  sites  are
              considered independently rather than in the context of introns.  As a result, the exons flanking a
              GT-AC or AT-AG intron might (misleadingly) be considered "clean".

              With  <b>--fshift</b> and <b>--nonsense</b>, it is possible for entries to pass through that have stop codons in
              the frame of the *reference* sequence, although they do not have any  in  their  own  frame.   Use
              <b>--clean-gaps</b>  instead  to  guarantee that no stop codons occur in any sequence in the frame of the
              reference sequence.

clean_genes 1.4                                     May 2016                                      <u><a href="../man1/CLEAN_GENES.1.html">CLEAN_GENES</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>