<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>cnvkit_import-rna - Convert a cohort of per-gene log2 ratios to CNVkit .cnr format.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/cnvkit">cnvkit_0.9.12-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       cnvkit_import-rna - Convert a cohort of per-gene log2 ratios to CNVkit .cnr format.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       usage: cnvkit.py import-rna [-h] [-f NAME] <b>-g</b> FILE [-c FILE]

       [--max-log2 FLOAT] [-n NORMAL [NORMAL ...]]
              [-d PATH] [-o FILE] [--no-gc] [--no-txlen] FILES [FILES ...]

   <b>positional</b> <b>arguments:</b>
       FILES  Tabular  files  with Ensembl gene ID and number of reads mapped to each gene, from RSEM or another
              transcript quantifier.

   <b>options:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>-f</b> NAME, <b>--format</b> NAME
              Input format name: 'rsem' for RSEM gene-level read counts (*_rsem.genes.results), or 'counts'  for
              generic 2-column gene IDs and their read counts (e.g. TCGA level 2 RNA expression).

       <b>-g</b> FILE, <b>--gene-resource</b> FILE
              Location of gene info table from Ensembl BioMart.

       <b>-c</b> FILE, <b>--correlations</b> FILE
              Correlation of each gene's copy number with expression. Output of cnv_expression_correlate.py.

       <b>--max-log2</b> FLOAT
              Maximum  log2  ratio in output. Observed values above this limit will be replaced with this value.
              [Default: 3.0]

       <b>-n</b> NORMAL [NORMAL ...], <b>--normal</b> NORMAL [NORMAL ...]
              Normal samples (same format as `gene_counts`) to be used as a  control  to  when  normalizing  and
              re-centering gene read depth ratios. All filenames following this option will be used.

       <b>-d</b> PATH, <b>--output-dir</b> PATH
              Directory to write a CNVkit .cnr file for each input sample. [Default: .]

       <b>-o</b> FILE, <b>--output</b> FILE
              Output file name (summary table).

   <b>To</b> <b>disable</b> <b>specific</b> <b>automatic</b> <b>bias</b> <b>corrections:</b>
       <b>--no-gc</b>
              Skip GC correction.

       <b>--no-txlen</b>
              Skip transcript length correction.

cnvkit.py import-rna 0.9.10                         July 2023                               <u><a href="../man1/CNVKIT_IMPORT-RNA.1.html">CNVKIT_IMPORT-RNA</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>