<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>eval_predictions - Compares predicted genes with "real" (or annotated) genes.</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/phast">phast_1.7+dfsg-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       eval_predictions - Compares predicted genes with "real" (or annotated) genes.

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>eval_predictions</b> <b>-r</b> &lt;real_fname_list&gt; <b>-p</b> &lt;pred_fname_list&gt; <b>-l</b> &lt;seq_len_list&gt; [OPTIONS]

</pre><h4><b>DESCRIPTION:</b></h4><pre>
       Compares  predicted  genes  with  "real"  (or  annotated) genes.  Reports standard measures of prediction
       quality.  The following measures are reported:

              - nucleotide sensitivity (Sn)

              - nucleotide specificity (Sp)

              - approximate correlation (AC)

              - correlation coefficient (CC)

              - exon sensitivity (ESn)

              - exon specificity (ESp)

              - proportion of real exons correctly predicted (CRa)

              - proportion of real exons partially predicted (PCa)

              - proportion of real exons with overlapping predictions (OLa)

              - missed exons (ME)

              - proportion of predicted exons that are correct (CRp)

              - proportion of predicted exons that are partially correct (PCp)

              - proportion of predicted exons that overlap real ones (OLp)

              - wrong exons (WE)

       All quantities are computed as described in "Evaluation of Gene-Finding Programs on Mammalian Sequences,"
       by Rogic et al. (Genome Research 11:817-832).  Note that CRa + PCa + OLa + ME = 1 and CRp + PCp +  OLp  +
       WE  =  1.   Note  also  that  each  set  (predicted and real) should consist of non-overlapping groups of
       features (see 'refeature').

</pre><h4><b>OPTIONS</b></h4><pre>

       <b>-r</b> &lt;real_fname_list&gt;

              (required) List of names of files defining real genes (GFF).

       <b>-p</b> &lt;pred_fname_list&gt;

              (required) List of names of files defining predicted genes (GFF).  Must  correspond  in  order  to
              &lt;real_fname_list&gt;.

       <b>-l</b> &lt;seq_len_list&gt; (required) List of lengths of sequences.  Needed to compute

              certain nucleotide-level statistics.

       <b>-f</b> &lt;feat_list&gt;

       List of names of all features denoting exon regions.
              By

              default, equal to the single name "CDS".

       <b>-d</b> &lt;fname_prefix&gt; Dump full coords of correct, partially correct, wrong, missed, and overlapping exons to
              a set of files having the specified file name prefix.

       <b>-n</b>  &lt;nbases&gt;  Also  report stats on "nearly correct" exons, that is, incorrect exons whose boundaries are
              within &lt;nbases&gt; of being correct.  Columns will be labeled "NCa" and "NCp".

       <b>-h</b> Print this help message.

       NOTE: be sure stop codons are included in CDSs in both the predicted and real sets, or in neither set.

eval_predictions 1.4                                May 2016                                 <u><a href="../man1/EVAL_PREDICTIONS.1.html">EVAL_PREDICTIONS</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>