<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>GBrowse2 - The Generic Genome Browser</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/gbrowse">gbrowse_2.56+dfsg-12build4_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       GBrowse2 - The Generic Genome Browser

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>libgbrowse-perl</b> [-h]

</pre><h4><b>INSTALLATION</b></h4><pre>
       If  Apache2  is  installed  on  the  system,  the  software  has  configured  it  to link URLs to GBrowse
       directories. A restart of Apache is required to take it  into  account.  Configuration  is  available  in
       /etc/gbrowse2/apache2.conf file.
        If another web server is used, one should refer to the apache2 template to map urls in the same way.

</pre><h4><b>DEPENDENCIES</b></h4><pre>
       Additional  optional  perl  dependencies  exist  to provide additional features. Please refer to Optional
       modules in INSTALL file for more information.

</pre><h4><b>CONFIGURATION</b></h4><pre>
       <b>gbrowse2</b> Configuration files are located in /etc/gbrowse. Main configuration  file  is  GBrowse.conf.  It
       contains  the  global  configuration  elements.  Specific  configuration  files  are  located in the same
       directory per data bank. For more details, one should refer to the GBrowse documentation.
        Installation is provided with a Yeast genome sample.

</pre><h4><b>DESCRIPTION</b></h4><pre>
       GBrowse is a simple but highly
        configurable web-based genome browser. It is a component of the
        Generic Model Organism Systems Database project (GMOD).
        Some of its features:
         * Simultaneous bird's eye and detailed views of the genome;
         * Scroll, zoom, center;
         * Attach arbitrary URLs to any annotation;
         * Order and appearance of tracks are customizable by administrator and
           end-user;
         * Search by annotation ID, name, or comment;
         * Supports third party annotation using GFF formats;
         * Settings persist across sessions;
         * DNA and GFF dumps;
         * Connectivity to different databases, including BioSQL and Chado;
         * Multi-language support;
         * Third-party feature loading;
         * Customizable plug-in architecture (e.g. run BLAST, dump &amp; import many
           formats, find oligonucleotides, design primers, create restriction maps,
           edit features).

</pre><h4><b>WEB</b> <b>ACCESS</b></h4><pre>
       GBrowse is accessible at URL <a href="http://localhost/gbrowse2">http://localhost/gbrowse2</a>

</pre><h4><b>AUTHOR</b></h4><pre>
       Olivier Sallou (olivier.sallou (at) irisa.fr) - Man page and packaging Lincoln Stein - GBrowse

version 1.1.0                                   February 07, 2011                                    <u><a href="../man1/GBrowse2.1.html">GBrowse2</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>