<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>gmap_build - Tool for genome database creation for GMAP or GSNAP</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/gmap">gmap_2024-11-20+ds-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       gmap_build - Tool for genome database creation for GMAP or GSNAP

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>gmap_build</b> [<u>options</u>...] <u>-d</u> <u>&lt;genome&gt;</u> [<u>-c</u> <u>&lt;transcriptome&gt;</u> <u>-T</u> <u>&lt;transcript_fasta&gt;</u>] <u>&lt;genome_fasta_files&gt;</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       gmap_build:  Builds  a  gmap  database  for  a genome to be used by GMAP or GSNAP.  Part of GMAP package,
       version 2024-11-20.

       You are free to name &lt;genome&gt; and &lt;transcriptome&gt; as  you  wish.   You  will  use  the  same  names  when
       performing alignments subsequently using GMAP or GSNAP.

       Note:  If  adding  a  transcriptome  to  an  existing  genome,  then  there  is  no  need  to specify the
       genome_fasta_files.  This way you can add transcriptome information to an existing genome database.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-D</b>, <b>--genomedir</b>=<u>STRING</u>
              Destination directory for installation (defaults to gmapdb directory specified at configure time)

       <b>-d</b>, <b>--genomedb</b>=<u>STRING</u>
              Genome name (required)

       <b>-n</b>, <b>--names</b>=<u>STRING</u>
              Substitute names for contigs, provided in a file.

              The file can have two formats:

       1.     A file with one column per line, with each line corresponding to a FASTA file, in the order  given
              to  gmap_build.   The  chromosome  name  for  each  FASTA  file  will be replaced with the desired
              chromosome name in the file.  Every chromosome in the FASTA must have a corresponding line in  the
              file.  This is useful if you want to rename chromosomes with a systematic numbering pattern.

       2.     A  file  with  two  columns  per line, separated by white space.  In each line, the original FASTA
              chromosome name should be in column 1 and the desired chromosome name will be in column 2.

              The meaning of file format 2 depends on whether <b>--limit-to-names</b> is specified.  If so, the  genome
              build  will be limited to those chromosomes in this file.  Otherwise, all chromosomes in the FASTA
              file will be included, but only those chromosomes in this file will be re-named, which provides an
              easy way to change just a few chromosome names.

              This file can be combined with the <b>--sort</b>=<u>names</u> option, in which the order of chromosomes is  that
              given  in the file.  In this case, every chromosome must be listed in the file, and for chromosome
              names that should not be changed, column 2 can be blank (or the same as column 1).  The option  of
              a  blank  column  2  is  allowed only when specifying <b>--sort</b>=<u>names</u>, because otherwise, the program
              cannot distinguish between a 1-column and 2-column names file.

       <b>-L</b>, <b>--limit-to-names</b>
              Determines whether to limit the genome build to the lines listed in the  <b>--names</b>  file.   You  can
              limit  a  genome  build  to  certain chromosomes with this option, plus a <b>--names</b> file that either
              renames chromosomes, or lists the same names in both columns for the desired chromosomes.

       <b>-k</b>, <b>--kmer</b>=<u>INT</u>
              k-mer value for genomic index (allowed: 15 or less, default is 15)

       <b>-q</b> INT sampling interval for genomoe (allowed: 1-3, default 3)

       <b>-s</b>, <b>--sort</b>=<u>STRING</u>
              Sort chromosomes using given method: none - use chromosomes as found in  FASTA  file(s)  (default)
              alpha  -  sort  chromosomes alphabetically (chr10 before chr 1) numeric-alpha - chr1, chr1U, chr2,
              chrM, chrU, chrX, chrY chrom - chr1, chr2, chrM, chrX, chrY, chr1U, chrU names - sort  chromosomes
              based on file provided to <b>--names</b> flag

       <b>-g</b>, <b>--gunzip</b>
              Files are gzipped, so need to gunzip each file first

       <b>-E</b>, <b>--fasta-pipe</b>=<u>STRING</u>
              Interpret argument as a command, instead of a list of FASTA files

       <b>-Q</b>, <b>--fastq</b>
              Files are in FASTQ format

       <b>-R</b>, <b>--revcomp</b>
              Reverse complement all contigs

       <b>-w</b> INT Wait (sleep) this many seconds after each step (default 2)

       <b>-o</b>, <b>--circular</b>=<u>STRING</u>
              Circular  chromosomes (either a list of chromosomes separated by a comma, or a filename containing
              circular chromosomes, one per line).  If you use the <b>--names</b> feature,  then  you  should  use  the
              substitute  name  of the chromosome, not the original name, for this option.  (NOTE: This behavior
              is different from previous versions, and starts with version 2020-10-20.)

       <b>-2</b>, <b>--altscaffold</b>=<u>STRING</u>
              File with alt scaffold info, listing alternate scaffolds, one per line,  tab-delimited,  with  the
              following  fields:  (1)  alt_scaf_acc,  (2)  parent_name, (3) orientation, (4) alt_scaf_start, (5)
              alt_scaf_stop, (6) parent_start, (7) parent_end.

       <b>-e</b>, <b>--nmessages</b>=<u>INT</u>
              Maximum number of messages (warnings, contig reports) to report (default 50)

       <b>--sarray</b>=<u>INT</u>
              Whether to build suffix array: 0=no (default), 1=yes

   <b>Options</b> <b>for</b> <b>older</b> <b>genome</b> <b>formats:</b>
       <b>-M</b>, <b>--mdflag</b>=<u>STRING</u>
              Use MD file from NCBI for mapping contigs to chromosomal coordinates

   <b>Options</b> <b>for</b> <b>transcriptome-guided</b> <b>alignment:</b>
       <b>-C</b>, <b>--transcriptomedir</b>=<u>STRING</u>
              Destination directory for installation (defaults to gmapdb directory specified at configure time)

       <b>-c</b>, <b>--transcriptomedb</b>=<u>STRING</u>
              Transcriptome name, plus one of these four flags:

       <b>--gtf</b>=<u>FILE</u>
              GTF file containing transcripts

       <b>--gff3</b>=<u>FILE</u>
              GFF3 file containing transcripts

       <b>-G</b>, <b>--genes</b>=<u>FILE</u>
              Genes file containing transcripts

       <b>-T</b>, <b>--transcripts</b>=<u>FILE</u>
              FASTA file containing transcripts

       <b>-t</b>, <b>--nthreads</b>=<u>INT</u>
              Number of  threads  for  GMAP  alignment  of  transcripts  to  genome  (default  8).   Applies  if
              <b>--transcripts</b> option is given

       Other tools of GMAP suite are located in /usr/lib/gmap

gmap_build 2024-11-20+ds-2                        February 2025                                    <u><a href="../man1/GMAP_BUILD.1.html">GMAP_BUILD</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>