<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>homGeneMapping - create summary of gene homology</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/augustus">augustus_3.5.0+dfsg-4build7_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       homGeneMapping - create summary of gene homology

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>homGeneMapping</b> [options] --gtfs=gffilenames.tbl --halfile=aln.hal

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>homGeneMapping</b> takes a set of gene predictions of different genomes and a hal alignment of the genomes
       and prints a summary for each gene, e.g.

       •   how many of its exons/introns are in agreement with genes of other genomes

       •   how many of its exons/introns are supported by extrinsic evidence from any of the genomes

       •   a list of geneids of homologous genes

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Mandatory</b> <b>parameters</b>
       <b>--halfile=aln.hal</b>
           input hal file

       <b>--gtfs=gtffilenames.tbl</b>
           a text file containing the locations of the input gene files and optionally the hints files (both in
           GTF format). The file is formatted as follows:

               name_of_genome_1  path/to/genefile/of/genome_1  path/to/hintsfile/of/genome_1
               name_of_genome_2  path/to/genefile/of/genome_2  path/to/hintsfile/of/genome_2
               ...
               name_of_genome_N  path/to/genefile/of/genome_N  path/to/hintsfile/of/genome_N

   <b>Additional</b> <b>options</b>
       <b>--cpus=N</b>
           N is the number of CPUs to use (default: 1)

       <b>--noDupes</b>
           do not map between duplications in hal graph (default: off)

       <b>--details</b>
           print detailed output (default: off)

       <b>--halLiftover_exec_dir=DIR</b>
           Directory that contains the executable halLiftover If not specified it must be in $PATH environment
           variable.

       <b>--unmapped</b>
           print a GTF attribute with a list of all genomes, that are not aligned to the corresponding gene
           feature, e.g. hgm_unmapped \"1,4,5\"; (default; off)

       <b>--tmpdir=DIR</b>
           a temporary file directory that stores lifted over files (default 'tmp/' in current directory)

       <b>--outdir=DIR</b>
           file directory that stores output gene files (default: current directory)

       <b>--printHomologs=FILE</b>
           prints disjunct sets of homologous transcripts to FILE, e.g.

               # 0     dana
               # 1     dere
               # 2     dgri
               # 3     dmel
               # 4     dmoj
               # 5     dper
               (0, jg4139.t1) (0, jg4140.t1) (1, jg7797.t1) (2, jg3247.t1) (4, jg6720.t1) (5, jg313.t1)
               (1, jg14269.t1) (3, jg89.t1) (5, jg290.t1)
               ...
               Two transcripts are in the same set, if all their exons/introns are homologs and their are
               no additional exons/introns.
               This option requires the Boost C++ Library

       <b>--dbaccess=db</b>
           retrieve hints from an SQLite database. In order to set up a database and populate it with hints a
           separate tool 'load2sqlitedb' is provided. For more information, see the documentation in
           docs/RUNNING-AUGUSTUS-IN-CGP-MODE.md (DATABASE ACCESS / SQLite) in the Augustus package. If both a
           database and hint files in 'gtffilenames.tbl' are specified, hints are retrieved from both sources.

</pre><h4><b>EXAMPLE</b></h4><pre>
       homGeneMapping --noDupes --halLiftover_exec_dir=<a href="file:~/tools/progressiveCactus/submodules/hal/bin">~/tools/progressiveCactus/submodules/hal/bin</a>
       --gtfs=gtffilenames.tbl --halfile=msca.hal homGeneMapping --gtfs=gtffilenames.tbl --halfile=aln.hal
       --outdir=outdir --cpus=4

</pre><h4><b>AUTHORS</b></h4><pre>
       AUGUSTUS was written by M. Stanke, O. Keller, S. König, L. Gerischer and L. Romoth.

</pre><h4><b>ADDITIONAL</b> <b>DOCUMENTATION</b></h4><pre>
       An exhaustive documentation can be found in the file /usr/share/doc/augustus/README.md.gz.

                                                                                               <u><a href="../man1/HOMGENEMAPPING.1.html">HOMGENEMAPPING</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>