<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>tigr-glimmer — Find/Score potential genes in genome-file using the probability model in icm-file</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/tigr-glimmer">tigr-glimmer_3.02b-6_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       tigr-glimmer — Find/Score potential genes in genome-file using the probability model in icm-file

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>tigr-glimmer3</b> [<u>genome-file]</u>  <u>[icm-file]</u>  <u>[[options]]</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       <b>tigr-glimmer</b>  is  a  system  for  finding  genes in microbial DNA, especially the genomes of bacteria and
       archaea. <b>tigr-glimmer</b> (Gene Locator and Interpolated Markov  Modeler)  uses  interpolated  Markov  models
       (IMMs)  to  identify  the  coding  regions  and  distinguish  them  from noncoding DNA. The IMM approach,
       described in our Nucleic Acids Research paper on <b>tigr-glimmer</b> 1.0 and in our subsequent  paper  on  <b>tigr-</b>
       <b>glimmer</b>  2.0,  uses  a  combination  of  Markov  models  from 1st through 8th-order, weighting each model
       according to its predictive power. <b>tigr-glimmer</b> 1.0 and 2.0 use 3-periodic nonhomogenous Markov models in
       their IMMs.

       <b>tigr-glimmer</b> is the primary microbial gene finder at TIGR, and has been used  to  annotate  the  complete
       genomes  of  B. burgdorferi (Fraser et al., Nature, Dec. 1997), T. pallidum (Fraser et al., Science, July
       1998), T. maritima, D. radiodurans, M. tuberculosis, and non-TIGR projects including C.  trachomatis,  C.
       pneumoniae,  and  others.  Its  analyses  of  some  of  these genomes and others is available at the TIGR
       microbial database site.

       A special version of <b>tigr-glimmer</b> designed for small eukaryotes, GlimmerM, was used to find the genes  in
       chromosome  2 of the malaria parasite, P. falciparum.. GlimmerM is described in S.L. Salzberg, M. Pertea,
       A.L. Delcher, M.J. Gardner, and H. Tettelin, "Interpolated Markov models for  eukaryotic  gene  finding,"
       Genomics  59  (1999),  24-31.   Click here (<a href="http://www.tigr.org/software/glimmerm/">http://www.tigr.org/software/glimmerm/</a>) to visit the GlimmerM
       site, which includes information on how to download the GlimmerM system.

       The <b>tigr-glimmer</b> system consists of two main programs. The first of these is the training program, build-
       imm. This program takes an input set of sequences  and  builds  and  outputs  the  IMM  for  them.  These
       sequences can be complete genes or just partial orfs. For a new genome, this training data can consist of
       those  genes  with  strong  database hits as well as very long open reading frames that are statistically
       almost certain to be genes. The second program is glimmer, which uses this IMM to identify putative genes
       in an entire genome. <b>tigr-glimmer</b> automatically resolves conflicts  between  most  overlapping  genes  by
       choosing  one  of them. It also identifies genes that are suspected to truly overlap, and flags these for
       closer inspection by the user. These ``suspect'' gene candidates have been a very small percentage of the
       total for all the genomes analyzed thus far.  <b>tigr-glimmer</b> is a program that...

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-C</b> <u>n</u>      Use n as GC percentage of independent model

                 Note:  n should be a percentage, e.g., -C 45.2

       -f        Use ribosome-binding energy to choose start codon

       <b>+f</b>        Use first codon in orf as start codon

       <b>-g</b> <u>n</u>      Set minimum gene length to n

       <b>-i</b> <u>filename</u>
                 Use <u>filename</u>   <u>to</u> <u>select</u> <u>regions</u> <u>of</u> <u>bases</u> <u>that</u> <u>are</u> <u>off</u> <u>limits,</u> <u>so</u> <u>that</u>  <u>no</u>  <u>bases</u>  <u>within</u>  <u>that</u>
                 <u>area</u> <u>will</u> <u>be</u> <u>examined</u>

       <b>-l</b>        Assume linear rather than circular genome, i.e., no wraparound

       <b>-L</b> <u>filename</u>
                 Use filename to specify a list of orfs that should be scored separately, with no overlap rules

       <b>-M</b>        Input is a multifasta file of separate genes to be scored separately, with no overlap rules

       <b>-o</b> <u>n</u>      Set minimum overlap length to n.  Overlaps shorter than this are ignored.

       <b>-p</b> <u>n</u>      Set  minimum overlap percentage to n%.  Overlaps shorter than this percentage of *both* strings
                 are ignored.

       <b>-q</b> <u>n</u>      Set the maximum length orf that can be rejected because of the  independent  probability  score
                 column to (n - 1)

       <b>-r</b>        Don't use independent probability score column

       <b>+r</b>        Use independent probability score column

       <b>-r</b>        Don't use independent probability score column

       <b>-s</b> <u>s</u>      Use string s as the ribosome binding pattern to find start codons.

       <b>+S</b>        Do  use  stricter independent intergenic model that doesn't give probabilities to in-frame stop
                 codons.  (Option is obsolete since this is now the only behaviour

       <b>-t</b> <u>n</u>      Set threshold score for calling as gene to n.  If the in-frame score &gt;= n, then the  region  is
                 given a number and considered a potential gene.

       <b>-w</b> <u>n</u>      Use  "weak"  scores  on  tentative  genes  n  or  longer.   Weak  scores ignore the independent
                 probability score.

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       tigr-adjust (1), tigr-anomaly   (1), tigr-build-icm (1),  tigr-check  (1),  tigr-codon-usage  (1),  tigr-
       compare-lists  (1),  tigr-extract  (1), tigr-generate (1), tigr-get-len (1), tigr-get-putative (1), tigr-
       glimmer3 (1), tigr-long-orfs (1)

       <a href="http://www.tigr.org/software/glimmer/">http://www.tigr.org/software/glimmer/</a>

       Please see the readme in /usr/share/doc/glimmer for a description on how to use Glimmer.

</pre><h4><b>AUTHOR</b></h4><pre>
       This manual page was quickly copied from the glimmer web site by Steffen Moeller  <a href="mailto:moeller@debian.org">moeller@debian.org</a>  for
       the <b>Debian</b> system.

                                                                                                 <u><a href="../man1/TIGR-GLIMMER.1.html">TIGR-GLIMMER</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>