<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bio::Tools::OddCodes - Object holding alternative alphabet coding for one protein sequence</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/questing/+package/libbio-perl-perl">libbio-perl-perl_1.7.8-1_all</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       Bio::Tools::OddCodes - Object holding alternative alphabet coding for one protein sequence

</pre><h4><b>SYNOPSIS</b></h4><pre>
         # Take a sequence object from eg, an inputstream, and creates an
         # object for the purposes of rewriting that sequence in another
         # alphabet.  These are abbreviated amino acid sequence alphabets,
         # designed to simplify the statistical aspects of analysing protein
         # sequences, by reducing the combinatorial explosion of the
         # 20-letter alphabet.  These abbreviated alphabets range in size
         # from 2 to 8.

         # Creating the OddCodes object, eg:

               my $inputstream = Bio::SeqIO-&gt;new( '-file' =&gt; "seqfile",
                                                  '-format' =&gt; 'Fasta');
               my $seqobj = $inputstream-&gt;next_seq();
               my $oddcode_obj = Bio::Tools::Oddcodes-&gt;new(-seq =&gt; $seqobj);

         # or:

               my $seqobj = Bio::PrimarySeq-&gt;new
                     (-seq=&gt;'[cut and paste a sequence here]',
                      -alphabet =&gt; 'protein',
                      -id =&gt; 'test');
               my $oddcode_obj  =  Bio::Tools::OddCodes-&gt;new(-seq =&gt; $seqobj);

         # do the alternative coding, returning the answer as a reference to
         # a string

               my $output = $oddcode_obj-&gt;structural();
               my $output = $oddcode_obj-&gt;chemical();
               my $output = $oddcode_obj-&gt;functional();
               my $output = $oddcode_obj-&gt;charge();
               my $output = $oddcode_obj-&gt;hydrophobic();
               my $output = $oddcode_obj-&gt;Dayhoff();
               my $output = $oddcode_obj-&gt;Sneath();
               my $output = $oddcode_obj-&gt;Stanfel();

         # display sequence in new form, eg:

               my $new_coding = $$output;
               print "\n$new_coding";

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Bio::Tools::Oddcodes is a welterweight object for rewriting a protein sequence in an alternative
       alphabet.  Eight of these are provided, ranging from the the 2-letter hydrophobic alphabet, to the
       8-letter chemical alphabet.  These are useful for the statistical analysis of protein sequences since
       they can partially avoid the combinatorial explosion produced by the full 20-letter alphabet (eg. 400
       dimers, 8000 trimers etc.)

       The objects will print out a warning if the input sequence is not a protein. If you know what you are
       doing, you can silence the warning by setting <b>verbose()</b> to a negative value.

       See SYNOPSIS above for object creation code.

</pre><h4><b>REFERENCES</b></h4><pre>
       Stanfel LE (1996) A new approach to clustering the amino acids.  J. theor.  Biol. 183, 195-205.

       Karlin S, Ost F and Blaisdell BE (1989)  Patterns in DNA and amino acid sequences and their statistical
       significance.  Chapter 6 of: Mathematical Methods for DNA Sequences.  Waterman MS (ed.)  CRC Press, Boca
       Raton , FL.

</pre><h4><b>FEEDBACK</b></h4><pre>
   <b>Mailing</b> <b>Lists</b>
       User feedback is an integral part of the evolution of this and other Bioperl modules. Send your comments
       and suggestions preferably to one of the Bioperl mailing lists.  Your participation is much appreciated.

         <a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a>                  - General discussion
         <a href="http://bioperl.org/wiki/Mailing_lists">http://bioperl.org/wiki/Mailing_lists</a>  - About the mailing lists

   <b>Support</b>
       Please direct usage questions or support issues to the mailing list:

       <u><a href="mailto:bioperl-l@bioperl.org">bioperl-l@bioperl.org</a></u>

       rather than to the module maintainer directly. Many experienced and reponsive experts will be able look
       at the problem and quickly address it. Please include a thorough description of the problem with code and
       data examples if at all possible.

   <b>Reporting</b> <b>Bugs</b>
       Report bugs to the Bioperl bug tracking system to help us keep track the bugs and their resolution.  Bug
       reports can be submitted via the web:

         https://github.com/bioperl/bioperl-live/issues

</pre><h4><b>AUTHOR</b></h4><pre>
       Derek Gatherer

</pre><h4><b>APPENDIX</b></h4><pre>
       The rest of the documentation details each of the object methods.  Internal methods are usually preceded
       with a _

   <b>structural</b>
        Title   : structural
        Usage   : $output = $oddcode_obj-&gt;structural();
        Function: turns amino acid sequence into 3-letter structural alphabet
                : A (ambivalent), E (external), I (internal)
        Example : a sequence ACDEFGH will become AAEEIAE
        Returns : Reference to the new sequence string
        Args    : none

   <b>functional</b>
        Title   : functional
        Usage   : $output = $oddcode_obj-&gt;functional();
        Function: turns amino acid sequence into 4-letter functional alphabet
                : A (acidic), C (basic), H (hydrophobic), P (polar)
        Example : a sequence ACDEFGH will become HPAAHHC
        Returns : Reference to the new sequence string
        Args    : none

   <b>hydrophobic</b>
        Title   : hydrophobic
        Usage   : $output = $oddcode_obj-&gt;hydrophobic();
        Function: turns amino acid sequence into 2-letter hydrophobicity alphabet
                : O (hydrophobic), I (hydrophilic)
        Example : a sequence ACDEFGH will become OIIIOII
        Returns : Reference to the new sequence string
        Args    : none

   <b>Dayhoff</b>
        Title   : Dayhoff
        Usage   : $output = $oddcode_obj-&gt;Dayhoff();
        Function: turns amino acid sequence into 6-letter Dayhoff alphabet
        Example : a sequence ACDEFGH will become CADDGCE
                : A (=C),   C (=AGPST), D (=DENQ),
                : E (=HKR), F (=ILMV),  G (=FWY)
        Returns : Reference to the new sequence string
        Args    : none

   <b>Sneath</b>
        Title   : Sneath
        Usage   : $output = $oddcode_obj-&gt;Sneath();
        Function: turns amino acid sequence into 7-letter Sneath alphabet
        Example : a sequence ACDEFGH will become CEFFHCF
                : A (=ILV), C (=AGP), D (=MNQ), E (=CST),
                : F (=DE),  G (=KR),  H (=FHWY)
        Returns : Reference to the new sequence string
        Args    : none

   <b>Stanfel</b>
        Title   : Stanfel
        Usage   : $output = $oddcode_obj-&gt;Stanfel();
        Function: turns amino acid sequence into 4-letter Stanfel alphabet
        Example : a sequence ACDEFGH will become AACCDAE
                : A (=ACGILMPSTV), C (=DENQ), D (=FWY), E (=HKR)
        Returns : Reference to the new sequence string
        Args    : none

   <b>chemical</b>
        Title   : chemical
        Usage   : $output = $oddcode_obj-&gt;chemical();
        Function: turns amino acid sequence into 8-letter chemical alphabet
                : A (acidic), L (aliphatic), M (amide), R (aromatic)
                : C (basic),  H (hydroxyl),  I (imino), S (sulphur)
        Example : a sequence ACDEFGH will become LSAARAC
        Returns : Reference to the new sequence string
        Args    : none

   <b>charge</b>
        Title   : charge
        Usage   : $output = $oddcode_obj-&gt;charge();
        Function: turns amino acid sequence into 3-letter charge alphabet
        Example : a sequence ACDEFGH will become NNAANNC
                : A (negative; NOT anode), C (positive; NOT cathode), N (neutral)
        Returns : Reference to the new sequence string
        Args    : none

perl v5.32.1                                       2021-08-15                          <u>Bio::Tools::<a href="../man3pm/OddCodes.3pm.html">OddCodes</a></u>(3pm)
</pre>
 </div>
</div></section>
</div>
</body>
</html>