<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>conv_gtf_to_bed - legacy script for invoking the PALEOMIX command "gtf_to_bed"</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/paleomix">paleomix_1.3.8-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       conv_gtf_to_bed - legacy script for invoking the PALEOMIX command "gtf_to_bed"

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>paleomix</b> <b>gtf_to_bed</b> [options] in.gtf out_prefix [in.scaffolds]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The  PALEOMIX  pipelines  are  a set of pipelines and tools designed to aid the rapid processing of High-
       Throughput Sequencing (HTS) data: The BAM pipeline  processes  de-multiplexed  reads  from  one  or  more
       samples,  through sequence processing and alignment, to generate BAM alignment files useful in downstream
       analyses; the Phylogenetic pipeline carries out genotyping and phylogenetic inference  on  BAM  alignment
       files, either produced using the BAM pipeline or generated elsewhere; and the Zonkey pipeline carries out
       a  suite  of analyses on low coverage equine alignments, in order to detect the presence of F1-hybrids in
       archaeological assemblages. In addition, PALEOMIX aids in metagenomic analysis of the extracts.

       The pipelines have been designed  with  ancient  DNA  (aDNA)  in  mind,  and  includes  several  features
       especially  useful  for  the  analyses  of  ancient  samples, but can all be for the processing of modern
       samples, in order to ensure consistent data processing.

       <b>conv_gtf_to_bed</b> is a legacy script for invoking the <a href="../man1/paleomix.1.html">paleomix</a>(1) command "gtf_to_bed"

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>positional</b> <b>arguments:</b>
       INPUT.gtf
              Input file in GTF format.

       OUTPUT_PREFIX
              Prefix of output files.

       SCAFFOLDS
              Mapping of scaffolds to contig positions; e.g. mapping individual Un* scaffolds onto chrUn.

   <b>optional</b> <b>arguments:</b>
       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>--keep-all-transcripts</b>
              Include all transcripts in the output BED files, not just the  longest  transcript  of  each  gene
              [default: off]

       <b>--keep-malformed-proteins</b>
              Include  transcripts  of  protein-coding  in  the output, even if the the length of the CDS is not
              divisible by 3 [default: off]

       <b>--contig-prefix</b> CONTIG_PREFIX
              Add prefix to contig names (e.g. 'chr') [default: no prefix].

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <a href="../man1/paleomix.1.html">paleomix</a>(1)

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

conv_gtf_to_bed 1.2.7                             February 2017                               <u><a href="../man1/CONV_GTF_TO_BED.1.html">CONV_GTF_TO_BED</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>