<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>paleomix - pipelines and tools for HTS data analyses</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/paleomix">paleomix_1.3.8-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       paleomix - pipelines and tools for HTS data analyses

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>paleomix</b> <u>&lt;command&gt;</u> [<u>options</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The  PALEOMIX  pipelines  are  a set of pipelines and tools designed to aid the rapid processing of High-
       Throughput Sequencing (HTS) data: The BAM pipeline  processes  de-multiplexed  reads  from  one  or  more
       samples,  through sequence processing and alignment, to generate BAM alignment files useful in downstream
       analyses; the Phylogenetic pipeline carries out genotyping and phylogenetic inference  on  BAM  alignment
       files, either produced using the BAM pipeline or generated elsewhere; and the Zonkey pipeline carries out
       a  suite  of analyses on low coverage equine alignments, in order to detect the presence of F1-hybrids in
       archaeological assemblages. In addition, PALEOMIX aids in metagenomic analysis of the extracts.

       The pipelines have been designed  with  ancient  DNA  (aDNA)  in  mind,  and  includes  several  features
       especially  useful  for  the  analyses  of  ancient  samples, but can all be for the processing of modern
       samples, in order to ensure consistent data processing.

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Pipelines</b>
       paleomix bam_pipeline
              Pipeline for trimming and mapping of NGS reads.

       paleomix trim_pipeline
              Equivalent to 'bam_pipeline', but only runs the trimming steps.

       paleomix phylo_pipeline
              Pipeline for genotyping and phylogenetic inference from BAMs.

       paleomix zonkey
              Pipeline for detecting F1 (equine) hybrids.

   <b>BAM/SAM</b> <b>tools:</b>
       paleomix cleanup
              Reads SAM file from STDIN, and outputs sorted, tagged, and filter BAM, for which NM  and  MD  tags
              have been updated.

       paleomix coverage
              Calculate coverage across reference sequences or regions of interest.

       paleomix depths
              Calculate depth histograms across reference sequences or regions of interest.

       paleomix duphist
              Generates PCR duplicate histogram; used with the 'Preseq' tool.

       paleomix rmdup_collapsed
              Filters PCR duplicates for collapsed pairedended reads generated by the AdapterRemoval tool.

   <b>VCF/GTF/BED/Pileup</b> <b>tools:</b>
       paleomix genotype
              Creates  bgzipped VCF for a set of (sparse) BED regions, or for entire chromosomes / contigs using
              SAMTools / BCFTools.

       paleomix gtf_to_bed
              Convert GTF file to BED files grouped by feature (coding, RNA, etc).

       paleomix sample_pileup
              Randomly sample sites in a pileup to generate a FASTA sequence.

       paleomix vcf_filter
              Quality filters for VCF records, similar to 'vcfutils.pl varFilter'.

       paleomix vcf_to_fasta
              Create most likely FASTA sequence from tabixindexed VCF file.

   <b>Misc</b> <b>tools:</b>
       paleomix cat
              Generalized cat command for gz, bz2 and uncompressed files.

</pre><h4><b>AUTHOR</b></h4><pre>
       If you make use of PALEOMIX in your work, please cite

              Schubert et al, "Characterization of ancient and modern genomes by SNP detection and  phylogenomic
              and  metagenomic  analysis  using  PALEOMIX".   Nature  Protocols.  2014  May; <a href="../man5/9.5.html">9</a>(5): 1056-82. doi:
              10.1038/nprot.2014.063

       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

paleomix 1.2.7                                    February 2017                                      <u><a href="../man1/PALEOMIX.1.html">PALEOMIX</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>