<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>phylo_pipeline - Phylo Pipeline of PALEOMIX suite</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/paleomix">paleomix_1.3.8-2_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       phylo_pipeline - Phylo Pipeline of PALEOMIX suite

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>paleomix</b> <b>phylo_pipeline</b> &lt;command&gt; [<u>options</u>] [<u>makefiles</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       The  PALEOMIX  pipelines  are  a set of pipelines and tools designed to aid the rapid processing of High-
       Throughput Sequencing (HTS) data: The BAM pipeline  processes  de-multiplexed  reads  from  one  or  more
       samples,  through sequence processing and alignment, to generate BAM alignment files useful in downstream
       analyses; the Phylogenetic pipeline carries out genotyping and phylogenetic inference  on  BAM  alignment
       files, either produced using the BAM pipeline or generated elsewhere; and the Zonkey pipeline carries out
       a  suite  of analyses on low coverage equine alignments, in order to detect the presence of F1-hybrids in
       archaeological assemblages. In addition, PALEOMIX aids in metagenomic analysis of the extracts.

       The pipelines have been designed  with  ancient  DNA  (aDNA)  in  mind,  and  includes  several  features
       especially  useful  for  the  analyses  of  ancient  samples, but can all be for the processing of modern
       samples, in order to ensure consistent data processing.

       <b>phylo_pipeline</b> is the phylo pipeline of <a href="../man1/paleomix.1.html">paleomix</a>(1).

</pre><h4><b>OPTIONS</b></h4><pre>
   <b>Commands</b>
       phylo_pipeline help
              Display this message.

       phylo_pipeline example [...]
              Copy example project to folder.

       phylo_pipeline makefile
              Print makefile template.

       phylo_pipeline genotype [...]
              Carry out genotyping according to makefile.

       phylo_pipeline msa [...]
              Carry out multiple sequence alignment.

       phylo_pipeline phylogeny
              Carry out phylogenetic inference.

   <b>Optional</b> <b>arguments</b>
       <b>--version</b>
              show program's version number and exit

       <b>-h</b>, <b>--help</b>
              show this help message and exit

       <b>--to-dot-file</b>=<u>DOT_FILE</u>
              Write dependency tree to the specified dot-file.

   <b>Progress</b> <b>reporting</b>
       <b>--progress-ui</b>=<u>PROGRESS_UI</u>
              Select method for displaying the progress of the pipeline:  'running'  =  Display  only  currently
              running  nodes;  'progress' = Display changes in state; 'summary'; one-line summary only. [Default
              is 'running']

       <b>--ui-colors</b>=<u>UI_COLORS</u>
              Enable, disable, or force the use of color codes when printing the command-line UI. Unless forced,
              colors will only be printed if STDOUT is a TTY [Default is 'on']

   <b>Logging</b>
       <b>--log-file</b>=<u>LOG_FILE</u>
              Create the specified log-file and write any messages to this file. By default, a log-file will  be
              generated in the folder specified using <b>--temp-root</b>, but only when messages are logged

       <b>--log-level</b>=<u>LOG_LEVEL</u>
              Log  messages  to log-file at and above the specified level; one of 'info', 'warning', 'error', or
              'debug' [warning]

   <b>Scheduling</b>
       <b>--samtools-max-threads</b>=<u>SAMTOOLS_MAX_THREADS</u>
              Maximum number of threads to use when genotyping or building pileups [1]

       <b>--examl-max-threads</b>=<u>EXAML_MAX_THREADS</u>
              Maximum number of threads to use for each instance of ExaML [1]

       <b>--max-threads</b>=<u>MAX_THREADS</u>
              Maximum number of threads to use in total [4]

       <b>--dry-run</b>
              If passed, only a dry-run in performed, the dependency tree is printed, and no tasks are executed.

   <b>Required</b> <b>paths:</b>
       <b>--temp-root</b>=<u>TEMP_ROOT</u>
              Location for temporary files and folders [/tmp/USERNAME/phylo_pipeline]

       <b>--samples-root</b>=<u>SAMPLES_ROOT</u>
              Location of BAM files for each sample [./data/samples]

       <b>--regions-root</b>=<u>REGIONS_ROOT</u>
              Location of BED files containing regions of interest [./data/regions]

       <b>--prefix-root</b>=<u>PREFIX_ROOT</u>
              Location of prefixes (FASTAs) [./data/prefixes]

       <b>--refseq-root</b>=<u>REFSEQ_ROOT</u>
              Location of reference sequences (FASTAs) [./data/refseqs]

       <b>--destination</b>=<u>DESTINATION</u>
              The destination folder for result files [./results]

   <b>Files</b> <b>and</b> <b>executables:</b>
       <b>--list-input-files</b>
              List all input files used by pipeline for the makefile(s), excluding any generated by the pipeline
              itself.

       <b>--list-output-files</b>
              List all output files generated by pipeline for the makefile(s).

       <b>--list-executables</b>
              List all executables required by the pipeline, with version requirements (if any).

   <b>Config</b> <b>files:</b>
       <b>--write-config-file</b>
              Write config using current settings to $HOME/.paleomix/phylo_pipeline.ini

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       <a href="../man1/paleomix.1.html">paleomix</a>(1)

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

phylo_pipeline 1.2.7                              February 2017                                <u><a href="../man1/PHYLO_PIPELINE.1.html">PHYLO_PIPELINE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>