<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>run_mapsembler2_pipeline - pipelines the mapsembler2 tools</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/mapsembler2">mapsembler2_2.2.4+dfsg1-4_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       run_mapsembler2_pipeline - pipelines the mapsembler2 tools

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>run_mapsembler_and_phaser.sh</b> <u>-s</u> <u>&lt;starter.fasta&gt;</u> <u>-r</u> <u>&lt;reads.faste&gt;</u> <u>-t</u> [<u>1/2/3/4</u>]<u>&lt;options&gt;</u>

</pre><h4><b>DESCRIPTION</b></h4><pre>
       run_mapsembler_pipeline.sh, a pipelining mapsembler2_extremities, mapsembler2_extend and kissread_g

</pre><h4><b>OPTIONS</b></h4><pre>

       <b>-s</b>: file containing starters (fasta)

       <b>-r</b>  list of reads separated by space, surrounded by the '"' character. Note that reads may be in fasta or
              fastq    format,    gzipped    or    not.    Example:    <b>-r</b>     "data_sample/reads_sequence1.fasta
              data_sample/reads_sequence2.fasta.gz".

       <b>-t</b>: kind of assembly: 1=unitig (fasta), 2=contig (fasta), 3=unitig (graph), 4=contig(graph)

   <b>OPTIONAL</b>

       <b>-p</b> prefix. All out files will start with this prefix. Example: <b>-p</b> my_prefix

       <b>-k</b> value. Set the length of used kmers. Must fit the compiled value. Default=31. Example <b>-k</b> 31

       <b>-c</b> value. Set the minimal coverage. Default=5. Example <b>-c</b> 5

       <b>-d</b>  value.  Set the number of authorized substitutions used while mapping reads on found SNPs. Default=1.
              Example: <b>-d</b> 1

       <b>-g</b> value. Estimated genome size. Used only to control memory usage. e.g. 3 billion (3000000000) uses  4Gb
              of RAM. Default=10 million. Example: <b>-d</b> 10000000

       <b>-f</b> value. Set the process of search in the graph (1=Breadth  and 2=Depth). Default=1. Example: <b>-f</b> 1

       <b>-x</b> value. Set the maximal nodes length . Default=40. Example: <b>-x</b> 40

       <b>-y</b> value. Set the maximal graph depth . Default=10000. Example: <b>-y</b> 10000

       <b>-h</b> Prints this message and exist

</pre><h4><b>SEE</b> <b>ALSO</b></h4><pre>
       Any further question: read the readme file or contact us: <a href="mailto:pierre.peterlongo@inria.fr">pierre.peterlongo@inria.fr</a>

</pre><h4><b>AUTHOR</b></h4><pre>
       This manpage was written by Andreas Tille for the Debian distribution and can be used for any other usage
       of the program.

run_mapsembler2_pipeline 2.2.4                    October 2018                       <u><a href="../man1/RUN_MAPSEMBLER2_PIPELINE.1.html">RUN_MAPSEMBLER2_PIPELINE</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>