<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>lambda - the Local Aligner for Massive Biological DatA</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/lambda-align">lambda-align_1.0.3-6_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       lambda - the Local Aligner for Massive Biological DatA

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>lambda</b> [<u>OPTIONS</u>] <u>-q</u> <u>QUERY.fasta</u> <u>-d</u> <u>DATABASE.fasta</u> [<u>-o</u> <u>output.m8</u>]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Lambda  is  a  local  aligner  optimized  for  many  query sequences and searches in protein space. It is
       compatible to BLAST, but much faster than BLAST and many other comparable tools.

       Detailed information is available in the wiki: &lt;https://github.com/seqan/lambda/wiki&gt;

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              Display the help message.

       <b>-hh</b>, <b>--full-help</b>
              Display the help message with advanced options.

       <b>--version-check</b> <u>BOOL</u>
              Turn this option off to disable version update notifications of the application.  One  of  <u>1</u>,  <u>ON</u>,
              <u>TRUE</u>, <u>T</u>, <u>YES</u>, <u>0</u>, <u>OFF</u>, <u>FALSE</u>, <u>F</u>, and <u>NO</u>. Default: <u>1</u>.

       <b>--version</b>
              Display version information.

       <b>--copyright</b>
              Display long copyright information.

       <b>-v</b>, <b>--verbosity</b> <u>INTEGER</u>
              Display  more/less diagnostic output during operation: 0 [only errors]; 1 [default]; 2 [+run-time,
              options and statistics]. In range [0..2]. Default: <u>1</u>.

   <b>Input</b> <b>Options:</b>
       <b>-q</b>, <b>--query</b> <u>INPUT_FILE</u>
              Query sequences. Valid filetypes are: <u>.sam[.*]</u>, <u>.raw[.*]</u>, <u>.gbk[.*]</u>, <u>.frn[.*]</u>,  <u>.fq[.*]</u>,  <u>.fna[.*]</u>,
              <u>.ffn[.*]</u>,  <u>.fastq[.*]</u>,  <u>.fasta[.*]</u>,  <u>.faa[.*]</u>, <u>.fa[.*]</u>, <u>.embl[.*]</u>, and <u>.bam</u>, where * is any of the
              following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent (de)compression.

       <b>-d</b>, <b>--database</b> <u>INPUT_FILE</u>
              Path to original database sequences (a precomputed index with .sa or .fm needs to  exist!).  Valid
              filetypes  are:  <u>.sam[.*]</u>,  <u>.raw[.*]</u>, <u>.gbk[.*]</u>, <u>.frn[.*]</u>, <u>.fq[.*]</u>, <u>.fna[.*]</u>, <u>.ffn[.*]</u>, <u>.fastq[.*]</u>,
              <u>.fasta[.*]</u>, <u>.faa[.*]</u>, <u>.fa[.*]</u>, <u>.embl[.*]</u>, and <u>.bam</u>, where * is any of  the  following  extensions:
              <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent (de)compression.

       <b>-di</b>, <b>--db-index-type</b> <u>STRING</u>
              database index is in this format. One of <u>sa</u> and <u>fm</u>. Default: <u>fm</u>.

   <b>Output</b> <b>Options:</b>
       <b>-o</b>, <b>--output</b> <u>OUTPUT_FILE</u>
              File  to  hold  reports  on  hits (.m* are blastall -m* formats; .m8 is tab-seperated, .m9 is tab-
              seperated with with comments, .m0 is pairwise format). Valid  filetypes  are:  <u>.sam[.*]</u>,  <u>.m9[.*]</u>,
              <u>.m8[.*]</u>,  <u>.m0[.*]</u>,  and  <u>.bam</u>,  where  * is any of the following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for
              transparent (de)compression. Default: <u>output.m8</u>.

       <b>-oc</b>, <b>--output-columns</b> <u>STRING</u>
              Print specified column combination and/or order (.m8 and .m9 outputs only); call -oc help for more
              details. Default: <u>std</u>.

       <b>-id</b>, <b>--percent-identity</b> <u>INTEGER</u>
              Output only matches above this threshold  (checked  before  e-value  check).  In  range  [0..100].
              Default: <u>0</u>.

       <b>-e</b>, <b>--e-value</b> <u>DOUBLE</u>
              Output only matches that score below this threshold. In range [0..inf]. Default: <u>0.1</u>.

       <b>-nm</b>, <b>--num-matches</b> <u>INTEGER</u>
              Print at most this number of matches per query. In range [1..inf]. Default: <u>500</u>.

       <b>--sam-with-refheader</b> <u>STRING</u>
              BAM  files require all subject names to be written to the header. For SAM this is not required, so
              Lambda does not automatically do it to save space (especially  for  protein  database  this  is  a
              lot!). If you still want them with SAM, e.g. for better BAM compatibility, use this option. One of
              <u>on</u> and <u>off</u>. Default: <u>off</u>.

       <b>--sam-bam-seq</b> <u>STRING</u>
              Write  matching  DNA  subsequence  into SAM/BAM file (BLASTN). For BLASTX and TBLASTX the matching
              protein sequence is "untranslated" and positions  retransformed  to  the  original  sequence.  For
              BLASTP  and  TBLASTN  there is no DNA sequence so a "*" is written to the SEQ column. The matching
              protein sequence can be written as an optional tag, see --sam-bam-tags. If set to  uniq  than  the
              sequence  is omitted iff it is identical to the previous match's subsequence. One of <u>always</u>, <u>uniq</u>,
              and <u>never</u>. Default: <u>uniq</u>.

       <b>--sam-bam-tags</b> <u>STRING</u>
              Write the specified optional columns to the  SAM/BAM  file.  Call  --sam-bam-tags  help  for  more
              details. Default: <u>AS</u> <u>NM</u> <u>ZE</u> <u>ZI</u> <u>ZF</u>.

       <b>--sam-bam-clip</b> <u>STRING</u>
              Whether  to  hard-clip  or soft-clip the regions beyond the local match. Soft-clipping retains the
              full sequence in the output file, but obviously uses more space. One of <u>hard</u>  and  <u>soft</u>.  Default:
              <u>hard</u>.

   <b>General</b> <b>Options:</b>
       <b>-t</b>, <b>--threads</b> <u>INTEGER</u>
              number of threads to run concurrently.

       <b>-qi</b>, <b>--query-index-type</b> <u>STRING</u>
              controls double-indexing. One of <u>radix</u> and <u>none</u>. Default: <u>none</u>.

   <b>Alphabets</b> <b>and</b> <b>Translation:</b>
       <b>-p</b>, <b>--program</b> <u>STRING</u>
              Blast Operation Mode. One of <u>blastn</u>, <u>blastp</u>, <u>blastx</u>, <u>tblastn</u>, and <u>tblastx</u>. Default: <u>blastx</u>.

       <b>-g</b>, <b>--genetic-code</b> <u>INTEGER</u>
              The  translation  table  to  use  for  nucl -&gt; amino acid translation(not for BlastN, BlastP). See
              https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=c for ids (default is generic).  Six
              frames are generated. Default: <u>1</u>.

       <b>-ar</b>, <b>--alphabet-reduction</b> <u>STRING</u>
              Alphabet  Reduction  for  seeding  phase  (ignored for BLASTN). One of <u>none</u> and <u>murphy10</u>. Default:
              <u>murphy10</u>.

   <b>Seeding</b> <b>/</b> <b>Filtration:</b>
       <b>-sl</b>, <b>--seed-length</b> <u>INTEGER</u>
              Length of the seeds (default = 14 for BLASTN). Default: <u>10</u>.

       <b>-so</b>, <b>--seed-offset</b> <u>INTEGER</u>
              Offset for seeding (if unset = seed-length, non-overlapping; default = 5 for BLASTN). Default: <u>10</u>.

       <b>-sd</b>, <b>--seed-delta</b> <u>INTEGER</u>
              maximum seed distance. Default: <u>1</u>.

   <b>Miscellaneous</b> <b>Heuristics:</b>
       <b>-ps</b>, <b>--pre-scoring</b> <u>INTEGER</u>
              evaluate score of a region NUM times the size of the seed before extension (0 -&gt; no pre-scoring, 1
              -&gt; evaluate seed, n-&gt; area around seed, as well; default = 1 if no reduction is  used).  In  range
              [1..inf]. Default: <u>2</u>.

       <b>-pt</b>, <b>--pre-scoring-threshold</b> <u>DOUBLE</u>
              minimum average score per position in pre-scoring region. Default: <u>2</u>.

       <b>-pd</b>, <b>--filter-putative-duplicates</b> <u>STRING</u>
              filter hits that will likely duplicate a match already found. One of <u>on</u> and <u>off</u>. Default: <u>on</u>.

       <b>-pa</b>, <b>--filter-putative-abundant</b> <u>STRING</u>
              If  the  maximum  number  of  matches per query are found already, stop searching if the remaining
              realm looks unfeasable. One of <u>on</u> and <u>off</u>. Default: <u>on</u>.

   <b>Scoring:</b>
       <b>-sc</b>, <b>--scoring-scheme</b> <u>INTEGER</u>
              use '45' for Blosum45; '62' for Blosum62  (default);  '80'  for  Blosum80;  [ignored  for  BlastN]
              Default: <u>62</u>.

       <b>-ge</b>, <b>--score-gap</b> <u>INTEGER</u>
              Score per gap character (default = -2 for BLASTN). Default: <u>-1</u>.

       <b>-go</b>, <b>--score-gap-open</b> <u>INTEGER</u>
              Additional cost for opening gap (default = -5 for BLASTN). Default: <u>-11</u>.

       <b>-ma</b>, <b>--score-match</b> <u>INTEGER</u>
              Match score [only BLASTN]) Default: <u>2</u>.

       <b>-mi</b>, <b>--score-mismatch</b> <u>INTEGER</u>
              Mismatch score [only BLASTN] Default: <u>-3</u>.

   <b>Extension:</b>
       <b>-x</b>, <b>--x-drop</b> <u>INTEGER</u>
              Stop  Banded  extension if score x below the maximum seen (-1 means no xdrop). In range [-1..inf].
              Default: <u>30</u>.

       <b>-b</b>, <b>--band</b> <u>INTEGER</u>
              Size of the DP-band used in extension (-3 means log2 of query  length;  -2  means  sqrt  of  query
              length; -1 means full dp; n means band of size 2n+1) In range [-3..inf]. Default: <u>-3</u>.

</pre><h4><b>TUNING</b></h4><pre>
       Tuning  the seeding parameters and (de)activating alphabet reduction has a strong influence on both speed
       and sensitivity. We recommend the following alternative profiles for protein searches:

       fast (high similarity):       -ar none -sl 7 -sd 0

       sensitive (lower similarity): -so 5

       For further information see the wiki: &lt;https://github.com/seqan/lambda/wiki&gt;

</pre><h4><b>LEGAL</b></h4><pre>
       <b>lambda</b> <b>Copyright:</b> 2013-2017 Hannes Hauswedell, released under the GNU GPL v3 (or later);  2016-2017  Knut
       Reinert and Freie Universität Berlin, released under the 3-clause-BSDL
       <b>SeqAn</b> <b>Copyright:</b> 2006-2015 Knut Reinert, FU-Berlin; released under the 3-clause BSDL.
       <b>In</b> <b>your</b> <b>academic</b> <b>works</b> <b>please</b> <b>cite:</b> Hauswedell et al (2014); doi: 10.1093/bioinformatics/btu439
       For full copyright and/or warranty information see <b>--copyright</b>.

lambda 1.0.2                                       Feb  8 2020                                         <u><a href="../man1/LAMBDA.1.html">LAMBDA</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>