<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>lambda2_mkindexp - the Local Aligner for Massive Biological DatA</title>
    <style>
        body { font-family: monospace; margin: 20px; line-height: 1.4; }
        a { color: #0066cc; }
        pre { white-space: pre-wrap; }
    </style>
</head>
<body>
    <div id="main-content">
<section class="p-strip is-bordered">
<div class="row">
<div class="col-3 u-hide--small u-hide" id="toc">
</div>
<div id="tableWrapper">
<p id="distroAndSection"></p>

Provided by: <a href="https://launchpad.net/ubuntu/plucky/+package/lambda-align2">lambda-align2_2.0.1-3_amd64</a> <br><br><pre>
</pre><h4><b>NAME</b></h4><pre>
       lambda2_mkindexp - the Local Aligner for Massive Biological DatA

</pre><h4><b>SYNOPSIS</b></h4><pre>
       <b>lambda2</b> <b>mkindexp</b> [<u>OPTIONS</u>] -d DATABASE.fasta [-i INDEX.lambda]

</pre><h4><b>DESCRIPTION</b></h4><pre>
       Lambda  is  a  local  aligner  optimized  for  many  query sequences and searches in protein space. It is
       compatible to BLAST, but much faster than BLAST and many other comparable tools.

       Detailed information is available in the wiki: &lt;https://github.com/seqan/lambda/wiki&gt;

       This is the indexer_binary for creating lambda-compatible databases.

</pre><h4><b>OPTIONS</b></h4><pre>
       <b>-h</b>, <b>--help</b>
              Display the help message.

       <b>-hh</b>, <b>--full-help</b>
              Display the help message with advanced options.

       <b>--version</b>
              Display version information.

       <b>--copyright</b>
              Display long copyright information.

       <b>-v</b>, <b>--verbosity</b> <u>INTEGER</u>
              Display more/less diagnostic output during operation: 0 [only errors]; 1 [default]; 2  [+run-time,
              options and statistics]. In range [0..2]. Default: <u>1</u>.

   <b>Input</b> <b>Options:</b>
       <b>-d</b>, <b>--database</b> <u>INPUT_FILE</u>
              Database  sequences.  Valid  filetypes  are:  <u>.sam[.*]</u>,  <u>.raw[.*]</u>,  <u>.gbk[.*]</u>,  <u>.frn[.*]</u>,  <u>.fq[.*]</u>,
              <u>.fna[.*]</u>, <u>.ffn[.*]</u>, <u>.fastq[.*]</u>, <u>.fasta[.*]</u>, <u>.faa[.*]</u>, <u>.fa[.*]</u>, <u>.embl[.*]</u>, and <u>.bam</u>, where * is any
              of the following extensions: <u>gz</u>, <u>bz2</u>, and <u>bgzf</u> for transparent (de)compression.

       <b>-m</b>, <b>--acc-tax-map</b> <u>INPUT_FILE</u>
              An     NCBI     or     UniProt     accession-to-taxid     mapping     file.     Download      from
              <a href="ftp://ftp.ncbi.nlm.nih.gov/pub/taxonomy/accession2taxid/">ftp://ftp.ncbi.nlm.nih.gov/pub/taxonomy/accession2taxid/</a>                                        or
              <a href="ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/idmapping/">ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/idmapping/</a>    .    Valid
              filetypes  are: <u>.dat[.*]</u> and <u>.accession2taxid[.*]</u>, where * is any of the following extensions: <u>gz</u>,
              <u>bz2</u>, and <u>bgzf</u> for transparent (de)compression.

       <b>-x</b>, <b>--tax-dump-dir</b> <u>INPUT_DIRECTORY</u>
              A     directory     that     contains     nodes.dmp     and     names.dmp;      unzipped      from
              <a href="ftp://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump.tar.gz">ftp://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump.tar.gz</a>

   <b>Output</b> <b>Options:</b>
       <b>-i</b>, <b>--index</b> <u>OUTPUT_DIRECTORY</u>
              The  output  directory  for  the  index  files (defaults to "DATABASE.lambda"). Valid filetype is:
              <u>.lambda</u>.

       <b>--db-index-type</b> <u>STRING</u>
              Suffix array or full-text minute space. One of <u>fm</u> and <u>bifm</u>. Default: <u>fm</u>.

       <b>--truncate-ids</b> <u>BOOL</u>
              Truncate IDs at first whitespace. This saves a lot of space  and  is  irrelevant  for  all  LAMBDA
              output  formats other than BLAST Pairwise (.m0). One of <u>1</u>, <u>ON</u>, <u>TRUE</u>, <u>T</u>, <u>YES</u>, <u>0</u>, <u>OFF</u>, <u>FALSE</u>, <u>F</u>, and
              <u>NO</u>. Default: <u>on</u>.

   <b>Alphabets</b> <b>and</b> <b>Translation:</b>
       <b>-a</b>, <b>--input-alphabet</b> <u>STRING</u>
              Alphabet of the database sequences (specify to override auto-detection); if input is Dna, it  will
              be translated. One of <u>auto</u>, <u>dna5</u>, and <u>aminoacid</u>. Default: <u>auto</u>.

       <b>-g</b>, <b>--genetic-code</b> <u>INTEGER</u>
              The       translation       table       to       use      if      input      is      Dna.      See
              https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=c  for  ids  (default  is  generic).
              Default: <u>1</u>.

       <b>-r</b>, <b>--alphabet-reduction</b> <u>STRING</u>
              Alphabet Reduction for seeding phase. One of <u>none</u> and <u>murphy10</u>. Default: <u>murphy10</u>.

   <b>Algorithm:</b>
       <b>--algorithm</b> <u>STRING</u>
              Algorithm  for  SA  construction  (also  used  for  FM;  see  Memory  Requirements below!). One of
              <u>mergesort</u>, <u>quicksortbuckets</u>, <u>quicksort</u>, <u>radixsort</u>, and <u>skew7ext</u>. Default: <u>radixsort</u>.

       <b>-t</b>, <b>--threads</b> <u>INTEGER</u>
              number of threads to run concurrently. Default: autodetected.

       <b>--tmp-dir</b> <u>OUTPUT_DIRECTORY</u>
              temporary directory used by skew, defaults to working directory.

</pre><h4><b>REMARKS</b></h4><pre>
       Please see the wiki (&lt;https://github.com/seqan/lambda/wiki&gt;) for more information  on  which  indexes  to
       chose and which algorithms to pick.

       Note  that the indexes created are binary and not compatible between different CPU endiannesses. Also the
       on-disk format is still subject to change between Lambda versions.

</pre><h4><b>LEGAL</b></h4><pre>
       <b>lambda2</b> <b>mkindexp</b> <b>Copyright:</b> 2013-2019 Hannes Hauswedell, released under  the  GNU  AGPL  v3  (or  later);
       2016-2019 Knut Reinert and Freie Universit√§t Berlin, released under the 3-clause-BSDL
       <b>SeqAn</b> <b>Copyright:</b> 2006-2015 Knut Reinert, FU-Berlin; released under the 3-clause BSDL.
       <b>In</b> <b>your</b> <b>academic</b> <b>works</b> <b>please</b> <b>cite:</b> Hauswedell et al (2014); doi: 10.1093/bioinformatics/btu439
       For full copyright and/or warranty information see <b>--copyright</b>.

lambda2 mkindexp 2.0.1                             Nov 28 2024                               <u><a href="../man1/LAMBDA2_MKINDEXP.1.html">LAMBDA2_MKINDEXP</a></u>(1)
</pre>
 </div>
</div></section>
</div>
</body>
</html>